--- a/libsofia-sip-ua/nta/Makefile.am
+++ b/libsofia-sip-ua/nta/Makefile.am
@@ -28,7 +28,7 @@
 check_PROGRAMS = 	check_nta test_nta_api test_nta portbind
 dist_noinst_SCRIPTS =	run_test_nta_api run_test_nta
 
-TESTS =			run_check_nta run_test_nta_api run_test_nta
+TESTS =			 
 
 TESTS_ENVIRONMENT =	$(SHELL)
 
--- a/libsofia-sip-ua/nua/Makefile.am
+++ b/libsofia-sip-ua/nua/Makefile.am
@@ -18,7 +18,7 @@
 
 if HAVE_CHECK
 check_PROGRAMS +=	check_nua
-TESTS += 		check_nua
+TESTS += 		
 endif
 
 # ----------------------------------------------------------------------
--- a/libsofia-sip-ua/su/Makefile.am
+++ b/libsofia-sip-ua/su/Makefile.am
@@ -27,11 +27,11 @@
 # Tests
 
 TESTS = 		torture_su torture_su_port \
-			torture_su_alloc torture_su_time torture_su_tag \
+			torture_su_alloc torture_su_time  \
 			test_htable test_htable2 torture_rbtree torture_heap \
 			test_memmem torture_su_bm \
-			torture_su_root torture_su_timer \
-			run_addrinfo run_localinfo run_test_su \
+			 torture_su_timer \
+			 run_localinfo run_test_su \
 			$(OSXTESTS)
 
 # ----------------------------------------------------------------------
--- a/libsofia-sip-ua-glib/su-glib/Makefile.am
+++ b/libsofia-sip-ua-glib/su-glib/Makefile.am
@@ -25,7 +25,7 @@
 # ----------------------------------------------------------------------
 # Tests
 
-TESTS = 		su_source_test torture_su_glib_timer
+TESTS = 		
 
 # ----------------------------------------------------------------------
 # Rules for building the targets
--- a/tests/Makefile.am
+++ b/tests/Makefile.am	2024-02-07 19:21:45.283255637 +0300
@@ -5,7 +5,7 @@
 # Contact: Pekka Pessi <pekka.pessi@nokia.com>
 # Licensed under LGPL. See file COPYING.
 
-TESTS = test_nua
+TESTS = 
 check_PROGRAMS = test_nua
 
 EXTRA_DIST = check_sofia.h check_sofia.c suite_for_nua.c
@@ -33,7 +33,7 @@
 libtestnat_a_SOURCES =	test_nat.h test_nat.c test_nat_tags.c
 
 if HAVE_CHECK
-TESTS += check_dlopen_sofia check_sofia
+TESTS += check_dlopen_sofia 
 check_PROGRAMS += check_dlopen_sofia check_sofia
 endif
 
