commit 156383a2ff84c01b347579ec8651a0a21384adf4
Author: Daniel Kolesa <daniel@octaforge.org>
Date:   Sun Apr 3 20:18:51 2022 +0200

    use correct SECCOMP_AUDIT_ARCH

diff --git a/src/privsep-linux.c b/src/privsep-linux.c
index 9534fb0..b866bce 100644
--- a/src/privsep-linux.c
+++ b/src/privsep-linux.c
@@ -213,7 +213,11 @@ ps_root_sendnetlink(struct dhcpcd_ctx *ctx, int protocol, struct msghdr *msg)
 #elif defined(__or1k__)
 #  define SECCOMP_AUDIT_ARCH AUDIT_ARCH_OPENRISC
 #elif defined(__powerpc64__)
-#  define SECCOMP_AUDIT_ARCH AUDIT_ARCH_PPC64
+#  ifdef __LITTLE_ENDIAN__
+#    define SECCOMP_AUDIT_ARCH AUDIT_ARCH_PPC64LE
+#  else
+#    define SECCOMP_AUDIT_ARCH AUDIT_ARCH_PPC64
+#  endif
 #elif defined(__powerpc__)
 #  define SECCOMP_AUDIT_ARCH AUDIT_ARCH_PPC
 #elif defined(__riscv)
