From 76ec6a63705e1c3591b4da94c10047a79f0f49db Mon Sep 17 00:00:00 2001
From: q66 <daniel@octaforge.org>
Date: Sun, 20 Dec 2020 15:16:08 +0100
Subject: [PATCH] fix privsep build on ppc*

---
 src/privsep-linux.c | 1 +
 1 file changed, 1 insertion(+)

diff --git src/privsep-linux.c src/privsep-linux.c
index e588ecd..9d335ca 100644
--- a/src/privsep-linux.c
+++ b/src/privsep-linux.c
@@ -42,6 +42,7 @@
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
+#include <termios.h>
 
 #include "common.h"
 #include "if.h"
-- 
2.29.2

