--- src/gui/app.cpp.orig	2017-09-09 05:03:59.000000000 +0200
+++ src/gui/app.cpp	2017-11-13 07:17:16.769223845 +0100
@@ -153,6 +153,10 @@
 
 #endif
 
+#ifndef ACCESSPERMS
+# define ACCESSPERMS (S_IRWXU|S_IRWXG|S_IRWXO) /* 0777 */
+#endif
+
 /* === U S I N G =========================================================== */
 
 using namespace std;
--- src/gui/autorecover.cpp.orig	2017-09-08 11:51:26.000000000 +0200
+++ src/gui/autorecover.cpp	2017-11-13 07:21:41.550697678 +0100
@@ -75,6 +75,10 @@
 
 #endif
 
+#ifndef ACCESSPERMS
+# define ACCESSPERMS (S_IRWXU|S_IRWXG|S_IRWXO) /* 0777 */
+#endif
+
 /* === U S I N G =========================================================== */
 
 using namespace std;
