From 17e57ed156450fe59001cb0a14f8d87c40d6bf77 Mon Sep 17 00:00:00 2001
From: Daniel Kolesa <daniel@octaforge.org>
Date: Sat, 15 Aug 2020 00:09:15 +0200
Subject: [PATCH] Allow disabling gir

---
 meson_options.txt | 5 +++++
 tepl/meson.build  | 4 ++++
 2 files changed, 9 insertions(+)

diff --git meson_options.txt meson_options.txt
index cad97dd..5e99084 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -3,3 +3,8 @@ option(
   type: 'boolean', value: false,
   description: 'Build API reference (requires gtk-doc)'
 )
+option(
+  'gir',
+  type: 'boolean', value: true,
+  description: 'Build gobject-introspection'
+)
diff --git tepl/meson.build tepl/meson.build
index 226d913..b35dd3a 100644
--- a/tepl/meson.build
+++ b/tepl/meson.build
@@ -164,6 +164,8 @@ PKG_CONFIG.generate(tepl_lib,
   libraries: TEPL_PUBLIC_DEPS,
 )
 
+if get_option('gir')
+
 GNOME.generate_gir(
   tepl_lib,
   export_packages: 'tepl-@0@'.format(TEPL_API_VERSION),
@@ -182,3 +184,5 @@ GNOME.generate_gir(
   # Support for deps being built as subprojects:
   dependencies: TEPL_DEPS,
 )
+
+endif
-- 
2.28.0

