From 460cd59bcc0735ca4e1f4e14b59f4e3459c2044d Mon Sep 17 00:00:00 2001
From: Nathan Owens <ndowens@artixlinux.org>
Date: Mon, 30 Nov 2020 09:44:57 -0600
Subject: [PATCH] fix configure error

---
 configure | 3 ---
 1 file changed, 3 deletions(-)

diff --git configure configure
index cb9a9f6..06e65e0 100755
--- a/configure
+++ b/configure
@@ -203,9 +203,6 @@ for opt;do
 		cf_check_lib=no ;;
 	*=*)
 		eval "$opt" ;;
-	*)
-		echo "unknown option: $opt"
-		exit 1 ;;
 	esac
 done
 
-- 
2.29.2

diff --git configure configure
index 06e65e0..b10aed0 100755
--- a/configure
+++ b/configure
@@ -201,8 +201,6 @@ for opt;do
 		cf_docdir=$tmp1 ;;
 	--no-check-lib)
 		cf_check_lib=no ;;
-	*=*)
-		eval "$opt" ;;
 	esac
 done
 
