--- configure.ac.orig
+++ configure.ac
@@ -30,7 +30,7 @@
 	[AC_DEFINE([HAVE_PTHREAD_H], [1], [pthread.h])],
 	AC_MSG_ERROR(pthread.h not found))
 
-AC_CHECK_FILE("$freetype_dir/ft2build.h",,AC_MSG_ERROR([(freetype) ft2build.h not found]))
+
 AC_CHECK_HEADER([fontconfig/fontconfig.h],,AC_MSG_ERROR(fontconfig.h not found))
 AC_CHECK_HEADER([zlib.h],,AC_MSG_ERROR(zlib.h not found))
 AC_CHECK_HEADER([png.h],,AC_MSG_ERROR(png.h not found))
