Author: @Johnnynator
Reason: musl defines stderr/out as (stderr)
---
diff -u -r svn.bak/svncleanupdialog.cpp svn/svncleanupdialog.cpp
--- svn.bak/svncleanupdialog.cpp	2020-08-18 21:40:19.812711204 +0200
+++ svn/svncleanupdialog.cpp	2020-08-18 21:43:40.088090871 +0200
@@ -74,7 +74,7 @@
         emit operationCompletedMessage(i18nc("@info:status", "SVN clean up completed successfully."));
     } else {
         emit errorMessage(i18nc("@info:status", "SVN clean up failed for %1", workDir));
-        qDebug() << result.stderr;
+        qDebug() << result.serr;
     }
 
     QDialog::accept();
diff -u -r svn.bak/svncommands.cpp svn/svncommands.cpp
--- svn.bak/svncommands.cpp	2020-08-18 21:40:19.809711258 +0200
+++ svn/svncommands.cpp	2020-08-18 21:43:40.090090838 +0200
@@ -311,8 +311,8 @@
     } else {
         result.success = true;
     }
-    result.stdout = process.readAllStandardOutput();
-    result.stderr = process.readAllStandardError();
+    result.sout = process.readAllStandardOutput();
+    result.serr = process.readAllStandardError();
 
     return result;
 }
@@ -399,7 +399,7 @@
         if (!process.waitForFinished() || process.exitCode() != 0) {
             process.setReadChannel( QProcess::StandardError );
 
-            // If stderr contains 'E195012' that means repo doesn't exist in the revision range.
+            // If serr contains 'E195012' that means repo doesn't exist in the revision range.
             // It's not an error: let's return everything we've got already.
             const QLatin1String errorCode("svn: E195012:"); // Error: 'Unable to find repository location for <path> in revision <revision>'.
             if (QTextStream(&process).readAll().indexOf(errorCode) != -1) {
diff -u -r svn.bak/svncommands.h svn/svncommands.h
--- svn.bak/svncommands.h	2020-08-18 21:40:19.811711222 +0200
+++ svn/svncommands.h	2020-08-18 21:43:40.089090854 +0200
@@ -35,8 +35,8 @@
  */
 struct CommandResult {
     bool success;       ///< True if return code is '0' (normal execution).
-    QString stdout;     ///< Process stdout.
-    QString stderr;     ///< Process stderr.
+    QString sout;     ///< Process stdout.
+    QString serr;     ///< Process stderr.
 };
 
 /**
diff -u -r svn.bak/svnprogressdialog.cpp svn/svnprogressdialog.cpp
--- svn.bak/svnprogressdialog.cpp	2020-08-18 21:40:19.811711222 +0200
+++ svn/svnprogressdialog.cpp	2020-08-18 21:43:40.088090871 +0200
@@ -115,7 +115,7 @@
         const CommandResult result = SvnCommands::cleanup(m_workingDir);
         if (!result.success) {
             qWarning() << QString("'svn cleanup' failed for %1").arg(m_workingDir);
-            qWarning() << result.stderr;
+            qWarning() << result.serr;
         }
         m_svnTerminated = false;
     }
