The definition of __WORDSIZE is found in <sys/reg.h>

--- usr/iscsi/iscsid.h	2015-11-03 04:58:36.000000000 +0100
+++ usr/iscsi/iscsid.h	2015-11-04 16:23:25.029877334 +0100
@@ -22,6 +22,7 @@
 #include <stdint.h>
 #include <inttypes.h>
 #include <netdb.h>
+#include <sys/reg.h>
 
 #include "transport.h"
 #include "list.h"
--- usr/tgtd.h	2015-11-04 16:25:05.435884573 +0100
+++ usr/tgtd.h	2015-11-04 16:25:52.415887960 +0100
@@ -2,6 +2,7 @@
 #ifndef __TARGET_DAEMON_H
 #define __TARGET_DAEMON_H
 
+#include <sys/reg.h>
 #include "log.h"
 #include "scsi_cmnd.h"
 #include "tgtadm_error.h"
--- usr/util.h	2015-11-04 16:26:22.262890112 +0100
+++ usr/util.h	2015-11-04 16:27:01.364892931 +0100
@@ -16,6 +16,7 @@
 #include <string.h>
 #include <limits.h>
 #include <linux/types.h>
+#include <sys/reg.h>
 
 #include "be_byteshift.h"
 
