diff --git a/src/core/ImageViewBase.h b/src/core/ImageViewBase.h
index e2df3a3..4a18be8 100644
--- a/src/core/ImageViewBase.h
+++ b/src/core/ImageViewBase.h
@@ -16,6 +16,7 @@
 #include <QTransform>
 #include <QWidget>
 #include <Qt>
+#include <QPainterPath>
 #include "ImagePixmapUnion.h"
 #include "ImageViewInfoProvider.h"
 #include "InteractionHandler.h"
diff --git a/src/core/filters/output/OutputGenerator.h b/src/core/filters/output/OutputGenerator.h
index 3e18cad..c6c3ee4 100644
--- a/src/core/filters/output/OutputGenerator.h
+++ b/src/core/filters/output/OutputGenerator.h
@@ -5,6 +5,7 @@
 #define OUTPUT_OUTPUTGENERATOR_H_
 
 #include <QRect>
+#include <QPainterPath>
 #include <memory>
 #include "ImageTransformation.h"
 #include "OutputImage.h"
diff --git a/src/core/filters/page_layout/ImageView.h b/src/core/filters/page_layout/ImageView.h
index c9e6f38..94e4b84 100644
--- a/src/core/filters/page_layout/ImageView.h
+++ b/src/core/filters/page_layout/ImageView.h
@@ -12,6 +12,7 @@
 #include <QRectF>
 #include <QSizeF>
 #include <QTransform>
+#include <QPainterPath>
 #include <unordered_map>
 #include "Alignment.h"
 #include "DragHandler.h"
diff --git a/src/core/filters/select_content/ContentBoxFinder.h b/src/core/filters/select_content/ContentBoxFinder.h
index 1eae24d..d54e7a0 100644
--- a/src/core/filters/select_content/ContentBoxFinder.h
+++ b/src/core/filters/select_content/ContentBoxFinder.h
@@ -5,6 +5,7 @@
 #define SELECT_CONTENT_CONTENTBOXFINDER_H_
 
 #include <BinaryThreshold.h>
+#include <QPainterPath>
 
 class TaskStatus;
 class DebugImages;
diff --git a/src/core/zones/ZoneContextMenuInteraction.h b/src/core/zones/ZoneContextMenuInteraction.h
index d3f712e..51c5922 100644
--- a/src/core/zones/ZoneContextMenuInteraction.h
+++ b/src/core/zones/ZoneContextMenuInteraction.h
@@ -7,6 +7,7 @@
 #include <QColor>
 #include <QObject>
 #include <QtGlobal>
+#include <QPainterPath>
 #include <boost/function.hpp>
 #include <map>
 #include <memory>
diff --git a/src/core/zones/ZoneDefaultInteraction.h b/src/core/zones/ZoneDefaultInteraction.h
index de87193..6e3504c 100644
--- a/src/core/zones/ZoneDefaultInteraction.h
+++ b/src/core/zones/ZoneDefaultInteraction.h
@@ -6,6 +6,7 @@
 
 #include <QCoreApplication>
 #include <QPointF>
+#include <QPainterPath>
 #include "BasicSplineVisualizer.h"
 #include "DragHandler.h"
 #include "DragWatcher.h"
