--- src/runtime/x86-linux-os.h.orig
+++ src/runtime/x86-linux-os.h
@@ -1,7 +1,7 @@
 #ifndef _X86_LINUX_OS_H
 #define _X86_LINUX_OS_H
 
-typedef struct ucontext os_context_t;
+typedef ucontext_t os_context_t;
 typedef greg_t os_context_register_t;
 
 static inline os_context_t *arch_os_get_context(void **void_context)
--- src/runtime/x86-64-linux-os.h.orig
+++ src/runtime/x86-64-linux-os.h
@@ -1,7 +1,7 @@
 #ifndef _X86_64_LINUX_OS_H
 #define _X86_64_LINUX_OS_H
 
-typedef struct ucontext os_context_t;
+typedef ucontext_t os_context_t;
 typedef long os_context_register_t;
 
 static inline os_context_t *arch_os_get_context(void **void_context)
