--- src/hardware/beaglelogic/beaglelogic.h
+++ src/hardware/beaglelogic/beaglelogic.h
@@ -30,8 +30,7 @@
 
 /* BeagleLogic device node name */
 #define BEAGLELOGIC_DEV_NODE        "/dev/beaglelogic"
-#define BEAGLELOGIC_SYSFS_ATTR(a)   "/sys/devices/virtual/misc/beaglelogic/"\
-					__STRING(a)
+#define BEAGLELOGIC_SYSFS_ATTR(a)   "/sys/devices/virtual/misc/beaglelogic/" #a
 
 /* Reproduced verbatim from beaglelogic.h in the kernel tree until the kernel
  * module hits the mainline. Contains the ABI, so DO NOT TOUCH this section */
