--- libsrc/odbcsdk/include/sqltypes.h	2013-12-09 13:28:32.000000000 +0100
+++ libsrc/odbcsdk/include/sqltypes.h	2020-01-27 00:16:39.381042209 +0100
@@ -393,6 +393,7 @@
       defined(_WCHAR_T)			|| \
       defined(_WCHAR_T_DEFINED)		|| \
       defined(_WCHAR_T_DECLARED)        || \
+      defined(__DEFINED_wchar_t)        || \
       defined(_BSD_WCHAR_T_DEFINED_)
 typedef wchar_t SQLWCHAR;
 #  else
