Index: wxWidgets-3.2.1/include/wx/defs.h
===================================================================
--- wxWidgets-3.2.1.orig/include/wx/defs.h
+++ wxWidgets-3.2.1/include/wx/defs.h
@@ -605,7 +605,7 @@ typedef short int WXTYPE;
     #if __has_cpp_attribute(deprecated)
         /* gcc 5 claims to support this attribute, but actually doesn't */
         #if !defined(__GNUC__) || wxCHECK_GCC_VERSION(6, 0)
-            #define wxHAS_DEPRECATED_ATTR
+            /* #define wxHAS_DEPRECATED_ATTR */
         #endif
     #endif
 #endif
