--- src/stc/scintilla/src/Editor.cxx	2014-10-06 23:33:44.000000000 +0200
+++ src/stc/scintilla/src/Editor.cxx	2016-09-18 15:34:29.359258750 +0200
@@ -5841,9 +5841,9 @@
 }
 
 static bool Close(Point pt1, Point pt2) {
-	if (abs(pt1.x - pt2.x) > 3)
+	if (abs(static_cast<int>(pt1.x - pt2.x)) > 3)
 		return false;
-	if (abs(pt1.y - pt2.y) > 3)
+	if (abs(static_cast<int>(pt1.y - pt2.y)) > 3)
 		return false;
 	return true;
 }
