Index: wxWidgets-3.2.1/src/unix/uilocale.cpp
===================================================================
--- wxWidgets-3.2.1.orig/src/unix/uilocale.cpp
+++ wxWidgets-3.2.1/src/unix/uilocale.cpp
@@ -619,7 +619,7 @@ wxString
 wxUILocaleImplUnix::GetLocalizedName(wxLocaleName name, wxLocaleForm form) const
 {
     wxString str;
-#if defined(HAVE_LANGINFO_H) && defined(__LINUX__)
+#if defined(HAVE_LANGINFO_H) && defined(__GLIBC__)
     switch (name)
     {
         case wxLOCALE_NAME_LOCALE:
