Index: wxWidgets-3.2.1/src/common/appbase.cpp
===================================================================
--- wxWidgets-3.2.1.orig/src/common/appbase.cpp
+++ wxWidgets-3.2.1/src/common/appbase.cpp
@@ -847,11 +847,8 @@ bool wxAppConsoleBase::CheckBuildOptions
         wxString prog = wxString::FromAscii(optionsSignature);
         wxString progName = wxString::FromAscii(componentName);
 
-        wxLogFatalError(wxT("Mismatch between the program and library build versions detected.\nThe library used %s,\nand %s used %s."),
+        wxLogWarning(wxT("Mismatch between the program and library build versions detected.\nThe library used %s,\nand %s used %s."),
                         lib, progName, prog);
-
-        // normally wxLogFatalError doesn't return
-        return false;
     }
 
     return true;
