From: Stefan Sauer <ensonic@users.sf.net>
Date: Mon, 23 Dec 2019 12:23:41 +0100
Subject: mkdb: ensure deterministic index files.

Origin: upstream, commit: 51762cfe2b47c886558ee5b311dca24bd5ddcd83
---
 gtkdoc/mkdb.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gtkdoc/mkdb.py b/gtkdoc/mkdb.py
index 087ede2..1e033d4 100644
--- a/gtkdoc/mkdb.py
+++ b/gtkdoc/mkdb.py
@@ -969,7 +969,7 @@ def OutputIndex(basename, apiindex):
 
 def OutputSinceIndexes():
     """Generate the 'since' api index files."""
-    for version in set(Since.values()):
+    for version in sorted(set(Since.values())):
         logging.info("Since : [%s]", version)
         index = {x: IndexEntriesSince[x] for x in IndexEntriesSince.keys() if Since[x] == version}
         OutputIndex("api-index-" + version, index)
