--- a/Kvantum/style/CMakeLists.txt	2023-12-09 19:15:19.658089615 -0500
+++ b/Kvantum/style/CMakeLists.txt	2023-12-09 19:15:31.014342407 -0500
@@ -36,10 +36,7 @@
 if(QT_VERSION_MAJOR EQUAL 6)
   get_target_property(REAL_QMAKE_EXECUTABLE Qt6::qmake
                       IMPORTED_LOCATION)
-  execute_process(COMMAND "${REAL_QMAKE_EXECUTABLE}" -query QT_INSTALL_PLUGINS
-                  OUTPUT_VARIABLE _Qt6_PLUGIN_INSTALL_DIR
-                  ERROR_QUIET OUTPUT_STRIP_TRAILING_WHITESPACE)
-  set(KVANTUM_STYLE_DIR "${_Qt6_PLUGIN_INSTALL_DIR}/styles/")
+  set(KVANTUM_STYLE_DIR "${QT6_INSTALL_PLUGINS}/styles/")
 
   include_directories(${CMAKE_SOURCE_DIR} ${CMAKE_CURRENT_BINARY_DIR}
                       ${Qt6Core_INCLUDE_DIRS} ${Qt6Widgets_INCLUDE_DIRS}
