diff -ruN Marker/meson.build Marker1/meson.build
--- Marker/meson.build	2019-11-06 14:44:46.000000000 +0200
+++ Marker1/meson.build	2019-12-06 22:39:56.434367811 +0200
@@ -5,13 +5,14 @@

 PREFIX = get_option('prefix')
 DATA_DIR = join_paths(PREFIX, 'share')
+LIBS_DIR = join_paths(PREFIX, 'lib')
 APP_DIR = join_paths(DATA_DIR, 'com.github.fabiocolacio.marker')
 ICONS_DIR = join_paths(APP_DIR, 'icons')
 STYLES_DIR = join_paths(APP_DIR, 'styles')
 COMMON_DIR = join_paths(APP_DIR, 'common')
 SCRIPTS_DIR = join_paths(APP_DIR, 'scripts')
 HIGHLIGHT_STYLES_DIR = join_paths(join_paths(SCRIPTS_DIR, 'highlight'),'styles')
-WEB_EXTENSIONS_DIRECTORY = join_paths(APP_DIR, 'extensions')
+WEB_EXTENSIONS_DIRECTORY = join_paths(LIBS_DIR, 'Marker.extensions')
 APPDATA_DIR = join_paths(DATA_DIR, 'metainfo')
 LOCALE_DIR = join_paths(PREFIX, get_option('localedir'))

