--- a/src/marker-markdown.c
+++ b/src/marker-markdown.c
@@ -87,8 +87,7 @@ char* html_header(MarkerMathJSMode    ma
       } else
       {
         mathjs_css = g_strdup(" ");
-        mathjs_script = g_strdup_printf("<script src=\"file://%smathjax/MathJax.js?config=TeX-AMS_HTML\"></script>",
-                                        SCRIPTS_DIR);
+        mathjs_script = g_strdup("<script src=\"file:///usr/share/mathjax/MathJax.js?config=TeX-AMS_HTML\"></script>");
         mathjs_auto = g_strdup(" ");
       }
       break;
