--- a/configure.ac	2016-12-31 00:25:38.000000000 +0100
+++ b/configure.ac	2020-09-20 00:16:14.950903683 +0200
@@ -84,10 +84,8 @@
 AC_SUBST(FRIBIDI_CFLAGS)
 AC_SUBST(FRIBIDI_LIBS)
 
-AC_CHECK_PROGS(FREETYPECONFIG, [freetype-config])
-if test -n "$FREETYPECONFIG"; then
-    FREETYPE_CPPFLAGS="`$FREETYPECONFIG --cflags`"
-    FREETYPE_LIBS="`$FREETYPECONFIG --libs $config_static`"
+PKG_CHECK_MODULES(FREETYPE, [freetype2],[
+    FREETYPE_CPPFLAGS="$FREETYPE_CFLAGS"
     AC_DEFINE(HAVE_FREETYPE, 1, [Whether FreeType is available])
 
     ac_save_CPPFLAGS="$CPPFLAGS"
@@ -96,9 +94,9 @@
     CPPFLAGS="$ac_save_CPPFLAGS"
     AC_SUBST(FREETYPE_CPPFLAGS)
     AC_SUBST(FREETYPE_LIBS)
-else
+        ],
     AC_MSG_ERROR([freetype not found])
-fi
+)
 
 
 AC_ARG_ENABLE([default-video-format],
