From 6a100b4a4a709f177f383988079a2a3c3ada08b8 Mon Sep 17 00:00:00 2001
From: Andrea Brancaleoni <miwaxe@gmail.com>
Date: Thu, 25 Jun 2015 01:59:58 +0200
Subject: [PATCH] default to runit-init

---
 init | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/init b/init
index 4891ed8..7534d23 100644
--- a/init
+++ b/init
@@ -2,7 +2,7 @@
 
 udevd_running=0
 mount_handler=default_mount_handler
-init=/sbin/init
+init=/usr/bin/runit-init
 rd_logmask=0
 
 . /init_functions
-- 
2.4.4

