--- a/Makefile.PL
+++ b/Makefile.PL
@@ -15,14 +15,8 @@ include 'ExtUtils::AutoInstall';
 
 perl_version '5.006';
 build_requires 'Test::More' => 0.47;
-eval { assertlibs lib => 'gmp', header => 'gmp.h'; };
-unless ($@) {
-  requires 'Math::BigInt::GMP' => 1.24;
-}
-else {
-  requires 'Math::BigInt::Pari' => 1.13;
-}
 
+requires('Math::BigInt::GMP' => '1.24');
 requires('Math::BigInt' => '1.60');
 
 githubmeta;
