--- m4/ssl.m4.orig	2014-09-09 21:46:51.073104613 +0200
+++ m4/ssl.m4	2014-09-09 21:47:09.448104838 +0200
@@ -16,7 +16,7 @@
 		AC_TRY_LINK(
 			[#include <openssl/ssl.h>
 			 #include <openssl/rand.h>],
-			[static SSL_CTX *ctx; SSL_new(ctx); RAND_status(); RAND_egd("file")],
+			[static SSL_CTX *ctx; SSL_new(ctx); RAND_status();],
 			[found_loc="$loc"])
 		LIBS="$old_LIBS"
 		LDFLAGS="$old_LDFLAGS"
--- src/lftp_ssl.cc.orig	2014-09-09 21:52:42.747108921 +0200
+++ src/lftp_ssl.cc	2014-09-09 21:54:39.347110350 +0200
@@ -762,8 +762,6 @@
 #endif
 
    RAND_file_name(file,sizeof(file));
-   if(RAND_egd(file)>0)
-      return;
 
    if(RAND_load_file(file,-1) && RAND_status()!=0)
       atexit(lftp_ssl_write_rnd);
