--- ./setup.py.orig	2025-11-17 14:12:01.808011870 -0500
+++ ./setup.py	2025-11-17 14:14:43.996459608 -0500
@@ -1,22 +1,9 @@
-import ast
 from io import open
 import os
 import sys
 
 from setuptools import Extension, setup
 
-
-def version():
-    filename = "src/pillow_avif/__init__.py"
-    with open(filename) as f:
-        tree = ast.parse(f.read(), filename)
-    for node in tree.body:
-        if isinstance(node, ast.Assign) and len(node.targets) == 1:
-            (target,) = node.targets
-            if isinstance(target, ast.Name) and target.id == "__version__":
-                return node.value.s
-
-
 def readme():
     try:
         with open("README.md") as f:
@@ -54,7 +41,7 @@
     description="A pillow plugin that adds avif support via libavif",
     long_description=readme(),
     long_description_content_type="text/markdown",
-    version=version(),
+    version='@@VERSION@@',
     ext_modules=[
         Extension(
             "pillow_avif._avif",
