--- setup.py.bck	2017-01-20 17:36:52.666818102 -0500
+++ setup.py	2017-01-20 17:37:23.204501483 -0500
@@ -12,9 +12,9 @@
     requires = load_requires_from_file('requirements.txt')
     if sys.version_info < (3, 4):
         # To enable Enum in Python < 3.4
-        requires.append('enum34 == 1.0.4')
+        requires.append('enum34 >= 1.0.4')
         # To enable pathlib in Python < 3.4
-        requires.append('pathlib == 1.0.1')
+        requires.append('pathlib >= 1.0.1')
     return requires
 
 
