From 52f81526dbbea59e5c8f6aee64db1a9c29509d58 Mon Sep 17 00:00:00 2001
From: Piotr Wójcik <chocimier@tlen.pl>
Date: Tue, 30 Mar 2021 22:58:45 +0200
Subject: [PATCH] default-PATH


diff --git a/include/pathnames.h b/include/pathnames.h
index 3845d4c..984b4d8 100644
--- a/include/pathnames.h
+++ b/include/pathnames.h
@@ -19,20 +19,12 @@
 /* DEFPATHs from <paths.h> don't include /usr/local */
 #undef _PATH_DEFPATH
 
-#ifdef USE_USRDIR_PATHS_ONLY
-# define _PATH_DEFPATH	        "/usr/local/bin:/usr/bin"
-#else
-# define _PATH_DEFPATH	        "/usr/local/bin:/bin:/usr/bin"
-#endif
+# define _PATH_DEFPATH	        "/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/sbin:/sbin:/bin"
 
 #undef _PATH_DEFPATH_ROOT
 
-#ifdef USE_USRDIR_PATHS_ONLY
-# define _PATH_DEFPATH_ROOT	"/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin"
-#else
-# define _PATH_DEFPATH_ROOT	"/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin"
-#endif
-
+# define _PATH_DEFPATH_ROOT	"/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/sbin:/sbin:/bin"
+	
 #define	_PATH_HUSHLOGIN		".hushlogin"
 #define	_PATH_HUSHLOGINS	"/etc/hushlogins"
 
-- 
2.31.0

