Fixes rendering issue with chsh

--- login-utils/chsh.c.orig	2015-08-01 23:56:41.281133707 +0200
+++ login-utils/chsh.c	2015-08-01 23:56:52.116132896 +0200
@@ -179,6 +179,7 @@ static char *ask_new_shell(char *questio
 	if (!oldshell)
 		oldshell = "";
 	printf("%s [%s]: ", question, oldshell);
+	fflush(stdout);
 	sz = getline(&ans, &dummy, stdin);
 	if (sz == -1)
 		return NULL;

--- libmount/src/context_mount.c.orig	2016-01-12 17:33:25.902659537 +0100
+++ libmount/src/context_mount.c	2016-01-12 17:34:31.198503965 +0100
@@ -605,27 +605,27 @@ static int exec_helper(struct libmnt_con
 		type = mnt_fs_get_fstype(cxt->fs);
 
 		args[i++] = cxt->helper;		/* 1 */
-		args[i++] = mnt_fs_get_srcpath(cxt->fs);/* 2 */
-		args[i++] = mnt_fs_get_target(cxt->fs);	/* 3 */
 
 		if (mnt_context_is_sloppy(cxt))
-			args[i++] = "-s";		/* 4 */
+			args[i++] = "-s";		/* 2 */
 		if (mnt_context_is_fake(cxt))
-			args[i++] = "-f";		/* 5 */
+			args[i++] = "-f";		/* 3 */
 		if (mnt_context_is_nomtab(cxt))
-			args[i++] = "-n";		/* 6 */
+			args[i++] = "-n";		/* 4 */
 		if (mnt_context_is_verbose(cxt))
-			args[i++] = "-v";		/* 7 */
+			args[i++] = "-v";		/* 5 */
 		if (o) {
-			args[i++] = "-o";		/* 8 */
-			args[i++] = o;			/* 9 */
+			args[i++] = "-o";		/* 6 */
+			args[i++] = o;			/* 7 */
 		}
 		if (type
 		    && strchr(type, '.')
 		    && !endswith(cxt->helper, type)) {
-			args[i++] = "-t";		/* 10 */
-			args[i++] = type;		/* 11 */
+			args[i++] = "-t";		/* 8 */
+			args[i++] = type;		/* 9 */
 		}
+		args[i++] = mnt_fs_get_srcpath(cxt->fs);/* 10 */
+		args[i++] = mnt_fs_get_target(cxt->fs);	/* 11 */
 		args[i] = NULL;				/* 12 */
 		for (i = 0; args[i]; i++)
 			DBG(CXT, ul_debugobj(cxt, "argv[%d] = \"%s\"",
