Define __SWORD_TYPE for musl.

--- a/sys-utils/switch_root.c.orig	2014-09-20 21:33:27.551345526 +0200
+++ b/sys-utils/switch_root.c	2014-09-20 21:34:17.958584374 +0200
@@ -47,6 +47,15 @@
 #define MNT_DETACH       0x00000002	/* Just detach from the tree */
 #endif
 
+#ifndef __SWORD_TYPE
+# if __WORDSIZE == 32	/* System word size */
+#  define __SWORD_TYPE	int
+# else /* __WORDSIZE == 64 */
+#  define __SWORD_TYPE	long int
+# endif
+#endif
+
+
 /* remove all files/directories below dirName -- don't cross mountpoints */
 static int recursiveRemove(int fd)
 {
