--- configure.orig	2012-07-27 11:11:26.767603020 +0200
+++ configure	2012-07-27 11:12:15.692451582 +0200
@@ -3383,23 +3383,7 @@ $as_echo "$as_me:   --libdir defaults to
   esac ;;
 esac
 
-case $prefix:$localstatedir in
-  NONE:'${prefix}/var')
-    localstatedir=/run
-    { $as_echo "$as_me:${as_lineno-$LINENO}:   --localstatedir defaults to /run" >&5
-$as_echo "$as_me:   --localstatedir defaults to /run" >&6;}
-    ;;
-  *:'${prefix}/var')
-    localstatedir=${prefix}/run
-    ;;
-  NONE:*)
-    ;;
-  *:*)
-    localstatedir=${prefix}/${localstatedir}
-    ;;
-esac
-
-
+localstatedir=/run
 libdirname=`basename "$libdir"`
 
 
