TIMESPEC_TO_TIMEVAL is a macro defined by glibc.

--- sys-utils/dmesg.c.orig	2014-01-07 21:40:23.490421172 +0100
+++ sys-utils/dmesg.c	2014-01-07 21:41:22.419080539 +0100
@@ -35,6 +35,12 @@
 #include "mangle.h"
 #include "pager.h"
 
+#ifndef TIMESPEC_TO_TIMEVAL
+#define TIMESPEC_TO_TIMEVAL(tv, ts) {		\
+	(tv)->tv_sec = (ts)->tv_sec;		\
+	(tv)->tv_usec = (ts)->tv_nsec / 1000;	\
+}
+#endif
 /* Close the log.  Currently a NOP. */
 #define SYSLOG_ACTION_CLOSE          0
 /* Open the log. Currently a NOP. */
