--- linux-5.7.9/include/uapi/linux/wireless.h.orig
+++ linux-5.7.9/include/uapi/linux/wireless.h
@@ -74,7 +74,7 @@
 #include <linux/socket.h>		/* for "struct sockaddr" et al	*/
 #include <linux/if.h>			/* for IFNAMSIZ and co... */
 
-#include <stddef.h>                     /* for offsetof */
+#include <linux/stddef.h>               /* for offsetof */
 
 /***************************** VERSION *****************************/
 /*
