--- configure.orig
+++ configure
@@ -39,7 +39,7 @@
     echo "The standalone Camlp4 requires OCaml >= 4.02."
     echo "For previous versions of OCaml use the Camlp4 distributed with OCaml."
     exit 2
-elif [ $major -ne 4 -o $minor -ne 8 ]; then
+elif [ $major -ne 4 -o $minor -ne 9 ]; then
     echo "This version of Camlp4 is for OCaml 4.07 but you are using OCaml $ocaml_version."
     if [ -d .git ] ; then
       if [ $PINNED -eq 1 ] ; then
