--- a/ddcci/ddcci.c
+++ b/ddcci/ddcci.c
@@ -1672,8 +1672,14 @@ static int ddcci_detect(struct i2c_clien
 }
 
 /* I2C probe function */
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 3, 0)
+static int ddcci_probe(struct i2c_client *client)
+{
+	const struct i2c_device_id *id = i2c_client_get_device_id(client);
+#else
 static int ddcci_probe(struct i2c_client *client, const struct i2c_device_id *id)
 {
+#endif
 	int i, ret = -ENODEV, tmp;
 	unsigned char main_addr, addr;
 	struct ddcci_bus_drv_data *drv_data;
