From 02fe362807c3b8743ec6696476bfe51ab590ff39 Mon Sep 17 00:00:00 2001
From: Jasper Chan <jasperchan515@gmail.com>
Date: Thu, 6 Sep 2018 21:47:43 -0700
Subject: [PATCH] Checking for -ltinfo is deprecated

---
 Makefile | 1 -
 1 file changed, 1 deletion(-)

diff --git Makefile Makefile
index 8626dd2..5653c67 100644
--- Makefile
+++ Makefile
@@ -55,7 +55,6 @@ CXX_FLAGS += -dynamic -D_OS_OSX
 # Linux
 else
 CXX_FLAGS += -D_LINUX
-LIBS += -ltinfo
 ifndef ENABLE_PULSE
 CHECK_PULSE=$(shell ldconfig -p | grep libpulse-simple)
 ifeq ($(strip $(CHECK_PULSE)),)
-- 
2.18.0

