Description: make overriding CFLAGS possible
Author: Jakub Wilk <jwilk@debian.org>
Forwarded: no
Last-Update: 2012-11-30

--- a/base64/Makefile
+++ b/base64/Makefile
@@ -3,7 +3,7 @@ BINARIES = base64
 # Build flags (uncomment one)
 #############################
 # Release build flags
-CFLAGS += -O3
+CFLAGS ?= -O3
 #############################
 # Debug build flags
 #CFLAGS += -g
--- a/src/Makefile
+++ b/src/Makefile
@@ -3,7 +3,7 @@ LIBRARIES = libb64.a
 # Build flags (uncomment one)
 #############################
 # Release build flags
-CFLAGS += -O3
+CFLAGS ?= -O3
 #############################
 # Debug build flags
 #CFLAGS += -g
