Description: use BUFSIZ as buffer size
Author: Jakub Wilk <jwilk@debian.org>
Bug: http://sourceforge.net/tracker/?func=detail&atid=785907&aid=3591336&group_id=152942
Forwarded: no
Last-Update: 2012-11-30

--- a/include/b64/decode.h
+++ b/include/b64/decode.h
@@ -8,6 +8,7 @@ For details, see http://sourceforge.net/
 #ifndef BASE64_DECODE_H
 #define BASE64_DECODE_H
 
+#include <cstdio>
 #include <iostream>
 
 namespace base64
@@ -22,7 +23,7 @@ namespace base64
 		base64_decodestate _state;
 		int _buffersize;
 
-		decoder(int buffersize_in = BUFFERSIZE)
+		decoder(int buffersize_in = BUFSIZ)
 		: _buffersize(buffersize_in)
 		{}
 
--- a/include/b64/encode.h
+++ b/include/b64/encode.h
@@ -8,6 +8,7 @@ For details, see http://sourceforge.net/
 #ifndef BASE64_ENCODE_H
 #define BASE64_ENCODE_H
 
+#include <cstdio>
 #include <iostream>
 
 namespace base64
@@ -22,7 +23,7 @@ namespace base64
 		base64_encodestate _state;
 		int _buffersize;
 
-		encoder(int buffersize_in = BUFFERSIZE)
+		encoder(int buffersize_in = BUFSIZ)
 		: _buffersize(buffersize_in)
 		{}
 
