diff -u -r lxde-common-0.99.0/startlxde.in lxde-common-0.99.0-dbus/startlxde.in
--- lxde-common-0.99.0/startlxde.in	2014-10-23 16:08:00.000000000 +0200
+++ lxde-common-0.99.0-dbus/startlxde.in	2015-09-25 19:00:01.152474183 +0200
@@ -34,10 +34,14 @@
 export SAL_USE_VCLPLUGIN=gtk
 
 # Launch DBus if needed
-if which dbus-launch >/dev/null && test -z "$DBUS_SESSION_BUS_ADDRESS"; then
-	if test -z "$XDG_RUNTIME_DIR" -o ! -S "$XDG_RUNTIME_DIR/bus" -o ! -O "$XDG_RUNTIME_DIR/bus"; then
-		eval "$(dbus-launch --sh-syntax --exit-with-session)"
-	fi
+if test -z "$DBUS_SESSION_BUS_ADDRESS"; then
+	if which dbus-launch >/dev/null; then
+		eval "$(dbus-launch --sh-syntax --exit-with-session)"
+	fi
+else
+	if which dbus-update-activation-environment >/dev/null; then
+		dbus-update-activation-environment --systemd --all
+	fi
 fi
 
 export XDG_MENU_PREFIX="lxde-"

