diff -Naur lxde-common-0.99.1.orig/desktop.conf lxde-common-0.99.1/desktop.conf
--- lxde-common-0.99.1.orig/desktop.conf	2016-02-26 14:51:28.558478250 +0000
+++ lxde-common-0.99.1/desktop.conf	2016-02-26 14:52:20.038264394 +0000
@@ -2,7 +2,7 @@
 window_manager=openbox-lxde
 
 [GTK]
-sNet/ThemeName=Clearlooks
+sNet/ThemeName=Adwaita
 sNet/IconThemeName=nuoveXT2
 sGtk/FontName=Sans 10
 iGtk/ToolbarStyle=3
