From f4be7f8d8e67583170ff036fe6313e8cf9a5d11d Mon Sep 17 00:00:00 2001
From: tibequadorian <tibequadorian@posteo.de>
Date: Sat, 22 Jan 2022 00:06:10 +0100
Subject: [PATCH] Use ayatana-appindicator

---
 Makefile       | 2 +-
 src/droidcam.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 7be3c15..b7f16c4 100644
--- a/Makefile
+++ b/Makefile
@@ -13,7 +13,7 @@ JPEG_LIB ?= $(JPEG_DIR)/lib`getconf LONG_BIT`
 CC   = gcc
 CFLAGS = -Wall -O2
 GTK   = `pkg-config --libs --cflags gtk+-3.0` `pkg-config --libs x11`
-GTK  += `pkg-config --cflags --libs appindicator3-0.1`
+GTK  += `pkg-config --cflags --libs ayatana-appindicator3-0.1`
 LIBAV = `pkg-config --libs --cflags libswscale libavutil`
 LIBS  =  -lspeex -lasound -lpthread -lm
 JPEG  = -I$(JPEG_INCLUDE) $(JPEG_LIB)/libturbojpeg.a
diff --git a/src/droidcam.c b/src/droidcam.c
index cfb6458..6857978 100644
--- a/src/droidcam.c
+++ b/src/droidcam.c
@@ -7,7 +7,7 @@
  */
 
 #include <gtk/gtk.h>
-#include <libappindicator/app-indicator.h>
+#include <libayatana-appindicator/app-indicator.h>
 #include <X11/Xlib.h>
 #include <stdint.h>
 
-- 
2.34.1

