diff -ru a/Makefile b/Makefile
--- a/Makefile	2024-05-07 11:35:07.033518288 -0700
+++ b/Makefile	2024-05-07 11:36:29.228732440 -0700
@@ -14,7 +14,7 @@
 
 CC           ?= gcc
 CFLAGS       ?=
-APPINDICATOR ?= appindicator3-0.1
+APPINDICATOR ?= ayatana-appindicator3-0.1
 
 GTK   = `pkg-config --libs --cflags gtk+-3.0` `pkg-config --libs x11`
 GTK  += `pkg-config --libs --cflags $(APPINDICATOR)`
diff -ru a/src/droidcam.c b/src/droidcam.c
--- a/src/droidcam.c	2024-05-07 11:51:14.240505923 -0700
+++ b/src/droidcam.c	2024-05-07 11:51:46.902382940 -0700
@@ -7,11 +7,7 @@
  */
 
 #include <gtk/gtk.h>
-#ifdef USE_AYATANA_APPINDICATOR
 #include <libayatana-appindicator/app-indicator.h>
-#else
-#include <libappindicator/app-indicator.h>
-#endif
 
 #include <X11/Xlib.h>
 #include <stdint.h>
