diff --git a/Makefile b/Makefile
index 7be3c15..b7f16c4 100644
--- a/Makefile
+++ b/Makefile
@@ -13,7 +13,7 @@ JPEG_LIB ?= $(JPEG_DIR)/lib`getconf LONG_BIT`
 CC   = gcc
 CFLAGS = -Wall -O2
 GTK   = `pkg-config --libs --cflags gtk+-3.0` `pkg-config --libs x11`
-GTK  += `pkg-config --cflags --libs appindicator3-0.1`
+GTK  += `pkg-config --cflags --libs ayatana-appindicator3-0.1`
 LIBAV = `pkg-config --libs --cflags libswscale libavutil`
 LIBS  =  -lspeex -lasound -lpthread -lm
 JPEG  = -I$(JPEG_INCLUDE) $(JPEG_LIB)/libturbojpeg.a
diff --git a/src/droidcam.c b/src/droidcam.c
index cad1ee4..28f4000 100644
--- a/src/droidcam.c
+++ b/src/droidcam.c
@@ -7,7 +7,7 @@
  */
 
 #include <gtk/gtk.h>
-#include <libappindicator/app-indicator.h>
+#include <libayatana-appindicator/app-indicator.h>
 #include <X11/Xlib.h>
 #include <stdint.h>
 
