From aeaa09619d4cfc57c1b98daea5cf7093340362e8 Mon Sep 17 00:00:00 2001
From: Gregory Hainaut <gregory.hainaut@gmail.com>
Date: Mon, 18 Apr 2016 19:27:06 +0200
Subject: [PATCH] x86emitter: gcc 6 compilation fix

Fix #1297
---
 common/src/x86emitter/cpudetect.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/common/src/x86emitter/cpudetect.cpp b/common/src/x86emitter/cpudetect.cpp
index b318909..38f506d 100644
--- common/src/x86emitter/cpudetect.cpp
+++ common/src/x86emitter/cpudetect.cpp
@@ -16,6 +16,7 @@
 #include "PrecompiledHeader.h"
 #include "cpudetect_internal.h"
 #include "internal.h"
+#include "x86_intrin.h"
 
 using namespace x86Emitter;
 
