Index: pcsx2-1.6.0/cmake/SearchForStuff.cmake
===================================================================
--- pcsx2-1.6.0.orig/cmake/SearchForStuff.cmake
+++ pcsx2-1.6.0/cmake/SearchForStuff.cmake
@@ -35,8 +35,6 @@ else()
     set(wxWidgets_CONFIG_OPTIONS --unicode=yes)
 endif()
 
-list(APPEND wxWidgets_CONFIG_OPTIONS --version=3.0)
-
 if(GTK3_API AND NOT APPLE)
     list(APPEND wxWidgets_CONFIG_OPTIONS --toolkit=gtk3)
 elseif(NOT APPLE)
@@ -227,4 +225,4 @@ if((GCC_VERSION VERSION_EQUAL "9.0" OR G
     https://gitweb.gentoo.org/proj/gcc-patches.git/commit/?id=275ab714637a64672c6630cfd744af2c70957d5a
     Even with that patch, compiling with LTO may still segfault. Use at your own risk!
     This text being in a compile log in an open issue may cause it to be closed.")
-endif()
\ No newline at end of file
+endif()
Index: pcsx2-1.6.0/pcsx2/gui/i18n.cpp
===================================================================
--- pcsx2-1.6.0.orig/pcsx2/gui/i18n.cpp
+++ pcsx2-1.6.0/pcsx2/gui/i18n.cpp
@@ -37,7 +37,9 @@ static wxString i18n_GetBetterLanguageNa
 	{
 		case wxLANGUAGE_CHINESE:				return L"Chinese (Traditional)";
 		case wxLANGUAGE_CHINESE_TRADITIONAL:	return L"Chinese (Traditional)";
+#if !wxCHECK_VERSION(3, 2, 0)
 		case wxLANGUAGE_CHINESE_TAIWAN:			return L"Chinese (Traditional)";
+#endif
 		case wxLANGUAGE_CHINESE_HONGKONG:		return L"Chinese (Traditional, Hong Kong)";
 		case wxLANGUAGE_CHINESE_MACAU:			return L"Chinese (Traditional, Macau)";
 	}
