--- a/meson.build
+++ b/meson.build
@@ -122,8 +122,10 @@ gnome_desktop = dependency('gnome-desktop-3.0', version: '>= 3.0.0')
 gtk = dependency('gtk+-3.0', version: '>= 3.22.27')
 libhandy = dependency('libhandy-1', version: '>= 1.1.90')
 libportal = []
+libportal_gtk3 = []
 if get_option('libportal')
-  libportal = dependency('libportal', version: '>= 0.3')
+  libportal = dependency('libportal', version: '>= 0.5')
+  libportal_gtk3 = dependency('libportal-gtk3', version: '>= 0.5')
 endif
 selinux = []
 if get_option('selinux')
--- a/src/meson.build
+++ b/src/meson.build
@@ -285,6 +285,7 @@ nautilus_deps = [
   libgd_dep,
   libhandy,
   libportal,
+  libportal_gtk3,
   nautilus_extension,
   selinux,
   tracker_sparql,
--- a/src/nautilus-files-view.c
+++ b/src/nautilus-files-view.c
@@ -93,7 +93,7 @@
 
 #ifdef HAVE_LIBPORTAL
 #include <libportal/portal.h>
-#include <libportal/portal-gtk3.h>
+#include <libportal-gtk3/portal-gtk3.h>
 #endif
 
 /* Minimum starting update inverval */
