See: https://github.com/ipython/ipython/pull/14223#issuecomment-1869776898

diff --git a/IPython/terminal/interactiveshell.py b/IPython/terminal/interactiveshell.py
index 532287f5e..d92d6b7b2 100644
--- a/IPython/terminal/interactiveshell.py
+++ b/IPython/terminal/interactiveshell.py
@@ -764,7 +764,7 @@ def get_message():
             "message": get_message,
             "prompt_continuation": (
                 lambda width, lineno, is_soft_wrap: PygmentsTokens(
-                    self.prompts.continuation_prompt_tokens(width, lineno=lineno)
+                    self.prompts.continuation_prompt_tokens(width)
                 )
             ),
             "multiline": True,
