diff --git IPython/core/pylabtools.py IPython/core/pylabtools.py
index c71b0cdb6..bb815956e 100644
--- IPython/core/pylabtools.py
+++ IPython/core/pylabtools.py
@@ -383,6 +383,6 @@ def configure_inline_support(shell, backend):
         stacklevel=2,
     )
 
-    from matplotlib_inline.backend_inline import configure_inline_support_orig
+    from matplotlib_inline.backend_inline import configure_inline_support as configure_inline_support_orig
 
     configure_inline_support_orig(shell, backend)
