--- rustc-1.48.0-src/compiler/rustc_llvm/src/lib.rs
+++ rustc-1.48.0-src/compiler/rustc_llvm/src/lib.rs
@@ -170,3 +170,12 @@
                  LLVMInitializeWebAssemblyTargetMC,
                  LLVMInitializeWebAssemblyAsmPrinter);
 }
+
+#[link(name = "ffi")]
+unsafe extern "C" {}
+#[link(name = "z")]
+unsafe extern "C" {}
+#[link(name = "z")]
+unsafe extern "C" {}
+#[link(name = "ncursesw")]
+unsafe extern "C" {}
