Remove useless redhat specific stuff from it.

--- smartd.service.in.orig	2012-12-10 23:46:12.941631937 +0100
+++ smartd.service.in	2012-12-10 23:46:38.323942716 +0100
@@ -1,12 +1,9 @@
 [Unit]
 Description=Self Monitoring and Reporting Technology (SMART) Daemon
-After=syslog.target
 
 [Service]
-EnvironmentFile=-/usr/local/etc/sysconfig/smartmontools
-ExecStart=/usr/local/sbin/smartd -n $smartd_opts
+ExecStart=/usr/local/sbin/smartd -n
 ExecReload=/bin/kill -HUP $MAINPID
-StandardOutput=syslog
 
 [Install]
 WantedBy=multi-user.target
