diff --git a/modules/ECMQueryQt.cmake b/modules/ECMQueryQt.cmake
index 89e7d4f..798a561 100644
--- a/modules/ECMQueryQt.cmake
+++ b/modules/ECMQueryQt.cmake
@@ -57,7 +57,7 @@ if (QT_MAJOR_VERSION STREQUAL "5")
     endif()
 elseif(QT_MAJOR_VERSION STREQUAL "6")
     # QUIET to accommodate the TRY option
-    find_package(Qt6 COMPONENTS CoreTools QUIET CONFIG)
+    find_package(Qt6 COMPONENTS Core QUIET CONFIG)
     set(_exec_name_text "Qt6 qtpaths")
     if (TARGET Qt6::qtpaths)
         get_target_property(_qtpaths_executable Qt6::qtpaths LOCATION)
