--- a/modules/ECMQueryQt.cmake
+++ b/modules/ECMQueryQt.cmake
@@ -56,7 +56,7 @@ if (QT_MAJOR_VERSION STREQUAL "5")
     endif()
 elseif(QT_MAJOR_VERSION STREQUAL "6")
     # QUIET to accommodate the TRY option
-    find_package(Qt6 COMPONENTS CoreTools QUIET CONFIG)
+    find_package(Qt6 COMPONENTS Core QUIET CONFIG)
     set(_exec_name_text "Qt6 qtpaths")
     if (TARGET Qt6::qtpaths)
         get_target_property(_qtpaths_executable Qt6::qtpaths LOCATION)
--- a/modules/ECMPoQmTools.cmake
+++ b/modules/ECMPoQmTools.cmake
@@ -150,6 +150,9 @@ function(ecm_process_po_files_as_qm lang
     if (QT_MAJOR_VERSION EQUAL "5")
         find_package(Qt5LinguistTools CONFIG REQUIRED)
     else()
+        # Find Linguist in target environment, and this Linguist will
+        # find LinguistTools via Qt6_HOST_PATH
+        find_package(Qt6 OPTIONAL_COMPONENTS Linguist CONFIG REQUIRED)
         find_package(Qt6 COMPONENTS LinguistTools CONFIG REQUIRED)
     endif()
 
