--- a/kde-modules/KDEInstallDirs5.cmake
+++ b/kde-modules/KDEInstallDirs5.cmake
@@ -43,7 +43,7 @@ deprecated variable name in square brack
     ``EXECROOTDIR/lib/<multiarch-tuple`` on Debian) [``LIB_INSTALL_DIR``]
 ``LIBEXECDIR``
     executables for internal use by programs and libraries (``BINDIR`` on
-    Windows, ``LIBDIR/libexec`` otherwise) [``LIBEXEC_INSTALL_DIR``]
+    Windows, ``EXECROOTDIR/libexec`` otherwise) [``LIBEXEC_INSTALL_DIR``]
 ``CMAKEPACKAGEDIR``
     CMake packages, including config files (``LIBDIR/cmake``)
     [``CMAKECONFIG_INSTALL_PREFIX``]
--- a/kde-modules/KDEInstallDirs6.cmake
+++ b/kde-modules/KDEInstallDirs6.cmake
@@ -43,7 +43,7 @@ where ``<dir>`` is one of (default value
     ``EXECROOTDIR/lib/<multiarch-tuple`` on Debian)
 ``LIBEXECDIR``
     executables for internal use by programs and libraries (``BINDIR`` on
-    Windows, ``LIBDIR/libexec`` otherwise)
+    Windows, ``EXECROOTDIR/libexec`` otherwise)
 ``CMAKEPACKAGEDIR``
     CMake packages, including config files (``LIBDIR/cmake``)
 ``QTPLUGINDIR``
--- a/kde-modules/KDEInstallDirsCommon.cmake
+++ b/kde-modules/KDEInstallDirsCommon.cmake
@@ -248,7 +248,7 @@ if(WIN32)
         "executables for internal use by programs and libraries"
         LIBEXEC_INSTALL_DIR)
 else()
-    _define_relative(LIBEXECDIR LIBDIR "libexec"
+    _define_relative(LIBEXECDIR EXECROOTDIR "libexec"
         "executables for internal use by programs and libraries"
         LIBEXEC_INSTALL_DIR)
 endif()
