--- a/scripts/sync-palm-jppy.py
+++ b/scripts/sync-palm-jppy.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # Script for syncing Palm addressbook data with iPod via Jppy
 # (c) 2005 Nick Piper <nick at nickpiper dot co dot uk>
 #
@@ -52,7 +52,7 @@
 (options, args) = parser.parse_args()
 
 if options.verbose:
-    print "Trying to export from Jppy:"
+    print("Trying to export from Jppy:")
 
 dir = os.path.join(options.mountpoint,"Contacts")
 if not os.path.exists(dir):
@@ -74,4 +74,4 @@
         os.unlink(fn)
         
 if options.verbose:
-    print "Saved %d records, removed %d records." % (n,r)
+    print("Saved %d records, removed %d records." % (n,r))
