include depreciated cdefs.h as needed for compilation with alpine

--- a/main/ast_expr2.c
+++ b/main/ast_expr2.c
@@ -95,6 +95,7 @@
 #include "asterisk.h"
 
 #include <sys/types.h>
+#include <sys/cdefs.h>
 #include <stdio.h>
 
 #if defined(STANDALONE) || defined(STANDALONE2)
--- a/utils/db1-ast/include/db.h
+++ b/utils/db1-ast/include/db.h
@@ -37,6 +37,7 @@
 #define	_DB_H 1
 
 #include <sys/types.h>
+#include <sys/cdefs.h>
 
 #include <limits.h>
