--- a/libgnucash/engine/test/CMakeLists.txt
+++ b/libgnucash/engine/test/CMakeLists.txt
@@ -45,9 +45,9 @@ set(test_engine_SOURCES
 
 # This test does not run on Win32
 if (NOT WIN32)
-  set(SOURCES ${test_qof_SOURCES} ${CMAKE_SOURCE_DIR}/common/test-core/unittest-support.c)
-  add_engine_test(test-qof "${SOURCES}")
-  target_compile_definitions(test-qof PRIVATE TESTPROG=test_qof)
+  # set(SOURCES ${test_qof_SOURCES} ${CMAKE_SOURCE_DIR}/common/test-core/unittest-support.c)
+  # add_engine_test(test-qof "${SOURCES}")
+  # target_compile_definitions(test-qof PRIVATE TESTPROG=test_qof)
 
   set(SOURCES ${test_engine_SOURCES} ${CMAKE_SOURCE_DIR}/common/test-core/unittest-support.c)
   add_engine_test(test-engine "${SOURCES}")
@@ -168,8 +168,8 @@ set(test_gnc_numeric_SOURCES
   ${CMAKE_SOURCE_DIR}/libgnucash/core-utils/gnc-locale-utils.cpp
   ${gtest_engine_win32_SOURCES}
   gtest-gnc-numeric.cpp)
-gnc_add_test(test-gnc-numeric "${test_gnc_numeric_SOURCES}"
-  gtest_engine_INCLUDES gtest_qof_LIBS)
+# gnc_add_test(test-gnc-numeric "${test_gnc_numeric_SOURCES}"
+#   gtest_engine_INCLUDES gtest_qof_LIBS)
 
 set(test_gnc_timezone_SOURCES
   ${MODULEPATH}/gnc-timezone.cpp
