diff --git a/src/main.c b/src/main.c
index 01155777c..0fbcd4390 100644
--- ./src/main.c
+++ ./src/main.c
@@ -438,6 +438,18 @@ main (int argc, char **argv)
   GError *error = NULL;
   int ecode;
 
+#ifndef __GLIBC__
+  pthread_attr_t attr;
+  if (pthread_getattr_default_np(&attr) != 0) {
+    perror("pthread_getattr_default_np");
+  } else {
+    pthread_attr_setstacksize(&attr, 8388608);
+    if (pthread_setattr_default_np(&attr) != 0) {
+      perror("pthread_setattr_default_np");
+    }
+  }
+#endif
+
   bindtextdomain (GETTEXT_PACKAGE, LOCALEDIR);
   bind_textdomain_codeset (GETTEXT_PACKAGE, "UTF-8");
   textdomain (GETTEXT_PACKAGE);
