diff --git a/data/meson.build b/data/meson.build
index e37f97bdf..5880b7a46 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -113,7 +113,7 @@ schema = configure_file(
 )
 install_data('00_org.gnome.shell.gschema.override', install_dir: schemadir)
 
-if have_systemd
+systemduserunitdir=get_option('systemduserunitdir')
   unitconf = configuration_data()
   unitconf.set('bindir', bindir)
   configure_file(
@@ -127,7 +127,6 @@ if have_systemd
     'org.gnome.Shell-disable-extensions.service',
     install_dir: systemduserunitdir,
   )
-endif
 
 # for unit tests - gnome.compile_schemas() only looks in srcdir
 compiled_schemas = custom_target('compile-schemas',
diff --git a/meson.options b/meson.options
index 01e0d5803..d6eb68faa 100644
--- a/meson.options
+++ b/meson.options
@@ -51,3 +51,9 @@ option('systemd',
   value: true,
   description: 'Enable systemd integration'
 )
+
+option('systemduserunitdir',
+  type: 'string',
+  value: '/usr/lib/systemd/user',
+  description: 'Directory for systemd user service files, or \'no\' to disable.'
+)
