--- a/ext/glslang/CMakeLists.txt	2021-02-21 10:08:39.000000000 +0100
+++ -	2022-11-01 15:26:49.372613144 +0100
@@ -327,7 +327,7 @@
     add_subdirectory(gtests)
 endif()
 
-if(ENABLE_CTEST AND BUILD_TESTING)
+if(ENABLE_CTEST AND BUILD_TESTING AND ENABLE_GLSLANG_BINARIES)
     # glslang-testsuite runs a bash script on Windows.
     # Make sure to use '-o igncr' flag to ignore carriage returns (\r).
     set(IGNORE_CR_FLAG "")
