--- ppsspp-1.20.4/Core/HW/MediaEngine.cpp.orig	2026-08-09 18:24:38.786651050 +0200
+++ ppsspp-1.20.4/Core/HW/MediaEngine.cpp	2026-08-09 18:25:00.738827543 +0200
@@ -429,11 +429,12 @@
 #if LIBAVFORMAT_VERSION_INT >= AV_VERSION_INT(57, 33, 100)
 			stream->codecpar->codec_type = AVMEDIA_TYPE_VIDEO;
 			stream->codecpar->codec_id = AV_CODEC_ID_H264;
+#else
+			stream->request_probe = 0;
 #endif
 #if LIBAVFORMAT_VERSION_MAJOR >= 59
 			avpriv_stream_set_need_parsing(stream, AVSTREAM_PARSE_FULL);
 #else
-			stream->request_probe = 0;
 			stream->need_parsing = AVSTREAM_PARSE_FULL;
 #endif
 			// We could set the width here, but we don't need to.
