--- ppsspp-1.19.3/CMakeLists.txt	2025-07-15 12:08:35.000000000 +0200
+++ -	2025-12-28 16:46:59.277413410 +0100
@@ -2855,7 +2855,7 @@
 	add_test(math_util PPSSPPUnitTest MathUtil)
 	add_test(parsers PPSSPPUnitTest Parsers)
 	add_test(jit PPSSPPUnitTest Jit)
-	add_test(matrix_transpose PPSSPPUnitTest MatrixTranspose)
+	add_test(matrix_transpose PPSSPPUnitTest VFPUMatrixTranspose)
 	add_test(parse_lbn PPSSPPUnitTest ParseLBN)
 	add_test(quick_texhash PPSSPPUnitTest QuickTexHash)
 	add_test(clz PPSSPPUnitTest CLZ)
