--- a/unittest/TestVertexJit.cpp	2021-03-09 23:26:11.430146058 +0100
+++ b/unittest/TestVertexJit.cpp	2021-03-09 23:26:11.430146058 +0100
@@ -15,6 +15,7 @@
 // Official git repository and contact information can be found at
 // https://github.com/hrydgard/ppsspp and http://www.ppsspp.org/.
 
+#include <math.h>
 #include "Common/Common.h"
 #include "Common/TimeUtil.h"
 #include "Core/Config.h"
--- a/CMakeLists.txt	2021-03-02 21:50:51.000000000 +0100
+++ b/CMakeLists.txt	2021-03-02 21:50:51.000000000 +0100
@@ -2100,6 +2100,7 @@
 
 if(UNITTEST)
 	add_executable(unitTest
+		unittest/TestShaderGenerators.cpp
 		unittest/UnitTest.cpp
 		unittest/TestArmEmitter.cpp
 		unittest/TestArm64Emitter.cpp
