from https://github.com/jjk-jacky/pam_rundir/pull/5

diff --git a/pam_rundir.c b/pam_rundir.c
index 2d2c201..67c7ca6 100644
--- a/pam_rundir.c
+++ b/pam_rundir.c
@@ -197,7 +197,7 @@ write_counter (int fd, int count)
             r = write (fd, buf + w, l - w);
             if (r < 0)
             {
-                if (errno = EINTR)
+                if (errno == EINTR)
                     continue;
                 if (w > 0)
                     emergency_invalidate_counter (fd);
