--- a/tsc/CMakeLists.txt	2020-05-17 12:21:43.000000000 +0200
+++ b/tsc/CMakeLists.txt	2020-09-25 15:46:16.492926109 +0200
@@ -297,9 +297,11 @@
 install(DIRECTORY "${TSC_SOURCE_DIR}/data/worlds/"
   DESTINATION ${CMAKE_INSTALL_DATADIR}/tsc/worlds
   COMPONENT worlds)
-install(DIRECTORY "${TSC_BINARY_DIR}/scriptdocs"
-  DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/doc/tsc
-  COMPONENT base)
+if (ENABLE_SCRIPT_DOCS)
+  install(DIRECTORY "${TSC_BINARY_DIR}/scriptdocs"
+    DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/doc/tsc
+    COMPONENT base)
+endif()
 
 if(NOT WIN32)
   install(FILES "${TSC_SOURCE_DIR}/docs/man/tsc.6"
