--- a/src/kssl/ksslcertificate.cpp	2017-11-05 01:28:12.000000000 -0700
+++ b/src/kssl/ksslcertificate.cpp	2017-11-13 01:45:27.716000000 -0700
@@ -168,7 +168,7 @@
         return rc;
     }
     rc = t;
-    d->kossl->OPENSSL_free(t);
+    d->kossl->OPENSSL_sk_free(t);
 #endif
     return rc;
 }
@@ -356,12 +356,12 @@
                 rc += x[i];
             }
             rc += '\n';
-            d->kossl->OPENSSL_free(x);
+            d->kossl->OPENSSL_sk_free(x);
 
             x = d->kossl->BN_bn2hex(e);
             rc += i18n("Exponent: 0x") + QLatin1String(x) +
                   QLatin1String("\n");
-            d->kossl->OPENSSL_free(x);
+            d->kossl->OPENSSL_sk_free(x);
         }
 #endif
 #ifndef NO_DSA
@@ -383,7 +383,7 @@
                 rc += x[i];
             }
             rc += '\n';
-            d->kossl->OPENSSL_free(x);
+            d->kossl->OPENSSL_sk_free(x);
 
             x = d->kossl->BN_bn2hex(q);
             rc += i18n("160 bit prime factor: ");
@@ -396,7 +396,7 @@
                 rc += x[i];
             }
             rc += '\n';
-            d->kossl->OPENSSL_free(x);
+            d->kossl->OPENSSL_sk_free(x);
 
             x = d->kossl->BN_bn2hex(g);
             rc += QString("g: ");
@@ -409,7 +409,7 @@
                 rc += x[i];
             }
             rc += '\n';
-            d->kossl->OPENSSL_free(x);
+            d->kossl->OPENSSL_sk_free(x);
 
             const BIGNUM *pub_key;
             d->kossl->DSA_get0_key(dsa, &pub_key, nullptr);
@@ -424,7 +424,7 @@
                 rc += x[i];
             }
             rc += '\n';
-            d->kossl->OPENSSL_free(x);
+            d->kossl->OPENSSL_sk_free(x);
         }
 #endif
         d->kossl->EVP_PKEY_free(pkey);
@@ -446,7 +446,7 @@
     }
 
     rc = t;
-    d->kossl->OPENSSL_free(t);
+    d->kossl->OPENSSL_sk_free(t);
 #endif
 
     return rc;
