diff --git a/Makefile b/Makefile
index 36a3778..2529515 100644
--- a/Makefile
+++ b/Makefile
@@ -6,10 +6,8 @@
 # ========================
 # For GMP set **both**  GMP_INC_DIR  **and**  ldflags_for_gmp
 # If you have a system installation of GMP with C++ support use next 2 lines:
-# GMP_INC_DIR=
-# ldflags_for_gmp=-lgmpxx -lgmp
-GMP_INC_DIR=/usr/local/include
-ldflags_for_gmp=-L/usr/local/lib -lgmpxx -lgmp
+GMP_INC_DIR=
+ldflags_for_gmp=-lgmpxx -lgmp
 
 
 
@@ -80,7 +78,7 @@ ifndef CXX
 endif
 
 ifndef BIN_INSTALL_DIR
-  BIN_INSTALL_DIR = "/usr/local/bin/"
+  BIN_INSTALL_DIR = "/usr/bin/"
 endif
 
 cxxflags = $(CXXFLAGS) $(CPPFLAGS) -I $(GMP_INC_DIR) -Wno-uninitialized -Wno-unused-parameter
@@ -252,7 +250,7 @@ endif
 
 -include $(objs:.o=.d)
 
-PREFIX ?= /usr/local
+PREFIX ?= /usr
 MAN1DIR ?= $(PREFIX)/share/man/man1
 
 # Installation.
