--- a/Makefile
+++ b/Makefile
@@ -22,15 +22,15 @@
 install-completions: install-bash-completion install-zsh-completion install-fish-completion
 
 install-bash-completion:
-	install -m644 auto-completion/bash/pdd.bash $(PREFIX)/share/bash-completion/compilations/pdd
+	install -D -m644 auto-completion/bash/pdd.bash $(DESTDIR)$(PREFIX)/share/bash-completion/completions/pdd
 
 install-fish-completion:
-	install -m644 auto-completion/fish/pdd.fish -t $(PREFIX)/share/fish/vendor_completions.d
+	install -D -m644 auto-completion/fish/pdd.fish -t $(DESTDIR)$(PREFIX)/share/fish/vendor_completions.d
 
 install-zsh-completion:
 	cp pdd pdd.py
 	auto-completion/zsh/zsh_completion.py
-	install -m644 _pdd -t $(PREFIX)/share/zsh/site-functions
+	install -D -m644 _pdd -t $(DESTDIR)$(PREFIX)/share/zsh/site-functions
 
 uninstall:
 	rm -f $(BINDIR)/pdd
