--- include/ixion/model_context.hpp	2019-08-09 19:53:33.659413255 +0200
+++ include/ixion/model_context.hpp	2019-08-09 19:54:16.188415450 +0200
@@ -62,7 +62,7 @@
         } value;
 
         /** Initializes the cell to be empty. */
-        input_cell(nullptr_t);
+        input_cell(std::nullptr_t);
         /** Boolean cell value. */
         input_cell(bool b);
         /** The char array must be null-terminated. */
