--- include/ixion/model_context.hpp	2019-03-09 05:05:04.000000000 +0100
+++ include/ixion/model_context.hpp	2020-01-29 21:50:49.831725062 +0100
@@ -61,7 +61,7 @@
         } value;
 
         /** Initializes the cell to be empty. */
-        input_cell(nullptr_t);
+        input_cell(std::nullptr_t);
         /** Boolean cell value. */
         input_cell(bool b);
         /** The char array must be null-terminated. */
