--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -57,7 +57,7 @@ endif( CMAKE_COMMAND AND CMAKE_CTEST_COM
 # Drop in New Tests
 ############################
 marble_add_test( MarbleWidgetSpeedTest)
-add_definitions(-DDGML_PATH="${CMAKE_CURRENT_SOURCE_DIR}/../data/maps/earth")
+add_definitions(-DDGML_PATH="@@WRKSRC@@/data/maps/earth")
 marble_add_test( TestGeoSceneWriter)
 
 marble_add_test( LocaleTest)               # Check MarbleLocale functionality
