diff --git a/compiler/generator/Text.hh b/compiler/generator/Text.hh
index 47ca2a7be..8fad938ea 100644
--- a/compiler/generator/Text.hh
+++ b/compiler/generator/Text.hh
@@ -31,6 +31,7 @@
 #include <string>
 #include <vector>
 #include <cmath>
+#include <cstdint>
 
 std::string subst(const std::string& m, const std::string& a0);
 std::string subst(const std::string& m, const std::vector<std::string>& vargs);
