--- a/compiler/global.cpp-orig	2023-09-09 10:38:02.642030383 +0000
+++ b/compiler/global.cpp	2023-09-09 10:41:55.199049192 +0000
@@ -1711,6 +1711,7 @@ void global::initDirectories(int argc, const char* argv[])
     gImportDirList.push_back(exepath::dirup(gFaustExeDir) + "/share/faust");
     gImportDirList.push_back("/usr/local/share/faust");
     gImportDirList.push_back("/usr/share/faust");
+    gImportDirList.push_back("/usr/lib/faust");
 
     //-------------------------------------------------------------------------------------
     // init gArchitectureDirList : a list of path where to search architectures files
@@ -1697,6 +1698,7 @@ void global::initDirectories(int argc, c
     gArchitectureDirList.push_back(exepath::dirup(gFaustExeDir) + "/include");
     gArchitectureDirList.push_back("/usr/local/share/faust");
     gArchitectureDirList.push_back("/usr/share/faust");
+    gArchitectureDirList.push_back("/usr/lib/faust");
     gArchitectureDirList.push_back("/usr/local/include");
     gArchitectureDirList.push_back("/usr/include");
     
@@ -2088,11 +2090,11 @@ void global::printIncludeDir()
 }
 void global::printArchDir()
 {
-    cout << gFaustRootDir << kPSEP << "share" << kPSEP << "faust" << endl;
+    cout << gFaustRootDir << kPSEP << "lib" << kPSEP << "faust" << endl;
 }
 void global::printDspDir()
 {
-    cout << gFaustRootDir << kPSEP << "share" << kPSEP << "faust" << endl;
+    cout << gFaustRootDir << kPSEP << "lib" << kPSEP << "faust" << endl;
 }
 void global::printPaths()
 {
