--- a/compiler/libcode.cpp-old	2019-12-22 19:50:13.663677186 +0100
+++ b/compiler/libcode.cpp	2019-12-22 19:51:42.011680485 +0100
@@ -733,11 +733,11 @@
 }
 static void printArchDir()
 {
-    cout << gGlobal->gFaustRootDir << kPSEP << "share" << kPSEP << "faust" << endl;
+    cout << gGlobal->gFaustRootDir << kPSEP << "lib" << kPSEP << "faust" << endl;
 }
 static void printDspDir()
 {
-    cout << gGlobal->gFaustRootDir << kPSEP << "share" << kPSEP << "faust" << endl;
+    cout << gGlobal->gFaustRootDir << kPSEP << "lib" << kPSEP << "faust" << endl;
 }
 static void printPaths()
 {
@@ -1101,6 +1101,7 @@
     gGlobal->gImportDirList.push_back(exepath::dirup(gGlobal->gFaustExeDir) + "/share/faust");
     gGlobal->gImportDirList.push_back("/usr/local/share/faust");
     gGlobal->gImportDirList.push_back("/usr/share/faust");
+    gGlobal->gImportDirList.push_back("/usr/lib/faust");
 
     //-------------------------------------------------------------------------------------
     // init gArchitectureDirList : a list of path where to search architectures files
@@ -1119,6 +1120,7 @@
     gGlobal->gArchitectureDirList.push_back(exepath::dirup(gGlobal->gFaustExeDir) + "/include");
     gGlobal->gArchitectureDirList.push_back("/usr/local/share/faust");
     gGlobal->gArchitectureDirList.push_back("/usr/share/faust");
+    gGlobal->gArchitectureDirList.push_back("/usr/lib/faust");
     gGlobal->gArchitectureDirList.push_back("/usr/local/include");
     gGlobal->gArchitectureDirList.push_back("/usr/include");
 
