--- examples/CMakeLists.txt	2020-10-07 20:28:14.000000000 +0200
+++ -	2020-10-10 21:40:10.823966203 +0200
@@ -1,6 +1,6 @@
 SUBDIRS( HelloWorld BasicDemo)
 IF(BUILD_BULLET3)
-	SUBDIRS( ExampleBrowser RobotSimulator SharedMemory ThirdPartyLibs/Gwen ThirdPartyLibs/BussIK ThirdPartyLibs/clsocket OpenGLWindow TwoJoint )
+	SUBDIRS( ExampleBrowser RobotSimulator ThirdPartyLibs/Gwen ThirdPartyLibs/BussIK ThirdPartyLibs/clsocket OpenGLWindow TwoJoint )
 ENDIF()
 
 IF(BUILD_PYBULLET)
