From b85741372fea0af6be79a6f40fda2da9cf985f5f Mon Sep 17 00:00:00 2001
From: Dotz0cat <Dotz0cat@gmail.com>
Date: Sat, 9 Jul 2022 17:19:52 -0500
Subject: [PATCH] void fix musl

---
 src/main.c | 44 ++++----------------------------------------
 1 file changed, 4 insertions(+), 40 deletions(-)

diff --git a/src/main.c b/src/main.c
index 3a13829..b062efa 100644
--- a/src/main.c
+++ b/src/main.c
@@ -146,46 +146,10 @@ static void init_daemon(const char* home_dir) {
 		close(x);
 	}
 
-	int char_count = snprintf(NULL, 0, "%s%s", home_dir, "/.walld/log");
-
-	if (char_count <= 0) {
-		abort();
-	}
-
-	char* log_file = malloc(char_count + 1U);
-
-	if (log_file == NULL) {
-		abort();
-	}
-
-	snprintf(log_file, char_count + 1U, "%s%s", home_dir, "/.walld/log");
-
-	stdin = fopen("/dev/null", "r");
-	stdout = fopen(log_file, "w");
-
-	free(log_file);
-
-	char_count = 0;
-
-	char_count = snprintf(NULL, 0, "%s%s", home_dir, "/.walld/error-log");
-
-	if (char_count <= 0) {
-		abort();
-	}
-
-	char* err_log_file = malloc(char_count + 1U);
-
-	if (err_log_file == NULL) {
-		abort();
-	}
-
-	snprintf(err_log_file, char_count + 1U, "%s%s", home_dir, "/.walld/error-log");
-
-
-	stderr = fopen(err_log_file, "w");
-
-	free(err_log_file);
-
+	fclose(stdin);
+	fclose(stdout);
+	fclose(stderr);
+	
 	openlog("walld", LOG_PID, LOG_DAEMON);
 }
 
-- 
2.37.0

