--- fdisk.c
+++ fdisk.c
@@ -59,6 +59,10 @@
  */
 
 
+#ifndef _LARGEFILE64_SOURCE
+#define _LARGEFILE64_SOURCE
+#endif
+#include <sys/types.h>
 #include <unistd.h>
 #include <stdio.h>
 #include <stdlib.h>
--- fdisklabel.c
+++ fdisklabel.c
@@ -35,6 +35,9 @@
    SUCH DAMAGE.
 */
 
+#ifndef _LARGEFILE64_SOURCE
+#define _LARGEFILE64_SOURCE
+#endif
 #include <sys/types.h>
 #include <unistd.h>
 #include <stdio.h>
--- io.c
+++ io.c
@@ -25,6 +25,7 @@
  * WITH THE USE OR PERFORMANCE OF THIS SOFTWARE. 
  */
 
+#define _LARGEFILE64_SOURCE
 #include <stdio.h>
 #include <stdlib.h>
 #include <fcntl.h>
@@ -37,6 +38,7 @@
 #endif
 #endif
 #include <linux/unistd.h>
+#include <sys/types.h>
 #include <unistd.h>
 #include <string.h>
 #include <stdarg.h>
