--- rizin-0.3.1/subprojects/tree-sitter-c/meson.build	2021-11-22 22:04:48.051969201 +0100
+++ -	2021-11-22 22:05:47.546282763 +0100
@@ -4,8 +4,11 @@
   'src/parser.c'
 ]
 
-tree_sitter_proj = subproject('tree-sitter', default_options: ['default_library=static'])
-tree_sitter_dep = tree_sitter_proj.get_variable('tree_sitter_dep')
+tree_sitter_dep = dependency('tree-sitter', required: true, static: false,   fallback: [])
+if not tree_sitter_dep.found()
+    tree_sitter_proj = subproject('tree-sitter', default_options: ['default_library=static'])
+    tree_sitter_dep = tree_sitter_proj.get_variable('tree_sitter_dep')
+endif
 
 libtsc = static_library('tree-sitter-c', ts_c_files,
   include_directories: ['src'],
