--- a/src/plugins/youtube/CMakeLists.txt
+++ b/src/plugins/youtube/CMakeLists.txt
@@ -1,7 +1,7 @@
 add_definitions(-DTRANSLATION_DOMAIN=\"purpose6_youtube\")

 if (KAccounts6_FOUND)
-    kaccounts_add_service(${CMAKE_CURRENT_SOURCE_DIR}/google-youtube.service.in)
+    kaccounts_add_service(${CMAKE_CURRENT_SOURCE_DIR}/google-youtube.service.in NO_INTLTOOL)
 endif()
 
 add_share_plugin(youtubeplugin youtubeplugin.cpp youtubejob.cpp youtubejobcomposite.cpp)

--- a/src/plugins/nextcloud/CMakeLists.txt
+++ b/src/plugins/nextcloud/CMakeLists.txt
@@ -1,7 +1,7 @@
 add_definitions(-DTRANSLATION_DOMAIN=\"purpose6_nextcloud\")
 
 if (KAccounts6_FOUND)
-    kaccounts_add_service(${CMAKE_CURRENT_SOURCE_DIR}/nextcloud-upload.service.in)
+    kaccounts_add_service(${CMAKE_CURRENT_SOURCE_DIR}/nextcloud-upload.service.in NO_INTLTOOL)
 endif()
 
 add_share_plugin(nextcloudplugin nextcloudplugin.cpp nextcloudjob.cpp)
