From b430f543c850acfb98751bfcf5343afb397cf049 Mon Sep 17 00:00:00 2001
From: Nathan Owens <ndowens@artixlinux.org>
Date: Thu, 31 Dec 2020 09:18:14 -0600
Subject: [PATCH] fix cross

---
 CMakeLists.txt | 5 -----
 1 file changed, 5 deletions(-)

diff --git CMakeLists.txt CMakeLists.txt
index 5f4cb14..28e622b 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -24,11 +24,6 @@ if( FORCE_CROSSCOMPILE )
 	set( CMAKE_CROSSCOMPILING TRUE )
 endif()
 
-if(CMAKE_CROSSCOMPILING)
-	set(IMPORT_EXECUTABLES "IMPORTFILE-NOTFOUND" CACHE FILEPATH "Export file from native build.")
-	include(${IMPORT_EXECUTABLES})
-endif()
-
 
 # Macro for building libraries without debugging information
 macro( make_release_only )
-- 
2.30.0

