--- src/cpu/sysdeps.h.orig	2016-11-03 22:25:43.000000000 +0100
+++ src/cpu/sysdeps.h	2017-03-20 09:43:25.919910461 +0100
@@ -52,7 +52,7 @@
 #define CPU_64_BIT 1
 #elif defined(__i386__) || defined(_M_IX86)
 #define CPU_i386 1
-#elif defined(__arm__) || defined(_M_ARM)
+#elif defined(__arm__) || defined(_M_ARM) || defined(__aarch64__)
 #define CPU_arm 1
 #elif defined(__powerpc__) || defined(_M_PPC) || defined(__ppc__) || defined(__ppc64__)
 #define CPU_powerpc 1
