--- ./src/utils_tools.c.orig	2014-10-24 12:58:35.151717616 -0200
+++ ./src/utils_tools.c	2014-10-24 13:00:42.716855265 -0200
@@ -105,10 +105,13 @@
 
 	case CRYPT_LOG_NORMAL:
 		fputs(msg, stdout);
+		fflush(stdout);
 		break;
 	case CRYPT_LOG_VERBOSE:
-		if (opt_verbose)
+		if (opt_verbose) {
 			fputs(msg, stdout);
+			fflush(stdout);
+		}
 		break;
 	case CRYPT_LOG_ERROR:
 		fputs(msg, stderr);
