diff --git a/lib/crypto_backend/crypto_openssl.c b/lib/crypto_backend/crypto_openssl.c
index 4929f82..463c3dd 100644
--- ./lib/crypto_backend/crypto_openssl.c
+++ ./lib/crypto_backend/crypto_openssl.c
@@ -52,7 +52,7 @@ struct crypt_hmac {
 /*
  * Compatible wrappers for OpenSSL < 1.1.0
  */
-#if OPENSSL_VERSION_NUMBER < 0x10100000L || defined(LIBRESSL_VERSION_NUMBER)
+#if OPENSSL_VERSION_NUMBER < 0x10100000L
 static void openssl_backend_init(void)
 {
 	OpenSSL_add_all_algorithms();
