From d9e71c87c7af8687750901d71409140f1bd181bd Mon Sep 17 00:00:00 2001
From: nerdopolis <bluescreen_avenger@verizon.net>
Date: Sun, 27 Nov 2022 18:46:05 -0500
Subject: [PATCH] Fix build on Debian Stable

---
 meson.build | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/meson.build b/meson.build
index ace7c933..13548b74 100644
--- a/meson.build
+++ b/meson.build
@@ -47,6 +47,7 @@ fish_comp = dependency('fish', required: false)
 libpam = cc.find_library('pam', required: get_option('pam'))
 crypt = cc.find_library('crypt', required: not libpam.found())
 math = cc.find_library('m')
+rt = cc.find_library('rt')
 
 git = find_program('git', required: false)
 scdoc = find_program('scdoc', required: get_option('man-pages'))
@@ -112,6 +113,7 @@ dependencies = [
 	client_protos,
 	gdk_pixbuf,
 	math,
+	rt,
 	xkbcommon,
 	wayland_client,
 ]
