--- a/src/vtsspp/Makefile	2019-08-19 09:51:34.000000000 -0700
+++ b/src/vtsspp/Makefile	2019-12-28 12:55:53.422853567 -0700
@@ -117,7 +117,7 @@
 _KBUILD_CFLAGS := $(call flags,KBUILD_CFLAGS)
 CHECK_BUILD    := $(CC) $(KBUILD_CPPFLAGS) $(CPPFLAGS) $(LINUXINCLUDE) \
                   $(_KBUILD_CFLAGS) $(CFLAGS_KERNEL) $(EXTRA_CFLAGS) $(CFLAGS) \
-                  -DKBUILD_BASENAME=\"$(DRIVER_NAME)\" -Werror -S -o /dev/null -xc >> $(obj)/.autoconf.log 2>&1
+                  -DKBUILD_BASENAME=\"$(DRIVER_NAME)\" -Werror -S -Wno-pointer-sign -o /dev/null -xc >> $(obj)/.autoconf.log 2>&1
 
 autoconf-srcs := $(sort $(wildcard $(M)/autoconf/*.c))
 $(addprefix $(obj)/,$($(DRIVER_NAME)-objs)): $(obj)/autoconf.h
