--- src/tools/arrow/arrowtool.h	2018-08-17 14:17:40.000000000 +0200
+++ src/tools/arrow/arrowtool.h	2020-09-01 14:52:47.715084317 +0200
@@ -19,6 +19,7 @@
 
 #include "src/tools/abstracttwopointtool.h"
 #include <QPainter>
+#include <QPainterPath>
 
 class ArrowTool : public AbstractTwoPointTool {
     Q_OBJECT
--- src/third-party/Qt-Color-Widgets/src/color_wheel.cpp	2018-08-17 14:17:40.000000000 +0200
+++ src/third-party/Qt-Color-Widgets/src/color_wheel.cpp	2020-09-01 14:54:24.670089318 +0200
@@ -24,6 +24,7 @@
 #include <cmath>
 #include <QMouseEvent>
 #include <QPainter>
+#include <QPainterPath>
 #include <QLineF>
 #include <QDragEnterEvent>
 #include <QMimeData>
