From c240fd5c1eda2cf697336cc234d66f6af38894e8 Mon Sep 17 00:00:00 2001
From: oreo639 <oreo6391@gmail.com>
Date: Sun, 15 Sep 2024 02:16:20 -0700
Subject: [PATCH] GlatexApp: implement new-window action

---
 src/glatex_app.vala | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/glatex_app.vala b/src/glatex_app.vala
index 3d93c8bd..4e913212 100644
--- a/src/glatex_app.vala
+++ b/src/glatex_app.vala
@@ -8,6 +8,7 @@ public class GlatexApp : Gtex.App
 {
     private const GLib.ActionEntry[] _app_actions =
     {
+        { "new-window", new_window_cb },
         { "new-document", new_document_cb },
         { "manage-build-tools", manage_build_tools_cb },
         { "quit", quit_cb }
@@ -52,6 +53,11 @@ public class GlatexApp : Gtex.App
         release ();
     }
 
+    private void new_window_cb ()
+    {
+        MainWindow? window = create_window ();
+    }
+
     private void new_document_cb ()
     {
         MainWindow? window = get_active_main_window ();
-- 
GitLab

