diff --git a/setup.py b/setup.py
index 1c0e620..6367695 100644
--- a/setup.py
+++ b/setup.py
@@ -54,7 +54,7 @@ class Install(_install, object):
             return
 
         real_confdir = os.path.join('/etc')
-        real_statedir = os.path.join('/lib', 'ufw')
+        real_statedir = os.path.join('/usr/lib', 'ufw')
         real_prefix = self.prefix
         if self.home != None:
             real_confdir = self.home + real_confdir
@@ -113,7 +113,7 @@ class Install(_install, object):
         if self.root != None:
             prefix = self.root + real_prefix
 
-        script = os.path.join(prefix, 'sbin', 'ufw')
+        script = os.path.join(prefix, 'bin', 'ufw')
         manpage = os.path.join(prefix, 'share', 'man', 'man8', 'ufw.8')
         manpage_f = os.path.join(prefix, 'share', 'man', 'man8', \
                                  'ufw-framework.8')
