--- src/modules/loaders/loader_xpm.c
+++ src/modules/loaders/loader_xpm.c.new
@@ -246,8 +246,8 @@
                                  return 0;
                               }
                             ptr = im->data;
-                            end = ptr + (sizeof(DATA32) * w * h);
                             pixels = w * h;
+                            end = ptr + pixels;
                          }
                        else
                          {

