Adapt DGifOpenFileHandle() to the 5.x API.

--- src/modules/loaders/loader_gif.c.orig	2012-10-15 16:29:26.667044054 +0200
+++ src/modules/loaders/loader_gif.c	2012-10-15 16:29:43.329649211 +0200
@@ -19,7 +19,7 @@ load(ImlibImage * im, ImlibProgressFunct
    int                 intoffset[] = { 0, 4, 2, 1 };
    int                 intjump[] = { 8, 8, 4, 2 };
    int                 transp;
-   int                 fd;
+   int                 fd, err = 0;
 
    done = 0;
    rows = NULL;
@@ -36,8 +36,8 @@ load(ImlibImage * im, ImlibProgressFunct
 #endif
    if (fd < 0)
       return 0;
-   gif = DGifOpenFileHandle(fd);
-   if (!gif)
+   gif = DGifOpenFileHandle(fd, &err);
+   if (err)
      {
         close(fd);
         return 0;
