--- a/src/CUPSNotifier.vala	2018-11-28 12:34:33.000000000 +0100
+++ b/src/CUPSNotifier.vala	2021-01-29 19:26:11.191992396 +0100
@@ -63,7 +63,7 @@
         Bus.get_proxy.begin<NotifierDBus> (BusType.SYSTEM, "org.cups.cupsd.Notifier", "/org/cups/cupsd/Notifier", GLib.DBusProxyFlags.NONE, null, (obj, res) => {
             try {
                 dbus_notifier = Bus.get_proxy.end (res);
-                ((GLib.DBusProxy) dbus_notifier).g_connection.signal_subscribe (null, "org.cups.cupsd.Notifier", null, "/org/cups/cupsd/Notifier", null, GLib.DBusSignalFlags.NONE, subscription_callback);
+                ((GLib.DBusProxy) dbus_notifier).g_connection.signal_subscribe (null, "org.cups.cupsd.Notifier", null, "/org/cups/cupsd/Notifier", null, GLib.DBusSignalFlags.NONE, (GLib.DBusSignalCallback)subscription_callback);
             } catch (IOError e) {
                 critical (e.message);
             }
