--- a/vendor/CMakeLists.sparse.txt	2022-07-13 18:44:44.000000000 +0200
+++ -	2022-11-20 01:25:48.530247621 +0100
@@ -5,7 +5,7 @@
 	core/libsparse/sparse_crc32.cpp
 	core/libsparse/sparse_err.cpp
 	core/libsparse/sparse_read.cpp)
-
+target_compile_definitions(libsparse PRIVATE SEEK_DATA=3 SEEK_HOLE=4)
 target_include_directories(libsparse PUBLIC
 	core/libsparse/include libbase/include)
 target_link_libraries(libsparse PRIVATE libbase)
