--- android-tools-34.0.1.orig/vendor/extras/libjsonpb/parse/jsonpb.cpp  2023-06-28 23:15:15.414236253 +0200
+++ android-tools-34.0.1/vendor/extras/libjsonpb/parse/jsonpb.cpp       2023-06-28 23:34:21.672962973 +0200
@@ -51,7 +51,7 @@
 #if GOOGLE_PROTOBUF_VERSION < 3016000
     return MakeError<std::string>(status.error_message().as_string());
 #else
-    return MakeError<std::string>(status.message().as_string());
+    return MakeError<std::string>(std::string(status.message()));
 #endif
   }
   return ErrorOr<std::string>(std::move(json));
@@ -68,7 +68,7 @@
 #if GOOGLE_PROTOBUF_VERSION < 3016000
     return MakeError<std::monostate>(status.error_message().as_string());
 #else
-    return MakeError<std::monostate>(status.message().as_string());
+    return MakeError<std::monostate>(std::string(status.message()));
 #endif
   }
   if (!message->ParseFromString(binary)) {
--- a/vendor/CMakeLists.partition.txt	2023-03-03 17:18:50.000000000 +0100
+++ -	2023-07-10 23:06:19.290718502 +0200
@@ -20,7 +20,7 @@
 
 add_library(libjsonpbparse STATIC
 	extras/libjsonpb/parse/jsonpb.cpp)
-target_link_libraries(libjsonpbparse PRIVATE libbase)
+target_link_libraries(libjsonpbparse PRIVATE libbase absl_status)
 target_include_directories(libjsonpbparse PUBLIC
 	extras/libjsonpb/parse/include)
 
