--- configure.orig	2009-03-23 06:29:01.403464750 +0100
+++ configure	2009-03-23 06:29:21.114421717 +0100
@@ -23082,7 +23082,7 @@ fi
 	ac_ext=c
 ac_cpp='$CPP $CPPFLAGS'
 ac_compile='$CC -c $CFLAGS $CPPFLAGS conftest.$ac_ext >&5'
-ac_link='$CC -o conftest$ac_exeext $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS >&5'
+ac_link='$CC -o conftest$ac_exeext $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext -lsmbios $LIBS >&5'
 ac_compiler_gnu=$ac_cv_c_compiler_gnu
 
 	if test "$LIB_SMBIOS" = "yes" ; then
