diff --git a/WHATS_NEW_DM b/WHATS_NEW_DM
index 2d0b05c..3f8d9c9 100644
--- a/WHATS_NEW_DM
+++ b/WHATS_NEW_DM
@@ -1,3 +1,7 @@
+Version 1.02.78 - 
+===================================
+  Fix dm_task_set_cookie to properly process udev flags if udev_sync disabled.
+
 Version 1.02.77 - 15th October 2012
 ===================================
   Support unmount of thin volumes from pool above thin pool threshold.
diff --git a/libdm/libdm-common.c b/libdm/libdm-common.c
index b8533ed..afdac89 100644
--- a/libdm/libdm-common.c
+++ b/libdm/libdm-common.c
@@ -1838,6 +1838,7 @@ int dm_task_set_cookie(struct dm_task *dmt, uint32_t *cookie, uint16_t flags)
 	if (dm_cookie_supported())
 		dmt->event_nr = flags << DM_UDEV_FLAGS_SHIFT;
 	*cookie = 0;
+	dmt->cookie_set = 1;
 
 	return 1;
 }
@@ -2207,6 +2208,7 @@ int dm_task_set_cookie(struct dm_task *dmt, uint32_t *cookie, uint16_t flags)
 
 	if (!dm_udev_get_sync_support()) {
 		*cookie = 0;
+		dmt->cookie_set = 1;
 		return 1;
 	}
 
