--- tools/dmsetup.c.orig	2012-05-26 16:02:35.578910688 +0200
+++ tools/dmsetup.c	2012-05-26 16:02:54.250473829 +0200
@@ -1009,7 +1009,7 @@ static int _udevcookies(CMD_ARGS)
 static int _set_up_udev_support(const char *dev_dir)
 {
 	struct udev *udev;
-	const char *udev_dev_dir;
+	const char *udev_dev_dir = "/dev";
 	size_t udev_dev_dir_len;
 	int dirs_diff;
 	const char *env;
@@ -1030,9 +1030,7 @@ static int _set_up_udev_support(const ch
 			  " defined by --udevcookie option.",
 			  _udev_cookie);
 
-	if (!(udev = udev_new()) ||
-	    !(udev_dev_dir = udev_get_dev_path(udev)) ||
-	    !*udev_dev_dir) {
+	if (!(udev = udev_new())) {
 		log_error("Could not get udev dev path.");
 		return 0;
 	}
--- lib/misc/lvm-wrappers.c.orig	2012-05-26 16:03:22.657809165 +0200
+++ lib/misc/lvm-wrappers.c	2012-05-26 16:03:46.664247487 +0200
@@ -72,7 +72,7 @@ const char *udev_get_dev_dir(void)
 		return NULL;
 	}
 
-	return udev_get_dev_path(_udev);
+	return "/dev";
 }
 
 struct udev* udev_get_library_context(void)
