--- a/vapi/libgdata.vapi	2019-08-12 11:01:50.347941770 +0200
+++ b/vapi/libgdata.vapi	2019-08-12 11:02:02.993946056 +0200
@@ -1916,7 +1916,6 @@ namespace GData {
 		public abstract bool is_authorized_for_domain (GData.AuthorizationDomain domain);
 		public abstract void process_request (GData.AuthorizationDomain? domain, Soup.Message message);
 		public virtual bool refresh_authorization (GLib.Cancellable? cancellable = null) throws GLib.Error;
-		public virtual async bool refresh_authorization_async (GLib.Cancellable? cancellable) throws GLib.Error;
 	}
 	[CCode (cheader_filename = "gdata/gdata.h", type_id = "gdata_batchable_get_type ()")]
 	public interface Batchable : GData.Service {
