--- a/src/mtcrypt.c
+++ b/src/mtcrypt.c
@@ -81,8 +81,8 @@
 	if ((passthru = HXmc_meminit(NULL, strlen(copt))) == NULL)
 		abort();
 
-	for (char *key = strtok_r(copt, ",", &copt); key != nullptr;
-	     key = strtok_r(nullptr, ",", &copt)) {
+	for (char *key = strtok_r(copt, ",", &copt); key != NULL;
+	     key = strtok_r(NULL, ",", &copt)) {
 		char *value = strchr(key, '=');
 
 		if (value != NULL)
--- a/src/rdconf1.c
+++ b/src/rdconf1.c
@@ -291,8 +291,8 @@
 	if (str == NULL || *str == '\0')
 		return true;
 
-	for (char *ptr = strtok_r(str, ",", &str); ptr != nullptr;
-	     ptr = strtok_r(nullptr, ",", &str)) {
+	for (char *ptr = strtok_r(str, ",", &str); ptr != NULL;
+	     ptr = strtok_r(NULL, ",", &str)) {
 		kvp = malloc(sizeof(struct kvp));
 		if (kvp == NULL) {
 			l0g("%s: malloc: %s\n", __func__, strerror(errno));
@@ -329,8 +329,8 @@
 	if (str == NULL || *str == '\0')
 		return true;
 
-	for (char *ptr = strtok_r(str, ",", &str); ptr != nullptr;
-	     ptr = strtok_r(nullptr, ",", &str)) {
+	for (char *ptr = strtok_r(str, ",", &str); ptr != NULL;
+	     ptr = strtok_r(NULL, ",", &str)) {
 		char *value = strchr(ptr, '=');
 		if (value != NULL) {
 			*value++ = '\0';
@@ -707,7 +707,7 @@
 		while ((arg = get_next_argument(&wp)) != NULL)
 			/*
 			 * The copy taken with strdup() is not freed in this
-			 * function, because it is used soon. 
+			 * function, because it is used soon.
 			 * @config->command[command][0] will be the pointer to
 			 * the block to free later.
 			 */
@@ -832,7 +832,7 @@
 		free(options);
 		if (!ret)
 			return "Error parsing required options";
-	}	
+	}
 
 	return NULL;
 }
--- a/src/crypto.c
+++ b/src/crypto.c
@@ -652,8 +652,8 @@
 		return -errno;
 
 	verdict = EHD_SECURITY_UNSPEC;
-	for (char *wp = strtok_r(base, ",-.:_", &tmp); wp != nullptr;
-	     wp = strtok_r(nullptr, ",-.:_", &tmp)) {
+	for (char *wp = strtok_r(base, ",-.:_", &tmp); wp != NULL;
+	     wp = strtok_r(NULL, ",-.:_", &tmp)) {
 		ret = __cipher_digest_security(wp);
 		if (verdict == EHD_SECURITY_UNSPEC)
 			verdict = ret;
