--- a/src/External/sol/sol.hpp
+++ b/src/External/sol/sol.hpp
@@ -6818,7 +6818,7 @@
 			}
 
 			static int push(lua_State* L, const wchar_t(&str)[N], std::size_t sz) {
-				return stack::push<const wchar_t*>(L, str, str + sz);
+				return stack::push<const wchar_t*>(L, str + 0, str + sz);
 			}
 		};
 
@@ -6829,7 +6829,7 @@
 			}
 
 			static int push(lua_State* L, const char16_t(&str)[N], std::size_t sz) {
-				return stack::push<const char16_t*>(L, str, str + sz);
+				return stack::push<const char16_t*>(L, str + 0, str + sz);
 			}
 		};
 
@@ -6840,7 +6840,7 @@
 			}
 
 			static int push(lua_State* L, const char32_t(&str)[N], std::size_t sz) {
-				return stack::push<const char32_t*>(L, str, str + sz);
+				return stack::push<const char32_t*>(L, str + 0, str + sz);
 			}
 		};
 
