--- a/src/Utility/Colour.h	2022-05-13 15:08:51.000000000 +0200
+++ -	2023-01-02 14:26:37.082908132 +0100
@@ -13,7 +13,7 @@
 
 	// Constructors
 	ColRGBA() = default;
-	ColRGBA(uint8_t r, uint8_t g, uint8_t b, uint8_t a = 255, char blend = -1, short index = -1) :
+	ColRGBA(uint8_t r, uint8_t g, uint8_t b, uint8_t a = 255, signed char blend = -1, short index = -1) :
 		r{ r },
 		g{ g },
 		b{ b },
@@ -25,7 +25,7 @@
 	explicit ColRGBA(const wxColour& c) : r{ c.Red() }, g{ c.Green() }, b{ c.Blue() }, a{ c.Alpha() } {}
 
 	// Functions
-	void set(uint8_t r, uint8_t g, uint8_t b, uint8_t a = 255, char blend = -1, short index = -1)
+	void set(uint8_t r, uint8_t g, uint8_t b, uint8_t a = 255, signed char blend = -1, short index = -1)
 	{
 		this->r     = r;
 		this->g     = g;
