--- systemstub_sdl.cpp
+++ systemstub_sdl.cpp
@@ -403,12 +403,12 @@ while (true) {
 			if (ev.key.keysym.mod & KMOD_ALT) {
 				if (ev.key.keysym.sym == SDLK_RETURN) {
 					switchGfxMode(!_fullscreen, _currentScaler);
-				} else if (ev.key.keysym.sym == SDLK_KP_PLUS) {
+				} else if (ev.key.keysym.sym == SDLK_PLUS) {
 					uint8 s = _currentScaler + 1;
 					if (s < NUM_SCALERS) {
 						switchGfxMode(_fullscreen, s);
 					}
-				} else if (ev.key.keysym.sym == SDLK_KP_MINUS) {
+				} else if (ev.key.keysym.sym == SDLK_MINUS) {
 					int8 s = _currentScaler - 1;
 					if (_currentScaler > 0) {
 						switchGfxMode(_fullscreen, s);
@@ -429,9 +429,9 @@ while (true) {
 					_pi.save = true;
 				} else if (ev.key.keysym.sym == SDLK_l) {
 					_pi.load = true;
-				} else if (ev.key.keysym.sym == SDLK_KP_PLUS) {
+				} else if (ev.key.keysym.sym == SDLK_PLUS) {
 					_pi.stateSlot = 1;
-				} else if (ev.key.keysym.sym == SDLK_KP_MINUS) {
+				} else if (ev.key.keysym.sym == SDLK_MINUS) {
 					_pi.stateSlot = -1;
 				} else if (ev.key.keysym.sym == SDLK_r) {
 					_pi.inpRecord = true;
