diff --git a/configure b/configure
index d317fd7..5560d75 100755
--- a/configure
+++ b/configure
@@ -60896,7 +60896,7 @@ $as_echo_n "checking for pg_config... " >&6; }
     fi
   done
 
-  if test -n "$PG_CONFIG"; then
+  if test -n "$PG_CONFIG" && test "x$cross_compiling" != "xyes"; then
     { $as_echo "$as_me:${as_lineno-$LINENO}: result: $PG_CONFIG" >&5
 $as_echo "$PG_CONFIG" >&6; }
     PGSQL_INCLUDE=`$PG_CONFIG --includedir`
@@ -62422,7 +62422,7 @@ $as_echo_n "checking for pg_config... " >&6; }
     fi
   done
 
-  if test -n "$PG_CONFIG"; then
+  if test -n "$PG_CONFIG" && test "x$cross_compiling" != "xyes"; then
     { $as_echo "$as_me:${as_lineno-$LINENO}: result: $PG_CONFIG" >&5
 $as_echo "$PG_CONFIG" >&6; }
     PGSQL_INCLUDE=`$PG_CONFIG --includedir`
