diff -ruN a/.cargo/config.toml b/.cargo/config.toml
--- a/.cargo/config.toml	2022-08-04 13:19:26.079217731 -0600
+++ b/.cargo/config.toml	2022-08-04 13:19:50.099215011 -0600
@@ -9,8 +9,8 @@
 linker = "aarch64-linux-gnu-gcc"
 
# I have not tested this, as I have no aarch64 devices and could not get any
# emulators to work, but this flag causes `rustc` to use system libraries /
# object files, which I think is desirable. I would appreciate someone letting
# me know if this causes errors on run.
 [target.aarch64-unknown-linux-musl]
-rustflags = ["-Clink-self-contained=yes"]
+rustflags = ["-Clink-self-contained=no"]
 linker = "rust-lld"
 
# Without this, check stage fails with a SIGSEGV. This workaround is from
# <https://github.com/rust-lang/rust/issues/85543>.
 [target.x86_64-unknown-linux-musl]
-rustflags = ["-Clink-self-contained=yes"]
+rustflags = ["-Clink-self-contained=no"]
