--- a/daemon/master.c
+++ b/daemon/master.c
@@ -17,6 +17,7 @@
  *
  * ----------------------------------------------------------------------- */
 
+#include <libgen.h>
 #include <stdlib.h>
 #include <string.h>
 #include <memory.h>
--- a/daemon/automount.c
+++ b/daemon/automount.c
@@ -21,6 +21,7 @@
 
 #include <dirent.h>
 #include <getopt.h>
+#include <libgen.h>
 #include <signal.h>
 #include <stdio.h>
 #include <stdlib.h>
--- a/modules/lookup_file.c
+++ b/modules/lookup_file.c
@@ -13,6 +13,7 @@
  *
  * ----------------------------------------------------------------------- */
 
+#include <libgen.h>
 #include <stdio.h>
 #include <malloc.h>
 #include <stdlib.h>
