--- lib/nss_parse.y.orig	2016-01-10 10:07:26.854512310 +0100
+++ lib/nss_parse.y	2016-01-10 10:07:53.134783321 +0100
@@ -28,6 +28,10 @@
 #include "nsswitch.h"
 #include "nss_parse.tab.h"
 
+#ifndef _PATH_NSSWITCH_CONF
+#define _PATH_NSSWITCH_CONF     "/etc/nsswitch.conf"
+#endif
+
 static pthread_mutex_t parse_mutex = PTHREAD_MUTEX_INITIALIZER;
 
 static struct list_head *nss_list;
--- daemon/automount.c.orig	2016-01-10 10:10:23.680333952 +0100
+++ daemon/automount.c	2016-01-10 10:10:35.599456586 +0100
@@ -37,6 +37,14 @@
 #include <sys/vfs.h>
 #include <sys/utsname.h>
 
+#ifndef __SWORD_TYPE
+# if __WORDSIZE == 32  /* System word size */
+#  define __SWORD_TYPE int
+# else /* __WORDSIZE == 64 */
+#  define __SWORD_TYPE long int
+# endif
+#endif
+
 #include "automount.h"
 #if defined(LIBXML2_WORKAROUND) || defined(TIRPC_WORKAROUND)
 #include <dlfcn.h>
