diff --git a/src/gr-recipe-formatter.c b/src/gr-recipe-formatter.c
index 0299562..4b6ed47 100644
--- src/gr-recipe-formatter.c
+++ src/gr-recipe-formatter.c
@@ -48,6 +48,7 @@ get_temperature_unit (void)
         GSettings *settings = gr_settings_get ();
         unit =  g_settings_get_enum (settings, "temperature-unit");
         if (unit == GR_TEMPERATURE_UNIT_LOCALE) {
+#ifdef __GLIBC__
                 const gchar *fmt;
 
                 fmt = nl_langinfo (_NL_MEASUREMENT_MEASUREMENT);
@@ -55,6 +56,9 @@ get_temperature_unit (void)
                         unit = GR_TEMPERATURE_UNIT_FAHRENHEIT;
                 else
                         unit = GR_TEMPERATURE_UNIT_CELSIUS;
+#else
+                unit = GR_TEMPERATURE_UNIT_CELSIUS;
+#endif
         }
 
         return unit;
