--- configure.orig
+++ configure
@@ -162,14 +162,6 @@
 	if ${COMP} -o "test-${1}" "${SOURCEDIR}/test-${1}.c" ${3} 1>&3 2>&3
 	then
 		echo "partial result of ${1}${3}: ${CC} succeeded" 1>&3
-	else
-		echo "result of ${1}${3}: ${CC} failed with exit status $?" 1>&3
-		echo "result of compiling ${1}${3}: no" 1>&3
-		echo 1>&3
-		return 1
-	fi
-
-	if ./test-${1} 1>&3 2>&3; then
 		echo "tested ${1}${3}: yes" 1>&2
 		echo "result of running ${1}${3}: yes" 1>&3
 		echo 1>&3
@@ -177,10 +169,9 @@
 		rm "test-${1}"
 		return 0
 	else
-		echo "result of ${1}${3}: execution failed with exit status $?" 1>&3
+		echo "result of ${1}${3}: ${CC} failed with exit status $?" 1>&3
 		echo "result of running ${1}${3}: no" 1>&3
 		echo 1>&3
-		rm "test-${1}"
 		return 1
 	fi
 }
