--- a/src/player/CodecsComponent.cpp	2020-05-19 19:32:23.000000000 +0200
+++ b/src/player/CodecsComponent.cpp	2020-07-28 08:00:34.966060781 +0200
@@ -684,9 +684,11 @@
 {
   Downloader::HeaderList headers;
   QString auth = SystemComponent::Get().authenticationToken();
+  QString prod = WITH_CODECS ? "Plex Media Player" : "openpmp";
+  QString plat = "Konvergo";
   if (auth.size())
     headers.append({"X-Plex-Token", auth});
-  headers.append({"X-Plex-Product", WITH_CODECS ? "Plex Media Player" : "openpmp"});
-  headers.append({"X-Plex-Platform", "Konvergo"});
+  headers.append({"X-Plex-Product", prod});
+  headers.append({"X-Plex-Platform", plat});
   return headers;
 }
