--- a/3rdParty/LibMiniUPnPc/SConscript	2016-09-07 10:29:12.251340787 +0200
+++ b/3rdParty/LibMiniUPnPc/SConscript	2016-09-07 10:29:49.165342691 +0200
@@ -46,11 +46,11 @@
 		if myenv["PLATFORM"] == "win32":
 			myenv.Append(CCFLAGS = ["-DWIN32", "-D_WIN32_WINNT=0x0501"])
 
-		if myenv["PLATFORM"] != "win32" and myenv["PLATFORM"] != "darwin":
-			conf = Configure(conf_env)
-			if conf.CheckTypeSize("struct ip_mreqn", includes="#include <netinet/in.h>") == 0:
-				myenv.Append(CCFLAGS = ["-DNEED_STRUCT_IP_MREQN"])
-			conf.Finish()
+		#if myenv["PLATFORM"] != "win32" and myenv["PLATFORM"] != "darwin":
+		#	conf = Configure(conf_env)
+		#	if conf.CheckTypeSize("struct ip_mreqn", includes="#include <netinet/in.h>") == 0:
+		#		myenv.Append(CCFLAGS = ["-DNEED_STRUCT_IP_MREQN"])
+		#	conf.Finish()
 
 		myenv.WriteVal("src/miniupnpc/miniupnpcstrings.h", myenv.Value(
 """
