--- configure.ac
+++ configure.ac
@@ -102,7 +102,7 @@ dnl We need libncurses
 dnl
 AC_DEFUN([AC_NCURSES_REQUIREMENTS],
 [
-    PKG_CHECK_MODULES(NCURSES, [ncurses >= 5], ncurses_has_pkgconfig=1, ncurses_has_pkgconfig=0)
+    PKG_CHECK_MODULES(NCURSES, [ncursesw >= 5], ncurses_has_pkgconfig=1, ncurses_has_pkgconfig=0)
     if test x$ncurses_has_pkgconfig = x0; then
       AC_PATH_PROG(NCURSES_CONFIG,ncurses6-config,false)
       if test "$NCURSES_CONFIG" = false; then
