--- inc/common/UFO/portable_compiler.h.orig	2020-05-13 11:41:37.725900359 +0200
+++ inc/common/UFO/portable_compiler.h	2020-05-13 11:42:26.407898739 +0200
@@ -125,7 +125,7 @@
 /* compile-time ASSERT */
 
 #ifndef C_ASSERT
-    #define __UNIQUENAME( a1, a2 )  __CONCAT( a1, a2 )
+    #define __UNIQUENAME( a1, a2 )  a1 ## a2
     #define UNIQUENAME( __text )    __UNIQUENAME( __text, __COUNTER__ )
 
 
--- visa/CISA.y.orig	2020-05-13 11:42:12.245899211 +0200
+++ visa/CISA.y	2020-05-13 11:42:26.410898739 +0200
@@ -25,6 +25,7 @@
 ======================= end_copyright_notice ==================================*/
 
 %{
+#include <inttypes.h>
 #include <stdio.h>
 #include <math.h>
 #include <string>
