--- libunique-1.16/unique/uniqueapp.c	2020-01-19 15:48:40.380617759 +0100
+++ libunique-1.16/unique/uniqueapp.c	2020-01-19 16:02:19.196585111 +0100
@@ -55,6 +55,9 @@
 #include <stdlib.h>
 #include <string.h>
 
+/* Avoid error for macros deprecated in glib-2.58 */
+#define GLIB_VERSION_MIN_REQUIRED GLIB_VERSION_2_56
+#define GLIB_VERSION_MAX_ALLOWED GLIB_VERSION_2_56
 #include <glib.h>
 #include <glib/gi18n-lib.h>
 
