--- a/GNUmakefile	2016-09-05 13:51:19.133051791 +0200
+++ b/GNUmakefile	2016-09-05 13:51:29.519052327 +0200
@@ -13,7 +13,7 @@
 includedir=	$(prefix)/include
 mandir=		$(prefix)/man
 
-PKG_CONFIG=	/usr/bin/pkg-config --silence-errors
+PKG_CONFIG=	pkg-config --silence-errors
 INSTALL=	/usr/bin/install
 STRIP=		/usr/bin/strip
 
@@ -33,7 +33,7 @@
   $(error You probably need to install "libbsd-dev" or "libbsd-devel" or something like that.)
 endif
 
-CURSES_LIBS:= $(shell $(PKG_CONFIG) --libs ncurses)
+CURSES_LIBS:= $(shell $(PKG_CONFIG) --libs ncursesw)
 ifeq ($(CURSES_LIBS),)
   $(error You probably need to install "libncurses5-dev" or "libnnurses6-devel" or something like that.)
 endif
