--- a/src/xsb_level.cpp	2012-09-15 22:21:06.000000000 +0200
+++ b/src/xsb_level.cpp	2016-09-23 10:56:51.900831887 +0200
@@ -784,7 +784,7 @@
 
 bool FromXyeDFS(int* mem, unsigned char x, unsigned char y)
 {
-    static const unsigned char dx[4] = {0,0,-1,1}, dy[4] = {-1,1,0,0};
+    static const unsigned char dx[4] = {0x00,0x00,0xff,0x01}, dy[4] = {0xff,0x01,0x00,0x00};
     int &res =  mem[y*XYE_HORZ+x];
     if( res==0)
     {
