--- a/src/src.pro	2018-06-04 11:43:53.000000000 +0200
+++ b/src/src.pro	2018-06-04 11:43:53.000000000 +0200
@@ -7,6 +7,32 @@
 CONFIG += c++11 staticlib
 CONFIG(release, debug|release):DEFINES += QT_NO_DEBUG_OUTPUT
 
+TRANSLATIONS    +=  ../localization/localization_nl_NL.ts \
+                    ../localization/localization_nl.ts \
+                    ../localization/localization_ca.ts \
+                    ../localization/localization_de_DE.ts \
+                    ../localization/localization_es_ES.ts \
+                    ../localization/localization_gl_ES.ts \
+                    ../localization/localization_hu_HU.ts \
+                    ../localization/localization_sv_SE.ts \
+                    ../localization/localization_pl_PL.ts \
+                    ../localization/localization_ru_RU.ts \
+                    ../localization/localization_he_IL.ts \
+                    ../localization/localization_zh_CN.ts \
+                    ../localization/localization_ar_MA.ts \
+                    ../localization/localization_fr_FR.ts \
+                    ../localization/localization_fr_BE.ts \
+                    ../localization/localization_nl_BE.ts \
+                    ../localization/localization_fr_LU.ts \
+                    ../localization/localization_de_LU.ts \
+                    ../localization/localization_lb_LU.ts \
+                    ../localization/localization_en_GB.ts \
+                    ../localization/localization_en_US.ts \
+                    ../localization/localization_el_GR.ts \
+                    ../localization/localization_cs_CZ.ts \
+                    ../localization/localization_it_IT.ts \
+                    ../localization/localization_pt_PT.ts
+
 SOURCES   += mainwindow.cpp \
              configdialog.cpp \
              storemodel.cpp \
