diff --git rpc/rpc-lib/src/rpcsvc.c rpc/rpc-lib/src/rpcsvc.c
index 5639abb..88ffd25 100644
--- rpc/rpc-lib/src/rpcsvc.c
+++ rpc/rpc-lib/src/rpcsvc.c
@@ -2287,6 +2287,7 @@ rpcsvc_program_register(rpcsvc_t *svc, rpcsvc_program_t *program,
     memcpy(newprog, program, sizeof(*program));
 
     INIT_LIST_HEAD(&newprog->program);
+#ifdef __GLIBC__
     pthread_mutexattr_init(&thr_attr);
     pthread_mutexattr_settype(&thr_attr, PTHREAD_MUTEX_ADAPTIVE_NP);
 
@@ -2301,7 +2302,7 @@ rpcsvc_program_register(rpcsvc_t *svc, rpcsvc_program_t *program,
 
     pthread_mutex_init(&newprog->thr_lock, &thr_attr);
     pthread_cond_init(&newprog->thr_cond, NULL);
-
+#endif
     newprog->alive = _gf_true;
 
     if (gf_async_ctrl.enabled) {
diff --git api/src/glfs.h api/src/glfs.h
index 279d11d..054730a 100644
--- api/src/glfs.h
+++ api/src/glfs.h
@@ -67,7 +67,7 @@
  * is false and __USE_FILE_OFFSET64 is true, then go on to define
  * off64_t using __off64_t.
  */
-#ifndef GF_BSD_HOST_OS
+#ifdef GF_BSD_HOST_OS
 #if defined(__USE_FILE_OFFSET64) && !defined(__off64_t_defined)
 typedef __off64_t off64_t;
 #endif /* defined(__USE_FILE_OFFSET64) && !defined(__off64_t_defined) */
diff --git contrib/fuse-lib/mount-common.c contrib/fuse-lib/mount-common.c
index cffd4c0..5300fc2 100644
--- contrib/fuse-lib/mount-common.c
+++ contrib/fuse-lib/mount-common.c
@@ -22,6 +22,9 @@
  * For changes that were made later and syncs with upstream,
  * see the commit log and per-function comments.
  */
+#ifndef __GLIBC__
+#include <paths.h>
+#endif
 
 #ifdef GF_LINUX_HOST_OS
 /* FUSE: cherry-picked bd99f9cf */
diff --git xlators/mgmt/glusterd/src/glusterd-utils.c xlators/mgmt/glusterd/src/glusterd-utils.c
index 5b84a3e..34a11d5 100644
--- xlators/mgmt/glusterd/src/glusterd-utils.c
+++ xlators/mgmt/glusterd/src/glusterd-utils.c
@@ -90,6 +90,9 @@
 #define NLM_PROGRAM 100021
 #define NLMV4_VERSION 4
 #define NLMV1_VERSION 1
+#ifndef __GLIBC__
+#include <paths.h>
+#endif
 
 gf_boolean_t
 is_brick_mx_enabled(void)
diff --git contrib/fuse-util/fusermount.c contrib/fuse-util/fusermount.c
index ff743f7..31afa8a 100644
--- contrib/fuse-util/fusermount.c
+++ contrib/fuse-util/fusermount.c
@@ -58,6 +58,9 @@
 #ifndef MS_PRIVATE
 #define MS_PRIVATE (1<<18)
 #endif
+#ifndef __GLIBC__
+#include <paths.h>
+#endif
 
 static const char *progname;
 
