From 3ad9559e59905cb54c0d97f3d8e163517e4e0afd Mon Sep 17 00:00:00 2001
From: mobinmob <mobinmob@disroot.org>
Date: Mon, 22 Nov 2021 23:16:32 +0200
Subject: [PATCH] limits.h patch

---
 src/FLTK/fl_callbacks.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/FLTK/fl_callbacks.h b/src/FLTK/fl_callbacks.h
index baea11b..026bfb9 100644
--- a/src/FLTK/fl_callbacks.h
+++ b/src/FLTK/fl_callbacks.h
@@ -15,6 +15,7 @@
 
 #ifndef FL_CALLBACKS_H
 #define FL_CALLBACKS_H
+#include <limits.h>
 
 enum { STREAM_TIME = 0, REC_TIME, SENT_DATA, REC_DATA };
 enum { STREAM = 0, RECORD };
-- 
2.34.0
