--- plugins/network/networksupport.cpp
+++ plugins/network/networksupport.cpp
@@ -89,7 +89,9 @@ Q_DECLARE_METATYPE(QNetworkInterface::InterfaceFlags)
 #endif
 Q_DECLARE_METATYPE(QNetworkProxy::Capabilities)
 Q_DECLARE_METATYPE(QNetworkProxy::ProxyType)
+#if QT_VERSION < QT_VERSION_CHECK(5, 15, 0)
 Q_DECLARE_METATYPE(QSocketNotifier::Type)
+#endif
 #ifndef QT_NO_SSL
 Q_DECLARE_METATYPE(QSsl::KeyAlgorithm)
 Q_DECLARE_METATYPE(QSsl::KeyType)
--- plugins/widgetinspector/overlaywidget.h
+++ plugins/widgetinspector/overlaywidget.h
@@ -30,6 +30,7 @@
 #define GAMMARAY_WIDGETINSPECTOR_OVERLAYWIDGET_H
 
 #include <QLayout>
+#include <QPainterPath>
 #include <QPointer>
 #include <QWidget>
 
