--- media/gpu/vaapi/BUILD.gn.i686permissive	2018-11-29 09:27:02.405909871 -0500
+++ media/gpu/vaapi/BUILD.gn	2018-11-29 09:29:50.648259696 -0500
@@ -10,6 +10,11 @@ import("//ui/ozone/ozone.gni")
 
 assert(use_vaapi)
 
+config("vaapi_permissive") {
+  cflags = [ "-fpermissive" ]
+}
+
+
 action("libva_generate_stubs") {
   extra_header = "va_stub_header.fragment"
 
@@ -98,6 +103,8 @@ source_set("vaapi") {
     "//third_party/libyuv",
   ]
 
+  configs += [ ":vaapi_permissive" ]
+
   if (use_x11) {
     configs += [ "//build/config/linux:x11" ]
     deps += [ "//ui/gfx/x" ]
--- media/gpu/vaapi/vaapi_wrapper.cc    2019-02-01 20:00:10.825233133 +0100
+++ media/gpu/vaapi/vaapi_wrapper.cc	2019-02-01 20:01:04.955228978 +0100
@@ -1009,7 +1009,7 @@
     }
     fds[i] = dmabuf_fd;
   }
-  va_attrib_extbuf.buffers = fds.data();
+  va_attrib_extbuf.buffers = (unsigned int *) fds.data();
   va_attrib_extbuf.num_buffers = fds.size();
 
   va_attrib_extbuf.flags = 0;