--- media/gpu/vaapi/BUILD.gn.orig	2019-06-18 11:48:14.000000000 -0400
+++ media/gpu/vaapi/BUILD.gn	2019-07-02 18:34:00.607864854 -0400
@@ -11,6 +11,10 @@
 
 assert(use_vaapi)
 
+config("vaapi_permissive") {
+  cflags = [ "-fpermissive" ]
+}
+
 generate_stubs("libva_stubs") {
   extra_header = "va_stub_header.fragment"
   sigs = [ "va.sigs" ]
@@ -89,6 +93,8 @@
     deps += [ "//media/gpu/linux" ]
   }
 
+  configs += [ ":vaapi_permissive" ]
+
   if (use_x11) {
     configs += [ "//build/config/linux:x11" ]
     deps += [ "//ui/gfx/x" ]
