--- files/usr/bin/pastebin.orig	2016-11-10 01:27:40.693838803 -0600
+++ files/usr/bin/pastebin	2016-11-10 01:28:05.234837754 -0600
@@ -22,7 +22,7 @@
         content = str_args
 
 if content != "":
-    for paster in ['/usr/bin/gist-paste', '/usr/bin/fpaste']:
+    for paster in ['/usr/bin/gist']:
         if os.path.exists(paster):
             p = subprocess.Popen([paster], stdin=subprocess.PIPE)
             p.communicate(content.encode("UTF-8"))
