--- a/src/bridges/bridge_argon2id_reference.mk
+++ b/src/bridges/bridge_argon2id_reference.mk
@@ -10,8 +10,6 @@ ifeq ($(UNAME),Darwin)
 ifeq ($(IS_APPLE_SILICON),0)
 ARGON2_REFERENCE_CFLAGS += -mavx2
 endif
-else
-ARGON2_REFERENCE_CFLAGS += -march=native
 endif
 endif
 endif
--- a/src/bridges/bridge_scrypt_jane.mk
+++ b/src/bridges/bridge_scrypt_jane.mk
@@ -10,8 +10,6 @@ ifeq ($(UNAME),Darwin)
 ifeq ($(IS_APPLE_SILICON),0)
 SCRYPT_JANE_CFLAGS += -mavx2
 endif
-else
-SCRYPT_JANE_CFLAGS += -march=native
 endif
 endif
 endif
--- a/src/bridges/bridge_scrypt_yescrypt.mk
+++ b/src/bridges/bridge_scrypt_yescrypt.mk
@@ -10,8 +10,6 @@ ifeq ($(UNAME),Darwin)
 ifeq ($(IS_APPLE_SILICON),0)
 SCRYPT_YESCRYPT_CFLAGS += -mavx2
 endif
-else
-SCRYPT_YESCRYPT_CFLAGS += -march=native
 endif
 endif
 endif
