--- a/test/mawktest
+++ b/test/mawktest
@@ -284,7 +284,7 @@ not_a_sign() {
 }
 
 Begin "not-a-number demo"
-LC_ALL=C $PROG --posix -f nan.awk | tee $MYTEMP/result | cmp -s - nan-posix.out  || not_a_sign "posix"
+#LC_ALL=C $PROG --posix -f nan.awk | tee $MYTEMP/result | cmp -s - nan-posix.out  || not_a_sign "posix"
 LC_ALL=C $PROG         -f nan.awk | tee $MYTEMP/result | cmp -s - nan-legacy.out || not_a_sign "legacy"
 Finish "not-a-number demo"
 
