diff --git src/CMakeLists.txt src/CMakeLists.txt
Source: https://invent.kde.org/libraries/kquickimageeditor/-/commit/c79657496326c10f81cf80a8c9a0e55bc2ba2798
Issue: https://invent.kde.org/libraries/kquickimageeditor/-/issues/1
---
index 301c05e5efe557970b528b14f9d9c46911b0f684..99b0a8731b5cdc32ee2ccc7b9490152a815ebaed 100644
--- src/CMakeLists.txt
+++ src/CMakeLists.txt
@@ -33,12 +33,6 @@ add_library(
     ${sources} ${pluginData}
 )
 
-target_compile_options(
-    kquickimageeditorplugin
-    PRIVATE
-        -Werror
-)
-
 target_link_libraries(
     kquickimageeditorplugin
     PRIVATE
