--- src/rpcb_svc_com.c
+++ src/rpcb_svc_com.c
@@ -42,8 +42,6 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <sys/param.h>
-#include <sys/poll.h>
-#include <bits/poll.h>
 #include <sys/socket.h>
 #include <rpc/rpc.h>
 #include <rpc/rpcb_prot.h>
@@ -54,6 +52,7 @@
 #include <unistd.h>
 #include <stdio.h>
 #include <time.h>
+#include <poll.h>
 #ifdef PORTMAP
 #include <netinet/in.h>
 #include <rpc/pmap_prot.h>
