--- a/build/premake/extern_libs5.lua
+++ b/build/premake/extern_libs5.lua
@@ -249,7 +249,7 @@ extern_lib_defs = {
 				-- The following are not strictly link dependencies on all systems, but
 				-- are included for compatibility with different versions of Boost
 				android_names = { "boost_filesystem-gcc-mt", "boost_system-gcc-mt" },
-				unix_names = { os.findlib("boost_filesystem-mt") and "boost_filesystem-mt" or "boost_filesystem", os.findlib("boost_system-mt") and "boost_system-mt" or "boost_system" },
+				unix_names = { "boost_filesystem" },
 				osx_names = { "boost_filesystem", "boost_system" },
 			})
 		end,
