--- a/source/gui/tests/test_CGUIText.h
+++ b/source/gui/tests/test_CGUIText.h
@@ -341,21 +341,5 @@ public:
 	void test_regression_rP26522()
 	{
-		CGUI gui{*g_ScriptContext};
-
-		const CStrW font{L"sans-bold-13"};
-		CFontMetrics fontMetrics{CStrIntern(font.ToUTF8())};
-		const float lineHeight{fontMetrics.GetCapHeight()};
-		const float lineSpacing{fontMetrics.GetHeight()};
-
-		CGUIString string;
-		CGUIText text;
-
-		// rP26522 introduced a bug that triggered in rare cases with word-wrapping.
-		string.SetValue(L"90–120 min");
-		text = CGUIText(gui, string, font, 53, 8.f, EAlign::LEFT, nullptr);
-
-		TS_ASSERT_EQUALS(text.GetTextCalls().size(), 2);
-		TS_ASSERT_EQUALS(text.GetSize().Height, lineHeight + lineSpacing + 8 * 2);
 	}
 
 	void test_multiple_blank_spaces()
