--- libraries/source/nvtt/src/src/nvmath/nvmath.h.orig	2018-06-16 16:01:48.511540946 +0200
+++ libraries/source/nvtt/src/src/nvmath/nvmath.h	2018-06-16 16:02:29.935000183 +0200
@@ -130,7 +130,7 @@
 {
 #if NV_OS_WIN32
 	return _isnan(f) != 0;
-#elif NV_OS_DARWIN || NV_OS_FREEBSD || NV_OS_OPENBSD
+#elif NV_OS_DARWIN || NV_OS_FREEBSD || NV_OS_OPENBSD || !defined(__GLIBC__)
 	return isnan(f);
 #elif NV_OS_LINUX
 	return isnanf(f);
--- source/lib/sysdep/os/linux/ldbg.cpp.orig	2018-06-16 16:27:55.639277094 +0200
+++ source/lib/sysdep/os/linux/ldbg.cpp	2018-06-16 16:28:17.342014019 +0200
@@ -33,7 +33,7 @@
 #include "lib/sysdep/sysdep.h"
 #include "lib/debug.h"
 
-#if OS_ANDROID
+#if OS_ANDROID || !defined(__GLIBC__)
 
 // Android NDK doesn't support backtrace()
 // TODO: use unwind.h or similar?
