Index: 0ad-0.0.26-alpha/libraries/source/spidermonkey/build.sh
===================================================================
--- 0ad-0.0.26-alpha.orig/libraries/source/spidermonkey/build.sh
+++ 0ad-0.0.26-alpha/libraries/source/spidermonkey/build.sh
@@ -129,7 +129,7 @@ else
 fi
 
 # Debug version of SM is broken on FreeBSD.
-if [ "$(uname -s)" != "FreeBSD" ]; then
+if false; then
   mkdir -p build-debug
   cd build-debug
   # SM configure checks for autoconf, but we don't actually need it.
