diff --git a/libraries/source/spidermonkey/build.sh b/libraries/source/spidermonkey/build.sh
index 0dc66e01847..edbe032b525 100755
--- a/libraries/source/spidermonkey/build.sh
+++ b/libraries/source/spidermonkey/build.sh
@@ -46,8 +46,9 @@ if [ ! -e "${FOLDER}.tar.xz" ]; then
 fi
 
 # unpack
-rm -Rf "${FOLDER}"
-tar xfJ "${FOLDER}.tar.xz"
+# done via xbps-src extract
+#rm -Rf "${FOLDER}"
+#tar xfJ "${FOLDER}.tar.xz"
 
 # patch
 (
diff --git a/libraries/source/cxxtest-4.4/build.sh b/libraries/source/cxxtest-4.4/build.sh
index 3243836ecb1..1ab6e46fff5 100755
--- a/libraries/source/cxxtest-4.4/build.sh1
+++ b/libraries/source/cxxtest-4.4/build.sh
@@ -39,8 +39,9 @@ if [ ! -e "cxxtest-${PV}.tar.gz" ]; then
 fi
 
 # unpack
-rm -Rf "cxxtest-${PV}"
-tar -xf "cxxtest-${PV}.tar.gz"
+# done via xbps-src extract
+#rm -Rf "cxxtest-${PV}"
+#tar -xf "cxxtest-${PV}.tar.gz"
 
 # patch
 patch -d "cxxtest-${PV}" -p1 <patches/0001-Add-Debian-python3-patch.patch
diff --git a/libraries/source/fcollada/build.sh b/libraries/source/fcollada/build.sh
index 4ba98b9d2a8..4edc51846fe 100755
--- a/libraries/source/fcollada/build.sh
+++ b/libraries/source/fcollada/build.sh
@@ -41,8 +41,9 @@ if [ ! -e "fcollada-${PV}.tar.xz" ]; then
 fi
 
 # unpack
-rm -Rf fcollada-${PV}
-tar xf fcollada-${PV}.tar.xz
+# done via xbps-src extract
+#rm -Rf fcollada-${PV}
+#tar xf fcollada-${PV}.tar.xz
 
 # build
 (
diff --git a/libraries/source/nvtt/build.sh b/libraries/source/nvtt/build.sh
index 80f29566e23..a5c5b0b7e66 100755
--- a/libraries/source/nvtt/build.sh
+++ b/libraries/source/nvtt/build.sh
@@ -41,8 +41,9 @@ if [ ! -e "nvtt-${PV}.tar.xz" ]; then
 fi
 
 # unpack
-rm -Rf nvtt-${PV}
-tar xf nvtt-${PV}.tar.xz
+# done via xbps-src extract
+#rm -Rf nvtt-${PV}
+#tar xf nvtt-${PV}.tar.xz
 
 # patch
 patch -d nvtt-${PV} -p1 <patches/0001-Don-t-overspecify-flags.patch
