Index: 0ad-0.0.26/libraries/source/spidermonkey/patch.sh
===================================================================
--- 0ad-0.0.26.orig/libraries/source/spidermonkey/patch.sh
+++ 0ad-0.0.26/libraries/source/spidermonkey/patch.sh
@@ -2,6 +2,24 @@
 # Apply patches if needed
 # This script gets called from build.sh.
 
+unset GIT_DIR
+HOME=/tmp
+PATH=/usr/libexec/chroot-git:$PATH
+GIT_CEILING_DIRECTORIES="$(cd .. && pwd)"
+export GIT_CEILING_DIRECTORIES HOME
+git config --global --add safe.directory "$PWD"
+git config --global user.name 'Void Linux'
+git config --global user.email 'none@voidlinux.org'
+git init . -b for-build
+git add . >/dev/null 2>&1
+git commit -m needs-git-because-virtual-env-has-binaries-diff
+
+git am --keep-cr ../1654457-virtualenv.patch
+git am --keep-cr ../python-3.11.patch
+patch -p1 < ../rust-target.patch
+
+ sed -i -e 's/exec python3 /exec python3.11 /' js/src/configure*
+
 # SM78 fails to create virtual envs on macs with python > 3.7
 # Unfortunately, 3.7 is mostly unavailable on ARM macs.
 # Therefore, replace the custom script with a more up-to-date version from pip
@@ -26,7 +41,7 @@ else
     #
     # It is assumed that the updated version fetched for macOS systems
     # above does not have this problem.
-    patch -p1 < ../FixVirtualenvForPython310.diff
+    : patch -p1 < ../FixVirtualenvForPython310.diff
 fi
 
 # Mozglue symbols need to be linked against static builds.
