--- source/ps/tests/stub_impl_hack.cpp.orig	2021-03-01 11:33:10.873986567 +0100
+++ source/ps/tests/stub_impl_hack.cpp	2021-03-01 11:35:37.991089218 +0100
@@ -13,7 +13,7 @@
 #include <cxxtest/TestMain.h>
 
 bool suite_TestDummy_init = false;
-#include "/Users/wfg/Jenkins/workspace/macOS-all-bundles/source/ps/tests/stub_impl_hack.h"
+#include "stub_impl_hack.h"
 
 static TestDummy suite_TestDummy;
 
