Index: libraries/source/fcollada/src/FColladaPlugins/FArchiveXML/FAXPhysicsExport.cpp
===================================================================
--- libraries/source/fcollada/src/FColladaPlugins/FArchiveXML/FAXPhysicsExport.cpp.orig
+++ libraries/source/fcollada/src/FColladaPlugins/FArchiveXML/FAXPhysicsExport.cpp
@@ -329,16 +329,3 @@ void FArchiveXML::WritePhysicsRigidBodyP
 		FArchiveXML::LetWriteObject(physicsRigidBodyParameters->GetPhysicsShape(i), techniqueNode);
 	}
 }
-
-template <class TYPE, int QUAL>
-xmlNode* FArchiveXML::AddPhysicsParameter(xmlNode* parentNode, const char* name, FCDParameterAnimatableT<TYPE,QUAL>& value)
-{
-	xmlNode* paramNode = AddChild(parentNode, name);
-	AddContent(paramNode, FUStringConversion::ToString((TYPE&) value));
-	if (value.IsAnimated())
-	{
-		const FCDAnimated* animated = value.GetAnimated();
-		FArchiveXML::WriteAnimatedValue(animated, paramNode, name);
-	}
-	return paramNode;
-}
Index: libraries/source/fcollada/src/FColladaPlugins/FArchiveXML/FArchiveXML.h
===================================================================
--- libraries/source/fcollada/src/FColladaPlugins/FArchiveXML/FArchiveXML.h.orig
+++ libraries/source/fcollada/src/FColladaPlugins/FArchiveXML/FArchiveXML.h
@@ -573,4 +573,17 @@ public:
 	static xmlNode* WriteLibrary(FCDLibrary<T>* library, xmlNode* node);
 };
 
+template <class TYPE, int QUAL>
+xmlNode* FArchiveXML::AddPhysicsParameter(xmlNode* parentNode, const char* name, FCDParameterAnimatableT<TYPE,QUAL>& value)
+{
+	xmlNode* paramNode = AddChild(parentNode, name);
+	AddContent(paramNode, FUStringConversion::ToString((TYPE&) value));
+	if (value.IsAnimated())
+	{
+		const FCDAnimated* animated = value.GetAnimated();
+		FArchiveXML::WriteAnimatedValue(animated, paramNode, name);
+	}
+	return paramNode;
+}
+
 #endif //_FCPARCHIVECOLLADA_H_
