From 3ff3f728900b89c4e00bddb03fc1aeb0bf7baad0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=C8=98tefan=20Talpalaru?= <stefantalpalaru@yahoo.com>
Date: Fri, 31 Jan 2025 13:01:40 +0100
Subject: [PATCH] build-source-libs.sh: fix "--with-system-premake"

---
 libraries/build-source-libs.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libraries/build-source-libs.sh b/libraries/build-source-libs.sh
index 323260a51e..da67b293cb 100755
--- a/libraries/build-source-libs.sh
+++ b/libraries/build-source-libs.sh
@@ -62,7 +62,7 @@ while [ "$#" -gt 0 ]; do
 		--with-system-cxxtest) with_system_cxxtest=true ;;
 		--with-system-nvtt) with_system_nvtt=true ;;
 		--with-system-mozjs) with_system_mozjs=true ;;
-		--with-system-premake) with_system_mozjs=true ;;
+		--with-system-premake) with_system_premake=true ;;
 		--with-spirv-reflect) with_spirv_reflect=true ;;
 		-j*) JOBS="$1" ;;
 		*)
