--- src/Makefile.in.orig	2008-10-08 14:39:38.000000000 +0200
+++ src/Makefile.in	2008-10-08 14:39:49.000000000 +0200
@@ -130,7 +130,7 @@
 #
 # Additional linker flags
 #	 
-lib@LIBRARY@_la_LIBADD = $(LIBM) $(X_LIBS) -lGL -lGLU -lXext -lX11 $(LIBXXF86VM)
+lib@LIBRARY@_la_LIBADD = $(LIBM) $(X_LIBS) -lusbhid -lGL -lGLU -lXext -lX11 $(LIBXXF86VM)
 lib@LIBRARY@_la_LDFLAGS = -version-info 11:0:8
 lib@LIBRARY@_la_CFLAGS = $(X_CFLAGS)
 
