*** a/src/test/contribs-lib/analysis/de/TestGermanStemFilter.cpp.orig	2020-08-27 19:02:04.024654943 -0300
--- a/src/test/contribs-lib/analysis/de/TestGermanStemFilter.cpp	2020-08-27 19:02:26.219672761 -0300
***************
*** 8,15 ****
  #include "test.h"
  #include "CLucene/util/CLStreams.h"
  #include "CLucene/analysis/Analyzers.h"
! #include "CLucene/analysis/de/GermanStemmer.h"
! #include "CLucene/analysis/de/GermanStemFilter.h"
  #include "CLucene/analysis/standard/StandardTokenizer.h"
  
  CL_NS_USE(util)
--- 8,15 ----
  #include "test.h"
  #include "CLucene/util/CLStreams.h"
  #include "CLucene/analysis/Analyzers.h"
! #include "CLucene/../../contribs-lib/CLucene/analysis/de/GermanStemmer.h"
! #include "CLucene/../../contribs-lib/CLucene/analysis/de/GermanStemFilter.h"
  #include "CLucene/analysis/standard/StandardTokenizer.h"
  
  CL_NS_USE(util)
