--- include/configs/usbarmory.h.orig	2015-04-29 15:11:43.064537918 +0200
+++ include/configs/usbarmory.h	2015-04-29 15:32:04.900446472 +0200
@@ -82,8 +82,8 @@
 #define CONFIG_BOOTCOMMAND						\
 	"run distro_bootcmd; "						\
 	"setenv bootargs console=${console} ${bootargs_default}; "	\
-	"ext2load mmc 0:1 ${kernel_addr_r} /boot/uImage; "		\
-	"ext2load mmc 0:1 ${fdt_addr_r} /boot/${fdtfile}; "		\
+	"fatload mmc 0:1 ${kernel_addr_r} /uImage; "		\
+	"fatload mmc 0:1 ${fdt_addr_r} /${fdtfile}; "		\
 	"bootm ${kernel_addr_r} - ${fdt_addr_r}"
 
 #define BOOT_TARGET_DEVICES(func) func(MMC, mmc, 0)
@@ -99,7 +99,7 @@
 
 #define CONFIG_EXTRA_ENV_SETTINGS				\
 	MEM_LAYOUT_ENV_SETTINGS					\
-	"bootargs_default=root=/dev/mmcblk0p1 rootwait rw\0"	\
+	"bootargs_default=root=/dev/mmcblk0p2 rootwait rw\0"	\
 	"fdtfile=imx53-usbarmory.dtb\0"				\
 	"console=ttymxc0,115200\0"				\
 	BOOTENV
