--- a/meson.build	2022-06-27 23:54:16.000000000 +1000
+++ b/meson.build	2022-10-03 11:53:08.442051982 +1100
@@ -135,7 +135,7 @@
 #
 # https://bugs.winehq.org/show_bug.cgi?id=49138
 wine_version = run_command(
-  'sh', '-c', '''wine --version | grep --only-matching -P '[0-9]+\.[0-9]+' | head -n1''',
+  'sh', '-c', '''wine --version | grep --only-matching -E '[0-9]+\.[0-9]+' | head -n1''',
   check : false
 )
 if wine_version.returncode() == 0
