--- a/Source/WebCore/css/makegrammar.pl	2020-04-14 00:51:51.000000000 +0200
+++ b/Source/WebCore/css/makegrammar.pl	2020-10-25 14:59:39.879927464 +0100
@@ -73,7 +73,6 @@
 }
 
 my $fileBase = File::Spec->join($outputDir, $filename);
-my @bisonCommand = ($bison, "-d", "-p", $symbolsPrefix, $grammarFilePath, "-o", "$fileBase.cpp");
+my @bisonCommand = ($bison, "--no-lines", "--defines=$fileBase.cpp.h", "-p", $symbolsPrefix, $grammarFilePath, "--output=$fileBase.cpp");
-push @bisonCommand, "--no-lines" if $^O eq "MSWin32"; # Work around bug in bison >= 3.0 on Windows where it puts backslashes into #line directives.
 system(@bisonCommand) == 0 or die;
 
@@ -91,6 +91,5 @@
 print HEADER "#endif\n";
 close HEADER;
 
-unlink("$fileBase.cpp.h");
 unlink("$fileBase.hpp");
 
