Description: add missing #include <functional>
Origin: upstream, https://github.com/annulen/webkit/commit/4ce8ebc4094512b9
Last-Update: 2018-04-15

--- Source/WebCore/dom/SlotAssignment.h
+++ Source/WebCore/dom/SlotAssignment.h
@@ -28,6 +28,7 @@
 
 #if ENABLE(SHADOW_DOM) || ENABLE(DETAILS_ELEMENT)
 
+#include <functional>
 #include <wtf/HashMap.h>
 #include <wtf/HashSet.h>
 #include <wtf/Vector.h>
