--- a/Source/JavaScriptCore/assembler/MacroAssemblerMIPS.h	2019-08-15 20:56:14.000000000 +0200
+++ b/Source/JavaScriptCore/assembler/MacroAssemblerMIPS.h	2020-03-04 10:43:29.468482593 +0100
@@ -315,11 +315,7 @@
 
     void countLeadingZeros32(RegisterID src, RegisterID dest)
     {
-#if WTF_MIPS_ISA_AT_LEAST(32)
         m_assembler.clz(dest, src);
-#else
-        static_assert(false, "CLZ opcode is not available for this ISA");
-#endif
     }
 
     void lshift32(RegisterID shiftAmount, RegisterID dest)
