From 06e4728ba7312efa0fd595e30745e60ce88f1a4f Mon Sep 17 00:00:00 2001
From: Marvin Schmidt <marv@exherbo.org>
Date: Sun, 3 May 2020 18:20:05 +0200
Subject: [PATCH] Build: Fix Python 3 incompatibility

---
 wscript | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git wscript wscript
index aefde70..e09fa78 100644
--- wscript
+++ wscript
@@ -20,7 +20,7 @@ FLAVORS = {
     'x11-gl' : 'glmark2',
     'x11-glesv2' : 'glmark2-es2',
 }
-FLAVORS_STR = ", ".join(sorted(FLAVORS.keys() + ['all-linux', 'all-win32']))
+FLAVORS_STR = ", ".join(sorted(list(FLAVORS) + ['all-linux', 'all-win32']))
 
 def linux_flavors():
     return [f for f in FLAVORS.keys() if not f.startswith('win32')]
