--- a/gio/tests/meson.build
+++ b/gio/tests/meson.build
@@ -56,7 +56,6 @@ gio_tests = {
     # FIXME: https://gitlab.gnome.org/GNOME/glib/-/issues/1392 / https://gitlab.gnome.org/GNOME/glib/-/issues/1251
     'should_fail' : host_system == 'darwin',
   },
-  'converter-stream' : {},
   'credentials' : {},
   'cxx' : {
     'source' : ['cxx.cpp'],
@@ -91,7 +90,6 @@ gio_tests = {
   'network-monitor' : {},
   'network-monitor-race' : {},
   'permission' : {},
-  'pollable' : {'dependencies' : [libdl_dep]},
   'power-profile-monitor' : {},
   'proxy-test' : {},
   'readwrite' : {},
@@ -174,7 +172,6 @@ endif
 #  Test programs buildable on UNIX only
 if host_machine.system() != 'windows'
   gio_tests += {
-    'file' : {},
     'gdbus-peer' : {
       'dependencies' : [libgdbus_example_objectmanager_dep],
       'install_rpath' : installed_tests_execdir,
@@ -798,14 +795,6 @@ if not meson.is_cross_build() or meson.has_exe_wrapper()
       test_resources_binary2,
     ]
   endif
-
-  gio_tests += {
-    'resources' : {
-      'extra_sources' : resources_extra_sources,
-      # FIXME: https://gitlab.gnome.org/GNOME/glib/-/issues/1392
-      'should_fail' : host_system == 'darwin',
-    },
-  }
 endif
 
 foreach test_name, extra_args : gio_tests
--- a/glib/tests/meson.build
+++ b/glib/tests/meson.build
@@ -11,25 +11,17 @@ glib_tests = {
   'cache' : {},
   'charset' : {},
   'checksum' : {},
-  'collate' : {},
   'completion' : {},
   'cond' : {},
-  'convert' : {},
   'cxx' : {
     'source' : ['cxx.cpp'],
   },
   'dataset' : {},
-  'date' : {
-    # FIXME: https://gitlab.gnome.org/GNOME/glib/-/issues/1392
-    'should_fail' : host_system == 'darwin',
-  },
   'dir' : {},
   'environment' : {
     # FIXME: https://gitlab.gnome.org/GNOME/glib/-/issues/1392
     'should_fail' : host_system == 'darwin',
   },
-  'error' : {},
-  'fileutils' : {},
   'gdatetime' : {
     'suite' : ['slow'],
   },
@@ -65,7 +57,6 @@ glib_tests = {
   'mutex' : {},
   'node' : {},
   'once' : {},
-  'option-context' : {},
   'option-argv0' : {},
   'overflow' : {},
   'overflow-fallback' : {
