diff --git gio/tests/meson.build gio/tests/meson.build
index a926ae01a..c2093cf4e 100644
--- gio/tests/meson.build
+++ gio/tests/meson.build
@@ -50,7 +50,6 @@ gio_tests = {
   'cancellable' : {},
   'contexts' : {},
   'contenttype' : {},
-  'converter-stream' : {},
   'credentials' : {},
   'data-input-stream' : {},
   'data-output-stream' : {},
@@ -74,7 +73,6 @@ gio_tests = {
   'network-monitor' : {},
   'network-monitor-race' : {},
   'permission' : {},
-  'pollable' : {'dependencies' : [libdl_dep]},
   'proxy-test' : {},
   'readwrite' : {},
   'simple-async-result' : {},
@@ -567,7 +565,7 @@ if installed_tests_enabled
   endforeach
 endif
 
-if not meson.is_cross_build() or meson.has_exe_wrapper()
+if false
 
   plugin_resources_c = custom_target('plugin-resources.c',
     input : 'test4.gresource.xml',
diff --git glib/tests/meson.build glib/tests/meson.build
index c77ccdd14..b0dab692a 100644
--- glib/tests/meson.build
+++ glib/tests/meson.build
@@ -11,21 +11,13 @@ glib_tests = {
   'cache' : {},
   'charset' : {},
   'checksum' : {},
-  'collate' : {},
   'cond' : {},
-  'convert' : {},
   'cxx' : {
     'source' : ['cxx.cpp'],
   },
   'dataset' : {},
-  'date' : {},
   'dir' : {},
   'environment' : {},
-  'error' : {},
-  'fileutils' : {},
-  'gdatetime' : {
-    'suite' : ['slow'],
-  },
   'guuid' : {},
   'gvariant' : {
     'suite' : ['slow'],
@@ -57,7 +49,6 @@ glib_tests = {
   'mutex' : {},
   'node' : {},
   'once' : {},
-  'option-context' : {},
   'option-argv0' : {},
   'overflow' : {},
   'overflow-fallback' : {
-- 
2.31.0

