--- a/gio/tests/meson.build
+++ b/gio/tests/meson.build
@@ -63,7 +63,6 @@ gio_tests = {
     # FIXME: https://gitlab.gnome.org/GNOME/glib/-/issues/1392 / https://gitlab.gnome.org/GNOME/glib/-/issues/1251
     'can_fail' : host_system == 'darwin',
   },
-  'converter-stream' : {},
   'credentials' : {},
   'data-input-stream' : {},
   'data-output-stream' : {},
@@ -227,7 +226,6 @@ endif
 #  Test programs buildable on UNIX only
 if host_machine.system() != 'windows'
   gio_tests += {
-    'file' : {},
     'gdbus-peer-object-manager' : {},
     'gdbus-sasl' : {},
     'live-g-file' : {},
@@ -951,13 +949,6 @@ if not meson.is_cross_build()
       test_resources_binary2,
     ]
   endif
-
-  gio_tests += {
-    'resources' : {
-      'extra_sources' : resources_extra_sources,
-      'depends' : resource_plugin,
-    },
-  }
 endif
 
 test_extra_programs_targets = {}
diff --git a/glib/tests/meson.build b/glib/tests/meson.build
index d74617823..1de81a825 100644
--- a/glib/tests/meson.build
+++ b/glib/tests/meson.build
@@ -12,22 +12,14 @@ glib_tests = {
   'cache' : {},
   'charset' : {},
   'checksum' : {},
-  'collate' : {},
   'completion' : {},
   'cond' : {},
-  'convert' : {},
   'dataset' : {},
-  'date' : {
-    # FIXME: https://gitlab.gnome.org/GNOME/glib/-/issues/1392
-    'can_fail' : host_system == 'darwin',
-  },
   'dir' : {},
   'environment' : {
     # FIXME: https://gitlab.gnome.org/GNOME/glib/-/issues/1392
     'can_fail' : host_system == 'darwin',
   },
-  'error' : {},
-  'fileutils' : {},
   'gdatetime' : {
     'suite' : ['slow'],
     'can_fail' : host_system == 'windows',
@@ -70,7 +62,6 @@ glib_tests = {
   'node' : {},
   'once' : {},
   'onceinit' : {},
-  'option-context' : {},
   'option-argv0' : {},
   'overflow' : {},
   'overflow-fallback' : {
