From 15437a7e5a89dcb2a812f829bc8e9e1db7ce350b Mon Sep 17 00:00:00 2001
From: Sam Thursfield <sam.thursfield@codethink.co.uk>
Date: Thu, 01 Dec 2011 15:48:27 +0000
Subject: Link to libpcre correctly when static linking

https://bugzilla.gnome.org/show_bug.cgi?id=665298
---
diff --git a/glib-2.0.pc.in b/glib-2.0.pc.in
index ec4256d..2e638f6 100644
--- a/glib-2.0.pc.in
+++ b/glib-2.0.pc.in
@@ -10,6 +10,7 @@ glib_mkenums=glib-mkenums
 Name: GLib
 Description: C Utility Library
 Version: @VERSION@
+Requires.private: @PCRE_REQUIRES@
 Libs: -L${libdir} -lglib-2.0 @INTLLIBS@
 Libs.private: @ICONV_LIBS@ @GLIB_RT_LIBS@
 Cflags: -I${includedir}/glib-2.0 -I${libdir}/glib-2.0/include @GLIB_EXTRA_CFLAGS@
--
cgit v0.9.0.2
