--- src/backtrace.c.orig	2017-08-05 16:57:00.000000000 +0000
+++ src/backtrace.c	2017-08-05 16:57:21.552276793 +0000
@@ -42,7 +42,7 @@
 #define MAX_BTSIZE 64
 
 void backtrace_handler(int n, siginfo_t *ist, void *extra) {
-  static struct ucontext *puc;
+  static ucontext_t *puc;
   static void *btinfo[MAX_BTSIZE];
   static char **messages = NULL;
   static size_t btsize = 0;
@@ -58,7 +58,7 @@
 
 #define SIC_CASE(c) case c: strerr = #c
 
-  puc = (struct ucontext *)extra;
+  puc = (ucontext_t *)extra;
   switch(n) {
   case SIGSEGV:
     switch(ist->si_code) {
