--- Makefile.orig	2020-05-05 18:42:58.044009729 -0400
+++ Makefile	2020-05-05 18:44:57.613239379 -0400
@@ -75,6 +75,14 @@
 .PHONY: distclean clean clean-doc doc dist install uninstall
 .PATH: src
 
+.ifdef CFLAGS_COMMON
+CFLAGS += ${CFLAGS_COMMON}
+.endif
+
+.ifdef LDFLAGS_COMMON
+LDFLAGS += ${LDFLAGS_COMMON}
+.endif
+
 .ifdef DEBUG
 CFLAGS += -g -O0 -fsanitize=address
 .else
@@ -160,7 +168,7 @@
 	echo "#define HIKARI_VERSION \"${VERSION}\"" >> version.h
 
 hikari: version.h xdg-shell-protocol.h wlr-layer-shell-unstable-v1-protocol.h ${OBJS}
-	${CC} ${LDFLAGS} ${CFLAGS} ${INCLUDES} ${LIBS} ${OBJS} -o ${.TARGET}
+	${CC} ${LDFLAGS} ${CFLAGS} ${INCLUDES} -o ${.TARGET} ${OBJS} ${LIBS}
 
 xdg-shell-protocol.h:
 	wayland-scanner server-header ${WAYLAND_PROTOCOLS}/stable/xdg-shell/xdg-shell.xml ${.TARGET}
@@ -169,7 +177,7 @@
 	wayland-scanner server-header protocol/wlr-layer-shell-unstable-v1.xml ${.TARGET}
 
 hikari-unlocker: hikari_unlocker.c
-	${CC} -lpam hikari_unlocker.c -o hikari-unlocker
+	${CC} ${LDFLAGS} ${CFLAGS} -o hikari-unlocker hikari_unlocker.c -lpam
 
 clean-doc:
 	@test -e _darcs && echo "cleaning manpage" ||:
@@ -216,14 +224,14 @@
 dist: distclean hikari-${VERSION}.tar.gz
 
 install: hikari hikari-unlocker share/man/man1/hikari.1
-	mkdir -p ${PREFIX}/bin
-	mkdir -p ${PREFIX}/share/man/man1
-	mkdir -p ${PREFIX}/share/examples/hikari
-	mkdir -p ${ETC_PREFIX}/pam.d
-	install -m 4555 -g ${INSTALL_GROUP} hikari hikari-unlocker ${PREFIX}/bin
-	install -m 644 -g ${INSTALL_GROUP} share/man/man1/hikari.1 ${PREFIX}/share/man/man1
-	install -m 644 -g ${INSTALL_GROUP} share/examples/hikari/hikari.conf ${PREFIX}/share/examples/hikari
-	install -m 644 -g ${INSTALL_GROUP} pam.d/hikari-unlocker.${OS} ${ETC_PREFIX}/pam.d/hikari-unlocker
+	mkdir -p ${DESTDIR}/${PREFIX}/bin
+	mkdir -p ${DESTDIR}/${PREFIX}/share/man/man1
+	mkdir -p ${DESTDIR}/${PREFIX}/share/examples/hikari
+	mkdir -p ${DESTDIR}/${ETC_PREFIX}/pam.d
+	install -m 4555 hikari hikari-unlocker ${DESTDIR}/${PREFIX}/bin
+	install -m 644 share/man/man1/hikari.1 ${DESTDIR}/${PREFIX}/share/man/man1
+	install -m 644 share/examples/hikari/hikari.conf ${DESTDIR}/${PREFIX}/share/examples/hikari
+	install -m 644 pam.d/hikari-unlocker.${OS} ${DESTDIR}/${ETC_PREFIX}/pam.d/hikari-unlocker
 
 uninstall:
 	-rm ${PREFIX}/bin/hikari
