--- a/src/xwayland_view.c	2023-10-02 23:25:53.794255011 -0400
+++ b/src/xwayland_view.c	2023-10-02 23:26:17.173265450 -0400
@@ -304,6 +304,7 @@
   struct wlr_xwayland_surface *xwayland_surface = xwayland_view->surface;
 
   wlr_xwayland_surface_activate(xwayland_surface, active);
+  wlr_xwayland_surface_restack(xwayland_surface, NULL, XCB_STACK_MODE_ABOVE);
   wlr_xwayland_set_seat(hikari_server.xwayland, hikari_server.seat);
 }
 
