From 8491e18e940a71a74874885436691e2443b21844 Mon Sep 17 00:00:00 2001
From: Federico Di Pierro <nierro92@gmail.com>
Date: Mon, 18 Mar 2019 20:24:48 +0100
Subject: [PATCH] Avoid flaot libsystemd/libelogind version. Fixes #70.

---
 CMakeLists.txt | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9d6f173..f6e0c3d 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -34,6 +34,12 @@ set_property(TARGET ${PROJECT_NAME} PROPERTY C_STANDARD 99)
 # Required dependencies
 pkg_check_modules(REQ_LIBS REQUIRED popt gsl libconfig)
 pkg_search_module(LOGIN_LIBS REQUIRED libelogind libsystemd>=221)
+
+# Avoid float versioning for libsystemd/libelogind
+string(REPLACE "." ";" LOGIN_LIBS_VERSION_LIST ${LOGIN_LIBS_VERSION})
+list(GET LOGIN_LIBS_VERSION_LIST 0 LOGIN_LIBS_VERSION_MAJOR)
+message(STATUS "Found lib${LOGIN_LIBS_LIBRARIES} version ${LOGIN_LIBS_VERSION_MAJOR}")
+
 target_link_libraries(${PROJECT_NAME}
                       m
                       ${REQ_LIBS_LIBRARIES}
@@ -44,7 +50,7 @@ target_include_directories(${PROJECT_NAME} PRIVATE
                            "${LOGIN_LIBS_INCLUDE_DIRS}"
 )
 target_compile_definitions(${PROJECT_NAME} PRIVATE
-    -DLIBSYSTEMD_VERSION=${LOGIN_LIBS_VERSION}
+    -DLIBSYSTEMD_VERSION=${LOGIN_LIBS_VERSION_MAJOR}
 )
 list(APPEND COMBINED_LDFLAGS ${REQ_LIBS_LDFLAGS})
 list(APPEND COMBINED_LDFLAGS ${LOGIN_LIBS_LDFLAGS})
