--- a/Source/dooble_application.cc
+++ b/Source/dooble_application.cc
@@ -76,16 +76,13 @@ void dooble_application::install_translator(void)

   if(dooble_settings::setting("language_index").toInt() == 1) // System
     {
-      QString path("");
+      QString path("/usr/share/translations/");
       auto name(QLocale::system().name());
       auto variable(qgetenv("DOOBLE_TRANSLATIONS_PATH").trimmed());

       if(!variable.isEmpty())
 	path = QString::fromLocal8Bit(variable.constData());

-      if(path.isEmpty())
-	path = QDir::currentPath() + QDir::separator() + "Translations";
-
       m_translator = new QTranslator(this);

       if(m_translator->load("dooble_" + name, path))