diff -ur a/tests/test_cli.py b/tests/test_cli.py
--- a/tests/test_cli.py	2025-12-01 09:46:31.000000000 -0500
+++ b/tests/test_cli.py	2026-01-03 14:31:57.560834056 -0500
@@ -6,9 +6,6 @@
 import sys
 from pathlib import Path
 
-BINDIR = Path(sys.executable).parent
-
-
 def test_module_run():
     """Simple test for no error when running the module. Output is not validated."""
     subprocess.check_call([sys.executable, "-m", "trove_classifiers"])
@@ -16,7 +13,7 @@
 
 def test_entry_point():
     """Simple test for no error when calling the entry point. Output is not validated."""
-    subprocess.check_call(f"{BINDIR}/trove-classifiers")
+    subprocess.check_call("trove-classifiers")
 
 
 def test_module_run_is_entry_point():
@@ -27,6 +24,6 @@
         encoding="utf-8",
     )
     entry_point_proc = subprocess.run(
-        f"{BINDIR}/trove-classifiers", capture_output=True, encoding="utf-8"
+        "trove-classifiers", capture_output=True, encoding="utf-8"
     )
     assert module_run_proc.stdout == entry_point_proc.stdout
