--- macros2/gnome-autogen.sh.orig	2012-05-31 14:42:38.034589137 +0200
+++ macros2/gnome-autogen.sh	2012-05-31 14:44:29.863881889 +0200
@@ -329,7 +329,7 @@ case $REQUIRED_AUTOMAKE_VERSION in
     1.6*) automake_progs="automake-1.11 automake-1.10 automake-1.9 automake-1.8 automake-1.7 automake-1.6" ;;
     1.7*) automake_progs="automake-1.11 automake-1.10 automake-1.9 automake-1.8 automake-1.7" ;;
     1.8*) automake_progs="automake-1.11 automake-1.10 automake-1.9 automake-1.8" ;;
-    1.9*) automake_progs="automake-1.11 automake-1.10 automake-1.9" ;;
+    1.9*) automake_progs="automake-1.12 automake-1.11 automake-1.10 automake-1.9" ;;
     1.10*) automake_progs="automake-1.11 automake-1.10" ;;
     1.11*) automake_progs="automake-1.11" ;;
 esac
