--- macros2/gnome-autogen.sh.orig	2013-01-07 10:21:58.723717192 +0100
+++ macros2/gnome-autogen.sh	2013-01-07 10:23:29.024673629 +0100
@@ -350,10 +350,11 @@ case $REQUIRED_AUTOMAKE_VERSION in
     1.6*) automake_progs="automake-1.12 automake-1.11 automake-1.10 automake-1.9 automake-1.8 automake-1.7 automake-1.6" ;;
     1.7*) automake_progs="automake-1.12 automake-1.11 automake-1.10 automake-1.9 automake-1.8 automake-1.7" ;;
     1.8*) automake_progs="automake-1.12 automake-1.11 automake-1.10 automake-1.9 automake-1.8" ;;
-    1.9*) automake_progs="automake-1.12 automake-1.11 automake-1.10 automake-1.9" ;;
+    1.9*) automake_progs="automake-1.13 utomake-1.12 automake-1.11 automake-1.10 automake-1.9" ;;
     1.10*) automake_progs="automake-1.12 automake-1.11 automake-1.10" ;;
     1.11*) automake_progs="automake-1.12 automake-1.11" ;;
     1.12*) automake_progs="automake-1.12" ;;
+    1.13*) automake_progs="automake-1.13" ;;
 esac
 version_check automake AUTOMAKE "$automake_progs" $REQUIRED_AUTOMAKE_VERSION \
     "http://ftp.gnu.org/pub/gnu/automake/automake-$REQUIRED_AUTOMAKE_VERSION.tar.gz"
