--- a/makefile_linuxLP64_utf8_ffi	2016-04-10 20:27:16.244225061 +0200
+++ b/makefile_linuxLP64_utf8_ffi	2016-04-10 20:27:57.323104312 +0200
@@ -6,19 +6,19 @@
 OBJS = newlisp.o nl-symbol.o nl-math.o nl-list.o nl-liststr.o nl-string.o nl-filesys.o \
 	nl-sock.o nl-import.o nl-xml-json.o nl-web.o nl-matrix.o nl-debug.o nl-utf8.o pcre.o
 
-CFLAGS = -fPIC -m64 -Wall -Wno-uninitialized -Wno-strict-aliasing -Wno-long-long -c -O2 -g -DREADLINE -DSUPPORT_UTF8 -DNEWLISP64 -DLINUX -DFFI -I/usr/local/lib/libffi-3.0.13/include 
+CFLAGS += -fPIC -Wall -Wno-uninitialized -Wno-strict-aliasing -Wno-long-long -c -g -DREADLINE -DSUPPORT_UTF8 -DNEWLISP64 -DLINUX -DFFI
 
 # replace -O2 with -Oz when using clang/llvm
 #CC = clang 
 
-CC = gcc
+CC ?= gcc
 
 default: $(OBJS)
 	$(CC) $(OBJS) -m64 -g -lm -ldl -lreadline -lffi -o newlisp # for UBUNTU Debian
 #	$(CC) $(OBJS) -m64 -g -lm -ldl -lreadline -ltermcap -o newlisp # slackware
 #	$(CC) $(OBJS) -m64 -g -lm -ldl -lreadline -lncurses -o newlisp # other Linux Dist
 #	$(CC) $(OBJS) -m64 -g -lm -ldl -o newlisp # without readline support
-	strip newlisp
+#	strip newlisp
 
 .c.o:
 	$(CC) $(CFLAGS) $<
--- a/makefile_linuxLP64_utf8_ffi	2016-04-10 20:30:01.588730930 +0200
+++ b/makefile_linuxLP64_utf8_ffi	2016-04-10 20:30:33.211641988 +0200
@@ -14,7 +14,7 @@ CFLAGS += -fPIC -Wall -Wno-uninitialized
 CC ?= gcc
 
 default: $(OBJS)
-	$(CC) $(OBJS) -m64 -g -lm -ldl -lreadline -lffi -o newlisp # for UBUNTU Debian
+	$(CC) $(OBJS) $(LDFLAGS) -lm -ldl -lreadline -lffi -o newlisp # for UBUNTU Debian
 #	$(CC) $(OBJS) -m64 -g -lm -ldl -lreadline -ltermcap -o newlisp # slackware
 #	$(CC) $(OBJS) -m64 -g -lm -ldl -lreadline -lncurses -o newlisp # other Linux Dist
 #	$(CC) $(OBJS) -m64 -g -lm -ldl -o newlisp # without readline support
