--- makefile_linuxLP64_utf8_ffi.orig	2013-03-21 01:03:17.261250308 +0100
+++ makefile_linuxLP64_utf8_ffi	2013-03-21 01:03:59.860419804 +0100
@@ -6,16 +6,15 @@
 OBJS = newlisp.o nl-symbol.o nl-math.o nl-list.o nl-liststr.o nl-string.o nl-filesys.o \
 	nl-sock.o nl-import.o nl-xml.o nl-web.o nl-matrix.o nl-debug.o nl-utf8.o pcre.o
 
-CFLAGS = -fPIC -m64 -Wall -pedantic -Wno-uninitialized -Wno-strict-aliasing -Wno-long-long -c -O2 -g -DREADLINE -DSUPPORT_UTF8 -DNEWLISP64 -DLINUX -DFFI -I/usr/lib/x86_64-linux-gnu
+CFLAGS += -fPIC -Wall -pedantic -Wno-uninitialized -Wno-strict-aliasing -Wno-long-long -c -DREADLINE -DSUPPORT_UTF8 -DNEWLISP64 -DLINUX -DFFI
 
-CC = gcc
+CC ?= gcc
 
 default: $(OBJS)
-	$(CC) $(OBJS) -m64 -g -lm -ldl -lreadline -lffi -o newlisp # for UBUNTU Debian
+	$(CC) $(OBJS) -lm -ldl -lreadline -lffi -o newlisp # for UBUNTU Debian
 #	$(CC) $(OBJS) -m64 -g -lm -ldl -lreadline -ltermcap -o newlisp # slackware
 #	$(CC) $(OBJS) -m64 -g -lm -ldl -lreadline -lncurses -o newlisp # other Linux Dist
 #	$(CC) $(OBJS) -m64 -g -lm -ldl -o newlisp # without readline support
-	strip newlisp
 
 .c.o:
 	$(CC) $(CFLAGS) $<
