diff -Nuar a/makefile_original_install b/makefile_original_install
--- a/makefile_original_install	2015-01-19 18:10:12.000000000 +0100
+++ b/makefile_original_install	2015-06-22 12:59:33.639980437 +0200
@@ -31,7 +31,9 @@
 	-install -d $(datadir)/newlisp/modules
 	-install -d $(datadir)/newlisp/util
 	-install -d $(datadir)/doc/newlisp
-	-rm $(bindir)/newlisp
+	-install -d $(datadir)/man
+	-install -d $(datadir)/man/man1
+	-install -d $(bindir)
 	-install -m 755  newlisp $(bindir)/newlisp-$(VERSION)
 	-ln -s $(bindir)/newlisp-$(VERSION) $(bindir)/newlisp
 	-install -m 755 util/newlispdoc $(bindir)/newlispdoc
