--- a/cli/build.clj
+++ b/cli/build.clj
@@ -134,8 +134,6 @@
                         "--features=clj_easy.graal_build_time.InitClojureClasses"
                         (when-not (fs/windows?) "-march=compatibility")
                         "-O1"
-                        (when-not (or (:pgo-instrument opts)
-                                      (fs/windows?)) "--pgo=graalvm/default.iprof")
                         (or (System/getenv "CLOJURE_LSP_XMX")
                             "-J-Xmx8g")
                         (when (= "true" (System/getenv "CLOJURE_LSP_STATIC"))
