--- m4/gnupg-pth.m4.orig
+++ m4/gnupg-pth.m4
@@ -51,9 +51,9 @@ AC_DEFUN([GNUPG_PTH_VERSION_CHECK],
          _gnupg_pth_save_cflags=$CFLAGS
          _gnupg_pth_save_ldflags=$LDFLAGS
          _gnupg_pth_save_libs=$LIBS
-         CFLAGS="$CFLAGS `$PTH_CONFIG --cflags`"
-         LDFLAGS="$LDFLAGS `$PTH_CONFIG --ldflags`"
-         LIBS="$LIBS `$PTH_CONFIG --libs --all`"
+         CFLAGS=
+         LDFLAGS=
+         LIBS=-lpth
          AC_LINK_IFELSE([AC_LANG_PROGRAM([#include <pth.h>
                                          ],
                                          [[ pth_init ();]])],
@@ -92,9 +92,8 @@ AC_DEFUN([GNUPG_PATH_PTH],
   if test "$PTH_CONFIG" != "no"; then
     GNUPG_PTH_VERSION_CHECK($tmp)
     if test $have_pth = yes; then
-       PTH_CFLAGS=`$PTH_CONFIG --cflags`
-       PTH_LIBS=`$PTH_CONFIG --ldflags`
-       PTH_LIBS="$PTH_LIBS `$PTH_CONFIG --libs --all`"
+       PTH_CFLAGS=
+       PTH_LIBS=-lpth
        AC_DEFINE(HAVE_PTH, 1,
                 [Defined if the GNU Pth is available])
     fi
