--- src/libs/tgfclient/guifont.h.orig
+++ src/libs/tgfclient/guifont.h
@@ -30,7 +30,7 @@
 //glFont structure (DO NOT MODIFY)
 typedef struct
 {
-    uint Tex;
+    unsigned int Tex;
     int TexWidth, TexHeight;
     int IntStart, IntEnd;
     GLFONTCHAR *Char;
--- src/libs/tgfclient/guifont.cpp.orig
+++ src/libs/tgfclient/guifont.cpp
@@ -106,7 +106,7 @@
 	FILE *Input;
 	char *TexBytes;
 	int	Num;
-	uint Tex;
+	unsigned int Tex;
 
 	font = NULL;
 	size = 8.0;
--- src/libs/tgfclient/guiedit.cpp.orig
+++ src/libs/tgfclient/guiedit.cpp
@@ -219,7 +219,7 @@
     tGfuiEditbox	*editbox;
     int			relX;
     char		buf[256];
-    uint		i;
+    unsigned int		i;
 
     object = GfuiScreen->hasFocus;
     if (object->state == GFUI_DISABLE) {
--- src/modules/simu/simuv2/SOLID-2.0/src/C-api.cpp.orig
+++ src/modules/simu/simuv2/SOLID-2.0/src/C-api.cpp
@@ -125,8 +125,8 @@
   Point p(x, y, z);
   int i = pointBuf.size()-20;
   if (i < 0) i = 0;
-  while ((uint)i < pointBuf.size() && !(pointBuf[i] == p)) ++i;
-  if ((uint)i == pointBuf.size()) pointBuf.push_back(p);
+  while ((unsigned int)i < pointBuf.size() && !(pointBuf[i] == p)) ++i;
+  if ((unsigned int)i == pointBuf.size()) pointBuf.push_back(p);
   indexBuf.push_back(i);
 }
 
@@ -162,7 +162,7 @@
 
 void dtVertexRange(DtPolyType type, DtIndex first, DtCount count) {
   DtIndex *indices = new DtIndex[count];
-  for (uint i = 0; i < count; ++i) indices[i] = first + i;
+  for (unsigned int i = 0; i < count; ++i) indices[i] = first + i;
   dtVertexIndices(type, count, indices);
   delete [] indices;
 }
