diff --git a/autotests/CMakeLists.txt b/autotests/CMakeLists.txt
index f2c7287..460e741 100644
--- a/autotests/CMakeLists.txt
+++ b/autotests/CMakeLists.txt
@@ -17,7 +17,6 @@ endfunction()
 quotient_add_test(NAME callcandidateseventtest)
 quotient_add_test(NAME utiltests)
 if(${PROJECT_NAME}_ENABLE_E2EE)
-    quotient_add_test(NAME testolmaccount)
     quotient_add_test(NAME testgroupsession)
     quotient_add_test(NAME testolmsession)
     if (NOT MSVC)
