Fix build with cups>=2.2

--- prnt/hpcups/CommonDefinitions.h.orig
+++ prnt/hpcups/CommonDefinitions.h
@@ -40,6 +40,7 @@
 #include <fcntl.h>
 #include <sys/stat.h>
 #include <cups/cups.h>
+#include <cups/ppd.h>
 #include <cups/raster.h>
 #include <stdarg.h>
 #include <stdio.h>
--- prnt/hpijs/hpcupsfax.cpp.orig
+++ prnt/hpijs/hpcupsfax.cpp
@@ -43,6 +43,7 @@
 #include <arpa/inet.h>
 #include <math.h>
 #include <cups/cups.h>
+#include <cups/ppd.h>
 #include <cups/raster.h>
 #include <string>
 #ifdef FALSE
