Fix error caused by removed URLopener

--- a/base/device.py
+++ b/base/device.py
@@ -2525,9 +2525,9 @@
                 data = None
 
             log.debug("Opening: %s" % url2)
-            opener = LocalOpener({})
+            opener = LocalOpener()
             try:
-                f = opener.open(url2, data)
+                f = opener.open_hp(url2, data)
                 
             except Error:
                 log.error("Status read failed: %s" % url2)
@@ -2547,7 +2547,7 @@
         try:
             url2 = "%s&loc=%s" % (self.device_uri.replace('hpfax:', 'hp:'), url)
             data = self
-            opener = LocalOpenerEWS_LEDM({})
+            opener = LocalOpenerEWS_LEDM()
             try:
                 if footer:
                     return opener.open_hp(url2, data, footer)
@@ -2562,7 +2562,7 @@
         try:
             url2 = "%s&loc=%s" % (self.device_uri.replace('hpfax:', 'hp:'), url)
             data = self
-            opener = LocalOpener_LEDM({})
+            opener = LocalOpener_LEDM()
             try:
                 if footer:
                     return opener.open_hp(url2, data, footer)
@@ -2579,7 +2579,7 @@
             url="/cdm/supply/v1/suppliesPublic"
             url2 = "%s&loc=%s" % (self.device_uri.replace('hpfax:', 'hp:'), url)
             data = self
-            opener = LocalOpener_CDM({})
+            opener = LocalOpener_CDM()
             try:
                 if footer:
                     return opener.open_hp(url2, data, footer)
@@ -2687,7 +2687,7 @@
 
 
 # URLs: hp:/usb/HP_LaserJet_3050?serial=00XXXXXXXXXX&loc=/hp/device/info_device_status.xml
-class LocalOpener(urllib_request.URLopener):
+class LocalOpener():
     def open_hp(self, url, dev):
         log.debug("open_hp(%s)" % url)
 
@@ -2717,7 +2717,7 @@
             return response#.fp
 
 # URLs: hp:/usb/HP_OfficeJet_7500?serial=00XXXXXXXXXX&loc=/hp/device/info_device_status.xml
-class LocalOpenerEWS_LEDM(urllib_request.URLopener):
+class LocalOpenerEWS_LEDM():
     def open_hp(self, url, dev, foot=""):
         log.debug("open_hp(%s)" % url)
 
@@ -2743,7 +2743,7 @@
 
 
 # URLs: hp:/usb/HP_OfficeJet_7500?serial=00XXXXXXXXXX&loc=/hp/device/info_device_status.xml
-class LocalOpener_LEDM(urllib_request.URLopener):
+class LocalOpener_LEDM():
     def open_hp(self, url, dev, foot=""):
         log.debug("open_hp(%s)" % url)
 
@@ -2770,7 +2770,7 @@
 
 
 # URLs: hp:/usb/HP_OfficeJet_7500?serial=00XXXXXXXXXX&loc=/hp/device/info_device_status.xml
-class LocalOpener_CDM(urllib_request.URLopener):
+class LocalOpener_CDM():
     def open_hp(self, url, dev, foot=""):
         log.debug("open_hp(%s)" % url)
         match_obj = http_pat_url.search(url)
