--- a/scan/sane/OrbliteScan/LinuxCommon.h	2018-10-10 22:20:30.798911195 +0200
+++ b/scan/sane/OrbliteScan/LinuxCommon.h	2018-10-10 22:21:28.815911201 +0200
@@ -3,6 +3,17 @@
 
 #include <sys/types.h>
 
+#ifndef __GLIBC__
+#include <bits/reg.h>
+#if __WORDSIZE == 32
+# define __S64_TYPE             long long int
+# define __U64_TYPE             unsigned long long int
+#elif __WORDSIZE == 64
+# define __S64_TYPE             long int
+# define __U64_TYPE             unsigned long int
+#endif
+#endif
+
 /* Common typedefs for Linux */
 
 typedef unsigned char *			StringPtr;
