diff --git a/Makefile b/Makefile
index 8297904..88a1c51 100644
--- a/Makefile
+++ b/Makefile
@@ -1,9 +1,9 @@
 CC = gcc
-LDFLAGS = -lImlib2 -lX11
+override LDFLAGS += -lImlib2 -lX11
 CFLAGS ?= -O2 -s
 #CFLAGS ?= -ggdb -g3 # For debugging
 
-all: xlunch entries.dsv
+all: xlunch # entries.dsv
 
 install: xlunch
 	mkdir -p $(DESTDIR)/etc/xlunch/
@@ -20,7 +20,7 @@ install: xlunch
 	cp extra/genentries.desktop.sh $(DESTDIR)/usr/bin
 	cp extra/genentries.desktop $(DESTDIR)/usr/share/applications/
 	cp default.conf $(DESTDIR)/etc/xlunch/default.conf
-	bash extra/genentries --path $(DESTDIR)/usr/share/xlunch/svgicons/ > $(DESTDIR)/etc/xlunch/entries.dsv
+	# bash extra/genentries --path $(DESTDIR)/usr/share/xlunch/svgicons/ > $(DESTDIR)/etc/xlunch/entries.dsv
 	cp -r svgicons/ $(DESTDIR)/usr/share/xlunch/ 2>/dev/null || :
 
 remove:
diff --git a/xlunch.c b/xlunch.c
index e613d37..50eed83 100644
--- a/xlunch.c
+++ b/xlunch.c
@@ -39,6 +39,9 @@ const int VERSION_PATCH = 5; // Patch version, changes when something is changed
 #include <sys/poll.h>
 #include <errno.h>
 
+/* file I/O stuff */
+#include <fcntl.h>
+
 /* some globals for our window & X display */
 Display *disp;
 Window   win;
