--- src/magellan.c.orig	2009-03-31 10:19:38.660028123 +0200
+++ src/magellan.c	2009-03-31 10:19:58.179207305 +0200
@@ -364,7 +364,7 @@ DeviceInit (DeviceIntPtr dev)
       return !Success;
     }
 
-  if (InitValuatorClassDeviceStruct (dev, 6, xf86GetMotionEvents,
+  if (InitValuatorClassDeviceStruct (dev, 6,
 				     pInfo->history_size, Absolute) == FALSE)
     {
       ErrorF ("Unable to allocate MAGELLAN ValuatorClassDeviceStruct\n");
