From 6e9805304ee436adb9d0363bd51d7cecf59bceb2 Mon Sep 17 00:00:00 2001
From: Eitan Isaacson <eitan@monotonous.org>
Date: Mon, 13 Jan 2025 12:04:16 -0800
Subject: [PATCH] Add soname versioning. Fixes #2

---
 libspeechprovider/meson.build | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libspeechprovider/meson.build b/libspeechprovider/meson.build
index e7294e1..39f804f 100644
--- a/libspeechprovider/meson.build
+++ b/libspeechprovider/meson.build
@@ -52,6 +52,7 @@ speech_provider_deps = [
 speech_provider_lib = shared_library('speech-provider-' + api_version,
   [speech_provider_sources, speech_provider_lib_generated],
   dependencies: speech_provider_deps,
+  version: meson.project_version(),
   install: true,
 )
 
