--- media/audio/BUILD.gn	2020-03-24 10:16:30.000000000 +0100
+++ -	2020-04-06 14:31:28.871450217 +0200
@@ -232,9 +232,19 @@
     deps += [ "//media/base/android:media_jni_headers" ]
   }
 
-  if (is_linux) {
+  if (is_linux) {
     sources += [ "linux/audio_manager_linux.cc" ]
   }
+  if (use_sndio) {
+    libs += [ "sndio" ]
+    sources += [
+      "openbsd/audio_manager_openbsd.cc",
+      "sndio/sndio_input.cc",
+      "sndio/sndio_input.h",
+      "sndio/sndio_output.cc",
+      "sndio/sndio_output.h"
+   ]
+ }
 
   if (use_alsa) {
     libs += [ "asound" ]
