--- a/lwip-v6/src/include/lwip/if.h
+++ b/lwip-v6/src/include/lwip/if.h
@@ -183,18 +183,14 @@
 # define _IOT_ifconf _IOT(_IOTS(struct ifconf),1,0,0,0,0) /* not right */
 #endif	/* Misc.  */
 
-__BEGIN_DECLS
-
 /* Convert an interface name to an index, and vice versa.  */
-extern unsigned int if_nametoindex (__const char *__ifname) __THROW;
-extern char *if_indextoname (unsigned int __ifindex, char *__ifname) __THROW;
+extern unsigned int if_nametoindex (__const char *__ifname);
+extern char *if_indextoname (unsigned int __ifindex, char *__ifname);
 
 /* Return a list of all interfaces and their indices.  */
-extern struct if_nameindex *if_nameindex (void) __THROW;
+extern struct if_nameindex *if_nameindex (void);
 
 /* Free the data returned from if_nameindex.  */
-extern void if_freenameindex (struct if_nameindex *__ptr) __THROW;
-
-__END_DECLS
+extern void if_freenameindex (struct if_nameindex *__ptr);
 
 #endif /* net/if.h */
--- a/lwip-v6/src/include/lwip/sockets.h
+++ b/lwip-v6/src/include/lwip/sockets.h
@@ -479,8 +479,6 @@
    this large value the readv/writev call will not fail because of this.  */
 #define UIO_MAXIOV      1024
 
-#define size_t          u32_t
-
 #if 0
 /* Structure for scatter/gather I/O.  */
 struct iovec {
