On big endian ppc64 glibc specifically, this causes ICEs.

--- a/stress-vecmath.c
+++ b/stress-vecmath.c
@@ -53,6 +53,11 @@ static const help_t help[] = {
 #undef HAVE_VECMATH
 #endif
 
+/* currently ICEs on our toolchain... */
+#if defined(STRESS_PPC64) && defined(__GLIBC__) && defined(__BIG_ENDIAN__)
+#undef HAVE_VECMATH
+#endif
+
 #if defined(HAVE_VECMATH)
 
 typedef int8_t  vint8_t  __attribute__ ((vector_size (16)));
