Index: initramfs-tools-0.142/scripts/functions
===================================================================
--- initramfs-tools-0.142.orig/scripts/functions
+++ initramfs-tools-0.142/scripts/functions
@@ -37,6 +37,22 @@ panic()
 {
 	local console rest IFS
 
+	# udev is not yet running, so load keyboard drivers
+	if [ "${quiet}" = "y" ]; then
+		opts="-q"
+	else
+		opts="-v"
+	fi
+	/sbin/modprobe ${opts} -a i8042 atkbd ehci-pci ehci-orion \
+		 ehci-hcd ohci-hcd ohci-pci uhci-hcd usbhid xhci \
+		 xhci-pci xhci-hcd
+	sleep 2
+	for modalias in /sys/bus/hid/devices/*/modalias; do
+		if [ -f "${modalias}" ]; then
+			/sbin/modprobe ${opts} -b "$(cat "${modalias}")"
+		fi
+	done
+
 	if command -v chvt >/dev/null 2>&1; then
 		chvt 1
 	fi
@@ -92,23 +108,6 @@ maybe_break()
 {
 	case ",${break?}," in
 	*,$1,*)
-		if [ "$1" = "top" ]; then
-			# udev is not yet running, so load keyboard drivers
-			if [ "${quiet}" = "y" ]; then
-				opts="-q"
-			else
-				opts="-v"
-			fi
-			/sbin/modprobe ${opts} -a i8042 atkbd ehci-pci ehci-orion \
-				 ehci-hcd ohci-hcd ohci-pci uhci-hcd usbhid xhci \
-				 xhci-pci xhci-hcd
-			sleep 2
-			for modalias in /sys/bus/hid/devices/*/modalias; do
-				if [ -f "${modalias}" ]; then
-					/sbin/modprobe ${opts} -b "$(cat "${modalias}")"
-				fi
-			done
-		fi
 		panic "Spawning shell within the initramfs"
 		;;
 	esac
