--- a/src/components/sde/tests/Makefile
+++ b/src/components/sde/tests/Makefile
@@ -1,7 +1,7 @@
 NAME=sde
 include ../../Makefile_comp_tests.target
 INCLUDE += -I$(datadir)/sde_lib -I..
-ifeq ($(notdir $(F77)),gfortran)
+ifeq ($(findstring gfortran,$(notdir $(F77))),gfortran)
 	FFLAGS +=-ffree-form -ffree-line-length-none
 else ifeq ($(notdir $(F77)),flang)
 	FFLAGS +=-ffree-form
--- a/src/components/sysdetect/tests/Makefile
+++ b/src/components/sysdetect/tests/Makefile
@@ -19,7 +19,7 @@ else
     FTESTS =
 endif
 
-ifeq ($(notdir $(F77)),gfortran)
+ifeq ($(findstring gfortran,$(notdir $(F77))),gfortran)
     FFLAGS +=-ffree-form -ffree-line-length-none
 else ifeq ($(notdir $(F77)),flang)
     FFLAGS +=-ffree-form
