--- a/_build/iolib/src/syscalls/ffi-functions-unix.lisp.orig	2022-02-27 09:44:00.327307802 -0700
+++ b/_build/iolib/src/syscalls/ffi-functions-unix.lisp	2022-02-27 09:48:30.875539336 -0700
@@ -12,7 +12,8 @@
 (eval-when (:compile-toplevel :load-toplevel :execute)
   (define-foreign-library
       (libfixposix :canary "lfp_buildinfo")
-    (t (:default "libfixposix")))
+    (:unix "libfixposix.so.3")
+    (t (:default "libfixposix")))
   (load-foreign-library 'libfixposix))
 
 
