--- a/_build/iolib/src/syscalls/ffi-functions-unix.lisp	2022-03-20 16:37:34.285638580 -0700
+++ b/_build/iolib/src/syscalls/ffi-functions-unix.lisp	2022-03-20 16:45:25.241621564 -0700
@@ -11,7 +11,10 @@
 ;; FIXME: move this into an ASDF operation
 (eval-when (:compile-toplevel :load-toplevel :execute)
   (define-foreign-library
-      (libfixposix :canary "lfp_buildinfo")
+      libfixposix
+       (:unix (:or "libfixposix.so.3.2.0"
+		   "libfixposix.so.3"
+		   "libfixposix.so"))
     (t (:default "libfixposix")))
   (load-foreign-library 'libfixposix))
 
