--- a/_build/cl-webkit/webkit2/webkit2.init.lisp	2022-01-14 03:22:05.000000000 -0700
+++ b/_build/cl-webkit/webkit2/webkit2.init.lisp	2022-02-27 09:59:29.175938024 -0700
@@ -18,9 +18,9 @@
               "libwebkit2gtk-4.0.37.dylib"
               "libwebkit2gtk-4.0.dylib"))
     (:unix (:or "libwebkit2gtk-4.1.so"
-                "libwebkit2gtk-4.0.so"
                 ;; Fedora only has this one?
-                "libwebkit2gtk-4.0.so.37")))
+                "libwebkit2gtk-4.0.so.37"
+                "libwebkit2gtk-4.0.so")))
   (use-foreign-library libwebkit2))
 
 (defcfun "webkit_get_major_version" :int)
