--- a/tools/cpp/src/cpp-build/generate_geocoding_data.cc	2021-12-13 14:43:42.507659848 +0100
+++ -	2021-12-13 14:45:11.191139520 +0100
@@ -567,7 +567,7 @@
   map<int32, set<string> > country_languages;
   vector<DirEntry> entries;
   if (!ListDirectory(data_path, &entries)) {
-    fprintf(stderr, "failed to read directory entries");
+    fprintf(stderr, "failed to read directory entries\n");
     return false;
   }
   for (vector<DirEntry>::const_iterator it = entries.begin();
