--- a/src/libs/utils/qtcassert.cpp
+++ b/src/libs/utils/qtcassert.cpp
@@ -11,7 +11,9 @@
 #if defined(Q_OS_UNIX)
 #include <stdio.h>
 #include <signal.h>
+#ifdef __GLIBC__
 #include <execinfo.h>
+#endif
 #elif defined(_MSC_VER)
 #ifdef QTCREATOR_PCH_H
 #define CALLBACK WINAPI
@@ -29,7 +31,7 @@ void dumpBacktrace(int maxdepth)
     const int ArraySize = 1000;
     if (maxdepth < 0 || maxdepth > ArraySize)
         maxdepth = ArraySize;
-#if defined(Q_OS_UNIX)
+#if defined(Q_OS_UNIX) && defined(__GLIBC__)
     void *bt[ArraySize] = {nullptr};
     int size = backtrace(bt, maxdepth);
     char **lines = backtrace_symbols(bt, size);
--- a/src/plugins/qmldesigner/libs/designercore/exceptions/exception.cpp
+++ b/src/plugins/qmldesigner/libs/designercore/exceptions/exception.cpp
@@ -3,7 +3,7 @@
 
 #include "exception.h"
 
-#ifdef Q_OS_LINUX
+#if defined(Q_OS_LINUX) && defined(__GLIBC__)
 #include <execinfo.h>
 #include <cxxabi.h>
 #endif
@@ -30,7 +30,7 @@
 
 namespace QmlDesigner {
 
-#ifdef Q_OS_LINUX
+#if defined(Q_OS_LINUX) && defined(__GLIBC__)
 const char* demangle(const char* name)
 {
    char buf[1024];
@@ -75,7 +75,7 @@ bool Exception::warnAboutException()
     return s_warnAboutException;
 }
 
-#ifdef Q_OS_LINUX
+#if defined(Q_OS_UNIX) && defined(__GLIBC__)
 static QString getBackTrace()
 {
     QString backTrace;
@@ -113,7 +113,7 @@ Exception::Exception(int line, const QByteArray &function,
   , m_function(QString::fromUtf8(function))
   , m_file(QString::fromUtf8(file))
   , m_description(description)
-  #ifdef Q_OS_LINUX
+  #if defined(Q_OS_UNIX) && defined(__GLIBC__)
   , m_backTrace(getBackTrace())
   #endif
 {
