--- src/plugins/clangformat/clangformatutils.cpp
+++ src/plugins/clangformat/clangformatutils.cpp
@@ -60,7 +60,7 @@ static clang::format::FormatStyle qtcStyle()
     style.AllowShortBlocksOnASingleLine = false;
     style.AllowShortCaseLabelsOnASingleLine = false;
     style.AllowShortFunctionsOnASingleLine = FormatStyle::SFS_Inline;
-    style.AllowShortIfStatementsOnASingleLine = false;
+    style.AllowShortIfStatementsOnASingleLine = FormatStyle::SIS_Never;
     style.AllowShortLoopsOnASingleLine = false;
     style.AlwaysBreakAfterReturnType = FormatStyle::RTBS_None;
     style.AlwaysBreakBeforeMultilineStrings = false;
