--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -107,7 +107,7 @@ if (MSVC AND QT_FEATURE_static_runtime)
 endif()
 
 find_package(Qt6 OPTIONAL_COMPONENTS Quick QuickWidgets Designer DesignerComponentsPrivate
-                                     Help SerialPort Svg Tools LinguistTools QUIET)
+                                     Help SerialPort Svg Tools Linguist LinguistTools QUIET)
 
 find_package(Threads)
 find_package(Clang QUIET)
