--- a/src/app/main.cpp	2020-08-25 13:26:21.000000000 +0200
+++ b/src/app/main.cpp	2020-09-03 14:21:38.975668939 +0200
@@ -718,8 +718,10 @@
         {
             m_trimTimer.setSingleShot(true);
             m_trimTimer.setInterval(60000);
+#if defined(__GLIBC__)
             // glibc may not actually free memory in free().
             connect(&m_trimTimer, &QTimer::timeout, this, [] { malloc_trim(0); });
+#endif
         }
 
         bool eventFilter(QObject *, QEvent *e) override
