--- dmd/compiler/src/dmd/link.d
+++ dmd/compiler/src/dmd/link.d
@@ -759,8 +759,7 @@
             perror("unable to create pipe to linker");
             return -1;
         }
-        // vfork instead of fork to avoid https://issues.dlang.org/show_bug.cgi?id=21089
-        childpid = vfork();
+        childpid = fork();
         if (childpid == 0)
         {
             // pipe linker stderr to fds[0]
https://github.com/dlang/dmd/pull/15971
--- dmd/druntime/src/core/sys/posix/net/if_.d
+++ dmd/druntime/src/core/sys/posix/net/if_.d
@@ -143,6 +143,21 @@ else version (CRuntime_Bionic)
     uint            if_nametoindex(const scope char*);
     char*           if_indextoname(uint, char*);
 }
+else version (CRuntime_Musl)
+{
+    struct if_nameindex_t
+    {
+        uint    if_index;
+        char*   if_name;
+    }
+
+    enum IF_NAMESIZE = 16;
+
+    uint            if_nametoindex(const scope char*);
+    char*           if_indextoname(uint, char*);
+    if_nameindex_t* if_nameindex();
+    void            if_freenameindex(if_nameindex_t*);
+}
 else version (CRuntime_UClibc)
 {
     struct if_nameindex_t
