--- a/configure.ac
+++ b/configure.ac
@@ -159,7 +159,7 @@ case "$host" in
   mips-*-linux-*)        ARCH="-D__MIPS__ -D__LINUX__" ;;
   mipsel-*-linux-*)      ARCH="-D__MIPS__ -D__LINUX__" ;;
   powerpc-*-linux-*)     ARCH="-D__PPC__  -D__LINUX__" ;;
-  powerpc64-*-linux-*)   ARCH="-D__PPC__  -D__LINUX__" ;;
+  powerpc64*-*-linux-*)   ARCH="-D__PPC__  -D__LINUX__" ;;
   powerpc-apple-darwin*) ARCH="-D__PPC__  -D__MACOSX__" ;;
   powerpc64-apple-darwin*) ARCH="-D__PPC__  -D__MACOSX__" ;;
   s390*-*-linux-*)       ARCH="-D__LINUX__" ;;
