--- a/src/fs.c	2009-09-14 22:54:54.000000000 +0200
+++ b/src/fs.c.orig	 2017-06-23 22:43:18.862270803 +0200
@@ -29,6 +29,7 @@
     { "ext3", "nodev,noauto,nosuid,user,errors=remount-ro", 0, NULL, 0 },
     { "ext2", "nodev,noauto,nosuid,user,errors=remount-ro", 0, NULL, 0 },
     { "ext4", "nodev,noauto,nosuid,user,errors=remount-ro", 0, NULL, 0 },
+    { "f2fs", "nodev,noauto,nosuid,user", 0, NULL, 0 },
     { "reiserfs", "nodev,noauto,nosuid,user", 0, NULL, 0 },
     { "reiser4", "nodev,noauto,nosuid,user", 0, NULL, 0 },
     { "xfs", "nodev,noauto,nosuid,user", 0, NULL, 0 },

