--- a/src/gs-manager.c
+++ b/src/gs-manager.c
@@ -820,8 +820,7 @@ apply_background_to_window (GSManager *manager,
         surface = gnome_bg_create_surface (manager->priv->bg,
                                            gdk_window,
                                            width,
-                                           height,
-                                           FALSE);
+                                           height);
         gs_window_set_background_surface (window, surface);
         cairo_surface_destroy (surface);
 }
