diff -Naur gnome-screensaver-3.6.1.orig/src/gs-manager.c gnome-screensaver-3.6.1/src/gs-manager.c
--- gnome-screensaver-3.6.1.orig/src/gs-manager.c	2020-03-08 08:53:46.408431578 +0000
+++ gnome-screensaver-3.6.1/src/gs-manager.c	2020-03-08 08:54:31.875738732 +0000
@@ -788,8 +788,7 @@
         surface = gnome_bg_create_surface (manager->priv->bg,
                                            gdk_window,
                                            width,
-                                           height,
-                                           FALSE);
+                                           height);
         gs_window_set_background_surface (window, surface);
         cairo_surface_destroy (surface);
 }
