diff --git src/3rdparty/chromium/third_party/yasm/source/config/linux/config.h src/3rdparty/chromium/third_party/yasm/source/config/linux/config.h
index 9e36539..f588083 100644
--- qtwebengine/src/3rdparty/chromium//third_party/yasm/source/config/linux/config.h
--- qtwebengine/src/3rdparty/chromium//third_party/yasm/source/config/linux/config.h
@@ -5,7 +5,7 @@
 #define CPP_PROG "gcc -E"
 
 /* */
-#define ENABLE_NLS 1
+/* #undef ENABLE_NLS 1 */
 
 /* Define to 1 if you have the `abort' function. */
 #define HAVE_ABORT 1
