--- qtwebengine/src/buildtools/configure.json	2020-09-01 13:02:34.000000000 +0200
+++ -	2020-09-10 16:39:33.309237834 +0200
@@ -269,7 +269,7 @@
             "test": {
                 "include": "features.h",
                 "tail": [
-                    "#if __GLIBC__ < 2 || __GLIBC_MINOR__ < 17",
+                    "#if false",
                     "#error glibc versions below 2.17 are not supported",
                     "#endif"
                 ]
