From 463f1234c57a36e78ff666bd55094a9d4e68f296 Mon Sep 17 00:00:00 2001
From: q66 <daniel@octaforge.org>
Date: Sat, 18 Jan 2020 23:52:55 +0100
Subject: [PATCH 1/3] Enable ppc64 builds

---
 configure.pri                  | 1 +
 mkspecs/features/functions.prf | 1 +
 2 files changed, 2 insertions(+)

diff --git a/configure.pri b/configure.pri
index 3a144e3f8..cabe8b24a 100644
--- a/configure.pri	2021-02-24 10:45:58.000000000 +0100
+++ -	2021-03-07 21:18:43.124754796 +0100
@@ -144,6 +144,7 @@
     contains(QT_ARCH, "arm")|contains(QT_ARCH, "arm64"): return(true)
     contains(QT_ARCH, "mips"): return(true)
     contains(QT_ARCH, "mips64"): return(true)
+    contains(QT_ARCH, "power64"): return(true)
     qtLog("Architecture not supported.")
     return(false)
 }
diff --git a/mkspecs/features/functions.prf b/mkspecs/features/functions.prf
index 512e2523b..e31123654 100644
--- a/mkspecs/features/functions.prf	2021-02-24 10:45:58.000000000 +0100
+++ -	2021-03-07 21:19:21.345036293 +0100
@@ -107,6 +107,7 @@
     contains(qtArch, "mips"): return(mipsel)
     contains(qtArch, "mips64"): return(mips64el)
     contains(qtArch, "mips64el"): return(mips64el)
+    contains(qtArch, "power64"): return(ppc64)
     return(unknown)
 }
 
-- 
2.26.0

