--- a/src/buildtools/config/linux.pri	2020-03-24 10:16:30.000000000 +0100
+++ -	2020-04-06 14:42:50.189729758 +0200
@@ -190,6 +190,11 @@
     } else {
         gn_args += use_alsa=false
     }
+    qtConfig(webengine-sndio) {
+        gn_args += use_sndio=true
+    } else {
+       gn_args += use_sndio=true
+    }
     !packagesExist(libpci): gn_args += use_libpci=false
 
     qtConfig(webengine-ozone-x11) {
