--- a/src/core/configure.json	2020-03-24 10:16:30.000000000 +0100
+++ -	2020-04-06 14:28:00.591236926 +0200
@@ -21,6 +21,7 @@
             "webengine-printing-and-pdf": "boolean",
             "webengine-proprietary-codecs": "boolean",
             "webengine-pulseaudio": "boolean",
+            "webengine-sndio": "boolean",
             "webengine-spellchecker": "boolean",
             "webengine-native-spellchecker": "boolean",
             "webengine-extensions": "boolean",
@@ -31,6 +32,7 @@
             "webengine-kerberos": "boolean",
             "alsa": { "type": "boolean", "name": "webengine-alsa" },
             "pulseaudio": { "type": "boolean", "name": "webengine-pulseaudio" },
+            "sndio": { "type": "boolean", "name": "webengine-sndio" },
             "ffmpeg": { "type": "enum", "name": "webengine-system-ffmpeg", "values": { "system": "yes", "qt": "no" } },
             "opus": { "type": "enum", "name": "webengine-system-opus", "values": { "system": "yes", "qt": "no" } },
             "webp": { "type": "enum", "name": "webengine-system-libwebp", "values": { "system": "yes", "qt": "no" } },
@@ -68,7 +70,13 @@
             "sources": [
                 { "type": "pkgConfig", "args": "libpulse >= 0.9.10 libpulse-mainloop-glib" }
             ]
-        }
+        },
+        "sndio": {
+            "label": "sndio",
+            "sources": [
+                { "type": "pkgConfig", "args": "libsndio >= 1.5.0 libsndio" }
+            ]
+	}
     },
     "tests" : {
         "webengine-host-compiler": {
@@ -136,6 +144,10 @@
             "condition": "libs.webengine-pulseaudio",
             "output": [ "privateFeature" ]
         },
+	"webengine-sndio": {
+            "label": "Use sndio",
+            "output": [ "privateFeature" ]
+        },
         "webengine-pepper-plugins": {
             "label": "Pepper Plugins",
             "purpose": "Enables use of Pepper Flash plugins.",
@@ -308,6 +320,11 @@
                     "condition": "config.unix"
                 },
                 {
+                   "type": "feature",
+                   "args": "webengine-sndio",
+                   "condition": "config.unix"
+                },
+                {
                     "type": "feature",
                     "args": "webengine-sanitizer",
                     "condition": "config.sanitizer"
