--- a/src/3rdparty/chromium/third_party/angle/include/platform/PlatformMethods.h
+++ b/src/3rdparty/chromium/third_party/angle/include/platform/PlatformMethods.h
@@ -239,11 +239,11 @@ inline void DefaultOverrideFeaturesMtl(P
 using ProgramKeyType   = std::array<uint8_t, 20>;
 using CacheProgramFunc = void (*)(PlatformMethods *platform,
                                   const ProgramKeyType &key,
-                                  size_t programSize,
+                                  std::size_t programSize,
                                   const uint8_t *programBytes);
 inline void DefaultCacheProgram(PlatformMethods *platform,
                                 const ProgramKeyType &key,
-                                size_t programSize,
+                                std::size_t programSize,
                                 const uint8_t *programBytes)
 {}
 
