--- a/src/3rdparty/chromium/third_party/crashpad/crashpad/util/linux/thread_info.h
+++ b/src/3rdparty/chromium/third_party/crashpad/crashpad/util/linux/thread_info.h
@@ -17,6 +17,17 @@
 
 #include <stdint.h>
 #include <sys/user.h>
+#if !defined(__GLIBC__)
+struct user_vfp {
+	unsigned long long fpregs[32];
+	unsigned long fpscr;
+};
+struct user_vfp_exc {
+	unsigned long fpexc;
+	unsigned long fpinst;
+	unsigned long fpinst2;
+};
+#endif /* !defined(__GLIBC__) */
 
 #include <type_traits>
 
