--- a/src/3rdparty/chromium/v8/src/base/cpu.cc	2020-09-01 13:02:34.000000000 +0200
+++ b/src/3rdparty/chromium/v8/src/base/cpu.cc	2020-10-06 21:28:45.124882085 +0200
@@ -109,6 +109,15 @@
 #define HWCAP_IDIV  (HWCAP_IDIVA | HWCAP_IDIVT)
 #define HWCAP_LPAE  (1 << 20)
 
+#ifndef __GLIBC__
+#include <elf.h>
+#ifdef __LP64__
+typedef Elf64_auxv_t elf_auxv_t;
+#else
+typedef Elf32_auxv_t elf_auxv_t;
+#endif
+#endif
+
 static uint32_t ReadELFHWCaps() {
 #if V8_GLIBC_PREREQ(2, 16)
   return static_cast<uint32_t>(getauxval(AT_HWCAP));
