--- a/src/3rdparty/chromium/third_party/webrtc/common_video/h264/pps_parser.h
+++ b/src/3rdparty/chromium/third_party/webrtc/common_video/h264/pps_parser.h
@@ -11,6 +11,7 @@
 #ifndef COMMON_VIDEO_H264_PPS_PARSER_H_
 #define COMMON_VIDEO_H264_PPS_PARSER_H_
 
+#include <stdint.h>
 #include "absl/types/optional.h"
 
 namespace rtc {
--- a/src/3rdparty/chromium/third_party/webrtc/common_video/h264/sps_parser.h
+++ b/src/3rdparty/chromium/third_party/webrtc/common_video/h264/sps_parser.h
@@ -11,6 +11,7 @@
 #ifndef COMMON_VIDEO_H264_SPS_PARSER_H_
 #define COMMON_VIDEO_H264_SPS_PARSER_H_
 
+#include <stdint.h>
 #include "absl/types/optional.h"
 
 namespace rtc {
--- a/src/3rdparty/chromium/third_party/webrtc/modules/audio_processing/aec3/reverb_model_estimator.cc
+++ b/src/3rdparty/chromium/third_party/webrtc/modules/audio_processing/aec3/reverb_model_estimator.cc
@@ -8,6 +8,7 @@
  *  be found in the AUTHORS file in the root of the source tree.
  */
 
+#include <memory>
 #include "modules/audio_processing/aec3/reverb_model_estimator.h"
 
 namespace webrtc {
--- a/src/3rdparty/chromium/third_party/webrtc/modules/include/module_common_types_public.h
+++ b/src/3rdparty/chromium/third_party/webrtc/modules/include/module_common_types_public.h
@@ -12,6 +12,7 @@
 #define MODULES_INCLUDE_MODULE_COMMON_TYPES_PUBLIC_H_
 
 #include <limits>
+#include <stdint.h>
 
 #include "absl/types/optional.h"
 
--- a/src/3rdparty/chromium/third_party/blink/renderer/core/page/scrolling/text_fragment_selector_generator.cc
+++ b/src/3rdparty/chromium/third_party/blink/renderer/core/page/scrolling/text_fragment_selector_generator.cc
@@ -6,6 +6,7 @@
 
 #include "base/metrics/histogram_macros.h"
 #include "base/time/default_tick_clock.h"
+#include "absl/base/attributes.h"
 #include "third_party/blink/public/common/browser_interface_broker_proxy.h"
 #include "third_party/blink/public/platform/interface_registry.h"
 #include "third_party/blink/renderer/core/editing/ephemeral_range.h"
--- a/src/3rdparty/chromium/mojo/public/cpp/platform/named_platform_channel_posix.cc
+++ b/src/3rdparty/chromium/mojo/public/cpp/platform/named_platform_channel_posix.cc
@@ -8,6 +8,7 @@
 #include <sys/socket.h>
 #include <sys/un.h>
 #include <unistd.h>
+#include <cstring>
 
 #include "base/files/file_util.h"
 #include "base/files/scoped_file.h"
--- a/src/3rdparty/chromium/third_party/breakpad/breakpad/src/common/linux/elf_core_dump.h
+++ b/src/3rdparty/chromium/third_party/breakpad/breakpad/src/common/linux/elf_core_dump.h
@@ -37,6 +37,9 @@
 #include <limits.h>
 #include <link.h>
 #include <stddef.h>
+#ifndef __GLIBC__
+#include <sys/reg.h>
+#endif
 
 #include "common/memory_range.h"
 
--- a/src/3rdparty/chromium/third_party/ots/include/opentype-sanitiser.h
+++ b/src/3rdparty/chromium/third_party/ots/include/opentype-sanitiser.h
@@ -21,6 +21,7 @@ typedef unsigned __int64 uint64_t;
 #define ots_htons(x) _byteswap_ushort (x)
 #else
 #include <arpa/inet.h>
+#include <sys/types.h>
 #include <stdint.h>
 #define ots_ntohl(x) ntohl (x)
 #define ots_ntohs(x) ntohs (x)
--- a/src/3rdparty/chromium/media/base/video_transformation.h
+++ b/src/3rdparty/chromium/media/base/video_transformation.h
@@ -5,6 +5,7 @@
 #ifndef MEDIA_BASE_VIDEO_TRANSFORMATION_H_
 #define MEDIA_BASE_VIDEO_TRANSFORMATION_H_
 
+#include <cstdint>
 #include <string>
 
 #include "base/numerics/math_constants.h"
--- a/src/3rdparty/chromium/ui/accessibility/ax_node_text_styles.h
+++ b/src/3rdparty/chromium/ui/accessibility/ax_node_text_styles.h
@@ -5,6 +5,7 @@
 #ifndef UI_ACCESSIBILITY_AX_NODE_TEXT_STYLES_H_
 #define UI_ACCESSIBILITY_AX_NODE_TEXT_STYLES_H_
 
+#include <cstdint>
 #include <string>
 
 #include "ui/accessibility/ax_base_export.h"
--- a/src/3rdparty/chromium/ui/accessibility/ax_active_popup.h
+++ b/src/3rdparty/chromium/ui/accessibility/ax_active_popup.h
@@ -9,6 +9,7 @@
 #include "base/no_destructor.h"
 #include "base/optional.h"
 #include "ui/accessibility/ax_export.h"
+#include <cstdint>
 
 namespace ui {
 AX_EXPORT base::Optional<int32_t> GetActivePopupAxUniqueId();
--- a/src/3rdparty/chromium/ui/accessibility/ax_table_info.h
+++ b/src/3rdparty/chromium/ui/accessibility/ax_table_info.h
@@ -5,6 +5,7 @@
 #ifndef UI_ACCESSIBILITY_AX_TABLE_INFO_H_
 #define UI_ACCESSIBILITY_AX_TABLE_INFO_H_
 
+#include <cstdint>
 #include <map>
 #include <set>
 #include <string>
--- a/src/3rdparty/chromium/components/metrics/environment_recorder.h
+++ b/src/3rdparty/chromium/components/metrics/environment_recorder.h
@@ -5,6 +5,7 @@
 #ifndef COMPONENTS_METRICS_ENVIRONMENT_RECORDER_H_
 #define COMPONENTS_METRICS_ENVIRONMENT_RECORDER_H_
 
+#include <cstdint>
 #include <string>
 
 #include "base/macros.h"
--- a/src/3rdparty/chromium/content/public/common/page_state.h
+++ b/src/3rdparty/chromium/content/public/common/page_state.h
@@ -5,6 +5,7 @@
 #ifndef CONTENT_PUBLIC_COMMON_PAGE_STATE_H_
 #define CONTENT_PUBLIC_COMMON_PAGE_STATE_H_
 
+#include <cstdint>
 #include <string>
 #include <vector>
 
--- a/src/3rdparty/chromium/content/public/browser/mhtml_generation_result.h
+++ b/src/3rdparty/chromium/content/public/browser/mhtml_generation_result.h
@@ -5,6 +5,7 @@
 #ifndef CONTENT_PUBLIC_BROWSER_MHTML_GENERATION_RESULT_H_
 #define CONTENT_PUBLIC_BROWSER_MHTML_GENERATION_RESULT_H_
 
+#include <cstdint>
 #include <string>
 
 #include "base/callback_forward.h"
--- a/src/3rdparty/chromium/extensions/common/api/messaging/port_context.h
+++ b/src/3rdparty/chromium/extensions/common/api/messaging/port_context.h
@@ -6,6 +6,7 @@
 #define EXTENSIONS_COMMON_API_MESSAGING_PORT_CONTEXT_H_
 
 #include <stddef.h>
+#include <stdint.h>
 
 #include <string>
 
