--- a/Makefile	2022-11-22 22:25:15.000000000 +0100
+++ -	2023-07-08 16:35:16.308351723 +0200
@@ -32,7 +32,7 @@
 	-Ikafel/include
 
 CXXFLAGS += $(USER_DEFINES) $(COMMON_FLAGS) $(shell pkg-config --cflags protobuf) \
-	-std=c++14 -fno-exceptions -Wno-unused -Wno-unused-parameter
+	-std=c++17 -fno-exceptions -Wno-unused -Wno-unused-parameter
 LDFLAGS += -pie -Wl,-z,noexecstack -lpthread $(shell pkg-config --libs protobuf)
 
 BIN = nsjail
