--- configure	2011-11-05 21:24:39.000000000 +1000
+++ configure	2011-12-27 15:43:54.832120449 +1000
@@ -6402,6 +6402,28 @@
	DEFAULT_SUPP="glibc-2.34567-NPTL-helgrind.supp ${DEFAULT_SUPP}"
	DEFAULT_SUPP="glibc-2.X-drd.supp ${DEFAULT_SUPP}"
	;;
+     2.15)
+	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: 2.15 family" >&5
+$as_echo "2.15 family" >&6; }
+
+$as_echo "#define GLIBC_2_15 1" >>confdefs.h
+
+	DEFAULT_SUPP="glibc-2.X.supp ${DEFAULT_SUPP}"
+	DEFAULT_SUPP="glibc-2.34567-NPTL-helgrind.supp ${DEFAULT_SUPP}"
+	DEFAULT_SUPP="glibc-2.X-drd.supp ${DEFAULT_SUPP}"
+	;;
+
+     2.16)
+	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: 2.16 family" >&5
+$as_echo "2.16 family" >&6; }
+
+$as_echo "#define GLIBC_2_16 1" >>confdefs.h
+
+	DEFAULT_SUPP="glibc-2.X.supp ${DEFAULT_SUPP}"
+	DEFAULT_SUPP="glibc-2.34567-NPTL-helgrind.supp ${DEFAULT_SUPP}"
+	DEFAULT_SUPP="glibc-2.X-drd.supp ${DEFAULT_SUPP}"
+	;;
+
      darwin)
	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: Darwin" >&5
 $as_echo "Darwin" >&6; }
@@ -6422,7 +6433,7 @@
      *)
	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: unsupported version ${GLIBC_VERSION}" >&5
 $as_echo "unsupported version ${GLIBC_VERSION}" >&6; }
-	as_fn_error "Valgrind requires glibc version 2.2 - 2.14" "$LINENO" 5
+	as_fn_error "Valgrind requires glibc version 2.2 - 2.16" "$LINENO" 5
	as_fn_error "or Darwin libc" "$LINENO" 5
	;;
 esac
