--- valgrind/configure.in.jj	2010-10-21 03:14:02.000000000 +0200
+++ valgrind/configure.in	2010-11-05 17:09:18.674455299 +0100
@@ -749,6 +759,13 @@ case "${GLIBC_VERSION}" in
 	DEFAULT_SUPP="glibc-2.34567-NPTL-helgrind.supp ${DEFAULT_SUPP}"
 	DEFAULT_SUPP="glibc-2.X-drd.supp ${DEFAULT_SUPP}"
 	;;
+     2.14)
+	AC_MSG_RESULT(2.14 family)
+	AC_DEFINE([GLIBC_2_14], 1, [Define to 1 if you're using glibc 2.14.x])
+	DEFAULT_SUPP="glibc-2.X.supp ${DEFAULT_SUPP}"
+	DEFAULT_SUPP="glibc-2.34567-NPTL-helgrind.supp ${DEFAULT_SUPP}"
+	DEFAULT_SUPP="glibc-2.X-drd.supp ${DEFAULT_SUPP}"
+	;;
      aix5)
 	AC_MSG_RESULT(AIX 5.1 or 5.2 or 5.3)
 	AC_DEFINE([AIX5_LIBC], 1, [Define to 1 if you're using AIX 5.1 or 5.2 or 5.3])
@@ -762,7 +779,7 @@ case "${GLIBC_VERSION}" in
 
      *)
 	AC_MSG_RESULT([unsupported version ${GLIBC_VERSION}])
-	AC_MSG_ERROR([Valgrind requires glibc version 2.2 - 2.13])
+	AC_MSG_ERROR([Valgrind requires glibc version 2.2 - 2.14])
 	AC_MSG_ERROR([or AIX 5.1 or 5.2 or 5.3 GLIBC_VERSION])
 	AC_MSG_ERROR([or Darwin libc])
 	;;
--- valgrind/config.h.in.jj	2010-10-20 22:20:49.000000000 +0200
+++ valgrind/config.h.in	2010-11-05 17:08:33.863454886 +0100
@@ -33,6 +33,9 @@
 /* Define to 1 if you're using glibc 2.13.x */
 #undef GLIBC_2_13
 
+/* Define to 1 if you're using glibc 2.14.x */
+#undef GLIBC_2_14
+
 /* Define to 1 if you're using glibc 2.2.x */
 #undef GLIBC_2_2
 
--- valgrind/configure.jj	2010-10-21 03:16:18.000000000 +0200
+++ valgrind/configure	2010-11-05 17:09:39.163455396 +0100
@@ -6367,6 +6389,18 @@ $as_echo "#define GLIBC_2_13 1" >>confde
 	DEFAULT_SUPP="glibc-2.34567-NPTL-helgrind.supp ${DEFAULT_SUPP}"
 	DEFAULT_SUPP="glibc-2.X-drd.supp ${DEFAULT_SUPP}"
 	;;
+     2.14)
+	echo "$as_me:$LINENO: result: 2.14 family" >&5
+echo "${ECHO_T}2.14 family" >&6
+
+cat >>confdefs.h <<\_ACEOF
+#define GLIBC_2_14 1
+_ACEOF
+
+	DEFAULT_SUPP="glibc-2.X.supp ${DEFAULT_SUPP}"
+	DEFAULT_SUPP="glibc-2.34567-NPTL-helgrind.supp ${DEFAULT_SUPP}"
+	DEFAULT_SUPP="glibc-2.X-drd.supp ${DEFAULT_SUPP}"
+	;;
      aix5)
 	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: AIX 5.1 or 5.2 or 5.3" >&5
 $as_echo "AIX 5.1 or 5.2 or 5.3" >&6; }
@@ -6387,7 +6421,7 @@ $as_echo "#define DARWIN_LIBC 1" >>confd
      *)
 	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: unsupported version ${GLIBC_VERSION}" >&5
 $as_echo "unsupported version ${GLIBC_VERSION}" >&6; }
-	as_fn_error "Valgrind requires glibc version 2.2 - 2.13" "$LINENO" 5
+	as_fn_error "Valgrind requires glibc version 2.2 - 2.14" "$LINENO" 5
 	as_fn_error "or AIX 5.1 or 5.2 or 5.3 GLIBC_VERSION" "$LINENO" 5
 	as_fn_error "or Darwin libc" "$LINENO" 5
 	;;
