--- configure	2012-09-19 05:19:23.000000000 +1000
+++ configure	2012-12-27 08:11:36.645429011 +1000
@@ -6612,6 +6612,24 @@

	# DEFAULT_SUPP set by kernel version check above.
	;;
+     2.17)
+	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: 2.17 family" >&5
+$as_echo "2.17 family" >&6; }
+
+$as_echo "#define GLIBC_2_17 1" >>confdefs.h
+
+	DEFAULT_SUPP="glibc-2.X.supp ${DEFAULT_SUPP}"
+	DEFAULT_SUPP="glibc-2.34567-NPTL-helgrind.supp ${DEFAULT_SUPP}"
+	DEFAULT_SUPP="glibc-2.X-drd.supp ${DEFAULT_SUPP}"
+	;;
+     darwin)
+	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: Darwin" >&5
+$as_echo "Darwin" >&6; }
+
+$as_echo "#define DARWIN_LIBC 1" >>confdefs.h
+
+	# DEFAULT_SUPP set by kernel version check above.
+	;;
      bionic)
	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: Bionic" >&5
 $as_echo "Bionic" >&6; }
@@ -6624,7 +6642,7 @@
      *)
	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: unsupported version ${GLIBC_VERSION}" >&5
 $as_echo "unsupported version ${GLIBC_VERSION}" >&6; }
-	as_fn_error "Valgrind requires glibc version 2.2 - 2.16" "$LINENO" 5
+	as_fn_error "Valgrind requires glibc version 2.2 - 2.17" "$LINENO" 5
	as_fn_error "or Darwin libc" "$LINENO" 5
	;;
 esac
