--- a/filters/reorient.cpp.orig	2020-12-21 23:44:16.000000000 -0700
+++ b/filters/reorient.cpp	2021-09-03 16:33:08.078102032 -0700
@@ -96,7 +96,7 @@ chomp (char *str)
 bool
 tesseract_version_before_(const char *cutoff)
 {
-  FILE *fp = popen ("tesseract --version 2>&1"
+  FILE *fp = popen ("tesseract-ocr --version 2>&1"
                     "| awk '/^tesseract/{ print $2 }'", "r");
   int errc = errno;
 
@@ -126,7 +126,7 @@ tesseract_version_before_(const char *cu
 bool
 have_tesseract_language_pack_(const char *lang)
 {
-  std::string cmd("tesseract --list-langs 2>&1"
+  std::string cmd("tesseract-ocr --list-langs 2>&1"
                   "| sed -n '/^");
   cmd += lang;
   cmd += "$/p'";
@@ -291,7 +291,7 @@ reorient::reorient ()
 
   if (found)
     {
-      if (have_tesseract_())  engine_ = "tesseract";
+      if (have_tesseract_())  engine_ = "tesseract-ocr";
       if (have_ocr_engine_()) engine_ = abs_path_name;
     }
   freeze_options ();   // initializes option tracking member variables
