--- a/filters/get-text-orientation.orig	2020-12-21 23:44:15.000000000 -0700
+++ b/filters/get-text-orientation	2021-09-03 22:11:53.401409017 -0700
@@ -48,7 +48,8 @@ if test $? != 0; then
 fi
 
 tmpfile=$(mktemp -q .reorient.XXX)
-trap "rm -f $tmpfile" 0 1 2 15
+tmpout=$(mktemp -q .reorient.XXX)
+trap "rm -f $tmpfile $tmpout.osd $tmpout.txt $tmpout" 0 1 2 15
 
 case "$engine" in
     */tesseract*|tesseract*)
@@ -51,7 +51,7 @@ tmpfile=$(mktemp -q .reorient.XXX)
 trap "rm -f $tmpfile" 0 1 2 15
 
 case "$engine" in
-    */tesseract|tesseract)
+    */tesseract*|tesseract*)
 
         #  Notwithstanding what the manual page says, tesseract
         #  doesn't support reading from standard input with the
@@ -66,7 +67,8 @@ case "$engine" in
         #  about ends up on standard error, but our caller looks
         #  for it on standard output.  Redirect to handle that.
 
-        $engine $tmpfile /dev/null -psm 0 -l osd 2>&1
+        $engine $tmpfile $tmpout --psm 0 -l osd >/dev/null 2>/dev/null
+        cat $tmpout.osd
         ;;
 
     */ocr-engine-getrotate)
