--- a/lib/tests/memory.cpp.orig	2020-12-21 23:44:15.000000000 -0700
+++ b/lib/tests/memory.cpp	2021-09-03 15:16:25.323531622 -0700
@@ -145,7 +145,7 @@ constant_octets (octet value)
   const streamsize margin = 10;
   const octet nul = 0x00;
   traits::assign (buffer, size, nul);
-  BOOST_MESSAGE ("value: " << value );
+  BOOST_TEST_MESSAGE ("value: " << value );
 
   setmem_idevice dev (shared_ptr<setmem_idevice::generator>
                       (new const_generator (value)));
