--- a/lib/tests/Makefile.in.orig	2020-12-21 23:45:00.000000000 -0700
+++ b/lib/tests/Makefile.in	2021-09-03 15:10:58.763604642 -0700
@@ -200,6 +200,7 @@ am__DEPENDENCIES_1 =
 @enable_boost_unit_test_framework_TRUE@	../libutsushi.la \
 @enable_boost_unit_test_framework_TRUE@	$(am__DEPENDENCIES_1) \
 @enable_boost_unit_test_framework_TRUE@	../../connexions/libcnx-usb.la \
+@enable_boost_unit_test_framework_TRUE@	../../connexions/libcnx-hexdump.la \
 @enable_boost_unit_test_framework_TRUE@	$(am__DEPENDENCIES_1)
 AM_V_lt = $(am__v_lt_@AM_V@)
 am__v_lt_ = $(am__v_lt_@AM_DEFAULT_V@)
@@ -212,6 +213,7 @@ connexion_utr_LDADD = $(LDADD)
 @enable_boost_unit_test_framework_TRUE@	../libutsushi.la \
 @enable_boost_unit_test_framework_TRUE@	$(am__DEPENDENCIES_1) \
 @enable_boost_unit_test_framework_TRUE@	../../connexions/libcnx-usb.la \
+@enable_boost_unit_test_framework_TRUE@	../../connexions/libcnx-hexdump.la \
 @enable_boost_unit_test_framework_TRUE@	$(am__DEPENDENCIES_1)
 descriptor_utr_SOURCES = descriptor.cpp
 descriptor_utr_OBJECTS = descriptor.$(OBJEXT)
@@ -220,6 +222,7 @@ descriptor_utr_LDADD = $(LDADD)
 @enable_boost_unit_test_framework_TRUE@	../libutsushi.la \
 @enable_boost_unit_test_framework_TRUE@	$(am__DEPENDENCIES_1) \
 @enable_boost_unit_test_framework_TRUE@	../../connexions/libcnx-usb.la \
+@enable_boost_unit_test_framework_TRUE@	../../connexions/libcnx-hexdump.la \
 @enable_boost_unit_test_framework_TRUE@	$(am__DEPENDENCIES_1)
 device_utr_SOURCES = device.cpp
 device_utr_OBJECTS = device.$(OBJEXT)
@@ -228,6 +231,7 @@ device_utr_LDADD = $(LDADD)
 @enable_boost_unit_test_framework_TRUE@	../libutsushi.la \
 @enable_boost_unit_test_framework_TRUE@	$(am__DEPENDENCIES_1) \
 @enable_boost_unit_test_framework_TRUE@	../../connexions/libcnx-usb.la \
+@enable_boost_unit_test_framework_TRUE@	../../connexions/libcnx-hexdump.la \
 @enable_boost_unit_test_framework_TRUE@	$(am__DEPENDENCIES_1)
 file_utr_SOURCES = file.cpp
 file_utr_OBJECTS = file.$(OBJEXT)
@@ -236,6 +240,7 @@ file_utr_LDADD = $(LDADD)
 @enable_boost_unit_test_framework_TRUE@	../libutsushi.la \
 @enable_boost_unit_test_framework_TRUE@	$(am__DEPENDENCIES_1) \
 @enable_boost_unit_test_framework_TRUE@	../../connexions/libcnx-usb.la \
+@enable_boost_unit_test_framework_TRUE@	../../connexions/libcnx-hexdump.la \
 @enable_boost_unit_test_framework_TRUE@	$(am__DEPENDENCIES_1)
 ipc_cnx_SOURCES = ipc-cnx.cpp
 ipc_cnx_OBJECTS = ipc-cnx.$(OBJEXT)
@@ -244,6 +249,7 @@ ipc_cnx_LDADD = $(LDADD)
 @enable_boost_unit_test_framework_TRUE@	../libutsushi.la \
 @enable_boost_unit_test_framework_TRUE@	$(am__DEPENDENCIES_1) \
 @enable_boost_unit_test_framework_TRUE@	../../connexions/libcnx-usb.la \
+@enable_boost_unit_test_framework_TRUE@	../../connexions/libcnx-hexdump.la \
 @enable_boost_unit_test_framework_TRUE@	$(am__DEPENDENCIES_1)
 am__log_checking_utr_SOURCES_DIST = log.cpp
 @enable_boost_unit_test_framework_TRUE@am_log_checking_utr_OBJECTS = log_checking_utr-log.$(OBJEXT)
@@ -253,6 +259,7 @@ log_checking_utr_LDADD = $(LDADD)
 @enable_boost_unit_test_framework_TRUE@	../libutsushi.la \
 @enable_boost_unit_test_framework_TRUE@	$(am__DEPENDENCIES_1) \
 @enable_boost_unit_test_framework_TRUE@	../../connexions/libcnx-usb.la \
+@enable_boost_unit_test_framework_TRUE@	../../connexions/libcnx-hexdump.la \
 @enable_boost_unit_test_framework_TRUE@	$(am__DEPENDENCIES_1)
 am__log_no_check_utr_SOURCES_DIST = log.cpp
 @enable_boost_unit_test_framework_TRUE@am_log_no_check_utr_OBJECTS = log_no_check_utr-log.$(OBJEXT)
@@ -262,6 +269,7 @@ log_no_check_utr_LDADD = $(LDADD)
 @enable_boost_unit_test_framework_TRUE@	../libutsushi.la \
 @enable_boost_unit_test_framework_TRUE@	$(am__DEPENDENCIES_1) \
 @enable_boost_unit_test_framework_TRUE@	../../connexions/libcnx-usb.la \
+@enable_boost_unit_test_framework_TRUE@	../../connexions/libcnx-hexdump.la \
 @enable_boost_unit_test_framework_TRUE@	$(am__DEPENDENCIES_1)
 memory_utr_SOURCES = memory.cpp
 memory_utr_OBJECTS = memory.$(OBJEXT)
@@ -270,6 +278,7 @@ memory_utr_LDADD = $(LDADD)
 @enable_boost_unit_test_framework_TRUE@	../libutsushi.la \
 @enable_boost_unit_test_framework_TRUE@	$(am__DEPENDENCIES_1) \
 @enable_boost_unit_test_framework_TRUE@	../../connexions/libcnx-usb.la \
+@enable_boost_unit_test_framework_TRUE@	../../connexions/libcnx-hexdump.la \
 @enable_boost_unit_test_framework_TRUE@	$(am__DEPENDENCIES_1)
 monitor_utr_SOURCES = monitor.cpp
 monitor_utr_OBJECTS = monitor.$(OBJEXT)
@@ -278,6 +287,7 @@ monitor_utr_LDADD = $(LDADD)
 @enable_boost_unit_test_framework_TRUE@	../libutsushi.la \
 @enable_boost_unit_test_framework_TRUE@	$(am__DEPENDENCIES_1) \
 @enable_boost_unit_test_framework_TRUE@	../../connexions/libcnx-usb.la \
+@enable_boost_unit_test_framework_TRUE@	../../connexions/libcnx-hexdump.la \
 @enable_boost_unit_test_framework_TRUE@	$(am__DEPENDENCIES_1)
 null_utr_SOURCES = null.cpp
 null_utr_OBJECTS = null.$(OBJEXT)
@@ -286,6 +296,7 @@ null_utr_LDADD = $(LDADD)
 @enable_boost_unit_test_framework_TRUE@	../libutsushi.la \
 @enable_boost_unit_test_framework_TRUE@	$(am__DEPENDENCIES_1) \
 @enable_boost_unit_test_framework_TRUE@	../../connexions/libcnx-usb.la \
+@enable_boost_unit_test_framework_TRUE@	../../connexions/libcnx-hexdump.la \
 @enable_boost_unit_test_framework_TRUE@	$(am__DEPENDENCIES_1)
 octet_utr_SOURCES = octet.cpp
 octet_utr_OBJECTS = octet.$(OBJEXT)
@@ -294,6 +305,7 @@ octet_utr_LDADD = $(LDADD)
 @enable_boost_unit_test_framework_TRUE@	../libutsushi.la \
 @enable_boost_unit_test_framework_TRUE@	$(am__DEPENDENCIES_1) \
 @enable_boost_unit_test_framework_TRUE@	../../connexions/libcnx-usb.la \
+@enable_boost_unit_test_framework_TRUE@	../../connexions/libcnx-hexdump.la \
 @enable_boost_unit_test_framework_TRUE@	$(am__DEPENDENCIES_1)
 option_utr_SOURCES = option.cpp
 option_utr_OBJECTS = option.$(OBJEXT)
@@ -302,6 +314,7 @@ option_utr_LDADD = $(LDADD)
 @enable_boost_unit_test_framework_TRUE@	../libutsushi.la \
 @enable_boost_unit_test_framework_TRUE@	$(am__DEPENDENCIES_1) \
 @enable_boost_unit_test_framework_TRUE@	../../connexions/libcnx-usb.la \
+@enable_boost_unit_test_framework_TRUE@	../../connexions/libcnx-hexdump.la \
 @enable_boost_unit_test_framework_TRUE@	$(am__DEPENDENCIES_1)
 quantity_utr_SOURCES = quantity.cpp
 quantity_utr_OBJECTS = quantity.$(OBJEXT)
@@ -310,6 +323,7 @@ quantity_utr_LDADD = $(LDADD)
 @enable_boost_unit_test_framework_TRUE@	../libutsushi.la \
 @enable_boost_unit_test_framework_TRUE@	$(am__DEPENDENCIES_1) \
 @enable_boost_unit_test_framework_TRUE@	../../connexions/libcnx-usb.la \
+@enable_boost_unit_test_framework_TRUE@	../../connexions/libcnx-hexdump.la \
 @enable_boost_unit_test_framework_TRUE@	$(am__DEPENDENCIES_1)
 run_time_utr_SOURCES = run-time.cpp
 run_time_utr_OBJECTS = run-time.$(OBJEXT)
@@ -318,6 +332,7 @@ run_time_utr_LDADD = $(LDADD)
 @enable_boost_unit_test_framework_TRUE@	../libutsushi.la \
 @enable_boost_unit_test_framework_TRUE@	$(am__DEPENDENCIES_1) \
 @enable_boost_unit_test_framework_TRUE@	../../connexions/libcnx-usb.la \
+@enable_boost_unit_test_framework_TRUE@	../../connexions/libcnx-hexdump.la \
 @enable_boost_unit_test_framework_TRUE@	$(am__DEPENDENCIES_1)
 scanner_utr_SOURCES = scanner.cpp
 scanner_utr_OBJECTS = scanner.$(OBJEXT)
@@ -326,6 +341,7 @@ scanner_utr_LDADD = $(LDADD)
 @enable_boost_unit_test_framework_TRUE@	../libutsushi.la \
 @enable_boost_unit_test_framework_TRUE@	$(am__DEPENDENCIES_1) \
 @enable_boost_unit_test_framework_TRUE@	../../connexions/libcnx-usb.la \
+@enable_boost_unit_test_framework_TRUE@	../../connexions/libcnx-hexdump.la \
 @enable_boost_unit_test_framework_TRUE@	$(am__DEPENDENCIES_1)
 stream_utr_SOURCES = stream.cpp
 stream_utr_OBJECTS = stream.$(OBJEXT)
@@ -334,6 +350,7 @@ stream_utr_LDADD = $(LDADD)
 @enable_boost_unit_test_framework_TRUE@	../libutsushi.la \
 @enable_boost_unit_test_framework_TRUE@	$(am__DEPENDENCIES_1) \
 @enable_boost_unit_test_framework_TRUE@	../../connexions/libcnx-usb.la \
+@enable_boost_unit_test_framework_TRUE@	../../connexions/libcnx-hexdump.la \
 @enable_boost_unit_test_framework_TRUE@	$(am__DEPENDENCIES_1)
 string_utr_SOURCES = string.cpp
 string_utr_OBJECTS = string.$(OBJEXT)
@@ -342,6 +359,7 @@ string_utr_LDADD = $(LDADD)
 @enable_boost_unit_test_framework_TRUE@	../libutsushi.la \
 @enable_boost_unit_test_framework_TRUE@	$(am__DEPENDENCIES_1) \
 @enable_boost_unit_test_framework_TRUE@	../../connexions/libcnx-usb.la \
+@enable_boost_unit_test_framework_TRUE@	../../connexions/libcnx-hexdump.la \
 @enable_boost_unit_test_framework_TRUE@	$(am__DEPENDENCIES_1)
 tiff_utr_SOURCES = tiff.cpp
 tiff_utr_OBJECTS = tiff_utr-tiff.$(OBJEXT)
@@ -349,6 +367,7 @@ tiff_utr_OBJECTS = tiff_utr-tiff.$(OBJEX
 @enable_boost_unit_test_framework_TRUE@	../libutsushi.la \
 @enable_boost_unit_test_framework_TRUE@	$(am__DEPENDENCIES_1) \
 @enable_boost_unit_test_framework_TRUE@	../../connexions/libcnx-usb.la \
+@enable_boost_unit_test_framework_TRUE@	../../connexions/libcnx-hexdump.la \
 @enable_boost_unit_test_framework_TRUE@	$(am__DEPENDENCIES_1)
 @enable_boost_unit_test_framework_TRUE@@have_libmagic_TRUE@@have_libtiff_TRUE@am__DEPENDENCIES_3 = $(am__DEPENDENCIES_1)
 @enable_boost_unit_test_framework_TRUE@@have_libtiff_TRUE@tiff_utr_DEPENDENCIES = $(am__DEPENDENCIES_2) \
@@ -364,6 +383,7 @@ value_utr_LDADD = $(LDADD)
 @enable_boost_unit_test_framework_TRUE@	../libutsushi.la \
 @enable_boost_unit_test_framework_TRUE@	$(am__DEPENDENCIES_1) \
 @enable_boost_unit_test_framework_TRUE@	../../connexions/libcnx-usb.la \
+@enable_boost_unit_test_framework_TRUE@	../../connexions/libcnx-hexdump.la \
 @enable_boost_unit_test_framework_TRUE@	$(am__DEPENDENCIES_1)
 AM_V_P = $(am__v_P_@AM_V@)
 am__v_P_ = $(am__v_P_@AM_DEFAULT_V@)
@@ -872,6 +892,7 @@ top_srcdir = @top_srcdir@
 @enable_boost_unit_test_framework_TRUE@LDADD = ../libutsushi.la \
 @enable_boost_unit_test_framework_TRUE@	$(LIBUTSUSHI_LIBS) \
 @enable_boost_unit_test_framework_TRUE@	../../connexions/libcnx-usb.la \
+@enable_boost_unit_test_framework_TRUE@	../../connexions/libcnx-hexdump.la \
 @enable_boost_unit_test_framework_TRUE@	$(BOOST_UNIT_TEST_FRAMEWORK_LIB)
 @enable_boost_unit_test_framework_TRUE@@have_libtiff_TRUE@tiff_utr_CXXFLAGS = $(AM_CXXFLAGS) \
 @enable_boost_unit_test_framework_TRUE@@have_libtiff_TRUE@	$(LIBTIFF_CFLAGS) \
