--- a/sane/tests/Makefile.in.orig	2020-12-21 23:45:00.000000000 -0700
+++ b/sane/tests/Makefile.in	2021-09-04 00:42:29.756027426 -0700
@@ -176,6 +176,7 @@ am__DEPENDENCIES_1 =
 @enable_boost_unit_test_framework_TRUE@	$(am__DEPENDENCIES_1) \
 @enable_boost_unit_test_framework_TRUE@	../../filters/libflt-all.la \
 @enable_boost_unit_test_framework_TRUE@	../../connexions/libcnx-usb.la \
+@enable_boost_unit_test_framework_TRUE@	../../connexions/libcnx-hexdump.la \
 @enable_boost_unit_test_framework_TRUE@	$(am__DEPENDENCIES_1)
 AM_V_lt = $(am__v_lt_@AM_V@)
 am__v_lt_ = $(am__v_lt_@AM_DEFAULT_V@)
@@ -188,6 +189,7 @@ value_utr_LDADD = $(LDADD)
 @enable_boost_unit_test_framework_TRUE@	$(am__DEPENDENCIES_1) \
 @enable_boost_unit_test_framework_TRUE@	../../filters/libflt-all.la \
 @enable_boost_unit_test_framework_TRUE@	../../connexions/libcnx-usb.la \
+@enable_boost_unit_test_framework_TRUE@	../../connexions/libcnx-hexdump.la \
 @enable_boost_unit_test_framework_TRUE@	$(am__DEPENDENCIES_1)
 AM_V_P = $(am__v_P_@AM_V@)
 am__v_P_ = $(am__v_P_@AM_DEFAULT_V@)
@@ -669,6 +671,7 @@ top_srcdir = @top_srcdir@
 @enable_boost_unit_test_framework_TRUE@	$(LIBUTSUSHI_LIBS) \
 @enable_boost_unit_test_framework_TRUE@	../../filters/libflt-all.la \
 @enable_boost_unit_test_framework_TRUE@	../../connexions/libcnx-usb.la \
+@enable_boost_unit_test_framework_TRUE@	../../connexions/libcnx-hexdump.la \
 @enable_boost_unit_test_framework_TRUE@	$(BOOST_UNIT_TEST_FRAMEWORK_LIB)
 EXTRA_DIST = sane-api.ipp
 CLEANFILES = $(am__append_7) $(am__append_9)
