--- a/makefile
+++ b/makefile
@@ -1,7 +1,7 @@
 #--------------------------------
 # jhead makefile for Unix
 #--------------------------------
-PREFIX=$(DESTDIR)/usr/local
+PREFIX=/usr
 BINDIR=$(PREFIX)/bin
 DOCDIR=$(PREFIX)/share/doc/jhead
 MANDIR=$(PREFIX)/share/man/man1
@@ -40,7 +40,7 @@
 	rm -f $(objs) jhead
 
 install: all
-	install -d $(BINDIR) $(DOCDIR) $(MANDIR)
-	install -m 0755 jhead $(BINDIR)
-	install -m 0644 $(docs) $(DOCDIR)
-	install -m 0644 jhead.1 $(MANDIR)
+	install -d $(DESTDIR)$(BINDIR) $(DESTDIR)$(DOCDIR) $(DESTDIR)$(MANDIR)
+	install -m 0755 jhead $(DESTDIR)$(BINDIR)
+	install -m 0644 $(docs) $(DESTDIR)$(DOCDIR)
+	install -m 0644 jhead.1 $(DESTDIR)$(MANDIR)
