--- a/Makefile	2016-05-24 14:29:52.349935460 +0200
+++ b/Makefile	2016-05-24 14:30:01.190934798 +0200
@@ -29,7 +29,7 @@ $(OBJ)/libbish.a: $(OBJECTS)
 	ranlib $@
 
 bish: $(SRC)/bish.cpp $(OBJ)/libbish.a
-	$(CXX) $(CXXFLAGS) -o bish $(SRC)/bish.cpp $(OBJ)/libbish.a $(CONFIG_CONSTANTS)
+	$(CXX) $(CXXFLAGS) $(LDFLAGS) -o bish $(SRC)/bish.cpp $(OBJ)/libbish.a $(CONFIG_CONSTANTS)
 
 .PHONY: clean
 clean:
