--- a/shared/source/compiler_interface/os_compiler_cache_helper.h
+++ b/shared/source/compiler_interface/os_compiler_cache_helper.h
@@ -7,10 +7,11 @@

 #pragma once
 #include <string>
+#include <ctime>

 namespace NEO {
 class EnvironmentVariableReader;
 bool checkDefaultCacheDirSettings(std::string &cacheDir, NEO::EnvironmentVariableReader &reader);
 time_t getFileModificationTime(const std::string &path);
 size_t getFileSize(const std::string &path);
-} // namespace NEO
\ No newline at end of file
+} // namespace NEO
