--- opencl/test/unit_test/linux/mock_os_layer.h.orig	2020-05-12 23:23:47.438463587 +0200
+++ opencl/test/unit_test/linux/mock_os_layer.h	2020-05-12 23:26:38.282470463 +0200
@@ -20,7 +20,11 @@
 
 extern "C" {
 int open(const char *pathname, int flags, ...);
+#ifdef __GLIBC__
 int ioctl(int fd, unsigned long int request, ...) throw();
+#else
+int ioctl(int fd, int request, ...) throw();
+#endif
 }
 
 extern int (*c_open)(const char *pathname, int flags, ...);
