--- shared/source/os_interface/linux/os_library_linux.cpp.orig	2020-05-12 21:02:19.995121993 +0200
+++ shared/source/os_interface/linux/os_library_linux.cpp	2020-05-12 21:03:20.426124425 +0200
@@ -30,7 +30,7 @@
         this->handle = dlopen(0, RTLD_LAZY);
     } else {
 #ifdef SANITIZER_BUILD
-        constexpr auto dlopenFlag = RTLD_LAZY;
+        constexpr auto dlopenFlag = RTLD_LAZY | RTLD_GLOBAL;
 #else
         constexpr auto dlopenFlag = RTLD_LAZY | RTLD_DEEPBIND;
 #endif
