--- info/window.c_old	2009-08-04 13:46:15.000000000 +0200
+++ info/window.c	2009-08-04 13:47:29.000000000 +0200
@@ -1581,7 +1581,7 @@ process_node_text (WINDOW *win, char *st
       const char *carried_over_ptr;
       size_t carried_over_len, carried_over_count;
       const char *cur_ptr = mbi_cur_ptr (iter);
-      int cur_len = mb_len (mbi_cur (iter));
+      size_t cur_len = mb_len (mbi_cur (iter));
       int replen;
       int delim = 0;
       int rc;
@@ -1754,7 +1754,7 @@ clean_manpage (char *manpage)
        mbi_advance (iter))
     {
       const char *cur_ptr = mbi_cur_ptr (iter);
-      int cur_len = mb_len (mbi_cur (iter));
+      size_t cur_len = mb_len (mbi_cur (iter));

       if (cur_len == 1)
	{
@@ -1852,8 +1852,8 @@ window_scan_line (WINDOW *win, int line,
        mbi_advance (iter))
     {
       const char *cur_ptr = mbi_cur_ptr (iter);
-      int cur_len = mb_len (mbi_cur (iter));
-      int replen;
+      size_t cur_len = mb_len (mbi_cur (iter));
+      size_t replen;

       if (cur_ptr >= endp)
	break;
