--- a/runtime/druntime/src/ldc/arm_unwind.c
+++ b/runtime/druntime/src/ldc/arm_unwind.c
@@ -17,6 +17,10 @@
 
 // clang's unwind.h doesn't have this
 typedef struct _Unwind_Context _Unwind_Context;
+#ifndef __GLIBC__
+typedef unsigned _Unwind_Ptr __attribute__((__mode__(__pointer__)));
+typedef unsigned _Unwind_Word __attribute__((__mode__(__word__)));
+#endif
 
 _Unwind_Ptr _d_eh_GetIP(_Unwind_Context *context)
 {
