--- a/runtime/druntime/src/core/sys/posix/sys/stat.d
+++ b/runtime/druntime/src/core/sys/posix/sys/stat.d
@@ -77,36 +77,36 @@ version (linux)
             struct stat_t
             {
                 dev_t       st_dev;
-                ushort      __pad1;
-                static if (!__USE_FILE_OFFSET64)
-                {
-                    ino_t       st_ino;
-                }
-                else
-                {
-                    uint        __st_ino;
-                }
+                int         __pad1;
+                long        __st_ino;
                 mode_t      st_mode;
                 nlink_t     st_nlink;
                 uid_t       st_uid;
                 gid_t       st_gid;
                 dev_t       st_rdev;
-                ushort      __pad2;
+                int         __pad2;
                 off_t       st_size;
                 blksize_t   st_blksize;
                 blkcnt_t    st_blocks;
-                private struct __timespec32
-                {
-                    c_long tv_sec;
-                    c_long tv_nsec;
+                version (CRuntime_Musl_Pre_Time64) {
+                    timespec    st_atim;
+                    timespec    st_mtim;
+                    timespec    st_ctim;
+                    ino_t       st_ino;
+                } else {
+                    private struct __timespec32
+                    {
+                        c_long tv_sec;
+                        c_long tv_nsec;
+                    }
+                    __timespec32 __st_atim32;
+                    __timespec32 __st_mtim32;
+                    __timespec32 __st_ctim32;
+                    ino_t       st_ino;
+                    timespec    st_atim;
+                    timespec    st_mtim;
+                    timespec    st_ctim;
                 }
-                __timespec32 __st_atim32;
-                __timespec32 __st_mtim32;
-                __timespec32 __st_ctim32;
-                ino_t       st_ino;
-                timespec    st_atim;
-                timespec    st_mtim;
-                timespec    st_ctim;
                 extern(D) @safe @property inout pure nothrow
                 {
                     ref inout(time_t) st_atime() return { return st_atim.tv_sec; }
@@ -114,7 +114,10 @@ version (linux)
                     ref inout(time_t) st_ctime() return { return st_ctim.tv_sec; }
                 }
             }
-            static assert(stat_t.sizeof == 144);
+            version (CRuntime_Musl_Pre_Time64)
+                static assert(stat_t.sizeof == 96);
+            else
+                static assert(stat_t.sizeof == 144);
         }
         else
         {
@@ -322,27 +325,37 @@ version (linux)
                 dev_t       st_rdev;
                 c_long[2]   __pad2;
                 off_t       st_size;
-                private struct __timespec32
-                {
-                    c_long tv_sec;
-                    c_long tv_nsec;
+                version (CRuntime_Musl_Pre_Time64) {
+                    timespec    st_atim;
+                    timespec    st_mtim;
+                    timespec    st_ctim;
+                } else {
+                    private struct __timespec32
+                    {
+                        c_long tv_sec;
+                        c_long tv_nsec;
+                    }
+                    __timespec32 __st_atim32;
+                    __timespec32 __st_mtim32;
+                    __timespec32 __st_ctim32;
                 }
-                __timespec32 __st_atim32;
-                __timespec32 __st_mtim32;
-                __timespec32 __st_ctim32;
                 blksize_t   st_blksize;
                 c_long      __pad3;
                 blkcnt_t    st_blocks;
-                timespec    st_atim;
-                timespec    st_mtim;
-                timespec    st_ctim;
+                version (CRuntime_Musl_Pre_Time64) {
+                    c_long[14] __pad4;
+                } else {
+                    timespec    st_atim;
+                    timespec    st_mtim;
+                    timespec    st_ctim;
+                    c_long[2]   __pad4;
+                }
                 extern(D) @safe @property inout pure nothrow
                 {
                     ref inout(time_t) st_atime() return { return st_atim.tv_sec; }
                     ref inout(time_t) st_mtime() return { return st_mtim.tv_sec; }
                     ref inout(time_t) st_ctime() return { return st_ctim.tv_sec; }
                 }
-                c_long[2]   __pad4;
             }
         }
         else
@@ -496,7 +509,39 @@ version (linux)
             off_t       st_size;
             blksize_t   st_blksize;
             blkcnt_t    st_blocks;
-            static if (_DEFAULT_SOURCE || _XOPEN_SOURCE >= 700)
+            version (CRuntime_Musl_Pre_Time64) {
+                timespec    st_atim;
+                timespec    st_mtim;
+                timespec    st_ctim;
+                extern(D) @safe @property inout pure nothrow
+                {
+                    ref inout(time_t) st_atime() return { return st_atim.tv_sec; }
+                    ref inout(time_t) st_mtime() return { return st_mtim.tv_sec; }
+                    ref inout(time_t) st_ctime() return { return st_ctim.tv_sec; }
+                }
+                c_ulong     __unused4;
+                c_ulong     __unused5;
+            } else version (CRuntime_Musl) {
+                private struct __timespec32
+                {
+                    c_long tv_sec;
+                    c_long tv_nsec;
+                }
+                __timespec32 __st_atim32;
+                __timespec32 __st_mtim32;
+                __timespec32 __st_ctim32;
+                c_ulong     __unused4;
+                c_ulong     __unused5;
+                timespec    st_atim;
+                timespec    st_mtim;
+                timespec    st_ctim;
+                extern(D) @safe @property inout pure nothrow
+                {
+                    ref inout(time_t) st_atime() return { return st_atim.tv_sec; }
+                    ref inout(time_t) st_mtime() return { return st_mtim.tv_sec; }
+                    ref inout(time_t) st_ctime() return { return st_ctim.tv_sec; }
+                }
+            } else static if (_DEFAULT_SOURCE || _XOPEN_SOURCE >= 700)
             {
                 timespec    st_atim;
                 timespec    st_mtim;
@@ -507,6 +552,8 @@ version (linux)
                     ref inout(time_t) st_mtime() return { return st_mtim.tv_sec; }
                     ref inout(time_t) st_ctime() return { return st_ctim.tv_sec; }
                 }
+                c_ulong     __unused4;
+                c_ulong     __unused5;
             }
             else
             {
@@ -516,11 +563,15 @@ version (linux)
                 c_ulong     st_mtimensec;
                 time_t      st_ctime;
                 c_ulong     st_ctimensec;
+                c_ulong     __unused4;
+                c_ulong     __unused5;
             }
-            c_ulong     __unused4;
-            c_ulong     __unused5;
         }
-        static if (__USE_FILE_OFFSET64)
+        version (CRuntime_Musl_Pre_Time64)
+            static assert(stat_t.sizeof == 104);
+        else version(CRuntime_Musl)
+            static assert(stat_t.sizeof == 104 + timespec.sizeof * 3);
+        else static if (__USE_FILE_OFFSET64)
             static assert(stat_t.sizeof == 104);
         else
             static assert(stat_t.sizeof == 88);
@@ -653,32 +704,41 @@ version (linux)
     {
         version (CRuntime_Musl)
         {
+            // Matches struct stat from musl arch/arm/bits/stat.h
+            // See: https://git.musl-libc.org/cgit/musl/tree/arch/arm/bits/stat.h?h=v1.2.3
             struct stat_t
             {
                 dev_t st_dev;
-                ushort __pad1;
-                c_long __st_ino;
+                int __st_dev_padding;
+                c_long __st_ino_truncated;
                 mode_t st_mode;
                 nlink_t st_nlink;
                 uid_t st_uid;
                 gid_t st_gid;
                 dev_t st_rdev;
-                ushort __pad2;
+                int __st_rdev_padding;
                 off_t st_size;
                 blksize_t st_blksize;
                 blkcnt_t st_blocks;
-                private struct __timespec32
-                {
-                    c_long tv_sec;
-                    c_long tv_nsec;
+                version (CRuntime_Musl_Pre_Time64) {
+                    timespec st_atim;
+                    timespec st_mtim;
+                    timespec st_ctim;
+                    ino_t st_ino;
+                } else {
+                    private struct __timespec32
+                    {
+                        c_long tv_sec;
+                        c_long tv_nsec;
+                    }
+                    __timespec32 __st_atim32;
+                    __timespec32 __st_mtim32;
+                    __timespec32 __st_ctim32;
+                    ino_t st_ino;
+                    timespec st_atim;
+                    timespec st_mtim;
+                    timespec st_ctim;
                 }
-                __timespec32 __st_atim32;
-                __timespec32 __st_mtim32;
-                __timespec32 __st_ctim32;
-                ino_t st_ino;
-                timespec st_atim;
-                timespec st_mtim;
-                timespec st_ctim;
                 extern(D) @safe @property inout pure nothrow
                 {
                     ref inout(time_t) st_atime() return { return st_atim.tv_sec; }
@@ -686,7 +746,10 @@ version (linux)
                     ref inout(time_t) st_ctime() return { return st_ctim.tv_sec; }
                 }
             }
-            static assert(stat_t.sizeof == 152);
+            version (CRuntime_Musl_Pre_Time64)
+                static assert(stat_t.sizeof == 104);
+            else
+                static assert(stat_t.sizeof == 152);
         }
         else
         {
--- a/runtime/druntime/src/core/sys/posix/time.d
+++ b/runtime/druntime/src/core/sys/posix/time.d
@@ -466,8 +466,6 @@ else version (CRuntime_Bionic)
 }
 else version (CRuntime_Musl)
 {
-    static assert(timespec.sizeof == 16);
-
     alias int clockid_t;
     alias void* timer_t;
 
