--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -513,7 +513,11 @@ append("-DLDC_HOST_${D_COMPILER_ID}=1" L
 append("-DLDC_HOST_FE_VER=${D_COMPILER_FE_VERSION}" LDC_CXXFLAGS)
 # If the LLVM is shared, add its lib dir to the hardcoded list used for library lookups.
 if(LLVM_IS_SHARED)
+  if(DEFINED LDC_LLVM_LIBDIR)
+    append("\"-DLDC_LLVM_LIBDIR=R\\\"(${LDC_LLVM_LIBDIR})\\\"\"" LDC_CXXFLAGS)
+  else()
     append("\"-DLDC_LLVM_LIBDIR=R\\\"(${LLVM_LIBRARY_DIRS})\\\"\"" LDC_CXXFLAGS)
+  endif()
 endif()
 
 #
@@ -615,6 +619,7 @@ set(LDMD_EXE_FULL ${PROJECT_BINARY_DIR}/
 
 # Figure out how to link the main LDC executable, for which we need to take the
 # LLVM flags into account.
+set (LLVM_LIBRARIES ${LLVM_LIBRARIES} -lzstd -lz)
 set(LDC_LINKERFLAG_LIST ${SANITIZE_LDFLAGS} ${LLVM_LIBRARIES} ${LLVM_LDFLAGS})
 if(MSVC)
     # Issue 1297 – set LDC's stack to 16 MiB for Windows builds (default: 1 MiB).
@@ -701,6 +706,7 @@ if(LDC_ENABLE_PLUGINS)
         endif()
     endif()
 endif()
+set(LDC_LINKERFLAG_LIST ${LLVM_LDFLAGS} ${LDC_LINKERFLAG_LIST})
 message(STATUS "-- Building LDC with plugin support (LDC_ENABLE_PLUGINS): ${LDC_ENABLE_PLUGINS}")
 message(STATUS "-- Linking LDC with flags: ${ALTERNATIVE_MALLOC_O};${LDC_LINKERFLAG_LIST}")
 
