--- a/ldc2.conf.in
+++ b/ldc2.conf.in
@@ -18,6 +18,7 @@ default:
 {
     // default switches injected before all explicit command-line switches
     switches = [
+        "-link-defaultlib-shared",
         "-defaultlib=druntime-ldc",@ADDITIONAL_DEFAULT_LDC_SWITCHES@
     ];
     // default switches appended after all explicit command-line switches
@@ -31,7 +32,7 @@ default:
         "@CMAKE_BINARY_DIR@/lib@LIB_SUFFIX@",@OPTIONAL_COMPILER_RT_DIR@
     ];
     // default rpath when linking against the shared default libs
-    rpath = "@SHARED_LIBS_RPATH@";
+    rpath = "";
 };
 
 "^wasm(32|64)-":
--- a/ldc2_install.conf.in
+++ b/ldc2_install.conf.in
@@ -18,6 +18,7 @@ default:
 {
     // default switches injected before all explicit command-line switches
     switches = [
+        "-link-defaultlib-shared",
         "-defaultlib=phobos2-ldc,druntime-ldc",@ADDITIONAL_DEFAULT_LDC_SWITCHES@
     ];
     // default switches appended after all explicit command-line switches
@@ -29,7 +30,7 @@ default:
         "@CMAKE_INSTALL_LIBDIR@",@OPTIONAL_COMPILER_RT_DIR@
     ];
     // default rpath when linking against the shared default libs
-    rpath = "@SHARED_LIBS_INSTALL_RPATH@";
+    rpath = "";
 };
 
 "^wasm(32|64)-":
--- a/ldc2_phobos.conf.in
+++ b/ldc2_phobos.conf.in
@@ -18,6 +18,7 @@ default:
 {
     // default switches injected before all explicit command-line switches
     switches = [
+        "-link-defaultlib-shared",
         "-defaultlib=phobos2-ldc,druntime-ldc",@ADDITIONAL_DEFAULT_LDC_SWITCHES@
     ];
     // default switches appended after all explicit command-line switches
@@ -32,7 +33,7 @@ default:
         "@CMAKE_BINARY_DIR@/lib@LIB_SUFFIX@",@OPTIONAL_COMPILER_RT_DIR@
     ];
     // default rpath when linking against the shared default libs
-    rpath = "@SHARED_LIBS_RPATH@";
+    rpath = "";
 };
 
 "^wasm(32|64)-":
