--- a/ldc2.conf.header
+++ b/ldc2.conf.header
@@ -25,7 +25,7 @@
 // "default": {
 //     // default switches injected before all explicit command-line switches
 //     switches = [
-//         "-link-defaultlib-shared=false",
+//         "-link-defaultlib-shared",
 //     ];
 //     // default switches appended after all explicit command-line switches
 //     post-switches = [
@@ -36,7 +36,8 @@
 //         "/usr/local/lib/",
 //     ];
 //     // default rpath when linking against the shared default libs
-//     rpath = "%%ldcbinarypath%%/../lib";
+//     // rpath = "%%ldcbinarypath%%/../lib";
+//     rpath = "";
 // }
 //
 // // A section that only applies when compiling for WebAssembly
