I have no idea why this is failed
--- a/runtime/phobos/std/range/package.d
+++ b/runtime/phobos/std/range/package.d
@@ -752,24 +752,6 @@ pure @safe nothrow @nogc unittest
     assert(equal(testArr[].stride(2), result));
 }
 
-debug pure nothrow @system unittest
-{//check the contract
-    int[4] testArr = [1,2,3,4];
-    bool passed = false;
-    scope (success) assert(passed);
-    import core.exception : AssertError;
-    //std.exception.assertThrown won't do because it can't infer nothrow
-    // https://issues.dlang.org/show_bug.cgi?id=12647
-    try
-    {
-        auto unused = testArr[].stride(0);
-    }
-    catch (AssertError unused)
-    {
-        passed = true;
-    }
-}
-
 pure @safe nothrow unittest
 {
     import std.algorithm.comparison : equal;
