--- a/runtime/CMakeLists.txt
+++ b/runtime/CMakeLists.txt
@@ -6,6 +6,10 @@ set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PAT
 include(CheckIncludeFile)
 
 # Verify required variables if this CMake project is NOT embedded in the LDC CMake project.
+if(DEFINED XLDC_EXE_FULL)
+    set(OLDC_EXE_FULL "${LDC_EXE_FULL}")
+    set(LDC_EXE_FULL "${XLDC_EXE_FULL}")
+endif()
 if(NOT LDC_EXE)
     if(NOT LDC_EXE_FULL)
         message(FATAL_ERROR "Please define the path to the LDC executable via -DLDC_EXE_FULL=...")
@@ -1116,3 +1120,6 @@ if(MULTILIB AND ${HOST_BITNESS} EQUAL 64
 endif()
 
 include(DRuntimeIntegrationTests)
+if(DEFINED XLDC_EXE_FULL)
+    set(LDC_EXE_FULL "${OLDC_EXE_FULL}")
+endif()
