diff -ru pyside-setup-opensource-src-5.15.6.orig/sources/shiboken2/libshiboken/embed/signature_bootstrap.py pyside-setup-opensource-src-5.15.6/sources/shiboken2/libshiboken/embed/signature_bootstrap.py
--- pyside-setup-opensource-src-5.15.6.orig/sources/shiboken2/libshiboken/embed/signature_bootstrap.py	2022-09-08 23:54:19.419724864 +0200
+++ pyside-setup-opensource-src-5.15.6/sources/shiboken2/libshiboken/embed/signature_bootstrap.py	2022-09-08 23:55:04.494277606 +0200
@@ -211,7 +211,7 @@
         return self if self._mod2path.get(fullname) else None
 
     def load_module(self, fullname):
-        import importlib
+        import importlib.machinery
         import sys
 
         filename = self._mod2path.get(fullname)
