--- setup.py.orig	2021-01-30 17:47:39.874134606 -0500
+++ setup.py	2021-01-30 17:47:54.435139352 -0500
@@ -74,10 +74,6 @@
     entry_points={
         "console_scripts": [
             "pip=pip._internal.cli.main:main",
-            "pip{}=pip._internal.cli.main:main".format(sys.version_info[0]),
-            "pip{}.{}=pip._internal.cli.main:main".format(
-                *sys.version_info[:2]
-            ),
         ],
     },
 
