diff --git a/tests/meson.build b/tests/meson.build
index 369abac8e7..9086609c87 100644
--- a/tests/meson.build
+++ b/tests/meson.build
@@ -45,13 +45,24 @@ test_install_dir = join_paths(get_option('datadir'), 'gobject-introspection-1.0'
 
 install_data(installed_sources, install_dir: test_install_dir)
 
+if get_option('enable-host-gi')
+    ev_scanner_command = [
+      'g-ir-scanner',
+    ]
+else
+    ev_scanner_command = [
+      python,
+      girscanner,
+    ]
+endif
+
+
 # Generate everything.h and everything.c:
 test_everything_files = custom_target('everything',
   output: ['everything.h', 'everything.c'],
   depends: typelibs,
   command: [
-    python,
-    girscanner,
+    ev_scanner_command,
     '--quiet',
     '--generate-typelib-tests=Everything,@OUTPUT0@,@OUTPUT1@',
     '--function-decoration=_GI_TEST_EXTERN',
@@ -109,14 +120,7 @@ if glib_dep.type_name() == 'pkgconfig'
     output: 'Everything-1.0.gir',
     depends: [everything_lib] + typelibs,
     command: [
-      python,
-      girscanner,
-      '--quiet',
-      '--output=@OUTPUT@',
-      '--no-libtool',
-      '--reparse-validate',
-      '--add-include-path', join_paths(meson.source_root(), 'gir'),
-      '--add-include-path', join_paths(meson.build_root(), 'gir'),
+      scanner_command,
       '--warn-all',
       '--warn-error',
       '--namespace=Everything',
@@ -134,14 +138,7 @@ if glib_dep.type_name() == 'pkgconfig'
     output: 'GIMarshallingTests-1.0.gir',
     depends: [gimarshallingtests_lib] + typelibs,
     command: [
-      python,
-      girscanner,
-      '--quiet',
-      '--output=@OUTPUT@',
-      '--no-libtool',
-      '--reparse-validate',
-      '--add-include-path', join_paths(meson.source_root(), 'gir'),
-      '--add-include-path', join_paths(meson.build_root(), 'gir'),
+      scanner_command,
       '--warn-all',
       '--warn-error',
       '--namespace=GIMarshallingTests',
