--- m4/python.m4.orig	2018-07-31 14:35:16.536402563 +0200
+++ m4/python.m4	2018-07-31 14:35:23.033408078 +0200
@@ -43,7 +43,7 @@ AC_DEFUN([AM_CHECK_PYTHON_HEADERS],
 [AC_REQUIRE([AM_PATH_PYTHON])
 AC_MSG_CHECKING(for headers required to compile python extensions)
 dnl deduce PYTHON_INCLUDES
-PYTHON_INCLUDES=`$PYTHON-config --includes`
+PYTHON_INCLUDES="-I${XBPS_CROSS_BASE}/include/python${PYTHON_VERSION}"
 AC_SUBST(PYTHON_INCLUDES)
 dnl check if the headers exist:
 save_CPPFLAGS="$CPPFLAGS"
--- Makefile-gir.am.orig	2018-07-31 14:39:06.856598070 +0200
+++ Makefile-gir.am	2018-07-31 14:40:09.763651457 +0200
@@ -55,8 +55,8 @@ endif
 endif
 
 # glib
-GLIB_INCLUDEDIR=$(shell "${PKG_CONFIG}" --variable=includedir glib-2.0)/glib-2.0
-GLIB_LIBDIR=$(shell "${PKG_CONFIG}" --variable=libdir glib-2.0)
+GLIB_INCLUDEDIR=${XBPS_CROSS_BASE}/usr/include/glib-2.0
+GLIB_LIBDIR=${XBPS_CROSS_BASE}/usr/lib
 
 GLIB_LIBRARY=glib-2.0
 
@@ -155,7 +155,7 @@ GIO_LIBDIR=$(shell "${PKG_CONFIG}" --var
 GIO_LIBRARY=gio-2.0
 
 if HAVE_GIO_UNIX
-GIO_UNIX_INCLUDEDIR = $(shell "${PKG_CONFIG}" --variable=includedir gio-unix-2.0)/gio-unix-2.0
+GIO_UNIX_INCLUDEDIR = ${XBPS_CROSS_BASE}/usr/include/gio-unix-2.0
 GIO_UNIX_HDRS = $(GIO_UNIX_INCLUDEDIR)/gio/*.h
 GIO_UNIX_PACKAGES = gio-unix-2.0
 else
--- common.mk.orig	2018-07-31 15:36:19.070510317 +0200
+++ common.mk	2018-07-31 15:36:33.636522675 +0200
@@ -6,7 +6,7 @@
 # module itself.
 #
 
-INTROSPECTION_SCANNER = \
+INTROSPECTION_SCANNER ?= \
     env PATH=".libs:$(PATH)" \
         LPATH=.libs \
         CC="$(CC)" \
@@ -27,7 +27,7 @@ INTROSPECTION_SCANNER_ARGS = \
 # GI_CROSS_LAUNCHER is the command to use for executing g-ir-compiler.
 # Normally will be undefined but can be set (e.g. to wine or qemu)
 # when cross-compiling
-INTROSPECTION_COMPILER = \
+INTROSPECTION_COMPILER ?= \
     env PATH=".libs:$(PATH)" \
         $(GI_CROSS_LAUNCHER) \
         $(top_builddir)/g-ir-compiler$(EXEEXT)
@@ -39,7 +39,7 @@ INTROSPECTION_COMPILER_ARGS = \
     --includedir=$(top_builddir) \
     --includedir=$(top_builddir)/gir
 
-INTROSPECTION_DOCTOOL = \
+INTROSPECTION_DOCTOOL ?= \
     env PATH=".libs:$(PATH)" \
         LPATH=.libs \
         PYTHONPATH=$(top_builddir):$(top_srcdir) \
