From 2767461bc4509440be3a9b8f553b9a0bea468e6b Mon Sep 17 00:00:00 2001
From: Tom Hacohen <tom@stosb.com>
Date: Mon, 23 Nov 2020 11:59:53 +0200
Subject: [PATCH] Dynamically link against libssl.

---
 Cargo.toml | 1 -
 1 file changed, 1 deletion(-)

diff --git a/Cargo.toml b/Cargo.toml
index 5700454..d525653 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -19,5 +19,4 @@ log = "^0.4.6"

 [dependencies]
 cpython = { version = "0.7", features = ["extension-module"] }
-openssl = { version = "0.10.30", features = ["vendored"] }
 etebase = { version = "^0.5.1" }
--
2.28.0
