diff --git a/src/openrct2/interface/InteractiveConsole.h b/src/openrct2/interface/InteractiveConsole.h
index 842c5a0..066216a 100644
--- a/src/openrct2/interface/InteractiveConsole.h
+++ b/src/openrct2/interface/InteractiveConsole.h
@@ -28,7 +28,7 @@ enum class ConsoleInput : uint8_t
 
 namespace OpenRCT2
 {
-    enum class FormatToken : uint8_t;
+    enum class FormatToken;
     struct TextInputSession;
 } // namespace OpenRCT2
 
diff --git a/src/openrct2/localisation/FormatCodes.h b/src/openrct2/localisation/FormatCodes.h
index bb0ccb7..94b32db 100644
--- a/src/openrct2/localisation/FormatCodes.h
+++ b/src/openrct2/localisation/FormatCodes.h
@@ -17,7 +17,7 @@
 
 namespace OpenRCT2
 {
-    enum class FormatToken : uint8_t
+    enum class FormatToken
     {
         Unknown,
         Literal,
