diff --git a/CMakeLists.txt b/CMakeLists.txt
index 07112b0..597a764 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -229,15 +229,16 @@ set_target_properties(${ALL_TOOLS} PROPERTIES
 )
 
 # tools command
+file(MAKE_DIRECTORY ${DATA_BIN_DIR})
 add_custom_command(
     OUTPUT
         ${ALL_DATA}
-    COMMAND ${CMAKE_COMMAND} -E make_directory ${DATA_BIN_DIR}
-    COMMAND ${CMAKE_COMMAND} -E chdir ${DATA_BIN_DIR} ${TOOLS_BIN_DIR}/init_database ${DATA_SRC_DIR}/phone.cin ${DATA_SRC_DIR}/tsi.src
+    COMMAND init_database ${DATA_SRC_DIR}/phone.cin ${DATA_SRC_DIR}/tsi.src
     DEPENDS
         ${ALL_TOOLS}
         ${DATA_SRC_DIR}/phone.cin
         ${DATA_SRC_DIR}/tsi.src
+    WORKING_DIRECTORY ${DATA_BIN_DIR}
 )
 
 # test
