diff --git a/data/dict/chewing/CMakeLists.txt b/data/dict/chewing/CMakeLists.txt
index 3d42fe3..4e87f10 100644
--- a/data/dict/chewing/CMakeLists.txt
+++ b/data/dict/chewing/CMakeLists.txt
@@ -5,7 +5,7 @@ foreach(TARGET ${TARGETS})
   string(REPLACE "dat" "csv" SRC ${TARGET})
   add_custom_command(
     OUTPUT ${TARGET}
-    COMMAND ${CHEWING_CLI} init --csv ${CMAKE_CURRENT_SOURCE_DIR}/${SRC} ${TARGET}
+    COMMAND ${CMAKE_CROSSCOMPILING_EMULATOR} ${CHEWING_CLI} init --csv ${CMAKE_CURRENT_SOURCE_DIR}/${SRC} ${TARGET}
     DEPENDS ${SRC}
   )
   install(FILES ${CMAKE_CURRENT_BINARY_DIR}/${TARGET} DESTINATION ${CMAKE_INSTALL_DATADIR}/libchewing)
