--- src/img2pdf.py
+++ src/img2pdf.py
@@ -3439,10 +3439,10 @@ RGB.""",
     outargs.add_argument(
         "--pdfa",
         nargs="?",
-        const="/usr/share/color/icc/sRGB.icc",
+        const="/usr/share/color/icc/colord/sRGB.icc",
         default=None,
         help="Output a PDF/A-1b compliant document. By default, this will "
-        "embed /usr/share/color/icc/sRGB.icc as the color profile.",
+        "embed /usr/share/color/icc/colord/sRGB.icc as the color profile.",
     )
 
     sizeargs = parser.add_argument_group(
--- src/img2pdf_test.py
+++ src/img2pdf_test.py
@@ -304,7 +304,7 @@ def compare(im1, im2, exact, icc, cmyk):
     else:
         iccargs = []
         if icc:
-            iccargs = ["-profile", "/usr/share/color/icc/sRGB.icc"]
+            iccargs = ["-profile", "/usr/share/color/icc/colord/sRGB.icc"]
         psnr = subprocess.run(
             ["compare"]
             + iccargs
@@ -424,7 +424,7 @@ def compare_pdfimages_png(tmpdir, img, pdf, exact=True, icc=False):
                     "PSNR",
                     "(",
                     "-profile",
-                    "/usr/share/color/icc/ghostscript/srgb.icc",
+                    "/usr/share/color/icc/colord/sRGB.icc",
                     "-depth",
                     "8",
                     str(img),
