--- a/public_suffix.gemspec
+++ b/public_suffix.gemspec
@@ -15,8 +15,8 @@ Gem::Specification.new do |s|
   s.required_ruby_version = ">= 2.3"
 
   s.require_paths    = ["lib"]
-  s.files            = `git ls-files`.split("\n")
-  s.test_files       = `git ls-files -- {test,spec,features}/*`.split("\n")
+  s.files            = `find . -type f -printf '%P\\n'`.split("\n")
+  s.test_files       = `find test spec features -type f`.split("\n")
   s.extra_rdoc_files = %w( LICENSE.txt )
 
   s.add_development_dependency "rake"
