--- Makefile.orig	2019-05-29 19:03:53.594802724 +0200
+++ Makefile	2019-05-29 19:11:57.443458314 +0200
@@ -1,11 +1,12 @@
 CPP	      = g++
+
 ifeq ($(ODROID),1)
 CXXFLAGS      = -O3 -fsigned-char -fdiagnostics-color=auto -mcpu=cortex-a9 -mfpu=neon -mfloat-abi=hard -fsingle-precision-constant -g -ffast-math -I./ -I/usr/include/GL -Wno-write-strings -DODROID
 LDFLAGS       = -lGL -lm
 DEST          = /usr/local
 else ifeq ($(LINUX),1)
-CXXFLAGS      = -O2 -g -I./ -I/usr/include/GL -Wno-write-strings
-LDFLAGS       = -lGL -lm
+CXXFLAGS      ?= -O2 -g -I./ -I/usr/include/GL -Wno-write-strings
+LIBS	       = -lGL -lm
 DEST          = /usr/local
 else ifeq ($(AMIGAOS4),1)
 ifeq ($(SDL2),1)
@@ -50,9 +51,9 @@ LDFLAGS      += -lfmod
 else
 CXXFLAGS     += -DNO_FMOD
 ifeq ($(SDL2),1)
-LDFLAGS      += -lSDL2_mixer 
+SDL_LDFLAGS      += -lSDL2_mixer 
 else
-LDFLAGS      += -lSDL_mixer 
+SDL_LDFLAGS      += -lSDL_mixer 
 endif
 endif
 endif
@@ -76,5 +77,5 @@ clean:
 
 $(PROGRAM):     $(OBJS) $(LIBS)
 		@echo "Linking $(PROGRAM) ..."
-		$(LD) -o $(PROGRAM) $(OBJS) $(LDFLAGS) $(SDL_LDFLAGS)
+		$(LD) -o $(PROGRAM) $(OBJS) $(LDFLAGS) $(LIBS) $(SDL_LDFLAGS)
 
