https://bugs.webkit.org/show_bug.cgi?id=221709

diff --git a/Source/WebKit/UIProcess/Launcher/glib/BubblewrapLauncher.cpp b/Source/WebKit/UIProcess/Launcher/glib/BubblewrapLauncher.cpp
index 69b4dc6a6a0b..dc660f399257 100644
--- Source/WebKit/UIProcess/Launcher/glib/BubblewrapLauncher.cpp
+++ Source/WebKit/UIProcess/Launcher/glib/BubblewrapLauncher.cpp
@@ -758,9 +758,15 @@ GRefPtr<GSubprocess> bubblewrapSpawn(GSubprocessLauncher* launcher, const Proces
         "--ro-bind-try", "/usr/lib", "/usr/lib",
         "--ro-bind-try", "/usr/local/lib", "/usr/local/lib",
         "--ro-bind-try", LIBDIR, LIBDIR,
+#if CPU(ADDRESS64)
         "--ro-bind-try", "/lib64", "/lib64",
         "--ro-bind-try", "/usr/lib64", "/usr/lib64",
         "--ro-bind-try", "/usr/local/lib64", "/usr/local/lib64",
+#else
+        "--ro-bind-try", "/lib32", "/lib32",
+        "--ro-bind-try", "/usr/lib32", "/usr/lib32",
+        "--ro-bind-try", "/usr/local/lib32", "/usr/local/lib32",
+#endif
 
         "--ro-bind-try", PKGLIBEXECDIR, PKGLIBEXECDIR,
     };
