--- Source/cmake/OptionsGTK.cmake.orig	2019-02-10 18:04:07.898417778 +0100
+++ Source/cmake/OptionsGTK.cmake	2019-02-10 18:04:36.673415569 +0100
@@ -357,7 +357,6 @@
 # Override the cached variables, gtk-doc and gobject-introspection do not really work when cross-building.
 if (CMAKE_CROSSCOMPILING)
     set(ENABLE_GTKDOC OFF)
-    set(ENABLE_INTROSPECTION OFF)
 endif ()
 
 # Override the cached variable, gtk-doc does not really work when building on Mac.
