--- Source/JavaScriptCore/heap/RegisterState.h	2018-09-16 19:01:30.092094651 +0200
+++ Source/JavaScriptCore/heap/RegisterState.h	2018-09-16 19:11:50.022069933 +0200
@@ -166,7 +166,7 @@
 #pragma warning(disable: 4611)
 #endif
 #define ALLOCATE_AND_GET_REGISTER_STATE(registers) \
-    alignas(std::max(alignof(void*), alignof(RegisterState))) RegisterState registers; \
+    alignas(alignof(void*) > alignof(RegisterState) ? alignof(void*) : alignof(RegisterState)) RegisterState registers; \
     setjmp(registers)
 
 #if COMPILER(MSVC)
