--- Source/JavaScriptCore/wasm/WasmFaultSignalHandler.cpp.orig	2017-09-14 10:03:40.908890971 +0200
+++ Source/JavaScriptCore/wasm/WasmFaultSignalHandler.cpp	2017-09-14 10:04:17.084859537 +0200
@@ -112,7 +112,11 @@ void unregisterCode(void* start, void* e
 
 bool fastMemoryEnabled()
 {
+#if ENABLE(WEBASSEMBLY_FAST_MEMORY)
     return fastHandlerInstalled;
+#else
+    return false;
+#endif // ENABLE(WEBASSEMBLY_FAST_MEMORY)
 }
 
 void enableFastMemory()
