From ef8f9a0b61509401e64c717c7cc6f025f1b49300 Mon Sep 17 00:00:00 2001
From: q66 <daniel@octaforge.org>
Date: Mon, 4 Jan 2021 19:12:25 +0100
Subject: [PATCH] portable little endian check

---
 Source/ThirdParty/xdgmime/src/xdgmimemagic.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git Source/ThirdParty/xdgmime/src/xdgmimemagic.c Source/ThirdParty/xdgmime/src/xdgmimemagic.c
index 3a62b9d..9bd4e1d 100644
--- a/Source/ThirdParty/xdgmime/src/xdgmimemagic.c
+++ b/Source/ThirdParty/xdgmime/src/xdgmimemagic.c
@@ -485,8 +485,7 @@ _xdg_mime_magic_parse_magic_line (FILE              *magic_file,
 	      _xdg_mime_magic_matchlet_free (matchlet);
 	      return XDG_MIME_MAGIC_ERROR;
 	    }
-	  /* FIXME: need to get this defined in a <config.h> style file */
-#if LITTLE_ENDIAN
+#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
 	  for (i = 0; i < matchlet->value_length; i = i + matchlet->word_size)
 	    {
 	      if (matchlet->word_size == 2)
-- 
2.30.0

