diff --git a/Source/JavaScriptCore/heap/MachineStackMarker.cpp b/Source/JavaScriptCore/heap/MachineStackMarker.cpp
index 1df2874..de8beb2 100644
--- Source/JavaScriptCore/heap/MachineStackMarker.cpp
+++ Source/JavaScriptCore/heap/MachineStackMarker.cpp
@@ -566,7 +566,7 @@
 #error Unknown Architecture
 #endif
 
-#elif defined(__GLIBC__) && ENABLE(JIT)
+#elif defined(__linux__) && ENABLE(JIT)
 
 #if CPU(X86)
     return reinterpret_cast<void*>((uintptr_t) regs.machineContext.gregs[REG_ESP]);
@@ -645,7 +645,7 @@ void* MachineThreads::Thread::Registers::framePointer() const
 #error Unknown Architecture
 #endif
 
-#elif defined(__GLIBC__)
+#elif defined(__linux__) // glibc and musl
 
 // The following sequence depends on glibc's sys/ucontext.h.
 #if CPU(X86)
@@ -727,7 +727,7 @@ void* MachineThreads::Thread::Registers::instructionPointer() const
 #error Unknown Architecture
 #endif
 
-#elif defined(__GLIBC__)
+#elif defined(__linux__) // glibc and musl
 
 // The following sequence depends on glibc's sys/ucontext.h.
 #if CPU(X86)
@@ -818,7 +818,7 @@ void* MachineThreads::Thread::Registers::llintPC() const
 #error Unknown Architecture
 #endif
 
-#elif defined(__GLIBC__)
+#elif defined(__linux__) // glibc and musl
 
 // The following sequence depends on glibc's sys/ucontext.h.
 #if CPU(X86)
