--- Source/JavaScriptCore/runtime/Options.cpp	2015-09-21 09:29:46.000000000 +0200
+++ Source/JavaScriptCore/runtime/Options.cpp	2015-09-30 12:32:25.447786694 +0200
@@ -610,7 +610,8 @@
     case Options::Type::unsignedType:
         return m_entry.unsignedVal == other.m_entry.unsignedVal;
     case Options::Type::doubleType:
-        return (m_entry.doubleVal == other.m_entry.doubleVal) || (isnan(m_entry.doubleVal) && isnan(other.m_entry.doubleVal));
+        return (m_entry.doubleVal == other.m_entry.doubleVal) ||
+		(std::isnan(m_entry.doubleVal) && std::isnan(other.m_entry.doubleVal));
     case Options::Type::int32Type:
         return m_entry.int32Val == other.m_entry.int32Val;
     case Options::Type::optionRangeType:
