There is no malloc_trim() in musl libc

--- Source/WebCore/platform/linux/MemoryPressureHandlerLinux.cpp
+++ Source/WebCore/platform/linux/MemoryPressureHandlerLinux.cpp
@@ -206,8 +206,10 @@
 
 void MemoryPressureHandler::platformReleaseMemory(bool)
 {
+#if defined(__GLIBC__)
     ReliefLogger log("Run malloc_trim");
     malloc_trim(0);
+#endif
 }
 
 void MemoryPressureHandler::ReliefLogger::platformLog()
