From c9fa538715b7a40a24ed187d14995ee67a04b718 Mon Sep 17 00:00:00 2001
From: coolbean <thecoolestofbeans@protonmail.com>
Date: Fri, 9 Sep 2022 05:22:55 +0200
Subject: [PATCH] apply patch

---
 .../graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp      | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp b/Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp
index 3237520e..d612c10a 100644
--- a/Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp
+++ b/Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp
@@ -3402,6 +3402,12 @@ GstElement* MediaPlayerPrivateGStreamer::createVideoSink()
 
     if (!m_player->isVideoPlayer()) {
         m_videoSink = makeGStreamerElement("fakevideosink", nullptr);
+	if (!m_videoSink) {
+		GST_DEBUG_OBJECT(m_pipeline.get(), "Falling back to fakesink for video rendering");
+		m_videoSink = gst_element_factory_make("fakesink", nullptr);
+		g_object_set(m_videoSink.get(), "sync", TRUE, nullptr);
+	}
+	
         return m_videoSink.get();
     }
 
-- 
2.37.3

