From 7cdeb3dfbe01fc681c35e89e3f509c8e7ac9522e Mon Sep 17 00:00:00 2001
From: Jasper Chan <jasperchan515@gmail.com>
Date: Mon, 27 May 2019 18:54:59 -0700
Subject: [PATCH] Replace sys/unistd.h with unistd.h for musl

---
 src/slic3r/Utils/Serial.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git src/slic3r/Utils/Serial.cpp src/slic3r/Utils/Serial.cpp
index 601719b50..43dab2390 100644
--- a/src/slic3r/Utils/Serial.cpp
+++ b/src/slic3r/Utils/Serial.cpp
@@ -42,7 +42,7 @@
 #ifndef _WIN32
 	#include <sys/ioctl.h>
 	#include <sys/time.h>
-	#include <sys/unistd.h>
+	#include <unistd.h>
 	#include <sys/select.h>
 #endif
 
-- 
2.21.0

