--- PrusaSlicer-version_2.5.0/src/occt_wrapper/CMakeLists.txt.orig	2022-09-17 11:47:02.398209546 +0200
+++ PrusaSlicer-version_2.5.0/src/occt_wrapper/CMakeLists.txt	2022-09-17 11:47:30.092595307 +0200
@@ -19,7 +19,7 @@ include(GenerateExportHeader)
 
 generate_export_header(OCCTWrapper)
 
-find_package(OpenCASCADE 7.6.2 REQUIRED)
+find_package(OpenCASCADE REQUIRED)
 
 set(OCCT_LIBS
     TKXDESTEP
