--- src/avrdude/CMakeLists.txt	2019-05-20 18:49:36.000000000 +0200
+++ -	2019-05-31 23:42:15.542838510 +0200
@@ -82,7 +82,7 @@
 add_custom_command(
     DEPENDS avrdude-conf-gen ${CMAKE_CURRENT_SOURCE_DIR}/avrdude-slic3r.conf
     OUTPUT ${CMAKE_CURRENT_SOURCE_DIR}/avrdude-slic3r.conf.h
-    COMMAND $<TARGET_FILE:avrdude-conf-gen> avrdude-slic3r.conf avrdude_slic3r_conf > avrdude-slic3r.conf.h
+    COMMAND avrdude-conf-gen avrdude-slic3r.conf avrdude_slic3r_conf > avrdude-slic3r.conf.h
     WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
 )
 
