diff --git CMakeLists.txt CMakeLists.txt
index 52be8e847..6340b9d31 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -363,8 +363,8 @@ if (NOT SLIC3R_STATIC)
 endif ()
 if (NOT EIGEN3_FOUND)
     set(EIGEN3_FOUND 1)
-    set(EIGEN3_INCLUDE_DIR ${LIBDIR}/eigen/)
 endif ()
+set(EIGEN3_INCLUDE_DIR ${LIBDIR}/eigen/)
 include_directories(BEFORE SYSTEM ${EIGEN3_INCLUDE_DIR})
 
 # Find expat or use bundled version
