--- AVMPI/MMgcPortUnix.cpp	2020-07-13 19:29:31.368000000 +1000
+++ AVMPI/MMgcPortUnix.cpp	2020-07-13 19:30:14.934000000 +1000
@@ -46,7 +46,7 @@
     #define FLUSHWIN()
 #endif
 
-#if defined linux && !defined VMCFG_MIPS && !defined VMCFG_ARM && !defined ANDROID
+#if defined linux && defined __GLIBC__ && !defined VMCFG_MIPS && !defined VMCFG_ARM && !defined ANDROID
 #include <execinfo.h>
 #define HAVE_BACKTRACE
 #endif
--- core/exec.cpp	2020-07-13 19:31:28.076000000 +1000
+++ core/exec.cpp	2020-07-13 19:31:36.176000000 +1000
@@ -806,7 +806,7 @@
 failure:
     AvmCore* core = env->core();
     env->toplevel()->throwTypeError(kCheckTypeFailedError, core->atomToErrorString(atom), core->toErrorString(t));
-    return unreachableAtom;
+    return 0;
 }
 
 // Coerce an argument to an expected type, but keep it represented as Atom.
