diff -u -r abiword-2.8.6.orig/src/Makefile.am abiword-2.8.6/src/Makefile.am
--- src/Makefile.am	2009-09-09 15:11:01.000000000 +0200
+++ src/Makefile.am	2011-08-02 23:35:51.904761961 +0200
@@ -107,7 +107,7 @@
 
 AbiWord_LDFLAGS = \
 	$(DEPS_LIBS) \
-	--no-undefined \
+	-Wl,--no-undefined \
 	-avoid-version \
 	-export-dynamic \
 	-headerpad_max_install_names 
@@ -129,7 +129,7 @@
 
 abiword_LDFLAGS = \
 	$(platform_ldflags) \
-	--no-undefined      \
+	-Wl,--no-undefined      \
 	-avoid-version      \
 	-export-dynamic
 
