--- goffice-bits/goffice/utils/go-glib-extras.c.orig	2012-05-24 12:07:28.399969286 +0200
+++ goffice-bits/goffice/utils/go-glib-extras.c	2012-05-24 12:08:35.993335129 +0200
@@ -11,7 +11,7 @@
 #include <goffice/goffice-config.h>
 #include "go-glib-extras.h"
 
-#include <glib/gi18n-lib.h>
+#include <glib.h>
 #include <gsf/gsf-impl-utils.h>
 #include <libxml/encoding.h>
 
--- goffice-bits/goffice/app/goffice-app.h.orig	2012-05-24 12:09:06.121606751 +0200
+++ goffice-bits/goffice/app/goffice-app.h	2012-05-24 12:09:15.321462357 +0200
@@ -22,7 +22,7 @@
 #ifndef GOFFICE_APP_H
 #define GOFFICE_APP_H
 
-#include <glib/gmacros.h>
+#include <glib.h>
 
 G_BEGIN_DECLS
 
--- src/af/util/xp/ut_go_file.h.orig	2012-05-24 12:11:30.229201580 +0200
+++ src/af/util/xp/ut_go_file.h	2012-05-24 12:11:41.052939975 +0200
@@ -31,7 +31,6 @@
 
 #include <glib.h>
 #include <gsf/gsf.h>
-#include <glib/gerror.h>
 #include <time.h>
 
 G_BEGIN_DECLS
