--- src/af/xap/xp/xap_Dialog.cpp	2021-07-03 17:46:07.000000000 +0200
+++ -	2023-02-23 13:21:03.557255506 +0100
@@ -267,7 +267,7 @@
 // This function constructs and returns the window name of a modeless dialog by
 // concatenating the active frame with the dialog name
 
-	*pWindowName = (char) NULL;
+	*pWindowName = (char) 0;
 	UT_UTF8String wn = UT_UTF8String(pDialogName);
 
 	XAP_Frame* pFrame = getActiveFrame();
