diff --git CMakeLists.txt CMakeLists.txt
index df6327a..3d7fa96 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -88,16 +88,6 @@ if(NOT PYTHON_MIN_VER_FOUND)
     message(FATAL_ERROR "Python 2.7 or greater required to build VOLK")
 endif()
 
-# Mako
-if(NOT MAKO_FOUND)
-    message(FATAL_ERROR "Mako templates required to build VOLK")
-endif()
-
-# Six
-if(NOT SIX_FOUND)
-    message(FATAL_ERROR "six - python 2 and 3 compatibility library required to build VOLK")
-endif()
-
 # Boost
 if(MSVC)
     if (NOT DEFINED BOOST_ALL_DYN_LINK)
