commit e483b8506640797a09ea30c126c081ea4070b2e2
Author: q66 <daniel@octaforge.org>
Date:   Mon Feb 28 19:15:39 2022 +0100

    prevent cpu_features from shitting itself on ppc32

diff --git a/cpu_features/src/impl_ppc_linux.c b/cpu_features/src/impl_ppc_linux.c
index 13a381a..fa72eee 100644
--- a/cpu_features/src/impl_ppc_linux.c
+++ b/cpu_features/src/impl_ppc_linux.c
@@ -69,6 +69,11 @@
   LINE(PPC_SCV, scv, "scv", 0, PPC_FEATURE2_SCV)                               \
   LINE(PPC_HTM_NO_SUSPEND, htm_no_suspend, "htm-no-suspend", 0,                \
        PPC_FEATURE2_HTM_NO_SUSPEND)
+
+#ifdef PPC
+#undef PPC
+#endif
+
 #define INTROSPECTION_PREFIX PPC
 #define INTROSPECTION_ENUM_PREFIX PPC
 #include "define_introspection_and_hwcaps.inl"
