diff --git a/gnulib-tests/test-canonicalize.c b/gnulib-tests/test-canonicalize.c
index 2c4fc23..89d33ed 100644
--- a/gnulib-tests/test-canonicalize.c
+++ b/gnulib-tests/test-canonicalize.c
@@ -394,9 +394,9 @@ main (void)
     ASSERT (stat ("/", &st1) == 0);
     ASSERT (stat ("//", &st2) == 0);
     bool same = psame_inode (&st1, &st2);
-#if defined __MVS__ || defined MUSL_LIBC
-    /* On IBM z/OS and musl libc, "/" and "//" both canonicalize to
-       themselves, yet they both have st_dev == st_ino == 1.  */
+#if defined __MVS__
+    /* On IBM z/OS, "/" and "//" both canonicalize to themselves, yet they both
+        have st_dev == st_ino == 1.  */
     same = false;
 #endif
     if (same)
