diff --git a/sources/src/runtime.4822e3c3aa77eb82b2fb33c9321f923cf11ddde6/eng/packaging.targets b/sources/src/runtime.4822e3c3aa77eb82b2fb33c9321f923cf11ddde6/eng/packaging.targets
index ecf7467..3add455 100644
--- a/sources/src/runtime.4822e3c3aa77eb82b2fb33c9321f923cf11ddde6/eng/packaging.targets
+++ b/sources/src/runtime.4822e3c3aa77eb82b2fb33c9321f923cf11ddde6/eng/packaging.targets
@@ -1,7 +1,7 @@
 <Project>
 
   <PropertyGroup>
-    <EnablePackageValidation>true</EnablePackageValidation>
+    <EnablePackageValidation>false</EnablePackageValidation>
     <!-- Don't restore prebuilt packages during sourcebuild. -->
     <DisablePackageBaselineValidation Condition="'$(DotNetBuildFromSource)' == 'true'">true</DisablePackageBaselineValidation>
     <PackageValidationBaselineVersion Condition="'$(PackageValidationBaselineVersion)' == ''">$([MSBuild]::Subtract($(MajorVersion), 1)).0.0</PackageValidationBaselineVersion>
