diff -rup src/main.cpp src/main.cpp
--- src/main.cpp	2015-02-25 10:46:48.000000000 +0100
+++ src/main.cpp	2015-03-12 23:01:53.971856726 +0100
@@ -125,16 +125,7 @@ int main(int argc, char **argv) {
 		return EXIT_SUCCESS;
 	else {
 		app.Init(mode, scnName, singleRun);
-
-		// If current directory doesn't have the "scenes" directory, move
-		// to where the executable is
-		if (!boost::filesystem::exists("./scenes")) {
-			// Look for the directory where LuxMark executable is installed
-			boost::filesystem::path exePath(boost::filesystem::initial_path<boost::filesystem::path>());
-			exePath = boost::filesystem::system_complete(boost::filesystem::path(argv[0])).parent_path();
-			// Set the current directory to where the binary is
-			boost::filesystem::current_path(exePath);
-		}
+		boost::filesystem::current_path("/usr/share/luxmark");

 		// Force C locale
 		setlocale(LC_NUMERIC, "C");
