--- a/tests/auto/cmake/test_generate_qmlls_ini/main.cpp
+++ b/tests/auto/cmake/test_generate_qmlls_ini/main.cpp
@@ -54,7 +54,7 @@ void tst_generate_qmlls_ini::qmllsIniAre
     }
 
     const QString &docPath = QLibraryInfo::path(QLibraryInfo::DocumentationPath);
-    const QStringList defaultImportPaths{ QLibraryInfo::path(QLibraryInfo::QmlImportsPath) };
+    const QStringList defaultImportPaths = {qgetenv("QML2_IMPORT_PATH") + ":" + QLibraryInfo::path(QLibraryInfo::QmlImportsPath)};
     const QString noCMakeCalls = "false"_L1;
 
     QTest::addRow("subfolders") << source.absolutePath()
