--- a/tests/auto/quickcontrols2/controls/universal/BLACKLIST
+++ b/tests/auto/quickcontrols2/controls/universal/BLACKLIST
@@ -6,5 +6,6 @@
 
 # QTBUG-95750
 [RangeSlider::test_overlappingHandles]
 b2qt
 qnx
+linux
--- a/tests/auto/quickcontrols2/controls/basic/BLACKLIST
+++ b/tests/auto/quickcontrols2/controls/basic/BLACKLIST
@@ -6,5 +6,6 @@
 
 # QTBUG-95750
 [RangeSlider::test_overlappingHandles]
 b2qt
 qnx
+linux
--- a/tests/auto/quickcontrols2/controls/fusion/BLACKLIST
+++ b/tests/auto/quickcontrols2/controls/fusion/BLACKLIST
@@ -6,5 +6,6 @@
 
 # QTBUG-95750
 [RangeSlider::test_overlappingHandles]
 b2qt
 qnx
+linux
--- a/tests/auto/quickcontrols2/controls/imagine/BLACKLIST
+++ b/tests/auto/quickcontrols2/controls/imagine/BLACKLIST
@@ -6,8 +6,9 @@
 
 # QTBUG-95750
 [RangeSlider::test_overlappingHandles]
 b2qt
 qnx
+linux
 
 # QTBUG-101704
 [ToolTip::test_attachedSizeBug]
--- a/tests/auto/quickcontrols2/controls/material/BLACKLIST
+++ b/tests/auto/quickcontrols2/controls/material/BLACKLIST
@@ -6,5 +6,6 @@
 
 # QTBUG-95750
 [RangeSlider::test_overlappingHandles]
 b2qt
 qnx
+linux
