--- a/tests/auto/cmake/test_generate_qmlls_ini/main.cpp	2025-02-23 14:05:00.700920877 +0100
+++ -	2025-02-23 14:07:13.399639405 +0100
@@ -42,7 +42,7 @@
 )"_s;
 
     const QString &docPath = QLibraryInfo::path(QLibraryInfo::DocumentationPath);
-    const QString defaultImportPath = QLibraryInfo::path(QLibraryInfo::QmlImportsPath);
+    const QString defaultImportPath = qgetenv("QML2_IMPORT_PATH") + ":" + QLibraryInfo::path(QLibraryInfo::QmlImportsPath);
     {
         auto file = QFile(source.absoluteFilePath(qmllsIniName));
         QVERIFY(file.exists());
