 libguile/fports.c | 22 +++++++++++-----------
 1 file changed, 11 insertions(+), 11 deletions(-)

diff --git a/libguile/fports.c b/libguile/fports.c
index 4bdc9ac..0c7d316 100644
--- libguile/fports.c
+++ libguile/fports.c
@@ -66,8 +66,8 @@
 #define OFF_T_MAX  LONG_MAX
 #define OFF_T_MIN  LONG_MIN
 #elif SIZEOF_OFF_T == SIZEOF_LONG_LONG
-#define OFF_T_MAX  LONG_LONG_MAX
-#define OFF_T_MIN  LONG_LONG_MIN
+#define OFF_T_MAX  LLONG_MAX
+#define OFF_T_MIN  LLONG_MIN
 #else
 #error Oops, unknown OFF_T size
 #endif
