Index: setup.py
===================================================================
--- a/setup.py
+++ b/setup.py
@@ -64,13 +64,6 @@ install_requires     = ['paramiko>=1.15.
                         'six>=1.12.0',
 ]
 
-# Check that the user has installed the Python development headers
-PythonH = os.path.join(get_python_inc(), 'Python.h')
-if not os.path.exists(PythonH):
-    print("You must install the Python development headers!", file=sys.stderr)
-    print("$ apt-get install python-dev", file=sys.stderr)
-    sys.exit(-1)
-
 # Convert README.md to reStructuredText for PyPI
 long_description = ''
 try:
