--- cmake/clang_libs.cmake.orig
+++ cmake/clang_libs.cmake
@@ -8,7 +8,7 @@
 if (${_llvm_coroutines} GREATER -1)
   list(APPEND llvm_raw_libs coroutines)
 endif()
-if (${LLVM_PACKAGE_VERSION} VERSION_GREATER "5")
+if (${LLVM_PACKAGE_VERSION} VERSION_GREATER "6")
   list(APPEND llvm_raw_libs bpfasmparser)
   list(APPEND llvm_raw_libs bpfdisassembler)
 endif()
--- src/cc/bcc_debug.cc.orig
+++ src/cc/bcc_debug.cc
@@ -48,6 +48,7 @@
 using namespace llvm;
 using DWARFLineTable = DWARFDebugLine::LineTable;
 
+#if 0
 void SourceDebugger::adjustInstSize(uint64_t &Size, uint8_t byte0,
                                     uint8_t byte1) {
 #ifdef WORKAROUND_FOR_LD_PSEUDO
@@ -215,5 +216,6 @@
       src_dbg_fmap_[func_name] = src_dbg_str;
     }
 }
+#endif
 
 }  // namespace ebpf
