From 60e5c02c9163de51f5d2bc9237da36d07c93f4b1 Mon Sep 17 00:00:00 2001
From: Andrea Brancaleoni <miwaxe@gmail.com>
Date: Wed, 26 Aug 2015 01:25:42 +0200
Subject: [PATCH] force pam

---
 configure.in | 1 +
 1 file changed, 1 insertion(+)

diff --git configure.in configure.in
index 590d5b2..22ed5d7 100644
--- a/configure.in
+++ b/configure.in
@@ -865,6 +865,7 @@ if test "$usepam" != "0" && echo "$LIBS" | grep -- "-lpam" > /dev/null ; then
 else
   usepam=0
 fi
+usepam=1
 USEPAM="$usepam"
 AC_SUBST(USEPAM)

--
2.5.0

