(Patch from @classabbyamp on GitHub)
--- a/build.sh
+++ b/build.sh
@@ -1,7 +1,7 @@
 #!/bin/bash
 
-cflags="-Wall -O3 -g -std=gnu11 -fno-strict-aliasing -Isrc"
-lflags="-lSDL2 -lm"
+cflags="$CFLAGS -Wall -O3 -g -std=gnu11 -fno-strict-aliasing -Isrc"
+lflags="$LDFLAGS -lSDL2 -lm"
 
 if [[ $* == *windows* ]]; then
   platform="windows"
@@ -14,7 +14,7 @@
 else
   platform="unix"
   outfile="lite"
-  compiler="gcc"
+  compiler="$CC"
   cflags="$cflags -DLUA_USE_POSIX"
   lflags="$lflags -o $outfile"
 fi
