diff --git a/src/base/convert_to.h b/src/base/convert_to.h
index ecd2fe88c..7485958d7 100644
--- a/src/base/convert_to.h
+++ b/src/base/convert_to.h
@@ -8,6 +8,7 @@
 
 #include "base/base.h"
 #include <string>
+#include <cstdint>
 
 namespace base {
 
diff --git a/src/base/sha1.h b/src/base/sha1.h
index 435faae24..1f0a5b873 100644
--- a/src/base/sha1.h
+++ b/src/base/sha1.h
@@ -8,6 +8,7 @@
 
 #include <vector>
 #include <string>
+#include <cstdint>
 
 extern "C" struct SHA1Context;
 
diff --git a/src/gfx/region.h b/src/gfx/region.h
index a59fb0011..301f21b87 100644
--- a/src/gfx/region.h
+++ b/src/gfx/region.h
@@ -9,6 +9,7 @@
 #include "gfx/rect.h"
 #include <vector>
 #include <iterator>
+#include <cstdint>
 
 namespace gfx {
 
