From 9a5721c8f3f67c2e549fb713051af92c69b30a7d Mon Sep 17 00:00:00 2001
From: Hannah von Reth <hannah.vonreth@owncloud.com>
Date: Fri, 18 Oct 2024 10:11:27 +0200
Subject: [PATCH] Qt6.8 build fix

---
 test/testcredentialmanager.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/test/testcredentialmanager.cpp b/test/testcredentialmanager.cpp
index 8881cb98388..90bead31e2a 100644
--- a/test/testcredentialmanager.cpp
+++ b/test/testcredentialmanager.cpp
@@ -112,8 +112,8 @@ private Q_SLOTS:
         QTest::qWait(1000);
         for (const auto s : spies) {
             QCOMPARE(s->count(), 1);
-            s->deleteLater();
         }
+        qDeleteAll(spies);
         spies.clear();
         {
             auto jobs = creds->clear(QStringLiteral("foo"));
@@ -131,8 +131,8 @@ private Q_SLOTS:
             QTest::qWait(1000);
             for (const auto s : spies) {
                 QCOMPARE(s->count(), 1);
-                s->deleteLater();
             }
+            qDeleteAll(spies);
         }
     }
 };
