From 133abb285359b6b2c9c4deab9563bd1763deab60 Mon Sep 17 00:00:00 2001
From: Hannah von Reth <hannah.vonreth@owncloud.com>
Date: Mon, 15 Jun 2020 12:45:18 +0200
Subject: [PATCH] Fix build with Qt 5.15

---
 src/libsync/networkjobs.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/libsync/networkjobs.cpp b/src/libsync/networkjobs.cpp
index d1e877df17..d1d75bad3e 100644
--- src/libsync/networkjobs.cpp
+++ src/libsync/networkjobs.cpp
@@ -31,6 +31,7 @@
 #include <QJsonObject>
 #ifndef TOKEN_AUTH_ONLY
 #include <QPainter>
+#include <QPainterPath>
 #endif
 
 #include "networkjobs.h"
