Index: zziplib-0.13.72/zzip/CMakeLists.txt
===================================================================
--- zziplib-0.13.72.orig/zzip/CMakeLists.txt
+++ zziplib-0.13.72/zzip/CMakeLists.txt
@@ -190,15 +190,18 @@ target_include_directories (libzzipmmapp
 endif()
 
 set_target_properties(libzzip PROPERTIES OUTPUT_NAME "zzip" RELEASE_POSTFIX "-${RELNUM}")
+set_target_properties(libzzip PROPERTIES OUTPUT_NAME "zzip" NONE_POSTFIX "-${RELNUM}")
 SET_TARGET_PROPERTIES(libzzip PROPERTIES VERSION ${VERNUM}.${FIXNUM} SOVERSION ${VERNUM})
 
 if(ZZIPFSEEKO)
 set_target_properties(libzzipfseeko PROPERTIES OUTPUT_NAME "zzipfseeko" RELEASE_POSTFIX "-${RELNUM}")
+set_target_properties(libzzipfseeko PROPERTIES OUTPUT_NAME "zzipfseeko" NONE_POSTFIX "-${RELNUM}")
 SET_TARGET_PROPERTIES(libzzipfseeko PROPERTIES VERSION ${VERNUM}.${FIXNUM} SOVERSION ${VERNUM})
 endif()
 
 if(ZZIPMMAPPED)
 set_target_properties(libzzipmmapped PROPERTIES OUTPUT_NAME "zzipmmapped" RELEASE_POSTFIX "-${RELNUM}")
+set_target_properties(libzzipmmapped PROPERTIES OUTPUT_NAME "zzipmmapped" NONE_POSTFIX "-${RELNUM}")
 SET_TARGET_PROPERTIES(libzzipmmapped PROPERTIES VERSION ${VERNUM}.${FIXNUM} SOVERSION ${VERNUM})
 endif()
 
Index: zziplib-0.13.72/zzipwrap/CMakeLists.txt
===================================================================
--- zziplib-0.13.72.orig/zzipwrap/CMakeLists.txt
+++ zziplib-0.13.72/zzipwrap/CMakeLists.txt
@@ -49,6 +49,7 @@ target_link_libraries(libzzipwrap libzzi
 target_include_directories(libzzipwrap PRIVATE "${CMAKE_SOURCE_DIR}" "${CMAKE_BINARY_DIR}" "${CMAKE_CURRENT_SOURCE_DIR}")
 
 set_target_properties(libzzipwrap PROPERTIES OUTPUT_NAME "zzipwrap" RELEASE_POSTFIX "-${RELNUM}")
+set_target_properties(libzzipwrap PROPERTIES OUTPUT_NAME "zzipwrap" NONE_POSTFIX "-${RELNUM}")
 SET_TARGET_PROPERTIES(libzzipwrap PROPERTIES VERSION ${VERNUM}.${FIXNUM} SOVERSION ${VERNUM})
 set_target_properties(libzzipwrap PROPERTIES DEFINE_SYMBOL "libzzip_EXPORTS")
 
