--- a/src/d_main.cpp.old	2022-10-23 21:12:16.501173842 -0400
+++ b/src/d_main.cpp	2022-10-23 21:13:13.549777412 -0400
@@ -3534,6 +3534,8 @@
 	
 	std::set_new_handler(NewFailure);
 	const char *batchout = Args->CheckValue("-errorlog");
+
+	D_DoomInit();
 	
 	// [RH] Make sure zdoom.pk3 is always loaded,
 	// as it contains magic stuff we need.
@@ -3566,9 +3568,7 @@
 		Printf("\n");
 	}
 
-	if (!batchrun) Printf(PRINT_LOG, "%s version %s\n", GAMENAME, GetVersionString());
-
-	D_DoomInit();
+	Printf("%s version %s\n", GAMENAME, GetVersionString());
 
 	extern void D_ConfirmSendStats();
 	D_ConfirmSendStats();
