diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 26c617e..e162638 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1361,11 +1361,11 @@ endif()
 if( ZD_CMAKE_COMPILER_IS_GNUCXX_COMPATIBLE )
 	# Need to enable intrinsics for these files.
 	if( SSE_MATTERS )
-		set_property( SOURCE
+		set_source_files_properties(
 			common/rendering/polyrenderer/poly_all.cpp
 			rendering/swrenderer/r_all.cpp
-			utility/palette.cpp
-			utility/x86.cpp
+			common/utility/palette.cpp
+			common/utility/x86.cpp
 			APPEND_STRING PROPERTY COMPILE_FLAGS " -msse2 -mmmx" )
 	endif()
 endif()
diff --git a/src/common/utility/palette.cpp b/src/common/utility/palette.cpp
index f2a2987..b8de208 100644
--- a/src/common/utility/palette.cpp
+++ b/src/common/utility/palette.cpp
@@ -33,6 +33,8 @@
 */
 
 #include <algorithm>
+#include <cfloat>
+#include <cmath>
 #include "palutil.h"
 #include "palentry.h"
 #include "sc_man.h"
