--- src/utility/palette.cpp	2019-10-20 16:16:08.000000000 +0200
+++ -	2019-10-30 16:45:10.882234214 +0100
@@ -34,6 +34,10 @@
 
 #include "palette.h"
 #include "palentry.h"
+#include <cfloat>
+#include <cstring>
+#include <cstdlib>
+#include <cmath>

 /****************************/
 /* Palette management stuff */
--- src/CMakeLists.txt	2019-10-30 16:50:46.769763172 +0100
+++ -	2019-10-30 17:11:18.507995925 +0100
@@ -1210,7 +1210,7 @@
 	endif()
 endif()
 
-target_link_libraries( zdoom ${ZDOOM_LIBS} gdtoa dumb lzma adl opn timidity timidityplus wildmidi oplsynth zmusic )
+target_link_libraries( zdoom ${ZDOOM_LIBS} dl gdtoa dumb lzma adl opn timidity timidityplus wildmidi oplsynth zmusic )
 
 include_directories( .
 	g_statusbar
--- libraries/zmusic/CMakeLists.txt	2019-10-20 16:16:08.000000000 +0200
+++ -	2019-10-30 17:15:11.581747039 +0100
@@ -94,7 +94,7 @@
 	zmusic/zmusic.cpp
 	${PLAT_SOURCES}
 	)
-target_link_libraries( zmusic adl dumb gme oplsynth opn timidity timidityplus wildmidi )
+target_link_libraries( zmusic dl adl dumb gme oplsynth opn timidity timidityplus wildmidi )
 
 if( NOT DYN_SNDFILE AND SNDFILE_FOUND )
 	include_directories( "${SNDFILE_INCLUDE_DIRS}" )
