diff --git a/src/gameconfigfile.cpp b/src/gameconfigfile.cpp
index b61279ebc..3577442be 100644
--- a/src/gameconfigfile.cpp
+++ b/src/gameconfigfile.cpp
@@ -124,6 +124,8 @@ FGameConfigFile::FGameConfigFile ()
 		SetValueForKey ("Path", "/usr/local/share/doom", true);
 		SetValueForKey ("Path", "/usr/local/share/games/doom", true);
 		SetValueForKey ("Path", "/usr/share/doom", true);
+		// Adds the correct locations of the pk3file for Void Linux
+		SetValueForKey ("Path", "/usr/share/gzdoom", true);
 		SetValueForKey ("Path", "/usr/share/games/doom", true);
 		SetValueForKey ("Path", SHARE_DIR "/doom", true);
 		SetValueForKey ("Path", SHARE_DIR "/games/doom", true);
@@ -151,6 +153,8 @@ FGameConfigFile::FGameConfigFile ()
 		SetValueForKey ("Path", "/usr/local/share/doom", true);
 		SetValueForKey ("Path", "/usr/local/share/games/doom", true);
 		SetValueForKey ("Path", "/usr/share/doom", true);
+		// Adds the correct locations of the pk3file for Void Linux
+		SetValueForKey ("Path", "/usr/share/gzdoom", true);
 		SetValueForKey ("Path", "/usr/share/games/doom", true);
 #endif
 		SetValueForKey ("Path", "$DOOMWADDIR", true);
