diff --git a/src/gameconfigfile.cpp b/src/gameconfigfile.cpp
index 0b3931e..c6856cc 100644
--- a/src/gameconfigfile.cpp
+++ b/src/gameconfigfile.cpp
@@ -124,6 +124,8 @@ FGameConfigFile::FGameConfigFile ()
 		SetValueForKey ("Path", "/usr/local/share/doom", true);
 		SetValueForKey ("Path", "/usr/local/share/games/doom", true);
 		SetValueForKey ("Path", "/usr/share/doom", true);
+                // Adds the correct locations of the pk3file for Voidlinux
+		SetValueForKey ("Path", "/usr/share/gzdoom", true);
 		SetValueForKey ("Path", "/usr/share/games/doom", true);
 #endif
 	}
@@ -146,6 +148,8 @@ FGameConfigFile::FGameConfigFile ()
 		SetValueForKey ("Path", "/usr/local/share/doom", true);
 		SetValueForKey ("Path", "/usr/local/share/games/doom", true);
 		SetValueForKey ("Path", "/usr/share/doom", true);
+                // Adds the correct locations of the pk3file for Voidlinux
+		SetValueForKey ("Path", "/usr/share/gzdoom", true);
 		SetValueForKey ("Path", "/usr/share/games/doom", true);
 #endif
 		SetValueForKey ("Path", "$DOOMWADDIR", true);
