--- a/cmake/FindIsoCodes.cmake
+++ b/cmake/FindIsoCodes.cmake
@@ -20,14 +20,17 @@ pkg_check_modules(PC_ISOCODES iso-codes)
 
 find_file(ISOCODES_ISO639_2_JSON iso_639-2.json
           HINTS "${PC_ISOCODES_PREFIX}/share/iso-codes/json/"
+          NO_CMAKE_FIND_ROOT_PATH
           )
 
 find_file(ISOCODES_ISO639_3_JSON iso_639-3.json
           HINTS "${PC_ISOCODES_PREFIX}/share/iso-codes/json/"
+          NO_CMAKE_FIND_ROOT_PATH
           )
 
 find_file(ISOCODES_ISO639_5_JSON iso_639-5.json
           HINTS "${PC_ISOCODES_PREFIX}/share/iso-codes/json/"
+          NO_CMAKE_FIND_ROOT_PATH
           )
 
 include(FindPackageHandleStandardArgs)
