Remove this nonsense to allow cross compilation...

--- configure.in.orig	2015-01-04 09:56:34.986063527 +0100
+++ configure.in	2015-01-04 09:56:43.842149084 +0100
@@ -44,15 +44,6 @@ if test -e Makefile.config; then
 fi
 
 # ===================================================================
-# Load feature configuration file and start logging features.
-# ===================================================================
-
-features="features.conf"
-AC_CHECK_FILE("$srcdir/$features", [ . $srcdir/$features ])
-AC_CHECK_FILE("$builddir/$features", [ . $builddir/$features ])
-echo "Feature summary:" > features.log
-
-# ===================================================================
 # Checks for programs.
 # ===================================================================
 
