--- a/src/loaders/SILLYPNGImageLoader.cpp	2006-06-19 00:42:12.000000000 +0200
+++ b/src/loaders/SILLYPNGImageLoader.cpp	2015-08-03 15:39:36.102606369 +0200
@@ -64,10 +64,8 @@
                         png_const_charp error)
 {
     //  printf("PNG Error: %s\n", error);
-    // copied from libpng's pngerror.cpp
-    jmp_buf buf;
-    memcpy(buf, png_ptr->jmpbuf, sizeof(jmp_buf));
-    longjmp(buf, 1);
+    // use png-1.6.18 png_longjmp() function
+    png_longjmp(png_ptr, 1);
 }
 
 
