--- squashfs-tools/action.c
+++ squashfs-tools/action.c
@@ -1905,6 +1905,9 @@
 	return 1;
 }
 
+#ifndef FNM_EXTMATCH
+#define FNM_EXTMATCH 0
+#endif
 
 TEST_FN(name, ACTION_ALL_LNK, \
 	return fnmatch(atom->argv[0], action_data->name,
--- squashfs-tools/mksquashfs.c
+++ squashfs-tools/mksquashfs.c
@@ -4391,6 +4391,9 @@
 	return paths;
 }
 
+#ifndef FNM_EXTMATCH
+#define FNM_EXTMATCH 0
+#endif
 
 int excluded_match(char *name, struct pathname *path, struct pathnames **new)
 {
--- squashfs-tools/pseudo.c
+++ squashfs-tools/pseudo.c
@@ -32,6 +32,7 @@
 #include <stdlib.h>
 #include <sys/types.h>
 #include <sys/wait.h>
+#include <sys/stat.h>
 #include <ctype.h>
 
 #include "pseudo.h"
--- squashfs-tools/unsquashfs.c
+++ squashfs-tools/unsquashfs.c
@@ -1410,6 +1410,9 @@
 	free(paths);
 }
 
+#ifndef FNM_EXTMATCH
+#define FNM_EXTMATCH 0
+#endif
 
 int matches(struct pathnames *paths, char *name, struct pathnames **new)
 {
