--- ./squashfs-tools/unsquashfs.c.orig
+++ ./squashfs-tools/unsquashfs.c
@@ -2099,7 +2099,9 @@
  */
 void *inflator(void *arg)
 {
-	char tmp[block_size];
+	char *tmp = malloc(block_size);
+	if(tmp == NULL)
+		EXIT_UNSQUASH("Out of memory allocating block buffer\n");
 
 	while(1) {
 		struct cache_entry *entry = queue_get(to_inflate);
@@ -2122,6 +2124,7 @@
  		 */ 
 		cache_block_ready(entry, res == -1);
 	}
+	free(tmp);
 }
 
 
