--- a/src/backend/main.ts
+++ b/src/backend/main.ts
@@ -208,7 +208,9 @@ async function initializeWindow(): Promi
     detectVCRedist(mainWindow)
   }
 
-  if (!app.isPackaged && process.env.CI !== 'e2e') {
+  if (!app.isPackaged &&
+      process.env.CI !== 'e2e' &&
+      process.env.HEROIC_NO_REACT_DEVTOOLS != undefined) {
     if (!process.env.HEROIC_NO_REACT_DEVTOOLS) {
       import('electron-devtools-installer').then((devtools) => {
         const { default: installExtension, REACT_DEVELOPER_TOOLS } = devtools
