--- a/Makefile	2024-02-13 15:37:10.490011066 -0300
+++ b/Makefile	2024-02-15 18:42:58.568694290 -0300
@@ -85,9 +85,9 @@
 install_unix:
 	mkdir -p $(EXECPREFIX)
 	cp $(EXECNAME) $(EXECPREFIX)/$(EXECNAME)
-	rm -rf $(PREFIX)/share/games/$(PROGNAME)
-	mkdir -p $(PREFIX)/share/games
-	cp -r assets $(PREFIX)/share/games/$(PROGNAME)
+	rm -rf $(PREFIX)/share/$(PROGNAME)
+	mkdir -p $(PREFIX)/share
+	cp -r assets $(PREFIX)/share/$(PROGNAME)
 	mkdir -p $(PREFIX)/share/pixmaps
 	cp icons/icon128.png $(PREFIX)/share/pixmaps/$(PROGNAME).png
 	mkdir -p $(PREFIX)/share/applications
