diff --git xlbiff.c xlbiff.c
index 23669f3..001e36b 100644
--- xlbiff.c
+++ xlbiff.c
@@ -235,6 +235,7 @@ main( int argc, char *argv[] )
 {
     progname = argv[0];
 
+    XtSetLanguageProc (NULL, NULL, NULL);
     topLevel = XtVaAppInitialize(&app_context,
                                 "XLbiff",
                                 optionDescList, XtNumber(optionDescList),
@@ -317,6 +318,7 @@ main( int argc, char *argv[] )
     textBox = XtVaCreateManagedWidget("text",
                                      commandWidgetClass,
                                      topLevel,
+                                      "international", True,
                                      NULL);
 
     XtAddCallback(textBox, XtNcallback, Popdown, textBox);
