--- src/libtomahawk/accounts/ResolverAccount.cpp	2016-11-08 20:50:59.907398017 +0100
+++ src/libtomahawk/accounts/ResolverAccount.cpp	2016-11-08 20:56:33.222289769 +0100
@@ -16,6 +16,9 @@
  *   You should have received a copy of the GNU General Public License
  *   along with Tomahawk. If not, see <http://www.gnu.org/licenses/>.
  */
+#if !defined(__GLIBC__)
+#include <bits/reg.h>
+#endif
 
 #include "ResolverAccount.h"
 
@@ -185,10 +185,17 @@
 #elif defined( Q_OS_MAC )
             myPlatform = "osx";
 #elif defined( Q_OS_LINUX )
+#if defined(__GLIBC__)
             if ( __WORDSIZE == 32 )
                 myPlatform = "linux-x86";
             else if ( __WORDSIZE == 64 )
                 myPlatform = "linux-x64";
+#else
+            if ( __WORDSIZE == 32 )
+                myPlatform = "linux-musl-x86";
+            else if ( __WORDSIZE == 64 )
+                myPlatform = "linux-musl-x64";
+#endif
 #endif
 
             if ( !myPlatform.contains( platform ) )
