diff --git CMakeModules/FindLibLastFm.cmake CMakeModules/FindLibLastFm.cmake
index a3488350e..545e39f9b 100644
--- CMakeModules/FindLibLastFm.cmake
+++ CMakeModules/FindLibLastFm.cmake
@@ -14,7 +14,7 @@ endif()
 # Include dir
 find_path(LIBLASTFM_INCLUDE_DIR
   # Track.h doesn't exist in liblastfm-0.3.1, was called Track back then
-  NAMES lastfm${LASTFM_LIB_SUFFIX}/Track.h
+  NAMES lastfm/Track.h
   PATHS ${KDE4_INCLUDE_DIR}
 )
 
diff --git src/libtomahawk/accounts/lastfm/LastFmConfig.cpp src/libtomahawk/accounts/lastfm/LastFmConfig.cpp
index 85b747be8..c1d2e1343 100644
--- src/libtomahawk/accounts/lastfm/LastFmConfig.cpp
+++ src/libtomahawk/accounts/lastfm/LastFmConfig.cpp
@@ -30,10 +30,10 @@
 #include "utils/NetworkAccessManager.h"
 
 #if QT_VERSION >= QT_VERSION_CHECK(5,0,0)
-#include <lastfm5/ws.h>
-#include <lastfm5/User.h>
-#include <lastfm5/XmlQuery.h>
-#include <lastfm5/Track.h>
+#include <lastfm/ws.h>
+#include <lastfm/User.h>
+#include <lastfm/XmlQuery.h>
+#include <lastfm/Track.h>
 #else
 #include <lastfm/ws.h>
 #include <lastfm/User.h>
diff --git src/libtomahawk/accounts/lastfm/LastFmInfoPlugin.cpp src/libtomahawk/accounts/lastfm/LastFmInfoPlugin.cpp
index 277a62e17..f5111fc6f 100644
--- src/libtomahawk/accounts/lastfm/LastFmInfoPlugin.cpp
+++ src/libtomahawk/accounts/lastfm/LastFmInfoPlugin.cpp
@@ -35,8 +35,8 @@
 #include "utils/NetworkAccessManager.h"
 
 #if QT_VERSION >= QT_VERSION_CHECK(5,0,0)
-#include <lastfm5/ws.h>
-#include <lastfm5/XmlQuery.h>
+#include <lastfm/ws.h>
+#include <lastfm/XmlQuery.h>
 #else
 #include <lastfm/ws.h>
 #include <lastfm/XmlQuery.h>
diff --git src/libtomahawk/accounts/lastfm/LastFmInfoPlugin.h src/libtomahawk/accounts/lastfm/LastFmInfoPlugin.h
index 0eb3b0749..c7de68e98 100644
--- src/libtomahawk/accounts/lastfm/LastFmInfoPlugin.h
+++ src/libtomahawk/accounts/lastfm/LastFmInfoPlugin.h
@@ -25,9 +25,9 @@
 #include "DllMacro.h"
 
 #if QT_VERSION >= QT_VERSION_CHECK(5,0,0)
-#include <lastfm5/Track.h>
-#include <lastfm5/Audioscrobbler.h>
-#include <lastfm5/ScrobblePoint.h>
+#include <lastfm/Track.h>
+#include <lastfm/Audioscrobbler.h>
+#include <lastfm/ScrobblePoint.h>
 #else
 #include <lastfm/Track.h>
 #include <lastfm/Audioscrobbler.h>
diff --git src/libtomahawk/utils/TomahawkUtils.cpp src/libtomahawk/utils/TomahawkUtils.cpp
index 0baa27c37..019db4af2 100644
--- src/libtomahawk/utils/TomahawkUtils.cpp
+++ src/libtomahawk/utils/TomahawkUtils.cpp
@@ -33,7 +33,7 @@
 
 #ifdef LIBLASTFM_FOUND
     #if QT_VERSION >= QT_VERSION_CHECK(5,0,0)
-    #include <lastfm5/ws.h>
+    #include <lastfm/ws.h>
     #else
     #include <lastfm/ws.h>
     #endif
diff --git src/tomahawk/Scrobbler.h src/tomahawk/Scrobbler.h
index 01c26959c..f117c896c 100644
--- src/tomahawk/Scrobbler.h
+++ src/tomahawk/Scrobbler.h
@@ -25,7 +25,7 @@
 
 
 #if QT_VERSION >= QT_VERSION_CHECK(5,0,0)
-#include <lastfm5/ScrobblePoint.h>
+#include <lastfm/ScrobblePoint.h>
 #else
 #include <lastfm/ScrobblePoint.h>
 #endif
diff --git src/tomahawk/TomahawkApp.h src/tomahawk/TomahawkApp.h
index 953140bd4..eab6b8de4 100644
--- src/tomahawk/TomahawkApp.h
+++ src/tomahawk/TomahawkApp.h
@@ -65,7 +65,7 @@ namespace Tomahawk
 
 #ifdef LIBLASTFM_FOUND
     #if QT_VERSION >= QT_VERSION_CHECK(5,0,0)
-    #include <lastfm5/NetworkAccessManager.h>
+    #include <lastfm/NetworkAccessManager.h>
     #else
     #include <lastfm/NetworkAccessManager.h>
     #endif
