--- py/emitglue.c	2018-12-20 16:15:21.717553255 +0100
+++ py/emitglue.c	2018-12-20 16:14:34.210702965 +0100
@@ -403,7 +403,7 @@
 // here we define mp_raw_code_load_file depending on the port
 // TODO abstract this away properly
 
-#if defined(__i386__) || defined(__x86_64__) || (defined(__arm__) && (defined(__unix__)))
+#if defined(__i386__) || defined(__x86_64__) || defined(__powerpc64__) || (defined(__arm__) && (defined(__unix__)))
 // unix file reader
 
 #include <sys/stat.h>
@@ -637,7 +637,7 @@
 // here we define mp_raw_code_save_file depending on the port
 // TODO abstract this away properly
 
-#if defined(__i386__) || defined(__x86_64__) || (defined(__arm__) && (defined(__unix__)))
+#if defined(__i386__) || defined(__x86_64__) || defined(__powerpc64__) || (defined(__arm__) && (defined(__unix__)))
 
 #include <unistd.h>
 #include <sys/stat.h>
