--- src/drivers/linux_wext.h
+++ src/drivers/linux_wext.h
@@ -21,11 +21,11 @@
 
 #include <sys/types.h>
 #include <net/if.h>
-typedef __uint32_t __u32;
-typedef __int32_t __s32;
-typedef __uint16_t __u16;
-typedef __int16_t __s16;
-typedef __uint8_t __u8;
+typedef u_int32_t __u32;
+typedef int32_t __s32;
+typedef u_int16_t __u16;
+typedef int16_t __s16;
+typedef u_int8_t __u8;
 #ifndef __user
 #define __user
 #endif /* __user */
