diff --git /src/meson.build b/src/meson.build
index 25eea68e811a78f8600cdea47223aab801e482a9..63fc56cf5451efcc155cb561d2cd712ef4d1db7a 100644
--- src/meson.build
+++ src/meson.build
@@ -113,7 +113,7 @@ libhandy = shared_library(
 )
 
 libhandy_dep = declare_dependency(
-              sources: libhandy_public_headers,
+              sources: libhandy_generated_headers,
          dependencies: libhandy_deps,
             link_with: libhandy,
   include_directories: include_directories('.'),
