--- a/basics.h	2019-01-23 12:32:20.420486153 +0700
+++ b/basics.h	2019-01-23 12:33:23.746320325 +0700
@@ -46,6 +46,7 @@
 
 #include <assert.h>
 #include <stdio.h>
+#include <stdint.h>
 
 #include "ladspa.h"
 
@@ -77,15 +78,6 @@
 #define MIN_GAIN .000001 /* -120 dB */
 #define NOISE_FLOOR .00000000000005 /* -266 dB */
 
-typedef __int8_t			int8;
-typedef __uint8_t			uint8;
-typedef __int16_t			int16;
-typedef __uint16_t		uint16;
-typedef __int32_t			int32;
-typedef __uint32_t		uint32;
-typedef __int64_t			int64;
-typedef __uint64_t		uint64;
-
 typedef struct {
 	const char * name;
 	LADSPA_PortDescriptor descriptor;
