--- a/libpisock/address.c	2007-01-15 01:29:59.000000000 +0100
+++ b/libpisock/address.c	2016-12-20 13:51:16.230458809 +0100
@@ -209,7 +209,7 @@
 	size_t 	i,
 		destlen = 4 + 16 * 22 + 2 + 2;
 
-	unsigned char *start = record;
+	const unsigned char *start = record;
 	unsigned long r;
 
 	ai->type = address_v1;
--- a/libpisock/memo.c	2007-01-15 01:30:00.000000000 +0100
+++ b/libpisock/memo.c	2016-12-20 13:55:47.344644240 +0100
@@ -122,7 +122,7 @@
 			 size_t len)
 {
 	int 	i = unpack_CategoryAppInfo(&appinfo->category, record, len);
-	unsigned char *start = record;
+	const unsigned char *start = record;
 
 	appinfo->type = memo_v1;
 
--- a/libpisock/todo.c	2007-01-15 01:30:00.000000000 +0100
+++ b/libpisock/todo.c	2016-12-20 14:33:36.738559189 +0100
@@ -232,7 +232,7 @@
 unpack_ToDoAppInfo(ToDoAppInfo_t *appinfo, const unsigned char *record, size_t len)
 {
 	int i;
-	unsigned char *start = record;
+	const unsigned char *start = record;
 
 	appinfo->type = todo_v1;
 
