From: Adrian Bunk <bunk@debian.org>
Date: Tue, 3 Jul 2018 10:03:37 +0100
Subject: Increase the test timeout for slower buildds

Bug-Debian: https://bugs.debian.org/878284
---
 js/src/Makefile.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/js/src/Makefile.in b/js/src/Makefile.in
index 9202f5e..20c8cf1 100644
--- a/js/src/Makefile.in
+++ b/js/src/Makefile.in
@@ -53,7 +53,7 @@ check:: check-js-msg
 
 check-jstests:
 	$(PYTHON3) -u $(srcdir)/tests/jstests.py \
-		--no-progress --format=automation --timeout 300 \
+		--no-progress --format=automation --timeout 600 \
 		$(JSTESTS_EXTRA_ARGS) \
 		$(DIST)/bin/js$(BIN_SUFFIX)
 
