--- a/js/src/jit/shared/AtomicOperations-shared-jit.cpp
+++ b/js/src/jit/shared/AtomicOperations-shared-jit.cpp
@@ -5,4 +5,9 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+// bug 1792159
+#if defined(__i386__) && defined(__GNUC__) && !defined(__clang__)
+#  include "../../../mfbt/Attributes.h"
+#endif
+
 #include "jit/AtomicOperations.h"
 
--- a/config/check_spidermonkey_style.py
+++ b/config/check_spidermonkey_style.py
@@ -68,2 +68,3 @@
         "jit/AtomicOperationsGenerated.h",  # generated in $OBJDIR
+        "../../../mfbt/Attributes.h",  # unrecognized path
         "jit/CacheIROpsGenerated.h",  # generated in $OBJDIR
