From: Simon McVittie <smcv@debian.org>
Date: Wed, 24 Aug 2022 20:52:33 +0100
Subject: js: Enable JIT by default on ARMv7 or up, but not ARMv6 or older

Bug: https://bugzilla.mozilla.org/show_bug.cgi?id=1786619
Bug-Debian: https://bugs.debian.org/1017961
Signed-off-by: Simon McVittie <smcv@debian.org>
---
 js/moz.configure | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/js/moz.configure b/js/moz.configure
index 593b93a..20d28a6 100644
--- a/js/moz.configure
+++ b/js/moz.configure
@@ -241,12 +241,14 @@ set_config(
 # =======================================================
 @depends(
     target,
+    arm_target,
     "--enable-record-tuple",
     "--enable-portable-baseline-interp",
     "--enable-explicit-resource-management",
 )
 def jit_default(
     target,
+    arm_target,
     enable_record_tuple,
     enable_portable_baseline_interp,
     enable_explicit_resource_management,
@@ -260,13 +262,16 @@ def jit_default(
     if target.cpu in (
         "x86",
         "x86_64",
-        "arm",
         "aarch64",
         "mips32",
         "mips64",
         "loongarch64",
     ):
         return True
+
+    if target.cpu == "arm" and arm_target.arm_arch >= 7:
+        return True
+
     return False
 
 
