diff --git src/mumble/ACLEditor.cpp src/mumble/ACLEditor.cpp
index c938ffc..bf321e7 100644
--- src/mumble/ACLEditor.cpp
+++ src/mumble/ACLEditor.cpp
@@ -37,11 +37,12 @@
 #include "Channel.h"
 #include "ClientUser.h"
 #include "Database.h"
-#include "Global.h"
 #include "Log.h"
 #include "ServerHandler.h"
 #include "User.h"
 
+#include "Global.h"
+
 ACLGroup::ACLGroup(const QString &name) : Group(NULL, name) {
 	bInherited = false;
 }
diff --git src/mumble/About.cpp src/mumble/About.cpp
index 4d2cf15..96db27a 100644
--- src/mumble/About.cpp
+++ src/mumble/About.cpp
@@ -32,10 +32,11 @@
 
 #include "About.h"
 
-#include "Global.h"
 #include "MainWindow.h"
 #include "licenses.h"
 
+#include "Global.h"
+
 AboutDialog::AboutDialog(QWidget *p) : QDialog(p) {
 	setWindowTitle(tr("About Mumble"));
 
diff --git src/mumble/AudioOutput.cpp src/mumble/AudioOutput.cpp
index 0695488..a9a849d 100644
--- src/mumble/AudioOutput.cpp
+++ src/mumble/AudioOutput.cpp
@@ -37,13 +37,14 @@
 #include "AudioOutputSample.h"
 #include "AudioOutputSpeech.h"
 #include "User.h"
-#include "Global.h"
 #include "Message.h"
 #include "Plugins.h"
 #include "PacketDataStream.h"
 #include "ServerHandler.h"
 #include "VoiceRecorder.h"
 
+#include "Global.h"
+
 // Remember that we cannot use static member classes that are not pointers, as the constructor
 // for AudioOutputRegistrar() might be called before they are initialized, as the constructor
 // is called from global initialization.
diff --git src/mumble/AudioWizard.cpp src/mumble/AudioWizard.cpp
index 511fd9b..25f7a33 100644
--- src/mumble/AudioWizard.cpp
+++ src/mumble/AudioWizard.cpp
@@ -34,10 +34,11 @@
 
 #include "AudioInput.h"
 #include "AudioOutputSample.h"
-#include "Global.h"
 #include "Log.h"
 #include "MainWindow.h"
 
+#include "Global.h"
+
 CompletablePage::CompletablePage(QWizard *p) : QWizardPage(p) {
 	bComplete = true;
 }
diff --git src/mumble/BanEditor.cpp src/mumble/BanEditor.cpp
index 36dad74..2b222ed 100644
--- src/mumble/BanEditor.cpp
+++ src/mumble/BanEditor.cpp
@@ -33,10 +33,11 @@
 #include "BanEditor.h"
 
 #include "Channel.h"
-#include "Global.h"
 #include "Net.h"
 #include "ServerHandler.h"
 
+#include "Global.h"
+
 BanEditor::BanEditor(const MumbleProto::BanList &msg, QWidget *p) : QDialog(p) {
 	setupUi(this);
 
diff --git src/mumble/ConnectDialog.cpp src/mumble/ConnectDialog.cpp
index 17d57e8..d6d87aa 100644
--- src/mumble/ConnectDialog.cpp
+++ src/mumble/ConnectDialog.cpp
@@ -40,10 +40,11 @@
 
 #include "Channel.h"
 #include "Database.h"
-#include "Global.h"
 #include "ServerHandler.h"
 #include "WebFetch.h"
 
+#include "Global.h"
+
 QMap<QString, QIcon> ServerItem::qmIcons;
 QList<PublicInfo> ConnectDialog::qlPublicServers;
 QString ConnectDialog::qsUserCountry, ConnectDialog::qsUserCountryCode, ConnectDialog::qsUserContinentCode;
diff --git src/mumble/CustomElements.cpp src/mumble/CustomElements.cpp
index b43f215..9a11524 100644
--- src/mumble/CustomElements.cpp
+++ src/mumble/CustomElements.cpp
@@ -34,10 +34,10 @@
 #include "CustomElements.h"
 
 #include "ClientUser.h"
-#include "Global.h"
 #include "MainWindow.h"
 #include "Log.h"
 
+#include "Global.h"
 
 LogTextBrowser::LogTextBrowser(QWidget *p)
 	: m_pos(0)
diff --git src/mumble/DBus.cpp src/mumble/DBus.cpp
index 1704395..3342f11 100644
--- src/mumble/DBus.cpp
+++ src/mumble/DBus.cpp
@@ -34,10 +34,10 @@
 
 #include "Channel.h"
 #include "ClientUser.h"
-#include "Global.h"
 #include "MainWindow.h"
 #include "ServerHandler.h"
 
+#include "Global.h"
 
 MumbleDBus::MumbleDBus(QObject *mw) : QDBusAbstractAdaptor(mw) {
 }
diff --git src/mumble/GlobalShortcut.cpp src/mumble/GlobalShortcut.cpp
index d35cf91..503a26c 100644
--- src/mumble/GlobalShortcut.cpp
+++ src/mumble/GlobalShortcut.cpp
@@ -36,9 +36,10 @@
 #include "ClientUser.h"
 #include "Channel.h"
 #include "Database.h"
-#include "Global.h"
 #include "MainWindow.h"
 
+#include "Global.h"
+
 /*!
   \class ShortcutKeyWidget
   Widget used to define and key combination for a shortcut. Once it gains
diff --git src/mumble/LCD.cpp src/mumble/LCD.cpp
index e83f692..1435a1a 100644
--- src/mumble/LCD.cpp
+++ src/mumble/LCD.cpp
@@ -35,10 +35,11 @@
 
 #include "ClientUser.h"
 #include "Channel.h"
-#include "Global.h"
 #include "Message.h"
 #include "ServerHandler.h"
 
+#include "Global.h"
+
 QList<LCDEngineNew> *LCDEngineRegistrar::qlInitializers;
 
 LCDEngineRegistrar::LCDEngineRegistrar(LCDEngineNew cons) {
diff --git src/mumble/Log.cpp src/mumble/Log.cpp
index 093f0b3..c6bdbd2 100644
--- src/mumble/Log.cpp
+++ src/mumble/Log.cpp
@@ -36,13 +36,14 @@
 #include "AudioOutput.h"
 #include "AudioOutputSample.h"
 #include "Channel.h"
-#include "Global.h"
 #include "MainWindow.h"
 #include "NetworkConfig.h"
 #include "RichTextEditor.h"
 #include "ServerHandler.h"
 #include "TextToSpeech.h"
 
+#include "Global.h"
+
 static ConfigWidget *LogConfigDialogNew(Settings &st) {
 	return new LogConfig(st);
 }
diff --git src/mumble/Log_unix.cpp src/mumble/Log_unix.cpp
index 0a18928..c64f3bf 100644
--- src/mumble/Log_unix.cpp
+++ src/mumble/Log_unix.cpp
@@ -32,10 +32,11 @@
 #include "mumble_pch.hpp"
 
 #include "Log.h"
-#include "Global.h"
 #include "MainWindow.h"
 #include "Settings.h"
 
+#include "Global.h"
+
 void Log::postNotification(MsgType mt, const QString &console, const QString &plain) {
 	// Message notification with balloon tooltips
 	QString qsIcon;
@@ -92,4 +93,4 @@ void Log::postNotification(MsgType mt, const QString &console, const QString &pl
 #endif
 		postQtNotification(mt, plain);
 	}
-}
\ No newline at end of file
+}
diff --git src/mumble/LookConfig.cpp src/mumble/LookConfig.cpp
index 5d30a2d..cddd914 100644
--- src/mumble/LookConfig.cpp
+++ src/mumble/LookConfig.cpp
@@ -35,9 +35,10 @@
 
 #include "AudioInput.h"
 #include "AudioOutput.h"
-#include "Global.h"
 #include "MainWindow.h"
 
+#include "Global.h"
+
 static ConfigWidget *LookConfigNew(Settings &st) {
 	return new LookConfig(st);
 }
diff --git src/mumble/MainWindow.cpp src/mumble/MainWindow.cpp
index eaa4573..65ab1bd 100644
--- src/mumble/MainWindow.cpp
+++ src/mumble/MainWindow.cpp
@@ -46,7 +46,6 @@
 #include "Connection.h"
 #include "ConnectDialog.h"
 #include "Database.h"
-#include "Global.h"
 #include "GlobalShortcut.h"
 #include "Log.h"
 #include "Net.h"
@@ -75,6 +74,8 @@
 #include "ConfigDialog_macx.h"
 #endif
 
+#include "Global.h"
+
 /*!
   \fn void MainWindow::findDesiredChannel()
   This function tries to join a desired channel on connect. It gets called
diff --git src/mumble/Messages.cpp src/mumble/Messages.cpp
index a9109bd..77d6a58 100644
--- src/mumble/Messages.cpp
+++ src/mumble/Messages.cpp
@@ -41,7 +41,6 @@
 #include "Connection.h"
 #include "ConnectDialog.h"
 #include "Database.h"
-#include "Global.h"
 #include "GlobalShortcut.h"
 #include "Log.h"
 #include "MainWindow.h"
@@ -55,6 +54,8 @@
 #include "VersionCheck.h"
 #include "ViewCert.h"
 
+#include "Global.h"
+
 #define ACTOR_INIT \
 	ClientUser *pSrc=NULL; \
 	if (msg.has_actor()) \
diff --git src/mumble/NetworkConfig.cpp src/mumble/NetworkConfig.cpp
index 5556e66..0cf14ac 100644
--- src/mumble/NetworkConfig.cpp
+++ src/mumble/NetworkConfig.cpp
@@ -34,10 +34,11 @@
 
 #include "NetworkConfig.h"
 
-#include "Global.h"
 #include "MainWindow.h"
 #include "OSInfo.h"
 
+#include "Global.h"
+
 static ConfigWidget *NetworkConfigNew(Settings &st) {
 	return new NetworkConfig(st);
 }
diff --git src/mumble/OSS.cpp src/mumble/OSS.cpp
index 4af1326..2611d8c 100644
--- src/mumble/OSS.cpp
+++ src/mumble/OSS.cpp
@@ -38,9 +38,10 @@
 #include <sys/ioctl.h>
 
 #include "User.h"
-#include "Global.h"
 #include "MainWindow.h"
 
+#include "Global.h"
+
 #define NBLOCKS 8
 
 class OSSEnumerator {
diff --git src/mumble/Overlay.cpp src/mumble/Overlay.cpp
index 619dd2f..d6f2b79 100644
--- src/mumble/Overlay.cpp
+++ src/mumble/Overlay.cpp
@@ -35,7 +35,6 @@
 #include "Channel.h"
 #include "ClientUser.h"
 #include "Database.h"
-#include "Global.h"
 #include "GlobalShortcut.h"
 #include "MainWindow.h"
 #include "Message.h"
@@ -45,6 +44,8 @@
 #include "User.h"
 #include "WebFetch.h"
 
+#include "Global.h"
+
 OverlayAppInfo::OverlayAppInfo(QString name, QIcon icon) {
 	qsDisplayName = name;
 	qiIcon = icon;
diff --git src/mumble/OverlayClient.cpp src/mumble/OverlayClient.cpp
index c3c408d..82412d1 100644
--- src/mumble/OverlayClient.cpp
+++ src/mumble/OverlayClient.cpp
@@ -34,7 +34,6 @@
 #include "OverlayText.h"
 #include "User.h"
 #include "Channel.h"
-#include "Global.h"
 #include "Message.h"
 #include "Database.h"
 #include "NetworkConfig.h"
@@ -42,6 +41,8 @@
 #include "MainWindow.h"
 #include "GlobalShortcut.h"
 
+#include "Global.h"
+
 OverlayClient::OverlayClient(QLocalSocket *socket, QObject *p) :
 		QObject(p),
 		ougUsers(&g.s.os) {
diff --git src/mumble/OverlayConfig.cpp src/mumble/OverlayConfig.cpp
index 9a45da3..d1f6ea4 100644
--- src/mumble/OverlayConfig.cpp
+++ src/mumble/OverlayConfig.cpp
@@ -34,7 +34,6 @@
 #include "OverlayText.h"
 #include "User.h"
 #include "Channel.h"
-#include "Global.h"
 #include "Message.h"
 #include "Database.h"
 #include "NetworkConfig.h"
@@ -42,6 +41,8 @@
 #include "MainWindow.h"
 #include "GlobalShortcut.h"
 
+#include "Global.h"
+
 static ConfigWidget *OverlayConfigDialogNew(Settings &st) {
 	return new OverlayConfig(st);
 }
diff --git src/mumble/OverlayEditor.cpp src/mumble/OverlayEditor.cpp
index 0263c1f..2d4dc2c 100644
--- src/mumble/OverlayEditor.cpp
+++ src/mumble/OverlayEditor.cpp
@@ -34,7 +34,6 @@
 #include "OverlayText.h"
 #include "User.h"
 #include "Channel.h"
-#include "Global.h"
 #include "Message.h"
 #include "Database.h"
 #include "NetworkConfig.h"
@@ -42,6 +41,8 @@
 #include "MainWindow.h"
 #include "GlobalShortcut.h"
 
+#include "Global.h"
+
 OverlayEditor::OverlayEditor(QWidget *p, QGraphicsItem *qgi, OverlaySettings *osptr) :
 		QDialog(p),
 		qgiPromote(qgi),
diff --git src/mumble/OverlayEditorScene.cpp src/mumble/OverlayEditorScene.cpp
index 95fe347..a2abb1a 100644
--- src/mumble/OverlayEditorScene.cpp
+++ src/mumble/OverlayEditorScene.cpp
@@ -34,7 +34,6 @@
 #include "OverlayText.h"
 #include "User.h"
 #include "Channel.h"
-#include "Global.h"
 #include "Message.h"
 #include "Database.h"
 #include "NetworkConfig.h"
@@ -42,6 +41,8 @@
 #include "MainWindow.h"
 #include "GlobalShortcut.h"
 
+#include "Global.h"
+
 OverlayEditorScene::OverlayEditorScene(const OverlaySettings &srcos, QObject *p) : QGraphicsScene(p), os(srcos) {
 	tsColor = Settings::Talking;
 	uiZoom = 2;
diff --git src/mumble/OverlayUser.cpp src/mumble/OverlayUser.cpp
index 979af17..5a4a4b9 100644
--- src/mumble/OverlayUser.cpp
+++ src/mumble/OverlayUser.cpp
@@ -35,7 +35,6 @@
 #include "User.h"
 #include "Channel.h"
 #include "ClientUser.h"
-#include "Global.h"
 #include "Message.h"
 #include "Database.h"
 #include "NetworkConfig.h"
@@ -43,6 +42,8 @@
 #include "MainWindow.h"
 #include "GlobalShortcut.h"
 
+#include "Global.h"
+
 OverlayUser::OverlayUser(ClientUser *cu, unsigned int height, OverlaySettings *osptr) : OverlayGroup(), os(osptr), uiSize(height), cuUser(cu), tsColor(Settings::Passive) {
 	setup();
 	updateLayout();
diff --git src/mumble/OverlayUserGroup.cpp src/mumble/OverlayUserGroup.cpp
index f8831d5..f12a113 100644
--- src/mumble/OverlayUserGroup.cpp
+++ src/mumble/OverlayUserGroup.cpp
@@ -35,7 +35,6 @@
 #include "User.h"
 #include "Channel.h"
 #include "ClientUser.h"
-#include "Global.h"
 #include "Message.h"
 #include "Database.h"
 #include "NetworkConfig.h"
@@ -43,6 +42,8 @@
 #include "MainWindow.h"
 #include "GlobalShortcut.h"
 
+#include "Global.h"
+
 template <typename T>
 QRectF OverlayGroup::boundingRect() const {
 	QRectF qr;
diff --git src/mumble/Overlay_macx.mm src/mumble/Overlay_macx.mm
index dd2a82f..925a9be 100644
--- src/mumble/Overlay_macx.mm
+++ src/mumble/Overlay_macx.mm
@@ -33,9 +33,10 @@
 #import <ScriptingBridge/ScriptingBridge.h>
 #include <Carbon/Carbon.h>
 #include "Overlay.h"
-#include "Global.h"
 #include "MainWindow.h"
 
+#include "Global.h"
+
 extern "C" {
 #include <xar/xar.h>
 }
diff --git src/mumble/Plugins.cpp src/mumble/Plugins.cpp
index 82b3ec6..568b9e7 100644
--- src/mumble/Plugins.cpp
+++ src/mumble/Plugins.cpp
@@ -32,7 +32,6 @@
 
 #include "Plugins.h"
 
-#include "Global.h"
 #include "Log.h"
 #include "MainWindow.h"
 #include "Message.h"
@@ -40,6 +39,8 @@
 #include "../../plugins/mumble_plugin.h"
 #include "WebFetch.h"
 
+#include "Global.h"
+
 static ConfigWidget *PluginConfigDialogNew(Settings &st) {
 	return new PluginConfig(st);
 }
diff --git src/mumble/PulseAudio.cpp src/mumble/PulseAudio.cpp
index b1ea1c1..9c32207 100644
--- src/mumble/PulseAudio.cpp
+++ src/mumble/PulseAudio.cpp
@@ -37,11 +37,11 @@
 #include <errno.h>
 #include <sys/ioctl.h>
 
-#include "Global.h"
 #include "MainWindow.h"
 #include "Timer.h"
 #include "User.h"
 
+#include "Global.h"
 
 static const char *mumble_sink_input = "Mumble Speakers";
 static const char *mumble_echo = "Mumble Speakers (Echo)";
diff --git src/mumble/RichTextEditor.cpp src/mumble/RichTextEditor.cpp
index ca4baa6..4c7f545 100644
--- src/mumble/RichTextEditor.cpp
+++ src/mumble/RichTextEditor.cpp
@@ -32,10 +32,11 @@
 
 #include "RichTextEditor.h"
 
-#include "Global.h"
 #include "Log.h"
 #include "MainWindow.h"
 
+#include "Global.h"
+
 RichTextHtmlEdit::RichTextHtmlEdit(QWidget *p) : QTextEdit(p) {
 	m_document = new LogDocument(this);
 	setDocument(m_document);
diff --git src/mumble/ServerHandler.cpp src/mumble/ServerHandler.cpp
index 60125c9..a7dfffc 100644
--- src/mumble/ServerHandler.cpp
+++ src/mumble/ServerHandler.cpp
@@ -37,7 +37,6 @@
 #include "Cert.h"
 #include "Connection.h"
 #include "Database.h"
-#include "Global.h"
 #include "MainWindow.h"
 #include "Message.h"
 #include "NetworkConfig.h"
@@ -47,6 +46,8 @@
 #include "SSL.h"
 #include "User.h"
 
+#include "Global.h"
+
 ServerHandlerMessageEvent::ServerHandlerMessageEvent(const QByteArray &msg, unsigned int mtype, bool flush) : QEvent(static_cast<QEvent::Type>(SERVERSEND_EVENT)) {
 	qbaMsg = msg;
 	uiType = mtype;
diff --git src/mumble/SocketRPC.cpp src/mumble/SocketRPC.cpp
index 928299a..d087e53 100644
--- src/mumble/SocketRPC.cpp
+++ src/mumble/SocketRPC.cpp
@@ -34,10 +34,11 @@
 
 #include "Channel.h"
 #include "ClientUser.h"
-#include "Global.h"
 #include "MainWindow.h"
 #include "ServerHandler.h"
 
+#include "Global.h"
+
 SocketRPCClient::SocketRPCClient(QLocalSocket *s, QObject *p) : QObject(p), qlsSocket(s), qbBuffer(NULL) {
 	qlsSocket->setParent(this);
 
diff --git src/mumble/Tokens.cpp src/mumble/Tokens.cpp
index 777ffc3..6446215 100644
--- src/mumble/Tokens.cpp
+++ src/mumble/Tokens.cpp
@@ -33,9 +33,10 @@
 #include "Tokens.h"
 
 #include "Database.h"
-#include "Global.h"
 #include "ServerHandler.h"
 
+#include "Global.h"
+
 Tokens::Tokens(QWidget *p) : QDialog(p) {
 	setupUi(this);
 
diff --git src/mumble/UserEdit.cpp src/mumble/UserEdit.cpp
index 9d7f456..4490047 100644
--- src/mumble/UserEdit.cpp
+++ src/mumble/UserEdit.cpp
@@ -32,10 +32,11 @@
 
 #include "UserEdit.h"
 
-#include "Global.h"
 #include "ServerHandler.h"
 #include "User.h"
 
+#include "Global.h"
+
 UserEdit::UserEdit(const MumbleProto::UserList &msg, QWidget *p) : QDialog(p) {
 	setupUi(this);
 
diff --git src/mumble/UserInformation.cpp src/mumble/UserInformation.cpp
index 6da8b80..41ba46c 100644
--- src/mumble/UserInformation.cpp
+++ src/mumble/UserInformation.cpp
@@ -34,11 +34,12 @@
 
 #include "Audio.h"
 #include "CELTCodec.h"
-#include "Global.h"
 #include "Net.h"
 #include "ServerHandler.h"
 #include "ViewCert.h"
 
+#include "Global.h"
+
 static QString decode_utf8_qssl_string(const QString &input) {
 	QString i = input;
 	return QUrl::fromPercentEncoding(i.replace(QLatin1String("\\x"), QLatin1String("%")).toLatin1());
diff --git src/mumble/UserModel.cpp src/mumble/UserModel.cpp
index a702e9d..1393317 100644
--- src/mumble/UserModel.cpp
+++ src/mumble/UserModel.cpp
@@ -36,7 +36,6 @@
 #include "ClientUser.h"
 #include "Channel.h"
 #include "Database.h"
-#include "Global.h"
 #include "LCD.h"
 #include "Log.h"
 #include "MainWindow.h"
@@ -46,6 +45,8 @@
 #include "Usage.h"
 #include "User.h"
 
+#include "Global.h"
+
 QHash <Channel *, ModelItem *> ModelItem::c_qhChannels;
 QHash <ClientUser *, ModelItem *> ModelItem::c_qhUsers;
 bool ModelItem::bUsersTop = false;
diff --git src/mumble/UserView.cpp src/mumble/UserView.cpp
index c462bf6..738a7a9 100644
--- src/mumble/UserView.cpp
+++ src/mumble/UserView.cpp
@@ -35,11 +35,12 @@
 #include "Channel.h"
 #include "ClientUser.h"
 #include "Log.h"
-#include "Global.h"
 #include "MainWindow.h"
 #include "ServerHandler.h"
 #include "UserModel.h"
 
+#include "Global.h"
+
 /*!
   \fn bool UserView::event(QEvent *evt)
   This implementation contains a special handler to display
diff --git src/mumble/VersionCheck.cpp src/mumble/VersionCheck.cpp
index e60a17a..58b3418 100644
--- src/mumble/VersionCheck.cpp
+++ src/mumble/VersionCheck.cpp
@@ -32,10 +32,11 @@
 
 #include "VersionCheck.h"
 
-#include "Global.h"
 #include "MainWindow.h"
 #include "WebFetch.h"
 
+#include "Global.h"
+
 VersionCheck::VersionCheck(bool autocheck, QObject *p, bool focus) : QObject(p) {
 	bSilent = autocheck;
 
diff --git src/mumble/VoiceRecorder.cpp src/mumble/VoiceRecorder.cpp
index 996b32c..0a9003b 100644
--- src/mumble/VoiceRecorder.cpp
+++ src/mumble/VoiceRecorder.cpp
@@ -36,11 +36,12 @@
 
 #include "AudioOutput.h"
 #include "ClientUser.h"
-#include "Global.h"
 #include "ServerHandler.h"
 
 #include "../Timer.h"
 
+#include "Global.h"
+
 VoiceRecorder::RecordBuffer::RecordBuffer(const ClientUser *cu,
         boost::shared_array<float> buffer, int samples, quint64 timestamp) :
 		cuUser(cu), fBuffer(buffer), iSamples(samples), uiTimestamp(timestamp) {
diff --git src/mumble/VoiceRecorderDialog.cpp src/mumble/VoiceRecorderDialog.cpp
index ecede42..045bd25 100644
--- src/mumble/VoiceRecorderDialog.cpp
+++ src/mumble/VoiceRecorderDialog.cpp
@@ -34,10 +34,11 @@
 #include "VoiceRecorderDialog.h"
 
 #include "AudioOutput.h"
-#include "Global.h"
 #include "ServerHandler.h"
 #include "VoiceRecorder.h"
 
+#include "Global.h"
+
 VoiceRecorderDialog::VoiceRecorderDialog(QWidget *p) : QDialog(p), qtTimer(new QTimer(this)) {
 	qtTimer->setObjectName(QLatin1String("qtTimer"));
 	qtTimer->setInterval(200);
diff --git src/mumble/WASAPI.cpp src/mumble/WASAPI.cpp
index 0e46871..abd59e7 100644
--- src/mumble/WASAPI.cpp
+++ src/mumble/WASAPI.cpp
@@ -32,10 +32,10 @@
 
 #include "WASAPI.h"
 #include "WASAPINotificationClient.h"
-#include "Global.h"
 #include "Timer.h"
 #include "User.h"
 
+#include "Global.h"
 
 // Now that Win7 is published, which includes public versions of these
 // interfaces, we simply inherit from those but use the "old" IIDs.
diff --git src/mumble/WASAPINotificationClient.cpp src/mumble/WASAPINotificationClient.cpp
index 4588778..a64ee71 100644
--- src/mumble/WASAPINotificationClient.cpp
+++ src/mumble/WASAPINotificationClient.cpp
@@ -29,13 +29,14 @@
 */
 
 #include <initguid.h>
-#include <Global.h>
 #include <QMutexLocker>
 #include "MainWindow.h"
 #include <boost/thread/once.hpp>
 
 #include "WASAPINotificationClient.h"
 
+#include <Global.h>
+
 HRESULT STDMETHODCALLTYPE WASAPINotificationClient::OnDefaultDeviceChanged(EDataFlow flow, ERole role, LPCWSTR pwstrDefaultDevice) {
 	const QString device = QString::fromWCharArray(pwstrDefaultDevice);
 
diff --git src/mumble/os_macx.mm src/mumble/os_macx.mm
index aa3c2f4..428fa45 100644
--- src/mumble/os_macx.mm
+++ src/mumble/os_macx.mm
@@ -30,10 +30,11 @@
 */
 
 #include "mumble_pch.hpp"
-#include "Global.h"
 #include "Overlay.h"
 #include "MainWindow.h"
 
+#include "Global.h"
+
 char *os_lang = NULL;
 static FILE *fConsole = NULL;
 

