--- src/mumble/AudioOutput.cpp.orig    2018-10-29 00:43:02.953895662 +0100
+++ src/mumble/AudioOutput.cpp 2018-10-29 00:43:49.407481605 +0100
@@ -431,7 +431,7 @@
                                        top[2] = 0.0f;
                                }

-                               if (std::abs<float>(front[0] * top[0] + front[1] * top[1] + front[2] * top[2]) > 0.01f) {
+                               if (std::abs(front[0] * top[0] + front[1] * top[1] + front[2] * top[2]) > 0.01f) {
                                        // Not perpendicular. Assume Y up and rotate 90 degrees.

                                        float azimuth = 0.0f;
