--- a/src/technical/util.cpp
+++ b/src/technical/util.cpp
@@ -92,18 +92,18 @@ int removeFileOrDirectory(const char *path, const struct stat *,
 		if (info->level != 0) {
 			if (0 != rmdir(path)) {
 				Logger::get().logWarning("Could not remove directory %s", path);
-				return FTW_STOP;
+				return 1;
 			}
 		}
 		break;
 	default:
 		if (0 != unlink(path)) {
 			Logger::get().logWarning("Could not remove file %s", path);
-			return FTW_STOP;
+			return 1;
 		}
 		break;
 	}
-	return FTW_CONTINUE;
+	return 0;
 }
 
 }
@@ -271,8 +271,8 @@ void Util::linkOrCopyFile(const char* newName, const char* oldName) {
 
 bool Util::removeDirectoryContents(const char* path) {
 	static const int maxDescripotorsToConsume = 8;
-	return FTW_STOP != nftw(path, removeFileOrDirectory,
-			maxDescripotorsToConsume, FTW_PHYS | FTW_DEPTH | FTW_ACTIONRETVAL);
+	return 1 != nftw(path, removeFileOrDirectory,
+			maxDescripotorsToConsume, FTW_PHYS | FTW_DEPTH);
 }
 
 void Util::ensurePathExists(const char* path) {
