--- a/diod/ops.c
+++ b/diod/ops.c
@@ -69,6 +69,7 @@
 #include <pthread.h>
 #include <errno.h>
 #include <sys/types.h>
+#include <sys/sysmacros.h>
 #include <sys/file.h>
 #include <sys/stat.h>
 #include <sys/statfs.h>
--- a/diod/xattr.c
+++ b/diod/xattr.c
@@ -39,7 +39,7 @@
 #include <errno.h>
 #include <sys/types.h>
 #include <sys/file.h>
-#include <attr/xattr.h>
+#include <sys/xattr.h>
 #include <sys/stat.h>
 #include <sys/statfs.h>
 #include <sys/socket.h>
