diff -wbBur bc-1.06/bc/scan.l bc-1.06.my/bc/scan.l
--- bc/scan.l   2000-09-13 22:25:47 +0400
+++ bc/scan.l    2006-12-25 12:39:34 +0300
@@ -143,7 +143,9 @@

 /* Definitions for readline access. */
 extern FILE *rl_instream;
+#if 0
 _PROTOTYPE(char *readline, (char *));
+#endif

 /* rl_input puts upto MAX characters into BUF with the number put in
    BUF placed in *RESULT.  If the yy input file is the same as
