--- src/xf86Calcomp.c.orig	2009-03-31 10:01:17.694998002 +0200
+++ src/xf86Calcomp.c	2009-03-31 10:01:30.975206736 +0200
@@ -308,8 +308,7 @@ DeviceInit (DeviceIntPtr dev)
 	 * this example device reports motions on 2 axes in absolute coordinates.
 	 * Device may reports touch pressure on the 3rd axis.
 	 */
-	if (InitValuatorClassDeviceStruct (dev, 3, xf86GetMotionEvents,
-									local->history_size, Absolute) == FALSE)
+	if (InitValuatorClassDeviceStruct (dev, 3, local->history_size, Absolute) == FALSE)
 	{
 		ErrorF ("Unable to allocate CALCOMP ValuatorClassDeviceStruct\n");
 		return !Success;
