--- lib/libnfs.c	2016-10-09 20:23:11.000000000 +0200
+++ lib/libnfs.c	2017-02-09 22:11:03.882598292 +0100
@@ -64,14 +64,6 @@
 #include <strings.h>
 #endif
 
-#ifdef MAJOR_IN_MKDEV
-#include <sys/mkdev.h>
-#endif
-
-#ifdef MAJOR_IN_SYSMACROS
-#include <sys/sysmacros.h>
-#endif
-
 #include <stdio.h>
 #include <stdarg.h>
 #include <stdlib.h>
@@ -82,6 +74,7 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <fcntl.h>
+#include <sys/sysmacros.h>
 #include "libnfs-zdr.h"
 #include "slist.h"
 #include "libnfs.h"
