From 6416d4e75017a4b1b7c593b9312675fdf4d36612 Mon Sep 17 00:00:00 2001
From: "Bradford D. Boyle" <bradford.d.boyle@gmail.com>
Date: Thu, 16 Apr 2026 01:50:39 +0000
Subject: [PATCH] Fix unknown type error

---
 rigs/harris/harris.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rigs/harris/harris.h b/rigs/harris/harris.h
index 070e2a97f..40f3e71df 100644
--- a/rigs/harris/harris.h
+++ b/rigs/harris/harris.h
@@ -46,7 +46,7 @@ int harris_get_mode(RIG *rig, vfo_t vfo, rmode_t *mode, pbwidth_t *width);
 int harris_set_level(RIG *rig, vfo_t vfo, setting_t level, value_t val);
 int harris_get_level(RIG *rig, vfo_t vfo, setting_t level, value_t *val);
 int harris_set_func(RIG *rig, vfo_t vfo, setting_t func, int status);
-int harris_get_bw(RIG *rig, vfo_t vfo, mode_t mode, pbwidth_t *width);
+int harris_get_bw(RIG *rig, vfo_t vfo, rmode_t mode, pbwidth_t *width);
 int harris_get_ptt(RIG *rig, vfo_t vfo, ptt_t *ptt);
 int harris_set_ptt(RIG *rig, vfo_t vfo, ptt_t ptt);
 
-- 
2.47.3

