 - we don't have threejs in void

--- a/src/sage/repl/ipython_kernel/install.py
+++ b/src/sage/repl/ipython_kernel/install.py
@@ -124,7 +124,7 @@ class SageKernelSpec(object):
             sage: spec = SageKernelSpec(prefix=tmp_dir())
             sage: spec.use_local_threejs()
             sage: threejs = os.path.join(spec.nbextensions_dir, 'threejs-sage')
-            sage: os.path.isdir(threejs)
+            sage: os.path.isdir(threejs)    # optional - threejs
             True
         """
         src = THREEJS_DIR
