--- libweston/weston-launch.c	2016-10-22 10:23:57.000000000 -0600
+++ libweston/weston-launch.c	2017-04-11 12:34:26.299944516 -0600
@@ -33,7 +33,6 @@
 #include <poll.h>
 #include <errno.h>
 
-#include <error.h>
 #include <getopt.h>
 
 #include <sys/types.h>
@@ -59,6 +58,7 @@
 #endif
 
 #include "weston-launch.h"
+#include "src/weston-error.h"
 
 #define DRM_MAJOR 226
 

--- /dev/null	2017-04-11 08:58:10.024649386 -0600
+++ src/weston-error.h	2017-04-11 12:38:10.575544091 -0600
@@ -0,0 +1,19 @@
+#ifndef _WESTON_ERROR_H
+#define _WESTON_ERROR_H
+
+#if defined(HAVE_ERROR_H)
+#include <error.h>
+#else
+#include <err.h>
+#include <string.h>
+#define _weston_error(S, E, F, ...) do { \
+       if (E) \
+               err(S, F ": %s", ##__VA_ARGS__, strerror(E)); \
+       else \
+               err(S, F, ##__VA_ARGS__); \
+} while(0)
+
+#define error _weston_error
+#endif
+#endif
+
