diff --git a/libopeniscsiusr/Makefile b/libopeniscsiusr/Makefile
index bf7c96c..589c69b 100644
--- libopeniscsiusr/Makefile
+++ libopeniscsiusr/Makefile
@@ -44,7 +44,7 @@ OBJS = context.o misc.o session.o sysfs.o iface.o idbm.o node.o default.o
 CFLAGS ?= -O2 -g
 CFLAGS += -Wall -Werror -Wextra -fvisibility=hidden -fPIC
 
-LIBADD =
+LIBADD = -lkmod
 
 all: $(LIBS) $(LIBS_MAJOR) $(TESTS) doc
 
diff --git a/usr/Makefile b/usr/Makefile
index f9445ad..13a719d 100644
--- usr/Makefile
+++ usr/Makefile
@@ -36,7 +36,7 @@ CFLAGS ?= -O2 -g
 WARNFLAGS ?= -Wall -Wstrict-prototypes
 CFLAGS += $(WARNFLAGS) -I../include -I. -D_GNU_SOURCE \
 	  -I$(TOPDIR)/libopeniscsiusr
-ISCSI_LIB = -L$(TOPDIR)/libopeniscsiusr -lopeniscsiusr
+ISCSI_LIB = -L$(TOPDIR)/libopeniscsiusr -lopeniscsiusr -lkmod
 PROGRAMS = iscsid iscsiadm iscsistart
 
 # libc compat files
