diff -ru gtk2hs-buildtools-old/c2hs/toplevel/C2HSConfig.hs gtk2hs-buildtools/c2hs/toplevel/C2HSConfig.hs
--- a/gtk2hs-buildtools/c2hs/toplevel/C2HSConfig.hs	2025-07-06 12:48:16.827097351 +0100
+++ b/gtk2hs-buildtools/c2hs/toplevel/C2HSConfig.hs	2025-07-06 12:50:54.803997950 +0100
@@ -70,8 +70,8 @@
 cppopts :: [String]
 cppopts  = case (os,cpp) of
   ("openbsd","cpp") -> ["-xc", "-w"]
-  (_,"cpp")         -> ["-x", "c", "-w"]
-  (_,"gcc")         -> ["-E", "-x", "c", "-w"]
+  (_,"cpp")         -> ["-x", "c", "-w", "-std=c99", "-D_Noreturn=", "-D_Nullable=", "-D_Nonnull="]
+  (_,"gcc")         -> ["-E", "-x", "c", "-w", "-std=c99", "-D_Noreturn=", "-D_Nullable=", "-D_Nonnull="]
   _                 -> []
 
 -- C preprocessor option for including only definitions (EXPORTED)
