diff --git a/deps/crypt/c_src/Makefile b/deps/crypt/c_src/Makefile
index 9c7e70b..ab31c50 100644
--- a/deps/crypt/c_src/Makefile
+++ b/deps/crypt/c_src/Makefile
@@ -25,8 +25,8 @@ else ifeq ($(UNAME_SYS), Linux)
 		CFLAGS ?= -O3 -std=c99 -finline-functions -Wall -Wmissing-prototypes
 		LDFLAGS ?= -lcrypt
   else
-		CFLAGS ?= -DHAVE_CRYPT_R -O3 -std=c99 -finline-functions -Wall -Wmissing-prototypes
-		LDFLAGS ?= -lpthread -lcrypt
+		CFLAGS += -DHAVE_CRYPT_R -O3 -std=c99 -finline-functions -Wall -Wmissing-prototypes
+		LDFLAGS += -lpthread -lcrypt
   endif
 else ifneq (,$(wildcard /usr/lib/libcrypt.*))
 	CC ?= cc
