diff --git a/config/config.exs b/config/config.exs
index b50c910..9cf622b 100644
--- a/config/config.exs
+++ b/config/config.exs
@@ -852,6 +852,8 @@
   {Pleroma.Web.ActivityPub.MRF.MediaProxyWarmingPolicy, [max_running: 5, max_waiting: 5]}
 ]
 
+config :tzdata, :data_dir, "/var/lib/pleroma/elixir_tzdata_data"
+
 # Import environment specific config. This must remain at the bottom
 # of this file so it overrides the configuration defined above.
 import_config "#{Mix.env()}.exs"
diff --git a/config/prod.secret.exs b/config/prod.secret.exs
new file mode 100644
index 0000000..23fd9f1
--- /dev/null
+++ b/config/prod.secret.exs
@@ -0,0 +1 @@
+import Mix.Config
