diff --git a/mix.exs b/mix.exs
index db2f1f0..115817c 100644
--- a/mix.exs
+++ b/mix.exs
@@ -38,7 +38,8 @@ def project do
           include_executables_for: [:unix],
           applications: [ex_syslogger: :load, syslog: :load, eldap: :transient],
           steps: [:assemble, &put_otp_version/1, &copy_files/1, &copy_nginx_config/1],
-          config_providers: [{Pleroma.Config.ReleaseRuntimeProvider, []}]
+          config_providers: [{Pleroma.Config.ReleaseRuntimeProvider, []}],
+          include_erts: "/usr/lib/erlang/erts-12.3"
         ]
       ]
     ]
