--- src/import/ImportFLAC.cpp	2016-01-08 23:05:48.000000000 +0100
+++ src/import/ImportFLAC.cpp	2016-10-14 22:19:44.434290227 +0200
@@ -296,7 +296,7 @@
    int cnt;
    wxFile binaryFile;
    if (!binaryFile.Open(filename)) {
-      return false; // File not found
+      return NULL; // File not found
    }
 
 #ifdef USE_LIBID3TAG
@@ -313,7 +313,7 @@
 
    if (cnt == wxInvalidOffset || strncmp(buf, FLAC_HEADER, 4) != 0) {
       // File is not a FLAC file
-      return false;
+      return NULL;
    }
 
    // Open the file for import
