--- ./src/effects/VST/VSTEffect.cpp.orig	2015-10-21 14:25:52.942716426 -0400
+++ ./src/effects/VST/VSTEffect.cpp	2015-10-21 14:28:45.521711160 -0400
@@ -2576,6 +2576,11 @@
    // symbols.
    //
    // Once we define a proper external API, the flags can be removed.
+
+#ifndef RTLD_DEEPBIND
+#define RTLD_DEEPBIND 0
+#endif
+
    void *lib = dlopen((const char *)wxString(realPath).ToUTF8(), RTLD_NOW | RTLD_LOCAL | RTLD_DEEPBIND);
    if (!lib) 
    {
