# audacity's pow10() clashes with musl's pow10()

--- a/src/uicomponents/components/internal/numeric/beatsformatter.cpp
+++ b/src/uicomponents/components/internal/numeric/beatsformatter.cpp
@@ -10,7 +10,7 @@
 
 using namespace au::uicomponents;
 
-static constexpr size_t pow10(int power)
+static constexpr size_t audacity_pow10(int power)
 {
     size_t result = 1;
     for (int i = 0; i < power; ++i) {
@@ -35,8 +35,8 @@
 
 static constexpr std::array<size_t, 3> MIN_DIGITS { 3, 2, 2 };
 static constexpr std::array<size_t, 3> UPPER_BOUNDS {
-    pow10(MIN_DIGITS[0] - 1) + 1, pow10(MIN_DIGITS[1] - 1) + 1,
-    pow10(MIN_DIGITS[2] - 1) + 1
+    audacity_pow10(MIN_DIGITS[0] - 1) + 1, audacity_pow10(MIN_DIGITS[1] - 1) + 1,
+    audacity_pow10(MIN_DIGITS[2] - 1) + 1
 };
 
 static QString barString()
@@ -194,9 +194,9 @@
     const auto digitsCount = m_fields[fieldIndex].digits;
 
     // Format always allows at least two digits
-    const auto lowerRange = digitsCount > MIN_DIGITS[fieldIndex] ? pow10(digitsCount - 1) : 0;
+    const auto lowerRange = digitsCount > MIN_DIGITS[fieldIndex] ? audacity_pow10(digitsCount - 1) : 0;
 
-    const auto upperRange = pow10(digitsCount);
+    const auto upperRange = audacity_pow10(digitsCount);
 
     return value >= int(lowerRange) && value < int(upperRange);
 }
