Index: audacity-Audacity-2.4.1/src/AudacityApp.cpp
===================================================================
--- audacity-Audacity-2.4.1.orig/src/AudacityApp.cpp
+++ audacity-Audacity-2.4.1/src/AudacityApp.cpp
@@ -1102,7 +1102,7 @@ locations of the missing files.").Format
 #define WL(lang,sublang)
 #endif
 
-#if wxCHECK_VERSION(3, 0, 1)
+#if 0
 wxLanguageInfo userLangs[] =
 {
    // Bosnian is defined in wxWidgets already
@@ -1366,7 +1366,7 @@ bool AudacityApp::OnInit()
    //
    // TODO:  The whole Language initialization really need to be reworked.
    //        It's all over the place.
-#if wxCHECK_VERSION(3, 0, 1)
+#if 0
    for (size_t i = 0, cnt = WXSIZEOF(userLangs); i < cnt; i++)
    {
       wxLocale::AddLanguage(userLangs[i]);
