--- a/configure.ac.old	2017-07-24 11:11:08.588672399 -0700
+++ b/configure.ac	2017-07-24 11:51:52.841737854 -0700
@@ -132,8 +132,12 @@
     AC_MSG_NOTICE([Using decorator interface version ${decor_ver}])
   ],
   [
-    AC_DEFINE(DECOR_INTERFACE_VERSION, 0, [Decorator interface version])
-    AC_MSG_ERROR([Failed to check the decorator interface version])
+    AC_DEFINE(DECOR_INTERFACE_VERSION, [20110504], [Decorator interface version])
+    AC_MSG_NOTICE([Unable to check decorator version at runtime. Defaulting to number for 0.8.14.])
+  ],
+  [
+    AC_DEFINE(DECOR_INTERFACE_VERSION, [20110504], [Decorator interface version])
+    AC_MSG_NOTICE([Unable to check decorator version at runtime. Defaulting to nnumber for 0.8.14.])
   ]
 )
 CFLAGS="$save_CFLAGS"
