--- a/src/Makefile
+++ b/src/Makefile
@@ -87,10 +87,7 @@
 INSTALL ?= install
 PIC = -fPIC
 EXE =
-WGET ?= $(shell \
-  if command -v wget >/dev/null 2>&1; then echo "wget --no-check-certificate --quiet -O"; \
-  elif command -v curl >/dev/null 2>&1; then echo "curl -k -L --silent -o"; \
-  else echo "echo 'Error: neither wget nor curl found' && false"; fi)
+WGET = false
 GIMPTOOL = $(shell \
   if command -v gimptool-3.2 >/dev/null 2>&1; then echo gimptool-3.2; \
   elif command -v gimptool-3.0 >/dev/null 2>&1; then echo gimptool-3.0; \
@@ -137,7 +134,7 @@
 else
 PREFIX ?= /opt/local
 endif
-WGET = curl -k -L --silent -o
+WGET = false
 endif

 ifeq ($(BUILD_OS),Windows)
@@ -160,10 +157,7 @@
 STRIP = strip
 endif

-QMAKE ?= $(shell \
-  if command -v qmake6 >/dev/null 2>&1; then echo qmake6; \
-  elif command -v qmake-qt5 >/dev/null 2>&1; then echo qmake-qt5; \
-  else echo qmake; fi)
+QMAKE = /usr/lib/qt5/bin/qmake

 #---------------------------------------------------------
 # Define compilation flags to use for each enabled feature
@@ -324,7 +318,7 @@

 # Enable native support of webcams and video streaming, using the OpenCV library.
 # (https://opencv.org/)
-OPENCV_CFLAGS = -Dcimg_use_opencv $(shell pkg-config opencv --cflags 2>/dev/null) -I/usr/include/opencv -I/usr/include/opencv4
+OPENCV_CFLAGS = -Dcimg_use_opencv $(shell pkg-config opencv --cflags 2>/dev/null) -I/usr/include/opencv -I/usr/include/opencv5
 OPENCV_LIBS = $(shell pkg-config opencv --libs 2>/dev/null || echo -lopencv_core -lopencv_highgui -lopencv_videoio)

 # Enable support of most classical image file formats, using the GraphicsMagick++ library.
@@ -419,7 +413,7 @@
 ifeq ($(BUILD_OS),Darwin)
 	$(CXX) -dynamiclib -o libgmic.$(SOVERSION).dylib -install_name $(PREFIX)/$(LIB)/libgmic.$(SOVERSION).dylib -compatibility_version $(SOVERSION) -current_version $(SVERSION) $(MANDATORY_CFLAGS) libgmic.o $(LIBS)
 else ifdef IS_CLANG
-	$(CXX) -shared -std=c++11 -pedantic -o libgmic.so.$(VERSION) libgmic.o $(LIBS)
+	$(CXX) -shared -std=c++11 -pedantic -Wl,-soname,libgmic.so.$(VERSION1) -o libgmic.so.$(VERSION) libgmic.o $(LIBS)
 else
 	$(CXX) -shared -std=c++11 -pedantic -Wl,-soname,libgmic.so.$(VERSION1) -o libgmic.so.$(VERSION) libgmic.o $(LIBS)
 endif
