--- src/CImg.h	2018-04-23 04:49:04.329501123 +0200
+++ src/CImg.h	2018-04-23 04:48:23.850499833 +0200
@@ -416,8 +416,8 @@
 #define _cimg_redefine_False
 #endif
 #include <cstddef>
-#include "cv.h"
-#include "highgui.h"
+#include <cv.h>
+#include <highgui.h>
 #endif
 
 // Configure LibPNG support.
@@ -429,7 +429,7 @@
 // (see methods 'CImg<T>::{load,save}_png()'.
 #ifdef cimg_use_png
 extern "C" {
-#include "png.h"
+#include <png.h>
 }
 #endif
 
@@ -442,8 +442,8 @@
 // (see methods 'CImg<T>::{load,save}_jpeg()').
 #ifdef cimg_use_jpeg
 extern "C" {
-#include "jpeglib.h"
-#include "setjmp.h"
+#include <jpeglib.h>
+#include <setjmp.h>
 }
 #endif
 
@@ -458,7 +458,7 @@
 extern "C" {
 #define uint64 uint64_hack_
 #define int64 int64_hack_
-#include "tiffio.h"
+#include <tiffio.h>
 #undef uint64
 #undef int64
 }
@@ -472,9 +472,9 @@
 // MINC2 library may be used to get a native support of '.mnc' files.
 // (see methods 'CImg<T>::{load,save}_minc2()').
 #ifdef cimg_use_minc2
-#include "minc_io_simple_volume.h"
-#include "minc_1_simple.h"
-#include "minc_1_simple_rw.h"
+#include <minc_io_simple_volume.h>
+#include <minc_1_simple.h>
+#include <minc_1_simple_rw.h>
 #endif
 
 // Configure Zlib support.
@@ -486,7 +486,7 @@
 // (see methods 'CImg[List]<T>::{load,save}_cimg()').
 #ifdef cimg_use_zlib
 extern "C" {
-#include "zlib.h"
+#include <zlib.h>
 }
 #endif
 
@@ -498,7 +498,7 @@
 // Libcurl may be used to get a native support of file downloading from the network.
 // (see method 'cimg::load_network()'.)
 #ifdef cimg_use_curl
-#include "curl/curl.h"
+#include <curl/curl.h>
 #endif
 
 // Configure Magick++ support.
@@ -509,7 +509,7 @@
 // Magick++ library may be used to get a native support of various image file formats.
 // (see methods 'CImg<T>::{load,save}()').
 #ifdef cimg_use_magick
-#include "Magick++.h"
+#include <Magick++.h>
 #endif
 
 // Configure FFTW3 support.
@@ -522,7 +522,7 @@
 // (see method 'CImg[List]<T>::FFT()').
 #ifdef cimg_use_fftw3
 extern "C" {
-#include "fftw3.h"
+#include <fftw3.h>
 }
 #endif
 
@@ -535,7 +535,7 @@
 // that can be saved as '.ps' or '.svg' files afterwards.
 // (see method 'CImg<T>::draw_object3d()').
 #ifdef cimg_use_board
-#include "Board.h"
+#include <Board.h>
 #endif
 
 // Configure OpenEXR support.
@@ -546,11 +546,11 @@
 // OpenEXR library may be used to get a native support of '.exr' files.
 // (see methods 'CImg<T>::{load,save}_exr()').
 #ifdef cimg_use_openexr
-#include "ImfRgbaFile.h"
-#include "ImfInputFile.h"
-#include "ImfChannelList.h"
-#include "ImfMatrixAttribute.h"
-#include "ImfArray.h"
+#include <ImfRgbaFile.h>
+#include <ImfInputFile.h>
+#include <ImfChannelList.h>
+#include <ImfMatrixAttribute.h>
+#include <ImfArray.h>
 #endif
 
 // Configure TinyEXR support.
@@ -563,7 +563,7 @@
 #ifndef TINYEXR_IMPLEMENTATION
 #define TINYEXR_IMPLEMENTATION
 #endif
-#include "tinyexr.h"
+#include <tinyexr.h>
 #endif
 
 // Lapack configuration.
