--- data/meson.build
+++ data/meson.build
@@ -30,12 +30,12 @@ if get_option('daemon')
   )
 endif
 
-if get_option('systemd')
-  con2 = configuration_data()
-  con2.set('libexecdir', libexecdir)
-  con2.set('bindir', bindir)
-  con2.set('datadir', datadir)
+con2 = configuration_data()
+con2.set('libexecdir', libexecdir)
+con2.set('bindir', bindir)
+con2.set('datadir', datadir)
 
+if get_option('systemd')
   rw_directories = []
   rw_directories += join_paths (localstatedir, 'lib', 'fwupd')
   rw_directories += join_paths (default_sysconfdir, 'fwupd', 'remotes.d')
@@ -57,17 +57,20 @@ if get_option('systemd')
   endif
   con2.set('dynamic_options', '\n'.join(dynamic_options))
 
-  # replace @libexecdir@
-  configure_file(
-    input : 'org.freedesktop.fwupd.service.in',
-    output : 'org.freedesktop.fwupd.service',
-    configuration : con2,
-    install: true,
-    install_dir: join_paths(datadir,
-                            'dbus-1',
-                            'system-services'),
-  )
+endif
 
+# replace @libexecdir@
+configure_file(
+  input : 'org.freedesktop.fwupd.service.in',
+  output : 'org.freedesktop.fwupd.service',
+  configuration : con2,
+  install: true,
+  install_dir: join_paths(datadir,
+                          'dbus-1',
+                          'system-services'),
+)
+
+if get_option('systemd')
   # replace @bindir@
   configure_file(
     input : 'fwupd-offline-update.service.in',
