diff --git meson.build meson.build
index c0463ef..c7d1567 100644
--- meson.build
+++ meson.build
@@ -241,6 +241,13 @@ if get_option('plugin_uefi')
   endif
   conf.set_quoted('EFI_MACHINE_TYPE_NAME', EFI_MACHINE_TYPE_NAME)
   r = run_command([python3, 'po/test-deps'])
+  if meson.is_cross_build()
+    qemu = '/usr/bin/qemu-' + gnu_efi_arch + '-static'
+    qemu_python3_cmd = qemu + ' -L @XBPS_CROSS_BASE@ @XBPS_CROSS_BASE@/bin/python3'
+    r = run_command('sh', '-c', qemu_python3_cmd + ' po/test-deps')
+  else
+    r = run_command([python3, 'po/test-deps'])
+  endif
   if r.returncode() != 0
     error(r.stderr())
   endif
diff --git po/meson.build po/meson.build
index 6e78d2e..3fdedb7 100644
--- po/meson.build
+++ po/meson.build
@@ -5,6 +5,12 @@ i18n.gettext(meson.project_name(),
   ]
 )
 
+if meson.is_cross_build()
+  python3_cmd = qemu_python3_cmd
+else
+  python3_cmd = python3.path()
+endif
+
 if get_option('plugin_uefi')
-meson.add_install_script('make-images.sh', localedir, python3.path())
+meson.add_install_script('make-images.sh', localedir, python3_cmd)
 endif

