From 22e9a074158a9eeeccc89e87a6aa941cbb743f58 Mon Sep 17 00:00:00 2001
From: Mario Limonciello <superm1@gmail.com>
Date: Sun, 22 Aug 2021 14:20:13 -0500
Subject: [PATCH] trivial: fix a compilation dependency error (Fixes: #3657)

---
 plugins/tpm-eventlog/meson.build | 1 +
 1 file changed, 1 insertion(+)

diff --git a/plugins/tpm-eventlog/meson.build b/plugins/tpm-eventlog/meson.build
index 3f72576f68..8fae3843d2 100644
--- a/plugins/tpm-eventlog/meson.build
+++ b/plugins/tpm-eventlog/meson.build
@@ -60,6 +60,7 @@ endif
 
 fwupdtpmevlog = executable(
   'fwupdtpmevlog',
+  fu_hash,
   sources : [
     'fu-tpm-eventlog.c',
     'fu-tpm-eventlog-common.c',
