From ae4fdeb44a4d7e3a668ce73778c65c3e37879090 Mon Sep 17 00:00:00 2001
From: Michal Vasilek <michal@vasilek.cz>
Date: Mon, 23 Aug 2021 19:21:58 +0200
Subject: [PATCH] trivial: make gi-docgen a host dependency on cross

---
 meson.build | 1 +
 1 file changed, 1 insertion(+)

diff --git a/meson.build b/meson.build
index bcd4cbf4d..c5f2f292a 100644
--- a/meson.build
+++ b/meson.build
@@ -555,6 +555,7 @@ elif get_option('docs') == 'docgen'
   endif
   gidocgen_dep = dependency('gi-docgen',
     version: '>= 2021.1',
+    native: true,
     fallback: ['gi-docgen', 'dummy_dep'],
   )
   gidocgen = find_program('gi-docgen')
