diff --git a/gdata/tests/streams.c b/gdata/tests/streams.c
index 7158754..1763bd9 100644
--- a/gdata/tests/streams.c
+++ b/gdata/tests/streams.c
@@ -193,7 +193,7 @@ create_server (SoupServerCallback callback, gpointer user_data, GMainLoop **main
 	g_main_context_push_thread_default (context);
 
 	soup_server_listen_local (server, 0  /* random port */,
-	                          SOUP_SERVER_LISTEN_HTTPS, &error);
+	                          SOUP_SERVER_LISTEN_HTTPS | SOUP_SERVER_LISTEN_IPV4_ONLY, &error);
 	g_assert_no_error (error);
 
 	g_main_context_pop_thread_default (context);
