From 7162c0af5651b8283d57de2bc4ca0e7fda10db80 Mon Sep 17 00:00:00 2001
From: Stefan Hundhammer <Stefan.Hundhammer@gmx.de>
Date: Tue, 19 Mar 2024 15:36:26 +0100
Subject: [PATCH] Fix build with musl libc (#266)

---
 src/DirTreeCache.cpp      | 1 +
 src/SystemFileChecker.cpp | 2 ++
 2 files changed, 3 insertions(+)

diff --git a/src/DirTreeCache.cpp b/src/DirTreeCache.cpp
index 2fda17a..932ba39 100644
--- a/src/DirTreeCache.cpp
+++ b/src/DirTreeCache.cpp
@@ -18,6 +18,7 @@
 #include "FormatUtil.h"
 #include "Logger.h"
 #include "Exception.h"
+#include "BrokenLibc.h"     // ALLPERMS
 
 #define KB 1024LL
 #define MB (1024LL*1024)
diff --git a/src/SystemFileChecker.cpp b/src/SystemFileChecker.cpp
index a1bbd8a..6440c15 100644
--- a/src/SystemFileChecker.cpp
+++ b/src/SystemFileChecker.cpp
@@ -7,6 +7,8 @@
  */
 
 
+#include <sys/types.h>          // uid_t
+
 #include "SystemFileChecker.h"
 #include "DirInfo.h"
 
-- 
2.44.0

