--- a/user/v83/drbdadm.h
+++ b/user/v83/drbdadm.h
@@ -4,6 +4,7 @@
 #include <linux/drbd_config.h>
 #include <sys/utsname.h>
 #include <sys/types.h>
+#include <sys/sysmacros.h>
 #include <net/if.h>
 #include <stdint.h>
 #include <stdarg.h>
--- a/user/v84/drbdadm.h
+++ b/user/v84/drbdadm.h
@@ -4,6 +4,7 @@
 #include <stdbool.h>
 #include <sys/utsname.h>
 #include <sys/types.h>
+#include <sys/sysmacros.h>
 #include <net/if.h>
 #include <stdint.h>
 #include <stdarg.h>
--- a/user/v9/drbdadm.h
+++ b/user/v9/drbdadm.h
@@ -4,6 +4,7 @@
 #include <stdbool.h>
 #include <sys/utsname.h>
 #include <sys/types.h>
+#include <sys/sysmacros.h>
 #include <net/if.h>
 /* We carry a copy of <sys/queue.h> around,
  * some older versions of it don't have the STAILQ_* variants */
