--- a/proc/module.mk.orig	2010-04-22 13:22:25.736775611 +0200
+++ b/proc/module.mk	2010-04-22 13:24:09.814780650 +0200
@@ -42,7 +42,7 @@ FPIC       := -fpic
 
 ifeq ($(SHARED),1)
 ALL        += proc/$(SONAME)
-INSTALL    += ldconfig
+INSTALL    += ldconfig
 LIBFLAGS   := -DSHARED=1 $(FPIC)
 # This is in gcc 3.5, but exported functions must be marked.
 #LIBFLAGS += $(call check_gcc,-fvisibility=hidden,)
@@ -96,7 +96,7 @@ endif
 #################### install rules ###########################
 
 $(lib)$(SOFILE) : proc/$(SONAME)
-	$(install) --mode a=rx $< $@
+	$(install) -m755 $< $@
 
 ifneq ($(SOLINK),$(SOFILE))
 .PHONY: $(lib)$(SOLINK)
@@ -115,7 +115,7 @@ ldconfig : $(lib)$(SONAME) $(lib)$(SOLIN
 	$(ldconfig)
 
 $(usr/lib)$(ANAME) : proc/$(ANAME)
-	$(install) --mode a=r $< $@
+	$(install) -m644 $< $@
 
 # Junk anyway... supposed to go in /usr/include/$(NAME)
 #INSTALL += $(addprefix $(include),$(HDRFILES))
--- a/ps/module.mk.orig	2010-04-22 13:26:49.156775917 +0200
+++ b/ps/module.mk	2010-04-22 13:27:05.019792210 +0200
@@ -33,8 +33,8 @@ ps/stacktrace.o: ps/stacktrace.c
 
 
 $(bin)ps: ps/ps
-	$(install) --mode a=rx $< $@
+	$(install) -m755 $< $@
 
 $(man1)ps.1 : ps/ps.1
-	$(install) --mode a=r $< $@
+	$(install) -m644 $< $@
 	-rm -f $(DESTDIR)/var/catman/cat1/ps.1.gz $(DESTDIR)/var/man/cat1/ps.1.gz
--- a/Makefile.orig	2009-05-10 20:50:48.000000000 +0200
+++ b/Makefile	2013-02-11 13:52:37.814939368 +0100
@@ -24,35 +24,35 @@ TARVERSION   := $(VERSION).$(SUBVERSION)
 ############ vars
 
 # so you can disable them or choose alternates
-ldconfig := ldconfig
+ldconfig :=
 ln_f     := ln -f
 ln_sf    := ln -sf
-install  := install -D --owner 0 --group 0
+install  := install -D
 
 # Lame x86-64 /lib64 and /usr/lib64 abomination:
 lib64    := lib$(shell [ -d /lib64 ] && echo 64)
 
 usr/bin                  := $(DESTDIR)/usr/bin/
-bin                      := $(DESTDIR)/bin/
-sbin                     := $(DESTDIR)/sbin/
+bin                      := $(DESTDIR)/usr/bin/
+sbin                     := $(DESTDIR)/usr/sbin/
 usr/proc/bin             := $(DESTDIR)/usr/bin/
 man1                     := $(DESTDIR)/usr/share/man/man1/
 man5                     := $(DESTDIR)/usr/share/man/man5/
 man8                     := $(DESTDIR)/usr/share/man/man8/
-lib                      := $(DESTDIR)/$(lib64)/
-usr/lib                  := $(DESTDIR)/usr/$(lib64)/
+lib                      := $(DESTDIR)/usr/lib/
+usr/lib                  := $(DESTDIR)/usr/lib/
 usr/include              := $(DESTDIR)/usr/include/
 
 #SKIP     := $(bin)kill $(man1)kill.1
 
-BINFILES := $(usr/bin)uptime $(usr/bin)tload $(usr/bin)free $(usr/bin)w \
+BINFILES := $(usr/bin)tload $(usr/bin)free $(usr/bin)w \
             $(usr/bin)top $(usr/bin)vmstat $(usr/bin)watch $(usr/bin)skill \
-            $(usr/bin)snice $(bin)kill $(sbin)sysctl $(usr/bin)pmap \
+            $(usr/bin)snice $(sbin)sysctl $(usr/bin)pmap \
             $(usr/proc/bin)pgrep $(usr/proc/bin)pkill $(usr/bin)slabtop \
             $(usr/proc/bin)pwdx
 
-MANFILES := $(man1)uptime.1 $(man1)tload.1 $(man1)free.1 $(man1)w.1 \
-            $(man1)top.1 $(man1)watch.1 $(man1)skill.1 $(man1)kill.1 \
+MANFILES := $(man1)tload.1 $(man1)free.1 $(man1)w.1 \
+            $(man1)top.1 $(man1)watch.1 $(man1)skill.1 \
             $(man1)snice.1 $(man1)pgrep.1 $(man1)pkill.1 $(man1)pmap.1 \
             $(man5)sysctl.conf.5 $(man8)vmstat.8 $(man8)sysctl.8 \
             $(man1)slabtop.1 $(man1)pwdx.1
@@ -174,7 +174,8 @@ INSTALL := $(BINFILES) $(MANFILES)
 # want this rule first, use := on ALL, and ALL not filled in yet
 all: do_all
 
--include */module.mk
+-include proc/module.mk
+-include ps/module.mk
 
 do_all:    $(ALL)
 
@@ -222,14 +223,14 @@ clean:
 ###### install
 
 $(BINFILES) : all
-	$(install) --mode a=rx $(notdir $@) $@
+	$(install) -m755 $(notdir $@) $@
 
 $(MANFILES) : all
-	$(install) --mode a=r $(notdir $@) $@
+	$(install) -m644 $(notdir $@) $@
 
 install: $(filter-out $(SKIP) $(addprefix $(DESTDIR),$(SKIP)),$(INSTALL))
-	cd $(usr/bin) && $(ln_f) skill snice
-	cd $(usr/proc/bin) && $(ln_f) pgrep pkill
+	cd $(usr/bin) && $(ln_sf) skill snice
+	cd $(usr/proc/bin) && $(ln_sf) pgrep pkill
 
 ############ prog.c --> prog.o
 
@@ -255,7 +256,7 @@ watch: % : %.o
 ############ progX --> progY
 
 snice kill: skill
-	$(ln_f) skill $@
+	$(ln_sf) skill $@
 
 pkill: pgrep
-	$(ln_f) pgrep pkill
+	$(ln_sf) pgrep pkill
