Author: <csmall@debian.org>
Description: Add CR to warning line 
Bug-Debian: http://bugs.debian.org/331419
Index: b/skill.c
===================================================================
--- a/skill.c	2009-11-24 21:00:31.000000000 +1100
+++ b/skill.c	2009-11-24 21:00:37.000000000 +1100
@@ -128,7 +128,7 @@
   sprintf(buf, "/proc/%d/stat", pid); /* pid (cmd) state ppid pgrp session tty */
   fd = open(buf,O_RDONLY);
   if(fd==-1){  /* process exited maybe */
-    if(pids && w_flag) printf("WARNING: process %d could not be found.",pid);
+    if(pids && w_flag) printf("WARNING: process %d could not be found.\n",pid);
     return;
   }
   fstat(fd, &statbuf);
@@ -342,7 +342,7 @@
   }else{
     fprintf(stderr,
       "Usage:   snice [new priority] [options] process selection criteria\n"
-      "Example: snice netscape crack +7\n"
+      "Example: snice +7 netscape crack \n"
       "\n"
       "The default priority is +4. (snice +4 ...)\n"
       "Priority numbers range from +20 (slowest) to -20 (fastest).\n"
