--- procps-3.2.7/w.c.kzak	2007-04-02 23:58:30.000000000 +0200
+++ procps-3.2.7/w.c	2007-04-03 00:09:15.000000000 +0200
@@ -149,11 +149,11 @@
 	const proc_t *restrict const tmp = *pptr;
 	if(unlikely(tmp->tgid == u->ut_pid)) {
 	    *found_utpid = 1;
-	    best = tmp;
+	    if (!best)
+	         best = tmp;
 	}
 	if(tmp->tty != line) continue;
 	(*jcpu) += tmp->utime + tmp->stime;
-	secondbest = tmp;
 	/* same time-logic here as for "best" below */
 	if(!  (secondbest && tmp->start_time <= secondbest->start_time)  ){
 	    secondbest = tmp;
