diff -up procps-3.2.7/top.c.clrscr procps-3.2.7/top.c
--- procps-3.2.7/top.c.clrscr	2008-04-08 08:49:48.000000000 +0200
+++ procps-3.2.7/top.c	2008-04-08 08:50:07.000000000 +0200
@@ -2109,6 +2109,7 @@ static void fields_sort (void)
       *p = x + 'A';
    Curwin->rc.sortindx = x;
    putp(Cap_curs_norm);
+   putp(Cap_clr_scr);
 }
 
 
@@ -2134,6 +2135,7 @@ static void fields_toggle (void)
          *p = i + 'A';
    }
    putp(Cap_curs_norm);
+   putp(Cap_clr_scr);
 }
 
 /*######  Windows/Field Groups support  #################################*/
