Description: kill prints perror 
Author: Justin Pryzby <justinpryzby@users.sourceforge.net>
Bug-Debian: http://bugs.debian.org/468363
Reviewed-by: Craig Small <csmall@debian.org>
Index: b/skill.c
===================================================================
--- a/skill.c	2009-11-24 20:53:05.000000000 +1100
+++ b/skill.c	2009-11-24 21:00:31.000000000 +1100
@@ -317,6 +317,7 @@
       // The standard says we return non-zero if an error occurs. Thus if
       // killing two processes gives 0 for one and EPERM for the other,
       // we are required to return both zero and non-zero. Quantum kill???
+      perror("kill");
       exitvalue = 1;
       continue;
     }
