Author: Craig Small <csmall@debian.org>
Description: Removed unneeded optarg and optind variables form tload.c
Index: b/tload.c
===================================================================
--- a/tload.c	2009-11-24 20:53:03.000000000 +1100
+++ b/tload.c	2009-11-24 21:00:42.000000000 +1100
@@ -30,9 +30,6 @@
 static int dly=5;
 static jmp_buf jb;
 
-extern int optind;
-extern char *optarg;
-
 static void alrm(int signo)
 {
     (void)signo;
