Author: Paolo Pantaleo <paolopantaleo@gmail.com>
Description: Fix highlighting problem in top
Bug-Debian: http://bugs.debian.org/351065
Index: b/top.c
===================================================================
--- a/top.c	2009-11-24 21:00:48.000000000 +1100
+++ b/top.c	2009-11-24 21:00:50.000000000 +1100
@@ -3054,9 +3054,10 @@
       snprintf(_z, sizeof(_z), f, ## va);                                    \
       snprintf(cbuf, sizeof(cbuf), "%s%s%s",                                 \
          q->capclr_rowhigh,                                                  \
-         _z,                                                                 \
+         _z+advance,                                                         \
          !(CHKw(q, Show_HIROWS) && 'R' == p->state) ? q->capclr_rownorm : "" \
       );                                                                     \
+      advance=0;                                                             \
       pad += q->len_rowhigh;                                                 \
       if (!(CHKw(q, Show_HIROWS) && 'R' == p->state)) pad += q->len_rownorm; \
    }                                                                         \
