Author: <csmall@debian.org>
Description: Make watch unsigned chars and 8bit clean
Index: b/watch.c
===================================================================
--- a/watch.c	2009-11-24 20:53:03.000000000 +1100
+++ b/watch.c	2009-11-24 21:00:40.000000000 +1100
@@ -297,7 +297,7 @@
 				move(y, x);
 				if (option_differences) {
 					chtype oldch = inch();
-					char oldc = oldch & A_CHARTEXT;
+					unsigned char oldc = oldch & A_CHARTEXT;
 					attr = !first_screen
 					    && ((char)c != oldc
 						||
