--- src/qjackctlGraphForm.cpp.orig	2018-05-27 18:39:33.051174395 +0200
+++ src/qjackctlGraphForm.cpp	2018-07-12 21:46:57.002185428 +0200
@@ -338,11 +338,11 @@
 	const QRectF& rect
 		= canvas->mapToScene(canvas->viewport()->rect()).boundingRect();
 	const QPointF& pos = rect.center();
-	const qreal w = 0.3 * qMax(rect.width(),  800.0);
-	const qreal h = 0.3 * qMax(rect.height(), 600.0);
+	const double w = 0.3 * qMax(static_cast<double>(rect.width()),  800.0);
+	const double h = 0.3 * qMax(static_cast<double>(rect.height()), 600.0);
 
-	qreal x = pos.x();
-	qreal y = pos.y();
+	double x = pos.x();
+	double y = pos.y();
 
 	switch (node->nodeMode()) {
 	case qjackctlGraphItem::Input:

