--- Makefile.orig	2017-12-10 19:54:55.223497878 -0500
+++ Makefile	2017-12-10 19:37:39.909790719 -0500
@@ -19,7 +19,7 @@
 EARLY_PREFIX ?=
 
 BINDIR ?= /usr/bin
-PYTHON ?= python
+PYTHON ?= python2
 DOCDIR ?= /usr/share/docs/$(PROGRAM_PREFIX)bootchart$(PROGRAM_SUFFIX)
 MANDIR ?= /usr/share/man/man1
 # never contains /usr; typically /lib, /lib64 or e.g. /lib/x86_64-linux-gnu
@@ -99,7 +99,7 @@
 	install -d $(DESTDIR)$(PKGLIBDIR)/tmpfs
 
 install-collector: all install-chroot
-	install -m 755 -D bootchartd $(DESTDIR)$(EARLY_PREFIX)/sbin/$(PROGRAM_PREFIX)bootchartd$(PROGRAM_SUFFIX)
+	install -m 755 -D bootchartd $(DESTDIR)$(EARLY_PREFIX)/bin/$(PROGRAM_PREFIX)bootchartd$(PROGRAM_SUFFIX)
 	install -m 644 -D bootchartd.conf $(DESTDIR)/etc/$(PROGRAM_PREFIX)bootchartd$(PROGRAM_SUFFIX).conf
 	install -m 755 -D bootchart-collector $(DESTDIR)$(PKGLIBDIR)/$(PROGRAM_PREFIX)bootchart$(PROGRAM_SUFFIX)-collector
 
