Index: pybootchartgui/draw.py
===================================================================
--- a/pybootchartgui/draw.py
+++ b/pybootchartgui/draw.py
@@ -508,7 +508,7 @@ def draw_processes_recursively(ctx, proc
 		cmdString = proc.cmd
 	else:
 		cmdString = ''
-	if (OPTIONS.show_pid or OPTIONS.show_all) and ipid is not 0:
+	if (OPTIONS.show_pid or OPTIONS.show_all) and ipid != 0:
 		cmdString = cmdString + " [" + str(ipid // 1000) + "]"
 	if OPTIONS.show_all:
 		if proc.args:
@@ -606,7 +606,7 @@ class CumlSample:
 		if self.color is None:
 			i = self.next() % HSV_MAX_MOD
 			h = 0.0
-			if i is not 0:
+			if i != 0:
 				h = (1.0 * i) / HSV_MAX_MOD
 			s = 0.5
 			v = 1.0
Index: pybootchartgui/parsing.py
===================================================================
--- a/pybootchartgui/parsing.py
+++ b/pybootchartgui/parsing.py
@@ -98,7 +98,7 @@ class Trace:
     def compile(self, writer):
 
         def find_parent_id_for(pid):
-            if pid is 0:
+            if pid == 0:
                 return 0
             ppid = self.parent_map.get(pid)
             if ppid:
