--- a/packaging/pep517_backend/_cython_configuration.py	2026-05-19 11:22:29.000000000 -0500
+++ b/packaging/pep517_backend/_cython_configuration.py	2026-05-19 16:24:45.294202045 -0500
@@ -8,7 +8,6 @@
 from pathlib import Path
 from sys import version_info as _python_version_tuple
 
-from expandvars import expandvars
 
 from ._compat import load_toml_from_string
 from ._transformers import (

--- a/pyproject.toml	2026-05-19 11:22:29.000000000 -0500
+++ b/pyproject.toml	2026-05-19 16:25:09.197896907 -0500
@@ -3,7 +3,6 @@
   # NOTE: The following build dependencies are necessary for initial
   # NOTE: provisioning of the in-tree build backend located under
   # NOTE: `packaging/pep517_backend/`.
-  "expandvars",
   "setuptools >= 82.0.1",  # Minimum required for `version = attr:`
   "tomli; python_version < '3.11'",
 ]
