From 92fe04c534581b8292c376ffd064691339d51f18 Mon Sep 17 00:00:00 2001
From: Eitan Isaacson <eitan@monotonous.org>
Date: Mon, 13 Jan 2025 13:06:21 -0800
Subject: [PATCH] Added version to soname

---
 libspiel/meson.build | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libspiel/meson.build b/libspiel/meson.build
index 5f99a42..128ae53 100644
--- a/libspiel/meson.build
+++ b/libspiel/meson.build
@@ -87,6 +87,7 @@ spiel_deps = [
 spiel_lib = shared_library('spiel-' + api_version,
   spiel_sources,
   dependencies: spiel_deps,
+  version: meson.project_version(),
   install: true,
 )
 
