--- script/lib/install-application.js
+++ script/lib/install-application.js
@@ -79,7 +79,8 @@ module.exports = function (packagedAppPath, installDir) {
     install(installationDirPath, packagedAppFileName, packagedAppPath)

     { // Install icons
-      const baseIconThemeDirPath = findBaseIconThemeDirPath()
+      const baseIconThemeDirPath = path.join(prefixDirPath, 'share/icons/hicolor')
+
       const fullIconName = atomExecutableName + '.png'

       let existingIconsFound = false
