--- a/Makefile	2023-04-02 10:04:13.000000000 +0200
+++ b/Makefile	2025-06-19 19:30:45.487560838 +0200
@@ -51,22 +51,20 @@
 	./bin/mbpfan-tests
 
 uninstall:
-	rm /usr/sbin/mbpfan
+	rm /usr/bin/mbpfan
 	rm /etc/mbpfan.conf
-	rm /lib/modules-load.d/mbpfan.depend.conf
-	rm /lib/systemd/system/mbpfan.service
+	rm /usr/lib/modules-load.d/mbpfan.depend.conf
 	rm /usr/share/man/man8/mbpfan.8.gz
 	rm -rf /usr/share/doc/mbpfan
 
 install: all
-	install -d $(DESTDIR)/usr/sbin
+	install -d $(DESTDIR)/usr/bin
 	install -d $(DESTDIR)/etc
-	install -d $(DESTDIR)/lib/systemd/system
 	install -d $(DESTDIR)/usr/share/doc/mbpfan
-	install -d $(DESTDIR)/lib/modules-load.d
-	install $(BIN) $(DESTDIR)/usr/sbin
+	install -d $(DESTDIR)/usr/lib/modules-load.d
+	install $(BIN) $(DESTDIR)/usr/bin
 	install -m644 $(CONF) $(DESTDIR)/etc
-	install -m644 $(DEPEND_MODULE) $(DESTDIR)/lib/modules-load.d
+	install -m644 $(DEPEND_MODULE) $(DESTDIR)/usr/lib/modules-load.d
 	install -m644 $(DOC) $(DESTDIR)/usr/share/doc/mbpfan
 	install -d $(DESTDIR)/usr/share/man/man8
 	install -m644 $(MAN) $(DESTDIR)/usr/share/man/man8
