--- OpenOrienteering-Mapper-0.9.6/src/printsupport/qt-5.12.4/printengine_advanced_pdf.cpp.orig	2026-08-09 18:01:00.732325349 +0200
+++ OpenOrienteering-Mapper-0.9.6/src/printsupport/qt-5.12.4/printengine_advanced_pdf.cpp	2026-08-09 18:01:32.713626426 +0200
@@ -34,6 +34,7 @@
 #include <qbuffer.h>
 #include "qprinterinfo.h"
 #include <QtGui/qpagelayout.h>
+#define _LARGEFILE64_SOURCE
 
 #ifdef Q_OS_UNIX
 #include <private/qcore_unix_p.h> // overrides QT_OPEN
