--- src/core/renderables/renderable.h	2020-05-15 19:47:28.000000000 +0200
+++ src/core/renderables/renderable.h	2020-09-02 14:37:36.038494590 +0200
@@ -32,9 +32,9 @@
 
 #include "core/map_color.h"
 
-class QColor;
-class QPainter;
-class QPainterPath;
+#include <QColor>
+#include <QPainter>
+#include <QPainterPath>
 // IWYU pragma: no_forward_declare QRectF
 
 namespace OpenOrienteering {
--- src/core/renderables/renderable_implementation.cpp	2020-05-15 19:47:28.000000000 +0200
+++ src/core/renderables/renderable_implementation.cpp	2020-09-02 14:37:49.659495293 +0200
@@ -31,6 +31,7 @@
 #include <QFontMetricsF>
 #include <QPaintEngine>
 #include <QPainter>
+#include <QPainterPath>
 #include <QPen>
 #include <QPoint>
 #include <QTransform>
--- src/core/symbols/point_symbol.h	2020-05-15 19:47:28.000000000 +0200
+++ src/core/symbols/point_symbol.h	2020-09-02 14:36:40.203491709 +0200
@@ -32,7 +32,7 @@
 
 // IWYU pragma: no_include "core/objects/object.h"
 
-class QPainterPath;
+#include <QPainterPath>
 class QXmlStreamReader;
 class QXmlStreamWriter;
 
--- src/gui/print_tool.cpp	2020-05-15 19:47:28.000000000 +0200
+++ src/gui/print_tool.cpp	2020-09-02 14:38:03.720496018 +0200
@@ -25,6 +25,7 @@
 
 #include <QMouseEvent>
 #include <QPainter>
+#include <QPainterPath>
 
 #include "core/map.h"
 #include "core/map_printer.h"
--- src/printsupport/qt-5.12.4/advanced_pdf.cpp	2020-05-15 19:47:28.000000000 +0200
+++ src/printsupport/qt-5.12.4/advanced_pdf.cpp	2020-09-02 14:40:12.152502644 +0200
@@ -43,6 +43,7 @@
 #include <qnumeric.h>
 #include <qtemporaryfile.h>
 #include <quuid.h>
+#include <QPainterPath>
 
 #ifndef QT_NO_COMPRESS
 #include <zlib.h>
--- src/printsupport/qt-5.12.4/advanced_pdf_p.h	2020-05-15 19:47:28.000000000 +0200
+++ src/printsupport/qt-5.12.4/advanced_pdf_p.h	2020-09-02 14:39:39.432500956 +0200
@@ -43,6 +43,7 @@
 #ifndef QT_NO_PDF
 
 #include "QtGui/qmatrix.h"
+#include "QtGui/qpainterpath.h"
 #include "QtCore/qstring.h"
 #include "QtCore/qvector.h"
 #include <private/qstroker_p.h>
--- src/printsupport/qt-5.5.1/advanced_pdf.cpp	2020-05-15 19:47:28.000000000 +0200
+++ src/printsupport/qt-5.5.1/advanced_pdf.cpp	2020-09-02 14:39:10.743499476 +0200
@@ -47,6 +47,7 @@
 #include <qimagewriter.h>
 #include "qbuffer.h"
 #include "QtCore/qdatetime.h"
+#include <QPainterPath>
 
 #ifndef QT_NO_COMPRESS
 #include <zlib.h>
--- src/printsupport/qt-5.5.1/advanced_pdf_p.h	2020-05-15 19:47:28.000000000 +0200
+++ src/printsupport/qt-5.5.1/advanced_pdf_p.h	2020-09-02 14:38:36.436497706 +0200
@@ -49,6 +49,7 @@
 #ifndef QT_NO_PDF
 
 #include "QtGui/qmatrix.h"
+#include "QtGui/qpainterpath.h"
 #include "QtCore/qstring.h"
 #include "QtCore/qvector.h"
 #include <private/qstroker_p.h>
