From b0997a4cb799433c8073229b507fc49a91d61284 Mon Sep 17 00:00:00 2001
From: guyzmo <guyzmo+github+pub@m0g.net>
Date: Sat, 25 Feb 2017 12:47:14 +0100
Subject: [PATCH] Fixes missing semi-colon compilation issue (#433)

Signed-off-by: Guyzmo <guyzmo+github+pub@m0g.net>
---
 getdomain.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/getdomain.c b/getdomain.c
index 97fd9f4..e939547 100644
--- getdomain.c
+++ getdomain.c
@@ -67,7 +67,7 @@ int getdnsdomainname (char *d, size_t len)
 
 #else /* !HAVE_GETADDRINFO_A */
 
-  getaddrinfo(node, NULL, &hints, &h)
+  getaddrinfo(node, NULL, &hints, &h);
 
 #endif
 
