--- a/Makefile	2017-06-02 21:17:59.554162598 -0500
+++ b/Makefile	2017-06-02 21:18:27.781905157 -0500
@@ -71,10 +71,8 @@
 all:
 	@echo "ROOTDIR=$(ROOTDIR)" > Makefile.global
 	@echo "XOBJ=$(XOBJ)" >> Makefile.global
-	@echo "CFLAGS=-g -ansi -pedantic -Wall -W -O2 -I $(ROOTDIR)/include $(shell sdl-config --cflags)" >> Makefile.global
-	@echo "LDFLAGS=-lz $(shell sdl-config --libs)" >> Makefile.global
-	@echo "CC=gcc" >> Makefile.global
-	@echo "CPP=gcc -E" >> Makefile.global
+	@echo "CFLAGS+=-g -ansi -pedantic -Wall -W -O2 -I $(ROOTDIR)/include $(shell sdl-config --cflags)" >> Makefile.global
+	@echo "LDFLAGS+=-lz $(shell sdl-config --libs)" >> Makefile.global
 	$(MAKE) -C src all
 
 clean:
