--- a/tools/ipv6calc-update-registries.sh~	2019-10-11 07:15:53.000000000 +0200
+++ b/tools/ipv6calc-update-registries.sh	2020-04-28 20:56:41.367743447 +0200
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 #
 # Project    : ipv6calc/databases/registries
 # File       : update-registries.sh
--- a/tools/ipv6calc-registries-check-run-create.sh~	2019-10-11 07:15:53.000000000 +0200
+++ b/tools/ipv6calc-registries-check-run-create.sh	2020-04-28 20:56:42.704763285 +0200
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 #
 # Project    : ipv6calc/databases/ipv4-assignment
 # File       : ipv6calc-registries-check-run-create.sh
