--- a/cmake/Functions.cmake
+++ b/cmake/Functions.cmake
@@ -844,6 +844,8 @@ macro(append_build_type_setup)
         endif()
     elseif(${config} STREQUAL "MinSizeRel")
         list(APPEND gnArgArg is_debug=false symbol_level=0 optimize_for_size=true)
+    elseif(${config} STREQUAL "None")
+        list(APPEND gnArgArg is_debug=false symbol_level=0)
     endif()
     if(FEATURE_developer_build OR (${config} STREQUAL "Debug") OR QT_FEATURE_webengine_sanitizer)
         list(APPEND gnArgArg
