--- a/cmake/QtToolchainHelpers.cmake.orig	2025-09-01 21:12:10.624286293 +0200
+++ b/cmake/QtToolchainHelpers.cmake	2025-09-01 21:12:12.071300756 +0200
@@ -289,6 +289,11 @@
             optimize_for_size=true
             enable_mojom_message_id_scrambling=true
         )
+    elseif(${config} STREQUAL "None")
+    		list(APPEND gnArgArg
+    				is_debug=false
+    				symbol_level=0
+    				enable_mojom_message_id_scrambling=true)
     endif()
     if(FEATURE_developer_build OR (${config} STREQUAL "Debug") OR QT_FEATURE_webengine_sanitizer)
         list(APPEND gnArgArg
