--- a/configure.cmake	2026-01-23 20:06:19.000000000 +0100
+++ -	2026-01-28 22:59:29.757508810 +0100
@@ -388,13 +388,6 @@
     DOCUMENTATION "A pkg-config binary."
     TAGS LINUX_PLATFORM
 )
-qt_webengine_configure_check("glibc"
-    MODULES QtWebEngine
-    CONDITION NOT LINUX OR TEST_glibc
-    MESSAGE "A suitable version >= ${QT_CONFIGURE_CHECK_glibc_version} of glibc is required."
-    DOCUMENTATION "Glibc library at least ${QT_CONFIGURE_CHECK_glibc_version} version or later."
-    TAGS LINUX_PLATFORM
-)
 qt_webengine_configure_check("mesa-headers"
     MODULES QtWebEngine
     CONDITION NOT LINUX OR TEST_khr
