--- a/configure.cmake
+++ b/configure.cmake
@@ -593,6 +593,9 @@ qt_feature("webengine-developer-build" P
     PURPOSE "Enables the developer build configuration."
     AUTODETECT QT_FEATURE_private_tests
 )
+qt_feature("webengine-widevine" PRIVATE
+    LABEL "widevine"
+)
 qt_feature("webengine-system-re2" PRIVATE
     LABEL "re2"
     AUTODETECT FALSE
--- a/src/core/CMakeLists.txt
+++ b/src/core/CMakeLists.txt
@@ -376,7 +376,6 @@ foreach(arch ${archs})
             pdf_enable_fontations=false
             enable_vr=false
             enable_web_speech=false
-            enable_widevine=true
             enable_bound_session_credentials=false
             fatal_linker_warnings=false
             has_native_accessibility=false
@@ -402,6 +401,10 @@ foreach(arch ${archs})
             CONDITION QT_FEATURE_webengine_printing_and_pdf
         )
         extend_gn_list(gnArgArg
+            ARGS enable_widevine
+            CONDITION QT_FEATURE_webengine_widevine
+        )
+        extend_gn_list(gnArgArg
             ARGS enable_plugins
             CONDITION QT_FEATURE_webengine_printing_and_pdf OR
             QT_FEATURE_webengine_pepper_plugins
