--- qt6-webengine-6.4.2.orig/configure.cmake
+++ qt6-webengine-6.4.2/configure.cmake
@@ -499,11 +499,6 @@ add_check_for_support(
 )
 add_check_for_support(
    MODULES QtWebEngine QtPdf
-   CONDITION NOT LINUX OR TEST_glibc
-   MESSAGE "A suitable version >= 2.17 of glibc is required."
-)
-add_check_for_support(
-   MODULES QtWebEngine QtPdf
    CONDITION NOT LINUX OR TEST_khr
    MESSAGE "Build requires Khronos development headers for build - see mesa/libegl1-mesa-dev"
 )
