--- a/src/3rdparty/chromium/base/memory/ref_counted.h
+++ b/src/3rdparty/chromium/base/memory/ref_counted.h
@@ -6,6 +6,7 @@
 #define BASE_MEMORY_REF_COUNTED_H_
 
 #include <stddef.h>
+#include <limits>
 
 #include <limits>
 #include <utility>
--- a/src/3rdparty/chromium/base/check_op.h
+++ b/src/3rdparty/chromium/base/check_op.h
@@ -5,6 +5,7 @@
 #ifndef BASE_CHECK_OP_H_
 #define BASE_CHECK_OP_H_
 
+#include <cstdint>
 #include <cstddef>
 #include <string>
 #include <string_view>
--- a/src/3rdparty/chromium/base/debug/profiler.h
+++ b/src/3rdparty/chromium/base/debug/profiler.h
@@ -8,6 +8,7 @@
 #include <stddef.h>
 #include <stdint.h>
 
+#include <cstdint>
 #include <string>
 
 #include "base/base_export.h"
--- a/src/3rdparty/chromium/gpu/config/gpu_feature_info.h
+++ b/src/3rdparty/chromium/gpu/config/gpu_feature_info.h
@@ -7,6 +7,7 @@
 
 #include <stdint.h>
 
+#include <cstdint>
 #include <string>
 #include <vector>
 
--- a/src/3rdparty/chromium/net/base/net_export.h
+++ b/src/3rdparty/chromium/net/base/net_export.h
@@ -5,6 +5,8 @@
 #ifndef NET_BASE_NET_EXPORT_H_
 #define NET_BASE_NET_EXPORT_H_
 
+#include <cstdint>
+
 // Defines NET_EXPORT so that functionality implemented by the net module can
 // be exported to consumers, and NET_EXPORT_PRIVATE that allows unit tests to
 // access features not intended to be used directly by real consumers.
--- a/src/3rdparty/chromium/sandbox/linux/syscall_broker/broker_file_permission.h
+++ b/src/3rdparty/chromium/sandbox/linux/syscall_broker/broker_file_permission.h
@@ -5,6 +5,7 @@
 #ifndef SANDBOX_LINUX_SYSCALL_BROKER_BROKER_FILE_PERMISSION_H_
 #define SANDBOX_LINUX_SYSCALL_BROKER_BROKER_FILE_PERMISSION_H_
 
+#include <cstdint>
 #include <bitset>
 #include <cstdint>
 #include <string>
--- a/src/3rdparty/chromium/third_party/abseil-cpp/absl/strings/string_view.h
+++ b/src/3rdparty/chromium/third_party/abseil-cpp/absl/strings/string_view.h
@@ -27,6 +27,7 @@
 #ifndef ABSL_STRINGS_STRING_VIEW_H_
 #define ABSL_STRINGS_STRING_VIEW_H_
 
+#include <cstdint>
 #include <algorithm>
 #include <cassert>
 #include <cstddef>
--- a/src/3rdparty/chromium/third_party/angle/include/GLSLANG/ShaderVars.h
+++ b/src/3rdparty/chromium/third_party/angle/include/GLSLANG/ShaderVars.h
@@ -10,6 +10,7 @@
 #ifndef GLSLANG_SHADERVARS_H_
 #define GLSLANG_SHADERVARS_H_
 
+#include <cstdint>
 #include <algorithm>
 #include <array>
 #include <cstdint>
--- a/src/3rdparty/chromium/third_party/blink/public/common/bluetooth/web_bluetooth_device_id.h
+++ b/src/3rdparty/chromium/third_party/blink/public/common/bluetooth/web_bluetooth_device_id.h
@@ -7,6 +7,7 @@
 
 #include <stdint.h>
 
+#include <cstdint>
 #include <array>
 #include <cstdint>
 #include <functional>
--- a/src/3rdparty/chromium/third_party/dawn/src/tint/lang/spirv/reader/ast_parser/namer.h
+++ b/src/3rdparty/chromium/third_party/dawn/src/tint/lang/spirv/reader/ast_parser/namer.h
@@ -15,6 +15,7 @@
 #ifndef SRC_TINT_LANG_SPIRV_READER_AST_PARSER_NAMER_H_
 #define SRC_TINT_LANG_SPIRV_READER_AST_PARSER_NAMER_H_
 
+#include <cstdint>
 #include <string>
 #include <unordered_map>
 #include <vector>
--- a/src/3rdparty/chromium/third_party/openscreen/src/discovery/dnssd/public/dns_sd_txt_record.h
+++ b/src/3rdparty/chromium/third_party/openscreen/src/discovery/dnssd/public/dns_sd_txt_record.h
@@ -7,6 +7,7 @@
 
 #include <stdint.h>
 
+#include <cstdint>
 #include <functional>
 #include <map>
 #include <set>
--- a/src/3rdparty/chromium/third_party/swiftshader/src/System/LRUCache.hpp
+++ b/src/3rdparty/chromium/third_party/swiftshader/src/System/LRUCache.hpp
@@ -17,6 +17,7 @@
 
 #include "System/Debug.hpp"
 
+#include <cstdint>
 #include <cstddef>
 #include <cstdint>
 #include <functional>
--- a/src/3rdparty/chromium/ui/gfx/geometry/linear_gradient.h
+++ b/src/3rdparty/chromium/ui/gfx/geometry/linear_gradient.h
@@ -7,6 +7,7 @@
 
 #include <stdint.h>
 
+#include <cstdint>
 #include <array>
 #include <cstddef>
 #include <cstdint>
--- a/src/3rdparty/chromium/third_party/ruy/src/ruy/profiler/instrumentation.h
+++ b/src/3rdparty/chromium/third_party/ruy/src/ruy/profiler/instrumentation.h
@@ -17,6 +17,7 @@ limitations under the License.
 #define RUY_RUY_PROFILER_INSTRUMENTATION_H_
 
 #ifdef RUY_PROFILER
+#include <string>
 #include <cstdio>
 #include <mutex>
 #include <vector>
--- a/src/3rdparty/chromium/base/containers/flat_map.h
+++ b/src/3rdparty/chromium/base/containers/flat_map.h
@@ -5,6 +5,7 @@
 #ifndef BASE_CONTAINERS_FLAT_MAP_H_
 #define BASE_CONTAINERS_FLAT_MAP_H_
 
+#include <cstdint>
 #include <functional>
 #include <tuple>
 #include <type_traits>
--- a/src/3rdparty/chromium/components/crash/core/app/crash_reporter_client.h
+++ b/src/3rdparty/chromium/components/crash/core/app/crash_reporter_client.h
@@ -7,6 +7,7 @@
 
 #include <stdint.h>
 
+#include <cstdint>
 #include <string>
 
 #include "build/build_config.h"
--- a/src/3rdparty/chromium/ui/base/prediction/kalman_filter.h
+++ b/src/3rdparty/chromium/ui/base/prediction/kalman_filter.h
@@ -7,6 +7,8 @@
 
 #include <stdint.h>
 
+#include <cstdint>
+
 #include "base/component_export.h"
 #include "ui/gfx/geometry/matrix3_f.h"
 
--- a/src/3rdparty/chromium/components/password_manager/core/browser/generation/password_generator.h
+++ b/src/3rdparty/chromium/components/password_manager/core/browser/generation/password_generator.h
@@ -7,6 +7,7 @@
 
 #include <stdint.h>
 
+#include <cstdint>
 #include <string>
 
 
--- a/src/3rdparty/chromium/components/feature_engagement/internal/event_storage_validator.h
+++ b/src/3rdparty/chromium/components/feature_engagement/internal/event_storage_validator.h
@@ -7,6 +7,7 @@
 
 #include <stdint.h>
 
+#include <cstdint>
 #include <string>
 
 namespace feature_engagement {
--- a/src/3rdparty/chromium/components/feature_engagement/internal/never_event_storage_validator.h
+++ b/src/3rdparty/chromium/components/feature_engagement/internal/never_event_storage_validator.h
@@ -5,6 +5,7 @@
 #ifndef COMPONENTS_FEATURE_ENGAGEMENT_INTERNAL_NEVER_EVENT_STORAGE_VALIDATOR_H_
 #define COMPONENTS_FEATURE_ENGAGEMENT_INTERNAL_NEVER_EVENT_STORAGE_VALIDATOR_H_
 
+#include <cstdint>
 #include <string>
 
 #include "components/feature_engagement/internal/event_storage_validator.h"
--- a/src/3rdparty/chromium/components/autofill/core/browser/autofill_ablation_study.h
+++ b/src/3rdparty/chromium/components/autofill/core/browser/autofill_ablation_study.h
@@ -7,6 +7,7 @@
 
 #include <stdint.h>
 
+#include <cstdint>
 #include <string>
 
 class GURL;
--- a/src/3rdparty/chromium/components/payments/content/utility/fingerprint_parser.h
+++ b/src/3rdparty/chromium/components/payments/content/utility/fingerprint_parser.h
@@ -5,6 +5,7 @@
 #ifndef COMPONENTS_PAYMENTS_CONTENT_UTILITY_FINGERPRINT_PARSER_H_
 #define COMPONENTS_PAYMENTS_CONTENT_UTILITY_FINGERPRINT_PARSER_H_
 
+#include <cstdint>
 #include <stddef.h>
 #include <stdint.h>
 
--- a/src/3rdparty/chromium/pdf/document_attachment_info.h
+++ b/src/3rdparty/chromium/pdf/document_attachment_info.h
@@ -7,6 +7,7 @@
 
 #include <stdint.h>
 
+#include <cstdint>
 #include <string>
 
 
--- a/src/3rdparty/chromium/third_party/pdfium/constants/annotation_flags.h
+++ b/src/3rdparty/chromium/third_party/pdfium/constants/annotation_flags.h
@@ -7,6 +7,8 @@
 
 #include <stdint.h>
 
+#include <cstdint>
+
 namespace pdfium {
 namespace annotation_flags {
 
--- a/src/3rdparty/chromium/third_party/vulkan-deps/vulkan-validation-layers/src/layers/external/vma/vk_mem_alloc.h
+++ b/src/3rdparty/chromium/third_party/vulkan-deps/vulkan-validation-layers/src/layers/external/vma/vk_mem_alloc.h
@@ -2884,6 +2884,7 @@ static void vma_aligned_free(void* VMA_N
 
 // Define this macro to 1 to enable functions: vmaBuildStatsString, vmaFreeStatsString.
 #if VMA_STATS_STRING_ENABLED
+#include <stdio.h>
     static inline void VmaUint32ToStr(char* VMA_NOT_NULL outStr, size_t strLen, uint32_t num)
     {
         snprintf(outStr, strLen, "%u", static_cast<unsigned int>(num));
--- a/src/3rdparty/chromium/gin/time_clamper.h
+++ b/src/3rdparty/chromium/gin/time_clamper.h
@@ -48,7 +48,7 @@ class GIN_EXPORT TimeClamper {
     const int64_t micros = now_micros % 1000;
     // abs() is necessary for devices with times before unix-epoch (most likely
     // configured incorrectly).
-    if (abs(micros) + kResolutionMicros < 1000) {
+    if (std::abs(micros) + kResolutionMicros < 1000) {
       return now_micros / 1000;
     }
     return ClampTimeResolution(now_micros) / 1000;
--- a/src/3rdparty/chromium/chrome/test/chromedriver/chrome/web_view_impl.cc
+++ b/src/3rdparty/chromium/chrome/test/chromedriver/chrome/web_view_impl.cc
@@ -11,6 +11,7 @@
 #include <queue>
 #include <utility>
 #include <vector>
+#include <cstring>
 
 #include "base/check.h"
 #include "base/files/file_path.h"
--- a/src/3rdparty/chromium/skia/ext/skcolorspace_trfn.cc
+++ b/src/3rdparty/chromium/skia/ext/skcolorspace_trfn.cc
@@ -3,6 +3,7 @@
 // found in the LICENSE file.
 
 #include "skia/ext/skcolorspace_trfn.h"
+#include <cmath>
 
 #include <cmath>
 
--- a/src/3rdparty/chromium/third_party/webrtc/common_video/h264/sps_parser.h
+++ b/src/3rdparty/chromium/third_party/webrtc/common_video/h264/sps_parser.h
@@ -11,6 +11,7 @@
 #ifndef COMMON_VIDEO_H264_SPS_PARSER_H_
 #define COMMON_VIDEO_H264_SPS_PARSER_H_
 
+#include <cstdint>
 #include "absl/types/optional.h"
 #include "rtc_base/bitstream_reader.h"
 
--- a/src/3rdparty/chromium/third_party/webrtc/modules/include/module_common_types_public.h
+++ b/src/3rdparty/chromium/third_party/webrtc/modules/include/module_common_types_public.h
@@ -11,6 +11,7 @@
 #ifndef MODULES_INCLUDE_MODULE_COMMON_TYPES_PUBLIC_H_
 #define MODULES_INCLUDE_MODULE_COMMON_TYPES_PUBLIC_H_
 
+#include <cstdint>
 #include <limits>
 
 #include "absl/types/optional.h"
--- a/src/3rdparty/chromium/ui/gfx/linux/drm_util_linux.h
+++ b/src/3rdparty/chromium/ui/gfx/linux/drm_util_linux.h
@@ -9,6 +9,8 @@
 
 #include "ui/gfx/buffer_types.h"
 
+#include <cstdint>
+
 namespace ui {
 
 int GetFourCCFormatFromBufferFormat(gfx::BufferFormat format);
--- a/src/3rdparty/chromium/third_party/webrtc/audio/utility/channel_mixer.cc
+++ b/src/3rdparty/chromium/third_party/webrtc/audio/utility/channel_mixer.cc
@@ -8,6 +8,8 @@
  *  be found in the AUTHORS file in the root of the source tree.
  */
 
+#include <cstring>
+
 #include "audio/utility/channel_mixer.h"
 
 #include "audio/utility/channel_mixing_matrix.h"
--- a/src/3rdparty/chromium/third_party/webrtc/modules/video_coding/utility/ivf_file_reader.cc
+++ b/src/3rdparty/chromium/third_party/webrtc/modules/video_coding/utility/ivf_file_reader.cc
@@ -10,6 +10,7 @@
 
 #include "modules/video_coding/utility/ivf_file_reader.h"
 
+#include <cstring>
 #include <string>
 #include <vector>
 
--- a/src/3rdparty/chromium/third_party/ipcz/src/ipcz/router_link.h
+++ b/src/3rdparty/chromium/third_party/ipcz/src/ipcz/router_link.h
@@ -5,6 +5,7 @@
 #ifndef IPCZ_SRC_IPCZ_ROUTER_LINK_H_
 #define IPCZ_SRC_IPCZ_ROUTER_LINK_H_
 
+#include <memory>
 #include <cstddef>
 #include <functional>
 #include <memory>
--- a/src/3rdparty/chromium/third_party/material_color_utilities/src/cpp/palettes/tones.cc
+++ b/src/3rdparty/chromium/third_party/material_color_utilities/src/cpp/palettes/tones.cc
@@ -14,6 +14,7 @@
  * limitations under the License.
  */
 
+#include <cmath>
 #include "cpp/palettes/tones.h"
 
 #include <cmath>
--- a/src/3rdparty/chromium/services/device/public/cpp/generic_sensor/sensor_reading.h
+++ b/src/3rdparty/chromium/services/device/public/cpp/generic_sensor/sensor_reading.h
@@ -8,6 +8,9 @@
 #include <stddef.h>
 #include <stdint.h>
 
+#include <cstddef>
+#include <cstdint>
+
 #include <type_traits>
 
 namespace device {
--- a/src/3rdparty/chromium/components/gwp_asan/client/lightweight_detector.h
+++ b/src/3rdparty/chromium/components/gwp_asan/client/lightweight_detector.h
@@ -5,6 +5,8 @@
 #ifndef COMPONENTS_GWP_ASAN_CLIENT_LIGHTWEIGHT_DETECTOR_H_
 #define COMPONENTS_GWP_ASAN_CLIENT_LIGHTWEIGHT_DETECTOR_H_
 
+#include <atomic>
+
 #include "base/gtest_prod_util.h"
 #include "components/gwp_asan/client/export.h"
 #include "components/gwp_asan/common/lightweight_detector_state.h"
