--- a/src/3rdparty/chromium/media/gpu/vaapi/BUILD.gn
+++ b/src/3rdparty/chromium/media/gpu/vaapi/BUILD.gn
@@ -13,6 +13,12 @@ import("//tools/generate_stubs/rules.gni
 assert(is_linux || is_chromeos)
 assert(use_vaapi)
 
+config("vaapi_permissive") {
+  if (target_cpu == "x86") {
+    cflags = [ "-fpermissive" ]
+  }
+}
+
 generate_stubs("libva_stubs") {
   extra_header = "va_stub_header.fragment"
   sigs = [ "va.sigs" ]
@@ -81,6 +87,7 @@ source_set("vaapi") {
   configs += [
     "//build/config/linux/libva",
     "//third_party/libvpx:libvpx_config",
+    ":vaapi_permissive",
   ]
 
   deps = [
