This was dropped for some reason in 6951c37cecd05979b232a39e5c10e6346a0f74ef
--- a/src/3rdparty/chromium/third_party/closure_compiler/compiler.py
+++ b/src/3rdparty/chromium/third_party/closure_compiler/compiler.py
@@ -13,8 +13,9 @@ import subprocess
 
 
 _CURRENT_DIR = os.path.join(os.path.dirname(__file__))
-_JAVA_PATH = os.path.join(_CURRENT_DIR, "..", "jdk", "current", "bin", "java")
-assert os.path.isfile(_JAVA_PATH), "java only allowed in android builds"
+_JAVA_BIN = "java"
+_JDK_PATH = os.path.join(_CURRENT_DIR, "..", "jdk", "current", "bin", "java")
+_JAVA_PATH = _JDK_PATH if os.path.isfile(_JDK_PATH) else _JAVA_BIN
 
 class Compiler(object):
   """Runs the Closure compiler on given source files to typecheck them
