--- a/src/3rdparty/chromium/third_party/node/node.py
+++ b/src/3rdparty/chromium/third_party/node/node.py
@@ -32,11 +32,12 @@ def GetBinaryPath():
 
   darwin_name = ('node-darwin-arm64' if platform.machine() == 'arm64' else
                  'node-darwin-x64')
-  return os_path.join(os_path.dirname(__file__), *{
-    'Darwin': ('mac', darwin_name, 'bin', 'node'),
-    'Linux': ('linux', 'node-linux-x64', 'bin', 'node'),
-    'Windows': ('win', 'node.exe'),
-  }[platform.system()])
+  #return os_path.join(os_path.dirname(__file__), *{
+  #  'Darwin': ('mac', darwin_name, 'bin', 'node'),
+  #  'Linux': ('linux', 'node-linux-x64', 'bin', 'node'),
+  #  'Windows': ('win', 'node.exe'),
+  #}[platform.system()])
+  return "/usr/bin/node"
 
 
 def RunNode(cmd_parts, stdout=None):
