--- a/src/3rdparty/chromium/base/allocator/partition_allocator/partition_root.cc
+++ b/src/3rdparty/chromium/base/allocator/partition_allocator/partition_root.cc
@@ -281,7 +281,7 @@ void PartitionAllocMallocInitOnce() {
     return;
   }
 
-#if BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS)
+#if (BUILDFLAG(IS_LINUX) && defined(__GLIBC__)) || BUILDFLAG(IS_CHROMEOS)
   // When fork() is called, only the current thread continues to execute in the
   // child process. If the lock is held, but *not* by this thread when fork() is
   // called, we have a deadlock.
