--- a/src/3rdparty/chromium/base/allocator/partition_allocator/partition_alloc_config.h
+++ b/src/3rdparty/chromium/base/allocator/partition_allocator/partition_alloc_config.h
@@ -153,7 +153,7 @@ static_assert(sizeof(void*) != 8, "");
    defined(ARCH_CPU_LITTLE_ENDIAN))
 
 #define PA_CONFIG_HAS_MEMORY_TAGGING()              \
-  (defined(ARCH_CPU_ARM64) && defined(__clang__) && \
+  (0 && defined(ARCH_CPU_ARM64) && defined(__clang__) && \
    !defined(ADDRESS_SANITIZER) &&                   \
    (BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_ANDROID)))
 
