--- a/src/3rdparty/chromium/mojo/public/tools/mojom/mojom/fileutil.py	2023-10-03 21:49:54.000000000 +0200
+++ -	2023-10-13 23:26:56.151025878 +0200
@@ -3,7 +3,6 @@
 # found in the LICENSE file.
 
 import errno
-import imp
 import os.path
 import sys
 
--- a/src/3rdparty/chromium/mojo/public/tools/mojom/mojom/parse/lexer.py	2023-10-03 21:49:54.000000000 +0200
+++ -	2023-10-13 23:28:56.391987051 +0200
@@ -2,7 +2,6 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-import imp
 import os.path
 import sys

