--- a/src/3rdparty/chromium/BUILD.gn
+++ b/src/3rdparty/chromium/BUILD.gn
@@ -1647,9 +1647,7 @@ if (!is_ios && !is_qtwebengine) {
 }
 
 # TODO(cassew): Add more OS's that don't support x86.
-is_valid_x86_target =
-    target_os != "ios" && target_os != "mac" &&
-    (target_os != "linux" || use_libfuzzer || !build_with_chromium)
+is_valid_x86_target = true
 
 # Note: v8_target_cpu == arm allows using the V8 arm simulator on x86 for fuzzing.
 assert(
--- a/src/host/CMakeLists.txt
+++ b/src/host/CMakeLists.txt
@@ -48,18 +48,3 @@ if(QT_FEATURE_qtpdf_build)
         DESTINATION ${WEBENGINE_ROOT_BUILD_DIR}/src/pdf/v8_toolchain
     )
 endif()
-
-# TODO: this could be run as part of main configure with execute_process
-
-if(CMAKE_CXX_COMPILER_ID STREQUAL GNU AND TEST_architecture_arch STREQUAL "x86_64"
-AND GN_TARGET_CPU STREQUAL "arm")
-    try_compile(
-        has32HostCompiler
-        "${CMAKE_CURRENT_BINARY_DIR}/config.tests/hostcompiler"
-        "${CMAKE_CURRENT_SOURCE_DIR}/config.tests/hostcompiler"
-        hostcompiler
-    )
-    if(NOT has32HostCompiler)
-        MESSAGE(FATAL_ERROR "Compiler does not support 32bit compilation")
-    endif()
-endif()
