Index: CMakeLists.txt
===================================================================
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6,8 +6,6 @@ project( zbackup )
 
 set( CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake" )
 
-set( CMAKE_BUILD_TYPE Release )
-
 find_package( ZLIB REQUIRED )
 include_directories( ${ZLIB_INCLUDE_DIRS} )
 
Index: tartool/CMakeLists.txt
===================================================================
--- a/tartool/CMakeLists.txt
+++ b/tartool/CMakeLists.txt
@@ -4,8 +4,6 @@
 cmake_minimum_required( VERSION 2.6.0 )
 project( tartool )
 
-set( CMAKE_BUILD_TYPE Release )
-
 add_executable( tartool tartool.cc ../file.cc ../dir.cc )
 
 install( TARGETS tartool DESTINATION bin )
