diff --git a/radicle/src/io.rs b/radicle/src/io.rs
index 8af88728..7e319ef4 100644
--- a/radicle/src/io.rs
+++ b/radicle/src/io.rs
@@ -1,12 +1,9 @@
 use std::fmt;
 use std::io;
 
-use libc::{getrlimit, rlimit, setrlimit, RLIMIT_NOFILE};
+use libc::{getrlimit, rlim_t, rlimit, setrlimit, RLIMIT_NOFILE};
 
-#[cfg(not(target_os = "freebsd"))]
-type Int = u64;
-#[cfg(target_os = "freebsd")]
-type Int = i64;
+type Int = rlim_t;
 
 /// Sets the open file limit to the given value, or the maximum allowed value.
 pub fn set_file_limit<N>(n: N) -> io::Result<Int>
