Author: Daniel Baumann <daniel.baumann@progress-linux.org>
Description: Using system zoneinfo by default.

diff -Naurp pytzdata.orig/pytzdata/__init__.py pytzdata/pytzdata/__init__.py
--- pytzdata.orig/pytzdata/__init__.py
+++ pytzdata/pytzdata/__init__.py
@@ -7,10 +7,7 @@ from ._timezones import timezones
 from ._compat import FileNotFoundError
 
 
-DEFAULT_DIRECTORY = os.path.join(
-    os.path.dirname(__file__),
-    'zoneinfo'
-)
+DEFAULT_DIRECTORY = '/usr/share/zoneinfo'
 
 _DIRECTORY = os.getenv('PYTZDATA_TZDATADIR', DEFAULT_DIRECTORY)
 
