--- a/Source/cmFileCommand.cxx
+++ b/Source/cmFileCommand.cxx
@@ -2153,10 +2153,6 @@ bool HandleDownloadCommand(std::vector<s
   check_curl_result(res, "DOWNLOAD cannot set http failure option: ");
 
   curl_version_info_data* cv = curl_version_info(CURLVERSION_FIRST);
-  res = ::curl_easy_setopt(
-    curl, CURLOPT_USERAGENT,
-    cmStrCat("curl/", cv ? cv->version : LIBCURL_VERSION).c_str());
-  check_curl_result(res, "DOWNLOAD cannot set user agent option: ");
 
   res = ::curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, cmWriteToFileCallback);
   check_curl_result(res, "DOWNLOAD cannot set write function: ");
@@ -2181,15 +2177,6 @@ bool HandleDownloadCommand(std::vector<s
     }
   }
 
-  // check to see if TLS verification is requested
-  if (tlsVerifyOpt.has_value() && tlsVerifyOpt.value()) {
-    res = ::curl_easy_setopt(curl, CURLOPT_SSL_VERIFYPEER, 1);
-    check_curl_result(res, "DOWNLOAD cannot set TLS/SSL Verify on: ");
-  } else {
-    res = ::curl_easy_setopt(curl, CURLOPT_SSL_VERIFYPEER, 0);
-    check_curl_result(res, "DOWNLOAD cannot set TLS/SSL Verify off: ");
-  }
-
   for (auto const& range : curl_ranges) {
     std::string curl_range = range.first + '-' +
       (range.second.has_value() ? range.second.value() : "");
@@ -2224,25 +2211,11 @@ bool HandleDownloadCommand(std::vector<s
   res = ::curl_easy_setopt(curl, CURLOPT_DEBUGDATA, &chunkDebug);
   check_curl_result(res, "DOWNLOAD cannot set debug data: ");
 
-  res = ::curl_easy_setopt(curl, CURLOPT_FOLLOWLOCATION, 1L);
-  check_curl_result(res, "DOWNLOAD cannot set follow-redirect option: ");
-
   if (!logVar.empty()) {
     res = ::curl_easy_setopt(curl, CURLOPT_VERBOSE, 1);
     check_curl_result(res, "DOWNLOAD cannot set verbose: ");
   }
 
-  if (timeout > 0) {
-    res = ::curl_easy_setopt(curl, CURLOPT_TIMEOUT, timeout);
-    check_curl_result(res, "DOWNLOAD cannot set timeout: ");
-  }
-
-  if (inactivity_timeout > 0) {
-    // Give up if there is no progress for a long time.
-    ::curl_easy_setopt(curl, CURLOPT_LOW_SPEED_LIMIT, 1);
-    ::curl_easy_setopt(curl, CURLOPT_LOW_SPEED_TIME, inactivity_timeout);
-  }
-
   // Need the progress helper's scope to last through the duration of
   // the curl_easy_perform call... so this object is declared at function
   // scope intentionally, rather than inside the "if(showProgress)"
@@ -2579,15 +2552,6 @@ bool HandleUploadCommand(std::vector<std
     }
   }
 
-  // check to see if TLS verification is requested
-  if (tlsVerifyOpt.has_value() && tlsVerifyOpt.value()) {
-    res = ::curl_easy_setopt(curl, CURLOPT_SSL_VERIFYPEER, 1);
-    check_curl_result(res, "UPLOAD cannot set TLS/SSL Verify on: ");
-  } else {
-    res = ::curl_easy_setopt(curl, CURLOPT_SSL_VERIFYPEER, 0);
-    check_curl_result(res, "UPLOAD cannot set TLS/SSL Verify off: ");
-  }
-
   // check to see if a CAINFO file has been specified
   // command arg comes first
   std::string const& cainfo_err = cmCurlSetCAInfo(curl, cainfo);
@@ -2605,25 +2569,11 @@ bool HandleUploadCommand(std::vector<std
   res = ::curl_easy_setopt(curl, CURLOPT_DEBUGDATA, &chunkDebug);
   check_curl_result(res, "UPLOAD cannot set debug data: ");
 
-  res = ::curl_easy_setopt(curl, CURLOPT_FOLLOWLOCATION, 1L);
-  check_curl_result(res, "UPLOAD cannot set follow-redirect option: ");
-
   if (!logVar.empty()) {
     res = ::curl_easy_setopt(curl, CURLOPT_VERBOSE, 1);
     check_curl_result(res, "UPLOAD cannot set verbose: ");
   }
 
-  if (timeout > 0) {
-    res = ::curl_easy_setopt(curl, CURLOPT_TIMEOUT, timeout);
-    check_curl_result(res, "UPLOAD cannot set timeout: ");
-  }
-
-  if (inactivity_timeout > 0) {
-    // Give up if there is no progress for a long time.
-    ::curl_easy_setopt(curl, CURLOPT_LOW_SPEED_LIMIT, 1);
-    ::curl_easy_setopt(curl, CURLOPT_LOW_SPEED_TIME, inactivity_timeout);
-  }
-
   // Need the progress helper's scope to last through the duration of
   // the curl_easy_perform call... so this object is declared at function
   // scope intentionally, rather than inside the "if(showProgress)"
--- a/Tests/RunCMake/CMakeLists.txt
+++ b/Tests/RunCMake/CMakeLists.txt
@@ -595,7 +595,6 @@ if(COVERAGE_COMMAND)
   add_RunCMake_test(ctest_coverage -DCOVERAGE_COMMAND=${COVERAGE_COMMAND})
 endif()
 add_RunCMake_test(ctest_start)
-add_RunCMake_test(ctest_submit)
 add_RunCMake_test(ctest_test
   -DIMAGE_DIR=${CMAKE_SOURCE_DIR}/Utilities/Sphinx/static
 )
@@ -623,7 +622,6 @@ foreach(var
     list(APPEND file-DOWNLOAD_ARGS -D${var}=${${var}})
   endif()
 endforeach()
-add_RunCMake_test(file-DOWNLOAD)
 add_RunCMake_test(file-MAKE_DIRECTORY)
 add_RunCMake_test(file-RPATH
   -DCMAKE_SYSTEM_NAME=${CMAKE_SYSTEM_NAME}
--- a/Utilities/cmcurl/CMakeLists.txt
+++ b/Utilities/cmcurl/CMakeLists.txt
@@ -36,13 +36,13 @@ set(CURL_DISABLE_DIGEST_AUTH OFF)
 set(CURL_DISABLE_DOH OFF)
 set(CURL_DISABLE_FILE OFF CACHE INTERNAL "Disable curl file protocol?")
 set(CURL_DISABLE_FORM_API OFF)
-set(CURL_DISABLE_FTP OFF CACHE INTERNAL "Disable curl ftp protocol?")
+set(CURL_DISABLE_FTP ON CACHE INTERNAL "Disable curl ftp protocol?")
 set(CURL_DISABLE_GETOPTIONS OFF)
 set(CURL_DISABLE_GOPHER ON CACHE INTERNAL "Disable curl gopher protocol?")
 set(CURL_DISABLE_HEADERS_API OFF)
 set(CURL_DISABLE_HSTS OFF)
-set(CURL_DISABLE_HTTP_AUTH OFF)
-set(CURL_DISABLE_HTTP OFF CACHE INTERNAL "Disable curl http protocol?")
+set(CURL_DISABLE_HTTP_AUTH ON)
+set(CURL_DISABLE_HTTP ON CACHE INTERNAL "Disable curl http protocol?")
 set(CURL_DISABLE_IMAP ON CACHE INTERNAL "Disable curl imap protocol?")
 set(CURL_DISABLE_INSTALL ON)
 set(CURL_DISABLE_IPFS OFF)
@@ -59,13 +59,13 @@ set(CURL_DISABLE_OPENSSL_AUTO_LOAD_CONFI
 set(CURL_DISABLE_PARSEDATE OFF)
 set(CURL_DISABLE_POP3 ON CACHE INTERNAL "Disable curl pop3 protocol?")
 set(CURL_DISABLE_PROGRESS_METER OFF)
-set(CURL_DISABLE_PROXY OFF CACHE INTERNAL "Do not disable curl proxy")
+set(CURL_DISABLE_PROXY ON CACHE INTERNAL "Do not disable curl proxy")
 set(CURL_DISABLE_RTSP ON CACHE INTERNAL "Disable curl rtsp protocol?")
 set(CURL_DISABLE_SHA512_256 OFF)
 set(CURL_DISABLE_SHUFFLE_DNS OFF)
 set(CURL_DISABLE_SMB OFF)
 set(CURL_DISABLE_SMTP ON CACHE INTERNAL "Disable curl smtp protocol?")
-set(CURL_DISABLE_SOCKETPAIR OFF)
+set(CURL_DISABLE_SOCKETPAIR ON)
 set(CURL_DISABLE_SRP OFF)
 set(CURL_DISABLE_TELNET ON CACHE INTERNAL "Disable curl telnet protocol?")
 set(CURL_DISABLE_TFTP ON CACHE INTERNAL "Disable curl tftp protocol?")
@@ -84,7 +84,7 @@ set(ENABLE_ALT_SVC OFF)
 set(ENABLE_CURLDEBUG OFF CACHE INTERNAL "No curl TrackMemory features")
 set(ENABLE_DEBUG OFF CACHE INTERNAL "No curl debug features")
 set(ENABLE_INET_PTON OFF CACHE INTERNAL "Set to OFF to prevent usage of inet_pton when building against modern SDKs while still requiring compatibility with older Windows versions, such as Windows XP, Windows Server 2003 etc.")
-set(ENABLE_IPV6 ON CACHE INTERNAL "Enable curl IPv6 support detection")
+set(ENABLE_IPV6 OFF CACHE INTERNAL "Enable curl IPv6 support detection")
 set(ENABLE_MANUAL OFF CACHE INTERNAL "No curl built-in manual")
 set(ENABLE_THREADED_RESOLVER OFF CACHE INTERNAL "No curl POSIX threaded DNS lookup")
 set(ENABLE_UNICODE OFF)
@@ -103,9 +103,9 @@ set(PICKY_COMPILER OFF)
 set(SHARE_LIB_OBJECT OFF)
 set(USE_ECH OFF)
 set(USE_HTTPSRR OFF)
-set(USE_LIBIDN2 ON)
+set(USE_LIBIDN2 OFF)
 set(USE_LIBRTMP OFF)
-set(USE_NGHTTP2 ON)
+set(USE_NGHTTP2 OFF)
 set(USE_NGTCP2 OFF)
 set(USE_OPENSSL_QUIC OFF)
 set(USE_QUICHE OFF)
--- a/Utilities/cmcurl/lib/strerror.c
+++ b/Utilities/cmcurl/lib/strerror.c
@@ -61,7 +61,7 @@ curl_easy_strerror(CURLcode error)
     return "No error";
 
   case CURLE_UNSUPPORTED_PROTOCOL:
-    return "Unsupported protocol";
+    return "Unsupported protocol, xbps-src's cmake has no network";
 
   case CURLE_FAILED_INIT:
     return "Failed initialization";
