--- Makefile.orig	2013-02-11 15:06:43.831345892 +0100
+++ Makefile	2013-02-11 15:07:53.196725322 +0100
@@ -25,7 +25,7 @@ apptest.o: base.h apptest.c app.h list.h
 list.o: base.h list.h list.c
 
 $(SONAME).$(VER):	$(OBJS)
-	$(CC) $(MY_CFLAGS) $(CFLAGS) $(LDFLAGS) -Wl,-soname=$(SONAME) -shared  -o $@ $(OBJS)
+	$(CC) $(MY_CFLAGS) $(CFLAGS) $(LDFLAGS) -Wl,-soname=$(SONAME).0 -shared  -o $@ $(OBJS)
 
 .c.o:
 	$(CC) $(MY_CFLAGS) $(CFLAGS) $(CPPFLAGS) -c $< -o $@
@@ -44,8 +44,9 @@ clean:
 
 install: $(SONAME) libapp.pc
 	install -d '$(LIBDIR)'
-	$(INSTALL_S) -t '$(LIBDIR)' $(SONAME).$(VER)
+	install -m 755 $(SONAME).$(VER) $(LIBDIR)
 	$(LN_SF) $(SONAME).$(VER) '$(LIBDIR)'/$(SONAME)
+	$(LN_SF) $(SONAME).$(VER) '$(LIBDIR)'/$(SONAME).0
 	install -d '$(INCAPPDIR)'
 	install -m 644 -t '$(INCAPPDIR)' $(HEADERS)
 	install -d '$(PKGCONFIGDIR)'
