--- Project/GNU/Library/test/utils.sh	2020-10-17 17:36:11.648662226 +0200
+++ -	2020-10-17 17:38:40.423209231 +0200
@@ -1,26 +1,6 @@
 #!/bin/sh
 UTILS_PATH=$(dirname "$0")
 
-if [ ! -f "$UTILS_PATH/mediatrace.xsd" ]
-then
-    wget https://github.com/MediaArea/MediaAreaXml/raw/master/mediatrace.xsd -O "$UTILS_PATH/mediatrace.xsd"
-fi
-
-if [ ! -f "$UTILS_PATH/micromediatrace.xsd" ]
-then
-    wget https://github.com/MediaArea/MediaAreaXml/raw/master/micromediatrace.xsd -O "$UTILS_PATH/micromediatrace.xsd"
-fi
-
-if [ ! -f "$UTILS_PATH/mediainfo_2_0.xsd" ]
-then
-    wget https://mediaarea.net/mediainfo/mediainfo_2_0.xsd -O "$UTILS_PATH/mediainfo.xsd"
-fi
-
-if [ ! -f "$UTILS_PATH/mediaarea.xsd" ]
-then
-    wget https://github.com/MediaArea/MediaAreaXml/raw/master/mediaarea.xsd -O "$UTILS_PATH/mediaarea.xsd"
-fi
-
 enabled()
 {
     if test -z "${RUN_TESTS+x}" ; then
