from arch's package: https://github.com/archlinux/svntogit-community/blob/packages/bibutils/trunk/generate-man-pages.patch

--- a/bibutils.dbk
+++ b/bibutils.dbk
@@ -45,12 +45,12 @@
     <refmiscinfo class='version'> &version; </refmiscinfo>
     <refmiscinfo class='manual'> User Commands </refmiscinfo>
   </refmeta>
-  <!-- Uncomment this division to have .so manpages generated for all
-  commands
     <refnamediv>
     <refname>bibutils</refname>
     <refname>bib2xml</refname>
+    <refname>biblatex2xml</refname>
     <refname>copac2xml</refname>
+    <refname>ebi2xml</refname>
     <refname>end2xml</refname>
     <refname>endx2xml</refname>
     <refname>isi2xml</refname>
@@ -58,15 +58,17 @@
     <refname>modsclean</refname>
     <refname>nbib2xml</refname>
     <refname>ris2xml</refname>
+    <refname>wordbib2xml</refname>
     <refname>xml2ads</refname>
     <refname>xml2bib</refname>
+    <refname>xml2biblatex</refname>
     <refname>xml2end</refname>
     <refname>xml2isi</refname>
+    <refname>xml2nbib</refname>
     <refname>xml2ris</refname>
     <refname>xml2wordbib</refname>
       <refpurpose>bibliography conversion utilities</refpurpose>
   </refnamediv>
-  -->
     <refnamediv>
     <refname>bibutils</refname>
       <refpurpose>bibliography conversion utilities</refpurpose>
