--- Makefile.orig	2009-03-14 01:28:57.237390838 +0100
+++ Makefile	2009-03-14 01:30:27.492620161 +0100
@@ -18,6 +18,9 @@ FACILITY=LOG_DAEMON
 # target system will never user NIS for hostname lookup, you can define
 # USE_DNS to add hostname tests in hosts.allow/deny.
 
+USE_DNS=yes
+RPCUSER=nobody
+
 ifeq ($(NO_TCP_WRAPPER),)
 CPPFLAGS += -DHOSTS_ACCESS
 WRAP_LIB  = -lwrap
@@ -136,12 +139,12 @@ portmap.man : portmap.8
 	sed $(MAN_SED) < portmap.8 > portmap.man
 
 install: all
-	install -o root -g root -m 0755 -s portmap ${BASEDIR}/sbin
-	install -o root -g root -m 0755 -s pmap_dump ${BASEDIR}/sbin
-	install -o root -g root -m 0755 -s pmap_set ${BASEDIR}/sbin
-	install -o root -g root -m 0644 portmap.man ${BASEDIR}/usr/share/man/man8/portmap.8
-	install -o root -g root -m 0644 pmap_dump.8 ${BASEDIR}/usr/share/man/man8
-	install -o root -g root -m 0644 pmap_set.8 ${BASEDIR}/usr/share/man/man8
+	install -D -m 0755 -s portmap ${BASEDIR}/sbin/portmap
+	install -D -m 0755 -s pmap_dump ${BASEDIR}/sbin/pmap_dump
+	install -D -m 0755 -s pmap_set ${BASEDIR}/sbin/pmap_set
+	install -D -m 0644 portmap.man ${BASEDIR}/usr/share/man/man8/portmap.8
+	install -D -m 0644 pmap_dump.8 ${BASEDIR}/usr/share/man/man8/pmap_dump.8
+	install -D -m 0644 pmap_set.8 ${BASEDIR}/usr/share/man/man8/pmap_set.8
 
 clean:
 	rm -f *.o portmap pmap_dump pmap_set from_local \
