--- a/Makefile.am	2025-11-27 06:25:28
+++ b/Makefile.am	2025-12-04 00:33:35
@@ -56,18 +56,14 @@ dnsdist%generated-body.hh dnsdist%generated.hh: dnsdis
 	@echo ')FFIContent"' >> $@
 SRC_JS_FILES := $(wildcard src_js/*.js)
 MIN_JS_FILES := $(patsubst src_js/%.js,html/js/%.min.js,$(SRC_JS_FILES))
 
 dnsdist%generated-body.hh dnsdist%generated.hh: dnsdist-rules-generator.py dnsdist-actions-definitions.yml dnsdist-response-actions-definitions.yml dnsdist-selectors-definitions.yml
-	@if test "$(PYTHON)" = ":"; then echo "Actions or selectors definitions have changed, python is needed to regenerate the related files but python was not found. Please install python and re-run configure"; exit 1; fi
-	@if ! $(PYTHON) --version | grep -q "Python 3"; then echo $(PYTHON) should be at least version 3. Please install python 3 and re-run configure; exit 1; fi
-	$(PYTHON) ${srcdir}/dnsdist-rules-generator.py ${srcdir} .
+	#
 
 docs/reference/yaml-%.rst: dnsdist-settings-documentation-generator.py dnsdist-settings-definitions.yml dnsdist-actions-definitions.yml dnsdist-response-actions-definitions.yml dnsdist-selectors-definitions.yml
-	@if test "$(PYTHON)" = ":"; then echo "Settings definitions have changed, python is needed to regenerate the related files but python was not found. Please install python and re-run configure"; exit 1; fi
-	@if ! $(PYTHON) --version | grep -q "Python 3"; then echo $(PYTHON) should be at least version 3. Please install python 3 and re-run configure; exit 1; fi
-	$(PYTHON) ${srcdir}/dnsdist-settings-documentation-generator.py ${srcdir}
+	#
 
 html/js/%.min.js: src_js/%.js
 	uglifyjs $< > $@
 
 min_js: $(MIN_JS_FILES)
@@ -659,11 +655,11 @@ docs/reference/yaml-settings.rst: dnsdist-settings-doc
 $(MANPAGES): %: docs/manpages/%.rst .venv
 	$(AM_V_GEN).venv/bin/python -msphinx -b man docs . $<
 endif # if !HAVE_MANPAGES
 
 docs/reference/yaml-settings.rst: dnsdist-settings-documentation-generator.py dnsdist-settings-definitions.yml dnsdist-actions-definitions.yml dnsdist-response-actions-definitions.yml dnsdist-selectors-definitions.yml
-	$(PYTHON) dnsdist-settings-documentation-generator.py .
+	#
 
 .venv: docs/requirements.txt
 	$(PYTHON) -m venv .venv
 	.venv/bin/pip install -U pip setuptools
 	.venv/bin/pip install -r $<
--- a/Makefile.in	2025-12-01 05:21:39
+++ b/Makefile.in	2025-12-04 00:34:12
@@ -3030,18 +3030,14 @@ dnsdist%generated-body.hh dnsdist%generated.hh: dnsdis
 	$(AM_V_GEN)echo 'R"FFIContent(' > $@
 	@cat $^ >> $@
 	@echo ')FFIContent"' >> $@
 
 dnsdist%generated-body.hh dnsdist%generated.hh: dnsdist-rules-generator.py dnsdist-actions-definitions.yml dnsdist-response-actions-definitions.yml dnsdist-selectors-definitions.yml
-	@if test "$(PYTHON)" = ":"; then echo "Actions or selectors definitions have changed, python is needed to regenerate the related files but python was not found. Please install python and re-run configure"; exit 1; fi
-	@if ! $(PYTHON) --version | grep -q "Python 3"; then echo $(PYTHON) should be at least version 3. Please install python 3 and re-run configure; exit 1; fi
-	$(PYTHON) ${srcdir}/dnsdist-rules-generator.py ${srcdir} .
+	#
 
 docs/reference/yaml-%.rst: dnsdist-settings-documentation-generator.py dnsdist-settings-definitions.yml dnsdist-actions-definitions.yml dnsdist-response-actions-definitions.yml dnsdist-selectors-definitions.yml
-	@if test "$(PYTHON)" = ":"; then echo "Settings definitions have changed, python is needed to regenerate the related files but python was not found. Please install python and re-run configure"; exit 1; fi
-	@if ! $(PYTHON) --version | grep -q "Python 3"; then echo $(PYTHON) should be at least version 3. Please install python 3 and re-run configure; exit 1; fi
-	$(PYTHON) ${srcdir}/dnsdist-settings-documentation-generator.py ${srcdir}
+	#
 
 html/js/%.min.js: src_js/%.js
 	uglifyjs $< > $@
 
 min_js: $(MIN_JS_FILES)
@@ -3060,11 +3056,11 @@ dnsdist-lua-ffi.$(OBJEXT): dnsdist-lua-ffi-interface.i
 
 @HAVE_MANPAGES_FALSE@@HAVE_VENV_TRUE@$(MANPAGES): %: docs/manpages/%.rst .venv
 @HAVE_MANPAGES_FALSE@@HAVE_VENV_TRUE@	$(AM_V_GEN).venv/bin/python -msphinx -b man docs . $<
 
 @HAVE_VENV_TRUE@docs/reference/yaml-settings.rst: dnsdist-settings-documentation-generator.py dnsdist-settings-definitions.yml dnsdist-actions-definitions.yml dnsdist-response-actions-definitions.yml dnsdist-selectors-definitions.yml
-@HAVE_VENV_TRUE@	$(PYTHON) dnsdist-settings-documentation-generator.py .
+@HAVE_VENV_TRUE@	#
 
 @HAVE_VENV_TRUE@.venv: docs/requirements.txt
 @HAVE_VENV_TRUE@	$(PYTHON) -m venv .venv
 @HAVE_VENV_TRUE@	.venv/bin/pip install -U pip setuptools
 @HAVE_VENV_TRUE@	.venv/bin/pip install -r $<
--- a/dnsdist-rust-lib/Makefile.am	2025-11-27 06:25:28
+++ b/dnsdist-rust-lib/Makefile.am	2025-12-04 00:34:41
@@ -9,9 +9,6 @@ rust/src/lib.rs ../dnsdist-configuration-yaml-items-ge
 	../dnsdist-configuration-yaml-items-generated.cc
 
 all: rust/src/lib.rs ../dnsdist-configuration-yaml-items-generated.cc
 
 rust/src/lib.rs ../dnsdist-configuration-yaml-items-generated.cc: dnsdist-settings-generator.py ../dnsdist-settings-definitions.yml rust-pre-in.rs rust-middle-in.rs rust-post-in.rs dnsdist-configuration-yaml-items-pre-in.cc ../dnsdist-actions-definitions.yml ../dnsdist-response-actions-definitions.yml ../dnsdist-selectors-definitions.yml
-	@if test "$(PYTHON)" = ":"; then echo "Settings definitions have changed, python is needed to regenerate the related settings files but python was not found. Please install python and re-run configure"; exit 1; fi
-	@if ! $(PYTHON) --version | grep -q "Python 3"; then echo $(PYTHON) should be at least version 3. Please install python 3 and re-run configure; exit 1; fi
-	$(MAKE) -C rust clean
-	(cd ${srcdir} && $(PYTHON) dnsdist-settings-generator.py ../ ./ ../)
+	#
--- a/dnsdist-rust-lib/Makefile.in	2025-12-01 05:21:40
+++ b/dnsdist-rust-lib/Makefile.in	2025-12-04 00:34:49
@@ -593,14 +593,11 @@ rust/src/lib.rs ../dnsdist-configuration-yaml-items-ge
 
 
 all: rust/src/lib.rs ../dnsdist-configuration-yaml-items-generated.cc
 
 rust/src/lib.rs ../dnsdist-configuration-yaml-items-generated.cc: dnsdist-settings-generator.py ../dnsdist-settings-definitions.yml rust-pre-in.rs rust-middle-in.rs rust-post-in.rs dnsdist-configuration-yaml-items-pre-in.cc ../dnsdist-actions-definitions.yml ../dnsdist-response-actions-definitions.yml ../dnsdist-selectors-definitions.yml
-	@if test "$(PYTHON)" = ":"; then echo "Settings definitions have changed, python is needed to regenerate the related settings files but python was not found. Please install python and re-run configure"; exit 1; fi
-	@if ! $(PYTHON) --version | grep -q "Python 3"; then echo $(PYTHON) should be at least version 3. Please install python 3 and re-run configure; exit 1; fi
-	$(MAKE) -C rust clean
-	(cd ${srcdir} && $(PYTHON) dnsdist-settings-generator.py ../ ./ ../)
+	#
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
 
