--- unix/Makefile.orig	2009-04-19 00:15:07.062942983 +0200
+++ unix/Makefile	2009-04-19 00:18:01.606837606 +0200
@@ -34,10 +34,11 @@ BINFLAGS = 755
 MANFLAGS = 644
 
 # target directories - where to install executables and man pages to
-prefix = /usr/local
+DESTDIR =
+prefix = /usr
 BINDIR = $(prefix)/bin
 manext=1
-MANDIR = $(prefix)/man/man$(manext)
+MANDIR = $(prefix)/share/man/man$(manext)
 ZIPMANUAL = MANUAL
 PKGDIR = IZzip
 VERSION = Version 2.32
@@ -127,12 +128,10 @@ $(ZIPMANUAL): man/zip.1
 
 # install
 install:        $(ZIPS)
-	-$(INSTALL_D) $(BINDIR)
-	$(INSTALL_PROGRAM) $(ZIPS) $(BINDIR)
-	-cd $(BINDIR); $(CHMOD) $(BINFLAGS) $(ZIPS)
-	-$(INSTALL_D) $(MANDIR)
-	$(INSTALL) man/zip.1 $(MANDIR)/zip.$(manext)
-	$(CHMOD) $(MANFLAGS) $(MANDIR)/zip.$(manext)
+	install -d $(DESTDIR)$(BINDIR)
+	install -m 755 $(ZIPS) $(DESTDIR)$(BINDIR)
+	install -d $(DESTDIR)$(MANDIR)
+	install -m 644 man/zip.1 $(DESTDIR)$(MANDIR)/zip.$(manext)
 
 uninstall:
 	-cd $(BINDIR); rm -f $(ZIPS)
