--- Makefile	2017-09-19 14:31:03.514658296 +0200
+++ Makefile	2017-09-19 14:32:36.773662266 +0200
@@ -78,8 +78,13 @@
 .PHONY: compile
 compile:
 # compile FreeDoko
-	$(MAKE) Version
-	$(MAKE) -C src/ FreeDoko
+# cannot build with network due to lacking gnet package
+	$(MAKE) \
+	  "USE_UI_TEXT=false" \
+	  "USE_UI_GTKMM=true" \
+	  "USE_NETWORK=false" \
+	  "CPPFLAGS+= -DRELEASE -DVERSION_DESCRIPTION='\"$(OPERATING_SYSTEM_NAME) binary\"' -DPUBLIC_DATA_DIRECTORY_VALUE='\"$(DATA_DIRECTORY)\"' -DMANUAL_DIRECTORY_VALUE='\"$(HELP_DIRECTORY)/manual\"'" \
+	  -C src/ FreeDoko
 	@echo "---"
 	@echo "* Program 'FreeDoko' created"
 
@@ -289,22 +294,11 @@
 ifeq ($(OPERATING_SYSTEM), Linux)
 .PHONY: install
 install :
-	$(MAKE) \
-	  --always-make \
-	  "USE_UI_TEXT=false" \
-	  "USE_UI_GTKMM=true" \
-	  "USE_NETWORK=true" \
-	  "CXXFLAGS=-Wall -Werror -Wno-parentheses -pipe -Os -std=c++14" \
-	  "CPPFLAGS=-DRELEASE -DVERSION_DESCRIPTION='\"$(OPERATING_SYSTEM_NAME) binary\"' -DPUBLIC_DATA_DIRECTORY_VALUE='\"$(DATA_DIRECTORY)\"' -DMANUAL_DIRECTORY_VALUE='\"$(HELP_DIRECTORY)/manual\"'" \
-	  -C src \
-	  $(PROGRAM)
-	$(MAKE) -C src strip
 #	Copying the program
 	echo "Copying the program"
 	mkdir -p $(DESTDIR)$(BIN_DIRECTORY)
 	cp src/FreeDoko $(DESTDIR)$(BIN_DIRECTORY)
-	-chgrp games $(DESTDIR)$(BIN_DIRECTORY)/FreeDoko
-	chmod 750 $(DESTDIR)$(BIN_DIRECTORY)/FreeDoko
+	chmod 755 $(DESTDIR)$(BIN_DIRECTORY)/FreeDoko
 	cd $(DESTDIR)$(BIN_DIRECTORY) && ln -sf FreeDoko freedoko
 #	Copying the data files
 	echo "Copying the data files"
@@ -315,21 +309,20 @@
 	echo "Copying the documentation"
 	mkdir -p $(DESTDIR)$(HELP_DIRECTORY)
 	cp AUTHORS COPYING ChangeLog $(DESTDIR)$(HELP_DIRECTORY)/
-	$(MAKE) -C doc install
+# cannot build docs due to lacking latex packages
+#	$(MAKE) -C doc install
 #	update the manual page
 	echo "update the manual page"
 	a2x -f manpage doc/freedoko.asciidoc
 	mkdir -p $(DESTDIR)$(MAN_DIRECTORY)
 	cp doc/freedoko.6 $(DESTDIR)$(MAN_DIRECTORY)/
-	-chgrp games $(DESTDIR)$(MAN_DIRECTORY)/freedoko.6
-	chmod 640 $(DESTDIR)$(MAN_DIRECTORY)/freedoko.6
+	chmod 644 $(DESTDIR)$(MAN_DIRECTORY)/freedoko.6
 	cd $(DESTDIR)$(MAN_DIRECTORY)/ && ln -sf freedoko.6 FreeDoko.6
 #	copy the icon
 	echo "copy the icon"
 	mkdir -p $(DESTDIR)$(ICON_DIRECTORY)
 	cp src/FreeDoko.png $(DESTDIR)$(ICON_DIRECTORY)/
-	-chgrp games $(DESTDIR)$(ICON_DIRECTORY)/FreeDoko.png
-	chmod 640 $(DESTDIR)$(ICON_DIRECTORY)/FreeDoko.png
+	chmod 644 $(DESTDIR)$(ICON_DIRECTORY)/FreeDoko.png
 
 .PHONY: uninstall
 uninstall :
--- Makefile.install.directories	2017-09-19 14:31:03.514658296 +0200
+++ Makefile.install.directories	2017-09-19 14:32:36.774662266 +0200
@@ -4,13 +4,13 @@
 # the different directories for the files
 
 # installing in /usr/local
-MAIN_DIRECTORY         = /usr/local
-BIN_DIRECTORY          = $(MAIN_DIRECTORY)/games
-DATA_DIRECTORY         = $(MAIN_DIRECTORY)/share/games/FreeDoko
-HELP_DIRECTORY         = $(MAIN_DIRECTORY)/share/doc/FreeDoko
-MAN_DIRECTORY          = $(MAIN_DIRECTORY)/share/man/man6
-ICON_DIRECTORY         = $(MAIN_DIRECTORY)/share/pixmaps
-DESKTOP_FILE_DIRECTORY = $(MAIN_DIRECTORY)/share/applications
+#MAIN_DIRECTORY         = /usr/local
+#BIN_DIRECTORY          = $(MAIN_DIRECTORY)/games
+#DATA_DIRECTORY         = $(MAIN_DIRECTORY)/share/games/FreeDoko
+#HELP_DIRECTORY         = $(MAIN_DIRECTORY)/share/doc/FreeDoko
+#MAN_DIRECTORY          = $(MAIN_DIRECTORY)/share/man/man6
+#ICON_DIRECTORY         = $(MAIN_DIRECTORY)/share/pixmaps
+#DESKTOP_FILE_DIRECTORY = $(MAIN_DIRECTORY)/share/applications
 
 # debian
 #MAIN_DIRECTORY         = /usr
@@ -57,6 +57,15 @@
 #ICON_DIRECTORY         = $(MAIN_DIRECTORY)/share/pixmaps
 #DESKTOP_FILE_DIRECTORY = $(MAIN_DIRECTORY)/share/applications
 
+# Void Linux
+MAIN_DIRECTORY         = /usr
+BIN_DIRECTORY          = $(MAIN_DIRECTORY)/bin
+DATA_DIRECTORY         = $(MAIN_DIRECTORY)/share/FreeDoko
+HELP_DIRECTORY         = $(MAIN_DIRECTORY)/share/doc/FreeDoko
+MAN_DIRECTORY          = $(MAIN_DIRECTORY)/share/man/man6
+ICON_DIRECTORY         = $(MAIN_DIRECTORY)/share/pixmaps
+DESKTOP_FILE_DIRECTORY = $(MAIN_DIRECTORY)/share/applications
+
 
 # release
 # where the temporary release files are saved
--- bin/FreeDoko.desktop	2017-09-19 14:31:03.515658296 +0200
+++ bin/FreeDoko.desktop	2017-09-19 14:32:36.775662266 +0200
@@ -6,7 +6,7 @@
 Comment=game of Doppelkopf
 Comment[de]=Doppelkopf-Spiel
 Comment[fr]=un jeu de Doppelkopf
-Exec=/usr/games/freedoko
+Exec=/usr/bin/freedoko
 Path=
 Icon=FreeDoko
 Categories=Application;Game;CardGame;
--- data/Makefile	2017-09-19 14:31:03.618658300 +0200
+++ data/Makefile	2017-09-19 14:32:36.775662266 +0200
@@ -40,8 +40,7 @@
 	for d in cardsets iconsets backgrounds sounds; do \
 	  $(MAKE) -C $$d install; \
 	done
-	-chgrp games $(DESTDIR)$(DATA_DIRECTORY) && chgrp -R games $(DESTDIR)$(DATA_DIRECTORY)
-	chmod -R o-rwx,g+rX,g-w $(DESTDIR)$(DATA_DIRECTORY)
+	chmod -R o-rwx,g+rX,g-w,+rX $(DESTDIR)$(DATA_DIRECTORY)
 endif
 
 ifeq ($(SHELLTYPE), sh)
--- doc/Makefile	2017-09-19 14:31:03.684658303 +0200
+++ doc/Makefile	2017-09-19 14:32:36.776662266 +0200
@@ -36,8 +36,7 @@
 	fi
 	cp -a Bugs GPL Roadmap hpux.required_libs-ia64.txt $(DESTDIR)$(HELP_DIRECTORY)/
 	for d in de en; do $(MAKE) -C $$d install; done
-	-chgrp games $(DESTDIR)$(HELP_DIRECTORY) && chgrp -R games $(DESTDIR)$(HELP_DIRECTORY)
-	chmod -R o-rwx,g+rX,g-w $(DESTDIR)$(HELP_DIRECTORY)
+	chmod -R o-rwx,g+rX,g-w,+rX $(DESTDIR)$(HELP_DIRECTORY)
 	cd $(DESTDIR)$(DATA_DIRECTORY) && ln -sf $(HELP_DIRECTORY)/ doc
 endif
 
--- src/Makefile	2017-09-19 14:31:03.383658290 +0200
+++ src/Makefile	2017-09-19 14:32:36.777662266 +0200
@@ -252,6 +252,7 @@
 	  $(INCLUDE_ALL) \
 	  $(LIBS_ALL) \
 	  $(CXXFLAGS) \
+	  $(LDFLAGS) \
 	  -o $@
 	@echo "$(PROGRAM) compiled successfully"
 
