--- a/Makefile	2018-01-04 07:11:48.000000000 +0100
+++ b/Makefile	2018-01-05 09:47:57.905760414 +0100
@@ -316,8 +316,7 @@
 	echo "Kopiere das Programm"
 	mkdir -p $(DESTDIR)$(BIN_DIRECTORY)
 	cp src/FreeDoko $(DESTDIR)$(BIN_DIRECTORY)
-	-chgrp games $(DESTDIR)$(BIN_DIRECTORY)/FreeDoko
-	chmod 750 $(DESTDIR)$(BIN_DIRECTORY)/FreeDoko
+	chmod 755 $(DESTDIR)$(BIN_DIRECTORY)/FreeDoko
 	cd $(DESTDIR)$(BIN_DIRECTORY) && ln -sf FreeDoko freedoko
 #	Copying the data files
 	echo "Kopiere die Daten"
@@ -330,22 +329,21 @@
 	echo "Kopiere die Dokumentation"
 	mkdir -p $(DESTDIR)$(HELP_DIRECTORY)
 	cp AUTHORS COPYING ChangeLog $(DESTDIR)$(HELP_DIRECTORY)/
-	$(MAKE) -C doc install
+# cannot build docs due to lacking latex packages
+#	$(MAKE) -C doc install
 	$(MAKE) -C manual install
 #	update the manual page
 	echo "Aktualisiere die man-Page"
 	a2x -f manpage doc/freedoko.adoc
 	mkdir -p $(DESTDIR)$(MAN_DIRECTORY)
 	cp doc/freedoko.6 $(DESTDIR)$(MAN_DIRECTORY)/
-	-chgrp games $(DESTDIR)$(MAN_DIRECTORY)/freedoko.6
-	chmod 640 $(DESTDIR)$(MAN_DIRECTORY)/freedoko.6
+	chmod 644 $(DESTDIR)$(MAN_DIRECTORY)/freedoko.6
 	cd $(DESTDIR)$(MAN_DIRECTORY)/ && ln -sf freedoko.6 FreeDoko.6
 #	copy the icon
 	echo "Kopiere das Icon"
 	mkdir -p $(DESTDIR)$(ICON_DIRECTORY)
 	cp src/FreeDoko.png $(DESTDIR)$(ICON_DIRECTORY)/
-	-chgrp games $(DESTDIR)$(ICON_DIRECTORY)/FreeDoko.png
-	chmod 640 $(DESTDIR)$(ICON_DIRECTORY)/FreeDoko.png
+	chmod 644 $(DESTDIR)$(ICON_DIRECTORY)/FreeDoko.png
 #	copy the desktop file
 	echo "Kopiere den Desktop-Starter"
 	mkdir -p "$(DESTDIR)$(DESKTOP_FILE_DIRECTORY)"
@@ -362,15 +360,15 @@
 	  || echo "Programm a2x nicht gefunden."
 	@(which flip || which unix2dos) >/dev/null \
 	  || echo "Weder Programm flip noch unix2dos gefunden."
-	@which pdflatex >/dev/null \
-	  || echo "Programm pdflatex nicht gefunden."
+#	@which pdflatex >/dev/null \
+#	  || echo "Programm pdflatex nicht gefunden."
 	@which rsync >/dev/null \
 	  || echo "Programm rsync nicht gefunden."
 	@test -f src/FreeDoko 
 	@which asciidoc >/dev/null
 	@which a2x >/dev/null
 	@(which flip || which unix2dos) >/dev/null
-	@which pdflatex >/dev/null
+#	@which pdflatex >/dev/null
 	@which rsync >/dev/null
 
 .PHONY: uninstall

--- a/Makefile.install.directories	2018-01-04 07:11:48.000000000 +0100
+++ b/Makefile.install.directories	2018-01-05 09:49:04.236760414 +0100
@@ -4,14 +4,14 @@
 # the different directories for the files
 
 # installing in /usr/local
-MAIN_DIRECTORY         = /usr/local
-BIN_DIRECTORY          = $(MAIN_DIRECTORY)/games
-DATA_DIRECTORY         = $(MAIN_DIRECTORY)/share/games/FreeDoko
-HELP_DIRECTORY         = $(MAIN_DIRECTORY)/share/doc/FreeDoko
-MAN_DIRECTORY          = $(MAIN_DIRECTORY)/share/man/man6
-ICON_DIRECTORY         = $(MAIN_DIRECTORY)/share/pixmaps
-LOCALE_DIRECTORY       = $(MAIN_DIRECTORY)/share/locale
-DESKTOP_FILE_DIRECTORY = $(MAIN_DIRECTORY)/share/applications
+#MAIN_DIRECTORY         = /usr/local
+#BIN_DIRECTORY          = $(MAIN_DIRECTORY)/games
+#DATA_DIRECTORY         = $(MAIN_DIRECTORY)/share/games/FreeDoko
+#HELP_DIRECTORY         = $(MAIN_DIRECTORY)/share/doc/FreeDoko
+#MAN_DIRECTORY          = $(MAIN_DIRECTORY)/share/man/man6
+#ICON_DIRECTORY         = $(MAIN_DIRECTORY)/share/pixmaps
+#LOCALE_DIRECTORY       = $(MAIN_DIRECTORY)/share/locale
+#DESKTOP_FILE_DIRECTORY = $(MAIN_DIRECTORY)/share/applications
 
 # debian
 #MAIN_DIRECTORY         = /usr
@@ -63,6 +63,16 @@
 #LOCALE_DIRECTORY       = $(MAIN_DIRECTORY)/share/locale
 #DESKTOP_FILE_DIRECTORY = $(MAIN_DIRECTORY)/share/applications
 
+# Void Linux
+MAIN_DIRECTORY         = /usr
+BIN_DIRECTORY          = $(MAIN_DIRECTORY)/bin
+DATA_DIRECTORY         = $(MAIN_DIRECTORY)/share/FreeDoko
+HELP_DIRECTORY         = $(MAIN_DIRECTORY)/share/doc/FreeDoko
+MAN_DIRECTORY          = $(MAIN_DIRECTORY)/share/man/man6
+ICON_DIRECTORY         = $(MAIN_DIRECTORY)/share/pixmaps
+LOCALE_DIRECTORY       = $(MAIN_DIRECTORY)/share/locale
+DESKTOP_FILE_DIRECTORY = $(MAIN_DIRECTORY)/share/applications
+
 
 # release
 # where the temporary release files are saved

--- a/data/Makefile	2018-01-04 07:11:48.000000000 +0100
+++ b/data/Makefile	2018-01-05 09:49:28.201760414 +0100
@@ -32,8 +32,7 @@
 	for d in cardsets iconsets backgrounds sounds; do \
 	  $(MAKE) -C $$d install; \
 	done
-	-chgrp games $(DESTDIR)$(DATA_DIRECTORY) && chgrp -R games $(DESTDIR)$(DATA_DIRECTORY)
-	chmod -R o-rwx,g+rX,g-w $(DESTDIR)$(DATA_DIRECTORY)
+	chmod -R o-rwx,g+rX,g-w,+rX $(DESTDIR)$(DATA_DIRECTORY)
 endif
 
 ifeq ($(SHELLTYPE), sh)

--- a/doc/Makefile	2018-01-04 07:11:48.000000000 +0100
+++ b/doc/Makefile	2018-01-05 09:49:46.401760414 +0100
@@ -34,8 +34,7 @@
 	fi
 	cp -a Bugs GPL Roadmap hpux.required_libs-ia64.txt $(DESTDIR)$(HELP_DIRECTORY)/
 	for d in de en; do $(MAKE) -C $$d install; done
-	-chgrp games $(DESTDIR)$(HELP_DIRECTORY) && chgrp -R games $(DESTDIR)$(HELP_DIRECTORY)
-	chmod -R o-rwx,g+rX,g-w $(DESTDIR)$(HELP_DIRECTORY)
+	chmod -R o-rwx,g+rX,g-w,+rX $(DESTDIR)$(HELP_DIRECTORY)
 	cd $(DESTDIR)$(DATA_DIRECTORY) && ln -sf $(HELP_DIRECTORY)/ doc
 endif
 

--- a/src/Makefile	2018-01-04 07:11:48.000000000 +0100
+++ b/src/Makefile	2018-01-05 09:50:07.057760414 +0100
@@ -282,7 +282,6 @@
 		-DMANUAL_DIRECTORY_VALUE='\"$(HELP_DIRECTORY)/manual\"' \
 		-DLOCALE_DIRECTORY='\"$(LOCALE_DIRECTORY)\"'" \
 	  $(PROGRAM)
-	$(MAKE) strip
 
 # strip the program from all symbols
 .PHONY: strip
