diff -Nru indicator-appmenu-12.10.3.orig/configure.ac indicator-appmenu-12.10.3/configure.ac
--- indicator-appmenu-12.10.3.orig/configure.ac	2013-01-07 06:59:33.344260743 -0500
+++ indicator-appmenu-12.10.3/configure.ac	2013-01-07 06:59:44.277538125 -0500
@@ -8,7 +8,7 @@
 
 AC_PREREQ(2.53)
 
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADER(config.h)
 AC_CONFIG_SRCDIR([configure.ac])
 AC_CONFIG_MACRO_DIR([m4])
 
