Remove a test which fails when run as root.

diff --git a/lib/tests/test-mu-store.cc b/lib/tests/test-mu-store.cc
index 872c56e3..c4e7eeb2 100644
--- a/lib/tests/test-mu-store.cc
+++ b/lib/tests/test-mu-store.cc
@@ -470,13 +470,6 @@ test_store_fail()
 		const auto store = Store::make("/root/non-existent-path/12345");
 		g_assert_false(!!store);
 	}
-
-	{
-		const auto store = Store::make_new("/../../root/non-existent-path/12345",
-						   "/../../root/non-existent-path/54321",
-						   {}, {});
-		g_assert_false(!!store);
-	}
 }
 
 int
