From 24f7cec6b4b84cc75b7de18b16fd73d85ee448f9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Martin=20Kj=C3=A6r=20J=C3=B8rgensen?= <me@lagy.org>
Date: Mon, 15 Sep 2025 11:37:57 +0200
Subject: [PATCH] remove broken test for void

---
 lib/tests/test-mu-store.cc | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/lib/tests/test-mu-store.cc b/lib/tests/test-mu-store.cc
index b1573e7a..9d77d046 100644
--- a/lib/tests/test-mu-store.cc
+++ b/lib/tests/test-mu-store.cc
@@ -633,12 +633,6 @@ test_store_fail()
 		const auto store = Store::make("/root/non-existent-path/12345");
 		g_assert_false(!!store);
 	}
-
-	{
-		const auto store = Store::make_new("/../../root/non-existent-path/12345",
-						   "/../../root/non-existent-path/54321");
-		g_assert_false(!!store);
-	}
 }
 
 
-- 
2.51.0

