Remove a test which fails when run as root.

diff --git a/lib/tests/test-mu-store.cc b/lib/tests/test-mu-store.cc
index c9b8013..01dbe6b 100644
--- a/lib/tests/test-mu-store.cc
+++ b/lib/tests/test-mu-store.cc
@@ -581,11 +581,6 @@ test_store_fail()
 		g_assert_false(!!store);
 	}
 
-	{
-		const auto store = Store::make_new("/../../root/non-existent-path/12345",
-						   "/../../root/non-existent-path/54321");
-		g_assert_false(!!store);
-	}
 }
 
 
