Subject: [PATCH] remove broken test for void

---
 lib/tests/test-mu-store.cc | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/lib/tests/test-mu-store.cc b/lib/tests/test-mu-store.cc
index b1573e7a..9d77d046 100644
--- a/lib/tests/test-mu-store.cc
+++ b/lib/tests/test-mu-store.cc
@@ -633,12 +633,6 @@ test_store_fail()
 		const auto store = Store::make("/root/non-existent-path/12345");
 		g_assert_false(!!store);
 	}
-
-	{
-		const auto store = Store::make_new("/../../root/non-existent-path/12345",
-						   "/../../root/non-existent-path/54321");
-		g_assert_false(!!store);
-	}
 }
 
 
-- 
2.51.0

