--- wendy.c.orig	2015-09-07 15:45:56.336193710 -0400
+++ wendy.c	2015-09-07 15:46:12.468193218 -0400
@@ -115,7 +115,7 @@
 }
 
 	int
-execvpe(const char *program, char **argv, char **envp)
+_execvpe(const char *program, char **argv, char **envp)
 {
 	char **saved = environ;
 	int rc;
@@ -258,7 +258,7 @@
 				 * Also, double-forking.
 				 */
 				if (!fork())
-				if (!fork()) execvpe(cmd[0], cmd, environ);
+				if (!fork()) _execvpe(cmd[0], cmd, environ);
 				else exit(0);
 				else wait(NULL);
 			}
