--- wendy.c.orig	2016-02-01 07:45:28.000000000 -0500
+++ wendy.c	2016-02-01 10:53:25.313600196 -0500
@@ -102,7 +102,7 @@
 }
 
 int
-execvpe(const char *program, char **argv, char **envp)
+_execvpe(const char *program, char **argv, char **envp)
 {
 	char **saved = environ;
 	int rc;
@@ -262,7 +262,7 @@
 				setenv(ENV_MASK, strmask, 1);
 				setenv(ENV_PATH, EVENT_PATH(ev), 1);
 				if (!fork())
-				if (!fork()) execvpe(cmd[0], cmd, environ);
+				if (!fork()) _execvpe(cmd[0], cmd, environ);
 				else exit(0);
 				else wait(NULL);
 			}
