--- old/make/linux/makefiles/gcc.make	2014-04-23 22:07:59.838205838 +0100
+++ new/make/linux/makefiles/gcc.make	2014-04-23 22:07:59.754204533 +0100
@@ -266,6 +266,9 @@
   ifeq ($(shell expr $(CC_VER_MAJOR) = 4 \& $(CC_VER_MINOR) = 3), 1)
     OPT_CFLAGS/mulnode.o += $(OPT_CFLAGS/NOOPT)
   endif
+  ifeq ($(shell expr $(CC_VER_MAJOR) = 4 \& $(CC_VER_MINOR) = 9), 1)
+    OPT_CFLAGS/macroAssembler_x86.o += -fno-devirtualize
+  endif
 endif
 
 # Flags for generating make dependency flags.
