Use ~0u instead of -1 to shift an unsigned value.

--- a/src/share/vm/code/dependencies.hpp	2016-04-25 18:32:00.000000000 +0200
+++ b/src/share/vm/code/dependencies.hpp	2016-09-13 19:46:22.684595758 +0200
@@ -166,7 +166,7 @@
     LG2_TYPE_LIMIT = 4,  // assert(TYPE_LIMIT <= (1<<LG2_TYPE_LIMIT))
 
     // handy categorizations of dependency types:
-    all_types           = ((1 << TYPE_LIMIT) - 1) & ((-1) << FIRST_TYPE),
+    all_types           = ((1 << TYPE_LIMIT) - 1) & (~0u << FIRST_TYPE),
 
     non_klass_types     = (1 << call_site_target_value),
     klass_types         = all_types & ~non_klass_types,
