diff -ru ../bglibs-2.03.orig/net/dns_ip6.c ./net/dns_ip6.c
--- ../bglibs-2.03.orig/net/dns_ip6.c	2015-02-06 17:57:35.000000000 -0600
+++ ./net/dns_ip6.c	2015-03-30 19:40:44.591110913 -0500
@@ -75,12 +75,9 @@
 MAIN
 {
   doit("1.2.3.4");
-  doit("google.ca");
 }
 #endif
 #ifdef SELFTEST_EXP_FNMATCH
 result=0
 1.2.3.4 1: ::ffff:1.2.3.4
-result=0
-google.ca 1: 2607:f8b0:400a:*
 #endif
diff -ru ../bglibs-2.03.orig/net/dns_mx.c ./net/dns_mx.c
--- ../bglibs-2.03.orig/net/dns_mx.c	2015-02-06 17:57:35.000000000 -0600
+++ ./net/dns_mx.c	2015-03-30 19:40:44.591110913 -0500
@@ -70,18 +70,10 @@
 MAIN
 {
   doit("untroubled.org");
-  doit("google.com");
 }
 #endif
 #ifdef SELFTEST_EXP
 result=0
 untroubled.org => 1
 0 12801: "mx.futurequest.net"
-result=0
-google.com => 5
-0 10: "aspmx.l.google.com"
-1 20: "alt1.aspmx.l.google.com"
-2 30: "alt2.aspmx.l.google.com"
-3 40: "alt3.aspmx.l.google.com"
-4 50: "alt4.aspmx.l.google.com"
 #endif
diff -ru ../bglibs-2.03.orig/net/dns_txt.c ./net/dns_txt.c
--- ../bglibs-2.03.orig/net/dns_txt.c	2015-02-06 17:57:35.000000000 -0600
+++ ./net/dns_txt.c	2015-03-30 20:06:33.828383880 -0500
@@ -76,15 +76,11 @@
 }
 MAIN
 {
-  doit("gmail.com");
   doit("2.0.0.127.sbl-xbl.spamhaus.org");
 }
 #endif
 #ifdef SELFTEST_EXP
 result=0
-1 record
-v=spf1 redirect=_spf.google.com
-result=0
 2 record
 http://www.spamhaus.org/sbl/query/SBL233
 http://www.spamhaus.org/query/bl?ip=127.0.0.2
Only in .: no_google_spf.patch
