Also link to -ldl to prevent undefined references.

--- a/libcxxabi/src/CMakeLists.txt.orig
+++ b/libcxxabi/src/CMakeLists.txt
@@ -75,6 +75,7 @@ else()
   endif()

   add_library_flags_if(LIBCXXABI_HAS_C_LIB c)
+  add_library_flags_if(LIBCXXABI_HAS_C_LIB dl)
 endif()

 if (LIBCXXABI_USE_COMPILER_RT)
--- a/libcxx/CMakeLists.txt
+++ b/libcxx/CMakeLists.txt
@@ -781,6 +781,10 @@
     target_link_libraries(${target} PRIVATE m)
   endif()
 
+  if (LIBCXX_HAS_C_LIB)
+    target_link_libraries(${target} PRIVATE dl)
+  endif()
+
   if (LIBCXX_HAS_RT_LIB)
     target_link_libraries(${target} PRIVATE rt)
   endif()
