--- source/fitz/load-jpx.c.orig
+++ source/fitz/load-jpx.c
@@ -444,14 +444,19 @@
 
 #else /* HAVE_LURATECH */
 
+
+#ifdef __cplusplus
+extern "C"
+{
 #define OPJ_STATIC
 #define OPJ_HAVE_INTTYPES_H
 #if !defined(_WIN32) && !defined(_WIN64)
 #define OPJ_HAVE_STDINT_H
 #endif
 #define USE_JPIP
+#endif
 
-#include <openjpeg.h>
+#include <openjpeg-2.1/openjpeg.h>
 
 struct fz_jpxd_s
 {
@@ -919,6 +924,9 @@
 	*yresp = state.yres;
 }
 
+#ifdef __cplusplus
+}      
+#endif
 #endif /* HAVE_LURATECH */
 
 #else /* FZ_ENABLE_JPX */
