--- Makerules	2014-04-11 15:10:41.000000000 +0000
+++ Makerules	2014-04-15 05:35:34.430028127 +0000
@@ -47,6 +47,8 @@ SYS_OPENJPEG_LIBS = -lopenjpeg
 SYS_JBIG2DEC_LIBS = -ljbig2dec
 SYS_JPEG_LIBS = -ljpeg
 SYS_ZLIB_LIBS = -lz
+SYS_CURL_CFLAGS = $(shell pkg-config --cflags libcurl) -I/usr/include/curl
+SYS_CURL_LIBS = $(shell pkg-config --libs libcurl)
 
 CC = xcrun cc
 AR = xcrun ar
--- Makethird	2014-04-11 15:10:41.000000000 +0000
+++ Makethird	2014-04-15 05:37:06.712810863 +0000
@@ -454,6 +454,10 @@ CURL_CFLAGS := -I$(CURL_DIR)/include
 CURL_LIBS := $(SYS_CURL_DEPS)
 
 HAVE_CURL := yes
+else
+HAVE_CURL := yes
+CURL_CFLAGS := $(SYS_CURL_CFLAGS)
+CURL_LIBS := $(SYS_CURL_LIBS) $(SYS_CURL_DEPS)
 endif
 
 # --- X11 ---
