--- Makerules	2013-08-27 14:35:11.243520354 +0000
+++ Makerules	2013-08-27 15:40:22.583629510 +0000
@@ -67,6 +67,8 @@
 SYS_JBIG2DEC_LIBS = -ljbig2dec
 SYS_JPEG_LIBS = -ljpeg
 SYS_ZLIB_LIBS = -lz
+SYS_CURL_CFLAGS = $(shell pkg-config --cflags libcurl)
+SYS_CURL_LIBS = $(shell pkg-config --libs libcurl)
 
 endif
 
--- Makethird	2013-08-27 15:41:18.043354692 +0000
+++ Makethird	2013-08-27 15:42:41.332916192 +0000
@@ -444,5 +444,6 @@
 CURL_CFLAGS := -I$(CURL_DIR)/include
 CURL_LIBS := $(SYS_CURL_DEPS)
 else
-NOCURL := yes
+CURL_CFLAGS := $(SYS_CURL_CFLAGS)
+CURL_LIBS := $(SYS_CURL_LIBS) $(SYS_CURL_DEPS)
 endif
