--- a/rules.mk	2015-04-25 03:13:58.762185962 +0200
+++ b/rules.mk	2015-04-25 03:40:23.891254518 +0200
@@ -6,12 +6,12 @@
 
 $(rastertoqpdl_TARGET): $(rastertoqpdl_OBJ)
 	$(call printCmd, $(cmd_link))
-	$(Q)g++ -o $@ $^ $(rastertoqpdl_CXXFLAGS) $(rastertoqpdl_LDFLAGS) \
+	$(Q)$(CXX) -o $@ $^ $(rastertoqpdl_CXXFLAGS) $(rastertoqpdl_LDFLAGS) \
 		$(rastertoqpdl_LIBS)
 
 $(pstoqpdl_TARGET): $(pstoqpdl_OBJ)
 	$(call printCmd, $(cmd_link))
-	$(Q)g++ -o $@ $^ $(pstoqpdl_CXXFLAGS) $(pstoqpdl_LDFLAGS) \
+	$(Q)$(CXX) -o $@ $^ $(pstoqpdl_CXXFLAGS) $(pstoqpdl_LDFLAGS) \
 		$(pstoqpdl_LIBS)
 
 .PHONY: install installcms
