From: Luca Niccoli <lultimouomo@gmail.com>
Date: Sun, 12 Jun 2011 15:30:15 +0200
Subject: Fix alignment problem on armel

---
 src/qpdl.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git src/qpdl.cpp src/qpdl.cpp
index 994e1c8..a95f361 100644
--- a/src/qpdl.cpp
+++ b/src/qpdl.cpp
@@ -132,7 +132,7 @@ static bool _renderBand(const Request& request, const Band* band, bool mono)
 {
     unsigned long version, subVersion, size, dataSize, checkSum;
     bool color, headerSent=false;
-    unsigned char header[0x20];
+    unsigned char header[0x20] __attribute__((aligned(4)));
     const BandPlane *plane;

     version = request.printer()->qpdlVersion();
