--- private.c.orig	2018-01-16 19:16:59.014506195 +0100
+++ private.c	2018-01-16 19:21:35.115406026 +0100
@@ -2389,7 +2389,7 @@
     *_global_p_len = (FriBidiStrIndex)(size1/sizeof(FriBidiCharType));
     arg1 = (FriBidiCharType *)(buf1);
     
-    result = fribidi_get_par_direction((fribidi_uint32 const *)arg1,arg2);
+    result = fribidi_get_par_direction((uint32_t const *)arg1,arg2);
     ST(argvi) = SWIG_From_unsigned_SS_long  SWIG_PERL_CALL_ARGS_1((unsigned long)(result)); argvi++ ;
     
     
@@ -2455,7 +2455,7 @@
     Newx(*temp4, ((*_global_p_len) + 1), FriBidiLevel);
     
     _saved[0] = ST(1);
-    result = (FriBidiLevel)fribidi_get_par_embedding_levels((fribidi_uint32 const *)arg1,arg2,arg3,arg4);
+    result = (FriBidiLevel)fribidi_get_par_embedding_levels((uint32_t const *)arg1,arg2,arg3,arg4);
     ST(argvi) = SWIG_From_signed_SS_char  SWIG_PERL_CALL_ARGS_1((signed char)(result)); argvi++ ;
     if (SWIG_IsTmpObj(res3)) {
       if (argvi >= items) EXTEND(sp,1);  ST(argvi) = SWIG_From_unsigned_SS_long  SWIG_PERL_CALL_ARGS_1((*arg3)); argvi++  ;
@@ -2555,7 +2555,7 @@
     Newx(*temp6, ((*_global_p_len) + 1), FriBidiLevel);
     
     _saved[0] = ST(5);
-    result = (FriBidiLevel)fribidi_reorder_line(arg1,(fribidi_uint32 const *)arg2,arg3,arg4,arg5,arg6,arg7,arg8);
+    result = (FriBidiLevel)fribidi_reorder_line(arg1,(uint32_t const *)arg2,arg3,arg4,arg5,arg6,arg7,arg8);
     ST(argvi) = SWIG_From_signed_SS_char  SWIG_PERL_CALL_ARGS_1((signed char)(result)); argvi++ ;
     
     // printf("Pushing into arg6\n");
@@ -2629,7 +2629,7 @@
     
     _saved[0] = ST(1);
     _saved[1] = ST(2);
-    fribidi_join_arabic((fribidi_uint32 const *)arg1,arg2,(signed char const *)arg3,arg4);
+    fribidi_join_arabic((uint32_t const *)arg1,arg2,(signed char const *)arg3,arg4);
     ST(argvi) = sv_newmortal();
     
     
@@ -2894,7 +2894,7 @@
     
     _saved[0] = ST(5);
     _saved[1] = ST(6);
-    result = (FriBidiLevel)reorder_map(arg1,(fribidi_uint32 const *)arg2,arg3,arg4,arg5,arg6,arg7);
+    result = (FriBidiLevel)reorder_map(arg1,(uint32_t const *)arg2,arg3,arg4,arg5,arg6,arg7);
     ST(argvi) = SWIG_From_signed_SS_char  SWIG_PERL_CALL_ARGS_1((signed char)(result)); argvi++ ;
     
     MXPUSHUA(arg6, (*_global_p_len))
