--- a/src/nix-manual/meson.build	2026-07-12 21:19:35.799738812 +0200
+++ a/src/nix-manual/meson.build	2026-07-12 21:20:53.978741525 +0200
@@ -79,7 +79,7 @@
 # Hacky way to figure out if `nix` is an `ExternalProgram` or
 # `Executable`. Only the latter can occur in custom target input lists.
 if nix.full_path().startswith(meson.build_root())
-  nix_input = nix
+  nix_input = []
 else
   nix_input = []
 endif
