https://github.com/ipython/ipython/pull/9417

diff --git IPython/utils/terminal.py IPython/utils/terminal.py
index 9e7be2a..a1f0f73 100644
--- IPython/utils/terminal.py
+++ IPython/utils/terminal.py
@@ -9,22 +9,18 @@ Authors:
 * Alexander Belchenko (e-mail: bialix AT ukr.net)
 """
 
-#-----------------------------------------------------------------------------
-#  Copyright (C) 2008-2011  The IPython Development Team
-#
-#  Distributed under the terms of the BSD License.  The full license is in
-#  the file COPYING, distributed as part of this software.
-#-----------------------------------------------------------------------------
-
-#-----------------------------------------------------------------------------
-# Imports
-#-----------------------------------------------------------------------------
+# Copyright (c) IPython Development Team.
+# Distributed under the terms of the Modified BSD License.
 
 import os
 import struct
 import sys
 import warnings
-import backports.shutil_get_terminal_size
+try:
+    from shutil import get_terminal_size as _get_terminal_size
+except ImportError:
+    # use backport on Python 2
+    from backports.shutil_get_terminal_size import get_terminal_size as _get_terminal_size
 
 from . import py3compat
 
@@ -122,4 +118,4 @@ def freeze_term_title():
 
 
 def get_terminal_size(defaultx=80, defaulty=25):
-    return backports.shutil_get_terminal_size.get_terminal_size((defaultx, defaulty))
+    return _get_terminal_size((defaultx, defaulty))
diff --git setup.py setup.py
index 6479c51..14b5142 100755
--- setup.py
+++ setup.py
@@ -195,7 +195,6 @@ install_requires = [
     'pickleshare',
     'simplegeneric>0.8',
     'traitlets',
-    'backports.shutil_get_terminal_size',
 ]
 
 # Platform-specific dependencies:
@@ -203,6 +202,7 @@ install_requires = [
 # but requires pip >= 6. pip < 6 ignores these.
 
 extras_require.update({
+    ':python_version == "2.7"': ['backports.shutil_get_terminal_size'],
     ':sys_platform != "win32"': ['pexpect'],
     ':sys_platform == "darwin"': ['appnope'],
     ':sys_platform == "darwin" and platform_python_implementation == "CPython"': ['gnureadline'],
