diff --git a/Makefile.am b/Makefile.am
index eba5ec4..46c1728 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -165,7 +165,7 @@ $(gummiboot_solib): $(gummiboot_objects)
 .DELETE_ON_ERROR: $(gummboot_solib)
 
 $(gummiboot): $(gummiboot_solib)
-	$(AM_V_GEN) objcopy -j .text -j .sdata -j .data -j .dynamic \
+	$(AM_V_GEN) $(OBJCOPY) -j .text -j .sdata -j .data -j .dynamic \
 	  -j .dynsym -j .rel -j .rela -j .reloc \
 	  $(FORMAT) $< $@
 
@@ -200,7 +200,7 @@ $(stub_solib): $(stub_objects)
 .DELETE_ON_ERROR: $(gummboot_solib)
 
 $(stub): $(stub_solib)
-	$(AM_V_GEN) objcopy -j .text -j .sdata -j .data -j .dynamic \
+	$(AM_V_GEN) $(OBJCOPY) -j .text -j .sdata -j .data -j .dynamic \
 	  -j .dynsym -j .rel -j .rela -j .reloc \
 	  $(FORMAT) $< $@
 

