--- a/src/efi/stub.c	2015-03-12 00:50:35.000000000 +0100
+++ b/src/efi/stub.c	2020-07-06 12:32:55.985327487 +0200
@@ -83,7 +83,7 @@
         cmdline_len = szs[0];

         /* if we are not in secure boot mode, accept a custom command line and replace the built-in one */
-        if (!secure && loaded_image->LoadOptionsSize > 0) {
+        if ((!secure || cmdline_len == 0) && loaded_image->LoadOptionsSize > 0 && *(CHAR16 *)loaded_image->LoadOptions > 0x1F) {
                 CHAR16 *options;
                 CHAR8 *line;
                 UINTN i;

