--- a/src/zimg/colorspace/matrix3.cpp	2020-03-02 05:50:02.000000000 +0100
+++ b/src/zimg/colorspace/matrix3.cpp	2020-08-23 20:46:14.298311898 +0200
@@ -17,7 +17,7 @@
 {
 	Vector3 ret;
 
-	for (size_t i = 0; i < 3; ++i) {
+	for (std::size_t i = 0; i < 3; ++i) {
 		ret[i] = v1[i] * v2[i];
 	}
 	return ret;
@@ -27,10 +27,10 @@
 {
 	Vector3 ret;
 
-	for (size_t i = 0; i < 3; ++i) {
+	for (std::size_t i = 0; i < 3; ++i) {
 		double accum = 0;
 
-		for (size_t k = 0; k < 3; ++k) {
+		for (std::size_t k = 0; k < 3; ++k) {
 			accum += m[i][k] * v[k];
 		}
 		ret[i] = accum;
@@ -42,11 +42,11 @@
 {
 	Matrix3x3 ret;
 
-	for (size_t i = 0; i < 3; ++i) {
-		for (size_t j = 0; j < 3; ++j) {
+	for (std::size_t i = 0; i < 3; ++i) {
+		for (std::size_t j = 0; j < 3; ++j) {
 			double accum = 0;
 
-			for (size_t k = 0; k < 3; ++k) {
+			for (std::size_t k = 0; k < 3; ++k) {
 				accum += a[i][k] * b[k][j];
 			}
 			ret[i][j] = accum;
@@ -98,8 +98,8 @@
 {
 	Matrix3x3 ret;
 
-	for (size_t i = 0; i < 3; ++i) {
-		for (size_t j = 0; j < 3; ++j) {
+	for (std::size_t i = 0; i < 3; ++i) {
+		for (std::size_t j = 0; j < 3; ++j) {
 			ret[i][j] = m[j][i];
 		}
 	}
