--- configure.ac.orig	2016-05-23 13:02:19.860132419 +0200
+++ configure.ac	2016-05-23 13:02:31.867133038 +0200
@@ -9,11 +9,9 @@
 
 LT_INIT([win32-dll])
 
+AC_CANONICAL_SYSTEM
 AC_PROG_CXX
 
-AC_CANONICAL_HOST
-
-
 AC_ARG_ENABLE([testapp],   AS_HELP_STRING([--enable-testapp],   [Compile the test application. (default=no)]))
 AC_ARG_ENABLE([example],   AS_HELP_STRING([--enable-example],   [Compile example programs. (default=no)]))
 AC_ARG_ENABLE([unit-test], AS_HELP_STRING([--enable-unit-test], [Compile unit tests. May result in slower code. (default=no)]))
@@ -26,7 +24,7 @@
 AS_IF([test "x$enable_unit_test" = "xyes"],
       [
         CFLAGS="$CFLAGS -fno-fast-math -ffp-contract=off" CXXFLAGS="$CXXFLAGS -fno-fast-math -ffp-contract=off"
-        AS_CASE([$host_cpu],
+        AS_CASE([$target_cpu],
                 [i?86],   [CFLAGS="$CFLAGS -mfpmath=sse -msse2" CXXFLAGS="$CXXFLAGS -mfpmath=sse -msse2"],
                 [x86_64], [],
                 [AC_MSG_WARN([Could not determine how to enable IEEE-754 compliance on host. Tests may fail.])])
@@ -44,7 +42,7 @@
 
 X86="no"
 
-AS_CASE([$host_cpu],
+AS_CASE([$target_cpu],
         [i?86],   [BITS="32" X86="yes"],
         [x86_64], [BITS="64" X86="yes"])
 
