From d57c28e205462e51063e787f9ebddaadff592f1e Mon Sep 17 00:00:00 2001
From: Patrick <cowinkkeydinkinc@gmail.com>
Date: Tue, 16 May 2023 17:27:17 -0600
Subject: [PATCH] Fix build failure from lack of <cstdint> headers

Signed-off-by: Patrick <cowinkkeydinkinc@gmail.com>
---
 src/PommeDebug.h                   | 1 +
 src/Utilities/bigendianstreams.cpp | 1 +
 2 files changed, 2 insertions(+)

diff --git a/src/PommeDebug.h b/src/PommeDebug.h
index 34cd409..58344c4 100644
--- a/extern/Pomme/src/PommeDebug.h
+++ b/extern/Pomme/src/PommeDebug.h
@@ -4,6 +4,7 @@
 
 #include <string>
 #include <sstream>
+#include <cstdint>
 
 #define POMME_DEBUG_MEMORY		false
 #define POMME_DEBUG_SOUND		false
diff --git a/src/Utilities/bigendianstreams.cpp b/src/Utilities/bigendianstreams.cpp
index cb1f006..f19cc5b 100644
--- a/extern/Pomme/src/Utilities/bigendianstreams.cpp
+++ b/extern/Pomme/src/Utilities/bigendianstreams.cpp
@@ -1,5 +1,6 @@
 #include "Utilities/bigendianstreams.h"
 #include "Utilities/IEEEExtended.h"
+#include <cstdint>
 
 Pomme::StreamPosGuard::StreamPosGuard(std::istream& theStream) :
 	stream(theStream)
