--- a/autogen.sh
+++ b/autogen.sh
@@ -4,7 +4,7 @@
 
 test -z "$srcdir" && srcdir=.
 
-GNOMEDOC=`which yelp-build`
+GNOMEDOC=`command -v yelp-build`
 if test -z $GNOMEDOC; then
         echo "*** The tools to build the documentation are not found,"
         echo "    please intall the yelp-tool package ***"
--- a/configure.ac
+++ b/configure.ac
@@ -92,7 +92,6 @@ AC_LINK_IFELSE([AC_LANG_PROGRAM([], [])]
    AC_MSG_ERROR([C++ Compiler required to compile Anjuta]))
 AM_PROG_CC_C_O
 
-GNOME_COMPILE_WARNINGS([yes])
 AM_CFLAGS="$AM_CFLAGS $WARN_CFLAGS"
 AM_CXXFLAGS="$AM_CXXFLAGS $WARN_CXXFLAGS"
 
