commit 8f8fa97
Author: Rico Tzschichholz <ricotz@ubuntu.com>
Date:   Fri Aug 20 08:26:49 2021 +0200

    build: Allow building with vala 0.50

diff --git a/plugins/language-support-vala/plugin.vala b/plugins/language-support-vala/plugin.vala
index 718d74c..d94634c 100644
--- a/plugins/language-support-vala/plugin.vala
+++ b/plugins/language-support-vala/plugin.vala
@@ -78,7 +78,11 @@ public class ValaPlugin : Plugin, IAnjuta.Preferences {
 
 	void init_context () {
 		context = new Vala.CodeContext();
+#if VALA_0_50
+		context.set_target_profile (Vala.Profile.GOBJECT);
+#else
 		context.profile = Vala.Profile.GOBJECT;
+#endif
 		context.report = report;
 		report.clear_error_indicators ();
 
