diff --git a/src/Display/SimpleGui.py b/src/Display/SimpleGui.py
index 42705a82..6a274c16 100644
--- a/src/Display/SimpleGui.py
+++ b/src/Display/SimpleGui.py
@@ -161,8 +161,8 @@ def init_display(backend_str: Optional[str]=None,
             def centerOnScreen(self) -> None:
                 '''Centers the window on the screen.'''
                 resolution = QtWidgets.QApplication.desktop().screenGeometry()
-                x = (resolution.width() - self.frameSize().width()) / 2
-                y = (resolution.height() - self.frameSize().height()) / 2
+                x = (resolution.width() - self.frameSize().width()) // 2
+                y = (resolution.height() - self.frameSize().height()) // 2
                 self.move(x, y)
 
             def add_menu(self, menu_name: str) -> None:
