"codeset" in gettext.translation() is deprecated with python 3.11

--- a/DisplayCAL/defaultpaths.py
+++ b/DisplayCAL/defaultpaths.py
@@ -231,7 +231,7 @@
 
             try:
                 obj.translation = gettext.translation(
-                    obj.GETTEXT_PACKAGE, locale_dir, codeset="UTF-8"
+                    obj.GETTEXT_PACKAGE, locale_dir
                 )
             except IOError as exception:
                 print("XDG:", exception)
