--- a/DisplayCAL/RealDisplaySizeMM.py
+++ b/DisplayCAL/RealDisplaySizeMM.py
@@ -17,14 +17,7 @@
         # TODO: Intel vs ARM (Apple Silicon) distinction
         from DisplayCAL.lib64.RealDisplaySizeMM import *
 else:
-    # elif sys.platform == "win32":
-    # Windows have separate files
-    if sys.version_info[:2] == (3, 8):
-        from DisplayCAL.lib64.python38.RealDisplaySizeMM import *
-    elif sys.version_info[:2] == (3, 9):
-        from DisplayCAL.lib64.python39.RealDisplaySizeMM import *
-    elif sys.version_info[:2] == (3, 10):
-        from DisplayCAL.lib64.python310.RealDisplaySizeMM import *
+    from DisplayCAL.lib64.python@PY_VER@.RealDisplaySizeMM import *
 # else:
 #     pass
 
