Index: trace-cmd-2.9.1/kernel-shark/CMakeLists.txt
===================================================================
--- trace-cmd-2.9.1.orig/kernel-shark/CMakeLists.txt
+++ trace-cmd-2.9.1/kernel-shark/CMakeLists.txt
@@ -29,6 +29,7 @@ find_package(Doxygen)
 set(OpenGL_GL_PREFERENCE LEGACY)
 find_package(OpenGL)
 find_package(GLUT)
+link_libraries(${GLUT_LIBRARIES})
 
 find_package(Qt5Widgets 5.7.1)
 find_package(Qt5Network)
