--- src/qt/paymentrequestplus.cpp.orig	2017-04-18 12:28:33.789257513 +0200
+++ src/qt/paymentrequestplus.cpp	2017-04-18 12:29:01.597171846 +0200
@@ -159,7 +159,7 @@ bool PaymentRequestPlus::getMerchant(X50
         std::string data_to_verify;                     // Everything but the signature
         rcopy.SerializeToString(&data_to_verify);
 
-#if OPENSSL_VERSION_NUMBER >= 0x10100000L
+#if (OPENSSL_VERSION_NUMBER >= 0x10100000L) && !defined(LIBRESSL_VERSION_NUMBER)
         EVP_MD_CTX *ctx = EVP_MD_CTX_new();
         if (!ctx) throw SSLVerifyError("Error allocating OpenSSL context.");
 #else
@@ -174,7 +174,7 @@ bool PaymentRequestPlus::getMerchant(X50
             !EVP_VerifyFinal(ctx, (const unsigned char*)paymentRequest.signature().data(), (unsigned int)paymentRequest.signature().size(), pubkey)) {
             throw SSLVerifyError("Bad signature, invalid payment request.");
         }
-#if OPENSSL_VERSION_NUMBER >= 0x10100000L
+#if (OPENSSL_VERSION_NUMBER >= 0x10100000L) && !defined(LIBRESSL_VERSION_NUMBER)
         EVP_MD_CTX_free(ctx);
 #endif
 
