diff --git a/caffeine/main.py b/caffeine/main.py
index 516a138..f48653b 100644
--- a/caffeine/main.py
+++ b/caffeine/main.py
@@ -32,9 +32,9 @@ try:
     gi.require_version("Notify", "0.7")
 except ValueError:
     gi.require_version("Notify", "0.8")
-gi.require_version("AppIndicator3", "0.1")
+gi.require_version("AyatanaAppIndicator3", "0.1")
 
-from gi.repository import AppIndicator3  # noqa: E402
+from gi.repository import AyatanaAppIndicator3 as AppIndicator3  # noqa: E402
 from gi.repository import GdkPixbuf  # noqa: E402
 from gi.repository import Gtk  # noqa: E402
 from gi.repository.Notify import Notification  # noqa: E402
