--- .cargo/config
+++ .cargo/config
@@ -1,6 +1,3 @@
-[source.crates-io]
-replace-with = "vendored-sources"
-
 [source."https://gitlab.gnome.org/haecker-felix/libhandy-rs.git"]
 git = "https://gitlab.gnome.org/haecker-felix/libhandy-rs.git"
 branch = "update_cargo_version"
--- build-aux/cargo.sh
+++ build-aux/cargo.sh
@@ -5,9 +5,15 @@ export MESON_SOURCE_ROOT="$2"
 export APP_OUTPUT="$3"
 export APP_LOCALEDIR="$4"
 export APP_PROFILE="$5"
+export XBPS_RUST_TARGET="$6"
 
 export CARGO_TARGET_DIR="$MESON_BUILD_ROOT"/target
-export CARGO_HOME="$CARGO_TARGET_DIR"/cargo-home
+
+export CARGO_OUTPUT_DIR="$CARGO_TARGET_DIR"
+if ! test -z "$XBPS_RUST_TARGET"
+then
+    export CARGO_OUTPUT_DIR="${CARGO_OUTPUT_DIR}/${XBPS_RUST_TARGET}"
+fi
 
 echo "** RUST VERSION **"
 rustc --version
@@ -17,10 +23,10 @@ then
     echo "** DEBUG MODE **"
     cargo build --manifest-path \
         "$MESON_SOURCE_ROOT"/Cargo.toml --message-format=short && \
-        cp "$CARGO_TARGET_DIR"/debug/shortwave $APP_OUTPUT
+        cp "$CARGO_OUTPUT_DIR"/debug/shortwave $APP_OUTPUT
 else
     echo "** RELEASE MODE **"
     cargo build --manifest-path \
         "$MESON_SOURCE_ROOT"/Cargo.toml --message-format=short --release && \
-        cp "$CARGO_TARGET_DIR"/release/shortwave $APP_OUTPUT
+        cp "$CARGO_OUTPUT_DIR"/release/shortwave $APP_OUTPUT
 fi
--- meson.build
+++ meson.build
@@ -26,10 +26,11 @@ dependency('gstreamer-bad-audio-1.0', version: '>= 1.16')
 name       = 'Shortwave'
 app_id     = 'de.haeckerfelix.Shortwave'
 profile    = get_option('profile')
-vcs_tag    = run_command('git', 'rev-parse', '--short', 'HEAD').stdout().strip()
+vcs_tag    = get_option('tag')
 localedir  = join_paths(get_option('prefix'), get_option('localedir'))
 datadir    = get_option('datadir')
 pkgdatadir = join_paths(get_option('prefix'), datadir, meson.project_name())
+xbps_rust_target = get_option('xbps_rust_target')
 
 # Change app id for development builds
 if profile == 'development'
--- meson_options.txt
+++ meson_options.txt
@@ -8,3 +8,13 @@ option (
   ],
   value: 'default'
 )
+option (
+  'tag',
+  type: 'string',
+  value: 'none'
+)
+option (
+  'xbps_rust_target',
+  type: 'string',
+  value: ''
+)
--- src/meson.build
+++ src/meson.build
@@ -106,6 +106,7 @@ cargo_release = custom_target(
     meson.source_root(),
     '@OUTPUT@',
     localedir,
-    profile
+    profile,
+    xbps_rust_target
   ]
 )
