diff -ru a/meson.build b/meson.build
--- a/meson.build	2023-02-07 02:12:17.000000000 +0300
+++ b/meson.build	2023-04-27 23:57:14.057890647 +0300
@@ -31,7 +31,7 @@
 base_id    = 'de.haeckerfelix.Shortwave'
 path_id    = '/de/haeckerfelix/Shortwave'
 app_id     = base_id
-vcs_tag    = run_command('git', 'rev-parse', '--short', 'HEAD').stdout().strip()
+vcs_tag    = 'none'
 profile    = get_option('profile')
 datadir    = join_paths(get_option('prefix'), get_option('datadir'))
 localedir  = join_paths(get_option('prefix'), get_option('localedir'))
