--- custom/conf/app.ini.sample.orig	2020-06-22 08:49:03.000000000 +1000
+++ custom/conf/app.ini.sample	2020-07-10 10:53:31.047000000 +1000
@@ -7,12 +7,12 @@
 ; App name that shows in every page title
 APP_NAME = Gitea: Git with a cup of tea
 ; Change it if you run locally
-RUN_USER = git
+RUN_USER = _gitea
 ; Either "dev", "prod" or "test", default is "dev"
-RUN_MODE = dev
+RUN_MODE = prod
 
 [repository]
-ROOT =
+ROOT = /var/lib/gitea/repositories
 SCRIPT_TYPE = bash
 ; Default ANSI charset
 ANSI_CHARSET =
@@ -62,16 +62,16 @@
 PREVIEWABLE_FILE_MODES = markdown
 
 [repository.local]
-; Path for local repository copy. Defaults to `tmp/local-repo`
-LOCAL_COPY_PATH = tmp/local-repo
-; Path for local wiki copy. Defaults to `tmp/local-wiki`
-LOCAL_WIKI_PATH = tmp/local-wiki
+; Path for local repository copy. Defaults to `/var/lib/gitea/tmp/local-repo`
+LOCAL_COPY_PATH = /var/lib/gitea/tmp/local-repo
+; Path for local wiki copy. Defaults to `/var/lib/gitea/tmp/local-wiki`
+LOCAL_WIKI_PATH = /var/lib/gitea/tmp/local-wiki
 
 [repository.upload]
 ; Whether repository file uploads are enabled. Defaults to `true`
 ENABLED = true
 ; Path for uploads. Defaults to `data/tmp/uploads` (tmp gets deleted on gitea restart)
-TEMP_PATH = data/tmp/uploads
+TEMP_PATH = /var/lib/gitea/tmp/uploads
 ; One or more allowed types, e.g. image/jpeg|image/png. Nothing means any file type
 ALLOWED_TYPES =
 ; Max size of each file in megabytes. Defaults to 3MB
@@ -248,7 +248,7 @@
 REDIRECT_OTHER_PORT = false
 PORT_TO_REDIRECT = 80
 ; Permission for unix socket
-UNIX_SOCKET_PERMISSION = 666
+UNIX_SOCKET_PERMISSION = 660
 ; Local (DMZ) URL for Gitea workers (such as SSH update) accessing web service.
 ; In most cases you do not need to change the default value.
 ; Alter it only if your SSH server node is not the same as HTTP node.
@@ -269,7 +269,7 @@
 ; The port number the builtin SSH server should listen on
 SSH_LISTEN_PORT = %(SSH_PORT)s
 ; Root path of SSH directory, default is '~/.ssh', but you have to use '/home/git/.ssh'.
-SSH_ROOT_PATH =
+SSH_ROOT_PATH = /var/lib/gitea
 ; Gitea will create a authorized_keys file by default when it is not using the internal ssh server
 ; If you intend to use the AuthorizedKeysCommand functionality then you should turn this off.
 SSH_CREATE_AUTHORIZED_KEYS_FILE = true
@@ -308,7 +308,7 @@
 KEY_FILE = https/key.pem
 ; Root directory containing templates and static files.
 ; default is the path where Gitea is executed
-STATIC_ROOT_PATH =
+STATIC_ROOT_PATH = /var/lib/gitea
 ; Default path for App data
 APP_DATA_PATH = data
 ; Application level GZIP support
@@ -325,7 +325,7 @@
 ; Enables git-lfs support. true or false, default is false.
 LFS_START_SERVER = false
 ; Where your lfs files reside, default is data/lfs.
-LFS_CONTENT_PATH = data/lfs
+LFS_CONTENT_PATH = /var/lib/gitea/lfs
 ; LFS authentication secret, change this yourself
 LFS_JWT_SECRET =
 ; LFS authentication validity period (in time.Duration), pushes taking longer than this may fail.
@@ -355,7 +355,7 @@
 
 [database]
 ; Database to use. Either "mysql", "postgres", "mssql" or "sqlite3".
-DB_TYPE = mysql
+DB_TYPE = sqlite3
 HOST = 127.0.0.1:3306
 NAME = gitea
 USER = root
@@ -372,7 +372,7 @@
 ; NOTICE: for "utf8mb4" you must use MySQL InnoDB > 5.6. Gitea is unable to check this.
 CHARSET = utf8
 ; For "sqlite3" and "tidb", use an absolute path when you start gitea as service
-PATH = data/gitea.db
+PATH = /var/lib/gitea/data/gitea.db
 ; For "sqlite3" only. Query timeout
 SQLITE_TIMEOUT = 500
 ; For iterate buffer, default is 50
@@ -398,7 +398,7 @@
 ; Issue indexer name, available when ISSUE_INDEXER_TYPE is elasticsearch
 ISSUE_INDEXER_NAME = gitea_issues
 ; Issue indexer storage path, available when ISSUE_INDEXER_TYPE is bleve
-ISSUE_INDEXER_PATH = indexers/issues.bleve
+ISSUE_INDEXER_PATH = /var/lib/gitea/indexers/issues.bleve
 ; Issue indexer queue, currently support: channel, levelqueue or redis, default is levelqueue
 ISSUE_INDEXER_QUEUE_TYPE = levelqueue
 ; When ISSUE_INDEXER_QUEUE_TYPE is levelqueue, this will be the queue will be saved path,
@@ -706,7 +706,7 @@
 SESSION_LIFE_TIME = 86400
 
 [picture]
-AVATAR_UPLOAD_PATH = data/avatars
+AVATAR_UPLOAD_PATH = /var/lib/gitea/data/avatars
 REPOSITORY_AVATAR_UPLOAD_PATH = data/repo-avatars
 ; How Gitea deals with missing repository avatars
 ; none = no avatar will be displayed; random = random avatar will be displayed; image = default image will be used
@@ -751,7 +751,7 @@
 DEFAULT_UI_LOCATION =
 
 [log]
-ROOT_PATH =
+ROOT_PATH = /var/log/gitea
 ; Either "console", "file", "conn", "smtp" or "database", default is "console"
 ; Use comma to separate multiple modes, e.g. "console, file"
 MODE = console
@@ -968,14 +968,6 @@
 sv-SE = sv
 ko-KR = ko
 
-[U2F]
-; NOTE: THE DEFAULT VALUES HERE WILL NEED TO BE CHANGED
-; Two Factor authentication with security keys
-; https://developers.yubico.com/U2F/App_ID.html
-;APP_ID = http://localhost:3000/
-; Comma seperated list of trusted facets
-;TRUSTED_FACETS = http://localhost:3000/
-
 ; Extension mapping to highlight class
 ; e.g. .toml=ini
 [highlight.mapping]
