From e6994dc234395f68fd7195bbe7424f83c39d0f3d Mon Sep 17 00:00:00 2001
From: Tuukka Pasanen <tuukka.pasanen@ilmi.fi>
Date: Thu, 22 Oct 2015 09:53:05 +0300
Subject: [PATCH 3/3] Fixed PList deprication
---
 tools/ipod-lockdown.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)
diff --git a/tools/ipod-lockdown.c b/tools/ipod-lockdown.c
index f345726..8503798 100644
--- a/tools/ipod-lockdown.c
+++ b/tools/ipod-lockdown.c
@@ -100,7 +100,7 @@ read_sysinfo_extended_by_uuid (const char *uuid)
 	plist_get_string_val(ptr, &str);
 	if (str != NULL) {
 	    ptr = plist_new_string(str);
-	    plist_dict_insert_item(value, "SerialNumber", ptr);
+	    plist_dict_set_item(value, "SerialNumber", ptr);
 	    free(str);
 	}
 
@@ -112,15 +112,15 @@ read_sysinfo_extended_by_uuid (const char *uuid)
 	plist_get_string_val(ptr, &str);
 	if (str != NULL) {
 	    ptr = plist_new_string(str);
-	    plist_dict_insert_item(value, "VisibleBuildID", ptr);
+	    plist_dict_set_item(value, "VisibleBuildID", ptr);
 	    free(str);
 	}
 
 	ptr = plist_new_string(uuid);
-	plist_dict_insert_item(value, "FireWireGUID", ptr);
+	plist_dict_set_item(value, "FireWireGUID", ptr);
 
 	ptr = plist_new_string(uuid);
-	plist_dict_insert_item(value, "UniqueDeviceID", ptr);
+	plist_dict_set_item(value, "UniqueDeviceID", ptr);
 
 	plist_to_xml(value, &xml, &xml_length);
 
-- 
2.1.4

