--- a/src/ithumb-writer.c
+++ b/src/ithumb-writer.c
@@ -728,7 +728,7 @@ ithumb_writer_handle_rotation (GdkPixbuf
   {
       return gdk_pixbuf_rotate_simple (pixbuf, *rotation);
   }
-  return g_object_ref (G_OBJECT (pixbuf));
+  return GDK_PIXBUF(g_object_ref (G_OBJECT (pixbuf)));
 }
 
 /* On the iPhone, thumbnails are presented as squares in a grid.
@@ -968,7 +968,7 @@ ithumb_writer_write_thumbnail (iThumbWri
     else if (thumb->data_type == ITDB_THUMB_TYPE_PIXBUF)
     {
         Itdb_Thumb_Pixbuf *thumb_pixbuf = (Itdb_Thumb_Pixbuf *)thumb;
-        pixbuf = g_object_ref (G_OBJECT (thumb_pixbuf->pixbuf));
+        pixbuf = GDK_PIXBUF(g_object_ref (G_OBJECT (thumb_pixbuf->pixbuf)));
     }
 
     if (pixbuf == NULL)
--- a/tools/generic-callout.c
+++ b/tools/generic-callout.c
@@ -664,7 +664,7 @@ static gboolean write_sysinfo_extended (
 }
 
 
-static char *get_info_from_usb (usb_bus_number, usb_device_number)
+static char *get_info_from_usb (gint usb_bus_number, gint usb_device_number)
 {
 #ifdef HAVE_LIBUSB
         return read_sysinfo_extended_from_usb (usb_bus_number,
