--- a/doc/Makefile.am
+++ b/doc/Makefile.am
@@ -1,7 +1,7 @@
 man_MANS = gmpc.1 gmpc-remote.1 gmpc-remote-stream.1
 EXTRA_DIST = $(man_MANS)
 
-SUBDIRS=Mallard
+SUBDIRS=
 
 .PHONY: doc
 doc: Manual.txt
--- a/doc/Makefile.in
+++ b/doc/Makefile.in
@@ -311,7 +311,7 @@ unique_CFLAGS = @unique_CFLAGS@
 unique_LIBS = @unique_LIBS@
 man_MANS = gmpc.1 gmpc-remote.1 gmpc-remote-stream.1
 EXTRA_DIST = $(man_MANS)
-SUBDIRS = Mallard
+SUBDIRS =
 all: all-recursive
 
 .SUFFIXES:
