diff --git src/notes.c src/notes.c
index 0c427dd..5d32740 100644
--- a/src/notes.c
+++ b/src/notes.c
@@ -731,7 +731,7 @@ correctNoteHeight( Note *note )
 /*
   Reset the width of note internals (e.g. the pango layout) to note->width.
 */
-inline void
+void
 resetNoteWidth( Note *note )
 {
     pango_layout_set_width( note->playout,
diff --git src/xnots.h src/xnots.h
index 8280ccb..ef7395b 100644
--- a/src/xnots.h
+++ b/src/xnots.h
@@ -315,7 +315,7 @@ void		correctNoteGeometry	( Note		*note,
 					  unsigned	height
 					);
 void		correctNoteHeight	( Note *note );
-inline void	resetNoteWidth		( Note *note );
+void		resetNoteWidth		( Note *note );
 void		freeNote		( Note *note );
 void		showNotes		();
 void		setNoteText		( Note *note, char *text);
