--- joystick_linux.cpp  2017-12-12 21:28:08.263329050 +0100
+++ platform/x11/joystick_linux.cpp     2017-12-12 21:29:25.500336429 +0100
@@ -42,6 +42,10 @@
 #include <libudev.h>
 #endif

+#if !defined(__GLIBC__)
+#include <byteswap.h>
+#endif
+
 #define LONG_BITS (sizeof(long) * 8)
 #define test_bit(nr, addr) (((1UL << ((nr) % LONG_BITS)) & ((addr)[(nr) / LONG_BITS])) != 0)
 #define NBITS(x) ((((x)-1) / LONG_BITS) + 1)

