--- Makefile	2020-07-31 18:42:11.268427773 +0100
+++ Makefile	2020-07-31 19:30:33.904122987 +0100
@@ -7,9 +7,9 @@
 PREFIX = /usr/local
 MANPREFIX = $(PREFIX)/share/man
 
-CC = cc
+CC ?= cc
 LD = ld
-CFLAGS = -std=c99 -pedantic -Wall -Wextra -Werror -Wno-unused-parameter -Os -s -D_GNU_SOURCE $(shell pkg-config --cflags libnotify)
+CFLAGS += -std=c99 -D_GNU_SOURCE $(shell pkg-config --cflags libnotify)
 LDLIBS = $(shell pkg-config --libs libnotify)
 
 SRC = main.c version.h
