--- a/work/sbin/update-ca-certificates	2021-11-29 23:24:29.772537834 -0600
+++ b/work/sbin/update-ca-certificates	2021-11-29 23:26:49.158620616 -0600
@@ -81,8 +81,8 @@
 # Helper files.  (Some of them are not simple arrays because we spawn
 # subshells later on.)
 TEMPBUNDLE="${ETCCERTSDIR}/${CERTBUNDLE}.new"
-ADDED="$(mktemp --tmpdir "ca-certificates.tmp.XXXXXX")"
-REMOVED="$(mktemp --tmpdir "ca-certificates.tmp.XXXXXX")"
+ADDED="$(mktemp -p "ca-certificates.tmp.XXXXXX")"
+REMOVED="$(mktemp -p "ca-certificates.tmp.XXXXXX")"
 
 # Adds a certificate to the list of trusted ones.  This includes a symlink
 # in /etc/ssl/certs to the certificate file and its inclusion into the
