Use $DESTDIR to override some paths.

--- sbin/update-ca-certificates.orig	2012-12-12 04:37:18.204915114 +0100
+++ sbin/update-ca-certificates	2012-12-12 04:38:15.068574773 +0100
@@ -37,11 +37,11 @@ do
   shift
 done
 
-CERTSCONF=/etc/ca-certificates.conf
-CERTSDIR=/usr/share/ca-certificates
-LOCALCERTSDIR=/usr/local/share/ca-certificates
+CERTSCONF=$DESTDIR/etc/ca-certificates.conf
+CERTSDIR=$DESTDIR/usr/share/ca-certificates
+LOCALCERTSDIR=$DESTDIR/usr/local/share/ca-certificates
 CERTBUNDLE=ca-certificates.crt
-ETCCERTSDIR=/etc/ssl/certs
+ETCCERTSDIR=$DESTDIR/etc/ssl/certs
 
 cleanup() {
   rm -f "$TEMPBUNDLE"
@@ -148,7 +148,7 @@ mv -f "$TEMPBUNDLE" "$CERTBUNDLE"
 
 echo "$ADDED_CNT added, $REMOVED_CNT removed; done."
 
-HOOKSDIR=/etc/ca-certificates/update.d
+HOOKSDIR=$DESTDIR/etc/ca-certificates/update.d
 echo -n "Running hooks in $HOOKSDIR...."
 VERBOSE_ARG=
 [ "$verbose" = 0 ] || VERBOSE_ARG=--verbose
