--- a/ca-certificates/sbin/update-ca-certificates
+++ b/ca-certificates/sbin/update-ca-certificates
@@ -24,12 +24,12 @@
 verbose=0
 fresh=0
 default=0
-CERTSCONF=/etc/ca-certificates.conf
+CERTSCONF=$DESTDIR/etc/ca-certificates.conf
 CERTSDIR=/usr/share/ca-certificates
-LOCALCERTSDIR=/usr/local/share/ca-certificates
+LOCALCERTSDIR=$DESTDIR/usr/local/share/ca-certificates
 CERTBUNDLE=ca-certificates.crt
-ETCCERTSDIR=/etc/ssl/certs
-HOOKSDIR=/etc/ca-certificates/update.d
+ETCCERTSDIR=$DESTDIR/etc/ssl/certs
+HOOKSDIR=$DESTDIR/etc/ca-certificates/update.d
 
 while [ $# -gt 0 ];
 do
