--- a/sbin/update-ca-certificates	2015-05-29 11:09:43.922158838 +0200
+++ b/sbin/update-ca-certificates	2015-05-29 11:10:06.842632933 +0200
@@ -24,12 +24,12 @@
 verbose=0
 fresh=0
 default=0
-CERTSCONF=/etc/ca-certificates.conf
-CERTSDIR=/usr/share/ca-certificates
-LOCALCERTSDIR=/usr/local/share/ca-certificates
+CERTSCONF=$DESTDIR/etc/ca-certificates.conf
+CERTSDIR=/usr/share/ca-certificates
+LOCALCERTSDIR=$DESTDIR/usr/local/share/ca-certificates
 CERTBUNDLE=ca-certificates.crt
-ETCCERTSDIR=/etc/ssl/certs
-HOOKSDIR=/etc/ca-certificates/update.d
+ETCCERTSDIR=$DESTDIR/etc/ssl/certs
+HOOKSDIR=$DESTDIR/etc/ca-certificates/update.d
 
 while [ $# -gt 0 ];
 do
