--- cimg.cc.orig	2017-01-05 00:43:20.664395461 +0100
+++ cimg.cc	2017-01-05 00:44:12.366397039 +0100
@@ -94,11 +94,11 @@
       ct == PNG_COLOR_TYPE_GRAY_ALPHA)
     png_set_gray_to_rgb(pngp);
 
-  alloc(pngp->width,pngp->height);
+  alloc(width,height);
   if (!ok) { fclose(f); return; }
   ok = 0;
 
-  for(i=0;i<pngp->height;i++) {
+  for(i=0;i<height;i++) {
     png_read_row(pngp, (png_bytep) (&data[i*rowlen]), NULL);
   }
 
