--- a/src/DateTime.h	2024-07-10 14:12:58.000000000 +0200
+++ -	2024-12-26 23:58:03.454800522 +0100
@@ -2,6 +2,7 @@
 #define DATETIME_H
 
 #include <string>
+#include <cstdint>
 
 namespace Time {
 
--- a/contrib/nanosockets/nanosockets.h	2024-07-10 14:12:58.000000000 +0200
+++ -	2024-12-27 00:00:02.236314638 +0100
@@ -142,6 +142,7 @@
 		#include <netdb.h>
 		#include <unistd.h>
 		#include <sys/socket.h>
+		#include <sys/select.h>
 	#endif
 
 	// Macros
