--- src/posix/OSPosix.cpp	2019-02-03 11:56:16.000000000 +0100
+++ -	2019-02-03 17:32:45.957535773 +0100
@@ -48,7 +48,7 @@
 
 	void EnableFPE()
 	{
-#if defined(_GNU_SOURCE) && !defined(__APPLE__)
+#if defined(_GNU_SOURCE) && !defined(__APPLE__) && defined(__GLIBC__)
 		// clear any outstanding exceptions before enabling, otherwise they'll
 		// trip immediately
 		feclearexcept(FE_DIVBYZERO | FE_INVALID | FE_OVERFLOW);
@@ -58,7 +58,7 @@
 
 	void DisableFPE()
 	{
-#if defined(_GNU_SOURCE) && !defined(__APPLE__)
+#if defined(_GNU_SOURCE) && !defined(__APPLE__) && defined(__GLIBC__)
 		fedisableexcept(FE_DIVBYZERO | FE_INVALID | FE_OVERFLOW);
 #endif
 	}
