diff --git a/lib/colord/meson.build b/lib/colord/meson.build
index d0f412b..49139dd 100644
--- lib/colord/meson.build
+++ lib/colord/meson.build
@@ -187,6 +187,7 @@ if get_option('introspection')
 	  export_packages : 'colord',
 	  extra_args : [
 		'--c-include=colord.h',
+		'-DCD_COMPILATION',
 		],
 	  link_with : colordprivate,
 	  dependencies : [

diff --git a/lib/colorhug/meson.build b/lib/colorhug/meson.build
index e177ef3..f88e8e4 100644
--- lib/colorhug/meson.build
+++ lib/colorhug/meson.build
@@ -102,6 +102,9 @@ if get_option('introspection')
 		'GUsb-1.0',
 		libcolord_gir,
 	  ],
+      extra_args : [
+        '-DCD_COMPILATION',
+      ],
 	  install : true
 	)
 endif

