--- a/data/cmf/meson.build
+++ b/data/cmf/meson.build
@@ -1,3 +1,7 @@
+if meson.is_cross_build()
+  cd_idt8 = find_program('cd-it8')
+endif
+
 foreach arg: [ 'CIE1964-10deg-XYZ', 'CIE1931-2deg-XYZ' ]
   custom_target(arg,
     input: arg + '.csv',
--- a/data/profiles/meson.build
+++ b/data/profiles/meson.build
@@ -47,6 +47,10 @@ if get_option('print_profiles')
   icc_profiles += icc_print_profiles
 endif
 
+if meson.is_cross_build()
+  cd_create_profile = find_program('cd-create-profile')
+endif
+
 generated_iccs = []
 foreach arg: icc_profiles
   xml_i18n = i18n.merge_file(

