diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8f3516e..7e1960f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -21,7 +21,7 @@ if (WIN32)
 elseif (APPLE)
 	set(KIMAGEANNOTATOR_LANG_INSTALL_DIR "../Resources")
 elseif (UNIX)
-	set(KIMAGEANNOTATOR_LANG_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_DATAROOTDIR}/kImageAnnotator/translations")
+	set(KIMAGEANNOTATOR_LANG_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_DATAROOTDIR}/kImageAnnotator-Qt5/translations")
 	string(REGEX REPLACE "//" "/" KIMAGEANNOTATOR_LANG_INSTALL_DIR "${KIMAGEANNOTATOR_LANG_INSTALL_DIR}")
 endif ()
 
@@ -97,6 +97,7 @@ set_target_properties(kImageAnnotator
 					  RUNTIME_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/bin
 					  VERSION ${PROJECT_VERSION}
 					  SOVERSION 0
+					  OUTPUT_NAME kImageAnnotator-Qt5
 					  )
 
 install(TARGETS kImageAnnotator
