diff --git a/tests/util/path/core.c b/tests/util/path/core.c
index d1935a8..3d98f35 100644
--- a/tests/util/path/core.c
+++ b/tests/util/path/core.c
@@ -761,7 +761,7 @@ void test_path_core__validate_current_user_ownership(void)
 	cl_git_fail(git_fs_path_owner_is_current_user(&is_cur, "c:\\path\\does\\not\\exist"));
 #else
 	cl_git_pass(git_fs_path_owner_is_current_user(&is_cur, "/"));
-	cl_assert_equal_i(is_cur, (geteuid() == 0));
+	/* cl_assert_equal_i(is_cur, (geteuid() == 0)); xbps-src chroot belongs to $USER, not root */
 
 	cl_git_fail(git_fs_path_owner_is_current_user(&is_cur, "/path/does/not/exist"));
 #endif
