Index: cryfs-0.10.2/src/blockstore/implementations/compressing/compressors/Gzip.cpp
===================================================================
--- cryfs-0.10.2.orig/src/blockstore/implementations/compressing/compressors/Gzip.cpp
+++ cryfs-0.10.2/src/blockstore/implementations/compressing/compressors/Gzip.cpp
@@ -1,6 +1,6 @@
 #include "cpp-utils/crypto/cryptopp_byte.h"
 #include "Gzip.h"
-#include <vendor_cryptopp/gzip.h>
+#include <cryptopp/gzip.h>
 
 using cpputils::Data;
 
Index: cryfs-0.10.2/src/cpp-utils/crypto/cryptopp_byte.h
===================================================================
--- cryfs-0.10.2.orig/src/cpp-utils/crypto/cryptopp_byte.h
+++ cryfs-0.10.2/src/cpp-utils/crypto/cryptopp_byte.h
@@ -2,7 +2,7 @@
 #ifndef _CPPUTILS_CRYPTO_CRYPTOPP_BYTE_H
 #define _CPPUTILS_CRYPTO_CRYPTOPP_BYTE_H
 
-#include <vendor_cryptopp/cryptlib.h>
+#include <cryptopp/cryptlib.h>
 
 // If we're running an older CryptoPP version, CryptoPP::byte isn't defined yet.
 // Define it. Refer to "byte" type in the global namespace (placed by CryptoPP).
Index: cryfs-0.10.2/src/cpp-utils/crypto/hash/Hash.cpp
===================================================================
--- cryfs-0.10.2.orig/src/cpp-utils/crypto/hash/Hash.cpp
+++ cryfs-0.10.2/src/cpp-utils/crypto/hash/Hash.cpp
@@ -1,6 +1,6 @@
 #include "Hash.h"
 #include <cpp-utils/random/Random.h>
-#include <vendor_cryptopp/sha.h>
+#include <cryptopp/sha.h>
 
 using cpputils::Random;
 using CryptoPP::SHA512;
Index: cryfs-0.10.2/src/cpp-utils/crypto/kdf/Scrypt.cpp
===================================================================
--- cryfs-0.10.2.orig/src/cpp-utils/crypto/kdf/Scrypt.cpp
+++ cryfs-0.10.2/src/cpp-utils/crypto/kdf/Scrypt.cpp
@@ -1,5 +1,5 @@
 #include "Scrypt.h"
-#include <vendor_cryptopp/scrypt.h>
+#include <cryptopp/scrypt.h>
 
 using std::string;
 
Index: cryfs-0.10.2/src/cpp-utils/crypto/symmetric/CFB_Cipher.h
===================================================================
--- cryfs-0.10.2.orig/src/cpp-utils/crypto/symmetric/CFB_Cipher.h
+++ cryfs-0.10.2/src/cpp-utils/crypto/symmetric/CFB_Cipher.h
@@ -7,7 +7,7 @@
 #include "../../data/Data.h"
 #include "../../random/Random.h"
 #include <boost/optional.hpp>
-#include <vendor_cryptopp/modes.h>
+#include <cryptopp/modes.h>
 #include "Cipher.h"
 #include "EncryptionKey.h"
 
Index: cryfs-0.10.2/src/cpp-utils/crypto/symmetric/GCM_Cipher.h
===================================================================
--- cryfs-0.10.2.orig/src/cpp-utils/crypto/symmetric/GCM_Cipher.h
+++ cryfs-0.10.2/src/cpp-utils/crypto/symmetric/GCM_Cipher.h
@@ -6,7 +6,7 @@
 #include "../../data/FixedSizeData.h"
 #include "../../data/Data.h"
 #include "../../random/Random.h"
-#include <vendor_cryptopp/gcm.h>
+#include <cryptopp/gcm.h>
 #include "Cipher.h"
 #include "EncryptionKey.h"
 
Index: cryfs-0.10.2/src/cpp-utils/crypto/symmetric/ciphers.h
===================================================================
--- cryfs-0.10.2.orig/src/cpp-utils/crypto/symmetric/ciphers.h
+++ cryfs-0.10.2/src/cpp-utils/crypto/symmetric/ciphers.h
@@ -2,11 +2,11 @@
 #ifndef MESSMER_CPPUTILS_CRYPTO_SYMMETRIC_CIPHERS_H_
 #define MESSMER_CPPUTILS_CRYPTO_SYMMETRIC_CIPHERS_H_
 
-#include <vendor_cryptopp/aes.h>
-#include <vendor_cryptopp/twofish.h>
-#include <vendor_cryptopp/serpent.h>
-#include <vendor_cryptopp/cast.h>
-#include <vendor_cryptopp/mars.h>
+#include <cryptopp/aes.h>
+#include <cryptopp/twofish.h>
+#include <cryptopp/serpent.h>
+#include <cryptopp/cast.h>
+#include <cryptopp/mars.h>
 #include "GCM_Cipher.h"
 #include "CFB_Cipher.h"
 
Index: cryfs-0.10.2/src/cpp-utils/data/Data.cpp
===================================================================
--- cryfs-0.10.2.orig/src/cpp-utils/data/Data.cpp
+++ cryfs-0.10.2/src/cpp-utils/data/Data.cpp
@@ -1,6 +1,6 @@
 #include "Data.h"
 #include <stdexcept>
-#include <vendor_cryptopp/hex.h>
+#include <cryptopp/hex.h>
 #include <cpp-utils/crypto/cryptopp_byte.h>
 
 using std::istream;
Index: cryfs-0.10.2/src/cpp-utils/data/FixedSizeData.h
===================================================================
--- cryfs-0.10.2.orig/src/cpp-utils/data/FixedSizeData.h
+++ cryfs-0.10.2/src/cpp-utils/data/FixedSizeData.h
@@ -2,7 +2,7 @@
 #ifndef MESSMER_CPPUTILS_DATA_FIXEDSIZEDATA_H_
 #define MESSMER_CPPUTILS_DATA_FIXEDSIZEDATA_H_
 
-#include <vendor_cryptopp/hex.h>
+#include <cryptopp/hex.h>
 #include <string>
 #include <cstring>
 #include "../assert/assert.h"
Index: cryfs-0.10.2/src/cpp-utils/random/OSRandomGenerator.h
===================================================================
--- cryfs-0.10.2.orig/src/cpp-utils/random/OSRandomGenerator.h
+++ cryfs-0.10.2/src/cpp-utils/random/OSRandomGenerator.h
@@ -4,7 +4,7 @@
 
 #include "cpp-utils/crypto/cryptopp_byte.h"
 #include "RandomGenerator.h"
-#include <vendor_cryptopp/osrng.h>
+#include <cryptopp/osrng.h>
 
 namespace cpputils {
     class OSRandomGenerator final : public RandomGenerator {
Index: cryfs-0.10.2/src/cpp-utils/random/RandomGeneratorThread.h
===================================================================
--- cryfs-0.10.2.orig/src/cpp-utils/random/RandomGeneratorThread.h
+++ cryfs-0.10.2/src/cpp-utils/random/RandomGeneratorThread.h
@@ -4,7 +4,7 @@
 
 #include "../thread/LoopThread.h"
 #include "ThreadsafeRandomDataBuffer.h"
-#include <vendor_cryptopp/osrng.h>
+#include <cryptopp/osrng.h>
 
 namespace cpputils {
     //TODO Test
Index: cryfs-0.10.2/src/cryfs/localstate/BasedirMetadata.cpp
===================================================================
--- cryfs-0.10.2.orig/src/cryfs/localstate/BasedirMetadata.cpp
+++ cryfs-0.10.2/src/cryfs/localstate/BasedirMetadata.cpp
@@ -1,7 +1,7 @@
 #include "BasedirMetadata.h"
 #include <boost/property_tree/ptree.hpp>
 #include <boost/property_tree/json_parser.hpp>
-#include <vendor_cryptopp/sha.h>
+#include <cryptopp/sha.h>
 #include <boost/filesystem/operations.hpp>
 #include "LocalStateDir.h"
 #include <cpp-utils/logging/logging.h>
Index: cryfs-0.10.2/test/cryfs/config/CompatibilityTest.cpp
===================================================================
--- cryfs-0.10.2.orig/test/cryfs/config/CompatibilityTest.cpp
+++ cryfs-0.10.2/test/cryfs/config/CompatibilityTest.cpp
@@ -3,7 +3,7 @@
 #include <vector>
 #include <boost/filesystem.hpp>
 #include <cpp-utils/data/Data.h>
-#include <vendor_cryptopp/hex.h>
+#include <cryptopp/hex.h>
 #include <cpp-utils/crypto/symmetric/ciphers.h>
 #include <cpp-utils/tempfile/TempFile.h>
 #include <cryfs/config/CryConfigFile.h>
Index: cryfs-0.10.2/vendor/CMakeLists.txt
===================================================================
--- cryfs-0.10.2.orig/vendor/CMakeLists.txt
+++ cryfs-0.10.2/vendor/CMakeLists.txt
@@ -1,3 +1 @@
 add_subdirectory(googletest)
-add_subdirectory(spdlog)
-add_subdirectory(cryptopp)
