--- a/src/blockstore/implementations/compressing/compressors/Gzip.cpp	2022-08-25 22:14:05.481898162 +0400
+++ b/src/blockstore/implementations/compressing/compressors/Gzip.cpp	2022-08-25 22:17:49.606653292 +0400
@@ -1,5 +1,5 @@
 #include "Gzip.h"
-#include <vendor_cryptopp/gzip.h>
+#include <cryptopp/gzip.h>
 
 using cpputils::Data;
 
--- a/src/cpp-utils/crypto/hash/Hash.cpp	2022-08-25 22:14:05.482898159 +0400
+++ b/src/cpp-utils/crypto/hash/Hash.cpp	2022-08-25 22:18:07.131800277 +0400
@@ -1,6 +1,6 @@
 #include "Hash.h"
 #include <cpp-utils/random/Random.h>
-#include <vendor_cryptopp/sha.h>
+#include <cryptopp/sha.h>
 
 using cpputils::Random;
 using CryptoPP::SHA512;
--- a/src/cpp-utils/crypto/kdf/Scrypt.cpp	2022-08-25 22:14:05.482898159 +0400
+++ b/src/cpp-utils/crypto/kdf/Scrypt.cpp	2022-08-25 22:18:17.754889373 +0400
@@ -1,5 +1,5 @@
 #include "Scrypt.h"
-#include <vendor_cryptopp/scrypt.h>
+#include <cryptopp/scrypt.h>
 
 using std::string;
 
--- a/src/cpp-utils/crypto/symmetric/CFB_Cipher.h	2022-08-25 22:14:05.482898159 +0400
+++ b/src/cpp-utils/crypto/symmetric/CFB_Cipher.h	2022-08-25 22:18:28.524979704 +0400
@@ -6,7 +6,7 @@
 #include "../../data/Data.h"
 #include "../../random/Random.h"
 #include <boost/optional.hpp>
-#include <vendor_cryptopp/modes.h>
+#include <cryptopp/modes.h>
 #include "Cipher.h"
 #include "EncryptionKey.h"
 
--- a/src/cpp-utils/crypto/symmetric/ciphers.h	2022-08-25 22:14:05.482898159 +0400
+++ b/src/cpp-utils/crypto/symmetric/ciphers.h	2022-08-25 22:18:52.181178110 +0400
@@ -2,12 +2,12 @@
 #ifndef MESSMER_CPPUTILS_CRYPTO_SYMMETRIC_CIPHERS_H_
 #define MESSMER_CPPUTILS_CRYPTO_SYMMETRIC_CIPHERS_H_
 
-#include <vendor_cryptopp/aes.h>
-#include <vendor_cryptopp/twofish.h>
-#include <vendor_cryptopp/serpent.h>
-#include <vendor_cryptopp/cast.h>
-#include <vendor_cryptopp/mars.h>
-#include <vendor_cryptopp/chachapoly.h>
+#include <cryptopp/aes.h>
+#include <cryptopp/twofish.h>
+#include <cryptopp/serpent.h>
+#include <cryptopp/cast.h>
+#include <cryptopp/mars.h>
+#include <cryptopp/chachapoly.h>
 #include "GCM_Cipher.h"
 #include "CFB_Cipher.h"
 
--- a/src/cpp-utils/crypto/symmetric/GCM_Cipher.h	2022-08-25 22:14:05.482898159 +0400
+++ b/src/cpp-utils/crypto/symmetric/GCM_Cipher.h	2022-08-25 22:18:38.836066183 +0400
@@ -3,7 +3,7 @@
 #define MESSMER_CPPUTILS_CRYPTO_SYMMETRIC_GCMCIPHER_H_
 
 #include "AEAD_Cipher.h"
-#include <vendor_cryptopp/gcm.h>
+#include <cryptopp/gcm.h>
 
 namespace cpputils {
 
--- a/src/cpp-utils/data/Data.cpp	2022-08-25 22:14:05.482898159 +0400
+++ b/src/cpp-utils/data/Data.cpp	2022-08-25 22:19:09.349322106 +0400
@@ -1,6 +1,6 @@
 #include "Data.h"
 #include <stdexcept>
-#include <vendor_cryptopp/hex.h>
+#include <cryptopp/hex.h>
 
 using std::istream;
 using std::ofstream;
--- a/src/cpp-utils/data/FixedSizeData.h	2022-08-25 22:14:05.482898159 +0400
+++ b/src/cpp-utils/data/FixedSizeData.h	2022-08-25 22:19:19.103403916 +0400
@@ -2,7 +2,7 @@
 #ifndef MESSMER_CPPUTILS_DATA_FIXEDSIZEDATA_H_
 #define MESSMER_CPPUTILS_DATA_FIXEDSIZEDATA_H_
 
-#include <vendor_cryptopp/hex.h>
+#include <cryptopp/hex.h>
 #include <string>
 #include <array>
 #include <cstring>
--- a/src/cpp-utils/random/OSRandomGenerator.h	2022-08-25 22:14:05.483898156 +0400
+++ b/src/cpp-utils/random/OSRandomGenerator.h	2022-08-25 22:19:29.602491973 +0400
@@ -3,7 +3,7 @@
 #define MESSMER_CPPUTILS_RANDOM_OSRANDOMGENERATOR_H
 
 #include "RandomGenerator.h"
-#include <vendor_cryptopp/osrng.h>
+#include <cryptopp/osrng.h>
 
 namespace cpputils {
     class OSRandomGenerator final : public RandomGenerator {
--- a/src/cpp-utils/random/RandomGeneratorThread.h	2022-08-25 22:14:05.484898153 +0400
+++ b/src/cpp-utils/random/RandomGeneratorThread.h	2022-08-25 22:19:36.713551613 +0400
@@ -4,7 +4,7 @@
 
 #include "../thread/LoopThread.h"
 #include "ThreadsafeRandomDataBuffer.h"
-#include <vendor_cryptopp/osrng.h>
+#include <cryptopp/osrng.h>
 
 namespace cpputils {
     //TODO Test
diff -Naurp0 -U3 cryfs-0.11.2/src/cryfs/impl/localstate/BasedirMetadata.cpp cryfs-0.11.2-mine/src/cryfs/impl/localstate/BasedirMetadata.cpp
--- a/src/cryfs/impl/localstate/BasedirMetadata.cpp	2022-08-25 22:14:05.487898144 +0400
+++ b/src/cryfs/impl/localstate/BasedirMetadata.cpp	2022-08-25 22:19:46.722635560 +0400
@@ -1,7 +1,7 @@
 #include "BasedirMetadata.h"
 #include <boost/property_tree/ptree.hpp>
 #include <boost/property_tree/json_parser.hpp>
-#include <vendor_cryptopp/sha.h>
+#include <cryptopp/sha.h>
 #include <boost/filesystem/operations.hpp>
 #include "LocalStateDir.h"
 #include <cpp-utils/logging/logging.h>
--- a/test/cryfs/impl/config/CompatibilityTest.cpp	2022-08-25 22:14:05.493898126 +0400
+++ b/test/cryfs/impl/config/CompatibilityTest.cpp	2022-08-25 22:19:54.589701542 +0400
@@ -2,7 +2,7 @@
 #include <vector>
 #include <boost/filesystem.hpp>
 #include <cpp-utils/data/Data.h>
-#include <vendor_cryptopp/hex.h>
+#include <cryptopp/hex.h>
 #include <cpp-utils/crypto/symmetric/ciphers.h>
 #include <cpp-utils/tempfile/TempFile.h>
 #include <cryfs/impl/config/CryConfigFile.h>
--- a/vendor/CMakeLists.txt	2022-08-25 22:14:05.497898115 +0400
+++ b/vendor/CMakeLists.txt	2022-08-25 22:20:11.176840657 +0400
@@ -1,2 +1 @@
 add_subdirectory(googletest)
-add_subdirectory(cryptopp)
