Index: cryfs-0.10.2/test/cpp-utils/logging/testutils/LoggingTest.h
===================================================================
--- cryfs-0.10.2.orig/test/cpp-utils/logging/testutils/LoggingTest.h
+++ cryfs-0.10.2/test/cpp-utils/logging/testutils/LoggingTest.h
@@ -11,8 +11,7 @@ class MockLogger final {
 public:
     MockLogger():
             _capturedLogData(),
-            _sink(std::make_shared<spdlog::sinks::ostream_sink<std::mutex>>(_capturedLogData, true)),
-            _logger(spdlog::create("MockLogger", {_sink})) {
+            _logger(spdlog::create<spdlog::sinks::ostream_sink_mt>("MockLogger", _capturedLogData, true)) {
     }
 
     ~MockLogger() {
@@ -28,7 +27,6 @@ public:
     }
 private:
     std::ostringstream _capturedLogData;
-    std::shared_ptr<spdlog::sinks::ostream_sink<std::mutex>> _sink;
     std::shared_ptr<spdlog::logger> _logger;
 };
 
