--- a/util/nvramtool/accessors/cmos-hw-unix.c	2018-08-27 12:20:14.139599118 +0200
+++ b/util/nvramtool/accessors/cmos-hw-unix.c	2018-08-27 12:19:45.558280906 +0200
@@ -11,7 +11,7 @@
 #define INW(x) __extension__ ({ u_int tmp = (x); inw(tmp); })
 #define INL(x) __extension__ ({ u_int tmp = (x); inl(tmp); })
 #else
-#if defined(__GLIBC__) && (defined(__i386__) || defined(__x86_64__))
+#if (defined(__i386__) || defined(__x86_64__)) && !(defined(__FreeBSD__) || defined(__OpenBSD__) || defined(__NetBSD__))
 #include <sys/io.h>
 #endif
 #if (defined(__MACH__) && defined(__APPLE__))
