--- a/util/superiotool/Makefile	2018-05-16 21:00:17.000000000 +0200
+++ b/util/superiotool/Makefile	2018-08-27 12:56:28.070741476 +0200
@@ -25,7 +25,7 @@
 VERSION := -D'SUPERIOTOOL_VERSION="$(shell git describe 2>/dev/null)"'
 
 CFLAGS += -O2 -Wall -Wstrict-prototypes -Wundef -Wstrict-aliasing \
-         -Werror-implicit-function-declaration -ansi -pedantic $(VERSION)
+         -Werror-implicit-function-declaration -pedantic $(VERSION)
 LDFLAGS += -lz
 
 OBJS = superiotool.o serverengines.o ali.o exar.o fintek.o ite.o nsc.o \
--- a/util/superiotool/superiotool.h	2018-05-16 21:00:17.000000000 +0200
+++ b/util/superiotool/superiotool.h	2018-08-27 12:50:25.627387315 +0200
@@ -25,7 +25,7 @@
 #include <stdint.h>
 #include <string.h>
 #include <getopt.h>
-#if defined(__GLIBC__)
+#if  !(defined(__FreeBSD__) || defined(__OpenBSD__) || defined(__NetBSD__))
 #include <sys/io.h>
 #endif
 #if (defined(__MACH__) && defined(__APPLE__))
