From a87fdb41a476b59c53416f714b3e49f6dcefd6b5 Mon Sep 17 00:00:00 2001
From: Nathan Owens <ndowens04@gmail.com>
Date: Wed, 30 Oct 2019 21:55:13 -0500
Subject: [PATCH] libnet-structure-musl-fix

---
 include/libnet/libnet-structures.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/include/libnet/libnet-structures.h b/include/libnet/libnet-structures.h
index 6084caa..34fffc6 100644
--- a/include/libnet/libnet-structures.h
+++ b/include/libnet/libnet-structures.h
@@ -49,9 +49,9 @@ struct libnet_port_list_chain
 /* libnet statistics structure */
 struct libnet_stats
 {
-    __int64_t packets_sent;               /* packets sent */
-    __int64_t packet_errors;              /* packets errors */
-    __int64_t bytes_written;              /* bytes written */
+    int64_t packets_sent;               /* packets sent */
+    int64_t packet_errors;              /* packets errors */
+    int64_t bytes_written;              /* bytes written */
 };
 
 
-- 
2.23.0

