--- a/setup.py
+++ b/setup.py
@@ -5,29 +5,11 @@
 import os
 import sys
 
-try:
-    import pypandoc
-    PANDOC = True
-except ImportError:
-    PANDOC = False
 from pathlib import Path
 
-from share import makedoc
 from setuptools import setup
 
 
-makedoc()
-if not PANDOC:
-    print("Warning: 'pypandoc' is required to generate the manual pages.")
-else:
-    try:
-        for name in ['pass-import', 'pimport']:
-            pypandoc.convert_file(f"share/man/man1/{name}.md", 'man',
-                                  outputfile=f"share/man/man1/{name}.1",
-                                  extra_args=['--standalone'])
-    except OSError:
-        print("Warning: 'pandoc' is not installed.")
-
 
 share = Path(sys.prefix, 'share')
 base = '/usr'
