--- a/dbeaver/product/community/DBeaver.product
+++ b/dbeaver/product/community/DBeaver.product
@@ -9,7 +9,7 @@
     </configIni>

     <launcherArgs>
-        <programArgs></programArgs>
+        <programArgs>-vm /usr/lib/jvm/openjdk17/bin</programArgs>
         <programArgsLin>--launcher.additionalVmargs %XDG_DATA_HOME%/DBeaverData/settings/dbeaver.vmoptions %HOME%/.local/share/DBeaverData/settings/dbeaver.vmoptions</programArgsLin>
         <programArgsWin>--launcher.additionalVmargs %APPDATA%\DBeaverData\settings\dbeaver.exe.vmoptions</programArgsWin>
         <programArgsMac>--launcher.additionalVmargs %HOME%/Library/DBeaverData/settings/dbeaver.vmoptions -vm ../Eclipse/jre/Contents/Home/lib/libjli.dylib</programArgsMac>
