From: Đoàn Trần Công Danh <congdanhqx@gmail.com>
---
 configure.in | 2 ++
 1 file changed, 2 insertions(+)

diff --git configure.in configure.in
index aa6fd6e..ad8db32 100755
--- a/configure.in
+++ b/configure.in
@@ -9,6 +9,8 @@ TEA_INIT([3.9])
 
 AC_CONFIG_AUX_DIR(tclconfig)
 
+AC_PROG_CC([cc gcc])
+
 #--------------------------------------------------------------------
 # Configure script for package 'Expect'.
 # TEA compliant.
-- 
2.26.0.292.g33ef6b2f38

