--- a/inc/private.h
+++ b/inc/private.h
@@ -12,12 +12,12 @@
 #ifdef __cplusplus
  	extern "C" {
 #endif
+#include <stdint.h>
+typedef int16_t			word;		/* 16 bit signed int	*/
+typedef int32_t			longword;	/* 32 bit signed int	*/
 
-typedef short			word;		/* 16 bit signed int	*/
-typedef long			longword;	/* 32 bit signed int	*/
-
-typedef unsigned short		uword;		/* unsigned word	*/
-typedef unsigned long		ulongword;	/* unsigned longword	*/
+typedef uint16_t		uword;		/* unsigned word	*/
+typedef uint32_t		ulongword;	/* unsigned longword	*/
 
 struct gsm_state {
 
