--- a/inc/private.h	2006-04-26 22:17:01.000000000 +0300
+++ b/inc/private.h	2007-08-19 12:23:02.000000000 +0300
@@ -10,10 +10,10 @@
 #define	PRIVATE_H
 
 typedef short			word;		/* 16 bit signed int	*/
-typedef long			longword;	/* 32 bit signed int	*/
+typedef int			longword;	/* 32 bit signed int	*/
 
 typedef unsigned short		uword;		/* unsigned word	*/
-typedef unsigned long		ulongword;	/* unsigned longword	*/
+typedef unsigned int		ulongword;	/* unsigned longword	*/
 
 struct gsm_state {
 
