--- a/data/meson.build
+++ b/data/meson.build
@@ -2,7 +2,7 @@ subdir('icons')
 
 desktop_conf = configuration_data()
 desktop_conf.set('icon', application_id)
-desktop_file = i18n.merge_file ('desktop-file',
+desktop_file = i18n.merge_file (
   type: 'desktop',
   input: configure_file(
     input: files('com.gitlab.newsflash.desktop.in.in'),
@@ -27,7 +27,7 @@ endif
 
 appdata_conf = configuration_data()
 appdata_conf.set('appid', application_id)
-appdata_file = i18n.merge_file ('appdata-file',
+appdata_file = i18n.merge_file (
   input: configure_file(
     input: files('com.gitlab.newsflash.appdata.xml.in.in'),
     output: '@BASENAME@',
