Parallel cythonization throws a RuntimeError with this package.

--- a/setup.py
+++ b/setup.py
@@ -209,9 +209,6 @@
         # 3str is the default in Cython3 and we do not support older Cython releases.
         # we only set this to avoid the related FutureWarning from Cython3.
         cython_opts = dict(compiler_directives={"language_level": "3str"})
-        if not is_win32:
-            # compile .pyx extensions to .c in parallel, does not work on windows
-            cython_opts["nthreads"] = cpu_threads
 
         # generate C code from Cython for ALL supported platforms, so we have them in the sdist.
         # the sdist does not require Cython at install time, so we need all as C.
