--- dao/CdrDriver.cc	2009-09-12 18:21:25.000000000 +0200
+++ dao/CdrDriver.cc	2016-09-13 09:21:23.465964372 +0200
@@ -495,7 +495,7 @@
   0, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0
 };
 
-char CdrDriver::REMOTE_MSG_SYNC_[4] = { 0xff, 0x00, 0xff, 0x00 };
+unsigned char CdrDriver::REMOTE_MSG_SYNC_[4] = { 0xff, 0x00, 0xff, 0x00 };
 
 
 /* Maps a string to the corresponding driver option value 
--- dao/CdrDriver.h	2009-02-21 11:27:31.000000000 +0100
+++ dao/CdrDriver.h	2016-09-13 09:21:41.068972901 +0200
@@ -440,7 +440,7 @@
   bool fullBurn_;
 
   static unsigned char syncPattern[12];
-  static char REMOTE_MSG_SYNC_[4];
+  static unsigned char REMOTE_MSG_SYNC_[4];
 
   static int speed2Mult(int);
   static int mult2Speed(int);
