--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -10,7 +10,6 @@
 
 add_subdirectory(data)
 add_subdirectory(src)
-add_subdirectory(deps)
 
 add_subdirectory(linux)
 
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -82,8 +82,11 @@
 	server/servermain.cpp
 	)
 
+find_package(PkgConfig REQUIRED)
 find_package(Boost REQUIRED)
 find_package(OpenGL)
+find_package(tinyxml2)
+pkg_check_modules(lua lua5.2)
 add_subdirectory(raknet)
 add_subdirectory(blobnet)
 
@@ -103,7 +103,7 @@
 	add_definitions("-DBLOBBY_DATA_DIR=\"${CMAKE_INSTALL_PREFIX}/share/blobby\"")
 endif()
 
-set(BLOBBY_COMMON_LIBS lua::lua blobnet::blobnet tinyxml2::tinyxml2 PhysFS::PhysFS SDL2::SDL2main SDL2::SDL2 Boost::boost)
+set(BLOBBY_COMMON_LIBS lua5.2 blobnet::blobnet tinyxml2::tinyxml2 PhysFS::PhysFS SDL2::SDL2main SDL2::SDL2 Boost::boost)
 
 # other additional dependencies
 if (NOT WIN32)
--- a/src/FileRead.cpp
+++ b/src/FileRead.cpp
@@ -30,7 +30,7 @@
 #include <boost/algorithm/string.hpp>
 
 #include "tinyxml2.h"
-#include "lua.hpp"
+#include <lua5.2/lua.hpp>
 
 
 /* implementation */
--- a/src/IScriptableComponent.cpp
+++ b/src/IScriptableComponent.cpp
@@ -1,6 +1,6 @@
 #include "IScriptableComponent.h"
 
-#include "lua.hpp"
+#include <lua5.2/lua.hpp>
 
 #include "Global.h"
 #include "GameConstants.h"
--- a/src/GameLogic.cpp
+++ b/src/GameLogic.cpp
@@ -26,7 +26,7 @@
 #include <iostream>
 #include <utility>
 
-#include "lua.hpp"
+#include <lua5.2/lua.hpp>
 
 #include "FileRead.h"
 #include "GameLogicState.h"
--- a/src/ScriptedInputSource.cpp
+++ b/src/ScriptedInputSource.cpp
@@ -26,7 +26,7 @@
 
 #include <SDL.h>
 
-#include "lua.hpp"
+#include <lua5.2/lua.hpp>
 
 #include "DuelMatch.h"
 #include "DuelMatchState.h"
