Index: alure-1.2/CMakeLists.txt
===================================================================
--- alure-1.2.orig/CMakeLists.txt
+++ alure-1.2/CMakeLists.txt
@@ -375,8 +375,7 @@ IF(FLUIDSYNTH)
     ELSE(NOT FLUIDSYNTH_FOUND)
         SET(HAS_FLUIDSYNTH 1)
         LINK_DIRECTORIES(${FLUIDSYNTH_LIBRARY_DIRS})
-        SET_SOURCE_FILES_PROPERTIES(src/codec_fluidsynth.cpp PROPERTIES
-                                    COMPILE_FLAGS "${FLUIDSYNTH_CFLAGS}")
+	add_compile_options(${FLUIDSYNTH_CFLAGS})
     ENDIF(NOT FLUIDSYNTH_FOUND)
     IF(HAS_FLUIDSYNTH)
         SET(ALURE_OBJS ${ALURE_OBJS} src/codec_fluidsynth.cpp)
