diff --git src/main.c src/main.c
index 154f052..b44463c 100644
--- a/src/main.c
+++ b/src/main.c
@@ -290,10 +290,10 @@ static void print_help(char *program_name, int long_help)
 
 static void print_version()
 {
-    fprintf(stderr, "HTTPDirFS version " VERSION "\n");
+    fprintf(stderr, "HTTPDirFS version: VERSION");
     /* --------- Print off SSL engine version  --------- */
     curl_version_info_data *data = curl_version_info(CURLVERSION_NOW);
-    fprintf(stderr, "libcurl SSL engine: %s\n", data->ssl_version);
+    fprintf(stderr, "\nlibcurl SSL engine: %s\n", data->ssl_version);
 }
 
 static void print_long_help()
diff --git src/util.h src/util.h
index d7c7c15..4ae9ce8 100644
--- a/src/util.h
+++ b/src/util.h
@@ -24,7 +24,7 @@
 /**
  * \brief the default user agent string
  */
-#define DEFAULT_USER_AGENT "HTTPDirFS-" VERSION
+#define DEFAULT_USER_AGENT "HTTPDirFS-VERSION"
 
 
 
