--- a/Jambase
+++ b/Jambase
@@ -1037,15 +1037,6 @@
 		HOSTTYPE = $(JAMUNAME[1]) ;
 	}
 
-	if $(HOSTTYPE) = x86_64 
-	 || $(HOSTTYPE) = x86_64-linux
-	 || $(HOSTTYPE) = amd64 {
-		ECHO "We're on a 64 bit host" ;
-		HOST64     = true ;
-		TARGET64   = true ;			# We're not allowing for cross-compiling here...
-		CCFLAGS	   += -m64 ;
-		C++FLAGS   += -m64 ;
-	}


 	# Hmm. Newer linux gcc ar defaults to no timestamps
 	# and introduces a new option to get old behaviour,
