diff -u -r a/kernel/nvidia-uvm/uvm8_tools.c b/kernel/nvidia-uvm/uvm8_tools.c
--- a/kernel/nvidia-uvm/uvm8_tools.c	2019-01-27 16:12:22.000000000 +0000
+++ b/kernel/nvidia-uvm/uvm8_tools.c	2019-07-09 04:42:40.134899491 +0000
@@ -204,12 +204,14 @@
     return event_tracker != NULL && !event_tracker->is_queue;
 }
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 2, 0)
 static void put_user_pages(struct page **pages, NvU64 page_count)
 {
     NvU64 i;
     for (i = 0; i < page_count; i++)
         put_page(pages[i]);
 }
+#endif
 
 static void unmap_user_pages(struct page **pages, void *addr, NvU64 size)
 {
