From 6032c0fc836da4c757860dfc71d19ed2f64a56b1 Mon Sep 17 00:00:00 2001
From: Komeil Parseh <komeilparseh@disroot.org>
Date: Thu, 31 Jul 2025 11:20:59 +0330
Subject: [PATCH] ivykis provided by xbps

---
 autogen.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/autogen.sh b/autogen.sh
index 7b78333..7fab5b4 100755
--- a/autogen.sh
+++ b/autogen.sh
@@ -73,7 +73,7 @@ autogen_submodules()
 			autoreconf -i
 		else
 			echo "Don't know how to bootstrap submodule '$submod'" >&2
-			exit 1
+			# exit 1
 		fi
 
 		CONFIGURE_OPTS="--disable-shared --enable-static --with-pic"
-- 
2.50.1

