--- src/ck-manager.c.orig	2012-07-09 12:38:27.749426822 +0200
+++ src/ck-manager.c	2012-07-09 12:39:08.292465239 +0200
@@ -181,8 +181,8 @@ ck_manager_dump (CkManager *manager)
 {
         int         fd;
         int         res;
-        const char *filename = LOCALSTATEDIR "/run/ConsoleKit/database";
-        const char *filename_tmp = LOCALSTATEDIR "/run/ConsoleKit/database~";
+        const char *filename = "/run/ConsoleKit/database";
+        const char *filename_tmp = "/run/ConsoleKit/database~";
 
         if (manager == NULL) {
                 return;
@@ -190,11 +190,11 @@ ck_manager_dump (CkManager *manager)
 
         /* always make sure we have a directory */
         errno = 0;
-        res = g_mkdir_with_parents (LOCALSTATEDIR "/run/ConsoleKit",
+        res = g_mkdir_with_parents ("/run/ConsoleKit",
                                     S_IRWXU | S_IRGRP | S_IXGRP | S_IROTH | S_IXOTH);
         if (res < 0) {
                 g_warning ("Unable to create directory %s (%s)",
-                           LOCALSTATEDIR "/run/ConsoleKit",
+                           "/run/ConsoleKit",
                            g_strerror (errno));
                 return;
         }
