--- a/project/cmake/modules/FindCrossGUID.cmake.orig	2017-09-04 14:23:17.801957041 +0200
+++ b/project/cmake/modules/FindCrossGUID.cmake	2017-09-04 14:23:55.258390977 +0200
@@ -44,10 +44,10 @@
   set(CROSSGUID_LIBRARIES ${CROSSGUID_LIBRARY})
   set(CROSSGUID_INCLUDE_DIRS ${CROSSGUID_INCLUDE_DIR})
 else()
-  find_path(CROSSGUID_INCLUDE_DIR NAMES guid.h)
+  find_path(CROSSGUID_INCLUDE_DIR NAMES Guid.hpp)
 
-  find_library(CROSSGUID_LIBRARY_RELEASE NAMES crossguid)
-  find_library(CROSSGUID_LIBRARY_DEBUG NAMES crossguidd)
+  find_library(CROSSGUID_LIBRARY_RELEASE NAMES xg)
+  find_library(CROSSGUID_LIBRARY_DEBUG NAMES xg)
 
   include(SelectLibraryConfigurations)
   select_library_configurations(CROSSGUID)
--- a/xbmc/utils/StringUtils.cpp.orig	2017-09-04 14:39:15.802986978 +0200
+++ b/xbmc/utils/StringUtils.cpp	2017-09-04 14:41:00.026267037 +0200
@@ -28,7 +28,7 @@
 //
 //------------------------------------------------------------------------
 
-#include <guid.h>
+#include <Guid.hpp>
 
 #include "StringUtils.h"
 #include "CharsetConverter.h"
@@ -1191,8 +1191,7 @@
 
 std::string StringUtils::CreateUUID()
 {
-  static GuidGenerator guidGenerator;
-  auto guid = guidGenerator.newGuid();
+  auto guid = xg::newGuid();
 
   std::stringstream strGuid; strGuid << guid;
   return strGuid.str();
