--- a/tools/depends/target/ffmpeg/CMakeLists.txt
+++ b/tools/depends/target/ffmpeg/CMakeLists.txt
@@ -12,10 +12,11 @@
                 )
 endif()

-if(CROSSCOMPILING)
+if(CMAKE_CROSSCOMPILING)
   set(pkgconf "PKG_CONFIG_LIBDIR=${DEPENDS_PATH}/lib/pkgconfig")
   list(APPEND ffmpeg_conf --pkg-config=${PKG_CONFIG_EXECUTABLE}
                           --pkg-config-flags=--static
+                          --pkg-config-flags=--libs-only-l
                           --enable-cross-compile
                           --enable-pic
                           --ar=${CMAKE_AR}
