diff --git a/xbmc/cores/VideoPlayer/VideoRenderers/ColorManager.h b/xbmc/cores/VideoPlayer/VideoRenderers/ColorManager.h
index 3fae32c023..b30d1a615c 100644
--- a/xbmc/cores/VideoPlayer/VideoRenderers/ColorManager.h
+++ b/xbmc/cores/VideoPlayer/VideoRenderers/ColorManager.h
@@ -12,6 +12,7 @@
 #include <lcms2.h>
 #endif
 
+#include <cstdint>
 #include <string>
 
 extern "C"
diff --git a/xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/VaapiEGL.cpp b/xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/VaapiEGL.cpp
index 0342c5441a..264cf79cae 100644
--- a/xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/VaapiEGL.cpp
+++ b/xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/VaapiEGL.cpp
@@ -12,6 +12,7 @@
 #include "utils/EGLUtils.h"
 #include "utils/log.h"
 
+#include <cstdint>
 #include <drm_fourcc.h>
 #include <va/va_drmcommon.h>
 
diff --git a/xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/VaapiEGL.h b/xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/VaapiEGL.h
index 66d1bf0200..10d54f3d7a 100644
--- a/xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/VaapiEGL.h
+++ b/xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/VaapiEGL.h
@@ -8,6 +8,7 @@
 
 #pragma once
 
+#include <cstdint>
 #include <array>
 
 #if defined(HAS_GL)
diff --git a/xbmc/pictures/Picture.h b/xbmc/pictures/Picture.h
index 5495cabc53..3017628b32 100644
--- a/xbmc/pictures/Picture.h
+++ b/xbmc/pictures/Picture.h
@@ -11,6 +11,7 @@
 #include "pictures/PictureScalingAlgorithm.h"
 #include "utils/Job.h"
 
+#include <cstdint>
 #include <string>
 #include <vector>
 
diff --git a/xbmc/platform/Filesystem.h b/xbmc/platform/Filesystem.h
index 3e6b7ec52b..732be9294c 100644
--- a/xbmc/platform/Filesystem.h
+++ b/xbmc/platform/Filesystem.h
@@ -8,6 +8,7 @@
 
 #pragma once
 
+#include <cstdint>
 #include <string>
 #include <system_error>
 namespace KODI
diff --git a/xbmc/windowing/X11/GLContext.h b/xbmc/windowing/X11/GLContext.h
index e3c68807b8..1082fb0236 100644
--- a/xbmc/windowing/X11/GLContext.h
+++ b/xbmc/windowing/X11/GLContext.h
@@ -8,6 +8,7 @@
 
 #pragma once
 
+#include <cstdint>
 #include <string>
 
 #include <X11/Xlib.h>
diff --git a/xbmc/windowing/wayland/WinSystemWayland.cpp b/xbmc/windowing/wayland/WinSystemWayland.cpp
index 6d07197bc7..3cace0954b 100644
--- a/xbmc/windowing/wayland/WinSystemWayland.cpp
+++ b/xbmc/windowing/wayland/WinSystemWayland.cpp
@@ -8,6 +8,8 @@
 
 #include "WinSystemWayland.h"
 
+#include <cstdint>
+
 #include "Application.h"
 #include "CompileInfo.h"
 #include "Connection.h"
diff --git a/xbmc/windowing/wayland/WinSystemWayland.h b/xbmc/windowing/wayland/WinSystemWayland.h
index 9f93d4bd46..e7caf6c227 100644
--- a/xbmc/windowing/wayland/WinSystemWayland.h
+++ b/xbmc/windowing/wayland/WinSystemWayland.h
@@ -8,6 +8,8 @@
 
 #pragma once
 
+#include <cstdint>
+
 #include "Connection.h"
 #include "Output.h"
 #include "Seat.h"
