diff --git a/xbmc/windowing/X11/WinSystemX11.cpp b/xbmc-18.7.1-Leia/xbmc/windowing/X11/WinSystemX11.cpp
index 8b0c56579b..5db641efbd 100644
--- a/xbmc/windowing/X11/WinSystemX11.cpp
+++ b/xbmc/windowing/X11/WinSystemX11.cpp
@@ -26,6 +26,7 @@
 #include "messaging/ApplicationMessenger.h"
 #include <X11/Xatom.h>
 #include <X11/extensions/Xrandr.h>
+#include "Application.h"
 
 #include "WinEventsX11.h"
 #include "input/InputManager.h"
@@ -557,8 +558,10 @@ void CWinSystemX11::NotifyXRREvent()
   {
     UpdateResolutions();
   }
-
+  if (!g_application.IsDPMSActive())
+  {
     RecreateWindow();
+  }
 }
 
 void CWinSystemX11::RecreateWindow()
