diff -ur a/cmake/scripts/common/ArchSetup.cmake b/cmake/scripts/common/ArchSetup.cmake
--- a/cmake/scripts/common/ArchSetup.cmake	2024-08-17 03:06:48.000000000 -0400
+++ b/cmake/scripts/common/ArchSetup.cmake	2024-11-23 18:01:32.568554400 -0500
@@ -175,15 +175,6 @@
     -g
     -D_DEBUG
   )
-
-  # these options affect only core code
-  if(NOT CORE_COMPILE_OPTIONS)
-    set(CORE_COMPILE_OPTIONS
-      -Werror=double-promotion
-      -Werror=missing-field-initializers
-      -Werror=sign-compare
-    )
-  endif()
 endif()
 
 # set for compile info to help detect binary addons
