--- a/cmake/scripts/linux/ArchSetup.cmake	2019-01-30 21:03:03.146025973 +0100
+++ b/cmake/scripts/linux/ArchSetup.cmake	2019-01-30 21:03:47.810441038 +0100
@@ -35,6 +35,12 @@
   elseif(CPU MATCHES riscv64)
     set(ARCH riscv64)
     set(NEON False)
+  elseif(CPU MATCHES ppc64le)
+    set(ARCH ppc64le-linux)
+    set(NEON False)
+  elseif(CPU MATCHES ppc64)
+    set(ARCH ppc64-linux)
+    set(NEON False)
   else()
     message(SEND_ERROR "Unknown CPU: ${CPU}")
   endif()
