--- a/xbmc/platform/posix/utils/PosixInterfaceForCLog.cpp	2019-01-30 19:51:15.744119125 +0100
+++ b/xbmc/platform/posix/utils/PosixInterfaceForCLog.cpp	2019-01-30 19:54:51.940377340 +0100
@@ -17,10 +17,6 @@
 #include "platform/android/activity/XBMCApp.h"
 #endif // TARGET_ANDROID
 
-struct FILEWRAP : public FILE
-{};
-
-
 CPosixInterfaceForCLog::CPosixInterfaceForCLog() :
   m_file(NULL)
 { }
@@ -40,7 +36,7 @@
   (void)remove(backupOldLogToFilename.c_str()); // if it's failed, try to continue
   (void)rename(logFilename.c_str(), backupOldLogToFilename.c_str()); // if it's failed, try to continue
 
-  m_file = (FILEWRAP*)fopen(logFilename.c_str(), "wb");
+  m_file = fopen(logFilename.c_str(), "wb");
   if (!m_file)
     return false; // error, can't open log file
 
--- a/xbmc/platform/posix/utils/PosixInterfaceForCLog.h	2019-01-30 19:55:21.460002043 +0100
+++ b/xbmc/platform/posix/utils/PosixInterfaceForCLog.h	2019-01-30 19:55:36.156814572 +0100
@@ -10,8 +10,6 @@
 
 #include <string>
 
-struct FILEWRAP; // forward declaration, wrapper for FILE
-
 class CPosixInterfaceForCLog
 {
 public:
@@ -23,5 +21,5 @@
   void PrintDebugString(const std::string& debugString);
   static void GetCurrentLocalTime(int& hour, int& minute, int& second, double& millisecond);
 private:
-  FILEWRAP* m_file;
+  FILE* m_file;
 };
