--- ./xbmc/threads/platform/pthreads/ThreadImpl.cpp.orig
+++ ./xbmc/threads/platform/pthreads/ThreadImpl.cpp
@@ -42,6 +42,7 @@
 {
   pthread_attr_t attr;
   pthread_attr_init(&attr);
+  if (!stacksize) stacksize = 1024*1024;
 #if !defined(TARGET_ANDROID) // http://code.google.com/p/android/issues/detail?id=7808
   if (stacksize > PTHREAD_STACK_MIN)
     pthread_attr_setstacksize(&attr, stacksize);
