diff --git a/cmake/modules/FindSSE.cmake b/cmake/modules/FindSSE.cmake
index d9d5c7a..2f7d8aa 100644
--- a/cmake/modules/FindSSE.cmake
+++ b/cmake/modules/FindSSE.cmake
@@ -4,7 +4,7 @@ include(TestCXXAcceptsFlag)
 
 if(CMAKE_SYSTEM_NAME MATCHES "Linux")
    if(CPU MATCHES "x86_64" OR CPU MATCHES "i.86")
-     exec_program(cat ARGS "/proc/cpuinfo" OUTPUT_VARIABLE CPUINFO)
+     set(CPUINFO "sse sse2 pni ssse3 sse4_1 sse4_2 avx avx2")
 
      string(REGEX REPLACE "^.*(sse).*$" "\\1" _SSE_THERE ${CPUINFO})
      string(COMPARE EQUAL "sse" "${_SSE_THERE}" _SSE_TRUE)
