Our gpgme package (>= 1.6.0) expects _FILE_OFFSET_BITS = 64
to be defined when gpgme.h is included.

--- kwalletd/backend/backendpersisthandler.cpp	2014-10-31 00:50:20.000000000 +0100
+++ kwalletd/backend/backendpersisthandler.cpp	2015-10-09 01:11:46.115305255 +0200
@@ -26,6 +26,8 @@
 #include <kmessagebox.h>
 #include <klocalizedstring.h>
 #ifdef HAVE_QGPGME
+#undef	_FILE_OFFSET_BITS
+#define	_FILE_OFFSET_BITS 64
 #include <gpgme.h>
 #include <gpgme++/context.h>
 #include <gpgme++/key.h>
--- kwalletd/knewwalletdialog.cpp	2014-10-31 00:50:20.000000000 +0100
+++ kwalletd/knewwalletdialog.cpp	2015-10-09 01:17:44.690277279 +0200
@@ -24,6 +24,8 @@
 #include <QTimer>
 #include <QTableWidget>
 #include <QTableWidgetItem>
+#undef	_FILE_OFFSET_BITS
+#define	_FILE_OFFSET_BITS 64
 #include <gpgme.h>
 #include <gpgme++/context.h>
 #include <gpgme++/key.h>
