--- a/data/lightdm-webkit2-greeter.conf	2018-06-26 15:29:27.461313219 +0200
+++ b/data/lightdm-webkit2-greeter.conf	2018-06-26 18:58:13.649378103 +0200
@@ -18,7 +18,7 @@
 secure_mode         = true
 time_format         = LT
 time_language       = auto
-webkit_theme        = antergos
+webkit_theme        = void
 
 #
 # [branding]
@@ -31,5 +31,6 @@
 
 [branding]
 background_images = /usr/share/backgrounds
-logo              = /usr/share/lightdm-webkit/themes/antergos/img/antergos.png
-user_image        = /usr/share/lightdm-webkit/themes/antergos/img/antergos-logo-user.png
+logo              = /usr/share/void-artwork/void-transparent.png
+user_image        = /usr/share/void-artwork/void-transparent.png
+
