From 1dadb93567e8607611979c515e8b6152cd36f3f3 Mon Sep 17 00:00:00 2001
From: Xiao-Long Chen <chenxiaolong@cxl.epac.to>
Date: Fri, 11 Apr 2014 15:46:47 -0400
Subject: [PATCH 2/6] Remove code depending on accountsservices patch

---
 lib/main.vala | 38 +-------------------------------------
 1 file changed, 1 insertion(+), 37 deletions(-)

diff --git a/lib/main.vala b/lib/main.vala
index fbaf5a1..eab8d1c 100644
--- a/lib/main.vala
+++ b/lib/main.vala
@@ -305,25 +305,7 @@ public class Indicator.Keyboard.Service : Object {
 				Act.User? user = manager.get_user ((!) greeter_user);
 
 				if (user != null && ((!) user).is_loaded) {
-					foreach (var outer in ((!) user).input_sources) {
-						foreach (var inner in (!) outer) {
-							unowned string key;
-							unowned string value;
-
-							((!) inner).get ("{&s&s}", out key, out value);
-
-							if (key == "xkb") {
-								source = value;
-								break;
-							}
-						}
-
-						if (source != null) {
-							break;
-						}
-					}
-
-					if (source == null) {
+					if (true) {
 						var layouts = ((!) user).xkeyboard_layouts;
 
 						if (layouts.length <= 0) {
@@ -495,24 +477,6 @@ public class Indicator.Keyboard.Service : Object {
 			if (user.is_loaded) {
 				var done = false;
 
-				foreach (var outer in user.input_sources) {
-					foreach (var inner in (!) outer) {
-						unowned string key;
-						unowned string source;
-
-						((!) inner).get ("{&s&s}", out key, out source);
-
-						if (key == "xkb") {
-							done = true;
-
-							if (!added.contains (source)) {
-								list.add (source);
-								added.add (source);
-							}
-						}
-					}
-				}
-
 				if (!done) {
 					var layouts = user.xkeyboard_layouts;
 					foreach (var layout in layouts) {
-- 
2.9.3

