From 41e1b2613e6c089a8f893cc44650e2bb59586ba4 Mon Sep 17 00:00:00 2001
From: Xiao-Long Chen <chenxiaolong@cxl.epac.to>
Date: Wed, 16 Apr 2014 17:06:46 -0400
Subject: [PATCH 4/6] Remove unavailable DBus method xkeyboard-layouts

---
 lib/main.vala | 20 ++------------------
 1 file changed, 2 insertions(+), 18 deletions(-)

diff --git a/lib/main.vala b/lib/main.vala
index d230d16..aa61a22 100644
--- a/lib/main.vala
+++ b/lib/main.vala
@@ -299,9 +299,9 @@ public class Indicator.Keyboard.Service : Object {
 
 				if (user != null && ((!) user).is_loaded) {
 					if (true) {
-						var layouts = ((!) user).xkeyboard_layouts;
+						string[] layouts = null;
 
-						if (layouts.length <= 0) {
+						if (true) {
 							var user_list = LightDM.UserList.get_instance ();
 							LightDM.User? light_user = user_list.get_user_by_name ((!) greeter_user);
 
@@ -471,22 +471,6 @@ public class Indicator.Keyboard.Service : Object {
 				var done = false;
 
 				if (!done) {
-					var layouts = user.xkeyboard_layouts;
-					foreach (var layout in layouts) {
-						done = true;
-
-						var source = layout;
-						source = source.replace (" ", "+");
-						source = source.replace ("\t", "+");
-
-						if (!added.contains (source)) {
-							list.add (source);
-							added.add (source);
-						}
-					}
-				}
-
-				if (!done) {
 					var user_list = LightDM.UserList.get_instance ();
 					LightDM.User? light_user = user_list.get_user_by_name (user.user_name);
 
-- 
2.9.3

