--- /src/ifcblender/io_import_scene_ifc/__init__.py
+++ /src/ifcblender/io_import_scene_ifc/__init__.py
@@ -64,8 +64,8 @@
 
 
 def import_ifc(filename, use_names, process_relations, blender_booleans):
-    from . import ifcopenshell
-    from .ifcopenshell import geom as ifcopenshell_geom
+    import ifcopenshell
+    from ifcopenshell import geom as ifcopenshell_geom
     print(f"Reading {bpy.path.basename(filename)}...")
     settings = ifcopenshell_geom.settings()
     settings.set(settings.DISABLE_OPENING_SUBTRACTIONS, blender_booleans)
