--- /src/ifcgeom/IfcGeomFunctions.cpp
+++ /src/ifcgeom/IfcGeomFunctions.cpp
@@ -3645,10 +3645,9 @@
 			if (i == n - 1 && j == 0) continue;
 			
 			double u11, u12, u21, u22, U1, U2;
-			GeomAPI_ExtremaCurveCurve ecc(
-				BRep_Tool::Curve(wd->Edge(i + 1), u11, u12),
-				BRep_Tool::Curve(wd->Edge(j + 1), u21, u22)
-			);
+      thread_local Handle(Geom_Curve) c0 = BRep_Tool::Curve(wd->Edge(i + 1), u11, u12);
+      thread_local Handle(Geom_Curve) c1 = BRep_Tool::Curve(wd->Edge(j + 1), u21, u22);
+      GeomAPI_ExtremaCurveCurve ecc(c0, c1);
 
 			// @todo: extend this to work in case of multiple extrema and curved segments.
 			const bool unbounded_intersects = (ecc.NbExtrema() == 1 && ecc.Distance(1) < eps);
@@ -4227,4 +4226,4 @@
 	if (loops_removed || (non_manifold && l->declaration().is(IfcSchema::IfcClosedShell::Class()))) {
 		Logger::Warning(boost::lexical_cast<std::string>(duplicate_faces) + " duplicate faces removed, " + boost::lexical_cast<std::string>(loops_removed) + " loops removed and " + boost::lexical_cast<std::string>(non_manifold) + " non-manifold edges for:", l);
 	}
-}
\ No newline at end of file
+}
