#inspired by alpine
#upstream: no

--- open-vm-tools/lib/include/guest_os.h
+++ open-vm-tools/lib/include/guest_os.h
@@ -241,6 +241,7 @@ Bool Gos_InSetArray(uint32 gos, const uint32 *set);
 #define STR_OS_AMAZON_LINUX        "amazonlinux"
 #define STR_OS_ANNVIX              "Annvix"
 #define STR_OS_ARCH                "Arch"
+#define STR_OS_VOID                "Void Linux"
 #define STR_OS_ARKLINUX            "Arklinux"
 #define STR_OS_ASIANUX_3           "asianux3"
 #define STR_OS_ASIANUX_4           "asianux4"
--- open-vm-tools/lib/misc/hostinfoPosix.c
+++ open-vm-tools/lib/misc/hostinfoPosix.c
@@ -902,6 +902,8 @@ HostinfoGetOSShortName(const char *distro,      // IN: full distro name
 
       Str_Sprintf(distroShort, distroShortSize, "%s%d", STR_OS_AMAZON_LINUX,
                   version);
+	} else if (strstr(distroLower, "void")) {
+		Str_Strcpy(distroShort, STR_OS_VOID, distroShortSize);
    } else if (strstr(distroLower, "annvix")) {
       Str_Strcpy(distroShort, STR_OS_ANNVIX, distroShortSize);
    } else if (strstr(distroLower, "arch")) {
