#inspired by alpine
#upstream: no

--- a/open-vm-tools/lib/include/guest_os.h
+++ b/open-vm-tools/lib/include/guest_os.h
@@ -309,6 +309,7 @@ Bool Gos_InSetArray(uint32 gos, const uint32 *set);
 #define STR_OS_ULTRAPENGUIN       "UltraPenguin"
 #define STR_OS_UNITEDLINUX        "UnitedLinux"
 #define STR_OS_VALINUX            "VALinux"
+#define STR_OS_VOID               "Void Linux"
 #define STR_OS_YELLOW_DOG         "Yellow Dog"
 #define STR_OS_ECOMSTATION        "eComStation"
 
diff --git a/open-vm-tools/lib/misc/hostinfoPosix.c b/open-vm-tools/lib/misc/hostinfoPosix.c
index 6ac15412..ea898e24 100644
--- a/open-vm-tools/lib/misc/hostinfoPosix.c
+++ b/open-vm-tools/lib/misc/hostinfoPosix.c
@@ -1259,6 +1259,7 @@ static const ShortNameSet shortNameArray[] = {
 {"united linux",        STR_OS_UNITEDLINUX,        HostinfoGenericSetShortName},
 {"va linux",            STR_OS_VALINUX,            HostinfoGenericSetShortName},
 {"vmware photon",       STR_OS_PHOTON,             HostinfoGenericSetShortName},
+{"void",                STR_OS_VOID,               HostinfoGenericSetShortName},
 {"yellow dog",          STR_OS_YELLOW_DOG,         HostinfoGenericSetShortName},
 {NULL,                  NULL,                      NULL} // MUST BE LAST
 };
