diff --git a/CMakeLists.txt b/CMakeLists.txt
index 87a6692..e48c81b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -14,9 +14,9 @@ if(NOT CMAKE_CONFIGURATION_TYPES)
   set(CMAKE_BUILD_TYPE RELWITHDEBINFO CACHE STRING "" FORCE)
   else()
     string(TOUPPER ${CMAKE_BUILD_TYPE} CMAKE_BUILD_TYPE)
-    if(NOT CMAKE_BUILD_TYPE IN_LIST allowableBuildTypes)
-      message(FATAL_ERROR "Invalid build type: ${CMAKE_BUILD_TYPE}")
-    endif()
+    # if(NOT CMAKE_BUILD_TYPE IN_LIST allowableBuildTypes)
+    #   message(FATAL_ERROR "Invalid build type: ${CMAKE_BUILD_TYPE}")
+    # endif()
   endif()
 endif()
 
