diff --git a/configure.in b/configure.in
index b7f4f84..130f54a 100644
--- a/configure.in
+++ b/configure.in
@@ -102,10 +102,10 @@ case "$target" in
         ;;
     *-*-beos*)
         ac_default_prefix=/boot/beos
-        GAME_INSTALLDIR="\$(prefix)/games/$PACKAGE"
+        GAME_INSTALLDIR="\$(prefix)/$PACKAGE"
         ;;
     *)
-        GAME_INSTALLDIR="\$(prefix)/games/$PACKAGE"
+        GAME_INSTALLDIR="\$(prefix)/$PACKAGE"
         ;;
 esac
 AC_SUBST(GAME_INSTALLDIR)
