
--- Makefile.in	2002-10-20 00:19:42.000000000 -0400
+++ Makefile.in	2009-04-14 14:45:59.000000000 -0400
@@ -431,7 +431,7 @@
 
 # Special install rule for the game
 install:
-	make install_gamedata target=@GAME_INSTALLDIR@
+	make install_gamedata target=$(DESTDIR)/@GAME_INSTALLDIR@
 
 install_gamedata:
 	sh mkinstalldirs $(target)/
@@ -444,8 +444,8 @@
 	chmod 666 $(target)/Maelstrom-Scores
 
 install_gamedocs:
-	sh mkinstalldirs $(target)/
-	cp -rv README* COPYING* Docs*	$(target)/
+	sh mkinstalldirs $(DESTDIR)/$(target)/
+	cp -rv README* COPYING* Docs*	$(DESTDIR)/$(target)/
 
 # Rule to build tar-gzipped distribution package
 $(PACKAGE)-$(VERSION).tar.gz: dist

--- configure	2002-10-19 23:43:07.000000000 -0400
+++ configure	2009-04-14 14:47:33.000000000 -0400
@@ -3232,7 +3232,7 @@
         GAME_INSTALLDIR="\$(prefix)/games/$PACKAGE"
         ;;
     *)
-        GAME_INSTALLDIR="\$(prefix)/games/$PACKAGE"
+        GAME_INSTALLDIR="\$(prefix)/share/$PACKAGE"
         ;;
 esac
 
