--- bin/yarn.js.orig	2017-09-14 18:50:29.701379167 -0500
+++ bin/yarn.js	2017-09-14 17:27:32.749746288 -0500
@@ -11,7 +11,7 @@
   console.error('Node version ' + ver + ' is not supported, please use Node.js 4.0 or higher.');
   process.exitCode = 1;
 } else {
-  var dirPath = '../lib/';
+  var dirPath = '/usr/lib/yarn/';
   var v8CompileCachePath = dirPath + 'v8-compile-cache';
   var fs = require('fs');
   // We don't have/need this on legacy builds and dev builds
