--- a/Makefile.in	2015-02-05 08:25:13.491988672 +0100
+++ b/Makefile.in	2015-02-05 08:25:51.774946087 +0100
@@ -19,7 +19,9 @@ ${PROG}: ${OBJS}
 	${CC} ${OBJS} -o ${PROG}
 
 install:
-	install -c -s ${PROG} ${DESTDIR}${PREFIX}/bin
+	install -d ${DESTDIR}/${PREFIX}/bin
+	install -c ${PROG} ${DESTDIR}${PREFIX}/bin
+	install -d ${DESTDIR}/${MANDIR}/man1
 	install -c netpgpverify.1 ${DESTDIR}${MANDIR}/man1
 
 tst:
