--- Makefile.in.orig	2009-03-28 14:36:17.871227905 +0100
+++ Makefile.in	2009-03-28 14:36:52.480413711 +0100
@@ -88,8 +88,8 @@ check test tests: newsyslog t/runtests
 	cd t && ./runtests TESTS
 
 install: newsyslog newsyslog.8
-	$(INSTALL) -m 0755 newsyslog $(sbindir)
-	$(INSTALL) -m 0644 newsyslog.8 $(mandir)/man8
+	$(INSTALL) -D -m 0755 newsyslog $(DESTDIR)/$(sbindir)/newsyslog
+	$(INSTALL) -D -m 0644 newsyslog.8 $(DESTDIR)/$(mandir)/man8/newsyslog.8
 
 clean:
 	rm -f *.o t/*.o core newsyslog newsyslog.pure .pure config.log \
