--- configure.old	2014-06-11 11:45:38.108575635 +0200
+++ configure	2014-06-11 11:45:46.907574976 +0200
@@ -1885,7 +1885,7 @@ echo "configure:1885: checking for $ac_f
 if eval "test \"`echo '$''{'ac_cv_file_$ac_safe'+set}'`\" = set"; then
   echo $ac_n "(cached) $ac_c" 1>&6
 else
-  if test "$cross_compiling" = yes; then
+  if test "$cross_compiling" = XXyes; then
     { echo "configure: error: Cannot check for file existence when cross compiling" 1>&2; exit 1; }
 else
   if test -r $ac_file; then
@@ -1971,7 +1971,7 @@ echo "configure:1971: checking for $ac_f
 if eval "test \"`echo '$''{'ac_cv_file_$ac_safe'+set}'`\" = set"; then
   echo $ac_n "(cached) $ac_c" 1>&6
 else
-  if test "$cross_compiling" = yes; then
+  if test "$cross_compiling" = XXyes; then
     { echo "configure: error: Cannot check for file existence when cross compiling" 1>&2; exit 1; }
 else
   if test -r $ac_file; then
