From 06eea47224a6015a48f592951298b1d80489452d Mon Sep 17 00:00:00 2001
From: Cameron Nemo <cnemo@tutanota.com>
Date: Thu, 12 Sep 2019 23:05:51 -0700
Subject: [PATCH] fix(vfs): include pthread.h for musl compat

Signed-off-by: Cameron Nemo <cnemo@tutanota.com>
---
 src/vfs.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/vfs.c b/src/vfs.c
index 31e138c..5d2a7c4 100644
--- src/vfs.c
+++ src/vfs.c
@@ -1,6 +1,7 @@
 #include <string.h>
 #include <sys/time.h>
 #include <errno.h>
+#include <pthread.h>
 
 #include <raft.h>
 
