diff --git a/cmd/rc/interactive.ha b/cmd/rc/interactive.ha
index 1ab4003..ef3c6a0 100644
--- a/cmd/rc/interactive.ha
+++ b/cmd/rc/interactive.ha
@@ -24,13 +24,13 @@ type prompter = struct {
 	prompt: str,
 };
 
-fn prompt_get(p: *made::prompter, buf: []u8, pos: size) (str, str) = {
+fn prompt_get(p: *made::prompter, s: *made::state) (str, str) = {
 	let p = p: *prompter;
 	let loc = p.parser.lex.loc;
 	defer p.parser.lex.loc = loc;
 	io::seek(p.in, 0, io::whence::SET)!;
 	memio::reset(p.in);
-	io::writeall(p.in, buf)!;
+	io::writeall(p.in, s.buf)!;
 	io::seek(p.in, 0, io::whence::SET)!;
 
 	let post = match (parse::parse_commands(p.parser)) {
@@ -67,6 +67,12 @@ fn interactive(state: *interp::state) void = {
 		hist = &hist,
 		prompt = &p,
 		split = made::split_sh,
+		cfg = match (made::config_default("example")) {
+		case let cfg: made::config =>
+			yield cfg;
+		case let e: made::error =>
+			fmt::fatal(made::strerror(e));
+		},
 		...
 	};
 	defer made::ctx_finish(&ctx)!;
