diff --git a/src/widgets/freespacebar.cpp b/src/widgets/freespacebar.cpp
index 3067f72..c9789d3 100644
--- a/src/widgets/freespacebar.cpp
+++ b/src/widgets/freespacebar.cpp
@@ -20,6 +20,7 @@
 
 #include <QLinearGradient>
 #include <QPainter>
+#include <QPainterPath>
 
 const int FreeSpaceBar::kBarHeight = 20;
 const int FreeSpaceBar::kBarBorderRadius = 8;
diff --git a/src/widgets/osdpretty.cpp b/src/widgets/osdpretty.cpp
index 9d95716..30cef7d 100644
--- a/src/widgets/osdpretty.cpp
+++ b/src/widgets/osdpretty.cpp
@@ -26,6 +26,7 @@
 #include <QLayout>
 #include <QMouseEvent>
 #include <QPainter>
+#include <QPainterPath>
 #include <QSettings>
 #include <QTimer>
 #include <QTimeLine>
diff --git a/src/widgets/sliderwidget.cpp b/src/widgets/sliderwidget.cpp
index 581a0de..27b81f7 100644
--- a/src/widgets/sliderwidget.cpp
+++ b/src/widgets/sliderwidget.cpp
@@ -24,6 +24,7 @@
 #include <QBrush>
 #include <QImage>
 #include <QPainter>
+#include <QPainterPath>
 #include <QSize>
 #include <QTimer>
 #include <QStyle>
