--- a/gstenginepipeline.cpp	2017-09-22 23:13:31.520359890 +0200
+++ b/src/engines/gstenginepipeline.cpp	2017-09-22 23:14:31.837362189 +0200
@@ -375,7 +375,7 @@
   // Link the elements with special caps
   // The scope path through the tee gets 16-bit ints.
   GstCaps* caps16 = gst_caps_new_simple("audio/x-raw", "format", G_TYPE_STRING,
-                                        "S16LE", NULL);
+                                        "S16LE", (char*)0);
   gst_element_link_filtered(probe_converter, probe_sink, caps16);
   gst_caps_unref(caps16);
 
--- a/chromaprinter.cpp	2017-09-22 23:18:59.324372383 +0200
+++ b/src/musicbrainz/chromaprinter.cpp	2017-09-22 23:19:35.520373762 +0200
@@ -78,7 +78,7 @@
   // Chromaprint expects mono 16-bit ints at a sample rate of 11025Hz.
   GstCaps* caps = gst_caps_new_simple(
       "audio/x-raw", "format", G_TYPE_STRING, "S16LE", "channels", G_TYPE_INT,
-      kDecodeChannels, "rate", G_TYPE_INT, kDecodeRate, NULL);
+      kDecodeChannels, "rate", G_TYPE_INT, kDecodeRate, (char*)0);
   gst_element_link_filtered(resample, sink, caps);
   gst_caps_unref(caps);
 
--- a/cddasongloader.cpp	2017-09-22 23:25:22.975387004 +0200
+++ b/src/devices/cddasongloader.cpp	2017-09-22 23:25:59.774388406 +0200
@@ -62,7 +62,7 @@
                  nullptr);
   }
   if (g_object_class_find_property (G_OBJECT_GET_CLASS (cdda_), "paranoia-mode")) {
-    g_object_set (cdda_, "paranoia-mode", 0, NULL);
+    g_object_set (cdda_, "paranoia-mode", 0, (char*)0);
   }
 
   // Change the element's state to ready and paused, to be able to query it
@@ -106,7 +106,7 @@
 
   GstElement* pipeline = gst_pipeline_new("pipeline");
   GstElement* sink = gst_element_factory_make ("fakesink", NULL);
-  gst_bin_add_many (GST_BIN (pipeline), cdda_, sink, NULL);
+  gst_bin_add_many (GST_BIN (pipeline), cdda_, sink, (char*)0);
   gst_element_link (cdda_, sink);
   gst_element_set_state(pipeline, GST_STATE_READY);
   gst_element_set_state(pipeline, GST_STATE_PAUSED);
