--- a/src/libical-glib/CMakeLists.txt
+++ b/src/libical-glib/CMakeLists.txt
@@ -3,7 +3,7 @@ add_definitions(-Dlibical_ical_EXPORTS)
 # build ical-glib-src-generator
 add_custom_command(
   OUTPUT ${EXECUTABLE_OUTPUT_PATH}/ical-glib-src-generator
-  COMMAND /usr/bin/cc ${CMAKE_CURRENT_SOURCE_DIR}/tools/generator.c ${CMAKE_CURRENT_SOURCE_DIR}/tools/xml-parser.c -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/libxml2 -I${CMAKE_CURRENT_SOURCE_DIR}/tools/ -lxml2 -lglib-2.0 -o ${EXECUTABLE_OUTPUT_PATH}/ical-glib-src-generator
+  COMMAND /usr/bin/cc ${CMAKE_CURRENT_SOURCE_DIR}/tools/generator.c ${CMAKE_CURRENT_SOURCE_DIR}/tools/xml-parser.c -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/libxml2 -I${CMAKE_CURRENT_SOURCE_DIR}/tools/ -l:libxml2.so -lglib-2.0 -o ${EXECUTABLE_OUTPUT_PATH}/ical-glib-src-generator
 )
 
 list(APPEND API_FILES
