--- a/EternalTerminal/CMakeLists.txt
+++ b/EternalTerminal/CMakeLists.txt
@@ -519,17 +519,8 @@
       ${CORE_LIBRARIES})
   decorate_target(htmd)
 
-  enable_testing()
-
-  file(GLOB TEST_SRCS test/*Test.cpp)
-  add_executable(
-  et-test
-  ${TEST_SRCS}
-  test/Main.cpp
-  )
-  add_dependencies(et-test TerminalCommon et-lib)
+  add_dependencies(TerminalCommon et-lib)
   target_link_libraries(
-    et-test
     TerminalCommon
     et-lib
     ${CMAKE_THREAD_LIBS_INIT}
@@ -539,8 +530,6 @@
     ${UTEMPTER_LIBRARIES}
     ${Boost_LIBRARIES}
       ${CORE_LIBRARIES})
-  add_test(et-test et-test)
-  decorate_target(et-test)
 
   if(FUZZING)
     add_executable(
