Index: EternalTerminal-et-v6.0.13/CMakeLists.txt
===================================================================
--- a/EternalTerminal/CMakeLists.txt
+++ b/EternalTerminal/CMakeLists.txt
@@ -187,12 +187,6 @@
       "${CMAKE_CXX_FLAGS} -DELPP_DISABLE_DEFAULT_CRASH_HANDLING")
 endif(DISABLE_CRASH_LOG)
 
-if(UNIX)
-  # Enable debug info
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -g -ggdb3")
-  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -g -ggdb3")
-endif()
-
 # Enable C++-17
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
