Index: EternalTerminal-et-v6.0.13/CMakeLists.txt
===================================================================
--- EternalTerminal-et-v6.0.13.orig/CMakeLists.txt
+++ EternalTerminal-et-v6.0.13/CMakeLists.txt
@@ -26,12 +26,6 @@ if(DISABLE_CRASH_LOG)
 SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DELPP_DISABLE_DEFAULT_CRASH_HANDLING")
 ENDIF(DISABLE_CRASH_LOG)
 
-if(UNIX)
-  # Enable debug info
-  SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -g -ggdb3")
-  SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -g -ggdb3")
-endif()
-
 # Enable C++-11
 set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
