diff --git CMakeLists.txt CMakeLists.txt
index 3945f33..dd5b572 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -3,6 +3,7 @@ project (EternalTCP VERSION 5.1.8)
 
 option(BUILD_TEST "Build all unit test" OFF)
 option(CODE_COVERAGE "Enable code coverage" OFF)
+option(EXECINFO "Enable use of libexecinfo" OFF)
 
 SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DET_VERSION='\"${PROJECT_VERSION}\"'")
 # For easylogging, disable default log file, enable crash log, ensure thread safe, and catch c++ exceptions
@@ -88,6 +89,8 @@ IF(FREEBSD)
   set(CORE_LIBRARIES util execinfo)
 ELSEIF(NETBSD)
   set(CORE_LIBRARIES util resolv execinfo)
+ELSEIF(EXECINFO)
+  set(CORE_LIBRARIES util resolv execinfo)
 ELSE()
   set(CORE_LIBRARIES util resolv)
 ENDIF()
