From ed993538c711d817e5365630b65f372e0dfd01a7 Mon Sep 17 00:00:00 2001
From: fujiwarat <takao.fujiwara1@gmail.com>
Date: Thu, 27 Jan 2022 15:15:13 +0900
Subject: [PATCH] data: Update era.t with 2022

---
 data/era.t | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/data/era.t b/data/era.t
index a6d297e..686271f 100644
--- a/data/era.t
+++ b/data/era.t
@@ -2,7 +2,7 @@
 #
 # ibus-anthy - The Anthy engine for IBus
 #
-# Copyright (c) 2010-2021 Takao Fujiwara <takao.fujiwara1@gmail.com>
+# Copyright (c) 2010-2022 Takao Fujiwara <takao.fujiwara1@gmail.com>
 # Copyright (c) 2010-2013 Red Hat, Inc.
 #
 # This program is free software; you can redistribute it and/or modify
@@ -338,6 +338,8 @@
 れいわ２ #T35*500 2020
 れいわ３ #T35*500 令和3
 れいわ３ #T35*500 2021
+れいわ４ #T35*500 令和4
+れいわ４ #T35*500 2022
 １８６８ #T35*500 明治1
 １８６９ #T35*500 明治2
 １８７０ #T35*500 明治3
@@ -496,3 +498,4 @@
 ２０１９ #T35*500 平成31
 ２０２０ #T35*500 令和2
 ２０２１ #T35*500 令和3
+２０２２ #T35*500 令和4
