--- mscore/CMakeLists.txt.orig
+++ mscore/CMakeLists.txt
@@ -514,7 +514,7 @@
         if (APPLE)
             set(PORTMIDI_LIB portmidi)
         else (APPLE)
-            set(PORTMIDI_LIB -lportmidi -lporttime) # Remove -lporttime on RPM-based systems where PortTime is part of PortMidi.
+            set(PORTMIDI_LIB -lportmidi) # Remove -lporttime on RPM-based systems where PortTime is part of PortMidi.
         endif (APPLE)
         target_link_libraries(mscore ${PORTMIDI_LIB})
     endif (USE_PORTMIDI)
