--- sip-6.7.7.orig/sipbuild/builder.py
+++ sip-6.7.7/sipbuild/builder.py
@@ -250,6 +250,10 @@ class Builder(AbstractBuilder):
             sip_include_dirs.append(local_bindings_dir)
 
             # Add any bindings from previously installed packages.
+            if "PYPREFIX" in os.environ:
+                sip_include_dirs.append(os.environ["PYPREFIX"] +
+                        os.path.join(project.target_dir,
+                                project.get_bindings_dir()))
             sip_include_dirs.append(
                     os.path.join(project.target_dir,
                             project.get_bindings_dir()))
