Index: syncthing-gtk-0.9.4.4/syncthing_gtk/tools.py
===================================================================
--- syncthing-gtk-0.9.4.4.orig/syncthing_gtk/tools.py
+++ syncthing-gtk-0.9.4.4/syncthing_gtk/tools.py
@@ -259,7 +259,6 @@ def init_locale(localedir=None):
 	global _localedir
 	_localedir = localedir
 	gettext.bindtextdomain(GETTEXT_DOMAIN, localedir)
-	gettext.bind_textdomain_codeset(GETTEXT_DOMAIN, "utf-8")
 	gettext.textdomain(GETTEXT_DOMAIN)
 
 def get_locale_dir():
