From 9d9e80239647d37e5cfe320c422cce879c2e0d2c Mon Sep 17 00:00:00 2001
From: oreo639 <31916379+Oreo639@users.noreply.github.com>
Date: Sat, 9 Apr 2022 05:39:56 -0700
Subject: [PATCH] udev: add udev_dir as synonym of udevdir

---
 src/udev/udev.pc.in | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/udev/udev.pc.in b/src/udev/udev.pc.in
index 4a9d867b8..cccebb245 100644
--- a/src/udev/udev.pc.in
+++ b/src/udev/udev.pc.in
@@ -3,4 +3,5 @@ Description: eudev
 Version: @UDEV_VERSION@
 prefix=@prefix@
 exec_prefix=@exec_prefix@
-udevdir=@udevlibexecdir@
+udev_dir=@udevlibexecdir@
+udevdir=${udev_dir}
