From 6d2006a0f72abe86f7883956aa56fa48b9804ecf Mon Sep 17 00:00:00 2001
From: Andras Fuchs <andras.fuchs@gmail.com>
Date: Thu, 25 Nov 2021 20:21:13 +0100
Subject: [PATCH] Remove deprecated Gradle settings

---
 build.gradle | 16 +++++++++-------
 1 file changed, 9 insertions(+), 7 deletions(-)

diff --git a/build.gradle b/build.gradle
index 2991bcb..fadfda4 100644
--- a/build.gradle
+++ b/build.gradle
@@ -1,7 +1,7 @@
 buildscript {
     repositories {
         mavenCentral()
-        jcenter()
+        google()
     }
 
     dependencies {
@@ -49,17 +49,17 @@ if (JavaVersion.current().isJava8Compatible()) {
 
 repositories {
     mavenCentral()
-    jcenter()
+    google()
 }
 
 dependencies {
-    testCompile group: 'junit', name: 'junit', version: '4.10'
+    testImplementation group: 'junit', name: 'junit', version: '4.10'
 
     // https://mvnrepository.com/artifact/javax.help/javahelp
-    compile group: 'javax.help', name: 'javahelp', version: '2.0.05'
+    implementation group: 'javax.help', name: 'javahelp', version: '2.0.05'
 
-    compile group: 'org.apache.logging.log4j', name: 'log4j-api', version: '2.13.0'
-    compile group: 'org.apache.logging.log4j', name: 'log4j-core', version: '2.13.0'
+    implementation group: 'org.apache.logging.log4j', name: 'log4j-api', version: '2.13.0'
+    implementation group: 'org.apache.logging.log4j', name: 'log4j-core', version: '2.13.0'
 }
 
 
@@ -88,8 +88,10 @@ jar {
 
 task executableJar(type: Jar) {
     classifier = 'executable'
+	
+	duplicatesStrategy(DuplicatesStrategy.EXCLUDE)
 
-    from configurations.runtime.asFileTree.files.collect { zipTree(it) }
+    from configurations.runtimeClasspath.collect { zipTree(it) }
 
     from files(sourceSets.main.output)
 
