From 62780b31095fcd37a5d9218d0393d552bc44149f Mon Sep 17 00:00:00 2001
From: q66 <daniel@octaforge.org>
Date: Mon Jun 13 14:01:59 2022 +0200
Subject: fix ftbfs on powerpc32

diff --git a/js/src/wasm/WasmDebugFrame.h b/js/src/wasm/WasmDebugFrame.h
index 47652e9..02a13559 100644
--- a/js/src/wasm/WasmDebugFrame.h
+++ b/js/src/wasm/WasmDebugFrame.h
@@ -112,7 +112,7 @@ class DebugFrame {
 
   // Avoid -Wunused-private-field warnings.
  protected:
-#if defined(JS_CODEGEN_ARM) || defined(JS_CODEGEN_X86) || defined(__wasi__)
+#if defined(JS_CODEGEN_ARM) || defined(JS_CODEGEN_X86) || defined(__wasi__) || (defined(__powerpc__) && !defined(__powerpc64__))
   // See alignmentStaticAsserts().  For ARM32 and X86 DebugFrame is only
   // 4-byte aligned, so we add another word to get up to 8-byte
   // alignment.
