--- a/src/gui/CMakeLists.txt
+++ b/src/gui/CMakeLists.txt
@@ -393,8 +393,8 @@ else()
    set_property(SOURCE ../3rdparty/qtlockedfile/qtlockedfile_win.cpp PROPERTY SKIP_UNITY_BUILD_INCLUSION ON)
 endif()
 
-find_package(Qt${QT_VERSION_MAJOR} ${REQUIRED_QT_VERSION} COMPONENTS LinguistTools)
-if(Qt${QT_MAJOR_VERSION}LinguistTools_FOUND)
+find_package(Qt${QT_VERSION_MAJOR} ${REQUIRED_QT_VERSION} COMPONENTS Linguist)
+if(Qt${QT_MAJOR_VERSION}Linguist_FOUND)
     qt_add_translation(client_I18N ${TRANSLATIONS})
 endif()
 
