is wrong if running within a user's chroot

--- test/testfolderman.cpp.ORIG	2021-04-21 23:33:49.296779423 +0200
+++ test/testfolderman.cpp	2021-04-21 23:34:03.289770264 +0200
@@ -122,9 +122,6 @@ private slots:
         QVERIFY(folderman->checkPathValidityForNewFolder(dirPath + "/link1/subfolder").isNull());
         QVERIFY(folderman->checkPathValidityForNewFolder(dirPath + "/link2/free/subfolder").isNull());

-        // Should not have the rights
-        QVERIFY(!folderman->checkPathValidityForNewFolder("/").isNull());
-        QVERIFY(!folderman->checkPathValidityForNewFolder("/usr/bin/somefolder").isNull());
 #endif

 #ifdef Q_OS_WIN // drive-letter tests
