is wrong if running within a user's chroot

--- a/test/testfolderman.cpp	2020-09-24 14:51:48.000000000 +0200
+++ b/test/testfolderman.cpp	2020-10-02 20:33:32.248359240 +0200
@@ -119,10 +119,6 @@
         // Subfolder of links
         QVERIFY(folderman->checkPathValidityForNewFolder(dirPath + "/link1/subfolder").isNull());
         QVERIFY(folderman->checkPathValidityForNewFolder(dirPath + "/link2/free/subfolder").isNull());
-
-        // Should not have the rights
-        QVERIFY(!folderman->checkPathValidityForNewFolder("/").isNull());
-        QVERIFY(!folderman->checkPathValidityForNewFolder("/usr/bin/somefolder").isNull());
 #endif
 
         // Invalid paths
