From 613c644f0479b2aae6be96fa389352883a2321c3 Mon Sep 17 00:00:00 2001
From: figsoda <figsoda@pm.me>
Date: Sat, 1 Apr 2023 08:22:47 -0400
Subject: [PATCH] Default workspace members to cli (#484)

---
 Cargo.toml | 1 +
 flake.nix  | 3 ---
 2 files changed, 1 insertion(+), 3 deletions(-)

diff --git a/Cargo.toml b/Cargo.toml
index 3db8b3b9c..3fe010947 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -6,6 +6,7 @@ edition = "2021"
 
 [workspace]
 members = ["cli", "docs", "library", "macros", "tests"]
+default-members = ["cli"]
 
 [lib]
 doctest = false
