Source: https://gitlab.gnome.org/GNOME/babl/issues/24#note_712527

--- babl/babl-fish-path.c
+++ babl/babl-fish-path.c
@@ -404,7 +404,7 @@ alias_conversion (Babl *babl,
                     (void*)conv->source, (void*)space),
               babl_remodel_with_space (
                     (void*)conv->destination, (void*)space),
-              "linear", conv->function,
+              "linear", conv->function.linear,
               NULL);
         break;
       case BABL_CONVERSION_PLANAR:
@@ -413,7 +413,7 @@ alias_conversion (Babl *babl,
                     (void*)conv->source, (void*)space),
               babl_remodel_with_space (
                     (void*)conv->destination, (void*)space),
-              "planar", conv->function,
+              "planar", conv->function.planar,
               NULL);
         break;
       case BABL_CONVERSION_PLANE:
@@ -422,7 +422,7 @@ alias_conversion (Babl *babl,
                     (void*)conv->source, (void*)space),
               babl_remodel_with_space (
                     (void*)conv->destination, (void*)space),
-              "plane", conv->function,
+              "plane", conv->function.plane,
               NULL);
         break;
       default:
