--- src/common/classes/alloc.h	2015-11-16 12:39:40.000000000 +0100
+++ src/common/classes/alloc.h	2016-09-14 14:28:38.810632773 +0200
@@ -49,14 +49,8 @@
 					   stdlib.h (EKU) */
 #endif
 
-// MSVC does not support exception specification, so it's unknown if that will be correct or not
-// from its POV. For now, use it and disable the C4290 warning.
-//
-//#if defined (_MSC_VER)
-//#define THROW_BAD_ALLOC
-//#else
-#define THROW_BAD_ALLOC throw (std::bad_alloc)
-//#endif
+// Handled by -fcheck-new compiler option
+#define THROW_BAD_ALLOC
 
 #ifdef USE_VALGRIND
 
