--- a/util.c	2012-01-13 23:25:58.000000000 +0100
+++ b/util.c	2018-05-26 20:14:12.290726591 +0200
@@ -386,7 +386,7 @@
 			msg = "Invalid back reference to a subexpression.";
 			break;
 
-#ifdef linux
+#ifdef __GLIBC__
 		case REG_EEND:
 			msg = "Non specific error.";
 			break;
@@ -400,7 +400,7 @@
 			msg = "Invalid use of pattern operators such as group or list.";
 			break;
 
-#ifdef linux
+#ifdef __GLIBC__
 		case REG_ESIZE:
 			msg = "Compiled regular expression requires a pattern buffer larger than  64Kb.";
 			break;
