diff --git a/hunt/hunt/playit.c b/hunt/hunt/playit.c
index 9acf86e..881a4e7 100644
--- a/hunt/hunt/playit.c
+++ b/hunt/hunt/playit.c
@@ -114,7 +114,7 @@ playit()
 		bad_con();
 		/* NOTREACHED */
 	}
-	if (ntohl(version) != (unsigned long)HUNT_VERSION) {
+	if (ntohl(version) != (uint32_t)HUNT_VERSION) {
 		bad_ver();
 		/* NOTREACHED */
 	}
@@ -649,7 +649,7 @@ do_message()
 		bad_con();
 		/* NOTREACHED */
 	}
-	if (ntohl(version) != (unsigned long)HUNT_VERSION) {
+	if (ntohl(version) != (uint32_t)HUNT_VERSION) {
 		bad_ver();
 		/* NOTREACHED */
 	}
