--- a/libntfs-3g/Makefile.am
+++ b/libntfs-3g/Makefile.am
@@ -59,15 +59,15 @@
 # And create ldscript or symbolic link from /usr
 install-exec-hook: install-rootlibLTLIBRARIES
 if INSTALL_LIBRARY
-	if [ ! "$(rootlibdir)" -ef "$(libdir)" ]; then \
+	if test "$(rootlibdir)" != "$(libdir)"; then \
 		$(MV) -f "$(DESTDIR)/$(libdir)"/libntfs-3g.so* "$(DESTDIR)/$(rootlibdir)";  \
 	fi
 if GENERATE_LDSCRIPT
-	if [ ! "$(rootlibdir)" -ef "$(libdir)" ]; then \
+	if test "$(rootlibdir)" != "$(libdir)"; then \
 		$(install_sh_PROGRAM) "libntfs-3g.script.so" "$(DESTDIR)/$(libdir)/libntfs-3g.so"; \
 	fi
 else
-	if [ ! "$(rootlibdir)" -ef "$(libdir)" ]; then \
+	if test "$(rootlibdir)" != "$(libdir)"; then \
 		$(LN_S) "$(rootlibdir)/libntfs-3g.so" "$(DESTDIR)/$(libdir)/libntfs-3g.so"; \
 	fi
 endif
--- a/libntfs-3g/Makefile.in
+++ b/libntfs-3g/Makefile.in
@@ -1170,13 +1170,13 @@
 # We may need to move .so files to root
 # And create ldscript or symbolic link from /usr
 install-exec-hook: install-rootlibLTLIBRARIES
-@INSTALL_LIBRARY_TRUE@	if [ ! "$(rootlibdir)" -ef "$(libdir)" ]; then \
+@INSTALL_LIBRARY_TRUE@	if test "$(rootlibdir)" != "$(libdir)"; then \
 @INSTALL_LIBRARY_TRUE@		$(MV) -f "$(DESTDIR)/$(libdir)"/libntfs-3g.so* "$(DESTDIR)/$(rootlibdir)";  \
 @INSTALL_LIBRARY_TRUE@	fi
-@GENERATE_LDSCRIPT_TRUE@@INSTALL_LIBRARY_TRUE@	if [ ! "$(rootlibdir)" -ef "$(libdir)" ]; then \
+@GENERATE_LDSCRIPT_TRUE@@INSTALL_LIBRARY_TRUE@	if test "$(rootlibdir)" != "$(libdir)"; then \
 @GENERATE_LDSCRIPT_TRUE@@INSTALL_LIBRARY_TRUE@		$(install_sh_PROGRAM) "libntfs-3g.script.so" "$(DESTDIR)/$(libdir)/libntfs-3g.so"; \
 @GENERATE_LDSCRIPT_TRUE@@INSTALL_LIBRARY_TRUE@	fi
-@GENERATE_LDSCRIPT_FALSE@@INSTALL_LIBRARY_TRUE@	if [ ! "$(rootlibdir)" -ef "$(libdir)" ]; then \
+@GENERATE_LDSCRIPT_FALSE@@INSTALL_LIBRARY_TRUE@	if test "$(rootlibdir)" != "$(libdir)"; then \
 @GENERATE_LDSCRIPT_FALSE@@INSTALL_LIBRARY_TRUE@		$(LN_S) "$(rootlibdir)/libntfs-3g.so" "$(DESTDIR)/$(libdir)/libntfs-3g.so"; \
 @GENERATE_LDSCRIPT_FALSE@@INSTALL_LIBRARY_TRUE@	fi
 
