diff --git a/resources/sioyek.desktop b/resources/sioyek.desktop
index cf8b6b2..917fd62 100644
--- a/resources/sioyek.desktop
+++ b/resources/sioyek.desktop
@@ -1,8 +1,12 @@
 [Desktop Entry]
-Version=1.0
-Type=Application
-MimeType=application/pdf;
 Name=Sioyek
-Exec=sioyek
+Comment=PDF viewer for reading research papers and technical books
+Keywords=pdf;viewer;reader;research;
+TryExec=sioyek
+Exec=sioyek %U
+StartupNotify=true
+Terminal=false
+Type=Application
 Icon=sioyek-icon-linux
-Categories=Development;
+Categories=Development;Viewer;
+MimeType=application/pdf;
