+++ b/external_libs/oscpack_1_1_0/osc/OscReceivedElements.cpp	2019-06-06 22:06:36.000000000 +0200
--- a/external_libs/oscpack_1_1_0/osc/OscReceivedElements.cpp	2019-06-07 10:54:40.107305153 +0200
@@ -105,7 +105,7 @@
 
     return u.i;
 #else
-	return *(int32*)p;
+    return *(int32*)p;
 #endif
 }
 
@@ -125,7 +125,7 @@
 
     return u.i;
 #else
-	return *(uint32*)p;
+    return *(uint32*)p;
 #endif
 }
 
@@ -149,7 +149,7 @@
 
     return u.i;
 #else
-	return *(int64*)p;
+    return *(int64*)p;
 #endif
 }
 
@@ -173,7 +173,7 @@
 
     return u.i;
 #else
-	return *(uint64*)p;
+    return *(uint64*)p;
 #endif
 }
 
@@ -249,7 +249,7 @@
 
     return u.i;
 #else
-	return *(int32*)argument_;
+    return *(int32*)argumentPtr_;
 #endif
 }
 
@@ -280,7 +280,7 @@
 
     return u.f;
 #else
-	return *(float*)argument_;
+    return *(float*)argumentPtr_;
 #endif
 }
 
@@ -400,7 +400,7 @@
 
     return u.d;
 #else
-	return *(double*)argument_;
+    return *(double*)argumentPtr_;
 #endif
 }
 
