From 7677dabea40a2d17a42a718ca8c7e450b09e153c Mon Sep 17 00:00:00 2001
From: Phil Hagelberg <phil@hagelb.org>
Date: Wed, 8 Jul 2020 14:31:03 -0700
Subject: [PATCH] Fix missing quote in bin/lein-pkg.

Fixes #2691
---
 bin/lein-pkg | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- bin/lein-pkg
+++ bin/lein-pkg
@@ -86,7 +86,7 @@ if [ -r .lein-fast-trampoline ]; then
 fi
 
 if [ "$LEIN_FAST_TRAMPOLINE" != "" ] && [ -r project.clj ]; then
-    INPUTS="$* $(cat project.clj) $LEIN_VERSION $(test -f "$LEIN_HOME/profiles.clj" && cat "$LEIN_HOME/profiles.clj") $(test -f profiles.clj && cat profiles.clj)
+    INPUTS="$* $(cat project.clj) $LEIN_VERSION $(test -f "$LEIN_HOME/profiles.clj" && cat "$LEIN_HOME/profiles.clj") $(test -f profiles.clj && cat profiles.clj)"
 
     if command -v shasum >/dev/null 2>&1; then
         SUM="shasum"
