diff -p1ru a/src/unity-greeter.vala b/src/unity-greeter.vala
--- a/src/unity-greeter.vala	2023-06-08 15:31:11.000000000 +0800
+++ b/src/unity-greeter.vala	2024-07-26 17:27:41.963293863 +0800
@@ -644,3 +644,4 @@ public class UnityGreeter
 
-                Shell.parse_argv ("systemctl --user start indicator-application indicator-power indicator-datetime indicator-keyboard indicator-session indicator-sound", out argv);
+                /* Missing indicators: application, keyboard */
+                Shell.parse_argv ("sv start indicator-power indicator-datetime indicator-session indicator-sound", out argv);
                 Process.spawn_async (null,
