--- a/third_party/skia/src/ports/SkFontHost_FreeType.cpp.orig	2019-07-19 11:08:34.770972665 +0000
+++ b/third_party/skia/src/ports/SkFontHost_FreeType.cpp	2019-07-19 11:08:44.274442065 +0000
@@ -128,9 +128,9 @@ public:
         : fGetVarDesignCoordinates(nullptr)
         , fGetVarAxisFlags(nullptr)
         , fLibrary(nullptr)
-        , fIsLCDSupported(false)
+        , fIsLCDSupported(true)
         , fLightHintingIsYOnly(false)
-        , fLCDExtra(0)
+        , fLCDExtra(2)
     {
         if (FT_New_Library(&gFTMemory, &fLibrary)) {
             return;
