diff --git a/sapi/apache2handler/config.m4 b/sapi/apache2handler/config.m4
index e335721..d824d5f 100644
--- a/sapi/apache2handler/config.m4
+++ b/sapi/apache2handler/config.m4
@@ -122,7 +122,7 @@ if test "$PHP_APXS2" != "no"; then
 
   AS_IF([$APXS_HTTPD -V 2>/dev/null | grep 'threaded:.*yes' >/dev/null 2>&1], [
     APACHE_THREADED_MPM=yes
-    enable_zts=yes
+    enable_zts=no
   ], [APACHE_THREADED_MPM=no])
 
 AC_CONFIG_COMMANDS([apache2handler], [AS_VAR_IF([enable_zts], [yes],,
