diff --git a/build/php.m4 b/build/php.m4
index d8a5cbf..a870ca4 100644
--- a/build/php.m4
+++ b/build/php.m4
@@ -1967,7 +1967,11 @@ AS_VAR_IF([found_pgsql], [no], [dnl
     AS_IF([test -x $i/pg_config], [PG_CONFIG="$i/pg_config"; break;])
   done
 
-  AS_VAR_IF([PG_CONFIG],, [dnl
+  AS_IF([test -n "$PG_CONFIG" && test "x$cross_compiling" != "xyes"], [dnl
+    AC_MSG_RESULT([$PG_CONFIG])
+    PGSQL_INCLUDE=$($PG_CONFIG --includedir)
+    PGSQL_LIBDIR=$($PG_CONFIG --libdir)
+  ], [dnl
     AC_MSG_RESULT([not found])
     AS_VAR_IF([pgsql_dir],,
       [pgsql_search_paths="/usr /usr/local /usr/local/pgsql"],
@@ -1983,10 +1987,6 @@ AS_VAR_IF([found_pgsql], [no], [dnl
           [PGSQL_LIBDIR=$i/$j])
       done
     done
-  ], [dnl
-    AC_MSG_RESULT([$PG_CONFIG])
-    PGSQL_INCLUDE=$($PG_CONFIG --includedir)
-    PGSQL_LIBDIR=$($PG_CONFIG --libdir)
   ])
 
   AS_IF([test -n "$PGSQL_INCLUDE" && test -n "PGSQL_LIBDIR"], [
