--- configure.orig	2011-10-14 11:52:37.150473742 +0200
+++ configure	2011-10-14 11:52:46.191239300 +0200
@@ -14181,7 +14181,7 @@ cat >>confdefs.h <<_ACEOF
 _ACEOF
 
 
-ac_config_files="$ac_config_files Makefile src/Makefile src/tests/Makefile data/Makefile data/ui/Makefile po/Makefile.in docs/Makefile docs/reference/Makefile docs/reference/libfm/Makefile libfm.pc libfm-gtk.pc"
+ac_config_files="$ac_config_files Makefile src/Makefile src/tests/Makefile data/Makefile data/ui/Makefile po/Makefile.in libfm.pc libfm-gtk.pc"
 
 
 cat >confcache <<\_ACEOF
