--- a/Makefile	2017-08-12 17:55:29.265988875 +0200
+++ b/Makefile	2017-08-12 17:56:17.492988011 +0200
@@ -122,7 +122,6 @@
 	rm -rf $(DISTDIR)
 	install -d -m 755 $(DISTDIR) $(DISTDIR)/src $(DISTDIR)/doc $(DISTDIR)/examples
 	install -m 755 src/ferm src/import-ferm $(DISTDIR)/src
-	install -m 644 ferm.service $(DISTDIR)
 	install -m 644 doc/ferm.pod doc/ferm.txt doc/ferm.html doc/ferm.1 doc/import-ferm.1 $(DISTDIR)/doc
 	install -m 644 config.mk Makefile AUTHORS COPYING NEWS README.rst TODO $(DISTDIR)
 	install -m 644 $(wildcard examples/*.ferm) $(DISTDIR)/examples
@@ -137,10 +136,10 @@
 .PHONY: install uninstall
 
 install: all
-	install -d -m 755 $(DOCDIR)/examples $(PREFIX)/sbin $(SERVICEDIR)
+	install -d -m 755 $(DOCDIR)/examples $(PREFIX)/bin $(SERVICEDIR)
 	install -m 644 AUTHORS COPYING NEWS README.rst TODO $(DOCDIR)
 	install -m 644 examples/*.ferm $(DOCDIR)/examples
-	install -m 755 src/ferm src/import-ferm $(PREFIX)/sbin/
+	install -m 755 src/ferm src/import-ferm $(PREFIX)/bin/
 	install -m 644 ferm.service $(SERVICEDIR)
 
 	install -d -m 755 $(DOCDIR) $(MANDIR)
@@ -152,7 +151,7 @@
 	rm -rf $(DOCDIR)
 	rm -f $(MANDIR)/ferm.1 $(MANDIR)/import-ferm.1
 	rm -f $(SERVICEDIR)/ferm.service
-	rm -f $(PREFIX)/sbin/ferm $(PREFIX)/sbin/import-ferm
+	rm -f $(PREFIX)/bin/ferm $(PREFIX)/bin/import-ferm
 
 #
 # misc targets
