diff -aur src/Basescape/BaseInfoState.h src/Basescape/BaseInfoState.h
--- src/Basescape/BaseInfoState.h	2018-03-14 15:06:59.665628271 +0000
+++ src/Basescape/BaseInfoState.h	2018-03-14 15:43:22.067605905 +0000
@@ -19,6 +19,7 @@
 #ifndef OPENXCOM_BASEINFOSTATE_H
 #define OPENXCOM_BASEINFOSTATE_H
 
+#include <cmath>
 #include "../Engine/State.h"
 
 namespace OpenXcom
diff -aur src/Basescape/CraftInfoState.h src/Basescape/CraftInfoState.h
--- src/Basescape/CraftInfoState.h	2018-03-14 15:02:09.783729176 +0000
+++ src/Basescape/CraftInfoState.h	2018-03-14 15:43:16.476703036 +0000
@@ -19,6 +19,7 @@
 #ifndef OPENXCOM_CRAFTINFOSTATE_H
 #define OPENXCOM_CRAFTINFOSTATE_H
 
+#include <cmath>
 #include "../Engine/State.h"
 
 namespace OpenXcom
diff -aur src/Engine/Language.h src/Engine/Language.h
--- src/Engine/Language.h	2018-03-14 15:11:37.236763021 +0000
+++ src/Engine/Language.h	2018-03-14 15:42:14.751775215 +0000
@@ -23,6 +23,7 @@
 #include <vector>
 #include <string>
 #include <yaml-cpp/yaml.h>
+#include <algorithm>
 #include "LocalizedText.h"
 #include "../Savegame/Soldier.h"
 
diff -aur src/Engine/OptionInfo.h src/Engine/OptionInfo.h
--- src/Engine/OptionInfo.h	2018-03-14 15:14:50.346384845 +0000
+++ src/Engine/OptionInfo.h	2018-03-14 15:42:14.750775233 +0000
@@ -23,6 +23,7 @@
 #include <string>
 #include <map>
 #include <SDL.h>
+#include <algorithm>
 
 namespace OpenXcom
 {
diff -aur src/Savegame/CraftWeapon.h src/Savegame/CraftWeapon.h
--- src/Savegame/CraftWeapon.h	2018-03-14 15:27:14.695387913 +0000
+++ src/Savegame/CraftWeapon.h	2018-03-14 15:42:14.748775267 +0000
@@ -21,6 +21,7 @@
 
 #include <string>
 #include <yaml-cpp/yaml.h>
+#include <cmath>
 
 namespace OpenXcom
 {
