# fix tmux wait condition
--- a/test/utils.py
+++ b/test/utils.py
@@ -19,7 +19,7 @@ class Nemu():
         wait_cur = 0
         while os.path.exists(self.pidfile):
             time.sleep(0.1)
             wait_cur += 1
-            if wait_cur == wait_max:
+            if wait_cur >= wait_max:
                 break
         sub = subprocess.run([os.getenv("NEMU_BIN_DIR") + "/nemu",
             "--cfg", "/tmp/nemu_" + self.uuid + "/nemu.cfg", "--cmd", vm],
@@ -55,7 +55,7 @@ class Tmux():
         wait_cur = 0
         while os.path.exists(pidfile):
             time.sleep(0.1)
             wait_cur += 1
-            if wait_max >= wait_cur:
+            if wait_cur >= wait_max:
                 break
         sub = subprocess.run(["tmux", "-f", "tmux.conf", "-L", self.uuid,
                 "new-session", "-d", os.getenv("NEMU_BIN_DIR") + "/nemu",
@@ -65,7 +65,7 @@ class Tmux():
         wait_cur = 0
         while not os.path.exists(pidfile):
             time.sleep(0.1)
             wait_cur += 1
-            if wait_max >= wait_cur:
+            if wait_cur >= wait_max:
                 break
         return sub.returncode
     def dump_screen(self, log_path):
