--- a/data/autostart/autostart.in~	2011-08-02 15:50:01.963531310 +0000
+++ b/data/autostart/autostart.in	2011-08-02 15:51:57.277879458 +0000
@@ -8,7 +8,7 @@
 #
 #if test -x @libexecdir@/gnome-settings-daemon >/dev/null; then
 #  @libexecdir@/gnome-settings-daemon &
-#elif which gnome-settings-daemon >/dev/null; then
+#elif which gnome-settings-daemon >/dev/null 2>&1; then
 #  gnome-settings-daemon &
 #fi
 
