--- a/tests/hypothesis/test_inherited_test.py	2023-07-12 07:11:21.000000000 -0300
+++ b/tests/hypothesis/test_inherited_test.py	2023-09-15 19:12:21.328680631 -0300
@@ -1,10 +1,11 @@
 import hypothesis.strategies as st
 import pytest
-from hypothesis import given
+from hypothesis import given, settings, HealthCheck
 
 
 class BaseClass:
     @pytest.mark.asyncio
+    @settings(suppress_health_check=(HealthCheck.differing_executors,))
     @given(value=st.integers())
     async def test_hypothesis(self, value: int) -> None:
         pass
