--- Makefile.in.orig	2013-05-17 06:48:20.607147825 +0200
+++ Makefile.in	2013-05-17 06:49:28.438511555 +0200
@@ -49,7 +49,7 @@ NORMAL_UNINSTALL = :
 PRE_UNINSTALL = :
 POST_UNINSTALL = :
 bin_PROGRAMS = xdelta3$(EXEEXT)
-noinst_PROGRAMS = xdelta3regtest$(EXEEXT) xdelta3decode$(EXEEXT)
+noinst_PROGRAMS = xdelta3decode$(EXEEXT)
 @DEBUG_SYMBOLS_TRUE@am__append_1 = -g
 subdir = .
 DIST_COMMON = README $(am__configure_deps) $(srcdir)/Makefile.am \
