--- a/configure.in
+++ b/configure.in
@@ -563,69 +563,6 @@ AC_MSG_RESULT([])
 AC_SUBST(KRB4)])dnl
 WITH_KRB4
 
-krb_h=
-AC_MSG_CHECKING([for krb.h])
-if test "$cross_compiling" != yes && test -r $KRB4/include/krb.h; then
-   hold_cflags=$CFLAGS
-   CFLAGS="$CFLAGS -I$KRB4/include"
-   AC_TRY_LINK([#include <krb.h>],[int i;],
-          [krb_h=yes krb_incdir=$KRB4/include],
-          [CFLAGS=$hold_cflags
-           AC_TRY_LINK([#include <krb.h>],[int i;],
-             [krb_h=yes krb_incdir=])])
-   CFLAGS=$hold_cflags
-else
-   AC_TRY_LINK([#include <krb.h>],[int i;],
-             [krb_h=yes krb_incdir=])
-fi
-if test -z "$krb_h"; then
-  AC_TRY_LINK([#include <krb.h>],[int i;],
-    [krb_h=yes krb_incdir=],
-    [if test "$cross_compiling" != yes && test -r $KRB4/include/kerberosIV/krb.h; then
-       hold_cflags=$CFLAGS
-       CFLAGS="$CFLAGS -I$KRB4/include/kerberosIV"
-       AC_TRY_LINK([#include <krb.h>],[int i;],
-	 [krb_h=yes krb_incdir=$KRB4/include/kerberosIV])
-       CFLAGS=$hold_cflags
-     fi])
-fi
-AC_MSG_RESULT($krb_h)
-
-if test -n "$krb_h"; then
-  krb_lib=
-  if test "$cross_compiling" != yes && test -r $KRB4/lib/libkrb.a; then
-       hold_ldflags=$LDFLAGS
-       LDFLAGS="-L${KRB4}/lib $LDFLAGS"
-       AC_CHECK_LIB(krb,printf,[krb_lib=yes krb_libdir=${KRB4}/lib],
-           [LDFLAGS=$hold_ldflags
-            # Using open here instead of printf so we don't
-            # get confused by the cached value for printf from above.
-            AC_CHECK_LIB(krb,open,[krb_lib=yes krb_libdir=])])
-       LDFLAGS=$hold_ldflags
-  else
-       AC_CHECK_LIB(krb,printf,[krb_lib=yes krb_libdir=])
-       AC_CHECK_FUNC(krb_recvauth,[krb_lib=yes krb_libdir=])
-  fi
-  if test -n "$krb_lib"; then
-    AC_DEFINE([HAVE_KERBEROS], 1,
-	      [Define if you have MIT Kerberos version 4 available.])
-    cvs_client_objects="$cvs_client_objects kerberos4-client.o"
-    test -n "${krb_libdir}" && LIBS="${LIBS} -L${krb_libdir}"
-    # Put -L${krb_libdir} in LDFLAGS temporarily so that it appears before
-    # -ldes in the command line.  Don't do it permanently so that we honor
-    # the user's setting for LDFLAGS
-    hold_ldflags=$LDFLAGS
-    test -n "${krb_libdir}" && LDFLAGS="$LDFLAGS -L${krb_libdir}"
-    AC_CHECK_LIB(des,printf,[LIBS="${LIBS} -ldes"])
-    AC_CHECK_LIB(krb,krb_recvauth)
-    AC_CHECK_LIB(krb4,krb_recvauth)
-    LDFLAGS=$hold_ldflags
-    if test -n "$krb_incdir"; then
-      CPPFLAGS="$CPPFLAGS -I$krb_incdir"
-    fi
-  fi
-fi
-AC_CHECK_FUNCS(krb_get_err_text)
 
 
 dnl
