--- src/sat.cc	2021-01-13 09:41:59.963038539 +1100
+++ src/sat.cc	2021-01-13 09:42:56.376419216 +1100
@@ -1485,6 +1485,7 @@
 // Return the worst case (largest) cache line size of the various levels of
 // cache actually prsent in the machine.
 int Sat::CacheLineSize() {
+#ifdef _SC_LEVEL1_DCACHE_LINESIZE
   int max_linesize = sysconf(_SC_LEVEL1_DCACHE_LINESIZE);
   int linesize = sysconf(_SC_LEVEL2_CACHE_LINESIZE);
   if (linesize > max_linesize) max_linesize = linesize;
@@ -1493,6 +1494,9 @@
   linesize = sysconf(_SC_LEVEL4_CACHE_LINESIZE);
   if (linesize > max_linesize) max_linesize = linesize;
   return max_linesize;
+#else
+  return 0;
+#endif
 }
 
 // Notify and reap worker threads.
