--- src/actions.cpp.orig	2017-09-19 11:49:12.000000000 +0000
+++ src/actions.cpp	2017-09-19 11:49:26.682396655 +0000
@@ -2049,7 +2049,11 @@
   /* This is the critical section object (statically allocated). */
   static pthread_mutex_t cs =  PTHREAD_RECURSIVE_MUTEX_INITIALIZER;
  #else
-  static pthread_mutex_t cs =  PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP;
+  #if !defined(__GLIBC__)
+   static pthread_mutex_t cs =  {{{PTHREAD_MUTEX_RECURSIVE}}};
+  #else
+   static pthread_mutex_t cs =  PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP;
+  #endif
  #endif
 #endif
 
