--- a/Makefile
+++ b/Makefile
@@ -99,7 +99,7 @@ else ifneq (,$(findstring rpi,$(platform)))
       GL_LIB := -lGLESv2
    else
       LLE = 0
-      CPUFLAGS += -DVC
+      CPUFLAGS += -DVC -DGL_GLEXT_PROTOTYPES
       GL_LIB := -L/opt/vc/lib -lbrcmGLESv2
       EGL_LIB := -lbrcmEGL
       INCFLAGS += -I/opt/vc/include -I/opt/vc/include/interface/vcos -I/opt/vc/include/interface/vcos/pthreads
--- a/libretro-common/include/glsm/glsm.h
+++ b/libretro-common/include/glsm/glsm.h
@@ -32,8 +32,8 @@
 RETRO_BEGIN_DECLS
 
 #ifdef HAVE_OPENGLES2
-typedef GLfloat GLdouble;
-typedef GLclampf GLclampd;
+typedef double GLdouble;
+typedef double GLclampd;
 #endif
 
 #if defined(HAVE_OPENGLES2)
