diff --git a/platformplugin/windoweventhook.cpp b/platformplugin/windoweventhook.cpp
index 742b41d..428dcb0 100644
--- a/platformplugin/windoweventhook.cpp
+++ b/platformplugin/windoweventhook.cpp
@@ -359,6 +359,8 @@ void WindowEventHook::handleFocusInEvent(const xcb_focus_in_event_t *event)
     if (relayFocusToModalWindow(w, xcbWindow->connection()))
         return;
 
+    xcbWindow->connection()->focusInTimer().stop();
+
 #if QT_VERSION >= QT_VERSION_CHECK(5, 9, 0)
     xcbWindow->connection()->setFocusWindow(w);
 #else
@@ -433,7 +435,7 @@ void WindowEventHook::handleFocusOutEvent(const xcb_focus_out_event_t *event)
     // Do not set the active window to 0 if there is a FocusIn coming.
     // There is however no equivalent for XPutBackEvent so register a
     // callback for QXcbConnection instead.
-    xcbWindow->connection()->addPeekFunc(focusInPeeker);
+    xcbWindow->connection()->focusInTimer().start(400);
 }
 
 void WindowEventHook::handlePropertyNotifyEvent(const xcb_property_notify_event_t *event)
