diff --git Makefile Makefile
index f6d69d3..58a9d98 100644
--- a/Makefile
+++ b/Makefile
@@ -4,7 +4,7 @@ UNAME_S := $(shell uname -s)
 
 DESTDIR ?=
 prefix  ?= /usr
-sbindir ?= $(prefix)/sbin
+sbindir ?= $(prefix)/bin
 
 INSTALL		:= install
 INSTALL_DIR	:= $(INSTALL) -m 755 -d
@@ -14,7 +14,7 @@ RM		:= rm -rf
 CC ?= gcc
 CFLAGS ?= -g -O0
 CFLAGS += -Wall -Wextra -std=c99 -pedantic
-GIT_VERSION := $(shell git describe --abbrev=8 --dirty --always --tags)
+GIT_VERSION := $(shell basename "$(CURDIR)" | cut -d '-' -f 2)
 ifeq ($(GIT_VERSION),)
     GIT_VERSION := $(shell cat VERSION)
 endif
