--- Makefile
+++ Makefile
@@ -4,7 +4,7 @@
 
 DESTDIR ?=
 prefix  ?= /usr
-sbindir ?= $(prefix)/sbin
+sbindir ?= $(prefix)/bin
 
 INSTALL		:= install
 INSTALL_DIR	:= $(INSTALL) -m 755 -d
@@ -14,7 +14,7 @@
 CC ?= gcc
 CFLAGS ?= -g -O0
 CFLAGS += -Wall -Wextra -std=c99 -pedantic
-GIT_VERSION := $(shell git describe --abbrev=4 --dirty --always --tags)
+GIT_VERSION := $(shell basename "$(CURDIR)" | cut -d '-' -f 2)
 CFLAGS += -DPROGRAM_VERSION=\"$(GIT_VERSION)\"
 
 ifeq ($(UNAME_S),Linux)
