 installation path

---
 data/anilibria.desktop | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/data/anilibria.desktop b/data/anilibria.desktop
index 3dfcc73..04c92a5 100644
--- a/data/anilibria.desktop
+++ b/data/anilibria.desktop
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Name=AniLibria
 Comment=AniLibria desktop client
-Exec=/opt/AniLibria/bin/AniLibria
+Exec=/usr/bin/AniLibria
 Icon=anilibria
 Type=Application
 Version=0.1.2
-- 
2.26.2

