From dd36cce8c47bb0e17a789cf2bd95a51e29b59e78 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Daniel=20Ekl=C3=B6f?= <daniel@ekloef.se>
Date: Tue, 30 Apr 2024 08:47:19 +0200
Subject: [PATCH] shm: MFD_NOEXEC_SEAL is not defined by musl libc
---
diff --git a/shm.c b/shm.c
index bd4ce26..c0fe0bc 100644
--- a/shm.c
+++ b/shm.c
@@ -19,6 +19,10 @@
  #define MAP_UNINITIALIZED 0
 #endif
 
+#if !defined(MFD_NOEXEC_SEAL)
+ #define MFD_NOEXEC_SEAL 0
+#endif
+
 static void
 buffer_destroy(struct buffer *buf)
 {
