Fix an obvious typo.

--- src/FXString.cpp	2016-06-16 06:16:00.000000000 +0200
+++ src/FXString.cpp	2016-11-01 18:16:42.717595139 +0100
@@ -2365,7 +2365,7 @@
 #else
     va_list ag;
 x:  va_copy(ag,args);
-    result=vsnprintf(str,length()+1,fmt,a);
+    result=vsnprintf(str,length()+1,fmt,ag);
     va_end(ag);
     if(result<0){ length(FXMAX(64,length()*2)); goto x; }
     if(length()<result){ length(result); goto x; }
