--- a/src/qdoc/qdoc/CMakeLists.txt
+++ b/src/qdoc/qdoc/CMakeLists.txt
@@ -103,7 +103,6 @@ qt_internal_return_unless_building_tools
 
 # If libclangTooling.a is not built with -fPIE enabled we cannot link it to qdoc.
 # TODO: Re-enable PIE once clang is built with PIE in provisioning.
-set_target_properties(${target_name} PROPERTIES POSITION_INDEPENDENT_CODE FALSE)
 
 qt_internal_extend_target(${target_name} CONDITION (WIN32 AND ICC) OR MSVC
     LINK_OPTIONS
