--- a/src/linguist/lupdate/CMakeLists.txt	2023-10-03 20:26:34.000000000 +0200
+++ -	2023-10-13 22:42:43.023536533 +0200
@@ -95,11 +95,11 @@
         WrapLibClang::WrapLibClang
 )
 
-if(QT_FEATURE_clangcpp)
-    # If libclangTooling.a is not built with -fPIE enabled we cannot link it to lupdate.
-    # TODO: Re-enable PIE once clang is built with PIE in provisioning.
-    set_target_properties(${target_name} PROPERTIES POSITION_INDEPENDENT_CODE FALSE)
-endif()
+#if(QT_FEATURE_clangcpp)
+#    # If libclangTooling.a is not built with -fPIE enabled we cannot link it to lupdate.
+#    # TODO: Re-enable PIE once clang is built with PIE in provisioning.
+#    set_target_properties(${target_name} PROPERTIES POSITION_INDEPENDENT_CODE FALSE)
+#endif()
 
 qt_internal_extend_target(${target_name} CONDITION MSVC
     DEFINES _SILENCE_CXX17_ITERATOR_BASE_CLASS_DEPRECATION_WARNING)
--- a/src/qdoc/qdoc/CMakeLists.txt	2023-10-03 20:26:34.000000000 +0200
+++ -	2023-10-13 22:43:26.354327742 +0200
@@ -100,7 +100,7 @@
 
 # If libclangTooling.a is not built with -fPIE enabled we cannot link it to qdoc.
 # TODO: Re-enable PIE once clang is built with PIE in provisioning.
-set_target_properties(${target_name} PROPERTIES POSITION_INDEPENDENT_CODE FALSE)
+#set_target_properties(${target_name} PROPERTIES POSITION_INDEPENDENT_CODE FALSE)
 
 qt_internal_extend_target(${target_name} CONDITION (WIN32 AND ICC) OR MSVC
     LINK_OPTIONS
