--- qt6-tools-6.4.2.orig/tests/auto/qdoc/generatedoutput/tst_generatedoutput.cpp
+++ qt6-tools-6.4.2/tests/auto/qdoc/generatedoutput/tst_generatedoutput.cpp
@@ -74,7 +74,10 @@ private:
 void tst_generatedOutput::initTestCase()
 {
     // Build the path to the QDoc binary the same way moc tests do for moc.
-    const auto binpath = QLibraryInfo::path(QLibraryInfo::BinariesPath);
+    auto binpath = QLibraryInfo::path(QLibraryInfo::BinariesPath);
+    char *build_binpath = getenv("QT_BUILD_BIN_PATH");
+    if (build_binpath && *build_binpath)
+    	    binpath = build_binpath;
     const auto extension = QSysInfo::productType() == "windows" ? ".exe" : "";
     m_qdoc = binpath + QLatin1String("/qdoc") + extension;
     m_expectedDir.setPath(QFINDTESTDATA("expected_output"));
