--- a/src/assistant/CMakeLists.txt
+++ b/src/assistant/CMakeLists.txt
@@ -34,7 +34,9 @@ if(TARGET qlitehtml)
     qt_internal_set_exceptions_flags(litehtml OFF)
     qt_disable_warnings(litehtml)
     qt_disable_warnings(qlitehtml)
-    qt_disable_warnings(gumbo)
+    if (TARGET gumbo)
+        qt_disable_warnings(gumbo)
+    endif()
     qt_handle_multi_config_output_dirs(qlitehtml)
     set_target_properties(qlitehtml PROPERTIES
         RUNTIME_OUTPUT_DIRECTORY "${QT_BUILD_DIR}/${INSTALL_BINDIR}"
--- a/src/assistant/qlitehtml/src/3rdparty/litehtml/CMakeLists.txt
+++ b/src/assistant/qlitehtml/src/3rdparty/litehtml/CMakeLists.txt
@@ -196,52 +196,6 @@ endif()
 set_source_files_properties(${CMAKE_CURRENT_SOURCE_DIR}/src/master.css.inc PROPERTIES GENERATED TRUE)
 
 # Tests
-
-if (BUILD_TESTING)
-    include(FetchContent)
-    FetchContent_Declare(
-      googletest
-      URL https://github.com/google/googletest/archive/609281088cfefc76f9d0ce82e1ff6c30cc3591e5.zip
-    )
-    # For Windows: Prevent overriding the parent project's compiler/linker settings
-    set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
-    FetchContent_GetProperties(googletest)
-    if(NOT googletest_POPULATED)
-      FetchContent_Populate(googletest)
-      add_subdirectory(${googletest_SOURCE_DIR} ${googletest_BINARY_DIR})
-    endif()
-
-    enable_testing()
-
-    set(TEST_NAME ${PROJECT_NAME}_tests)
-
-    add_executable(
-        ${TEST_NAME}
-        ${TEST_LITEHTML}
-        ${CMAKE_CURRENT_SOURCE_DIR}/src/master.css.inc
-    )
-
-    set_target_properties(${TEST_NAME} PROPERTIES
-        CXX_STANDARD 11
-        C_STANDARD 99
-        PUBLIC_HEADER "${HEADER_LITEHTML}"
-    )
-
-    target_include_directories(
-        ${TEST_NAME}
-        PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/containers
-    )
-
-    target_link_libraries(
-        ${TEST_NAME}
-        ${PROJECT_NAME}
-        gtest_main
-    )
-
-    include(GoogleTest)
-    gtest_discover_tests(${TEST_NAME})
-endif()
-
 #     set(TEST_NAME ${PROJECT_NAME}_tests)
 #     add_executable(${TEST_NAME} ${TEST_LITEHTML} ${CMAKE_CURRENT_SOURCE_DIR}/src/master.css.inc)
 #     set_target_properties(${TEST_NAME} PROPERTIES
