diff --git a/Makefile b/Makefile
index c4c532f..572c898 100644
--- a/Makefile
+++ b/Makefile
@@ -83,7 +83,7 @@ debug: $(LIBRARY_DEBUG_$(OS)) libkrun.pc
 ifeq ($(BUILD_INIT),1)
 INIT_BINARY = init/init
 $(INIT_BINARY): $(INIT_SRC)
-	gcc -O2 -static -Wall $(INIT_DEFS) -o $@ $(INIT_SRC) $(INIT_DEFS)
+	$(CC) -O2 -static -Wall $(INIT_DEFS) -o $@ $(INIT_SRC) $(INIT_DEFS)
 endif
 
 $(LIBRARY_RELEASE_$(OS)): $(INIT_BINARY)
@@ -92,7 +92,7 @@ ifeq ($(SEV),1)
 	mv target/release/libkrun.so target/release/$(KRUN_BASE_$(OS))
 endif
 ifeq ($(OS),Linux)
-	patchelf --set-soname $(KRUN_SONAME_$(OS)) --output $(LIBRARY_RELEASE_$(OS)) target/release/$(KRUN_BASE_$(OS))
+	patchelf --set-soname $(KRUN_SONAME_$(OS)) --output $(LIBRARY_RELEASE_$(OS)) target/$(XBPS_CROSS_RUST_TARGET)/release/$(KRUN_BASE_$(OS))
 else
 ifeq ($(EFI),1)
 ifeq ($(OS),Darwin)
