From 3599e8476d11127fe49fe696fc6e2b05e7687d12 Mon Sep 17 00:00:00 2001
From: Veloman Yunkan <veloman.yunkan@gmail.com>
Date: Mon, 16 Sep 2024 17:50:20 +0400
Subject: [PATCH] libkiwix now identifies itself as libkiwix

---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 53a477c5..8b1454d7 100644
--- a/meson.build
+++ b/meson.build
@@ -21,6 +21,6 @@ endif
 
 thread_dep = dependency('threads')
 libzim_dep = dependency('libzim', version:['>=9.0.0', '<10.0.0'], static:static_linkage)
-kiwixlib_dep = dependency('kiwix', version:['>=14.0.0', '<15.0.0'], static:static_linkage)
+kiwixlib_dep = dependency('libkiwix', version:['>=14.0.0', '<15.0.0'], static:static_linkage)
 
 all_deps = [thread_dep, kiwixlib_dep, libzim_dep]
