--- kiwix-tools-3.7.0/src/server/kiwix-serve.cpp	2024-03-13 15:17:32.000000000 +0100
+++ -	2025-03-15 23:48:03.235468369 +0100
@@ -413,7 +413,7 @@
     exit(1);
   }
 
-  std::string url = "http://" + server.getAddress() + ":" + std::to_string(server.getPort()) + normalizeRootUrl(rootLocation);
+  std::string url = "http://" + server.getAddress().addr + ":" + std::to_string(server.getPort()) + normalizeRootUrl(rootLocation);
   std::cout << "The Kiwix server is running and can be accessed in the local network at: "
             << url << std::endl;
 
