--- distribute.egg-info/entry_points.txt	2012-05-18 15:46:04.000000000 -0400
+++ distribute.egg-info/entry_points.txt	2012-05-22 18:14:17.443383039 -0400
@@ -11,6 +11,8 @@
 install_egg_info = setuptools.command.install_egg_info:install_egg_info
 alias = setuptools.command.alias:alias
 easy_install = setuptools.command.easy_install:easy_install
+easy_install-2.7 = setuptools.command.easy_install:main
+easy_install-3.2 = setuptools.command.easy_install:main
 install_scripts = setuptools.command.install_scripts:install_scripts
 bdist_wininst = setuptools.command.bdist_wininst:bdist_wininst
 bdist_egg = setuptools.command.bdist_egg:bdist_egg
--- distribute_setup.py	2012-05-18 15:45:03.000000000 -0400
+++ distribute_setup.py	2012-05-22 18:14:17.443383039 -0400
@@ -299,8 +299,7 @@
         log.warn('Could not find the install location')
         return
     pyver = '%s.%s' % (sys.version_info[0], sys.version_info[1])
-    setuptools_file = 'setuptools-%s-py%s.egg-info' % \
-            (SETUPTOOLS_FAKED_VERSION, pyver)
+    setuptools_file = 'setuptools-%s.egg-info' % SETUPTOOLS_FAKED_VERSION
     pkg_info = os.path.join(placeholder, setuptools_file)
     if os.path.exists(pkg_info):
         log.warn('%s already exists', pkg_info)
--- setup.py	2012-05-18 15:19:51.000000000 -0400
+++ setup.py	2012-05-22 18:14:17.443383039 -0400
@@ -100,6 +100,7 @@
             finally:
                 f.close()
 
+console_scripts = ["easy_install-%s = setuptools.command.easy_install:main" % sys.version[:3]]

 # if we are installing Distribute using "python setup.py install"
 # we need to get setuptools out of the way
