--- a/qtbase/src/gui/Qt5GuiConfigExtras.cmake.in	2020-09-19 13:27:22.130997854 +0200
+++ b/qtbase/src/gui/Qt5GuiConfigExtras.cmake.in	2020-09-19 13:27:22.130997854 +0200
@@ -106,12 +106,6 @@
                 set(Qt5Gui_${_cmake_lib_name}_LIBRARY ${_lib})
             else()
                 find_library(Qt5Gui_${_cmake_lib_name}_LIBRARY ${_lib}
-!!IF !isEmpty(CROSS_COMPILE)
-                    PATHS \"${LibDir}\"
-!!IF !mac
-                    NO_DEFAULT_PATH
-!!ENDIF
-!!ENDIF
                 )
             endif()
 !!IF mac
