--- a/qtbase/src/corelib/codecs/qiconvcodec.cpp	2017-01-18 15:20:58.000000000 +0100
+++ b/qtbase/src/corelib/codecs/qiconvcodec.cpp	2017-02-21 14:33:32.423808603 +0100
@@ -64,7 +64,7 @@
 #elif defined(Q_OS_AIX)
 #  define NO_BOM
 #  define UTF16 "UCS-2"
-#elif defined(Q_OS_FREEBSD)
+#elif defined(Q_OS_FREEBSD) || (defined(Q_OS_LINUX) && !defined(__GLIBC__))
 #  define NO_BOM
 #  if Q_BYTE_ORDER == Q_BIG_ENDIAN
 #    define UTF16 "UTF-16BE"
