--- qtwebengine/src/3rdparty/chromium/third_party/lss/linux_syscall_support.h	2018-09-07 21:25:26.639964100 +0200
+++ qtwebengine/src/3rdparty/chromium/third_party/lss/linux_syscall_support.h	2018-09-07 21:25:11.611964995 +0200
@@ -1794,6 +1794,15 @@
 /* End of s390/s390x definitions                                             */
 #endif
 
+#ifndef __GLIBC__
+  /* For Musl libc pread/pread is the same as pread64/pwrite64 */
+#ifndef __NR_pread
+#define __NR_pread              __NR_pread64
+#endif
+#ifndef __NR_pwrite
+#define __NR_pwrite             __NR_pwrite64
+#endif
+#endif /* ifndef __GLIBC__ */
 
 /* After forking, we must make sure to only call system calls.               */
 #if defined(__BOUNDED_POINTERS__)
