--- qtwebengine/src/core/api/qtbug-61521.cpp	2017-11-29 09:42:29.000000000 +0100
+++ qtwebengine/src/core/api/qtbug-61521.cpp	2018-01-28 06:49:29.454175725 +0100
@@ -111,7 +111,11 @@
 }
 
 SHIM_HIDDEN void* ShimPvalloc(size_t size) {
+#if defined(__GLIBC__)
     return pvalloc(size);
+#else
+    return valloc((size+4095)&~4095);
+#endif
 }
 
 SHIM_HIDDEN int ShimPosixMemalign(void** r, size_t a, size_t s) {
