--- qtwebengine/src/3rdparty/chromium/base/threading/platform_thread_linux.cc.orig	2018-09-19 18:05:15.523923703 +0200
+++ qtwebengine/src/3rdparty/chromium/base/threading/platform_thread_linux.cc	2018-09-19 18:08:05.150932454 +0200
@@ -173,7 +173,12 @@
 
 size_t GetDefaultThreadStackSize(const pthread_attr_t& attributes) {
 #if !defined(THREAD_SANITIZER)
+#if defined(__GLIBC__)
   return 0;
+#else // defined(__GLIBC__)
+  // For Musl libc try with a default stack size of 2 MiB
+  return 2 * 1024 * 1024;
+#endif // !defined(__GLIBC__)
 #else
   // ThreadSanitizer bloats the stack heavily. Evidence has been that the
   // default stack size isn't enough for some browser tests.
