--- qtwebengine/src/3rdparty/chromium/third_party/skia/include/gpu/GrTypes.h.orig	2018-09-11 14:37:56.832822055 +0200
+++ qtwebengine/src/3rdparty/chromium/third_party/skia/include/gpu/GrTypes.h	2018-09-11 14:38:11.464821183 +0200
@@ -348,9 +348,6 @@
 static const int kGrPixelConfigCnt = kLast_GrPixelConfig + 1;
 
 // Aliases for pixel configs that match skia's byte order.
-#ifndef SK_CPU_LENDIAN
-    #error "Skia gpu currently assumes little endian"
-#endif
 #if SK_PMCOLOR_BYTE_ORDER(B,G,R,A)
     static const GrPixelConfig kSkia8888_GrPixelConfig = kBGRA_8888_GrPixelConfig;
 #elif SK_PMCOLOR_BYTE_ORDER(R,G,B,A)
