--- qtscript/src/3rdparty/javascriptcore/JavaScriptCore/wtf/FastMalloc.cpp	2018-10-07 21:25:14.377211462 +0200
+++ qtscript/src/3rdparty/javascriptcore/JavaScriptCore/wtf/FastMalloc.cpp	2018-10-07 21:26:13.987214537 +0200
@@ -1456,7 +1456,12 @@
   pthread_cond_init(&m_scavengeCondition, 0);
   m_scavengeThreadActive = true;
   pthread_t thread;
+  pthread_attr_t attr;
+  pthread_attr_init(&attr);
+#if !defined(__GLIBC__)
+  pthread_attr_setstacksize(&attr, 2 * 1024 * 1024);
+#endif
-  pthread_create(&thread, 0, runScavengerThread, this);
+  pthread_create(&thread, &attr, runScavengerThread, this);
 }
 
 void* TCMalloc_PageHeap::runScavengerThread(void* context)
