--- qtbase/src/network/ssl/qsslsocket_openssl.cpp	2019-06-13 06:18:18.000000000 +0200
+++ -	2019-06-17 14:59:22.501460882 +0200
@@ -604,7 +604,7 @@
             q_SSL_set_psk_server_callback(ssl, &q_ssl_psk_server_callback);
     }
 #endif
-#if OPENSSL_VERSION_NUMBER >= 0x10101006L
+#if OPENSSL_VERSION_NUMBER >= 0x10101006L && !defined(LIBRESSL_VERSION_NUMBER)
     // Set the client callback for TLSv1.3 PSK
     if (mode == QSslSocket::SslClientMode
         && QSslSocket::sslLibraryBuildVersionNumber() >= 0x10101006L) {
