--- qtwebengine/src/3rdparty/chromium//third_party/openmax_dl/dl/BUILD.gn	2019-04-09 09:31:16.000000000 +0200
+++ -	2019-05-06 23:08:09.537346546 +0200
@@ -206,18 +206,6 @@
       "sp/src/arm/armv7/omxSP_FFTFwd_RToCCS_F32_Sfs_s.S",
       "sp/src/arm/armv7/omxSP_FFTInv_CCSToR_F32_Sfs_s.S",
     ]
-    if (arm_optionally_use_neon) {
-      # Run-time NEON detection.
-      deps = [
-        "//third_party/android_tools:cpu_features",
-      ]
-
-      # To get the __android_log_print routine
-      libs = [ "log" ]
-
-      # Detection routine
-      sources += [ "sp/src/arm/detect.c" ]
-    }
   }
 
   # NEON implementation of FFT. This library is NOT
