--- qtwebengine/src/3rdparty/chromium/third_party/skia/third_party/skcms/skcms.cc
+++ qtwebengine/src/3rdparty/chromium/third_party/skia/third_party/skcms/skcms.cc
@@ -1828,6 +1828,14 @@ typedef enum {
     #endif
 #endif
 
+// with current gcc 8.3.0 we get an internal compiler error when compiling
+// for ppc64 big endian - so switch to portable until this is resolved
+#if defined(__powerpc64__)
+#  if defined(__BIG_ENDIAN__) && !defined(SKCMS_PORTABLE)
+#    define SKCMS_PORTABLE
+#  endif
+#endif
+
 #if defined(__clang__)
     template <int N, typename T> using Vec = T __attribute__((ext_vector_type(N)));
 #elif defined(__GNUC__)
