--- qtwebengine/src/3rdparty/chromium/base/allocator/partition_allocator/page_allocator_internals_posix.h	2019-10-21 10:14:54.000000000 +0200
+++ qtwebengine/src/3rdparty/chromium/base/allocator/partition_allocator/page_allocator_internals_posix.h.patched	2019-11-15 07:04:36.673637152 +0100
@@ -53,6 +53,9 @@
 // Multiple guarded memory regions may exceed the process address space limit.
 // This function will raise or lower the limit by |amount|.
 bool AdjustAddressSpaceLimit(int64_t amount) {
+#if !defined(__GLIBC__)
+  return false;
+#else
   struct rlimit old_rlimit;
   if (getrlimit(RLIMIT_AS, &old_rlimit))
     return false;
@@ -62,6 +65,7 @@
                                     old_rlimit.rlim_max};
   // setrlimit will fail if limit > old_rlimit.rlim_max.
   return setrlimit(RLIMIT_AS, &new_rlimit) == 0;
+#endif
 }
 
 // Current WASM guarded memory regions have 8 GiB of address space. There are
