--- qtwebengine/src/3rdparty/chromium/third_party/openmax_dl/dl/BUILD.gn	2017-11-28 14:06:53.000000000 +0100
+++ qtwebengine/src/3rdparty/chromium/third_party/openmax_dl/dl/BUILD.gn	2018-01-30 16:42:15.332826020 +0100
@@ -194,14 +194,6 @@
       "sp/src/arm/armv7/omxSP_FFTFwd_RToCCS_F32_Sfs_s.S",
       "sp/src/arm/armv7/omxSP_FFTInv_CCSToR_F32_Sfs_s.S",
     ]
-    if (arm_optionally_use_neon) {
-      # Run-time NEON detection.
-      deps = [ "//third_party/android_tools:cpu_features" ]
-      # To get the __android_log_print routine
-      libs = [ "log" ]
-      # Detection routine
-      sources += [ "sp/src/arm/detect.c" ]
-    }
   }
 
   # GYP: third_party/openmax_dl/dl/dl.gyp:openmax_dl_neon
