--- qtbase/config.tests/unix/alsa/alsatest.cpp	2016-03-03 12:35:22.000000000 +0100
+++ qtbase/config.tests/unix/alsa/alsatest.cpp	2016-04-24 11:21:33.168021766 +0200
@@ -32,7 +32,7 @@
 ****************************************************************************/
 
 #include <alsa/asoundlib.h>
-#if(!(SND_LIB_MAJOR == 1 && SND_LIB_MINOR == 0 && SND_LIB_SUBMINOR >= 10))
+#if(!(SND_LIB_MAJOR == 1 && (SND_LIB_MINOR > 0) || (SND_LIB_MINOR == 0 && SND_LIB_SUBMINOR >= 10)))
 #error "Alsa version found too old, require >= 1.0.10"
 #endif
 
