--- qtwebengine/src/3rdparty/chromium/third_party/pdfium/fpdfsdk/javascript/global.cpp	2017-05-20 02:18:35.281893421 +0200
+++ qtwebengine/src/3rdparty/chromium/third_party/pdfium/fpdfsdk/javascript/global.cpp	2017-05-20 02:23:36.260127027 +0200
@@ -24,7 +24,7 @@
 struct CHash;
 
 // Only needed to hash single-character strings.
-template <wchar_t N>
+template <unsigned N>
 struct CHash<N> {
   static const unsigned value = N;
 };
