--- qtwebengine/src/3rdparty/chromium/build/compiler_version.py	2016-11-07 15:46:18.000000000 +0100
+++ qtwebengine/src/3rdparty/chromium/build/compiler_version.py	2016-12-21 17:11:17.027999078 +0100
@@ -116,6 +116,9 @@
   if mode == 'host':
     environments = ['CXX_host'] + environments;
   compiler = GetEnvironFallback(environments, 'c++')
+  # Hack: remove leading "@echo compiling xyz && " from compiler string
+  if compiler.startswith("@echo"):
+    compiler = compiler[compiler.find("&&")+3:]
 
   if compiler:
     compiler_version = GetVersion(compiler, tool)
