--- qtbase/src/corelib/io/qfilesystemengine_unix.cpp.orig	2018-11-11 14:00:30.431950676 +0100
+++ qtbase/src/corelib/io/qfilesystemengine_unix.cpp	2018-11-11 14:02:57.134958244 +0100
@@ -104,11 +104,6 @@
 #  endif
 #  endif
 
-#  if !QT_CONFIG(statx) && defined(SYS_statx) && QT_HAS_INCLUDE(<linux/stat.h>)
-#    include <linux/stat.h>
-static int statx(int dirfd, const char *pathname, int flag, unsigned mask, struct statx *statxbuf)
-{ return syscall(SYS_statx, dirfd, pathname, flag, mask, statxbuf); }
-#  endif
 #endif
 
 #ifndef STATX_BASIC_STATS
