--- qttools/src/src.pro	2017-01-02 07:50:13.000000000 +0100
+++ qttools/src/src.pro	2017-12-21 12:27:33.414867228 +0100
@@ -1,20 +1,13 @@
 TEMPLATE = subdirs
 
-qtHaveModule(widgets) {
-    no-png {
-        message("Some graphics-related tools are unavailable without PNG support")
-    } else {
-        SUBDIRS = assistant \
-                  pixeltool \
-                  designer
+SUBDIRS = assistant \
+	pixeltool \
+	designer \
+	linguist \
+	qdoc \
+	qtattributionsscanner
 
-        linguist.depends = designer
-    }
-}
-
-SUBDIRS += linguist \
-    qdoc \
-    qtattributionsscanner
+linguist.depends = designer
 
 !android|android_app: SUBDIRS += qtplugininfo
 
