--- qtwebengine/src/3rdparty/chromium/third_party/WebKit/Source/wtf/SpinLock.cpp	2017-01-03 10:28:53.000000000 +0100
+++ qtwebengine/src/3rdparty/chromium/third_party/WebKit/Source/wtf/SpinLock.cpp	2017-03-05 16:31:09.299623422 +0100
@@ -29,8 +29,10 @@
 #elif COMPILER(GCC) || COMPILER(CLANG)
 #if CPU(X86_64) || CPU(X86)
 #define YIELD_PROCESSOR __asm__ __volatile__("pause")
-#elif CPU(ARM) || CPU(ARM64)
+#elif CPU(ARM64)
 #define YIELD_PROCESSOR __asm__ __volatile__("yield")
+#elif CPU(ARM)
+#define YIELD_PROCESSOR __asm__ __volatile__("nop")
 #elif CPU(MIPS)
 // The MIPS32 docs state that the PAUSE instruction is a no-op on older
 // architectures (first added in MIPS32r2). To avoid assembler errors when
