--- a/setup.py.orig	2022-09-19 15:51:09 UTC
+++ b/setup.py
@@ -36,8 +36,6 @@ class jq_build_ext(build_ext):
     def run(self):
         if not os.path.exists(_dep_build_path(".")):
             os.makedirs(_dep_build_path("."))
-        self._build_oniguruma()
-        self._build_libjq()
         build_ext.run(self)

     def _build_oniguruma(self):
@@ -87,11 +85,7 @@ jq_extension = Extension(
     "jq",
     sources=["jq.c"],
     include_dirs=[os.path.join(jq_lib_dir, "src")],
-    extra_link_args=["-lm"],
-    extra_objects=[
-        os.path.join(jq_lib_dir, ".libs/libjq.a"),
-        os.path.join(oniguruma_lib_install_dir, "lib/libonig.a"),
-    ],
+    extra_link_args=["-lm", "-ljq", "-lonig"],
 )

 setup(
