From 8c468112fe4c00f70c9bcc16698928eedf74b639 Mon Sep 17 00:00:00 2001
From: Kai Uwe Broulik <kde@privat.broulik.de>
Date: Sun, 10 Aug 2025 12:19:40 +0200
Subject: [PATCH] Fix camera selection

---
 src/qml/ImagesView.qml | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/qml/ImagesView.qml b/src/qml/ImagesView.qml
index f049700a..64e5bc1e 100644
--- a/src/qml/ImagesView.qml
+++ b/src/qml/ImagesView.qml
@@ -282,8 +282,8 @@ QQC2.StackView {
                     model: DeviceManager
                     textRole: "display"
                     visible: count>1
-                    onActivated: {
-                        DevicesModel.playingDevice = DevicesModel.data(index, ObjectSerialRole)
+                    onActivated: (row) => {
+                        DeviceManager.playingDevice = DeviceManager.deviceAt(row);
                     }
                 }
 
-- 
GitLab

