--- a/src/util.c	2019-07-15 00:30:31.000000000 +0200
+++ b/src/util.c	2019-08-31 15:56:13.580212122 +0200
@@ -5457,10 +5457,10 @@
   // NON_LSB VERSION
   // Check first for newer pthread function
 #  ifdef HAVE_PTHREAD_MUTEXATTR_SETTYPE
-  (void)pthread_mutexattr_settype(&attr, PTHREAD_MUTEX_ERRORCHECK_NP);
+  (void)pthread_mutexattr_settype(&attr, PTHREAD_MUTEX_ERRORCHECK);
 #  else
   // Use older, deprecated pthread function
-  (void)pthread_mutexattr_setkind_np(&attr, PTHREAD_MUTEX_ERRORCHECK_NP);
+  (void)pthread_mutexattr_setkind_np(&attr, PTHREAD_MUTEX_ERRORCHECK);
 #  endif  // HAVE_PTHREAD_MUTEXATTR_SETTYPE
 # endif  // __LSB__
 
