From f9c4621389dae5999ca9d2f3c8402c2512a9ea60 Mon Sep 17 00:00:00 2001
From: Jerry Casiano <JerryCasiano@gmail.com>
Date: Thu, 30 Sep 2021 18:31:10 -0400
Subject: [PATCH] - Add (skip) annotation to fsType_to_string - Issue #240

---
 lib/common/font-manager-freetype.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/common/font-manager-freetype.c b/lib/common/font-manager-freetype.c
index cd0236a4..9d3bfac6 100644
--- a/lib/common/font-manager-freetype.c
+++ b/lib/common/font-manager-freetype.c
@@ -278,7 +278,7 @@ font_manager_fsType_get_type (void)
 }
 
 /**
- * font_manager_fsType_to_string:
+ * font_manager_fsType_to_string: (skip)
  * @fstype: #FontManagerfsType
  *
  * Returns a description of the fsType field suitable for display.
