From 12a458db3c590c3f1972594e091a24cb4d65b610 Mon Sep 17 00:00:00 2001
From: Mathieu Comandon <strycore@gmail.com>
Date: Wed, 1 Jan 2020 12:47:55 -0800
Subject: [PATCH] Support both sort_new_with_model and new_with_model

---
 lutris/gui/views/store.py | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/lutris/gui/views/store.py b/lutris/gui/views/store.py
index 9e843e9e4..8fe5decf5 100644
--- a/lutris/gui/views/store.py
+++ b/lutris/gui/views/store.py
@@ -140,7 +140,11 @@ def __init__(
         self.prevent_sort_update = False  # prevent recursion with signals
         self.modelfilter = self.store.filter_new()
         self.modelfilter.set_visible_func(self.filter_view)
-        self.modelsort = Gtk.TreeModelSort.sort_new_with_model(self.modelfilter)
+        try:
+            self.modelsort = Gtk.TreeModelSort.sort_new_with_model(self.modelfilter)
+        except AttributeError:
+            # Apparently some API breaking changes on GTK minor versions.
+            self.modelsort = Gtk.TreeModelSort.new_with_model(self.modelfilter)
         self.modelsort.connect("sort-column-changed", self.on_sort_column_changed)
         self.modelsort.set_sort_func(sort_col, sort_func, sort_col)
         self.sort_view(sort_key, sort_ascending)
