From ed0b153f320a95a11bd561726d2405823573500e Mon Sep 17 00:00:00 2001
From: Adam Jackson <ajax@redhat.com>
Date: Thu, 10 Jan 2013 03:16:26 +0000
Subject: Remove mibstore.h

Signed-off-by: Adam Jackson <ajax@redhat.com>
---
diff --git a/src/geode_driver.c b/src/geode_driver.c
index b27bfd5..9cd0af9 100644
--- src/geode_driver.c
+++ src/geode_driver.c
@@ -59,8 +59,6 @@
 /* Machine independent stuff */
 #include "mipointer.h"
 #include "micmap.h"
-/* All drivers implementing backing store need this */
-#include "mibstore.h"
 #include "vgaHW.h"
 #include "vbe.h"
 
diff --git a/src/gx_driver.c b/src/gx_driver.c
index 5268f29..7f44e19 100644
--- src/gx_driver.c
+++ src/gx_driver.c
@@ -1350,7 +1350,6 @@ GXScreenInit(SCREEN_INIT_ARGS_DECL)
     if (!pGeode->NoAccel)
         GXAccelInit(pScrn);
 
-    miInitializeBackingStore(pScrn);
     xf86SetBackingStore(pScrn);
 
     /* Set up the soft cursor */
diff --git a/src/lx_driver.c b/src/lx_driver.c
index 6d7f37f..146578e 100644
--- src/lx_driver.c
+++ src/lx_driver.c
@@ -932,7 +932,6 @@ LXScreenInit(SCREEN_INIT_ARGS_DECL)
     if (!pGeode->NoAccel)
         pGeode->NoAccel = LXExaInit(pScrn) ? FALSE : TRUE;
 
-    miInitializeBackingStore(pScrn);
     xf86SetBackingStore(pScrn);
 
     /* Set up the soft cursor */
--
cgit v0.9.0.2-2-gbebe
