--- projectlibre.sh
+++ projectlibre.sh
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 #
 #The contents of this file are subject to the Common Public Attribution License
 #Version 1.0 (the "License"); you may not use this file except in compliance with
@@ -179,7 +179,7 @@
 	OPENPROJ_HOME="$OPENPROJ_HOME0"
 	#OPENPROJ_HOME can be harmful with multiple versions. disable
 	JAVA_OK="1"
-	run_openproj --silentlyFail true "$@" || if [[ $? -eq 126 || $? -eq 127 || $? -eq 64 ]]; then
+	run_openproj --silentlyFail true "$@" || if [ $? -eq 126 -o $? -eq 127 -o $? -eq 64 ]; then
 		#126 command invoked cannot execute
 		#127 command not found
 		#64 bad java version/impl
