--- a/src/VBox/Frontends/VirtualBox/nls/ApprovedLanguages.kmk   2012-10-18 16:22:52.000000000 +0200
+++ b/src/VBox/Frontends/VirtualBox/nls/ApprovedLanguages.kmk   2012-10-19 03:03:52.640867388 +0200
@@ -23,7 +23,6 @@
	de    \
	en    \
	es    \
-	fa_IR \
	hu    \
	it    \
	ja    \

--- a/src/VBox/Frontends/VirtualBox/VBoxUI.pro	2012-10-18 16:22:47.000000000 +0200
+++ b/src/VBox/Frontends/VirtualBox/VBoxUI.pro	2012-10-20 02:48:37.525765901 +0200
@@ -63,7 +63,6 @@
	nls/VirtualBox_es.ts \
	nls/VirtualBox_eu.ts \
	nls/VirtualBox_fi.ts \
-	nls/VirtualBox_fa_IR.ts \
	nls/VirtualBox_fr.ts \
	nls/VirtualBox_gl_ES.ts \
	nls/VirtualBox_hu.ts \
