--- src/VBox/RDP/client-1.8.3/Makefile.kmk.orig	2015-05-14 07:23:17.351248014 +0200
+++ src/VBox/RDP/client-1.8.3/Makefile.kmk	2015-05-14 07:23:34.138442785 +0200
@@ -41,7 +41,7 @@ rdesktop-vrdp_DEFS = \
 	EGD_SOCKET=\"/var/run/egd-pool\" WITH_RDPSND=1 HAVE_DIRENT_H=1 \
 	HAVE_ICONV=1 HAVE_SYS_STATVFS_H=1 \
 	HAVE_SYS_PARAM_H=1 HAVE_SYS_MOUNT_H=1 HAVE_STRUCT_STATVFS_F_NAMEMAX=1 \
-	KEYMAP_PATH=$(if $(findstring RTPATH_APP_PRIVATE,$(CDEFS)),RTPATH_APP_PRIVATE,\"/opt/VirtualBox\")\"/rdesktop-vrdp-keymaps\" \
+	KEYMAP_PATH=$(if $(findstring RTPATH_APP_PRIVATE,$(CDEFS)),RTPATH_APP_PRIVATE,\"/usr/share/virtualbox\")\"/rdesktop-vrdp-keymaps\" \
 	RDESKTOP=1
 ifdef WITH_BIRD_VD_HACKS
  rdesktop-vrdp_DEFS += WITH_BIRD_VD_HACKS
