--- src/VBox/RDP/client/Makefile.kmk	2013-02-19 23:12:20.207464901 +0100
+++ src/VBox/RDP/client/Makefile.kmk	2014-01-04 04:17:26.700618600 +0100
@@ -41,7 +41,7 @@
 	EGD_SOCKET=\"/var/run/egd-pool\" WITH_RDPSND=1 HAVE_DIRENT_H=1 \
 	HAVE_ICONV=1 HAVE_SYS_STATVFS_H=1 \
 	HAVE_SYS_PARAM_H=1 HAVE_SYS_MOUNT_H=1 HAVE_STRUCT_STATVFS_F_NAMEMAX=1 \
-	KEYMAP_PATH=$(if $(findstring RTPATH_APP_PRIVATE,$(DEFS)),RTPATH_APP_PRIVATE,\"/opt/VirtualBox\")\"/rdesktop-vrdp-keymaps\" \
+	KEYMAP_PATH=$(if $(findstring RTPATH_APP_PRIVATE,$(DEFS)),RTPATH_APP_PRIVATE,\"/usr/share/virtualbox\")\"/rdesktop-vrdp-keymaps\" \
 	RDESKTOP=1
 ifdef WITH_BIRD_VD_HACKS
  rdesktop-vrdp_DEFS += WITH_BIRD_VD_HACKS
