--- src/VBox/Additions/x11/vboxvideo/vboxvideo.h	2015-03-04 00:38:45.381480285 +0100
+++ src/VBox/Additions/x11/vboxvideo/vboxvideo.h	2015-03-04 00:40:57.536942966 +0100
@@ -130,15 +130,7 @@
 #define VBOX_MAX_DRAWABLES    256          /* At random. */
 
 #define VBOXPTR(p) ((VBOXPtr)((p)->driverPrivate))
-
-/** Helper to work round different ways of getting the root window in different
- * server versions. */
-#if defined(XORG_VERSION_CURRENT) && XORG_VERSION_CURRENT < 70000000 \
-    && XORG_VERSION_CURRENT >= 10900000
-# define ROOT_WINDOW(pScrn) screenInfo.screens[(pScrn)->scrnIndex]->root
-#else
-# define ROOT_WINDOW(pScrn) WindowTable[(pScrn)->scrnIndex]
-#endif
+#define ROOT_WINDOW(pScrn) screenInfo.screens[(pScrn)->scrnIndex]->root
 
 /** Structure containing all virtual monitor-specific information. */
 struct VBoxScreen
