Index: virtualbox-ose-7.0.4/src/VBox/Frontends/VirtualBox/src/networking/UINetworkReply.h
===================================================================
--- virtualbox-ose-7.0.4.orig/src/VBox/Frontends/VirtualBox/src/networking/UINetworkReply.h
+++ virtualbox-ose-7.0.4/src/VBox/Frontends/VirtualBox/src/networking/UINetworkReply.h
@@ -32,6 +32,7 @@
 #endif
 
 /* Qt includes: */
+#include <QObject>
 #include <QPointer>
 #include <QUrl>
 
