diff -Nur VirtualBox-4.1.0_OSE.orig/src/VBox/Additions/common/crOpenGL/fakedri_drv.c VirtualBox-4.1.0_OSE/src/VBox/Additions/common/crOpenGL/fakedri_drv.c
--- VirtualBox-4.1.0_OSE.orig/src/VBox/Additions/common/crOpenGL/fakedri_drv.c	2011-07-19 15:11:37.903456612 +0000
+++ VirtualBox-4.1.0_OSE/src/VBox/Additions/common/crOpenGL/fakedri_drv.c	2011-07-19 15:12:41.024021274 +0000
@@ -36,13 +36,8 @@
 //#define DEBUG_DRI_CALLS
 
 //@todo this could be different...
-#ifdef RT_ARCH_AMD64
-# define DRI_DEFAULT_DRIVER_DIR "/usr/lib64/dri:/usr/lib/dri:/usr/lib/x86_64-linux-gnu/dri"
+# define DRI_DEFAULT_DRIVER_DIR "/usr/lib/xorg/modules/dri"
 # define DRI_XORG_DRV_DIR "/usr/lib/xorg/modules/drivers/"
-#else
-# define DRI_DEFAULT_DRIVER_DIR "/usr/lib/dri:/usr/lib/i386-linux-gnu/dri"
-# define DRI_XORG_DRV_DIR "/usr/lib/xorg/modules/drivers/"
-#endif
 
 #ifdef DEBUG_DRI_CALLS
  #define SWDRI_SHOWNAME(pext, func) \
