--- configure.orig	2011-04-21 03:29:30.000000000 -0700
+++ configure		2011-04-21 11:42:57.517216676 -0700
@@ -412,9 +412,9 @@
       elif [ $cc_maj -lt 3 \
              -o \( $cc_maj -eq 3 -a $cc_min -lt 2 \) \
              -o \( $cc_maj -eq 4 -a $cc_min -lt 1 -a "$OS" != "darwin" \) \
-             -o \( $cc_maj -eq 4 -a $cc_min -gt 5 \) \
+             -o \( $cc_maj -eq 4 -a $cc_min -gt 6 \) \
              -o $cc_maj -gt 4 ]; then
-        log_failure "gcc version $cc_ver found, expected gcc 3.x with x>1 or gcc 4.x with 0<x<5"
+        log_failure "gcc version $cc_ver found, expected gcc 3.x with x>1 or gcc 4.x with 0<x<=6"
         fail really
       else
         log_success "found version $cc_ver"
