Index: Config.kmk
===================================================================
--- foo.orig/Config.kmk	(revision 37751)
+++ foo/Config.kmk	(revision 37752)
@@ -409,6 +409,9 @@
 ifdef VBOX_WITH_USB
  VBOX_WITH_VUSB = 1
 endif
+ifdef VBOX_WITH_EHCI
+ VBOX_WITH_EHCI_IMPL = 1
+endif
 # Enable the hal/sysfs USB code on Linux.
 ifeq ($(KBUILD_TARGET),linux)
  VBOX_USB_WITH_SYSFS = 1
@@ -920,7 +923,7 @@
 #
 ifdef VBOX_OSE
  VBOX_WITH_VRDP=
- VBOX_WITH_EHCI=
+ VBOX_WITH_EHCI_IMPL=
  VBOX_WITH_EXTPACK_PUEL=
  VBOX_WITH_PCI_PASSTHROUGH=
  VBOX_WITH_OS2_ADDITIONS_BIN=
Index: src/VBox/Devices/Makefile.kmk
===================================================================
--- foo.orig/src/VBox/Devices/Makefile.kmk	(revision 37751)
+++ foo/src/VBox/Devices/Makefile.kmk	(revision 37752)
@@ -57,8 +57,8 @@
  ifdef VBOX_WITH_USB
   VBoxDDU_DEFS           += VBOX_WITH_USB IN_USBLIB
   VBoxDDU_SDKS.win        = WINPSDK W2K3DDK
-  if defined(VBOX_WITH_EHCI) && !defined(VBOX_WITH_EXTPACK_PUEL)
-   VBoxDDU_DEFS          += VBOX_WITH_EHCI
+  if defined(VBOX_WITH_EHCI_IMPL) && !defined(VBOX_WITH_EXTPACK_PUEL)
+   VBoxDDU_DEFS          += VBOX_WITH_EHCI_IMPL
   endif
   ifdef VBOX_WITH_NEW_USB_CODE_ON_DARWIN
    VBoxDDU_DEFS.darwin   += VBOX_WITH_NEW_USB_CODE_ON_DARWIN
@@ -122,8 +122,8 @@
  VBoxDD_DEFS             = VBOX_ACPI
  ifdef VBOX_WITH_USB
   VBoxDD_DEFS           += VBOX_WITH_USB
-  if defined(VBOX_WITH_EHCI) && !defined(VBOX_WITH_EXTPACK_PUEL)
-   VBoxDD_DEFS          += VBOX_WITH_EHCI
+  if defined(VBOX_WITH_EHCI_IMPL) && !defined(VBOX_WITH_EXTPACK_PUEL)
+   VBoxDD_DEFS          += VBOX_WITH_EHCI_IMPL
   endif
  endif
  ifdef VBOX_WITH_VUSB
@@ -448,8 +448,8 @@
  endif
  ifdef VBOX_WITH_USB
   DevicesR3_DEFS        += VBOX_WITH_USB
-  if defined(VBOX_WITH_EHCI) && !defined(VBOX_WITH_EXTPACK_PUEL)
-   DevicesR3_DEFS       += VBOX_WITH_EHCI
+  if defined(VBOX_WITH_EHCI_IMPL) && !defined(VBOX_WITH_EXTPACK_PUEL)
+   DevicesR3_DEFS       += VBOX_WITH_EHCI_IMPL
    DevicesR3_SOURCES    += \
  	USB/DevEHCI.cpp
   endif
@@ -603,8 +603,8 @@
 
   ifdef VBOX_WITH_USB
    VBoxDDGC_DEFS        += VBOX_WITH_USB
-   if defined(VBOX_WITH_EHCI) && !defined(VBOX_WITH_EXTPACK_PUEL)
-    VBoxDDGC_DEFS       += VBOX_WITH_EHCI
+   if defined(VBOX_WITH_EHCI_IMPL) && !defined(VBOX_WITH_EXTPACK_PUEL)
+    VBoxDDGC_DEFS       += VBOX_WITH_EHCI_IMPL
     VBoxDDGC_SOURCES    += \
   	USB/DevEHCI.cpp
    endif
@@ -708,8 +708,8 @@
  endif
  ifdef VBOX_WITH_USB
  VBoxDDR0_DEFS          += VBOX_WITH_USB
-  if defined(VBOX_WITH_EHCI) && !defined(VBOX_WITH_EXTPACK_PUEL)
-   VBoxDDR0_DEFS        += VBOX_WITH_EHCI
+  if defined(VBOX_WITH_EHCI_IMPL) && !defined(VBOX_WITH_EXTPACK_PUEL)
+   VBoxDDR0_DEFS        += VBOX_WITH_EHCI_IMPL
    VBoxDDR0_SOURCES     += \
  	USB/DevEHCI.cpp
   endif
Index: src/VBox/Devices/build/VBoxDD.cpp
===================================================================
--- foo.orig/src/VBox/Devices/build/VBoxDD.cpp	(revision 37751)
+++ foo/src/VBox/Devices/build/VBoxDD.cpp	(revision 37752)
@@ -132,7 +132,7 @@
     if (RT_FAILURE(rc))
         return rc;
 #endif
-#ifdef VBOX_WITH_EHCI
+#ifdef VBOX_WITH_EHCI_IMPL
     rc = pCallbacks->pfnRegister(pCallbacks, &g_DeviceEHCI);
     if (RT_FAILURE(rc))
         return rc;
Index: src/VBox/Devices/testcase/tstDeviceStructSizeRC.cpp
===================================================================
--- foo.orig/src/VBox/Devices/testcase/tstDeviceStructSizeRC.cpp	(revision 37751)
+++ foo/src/VBox/Devices/testcase/tstDeviceStructSizeRC.cpp	(revision 37752)
@@ -62,7 +62,7 @@
 #ifdef VBOX_WITH_USB
 # undef LOG_GROUP
 # include "../USB/DevOHCI.cpp"
-# ifdef VBOX_WITH_EHCI
+# ifdef VBOX_WITH_EHCI_IMPL
 #  include "../USB/DevEHCI.cpp"
 # endif
 #endif
@@ -928,7 +928,7 @@
     GEN_CHECK_OFF(OHCI, StatDroppedUrbs);
     GEN_CHECK_OFF(OHCI, StatTimer);
 # endif
-# ifdef VBOX_WITH_EHCI
+# ifdef VBOX_WITH_EHCI_IMPL
     /* USB/DevEHCI.cpp */
     GEN_CHECK_SIZE(EHCIHUBPORT);
     GEN_CHECK_OFF(EHCIHUBPORT, fReg);
@@ -1005,7 +1005,7 @@
     GEN_CHECK_OFF(EHCI, pEOFTimerNoSyncRC);
     GEN_CHECK_OFF(EHCI, pEOFTimerNoSyncR3);
     GEN_CHECK_OFF(EHCI, pEOFTimerNoSyncR0);
-# endif /* VBOX_WITH_EHCI */
+# endif /* VBOX_WITH_EHCI_IMPL */
 #endif /* VBOX_WITH_USB */
 
     /* VMMDev/VBoxDev.cpp */
Index: src/VBox/Devices/testcase/tstDeviceStructSize.cpp
===================================================================
--- foo.orig/src/VBox/Devices/testcase/tstDeviceStructSize.cpp	(revision 37751)
+++ foo/src/VBox/Devices/testcase/tstDeviceStructSize.cpp	(revision 37752)
@@ -67,7 +67,7 @@
 #ifdef VBOX_WITH_USB
 # undef LOG_GROUP
 # include "../USB/DevOHCI.cpp"
-# ifdef VBOX_WITH_EHCI
+# ifdef VBOX_WITH_EHCI_IMPL
 #  include "../USB/DevEHCI.cpp"
 # endif
 #endif
@@ -284,7 +284,7 @@
 #endif
     //CHECK_MEMBER_ALIGNMENT(E1KSTATE, csTx, 8);
 #ifdef VBOX_WITH_USB
-# ifdef VBOX_WITH_EHCI
+# ifdef VBOX_WITH_EHCI_IMPL
     CHECK_MEMBER_ALIGNMENT(EHCI, RootHub, 8);
 #  ifdef VBOX_WITH_STATISTICS
     CHECK_MEMBER_ALIGNMENT(EHCI, StatCanceledIsocUrbs, 8);
Index: src/VBox/Devices/testcase/Makefile.kmk
===================================================================
--- foo.orig/src/VBox/Devices/testcase/Makefile.kmk	(revision 37751)
+++ foo/src/VBox/Devices/testcase/Makefile.kmk	(revision 37752)
@@ -28,7 +28,7 @@
 VBOX_DEVICES_TESTS_FEATURES = \
 	$(if $(VBOX_WITH_RAW_MODE),VBOX_WITH_RAW_MODE,) \
 	$(if $(VBOX_WITH_USB),VBOX_WITH_USB,) \
-	$(if $(VBOX_WITH_EHCI),VBOX_WITH_EHCI,) \
+	$(if $(VBOX_WITH_EHCI_IMPL),VBOX_WITH_EHCI_IMPL,) \
 	$(if $(VBOX_WITH_AHCI),VBOX_WITH_AHCI,) \
 	$(if $(VBOX_WITH_E1000),VBOX_WITH_E1000,) \
 	$(if $(VBOX_WITH_VIRTIO),VBOX_WITH_VIRTIO,) \
