--- src/VBox/Additions/x11/vboxvideo/edid.c	2014-11-21 16:16:53.000000000 +0100
+++ src/VBox/Additions/x11/vboxvideo/edid.c	2015-02-05 07:37:15.213061546 +0100
@@ -44,6 +44,7 @@
  *    Dave Airlie <airlied@redhat.com>
  */
 
+#include <xorg/xorg-server.h>
 #include <misc.h>
 #include <xf86DDC.h>
 #include <xf86Crtc.h>
--- src/VBox/Additions/x11/vboxvideo/pointer.c	2014-11-21 16:16:53.000000000 +0100
+++ src/VBox/Additions/x11/vboxvideo/pointer.c	2015-02-05 07:48:43.910512564 +0100
@@ -22,6 +22,7 @@
 # include <Pci.h>
 #endif
 
+#include <xorg/xorg-server.h>
 #include "xf86.h"
 #define NEED_XF86_TYPES
 #include <iprt/string.h>
--- src/VBox/Additions/x11/vboxvideo/vboxvideo.c	2014-11-21 16:16:53.000000000 +0100
+++ src/VBox/Additions/x11/vboxvideo/vboxvideo.c	2015-02-05 07:38:16.019638643 +0100
@@ -59,6 +59,7 @@
 # include <string.h>
 #endif
 
+#include <xorg/xorg-server.h>
 #include "xf86.h"
 #include "xf86_OSproc.h"
 #if GET_ABI_MAJOR(ABI_VIDEODRV_VERSION) < 6
