diff --git buku buku
index b93bdb4..1a55f76 100755
--- buku
+++ buku
@@ -41,7 +41,6 @@ try:
 except ImportError:
     pass
 from bs4 import BeautifulSoup
-import certifi
 import urllib3
 from urllib3.exceptions import LocationParseError
 from urllib3.util import parse_url, make_headers
@@ -88,6 +87,8 @@ LOGGER = logging.getLogger()
 LOGDBG = LOGGER.debug
 LOGERR = LOGGER.error
 
+def where():
+    return '/etc/ssl/certs/ca-certificates.crt'
 
 class BukuCrypt:
     """Class to handle encryption and decryption of
@@ -2553,12 +2554,12 @@ class BukuDb:
                 num_pools=1,
                 headers=MYHEADERS,
                 cert_reqs='CERT_REQUIRED',
-                ca_certs=certifi.where())
+                ca_certs=where())
         else:
             manager = urllib3.PoolManager(num_pools=1,
                                           headers={'User-Agent': USER_AGENT},
                                           cert_reqs='CERT_REQUIRED',
-                                          ca_certs=certifi.where())
+                                          ca_certs=where())
 
         try:
             r = manager.request(
@@ -3395,14 +3396,14 @@ def get_PoolManager():
 
     if MYPROXY:
         return urllib3.ProxyManager(MYPROXY, num_pools=1, headers=MYHEADERS, timeout=15,
-                                    cert_reqs='CERT_REQUIRED', ca_certs=certifi.where())
+                                    cert_reqs='CERT_REQUIRED', ca_certs=where())
 
     return urllib3.PoolManager(
         num_pools=1,
         headers=MYHEADERS,
         timeout=15,
         cert_reqs='CERT_REQUIRED',
-        ca_certs=certifi.where())
+        ca_certs=where())
 
 
 def network_handler(url, http_head=False):
@@ -4224,13 +4225,13 @@ def check_upstream_release():
             num_pools=1,
             headers=MYHEADERS,
             cert_reqs='CERT_REQUIRED',
-            ca_certs=certifi.where()
+            ca_certs=where()
         )
     else:
         manager = urllib3.PoolManager(num_pools=1,
                                       headers={'User-Agent': USER_AGENT},
                                       cert_reqs='CERT_REQUIRED',
-                                      ca_certs=certifi.where())
+                                      ca_certs=where())
 
     try:
         r = manager.request(
diff --git setup.py setup.py
index 91f9fbb..b6f4cfa 100644
--- setup.py
+++ setup.py
@@ -57,7 +57,6 @@ setup(
     py_modules=['buku'],
     install_requires=[
         'beautifulsoup4>=4.4.1',
-        'certifi',
         'cryptography>=1.2.3',
         'urllib3>=1.23',
         'html5lib>=1.0.1',
