--- a/test/test-list.h	2026-07-23 17:58:08.555904735 +0200
+++ b/test/test-list.h	2026-03-06 21:05:38.000000000 +0100
@@ -379,7 +379,7 @@
 #ifdef _WIN32
 TEST_FS_DECLARE   (fs_unlink_archive_readonly)
 #endif
-TEST_FS_DECLARE   (fs_chown)
+/* TEST_FS_DECLARE   (fs_chown) */
 TEST_FS_DECLARE   (fs_link)
 TEST_FS_DECLARE   (fs_readlink)
 TEST_FS_DECLARE   (fs_realpath)
@@ -1107,7 +1107,7 @@
 #ifdef _WIN32
   TEST_FS_ENTRY  (fs_unlink_archive_readonly)
 #endif
-  TEST_FS_ENTRY  (fs_chown)
+  /* TEST_FS_ENTRY  (fs_chown) */
   TEST_FS_ENTRY  (fs_link)
   TEST_FS_ENTRY  (fs_utime)
   TEST_FS_ENTRY  (fs_utime_round)
