--- test/test-list.h.orig	2018-04-16 19:13:01.184197428 +0200
+++ test/test-list.h	2018-04-16 19:13:28.144260600 +0200
@@ -287,7 +287,7 @@ TEST_DECLARE   (fs_unlink_readonly)
 #ifdef _WIN32
 TEST_DECLARE   (fs_unlink_archive_readonly)
 #endif
-TEST_DECLARE   (fs_chown)
+/* TEST_DECLARE   (fs_chown) */
 TEST_DECLARE   (fs_link)
 TEST_DECLARE   (fs_readlink)
 TEST_DECLARE   (fs_realpath)
@@ -821,7 +821,7 @@ TASK_LIST_START
 #ifdef _WIN32
   TEST_ENTRY  (fs_unlink_archive_readonly)
 #endif
-  TEST_ENTRY  (fs_chown)
+  /* TEST_ENTRY  (fs_chown) */
   TEST_ENTRY  (fs_utime)
   TEST_ENTRY  (fs_futime)
   TEST_ENTRY  (fs_readlink)
