--- a/test/test-list.h	2021-03-02 21:59:17.718832425 +1100
+++ b/test/test-list.h	2021-03-02 21:59:41.809157112 +1100
@@ -344,7 +344,7 @@
 #ifdef _WIN32
 TEST_DECLARE   (fs_unlink_archive_readonly)
 #endif
-TEST_DECLARE   (fs_chown)
+/* TEST_DECLARE   (fs_chown) */
 TEST_DECLARE   (fs_link)
 TEST_DECLARE   (fs_readlink)
 TEST_DECLARE   (fs_realpath)
@@ -997,7 +997,7 @@
 #ifdef _WIN32
   TEST_ENTRY  (fs_unlink_archive_readonly)
 #endif
-  TEST_ENTRY  (fs_chown)
+  /* TEST_ENTRY  (fs_chown) */
   TEST_ENTRY  (fs_utime)
   TEST_ENTRY  (fs_utime_round)
   TEST_ENTRY  (fs_futime)
