--- test/test-list.h
+++ test/test-list.h
@@ -279,7 +279,7 @@ TEST_DECLARE   (fs_access)
 TEST_DECLARE   (fs_chmod)
 TEST_DECLARE   (fs_copyfile)
 TEST_DECLARE   (fs_unlink_readonly)
-TEST_DECLARE   (fs_chown)
+/* TEST_DECLARE   (fs_chown) */
 TEST_DECLARE   (fs_link)
 TEST_DECLARE   (fs_readlink)
 TEST_DECLARE   (fs_realpath)
@@ -797,7 +797,7 @@ TASK_LIST_START
   TEST_ENTRY  (fs_chmod)
   TEST_ENTRY  (fs_copyfile)
   TEST_ENTRY  (fs_unlink_readonly)
-  TEST_ENTRY  (fs_chown)
+  /* TEST_ENTRY  (fs_chown) */
   TEST_ENTRY  (fs_utime)
   TEST_ENTRY  (fs_futime)
   TEST_ENTRY  (fs_readlink)
