diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7007a12..aa5da09 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -23,7 +23,7 @@ set(QT_MINIMUM_VERSION "6.6.0")
 set(GLIB_MINIMUM_VERSION "2.41.0") # Mime Apps new implementation
 
 find_package(lxqt2-build-tools ${LXQTBT_MINIMUM_VERSION} REQUIRED)
-find_package(Qt6 ${QT_MINIMUM_VERSION} CONFIG REQUIRED Widgets Svg Xml DBus)
+find_package(Qt6 ${QT_MINIMUM_VERSION} CONFIG REQUIRED Widgets GuiPrivate Svg Xml DBus)
 find_package(GLIB ${GLIB_MINIMUM_VERSION} REQUIRED COMPONENTS gobject gio gio-unix)
 find_package(XTerm)
 
diff --git a/src/qtxdg/CMakeLists.txt b/src/qtxdg/CMakeLists.txt
index edd3741..a86278f 100644
--- a/src/qtxdg/CMakeLists.txt
+++ b/src/qtxdg/CMakeLists.txt
@@ -84,6 +84,8 @@ target_link_libraries(${QTXDGX_LIBRARY_NAME}
         ${GLIB_LIBRARIES}
         ${GLIB_GOBJECT_LIBRARIES}
         ${GLIB_GIO_LIBRARIES}
+    PRIVATE
+        Qt6::GuiPrivate
 )
 
 set_target_properties(${QTXDGX_LIBRARY_NAME} PROPERTIES
diff --git a/src/xdgiconloader/CMakeLists.txt b/src/xdgiconloader/CMakeLists.txt
index fba72e9..4a7c215 100644
--- a/src/xdgiconloader/CMakeLists.txt
+++ b/src/xdgiconloader/CMakeLists.txt
@@ -53,6 +53,8 @@ target_link_libraries(${QTXDGX_ICONLOADER_LIBRARY_NAME}
     PUBLIC
         Qt6::Gui
         Qt6::Svg
+    PRIVATE
+        Qt6::GuiPrivate
 )
 
 set_target_properties(${QTXDGX_ICONLOADER_LIBRARY_NAME}
diff --git a/src/xdgiconloader/plugin/CMakeLists.txt b/src/xdgiconloader/plugin/CMakeLists.txt
index 04564c4..89a27a5 100644
--- a/src/xdgiconloader/plugin/CMakeLists.txt
+++ b/src/xdgiconloader/plugin/CMakeLists.txt
@@ -14,6 +14,8 @@ target_link_libraries(${QTXDGX_ICONENGINEPLUGIN_LIBRARY_NAME}
     PUBLIC
         Qt6::Gui
         "${QTXDGX_ICONLOADER_LIBRARY_NAME}"
+    PRIVATE
+        Qt6::GuiPrivate
 )
 
 target_include_directories("${QTXDGX_ICONENGINEPLUGIN_LIBRARY_NAME}"
