From f34f482530170972947b259844d00830a15f81a4 Mon Sep 17 00:00:00 2001
From: chedi toueiti <chedi.toueiti@gmail.com>
Date: Tue, 12 Jul 2022 21:01:31 +0100
Subject: [PATCH] Removing deprecated pytest-runner and adding missing test
 dependencies

---
 setup.cfg | 5 +++++
 setup.py  | 4 ++--
 2 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/setup.cfg b/setup.cfg
index 21c0a34..db0c26b 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -23,3 +23,8 @@ test = pytest
 [tool:pytest]
 collect_ignore = ['setup.py']
 
+[options.extras_require]
+test =
+    pytest >= 3
+    rfc3987
+    hypothesis
diff --git a/setup.py b/setup.py
index 53ebea7..1652285 100644
--- a/setup.py
+++ b/setup.py
@@ -10,9 +10,9 @@
 
 requirements = []
 
-setup_requirements = ['pytest-runner', ]
+setup_requirements = []
 
-test_requirements = ['pytest>=3', ]
+test_requirements = []
 
 setup(
     author="Nicolas Aimetti",
