From: =?utf-8?q?Daniel_Gr=C3=B6ber?= <dxld@darkboxed.org>
Date: Tue, 13 Jun 2023 12:50:07 +0200
Subject: Remove build-date/time reproducibility hazard

---
 src/base/cmd/cmdUtils.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/src/base/cmd/cmdUtils.c
+++ b/src/base/cmd/cmdUtils.c
@@ -593,7 +593,7 @@ void CmdCommandPrint( Abc_Frame_t * pAbc
     nColumns = 79 / (LenghtMax + 2);
 
     // print the starting message 
-    fprintf( pAbc->Out, "      Welcome to ABC compiled on %s %s!", __DATE__, __TIME__ );
+    fprintf( pAbc->Out, "      Welcome to ABC!");
 
     // print the command by group
     sGroupCur = NULL;
