--- a/src/hts/hts_engine_API/lib/HTS_misc.c	2020-05-06 08:48:23.207180474 -0400
+++ b/src/hts/hts_engine_API/lib/HTS_misc.c	2020-05-06 08:49:38.850713440 -0400
@@ -247,6 +247,8 @@
       fgetpos((FILE *) fp->pointer, &pos);
 #if defined(_WIN32) || defined(__CYGWIN__) || defined(__APPLE__) || defined(__ANDROID__)
       return (size_t) pos;
+#elif defined(__linux__) && !defined(__GLIBC__) /* musl */
+      return (size_t) pos.__lldata;
 #else
       return (size_t) pos.__pos;
 #endif                          /* _WIN32 || __CYGWIN__ || __APPLE__ || __ANDROID__ */
