--- a/src/api/KWallet/KF6WalletConfig.cmake.in
+++ b/src/api/KWallet/KF6WalletConfig.cmake.in
@@ -1,7 +1,11 @@
 @PACKAGE_INIT@
 
+set(XBPS_PACKAGE_PREFIX_DIR "${PACKAGE_PREFIX_DIR}")
+set(PACKAGE_PREFIX_DIR "/usr")
 set(KWALLET_DBUS_INTERFACES_DIR "@PACKAGE_KDE_INSTALL_DBUSINTERFACEDIR@")
 set(KWALLETD_BIN_PATH "@PACKAGE_KDE_INSTALL_BINDIR@/kwalletd6")
+set(PACKAGE_PREFIX_DIR "${XBPS_PACKAGE_PREFIX_DIR}")
+unset(XBPS_PACKAGE_PREFIX_DIR)
 
 include(CMakeFindDependencyMacro)
 find_dependency(Qt6Gui @REQUIRED_QT_VERSION@)
