--- a/cmake/Libraries.cmake
+++ b/cmake/Libraries.cmake
@@ -25,7 +25,7 @@ if(IOS OR ANDROID)
 endif()
 
 set(Qt Qt6)
-find_package(${Qt} ${MIN_QT_VERSION} REQUIRED COMPONENTS Core Concurrent Network StateMachine CMAKE_FIND_ROOT_PATH_BOTH)
+find_package(${Qt} ${MIN_QT_VERSION} REQUIRED COMPONENTS Core CorePrivate Concurrent Network StateMachine CMAKE_FIND_ROOT_PATH_BOTH)
 set(QT_VERSION "${Qt6Core_VERSION}")
 
 if(NOT CONTAINER_SDK)
