--- clang/lib/Driver/ToolChains/Linux.cpp	2018-12-16 23:52:16.174867512 +0100
+++ clang/lib/Driver/ToolChains/Linux.cpp	2018-12-16 23:56:25.040531791 +0100
@@ -502,12 +502,12 @@
     Loader = "ld.so.1";
     break;
   case llvm::Triple::ppc64:
-    LibDir = "lib64";
+    LibDir = "lib";
     Loader =
         (tools::ppc::hasPPCAbiArg(Args, "elfv2")) ? "ld64.so.2" : "ld64.so.1";
     break;
   case llvm::Triple::ppc64le:
-    LibDir = "lib64";
+    LibDir = "lib";
     Loader =
         (tools::ppc::hasPPCAbiArg(Args, "elfv1")) ? "ld64.so.1" : "ld64.so.2";
     break;
