--- a/templates/qgo.desktop	2022-04-27 13:03:23.000000000 -0500
+++ b/templates/qgo.desktop	2022-05-16 18:26:45.485870905 -0500
@@ -1,10 +1,10 @@
 [Desktop Entry]
 Encoding=UTF-8
-Comment=qgo
-Exec=qgo
-Icon=/usr/share/qGo/pics/Bowl.png
-MimeType=text/plain;text/sfg;Application
-Name=qgo
+Comment=q5go
+Exec=q5go
+Icon=/usr/share/q5go/pics/Bowl.ico
+MimeType=application/x-go-sgf;
+Name=q5go
 GenericName=Go
 Name[fr]=qgo
 GenericName[ar]=??
