--- qt.orig/src/3rdparty/javascriptcore/JavaScriptCore/wtf/Platform.h	2015-05-07 16:14:48.000000000 +0200
+++ qt/src/3rdparty/javascriptcore/JavaScriptCore/wtf/Platform.h	2017-01-14 19:34:43.375000207 +0100
@@ -336,7 +336,7 @@
 /* CPU(MIPS) - MIPS, any version */
 #if (defined(mips) || defined(__mips__) || defined(MIPS) || defined(_MIPS_))
 #define WTF_CPU_MIPS 1
-#include <sgidefs.h>
+#include <asm/sgidefs.h>
 #if defined(__MIPSEB__)
 #define WTF_CPU_BIG_ENDIAN 1
 #endif
