Remove vendor and var 'lib' from setup.py
diff -u hiredis-0.2.0.orig/setup.py hiredis-0.2.0/setup.py
--- setup.py	2015-06-21 11:47:28.796472027 +0800
+++ setup.py	2015-06-21 11:49:53.047949803 +0800
@@ -40,12 +40,9 @@
 # supported Python versions is worse...
 #
 # Also see: https://github.com/pietern/hiredis-py/issues/15
-lib = ("hiredis_for_hiredis_py", {
-  "sources": ["vendor/hiredis/%s.c" % src for src in ("read", "sds")]})
 
 ext = Extension("hiredis.hiredis",
-  sources=glob.glob("src/*.c"),
-  include_dirs=["vendor"])
+  sources=glob.glob("src/*.c"))
 
 setup(
   name="hiredis",
@@ -57,7 +54,6 @@
   keywords=["Redis"],
   license="BSD",
   packages=["hiredis"],
-  libraries=[lib],
   ext_modules=[ext],
 
   # Override "install_lib" command
Common subdirectories: hiredis-0.2.0.orig/src and hiredis-0.2.0/src
Common subdirectories: hiredis-0.2.0.orig/vendor and hiredis-0.2.0/vendor
