diff --git a/shell/containmentconfigview.cpp b/shell/containmentconfigview.cpp
index 4d107bec..8d6df0d4 100644
--- shell/containmentconfigview.cpp
+++ shell/containmentconfigview.cpp
@@ -245,4 +245,4 @@ void ContainmentConfigView::syncWallpaperObjects()
     m_currentWallpaperConfig = static_cast<KDeclarative::ConfigPropertyMap *>(wallpaperGraphicsObject->property("configuration").value<QObject *>());
 }
 
-#include "private/moc_containmentconfigview.cpp"
+#include "moc_containmentconfigview.cpp"
diff --git a/shell/currentcontainmentactionsmodel.cpp b/shell/currentcontainmentactionsmodel.cpp
index 08317ed9..521d1c98 100644
--- shell/currentcontainmentactionsmodel.cpp
+++ shell/currentcontainmentactionsmodel.cpp
@@ -275,4 +275,4 @@ void CurrentContainmentActionsModel::save()
     }
 }
 
-#include "private/moc_currentcontainmentactionsmodel.cpp"
+#include "moc_currentcontainmentactionsmodel.cpp"
