--- a/login-sessions/plasmax11.desktop.cmake	2020-05-14 12:53:54.000000000 +0200
+++ b/login-sessions/plasmax11.desktop.cmake	2020-05-14 12:53:54.000000000 +0200
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Type=XSession
-Exec=${CMAKE_INSTALL_FULL_BINDIR}/startplasma-x11
+Exec=dbus-launch --exit-with-session ${CMAKE_INSTALL_FULL_BINDIR}/startplasma-x11
 TryExec=${CMAKE_INSTALL_FULL_BINDIR}/startplasma-x11
 DesktopNames=KDE
 Name=Plasma
