Fix build with glibc-2.10.x, patch from gentoo.

--- include/ntp_syscall.h.orig	2009-05-19 16:44:55.048156467 -0400
+++ include/ntp_syscall.h	2009-05-19 16:46:19.293323686 -0400
@@ -14,6 +14,14 @@
 # include <sys/timex.h>
 #endif
 
+#if defined(ADJ_NANO) && !defined(MOD_NANO)
+#define MOD_NANO ADJ_NANO
+#endif
+
+#if defined(ADJ_TAI) && !defined(MOD_TAI)
+#define MOD_TAI ADJ_TAI
+#endif
+
 #ifndef NTP_SYSCALLS_LIBC
 #ifdef NTP_SYSCALLS_STD
 # define ntp_adjtime(t)		syscall(SYS_ntp_adjtime, (t))
