commit a3db0a3574271fea9be5f9a8832bead210686130
Author: q66 <daniel@octaforge.org>
Date:   Tue Jan 7 02:03:11 2020 +0100

    Include byteswap.h to prevent undefined references on BE musl

diff --git endian.h endian.h
index 7c78807..3cae0e1 100644
--- a/endian.h
+++ b/endian.h
@@ -20,6 +20,7 @@
 #define ENDIAN_H
 
 #include <endian.h>
+#include <byteswap.h>
 
 #if __BYTE_ORDER == __LITTLE_ENDIAN
 #define cpu_to_le16(x) (x)
