commit 167c895b28df8f9be619dd5a7ed1e2e3f31dd4d8
Author: q66 <daniel@octaforge.org>
Date:   Sun Jan 5 22:23:53 2020 +0100

    Fix waflib on python 3.8

diff --git waflib/Node.py waflib/Node.py
index c86c4c1..a81779a 100644
--- waflib/Node.py
+++ waflib/Node.py
@@ -487,7 +487,6 @@ class Node(object):
 					if maxdepth:
 						for k in node.ant_iter(accept=accept, maxdepth=maxdepth - 1, pats=npats, dir=dir, src=src, remove=remove):
 							yield k
-		raise StopIteration
 
 	def ant_glob(self, *k, **kw):
 		"""
