diff -aur kupfer-v208/kupfer/core/settings.py 
kupfer-v208.modified/kupfer/core/settings.py
--- kupfer-v208/kupfer/core/settings.py 2012-06-01 10:51:14.000000000 -0700
+++ kupfer-v208.modified/kupfer/core/settings.py	2012-11-21 14:48:11.876592441 
-0800
@@ -393,6 +393,8 @@
		"""
		Get a list of (id_, name) tuples for the given @category_key
		"""
+		if not category_key in self._alternative_validators:
+			return
		validator = self._alternative_validators[category_key]
		for (id_, alternative) in self._alternatives[category_key].iteritems():
			name = alternative["name"]
@@ -400,7 +402,8 @@
				yield (id_, name)
 
	def get_all_alternatives(self, category_key):
-		return self._alternatives[category_key]
+		if category_key in self._alternatives:
+			return self._alternatives[category_key]
 
	def get_preferred_alternative(self, category_key):
