--- linphone-sdk-5.5.15.orig/cmake/BCProjects.cmake	2026-08-06 09:00:06.000000000 +0200
+++ linphone-sdk-5.5.15/cmake/BCProjects.cmake	2026-08-10 11:51:50.495939257 +0200
@@ -4,7 +4,7 @@
 #
 ############################################################################
 #
-# This file is part of Linphone SDK 
+# This file is part of Linphone SDK
 # (see https://gitlab.linphone.org/BC/public/linphone-sdk).
 #
 # This program is free software: you can redistribute it and/or modify
@@ -73,7 +73,7 @@
 
 if(ENABLE_MKV)
 	function(add_bcmatroska2)
-		set(BUILD_SHARED_LIBS OFF) # Always build statically, otherwise linking is failing on Windows
+		set(BUILD_SHARED_LIBS ON)
 		add_subdirectory("bcmatroska2")
 		add_dependencies(sdk bcmatroska2)
 	endfunction()
