--- linphone-sdk-5.5.15.orig/bctoolbox/src/crypto/mbedtls.cc	2026-08-06 09:00:06.000000000 +0200
+++ linphone-sdk-5.5.15/bctoolbox/src/crypto/mbedtls.cc	2026-08-10 10:35:48.724326469 +0200
@@ -38,6 +38,8 @@
 namespace bctoolbox {
 
 namespace {
+
+#if defined(MBEDTLS_THREADING_ALT)
 // This is also defined in mbedtls source code by a custom modification
 using mbedtls_threading_mutex_t = void *;
 
@@ -71,6 +73,7 @@
 	static_cast<std::mutex *>(*mutex)->unlock();
 	return 0;
 }
+#endif
 
 class mbedtlsStaticContexts {
 public:
@@ -80,8 +83,10 @@
 
 	std::unique_ptr<RNG> sRNG;
 	mbedtlsStaticContexts() {
+		#if defined(MBEDTLS_THREADING_ALT)
 		mbedtls_threading_set_alt(threading_mutex_init_cpp, threading_mutex_free_cpp, threading_mutex_lock_cpp,
 		                          threading_mutex_unlock_cpp);
+		#endif
 		if (psa_crypto_init() != PSA_SUCCESS) {
 			bctbx_error("MbedTLS PSA init fail");
 		}
@@ -92,7 +97,9 @@
 		// before destroying mbedtls internal context, destroy the static RNG
 		sRNG = nullptr;
 		mbedtls_psa_crypto_free();
+		#if defined(MBEDTLS_THREADING_ALT)
 		mbedtls_threading_free_alt();
+		#endif
 	}
 };
 static const auto mbedtlsStaticContextsInstance = std::make_unique<mbedtlsStaticContexts>();
