--- linphone-sdk-5.5.15.orig/liblinphone/cmake/FindJsonCPP.cmake	2026-08-06 09:00:06.000000000 +0200
+++ linphone-sdk-5.5.15/liblinphone/cmake/FindJsonCPP.cmake	2026-08-10 09:51:27.359104073 +0200
@@ -57,28 +57,9 @@
 	mark_as_advanced(JsonCPP_TARGET)
 
 else()
-	
-	set(_OPTIONS "CONFIG")
-	if(JsonCPP_FIND_REQUIRED)
-		list(APPEND _OPTIONS "REQUIRED")
-	endif()  
-	if(JsonCPP_FIND_QUIETLY)
-		list(APPEND _OPTIONS "QUIET")
-	endif()
-	if(JsonCPP_FIND_VERSION)
-		list(PREPEND _OPTIONS "${jsoncpp_FIND_VERSION}")
-	endif()
-	if(JsonCPP_FIND_EXACT)
-		list(APPEND _OPTIONS "EXACT")
-	endif()
+	find_package(PkgConfig QUIET REQUIRED)
+	pkg_check_modules(JsonCPP REQUIRED IMPORTED_TARGET jsoncpp)
+	set(JsonCPP_TARGET PkgConfig::JsonCPP)
 
-	find_package(jsoncpp ${_OPTIONS})
-
-	if(TARGET jsoncpp_lib)
-		set(JsonCPP_TARGET jsoncpp_lib)
-	else()
-		set(JsonCPP_TARGET jsoncpp_static)
-	endif()
-
-	set(JsonCPP_FOUND ${jsoncpp_FOUND})
+	set(JsonCPP_FOUND True)
 endif()
