--- linphone-sdk-5.5.15.orig/liblinphone/src/factory/factory.cpp	2026-08-06 09:00:06.000000000 +0200
+++ linphone-sdk-5.5.15/liblinphone/src/factory/factory.cpp	2026-08-10 11:04:20.458895757 +0200
@@ -35,7 +35,6 @@
 #include <ZXing/BitMatrixIO.h>
 #include <ZXing/CharacterSet.h>
 #include <ZXing/MultiFormatWriter.h>
-#include <ZXing/TextUtfEncoding.h>
 #endif
 #ifdef JPEG_ENABLED
 #include <mediastreamer2/msvideo.h>
@@ -935,7 +934,7 @@
 	}
 
 	auto writer = ZXing::MultiFormatWriter(format).setMargin((int)margin).setEncoding(encoding).setEccLevel(eccLevel);
-	auto matrix = writer.encode(ZXing::TextUtfEncoding::FromUtf8(code), (int)width, (int)height);
+	auto matrix = writer.encode(code, (int)width, (int)height);
 	auto bitmap = ZXing::ToMatrix<uint8_t>(matrix);
 
 	LinphoneContent *content = Factory::createContent();
