--- linphone-sdk-5.5.15.orig/liblinphone/wrappers/cpp/CMakeLists.txt	2026-08-06 09:00:06.000000000 +0200
+++ linphone-sdk-5.5.15/liblinphone/wrappers/cpp/CMakeLists.txt	2026-08-10 13:54:16.945702001 +0200
@@ -4,7 +4,7 @@
 #
 ############################################################################
 #
-# This file is part of Liblinphone 
+# This file is part of Liblinphone
 # (see https://gitlab.linphone.org/BC/public/liblinphone).
 #
 # This program is free software: you can redistribute it and/or modify
@@ -127,12 +127,12 @@
 
 install(EXPORT LinphoneCxxTargets
 	FILE LinphoneCxxTargets.cmake
-	DESTINATION "${CMAKE_INSTALL_DATADIR}/LinphoneCxx/cmake"
+	DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/LinphoneCxx"
 )
 install(FILES
 	"${CMAKE_CURRENT_BINARY_DIR}/LinphoneCxxConfig.cmake"
 	"${CMAKE_CURRENT_BINARY_DIR}/LinphoneCxxVersion.cmake"
-	DESTINATION "${CMAKE_INSTALL_DATADIR}/LinphoneCxx/cmake"
+	DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/LinphoneCxx"
 )
 
 if(ENABLE_DOC OR ENABLE_CXX_WRAPPER)
