--- linphone-sdk-5.5.15.orig/mediastreamer2/src/videofilters/zxing_qrcode.cpp	2026-08-06 09:00:06.000000000 +0200
+++ linphone-sdk-5.5.15/mediastreamer2/src/videofilters/zxing_qrcode.cpp	2026-08-10 10:56:19.348603679 +0200
@@ -124,10 +124,10 @@
 static void read_qrcode(MSFilter *f) {
 	QRCodeReaderStruct *qrc = (QRCodeReaderStruct *)f->data;
 	if (qrc->image.data(0, 0)) {
-		DecodeHints hints;
-		hints.setFormats(BarcodeFormat::QRCode); // Search optimization : Only QRCode symbols are used.
-		hints.setReturnErrors(true);
-		Results results = ReadBarcodes(qrc->image, hints);
+		ReaderOptions readerOptions;
+		readerOptions.setFormats(BarcodeFormat::QRCode); // Search optimization : Only QRCode symbols are used.
+		readerOptions.setReturnErrors(true);
+		Barcodes results = ReadBarcodes(qrc->image, readerOptions);
 		for (size_t i = 0; i < results.size(); ++i) {
 			if (results[i].error())
 				ms_warning("[MSQRCodeReader] Cannot decode QRCode : %s", ToString(results[i].error()).c_str());
