--- old/hotspot/src/os_cpu/linux_arm/vm/os_linux_arm.cpp
+++ new/hotspot/src/os_cpu/linux_arm/vm/os_linux_arm.cpp
@@ -71,7 +71,6 @@
 # include <pwd.h>
 # include <poll.h>
 # include <ucontext.h>
-# include <fpu_control.h>
 # include <asm/ptrace.h>
 
 #define SPELL_REG_SP  "sp"
@@ -104,6 +103,18 @@
 #define ARM_REGS_IN_CONTEXT  31
 
 #else
+
+// Stupid hack as the origin if below doesnt compile with gcc 8.2.0:
+//
+// os_linux_arm.cpp:114:5: error: missing binary operator before token "("
+//  #if NGREG == 16
+//       ^~~~~
+//
+// The NGREG is 18, so force it to that value.
+#ifdef NGREG
+#  undef NGREG
+#endif
+#define NGREG 18
 
 #if NGREG == 16
 // These definitions are based on the observation that until
