--- a/include/headers.h	2024-11-06 18:47:34.000000000 -0300
+++ b/include/headers.h	2024-04-07 00:47:17.000000000 -0300
@@ -134,8 +134,8 @@
 // Bummer, AF_PACKET requires kernel headers as <netpacket/packet.h> isn't sufficient
 #include <linux/filter.h>
 #include <linux/if_packet.h>
-#include <netinet/ip.h>
-#include <netinet/udp.h>
+#include <linux/ip.h>
+#include <linux/udp.h>
 // In older linux kernels, the kernel headers and glibc headers are in conflict,
 // specificially <linux/in6.h> and <netinit/in.h>
 // preventing the following include:
