From 478eb189c47b555204c132317c9cc28effbdc45a Mon Sep 17 00:00:00 2001
From: g1a55er <x@g1a55er.net>
Date: Sat, 1 Jun 2024 19:32:07 -0400
Subject: [PATCH] Remove domain that does not exist

---
 ts/scripts/generate-dns-fallback.ts | 1 -
 1 file changed, 1 deletion(-)

diff --git a/ts/scripts/generate-dns-fallback.ts b/ts/scripts/generate-dns-fallback.ts
index dec7023df3..39f442f4b9 100644
--- a/ts/scripts/generate-dns-fallback.ts
+++ b/ts/scripts/generate-dns-fallback.ts
@@ -20,7 +20,6 @@ const FALLBACK_DOMAINS = [
   'cdn3.signal.org',
   'updates2.signal.org',
   'sfu.voip.signal.org',
-  'create.signal.art',
 ];
 
 async function main() {
