diff -Nur a/util/grub.d/10_linux.in b/util/grub.d/10_linux.in
--- a/util/grub.d/10_linux.in	2015-09-27 14:14:45.544730230 -0300
+++ b/util/grub.d/10_linux.in	2015-09-27 14:58:23.533917184 -0300
@@ -31,7 +31,17 @@
 if [ "x${GRUB_DISTRIBUTOR}" = "x" ] ; then
   OS=GNU/Linux
 else
-  OS="${GRUB_DISTRIBUTOR} GNU/Linux"
+  if echo ${GRUB_DISTRIBUTOR} | grep -qi Parabola ; then
+      OS="${GRUB_DISTRIBUTOR} GNU/Linux-libre"
+  elif echo ${GRUB_DISTRIBUTOR} | grep -qi Blag ; then
+      OS="${GRUB_DISTRIBUTOR} Linux and GNU"
+  elif echo ${GRUB_DISTRIBUTOR} | grep -qi Musix ; then
+      OS="${GRUB_DISTRIBUTOR} GNU+Linux"
+  elif echo ${GRUB_DISTRIBUTOR} | grep -qi Dragora ; then
+      OS="${GRUB_DISTRIBUTOR} GNU/Linux-libre"
+  else
+      OS="${GRUB_DISTRIBUTOR} GNU/Linux"
+  fi
   CLASS="--class $(echo ${GRUB_DISTRIBUTOR} | tr 'A-Z' 'a-z' | cut -d' ' -f1|LC_ALL=C sed 's,[^[:alnum:]_],_,g') ${CLASS}"
 fi
 
diff -Nur a/util/grub.d/20_linux_xen.in b/util/grub.d/20_linux_xen.in
--- a/util/grub.d/20_linux_xen.in	2015-09-27 14:14:45.544730230 -0300
+++ b/util/grub.d/20_linux_xen.in	2015-09-27 14:59:16.319319658 -0300
@@ -31,7 +31,17 @@
 if [ "x${GRUB_DISTRIBUTOR}" = "x" ] ; then
   OS=GNU/Linux
 else
-  OS="${GRUB_DISTRIBUTOR} GNU/Linux"
+  if echo ${GRUB_DISTRIBUTOR} | grep -qi Parabola ; then
+      OS="${GRUB_DISTRIBUTOR} GNU/Linux-libre"
+  elif echo ${GRUB_DISTRIBUTOR} | grep -qi Blag ; then
+      OS="${GRUB_DISTRIBUTOR} Linux and GNU"
+  elif echo ${GRUB_DISTRIBUTOR} | grep -qi Musix ; then
+      OS="${GRUB_DISTRIBUTOR} GNU+Linux"
+  elif echo ${GRUB_DISTRIBUTOR} | grep -qi Dragora ; then
+      OS="${GRUB_DISTRIBUTOR} GNU/Linux-libre"
+  else
+      OS="${GRUB_DISTRIBUTOR} GNU/Linux"
+  fi
   CLASS="--class $(echo ${GRUB_DISTRIBUTOR} | tr 'A-Z' 'a-z' | cut -d' ' -f1|LC_ALL=C sed 's,[^[:alnum:]_],_,g') ${CLASS}"
 fi
 
