--- grub-core/gnulib/stdio.in.h.orig	2012-07-05 11:43:24.596994420 +0200
+++ grub-core/gnulib/stdio.in.h	2012-07-05 11:43:28.683970680 +0200
@@ -141,7 +141,6 @@ _GL_WARN_ON_USE (fflush, "fflush is not
    so any use of gets warrants an unconditional warning.  Assume it is
    always declared, since it is required by C89.  */
 #undef gets
-_GL_WARN_ON_USE (gets, "gets is a security hole - use fgets instead");
 
 #if @GNULIB_FOPEN@
 # if @REPLACE_FOPEN@
