diff --git a/libgweather/meson.build b/libgweather/meson.build
index c32261e..f7dff20 100644
--- libgweather/meson.build
+++ libgweather/meson.build
@@ -76,7 +76,10 @@ if get_option('enable_gir')
     identifier_prefix: 'GWeather',
     export_packages: 'gweather-3.0',
     header: 'libgweather/gweather.h',
-    extra_args: ['--warn-all'],
+    extra_args: [
+      '--warn-all',
+      '-DGWEATHER_COMPILATION',
+    ],
     install: true,
   )
   

