--- configure.in	2012-01-02 01:47:37.000000000 +0100
+++ configure.in	2012-02-09 19:29:29.405603749 +0100
@@ -433,12 +433,13 @@
 		#
 		# Try libnl 2.x first.
 		#
-		AC_CHECK_LIB(nl, nl_socket_alloc,
+		AC_CHECK_LIB(nl-3, nl_socket_alloc,
 		[
 			#
 			# Yes, we have libnl 2.x.
 			#
-			LIBS="-lnl-genl -lnl $LIBS"
+			LIBS="$(pkg-config --libs libnl-genl-3.0 libnl-3.0) $LIBS"
+			V_INCLS="$V_INCLS $(pkg-config --cflags libnl-genl-3.0 libnl-3.0)"
 			AC_DEFINE(HAVE_LIBNL,1,[if libnl exists])
 			AC_DEFINE(HAVE_LIBNL_2_x,1,[if libnl exists and is version 2.x])
 		],
