From 64ac3a6f94cd299e5e14945dc31b48f009dec152 Mon Sep 17 00:00:00 2001
From: Rico Tzschichholz <ricotz@ubuntu.com>
Date: Wed, 30 Dec 2020 16:50:32 +0100
Subject: [PATCH] Use GenericArray API only

---
 libzeitgeist/where-clause.vala | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libzeitgeist/where-clause.vala b/libzeitgeist/where-clause.vala
index 204e9b1a..c3e6fa62 100644
--- a/libzeitgeist/where-clause.vala
+++ b/libzeitgeist/where-clause.vala
@@ -216,7 +216,7 @@ namespace Zeitgeist
 #else
             long[] pointers = new long[gptrarr.length + 1];
 #endif
-            Memory.copy(pointers, ((PtrArray *) gptrarr)->pdata,
+            Memory.copy(pointers, (void*) gptrarr.data,
                 gptrarr.length * sizeof (void *));
             return (T[]) pointers;
         }
-- 
2.31.1

