https://gitlab.freedesktop.org/zeitgeist/zeitgeist/-/issues/25

--- a/test/c/Makefile.am	2016-09-17 08:57:19.000000000 +0100
+++ b/test/c/Makefile.am	2023-12-04 14:20:58.941739810 +0000
@@ -26,7 +26,6 @@
 	test-timestamp \
 	test-symbols \
 	test-monitor \
-	test-log \
 	test-event \
 	test-datasource \
 	test-mimetypes \
@@ -38,7 +37,6 @@
 test_timestamp_SOURCES = test-timestamp.c
 test_symbols_SOURCES = test-symbols.c
 test_monitor_SOURCES = test-monitor.c
-test_log_SOURCES = test-log.c
 test_event_SOURCES = test-event.c
 test_datasource_SOURCES = test-datasource.c
 test_mimetypes_SOURCES = test-mimetypes.c
--- a/test/direct/Makefile.am	2016-09-17 08:57:19.000000000 +0100
+++ b/test/direct/Makefile.am	2023-12-04 14:15:59.567828861 +0000
@@ -47,7 +47,6 @@
 	datamodel-test \
 	datasource-test \
 	event-test \
-	log-test \
 	marshalling-test \
 	mimetype-test \
 	monitor-test \
@@ -62,7 +61,6 @@
 datamodel_test_SOURCES = datamodel-test.vala
 datasource_test_SOURCES = datasource-test.vala
 event_test_SOURCES = event-test.vala
-log_test_SOURCES = log-test.vala
 marshalling_test_SOURCES = marshalling-test.vala
 query_operators_test_SOURCES = query-operators-test.vala
 symbol_test_SOURCES = symbol-test.vala
