--- configure.ac.orig	2018-01-16 18:25:01.896527503 +0100
+++ configure.ac	2018-01-16 18:25:27.827424062 +0100
@@ -1310,36 +1310,6 @@ AC_SUBST([sockdir])
 
 # Where to place smtpd.pid
 piddir=/var/run
-AC_RUN_IFELSE(
-	[
-		AC_LANG_PROGRAM([[
-#include <stdio.h>
-#include <stdlib.h>
-#ifdef HAVE_PATHS_H
-#include <paths.h>
-#endif
-#define DATA "conftest.piddir"
-		]],
-		[[
-#ifdef _PATH_VARRUN
-FILE *fd;
-int rc;
-
-if ((fd = fopen(DATA,"w")) == NULL) { exit(1); }
-if ((rc = fprintf(fd ,"%s\n", _PATH_VARRUN)) < 0) { exit(2); }
-exit(0);
-#else
-exit(-1);
-#endif
-		]])
-	],		[
-		piddir=`cat conftest.piddir`
-		AC_MSG_RESULT([using $piddir from paths.h])
-	],
-	[
-		AC_MSG_RESULT([using $piddir from default value])
-	]
-)
 
 AC_ARG_WITH([path-pidfile],
 	[  --with-path-pidfile=PATH	Specify path to smtpd.pid directory (default=/var/run)],
