--- smtpd/table.c	2015-10-02 01:46:21.000000000 +0200
+++ smtpd/table.c	2015-10-02 08:48:34.971352820 +0200
@@ -44,6 +44,10 @@
 #include "smtpd.h"
 #include "log.h"
 
+#if !defined(SCOPE_DELIMITER)
+#define	SCOPE_DELIMITER	'%'
+#endif
+
 struct table_backend *table_backend_lookup(const char *);
 
 extern struct table_backend table_backend_static;
--- smtpd/parser.c	2015-10-02 01:46:21.000000000 +0200
+++ smtpd/parser.c	2015-10-02 10:07:39.515982655 +0200
@@ -36,6 +36,10 @@
 
 #include "parser.h"
 
+#if !defined(SCOPE_DELIMITER)
+#define	SCOPE_DELIMITER	'%'
+#endif
+
 uint64_t text_to_evpid(const char *);
 uint32_t text_to_msgid(const char *);
 
