--- mk/smtpctl/Makefile.am
+++ mk/smtpctl/Makefile.am
@@ -4,6 +4,7 @@ sbin_PROGRAMS=    smtpctl
 
 smtpctl_SOURCES= $(smtpd_srcdir)/enqueue.c
 smtpctl_SOURCES+=  $(smtpd_srcdir)/parser.c
+smtpctl_SOURCES+=  $(smtpd_srcdir)/config.c
 smtpctl_SOURCES+=  $(smtpd_srcdir)/log.c
 smtpctl_SOURCES+=  $(smtpd_srcdir)/envelope.c
 smtpctl_SOURCES+=  $(smtpd_srcdir)/queue_backend.c
@@ -35,7 +36,7 @@
 smtpctl_SOURCES+=  $(smtpd_srcdir)/crypto.c
 endif
 
-smtpctl_CFLAGS=    -DNO_IO
+smtpctl_CFLAGS=    -DNO_IO -DCONFIG_MINIMUM
 smtpctl_CFLAGS+= -DPATH_GZCAT=\"$(ZCAT)\" \
      -DPATH_ENCRYPT=\"$(pkglibexecdir)/encrypt\"
 
@@ -79,8 +80,6 @@

 install-exec-hook: $(CONFIGFILES) $(MANPAGES)                                 
        $(MKDIR_P) $(DESTDIR)$(mandir)/$(mansubdir)8                           
-       chgrp $(SMTPD_QUEUE_USER) $(DESTDIR)$(sbindir)/smtpctl                 
-       chmod 2555 $(DESTDIR)$(sbindir)/smtpctl                                
        $(INSTALL) -m 644 smtpctl.8.out         $(DESTDIR)$(mandir)/$(mansubdir)8/smtpctl.8
        $(INSTALL) -m 644 sendmail.8.out        $(DESTDIR)$(mandir)/$(mansubdir)8/sendmail.8
        $(INSTALL) -m 644 makemap.8.out         $(DESTDIR)$(mandir)/$(mansubdir)8/makemap.8

