Else it smashes the stack when IPv6 is used.

--- smtpd/mta_session.c
+++ smtpd/mta_session.c
@@ -1811,15 +1811,15 @@
 static void
 mta_connected(struct mta_session *s)
 {
-	struct sockaddr sa_src;
-	struct sockaddr sa_dest;
+	struct sockaddr_storage sa_src;
+	struct sockaddr_storage sa_dest;
 	int sa_len;
 
 	log_info("%016"PRIx64" mta connected", s->id);
 
-	if (getsockname(io_fileno(s->io), &sa_src, &sa_len) == -1)
+	if (getsockname(io_fileno(s->io), (struct sockaddr *)(void *)&sa_src, &sa_len) == -1)
 		bzero(&sa_src, sizeof sa_src);
-	if (getpeername(io_fileno(s->io), &sa_dest, &sa_len) == -1)
+	if (getpeername(io_fileno(s->io), (struct sockaddr *)(void *)&sa_dest, &sa_len) == -1)
 		bzero(&sa_dest, sizeof sa_dest);
 
 	mta_report_link_connect(s,
