From 8ad26a05598810b9d64722ac48507c145b010e30 Mon Sep 17 00:00:00 2001
From: Leon Marz <main@lmarz.org>
Date: Wed, 7 Dec 2022 21:18:58 +0100
Subject: [PATCH 1/2] musl fixes

---
 extern/glog/src/config_linux.h | 1 -
 1 file changed, 1 deletion(-)

diff --git a/extern/glog/src/config_linux.h b/extern/glog/src/config_linux.h
index b3a3325..946095a 100644
--- a/extern/glog/src/config_linux.h
+++ b/extern/glog/src/config_linux.h
@@ -14,7 +14,9 @@
 #define HAVE_DLFCN_H

 /* Define to 1 if you have the <execinfo.h> header file. */
+#ifdef __GLIBC__
 #define HAVE_EXECINFO_H
+#endif

 /* Define if you have the `fcntl' function */
 #define HAVE_FCNTL