Index: blender-3.3.0/intern/cycles/kernel/CMakeLists.txt
===================================================================
--- blender-3.3.0.orig/intern/cycles/kernel/CMakeLists.txt
+++ blender-3.3.0/intern/cycles/kernel/CMakeLists.txt
@@ -947,9 +947,7 @@ endif()
 # Warnings to avoid using doubles in the kernel.
 if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_C_COMPILER_ID MATCHES "Clang")
   ADD_CHECK_CXX_COMPILER_FLAG(CMAKE_CXX_FLAGS _has_cxxflag_float_conversion "-Werror=float-conversion")
-  ADD_CHECK_CXX_COMPILER_FLAG(CMAKE_CXX_FLAGS _has_cxxflag_double_promotion "-Werror=double-promotion")
   unset(_has_cxxflag_float_conversion)
-  unset(_has_cxxflag_double_promotion)
 endif()
 
 cycles_add_library(cycles_kernel "${LIB}"
