From 90ac05e3d81c604df532122b7da53a4a15c79a37 Mon Sep 17 00:00:00 2001
From: q66 <daniel@octaforge.org>
Date: Mon, 23 Dec 2019 22:48:50 +0100
Subject: [PATCH] add ppc support in build_config.h

---
 intern/numaapi/source/build_config.h | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/intern/numaapi/source/build_config.h b/intern/numaapi/source/build_config.h
index 444adcc..f7c072a 100644
--- a/intern/numaapi/source/build_config.h
+++ b/intern/numaapi/source/build_config.h
@@ -307,6 +307,20 @@
 #    define ARCH_CPU_32_BITS 1
 #    define ARCH_CPU_BIG_ENDIAN 1
 #  endif
+#elif defined(__powerpc64__)
+#  define ARCH_CPU_PPC64_FAMILY 1
+#  define ARCH_CPU_PPC64 1
+#  define ARCH_CPU_64_BITS 1
+#  if defined(__LITTLE_ENDIAN__)
+#    define ARCH_CPU_LITTLE_ENDIAN 1
+#  else
+#    define ARCH_CPU_BIG_ENDIAN 1
+#  endif
+#elif defined(__powerpc__)
+#  define ARCH_CPU_PPC_FAMILY 1
+#  define ARCH_CPU_PPC 1
+#  define ARCH_CPU_32_BITS 1
+#  define ARCH_CPU_BIG_ENDIAN 1
 #else
 #  error Please add support for your architecture in build_config.h
 #endif
@@ -337,6 +351,12 @@
 #if !defined(ARCH_CPU_MIPS64_FAMILY)
 #  define ARCH_CPU_MIPS64_FAMILY 0
 #endif
+#if !defined(ARCH_CPU_PPC64_FAMILY)
+#  define ARCH_CPU_PPC64_FAMILY 0
+#endif
+#if !defined(ARCH_CPU_PPC_FAMILY)
+#  define ARCH_CPU_PPC_FAMILY 0
+#endif
 
 ////////////////////////////////////////////////////////////////////////////////
 // Sizes of platform-dependent types.
-- 
2.24.0

