--- a/scripts/addons/io_import_dxf/dxfgrabber/dxfentities.py
+++ b/scripts/addons/io_import_dxf/dxfgrabber/dxfentities.py
@@ -852,7 +852,7 @@ class MText(DXFEntity):
         self.set_default_extrusion()
 
     def lines(self):
-        return self.raw_text.split('\P')
+        return self.raw_text.split(r'\P')
 
     def plain_text(self, split=False):
         chars = []
--- a/scripts/addons/io_scene_gltf2/io/exp/gltf2_io_image_data.py
+++ b/scripts/addons/io_scene_gltf2/io/exp/gltf2_io_image_data.py
@@ -22,9 +22,9 @@ class ImageData:
         return hash(self._data)
 
     def adjusted_name(self):
-        regex_dot = re.compile("\.")
+        regex_dot = re.compile(r"\.")
         adjusted_name = re.sub(regex_dot, "_", self.name)
-        new_name = "".join([char for char in adjusted_name if char not in "!#$&'()*+,/:;<>?@[\]^`{|}~"])
+        new_name = "".join([char for char in adjusted_name if char not in r"!#$&'()*+,/:;<>?@[\]^`{|}~"])
         return new_name
 
     @property
--- a/scripts/modules/bl_i18n_utils/settings.py
+++ b/scripts/modules/bl_i18n_utils/settings.py
@@ -302,7 +302,7 @@ PYGETTEXT_KEYWORDS = (() +
     # becomes extremely slow to process some (unrelated) source files.
     ((r"\{(?:(?:\s*\"[^\",]+\"\s*,)|(?:\s*\"\\\"\",)|(?:\s*nullptr\s*,)){4}\s*" +
       _msg_re + r"\s*,(?:(?:\s*\"[^\"',]+\"\s*,)|(?:\s*nullptr\s*,))(?:[^,]+,){2}"
-      + "(?:\|?\s*B_UNIT_DEF_[_A-Z]+\s*)+\}"),) +
+      + r"(?:\|?\s*B_UNIT_DEF_[_A-Z]+\s*)+\}"),) +
 
     tuple((r"{}\(\s*" + _msg_re + r"\s*,\s*(?:" +
            r"\s*,\s*)?(?:".join(_ctxt_re_gen(i) for i in range(PYGETTEXT_MAX_MULTI_CTXT)) + r")?\s*\)").format(it)
