--- a/intern/itasc/kdl/tree.hpp	2018-03-23 16:22:25.000000000 +0100
+++ b/intern/itasc/kdl/tree.hpp	2018-08-13 19:31:32.101185313 +0200
@@ -34,7 +34,7 @@
     //Forward declaration
     class TreeElement;
     // Eigen allocator is needed for alignment of Eigen data types
-    typedef std::map<std::string,TreeElement, std::less<std::string>, Eigen::aligned_allocator<std::pair<std::string, TreeElement> > > SegmentMap;
+    typedef std::map<std::string,TreeElement, std::less<std::string>, Eigen::aligned_allocator<std::pair<const std::string, TreeElement> > > SegmentMap;
 
     class TreeElement
     {
