# reason: print to $HOME by default

--- a/extra/cups-pdf.conf
+++ b/extra/cups-pdf.conf
@@ -42,17 +42,17 @@
 ### Key: Out (config)
 ##  CUPS-PDF output directory 
 ##  special qualifiers: 
 ##     ${HOME} will be expanded to the user's home directory
 ##     ${USER} will be expanded to the user name
 ##  in case it is an NFS export make sure it is exported without
 ##  root_squash! 
 ### Default: /var/spool/cups-pdf/${USER}
 
-#Out /var/spool/cups-pdf/${USER}
+Out ${HOME}/PDF
 
 ### Key: AnonDirName (config)
 ##  ABSOLUTE path for anonymously created PDF files
 ##  if anonymous access is disabled this setting has no effect
 ### Default: /var/spool/cups-pdf/ANONYMOUS
 
 #AnonDirName /var/spool/cups-pdf/ANONYMOUS
