diff --git a/Makefile b/Makefile
index 31d7c9a..e205e3f 100644
--- a/Makefile
+++ b/Makefile
@@ -18,7 +18,7 @@ INC_CFLAGS :=
 THR_CFLAGS :=
 CFLAGS     := $(OPT_CFLAGS) $(CPU_CFLAGS) $(DEB_CFLAGS) $(DEF_CFLAGS) $(USR_CFLAGS) $(INC_CFLAGS) $(THR_CFLAGS)
 
-LD         := $(CC)
+LD         := $(CROSS_COMPILE)cc
 DEB_LFLAGS := -g
 USR_LFLAGS :=
 LIB_LFLAGS :=
@@ -67,7 +67,6 @@ bin/%: src/%.o
 install: install-bins
 
 install-bins: bins
-	$(cmd_STRIP) $(BINS)
 	$(Q)[ -d "$(DESTDIR)$(PREFIX)/bin/." ] || mkdir -p -m 0755 $(DESTDIR)$(PREFIX)/bin
 	$(cmd_INSTALL) -m 0755 $(BINS) $(DESTDIR)$(PREFIX)/bin
 
