--- src/plugins/thumbnailffmpeg_extractor.c.orig	2014-05-08 21:07:41.546845538 +0200
+++ src/plugins/thumbnailffmpeg_extractor.c	2014-05-08 21:08:30.389842241 +0200
@@ -335,7 +335,7 @@
  * @param ec extraction context to use
  */
 static void
-extract_image (enum CodecID image_codec_id,
+extract_image (enum AVCodecID image_codec_id,
                struct EXTRACTOR_ExtractContext *ec)
 {
   AVDictionary *opts;
@@ -631,7 +631,7 @@
   /**
    * Corresponding ffmpeg decoder ID.
    */
-  enum CodecID codec_id;
+  enum AVCodecID codec_id;
 };
 
 
