--- src/plugins/thumbnailffmpeg_extractor.c.orig        2018-05-09 23:14:02.721105141 +0200
+++ src/plugins/thumbnailffmpeg_extractor.c     2018-05-09 23:14:48.491629162 +0200
@@ -107,7 +107,7 @@
 /**
  * Number of bytes to feed to libav in one go, with padding (padding is zeroed).
  */
-#define PADDED_BUFFER_SIZE (BUFFER_SIZE + FF_INPUT_BUFFER_PADDING_SIZE)
+#define PADDED_BUFFER_SIZE (BUFFER_SIZE + AV_INPUT_BUFFER_PADDING_SIZE)

 /**
  * Global handle to MAGIC data.
@@ -355,7 +355,7 @@
    encoder_codec_ctx->mb_lmin = encoder_codec_ctx->qmin * FF_QP2LAMBDA;
    encoder_codec_ctx->mb_lmax = encoder_codec_ctx->qmax * FF_QP2LAMBDA;
 #endif
-   encoder_codec_ctx->flags          = CODEC_FLAG_QSCALE;
+   encoder_codec_ctx->flags          = AV_CODEC_FLAG_QSCALE;
    encoder_codec_ctx->global_quality = encoder_codec_ctx->qmin * FF_QP2LAMBDA;

    dst_frame->pts     = 1;

