--- a/src/plugins/test_exiv2.c	2021-01-30 13:20:23.000000000 +0000
+++ b/src/plugins/test_exiv2.c	2023-11-03 19:39:43.115274492 +0000
@@ -49,8 +49,8 @@
       EXTRACTOR_METATYPE_GPS_LATITUDE,
       EXTRACTOR_METAFORMAT_UTF8,
       "text/plain",
-      "28deg 8' (18|17.585)\" ?",
-      strlen ("28deg 8' (18|17.585)\" ?") + 1,
+      "28 deg 8' (18|17.58)\" ?",
+      strlen ("28 deg 8' (18|17.58)\" ?") + 1,
       0,
       .regex = 1
     },
@@ -66,8 +66,8 @@
       EXTRACTOR_METATYPE_GPS_LONGITUDE,
       EXTRACTOR_METAFORMAT_UTF8,
       "text/plain",
-      "14deg 14' (22|21.713)\" ?",
-      strlen ("14deg 14' (22|21.713)\" ?"),
+      "14 deg 14' (22|21.71)\" ?",
+      strlen ("14 deg 14' (22|21.71)\" ?"),
       0,
       .regex = 1
     },
