--- Makefile.am.orig	2015-12-22 07:53:14.124104461 +0100
+++ Makefile.am	2015-12-22 07:53:24.434208468 +0100
@@ -93,27 +93,5 @@ release: distcheck
 	rm -fr $(PACKAGE_NAME)-$(PACKAGE_VERSION)
 	ls -l $(PACKAGE_NAME)-$(PACKAGE_VERSION).tar.gz
 
-install-data-hook:
-	$(DESTDIR)$(bindir)/vnstat$(EXEEXT) --showconfig >vnstat.conf && \
-	  $(MKDIR_P) $(DESTDIR)$(sysconfdir) && \
-	  mv -f vnstat.conf $(DESTDIR)$(sysconfdir)/vnstat.conf
-	@echo
-	@echo "============================================================================"
-	@echo "  Note: No service file has been installed for starting the daemon"
-	@echo "        automatically. Such setup will need to be created manually"
-	@echo "        depending on the used environment. See the included documentation."
-	@echo "============================================================================"
-	@echo
-
-uninstall-hook:
-	-rm -f $(DESTDIR)$(sysconfdir)/vnstat.conf
-	-rmdir $(DESTDIR)$(sysconfdir)
-	@echo
-	@echo "=================================================================="
-	@echo "  Note: The database directory has been left untouched."
-	@echo "        It will need to be removed manually if no longer needed."
-	@echo "=================================================================="
-	@echo
-
 clean-local:
 	-rm -fr coverage *.lcov *.gz test*.log test*.xml src/*.gc* tests/*.gc*
