From f054c63bd56704b5dd6b144622be9de95ba2ed72 Mon Sep 17 00:00:00 2001
From: "A. Maitland Bottoms" <bottoms@debian.org>
Date: Sun, 1 Dec 2019 17:23:43 -0500
Subject: [PATCH 2/2] find gr38 blocks

---
 CMakeLists.txt     | 8 ++++----
 lib/CMakeLists.txt | 4 ++--
 2 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 76375dc..0cb94d5 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -112,7 +112,7 @@ ENDIF()
 ########################################################################
 # Install directories
 ########################################################################
-find_package(Gnuradio REQUIRED) # Side effect of MODULE_PATH getting prepended with system components
+find_package(Gnuradio REQUIRED COMPONENTS blocks) # Side effect of MODULE_PATH getting prepended with system components
 list(REVERSE CMAKE_MODULE_PATH)
 set(MIN_GR_VERSION "3.8")
 if("${Gnuradio_VERSION}" VERSION_LESS MIN_GR_VERSION)
@@ -167,13 +167,13 @@ endif(APPLE)
 ########################################################################
 set(GR_REQUIRED_COMPONENTS RUNTIME PMT BLOCKS)
 
-find_package(gnuradio-blocks PATHS ${GR_PREFIX}/lib/cmake/gnuradio/)
+find_package(gnuradio-blocks PATHS ${Gnuradio_DIR})
 message(STATUS "Found Block Block: ${gnuradio-blocks_FOUND}")
 
-#[[find_package(gnuradio-pmt PATHS ${GR_PREFIX}/lib/cmake/gnuradio/)
+#[[find_package(gnuradio-pmt PATHS ${Gnuradio_DIR})
 message(STATUS "Found PMT Block: ${gnuradio-pmt_FOUND}")
 
-find_package(gnuradio-runtime PATHS ${GR_PREFIX}/lib/cmake/gnuradio/)
+find_package(gnuradio-runtime PATHS ${Gnuradio_DIR})
 message(STATUS "Found Runtime Block: ${gnuradio-runtime_FOUND}")
 
 # Software Components part of GNU Radio - These should all be present for a default install
diff --git a/lib/CMakeLists.txt b/lib/CMakeLists.txt
index 2171df2..5e3098d 100644
--- lib/CMakeLists.txt
+++ lib/CMakeLists.txt
@@ -137,7 +137,7 @@ endif(ENABLE_OSMOSDR)
 ########################################################################
 # Setup File component
 ########################################################################
-GR_REGISTER_COMPONENT("IQ File Source & Sink" ENABLE_FILE GNURADIO_BLOCKS_FOUND)
+GR_REGISTER_COMPONENT("IQ File Source & Sink" ENABLE_FILE gnuradio-blocks_FOUND)
 if(ENABLE_FILE)
 GR_INCLUDE_SUBDIRECTORY(file)
 endif(ENABLE_FILE)
@@ -153,7 +153,7 @@ endif(ENABLE_RTL)
 ########################################################################
 # Setup RTL_TCP component
 ########################################################################
-GR_REGISTER_COMPONENT("RTLSDR TCP Client" ENABLE_RTL_TCP GNURADIO_BLOCKS_FOUND)
+GR_REGISTER_COMPONENT("RTLSDR TCP Client" ENABLE_RTL_TCP gnuradio-blocks_FOUND)
 if(ENABLE_RTL_TCP)
 GR_INCLUDE_SUBDIRECTORY(rtl_tcp)
 endif(ENABLE_RTL_TCP)
-- 
2.20.1

