diff -u --recursive --new-file a/test/layer-tests/meson.build b/test/layer-tests/meson.build
--- a/test/layer-tests/meson.build	2025-01-29 12:10:41.000000000 -0500
+++ b/test/layer-tests/meson.build	2025-03-02 13:59:32.437263471 -0500
@@ -19,7 +19,6 @@
     'test-get-explicit-exclusive-zone',
     'test-get-auto-exclusive-zone',
     'test-exclusive-zone-below-negative-1',
-    'test-layer-surface-popup',
     'test-xdg-toplevel-popup-with-layer-surface',
     'test-xdg-toplevel-popup-without-layer-surface',
     'test-close-layer-surface',
diff -u --recursive --new-file a/test/layer-tests/test-layer-surface-popup.c b/test/layer-tests/test-layer-surface-popup.c
--- a/test/layer-tests/test-layer-surface-popup.c	2025-01-29 12:10:41.000000000 -0500
+++ b/test/layer-tests/test-layer-surface-popup.c	1969-12-31 19:00:00.000000000 -0500
@@ -1,44 +0,0 @@
-#include "integration-test-common.h"
-
-static GtkWindow *layer_window;
-static GtkWidget *layer_dropdown;
-static const char *options[] = {"Foo", "Bar", "Baz", NULL};
-
-static void callback_0() {
-    EXPECT_MESSAGE(zwlr_layer_shell_v1 .get_layer_surface);
-
-    // The popup is weirdly slow to open, so slow the tests down
-    step_time = 600;
-
-    layer_window = GTK_WINDOW(gtk_window_new());
-    layer_dropdown = gtk_drop_down_new_from_strings(options);
-    gtk_window_set_child(layer_window, layer_dropdown);
-    gtk_layer_init_for_window(layer_window);
-    gtk_window_present(layer_window);
-}
-
-static void callback_1() {
-    EXPECT_MESSAGE(xdg_wm_base .get_xdg_surface);
-    EXPECT_MESSAGE(xdg_surface .get_popup nil);
-    EXPECT_MESSAGE(zwlr_layer_surface_v1 .get_popup);
-    EXPECT_MESSAGE(xdg_popup .grab);
-
-    UNEXPECT_MESSAGE(xdg_popup .destroy);
-
-    g_signal_emit_by_name(layer_dropdown, "activate", NULL);
-}
-
-static void callback_2() {
-    EXPECT_MESSAGE(xdg_popup .destroy);
-    EXPECT_MESSAGE(xdg_surface .destroy);
-    EXPECT_MESSAGE(zwlr_layer_surface_v1 .destroy);
-
-    gtk_window_close(layer_window);
-    gtk_window_close(layer_window);
-}
-
-TEST_CALLBACKS(
-    callback_0,
-    callback_1,
-    callback_2,
-)
diff -u --recursive --new-file a/test/smoke-tests/meson.build b/test/smoke-tests/meson.build
--- a/test/smoke-tests/meson.build	2025-01-29 12:10:41.000000000 -0500
+++ b/test/smoke-tests/meson.build	2025-03-02 13:59:18.690961683 -0500
@@ -3,6 +3,5 @@
     'test-c-demo',
     'test-c-session-lock',
     'test-python-example',
-    'test-python-session-lock',
     'test-vala-example',
 ]
diff -u --recursive --new-file a/test/smoke-tests/test-python-session-lock.py b/test/smoke-tests/test-python-session-lock.py
--- a/test/smoke-tests/test-python-session-lock.py	2025-01-29 12:10:41.000000000 -0500
+++ b/test/smoke-tests/test-python-session-lock.py	1969-12-31 19:00:00.000000000 -0500
@@ -1,13 +0,0 @@
-#!/usr/bin/env python3
-
-import os
-import smoke_test_common
-
-script_path = os.path.join(os.path.dirname(__file__), '..', '..', 'examples', 'session-lock.py')
-assert os.path.isfile(script_path), 'script not found at ' + script_path
-src_build_dir = smoke_test_common.build_dir() + '/src'
-env = {
-    'GI_TYPELIB_PATH': src_build_dir,
-    'LD_LIBRARY_PATH': src_build_dir,
-}
-smoke_test_common.run(['python3', script_path], 'lock', env)
