--- a/src/plugins/styles/gtk2/qgtkstyle_p.cpp	2017-03-11 10:23:02.000000000 +0100
+++ b/src/plugins/styles/gtk2/qgtkstyle_p.cpp	2017-03-11 10:23:02.000000000 +0100
@@ -508,7 +508,6 @@
       if (oldTheme != QGtkStylePrivate::getThemeName()) {
           oldTheme = QGtkStylePrivate::getThemeName();
           QPalette newPalette = qApp->style()->standardPalette();
-          QApplicationPrivate::setSystemPalette(newPalette);
           QApplication::setPalette(newPalette);
           if (!QGtkStylePrivate::instances.isEmpty()) {
               QGtkStylePrivate::instances.last()->initGtkWidgets();
--- a/src/plugins/styles/gtk2/qgtkstyle.cpp	2020-06-13 19:48:41.746108796 +0200
+++ b/src/plugins/styles/gtk2/qgtkstyle.cpp	2020-06-13 19:48:41.746108796 +0200
@@ -440,7 +440,7 @@
     // not supported as these should be entirely determined by
     // current Gtk settings
     if (app->desktopSettingsAware() && d->isThemeAvailable()) {
-	QApplicationPrivate::setSystemPalette(standardPalette());
+        QApplication::setPalette(standardPalette());
         QApplicationPrivate::setSystemFont(d->getThemeFont());
         d->applyCustomPaletteHash();
         if (!d->isKDE4Session())
