diff --git src/util.c src/util.c
index 9dbf8af..7155da2 100644
--- src/util.c
+++ src/util.c
@@ -435,7 +435,7 @@ gchar **CacheUtilGlob(const gchar *pattern) {
   glob_t glob_result = {0, NULL, 0};
   gchar **result = NULL;
 
-  switch (glob(pattern, GLOB_BRACE | GLOB_ERR, NULL, &glob_result)) {
+  switch (glob(pattern, GLOB_ERR, NULL, &glob_result)) {
     case 0:
       break;
     case GLOB_NOMATCH:
