--- renderdoc/driver/gl/egl_dispatch_table.h
+++ renderdoc/driver/gl/egl_dispatch_table.h
@@ -59,13 +59,13 @@ typedef const char *(EGLAPIENTRY *PFN_eglQueryString)(EGLDisplay dpy, EGLint nam
 typedef PFNEGLPOSTSUBBUFFERNVPROC PFN_eglPostSubBufferNV;
 
 #define EGL_HOOKED_SYMBOLS(FUNC)    \
+  FUNC(GetProcAddress, false);      \
   FUNC(GetDisplay, false);          \
   FUNC(CreateContext, false);       \
   FUNC(DestroyContext, false);      \
   FUNC(CreateWindowSurface, false); \
   FUNC(MakeCurrent, false);         \
   FUNC(SwapBuffers, false);         \
-  FUNC(GetProcAddress, false);      \
   FUNC(PostSubBufferNV, true);
 
 #define EGL_NONHOOKED_SYMBOLS(FUNC)  \
--- renderdoc/driver/gl/glx_dispatch_table.h
+++ renderdoc/driver/gl/glx_dispatch_table.h
@@ -62,6 +62,8 @@ typedef void (*PFN_glTexCoord2f)(float, float);
 typedef void (*PFN_glEnd)();
 
 #define GLX_HOOKED_SYMBOLS(FUNC)    \
+  FUNC(glXGetProcAddress);          \
+  FUNC(glXGetProcAddressARB);       \
   FUNC(glXCreateContext);           \
   FUNC(glXDestroyContext);          \
   FUNC(glXCreateContextAttribsARB); \
@@ -69,9 +71,7 @@ typedef void (*PFN_glEnd)();
   FUNC(glXMakeContextCurrent);      \
   FUNC(glXSwapBuffers);             \
   FUNC(glXCreateWindow);            \
-  FUNC(glXDestroyWindow);           \
-  FUNC(glXGetProcAddress);          \
-  FUNC(glXGetProcAddressARB);
+  FUNC(glXDestroyWindow);
 
 #define GLX_NONHOOKED_SYMBOLS(FUNC) \
   FUNC(glXGetConfig);               \
