--- a/qrenderdoc/Widgets/RangeHistogram.cpp	2019-04-29 11:50:14.000000000 +0200
+++ b/qrenderdoc/Widgets/RangeHistogram.cpp	2020-08-31 19:10:09.169423623 +0200
@@ -27,6 +27,7 @@
 #include <math.h>
 #include <QMouseEvent>
 #include <QPainter>
+#include <QPainterPath>
 
 RangeHistogram::RangeHistogram(QWidget *parent) : QWidget(parent)
 {
--- a/qrenderdoc/Styles/RDTweakedNativeStyle/RDTweakedNativeStyle.cpp.orig	2019-04-29 11:50:14.000000000 +0200
+++ b/qrenderdoc/Styles/RDTweakedNativeStyle/RDTweakedNativeStyle.cpp	2020-08-31 19:14:26.336436890 +0200
@@ -26,6 +26,7 @@
 #include <QDebug>
 #include <QFontMetrics>
 #include <QPainter>
+#include <QPainterPath>
 #include <QPen>
 #include <QStyleOption>
 
--- a/qrenderdoc/3rdparty/toolwindowmanager/ToolWindowManager.cpp	2019-04-29 11:50:14.000000000 +0200
+++ b/qrenderdoc/3rdparty/toolwindowmanager/ToolWindowManager.cpp	2020-08-31 19:15:33.841440372 +0200
@@ -32,6 +32,7 @@
 #include <QMimeData>
 #include <QMouseEvent>
 #include <QPainter>
+#include <QPainterPath>
 #include <QScreen>
 #include <QSplitter>
 #include <QTabBar>
--- a/qrenderdoc/Windows/TimelineBar.cpp	2019-04-29 11:50:14.000000000 +0200
+++ b/qrenderdoc/Windows/TimelineBar.cpp	2020-08-31 19:16:36.087443584 +0200
@@ -25,6 +25,7 @@
 #include "TimelineBar.h"
 #include <QMouseEvent>
 #include <QPainter>
+#include <QPainterPath>
 #include <QScrollBar>
 #include <QWheelEvent>
 #include "Code/QRDUtils.h"
