--- a/src
+++ b/src
@@ -3103,7 +3103,7 @@ class RCS(RevisionMixin):
         # rcs --version fails under OS X
         # rcs -V throws an obsolescence warning under GNU RCS 5.9.4
         # We choose the more portable option here.
-        rawversion = capture_or_die("rcs -V 2>/dev/null")
+        rawversion = capture_or_die("rcs --version 2>/dev/null")
         m = re.search(r"[0-9]+\.[0-9]+\.[0-9]+".encode("ascii"), rawversion)
         return m and polystr(m.group(0))
