Only in src-1.32: __pycache__
diff -c src-1.32-orig/src src-1.32/src
*** src-1.32-orig/src	2023-04-24 23:14:32.000000000 +1000
--- src-1.32/src	2023-08-17 18:22:48.922724966 +1000
***************
*** 78,84 ****
  
  from __future__ import print_function
  
! # pylint: disable=fixme,line-too-long,too-many-lines,invalid-name,no-value-for-parameter,no-self-use,no-else-break,no-else-return,no-else-continue,missing-function-docstring,comparison-with-callable,consider-using-f-string,too-many-public-methods,too-many-nested-blocks,consider-using-with,too-many-locals.too-many-branches
  
  # pylint: disable=multiple-imports
  import sys, os, subprocess, datetime, time, calendar, stat, glob
--- 78,84 ----
  
  from __future__ import print_function
  
! # pylint: disable=fixme,line-too-long,too-many-lines,invalid-name,no-value-for-parameter,no-else-break,no-else-return,no-else-continue,missing-function-docstring,comparison-with-callable,consider-using-f-string,too-many-public-methods,too-many-nested-blocks,consider-using-with,too-many-locals.too-many-branches
  
  # pylint: disable=multiple-imports
  import sys, os, subprocess, datetime, time, calendar, stat, glob
***************
*** 120,125 ****
--- 120,126 ----
  
  if str is bytes:  # Python 2
  
+     # pylint: disable=deprecated-module
      import cgi
      # pylint: disable=no-member
      htmlescape = cgi.escape
***************
*** 1276,1281 ****
--- 1277,1283 ----
              return metadata.native_to_revno(part)
          self.lo = subresolve(self.start)
          self.hi = subresolve(self.end)
+         # pylint: disable=superfluous-parens
          mreversed = (self.lo > self.hi)
          if mreversed:
              swapme = self.hi
***************
*** 1485,1491 ****
      if differ == difflib.unified_diff:
          colorizer = colorize_unified
      else:
!         colorizer = lambda x: x
      with backend.lifter(metadata.name):
          for line in compute_diff(metadata, lo, hi, differ, ignore_ws):
              sys.stdout.write(colorizer(line) + "\n")
--- 1487,1493 ----
      if differ == difflib.unified_diff:
          colorizer = colorize_unified
      else:
!         colorizer = lambda x: x # pylint: disable=unnecessary-lambda-assignment
      with backend.lifter(metadata.name):
          for line in compute_diff(metadata, lo, hi, differ, ignore_ws):
              sys.stdout.write(colorizer(line) + "\n")
