--- ext/libstrawberry-common/core/logging.cpp	2019-01-03 23:40:30.761082917 +0100
+++ ext/libstrawberry-common/core/logging.cpp	2019-01-03 23:41:07.721257707 +0100
@@ -32,7 +32,7 @@
 #include <stdlib.h>
 #include <string>
 #include <iostream>
-#ifdef Q_OS_UNIX
+#if defined Q_OS_UNIX && defined __GLIBC__
   #include <execinfo.h>
 #endif
 
@@ -249,7 +249,7 @@
 }
 
 void DumpStackTrace() {
-#ifdef Q_OS_UNIX
+#if defined Q_OS_UNIX && defined __GLIBC__
   void* callstack[128];
   int callstack_size = backtrace(reinterpret_cast<void**>(&callstack), sizeof(callstack));
   char** symbols = backtrace_symbols(reinterpret_cast<void**>(&callstack), callstack_size);
