--- lib/meta.py	2020-02-15 16:05:22.000000000 +0100
+++ lib/meta.py	2020-03-29 11:40:40.275375283 +0200
@@ -264,7 +264,7 @@
         # If release information from release.sh exists, use that
         relinfo = {}
         with open("release_info", "rb") as relinfo_fp:
-            exec(relinfo_fp, relinfo)
+            exec(relinfo_fp.read(), relinfo)
         base_version = relinfo.get(
             "MYPAINT_VERSION_BASE",
             base_version,
