--- src/k3bsystemproblemdialog.cpp.orig	2018-05-31 08:02:32.713467562 +0200
+++ src/k3bsystemproblemdialog.cpp	2018-05-31 08:02:42.902478072 +0200
@@ -57,7 +57,7 @@
 #include <langinfo.h>
 #endif
 
-#ifndef Q_OS_WIN32
+#ifdef __GLIBC__
 #include <fstab.h>
 #endif
 #include <unistd.h>
@@ -394,7 +394,7 @@
             dvd_r_dl = true;
     }
 
-#ifndef Q_OS_WIN32
+#ifdef __GLIBC__
     // check automounted devices
     QList<K3b::Device::Device*> automountedDevices = checkForAutomounting();
     for( QList<K3b::Device::Device *>::const_iterator it = automountedDevices.constBegin();
@@ -681,7 +681,7 @@
 }
 
 
-#ifndef Q_OS_WIN32
+#ifdef __GLIBC__
 QList<K3b::Device::Device*> K3b::SystemProblemDialog::checkForAutomounting()
 {
     QList<K3b::Device::Device *> l;
