In musl libc there is no fstab.h

--- src/k3bsystemproblemdialog.cpp	2014-11-04 19:37:31.000000000 +0100
+++ src/k3bsystemproblemdialog.cpp	2016-09-22 17:06:44.365902832 +0200
@@ -53,7 +53,7 @@
 #include <langinfo.h>
 #endif
 
-#ifndef Q_OS_WIN32
+#if !defined(Q_OS_WIN32) && defined(__GLIBC__)
 #include <fstab.h>
 #endif
 #include <unistd.h>
@@ -651,6 +651,7 @@
 QList<K3b::Device::Device*> K3b::SystemProblemDialog::checkForAutomounting()
 {
     QList<K3b::Device::Device *> l;
+#if defined(__GLIBC__)
     ::setfsent();
 
     struct fstab * mountInfo = 0;
@@ -676,6 +677,7 @@
     } // while mountInfo
 
     ::endfsent();
+#endif
     return l;
 }
 #endif
