--- misc/sys/xattr.h.orig	2009-04-24 15:56:12.517504626 +0200
+++ misc/sys/xattr.h	2009-04-24 15:56:27.825593128 +0200
@@ -30,9 +30,7 @@ __BEGIN_DECLS
 enum
 {
   XATTR_CREATE = 1,	/* set value, fail if attr already exists.  */
-#define XATTR_CREATE	XATTR_CREATE
   XATTR_REPLACE = 2	/* set value, fail if attr does not exist.  */
-#define XATTR_REPLACE	XATTR_REPLACE
 };
 
 /* Set the attribute NAME of the file pointed to by PATH to VALUE (which
