diff --git a/src/git.cc b/src/git.cc
index 9da2d7b..45e9007 100644
--- src/git.cc
+++ src/git.cc
@@ -6,7 +6,7 @@ bool Git::initialized = false;
 std::mutex Git::mutex;
 
 std::string Git::Error::message() noexcept {
-  const git_error *last_error = giterr_last();
+  const git_error *last_error = git_error_last();
   if(last_error == nullptr)
     return std::string();
   else
@@ -244,7 +244,7 @@ boost::filesystem::path Git::Repository::get_root_path(const boost::filesystem::
       throw std::runtime_error(error.message());
   }
   auto root_path = Git::path(root.ptr, root.size);
-  git_buf_free(&root);
+  git_buf_dispose(&root);
   return root_path;
 }
 

