--- a/sip/cpp/sip_propgridwxPGPropArgCls.cpp
+++ b/sip/cpp/sip_propgridwxPGPropArgCls.cpp
@@ -317,7 +317,7 @@ static int convertTo_wxPGPropArgCls(PyOb
             return sipGetState(sipTransferObj);
         }
         else if (sipPy == Py_None) {
-            *sipCppPtr = new wxPGPropArgCls(reinterpret_cast< wxPGProperty * >(NULL));
+            *sipCppPtr = new wxPGPropArgCls(static_cast< wxPGProperty * >(NULL));
             return sipGetState(sipTransferObj);
         }
         else {
