Include QDataStream for the required definitions

--- src/lib/3rdparty/qtsingleapplication/qtlocalpeer.cpp	2015-01-26 15:26:15.000000000 +0100
+++ src/lib/3rdparty/qtsingleapplication/qtlocalpeer.cpp	2015-07-06 18:32:29.862136464 +0200
@@ -48,6 +48,7 @@
 #include "qtlocalpeer.h"
 #include <QtCore/QCoreApplication>
 #include <QtCore/QTime>
+#include <QtCore/QDataStream>
 
 #if defined(Q_OS_WIN)
 #include <QtCore/QLibrary>
