Index: libgcrypt-1.10.1/src/libgcrypt-config.in
===================================================================
--- libgcrypt-1.10.1.orig/src/libgcrypt-config.in
+++ libgcrypt-1.10.1/src/libgcrypt-config.in
@@ -154,7 +154,7 @@ if test "$echo_cflags" = "yes"; then
 
     tmp=""
     for i in $includes $cflags_final; do
-       if echo "$tmp" | fgrep -v -- "$i" >/dev/null; then
+       if echo "$tmp" | grep -F -v -- "$i" >/dev/null; then
            tmp="$tmp $i"
        fi
     done
@@ -175,7 +175,7 @@ if test "$echo_libs" = "yes"; then
 
     tmp=""
     for i in $libdirs $libs_final; do
-       if echo "$tmp" | fgrep -v -- "$i" >/dev/null; then
+       if echo "$tmp" | grep -F -v -- "$i" >/dev/null; then
            tmp="$tmp $i"
        fi
     done
