--- a/cert-to-efi-hash-list.c	2018-04-21 20:59:24.814748503 +0200
+++ b/cert-to-efi-hash-list.c	2018-04-21 20:59:51.868581307 +0200
@@ -135,7 +135,7 @@
         X509 *cert = PEM_read_bio_X509(cert_bio, NULL, NULL, NULL);
 	unsigned char *cert_buf = NULL;
 
-#if OPENSSL_VERSION_NUMBER < 0x10100000L
+#if OPENSSL_VERSION_NUMBER < 0x10100000L || defined(LIBRESSL_VERSION_NUMBER)
 	int cert_len = i2d_X509_CINF(cert->cert_info, &cert_buf);
 #else
 	int cert_len = i2d_re_X509_tbs(cert, &cert_buf);
