diff --git a/Make.rules b/Make.rules
index 903a5a4..4a400d5 100644
--- a/Make.rules
+++ b/Make.rules
@@ -132,4 +132,12 @@ getvar = $(shell if [ "$(1)" = "PK" -o "$(1)" = "KEK" ]; then echo $(1); else ec
 	ar rcv $@ $^
 
 doc/%.1: doc/%.1.in %
-	$(HELP2MAN) --no-info -i $< -o $@ ./$*
+	mkdir -p help2man-cross-shims
+	printf '#!/bin/sh\ncase "$$1" in\n\t--version) printf "$* %%s\\n" ' > help2man-cross-shims/$*
+	sed -nE 's/^#define VERSION ("[^"]+")$$/\1/p' include/version.h >> help2man-cross-shims/$*
+	printf '\t\t;;\n\t--help) cat <<"EOT"\n' >> help2man-cross-shims/$*
+	grep -Eazo 'Usage: %s.*$$' $* | head -c -1 | sed -E 's/%s/$*/' >> help2man-cross-shims/$*
+	grep -Eazo '[[:print:][:space:]]*Options:.*' $* | head -c -1 >> help2man-cross-shims/$*
+	printf 'EOT\n\t\t;;\nesac\n' >> help2man-cross-shims/$*
+	chmod +x help2man-cross-shims/$*
+	$(HELP2MAN) --no-info -i $< -o $@ help2man-cross-shims/$*
