--- a/wezterm-font/src/parser.rs	2022-09-05 10:28:02.000000000 -0700
+++ b/wezterm-font/src/parser.rs	2022-10-24 09:06:02.529054286 -0700
@@ -715,19 +715,6 @@
         font!("../../assets/fonts/JetBrainsMono-Regular.ttf"),
         font!("../../assets/fonts/JetBrainsMono-ThinItalic.ttf"),
         font!("../../assets/fonts/JetBrainsMono-Thin.ttf"),
-        font!("../../assets/fonts/Roboto-Black.ttf"),
-        font!("../../assets/fonts/Roboto-BlackItalic.ttf"),
-        font!("../../assets/fonts/Roboto-Bold.ttf"),
-        font!("../../assets/fonts/Roboto-BoldItalic.ttf"),
-        font!("../../assets/fonts/Roboto-Italic.ttf"),
-        font!("../../assets/fonts/Roboto-Light.ttf"),
-        font!("../../assets/fonts/Roboto-LightItalic.ttf"),
-        font!("../../assets/fonts/Roboto-Medium.ttf"),
-        font!("../../assets/fonts/Roboto-MediumItalic.ttf"),
-        font!("../../assets/fonts/Roboto-Regular.ttf"),
-        font!("../../assets/fonts/Roboto-Thin.ttf"),
-        font!("../../assets/fonts/Roboto-ThinItalic.ttf"),
-        font!("../../assets/fonts/NotoColorEmoji.ttf"),
         font!("../../assets/fonts/Symbols-Nerd-Font-Mono.ttf"),
         font!("../../assets/fonts/LastResortHE-Regular.ttf"),
     ] {
