--- deps/freetype/build.rs
+++ deps/freetype/build.rs
@@ -74,9 +74,16 @@ fn freetype() {
 }
 
 fn main() {
-    zlib();
-    libpng();
-    freetype();
+    println!("cargo:rerun-if-env-changed=WEZTERM_SYSDEPS");
+    if cfg!(unix) && env::var("WEZTERM_SYSDEPS").map(|x| x == "1").unwrap_or(false) {
+        println!("cargo:rustc-link-lib=z");
+        println!("cargo:rustc-link-lib=png");
+        println!("cargo:rustc-link-lib=freetype");
+    } else {
+        zlib();
+        libpng();
+        freetype();
+    }
     let out_dir = env::var("OUT_DIR").unwrap();
     println!("cargo:outdir={}", out_dir);
     println!("cargo:rustc-env=MACOSX_DEPLOYMENT_TARGET=10.9");
--- deps/harfbuzz/build.rs
+++ deps/harfbuzz/build.rs
@@ -45,7 +45,12 @@ fn emit_libdirs(p: &Path) {
 }
 
 fn main() {
-    harfbuzz();
+    println!("cargo:rerun-if-env-changed=WEZTERM_SYSDEPS");
+    if cfg!(unix) && env::var("WEZTERM_SYSDEPS").map(|x| x == "1").unwrap_or(false) {
+        println!("cargo:rustc-link-lib=harfbuzz");
+    } else {
+        harfbuzz();
+    }
     let out_dir = env::var("OUT_DIR").unwrap();
     println!("cargo:outdir={}", out_dir);
     println!("cargo:rustc-env=MACOSX_DEPLOYMENT_TARGET=10.9");
