--- a/config/src/font.rs	2022-09-05 10:28:02.000000000 -0700
+++ b/config/src/font.rs	2022-11-24 11:27:15.115251410 -0700
@@ -407,7 +407,7 @@
 impl Default for FontAttributes {
     fn default() -> Self {
         Self {
-            family: "JetBrains Mono".into(),
+            family: "monospace".into(),
             weight: FontWeight::default(),
             stretch: FontStretch::default(),
             style: FontStyle::Normal,
