--- a/internal/frontend/bridge-gui/bridgepp/CMakeLists.txt.orig
+++ b/internal/frontend/bridge-gui/bridgepp/CMakeLists.txt
@@ -48,7 +48,7 @@
 #****************************************************************************************************************************************************
 # gRPC / Protobuf
 #****************************************************************************************************************************************************
-find_package(Protobuf CONFIG REQUIRED)
+find_package(Protobuf REQUIRED)
 message(STATUS "Using protobuf ${Protobuf_VERSION}")
 
 find_package(gRPC CONFIG REQUIRED)
