diff --git requests/certs.py requests/certs.py
index 07e6475..2f2d79b 100644
--- requests/certs.py
+++ requests/certs.py
@@ -18,8 +18,7 @@ try:
 except ImportError:
     def where():
         """Return the preferred certificate bundle."""
-        # vendored bundle inside Requests
-        return os.path.join(os.path.dirname(__file__), 'cacert.pem')
+        return '/etc/ssl/certs/ca-certificates.crt'
 
 if __name__ == '__main__':
     print(where())
