--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -283,7 +283,6 @@ find_package(Boost ${MINIMUM_BOOST_VERSI
         log
         log_setup
         serialization
-        system
         thread
     REQUIRED)
 
--- a/GG/CMakeLists.txt
+++ b/GG/CMakeLists.txt
@@ -66,7 +66,7 @@ endif()
 
 set(MINIMUM_BOOST_VERSION 1.69.0)
 
-find_package(Boost ${MINIMUM_BOOST_VERSION} COMPONENTS filesystem regex system REQUIRED)
+find_package(Boost ${MINIMUM_BOOST_VERSION} COMPONENTS filesystem REQUIRED)
 
 find_package(OpenGL REQUIRED)
 find_package(GLEW REQUIRED)
