Index: CMakeLists.txt
===================================================================
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -432,13 +432,6 @@ if(MSVC)
     )
 endif()
 
-target_compile_options(freeorionparseobj
-    PRIVATE
-        $<$<CXX_COMPILER_ID:Clang>:-ftemplate-depth=512>
-        $<$<CXX_COMPILER_ID:AppleClang>:-ftemplate-depth=512>
-        $<$<AND:$<NOT:$<BOOL:${BUILD_TESTING}>>,$<CXX_COMPILER_ID:GNU>>:-O3>
-)
-
 target_compile_definitions(freeorionparseobj
     PUBLIC
         -DBOOST_ALL_DYN_LINK
