--- src/compositor/meta-compositor-vulkan.c	2020-03-06 18:04:14.000000000 +0100
+++ src/compositor/meta-compositor-vulkan.c	2020-09-03 20:33:53.599821164 +0200
@@ -810,7 +810,9 @@
         return "cpu";
         break;
 
+#if defined(VK_PHYSICAL_DEVICE_TYPE_RANGE_SIZE)
       case VK_PHYSICAL_DEVICE_TYPE_RANGE_SIZE:
+#endif
       case VK_PHYSICAL_DEVICE_TYPE_MAX_ENUM:
       default:
         break;
