From 54aec0a4738250043726cb0e1276369fbb045eb1 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Piotr=20W=C3=B3jcik?= <chocimier@tlen.pl>
Date: Mon, 13 May 2019 12:07:02 +0200
Subject: [PATCH] cross


diff --git CMakeLists.txt CMakeLists.txt
index 16f3e5a..f959bf0 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -1,15 +1,15 @@
 cmake_minimum_required(VERSION 3.0.0)
+project(QAccessibilityClient VERSION "0.4.0")
 
-include(FeatureSummary)
 find_package(ECM "5.18.0" REQUIRED NO_MODULE)
-set_package_properties(ECM PROPERTIES TYPE REQUIRED DESCRIPTION "Extra CMake Modules." URL "https://projects.kde.org/projects/kdesupport/extra-cmake-modules")
-feature_summary(WHAT REQUIRED_PACKAGES_NOT_FOUND FATAL_ON_MISSING_REQUIRED_PACKAGES)
-set(CMAKE_MODULE_PATH ${ECM_MODULE_PATH})
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${ECM_MODULE_PATH})
 
+include(KDEInstallDirs)
+include(KDECompilerSettings NO_POLICY_SCOPE)
+include(KDECMakeSettings)
 include(ECMSetupVersion)
 
 cmake_policy(SET CMP0048 NEW)
-project(QAccessibilityClient VERSION "0.4.0")
 
 ecm_setup_version(${PROJECT_VERSION}
     VARIABLE_PREFIX QACCESSIBILITYCLIENT
@@ -60,11 +60,8 @@ else (WIN32)
    set(INCLUDE_INSTALL_DIR  "${CMAKE_INSTALL_PREFIX}/include")
 endif (WIN32)
 
-set(_QAccessibilityClient_CONFIG_DEST "${LIB_INSTALL_DIR}/cmake/${PROJECT_NAME}")
+set(_QAccessibilityClient_CONFIG_DEST "${KDE_INSTALL_CMAKEPACKAGEDIR}/${PROJECT_NAME}")
 
-set(INSTALL_TARGETS_DEFAULT_ARGS  RUNTIME DESTINATION "${BIN_INSTALL_DIR}"
-                                  LIBRARY DESTINATION "${LIB_INSTALL_DIR}"
-                                  ARCHIVE DESTINATION "${LIB_INSTALL_DIR}" COMPONENT Devel )
 if(APPLE)
    set(INSTALL_TARGETS_DEFAULT_ARGS  ${INSTALL_TARGETS_DEFAULT_ARGS}
                                BUNDLE DESTINATION "${BUNDLE_INSTALL_DIR}" )
@@ -103,3 +100,7 @@ install( FILES
          "${QAccessibilityClient_BINARY_DIR}/QAccessibilityClientConfig.cmake"
          "${QAccessibilityClient_BINARY_DIR}/QAccessibilityClientConfigVersion.cmake"
          DESTINATION "${_QAccessibilityClient_CONFIG_DEST}" )
+
+include(FeatureSummary)
+set_package_properties(ECM PROPERTIES TYPE REQUIRED DESCRIPTION "Extra CMake Modules." URL "https://projects.kde.org/projects/kdesupport/extra-cmake-modules")
+feature_summary(WHAT REQUIRED_PACKAGES_NOT_FOUND FATAL_ON_MISSING_REQUIRED_PACKAGES)
diff --git src/CMakeLists.txt src/CMakeLists.txt
index ecf8deb..9b05c79 100644
--- src/CMakeLists.txt
+++ src/CMakeLists.txt
@@ -31,10 +31,10 @@ set_target_properties(${QACCESSIBILITYCLIENT_LIB_NAME}
     PROPERTIES VERSION ${PROJECT_VERSION}
     SOVERSION ${QACCESSIBILITYCLIENT_SOVERSION}
 )
-target_include_directories(${QACCESSIBILITYCLIENT_LIB_NAME} INTERFACE "$<INSTALL_INTERFACE:${INCLUDE_INSTALL_DIR}>" )
+target_include_directories(${QACCESSIBILITYCLIENT_LIB_NAME} INTERFACE "$<INSTALL_INTERFACE:${KDE_INCLUDE_INSTALL_DIR}>" )
 
 
-install(TARGETS ${QACCESSIBILITYCLIENT_LIB_NAME} EXPORT QAccessibilityClient ${INSTALL_TARGETS_DEFAULT_ARGS})
+install(TARGETS ${QACCESSIBILITYCLIENT_LIB_NAME} EXPORT QAccessibilityClient ${KDE_INSTALL_TARGETS_DEFAULT_ARGS})
 install(EXPORT QAccessibilityClient DESTINATION ${_QAccessibilityClient_CONFIG_DEST})
 
 ########### install files ###############
-- 
2.21.0

