--- a/src/controller.cpp
+++ b/src/controller.cpp
@@ -98,7 +98,7 @@ Controller::set_active(bool v)
 {
   if (m_is_active != v)
   {
-    log_debug("activation status: " << v << " " << m_activation_cb);
+    log_debug("activation status: " << v << " m_activation_cb");
     m_is_active = v;
     if (m_activation_cb)
     {
--- a/src/controller_slot.cpp
+++ b/src/controller_slot.cpp
@@ -69,7 +69,7 @@ ControllerSlot::disconnect()
 bool
 ControllerSlot::is_connected() const
 {
-  return m_thread;
+  return m_thread != nullptr;
 }
 
 /* EOF */
