diff -ru a/tests/meson.build b/tests/meson.build
--- a/tests/meson.build
+++ b/tests/meson.build
@@ -61,7 +61,6 @@
 
     if host_system != 'darwin'
       tests += [
-        [ 'test-layout', [ 'test-layout.c', 'test-common.c' ], [ libpangocairo_dep, libpangoft2_dep ] ],
         [ 'test-fonts', [ 'test-fonts.c', 'test-common.c' ], [ libpangocairo_dep, libpangoft2_dep ] ],
         [ 'test-no-fonts', [ 'test-no-fonts.c' ], [ libpangocairo_dep, libpangoft2_dep ] ],
       ]
