diff --git Makefile.am Makefile.am
index 44d0d76..ee66e54 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -28,7 +28,7 @@ numademo_CFLAGS = $(AM_CFLAGS) -O3 -ffast-math -funroll-loops
 if HAVE_TREE_VECTORIZE
 numademo_CFLAGS += -ftree-vectorize
 endif
-numademo_LDADD = libnuma.la -lm
+numademo_LDADD = libnuma.la -lm $(AM_LDFLAGS)
 
 migratepages_SOURCES = migratepages.c util.c
 migratepages_LDADD = libnuma.la
