--- a/gradia/graphics/gradient.py
+++ b/gradia/graphics/gradient.py
@@ -97,9 +97,7 @@
         if cls._c_lib is not None:
             return

-        from importlib.resources import files
-        gradia_path = files("gradia").joinpath("libgradient_gen.so")
-        cls._c_lib = CDLL(str(gradia_path))
+        cls._c_lib = CDLL("libgradient_gen.so")
         cls._c_lib.generate_gradient.argtypes = [
             POINTER(c_uint8), c_int, c_int,
             POINTER(ColorStop), c_int,
