--- a/configure
+++ b/configure
@@ -8658,6 +8658,10 @@ case "$am__api_version" in
     ;;
 esac
 
+INTLTOOL_UPDATE=/bin/true
+INTLTOOL_MERGE=/bin/true
+INTLTOOL_EXTRACT=/bin/true
+INTLTOOL_PERL=/bin/true
 INTLTOOL_REQUIRED_VERSION_AS_INT=`echo  | awk -F. '{ print $ 1 * 1000 + $ 2 * 100 + $ 3; }'`
 INTLTOOL_APPLIED_VERSION=`intltool-update --version | head -1 | cut -d" " -f3`
 INTLTOOL_APPLIED_VERSION_AS_INT=`echo $INTLTOOL_APPLIED_VERSION | awk -F. '{ print $ 1 * 1000 + $ 2 * 100 + $ 3; }'`
--- a/configure.ac
+++ b/configure.ac
@@ -44,7 +44,6 @@ AX_PROG_CXX_FOR_BUILD
 AC_PROG_LN_S
 AC_PROG_EGREP
 
-IT_PROG_INTLTOOL
 LT_INIT
 
 dnl  ------------------------------------
--- a/data/Makefile.in
+++ b/data/Makefile.in
@@ -579,7 +579,8 @@ uninstall-am: uninstall-xmlDATA
 .PRECIOUS: Makefile
 
 
-@INTLTOOL_QALCULATE_DEFINITIONS_RULE@
+%.xml: %.xml.in
+	$(AM_V_GEN)GETTEXTDATADIRS=$(srcdir) $(MSGFMT) --xml --template $< -d $(top_srcdir)/po-defs -o $@
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
--- /dev/null
+++ b/data/its/qalculate.its
@@ -0,0 +1,7 @@
+<?xml version="1.0"?>
+<its:rules xmlns:its="http://www.w3.org/2005/11/its" version="2.0">
+  <its:translateRule selector="//*" translate="no"/>
+  <its:translateRule
+    selector="//title|//name|//names|//countries|//copyright|//description"
+    translate="yes"/>
+</its:rules>
--- /dev/null
+++ b/data/its/qalculate.loc
@@ -0,0 +1,6 @@
+<?xml version="1.0"?>
+<locatingRules>
+  <locatingRule name="qalculate" pattern="*.xml">
+    <documentRule localName="QALCULATE" target="qalculate.its"/>
+  </locatingRule>
+</locatingRules>
--- a/data/Makefile.am
+++ b/data/Makefile.am
@@ -11,7 +11,8 @@ else
 xml_DATA = $(xml_in_files:.xml.in=.xml) eurofxref-daily.xml rates.json
 endif
 
-@INTLTOOL_QALCULATE_DEFINITIONS_RULE@
+%.xml: %.xml.in
+	$(AM_V_GEN)GETTEXTDATADIRS=$(srcdir) $(MSGFMT) --xml --template $< -d $(top_srcdir)/po-defs -o $@
 
 EXTRA_DIST = $(xml_in_files) eurofxref-daily.xml rates.json
 
