Fix the build for python3-poetry-core>=2.0

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,9 +1,9 @@
-[tool.poetry]
+[project]
 name = "anytree"
 version = "2.12.1"
 description = "Powerful and Lightweight Python Tree Data Structure with various plugins"
 authors = [
-    "c0fec0de <c0fec0de@gmail.com>"
+    { name = "c0fec0de", email = "c0fec0de@gmail.com" }
 ]
 readme = "README.rst"
 license = "Apache-2.0"
