--- bloaty-1.1/CMakeLists.txt	2020-05-24 06:43:55.000000000 +0200
+++ -	2023-07-08 12:51:24.115742363 +0200
@@ -2,7 +2,7 @@
 cmake_policy(SET CMP0048 NEW)
 project (Bloaty VERSION 1.0)
 project (Bloaty VERSION 1.1)
-set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD 17)
 
 # Options we define for users.
 option(BLOATY_ENABLE_ASAN "Enable address sanitizer." OFF)
@@ -93,7 +93,7 @@
 include_directories("${CMAKE_CURRENT_BINARY_DIR}/src")
 
 # Baseline build flags.
-set(CMAKE_CXX_FLAGS "-std=c++11 -W -Wall -Wno-sign-compare")
+set(CMAKE_CXX_FLAGS "-W -Wall -Wno-sign-compare")
 set(CMAKE_CXX_FLAGS_DEBUG "-g1")
 set(CMAKE_CXX_FLAGS_RELEASE "-O2")
 set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "-O2 -g1")
