--- a/meson.build	2021-04-05 19:05:59.233949995 +0200
+++ b/meson.build	2021-04-05 19:06:10.010949599 +0200
@@ -61,7 +61,7 @@ includedir = join_paths(prefix, get_opti
 
 # gobject introspection
 gir = find_program('g-ir-scanner', required: get_option('introspection'))
-enable_introspection = gir.found() and not meson.is_cross_build()
+enable_introspection = gir.found()
 
 # vala
 vapi_opt = get_option('vapi')
@@ -76,7 +76,7 @@ endif
 # gtk-doc
 if not get_option('docs').disabled()
   gtk_doc = find_program('gtkdoc-scan', required: get_option('docs'))
-  enable_doc = gtk_doc.found() and not meson.is_cross_build()
+  enable_doc = gtk_doc.found()
 else
   enable_doc = false
 endif
