--- src/pingus/pingus_main.cpp~	2011-12-24 15:46:47.000000000 -0600
+++ src/pingus/pingus_main.cpp	2012-01-03 13:13:03.034195720 -0600
@@ -465,7 +465,7 @@
 void
 PingusMain::print_greeting_message()
 {
-  std::string greeting = "Welcome to Pingus "VERSION;
+  std::string greeting = "Welcome to Pingus " VERSION;
   greeting += "!";
   std::cout <<  greeting << std::endl;
   for (unsigned int i = 0; i < greeting.length(); ++i)
--- src/pingus/screens/pingus_menu.cpp~	2011-12-24 15:46:47.000000000 -0600
+++ src/pingus/screens/pingus_menu.cpp	2012-01-03 13:30:28.967700554 -0600
@@ -153,7 +153,7 @@
                          gc.get_height()/2 - 280));
 
   gc.print_left(Fonts::pingus_small, Vector2i(gc.get_width()/2 - 400 + 25, gc.get_height()-140),
-                "Pingus "VERSION" - Copyright (C) 1998-2011 Ingo Ruhnke <grumbel@gmail.com>\n"
+                "Pingus " VERSION " - Copyright (C) 1998-2011 Ingo Ruhnke <grumbel@gmail.com>\n"
                 "See the file AUTHORS for a complete list of contributors.\n"
                 "Pingus comes with ABSOLUTELY NO WARRANTY. This is free software, and you are\n"
                 "welcome to redistribute it under certain conditions; see the file COPYING for details.\n");
