Description: Use python3 print()
Author: Phil Wyett <philip.wyett@kathenas.org>
Last-Update: 2019-12-29
Bug-Debian: https://bugs.debian.org/947574

Index: pingus/SConscript
===================================================================
--- a/SConscript
+++ b/SConscript
@@ -126,12 +126,12 @@ class Project:
     def configure_end(self):
         self.env = self.conf.Finish()
 
-        print "Reports:"
-        print self.reports
+        print ("Reports:")
+        print (self.reports)
 
         if not self.fatal_error == "":
-            print "Fatal Errors:"
-            print self.fatal_error
+            print ("Fatal Errors:")
+            print (self.fatal_error)
             Exit(1)
 
     def configure_gxx(self): 
