From ebb50839a560134d44d7a7512faae15deae9c0fb Mon Sep 17 00:00:00 2001
From: Stefano Ragni <stefano.ragni@outlook.com>
Date: Sun, 18 Apr 2021 19:55:19 +0200
Subject: [PATCH] wlr_screencast: fix return value on cmd failure

---
 src/screencast/wlr_screencast.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/screencast/wlr_screencast.c b/src/screencast/wlr_screencast.c
index 45bea73..a3b0637 100644
--- a/src/screencast/wlr_screencast.c
+++ b/src/screencast/wlr_screencast.c
@@ -406,7 +406,7 @@ static bool wlr_output_chooser(struct xdpw_output_chooser *chooser,
 
 	if (!wait_chooser(pid)) {
 		close(chooser_out[0]);
-		goto end;
+		return false;
 	}
 
 	FILE *f = fdopen(chooser_out[0], "r");
