--- es-core/src/Util.cpp	2016-03-13 20:31:00.000000000 +0100
+++ -	2018-10-05 11:16:58.054469211 +0200
@@ -32,7 +32,7 @@
 }
 
 
-#if _MSC_VER < 1800
+#if 0
 float round(float num)
 {
 	return (float)((int)(num + 0.5f));
--- es-core/src/Util.h	2016-03-13 20:31:00.000000000 +0100
+++ -	2018-10-05 11:17:16.991980845 +0200
@@ -15,7 +15,7 @@
 Eigen::Vector3f roundVector(const Eigen::Vector3f& vec);
 Eigen::Vector2f roundVector(const Eigen::Vector2f& vec);
 
-#if _MSC_VER < 1800
+#if 0
 float round(float num);
 #endif /* _MSC_VER */
 
