--- liblangtag/lt-tag.h	2016-03-23 03:58:27.000000000 +0100
+++ liblangtag/lt-tag.h	2016-04-08 15:50:43.747557016 +0200
@@ -53,8 +53,6 @@
  * This is used as a binary count sequence with lt_tag_get_string_with_filter()
  * to allow only given tags to be the outcome.
  */
-typedef enum _lt_tag_filter_t	lt_tag_filter_t;
-
 enum _lt_tag_filter_t {
 	LT_TAG_FILTER_NONE          = 0,
 	LT_TAG_FILTER_LANGUAGE      = 1 << 0,
@@ -68,6 +66,8 @@
 	LT_TAG_FILTER_ALL           = -1
 };
 
+typedef enum _lt_tag_filter_t	lt_tag_filter_t;
+
 lt_tag_t                 *lt_tag_new                         (void);
 lt_tag_t                 *lt_tag_ref                         (lt_tag_t       *tag);
 void                      lt_tag_unref                       (lt_tag_t       *tag);
