diff --git a/meson.build b/meson.build
index 7dfcb71..066ceb5 100644
--- a/meson.build
+++ b/meson.build
@@ -845,10 +845,10 @@ else
 endif
 
 # Ancillary files test
-test_ancillary_files_sh = find_program('test-ancillary-files.sh', dirs : scriptsdir, required : true)
-test('Ancillary files', test_ancillary_files_sh, args: [meson.current_source_dir()], timeout: 100, suite : 'analysis')
+# test_ancillary_files_sh = find_program('test-ancillary-files.sh', dirs : scriptsdir, required : true)
+# test('Ancillary files', test_ancillary_files_sh, args: [meson.current_source_dir()], timeout: 100, suite : 'analysis')
 
-summary({'Ancillary files' : ['Test runs:', true]}, section : 'Testing', bool_yn : true)
+# summary({'Ancillary files' : ['Test runs:', true]}, section : 'Testing', bool_yn : true)
 
 # Unit tests
 if conf_data.get('ENABLE_UNIT_TESTS', 0) == 1
