--- a/test/unit/account_info/fixtures.py
+++ b/test/unit/account_info/fixtures.py
@@ -9,6 +9,7 @@
 ######################################################################
 
 import pytest
+import pytest_lazyfixture
 
 from apiver_deps import InMemoryAccountInfo, SqliteAccountInfo
 
@@ -84,8 +85,8 @@ def sqlite_account_info(sqlite_account_i
 
 @pytest.fixture(
     params=[
-        pytest.lazy_fixture('in_memory_account_info_factory'),
-        pytest.lazy_fixture('sqlite_account_info_factory'),
+        pytest_lazyfixture.lazy_fixture('in_memory_account_info_factory'),
+        pytest_lazyfixture.lazy_fixture('sqlite_account_info_factory'),
     ]
 )
 def account_info_factory(request):
@@ -94,8 +95,8 @@ def account_info_factory(request):
 
 @pytest.fixture(
     params=[
-        pytest.lazy_fixture('in_memory_account_info'),
-        pytest.lazy_fixture('sqlite_account_info'),
+        pytest_lazyfixture.lazy_fixture('in_memory_account_info'),
+        pytest_lazyfixture.lazy_fixture('sqlite_account_info'),
     ]
 )
 def account_info(request):
