From 6f0c8a73fd25fd05f528f79ac204a25657cebab7 Mon Sep 17 00:00:00 2001
From: Cristian Le <git@lecris.dev>
Date: Wed, 4 Sep 2024 13:42:53 +0200
Subject: [PATCH 1/2] type-erase lo_blob_dataptr input

---
 pyliblo3/_liblo.pyx | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pyliblo3/_liblo.pyx b/pyliblo3/_liblo.pyx
index 536cbf9..4fa70df 100644
--- a/pyliblo3/_liblo.pyx
+++ b/pyliblo3/_liblo.pyx
@@ -271,7 +271,7 @@ cdef int _msg_callback(const_char *path, const_char *types, lo_arg **argv,
         elif t == 'm': v = (argv[i].m[0], argv[i].m[1], argv[i].m[2], argv[i].m[3])
         elif t == 't': v = _timetag_to_double(argv[i].t)
         elif t == 'b':
-            v = bytes(<unsigned char*>lo_blob_dataptr(argv[i]))
+            v = bytes(<unsigned char*>lo_blob_dataptr(<lo_blob>argv[i]))
         else:
             v = None  # unhandled data type
