--- a/3rdparty/simple-protobuf/cmake/spb_protobuf.cmake
+++ b/3rdparty/simple-protobuf/cmake/spb_protobuf.cmake
@@ -129,7 +129,7 @@ function(spb_protobuf_generate)
     else()
       add_custom_command(
         OUTPUT ${_generated_srcs}
-        COMMAND $<TARGET_FILE:spb-protoc> ARGS ${spb_PROTOC_OPTIONS} --cpp_out ${spb_PROTOC_OUT_DIR} ${_protobuf_include_path} ${_abs_file}
+        COMMAND ${CMAKE_CROSSCOMPILING_EMULATOR} $<TARGET_FILE:spb-protoc> ARGS ${spb_PROTOC_OPTIONS} --cpp_out ${spb_PROTOC_OUT_DIR} ${_protobuf_include_path} ${_abs_file}
         DEPENDS ${_abs_file} spb-protoc ${spb_DEPENDENCIES}
         COMMENT ${_comment}
         VERBATIM )