--- go/src/github.com/influxdata/chronograf/Makefile.orig	2017-07-04 19:52:11.603513489 -0500
+++ go/src/github.com/influxdata/chronograf/Makefile	2017-07-04 19:53:43.031014354 -0500
@@ -8,7 +8,7 @@
 SOURCES := $(shell find . -name '*.go' ! -name '*_gen.go' -not -path "./vendor/*" )
 UISOURCES := $(shell find ui -type f -not \( -path ui/build/\* -o -path ui/node_modules/\* -prune \) )
 
-LDFLAGS=-ldflags "-s -X main.version=${VERSION} -X main.commit=${COMMIT}"
+GOLDFLAGS=-ldflags "-s -X main.version=${VERSION} -X main.commit=${COMMIT}"
 BINARY=chronograf
 
 .DEFAULT_GOAL := all
@@ -20,7 +20,7 @@
 dev: dep dev-assets ${BINARY}
 
 ${BINARY}: $(SOURCES) .bindata .jsdep .godep
-	go build -o ${BINARY} ${LDFLAGS} ./cmd/chronograf/main.go
+	go build -o ${BINARY} ${GOLDFLAGS} ./cmd/chronograf/main.go
 
 define CHRONOGIRAFFE
              ._ o o
@@ -37,7 +37,7 @@
 	@echo "$$CHRONOGIRAFFE"
 
 docker-${BINARY}: $(SOURCES)
-	CGO_ENABLED=0 GOOS=linux go build -installsuffix cgo -o ${BINARY} ${LDFLAGS} \
+	CGO_ENABLED=0 GOOS=linux go build -installsuffix cgo -o ${BINARY} ${GOLDFLAGS} \
 		./cmd/chronograf/main.go
 
 docker: dep assets docker-${BINARY}
