Index: rmlint-2.10.1/SConstruct
===================================================================
--- rmlint-2.10.1.orig/SConstruct
+++ rmlint-2.10.1/SConstruct
@@ -718,25 +718,19 @@ O_DEBUG   = 'g' # The optimisation level
 O_RELEASE = '2' # The optimisation level for a release build
 
 # build modes
+O_value = None
 if ARGUMENTS.get('DEBUG') == "1":
     print("Compiling in debug mode")
     conf.env.Append(CCFLAGS=['-DRM_DEBUG', '-fno-inline'])
     O_value = ARGUMENTS.get('O', O_DEBUG)
 else:
     conf.env.Append(CCFLAGS=['-DG_DISABLE_ASSERT', '-DNDEBUG'])
-    conf.env.Append(LINKFLAGS=['-s'])
-    O_value = ARGUMENTS.get('O', O_RELEASE)
 
 if O_value == 'debug':
     O_value = O_DEBUG
 elif O_value == 'release':
     O_value = O_RELEASE
 
-cc_O_option = '-O' + O_value
-
-print("Using compiler optimisation {} (to change, run scons with O=[0|1|2|3|s|fast])".format(cc_O_option))
-conf.env.Append(CCFLAGS=[cc_O_option])
-
 if ARGUMENTS.get('SYMBOLS') == '1':
     print("Compiling with debugging symbols")
     conf.env.Append(CCFLAGS='-g3')
