From f1e924628a59e7b658493514297aea0991f76b74 Mon Sep 17 00:00:00 2001
From: Erik Reider <35975961+ErikReider@users.noreply.github.com>
Date: Sun, 14 Sep 2025 18:43:15 +0200
Subject: [PATCH] Fixed compiling with elogind

---
 include/idle.hpp | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/include/idle.hpp b/include/idle.hpp
index fa4abcb..3db065e 100644
--- a/include/idle.hpp
+++ b/include/idle.hpp
@@ -1,9 +1,10 @@
 #pragma once
 
-#if HAVE_SYSTEMD
+#ifdef HAVE_SYSTEMD
 #include <systemd/sd-bus.h>
 #include <systemd/sd-login.h>
-#elif HAVE_ELOGIND
+#endif
+#ifdef HAVE_ELOGIND
 #include <elogind/sd-bus.h>
 #include <elogind/sd-login.h>
 #endif

