Work around https://github.com/thp/urlwatch/issues/677

diff --git a/lib/urlwatch/tests/test_handler.py b/lib/urlwatch/tests/test_handler.py
index 736536a..fac9449 100644
--- a/lib/urlwatch/tests/test_handler.py
+++ b/lib/urlwatch/tests/test_handler.py
@@ -89,6 +89,7 @@ def test_pep8_conformance():
 
 class ConfigForTest(CommandConfig):
     def __init__(self, config, urls, cache, hooks, verbose):
+        sys.argv = sys.argv[:1]
         (prefix, bindir) = os.path.split(os.path.dirname(os.path.abspath(sys.argv[0])))
         super().__init__('urlwatch', os.path.dirname(__file__), bindir, prefix, config, urls, hooks, cache, verbose)
 
