--- a/src/procattr.c	2025-07-16 11:55:13.000000000 +0100
+++ b/src/procattr.c	2025-07-29 23:01:20.086660475 +0100
@@ -27,18 +27,6 @@
 static int destructor_key_initialized = 0;
 static __thread char destructor_initialized;
 
-/* Bionic and glibc >= 2.30 declare gettid() system call wrapper in unistd.h and
- * has a definition for it */
-#ifdef __BIONIC__
-  #define HAVE_GETTID 1
-#elif !defined(__GLIBC_PREREQ)
-  #define HAVE_GETTID 0
-#elif !__GLIBC_PREREQ(2,30)
-  #define HAVE_GETTID 0
-#else
-  #define HAVE_GETTID 1
-#endif
-
 static pid_t selinux_gettid(void)
 {
 #if HAVE_GETTID
