# Contributed by Chocimier@github
diff --git src/foreign/zstr/strict_fstream.hpp src/foreign/zstr/strict_fstream.hpp
index c0fae8f..313f71a 100644
--- src/foreign/zstr/strict_fstream.hpp
+++ src/foreign/zstr/strict_fstream.hpp
@@ -33,7 +33,7 @@ static std::string strerror()
     {
         buff = "Unknown error";
     }
-#elif __APPLE__ || ((_POSIX_C_SOURCE >= 200112L || _XOPEN_SOURCE >= 600) && ! _GNU_SOURCE)
+#elif __APPLE__ || ((_POSIX_C_SOURCE >= 200112L || _XOPEN_SOURCE >= 600) && ! _GNU_SOURCE) || !defined(__GLIBC__)
 // XSI-compliant strerror_r()
     if (strerror_r(errno, &buff[0], buff.size()) != 0)
     {
