--- lib/export/dns/Makefile.in.orig	2011-05-17 01:47:16.000000000 +0200
+++ lib/export/dns/Makefile.in	2015-03-28 19:34:39.143820772 +0100
@@ -166,7 +166,8 @@ code.h:	gen
 	./gen -s ${srcdir} > code.h
 
 gen: ${srcdir}/gen.c
-	${CC} ${ALL_CFLAGS} ${LDFLAGS} -o $@ ${srcdir}/gen.c ${LIBS}
+	${BUILD_CC} ${BUILD_CFLAGS} -I${top_srcdir}/lib/isc/include \
+	${BUILD_CPPFLAGS} ${BUILD_LDFLAGS} -o $@ ${srcdir}/gen.c ${BUILD_LIBS}
 
 #We don't need rbtdb64 for this library
 #rbtdb64.@O@: rbtdb.c
