--- a/config.h	2021-03-16 12:22:48.315849306 -0500
+++ b/config.h	2021-03-16 12:23:00.273828304 -0500
@@ -15,7 +15,7 @@
 
 /* Phone home the name of images downloaded for image popularity ranking */
 #define TELEMETRY_URL                     "https://rpi-imager-stats.raspberrypi.org/downloads"
-#define TELEMETRY_ENABLED_DEFAULT         true
+#define TELEMETRY_ENABLED_DEFAULT         false
 
 /* Hash algorithm for verifying (uncompressed image) checksum */
 #define OSLIST_HASH_ALGORITHM             QCryptographicHash::Sha256
