--- Source/WTF/wtf/DisallowCType.h.orig	2014-04-07 06:37:46.851957952 +0000
+++ Source/WTF/wtf/DisallowCType.h	2014-04-07 06:39:30.653205023 +0000
@@ -41,7 +41,7 @@
 // or <glib/gi18n-lib.h>, which in turn include <xlocale/_ctype.h> which uses
 // isacii(). 
 #include <wtf/Platform.h>
-#if !(OS(DARWIN) && PLATFORM(GTK)) && !PLATFORM(EFL) && !defined(_LIBCPP_VERSION)
+#if !(OS(DARWIN) && PLATFORM(GTK)) && !PLATFORM(EFL) && !defined(_LIBCPP_VERSION) && defined(__GLIBC__)
 
 #include <ctype.h>
 
