--- src/nrpe.c	2017-09-01 09:59:54.000000000 -0400
+++ src/nrpe.c.new	2018-10-06 20:05:28.068549927 -0400
@@ -147,7 +147,7 @@
 	ClntCerts client_certs;
 	SslLogging log_opts;
 } sslprm = {
-#if OPENSSL_VERSION_NUMBER >= 0x10100000
+#if OPENSSL_VERSION_NUMBER >= 0x10100000 && !defined(LIBRESSL_VERSION_NUMBER)
 NULL, NULL, NULL, "ALL:!MD5:@STRENGTH:@SECLEVEL=0", TLSv1_plus, TRUE, 0, SSL_NoLogging};
 #else
 NULL, NULL, NULL, "ALL:!MD5:@STRENGTH", TLSv1_plus, TRUE, 0, SSL_NoLogging};
@@ -344,7 +344,7 @@
 		exit(STATE_CRITICAL);
 	}
 
-#if OPENSSL_VERSION_NUMBER >= 0x10100000
+#if OPENSSL_VERSION_NUMBER >= 0x10100000 && !defined(LIBRESSL_VERSION_NUMBER)
 
 	SSL_CTX_set_max_proto_version(ctx, 0);
 
@@ -445,7 +445,7 @@
 	} else {
 		/* use anonymous DH ciphers */
 		if (sslprm.allowDH == 2) {
-#if OPENSSL_VERSION_NUMBER >= 0x10100000
+#if OPENSSL_VERSION_NUMBER >= 0x10100000 && !defined(LIBRESSL_VERSION_NUMBER)
 			strncpy(sslprm.cipher_list, "ADH@SECLEVEL=0", MAX_FILENAME_LENGTH - 1);
 #else
 			strncpy(sslprm.cipher_list, "ADH", MAX_FILENAME_LENGTH - 1);
