--- include/curses.h.in	2015-03-17 14:53:23.966261378 +0100
+++ include/curses.h.in	2015-03-17 14:57:43.902100942 +0100
@@ -149,14 +149,10 @@
  */
 #include <stdio.h>
 
-/*
- * With XPG4, you must define _XOPEN_SOURCE_EXTENDED, it is redundant (or
- * conflicting) when _XOPEN_SOURCE is 500 or greater.
- */
+/* Enable wide char without consideration of _XOPEN_SOURCE.
+ * We only support ncursesw anyway.  */
 #undef NCURSES_WIDECHAR
-#if defined(_XOPEN_SOURCE_EXTENDED) || defined(_XPG5)
 #define NCURSES_WIDECHAR
-#endif
 
 #include <stdarg.h>	/* we need va_list */
 #ifdef NCURSES_WIDECHAR
