diff -ruN ncurses-6.3_orig/misc/terminfo.src ncurses-6.3/misc/terminfo.src
--- ncurses-6.3_orig/misc/terminfo.src	2022-05-17 17:22:53.886017351 +0200
+++ ncurses-6.3/misc/terminfo.src	2022-05-17 17:23:31.941720306 +0200
@@ -7627,60 +7627,6 @@
 
 ######## WAYLAND CLIENTS
 
-#### Foot
-# https://codeberg.org/dnkl/foot/
-# Version 1.8.2
-#
-# This identifies as a VT220 with 4=sixel and 22=color, however:
-# tack:
-#	bell does not work
-#	status-line does not work because foot does not set the window title
-#	sends escape when meta key is used, whether or not smm/rmm enabled
-# vttest:
-#	wrap-test fails
-#	no application-mode for numeric keypad (unless private mode 1035 is set)
-#	no NRCS
-#	no VT52
-#	no SRM
-#	protected areas do not work
-#	SU/SD work, SL/SR do not
-#	DECRPM responds, but not the corresponding ANSI reports.
-#	otherwise few reports, except cursor-position and mouse and some dtterm
-#	VT520 cursor-movement works, except for left/right margins
-#	supports xterm/DECSCUSR, though default case in vttest does not blink
-#		Send: <27> [ 0 <32> q 
-#		Text: The cursor should be a blinking rectangle
-#	partial support for xterm mouse any-event mode and button-event mode:
-#	+ does not report focus-in/focus-out
-#	+ does not report buttons 6/7
-#	alternate-screen works
-foot|foot terminal emulator,
-	oc=\E]104\E\\, use=xterm+256color2, use=foot+base,
-
-foot-direct|foot with direct color indexing,
-	use=xterm+direct, use=foot+base,
-
-foot+base|foot base fragment,
-	am, bce, bw, hs, mir, msgr, npc, xenl, AX, XT,
-	cols#80, it#8, lines#24,
-	acsc=``aaffggiijjkkllmmnnooppqqrrssttuuvvwwxxyyzz{{||}}~~,
-	bel=^G, blink=\E[5m, bold=\E[1m, clear=\E[H\E[2J, cr=\r,
-	cub1=^H, cud1=\n, dim=\E[2m, ech=\E[%p1%dX, ed=\E[J, el=\E[K,
-	el1=\E[1K, flash=\E]555\E\\, ind=\n, invis=\E[8m,
-	is2=\E[!p\E[?3;4l\E[4l\E>, kbs=^?, kcbt=\E[Z,
-	oc=\E]104\E\\, op=\E[39;49m, rev=\E[7m, ri=\EM, rmacs=\E(B,
-	rmam=\E[?7l, rmkx=\E[?1l\E>, rmso=\E[27m, rmul=\E[24m,
-	rs1=\Ec, rs2=\E[!p\E[?3;4l\E[4l\E>,
-	sgr=%?%p9%t\E(0%e\E(B%;\E[0%?%p6%t;1%;%?%p5%t;2%;%?%p2%t;4%;
-	    %?%p1%p3%|%t;7%;%?%p4%t;5%;%?%p7%t;8%;m,
-	sgr0=\E(B\E[m, smacs=\E(0, smam=\E[?7h, smkx=\E[?1h\E=,
-	smso=\E[7m, smul=\E[4m, E3=\E[3J, use=att610+cvis,
-	use=ansi+csr, use=ansi+cup, use=ansi+enq, use=ansi+local,
-	use=ansi+idc, use=ansi+idl, use=ansi+rca2, use=ansi+rep,
-	use=ansi+tabs, use=ecma+index, use=ecma+italics,
-	use=ecma+strikeout, use=xterm+alt+title,
-	use=xterm+pcfkeys, use=xterm+sm+1006, use=xterm+tmux2,
-
 ######## WEB CLIENTS
 
 #### DomTerm
