--- a/GNUmakefile
+++ b/GNUmakefile
@@ -278,42 +278,42 @@ gtkd-$(MAJOR).pc:
 	echo Name: GtkD > $@
 	echo Description: A D binding and OO wrapper for GTK+. >> $@
 	echo Version: $(GTKD_VERSION) >> $@
-	echo Libs: $(LINKERFLAG)-L$(prefix)/$(libdir)/ $(LINKERFLAG)-lgtkd-$(MAJOR) $(LINKERFLAG)-ldl >> $@
+	echo Libs: -L$(prefix)/$(libdir) -lgtkd-$(MAJOR) -ldl >> $@
 	echo Cflags: -I$(prefix)/include/d/gtkd-$(MAJOR)/ >> $@
 
 gtkdgl-$(MAJOR).pc:
 	echo Name: GtkDGL > $@
 	echo Description: OpenGL capabilities for GtkD. >> $@
 	echo Version: $(GTKD_VERSION) >> $@
-	echo Libs: $(LINKERFLAG)-lgtkdgl-$(MAJOR) >> $@
+	echo Libs: -lgtkdgl-$(MAJOR) >> $@
 	echo Requires: gtkd-$(MAJOR) >> $@
 
 gtkdsv-$(MAJOR).pc:
 	echo Name: GtkD SourceView > $@
 	echo Description: A D binding and OO wrapper for GtkSourceView. >> $@
 	echo Version: $(GTKD_VERSION) >> $@
-	echo Libs: $(LINKERFLAG)-lgtkdsv-$(MAJOR) >> $@
+	echo Libs: -lgtkdsv-$(MAJOR) >> $@
 	echo Requires: gtkd-$(MAJOR) >> $@
 
 gstreamerd-$(MAJOR).pc:
 	echo Name: GstreamerD > $@
 	echo Description: A D binding and OO wrapper for Gstreamer. >> $@
 	echo Version: $(GTKD_VERSION) >> $@
-	echo Libs: $(LINKERFLAG)-lgstreamerd-$(MAJOR) >> $@
+	echo Libs: -lgstreamerd-$(MAJOR) >> $@
 	echo Requires: gtkd-$(MAJOR) >> $@
 
 vted-$(MAJOR).pc:
 	echo Name: VteD > $@
 	echo Description: A D binding and OO wrapper for Vte. >> $@
 	echo Version: $(GTKD_VERSION) >> $@
-	echo Libs: $(LINKERFLAG)-lvted-$(MAJOR) >> $@
+	echo Libs: -lvted-$(MAJOR) >> $@
 	echo Requires: gtkd-$(MAJOR) >> $@
 
 peasd-$(MAJOR).pc:
 	echo Name: PeasD > $@
 	echo Description: A D binding and OO wrapper for Peas. >> $@
 	echo Version: $(GTKD_VERSION) >> $@
-	echo Libs: $(LINKERFLAG)-lpeasd-$(MAJOR) >> $@
+	echo Libs: -lpeasd-$(MAJOR) >> $@
 	echo Requires: gtkd-$(MAJOR) >> $@
 
 #######################################################################
