--- configure.ac
+++ configure.ac
@@ -279,7 +279,7 @@
 dnl Check for GDK Quartz and MacOSX integration package 
 dnl ================================================================
 _gdk_tgt=`$PKG_CONFIG --variable=targets gdk-3.0`
-AM_CONDITIONAL([GDK_TARGET_QUARTZ], [test x$_gdk_tgt = xquartz])
+AM_CONDITIONAL([GDK_TARGET_QUARTZ], [test "x$_gdk_tgt" = xquartz])
 if test "x$_gdk_tgt" = xquartz; then 
    PKG_CHECK_MODULES(GTK_MAC, gtk-mac-integration-gtk3)
 
