From 61304b2e8bac8ded76643cb7c3e781f73881dd2b Mon Sep 17 00:00:00 2001
From: Mike Gilbert <floppym@gentoo.org>
Date: Thu, 20 Jan 2022 14:18:09 -0500
Subject: [PATCH 1/2] meson: i18n.merge_file() does not take a positional
 argument

---
 data/meson.build | 2 --
 1 file changed, 2 deletions(-)

diff --git a/data/meson.build b/data/meson.build
index 9f244566..bfbc514c 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -2,7 +2,6 @@ desktop_conf = configuration_data()
 desktop_conf.set('VERSION', glade_version)
 
 i18n.merge_file(
-  'desktop',
   type: 'desktop',
   input: configure_file(
     input: 'org.gnome.Glade.desktop.in.in',
@@ -18,7 +17,6 @@ i18n.merge_file(
 appdata = glade_name + '.appdata.xml'
 
 i18n.merge_file(
-  'appdata',
   input: 'org.gnome.Glade.appdata.xml.in',
   output: '@BASENAME@',
   po_dir: po_dir,
-- 
GitLab


From 04ba6f969f716fbfe3c7feb7e4bab8678cc1e9eb Mon Sep 17 00:00:00 2001
From: Mike Gilbert <floppym@gentoo.org>
Date: Thu, 20 Jan 2022 14:19:03 -0500
Subject: [PATCH 2/2] Remove duplicate entry for 'da' from help/LINGUAS

---
 help/LINGUAS | 1 -
 1 file changed, 1 deletion(-)

diff --git a/help/LINGUAS b/help/LINGUAS
index fc3cd4c3..d67e42c8 100644
--- a/help/LINGUAS
+++ b/help/LINGUAS
@@ -1,6 +1,5 @@
 # please keep this list sorted alphabetically
 #
-da
 bg
 ca
 cs
-- 
GitLab

