Index: CMakeLists.txt
===================================================================
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -209,6 +209,8 @@ elseif(CMAKE_BUILD_TYPE STREQUAL "RelWithDebInfo")
   endif ()
   set(WL_DEBUG_FLAGS "-DNDEBUG -DNOPARACHUTE")
   option(OPTION_ASAN "Build with AddressSanitizer" ON)
+elseif(CMAKE_BUILD_TYPE STREQUAL "None")
+  message(STATUS "Not setting any default flags.")
 else()
   message(FATAL_ERROR "Unknown CMAKE_BUILD_TYPE: ${CMAKE_BUILD_TYPE}")
 endif()
@@ -415,7 +417,7 @@ endif (OPTION_BUILD_TESTS)
 install (
   FILES ${CMAKE_CURRENT_BINARY_DIR}/VERSION
   DESTINATION ${WL_INSTALL_BASEDIR}
-  CONFIGURATIONS Debug;Release
+  CONFIGURATIONS Debug;Release;None
   COMPONENT CoreVersionFile
 )
 
@@ -440,7 +442,7 @@ install(
     data/txts
     data/world
   DESTINATION ${WL_INSTALL_DATADIR}
-  CONFIGURATIONS Debug;Release
+  CONFIGURATIONS Debug;Release;None
   COMPONENT CoreDataFiles
 )
 
@@ -448,7 +450,7 @@ install(
   DIRECTORY
     data/maps
   DESTINATION ${WL_INSTALL_DATADIR}
-  CONFIGURATIONS Debug;Release
+  CONFIGURATIONS Debug;Release;None
   COMPONENT MapFiles
 )
 
@@ -457,7 +459,7 @@ install(
     data/music
     data/sound
   DESTINATION ${WL_INSTALL_DATADIR}
-  CONFIGURATIONS Debug;Release
+  CONFIGURATIONS Debug;Release;None
   COMPONENT MusicFiles
 )
 
@@ -467,7 +469,7 @@ install(
     CREDITS
     ChangeLog
   DESTINATION ${WL_INSTALL_BASEDIR}
-  CONFIGURATIONS Debug;Release
+  CONFIGURATIONS Debug;Release;None
   COMPONENT CoreLicenseFiles
 )
 
@@ -486,7 +488,7 @@ install(
   DIRECTORY
     ${CMAKE_CURRENT_BINARY_DIR}/locale/
   DESTINATION ${WL_INSTALL_DATADIR}/locale
-  CONFIGURATIONS Debug;Release
+  CONFIGURATIONS Debug;Release;None
   COMPONENT CoreLanguageFiles
 )
 
