--- src/i18n.cc	2014-02-22 10:06:51.000000000 +0100
+++ src/i18n.cc	2015-06-01 10:44:01.035598493 +0200
@@ -192,6 +192,7 @@
 #endif
 
 #ifdef __linux__
+#ifdef __GLIBC__
 	char * res = nullptr;
 	char const * encoding[] = {"", ".utf-8", "@euro", ".UTF-8"};
 	std::size_t found = alt_str.find(',', 0);
@@ -236,6 +237,7 @@
 	/* Finally make changes known.  */
 	++_nl_msg_cat_cntr;
 #endif
+#endif
 
 	SETLOCALE(LC_ALL, ""); //  call to libintl
 
