--- CMakeLists.txt	2020-07-12 21:33:44.000000000 +0200
+++ CMakeLists.txt	2020-07-19 08:28:28.360260489 +0200
@@ -312,19 +312,14 @@
   endif (CMAKE_SIZEOF_VOID_P EQUAL 4)
 endif (WIN32)
 
-# on BSD this must be explicitly linked
-if (CMAKE_SYSTEM_NAME MATCHES "FreeBSD" OR CMAKE_SYSTEM_NAME MATCHES "OpenBSD")
-  # Not needed on Debian GNU/kFreeBSD..
-  if (NOT CMAKE_SYSTEM_NAME MATCHES "kFreeBSD")
-    find_library(EXECINFO_LIBRARY NAMES execinfo)
-  endif (NOT CMAKE_SYSTEM_NAME MATCHES "kFreeBSD")
+# Find libexecinfo for musl
+find_library(EXECINFO_LIBRARY NAMES execinfo)
 
-  # OpenBSD needs the X11 include directory in order to find GL/glu.h
-  if (CMAKE_SYSTEM_NAME MATCHES "OpenBSD")
-    find_package(X11 REQUIRED)
-    include_directories(${X11_INCLUDE_DIR})
-  endif ()
-endif (CMAKE_SYSTEM_NAME MATCHES "FreeBSD" OR CMAKE_SYSTEM_NAME MATCHES "OpenBSD")
+# OpenBSD needs the X11 include directory in order to find GL/glu.h
+if (CMAKE_SYSTEM_NAME MATCHES "OpenBSD")
+  find_package(X11 REQUIRED)
+  include_directories(${X11_INCLUDE_DIR})
+endif ()
 
 if (NOT DEFINED WL_VERSION)
   include (${CMAKE_SOURCE_DIR}/cmake/BzrRevision.cmake)
--- src/CMakeLists.txt	2020-07-12 21:33:44.000000000 +0200
+++ src/CMakeLists.txt	2020-07-19 08:28:57.894258728 +0200
@@ -140,9 +140,9 @@
     logic_filesystem_constants
 )
 
-if (CMAKE_SYSTEM_NAME MATCHES "FreeBSD" OR CMAKE_SYSTEM_NAME MATCHES "OpenBSD")
+if (EXECINFO_LIBRARY)
   target_link_libraries(widelands_ball_of_mud ${EXECINFO_LIBRARY})
-endif (CMAKE_SYSTEM_NAME MATCHES "FreeBSD" OR CMAKE_SYSTEM_NAME MATCHES "OpenBSD")
+endif (EXECINFO_LIBRARY)
 
 if (WIN32)
   target_link_libraries(widelands_ball_of_mud wsock32)
