--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -329,7 +329,7 @@
 endif()
 
 if(NOT MSVC)
-  if((CMAKE_BUILD_TYPE STREQUAL "Release") OR (CMAKE_BUILD_TYPE STREQUAL "RelWithDebInfo"))
+  if((CMAKE_BUILD_TYPE STREQUAL "Release") OR (CMAKE_BUILD_TYPE STREQUAL "None") OR (CMAKE_BUILD_TYPE STREQUAL "RelWithDebInfo"))
     message(STATUS "Compiler warnings will be ignored.")
   elseif(("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU") AND (
          ((NOT (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 12.0)) AND (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 12.2))
