Index: widelands-1.2.1/src/base/macros.h
===================================================================
--- a/src/base/macros.h
+++ b/src/base/macros.h
@@ -137,13 +137,9 @@
 #else
 #define PRIuS PRIu32
 #endif
-#else
-#if SIZE_MAX == UINT64_MAX
-#define PRIuS "lu"
-#else
-#define PRIuS "u"
-#endif
-#endif
+#else // !_WIN32
+#define PRIuS "zu"
+#endif
 #endif
 
 #endif  // end of include guard: WL_BASE_MACROS_H
