Index: widelands-1.1/src/base/macros.h
===================================================================
--- widelands-1.1.orig/src/base/macros.h
+++ widelands-1.1/src/base/macros.h
@@ -134,16 +134,8 @@
 #else
 #define PRIuS PRIu32
 #endif
-#else
-#if __WORDSIZE == 64
-#define PRIuS "lu"
-#else
-#if defined(__WORDSIZE32_SIZE_ULONG) && __WORDSIZE32_SIZE_ULONG
-#define PRIuS "lu"
-#else
-#define PRIuS "u"
-#endif
-#endif
+#else // !_WIN32
+#define PRIuS "zu"
 #endif
 
 #endif  // end of include guard: WL_BASE_MACROS_H
