--- setup.py.orig
+++ setup.py
@@ -16,14 +16,14 @@ setup(
         "content, and resources (like CSS and images) into a neat stack of "
         "plain HTML. You run it on your local computer, and it generates a "
         "directory of web files that you can upload to your web server, or "
-        "serve directly."
+        "serve directly.",
     download_url="http://wok.mythmon.com/download",
     classifiers=[
         "Development Status :: 4 - Beta",
         "License :: OSI Approved :: MIT License",
         'Operating System :: POSIX',
         'Programming Language :: Python',
-    ]
+    ],
     requires=['pyyaml', 'jinja2', 'Markdown', 'docutils', 'Pygments'],
     packages=['wok'],
     scripts=['scripts/wok'],
