--- a/pv/util.cpp	2020-11-21 15:56:56.690899190 +0700
+++ b/pv/util.cpp	2020-11-21 16:00:40.981375801 +0700
@@ -137,7 +137,7 @@
 	QString s;
 	QTextStream ts(&s);
 	if (sign && !v.is_zero())
-		ts << forcesign;
+		Qt::forcesign(ts);
 	ts << qSetRealNumberPrecision(precision) << (v * multiplier);
 	ts << ' ' << prefix << unit;
 
@@ -175,7 +175,7 @@
 	QString s;
 	QTextStream ts(&s);
 	if (sign && (v != 0))
-		ts << forcesign;
+		Qt::forcesign(ts);
 	ts.setRealNumberNotation(QTextStream::FixedNotation);
 	ts.setRealNumberPrecision(precision);
 	ts << (v * multiplier) << ' ' << prefix << unit;
