Index: pulseview-0.4.2/pv/data/decode/annotation.cpp
===================================================================
--- pulseview-0.4.2.orig/pv/data/decode/annotation.cpp
+++ pulseview-0.4.2/pv/data/decode/annotation.cpp
@@ -17,9 +17,7 @@
  * along with this program; if not, see <http://www.gnu.org/licenses/>.
  */
 
-extern "C" {
 #include <libsigrokdecode/libsigrokdecode.h>
-}
 
 #include <cassert>
 #include <vector>
Index: pulseview-0.4.2/pv/views/trace/decodetrace.cpp
===================================================================
--- pulseview-0.4.2.orig/pv/views/trace/decodetrace.cpp
+++ pulseview-0.4.2/pv/views/trace/decodetrace.cpp
@@ -17,9 +17,7 @@
  * along with this program; if not, see <http://www.gnu.org/licenses/>.
  */
 
-extern "C" {
 #include <libsigrokdecode/libsigrokdecode.h>
-}
 
 #include <limits>
 #include <mutex>
