diff -Np1 a/BastetBlockChooser.hpp b/BastetBlockChooser.hpp
*** a/BastetBlockChooser.hpp	2015-08-30 17:04:24.000000000 +1000
--- b/BastetBlockChooser.hpp	2022-09-30 12:18:43.100123811 +1000
***************
*** 25,28 ****
  
! #include <boost/tr1/tr1/unordered_set>
! #include <set>
  #include <boost/functional/hash.hpp>
--- 25,27 ----
  
! #include <boost/unordered_set.hpp>
  #include <boost/functional/hash.hpp>
*************** namespace Bastet{
*** 77,79 ****
    private:
!     std::tr1::unordered_set<Vertex> _visited;
      //std::set<Vertex> _visited; ^^ the above is more efficient, we need to do many inserts
--- 76,78 ----
    private:
!     boost::unordered_set<Vertex> _visited;
      //std::set<Vertex> _visited; ^^ the above is more efficient, we need to do many inserts
