--- a/Config.cpp
+++ b/Config.cpp
@@ -187,7 +187,7 @@ namespace Bastet{
     for(int difficulty=0;difficulty<num_difficulties;++difficulty){
       int i=0;
       BOOST_FOREACH(const HighScore &hs, _hs[difficulty]){
-	ofs2<<str(scorer % difficulty % i) << " = \"" << hs.Scorer << "\"\n";
+	ofs2<<str(scorer % difficulty % i) << " = " << hs.Scorer << "\n";
 	ofs2<<str(score % difficulty % i) << " = " << hs.Score <<"\n";
 	i++;
       }
