diff -Np1 a/Config.cpp b/Config.cpp
*** a/Config.cpp	2015-08-30 17:04:24.000000000 +1000
--- b/Config.cpp	2022-10-03 21:45:30.198811697 +1000
*************** namespace Bastet{
*** 189,191 ****
        BOOST_FOREACH(const HighScore &hs, _hs[difficulty]){
! 	ofs2<<str(scorer % difficulty % i) << " = \"" << hs.Scorer << "\"\n";
  	ofs2<<str(score % difficulty % i) << " = " << hs.Score <<"\n";
--- 189,191 ----
        BOOST_FOREACH(const HighScore &hs, _hs[difficulty]){
! 	ofs2<<str(scorer % difficulty % i) << " = " << hs.Scorer << "\n";
  	ofs2<<str(score % difficulty % i) << " = " << hs.Score <<"\n";
