diff --git a/src/java.base/share/classes/sun/security/util/DomainName.java.orig b/src/java.base/share/classes/sun/security/util/DomainName.java
index 4f577f1..116693c 100644
--- a/src/java.base/share/classes/sun/security/util/DomainName.java.orig
+++ b/src/java.base/share/classes/sun/security/util/DomainName.java
@@ -204,8 +204,11 @@ private static Rules createRules(String tld) {
 
         private static InputStream getPubSuffixStream() {
             InputStream is = null;
-            File f = new File(System.getProperty("java.home"),
-                "lib/security/public_suffix_list.dat");
+            File f = new File("/usr/share/publicsuffix/public_suffix_list.dat");
+            if (!f.exists()) {
+                f = new File(System.getProperty("java.home"),
+                    "lib/security/public_suffix_list.dat");
+            }
             try {
                 is = new FileInputStream(f);
             } catch (FileNotFoundException e) { }
