--- a/jax/jax.c	2004-06-10 03:41:50.000000000 +0200
+++ b/jax/jax.c	2020-09-20 17:22:42.245080921 +0200
@@ -1,4 +1,5 @@
 #include <stdio.h>
+#include <stdint.h>
 #include "jax.h"
 
 /*****************************************************************************/
@@ -49,7 +49,7 @@
 				(jax->options[j].name2 && !strcmp(jax->options[j].name2, argv[i])))
 			{
 				c++;
-				if((int)(*jax->options[j].str)>1)
+				if((intptr_t)(*jax->options[j].str)>1)
 				{
 					if (*argc>i+1)
 					{
@@ -65,7 +65,7 @@
 				}
 				else
 				{
-					*jax->options[j].str=(char*)(((int)(*jax->options[j].str+1))%2);
+					*jax->options[j].str=(char*)(((intptr_t)(*jax->options[j].str+1))%2);
 					JAXshiftopts(i, 1, argc, argv);
 					j=-1;
 				}
