--- a/configure.in	2004-08-29 12:29:09.000000000 +0200
+++ b/configure.in	2020-09-20 17:17:33.079064972 +0200
@@ -206,66 +206,10 @@
 AC_SUBST(extras)
 AC_SUBST(extras_dist)
 
-
-dnl Check for gpm-imwheel source.
-NO_GPM_DOC=
-GPM_DIR="gpm-1.19.3"
-AC_SUBST(GPM_DIR)
-AC_CHECK_FILE($GPM_DIR/gpm.c,HAVE_GPM_SRC=yes,HAVE_GPM_SRC=no)
-AC_SUBST(HAVE_GPM_SRC)
-AC_MSG_CHECKING(if we build gpm-imwheel)
-gpm_enabled=no
-gpm_doc=no
-gpm_imwheel=$GPM_DIR
-AC_ARG_ENABLE(gpm,
-changequote({, })dnl
-{  --enable-gpm           build gpm-imwheel [build if source present]},
-changequote([, ])dnl
-[case "$enableval" in
-	yes) gpm_imwheel="$GPM_DIR"
-	     gpm_enabled=yes ;;
-	no)  gpm_imwheel= ;;
-	*)   AC_MSG_ERROR([bad value $enableval for --enable-gpm]) ;;
-esac],[gpm_imwheel=])
-AC_ARG_ENABLE(gpm_doc,
-changequote({, })dnl
-{  --enable-gpm-doc       build in gpm/doc  [build if source present]},
-changequote([, ])dnl
-[case "$enableval" in
-	yes) no_gpm_doc=true ;;
-	no)  no_gpm_doc=false ;;
-	*)   AC_MSG_ERROR([bad value $enableval for --enable-gpm-doc]) ;;
-esac],[no_gpm_doc=true])
-if [[ "x$HAVE_GPM_SRC" = "$Good" ]]
-then
-	if [[ "x$gpm_imwheel" = "x$GPM_DIR" ]]
-	then
-		AC_MSG_RESULT(yes)
-		AC_CONFIG_SUBDIRS(gpm-1.19.3)
-		AC_DEFINE_UNQUOTED(HAVE_GPM_SRC,1,[Define only if you are building the imwheel version of gpm. ])
-		AM_CONDITIONAL(NO_GPM_DOC, $no_gpm_doc)
-	else
-		AC_MSG_RESULT(no)
-		AM_CONDITIONAL(NO_GPM_DOC, false)
-	fi
-else
-	gpm_imwheel=
-	if [[ "x$gpm_enabled" = "xyes" ]]
-	then
-		AC_MSG_RESULT(unable)
-		AM_CONDITIONAL(NO_GPM_DOC, false)
-		AC_MSG_WARN([$GPM_DIR building disabled, source not present!])
-		Warnings=`expr $Warnings + 1`
-	else
-		AC_MSG_RESULT(no)
-		AM_CONDITIONAL(NO_GPM_DOC, false)
-	fi
-fi
-gpm_dist="$GPM_DIR"
-AC_SUBST(gpm_imwheel)
-AC_SUBST(gpm_dist)
-
-
+AC_SUBST(gpm_imwheel)
+AC_SUBST(gpm_dist)
+AC_DEFINE_UNQUOTED(HAVE_GPM_SRC,,[Define only if you are building the imwheel version of gpm. ])
+AM_CONDITIONAL(NO_GPM_DOC, true)
 dnl Checks for X.
 AC_PATH_X
 AC_PATH_XTRA
