--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -507,6 +507,7 @@ elseif (SUBSURFACE_TARGET_EXECUTABLE MATCHES "DesktopExecutable")
 	add_dependencies(subsurface_interface subsurface_generated_ui)
 	add_dependencies(subsurface_profile subsurface_generated_ui)
 	add_dependencies(subsurface_models_desktop subsurface_generated_ui)
+	add_dependencies(subsurface_corelib subsurface_generated_ui)
 	add_dependencies(subsurface_generated_ui version)
 elseif (SUBSURFACE_TARGET_EXECUTABLE MATCHES "DownloaderExecutable")
 	set(DOWNLOADER_APP
