Upstream: No
Reason: Avoid certifi dependency

diff --git contrib/requirements/requirements.txt contrib/requirements/requirements.txt
index 4ca24227..ac60e57c 100644
--- contrib/requirements/requirements.txt
+++ contrib/requirements/requirements.txt
@@ -8,4 +8,3 @@ qdarkstyle<3.0
 aiorpcx>=0.17,<0.18
 aiohttp>=3.3.0
 aiohttp_socks
-certifi
diff --git electrum/interface.py electrum/interface.py
index 8a019afb..7a5a7e76 100644
--- electrum/interface.py
+++ electrum/interface.py
@@ -38,7 +38,6 @@ import aiorpcx
 from aiorpcx import RPCSession, Notification, NetAddress
 from aiorpcx.curio import timeout_after, TaskTimeout
 from aiorpcx.jsonrpc import JSONRPC
-import certifi
 
 from .util import ignore_exceptions, log_exceptions, bfh, SilentTaskGroup
 from . import util
@@ -55,7 +54,7 @@ if TYPE_CHECKING:
     from .network import Network
 
 
-ca_path = certifi.where()
+ca_path = '/etc/ssl/certs/ca-certificates.crt'
 
 BUCKET_NAME_OF_ONION_SERVERS = 'onion'
 
diff --git electrum/paymentrequest.py electrum/paymentrequest.py
index 881a3eea..6221c4f4 100644
--- electrum/paymentrequest.py
+++ electrum/paymentrequest.py
@@ -28,7 +28,6 @@ import time
 import traceback
 import json
 
-import certifi
 import urllib.parse
 import aiohttp
 
@@ -53,7 +52,7 @@ _logger = get_logger(__name__)
 REQUEST_HEADERS = {'Accept': 'application/bitcoin-paymentrequest', 'User-Agent': 'Electrum'}
 ACK_HEADERS = {'Content-Type':'application/bitcoin-payment','Accept':'application/bitcoin-paymentack','User-Agent':'Electrum'}
 
-ca_path = certifi.where()
+ca_path = '/etc/ssl/certs/ca-certificates.crt'
 ca_list = None
 ca_keyID = None
 
diff --git electrum/util.py electrum/util.py
index 41d99c9f..0fa0ef2f 100644
--- electrum/util.py
+++ electrum/util.py
@@ -44,7 +44,6 @@ import ssl
 import aiohttp
 from aiohttp_socks import SocksConnector, SocksVer
 from aiorpcx import TaskGroup
-import certifi
 
 from .i18n import _
 from .logging import get_logger, Logger
@@ -62,7 +61,7 @@ def inv_dict(d):
     return {v: k for k, v in d.items()}
 
 
-ca_path = certifi.where()
+ca_path = '/etc/ssl/certs/ca-certificates.crt'
 
 
 base_units = {'BTC':8, 'mBTC':5, 'bits':2, 'sat':0}
diff --git electrum/x509.py electrum/x509.py
index c38edc77..1c406544 100644
--- electrum/x509.py
+++ electrum/x509.py
@@ -343,7 +343,5 @@ def load_certificates(ca_path):
 
 
 if __name__ == "__main__":
-    import certifi
-
-    ca_path = certifi.where()
+    ca_path = '/etc/ssl/certs/ca-certificates.crt'
     ca_list, ca_keyID = load_certificates(ca_path)
diff --git run_electrum run_electrum
index 654cf15f..94cca1a0 100755
--- run_electrum
+++ run_electrum
@@ -55,7 +55,6 @@ def check_imports():
         import dns
         import pyaes
         import ecdsa
-        import certifi
         import qrcode
         import google.protobuf
         import jsonrpclib
@@ -69,7 +68,7 @@ def check_imports():
     from google.protobuf import descriptor_pb2
     from jsonrpclib import SimpleJSONRPCServer
     # make sure that certificates are here
-    assert os.path.exists(certifi.where())
+    assert os.path.exists('/etc/ssl/certs/ca-certificates.crt')
 
 
 if not is_android:
