diff --git a/tool/yubico-piv-tool.c b/tool/yubico-piv-tool.c
index d7e11d5..7cd15e3 100644
--- a/tool/yubico-piv-tool.c
+++ b/tool/yubico-piv-tool.c
@@ -751,7 +751,7 @@ static bool request_certificate(ykpiv_state *state, enum enum_key_format key_for
       fprintf(stderr, "Failed signing request.\n");
       goto request_out;
     }
-    M_ASN1_BIT_STRING_set(req->signature, signature, sig_len);
+    ASN1_STRING_set(req->signature, signature, sig_len);
     /* mark that all bits should be used. */
     req->signature->flags = ASN1_STRING_FLAG_BITS_LEFT;
   }
@@ -1007,7 +1007,7 @@ static bool selfsign_certificate(ykpiv_state *state, enum enum_key_format key_fo
       fprintf(stderr, "Failed signing certificate.\n");
       goto selfsign_out;
     }
-    M_ASN1_BIT_STRING_set(x509->signature, signature, sig_len);
+    ASN1_STRING_set(x509->signature, signature, sig_len);
     /* setting flags to ASN1_STRING_FLAG_BITS_LEFT here marks that no bits
      * should be subtracted from the bit string, thus making sure that the
      * certificate can be validated. */
