--- src/kptview.h
+++ src/kptview.h
@@ -30,6 +30,7 @@
 
 #include <QDockWidget>
 #include <QMap>
+#include <QActionGroup>
 
 #include <kconfigdialog.h>
 #include <KDirWatch>
--- src/kptviewlist.cpp
+++ src/kptviewlist.cpp
@@ -26,6 +26,7 @@
 #include <QBrush>
 #include <QContextMenuEvent>
 #include <QMenu>
+#include <QHeaderView>
 
 #include <kmessagebox.h>
 #include <kcombobox.h>
--- src/libs/ui/kptaccountseditor.cpp
+++ src/libs/ui/kptaccountseditor.cpp
@@ -40,6 +40,7 @@
 #include <QContextMenuEvent>
 #include <QMenu>
 #include <QAction>
+#include <QHeaderView>
 
 #include <KLocalizedString>
 #include <kactioncollection.h>
--- src/libs/ui/kptdocumentspanel.h
+++ src/libs/ui/kptdocumentspanel.h
@@ -27,6 +27,7 @@
 #include "kptdocuments.h"
 
 #include <QWidget>
+#include <QModelIndex>
 #include <kundo2qstack.h>
 
 class QModelIndex;
--- src/libs/ui/kptitemviewsettup.cpp
+++ src/libs/ui/kptitemviewsettup.cpp
@@ -26,7 +26,7 @@
 #include "KoPageLayoutWidget.h"
 
 #include <QPushButton>
-
+#include <QHeaderView>
 
 namespace KPlato
 {
--- src/libs/ui/kpttaskeditor.cpp
+++ src/libs/ui/kpttaskeditor.cpp
@@ -46,6 +46,7 @@
 #include <QDragMoveEvent>
 #include <QAction>
 #include <QMenu>
+#include <QHeaderView>
 
 #include <kactionmenu.h>
 #include <KLocalizedString>
--- src/libs/ui/reportsgenerator/ReportsGeneratorView.cpp
+++ src/libs/ui/reportsgenerator/ReportsGeneratorView.cpp
@@ -44,6 +44,7 @@
 #include <QMap>
 #include <QMessageBox>
 #include <QDebug>
+#include <QHeaderView>
 
 namespace KPlato
 {
--- src/libs/widgets/KoPageLayoutWidget.h
+++ src/libs/widgets/KoPageLayoutWidget.h
@@ -25,6 +25,7 @@
 //#include <KoText.h>
 #include <KoPageLayout.h>
 #include <QWidget>
+#include <QButtonGroup>
 
 class KoUnit;
 
From fe36bf6f0dc1f0c1e927850dd78093cfe3b7aae2 Mon Sep 17 00:00:00 2001
From: Albert Astals Cid <aacid@kde.org>
Date: Sun, 28 Jul 2019 00:07:33 +0200
Subject: [PATCH] Fix compile on CI

src/workpackage/view.h:148:31: error: field m_scheduleActions has incomplete type QMap<QAction*, KPlato::Schedule*>
---
 src/workpackage/view.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/workpackage/view.h b/src/workpackage/view.h
index 6cae32b7..cd2fc4ea 100644
--- src/workpackage/view.h
+++ src/workpackage/view.h
@@ -24,6 +24,7 @@
 
 #include <KoView.h>
 
+#include <QMap>
 #include <QStackedWidget>
 
 class QPrinter;
