--- a/src/plugins/schedulers/tj/taskjuggler/TaskList.cpp
+++ b/src/plugins/schedulers/tj/taskjuggler/TaskList.cpp
@@ -146,17 +146,25 @@ TaskList::compareItemsLevel(CoreAttribut
     case ResponsibleUp:
     {
         QString fn1;
-        t1->responsible->getFullName(fn1);
+        if (t1->responsible) {
+            t1->responsible->getFullName(fn1);
+        }
         QString fn2;
-        t2->responsible->getFullName(fn2);
+        if (t2->responsible) {
+            t2->responsible->getFullName(fn2);
+        }
         return fn1.compare(fn2);
     }
     case ResponsibleDown:
     {
         QString fn1;
-        t1->responsible->getFullName(fn1);
+        if (t1->responsible) {
+            t1->responsible->getFullName(fn1);
+        }
         QString fn2;
-        t2->responsible->getFullName(fn2);
+        if (t2->responsible) {
+            t2->responsible->getFullName(fn2);
+        }
         return -fn1.compare(fn2);
     }
     case CriticalnessUp:
