diff -u clockspeed-0.62.old/clockview.c clockspeed-0.62/clockview.c
--- clockspeed-0.62.old/clockview.c	1998-10-13 12:37:49.000000000 -0500
+++ clockspeed-0.62/clockview.c	2003-05-08 09:59:26.000000000 -0500
@@ -1,5 +1,6 @@
 #include <sys/types.h>
 #include <sys/time.h>
+#include <time.h>
 #include "substdio.h"
 #include "readwrite.h"
 #include "strerr.h"
diff -u clockspeed-0.62.old/error.h clockspeed-0.62/error.h
--- clockspeed-0.62.old/error.h	1998-10-13 12:37:49.000000000 -0500
+++ clockspeed-0.62/error.h	2003-05-08 09:07:13.000000000 -0500
@@ -1,7 +1,7 @@
 #ifndef ERROR_H
 #define ERROR_H
 
-extern int errno;
+#include <errno.h>
 
 extern int error_intr;
 extern int error_nomem;
diff -u clockspeed-0.62.old/leapsecs_read.c clockspeed-0.62/leapsecs_read.c
--- clockspeed-0.62.old/leapsecs_read.c	1998-10-13 12:37:49.000000000 -0500
+++ clockspeed-0.62/leapsecs_read.c	2003-05-08 09:07:13.000000000 -0500
@@ -2,7 +2,6 @@
 #include <sys/stat.h>
 #include <fcntl.h>
 #include <errno.h>
-extern int errno;
 #include "tai.h"
 #include "leapsecs.h"
 
