--- src/libdmr/player_engine.cpp	2018-05-04 10:15:32.000000000 +0200
+++ -	2018-05-05 14:43:52.094245514 +0200
@@ -48,6 +48,8 @@
     auto *l = new QVBoxLayout(this);
     l->setContentsMargins(0, 0, 0, 0);
 
+    setlocale(LC_NUMERIC, "C");
+    
     _current = new MpvProxy(this);
     if (_current) {
         connect(_current, &Backend::stateChanged, this, &PlayerEngine::onBackendStateChanged);
