diff --git a/Makefile.am b/Makefile.am
index ce19aac3..22a8c075 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -498,10 +498,9 @@ fapi-config.json: dist/fapi-config.json.in
 		-e 's|[@]userstatedir@|$(userstatedir)|g' \
 		< "$<" > "$@"
 
-sysusers_DATA = dist/sysusers.d/tpm2-tss.conf
 tmpfiles_DATA = tpm2-tss-fapi.conf
 
-EXTRA_DIST += dist/sysusers.d/tpm2-tss.conf dist/tmpfiles.d/tpm2-tss-fapi.conf.in
+EXTRA_DIST += dist/tmpfiles.d/tpm2-tss-fapi.conf.in
 CLEANFILES += tpm2-tss-fapi.conf
 
 # We have to do this ourselves, in order to get absolute paths
@@ -726,13 +725,6 @@ EXTRA_DIST += dist/tpm-udev.rules
 
 install-dirs:
 if HOSTOS_LINUX
-if SYSD_SYSUSERS
-	@echo "systemd-sysusers $(DESTDIR)$(sysconfdir)/sysusers.d/tpm2-tss.conf"
-	@systemd-sysusers $(DESTDIR)$(sysconfdir)/sysusers.d/tpm2-tss.conf || echo "WARNING Failed to create the tss user and group"
-else
-	@echo "call make_tss_user_and_group"
-	@$(call make_tss_user_and_group) || echo "WARNING Failed to create the tss user and group"
-endif
 if SYSD_TMPFILES
 	@echo "systemd-tmpfiles --create $(DESTDIR)$(sysconfdir)/tmpfiles.d/tpm2-tss-fapi.conf"
 	@systemd-tmpfiles --create $(DESTDIR)$(sysconfdir)/tmpfiles.d/tpm2-tss-fapi.conf|| echo "WARNING Failed to create the FAPI directories with the correct permissions"
diff --git a/configure.ac b/configure.ac
index 6482944f..44c0e383 100644
--- a/configure.ac
+++ b/configure.ac
@@ -483,22 +483,9 @@ AS_IF([test "x$enable_integration" = "xyes" && test "x$enable_self_generated_cer
       [AC_MSG_WARN([Running integration tests without EK certificate verification, use --enable-self-generated-certificate for full test coverage])])
 
 # Check for systemd helper tools used by make install
-AC_CHECK_PROG(systemd_sysusers, systemd-sysusers, yes)
-AM_CONDITIONAL(SYSD_SYSUSERS, test "x$systemd_sysusers" = "xyes")
 AC_CHECK_PROG(systemd_tmpfiles, systemd-tmpfiles, yes)
 AM_CONDITIONAL(SYSD_TMPFILES, test "x$systemd_tmpfiles" = "xyes")
 
-# Check all tools used by make install
-AS_IF([test "$HOSTOS" = "Linux"],
-    [ AC_CHECK_PROG(useradd, useradd, yes)
-      AC_CHECK_PROG(groupadd, groupadd, yes)
-      AC_CHECK_PROG(adduser, adduser, yes)
-      AC_CHECK_PROG(addgroup, addgroup, yes)
-      AS_IF([test "x$addgroup" != "xyes" && test "x$groupadd" != "xyes" ],
-         [AC_MSG_ERROR([addgroup or groupadd are needed.])])
-      AS_IF([test "x$adduser" != "xyes" && test "x$useradd" != "xyes" ],
-         [AC_MSG_ERROR([adduser or useradd are needed.])])])
-
 AC_SUBST([PATH])
 
 dnl --------- Doxy Gen -----------------------

