diff -p1ru a/src/menu.cpp b/src/menu.cpp
--- a/src/menu.cpp	2021-03-05 05:44:01.868501000 +0800
+++ b/src/menu.cpp	2024-08-08 12:57:10.803323830 +0800
@@ -346,4 +346,4 @@ private:
             g_menu_item_set_action_and_target_value (item, "indicator.calendar", v);
-            g_menu_item_set_attribute (item, "x-canonical-type",
-                                       "s", "com.canonical.indicator.calendar");
+            g_menu_item_set_attribute (item, "x-ayatana-type",
+                                       "s", "org.ayatana.indicator.calendar");
             if (action_name != nullptr)
diff -p1ru a/tests/test-menus.cpp b/tests/test-menus.cpp
--- a/tests/test-menus.cpp	2021-03-05 05:44:01.868501000 +0800
+++ b/tests/test-menus.cpp	2024-08-08 12:58:25.775818817 +0800
@@ -133,4 +133,4 @@ protected:
         {
-            g_menu_model_get_item_attribute(section, 1, "x-canonical-type", "s", &str);
-            EXPECT_STREQ("com.canonical.indicator.calendar", str);
+            g_menu_model_get_item_attribute(section, 1, "x-ayatana-type", "s", &str);
+            EXPECT_STREQ("org.ayatana.indicator.calendar", str);
             g_clear_pointer(&str, g_free);
