diff -p1ru a/src/menu.cpp b/src/menu.cpp
--- a/src/menu.cpp	2021-03-05 05:44:01.868501000 +0800
+++ b/src/menu.cpp	2024-07-31 09:43:50.165020138 +0800
@@ -295,4 +295,4 @@ private:
         auto header = g_menu_item_new(nullptr, detailed_action.c_str());
-        g_menu_item_set_attribute(header, "x-canonical-type", "s",
-                                  "com.canonical.indicator.root");
+        g_menu_item_set_attribute(header, "x-ayatana-type", "s",
+                                  "org.ayatana.indicator.root");
         g_menu_item_set_attribute(header, "submenu-action", "s",
diff -p1ru a/tests/test-menus.cpp b/tests/test-menus.cpp
--- a/tests/test-menus.cpp	2021-03-05 05:44:01.868501000 +0800
+++ b/tests/test-menus.cpp	2024-07-31 09:44:32.703740129 +0800
@@ -64,4 +64,4 @@ protected:
         gchar* str = nullptr;
-        g_menu_model_get_item_attribute(menu_model, 0, "x-canonical-type", "s", &str);
-        EXPECT_STREQ("com.canonical.indicator.root", str);
+        g_menu_model_get_item_attribute(menu_model, 0, "x-ayatana-type", "s", &str);
+        EXPECT_STREQ("org.ayatana.indicator.root", str);
         g_clear_pointer(&str, g_free);
