--- a/base/allocator/partition_allocator/src/partition_alloc/partition_alloc_config.h
+++ b/base/allocator/partition_allocator/src/partition_alloc/partition_alloc_config.h
@@ -162,11 +162,16 @@
 
 #if defined(ARCH_CPU_ARM64) && defined(__clang__) && \
     !defined(ADDRESS_SANITIZER) &&                   \
-    (BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_ANDROID))
+    (BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_ANDROID)) && \
+    defined(__GLIBC__) && defined(__GLIBC_PREREQ)
+#if __GLIBC_PREREQ(2, 29)
 #define PA_CONFIG_HAS_MEMORY_TAGGING() 1
 #else
 #define PA_CONFIG_HAS_MEMORY_TAGGING() 0
 #endif
+#else
+#define PA_CONFIG_HAS_MEMORY_TAGGING() 0
+#endif
 
 #if PA_CONFIG(HAS_MEMORY_TAGGING)
 static_assert(sizeof(void*) == 8);
