--- net/third_party/quiche/src/quic/core/quic_path_validator.cc.orig	2021-03-02 18:48:06.000000000 +0100
+++ net/third_party/quiche/src/quic/core/quic_path_validator.cc	2021-03-03 13:17:56.624281475 +0100
@@ -68,7 +68,7 @@
 void QuicPathValidator::StartPathValidation(
     std::unique_ptr<QuicPathValidationContext> context,
     std::unique_ptr<ResultDelegate> result_delegate) {
-  DCHECK_NE(nullptr, context);
+  DCHECK(context);
   QUIC_DLOG(INFO) << "Start validating path " << *context
                   << " via writer: " << context->WriterToUse();
   if (path_context_ != nullptr) {
