diff -up chromium-121.0.6167.57/content/browser/interest_group/header_direct_from_seller_signals.cc.me chromium-121.0.6167.57/content/browser/interest_group/header_direct_from_seller_signals.cc
--- chromium-121.0.6167.57/content/browser/interest_group/header_direct_from_seller_signals.cc.me	2024-01-23 15:13:11.469104694 +0100
+++ chromium-121.0.6167.57/content/browser/interest_group/header_direct_from_seller_signals.cc	2024-01-23 15:11:40.117842015 +0100
@@ -46,7 +46,7 @@ size_t GetResultSizeBytes(const HeaderDi
 
 }  // namespace
 
-HeaderDirectFromSellerSignals::Result::Result() = default;
+HeaderDirectFromSellerSignals::Result::Result() {}
 
 HeaderDirectFromSellerSignals::Result::Result(
     absl::optional<std::string> seller_signals,
