From 7a0d8ee35637aaa73b041e6ba9b381ffe18e88b5 Mon Sep 17 00:00:00 2001
From: LN Liberda <lauren@selfisekai.rocks>
Date: Sun, 19 Jul 2026 11:19:09 +0200
Subject: [PATCH] dawn: build with clang 22

---
 src/utils/BUILD.gn | 2 --
 1 file changed, 2 deletions(-)

diff --git a/third_party/dawn/src/utils/BUILD.gn b/third_party/dawn/src/utils/BUILD.gn
index b9f0be693e..ed7ad3b6d3 100644
--- a/third_party/dawn/src/utils/BUILD.gn
+++ b/third_party/dawn/src/utils/BUILD.gn
@@ -218,8 +218,6 @@ config("dawn_warnings") {
       # the analysis pass.
       "-Wno-lifetime-safety",
       "-Wno-lifetime-safety-all",
-      "-Xclang=-fno-lifetime-safety-inference",
-      "-Xclang=-fno-experimental-lifetime-safety-tu-analysis",
     ]
 
     # P3. Checks that could be nice but probably don't help with hardening.
