diff -up chromium-119.0.6045.59/third_party/blink/renderer/core/paint/fragment_data_iterator.h.me chromium-119.0.6045.59/third_party/blink/renderer/core/paint/fragment_data_iterator.h
--- chromium-119.0.6045.59/third_party/blink/renderer/core/paint/fragment_data_iterator.h.me	2023-10-27 20:07:16.421230815 +0200
+++ chromium-119.0.6045.59/third_party/blink/renderer/core/paint/fragment_data_iterator.h	2023-10-27 20:07:36.883600085 +0200
@@ -52,7 +52,7 @@ class FragmentDataIterator
  public:
   explicit FragmentDataIterator(const LayoutObject& object)
       : FragmentDataIteratorBase(&object.FirstFragment()) {}
-  explicit FragmentDataIterator(nullptr_t)
+  explicit FragmentDataIterator(std::nullptr_t)
       : FragmentDataIteratorBase(nullptr) {}
 };
 
@@ -63,7 +63,7 @@ class MutableFragmentDataIterator
   explicit MutableFragmentDataIterator(const LayoutObject& object)
       : FragmentDataIteratorBase(
             &object.GetMutableForPainting().FirstFragment()) {}
-  explicit MutableFragmentDataIterator(nullptr_t)
+  explicit MutableFragmentDataIterator(std::nullptr_t)
       : FragmentDataIteratorBase(nullptr) {}
 };
 
