--- chrome/browser/ui/webui/print_preview_handler.cc.orig	2011-04-23 09:05:25.000000000 +0200
+++ chrome/browser/ui/webui/print_preview_handler.cc	2011-04-23 09:13:07.000000000 +0200
@@ -28,7 +28,7 @@
 #include "printing/metafile_impl.h"
 #include "printing/print_job_constants.h"
 
-#if defined(OS_POSIX) && !defined(OS_CHROMEOS)
+#if defined(CP_PRINT_SYSTEM_AVAILABLE)
 #include <cups/cups.h>
 
 #include "base/file_util.h"
@@ -42,7 +42,7 @@
 const char kDisableColorOption[] = "disableColorOption";
 const char kSetColorAsDefault[] = "setColorAsDefault";
 
-#if defined(OS_POSIX) && !defined(OS_CHROMEOS)
+#if defined(CP_PRINT_SYSTEM_AVAILABLE)
 const char kColorDevice[] = "ColorDevice";
 #endif
 
@@ -131,7 +131,7 @@
       return;
     }
 
-  #if defined(OS_POSIX) && !defined(OS_CHROMEOS)
+#if defined(CP_PRINT_SYSTEM_AVAILABLE)
     FilePath ppd_file_path;
     if (!file_util::CreateTemporaryFile(&ppd_file_path))
       return;
@@ -153,9 +153,9 @@
       ppdClose(ppd);
     }
     file_util::Delete(ppd_file_path, false);
-  #elif defined(OS_WIN) || defined(OS_CHROMEOS)
+#else
     NOTIMPLEMENTED();
-  #endif
+#endif
 
     DictionaryValue settings_info;
     settings_info.SetBoolean(kDisableColorOption, !supports_color);
