--- ./components/autofill/core/common/autofill_l10n_util.cc
+++ ./components/autofill/core/common/autofill_l10n_util.cc
@@ -22,7 +22,7 @@ CaseInsensitiveCompare::CaseInsensitiveCompare(const icu::Locale& locale) {
     // library. This could be due to a device-specific issue (has been seen in
     // the wild on Android devices). In the failure case, |collator_| will be
     // null. See http://crbug.com/558625.
-    icu_54::UnicodeString name;
+    icu::UnicodeString name;
     std::string locale_name;
     locale.getDisplayName(name).toUTF8String(locale_name);
     LOG(ERROR) << "Failed to initialize the ICU Collator for "
