--- chromium-20.0.1132.43/base/message_pump_libevent.cc.gcc47	2012-02-17 14:52:00.527217354 -0500
+++ chromium-20.0.1132.43/base/message_pump_libevent.cc	2012-02-17 14:52:10.172135400 -0500
@@ -6,6 +6,7 @@

 #include <errno.h>
 #include <fcntl.h>
+#include <unistd.h>

 #include "base/auto_reset.h"
 #include "base/compiler_specific.h"
--- chromium-20.0.1132.43/chrome/browser/policy/policy_path_parser_linux.cc.gcc47	2012-02-17 15:26:32.086746278 -0500
+++ chromium-20.0.1132.43/chrome/browser/policy/policy_path_parser_linux.cc	2012-02-17 15:26:53.039544290 -0500
@@ -3,6 +3,7 @@
 // found in the LICENSE file.

 #include <pwd.h>
+#include <unistd.h>

 #include "chrome/browser/policy/policy_path_parser.h"

--- chromium-20.0.1132.43/ipc/ipc_channel.h.gcc47	2012-01-27 03:02:07.000000000 -0500
+++ chromium-20.0.1132.43/ipc/ipc_channel.h	2012-02-17 14:46:31.821010460 -0500
@@ -9,6 +9,7 @@
 #include "base/process.h"
 #include "ipc/ipc_channel_handle.h"
 #include "ipc/ipc_message.h"
+#include <sys/types.h>

 namespace IPC {

--- chromium-20.0.1132.43/ipc/ipc_platform_file.cc.gcc47	2012-01-27 03:02:07.000000000 -0500
+++ chromium-20.0.1132.43/ipc/ipc_platform_file.cc	2012-02-17 14:46:31.824010435 -0500
@@ -3,10 +3,7 @@
 // found in the LICENSE file.

 #include "ipc/ipc_platform_file.h"
-
-#if defined(OS_ANDROID)
 #include <unistd.h>
-#endif

 namespace IPC {

--- chromium-20.0.1132.43/seccompsandbox/library.h.gcc47	2012-01-27 03:03:35.000000000 -0500
+++ chromium-20.0.1132.43/seccompsandbox/library.h	2012-02-17 14:46:31.826010418 -0500
@@ -12,6 +12,7 @@
 #include <string>
 #include <string.h>
 #include <sys/mman.h>
+#include <sys/types.h>

 #include "maps.h"
