From 980ae0fbe6d985efce517c8f6c9aa139b076322b Mon Sep 17 00:00:00 2001
From: Stephan Hartmann <stha09@googlemail.com>
Date: Sun, 28 Feb 2021 12:55:19 +0000
Subject: [PATCH] IWYU: add missing cstdint for uint32_t

---
 .../translate/core/language_detection/quantization_utils.h       | 1 +
 1 file changed, 1 insertion(+)

diff --git a/components/translate/core/language_detection/quantization_utils.h b/components/translate/core/language_detection/quantization_utils.h
index eb8f6d1..699a488 100644
--- components/translate/core/language_detection/quantization_utils.h
+++ components/translate/core/language_detection/quantization_utils.h
@@ -7,6 +7,7 @@
 
 #include <algorithm>
 #include <cmath>
+#include <cstdint>
 
 namespace translate {
 
-- 
2.26.2

