--- chromium-121.0.6167.57/third_party/libc++/src/include/__config.orig	2024-01-16 18:38:00.995930774 +0100
+++ chromium-121.0.6167.57/third_party/libc++/src/include/__config	2024-01-16 18:43:47.056517289 +0100
@@ -32,11 +32,7 @@
 
 // Warn if a compiler version is used that is not supported anymore
 // LLVM RELEASE Update the minimum compiler versions
-#  if defined(_LIBCPP_CLANG_VER)
-#    if _LIBCPP_CLANG_VER < 1600
-#      warning "Libc++ only supports Clang 16 and later"
-#    endif
-#  elif defined(_LIBCPP_APPLE_CLANG_VER)
+#  if defined(_LIBCPP_APPLE_CLANG_VER)
 #    if _LIBCPP_APPLE_CLANG_VER < 1500
 #      warning "Libc++ only supports AppleClang 15 and later"
 #    endif
