--- chrome/common/chrome_content_client.cc	2016-01-20 22:01:20.000000000 +0200
+++ chrome/common/chrome_content_client.cc	2016-01-21 20:02:02.788936626 +0200
@@ -158,7 +158,12 @@ void ComputeBuiltInPlugins(std::vector<c
 #if defined(WIDEVINE_CDM_AVAILABLE) && defined(ENABLE_PEPPER_CDMS) && \
     !defined(WIDEVINE_CDM_IS_COMPONENT)
   static bool skip_widevine_cdm_file_check = false;
-  if (PathService::Get(chrome::FILE_WIDEVINE_CDM_ADAPTER, &path)) {
+  bool widevine_cdm_present =
+      skip_widevine_cdm_file_check ||
+      (PathService::Get(chrome::FILE_WIDEVINE_CDM, &path) &&
+       base::PathExists(path));
+  if (widevine_cdm_present &&
+      PathService::Get(chrome::FILE_WIDEVINE_CDM_ADAPTER, &path)) {
     if (skip_widevine_cdm_file_check || base::PathExists(path)) {
       content::PepperPluginInfo widevine_cdm;
       widevine_cdm.is_out_of_process = true;
--- chrome/common/chrome_paths.cc	2016-01-20 22:01:20.000000000 +0200
+++ chrome/common/chrome_paths.cc	2016-01-21 19:18:51.287978456 +0200
@@ -375,6 +375,11 @@ bool PathProvider(int key, base::FilePat
         return false;
       cur = cur.AppendASCII(kWidevineCdmAdapterFileName);
       break;
+    case chrome::FILE_WIDEVINE_CDM:
+      if (!GetInternalPluginsDirectory(&cur))
+        return false;
+      cur = cur.AppendASCII(kWidevineCdmFileName);
+      break;
 #endif  // defined(WIDEVINE_CDM_AVAILABLE) && defined(ENABLE_PEPPER_CDMS)
     case chrome::FILE_RESOURCES_PACK:
 #if defined(OS_MACOSX) && !defined(OS_IOS)
--- chrome/common/chrome_paths.h	2016-01-14 03:49:22.000000000 +0200
+++ chrome/common/chrome_paths.h	2016-01-21 19:18:51.287978456 +0200
@@ -99,6 +99,7 @@ enum {
   DIR_COMPONENT_WIDEVINE_CDM,   // Directory that contains component-updated
                                 // Widevine CDM files.
   FILE_WIDEVINE_CDM_ADAPTER,    // Full path to the Widevine CDM adapter file.
+  FILE_WIDEVINE_CDM,            // Full path to the Widevine CDM file.
   FILE_RESOURCES_PACK,          // Full path to the .pak file containing
                                 // binary data (e.g., html files and images
                                 // used by internal pages).
--- third_party/widevine/cdm/stub/widevine_cdm_version.h	2016-01-14 01:05:17.000000000 +0200
+++ third_party/widevine/cdm/stub/widevine_cdm_version.h	2016-01-21 19:18:51.287978456 +0200
@@ -12,4 +12,6 @@
 
 #define WIDEVINE_CDM_AVAILABLE
 
+#define WIDEVINE_CDM_VERSION_STRING "@WIDEVINE_VERSION@"
+
 #endif  // WIDEVINE_CDM_VERSION_H_
