diff -up chromium-120.0.6099.56/media/base/media_switches.cc.me chromium-120.0.6099.56/media/base/media_switches.cc
--- chromium-120.0.6099.56/media/base/media_switches.cc.me	2023-12-02 11:43:21.990775897 +0100
+++ chromium-120.0.6099.56/media/base/media_switches.cc	2023-12-02 11:45:23.248006377 +0100
@@ -1636,7 +1636,7 @@ BASE_FEATURE(kUseSharedImagesForPepperVi
 // Enables FFmpeg allow lists for supported codecs / containers.
 BASE_FEATURE(kFFmpegAllowLists,
              "FFmpegAllowLists",
-             base::FEATURE_ENABLED_BY_DEFAULT);
+             base::FEATURE_DISABLED_BY_DEFAULT);
 
 #if BUILDFLAG(ENABLE_FFMPEG_VIDEO_DECODERS)
 // Allows decoding of theora / vp3 content.
