From 3a7b8dd0fcceffcfd0ea7e3186d2850deed7a00b Mon Sep 17 00:00:00 2001
From: Stephan Hartmann <stha09@googlemail.com>
Date: Wed, 6 Oct 2021 15:36:47 +0000
Subject: [PATCH] IWYU: add vector for std::vector in CouponDB

---
 chrome/browser/commerce/coupons/coupon_db.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/chrome/browser/commerce/coupons/coupon_db.h b/chrome/browser/commerce/coupons/coupon_db.h
index f0758f4..93e2dd3 100644
--- a/chrome/browser/commerce/coupons/coupon_db.h
+++ b/chrome/browser/commerce/coupons/coupon_db.h
@@ -5,6 +5,8 @@
 #ifndef CHROME_BROWSER_COMMERCE_COUPONS_COUPON_DB_H_
 #define CHROME_BROWSER_COMMERCE_COUPONS_COUPON_DB_H_
 
+#include <vector>
+
 #include "base/callback_helpers.h"
 #include "base/memory/weak_ptr.h"
 #include "url/gurl.h"
-- 
2.32.0

