--- build/linux/system.gyp.orig	2013-04-02 00:42:20.645560574 +0000
+++ build/linux/system.gyp	2013-04-02 00:43:15.615870417 +0000
@@ -491,7 +491,7 @@
                      '--name', 'LibSpeechdLoader',
                      '--output-h', '<(output_h)',
                      '--output-cc', '<(output_cc)',
-                     '--header', '<libspeechd.h>',
+                     '--header', '<speech-dispatcher/libspeechd.h>',
                      '--link-directly=<(linux_link_libspeechd)',
                      'spd_open',
                      'spd_say',
