--- chromium-21.0.1180.79/ppapi/ppapi_untrusted.gyp.orig	2012-06-12 10:20:16.000000000 +0200
+++ chromium-21.0.1180.79/ppapi/ppapi_untrusted.gyp	2012-06-12 10:20:52.000000000 +0200
@@ -80,7 +80,7 @@
         ],
       },
       'conditions': [
-        ['target_arch!="arm"', {
+        ['target_arch!="arm" and disable_glibc==0', {
           'variables': {
             'build_glibc': 1,
             # NOTE: Use /lib, not /lib64 here; it is a symbolic link which
