Subversion Revision: 143170
diff --git Source/WebCore/platform/graphics/harfbuzz/HarfBuzzShaper.cpp Source/WebCore/platform/graphics/harfbuzz/HarfBuzzShaper.cpp
index 1cf3979a560292679e0bb706ec3a63ae4bcf2b51..69ba6ff231e069cdd59bf0e411c8731fe23c229c 100644
--- third_party/WebKit/Source/WebCore/platform/graphics/harfbuzz/HarfBuzzShaper.cpp
+++ third_party/WebKit/Source/WebCore/platform/graphics/harfbuzz/HarfBuzzShaper.cpp
@@ -329,6 +329,9 @@ bool HarfBuzzShaper::shapeHarfBuzzRuns(bool shouldSetDirection)
         hb_buffer_set_script(harfBuzzBuffer.get(), currentRun->script());
         if (shouldSetDirection)
             hb_buffer_set_direction(harfBuzzBuffer.get(), currentRun->rtl() ? HB_DIRECTION_RTL : HB_DIRECTION_LTR);
+        else
+            // Leaving direction to HarfBuzz to guess is *really* bad, but will do for now.
+            hb_buffer_guess_segment_properties(harfBuzzBuffer.get());
 
         // Add a space as pre-context to the buffer. This prevents showing dotted-circle
         // for combining marks at the beginning of runs.
