Source: Tom Briden <tom@decompile.me.uk>
Upstream: No

diff --git a/chrome/browser/ui/views/side_panel/BUILD.gn b/chrome/browser/ui/views/side_panel/BUILD.gn
index ea57f9b16f6b8..20d8681ee9af2 100644
--- a/chrome/browser/ui/views/side_panel/BUILD.gn
+++ b/chrome/browser/ui/views/side_panel/BUILD.gn
@@ -142,18 +142,30 @@ static_library("side_panel") {
   }
   public_deps = [
     "//base",
+    "//chrome/browser/cart:mojo_bindings",
     "//chrome/browser/companion/core/mojom:mojo_bindings",
     "//chrome/browser/profiles:profile",
     "//chrome/browser/ui/actions:actions_headers",
     "//chrome/browser/ui/color:color_headers",
+    "//chrome/browser/ui/webui/side_panel/customize_chrome:mojo_bindings",
+    "//chrome/browser/ui/webui/side_panel/bookmarks:mojo_bindings",
     "//chrome/browser/ui/webui/side_panel/performance_controls:mojo_bindings",
+    "//chrome/browser/ui/webui/side_panel/reading_list:mojo_bindings",
+    "//chrome/browser/ui:webui_name_variants",
     "//chrome/common",
     "//chrome/common/accessibility:mojo_bindings",
+    "//chrome/common/companion:mojo_bindings",
+    "//components/enterprise/buildflags:buildflags",
+    "//components/enterprise/common/proto:connectors_proto",
     "//components/lens",
     "//components/lens:buildflags",
     "//components/omnibox/browser",
+    "//components/page_image_service/mojom:mojo_bindings",
+    "//components/paint_preview/buildflags:buildflags",
     "//components/prefs",
     "//components/search_engines",
+    "//components/segmentation_platform/public/proto:proto",
+    "//components/webapps/common:mojo_bindings",
     "//content/public/browser",
     "//extensions/browser",
     "//extensions/common",
@@ -165,6 +177,8 @@ static_library("side_panel") {
     "//ui/gfx/geometry",
     "//ui/views",
     "//ui/views/controls/webview",
+    "//ui/webui/resources/cr_components/commerce:mojo_bindings",
+    "//ui/webui/resources/cr_components/help_bubble:mojo_bindings",
     "//url",
   ]
   deps = [
diff --git a/chrome/browser/ui/webui/top_chrome/BUILD.gn b/chrome/browser/ui/webui/top_chrome/BUILD.gn
index 5830f1f5e234a..6749b8fd0da56 100644
--- a/chrome/browser/ui/webui/top_chrome/BUILD.gn
+++ b/chrome/browser/ui/webui/top_chrome/BUILD.gn
@@ -19,6 +19,7 @@ source_set("top_chrome") {
   deps = [
     "//base",
     "//chrome/browser/profiles:profile",
+    "//chrome/browser/ui:webui_name_variants",
     "//components/site_engagement/content:content",
     "//content/public/browser",
     "//ui/webui",
