--- base/check.h.orig	2021-04-15 19:44:05.656259734 -0400
+++ base/check.h	2021-04-15 19:44:13.386279197 -0400
@@ -85,7 +85,7 @@
   // Stream for adding optional details to the error message.
   std::ostream& stream();
 
-  NOMERGE ~CheckError();
+  ~CheckError();
 
   CheckError(const CheckError& other) = delete;
   CheckError& operator=(const CheckError& other) = delete;
--- base/compiler_specific.h.orig	2021-04-15 19:44:28.781317968 -0400
+++ base/compiler_specific.h	2021-04-15 19:44:54.136381820 -0400
@@ -332,11 +332,4 @@
 
 #endif  // defined(__clang_analyzer__)
 
-// Use nomerge attribute to disable optimization of merging multiple same calls.
-#if defined(__clang__) && __has_attribute(nomerge) && !defined(OS_CHROMEOS)
-#define NOMERGE [[clang::nomerge]]
-#else
-#define NOMERGE
-#endif
-
 #endif  // BASE_COMPILER_SPECIFIC_H_
