--- chromium-23.0.1271.95/third_party/tcmalloc/chromium/src/base/linux_syscall_support.h	2012-07-10 10:48:31.000000000 +0300
+++ chromium-23.0.1271.95/third_party/tcmalloc/chromium/src/base/linux_syscall_support.h	2012-07-11 23:44:11.000000000 +0300
@@ -243,14 +243,13 @@ struct kernel_rusage {
   long               ru_nivcsw;
 };

-struct siginfo;
 #if defined(__i386__) || defined(__arm__) || defined(__PPC__)

 /* include/asm-{arm,i386,mips,ppc}/signal.h                                  */
 struct kernel_old_sigaction {
   union {
     void             (*sa_handler_)(int);
-    void             (*sa_sigaction_)(int, struct siginfo *, void *);
+    void             (*sa_sigaction_)(int, siginfo_t *, void *);
   };
   unsigned long      sa_mask;
   unsigned long      sa_flags;
@@ -287,13 +286,13 @@ struct kernel_sigaction {
   unsigned long      sa_flags;
   union {
     void             (*sa_handler_)(int);
-    void             (*sa_sigaction_)(int, struct siginfo *, void *);
+    void             (*sa_sigaction_)(int, siginfo_t *, void *);
   };
   struct kernel_sigset_t sa_mask;
 #else
   union {
     void             (*sa_handler_)(int);
-    void             (*sa_sigaction_)(int, struct siginfo *, void *);
+    void             (*sa_sigaction_)(int, siginfo_t *, void *);
   };
   unsigned long      sa_flags;
   void               (*sa_restorer)(void);
