--- a/build/toolchain/toolchain.gni
+++ b/build/toolchain/toolchain.gni
@@ -51,6 +51,10 @@
   }
 }
 
+declare_args() {
+  is_musl = false
+}
+
 # Extension for shared library files (including leading dot).
 if (is_apple) {
   shlib_extension = ".dylib"
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -1216,8 +1220,13 @@
     if (current_cpu == "x64") {
       if (is_clang && !is_android && !is_nacl && !is_fuchsia &&
           !is_chromeos_device) {
-        cflags += [ "--target=x86_64-unknown-linux-gnu" ]
-        ldflags += [ "--target=x86_64-unknown-linux-gnu" ]
+        if (is_musl) {
+          cflags += [ "--target=x86_64-unknown-linux-musl" ]
+          ldflags += [ "--target=x86_64-unknown-linux-musl" ]
+        } else {
+          cflags += [ "--target=x86_64-unknown-linux-gnu" ]
+          ldflags += [ "--target=x86_64-unknown-linux-gnu" ]
+        }
       } else {
         cflags += [ "-m64" ]
         ldflags += [ "-m64" ]
@@ -1225,8 +1234,13 @@
       cflags += [ "-msse3" ]
     } else if (current_cpu == "x86") {
       if (is_clang && !is_android && !is_nacl && !is_chromeos_device) {
-        cflags += [ "--target=i386-unknown-linux-gnu" ]
-        ldflags += [ "--target=i386-unknown-linux-gnu" ]
+        if (is_musl) {
+          cflags += [ "--target=i386-unknown-linux-musl" ]
+          ldflags += [ "--target=i386-unknown-linux-musl" ]
+        } else {
+          cflags += [ "--target=i386-unknown-linux-gnu" ]
+          ldflags += [ "--target=i386-unknown-linux-gnu" ]
+        }
       } else {
         cflags += [ "-m32" ]
         ldflags += [ "-m32" ]
@@ -1254,8 +1268,13 @@
     } else if (current_cpu == "arm64") {
       if (is_clang && !is_android && !is_nacl && !is_fuchsia &&
           !is_chromeos_device) {
-        cflags += [ "--target=aarch64-linux-gnu" ]
-        ldflags += [ "--target=aarch64-linux-gnu" ]
+        if (is_musl) {
+          cflags += [ "--target=aarch64-linux-musl" ]
+          ldflags += [ "--target=aarch64-linux-musl" ]
+        } else {
+          cflags += [ "--target=aarch64-linux-gnu" ]
+          ldflags += [ "--target=aarch64-linux-gnu" ]
+        }
       }
     } else if (current_cpu == "mipsel" && !is_nacl) {
       ldflags += [ "-Wl,--hash-style=sysv" ]
