diff -up chromium-124.0.6367.49/third_party/libavif/BUILD.gn.me chromium-124.0.6367.49/third_party/libavif/BUILD.gn
--- chromium-124.0.6367.49/third_party/libavif/BUILD.gn.me	2024-04-15 22:29:28.024287328 +0200
+++ chromium-124.0.6367.49/third_party/libavif/BUILD.gn	2024-04-15 22:29:33.205411066 +0200
@@ -85,7 +85,6 @@ component("libavif_enc") {
   configs += [ ":avif_config" ]
 
   deps = [
-    "//third_party/libwebp:libwebp_sharpyuv",
     "//third_party/libyuv",
   ]
   defines = [ "AVIF_LIBSHARPYUV_ENABLED" ]
