From 127ec3b1bf26ab37f2ae8333f284008868756274 Mon Sep 17 00:00:00 2001
From: Stephan Hartmann <stha09@googlemail.com>
Date: Sat, 21 Nov 2020 15:59:23 +0000
Subject: [PATCH] IWYU: size_t is defined stddef.h

Change-Id: I4400ac7c6004b49ec6e72c44f2754e2166642f88
---

diff --git a/components/bookmarks/browser/bookmark_model_observer.h b/components/bookmarks/browser/bookmark_model_observer.h
index 69c40e7..0e5eb96 100644
--- components/bookmarks/browser/bookmark_model_observer.h
+++ components/bookmarks/browser/bookmark_model_observer.h
@@ -7,6 +7,8 @@
 
 #include <set>
 
+#include <stddef.h>
+
 class GURL;
 
 namespace bookmarks {
