--- build/gyp_chromium.orig	2014-01-29 01:22:52.527247671 +0000
+++ build/gyp_chromium	2014-01-29 01:23:49.857623290 +0000
@@ -38,8 +38,6 @@
 sys.path.insert(1, os.path.join(chrome_src, 'third_party', 'WebKit',
     'Source', 'build', 'scripts'))
 
-import find_depot_tools
-
 # On Windows, Psyco shortens warm runs of build/gyp_chromium by about
 # 20 seconds on a z600 machine with 12 GB of RAM, from 90 down to 70
 # seconds.  Conversely, memory usage of build/gyp_chromium with Psyco
