--- /dev/null   2026-04-08 02:08:03.222890460 +0200
+++ b/build/config/linux/musl.gni   2026-04-10 01:18:23.479529250 +0200
@@ -0,0 +1,3 @@
+declare_args() {
+  is_musl = false
+}
--- a/build/config/compiler_cpu_abi.gn
+++ b/build/config/compiler_cpu_abi.gn
@@ -3,6 +3,7 @@
 # found in the LICENSE file.
 
 import("//build/config/chromeos/args.gni")
+import("//build/config/linux/musl.gni")
 import("//build/toolchain/toolchain.gni")
 
 if (current_cpu == "arm" || current_cpu == "arm64") {
@@ -49,8 +50,13 @@
   # simplicity we always explicitly set the architecture.
   if (current_cpu == "x64") {
     if (is_clang && !is_android && !is_fuchsia && !is_chromeos_device) {
-      cpu_abi_cflags += [ "--target=x86_64-unknown-linux-gnu" ]
-      cpu_abi_ldflags += [ "--target=x86_64-unknown-linux-gnu" ]
+      if (is_musl) {
+        cpu_abi_cflags += [ "--target=x86_64-unknown-linux-musl" ]
+        cpu_abi_ldflags += [ "--target=x86_64-unknown-linux-musl" ]
+      } else{
+        cpu_abi_cflags += [ "--target=x86_64-unknown-linux-gnu" ]
+        cpu_abi_ldflags += [ "--target=x86_64-unknown-linux-gnu" ]
+      }
     } else {
       cpu_abi_cflags += [ "-m64" ]
       cpu_abi_ldflags += [ "-m64" ]
@@ -63,8 +69,13 @@
     }
   } else if (current_cpu == "x86") {
     if (is_clang && !is_android && !is_chromeos_device) {
-      cpu_abi_cflags += [ "--target=i386-unknown-linux-gnu" ]
-      cpu_abi_ldflags += [ "--target=i386-unknown-linux-gnu" ]
+      if (is_musl) {
+        cpu_abi_cflags += [ "--target=i386-unknown-linux-musl" ]
+        cpu_abi_ldflags += [ "--target=i386-unknown-linux-musl" ]
+      } else {
+        cpu_abi_cflags += [ "--target=i386-unknown-linux-gnu" ]
+        cpu_abi_ldflags += [ "--target=i386-unknown-linux-gnu" ]
+      }
     } else {
       cpu_abi_cflags += [ "-m32" ]
       cpu_abi_ldflags += [ "-m32" ]
@@ -75,8 +86,13 @@
     ]
   } else if (current_cpu == "arm") {
     if (is_clang && !is_android && !is_chromeos_device) {
-      cpu_abi_cflags += [ "--target=arm-linux-gnueabihf" ]
-      cpu_abi_ldflags += [ "--target=arm-linux-gnueabihf" ]
+      if (is_musl) {
+        cpu_abi_cflags += [ "--target=arm-linux-musl" ]
+        cpu_abi_ldflags += [ "--target=arm-linux-musl" ]
+      } else {
+        cpu_abi_cflags += [ "--target=arm-linux-gnueabihf" ]
+        cpu_abi_ldflags += [ "--target=arm-linux-gnueabihf" ]
+      }
     }
     cpu_abi_cflags += [
       "-march=$arm_arch",
@@ -100,8 +100,13 @@
     ]
   } else if (current_cpu == "arm64") {
     if (is_clang && !is_android && !is_fuchsia && !is_chromeos_device) {
-      cpu_abi_cflags += [ "--target=aarch64-linux-gnu" ]
-      cpu_abi_ldflags += [ "--target=aarch64-linux-gnu" ]
+      if (is_musl) {
+        cpu_abi_cflags += [ "--target=aarch64-linux-musl" ]
+        cpu_abi_ldflags += [ "--target=aarch64-linux-musl" ]
+      } else {
+        cpu_abi_cflags += [ "--target=aarch64-linux-gnu" ]
+        cpu_abi_ldflags += [ "--target=aarch64-linux-gnu" ]
+      }
     }
     if (target_os == "chromeos" && cros_target_cpu_arch != "" &&
         is_a_target_toolchain) {
--- a/buildtools/third_party/libc++/__config_site
+++ b/buildtools/third_party/libc++/__config_site
@@ -26,12 +26,6 @@
 #define _LIBCPP_HAS_MONOTONIC_CLOCK 1
 #define _LIBCPP_HAS_TERMINAL 1
 
-#ifdef ANDROID_HOST_MUSL
-#define _LIBCPP_HAS_MUSL_LIBC 1
-#else
-#define _LIBCPP_HAS_MUSL_LIBC 0
-#endif
-
 #ifdef _WIN32
 #define _LIBCPP_HAS_THREAD_API_PTHREAD 0
 #define _LIBCPP_HAS_THREAD_API_EXTERNAL 0
--- a/build/config/rust.gni
+++ b/build/config/rust.gni
@@ -5,6 +5,7 @@
 import("//build/config/chrome_build.gni")
 import("//build/config/compiler/compiler.gni")
 import("//build/config/sanitizers/sanitizers.gni")
+import("//build/config/linux/musl.gni")
 import("//build/toolchain/toolchain.gni")
 
 if (is_android) {
--- a/build/config/c++/BUILD.gn
+++ b/build/config/c++/BUILD.gn
@@ -3,6 +3,7 @@
 import("//build/config/chromeos/ui_mode.gni")
 import("//build/config/compiler/compiler.gni")
 import("//build/config/dcheck_always_on.gni")
+import("//build/config/linux/musl.gni")
 import("//buildtools/deps_revisions.gni")
 
 assert(use_custom_libcxx, "should only be used if use_custom_libcxx is set")
@@ -60,6 +61,12 @@
 
   defines += [ "CR_LIBCXX_REVISION=$libcxx_revision" ]
 
+  if (is_musl) {
+    defines += [ "_LIBCPP_HAS_MUSL_LIBC=1" ]
+  } else {
+    defines += [ "_LIBCPP_HAS_MUSL_LIBC=0" ]
+  }
+
   if (is_win) {
     # Intentionally not using libc++abi on Windows because libc++abi only
     # implements the Itanium C++ ABI, and not the Microsoft ABI which we use on
