--- chrome/browser/ui/views/accessibility/invert_bubble_view.cc.orig	2019-12-11 10:41:00.069472967 -0500
+++ chrome/browser/ui/views/accessibility/invert_bubble_view.cc	2019-12-11 10:41:48.254466283 -0500
@@ -47,7 +47,7 @@
 // Tag value used to uniquely identify the "learn more" (?) button.
 constexpr int kLearnMoreButton = 100;
 
-std::unique_ptr<views::View> CreateExtraView(views::ButtonListener* listener) {
+std::unique_ptr<views::View> CreateExtraViewInternal(views::ButtonListener* listener) {
   auto learn_more = views::CreateVectorImageButton(listener);
   views::SetImageFromVectorIcon(learn_more.get(),
                                 vector_icons::kHelpOutlineIcon);
@@ -95,7 +95,7 @@
       dark_theme_(nullptr) {
   DialogDelegate::set_button_label(ui::DIALOG_BUTTON_OK,
                                    l10n_util::GetStringUTF16(IDS_DONE));
-  DialogDelegate::SetExtraView(::CreateExtraView(this));
+  DialogDelegate::SetExtraView(::CreateExtraViewInternal(this));
   set_margins(gfx::Insets());
   chrome::RecordDialogCreation(chrome::DialogIdentifier::INVERT);
 }
