--- chrome/browser/about_flags.cc.orig	2018-09-28 20:23:00.831827041 +0200
+++ chrome/browser/about_flags.cc	2018-09-28 20:23:27.702828427 +0200
@@ -1837,7 +1837,7 @@
      SINGLE_VALUE_TYPE(switches::kEnableWebGLDraftExtensions)},
 #if BUILDFLAG(ENABLE_DICE_SUPPORT)
     {"account-consistency", flag_descriptions::kAccountConsistencyName,
-     flag_descriptions::kAccountConsistencyDescription, kOsAll,
+     flag_descriptions::kAccountConsistencyDescription, 0,
      FEATURE_WITH_PARAMS_VALUE_TYPE(kAccountConsistencyFeature,
                                     kAccountConsistencyFeatureVariations,
                                     "AccountConsistencyVariations")},
