--- a/base/files/file_util_linux.cc
+++ b/base/files/file_util_linux.cc
@@ -23,14 +23,14 @@
 
   // Not all possible |statfs_buf.f_type| values are in linux/magic.h.
   // Missing values are copied from the statfs man page.
-  switch (statfs_buf.f_type) {
+  switch (static_cast<uintmax_t>(statfs_buf.f_type)) {
     case 0:
       *type = FILE_SYSTEM_0;
       break;
     case EXT2_SUPER_MAGIC:  // Also ext3 and ext4
     case MSDOS_SUPER_MAGIC:
     case REISERFS_SUPER_MAGIC:
-    case static_cast<int>(BTRFS_SUPER_MAGIC):
+    case BTRFS_SUPER_MAGIC:
     case 0x5346544E:  // NTFS
     case 0x58465342:  // XFS
     case 0x3153464A:  // JFS
@@ -40,14 +40,14 @@
       *type = FILE_SYSTEM_NFS;
       break;
     case SMB_SUPER_MAGIC:
-    case static_cast<int>(0xFF534D42):  // CIFS
+    case 0xFF534D42:  // CIFS
       *type = FILE_SYSTEM_SMB;
       break;
     case CODA_SUPER_MAGIC:
       *type = FILE_SYSTEM_CODA;
       break;
-    case static_cast<int>(HUGETLBFS_MAGIC):
-    case static_cast<int>(RAMFS_MAGIC):
+    case HUGETLBFS_MAGIC:
+    case RAMFS_MAGIC:
     case TMPFS_MAGIC:
       *type = FILE_SYSTEM_MEMORY;
       break;
--- a/base/system/sys_info_posix.cc
+++ b/base/system/sys_info_posix.cc
@@ -100,10 +100,10 @@
   if (HANDLE_EINTR(statfs(path.value().c_str(), &stats)) != 0)
     return false;
 
-  switch (stats.f_type) {
+  switch (static_cast<uintmax_t>(stats.f_type)) {
     case TMPFS_MAGIC:
-    case static_cast<int>(HUGETLBFS_MAGIC):
-    case static_cast<int>(RAMFS_MAGIC):
+    case HUGETLBFS_MAGIC:
+    case RAMFS_MAGIC:
       return true;
   }
   return false;
