--- third_party/WebKit/Source/platform/image-decoders/png/PNGImageDecoder.cpp.orig	2016-06-01 17:42:30.396115414 +0200
+++ third_party/WebKit/Source/platform/image-decoders/png/PNGImageDecoder.cpp	2016-06-01 17:43:05.182117208 +0200
@@ -234,7 +234,7 @@
 #endif
             png_uint_32 profileLength = 0;
             if (png_get_iCCP(png, info, &profileName, &compressionType, &profile, &profileLength)) {
-                setColorProfileAndTransform(profile, profileLength, imageHasAlpha, false /* useSRGB */);
+                setColorProfileAndTransform((const char*) profile, profileLength, imageHasAlpha, false /* useSRGB */);
             }
         }
 #endif // PNG_iCCP_SUPPORTED
