This macro is defined in glibc, but not musl.

--- a/sandbox/linux/suid/process_util.h.orig
+++ b/sandbox/linux/suid/process_util.h
@@ -11,6 +11,16 @@
 #include <stdint.h>
 #include <sys/types.h>
 
+// Some additional functions
+#if !defined(TEMP_FAILURE_RETRY)
+# define TEMP_FAILURE_RETRY(expression) \
+	(__extension__			\
+	 ({ long int __result;		\
+	  do __result = (long int) (expression); \
+	  while (__result == -1L && errno == EINTR); \
+	  __result; }))
+#endif
+
 // This adjusts /proc/process/oom_score_adj so the Linux OOM killer
 // will prefer certain process types over others. The range for the
 // adjustment is [-1000, 1000], with [0, 1000] being user accessible.
