Index: net/quic/crypto/proof_verifier_chromium.cc
diff --git net/quic/crypto/proof_verifier_chromium.cc net/quic/crypto/proof_verifier_chromium.cc
index 546b00740fef947a95a109e7e9bc22ffdedce785..160191bf6226292d49997bec94429e3a1d0fe990 100644
--- net/quic/crypto/proof_verifier_chromium.cc
+++ net/quic/crypto/proof_verifier_chromium.cc
@@ -426,6 +426,8 @@ int ProofVerifierChromium::Job::DoVerifyCertComplete(int result) {
     int ct_result = OK;
     if (verify_details_->ct_verify_result.cert_policy_compliance !=
             ct::CertPolicyCompliance::CERT_POLICY_COMPLIES_VIA_SCTS &&
+        verify_details_->ct_verify_result.cert_policy_compliance !=
+            ct::CertPolicyCompliance::CERT_POLICY_BUILD_NOT_TIMELY &&
         transport_security_state_->ShouldRequireCT(
             hostname_, cert_verify_result.verified_cert.get(),
             cert_verify_result.public_key_hashes)) {
Index: net/socket/ssl_client_socket_impl.cc
diff --git net/socket/ssl_client_socket_impl.cc net/socket/ssl_client_socket_impl.cc
index bb76bf8c46d6d84f7d5ad7d48eacfe751204d151..a5d2afd7d8483a4996e04f9b68e3756f55af5d1a 100644
--- net/socket/ssl_client_socket_impl.cc
+++ net/socket/ssl_client_socket_impl.cc
@@ -1556,6 +1556,8 @@ int SSLClientSocketImpl::VerifyCT() {
 
   if (ct_verify_result_.cert_policy_compliance !=
           ct::CertPolicyCompliance::CERT_POLICY_COMPLIES_VIA_SCTS &&
+      ct_verify_result_.cert_policy_compliance !=
+          ct::CertPolicyCompliance::CERT_POLICY_BUILD_NOT_TIMELY &&
       transport_security_state_->ShouldRequireCT(
           host_and_port_.host(), server_cert_verify_result_.verified_cert.get(),
           server_cert_verify_result_.public_key_hashes)) {
Index: net/spdy/spdy_session.cc
diff --git net/spdy/spdy_session.cc net/spdy/spdy_session.cc
index b79c5e62e3d17f2745c369428c8ac35d9d663349..9315380e45f7f7d9ebfa5bdfe62d11c93705cddc 100644
--- net/spdy/spdy_session.cc
+++ net/spdy/spdy_session.cc
@@ -617,6 +617,8 @@ bool SpdySession::CanPool(TransportSecurityState* transport_security_state,
 
   if (ssl_info.ct_cert_policy_compliance !=
           ct::CertPolicyCompliance::CERT_POLICY_COMPLIES_VIA_SCTS &&
+      ssl_info.ct_cert_policy_compliance !=
+          ct::CertPolicyCompliance::CERT_POLICY_BUILD_NOT_TIMELY &&
       transport_security_state->ShouldRequireCT(
           new_hostname, ssl_info.cert.get(), ssl_info.public_key_hashes)) {
     return false;
