--- base/process/memory_linux.cc.orig	2021-03-02 18:44:58.000000000 +0100
+++ base/process/memory_linux.cc	2021-03-03 12:05:57.952296984 +0100
@@ -151,7 +151,7 @@
     (!defined(LIBC_GLIBC) && !BUILDFLAG(USE_TCMALLOC))
   *result = malloc(size);
 #elif defined(LIBC_GLIBC) && !BUILDFLAG(USE_TCMALLOC)
-  *result = __libc_malloc(size);
+  *result = malloc(size);
 #elif BUILDFLAG(USE_TCMALLOC)
   *result = tc_malloc_skip_new_handler(size);
 #endif
