--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -1641,7 +1641,7 @@
   # Some build configs use older versions of clang that don't support WSMs
   if (!is_nacl && default_toolchain != "//build/toolchain/cros:target" &&
       !llvm_android_mainline && is_clang &&
-      clang_warning_suppression_file != "") {
+      clang_warning_suppression_file != "" && clang_version != "19") {
     from_build_root =
         rebase_path(clang_warning_suppression_file, root_build_dir)
     inputs = [ clang_warning_suppression_file ]
