--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -328,6 +328,10 @@
     configs -= [ ":clang_warning_suppression" ]
   }
 
+  if (clang_version == "19") {
+    configs -= [ ":clang_warning_suppression" ]
+  }
+
   # Here we enable -fno-delete-null-pointer-checks, which makes various nullptr
   # operations (e.g. dereferencing) into defined behavior. This avoids deletion
   # of some security-critical code: see https://crbug.com/1139129.
