--- build/config/compiler/BUILD.gn.orig	2021-01-20 09:01:44.362114311 -0500
+++ build/config/compiler/BUILD.gn	2021-01-20 09:03:41.880232248 -0500
@@ -1534,9 +1534,6 @@
 
           # TODO(https://crbug.com/1028110): Evaluate and possible enable.
           "-Wno-deprecated-copy",
-
-          # TODO(https://crbug.com/1050281): Clean up, enable.
-          "-Wno-non-c-typedef-for-linkage",
         ]
 
         cflags_c += [
@@ -1548,13 +1545,6 @@
           # TODO(https://crbug.com/1143700): Clean up and re-enable.
           cflags_objc = [ "-Wno-gnu-folding-constant" ]
         }
-
-        if (enable_wmax_tokens) {
-          cflags += [ "-Wmax-tokens" ]
-        } else {
-          # TODO(https://crbug.com/1049569): Remove after Clang 87b235db.
-          cflags += [ "-Wno-max-tokens" ]
-        }
       }
     }
   }
