--- build/config/compiler/BUILD.gn.orig	2020-04-03 00:11:19.000000000 -0400
+++ build/config/compiler/BUILD.gn	2020-04-11 23:49:51.153896808 -0400
@@ -702,14 +702,6 @@
     ldflags += [ "-Wl,--no-rosegment" ]
   }
 
-  # This flag enforces that member pointer base types are complete. It helps
-  # prevent us from running into problems in the Microsoft C++ ABI (see
-  # https://crbug.com/847724).
-  if (is_clang && !is_nacl && target_os != "chromeos" && !use_xcode_clang &&
-      (is_win || use_custom_libcxx)) {
-    cflags += [ "-fcomplete-member-pointers" ]
-  }
-
   # Pass the same C/C++ flags to the objective C/C++ compiler.
   cflags_objc += cflags_c
   cflags_objcc += cflags_cc
@@ -1506,42 +1498,6 @@
         # Currently goma can not handle case sensitiveness for windows well.
         cflags += [ "-Wno-nonportable-include-path" ]
       }
-
-      if (current_toolchain == host_toolchain || !use_xcode_clang) {
-        # Flags NaCl (Clang 3.7) and Xcode 9.2 (Clang clang-900.0.39.2) do not
-        # recognize.
-        cflags += [
-          # Ignore warnings about MSVC optimization pragmas.
-          # TODO(thakis): Only for no_chromium_code? http://crbug.com/912662
-          "-Wno-ignored-pragma-optimize",
-
-          # TODO(https://crbug.com/989932): Evaluate and possibly enable.
-          "-Wno-implicit-int-float-conversion",
-
-          # TODO(https://crbug.com/999886): Clean up, enable.
-          "-Wno-final-dtor-non-final-class",
-
-          # TODO(https://crbug.com/1016947) Clean up, enable.
-          "-Wno-bitwise-conditional-parentheses",
-
-          # TODO(https://crbug.com/1016945) Clean up, enable.
-          "-Wno-builtin-assume-aligned-alignment",
-
-          # TODO(https://crbug.com/1028110): Evaluate and possible enable.
-          "-Wno-deprecated-copy",
-
-          # TODO(https://crbug.com/1031169): Clean up and enable.
-          "-Wno-misleading-indentation",
-
-          # TODO(https://crbug.com/1039697): Evaluate and possibly enable.
-          "-Wno-range-loop-analysis",
-        ]
-
-        cflags_c += [
-          # TODO(https://crbug.com/995993): Clean up and enable.
-          "-Wno-implicit-fallthrough",
-        ]
-      }
     }
   }
 }
--- third_party/blink/renderer/BUILD.gn.orig	2020-04-12 00:55:11.146544560 -0400
+++ third_party/blink/renderer/BUILD.gn	2020-04-12 00:55:29.106560641 -0400
@@ -48,13 +48,6 @@
       "-Wno-implicit-float-conversion",
       "-Wno-implicit-int-conversion",
     ]
-
-    if (!is_chromeos || default_toolchain != "//build/toolchain/cros:target") {
-      cflags += [
-        "-Wno-enum-enum-conversion",
-        "-Wno-enum-float-conversion",
-      ]
-    }
   }
 }
 
