--- build/config/compiler/BUILD.gn.orig	2020-07-18 09:24:00.272560391 -0400
+++ build/config/compiler/BUILD.gn	2020-07-18 10:09:09.225465782 -0400
@@ -1495,9 +1495,6 @@
 
           # TODO(https://crbug.com/1028110): Evaluate and possible enable.
           "-Wno-deprecated-copy",
-
-          # TODO(https://crbug.com/1050281): Clean up, enable.
-          "-Wno-non-c-typedef-for-linkage",
         ]
 
         cflags_c += [
