--- build/config/compiler/BUILD.gn	2019-10-24 11:01:40.459910040 -0400
+++ build/config/compiler/BUILD.gn	2019-10-24 11:04:19.690063743 -0400
@@ -1492,37 +1492,6 @@
         # Currently goma can not handle case sensitiveness for windows well.
         cflags += [ "-Wno-nonportable-include-path" ]
       }
-
-      if (current_toolchain == host_toolchain || !use_xcode_clang) {
-        # Flags NaCl (Clang 3.7) and Xcode 9.2 (Clang clang-900.0.39.2) do not
-        # recognize.
-        cflags += [
-          # Ignore warnings about MSVC optimization pragmas.
-          # TODO(thakis): Only for no_chromium_code? http://crbug.com/912662
-          "-Wno-ignored-pragma-optimize",
-
-          # TODO(https://crbug.com/989932): Evaluate and possibly enable.
-          "-Wno-implicit-int-float-conversion",
-
-          # TODO(https://crbug.com/995200): Clean up and enable.
-          "-Wno-xor-used-as-pow",
-
-          # TODO(https://crbug.com/999871): Decide if we want to clean up the
-          # codebase or just disable this.  Doesn't seem super useful, but
-          # also fires in only 4 files.
-          "-Wno-c99-designator",
-
-          # This is a side effect of -Wc99-designator; easier to clean up.
-          "-Wno-reorder-init-list",
-
-          # TODO(https://crbug.com/999886): Clean up, enable.
-          "-Wno-final-dtor-non-final-class",
-        ]
-        cflags_c += [
-          # TODO(https://crbug.com/995993): Clean up and enable.
-          "-Wno-implicit-fallthrough",
-        ]
-      }
     }
   }
 }
