--- third_party/WebKit/Source/WebCore/make-hash-tools.pl	(revision 72664)
+++ third_party/WebKit/Source/WebCore/make-hash-tools.pl	(revision 89265)
@@ -21,5 +21,4 @@
 
 use strict;
-use Switch;
 use File::Basename;
 
@@ -29,7 +28,5 @@
 
 
-switch ($option) {
-
-case "DocTypeStrings" {
+if ($option eq "DocTypeStrings") {
 
     my $docTypeStringsGenerated    = "$outdir/DocTypeStrings.cpp";
@@ -39,7 +36,5 @@
     system("gperf --key-positions=\"*\" -s 2 $docTypeStringsGperf > $docTypeStringsGenerated") == 0 || die "calling gperf failed: $?";
 
-} # case "DocTypeStrings"
-
-case "ColorData" {
+} elsif ($option eq "ColorData") {
 
     my $colorDataGenerated         = "$outdir/ColorData.cpp";
@@ -49,5 +44,5 @@
     system("gperf --key-positions=\"*\" -D -s 2 $colorDataGperf > $colorDataGenerated") == 0 || die "calling gperf failed: $?";
 
-} # case "ColorData"
-
-} # switch ($option)
+} else {
+    die "Unknown option.";
+}
