--- a/base/files/file.h.orig
+++ b/base/files/file.h
@@ -19,7 +19,8 @@
 #include "build/build_config.h"
 
 #if defined(OS_BSD) || defined(OS_APPLE) || defined(OS_NACL) || \
-    defined(OS_FUCHSIA) || (defined(OS_ANDROID) && __ANDROID_API__ < 21)
+    defined(OS_FUCHSIA) || (defined(OS_ANDROID) && __ANDROID_API__ < 21) || \
+    (defined(OS_LINUX) && !defined(__GLIBC__))
 struct stat;
 namespace base {
 typedef struct stat stat_wrapper_t;
