--- ui/accessibility/ax_tree_serializer.h.orig	2021-03-02 18:46:00.000000000 +0100
+++ ui/accessibility/ax_tree_serializer.h	2021-03-03 14:03:43.926160333 +0100
@@ -746,7 +746,7 @@
 #endif  // defined(AX_FAIL_FAST_BUILD)
         static bool has_sent_dupe_id_err = false;
         if (!has_sent_dupe_id_err) {
-          std::srand(std::time(nullptr));  // use current time as seed.
+          std::srand(time(nullptr));  // use current time as seed.
           if (std::rand() % 50 == 0) {     // Roughly 2% of the time.
             base::debug::DumpWithoutCrashing();
             has_sent_dupe_id_err = true;  // Only send once.
