misdetected on musl...

--- third_party/angle/third_party/vulkan-validation-layers/src/loader/loader.c.orig	2017-07-25 19:06:31.000000000 +0000
+++ third_party/angle/third_party/vulkan-validation-layers/src/loader/loader.c	2017-07-27 16:20:54.721755713 +0000
@@ -234,10 +234,10 @@ static inline char *loader_secure_getenv(const char *name, const struct loader_i
     return geteuid() != getuid() || getegid() != getgid() ? NULL : loader_getenv(name, inst);
 #else
 // Linux
-#ifdef HAVE_SECURE_GETENV
+#if defined(HAVE_SECURE_GETENV) && defined(__GLIBC__)
     (void)inst;
     return secure_getenv(name);
-#elif defined(HAVE___SECURE_GETENV)
+#elif defined(HAVE___SECURE_GETENV) && defined(__GLIBC__)
     (void)inst;
     return __secure_getenv(name);
 #else

