--- a/third_party/node/node.gni
+++ b/third_party/node/node.gni
@@ -35,16 +35,5 @@
         inputs += [ "//third_party/node/mac/node-darwin-x64/bin/node" ]
       }
     }
-
-    # Automatically add a dependency to ":check_version" to ensure NodeJS is
-    # always running the expected version, except when the ':check_version'
-    # target itself is running in which case it shouldn't depend on itself.
-    if (get_label_info(":" + target_name, "label_no_toolchain") !=
-        "//third_party/node:check_version") {
-      if (!defined(deps)) {
-        deps = []
-      }
-      deps += [ "//third_party/node:check_version" ]
-    }
   }
 }
