diff -ur ./third_party/jinja2.orig/runtime.py ./third_party/jinja2/runtime.py
--- a/third_party/jinja2.orig/runtime.py	2021-09-24 00:26:35.000000000 -0400
+++ b/third_party/jinja2/runtime.py	2021-10-13 11:51:16.493986556 -0400
@@ -315,7 +315,7 @@
 
 # register the context as mapping if possible
 try:
-    from collections import Mapping
+    from collections.abc import Mapping
     Mapping.register(Context)
 except ImportError:
     pass
diff -ur ./third_party/jinja2.orig/sandbox.py ./third_party/jinja2/sandbox.py
--- a/third_party/jinja2.orig/sandbox.py	2021-09-24 00:26:35.000000000 -0400
+++ b/third_party/jinja2/sandbox.py	2021-10-13 11:50:51.513980399 -0400
@@ -14,7 +14,7 @@
 """
 import types
 import operator
-from collections import Mapping
+from collections.abc import Mapping
 from jinja2.environment import Environment
 from jinja2.exceptions import SecurityError
 from jinja2._compat import string_types, PY2
@@ -79,7 +79,7 @@
     pass
 
 #: register Python 2.6 abstract base classes
-from collections import MutableSet, MutableMapping, MutableSequence
+from collections.abc import MutableSet, MutableMapping, MutableSequence
 _mutable_set_types += (MutableSet,)
 _mutable_mapping_types += (MutableMapping,)
 _mutable_sequence_types += (MutableSequence,)
diff -ur ./third_party/jinja2.orig/tests.py ./third_party/jinja2/tests.py
--- a/third_party/jinja2.orig/tests.py	2021-09-24 00:26:35.000000000 -0400
+++ b/third_party/jinja2/tests.py	2021-10-13 11:51:51.693995227 -0400
@@ -10,7 +10,7 @@
 """
 import operator
 import re
-from collections import Mapping
+from collections.abc import Mapping
 from jinja2.runtime import Undefined
 from jinja2._compat import text_type, string_types, integer_types
 import decimal
diff -ur ./third_party/jinja2.orig/utils.py ./third_party/jinja2/utils.py
--- a/third_party/jinja2.orig/utils.py	2021-09-24 00:26:35.000000000 -0400
+++ b/third_party/jinja2/utils.py	2021-10-13 11:51:34.969991106 -0400
@@ -482,7 +482,7 @@
 
 # register the LRU cache as mutable mapping if possible
 try:
-    from collections import MutableMapping
+    from collections.abc import MutableMapping
     MutableMapping.register(LRUCache)
 except ImportError:
     pass
