--- a/chrome/test/chromedriver/BUILD.gn.orig
+++ b/chrome/test/chromedriver/BUILD.gn
@@ -308,11 +308,7 @@
   }
 }
 
-if (is_linux) {
-  chromedriver_output = "chromedriver.unstripped"
-} else {
-  chromedriver_output = "chromedriver"
-}
+chromedriver_output = "chromedriver"
 
 executable("$chromedriver_output") {
   testonly = true
@@ -336,16 +332,6 @@
   }
 }
 
-if (is_linux) {
-  strip_binary("chromedriver") {
-    testonly = true
-    binary_input = "$root_out_dir/$chromedriver_output"
-    symbol_output = "$root_out_dir/chromedriver.debug"
-    stripped_binary_output = "$root_out_dir/chromedriver"
-    deps = [ ":$chromedriver_output" ]
-  }
-}
-
 python_library("chromedriver_py_tests") {
   testonly = true
   deps = [
