--- sandbox/linux/seccomp-bpf-helpers/syscall_sets.cc.orig	2021-03-03 18:04:45.167674791 +0100
+++ sandbox/linux/seccomp-bpf-helpers/syscall_sets.cc	2021-03-03 18:13:17.036547592 +0100
@@ -394,11 +394,12 @@
   switch (sysno) {
     case __NR_exit:
     case __NR_exit_group:
+    case __NR_membarrier:
     case __NR_wait4:
     case __NR_waitid:
 #if defined(__i386__)
     case __NR_waitpid:
 #endif
       return true;
     case __NR_clone:  // Should be parameter-restricted.
     case __NR_setns:  // Privileged.
