diff --git a/third_party/swiftshader/src/Common/Socket.cpp b/third_party/swiftshader/src/Common/Socket.cpp
index a19f574..3e6655d 100644
--- third_party/swiftshader/src/Common/Socket.cpp
+++ swiftshader/src/Common/Socket.cpp
@@ -17,6 +17,7 @@
 #if defined(_WIN32)
 	#include <ws2tcpip.h>
 #else
+	#include <sys/select.h>
 	#include <unistd.h>
 	#include <netdb.h>
 	#include <netinet/in.h>
diff --git a/third_party/swiftshader/third_party/LLVM/include-linux/llvm/Config/config.h b/third_party/swiftshader/third_party/LLVM/include-linux/llvm/Config/config.h
index 509bc26..6522713 100644
--- third_party/swiftshader/third_party/LLVM/include-linux/llvm/Config/config.h
+++ swiftshader/third_party/LLVM/include-linux/llvm/Config/config.h
@@ -125,7 +125,7 @@
 #define HAVE_ERROR_T 1
 
 /* Define to 1 if you have the <execinfo.h> header file. */
-#define HAVE_EXECINFO_H 1
+/* #undef HAVE_EXECINFO_H */
 
 /* Define to 1 if you have the <fcntl.h> header file. */
 #define HAVE_FCNTL_H 1
@@ -240,10 +240,10 @@
 /* #undef HAVE_MACH_O_DYLD_H */
 
 /* Define if mallinfo() is available on this platform. */
-#define HAVE_MALLINFO 1
+/* #undef HAVE_MALLINFO */
 
 /* Define to 1 if you have the <malloc.h> header file. */
-#define HAVE_MALLOC_H 1
+/* #undef HAVE_MALLOC_H */
 
 /* Define to 1 if you have the <malloc/malloc.h> header file. */
 /* #undef HAVE_MALLOC_MALLOC_H */
diff --git a/third_party/swiftshader/third_party/LLVM/lib/Support/DynamicLibrary.cpp b/third_party/swiftshader/third_party/LLVM/lib/Support/DynamicLibrary.cpp
index bd68083..f703e13 100644
--- third_party/swiftshader/third_party/LLVM/lib/Support/DynamicLibrary.cpp
+++ swiftshader/third_party/LLVM/lib/Support/DynamicLibrary.cpp
@@ -160,7 +160,7 @@ void* DynamicLibrary::SearchForAddressOfSymbol(const char *symbolName) {
 // On linux we have a weird situation. The stderr/out/in symbols are both
 // macros and global variables because of standards requirements. So, we
 // boldly use the EXPLICIT_SYMBOL macro without checking for a #define first.
-#if defined(__linux__) && !defined(__ANDROID__)
+#if defined(__GLIBC__)
   {
     EXPLICIT_SYMBOL(stderr);
     EXPLICIT_SYMBOL(stdout);
diff --git a/third_party/swiftshader/third_party/llvm-subzero/build/Linux/include/llvm/Config/config.h b/third_party/swiftshader/third_party/llvm-subzero/build/Linux/include/llvm/Config/config.h
index e14edbd..8d7e41e 100644
--- third_party/swiftshader/third_party/llvm-subzero/build/Linux/include/llvm/Config/config.h
+++ swiftshader/third_party/llvm-subzero/build/Linux/include/llvm/Config/config.h
@@ -14,7 +14,7 @@
 #define ENABLE_CRASH_OVERRIDES 1
 
 /* Define to 1 if you have the `backtrace' function. */
-#define HAVE_BACKTRACE 1
+/* #undef  HAVE_BACKTRACE */
 
 /* Define to 1 if you have the <CrashReporterClient.h> header file. */
 /* #undef HAVE_CRASHREPORTERCLIENT_H */
@@ -55,7 +55,7 @@
 #define HAVE_ERRNO_H 1
 
 /* Define to 1 if you have the <execinfo.h> header file. */
-#define HAVE_EXECINFO_H 1
+/* #undef HAVE_EXECINFO_H */
 
 /* Define to 1 if you have the <fcntl.h> header file. */
 #define HAVE_FCNTL_H 1
@@ -130,10 +130,10 @@
 /* #undef HAVE_MALLCTL */
 
 /* Define to 1 if you have the `mallinfo' function. */
-#define HAVE_MALLINFO 1
+/* #undef HAVE_MALLINFO */
 
 /* Define to 1 if you have the <malloc.h> header file. */
-#define HAVE_MALLOC_H 1
+/* #undef HAVE_MALLOC_H */
 
 /* Define to 1 if you have the <malloc/malloc.h> header file. */
 /* #undef HAVE_MALLOC_MALLOC_H */
