--- buildtools/third_party/libc++/trunk/include/__locale.orig	2017-09-14 19:06:26.000000000 +0000
+++ buildtools/third_party/libc++/trunk/include/__locale	2017-09-15 08:58:37.520038686 +0000
@@ -34,7 +34,7 @@
 # include <support/solaris/xlocale.h>
 #elif defined(_NEWLIB_VERSION)
 # include <support/newlib/xlocale.h>
-#elif (defined(__GLIBC__) || defined(__APPLE__)      || defined(__FreeBSD__) \
+#elif (defined(__APPLE__)      || defined(__FreeBSD__) \
     || defined(__EMSCRIPTEN__) || defined(__IBMCPP__))
 # include <xlocale.h>
 #elif defined(__Fuchsia__)
