--sysroot should be enough.

--- a/scripts/common.m4	2014-02-02 11:58:17.361448885 +0100
+++ b/scripts/common.m4	2014-02-02 11:59:13.048185911 +0100
@@ -63,8 +63,7 @@ AC_DEFUN([TORRENT_WITH_SYSROOT], [
       else
         AC_MSG_RESULT($withval)
         
-        CXXFLAGS="$CXXFLAGS -isysroot $withval"
-        LDFLAGS="$LDFLAGS -Wl,-syslibroot,$withval"
+        CXXFLAGS="$CXXFLAGS --sysroot=$withval"
       fi
     ])
 ])
