--- a/scripts/checks.m4	2015-07-02 00:32:45.000000000 +0200
+++ b/scripts/checks.m4	2015-07-14 15:39:07.334398349 +0200
@@ -463,19 +463,9 @@
       fi
       
       if eval $xmlrpc_cc_prg --version 2>/dev/null >/dev/null; then
+        AC_MSG_RESULT(ok)
         CXXFLAGS="$CXXFLAGS `$xmlrpc_cc_prg --cflags server-util`"
         LIBS="$LIBS `$xmlrpc_cc_prg server-util --libs`"
-
-        AC_TRY_LINK(
-        [ #include <xmlrpc-c/server.h>
-        ],[ xmlrpc_registry_new(NULL); ],
-        [
-          AC_MSG_RESULT(ok)
-        ], [
-          AC_MSG_RESULT(failed)
-          AC_MSG_ERROR(Could not compile XMLRPC-C test.)
-        ])
-
         AC_DEFINE(HAVE_XMLRPC_C, 1, Support for XMLRPC-C.)
 
       else
