--- communicate.h.orig	2014-01-08 10:28:04.825714200 +0100
+++ communicate.h	2014-01-08 10:29:13.825221763 +0100
@@ -70,7 +70,7 @@
 # define STAT64_SUPPORT
 #else
 # ifndef __APPLE__
-#  warning Not using stat64 support
+#  define STAT64_SUPPORT
 # endif
 /* if glibc is 2.0 or older, undefine these again */
 # undef STAT64_SUPPORT
--- libfakeroot.c.orig	2014-01-08 10:44:53.185797653 +0100
+++ libfakeroot.c	2014-01-08 10:45:18.249966551 +0100
@@ -31,6 +31,10 @@
 */
 #define _GNU_SOURCE
 
+#ifndef _STAT_VER
+#define _STAT_VER 0
+#endif
+
 #define FAKEROOT_LIBFAKEROOT
 
 #ifdef __APPLE__
