--- firejail-0.9.66/src/firejail/main.c.orig
+++ firejail-0.9.66/src/firejail/main.c
@@ -32,7 +32,7 @@
 #include <dirent.h>
 #include <pwd.h>
 #include <errno.h>
-//#include <limits.h>
+#include <limits.h>
 #include <sys/file.h>
 #include <sys/prctl.h>
 #include <signal.h>
--- firejail-0.9.66/src/firejail/env.c.orig
+++ firejail-0.9.66/src/firejail/env.c
@@ -22,6 +22,7 @@
 #include <sys/stat.h>
 #include <unistd.h>
 #include <dirent.h>
+#include <limits.h>
 
 typedef struct env_t {
 	struct env_t *next;
