--- src/DkGui/DkPong.cpp.orig	2016-04-28 11:31:15.000000000 +0000
+++ src/DkGui/DkPong.cpp	2016-12-30 18:51:24.147034114 +0000
@@ -655,7 +655,7 @@
 	double angle = mDirection.angle();
 	double range = CV_PI / 5.0;
 	double sign = angle > 0 ? 1.0 : -1.0;
-	angle = abs(angle);
+	angle = std::abs(angle);
 	double newAngle = 0.0;
 
 	if (angle < CV_PI*0.5 && angle > CV_PI*0.5 - range) {
--- src/DkGui/DkViewPort.cpp.orig	2016-04-28 11:31:15.000000000 +0000
+++ src/DkGui/DkViewPort.cpp	2016-12-30 18:54:36.602044043 +0000
@@ -1251,7 +1251,7 @@
 	mAnimationValue = 1.0f-(float)(mAnimationTime.elapsed()/1000.0)/Settings::param().display().animationDuration;
 	
 	// slow in - slow out
-	double speed = mAnimationValue > 0.5 ? abs(1.0 - mAnimationValue) : abs(mAnimationValue);
+	double speed = mAnimationValue > 0.5 ? std::abs(1.0 - mAnimationValue) : std::abs(mAnimationValue);
 	speed *= .05;
 
 	mAnimationValue += (float)speed;
