This macro cannot be normally defined in cross.

--- a/src/fbdev.c	2019-04-30 17:41:01.660203681 +0200
+++ b/src/fbdev.c	2019-04-30 17:41:07.756263574 +0200
@@ -1010,7 +1010,6 @@
 	fPtr->CloseScreen = pScreen->CloseScreen;
 	pScreen->CloseScreen = FBDevCloseScreen;
 
-#if XV
 	{
 	    XF86VideoAdaptorPtr *ptr;
 
@@ -1019,7 +1018,6 @@
 		xf86XVScreenInit(pScreen,ptr,n);
 	    }
 	}
-#endif
 
 	TRACE_EXIT("FBDevScreenInit");
 
