Fix regression tests on musl libc
index 19fece83f5..641e1e6de2 100644
--- a/openttd-13.4/cmake/scripts/Regression.cmake
+++ b/openttd-13.4/cmake/scripts/Regression.cmake
@@ -53,6 +53,7 @@ endif()
 string(REPLACE "0x(nil)" "0x00000000" REGRESSION_RESULT "${REGRESSION_RESULT}")
 string(REPLACE "0x0000000000000000" "0x00000000" REGRESSION_RESULT "${REGRESSION_RESULT}")
 string(REPLACE "0x0x0" "0x00000000" REGRESSION_RESULT "${REGRESSION_RESULT}")
+string(REPLACE "(null : 0x0)" "(null : 0x00000000)" REGRESSION_RESULT "${REGRESSION_RESULT}")
 
 # Remove timestamps if any
 string(REGEX REPLACE "\[[0-9-]+ [0-9:]+\] " "" REGRESSION_RESULT "${REGRESSION_RESULT}")
