There is no <execinfo.h> in musl libc

--- a/libvtv/vtv_rts.cc
+++ b/libvtv/vtv_rts.cc
@@ -125,7 +125,7 @@
 #include <windows.h>
 #include <winternl.h>
 #include <psapi.h>
-#else
+#elif defined (__GLIBC__)
 #include <execinfo.h>
 #endif
 
--- a/libvtv/vtv_utils.cc
+++ b/libvtv/vtv_utils.cc
@@ -35,7 +35,7 @@
 #if defined (__CYGWIN__) || defined (__MINGW32__)
 #define WIN32_LEAN_AND_MEAN
 #include <windows.h>
-#else
+#elif defined (__GLIBC__)
 #include <execinfo.h>
 #endif
 
@@ -179,7 +179,7 @@ __vtv_log_verification_failure (const ch
 
   __vtv_add_to_log (vtv_failures_log_fd, "%s", log_msg);
 
-#if !defined (__CYGWIN__) && !defined (__MINGW32__)
+#if !defined (__CYGWIN__) && !defined (__MINGW32__) && defined(__GLIBC__)
   if (generate_backtrace)
     {
 #define STACK_DEPTH 20
