--- configure	2011-12-18 20:03:44.000000000 +1000
+++ configure	2012-02-03 17:55:14.885990135 +1000
@@ -6049,8 +6049,8 @@
     LDFLAGS="${_cloog_saved_LDFLAGS} ${clooglibs} ${ppllibs}"

     if test "${cloog_org}" = yes ; then
-      { $as_echo "$as_me:${as_lineno-$LINENO}: checking for version 0.16.1 of CLooG" >&5
-$as_echo_n "checking for version 0.16.1 of CLooG... " >&6; }
+      { $as_echo "$as_me:${as_lineno-$LINENO}: checking for version 0.17 of CLooG" >&5
+$as_echo_n "checking for version 0.17 of CLooG... " >&6; }
 if test "${gcc_cv_cloog_ct_0_14_0+set}" = set; then :
   $as_echo_n "(cached) " >&6
 else
@@ -6061,8 +6061,7 @@
 main ()
 {
 #if CLOOG_VERSION_MAJOR != 0 \
-    || CLOOG_VERSION_MINOR != 16 \
-    || CLOOG_VERSION_REVISION < 1
+    || CLOOG_VERSION_MINOR != 17
     choke me
    #endif
   ;
