--- a/Makefile.in
+++ b/Makefile.in
@@ -54363,7 +54363,7 @@ configure-target-libphobos:
 	module_srcdir=libphobos; \
 	rm -f no-such-file || : ; \
 	GDCFLAGS="$(GDCFLAGS_FOR_TARGET)" \
-	CONFIG_SITE=no-such-file $(SHELL) \
+	CONFIG_SITE=no-such-file LIBS="$(LIBUCONTEXT) $$LIBS" $(SHELL) \
 	  $$s/$$module_srcdir/configure \
 	  --srcdir=$${topdir}/$$module_srcdir \
 	  $(TARGET_CONFIGARGS) --build=${build_alias} --host=${target_alias} \
--- a/libphobos/configure
+++ b/libphobos/configure
@@ -15383,6 +15383,11 @@ fi
       druntime_fiber_asm_external=yes
       ;;
   esac
+  case "$target" in
+    *-musl)
+      druntime_fiber_asm_external=no
+      ;;
+  esac
   if test "$druntime_fiber_asm_external" = no; then
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking for library containing swapcontext" >&5
 $as_echo_n "checking for library containing swapcontext... " >&6; }
--- a/libphobos/m4/druntime/libraries.m4
+++ b/libphobos/m4/druntime/libraries.m4
@@ -228,6 +228,11 @@ AC_DEFUN([DRUNTIME_LIBRARIES_UCONTEXT],
       druntime_fiber_asm_external=yes
       ;;
   esac
+  case "$target" in
+    *-musl)
+      druntime_fiber_asm_external=no
+      ;;
+  esac
   if test "$druntime_fiber_asm_external" = no; then
     AC_SEARCH_LIBS([swapcontext], [c ucontext], [],
       AC_MSG_ERROR([swapcontext required but not found]))
