--- a/libphobos/libdruntime/core/sys/posix/time.d
+++ b/libphobos/libdruntime/core/sys/posix/time.d
@@ -473,8 +473,12 @@ else version (CRuntime_Bionic)
 }
 else version (CRuntime_Musl)
 {
-    static assert(timespec.sizeof == 16);
-
+    static assert(timespec.sizeof == 2 * time_t.sizeof);
+    version (CRuntime_Musl_Pre_Time64) {
+        static assert(time_t.sizeof == c_long.sizeof);
+    } else {
+        static assert(time_t.sizeof == 8);
+    }
     alias clockid_t = int;
     alias timer_t = void*;
 
