--- a/libphobos/libdruntime/core/sys/posix/dirent.d
+++ b/libphobos/libdruntime/core/sys/posix/dirent.d
@@ -368,6 +368,7 @@ else version (CRuntime_Bionic)
 }
 else version (CRuntime_Musl)
 {
+    // NOTE: marker for diff
     enum
     {
         DT_UNKNOWN  = 0,
@@ -385,15 +386,7 @@ else version (CRuntime_Musl)
     {
     }
 
-    static if ( __USE_FILE_OFFSET64 )
-    {
-        dirent* readdir64(DIR*);
-        alias   readdir64 readdir;
-    }
-    else
-    {
-        dirent* readdir(DIR*);
-    }
+    dirent* readdir(DIR*);
 }
 else version (CRuntime_UClibc)
 {
--- a/libphobos/libdruntime/core/sys/posix/sys/mman.d
+++ b/libphobos/libdruntime/core/sys/posix/sys/mman.d
@@ -293,11 +293,7 @@ else version (CRuntime_Bionic)
 }
 else version (CRuntime_Musl)
 {
-    static if (__USE_LARGEFILE64) void* mmap64(void*, size_t, int, int, int, off_t);
-    static if (__USE_FILE_OFFSET64)
-        alias mmap = mmap64;
-    else
-        void* mmap(void*, size_t, int, int, int, off_t);
+    void* mmap(void*, size_t, int, int, int, off_t);
     int munmap(void*, size_t);
 }
 else version (CRuntime_UClibc)
--- a/libphobos/libdruntime/core/sys/posix/sys/types.d
+++ b/libphobos/libdruntime/core/sys/posix/sys/types.d
@@ -111,6 +111,7 @@ version (linux)
 
     version (CRuntime_Musl)
     {
+        static assert(off_t.sizeof == 8);
         /**
          * Musl versions before v1.2.0 (up to v1.1.24) had different
          * definitions for `time_t` for 32 bits.
--- a/libphobos/libdruntime/core/sys/posix/stdio.d
+++ b/libphobos/libdruntime/core/sys/posix/stdio.d
@@ -182,10 +182,9 @@ else version (CRuntime_UClibc)
 }
 else version (CRuntime_Musl)
 {
-    static if ( __USE_FILE_OFFSET64 )
+    static if (false)
     {
         int   fgetpos64(FILE*, fpos_t *);
-        alias fgetpos64 fgetpos;
 
         FILE* fopen64(const scope char*, const scope char*);
         alias fopen64 fopen;
@@ -320,26 +319,8 @@ else version (CRuntime_Musl)
 {
     enum L_ctermid = 20;
 
-    static if ( __USE_FILE_OFFSET64 )
-    {
-        int   fseeko64(FILE*, off_t, int);
-        alias fseeko64 fseeko;
-    }
-    else
-    {
-        int   fseeko(FILE*, off_t, int);
-    }
-
-    static if ( __USE_FILE_OFFSET64 )
-    {
-        off_t ftello64(FILE*);
-        alias ftello64 ftello;
-    }
-    else
-    {
-        off_t ftello(FILE*);
-    }
-
+    int   fseeko(FILE*, off_t, int);
+    off_t ftello(FILE*);
     ssize_t getdelim(char**, size_t*, int, FILE*);
     ssize_t getline(char**, size_t*, FILE*);
 }
--- a/libphobos/libdruntime/core/sys/posix/stdlib.d
+++ b/libphobos/libdruntime/core/sys/posix/stdlib.d
@@ -604,16 +604,6 @@ else version (CRuntime_Musl)
     void   srandom(uint);
     int    unlockpt(int);
 
-  static if ( __USE_LARGEFILE64 )
-  {
-    int    mkstemp64(char*);
-    alias  mkstemp64 mkstemp;
-  }
-  else
-  {
-    int    mkstemp(char*);
-  }
-
 }
 else version (Solaris)
 {
