--- a/libffi/src/mips/o32.S	2020-03-12 12:07:23.000000000 +0100
+++ b/libffi/src/mips/o32.S	2020-05-14 20:52:45.701809054 +0200
@@ -278,6 +278,7 @@
 	li	$9, FFI_TYPE_VOID
 	beq	$8, $9, closure_done
 
+#ifndef __mips_soft_float
 	li	$13, 1		# FFI_O32
 	bne	$16, $13, 1f	# Skip fp restore if FFI_O32_SOFT_FLOAT
 
@@ -289,6 +290,7 @@
 	l.d	$f0, V0_OFF2($fp)
 	beq	$8, $9, closure_done
 1:	
+#endif
 	REG_L	$3, V1_OFF2($fp)
 	REG_L	$2, V0_OFF2($fp)
 
