--- src/bezier.cpp.orig	2014-08-11 18:08:01.000000000 +0000
+++ src/bezier.cpp	2017-09-22 19:42:14.914232067 +0000
@@ -463,7 +463,7 @@
 
 void Bezier::ReadFrom(std::istream &openfile)
 {
-	assert(openfile);
+	assert(static_cast<bool>(openfile));
 
 	for (int x = 0; x < 4; x++)
 	{
@@ -480,7 +480,7 @@
 
 void Bezier::ReadFromYZX(std::istream &openfile)
 {
-	assert(openfile);
+	assert(static_cast<bool>(openfile));
 	for (int x = 0; x < 4; x++)
 	{
 		for (int y = 0; y < 4; y++)
@@ -494,7 +494,7 @@
 
 void Bezier::WriteTo(std::ostream &openfile) const
 {
-	assert(openfile);
+	assert(static_cast<bool>(openfile));
 
 	for (int x = 0; x < 4; x++)
 	{
--- src/roadstrip.cpp.orig	2014-08-01 06:18:06.000000000 +0000
+++ src/roadstrip.cpp	2017-09-22 19:44:11.794238097 +0000
@@ -31,7 +31,7 @@
 	bool reverse,
 	std::ostream & error_output)
 {
-	assert(openfile);
+	assert(static_cast<bool>(openfile));
 
 	int num = 0;
 	openfile >> num;
