--- perl-5.40/cnf/configure_misc.sh
+++ perl-5.40/cnf/configure_misc.sh
@@ -240,11 +240,11 @@
 define stdio_stream_array
 define d_lc_monetary_2008 'undef'
 
-define d_perl_lc_all_uses_name_value_pairs 'define'
-define d_perl_lc_all_category_positions_init 'undef'
-define d_perl_lc_all_separator 'undef'
-define perl_lc_all_separator ''
-define perl_lc_all_category_positions_init ''
+define d_perl_lc_all_uses_name_value_pairs 'undef'
+define d_perl_lc_all_category_positions_init 'define'
+define d_perl_lc_all_separator 'define'
+define perl_lc_all_separator '";"'
+define perl_lc_all_category_positions_init '{0,1,2,3,4,5}'
 
 define sharpbang '#!'
 define startperl "$sharpbang$perlpath"
