--- a/higan/target-higan/GNUmakefile	2020-03-20 08:59:06.000000000 +0100
+++ b/higan/target-higan/GNUmakefile	2021-09-09 10:58:54.370975389 +0200
@@ -37,8 +37,6 @@
 ifeq ($(platform),windows)
 	mkdir -p $(prefix)/$(name)/
 	cp -R System/* $(prefix)/$(name)/
-else ifeq ($(shell id -un),root)
-	$(error "make install should not be run as root")
 else ifeq ($(platform),macos)
 	mkdir -p ~/Library/Application\ Support\$(name)/
 	mkdir -p ~/Library/Application\ Support\$(name)/System/
--- a/icarus/GNUmakefile	2020-03-20 08:59:06.000000000 +0100
+++ b/icarus/GNUmakefile	2021-09-09 12:11:52.597265710 +0200
@@ -46,8 +46,6 @@
 	mkdir -p $(prefix)/$(name)/Firmware/
 	cp -R Database/* $(prefix)/$(name)/Database/
 	cp -R Firmware/* $(prefix)/$(name)/Firmware/
-else ifeq ($(shell id -un),root)
-	$(error "make install should not be run as root")
 else ifeq ($(platform),macos)
 	cp -R out/$(name).app /Applications/$(name).app
 else ifneq ($(filter $(platform),linux bsd),)
