commit b26d28d27e7f426b202da82bf0390b7f07469a44
Author: Daniel Martinez <danielmartinez@cock.li>
Date:   Sun Apr 19 11:43:48 2026 -0400

    Only depend on cpufeatures on supported architechtures.

diff --git a/common/Cargo.toml b/common/Cargo.toml
index 9a1ff61..94b9157 100644
--- a/common/Cargo.toml
+++ b/common/Cargo.toml
@@ -23,6 +23,7 @@ rustix = { version = "1.1", default-features = false, features = [
   "thread",
   "time",
 ] }
+[target.'cfg(any(target_arch = "aarch64", target_arch = "x86_64", target_arch = "x86"))'.dependencies]
 cpufeatures = { version = "0.3" }
 
 [build-dependencies]
