diff --git a/irc/Cargo.toml b/irc/Cargo.toml
index 19013b0..62afba3 100644
--- a/irc/Cargo.toml
+++ b/irc/Cargo.toml
@@ -16,7 +16,7 @@ tokio-rustls = { version = "0.26.0", default-features = false, features = ["tls1
 tokio-util = { version = "0.7", features = ["codec"] }
 rustls-native-certs = "0.7.0"
 rustls-pemfile = "2.1.1"
-xz2 = { version = "0.1.7", features = ["static"] }
+xz2 = "0.1.7"
 
 [dependencies.proto]
 path = "proto"
