diff --git a/Cargo.toml b/Cargo.toml
index 32020be..5d69318 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -40,7 +40,7 @@ palette = "0.7.4"
 log = { version = "0.4.26", features = ['std'] }
 uptime_lib = "0.3.1"
 nucleo-matcher = "0.3.1"
-reqwest = { version = "0.13", features = ["json", "socks"] }
+reqwest = { version = "0.13", default-features = false, features = ["json", "socks", "charset", "http2", "system-proxy", "native-tls"] }
 
 [dependencies]
 tokio = { workspace = true, features = ["rt", "fs", "process"] }
