diff --git a/Makefile b/Makefile
index 43c4227a..9ca8587b 100644
--- a/Makefile
+++ b/Makefile
@@ -48,11 +48,10 @@ HARE_BUILD_ENV = HAREPATH=. HAREC="$(HAREC)" QBE="$(QBE)" AS="$(AS)" \
 	LD="$(LD)" HAREFLAGS="$(HAREFLAGS)" HARECFLAGS="$(HARECFLAGS)" \
 	QBEFLAGS="$(QBEFLAGS)" ASFLAGS="$(ASFLAGS)" LDLINKFLAGS="$(LDLINKFLAGS)"
 
-$(BINOUT)/haredoc: $(BINOUT)/hare
+$(BINOUT)/haredoc:
 	@mkdir -p $(BINOUT)
 	@printf 'HARE\t%s\n' "$@"
-	@env $(HARE_BUILD_ENV) \
-		$(BINOUT)/hare build $(HARE_DEFINES) -o $(BINOUT)/haredoc ./cmd/haredoc
+	@hare build $(HARE_DEFINES) $(HAREFLAGS) -o $(BINOUT)/haredoc ./cmd/haredoc
 
 docs/html: $(BINOUT)/haredoc
 	mkdir -p docs/html
