diff --git a/Makefile b/Makefile
index 2482be1f..0f86869e 100644
--- a/Makefile
+++ b/Makefile
@@ -52,11 +52,10 @@ $(BINOUT)/harec2: $(BINOUT)/hare
 		LDLINKFLAGS="$(LDLINKFLAGS)" \
 		$(BINOUT)/hare build $(HARE_DEFINES) -o $(BINOUT)/harec2 cmd/harec
 
-$(BINOUT)/haredoc: $(BINOUT)/hare
+$(BINOUT)/haredoc:
 	@mkdir -p $(BINOUT)
 	@printf 'HARE\t%s\n' "$@"
-	@env HAREPATH=. HAREC="$(HAREC)" QBE="$(QBE)" $(BINOUT)/hare build \
-		$(HARE_DEFINES) -o $(BINOUT)/haredoc ./cmd/haredoc
+	@hare build $(HAREFLAGS) -o $(BINOUT)/haredoc ./cmd/haredoc
 
 docs/html: $(BINOUT)/haredoc
 	mkdir -p docs/html
