--- a/database.c	2026-01-26 11:04:38.000000000 +0100
+++ b/database.c	2026-03-01 02:11:46.310427997 +0000
@@ -12,6 +12,7 @@
 #include <string.h>
 #include <unistd.h>
 #include <assert.h>
+#include <ctype.h>
 #ifdef HAVE_CONFIG_H
 #	include "config.h"
 #endif
@@ -737,7 +738,7 @@
 {
 	char* p = str;
 	while (*str) {
-		if (isalnum (*str)) *(p++) = *str;
+		if (isalnum((unsigned char)*str)) *(p++) = *str;
 		++str;
 	}
 	*p = 0;
