diff -ur database.h database.h
--- a/database.h	2015-10-04 10:38:25.000000000 +0800
+++ b/database.h	2018-02-15 21:11:46.311673251 +0800
@@ -61,7 +61,7 @@
 /*
  * Field operations
  */
-inline int field_id(int i);
+int field_id(int i);
 abook_field *find_standard_field(char *key, int do_declare);
 abook_field *real_find_field(char *key, abook_field_list *list, int *nb);
 #define find_field(key, list)		real_find_field(key, list, NULL)
