diff --git fpcsrc/rtl/linux/i386/si_c21.inc fpcsrc/rtl/linux/i386/si_c21.inc
index 71df29b6..222004d9 100644
--- fpcbuild-3.2.0/fpcsrc/rtl/linux/i386/si_c21.inc
+++ fpcbuild-3.2.0/fpcsrc/rtl/linux/i386/si_c21.inc
@@ -40,6 +40,9 @@ procedure __libc_csu_fini; cdecl; external;
 procedure libc_start_main; external name '__libc_start_main';
 procedure libc_exit(code: longint); cdecl; external name 'exit';
 
+procedure ini_dummy;
+  begin
+  end;
 
 {******************************************************************************
                        glibc 2.1 lib + profiling start/halt
@@ -81,20 +84,20 @@ asm
   {$endif FPC_PIC}
 
         { int __libc_start_main(
-		int *(main) (int, char * *, char * *), 
-		int argc, 
-		char * * ubp_av, 
-		void (*init) (void), 
-		void (*fini) (void), 
-		void (*rtld_fini) (void), 
-		void (* stack_end)); } 
+		int *(main) (int, char * *, char * *),
+		int argc,
+		char * * ubp_av,
+		void (*init) (void),
+		void (*fini) (void),
+		void (*rtld_fini) (void),
+		void (* stack_end)); }
 
         pushl %ebp   			{ padding }
         pushl %esp   			{ stack_end }
         pushl %edx   			{ function to be registered with
                       			  atexit(), passed by loader }
-        pushl $__libc_csu_fini
-        pushl $__libc_csu_init
+        pushl $ini_dummy
+        pushl $ini_dummy
         pushl %esi                      { Push second argument: argv.  }
         pushl %ecx                      { Push first argument: argc.  }
 
