commit 5120237db67c066764563df36fcf171699ff7c6f
Author: q66 <daniel@octaforge.org>
Date:   Mon Apr 5 01:50:14 2021 +0200

    always use elfv2 abi by default on ppc64

diff --git fpcsrc/compiler/options.pas fpcsrc/compiler/options.pas
index 2cc36670..c35cc785 100644
--- fpcbuild-3.2.0/fpcsrc/compiler/options.pas
+++ fpcbuild-3.2.0/fpcsrc/compiler/options.pas
@@ -4299,13 +4299,8 @@ begin
      likely to fail in spectacular ways" }
   if not option.ABISetExplicitly then
     begin
-      if (target_info.abi=abi_powerpc_sysv) and
-         (target_info.endian=endian_little) then
+      if (target_info.abi=abi_powerpc_sysv) then
         target_info.abi:=abi_powerpc_elfv2
-      else
-        if (target_info.abi=abi_powerpc_elfv2) and
-         (target_info.endian=endian_big) then
-        target_info.abi:=abi_powerpc_sysv
     end;
 {$endif}
 
