--- a/configure	2016-09-29 15:59:30.000000000 +0200
+++ b/configure	2017-02-14 01:48:10.874916913 +0100
@@ -23873,6 +23873,14 @@
                  POSTGRESQL_LIBDIR=/usr/lib/postgresql
               fi
               POSTGRESQL_BINDIR=/usr/bin
+          elif test -f ${XBPS_CROSS_BASE}/usr/include/libpq-fe.h; then
+              POSTGRESQL_INCDIR=${XBPS_CROSS_BASE}/usr/include
+              if test -d ${XBPS_CROSS_BASE}/usr/lib64; then
+                 POSTGRESQL_LIBDIR=${XBPS_CROSS_BASE}/usr/lib64
+              else
+                 POSTGRESQL_LIBDIR=${XBPS_CROSS_BASE}/usr/lib
+              fi
+              POSTGRESQL_BINDIR=${XBPS_CROSS_BASE}/usr/bin
           else
               { $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
 $as_echo "no" >&6; }
@@ -24182,6 +24190,15 @@
                  MYSQL_LIBDIR=/usr/local/lib
               fi
               MYSQL_BINDIR=/usr/local/bin
+           elif test -f ${XBPS_CROSS_BASE}/usr/include/mysql/mysql.h; then
+              MYSQL_INCDIR=${XBPS_CROSS_BASE}/usr/include/mysql
+              if test -f ${XBPS_CROSS_BASE}/usr/lib64/libmysqlclient_r.a \
+                      -o -f ${XBPS_CROSS_BASE}/usr/lib64/libmysqlclient_r.so; then
+                 MYSQL_LIBDIR=${XBPS_CROSS_BASE}/usr/lib64
+              else
+                 MYSQL_LIBDIR=${XBPS_CROSS_BASE}/usr/lib
+              fi
+              MYSQL_BINDIR=${XBPS_CROSS_BASE}/usr/bin
            else
               { $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
 $as_echo "no" >&6; }
@@ -24587,6 +24604,14 @@
               SQLITE_LIBDIR=$prefix/lib
            fi
            SQLITE_BINDIR=$prefix/bin
+        elif test -f ${XBPS_CROSS_BASE}/usr/include/sqlite3.h; then
+           SQLITE_INCDIR=${XBPS_CROSS_BASE}/usr/include
+           if test -d ${XBPS_CROSS_BASE}/usr/lib64; then
+              SQLITE_LIBDIR=${XBPS_CROSS_BASE}/usr/lib64
+           else
+              SQLITE_LIBDIR=${XBPS_CROSS_BASE}/usr/lib
+           fi
+           SQLITE_BINDIR=${XBPS_CROSS_BASE}/usr/bin
         else
            { $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
 $as_echo "no" >&6; }
