# gethostbyname returns a nullptr, because the host can't resolve itself

--- a/src/OSD/OSD_Host.cxx
+++ b/src/OSD/OSD_Host.cxx
@@ -124,7 +124,7 @@ TCollection_AsciiString OSD_Host::InternetAddress(){
 
  host = HostName();
  memcpy(&internet_address,
-        gethostbyname(host.ToCString()),
+        gethostbyname("localhost"),
         sizeof(struct hostent));
 
  // Gets each bytes into integers
