# gethostbyname returns a nullptr, because the host can't resolve itself

--- a/src/OSD/OSD_Host.cxx
+++ b/src/OSD/OSD_Host.cxx
@@ -140,7 +140,7 @@ TCollection_AsciiString OSD_Host::InternetAddress(){
   TCollection_AsciiString result, host;
 
   host = HostName();
-  memcpy(&internet_address, gethostbyname(host.ToCString()), sizeof(struct hostent));
+  memcpy(&internet_address, gethostbyname("localhost"), sizeof(struct hostent));
 
   // Gets each bytes into integers
   a = (unsigned char)internet_address.h_addr_list[0][0];
