--- a/cmd/perl/ssh-last/ssh-last	2026-04-30 20:55:20.000000000 +0000
+++ b/cmd/perl/ssh-last/ssh-last	2026-04-30 21:26:12.260751568 +0000
@@ -452,6 +452,10 @@
                            . ' /var/log/messages'
                            . ' 2>/dev/null';
 
+my $log_cmd_files_void     = 'grep -hE "Accepted|Disconnected"'
+                           . ' /var/log/socklog/secure/current'
+                           . ' 2>/dev/null';
+
 my $log_cmd_files_openbsd  = 'zgrep -hE "Accepted|Disconnected"'
                            . ' /var/log/authlog.6.gz'
                            . ' /var/log/authlog.5.gz'
@@ -753,12 +757,13 @@
 
     if ( $os_data{ID} eq 'void' ) {
 
-        $log_cmd        = $log_cmd_files_messages;
+        $log_cmd        = $log_cmd_files_void;
 
         $matches_login  = '^(?<TS>\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2})'
                         . '\.(?<IGNORE>\S+)'
                         . '\s+(?<HOSTNAME>\S+)'
-                        . '\s+sshd\[(?<PID>\d+)\]:'
+                        . '.*'
+                        . '\s+sshd(?:-session)?\[(?<PID>\d+)\]:'
                         . '\s+Accepted\s+(?<AUTH_TYPE>\S+)'
                         . '\s+for\s+(?<USER>\S+)'
                         . '\s+from\s+(?<HOST>\S+)'
@@ -768,7 +773,8 @@
         $matches_logout = '^(?<TS>\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2})'
                         . '\.(?<IGNORE>\S+)'
                         . '\s+(?<HOSTNAME>\S+)'
-                        . '\s+sshd\[(?<PID>\d+)\]:'
+                        . '.*'
+                        . '\s+sshd(?:-session)?\[(?<PID>\d+)\]:'
                         . '\s+Disconnected'
                         . '\s+from\s+user\s+(?<USER>\S+)'
                         . '\s+(?<HOST>\S+)'
