From 1bbfda77ea96fb8399a4c12c0c096a52371e5864 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Piotr=20W=C3=B3jcik?= <chocimier@tlen.pl>
Date: Wed, 11 Mar 2020 23:02:33 +0100
Subject: [PATCH] disable xar test


diff --git autotests/kerfuffle/extracttest.cpp autotests/kerfuffle/extracttest.cpp
index 9178611..c1d3f13 100644
--- a/autotests/kerfuffle/extracttest.cpp
+++ b/autotests/kerfuffle/extracttest.cpp
@@ -350,23 +350,6 @@ void ExtractTest::testExtraction_data()
         qDebug() << "lz4 executable not found in path. Skipping lz4 test.";
     }

-    archivePath = QFINDTESTDATA("data/simplearchive.xar");
-    QTest::newRow("extract selected entries from a xar archive without path")
-            << archivePath
-            << QVector<Archive::Entry*> {
-                   new Archive::Entry(this, QStringLiteral("dir1/file11.txt"), QString()),
-                   new Archive::Entry(this, QStringLiteral("file4.txt"), QString())
-               }
-            << optionsNoPaths
-            << 2;
-
-    archivePath = QFINDTESTDATA("data/simplearchive.xar");
-    QTest::newRow("extract all entries from a xar archive with path")
-            << archivePath
-            << QVector<Archive::Entry*>()
-            << optionsPreservePaths
-            << 6;
-
     archivePath = QFINDTESTDATA("data/hello-1.0-x86_64.AppImage");
     QTest::newRow("extract all entries from an AppImage with path")
             << archivePath
diff --git autotests/kerfuffle/loadtest.cpp autotests/kerfuffle/loadtest.cpp
index 8c5f52d..cbb47cf 100644
--- a/autotests/kerfuffle/loadtest.cpp
+++ b/autotests/kerfuffle/loadtest.cpp
@@ -181,13 +181,6 @@ void LoadTest::testProperties_data()
         qDebug() << "lz4 executable not found in path. Skipping lz4 test.";
     }

-    QTest::newRow("xar archive")
-            << QFINDTESTDATA("data/simplearchive.xar")
-            << QStringLiteral("simplearchive")
-            << true << false << false << false << false << 0 << Archive::Unencrypted
-            << QStringLiteral("simplearchive")
-            << QString();
-
     QTest::newRow("mimetype child of application/zip")
             << QFINDTESTDATA("data/test.odt")
             << QStringLiteral("test")
--
2.23.0

