fix https://invent.kde.org/utilities/okteta/-/issues/1

--- kasten/controllers/CMakeLists.txt.ORIG	2020-07-28 23:41:01.000000000 +0200
+++ kasten/controllers/CMakeLists.txt	2020-11-01 11:15:35.947523996 +0100
@@ -583,14 +583,29 @@
     test/scriptvalueconvertertest.cpp
     test/osdparsertest.cpp
     test/commonparsertest.cpp
-    test/jsparsertest.cpp
-    test/customtostringtest.cpp
     test/locktooffsettest.cpp
 
     NAME_PREFIX structures-
     LINK_LIBRARIES structureview-tested-srcs Qt5::Test poddecoder-typeeditors
   )
 
+  #these tests are using QFINDTESTDATA, that does not work since managed by
+  #CMake from this parent directory: WORKING_DIRECTORY must be set accordindly
+  add_executable(jsparsertest test/jsparsertest.cpp)
+  add_test(
+    NAME structures-jsparsertest
+    COMMAND jsparsertest
+    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/test
+  )
+  target_link_libraries(jsparsertest structureview-tested-srcs Qt5::Test poddecoder-typeeditors)
+  add_executable(customtostringtest test/customtostringtest.cpp)
+  add_test(
+    NAME structures-customtostringtest
+    COMMAND customtostringtest
+    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/test
+  )
+  target_link_libraries(customtostringtest structureview-tested-srcs Qt5::Test poddecoder-typeeditors)
+
   #replace tool unit tests
   ecm_add_test(
     view/search/searchjob.cpp
