--- src/netpgpverify/verify.c.orig	2010-05-04 01:00:46.490355473 +0200
+++ src/netpgpverify/verify.c	2010-03-14 00:30:41.000000000 +0100
@@ -569,6 +569,7 @@ typedef struct __ops_ptag_t {
  * \see RFC4880 9.1
  */
 typedef enum {
+	OPS_PKA_NOTHING = 0,
 	OPS_PKA_RSA = 1,	/* RSA (Encrypt or Sign) */
 	OPS_PKA_RSA_ENCRYPT_ONLY = 2,	/* RSA Encrypt-Only (deprecated -
 					 * \see RFC4880 13.5) */
@@ -5551,7 +5550,7 @@ __ops_pubkey_free(__ops_pubkey_t *p)
 		free_BN(&p->key.elgamal.y);
 		break;
 
+	case OPS_PKA_NOTHING:
-	case 0:
 		/* nothing to free */
 		break;
 
