--- src/lib/validate.c.orig	2010-05-04 02:07:27.512355384 +0200
+++ src/lib/validate.c	2010-05-04 02:08:47.038371812 +0200
@@ -814,7 +814,8 @@ __ops_validate_file(__ops_io_t *io,
 	sigsize = st.st_size;
 	detachname = NULL;
 	cc = snprintf(origfile, sizeof(origfile), "%s", infile);
-	if (strcmp(&origfile[cc - 4], ".sig") == 0) {
+	if ((strcmp(&origfile[cc - 4], ".sig") == 0) ||
+	    (strcmp(&origfile[cc - 4], ".asc") == 0)) {
 		origfile[cc - 4] = 0x0;
 		if (stat(origfile, &st) == 0 &&
 		    st.st_size > sigsize - SIG_OVERHEAD) {
