From 4d5fb96252d289e7899bc0bb87262127d19de949 Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Tue, 07 May 2013 10:47:30 +0000
Subject: login: add missing _public_ to sd_get_machine_names()

---
diff --git a/src/login/sd-login.c b/src/login/sd-login.c
index 66c4487..d0dc42f 100644
--- src/login/sd-login.c
+++ src/login/sd-login.c
@@ -591,7 +591,7 @@ _public_ int sd_get_uids(uid_t **users) {
         return r;
 }
 
-int sd_get_machine_names(char ***machines) {
+_public_ int sd_get_machine_names(char ***machines) {
         _cleanup_closedir_ DIR *d = NULL;
         _cleanup_strv_free_ char **l = NULL;
         _cleanup_free_ char *md = NULL;
--
cgit v0.9.0.2-2-gbebe
