--- a/meson.build
+++ b/meson.build
@@ -1755,6 +1755,14 @@
                                 '/usr/include/@0@'.format(target_triplet)
                         ]
                 endif
+        else
+                sysroot = run_command(meson.get_compiler('c').cmd_array() + ['-print-sysroot']).stdout().split('\n')[0]
+                if sysroot == ''
+                         error('sysroot not found')
+                endif
+                bpf_o_unstripped_cmd += [
+                        '--sysroot=@0@'.format(sysroot),
+                ]
         endif
 
         bpf_o_unstripped_cmd += [
@@ -2685,6 +2693,15 @@
                 install : dbus_interfaces_dir != 'no',
                 install_dir : fs.parent(dbus_interfaces_dir),
                 command : [export_dbus_interfaces_py, '@OUTPUT@', dbus_programs])
+else
+        qemu_arch = {}.get(host_machine.cpu_family(), host_machine.cpu_family())
+        custom_target(
+                'export-dbus-interfaces',
+                output : fs.name(dbus_interfaces_dir),
+                install : dbus_interfaces_dir != 'no',
+                install_dir : fs.parent(dbus_interfaces_dir),
+                command : [export_dbus_interfaces_py, '@OUTPUT@', dbus_programs],
+                env : ['QEMU=/usr/bin/qemu-@0@-static'.format(qemu_arch)])
 endif
 
 ############################################################
--- a/tools/dbus_exporter.py
+++ b/tools/dbus_exporter.py
@@ -3,11 +3,15 @@
 from argparse import ArgumentParser
 from pathlib import Path
 from subprocess import PIPE, run
+import os
 
 
 def extract_interfaces_xml(output_dir, executable):
+    wrap = []
+    if qemu := os.environ.get('QEMU'):
+        wrap.append(qemu)
     proc = run(
-        args=[executable.absolute(), '--bus-introspect', 'list'],
+        args=wrap + [executable.absolute(), '--bus-introspect', 'list'],
         stdout=PIPE,
         check=True,
         universal_newlines=True)
@@ -16,7 +20,7 @@
 
     for interface_name in interface_names:
         proc = run(
-            args=[executable.absolute(), '--bus-introspect', interface_name],
+            args=wrap + [executable.absolute(), '--bus-introspect', interface_name],
             stdout=PIPE,
             check=True,
             universal_newlines=True)
