From e2c8b07dcb50c2adf64cdfb22e4a496fc76576fb Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Fri, 21 Sep 2012 22:16:13 +0000
Subject: journal: bring mmap cache prototype in sync

---
diff --git a/src/journal/mmap-cache.h b/src/journal/mmap-cache.h
index de34634..0c42fb8 100644
--- src/journal/mmap-cache.h
+++ src/journal/mmap-cache.h
@@ -31,6 +31,6 @@ MMapCache* mmap_cache_new(void);
 MMapCache* mmap_cache_ref(MMapCache *m);
 MMapCache* mmap_cache_unref(MMapCache *m);
 
-int mmap_cache_get(MMapCache *m, int fd, int prot, unsigned context, bool keep_always, uint64_t offset, uint64_t size, struct stat *st, void **ret);
+int mmap_cache_get(MMapCache *m, int fd, int prot, unsigned context, bool keep_always, uint64_t offset, size_t size, struct stat *st, void **ret);
 void mmap_cache_close_fd(MMapCache *m, int fd);
 void mmap_cache_close_context(MMapCache *m, unsigned context);
--
cgit v0.9.0.2-2-gbebe
