--- configure.ac.orig	2017-10-11 07:45:18.000000000 +0000
+++ configure.ac	2017-10-11 07:49:25.955714462 +0000
@@ -30,12 +30,12 @@
 
 # Check version number coherency between RtMidi.h and configure.ac
 AC_MSG_CHECKING([that version numbers are coherent])
-AC_RUN_IFELSE(
-   [AC_LANG_PROGRAM([#include <string.h>
-                     `grep "define RTMIDI_VERSION" $srcdir/RtMidi.h`],
-                    [return strcmp(RTMIDI_VERSION, PACKAGE_VERSION);])],
-   [AC_MSG_RESULT([yes])],
-   [AC_MSG_FAILURE([testing RTMIDI_VERSION==PACKAGE_VERSION failed, check that RtMidi.h defines RTMIDI_VERSION as "$PACKAGE_VERSION" or that the first line of configure.ac has been updated.])])
+# AC_RUN_IFELSE(
+#    [AC_LANG_PROGRAM([#include <string.h>
+#                      `grep "define RTMIDI_VERSION" $srcdir/RtMidi.h`],
+#                     [return strcmp(RTMIDI_VERSION, PACKAGE_VERSION);])],
+#    [AC_MSG_RESULT([yes])],
+#    [AC_MSG_FAILURE([testing RTMIDI_VERSION==PACKAGE_VERSION failed, check that RtMidi.h defines RTMIDI_VERSION as "$PACKAGE_VERSION" or that the first line of configure.ac has been updated.])])
 
 # Enable some nice automake features if they are available
 m4_ifdef([AM_MAINTAINER_MODE], [AM_MAINTAINER_MODE])
