--- src/nvc.c	2019-09-05 01:45:00.000000000 +0200
+++ src/nvc.c	2019-12-13 17:35:30.692519318 +0100
@@ -189,13 +189,13 @@
                 }

                 log_info("loading kernel module nvidia");
-                if (nvidia_modprobe(0, -1) == 0)
+                if (nvidia_modprobe(0) == 0)
                         log_err("could not load kernel module nvidia");
                 else {
-                        if (nvidia_mknod(NV_CTL_DEVICE_MINOR, -1) == 0)
+                        if (nvidia_mknod(NV_CTL_DEVICE_MINOR) == 0)
                                 log_err("could not create kernel module device node");
                         for (int i = 0; i < (int)devs.num_matches; ++i) {
-                                if (nvidia_mknod(i, -1) == 0)
+                                if (nvidia_mknod(i) == 0)
                                         log_err("could not create kernel module device node");
                         }
                 }

