diff --git a/src/language.cpp b/src/language.cpp
index 8b7b0ee..9a97093 100644
--- src/language.cpp
+++ src/language.cpp
@@ -75,7 +75,7 @@ const char *Language::Translate(const NickCore *nc, const char *string)
 
 #if GETTEXT_FOUND
 
-#ifdef __USE_GNU_GETTEXT
+#if defined(__USE_GNU_GETTEXT) && defined(_nl_msg_cat_cntr)
 extern "C" int _nl_msg_cat_cntr;
 #endif
 
@@ -87,7 +87,7 @@ const char *Language::Translate(const char *lang, const char *string)
 	if (!lang || !*lang)
 		lang = Config->DefLanguage.c_str();
 
-#ifdef __USE_GNU_GETTEXT
+#if defined(__USE_GNU_GETTEXT) && defined(_nl_msg_cat_cntr)
 	++_nl_msg_cat_cntr;
 #endif
 

