--- plugins/python/pybind11/tools/FindPythonLibsNew.cmake
+++ plugins/python/pybind11/tools/FindPythonLibsNew.cmake
@@ -57,9 +57,9 @@ endif()
 
 # Use the Python interpreter to find the libs.
 if(PythonLibsNew_FIND_REQUIRED)
-    find_package(PythonInterp ${PythonLibsNew_FIND_VERSION} REQUIRED)
+	find_package(PythonLibs ${PythonLibsNew_FIND_VERSION} REQUIRED)
 else()
-    find_package(PythonInterp ${PythonLibsNew_FIND_VERSION})
+	find_package(PythonLibs ${PythonLibsNew_FIND_VERSION})
 endif()
 
 if(NOT PYTHONINTERP_FOUND)
