--- a/src/ptrace/ptrace.c	2014-12-15 15:18:11.000000000 +0100
+++ b/src/ptrace/ptrace.c	2015-07-12 13:13:45.970093333 +0200
@@ -28,6 +28,7 @@
 #include <sys/uio.h>    /* struct iovec, */
 #include <sys/param.h>  /* MIN(), MAX(), */
 #include <string.h>     /* memcpy(3), */
+#include <linux/wait.h> /* __WALL, __WCLONE */
 
 #include "ptrace/ptrace.h"
 #include "ptrace/user.h"
--- a/src/ptrace/wait.c	2014-12-15 15:18:11.000000000 +0100
+++ b/src/ptrace/wait.c	2015-07-12 13:16:41.930106019 +0200
@@ -26,6 +26,7 @@
 #include <stdbool.h>    /* bool, true, false, */
 #include <signal.h>     /* SIG*, */
 #include <talloc.h>     /* talloc*, */
+#include <linux/wait.h> /* __WALL, __WCLONE */
 
 #include "ptrace/wait.h"
 #include "ptrace/ptrace.h"
--- a/src/tracee/tracee.c	2014-12-15 15:18:11.000000000 +0100
+++ b/src/tracee/tracee.c	2015-07-12 13:20:01.058120375 +0200
@@ -31,6 +31,11 @@
 #include <signal.h>     /* kill(2), SIGKILL, */
 #include <sys/ptrace.h> /* ptrace(2), PTRACE_*, */
 #include <errno.h>      /* E*, */
+#include <linux/wait.h> /* __WALL, __WCLONE */
+
+#if !defined(__W_STOPCODE)
+#define __W_STOPCODE(sig) ((sig) << 8 | 0x7f)
+#endif
 
 #include "tracee/tracee.h"
 #include "tracee/reg.h"
