diff --git a/src/GNUmakefile b/src/GNUmakefile
index 24662e8..71f9353 100644
--- a/src/GNUmakefile
+++ b/src/GNUmakefile
@@ -8,12 +8,12 @@ SRC    = $(dir $(firstword $(MAKEFILE_LIST)))
 GIT        = git
 RM         = rm
 INSTALL    = install
-CC         = $(CROSS_COMPILE)gcc
+CC         ?= $(CROSS_COMPILE)gcc
 LD         = $(CC)
 LD_CMD     = $(CROSS_COMPILE)ld
 STRIP      = $(CROSS_COMPILE)strip
-OBJCOPY    = $(CROSS_COMPILE)objcopy
-OBJDUMP    = $(CROSS_COMPILE)objdump
+OBJCOPY    ?= $(CROSS_COMPILE)objcopy
+OBJDUMP    ?= $(CROSS_COMPILE)objdump
 PKG_CONFIG = $(CROSS_COMPILE)pkg-config
 PYTHON     = python3

@@ -145,7 +145,7 @@ quiet_CC  = @echo "  CC     $@"; $(CC)
 quiet_LD  = @echo "  LD        $@"; $(LD)
 quiet_INSTALL = @echo "  INSTALL       $?"; $(INSTALL)

-V = 0
+V = 1
 ifeq ($(V), 0)
     quiet = quiet_
     Q     = @
