diff --git a/src/GNUmakefile b/src/GNUmakefile
index 777b058..3f25193 100644
--- a/src/GNUmakefile
+++ b/src/GNUmakefile
@@ -8,11 +8,11 @@ SRC    = $(dir $(firstword $(MAKEFILE_LIST)))
 GIT      = git
 RM       = rm
 INSTALL  = install
-CC       = $(CROSS_COMPILE)gcc
+CC       ?= $(CROSS_COMPILE)gcc
 LD       = $(CC)
-STRIP    = $(CROSS_COMPILE)strip
-OBJCOPY  = $(CROSS_COMPILE)objcopy
-OBJDUMP  = $(CROSS_COMPILE)objdump
+STRIP    = :
+OBJCOPY  ?= $(CROSS_COMPILE)objcopy
+OBJDUMP  ?= $(CROSS_COMPILE)objdump
 PYTHON   = python3
 
 HAS_SWIG := $(shell swig -version 2>/dev/null)
@@ -104,7 +104,7 @@ quiet_CC  = @echo "  CC	$@"; $(CC)
 quiet_LD  = @echo "  LD	$@"; $(LD)
 quiet_INSTALL = @echo "  INSTALL	$?"; $(INSTALL)
 
-V = 0
+V = 1
 ifeq ($(V), 0)
     quiet = quiet_
     Q     = @
