--- a/third_party/blink/renderer/platform/disk_data_allocator.h	2020-09-21 20:39:25.000000000 +0200
+++ -	2021-05-24 22:03:47.238393816 +0200
@@ -36,12 +36,12 @@
    public:
     int64_t start_offset() const { return start_offset_; }
     size_t size() const { return size_; }
-    Metadata(Metadata&& other) = delete;
 
    private:
     Metadata(int64_t start_offset, size_t size)
         : start_offset_(start_offset), size_(size) {}
     Metadata(const Metadata& other) = default;
+    Metadata(Metadata&& other) = default;
     Metadata& operator=(const Metadata& other) = default;
 
     int64_t start_offset_;
