--- a/base/process/memory_linux.cc	2020-09-21 20:39:01.000000000 +0200
+++ -	2020-10-02 20:23:17.972108815 +0200
@@ -23,6 +23,12 @@
 #include "third_party/tcmalloc/chromium/src/gperftools/tcmalloc.h"
 #endif
 
+#if defined(LIBC_GLIBC)
+extern "C" {
+extern void *__libc_malloc(size_t size);
+}
+#endif
+
 namespace base {
 
 size_t g_oom_size = 0U;
@@ -135,7 +141,7 @@
     (!defined(LIBC_GLIBC) && !BUILDFLAG(USE_TCMALLOC))
   *result = malloc(size);
 #elif defined(LIBC_GLIBC) && !BUILDFLAG(USE_TCMALLOC)
-  *result = __libc_malloc(size);
+  *result = ::__libc_malloc(size);
 #elif BUILDFLAG(USE_TCMALLOC)
   *result = tc_malloc_skip_new_handler(size);
 #endif
