--- ipxe/src/include/nic.h	2016-10-13 11:36:49.397750709 +0200
+++ ipxe/src/include/nic.h	2016-10-13 11:37:55.082691367 +0200
@@ -199,7 +199,6 @@
 
 #undef DRIVER
 #define DRIVER(_name_text,_unused2,_unused3,_name,_probe,_disable)	  \
-	static const char _name ## _text[] = _name_text;		  \
 	static inline int						  \
 	_name ## _probe ( struct nic *nic, void *hwdev ) {		  \
 		return _probe ( nic, hwdev );				  \
