--- ipxe/src/drivers/net/via-rhine.c	2011-12-11 03:28:04.000000000 +0100
+++ ipxe/src/drivers/net/via-rhine.c	2016-10-13 12:31:46.766898745 +0200
@@ -947,14 +947,14 @@
             // if (tp->chip_id == 0x3065)
             if( tp->chip_revision < 0x80 && tp->chip_revision >=0x40 )
                 intr_status |= inb(nic->ioaddr + IntrStatus2) << 16;
-                intr_status = (intr_status & ~DEFAULT_INTR);
-                if ( action == ENABLE ) 
-                    intr_status = intr_status | DEFAULT_INTR;
-                    outw(intr_status, nic->ioaddr + IntrEnable);
-                break;
+            intr_status = (intr_status & ~DEFAULT_INTR);
+            if ( action == ENABLE ) 
+                intr_status = intr_status | DEFAULT_INTR;
+                outw(intr_status, nic->ioaddr + IntrEnable);
+            break;
         case FORCE :
             outw(0x0010, nic->ioaddr + 0x84);
-           break;
+            break;
         }
 }

