From alpine: https://gitlab.alpinelinux.org/alpine/aports/-/tree/master/main/py3-lxml

the find above returns None with this version
--
diff --git a/src/lxml/tests/test_etree.py b/src/lxml/tests/test_etree.py
index 0339796..790c57a 100644
--- a/src/lxml/tests/test_etree.py
+++ b/src/lxml/tests/test_etree.py
@@ -3073,7 +3073,7 @@ class ETreeOnlyTestCase(HelperTestCase):
         if etree.LIBXML_VERSION < (2, 9, 11):
             self.assertEqual({'hha': None}, el.nsmap)
         else:
-            self.assertEqual({}, el.nsmap)
+            self.assertEqual(None, el)
 
     def test_getchildren(self):
         Element = self.etree.Element
