from alpine: https://gitlab.alpinelinux.org/alpine/aports/-/commit/369de77b6197a5d04beb551537533fc82141a516

From: Patrycja Rosa <alpine@ptrcnull.me>
Date: Thu, 10 Feb 2022 14:09:24 +0100
Subject: test_incremental_xmlfile: fix encoding name

diff --git a/src/lxml/tests/test_incremental_xmlfile.py b/src/lxml/tests/test_incremental_xmlfile.py
--- a/src/lxml/tests/test_incremental_xmlfile.py
+++ b/src/lxml/tests/test_incremental_xmlfile.py
@@ -173,4 +173,4 @@
-        with etree.xmlfile(self._file, encoding='utf16') as xf:
+        with etree.xmlfile(self._file, encoding='utf-16') as xf:
             with xf.element('test'):
                 xf.write('toast')
-        self.assertXml('<test>toast</test>', encoding='utf16')
+        self.assertXml('<test>toast</test>', encoding='utf-16')
