--- a/build/config/compiler/BUILD.gn	2022-07-08 23:17:39.889019407 +0200
+++ -	2022-07-08 23:23:22.740712991 +0200
@@ -1531,14 +1531,6 @@
         cflags += [ "-Wno-deprecated-non-prototype" ]
       }
 
-      if (!is_nacl && !(is_chromeos ||
-                        default_toolchain == "//build/toolchain/cros:target")) {
-        # TODO(https://crbug.com/1316298): Re-enable once test failure is figured out
-        cflags += [
-          "-Xclang",
-          "-no-opaque-pointers",
-        ]
-      }
 
       if (is_fuchsia) {
         # TODO(https://bugs.chromium.org/p/fuchsia/issues/detail?id=77383)
