--- a/sandbox/linux/system_headers/linux_stat.h
+++ b/sandbox/linux/system_headers/linux_stat.h
@@ -155,6 +155,28 @@ struct kernel_stat {
   unsigned int __unused4;
   unsigned int __unused5;
 };
+#elif defined(__powerpc64__)
+struct kernel_stat {
+  unsigned long	st_dev;
+  unsigned long	st_ino;
+  unsigned long	st_nlink;
+  unsigned int	st_mode;
+  unsigned int	st_uid;
+  unsigned int	st_gid;
+  unsigned long	st_rdev;
+  long		st_size;
+  unsigned long	st_blksize;
+  unsigned long	st_blocks;
+  unsigned long	st_atime_;
+  unsigned long	st_atime_nsec_;
+  unsigned long	st_mtime_;
+  unsigned long	st_mtime_nsec_;
+  unsigned long	st_ctime_;
+  unsigned long	st_ctime_nsec_;
+  unsigned long	__unused4;
+  unsigned long	__unused5;
+  unsigned long	__unused6;
+};
 #endif
 
 // On 32-bit systems, we default to the 64-bit stat struct like libc
