--- a/third_party/node/node.py	2022-02-23 04:45:01.000000000 +0100
+++ -	2022-04-24 21:49:27.122211926 +0200
@@ -13,12 +13,7 @@
 def GetBinaryPath():
   darwin_name = ('node-darwin-arm64' if platform.machine() == 'arm64' else
                  'node-darwin-x64')
-  return os_path.join(os_path.dirname(__file__), *{
-    'Darwin': ('mac', darwin_name, 'bin', 'node'),
-    'Linux': ('linux', 'node-linux-x64', 'bin', 'node'),
-    'Windows': ('win', 'node.exe'),
-  }[platform.system()])
-
+  return "/usr/bin/node"
 
 def RunNode(cmd_parts, stdout=None):
   cmd = [GetBinaryPath()] + cmd_parts
