--- a/third_party/swiftshader/third_party/llvm-subzero/build/Linux/include/llvm/Config/config.h
+++ b/third_party/swiftshader/third_party/llvm-subzero/build/Linux/include/llvm/Config/config.h
@@ -17,7 +17,11 @@
 #define ENABLE_CRASH_OVERRIDES 0
 
 /* Define to 1 if you have the `backtrace' function. */
+#ifdef __GLIBC__
 #define HAVE_BACKTRACE 1
+#else
+#define HAVE_BACKTRACE 0
+#endif
 
 /* Define to 1 if you have the <CrashReporterClient.h> header file. */
 /* #undef HAVE_CRASHREPORTERCLIENT_H */
@@ -58,7 +62,11 @@
 #define HAVE_ERRNO_H 1
 
 /* Define to 1 if you have the <execinfo.h> header file. */
+#ifdef __GLIBC__
 #define HAVE_EXECINFO_H 1
+#else
+#define HAVE_EXECINFO_H 0
+#endif
 
 /* Define to 1 if you have the <fcntl.h> header file. */
 #define HAVE_FCNTL_H 1
