--- a/package.json	2022-07-06 17:31:50.000000000 +0200
+++ -	2022-07-08 23:04:43.654812957 +0200
@@ -98,7 +98,6 @@
     "precommit": "lint-staged",
     "preinstall": "node -e 'process.exit(0)'",
     "prepack": "check-for-leaks",
-    "prepare": "husky install",
     "repl": "node ./script/start.js --interactive",
     "start": "node ./script/start.js",
     "test": "node ./script/spec-runner.js",
@@ -143,4 +142,4 @@
       "node script/gen-hunspell-filenames.js"
     ]
   }
-}
\ No newline at end of file
+}
