From 3b274fe8f31d61996343b17402f30408a6e447cf Mon Sep 17 00:00:00 2001
From: John Brandwood <john.brandwood@telzey.com>
Date: Fri, 15 Mar 2024 15:15:53 -0400
Subject: [PATCH] Fix build with GCC 14.x by splitting a NULL assignment into 2
 lines.

---
 src/expand.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/expand.c b/src/expand.c
index 08f829a..ea2d2a9 100644
--- a/src/expand.c
+++ b/src/expand.c
@@ -710,7 +710,8 @@ static char *   replace(
         } else {
             m_inf->locs.start_col = m_inf->locs.start_line = 0L;
         }
-        m_inf->args = m_inf->loc_args = NULL;       /* Default args */
+        m_inf->args = NULL;                         /* Default args */
+        m_inf->loc_args = NULL;
         for (num = 1, recurs = 0; num < m_num; num++)
             if (mac_inf[ num].defp == defp)
                 recurs++;           /* Recursively nested macro     */

