--- Makefile
+++ Makefile
@@ -11,9 +11,9 @@
 
 install: 
 	@echo installing executable to "${DESTDIR}${PREFIX}/bin"
-	@install -D "${BIN}" "${DESTDIR}${BINDIR}/${BIN}"
+	@install -D -m 755 "${BIN}" "${DESTDIR}${BINDIR}/${BIN}"
 	@echo installing manual page to ${DESTDIR}${MANDIR}/man1
-	@install -D "${BIN}.1" "${DESTDIR}${MANDIR}/man1/${BIN}.1"
+	@install -D -m 644 "${BIN}.1" "${DESTDIR}${MANDIR}/man1/${BIN}.1"
 
 uninstall:
 	@echo removing executable file from "${DESTDIR}${PREFIX}/bin"
