--- a/cmake/install_layout.cmake
+++ b/cmake/install_layout.cmake
@@ -86,7 +86,7 @@ IF(UNIX)
   ELSEIF(INSTALL_LAYOUT MATCHES "SVR4")
     SET(default_prefix "/opt/mysql/mysql")
   ELSE()
-    SET(default_prefix "/usr/local/mysql")
+    SET(default_prefix "/usr")
   ENDIF()
   IF(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
     SET(CMAKE_INSTALL_PREFIX ${default_prefix}
--- a/scripts/CMakeLists.txt
+++ b/scripts/CMakeLists.txt
@@ -182,7 +182,7 @@ ENDIF(UNIX)
 # the documentation, so better not change this.
 
 IF(INSTALL_LAYOUT MATCHES "STANDALONE")
-  SET(prefix ".")
+  SET(prefix "${CMAKE_INSTALL_PREFIX}")
   SET(bindir ${prefix}/${INSTALL_BINDIR})
   SET(sbindir ${prefix}/${INSTALL_SBINDIR})
   SET(scriptdir ${prefix}/${INSTALL_BINDIR})
