--- CMakeLists.txt.orig	2019-04-13 15:56:16.853115799 -0400
+++ CMakeLists.txt	2019-04-13 15:56:43.169656287 -0400
@@ -356,10 +356,10 @@
 
 CHECK_SYSTEMD()
 
-IF(CMAKE_CROSSCOMPILING)
-  SET(IMPORT_EXECUTABLES "IMPORTFILE-NOTFOUND" CACHE FILEPATH "Path to import_executables.cmake from a native build")
-  INCLUDE(${IMPORT_EXECUTABLES})
-ENDIF()
+# IF(CMAKE_CROSSCOMPILING)
+#   SET(IMPORT_EXECUTABLES "IMPORTFILE-NOTFOUND" CACHE FILEPATH "Path to import_executables.cmake from a native build")
+#   INCLUDE(${IMPORT_EXECUTABLES})
+# ENDIF()
 
 #
 # Setup maintainer mode options. Platform checks are
