--- ./strings/my_vsnprintf.c.orig	2015-09-16 23:54:27.348529851 -0400
+++ ./strings/my_vsnprintf.c	2015-09-16 23:57:11.968524828 -0400
@@ -827,11 +827,7 @@
     */
 #if defined(__WIN__)
     strerror_s(buf, len, nr);
-#elif ((defined _POSIX_C_SOURCE && (_POSIX_C_SOURCE >= 200112L)) ||    \
-       (defined _XOPEN_SOURCE   && (_XOPEN_SOURCE >= 600)))      &&    \
-      ! defined _GNU_SOURCE
-    strerror_r(nr, buf, len);             /* I can build with or without GNU */
-#elif defined _GNU_SOURCE
+#elif defined(__GLIBC__) && defined (_GNU_SOURCE)
     char *r= strerror_r(nr, buf, len);
     if (r != buf)                         /* Want to help, GNU? */
       strmake(buf, r, len - 1);           /* Then don't. */

--- ./storage/mroonga/vendor/groonga/lib/com.c.orig	2015-09-16 23:53:30.468531587 -0400
+++ ./storage/mroonga/vendor/groonga/lib/com.c	2015-09-16 23:53:43.165531199 -0400
@@ -351,7 +351,7 @@
     struct epoll_event e;
     memset(&e, 0, sizeof(struct epoll_event));
     e.data.fd = (fd);
-    e.events = (__uint32_t) events;
+    e.events = (uint32_t) events;
     if (epoll_ctl(ev->epfd, EPOLL_CTL_ADD, (fd), &e) == -1) {
       SERR("epoll_ctl");
       return ctx->rc;
@@ -399,7 +399,7 @@
       struct epoll_event e;
       memset(&e, 0, sizeof(struct epoll_event));
       e.data.fd = (fd);
-      e.events = (__uint32_t) events;
+      e.events = (uint32_t) events;
       if (epoll_ctl(ev->epfd, EPOLL_CTL_MOD, (fd), &e) == -1) {
         SERR("epoll_ctl");
         return ctx->rc;
