--- ./CMakeLists.txt.orig	2015-09-17 15:47:59.794784111 -0400
+++ ./CMakeLists.txt	2015-09-19 19:42:16.751348473 -0400
@@ -362,10 +376,10 @@
 
 CHECK_PCRE()
 
-IF(CMAKE_CROSSCOMPILING)
-  SET(IMPORT_EXECUTABLES "IMPORTFILE-NOTFOUND" CACHE FILEPATH "Path to import_executables.cmake from a native build")
-  INCLUDE(${IMPORT_EXECUTABLES})
-ENDIF()
+# IF(CMAKE_CROSSCOMPILING)
+#   SET(IMPORT_EXECUTABLES "IMPORTFILE-NOTFOUND" CACHE FILEPATH "Path to import_executables.cmake from a native build")
+#   INCLUDE(${IMPORT_EXECUTABLES})
+# ENDIF()
 
 #
 # Setup maintainer mode options. Platform checks are
