From: Debian MySQL Maintainers <pkg-mysql-maint@lists.alioth.debian.org>
Date: Thu, 10 Aug 2017 20:40:29 +0200
Subject: mips-innobase-atomic

---
 storage/innobase/include/os0sync.h | 1 +
 1 file changed, 1 insertion(+)

diff --git storage/innobase/include/os0sync.h storage/innobase/include/os0sync.h
index bb225c5..6520237 100644
--- storage/innobase/include/os0sync.h
+++ storage/innobase/include/os0sync.h
@@ -37,6 +37,7 @@ Created 9/6/1995 Heikki Tuuri
 
 #include "univ.i"
 #include "ut0lst.h"
+#include "sync0types.h"
 
 /** CPU cache line size */
 #ifdef __powerpc__
