--- ./storage/tokudb/PerconaFT/cmake_modules/TokuFeatureDetection.cmake.orig	2015-12-20 16:08:01.849875314 -0500
+++ ./storage/tokudb/PerconaFT/cmake_modules/TokuFeatureDetection.cmake	2015-12-20 16:08:16.357758151 -0500
@@ -93,7 +93,7 @@
   if (HAVE_BACKTRACE_WITH_EXECINFO)
     list(APPEND EXTRA_SYSTEM_LIBS execinfo)
   else ()
-	  message(FATAL_ERROR "Cannot find backtrace(), even with -lexecinfo.")
+	  message(WARNING "Cannot find backtrace(), even with -lexecinfo.")
   endif ()
 endif ()
 
