--- ./sql/signal_handler.cc.orig	2016-07-22 18:19:35.495952401 -0400
+++ ./sql/signal_handler.cc	2016-07-22 18:20:57.613943892 -0400
@@ -276,6 +276,7 @@
       "\"mlockall\" bugs.\n");
   }
 
+#ifdef HAVE_STACKTRACE
   if (print_invalid_query_pointer)
   {
     my_safe_printf_stderr(
@@ -285,6 +286,7 @@
     my_write_stderr(thd->query(), MY_MIN(65536U, thd->query_length()));
     my_safe_printf_stderr("\n\n");
   }
+#endif
 
 #ifdef HAVE_WRITE_CORE
   if (test_flags & TEST_CORE_ON_SIGNAL)
