--- src/microtouch.c.orig	2009-03-31 10:22:46.455076331 +0200
+++ src/microtouch.c	2009-03-31 10:23:00.507008801 +0200
@@ -425,8 +425,7 @@ DeviceInit (DeviceIntPtr dev)
 	 * Device reports motions on 2 axes in absolute coordinates.
 	 * Axes min and max values are reported in raw coordinates.
 	 */
-	if (InitValuatorClassDeviceStruct (dev, 2, xf86GetMotionEvents,
-									local->history_size, Absolute) == FALSE)
+	if (InitValuatorClassDeviceStruct (dev, 2, local->history_size, Absolute) == FALSE)
 	{
 		ErrorF ("Unable to allocate MicroTouch touchscreen ValuatorClassDeviceStruct\n");
 		return !Success;
