diff --git a/build/config/ui.gni b/build/config/ui.gni
index 547b42f..966b00c 100644
--- a/build/config/ui.gni
+++ b/build/config/ui.gni
@@ -51,8 +51,10 @@ if (use_ozone) {
   use_glib = false
 }

-# Whether to use atk, the Accessibility ToolKit library
-use_atk = is_desktop_linux && use_x11
+declare_args() {
+  # Whether to use atk, the Accessibility ToolKit library
+  use_atk = is_desktop_linux && use_x11
+}
 # =============================================
 #   PLEASE DO NOT ADD MORE FLAGS TO THIS FILE
 # =============================================
