diff --git a/ui/platform_window/x11/x11_window.cc b/ui/platform_window/x11/x11_window.cc
index 7a3a6eb..7c01534 100644
--- a/ui/platform_window/x11/x11_window.cc
+++ b/ui/platform_window/x11/x11_window.cc
@@ -67,6 +67,7 @@ ui::XWindow::Configuration ConvertInitPropertiesToXWindowConfig(
   config.wm_role_name = properties.wm_role_name;
   config.activatable = properties.activatable;
   config.visual_id = properties.x_visual_id;
+  config.icon = properties.icon;
   return config;
 }
 
