From d48486507d6b4ed130696721a758e2a231066cf7 Mon Sep 17 00:00:00 2001
From: Elvis Pranskevichus <elvis@magic.io>
Date: Mon, 6 Jul 2020 16:33:14 -0700
Subject: [PATCH] GCC-X11-Success-enum

---
 components/viz/host/host_display_client.h | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/components/viz/host/host_display_client.h b/components/viz/host/host_display_client.h
index 72a807e..257bc86 100644
--- a/components/viz/host/host_display_client.h
+++ b/components/viz/host/host_display_client.h
@@ -13,7 +13,14 @@
 #include "build/build_config.h"
 #include "components/viz/host/viz_host_export.h"
 #include "mojo/public/cpp/bindings/binding.h"
+#ifdef Success
+#undef Success
+#define __X11_SUCCESS
+#endif
 #include "services/viz/privileged/mojom/compositing/display_private.mojom.h"
+#ifdef __X11_SUCCESS
+#define Success 0
+#endif
 #include "ui/gfx/native_widget_types.h"
 
 namespace viz {
-- 
2.26.2

