--- src/plugins/preauth/pkinit/pkinit_crypto_openssl.c.orig	2014-07-16 00:02:32.311738055 +0200
+++ src/plugins/preauth/pkinit/pkinit_crypto_openssl.c	2014-07-16 00:03:04.729003461 +0200
@@ -172,7 +172,7 @@ static char *
 pkinit_pkcs11_code_to_text(int err);
 
 
-#if OPENSSL_VERSION_NUMBER >= 0x10000000L
+#if OPENSSL_VERSION_NUMBER >= 0x10000000L && OPENSSL_VERSION_NUMBER <  0x20000000L
 /* Use CMS support present in OpenSSL 1.0 and later. */
 #include <openssl/cms.h>
 #define pkinit_CMS_get0_content_signed(_cms) CMS_get0_content(_cms)
