--- src/botantools/botan/botan/secmem.h.orig	2012-01-07 20:09:35.427999593 +0100
+++ src/botantools/botan/botan/secmem.h	2012-01-07 20:09:52.540001422 +0100
@@ -214,11 +214,11 @@
 
       SecureVector(u32bit n = 0) { MemoryRegion<T>::init(true, n); }
       SecureVector(const T in[], u32bit n)
-         { MemoryRegion<T>::init(true); set(in, n); }
+         { MemoryRegion<T>::init(true); this->set(in, n); }
       SecureVector(const MemoryRegion<T>& in)
-         { MemoryRegion<T>::init(true); set(in); }
+         { MemoryRegion<T>::init(true); this->set(in); }
       SecureVector(const MemoryRegion<T>& in1, const MemoryRegion<T>& in2)
-         { MemoryRegion<T>::init(true); set(in1); append(in2); }
+         { MemoryRegion<T>::init(true); this->set(in1); append(in2); }
    };
 
 /*************************************************
