--- a/dbus/dbus-gvalue.c	2013-04-03 11:46:39.075287327 +0200
+++ b/dbus/dbus-gvalue.c	2013-04-03 11:46:59.178392995 +0200
@@ -1932,8 +1932,6 @@ G_STATIC_ASSERT (sizeof (dbus_bool_t) ==
 G_STATIC_ASSERT (sizeof (dbus_int32_t) == sizeof (gint));
 G_STATIC_ASSERT (sizeof (dbus_uint32_t) == sizeof (guint));
 /* These should be true regardless, but just for completeness... */
-G_STATIC_ASSERT (sizeof (dbus_int64_t) == sizeof (gint64));
-G_STATIC_ASSERT (sizeof (dbus_uint64_t) == sizeof (guint64));
 G_STATIC_ASSERT (sizeof (double) == sizeof (gdouble));
 
 static gboolean
