--- a/include/image.h	2014-08-19 08:42:46.000000000 +0200
+++ b/include/image.h	2020-09-15 10:43:48.410147914 +0200
@@ -19,6 +19,11 @@
 #include "compiler.h"
 #include <asm/byteorder.h>
 
+#if !defined(__GLIBC__)
+typedef unsigned long int ulong;
+typedef unsigned long int uintptr_t;
+#endif
+
 /* Define this to avoid #ifdefs later on */
 struct lmb;
 
--- a/tools/mkimage.h	2014-08-19 08:42:46.000000000 +0200
+++ b/tools/mkimage.h	2020-09-15 10:46:44.221140904 +0200
@@ -22,6 +22,11 @@
 #include "fdt_host.h"
 #include "imagetool.h"
 
+#if !defined(__GLIBC__)
+typedef unsigned long int ulong;
+typedef unsigned long int uintptr_t;
+#endif
+
 #undef MKIMAGE_DEBUG
 
 #ifdef MKIMAGE_DEBUG
--- a/tools/proftool.c	2014-08-19 08:42:46.000000000 +0200
+++ b/tools/proftool.c	2020-09-15 10:48:28.212136757 +0200
@@ -20,6 +20,11 @@
 #include <compiler.h>
 #include <trace.h>
 
+#if !defined(__GLIBC__)
+typedef unsigned long int ulong;
+typedef unsigned long int uintptr_t;
+#endif
+
 #define MAX_LINE_LEN 500
 
 enum {
