--- lib/pki_temp.cpp.orig	2016-10-07 18:41:22.920435711 +0400
+++ lib/pki_temp.cpp	2016-10-07 18:48:19.459988316 +0400
@@ -367,8 +367,8 @@
 	PEM_write_bio(b, PEM_STRING_XCA_TEMPLATE, (char*)"",
 		(unsigned char*)(ba.data()), ba.size());
 #else
-	PEM_write_bio(b, PEM_STRING_XCA_TEMPLATE, "",
-		(const unsigned char*)(ba.constData()), ba.size());
+	PEM_write_bio(b, PEM_STRING_XCA_TEMPLATE, (char*)"",
+		(unsigned char*)(ba.constData()), ba.size());
 #endif
 	pki_openssl_error();
 	return b;
