--- src/kj/exception.c++.fix	2016-11-10 10:30:39.182645852 +0100
+++ src/kj/exception.c++	2016-11-10 10:30:26.848646578 +0100
@@ -28,7 +28,7 @@
 #include <exception>
 #include <new>
 
-#if (__linux__ && !__ANDROID__) || __APPLE__
+#if (__GLIBC__) || __APPLE__
 #define KJ_HAS_BACKTRACE 1
 #include <execinfo.h>
 #endif
@@ -43,7 +43,7 @@
 namespace {
 
 String getStackSymbols(ArrayPtr<void* const> trace) {
-#if (__linux__ || __APPLE__) && !__ANDROID__ && defined(KJ_DEBUG)
+#if (__GLIBC__ || __APPLE__) && !__ANDROID__ && defined(KJ_DEBUG)
   // We want to generate a human-readable stack trace.
 
   // TODO(someday):  It would be really great if we could avoid farming out to another process
