--- openjdk.orig/jdk/src/solaris/native/com/sun/management/UnixOperatingSystem_md.c	2019-04-16 05:24:16.000000000 +0200
+++ openjdk/jdk/src/solaris/native/com/sun/management/UnixOperatingSystem_md.c	2021-01-16 09:43:09.253748103 +0100
@@ -33,7 +33,7 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #if defined(_ALLBSD_SOURCE)
-#include <sys/sysctl.h>
+#include <linux/sysctl.h>
 #ifdef __APPLE__
 #include <sys/param.h>
 #include <sys/mount.h>
--- openjdk.orig/jdk/src/solaris/native/java/net/PlainDatagramSocketImpl.c	2019-04-16 05:24:16.000000000 +0200
+++ openjdk/jdk/src/solaris/native/java/net/PlainDatagramSocketImpl.c	2021-01-16 09:43:09.253748103 +0100
@@ -41,7 +41,7 @@
 #endif
 #ifdef __linux__
 #include <unistd.h>
-#include <sys/sysctl.h>
+#include <linux/sysctl.h>
 #include <sys/utsname.h>
 #include <netinet/ip.h>
 
--- openjdk.orig/jdk/src/solaris/native/java/net/PlainSocketImpl.c	2019-04-16 05:24:16.000000000 +0200
+++ openjdk/jdk/src/solaris/native/java/net/PlainSocketImpl.c	2021-01-16 09:43:09.252748056 +0100
@@ -43,7 +43,7 @@
 #endif
 #ifdef __linux__
 #include <unistd.h>
-#include <sys/sysctl.h>
+#include <linux/sysctl.h>
 #endif
 
 #include "jvm.h"
--- openjdk.orig/jdk/src/solaris/native/java/net/net_util_md.c	2019-04-16 05:24:16.000000000 +0200
+++ openjdk/jdk/src/solaris/native/java/net/net_util_md.c	2021-01-16 09:43:09.252748056 +0100
@@ -40,7 +40,7 @@
 #else
 #include <limits.h>
 #include <sys/param.h>
-#include <sys/sysctl.h>
+#include <linux/sysctl.h>
 #ifndef MAXINT
 #define MAXINT INT_MAX
 #endif
--- openjdk.orig/jdk/src/solaris/native/sun/net/portconfig.c	2019-04-16 05:24:16.000000000 +0200
+++ openjdk/jdk/src/solaris/native/sun/net/portconfig.c	2021-01-16 09:43:09.252748056 +0100
@@ -29,7 +29,7 @@
 #include <errno.h>
 
 #if defined(_ALLBSD_SOURCE)
-#include <sys/sysctl.h>
+#include <linux/sysctl.h>
 #endif
 
 #include "jni.h"
