--- IlmThread/Makefile.am.no_undefined	2013-06-18 14:51:38.000000000 -0500
+++ IlmThread/Makefile.am	2013-08-28 21:04:25.793391766 -0500
@@ -14,7 +14,7 @@ libIlmThread_la_LDFLAGS = -version-info
 if LIB_SUFFIX_EXISTS
 libIlmThread_la_LDFLAGS += -release @LIB_SUFFIX@
 endif
-libIlmThread_la_LIBADD = ../Iex/libIex.la
+libIlmThread_la_LIBADD = ../Iex/libIex.la $(PTHREAD_LIBS)
 
 libIlmThreadincludedir = $(includedir)/OpenEXR
 
--- IlmThread/Makefile.in.no_undefined	2013-06-18 14:55:24.000000000 -0500
+++ IlmThread/Makefile.in	2013-08-28 21:04:55.395049371 -0500
@@ -253,7 +253,7 @@ libIlmThread_la_SOURCES = IlmThreadPool.
 
 libIlmThread_la_LDFLAGS = -version-info @LIBTOOL_VERSION@ \
 	-no-undefined $(am__append_1)
-libIlmThread_la_LIBADD = ../Iex/libIex.la
+libIlmThread_la_LIBADD = ../Iex/libIex.la $(PTHREAD_LIBS)
 libIlmThreadincludedir = $(includedir)/OpenEXR
 libIlmThreadinclude_HEADERS = IlmThreadPool.h IlmThread.h \
 			      IlmThreadSemaphore.h IlmThreadMutex.h \
