--- a/IlmBase/IlmBase.pc.in	2019-09-18 03:02:06.000000000 +0200
+++ b/IlmBase/IlmBase.pc.in	2019-12-01 19:18:24.067056545 +0100
@@ -4,9 +4,9 @@
 ##
 
 prefix=@prefix@
-exec_prefix=@exec_prefix@
-libdir=@libdir@
-includedir=@includedir@
+exec_prefix=${prefix}/@exec_prefix@
+libdir=${prefix}/@libdir@
+includedir=${prefix}/@includedir@
 libsuffix=@LIB_SUFFIX_DASH@
 Name: IlmBase
 Description: Base math and exception libraries
