diff --git a/resources/linux/bin/code.sh b/resources/linux/bin/code.sh
index 4f11f5b82e0..dda9effb53a 100755
--- a/resources/linux/bin/code.sh
+++ b/resources/linux/bin/code.sh
@@ -53,7 +53,7 @@ else
 		VSCODE_PATH="$(dirname "$(readlink -f "$0")")/.."
 	else
 		# else use the standard install location
-		VSCODE_PATH="/usr/share/@@APPNAME@@"
+		VSCODE_PATH="/usr/lib/@@APPNAME@@"
 	fi
 fi
 
diff --git a/resources/linux/code-url-handler.desktop b/resources/linux/code-url-handler.desktop
index 7106e0e0969..faaff89d71c 100644
--- a/resources/linux/code-url-handler.desktop
+++ b/resources/linux/code-url-handler.desktop
@@ -2,7 +2,7 @@
 Name=@@NAME_LONG@@ - URL Handler
 Comment=Code Editing. Redefined.
 GenericName=Text Editor
-Exec=@@EXEC@@ --open-url %U
+Exec=/usr/lib/@@NAME@@/bin/@@NAME@@ --open-url %U
 Icon=@@ICON@@
 Type=Application
 NoDisplay=true
diff --git a/resources/linux/code.desktop b/resources/linux/code.desktop
index 4c939a2fab8..1f94f5fccfb 100755
--- a/resources/linux/code.desktop
+++ b/resources/linux/code.desktop
@@ -2,7 +2,7 @@
 Name=@@NAME_LONG@@
 Comment=Code Editing. Redefined.
 GenericName=Text Editor
-Exec=@@EXEC@@ %F
+Exec=/usr/lib/@@NAME@@/bin/@@NAME@@ %F
 Icon=@@ICON@@
 Type=Application
 StartupNotify=false
@@ -23,5 +23,5 @@ Name[ko]=새 빈 창
 Name[ru]=Новое пустое окно
 Name[zh_CN]=新建空窗口
 Name[zh_TW]=開新空視窗
-Exec=@@EXEC@@ --new-window %F
+Exec=/usr/lib/@@NAME@@/bin/@@NAME@@ --new-window %F
 Icon=@@ICON@@
