diff --git a/resources/linux/bin/code.sh b/resources/linux/bin/code.sh
index 55f50b6f1c14..16b73343e6ea 100755
--- a/resources/linux/bin/code.sh
+++ b/resources/linux/bin/code.sh
@@ -26,7 +26,7 @@ else
 		VSCODE_PATH="$(dirname $(readlink -f $0))/.."
 	else
 		# else use the standard install location
-		VSCODE_PATH="/usr/share/@@NAME@@"
+		VSCODE_PATH="/usr/lib/@@NAME@@"
 	fi
 fi
 
diff --git a/resources/linux/code.desktop b/resources/linux/code.desktop
index dbc7818cecf3..eefe01f43397 100644
--- a/resources/linux/code.desktop
+++ b/resources/linux/code.desktop
@@ -2,7 +2,7 @@
 Name=@@NAME_LONG@@
 Comment=Code Editing. Redefined.
 GenericName=Text Editor
-Exec=/usr/share/@@NAME@@/@@NAME@@ --unity-launch %F
+Exec=/usr/lib/@@NAME@@/@@NAME@@ --unity-launch %F
 Icon=@@ICON@@
 Type=Application
 StartupNotify=false
@@ -14,5 +14,5 @@ Keywords=vscode;
 
 [Desktop Action new-empty-window]
 Name=New Empty Window
-Exec=/usr/share/@@NAME@@/@@NAME@@ --new-window %F
+Exec=/usr/lib/@@NAME@@/@@NAME@@ --new-window %F
 Icon=@@ICON@@
