diff --git a/resources/linux/bin/code.sh b/resources/linux/bin/code.sh
index 06973937f14..78f72c9e9ed 100755
--- a/resources/linux/bin/code.sh
+++ b/resources/linux/bin/code.sh
@@ -44,7 +44,7 @@ else
 		VSCODE_PATH="$(dirname "$(readlink -f "$0")")/.."
 	else
 		# else use the standard install location
-		VSCODE_PATH="/usr/share/@@NAME@@"
+		VSCODE_PATH="/usr/lib/@@NAME@@"
 	fi
 fi
 
diff --git a/resources/linux/code-url-handler.desktop b/resources/linux/code-url-handler.desktop
index 7106e0e0969..faaff89d71c 100644
--- a/resources/linux/code-url-handler.desktop
+++ b/resources/linux/code-url-handler.desktop
@@ -2,7 +2,7 @@
 Name=@@NAME_LONG@@ - URL Handler
 Comment=Code Editing. Redefined.
 GenericName=Text Editor
-Exec=@@EXEC@@ --open-url %U
+Exec=/usr/lib/@@NAME@@/bin/@@NAME@@ --open-url %U
 Icon=@@ICON@@
 Type=Application
 NoDisplay=true
diff --git a/resources/linux/code.desktop b/resources/linux/code.desktop
index ab3b79a011b..cdc88ef68a4 100755
--- a/resources/linux/code.desktop
+++ b/resources/linux/code.desktop
@@ -2,7 +2,7 @@
 Name=@@NAME_LONG@@
 Comment=Code Editing. Redefined.
 GenericName=Text Editor
-Exec=@@EXEC@@ --unity-launch %F
+Exec=/usr/lib/@@NAME@@/bin/@@NAME@@ --unity-launch %F
 Icon=@@ICON@@
 Type=Application
 StartupNotify=false
@@ -14,5 +14,5 @@ Keywords=vscode;
 
 [Desktop Action new-empty-window]
 Name=New Empty Window
-Exec=@@EXEC@@ --new-window %F
+Exec=/usr/lib/@@NAME@@/bin/@@NAME@@ --new-window %F
 Icon=@@ICON@@
