diff --git a/build/gulpfile.vscode.js b/build/gulpfile.vscode.js
index 090db00ffebb..0c54098d2601 100644
--- a/build/gulpfile.vscode.js
+++ b/build/gulpfile.vscode.js
@@ -323,7 +323,15 @@ function packageTask(platform, arch, opts) {
		if (platform === 'win32') {
			all = es.merge(all, gulp.src(['resources/win32/code_file.ico', 'resources/win32/code_70x70.png', 'resources/win32/code_150x150.png'], { base: '.' }));
		} else if (platform === 'linux') {
-			all = es.merge(all, gulp.src('resources/linux/code.png', { base: '.' }));
+			const icon = gulp.src('resources/linux/code.png', { base: '.' });
+
+			const desktop = gulp.src('resources/linux/code.desktop', { base: '.' })
+				.pipe(replace('@@NAME_LONG@@', product.nameLong))
+				.pipe(replace('@@NAME_SHORT@@', product.nameShort))
+				.pipe(replace('@@NAME@@', product.applicationName))
+				.pipe(replace('@@ICON@@', `/usr/share/pixmaps/${product.applicationName}.png`));
+
+			all = es.merge(all, icon, desktop);
		} else if (platform === 'darwin') {
			const shortcut = gulp.src('resources/darwin/bin/code.sh')
				.pipe(rename('bin/code'));
