diff --git a/build/npm/postinstall.js b/build/npm/postinstall.js
index ef8fa4a47f66..7b8b710636a7 100644
--- a/build/npm/postinstall.js
+++ b/build/npm/postinstall.js
@@ -73,5 +73,3 @@ yarnInstall('test/automation'); // node modules required for smoketest
 yarnInstall('test/smoke'); // node modules required for smoketest
 yarnInstall('test/integration/browser'); // node modules required for integration
 yarnInstallBuildDependencies(); // node modules for watching, specific to host node version, not electron
-
-cp.execSync('git config pull.rebase true');
