From 4c69751d3ed6c0623aceee3d77499f1c477136ab Mon Sep 17 00:00:00 2001
From: Tao Nelson <taobert@gmail.com>
Date: Fri, 23 Sep 2022 19:17:47 +1000
Subject: [PATCH] Accommodate glib now using pcre2

---
 common/clib/regex.c | 2 +-
 common/util.c       | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/common/clib/regex.c b/common/clib/regex.c
index c4f42317c..e87156cf8 100644
--- a/common/clib/regex.c
+++ b/common/clib/regex.c
@@ -59,7 +59,7 @@ luaH_regenerate_regex(lua_State *L, lregex_t *regex)
 
     GError *error = NULL;
     regex->reg = g_regex_new(regex->pattern,
-            G_REGEX_DOTALL|G_REGEX_OPTIMIZE|G_REGEX_JAVASCRIPT_COMPAT, 0, &error);
+            G_REGEX_DOTALL|G_REGEX_OPTIMIZE, 0, &error);
     if (error) {
         lua_pushstring(L, error->message);
         g_error_free(error);
diff --git a/common/util.c b/common/util.c
index 4b44d8515..738e1e5cb 100644
--- a/common/util.c
+++ b/common/util.c
@@ -61,9 +61,9 @@ strip_ansi_escapes(const gchar *in)
     static GRegex *reg;
 
     if (!reg) {
-        const gchar *expr = "[\\u001b\\u009b][[()#;?]*(?:[0-9]{1,4}(?:;[0-9]{0,4})*)?[0-9A-ORZcf-nqry=><]";
+        const gchar *expr = "[\x1b\x9b][[()#;?]*(?:[0-9]{1,4}(?:;[0-9]{0,4})*)?[0-9A-ORZcf-nqry=><]";
         GError *err = NULL;
-        reg = g_regex_new(expr, G_REGEX_JAVASCRIPT_COMPAT | G_REGEX_DOTALL | G_REGEX_EXTENDED | G_REGEX_RAW | G_REGEX_OPTIMIZE, 0, &err);
+        reg = g_regex_new(expr, G_REGEX_DOTALL | G_REGEX_EXTENDED | G_REGEX_RAW | G_REGEX_OPTIMIZE, 0, &err);
         g_assert_no_error(err);
     }
 
