--- a/Makefile	2022-08-30 17:17:46.000000000 +0200
+++ b/Makefile	2022-09-09 13:20:16.288047152 +0200
@@ -17,7 +17,7 @@
 # Must be kept in sync with doc/docgen.ld
 DOC_SRCS = $(filter-out lib/markdown.lua lib/lousy/init.lua,$(shell for d in doc/luadoc lib common/clib; do find $$d -type f; done)) tests/lib.lua
 
-all: options newline luakit luakit.1.gz luakit.so apidoc
+all: options newline luakit luakit.1.gz luakit.so
 
 options:
 	@echo luakit build options:
@@ -80,28 +80,11 @@
 luakit.1.gz: luakit.1
 	@gzip -c $< > $@
 
-doc/apidocs/index.html: $(DOC_SRCS) $(wildcard build-utils/docgen/*)
-	rm -rf doc/apidocs
-	mkdir doc/apidocs
-	$(LUA_BIN_NAME) ./build-utils/docgen/makedoc.lua
-
-apidoc: doc/apidocs/index.html
-
-doc: buildopts.h $(THEAD) $(TSRC)
-	doxygen -s doc/luakit.doxygen
-
-clean:
-	rm -rf doc/apidocs doc/html luakit $(OBJS) $(EXT_OBJS) $(TSRC) $(THEAD) buildopts.h luakit.1 luakit.1.gz luakit.so
-
 install: all
 	install -d $(DESTDIR)$(DOCDIR)/classes
 	install -d $(DESTDIR)$(DOCDIR)/modules
 	install -d $(DESTDIR)$(DOCDIR)/pages
 	install -m644 README.md AUTHORS COPYING.GPLv3 $(DESTDIR)$(DOCDIR)
-	install -m644 doc/apidocs/classes/* $(DESTDIR)$(DOCDIR)/classes
-	install -m644 doc/apidocs/modules/* $(DESTDIR)$(DOCDIR)/modules
-	install -m644 doc/apidocs/pages/* $(DESTDIR)$(DOCDIR)/pages
-	install -m644 doc/apidocs/*.html $(DESTDIR)$(DOCDIR)
 	install -d $(DESTDIR)$(PREFIX)/share/luakit/lib/lousy/widget
 	install -m644 lib/*.* $(DESTDIR)$(PREFIX)/share/luakit/lib
 	install -m644 lib/lousy/*.* $(DESTDIR)$(PREFIX)/share/luakit/lib/lousy
