--- configure.in.orig	2013-03-21 17:43:57.269562981 +0100
+++ configure.in	2013-03-21 17:44:23.956687605 +0100
@@ -44,7 +44,7 @@ AC_FUNC_FSEEKO
 AC_CHECK_SIZEOF(void*,0)
 
 #@@@ new name is AC_CONFIG_HEADERS
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 
 AC_C_BIGENDIAN
 
@@ -256,8 +256,8 @@ AH_TEMPLATE(FLAC__HAS_OGG, [define if yo
 fi
 
 dnl check for i18n(internationalization); these are from libiconv/gettext
-AM_ICONV
-AM_LANGINFO_CODESET
+AC_DEFINE([HAVE_ICONV], [], [Whether we have libiconv available]) LIBICONV=""
+AC_SUBST(LIBICONV)
 
 AC_CHECK_PROGS(DOCBOOK_TO_MAN, docbook-to-man docbook2man)
 AM_CONDITIONAL(FLaC__HAS_DOCBOOK_TO_MAN, test -n "$DOCBOOK_TO_MAN")
