From 3c65bd1fcb4648181bf86a5c7a45813ab0f27b8b Mon Sep 17 00:00:00 2001
From: Andrea Brancaleoni <miwaxe@gmail.com>
Date: Sun, 30 Aug 2015 15:59:42 +0200
Subject: [PATCH] musl

---
 lib/Support/Unix/FileSystem.inc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/Support/Unix/FileSystem.inc b/lib/Support/Unix/FileSystem.inc
index 67fdd73..0cf873d 100644
--- a/lib/Support/Unix/FileSystem.inc
+++ b/lib/Support/Unix/FileSystem.inc
@@ -19,6 +19,7 @@
 #include <sys/stat.h>
 #include <sys/mman.h>
 #include <unistd.h>
+#include <limits.h>

 namespace mcld {
 namespace sys {
--
2.5.1

