From 2e300dfa4f6915a752d17c465b8d932849e17910 Mon Sep 17 00:00:00 2001
From: Isarhamster <frodo.hobbit@gmx.net>
Date: Sun, 17 May 2020 18:23:30 +0200
Subject: [PATCH] Remove a couple of unmaintained translations from resource

---
 resources.qrc | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/resources.qrc b/resources.qrc
index c826912f..770f9934 100644
--- resources.qrc
+++ resources.qrc
@@ -12,11 +12,7 @@
         <file alias="help/about4.html">data/help/about4.html</file>
         <file alias="help/about5.html">data/help/about5.html</file>
         <file alias="help/about6.html">data/help/about6.html</file>
-        <file alias="i18n/chessx_da.qm">i18n/chessx_da.qm</file>
         <file alias="i18n/chessx_de.qm">i18n/chessx_de.qm</file>
-        <file alias="i18n/chessx_fr.qm">i18n/chessx_fr.qm</file>
-        <file alias="i18n/chessx_it.qm">i18n/chessx_it.qm</file>
-        <file alias="i18n/chessx_cz.qm">i18n/chessx_cz.qm</file>
         <file alias="images/accept.png">data/images/accept.png</file>
         <file alias="images/annotate.png">data/images/annotate.png</file>
         <file alias="images/arrow_green.png">data/images/arrow_green.png</file>
@@ -180,7 +176,6 @@
         <file alias="themes/shadow/leipzig.png">data/themes/shadow/leipzig.png</file>
         <file alias="themes/shadow/merida.png">data/themes/shadow/merida.png</file>
         <file alias="themes/shadow/spatial.png">data/themes/shadow/spatial.png</file>
-        <file alias="i18n/chessx_ru.qm">i18n/chessx_ru.qm</file>
         <file alias="images/threat_black.png">data/images/threat_black.png</file>
         <file alias="images/threat_white.png">data/images/threat_white.png</file>
         <file alias="images/black_wall.png">data/images/black_wall.png</file>
