--- src/dialogs/preferences.cpp
+++ src/dialogs/preferences.cpp
@@ -33,6 +33,7 @@
 #include <QFileDialog>
 #include <QDesktopServices>
 #include <QTextStream>
+#include <QAction>
 
 #if defined(_MSC_VER) && defined(_DEBUG)
 #define DEBUG_NEW new( _NORMAL_BLOCK, __FILE__, __LINE__ )
--- src/dialogs/savedialog.cpp
+++ src/dialogs/savedialog.cpp
@@ -24,6 +24,7 @@
 
 #include <QLineEdit>
 #include <QCompleter>
+#include <QButtonGroup>
 
 #if defined(_MSC_VER) && defined(_DEBUG)
 #define DEBUG_NEW new( _NORMAL_BLOCK, __FILE__, __LINE__ )
--- src/gui/gamewindow.cpp
+++ src/gui/gamewindow.cpp
@@ -9,6 +9,7 @@
 #include "settings.h"
 
 #include <QToolButton>
+#include <QAction>
 
 #if defined(_MSC_VER) && defined(_DEBUG)
 #define DEBUG_NEW new( _NORMAL_BLOCK, __FILE__, __LINE__ )
