diff --git a/plugins/externaltools/tools/library.py b/plugins/externaltools/tools/library.py
index ed66e26..0eb4e12 100755
--- a/plugins/externaltools/tools/library.py
+++ b/plugins/externaltools/tools/library.py
@@ -197,7 +197,7 @@ class ToolDirectory(object):
 
 
 class Tool(object):
-    RE_KEY = re.compile('^([a-zA-Z_][a-zA-Z0-9_.\-]*)(\[([a-zA-Z_@]+)\])?$')
+    RE_KEY = re.compile(r'^([a-zA-Z_][a-zA-Z0-9_.\-]*)(\[([a-zA-Z_@]+)\])?$')
 
     def __init__(self, parent, filename=None):
         super(Tool, self).__init__()
diff --git a/plugins/snippets/snippets/Document.py b/plugins/snippets/snippets/Document.py
index 1c2960b..65c4b43 100644
--- a/plugins/snippets/snippets/Document.py
+++ b/plugins/snippets/snippets/Document.py
@@ -827,8 +827,8 @@ class Document:
             return components
 
     def relative_path(self, first, second, mime):
-        prot1 = re.match('(^[a-z]+:\/\/|\/)(.*)', first)
-        prot2 = re.match('(^[a-z]+:\/\/|\/)(.*)', second)
+        prot1 = re.match(r'(^[a-z]+:\/\/|\/)(.*)', first)
+        prot2 = re.match(r'(^[a-z]+:\/\/|\/)(.*)', second)
 
         if not prot1 or not prot2:
             return second
diff --git a/plugins/snippets/snippets/SubstitutionParser.py b/plugins/snippets/snippets/SubstitutionParser.py
index e6b4647..a03382b 100644
--- a/plugins/snippets/snippets/SubstitutionParser.py
+++ b/plugins/snippets/snippets/SubstitutionParser.py
@@ -159,7 +159,7 @@ class SubstitutionParser:
         return match.group(1), tokens[match.end():]
 
     def _condition_value(self, tokens):
-        match = re.match('\\\\?%s\s*' % self.REG_GROUP, tokens)
+        match = re.match('\\\\?%s\\s*' % self.REG_GROUP, tokens)
 
         if not match:
             return None, tokens
