--- a/src/modules/audio_coding/neteq/reorder_optimizer.cc
+++ b/src/modules/audio_coding/neteq/reorder_optimizer.cc
@@ -11,6 +11,7 @@
 #include "modules/audio_coding/neteq/reorder_optimizer.h"
 
 #include <algorithm>
+#include <cstdint>
 #include <limits>
 #include <vector>
 
--- a/src/rtc_base/system/file_wrapper.h
+++ -
@@ -15,6 +15,7 @@
 #include <stdio.h>
 
 #include <string>
+#include <cstdint>
 
 #include "absl/strings/string_view.h"
 
