diff --git CMakeLists.txt CMakeLists.txt
index d970719..c14226e 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -45,7 +45,6 @@ include(cmake/libpffft.cmake)
 include(cmake/librnnoise.cmake)
 include(cmake/libsrtp.cmake)
 include(cmake/libusrsctp.cmake)
-include(cmake/libvpx.cmake)
 include(cmake/libwebrtcbuild.cmake)
 include(cmake/libyuv.cmake)
 if (NOT WIN32 AND NOT APPLE)
@@ -77,13 +76,6 @@ if (is_x86 OR is_x64)
         tg_owt::librnnoise
         tg_owt::libsrtp
         tg_owt::libusrsctp
-        tg_owt::libvpx
-        tg_owt::libvpx_mmx
-        tg_owt::libvpx_sse2
-        tg_owt::libvpx_ssse3
-        tg_owt::libvpx_sse4
-        tg_owt::libvpx_avx
-        tg_owt::libvpx_avx2
         tg_owt::libyuv
     )
 else()
@@ -97,7 +89,6 @@ else()
         tg_owt::librnnoise
         tg_owt::libsrtp
         tg_owt::libusrsctp
-        tg_owt::libvpx
         tg_owt::libyuv
     )
 endif()
@@ -122,6 +113,7 @@ if (APPLE)
     target_link_libraries(tg_owt PUBLIC tg_owt::libsdkmacos)
 endif()
 
+link_vpx(tg_owt)
 link_openssl(tg_owt)
 link_ffmpeg(tg_owt)
 link_opus(tg_owt)
@@ -1868,16 +1860,6 @@ elseif (APPLE)
 endif()
 
 set(vpx_export)
-if (is_x86 OR is_x64)
-    set(vpx_export
-        libvpx_mmx
-        libvpx_sse2
-        libvpx_ssse3
-        libvpx_sse4
-        libvpx_avx
-        libvpx_avx2
-    )
-endif()
 
 set(export_targets
     tg_owt
@@ -1887,7 +1869,6 @@ set(export_targets
     librnnoise
     libsrtp
     libusrsctp
-    libvpx
     ${vpx_export}
     libwebrtcbuild
     libyuv
diff --git cmake/external.cmake cmake/external.cmake
index 964e260..82d68d0 100644
--- cmake/external.cmake
+++ cmake/external.cmake
@@ -129,3 +129,15 @@ function(link_dl target_name)
         target_link_libraries(${target_name} PRIVATE ${CMAKE_DL_LIBS})
     endif()
 endfunction()
+
+# libvpx
+function(link_vpx target_name)
+    if (TG_OWT_PACKAGED_BUILD)
+        find_package(PkgConfig REQUIRED)
+        pkg_check_modules(VPX vpx)
+    endif()
+    if (VPX_FOUND)
+        target_include_directories(${target_name} PRIVATE ${VPX_INCLUDE_DIRS})
+        target_link_libraries(${target_name} PRIVATE ${VPX_LINK_LIBRARIES})
+    endif()
+endfunction()
