--- a/src/migration/mdb/3rdparty/mdbtools/libmdb/catalog.c	2018-03-06 22:46:18.503349370 +0100
+++ b/src/migration/mdb/3rdparty/mdbtools/libmdb/catalog.c	2018-03-06 22:46:18.503349370 +0100
@@ -17,6 +17,7 @@
  */
 
 #include "mdbtools.h"
+#include <strings.h>
 
 #ifdef DMALLOC
 #include "dmalloc.h"
--- a/src/migration/mdb/3rdparty/mdbtools/libmdb/data.c	2018-03-06 22:46:18.503349370 +0100
+++ b/src/migration/mdb/3rdparty/mdbtools/libmdb/data.c	2018-03-06 22:46:18.503349370 +0100
@@ -18,6 +18,7 @@
 
 #include <time.h>
 #include <math.h>
+#include <strings.h>
 #include "mdbtools.h"
 
 #ifdef DMALLOC
--- a/src/migration/mdb/3rdparty/mdbtools/libmdb/sargs.c	2018-03-06 22:46:18.507349448 +0100
+++ b/src/migration/mdb/3rdparty/mdbtools/libmdb/sargs.c	2018-03-06 22:46:18.507349448 +0100
@@ -27,6 +27,7 @@
  * a mdb_test_[type]() function and invoke it from mdb_test_sarg()
  */
 
+#include <strings.h>
 #include <time.h>
 #include "mdbtools.h"
 #ifdef DMALLOC
--- a/src/migration/mdb/3rdparty/mdbtools/libmdb/table.c	2018-03-06 22:46:18.507349448 +0100
+++ b/src/migration/mdb/3rdparty/mdbtools/libmdb/table.c	2018-03-06 22:46:18.507349448 +0100
@@ -16,6 +16,7 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
  */
 
+#include <strings.h>
 #include "mdbtools.h"
 
 #ifdef DMALLOC
