--- a/accel-pppd/ctrl/ipoe/arp.c
+++ b/accel-pppd/ctrl/ipoe/arp.c
@@ -13,7 +13,6 @@
 #include <netinet/ip.h>
 #include <sys/socket.h>
 #include <sys/ioctl.h>
-#include <linux/if_arp.h>
 #include <linux/if_packet.h>
 
 #include "list.h"
--- a/accel-pppd/ctrl/ipoe/ipoe.c.orig
+++ b/accel-pppd/ctrl/ipoe/ipoe.c
@@ -15,7 +15,6 @@
 #include <sys/socket.h>
 #include <sys/ioctl.h>
 #include <linux/if.h>
-#include <linux/if_arp.h>
 #include <linux/route.h>
 
 #include <pcre.h>
