--- a/src/base/omx_base_component.c	2018-09-10 11:38:00.018615437 +0200
+++ b/src/base/omx_base_component.c	2018-09-10 11:38:58.672611941 +0200
@@ -827,7 +827,7 @@
   uuid[0] = (OMX_U32)omx_component;
   uuid[1] = getpid();
   uuid[2] = getuid();
-  memcpy(*pComponentUUID, uuid, 3*sizeof(uuid));
+  memcpy(*pComponentUUID, uuid, sizeof(uuid));
 
   DEBUG(DEB_LEV_FUNCTION_NAME, "Out of %s for component %p\n", __func__, hComponent);
   return OMX_ErrorNone;
