From 88b51489346e6ef352c56fff8dd0298341e0108e Mon Sep 17 00:00:00 2001
From: Vincent Schweiger <vincent.schweiger@icloud.com>
Date: Thu, 9 Apr 2026 19:34:47 +0200
Subject: [PATCH] fix cross build

---
 cmake/FindBpfObject.cmake | 17 +++--------------
 1 file changed, 3 insertions(+), 14 deletions(-)

diff --git a/cmake/FindBpfObject.cmake b/cmake/FindBpfObject.cmake
index 2f2ada3..570a1a5 100644
--- a/cmake/FindBpfObject.cmake
+++ b/cmake/FindBpfObject.cmake
@@ -128,22 +128,11 @@ find_package_handle_standard_args(BpfObject
     GENERATED_VMLINUX_DIR)
 
 # Get clang bpf system includes
-execute_process(
-  COMMAND bash -c "${BPFOBJECT_CLANG_EXE} -v -E - < /dev/null 2>&1 |
-          sed -n '/<...> search starts here:/,/End of search list./{ s| \\(/.*\\)|-idirafter \\1|p }'"
-  OUTPUT_VARIABLE CLANG_SYSTEM_INCLUDES_output
-  ERROR_VARIABLE CLANG_SYSTEM_INCLUDES_error
-  RESULT_VARIABLE CLANG_SYSTEM_INCLUDES_result
-  OUTPUT_STRIP_TRAILING_WHITESPACE)
-if(${CLANG_SYSTEM_INCLUDES_result} EQUAL 0)
-  string(REPLACE "\n" " " CLANG_SYSTEM_INCLUDES ${CLANG_SYSTEM_INCLUDES_output})
-  message(STATUS "BPF system include flags: ${CLANG_SYSTEM_INCLUDES}")
-else()
-  message(FATAL_ERROR "Failed to determine BPF system includes: ${CLANG_SYSTEM_INCLUDES_error}")
-endif()
+set(CLANG_SYSTEM_INCLUDES "-I${XBPS_CROSS_BASE}/usr/include -I${XBPS_CROSS_BASE}/usr/include/${XBPS_TARGET_TRIPLET}")
+message(STATUS "BPF system include flags: ${CLANG_SYSTEM_INCLUDES}")
 
 # Get target arch
-execute_process(COMMAND uname -m
+execute_process(COMMAND printf "${CMAKE_SYSTEM_PROCESSOR}"
   COMMAND sed -e "s/x86_64/x86/" -e "s/aarch64/arm64/" -e "s/ppc64le/powerpc/" -e "s/mips.*/mips/"
   OUTPUT_VARIABLE ARCH_output
   ERROR_VARIABLE ARCH_error
-- 
2.53.0

