diff --git testsuite/Makefile.am testsuite/Makefile.am
index ccb84d3..9774a9c 100644
--- testsuite/Makefile.am
+++ testsuite/Makefile.am
@@ -7,9 +7,9 @@ AM_CPPFLAGS = -I$(top_srcdir) $(DBUS_CFLAGS)
 LDADD = ../snapper/libsnapper.la ../dbus/libdbus.la -lboost_unit_test_framework
 
 check_PROGRAMS = sysconfig-get1.test dirname1.test basename1.test 		\
-	equal-date.test dbus-escape.test cmp-lt.test humanstring.test uuid.test	\
-	table.test table-formatter.test csv-formatter.test json-formatter.test	\
-	getopts.test scan-datetime.test root-prefix.test range.test limit.test
+	equal-date.test dbus-escape.test uuid.test	\
+	csv-formatter.test json-formatter.test	\
+	getopts.test scan-datetime.test root-prefix.test
 
 if ENABLE_BTRFS_QUOTA
 check_PROGRAMS += qgroup1.test
@@ -29,14 +29,8 @@ equal_date_test_LDADD = -lboost_unit_test_framework ../client/utils/libutils.la
 
 scan_datetime_test_LDADD = -lboost_unit_test_framework ../client/utils/libutils.la
 
-humanstring_test_LDADD = -lboost_unit_test_framework ../client/utils/libutils.la
-
 uuid_test_LDADD = -lboost_unit_test_framework ../client/utils/libutils.la
 
-table_test_LDADD = -lboost_unit_test_framework ../client/utils/libutils.la
-
-table_formatter_test_LDADD = -lboost_unit_test_framework ../client/utils/libutils.la
-
 csv_formatter_test_LDADD = -lboost_unit_test_framework ../client/utils/libutils.la
 
 json_formatter_test_LDADD = -lboost_unit_test_framework ../client/utils/libutils.la $(JSONC_LIBS)
@@ -44,7 +38,3 @@ json_formatter_test_LDADD = -lboost_unit_test_framework ../client/utils/libutils
 getopts_test_LDADD = -lboost_unit_test_framework ../client/utils/libutils.la
 
 lvm_utils_test_LDADD = -lboost_unit_test_framework ../snapper/libsnapper.la
-
-range_test_LDADD = -lboost_unit_test_framework ../client/utils/libutils.la
-
-limit_test_LDADD = -lboost_unit_test_framework ../client/utils/libutils.la
