From a4068f03b471777151c4eb43fc70c3b6ebf8df3a Mon Sep 17 00:00:00 2001
From: Nathan Owens <ndowens04@gmail.com>
Date: Wed, 18 Dec 2019 18:23:38 -0600
Subject: [PATCH 1/4] musl-sterror_r

Signed-off-by: Nathan Owens <ndowens04@gmail.com>
---
 snapper/AppUtil.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/snapper/AppUtil.cc b/snapper/AppUtil.cc
index 553065c..2897276 100644
--- snapper/AppUtil.cc
+++ snapper/AppUtil.cc
@@ -223,7 +223,7 @@ namespace snapper
     string
     stringerror(int errnum)
     {
-#if (_POSIX_C_SOURCE >= 200112L) && ! _GNU_SOURCE
+#if (_POSIX_C_SOURCE >= 200112L) && ! _GNU_SOURCE || MUSL_LIBC
 	char buf1[100];
 	if (strerror_r(errnum, buf1, sizeof(buf1) - 1) == 0)
 	    return string(buf1);
-- 
2.24.1

