diff --git client/cleanup.cc client/cleanup.cc
index 801c0f5..e57a0a8 100644
--- a/client/cleanup.cc
+++ b/client/cleanup.cc
@@ -51,7 +51,7 @@ struct Parameters
     MinFreeLimit free_limit;
 
 
-    void read(const ProxyConfig& config, const char* name, long int& value)
+    void read(const ProxyConfig& config, const char* name, time_t& value)
     {
        const map<string, string>& raw = config.getAllValues();
        map<string, string>::const_iterator pos = raw.find(name);
