From 6385e34fb13dbfabd0b4b93f7670eac0c490637f Mon Sep 17 00:00:00 2001
From: Nathan Owens <ndowens04@gmail.com>
Date: Wed, 18 Dec 2019 18:24:08 -0600
Subject: [PATCH 2/4] musl-mktime

Signed-off-by: Nathan Owens <ndowens04@gmail.com>
---
 snapper/AppUtil.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/snapper/AppUtil.cc b/snapper/AppUtil.cc
index 2897276..64812d7 100644
--- snapper/AppUtil.cc
+++ snapper/AppUtil.cc
@@ -288,7 +288,7 @@ namespace snapper
 	const char* p = strptime(str.c_str(), "%F %T", &s);
 	if (!p || *p != '\0')
 	    return (time_t)(-1);
-	return utc ? timegm(&s) : timelocal(&s);
+	return utc ? timegm(&s) : mktime(&s);
     }
 
 
-- 
2.24.1

