From d519c3c94a029d6891b0703ce5adb6ea0ac8b9fa Mon Sep 17 00:00:00 2001
From: Nathan Owens <ndowens04@gmail.com>
Date: Wed, 18 Dec 2019 18:27:17 -0600
Subject: [PATCH 4/4] musl-stdout_stderr

Signed-off-by: Nathan Owens <ndowens04@gmail.com>
---
 snapper/SystemCmd.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/snapper/SystemCmd.h b/snapper/SystemCmd.h
index 3d38604..72bbb2b 100644
--- snapper/SystemCmd.h
+++ snapper/SystemCmd.h
@@ -32,6 +32,8 @@
 #include <list>
 #include <boost/noncopyable.hpp>
 
+#undef stdout
+#undef stderr
 
 namespace snapper
 {
-- 
2.24.1

