fixing btrfs4.19 issues conflicting with boost
--- snapper/Btrfs.cc
+++ snapper/Btrfs.cc
@@ -41,6 +41,9 @@
 #include <btrfs/send.h>
 #include <btrfs/send-stream.h>
 #include <btrfs/send-utils.h>
+#ifdef swap
+#undef swap
+#endif
 #include <boost/version.hpp>
 #include <boost/thread.hpp>
 #endif


--- snapper/BtrfsUtils.cc
+++ snapper/BtrfsUtils.cc
@@ -34,6 +34,9 @@
 #ifdef HAVE_LIBBTRFS
 #include <btrfs/ioctl.h>
 #include <btrfs/send-utils.h>
+#ifdef swap
+#undef swap
+#endif
 #endif
 #include <algorithm>
 #include <functional>
