From 7ddd58ee5e6b7c3a8df3086ff5a6d17d19362761 Mon Sep 17 00:00:00 2001
From: Nathan Owens <ndowens04@gmail.com>
Date: Wed, 18 Dec 2019 18:35:55 -0600
Subject: [PATCH] musl-_u_int-types

Signed-off-by: Nathan Owens <ndowens04@gmail.com>
---
 client/utils/JsonFormatter.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git client/utils/JsonFormatter.h client/utils/JsonFormatter.h
index 79f7672..501b77d 100644
--- client/utils/JsonFormatter.h
+++ client/utils/JsonFormatter.h
@@ -25,6 +25,9 @@
 #include <string>
 #include <vector>
 #include <utility>
+#ifndef __GLIBC__
+#include <sys/types.h>
+#endif
 
 namespace snapper
 {
-- 
2.24.1

