diff --git a/src/intern_blks/ya_intern.c b/src/intern_blks/ya_intern.c
index 29bb11535f23..55610950302a 100644
--- a/src/intern_blks/ya_intern.c
+++ b/src/intern_blks/ya_intern.c
@@ -18,7 +18,7 @@ void ya_int_cpu(ya_block_t *blk);
 void ya_int_diskio(ya_block_t *blk);
 void ya_int_network(ya_block_t *blk);

-struct reserved_blk ya_reserved_blks[YA_INTERNAL_LEN] = {
+struct reserved_blk ya_reserved_blks[YA_INTERNAL_LEN] = {
	{"YABAR_DATE", ya_int_date},
	{"YABAR_UPTIME", ya_int_uptime},
	{"YABAR_THERMAL", ya_int_thermal},
@@ -32,14 +32,14 @@ struct reserved_blk ya_reserved_blks[YA_INTERNAL_LEN] = {
	{"YABAR_TITLE", NULL},
	{"YABAR_WORKSPACE", NULL}
 #endif
-};
+};

 //#define YA_INTERNAL

 #ifdef YA_INTERNAL

-inline void ya_setup_prefix_suffix(ya_block_t *blk, size_t * prflen, size_t *suflen, char **startstr) {
-	if(blk->internal->prefix) {
+__attribute__ ((gnu_inline)) inline void ya_setup_prefix_suffix(ya_block_t *blk, size_t * prflen, size_t *suflen, char **startstr) {
+ 	if(blk->internal->prefix) {
		*prflen = strlen(blk->internal->prefix);
		if(*prflen) {
			strcpy(blk->buf, blk->internal->prefix);
@@ -194,7 +194,7 @@ void ya_int_brightness(ya_block_t *blk) {

 void ya_int_bandwidth(ya_block_t * blk) {
	unsigned long rx, tx, orx, otx;
-	unsigned int rxrate, txrate;
+	unsigned int rxrate, txrate;
	FILE *rxfile, *txfile;
	char rxpath[128];
	char txpath[128];
@@ -259,7 +259,7 @@ void ya_int_bandwidth(ya_block_t * blk) {
		fclose(txfile);
		sleep(blk->sleep);
	}
-
+
 }

 void ya_int_memory(ya_block_t *blk) {
@@ -421,8 +421,8 @@ void ya_int_network(ya_block_t *blk) {
			continue;
		family = ifa->ifa_addr->sa_family;
		//printf("%s\n", ifa->ifa_name);
-		if (family == AF_INET || family == AF_INET6) {
-			s = getnameinfo(ifa->ifa_addr,
+		if (family == AF_INET || family == AF_INET6) {
+			s = getnameinfo(ifa->ifa_addr,
					(family == AF_INET) ? sizeof(struct sockaddr_in) :
					sizeof(struct sockaddr_in6), host, 1025,
					NULL, 0, NI_NUMERICHOST);
