diff --git a/Makefile b/Makefile
index bfbf543..9fa6deb 100644
--- a/Makefile
+++ b/Makefile
@@ -55,9 +55,11 @@ $(LIBDIR)/libmcpp.a: $(OBJS)
 	$(AR) rcs $(LIBDIR)/libmcpp.a $(OBJS)
 
 install: $(LIBDIR)/libmcpp.a
-	@mkdir -p $(PREFIX)/$(LIBDIR)
-	cp $(LIBDIR)/libmcpp.a $(PREFIX)/$(LIBDIR)
+	install -Dm644 $(LIBDIR)/libmcpp.a $(DESTDIR)$(PREFIX)/$(LIBDIR)/libmcpp.a
+	install -Dm644 mcpp_lib.h $(DESTDIR)$(PREFIX)/include/mcpp_lib.h
+	install -m644 mcpp_out.h $(DESTDIR)$(PREFIX)/include/mcpp_out.h
 
+.PHONY: clean
 clean:
 	rm -f $(OBJS)
 	rm -rf $(LIBDIR)
