--- a/lib/vecmath/include/vecmath/scalar.h	2021-10-27 17:02:35.220565752 -0500
+++ -	2021-10-27 17:02:50.567780673 -0500
@@ -27,6 +27,7 @@
 #include <tuple>
 #include <type_traits>
 #include <utility>
+#include <cstdint>
 
 namespace vm {
     /**
@@ -377,7 +378,7 @@
      */
     template <typename T>
     constexpr T trunc(const T v) {
-        return static_cast<T>(static_cast<int64_t>(v));
+        return static_cast<T>(static_cast<std::int64_t>(v));
     }
 
     /**
--- a/common/src/TrenchBroomStackWalker.cpp	2021-03-15 12:54:02.000000000 -0500
+++ -	2021-10-29 17:05:18.439743183 -0500
@@ -23,8 +23,10 @@
 #include <QMutexLocker>
 #endif
 #else
+#if defined(__GLIBC__)
 #include <execinfo.h>
 #endif
+#endif
 
 #include "TrenchBroomStackWalker.h"
 
@@ -91,6 +93,7 @@
     }
 #endif
 #else
+#if defined(__GLIBC__)
     std::string TrenchBroomStackWalker::getStackTrace() {
         const int MaxDepth = 256;
         void *callstack[MaxDepth];
@@ -109,5 +112,10 @@
         free(strs);
         return ss.str();
     }
+#else
+    std::string TrenchBroomStackWalker::getStackTrace() {
+        return "";
+    }
+#endif
 #endif
 }
