Subject: Re: Regression in dash 0.5.10 related to subshells
Message-ID: <20180505160243.t5rujv3eifiust5a@gondor.apana.org.au>

--- src/jobs.c
+++ src/jobs.c
@@ -975,8 +975,8 @@ waitforjob(struct job *jp)
 	int st;
 
 	TRACE(("waitforjob(%%%d) called\n", jp ? jobno(jp) : 0));
-	while ((jp && jp->state == JOBRUNNING) || gotsigchld)
-		dowait(DOWAIT_BLOCK, jp);
+	while (jp ? jp->state == JOBRUNNING : gotsigchld)
+		dowait(jp ? DOWAIT_BLOCK : DOWAIT_NORMAL, jp);
 	if (!jp)
 		return exitstatus;
 	st = getstatus(jp);
