--- a/meson.build
+++ b/meson.build
@@ -182,8 +182,8 @@ elif get_option('epub')
   error('"mathjax-directory" is undefined and could not be autodetected')
 endif
 
-mate_submodules_dep = dependency('mate-submodules', version: '1.24.0',
-fallback: ['mate-submodules', 'mate_submodules_dep'])
+mate_submodules_proj = subproject('mate-submodules')
+mate_submodules_dep = mate_submodules_proj.get_variable('libegg_dep')
 
 # Generate config.h
 config_h_file = configure_file(
--- a/subprojects/mate-submodules/libegg/meson.build
+++ b/subprojects/mate-submodules/libegg/meson.build
@@ -23,3 +23,8 @@ egg = static_library('libegg',
   c_args : cargs,
   dependencies : libegg_deps,
 )
+
+libegg_dep = declare_dependency(
+  include_directories: include_directories('.'),
+  link_with: egg,
+  )
