--- a/CMakeLists.txt	2020-08-27 15:31:42.000000000 +0700
+++ b/CMakeLists.txt	2021-01-14 21:44:51.704505119 +0700
@@ -10,19 +10,6 @@
 option(SNDIO "SNDIO" ON)
 option(PORTAUDIO "PORTAUDIO" ON)
 
-# Fix pkg-config for cross-builds (such as MinGW on ArchLinux)
-if(CMAKE_FIND_ROOT_PATH)
-    set(CMAKE_SYSROOT "${CMAKE_FIND_ROOT_PATH}")
-
-    set(ENV{PKG_CONFIG_DIR} "")
-    set(ENV{PKG_CONFIG_LIBDIR} "${CMAKE_SYSROOT}/lib/pkgconfig:${CMAKE_SYSROOT}/share/pkgconfig")
-    set(ENV{PKG_CONFIG_SYSROOT_DIR} ${CMAKE_SYSROOT})
-endif()
-
-if(NOT CMAKE_BUILD_TYPE)
-	set(CMAKE_BUILD_TYPE Debug)
-endif()
-
 set(CMAKE_C_STANDARD 11)
 set(xava_VERSION_MAJOR 0)
 set(xava_VERSION_MINOR 6)
