--- src/game-view.vala	2019-09-09 21:09:07.000000000 +0200
+++ src/game-view.vala	2019-10-13 20:40:46.515571474 +0200
@@ -398,8 +398,7 @@
 
         set_content_gravity (Clutter.ContentGravity.CENTER);
 
-        pivot_point.x = 0.5f;
-        pivot_point.y = 0.5f;
+	set_pivot_point (0.5f, 0.5f);
     }
 
     /* Destroy the tile */
@@ -438,8 +437,7 @@
 
         set_content_gravity (Clutter.ContentGravity.CENTER);
 
-        pivot_point.x = 0.5f;
-        pivot_point.y = 0.5f;
+	set_pivot_point (0.5f, 0.5f);
     }
 }
 
@@ -460,8 +458,7 @@
 
         add_child (label);
 
-        pivot_point.x = 0.5f;
-        pivot_point.y = 0.5f;
+	set_pivot_point (0.5f, 0.5f);
 
         this.scene_width = (float) width;
         this.scene_height = (float) height;
