From a366bf649c032c56f52138d14e2b96b4a48525de Mon Sep 17 00:00:00 2001
From: Rico Tzschichholz <ricotz@ubuntu.com>
Date: Thu, 2 Jul 2020 15:28:02 +0200
Subject: [PATCH] Don't use unsupported vala syntax

---
 src/game-view.vala | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git src/game-view.vala src/game-view.vala
index 1fb0c4b..69b50b8 100644
--- src/game-view.vala
+++ src/game-view.vala
@@ -451,7 +451,7 @@ private class Theme : Object
     internal Theme (string name)
     {
         textures = new Clutter.Image [4];
-        string[4] colors = {"blue", "green", "yellow", "red"};
+        string[] colors = {"blue", "green", "yellow", "red"};
 
         /* Create the textures required to render */
         try
-- 
GitLab

