--- sfxr-sdl-1.2.1/Makefile	2012-04-28 07:55:05.000000000 -0500
+++ sfxr-sdl-1.2.1/Makefile	2020-04-18 21:08:53.055807782 -0500
@@ -1,7 +1,7 @@
-CFLAGS=-ggdb
+override CFLAGS += -ggdb
 GTK=3.0
-CXXFLAGS=$(CFLAGS) `sdl-config --cflags` `pkg-config gtk+-${GTK} --cflags`
-LDFLAGS=`sdl-config --libs` `pkg-config gtk+-${GTK} --libs`
+override CXXFLAGS += $(CFLAGS) `sdl-config --cflags` `pkg-config gtk+-${GTK} --cflags`
+override LDFLAGS += `sdl-config --libs` `pkg-config gtk+-${GTK} --libs`
 
 sfxr: main.cpp tools.h sdlkit.h
 	$(CXX) $< $(CXXFLAGS) $(LDFLAGS) -o $@
