--- include/bsd/string.h	2017-08-05 13:20:54.000000000 +0200
+++ include/bsd/string.h	2018-02-24 12:07:30.705680366 +0100
@@ -42,7 +42,7 @@
 char *strnstr(const char *str, const char *find, size_t str_len);
 void strmode(mode_t mode, char *str);
 
-#if defined(_GNU_SOURCE) && defined(__GLIBC__) && !__GLIBC_PREREQ(2, 25)
+#if defined(_GNU_SOURCE)
 void explicit_bzero(void *buf, size_t len);
 #endif
 __END_DECLS
--- include/bsd/stdlib.h	2017-08-05 13:20:00.000000000 +0200
+++ include/bsd/stdlib.h	2018-02-24 12:09:25.072688523 +0100
@@ -67,7 +67,7 @@
                const unsigned char *table, unsigned endbyte);
 
 void *reallocf(void *ptr, size_t size);
-#if defined(_GNU_SOURCE) && defined(__GLIBC__) && !__GLIBC_PREREQ(2, 26)
+#if defined(_GNU_SOURCE)
 void *reallocarray(void *ptr, size_t nmemb, size_t size);
 #endif
 
