From ecf2a6c42122946cc84150f6927ef69c1f67c909 Mon Sep 17 00:00:00 2001
From: Xeonacid <h.dwwwwww@gmail.com>
Date: Sat, 8 Jan 2022 01:58:44 +0800
Subject: [PATCH] Meson: remove positional arguments from i18n.merge_file
 (#226)

---
 data/meson.build | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/data/meson.build b/data/meson.build
index c456813..109554d 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -1,5 +1,5 @@
 po_extra_dir = join_paths(meson.source_root(), 'po', 'extra')
-i18n.merge_file('desktop',
+i18n.merge_file(
     input: 'io.elementary.switchboard.desktop.in',
     output: 'io.elementary.switchboard.desktop',
     install: true,
@@ -8,7 +8,7 @@ i18n.merge_file('desktop',
     type: 'desktop'
 )
 
-i18n.merge_file('appdata',
+i18n.merge_file(
     input: 'io.elementary.switchboard.appdata.xml.in',
     output: 'io.elementary.switchboard.appdata.xml',
     install: true,
