diff --git a/ipe-tools/pdftoipe/parseargs.h b/ipe-tools/pdftoipe/parseargs.h
index 8530bd1..bf87418 100644
--- ipe-tools/pdftoipe/parseargs.h
+++ ipe-tools/pdftoipe/parseargs.h
@@ -27,8 +27,6 @@
 extern "C" {
 #endif
 
-#include "goo/gtypes.h"
-
 /*
  * Argument kinds.
  */
diff --git a/ipe-tools/pdftoipe/xmloutputdev.cpp b/ipe-tools/pdftoipe/xmloutputdev.cpp
index 3c38f2e..7186437 100644
--- ipe-tools/pdftoipe/xmloutputdev.cpp
+++ ipe-tools/pdftoipe/xmloutputdev.cpp
@@ -330,7 +330,7 @@ void XmlOutputDev::drawImage(GfxState *state, Object *ref, Stream *str,
   finishText();
 
   ImageStream *imgStr;
-  Guchar *p;
+  unsigned char *p;
   GfxRGB rgb;
   int x, y;
   int c;

