--- a/ipe-tools/pdftoipe/pdftoipe.cpp
+++ b/ipe-tools/pdftoipe/pdftoipe.cpp
@@ -100,7 +100,7 @@ int main(int argc, char *argv[])
   }
 
   // open PDF file
-  PDFDoc *doc = new PDFDoc(std::make_unique<GooString>(fileName), ownerPW, userPW);
+  PDFDoc *doc = new PDFDoc(std::unique_ptr<GooString>(fileName), ownerPW, userPW);
 
   if (!doc->isOk())
     return 1;
@@ -110,10 +110,10 @@ int main(int argc, char *argv[])
   if (argc == 3) {
     xmlFileName = argv[2];
   } else {
-    const char *p = fileName->c_str() + fileName->getLength() - 4;
+    const char *p = fileName->c_str() + fileName->size() - 4;
     if (!strcmp(p, ".pdf") || !strcmp(p, ".PDF")) {
         xmlFileName = std::string(fileName->c_str(),
-                                  fileName->getLength() - 4);
+                                  fileName->size() - 4);
     } else {
       xmlFileName = fileName->c_str();
     }
