diff --git build-aux/run-ag.sh build-aux/run-ag.sh
index 6ee7b85..f604462 100755
--- a/build-aux/run-ag.sh
+++ b/build-aux/run-ag.sh
@@ -25,7 +25,7 @@
 # any containing directory must be created. The target is created with a
 # very old time stamp.
 #
-AGexe=/u/bkorb/tools/ag/autogen-bld/agen5/.libs/autogen
+AGexe=/usr/bin/autogen
 find_exe() {
   eval local exe=\${$1}
   test -x "$exe" && return 0
diff --git getdefs/Makefile.in getdefs/Makefile.in
index 058d0a6..7a17783 100644
--- a/getdefs/Makefile.in
+++ b/getdefs/Makefile.in
@@ -275,7 +275,7 @@ AG_STATIC_AUTOGEN = @AG_STATIC_AUTOGEN@
 AG_TIMEOUT = @AG_TIMEOUT@
 AG_VERSION = @AG_VERSION@
 AG_XML2 = @AG_XML2@
-AGexe = @AGexe@
+AGexe = /usr/bin/autogen
 AGnam = @AGnam@
 AMTAR = @AMTAR@
 AM_DEFAULT_VERBOSITY = @AM_DEFAULT_VERBOSITY@
