diff -ru a/Makefile b/Makefile
--- a/Makefile	2015-09-09 07:54:02.000000000 -0700
+++ b/Makefile	2015-11-23 02:32:22.886247897 -0800
@@ -26,7 +26,7 @@
 	install -Dm644 doc/$(PN).1 "$(DESTDIR)$(MANDIR)/$(PN).1"
 	gzip -9 "$(DESTDIR)$(MANDIR)/$(PN).1"
 
-install: install-bin install-man install-systemd
+install: install-bin install-man
 
 install-systemd:
 	$(Q)echo -e '\033[1;32mInstalling systemd files...\033[0m'
