chskshsgr must be built for the host.

--- a/src/Makefile
+++ b/src/Makefile
@@ -179,6 +179,11 @@ choose: choose.sh warn-auto.sh
 coe.o: coe.c coe.h compile
 	./compile coe.c
 
+compilehost:
+	cat warn-auto.sh > compilehost
+	@echo 'exec cc -D_GNU_SOURCE -c $${1+"$$@"}' >> compilehost
+	chmod 555 compilehost
+
 compile: conf-cc print-cc.sh systype warn-auto.sh
 	rm -f compile
 	sh print-cc.sh > compile
@@ -229,12 +234,21 @@ hassgprm.h: choose compile hassgprm.h1 h
 haswaitp.h: choose compile haswaitp.h1 haswaitp.h2 load trywaitp.c
 	./choose cl trywaitp haswaitp.h1 haswaitp.h2 > haswaitp.h
 
-iopause.h: choose compile iopause.h1 iopause.h2 load trypoll.c
+iopause.h: choose compilehost iopause.h1 iopause.h2 loadhost trypoll.c
 	./choose clr trypoll iopause.h1 iopause.h2 > iopause.h
 
 iopause.o: compile iopause.c iopause.h select.h tai.h taia.h uint64.h
 	./compile iopause.c
 
+loadhost:
+	rm -f loadhost
+	@echo > loadhost
+	@echo 'systype="`cat systype`"' >> loadhost
+	@echo 'cat warn-auto.sh' >> loadhost
+	@echo 'main="$$1"; shift' >> loadhost
+	@echo 'exec cc -o "$$main" "$$main".o $${1+"$$@"}' >> loadhost
+	chmod 555 loadhost
+
 load: conf-ld print-ld.sh systype warn-auto.sh
 	rm -f load
 	sh print-ld.sh > load
@@ -422,7 +436,7 @@ taia_pack.o taia_sub.o taia_uint.o
 	taia_now.o taia_pack.o taia_sub.o taia_uint.o
 
 uint64.h: choose compile load tryulong64.c uint64.h1 uint64.h2
-	./choose clr tryulong64 uint64.h1 uint64.h2 > uint64.h
+	./choose cl tryulong64 uint64.h1 uint64.h2 > uint64.h
 
 unix.a: alloc.o alloc_re.o buffer.o buffer_0.o buffer_1.o buffer_2.o \
 buffer_get.o buffer_put.o buffer_read.o buffer_write.o coe.o env.o \
--- a/src/choose.sh
+++ b/src/choose.sh
@@ -2,10 +2,12 @@
 result="$4"
 
 case "$1" in
+  *c*r*) ./compilehost $2.c >/dev/null 2>&1 || result="$3" ;;
   *c*) ./compile $2.c >/dev/null 2>&1 || result="$3" ;;
 esac
 
 case "$1" in
+  *l*r*) ./loadhost $2 >/dev/null 2>&1 || result="$3" ;;
   *l*) ./load $2 >/dev/null 2>&1 || result="$3" ;;
 esac
 
--- a/src/tryulong64.c
+++ b/src/tryulong64.c
@@ -1,15 +1,10 @@
 /* Public domain. */
 
 #include <unistd.h>
+#include <stdint.h>
 
 int main()
 {
-  unsigned long u;
-  u = 1;
-  u += u; u += u; u += u; u += u; u += u; u += u; u += u; u += u;
-  u += u; u += u; u += u; u += u; u += u; u += u; u += u; u += u;
-  u += u; u += u; u += u; u += u; u += u; u += u; u += u; u += u;
-  u += u; u += u; u += u; u += u; u += u; u += u; u += u; u += u;
-  if (!u) _exit(1);
-  _exit(0);
+  char u[(sizeof(unsigned long) == sizeof(uint64_t)) * 2 - 1] = { 0 };
+  return u[0];
 }
