From 09ae1f95dbcd048031aa6ec9e581211809b3ab28 Mon Sep 17 00:00:00 2001
From: Fabian Winter <5821180+fdw@users.noreply.github.com>
Date: Tue, 14 May 2024 21:34:54 +0200
Subject: [PATCH] Fix rofi mode

---
 src/picker/mode.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/picker/mode.py b/src/picker/mode.py
index 4ee8db0..d2e0288 100644
--- a/src/picker/mode.py
+++ b/src/picker/mode.py
@@ -225,7 +225,7 @@ def choose_action(self, state: State) -> None:
         state.step += 1
 
     def execute_actions(self, state: State) -> Optional[str]:
-        save_recent_characters(state.processed_characters, self.args.max_recent)
-        execute_action(state.processed_characters, state.actions)
+        save_recent_characters(state.processed_characters, self.args.max_recent, self.args.files)
+        execute_action(state.processed_characters, state.actions, '')
         state.step += 1
         return
