--- configure	2014-04-13 23:37:52.355000000 +0200
+++ configure	2014-04-13 23:38:32.853000000 +0200
@@ -6140,14 +6140,6 @@
   am__fastdepCXX_FALSE=
 fi
 
-
-# check for C++ compiler explicitly and fail if none is found, do this check
-# after AC_PROG_CXX has set the CXX environment variable
-if ! which $CXX >/dev/null 2>&1; then
-	as_fn_error $? "No C++ compiler found. Please install a C++ compiler." "$LINENO" 5
-fi
-
-
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether ln -s works" >&5
 $as_echo_n "checking whether ln -s works... " >&6; }
 LN_S=$as_ln_s
