Index: Source/WebCore/dom/make_names.pl
===================================================================
--- Source/WebCore/dom/make_names.pl	(revision 73989)
+++ Source/WebCore/dom/make_names.pl	(revision 84123)
@@ -66,5 +66,5 @@
     $gccLocation = "/usr/bin/gcc";
 }
-my $preprocessor = $gccLocation . " -E -P -x c++";
+my $preprocessor = $gccLocation . " -E -x c++";
 
 GetOptions(
