Index: /trunk/Source/WebKit2/UIProcess/API/gtk/WebKitDefines.h
===================================================================
--- Source/WebKit2/UIProcess/API/gtk/WebKitDefines.h	(revision 109222)
+++ Source/WebKit2/UIProcess/API/gtk/WebKitDefines.h	(revision 109329)
@@ -34,4 +34,6 @@
 
 typedef struct _WebKitPrintOperation WebKitPrintOperation;
+typedef struct _WebKitFindController WebKitFindController;
+typedef struct _WebKitWebView        WebKitWebView;
 
 #ifdef G_OS_WIN32
@@ -47,6 +49,3 @@
 #endif
 
-typedef struct _WebKitFindController WebKitFindController;
-typedef struct _WebKitWebView        WebKitWebView;
-
 #endif // WebKitDefines_h
Index: /trunk/Source/WebKit2/UIProcess/API/gtk/WebKitWebView.h
===================================================================
--- Source/WebKit2/UIProcess/API/gtk/WebKitWebView.h	(revision 109222)
+++ Source/WebKit2/UIProcess/API/gtk/WebKitWebView.h	(revision 109329)
@@ -49,5 +49,4 @@
 #define WEBKIT_WEB_VIEW_GET_CLASS(obj)  (G_TYPE_INSTANCE_GET_CLASS((obj),  WEBKIT_TYPE_WEB_VIEW, WebKitWebViewClass))
 
-typedef struct _WebKitWebView WebKitWebView;
 typedef struct _WebKitWebViewClass WebKitWebViewClass;
 typedef struct _WebKitWebViewPrivate WebKitWebViewPrivate;
