From 5677863e61e2c115eb86613b09f636b84cf4ada3 Mon Sep 17 00:00:00 2001
From: Alexis Hildebrandt <afh@surryhill.net>
Date: Tue, 3 Oct 2017 00:00:07 +0200
Subject: [PATCH] curses: Add CTRL-N/CTRL-P for next/previous element

---
 src/out_curses.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git src/out_curses.c src/out_curses.c
index 51b6391..330e8f6 100644
--- a/src/out_curses.c
+++ b/src/out_curses.c
@@ -45,6 +45,8 @@ enum {
 	KEY_TOGGLE_DETAILS	= 'd',
 	KEY_TOGGLE_INFO		= 'i',
 	KEY_COLLECT_HISTORY	= 'h',
+	KEY_CTRL_N	= 14,
+	KEY_CTRL_P	= 16,
 };
 
 #define DETAILS_COLS		40
@@ -1186,10 +1188,12 @@ static int handle_input(int ch)
 			return 1;
 
 		case KEY_DOWN:
+		case KEY_CTRL_N:
 			element_select_next();
 			return 1;
 
 		case KEY_UP:
+		case KEY_CTRL_P:
 			element_select_prev();
 			return 1;
 
