From 1333ae592d0261e6a014ef4c0fb25086ab0919dc Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20G=C3=B3rny?= <mgorny@gentoo.org>
Date: Wed, 5 Aug 2026 20:24:55 +0200
Subject: [PATCH] Use the PEP 621 project metadata to support flit-core>=5
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Michał Górny <mgorny@gentoo.org>
---
 pyproject.toml | 22 ++++++++++++----------
 1 file changed, 12 insertions(+), 10 deletions(-)

diff --git a/pyproject.toml b/pyproject.toml
index 1279b10..810224a 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,18 +1,20 @@
 [build-system]
-requires = ["flit_core  >=2,<4"]
+requires = ["flit_core >=2,<5"]
 build-backend = "flit_core.buildapi"
 
-[tool.flit.metadata]
-module = "entrypoints"
-author = "Thomas Kluyver"
-author-email = "thomas@kluyver.me.uk"
-home-page = "https://github.com/takluyver/entrypoints"
-description-file = "README.rst"
+[project]
+name = "entrypoints"
+authors = [
+    { name = "Thomas Kluyver", email = "thomas@kluyver.me.uk" },
+]
+readme = "README.rst"
+license = "MIT"
+requires-python = ">=3.6"
+dynamic = ["version", "description"]
 classifiers = [
-    "License :: OSI Approved :: MIT License",
     "Programming Language :: Python :: 3"
 ]
-requires-python = ">=3.6"
 
-[tool.flit.metadata.urls]
+[project.urls]
+Homepage = "https://github.com/takluyver/entrypoints"
 Documentation = "https://entrypoints.readthedocs.io/en/latest/"
