--- /dev/null
+++ b/setup.py
@@ -0,0 +1,14 @@
+from setuptools import setup
+
+setup(
+    name='entrypoints',
+    version='@VERSION@',
+    description='Discover and load entry points from installed packages',
+    author='Thomas Kluyver',
+    author_email='thomas@kluyver.me.uk',
+    url='https://github.com/takluyver/entrypoints',
+    license='MIT',
+    extras_require={
+        ':python_version == "2.7"': ['configparser'],
+    },
+)
