diff --git a/CMakeLists.txt b/CMakeLists.txt
index 333b529..aa508e5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -39,9 +39,9 @@ if(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
     set(CMAKE_INSTALL_PREFIX "${PROJECT_SOURCE_DIR}/install" CACHE PATH "Install Path" FORCE)
 endif()
 
-add_subdirectory(external/OpenCL-Headers)
-add_subdirectory(external/OpenCL-ICD-Loader)
-add_subdirectory(external/OpenCL-CLHPP)
+find_package (OpenCLHeaders REQUIRED)
+find_package (OpenCL REQUIRED)
+find_package (OpenCLHeadersCpp REQUIRED)
 
 list(APPEND CMAKE_MODULE_PATH
   ${PROJECT_SOURCE_DIR}/cmake
