--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -201,7 +201,7 @@ if (WITH_LUA)
     find_program(LUA_EXE lua)
 endif()
 
-find_package(Boost 1.50 REQUIRED COMPONENTS system filesystem)
+find_package(Boost 1.50 REQUIRED)
 include_directories(SYSTEM ${Boost_INCLUDE_DIR})
 
 find_package(PostgreSQL REQUIRED)
--- a/src/output-flex.cpp
+++ b/src/output-flex.cpp
@@ -36,7 +36,7 @@ extern "C"
 #include <rapidjson/stringbuffer.h>
 #include <rapidjson/writer.h>
 
-#include <boost/filesystem.hpp>
+#include <filesystem>
 
 #include <cassert>
 #include <cstdlib>
@@ -1633,7 +1633,7 @@ void output_flex_t::init_lua(std::string
     luaX_add_table_int(lua_state(), "stage", 1);
 
     std::string const dir_path =
-        boost::filesystem::path{filename}.parent_path().string();
+        std::filesystem::path{filename}.parent_path().string();
     luaX_add_table_str(lua_state(), "config_dir", dir_path.c_str());
 
     luaX_add_table_func(lua_state(), "define_table",
--- a/tests/common-cleanup.hpp
+++ b/tests/common-cleanup.hpp
@@ -14,7 +14,7 @@
 
 #include <string>
 
-#include <boost/filesystem.hpp>
+#include <filesystem>
 
 namespace testing {
 namespace cleanup {
@@ -46,7 +46,7 @@ private:
         }
 
         boost::system::error_code ec;
-        boost::filesystem::remove(m_filename, ec);
+        std::filesystem::remove(m_filename, ec);
         if (ec && warn) {
             fmt::print(stderr, "WARNING: Unable to remove \"{}\": {}\n",
                        m_filename, ec.message());
