--- a/configure	2019-05-08 10:44:58.580806039 +0200
+++ b/configure	2019-05-08 10:45:35.185289121 +0200
@@ -17628,7 +17628,7 @@
   "$ac_abs_confdir/src/fzshellext/configure" --prefix="$prefix" --exec-prefix="$exec_prefix" --host=x86_64-w64-mingw32 || exit 1
 else
   cd "$OLD_DIR/src/fzshellext/64" || exit 1
-  "$ac_abs_confdir/src/fzshellext/configure" --disable-shellext || exit 1
+  "$ac_abs_confdir/src/fzshellext/configure" --host="$host_alias" --disable-shellext || exit 1
 fi
 cd "$OLD_DIR"
 
