--- src/engine/sftpcontrolsocket.cpp.orig	2016-09-26 03:00:12.401188401 +0200
+++ src/engine/sftpcontrolsocket.cpp	2016-09-26 03:00:23.531188975 +0200
@@ -2031,7 +2031,7 @@
 
 std::wstring CSftpControlSocket::QuoteFilename(std::wstring const& filename)
 {
-	return L"\"" + fz::replace_substrings(filename, L"\"", L"\"\"") + L"\"";
+	return L"\"" + fz::replaced_substrings(filename, L"\"", L"\"\"") + L"\"";
 }
 
 int CSftpControlSocket::Delete(const CServerPath& path, std::deque<std::wstring>&& files)
