--- src/engine/ControlSocket.cpp.orig	2017-10-02 08:05:25.000000000 +0000
+++ src/engine/ControlSocket.cpp	2017-10-02 08:05:53.129649637 +0000
@@ -682,7 +682,7 @@
 
 bool CControlSocket::IsLocked(locking_reason reason, CServerPath const& directory)
 {
-	assert(currentServer_);
+	assert(static_cast<bool>(currentServer_));
 
 	std::list<t_lockInfo>::iterator own = GetLockStatus();
 	if (own != m_lockInfoList.end()) {
--- src/engine/ControlSocket.cpp.orig	2017-10-02 08:06:50.000000000 +0000
+++ src/engine/ControlSocket.cpp	2017-10-02 08:07:35.442654915 +0000
@@ -629,7 +629,7 @@
 
 bool CControlSocket::TryLockCache(locking_reason reason, CServerPath const& directory)
 {
-	assert(currentServer_);
+	assert(static_cast<bool>(currentServer_));
 	assert(!operations_.empty());
 
 	std::list<t_lockInfo>::iterator own = GetLockStatus();
--- src/engine/engineprivate.cpp.orig	2017-09-19 21:36:39.000000000 +0000
+++ src/engine/engineprivate.cpp	2017-10-02 08:09:57.375662238 +0000
@@ -554,7 +554,7 @@
 
 	assert(m_pControlSocket);
 	CServer const& ownServer  = m_pControlSocket->GetCurrentServer();
-	assert(ownServer);
+	assert(static_cast<bool>(ownServer));
 
 	for (auto & engine : m_engineList) {
 		if (!engine || engine == this) {
@@ -821,7 +821,7 @@
 		return FZ_REPLY_ERROR;
 	}
 
-	assert(m_pControlSocket->GetCurrentServer());
+	assert(static_cast<bool>(m_pControlSocket->GetCurrentServer()));
 
 	bool is_outdated = false;
 	if (!directory_cache_.Lookup(listing, m_pControlSocket->GetCurrentServer(), path, true, is_outdated)) {
--- src/interface/loginmanager.cpp.orig	2017-09-19 21:36:39.000000000 +0000
+++ src/interface/loginmanager.cpp	2017-10-02 08:12:38.959670574 +0000
@@ -69,7 +69,7 @@
 
 bool CLoginManager::DisplayDialogForEncrypted(ServerWithCredentials &server, std::wstring const& name)
 {
-	assert(server.credentials.encrypted_);
+	assert(static_cast<bool>(server.credentials.encrypted_));
 
 	wxDialogEx pwdDlg;
 	if (!pwdDlg.Load(wxGetApp().GetTopWindow(), _T("ID_ENTERMASTERPASSWORD"))) {
