--- a/src/commonui/Makefile.in	2021-07-21 21:11:45.600907372 +0200
+++ b/src/commonui/Makefile.in	2021-07-21 21:12:01.772920575 +0200
@@ -191,7 +191,7 @@
 am__v_lt_ = $(am__v_lt_@AM_DEFAULT_V@)
 am__v_lt_0 = --silent
 am__v_lt_1 = 
-libfzclient_commonui_private_la_LINK = $(LIBTOOL) $(AM_V_lt) \
+libfzclient_commonui_private_la_LINK = $(LIBTOOL) $(AM_V_lt) --tag=CXX \
 	$(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=link $(OBJCXXLD) \
 	$(AM_OBJCXXFLAGS) $(OBJCXXFLAGS) \
 	$(libfzclient_commonui_private_la_LDFLAGS) $(LDFLAGS) -o $@
