--- src/lib/sandbox/sandbox.c	2018-12-20 17:33:15.815307612 +0100
+++ src/lib/sandbox/sandbox.c	2018-12-20 17:36:59.795797024 +0100
@@ -111,6 +111,11 @@
 #define REG_SYSCALL 8
 #define M_SYSCALL regs[REG_SYSCALL]
 
+#elif defined(__powerpc64__)
+
+#define REG_SYSCALL 0
+#define M_SYSCALL gp_regs[REG_SYSCALL]
+
 #endif /* defined(__i386__) || ... */
 
 /**Determines if at least one sandbox is active.*/
