--- src/ftrace.h
+++ src/ftrace.h
@@ -11,6 +11,7 @@
 
 #include <kwinglobals.h>
 
+#include <type_traits>
 #include <QFile>
 #include <QMutex>
 #include <QMutexLocker>
@@ -70,10 +71,15 @@ private:
 
 class KWIN_EXPORT FTraceDuration
 {
+    using ftrace_context_t = std::conditional<
+        QAtomicOpsSupport<sizeof(qulonglong)>::IsSupported,
+        qulonglong,
+        ulong>::type;
+
 public:
     template<typename... Args> FTraceDuration(Args... args)
     {
-        static QAtomicInteger<qulonglong> s_context = 0;
+        static QAtomicInteger<ftrace_context_t> s_context = 0;
         QTextStream stream(&m_message);
         (stream << ... << args);
         stream.flush();
@@ -85,7 +91,7 @@ public:
 
 private:
     QByteArray m_message;
-    qulonglong m_context;
+    ftrace_context_t m_context;
 };
 
 } // namespace KWin
