--- kwin-6.0.0/src/wayland/tools/CMakeLists.txt	2024-02-22 10:52:24.000000000 +0100
+++ -	2024-03-05 14:58:18.547351388 +0100
@@ -13,7 +13,7 @@
 if(CMAKE_CROSSCOMPILING AND QTWAYLANDSCANNER_KDE_EXECUTABLE)
     add_executable(qtwaylandscanner_kde IMPORTED GLOBAL)
     set_target_properties(qtwaylandscanner_kde PROPERTIES IMPORTED_LOCATION ${QTWAYLANDSCANNER_KDE_EXECUTABLE})
-elseif(CMAKE_CROSSCOMPILING)
+elseif(CMAKE_CROSSCOMPILING AND NOT FORCE_CROSSCOMPILED_TOOLS)
     if (NOT KF6_HOST_TOOLING)
         message(FATAL_ERROR "Please provide a prefix with a native Qt build and pass -DKF6_HOST_TOOLING=path")
     endif()
