--- a/src/effects/CMakeLists.txt	2022-02-07 13:48:20.000000000 +0100
+++ -	2022-02-10 15:23:10.228131146 +0100
@@ -8,7 +8,7 @@
     kcoreaddons_add_plugin(${name} INSTALL_NAMESPACE "kwin/effects/configs" SOURCES ${ARGV})
 endfunction()
 
-if (CMAKE_CROSSCOMPILING)
+if (CMAKE_CROSSCOMPILING AND NOT FORCE_CROSSCOMPILED_TOOLS)
     if (NOT KWIN_HOST_TOOLING)
         message(FATAL_ERROR "Please provide the path to a native build and pass -DKWIN_HOST_TOOLING=path")
     endif()
