diff --git kadu-core/plugin/dependency-graph/plugin-dependency-graph-builder.h kadu-core/plugin/dependency-graph/plugin-dependency-graph-builder.h
index 1e6a500..a231cba 100644
--- kadu-core/plugin/dependency-graph/plugin-dependency-graph-builder.h
+++ kadu-core/plugin/dependency-graph/plugin-dependency-graph-builder.h
@@ -22,6 +22,7 @@
 #include <QtCore/QObject>
 #include <map>
 #include <set>
+#include <functional>
 
 #include "exports.h"
 #include "plugin/dependency-graph/plugin-dependency-graph.h"
