Qt5 since 5.5.0 needs explicit include of QDataStream

--- kadu-core/os/single-application/single-application.cpp	2015-03-24 22:13:18.000000000 +0100
+++ kadu-core/os/single-application/single-application.cpp	2015-07-10 22:10:02.633790611 +0200
@@ -30,6 +30,7 @@
 #include <QtCore/QRegExp>
 #include <QtNetwork/QLocalServer>
 #include <QtNetwork/QLocalSocket>
+#include <QDataStream>
 #include <memory>
 
 #if defined(Q_OS_WIN)
