--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -21,7 +21,7 @@ endif()
 find_package(PkgConfig REQUIRED)
 find_package(Threads REQUIRED)
 find_package(Filesystem REQUIRED)
-find_package(Boost REQUIRED COMPONENTS system program_options iostreams)
+find_package(Boost REQUIRED COMPONENTS program_options iostreams)
 find_package(Wt REQUIRED COMPONENTS Wt Dbo DboSqlite3 HTTP)
 
 # WT
--- a/src/libs/core/CMakeLists.txt
+++ b/src/libs/core/CMakeLists.txt
@@ -49,7 +49,6 @@ target_link_libraries(lmscore PRIVATE
 	)
 
 target_link_libraries(lmscore PUBLIC
-	Boost::system
 	std::filesystem
 	Wt::Wt
 	)
--- a/src/libs/services/auth/CMakeLists.txt
+++ b/src/libs/services/auth/CMakeLists.txt
@@ -19,7 +19,6 @@ target_include_directories(lmsauth PRIVA
 	)
 
 target_link_libraries(lmsauth PUBLIC
-	Boost::system
 	Wt::Wt
 	lmscore
 	lmsdatabase
