--- a/imagemagick.m4	2021-09-09 10:53:42.287881616 +0200
+++ b/imagemagick.m4	2021-09-09 11:14:47.662312430 +0200
@@ -167,6 +167,14 @@ AC_DEFUN([IM_FIND_IMAGEMAGICK],[
   
     AC_MSG_RESULT([user location ${IM_IMAGEMAGICK_PREFIX}/include/ImageMagick-${IM_MAJOR_VERSION}/wand/MagickWand.h])
 
+  elif test -r "${IM_IMAGEMAGICK_PREFIX}/include/ImageMagick-${IM_MAJOR_VERSION}/MagickWand/MagickWand.h"; then
+
+    IM_INCLUDE_FORMAT="MagickWand/MagickWand.h"
+    IM_HEADER_STYLE="SEVEN"
+    AC_DEFINE([IM_MAGICKWAND_HEADER_STYLE_SEVEN], [1], [ImageMagick 7.x style header])
+
+    AC_MSG_RESULT([${IM_PREFIX}/include/ImageMagick-${IM_MAJOR_VERSION}/MagickWand/MagickWand.h])
+
   elif test -r "${IM_PREFIX}/include/ImageMagick-${IM_MAJOR_VERSION}/MagickWand/MagickWand.h"; then
 
     IM_INCLUDE_FORMAT="MagickWand/MagickWand.h"
