--- a/application/third_party/fbx/src/fbxproperty.h
+++ b/application/third_party/fbx/src/fbxproperty.h
@@ -4,6 +4,7 @@
 #include <memory>
 #include <iostream>
 #include <vector>
+#include <cstdint>
 
 namespace fbx {
 
--- a/dust3d/base/ds3_file.h
+++ b/dust3d/base/ds3_file.h
@@ -26,6 +26,7 @@
 #include <map>
 #include <string>
 #include <vector>
+#include <cstdint>
 
 namespace dust3d {
 
--- a/dust3d/mesh/solid_mesh_boolean_operation.h
+++ b/dust3d/mesh/solid_mesh_boolean_operation.h
@@ -26,6 +26,7 @@
 #include <dust3d/base/position_key.h>
 #include <dust3d/base/vector3.h>
 #include <dust3d/mesh/solid_mesh.h>
+#include <cstdint>
 #include <map>
 #include <set>
 
