--- a/host/CMakeLists.txt
+++ b/host/CMakeLists.txt
@@ -297,10 +297,8 @@ message(STATUS "Configuring Boost C++ Li
 # ALL required Boost components!
 set(UHD_BOOST_REQUIRED_COMPONENTS
     chrono
-    date_time
     filesystem
     program_options
-    system
     serialization
     thread
     unit_test_framework
--- a/host/cmake/Modules/UHDBoost.cmake
+++ b/host/cmake/Modules/UHDBoost.cmake
@@ -108,7 +108,6 @@ endif()
 list(FIND UHD_BOOST_REQUIRED_COMPONENTS "system" SYSTEM_NDX)
 if(NOT ${SYSTEM_NDX} EQUAL -1)
     list(REMOVE_AT UHD_BOOST_REQUIRED_COMPONENTS ${SYSTEM_NDX})
-    list(APPEND UHD_BOOST_REQUIRED_COMPONENTS "system")
 endif()
 
 # special library directory that's used by some Linux
