--- a/host/examples/rx_samples_to_file.cpp
+++ b/host/examples/rx_samples_to_file.cpp
@@ -15,7 +15,9 @@
 #include <boost/program_options.hpp>
 #ifdef __linux__
 #    include <boost/filesystem.hpp>
-#    include <boost/process.hpp>
+#    include <boost/process/v1/child.hpp>
+#    include <boost/process/v1/io.hpp>
+#    include <boost/process/v1/pipe.hpp>
 #endif
 #include <chrono>
 #include <complex>
@@ -29,6 +31,7 @@
 using namespace std::chrono_literals;
 
 namespace po = boost::program_options;
+namespace bp = boost::process::v1;
 
 std::mutex recv_mutex;
 
@@ -59,7 +62,7 @@ double disk_rate_check(const size_t samp
 {
     std::string err_msg =
         "Disk benchmark tool 'dd' did not run or returned an unexpected output format";
-    boost::process::ipstream pipe_stream;
+    bp::ipstream pipe_stream;
     boost::filesystem::path temp_file =
         boost::filesystem::path(file).parent_path() / boost::filesystem::unique_path();
 
@@ -69,8 +72,8 @@ double disk_rate_check(const size_t samp
         + " count=100";
 
     try {
-        boost::process::child c(
-            disk_check_proc_str, boost::process::std_err > pipe_stream);
+        bp::child c(
+            disk_check_proc_str, bp::std_err > pipe_stream);
         std::this_thread::sleep_for(std::chrono::seconds(1));
         if (c.running()) {
             c.terminate();
