Index: uhd-4.0.0.0/host/lib/usrp/dboard/e3xx/e3xx_radio_control_impl.cpp
===================================================================
--- uhd-4.0.0.0.orig/host/lib/usrp/dboard/e3xx/e3xx_radio_control_impl.cpp
+++ uhd-4.0.0.0/host/lib/usrp/dboard/e3xx/e3xx_radio_control_impl.cpp
@@ -17,6 +17,7 @@
 #include <cmath>
 #include <cstdlib>
 #include <sstream>
+#include <thread>
 
 using namespace uhd;
 using namespace uhd::usrp;
Index: uhd-4.0.0.0/host/lib/include/uhdlib/utils/rpc.hpp
===================================================================
--- uhd-4.0.0.0.orig/host/lib/include/uhdlib/utils/rpc.hpp
+++ uhd-4.0.0.0/host/lib/include/uhdlib/utils/rpc.hpp
@@ -12,7 +12,10 @@
 #include <rpc/client.h>
 #include <rpc/rpc_error.h>
 #include <boost/format.hpp>
+#include <chrono>
 #include <memory>
+#include <string>
+#include <thread>
 
 namespace {
 
