diff -Naur iODBC-3.52.8.orig/include/sqltypes.h iODBC-3.52.8/include/sqltypes.h
--- a/include/sqltypes.h	2012-03-27 05:15:02.000000000 -0400
+++ b/include/sqltypes.h	2015-07-18 14:05:24.514478508 -0400
@@ -399,19 +399,7 @@
 typedef unsigned short SQLWCHAR;
 #else
 #  include <stdlib.h>
-
-#  if defined(__cplusplus)		|| \
-      defined(_WCHAR_T)			|| \
-      defined(_WCHAR_T_DEFINED)		|| \
-      defined(_WCHAR_T_DEFINED_)	|| \
-      defined(_WCHAR_T_DECLARED)	|| \
-      defined(_BSD_WCHAR_T_DEFINED_)	|| \
-      defined(_BSD_WCHAR_T_)		|| \
-      defined(_BSD_CT_RUNE_T_)
 typedef wchar_t SQLWCHAR;
-#  else
-#    error Please make sure your system supports the wchar_t type
-#  endif
 #endif /* WIN32 */
 
 
