--- a/CMakeLists.txt	2016-09-15 01:51:53.168404310 -0500
+++ b/CMakeLists.txt	2016-09-15 16:00:14.326726871 -0500
@@ -4,6 +4,11 @@
 
 include (StepmaniaCore.cmake)
 
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${CXXFLAGS}")
+set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} ${LDFLAGS}")
+set(WITH_MINIMAID OFF)
+set(CMAKE_INSTALL_PREFIX "${DESTDIR}/opt")
+
 # The external libraries need to be included.
 add_subdirectory(extern)
 
