From 0c2b81ac1f9ff29f5012a98f530880b87f416337 Mon Sep 17 00:00:00 2001
From: Benau <Benau@users.noreply.github.com>
Date: Thu, 3 Nov 2022 11:39:25 +0800
Subject: [PATCH] Fix missing rotation when lto is used, see #4811

---
 src/physics/physical_object.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/physics/physical_object.cpp b/src/physics/physical_object.cpp
index c389b7a28b2..09d57034ece 100644
--- a/src/physics/physical_object.cpp
+++ b/src/physics/physical_object.cpp
@@ -637,10 +637,12 @@ void PhysicalObject::updateGraphics(float dt)
 
     Vec3 hpr;
     hpr.setHPR(SmoothNetworkBody::getSmoothedTrans().getRotation());
+    // Fix missing rotation when lto is used, see #4811
+    hpr *= RAD_TO_DEGREE;
 
     // This will only update the visual position, so it can be
     // called in updateGraphics()
-    m_object->move(xyz.toIrrVector(), hpr.toIrrVector()*RAD_TO_DEGREE,
+    m_object->move(xyz.toIrrVector(), hpr.toIrrVector(),
                    m_init_scale, /*updateRigidBody*/false, 
                    /* isAbsoluteCoord */true);
 }   // updateGraphics
