--- lib/irrlicht/source/Irrlicht/COSOperator.cpp.orig	2017-11-20 11:54:09.862922068 +0100
+++ lib/irrlicht/source/Irrlicht/COSOperator.cpp	2017-11-20 11:54:36.294203640 +0100
@@ -14,7 +14,7 @@
 #if !defined(_IRR_SOLARIS_PLATFORM_) && !defined(__CYGWIN__)
 #include <sys/param.h>
 #include <sys/types.h>
-#ifdef ANDROID
+#if defined(ANDROID) || !defined(__GLIBC__)
 #include <linux/sysctl.h>
 #else
 #include <sys/sysctl.h>
