Musl provides encrypt() with different signature in unistd.h.

--- src/pidgin-gpg.c
+++ src/pidgin-gpg.c
@@ -458,7 +458,7 @@ static char* verify(const char* sig_str)
 /* ------------------
  * encrypt a plain string with the key found with fingerprint fpr
  * ------------------ */
-static char* encrypt(const char* plain_str, const char* fpr)
+static char* p_encrypt(const char* plain_str, const char* fpr)
 {
 	gpgme_error_t error;
 	gpgme_ctx_t ctx;
@@ -829,7 +829,7 @@ void jabber_send_signal_cb(PurpleConnection *pc, xmlnode **packet,
 			free(bare_jid);
 
 			// encrypt message
-			enc_str = encrypt(message,fpr_to);
+			enc_str = p_encrypt(message,fpr_to);
 			if (enc_str != NULL)
 			{
 				// remove message from body
