diff -baur src/global.h src/global.h
--- a/src/global.h	2011-07-12 05:13:31.367583829 +0800
+++ b/src/global.h	2011-07-12 05:15:20.670325018 +0800
@@ -99,7 +99,7 @@
 #include <X11/xpm.h>
 #include <jack/jack.h>
 #include <jack/midiport.h>
-#include <Fl/Fl_Preferences.H>
+#include <FL/Fl_Preferences.H>
 #include "FPreset.h"
 #include "Reverb.h"
 #include "Chorus.h"
diff -baur src/process.C src/process.C
--- a/src/process.C	2011-07-12 05:13:31.367583829 +0800
+++ b/src/process.C	2011-07-12 05:15:31.293600800 +0800
@@ -28,7 +28,7 @@
 #include <fcntl.h>
 #include <sys/types.h>
 #include <unistd.h>
-#include <Fl/Fl_Preferences.H>
+#include <FL/Fl_Preferences.H>
 #include "global.h"

 int Pexitprogram, preset;
