--- Makefile.orig	2011-02-21 21:24:40.916848965 +0100
+++ Makefile	2011-02-21 21:26:53.843354443 +0100
@@ -23,23 +23,18 @@ install: build
 	install -d $(DESTDIR)$(socketdir)
 	install -p -m 755 spice-vdagent $(DESTDIR)$(bindir)
 	install -p -m 755 spice-vdagentd $(DESTDIR)$(sbindir)
-	install -d $(DESTDIR)$(initdir)
-	install -p -m 755 spice-vdagentd.sh $(DESTDIR)$(initdir)/spice-vdagentd
 	install -d $(DESTDIR)$(xdgautostartdir)
-	install -d $(DESTDIR)$(gdmautostartdir)
 	desktop-file-install --dir=$(DESTDIR)$(xdgautostartdir) \
 		spice-vdagent.desktop
-	desktop-file-install --dir=$(DESTDIR)$(gdmautostartdir) \
-		spice-vdagent.desktop
 
 clean:
 	rm -f $(TARGETS) *.o *~
 
 spice-vdagentd: vdagentd.o vdagentd-uinput.o udscs.o vdagent-virtio-port.o console-kit.o
-	$(CC) -o $@ $^ $(shell pkg-config --libs dbus-1)
+	$(CC) $(CFLAGS) $^ $(LDFLAGS) $(shell pkg-config --libs dbus-1) -o $@
 
 spice-vdagent: vdagent.o vdagent-x11.o udscs.o
-	$(CC) -o $@ $^ -lX11 -lXrandr -lXfixes
+	$(CC) $(CFLAGS) $^ $(LDFLAGS) -lX11 -lXrandr -lXfixes -o $@
 
 tag:
 	@git tag -a -m "Tag as spice-vdagent-$(VERSION)" spice-vdagent-$(VERSION)
