--- compat/imsg.c.orig	2015-05-07 12:10:26.647474701 +0200
+++ compat/imsg.c	2015-05-07 12:10:34.315439634 +0200
@@ -17,7 +17,11 @@
  */
 
 #include <sys/types.h>
+#ifdef HAVE_QUEUE_H
 #include <sys/queue.h>
+#else
+#include "compat/queue.h"
+#endif
 #include <sys/socket.h>
 #include <sys/uio.h>
 
--- compat/imsg-buffer.c.orig	2015-05-07 13:21:12.542453035 +0200
+++ compat/imsg-buffer.c	2015-05-07 13:20:34.809621188 +0200
@@ -17,7 +17,11 @@
  */
 
 #include <sys/types.h>
+#ifdef HAVE_QUEUE_H
 #include <sys/queue.h>
+#else
+#include "compat/queue.h"
+#endif
 #include <sys/socket.h>
 #include <sys/uio.h>
 
