Index: tmux-3.2/configure.ac
===================================================================
--- tmux-3.2.orig/configure.ac
+++ tmux-3.2/configure.ac
@@ -158,18 +158,12 @@ AC_FUNC_STRNLEN
 # system. When compiled it always returns NULL and crashes the program. To
 # detect this we need a more complicated test.
 AC_MSG_CHECKING([for working reallocarray])
-AC_RUN_IFELSE([AC_LANG_PROGRAM(
-		[#include <stdlib.h>],
-		[return (reallocarray(NULL, 1, 1) == NULL);]
-	)],
+AC_CHECK_FUNC(reallocarray,
 	AC_MSG_RESULT(yes),
 	[AC_LIBOBJ(reallocarray) AC_MSG_RESULT([no])]
 )
 AC_MSG_CHECKING([for working recallocarray])
-AC_RUN_IFELSE([AC_LANG_PROGRAM(
-		[#include <stdlib.h>],
-		[return (recallocarray(NULL, 1, 1, 1) == NULL);]
-	)],
+AC_CHECK_FUNC(recallocarray,
 	AC_MSG_RESULT(yes),
 	[AC_LIBOBJ(recallocarray) AC_MSG_RESULT([no])]
 )
