--- src/storage/storage_backend.h	2015-06-21 17:25:00.303000000 -0200
+++ src/storage/storage_backend.h	2015-06-21 17:26:31.583000000 -0200
@@ -31,6 +31,10 @@
 # include "vircommand.h"
 # include "storage_driver.h"
 
+#ifndef _PATH_MOUNTED
+#define _PATH_MOUNTED MOUNTED
+#endif
+
 typedef char * (*virStorageBackendFindPoolSources)(virConnectPtr conn,
                                                    const char *srcSpec,
                                                    unsigned int flags);
