Extend the guard to be also true for libressl.

--- src/ne_openssl.c	2016-09-30 10:46:00.000000000 +0200
+++ src/ne_openssl.c	2016-10-04 12:57:24.507084147 +0200
@@ -66,7 +66,7 @@
 typedef const unsigned char ne_d2i_uchar;
 #endif
 
-#if OPENSSL_VERSION_NUMBER < 0x10100000L
+#if (OPENSSL_VERSION_NUMBER < 0x10100000L) || defined(LIBRESSL_VERSION_NUMBER)
 #define X509_up_ref(x) x->references++
 #define EVP_PKEY_up_ref(x) x->references++
 #define EVP_PKEY_get0_RSA(evp) (evp->pkey.rsa)
