diff --git Makefile Makefile
index e191e72..6922391 100644
--- Makefile
+++ Makefile
@@ -47,7 +47,7 @@ CFLAGS = $(CXXFLAGS) $(CPPFLAGS) -c                \
    -I/usr/include/libchamplain-0.12/
 
 LIBS = `$(PKG_CONFIG) --libs gtk+-3.0`                                              \
-         -lrt -lpthread -llcms2 -ltiff -lpng -lraw -ljpeg                           \
+         -lrt -lexecinfo -lpthread -llcms2 -ltiff -lpng -lraw -ljpeg                           \
          -lclutter-1.0 -lclutter-gtk-1.0 -lchamplain-0.12 -lchamplain-gtk-0.12
 
 ALLFILES = fotoxx.o f.widgets.o f.file.o f.gallery.o f.albums.o f.area.o f.meta.o   \
