diff --git a/test/Makefile.in b/test/Makefile.in
index fbe2801..3714842 100644
--- a/test/Makefile.in
+++ b/test/Makefile.in
@@ -659,7 +659,7 @@ top_srcdir = @top_srcdir@
 
 # Ideally want to use -fsanitize=address and/or -fsanitize=leak too;
 #  but either bugs in Viking or GTK libs not auto cleaning up - means can't use
-AM_CFLAGS = -Wall -fsanitize=undefined -fstack-protector-all \
+AM_CFLAGS = -Wall -fstack-protector-all \
 	-I$(top_srcdir)/src \
 	$(PACKAGE_CFLAGS) $(GTK_CFLAGS)
 
