upstream: yes

From da4e97b0766243f937d11a4a4cbd6ab5aaac4e0f Mon Sep 17 00:00:00 2001
From: Rob Norris <rw_norris@hotmail.com>
Date: Sun, 9 Feb 2020 11:28:30 +0000
Subject: [PATCH] Github #80: Fix build with '--with-libexif'

---
 src/geotag_exif.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/geotag_exif.c b/src/geotag_exif.c
index 0e879a39..6e9514a7 100644
--- a/src/geotag_exif.c
+++ b/src/geotag_exif.c
@@ -362,7 +362,7 @@ VikWaypoint* a_geotag_create_waypoint_from_file ( const gchar *filename, VikCoor
 	// Not worried if none of the other fields exist, as can default the values to something
 	//
 
-	gdouble alt = VIK_DEFAULT_ALTITUDE;
+	gdouble alt = NAN;
 	ee = exif_content_get_entry (ed->ifd[EXIF_IFD_GPS], EXIF_TAG_GPS_ALTITUDE);
 	if ( ee && ee->components == 1 && ee->format == EXIF_FORMAT_RATIONAL ) {
 		alt = Rational2Double ( ee->data,
