--- Makefile.orig	2017-06-27 19:08:03.195849277 +0200
+++ Makefile	2017-06-27 19:09:14.208045274 +0200
@@ -160,7 +160,7 @@
 		  -Wall -pipe -g -fwrapv
 CFLAGS += $(OSX_ARCH)
 else
-CFLAGS := -std=gnu99 -O2 -fno-strict-aliasing \
+CFLAGS += -std=gnu99 -O2 -fno-strict-aliasing \
 		  -Wall -pipe -g -ggdb -MMD -fwrapv
 endif
 
@@ -237,7 +237,7 @@
 
 # Base LDFLAGS.
 ifeq ($(YQ2_OSTYPE),Linux)
-LDFLAGS := -L/usr/lib -lm -ldl -rdynamic
+LDFLAGS += -L/usr/lib -lm -ldl -rdynamic
 else ifeq ($(YQ2_OSTYPE),FreeBSD)
 LDFLAGS := -L/usr/local/lib -lm
 else ifeq ($(YQ2_OSTYPE),OpenBSD)
