--- src/backends/unix/signalhandler.c.orig	2015-05-31 13:02:37.500133271 +0200
+++ src/backends/unix/signalhandler.c	2015-05-31 13:02:54.932317118 +0200
@@ -28,14 +28,13 @@
 
 #include <signal.h>
 
-#ifdef __linux__
+#if defined(__linux__) && defined(__GLIBC__)
 #include <execinfo.h>
 #endif
 
 #include "../../common/header/common.h"
 
-#ifdef __linux__
-
+#if defined(__linux__) && defined(__GLIBC__)
 void
 printBacktrace(int sig)
 {
--- src/backends/unix/system.c
+++ src/backends/unix/system.c
@@ -34,6 +34,7 @@
 #include <time.h>
 #include <unistd.h>
 #include <sys/stat.h>
+#include <sys/select.h>
 
 #ifdef __APPLE__
 #include <mach/clock.h>
