/usr/man is forbidden, use /usr/share/man

--- man/meson.build
+++ man/meson.build
@@ -1,7 +1,7 @@
 configure_file(input: 'wayfire.1.in',
                output: 'wayfire.1',
                install: true,
-               install_dir: 'man',
+               install_dir: '/usr/share/man',
                configuration: conf_data)
 
 install_man(join_paths(meson.project_build_root(), 'man', 'wayfire.1'))
