--- src/api/img.hpp	2019-02-05 15:46:42.000000000 +0100
+++ -	2019-02-13 14:59:35.302813580 +0100
@@ -5,8 +5,6 @@
 #include <GLES2/gl2.h>
 #include <string>
 
-#define ulong unsigned long
-
 namespace image_io
 {
     /* Load the image from the given file, binding it to the given GL texture target
--- src/core/img.cpp	2019-02-05 15:46:42.000000000 +0100
+++ -	2019-02-13 14:59:57.760169039 +0100
@@ -19,7 +19,7 @@
 
 namespace image_io {
     using Loader = std::function<bool(const char *, GLuint)>;
-    using Writer = std::function<void(const char *name, uint8_t *pixels, ulong, ulong)>;
+    using Writer = std::function<void(const char *name, uint8_t *pixels, unsigned long, unsigned long)>;
     namespace {
         std::unordered_map<std::string, Loader> loaders;
         std::unordered_map<std::string, Writer> writers;
