diff --git a/geocode-glib/meson.build b/geocode-glib/meson.build
index 8bc2bfc..0377535 100644
--- geocode-glib/meson.build
+++ geocode-glib/meson.build
@@ -69,10 +69,9 @@ pkgconf.generate(libraries: libgcglib,
                  ])
 
 gir = find_program('g-ir-scanner', required: false)
-cross_build = meson.is_cross_build()
 enable_gir = get_option('enable-introspection')
 
-if gir.found() and not cross_build and enable_gir
+if gir.found() and enable_gir
   gir_args = [
     '--quiet',
 	'--c-include=geocode-glib/geocode-glib.h'

