In musl libc there is no <execinfo.h> and backtrace(3)

--- src/common/Debug.cpp	2010-02-05 19:22:18.000000000 +0100
+++ src/common/Debug.cpp	2015-11-05 21:38:40.336469931 +0100
@@ -407,7 +407,7 @@
 // --------------- backtrace stuff start --------------
 
 #ifndef HAVE_EXECINFO
-#	if defined(__linux__)
+#	if defined(__GLIBC__)
 #		define HAVE_EXECINFO 1
 #	elif defined(__DARWIN_VERS_1050)
 #		define HAVE_EXECINFO 1
