--- a/Make.defaults
+++ b/Make.defaults
@@ -256,11 +256,11 @@ ifeq ($(IS_MINGW32),)
 endif
 
 ifeq ($(USING_FREEBSD),1)
-CFLAGS  += $(ARCH3264) -g -O2 -Wall -Wextra -Werror \
+CFLAGS  += $(ARCH3264) -g -O2 -Wall -Wextra \
            -fno-strict-aliasing \
            -ffreestanding -fno-stack-protector
 else
-CFLAGS  += $(ARCH3264) -g -O2 -Wall -Wextra -Wno-pointer-sign -Werror \
+CFLAGS  += $(ARCH3264) -g -O2 -Wall -Wextra -Wno-pointer-sign \
            -fno-strict-aliasing \
            -ffreestanding -fno-stack-protector -fno-stack-check \
            $(if $(findstring 0,$(USING_CLANG)),-Wno-error=maybe-uninitialized,) \
