--- masterdir/builddir/deepin-terminal-3.0.0/CMakeLists.txt	2018-05-14 05:47:46.000000000 +0200
+++ CMakeLists.txt	2018-05-24 15:11:43.457568178 +0200
@@ -24,6 +24,7 @@
 pkg_check_modules(WNCK REQUIRED libwnck-3.0)
 pkg_check_modules(FONTCONFIG REQUIRED fontconfig)
 pkg_check_modules(GLIB REQUIRED glib-2.0)
+pkg_check_modules(VTE REQUIRED vte-2.91)
 
 OPTION(TEST_BUILD "Build with native testing" ON)
 
@@ -44,14 +45,6 @@
 )
 endif()
 
-ExternalProject_Add(Vte-0.29
-  SOURCE_DIR ${CMAKE_SOURCE_DIR}/3rdparty/vte-0.50.1
-  CONFIGURE_COMMAND ./configure --enable-static
-  BUILD_COMMAND make
-  INSTALL_COMMAND ""
-  BUILD_IN_SOURCE 1
-  )
-
 ExternalProject_Add(zssh-1.5c
   SOURCE_DIR ${CMAKE_SOURCE_DIR}/3rdparty/zssh-1.5c
   CONFIGURE_COMMAND ./configure
@@ -79,7 +72,7 @@
 	${RSVG_INCLUDE_DIRS}
 	${JsonGlib_INCLUDE_DIRS}
 	${LIBSECRET_INCLUDE_DIRS}
-	${CMAKE_SOURCE_DIR}/3rdparty/vte-0.50.1/src
+	${VTE_INCLUDE_DIRS}
 )
 find_library(VTE_LIB NAMES libvte-2.91.a tcmalloc_minimal)
 include_directories(${INCLUDE_PATHS})
@@ -96,6 +89,7 @@
 	${XTST_LIBRARIES}
 	${FONTCONFIG_LIBRARY}
 	${M_LIBRARIES}
+	${VTE_LIBRARIES}
 	stdc++
 	m
 	z
@@ -137,8 +131,7 @@
 	librsvg-2.0
 )
 add_executable(${target} ${VALA_C} ${CMAKE_SOURCE_DIR}/project_path.c ${CMAKE_SOURCE_DIR}/lib/font.c)
-target_link_libraries(${target} ${LIBS} ${CMAKE_SOURCE_DIR}/3rdparty/vte-0.50.1/src/.libs/libvte-2.91.a)
-add_dependencies(${target} Vte-0.29)
+target_link_libraries(${target} ${LIBS})
 
 FILE(GLOB children LIST_DIRECTORIES true po/*)
 SET(dirlist "")
