--- a/ext_src/yocto/ext/filesystem.hpp	2020-01-04 07:42:25.000000000 +0100
+++ b/ext_src/yocto/ext/filesystem.hpp	2020-01-04 07:42:25.000000000 +0100
@@ -1525,7 +1525,7 @@
     std::string msg = toUtf8(std::wstring((LPWSTR)msgBuf));
     LocalFree(msgBuf);
     return msg;
-#elif defined(GHC_OS_MACOS) || ((_POSIX_C_SOURCE >= 200112L || _XOPEN_SOURCE >= 600) && !defined(_GNU_SOURCE)) || (defined(GHC_OS_ANDROID) && __ANDROID_API__ < 23) || defined(EMSCRIPTEN)
+#elif defined(GHC_OS_MACOS) || ((_POSIX_C_SOURCE >= 200112L || _XOPEN_SOURCE >= 600) && !defined(_GNU_SOURCE)) || (defined(GHC_OS_ANDROID) && __ANDROID_API__ < 23) || defined(EMSCRIPTEN) || !defined(__GLIBC__)
     char buffer[512];
     int rc = strerror_r(code ? code : errno, buffer, sizeof(buffer));
     return rc == 0 ? (const char*)buffer : "Error in strerror_r!";
