From c9ec69850428e599fba282c32e51868b815e3019 Mon Sep 17 00:00:00 2001
From: Chih-Hsuan Yen <yan12125@gmail.com>
Date: Sun, 15 Dec 2019 09:33:12 +0800
Subject: [PATCH] Fix building with Qt 5.14

---
 daemon/core.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/daemon/core.cpp b/daemon/core.cpp
index 1be64ec..53bdf7b 100644
--- daemon/core.cpp
+++ daemon/core.cpp
@@ -33,6 +33,7 @@
 #include <QTimer>
 #include <QDBusConnectionInterface>
 #include <QDBusServiceWatcher>
+#include <QFile>
 
 #include <cstddef>
 #include <cstdlib>
