From 2e8be9e242ef146a7baa1f092e56c0ff719bdff1 Mon Sep 17 00:00:00 2001
From: Weng Xuetian <wengxt@gmail.com>
Date: Fri, 15 May 2020 21:40:08 -0700
Subject: [PATCH] Use relative path for header

Close #39
---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d10bb7f..7263bd2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -37,7 +37,7 @@ find_package(Fcitx 4.2.8 REQUIRED)
 find_package(LibIntl REQUIRED)
 endif ()
 
-set(FcitxQt5_INCLUDE_INSTALL_DIR ${CMAKE_INSTALL_FULL_INCLUDEDIR}/FcitxQt5)
+set(FcitxQt5_INCLUDE_INSTALL_DIR ${CMAKE_INSTALL_INCLUDEDIR}/FcitxQt5)
 
 if (ENABLE_LIBRARY)
 add_subdirectory(dbusaddons)
