--- a/execunix.c
+++ b/execunix.c
@@ -42,6 +42,8 @@
 # include "lists.h"
 # include "execcmd.h"
 # include <errno.h>
+# include <unistd.h>
+# include <sys/wait.h>
 
 # ifdef USE_EXECUNIX
 
@@ -102,7 +104,7 @@ execcmd(
 {
 	int pid;
 	int slot;
-	const char *argv[ MAXARGC + 1 ];	/* +1 for NULL */
+	char *argv[ MAXARGC + 1 ];	/* +1 for NULL */
 
 # ifdef USE_EXECNT
 	char *p;
--- a/fileunix.c
+++ b/fileunix.c
@@ -36,6 +36,7 @@
 # include "jam.h"
 # include "filesys.h"
 # include "pathsys.h"
+# include <unistd.h>
 
 # ifdef USE_FILEUNIX
 
--- a/make1.c
+++ b/make1.c
@@ -61,6 +61,7 @@
 # include "make.h"
 # include "command.h"
 # include "execcmd.h"
+# include <unistd.h>
 
 static void make1a( TARGET *t, TARGET *parent );
 static void make1b( TARGET *t );
--- a/mkjambase.c
+++ b/mkjambase.c
@@ -24,6 +24,7 @@
 # include <stdio.h>
 # include <string.h>
 
+int
 main( int argc, char **argv, char **envp )
 {
 	char buf[ 1024 ];
