Index: termbox-1.1.2/wscript
===================================================================
--- termbox-1.1.2.orig/wscript
+++ termbox-1.1.2/wscript
@@ -21,10 +21,6 @@ def configure(conf):
 	conf.load('gnu_dirs')
 	conf.load('compiler_c')
 	conf.env.append_unique('CFLAGS', ['-std=gnu99', '-Wall', '-Wextra', '-D_XOPEN_SOURCE'])
-	if conf.options.debug:
-		conf.env.append_unique('CFLAGS', ['-g', '-Og'])
-	else:
-		conf.env.append_unique('CFLAGS', '-O3')
 
 def build(bld):
 	bld.recurse('src')
