--- psutils/psutil.c	2021-02-05 23:12:50.141587998 +0100
+++ psutils/psutil.c	2021-02-05 23:15:17.861925136 +0100
@@ -38,7 +38,7 @@
 extern char pagelabel[BUFSIZ];
 extern int pageno;
 
-static char buffer[BUFSIZ];
+static char buffer[BUFSIZ+24];
 static long bytes = 0;
 static off_t pagescmt = 0;
 static off_t headerpos = 0;
@@ -97,7 +97,7 @@
   long r, w ;
 #endif
   char *p;
-  char buffer[BUFSIZ] ;
+  char buffer[BUFSIZ+24] ;
 #if defined(WINNT)
   struct _stat fs ;
 #else
