--- a/include/abg-comparison.h
+++ b/include/abg-comparison.h
@@ -27,6 +27,7 @@
 
 #include <ostream>
+#include <sys/types.h>
 #include "abg-cxx-compat.h"
 #include "abg-corpus.h"
 #include "abg-diff-utils.h"
 #include "abg-ini.h"
--- a/src/abg-dwarf-reader.cc
+++ b/src/abg-dwarf-reader.cc
@@ -37,6 +37,10 @@
 #include <cmath>
 #include <elfutils/libdwfl.h>
 #include <dwarf.h>
+#include <elf.h>
+#ifndef EM_OPENRISC
+#  define EM_OPENRISC 92
+#endif
 #include <algorithm>
 #include <iostream>
 #include <tr1/unordered_map>
--- a/tools/abisym.cc
+++ b/tools/abisym.cc
@@ -151,10 +151,10 @@
       cout << "could not find symbol '"
 	   << opts.symbol_name
 	   << "' in file '";
-      if (opts.absolute_path)
+      //if (opts.absolute_path)
 	cout << opts.elf_path << "'\n";
-      else
-	cout << basename(opts.elf_path);
+      //else
+	//cout << basename(opts.elf_path);
       return 0;
     }
 
