--- CMakeLists.txt.orig	2016-03-13 20:55:53.906485652 +0100
+++ CMakeLists.txt	2016-03-13 20:56:22.320483526 +0100
@@ -177,7 +177,7 @@ if(APPLE)
 		USE_SOURCE_PERMISSIONS)
 else()
 	install(
-		PROGRAMS src/cdogs-sdl${EXE_EXTENSION} src/cdogs-sdl-editor${EXE_EXTENSION}
+		PROGRAMS build/src/cdogs-sdl${EXE_EXTENSION} build/src/cdogs-sdl-editor${EXE_EXTENSION}
 		DESTINATION ${INSTALL_PREFIX}/bin)
 endif()
 
