--- src/cdogs/yajl_utils.h.orig	2016-03-15 08:17:24.508938965 +0100
+++ src/cdogs/yajl_utils.h	2016-03-15 08:18:21.256934718 +0100
@@ -28,7 +28,7 @@
 #include <stdbool.h>
 
 #include "vector.h"
-#include "yajl/api/yajl_tree.h"
+#include <yajl/yajl_tree.h>
 
 
 yajl_val YAJLReadFile(const char *filename);
--- src/cdogs/CMakeLists.txt	2016-03-15 08:35:00.996859895 +0100
+++ src/cdogs/CMakeLists.txt.orig	2016-03-15 08:34:55.824860282 +0100
@@ -210,7 +210,7 @@ add_library(cdogs STATIC
 target_link_libraries(cdogs
 	json
 	SDL_joystickbuttonnames
-	yajl_s
+	yajl
 	${SDL2_LIBRARY}
 	${SDL2_IMAGE_LIBRARIES}
 	${SDL2_MIXER_LIBRARIES}
