--- src/pkcs11/openssl.c.orig	2016-06-05 18:51:11.947157265 +0200
+++ src/pkcs11/openssl.c	2016-06-05 18:55:21.067170117 +0200
@@ -180,7 +180,7 @@
 	e = ENGINE_by_id("gost");
 	if (!e)
 	{
-#if !defined(OPENSSL_NO_STATIC_ENGINE) && !defined(OPENSSL_NO_GOST)
+#if !defined(LIBRESSL_VERSION_NUMBER) && !defined(OPENSSL_NO_STATIC_ENGINE) && !defined(OPENSSL_NO_GOST)
 		ENGINE_load_gost();
 		e = ENGINE_by_id("gost");
 #else
@@ -195,7 +195,7 @@
 			ENGINE_free(e);
 			e = NULL;
 		}
-#endif /* !OPENSSL_NO_STATIC_ENGINE && !OPENSSL_NO_GOST */
+#endif /* !defined(LIBRESSL_VERSION_NUMBER) && !defined(OPENSSL_NO_STATIC_ENGINE) && !defined(OPENSSL_NO_GOST) */
 	}
 	if (e) {
 		ENGINE_set_default(e, ENGINE_METHOD_ALL);
