--- libmcrypt-2.5.8/libltdl/configure.in
+++ libmcrypt-2.5.8/libltdl/configure.in
@@ -2,13 +2,6 @@
 
 AC_INIT(ltdl.c)
 
-dnl We shouldn't be using these internal macros of autoconf,
-dnl but CONFIG_AUX_DIR($with_auxdir) breaks automake.
-AC_ARG_WITH(auxdir,
-[  --with-auxdir=DIR   path to autoconf auxiliary files],
-[AC_CONFIG_AUX_DIRS($with_auxdir)],
-[AC_CONFIG_AUX_DIR_DEFAULT])
-
 if test -z "$enable_ltdl_install$enable_ltdl_convenience"; then
   if test -f ${srcdir}/ltmain.sh; then
     # if libltdl is libtoolized, it is assumed to be stand-alone and
