--- a/test/merge.test
+++ b/test/merge.test
@@ -39,8 +39,9 @@ Test the patch merging functionality of
 	> Applying patch %{P}c.diff
 	> Now at patch %{P}c.diff
 
-	$ quilt diff -P b.diff | grep -v "^\\(---\\|+++\\)"
+	$ quilt diff -P b.diff >/dev/null
 	> Warning: more recent patches modify files in patch %{P}b.diff
+	$ quilt diff -P b.diff 2>/dev/null | grep -v "^\\(---\\|+++\\)"
 	>~ Index: [^/]+/abc\.txt
 	> ===================================================================
 	> @@ -1,3 +1,3 @@
@@ -49,8 +50,9 @@ Test the patch merging functionality of
 	> +b+
 	>  c
 
-	$ quilt diff --combine a.diff -P b.diff | grep -v "^\\(---\\|+++\\)"
+	$ quilt diff --combine a.diff -P b.diff >/dev/null
 	> Warning: more recent patches modify files in patch %{P}b.diff
+	$ quilt diff --combine a.diff -P b.diff 2>/dev/null | grep -v "^\\(---\\|+++\\)"
 	>~ Index: [^/]+/abc\.txt
 	> ===================================================================
 	> @@ -1,3 +1,3 @@
