diff -ur a/Makefile b/Makefile
--- a/Makefile	2022-08-22 07:18:44.000000000 -0400
+++ b/Makefile	2025-11-17 14:04:07.236703698 -0500
@@ -31,7 +31,7 @@
 FEATURES=
 FEATURES_INC=
 
-CFLAGS+=-Wall -Werror -Wstrict-prototypes -Wmissing-prototypes
+CFLAGS+=-Wall -Wstrict-prototypes -Wmissing-prototypes
 CFLAGS+=-Wmissing-declarations -Wshadow -Wpointer-arith -Wcast-qual
 CFLAGS+=-Wsign-compare -Iinclude/kore -I$(OBJDIR) -std=c99 -pedantic
 CFLAGS+=-Wtype-limits -fno-common
diff -ur a/kodev/Makefile b/kodev/Makefile
--- a/kodev/Makefile	2022-08-22 07:18:44.000000000 -0400
+++ b/kodev/Makefile	2025-11-17 14:04:04.482696142 -0500
@@ -9,7 +9,7 @@
 
 S_SRC=	../src/cli.c
 
-CFLAGS+=-Wall -Werror -Wstrict-prototypes -Wmissing-prototypes
+CFLAGS+=-Wall -Wstrict-prototypes -Wmissing-prototypes
 CFLAGS+=-Wmissing-declarations -Wshadow -Wpointer-arith -Wcast-qual
 CFLAGS+=-Wsign-compare -Iincludes -std=c99 -pedantic
 CFLAGS+=-DPREFIX='"$(PREFIX)"'
diff -ur a/tools/kore-serve/conf/build.conf b/tools/kore-serve/conf/build.conf
--- a/tools/kore-serve/conf/build.conf	2022-08-22 07:18:44.000000000 -0400
+++ b/tools/kore-serve/conf/build.conf	2025-11-17 14:04:15.480726315 -0500
@@ -2,7 +2,7 @@
 kore_source=../../
 kore_flavor=NOTLS=1
 
-cflags=-std=c99 -Werror
+cflags=-std=c99
 cflags=-Wall -Wmissing-declarations -Wshadow
 cflags=-Wstrict-prototypes -Wmissing-prototypes
 cflags=-Wpointer-arith -Wcast-qual -Wsign-compare
