diff --git a/src/python.c b/src/python.c
index ab0f3c6..526d443 100644
--- a/src/python.c
+++ b/src/python.c
@@ -1192,8 +1192,14 @@ python_coro_trace(const char *label, struct python_coro *coro)
 
 	gen = (PyGenObject *)coro->obj;
 
-	if (gen->gi_frame != NULL && gen->gi_frame->f_code != NULL) {
-		code = gen->gi_frame->f_code;
+#if PY_VERSION_HEX < 0x030B0000
+	code = NULL;
+	if (gen->gi_frame != NULL)
+		code = gen->gi_frame->f_code;
+#else
+	code = gen->gi_code;
+#endif
+	if (code != NULL) {
 		func = PyUnicode_AsUTF8AndSize(code->co_name, NULL);
 		file = PyUnicode_AsUTF8AndSize(code->co_filename, NULL);
 
@@ -1206,8 +1206,13 @@ python_coro_trace(const char *label, struct python_coro *coro)
 		fname = "unknown";
 	}
 
+#if PY_VERSION_HEX <= 0x030B0000
 	if (gen->gi_frame != NULL)
 		line = PyFrame_GetLineNumber(gen->gi_frame);
+#else
+	if (code != NULL)
+		line = PyCode_Addr2Line(code, 0);
+#endif
 	else
 		line = -1;
 
