Index: console-setup-1.211/Makefile
===================================================================
--- console-setup-1.211.orig/Makefile
+++ console-setup-1.211/Makefile
@@ -28,9 +28,9 @@ build-common: gzipped-acmfiles
 install-common: build-common
 	install -d  $(bootprefix)/bin/
 	install setupcon $(bootprefix)/bin/
-	install -d $(etcdir)/default
-	install -m 644 config/keyboard $(etcdir)/default/
-	install -m 644 config/console-setup $(etcdir)/default/
+	install -d $(etcdir)/console-setup
+	install -m 644 config/keyboard $(etcdir)/console-setup/
+	install -m 644 config/console-setup $(etcdir)/console-setup/
 	install -d $(mandir)/man1/
 	install -m 644 man/setupcon.1 $(mandir)/man1/
 	install -d $(mandir)/man5/
Index: console-setup-1.211/setupcon
===================================================================
--- console-setup-1.211.orig/setupcon
+++ console-setup-1.211/setupcon
@@ -356,9 +356,9 @@ Sets up the font and the keyboard on Lin
 
 If VARIANT is not specified setupcon looks for the configuration files
 (in this order) ~/.console-setup and if this doesn't exist then the
-combination /etc/default/keyboard + /etc/default/console-setup.  When
+combination /etc/console-setup/keyboard + /etc/console-setup/console-setup.  When
 a VARIANT is specified then setupcon looks for the configuration files
-~/.console-setup.VARIANT and /etc/default/console-setup.VARIANT.
+~/.console-setup.VARIANT and /etc/console-setup/console-setup.VARIANT.
 EOF
 	    exit 0
 	    ;;
@@ -407,14 +407,14 @@ fi
 
 USER_CONFIG=${HOME}/.console-setup"$VARIANT"
 USER_CONFIG2=${HOME}/.keyboard"$VARIANT"
-MAIN_CONFIG=/etc/default/keyboard"$VARIANT"
+MAIN_CONFIG=/etc/console-setup/keyboard"$VARIANT"
 [ -f "$MAIN_CONFIG" ] \
-    || [ ! -f "$installdir"/etc/default/keyboard"$VARIANT" ] \
-    || MAIN_CONFIG="$installdir"/etc/default/keyboard"$VARIANT"
-MAIN_CONFIG2=/etc/default/console-setup"$VARIANT"
+    || [ ! -f "$installdir"/etc/console-setup/keyboard"$VARIANT" ] \
+    || MAIN_CONFIG="$installdir"/etc/console-setup/keyboard"$VARIANT"
+MAIN_CONFIG2=/etc/console-setup/console-setup"$VARIANT"
 [ -f "$MAIN_CONFIG2" ] \
-    || [ ! -f "$installdir"/etc/default/console-setup"$VARIANT" ] \
-    || MAIN_CONFIG2="$installdir"/etc/default/console-setup"$VARIANT"
+    || [ ! -f "$installdir"/etc/console-setup/console-setup"$VARIANT" ] \
+    || MAIN_CONFIG2="$installdir"/etc/console-setup/console-setup"$VARIANT"
 
 if [ -f "$USER_CONFIG" -o -f "$USER_CONFIG2" ]; then
     CONFIG="$USER_CONFIG"
