From ff48fe5c97b2d36cecf7492c4c04b79cae8099d2 Mon Sep 17 00:00:00 2001
From: lemmi <lemmi@nerd2nerd.org>
Date: Thu, 4 May 2023 09:27:39 +0200
Subject: [PATCH] disable -Werror

---
 cmake/Modules/CompilerConfig.cmake | 1 -
 1 file changed, 1 deletion(-)

diff --git a/cmake/Modules/CompilerConfig.cmake b/cmake/Modules/CompilerConfig.cmake
index 41c56da8b..b8f8bf330 100644
--- a/cmake/Modules/CompilerConfig.cmake
+++ b/cmake/Modules/CompilerConfig.cmake
@@ -101,7 +101,6 @@ else()
 
   option(CALM_DEPRECATION "Keep deprecated-declarations as warnings" OFF)
   add_compile_options(
-    -Werror
     -Wextra
     -Wvla
     -Wswitch
-- 
2.40.1

