From 8105e17c547c80fc9940a46a0f2ae385d48b9ebb Mon Sep 17 00:00:00 2001
From: Daniel Kolesa <daniel@octaforge.org>
Date: Tue, 2 May 2023 16:17:41 +0200
Subject: [PATCH 16/27] libc++: musl locale workarounds

---
 libcxx/include/locale | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/libcxx/include/locale b/libcxx/include/locale
index 874866f69..befed29bf 100644
--- a/libcxx/include/locale
+++ b/libcxx/include/locale
@@ -758,7 +758,11 @@ __num_get_signed_integral(const char* __a, const char* __a_end,
         __libcpp_remove_reference_t<decltype(errno)> __save_errno = errno;
         errno = 0;
         char *__p2;
+#if defined(__linux__) && !defined(__GLIBC__)
+        long long __ll = strtoll(__a, &__p2, __base);
+#else
         long long __ll = strtoll_l(__a, &__p2, __base, _LIBCPP_GET_C_LOCALE);
+#endif
         __libcpp_remove_reference_t<decltype(errno)> __current_errno = errno;
         if (__current_errno == 0)
             errno = __save_errno;
@@ -798,7 +802,11 @@ __num_get_unsigned_integral(const char* __a, const char* __a_end,
         __libcpp_remove_reference_t<decltype(errno)> __save_errno = errno;
         errno = 0;
         char *__p2;
+#if defined(__linux__) && !defined(__GLIBC__)
+        unsigned long long __ll = strtoull(__a, &__p2, __base);
+#else
         unsigned long long __ll = strtoull_l(__a, &__p2, __base, _LIBCPP_GET_C_LOCALE);
+#endif
         __libcpp_remove_reference_t<decltype(errno)> __current_errno = errno;
         if (__current_errno == 0)
             errno = __save_errno;
-- 
2.39.0

