--- a/meson.build
+++ b/meson.build
@@ -20,7 +20,6 @@ glib_dep = dependency('glib-2.0', version: '>= 2.68')
 gio_dep = dependency('gio-2.0')
 gio_unix_dep = dependency('gio-unix-2.0')
 pipewire_dep = dependency('libpipewire-0.3', version: '>= 0.3.0')
-systemd_dep = dependency('systemd', required: get_option('systemd'))
 libsecret_dep = dependency('libsecret-1')
 libnotify_dep = dependency('libnotify')
 epoxy_dep = dependency('epoxy')
@@ -83,15 +82,6 @@ configure_file(input: 'config.h.meson',
 configinc = include_directories('.')
 
 servicedir = get_option('systemd_user_unit_dir')
-if systemd_dep.found()
-  if servicedir == ''
-    servicedir = systemd_dep.get_pkgconfig_variable('systemduserunitdir')
-  endif
-
-  if servicedir == ''
-    error('Couldn\'t determine systemd user unit service directory')
-  endif
-endif
 
 top_srcdir = meson.current_source_dir()
 builddir = meson.current_build_dir()
