From b903d0dc91f482b348e5840e5a1df599ebd5e6f9 Mon Sep 17 00:00:00 2001
From: David Tardon <dtardon@redhat.com>
Date: Sun, 30 Aug 2015 13:57:14 +0200
Subject: [PATCH] fix build with boost 1.59

---
 src/lib/Makefile.am | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/lib/Makefile.am b/src/lib/Makefile.am
index 9931a22..35d35be 100644
--- src/lib/Makefile.am
+++ src/lib/Makefile.am
@@ -64,6 +64,7 @@ libe_book_@EBOOK_MAJOR_VERSION@_@EBOOK_MINOR_VERSION@_la_LDFLAGS = $(version_inf
 libe_book_@EBOOK_MAJOR_VERSION@_@EBOOK_MINOR_VERSION@_la_SOURCES = \
 	EBOOKDocument.cpp
 
+libe_book_internal_la_CPPFLAGS = -DBOOST_ERROR_CODE_HEADER_ONLY
 libe_book_internal_la_SOURCES = \
 	EBOOKBitStream.cpp \
 	EBOOKBitStream.h \
-- 
2.6.2

