commit 44136a2e5c52623b0e62c38846f511f307615ce2
Author: q66 <daniel@octaforge.org>
Date:   Sun May 3 19:52:09 2020 +0200

    fix build with libgit2-1.0.0

diff --git a/src/plugins/git/daemon/ipc-git-repository-impl.c b/src/plugins/git/daemon/ipc-git-repository-impl.c
index 7f01042..2ccc4df 100644
--- a/src/plugins/git/daemon/ipc-git-repository-impl.c
+++ b/src/plugins/git/daemon/ipc-git-repository-impl.c
@@ -33,11 +33,7 @@
 #include "ipc-git-types.h"
 #include "ipc-git-util.h"
 
-#if LIBGIT2_SOVERSION >= 28
 G_DEFINE_AUTO_CLEANUP_CLEAR_FUNC (git_buf, git_buf_dispose)
-#else
-G_DEFINE_AUTO_CLEANUP_CLEAR_FUNC (git_buf, git_buf_free)
-#endif
 
 typedef enum
 {
