--- src/plugins/sysprof/gbp-sysprof-plugin.c	2018-03-14 03:10:22.000000000 +0100
+++ src/plugins/sysprof/gbp-sysprof-plugin.c	2018-04-10 15:20:46.416250960 +0200
@@ -16,6 +16,11 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include <stdlib.h>
+#if !defined(__GLIBC__)
+#include <bits/reg.h>	/* for __WORDSIZE */
+#endif
+
 #include <libpeas/peas.h>
 #include <ide.h>
 #include <sysprof.h>
--- src/plugins/sysprof/gbp-sysprof-perspective.c	2018-03-14 03:10:22.000000000 +0100
+++ src/plugins/sysprof/gbp-sysprof-perspective.c	2018-04-10 15:12:49.793279369 +0200
@@ -17,6 +17,10 @@
  */
 
 #define G_LOG_DOMAIN "gbp-sysprof-perspective"
+#include <stdlib.h>
+#if !defined(__GLIBC__)
+#include <bits/reg.h>	/* for __WORDSIZE */
+#endif
 
 #include <glib/gi18n.h>
 #include <sysprof.h>
--- src/plugins/sysprof/gbp-sysprof-workbench-addin.c	2018-03-14 03:10:22.000000000 +0100
+++ src/plugins/sysprof/gbp-sysprof-workbench-addin.c	2018-04-10 15:19:28.473255605 +0200
@@ -16,6 +16,11 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include <stdlib.h>
+#if !defined(__GLIBC__)
+#include <bits/reg.h>	/* for __WORDSIZE */
+#endif
+
 #include <glib/gi18n.h>
 #include <sysprof.h>
 
