--- a/data/meson.build
+++ b/data/meson.build
@@ -2,7 +2,6 @@ if host_machine.system() != 'windows'
   desktop = 'remote-viewer.desktop'
 
   i18n.merge_file (
-    desktop,
     type: 'desktop',
     input: desktop + '.in',
     output: desktop,
@@ -14,7 +13,6 @@ if host_machine.system() != 'windows'
   mimetypes = 'virt-viewer-mime.xml'
 
   i18n.merge_file (
-    mimetypes,
     type: 'xml',
     input: mimetypes + '.in',
     output: mimetypes,
@@ -27,7 +25,6 @@ if host_machine.system() != 'windows'
   metainfo = 'remote-viewer.appdata.xml'
 
   i18n.merge_file (
-    metainfo,
     type: 'xml',
     input: metainfo + '.in',
     output: metainfo,
