From 4acfdaf95adb0a05c2a25550bdde036c865902f4 Mon Sep 17 00:00:00 2001
From: Julien Cristau <jcristau@debian.org>
Date: Mon, 02 Dec 2013 15:26:24 +0000
Subject: Use FT_*_H macros instead of including <freetype/*.h>

freetype moved its headers around in 2.5.1.

Signed-off-by: Julien Cristau <jcristau@debian.org>
---
diff --git a/src/xftglyphs.c b/src/xftglyphs.c
index 2f3dc5a..4b5fb82 100644
--- src/xftglyphs.c
+++ src/xftglyphs.c
@@ -21,10 +21,10 @@
  */
 
 #include "xftint.h"
-#include <freetype/ftoutln.h>
-#include <freetype/ftlcdfil.h>
+#include FT_OUTLINE_H
+#include FT_LCD_FILTER_H
 
-#include <freetype/ftsynth.h>
+#include FT_SYNTHESIS_H
 
 /*
  * Validate the memory info for a font
--
cgit v0.9.0.2-2-gbebe

