Description: CBR: High quality settings (e.g, -q 0) degrade quality over -q 4
Origin: https://sourceforge.net/p/lame/bugs/516/#a26d
Forwarded: https://sourceforge.net/p/lame/bugs/516/

--- a/libmp3lame/lame.c
+++ b/libmp3lame/lame.c
@@ -1050,6 +1050,11 @@ lame_init_params(lame_global_flags * gfp
             if (gfp->quality < 0)
                 gfp->quality = LAME_DEFAULT_QUALITY;
 
+            /* VBR and ABR don't give good results with noise shaping
+             * settings used in quality levels below 4.
+             */
+            if (gfp->quality < 4)
+                gfp->quality = 4;
 
             if (cfg->vbr == vbr_off)
                 (void) lame_set_VBR_mean_bitrate_kbps(gfp, gfp->brate);

