--- CMakeLists.txt.orig	2015-09-21 18:07:53.311175825 +0200
+++ CMakeLists.txt	2015-09-21 18:08:14.138130904 +0200
@@ -415,8 +415,7 @@ if(NOT ANDROID)
 			add_definitions(-DHAVE_WAYLAND=0)
 		endif(USE_WAYLAND AND WAYLAND_FOUND)
 
-		# Note: We do not need to explicitly check for X11 as it is done in the cmake
-		# FindOpenGL module on linux.
+		include(FindX11)
 		if(USE_X11 AND X11_FOUND)
 			set(USE_X11 1)
 			add_definitions(-DHAVE_X11=1)
