--- ./Source/Core/Common/Src/Thread.cpp.orig	2015-10-21 12:01:13.444981266 -0400
+++ ./Source/Core/Common/Src/Thread.cpp	2015-10-21 12:01:57.739979915 -0400
@@ -123,7 +123,7 @@
 {
 #ifdef __APPLE__
 	pthread_setname_np(szThreadName);
-#else
+#elif defined __GLIBC__
 	pthread_setname_np(pthread_self(), szThreadName);
 #endif
 }

--- ./Source/Core/Core/Src/x64MemTools.cpp.orig	2015-10-21 12:16:39.401953012 -0400
+++ ./Source/Core/Core/Src/x64MemTools.cpp	2015-10-21 12:16:55.380952525 -0400
@@ -55,7 +55,7 @@
 namespace EMM
 {
 
-#if defined __APPLE__ || defined __linux__ || defined __FreeBSD__
+#if defined __APPLE__ || defined __GLIBC__ || defined __FreeBSD__
 #include <execinfo.h>
 void print_trace(const char * msg)
 {
