--- a/Source/Core/Core/HW/EXI/BBA/BuiltIn.h	2023-05-11 17:02:30.000000000 -0400
+++ b/Source/Core/Core/HW/EXI/BBA/BuiltIn.h	2024-06-04 09:24:19.476262345 -0400
@@ -20,8 +20,10 @@
 constexpr u16 TCP_FLAG_FIN = 0x1;
 constexpr u16 TCP_FLAG_RST = 0x4;

+#if not defined(__GLIBC__) and (defined(__aarch64__) || defined(__arm64__) || defined(__x86_64__) )
 constexpr u16 IP_PROTOCOL = 0x800;
 constexpr u16 ARP_PROTOCOL = 0x806;
+#endif

 constexpr u8 MAX_TCP_BUFFER = 4;
 constexpr u16 MAX_UDP_LENGTH = 1500;
