--- a/gtk/configure.ac
+++ b/gtk/configure.ac
@@ -199,7 +199,7 @@
 
 AM_CONDITIONAL([MINGW], [test "x$mingw_flag" = "xyes"])
 
-HB_LIBS="$HB_LIBS -lhandbrake -lavformat -lavfilter -lavcodec -lavutil -ldav1d -lswresample -lpostproc -ldvdnav -ldvdread -lmp3lame -lvorbis -lvorbisenc -logg -lswscale -ltheoraenc -ltheoradec -lvpx -lz -lbz2 -lbluray -lass -lfontconfig -lfreetype -lxml2 -ljansson -lopus -lspeex -llzma"
+HB_LIBS="$HB_LIBS -lhandbrake -lavformat -lavfilter -lavcodec -lavutil -ldav1d -lswresample -lpostproc -ldvdnav -ldvdread -lmp3lame -lvorbis -lvorbisenc -logg -lswscale -ltheoraenc -ltheoradec -lvpx -lz -lbz2 -lbluray -lass -lfontconfig -lfreetype -lxml2 -ljansson -lopus -lspeex -llzma -ldl"
 HB_CPPFLAGS="$HB_CPPFLAGS $HBINC"
 
 PKG_CHECK_MODULES([x264], [x264], sys_x264=yes, sys_x264=no)

--- a/test/module.defs
+++ b/test/module.defs
@@ -69,6 +69,9 @@ else ifeq ($(HOST.system),linux)
 ifeq (1, $(FEATURE.numa))
     TEST.GCC.l += numa
 endif
+ifeq (1, $(FEATURE.x265))
+    TEST.GCC.l += x265
+endif
 else ifeq ($(HOST.system),kfreebsd)
     TEST.GCC.l += pthread dl m
 else ifeq ($(HOST.system),freebsd)
