--- a/libhb/module.defs
+++ b/libhb/module.defs
@@ -52,11 +52,7 @@
 LIBHB.GCC.I += $(LIBHB.build/) $(CONTRIB.build/)include

 ifeq (1,$(FEATURE.qsv))
-    ifeq ($(HOST.system),freebsd)
-        LIBHB.GCC.I += $(LOCALBASE)/include/vpl
-    else
-        LIBHB.GCC.I += $(CONTRIB.build/)include/vpl
-    endif
+    LIBHB.GCC.I += /usr/include/vpl
 endif

 ifneq (,$(filter $(HOST.system),freebsd netbsd openbsd))
