--- a/libhb/encsvtav1.c
+++ b/libhb/encsvtav1.c
@@ -16,6 +16,7 @@
 #include "handbrake/extradata.h"
 
 #include "libavutil/avutil.h"
+#include "svt-av1/EbSvtAv1.h"
 #include "svt-av1/EbSvtAv1ErrorCodes.h"
 #include "svt-av1/EbSvtAv1Enc.h"
 #include "svt-av1/EbSvtAv1Metadata.h"
@@ -99,7 +100,11 @@ int encsvtInit(hb_work_object_t *w, hb_j
     EbErrorType svt_ret;
     int ret;
 
+#if SVT_AV1_VERSION_MAJOR > 2
     svt_ret = svt_av1_enc_init_handle(&pv->svt_handle, &pv->enc_params);
+#else
+    svt_ret = svt_av1_enc_init_handle(&pv->svt_handle, pv, &pv->enc_params);
+#endif
     if (svt_ret != EB_ErrorNone)
     {
         hb_error("encsvtav1: error initializing encoder handle");
