*** a/libhb/encx265.c	2024-06-21 01:12:14.000000000 -0400
--- b/libhb/encx265.c	2025-05-31 21:50:54.619402956 -0400
*************** static hb_buffer_t* x265_encode(hb_work_
*** 778,781 ****
      save_frame_info(pv, in);
! 
!     ret = pv->api->encoder_encode(pv->x265, &nal, &nnal, &pic_in, &pic_out);
  
--- 778,781 ----
      save_frame_info(pv, in);
!     x265_picture *pic_out_ptr = &pic_out;
!     ret = pv->api->encoder_encode(pv->x265, &nal, &nnal, &pic_in, &pic_out_ptr);
  
*************** int encx265Work(hb_work_object_t *w, hb_
*** 807,814 ****
          hb_buffer_list_clear(&list);
! 
          // flush delayed frames
          while (pv->api->encoder_encode(pv->x265, &nal,
!                                        &nnal, NULL, &pic_out) > 0)
          {
!             hb_buffer_t *buf = nal_encode(w, &pic_out, nal, nnal);
              hb_buffer_list_append(&list, buf);
--- 807,814 ----
          hb_buffer_list_clear(&list);
!         x265_picture *pic_out_ptr = &pic_out;
          // flush delayed frames
          while (pv->api->encoder_encode(pv->x265, &nal,
!                                        &nnal, NULL, &pic_out_ptr) > 0)
          {
!             hb_buffer_t *buf = nal_encode(w, pic_out_ptr, nal, nnal);
              hb_buffer_list_append(&list, buf);
