--- flowblade-trunk/flowblade	2019-02-04 02:29:07.000000000 -0700
+++ flowblade-trunk/flowblade	2019-07-06 17:48:47.295155396 -0600
@@ -34,7 +34,7 @@ print "Launch script dir:", launch_dir
 
 # Update sys.path to include modules
 # When running on distro
-if launch_dir == "/usr/bin":
+if launch_dir in {"/usr/bin", "/bin", "/usr/local/bin"}:
     print "Running from installation..."
     modules_path = "/usr/share/flowblade/Flowblade"
     if not os.path.isdir(modules_path):
