--- flowblade-trunk/flowblade	2019-08-31 20:09:32.877338976 +0200
+++ flowblade-trunk/flowblade	2019-08-31 20:09:50.706337258 +0200
@@ -34,7 +34,7 @@
 
 # Update sys.path to include modules
 # When running on distro
-if launch_dir in {"bin", "/usr/bin", "/usr/local/bin"}:
+if launch_dir in {"/bin", "/usr/bin", "/usr/local/bin"}:
     print "Running from installation..."
     modules_path = "/usr/share/flowblade/Flowblade"
     if not os.path.isdir(modules_path):
