diff --git flowblade-trunk/flowblade flowblade-trunk/flowblade
index faeba629..9acad76e 100755
--- flowblade-trunk/flowblade
+++ flowblade-trunk/flowblade
@@ -34,7 +34,7 @@ print ("Launch script dir:", launch_dir)
 
 # Update sys.path to include modules.
 # When running on distro.
-if launch_dir == "/usr/bin":
+if os.path.realpath(launch_dir) == "/usr/bin":
     print ("Running from installation...")
     modules_path = "/usr/share/flowblade/Flowblade"
     if not os.path.isdir(modules_path):
