From 4868300e3295dce16f3672176caf4dad413d1d6d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ferm=C3=ADn=20Olaiz?= <fermino@users.noreply.github.com>
Date: Tue, 9 Jan 2024 17:59:28 -0300
Subject: [PATCH] install-helpers.sh: fix date template syntax

---
 zfsbootmenu/install-helpers.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/zfsbootmenu/install-helpers.sh b/zfsbootmenu/install-helpers.sh
index 5af273e7d..8eb1ef41c 100644
--- a/zfsbootmenu/install-helpers.sh
+++ b/zfsbootmenu/install-helpers.sh
@@ -262,7 +262,7 @@ install_zbm_osver() {
   local build_date
   [ -r "${zfsbootmenu_module_root}/zbm-release" ] || return 0
   zbm_install_file "${zfsbootmenu_module_root}/zbm-release" "/etc/zbm-release"
-  if build_date="$(date %Y-%m-%d)"; then
+  if build_date="$(date '+%Y-%m-%d')"; then
     cat >> "${BUILDROOT}/etc/zbm-release" <<-EOF
 	BUILD_ID="${build_date}"
 	EOF
