From 20210a4abd417f067483875fbb8aaabe56a344d0 Mon Sep 17 00:00:00 2001
From: q66 <daniel@octaforge.org>
Date: Wed, 23 Jun 2021 03:02:10 +0200
Subject: [PATCH] fix broken shared library creation

---
 mk/lib.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mk/lib.mk b/mk/lib.mk
index c397941..ca29e8f 100644
--- a/bmake/mk/lib.mk
+++ b/bmake/mk/lib.mk
@@ -170,7 +170,7 @@ LD_solib= lib${LIB}_pic.a
 .elif ${TARGET_OSNAME} == "Linux"
 SHLIB_LD = ${CC}
 # this is ambiguous of course
-LD_shared=-shared -Wl,"-soname lib${LIB}.so.${SHLIB_MAJOR}"
+LD_shared=-shared -Wl,-soname,lib${LIB}.so.${SHLIB_MAJOR}
 LD_solib= -Wl,--whole-archive lib${LIB}_pic.a -Wl,--no-whole-archive
 .if ${COMPILER_TYPE} == "gcc"
 # Linux uses GNU ld, which is a multi-pass linker
-- 
2.31.1

