--- CMakeLists.txt.orig
+++ CMakeLists.txt
@@ -109,8 +109,8 @@ function(new_ec_test_full_ec_file_path name ec_file src_file regex)
     set_tests_properties(${name}-open ${name}-parse PROPERTIES PASS_REGULAR_EXPRESSION "${regex}")
 endfunction()
 
-add_subdirectory(tests/glob)
-add_subdirectory(tests/properties)
-add_subdirectory(tests/parser)
-add_subdirectory(tests/filetree)
+#add_subdirectory(tests/glob)
+#add_subdirectory(tests/properties)
+#add_subdirectory(tests/parser)
+#add_subdirectory(tests/filetree)
 #add_subdirectory(tests/cli)

