From b64df1afa8be8bf784e970aa47973ec420ac4381 Mon Sep 17 00:00:00 2001
From: Jeremy Bicha <jbicha@ubuntu.com>
Date: Thu, 7 Dec 2017 17:18:05 -0500
Subject: [PATCH] Always install executables

---
 CMakeLists.txt | 10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ecfbb83..29c45b9 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -260,12 +260,10 @@ generate_pkg_config ("${CMAKE_CURRENT_BINARY_DIR}/libwoff2enc.pc"
   LIBRARIES woff2enc)
 
 # Installation
-if (NOT BUILD_SHARED_LIBS)
-  install(
-    TARGETS woff2_decompress woff2_compress woff2_info
-    RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
-  )
-endif()
+install(
+  TARGETS woff2_decompress woff2_compress woff2_info
+  RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
+)
 
 install(
   TARGETS woff2common woff2dec woff2enc

