--- a/Telegram/lib_base/base/platform/linux/base_info_linux.cpp	2020-11-21 11:09:51.647955733 +0100
+++ b/Telegram/lib_base/base/platform/linux/base_info_linux.cpp	2020-11-21 11:09:51.647955733 +0100
@@ -12,11 +12,6 @@
 #include <QtCore/QDate>
 #include <QtGui/QGuiApplication>
 
-// this file is used on both Linux & BSD
-#ifdef Q_OS_LINUX
-#include <gnu/libc-version.h>
-#endif // Q_OS_LINUX
-
 namespace Platform {
 namespace {
 
@@ -95,11 +90,7 @@
 
 	if (IsLinux32Bit()) {
 		return QDate(2020, 9, 1);
-	} else if (libcName == qstr("glibc") && !libcVersion.isEmpty()) {
-		if (QVersionNumber::fromString(libcVersion) < QVersionNumber(2, 23)) {
-			return QDate(2020, 9, 1); // Older than Ubuntu 16.04.
-		}
-	}
+	} 
 
 	return QDate();
 }
@@ -123,22 +114,10 @@
 }
 
 QString GetLibcName() {
-#ifdef Q_OS_LINUX
-	return "glibc";
-#endif // Q_OS_LINUX
-
 	return QString();
 }
 
 QString GetLibcVersion() {
-#ifdef Q_OS_LINUX
-	static const auto result = [&] {
-		const auto version = QString::fromLatin1(gnu_get_libc_version());
-		return QVersionNumber::fromString(version).isNull() ? QString() : version;
-	}();
-	return result;
-#endif // Q_OS_LINUX
-
 	return QString();
 }
 
