From af5f8d19f0fdded987beee4568f02cc8dcbd20f4 Mon Sep 17 00:00:00 2001
From: John Zimmermann <johnz@posteo.net>
Date: Thu, 2 Jan 2020 12:58:45 +0100
Subject: [PATCH] PPC config

---
 base/build_config.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git Telegram/lib_base/base/build_config.h Telegram/lib_base/base/build_config.h
index a02c9b0..b3f4860 100644
--- Telegram/lib_base/base/build_config.h
+++ Telegram/lib_base/base/build_config.h
@@ -46,9 +46,9 @@
 #define ARCH_CPU_X86_FAMILY 1
 #define ARCH_CPU_X86 1
 #define ARCH_CPU_32_BITS 1
-#elif defined(__aarch64__)
+#elif defined(__aarch64__) || defined(__powerpc64__)
 #define ARCH_CPU_64_BITS 1
-#elif defined(_M_ARM) || defined(__arm__)
+#elif defined(_M_ARM) || defined(__arm__) || defined(__powerpc__)
 #define ARCH_CPU_32_BITS 1
 #else
 #error Please add support for your architecture in base/build_config.h
-- 
2.24.1

