From ef30081973681f4f8d261917fa763b00adfbe6e9 Mon Sep 17 00:00:00 2001
From: John Zimmermann <johnz@posteo.net>
Date: Thu, 2 Jan 2020 13:10:31 +0100
Subject: [PATCH] gtk3

---
 Telegram/CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git Telegram/CMakeLists.txt Telegram/CMakeLists.txt
index cf5da1c13..0d155d54a 100644
--- Telegram/CMakeLists.txt
+++ Telegram/CMakeLists.txt
@@ -1018,8 +1018,8 @@ elseif (LINUX)
     if (NOT TDESKTOP_DISABLE_GTK_INTEGRATION)
         find_package(PkgConfig REQUIRED)
 
-        pkg_check_modules(GTK2 REQUIRED gtk+-2.0)
-        target_include_directories(Telegram PRIVATE ${GTK2_INCLUDE_DIRS})
+        pkg_check_modules(GTK3 REQUIRED gtk+-3.0)
+        target_include_directories(Telegram PRIVATE ${GTK3_INCLUDE_DIRS})
         target_compile_options(Telegram PRIVATE -Wno-register)
 
         set(appindicator_packages
-- 
2.24.1

