--- lib/xdg/telegramdesktop.appdata.xml 2019-12-31 16:06:54.000000000 +0100
+++ -	2020-01-03 11:05:49.315124450 +0100
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 <component type="desktop">
-    <id>org.telegram.desktop</id>
+    <id>telegramdesktop.desktop</id>
     <metadata_license>CC0-1.0</metadata_license>
     <project_license>GPL-3.0</project_license>
     <name>Telegram Desktop</name>
--- lib/xdg/telegramdesktop.appdata.xml	2019-12-31 16:06:54.000000000 +0100
+++ -	2020-01-03 11:05:49.315124450 +0100
@@ -63,6 +63,6 @@
         <content_attribute id="money-gambling">none</content_attribute>
     </content_rating>
     <provides>
-        <binary>telegram-desktop</binary>
+        <binary>Telegram</binary>
     </provides>
 </component>
--- lib/xdg/telegramdesktop.desktop	2019-12-31 16:06:54.000000000 +0100
+++ -	2020-01-03 11:06:05.345353255 +0100
@@ -2,7 +2,7 @@
 Version=1.0
 Name=Telegram Desktop
 Comment=Official desktop version of Telegram messaging app
-Exec=telegram-desktop -- %u
+Exec=Telegram -- %u
 Icon=telegram
 Terminal=false
 StartupWMClass=TelegramDesktop
--- lib/xdg/tg.protocol	2019-12-31 16:06:54.000000000 +0100
+++ -	2020-01-03 11:06:20.712139153 +0100
@@ -1,5 +1,5 @@
 [Protocol]
-exec=/usr/bin/telegram-desktop -- %u
+exec=Telegram -- %u
 protocol=tg
 input=none
 output=none
