--- Telegram/SourceFiles/core/utils.cpp	2018-02-18 16:50:35.666997894 +0100
+++ -	2018-02-18 16:51:32.647313078 +0100
@@ -295,7 +295,9 @@
 		av_lockmgr_register(nullptr);
 
 		CRYPTO_cleanup_all_ex_data();
+#if !defined(LIBRESSL_VERSION_NUMBER)
 		FIPS_mode_set(0);
+#endif
 		ENGINE_cleanup();
 		CONF_modules_unload(1);
 		ERR_remove_state(0);
--- Telegram/SourceFiles/mtproto/rsa_public_key.cpp	2018-01-03 11:46:01.000000000 +0100
+++ -	2018-02-18 17:04:49.956395054 +0100
@@ -16,7 +16,7 @@
 namespace MTP {
 namespace internal {
 namespace {
-#if OPENSSL_VERSION_NUMBER < 0x10100000L
+#if OPENSSL_VERSION_NUMBER < 0x10100000L || defined(LIBRESSL_VERSION_NUMBER)
 
 // This is a key setter for compatibility with OpenSSL 1.0
 int RSA_set0_key(RSA *r, BIGNUM *n, BIGNUM *e, BIGNUM *d) {
