--- Telegram/ThirdParty/libtgvoip/os/linux/AudioInputPulse.cpp	2018-02-18 16:15:10.850216642 +0100
+++ -	2018-02-18 16:17:18.277060386 +0100
@@ -4,6 +4,9 @@
 // you should have received with this source code distribution.
 //
 
+#if !defined(__GLIBC__)
+# include <libgen.h>
+#endif
 #include <assert.h>
 #include <dlfcn.h>
 #include <unistd.h>
--- Telegram/ThirdParty/libtgvoip/os/linux/AudioOutputPulse.cpp	2017-12-27 19:47:58.000000000 +0100
+++ -	2018-02-18 16:18:27.461801108 +0100
@@ -4,7 +4,9 @@
 // you should have received with this source code distribution.
 //
 
-
+#if !defined(__GLIBC__)
+# include <libgen.h>
+#endif
 #include <assert.h>
 #include <dlfcn.h>
 #include <unistd.h>
