diff --git Telegram/gyp/qt.gypi Telegram/gyp/qt.gypi
index c9eca6ad..1404f91b 100644
--- Telegram/gyp/qt.gypi
+++ Telegram/gyp/qt.gypi
@@ -176,14 +176,10 @@
   ],
   'conditions': [
     [ 'build_linux', {
-      'dependencies': [
-        '<(DEPTH)/linux_glibc_wraps.gyp:linux_glibc_wraps',
-      ],
       'library_dirs': [
         '<(qt_loc)/plugins/platforminputcontexts',
       ],
       'libraries': [
-        '<(PRODUCT_DIR)/obj.target/liblinux_glibc_wraps.a',
         '<@(qt_libs_release)',
         '-lcrypto',
         '-lX11',
diff --git Telegram/gyp/telegram_linux.gypi Telegram/gyp/telegram_linux.gypi
index 1e118538..c02e57a0 100644
--- Telegram/gyp/telegram_linux.gypi
+++ Telegram/gyp/telegram_linux.gypi
@@ -58,9 +58,6 @@
       '-Wno-maybe-uninitialized',
     ],
     'ldflags': [
-      '-Wl,-wrap,aligned_alloc',
-      '-Wl,-wrap,secure_getenv',
-      '-Wl,-wrap,clock_gettime',
       '-Wl,--no-as-needed,-lrt',
     ],
     'configurations': {
