diff --git Telegram/SourceFiles/mtproto/rsa_public_key.cpp Telegram/SourceFiles/mtproto/rsa_public_key.cpp
index fcb11979..add50d03 100644
--- Telegram/SourceFiles/mtproto/rsa_public_key.cpp
+++ Telegram/SourceFiles/mtproto/rsa_public_key.cpp
@@ -16,7 +16,7 @@ https://github.com/telegramdesktop/tdesktop/blob/master/LEGAL
 namespace MTP {
 namespace internal {
 namespace {
-#if OPENSSL_VERSION_NUMBER < 0x10100000L || defined(LIBRESSL_VERSION_NUMBER)
+#if OPENSSL_VERSION_NUMBER < 0x10100000L || (defined(LIBRESSL_VERSION_NUMBER) && LIBRESSL_VERSION_NUMBER < 0x2070000fL)
 
 // This is a key setter for compatibility with OpenSSL 1.0
 int RSA_set0_key(RSA *r, BIGNUM *n, BIGNUM *e, BIGNUM *d) {
