--- cmake/external/rlottie/CMakeLists.txt	2020-06-17 18:07:24.000000000 +0200
+++ -	2020-06-19 10:17:31.071247004 +0200
@@ -8,14 +8,9 @@
     add_library(external_rlottie INTERFACE IMPORTED GLOBAL)
     add_library(desktop-app::external_rlottie ALIAS external_rlottie)
 
-    find_package(rlottie)
-    if (rlottie_FOUND)
-        target_link_libraries(external_rlottie INTERFACE rlottie::rlottie)
-    else()
         find_package(PkgConfig REQUIRED)
         pkg_check_modules(RLOTTIE REQUIRED IMPORTED_TARGET rlottie)
         target_link_libraries(external_rlottie INTERFACE PkgConfig::RLOTTIE)
-    endif()
 else()
     add_library(external_rlottie OBJECT)
     add_library(desktop-app::external_rlottie ALIAS external_rlottie)
