--- a/cmake/external/kwayland/CMakeLists.txt	2022-04-19 14:57:00.000000000 +0200
+++ -	2022-04-26 18:18:21.919838146 +0200
@@ -32,6 +32,8 @@
         -DCMAKE_INSTALL_PREFIX=<BINARY_DIR>
         -DCMAKE_INSTALL_DATADIR=share
         -DBUILD_TESTING=OFF
+	-DQT_HOST_PATH=/usr
+	-DQtWaylandScanner_EXECUTABLE=/usr/lib/qt6/libexec/qtwaylandscanner
     BUILD_COMMAND ${CMAKE_COMMAND} --build <BINARY_DIR>
     COMMAND ${CMAKE_COMMAND} --install <BINARY_DIR>
     STEP_TARGETS build
