--- a/Telegram/lib_ui/ui/widgets/buttons.h	2025-03-10 07:58:29.000000000 +0100
+++ -	2025-03-13 21:21:45.309846801 +0100
@@ -13,6 +13,7 @@
 #include "styles/style_widgets.h"
 
 #include <memory>
+#include <cstddef>
 
 class Painter;
 
@@ -273,7 +274,7 @@
 		const style::SettingsButton &st = st::defaultSettingsButton);
 	SettingsButton(
 		QWidget *parent,
-		nullptr_t,
+		std::nullptr_t,
 		const style::SettingsButton &st = st::defaultSettingsButton);
 	~SettingsButton();
 
--- a/Telegram/lib_ui/ui/widgets/buttons.cpp	2025-03-10 07:58:29.000000000 +0100
+++ -	2025-03-13 21:27:09.133348285 +0100
@@ -795,7 +795,7 @@
 
 SettingsButton::SettingsButton(
 	QWidget *parent,
-	nullptr_t,
+	std::nullptr_t,
 	const style::SettingsButton &st)
 : RippleButton(parent, st.ripple)
 , _st(st)
