--- Telegram/SourceFiles/base/build_config.h
+++ Telegram/SourceFiles/base/build_config.h
@@ -47,9 +47,9 @@ https://github.com/telegramdesktop/tdesktop/blob/master/LEGAL
 #define ARCH_CPU_X86_FAMILY 1
 #define ARCH_CPU_X86 1
 #define ARCH_CPU_32_BITS 1
-#elif defined(__aarch64__)
+#elif defined(__aarch64__) || defined(__powerpc64__)
 #define ARCH_CPU_64_BITS 1
-#elif defined(_M_ARM) || defined(__arm__)
+#elif defined(_M_ARM) || defined(__arm__) || defined(__powerpc__)
 #define ARCH_CPU_32_BITS 1
 #else
 #error Please add support for your architecture in base/build_config.h
--- Telegram/SourceFiles/config.h
+++ Telegram/SourceFiles/config.h
@@ -194,7 +194,7 @@ constexpr auto ApiHash = "344583e45741c457fe1862106095a5eb";
 #endif // TDESKTOP_API_ID && TDESKTOP_API_HASH
 
 #if Q_BYTE_ORDER == Q_BIG_ENDIAN
-#error "Only little endian is supported!"
+#warning "Only little endian is supported!"
 #endif // Q_BYTE_ORDER == Q_BIG_ENDIAN
 
 #if (TDESKTOP_ALPHA_VERSION != 0)
