--- cmake/external/rlottie/CMakeLists.txt	2020-01-17 18:21:17.000000000 +0100
+++ -	2020-01-18 00:37:40.280438388 +0100
@@ -8,8 +8,9 @@
     add_library(external_rlottie INTERFACE IMPORTED GLOBAL)
     add_library(desktop-app::external_rlottie ALIAS external_rlottie)
 
-    find_package(rlottie REQUIRED)
-    target_link_libraries(external_rlottie INTERFACE rlottie::rlottie)
+    find_package(PkgConfig REQUIRED)
+    pkg_check_modules(RLOTTIE REQUIRED rlottie)
+    target_link_libraries(external_rlottie INTERFACE ${RLOTTIE_LIBRARIES})
 else()
     add_library(external_rlottie OBJECT)
     add_library(desktop-app::external_rlottie ALIAS external_rlottie)
