--- a/cmake/external/rlottie/CMakeLists.txt	2020-08-11 15:21:17.000000000 +0200
+++ b/cmake/external/rlottie/CMakeLists.txt	2020-08-11 15:21:17.000000000 +0200
@@ -8,7 +8,6 @@
 add_library(desktop-app::external_rlottie ALIAS external_rlottie)
 
 if (DESKTOP_APP_USE_PACKAGED)
-    find_package(rlottie QUIET)
 
     if (rlottie_FOUND)
         target_link_libraries(external_rlottie INTERFACE rlottie::rlottie)
