diff --git libdwarf/Makefile.in libdwarf/Makefile.in
index 28c62a9..da69093 100644
--- libdwarf/Makefile.in
+++ libdwarf/Makefile.in
@@ -129,7 +129,7 @@
 	$(AR) $(ARFLAGS) $@ $(OBJS)  dwarf_names.o > ar-output-temp
 
 libdwarf.so: dwarf_names.h dwarf_names.c $(OBJS) dwarf_names.o
-	$(CC) $(CFLAGS) -shared $(OBJS)  dwarf_names.o $(dwfzlib) -o $@
+	$(CC) $(CFLAGS) -shared $(OBJS)  dwarf_names.o $(dwfzlib) -Wl,-soname,$@ -o $@
 
 none:
 	echo "do nothing " $@
