diff --git Makefile Makefile
index f8aa905..d6ded67 100644
--- Makefile
+++ Makefile
@@ -676,6 +676,8 @@ $(DESTDIR)$(libdir)/${LIB_REL_NAME} :: \
 	$(LIBTOOL_CREATE) ${LIB_OBJECT} ${LIBS} ${LDFLAGS}
 	cd $(DESTDIR)$(libdir) && ($(LN_S) ${LIB_REL_NAME} ${LIB_ABI_NAME}; $(LN_S) ${LIB_ABI_NAME} ${LIB_BASENAME}; )
 	- $(SHELL) -c "if test -z "$(DESTDIR)" ; then /sbin/ldconfig; fi"
+	ar rcs libcdk.a *.o
+	install -m644 libcdk.a $(DESTDIR)$(libdir)
 
 uninstall \
 uninstall.libs ::
