--- a/modules/QtPDF/CMake/Modules/FindPoppler.cmake
+++ b/modules/QtPDF/CMake/Modules/FindPoppler.cmake
@@ -79,15 +79,7 @@ if (Poppler_PRIVATE_INCLUDE_DIR)
 endif ()
 
 
-# Scan poppler libraries for dependencies on Fontconfig
-include(GetPrerequisites)
-mark_as_advanced(gp_cmd)
-get_prerequisites("${Poppler_LIBRARY}" Poppler_PREREQS TRUE FALSE "" "")
-if ("${Poppler_PREREQS}" MATCHES "fontconfig")
-  set(Poppler_NEEDS_FONTCONFIG TRUE)
-else ()
-  set(Poppler_NEEDS_FONTCONFIG FALSE)
-endif ()
+set(Poppler_NEEDS_FONTCONFIG TRUE)
 
 # -----------------------
 # Find Poppler components
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -212,7 +212,7 @@ set (CMAKE_CXX_STANDARD 11)
 
 if (QT_DEFAULT_MAJOR_VERSION EQUAL 6)
 	# Check for Qt6
-	find_package(Qt6 REQUIRED COMPONENTS Core Core5Compat Widgets Gui UiTools Concurrent Xml LinguistTools Qml)
+	find_package(Qt6 REQUIRED COMPONENTS Core Core5Compat Widgets Gui UiTools Concurrent Xml Linguist Qml)
 	set(QT_LIBRARIES Qt6::Core Qt6::Core5Compat Qt6::Widgets Qt6::Gui Qt6::UiTools Qt6::Concurrent Qt6::Xml Qt6::Qml)
 
 	if (UNIX AND NOT APPLE)
