--- a/modules/QtPDF/CMake/Modules/FindPoppler.cmake
+++ b/modules/QtPDF/CMake/Modules/FindPoppler.cmake
@@ -79,15 +79,7 @@
 endif ()
 
 
-# Scan poppler libraries for dependencies on Fontconfig
-include(GetPrerequisites)
-mark_as_advanced(gp_cmd)
-get_prerequisites("${Poppler_LIBRARY}" Poppler_PREREQS TRUE FALSE "" "")
-if ("${Poppler_PREREQS}" MATCHES "fontconfig")
-  set(Poppler_NEEDS_FONTCONFIG TRUE)
-else ()
-  set(Poppler_NEEDS_FONTCONFIG FALSE)
-endif ()
+set(Poppler_NEEDS_FONTCONFIG TRUE)
 
 # -----------------------
 # Find Poppler components
