From a5352a3a94e3685125650b65e6197de060326cc2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Stefan=20L=C3=B6ffler?= <st.loeffler@gmail.com>
Date: Fri, 1 May 2020 08:05:46 +0200
Subject: [PATCH] <QtPDF> Fix compilation with Qt 5.15

---
 modules/QtPDF/src/PDFBackend.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/modules/QtPDF/src/PDFBackend.cpp b/modules/QtPDF/src/PDFBackend.cpp
index 0305d87a..500d20aa 100644
--- a/modules/QtPDF/src/PDFBackend.cpp
+++ b/modules/QtPDF/src/PDFBackend.cpp
@@ -14,6 +14,7 @@
 
 #include <PDFBackend.h>
 #include <QPainter>
+#include <QPainterPath>
 #include <QApplication>
 
 namespace QtPDF {
