*** a/ax25ipd/io.c	2019-03-20 17:48:59.000000000 +0100
--- b/ax25ipd/io.c	2021-09-02 00:32:05.815175217 +0200
***************
*** 21,23 ****
  #include <syslog.h>
! #include <termio.h>
  #include <time.h>
--- 21,24 ----
  #include <syslog.h>
! #include <termios.h>
! #include <sys/ioctl.h>
  #include <time.h>
***************
*** 34,36 ****
  
! static struct termio nterm;
  
--- 35,37 ----
  
! static struct termios nterm;
  
*** a/call/call.h	2019-03-20 17:48:59.000000000 +0100
--- b/call/call.h	2021-09-02 00:36:47.367157222 +0200
***************
*** 6,7 ****
--- 6,13 ----
  #endif
+ 
+ #ifndef __CALL_H
+ #define __CALL_H
+ 
+ #include <time.h>
+ 
  extern int fd;
*************** extern int yapp2unix(char *);
*** 21 ****
--- 27,29 ----
  extern void unix2yapp(time_t unix_date, char *buffer);
+ 
+ #endif
