--- a/src/ui/lv2_ui.cxx	2016-12-24 20:36:45.000000000 +0100
+++ b/src/ui/lv2_ui.cxx	2020-12-25 22:17:21.206336431 +0100
@@ -33,7 +33,7 @@
 #include "whaaa.hxx"
 
 
-static LV2UI_Handle artyfx_instantiate(const struct _LV2UI_Descriptor * descriptor,
+static LV2UI_Handle artyfx_instantiate(const struct LV2UI_Descriptor * descriptor,
                                        const char * plugin_uri,
                                        const char * bundle_path,
                                        LV2UI_Write_Function write_function,
