--- a/CMakeScripts/DefineDependsandFlags.cmake
+++ b/CMakeScripts/DefineDependsandFlags.cmake
@@ -309,12 +309,12 @@ elseif(APPLE)
     list(APPEND INKSCAPE_CXX_FLAGS "-D_GNU_SOURCE")
 else()
     # The package stacktrace_backtrace may not be available on all distros.
-    find_package(Boost 1.19.0 COMPONENTS stacktrace_backtrace)
-    if (BOOST_FOUND)
+    find_package(Boost 1.19.0)
+    option(BOOST_STACKTRACE_USE_BACKTRACE "BOOST_STACKTRACE_USE_BACKTRACE" OFF)
+    if (BOOST_STACKTRACE_USE_BACKTRACE)
         list(APPEND INKSCAPE_LIBS "-lbacktrace")
         add_definitions("-DBOOST_STACKTRACE_USE_BACKTRACE")
     else() # fall back to stacktrace_basic
-        find_package(Boost 1.19.0 REQUIRED COMPONENTS stacktrace_basic)
         list(APPEND INKSCAPE_CXX_FLAGS "-D_GNU_SOURCE")
     endif()
 endif()
