--- a/src/display/control/ctrl-handle-rendering.h
+++ b/src/display/control/ctrl-handle-rendering.h
@@ -13,6 +13,7 @@
  * Released under GNU GPL v2+, read the file 'COPYING' for more information.
  */
 
+#include <cstdint>
 #include <memory>
 #include <compare>
 
--- a/src/util/pool.cpp
+++ b/src/util/pool.cpp
@@ -3,6 +3,7 @@
 #include "pool.h"
 
 #include <cassert>
+#include <cstdint>
 #include <utility>
 
 namespace Inkscape::Util {
