diff -aur old/src/2geom/utils.h new/src/2geom/utils.h
--- old/src/2geom/utils.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/2geom/utils.h	2011-05-26 14:38:55.912288964 +0000
@@ -35,6 +35,7 @@
 
 #include <cmath>
 #include <vector>
+#include <cstddef>
 
 namespace Geom {
 
diff -aur old/src/arc-context.h new/src/arc-context.h
--- old/src/arc-context.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/arc-context.h	2011-05-26 14:37:30.814792031 +0000
@@ -15,6 +15,7 @@
  * Released under GNU GPL, read the file 'COPYING' for more information
  */
 
+#include <stddef.h>
 #include <sigc++/connection.h>
 
 #include <2geom/point.h>
diff -aur old/src/box3d-context.h new/src/box3d-context.h
--- old/src/box3d-context.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/box3d-context.h	2011-05-26 14:37:30.814792031 +0000
@@ -15,6 +15,7 @@
  * Released under GNU GPL
  */
 
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 #include "event-context.h"
 #include "proj_pt.h"
diff -aur old/src/conn-avoid-ref.h new/src/conn-avoid-ref.h
--- old/src/conn-avoid-ref.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/conn-avoid-ref.h	2011-05-26 14:37:30.814792031 +0000
@@ -14,6 +14,7 @@
  */
 
 #include <glib/gslist.h>
+#include <stddef.h>
 #include <sigc++/connection.h>
 
 struct SPDesktop;
diff -aur old/src/connection-pool.h new/src/connection-pool.h
--- old/src/connection-pool.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/connection-pool.h	2011-05-26 14:37:30.814792031 +0000
@@ -3,6 +3,7 @@
 
 #include <glib-object.h>
 #include <gtkmm.h>
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 
 namespace Inkscape
diff -aur old/src/connector-context.h new/src/connector-context.h
--- old/src/connector-context.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/connector-context.h	2011-05-26 14:37:30.814792120 +0000
@@ -12,6 +12,7 @@
  * Released under GNU GPL, read the file 'COPYING' for more information
  */
 
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 #include <sigc++/connection.h>
 #include "event-context.h"
diff -aur old/src/desktop.h new/src/desktop.h
--- old/src/desktop.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/desktop.h	2011-05-26 14:37:30.814792120 +0000
@@ -27,6 +27,7 @@
 
 #include <gdk/gdkevents.h>
 #include <gtk/gtktypeutils.h>
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 
 #include <2geom/matrix.h>
diff -aur old/src/dialogs/object-attributes.cpp new/src/dialogs/object-attributes.cpp
--- old/src/dialogs/object-attributes.cpp	2010-07-13 03:48:40.691600000 +0000
+++ new/src/dialogs/object-attributes.cpp	2011-05-26 14:37:30.814792120 +0000
@@ -13,6 +13,7 @@
 #include <glibmm/i18n.h>
 #include <string>
 #include <cstring>
+#include <stddef.h>
 #include <sigc++/connection.h>
 #include <sigc++/functors/ptr_fun.h>
 #include <sigc++/adaptors/bind.h>
diff -aur old/src/display/canvas-temporary-item.h new/src/display/canvas-temporary-item.h
--- old/src/display/canvas-temporary-item.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/display/canvas-temporary-item.h	2011-05-26 14:37:30.814792120 +0000
@@ -14,6 +14,7 @@
 
 #include "display/display-forward.h"
 
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 
 namespace Inkscape {
diff -aur old/src/document.h new/src/document.h
--- old/src/document.h	2010-08-04 16:45:00.868588000 +0000
+++ new/src/document.h	2011-05-26 14:37:30.814792120 +0000
@@ -17,6 +17,7 @@
 
 #include <glib-object.h>
 #include <gtk/gtksignal.h>
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 #include <sigc++/class_slot.h>
 
diff -aur old/src/document-private.h new/src/document-private.h
--- old/src/document-private.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/document-private.h	2011-05-26 14:37:30.814792120 +0000
@@ -14,6 +14,7 @@
  */
 
 #include <map>
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 #include "xml/event-fns.h"
 #include "sp-defs.h"
diff -aur old/src/document-subset.h new/src/document-subset.h
--- old/src/document-subset.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/document-subset.h	2011-05-26 14:37:30.814792120 +0000
@@ -10,6 +10,7 @@
 #ifndef SEEN_INKSCAPE_DOCUMENT_SUBSET_H
 #define SEEN_INKSCAPE_DOCUMENT_SUBSET_H
 
+#include <stddef.h>
 #include <sigc++/connection.h>
 #include <sigc++/functors/slot.h>
 
diff -aur old/src/draw-context.h new/src/draw-context.h
--- old/src/draw-context.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/draw-context.h	2011-05-26 14:37:30.814792120 +0000
@@ -14,6 +14,7 @@
  * Released under GNU GPL
  */
 
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 #include "event-context.h"
 #include <forward.h>
diff -aur old/src/extension/timer.h new/src/extension/timer.h
--- old/src/extension/timer.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/extension/timer.h	2011-05-26 14:37:30.814792120 +0000
@@ -13,6 +13,7 @@
 #ifndef INKSCAPE_EXTENSION_TIMER_H__
 #define INKSCAPE_EXTENSION_TIMER_H__
 
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 #include <glibmm/timeval.h>
 #include "extension-forward.h"
diff -aur old/src/flood-context.h new/src/flood-context.h
--- old/src/flood-context.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/flood-context.h	2011-05-26 14:37:30.814792120 +0000
@@ -11,6 +11,7 @@
  * Released under GNU GPL
  */
 
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 #include <gtk/gtk.h>
 #include "event-context.h"
diff -aur old/src/gc-alloc.h new/src/gc-alloc.h
--- old/src/gc-alloc.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/gc-alloc.h	2011-05-26 14:37:30.814792120 +0000
@@ -16,6 +16,7 @@
 #define SEEN_INKSCAPE_GC_ALLOC_H
 
 #include <limits>
+#include <cstddef>
 #include "gc-core.h"
 
 namespace Inkscape {
diff -aur old/src/gc.cpp new/src/gc.cpp
--- old/src/gc.cpp	2010-07-13 03:48:40.691600000 +0000
+++ new/src/gc.cpp	2011-05-26 14:37:30.814792120 +0000
@@ -16,6 +16,7 @@
 #include <glib/gmessages.h>
 #include <sigc++/functors/ptr_fun.h>
 #include <glibmm/main.h>
+#include <cstddef>
 
 namespace Inkscape {
 namespace GC {
diff -aur old/src/gc-finalized.h new/src/gc-finalized.h
--- old/src/gc-finalized.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/gc-finalized.h	2011-05-26 14:37:30.814792120 +0000
@@ -17,6 +17,7 @@
 #define SEEN_INKSCAPE_GC_FINALIZED_H
 
 #include <new>
+#include <cstddef>
 #include "gc-core.h"
 
 namespace Inkscape {
diff -aur old/src/gradient-context.h new/src/gradient-context.h
--- old/src/gradient-context.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/gradient-context.h	2011-05-26 14:37:30.814792120 +0000
@@ -15,6 +15,7 @@
  * Released under GNU GPL
  */
 
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 #include "event-context.h"
 
diff -aur old/src/gradient-drag.h new/src/gradient-drag.h
--- old/src/gradient-drag.h	2010-08-04 16:45:00.868588000 +0000
+++ new/src/gradient-drag.h	2011-05-26 14:37:30.814792120 +0000
@@ -15,6 +15,7 @@
  */
 
 #include <glib/gslist.h>
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 #include <vector>
 
diff -aur old/src/knot.h new/src/knot.h
--- old/src/knot.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/knot.h	2011-05-26 14:37:30.814792120 +0000
@@ -20,6 +20,7 @@
 #include "forward.h"
 #include <2geom/point.h>
 #include "knot-enums.h"
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 
 class SPKnot;
diff -aur old/src/libavoid/geomtypes.h new/src/libavoid/geomtypes.h
--- old/src/libavoid/geomtypes.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/libavoid/geomtypes.h	2011-05-26 14:37:30.814792120 +0000
@@ -29,6 +29,7 @@
 #ifndef AVOID_GEOMTYPES_H
 #define AVOID_GEOMTYPES_H
 
+#include <cstddef>
 #include <vector>
 #include <utility>
 
diff -aur old/src/live_effects/lpeobject-reference.h new/src/live_effects/lpeobject-reference.h
--- old/src/live_effects/lpeobject-reference.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/live_effects/lpeobject-reference.h	2011-05-26 14:37:30.818124452 +0000
@@ -11,6 +11,7 @@
 
 #include <forward.h>
 #include <uri-references.h>
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 
 namespace Inkscape {
diff -aur old/src/live_effects/parameter/path.h new/src/live_effects/parameter/path.h
--- old/src/live_effects/parameter/path.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/live_effects/parameter/path.h	2011-05-26 14:37:30.818124452 +0000
@@ -16,6 +16,7 @@
 
 #include "live_effects/parameter/parameter.h"
 #include "live_effects/parameter/path-reference.h"
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 
 namespace Inkscape {
diff -aur old/src/live_effects/parameter/path-reference.h new/src/live_effects/parameter/path-reference.h
--- old/src/live_effects/parameter/path-reference.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/live_effects/parameter/path-reference.h	2011-05-26 14:37:30.818124452 +0000
@@ -11,6 +11,7 @@
 
 #include <forward.h>
 #include <uri-references.h>
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 
 class Path;
diff -aur old/src/Makefile.am new/src/Makefile.am
--- old/src/Makefile.am	2010-07-13 03:48:40.691600000 +0000
+++ new/src/Makefile.am	2011-05-26 14:37:30.814792031 +0000
@@ -205,7 +205,7 @@
 
 inkscape_SOURCES += main.cpp $(win32_sources)
 inkscape_LDADD = $(all_libs)
-inkscape_LDFLAGS = --export-dynamic $(kdeldflags) $(mwindows)
+inkscape_LDFLAGS = $(kdeldflags) $(mwindows)
 
 inkview_SOURCES += inkview.cpp $(win32_sources)
 inkview_LDADD = $(all_libs)
diff -aur old/src/message-stack.h new/src/message-stack.h
--- old/src/message-stack.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/message-stack.h	2011-05-26 14:37:30.818124452 +0000
@@ -14,6 +14,7 @@
 #ifndef SEEN_INKSCAPE_MESSAGE_STACK_H
 #define SEEN_INKSCAPE_MESSAGE_STACK_H
 
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 #include <glib.h>
 #include <stdarg.h>
diff -aur old/src/object-hierarchy.h new/src/object-hierarchy.h
--- old/src/object-hierarchy.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/object-hierarchy.h	2011-05-26 14:37:30.818124452 +0000
@@ -14,6 +14,7 @@
 
 #include <exception>
 #include <list>
+#include <stddef.h>
 #include <sigc++/connection.h>
 #include <sigc++/signal.h>
 #include <glib/gmessages.h>
diff -aur old/src/persp3d-reference.h new/src/persp3d-reference.h
--- old/src/persp3d-reference.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/persp3d-reference.h	2011-05-26 14:37:30.818124452 +0000
@@ -11,6 +11,7 @@
  */
 
 #include "uri-references.h"
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 #include "persp3d.h"
 
diff -aur old/src/rect-context.h new/src/rect-context.h
--- old/src/rect-context.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/rect-context.h	2011-05-26 14:37:30.818124452 +0000
@@ -14,6 +14,7 @@
  * Released under GNU GPL
  */
 
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 #include "event-context.h"
 #include "libnr/nr-point.h"
diff -aur old/src/selcue.h new/src/selcue.h
--- old/src/selcue.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/selcue.h	2011-05-26 14:37:30.818124452 +0000
@@ -14,6 +14,7 @@
  */
 
 #include <list>
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 
 class SPDesktop;
diff -aur old/src/selection-describer.h new/src/selection-describer.h
--- old/src/selection-describer.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/selection-describer.h	2011-05-26 14:37:30.818124452 +0000
@@ -12,6 +12,7 @@
 #ifndef SEEN_INKSCAPE_SELECTION_DESCRIPTION_HANDLER_H
 #define SEEN_INKSCAPE_SELECTION_DESCRIPTION_HANDLER_H
 
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 #include "message-context.h"
 
diff -aur old/src/selection.h new/src/selection.h
--- old/src/selection.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/selection.h	2011-05-26 14:37:30.818124452 +0000
@@ -19,6 +19,7 @@
 #include <vector>
 #include <map>
 #include <list>
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 
 //#include "libnr/nr-rect.h"
diff -aur old/src/seltrans.h new/src/seltrans.h
--- old/src/seltrans.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/seltrans.h	2011-05-26 14:37:30.818124452 +0000
@@ -15,6 +15,7 @@
  * Released under GNU GPL, read the file 'COPYING' for more information
  */
 
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 #include <2geom/point.h>
 #include <2geom/matrix.h>
diff -aur old/src/sp-conn-end.h new/src/sp-conn-end.h
--- old/src/sp-conn-end.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/sp-conn-end.h	2011-05-26 14:37:30.818124452 +0000
@@ -2,6 +2,7 @@
 #define SEEN_SP_CONN_END
 
 #include <glib/gtypes.h>
+#include <stddef.h>
 #include <sigc++/connection.h>
 
 #include "sp-use-reference.h"
diff -aur old/src/sp-conn-end-pair.h new/src/sp-conn-end-pair.h
--- old/src/sp-conn-end-pair.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/sp-conn-end-pair.h	2011-05-26 14:37:30.818124452 +0000
@@ -15,6 +15,7 @@
 
 #include "forward.h"
 #include "libnr/nr-point.h"
+#include <stddef.h>
 #include <sigc++/connection.h>
 #include <sigc++/functors/slot.h>
 #include <sigc++/signal.h>
diff -aur old/src/sp-gradient.h new/src/sp-gradient.h
--- old/src/sp-gradient.h	2010-08-04 16:45:00.868588000 +0000
+++ new/src/sp-gradient.h	2011-05-26 14:37:30.818124452 +0000
@@ -25,6 +25,7 @@
 #include "sp-gradient-units.h"
 #include "sp-gradient-vector.h"
 
+#include <stddef.h>
 #include <sigc++/connection.h>
 
 struct SPGradientReference;
diff -aur old/src/spiral-context.h new/src/spiral-context.h
--- old/src/spiral-context.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/spiral-context.h	2011-05-26 14:37:30.821456796 +0000
@@ -16,6 +16,7 @@
  */
 
 #include <gtk/gtktypeutils.h>
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 #include "event-context.h"
 #include "libnr/nr-point.h"
diff -aur old/src/sp-object.h new/src/sp-object.h
--- old/src/sp-object.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/sp-object.h	2011-05-26 14:37:30.818124452 +0000
@@ -56,6 +56,7 @@
 
 
 #include <glib-object.h>
+#include <stddef.h>
 #include <sigc++/connection.h>
 #include <sigc++/functors/slot.h>
 #include <sigc++/signal.h>
diff -aur old/src/sp-offset.h new/src/sp-offset.h
--- old/src/sp-offset.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/sp-offset.h	2011-05-26 14:37:30.818124452 +0000
@@ -16,6 +16,7 @@
 
 #include "sp-shape.h"
 
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 
 #define SP_TYPE_OFFSET            (sp_offset_get_type ())
diff -aur old/src/sp-pattern.h new/src/sp-pattern.h
--- old/src/sp-pattern.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/sp-pattern.h	2011-05-26 14:37:30.818124452 +0000
@@ -32,6 +32,7 @@
 #include "sp-paint-server.h"
 #include "uri-references.h"
 
+#include <stddef.h>
 #include <sigc++/connection.h>
 
 class SPPatternReference : public Inkscape::URIReference {
diff -aur old/src/sp-shape.h new/src/sp-shape.h
--- old/src/sp-shape.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/sp-shape.h	2011-05-26 14:37:30.818124452 +0000
@@ -19,6 +19,7 @@
 #include "sp-marker-loc.h"
 #include <2geom/forward.h>
 
+#include <stddef.h>
 #include <sigc++/connection.h>
 
 #define SP_TYPE_SHAPE (sp_shape_get_type ())
diff -aur old/src/sp-switch.h new/src/sp-switch.h
--- old/src/sp-switch.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/sp-switch.h	2011-05-26 14:37:30.821456796 +0000
@@ -14,6 +14,7 @@
 
 #include "sp-item-group.h"
 
+#include <stddef.h>
 #include <sigc++/connection.h>
 
 #define SP_TYPE_SWITCH            (CSwitch::getType())
diff -aur old/src/sp-text.h new/src/sp-text.h
--- old/src/sp-text.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/sp-text.h	2011-05-26 14:37:30.821456796 +0000
@@ -14,6 +14,7 @@
  */
 
 #include <glib/gtypes.h>
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 #include "sp-item.h"
 #include "sp-string.h"
diff -aur old/src/sp-tref-reference.h new/src/sp-tref-reference.h
--- old/src/sp-tref-reference.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/sp-tref-reference.h	2011-05-26 14:37:30.821456796 +0000
@@ -13,6 +13,7 @@
 
 #include <forward.h>
 #include <uri-references.h>
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 
 #include "util/share.h"
diff -aur old/src/sp-use.h new/src/sp-use.h
--- old/src/sp-use.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/sp-use.h	2011-05-26 14:37:30.821456796 +0000
@@ -13,6 +13,7 @@
  * Released under GNU GPL, read the file 'COPYING' for more information
  */
 
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 #include "svg/svg-length.h"
 #include "sp-item.h"
diff -aur old/src/sp-use-reference.h new/src/sp-use-reference.h
--- old/src/sp-use-reference.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/sp-use-reference.h	2011-05-26 14:37:30.821456796 +0000
@@ -11,6 +11,7 @@
 
 #include <forward.h>
 #include <uri-references.h>
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 
 class Path;
diff -aur old/src/star-context.h new/src/star-context.h
--- old/src/star-context.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/star-context.h	2011-05-26 14:37:30.821456796 +0000
@@ -14,6 +14,7 @@
  * Released under GNU GPL, read the file 'COPYING' for more information
  */
 
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 #include "event-context.h"
 #include "libnr/nr-point.h"
diff -aur old/src/style.h new/src/style.h
--- old/src/style.h	2010-08-04 16:45:00.868588000 +0000
+++ new/src/style.h	2011-05-26 14:37:30.821456796 +0000
@@ -24,6 +24,7 @@
 #include "uri.h"
 #include "sp-paint-server.h"
 
+#include <stddef.h>
 #include <sigc++/connection.h>
 
 namespace Inkscape {
diff -aur old/src/text-context.h new/src/text-context.h
--- old/src/text-context.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/text-context.h	2011-05-26 14:37:30.821456796 +0000
@@ -15,6 +15,7 @@
  */
 
 /*  #include <gdk/gdkic.h> */
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 #include <gtk/gtkimcontext.h>
 
diff -aur old/src/ui/dialog/desktop-tracker.h new/src/ui/dialog/desktop-tracker.h
--- old/src/ui/dialog/desktop-tracker.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/ui/dialog/desktop-tracker.h	2011-05-26 14:37:30.821456796 +0000
@@ -11,6 +11,7 @@
 #ifndef SEEN_DIALOG_DESKTOP_TRACKER
 #define SEEN_DIALOG_DESKTOP_TRACKER
 
+#include <stddef.h>
 #include <sigc++/connection.h>
 #include <glib/gtypes.h>
 
diff -aur old/src/ui/dialog/document-metadata.h new/src/ui/dialog/document-metadata.h
--- old/src/ui/dialog/document-metadata.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/ui/dialog/document-metadata.h	2011-05-26 14:37:30.821456796 +0000
@@ -14,6 +14,7 @@
 #define INKSCAPE_UI_DIALOG_DOCUMENT_METADATA_H
 
 #include <list>
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 #include <gtkmm/notebook.h>
 #include <glibmm/i18n.h>
diff -aur old/src/ui/dialog/document-properties.h new/src/ui/dialog/document-properties.h
--- old/src/ui/dialog/document-properties.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/ui/dialog/document-properties.h	2011-05-26 14:37:30.821456796 +0000
@@ -15,6 +15,7 @@
 #define INKSCAPE_UI_DIALOG_DOCUMENT_PREFERENCES_H
 
 #include <list>
+#include <stddef.h>
 #include <sigc++/sigc++.h>//
 #include <gtkmm/notebook.h>
 #include <glibmm/i18n.h>
diff -aur old/src/ui/dialog/inkscape-preferences.h new/src/ui/dialog/inkscape-preferences.h
--- old/src/ui/dialog/inkscape-preferences.h	2010-11-06 18:47:19.806162000 +0000
+++ new/src/ui/dialog/inkscape-preferences.h	2011-05-26 14:37:30.821456796 +0000
@@ -28,6 +28,7 @@
 #include <gtkmm/frame.h>
 #include <gtkmm/notebook.h>
 #include "ui/widget/preferences-widget.h"
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 #include <glibmm/i18n.h>
 #include <gtkmm/textview.h>
diff -aur old/src/ui/dialog/undo-history.cpp new/src/ui/dialog/undo-history.cpp
--- old/src/ui/dialog/undo-history.cpp	2010-07-13 03:48:40.691600000 +0000
+++ new/src/ui/dialog/undo-history.cpp	2011-05-26 14:37:30.821456796 +0000
@@ -14,6 +14,7 @@
 
 #include <glibmm/i18n.h>
 #include <gtk/gtkimage.h>
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 
 
diff -aur old/src/ui/tool/control-point.h new/src/ui/tool/control-point.h
--- old/src/ui/tool/control-point.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/ui/tool/control-point.h	2011-05-26 14:37:30.821456796 +0000
@@ -12,6 +12,7 @@
 #define SEEN_UI_TOOL_CONTROL_POINT_H
 
 #include <boost/utility.hpp>
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 #include <gdkmm.h>
 #include <gtkmm.h>
diff -aur old/src/ui/tool/control-point-selection.h new/src/ui/tool/control-point-selection.h
--- old/src/ui/tool/control-point-selection.h	2010-08-04 16:45:00.868588000 +0000
+++ new/src/ui/tool/control-point-selection.h	2011-05-26 14:37:30.821456796 +0000
@@ -14,6 +14,7 @@
 
 #include <memory>
 #include <boost/optional.hpp>
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 #include <2geom/forward.h>
 #include <2geom/point.h>
diff -aur old/src/ui/tool/manipulator.h new/src/ui/tool/manipulator.h
--- old/src/ui/tool/manipulator.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/ui/tool/manipulator.h	2011-05-26 14:37:30.821456796 +0000
@@ -13,6 +13,7 @@
 
 #include <set>
 #include <map>
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 #include <glib.h>
 #include <gdk/gdk.h>
diff -aur old/src/ui/tool/multi-path-manipulator.h new/src/ui/tool/multi-path-manipulator.h
--- old/src/ui/tool/multi-path-manipulator.h	2010-11-06 18:47:19.806162000 +0000
+++ new/src/ui/tool/multi-path-manipulator.h	2011-05-26 14:37:30.821456796 +0000
@@ -11,6 +11,7 @@
 #ifndef SEEN_UI_TOOL_MULTI_PATH_MANIPULATOR_H
 #define SEEN_UI_TOOL_MULTI_PATH_MANIPULATOR_H
 
+#include <stddef.h>
 #include <sigc++/connection.h>
 #include "display/display-forward.h"
 #include "forward.h"
diff -aur old/src/ui/tool/node.h new/src/ui/tool/node.h
--- old/src/ui/tool/node.h	2011-01-02 04:34:09.814902000 +0000
+++ new/src/ui/tool/node.h	2011-05-26 14:37:30.821456796 +0000
@@ -15,6 +15,7 @@
 #include <iterator>
 #include <iosfwd>
 #include <stdexcept>
+#include <cstddef>
 #include <tr1/functional>
 #include <boost/utility.hpp>
 #include <boost/shared_ptr.hpp>
diff -aur old/src/ui/tool/node-tool.h new/src/ui/tool/node-tool.h
--- old/src/ui/tool/node-tool.h	2010-11-06 18:47:19.806162000 +0000
+++ new/src/ui/tool/node-tool.h	2011-05-26 14:37:30.821456796 +0000
@@ -14,6 +14,7 @@
 #include <memory>
 #include <boost/ptr_container/ptr_map.hpp>
 #include <glib.h>
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 #include "event-context.h"
 #include "forward.h"
diff -aur old/src/ui/view/view.h new/src/ui/view/view.h
--- old/src/ui/view/view.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/ui/view/view.h	2011-05-26 14:37:30.821456796 +0000
@@ -15,6 +15,7 @@
  */
 
 #include <gdk/gdktypes.h>
+#include <stddef.h>
 #include <sigc++/connection.h>
 #include "message.h"
 #include "gc-managed.h"
diff -aur old/src/ui/widget/color-picker.h new/src/ui/widget/color-picker.h
--- old/src/ui/widget/color-picker.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/ui/widget/color-picker.h	2011-05-26 14:37:30.824789230 +0000
@@ -13,6 +13,7 @@
 #ifndef __COLOR_PICKER_H__
 #define __COLOR_PICKER_H__
 
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 #include <gtkmm/button.h>
 #include <gtkmm/dialog.h>
diff -aur old/src/ui/widget/page-sizer.h new/src/ui/widget/page-sizer.h
--- old/src/ui/widget/page-sizer.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/ui/widget/page-sizer.h	2011-05-26 14:37:30.824789230 +0000
@@ -13,6 +13,7 @@
 #define INKSCAPE_UI_WIDGET_PAGE_SIZER__H
 
 #include <gtkmm.h>
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 
 #include "helper/units.h"
diff -aur old/src/ui/widget/preferences-widget.h new/src/ui/widget/preferences-widget.h
--- old/src/ui/widget/preferences-widget.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/ui/widget/preferences-widget.h	2011-05-26 14:37:30.824789230 +0000
@@ -26,6 +26,7 @@
 #include <gtkmm/drawingarea.h>
 #include <gtkmm/frame.h>
 #include <gtkmm/filechooserbutton.h>
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 #include <glibmm/i18n.h>
 
diff -aur old/src/ui/widget/selected-style.h new/src/ui/widget/selected-style.h
--- old/src/ui/widget/selected-style.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/ui/widget/selected-style.h	2011-05-26 14:37:30.824789230 +0000
@@ -23,6 +23,7 @@
 #include <gtkmm/adjustment.h>
 #include <gtkmm/spinbutton.h>
 
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 
 #include <glibmm/i18n.h>
diff -aur old/src/ui/widget/style-subject.h new/src/ui/widget/style-subject.h
--- old/src/ui/widget/style-subject.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/ui/widget/style-subject.h	2011-05-26 14:37:30.824789230 +0000
@@ -14,6 +14,7 @@
 #include "libnr/nr-rect.h"
 #include <2geom/rect.h>
 #include "sp-item.h"
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 
 class SPDesktop;
diff -aur old/src/uri-references.h new/src/uri-references.h
--- old/src/uri-references.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/uri-references.h	2011-05-26 14:37:30.824789230 +0000
@@ -13,6 +13,7 @@
  * Released under GNU GPL, read the file 'COPYING' for more information
  */
 
+#include <stddef.h>
 #include <sigc++/connection.h>
 #include <sigc++/trackable.h>
 
diff -aur old/src/util/forward-pointer-iterator.h new/src/util/forward-pointer-iterator.h
--- old/src/util/forward-pointer-iterator.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/util/forward-pointer-iterator.h	2011-05-26 14:37:30.824789230 +0000
@@ -15,6 +15,7 @@
 #define SEEN_INKSCAPE_UTIL_FORWARD_POINTER_ITERATOR_H
 
 #include <iterator>
+#include <cstddef>
 #include "util/reference.h"
 
 namespace Inkscape {
diff -aur old/src/util/share.h new/src/util/share.h
--- old/src/util/share.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/util/share.h	2011-05-26 14:37:30.824789230 +0000
@@ -14,6 +14,7 @@
 
 #include "gc-core.h"
 #include <cstring>
+#include <cstddef>
 
 namespace Inkscape {
 namespace Util {
diff -aur old/src/util/unordered-containers.h new/src/util/unordered-containers.h
--- old/src/util/unordered-containers.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/util/unordered-containers.h	2011-05-26 14:37:30.824789230 +0000
@@ -41,6 +41,8 @@
 # define INK_UNORDERED_MAP __gnu_cxx::hash_map
 # define INK_HASH __gnu_cxx::hash
 
+#include <cstddef>
+
 namespace __gnu_cxx {
 // hash function for pointers
 // TR1 and Boost have this defined by default, __gnu_cxx doesn't
diff -aur old/src/widgets/desktop-widget.h new/src/widgets/desktop-widget.h
--- old/src/widgets/desktop-widget.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/widgets/desktop-widget.h	2011-05-26 14:37:30.824789230 +0000
@@ -21,6 +21,7 @@
 #include "ui/view/view-widget.h"
 #include "ui/view/edit-widget-interface.h"
 
+#include <stddef.h>
 #include <sigc++/connection.h>
 
 // forward declaration
diff -aur old/src/widgets/gradient-image.h new/src/widgets/gradient-image.h
--- old/src/widgets/gradient-image.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/widgets/gradient-image.h	2011-05-26 14:37:30.824789230 +0000
@@ -19,6 +19,7 @@
 
 #include <glib.h>
 
+#include <stddef.h>
 #include <sigc++/connection.h>
 
 #define SP_TYPE_GRADIENT_IMAGE (sp_gradient_image_get_type ())
diff -aur old/src/widgets/gradient-vector.h new/src/widgets/gradient-vector.h
--- old/src/widgets/gradient-vector.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/widgets/gradient-vector.h	2011-05-26 14:37:30.824789230 +0000
@@ -17,6 +17,7 @@
 
 #include <glib.h>
 
+#include <stddef.h>
 #include <sigc++/connection.h>
 
 #include <gtk/gtkvbox.h>
diff -aur old/src/widgets/sp-attribute-widget.h new/src/widgets/sp-attribute-widget.h
--- old/src/widgets/sp-attribute-widget.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/widgets/sp-attribute-widget.h	2011-05-26 14:37:30.824789230 +0000
@@ -14,6 +14,7 @@
 #define SEEN_DIALOGS_SP_ATTRIBUTE_WIDGET_H
 
 #include <glib.h>
+#include <stddef.h>
 #include <sigc++/connection.h>
 
 #define SP_TYPE_ATTRIBUTE_WIDGET (sp_attribute_widget_get_type ())
diff -aur old/src/widgets/sp-color-notebook.cpp new/src/widgets/sp-color-notebook.cpp
--- old/src/widgets/sp-color-notebook.cpp	2010-07-13 03:48:40.691600000 +0000
+++ new/src/widgets/sp-color-notebook.cpp	2011-05-26 14:37:30.824789230 +0000
@@ -22,6 +22,7 @@
 #include <cstring>
 #include <string>
 #include <cstdlib>
+#include <cstddef>
 #include <gtk/gtk.h>
 #include <glibmm/i18n.h>
 
diff -aur old/src/xml/helper-observer.h new/src/xml/helper-observer.h
--- old/src/xml/helper-observer.h	2010-07-13 03:48:40.691600000 +0000
+++ new/src/xml/helper-observer.h	2011-05-26 14:37:30.824789230 +0000
@@ -5,6 +5,7 @@
 #include "node.h"
 #include "../sp-object.h"
 //#include "../sp-object-repr.h"
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 
 namespace Inkscape {
