--- src/extension/internal/pdfinput/pdf-parser.cpp.orig	2014-04-28 13:42:17.843140695 +0000
+++ src/extension/internal/pdfinput/pdf-parser.cpp	2014-04-28 14:05:10.646106685 +0000
@@ -861,7 +861,7 @@
 	  isolated = knockout = gFalse;
 	  if (!obj4.dictLookup(const_cast<char*>("CS"), &obj5)->isNull()) {
 #if defined(POPPLER_NEW_COLOR_SPACE_API) || defined(POPPLER_NEW_ERRORAPI)
-	    blendingColorSpace = GfxColorSpace::parse(&obj5, NULL);
+	    blendingColorSpace = GfxColorSpace::parse(&obj5, NULL, NULL);
 #else
 	    blendingColorSpace = GfxColorSpace::parse(&obj5);
 #endif
@@ -1085,9 +1085,9 @@
   res->lookupColorSpace(args[0].getName(), &obj);
 #if defined(POPPLER_NEW_COLOR_SPACE_API) || defined(POPPLER_NEW_ERRORAPI)
   if (obj.isNull()) {
-    colorSpace = GfxColorSpace::parse(&args[0], NULL);
+    colorSpace = GfxColorSpace::parse(&args[0], NULL, NULL);
   } else {
-    colorSpace = GfxColorSpace::parse(&obj, NULL);
+    colorSpace = GfxColorSpace::parse(&obj, NULL, NULL);
   }
 #else
   if (obj.isNull()) {
@@ -1120,9 +1120,9 @@
   res->lookupColorSpace(args[0].getName(), &obj);
 #if defined(POPPLER_NEW_COLOR_SPACE_API) || defined(POPPLER_NEW_ERRORAPI)
   if (obj.isNull()) {
-    colorSpace = GfxColorSpace::parse(&args[0], NULL);
+    colorSpace = GfxColorSpace::parse(&args[0], NULL, NULL);
   } else {
-    colorSpace = GfxColorSpace::parse(&obj, NULL);
+    colorSpace = GfxColorSpace::parse(&obj, NULL, NULL);
   }
 #else
   if (obj.isNull()) {
@@ -1213,7 +1213,7 @@
     }
 #if defined(POPPLER_NEW_COLOR_SPACE_API) || defined(POPPLER_NEW_ERRORAPI)
     if (args[numArgs-1].isName() &&
-	(pattern = res->lookupPattern(args[numArgs-1].getName(), NULL))) {
+	(pattern = res->lookupPattern(args[numArgs-1].getName(), NULL, NULL))) {
       state->setFillPattern(pattern);
       builder->updateStyle(state);
     }
@@ -1273,7 +1273,7 @@
     }
 #if defined(POPPLER_NEW_COLOR_SPACE_API) || defined(POPPLER_NEW_ERRORAPI)
     if (args[numArgs-1].isName() &&
-	(pattern = res->lookupPattern(args[numArgs-1].getName(), NULL))) {
+	(pattern = res->lookupPattern(args[numArgs-1].getName(), NULL, NULL))) {
       state->setStrokePattern(pattern);
       builder->updateStyle(state);
     }
@@ -1711,7 +1711,7 @@
   GBool savedState = gFalse;
 
 #if defined(POPPLER_NEW_COLOR_SPACE_API) || defined(POPPLER_NEW_ERRORAPI)
-  if (!(shading = res->lookupShading(args[0].getName(), NULL))) {
+  if (!(shading = res->lookupShading(args[0].getName(), NULL, NULL))) {
     return;
   }
 #else
@@ -2727,7 +2727,7 @@
     }
     if (!obj1.isNull()) {
 #if defined(POPPLER_NEW_COLOR_SPACE_API) || defined(POPPLER_NEW_ERRORAPI)
-      colorSpace = GfxColorSpace::parse(&obj1, NULL);
+      colorSpace = GfxColorSpace::parse(&obj1, NULL, NULL);
 #else
       colorSpace = GfxColorSpace::parse(&obj1);
 #endif
@@ -2816,7 +2816,7 @@
 	}
       }
 #if defined(POPPLER_NEW_COLOR_SPACE_API) || defined(POPPLER_NEW_ERRORAPI)
-      maskColorSpace = GfxColorSpace::parse(&obj1, NULL);
+      maskColorSpace = GfxColorSpace::parse(&obj1, NULL, NULL);
 #else
       maskColorSpace = GfxColorSpace::parse(&obj1);
 #endif
@@ -3007,7 +3007,7 @@
       transpGroup = gTrue;
       if (!obj1.dictLookup(const_cast<char*>("CS"), &obj3)->isNull()) {
 #if defined(POPPLER_NEW_COLOR_SPACE_API) || defined(POPPLER_NEW_ERRORAPI)
-	blendingColorSpace = GfxColorSpace::parse(&obj3, NULL);
+	blendingColorSpace = GfxColorSpace::parse(&obj3, NULL, NULL);
 #else
 	blendingColorSpace = GfxColorSpace::parse(&obj3);
 #endif
