--- src/extension/implementation/script.cpp.orig
+++ src/extension/implementation/script.cpp
@@ -88,7 +88,7 @@
 #ifdef WIN32
         {"python", "python-interpreter", "pythonw" },
 #else
         {"perl",   "perl-interpreter",   "perl"   },
-        {"python", "python-interpreter", "python" },
+        {"python", "python-interpreter", "python2" },
 #endif
         {"ruby",   "ruby-interpreter",   "ruby"   },
         {"shell",  "shell-interpreter",  "sh"     },
