--- a/plugins/player_mpv/src/mpvwidget.cpp
+++ b/plugins/player_mpv/src/mpvwidget.cpp
@@ -69,7 +69,7 @@
 }
 
 void MpvWidget::initializeGL() {
-    mpv_opengl_init_params gl_init_params{get_proc_address, nullptr, nullptr};
+    mpv_opengl_init_params gl_init_params{get_proc_address, nullptr};
     mpv_render_param params[]{
         {MPV_RENDER_PARAM_API_TYPE, const_cast<char *>(MPV_RENDER_API_TYPE_OPENGL)},
         {MPV_RENDER_PARAM_OPENGL_INIT_PARAMS, &gl_init_params},
