diff -ruN a/qimgv/3rdparty/QtOpenCV/cvmatandqimage.cpp qimgv/qimgv/3rdparty/QtOpenCV/cvmatandqimage.cpp
--- a/qimgv/3rdparty/QtOpenCV/cvmatandqimage.cpp	2021-09-29 23:46:05.000000000 +0200
+++ b/qimgv/3rdparty/QtOpenCV/cvmatandqimage.cpp	2026-08-14 19:24:42.931681534 +0200
@@ -28,7 +28,7 @@
 #include <QSysInfo>
 #include <QDebug>
 #include <cstring>
-#include <opencv4/opencv2/imgproc.hpp>
+#include <opencv2/imgproc.hpp>
 
 namespace QtOcv {
 namespace {
@@ -267,7 +267,7 @@
 #else
         format = QImage::Format_RGB32;
         cv::Mat mat_tmp;
-        cv::cvtColor(mat, mat_tmp, order == MCO_BGR ? CV_BGR2BGRA : CV_RGB2BGRA);
+        cv::cvtColor(mat, mat_tmp, order == MCO_BGR ? cv::COLOR_BGR2BGRA : cv::COLOR_RGB2BGRA);
 #if Q_BYTE_ORDER == Q_LITTLE_ENDIAN
         mat_adjustCn = mat_tmp;
 #else
diff -ruN a/qimgv/3rdparty/QtOpenCV/cvmatandqimage.h qimgv/qimgv/3rdparty/QtOpenCV/cvmatandqimage.h
--- a/qimgv/3rdparty/QtOpenCV/cvmatandqimage.h	2021-09-29 23:46:05.000000000 +0200
+++ b/qimgv/3rdparty/QtOpenCV/cvmatandqimage.h	2026-08-14 19:24:42.930514580 +0200
@@ -27,7 +27,7 @@
 #define CVMATANDQIMAGE_H
 
 #include <QtGui/qimage.h>
-#include <opencv4/opencv2/core.hpp>
+#include <opencv2/core.hpp>
 
 namespace QtOcv {
 
