--- a/pytest.ini	2023-06-16 22:48:37.211072200 -0300
+++ b/pytest.ini	2023-06-19 22:52:00.230063231 -0300
@@ -27,3 +27,5 @@
     ignore:\n\n  `numpy.distutils`:DeprecationWarning
 # Ignore mypy >= 0.971 DeprecationWarnings
     ignore:path is deprecated\. Use files\(\) instead:DeprecationWarning:mypy
+# Ignore warning raised when importing setuptools.command.
+    ignore:pkg_resources is deprecated as an API:DeprecationWarning
