--- a/pyproject.toml
+++ b/pyproject.toml
@@ -11,7 +11,7 @@
 build-backend = 'mesonpy'
 requires = [
     "meson-python>=0.12.1,<0.14.0", # already working with 0.13.x series at branch time
-    "Cython>=0.29.35,<3.0",  # when updating version, also update check in meson.build
+    "Cython>=0.29.35",
     "pybind11>=2.10.4,<2.11.0",
     "pythran>=0.12.0,<0.14.0", # already working with 0.13.x series at branch time
     # `wheel` is needed for non-isolated builds, given that `meson-python`
