Index: CMakeLists.txt
===================================================================
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -11,7 +11,7 @@ option (OPSTEP_FAST_AND_ROUGH "Fast and
 
 #ALL_CFLAGS := -fPIC -fno-common -ansi -pedantic -Wall -pipe -O2 -I. -I./include
 if (CMAKE_COMPILER_IS_GNUCC)
-    set (CMAKE_C_FLAGS "-fPIC -fno-common -ansi -pedantic -Wall -pipe -O2")
+    set (CMAKE_C_FLAGS "-fPIC -fno-common -ansi -pedantic -Wall -pipe ${CMAKE_C_FLAGS}")
 endif ()
 
 include_directories(BEFORE . include)
