--- a/cmake/modules/QtConfig.cmake
+++ b/cmake/modules/QtConfig.cmake
@@ -495,6 +495,9 @@ if (ENABLE_QT_TRANSLATIONS AND TS_FILES)
     set(APP_SPECIFIC_QT_TRANSLATIONS_AVAILABLE YES)
 
     # require the LinguistTools module (not adding it to QT_MODULES because we don't link against it)
+    if ("${QT_PACKAGE_PREFIX}" STREQUAL "Qt6")
+        find_package(Qt6 OPTIONAL_COMPONENTS Linguist CONFIG)
+    endif()
     find_package("${QT_LINGUIST_TOOLS_PACKAGE}")
     if (NOT "${${QT_LINGUIST_TOOLS_PACKAGE}_FOUND}" AND QT_HOST_PATH)
         # find the module within the host path when set (required for cross compilation with Qt 6 as the module is absent in
