--- a/prosody.cfg.lua.dist	Sun Mar 22 15:12:41 2026 +0100
+++ b/prosody.cfg.lua.dist	Tue Mar 24 12:21:26 2026 +0100
@@ -15,6 +15,9 @@
 -- blanks. Good luck, and happy Jabbering!
 
 
+------ Void settings ------
+pidfile = "/run/prosody/prosody.pid"
+
 ---------- Server-wide settings ----------
 -- Settings in this section apply to the whole server and are the default settings
 -- for any virtual hosts
@@ -185,8 +188,8 @@
 -- Logging configuration
 -- For advanced logging see https://prosody.im/doc/logging
 log = {
-	info = "prosody.log"; -- Change 'info' to 'debug' for verbose logging
-	error = "prosody.err";
+	info = "/var/log/prosody/prosody.log"; -- Change 'info' to 'debug' for verbose logging
+	error = "/var/log/prosody/prosody.err";
 	-- "*syslog"; -- Uncomment this for logging to syslog
 	-- "*console"; -- Log to the console, useful for debugging when running in the foreground
 }
