diff -ur a/packaging/pep517_backend/_cython_configuration.py b/packaging/pep517_backend/_cython_configuration.py
--- a/packaging/pep517_backend/_cython_configuration.py	2024-11-16 13:10:48.821864296 -0500
+++ b/packaging/pep517_backend/_cython_configuration.py	2024-11-16 13:12:37.985262597 -0500
@@ -7,7 +7,7 @@
 from pathlib import Path
 from sys import version_info as _python_version_tuple
 
-from expandvars import expandvars
+from os.path import expandvars
 
 from ._compat import load_toml_from_string  # noqa: WPS436
 from ._transformers import (  # noqa: WPS436
diff -ur a/pyproject.toml b/pyproject.toml
--- a/pyproject.toml	2024-11-16 13:10:48.821864296 -0500
+++ b/pyproject.toml	2024-11-16 13:10:59.594903606 -0500
@@ -3,7 +3,6 @@
   # NOTE: The following build dependencies are necessary for initial
   # NOTE: provisioning of the in-tree build backend located under
   # NOTE: `packaging/pep517_backend/`.
-  "expandvars",
   "setuptools >= 47",  # Minimum required for `version = attr:`
   "tomli; python_version < '3.11'",
 ]
