diff --git a/tests/input/local_exec_out_01.pln b/tests/input/local_exec_out_01.pln
index 6abff11..31e034a 100644
--- a/tests/input/local_exec_out_01.pln
+++ b/tests/input/local_exec_out_01.pln
@@ -1,12 +1,12 @@
 one:
 {
-	echo "LOCAL_HOSTNAME=$(hostname -s)"
+	echo "LOCAL_HOSTNAME=$(inetutils-hostname -s)"
 }
 	echo "$LOCAL_HOSTNAME"
 
 two:
 {
-	printf "LOCAL_HOSTNAME=$(hostname -s)"
+	printf "LOCAL_HOSTNAME=$(inetutils-hostname -s)"
 }
 	echo "$LOCAL_HOSTNAME"
 
