From 2c26cbb9292c5b06febe9b4a135e140ece4a7e46 Mon Sep 17 00:00:00 2001
From: q66 <daniel@octaforge.org>
Date: Thu, 17 Dec 2020 01:41:54 +0100
Subject: [PATCH] add powerpcle triple

---
 configure    | 4 ++++
 configure.ac | 4 ++++
 2 files changed, 8 insertions(+)

diff --git configure configure
index 2d379fe..97d1f9b 100755
--- a/configure
+++ b/configure
@@ -5302,7 +5302,11 @@ cat >> conftest.c <<EOF
         powerpc64-linux-gnu
 #  endif
 # elif defined(__powerpc__)
+#  if defined(__LITTLE_ENDIAN__)
+        powerpcle-linux-gnu
+#  else
         powerpc-linux-gnu
+#  endif
 # elif defined(__s390x__)
         s390x-linux-gnu
 # elif defined(__s390__)
diff --git configure.ac configure.ac
index c968d14..499ef7a 100644
--- a/configure.ac
+++ b/configure.ac
@@ -826,7 +826,11 @@ cat >> conftest.c <<EOF
         powerpc64-linux-gnu
 #  endif
 # elif defined(__powerpc__)
+#  if defined(__LITTLE_ENDIAN__)
+        powerpcle-linux-gnu
+#  else
         powerpc-linux-gnu
+#  endif
 # elif defined(__s390x__)
         s390x-linux-gnu
 # elif defined(__s390__)
-- 
2.29.2

