--- a/configure
+++ b/configure
@@ -5205,6 +5205,14 @@
 
 MULTIARCH=$($CC --print-multiarch 2>/dev/null)
 
+if test x$MULTIARCH = xpowerpc-linux-musl
+then
+	MULTIARCH="powerpc-linux-gnu"
+fi
+if test x$MULTIARCH = xpowerpcle-linux-musl
+then
+	MULTIARCH="powerpcle-linux-gnu"
+fi
 
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for the platform triplet based on compiler characteristics" >&5
 $as_echo_n "checking for the platform triplet based on compiler characteristics... " >&6; }
