Source: https://gitlab.gnome.org/GNOME/libpeas/-/issues/65

diff --git a/tests/libpeas/extension-py.c b/tests/libpeas/extension-py.c
index 4ac6fe5..995e698 100644
--- a/tests/libpeas/extension-py.c
+++ b/tests/libpeas/extension-py.c
@@ -51,7 +51,7 @@ test_extension_py_instance_refcount (PeasEngine     *engine,
 
   g_object_add_weak_pointer (extension, (gpointer *) &extension);
 
-  g_assert_cmpint (extension->ref_count, ==, 2);
+  g_assert_cmpint (extension->ref_count, ==, 1);
 
   g_object_unref (extension);
   g_assert_true (extension == NULL);
@@ -88,7 +88,7 @@ test_extension_py_activatable_subject_refcount (PeasEngine     *engine,
   wrapper = g_object_get_data (object, "PyGObject::wrapper");
   g_assert_cmpint (wrapper->ob_refcnt, ==, 1);
 
-  g_assert_cmpint (G_OBJECT (extension)->ref_count, ==, 2);
+  g_assert_cmpint (G_OBJECT (extension)->ref_count, ==, 1);
   g_object_unref (extension);
 
   /* We unreffed the extension, so it should have been destroyed and our dummy

