--- a/configure.ac
+++ b/configure.ac
@@ -56,18 +56,8 @@
 	exec_prefix="$prefix"
 fi
 
-if test "x$WGET" = "x"; then
-	if test "x$CURL" = "x"; then
-		AC_MSG_WARN([** Neither wget nor curl could be found.])
-		AC_MSG_WARN([** IANA PEN database will not be installed by `make install` !])
-	else
-		DOWNLOAD="$CURL --location --progress-bar"
-		AM_CONDITIONAL([DOWNLOAD], [true])
-	fi
-else
-	DOWNLOAD="$WGET -c -nd -O -"
-	AM_CONDITIONAL([DOWNLOAD], [true])
-fi
+DOWNLOAD=
+AM_CONDITIONAL([DOWNLOAD], [false])
 
 AC_MSG_WARN([** Download is:])
 AC_MSG_WARN($DOWNLOAD)
--- a/Makefile.am
+++ b/Makefile.am
@@ -69,9 +69,9 @@
 
 else
 
-install-pen-database:
-	@echo "*** NOT installing the IANA PEN database."
-	@echo "*** Don't know how to download it."
+install-pen-database:
+	mkdir -m 755 -p $(DESTDIR)$(IANADIR)
+	$(INSTALL_DATA) enterprise-numbers $(DESTDIR)$(IANADIR)/
 
 uninstall-pen-database:
 	@echo "*** NOT uninstalling the IANA PEN database."
