--- a/data/info.febvre.Komikku.appdata.xml.in
+++ b/data/info.febvre.Komikku.appdata.xml.in
@@ -49,28 +49,28 @@
 
     <screenshots>
         <screenshot type="default">
-            <image>https://gitlab.com/valos/Komikku/raw/master/screenshots/library.png</image>
+            <image>https://gitlab.com/valos/Komikku/raw/main/screenshots/library.png</image>
         </screenshot>
         <screenshot>
-            <image>https://gitlab.com/valos/Komikku/raw/master/screenshots/library-with-categories.png</image>
+            <image>https://gitlab.com/valos/Komikku/raw/main/screenshots/library-with-categories.png</image>
         </screenshot>
         <screenshot>
-            <image>https://gitlab.com/valos/Komikku/raw/master/screenshots/preferences.png</image>
+            <image>https://gitlab.com/valos/Komikku/raw/main/screenshots/preferences.png</image>
         </screenshot>
         <screenshot>
-            <image>https://gitlab.com/valos/Komikku/raw/master/screenshots/explorer.png</image>
+            <image>https://gitlab.com/valos/Komikku/raw/main/screenshots/explorer.png</image>
         </screenshot>
         <screenshot>
-            <image>https://gitlab.com/valos/Komikku/raw/master/screenshots/categories-editor.png</image>
+            <image>https://gitlab.com/valos/Komikku/raw/main/screenshots/categories-editor.png</image>
         </screenshot>
         <screenshot>
-            <image>https://gitlab.com/valos/Komikku/raw/master/screenshots/card-info.png</image>
+            <image>https://gitlab.com/valos/Komikku/raw/main/screenshots/card-info.png</image>
         </screenshot>
         <screenshot>
-            <image>https://gitlab.com/valos/Komikku/raw/master/screenshots/card-chapters.png</image>
+            <image>https://gitlab.com/valos/Komikku/raw/main/screenshots/card-chapters.png</image>
         </screenshot>
         <screenshot>
-            <image>https://gitlab.com/valos/Komikku/raw/master/screenshots/reader.png</image>
+            <image>https://gitlab.com/valos/Komikku/raw/main/screenshots/reader.png</image>
         </screenshot>
     </screenshots>
 
