diff --git a/configure b/configure
index 2075cf724d22b4799727896a67d74ecc20900f88..24624537c398efc3abc2470e934132c8bbdae33e 100755
--- a/configure
+++ b/configure
@@ -317,14 +317,14 @@ fi
 stripdir prefix
 for i in exec_prefix dynlibdir libexecdir bindir libdir includedir sysconfdir \
 		datarootdir mandir shebangdir livedir skel system_dir system_log \
-		service_system module_system script_module service_adm module_adm service_admconf sproot sysdeps ; do
+		service_system module_system script_system service_adm module_adm service_admconf sproot sysdeps ; do
   eval tmp=\${$i}
   eval $i=$tmp
   stripdir $i
 done
 
 stripdir datarootdir
-for i in service_system module_system script_module mandir; do
+for i in service_system module_system script_system mandir; do
   eval tmp=\${$i}
   eval $i=$tmp
   stripdir $i
