diff --git a/src/lib66/ss_environ.c b/src/lib66/ss_environ.c
index 05f2697c45d3958147d9243c24a5389897c2bd41..12c2b8fcbc99fcb83adf576c1541984a4b23e8e5 100644
--- a/src/lib66/ss_environ.c
+++ b/src/lib66/ss_environ.c
@@ -185,7 +185,7 @@ int env_make_symlink(stralloc *dst,stralloc *old_dst,sv_alltype *sv,uint8_t conf
 	 * a directory. File means old format, in this case we enforce to pass
 	 * to the new one. */
 	r = lstat(sym_version,&st) ;
-	if (S_ISLNK(st.st_mode))
+	if (S_ISLNK(st.st_mode) && !r)
 	{
 		if (sarealpath(&saversion,sym_version) == -1)
 			log_warn_return(LOG_EXIT_ZERO,"sarealpath of: ",sym_version) ;
diff --git a/src/lib66/ssexec_disable.c b/src/lib66/ssexec_disable.c
index ed884021cea729be8d22af9323fa5ec67c7a7b56..5e056509171fcaaa48ae52ea2390400edc1a3312 100644
--- a/src/lib66/ssexec_disable.c
+++ b/src/lib66/ssexec_disable.c
@@ -213,7 +213,7 @@ int ssexec_disable(int argc, char const *const *argv,char const *const *envp,sse
 		
 		/** The force options can be only used if the service is not marked initialized.
 		 * This option should only be used when we have a inconsistent state between 
-		 * the /var/lib/66/system/<tree>/servicedirs/* and /var/lib/66/system/<tree>/.resolve
+		 * the /var/lib/66/system/<tree>/servicedirs/ and /var/lib/66/system/<tree>/.resolve
 		 * directory meaning a service which is not present in the compiled db but its resolve file
 		 * exist.*/
 		if (FORCE && ss_state_check(state,name))
