diff --git a/doc/make-man.sh b/doc/make-man.sh
index fd5d9cd..0c3fee6 100755
--- a/doc/make-man.sh
+++ b/doc/make-man.sh
@@ -14,29 +14,26 @@ done
 
 for i in ${man1}; do
     lowdown -s -Tman doc/"${i}".md -o doc/man/man1/"${i}".1 || exit 1
-    var=$(head -n1 < doc/man/man1/"${i}".1)
+    var=$( sed -n -e '/^.TH/p' < doc/man/man1/"${i}".1)
     var=$(printf '%s' "$var" | tr '7' '1')
-    var="${var} \"\" \"General Commands Manual\""
     sed -i "s!^.TH.*!${var}!" doc/man/man1/"${i}".1 || exit 1
-    sed -i '2,5d' doc/man/man1/"${i}".1 || exit 1
+    sed -i '4,8d' doc/man/man1/"${i}".1 || exit 1
 done
 
 for i in ${man5}; do
     lowdown -s -Tman doc/"${i}".md -o doc/man/man5/"${i}".5 || exit 1
-    var=$(head -n1 < doc/man/man5/"${i}".5)
+    var=$( sed -n -e '/^.TH/p' < doc/man/man5/"${i}".5)
     var=$(printf '%s' "$var" | tr '7' '5')
-    var="${var} \"\" \"File Formats Manual\""
     sed -i "s!^.TH.*!${var}!" doc/man/man5/"${i}".5 || exit 1
-    sed -i '2,5d' doc/man/man5/"${i}".5 || exit 1
+    sed -i '4,8d' doc/man/man5/"${i}".5 || exit 1
 done
 
 for i in ${man8}; do
     lowdown -s -Tman doc/"${i}".md -o doc/man/man8/"${i}".8 || exit 1
-    var=$(head -n1 < doc/man/man8/"${i}".8)
+    var=$( sed -n -e '/^.TH/p' < doc/man/man8/"${i}".8)
     var=$(printf '%s' "$var" | tr '7' '8')
-    var="${var} \"\" \"System Administration\""
     sed -i "s!^.TH.*!${var}!" doc/man/man8/"${i}".8 || exit 1
-    sed -i '2,5d' doc/man/man8/"${i}".8 || exit 1
+    sed -i '4,8d' doc/man/man8/"${i}".8 || exit 1
 done
 
 exit 0
