diff -p1ru a/src/sound-menu.vala b/src/sound-menu.vala
--- a/src/sound-menu.vala	2018-06-13 04:42:13.000000000 +0800
+++ b/src/sound-menu.vala	2024-08-08 12:28:25.462427836 +0800
@@ -459,3 +459,3 @@ public class SoundMenu: Object
 		var slider = new MenuItem (label, action);
-		slider.set_attribute ("x-canonical-type", "s", "com.canonical.unity.slider");
+		slider.set_attribute ("x-ayatana-type", "s", "org.ayatana.indicator.slider");
 		slider.set_attribute_value ("min-icon", min_icon.serialize ());
diff -p1ru a/tests/integration/indicator-sound-test-base.cpp b/tests/integration/indicator-sound-test-base.cpp
--- a/tests/integration/indicator-sound-test-base.cpp	2018-06-13 04:42:13.000000000 +0800
+++ b/tests/integration/indicator-sound-test-base.cpp	2024-08-08 12:29:03.730284327 +0800
@@ -409,3 +409,3 @@ unity::gmenuharness::MenuItemMatcher Ind
             .double_attribute("step", 0.01)
-            .string_attribute("x-canonical-type", "com.canonical.unity.slider")
+            .string_attribute("x-ayatana-type", "org.ayatana.indicator.slider")
             .themed_icon("max-icon", {"audio-volume-high-panel", "audio-volume-high", "audio-volume", "audio"})
@@ -423,3 +423,3 @@ unity::gmenuharness::MenuItemMatcher Ind
             .double_attribute("step", 0.01)
-            .string_attribute("x-canonical-type", "com.canonical.unity.slider")
+            .string_attribute("x-ayatana-type", "org.ayatana.indicator.slider")
             .themed_icon("max-icon", {"audio-input-microphone-high-panel", "audio-input-microphone-high", "audio-input-microphone", "audio-input", "audio"})
