From 46e86c50044e07c7427233495c0c21adfdb47891 Mon Sep 17 00:00:00 2001
From: Jan-Michael Brummer <jan-michael.brummer1@volkswagen.de>
Date: Fri, 8 Mar 2024 19:58:43 +0100
Subject: [PATCH] Add HTTP/2 support

---
 libuhttpmock/uhm-server.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/libuhttpmock/uhm-server.c b/libuhttpmock/uhm-server.c
index 4451a91..da5d3a5 100644
--- a/libuhttpmock/uhm-server.c
+++ b/libuhttpmock/uhm-server.c
@@ -873,6 +873,9 @@ trace_to_soup_message (const gchar *trace, GUri *base_uri)
 	} else if (strncmp (trace, "HTTP/1.0", strlen ("HTTP/1.0")) == 0) {
 		http_version = SOUP_HTTP_1_0;
 		trace += strlen ("HTTP/1.0");
+	} else if (strncmp (trace, "HTTP/2", strlen ("HTTP/2")) == 0) {
+		http_version = SOUP_HTTP_2_0;
+		trace += strlen ("HTTP/2");
 	} else {
 		g_warning ("Unrecognised HTTP version ‘%s’.", trace);
 		http_version = SOUP_HTTP_1_1;
@@ -914,6 +917,9 @@ trace_to_soup_message (const gchar *trace, GUri *base_uri)
 	} else if (strncmp (trace, "HTTP/1.0", strlen ("HTTP/1.0")) == 0) {
 		http_version = SOUP_HTTP_1_0;
 		trace += strlen ("HTTP/1.0");
+	} else if (strncmp (trace, "HTTP/2", strlen ("HTTP/2")) == 0) {
+		http_version = SOUP_HTTP_2_0;
+		trace += strlen ("HTTP/2");
 	} else {
 		g_warning ("Unrecognised HTTP version ‘%s’.", trace);
 	}
-- 
GitLab


