From f06950a6d4af218e4856019180a169f5b4026ef5 Mon Sep 17 00:00:00 2001
From: elbachir-one <bachiralfa@gmail.com>
Date: Wed, 2 Jul 2025 18:01:19 +0100
Subject: [PATCH] Fixed man pages location

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index dcf2347..76baaf2 100644
--- a/setup.py
+++ b/setup.py
@@ -38,7 +38,7 @@ setuptools.setup(
         "Programming Language :: Python :: 3.5",
         "Programming Language :: Python :: 3.6",
     ],
-    data_files=[('man/man1', ['data/man/man1/wal.1'])],
+    data_files=[('share/man/man1', ['data/man/man1/wal.1'])],
     packages=["pywal"],
     entry_points={"console_scripts": ["wal=pywal.__main__:main"]},
     python_requires=">=3.5",
-- 
2.50.0

