--- a/qt/meson.build	2024-02-24 22:22:21.000000000 +0100
+++ -	2024-03-12 13:03:47.811844901 +0100
@@ -127,6 +127,7 @@
 
 cmake_data = configuration_data()
 cmake_data.set('LIBDIR_FULL', join_paths(get_option('prefix'), get_option('libdir')))
+cmake_data.set('LIBDIR', get_option('libdir'))
 cmake_data.set('PREFIX', get_option('prefix'))
 cmake_data.set('VERSION', as_version)
 
--- a/qt/cmake/AppStreamQtConfig.cmake.in	2024-02-24 22:22:21.000000000 +0100
+++ -	2024-03-12 13:04:12.791711630 +0100
@@ -32,7 +32,7 @@
 set_target_properties(AppStreamQt PROPERTIES
   INTERFACE_INCLUDE_DIRECTORIES "${PACKAGE_PREFIX_DIR}/include/"
   INTERFACE_LINK_LIBRARIES "Qt::Core"
-  IMPORTED_LOCATION "@LIBDIR_FULL@/libAppStreamQt.so.${AppStreamQt_VERSION}"
+  IMPORTED_LOCATION "${PACKAGE_PREFIX_DIR}/@LIBDIR@/libAppStreamQt.so.${AppStreamQt_VERSION}"
   IMPORTED_SONAME "libAppStreamQt.${AppStreamQt_VERSION_MAJOR}"
 )
 
