From 717f1234b6f26c07af4a7758840dcd0e846133a9 Mon Sep 17 00:00:00 2001
From: Naz <ndpm13@ch-naseem.com>
Date: Fri, 17 Apr 2026 16:56:04 +0100
Subject: [PATCH] update YAML test data for libfyaml>=0.9.5 compatibility

---
 tests/test-yamldata.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/tests/test-yamldata.c b/tests/test-yamldata.c
index b470595..48001d2 100644
--- a/tests/test-yamldata.c
+++ b/tests/test-yamldata.c
@@ -1057,7 +1057,7 @@ static const gchar *yamldata_relations_field = "Type: generic\n"
 					       "- id: org.example.old_test\n"
 					       "Requires:\n"
 					       "- kernel: Linux\n"
-					       "  version: \">= 4.15\"\n"
+					       "  version: \'>= 4.15\'\n"
 					       "- id: org.example.TestDependency\n"
 					       "  version: == 1.2\n"
 					       "- display_length: 4200\n"
@@ -1881,10 +1881,10 @@ test_yaml_rw_branding (void)
 				       "  colors:\n"
 				       "  - type: primary\n"
 				       "    scheme-preference: light\n"
-				       "    value: \"#ff00ff\"\n"
+				       "    value: \'#ff00ff\'\n"
 				       "  - type: primary\n"
 				       "    scheme-preference: dark\n"
-				       "    value: \"#993d3d\"\n";
+				       "    value: \'#993d3d\'\n";
 	g_autoptr(AsComponent) cpt = NULL;
 	g_autofree gchar *res = NULL;
 	AsBranding *branding;
-- 
2.53.0

