From c44560949ec78dd13fe1394bf2957e4fd5adec79 Mon Sep 17 00:00:00 2001
From: Brian Evans <grknight@gentoo.org>
Date: Wed, 14 Mar 2018 20:00:22 -0400
Subject: [PATCH] Get rid of custom target

---
 CMakeLists.txt | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

--- CMakeLists.txt
+++ CMakeLists.txt
@@ -45,7 +45,5 @@ project(SPIRV-Headers)
 #   3. cmake --build . install-headers
 
 file(GLOB_RECURSE FILES include/spirv/*)
-add_custom_target(install-headers
-  COMMAND cmake -E copy_directory ${CMAKE_CURRENT_SOURCE_DIR}/include/spirv ${CMAKE_INSTALL_PREFIX}/include/spirv)
-
+INSTALL(DIRECTORY include/spirv/ DESTINATION include/spirv)
 add_subdirectory(example)
-- 
2.16.2


