From 6395b93cc8461f8a3da1877c18b4abff490a3965 Mon Sep 17 00:00:00 2001
From: Ailin Nemui <ailin@d5421s.localdomain>
Date: Sun, 13 Jul 2025 19:41:36 +0200
Subject: [PATCH] new pm_to_blib syntax in perl 5.42

---
 src/perl/Makefile_silent.pm | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/perl/Makefile_silent.pm b/src/perl/Makefile_silent.pm
index b5d71d66d..11259e176 100644
--- a/src/perl/Makefile_silent.pm
+++ b/src/perl/Makefile_silent.pm
@@ -22,7 +22,7 @@ my $verb = $AM_DEFAULT_VERBOSITY;
     }
     sub pm_to_blib {
 	my $ret = shift->SUPER::pm_to_blib(@_);
-	$ret =~ s{^(\t(?:- ?)?)(?:\$\(NOECHO\) ?)?(.*-e ['"]pm_to_blib(.*\\\n)*.*)$}{$1\$(PL_AM_V_BLIB)$2\$(PL_AM_V_BLIB_Hide)}mg;
+	$ret =~ s{^(\t(?:- ?)?)(?:\$\(NOECHO\) ?)?((?:\\\r?\n|.)*-e ['"]pm_to_blib(.*\\\n)*.*)$}{$1\$(PL_AM_V_BLIB)$2\$(PL_AM_V_BLIB_Hide)}mg;
 	$ret
     }
     sub post_constants {
