Author: <gaudenz@debian.org>
Description: Pass through key events of modifier keys
--- a/mouseemu.c
+++ b/mouseemu.c
@@ -174,7 +174,7 @@
 	/* I think its best not to pass scroll, or experiment with not passing the release if
 	 * we actually used it for scrolling (but some apps may get stuck?)
 	 */
-		if (inp.code != b2_key && inp.code != b3_key && !b2_mod_pressed && !b3_mod_pressed) {
+		if (inp.code != b2_key && inp.code != b3_key) {
 			passthrough(ui_keyb_fd, inp);
 		}
 	}
