Author: Colin Watson <cjwatson@ubuntu.com>
Description: No description.
--- a/mouseemu.c
+++ b/mouseemu.c
@@ -30,6 +30,9 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <linux/input.h>
+#ifndef BUS_VIRTUAL /* new in Linux 2.6.19 */
+#define BUS_VIRTUAL 0x06
+#endif
 #include <linux/uinput.h>
 #include "mouseemu.h"
 #include "defkeys.h"
@@ -434,7 +437,7 @@
 	
 	strcpy(device.name, "Mouseemu virtual keyboard");
 
-	device.id.bustype = 0;
+	device.id.bustype = BUS_VIRTUAL;
 	device.id.vendor = 0x1F;
 	device.id.product = 0x1F;
 	device.id.version = 0;
@@ -476,7 +479,7 @@
 
         strcpy(device.name, "Mouseemu virtual mouse");
 
-        device.id.bustype = 0;
+        device.id.bustype = BUS_VIRTUAL;
         device.id.vendor = 0x1F;
         device.id.product = 0x1E;
         device.id.version = 0;
