From f97247434ecda2cd2c451da5c73c3af98dfaea35 Mon Sep 17 00:00:00 2001
From: Mark Harmstone <mark@harmstone.com>
Date: Thu, 7 Oct 2021 17:59:35 +0100
Subject: [PATCH] CMakeLists.txt: use GNUInstallDirs to determine sbin location

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index cfaa571..17ded78 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -89,5 +89,5 @@ else()
     target_compile_options(ntfs2btrfs PRIVATE -Wall -Wextra -Wno-address-of-packed-member -Wconversion -Wno-unknown-pragmas -Werror=pointer-arith)
 endif()
 
-install(TARGETS ntfs2btrfs DESTINATION sbin)
+install(TARGETS ntfs2btrfs DESTINATION ${CMAKE_INSTALL_SBINDIR})
 install(FILES ${CMAKE_CURRENT_BINARY_DIR}/ntfs2btrfs.8 DESTINATION ${CMAKE_INSTALL_MANDIR}/man8)
-- 
2.33.0

