--- src/libserver/dkim.c.orig	2016-04-30 12:01:07.681284455 +0200
+++ src/libserver/dkim.c	2016-04-30 12:01:17.738283703 +0200
@@ -886,7 +886,7 @@ rspamd_create_dkim_context (const gchar
 
 		return NULL;
 	}
-#if OPENSSL_VERSION_NUMBER < 0x10100000L
+#if OPENSSL_VERSION_NUMBER < 0x10100000L || 1
 	ctx->body_hash = EVP_MD_CTX_create ();
 	EVP_DigestInit_ex (ctx->body_hash, md_alg, NULL);
 	ctx->headers_hash = EVP_MD_CTX_create ();
