diff --git a/CMakeLists.txt b/CMakeLists.txt
index 98e8c2ec0..5f8939bb5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -131,7 +131,6 @@ include_directories(
         "${CMAKE_SOURCE_DIR}/contrib/fpconv"
         "${CMAKE_SOURCE_DIR}/contrib/libottery"
         "${CMAKE_SOURCE_DIR}/contrib/cdb"
-        "${CMAKE_SOURCE_DIR}/contrib/snowball/include"
         "${CMAKE_SOURCE_DIR}/contrib/librdns"
         "${CMAKE_SOURCE_DIR}/contrib/aho-corasick"
         "${CMAKE_SOURCE_DIR}/contrib/lc-btrie"
diff --git a/cmake/AddDependencySubdirectories.cmake b/cmake/AddDependencySubdirectories.cmake
index 61781b511..560a4383e 100644
--- a/cmake/AddDependencySubdirectories.cmake
+++ b/cmake/AddDependencySubdirectories.cmake
@@ -32,7 +32,6 @@ function(AddDependencySubdirectories)
 
     # Optional dependencies based on configuration
     if (ENABLE_SNOWBALL)
-        add_subdirectory(contrib/snowball)
         set(WITH_SNOWBALL 1 PARENT_SCOPE)
     endif ()

