diff --git a/src/tbbmalloc_proxy/proxy.cpp b/src/tbbmalloc_proxy/proxy.cpp
index 6cf1f49f64..3acbaa8d9d 100644
--- a/src/tbbmalloc_proxy/proxy.cpp
+++ b/src/tbbmalloc_proxy/proxy.cpp
@@ -24,6 +24,7 @@
 // of aligned_alloc as required by new C++ standard, this makes it hard to
 // redefine aligned_alloc here. However, running on systems with new libc
 // version, it still needs it to be redefined, thus tricking system headers
+#if defined(__GLIBC__)
 #if defined(__GLIBC_PREREQ)
 #if !__GLIBC_PREREQ(2, 16) && _GLIBCXX_HAVE_ALIGNED_ALLOC
 // tell <cstdlib> that there is no aligned_alloc
@@ -35,6 +36,7 @@
 #undef aligned_alloc
 #endif // !__GLIBC_PREREQ(2, 16) && _GLIBCXX_HAVE_ALIGNED_ALLOC
 #endif // defined(__GLIBC_PREREQ)
+#endif // defined(__GLIBC__)
 #include <cstdlib>
 #endif // __unix__ && !__ANDROID__
 
