--- a/common/libBuffer/include/CommonFWriteBufferFlusher.h	2016-02-23 00:57:35.000000000 +0100
+++ b/common/libBuffer/include/CommonFWriteBufferFlusher.h	2016-02-29 02:18:38.673566829 +0100
@@ -44,6 +44,7 @@
 typedef int64_t __int64;
 #else
 #if (defined(__GNUC__) && !defined(__STRICT_ANSI__) && !defined(__MINGW32__)) || (__STDC_VERSION__ >= 199901L)
+#include <stdint.h>
 typedef int64_t __int64;
 #endif
 #endif
