--- a/common/libBuffer/include/CommonFWriteBufferFlusher.h	2016-04-11 20:10:51.000000000 +0200
+++ b/common/libBuffer/include/CommonFWriteBufferFlusher.h	2016-04-12 10:56:39.685988592 +0200
@@ -35,7 +35,7 @@
 #ifdef __GNUC__
 #   include <cstdlib> /* size_t */
 #   include <cstdio>  /* FILE */
-#ifdef __FreeBSD__
+#if defined(__FreeBSD__) || !defined(__GLIBC__)
 #include <stdint.h> /* int64_t */
 #endif
 #endif
