--- a/COLLADAStreamWriter/src/COLLADASWLibraryAnimations.cpp	2018-11-26 23:43:10.000000000 +0100
+++ -	2023-02-23 13:47:42.819812753 +0100
@@ -62,7 +62,7 @@
 
     //---------------------------------------------------------------
     LibraryAnimations::LibraryAnimations ( COLLADASW::StreamWriter * streamWriter )
-            : Library ( streamWriter, CSWC::CSW_ELEMENT_LIBRARY_ANIMATIONS ), mOpenAnimations ( NULL )
+            : Library ( streamWriter, CSWC::CSW_ELEMENT_LIBRARY_ANIMATIONS ), mOpenAnimations ( std::vector<Animation*> {} )
     {}
 
     //---------------------------------------------------------------
