From c6891596fc4a1f41dae60c5a8bd08dd7a933d625 Mon Sep 17 00:00:00 2001
From: Nathan Owens <ndowens@artixlinux.org>
Date: Sun, 22 Nov 2020 19:56:11 -0600
Subject: [PATCH] musl build fix

---
 tools/find_event_devices.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git tools/find_event_devices.c tools/find_event_devices.c
index 2a43d54..70314e8 100644
--- a/tools/find_event_devices.c
+++ b/tools/find_event_devices.c
@@ -32,7 +32,7 @@
 #include <sys/stat.h>
 #include <fcntl.h>
 #include <unistd.h>
-
+#include <limits.h>
 
 #define DEV_INPUT_EVENT "/dev/input"
 #define EVENT_DEV_NAME "event"
-- 
2.29.2

