diff -up crack-attack-1.1.14/src/Sound.cxx~ crack-attack-1.1.14/src/Sound.cxx
--- src/Sound.cxx	2008-06-24 11:20:22.000000000 +0200
+++ src/Sound.cxx	2008-06-24 11:20:22.000000000 +0200
@@ -41,7 +41,7 @@ ChunkMap chunks;
 int audio_rate          = 22050;
 int audio_channels      = 2;
 int audio_buffers       = 1024; /* small enough buffer to get synchronized sound */
-Uint16 audio_format     = AUDIO_S16; /* 16-bit stereo */
+Uint16 audio_format     = AUDIO_S16SYS; /* 16-bit stereo */
 int has_audio_available = 0;
 
 void Sound::initialize( void )
