--- src/Music.h	2007-01-12 10:54:25.000000000 +0100
+++ src/Music.h	2007-01-12 10:54:25.000000000 +0100
@@ -34,10 +34,10 @@
 #include <vector>
 
 #define GC_MUSIC_FILENAME_LENGTH 128
-#define GC_MUSIC_PRELUDE_TRACK   "prelude"
-#define GC_MUSIC_GAME_TRACK      "game"
-#define GC_MUSIC_GAMEOVER_TRACK  "gameover"
-#define GC_MUSIC_YOUWIN_TRACK    "youwin"
+#define GC_MUSIC_PRELUDE_TRACK   "prelude.xm"
+#define GC_MUSIC_GAME_TRACK      "game.xm"
+#define GC_MUSIC_GAMEOVER_TRACK  "gameover.xm"
+#define GC_MUSIC_YOUWIN_TRACK    "youwin.xm"
 
 class Music {
 	public:
--- src/Music.cxx	2007-01-12 11:23:15.000000000 +0100
+++ src/Music.cxx	2007-01-12 11:23:15.000000000 +0100
@@ -91,7 +91,7 @@
 	cout << "Playing " << music_filelist[current_track].c_str() << endl;
 	#endif
 	music = Mix_LoadMUS( music_filelist[current_track].c_str() );
-	Mix_VolumeMusic( MIX_MAX_VOLUME / 4 );
+	Mix_VolumeMusic( (MIX_MAX_VOLUME * 8) / 10 );
 	Mix_PlayMusic( music , 0 );
 	Mix_HookMusicFinished(Music::finished);
 	keep_playing = 1;
@@ -126,7 +126,7 @@
 	cout << "Playing " << Track.c_str() << endl;
 	#endif
 	music = Mix_LoadMUS( Track.c_str() );
-	Mix_VolumeMusic( MIX_MAX_VOLUME / 4 );
+	Mix_VolumeMusic( (MIX_MAX_VOLUME * 8) / 10 );
 	Mix_PlayMusic( music , 0 );
 	keep_playing = 0;
 }
