--- Makefile.in.orig	2009-05-13 04:24:38.602959508 +0200
+++ Makefile.in	2009-05-13 04:25:11.570694167 +0200
@@ -250,9 +250,7 @@ dist_noinst_DATA = $(dist_man_MANS)
 #
 ul_libblkid_srcdir = $(top_srcdir)/libs/blkid/src
 ul_libblkid_la = $(top_builddir)/libs/blkid/src/libblkid.la
-SUBDIRS = include disk-utils fdisk getopt lib libs login-utils \
-	misc-utils po schedutils sys-utils text-utils tests \
-	$(am__append_1) $(am__append_2) $(am__append_3)
+SUBDIRS = libs
 RCS_FIND_IGNORE := \( -name SCCS -o -name BitKeeper -o -name .svn -o \
 		-name CVS -o -name .pc -o -name .hg -o -name .git \) -prune -o
 
