--- make-linux.mk	2019-07-31 22:25:58.000000000 +0200
+++ make-linux.mk	2019-08-10 07:30:54.288571780 +0200
@@ -147,6 +147,9 @@
 ifeq ($(CC_MACH),ppc64el)
 	ZT_ARCHITECTURE=8
 endif
+ifeq ($(CC_MACH),powerpc64)
+	ZT_ARCHITECTURE=8
+endif
 ifeq ($(CC_MACH),i386)
 	ZT_ARCHITECTURE=1
 endif
@@ -177,22 +177,22 @@
 ifeq ($(CC_MACH),armv6)
 	ZT_ARCHITECTURE=3
 	override DEFS+=-DZT_NO_TYPE_PUNNING
-	ZT_USE_ARM32_NEON_ASM_CRYPTO=1
+	ZT_USE_ARM32_NEON_ASM_CRYPTO=0
 endif
 ifeq ($(CC_MACH),armv6l)
 	ZT_ARCHITECTURE=3
 	override DEFS+=-DZT_NO_TYPE_PUNNING
-	ZT_USE_ARM32_NEON_ASM_CRYPTO=1
+	ZT_USE_ARM32_NEON_ASM_CRYPTO=0
 endif
 ifeq ($(CC_MACH),armv6zk)
 	ZT_ARCHITECTURE=3
 	override DEFS+=-DZT_NO_TYPE_PUNNING
-	ZT_USE_ARM32_NEON_ASM_CRYPTO=1
+	ZT_USE_ARM32_NEON_ASM_CRYPTO=0
 endif
 ifeq ($(CC_MACH),armv6kz)
 	ZT_ARCHITECTURE=3
 	override DEFS+=-DZT_NO_TYPE_PUNNING
-	ZT_USE_ARM32_NEON_ASM_CRYPTO=1
+	ZT_USE_ARM32_NEON_ASM_CRYPTO=0
 endif
 ifeq ($(CC_MACH),armv7)
 	ZT_ARCHITECTURE=3
@@ -261,19 +261,6 @@
 	override LDFLAGS+=-Wl,--wrap=memcpy -static-libstdc++
 endif
 
-# ARM32 hell -- use conservative CFLAGS
-ifeq ($(ZT_ARCHITECTURE),3)
-	ifeq ($(shell if [ -e /usr/bin/dpkg ]; then dpkg --print-architecture; fi),armel)
-		override CFLAGS+=-march=armv5 -mfloat-abi=soft -msoft-float -mno-unaligned-access -marm
-		override CXXFLAGS+=-march=armv5 -mfloat-abi=soft -msoft-float -mno-unaligned-access -marm
-		ZT_USE_ARM32_NEON_ASM_CRYPTO=0
-	else
-		override CFLAGS+=-march=armv5 -mno-unaligned-access -marm
-		override CXXFLAGS+=-march=armv5 -mno-unaligned-access -marm
-		ZT_USE_ARM32_NEON_ASM_CRYPTO=0
-	endif
-endif
-
 # Build faster crypto on some targets
 ifeq ($(ZT_USE_X64_ASM_SALSA),1)
 	override DEFS+=-DZT_USE_X64_ASM_SALSA2012
