--- make-linux.mk	2020-12-01 01:33:39.000000000 +0100
+++ make-linux.mk	2020-12-19 13:00:35.311711026 +0100
@@ -144,6 +144,9 @@
 ifeq ($(CC_MACH),ppc64el)
 	ZT_ARCHITECTURE=8
 endif
+ifeq ($(CC_MACH),powerpc64)
+	ZT_ARCHITECTURE=8
+endif
 ifeq ($(CC_MACH),e2k)
 	ZT_ARCHITECTURE=2
 endif
@@ -177,22 +180,22 @@
 ifeq ($(CC_MACH),armv6)
 	ZT_ARCHITECTURE=3
 	override DEFS+=-DZT_NO_TYPE_PUNNING
-	ZT_USE_ARM32_NEON_ASM_CRYPTO=1
+	ZT_USE_ARM32_NEON_ASM_CRYPTO=0
 endif
 ifeq ($(CC_MACH),armv6l)
 	ZT_ARCHITECTURE=3
 	override DEFS+=-DZT_NO_TYPE_PUNNING
-	ZT_USE_ARM32_NEON_ASM_CRYPTO=1
+	ZT_USE_ARM32_NEON_ASM_CRYPTO=0
 endif
 ifeq ($(CC_MACH),armv6zk)
 	ZT_ARCHITECTURE=3
 	override DEFS+=-DZT_NO_TYPE_PUNNING
-	ZT_USE_ARM32_NEON_ASM_CRYPTO=1
+	ZT_USE_ARM32_NEON_ASM_CRYPTO=0
 endif
 ifeq ($(CC_MACH),armv6kz)
 	ZT_ARCHITECTURE=3
 	override DEFS+=-DZT_NO_TYPE_PUNNING
-	ZT_USE_ARM32_NEON_ASM_CRYPTO=1
+	ZT_USE_ARM32_NEON_ASM_CRYPTO=0
 endif
 ifeq ($(CC_MACH),armv7)
 	ZT_ARCHITECTURE=3
@@ -275,19 +278,6 @@
 	override INCLUDES+=-I/usr/pgsql-10/include -Iext/hiredis-0.14.1/include/ -Iext/redis-plus-plus-1.1.1/install/centos8/include/sw/
 endif
 
-# ARM32 hell -- use conservative CFLAGS
-ifeq ($(ZT_ARCHITECTURE),3)
-	ifeq ($(shell if [ -e /usr/bin/dpkg ]; then dpkg --print-architecture; fi),armel)
-		override CFLAGS+=-march=armv5t -mfloat-abi=soft -msoft-float -mno-unaligned-access -marm
-		override CXXFLAGS+=-march=armv5t -mfloat-abi=soft -msoft-float -mno-unaligned-access -marm
-		ZT_USE_ARM32_NEON_ASM_CRYPTO=0
-	else
-		override CFLAGS+=-march=armv5t -mno-unaligned-access -marm -fexceptions
-		override CXXFLAGS+=-march=armv5t -mno-unaligned-access -marm -fexceptions
-		ZT_USE_ARM32_NEON_ASM_CRYPTO=0
-	endif
-endif
-
 # Build faster crypto on some targets
 ifeq ($(ZT_USE_X64_ASM_SALSA),1)
 	override DEFS+=-DZT_USE_X64_ASM_SALSA2012
