--- Makefile.orig	2016-12-15 13:28:15.000000000 +0000
+++ Makefile	2016-12-15 13:29:23.893597661 +0000
@@ -3,13 +3,13 @@
 MANDIR=$(PREFIX)/share/man/man1
 
 CC=gcc
-CFLAGS=-std=c89 -O2 -pedantic -Wall -I"./include"
+ORG_CFLAGS=-std=c89 -O2 -pedantic -Wall -I"./include"
 MANFLAGS=-h -h -v -V
 
 HELP2MAN_VERSION := $(shell help2man --version 2>/dev/null)
 
 all:
-	$(CC) $(CFLAGS) -g -o light src/helpers.c src/light.c src/main.c
+	$(CC) $(CFLAGS) $(ORG_CFLAGS) $(LDFLAGS) -g -o light src/helpers.c src/light.c src/main.c
 exp:
 	$(CC) $(CFLAGS) -E  src/helpers.c src/light.c
 man:
