--- lldb-5.0.0.src/include/lldb/Host/Editline.h.orig
+++ lldb-5.0.0.src/include/lldb/Host/Editline.h
@@ -44,7 +44,7 @@
 // used in cases where this is true.  This is a compile time dependecy, for now
 // selected per target Platform
 #if defined(__APPLE__) || defined(__FreeBSD__) || defined(__NetBSD__) ||       \
-    defined(__OpenBSD__)
+    defined(__OpenBSD__) || defined(__linux__)
 #define LLDB_EDITLINE_USE_WCHAR 1
 #include <codecvt>
 #else
