--- a/mandrel-mandrel-21.1.0.0-Final/substratevm/src/com.oracle.svm.driver/src/com/oracle/svm/driver/NativeImage.java
+++ b/mandrel-mandrel-21.1.0.0-Final/substratevm/src/com.oracle.svm.driver/src/com/oracle/svm/driver/NativeImage.java
@@ -1829,6 +1829,9 @@
         if (Long.compareUnsigned(memMax, SubstrateOptionsParser.parseLong(maxXmx)) >= 0) {
             return maxXmx;
         }
+        if (memMax == 0) {
+            return "2g";
+        }
         return Long.toUnsignedString(memMax);
     }
 
