diff --git a/src/mx/_impl/mx.py b/src/mx/_impl/mx.py
index 60e14f6d..ce796dac 100755
--- a/mx/src/mx/_impl/mx.py
+++ b/mx/src/mx/_impl/mx.py
@@ -8444,6 +8444,10 @@ class TarExtractor(Extractor):
         return ar.getnames()
 
     def _extractall(self, ar, dst):
+        # Force the filter to 'fully_trusted' when available
+        if hasattr(tarfile, 'fully_trusted_filter'):
+            return ar.extractall(dst, filter=tarfile.fully_trusted_filter)
+
         return ar.extractall(dst)
