From d834971874bddc389ea7a6313b4d72f46b68be72 Mon Sep 17 00:00:00 2001
From: JiCiT <jason@infinitebubble.com>
Date: Wed, 1 Oct 2014 17:32:30 -0400
Subject: [PATCH 1/4] Accept LibreSSL in lieu of OpenSSL at link-time. (PR
 #209)

---
 src/module.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/module.c b/src/module.c
index 1b3ee59..ff97f6c 100644
--- src/module.c
+++ src/module.c
@@ -299,6 +299,8 @@ initpycurl(void)
         runtime_ssl_lib = "none/other";
     } else if (!strncmp(vi->ssl_version, "OpenSSL/", 8)) {
         runtime_ssl_lib = "openssl";
+    } else if (!strncmp(vi->ssl_version, "LibreSSL/", 9)) {
+        runtime_ssl_lib = "openssl";
     } else if (!strncmp(vi->ssl_version, "GnuTLS/", 7)) {
         runtime_ssl_lib = "gnutls";
     } else if (!strncmp(vi->ssl_version, "NSS/", 4)) {
-- 
2.1.2

