--- make/build_jconsole.sh	2019-12-17 18:25:11.713700768 +0900
+++ make/build_jconsole.sh	2019-12-17 18:26:45.090341029 +0900
@@ -5,6 +5,7 @@
 cd ~
 
 macmin="-mmacosx-version-min=10.6"
+ccver=$(${CC} --version)
 USE_LINENOISE="${USE_LINENOISE:=1}"
 
 if [ "x$CC" = x'' ] ; then
@@ -20,7 +21,7 @@
 export CC
 fi
 
-if [ $CC = "gcc" ] ; then
+if [ -z "${ccver##*(GCC)*}" ]; then
 # gcc
 common=" -fPIC -O1 -fwrapv -fno-strict-aliasing -Wextra -Wno-maybe-uninitialized -Wno-unused-parameter -Wno-sign-compare -Wno-clobbered -Wno-empty-body -Wno-unused-value -Wno-pointer-sign -Wno-parentheses"
 OVER_GCC_VER6=$(echo `$CC -dumpversion | cut -f1 -d.` \>= 6 | bc)
