From 8613c2f04b2343895d993558dab6c678df0f8750 Mon Sep 17 00:00:00 2001
From: Xiao-Long Chen <chenxiaolong@cxl.epac.to>
Date: Sat, 11 Oct 2014 18:50:22 -0400
Subject: [PATCH 2/8] Skip region panel

---
 panels/Makefile.am | 1 -
 1 file changed, 1 deletion(-)

diff --git a/panels/Makefile.am b/panels/Makefile.am
index 5ed02c6..5a67273 100644
--- a/panels/Makefile.am
+++ b/panels/Makefile.am
@@ -7,7 +7,6 @@ SUBDIRS= \
 	color \
 	display \
 	mouse \
-	region \
 	info \
 	sound \
 	keyboard \
-- 
2.8.2

