From be3be4c5320920a23d7e09fa08ef2a3cf066dda7 Mon Sep 17 00:00:00 2001
From: Xiao-Long Chen <chenxiaolong@cxl.epac.to>
Date: Sun, 8 Jun 2014 16:54:41 -0400
Subject: [PATCH 5/8] Allow gnome-control-center to run at the same time

---
 shell/control-center.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/shell/control-center.c b/shell/control-center.c
index de27113..a91eba1 100644
--- a/shell/control-center.c
+++ b/shell/control-center.c
@@ -308,7 +308,7 @@ main (int argc, char **argv)
   shell = gnome_control_center_new ();
 
   /* enforce single instance of this application */
-  application = gtk_application_new ("org.gnome.ControlCenter", G_APPLICATION_HANDLES_COMMAND_LINE);
+  application = gtk_application_new ("org.gnome.ControlCenterUnity", G_APPLICATION_HANDLES_COMMAND_LINE);
   g_signal_connect (application, "startup",
                     G_CALLBACK (application_startup_cb), shell);
   g_signal_connect (application, "command-line",
-- 
2.8.2

