From f76eb169510d94f028001423f759ed4cd142fe59 Mon Sep 17 00:00:00 2001
From: Xiao-Long Chen <chenxiaolong@cxl.epac.to>
Date: Mon, 9 Jun 2014 15:32:07 -0400
Subject: [PATCH 6/8] Remove overlay-scrollbar from GTK_MODULES

---
 shell/control-center.c | 23 +++++++++++++++++++++++
 1 file changed, 23 insertions(+)

diff --git a/shell/control-center.c b/shell/control-center.c
index a91eba1..f484351 100644
--- a/shell/control-center.c
+++ b/shell/control-center.c
@@ -281,9 +281,32 @@ application_startup_cb (GApplication       *application,
    */
 }
 
+void
+remove_substring(char *src, const char *remove)
+{
+  while (src = strstr (src, remove)) {
+    memmove (src, src + strlen (remove), strlen (src + strlen(remove)) + 1);
+  }
+}
+
+void
+remove_overlay_scrollbars ()
+{
+  char *temp = getenv ("GTK_MODULES");
+  if (temp == NULL) {
+    return;
+  }
+
+  char *modules = strdup (temp);
+  remove_substring (modules, "overlay-scrollbar");
+  setenv ("GTK_MODULES", modules, 1);
+  free (modules);
+}
+
 int
 main (int argc, char **argv)
 {
+  remove_overlay_scrollbars ();
   GnomeControlCenter *shell;
   GtkApplication *application;
   int status;
-- 
2.8.2

