diff -p1ru a/panels/screen/cc-screen-panel.c b/panels/screen/cc-screen-panel.c
--- a/panels/screen/cc-screen-panel.c	2023-02-21 02:17:29.000000000 +0800
+++ b/panels/screen/cc-screen-panel.c	2024-08-06 07:49:35.315105356 +0800
@@ -541,9 +541,2 @@ cc_screen_panel_init (CcScreenPanel *sel
 
-  /* bind the screen lock suspend checkbutton */
-  widget = WID ("screen_lock_suspend_checkbutton");
-  g_settings_bind (self->priv->lock_settings,
-                   "ubuntu-lock-on-suspend",
-                   widget, "active",
-                   G_SETTINGS_BIND_DEFAULT);
-
   widget = WID ("screen_vbox");
diff -p1ru a/panels/screen/screen.ui b/panels/screen/screen.ui
--- a/panels/screen/screen.ui	2023-02-21 02:17:29.000000000 +0800
+++ b/panels/screen/screen.ui	2024-08-06 07:51:13.397039849 +0800
@@ -300,27 +300,2 @@
                         <child>
-                          <object class="GtkHBox" id="screen_suspend_hbox">
-                            <property name="visible">True</property>
-                            <property name="can_focus">False</property>
-                            <property name="spacing">6</property>
-                            <child>
-                              <object class="GtkCheckButton" id="screen_lock_suspend_checkbutton">
-                                <property name="label" translatable="yes">Require my password when waking from suspend</property>
-                                <property name="use_action_appearance">False</property>
-                                <property name="visible">True</property>
-                                <property name="can_focus">True</property>
-                              </object>
-                              <packing>
-                                <property name="expand">False</property>
-                                <property name="fill">False</property>
-                                <property name="position">0</property>
-                              </packing>
-                            </child>
-                          </object>
-                          <packing>
-                            <property name="expand">False</property>
-                            <property name="fill">False</property>
-                            <property name="position">2</property>
-                          </packing>
-                        </child>
-                        <child>
                           <object class="GtkHBox" id="hbox2">
