- remove libpthread stubs
- detect static_assert() correctly for musl

--- configure.ac.orig	2016-11-01 14:39:59.000000000 +0100
+++ configure.ac	2016-11-04 13:07:05.317813259 +0100
@@ -828,7 +828,7 @@
 
 dnl pthread-stubs is mandatory on targets where it exists
 case "$host_os" in
-cygwin* )
+cygwin* | linux* )
     pthread_stubs_possible="no"
     ;;
 * )
@@ -1919,6 +1919,11 @@
 fi
 AM_CONDITIONAL(HAVE_ST_NINE, test "x$enable_nine" = xyes)
 
+AC_CHECK_DECL(static_assert, [],
+	[AC_DEFINE(static_assert(...), [/* */], [noop static_assert() replacement]),
+	AC_MSG_RESULT([no])],
+	[[#include <assert.h>]])
+
 dnl
 dnl OpenCL configuration
 dnl
