From 0d53c1ed54e58b6db1fb30440d8f0f0ff19efa8b Mon Sep 17 00:00:00 2001
From: Daeho Ro <lamanus@techready.ai>
Date: Sat, 11 Jul 2026 22:19:08 +0900
Subject: [PATCH] Build with OpenCV 5

---
 ChangeLog   |  1 +
 meson.build | 14 +++++++++-----
 2 files changed, 10 insertions(+), 5 deletions(-)

diff --git a/meson.build b/meson.build
index 8dc7f1c1b..c47f322a4 100644
--- a/meson.build
+++ b/meson.build
@@ -282,11 +282,16 @@
 endif
 cairo_dep = dependency('cairo', required : true)
 kplot_dep = dependency('kplot', fallback : ['kplot', 'kplot_dep'], required : true)
-opencv4_dep = dependency('opencv4', required : true, version: '>= 4.2.0')
+opencv_dep = dependency('opencv5', version : '>= 5.0.0', required : false)
+if not opencv_dep.found()
+  opencv_dep = dependency('opencv4', version : '>= 4.2.0', required : true)
+endif
 # checking required opencv modules
-opencv_symbols = ['HAVE_OPENCV_CORE', 'HAVE_OPENCV_IMGPROC', 'HAVE_OPENCV_CALIB3D', 'HAVE_OPENCV_STITCHING']
+opencv_calib_module = opencv_dep.version().version_compare('>= 5.0.0') ? 'HAVE_OPENCV_GEOMETRY' : 'HAVE_OPENCV_CALIB3D'
+opencv_symbols = ['HAVE_OPENCV_CORE', 'HAVE_OPENCV_IMGPROC', opencv_calib_module, 'HAVE_OPENCV_STITCHING']
+opencv_incdir = '-I'+opencv_dep.get_variable(pkgconfig : 'includedir')
 foreach s : opencv_symbols
-  cpp.has_header_symbol('opencv2/opencv_modules.hpp', s, dependencies: opencv4_dep, required : true)
+  cpp.has_header_symbol('opencv2/opencv_modules.hpp', s, args: [opencv_incdir], required : true)
 endforeach

 rtprocess_dep = dependency('rtprocess', required : false)
@@ -564,7 +569,7 @@
 endforeach

 ## Dependencies configuration
-if opencv4_dep.version().version_compare('>=4.4.0')
+if opencv_dep.version().version_compare('>=4.4.0')
   conf_data.set('HAVE_CV44', 1, description : 'Using OpenCV 4.4 and all registration features.')
 else
   warning('OpenCV 4.4 was not found, using a lower version that does not support translation-only registration instead.')
@@ -688,7 +693,7 @@
     libcurl_dep,
     libgit2_dep,
     gsl_dep,
-    opencv4_dep,
+    opencv_dep,
     rtprocess_dep,
     htmesh_dep,
     healpix_dep,
