From 829ba7f6807f6da409fe266e8d027654ae1c179b Mon Sep 17 00:00:00 2001
From: crvi <crvisqr@gmail.com>
Date: Thu, 20 Feb 2020 18:36:23 +0530
Subject: [PATCH 1/2] update app icon name

Fixes: #1775
---
 lib/rb-stock-icons.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git lib/rb-stock-icons.c lib/rb-stock-icons.c
index cf52c7a38..cbc221a7d 100644
--- a/lib/rb-stock-icons.c
+++ b/lib/rb-stock-icons.c
@@ -37,7 +37,7 @@
 #include "rb-file-helpers.h"
 #include "rb-stock-icons.h"
 
-const char RB_APP_ICON[] = "rhythmbox";
+const char RB_APP_ICON[] = "org.gnome.Rhythmbox";
 const char RB_STOCK_SET_STAR[] = "rhythmbox-set-star";
 const char RB_STOCK_UNSET_STAR[] = "rhythmbox-unset-star";
 const char RB_STOCK_NO_STAR[] = "rhythmbox-no-star";
-- 
2.24.1


From b2c73c7efe28eefb89388bfe3dd0bc55a91a0bf4 Mon Sep 17 00:00:00 2001
From: crvi <crvisqr@gmail.com>
Date: Thu, 20 Feb 2020 18:39:23 +0530
Subject: [PATCH 2/2] use RB_APP_ICON instead of hardcoding icon names

---
 shell/main.c           | 3 ++-
 shell/rb-application.c | 2 +-
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git shell/main.c shell/main.c
index 5060f6545..7170f29d5 100644
--- a/shell/main.c
+++ b/shell/main.c
@@ -45,6 +45,7 @@
 #include "rb-util.h"
 #include "rb-debug.h"
 #include "rb-application.h"
+#include "rb-stock-icons.h"
 
 int
 main (int argc, char **argv)
@@ -76,7 +77,7 @@ main (int argc, char **argv)
 #endif
 
 	g_set_application_name (_("Rhythmbox"));
-	gtk_window_set_default_icon_name ("rhythmbox");
+	gtk_window_set_default_icon_name (RB_APP_ICON);
 
 #if defined(USE_UNINSTALLED_DIRS)
 	g_irepository_prepend_search_path (SHARE_UNINSTALLED_BUILDDIR "/../bindings/gi");
diff --git shell/rb-application.c shell/rb-application.c
index 37afb36fb..8492bb03b 100644
--- a/shell/rb-application.c
+++ b/shell/rb-application.c
@@ -205,7 +205,7 @@ about_action_cb (GSimpleAction *action, GVariant *parameters, gpointer user_data
 			       "authors", (const char **) authors,
 			       "documenters", (const char **) documenters,
 			       "translator-credits", strcmp (translator_credits, "translator-credits") != 0 ? translator_credits : NULL,
-			       "logo-icon-name", "rhythmbox",
+			       "logo-icon-name", RB_APP_ICON,
 			       NULL);
 	g_string_free (comment, TRUE);
 	g_free (license_trans);
-- 
2.24.1

