--- a/src/3rdparty/btparse/lex_auxiliary.c	2018-12-19 01:27:45.874263697 +0000
+++ b/src/3rdparty/btparse/lex_auxiliary.c	2018-12-19 01:27:45.874263697 +0000
@@ -32,6 +32,7 @@
 #include "bt_debug.h"
 #include <stdlib.h>
 #include <string.h>
+#include <strings.h>
 #include <ctype.h>
 #include <stdarg.h>
 #include <assert.h>
--- a/src/3rdparty/btparse/sym.c	2018-12-19 02:27:45.878263787 +0100
+++ b/src/3rdparty/btparse/sym.c	2018-12-19 02:27:45.878263787 +0100
@@ -110,6 +110,7 @@
 /*#include "bt_config.h"*/
 #include <stdio.h>
 #include <string.h>
+#include <strings.h>
 #include <stdlib.h>
 #ifdef MEMCHK
 #include "trax.h"
