From c27d10fcd8932c3d83fe73b07412969cc42ff686 Mon Sep 17 00:00:00 2001
From: Martin Donlon <github-martin@donlons.com>
Date: Sun, 3 Nov 2019 19:29:06 -0800
Subject: [PATCH] impl AsPyRefDispatch for every pyclass

---
 pyo3-derive-backend/src/pyclass.rs | 3 +--

diff --git a/pyo3-derive-backend/src/pyclass.rs b/pyo3-derive-backend/src/pyclass.rs
index 59a0ab29f..ea6c0f513 100644
--- a/pyo3-derive-backend/src/pyclass.rs
+++ b/pyo3-derive-backend/src/pyclass.rs
@@ -422,8 +422,7 @@ fn impl_class(
             }
         }
 
-        //impl<T: pyo3::type_object::PyTypeInfo> pyo3::AsPyRefDispatch<T> for #cls {}
-        //impl pyo3::ManagedPyRefDispatch for #cls {}
+        impl pyo3::AsPyRefDispatch<#cls> for Py<#cls> {}
 
         #inventory_impl
 
