--- a/build.rs	2020-06-01 19:27:44.819141296 +1000
+++ b/build.rs	2020-06-01 19:29:51.770147574 +1000
@@ -539,14 +539,6 @@
             MIN_VERSION, actual_version
         )
     }
-
-    let actual_date = Date::read().expect("Failed to determine the rustc date");
-    if !actual_date.at_least(MIN_DATE) {
-        panic!(
-            "pyo3 requires at least rustc {}, while the current rustc date is {}",
-            MIN_DATE, actual_date
-        )
-    }
 }
 
 fn main() -> Result<(), String> {
