--- ghidra-Ghidra_9.0.4_build/build.gradle	2019-08-01 14:38:35.228473377 -0500
+++ ghidra-Ghidra_9.0.4_build/build.gradle	2019-08-01 14:38:16.382472910 -0500
@@ -6,7 +6,7 @@
 /***************************************************************************************
  * Make sure the correct version of gradle is being used
  ***************************************************************************************/
-if (gradle.gradleVersion != "5.0") {
+if (!gradle.gradleVersion.matches("5\\.(.*)")) {
 	throw new GradleException("Requires Gradle 5.0, but was run with $gradle.gradleVersion")
 }
 
