--- ghidra-10.0.1/ghidra-Ghidra_10.0.1_build/Ghidra/Debug/Debugger-gadp/build.gradle	2021-07-08 12:16:19.000000000 -0500
+++ -	2021-07-19 04:53:54.446268239 -0500
@@ -27,7 +27,6 @@
 
 configurations {
 	allProtocArtifacts
-	protocArtifact
 }
 
 def os = org.gradle.internal.os.OperatingSystem.current()
@@ -37,16 +36,6 @@
     allProtocArtifacts 'com.google.protobuf:protoc:3.11.1:linux-x86_64@exe'
     allProtocArtifacts 'com.google.protobuf:protoc:3.11.1:osx-x86_64@exe'
 
-	if (os.isWindows()) {
-    	protocArtifact 'com.google.protobuf:protoc:3.11.1:windows-x86_64@exe'
-	}
-	if (os.isLinux()) {
-    	protocArtifact 'com.google.protobuf:protoc:3.11.1:linux-x86_64@exe'
-	}
-	if (os.isMacOsX()) {
-    	protocArtifact 'com.google.protobuf:protoc:3.11.1:osx-x86_64@exe'
-    }
-
 	api 'com.google.protobuf:protobuf-java:3.11.1'
 	api project(':Framework-AsyncComm')
 	api project(':Framework-Debugging')
@@ -70,14 +59,9 @@
 	ext.outdir = file("build/generated/source/proto/main/java")
 	outputs.dir(outdir)
 	inputs.files(src)
-	dependsOn(configurations.protocArtifact)
 	doLast {
-		def exe = configurations.protocArtifact.first()
-		if (!os.isWindows()) {
-			exe.setExecutable(true)
-		}
 		exec {
-			commandLine exe, "--java_out=$outdir", "-I$srcdir"
+			commandLine '/usr/bin/protoc', "--java_out=$outdir", "-I$srcdir"
 			args src
 		}
 	}
