--- ghidra-10.1/ghidra_build/Ghidra/Debug/Debugger-gadp/build.gradle	2021-12-10 19:03:38.000000000 +0100
+++ -	2021-12-11 13:01:51.089861724 +0100
@@ -33,33 +33,33 @@
 def platform = getCurrentPlatformName()
 
 dependencies {
-	allProtocArtifacts 'com.google.protobuf:protoc:3.17.3:windows-x86_64@exe'
-	allProtocArtifacts 'com.google.protobuf:protoc:3.17.3:linux-x86_64@exe'
-	allProtocArtifacts 'com.google.protobuf:protoc:3.17.3:linux-aarch_64@exe'
-	allProtocArtifacts 'com.google.protobuf:protoc:3.17.3:osx-x86_64@exe'
-	allProtocArtifacts 'com.google.protobuf:protoc:3.17.3:osx-aarch_64@exe'
+	allProtocArtifacts 'com.google.protobuf:protoc:3.18.1:windows-x86_64@exe'
+	allProtocArtifacts 'com.google.protobuf:protoc:3.18.1:linux-x86_64@exe'
+	allProtocArtifacts 'com.google.protobuf:protoc:3.18.1:linux-aarch_64@exe'
+	allProtocArtifacts 'com.google.protobuf:protoc:3.18.1:osx-x86_64@exe'
+	allProtocArtifacts 'com.google.protobuf:protoc:3.18.1:osx-aarch_64@exe'
 
 	if (isCurrentWindows()) {
-		protocArtifact 'com.google.protobuf:protoc:3.17.3:windows-x86_64@exe'
+		protocArtifact 'com.google.protobuf:protoc:3.18.1:windows-x86_64@exe'
 	}
 	if (isCurrentLinux()) {
 		if (platform.endsWith("x86_64")) {
-			protocArtifact 'com.google.protobuf:protoc:3.17.3:linux-x86_64@exe'
+			protocArtifact 'com.google.protobuf:protoc:3.18.1:linux-x86_64@exe'
 		}
 		else {
-			protocArtifact 'com.google.protobuf:protoc:3.17.3:linux-aarch_64@exe'
+			protocArtifact 'com.google.protobuf:protoc:3.18.1:linux-aarch_64@exe'
 		}
 	}
 	if (isCurrentMac()) {
 		if (platform.endsWith("x86_64")) {
-			protocArtifact 'com.google.protobuf:protoc:3.17.3:osx-x86_64@exe'
+			protocArtifact 'com.google.protobuf:protoc:3.18.1:osx-x86_64@exe'
 		}
 		else {
-			protocArtifact 'com.google.protobuf:protoc:3.17.3:osx-aarch_64@exe'
+			protocArtifact 'com.google.protobuf:protoc:3.18.1:osx-aarch_64@exe'
 		}
 	}
 
-	api 'com.google.protobuf:protobuf-java:3.17.3'
+	api 'com.google.protobuf:protobuf-java:3.18.1'
 	api project(':Framework-AsyncComm')
 	api project(':Framework-Debugging')
 	api project(':ProposedUtils')
@@ -70,7 +70,7 @@
 
 /*protobuf {
 	protoc {
-		artifact = 'com.google.protobuf:protoc:3.17.3'
+		artifact = 'com.google.protobuf:protoc:3.18.1'
 	}
 }*/
 
@@ -82,14 +82,9 @@
 	ext.outdir = file("build/generated/source/proto/main/java")
 	outputs.dir(outdir)
 	inputs.files(src)
-	dependsOn(configurations.protocArtifact)
 	doLast {
-		def exe = configurations.protocArtifact.first()
-		if (!isCurrentWindows()) {
-			exe.setExecutable(true)
-		}
 		exec {
-			commandLine exe, "--java_out=$outdir", "-I$srcdir"
+			commandLine '/usr/bin/protoc', "--java_out=$outdir", "-I$srcdir"
 			args src
 		}
 	}
--- ghidra-10.1/ghidra_build/Ghidra/Debug/Debugger-gadp/Module.manifest	2021-12-10 19:03:38.000000000 +0100
+++ -	2021-12-11 13:02:09.420069640 +0100
@@ -1 +1 @@
-MODULE FILE LICENSE: lib/protobuf-java-3.17.3.jar BSD
+MODULE FILE LICENSE: lib/protobuf-java-3.18.1.jar BSD
