--- a/ghidra_build/Ghidra/Debug/Debugger-gadp/build.gradle	2021-09-28 20:47:24.000000000 +0200
+++ -	2021-10-21 16:35:42.275129993 +0200
@@ -33,21 +33,11 @@
 def os = org.gradle.internal.os.OperatingSystem.current()
 
 dependencies {
-	allProtocArtifacts 'com.google.protobuf:protoc:3.11.1:windows-x86_64@exe'
-    allProtocArtifacts 'com.google.protobuf:protoc:3.11.1:linux-x86_64@exe'
-    allProtocArtifacts 'com.google.protobuf:protoc:3.11.1:osx-x86_64@exe'
+	allProtocArtifacts 'com.google.protobuf:protoc:3.18.1:windows-x86_64@exe'
+    allProtocArtifacts 'com.google.protobuf:protoc:3.18.1:linux-x86_64@exe'
+    allProtocArtifacts 'com.google.protobuf:protoc:3.18.1:osx-x86_64@exe'
 
-	if (os.isWindows()) {
-    	protocArtifact 'com.google.protobuf:protoc:3.11.1:windows-x86_64@exe'
-	}
-	if (os.isLinux()) {
-    	protocArtifact 'com.google.protobuf:protoc:3.11.1:linux-x86_64@exe'
-	}
-	if (os.isMacOsX()) {
-    	protocArtifact 'com.google.protobuf:protoc:3.11.1:osx-x86_64@exe'
-    }
-
-	api 'com.google.protobuf:protobuf-java:3.11.1'
+	api 'com.google.protobuf:protobuf-java:3.18.1'
 	api project(':Framework-AsyncComm')
 	api project(':Framework-Debugging')
 	api project(':ProposedUtils')
@@ -58,7 +48,7 @@
 
 /*protobuf {
 	protoc {
-		artifact = 'com.google.protobuf:protoc:3.11.1'
+		artifact = 'com.google.protobuf:protoc:3.18.1'
 	}
 }*/
 
@@ -72,12 +62,7 @@
 	inputs.files(src)
-	dependsOn(configurations.protocArtifact)
 	doLast {
-		def exe = configurations.protocArtifact.first()
-		if (!os.isWindows()) {
-			exe.setExecutable(true)
-		}
 		exec {
-			commandLine exe, "--java_out=$outdir", "-I$srcdir"
+			commandLine '/usr/bin/protoc', "--java_out=$outdir", "-I$srcdir"
 			args src
 		}
 	}
--- a/ghidra_build/Ghidra/Debug/Debugger-gadp/Module.manifest	2021-09-28 20:47:24.000000000 +0200
+++ -	2021-10-21 16:40:18.629990242 +0200
@@ -1 +1 @@
-MODULE FILE LICENSE: lib/protobuf-java-3.11.1.jar BSD
+MODULE FILE LICENSE: lib/protobuf-java-3.18.1.jar BSD
