--- ghidra-10.3/ghidra_build/Ghidra/Debug/Debugger-gadp/build.gradle.orig
+++ ghidra-10.3/ghidra_build/Ghidra/Debug/Debugger-gadp/build.gradle
@@ -81,14 +81,9 @@
 	ext.outdir = file("build/generated/source/proto/main/java")
 	outputs.dir(outdir)
 	inputs.files(src)
-	dependsOn(configurations.protocArtifact)
 	doLast {
-		def exe = configurations.protocArtifact.first()
-		if (!isCurrentWindows()) {
-			exe.setExecutable(true)
-		}
 		exec {
-			commandLine exe, "--java_out=$outdir", "-I$srcdir"
+			commandLine '/usr/bin/protoc', "--java_out=$outdir", "-I$srcdir"
 			args src
 		}
 	}
--- ghidra-10.3/ghidra_build/Ghidra/Debug/Debugger-isf/build.gradle.orig
+++ ghidra-10.3/ghidra_build/Ghidra/Debug/Debugger-isf/build.gradle
@@ -72,14 +72,9 @@
 	ext.outdir = file("build/generated/source/proto/main/java")
 	outputs.dir(outdir)
 	inputs.files(src)
-	dependsOn(configurations.protocArtifact)
 	doLast {
-		def exe = configurations.protocArtifact.first()
-		if (!isCurrentWindows()) {
-			exe.setExecutable(true)
-		}
 		exec {
-			commandLine exe, "--java_out=$outdir", "-I$srcdir"
+			commandLine '/usr/bin/protoc', "--java_out=$outdir", "-I$srcdir"
 			args src
 		}
 	}
