--- a/ghidra_build/Ghidra/Debug/ProposedUtils/Module.manifest
+++ b/ghidra_build/Ghidra/Debug/ProposedUtils/Module.manifest
@@ -1 +1 @@
-MODULE FILE LICENSE: lib/protobuf-java-3.21.8.jar BSD-3-GOOGLE
+MODULE FILE LICENSE: lib/protobuf-java-3.25.5.jar BSD-3-GOOGLE
--- a/ghidra_build/Ghidra/Extensions/MachineLearning/Module.manifest
+++ b/ghidra_build/Ghidra/Extensions/MachineLearning/Module.manifest
@@ -1,6 +1,6 @@
 MODULE FILE LICENSE: lib/olcut-config-protobuf-5.2.0.jar BSD-2-ORACLE
 MODULE FILE LICENSE: lib/olcut-core-5.2.0.jar BSD-2-ORACLE
-MODULE FILE LICENSE: lib/protobuf-java-3.21.8.jar BSD-3-GOOGLE
+MODULE FILE LICENSE: lib/protobuf-java-3.25.5.jar BSD-3-GOOGLE
 MODULE FILE LICENSE: lib/tribuo-classification-core-4.2.0.jar Apache License 2.0
 MODULE FILE LICENSE: lib/tribuo-classification-tree-4.2.0.jar Apache License 2.0
 MODULE FILE LICENSE: lib/tribuo-common-tree-4.2.0.jar Apache License 2.0
--- a/ghidra_build/gradle/hasProtobuf.gradle
+++ b/ghidra_build/gradle/hasProtobuf.gradle
@@ -67,7 +67,7 @@
 		if (!isCurrentWindows()) {
 			exe.setExecutable(true)
 		}
-		generateProto.commandLine exe, "--java_out=${generateProto.outdir}", "-I${generateProto.srcdir}"
+		generateProto.commandLine "/usr/bin/protoc", "--java_out=${generateProto.outdir}", "-I${generateProto.srcdir}"
 		generateProto.args generateProto.src
 	}
 }
--- a/ghidra_build/gradle/hasProtobuf.gradle
+++ b/ghidra_build/gradle/hasProtobuf.gradle
@@ -26,36 +26,36 @@
 
 
 dependencies {
-	allProtocArtifacts 'com.google.protobuf:protoc:3.21.8:windows-x86_64@exe'
-	allProtocArtifacts 'com.google.protobuf:protoc:3.21.8:linux-x86_64@exe'
-	allProtocArtifacts 'com.google.protobuf:protoc:3.21.8:linux-aarch_64@exe'
-	allProtocArtifacts 'com.google.protobuf:protoc:3.21.8:osx-x86_64@exe'
-	allProtocArtifacts 'com.google.protobuf:protoc:3.21.8:osx-aarch_64@exe'
+	allProtocArtifacts 'com.google.protobuf:protoc:3.25.5:windows-x86_64@exe'
+	allProtocArtifacts 'com.google.protobuf:protoc:3.25.5:linux-x86_64@exe'
+	allProtocArtifacts 'com.google.protobuf:protoc:3.25.5:linux-aarch_64@exe'
+	allProtocArtifacts 'com.google.protobuf:protoc:3.25.5:osx-x86_64@exe'
+	allProtocArtifacts 'com.google.protobuf:protoc:3.25.5:osx-aarch_64@exe'
 
 	if (isCurrentWindows()) {
-		protocArtifact 'com.google.protobuf:protoc:3.21.8:windows-x86_64@exe'
+		protocArtifact 'com.google.protobuf:protoc:3.25.5:windows-x86_64@exe'
 	}
 	if (isCurrentLinux()) {
 		if (platform.endsWith("x86_64")) {
-			protocArtifact 'com.google.protobuf:protoc:3.21.8:linux-x86_64@exe'
+			protocArtifact 'com.google.protobuf:protoc:3.25.5:linux-x86_64@exe'
 		}
 		else {
-			protocArtifact 'com.google.protobuf:protoc:3.21.8:linux-aarch_64@exe'
+			protocArtifact 'com.google.protobuf:protoc:3.25.5:linux-aarch_64@exe'
 		}
 	}
 	if (isCurrentMac()) {
 		if (platform.endsWith("x86_64")) {
-			protocArtifact 'com.google.protobuf:protoc:3.21.8:osx-x86_64@exe'
+			protocArtifact 'com.google.protobuf:protoc:3.25.5:osx-x86_64@exe'
 		}
 		else {
-			protocArtifact 'com.google.protobuf:protoc:3.21.8:osx-aarch_64@exe'
+			protocArtifact 'com.google.protobuf:protoc:3.25.5:osx-aarch_64@exe'
 		}
 	}
 }
 
 /*protobuf {
 	protoc {
-		artifact = 'com.google.protobuf:protoc:3.21.8'
+		artifact = 'com.google.protobuf:protoc:3.25.5'
 	}
 }*/
 
--- a/ghidra_build/Ghidra/Debug/ProposedUtils/build.gradle
+++ b/ghidra_build/Ghidra/Debug/ProposedUtils/build.gradle
@@ -24,7 +24,7 @@
 // val autoServiceVersion = "1.0-rc5"

 dependencies {
-	api 'com.google.protobuf:protobuf-java:3.21.8'
+	api 'com.google.protobuf:protobuf-java:3.25.5'
 	api project(':DB')
 	api project(':Project')
 	api project(':SoftwareModeling')
--- a/ghidra_build/Ghidra/Extensions/MachineLearning/build.gradle
+++ b/ghidra_build/Ghidra/Extensions/MachineLearning/build.gradle
@@ -26,7 +26,7 @@
 
     api "com.oracle.labs.olcut:olcut-config-protobuf:5.2.0" //{exclude group: "com.google.protobuf", module: "protobuf-java"}
     api ("com.oracle.labs.olcut:olcut-core:5.2.0") {exclude group: "org.jline"}
-    api "com.google.protobuf:protobuf-java:3.21.8"   //only needed for running junits
+    api "com.google.protobuf:protobuf-java:3.25.5"   //only needed for running junits
     api "org.tribuo:tribuo-classification-core:4.2.0"
     api "org.tribuo:tribuo-classification-tree:4.2.0"
     api "org.tribuo:tribuo-common-tree:4.2.0"
