--- ghidra-10.4/ghidra_build/Ghidra/Debug/Framework-AsyncComm/Module.manifest
+++ ghidra-10.4/ghidra_build/Ghidra/Debug/Framework-AsyncComm/Module.manifest
@@ -1 +1 @@
-MODULE FILE LICENSE: lib/protobuf-java-3.21.8.jar BSD-3-GOOGLE
+MODULE FILE LICENSE: lib/protobuf-java-3.24.3.jar BSD-3-GOOGLE
--- ghidra-10.4/ghidra_build/Ghidra/Debug/Framework-AsyncComm/build.gradle
+++ ghidra-10.4/ghidra_build/Ghidra/Debug/Framework-AsyncComm/build.gradle
@@ -22,7 +22,7 @@
 eclipse.project.name = 'Debug Framework-AsyncComm'
 
 dependencies {
-	api 'com.google.protobuf:protobuf-java:3.21.8'
+	api 'com.google.protobuf:protobuf-java:3.24.3'
 	api project(':Generic')
 	api project(':Graph')
 	api project(':ProposedUtils')
--- ghidra-10.4/ghidra_build/gradle/debugger/hasProtobuf.gradle
+++ ghidra-10.4/ghidra_build/gradle/debugger/hasProtobuf.gradle
@@ -26,36 +26,36 @@
 
 
 dependencies {
-	allProtocArtifacts 'com.google.protobuf:protoc:3.21.8:windows-x86_64@exe'
-	allProtocArtifacts 'com.google.protobuf:protoc:3.21.8:linux-x86_64@exe'
-	allProtocArtifacts 'com.google.protobuf:protoc:3.21.8:linux-aarch_64@exe'
-	allProtocArtifacts 'com.google.protobuf:protoc:3.21.8:osx-x86_64@exe'
-	allProtocArtifacts 'com.google.protobuf:protoc:3.21.8:osx-aarch_64@exe'
+	allProtocArtifacts 'com.google.protobuf:protoc:3.24.3:windows-x86_64@exe'
+	allProtocArtifacts 'com.google.protobuf:protoc:3.24.3:linux-x86_64@exe'
+	allProtocArtifacts 'com.google.protobuf:protoc:3.24.3:linux-aarch_64@exe'
+	allProtocArtifacts 'com.google.protobuf:protoc:3.24.3:osx-x86_64@exe'
+	allProtocArtifacts 'com.google.protobuf:protoc:3.24.3:osx-aarch_64@exe'
 
 	if (isCurrentWindows()) {
-		protocArtifact 'com.google.protobuf:protoc:3.21.8:windows-x86_64@exe'
+		protocArtifact 'com.google.protobuf:protoc:3.24.3:windows-x86_64@exe'
 	}
 	if (isCurrentLinux()) {
 		if (platform.endsWith("x86_64")) {
-			protocArtifact 'com.google.protobuf:protoc:3.21.8:linux-x86_64@exe'
+			protocArtifact 'com.google.protobuf:protoc:3.24.3:linux-x86_64@exe'
 		}
 		else {
-			protocArtifact 'com.google.protobuf:protoc:3.21.8:linux-aarch_64@exe'
+			protocArtifact 'com.google.protobuf:protoc:3.24.3:linux-aarch_64@exe'
 		}
 	}
 	if (isCurrentMac()) {
 		if (platform.endsWith("x86_64")) {
-			protocArtifact 'com.google.protobuf:protoc:3.21.8:osx-x86_64@exe'
+			protocArtifact 'com.google.protobuf:protoc:3.24.3:osx-x86_64@exe'
 		}
 		else {
-			protocArtifact 'com.google.protobuf:protoc:3.21.8:osx-aarch_64@exe'
+			protocArtifact 'com.google.protobuf:protoc:3.24.3:osx-aarch_64@exe'
 		}
 	}
 }
 
 /*protobuf {
 	protoc {
-		artifact = 'com.google.protobuf:protoc:3.21.8'
+		artifact = 'com.google.protobuf:protoc:3.24.3'
 	}
 }*/
 
@@ -74,7 +74,7 @@
 			exe.setExecutable(true)
 		}
 		exec {
-			commandLine exe, "--java_out=$outdir", "-I$srcdir"
+			commandLine '/usr/bin/protoc', "--java_out=$outdir", "-I$srcdir"
 			args src
 		}
 	}
