*** ghidra-Ghidra_9.0.4_build/Ghidra/RuntimeScripts/Linux/ghidraRun	2019-05-16 14:36:27.000000000 -0500
--- ghidra-Ghidra_9.0.4_build/Ghidra/RuntimeScripts/Linux/ghidraRun.new	2019-08-29 12:24:32.300079737 -0500
***************
*** 15,19 ****
--- 15,24 ----
  SCRIPT_FILE="$(readlink -f "$0" 2>/dev/null || readlink "$0" 2>/dev/null || echo "$0")"
  SCRIPT_DIR="${SCRIPT_FILE%/*}"
  
+ ## Make Ghidra's UI work on wayland
+ if [ "$XDG_SESSION_TYPE" = "wayland" ]; then
+     export _JAVA_AWT_WM_NONREPARENTING=1
+ fi
+ 
  # Launch Ghidra
  "${SCRIPT_DIR}"/support/launch.sh bg Ghidra "${MAXMEM}" "" ghidra.GhidraRun "$@"
