diff -p1ru a/95dbus_update-activation-env b/95dbus_update-activation-env
--- a/95dbus_update-activation-env	2023-12-05 23:36:45.000000000 +0800
+++ b/95dbus_update-activation-env	2024-07-19 18:27:30.809000000 +0800
@@ -13,6 +13,6 @@ if [ -n "$DBUS_SESSION_BUS_ADDRESS" ] &&
 
-    # tell dbus-daemon --session (and systemd --user, if running)
+    # tell dbus-daemon --session
     # to put the Xsession's environment in activated services'
     # environments
-    dbus-update-activation-environment --verbose --systemd --all
+    dbus-update-activation-environment --verbose --all
   )
