$OpenBSD: patch-src_openssl_stream_h,v 1.8 2018/03/19 18:28:12 jasper Exp $

Index: src/openssl_stream.h
--- src/openssl_stream.h.orig
+++ src/openssl_stream.h
@@ -27,7 +27,8 @@ extern int git_openssl_stream_new(git_stream **out, co
 
 
 
-# if OPENSSL_VERSION_NUMBER < 0x10100000L || defined(LIBRESSL_VERSION_NUMBER)
+# if OPENSSL_VERSION_NUMBER < 0x10100000L || \
+    (defined(LIBRESSL_VERSION_NUMBER) && LIBRESSL_VERSION_NUMBER < 0x2070000fL)
 
 GIT_INLINE(BIO_METHOD*) BIO_meth_new(int type, const char *name)
 {
