--- crypto/wvx509.cc.orig	2019-05-16 20:25:58.171570222 +0200
+++ crypto/wvx509.cc	2019-05-16 20:26:15.784410788 +0200
@@ -974,7 +974,7 @@
     sk_ACCESS_DESCRIPTION_push(ainfo, acc);
     acc->method = OBJ_txt2obj(type.cstr(), 0);
     acc->location->type = GEN_URI;
-    acc->location->d.ia5 = M_ASN1_IA5STRING_new();
+    acc->location->d.ia5 = ASN1_IA5STRING_new();
     unsigned char *cident 
 	= reinterpret_cast<unsigned char *>(identifier.edit());
     ASN1_STRING_set(acc->location->d.ia5, cident, identifier.len());
@@ -1059,7 +1059,7 @@
         GENERAL_NAMES *uris = GENERAL_NAMES_new();
         GENERAL_NAME *uri = GENERAL_NAME_new();
         uri->type = GEN_URI;
-        uri->d.ia5 = M_ASN1_IA5STRING_new();
+        uri->d.ia5 = ASN1_IA5STRING_new();
         unsigned char *cident
 	    = reinterpret_cast<unsigned char *>(i().edit());    
         ASN1_STRING_set(uri->d.ia5, cident, i().len());
@@ -1130,7 +1130,7 @@
 	pol->qualifiers = sk_POLICYQUALINFO_new_null();
 	qual = POLICYQUALINFO_new();
 	qual->pqualid = OBJ_nid2obj(NID_id_qt_cps);
-	qual->d.cpsouri = M_ASN1_IA5STRING_new();
+	qual->d.cpsouri = ASN1_IA5STRING_new();
 	ASN1_STRING_set(qual->d.cpsuri, url.edit(), url.len());
 	sk_POLICYQUALINFO_push(pol->qualifiers, qual);
     }
@@ -1450,19 +1450,19 @@
 {
     CHECK_CERT_EXISTS_SET("ski");
 
-    ASN1_OCTET_STRING *oct = M_ASN1_OCTET_STRING_new();
+    ASN1_OCTET_STRING *oct = ASN1_OCTET_STRING_new();
     ASN1_BIT_STRING *pk = cert->cert_info->key->public_key;
     unsigned char pkey_dig[EVP_MAX_MD_SIZE];
     unsigned int diglen;
 
     EVP_Digest(pk->data, pk->length, pkey_dig, &diglen, EVP_sha1(), NULL);
 
-    M_ASN1_OCTET_STRING_set(oct, pkey_dig, diglen);
+    ASN1_OCTET_STRING_set(oct, pkey_dig, diglen);
     X509_EXTENSION *ext = X509V3_EXT_i2d(NID_subject_key_identifier, 0, 
 					oct);
     X509_add_ext(cert, ext, -1);
     X509_EXTENSION_free(ext);
-    M_ASN1_OCTET_STRING_free(oct);
+    ASN1_OCTET_STRING_free(oct);
 }
 
 
