--- a/argp/argp-fmtstream.h	2016-11-30 11:59:03.981286513 +0400
+++ b/argp/argp-fmtstream.h	2016-11-30 12:00:49.524271289 +0400
@@ -199,8 +199,8 @@
 extern void __argp_fmtstream_update (argp_fmtstream_t __fs);
 extern int _argp_fmtstream_ensure (argp_fmtstream_t __fs, size_t __amount);
 extern int __argp_fmtstream_ensure (argp_fmtstream_t __fs, size_t __amount);
-
-#ifdef __OPTIMIZE__
+
+#if 0
 /* Inline versions of above routines.  */
 
 #if !_LIBC
