--- include/wvtripledes.h.old	2014-12-11 01:53:28.157613918 +0100
+++ include/wvtripledes.h	2014-12-11 01:53:46.005612583 +0100
@@ -70,11 +70,11 @@ protected:
 
 private:
     Mode mode;
-    des_cblock key;
-    des_key_schedule deskey1;
-    des_key_schedule deskey2;
-    des_key_schedule deskey3;
-    des_cblock ivec; // initialization vector
+    DES_cblock key;
+    DES_key_schedule deskey1;
+    DES_key_schedule deskey2;
+    DES_key_schedule deskey3;
+    DES_cblock ivec; // initialization vector
     int ivecoff; // current offset into initvec
 };
 
