--- a/uniconf/wvconfemu.cc.old	2015-09-22 02:07:38.395970444 -0400
+++ b/uniconf/wvconfemu.cc	2015-09-22 02:02:26.111102649 -0400
@@ -11,7 +11,7 @@
 #include "strutils.h"
 
 //#define DEBUG_DEL_CALLBACK 1
-#ifdef DEBUG_DEL_CALLBACK
+#if defined(DEBUG_DEL_CALLBACK) && defined(HAVE_EXECINFO_H)
 #include <execinfo.h>
 #endif
 
@@ -338,7 +338,7 @@
 	    return;
     }
 
-#ifdef DEBUG_DEL_CALLBACK
+#if defined(DEBUG_DEL_CALLBACK) && defined(HAVE_EXECINFO_H)
     void* trace[10];
     int count = backtrace(trace, sizeof(trace)/sizeof(trace[0]));
     char** tracedump = backtrace_symbols(trace, count);
