--- a/utils/wvtask.cc.old	2015-09-22 05:49:32.911132004 -0400
+++ b/utils/wvtask.cc	2015-09-22 05:50:36.230115961 -0400
@@ -540,7 +540,7 @@
 
 const void *WvTaskMan::current_top_of_stack()
 {
-#ifdef HAVE_LIBC_STACK_END
+#if defined(HAVE_LIBC_STACK_END) && defined(__GLIBC__)
     extern const void *__libc_stack_end;
     if (use_shared_stack() || current_task == NULL)
         return __libc_stack_end;
