diff --git a/fuzzylite/test/catch.hpp b/fuzzylite/test/catch.hpp
index 6f9334ba..0ef1af21 100644
--- a/fuzzylite/test/catch.hpp
+++ b/fuzzylite/test/catch.hpp
@@ -6375,7 +6375,7 @@ namespace Catch {
         static bool isSet;
         static struct sigaction oldSigActions [sizeof(signalDefs)/sizeof(SignalDefs)];
         static stack_t oldSigStack;
-        static char altStackMem[SIGSTKSZ];
+        static char altStackMem[4096];
 
         static void handleSignal( int sig ) {
             std::string name = "<unknown signal>";
@@ -6395,7 +6395,7 @@ namespace Catch {
             isSet = true;
             stack_t sigStack;
             sigStack.ss_sp = altStackMem;
-            sigStack.ss_size = SIGSTKSZ;
+            sigStack.ss_size = 4096;
             sigStack.ss_flags = 0;
             sigaltstack(&sigStack, &oldSigStack);
             struct sigaction sa = { 0 };
@@ -6426,7 +6426,7 @@ namespace Catch {
     bool FatalConditionHandler::isSet = false;
     struct sigaction FatalConditionHandler::oldSigActions[sizeof(signalDefs)/sizeof(SignalDefs)] = {};
     stack_t FatalConditionHandler::oldSigStack = {};
-    char FatalConditionHandler::altStackMem[SIGSTKSZ] = {};
+    char FatalConditionHandler::altStackMem[4096] = {};
 
 } // namespace Catch
 
