diff --git a/src/plugins/openfileslist/openfileslistplugin.h b/src/plugins/openfileslist/openfileslistplugin.h
index 7fd7a95..87b08ee 100644
--- a/src/plugins/openfileslist/openfileslistplugin.h
+++ b/src/plugins/openfileslist/openfileslistplugin.h
@@ -23,7 +23,7 @@ struct TargetFilesData
     // Functor for the std::set predicate to sort the opened editor files according to their tab order
     struct compareLess
     {
-        bool operator()(const ProjectFile* lhs, const ProjectFile* rhs) { return lhs->editorTabPos < rhs->editorTabPos; }
+        bool operator()(const ProjectFile* lhs, const ProjectFile* rhs) const { return lhs->editorTabPos < rhs->editorTabPos; }
     };
     typedef std::set<ProjectFile*, compareLess> OpenFilesSet;
     ProjectFile* activeFile;
