Index: codeblocks-20.03/src/sdk/scripting/bindings/sc_wxtypes.cpp
===================================================================
--- codeblocks-20.03.orig/src/sdk/scripting/bindings/sc_wxtypes.cpp
+++ codeblocks-20.03/src/sdk/scripting/bindings/sc_wxtypes.cpp
@@ -322,6 +322,7 @@ namespace ScriptBindings
         typedef bool(wxFileName::*WXFN_SETCWD)();
 #endif
         typedef bool(wxFileName::*WXFN_ISFILEWRITEABLE)()const;
+        typedef bool(wxFileName::*WXFN_NORMALIZE)(int, const wxString&, wxPathFormat);
 
         SqPlus::SQClassDef<wxFileName>("wxFileName").
                 emptyCtor().
@@ -354,7 +355,7 @@ namespace ScriptBindings
                 func(&wxFileName::IsDir, "IsDir").
                 func(&wxFileName::MakeAbsolute, "MakeAbsolute").
                 func(&wxFileName::MakeRelativeTo, "MakeRelativeTo").
-                func(&wxFileName::Normalize, "Normalize").
+                func<WXFN_NORMALIZE>(&wxFileName::Normalize, "Normalize").
                 func(&wxFileName::PrependDir, "PrependDir").
                 func(&wxFileName::RemoveDir, "RemoveDir").
                 func(&wxFileName::RemoveLastDir, "RemoveLastDir").
