diff --git a/Cargo.toml b/Cargo.toml
index ff57424af7e..27e0ed3c154 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -78,7 +78,7 @@ nu-std = { path = "./crates/nu-std", version = "0.90.1" }
 nu-utils = { path = "./crates/nu-utils", version = "0.90.1" }
 
 nu-ansi-term = "0.50.0"
-reedline = { version = "0.29.0", features = ["bashisms", "sqlite"] }
+reedline = { version = "0.29.0", features = ["bashisms", "sqlite-dynlib"] }
 
 crossterm = "0.27"
 ctrlc = "3.4"
diff --git a/crates/nu-cli/Cargo.toml b/crates/nu-cli/Cargo.toml
index cbcfe333e0f..ab349ecbc6a 100644
--- a/crates/nu-cli/Cargo.toml
+++ b/crates/nu-cli/Cargo.toml
@@ -25,7 +25,7 @@ nu-protocol = { path = "../nu-protocol", version = "0.90.1" }
 nu-utils = { path = "../nu-utils", version = "0.90.1" }
 nu-color-config = { path = "../nu-color-config", version = "0.90.1" }
 nu-ansi-term = "0.50.0"
-reedline = { version = "0.29.0", features = ["bashisms", "sqlite"] }
+reedline = { version = "0.29.0", features = ["bashisms", "sqlite-dynlib"] }
 
 chrono = { default-features = false, features = ["std"], version = "0.4" }
 crossterm = "0.27"
diff --git a/crates/nu-command/Cargo.toml b/crates/nu-command/Cargo.toml
index 56f6a3ab622..9779f991a18 100644
--- a/crates/nu-command/Cargo.toml
+++ b/crates/nu-command/Cargo.toml
@@ -72,7 +72,7 @@ rand = "0.8"
 rayon = "1.8"
 regex = "1.9.5"
 roxmltree = "0.18"
-rusqlite = { version = "0.29", features = ["bundled", "backup", "chrono"], optional = true }
+rusqlite = { version = "0.29", features = ["backup", "chrono"], optional = true }
 same-file = "1.0"
 serde = { version = "1.0", features = ["derive"] }
 serde_json = { version = "1.0", features = ["preserve_order"] }
