diff --git a/Cargo.lock b/Cargo.lock
index caa34e62..2e52d8b8 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -2135,7 +2135,6 @@ version = "0.26.0"
 source = "registry+https://github.com/rust-lang/crates.io-index"
 checksum = "afc22eff61b133b115c6e8c74e818c628d6d5e7a502afea6f64dee076dd94326"
 dependencies = [
- "cc",
  "pkg-config",
  "vcpkg",
 ]
diff --git a/Cargo.toml b/Cargo.toml
index 601e18cb..0458e0c8 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -67,7 +67,7 @@ nu-term-grid = { path = "./crates/nu-term-grid", version = "0.84.0" }
 nu-std = { path = "./crates/nu-std", version = "0.84.0" }
 nu-utils = { path = "./crates/nu-utils", version = "0.84.0" }
 nu-ansi-term = "0.49.0"
-reedline = { version = "0.23.0", features = ["bashisms", "sqlite"]}
+reedline = { version = "0.23.0", features = ["bashisms", "sqlite-dynlib"]}
 
 crossterm = "0.26"
 ctrlc = "3.4"
diff --git a/crates/nu-cli/Cargo.toml b/crates/nu-cli/Cargo.toml
index a3871ff2..41873e16 100644
--- a/crates/nu-cli/Cargo.toml
+++ b/crates/nu-cli/Cargo.toml
@@ -25,7 +25,7 @@ nu-protocol = { path = "../nu-protocol", version = "0.84.0" }
 nu-utils = { path = "../nu-utils", version = "0.84.0" }
 nu-color-config = { path = "../nu-color-config", version = "0.84.0" }
 nu-ansi-term = "0.49.0"
-reedline = { version = "0.23.0", features = ["bashisms", "sqlite"]}
+reedline = { version = "0.23.0", features = ["bashisms", "sqlite-dynlib"]}
 
 chrono = { default-features = false, features = ["std"], version = "0.4" }
 crossterm = "0.26"
diff --git a/crates/nu-command/Cargo.toml b/crates/nu-command/Cargo.toml
index f6c6f62f..7185e1be 100644
--- a/crates/nu-command/Cargo.toml
+++ b/crates/nu-command/Cargo.toml
@@ -74,7 +74,7 @@ rand = "0.8"
 rayon = "1.7"
 regex = "1.7"
 roxmltree = "0.18"
-rusqlite = { version = "0.29", features = ["bundled"], optional = true }
+rusqlite = { version = "0.29", optional = true }
 same-file = "1.0"
 serde = { version = "1.0", features = ["derive"] }
 serde_json = "1.0"
