diff --git a/Cargo.toml.orig b/Cargo.toml
index 7c53d6d1ed..510ef7698b 100644
--- a/Cargo.toml.orig
+++ b/Cargo.toml
@@ -227,3 +227,7 @@ required-features = ["sqlite"]
 [[bin]]
 name = "nu"
 path = "src/main.rs"
+
+[dependencies.zstd]
+features = ["pkg-config"]
+version = ">=0.0.0"
====
diff --git a/crates/nu_plugin_from_sqlite/Cargo.toml.orig b/crates/nu_plugin_from_sqlite/Cargo.toml
index ef28788f42..ad439b5fd3 100644
--- a/crates/nu_plugin_from_sqlite/Cargo.toml.orig
+++ b/crates/nu_plugin_from_sqlite/Cargo.toml
@@ -18,7 +18,7 @@ nu-source = { path="../nu-source", version = "0.42.0" }
 tempfile = "3.2.0"
 
 [dependencies.rusqlite]
-features = ["bundled", "blob"]
+features = ["blob"]
 version = "0.26.1"
 
 [build-dependencies]
====
diff --git a/crates/nu_plugin_to_sqlite/Cargo.toml.orig b/crates/nu_plugin_to_sqlite/Cargo.toml
index 6e26ec5fda..a551200402 100644
--- a/crates/nu_plugin_to_sqlite/Cargo.toml.orig
+++ b/crates/nu_plugin_to_sqlite/Cargo.toml
@@ -18,7 +18,7 @@ nu-source = { path="../nu-source", version = "0.42.0" }
 tempfile = "3.2.0"
 
 [dependencies.rusqlite]
-features = ["bundled", "blob"]
+features = ["blob"]
 version = "0.26.1"
 
 [build-dependencies]
