Index: nushell-0.34.0/Cargo.toml
===================================================================
--- nushell-0.34.0.orig/Cargo.toml
+++ nushell-0.34.0/Cargo.toml
@@ -259,3 +259,7 @@ required-features = ["sqlite"]
 [[bin]]
 name = "nu"
 path = "src/main.rs"
+
+[dependencies.zstd]
+features = ["pkg-config"]
+version = ">=0.0.0"
Index: nushell-0.34.0/crates/nu_plugin_from_sqlite/Cargo.toml
===================================================================
--- nushell-0.34.0.orig/crates/nu_plugin_from_sqlite/Cargo.toml
+++ nushell-0.34.0/crates/nu_plugin_from_sqlite/Cargo.toml
@@ -20,7 +20,7 @@ num-traits = "0.2.14"
 tempfile = "3.2.0"
 
 [dependencies.rusqlite]
-features = ["bundled", "blob"]
+features = ["blob"]
 version = "0.25.3"
 
 [build-dependencies]
Index: nushell-0.34.0/crates/nu_plugin_to_sqlite/Cargo.toml
===================================================================
--- nushell-0.34.0.orig/crates/nu_plugin_to_sqlite/Cargo.toml
+++ nushell-0.34.0/crates/nu_plugin_to_sqlite/Cargo.toml
@@ -20,7 +20,7 @@ num-traits = "0.2.14"
 tempfile = "3.2.0"
 
 [dependencies.rusqlite]
-features = ["bundled", "blob"]
+features = ["blob"]
 version = "0.25.3"
 
 [build-dependencies]
