--- configure
+++ configure
@@ -214,19 +215,28 @@
 		return 1
 	fi
 
-	if ./test-${1} 1>&3 2>&3; then
-		echo "${1}: yes" 1>&2
-		echo "${1}: yes" 1>&3
-		echo 1>&3
-		eval HAVE_${2}=1
-		rm "test-${1}"
-		return 0
-	else
-		echo "${1}: execution failed with $?" 1>&3
-		echo 1>&3
-		rm "test-${1}"
-		return 1
-	fi
+	echo "${1}: yes" 1>&2
+	echo "${1}: yes" 1>&3
+	echo 1>&3
+	eval HAVE_${2}=1
+	rm "test-${1}"
+	return 0
+
+	# Don't actually run the test: none of our tests check for
+	# run-time behaviour.
+	# if ./test-${1} 1>&3 2>&3; then
+	# 	echo "${1}: yes" 1>&2
+	# 	echo "${1}: yes" 1>&3
+	# 	echo 1>&3
+	# 	eval HAVE_${2}=1
+	# 	rm "test-${1}"
+	# 	return 0
+	# else
+	# 	echo "${1}: execution failed with $?" 1>&3
+	# 	echo 1>&3
+	# 	rm "test-${1}"
+	# 	return 1
+	# fi
 }
 
 # Run a complete autoconfiguration test, including the check for
