diff --git DLL430_v3/src/TI/DLL430/HidUpdateManager.cpp DLL430_v3/src/TI/DLL430/HidUpdateManager.cpp
index 5b74a2a..e1358df 100644
--- DLL430_v3/src/TI/DLL430/HidUpdateManager.cpp
+++ DLL430_v3/src/TI/DLL430/HidUpdateManager.cpp
@@ -38,7 +38,7 @@
 #include <pch.h>
 #include "HidUpdateManager.h"
 
-#include <hidapi.h>
+#include <hidapi/hidapi.h>
 
 #include <BSL430_DLL/Connections/MSPBSL_Connection5xxUSB.h>
 #include <BSL430_DLL/MSPBSL_Factory.h>
diff --git Makefile Makefile
index a4cfbed..7c65736 100644
--- Makefile
+++ Makefile
@@ -55,7 +55,7 @@ ifeq ($(PLATFORM),Linux)
 	LIBS += -lusb-1.0
 	endif
 
-	LIBS += -lusb-1.0 -lrt -lpthread
+	LIBS += -lusb-1.0 -lrt -lpthread -lhidapi-libusb
 
 	ifdef BOOST_DIR
 	INCLUDES += -I$(BOOST_DIR)
@@ -66,7 +66,7 @@ ifeq ($(PLATFORM),Linux)
 	BSTATIC := -Wl,-Bstatic
 	BDYNAMIC := -Wl,-Bdynamic
 
-	HIDOBJ := $(LIBTHIRD)/hid-libusb.o
+	HIDOBJ :=
 else
 	CXX:= clang++
 
diff --git ThirdParty/BSL430_DLL/BSL430_DLL/Physical_Interfaces/MSPBSL_PhysicalInterfaceUSB.h ThirdParty/BSL430_DLL/BSL430_DLL/Physical_Interfaces/MSPBSL_PhysicalInterfaceUSB.h
index 37825c6..d1105ed 100644
--- ThirdParty/BSL430_DLL/BSL430_DLL/Physical_Interfaces/MSPBSL_PhysicalInterfaceUSB.h
+++ ThirdParty/BSL430_DLL/BSL430_DLL/Physical_Interfaces/MSPBSL_PhysicalInterfaceUSB.h
@@ -38,7 +38,7 @@
 
 #pragma once
 
-#include <hidapi.h>
+#include <hidapi/hidapi.h>
 
 #include "MSPBSL_PhysicalInterface.h"
 
