--- a/src-qt5/OS-detect.pri	2017-07-05 20:02:49.727200592 +0200
+++ b/src-qt5/OS-detect.pri	2017-07-05 20:03:10.463911159 +0200
@@ -94,7 +94,7 @@
   isEmpty(L_SHAREDIR){ L_SHAREDIR = $${PREFIX}/share }
   isEmpty(L_INCLUDEDIR){ L_INCLUDEDIR = $${PREFIX}/include }
   isEmpty(L_SESSDIR){ L_SESSDIR = $${L_SHAREDIR}/xsessions }
-  isEmpty(L_MANDIR){ L_MANDIR = $${PREFIX}/man }
+  isEmpty(L_MANDIR){ L_MANDIR = $${PREFIX}/share/man }
   isEmpty(LRELEASE){ LRELEASE = $$[QT_INSTALL_BINS]/lrelease }
   isEmpty(MAN_ZIP){ MAN_ZIP = "gzip -c" }
   !exists(LRELEASE){ NO_I18N=true } #translations unavailable
