--- Makefile
+++ patches/Makefile
@@ -6,21 +6,21 @@
 CC     = cc
 CFLAGS = -std=c11 -D_POSIX_C_SOURCE=200809L -Wall -Wextra -pedantic -O2
 LDLIBS = -s -lX11
+PREFIX = /usr
 
-BIN_DIR = /usr/local/bin
-
 SRC = devour.c
 OBJ = devour.o
 
 all: $(NAME)
 $(NAME): $(OBJ)
 install: all
-	@mkdir -p $(BIN_DIR)
-	@mv $(NAME) $(BIN_DIR)
-	@rm -f $(OBJ)
-	@echo Done moving the binary to ${DESTDIR}${BIN_DIR}
+	@mkdir -p ${DESTDIR}${PREFIX}/bin
+	@mv devour ${DESTDIR}${PREFIX}/bin
+	@rm -f ${OBJ}
+	@echo Done installing executable files to ${DESTDIR}${PREFIX}/bin
+
 uninstall:
-	@rm -f $(BIN_DIR)/$(NAME)
-	@echo Done removing the binary from $(BIN_DIR)
+	@rm -f ${DESTDIR}${PREFIX}/bin/devour
+	@echo Done removing executable files from ${DESTDIR}${PREFIX}/bin
 
 .PHONY: all install uninstall
