diff --git a/extern/ESFMu/esfm.c b/extern/ESFMu/esfm.c
index 6c74324..a07571f 100644
--- a/extern/ESFMu/esfm.c
+++ b/extern/ESFMu/esfm.c
@@ -1908,7 +1908,7 @@ ESFM_process_feedback(esfm_chip *chip)
 				  [i]      "m"   (iter_counter)
 				: "cc", "ax", "bx", "cx", "di"
 			);
-#elif defined(__GNUC__) && defined(__arm__)
+#elif defined(__GNUC__) && defined(__arm__) && 0
 			asm (
 				"ldr     r3, =%[sinrom]             \n\t"
 				"ldrb    r0, %[wave]                \n\t"
