Index: CMakeLists.txt
===================================================================
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -9,7 +9,7 @@ set (CMAKE_MODULE_PATH ${CMAKE_MODULE_PA
      "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
 
 add_definitions(-D_GNU_SOURCE)
-set(CMAKE_C_FLAGS "-Wall -std=c99")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC -Wall -std=c99")
 
 # find fuse library
 find_package (FUSE REQUIRED)
