From 73d1c4ecbddf64b729a16aa4ebe9e9151c975891 Mon Sep 17 00:00:00 2001
From: Daniel Kolesa <daniel@octaforge.org>
Date: Thu, 23 Jan 2020 19:49:43 +0100
Subject: [PATCH xf86-video-mach64 1/4] avoid including VBE paths when not used

---
 src/atipreinit.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git src/atipreinit.c src/atipreinit.c
index 96ec68f..be30141 100644
--- a/src/atipreinit.c
+++ b/src/atipreinit.c
@@ -51,7 +51,9 @@
 #include "atixv.h"
 #include "atiadjust.h"
 
+#if !defined(AVOID_CPIO) || defined(TV_OUT)
 #include "vbe.h"
+#endif
 #if GET_ABI_MAJOR(ABI_VIDEODRV_VERSION) < 6
 #include "xf86RAC.h"
 #endif
@@ -572,7 +574,9 @@ ATIPreInit
         return FALSE;
     }
 #endif
+#if !defined(AVOID_CPIO) || defined(TV_OUT)
     ConfiguredMonitor = NULL;
+#endif
     (void)memset(BIOS, 0, SizeOf(BIOS));
 
     if (!(flags & PROBE_DETECT))
-- 
2.25.0

