fix cross similar to xf86-video-mga's configure.ac

diff --git a/configure.ac.orig b/configure.ac
index 6170c94..66bb1da 100644
--- a/configure.ac
+++ b/configure.ac
@@ -79,6 +79,7 @@ sdkdir=`$PKG_CONFIG --variable=sdkdir xorg-server`
 # Checks for libraries.
 
 if test "$DRI" != no; then
+	if test "$cross_compiling" = no; then
 	AC_CHECK_FILE([${sdkdir}/dri.h],
                       [have_dri_h="yes"], [have_dri_h="no"])
 	AC_CHECK_FILE([${sdkdir}/sarea.h],
@@ -87,6 +88,12 @@ if test "$DRI" != no; then
                       [have_dristruct_h="yes"], [have_dristruct_h="no"])
 	AC_CHECK_FILE([${sdkdir}/damage.h],
                       [have_damage_h="yes"], [have_damage_h="no"])
+	else
+		   have_dri_h="yes"
+		   have_sarea_h="yes"
+		   have_dristruct_h="yes"
+           have_damage_h="yes"
+	fi
 fi
 
 AC_MSG_CHECKING([whether to include DRI support])
