diff --git a/src/notifications.c b/src/notifications.c
index f5ac1f8..e366a12 100644
--- a/src/notifications.c
+++ b/src/notifications.c
@@ -7,6 +7,7 @@
  *
  */
 
+#include <stdio.h>
 #include <libnotify/notify.h>
 
 #include "audio_status.h"
@@ -73,7 +74,13 @@ void notifications_flash(void)
     notify_notification_set_hint_int32(notification, "value", (gint)as->volume);
 
     // Update the notification icon
-    notify_notification_update(notification, PROGRAM_NAME, NULL, icon_name);
+    char vol_text[5];
+    if (as->muted) {
+      snprintf(vol_text, 5, "mute");
+    } else {
+      snprintf(vol_text, 5, "%3.0f%%", as->volume);
+    }
+    notify_notification_update(notification, PROGRAM_NAME, vol_text, icon_name);
 
     // Show the notification
     notify_notification_show(notification, NULL);
