--- src/glshim/src/glx/glx.c	2018-11-07 04:14:41.000000000 +0100
+++ -	2019-01-30 20:49:46.097821225 +0100
@@ -1,4 +1,6 @@
+#ifdef __GLIBC__
 #include <execinfo.h>
+#endif
 #include <fcntl.h>
 #include <linux/fb.h>
 #include <signal.h>
@@ -168,7 +170,7 @@
         bcm_host_deinit();
     }
 #endif
-
+#ifdef __GLIBC__
     if (g_stacktrace) {
         switch (sig) {
             case SIGBUS:
@@ -187,6 +189,7 @@
             }
         }
     }
+#endif
     signal(sig, SIG_DFL);
     raise(sig);
 }
