--- a/tests/decorate-errors.pl
+++ b/tests/decorate-errors.pl
@@ -149,7 +149,7 @@
     {ERR=>"$prog: invalid dot-decimal IPv4 address 'FOO'\n" .
           "$prog: conversion failed in line 1\n" }],
   ['c4', '--decorate -k1,1:ipv4inet' , {IN_PIPE=>"FOO\n"}, {OUT => " "},
-    {EXIT=>2},
+    {EXIT=>2}, {ERR_SUBST=>'s/: Invalid argument//'},
     {ERR=>"$prog: invalid IPv4 address 'FOO'\n" .
           "$prog: conversion failed in line 1\n" }],
   ['c5', '--decorate -k1,1:ipv6' , {IN_PIPE=>"FOO\n"}, {OUT => " "}, {EXIT=>2},
