--- base/process/memory_linux.cc.orig	2017-09-15 08:41:43.000000000 +0000
+++ base/process/memory_linux.cc	2017-09-15 08:44:39.804995469 +0000
@@ -21,6 +21,12 @@
 #include "third_party/tcmalloc/chromium/src/gperftools/tcmalloc.h"
 #endif
 
+#if defined(LIBC_GLIBC)
+extern "C" {
+extern void *__libc_malloc(size_t size);
+}
+#endif
+
 namespace base {
 
 size_t g_oom_size = 0U;
--- base/process/memory_linux.cc.orig	2020-08-30 14:18:35.401132593 -0400
+++ base/process/memory_linux.cc	2020-08-30 14:19:08.030199189 -0400
@@ -141,7 +141,7 @@
     (!defined(LIBC_GLIBC) && !BUILDFLAG(USE_TCMALLOC))
   *result = malloc(size);
 #elif defined(LIBC_GLIBC) && !BUILDFLAG(USE_TCMALLOC)
-  *result = __libc_malloc(size);
+  *result = ::__libc_malloc(size);
 #elif BUILDFLAG(USE_TCMALLOC)
   *result = tc_malloc_skip_new_handler(size);
 #endif
