--- a/cmake/modules/FindCrossGUID.cmake	2019-01-30 18:31:51.519546398 +0100
+++ b/cmake/modules/FindCrossGUID.cmake	2019-01-30 18:32:51.197608565 +0100
@@ -49,10 +49,10 @@
   set(CROSSGUID_LIBRARIES ${CROSSGUID_LIBRARY})
   set(CROSSGUID_INCLUDE_DIRS ${CROSSGUID_INCLUDE_DIR})
 else()
-  find_path(CROSSGUID_INCLUDE_DIR NAMES guid.h)
+  find_path(CROSSGUID_INCLUDE_DIR NAMES Guid.hpp)
 
-  find_library(CROSSGUID_LIBRARY_RELEASE NAMES crossguid)
-  find_library(CROSSGUID_LIBRARY_DEBUG NAMES crossguidd)
+  find_library(CROSSGUID_LIBRARY_RELEASE NAMES xg)
+  find_library(CROSSGUID_LIBRARY_DEBUG NAMES xg)
 
   include(SelectLibraryConfigurations)
   select_library_configurations(CROSSGUID)
--- a/xbmc/utils/StringUtils.cpp	2019-01-30 18:35:42.588517593 +0100
+++ b/xbmc/utils/StringUtils.cpp	2019-01-30 18:37:38.058555463 +0100
@@ -16,7 +16,7 @@
 //
 //------------------------------------------------------------------------
 
-#include <guid.h>
+#include <Guid.hpp>
 
 #if defined(TARGET_ANDROID)
 #include <androidjni/JNIThreading.h>
@@ -1126,8 +1126,7 @@
 
 std::string StringUtils::CreateUUID()
 {
-  static GuidGenerator guidGenerator;
-  auto guid = guidGenerator.newGuid();
+  auto guid = xg::newGuid();
 
   std::stringstream strGuid; strGuid << guid;
   return strGuid.str();
