--- a/configure.ac.orig	2015-07-23 23:49:37.887069973 +0200
+++ b/configure.ac	2015-07-23 23:51:40.074074325 +0200
@@ -2170,15 +2170,12 @@ else
   final_message="$final_message\n  Bluray:\tNo"
 fi
 
-# we don't compile with default upstream squish, so we use our own one
-# even on non unified deps platforms
-if test "$cross_compiling" != "yes"; then
-  make -C ${abs_top_srcdir}/tools/depends/native/libsquish-native CXX=${CXX}
-  export PKG_CONFIG_PATH="${abs_top_srcdir}/tools/depends/native/libsquish-native/squish-install/lib/pkgconfig:$PKG_CONFIG_PATH"
-fi
-PKG_CHECK_MODULES([SQUISH], [squish],
- [INCLUDES="$INCLUDES $SQUISH_CFLAGS" LIBS="$LIBS $SQUISH_LIBS"],
- AC_MSG_ERROR("squish not found"))
+make -C ${abs_top_srcdir}/tools/depends/native/libsquish-native CXX=${CXX}
+SQUISH_CFLAGS="-I${abs_top_srcdir}/tools/depends/native/libsquish-native/squish-install/include "
+SQUISH_LIBS="-L${abs_top_srcdir}/tools/depends/native/libsquish-native/squish-install/lib -lsquish "
+INCLUDES="$INCLUDES $SQUISH_CFLAGS"
+LIBS="$LIBS $SQUISH_LIBS"
+
 
 if test "x$use_texturepacker" != "xno"; then
   AC_PATH_PROG([TEXTUREPACKER], [TexturePacker], ["none"],
