--- a/configure.ac.orig	2017-05-21 19:04:34.705518286 +0200
+++ b/configure.ac	2017-05-21 19:05:10.431442782 +0200
@@ -611,7 +611,7 @@ case $host in
      CORE_SYSTEM_NAME=linux
      ARCH="powerpc64-linux"
      ;;
-  arm*-*-linux-gnu*|arm*-*-linux-uclibc*)
+  arm*-*-linux-gnu*|arm*-*-linux-uclibc*|arm*-*-linux-musl*)
      target_platform=target_linux
      CORE_SYSTEM_NAME=linux
      ARCH="arm"
@@ -621,7 +621,7 @@ case $host in
      use_gl=no
      USE_STATIC_FFMPEG=1
      ;;
-  aarch64*-*-linux-gnu*|aarch64*-*-linux-uclibc*)
+  aarch64*-*-linux-gnu*|aarch64*-*-linux-uclibc*|aarch64*-*-linux-musl*)
      target_platform=target_linux
      ARCH="aarch64"
      use_arch="aarch64"
