https://savannah.gnu.org/bugs/?33873

Index: main.c
===================================================================
RCS file: /sources/make/make/main.c,v
retrieving revision 1.246
diff -u -r1.246 main.c
--- main.c	29 Aug 2010 23:05:27 -0000	1.246
+++ main.c	27 Jul 2011 22:03:12 -0000
@@ -2091,10 +2095,16 @@

           /* Reset makeflags in case they were changed.  */
           {
+            if (master_job_slots)
+              {
+                assert (job_slots == default_job_slots);
+                job_slots = master_job_slots;
+              }
             const char *pv = define_makeflags (1, 1);
             char *p = alloca (sizeof ("MAKEFLAGS=") + strlen (pv) + 1);
             sprintf (p, "MAKEFLAGS=%s", pv);
             putenv (allocated_variable_expand (p));
+            job_slots = default_job_slots;
           }

	  if (ISDB (DB_BASIC))
@@ -2825,9 +2852,11 @@
		       && (*(unsigned int *) cs->value_ptr ==
			   *(unsigned int *) cs->noarg_value))
		ADD_FLAG ("", 0); /* Optional value omitted; see below.  */
+#if 0
	      else if (cs->c == 'j')
		/* Special case for `-j'.  */
		ADD_FLAG ("1", 1);
+#endif
	      else
		{
		  char *buf = alloca (30);
