--- job.c~	2010-08-11 16:13:33.000000000 +0200
+++ job.c	2010-08-12 14:20:08.000000000 +0200
@@ -2442,7 +2442,11 @@ construct_command_argv_internal (char *l
 
   /* See if it is safe to parse commands internally.  */
   if (shell == 0)
-    shell = default_shell;
+    {
+      shell = default_shell;
+      if (shellflags == 0)
+	shellflags = "-c";
+    }
 #ifdef WINDOWS32
   else if (strcmp (shell, default_shell))
   {
