https://bugs.archlinux.org/task/22733 (fix from Fedora)

--- function.c	2010-07-13 03:20:39.000000000 +0200
+++ function.c	2010-10-27 01:43:27.000000000 +0200
@@ -1138,12 +1138,12 @@ func_sort (char *o, char **argv, const c
     {
       char c = *(t++);
 
-      if (! isspace ((unsigned char)c))
+      if (! isblank ((unsigned char)c))
         continue;
 
       ++wordi;
 
-      while (isspace ((unsigned char)*t))
+      while (isblank ((unsigned char)*t))
         ++t;
     }
