--- a/re2.pc.in
+++ b/re2.pc.in
@@ -3,7 +3,7 @@ libdir=@CMAKE_INSTALL_FULL_LIBDIR@
 
 Name: re2
 Description: RE2 is a fast, safe, thread-friendly regular expression engine.
-Requires: @REQUIRES@
+Requires: absl_base absl_core_headers absl_strings
 Version: @SONAME@.0.0
 Cflags: -pthread -I${includedir}
 Libs: -pthread -L${libdir} -lre2
--- a/re2Config.cmake.in
+++ b/re2Config.cmake.in
@@ -13,10 +13,6 @@ endif()
 
 find_dependency(absl REQUIRED)
 
-if(@RE2_USE_ICU@)
-  find_dependency(ICU REQUIRED COMPONENTS uc)
-endif()
-
 check_required_components(re2)
 
 if(TARGET re2::re2)
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -156,11 +156,12 @@ foreach(dep ${ABSL_DEPS})
   # Work around https://gitlab.kitware.com/cmake/cmake/-/issues/16899. >:(
   string(PREPEND dep "^")
   string(REGEX REPLACE "\\^absl_" "absl::" dep ${dep})
-  target_link_libraries(re2 PUBLIC ${dep})
+  target_link_libraries(re2 PRIVATE ${dep})
 endforeach()
+target_link_libraries(re2 PUBLIC absl::base absl::core_headers absl::strings)
 
 if(RE2_USE_ICU)
-  target_link_libraries(re2 PUBLIC ICU::uc)
+  target_link_libraries(re2 PRIVATE ICU::uc)
 endif()
 
 if(RE2_BUILD_TESTING)
