--- a/plugins/perl/meson.build	2018-08-15 13:30:17.108881970 +0200
+++ b/plugins/perl/meson.build	2018-08-15 13:30:36.682882751 +0200
@@ -39,7 +39,7 @@
 perl_ldflags = []
 perl_rpath = ''
 foreach flag : ret.stdout().strip().split(' ')
-  if flag.startswith('-L') or flag.startswith('-l')
+  if flag.startswith('-l')
     perl_ldflags += flag
   endif
   if flag.startswith('-Wl,-rpath,')
@@ -52,6 +52,8 @@
   endif
 endforeach
 
+perl_ldflags += '-L${XBPS_CROSS_BASE}/usr/lib/perl5/core_perl/CORE'
+
 perl_cflags += [
   # Perl has its own 'config.h' that we must override
   # TODO: Just rename ours to something more unique.
