--- a/makefile	2016-05-24 14:50:13.551844062 +0200
+++ b/makefile	2016-05-24 14:50:22.173843417 +0200
@@ -128,28 +128,28 @@ MYDOCS=sloccount.html README TODO Change
 all: $(COMPILED_EXECUTABLES)
 
 lexcount1$(EXE_SUFFIX): lexcount1.c
-	$(CC) lexcount1.c -o lexcount1$(EXE_SUFFIX)
+	$(CC) $(CFLAGS) $(LDFLAGS) lexcount1.c -o lexcount1$(EXE_SUFFIX)
 
 c_count$(EXE_SUFFIX): c_count.c
-	$(CC) c_count.c -o c_count$(EXE_SUFFIX)
+	$(CC) $(CFLAGS) $(LDFLAGS) c_count.c -o c_count$(EXE_SUFFIX)
 
 php_count$(EXE_SUFFIX): php_count.c
-	$(CC) php_count.c -o php_count$(EXE_SUFFIX)
+	$(CC) $(CFLAGS) $(LDFLAGS) php_count.c -o php_count$(EXE_SUFFIX)
 
 pascal_count.c: pascal_count.l driver.c driver.h
 	flex -Cfe -t pascal_count.l > pascal_count.c
 
 pascal_count$(EXE_SUFFIX): pascal_count.c
-	$(CC) pascal_count.c -o pascal_count$(EXE_SUFFIX)
+	$(CC) $(CFLAGS) $(LDFLAGS) pascal_count.c -o pascal_count$(EXE_SUFFIX)
 
 jsp_count.c: jsp_count.l driver.c driver.h
 	flex -Cfe -t jsp_count.l > jsp_count.c
 
 jsp_count$(EXE_SUFFIX): jsp_count.c
-	$(CC) jsp_count.c -o jsp_count$(EXE_SUFFIX)
+	$(CC) $(CFLAGS) $(LDFLAGS) jsp_count.c -o jsp_count$(EXE_SUFFIX)
 
 ml_count$(EXE_SUFFIX): ml_count.c
-	$(CC) ml_count.c -o ml_count$(EXE_SUFFIX)
+	$(CC) $(CFLAGS) $(LDFLAGS) ml_count.c -o ml_count$(EXE_SUFFIX)
 
 sloccount.1.gz: sloccount.1
 	gzip -c sloccount.1 > sloccount.1.gz
@@ -160,7 +160,7 @@ java_count$(EXE_SUFFIX): c_count$(EXE_SU
 
 # This is USC's code counter, not built by default:
 c_lines: C_LINES.C
-	$(CC) C_LINES.C -o c_lines$(EXE_SUFFIX)
+	$(CC) $(CFLAGS) $(LDFLAGS) C_LINES.C -o c_lines$(EXE_SUFFIX)
 
 
 install_programs: all
