diff --git src/declarativeimports/core/CMakeLists.txt src/declarativeimports/core/CMakeLists.txt
index 1d97bb5..bdf3240 100644
--- a/src/declarativeimports/core/CMakeLists.txt
+++ b/src/declarativeimports/core/CMakeLists.txt
@@ -60,7 +60,7 @@ if(HAVE_EGL)
 endif()
 
 if (${Qt5Gui_OPENGL_IMPLEMENTATION} STREQUAL "GLESv2")
-    target_link_libraries(corebindingsplugin Qt5::Gui_GLESv2)
+    target_link_libraries(corebindingsplugin GLESv2)
 endif()
 
 install(TARGETS corebindingsplugin DESTINATION ${KDE_INSTALL_QMLDIR}/org/kde/plasma/core)
