--- a/aclocal.m4
+++ b/aclocal.m4
@@ -315,51 +315,11 @@
 [
   # https://raw.githubusercontent.com/tmux/tmux/2dd9a4fb9cd73987bdca5b8b2f85ca8b1a6e4e73/configure.ac
 
-  # Check for b64_ntop. If we have b64_ntop, we assume b64_pton as well.
-  AC_MSG_CHECKING(for b64_ntop)
-  AC_RUN_IFELSE([AC_LANG_PROGRAM([[
-      #include <sys/types.h>
-      #include <netinet/in.h>
-      #include <resolv.h>
-    ]], [[b64_ntop(NULL, 0, NULL, 0);]])],[found_b64_ntop=yes],[found_b64_ntop=no
-  ])
-  if test "x$found_b64_ntop" = xno; then
-    AC_MSG_RESULT(no)
-
-    AC_MSG_CHECKING(for b64_ntop with -lresolv)
-    OLD_LIBS="$LIBS"
-    LIBS="$LIBS -lresolv"
-    AC_RUN_IFELSE([AC_LANG_PROGRAM([[
-        #include <sys/types.h>
-        #include <netinet/in.h>
-        #include <resolv.h>
-      ]], [[b64_ntop(NULL, 0, NULL, 0);]])],[found_b64_ntop=yes],[found_b64_ntop=no
-    ])
-    if test "x$found_b64_ntop" = xno; then
-      LIBS="$OLD_LIBS"
-      AC_MSG_RESULT(no)
-
-      AC_MSG_CHECKING(for b64_ntop with -lnetwork)
-      OLD_LIBS="$LIBS"
-      LIBS="-lnetwork"
-      AC_RUN_IFELSE([AC_LANG_PROGRAM([[
-        #include <sys/types.h>
-        #include <netinet/in.h>
-        #include <resolv.h>
-        ]], [[b64_ntop(NULL, 0, NULL, 0);]])],[found_b64_ntop=yes],[found_b64_ntop=no
-      ])
-      if test "x$found_b64_ntop" = xno; then
-        LIBS="$OLD_LIBS"
-        AC_MSG_RESULT(no)
-      fi
-    fi
-  fi
-  if test "x$found_b64_ntop" = xyes; then
-    AC_DEFINE([HAVE_BASE64], [1], [Define if b64_ntop exists.])
-    AC_MSG_RESULT(yes)
-  else
-    AC_LIBOBJ(base64)
-  fi
+  AC_MSG_CHECKING(for b64_ntop with -lresolv)  
+  OLD_LIBS="$LIBS"
+  LIBS="$LIBS -lresolv"
+  AC_DEFINE([HAVE_BASE64], [1], [Define if b64_ntop exists.])
+  AC_MSG_RESULT(yes)
 ])
 
 
