
--- a/gemini/qml/welcomepages/WelcomePageCloud.qml
+++ b/gemini/qml/welcomepages/WelcomePageCloud.qml
@@ -8,7 +8,7 @@ import QtQuick 2.0
 import QtQuick.Layouts 1.11 as QtLayouts
 import QtQuick.Controls 2.2 as QtControls
 import org.kde.kirigami 2.7 as Kirigami
-import Calligra.Gemini.Dropbox 1.0
+// import Calligra.Gemini.Dropbox 1.0
 import org.calligra 1.0
 import "../components"
 import "cloud"
@@ -37,18 +37,9 @@ Kirigami.ScrollablePage {
     Connections {
         target: controllerMIT;
         onNeedAuthenticateChanged: {
-            if(controllerMIT.needAuthenticate) {
-                cloudAccounts.removeAccountByName("Dropbox");
-            }
-            else {
-                cloudAccounts.addAccount("Dropbox", "DropBox", "accountsPageDropbox", 0, true);
-            }
         }
     }
     Component.onCompleted: {
-        if(!controllerMIT.needAuthenticate) {
-            cloudAccounts.addAccount("Dropbox", "DropBox", "accountsPageDropbox", 0, true);
-        }
     }
     ListView {
         id: accountsView;
@@ -76,7 +67,7 @@ Kirigami.ScrollablePage {
                 spacing: Kirigami.Units.largeSpacing;
                 Repeater {
                     model: ListModel {
-                        ListElement { text: "Dropbox"; accountType: "DropBox"; serviceName: "dropbox"; }
+                        // ListElement { text: "Dropbox"; accountType: "DropBox"; serviceName: "dropbox"; }
                         ListElement { text: "WebDav"; accountType: "WebDav"; serviceName: ""; }
                         //ListElement { text: "ownCloud"; accountType: "WebDav"; serviceName: "ownCloud"; }
                         ListElement { text: "Git"; accountType: "Git"; serviceName: ""; }
@@ -178,25 +169,25 @@ Kirigami.ScrollablePage {
 
     function addComponentFromName(name) {
         var elements = {
-            "DropBox": addDropBox,
+            // "DropBox": addDropBox,
             "WebDav": addWebDav,
             "Git": addGit
         };
         return elements[name];
     }
-    Component { id: addDropBox; AddDropbox { } }
+    // Component { id: addDropBox; AddDropbox { } }
     Component { id: addWebDav; AddWebdav { } }
     Component { id: addGit; AddGit { } }
 
     function editComponentFromName(name) {
         var elements = {
-            "DropBox": editDropBox,
+            // "DropBox": editDropBox,
             "WebDav": editWebDav,
             "Git": editGit
         };
         return elements[name];
     }
-    Component { id: editDropBox; AddDropbox { } }
+    // Component { id: editDropBox; AddDropbox { } }
     Component { id: editWebDav; EditDetailsBase { } }
     Component { id: editGit; EditGit { } }
 
@@ -204,13 +195,13 @@ Kirigami.ScrollablePage {
 
     function elementFromName(name) {
         var elements = {
-            "accountsPageDropbox": accountsPageDropbox,
+            // "accountsPageDropbox": accountsPageDropbox,
             "accountsPageWebdav": accountsPageWebdav,
             "accountsPageGit": accountsPageGit
         };
         return elements[name];
     }
-    Component { id: accountsPageDropbox; Dropbox { } }
+    // Component { id: accountsPageDropbox; Dropbox { } }
     Component { id: accountsPageWebdav; Webdav { } }
     Component { id: accountsPageGit; Git { } }
 }
