diff -Naur xarchiver.desktop.in.orig xarchiver.desktop.in
--- xarchiver.desktop.in.orig	2014-05-12 01:08:12.159131913 +0000
+++ xarchiver.desktop.in	2014-05-12 01:09:02.585526393 +0000
@@ -4,11 +4,11 @@
 _Name=Xarchiver
 _Comment=A GTK+2 only archive manager
 _GenericName=Archive manager
-Exec=xarchiver
+Exec=xarchiver %f
 Icon=xarchiver
 Terminal=false
 Type=Application
 X-MultipleArgs=false
 Categories=GTK;Archiving;Utility;
 StartupNotify=true
-MimeType=application/x-arj;application/arj;application/x-bzip;application/x-bzip-compressed-tar;application/x-gzip;application/x-rar;application/x-rar-compressed;application/x-tar;application/x-zip;application/x-zip-compressed;application/zip;multipart/x-zip;application/x-7z-compressed;application/x-compressed-tar;application/x-bzip2;application/x-bzip2-compressed-tar;
+MimeType=application/x-arj;application/arj;application/x-bzip;application/x-bzip-compressed-tar;application/x-gzip;application/x-rar;application/x-rar-compressed;application/x-tar;application/x-zip;application/x-zip-compressed;application/zip;application/x-7z-compressed;application/x-compressed-tar;application/x-bzip2;application/x-bzip2-compressed-tar;application/x-lzma;application/x-lzma-compressed-tar;application/x-lzop;application/x-lzop-compressed-tar;application/x-xz;application/x-xz-compressed-tar;application/x-deb;application/x-rpm;
