Description: Nonzero return code if tests fail
Origin: https://git.sagemath.org/sage.git/commit/build/pkgs/gfan/patches/maketestsreturnerror.patch?h=develop&id=4ad830c4cbb10ce81d49eb92cbd3b1be2df31e7b
Forwarded: yes
Last-Update: 2020-11-13

--- a/src/app_test.cpp
+++ b/src/app_test.cpp
@@ -1414,6 +1414,9 @@
 			  failed.push_back(i->folder);
 		  }
 	  cout<<"\n";
+	  cout<<"Number of successful tests "<<good<<endl;
+	  cout<<"Number of failed tests "<<bad<<endl;
+
 	  if(!failed.empty())
 	  {
 		  cout<<"Failed tests:\n-------------\n";
@@ -1421,6 +1424,7 @@
 		  {
 			  cout<<*i<<" FAILED!\n";
 		  }
+		  return 1;
 	  }
 
 	  cout<<"Number of succesful tests "<<good<<endl;
