diff --git a/src/app_test.cpp b/src/app_test.cpp
index 755bfe6..183c735 100644
--- a/src/app_test.cpp
+++ b/src/app_test.cpp
@@ -562,6 +562,9 @@ int testIntegers()
 			  failed.push_back(i->folder);
 		  }
 	  cout<<"\n";
+	  cout<<"Number of succesful tests "<<good<<endl;
+	  cout<<"Number of failed tests "<<bad<<endl;
+
 	  if(!failed.empty())
 	  {
 		  cout<<"Failed tests:\n-------------\n";
@@ -569,11 +572,9 @@ int testIntegers()
 		  {
 			  cout<<*i<<" FAILED!\n";
 		  }
+		  return 1;
 	  }
 
-	  cout<<"Number of succesful tests "<<good<<endl;
-	  cout<<"Number of failed tests "<<bad<<endl;
-
 	  return 0;
   }
 };
