--- a/src/packedmonomial.h	2024-08-12 13:00:15.000000000 -0300
+++ b/src/packedmonomial.h	2024-10-29 22:59:54.666573193 -0300
@@ -13,6 +13,9 @@
 #include "monomial.h"
 #include "printer.h"
 
+#define u_int64_t uint64_t
+#define u_int16_t uint16_t
+
 /*
  * This is an attempt to implement packed monomials in a similar way as Singular does.
  * Methods are designed for speed, not flexibility.
--- a/src/gfanlib_circuittableint.h	2024-10-29 23:01:48.248324088 -0300
+++ b/src/gfanlib_circuittableint.h	2024-10-29 23:12:51.154543137 -0300
@@ -23,6 +23,11 @@
 typedef absl::uint128 __uint128_t;
 #endif
 
+#define __int64_t int64_t
+#define __uint64_t uint64_t
+#define __int32_t int32_t
+#define __uint32_t uint32_t
+
 namespace gfan{
 
 
