$OpenBSD: patch-src_httperf_c,v 1.1 2015/07/18 17:13:49 jca Exp $
--- src/httperf.c.orig	Sat Jul 18 11:10:17 2015
+++ src/httperf.c	Sat Jul 18 11:10:41 2015
@@ -807,8 +807,7 @@ main (int argc, char **argv)
       SSL_load_error_strings ();
       SSLeay_add_ssl_algorithms ();
 
-      /* for some strange reason, SSLv23_client_method () doesn't work here */
-      ssl_ctx = SSL_CTX_new (SSLv3_client_method ());
+      ssl_ctx = SSL_CTX_new (SSLv23_client_method ());
       if (!ssl_ctx)
 	{
 	  ERR_print_errors_fp (stderr);
