--- CMakeLists.txt	2018-10-01 14:15:08.000000000 +0200
+++ -	2018-10-15 16:05:46.593091232 +0200
@@ -28,7 +28,7 @@
 
 vc_determine_compiler()
 
-if("${CMAKE_SYSTEM_PROCESSOR}" MATCHES "(x86|AMD64|amd64)")
+if("${CMAKE_SYSTEM_PROCESSOR}" MATCHES "(i686|x86|AMD64|amd64)")
    set(Vc_X86 TRUE)
 elseif("${CMAKE_SYSTEM_PROCESSOR}" MATCHES "(arm|aarch32|aarch64)")
    message(WARNING "No optimized implementation of the Vc types available for ${CMAKE_SYSTEM_PROCESSOR}")
