diff -Naur a/resources/default_schema/com.system76.CosmicTheme.Dark/v2/list_button b/resources/default_schema/com.system76.CosmicTheme.Dark/v2/list_button
--- a/resources/default_schema/com.system76.CosmicTheme.Dark/v2/list_button	1970-01-01 03:00:00.000000000 +0300
+++ b/resources/default_schema/com.system76.CosmicTheme.Dark/v2/list_button	2026-07-19 14:04:30.216805052 +0300
@@ -0,0 +1,74 @@
+(
+    base: (
+        red: 0.0,
+        green: 0.0,
+        blue: 0.0,
+        alpha: 0.0,
+    ),
+    hover: (
+        red: 0.0,
+        green: 0.0,
+        blue: 0.0,
+        alpha: 0.0,
+    ),
+    pressed: (
+        red: 0.08610422,
+        green: 0.08610421,
+        blue: 0.08610421,
+        alpha: 0.5,
+    ),
+    selected: (
+        red: 0.0,
+        green: 0.0,
+        blue: 0.0,
+        alpha: 0.0,
+    ),
+    selected_text: (
+        red: 0.3882353,
+        green: 0.8156863,
+        blue: 0.8745098,
+        alpha: 1.0,
+    ),
+    focus: (
+        red: 0.3882353,
+        green: 0.8156863,
+        blue: 0.8745098,
+        alpha: 1.0,
+    ),
+    divider: (
+        red: 0.7532969,
+        green: 0.7532969,
+        blue: 0.75329685,
+        alpha: 0.2,
+    ),
+    on: (
+        red: 0.7532969,
+        green: 0.7532969,
+        blue: 0.75329685,
+        alpha: 1.0,
+    ),
+    disabled: (
+        red: 0.0,
+        green: 0.0,
+        blue: 0.0,
+        alpha: 0.0,
+    ),
+    on_disabled: (
+        red: 0.7532969,
+        green: 0.7532969,
+        blue: 0.75329685,
+        alpha: 0.65,
+    ),
+    border: (
+        red: 0.743206,
+        green: 0.74320585,
+        blue: 0.74320585,
+        alpha: 1.0,
+    ),
+    disabled_border: (
+        red: 0.743206,
+        green: 0.74320585,
+        blue: 0.74320585,
+        alpha: 0.5,
+    ),
+)
\ No newline at end of file
diff -Naur a/resources/default_schema/com.system76.CosmicTheme.Dark.Builder/v2/palette b/resources/default_schema/com.system76.CosmicTheme.Dark.Builder/v2/palette
--- a/resources/default_schema/com.system76.CosmicTheme.Dark.Builder/v2/palette	2026-07-19 14:04:30.158032559 +0300
+++ b/resources/default_schema/com.system76.CosmicTheme.Dark.Builder/v2/palette	2026-07-19 14:04:30.179306920 +0300
@@ -1,4 +1,4 @@
-(
+Dark((
     name: "cosmic-dark",
     bright_red: "#FFA09AFF",
     bright_green: "#5EDB8CFF",
@@ -32,4 +32,4 @@
     ext_purple: "#CF7DFFFF",
     ext_pink: "#F93A83FF",
     ext_indigo: "#3E88FFFF",
-)
\ No newline at end of file
+))
diff -Naur a/resources/default_schema/com.system76.CosmicTheme.Light/v2/list_button b/resources/default_schema/com.system76.CosmicTheme.Light/v2/list_button
--- a/resources/default_schema/com.system76.CosmicTheme.Light/v2/list_button	1970-01-01 03:00:00.000000000 +0300
+++ b/resources/default_schema/com.system76.CosmicTheme.Light/v2/list_button	2026-07-19 14:04:30.217885149 +0300
@@ -0,0 +1,74 @@
+(
+    base: (
+        red: 0.0,
+        green: 0.0,
+        blue: 0.0,
+        alpha: 0.0,
+    ),
+    hover: (
+        red: 0.0,
+        green: 0.0,
+        blue: 0.0,
+        alpha: 0.0,
+    ),
+    pressed: (
+        red: 0.74320567,
+        green: 0.743206,
+        blue: 0.74320585,
+        alpha: 0.5,
+    ),
+    selected: (
+        red: 0.0,
+        green: 0.0,
+        blue: 0.0,
+        alpha: 0.0,
+    ),
+    selected_text: (
+        red: 0.0,
+        green: 0.32156864,
+        blue: 0.3529412,
+        alpha: 1.0,
+    ),
+    focus: (
+        red: 0.0,
+        green: 0.32156864,
+        blue: 0.3529412,
+        alpha: 1.0,
+    ),
+    divider: (
+        red: 0.15292808,
+        green: 0.15292814,
+        blue: 0.15292796,
+        alpha: 0.2,
+    ),
+    on: (
+        red: 0.15292808,
+        green: 0.15292814,
+        blue: 0.15292796,
+        alpha: 1.0,
+    ),
+    disabled: (
+        red: 0.0,
+        green: 0.0,
+        blue: 0.0,
+        alpha: 0.0,
+    ),
+    on_disabled: (
+        red: 0.15292808,
+        green: 0.15292814,
+        blue: 0.15292796,
+        alpha: 0.65,
+    ),
+    border: (
+        red: 0.08610404,
+        green: 0.08610427,
+        blue: 0.08610421,
+        alpha: 1.0,
+    ),
+    disabled_border: (
+        red: 0.08610404,
+        green: 0.08610427,
+        blue: 0.08610421,
+        alpha: 0.5,
+    ),
+)
\ No newline at end of file
