diff -Naur a/resources/default_schema/com.system76.CosmicSettings.Shortcuts/v1/defaults b/resources/default_schema/com.system76.CosmicSettings.Shortcuts/v1/defaults
--- a/resources/default_schema/com.system76.CosmicSettings.Shortcuts/v1/defaults	1970-01-01 03:00:00.000000000 +0300
+++ b/resources/default_schema/com.system76.CosmicSettings.Shortcuts/v1/defaults	2026-07-20 13:06:01.992767894 +0300
@@ -0,0 +1 @@
+{}
\ No newline at end of file
diff -Naur a/resources/default_schema/com.system76.CosmicSettings.WindowRules/v1/tiling_exception_defaults b/resources/default_schema/com.system76.CosmicSettings.WindowRules/v1/tiling_exception_defaults
--- a/resources/default_schema/com.system76.CosmicSettings.WindowRules/v1/tiling_exception_defaults	1970-01-01 03:00:00.000000000 +0300
+++ b/resources/default_schema/com.system76.CosmicSettings.WindowRules/v1/tiling_exception_defaults	2026-07-20 13:06:01.992767894 +0300
@@ -0,0 +1 @@
+[]
\ No newline at end of file
