From 4cf755ad73587fba904debe918ef281491cdd919 Mon Sep 17 00:00:00 2001
From: John Zimmermann <me@johnnynator.dev>
Date: Tue, 18 Feb 2025 23:57:44 +0100
Subject: [PATCH] musl fix

---
 src/gl/shim.c | 55 +++++++++++++++++++++++++++++++++++++++++++++++++--
 1 file changed, 53 insertions(+), 2 deletions(-)

diff --git a/src/gl/shim.c b/src/gl/shim.c
index 98b61e9..322af56 100644
--- a/src/gl/shim.c
+++ b/src/gl/shim.c
@@ -77,6 +77,59 @@ static bool load_adjacent_opengl_lib(void)
 
     return true;
 }
+#else
+static inline void
+free_indirect(char **p)
+{
+    free(*p);
+}
+static bool load_adjacent_opengl_lib(void)
+{
+    __attribute__((cleanup(free_indirect))) char *location = NULL;
+    __attribute__((cleanup(free_indirect))) char *lib = NULL;
+    Dl_info info = {};
+
+    // The first argument can be any symbol in this shared library,
+    // mangoHudLoaded is a convenient one
+    if (!dladdr(&mangoHudLoaded, &info))
+    {
+        fprintf(stderr, "shim: Unable to find my own location: %s\n", dlerror());
+        return false;
+    }
+
+    if (info.dli_fname == NULL)
+    {
+        fprintf(stderr, "shim: Unable to find my own location: NULL dli_fname\n");
+        return false;
+    }
+
+    location = realpath(info.dli_fname, NULL);
+    char *slash = strrchr(location, '/');
+
+    if (slash == NULL)
+    {
+        fprintf(stderr, "shim: Unable to find my own location: no directory separator\n");
+        return false;
+    }
+
+    *slash = '\0';
+
+    if (asprintf(&lib, "%s/libMangoHud_opengl.so", location) < 0)
+    {
+        fprintf(stderr, "shim: asprintf: %s\n", strerror(errno));
+        return false;
+    }
+
+    handle = dlopen(lib, RTLD_NOW | RTLD_LOCAL | RTLD_DEEPBIND);
+
+    if (handle == NULL)
+    {
+        fprintf(stderr, "shim: Failed to load from \"%s\": %s\n", lib, dlerror());
+        return false;
+    }
+
+    return true;
+}
 #endif
 
 // Load MangoHud after EGL/GLX functions have been intercepted
@@ -110,13 +163,11 @@ static void loadMangoHud() {
         }
     }
 
-#ifdef __GLIBC__
     if (load_adjacent_opengl_lib())
     {
         mangoHudLoaded = true;
         return;
     }
-#endif
 
     if (!mangoHudLoaded)
     {
-- 
2.48.1

