diff --git a/tests/python/repack.py b/tests/python/repack.py
index 0844cd2..189789e 100644
--- tests/python/repack.py
+++ tests/python/repack.py
@@ -194,7 +194,6 @@ class GUIDIndexAndPackFormatTests(TestCase):
         self.assertEqual(pack_formats, [ldb.PACKING_FORMAT_V2])
         self.assertEqual(self.get_database(), expect_db)
 
-        self.toggle_guidindex_check_pack()
 
 
 if __name__ == '__main__':
