--- a/tests/ldb_match_test.c
+++ b/tests/ldb_match_test.c
@@ -32,6 +32,8 @@
  * This allows test applications to use custom definitions of C standard
  * library functions and types.
  */
+// fails to compile on musl, so limit to glibc
+#ifdef __GLIBC__
 #include <stdarg.h>
 #include <stddef.h>
 #include <stdint.h>
@@ -189,3 +191,9 @@
 
 	return cmocka_run_group_tests(tests, NULL, NULL);
 }
+#else
+int main(int argc, const char **argv)
+{
+	return 0;
+}
+#endif
