diff --git a/deps/crypt/c_src/Makefile b/deps/crypt/c_src/Makefile
index 03d6380..bfc9f5f 100644
--- a/deps/crypt/c_src/Makefile
+++ b/deps/crypt/c_src/Makefile
@@ -24,12 +24,12 @@ else ifeq ($(UNAME_SYS), Linux)
 		CFLAGS ?= -O3 -std=c99 -finline-functions -Wall -Wmissing-prototypes
 		LDFLAGS ?= -lcrypt
   else
-		CFLAGS ?= -DHAVE_CRYPT_R -O3 -std=c99 -finline-functions -Wall -Wmissing-prototypes \
+		CFLAGS += -DHAVE_CRYPT_R -O3 -std=c99 -finline-functions -Wall -Wmissing-prototypes \
 			-pedantic -fwrapv -D_FORTIFY_SOURCE=2 \
 			-fstack-protector-strong -Wformat -Werror=format-security \
 			-fno-strict-aliasing -Wconversion -Wshadow -Wpointer-arith \
 			-Wcast-qual
-		LDFLAGS ?= -lpthread -lcrypt -Wl,-z,relro,-z,now -Wl,-z,noexecstack
+		LDFLAGS += -lpthread -lcrypt -Wl,-z,relro,-z,now -Wl,-z,noexecstack
   endif
 else ifneq (,$(wildcard /usr/lib/libcrypt.*))
 	CFLAGS ?= -O3 -std=c99 -finline-functions -Wall -Wmissing-prototypes
