--- chuck-1.3.5.1/src/makefile
+++ chuck-1.3.5.1/src/makefile
@@ -40,8 +40,6 @@
 
 ifneq ($(CHUCK_DEBUG),)
 CFLAGS+= -g
-else
-CFLAGS+= -O3
 endif
 
 ifneq ($(USE_64_BIT_SAMPLE),)
--- chuck-1.3.5.1/src/makefile.alsa
+++ chuck-1.3.5.1/src/makefile.alsa
@@ -1,4 +1,4 @@
 
-CFLAGS+= -D__LINUX_ALSA__ -D__PLATFORM_LINUX__ -O3 -fno-strict-aliasing -D__CK_SNDFILE_NATIVE__
+CFLAGS+= -D__LINUX_ALSA__ -D__PLATFORM_LINUX__ -fno-strict-aliasing -D__CK_SNDFILE_NATIVE__
 LDFLAGS+= -lasound -lstdc++ -ldl -lm -lsndfile -lpthread
 
--- chuck-1.3.5.1/src/makefile.jack
+++ chuck-1.3.5.1/src/makefile.jack
@@ -1,4 +1,4 @@
 
-CFLAGS+= -D__UNIX_JACK__ -D__PLATFORM_LINUX__ -O3 -fno-strict-aliasing -D__CK_SNDFILE_NATIVE__
+CFLAGS+= -D__UNIX_JACK__ -D__PLATFORM_LINUX__ -fno-strict-aliasing -D__CK_SNDFILE_NATIVE__
 LDFLAGS+= -lasound -ljack -lstdc++ -ldl -lm -lsndfile -lpthread
 
