In musl libc there is no GLOB_TILDE defined.

--- chuck-1.5.4.3/src/core/util_string.cpp	2024-12-03 05:52:05.000000000 +0100
+++ chuck-1.5.4.3/src/core/util_string.cpp	2024-12-07 00:11:07.698312447 +0100
@@ -48,8 +48,12 @@
 #include <time.h>
 #include <limits.h>
 #include <stdio.h>
-
 #include <algorithm>
+
+#if !defined(__GLIBC__)
+#define GLOB_TILDE 0	// unsupported GNU libc extension
+#endif
+
 using namespace std;
 
 
