diff --git a/Makefile.am b/Makefile.am
index 752a022..a99c983 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -96,7 +96,7 @@ compilerflags.h: gencompilerflags
 	./gencompilerflags > compilerflags.h
 
 gencompilerflags:
-	+$(CC) $(gencompilerflags_CFLAGS) gencompilerflags.c -o gencompilerflags
+	+$(BUILD_CC) $(gencompilerflags_CFLAGS) gencompilerflags.c -o gencompilerflags
 
 dist_man_MANS = man/man1/clsync.1
 endif
