commit df82dc59c7b3ef5fa65898cbdd459d1fb8f3c9bf
Author: q66 <daniel@octaforge.org>
Date:   Mon Feb 15 19:45:23 2021 +0100

    fix segfault under wayland

diff --git src/lxappearance.c src/lxappearance.c
index a4b7db3..b6441ae 100644
--- a/src/lxappearance.c
+++ b/src/lxappearance.c
@@ -86,6 +86,10 @@ static gboolean check_lxde_dbus()
 
 static void check_lxsession()
 {
+    /* on wayland, avoid touching X bits */
+    if (g_getenv("WAYLAND_DISPLAY"))
+        return;
+
     lxsession_atom = XInternAtom( GDK_DISPLAY_XDISPLAY(gdk_display_get_default()), "_LXSESSION", True );
     if( lxsession_atom != None )
     {
