diff --git b64.c b64.c
index 0649c1e..7788089 100644
--- a/b64.c
+++ b/b64.c
@@ -4,6 +4,7 @@
 
 #include <openssl/bio.h>
 #include <openssl/evp.h>
+#include <limits.h>
 #include <stdint.h>
 #include <string.h>
 
diff --git util.c util.c
index 3ec4e71..5989c6c 100644
--- a/util.c
+++ b/util.c
@@ -9,6 +9,7 @@
 #include <openssl/ec.h>
 #include <openssl/obj_mac.h>
 
+#include <limits.h>
 #include <stdlib.h>
 #include <fcntl.h>
 #include <sys/stat.h>
