From 72a356e12448f7d1e985075863bf20e975e9a418 Mon Sep 17 00:00:00 2001
From: John <me@johnnynator.dev>
Date: Tue, 26 Aug 2025 00:48:14 +0200
Subject: [PATCH] fix console for new callback api

---
 console/linphonec.c | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/console/linphonec.c b/console/linphonec.c
index 3c6709b26..1e2343178 100644
--- a/console/linphonec.c
+++ b/console/linphonec.c
@@ -105,7 +105,10 @@ static char **linephonec_readline_completion(const char *text, int start, int en
 
 /* These are callback for linphone core */
 static void linphonec_prompt_for_auth(LinphoneCore *lc, const char *realm, const char *username, const char *domain);
-static void linphonec_display_refer(LinphoneCore *lc, const char *refer_to);
+static void linphonec_display_refer(LinphoneCore *lc,
+                                    const LinphoneAddress *refer_to_addr,
+                                    const LinphoneHeaders *custom_headers,
+                                    const LinphoneContent *content);
 static void linphonec_transfer_state_changed(LinphoneCore *lc, LinphoneCall *call, LinphoneCallState new_call_state);
 
 static void linphonec_notify_presence_received(LinphoneCore *lc, LinphoneFriend *fid);
@@ -188,7 +191,11 @@ LinphoneCall *linphonec_get_call(int id) {
 /*
  * Linphone core callback
  */
-static void linphonec_display_refer(BCTBX_UNUSED(LinphoneCore *lc), const char *refer_to) {
+static void linphonec_display_refer(BCTBX_UNUSED(LinphoneCore *lc),
+                                    const LinphoneAddress *refer_to_addr,
+                                  	BCTBX_UNUSED(const LinphoneHeaders *custom_header),
+                                  	BCTBX_UNUSED(const LinphoneContent *content)) {
+	char * refer_to = linphone_address_as_string(refer_to_addr);
 	linphonec_out("Receiving out of call refer to %s\n", refer_to);
 }
 
-- 
2.50.1

