--- a/CMakeLists.txt	2024-06-27 15:40:03.000000000 +0200
+++ -	2024-07-03 16:56:27.923999881 +0200
@@ -418,7 +418,7 @@
 endif()
 
 include(CMakePackageConfigHelpers)
-set(CMAKE_MODULES_INSTALL_DIR "${CMAKE_INSTALL_DATADIR}/${PROJECT_NAME}/cmake")
+set(CMAKE_MODULES_INSTALL_DIR "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}/")
 configure_package_config_file("cmake/${PROJECT_NAME}Config.cmake.in" "${PROJECT_BINARY_DIR}/${PROJECT_NAME}Config.cmake"
 	INSTALL_DESTINATION "${CMAKE_MODULES_INSTALL_DIR}"
 	PATH_VARS LIBLINPHONE_PLUGINS_DIR
--- a/wrappers/cpp/CMakeLists.txt	2024-06-27 15:40:03.000000000 +0200
+++ -	2024-07-03 17:47:42.583013104 +0200
@@ -100,12 +100,12 @@
 
 install(EXPORT LinphoneCxxTargets
 	FILE LinphoneCxxTargets.cmake
-	DESTINATION "${CMAKE_INSTALL_DATADIR}/LinphoneCxx/cmake"
+	DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/LinphoneCxx"
 )
 install(FILES
 	"${CMAKE_CURRENT_BINARY_DIR}/LinphoneCxxConfig.cmake"
 	"${CMAKE_CURRENT_BINARY_DIR}/LinphoneCxxVersion.cmake"
-	DESTINATION "${CMAKE_INSTALL_DATADIR}/LinphoneCxx/cmake"
+	DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/LinphoneCxx"
 )
 
 if(ENABLE_DOC OR ENABLE_CXX_WRAPPER)
--- a/cmake/LibLinphoneConfig.cmake.in	2024-07-03 17:48:12.683720688 +0200
+++ -	2024-07-03 17:48:16.620948799 +0200
@@ -44,7 +44,7 @@
 include("${CMAKE_CURRENT_LIST_DIR}/LibLinphoneTargets.cmake")
 
 set(LibLinphone_TARGET liblinphone)
-set_and_check(LibLinphone_PLUGINS_DIR "@PACKAGE_LIBLINPHONE_PLUGINS_DIR@")
+set(LibLinphone_PLUGINS_DIR "@PACKAGE_LIBLINPHONE_PLUGINS_DIR@")
 
 # We must propagate the public dependencies and the private dependencies for static build
 include(CMakeFindDependencyMacro)
