--- linphone-5.4.39/cmake/FindJsonCPP.cmake.bak	2025-08-26 00:05:04.390781078 +0200
+++ linphone-5.4.39/cmake/FindJsonCPP.cmake	2025-08-26 00:05:06.024747302 +0200
@@ -72,13 +72,9 @@
 		list(APPEND _OPTIONS "EXACT")
 	endif()
 
-	find_package(jsoncpp ${_OPTIONS})
-
-	if(TARGET jsoncpp_lib)
-		set(JsonCPP_TARGET jsoncpp_lib)
-	else()
-		set(JsonCPP_TARGET jsoncpp_static)
-	endif()
+	include(FindPkgConfig)
+	pkg_check_modules(jsoncpp REQUIRED IMPORTED_TARGET jsoncpp)
+	set(JsonCPP_TARGET PkgConfig::jsoncpp)
 
 	set(JsonCPP_FOUND ${jsoncpp_FOUND})
 endif()
--- a/cmake/LibLinphoneConfig.cmake.in	2025-08-21 15:01:39.000000000 +0200
+++ -	2025-08-26 01:22:17.847416047 +0200
@@ -48,6 +48,7 @@
 
 # We must propagate the public dependencies and the private dependencies for static build
 include(CMakeFindDependencyMacro)
+include(FindPkgConfig)
 find_dependency(BelleSIP)
 find_dependency(Mediastreamer2)
 find_dependency(Ortp)
@@ -60,6 +60,7 @@
 if(@Lime_FOUND@)
 	find_dependency(Lime)
 endif()
+pkg_check_modules(jsoncpp REQUIRED IMPORTED_TARGET jsoncpp)
 if(@BUILD_SHARED_LIBS@)
 else()
 	list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}")
