--- kdm/kfrontend/genkdmconf.c~	2011-10-04 22:11:24.787771512 +0000
+++ kdm/kfrontend/genkdmconf.c	2011-10-04 22:13:58.612702824 +0000
@@ -742,6 +742,15 @@
 "[ -f /etc/xprofile ] && . /etc/xprofile\n"
 "[ -f $HOME/.xprofile ] && . $HOME/.xprofile\n"
 "\n"
+"# run all system xinitrc shell scripts.\n"
+"if [ -d /etc/X11/xinit/xinitrc.d ]; then\n"
+"  for i in /etc/X11/xinit/xinitrc.d/* ; do\n"
+"  if [ -x \"$i\" ]; then\n"
+"    . \"$i\"\n"
+"  fi\n"
+"  done\n"
+"fi\n"
+"\n"
 "if [ -d /etc/X11/Xresources ]; then\n"
 "  for i in /etc/X11/Xresources/*; do\n"
 "    [ -f $i ] && xrdb -merge $i\n"
