--- khotkeys/kcm_hotkeys/actions/dbus_action_widget.cpp~	2014-03-05 18:41:23.126818552 +0000
+++ khotkeys/kcm_hotkeys/actions/dbus_action_widget.cpp	2014-03-05 18:42:05.563172036 +0000
@@ -117,7 +117,7 @@
 
 void DbusActionWidget::launchDbusBrowser() const
     {
-    if( KRun::runCommand( "qdbusviewer", window()) == 0 )
+    if( KRun::runCommand( "qdbusviewer-qt4", window()) == 0 )
         {
         KMessageBox::sorry( window(), i18n( "Failed to run qdbusviewer" ));
         }
--- khotkeys/libkhotkeysprivate/actions/dbus_action.cpp~	2014-03-05 18:41:29.120107842 +0000
+++ khotkeys/libkhotkeysprivate/actions/dbus_action.cpp	2014-03-05 18:41:52.349936188 +0000
@@ -133,7 +133,7 @@
         }
     kDebug() << "D-Bus call:" << _application << ":" << _object << ":" << _function << ":" << args_list;
     KProcess proc;
-    proc << "qdbus" << _application << _object << _function << args_list;
+    proc << "qdbus-qt4" << _application << _object << _function << args_list;
     proc.startDetached();
     }
 
