diff --git configure configure
index 4cb3d5b..a9db926 100755
--- configure
+++ configure
@@ -6159,7 +6159,7 @@ if test "x$ac_cv_func_strlcpy" = xyes; then :
 
 $as_echo "#define HAVE_SYSTEM_STRLCPY /**/" >>confdefs.h
 
-		if test "$cross_compiling" = yes; then :
+		if test "$cross_compiling" = xxx; then :
   { { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
 $as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
 as_fn_error $? "cannot run test program while cross compiling
@@ -6238,7 +6238,7 @@ if test "x$ac_cv_func_strlcat" = xyes; then :
 
 $as_echo "#define HAVE_SYSTEM_STRLCAT /**/" >>confdefs.h
 
-		if test "$cross_compiling" = yes; then :
+		if test "$cross_compiling" = xxx; then :
   { { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
 $as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
 as_fn_error $? "cannot run test program while cross compiling
