diff -rup recode-3.7.15/src/task.c.orig recode-3.7.15/src/task.c
--- recode-3.7.15/src/task.c.orig	2025-03-20 17:10:53.973167052 -0700
+++ recode-3.7.15/src/task.c	2025-03-20 17:11:46.080614094 -0700
@@ -432,6 +432,7 @@ recode_perform_task (RECODE_TASK task)
   /* Final clean up.  */
  exit:
 
+  if (subtask->output.file) fflush (subtask->output.file);
   if (subtask->input.file && subtask->input.file != task->input.file && fclose (subtask->input.file) != 0)
     {
       recode_perror (NULL, "fclose (%s)", subtask->input.name ? subtask->input.name : "stdin");
