diff --git plugins/signal-protocol/CMakeLists.txt plugins/signal-protocol/CMakeLists.txt
index 0ae8411..a687c79 100644
--- plugins/signal-protocol/CMakeLists.txt
+++ plugins/signal-protocol/CMakeLists.txt
@@ -32,7 +32,7 @@ if(NOT BUILD_LIBSIGNAL_IN_TREE)
     # libsignal-protocol-c has a history of breaking compatibility on the patch level
     # we'll have to check compatibility for every new release
     # distro maintainers may update this dependency after compatibility tests
-    find_package(SignalProtocol 2.3.2 EXACT REQUIRED)
+    find_package(SignalProtocol 2.3.2 REQUIRED)
 else()
     add_subdirectory(libsignal-protocol-c EXCLUDE_FROM_ALL)
     set_property(TARGET curve25519 PROPERTY POSITION_INDEPENDENT_CODE ON)
