From 1185fed36da8aaa2e06a8229ea308a63fe0a5c75 Mon Sep 17 00:00:00 2001
From: oreo639 <oreo6391@gmail.com>
Date: Sat, 24 Feb 2024 19:04:38 -0800
Subject: [PATCH] allow building with webkit2gtk-4.1

---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 252cace..e6f72c5 100644
--- a/configure.ac
+++ b/configure.ac
@@ -34,7 +34,7 @@ AC_ARG_ENABLE([html],
 			[Build YAD with HTML widget])],
 	[build_html=$enableval], [build_html=yes])
 if test x$build_html = xyes; then
-    PKG_CHECK_MODULES([HTML], [webkit2gtk-4.0], [have_html=yes], [have_html=no])
+    PKG_CHECK_MODULES([HTML], [webkit2gtk-4.1], [have_html=yes], [PKG_CHECK_MODULES([HTML], [webkit2gtk-4.0], [have_html=yes], [have_html=no])])
 else
     have_html=no
 fi
