From a3f22386938bd54214011b2b711b55b5fa277698 Mon Sep 17 00:00:00 2001
From: Nathan Owens <ndowens@artixlinux.org>
Date: Mon, 14 Dec 2020 10:18:27 -0600
Subject: [PATCH] Fix cross

---
 configure | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git configure configure
index 3faf65b..ddbdb2f 100755
--- configure
+++ configure
@@ -13359,7 +13359,7 @@ $as_echo "#define _GNU_SOURCE 1" >>confdefs.h
 
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for GNU extensions of strptime()" >&5
 $as_echo_n "checking for GNU extensions of strptime()... " >&6; }
-if test "$cross_compiling" = yes; then :
+if test "$cross_compiling" = xyes; then :
   { { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
 $as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
 as_fn_error $? "cannot run test program while cross compiling
@@ -13395,7 +13395,7 @@ fi
 
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking size of time_t type" >&5
 $as_echo_n "checking size of time_t type... " >&6; }
-if test "$cross_compiling" = yes; then :
+if test "$cross_compiling" = xyes; then :
   { { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
 $as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
 as_fn_error $? "cannot run test program while cross compiling
-- 
2.29.2

