From da9a3a95cfe9965ff6e15e3a0e1a98f8840a34d7 Mon Sep 17 00:00:00 2001
From: Alexander Gehrke <void@qwertyuiop.de>
Date: Tue, 25 May 2021 23:47:51 +0000
Subject: [PATCH] Fix missing header for musl

---
 src/utils/client-lock.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/utils/client-lock.cpp b/src/utils/client-lock.cpp
index fb1e67b..27c6c6a 100644
--- a/src/utils/client-lock.cpp
+++ b/src/utils/client-lock.cpp
@@ -19,6 +19,7 @@
 
 #include <sys/file.h>
 #include <unistd.h>
+#include <fcntl.h>
 
 #include <exception>
 #include <stdexcept>
-- 
2.31.1

