--- lib/matplotlib/__init__.py.orig
+++ lib/matplotlib/__init__.py
@@ -681,6 +681,8 @@
         if path.is_dir():
             return str(path)
 
+    return '/usr/share/matplotlib/mpl-data'
+
     raise RuntimeError('Could not find the matplotlib data files')
 
 
@@ -742,6 +744,7 @@
             yield os.path.join(matplotlibrc, 'matplotlibrc')
         yield os.path.join(get_configdir(), 'matplotlibrc')
         yield os.path.join(get_data_path(), 'matplotlibrc')
+        yield '/etc/matplotlibrc'
 
     for fname in gen_candidates():
         if os.path.exists(fname):
