--- a/src/org/hsqldb/jdbc/jdbcBlob.java
+++ b/src/org/hsqldb/jdbc/jdbcBlob.java
@@ -462,7 +462,6 @@ public class jdbcBlob implements Blob {
     }
 
 //#ifdef JAVA6
-/*
     public void free() throws SQLException
     {
         throw new UnsupportedOperationException("Not supported yet.");
@@ -472,7 +471,6 @@ public class jdbcBlob implements Blob {
     {
         throw new UnsupportedOperationException("Not supported yet.");
     }
-*/
 
 //#endif JAVA6
 
--- a/src/org/hsqldb/jdbc/jdbcCallableStatement.java
+++ b/src/org/hsqldb/jdbc/jdbcCallableStatement.java
@@ -37,11 +37,9 @@ import java.math.BigDecimal;
 import java.sql.CallableStatement;
 import java.sql.Date;
 //#ifdef JAVA6
-/*
 import java.sql.NClob;
 import java.sql.RowId;
 import java.sql.SQLXML;
-*/
 //#endif JAVA6
 import java.sql.Time;
 import java.sql.Timestamp;
@@ -3106,7 +3104,6 @@ implements CallableStatement {
     
 //#endif JAVA4
 //#ifdef JAVA6
-/*
     public void setPoolable(boolean poolable) throws SQLException
     {
         throw new UnsupportedOperationException("Not supported yet.");
@@ -3376,7 +3373,6 @@ implements CallableStatement {
     {
         throw new UnsupportedOperationException("Not supported yet.");
     }
-*/
 
 //#endif JAVA6
 }
--- a/src/org/hsqldb/jdbc/jdbcClob.java
+++ b/src/org/hsqldb/jdbc/jdbcClob.java
@@ -464,7 +464,6 @@ public final class jdbcClob implements C
     }
 
 //#ifdef JAVA6
-/*
     public void free() throws SQLException
     {
         throw new UnsupportedOperationException("Not supported yet.");
@@ -474,6 +473,5 @@ public final class jdbcClob implements C
     {
         throw new UnsupportedOperationException("Not supported yet.");
     }
-*/
 //#endif JAVA6
 }
--- a/src/org/hsqldb/jdbc/jdbcConnection.java
+++ b/src/org/hsqldb/jdbc/jdbcConnection.java
@@ -43,11 +43,9 @@ import java.sql.Connection;
 import java.sql.DatabaseMetaData;
 
 //#ifdef JAVA6
-/*
 import java.sql.NClob;
 import java.sql.SQLClientInfoException;
 import java.sql.SQLXML;
-*/
 
 //#endif JAVA6
 import java.sql.PreparedStatement;
@@ -2729,7 +2727,6 @@ public class jdbcConnection implements C
     }
 
 //#ifdef JAVA6
-/*
     public Clob createClob() throws SQLException
     {
         throw new UnsupportedOperationException("Not supported yet.");
@@ -2794,7 +2791,6 @@ public class jdbcConnection implements C
     {
         throw new UnsupportedOperationException("Not supported yet.");
     }
-*/
 
 //#endif JAVA6
 }
--- a/src/org/hsqldb/jdbc/jdbcDataSource.java
+++ b/src/org/hsqldb/jdbc/jdbcDataSource.java
@@ -312,7 +312,6 @@ public class jdbcDataSource implements S
     }
 
 //#ifdef JAVA6
-/*
     public <T> T unwrap(Class<T> iface) throws SQLException
     {
         throw new UnsupportedOperationException("Not supported yet.");
@@ -322,7 +321,6 @@ public class jdbcDataSource implements S
     {
         throw new UnsupportedOperationException("Not supported yet.");
     }
-*/
 
 //#endif JAVA6
 }
--- a/src/org/hsqldb/jdbc/jdbcDatabaseMetaData.java
+++ b/src/org/hsqldb/jdbc/jdbcDatabaseMetaData.java
@@ -37,9 +37,7 @@ import java.sql.ResultSet;
 import java.sql.SQLException;
 
 //#ifdef JAVA6
-/*
 import java.sql.RowIdLifetime;
-*/
 
 //#endif JAVA6
 import org.hsqldb.Column;
@@ -5650,7 +5648,6 @@ public class jdbcDatabaseMetaData implem
     }
 
 //#ifdef JAVA6
-/*
     public RowIdLifetime getRowIdLifetime() throws SQLException
     {
         throw new UnsupportedOperationException("Not supported yet.");
@@ -5694,7 +5691,6 @@ public class jdbcDatabaseMetaData implem
     {
         throw new UnsupportedOperationException("Not supported yet.");
     }
-*/
 
 //#endif JAVA6
 }
--- a/src/org/hsqldb/jdbc/jdbcParameterMetaData.java
+++ b/src/org/hsqldb/jdbc/jdbcParameterMetaData.java
@@ -383,7 +383,6 @@ public class jdbcParameterMetaData imple
     }
 
 //#ifdef JAVA6
-/*
     public <T> T unwrap(Class<T> iface) throws SQLException
     {
         throw new UnsupportedOperationException("Not supported yet.");
@@ -393,7 +392,6 @@ public class jdbcParameterMetaData imple
     {
         throw new UnsupportedOperationException("Not supported yet.");
     }
-*/
 
 //#endif JAVA6
 }
--- a/src/org/hsqldb/jdbc/jdbcPreparedStatement.java
+++ b/src/org/hsqldb/jdbc/jdbcPreparedStatement.java
@@ -39,11 +39,9 @@ import java.math.BigDecimal;
 import java.sql.Date;
 
 //#ifdef JAVA6
-/*
 import java.sql.NClob;
 import java.sql.RowId;
 import java.sql.SQLXML;
-*/
 
 //#endif JAVA6
 import java.sql.PreparedStatement;
@@ -2293,7 +2291,6 @@ implements PreparedStatement {
     }
 
 //#ifdef JAVA6
-/*
     public void setPoolable(boolean poolable) throws SQLException
     {
         throw new UnsupportedOperationException("Not supported yet.");
@@ -2403,7 +2400,6 @@ implements PreparedStatement {
     {
         throw new UnsupportedOperationException("Not supported yet.");
     }
-*/
 
 //#endif JAVA6
 }
--- a/src/org/hsqldb/jdbc/jdbcResultSet.java
+++ b/src/org/hsqldb/jdbc/jdbcResultSet.java
@@ -76,11 +76,9 @@ import java.sql.ResultSet;
 import java.sql.ResultSetMetaData;
 
 //#ifdef JAVA6
-/*
 import java.sql.RowId;
 import java.sql.NClob;
 import java.sql.SQLXML;
-*/
 
 //#endif JAVA6
 import java.sql.SQLException;
@@ -5075,7 +5073,6 @@ public class jdbcResultSet implements Re
     }
 
 //#ifdef JAVA6
-/*
     public RowId getRowId(int columnIndex) throws SQLException
     {
         throw new UnsupportedOperationException("Not supported yet.");
@@ -5325,7 +5322,6 @@ public class jdbcResultSet implements Re
     {
         throw new UnsupportedOperationException("Not supported yet.");
     }
-*/
 
 //#endif JAVA6
 }
--- a/src/org/hsqldb/jdbc/jdbcResultSetMetaData.java
+++ b/src/org/hsqldb/jdbc/jdbcResultSetMetaData.java
@@ -1145,7 +1145,6 @@ public class jdbcResultSetMetaData imple
         }
     }
 //#ifdef JAVA6
-/*
     public <T> T unwrap(Class<T> iface) throws SQLException
     {
         throw new UnsupportedOperationException("Not supported yet.");
@@ -1155,6 +1154,5 @@ public class jdbcResultSetMetaData imple
     {
         throw new UnsupportedOperationException("Not supported yet.");
     }
-*/
 //#endif JAVA6
 }
--- a/src/org/hsqldb/jdbc/jdbcStatement.java
+++ b/src/org/hsqldb/jdbc/jdbcStatement.java
@@ -1588,7 +1588,6 @@ public class jdbcStatement implements St
         }
     }
 //#ifdef JAVA6
-/*
     public void setPoolable(boolean poolable) throws SQLException
     {
         throw new UnsupportedOperationException("Not supported yet.");
@@ -1608,6 +1607,5 @@ public class jdbcStatement implements St
     {
         throw new UnsupportedOperationException("Not supported yet.");
     }
-*/
 //#endif JAVA6
 }
