From 1d3f2ed0606cc971513dab5932ec7d1dd2a15f90 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fridrich=20=C5=A0trba?= <fridrich.strba@bluewin.ch>
Date: Thu, 12 Jul 2018 11:57:07 +0200
Subject: call System.runFinalizersOnExit by reflection, since it was removed
 in jdk11

--- a/src/org/hsqldb/lib/java/JavaSystem.java	2008-03-16 23:51:35.000000000 +0100
+++ b/src/org/hsqldb/lib/java/JavaSystem.java	2018-07-12 11:46:57.997837180 +0200
@@ -160,8 +160,9 @@
     public static void runFinalizers() {
 
 //#ifdef JAVA2FULL
-        System.runFinalizersOnExit(true);
-
+        try {
+            System.class.getMethod("runFinalizersOnExit", boolean.class).invoke(null, true);
+        } catch (Exception e) {}
 //#endif
     }
 
-- 
cgit v1.1
