--- src/elo.c.orig	2009-03-31 10:16:32.517051719 +0200
+++ src/elo.c	2009-03-31 10:16:49.296056335 +0200
@@ -309,8 +309,7 @@ DeviceInit( DeviceIntPtr dev )
 	 * Device reports motions on 2 axes in absolute coordinates.
 	 * Device may reports touch pressure on the 3rd axis.
 	 */
-	if (InitValuatorClassDeviceStruct (dev, priv->axes, xf86GetMotionEvents,
-									local->history_size, Absolute) == FALSE)
+	if (InitValuatorClassDeviceStruct (dev, priv->axes, local->history_size, Absolute) == FALSE)
 	{
 		ErrorF ("Unable to allocate Elographics touchscreen ValuatorClassDeviceStruct\n");
 		return !Success;
