--- src/common/pipe.h	2017-08-03 21:42:22.098586810 +0200
+++ src/common/pipe.h	2017-12-16 20:27:26.546173345 +0100
@@ -18,6 +18,7 @@
 #ifndef LTTNG_PIPE_H
 #define LTTNG_PIPE_H
 
+#include <sys/types.h>
 #include <pthread.h>
 #include <common/macros.h>
 
--- tests/regression/ust/ust-dl/prog.c	2017-08-03 21:42:22.130585910 +0200
+++ tests/regression/ust/ust-dl/prog.c	2017-12-16 20:33:52.906200900 +0100
@@ -13,6 +13,10 @@
  */
 int main(int argc, char **argv)
 {
+/*
+ * Note: musl libc does not support dlmopen(3) so make this fail
+ */
+#if defined(__GLIBC__)
 	void *h0, *h1, *h2, *h3, *h4;
 	char *error;
 	int (*foo)(void);
@@ -67,6 +71,9 @@
 get_error:
 	error = dlerror();
 error:
+#else	/* defined(__GLIBC__) */
+	const char* error = "dlmopen(3) not supported";
+#endif	/* defined(__GLIBC__) */
 	fprintf(stderr, "%s\n", error);
 	exit(EXIT_FAILURE);
 }
