--- lib/Widgets/SeekBar.vala
+++ lib/Widgets/SeekBar.vala
@@ -26,7 +26,7 @@
 
 public class Granite.SeekBar : Gtk.Grid {
     private double _playback_duration;
-    private double _playback_progress;
+    private double _playback_progress = 0.0;
 
     /*
      * The time of the full duration of the playback.
@@ -68,7 +68,6 @@ public class Granite.SeekBar : Gtk.Grid {
             scale.set_value (progress);
             progression_label.label = DateTime.seconds_to_time ((int) (progress * playback_duration));
         }
-        default = 0.0;
     }
 
     /*
--- lib/Widgets/StorageBar.vala
+++ lib/Widgets/StorageBar.vala
@@ -83,7 +83,7 @@ public class Granite.Widgets.StorageBar : Gtk.Box {
         }
     }
 
-    private uint64 _total_usage;
+    private uint64 _total_usage = 0;
 
     public uint64 total_usage {
         get {
@@ -94,8 +94,6 @@ public class Granite.Widgets.StorageBar : Gtk.Box {
             _total_usage = uint64.min (value, storage);
             update_size_description ();
         }
-
-        default = 0;
     }
 
     public int inner_margin_sides {

