Revert of a316f8ab8cf79895062c1d1fcc5d47c3d2cc6f6a

--- b/apps/nc/Makefile.am
+++ a/apps/nc/Makefile.am
@@ -1,8 +1,6 @@
 include $(top_srcdir)/Makefile.am.common
 
 -include $(abs_top_builddir)/crypto/libcrypto_la_objects.mk
--include $(abs_top_builddir)/ssl/libssl_la_objects.mk
--include $(abs_top_builddir)/tls/libtls_la_objects.mk
 
 if BUILD_NC
 
@@ -16,13 +14,11 @@
 EXTRA_DIST = nc.1
 EXTRA_DIST += CMakeLists.txt
 
+nc_LDADD = $(abs_top_builddir)/tls/libtls.la
+nc_LDADD += $(PLATFORM_LDADD) $(PROG_LDADD)
+
-nc_LDADD = $(libcrypto_la_objects)
 nc_LDADD += $(libcompat_la_objects)
 nc_LDADD += $(libcompatnoopt_la_objects)
-nc_LDADD += $(libssl_la_objects)
-nc_LDADD += $(libtls_la_objects)
-
-nc_LDADD += $(PLATFORM_LDADD) $(PROG_LDADD)
 
 AM_CPPFLAGS += -I$(top_srcdir)/apps/nc/compat
 
--- b/apps/ocspcheck/Makefile.am
+++ a/apps/ocspcheck/Makefile.am
@@ -1,9 +1,5 @@
 include $(top_srcdir)/Makefile.am.common
 
--include $(abs_top_builddir)/crypto/libcrypto_la_objects.mk
--include $(abs_top_builddir)/ssl/libssl_la_objects.mk
--include $(abs_top_builddir)/tls/libtls_la_objects.mk
-
 if !ENABLE_LIBTLS_ONLY
 bin_PROGRAMS = ocspcheck
 dist_man_MANS = ocspcheck.8
@@ -14,11 +10,9 @@
 EXTRA_DIST = ocspcheck.8
 EXTRA_DIST += CMakeLists.txt
 
+ocspcheck_LDADD = $(abs_top_builddir)/crypto/libcrypto.la
+ocspcheck_LDADD += $(abs_top_builddir)/ssl/libssl.la
+ocspcheck_LDADD += $(abs_top_builddir)/tls/libtls.la
-ocspcheck_LDADD = $(libcrypto_la_objects)
-ocspcheck_LDADD += $(libcompat_la_objects)
-ocspcheck_LDADD += $(libcompatnoopt_la_objects)
-ocspcheck_LDADD += $(libssl_la_objects)
-ocspcheck_LDADD += $(libtls_la_objects)
 ocspcheck_LDADD += $(PLATFORM_LDADD) $(PROG_LDADD)
 
 ocspcheck_SOURCES = http.c
--- b/apps/openssl/Makefile.am
+++ a/apps/openssl/Makefile.am
@@ -1,8 +1,5 @@
 include $(top_srcdir)/Makefile.am.common
 
--include $(abs_top_builddir)/crypto/libcrypto_la_objects.mk
--include $(abs_top_builddir)/ssl/libssl_la_objects.mk
-
 if !ENABLE_LIBTLS_ONLY
 bin_PROGRAMS = openssl
 dist_man_MANS = openssl.1
@@ -10,10 +7,8 @@
 noinst_PROGRAMS = openssl
 endif
 
+openssl_LDADD = $(abs_top_builddir)/ssl/libssl.la
+openssl_LDADD += $(abs_top_builddir)/crypto/libcrypto.la
-openssl_LDADD = $(libcrypto_la_objects)
-openssl_LDADD += $(libcompat_la_objects)
-openssl_LDADD += $(libcompatnoopt_la_objects)
-openssl_LDADD += $(libssl_la_objects)
 openssl_LDADD += $(PLATFORM_LDADD) $(PROG_LDADD)
 
 openssl_SOURCES = apps.c
