Index: libressl-3.2.4/apps/nc/nc.1
===================================================================
--- libressl-3.2.4.orig/apps/nc/nc.1
+++ libressl-3.2.4/apps/nc/nc.1
@@ -219,7 +219,7 @@ Cannot be used together with
 Load the root CA bundle for TLS certificate verification from
 .Ar CAfile ,
 in PEM format, instead of
-.Pa /etc/ssl/cert.pem .
+.Pa /etc/ssl/certs.pem .
 Requires
 .Fl c .
 .It Fl r
Index: libressl-3.2.4/tls/Makefile.am
===================================================================
--- libressl-3.2.4.orig/tls/Makefile.am
+++ libressl-3.2.4/tls/Makefile.am
@@ -18,9 +18,9 @@ libtls_la_LIBADD += $(PLATFORM_LDADD)
 
 libtls_la_CPPFLAGS = $(AM_CPPFLAGS)
 if OPENSSLDIR_DEFINED
-libtls_la_CPPFLAGS += -DTLS_DEFAULT_CA_FILE=\"@OPENSSLDIR@/cert.pem\"
+libtls_la_CPPFLAGS += -DTLS_DEFAULT_CA_FILE=\"@OPENSSLDIR@/certs.pem\"
 else
-libtls_la_CPPFLAGS += -DTLS_DEFAULT_CA_FILE=\"$(sysconfdir)/ssl/cert.pem\"
+libtls_la_CPPFLAGS += -DTLS_DEFAULT_CA_FILE=\"$(sysconfdir)/ssl/certs.pem\"
 endif
 
 libtls_la_SOURCES = tls.c
