--- a/CMakeModules/Translations.cmake
+++ b/CMakeModules/Translations.cmake
@@ -12,7 +12,7 @@
     set(do_translate FALSE)
 endif()
 
-find_package(Qt${QT_VERSION_MAJOR} COMPONENTS LinguistTools)
+find_package(Qt${QT_VERSION_MAJOR} COMPONENTS Linguist)
 if (${do_translate} AND NOT TARGET Qt${QT_VERSION_MAJOR}::lrelease)
     set(do_translate FALSE)
     message(STATUS "Built-in translations disabled becuase Qt lrelease tool was not found."
@@ -81,7 +81,7 @@
     set(HAVE_QRC_TRANSLATIONS TRUE)
 
     add_custom_target(ts)
-    
+
     foreach(lang ${LANGUAGES})
         add_custom_target(
             ts_${lang}
