--- a/src/FDM/JSBSim/JSBSim.cpp	2017-03-01 21:06:31.000000000 +0100
+++ b/src/FDM/JSBSim/JSBSim.cpp	2017-03-17 09:49:13.115544107 +0100
@@ -287,7 +287,7 @@
   _clearfp();
   _controlfp(_controlfp(0, 0) & ~(_EM_INVALID | _EM_ZERODIVIDE | _EM_OVERFLOW),
            _MCW_EM);
-#elif defined(__GNUC__) && !defined(sgi) && !defined(__APPLE__)
+#elif defined(__GNUC__) && !defined(sgi) && !defined(__APPLE__) && defined(__GLIBC__)
   feenableexcept(FE_DIVBYZERO | FE_INVALID);
 #endif
 
