--- a/src/Main/bootstrap.cxx
+++ b/src/Main/bootstrap.cxx
@@ -192,13 +192,15 @@
 #endif

 #if defined(__GNUC__)
+#if defined(__GLIBC__)
 #include <execinfo.h>
+#endif
 #include <cxxabi.h>
 void segfault_handler(int signo) {

   fprintf(stderr, "Error: caught signal %d:\n", signo);

-  #ifndef __OpenBSD__
+  #if defined(__GLIBC__)
   void *array[128];
   size_t size;
   size = backtrace(array, 128);
