--- a/src/Input/FGLinuxEventInput.cxx	2020-06-26 00:08:15.000000000 +0200
+++ b/src/Input/FGLinuxEventInput.cxx	2020-12-07 07:51:45.332825252 +0100
@@ -415,8 +415,8 @@
   evt.type=typeCode.type;
   evt.code = typeCode.code;
   evt.value = (long)value;
-  evt.time.tv_sec = 0;
-  evt.time.tv_usec = 0;
+  evt.input_event_sec = 0;
+  evt.input_event_usec = 0;
   size_t bytes_written = write(fd, &evt, sizeof(evt));
 
   if( bytes_written == sizeof(evt) )
