--- bin/named/include/named/seccomp.h.orig
+++ bin/named/include/named/seccomp.h
@@ -58,6 +58,7 @@
 	SCMP_SYS(fstat),
 	SCMP_SYS(epoll_ctl),
 	SCMP_SYS(gettimeofday),
+	SCMP_SYS(getrandom),
 	SCMP_SYS(unlink),
 	SCMP_SYS(socket),
 	SCMP_SYS(sendto),
@@ -121,6 +122,7 @@
 	"fstat",
 	"epoll_ctl",
 	"gettimeofday",
+	"getrandom",
 	"unlink",
 	"socket",
 	"sendto",
@@ -190,6 +192,7 @@
 	SCMP_SYS(fstat64),
 	SCMP_SYS(epoll_ctl),
 	SCMP_SYS(gettimeofday),
+	SCMP_SYS(getrandom),
 	SCMP_SYS(unlink),
 #ifndef ISC_PLATFORM_USETHREADS
 	SCMP_SYS(fcntl64),
@@ -226,6 +229,7 @@
 	"fstat64",
 	"epoll_ctl",
 	"gettimeofday",
+	"getrandom",
 	"unlink",
 #ifndef ISC_PLATFORM_USETHREADS
 	"fcntl64",
