--- a/fs/src/Makevars
+++ b/fs/src/Makevars
@@ -2,7 +2,7 @@ UNAME := $(shell uname)
 
 LIBUV := libuv-1.44.2
 
-PKG_LIBS = ./$(LIBUV)/.libs/libuv.pa -pthread
+PKG_LIBS = -luv -pthread
 
 SOURCES = $(wildcard *.cc unix/*.cc)
 OBJECTS = $(SOURCES:.cc=.o)
@@ -24,11 +24,11 @@ ifneq ($(filter $(UNAME), Linux AIX OS40
 OBJECTS +=  bsd/setmode.o bsd/strmode.o bsd/reallocarray.o
 endif
 
-PKG_CPPFLAGS = -I./$(LIBUV)/include -I. -pthread
+PKG_CPPFLAGS = -I. -pthread
 
 all: $(SHLIB)
 
-$(SHLIB): $(LIBUV)/.libs/libuv.pa
+$(SHLIB):
 
 # Avoid re-running autoconf/automake/aclocal.
 # Need the timestamps in the right order.
