--- configure
+++ configure
@@ -5669,7 +5669,7 @@ if test "${with_xmlrpc_dir+set}" = set; then :
 fi
 
 
-if test -n $XMLRPCDIR ; then
+if test -n "$XMLRPCDIR" ; then
     CFLAGS="-I$XMLRPCDIR/include -I$XMLRPCDIR/include/xmlrpcpp $CFLAGS"
     CXXFLAGS="-I$XMLRPCDIR/include -I$XMLRPCDIR/include/xmlrpcpp $CXXFLAGS"
     CPPFLAGS="-I$XMLRPCDIR/include -I$XMLRPCDIR/include/xmlrpcpp $CPPFLAGS"
-- 
2.21.0

