*** configure.ac	2020-05-27 04:58:09.858735786 -0300
--- configure.ac	2020-05-27 04:53:37.731383580 -0300
***************
*** 964,970 ****
  *** Detecting GNU Guile...
  ])
  
!    AC_PATH_PROG([guile_snarf], [guile-snarf])
     if test "x$guile_snarf" = "x"; then
        AC_MSG_WARN([`guile-snarf' from Guile not found.  Guile bindings not built.])
        opt_guile_bindings=no
--- 964,970 ----
  *** Detecting GNU Guile...
  ])
  
!    AC_PATH_PROG([guile_snarf], [guile-snarf3])
     if test "x$guile_snarf" = "x"; then
        AC_MSG_WARN([`guile-snarf' from Guile not found.  Guile bindings not built.])
        opt_guile_bindings=no
