diff --git libs/ardour/wscript libs/ardour/wscript
index 49cea6a..a7f8e84 100644
--- a/libs/ardour/wscript
+++ b/libs/ardour/wscript
@@ -384,7 +384,7 @@ def build(bld):
 
     # remove '${DEFINES_ST:DEFINES}' from run_str.
     # x86_64-w64-mingw32-as (mingw) -D flag is for debug messages
-    if bld.env['build_target'] == 'mingw':
+    if True:
         class asm(Task.classes['asm']): run_str = '${AS} ${ASFLAGS} ${ASMPATH_ST:INCPATHS} ${AS_SRC_F}${SRC} ${AS_TGT_F}${TGT}'
 
     # operate on copy to avoid adding sources twice
-- 
2.26.2

