--- a/libs/pbd/pbd/pthread_utils.h	2020-09-18 22:30:18.353441907 +0700
+++ b/libs/pbd/pbd/pthread_utils.h	2020-09-18 22:37:07.059363532 +0700
@@ -49,7 +49,7 @@
 #include "pbd/signals.h"
 
 #define PBD_RT_STACKSIZE_PROC 0x80000 // 512kB
-#define PBD_RT_STACKSIZE_HELP 0x08000 // 32kB
+#define PBD_RT_STACKSIZE_HELP PBD_RT_STACKSIZE_PROC // 512kB
 
 /* these are relative to sched_get_priority_max()
  * see pbd_absolute_rt_priority()
