From a5ec4fa3acd4837e5c5914d9ce67b7bc32d4fcfb Mon Sep 17 00:00:00 2001
From: Guido Aulisi <guido.aulisi@gmail.com>
Date: Wed, 27 May 2020 17:23:04 +0200
Subject: [PATCH] Do not install wscript

---
 share/plugin_metadata/wscript | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git share/plugin_metadata/wscript share/plugin_metadata/wscript
index b7aaa92ac1..9f2559095c 100644
--- share/plugin_metadata/wscript
+++ share/plugin_metadata/wscript
@@ -9,7 +9,7 @@ def configure(conf):
     pass
 
 def build(bld):
-    maps = bld.path.ant_glob ('*')
+    maps = bld.path.ant_glob ('plugin*')
     bld.install_files (os.path.join(bld.env['DATADIR'], 'plugin_metadata'),
                        maps)
 
