From 2b5e890480691ce5a6c9f05b55e83a3b9ed8a8b9 Mon Sep 17 00:00:00 2001
From: q66 <daniel@octaforge.org>
Date: Mon, 9 Mar 2020 16:33:43 +0100
Subject: [PATCH] use bundled qm-dsp

---
 libs/qm-dsp/wscript | 11 ++---------
 wscript             |  2 --
 2 files changed, 2 insertions(+), 11 deletions(-)

diff --git libs/qm-dsp/wscript libs/qm-dsp/wscript
index 92dc9502b1..dbbff16722 100644
--- a/libs/qm-dsp/wscript
+++ b/libs/qm-dsp/wscript
@@ -23,17 +23,10 @@ def options(opt):
     autowaf.set_options(opt)
 
 def configure(conf):
-    if conf.is_defined('USE_EXTERNAL_LIBS'):
-        conf.check_cxx(header_name="base/Pitch.h", mandatory=True)
-        conf.check_cxx(lib="qm-dsp", uselib_store="QMDSP", mandatory=True)
-    else:
-        conf.load('compiler_cxx')
-        autowaf.configure(conf)
+    conf.load('compiler_cxx')
+    autowaf.configure(conf)
 
 def build(bld):
-    if bld.is_defined('USE_EXTERNAL_LIBS'):
-        return
-
     # Host Library
     obj = bld(features = 'c cxx cxxstlib')
     obj.source = '''
diff --git wscript wscript
index 4b5acb0ad7..c0358af37e 100644
--- a/wscript
+++ b/wscript
@@ -991,8 +991,6 @@ def configure(conf):
 
     if Options.options.use_external_libs:
         conf.define('USE_EXTERNAL_LIBS', 1)
-        conf.env.append_value(
-            'CXXFLAGS', '-I' + Options.options.qm_dsp_include)
 
     if Options.options.boost_include != '':
         conf.env.append_value('CXXFLAGS', '-I' + Options.options.boost_include)
-- 
2.25.1

