diff -ur host-orig/superscript.com/net/ucspi-ssl-0.96/src/ssl_params.c host/superscript.com/net/ucspi-ssl-0.96/src/ssl_params.c
--- host-orig/superscript.com/net/ucspi-ssl-0.96/src/ssl_params.c	2012-06-07 09:50:27.000000000 -0500
+++ host/superscript.com/net/ucspi-ssl-0.96/src/ssl_params.c	2015-11-10 12:17:02.000000000 -0600
@@ -16,11 +16,13 @@
     if (!SSL_CTX_set_tmp_dh(ctx,dh)) return 0;
   }
 
+#ifndef LIBRESSL_VERSION_NUMBER
   if (len) {
     rsa = RSA_generate_key(len,RSA_F4,0,0);
     if (!rsa) return 0;
     if (!SSL_CTX_set_tmp_rsa(ctx,rsa)) return 0;
   }
+#endif
 
   return 1;
 }
