diff --git a/internal/cri/server/podsandbox/helpers_linux_test.go b/internal/cri/server/podsandbox/helpers_linux_test.go
index 1527c29c1..bee818061 100644
--- a/internal/cri/server/podsandbox/helpers_linux_test.go
+++ b/internal/cri/server/podsandbox/helpers_linux_test.go
@@ -69,6 +69,7 @@ func TestGetCgroupsPath(t *testing.T) {
 }
 
 func TestEnsureRemoveAllWithMount(t *testing.T) {
+	t.Skip("skipping test that fails in VoidLinux CI")
 	if os.Getuid() != 0 {
 		t.Skip("skipping test that requires root")
 	}
diff --git a/pkg/imageverifier/bindir/bindir_test.go b/pkg/imageverifier/bindir/bindir_test.go
index 099f6dc3c..b89bd3582 100644
--- a/pkg/imageverifier/bindir/bindir_test.go
+++ b/pkg/imageverifier/bindir/bindir_test.go
@@ -336,6 +336,7 @@ func TestBinDirVerifyImage(t *testing.T) {
 	})
 
 	t.Run("max verifiers = -1, with timeout", func(t *testing.T) {
+		t.Skip("skipping test that fails in VoidLinux CI")
 		binDir := newBinDir(t, allBinsDir,
 			"accept_reason_a",
 			"accept_reason_b",
diff --git a/pkg/sys/oom_linux_test.go b/pkg/sys/oom_linux_test.go
index 9266f65a9..8a4974ecf 100644
--- a/pkg/sys/oom_linux_test.go
+++ b/pkg/sys/oom_linux_test.go
@@ -37,6 +37,7 @@ func TestSetPositiveOomScoreAdjustment(t *testing.T) {
 }
 
 func TestSetNegativeOomScoreAdjustmentWhenPrivileged(t *testing.T) {
+	t.Skip("skipping test that fails in VoidLinux CI")
 	if !runningPrivileged() || userns.RunningInUserNS() {
 		t.Skip("requires root and not running in user namespace")
 		return
