--- Makefile	2009-11-11 22:05:21.251407668 +0100
+++ Makefile	2009-11-11 22:07:09.891833516 +0100
@@ -1,11 +1,12 @@
 DESTDIR=/usr/
 ROOTDIR=$(DESTDIR)
 LIBDIR=/usr/lib/
-SBINDIR=/sbin
+SBINDIR=/usr/sbin
 CONFDIR=/etc/iproute2
-DOCDIR=/share/doc/iproute2
-MANDIR=/share/man
+DOCDIR=/usr/share/doc/iproute2
+MANDIR=/usr/share/man
 ARPDDIR=/var/lib/arpd
+SHAREDIR=/usr/share

 # Path to db_185.h include
 DBM_INCLUDE:=$(ROOTDIR)/usr/include
--- netem/Makefile	2009-11-11 22:05:21.284750207 +0100
+++ netem/Makefile	2009-11-11 22:07:54.674736924 +0100
@@ -20,9 +20,9 @@
 	$(HOSTCC) $(CCOPTS) -I../include -o $@ $@.c -lm
 
 install: all
-	mkdir -p $(DESTDIR)/lib/tc
+	mkdir -p $(DESTDIR)/${SHAREDIR}/tc
 	for i in $(DISTDATA); \
-	do install -m 755 $$i $(DESTDIR)/lib/tc; \
+	do install -m 755 $$i $(DESTDIR)/${SHAREDIR}/tc; \
 	done
 
 clean:
--- tc/tc_util.c	2009-11-11 22:05:21.298076943 +0100
+++ tc/tc_util.c	2009-11-11 22:09:32.865152646 +0100
@@ -24,8 +24,8 @@
 #include "utils.h"
 #include "tc_util.h"
 
-#ifndef LIBDIR
-#define LIBDIR "/usr/lib/"
+#ifndef SHAREDIR
+#define SHAREDIR "/usr/share"
 #endif
 
 const char *get_tc_lib(void)
@@ -34,7 +34,7 @@
 
 	lib_dir = getenv("TC_LIB_DIR");
 	if (!lib_dir)
-		lib_dir = LIBDIR "/tc/";
+		lib_dir = SHAREDIR "/tc/";
 
 	return lib_dir;
 }
