--- src/core/sipe-digest-openssl.c.orig	2017-06-11 00:58:07.000000000 +0000
+++ src/core/sipe-digest-openssl.c	2017-07-15 11:50:28.560520967 +0000
@@ -77,7 +77,7 @@
 /* Stream HMAC(SHA1) digest for file transfer */
 gpointer sipe_digest_ft_start(const guchar *sha1_digest)
 {
-#if OPENSSL_VERSION_NUMBER < 0x10100000L
+#if OPENSSL_VERSION_NUMBER < 0x10100000L || defined(LIBRESSL_VERSION_NUMBER)
 	HMAC_CTX *ctx = g_malloc(sizeof(HMAC_CTX));
 	HMAC_CTX_init(ctx);
 #else
@@ -101,7 +101,7 @@
 
 void sipe_digest_ft_destroy(gpointer context)
 {
-#if OPENSSL_VERSION_NUMBER < 0x10100000L
+#if OPENSSL_VERSION_NUMBER < 0x10100000L || defined(LIBRESSL_VERSION_NUMBER)
 	HMAC_CTX_cleanup(context);
 	g_free(context);
 #else
