In Void linux, elogind provides libsystemd

diff --git a/.cmake/AddOsSpecificPlugins.cmake b/.cmake/AddOsSpecificPlugins.cmake
index ce0eea9b..5b067671 100644
--- a/.cmake/AddOsSpecificPlugins.cmake
+++ b/.cmake/AddOsSpecificPlugins.cmake
@@ -6,7 +6,7 @@ endif()
 
 # systemd / MPRIS detection
 if (NOT ENABLE_MPRIS MATCHES "false")
-  find_library(LIB_SYSTEMD NAMES systemd)
+  find_library(LIB_SYSTEMD NAMES elogind)
   if (NOT LIB_SYSTEMD MATCHES "LIB_SYSTEMD-NOTFOUND")
     message(STATUS "[mpris] systemd found at " ${LIB_SYSTEMD})
     message(STATUS "[mpris] setting ENABLE_MPRIS=true")
diff --git a/src/plugins/mpris/CMakeLists.txt b/src/plugins/mpris/CMakeLists.txt
index fdee62d2..7a941d08 100644
--- a/src/plugins/mpris/CMakeLists.txt
+++ b/src/plugins/mpris/CMakeLists.txt
@@ -6,4 +6,4 @@ find_package(PkgConfig)
 pkg_check_modules (SYSTEMD REQUIRED libsystemd)
 
 add_library(mpris SHARED ${mpris_SOURCES})
-target_link_libraries(mpris systemd)
+target_link_libraries(mpris elogind)
