--- a/src/musikcube/app/model/DirectoryAdapter.cpp
+++ b/src/musikcube/app/model/DirectoryAdapter.cpp
@@ -127,19 +127,23 @@ static void buildDirectoryList(
     }
     catch (...) {
         /* may throw trying to open the directory */
     }
 
+#ifdef __GLIBC__
     try {
         std::sort(
             target.begin(),
             target.end(),
             std::locale(setlocale(LC_ALL, nullptr)));
     }
     catch (...) {
         std::sort(target.begin(), target.end());
     }
+#else
+    std::sort(target.begin(), target.end());
+#endif
 }
 
 static std::string pathToString(fs::path path) {
     return musik::core::NormalizeDir(path.u8string());
 }
