Taken from libgme patches.

--- a/src/plugins/gmedecoder/gme/blargg_endian.h
+++ b/src/plugins/gmedecoder/gme/blargg_endian.h
@@ -36,7 +36,7 @@
 #endif
 
 #if defined (MSB_FIRST)     || defined (__BIG_ENDIAN__) || defined (WORDS_BIGENDIAN) || \
-	defined (__sparc__)     ||  BLARGG_CPU_POWERPC || \
+	defined (__sparc__) || \
 	(defined (BIG_ENDIAN) && BIG_ENDIAN+0 != 4321)
 	#define BLARGG_BIG_ENDIAN 1
 #elif !defined (__mips__)
