--- configure.py.orig	2017-03-30 10:46:57.000000000 +0200
+++ configure.py	2017-12-17 14:51:45.159142941 +0100
@@ -2276,6 +2276,7 @@
     name_pro = name + '.pro'
     name_makefile = name + '.mk'
     name_source = name + '.cpp'
+    name_inc = name + '.inc'
 
     # Create the .pro file.
     pro_lines = []
@@ -2283,6 +2284,9 @@
     pro_lines.append('TARGET = %s' % name)
     pro_lines.append('SOURCES = %s' % name_source)
 
+    if (os.path.isfile(name_inc)):
+        pro_lines.append('include(%s)' % name_inc)
+
     f = open_for_writing(name_pro)
     f.write('\n'.join(pro_lines))
     f.close()
