--- ./configure.py.dbus	2016-04-24 22:16:43.120281591 -0500
+++ ./configure.py	2016-04-25 10:16:52.465218805 -0500
@@ -2141,7 +2141,7 @@ def check_dbus(target_config, verbose):
     else:
         dlist = target_config.dbus_inc_dirs
 
-    target_config.dbus_inc_dirs = []
+    #target_config.dbus_inc_dirs = []
 
     for d in dlist:
         if os.access(os.path.join(d, 'dbus', 'dbus-python.h'), os.F_OK):
