diff --git a/bitops.h b/bitops.h
index 323c571..7f30b06 100644
--- bitops.h
+++ bitops.h
@@ -65,6 +65,9 @@ static inline unsigned long hweight_long(unsigned long w)
 }
 
 #define BITOP_WORD(nr)		((nr) / BITS_PER_LONG)
+#ifndef __always_inline
+#define __always_inline         inline __attribute__((always_inline))
+#endif
 
 /**
  * __ffs - find first bit in word.
diff --git a/kerncompat.h.orig b/kerncompat.h
index bb03194e127e..28f832a2e625 100644
--- kerncompat.h
+++ kerncompat.h
@@ -28,6 +28,7 @@
 #include <assert.h>
 #include <stddef.h>
 #include <linux/types.h>
+#include <linux/limits.h>
 #include <stdint.h>
 
 #define ptr_to_u64(x)	((u64)(uintptr_t)x)
--- rbtree.h
+++ rbtree.h
@@ -38,6 +38,10 @@
 extern "C" {
 #endif
 
+#ifndef __always_inline
+#define __always_inline         inline __attribute__((always_inline))
+#endif
+
 struct rb_node {
 	unsigned long  __rb_parent_color;
 	struct rb_node *rb_right;
