Fixes /usr/lib/libbtrfs.so: undefined reference to `__error'

--- Makefile.orig	2017-03-16 08:47:29.000000000 +0000
+++ Makefile	2017-03-16 08:53:24.313214267 +0000
@@ -105,7 +105,7 @@
 	       mkfs/common.o
 libbtrfs_objects = send-stream.o send-utils.o kernel-lib/rbtree.o btrfs-list.o \
 		   kernel-lib/crc32c.o \
-		   uuid-tree.o utils-lib.o rbtree-utils.o
+		   uuid-tree.o utils-lib.o rbtree-utils.o messages.o
 libbtrfs_headers = send-stream.h send-utils.h send.h kernel-lib/rbtree.h btrfs-list.h \
 	       kernel-lib/crc32c.h kernel-lib/list.h kerncompat.h \
 	       kernel-lib/radix-tree.h kernel-lib/sizes.h extent-cache.h \
