--- convert/source-fs.h.orig	2017-09-08 14:20:05.000000000 +0000
+++ convert/source-fs.h	2017-09-09 08:48:46.648264368 +0000
@@ -20,6 +20,7 @@
 #include "kerncompat.h"
 #include <linux/kdev_t.h>
 #include <pthread.h>
+#include <sys/types.h>
 
 #define CONV_IMAGE_SUBVOL_OBJECTID BTRFS_FIRST_FREE_OBJECTID
 
--- mkfs/main.c
+++ mkfs/main.c
@@ -26,6 +26,7 @@
 #include <stdlib.h>
 /* #include <sys/dir.h> included via androidcompat.h */
 #include <fcntl.h>
+#include <limits.h> /* for PATH_MAX */
 #include <unistd.h>
 #include <getopt.h>
 #include <uuid/uuid.h>
