diff -p1ru a/data/CMakeLists.txt b/data/CMakeLists.txt
--- a/data/CMakeLists.txt	2023-02-23 02:37:44.000000000 +0800
+++ b/data/CMakeLists.txt	2024-07-27 09:32:29.100314406 +0800
@@ -41,3 +41,3 @@ install(FILES ${CMAKE_CURRENT_BINARY_DIR
 configure_file(unity7.service.in ${CMAKE_CURRENT_BINARY_DIR}/unity7.service @ONLY)
-pkg_get_variable(SYSTEMD_USER_DIR systemd systemduserunitdir)
+set(SYSTEMD_USER_DIR "${CMAKE_INSTALL_PREFIX}/lib/systemd/user")
 install(FILES ${CMAKE_CURRENT_BINARY_DIR}/unity7.service DESTINATION ${SYSTEMD_USER_DIR})
