diff -p1ru a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2023-02-23 02:37:59.000000000 +0800
+++ b/CMakeLists.txt	2024-07-27 08:48:23.292047534 +0800
@@ -175,5 +175,5 @@ execute_process (COMMAND ${PKG_CONFIG_EX
 execute_process (COMMAND ${PKG_CONFIG_EXECUTABLE} unity --variable lensesdir OUTPUT_VARIABLE LENSES_DIR OUTPUT_STRIP_TRAILING_WHITESPACE)
-execute_process (COMMAND ${PKG_CONFIG_EXECUTABLE} indicator3-0.4 --variable indicatordir OUTPUT_VARIABLE INDICATORDIR OUTPUT_STRIP_TRAILING_WHITESPACE)
-execute_process (COMMAND ${PKG_CONFIG_EXECUTABLE} indicator3-0.4 --variable iconsdir OUTPUT_VARIABLE INDICATORICONDIR OUTPUT_STRIP_TRAILING_WHITESPACE)
-execute_process (COMMAND ${PKG_CONFIG_EXECUTABLE} indicator3-0.4 --variable prefix OUTPUT_VARIABLE INDICATORPREFIX OUTPUT_STRIP_TRAILING_WHITESPACE)
+execute_process (COMMAND ${PKG_CONFIG_EXECUTABLE} ayatana-indicator3-0.4 --variable indicatordir OUTPUT_VARIABLE INDICATORDIR OUTPUT_STRIP_TRAILING_WHITESPACE)
+execute_process (COMMAND ${PKG_CONFIG_EXECUTABLE} ayatana-indicator3-0.4 --variable iconsdir OUTPUT_VARIABLE INDICATORICONDIR OUTPUT_STRIP_TRAILING_WHITESPACE)
+execute_process (COMMAND ${PKG_CONFIG_EXECUTABLE} ayatana-indicator3-0.4 --variable prefix OUTPUT_VARIABLE INDICATORPREFIX OUTPUT_STRIP_TRAILING_WHITESPACE)
 set (INDICATOR_SERVICE_DIR "${INDICATORPREFIX}/share/${PROJECT_NAME}/indicators")
@@ -243,3 +243,3 @@ set(UNITY_PLUGIN_SHARED_DEPS
     gtk+-3.0>=3.1
-    indicator3-0.4>=0.4.90
+    ayatana-indicator3-0.4
     json-glib-1.0
diff -p1ru a/launcher/ApplicationLauncherIcon.h b/launcher/ApplicationLauncherIcon.h
--- a/launcher/ApplicationLauncherIcon.h	2023-02-23 02:37:44.000000000 +0800
+++ b/launcher/ApplicationLauncherIcon.h	2024-07-27 08:49:55.159958565 +0800
@@ -26,3 +26,3 @@
 
-#include <libindicator/indicator-desktop-shortcuts.h>
+#include <libayatana-indicator/indicator-desktop-shortcuts.h>
 
diff -p1ru a/services/CMakeLists.txt b/services/CMakeLists.txt
--- a/services/CMakeLists.txt	2023-02-23 02:37:44.000000000 +0800
+++ b/services/CMakeLists.txt	2024-07-27 08:48:57.861323543 +0800
@@ -11,5 +11,5 @@ set(UNITY_PANEL_SERVICE_DEPS
     gtk+-3.0>=3.3
-    indicator3-0.4>=12.10.2
+    ayatana-indicator3-0.4
     x11
-    libido3-0.1>=13.0.0
+    libayatana-ido3-0.4
 )
diff -p1ru a/services/panel-indicator-accessible.h b/services/panel-indicator-accessible.h
--- a/services/panel-indicator-accessible.h	2023-02-23 02:37:44.000000000 +0800
+++ b/services/panel-indicator-accessible.h	2024-07-27 08:51:14.274496085 +0800
@@ -22,4 +22,4 @@
 #include <atk/atk.h>
-#include <libindicator/indicator.h>
-#include <libindicator/indicator-object.h>
+#include <libayatana-indicator/indicator.h>
+#include <libayatana-indicator/indicator-object.h>
 
diff -p1ru a/services/panel-indicator-entry-accessible.h b/services/panel-indicator-entry-accessible.h
--- a/services/panel-indicator-entry-accessible.h	2023-02-23 02:37:44.000000000 +0800
+++ b/services/panel-indicator-entry-accessible.h	2024-07-27 08:50:43.352042575 +0800
@@ -22,4 +22,4 @@
 #include <atk/atk.h>
-#include <libindicator/indicator.h>
-#include <libindicator/indicator-object.h>
+#include <libayatana-indicator/indicator.h>
+#include <libayatana-indicator/indicator-object.h>
 
diff -p1ru a/services/panel-main.c b/services/panel-main.c
--- a/services/panel-main.c	2023-02-23 02:37:44.000000000 +0800
+++ b/services/panel-main.c	2024-07-27 08:51:58.115405592 +0800
@@ -24,3 +24,3 @@
 #include <gtk/gtk.h>
-#include <libido/libido.h>
+#include <libayatana-ido/libayatana-ido.h>
 
diff -p1ru a/services/panel-service.c b/services/panel-service.c
--- a/services/panel-service.c	2023-02-23 02:37:44.000000000 +0800
+++ b/services/panel-service.c	2024-07-27 08:51:36.358532591 +0800
@@ -30,3 +30,3 @@
 #include <glib/gi18n-lib.h>
-#include <libindicator/indicator-ng.h>
+#include <libayatana-indicator/indicator-ng.h>
 
diff -p1ru a/services/panel-service.h b/services/panel-service.h
--- a/services/panel-service.h	2023-02-23 02:37:44.000000000 +0800
+++ b/services/panel-service.h	2024-07-27 08:50:56.180453579 +0800
@@ -22,4 +22,4 @@
 #include <glib-object.h>
-#include <libindicator/indicator.h>
-#include <libindicator/indicator-object.h>
+#include <libayatana-indicator/indicator.h>
+#include <libayatana-indicator/indicator-object.h>
 
diff -p1ru a/tests/mock_indicator_object.h b/tests/mock_indicator_object.h
--- a/tests/mock_indicator_object.h	2023-02-23 02:37:45.000000000 +0800
+++ b/tests/mock_indicator_object.h	2024-07-27 08:50:18.488617071 +0800
@@ -24,3 +24,3 @@
 
-#include <libindicator/indicator-object.h>
+#include <libayatana-indicator/indicator-object.h>
 
