diff --git config/meson.build config/meson.build
index 2bafea5..6b15f13 100644
--- a/config/meson.build
+++ b/config/meson.build
@@ -50,11 +50,7 @@ meson.add_install_script('../buildtools/symlink-drivers-solibs.sh',
 		pmd_subdir_opt)
 
 # set the machine type and cflags for it
-if meson.is_cross_build()
-	machine = host_machine.cpu()
-else
-	machine = get_option('machine')
-endif
+machine = get_option('machine')
 
 # machine type 'default' is special, it defaults to the per arch agreed common
 # minimal baseline needed for DPDK.
