Index: dpdk-22.11/config/meson.build
===================================================================
--- dpdk-22.11.orig/config/meson.build
+++ dpdk-22.11/config/meson.build
@@ -74,7 +74,7 @@ enable_drivers = ''
 platform = get_option('platform')
 
 # set the cpu_instruction_set and cflags for it
-if meson.is_cross_build()
+if false
     cpu_instruction_set = host_machine.cpu()
 else
     cpu_instruction_set = get_option('cpu_instruction_set')
Index: dpdk-22.11/config/arm/meson.build
===================================================================
--- dpdk-22.11.orig/config/arm/meson.build
+++ dpdk-22.11/config/arm/meson.build
@@ -501,7 +501,7 @@ soc_flags = []
 if dpdk_conf.get('RTE_ARCH_32')
     # 32-bit build
     dpdk_conf.set('RTE_CACHE_LINE_SIZE', 64)
-    if meson.is_cross_build()
+    if false
         update_flags = true
         soc = meson.get_cross_property('platform', '')
         if soc == ''
@@ -523,14 +523,14 @@ else
     dpdk_conf.set('RTE_ARCH', 'armv8')
     update_flags = true
     soc_config = {}
-    if not meson.is_cross_build()
+    if true
         # for backwards compatibility:
         #   machine=native is the same behavior as soc=native
         #   machine=generic/default is the same as soc=generic
         # cpu_instruction_set holds the proper value - native, generic or cpu
         # the old behavior only distinguished between generic and native build
         if machine != 'auto'
-            if cpu_instruction_set == 'generic'
+            if cpu_instruction_set != 'native'
                 soc = 'generic'
             else
                 soc = 'native'
