Index: dpdk-19.08/doc/guides/conf.py
===================================================================
--- dpdk-19.08.orig/doc/guides/conf.py
+++ dpdk-19.08/doc/guides/conf.py
@@ -410,4 +410,4 @@ def setup(app):
         # Process the numref references once the doctree has been created.
         app.connect('doctree-resolved', process_numref)
 
-    app.add_stylesheet('css/custom.css')
+    app.add_css_file('css/custom.css')
