--- qt6-location-6.4.2.orig/tests/auto/CMakeLists.txt
+++ qt6-location-6.4.2/tests/auto/CMakeLists.txt
@@ -14,7 +14,6 @@ add_subdirectory(qgeopositioninfo)
 add_subdirectory(qgeosatelliteinfo)
 add_subdirectory(qgeosatelliteinfosource)
 add_subdirectory(qnmeasatelliteinfosource)
-add_subdirectory(cmake)
 if (QT6_IS_SHARED_LIBS_BUILD)
     add_subdirectory(positionplugin)
     add_subdirectory(positionplugintest)
