--- a/js/src/build/moz.build
+++ b/js/src/build/moz.build
@@ -80,14 +80,3 @@
 NO_EXPAND_LIBS = True
 
 DIST_INSTALL = True
-
-# Run SpiderMonkey style checker after linking the static library. This avoids
-# running the script for no-op builds.
-GeneratedFile(
-    'spidermonkey_checks', script='/config/run_spidermonkey_checks.py',
-    inputs=[
-        '!%sjs_static.%s' % (CONFIG['LIB_PREFIX'], CONFIG['LIB_SUFFIX']),
-        '/config/check_spidermonkey_style.py',
-        '/config/check_macroassembler_style.py',
-        '/config/check_js_opcode.py'
-        ])
