Description: pbbuttonsd is not rewritten for kernel 3.2
Author: Mathieu Malaterre <malat@debian.org>
Origin: https://bugs.gentoo.org/show_bug.cgi?id=429306#c2
Bug-Debian: http://bugs.debian.org/711685
Reviewed-by: Mathieu Malaterre <malat@debian.org>

Index: pbbuttonsd-0.7.9/scripts/scripts.d/laptopmode.sh
===================================================================
--- pbbuttonsd-0.7.9.orig/scripts/scripts.d/laptopmode.sh
+++ pbbuttonsd-0.7.9/scripts/scripts.d/laptopmode.sh
@@ -122,7 +122,7 @@ KLEVEL=$(
 	     )
 	 )
 case "$KLEVEL" in
-	"2.4"|"2.6")
+	"2.4"|"2.6"|"3."*)
 		true
 		;;
 	*)
@@ -222,7 +222,7 @@ case "$1" in
 				echo "1"				> /proc/sys/vm/laptop_mode
 				echo "30 500 0 0 $AGE $AGE 60 20 0"	> /proc/sys/vm/bdflush
 				;;
-			"2.6")
+			"2.6"|"3."*)
 				echo "5"				> /proc/sys/vm/laptop_mode
 				echo "$AGE"				> /proc/sys/vm/dirty_writeback_centisecs
 				echo "$AGE"				> /proc/sys/vm/dirty_expire_centisecs
@@ -268,7 +268,7 @@ case "$1" in
 			"2.4")
 				echo "30 500 0 0 $U_AGE $B_AGE 60 20 0"	> /proc/sys/vm/bdflush
 				;;
-			"2.6")
+			"2.6"|"3."*)
 				echo "$U_AGE"				> /proc/sys/vm/dirty_writeback_centisecs
 				echo "$B_AGE"				> /proc/sys/vm/dirty_expire_centisecs
 				echo "$DEF_DIRTY_RATIO"			> /proc/sys/vm/dirty_ratio
