--- a/tests/meson.build	2026-09-01 11:38:03.896222339 +0200
+++ b/tests/meson.build	2026-09-01 11:42:16.446248925 +0200
@@ -1,6 +1,6 @@
 test('check-format', find_program('check-format.sh'))
 
-libmocklibc = subproject('mocklibc').get_variable('libmocklibc')
+#libmocklibc = subproject('mocklibc').get_variable('libmocklibc')
 
 envs = environment()
 # Enable debug messages and abort on critical warnings
@@ -8,7 +8,7 @@
 envs.set('G_MESSAGES_DEBUG', 'all')
 envs.set('top_srcdir', meson.source_root())
 envs.set('top_builddir', meson.build_root())
-envs.set('MOCKLIBC_LD_PRELOAD', libmocklibc.full_path())
+envs.set('MOCKLIBC_LD_PRELOAD', '')
 
 # Setup paths
 accountsservice_build_dir = meson.build_root() / 'src' / 'libaccountsservice'
