diff --git a/libharvid/timecode.h b/libharvid/timecode.h
index 204c979..62c1684 100644
--- a/libharvid/timecode.h
+++ b/libharvid/timecode.h
@@ -27,13 +27,6 @@
 typedef int int32_t;
 #endif
 
-#if (!defined int64_t && !defined __int8_t_defined && !defined _UINT64_T)
-#  if __WORDSIZE == 64
-typedef long int int64_t;
-#else
-typedef long long int int64_t;
-#endif
-#endif
 
 /**
  * classical timecode
