diff --git src/main.nqp src/main.nqp
index b53302759..bd9aeb5d5 100644
--- src/main.nqp
+++ src/main.nqp
@@ -33,7 +33,7 @@ my $install-dir := $execname eq ''
 
 my $perl6-home := $comp.config<static_perl6_home>
     // nqp::getenvhash()<PERL6_HOME>
-    // $install-dir ~ '/share/perl6';
+    // $comp.config<libdir> ~ '/perl6';
 if nqp::substr($perl6-home, nqp::chars($perl6-home) - 1) eq $sep {
     $perl6-home := nqp::substr($perl6-home, 0, nqp::chars($perl6-home) - 1);
 }
diff --git tools/lib/NQP/Config/Rakudo.pm tools/lib/NQP/Config/Rakudo.pm
index 90839c229..2819b60b6 100644
--- tools/lib/NQP/Config/Rakudo.pm
+++ tools/lib/NQP/Config/Rakudo.pm
@@ -347,7 +347,7 @@ sub configure_moar_backend {
           );
         $config->{static_perl6_home} =
           File::Spec->rel2abs(
-            File::Spec->catdir( $config->{prefix}, 'share', 'perl6' ) );
+            File::Spec->catdir( $config->{libdir}, 'perl6' ) );
         $config->{static_nqp_home_define} =
           '-DSTATIC_NQP_HOME='
           . $self->c_escape_string( $config->{static_nqp_home} );
