pyside6-$tool is equivalent to $tool -g python

--- a/src/tools/ui_compile.py
+++ b/src/tools/ui_compile.py
@@ -19,7 +19,9 @@
     python_name = f"{ui_file_name.replace('.ui', '')}.py"
 
     cmd = [
-        "pyside6-uic",
+        "/usr/lib/qt6/libexec/uic",
+        "-g",
+        "python",
         str(ui_file),
         "-o",
         str(dir / python_name),
@@ -36,7 +36,7 @@
     ui_file_name = qrc_file.name
     python_name = f"{ui_file_name.replace('.qrc', '')}_rc.py"
 
-    cmd = ["pyside6-rcc", str(qrc_file), "-o", str(dir / python_name)]
+    cmd = ["/usr/lib/qt6/libexec/rcc", "-g", "python", str(qrc_file), "-o", str(dir / python_name)]
     print(f"{' '.join(cmd)}")
     subprocess.run(cmd, check=True)
 
