--- a/mozc/src/config.bzl
+++ b/mozc/src/config.bzl
@@ -44,7 +44,7 @@ LINUX_MOZC_DOCUMENT_DIR = LINUX_MOZC_SER
 IBUS_COMPONENT_DIR = "/usr/share/ibus/component"
 IBUS_MOZC_INSTALL_DIR = "/usr/share/ibus-mozc"
 IBUS_MOZC_ICON_PATH = IBUS_MOZC_INSTALL_DIR + "/product_icon.png"
-IBUS_MOZC_PATH = "/usr/lib/ibus-mozc/ibus-engine-mozc"
+IBUS_MOZC_PATH = "/usr/libexec/ibus-engine-mozc"
 EMACS_MOZC_CLIENT_DIR = "/usr/share/emacs/site-lisp/emacs-mozc"
 EMACS_MOZC_HELPER_DIR = "/usr/bin"
 
