--- mozc/src/build_mozc.py
+++ mozc/src/build_mozc.py
@@ -728,6 +728,7 @@ def GypMain(options, unused_args):
   gyp_options.extend(['--generator-output=.'])
   short_basename = GetBuildShortBaseName(target_platform)
   gyp_options.extend(['-G', 'output_dir=%s' % short_basename])
+  gyp_options.extend(['--no-parallel'])  # Parallel generation requires writable /dev/shm
 
   # Enable cross-compile
   # TODO(yukawa): Enable GYP_CROSSCOMPILE for Windows.
