--- a/src/bazel/pkg_config_repository.bzl
+++ b/src/bazel/pkg_config_repository.bzl
@@ -59,9 +59,7 @@ package(
 
 cc_library(
     name = "{name}",
-    hdrs = glob([
-        {hdrs}
-    ]),
+    {hdrs}
     copts = [
         {copts}
     ],
@@ -96,15 +94,18 @@ def _symlinks(repo_ctx, paths):
 
 def _pkg_config_repository_impl(repo_ctx):
     includes = _exec_pkg_config(repo_ctx, "--cflags-only-I")
-    includes = [item[len("-I/"):] for item in includes]
+    includes = [item[len("-I/"):] for item in includes if item]
     _symlinks(repo_ctx, includes)
     data = {
         "name": repo_ctx.attr.name,
-        "hdrs": _make_strlist([item + "/**" for item in includes]),
         "copts": _make_strlist(_exec_pkg_config(repo_ctx, "--cflags-only-other")),
         "includes": _make_strlist(includes),
         "linkopts": _make_strlist(_exec_pkg_config(repo_ctx, "--libs-only-l")),
     }
+    if includes:
+        data['hdrs'] = 'hdrs = glob([' + _make_strlist([item + "/**" for item in includes]) + ']),'
+    else:
+        data['hdrs'] = ''
     build_file_data = BUILD_TEMPLATE.format(**data)
 
     libexecdir = _exec_pkg_config(repo_ctx, "--variable=libexecdir")
