Upstream: No
Reason: musl doesn't have execinfo.h

Index: scribus/util.cpp
===================================================================
--- a/scribus/util.cpp	(revision 22789)
+++ b/scribus/util.cpp	(working copy)
@@ -41,7 +41,6 @@
 #include <signal.h>
 
 #if !defined(_WIN32) && !defined(Q_OS_MAC) 
-#include <execinfo.h>
 #include <cxxabi.h>
 #endif
 #if defined(_WIN32)
Index: scribus/util_debug.cpp
===================================================================
--- a/scribus/util_debug.cpp	(revision 22789)
+++ b/scribus/util_debug.cpp	(working copy)
@@ -22,7 +22,6 @@
  ***************************************************************************/
 
 #if !defined(_WIN32) && !defined(Q_OS_MAC)
-#include <execinfo.h>
 #include <cxxabi.h>
 #endif
 #if defined(_WIN32)
@@ -51,7 +50,7 @@
  */
 void printBacktrace ( int nFrames )
 {
-#if !defined(_WIN32) && !defined(Q_OS_MAC) && !defined(Q_OS_OPENBSD) && !defined(Q_OS_FREEBSD)
+#if 0
 	void ** trace = new void*[nFrames + 1];
 	char **messages = ( char ** ) NULL;
 	int i, trace_size = 0;
