--- source/puddlestuff/puddleobjects.py	2015-11-02 15:55:52.830814590 +0100
+++ source/puddlestuff/puddleobjects.py	2015-11-02 15:55:59.994099731 +0100
@@ -1418,7 +1418,7 @@
         super(ArtworkLabel, self).__init__(*args, **kwargs)
 
         pal = self.palette()
-        pal.setBrush(self.backgroundRole(), QBrush(QPalette.Window))
+        pal.setBrush(self.backgroundRole(), QBrush(pal.window()))
         self.setAutoFillBackground(True)
         self.setPalette(pal)
 
