From bde0dff47d76a874bcbb5e75a74f6db83b90da5f Mon Sep 17 00:00:00 2001
From: Daniel Kolesa <daniel@octaforge.org>
Date: Thu, 5 Nov 2020 02:41:14 +0100
Subject: [PATCH] add support for using ncursesw pkg-config library

---
 aclocal.m4   | 3 +++
 configure    | 4 ++++
 configure.ac | 1 +
 3 files changed, 8 insertions(+)

diff --git aclocal.m4 aclocal.m4
index 1413267..16775a9 100644
--- a/aclocal.m4
+++ b/aclocal.m4
@@ -981,6 +981,9 @@ TERMCAP_DEP=
 elif test $bash_cv_termcap_lib = libncurses; then
 TERMCAP_LIB=-lncurses
 TERMCAP_DEP=
+elif test $bash_cv_termcap_lib = libncursesw; then
+TERMCAP_LIB=-lncursesw
+TERMCAP_DEP=
 elif test $bash_cv_termcap_lib = libc; then
 TERMCAP_LIB=
 TERMCAP_DEP=
diff --git configure configure
index c9b7ade..7d7bce0 100755
--- a/configure
+++ b/configure
@@ -6328,6 +6328,9 @@ TERMCAP_DEP=
 elif test $bash_cv_termcap_lib = libncurses; then
 TERMCAP_LIB=-lncurses
 TERMCAP_DEP=
+elif test $bash_cv_termcap_lib = libncursesw; then
+TERMCAP_LIB=-lncursesw
+TERMCAP_DEP=
 elif test $bash_cv_termcap_lib = libc; then
 TERMCAP_LIB=
 TERMCAP_DEP=
@@ -6363,6 +6366,7 @@ case "$TERMCAP_LIB" in
 -ltinfo)  TERMCAP_PKG_CONFIG_LIB=tinfo ;;
 -lcurses) TERMCAP_PKG_CONFIG_LIB=ncurses ;;
 -lncurses) TERMCAP_PKG_CONFIG_LIB=ncurses ;;
+-lncursesw) TERMCAP_PKG_CONFIG_LIB=ncursesw ;;
 -ltermcap) TERMCAP_PKG_CONFIG_LIB=termcap ;;
 *) TERMCAP_PKG_CONFIG_LIB=termcap ;;
 esac
diff --git configure.ac configure.ac
index b73e37b..db788a6 100644
--- a/configure.ac
+++ b/configure.ac
@@ -202,6 +202,7 @@ case "$TERMCAP_LIB" in
 -ltinfo)  TERMCAP_PKG_CONFIG_LIB=tinfo ;;
 -lcurses) TERMCAP_PKG_CONFIG_LIB=ncurses ;;
 -lncurses) TERMCAP_PKG_CONFIG_LIB=ncurses ;;
+-lncursesw) TERMCAP_PKG_CONFIG_LIB=ncursesw ;;
 -ltermcap) TERMCAP_PKG_CONFIG_LIB=termcap ;;
 *) TERMCAP_PKG_CONFIG_LIB=termcap ;;
 esac
-- 
2.29.2

