From ad5b290816c57cae153ebf2dfbd64e8c7924344c Mon Sep 17 00:00:00 2001
From: Johannes Rosenberger <johannes@jorsn.eu>
Date: Mon, 5 Jan 2026 17:14:37 +0100
Subject: [PATCH] gir_main plugin: add compatibility with girepository-2.0

fix #765
---
 src/plugin/gir_main.c | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/src/plugin/gir_main.c b/src/plugin/gir_main.c
index 2afda0835..6be395984 100644
--- a/src/plugin/gir_main.c
+++ b/src/plugin/gir_main.c
@@ -4,7 +4,13 @@
  *
  */
 
-# include <girepository.h>
+// compatibility with girepository-1.0 and 2.0
+#if __has_include(<girepository.h>) // girepository-1.0
+  #include <girepository.h>
+  #define gi_repository_get_option_group g_irepository_get_option_group
+#else
+  #include <girepository/girepository.h>
+#endif
 
 # include "astroid_activatable.h"
 
@@ -14,7 +20,7 @@ int main (int argc, char ** argv) {
   GError *error = NULL;
 
   ctx = g_option_context_new (NULL);
-  g_option_context_add_group (ctx, g_irepository_get_option_group ());
+  g_option_context_add_group (ctx, gi_repository_get_option_group ());
 
   if (!g_option_context_parse (ctx, &argc, &argv, &error)) {
     g_print ("astroid girmain: %s\n", error->message);
