--- src/wwwsrv.c.old	2010-10-26 14:16:31.000000000 +0200
+++ src/wwwsrv.c	2010-10-26 14:17:35.000000000 +0200
@@ -100,12 +100,12 @@
 
   data->privacy = privacy;
   data->log = log;
-  strncpy (data->metaserver, metaserver, URL_SIZE);
+  strncpy (data->metaserver, metaserver, sizeof(data->metaserver));
   data->port = port;
   data->busy_players = busy_players;
   data->max_players = max_players;
-  strncpy (data->password, password, URL_SIZE);
-  strncpy (data->comment, comment, URL_SIZE);
+  strncpy (data->password, password, sizeof(data->password));
+  strncpy (data->comment, comment, sizeof(data->comment));
 }
 
 /*------------------------------------------------------------------*/
