--- src/Makefile.in.old	2011-06-22 07:45:05.756127862 +0200
+++ src/Makefile.in	2011-06-22 07:46:00.681613957 +0200
@@ -223,7 +223,7 @@
     EXTERN_LIBS = `allegro-config --static $(ALLEGRO_LIBRARY)` $(EFENCE_LIBRARY)
 else
     LINK_OPTIONS = 
-    EXTERN_LIBS = `allegro-config --libs $(ALLEGRO_LIBRARY)` $(EFENCE_LIBRARY)
+    EXTERN_LIBS = `allegro-config --libs $(ALLEGRO_LIBRARY)` $(EFENCE_LIBRARY) -lm
 endif
 
 CONFIG_OPTIONS = \
