--- a/kernel/uvm/nvidia_uvm_lite.c
+++ b/kernel/uvm/nvidia_uvm_lite.c
@@ -822,8 +822,10 @@ done:
 #if defined(NV_VM_OPERATIONS_STRUCT_HAS_FAULT)
 #if LINUX_VERSION_CODE < KERNEL_VERSION(4, 11, 0)
 int _fault(struct vm_area_struct *vma, struct vm_fault *vmf)
-#else 	
-int _fault(struct vm_fault *vmf) 	
+#elif LINUX_VERSION_CODE < KERNEL_VERSION(5, 0, 0)     
+int _fault(struct vm_fault *vmf)       
+#else
+unsigned int _fault(struct vm_fault *vmf)      
 #endif
 {
 #if defined(NV_VM_FAULT_HAS_ADDRESS)
@@ -878,8 +880,10 @@ static struct vm_operations_struct uvmli
 #if defined(NV_VM_OPERATIONS_STRUCT_HAS_FAULT)
 #if LINUX_VERSION_CODE < KERNEL_VERSION(4, 11, 0)
 int _sigbus_fault(struct vm_area_struct *vma, struct vm_fault *vmf)
-#else
+#elif LINUX_VERSION_CODE < KERNEL_VERSION(5, 0, 0)
 int _sigbus_fault(struct vm_fault *vmf)
+#else
+unsigned int _sigbus_fault(struct vm_fault *vmf)
 #endif
 {
     vmf->page = NULL;
