diff -Nur nv/kernel/nv-drm.c nv2/kernel/nv-drm.c
--- kernel/nv-drm.c	2018-05-25 06:16:20.000000000 +0200
+++ kernel/nv-drm.c	2019-05-09 20:51:33.364278935 +0200
@@ -146,11 +146,7 @@
 };
 
 static struct drm_driver nv_drm_driver = {
-#if defined(DRIVER_LEGACY)
     .driver_features = DRIVER_GEM | DRIVER_PRIME | DRIVER_LEGACY,
-#else
-    .driver_features = DRIVER_GEM | DRIVER_PRIME,
-#endif
     .load = nv_drm_load,
     .unload = nv_drm_unload,
     .fops = &nv_drm_fops,

