--- kernel/uvm/nvidia_uvm_lite.c	2019-09-10 16:26:28.674315544 +0200
+++ kernel/uvm/nvidia_uvm_lite.c	2019-09-10 16:28:05.224054277 +0200
@@ -820,8 +830,10 @@
 #if defined(NV_VM_OPERATIONS_STRUCT_HAS_FAULT)
 #if LINUX_VERSION_CODE < KERNEL_VERSION(4, 11, 0)
 int _fault(struct vm_area_struct *vma, struct vm_fault *vmf)
-#else 	
-int _fault(struct vm_fault *vmf) 	
+#elif LINUX_VERSION_CODE < KERNEL_VERSION(5, 0, 0)     
+int _fault(struct vm_fault *vmf)       
+#else
+unsigned int _fault(struct vm_fault *vmf)      
 #endif
 {
 #if defined(NV_VM_FAULT_HAS_ADDRESS)
@@ -876,8 +888,10 @@
 #if defined(NV_VM_OPERATIONS_STRUCT_HAS_FAULT)
 #if LINUX_VERSION_CODE < KERNEL_VERSION(4, 11, 0)
 int _sigbus_fault(struct vm_area_struct *vma, struct vm_fault *vmf)
-#else
+#elif LINUX_VERSION_CODE < KERNEL_VERSION(5, 0, 0)
 int _sigbus_fault(struct vm_fault *vmf)
+#else
+unsigned int _sigbus_fault(struct vm_fault *vmf)
 #endif
 {
     vmf->page = NULL;
