diff --git a/headers/meson.build b/headers/meson.build
index 3ce8b61..1132695 100644
--- headers/meson.build
+++ headers/meson.build
@@ -19,7 +19,7 @@ enums_xml = custom_target(
   install: true,
   install_dir: schemasdir)
 
-if not meson.is_cross_build() and get_option('introspection')
+if get_option('introspection')
   noinst_lib = shared_library('noinst',
                               headers,
                               install: false)
