--- a/XADMaster/XADRAR5Parser.h
+++ b/XADMaster/XADRAR5Parser.h
@@ -67,6 +67,7 @@
 +(int)requiredHeaderSize;
 +(BOOL)recognizeFileWithHandle:(CSHandle *)handle firstBytes:(NSData *)data name:(NSString *)name;
 +(NSArray *)volumesForHandle:(CSHandle *)handle firstBytes:(NSData *)data name:(NSString *)name;
++ (off_t)signatureLocationInData:(NSData *)data;
 
 -(void)parse;
 -(void)addEntryWithDictionary:(NSMutableDictionary *)dict
