--- a/XADMaster/XADRegex.h
+++ b/XADMaster/XADRegex.h
@@ -20,8 +20,8 @@
  */
 #import <Foundation/Foundation.h>
 
-#ifdef _WIN32
-#import "regex.h"
+#ifndef __GLIBC__
+#import "xad_regex.h"
 #else
 #import <regex.h>
 #endif
@@ -106,4 +106,4 @@
 -(void)replaceEveryPattern:(NSString *)pattern usingSelector:(SEL)selector onObject:(id)object;
 -(void)replaceEveryPattern:(NSString *)pattern usingSelector:(SEL)selector onObject:(id)object options:(int)options;
 
-@end*/
\ No newline at end of file
+@end*/
