--- a/CMakeLists.txt	2024-08-04 15:07:13.016418509 +0200
+++ -	2024-08-04 15:11:35.036746347 +0200
@@ -332,7 +332,7 @@
 endif()
 
 if(ENABLE_G729 OR ENABLE_G729B_CNG)
-	find_package(BCG729 1.1.1)
+	find_package(Bcg729 1.1.1)
 	if(NOT BCG729_FOUND)
 		message(WARNING "Could not find bcg729, mediastreamer2 will be compiled without G729 codec and G729 AnnexB in RFC3389.")
 		set(ENABLE_G729 OFF CACHE BOOL "Build mediastreamer2 with the G729 codec." FORCE)
@@ -488,7 +488,7 @@
 	if(ENABLE_G729_CNG)
 		add_definitions("-DHAVE_G729B=1")
 	endif()
-	list(APPEND LINK_LIBS ${BCG729_TARGET})
+	list(APPEND LINK_LIBS bcg729)
 endif()
 if(GSM_FOUND)
 	list(APPEND LINK_LIBS ${GSM_TARGET})
