--- a/cmake/Mediastreamer2Config.cmake.in	2024-06-19 16:37:43.000000000 +0200
+++ -	2024-07-01 23:50:21.114340308 +0200
@@ -44,7 +44,7 @@
 include("${CMAKE_CURRENT_LIST_DIR}/Mediastreamer2Targets.cmake")
 
 set(Mediastreamer2_TARGET mediastreamer2)
-set_and_check(Mediastreamer2_PLUGINS_DIR "@PACKAGE_MS2_PLUGINS_DIR@")
+set(Mediastreamer2_PLUGINS_DIR "@PACKAGE_MS2_PLUGINS_DIR@")
 
 # We must propagate the public dependencies and the private dependencies for static build
 include(CMakeFindDependencyMacro)
