--- configure-master	2017-07-01 11:07:01.170862696 -0600
+++ configure	2017-07-01 11:17:37.613785126 -0600
@@ -6544,12 +6544,6 @@
 $as_echo_n "checking whether the compiler accepts -O2... " >&6; }
 	_cflags="$CFLAGS"
 	CFLAGS="$HITCH_CFLAGS -O2 $CFLAGS"
-	if test "$cross_compiling" = yes; then :
-  { { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
-$as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
-as_fn_error $? "cannot run test program while cross compiling
-See \`config.log' for more details" "$LINENO" 5; }
-else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 int main(void) { return (0); }
@@ -6563,7 +6557,6 @@
 fi
 rm -f core *.core core.conftest.* gmon.out bb.out conftest$ac_exeext \
   conftest.$ac_objext conftest.beam conftest.$ac_ext
-fi
 
 	CFLAGS="$_cflags"
 
@@ -6572,12 +6565,6 @@
 $as_echo_n "checking whether the compiler accepts -g... " >&6; }
 	_cflags="$CFLAGS"
 	CFLAGS="$HITCH_CFLAGS -g $CFLAGS"
-	if test "$cross_compiling" = yes; then :
-  { { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
-$as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
-as_fn_error $? "cannot run test program while cross compiling
-See \`config.log' for more details" "$LINENO" 5; }
-else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 int main(void) { return (0); }
@@ -6591,7 +6578,6 @@
 fi
 rm -f core *.core core.conftest.* gmon.out bb.out conftest$ac_exeext \
   conftest.$ac_objext conftest.beam conftest.$ac_ext
-fi
 
 	CFLAGS="$_cflags"
 
@@ -6600,12 +6586,6 @@
 $as_echo_n "checking whether the compiler accepts -Wall... " >&6; }
 	_cflags="$CFLAGS"
 	CFLAGS="$HITCH_CFLAGS -Wall $CFLAGS"
-	if test "$cross_compiling" = yes; then :
-  { { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
-$as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
-as_fn_error $? "cannot run test program while cross compiling
-See \`config.log' for more details" "$LINENO" 5; }
-else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 int main(void) { return (0); }
@@ -6619,7 +6599,6 @@
 fi
 rm -f core *.core core.conftest.* gmon.out bb.out conftest$ac_exeext \
   conftest.$ac_objext conftest.beam conftest.$ac_ext
-fi
 
 	CFLAGS="$_cflags"
 
@@ -6628,12 +6607,6 @@
 $as_echo_n "checking whether the compiler accepts -W... " >&6; }
 	_cflags="$CFLAGS"
 	CFLAGS="$HITCH_CFLAGS -W $CFLAGS"
-	if test "$cross_compiling" = yes; then :
-  { { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
-$as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
-as_fn_error $? "cannot run test program while cross compiling
-See \`config.log' for more details" "$LINENO" 5; }
-else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 int main(void) { return (0); }
@@ -6647,7 +6620,6 @@
 fi
 rm -f core *.core core.conftest.* gmon.out bb.out conftest$ac_exeext \
   conftest.$ac_objext conftest.beam conftest.$ac_ext
-fi
 
 	CFLAGS="$_cflags"
 
@@ -6664,12 +6636,6 @@
 $as_echo_n "checking whether the compiler accepts -fno-strict-aliasing... " >&6; }
 	_cflags="$CFLAGS"
 	CFLAGS="$EV_CFLAGS -fno-strict-aliasing $CFLAGS"
-	if test "$cross_compiling" = yes; then :
-  { { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
-$as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
-as_fn_error $? "cannot run test program while cross compiling
-See \`config.log' for more details" "$LINENO" 5; }
-else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 int main(void) { return (0); }
@@ -6683,7 +6649,6 @@
 fi
 rm -f core *.core core.conftest.* gmon.out bb.out conftest$ac_exeext \
   conftest.$ac_objext conftest.beam conftest.$ac_ext
-fi
 
 	CFLAGS="$_cflags"
 
