--- src/hitch.c.orig	2017-02-16 01:37:27.743263070 -0500
+++ src/hitch.c	2017-02-16 01:37:59.846392318 -0500
@@ -683,7 +683,7 @@
 		return NULL;
 	}
 
-#if OPENSSL_VERSION_NUMBER < 0x10100000L
+#if OPENSSL_VERSION_NUMBER < 0x10100000L || defined(LIBRESSL_VERSION_NUMBER)
 #define SSL_CTX_get_default_passwd_cb(ctx) (ctx->default_passwd_callback)
 #define SSL_CTX_get_default_passwd_cb_userdata(ctx) (ctx->default_passwd_callback_userdata)
 #endif
