Index: enblend-enfuse-4.2/configure.ac
===================================================================
--- enblend-enfuse-4.2.orig/configure.ac
+++ enblend-enfuse-4.2/configure.ac
@@ -287,15 +287,6 @@ AC_CHECK_FUNCS([atexit clock_gettime \
 AC_C99_FUNC_LRINT
 AC_C99_FUNC_LRINTF
 
-AC_MSG_CHECKING([for BOOST_FALLTHROUGH])
-AC_PREPROC_IFELSE([AC_LANG_PROGRAM([[#include <boost/config/suffix.hpp>]],
-                                   [[BOOST_FALLTHROUGH]])],
-                  [ac_found=yes; AC_MSG_RESULT([yes])],
-                  [ac_found=no]; AC_MSG_RESULT([no]))
-if test "$ac_found" = no; then
-    AC_DEFINE(BOOST_FALLTHROUGH, [((void) 0)],
-              [Define if Boost is not recent enough to define BOOST_FALLTHROUGH itself])
-fi
 
 AC_PATH_PROG(PERL, perl, false)
 if test "$PERL" = false; then
