From 7d3391fd6fe9279ec02abf28763a0249998d68d4 Mon Sep 17 00:00:00 2001
From: Patrick Kostas <patrick.kostas@mailbox.org>
Date: Fri, 11 Jun 2021 00:59:15 +0200
Subject: [PATCH 1/2] fix gcc11 compilation error

issue #118
---
 src/utils.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/utils.h b/src/utils.h
index a9ccc2a..3ccf50b 100644
--- a/src/utils.h
+++ b/src/utils.h
@@ -8,6 +8,7 @@
 
 #include <array>
 #include <vector>
+#include <cstddef>
 #include "stdint.h"
 
 int    calcBrightness(uint8_t *buf, uint64_t buf_sz, int bytes_per_pixel, int stride);

From 54bf2eeade2c792c63e35b0379c0a7dac70b0307 Mon Sep 17 00:00:00 2001
From: Patrick Kostas <patrick.kostas@mailbox.org>
Date: Sat, 12 Jun 2021 20:59:54 +0200
Subject: [PATCH 2/2] update header files inclusion

---
 src/utils.h | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/src/utils.h b/src/utils.h
index 3ccf50b..0088e34 100644
--- a/src/utils.h
+++ b/src/utils.h
@@ -6,10 +6,8 @@
 #ifndef UTILS_H
 #define UTILS_H
 
-#include <array>
-#include <vector>
 #include <cstddef>
-#include "stdint.h"
+#include <cstdint>
 
 int    calcBrightness(uint8_t *buf, uint64_t buf_sz, int bytes_per_pixel, int stride);
 double lerp(double x, double a, double b);
