--- a/conf/nginx.conf
+++ b/conf/nginx.conf
@@ -1,5 +1,5 @@
 
-#user  nobody;
+#user  nginx;
 worker_processes  1;
 
 #error_log  logs/error.log;
@@ -39,7 +39,7 @@
         #access_log  logs/host.access.log  main;
 
         location / {
-            root   html;
+            root   /usr/share/nginx/html;
             index  index.html index.htm;
         }
 
@@ -49,7 +49,7 @@
         #
         error_page   500 502 503 504  /50x.html;
         location = /50x.html {
-            root   html;
+            root   /usr/share/nginx/html;
         }
 
         # proxy the PHP scripts to Apache listening on 127.0.0.1:80
@@ -61,7 +61,7 @@
         # pass the PHP scripts to FastCGI server listening on 127.0.0.1:9000
         #
         #location ~ \.php$ {
-        #    root           html;
+        #    root           /usr/share/nginx/html;
         #    fastcgi_pass   127.0.0.1:9000;
         #    fastcgi_index  index.php;
         #    fastcgi_param  SCRIPT_FILENAME  /scripts$fastcgi_script_name;
@@ -85,7 +85,7 @@
     #    server_name  somename  alias  another.alias;
 
     #    location / {
-    #        root   html;
+    #        root   /usr/share/nginx/html;
     #        index  index.html index.htm;
     #    }
     #}
@@ -107,7 +107,7 @@
     #    ssl_prefer_server_ciphers  on;
 
     #    location / {
-    #        root   html;
+    #        root   /usr/share/nginx/html;
     #        index  index.html index.htm;
     #    }
     #}
