diff --git a/desktop/CMakeLists.txt b/desktop/CMakeLists.txt
index 2ee308c..8dc2d03 100644
--- a/desktop/CMakeLists.txt
+++ b/desktop/CMakeLists.txt
@@ -8,7 +8,7 @@ if(NOT RSVG_CONVERT)
     message(WARNING "rsvg-convert not found; desktop icons will not be generated")
 endif()
 
-find_program(XMLSTARLET xmlstarlet)
+find_program(XMLSTARLET xml)
 if(NOT XMLSTARLET)
     message(WARNING "xmlstarlet not found; desktop icons will not be generated")
 endif()
