diff --git a/Makefile b/Makefile
index 06cbaf2..764caab 100644
--- a/Makefile
+++ b/Makefile
@@ -251,6 +251,9 @@ ifeq ($(WINDOWS_STATIC),1)
  CPPFLAGS += -DSFML_STATIC $(shell $(PKG_CONFIG) --static --cflags $(SFML_PC))
  FE_WINDOWS_COMPILE=1
 
+else ifeq ($(shell $(PKG_CONFIG) --exists sfml-all && echo "1" || echo "0"), 1)
+  TEMP_LIBS += sfml-all
+  CPPFLAGS += $(shell $(PKG_CONFIG) --cflags sfml-all)
 else
   TEMP_LIBS += sfml-graphics sfml-window sfml-system
 endif
@@ -459,6 +462,7 @@ endif
 CPPFLAGS += -D__STDC_CONSTANT_MACROS
 
 LIBS := $(LIBS) $(shell $(PKG_CONFIG) --libs $(TEMP_LIBS))
+LIBS += -lbz2 -lcrypto -lzstd -llz4 -llzma -lacl
 CPPFLAGS := $(CPPFLAGS) $(shell $(PKG_CONFIG) --cflags $(TEMP_LIBS))
 
 EXE = $(EXE_BASE)$(EXE_EXT)
