--- a/extlibs/gameswf/base/utility.cpp	2016-11-08 08:06:21.000000000 +0100
+++ b/extlibs/gameswf/base/utility.cpp	2016-12-06 22:03:53.042131946 +0100
@@ -41,7 +41,7 @@
 {
 
 // This doesn't compile on Windows.
-#if  !defined(_WIN32) && !defined(__APPLE_CC__)
+#if  !defined(_WIN32) && !defined(__APPLE_CC__) && defined(__GLIBC__)
   
 // This does not work with DMALLOC, since the internal data structures
 // differ.
