diff --git a/Makefile b/Makefile
index 4f4793a..82387e0 100644
--- a/Makefile
+++ b/Makefile
@@ -69,15 +69,7 @@
 
 FE_VERSION=v2.7.0
 
-CC=gcc
-CXX=g++
-CFLAGS=
-CXXFLAGS=-std=c++11 $(EXTRA_CXXFLAGS)
-CPPFLAGS=
-STRIP=strip
-OBJCOPY=objcopy
 PKG_CONFIG=pkg-config
-AR=ar
 ARFLAGS=rc
 RM=rm -f
 MD=mkdir -p
@@ -308,7 +300,7 @@ ifeq ($(FE_DEBUG),1)
  FE_FLAGS += -DFE_DEBUG
  ENABLE_DEBUG_SYMBOLS=1
 else
- CPPFLAGS += -O$(OPTIMIZE) -DNDEBUG
+ CPPFLAGS += -DNDEBUG
 endif
 
 ifeq ($(USE_GLES),1)
@@ -512,7 +504,7 @@ $(OBJ_DIR)/%.o: $(SRC_DIR)/%.mm $(DEP) | $(OBJ_DIR)
 
 $(EXE): $(OBJ) $(EXPAT) $(SQUIRREL) $(AUDIO)
 	$(EXE_MSG)
-	$(CXX) -o $@ $^ $(CXXFLAGS) $(CPPFLAGS) $(FE_FLAGS) $(LIBS)
+	$(CXX) -o $@ $^ $(LDFLAGS) $(CXXFLAGS) $(CPPFLAGS) $(FE_FLAGS) $(LIBS)
 ifeq ($(ENABLE_DEBUG_SYMBOLS),1)
 	$(DEBUG_MSG)
 	$(SILENT)$(OBJCOPY) --only-keep-debug $@ $@.debug
