diff --git a/Makefile.orig b/Makefile
index 4f4793a9a69..b1f7cf469f4 100644
--- a/Makefile.orig
+++ b/Makefile
@@ -69,15 +69,10 @@
 
 FE_VERSION=v2.7.0
 
-CC=gcc
-CXX=g++
 CFLAGS=
 CXXFLAGS=-std=c++11 $(EXTRA_CXXFLAGS)
 CPPFLAGS=
-STRIP=strip
-OBJCOPY=objcopy
 PKG_CONFIG=pkg-config
-AR=ar
 ARFLAGS=rc
 RM=rm -f
 MD=mkdir -p
@@ -512,7 +507,7 @@ $(OBJ_DIR)/%.o: $(SRC_DIR)/%.mm $(DEP) | $(OBJ_DIR)
 
 $(EXE): $(OBJ) $(EXPAT) $(SQUIRREL) $(AUDIO)
 	$(EXE_MSG)
-	$(CXX) -o $@ $^ $(CXXFLAGS) $(CPPFLAGS) $(FE_FLAGS) $(LIBS)
+	$(CXX) -o $@ $^ $(LDFLAGS) $(CXXFLAGS) $(CPPFLAGS) $(FE_FLAGS) $(LIBS)
 ifeq ($(ENABLE_DEBUG_SYMBOLS),1)
 	$(DEBUG_MSG)
 	$(SILENT)$(OBJCOPY) --only-keep-debug $@ $@.debug
