--- shlr/Makefile.orig	2013-11-11 11:33:05.197480920 +0100
+++ shlr/Makefile	2013-11-11 11:33:14.263944850 +0100
@@ -23,7 +23,7 @@ $(SDBLIB) sdb/sdb:
 	cd sdb ; ${MAKE} clean
 	cd sdb ; \
 		${MAKE} clean && \
-		${MAKE} CC=${HOST_CC} && \
+		${MAKE} CC=${HOST_CC} CFLAGS="${HOST_CFLAGS}" && \
 		cp -f src/sdb src/.sdb
 	cd sdb ; ${MAKE} clean
 	cd sdb ; ${MAKE} src/sdb-version.h
--- configure.orig	2013-11-11 11:39:27.476221882 +0100
+++ configure	2013-11-11 11:39:51.791056853 +0100
@@ -36,7 +36,7 @@ while : ; do
 ENVWORDS="${ENVWORDS} $1_CPU $1_OS"
 STR=`eval "echo ${S}$1"`
 SPLIT_CPU="`echo "$STR" | cut -d - -f 1`"
-SPLIT_OS="`echo "$STR" | awk -F - '{if(NF==4){print $4}else{print $3}}'`"
+SPLIT_OS="`echo "$STR" | awk -F - '{print $(NF-1)}'`"
 eval "$1_CPU=\"$SPLIT_CPU\""
 eval "$1_OS=\"$SPLIT_OS\""
 shift
