diff -ru configure configure
--- configure	2013-03-27 12:41:35.959376909 +0100
+++ configure	2013-03-27 12:41:55.542960992 +0100
@@ -36,7 +36,7 @@
 ENVWORDS="${ENVWORDS} $1_CPU $1_OS"
 STR=`eval "echo ${S}$1"`
 SPLIT_CPU="`echo "$STR" | cut -d - -f 1`"
-SPLIT_OS="`echo "$STR" | awk -F - '{if(NF==4){print $4}else{print $3}}'`"
+SPLIT_OS="`echo "$STR" | awk -F - '{if(NF==4){print $4}else{print $2}}'`"
 eval "$1_CPU=\"$SPLIT_CPU\""
 eval "$1_OS=\"$SPLIT_OS\""
 shift
diff -ru libr/syscall/d/Makefile libr/syscall/d/Makefile
--- libr/syscall/d/Makefile	2013-03-27 12:41:35.948377143 +0100
+++ libr/syscall/d/Makefile	2013-03-27 12:42:15.245542518 +0100
@@ -39,7 +39,7 @@
 ${SDB}:
 	cp -rf $(LTOP)/db/sdb/src $(LTOP)/db/sdb/src-native
 	#cd $(LTOP)/db/sdb/src-native && ${MAKE} -j ${MAKE_JOBS} ARCH=xxx CC=gcc clean sdb
-	cd $(LTOP)/db/sdb/src-native && ${MAKE} -j 1 ARCH=xxx CC=gcc clean sdb
+	cd $(LTOP)/db/sdb/src-native && CFLAGS='' ${MAKE} -j 1 ARCH=xxx CC=gcc clean sdb
 	cp -f $(LTOP)/db/sdb/src-native/sdb $(LTOP)/db/sdb/sdb
 	rm -rf $(LTOP)/db/sdb/src-native
 
