From bdaff5a9fdd628482cd03a384a8804f228eecf9a Mon Sep 17 00:00:00 2001
From: ypsvlq <86775241+ypsvlq@users.noreply.github.com>
Date: Tue, 10 May 2022 08:24:14 +0100
Subject: [PATCH] Fix meson build with use_sys_openssl ##build

---
 libr/socket/meson.build | 1 -
 1 file changed, 1 deletion(-)

diff --git a/libr/socket/meson.build b/libr/socket/meson.build
index 4e8614e9e57b..135c7b50efb9 100644
--- a/libr/socket/meson.build
+++ b/libr/socket/meson.build
@@ -20,7 +20,6 @@ endif
 
 if use_sys_openssl
   r_util_deps += [sys_openssl]
-  r_util_static_deps += [sys_openssl]
 endif
 
 r_socket = library('r_socket', r_socket_sources,
