--- configure.orig
+++ configure
@@ -330,10 +330,6 @@
 COMPILER=CC
 printf "checking for c compiler... "
 HAVE_LANG_C=1
-if [ "${CROSSBUILD}" = 1 ]; then
- (command -v ${HOST}-${CC} >/dev/null 2>&1)
- if [ $? = 0 ]; then CC="${HOST}-${CC}"; fi
-fi
 echo "int main(int argc, char **argv){return 0;}" > test.c
  (exec ${CC} ${CFLAGS} ${CPPFLAGS} ${LDFLAGS} test.c >/dev/null 2>&1)
 if [ $? = 0 ]; then echo ${CC}; else
