--- a/external_libraries/link/modules/asio-standalone/asio/include/asio/impl/error_code.ipp	2020-10-26 23:22:21.954194075 +0700
+++ b/external_libraries/link/modules/asio-standalone/asio/include/asio/impl/error_code.ipp	2020-10-26 23:25:46.716820740 +0700
@@ -106,9 +106,14 @@
     using namespace std;
     strerror_r(value, buf, sizeof(buf));
     return buf;
-#else
+#elif defined(__GLIBC__)
     char buf[256] = "";
     return strerror_r(value, buf, sizeof(buf));
+#else
+    char buf[256] = "";
+    using namespace std;
+    strerror_r(value, buf, sizeof(buf));
+    return buf;
 #endif
 #endif // defined(ASIO_WINDOWS_DESKTOP) || defined(__CYGWIN__)
   }
