--- QtCollider/primitives/prim_QPen.cpp	2020-01-17 02:32:26.000000000 +0100
+++ QtCollider/primitives/prim_QPen.cpp	2020-09-02 15:48:21.747713623 +0200
@@ -27,6 +27,7 @@
 #include "PyrKernel.h"
 
 #include <QPainter>
+#include <QPainterPath>
 #include <QVector2D>
 #include <QVector3D>
 #include <cmath>
--- QtCollider/widgets/QcGraph.cpp	2020-01-17 02:32:26.000000000 +0100
+++ QtCollider/widgets/QcGraph.cpp	2020-09-02 15:48:13.248713184 +0200
@@ -24,6 +24,7 @@
 #include "../style/routines.hpp"
 
 #include <QPainter>
+#include <QPainterPath>
 #include <QMouseEvent>
 #include <QApplication>
 #include <QtCore/qmath.h>
--- QtCollider/widgets/QcGraph.h	2020-01-17 02:32:26.000000000 +0100
+++ QtCollider/widgets/QcGraph.h	2020-09-02 15:47:36.817711305 +0200
@@ -26,6 +26,7 @@
 #include "../style/style.hpp"
 
 #include <QWidget>
+#include <QPainterPath>
 
 struct QcGraphElement {
     friend class QcGraphModel;
--- QtCollider/widgets/QcLevelIndicator.cpp	2020-01-17 02:32:26.000000000 +0100
+++ QtCollider/widgets/QcLevelIndicator.cpp	2020-09-02 15:47:52.371712107 +0200
@@ -23,6 +23,7 @@
 #include "../QcWidgetFactory.h"
 
 #include <QPainter>
+#include <QPainterPath>
 
 QC_DECLARE_QWIDGET_FACTORY(QcLevelIndicator);
 
--- QtCollider/widgets/QcMultiSlider.cpp	2020-01-17 02:32:26.000000000 +0100
+++ QtCollider/widgets/QcMultiSlider.cpp	2020-09-02 15:47:59.289712464 +0200
@@ -26,6 +26,7 @@
 #include <QApplication>
 #include <QMouseEvent>
 #include <QPainter>
+#include <QPainterPath>
 
 #include <cmath>
 
--- QtCollider/widgets/QcScopeShm.cpp	2020-01-17 02:32:26.000000000 +0100
+++ QtCollider/widgets/QcScopeShm.cpp	2020-09-02 15:47:45.719711764 +0200
@@ -25,6 +25,7 @@
 #include "../debug.h"
 
 #include <QPainter>
+#include <QPainterPath>
 #include <QTimer>
 #include <QResizeEvent>
 #include <QWindow>
--- QtCollider/widgets/soundfileview/view.cpp	2020-01-17 02:32:26.000000000 +0100
+++ QtCollider/widgets/soundfileview/view.cpp	2020-09-02 15:48:06.605712841 +0200
@@ -24,6 +24,7 @@
 
 #include <QGridLayout>
 #include <QPainter>
+#include <QPainterPath>
 #include <QApplication>
 #include <QPaintEvent>
 #include <QCursor>
