--- CMakeLists.txt.orig	2017-10-06 13:15:21.282961157 +0200
+++ CMakeLists.txt	2017-10-06 13:17:15.567953674 +0200
@@ -462,61 +462,61 @@
 #
 
 # CTPP Compiler
-ADD_EXECUTABLE(ctpp2c                       tests/CTPP2Compiler.cpp)
-TARGET_LINK_LIBRARIES(ctpp2c                ctpp2)
+ADD_EXECUTABLE(ctpp2c                       tests/CTPP2Compiler.cpp)
+TARGET_LINK_LIBRARIES(ctpp2c                ctpp2)
 
 # CTPP2 Interpreter
-ADD_EXECUTABLE(ctpp2i                       tests/CTPP2Interpreter.cpp)
-TARGET_LINK_LIBRARIES(ctpp2i                ctpp2)
+ADD_EXECUTABLE(ctpp2i                       tests/CTPP2Interpreter.cpp)
+TARGET_LINK_LIBRARIES(ctpp2i                ctpp2)
 
 # CTPP2 virtual machine
-ADD_EXECUTABLE(ctpp2vm                      tests/CTPP2Emulator.cpp)
-TARGET_LINK_LIBRARIES(ctpp2vm               ctpp2)
+ADD_EXECUTABLE(ctpp2vm                      tests/CTPP2Emulator.cpp)
+TARGET_LINK_LIBRARIES(ctpp2vm               ctpp2)
 
 # CTPP2 JSON parser
-ADD_EXECUTABLE(ctpp2json                    tests/CTPP2JSONParser.cpp)
-TARGET_LINK_LIBRARIES(ctpp2json             ctpp2)
+ADD_EXECUTABLE(ctpp2json                    tests/CTPP2JSONParser.cpp)
+TARGET_LINK_LIBRARIES(ctpp2json             ctpp2)
 
 # ################################################################################
 # Test suite
 #
 ENABLE_TESTING()
 
-ADD_EXECUTABLE(CDTTest                      tests/CDTTest.cpp)
-TARGET_LINK_LIBRARIES(CDTTest               ctpp2)
+#ADD_EXECUTABLE(CDTTest                      tests/CDTTest.cpp)
+#TARGET_LINK_LIBRARIES(CDTTest               ctpp2)
 
-ADD_EXECUTABLE(CDTPerfTest                  tests/CDTPerfTest.cpp)
-TARGET_LINK_LIBRARIES(CDTPerfTest           ctpp2)
+#ADD_EXECUTABLE(CDTPerfTest                  tests/CDTPerfTest.cpp)
+#TARGET_LINK_LIBRARIES(CDTPerfTest           ctpp2)
 
-ADD_EXECUTABLE(BitIndexTest                 tests/BitIndexText.cpp)
-TARGET_LINK_LIBRARIES(BitIndexTest          ctpp2)
+#ADD_EXECUTABLE(BitIndexTest                 tests/BitIndexText.cpp)
+#TARGET_LINK_LIBRARIES(BitIndexTest          ctpp2)
 
-ADD_EXECUTABLE(HashTest                     tests/HashTest.cpp)
-TARGET_LINK_LIBRARIES(HashTest              ctpp2)
+#ADD_EXECUTABLE(HashTest                     tests/HashTest.cpp)
+#TARGET_LINK_LIBRARIES(HashTest              ctpp2)
 
-ADD_EXECUTABLE(StaticTextTest               tests/StaticTextTest.cpp)
-TARGET_LINK_LIBRARIES(StaticTextTest        ctpp2)
+#ADD_EXECUTABLE(StaticTextTest               tests/StaticTextTest.cpp)
+#TARGET_LINK_LIBRARIES(StaticTextTest        ctpp2)
 
 #ADD_EXECUTABLE(StaticDataTest               tests/StaticDataTest.cpp)
 #TARGET_LINK_LIBRARIES(StaticDataTest        ctpp2)
 
-ADD_EXECUTABLE(VMArgStackTest               tests/VMArgStackTest.cpp)
-TARGET_LINK_LIBRARIES(VMArgStackTest        ctpp2)
+#ADD_EXECUTABLE(VMArgStackTest               tests/VMArgStackTest.cpp)
+#TARGET_LINK_LIBRARIES(VMArgStackTest        ctpp2)
 
-ADD_EXECUTABLE(VMCodeStackTest              tests/VMCodeStackTest.cpp)
-TARGET_LINK_LIBRARIES(VMCodeStackTest       ctpp2)
+#ADD_EXECUTABLE(VMCodeStackTest              tests/VMCodeStackTest.cpp)
+#TARGET_LINK_LIBRARIES(VMCodeStackTest       ctpp2)
 
-ADD_EXECUTABLE(CTPP2VMTest                  tests/CTPP2VMTest.cpp)
-TARGET_LINK_LIBRARIES(CTPP2VMTest           ctpp2)
+#ADD_EXECUTABLE(CTPP2VMTest                  tests/CTPP2VMTest.cpp)
+#TARGET_LINK_LIBRARIES(CTPP2VMTest           ctpp2)
 
-ADD_EXECUTABLE(CTPP2DTOATest                tests/CTPP2DTOA.cpp)
-TARGET_LINK_LIBRARIES(CTPP2DTOATest         ctpp2)
+#ADD_EXECUTABLE(CTPP2DTOATest                tests/CTPP2DTOA.cpp)
+#TARGET_LINK_LIBRARIES(CTPP2DTOATest         ctpp2)
 
-ADD_EXECUTABLE(simple-compiler              tests/SimpleCompilerTest.cpp)
-TARGET_LINK_LIBRARIES(simple-compiler       ctpp2)
+#ADD_EXECUTABLE(simple-compiler              tests/SimpleCompilerTest.cpp)
+#TARGET_LINK_LIBRARIES(simple-compiler       ctpp2)
 
-ADD_EXECUTABLE(simple-vm                    tests/SimpleVMTest.cpp)
-TARGET_LINK_LIBRARIES(simple-vm             ctpp2)
+#ADD_EXECUTABLE(simple-vm                    tests/SimpleVMTest.cpp)
+#TARGET_LINK_LIBRARIES(simple-vm             ctpp2)
 
 ADD_TEST(CDT_performance_test               CDTPerfTest)
 ADD_TEST(CDT_ops_test                       CDTTest)
--- CMakeLists.txt.orig	2017-10-06 13:17:51.168951343 +0200
+++ CMakeLists.txt	2017-10-06 13:18:07.934950245 +0200
@@ -530,8 +530,8 @@
 ADD_TEST(VM_self_test                       ctpp2vm selftest.ct2)
 SET_TESTS_PROPERTIES(VM_self_test PROPERTIES DEPENDS Create_executable_file)
 
-ADD_EXECUTABLE(CDT2JSONBenchmark            benchmarks/CDT2JSON.cpp)
-TARGET_LINK_LIBRARIES(CDT2JSONBenchmark     ctpp2)
+#ADD_EXECUTABLE(CDT2JSONBenchmark            benchmarks/CDT2JSON.cpp)
+#TARGET_LINK_LIBRARIES(CDT2JSONBenchmark     ctpp2)
 
 ADD_TEST(StreamLikeJsonDump_1                 CDT2JSONBenchmark -t ${CMAKE_CURRENT_SOURCE_DIR}/benchmarks/data/test.json)
 ADD_TEST(StreamLikeJsonDump_2                 CDT2JSONBenchmark -t ${CMAKE_CURRENT_SOURCE_DIR}/benchmarks/data/lebowski-bench.json)
--- CMakeLists.txt.orig	2017-10-06 13:18:43.220947934 +0200
+++ CMakeLists.txt	2017-10-06 13:19:29.463944906 +0200
