--- a/mojo/public/cpp/platform/named_platform_channel_posix.cc.orig       2020-07-19 13:26:10.696171063 -0400
+++ b/mojo/public/cpp/platform/named_platform_channel_posix.cc    2020-07-19 13:26:43.680151714 -0400
@@ -5,6 +5,7 @@
 #include "mojo/public/cpp/platform/named_platform_channel.h"
 
 #include <errno.h>
+#include <string.h>
 #include <sys/socket.h>
 #include <sys/un.h>
 #include <unistd.h>
