--- a/clutter/clutter/meson.build
+++ b/clutter/clutter/meson.build
@@ -487,7 +487,7 @@
   soversion: 0,
   c_args: clutter_c_args,
   include_directories: clutter_includes,
-  dependencies: [clutter_deps],
+  dependencies: [clutter_deps,libmutter_cogl_path_dep],
   gnu_symbol_visibility: 'hidden',
   link_with: [
     libmutter_cogl,
@@ -499,7 +499,7 @@
   install: true,
 )
 libmutter_clutter_dep = declare_dependency(
-  sources: [clutter_enum_types[1]],
+  sources: [clutter_enum_types[1],libmutter_cogl_path_enum_types[1]],
   link_with: libmutter_clutter,
 )
