--- cogl/cogl/meson.build	2019-10-08 19:15:10.000000000 +0200
+++ cogl/cogl/meson.build	2019-12-16 15:45:44.037640548 +0100
@@ -48,7 +48,7 @@
 built_headers += [cogl_gl_header_h]
 
 if have_egl
-  cogl_egl_includes_string = '#include <EGL/egl.h>\n#include <EGL/eglext.h>'
+  cogl_egl_includes_string = '#include <EGL/egl.h>\n#include <EGL/eglext.h>\n#include <EGL/eglmesaext.h>'
 else
   cogl_egl_includes_string = ''
 endif
