diff --git a/clutter/clutter/meson.build b/clutter/clutter/meson.build
index 44ee1a315e..7f8f5d8534 100644
--- a/clutter/clutter/meson.build
+++ b/clutter/clutter/meson.build
@@ -440,7 +440,7 @@ libmutter_clutter = shared_library(libmutter_clutter_name,
   soversion: 0,
   c_args: clutter_c_args,
   include_directories: clutter_includes,
-  dependencies: [clutter_deps],
+  dependencies: [clutter_deps,libmutter_cogl_path_dep],
   gnu_symbol_visibility: 'hidden',
   link_with: [
     libmutter_cogl,
@@ -452,7 +452,7 @@ libmutter_clutter = shared_library(libmutter_clutter_name,
   install: true,
 )
 libmutter_clutter_dep = declare_dependency(
-  sources: [clutter_enum_types[1]],
+  sources: [clutter_enum_types[1],libmutter_cogl_path_enum_types[1]],
   link_with: libmutter_clutter,
 )
 
