https://github.com/mesonbuild/meson/issues/5050

--- clutter/clutter/meson.build
+++ clutter/clutter/meson.build
@@ -577,16 +577,16 @@ if have_introspection
 endif
 
 install_headers(clutter_headers,
-  subdir: clutter_clutter_includedir)
+  install_dir: clutter_clutter_includedir)
 
 install_headers(cally_headers,
-  subdir: join_paths(clutter_includedir, 'cally'))
+  install_dir: join_paths(clutter_includedir, 'cally'))
 
 install_headers(clutter_deprecated_headers,
-  subdir: join_paths(clutter_clutter_includedir, 'deprecated'))
+  install_dir: join_paths(clutter_clutter_includedir, 'deprecated'))
 
 install_headers(clutter_x11_headers,
-  subdir: join_paths(clutter_clutter_includedir, 'x11'))
+  install_dir: join_paths(clutter_clutter_includedir, 'x11'))
 
 pkg.generate(libmutter_clutter,
   name: 'Mutters Clutter',
--- cogl/cogl-gles2/meson.build
+++ cogl/cogl-gles2/meson.build
@@ -29,7 +29,7 @@ libmutter_cogl_gles2 = shared_library('mutter-cogl-gles2-' + libmutter_api_versi
 
 cogl_gles2_includedir = join_paths(cogl_includedir, 'cogl-gles2/GLES2')
 install_headers(cogl_gles2_public_headers,
-  subdir: cogl_gles2_includedir)
+  install_dir: cogl_gles2_includedir)
 
 pkg.generate(libmutter_cogl_gles2,
   name: 'CoglGles2',
--- cogl/cogl-pango/meson.build
+++ cogl/cogl-pango/meson.build
@@ -71,7 +71,7 @@ if have_introspection
 endif
 
 cogl_pango_includedir = join_paths(cogl_includedir, 'cogl-pango')
-install_headers(cogl_pango_public_headers, subdir: cogl_pango_includedir)
+install_headers(cogl_pango_public_headers, install_dir: cogl_pango_includedir)
 
 pkg.generate(libmutter_cogl_pango,
   name: 'CoglPango',
--- cogl/cogl-path/meson.build
+++ cogl/cogl-path/meson.build
@@ -76,7 +76,7 @@ libmutter_cogl_path_dep = declare_dependency(
 )
 
 install_headers(cogl_path_public_headers,
-  subdir: cogl_path_includedir)
+  install_dir: cogl_path_includedir)
 
 pkg.generate(libmutter_cogl_path,
   name: 'CoglPath',
--- cogl/cogl/meson.build
+++ cogl/cogl/meson.build
@@ -514,16 +514,16 @@ install_headers([
     cogl_headers,
     cogl_nonintrospected_headers,
   ],
-  subdir: cogl_cogl_includedir)
+  install_dir: cogl_cogl_includedir)
 
 install_headers([
     cogl_deprecated_headers,
     cogl_deprecated_nonintrospected_headers,
   ],
-  subdir: join_paths(cogl_cogl_includedir, 'deprecated'))
+  install_dir: join_paths(cogl_cogl_includedir, 'deprecated'))
 
 install_headers(cogl_gl_prototype_headers,
-  subdir: join_paths(cogl_cogl_includedir, 'gl-prototypes'))
+  install_dir: join_paths(cogl_cogl_includedir, 'gl-prototypes'))
 
 pkg.generate(libmutter_cogl,
   name: 'Cogl',
--- src/meta/meson.build
+++ src/meta/meson.build
@@ -48,7 +48,7 @@ if have_x11
 endif
 
 install_headers(mutter_public_headers,
-  subdir: mutter_includedir
+  install_dir: mutter_includedir
 )
 
 mutter_public_header_files = files(mutter_public_headers)
