--- src/backends/x11/meta-backend-x11.c	2017-05-11 06:30:09.000000000 -0600
+++ src/backends/x11/meta-backend-x11.c	2017-06-05 12:10:43.010474780 -0600
@@ -612,12 +612,6 @@
 static void
 meta_backend_x11_init (MetaBackendX11 *x11)
 {
-  /* XInitThreads() is needed to use the "threaded swap wait" functionality
-   * in Cogl - see meta_renderer_x11_create_cogl_renderer(). We call it here
-   * to hopefully call it before any other use of XLib.
-   */
-  XInitThreads();
-
   clutter_x11_request_reset_on_video_memory_purge ();
 
   /* We do X11 event retrieval ourselves */
