diff --git a/arch/powerpc/platforms/book3s/Kconfig b/arch/powerpc/platforms/book3s/Kconfig
index 34c9315..88f4f87 100644
--- a/arch/powerpc/platforms/book3s/Kconfig
+++ b/arch/powerpc/platforms/book3s/Kconfig
@@ -1,7 +1,7 @@
 # SPDX-License-Identifier: GPL-2.0
 config PPC_VAS
 	bool "IBM Virtual Accelerator Switchboard (VAS)"
-	depends on (PPC_POWERNV || PPC_PSERIES) && PPC_64K_PAGES
+	depends on PPC_POWERNV || PPC_PSERIES
 	default y
 	help
 	  This enables support for IBM Virtual Accelerator Switchboard (VAS).
