--- src/seccomp.c.orig	2015-05-28 20:49:40.000000000 +0200
+++ src/seccomp.c	2017-02-27 20:02:02.490783181 +0100
@@ -48,6 +48,8 @@
 #   define EM_ARM 40
 # endif
 #  define SECCOMP_AUDIT_ARCH AUDIT_ARCH_ARM
+#elif defined(__aarch64__)
+#  define SECCOMP_AUDIT_ARCH AUDIT_ARCH_AARCH64
 #else
 #  error "Platform does not support seccomp filter yet"
 #endif
@@ -87,7 +89,9 @@
     SC_ALLOW (exit_group),
     SC_ALLOW (exit),
 
+#ifdef __NR_open
     SC_DENY (open, EINVAL),
+#endif
     SC_DENY (fcntl, EINVAL),
     SC_DENY (fstat, EINVAL),
 #ifdef __NR_mmap
