--- a/configure.ac
+++ b/configure.ac
@@ -147,7 +147,9 @@ if test $GPGME_CONFIG != "failed" ; then
 	micro=`echo $gpgme_config_version | \
 		sed 's/\([[0-9]]*\)\.\([[0-9]]*\)\.\([[0-9]]*\).*/\3/'`
 
-	if test "$major" -eq "$req_major"; then
+	if test "$major" -gt "$req_major"; then
+		ok=yes
+	elif test "$major" -eq "$req_major"; then
 		if test "$minor" -ge "$req_minor"; then
 			if test "$micro" -ge "$req_micro"; then
 				ok="yes"
--- a/daemon/seahorse-gpgme-operation.c
+++ b/daemon/seahorse-gpgme-operation.c
@@ -302,7 +302,6 @@ event_cb (void *data, gpgme_event_io_t t
         break;
     
     case GPGME_EVENT_NEXT_KEY:
-    case GPGME_EVENT_NEXT_TRUSTITEM:
     default:
         /* Ignore unsupported event types */
         break;
