--- a/apps/openmw/mwgui/controllers.hpp
+++ b/apps/openmw/mwgui/controllers.hpp
@@ -1,6 +1,7 @@
 #ifndef MWGUI_CONTROLLERS_H
 #define MWGUI_CONTROLLERS_H

+#include <cstdint>
 #include <string>
 #include <MyGUI_ControllerItem.h>

--- a/apps/openmw/mwgui/itemselection.hpp
+++ b/apps/openmw/mwgui/itemselection.hpp
@@ -1,6 +1,7 @@
 #ifndef OPENMW_GAME_MWGUI_ITEMSELECTION_H
 #define OPENMW_GAME_MWGUI_ITEMSELECTION_H

+#include <cstdint>
 #include <MyGUI_Delegate.h>

 #include "windowbase.hpp"
--- a/apps/openmw/mwgui/timeadvancer.hpp
+++ b/apps/openmw/mwgui/timeadvancer.hpp
@@ -1,6 +1,7 @@
 #ifndef MWGUI_TIMEADVANCER_H
 #define MWGUI_TIMEADVANCER_H

+#include <cstdint>
 #include <MyGUI_Delegate.h>

 namespace MWGui

--- a/apps/openmw/mwgui/widgets.hpp
+++ b/apps/openmw/mwgui/widgets.hpp
@@ -1,6 +1,7 @@
 #ifndef MWGUI_WIDGETS_H
 #define MWGUI_WIDGETS_H

+#include <cstdint>
 #include "../mwmechanics/stat.hpp"

 #include <components/esm/effectlist.hpp>
--- a/apps/openmw/mwinput/controlswitch.hpp
+++ b/apps/openmw/mwinput/controlswitch.hpp
@@ -1,6 +1,7 @@
 #ifndef MWINPUT_CONTROLSWITCH_H
 #define MWINPUT_CONTROLSWITCH_H

+#include <cstdint>
 #include <map>
 #include <string>

--- a/components/misc/utf8stream.hpp
+++ b/components/misc/utf8stream.hpp
@@ -1,6 +1,7 @@
 #ifndef MISC_UTF8ITER_HPP
 #define MISC_UTF8ITER_HPP

+#include <cstdint>
 #include <cstring>
 #include <string>
 #include <string_view>
--- a/components/myguiplatform/myguidatamanager.cpp
+++ b/components/myguiplatform/myguidatamanager.cpp
@@ -1,5 +1,6 @@
 #include "myguidatamanager.hpp"

+#include <cstdint>
 #include <memory>
 #include <string>
 #include <stdexcept>
