diff --git a/test/Makefile.in b/test/Makefile.in
index b2c8d43..5747310 100644
--- a/test/Makefile.in
+++ b/test/Makefile.in
@@ -13,9 +13,7 @@ one: venv/bin/pytest fakekey
 fakekey:
 	../dropbearkey -t ecdsa -f $@
 
-venv/bin/pytest: $(srcdir)/requirements.txt
+venv/bin/pytest:
 	python3 -m venv init venv
-	./venv/bin/pip install --upgrade pip
-	./venv/bin/pip install -r $(srcdir)/requirements.txt
 
 .PHONY: test
