--- docs/libs/Makefile.in.orig	2010-10-24 18:17:50.476548975 +0200
+++ docs/libs/Makefile.in	2010-10-24 18:18:01.931680444 +0200
@@ -955,7 +955,7 @@ dist-hook: dist-check-gtkdoc dist-hook-l
 	-cp $(srcdir)/$(DOC_MODULE).types $(distdir)/
 	-cp $(srcdir)/$(DOC_MODULE)-sections.txt $(distdir)/
 	cd $(distdir) && rm -f $(DISTCLEANFILES)
-        -gtkdoc-rebase --online --relative --html-dir=$(distdir)/html
+	-gtkdoc-rebase --online --relative --html-dir=$(distdir)/html
 
 .PHONY : dist-hook-local docs
 
--- docs/plugins/Makefile.in.orig	2010-10-24 18:20:51.027378905 +0200
+++ docs/plugins/Makefile.in	2010-10-24 18:21:03.513611634 +0200
@@ -1101,7 +1101,7 @@ dist-hook: dist-check-gtkdoc dist-hook-l
 	-cp $(srcdir)/$(DOC_MODULE).types $(distdir)/
 	-cp $(srcdir)/$(DOC_MODULE)-sections.txt $(distdir)/
 	cd $(distdir) && rm -f $(DISTCLEANFILES)
-        -gtkdoc-rebase --online --relative --html-dir=$(distdir)/html
+	-gtkdoc-rebase --online --relative --html-dir=$(distdir)/html
 
 .PHONY : dist-hook-local docs
 
