--- a/src/chm_lib.c	2009-05-23 08:43:31.000000000 -0600
+++ b/src/chm_lib.c	2011-01-21 17:42:45.000000000 -0700
@@ -153,7 +153,7 @@
 /* Sparc        */
 /* MIPS         */
 /* PPC          */
-#elif __i386__ || __sun || __sgi || __ppc__
+#elif __i386__ || __sun || __sgi || (__powerpc__ && !defined(__powerpc64__)) || __arm__
 typedef unsigned char           UChar;
 typedef short                   Int16;
 typedef unsigned short          UInt16;
