--- a/Source/WebKit/WebProcess/InjectedBundle/API/gtk/DOM/WebKitDOMDocumentGtk.cpp
+++ b/Source/WebKit/WebProcess/InjectedBundle/API/gtk/DOM/WebKitDOMDocumentGtk.cpp
@@ -54,6 +54,7 @@
 #include "WebKitDOMXPathResultPrivate.h"
 #include <WebCore/CSSImportRule.h>
 #include <WebCore/CSSStyleProperties.h>
+#include <WebCore/CustomElementRegistry.h>
 #include <WebCore/DOMException.h>
 #include <WebCore/DocumentFullscreen.h>
 #include <WebCore/DocumentInlines.h>
--- a/Source/WebKit/WebProcess/GPU/graphics/RemoteDisplayListRecorderIdentifier.h
+++ b/Source/WebKit/WebProcess/GPU/graphics/RemoteDisplayListRecorderIdentifier.h
@@ -25,7 +25,7 @@
 
 #pragma once
 
-#if ENABLE(GPU_PROCESS) && ENABLE(WEBGL)
+#if ENABLE(GPU_PROCESS)
 
 #include <wtf/ObjectIdentifier.h>
 
