commit 7366988f033d29f2b2e2e68fedb8827ba09a2cc1
Author: Daniel Kolesa <daniel@octaforge.org>
Date:   Wed Jan 5 17:50:07 2022 +0100

    reproducible build

--- a/Source/WebCore/bindings/scripts/CodeGenerator.pm
+++ b/Source/WebCore/bindings/scripts/CodeGenerator.pm
@@ -325,7 +325,7 @@ sub MergeExtendedAttributesFromSupplemen
 {
     my ($object, $supplementalExtendedAttributes, $property, $context) = @_;
 
-    foreach my $extendedAttributeName (keys %{$supplementalExtendedAttributes}) {
+    foreach my $extendedAttributeName (sort keys %{$supplementalExtendedAttributes}) {
         my $isAllowed = 0;
         for my $contextAllowed (@{$idlAttributes->{$extendedAttributeName}->{"contextsAllowed"}}) {
             if ($contextAllowed eq $context) {
--- a/Source/WebCore/bindings/scripts/CodeGeneratorJS.pm
+++ b/Source/WebCore/bindings/scripts/CodeGeneratorJS.pm
@@ -3431,7 +3431,7 @@ sub GenerateHeader
         } else {
             push(@headerContent, "    static constexpr unsigned StructureFlags = Base::StructureFlags");
         }
-        foreach my $structureFlag (sort (keys %structureFlags)) {
+        foreach my $structureFlag (sort (sort keys %structureFlags)) {
             push(@headerContent, " | " . $structureFlag);
         }
         push(@headerContent, ";\n");
@@ -8041,7 +8041,7 @@ sub WriteData
 
     my @includes = ();
     my %implIncludeConditions = ();
-    foreach my $include (keys %implIncludes) {
+    foreach my $include (sort keys %implIncludes) {
         next if $headerIncludes{$include};
         next if $headerTrailingIncludes{$include};
 
@@ -8084,7 +8084,7 @@ sub WriteData
     $contents = join "", @headerContentHeader;
 
     @includes = ();
-    foreach my $include (keys %headerIncludes) {
+    foreach my $include (sort keys %headerIncludes) {
         $include = "\"$include\"" unless $include =~ /^["<]/; # "
         $include = SubstituteHeader($include);
         push @includes, $include;
@@ -8098,7 +8098,7 @@ sub WriteData
     $contents .= join "", @headerContent;
 
     @includes = ();
-    foreach my $include (keys %headerTrailingIncludes) {
+    foreach my $include (sort keys %headerTrailingIncludes) {
         $include = "\"$include\"" unless $include =~ /^["<]/; # "
         push @includes, $include;
     }
@@ -8171,7 +8171,7 @@ sub GeneratePrototypeDeclaration
     if (%structureFlags) {
         push(@$outputArray, "public:\n");
         push(@$outputArray, "    static constexpr unsigned StructureFlags = Base::StructureFlags");
-        foreach my $structureFlag (sort (keys %structureFlags)) {
+        foreach my $structureFlag (sort (sort keys %structureFlags)) {
             push(@$outputArray, " | " . $structureFlag);
         }
         push(@$outputArray, ";\n");
