commit 4f3ac6f8bebe6ed32593d6201f971a47badb7d81
Author: q66 <q66@chimera-linux.org>
Date:   Fri Mar 22 00:12:42 2024 +0100

    disable wasm on riscv64

diff --git a/Source/WTF/wtf/PlatformEnable.h b/Source/WTF/wtf/PlatformEnable.h
index 8460fc0..48a0a0c 100644
--- a/Source/WTF/wtf/PlatformEnable.h
+++ b/Source/WTF/wtf/PlatformEnable.h
@@ -623,7 +623,7 @@
 
 #if CPU(RISCV64)
 #undef ENABLE_WEBASSEMBLY
-#define ENABLE_WEBASSEMBLY 1
+#define ENABLE_WEBASSEMBLY 0
 #undef ENABLE_WEBASSEMBLY_OMGJIT
 #define ENABLE_WEBASSEMBLY_OMGJIT 0
 #undef ENABLE_WEBASSEMBLY_BBQJIT
