--- doc/make-man.sh
+++ doc/make-man.sh
@@ -10,11 +10,9 @@ done
 
 for i in ${man1}; do
     lowdown -s -Tman doc/"${i}".md -o doc/man/man1/"${i}".1
-    var=$(head -n1 < doc/man/man1/"${i}".1)
-    var=$(printf '%s' "$var" | tr '7' '1')
-    var="${var} \"\" \"General Commands Manual\""
-    sed -i "s!^.TH.*!${var}!" doc/man/man1/"${i}".1
-    sed -i '2,7d' doc/man/man1/"${i}".1
+    var=$(sed -n '/^.TH/p' doc/man/man1/"${f}".1 | tr '7' '1')
+    sed -i "s!^.TH.*!${var}!" doc/man/man1/"${f}".1
+    sed -i '4,8d' doc/man/man1/"${f}".1
 done
 
 man_5(){
