--- a/src/ft2_about.c
+++ b/src/ft2_about.c
@@ -433,7 +433,8 @@ void initAboutScreen(void)
 	for (int32_t i = 0; i < SINUS_PHASES; i++)
 		sin16[i] = (int16_t)round(32767.0 * sin(i * PI * 2.0 / SINUS_PHASES));
 
-	sprintf(customText3, "v%s (%s)", PROG_VER_STR, __DATE__);
+	sprintf(customText3, "v%s (%s)", PROG_VER_STR, PROG_VER_DATE);
+
 	customText0X = (SCREEN_W    - textWidth(customText0)) / 2;
 	customText1X = (SCREEN_W    - textWidth(customText1)) / 2;
 	customText2X = (SCREEN_W-8) - textWidth(customText2);
--- a/src/ft2_header.h
+++ b/src/ft2_header.h
@@ -13,6 +13,7 @@
 #include "ft2_replayer.h"
 
 #define PROG_VER_STR "@PKGVER@"
+#define PROG_VER_DATE "@DATE@"
 
 // do NOT change these! It will only mess things up...
 
