--- CMakeLists.txt	2018-10-11 18:55:40.000000000 +0200
+++ -	2020-01-10 18:45:11.754965457 +0100
@@ -42,7 +42,7 @@
     set(BOOST_COMPONENTS regex)
 endif()
 
-list(APPEND BOOST_COMPONENTS thread date_time chrono system program_options)
+list(APPEND BOOST_COMPONENTS thread date_time chrono system program_options filesystem)
 
 # Add other dependencies
 find_package(Boost 1.54 REQUIRED COMPONENTS ${BOOST_COMPONENTS})
