--- usr/bin/debootstrap	2014-09-25 13:44:06.000000000 +0900
+++ usr/bin/debootstrap	2014-10-17 11:57:17.000000000 +0900
@@ -50,7 +50,7 @@
 DEF_MIRROR="http://ftp.us.debian.org/debian"
 DEF_HTTPS_MIRROR="https://mirrors.kernel.org/debian"
 
-export LANG USE_COMPONENTS
+export LANG USE_COMPONENTS EXCLUDE
 umask 022
 
 ###########################################################################
--- usr/share/debootstrap/functions	2014-09-15 18:23:00.000000000 +0900
+++ usr/share/debootstrap/functions	2014-10-17 11:57:59.000000000 +0900
@@ -1171,6 +1171,8 @@
 			perl -e '
 $prevpkg = "";
 @d = ();
+my %exclude;
+$exclude{$_} = 1 for split(" ", $ENV{"EXCLUDE"});
 while (<STDIN>) {
 	chomp;
 	if (/^Package: (.*)$/) {
@@ -1187,9 +1189,13 @@
 	$in = 0 if (/^$/);
 	if ($in and (/^Depends: (.*)$/ or /^Pre-Depends: (.*)$/)) {
 		for $d (split /\s*,\s*/, $1) {
-			$d =~ s/\s*[|].*$//;
-			$d =~ s/\s*[(].*[)]\s*//;
-			push @d, $d;
+			for $p (split /\s*\|\s*/, $d) {
+				$p =~ s/\s*[(].*[)]\s*//;
+				if (!$exclude{$p}) {
+					push @d, $p;
+					last;
+				}
+			}
 		}
 	}
 }
