--- docs/libs/Makefile.in.orig	2010-10-24 18:03:12.803730082 +0200
+++ docs/libs/Makefile.in	2010-10-24 18:03:28.459281237 +0200
@@ -931,7 +931,7 @@ dist-hook: dist-check-gtkdoc dist-hook-l
 	-cp $(srcdir)/$(DOC_MODULE).types $(distdir)/
 	-cp $(srcdir)/$(DOC_MODULE)-sections.txt $(distdir)/
 	cd $(distdir) && rm -f $(DISTCLEANFILES)
-        -gtkdoc-rebase --online --relative --html-dir=$(distdir)/html
+	-gtkdoc-rebase --online --relative --html-dir=$(distdir)/html
 
 .PHONY : dist-hook-local docs
 
--- docs/gst/Makefile.in.orig	2010-10-24 18:04:38.452214860 +0200
+++ docs/gst/Makefile.in	2010-10-24 18:04:50.010359626 +0200
@@ -936,7 +936,7 @@ dist-hook: dist-check-gtkdoc dist-hook-l
 	-cp $(srcdir)/$(DOC_MODULE).types $(distdir)/
 	-cp $(srcdir)/$(DOC_MODULE)-sections.txt $(distdir)/
 	cd $(distdir) && rm -f $(DISTCLEANFILES)
-        -gtkdoc-rebase --online --relative --html-dir=$(distdir)/html
+	-gtkdoc-rebase --online --relative --html-dir=$(distdir)/html
 
 .PHONY : dist-hook-local docs
 
--- docs/plugins/Makefile.in.orig	2010-10-24 18:07:05.705795572 +0200
+++ docs/plugins/Makefile.in	2010-10-24 18:07:20.169227262 +0200
@@ -1043,7 +1043,7 @@ dist-hook: dist-check-gtkdoc dist-hook-l
 	-cp $(srcdir)/$(DOC_MODULE).types $(distdir)/
 	-cp $(srcdir)/$(DOC_MODULE)-sections.txt $(distdir)/
 	cd $(distdir) && rm -f $(DISTCLEANFILES)
-        -gtkdoc-rebase --online --relative --html-dir=$(distdir)/html
+	-gtkdoc-rebase --online --relative --html-dir=$(distdir)/html
 
 .PHONY : dist-hook-local docs
 
