--- old/meson.build     2025-07-24 12:10:47.000000000 +0200
+++ new/meson.build     2026-10-11 10:41:51.499761097 +0200
@@ -39,7 +39,7 @@ xorg_include = xorg_dep.get_variable(pkg
-have_fbGlyphs = cc.has_header('fbpict.h', args: ['-I' + xorg_include])
-have_dri3_h = cc.has_header('dri3.h', args: ['-I' + xorg_include])
-have_present_h = cc.has_header('present.h', args: ['-I' + xorg_include])
-have_regionduplicate = cc.has_header('regionstr.h', args: ['-I' + xorg_include])
-have_xf86CursorResetCursor = cc.has_header('xf86Cursor.h', args: ['-I' + xorg_include])
-have_misyncshm = cc.has_header('misyncshm.h', args: ['-I' + xorg_include])
-have_glamor = cc.has_header('glamor.h', args: ['-I' + xorg_include])
+have_fbGlyphs = cc.has_header('fbpict.h', dependencies: xorg_dep)
+have_dri3_h = cc.has_header('dri3.h', dependencies: xorg_dep)
+have_present_h = cc.has_header('present.h', dependencies: xorg_dep)
+have_regionduplicate = cc.has_header('regionstr.h', dependencies: xorg_dep)
+have_xf86CursorResetCursor = cc.has_header('xf86Cursor.h', dependencies: xorg_dep)
+have_misyncshm = cc.has_header('misyncshm.h', dependencies: xorg_dep)
+have_glamor = cc.has_header('glamor.h', dependencies: xorg_dep)
