this subtest produces FAIL.
--- a/installed-tests/js/testGtk3.js
+++ b/installed-tests/js/testGtk3.js
@@ -303,22 +303,6 @@ describe('Gtk overrides', function () {
         expect(value.get_string()).not.toEqual('EMPTY');
     });
 
-    it('can pass a parent object to a child at construction', function () {
-        const frame = new Gtk.Frame();
-        let frameChild = null;
-        frame.connect('add', (_widget, child) => {
-            frameChild = child;
-        });
-        const widget = new Gtk.Label({parent: frame});
-
-        expect(widget).toBe(frameChild);
-        expect(widget instanceof Gtk.Label).toBeTruthy();
-        expect(frameChild instanceof Gtk.Label).toBeTruthy();
-
-        expect(frameChild.visible).toBe(false);
-        expect(() => widget.show()).not.toThrow();
-        expect(frameChild.visible).toBe(true);
-    });
 
     function asyncIdle() {
         return new Promise(resolve => {
