--- Makefile-test.am.orig	2012-01-14 08:54:37.862659814 +0100
+++ Makefile-test.am	2012-01-14 08:55:24.061568378 +0100
@@ -69,8 +69,8 @@ testlibdir = $(prefix)/unused
 install-testlibLTLIBRARIES: # prevent it from being installed
 
 nodist_libregress_la_SOURCES = $(GI_DATADIR)/tests/regress.c $(GI_DATADIR)/tests/regress.h
-libregress_la_CFLAGS = $(GJS_CFLAGS) $(GJS_CAIRO_CFLAGS)
-libregress_la_LDFLAGS = -avoid-version $(GJS_LIBS) $(GJS_CAIRO_LIBS)
+libregress_la_CFLAGS = $(GJS_CFLAGS) $(GJS_CAIRO_CFLAGS) $(GJS_GDBUS_CFLAGS)
+libregress_la_LDFLAGS = -avoid-version $(GJS_LIBS) $(GJS_CAIRO_LIBS) $(GJS_GDBUS_LIBS)
 nodist_libgimarshallingtests_la_SOURCES = $(GI_DATADIR)/tests/gimarshallingtests.c $(GI_DATADIR)/tests/gimarshallingtests.h
 libgimarshallingtests_la_CFLAGS = $(GJS_CFLAGS)
 libgimarshallingtests_la_LDFLAGS = -avoid-version $(GJS_LIBS)
