--- a/compel/arch/aarch64/src/lib/include/uapi/asm/sigframe.h
+++ b/compel/arch/aarch64/src/lib/include/uapi/asm/sigframe.h
@@ -1,7 +1,11 @@
 #ifndef UAPI_COMPEL_ASM_SIGFRAME_H__
 #define UAPI_COMPEL_ASM_SIGFRAME_H__
 
+#ifdef __GLIBC__
 #include <asm/sigcontext.h>
+#else
+#include <linux/types.h>
+#endif
 #include <sys/ucontext.h>
 
 #include <stdint.h>
--- a/criu/arch/aarch64/include/asm/restorer.h
+++ b/criu/arch/aarch64/include/asm/restorer.h
@@ -1,7 +1,11 @@
 #ifndef __CR_ASM_RESTORER_H__
 #define __CR_ASM_RESTORER_H__
 
+#ifdef __GLIBC__
 #include <asm/sigcontext.h>
+#else
+#include <linux/types.h>
+#endif
 #include <sys/ucontext.h>
 
 #include "asm/types.h"
