--- Makefile.in.orig	2010-04-16 21:38:30.707429111 +0200
+++ Makefile.in	2010-04-16 21:39:03.620439149 +0200
@@ -1132,24 +1132,7 @@ abi_check_all:	$(TEST_PREPROCESSOR_HEADE
 	$(MAKE) abi_headers="$^" do_abi_check
 
 do_abi_check:
-	set -ex; \
-	for file in $(abi_headers); do \
-	         @CC@ -E -nostdinc -dI \
-	                  -I$(top_srcdir)/include \
-	                  -I$(top_srcdir)/include/mysql \
-	                  -I$(top_srcdir)/sql \
-	                  -I$(top_builddir)/include \
-	                  -I$(top_builddir)/include/mysql \
-	                  -I$(top_builddir)/sql \
-	                                 $$file 2>/dev/null | \
-	                  @SED@ -e '/^# /d' \
-	                            -e '/^[ 	]*$$/d' \
-	                            -e '/^#pragma GCC set_debug_pwd/d' \
-	                            -e '/^#ident/d' > \
-	                                       $(top_builddir)/abi_check.out; \
-	                  @DIFF@ -w $$file.pp $(top_builddir)/abi_check.out; \
-	                  @RM@ $(top_builddir)/abi_check.out; \
-	done
+	@echo "Skipping do_abi_check!"
 
 # Don't update the files from bitkeeper
 %::SCCS/s.%
