--- sql-common/client_authentication.cc	2016-03-03 16:35:33.000000000 +0100
+++ sql-common/client_authentication.cc	2017-05-09 16:14:10.298254686 +0200
@@ -83,8 +83,7 @@
   FILE *pub_key_file= NULL;
 
   if (mysql->options.extension != NULL &&
-      mysql->options.extension->server_public_key_path != NULL &&
-      mysql->options.extension->server_public_key_path != '\0')
+      mysql->options.extension->server_public_key_path != NULL)
   {
     pub_key_file= fopen(mysql->options.extension->server_public_key_path,
                         "r");
