From 5ee7d58f19542d1dac872badd72669ee288bd316 Mon Sep 17 00:00:00 2001
From: Daniel Kolesa <daniel@octaforge.org>
Date: Wed, 16 Sep 2020 22:11:50 +0200
Subject: [PATCH] drop test that fails on musl

---
 tests/libtracker-common/meson.build | 1 -
 1 file changed, 1 deletion(-)

diff --git tests/libtracker-common/meson.build tests/libtracker-common/meson.build
index 55b7ada..69688e0 100644
--- a/tests/libtracker-common/meson.build
+++ b/tests/libtracker-common/meson.build
@@ -2,7 +2,6 @@ libtracker_common_tests = [
     'date-time',
     'dbus',
     'file-utils',
-    'sched',
     'type-utils',
     'utils',
 ]
-- 
2.28.0

