diff --git a/src/libtracker-miner/meson.build b/src/libtracker-miner/meson.build
index 49c879057..a3bd2d9d9 100644
--- src/libtracker-miner/meson.build
+++ src/libtracker-miner/meson.build
@@ -91,6 +91,7 @@ tracker_miner_gir = gnome.generate_gir(libtracker_miner,
     install: true,
     extra_args: tracker_c_args + [
         '--c-include=libtracker-miner/tracker-miner.h',
+        '-DTRACKER_COMPILATION',
     ])
 
 gnome.generate_vapi(
diff --git a/src/libtracker-sparql-backend/meson.build b/src/libtracker-sparql-backend/meson.build
index e54333bec..8f4a9e4a0 100644
--- src/libtracker-sparql-backend/meson.build
+++ src/libtracker-sparql-backend/meson.build
@@ -45,7 +45,9 @@ tracker_sparql_c_gir = gnome.generate_gir(libtracker_sparql,
     includes : ['GLib-2.0', 'GObject-2.0', 'Gio-2.0' ],
     link_with: libtracker_sparql,
     extra_args: [
-        '--c-include', 'libtracker-sparql/tracker-sparql.h',
+        '--c-include',
+        'libtracker-sparql/tracker-sparql.h',
+        '-DTRACKER_COMPILATION',
     ])
 
 tracker_sparql_c_gir = tracker_sparql_c_gir[0]  # Ignore this .typelib

