Adapt DGifOpenFileHandle() to the 5.x API.

--- src/tracker-extract/tracker-extract-gif.c.orig	2012-10-15 16:37:00.360314076 +0200
+++ src/tracker-extract/tracker-extract-gif.c	2012-10-15 16:37:48.823166861 +0200
@@ -592,7 +592,7 @@ tracker_extract_get_metadata (TrackerExt
 	const gchar *graph;
 	gchar *filename, *uri;
 	GFile *file;
-	int fd;
+	int fd, err = 0;
 
 	preupdate = tracker_extract_info_get_preupdate_builder (info);
 	metadata = tracker_extract_info_get_metadata_builder (info);
@@ -617,7 +617,8 @@ tracker_extract_get_metadata (TrackerExt
 		return FALSE;
 	}	
 
-	if ((gifFile = DGifOpenFileHandle (fd)) == NULL) {
+	gifFile = DGifOpenFileHandle (fd, &err);
+	if (err) {
 		PrintGifError ();
 		close (fd);
 		return FALSE;
