--- a/src/modules/graphic/osggraph/Utils/OsgAtomic.h	2021-06-04 21:21:06.000000000 +0200
+++ -	2022-02-05 22:05:41.666948753 +0100
@@ -23,7 +23,7 @@
 #if defined(__GNUC__) && ((4 < __GNUC__)||(4 == __GNUC__ && 1 <= __GNUC_MINOR__)) && (defined(__x86_64__) || defined(__powerpc64__))
 // No need to include something. Is a Compiler API ...
 # define SDATOMIC_USE_GCC4_BUILTINS
-#elif defined(__GNUC__) && (defined(__i386__) || defined(__arm__))
+#elif defined(__GNUC__) && (defined(__i386__) || defined(__arm__) || defined(__aarch64__))
 # define SDATOMIC_USE_LIBRARY_FUNCTIONS
 #elif defined(__sgi) && defined(_COMPILER_VERSION) && (_COMPILER_VERSION>=730)
 // No need to include something. Is a Compiler API ...
