Index: cmake/robot.cmake
===================================================================
--- a/cmake/robot.cmake
+++ b/cmake/robot.cmake
@@ -229,7 +229,7 @@ ENDMACRO(ROBOT_MODULE)
 #  
 MACRO(ROBOT_DATA)
 
-  SET(RBD_SYNTAX "NAME,1,1,RBD_HAS_NAME,RBD_NAME")
+  SET(RBD_SYNTAX "RNAME,1,1,RBD_HAS_NAME,RBD_NAME")
   SET(RBD_SYNTAX ${RBD_SYNTAX} "PREFIX,0,1,RBD_HAS_PREFIX,RBD_PREFIX")
   SET(RBD_SYNTAX ${RBD_SYNTAX} "FILES,0,-1,RBD_HAS_FILES,RBD_FILES")
   SET(RBD_SYNTAX ${RBD_SYNTAX} "SUBDIRS,0,-1,RBD_HAS_SUBDIRS,RBD_SUBDIRS")
@@ -311,6 +311,6 @@ MACRO(ROBOT)
 
   ROBOT_MODULE(NAME ${RB_NAME} INTERFACE ${RB_INTERFACE} SOURCES ${RB_SOURCES})
 
-  ROBOT_DATA(NAME ${RB_NAME} PREFIX ${RB_PREFIX} FILES ${RB_FILES} SUBDIRS ${RB_SUBDIRS})
+  ROBOT_DATA(RNAME ${RB_NAME} PREFIX ${RB_PREFIX} FILES ${RB_FILES} SUBDIRS ${RB_SUBDIRS})
 
 ENDMACRO(ROBOT)
Index: data/drivers/dandroid_36GP/CMakeLists.txt
===================================================================
--- a/data/drivers/dandroid_36GP/CMakeLists.txt
+++ b/data/drivers/dandroid_36GP/CMakeLists.txt
@@ -6,6 +6,6 @@ SET(ROBOT_SUBDIRS 0 1 2 3 4 5 6 7 8 9 10
                   36gp-tridenti-6c34 36gp-tridenti-v8ri
                   tracks)
 
-ROBOT_DATA(NAME dandroid_36GP
+ROBOT_DATA(RNAME dandroid_36GP
            FILES dandroid_36GP.xml default.xml logo.rgb readme.txt
            SUBDIRS ${ROBOT_SUBDIRS} PATTERNS *.xml logo.rgb *.png *.jpg *.csv)
Index: data/drivers/dandroid_ls1/CMakeLists.txt
===================================================================
--- a/data/drivers/dandroid_ls1/CMakeLists.txt
+++ b/data/drivers/dandroid_ls1/CMakeLists.txt
@@ -6,6 +6,6 @@ SET(ROBOT_SUBDIRS 0 1 2 3 4 5 6 7 8 9 10
                   ls1-taipan-ltsr ls1-marisatech-gt4r
                   tracks)
 
-ROBOT_DATA(NAME dandroid_ls1
+ROBOT_DATA(RNAME dandroid_ls1
            FILES dandroid_ls1.xml default.xml logo.rgb readme.txt
            SUBDIRS ${ROBOT_SUBDIRS} PATTERNS *.xml logo.rgb *.png *.jpg *.csv)
Index: data/drivers/dandroid_ls2/CMakeLists.txt
===================================================================
--- a/data/drivers/dandroid_ls2/CMakeLists.txt
+++ b/data/drivers/dandroid_ls2/CMakeLists.txt
@@ -5,6 +5,6 @@ SET(ROBOT_SUBDIRS 0 1 2 3 4 5 6 7 8 9 10
                   ls2-cavallo-360r ls2-morley-r900 ls2-vulture-v6rs2
                   tracks)
 
-ROBOT_DATA(NAME dandroid_ls2
+ROBOT_DATA(RNAME dandroid_ls2
            FILES dandroid_ls2.xml default.xml logo.rgb readme.txt
            SUBDIRS ${ROBOT_SUBDIRS} PATTERNS *.xml logo.rgb *.png *.jpg *.csv)
Index: data/drivers/dandroid_mpa1/CMakeLists.txt
===================================================================
--- a/data/drivers/dandroid_mpa1/CMakeLists.txt
+++ b/data/drivers/dandroid_mpa1/CMakeLists.txt
@@ -5,6 +5,6 @@ SET(ROBOT_SUBDIRS 0 1 2 3 4 5 6 7 8 9
                   mpa1-team-lynx mpa1-ffr
                   tracks)
 
-ROBOT_DATA(NAME dandroid_mpa1
+ROBOT_DATA(RNAME dandroid_mpa1
            FILES dandroid_mpa1.xml default.xml logo.rgb readme.txt
            SUBDIRS ${ROBOT_SUBDIRS} PATTERNS *.xml logo.rgb *.png *.jpg *.csv)
Index: data/drivers/dandroid_rs/CMakeLists.txt
===================================================================
--- a/data/drivers/dandroid_rs/CMakeLists.txt
+++ b/data/drivers/dandroid_rs/CMakeLists.txt
@@ -5,6 +5,6 @@ SET(ROBOT_SUBDIRS 0 1 2 3 4 5 6 7 8 9
                   rs2-fai-dot rs2-lion-2m
                   tracks)
 
-ROBOT_DATA(NAME dandroid_rs
+ROBOT_DATA(RNAME dandroid_rs
            FILES dandroid_rs.xml default.xml logo.rgb readme.txt
            SUBDIRS ${ROBOT_SUBDIRS} PATTERNS *.xml logo.rgb *.png *.jpg *.csv)
Index: data/drivers/dandroid_sc/CMakeLists.txt
===================================================================
--- a/data/drivers/dandroid_sc/CMakeLists.txt
+++ b/data/drivers/dandroid_sc/CMakeLists.txt
@@ -5,6 +5,6 @@ SET(ROBOT_SUBDIRS 0 1 2 3 4 5 6 7 8 9 10
                   sc-lynx-220 sc-murasama-nsx sc-kanagawa-z35gts sc-kongei-kk8s
                   tracks)
 
-ROBOT_DATA(NAME dandroid_sc
+ROBOT_DATA(RNAME dandroid_sc
            FILES dandroid_sc.xml default.xml logo.rgb readme.txt
            SUBDIRS ${ROBOT_SUBDIRS} PATTERNS *.xml logo.rgb *.png *.jpg *.csv)
Index: data/drivers/dandroid_srw/CMakeLists.txt
===================================================================
--- a/data/drivers/dandroid_srw/CMakeLists.txt
+++ b/data/drivers/dandroid_srw/CMakeLists.txt
@@ -9,6 +9,6 @@ SET(ROBOT_SUBDIRS 0 1 2 3 4 5 6 7
                   srw-kanagawa-gtr
                   tracks)
 
-ROBOT_DATA(NAME dandroid_srw
+ROBOT_DATA(RNAME dandroid_srw
            FILES dandroid_srw.xml default.xml logo.rgb readme.txt
            SUBDIRS ${ROBOT_SUBDIRS} PATTERNS *.xml logo.rgb *.png *.jpg)
Index: data/drivers/dandroid_trb1/CMakeLists.txt
===================================================================
--- a/data/drivers/dandroid_trb1/CMakeLists.txt
+++ b/data/drivers/dandroid_trb1/CMakeLists.txt
@@ -6,6 +6,6 @@ SET(ROBOT_SUBDIRS 0 1 2 3 4 5 6 7 8 9 10
                   trb1-zaxxon-rb trb1-vieringe-5rb
                   tracks)
 
-ROBOT_DATA(NAME dandroid_trb1
+ROBOT_DATA(RNAME dandroid_trb1
            FILES dandroid_trb1.xml default.xml logo.rgb readme.txt
            SUBDIRS ${ROBOT_SUBDIRS} PATTERNS *.xml logo.rgb *.png *.jpg *.csv)
Index: data/drivers/human/CMakeLists.txt
===================================================================
--- a/data/drivers/human/CMakeLists.txt
+++ b/data/drivers/human/CMakeLists.txt
@@ -2,10 +2,10 @@ INCLUDE(../../../cmake/macros.cmake)
 
 SET(ROBOT_NAME "human")
 
-ROBOT_DATA(NAME ${ROBOT_NAME}
+ROBOT_DATA(RNAME ${ROBOT_NAME}
            FILES logo.png logo.rgb
            SUBDIRS cars PATTERNS *.xml)
 
-ROBOT_DATA(NAME ${ROBOT_NAME}
+ROBOT_DATA(RNAME ${ROBOT_NAME}
            FILES ${ROBOT_NAME}.xml preferences.xml car.xml
            USER)
Index: data/drivers/networkhuman/CMakeLists.txt
===================================================================
--- a/data/drivers/networkhuman/CMakeLists.txt
+++ b/data/drivers/networkhuman/CMakeLists.txt
@@ -2,10 +2,10 @@ INCLUDE(../../../cmake/macros.cmake)
 
 SET(ROBOT_NAME "networkhuman")
 
-ROBOT_DATA(NAME ${ROBOT_NAME}
+ROBOT_DATA(RNAME ${ROBOT_NAME}
            FILES logo.png)
 #           SUBDIRS cars PATTERNS *.xml)
 
-ROBOT_DATA(NAME ${ROBOT_NAME}
+ROBOT_DATA(RNAME ${ROBOT_NAME}
            FILES ${ROBOT_NAME}.xml preferences.xml car.xml
            USER)
Index: data/drivers/shadow_sc/CMakeLists.txt
===================================================================
--- a/data/drivers/shadow_sc/CMakeLists.txt
+++ b/data/drivers/shadow_sc/CMakeLists.txt
@@ -7,6 +7,6 @@ SET(ROBOT_SUBDIRS 1 2 3 4 5 6 7 8 9
                   sc-lynx-220 sc-murasama-nsx sc-spirit-300 
                   tracks)
 
-ROBOT_DATA(NAME shadow_sc
+ROBOT_DATA(RNAME shadow_sc
            FILES shadow_sc.xml default.xml logo.rgb readme.txt
            SUBDIRS ${ROBOT_SUBDIRS} PATTERNS *.xml logo.rgb *.png *.jpg)
Index: data/drivers/simplix/CMakeLists.txt
===================================================================
--- a/data/drivers/simplix/CMakeLists.txt
+++ b/data/drivers/simplix/CMakeLists.txt
@@ -10,6 +10,6 @@ SET(ROBOT_SUBDIRS trb1-cavallo-360rb
                   tracks
                   0)
 
-ROBOT_DATA(NAME simplix
+ROBOT_DATA(RNAME simplix
            FILES simplix.xml default.xml logo.rgb readme.txt
            SUBDIRS ${ROBOT_SUBDIRS} PATTERNS *.xml *.rgb)
Index: data/drivers/simplix_36GP/CMakeLists.txt
===================================================================
--- a/data/drivers/simplix_36GP/CMakeLists.txt
+++ b/data/drivers/simplix_36GP/CMakeLists.txt
@@ -7,6 +7,6 @@ SET(ROBOT_SUBDIRS 0 1 2 3 4 5 6 7 8 9
                   36gp-tridenti-6c34 36gp-tridenti-v8ri
                   tracks)
 
-ROBOT_DATA(NAME simplix_36GP
+ROBOT_DATA(RNAME simplix_36GP
            FILES simplix_36GP.xml default.xml logo.rgb readme.txt
            SUBDIRS ${ROBOT_SUBDIRS} PATTERNS *.xml logo.rgb *.png *.jpg)
Index: data/drivers/simplix_ls1/CMakeLists.txt
===================================================================
--- a/data/drivers/simplix_ls1/CMakeLists.txt
+++ b/data/drivers/simplix_ls1/CMakeLists.txt
@@ -8,6 +8,6 @@ SET(ROBOT_SUBDIRS 0 1 2 3 4 5 6 7 8 9 10
                   ls1-taipan-ltsr
                   tracks)
 
-ROBOT_DATA(NAME simplix_ls1
+ROBOT_DATA(RNAME simplix_ls1
            FILES simplix_ls1.xml default.xml logo.rgb readme.txt
            SUBDIRS ${ROBOT_SUBDIRS} PATTERNS *.xml logo.rgb *.png *.jpg)
Index: data/drivers/simplix_ls2/CMakeLists.txt
===================================================================
--- a/data/drivers/simplix_ls2/CMakeLists.txt
+++ b/data/drivers/simplix_ls2/CMakeLists.txt
@@ -6,6 +6,6 @@ SET(ROBOT_SUBDIRS 0
                   ls2-bavaria-g3gtr
                   tracks)
 
-ROBOT_DATA(NAME simplix_ls2
+ROBOT_DATA(RNAME simplix_ls2
            FILES simplix_ls2.xml default.xml logo.rgb readme.txt
            SUBDIRS ${ROBOT_SUBDIRS} PATTERNS *.xml logo.rgb *.png *.jpg)
Index: data/drivers/simplix_mp5/CMakeLists.txt
===================================================================
--- a/data/drivers/simplix_mp5/CMakeLists.txt
+++ b/data/drivers/simplix_mp5/CMakeLists.txt
@@ -6,6 +6,6 @@ SET(ROBOT_SUBDIRS 0 1 2 3 4 5
                   mp5-fmc mp5-fmc-drift mp5-fmc-speedy
                   tracks)
 
-ROBOT_DATA(NAME simplix_mp5
+ROBOT_DATA(RNAME simplix_mp5
            FILES simplix_mp5.xml default.xml logo.rgb readme.txt
            SUBDIRS ${ROBOT_SUBDIRS} PATTERNS *.xml logo.rgb *.png *.jpg)
Index: data/drivers/simplix_mpa1/CMakeLists.txt
===================================================================
--- a/data/drivers/simplix_mpa1/CMakeLists.txt
+++ b/data/drivers/simplix_mpa1/CMakeLists.txt
@@ -7,6 +7,6 @@ SET(ROBOT_SUBDIRS 0 1 2 3 4 5 6 7 8 9
                   mpa1-team-lynx mpa1-ffr
                   tracks)
 
-ROBOT_DATA(NAME simplix_mpa1
+ROBOT_DATA(RNAME simplix_mpa1
            FILES simplix_mpa1.xml default.xml logo.rgb readme.txt
            SUBDIRS ${ROBOT_SUBDIRS} PATTERNS *.xml logo.rgb *.png *.jpg)
Index: data/drivers/simplix_sc/CMakeLists.txt
===================================================================
--- a/data/drivers/simplix_sc/CMakeLists.txt
+++ b/data/drivers/simplix_sc/CMakeLists.txt
@@ -7,6 +7,6 @@ SET(ROBOT_SUBDIRS 0 1 2 3 4 5 6 7 8 9 10
                   sc-lynx-220 sc-murasama-nsx
                   tracks)
 
-ROBOT_DATA(NAME simplix_sc
+ROBOT_DATA(RNAME simplix_sc
            FILES simplix_sc.xml default.xml logo.rgb readme.txt
            SUBDIRS ${ROBOT_SUBDIRS} PATTERNS *.xml logo.rgb *.png *.jpg)
Index: data/drivers/simplix_srw/CMakeLists.txt
===================================================================
--- a/data/drivers/simplix_srw/CMakeLists.txt
+++ b/data/drivers/simplix_srw/CMakeLists.txt
@@ -9,6 +9,6 @@ SET(ROBOT_SUBDIRS 0 1 2 3 4 5 6 7
                   srw-kanagawa-gtr
                   tracks)
 
-ROBOT_DATA(NAME simplix_srw
+ROBOT_DATA(RNAME simplix_srw
            FILES simplix_srw.xml default.xml logo.rgb readme.txt
            SUBDIRS ${ROBOT_SUBDIRS} PATTERNS *.xml logo.rgb *.png *.jpg)
Index: data/drivers/simplix_trb1/CMakeLists.txt
===================================================================
--- a/data/drivers/simplix_trb1/CMakeLists.txt
+++ b/data/drivers/simplix_trb1/CMakeLists.txt
@@ -8,6 +8,6 @@ SET(ROBOT_SUBDIRS 0 1 2 3 4 5 6 7 8 9 10
                   trb1-zaxxon-rb trb1-vieringe-5rb
                   tracks)
 
-ROBOT_DATA(NAME simplix_trb1
+ROBOT_DATA(RNAME simplix_trb1
            FILES simplix_trb1.xml default.xml logo.rgb readme.txt
            SUBDIRS ${ROBOT_SUBDIRS} PATTERNS *.xml logo.rgb *.png *.jpg)
Index: data/drivers/usr_36GP/CMakeLists.txt
===================================================================
--- a/data/drivers/usr_36GP/CMakeLists.txt
+++ b/data/drivers/usr_36GP/CMakeLists.txt
@@ -8,7 +8,7 @@ SET(ROBOT_SUBDIRS 1 2 3 4 5 6 7 8 9 10 1
                   36gp-tridenti-v8ri
                   tracks)
 
-ROBOT_DATA(NAME usr_36GP
+ROBOT_DATA(RNAME usr_36GP
            FILES usr_36GP.xml default.xml logo.rgb readme.txt
            SUBDIRS ${ROBOT_SUBDIRS} PATTERNS *.xml logo.rgb *.png *.jpg)
 
Index: data/drivers/usr_ls1/CMakeLists.txt
===================================================================
--- a/data/drivers/usr_ls1/CMakeLists.txt
+++ b/data/drivers/usr_ls1/CMakeLists.txt
@@ -8,6 +8,6 @@ SET(ROBOT_SUBDIRS 1 2 3 4 5 6 7 8 9 10 1
                   ls1-taipan-ltsr ls1-toro-rgt
                   tracks)
 
-ROBOT_DATA(NAME usr_ls1
+ROBOT_DATA(RNAME usr_ls1
            FILES usr_ls1.xml default.xml logo.rgb readme.txt
            SUBDIRS ${ROBOT_SUBDIRS} PATTERNS *.xml logo.rgb *.png *.jpg)
Index: data/drivers/usr_ls2/CMakeLists.txt
===================================================================
--- a/data/drivers/usr_ls2/CMakeLists.txt
+++ b/data/drivers/usr_ls2/CMakeLists.txt
@@ -9,6 +9,6 @@ SET(ROBOT_SUBDIRS 1 2 3 4 5 6 7 8
 		  ls2-vortex-gt ls2-lynx-xkr
                   )
 
-ROBOT_DATA(NAME usr_ls2
+ROBOT_DATA(RNAME usr_ls2
            FILES usr_ls2.xml default.xml logo.rgb readme.txt
            SUBDIRS ${ROBOT_SUBDIRS} PATTERNS *.xml logo.rgb *.png *.jpg)
Index: data/drivers/usr_mpa1/CMakeLists.txt
===================================================================
--- a/data/drivers/usr_mpa1/CMakeLists.txt
+++ b/data/drivers/usr_mpa1/CMakeLists.txt
@@ -6,6 +6,6 @@ SET(ROBOT_SUBDIRS 1 2 3 4 5 6 7 8 #9 10
                   mpa1-ffr mpa1-furia mpa1-hartbill-2002
                   mpa1-murasama mpa1-team-lynx)
 
-ROBOT_DATA(NAME usr_mpa1
+ROBOT_DATA(RNAME usr_mpa1
            FILES usr_mpa1.xml default.xml logo.rgb readme.txt
            SUBDIRS ${ROBOT_SUBDIRS} PATTERNS *.xml logo.rgb *.png *.jpg)
Index: data/drivers/usr_rs/CMakeLists.txt
===================================================================
--- a/data/drivers/usr_rs/CMakeLists.txt
+++ b/data/drivers/usr_rs/CMakeLists.txt
@@ -6,6 +6,6 @@ SET(ROBOT_SUBDIRS 1
                   rs1-kenji-08
                   )
 
-ROBOT_DATA(NAME usr_rs
+ROBOT_DATA(RNAME usr_rs
            FILES usr_rs.xml default.xml logo.rgb readme.txt
            SUBDIRS ${ROBOT_SUBDIRS} PATTERNS *.xml logo.rgb *.png *.jpg)
Index: data/drivers/usr_sc/CMakeLists.txt
===================================================================
--- a/data/drivers/usr_sc/CMakeLists.txt
+++ b/data/drivers/usr_sc/CMakeLists.txt
@@ -7,6 +7,6 @@ SET(ROBOT_SUBDIRS 1 2 3 4 5 6 7 8 9 10 1
                   sc-lynx-220 sc-murasama-nsx sc-spirit-300 
                   tracks)
 
-ROBOT_DATA(NAME usr_sc
+ROBOT_DATA(RNAME usr_sc
            FILES usr_sc.xml default.xml logo.rgb readme.txt
            SUBDIRS ${ROBOT_SUBDIRS} PATTERNS *.xml logo.rgb *.png *.jpg)
Index: data/drivers/usr_trb1/CMakeLists.txt
===================================================================
--- a/data/drivers/usr_trb1/CMakeLists.txt
+++ b/data/drivers/usr_trb1/CMakeLists.txt
@@ -8,6 +8,6 @@ SET(ROBOT_SUBDIRS 1 2 3 4 5 6 7 8 9 10 1
                   trb1-zaxxon-rb
                   tracks)
 
-ROBOT_DATA(NAME usr_trb1
+ROBOT_DATA(RNAME usr_trb1
            FILES usr_trb1.xml default.xml logo.rgb readme.txt
            SUBDIRS ${ROBOT_SUBDIRS} PATTERNS *.xml logo.rgb *.png *.jpg)
