Index: src/modules/graphic/osggraph/Utils/OsgAtomic.h
===================================================================
--- a/src/modules/graphic/osggraph/Utils/OsgAtomic.h      (revision 6789)
+++ b/src/modules/graphic/osggraph/Utils/OsgAtomic.h      (working copy)
@@ -20,7 +20,7 @@
 #ifndef OsgAtomic_H
 #define OsgAtomic_H
 
-#if defined(__GNUC__) && ((4 < __GNUC__)||(4 == __GNUC__ && 1 <= __GNUC_MINOR__)) && defined(__x86_64__)
+#if defined(__GNUC__) && ((4 < __GNUC__)||(4 == __GNUC__ && 1 <= __GNUC_MINOR__)) && (defined(__x86_64__) || defined(__powerpc64__))
 // No need to include something. Is a Compiler API ...
 # define SDATOMIC_USE_GCC4_BUILTINS
 #elif defined(__GNUC__) && defined(__i386__)
