--- a/Makefile.in
+++ b/Makefile.in
@@ -67,10 +67,10 @@
 SIGFILES = @SIGFILES@
 
 es	: ${OFILES} initial.o
-	${CC} -o es ${LDFLAGS} ${OFILES} initial.o ${LIBS}
+	${CC} -o es ${OFILES} initial.o ${LDFLAGS} ${LIBS}
 
 esdump	: ${OFILES} dump.o
-	${CC} -o esdump ${LDFLAGS} ${OFILES} dump.o ${LIBS}
+	${CC} -o esdump ${OFILES} dump.o ${LDFLAGS} ${LIBS}
 
 clean	:
 	rm -f es ${OFILES} ${GEN} dump.o initial.o
