diff --git willuslib/gslpolyfit.c willuslib/gslpolyfit.c
index 5d2b6fbb85..f127bf054f 100644
--- willuslib/gslpolyfit.c
+++ willuslib/gslpolyfit.c
@@ -24,7 +24,10 @@
 #include <stdlib.h>
 #include "willus.h"
 #ifdef HAVE_GSL_LIB
-#include <gsl.h>
+#include <gsl/gsl_poly.h>
+#include <gsl/gsl_vector.h>
+#include <gsl/gsl_multifit.h>
+#include <gsl/gsl_matrix.h>
 #endif
 
 void gslpolyfit(double *x,double *y0,int n,int d,double *c0)
diff --git willuslib/bmp.c willuslib/bmp.c
index 583644d565..b2c36d5673 100644
--- willuslib/bmp.c
+++ willuslib/bmp.c
@@ -37,7 +37,7 @@
 #include <jpeglib.h>
 #endif
 #ifdef HAVE_JASPER_LIB
-#include <jasper.h>
+#include <jasper/jasper.h>
 #endif
 
 #define BOUND(x,xmin,xmax)  if ((x)<(xmin)) (x)=(xmin); else { if ((x)>(xmax)) (x)=(xmax); }
diff --git willuslib/ocrtess.c willuslib/ocrtess.c
index e1366e1aab..58aaf97fb9 100644
--- willuslib/ocrtess.c
+++ willuslib/ocrtess.c
@@ -24,8 +24,9 @@
 #include <stdlib.h>
 #include <string.h>
 #include <ctype.h>
-#include <leptonica.h>
-#include <tesseract.h>
+#include <locale.h>
+#include <leptonica/allheaders.h>
+#include <tesseract/capi.h>
 #include "willus.h"
 
 
diff --git willuslib/wleptonica.c willuslib/wleptonica.c
index a4e3cfd61d..a85be0949d 100644
--- willuslib/wleptonica.c
+++ willuslib/wleptonica.c
@@ -23,7 +23,7 @@
 #include "willus.h"
 
 #ifdef HAVE_LEPTONICA_LIB
-#include <leptonica.h>
+#include <leptonica/allheaders.h>
 
 static void wlept_pix_from_bmp(PIX **pixptr,WILLUSBITMAP *bmp);
 static void wlept_bmp_from_pix(WILLUSBITMAP *bmp,PIX *pix);
@@ -135,7 +135,7 @@ pixWrite("pixb.png",pixb,IFF_PNG);
     dew1=dewarpCreate(pixb,1);
     pixDestroy(&pixb);
     dewarpaInsertDewarp(dewa,dew1);
-    dewarpBuildPageModel_ex(dew1,debug,fit_order);
+    dewarpBuildPageModel(dew1,debug);
     if (bmp1!=NULL)
         {
         PIX *pix2,*pix2d;

diff --git willuslib/wmupdf.c willuslib/wmupdf.c
index 81627efd4c..fcab84c1db 100644
--- willuslib/wmupdf.c
+++ willuslib/wmupdf.c
@@ -189,8 +189,8 @@ int wmupdf_remake_pdf(char *infile,char *outfile,WPDFPAGEINFO *pageinfo,int use_
     pdf_write_opts.do_compress=1;
     pdf_write_opts.do_linear=0;
     pdf_write_opts.do_garbage=1; /* 2 and 3 don't work for this. */
-    pdf_write_opts.continue_on_error=0;
-    pdf_write_opts.errors=NULL;
+    //pdf_write_opts.continue_on_error=0;
+    //pdf_write_opts.errors=NULL;
     write_failed=0;
     wpdfpageinfo_sort(pageinfo);
     xref=NULL;
