Index: b/willuslib/gslpolyfit.c
===================================================================
--- a/willuslib/gslpolyfit.c
+++ b/willuslib/gslpolyfit.c
@@ -24,7 +24,10 @@
 #include <stdlib.h>
 #include "willus.h"
 #ifdef HAVE_GSL_LIB
-#include <gsl.h>
+#include <gsl/gsl_poly.h>
+#include <gsl/gsl_vector.h>
+#include <gsl/gsl_multifit.h>
+#include <gsl/gsl_matrix.h>
 #endif
 
 void gslpolyfit(double *x,double *y0,int n,int d,double *c0)
Index: b/willuslib/bmp.c
===================================================================
--- a/willuslib/bmp.c
+++ b/willuslib/bmp.c
@@ -37,7 +37,7 @@
 #include <jpeglib.h>
 #endif
 #ifdef HAVE_JASPER_LIB
-#include <jasper.h>
+#include <jasper/jasper.h>
 #endif
 
 #define BOUND(x,xmin,xmax)  if ((x)<(xmin)) (x)=(xmin); else { if ((x)>(xmax)) (x)=(xmax); }
Index: b/willuslib/ocrtess.c
===================================================================
--- a/willuslib/ocrtess.c
+++ b/willuslib/ocrtess.c
@@ -25,8 +25,8 @@
 #include <string.h>
 #include <ctype.h>
 #include <locale.h>
-#include <leptonica.h>
-#include <tesseract.h>
+#include <leptonica/allheaders.h>
+#include <tesseract/capi.h>
 #include "willus.h"
 
 char *ocrtess_langnames[] =
Index: b/willuslib/wleptonica.c
===================================================================
--- a/willuslib/wleptonica.c
+++ b/willuslib/wleptonica.c
@@ -23,7 +23,7 @@
 #include "willus.h"
 
 #ifdef HAVE_LEPTONICA_LIB
-#include <leptonica.h>
+#include <leptonica/allheaders.h>
 
 static void wlept_pix_from_bmp(PIX **pixptr,WILLUSBITMAP *bmp);
 static void wlept_bmp_from_pix(WILLUSBITMAP *bmp,PIX *pix);
@@ -135,7 +135,7 @@ pixWrite("pixb.png",pixb,IFF_PNG);
     dew1=dewarpCreate(pixb,1);
     pixDestroy(&pixb);
     dewarpaInsertDewarp(dewa,dew1);
-    dewarpBuildPageModel_ex(dew1,debug,fit_order);
+    dewarpBuildPageModel(dew1,debug);
     if (bmp1!=NULL)
         {
         PIX *pix2,*pix2d;
Index: b/willuslib/bmpdjvu.c
===================================================================
--- a/willuslib/bmpdjvu.c
+++ b/willuslib/bmpdjvu.c
@@ -23,7 +23,7 @@
 #include "willus.h"
 
 #ifdef HAVE_DJVU_LIB
-#include <djvu.h>
+#include <libdjvu/ddjvuapi.h>
 
 static void handle(int wait,ddjvu_context_t *ctx);
 static void djvu_add_page_info(char *buf,ddjvu_document_t *doc,int page,int npages);
