Index: b/willuslib/bmpmupdf.c
===================================================================
--- a/willuslib/bmpmupdf.c
+++ b/willuslib/bmpmupdf.c
@@ -25,7 +25,9 @@
 
 #ifdef HAVE_MUPDF_LIB
 #include <mupdf/pdf.h>
-void pdf_install_load_system_font_funcs(fz_context *ctx);
+void pdf_install_load_system_font_funcs(fz_context *ctx)
+{
+}
 
 static void mupdf_cbz_add_page_info(char *buf,fz_context *ctx,fz_document *doc,
                                     int pageno,int npages);
