Index: b/willuslib/ocrtess.c
===================================================================
--- a/willuslib/ocrtess.c
+++ b/willuslib/ocrtess.c
@@ -482,14 +482,12 @@ int ocrtess_lang_count(char *langstr)
 void ocrtess_set_logfile(char *filename)
 
     {
-    tess_capi_set_logfile(filename);
     }
 
 
 void ocrtess_debug_message(char *message)
 
     {
-    tess_capi_debug_message(message);
     }
 
 
