--- a/Source/dooble_settings.cc
+++ b/Source/dooble_settings.cc
@@ -2307,7 +2307,7 @@ void dooble_settings::show_qtwebengine_dictionaries_warning_label(void)
   if(bytes.trimmed().isEmpty())
     {
       auto const directory
-	(QDir::currentPath() + QDir::separator() + "qtwebengine_dictionaries");
+ ("/usr/share/qt6/qtwebengine_dictionaries");
 
       m_ui.qtwebengine_dictionaries_warning_label->setText
 	(tr("<b>Warning!</b> "
