--- a/Source/dooble_settings.cc
+++ b/Source/dooble_settings.cc
@@ -188,17 +188,11 @@ dooble_settings::dooble_settings(void):dooble_main_window()
     }
   else
     {
-      QString path("");
+      QString path("/usr/share/dooble/translations/");
       auto variable(qgetenv("DOOBLE_TRANSLATIONS_PATH").trimmed());

       if(!variable.isEmpty())
 	path = QString::fromLocal8Bit(variable.constData());
-      else
-	{
-	  path = QDir::currentPath();
-	  path.append(QDir::separator());
-	  path.append("Translations");
-	}

       if(!path.endsWith(QDir::separator()))
 	path.append(QDir::separator());