--- a/src/FileClasses/music/DirectoryPlayer.cpp	2016-11-20 15:57:02.000000000 +0100
+++ b/src/FileClasses/music/DirectoryPlayer.cpp		2018-01-13 17:29:35.816857954 +0100
@@ -61,7 +61,7 @@
 
     music = nullptr;
 
-    Mix_Init(MIX_INIT_FLUIDSYNTH | MIX_INIT_FLAC | MIX_INIT_MP3 | MIX_INIT_OGG);
+    Mix_Init(MIX_INIT_MID | MIX_INIT_FLAC | MIX_INIT_MP3 | MIX_INIT_OGG);
 }
 
 DirectoryPlayer::~DirectoryPlayer() {
--- a/src/FileClasses/music/XMIPlayer.cpp	2016-11-18 00:06:11.000000000 +0100
+++ b/src/FileClasses/music/XMIPlayer.cpp		2018-01-13 17:29:45.983900596 +0100
@@ -32,7 +32,7 @@
 XMIPlayer::XMIPlayer() : MusicPlayer(settings.audio.playMusic, settings.audio.musicVolume) {
     music = nullptr;
 
-    if((Mix_Init(MIX_INIT_FLUIDSYNTH) & MIX_INIT_FLUIDSYNTH) == 0) {
+    if((Mix_Init(MIX_INIT_MID) & MIX_INIT_MID) == 0) {
         SDL_Log("XMIPlayer: Failed to init required midi support: %s", SDL_GetError());
     }
 }
