diff -p1ru a/usr/bin/unity-indicator-appearance b/usr/bin/unity-indicator-appearance
--- a/usr/bin/unity-indicator-appearance	2022-09-09 23:39:17.000000000 +0800
+++ b/usr/bin/unity-indicator-appearance	2024-07-31 11:16:27.596805842 +0800
@@ -5,3 +5,3 @@ import gi
 gi.require_version("Gtk", "3.0")
-gi.require_version('AppIndicator3', '0.1')
+gi.require_version('AyatanaAppIndicator3', '0.1')
 gi.require_version('Notify', '0.7')
@@ -18,3 +18,3 @@ from gi.repository import Gtk as gtk
 from gi.repository import Gio as gio
-from gi.repository import AppIndicator3 as appindicator
+from gi.repository import AyatanaAppIndicator3 as appindicator
 from gi.repository import Notify as notify
@@ -150,2 +150,2 @@ if __name__ == "__main__":
     signal.signal(signal.SIGINT, signal.SIG_DFL)
-    main()
\ No newline at end of file
+    main()
