diff -up krename-4.0.9/src/batchrenamer.cpp.fix krename-4.0.9/src/batchrenamer.cpp
--- src/batchrenamer.cpp.fix	2016-12-08 15:11:53.559881534 +0100
+++ src/batchrenamer.cpp	2016-12-08 15:14:00.345352643 +0100
@@ -402,7 +402,7 @@ void BatchRenamer::undoFiles( ProgressDi
                 break;
             case eRenameMode_Link:
                 // In case of link delete created file
-                job = KIO::file_delete( dstUrl, false );
+                job = KIO::file_delete( dstUrl, KIO::HideProgressInfo );
                 break;
             case eRenameMode_Copy: // no undo possible
                 // TODO: Maybe we should delete the created files
