--- pick.c.orig	2017-09-21 16:36:29.000000000 +0000
+++ pick.c	2017-09-21 16:36:43.917200522 +0000
@@ -28,6 +28,10 @@
 
 #include "compat.h"
 
+#ifndef CCEQ
+	#define CCEQ(val, c)   ((c) == (val) && (val) != _POSIX_VDISABLE)
+#endif
+
 #define tty_putp(capability, fatal) do {				\
 	if (tputs((capability), 1, tty_putc) == ERR && (fatal))		\
 		errx(1, #capability ": unknown terminfo capability");	\
