--- src/gajim.py	2017-11-27 19:00:24.350830956 +0100
+++ src/gajim.py	2017-11-27 19:02:54.072648722 +0100
@@ -307,7 +307,8 @@
     OpenSSL.rand.load_file(str(RNG_SEED))
     crypto.add_entropy_sources_OpenSSL()
     OpenSSL.rand.write_file(str(RNG_SEED))
-except ImportError:
+except (ImportError, AttributeError):
+    PYOPENSSL_PRNG_PRESENT = False
     log.info("PyOpenSSL PRNG not available")
 
 import traceback
