diff --git a/MultiMC.cpp b/MultiMC.cpp.fixed
index 39cc850..be10ffc 100644
--- application/MultiMC.cpp
+++ application/MultiMC.cpp
@@ -154,8 +154,7 @@ MultiMC::MultiMC(int &argc, char **argv, bool test_mode) : QApplication(argc, ar
 
 // static data paths... mostly just for translations
 #ifdef Q_OS_LINUX
-       QDir foo(PathCombine(binPath, ".."));
-       staticDataPath = foo.absolutePath();
+       staticDataPath = dataPath;
 #elif defined(Q_OS_WIN32)
        staticDataPath = binPath;
 #elif defined(Q_OS_MAC)

