Fixes build on big endian.

--- libraries/classparser/src/javaendian.h
+++ libraries/classparser/src/javaendian.h
@@ -11,32 +11,26 @@ inline uint64_t bigswap(uint64_t x)
 {
     return x;
 }
-;
 inline uint32_t bigswap(uint32_t x)
 {
     return x;
 }
-;
 inline uint16_t bigswap(uint16_t x)
 {
     return x;
 }
-;
 inline int64_t bigswap(int64_t x)
 {
     return x;
 }
-;
 inline int32_t bigswap(int32_t x)
 {
     return x;
 }
-;
 inline int16_t bigswap(int16_t x)
 {
     return x;
 }
-;
 #else
 inline uint64_t bigswap(uint64_t x)
 {
