--- application/MultiMC.h
+++ application/MultiMC.h
@@ -6,6 +6,7 @@
 #include <QFlag>
 #include <QIcon>
 #include <QDateTime>
+#include <QTabBar>
 #include <updater/GoUpdate.h>
 
 #include <BaseInstance.h>
--- application/dialogs/NotificationDialog.cpp
+++ application/dialogs/NotificationDialog.cpp
@@ -2,6 +2,7 @@
 #include "ui_NotificationDialog.h"
 
 #include <QTimerEvent>
+#include <QStyle>
 
 NotificationDialog::NotificationDialog(const NotificationChecker::NotificationEntry &entry, QWidget *parent) :
 	QDialog(parent, Qt::MSWindowsFixedSizeDialogHint | Qt::WindowTitleHint | Qt::CustomizeWindowHint),
--- application/pages/modplatform/ImportPage.cpp
+++ application/pages/modplatform/ImportPage.cpp
@@ -7,6 +7,7 @@
 #include "dialogs/ProgressDialog.h"
 #include "dialogs/NewInstanceDialog.h"
 #include <QFileDialog>
+#include <QValidator>
 #include <InstanceImportTask.h>
 
 class UrlValidator : public QValidator
