--- cmake/FindWrapAtomic.cmake	2021-04-30 11:31:33.000000000 +0200
+++ -	2021-05-06 15:56:49.553742705 +0200
@@ -31,9 +31,9 @@
 check_cxx_source_compiles("${atomic_test_sources}" HAVE_STDATOMIC)
 if(NOT HAVE_STDATOMIC)
-    set(_req_libraries "${CMAKE_REQUIRE_LIBRARIES}")
-    set(CMAKE_REQUIRE_LIBRARIES "atomic")
+    set(_req_libraries "${CMAKE_REQUIRED_LIBRARIES}")
+    set(CMAKE_REQUIRED_LIBRARIES "atomic")
     check_cxx_source_compiles("${atomic_test_sources}" HAVE_STDATOMIC_WITH_LIB)
-    set(CMAKE_REQUIRE_LIBRARIES "${_req_libraries}")
+    set(CMAKE_REQUIRED_LIBRARIES "${_req_libraries}")
 endif()
 
 add_library(WrapAtomic::WrapAtomic INTERFACE IMPORTED)
