--- qt6-base-6.4.2.orig/tests/auto/corelib/text/qlocale/tst_qlocale.cpp
+++ qt6-base-6.4.2/tests/auto/corelib/text/qlocale/tst_qlocale.cpp
@@ -1502,8 +1502,10 @@ void tst_QLocale::fpExceptions()
     fenv_t envp;
     fegetenv(&envp);
     feclearexcept(FE_ALL_EXCEPT);
+#ifdef __GLIBC__
     feenableexcept(FE_DIVBYZERO | FE_OVERFLOW | FE_UNDERFLOW | FE_INVALID);
 #endif
+#endif
 
     QString::number(1000.1245);
     QString::number(1.1);
