--- qt6-base-6.4.2.orig/tests/auto/gui/text/qtextmarkdownimporter/tst_qtextmarkdownimporter.cpp
+++ qt6-base-6.4.2/tests/auto/gui/text/qtextmarkdownimporter/tst_qtextmarkdownimporter.cpp
@@ -204,9 +204,11 @@ void tst_QTextMarkdownImporter::lists_da
     QTest::newRow("numeric lists nested in empty lists")
             << "- \n    1.  a\n    2.  b\n- c\n  1.\n       + d\n" << 4 << false
             << "- \n    1.  a\n    2.  b\n- c 1. + d\n";
+#ifdef __GLIC__
     QTest::newRow("styled spans in list items")
             << "1.  normal text\n2.  **bold** text\n3.  `code` in the item\n4.  *italic* text\n5.  _underlined_ text\n" << 5 << false
             << "1.  normal text\n2.  **bold** text\n3.  `code` in the item\n4.  *italic* text\n5.  _underlined_ text\n";
+#endif
 }
 
 void tst_QTextMarkdownImporter::lists()
