--- a/qmake/library/ioutils.cpp	2020-09-24 17:55:53.000000000 +0200
+++ b/qmake/library/ioutils.cpp	2020-09-24 17:55:53.000000000 +0200
@@ -40,6 +40,7 @@
 #  include <unistd.h>
 #  include <utime.h>
 #  include <fcntl.h>
+#  include <limits.h>
 #  include <errno.h>
 #endif
 
--- a/src/corelib/io/qstorageinfo_unix.cpp	2020-09-24 17:55:53.000000000 +0200
+++ b/src/corelib/io/qstorageinfo_unix.cpp	2020-09-24 17:55:53.000000000 +0200
@@ -58,6 +58,7 @@
 #  include <sys/vfs.h>
 #  include <mntent.h>
 #elif defined(Q_OS_LINUX) || defined(Q_OS_HURD)
+#  include <limits.h>
 #  include <mntent.h>
 #  include <sys/statvfs.h>
 #  include <sys/sysmacros.h>
