--- a/tests/auto/corelib/text/qlatin1stringmatcher/tst_qlatin1stringmatcher.cpp
+++ b/tests/auto/corelib/text/qlatin1stringmatcher/tst_qlatin1stringmatcher.cpp
@@ -517,7 +517,9 @@ void tst_QLatin1StringMatcher::indexIn()
 
 void tst_QLatin1StringMatcher::haystacksWithMoreThan4GiBWork()
 {
-#if QT_POINTER_SIZE > 4
+#if 1
+    QSKIP("memory intensive");
+#elif QT_POINTER_SIZE > 4
     // use a large needle to trigger long skips in the Boyer-Moore algorithm
     // (to speed up the test)
     constexpr std::string_view needle = LONG_STRING_256;
