--- qtbase-everywhere-src-6.6.2/tests/auto/corelib/io/qstorageinfo/tst_qstorageinfo.cpp	2024-02-08 18:01:05.000000000 +0200
+++ sth	2024-03-26 15:56:00.112000000 +0200
@@ -180,6 +180,11 @@
                      __FILE__, __LINE__);
         return false;
     }
+    if (storage.fileSystemType() == "zfs") {
+	QTest::qSkip("This test doesn't work on zfs",
+		    __FILE__, __LINE__);
+	return false;
+    }
 #else
     Q_UNUSED(file);
     Q_UNUSED(storage);
