libavutil/audioconvert.h is gone in ffmpeg3.x

--- a/src/mpv/audio/filter/af_lavcac3enc.c	2015-05-20 11:31:58.000000000 +0200
+++ b/src/mpv/audio/filter/af_lavcac3enc.c	2016-09-22 19:52:47.730132367 +0200
@@ -26,7 +26,6 @@
 #include <assert.h>
 
 #include <libavcodec/avcodec.h>
-#include <libavutil/audioconvert.h>
 #include <libavutil/intreadwrite.h>
 #include <libavutil/common.h>
 #include <libavutil/bswap.h>
--- a/src/mpv/audio/out/ao_lavc.c	2015-05-20 11:31:58.000000000 +0200
+++ b/src/mpv/audio/out/ao_lavc.c	2016-09-22 19:54:50.291163189 +0200
@@ -25,7 +25,6 @@
 #include <limits.h>
 
 #include <libavutil/common.h>
-#include <libavutil/audioconvert.h>
 
 #include "config.h"
 #include "options/options.h"
--- a/src/mpv/audio/filter/af_lavrresample.c	2015-05-20 11:31:58.000000000 +0200
+++ b/src/mpv/audio/filter/af_lavrresample.c	2016-09-22 19:57:13.539200765 +0200
@@ -28,7 +28,6 @@
 #include <assert.h>
 
 #include <libavutil/opt.h>
-#include <libavutil/audioconvert.h>
 #include <libavutil/common.h>
 #include <libavutil/samplefmt.h>
 #include <libavutil/mathematics.h>
