--- a/src/bomi/bomi.pro	2015-05-20 11:31:58.000000000 +0200
+++ b/src/bomi/bomi.pro	2016-09-22 20:03:57.452315214 +0200
@@ -19,6 +19,7 @@
 }
 
 QMAKE_CXXFLAGS_CXX11 = -std=c++1y
+CONFIG += c++14
 
 contains(QMAKE_CXX, clang++) {
 QMAKE_CXXFLAGS += -Wno-missing-braces
--- a/src/bomi/misc/filenamegenerator.cpp	2015-05-20 11:31:58.000000000 +0200
+++ b/src/bomi/misc/filenamegenerator.cpp	2016-09-22 20:06:10.523355518 +0200
@@ -51,8 +51,8 @@
         };
         get[u"%MEDIA_DISPLAY_NAME%"_q] = [] (const FileNameGenerator *g) { return g->mediaName; };
 
-        get[u"%UNIX%"_q]     = [] (const FileNameGenerator *g) { return _N(g->unix / 1000llu); };
-        get[u"%UNIX_MS%"_q]  = [] (const FileNameGenerator *g) { return _N(g->unix); };
+        get[u"%UNIX%"_q]     = [] (const FileNameGenerator *g) { return _N(g->_unix / 1000llu); };
+        get[u"%UNIX_MS%"_q]  = [] (const FileNameGenerator *g) { return _N(g->_unix); };
         return get;
     }();
     auto ret = func.value(ph);
--- a/src/bomi/misc/filenamegenerator.hpp	2015-05-20 11:31:58.000000000 +0200
+++ b/src/bomi/misc/filenamegenerator.hpp	2016-09-22 20:07:13.762375106 +0200
@@ -11,7 +11,7 @@
     QTime start, end;
     QString mediaName;
     Mrl mrl;
-    quint64 unix = 0;
+    quint64 _unix = 0;
 };
 
 #endif // FILENAMEGENERATOR_HPP
--- a/src/bomi/player/mainwindow_p.cpp	2015-05-20 11:31:58.000000000 +0200
+++ b/src/bomi/player/mainwindow_p.cpp	2016-09-22 20:08:20.590396090 +0200
@@ -910,7 +910,7 @@
     g.dateTime = QDateTime::currentDateTime();
     g.start = QTime::fromMSecsSinceStartOfDay(e.time());
     g.end = end.isNull() ? g.start : end;
-    g.unix = QDateTime::currentMSecsSinceEpoch();
+    g._unix = QDateTime::currentMSecsSinceEpoch();
     g.mrl = e.mrl();
     g.mediaName = e.media()->name();
     return g;
--- a/src/bomi/player/mainwindow_p.hpp	2015-05-20 11:31:58.000000000 +0200
+++ b/src/bomi/player/mainwindow_p.hpp	2016-09-22 20:09:18.003414361 +0200
@@ -101,7 +101,7 @@
 
     struct {
         QDate date; QTime time, position;
-        quint64 unix = 0;
+        quint64 _unix = 0;
         QMap<QString, std::function<QString(void)>> get;
     } ph;
     QTimer waiter, hider, dialogWorkaround;
--- a/src/bomi/dialog/encoderdialog.cpp	2015-05-20 11:31:58.000000000 +0200
+++ b/src/bomi/dialog/encoderdialog.cpp	2016-09-22 20:05:18.684339670 +0200
@@ -255,7 +255,7 @@
 {
     if (d->size.isEmpty())
         return tr("No video stream exists.");
-    d->g.unix = QDateTime::currentMSecsSinceEpoch();
+    d->g._unix = QDateTime::currentMSecsSinceEpoch();
     d->g.dateTime = QDateTime::currentDateTime();
     d->g.start = d->ui.a->time();
     d->g.end = d->ui.b->time();
