Fix type to be consistent with intel-media-driver

diff -rup libavutil/hwcontext_vaapi.c.orig libavutil/hwcontext_vaapi.c
--- a/libavutil/hwcontext_vaapi.c
+++ b/libavutil/hwcontext_vaapi.c
@@ -1032,7 +1032,7 @@ static int vaapi_map_from_drm(AVHWFrames
     uint32_t va_fourcc;
     int err, i, j, k;
 
-    unsigned long buffer_handle;
+    uintptr_t buffer_handle;
     VASurfaceAttribExternalBuffers buffer_desc;
     VASurfaceAttrib attrs[2] = {
         {

