From aad04fa1c1d953094d570fbb26d8d015ae0e4462 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Piotr=20W=C3=B3jcik?= <chocimier@tlen.pl>
Date: Fri, 6 Nov 2020 21:17:30 +0100
Subject: [PATCH] fix-tests


diff --git src/kbookmarkmodel/tests/CMakeLists.txt src/kbookmarkmodel/tests/CMakeLists.txt
index 54df341..c07a310 100644
--- a/src/kbookmarkmodel/tests/CMakeLists.txt
+++ b/src/kbookmarkmodel/tests/CMakeLists.txt
@@ -5,7 +5,11 @@ set( EXECUTABLE_OUTPUT_PATH ${CMAKE_CURRENT_BINARY_DIR} )
 ###### kbookmarkmodeltest ######
 
 set(kbookmarkmodeltest_SRCS kbookmarkmodeltest.cpp)
-ecm_add_test(${kbookmarkmodeltest_SRCS}
-    TEST_NAME kbookmarkmodeltest
-    LINK_LIBRARIES kbookmarkmodel_private KF5::Bookmarks Qt5::Test
+add_executable(kbookmarkmodeltest ${kbookmarkmodeltest_SRCS})
+add_test(
+    NAME kbookmarkmodeltest
+    COMMAND kbookmarkmodeltest
+    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
 )
+target_link_libraries(kbookmarkmodeltest kbookmarkmodel_private KF5::Bookmarks Qt5::Test)
+
-- 
2.29.2

