From f239d0499875a5dcadc2f9ea4586c4d933bbbcf1 Mon Sep 17 00:00:00 2001
From: Estevan Castilho <tevo@riajuu.net>
Date: Sat, 29 Mar 2025 22:19:53 +0000
Subject: [PATCH] Do not install cross-compiled built artifacts under a
 separate tree

---
 Makefile.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.in b/Makefile.in
index 880262b..521e04f 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -40,7 +40,7 @@ ifeq (${HOST},${TARGET})
 IPREFIX=${DESTDIR}${prefix}
 else
 GPROPTS+=--target=${TARGET_ALIAS}
-IPREFIX=${DESTDIR}${prefix}/${TARGET_ALIAS}
+IPREFIX=${DESTDIR}${prefix}
 endif
 
 ifdef RTS
-- 
2.48.1

