--- atspi/meson.build	2019-03-12 03:54:43.000000000 +0100
+++ atspi/meson.build	2019-03-14 12:13:54.759466260 +0100
@@ -57,7 +57,7 @@
 
 atspi_includedir = join_paths(get_option('prefix'), get_option('includedir'), 'at-spi-2.0', 'atspi')
 
-install_headers(atspi_headers, subdir: atspi_includedir)
+install_headers(atspi_headers, subdir: 'at-spi-2.0/atspi')
 
 atspi_enums = gnome.mkenums('atspi-enum-types',
                             sources: [ 'atspi-constants.h', 'atspi-types.h' ],
