diff -ur Nagstamon.orig/Nagstamon/QUI/__init__.py Nagstamon/Nagstamon/QUI/__init__.py
--- a/Nagstamon/QUI/__init__.py	2020-01-24 10:12:12.000000000 -0500
+++ b/Nagstamon/QUI/__init__.py	2020-10-08 00:20:39.437758862 -0400
@@ -5770,7 +5770,7 @@
         self.ui.button_choose_custom_cert_ca_file.clicked.connect(self.choose_custom_cert_ca_file)
 
         # fill authentication combobox
-        self.ui.input_combobox_authentication.addItems(['Basic', 'Digest', 'Kerberos'])
+        self.ui.input_combobox_authentication.addItems(['Basic', 'Digest'])
 
         # detect change of server type which leads to certain options shown or hidden
         self.ui.input_combobox_type.activated.connect(self.toggle_type)
diff -ur Nagstamon.orig/Nagstamon/Servers/Generic.py Nagstamon/Nagstamon/Servers/Generic.py
--- a/Nagstamon/Servers/Generic.py	2020-01-24 10:12:12.000000000 -0500
+++ b/Nagstamon/Servers/Generic.py	2020-10-08 00:21:28.460699656 -0400
@@ -53,15 +53,6 @@
                               RESOURCES)
 
 
-# requests_gssapi is newer but not available everywhere
-try:
-    # extra imports needed to get it compiled on macOS
-    import numbers
-    import gssapi.raw.cython_converters
-    from requests_gssapi import HTTPSPNEGOAuth as HTTPSKerberos
-except ImportError:
-    from requests_kerberos import HTTPKerberosAuth as HTTPSKerberos
-
 # disable annoying SubjectAltNameWarning warnings
 try:
     from requests.packages.urllib3.exceptions import SubjectAltNameWarning
@@ -260,8 +251,6 @@
                 self.session.auth = requests.auth.HTTPBasicAuth(self.username, self.password)
             elif self.authentication == 'digest':
                 self.session.auth = requests.auth.HTTPDigestAuth(self.username, self.password)
-            elif self.authentication == 'kerberos':
-                self.session.auth = HTTPSKerberos()
 
             # default to check TLS validity
             if self.ignore_cert:
diff -ur Nagstamon.orig/setup.py Nagstamon/setup.py
--- a/setup.py	2020-10-08 00:18:37.532906083 -0400
+++ b/setup.py	2020-10-08 00:21:43.580681397 -0400
@@ -89,7 +89,6 @@
         'python3-psutil '
         'python3-qt5 '
         'python3-requests '
-        'python3-requests-kerberos '
         'python3-SecretStorage '
         'qt5-qtmultimedia '
         'qt5-qtsvg ',
