--- a/Nagstamon/QUI/__init__.py	2022-11-04 07:04:14.000000000 +0100
+++ -	2023-02-16 15:17:33.465042825 +0100
@@ -5701,7 +5701,7 @@
         self.window.button_choose_custom_cert_ca_file.clicked.connect(self.choose_custom_cert_ca_file)
 
         # fill authentication combobox
-        self.window.input_combobox_authentication.addItems(['Basic', 'Digest', 'Kerberos', 'Bearer'])
+        self.window.input_combobox_authentication.addItems(['Basic', 'Digest', 'Bearer'])
         if ECP_AVAILABLE is True:
             self.window.input_combobox_authentication.addItems(['ECP'])
 
--- a/Nagstamon/Servers/Generic.py	2022-11-04 07:04:14.000000000 +0100
+++ -	2023-02-16 15:18:28.346371050 +0100
@@ -59,20 +59,6 @@
                               OS_DARWIN,
                               RESOURCES)
 
-if OS == OS_DARWIN:
-    # requests_gssapi is newer but not available everywhere
-    try:
-        # extra imports needed to get it compiled on macOS
-        import numbers
-        import gssapi.raw.cython_converters
-        from requests_gssapi import HTTPSPNEGOAuth as HTTPSKerberos
-    except ImportError:
-        from requests_kerberos import HTTPKerberosAuth as HTTPSKerberos
-else:
-    # requests_gssapi needs installation of KfW - Kerberos for Windows
-    # requests_kerberoes doesn't
-    from requests_kerberos import HTTPKerberosAuth as HTTPSKerberos
-
 # disable annoying SubjectAltNameWarning warnings
 try:
     from requests.packages.urllib3.exceptions import SubjectAltNameWarning
@@ -303,8 +289,6 @@
             session.auth = requests.auth.HTTPDigestAuth(self.username, self.password)
         elif self.authentication == 'ecp' and ECP_AVAILABLE:
             session.auth = HTTPECPAuth(self.idp_ecp_endpoint, username=self.username, password=self.password)
-        elif self.authentication == 'kerberos':
-            session.auth = HTTPSKerberos()
         elif self.authentication == 'bearer':
             session.auth = BearerAuth(self.password)
 
--- a/setup.py	2022-11-04 07:04:14.000000000 +0100
+++ -	2023-02-16 15:18:47.726901893 +0100
@@ -98,7 +98,6 @@
                                   'python3-pysocks '
                                   'python3-qt5 '
                                   'python3-requests '
-                                  'python3-requests-kerberos '
                                   'python3-SecretStorage '
                                   'qt5-qtmultimedia '
                                   'qt5-qtsvg ',
