--- Nagstamon/QUI/__init__.py.orig	2018-09-12 16:40:47.350735530 +0200
+++ Nagstamon/QUI/__init__.py	2018-09-12 16:47:18.330913802 +0200
@@ -5279,7 +5279,7 @@
         self.ui.button_choose_custom_cert_ca_file.clicked.connect(self.choose_custom_cert_ca_file)
 
         # fill authentication combobox
-        self.ui.input_combobox_authentication.addItems(['Basic', 'Digest', 'Kerberos'])
+        self.ui.input_combobox_authentication.addItems(['Basic', 'Digest'])
 
         # detect change of server type which leads to certain options shown or hidden
         self.ui.input_combobox_type.activated.connect(self.toggle_type)
--- Nagstamon/Servers/Generic.py,orug	2018-09-12 16:40:47.353735516 +0200
+++ Nagstamon/Servers/Generic.py	2018-09-12 16:47:52.962752440 +0200
@@ -45,7 +45,6 @@
 from collections import OrderedDict
 
 import requests
-import requests_kerberos
 
 # disable annoying SubjectAltNameWarning warnings
 try:
@@ -231,8 +230,6 @@
                 self.session.auth = requests.auth.HTTPBasicAuth(self.username, self.password)
             elif self.authentication == 'digest':
                 self.session.auth = requests.auth.HTTPDigestAuth(self.username, self.password)
-            elif self.authentication == 'kerberos':
-                self.session.auth = requests_kerberos.HTTPKerberosAuth()
 
             # default to check TLS validity
             if self.ignore_cert:
--- setup.py.orig	2018-09-12 16:40:47.363735470 +0200
+++ setup.py	2018-09-12 16:48:02.146709650 +0200
@@ -85,7 +85,6 @@
         'python3-psutil '
         'python3-qt5 '
         'python3-requests '
-        'python3-requests-kerberos '
         'python3-SecretStorage '
         'qt5-qtmultimedia '
         'qt5-qtsvg ',
