From a1a3516309bd45ec1783321d262b0e2f17a2fc4a Mon Sep 17 00:00:00 2001
From: Henri Wahl <2835065+HenriWahl@users.noreply.github.com>
Date: Sun, 12 Feb 2023 22:43:43 +0100
Subject: [PATCH] version without OS

---
 setup.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/setup.py b/setup.py
index 5ff72dda..271072e3 100644
--- a/setup.py
+++ b/setup.py
@@ -45,8 +45,8 @@
     else:
         DIST, DIST_VERSION, DIST_NAME = platform.dist()
     NAME = NAME.lower()
-VERSION = AppInfo.VERSION.replace('-', '.') + '.' + DIST + DIST_VERSION
-
+#VERSION = AppInfo.VERSION.replace('-', '.') + '.' + DIST + DIST_VERSION
+VERSION = AppInfo.VERSION.replace('-', '.')
 NAGSTAMON_SCRIPT = 'nagstamon.py'
 
 from setuptools import setup
