diff --git configure configure
index d3702c3..79a958b 100755
--- a/configure
+++ b/configure
@@ -11188,7 +11188,7 @@ LIBS=$RT_LIBS
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for CLOCK_MONOTONIC" >&5
 $as_echo_n "checking for CLOCK_MONOTONIC... " >&6; }
 
-if test "$cross_compiling" = yes; then :
+if test "$cross_compiling" = xxx; then :
   { { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
 $as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
 as_fn_error "cannot run test program while cross compiling
