Use ELFv2 ABI for big endian.

--- a/purgatory/arch/ppc64/Makefile
+++ b/purgatory/arch/ppc64/Makefile
@@ -14,8 +14,8 @@ ppc64_PURGATORY_EXTRA_CFLAGS += -m64 -msoft-float \
 ppc64_PURGATORY_EXTRA_ASFLAGS += -m64
 ifeq ($(SUBARCH),BE)
 	ppc64_PURGATORY_EXTRA_LDFLAGS += -melf64ppc
-	ppc64_PURGATORY_EXTRA_CFLAGS += -mcall-aixdesc
-	ppc64_PURGATORY_EXTRA_ASFLAGS += -mcall-aixdesc
+	ppc64_PURGATORY_EXTRA_CFLAGS += -mabi=elfv2
+	ppc64_PURGATORY_EXTRA_ASFLAGS += -mabi=elfv2
 else
 	ppc64_PURGATORY_EXTRA_LDFLAGS += -melf64lppc
 	ppc64_PURGATORY_EXTRA_CFLAGS += -mlittle-endian
