--- mfc5490cn/cupswrapper/cupswrappermfc5490cn	2008-10-21 09:07:10.000000000 +0000
+++ mfc5490cn/cupswrapper/cupswrappermfc5490cn.void	2017-01-31 11:48:41.119329004 +0000
@@ -22,30 +22,25 @@
 printer_name=`echo $printer_model | tr '[a-z]' '[A-Z]'`
 device_name=`echo $printer_name | eval sed -e 's/MFC/MFC-/' -e 's/DCP/DCP-/' -e 's/FAX/FAX-/'`
 pcfilename=`echo $printer_name | tr -d '[A-Z]'`
-device_model="Printer"
+device_model="Printers"
 if [ "$1" = '-e' ]; then
   lpadmin -x ${printer_name}
   rm -f /usr/share/cups/model/br${printer_model}.ppd
   rm -f /usr/share/ppd/br${printer_model}.ppd
   rm -f /usr/lib/cups/filter/brlpdwrapper${printer_model}
   rm -f /usr/lib64/cups/filter/brlpdwrapper${printer_model}
-  rm -f /usr/local/Brother/${device_model}/${printer_model}/cupswrapper/brcupsconfpt1
-if [  -e /etc/init.d/cups ]; then
-   /etc/init.d/cups restart
-elif [  -e /etc/init.d/cupsys ]; then
-   /etc/init.d/cupsys restart
+if [  -e /var/service/cupsd ]; then
+   sv restart cupsd
 fi
-#  /etc/init.d/cups restart
+#  sv restart cupsd
   exit 0
 fi
 if [ "$1" = "-r" ]; then
   lpadmin -x ${printer_name}
-if [  -e /etc/init.d/cups ]; then
-   /etc/init.d/cups restart
-elif [  -e /etc/init.d/cupsys ]; then
-   /etc/init.d/cupsys restart
+if [  -e /var/service/cupsd ]; then
+   sv restart cupsd
 fi
-#  /etc/init.d/cups restart
+#  sv restart cupsd
   exit 0
 fi
 if [ "$1" = "help" ] || [ "$1" = "-h" ]; then
@@ -55,10 +50,10 @@
   echo   '       -r : remove printer'
   exit 0
 fi
-#mkdir -p /usr/local/Brother/${device_model}/${printer_model}/filter
+#mkdir -p /opt/brother/${device_model}/${printer_model}/filter
 #mkdir -p /usr/lib/cups/filter
 
-if [ -e "/usr/local/Brother/${device_model}/${printer_model}/lpd/filter${printer_model}" ]; then
+if [ -e "/opt/brother/${device_model}/${printer_model}/lpd/filter${printer_model}" ]; then
   :
 else
   echo "ERROR : Brother LPD filter is not installed."
@@ -67,7 +62,9 @@
 if [ -d "/usr/share/cups/model" ]; then
   ppd_file_name=/usr/share/cups/model/br${printer_model}.ppd
 else
-  ppd_file_name=/usr/share/ppd/br${printer_model}.ppd
+  ppd_dir=/usr/share/ppd
+  mkdir -p ${ppd_dir}
+  ppd_file_name=${ppd_dir}/br${printer_model}.ppd
 fi
 #ppd_file_name=/usr/share/cups/model/br${printer_model}.ppd
 
@@ -829,25 +826,25 @@
       cat    > \$INPUT_TEMP_PS
    fi
 fi
-if [ -e "/usr/local/Brother/${device_model}/${printer_model}/lpd/filter${printer_model}" ]; then
+if [ -e "/opt/brother/${device_model}/${printer_model}/lpd/filter${printer_model}" ]; then
        :
 else
-    echo "ERROR: /usr/local/Brother/${device_model}/${printer_model}/lpd/filter${printer_model} does not exist"   >>\$LOGFILE
+    echo "ERROR: /opt/brother/${device_model}/${printer_model}/lpd/filter${printer_model} does not exist"   >>\$LOGFILE
     errorcode=30
     exit
 fi
 
 CUPSOPTION=\`echo "\$5 Copies=\$4" | sed -e 's/BrMirror=OFF/MirrorPrint=OFF/' -e 's/BrMirror=ON/MirrorPrint=ON/' -e 's/BrChain/Chain/' -e 's/BrBrightness/Brightness/' -e 's/BrContrast/Contrast/' -e 's/BrHalfCut/HalfCut/' -e 's/BrAutoTapeCut/AutoCut/' -e 's/BrHalftonePattern/Halftone/' -e 's/Binary/Binary/' -e 's/Dither/Dither/' -e 's/ErrorDiffusion/ErrorDiffusion/' -e 's/PageSize/media/' -e 's/BrSheets/Sheets/' -e 's/multiple-document-handling/Collate/' -e 's/separate-documents-collated-copies/ON/' -e 's/separate-documents-uncollated-copies/OFF/'\`
-if [ -e "/usr/local/Brother/${device_model}/${printer_model}/cupswrapper/brcupsconfpt1" ]; then
+if [ -e "/opt/brother/${device_model}/${printer_model}/cupswrapper/brcupsconfpt1" ]; then
   if [ \$DEBUG = 0 ]; then
-     /usr/local/Brother/${device_model}/${printer_model}/cupswrapper/brcupsconfpt1  ${printer_name}  \$PPDC 0 "\$CUPSOPTION" "${printer_model}">> /dev/null
+     /opt/brother/${device_model}/${printer_model}/cupswrapper/brcupsconfpt1  ${printer_name}  \$PPDC 0 "\$CUPSOPTION" "${printer_model}">> /dev/null
   else
-     /usr/local/Brother/${device_model}/${printer_model}/cupswrapper/brcupsconfpt1  ${printer_name}  \$PPDC \$LOGCLEVEL "\$CUPSOPTION" "${printer_model}">>\$LOGFILE
+     /opt/brother/${device_model}/${printer_model}/cupswrapper/brcupsconfpt1  ${printer_name}  \$PPDC \$LOGCLEVEL "\$CUPSOPTION" "${printer_model}">>\$LOGFILE
   fi
 fi
 
 if [ \$DEBUG -lt 10 ]; then
-    cat    \$INPUT_TEMP_PS | /usr/local/Brother/${device_model}/${printer_model}/lpd/filter${printer_model} "\$\$" "CUPS" "USB"
+    cat    \$INPUT_TEMP_PS | /opt/brother/${device_model}/${printer_model}/lpd/filter${printer_model} "\$\$" "CUPS" "USB"
 
     if [ \$LOGLEVEL -gt 2 ];  then
 	   if [ \$LOGFILE != "/dev/null" ]; then
@@ -868,40 +865,13 @@
 #   cp $brotherlpdwrapper  $brotherlpdwrapper64
 #fi
 
-chmod a+w /usr/local/Brother/${device_model}/${printer_model}/inf/br${printer_model}rc
-chmod a+w /usr/local/Brother/${device_model}/${printer_model}/inf
-if [ -e /etc/init.d/lpd ]; then
-   /etc/init.d/lpd stop
-fi
-if [  -e /etc/init.d/lprng ]; then
-   /etc/init.d/lprng stop
-fi
-
-
-if [  -e /etc/init.d/cups ]; then
-   /etc/init.d/cups restart
-elif [  -e /etc/init.d/cupsys ]; then
-   /etc/init.d/cupsys restart
+chmod a+w /opt/brother/${device_model}/${printer_model}/inf/br${printer_model}rc
+chmod a+w /opt/brother/${device_model}/${printer_model}/inf
+if [  -e /var/service/cupsd ]; then
+   sv restart cupsd
 fi
 
 sleep 2s
 
-
-port2=`lpinfo -v | grep -i 'usb://Brother/${device_name}' | head -1`
-if [ "$port2" = '' ];then
-    port2=`lpinfo -v | grep 'usb://Brother' | head -1`
-fi
-
-if [ "$port2" = '' ];then
-    port2=`lpinfo -v | grep 'usb://' | head -1`
-fi
-port=`echo $port2| sed s/direct//g`
-
-if [ "$port" = '' ];then
-    port=usb:/dev/usb/lp0
-fi
-#lpadmin -p ${printer_name} -E -v $port -m br${printer_model}.ppd
-lpadmin -p ${printer_name} -E -v $port -P $ppd_file_name
-
 exit 0
 
