--- a/src/cl_api_kernel.c	2017-09-22 10:05:22.000000000 +0200
+++ b/src/cl_api_kernel.c	2020-03-24 18:27:40.186485887 +0100
@@ -151,7 +151,7 @@
         fixed_local_sz[0] = 16;
         fixed_local_sz[1] = 1;
       } else {
-        uint j, maxDimSize = 64 /* from 64? */, maxGroupSize = 256; //MAX_WORK_GROUP_SIZE may too large
+        cl_uint j, maxDimSize = 64 /* from 64? */, maxGroupSize = 256; //MAX_WORK_GROUP_SIZE may too large
         size_t realGroupSize = 1;
         for (i = 0; i < work_dim; i++) {
           for (j = maxDimSize; j > 1; j--) {
--- a/src/cl_enqueue.c	2017-09-22 10:05:22.000000000 +0200
+++ b/src/cl_enqueue.c	2020-03-24 18:30:12.031479833 +0100
@@ -494,7 +494,7 @@
 cl_int cl_enqueue_svm_free(enqueue_data *data, cl_int status) {
   int i;
   void **pointers = data->pointers;
-  uint num_svm_ptrs = data->size;
+  cl_uint num_svm_ptrs = data->size;
   cl_int err = CL_SUCCESS;
 
   if (status != CL_COMPLETE)
