--- a/configure.ac
+++ b/configure.ac
@@ -403,7 +403,7 @@
                              [$accheck_includes])])
 
 AC_MSG_CHECKING([for va_copy])
-AC_RUN_IFELSE([AC_LANG_PROGRAM([[
+AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
 $accheck_includes
 ]], [[
 va_list ap, ap2;
@@ -415,7 +415,7 @@
               [AC_MSG_RESULT([no])])
 
 AC_MSG_CHECKING([for __builtin_va_copy])
-AC_RUN_IFELSE([AC_LANG_PROGRAM([[
+AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
 $accheck_includes
 ]], [[
 va_list ap, ap2;
