Source: sgn
Index: lc0-0.29.0/src/mcts/node.h
===================================================================
--- lc0-0.29.0.orig/src/mcts/node.h
+++ lc0-0.29.0/src/mcts/node.h
@@ -328,6 +328,10 @@ class Node {
   // Whether the child_ is actually an array of equal length to edges.
   bool solid_children_ : 1;
 
+  // either uint8_t, uint16_t or uint32_t is fine.
+  // force size to 48 bytes in 32-bit platforms with 4 bytes alignment
+  uint8_t padding_ = 0;
+
   // TODO(mooskagh) Unfriend NodeTree.
   friend class NodeTree;
   friend class Edge_Iterator<true>;
