--- a/src/findFile.cpp	2013-03-27 22:57:44.000000000 +0100
+++ b/src/findFile.cpp	2014-02-17 17:16:01.256683107 +0100
@@ -37,14 +37,14 @@
     }
 
     #if defined(__UNIX__)
-    testPath = wxString(wxStandardPaths().GetInstallPrefix() + _T("/share/") + _T(PACKAGE) + _T("/") + str);
+    testPath = wxString(wxStandardPaths::Get().GetInstallPrefix() + _T("/share/") + _T(PACKAGE) + _T("/") + str);
     if(wxFileExists(testPath) || wxDirExists(testPath)) // Si c'est dans le dossier /usr/.../share/nomprog
     {
         return testPath;
     }
     #endif // defined(__UNIX__)
 
-    testPath = wxString(wxStandardPaths().GetDataDir() + _T("/") + str);
+    testPath = wxString(wxStandardPaths::Get().GetDataDir() + _T("/") + str);
     if(wxFileExists(testPath) || wxDirExists(testPath)) // Si c'est dans le dossier /usr/.../share/nomPaquet
     {
         return testPath;
