Index: mako-1.6/meson.build
===================================================================
--- mako-1.6.orig/meson.build
+++ mako-1.6/meson.build
@@ -130,7 +130,7 @@ configure_file(
 	install_dir: datadir + '/dbus-1/services',
 )
 
-scdoc = dependency('scdoc', required: get_option('man-pages'), version: '>= 1.9.7')
+scdoc = dependency('scdoc', required: get_option('man-pages'), version: '>= 1.9.7', native: true)
 
 if scdoc.found()
 	sh = find_program('sh')
