diff --git a/x2goserver/sbin/x2gocleansessions b/x2goserver/sbin/x2gocleansessions
index e78f677..8e2ba43 100755
--- a/x2goserver/sbin/x2gocleansessions
+++ b/x2goserver/sbin/x2gocleansessions
@@ -40,15 +40,17 @@ use Pod::Usage;
 Getopt::Long::Configure("gnu_getopt", "no_auto_abbrev");
 
 my $debug = 0;
+my $nofork = 0;
 my $help = 0;
 my $man = 0;
-GetOptions('debug|d' => \$debug, 'help|?|h' => \$help, 'man' => \$man) or pod2usage(2);
+GetOptions('debug|d' => \$debug, 'nofork|n' => \$nofork, 'help|?|h' => \$help, 'man' => \$man) or pod2usage(2);
 pod2usage(1) if $help;
 pod2usage(-verbose => 2, -exitval => 0) if $man;
 
 openlog($0,'cons,pid','user');
 if ($debug)
 {
+	$nofork = 1;
 	setlogmask( LOG_UPTO (LOG_DEBUG) );
 }
 else
@@ -99,12 +101,12 @@ my $uname;
 my $serv = hostname;
 my $pid;
 
-if (! $debug)
+if (! $nofork)
 {
 	$pid = fork();
 }
 
-if ((!$debug) && (not defined $pid))
+if ((!$nofork) && (not defined $pid))
 {
 	print "resources not avilable.\n";
 }
@@ -128,7 +130,7 @@ elsif ($pid == 0 )
 	my $superenice_idle=$Config->param("superenicer.idle-nice-level");
 	my $superenice_ignoredusers=$Config->param("superenicer.ignored-users");
 
-	if ( ! $debug )
+	if ( ! $nofork )
 	{
 		# close any open file descriptor left open by our parent before the fork
 		my $fd;
@@ -377,7 +379,8 @@ x2gocleansessions [options]
   Options:
     --help|-h|-?            brief help message
     --man                   full documentation
-    --debug|-d              enable debugging and don't daemonize
+    --debug|-d              enable debugging; implies --nofork
+    --nofork|-n             don't daemonize
 
 =head1 OPTIONS
 
@@ -393,8 +396,11 @@ Prints the manual page and exits.
 
 =item B<--debug>|B<-d>
 
-Override debugging setting in global config and keep application in foreground
-instead of daemonizing.
+Override debugging setting in global config; implies B<--nofork>.
+
+=item B<--nofork>|B<-n>
+
+Keep application in foreground instead of daemonizing.
 
 =back
 
