--- include/libssl_compat.h.orig	2016-11-21 13:28:40.000000000 +0100
+++ include/libssl_compat.h	2016-11-24 12:43:14.297426937 +0100
@@ -25,7 +25,7 @@
 #include "openssl/rsa.h"
 
 /* ----------------------------------------------------------------- */
-#if OPENSSL_VERSION_NUMBER < 0x10100000L
+#if OPENSSL_VERSION_NUMBER < 0x10100000L || defined(LIBRESSL_VERSION_NUMBER)
 /* ----------------------------------------------------------------- */
 
 # include <openssl/objects.h>
--- libntp/libssl_compat.c.orig	2017-02-01 10:47:13.000000000 +0100
+++ libntp/libssl_compat.c	2017-03-23 13:03:13.569465673 +0100
@@ -26,7 +26,7 @@
 /* ----------------------------------------------------------------- */
 
 /* ----------------------------------------------------------------- */
-#if defined(OPENSSL) && OPENSSL_VERSION_NUMBER < 0x10100000L
+#if defined(OPENSSL) && (OPENSSL_VERSION_NUMBER < 0x10100000L || defined(LIBRESSL_VERSION_NUMBER))
 /* ----------------------------------------------------------------- */
 
 #include "libssl_compat.h"
