From 5d7ddc111e761e526ec05e8967cd4b95b8fc5eaf Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Piotr=20W=C3=B3jcik?= <chocimier@tlen.pl>
Date: Fri, 17 Apr 2020 18:24:33 +0200
Subject: [PATCH] pdf_test


diff --git tests/comparators/test_directory.py tests/comparators/test_directory.py
index a1b80d4..2434d82 100644
--- tests/comparators/test_directory.py
+++ tests/comparators/test_directory.py
@@ -73,10 +73,6 @@ def test_content(differences):
     assert differences[0].details[0].unified_diff == expected_diff
 
 
-def test_stat(differences):
-    assert 'stat' in differences[0].details[0].details[0].source1
-
-
 def test_compare_to_file(tmpdir):
     path = str(tmpdir.join('file'))
 
-- 
2.26.0

