commit 09e233ed73aa3335642f97c080ef27e42047c527
Author: q66 <daniel@octaforge.org>
Date:   Fri Mar 11 04:37:15 2022 +0100

    constrain aeson to prevent build failures

diff --git a/stack.cabal b/stack.cabal
index dba5bf0..0195f2c 100644
--- a/stack.cabal
+++ b/stack.cabal
@@ -231,7 +231,7 @@ library
 
     build-depends:
         Cabal >=3.2.1.0,
-        aeson >=1.5.6.0,
+        aeson >=1.5.6.0 && <1.6,
         annotated-wl-pprint >=0.7.0,
         ansi-terminal >=0.10.3,
         array >=0.5.4.0,
@@ -357,7 +357,7 @@ executable stack
 
     build-depends:
         Cabal >=3.2.1.0,
-        aeson >=1.5.6.0,
+        aeson >=1.5.6.0 && <1.6,
         annotated-wl-pprint >=0.7.0,
         ansi-terminal >=0.10.3,
         array >=0.5.4.0,
@@ -485,7 +485,7 @@ executable stack-integration-test
 
     build-depends:
         Cabal >=3.2.1.0,
-        aeson >=1.5.6.0,
+        aeson >=1.5.6.0 && <1.6,
         annotated-wl-pprint >=0.7.0,
         ansi-terminal >=0.10.3,
         array >=0.5.4.0,
@@ -621,7 +621,7 @@ test-suite stack-test
     build-depends:
         Cabal >=3.2.1.0,
         QuickCheck >=2.14.2,
-        aeson >=1.5.6.0,
+        aeson >=1.5.6.0 && <1.6,
         annotated-wl-pprint >=0.7.0,
         ansi-terminal >=0.10.3,
         array >=0.5.4.0,
