--- a/src/Stack/Storage/User.hs	2020-10-15 15:51:28.000000000 +0200
+++ b/src/Stack/Storage/User.hs	2020-12-06 20:51:06.632607164 +0100
@@ -1,3 +1,4 @@
+{-# LANGUAGE DataKinds #-}
 {-# LANGUAGE DerivingStrategies #-}
 {-# LANGUAGE FlexibleInstances #-}
 {-# LANGUAGE GADTs #-}
--- a/src/Stack/Storage/Project.hs	2020-10-15 15:51:28.000000000 +0200
+++ b/src/Stack/Storage/Project.hs	2020-12-06 21:55:21.349484738 +0100
@@ -1,3 +1,4 @@
+{-# LANGUAGE DataKinds #-}
 {-# LANGUAGE DerivingStrategies #-}
 {-# LANGUAGE FlexibleInstances #-}
 {-# LANGUAGE GADTs #-}
