--- a/src/Stack/Build/Execute.hs
+++ b/src/Stack/Build/Execute.hs
@@ -26,7 +26,7 @@ import           Control.Concurrent.Exec
 import           Control.Concurrent.STM (check)
 import           Stack.Prelude hiding (Display (..))
 import           Crypto.Hash
-import           Data.Attoparsec.Text hiding (try)
+import           Data.Attoparsec.Text as P hiding (try)
 import qualified Data.ByteArray as Mem (convert)
 import qualified Data.ByteString as S
 import qualified Data.ByteString.Builder
@@ -2160,7 +2160,7 @@ mungeBuildOutput excludeTHLoading makeAb
     lineCol = char ':'
            >> choice
                 [ num >> char ':' >> num >> optional (char '-' >> num) >> return ()
-                , char '(' >> num >> char ',' >> num >> string ")-(" >> num >> char ',' >> num >> char ')' >> return ()
+                , char '(' >> num >> char ',' >> num >> P.string ")-(" >> num >> char ',' >> num >> char ')' >> return ()
                 ]
            >> char ':'
            >> return ()
