constrain http-download to prevent build failures

--- a/stack.cabal
+++ b/stack.cabal
@@ -267,7 +267,7 @@ library
         http-client >=0.6.4.1,
         http-client-tls >=0.3.5.3,
         http-conduit >=2.3.8,
-        http-download >=0.2.0.0,
+        http-download >=0.2.0.0 && <0.2.1,
         http-types >=0.12.3,
         memory >=0.15.0,
         microlens >=0.4.11.2,
@@ -393,7 +393,7 @@ executable stack
         http-client >=0.6.4.1,
         http-client-tls >=0.3.5.3,
         http-conduit >=2.3.8,
-        http-download >=0.2.0.0,
+        http-download >=0.2.0.0 && <0.2.1,
         http-types >=0.12.3,
         memory >=0.15.0,
         microlens >=0.4.11.2,
@@ -522,7 +522,7 @@ executable stack-integration-test
         http-client >=0.6.4.1,
         http-client-tls >=0.3.5.3,
         http-conduit >=2.3.8,
-        http-download >=0.2.0.0,
+        http-download >=0.2.0.0 && <0.2.1,
         http-types >=0.12.3,
         memory >=0.15.0,
         microlens >=0.4.11.2,
@@ -658,7 +658,7 @@ test-suite stack-test
         http-client >=0.6.4.1,
         http-client-tls >=0.3.5.3,
         http-conduit >=2.3.8,
-        http-download >=0.2.0.0,
+        http-download >=0.2.0.0 && <0.2.1,
         http-types >=0.12.3,
         memory >=0.15.0,
         microlens >=0.4.11.2,
