commit 188d7ece06ee06ed6ccaf0df0c46285bd43898f4
Author: q66 <daniel@octaforge.org>
Date:   Thu Apr 28 05:02:09 2022 +0200

    constrain persistent to fix build

diff --git a/stack.cabal b/stack.cabal
index dba5bf0..e221321 100644
--- a/stack.cabal
+++ b/stack.cabal
@@ -281,7 +281,7 @@ library
         pantry >=0.5.3,
         path >=0.7.0,
         path-io >=1.6.3,
-        persistent >=2.11.0.4,
+        persistent >=2.11.0.4 && <2.14.0.0,
         persistent-sqlite >=2.11.1.0,
         persistent-template >=2.9.1.0,
         pretty >=1.1.3.6,
@@ -407,7 +407,7 @@ executable stack
         pantry >=0.5.3,
         path >=0.7.0,
         path-io >=1.6.3,
-        persistent >=2.11.0.4,
+        persistent >=2.11.0.4 && <2.14.0.0,
         persistent-sqlite >=2.11.1.0,
         persistent-template >=2.9.1.0,
         pretty >=1.1.3.6,
@@ -537,7 +537,7 @@ executable stack-integration-test
         pantry >=0.5.3,
         path >=0.7.0,
         path-io >=1.6.3,
-        persistent >=2.11.0.4,
+        persistent >=2.11.0.4 && <2.14.0.0,
         persistent-sqlite >=2.11.1.0,
         persistent-template >=2.9.1.0,
         pretty >=1.1.3.6,
@@ -672,7 +672,7 @@ test-suite stack-test
         pantry >=0.5.3,
         path >=0.7.0,
         path-io >=1.6.3,
-        persistent >=2.11.0.4,
+        persistent >=2.11.0.4 && <2.14.0.0,
         persistent-sqlite >=2.11.1.0,
         persistent-template >=2.9.1.0,
         pretty >=1.1.3.6,
