--- a/src/Stack/Setup.hs
+++ b/src/Stack/Setup.hs
@@ -837,9 +837,8 @@
    if compilerTool `elem` installed
      then return (compilerTool,CompilerBuildStandard)
      else do
-       let repo = Repo
-            { repoCommit = commitId
-            , repoUrl    = url
-            , repoType   = RepoGit
-            , repoSubdir = mempty
+       let repo = SimpleRepo
+            { sRepoCommit = commitId
+            , sRepoUrl    = url
+            , sRepoType   = RepoGit
             }

        -- clone the repository and execute the given commands
