constrain unix-compat version to prevent build failures

--- a/stack.cabal
+++ b/stack.cabal
@@ -307,7 +308,7 @@ library
         transformers >=0.5.6.2,
         typed-process >=0.2.6.0,
         unicode-transforms >=0.3.7.1,
-        unix-compat >=0.5.3,
+        unix-compat >=0.5.3 && < 0.7,
         unliftio >=0.2.18,
         unordered-containers >=0.2.14.0,
         vector >=0.12.1.2,
@@ -434,7 +435,7 @@ executable stack
         transformers >=0.5.6.2,
         typed-process >=0.2.6.0,
         unicode-transforms >=0.3.7.1,
-        unix-compat >=0.5.3,
+        unix-compat >=0.5.3 && < 0.7,
         unliftio >=0.2.18,
         unordered-containers >=0.2.14.0,
         vector >=0.12.1.2,
@@ -563,7 +564,7 @@ executable stack-integration-test
         transformers >=0.5.6.2,
         typed-process >=0.2.6.0,
         unicode-transforms >=0.3.7.1,
-        unix-compat >=0.5.3,
+        unix-compat >=0.5.3 && < 0.7,
         unliftio >=0.2.18,
         unordered-containers >=0.2.14.0,
         vector >=0.12.1.2,
@@ -701,7 +702,7 @@ test-suite stack-test
         transformers >=0.5.6.2,
         typed-process >=0.2.6.0,
         unicode-transforms >=0.3.7.1,
-        unix-compat >=0.5.3,
+        unix-compat >=0.5.3 && < 0.7,
         unliftio >=0.2.18,
         unordered-containers >=0.2.14.0,
         vector >=0.12.1.2,
