--- a/src/CMakeLists.txt	2017-12-02 13:38:52.784843265 +0100
+++ b/src/CMakeLists.txt	2017-12-02 13:39:15.028841808 +0100
@@ -1,6 +1,8 @@
 include_directories(${LIBINTL_INCLUDE_DIR})
 include_directories(${LIBICONV_INCLUDE_DIR})
-include_directories(${LIBEXECINFO_INCLUDE_DIR})
+if(ENABLE_BACKTRACE)
+	include_directories(${LIBEXECINFO_INCLUDE_DIR})
+endif()
 include_directories(${DL_INCLUDE_DIR})
 include_directories(${PTHREAD_INCLUDE_DIR})
 
