diff --git src/lib/fcitx-gclient/CMakeLists.txt src/lib/fcitx-gclient/CMakeLists.txt
index ea1d50f..20b2359 100644
--- a/src/lib/fcitx-gclient/CMakeLists.txt
+++ b/src/lib/fcitx-gclient/CMakeLists.txt
@@ -65,6 +65,8 @@ install(FILES ${FCITX_GCLIENT_HEADERS}
 
 if(ENABLE_GIR)
   include(GObjectIntrospection)
+  string(REPLACE "///" "/" GOBJECT_INTROSPECTION_GIRDIR "${GOBJECT_INTROSPECTION_GIRDIR}")
+  string(REPLACE "///" "/" GOBJECT_INTROSPECTION_TYPELIBDIR "${GOBJECT_INTROSPECTION_TYPELIBDIR}")
   gobject_introspection(
     FILENAME Fcitx-1.0.gir
     NSVERSION 1.0

