diff --git a/src/im/pinyin/py.c b/src/im/pinyin/py.c
index e0028854..c45c8b4d 100644
--- a/src/im/pinyin/py.c
+++ b/src/im/pinyin/py.c
@@ -19,6 +19,7 @@
  ***************************************************************************/
 #include "config.h"
 
+#define _FILE_OFFSET_BITS 64
 #include <libintl.h>
 #include <stdio.h>
 #include <limits.h>
diff --git a/src/lib/fcitx-config/xdg.c b/src/lib/fcitx-config/xdg.c
index 1559e7bb..cdfbae91 100644
--- a/src/lib/fcitx-config/xdg.c
+++ b/src/lib/fcitx-config/xdg.c
@@ -27,7 +27,7 @@
  */
 
 #define FCITX_CONFIG_XDG_DEPRECATED
-
+#define _FILE_OFFSET_BITS 64
 #include "config.h"
 
 #include <limits.h>
diff --git a/src/lib/fcitx-utils/utils.h b/src/lib/fcitx-utils/utils.h
index 0d6bf309..a930ced1 100644
--- a/src/lib/fcitx-utils/utils.h
+++ b/src/lib/fcitx-utils/utils.h
@@ -60,6 +60,7 @@
 #ifndef _FCITX_UTILS_H_
 #define _FCITX_UTILS_H_
 
+#define _FILE_OFFSET_BITS 64
 #include <stdio.h>
 #include <stdint.h>
 #include <unistd.h>
diff --git a/src/module/spell/dict/comp_spell_dict.c b/src/module/spell/dict/comp_spell_dict.c
index 707226dd..4b8eb2a8 100644
--- a/src/module/spell/dict/comp_spell_dict.c
+++ b/src/module/spell/dict/comp_spell_dict.c
@@ -17,7 +17,7 @@
  *   Free Software Foundation, Inc.,                                       *
  *   51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA.              *
  ***************************************************************************/
-
+#define _FILE_OFFSET_BITS 64
 #include "fcitx-utils/log.h"
 #include <stdlib.h>
 #include <stdio.h>
diff --git a/src/module/spell/spell-custom-dict.c b/src/module/spell/spell-custom-dict.c
index 93b96266..5a45984b 100644
--- a/src/module/spell/spell-custom-dict.c
+++ b/src/module/spell/spell-custom-dict.c
@@ -17,7 +17,7 @@
  *   Free Software Foundation, Inc.,                                       *
  *   51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA.              *
  ***************************************************************************/
-
+#define _FILE_OFFSET_BITS 64
 #include "fcitx/fcitx.h"
 #include "config.h"
 
