diff --git a/./electron/shell/browser/ui/file_dialog.h b/./electron/shell/browser/ui/file_dialog.h
index 6cdfc7b..f7757da 100644
--- a/shell/browser/ui/file_dialog.h
+++ b/shell/browser/ui/file_dialog.h
@@ -13,10 +13,6 @@
 #include "base/files/file_path.h"
 #include "base/memory/raw_ptr_exclusion.h"
 
-#if BUILDFLAG(IS_LINUX)
-#include <bits/stdint-uintn.h>
-#endif
-
 namespace electron {
 class NativeWindow;
 }
