--- a/src/base/debug/profiler.h
+++ b/src/base/debug/profiler.h
@@ -6,7 +6,7 @@
 #define BASE_DEBUG_PROFILER_H_
 
 #include <stddef.h>
-
+#include <stdint.h>
 #include <string>
 
 #include "base/base_export.h"
--- a/src/cc/trees/target_property.cc
+++ b/src/cc/trees/target_property.cc
@@ -5,6 +5,7 @@
 #include "cc/trees/target_property.h"
 
 #include "ui/gfx/animation/keyframe/target_property.h"
+#include <cstdint>
 
 namespace cc {
 
--- a/src/components/crash/core/app/crash_reporter_client.h
+++ b/src/components/crash/core/app/crash_reporter_client.h
@@ -5,6 +5,7 @@
 #ifndef COMPONENTS_CRASH_CORE_APP_CRASH_REPORTER_CLIENT_H_
 #define COMPONENTS_CRASH_CORE_APP_CRASH_REPORTER_CLIENT_H_
 
+#include <cstdint>
 #include <map>
 #include <string>
 
--- a/src/components/feature_engagement/internal/event_storage_validator.h
+++ b/src/components/feature_engagement/internal/event_storage_validator.h
@@ -5,6 +5,7 @@
 #ifndef COMPONENTS_FEATURE_ENGAGEMENT_INTERNAL_EVENT_STORAGE_VALIDATOR_H_
 #define COMPONENTS_FEATURE_ENGAGEMENT_INTERNAL_EVENT_STORAGE_VALIDATOR_H_
 
+#include <cstdint>
 #include <string>
 
 namespace feature_engagement {
--- a/src/components/password_manager/core/browser/generation/password_generator.h
+++ b/src/components/password_manager/core/browser/generation/password_generator.h
@@ -5,6 +5,7 @@
 #ifndef COMPONENTS_PASSWORD_MANAGER_CORE_BROWSER_GENERATION_PASSWORD_GENERATOR_H_
 #define COMPONENTS_PASSWORD_MANAGER_CORE_BROWSER_GENERATION_PASSWORD_GENERATOR_H_
 
+#include <cstdint>
 #include <string>
 
 
--- a/src/components/viz/common/view_transition_element_resource_id.h
+++ b/src/components/viz/common/view_transition_element_resource_id.h
@@ -5,6 +5,7 @@
 #ifndef COMPONENTS_VIZ_COMMON_VIEW_TRANSITION_ELEMENT_RESOURCE_ID_H_
 #define COMPONENTS_VIZ_COMMON_VIEW_TRANSITION_ELEMENT_RESOURCE_ID_H_
 
+#include <cstdint>
 #include <string>
 #include <vector>
 
--- a/src/device/base/synchronization/one_writer_seqlock.h
+++ b/src/device/base/synchronization/one_writer_seqlock.h
@@ -6,6 +6,7 @@
 #define DEVICE_BASE_SYNCHRONIZATION_ONE_WRITER_SEQLOCK_H_
 
 #include <atomic>
+#include <cstdint>
 #include <type_traits>
 
 #include "base/atomicops.h"
--- a/src/gpu/config/gpu_feature_info.h
+++ b/src/gpu/config/gpu_feature_info.h
@@ -5,6 +5,7 @@
 #ifndef GPU_CONFIG_GPU_FEATURE_INFO_H_
 #define GPU_CONFIG_GPU_FEATURE_INFO_H_
 
+#include <cstdint>
 #include <string>
 #include <vector>
 
--- a/src/net/cert/pki/string_util.h
+++ b/src/net/cert/pki/string_util.h
@@ -7,6 +7,7 @@
 
 #include "net/base/net_export.h"
 
+#include <cstdint>
 #include <string_view>
 #include <vector>
 
--- a/src/net/cookies/cookie_inclusion_status.h
+++ b/src/net/cookies/cookie_inclusion_status.h
@@ -5,6 +5,7 @@
 #ifndef NET_COOKIES_COOKIE_INCLUSION_STATUS_H_
 #define NET_COOKIES_COOKIE_INCLUSION_STATUS_H_
 
+#include <cstdint>
 #include <bitset>
 #include <ostream>
 #include <string>
--- a/src/net/third_party/quiche/src/quiche/quic/core/crypto/quic_hkdf.h
+++ b/src/net/third_party/quiche/src/quiche/quic/core/crypto/quic_hkdf.h
@@ -5,6 +5,7 @@
 #ifndef QUICHE_QUIC_CORE_CRYPTO_QUIC_HKDF_H_
 #define QUICHE_QUIC_CORE_CRYPTO_QUIC_HKDF_H_
 
+#include <cstdint>
 #include <vector>
 
 #include "absl/strings/string_view.h"
--- a/src/net/third_party/quiche/src/quiche/quic/core/quic_connection_id.h
+++ b/src/net/third_party/quiche/src/quiche/quic/core/quic_connection_id.h
@@ -5,6 +5,7 @@
 #ifndef QUICHE_QUIC_CORE_QUIC_CONNECTION_ID_H_
 #define QUICHE_QUIC_CORE_QUIC_CONNECTION_ID_H_
 
+#include <cstdint>
 #include <string>
 #include <vector>
 
--- a/src/third_party/dawn/src/dawn/native/stream/ByteVectorSink.h
+++ b/src/third_party/dawn/src/dawn/native/stream/ByteVectorSink.h
@@ -15,6 +15,7 @@
 #ifndef SRC_DAWN_NATIVE_STREAM_BYTEVECTORSINK_H_
 #define SRC_DAWN_NATIVE_STREAM_BYTEVECTORSINK_H_
 
+#include <cstdint>
 #include <ostream>
 #include <vector>
 
--- a/src/third_party/maldoca/src/maldoca/ole/header.h
+++ b/src/third_party/maldoca/src/maldoca/ole/header.h
@@ -43,6 +43,7 @@
 #ifndef MALDOCA_OLE_HEADER_H_
 #define MALDOCA_OLE_HEADER_H_
 
+#include <cstdint>
 #include "absl/strings/string_view.h"
 
 namespace maldoca {
--- a/src/third_party/openscreen/src/discovery/dnssd/public/dns_sd_txt_record.h
+++ b/src/third_party/openscreen/src/discovery/dnssd/public/dns_sd_txt_record.h
@@ -5,6 +5,7 @@
 #ifndef DISCOVERY_DNSSD_PUBLIC_DNS_SD_TXT_RECORD_H_
 #define DISCOVERY_DNSSD_PUBLIC_DNS_SD_TXT_RECORD_H_
 
+#include <cstdint>
 #include <functional>
 #include <map>
 #include <set>
--- a/src/third_party/openscreen/src/util/base64.h
+++ b/src/third_party/openscreen/src/util/base64.h
@@ -5,6 +5,7 @@
 #ifndef UTIL_BASE64_H_
 #define UTIL_BASE64_H_
 
+#include <cstdint>
 #include <string>
 #include <vector>
 
--- a/src/third_party/perfetto/include/perfetto/ext/base/uuid.h
+++ b/src/third_party/perfetto/include/perfetto/ext/base/uuid.h
@@ -18,6 +18,7 @@
 #define INCLUDE_PERFETTO_EXT_BASE_UUID_H_
 
 #include <array>
+#include <cstdint>
 #include <string>
 
 #include "perfetto/ext/base/optional.h"
--- a/src/third_party/ruy/src/ruy/profiler/instrumentation.h
+++ b/src/third_party/ruy/src/ruy/profiler/instrumentation.h
@@ -20,6 +20,7 @@ limitations under the License.
 #include <cstdio>
 #include <mutex>
 #include <vector>
+#include <string>
 #endif
 
 namespace ruy {
--- a/src/third_party/swiftshader/src/System/LRUCache.hpp
+++ b/src/third_party/swiftshader/src/System/LRUCache.hpp
@@ -18,6 +18,7 @@
 #include "System/Debug.hpp"
 
 #include <cstddef>
+#include <cstdint>
 #include <functional>
 #include <unordered_set>
 #include <vector>
--- a/src/third_party/tflite/src/tensorflow/lite/kernels/internal/spectrogram.cc
+++ b/src/third_party/tflite/src/tensorflow/lite/kernels/internal/spectrogram.cc
@@ -17,6 +17,7 @@ limitations under the License.
 
 #include <assert.h>
 #include <math.h>
+#include <stdint.h>
 
 #include "third_party/fft2d/fft.h"
 
--- a/src/third_party/vulkan_memory_allocator/include/vk_mem_alloc.h
+++ b/src/third_party/vulkan_memory_allocator/include/vk_mem_alloc.h
@@ -2389,6 +2389,7 @@ VMA_CALL_PRE void VMA_CALL_POST vmaFreeV
 #undef VMA_IMPLEMENTATION
 
 #include <cstdint>
+#include <cstdio>
 #include <cstdlib>
 #include <cstring>
 #include <utility>
--- a/src/third_party/webrtc/rtc_base/third_party/base64/base64.h
+++ b/src/third_party/webrtc/rtc_base/third_party/base64/base64.h
@@ -12,6 +12,7 @@
 #ifndef RTC_BASE_THIRD_PARTY_BASE64_BASE64_H_
 #define RTC_BASE_THIRD_PARTY_BASE64_BASE64_H_
 
+#include <cstdint>
 #include <string>
 #include <vector>
 
--- a/src/ui/base/prediction/kalman_filter.h
+++ b/src/ui/base/prediction/kalman_filter.h
@@ -5,6 +5,7 @@
 #ifndef UI_BASE_PREDICTION_KALMAN_FILTER_H_
 #define UI_BASE_PREDICTION_KALMAN_FILTER_H_
 
+#include <cstdint>
 #include "base/component_export.h"
 #include "ui/gfx/geometry/matrix3_f.h"
 
--- a/src/ui/events/types/scroll_types.h
+++ b/src/ui/events/types/scroll_types.h
@@ -5,6 +5,8 @@
 #ifndef UI_EVENTS_TYPES_SCROLL_TYPES_H_
 #define UI_EVENTS_TYPES_SCROLL_TYPES_H_
 
+#include <cstdint>
+
 namespace ui {
 
 enum class ScrollGranularity : uint8_t {
--- a/src/ui/gfx/geometry/linear_gradient.h
+++ b/src/ui/gfx/geometry/linear_gradient.h
@@ -6,6 +6,7 @@
 #define UI_GFX_LINEAR_GRADIENT_H_
 
 #include <array>
+#include <cstdint>
 #include <string>
 
 #include "ui/gfx/geometry/geometry_skia_export.h"
--- a/src/components/autofill/core/browser/autofill_ablation_study.h
+++ b/src/components/autofill/core/browser/autofill_ablation_study.h
@@ -5,6 +5,7 @@
 #ifndef COMPONENTS_AUTOFILL_CORE_BROWSER_AUTOFILL_ABLATION_STUDY_H_
 #define COMPONENTS_AUTOFILL_CORE_BROWSER_AUTOFILL_ABLATION_STUDY_H_
 
+#include <cstdint>
 #include <string>
 
 class GURL;
--- a/src/third_party/blink/public/common/bluetooth/web_bluetooth_device_id.h
+++ b/src/third_party/blink/public/common/bluetooth/web_bluetooth_device_id.h
@@ -6,6 +6,7 @@
 #define THIRD_PARTY_BLINK_PUBLIC_COMMON_BLUETOOTH_WEB_BLUETOOTH_DEVICE_ID_H_
 
 #include <array>
+#include <cstdint>
 #include <string>
 
 #include "third_party/blink/public/common/common_export.h"
--- a/src/gin/time_clamper.h
+++ b/src/gin/time_clamper.h
@@ -6,6 +6,7 @@
 #define GIN_TIME_CLAMPER_H_
 
 #include <algorithm>
+#include <cstdint>
 
 #include "base/rand_util.h"
 #include "base/time/time.h"
--- a/src/pdf/document_attachment_info.h
+++ b/src/pdf/document_attachment_info.h
@@ -5,6 +5,7 @@
 #ifndef PDF_DOCUMENT_ATTACHMENT_INFO_H_
 #define PDF_DOCUMENT_ATTACHMENT_INFO_H_
 
+#include <cstdint>
 #include <string>
 
 
--- a/src/third_party/pdfium/constants/annotation_flags.h
+++ b/src/third_party/pdfium/constants/annotation_flags.h
@@ -5,6 +5,8 @@
 #ifndef CONSTANTS_ANNOTATION_FLAGS_H_
 #define CONSTANTS_ANNOTATION_FLAGS_H_
 
+#include <stdint.h>
+
 namespace pdfium {
 namespace annotation_flags {
 
--- a/src/third_party/openscreen/src/platform/base/error.h
+++ b/src/third_party/openscreen/src/platform/base/error.h
@@ -6,6 +6,7 @@
 #define PLATFORM_BASE_ERROR_H_
 
 #include <cassert>
+#include <cstdint>
 #include <ostream>
 #include <string>
 #include <utility>
--- a/src/third_party/dawn/src/tint/reader/wgsl/token.h
+++ b/src/third_party/dawn/src/tint/reader/wgsl/token.h
@@ -15,6 +15,7 @@
 #ifndef SRC_TINT_READER_WGSL_TOKEN_H_
 #define SRC_TINT_READER_WGSL_TOKEN_H_
 
+#include <cstdint>
 #include <string>
 #include <string_view>
 #include <variant>
--- a/src/third_party/webrtc/rtc_base/system/file_wrapper.h
+++ b/src/third_party/webrtc/rtc_base/system/file_wrapper.h
@@ -12,6 +12,7 @@
 #define RTC_BASE_SYSTEM_FILE_WRAPPER_H_
 
 #include <stddef.h>
+#include <stdint.h>
 #include <stdio.h>
 
 #include <string>
--- a/src/components/dom_distiller/core/url_utils.h
+++ b/src/components/dom_distiller/core/url_utils.h
@@ -5,6 +5,7 @@
 #ifndef COMPONENTS_DOM_DISTILLER_CORE_URL_UTILS_H_
 #define COMPONENTS_DOM_DISTILLER_CORE_URL_UTILS_H_
 
+#include <cstdint>
 #include <string>
 
 #include "base/strings/string_piece_forward.h"
--- a/src/gpu/config/gpu_driver_bug_workarounds.h
+++ b/src/gpu/config/gpu_driver_bug_workarounds.h
@@ -5,6 +5,7 @@
 #ifndef GPU_CONFIG_GPU_DRIVER_BUG_WORKAROUNDS_H_
 #define GPU_CONFIG_GPU_DRIVER_BUG_WORKAROUNDS_H_
 
+#include <cstdint>
 #include <vector>
 
 #include "build/build_config.h"
--- a/src/media/base/video_transformation.h
+++ b/src/media/base/video_transformation.h
@@ -5,6 +5,7 @@
 #ifndef MEDIA_BASE_VIDEO_TRANSFORMATION_H_
 #define MEDIA_BASE_VIDEO_TRANSFORMATION_H_
 
+#include <cstdint>
 #include <string>
 
 #include "base/numerics/math_constants.h"
--- a/src/third_party/blink/public/common/page_state/page_state.h
+++ b/src/third_party/blink/public/common/page_state/page_state.h
@@ -5,6 +5,7 @@
 #ifndef THIRD_PARTY_BLINK_PUBLIC_COMMON_PAGE_STATE_PAGE_STATE_H_
 #define THIRD_PARTY_BLINK_PUBLIC_COMMON_PAGE_STATE_PAGE_STATE_H_
 
+#include <stdint.h>
 #include <string>
 #include <vector>
 
--- a/src/components/autofill/core/browser/strike_databases/strike_database_base.h
+++ b/src/components/autofill/core/browser/strike_databases/strike_database_base.h
@@ -5,6 +5,7 @@
 #ifndef COMPONENTS_AUTOFILL_CORE_BROWSER_STRIKE_DATABASES_STRIKE_DATABASE_BASE_H_
 #define COMPONENTS_AUTOFILL_CORE_BROWSER_STRIKE_DATABASES_STRIKE_DATABASE_BASE_H_
 
+#include <cstdint>
 #include <map>
 #include <string>
 #include <vector>
--- a/src/ui/accessibility/ax_text_attributes.h
+++ b/src/ui/accessibility/ax_text_attributes.h
@@ -5,6 +5,7 @@
 #ifndef UI_ACCESSIBILITY_AX_TEXT_ATTRIBUTES_H_
 #define UI_ACCESSIBILITY_AX_TEXT_ATTRIBUTES_H_
 
+#include <cstdint>
 #include <string>
 #include <vector>
 
