--- src/linux/LinuxForceFeedback.cpp	2010-04-01 05:06:55.000000000 +0200
+++ src/linux/LinuxForceFeedback.cpp	2015-08-03 13:47:54.690123238 +0200
@@ -24,6 +24,7 @@
 #include "OISException.h"

 #include <cstdlib>
+#include <unistd.h>
 #include <errno.h>
 #include <memory.h>

--- src/linux/LinuxJoyStickEvents.cpp	2010-07-26 03:28:41.000000000 +0200
+++ src/linux/LinuxJoyStickEvents.cpp	2015-08-03 13:48:14.720124682 +0200
@@ -30,6 +30,7 @@
 #include "OISEvents.h"
 #include "OISException.h"

+#include <unistd.h>
 #include <fcntl.h>        //Needed to Open a file descriptor
 #include <cassert>
 #include <linux/input.h>
--- demos/OISConsole.cpp	2010-08-16 00:51:34.000000000 +0200
+++ demos/OISConsole.cpp	2015-08-03 13:52:09.446141604 +0200
@@ -9,6 +9,7 @@
 //Advanced Usage
 #include "OISForceFeedback.h"

+#include <unistd.h>
 #include <iostream>
 #include <vector>
 #include <sstream>
--- demos/FFConsoleDemo.cpp	2010-04-01 05:38:04.000000000 +0200
+++ demos/FFConsoleDemo.cpp	2015-08-03 13:53:47.366148664 +0200
@@ -1,6 +1,7 @@
 #include "OIS.h"

 #include <math.h>
+#include <unistd.h>
 #include <cstdlib>
 #include <iostream>
 #include <iomanip>
