--- taskcluster/docker/firefox-snap/firefox.desktop.orig	2019-01-18 19:31:39.428839442 +0100
+++ taskcluster/docker/firefox-snap/firefox.desktop	2019-01-18 19:32:20.689063456 +0100
@@ -154,7 +154,7 @@
 Terminal=false
 X-MultipleArgs=false
 Type=Application
-Icon=/browser/chrome/icons/default/default128.png
+Icon=firefox
 Categories=GNOME;GTK;Network;WebBrowser;
 MimeType=text/html;text/xml;application/xhtml+xml;application/xml;application/rss+xml;application/rdf+xml;image/gif;image/jpeg;image/png;x-scheme-handler/http;x-scheme-handler/https;x-scheme-handler/ftp;x-scheme-handler/chrome;video/webm;application/x-xpinstall;
 StartupNotify=true
