 build/moz.configure/rust.configure | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git c/build/moz.configure/rust.configure i/build/moz.configure/rust.configure
index cd86b24153debb1b..44911715e25d95e3 100644
--- build/moz.configure/rust.configure
+++ build/moz.configure/rust.configure
@@ -81,7 +81,7 @@ def rust_target(rust_compiler, rustc, target, cross_compiling):
             # OpenBSD
             ('x86_64', 'OpenBSD'): 'x86_64-unknown-openbsd',
             # Linux
-            ('x86', 'Linux'): 'i586-unknown-linux-gnu',
+            ('x86', 'Linux'): 'i686-unknown-linux-gnu',
             # Linux
             ('x86_64', 'Linux'): 'x86_64-unknown-linux-gnu',
             # OS X and iOS
