diff -Nur dev-requirements.txt dev-requirements.txt
--- dev-requirements.txt	2014-07-07 22:56:01.000000000 +0200
+++ dev-requirements.txt	2014-08-19 02:10:38.507546859 +0200
@@ -3,3 +3,5 @@
 tornado==3.2.2
 coverage==3.7.1
 tox==1.7.1
+six
+backports.ssl_match_hostname
diff -Nur dummyserver/handlers.py dummyserver/handlers.py
--- dummyserver/handlers.py	2014-07-07 22:56:01.000000000 +0200
+++ dummyserver/handlers.py	2014-08-19 02:03:31.933539957 +0200
@@ -211,7 +211,7 @@
     """
     import tornado.httputil
     import email.utils
-    from urllib3.packages import six
+    import six
     if not six.PY3:
         line = line.encode('utf-8')
     parts = tornado.httputil._parseparam(';' + line)
diff -Nur setup.py setup.py
--- setup.py	2014-07-07 22:56:01.000000000 +0200
+++ setup.py	2014-08-19 01:51:36.502528381 +0200
@@ -42,7 +42,6 @@
       url='http://urllib3.readthedocs.org/',
       license='MIT',
       packages=['urllib3',
-                'urllib3.packages', 'urllib3.packages.ssl_match_hostname',
                 'urllib3.contrib', 'urllib3.util',
                 ],
       requires=[],
diff -Nur test/test_collections.py test/test_collections.py
--- test/test_collections.py	2014-03-15 01:05:07.000000000 +0100
+++ test/test_collections.py	2014-08-19 01:52:16.527529028 +0200
@@ -4,7 +4,7 @@
     HTTPHeaderDict,
     RecentlyUsedContainer as Container
 )
-from urllib3.packages import six
+import six
 xrange = six.moves.xrange
 
 
diff -Nur test/test_connectionpool.py test/test_connectionpool.py
--- test/test_connectionpool.py	2014-07-07 22:56:01.000000000 +0200
+++ test/test_connectionpool.py	2014-08-19 01:54:30.020531188 +0200
@@ -6,7 +6,10 @@
     HTTPConnectionPool,
 )
 from urllib3.util.timeout import Timeout
-from urllib3.packages.ssl_match_hostname import CertificateError
+try:
+    from ssl import CertificateError
+except ImportError:
+    from backports.ssl_match_hostname import CertificateError
 from urllib3.exceptions import (
     ClosedPoolError,
     EmptyPoolError,
diff -Nur test/test_fields.py test/test_fields.py
--- test/test_fields.py	2014-03-04 20:08:03.000000000 +0100
+++ test/test_fields.py	2014-08-19 01:54:57.941531640 +0200
@@ -1,7 +1,7 @@
 import unittest
 
 from urllib3.fields import guess_content_type, RequestField
-from urllib3.packages.six import u
+from six import u
 
 
 class TestRequestField(unittest.TestCase):
diff -Nur test/test_filepost.py test/test_filepost.py
--- test/test_filepost.py	2014-03-15 01:05:07.000000000 +0100
+++ test/test_filepost.py	2014-08-19 01:55:14.829531913 +0200
@@ -2,7 +2,7 @@
 
 from urllib3.filepost import encode_multipart_formdata, iter_fields
 from urllib3.fields import RequestField
-from urllib3.packages.six import b, u
+from six import b, u
 
 
 BOUNDARY = '!! test boundary !!'
diff -Nur test/test_retry.py test/test_retry.py
--- test/test_retry.py	2014-07-07 22:56:01.000000000 +0200
+++ test/test_retry.py	2014-08-19 01:56:07.295532762 +0200
@@ -1,6 +1,6 @@
 import unittest
 
-from urllib3.packages.six.moves import xrange
+from six.moves import xrange
 from urllib3.util.retry import Retry
 from urllib3.exceptions import (
     ConnectTimeoutError,
diff -Nur urllib3/_collections.py urllib3/_collections.py
--- urllib3/_collections.py	2014-07-07 22:56:01.000000000 +0200
+++ urllib3/_collections.py	2014-08-19 02:07:34.678543884 +0200
@@ -10,11 +10,8 @@
             pass
 
 
-try: # Python 2.7+
-    from collections import OrderedDict
-except ImportError:
-    from .packages.ordered_dict import OrderedDict
-from .packages.six import itervalues
+from collections import OrderedDict
+from six import itervalues
 
 
 __all__ = ['RecentlyUsedContainer', 'HTTPHeaderDict']
diff -Nur urllib3/connectionpool.py urllib3/connectionpool.py
--- urllib3/connectionpool.py	2014-07-07 22:56:01.000000000 +0200
+++ urllib3/connectionpool.py	2014-08-19 01:57:57.899534552 +0200
@@ -26,8 +26,11 @@
     TimeoutError,
     InsecureRequestWarning,
 )
-from .packages.ssl_match_hostname import CertificateError
-from .packages import six
+try:
+    from ssl import CertificateError
+except ImportError:
+    from backports.ssl_match_hostname import CertificateError
+import six
 from .connection import (
     port_by_scheme,
     DummyConnection,
diff -Nur urllib3/connection.py urllib3/connection.py
--- urllib3/connection.py	2014-07-07 22:56:01.000000000 +0200
+++ urllib3/connection.py	2014-08-19 01:59:00.332535562 +0200
@@ -27,8 +27,11 @@
 from .exceptions import (
     ConnectTimeoutError,
 )
-from .packages.ssl_match_hostname import match_hostname
-from .packages import six
+try:
+    from ssl import match_hostname
+except ImportError:
+    from backports.ssl_match_hostname import match_hostname
+import six
 
 from .util.ssl_ import (
     resolve_cert_reqs,
diff -Nur urllib3/fields.py urllib3/fields.py
--- urllib3/fields.py	2014-07-07 22:56:01.000000000 +0200
+++ urllib3/fields.py	2014-08-19 01:59:25.757535973 +0200
@@ -1,7 +1,7 @@
 import email.utils
 import mimetypes
 
-from .packages import six
+import six
 
 
 def guess_content_type(filename, default='application/octet-stream'):
diff -Nur urllib3/filepost.py urllib3/filepost.py
--- urllib3/filepost.py	2014-07-07 22:56:01.000000000 +0200
+++ urllib3/filepost.py	2014-08-19 02:00:05.271536613 +0200
@@ -3,8 +3,8 @@
 from uuid import uuid4
 from io import BytesIO
 
-from .packages import six
-from .packages.six import b
+import six
+from six import b
 from .fields import RequestField
 
 writer = codecs.lookup('utf-8')[3]
diff -Nur urllib3/response.py urllib3/response.py
--- urllib3/response.py	2014-07-07 22:56:01.000000000 +0200
+++ urllib3/response.py	2014-08-19 02:00:38.448537150 +0200
@@ -4,7 +4,7 @@
 
 from ._collections import HTTPHeaderDict
 from .exceptions import ProtocolError, DecodeError, ReadTimeoutError
-from .packages.six import string_types as basestring, binary_type
+from six import string_types as basestring, binary_type
 from .connection import HTTPException, BaseSSLError
 from .util.response import is_fp_closed
 
diff -Nur urllib3/util/request.py urllib3/util/request.py
--- urllib3/util/request.py	2014-07-07 22:56:01.000000000 +0200
+++ urllib3/util/request.py	2014-08-19 02:00:57.616537460 +0200
@@ -1,6 +1,6 @@
 from base64 import b64encode
 
-from ..packages.six import b
+from six import b
 
 ACCEPT_ENCODING = 'gzip,deflate'
 
diff -Nur urllib3/util/retry.py urllib3/util/retry.py
--- urllib3/util/retry.py	2014-07-07 22:56:01.000000000 +0200
+++ urllib3/util/retry.py	2014-08-19 02:01:18.385537796 +0200
@@ -7,7 +7,7 @@
     ReadTimeoutError,
     MaxRetryError,
 )
-from ..packages import six
+import six
 
 
 log = logging.getLogger(__name__)
