--- a/src/spy/pcsc-spy	2019-12-26 20:48:20.222069705 +0700
+++ b/src/spy/pcsc-spy	2019-12-26 20:48:41.407288561 +0700
@@ -1,4 +1,4 @@
-#! /usr/bin/python
+#!/usr/bin/python3
 
 """
 #    Display PC/SC functions arguments
@@ -22,12 +22,7 @@
 import os
 import signal
 import time
-try:
-    # for Python3
-    from queue import Queue
-except ImportError:
-    # for Python2
-    from Queue import Queue
+from queue import Queue
 from threading import Thread
 from operator import attrgetter
 
