--- a/utils/keytable/keytable.c	2019-09-22 11:22:54.000000000 +0200
+++ b/utils/keytable/keytable.c	2021-01-15 14:48:30.569549169 +0100
@@ -1533,7 +1533,7 @@
 
 		for (i = 0; i < rd / sizeof(struct input_event); i++) {
 			printf(_("%ld.%06ld: event type %s(0x%02x)"),
-				ev[i].time.tv_sec, ev[i].time.tv_usec,
+				ev[i].input_event_sec, ev[i].input_event_usec,
 				get_event_name(events_type, ev[i].type), ev[i].type);
 
 			switch (ev[i].type) {
