--- a/vapi/libmarkdown.vapi	2020-11-21 13:32:35.540125253 +0700
+++ b/vapi/libmarkdown.vapi	2020-11-21 13:40:04.961780640 +0700
@@ -41,7 +41,7 @@
 	public void initialize ();
 	public void with_html5_tags ();
 	public void shlib_destructor ();
-	public char markdown_version[];
+	public string markdown_version;
 
 	[Compact]
 	[CCode (cname = "MMIOT", cprefix = "mkd_", free_function = "mkd_cleanup")]
@@ -125,4 +125,4 @@
 		NOSTYLE,
 		EMBED
 	}
-}
\ No newline at end of file
+}
