Index: x-unikey-1.0.4/src/gui/gui.c
===================================================================
--- x-unikey-1.0.4.orig/src/gui/gui.c
+++ x-unikey-1.0.4/src/gui/gui.c
@@ -40,6 +40,10 @@
 #include "uksync.h"
 #include "guiopt.h"
 
+#ifndef WAIT_ANY
+#define WAIT_ANY -1
+#endif
+
 #define PROG_NAME "Unikey XIM Simple GUI"
 
 /* using global variables is a very bad way of communication. Must do something better!!! */
