--- swift/utils/build-presets.ini
+++ swift/utils/build-presets.ini
@@ -736,12 +736,12 @@ mixin-preset=
 llbuild
 swiftpm
 xctest
-libicu
+#libicu
 libcxx
 
 dash-dash
 
-build-ninja
+#build-ninja
 install-swift
 install-lldb
 install-llbuild
@@ -758,7 +758,7 @@ build-swift-stdlib-unittest-extra
 # Executes the lit tests for the installable package that is created
 # Assumes the swift-integration-tests repo is checked out
 
-test-installable-package
+#test-installable-package
 
 # Build the benchmarks against the toolchain.
 toolchain-benchmarks
