From 7947941ce54372e8dc65e4c731991460fea41746 Mon Sep 17 00:00:00 2001
From: Artem Bityutskiy <dedekind1@gmail.com>
Date: Fri, 6 Mar 2020 18:16:46 +0200
Subject: [PATCH 5/5] Update CLI.py

---
 bmaptools/CLI.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/bmaptools/CLI.py b/bmaptools/CLI.py
index 4f772db..49a3d3f 100644
--- a/bmaptools/CLI.py
+++ b/bmaptools/CLI.py
@@ -385,10 +385,10 @@ def open_files(args):
 
     if bmap_path == args.image:
         # Most probably the user specified the bmap file instead of the image
-        # file by mistake. Don't let bmaptool copy any data
+        # file by mistake.
         bmap_obj.close()
-        error_out("Make sure you are flashing your image and not the bmap file "
-                  "because they have the same path.")
+        error_out("Make sure you are writing your image and not the bmap file "
+                  "(you specified the same path for them)")
 
     # If the destination file is under "/dev", but does not exist, print a
     # warning. This is done in order to be more user-friendly, because
-- 
2.27.0

