--- ./lib_pypy/_cffi_ssl/_cffi_src/openssl/x509.py.orig 2019-02-17 04:27:29.778408202 +0100
+++ ./lib_pypy/_cffi_ssl/_cffi_src/openssl/x509.py      2019-02-17 04:29:05.433404279 +0100
@@ -258,7 +258,7 @@
 const X509_ALGOR *X509_get0_tbs_sigalg(const X509 *);
 
 /* in 1.1.0 becomes const ASN1_BIT_STRING, const X509_ALGOR */
-void X509_get0_signature(ASN1_BIT_STRING **, X509_ALGOR **, X509 *);
+void X509_get0_signature(const ASN1_BIT_STRING **, const X509_ALGOR **, const X509 *);
 
 long X509_get_version(X509 *);
 
@@ -343,7 +343,7 @@
    opaquing. */
 #if CRYPTOGRAPHY_OPENSSL_LESS_THAN_102 || defined(LIBRESSL_VERSION_NUMBER)
 /* from x509/x_x509.c version 1.0.2 */
-void X509_get0_signature(ASN1_BIT_STRING **psig, X509_ALGOR **palg,
+void X509_get0_signature(const ASN1_BIT_STRING **psig, const X509_ALGOR **palg,
                          const X509 *x)
 {
     if (psig)
--- ./lib_pypy/_cffi_ssl/_cffi_src/openssl/x509_vfy.py.orig     2019-02-17 04:31:05.820399342 +0100
+++ ./lib_pypy/_cffi_ssl/_cffi_src/openssl/x509_vfy.py  2019-02-17 04:31:27.573398450 +0100
@@ -204,7 +204,7 @@
 X509_OBJECT *sk_X509_OBJECT_value(Cryptography_STACK_OF_X509_OBJECT *, int);
 X509_VERIFY_PARAM *X509_STORE_get0_param(X509_STORE *);
 Cryptography_STACK_OF_X509_OBJECT *X509_STORE_get0_objects(X509_STORE *);
-X509 *X509_OBJECT_get0_X509(X509_OBJECT *);
+X509 *X509_OBJECT_get0_X509(const X509_OBJECT *);
 int X509_OBJECT_get_type(const X509_OBJECT *);
 """
 
@@ -293,7 +293,7 @@
 #endif
 
 #if CRYPTOGRAPHY_OPENSSL_LESS_THAN_110PRE5 || defined(LIBRESSL_VERSION_NUMBER)
-X509 *X509_OBJECT_get0_X509(X509_OBJECT *x) {
+X509 *X509_OBJECT_get0_X509(const X509_OBJECT *x) {
     return x->data.x509;
 }
 #endif
