--- /dev/null
+++ setup.py
@@ -0,0 +1,20 @@
+from setuptools import setup
+
+def get_version():
+    with open('entrypoints.py') as f:
+        for line in f:
+            if line.startswith('__version__'):
+                return eval(line.split('=')[-1])
+
+setup(
+    name='entrypoints',
+    version=get_version(),
+    description='Discover and load entry points from installed packages',
+    author='Thomas Kluyver',
+    author_email='thomas@kluyver.me.uk',
+    url='https://github.com/takluyver/entrypoints',
+    license='MIT',
+    extras_require={
+        ':python_version == "2.7"': ['configparser'],
+    },
+)
