diff --git a/src/utils/cpu-probe.h b/src/utils/cpu-probe.h
index df7af7c0..e0564998 100644
--- a/src/utils/cpu-probe.h
+++ b/src/utils/cpu-probe.h
@@ -2,37 +2,6 @@
 
 #include <glib.h>
 
-typedef struct _ProtonPlusCpuFeatureProbe {
-    gboolean available;
-    gboolean cmpxchg16b;
-    gboolean lahf_sahf;
-    gboolean popcnt;
-    gboolean sse3;
-    gboolean sse4_1;
-    gboolean sse4_2;
-    gboolean ssse3;
-    gboolean avx;
-    gboolean avx2;
-    gboolean bmi1;
-    gboolean bmi2;
-    gboolean f16c;
-    gboolean fma;
-    gboolean lzcnt;
-    gboolean movbe;
-    gboolean osxsave;
-    gboolean xcr0_xmm;
-    gboolean xcr0_ymm;
-    gboolean avx512f;
-    gboolean avx512bw;
-    gboolean avx512cd;
-    gboolean avx512dq;
-    gboolean avx512vl;
-    gboolean xcr0_opmask;
-    gboolean xcr0_zmm_hi256;
-    gboolean xcr0_hi16_zmm;
-    gboolean aarch64_crc32;
-    gboolean aarch64_lse_atomics;
-    gboolean aarch64_rdma;
-} ProtonPlusCpuFeatureProbe;
+typedef struct _ProtonPlusCpuFeatureProbe ProtonPlusCpuFeatureProbe;
 
 void protonplus_cpu_get_feature_probe (ProtonPlusCpuFeatureProbe *out_probe);
diff --git a/src/utils/cpu-probe.c b/src/utils/cpu-probe.c
index 48ab7fa2..b3b5f872 100644
--- a/src/utils/cpu-probe.c
+++ b/src/utils/cpu-probe.c
@@ -2,6 +2,39 @@
 
 #include <stdint.h>
 
+struct _ProtonPlusCpuFeatureProbe {
+    gboolean available;
+    gboolean cmpxchg16b;
+    gboolean lahf_sahf;
+    gboolean popcnt;
+    gboolean sse3;
+    gboolean sse4_1;
+    gboolean sse4_2;
+    gboolean ssse3;
+    gboolean avx;
+    gboolean avx2;
+    gboolean bmi1;
+    gboolean bmi2;
+    gboolean f16c;
+    gboolean fma;
+    gboolean lzcnt;
+    gboolean movbe;
+    gboolean osxsave;
+    gboolean xcr0_xmm;
+    gboolean xcr0_ymm;
+    gboolean avx512f;
+    gboolean avx512bw;
+    gboolean avx512cd;
+    gboolean avx512dq;
+    gboolean avx512vl;
+    gboolean xcr0_opmask;
+    gboolean xcr0_zmm_hi256;
+    gboolean xcr0_hi16_zmm;
+    gboolean aarch64_crc32;
+    gboolean aarch64_lse_atomics;
+    gboolean aarch64_rdma;
+};
+
 #if defined(__aarch64__) && defined(__linux__)
 #include <asm/hwcap.h>
 #include <sys/auxv.h>
