--- a/etc/clamav-milter.conf.sample	2023-09-23 21:48:44.563130754 +0900
+++ b/etc/clamav-milter.conf.sample	2023-09-23 21:52:44.693134403 +0900
@@ -38,7 +38,7 @@
 # to work)
 #
 # Default: unset (don't drop privileges)
-#User clamav
+#User _clamav
 
 # Waiting for data from clamd will timeout after this time (seconds).
 # Value of 0 disables the timeout.
--- a/etc/clamd.conf.sample	2023-09-23 21:20:27.717104965 +0900
+++ b/etc/clamd.conf.sample	2023-09-23 21:57:28.716138720 +0900
@@ -81,7 +81,7 @@
 #TemporaryDirectory /var/tmp
 
 # Path to the database directory.
-# Default: hardcoded (depends on installation options)
+# Default: /var/lib/clamav
 #DatabaseDirectory /var/lib/clamav
 
 # Only load the official signatures published by the ClamAV project.
@@ -224,7 +224,7 @@
 
 # Run as another user (clamd must be started by root for this option to work)
 # Default: don't drop privileges
-#User clamav
+#User _clamav
 
 # Stop daemon when libclamav reports out of memory condition.
 #ExitOnOOM yes
@@ -706,7 +706,7 @@
 # Modifies fanotify blocking behaviour when handling permission events.
 # If off, fanotify will only notify if the file scanned is a virus,
 # and not perform any blocking.
-# Default: no
+# Default: no (Void Linux does not support this option; setting this to 'yes' will do nothing)
 #OnAccessPrevention yes
 
 # When using prevention, if this option is turned on, any errors that occur
@@ -766,7 +766,7 @@
 # It has the same potential race condition limitations of the
 # OnAccessExcludeUID option.
 # Default: disabled
-#OnAccessExcludeUname clamav
+#OnAccessExcludeUname _clamav
 
 # Number of times the OnAccess client will retry a failed scan due to
 # connection problems (or other issues).
--- a/etc/freshclam.conf.sample	2023-09-23 21:52:51.220134502 +0900
+++ b/etc/freshclam.conf.sample	2023-09-23 21:57:12.082138467 +0900
@@ -9,7 +9,7 @@
 
 # Path to the database directory.
 # WARNING: It must match clamd.conf's directive!
-# Default: hardcoded (depends on installation options)
+# Default: /var/lib/clamav
 #DatabaseDirectory /var/lib/clamav
 
 # Path to the log file (make sure it has proper permissions)
@@ -56,8 +56,8 @@
 
 # By default when started freshclam drops privileges and switches to the
 # "clamav" user. This directive allows you to change the database owner.
-# Default: clamav (may depend on installation options)
-#DatabaseOwner clamav
+# Default: _clamav
+#DatabaseOwner _clamav
 
 # Use DNS to verify virus database version. FreshClam uses DNS TXT records
 # to verify database and software versions. With this directive you can change
