--- libclamav/readdb.c	2015-11-24 00:13:46.000000000 +0100
+++ libclamav/readdb.c	2015-12-02 10:42:08.374736882 +0100
@@ -4163,7 +4163,9 @@
 
         /* use the tdb to track filetypes and check flevels */
         memset(&tdb, 0, sizeof(tdb));
+#ifdef USE_MPOOL
         tdb.mempool = engine->mempool;
+#endif
         ret = init_tdb(&tdb, engine, attribs, passname);
         free(attribs);
         if(ret != CL_SUCCESS) {
