--- nss/lib/freebl/Makefile.orig	2018-05-04 18:40:51.000000000 +0200
+++ nss/lib/freebl/Makefile	2018-05-14 18:14:14.316003320 +0200
@@ -535,11 +535,12 @@
     # All intel architectures get the 64 bit version
     # With custom uint128 if necessary (faster than generic 32 bit version).
     ECL_SRCS += curve25519_64.c
-    VERIFIED_SRCS += Hacl_Curve25519.c FStar.c
+    VERIFIED_SRCS += Hacl_Curve25519.c
 else
     # All non intel architectures get the generic 32 bit implementation (slow!)
     ECL_SRCS += curve25519_32.c
 endif
+VERIFIED_SRCS += FStar.c
 
 #######################################################################
 # (5) Execute "global" rules. (OPTIONAL)                              #