--- a/tools/tracer/tracer/tracer_linux.cpp	2022-11-18 03:02:59.000000000 -0500
+++ b/tools/tracer/tracer/tracer_linux.cpp	2022-12-18 01:44:51.341542572 -0500
@@ -32,6 +32,11 @@
 #include <dlfcn.h>
 #include "tracer.h"

+#ifndef RTLD_DEEPBIND
+// musl neither defines nor can use RTLD_DEEPBIND
+#define RTLD_DEEPBIND 0
+#endif
+
 #if defined(__i386__)
     #define LIBMFXHW "libmfxhw32.so.1"
 #elif defined(__x86_64__)
