diff --git a/build/npm/preinstall.js b/build/npm/preinstall.js
index 0c5d36eab68e..8b137891791f 100644
--- a/build/npm/preinstall.js
+++ b/build/npm/preinstall.js
@@ -1,23 +1 @@
-/*---------------------------------------------------------------------------------------------
- *  Copyright (c) Microsoft Corporation. All rights reserved.
- *  Licensed under the MIT License. See License.txt in the project root for license information.
- *--------------------------------------------------------------------------------------------*/

-let err = false;
-
-const major = parseInt(/^(\d+)\./.exec(process.versions.node)[1]);
-
-if (major < 8) {
-	console.error('\033[1;31m*** Please use node>=8.\033[0;0m');
-	err = true;
-}
-
-if (!/yarn\.js$|yarnpkg$/.test(process.env['npm_execpath'])) {
-	console.error('\033[1;31m*** Please use yarn to install dependencies.\033[0;0m');
-	err = true;
-}
-
-if (err) {
-	console.error('');
-	process.exit(1);
-}
\ No newline at end of file
