--- source/modules/water/memory/ByteOrder.h	2018-01-22 21:05:35.000000000 +0000
+++ -	2018-06-18 09:58:25.563913205 +0000
@@ -153,7 +153,7 @@
 {
    #ifdef CARLA_OS_MAC
     return OSSwapInt32 (n);
-   #elif defined(CARLA_OS_WIN) || ! (defined (__arm__) || defined (__arm64__))
+   #elif defined(CARLA_OS_WIN) || ! (defined (__arm__) || ! defined (__arm64__))
     asm("bswap %%eax" : "=a"(n) : "a"(n));
     return n;
    #else
