--- source/Makefile.mk
+++ source/Makefile.mk
@@ -355,13 +355,7 @@ endif
 
 ifneq ($(WIN32),true)
 
-ifeq ($(shell $(PKG_CONFIG) --exists libmagic && echo true),true)
 HAVE_LIBMAGIC = true
-else
-# old libmagic versions don't have a pkg-config file, so we need to call the compiler to test it
-CFLAGS_WITHOUT_ARCH = $(subst -arch arm64,,$(CFLAGS))
-HAVE_LIBMAGIC = $(shell echo '\#include <magic.h>' | $(CC) $(CFLAGS_WITHOUT_ARCH) -x c -w -c - -o /dev/null 2>/dev/null && echo true)
-endif
 
 endif
 
