--- a/source/Makefile.mk
+++ b/source/Makefile.mk
@@ -233,10 +233,7 @@
 # Check for optional libs (special non-pkgconfig tests)
 
 ifneq ($(WIN32),true)
-
-# libmagic doesn't have a pkg-config file, so we need to call the compiler to test it
-HAVE_LIBMAGIC = $(shell echo '\#include <magic.h>' | $(CC) $(CFLAGS) -x c -w -c - -o .libmagic-tmp 2>/dev/null && echo true)
-
+HAVE_LIBMAGIC = true
 endif
 
 # ---------------------------------------------------------------------------------------------------------------------
