--- source/modules/hylia/link/asio/impl/error_code.ipp	2018-01-22 21:05:35.000000000 +0000
+++ -	2018-06-18 09:13:32.145486974 +0000
@@ -101,7 +101,7 @@
 #elif defined(__MACH__) && defined(__APPLE__) \
   || defined(__NetBSD__) || defined(__FreeBSD__) || defined(__OpenBSD__) \
   || defined(_AIX) || defined(__hpux) || defined(__osf__) \
-  || defined(__ANDROID__)
+  || defined(__ANDROID__) || !defined(__GLIBC__)
     char buf[256] = "";
     using namespace std;
     strerror_r(value, buf, sizeof(buf));

