diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 269c01d..0bc471c 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -34,4 +34,4 @@ install(TARGETS btrfs-assistant-bin RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
 
 find_library(BTRFSUTIL_LIB btrfsutil)
 target_link_libraries(btrfs-assistant-bin PRIVATE Qt${QT_VERSION_MAJOR}::Widgets ${BTRFSUTIL_LIB})
-target_compile_options(btrfs-assistant-bin PRIVATE -Werror -Wall -Wextra -Wconversion)
+target_compile_options(btrfs-assistant-bin PRIVATE -Wall -Wextra -Wconversion)

