diff '--color=auto' -c4 orig/utils.py patch/utils.py
*** orig/radicale/utils.py	2025-09-24 13:11:26.299249118 +0200
--- patch/radicale/utils.py	2025-09-24 13:19:35.689249351 +0200
***************
*** 279,290 ****
      if sys.platform == "win32":
          # TODO check how to support this better
          return str(unixtime)
      if unixtime < DATETIME_MAX_UNIXTIME:
!         if sys.version_info < (3, 11):
!             dt = datetime.datetime.utcfromtimestamp(unixtime)
!         else:
!             dt = datetime.datetime.fromtimestamp(unixtime, datetime.UTC)
          r = str(unixtime) + "(" + dt.strftime('%Y-%m-%dT%H:%M:%SZ') + ")"
      else:
          r = str(unixtime) + "(>MAX:" + str(DATETIME_MAX_UNIXTIME) + ")"
      return r
--- 279,287 ----
      if sys.platform == "win32":
          # TODO check how to support this better
          return str(unixtime)
      if unixtime < DATETIME_MAX_UNIXTIME:
!         dt = datetime.datetime(1970, 1, 1, tzinfo=datetime.timezone.utc) + datetime.timedelta(seconds=unixtime)
          r = str(unixtime) + "(" + dt.strftime('%Y-%m-%dT%H:%M:%SZ') + ")"
      else:
          r = str(unixtime) + "(>MAX:" + str(DATETIME_MAX_UNIXTIME) + ")"
      return r
