--- a/radicale/utils.py	2026-01-10 00:52:03.000000000 -0600
+++ b/radicale/utils.py	2026-01-10 13:31:34.039736783 -0600
@@ -333,16 +333,11 @@
     if sys.platform == "win32":
         # TODO check how to support this better
         return str(unixtime)
-    if unixtime <= DATETIME_MIN_UNIXTIME:
-        r = str(unixtime) + "(<=MIN:" + str(DATETIME_MIN_UNIXTIME) + ")"
-    elif unixtime >= DATETIME_MAX_UNIXTIME:
-        r = str(unixtime) + "(>=MAX:" + str(DATETIME_MAX_UNIXTIME) + ")"
-    else:
-        if sys.version_info < (3, 11):
-            dt = datetime.datetime.utcfromtimestamp(unixtime)
-        else:
-            dt = datetime.datetime.fromtimestamp(unixtime, datetime.UTC)
+    if unixtime < DATETIME_MIN_UNIXTIME:
+        dt = datetime.datetime(1970, 1, 1, tzinfo=datetime.timezone.utc) + datetime.timedelta(seconds=unixtime)
         r = str(unixtime) + "(" + dt.strftime('%Y-%m-%dT%H:%M:%SZ') + ")"
+    else:
+        r = str(unixtime) + "(>MAX:" + str(DATETIME_MAX_UNIXTIME) + ")"
     return r
 
 
