--- astyle/src/astyle_main.cpp	2017-05-30 20:26:37.000000000 -0400
+++ astyle/src/astyle_main.cpp	2019-09-11 20:38:02.155936887 -0400
@@ -3749,7 +3749,7 @@
 // called by a java program to get the version number
 // the function name is constructed from method names in the calling java program
 extern "C"  EXPORT
-jstring STDCALL Java_AStyleInterface_AStyleGetVersion(JNIEnv* env, jclass)
+jstring STDCALL Java_cc_arduino_packages_formatter_AStyleInterface_AStyleGetVersion(JNIEnv* env, jclass)
 {
 	return env->NewStringUTF(g_version);
 }
@@ -3757,7 +3757,7 @@
 // called by a java program to format the source code
 // the function name is constructed from method names in the calling java program
 extern "C"  EXPORT
-jstring STDCALL Java_AStyleInterface_AStyleMain(JNIEnv* env,
+jstring STDCALL Java_cc_arduino_packages_formatter_AStyleInterface_AStyleMain(JNIEnv* env,
                                                 jobject obj,
                                                 jstring textInJava,
                                                 jstring optionsJava)
--- astyle/src/astyle_main.h	2017-05-30 20:26:37.000000000 -0400
+++ astyle/src/astyle_main.h	2019-09-11 20:39:15.953536523 -0400
@@ -406,9 +406,9 @@
 char* STDCALL javaMemoryAlloc(unsigned long memoryNeeded);
 // the following function names are constructed from method names in the calling java program
 extern "C" EXPORT
-jstring STDCALL Java_AStyleInterface_AStyleGetVersion(JNIEnv* env, jclass);
+jstring STDCALL Java_cc_arduino_packages_formatter_AStyleInterface_AStyleGetVersion(JNIEnv* env, jclass);
 extern "C" EXPORT
-jstring STDCALL Java_AStyleInterface_AStyleMain(JNIEnv* env,
+jstring STDCALL Java_cc_arduino_packages_formatter_AStyleInterface_AStyleMain(JNIEnv* env,
                                                 jobject obj,
                                                 jstring textInJava,
                                                 jstring optionsJava);
