--- a/lib/dpkg/t/t-file.c
+++ b/lib/dpkg/t/t-file.c
@@ -67,7 +67,11 @@
 	free(path);
 
 	path = file_realpath("//./..///../././..///..");
+#ifdef __GLIBC__
 	test_str(path, ==, "/");
+#else
+	test_str(path, ==, "//");
+#endif
 	free(path);
 }
 
@@ -86,7 +90,11 @@
 	free(path);
 
 	path = file_canonicalize("//./..///../././..///..");
+#ifdef __GLIBC__
 	test_str(path, ==, "/");
+#else
+	test_str(path, ==, "//");
+#endif
 	free(path);
 
 	/* Canonicalize filenames that do not exist. */
