--- a/effects/internal/texteffect.cpp	2020-08-31 20:39:58.048701631 +0200
+++ b/effects/internal/texteffect.cpp	2019-11-11 07:05:02.000000000 +0100
@@ -25,6 +25,7 @@
 #include <QOpenGLTexture>
 #include <QTextEdit>
 #include <QPainter>
+#include <QPainterPath>
 #include <QPushButton>
 #include <QColorDialog>
 #include <QFontDatabase>
--- a/effects/internal/timecodeeffect.cpp	2020-08-31 20:39:49.809701206 +0200
+++ b/effects/internal/timecodeeffect.cpp	2019-11-11 07:05:02.000000000 +0100
@@ -25,6 +25,7 @@
 #include <QOpenGLTexture>
 #include <QTextEdit>
 #include <QPainter>
+#include <QPainterPath>
 #include <QPushButton>
 #include <QColorDialog>
 #include <QFontDatabase>
--- a/ui/graphview.cpp	2020-08-31 20:39:43.016700855 +0200
+++ b/ui/graphview.cpp	2019-11-11 07:05:02.000000000 +0100
@@ -21,6 +21,7 @@
 #include "graphview.h"
 
 #include <QPainter>
+#include <QPainterPath>
 #include <QMouseEvent>
 #include <QtMath>
 #include <cfloat>
--- a/ui/timelineheader.cpp	2020-08-31 20:39:36.992700545 +0200
+++ b/ui/timelineheader.cpp	2019-11-11 07:05:02.000000000 +0100
@@ -21,6 +21,7 @@
 #include "timelineheader.h"
 
 #include <QPainter>
+#include <QPainterPath>
 #include <QMouseEvent>
 #include <QScrollBar>
 #include <QtMath>
