--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -85,7 +85,7 @@
 
 set(QT_MAJOR Qt6)
 
-set(QT_COMPONENTS Qml Quick Gui Multimedia Core Svg Network LinguistTools Sensors QuickControls2 QuickTemplates2 Charts Widgets QmlWorkerScript)
+set(QT_COMPONENTS Qml Quick Gui Multimedia CorePrivate Svg Network LinguistTools Sensors QuickControls2BasicPrivate QuickTemplates2 Charts Widgets QmlWorkerScript)
 
 find_package(${QT_MAJOR} ${QT_REQUIRED_VERSION} COMPONENTS
     ${QT_COMPONENTS})
@@ -114,7 +114,7 @@
 )
 
 if(UNIX AND NOT APPLE AND NOT ANDROID)
-  find_package(${QT_MAJOR} ${QT_REQUIRED_VERSION} OPTIONAL_COMPONENTS WaylandClient)
+  find_package(${QT_MAJOR} ${QT_REQUIRED_VERSION} OPTIONAL_COMPONENTS WaylandClientPrivate)
 endif()
 
 # For systems where ICU is not packaged with Qt installer

