--- lib/mega.c.orig	2017-09-27 13:56:24.880875673 +0200
+++ lib/mega.c	2017-09-27 13:56:28.285869611 +0200
@@ -803,7 +803,7 @@ static gboolean rsa_key_gen(rsa_key* k)
     return FALSE;
   }
 
-#if OPENSSL_VERSION_NUMBER >= 0x10100000L
+#if OPENSSL_VERSION_NUMBER >= 0x10100000L && !defined(LIBRESSL_VERSION_NUMBER)
   RSA_get0_key(key, &k->m, &k->e, &k->d);
   RSA_get0_factors(key, &k->q, &k->p);
   RSA_get0_crt_params(key, NULL, NULL, &k->u);
