From 31f7ded7687c443b8e4570a9364340aace022dea Mon Sep 17 00:00:00 2001
From: tux3 <barrdetwix@gmail.com>
Date: Fri, 5 Jun 2015 03:07:42 +0200
Subject: [PATCH] Fix #1794

---
 src/video/videomode.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git src/video/videomode.h src/video/videomode.h
index ec2ffb9..0cbe52f 100644
--- src/video/videomode.h
+++ src/video/videomode.h
@@ -8,12 +8,12 @@ struct VideoMode
     unsigned short FPS; ///< Max frames per second supported by the device at this resolution

     /// All zeros means a default/unspecified mode
-    operator bool()
+    operator bool() const
     {
         return width || height || FPS;
     }

-    bool operator==(const VideoMode& other)
+    bool operator==(const VideoMode& other) const
     {
         return width == other.width
                 && height == other.height
