diff --git a/internal/txlib/update.go b/internal/txlib/update.go
index 98b6f6d..8b397a9 100644
--- a/internal/txlib/update.go
+++ b/internal/txlib/update.go
@@ -18,6 +18,9 @@ type UpdateCommandArguments struct {
 }
 
 func UpdateCommand(arguments UpdateCommandArguments) error {
+	fmt.Println("This version cannot perform self-updates")
+	return nil
+
 	if arguments.Debug {
 		selfupdate.EnableLog()
 	}
