--- a/setup.py	2021-10-06 20:25:26.847727634 -0400
+++ b/setup.py	2021-10-06 20:25:54.167862198 -0400
@@ -18,11 +18,6 @@
     'cffi',
 ]
 
-setup_requirements = [
-    'cffi',
-    'pytest-runner',
-]
-
 test_requirements = [
     'pytest',
 ]
@@ -60,6 +55,5 @@
         ],
         test_suite='tests',
         tests_require=test_requirements,
-        setup_requires=setup_requirements,
         cffi_modules=["reflink/native.py:ffibuilder"]
     )
