configure several paths to sane/expected locations

diff --git a/.env.template b/.env.template
index 66a0434..cebac33 100644
--- a/.env.template
+++ b/.env.template
@@ -10,7 +10,7 @@
 ## vaultwarden.
 
 ## Main data folder
-# DATA_FOLDER=data
+DATA_FOLDER=/var/lib/vaultwarden
 
 ## Database URL
 ## When using SQLite, this is the path to the DB file, default to %DATA_FOLDER%/db.sqlite3
@@ -61,8 +61,8 @@
 # ICON_CACHE_NEGTTL=259200
 
 ## Web vault settings
-# WEB_VAULT_FOLDER=web-vault/
-# WEB_VAULT_ENABLED=true
+# WEB_VAULT_FOLDER=/usr/share/webapps/vaultwarden-web
+WEB_VAULT_ENABLED=false
 
 ## Enables websocket notifications
 # WEBSOCKET_ENABLED=false
@@ -117,7 +117,7 @@
 # LOG_TIMESTAMP_FORMAT="%Y-%m-%d %H:%M:%S.%3f"
 
 ## Logging to file
-# LOG_FILE=/path/to/log
+# LOG_FILE=/var/log/vaultwarden.log
 
 ## Logging to Syslog
 ## This requires extended logging
