diff --git a/.env.template b/.env.template
index 6d27238..c0731ce 100644
--- a/.env.template
+++ b/.env.template
@@ -23,7 +23,7 @@
 ## When using an external location, make sure to set TMP_FOLDER,
 ## TEMPLATES_FOLDER, and DATABASE_URL to local paths and/or a remote database
 ## location.
-# DATA_FOLDER=data
+DATA_FOLDER=/var/lib/vaultwarden
 
 ## Individual folders, these override %DATA_FOLDER%
 # RSA_KEY_FILENAME=data/rsa_key
@@ -42,8 +42,8 @@
 # RELOAD_TEMPLATES=false
 
 ## Web vault settings
-# WEB_VAULT_FOLDER=web-vault/
-# WEB_VAULT_ENABLED=true
+# WEB_VAULT_FOLDER=/usr/share/webapps/vaultwarden-web
+WEB_VAULT_ENABLED=false
 
 #########################
 ### Database settings ###
@@ -386,7 +386,7 @@
 # USE_SYSLOG=false
 
 ## Logging to file
-# LOG_FILE=/path/to/log
+# LOG_FILE=/var/log/vaultwarden.log
 
 ## Log level
 ## Change the verbosity of the log output
