--- a/src/pty.c
+++ b/src/pty.c
@@ -46,6 +46,14 @@
 
 #define KMSCON_NREAD 16384
 
+/* ever since glibc 2.26 SIGUNUSED is no longer defined on any architerture
+ * http://man7.org/linux/man-pages/man7/signal.7.html
+ * according to signal(7) SIGUNUSED is synonymous with SIGSYS, musl defines it as such
+ */
+#ifdef __GLIBC__
+#define SIGUNUSED SIGSYS
+#endif
+
 struct kmscon_pty {
 	unsigned long ref;
 	struct ev_eloop *eloop;
