--- ext/faac/gstfaac.c	2017-05-15 17:46:30.000000000 +0200
+++ ext/faac/gstfaac.c	2017-11-06 11:40:30.458137107 +0100
@@ -490,7 +490,9 @@
   conf = faacEncGetCurrentConfiguration (faac->handle);
   conf->mpegVersion = (faac->mpegversion == 4) ? MPEG4 : MPEG2;
   conf->aacObjectType = faac->profile;
+#if FAAC_CFG_VERSION < 105
   conf->allowMidside = faac->midside;
+#endif
   conf->useLfe = 0;
   conf->useTns = faac->tns;
 
