--- a/ext/dtls/gstdtlsagent.c	2018-03-23 21:45:15.000000000 +0100
+++ b/ext/dtls/gstdtlsagent.c	2018-03-23 21:45:15.000000000 +0100
@@ -176,7 +176,7 @@
 
   ERR_clear_error ();
 
-#if OPENSSL_VERSION_NUMBER >= 0x1000200fL
+#if OPENSSL_VERSION_NUMBER >= 0x1000200fL && !defined(LIBRESSL_VERSION_NUMBER)
   priv->ssl_context = SSL_CTX_new (DTLS_method ());
 #else
   priv->ssl_context = SSL_CTX_new (DTLSv1_method ());
