diff --git a/tests/Makefile b/tests/Makefile
index 760294b..9b3f365 100644
--- a/tests/Makefile
+++ b/tests/Makefile
@@ -10,14 +10,17 @@ NUMS_CMA=nums.cma
 NUMS_CMXA=nums.cmxa
 endif
 
+ifeq ($(HASOCAMLOPT),yes)
 test:: zq.exe
 	@echo "Testing zq (native)..."
 	@if ./zq.exe | cmp -s zq.output$(WORDSIZE) - ; then echo "zq: passed"; else echo "zq: FAILED"; exit 2; fi
+endif
 
 test:: zq.byt
 	@echo "Testing zq (bytecode)..."
 	@if ocamlrun -I .. ./zq.byt | cmp -s zq.output$(WORDSIZE) - ; then echo "zq: passed"; else echo "zq: FAILED"; exit 2; fi
 
+ifeq ($(HASOCAMLOPT),yes)
 ifeq ($(HAS_NUM),true)
 test:: bi.exe
 	@echo "Testing bi..."
@@ -49,6 +52,7 @@ bench:: pi.exe
 test:: tst_extract.exe
 	@echo "Testing extract..."
 	@if ./tst_extract.exe; then echo "tst_extract: passed"; else echo "tst_extract: FAILED"; exit 2; fi
+endif
 
 tofloat.exe: tofloat.ml setround.o ../zarith.cmxa
 	ocamlopt -I .. -ccopt "-L.." zarith.cmxa -o tofloat.exe \
