# See https://github.com/lierdakil/pandoc-crossref/issues/342#issuecomment-1073256586

--- a/pandoc-crossref-0.3.17.0/pandoc-crossref.cabal	2001-09-08 22:46:40.000000000 -0300
+++ b/pandoc-crossref-0.3.17.0/pandoc-crossref.cabal	2024-03-26 17:16:02.918173738 -0300
@@ -127,20 +127,6 @@
 library
   exposed-modules:
       Text.Pandoc.CrossRef
-  hs-source-dirs:
-      lib
-  ghc-options: -Wall
-  build-depends:
-      base >=4.11 && <5
-    , mtl >=1.1 && <2.4
-    , pandoc >=3.1.8 && <3.2
-    , pandoc-crossref-internal
-    , pandoc-types ==1.23.*
-    , text >=1.2.2 && <2.1
-  default-language: Haskell2010
-
-library pandoc-crossref-internal
-  exposed-modules:
       Text.Numeral.Roman
       Text.Pandoc.CrossRef.References
       Text.Pandoc.CrossRef.References.Blocks
@@ -165,7 +151,9 @@
       Text.Pandoc.CrossRef.Util.Template
       Text.Pandoc.CrossRef.Util.Util
   hs-source-dirs:
+      lib
       lib-internal
+  ghc-options: -Wall
   build-depends:
       base >=4.11 && <5
     , containers >=0.1 && <0.7
@@ -247,7 +235,6 @@
     , mtl >=1.1 && <2.4
     , pandoc >=3.1.8 && <3.2
     , pandoc-crossref
-    , pandoc-crossref-internal
     , pandoc-types ==1.23.*
     , text >=1.2.2 && <2.1
   if flag(enable_flaky_tests)
