# See https://github.com/lierdakil/pandoc-crossref/issues/342#issuecomment-1073256586

--- a/pandoc-crossref-0.3.18.1/pandoc-crossref.cabal	2001-09-09 02:46:40.000000000 +0100
+++ a/pandoc-crossref-0.3.18.1/pandoc-crossref.cabal	2025-07-03 17:23:13.941135615 +0100
@@ -133,20 +133,6 @@
 library
   exposed-modules:
       Text.Pandoc.CrossRef
-  hs-source-dirs:
-      lib
-  ghc-options: -Wall
-  build-depends:
-      base >=4.11 && <5
-    , mtl >=1.1 && <2.4
-    , pandoc >=3.1.8 && <3.7
-    , pandoc-crossref-internal
-    , pandoc-types ==1.23.*
-    , text >=1.2.2 && <2.2
-  default-language: Haskell2010
-
-library pandoc-crossref-internal
-  exposed-modules:
       Text.Numeral.Roman
       Text.Pandoc.CrossRef.References
       Text.Pandoc.CrossRef.References.Blocks
@@ -171,6 +157,7 @@
       Text.Pandoc.CrossRef.Util.Template
       Text.Pandoc.CrossRef.Util.Util
   hs-source-dirs:
+      lib
       lib-internal
   build-depends:
       base >=4.11 && <5
@@ -253,7 +240,6 @@
     , mtl >=1.1 && <2.4
     , pandoc >=3.1.8 && <3.7
     , pandoc-crossref
-    , pandoc-crossref-internal
     , pandoc-types ==1.23.*
     , text >=1.2.2 && <2.2
   default-language: Haskell2010
