From 45e8b302bce1d318f310ea13599d7ce84acc477e Mon Sep 17 00:00:00 2001
From: Carl Schwan <carl@carlschwan.eu>
Date: Sun, 5 Oct 2025 14:21:49 +0200
Subject: [PATCH] Bump poppler to 25.02.0

diff --git a/filters/karbon/pdf/SvgOutputDev.cpp b/filters/karbon/pdf/SvgOutputDev.cpp
index 1e553ab391e..436243da7bd 100644
--- a/filters/karbon/pdf/SvgOutputDev.cpp
+++ b/filters/karbon/pdf/SvgOutputDev.cpp
@@ -386,23 +386,29 @@ void SvgOutputDev::drawString(GfxState *state, const GooString *s)
 {
     int render = state->getRender();
     // check for invisible text -- this is used by Acrobat Capture
-    if (render == 3)
+    if (render == 3) {
         return;
+    }
 
     // ignore empty strings
-    if (s->getLength() == 0)
+#if POPPLER_VERSION_MACRO < QT_VERSION_CHECK(25, 10, 0)
+    if (s->getLength() == 0) {
+#else
+    if (s->size() == 0) {
+#endif
         return;
+    }
 
-#if POPPLER_VERSION_MACRO < QT_VERSION_CHECK(22, 04, 0)
-    GfxFont *font = state->getFont();
-#else
     std::shared_ptr<GfxFont> font = state->getFont();
-#endif
 
     QString str;
 
     const char *p = s->c_str();
+#if POPPLER_VERSION_MACRO < QT_VERSION_CHECK(25, 10, 0)
     int len = s->getLength();
+#else
+    int len = s->size();
+#endif
     CharCode code;
     const Unicode *u = nullptr;
     int uLen;
-- 
GitLab

