diff --git a/filters/karbon/pdf/SvgOutputDev.cpp b/filters/karbon/pdf/SvgOutputDev.cpp
index 80f01a5f76f..ea69228dbc1 100644
--- filters/karbon/pdf/SvgOutputDev.cpp
+++ filters/karbon/pdf/SvgOutputDev.cpp
@@ -407,7 +407,7 @@ void SvgOutputDev::drawString(GfxState * state, const GooString * s)
 
     QString str;
 
-    const char * p = s->getCString();
+    const char * p = s->c_str();
     int len = s->getLength();
     CharCode code;
     Unicode *u = nullptr;
@@ -459,11 +459,11 @@ void SvgOutputDev::drawString(GfxState * state, const GooString * s)
     *d->body << " y=\"" << y << "px\"";
 
     if (font && font->getFamily()) {
-        *d->body << " font-family=\"" << QString::fromLatin1(font->getFamily()->getCString()) << "\"";
-        //debugPdf << "font family:" << QString::fromLatin1( font->getFamily()->getCString() );
+        *d->body << " font-family=\"" << QString::fromLatin1(font->getFamily()->c_str()) << "\"";
+        //debugPdf << "font family:" << QString::fromLatin1( font->getFamily()->c_str() );
     } else if (font && font->getName()) {
-        *d->body << " font-family=\"" << QString::fromLatin1(font->getName()->getCString()) << "\"";
-        //debugPdf << "font name:" << QString::fromLatin1( font->getName()->getCString() );
+        *d->body << " font-family=\"" << QString::fromLatin1(font->getName()->c_str()) << "\"";
+        //debugPdf << "font name:" << QString::fromLatin1( font->getName()->c_str() );
     }
     *d->body << " font-size=\"" << qMax(state->getFontSize(), state->getTransformedFontSize()) << "px\"";
 

