From 0c5430697bdcf41a45046107b28014e40c49a11a Mon Sep 17 00:00:00 2001
From: David Faure <faure@kde.org>
Date: Tue, 23 Jul 2019 13:02:34 +0200
Subject: [PATCH] Fix compilation with Qt 5.13 (missing include)

---
 sheets/plugins/calendar/CalendarToolWidget.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/sheets/plugins/calendar/CalendarToolWidget.cpp b/sheets/plugins/calendar/CalendarToolWidget.cpp
index b10b23022a2..0c5eeb68cf3 100644
--- sheets/plugins/calendar/CalendarToolWidget.cpp
+++ sheets/plugins/calendar/CalendarToolWidget.cpp
@@ -30,6 +30,7 @@
 #include <kdatepicker.h>
 
 #include <QPushButton>
+#include <QDate>
 
 namespace Calligra
 {
