--- a/src/core/zip/zip.h
+++ b/src/core/zip/zip.h
@@ -19,15 +19,6 @@
 extern "C" {
 #endif
 
-#if !defined(_SSIZE_T_DEFINED) && !defined(_SSIZE_T_DEFINED_) && \
-    !defined(_SSIZE_T) && !defined(_SSIZE_T_) && \
-    !defined(__ssize_t_defined) && !defined(ssize_t) && \
-    !defined(_SSIZE_T_DECLARED) && !defined(HAVE_SSIZE_T)
-#define _SSIZE_T
-#define __ssize_t_defined
-typedef long  ssize_t;  /* byte count or error */
-#endif
-
 #ifndef MAX_PATH
 #define MAX_PATH 32767 /* # chars in a path name including NULL */
 #endif
