From cbace147456d6ccc1921f62582a791b340b8a88d Mon Sep 17 00:00:00 2001
From: q66 <daniel@octaforge.org>
Date: Fri, 10 Apr 2020 20:31:00 +0200
Subject: [PATCH] fix build on big endian

---
 Source/MACLib/MD5.cpp | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git Source/MACLib/MD5.cpp Source/MACLib/MD5.cpp
index c122fb5..89079b2 100644
--- a/Source/MACLib/MD5.cpp
+++ b/Source/MACLib/MD5.cpp
@@ -35,11 +35,11 @@ CopyToLittleEndian ( uint32_t*       dst,
                      const uint8_t*  src, 
              size_t          length ) 
 {
-    for ( ; length--; src += 4; dst++  ) {
-    *dst = (( (uint32_t) src [3] ) << 24) |
-           (( (uint32_t) src [2] ) << 16) |
-           (( (uint32_t) src [1] ) <<  8) |
-           (( (uint32_t) src [0] ) <<  0);
+    for ( ; length--; src += 4 ) {
+    *dst++ = (( (uint32_t) src [3] ) << 24) |
+             (( (uint32_t) src [2] ) << 16) |
+             (( (uint32_t) src [1] ) <<  8) |
+             (( (uint32_t) src [0] ) <<  0);
        
        
     }
