diff --git a/include/util.h b/include/util.h
index b45fd53..3f2d80a 100644
--- a/include/util.h
+++ b/include/util.h
@@ -1,11 +1,11 @@
 #ifndef UTILH_
 #define UTILH_
 
-#include <bits/types/struct_timeval.h>
 #include <stdbool.h>
 #include <stddef.h>
 #include <stdint.h>
 #include <stdio.h>
+#include <sys/time.h>
 #include <sys/types.h>
 
 #include "keys.h"
diff --git a/src/auth.c b/src/auth.c
index d686cd1..1c06836 100644
--- a/src/auth.c
+++ b/src/auth.c
@@ -245,7 +245,7 @@ bool launch(char* user, char* passwd, struct session session, void (*cb)(void),
     perror("execl error");
     (void)fputs("failure calling session\n", stderr);
   } else {
-    __pid_t child_pid = (__pid_t)pid;
+    pid_t child_pid = (pid_t)pid;
     waitpid(child_pid, NULL, 0);
 
     pam_setcred(pamh, PAM_DELETE_CRED);
