--- a/source/ImageBuffer.h
+++ b/source/ImageBuffer.h
@@ -13,6 +13,7 @@ PARTICULAR PURPOSE.  See the GNU General
 #ifndef IMAGE_BUFFER_H_
 #define IMAGE_BUFFER_H_
 
+#include <cstdint>
 #include <string>
 
 
--- a/source/Sound.cpp
+++ b/source/Sound.cpp
@@ -21,6 +21,7 @@ PARTICULAR PURPOSE.  See the GNU General
 #include <OpenAL/al.h>
 #endif
 
+#include <cstdint>
 #include <cstdio>
 #include <vector>
 
