--- a/SConstruct.old	2017-12-06 12:54:20.524502000 -0800
+++ b/SConstruct	2017-12-12 12:39:21.764384877 -0800
@@ -18,7 +18,7 @@
 	env.Append(LINKFLAGS = ["-static-libstdc++"])
 
 opts = Variables()
-opts.Add(PathVariable("PREFIX", "Directory to install under", "/usr/local", PathVariable.PathIsDirCreate))
+opts.Add(PathVariable("PREFIX", "Directory to install under", "/usr", PathVariable.PathIsDirCreate))
 opts.Add(PathVariable("DESTDIR", "Destination root directory", "", PathVariable.PathAccept))
 opts.Add(EnumVariable("mode", "Compilation mode", "release", allowed_values=("release", "debug", "profile")))
 opts.Update(env)
@@ -66,7 +66,7 @@
 
 
 # Install the binary:
-env.Install("$DESTDIR$PREFIX/games", sky)
+env.Install("$DESTDIR$PREFIX/bin", sky)
 
 # Install the desktop file:
 env.Install("$DESTDIR$PREFIX/share/applications", "endless-sky.desktop")
