diff --git a/pyproject.toml b/pyproject.toml
index 74ca2694..a61246c6 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -192,12 +192,6 @@ method = { module = "build_backend.onbuild", value = "onbuild" }
 # https://docs.pytest.org/en/latest/reference/customize.html#configuration
 # https://docs.pytest.org/en/latest/reference/reference.html#ini-options-ref
 [tool.pytest.ini_options]
-minversion = "8.4.0"
-addopts = """
---disable-plugin-autoload
--p trio
--p requests_mock
-"""
 testpaths = [
   "build_backend",
   "tests",
