exceptiongroup is for python<3.11 compat

--- a/src/streamlink.egg-info/requires.txt
+++ b/src/streamlink.egg-info/requires.txt
@@ -1,5 +1,4 @@
 certifi
-exceptiongroup
 isodate
 lxml<6,>=4.6.4
 pycountry
--- a/src/streamlink.egg-info/PKG-INFO
+++ b/src/streamlink.egg-info/PKG-INFO
@@ -32,7 +32,6 @@
 Description-Content-Type: text/markdown
 License-File: LICENSE
 Requires-Dist: certifi
-Requires-Dist: exceptiongroup
 Requires-Dist: isodate
 Requires-Dist: lxml<6,>=4.6.4
 Requires-Dist: pycountry
--- a/PKG-INFO
+++ b/PKG-INFO
@@ -32,7 +32,6 @@
 Description-Content-Type: text/markdown
 License-File: LICENSE
 Requires-Dist: certifi
-Requires-Dist: exceptiongroup
 Requires-Dist: isodate
 Requires-Dist: lxml<6,>=4.6.4
 Requires-Dist: pycountry
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -56,7 +56,6 @@
 requires-python = ">=3.8"
 dependencies = [
   "certifi",
-  "exceptiongroup",
   "isodate",
   "lxml >=4.6.4,<6",
   "pycountry",
--- a/src/streamlink/plugins/twitch.py
+++ b/src/streamlink/plugins/twitch.py
@@ -541,7 +541,6 @@
         headers: Mapping[str, str],
         device_id: str,
     ) -> Optional[Tuple[str, int]]:
-        from exceptiongroup import BaseExceptionGroup  # noqa: PLC0415, I001
         from streamlink.webbrowser.cdp import CDPClient, CDPClientSession, devtools  # noqa: PLC0415
 
         url = f"https://www.twitch.tv/{channel}"
--- a/src/streamlink/webbrowser/webbrowser.py
+++ b/src/streamlink/webbrowser/webbrowser.py
@@ -8,7 +8,6 @@
 from typing import AsyncContextManager, AsyncGenerator, Generator, List, Optional, Union
 
 import trio
-from exceptiongroup import BaseExceptionGroup
 
 from streamlink.utils.path import resolve_executable
 from streamlink.webbrowser.exceptions import WebbrowserError
--- a/src/streamlink/compat.py
+++ b/src/streamlink/compat.py
@@ -5,9 +5,6 @@
 import warnings
 from typing import Any, Callable, Dict, Optional, Tuple
 
-# import exceptiongroup, so it can monkeypatch ExceptionGroup logic on <=py311
-import exceptiongroup  # noqa: F401
-
 from streamlink.exceptions import StreamlinkDeprecationWarning
 
 
--- a/tests/webbrowser/test_webbrowser.py
+++ b/tests/webbrowser/test_webbrowser.py
@@ -7,7 +7,6 @@
 
 import pytest
 import trio
-from exceptiongroup import BaseExceptionGroup
 
 from streamlink.compat import is_win32
 from streamlink.webbrowser.exceptions import WebbrowserError
--- a/tests/webbrowser/cdp/test_connection.py
+++ b/tests/webbrowser/cdp/test_connection.py
@@ -7,7 +7,6 @@
 
 import pytest
 import trio
-from exceptiongroup import ExceptionGroup
 from trio.testing import MockClock, wait_all_tasks_blocked
 from trio_websocket import CloseReason, ConnectionClosed, ConnectionTimeout  # type: ignore[import]
 
--- a/tests/webbrowser/cdp/test_client.py
+++ b/tests/webbrowser/cdp/test_client.py
@@ -4,7 +4,6 @@
 
 import pytest
 import trio
-from exceptiongroup import ExceptionGroup
 from trio.testing import wait_all_tasks_blocked
 
 from streamlink.session import Streamlink
