diff -urN gpsd-3.15-old/SConstruct gpsd-3.15-new/SConstruct
--- gpsd-3.15-old/SConstruct	2015-06-03 14:09:31.000000000 +0000
+++ gpsd-3.15-new/SConstruct	2015-09-17 18:17:49.140818485 +0000
@@ -539,8 +539,8 @@
     # are like FreeBSD.
     ncurseslibs= []
     if env['ncurses']:
-        if config.CheckPKG('ncurses'):
-            ncurseslibs = pkg_config('ncurses')
+        if config.CheckPKG('ncursesw'):
+            ncurseslibs = pkg_config('ncursesw')
             if config.CheckPKG('tinfo'):
                 ncurseslibs += pkg_config('tinfo')
         elif WhereIs('ncurses5-config'):
