
--- Makefile.in.orig
+++ Makefile.in
@@ -2,7 +2,7 @@
 
 prefix = @prefix@
 exec_prefix = @exec_prefix@
-BINDIR = ${exec_prefix}/@bindir@
+BINDIR = @bindir@
 CC = @CC@
 FC = @FC@
 HAVE_NR = @HAVE_NR@

--- source_c/Makefile.in.orig
+++ source_c/Makefile.in
@@ -2,7 +2,7 @@
 
 prefix = @prefix@
 exec_prefix = @exec_prefix@
-BINDIR = ${exec_prefix}/@bindir@
+BINDIR = @bindir@
 
 CC = @CC@
 CFLAGS = @CFLAGS@

--- source_f/Makefile.in.orig
+++ source_f/Makefile.in
@@ -2,7 +2,7 @@
 
 prefix = @prefix@
 exec_prefix = @exec_prefix@
-BINDIR = ${exec_prefix}/@bindir@
+BINDIR = @bindir@
 
 FC = @FC@
 FFLAGS = @FFLAGS@

--- source_f/randomize/Makefile.in.orig
+++ source_f/randomize/Makefile.in
@@ -2,7 +2,7 @@
 
 prefix = @prefix@
 exec_prefix = @exec_prefix@
-BINDIR = ${exec_prefix}/@bindir@
+BINDIR = @bindir@
 
 FC = @FC@
 FFLAGS = @FFLAGS@
