--- preload-0.6.4/src/Makefile.in	2009-04-15 17:49:28.000000000 -0400
+++ preload-0.6.4.new/src/Makefile.in	2023-05-30 22:23:16.820648202 -0400
@@ -222,6 +222,12 @@
 dist_man_MANS = preload.8
 RUNPREQ = preload preload.conf.debug
 RUNCMD = ./preload -c preload.conf.debug -s preload.state -d $(ARGS)
+preload_binary = `if [ -n "$$CROSS_BUILD" ]; then \
+				 	echo "qemu-$$XBPS_TARGET_QEMU_MACHINE-static ./preload"; \
+				else \
+				 	echo './preload'; \
+				fi;`
+
 all: all-am
 
 .SUFFIXES:
@@ -678,11 +684,11 @@
 	@$(top_builddir)/missing --run \
 	   help2man --no-info --section=8 --include=preload.8.i \
 	   	    --help-option="-H" --output="$@.tmp" \
-		    --name 'Adaptive readahead daemon' ./preload \
+		    --name 'Adaptive readahead daemon' "$(preload_binary)" \
 	 && mv "$@.tmp" "$@" \
 	  || ($(RM) "$@"; \
-	      echo Failed to update preload.8, the man page may be outdated >&2; \
-	      (test -f "$@" || echo help2man is required to generate this file. >> "$@"));
+	      echo Failed to create preload.8 >&2; \
+		  exit 1);
 
 install-data-hook:
 	@cd "$(DESTDIR)$(man8dir)" && gzip -c preload.8 > preload.8.gz.tmp && mv preload.8.gz.tmp preload.8.gz && $(RM) preload.8
