diff --git a/betterlockscreen b/betterlockscreen
index 0b92ad9..fdd7b60 100755
--- a/betterlockscreen
+++ b/betterlockscreen
@@ -20,7 +20,7 @@ init_config () {
     description=""
     quiet=false
     i3lockcolor_bin="i3lock-color"
-    suspend_command="systemctl suspend"
+    suspend_command="loginctl suspend"
     convert_command="magick"
     composite_command="magick composite"
 
diff --git a/examples/betterlockscreenrc b/examples/betterlockscreenrc
index 84e355c..ed69e66 100644
--- a/examples/betterlockscreenrc
+++ b/examples/betterlockscreenrc
@@ -43,7 +43,7 @@ bgcolor=000000ff
 #
 
 # i3lockcolor_bin="i3lock-color"      # Manually set command for i3lock-color
-# suspend_command="systemctl suspend" # Manually change action e.g. hibernate/suspend-command
+# suspend_command="loginctl suspend" # Manually change action e.g. hibernate/suspend-command
 
 # i3lock-color - custom arguments
 # lockargs=() 						  # overwriting default "(-n)"

