diff -Naur betterlockscreen-4.0.3/betterlockscreen betterlockscreen-4.0.3-patched/betterlockscreen
--- betterlockscreen-4.0.3/betterlockscreen	2021-08-22 03:39:34.000000000 +0800
+++ betterlockscreen-4.0.3-patched/betterlockscreen	2022-09-05 17:29:51.165948726 +0800
@@ -965,6 +965,6 @@
 # Activate lockscreen
 [[ $runsuspend ]] || lockargs+=(-n)
 [[ $runlock ]] && lockselect "$lockstyle" && \
-    { [[ $runsuspend ]] && systemctl suspend; }
+    { [[ $runsuspend ]] && loginctl suspend; }
 
 exit 0
\ No newline at end of file
