--- configure.ac	2019-04-23 12:08:03.000000000 +0200
+++ configure.ac	2019-12-16 13:59:13.310895364 +0100
@@ -685,7 +685,7 @@
             AC_SUBST([COGL_EGL_INCLUDES])
 
             AC_CHECK_HEADERS([EGL/eglext.h],
-                             [COGL_EGL_INCLUDES="$COGL_EGL_INCLUDE
+                             [COGL_EGL_INCLUDES="$COGL_EGL_INCLUDES
 #include <EGL/eglext.h>"],
                              [],
                              [$COGL_EGL_INCLUDES])
@@ -1226,7 +1226,8 @@
           )
 
         COGL_EGL_INCLUDES="#include <EGL/egl.h>
-#include <EGL/eglext.h>"
+#include <EGL/eglext.h>
+#include <EGL/eglmesaext.h>"
         AC_SUBST([COGL_EGL_INCLUDES])
       ])
 
