--- a/plugins/avrmcu/plugin/firmware.c
+++ b/plugins/avrmcu/plugin/firmware.c
@@ -118,10 +118,6 @@
             case AVR_MMCU_TAG_SIMAVR_CONSOLE: {
                 firmware->console_register_addr = src[0] | (src[1] << 8);
             }	break;
-			case AVR_MMCU_TAG_BITBANG: {
-				firmware->bitbang_on_mask =
-					src[0] | (src[1] << 8) | (src[2] << 16) | (src[3] << 24);
-			}	break;
         }
         size -= next;
         src += next - 2; // already incremented
