--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -693,15 +693,6 @@ else()
   set(mi_install_cmakedir   "${CMAKE_INSTALL_LIBDIR}/cmake/mimalloc-${mi_version}") # for cmake package info
 endif()
 set(mi_libname "mimalloc")
-if(MI_SECURE)
-  set(mi_libname "${mi_libname}-secure")
-endif()
-if(MI_TRACK STREQUAL "VALGRIND")
-  set(mi_libname "${mi_libname}-valgrind")
-endif()
-if(MI_TRACK STREQUAL "ASAN")
-  set(mi_libname "${mi_libname}-asan")
-endif()
 string(TOLOWER "${CMAKE_BUILD_TYPE}" CMAKE_BUILD_TYPE_LC)
 list(APPEND mi_defines "MI_CMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE_LC}")  #todo: multi-config project needs $<CONFIG> ?
 if(CMAKE_BUILD_TYPE_LC MATCHES "^(release|relwithdebinfo|minsizerel|none)$")
