--- a/CMakeLists.txt	2026-01-08 14:53:15.000000000 -0600
+++ b/CMakeLists.txt	2026-01-09 09:23:03.059221454 -0600
@@ -560,15 +560,6 @@
 endif()
 
 set(mi_libname "mimalloc")
-if(MI_SECURE)
-  set(mi_libname "${mi_libname}-secure")
-endif()
-if(MI_TRACK_VALGRIND)
-  set(mi_libname "${mi_libname}-valgrind")
-endif()
-if(MI_TRACK_ASAN)
-  set(mi_libname "${mi_libname}-asan")
-endif()
 string(TOLOWER "${CMAKE_BUILD_TYPE}" CMAKE_BUILD_TYPE_LC)
 list(APPEND mi_defines "MI_CMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE_LC}")  #todo: multi-config project needs $<CONFIG> ?
 if(CMAKE_BUILD_TYPE_LC MATCHES "^(release|relwithdebinfo|minsizerel|none)$")
