--- ./afew/Settings.py.orig	2023-09-28 15:08:53.611940644 -0400
+++ ./afew/Settings.py	2023-09-28 15:09:06.740017084 -0400
@@ -18,7 +18,7 @@
 # preserve the capitalization of the keys.
 settings.optionxform = str
 
-settings.readfp(open(os.path.join(os.path.dirname(__file__), 'defaults', 'afew.config')))
+settings.read_file(open(os.path.join(os.path.dirname(__file__), 'defaults', 'afew.config')))
 settings.read(os.path.join(user_config_dir, 'config'))
 
 # All the values for keys listed here are interpreted as ;-delimited lists
