From 0b78aea138daee5de1ba5fad126625f245134c6b Mon Sep 17 00:00:00 2001
From: intrigeri <intrigeri@boum.org>
Date: Mon, 9 Sep 2019 08:32:52 +0000
Subject: [PATCH] AppArmor: allow new path for the Firefox 68 IPC

---
 apparmor/torbrowser.Browser.firefox | 1 +
 1 file changed, 1 insertion(+)

diff --git a/apparmor/torbrowser.Browser.firefox b/apparmor/torbrowser.Browser.firefox
index f782f35..c6d8a26 100644
--- apparmor/torbrowser.Browser.firefox
+++ apparmor/torbrowser.Browser.firefox
@@ -105,6 +105,7 @@ profile torbrowser_firefox @{torbrowser_firefox_executable} {
 
   # Required for multiprocess Firefox (aka Electrolysis, i.e. e10s)
   owner /{dev,run}/shm/org.chromium.* rw,
+  owner /dev/shm/org.mozilla.ipc.[0-9]*.[0-9]* rw, # for Chromium IPC
 
   # Deny access to DRM nodes, that's granted by the X abstraction, which is
   # sourced by the gnome abstraction, that we include.
