--- src/c.orig	2017-09-04 21:17:26.180830076 -0400
+++ src/c	2017-09-04 21:34:22.561799357 -0400
@@ -154,8 +154,7 @@
 #
 #  Generic modern GCC system
 #
-case "$CCNAME" in
-*gcc*)
+if [ "$(${CCNAME} --version | grep -c \(GCC\))" -gt 0 ]; then
     [ -z "$BOOM_MODEL_NOOPT" ] && CCDEBUG="-O2"
     [ -z "$BOOM_MODEL_NOOPT" ] && CCNODEBUG="$CCNODEBUG -O2"
     CCOPTS="-D_REENTRANT -D_GNU_SOURCE -Wall -Wno-unused -fno-strict-aliasing"
@@ -212,20 +211,18 @@
             STDLIBS="-lapr $STDLIBS"
         fi
     fi
-    ;;
-*)
-#
-#  AIX with xlc
-#
-if [ "$UTYPE" = "AIX" -a "$CCNAME" = "xlc_r" ]; then
-    [ -z "$BOOM_MODEL_NOOPT" ] && CCNODEBUG="-O"
-    CCNAME="xlc_r"            #  Use VAC Threaded Mode
-    CCPLUS="xlC_r"            #  Use VAC Threaded Mode
-    CCOPTS="$CCOPTS -D_REENTRANT"
-    STDLIBS="-lpthread"
+else
+	#
+	#  AIX with xlc
+	#
+	if [ "$UTYPE" = "AIX" -a "$CCNAME" = "xlc_r" ]; then
+		[ -z "$BOOM_MODEL_NOOPT" ] && CCNODEBUG="-O"
+		CCNAME="xlc_r"            #  Use VAC Threaded Mode
+		CCPLUS="xlC_r"            #  Use VAC Threaded Mode
+		CCOPTS="$CCOPTS -D_REENTRANT"
+		STDLIBS="-lpthread"
+	fi
 fi
-;;
-esac
 
 #   Patch together the CC options and defines into one variable
 CCOPTS="$CCOPTS $CCDEFINES"
