--- src/c.fix	2017-02-10 21:50:13.882583271 -0500
+++ src/c	2017-02-10 21:55:55.064681445 -0500
@@ -154,7 +154,7 @@
 #
 #  Generic modern GCC system
 #
-if [ "$CCNAME" = "gcc" ]; then
+if [ "$(${CCNAME} --version | grep -c \(GCC\))" -gt 0 ]; then
     [ -z "$BOOM_MODEL_NOOPT" ] && CCDEBUG="-O2"
     [ -z "$BOOM_MODEL_NOOPT" ] && CCNODEBUG="$CCNODEBUG -O2"
     CCOPTS="-D_REENTRANT -D_GNU_SOURCE -Wall -Wno-unused -fno-strict-aliasing"
