--- src/Makefile.bck	2017-09-06 02:54:04.375680332 -0400
+++ src/Makefile	2017-09-06 02:54:16.048706017 -0400
@@ -185,7 +185,6 @@
 smtxlog$(OBJ): smtxlog.c
 
 everything: libgsl$(LIB) gsl$(EXE)
-	./gsl$(EXE) -v
 
 code:
 	./gsl$(EXE) -q ggconv.gxl ggdiag.gxl ggenvt.gxl ggfile.gxl gggsl.gxl ggmath.gxl ggpcre.gxl ggproc.gxl ggscrp.gxl ggsock.gxl ggstrn.gxl ggsymb.gxl ggthrd.gxl ggtime.gxl ggxml.gxl
