--- a/src/sfl.h	2017-09-13 15:35:49.000000000 +0200
+++ -	2023-02-07 16:09:26.971059849 +0100
@@ -130,7 +130,8 @@
 
 #if (defined (__64BIT__) || defined (__x86_64__) || defined (__AARCH64EL__) \
     || defined (__PPC64__) || defined (__powerpc64__) || defined (__ppc64__) \
-    || defined (__s390x__))
+    || defined (__s390x__)) \
+    || defined (__riscv64) || (defined (__riscv_xlen) && __riscv_xlen == 64)
 #    define __IS_64BIT__                /*  May have 64-bit OS/compiler      */
 #else
 #    define __IS_32BIT__                /*  Else assume 32-bit OS/compiler   */
--- a/src/prelude.h	2017-09-13 15:35:49.000000000 +0200
+++ -	2023-02-07 16:09:50.387722331 +0100
@@ -91,7 +91,8 @@
 
 #if (defined (__64BIT__) || defined (__x86_64__) || defined (__AARCH64EL__) \
     || defined (__PPC64__) || defined (__powerpc64__) || defined (__ppc64__) \
-    || defined (__s390x__))
+    || defined (__s390x__)) \
+    || defined (__riscv64) || (defined (__riscv_xlen) && __riscv_xlen == 64)
 #    define __IS_64BIT__                /*  May have 64-bit OS/compiler      */
 #else
 #    define __IS_32BIT__                /*  Else assume 32-bit OS/compiler   */
