diff --git a/data/meson.build b/data/meson.build
index 9b2ea4b..d4fbb8e 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -49,15 +49,15 @@ metainfo_file = i18n.merge_file(
 )
 
 # Validate MetaInfo file
-if appstreamcli.found()
-  test (
-    'Validate metainfo file',
-    appstreamcli,
-    args: ['validate', '--no-net',
-           join_paths(meson.current_source_dir(), 'metainfo', '@0@.in'.format(tilix_metainfo_name))
-    ]
-  )
-endif
+# if appstreamcli.found()
+#   test (
+#     'Validate metainfo file',
+#     appstreamcli,
+#     args: ['validate', '--no-net',
+#            join_paths(meson.current_source_dir(), 'metainfo', '@0@.in'.format(tilix_metainfo_name))
+#     ]
+#   )
+# endif
 
 # Build resources
 gresource = gnome.compile_resources(
