--- a/source/gx/tilix/prefeditor/prefdialog.d
+++ b/source/gx/tilix/prefeditor/prefdialog.d
@@ -957,7 +957,7 @@ private:
             return;
         }
 
-        import std.xml: DocumentParser, ElementParser, Element, XMLException;
+        import undead.xml: DocumentParser, ElementParser, Element, XMLException;
 
         try {
             DocumentParser parser = new DocumentParser(ui);
--- a/meson.build
+++ b/meson.build
@@ -30,6 +30,7 @@
 appdir = datadir / 'applications'
 
 tilix_sources = [
+    'undead/xml.d',
     'source/gx/gtk/actions.d',
     'source/gx/gtk/cairo.d',
     'source/gx/gtk/clipboard.d',
