--- a/source/gx/tilix/prefeditor/prefdialog.d
+++ b/source/gx/tilix/prefeditor/prefdialog.d
@@ -957,7 +957,7 @@ private:
             return;
         }
 
-        import std.xml: DocumentParser, ElementParser, Element, XMLException;
+        import undead.xml: DocumentParser, ElementParser, Element, XMLException;
 
         try {
             DocumentParser parser = new DocumentParser(ui);
--- a/meson.build
+++ b/meson.build
@@ -91,6 +91,7 @@ tilix_sources = [
     'source/secretc/secrettypes.d',
     'source/x11/X.d',
     'source/x11/Xlib.d',
+    'source/undead/xml.d',
     'source/app.d'
 ]
 
