--- src/master/master-settings.c.orig
+++ src/master/master-settings.c
@@ -209,9 +209,9 @@
 	.protocols = "imap pop3 lmtp",
 	.listen = "*, ::",
 	.ssl = "yes:no:required",
-	.default_internal_user = "dovecot",
-	.default_internal_group = "dovecot",
-	.default_login_user = "dovenull",
+	.default_internal_user = "_dovecot",
+	.default_internal_group = "_dovecot",
+	.default_login_user = "_dovenull",
 	.default_process_limit = 100,
 	.default_client_limit = 1000,
 	.default_idle_kill = 60,
