--- a/_scripts/build.js	2022-08-12 04:09:42.000000000 +0200
+++ b/_scripts/build.js	2022-08-31 21:06:38.025503775 +0200
@@ -33,7 +33,7 @@
     arch = Arch.armv7l
   }
 
-  targets = Platform.LINUX.createTarget(['deb', 'zip', 'apk', 'rpm', 'AppImage', 'pacman'], arch)
+  targets = Platform.LINUX.createTarget(['dir'], arch)
 }
 
 const config = {
@@ -97,7 +97,7 @@
   linux: {
     category: 'Network',
     icon: '_icons/icon.svg',
-    target: ['deb', 'zip', 'apk', 'rpm', 'AppImage', 'pacman'],
+    target: ['dir'],
   },
   // See the following issues for more information
   // https://github.com/jordansissel/fpm/issues/1503
