--- src/cairo-xlib-display.c.ubuntu	2010-08-04 11:57:49.000000000 +0200
+++ src/cairo-xlib-display.c	2010-08-04 11:58:28.000000000 +0200
@@ -353,11 +353,7 @@
     /* Prior to Render 0.10, there is no protocol support for gradients and
      * we call function stubs instead, which would silently consume the drawing.
      */
-#if RENDER_MAJOR == 0 && RENDER_MINOR < 10
     display->buggy_gradients = TRUE;
-#else
-    display->buggy_gradients = FALSE;
-#endif
     display->buggy_pad_reflect = FALSE;
     display->buggy_repeat = FALSE;
