--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -237,7 +237,7 @@ else()
 endif()
 
 add_custom_command( TARGET docbookl10nhelper POST_BUILD
-    COMMAND $<TARGET_FILE:KF6::docbookl10nhelper>
+    COMMAND KF6::docbookl10nhelper
     "${DOCBOOKXSL_DIR}" "${CMAKE_CURRENT_SOURCE_DIR}/customization/xsl"
     "${CMAKE_CURRENT_BINARY_DIR}/customization/xsl"
 )
--- a/KF6DocToolsConfig.cmake.in
+++ b/KF6DocToolsConfig.cmake.in
@@ -8,7 +8,7 @@ endif()
 include("${CMAKE_CURRENT_LIST_DIR}/KF6DocToolsTargets.cmake")
 
     if (CMAKE_CROSSCOMPILING AND KF6_HOST_TOOLING)
-    find_file(KDOCTOOLS_TARGETSFILE KF6DocTools/KF6DocToolsToolsTargets.cmake PATHS ${KF6_HOST_TOOLING} ${CMAKE_CURRENT_LIST_DIR} NO_DEFAULT_PATH)
+    find_file(KDOCTOOLS_TARGETSFILE KF6DocTools/KF6DocToolsToolsTargets.cmake PATHS ${KF6_HOST_TOOLING} ${CMAKE_CURRENT_LIST_DIR} NO_DEFAULT_PATH NO_CMAKE_FIND_ROOT_PATH)
     include("${KDOCTOOLS_TARGETSFILE}")
 else()
     include("${CMAKE_CURRENT_LIST_DIR}/KF6DocToolsToolsTargets.cmake")
