--- a/lib/igt_vc4.h
+++ b/lib/igt_vc4.h
@@ -31,7 +31,9 @@
 #include "igt_fb.h"
 #include "vc4_drm.h"

+#ifndef PAGE_SIZE
 #define PAGE_SIZE 4096
+#endif

 uint32_t igt_vc4_get_cleared_bo(int fd, size_t size, uint32_t clearval);
 int igt_vc4_create_bo(int fd, size_t size);

--- a/lib/xe/xe_sriov_debugfs.c
+++ b/lib/xe/xe_sriov_debugfs.c
@@ -3,6 +3,7 @@
  * Copyright(c) 2024 Intel Corporation. All rights reserved.
  */

+#include <limits.h>
 #include <dirent.h>
 #include <fcntl.h>

--- a/lib/svga/vm_basic_types.h
+++ b/lib/svga/vm_basic_types.h
@@ -27,6 +27,9 @@
 #define VM_BASIC_TYPES_H

 #define PAGE_SHIFT 12
+#ifdef PAGE_SIZE
+#undef PAGE_SIZE
+#endif
 #define PAGE_SIZE (1 << PAGE_SHIFT)
 #define PAGE_MASK (~(PAGE_SIZE - 1))
 #define PAGE_ALIGN(addr) (((addr) + PAGE_SIZE - 1) & PAGE_MASK)
