diff --git src/libqhull/io.c src/libqhull/io.c
index 09a0079..7ad4b35 100644
--- src/libqhull/io.c
+++ src/libqhull/io.c
@@ -68,8 +68,8 @@ void qh_produce_output2(void) {
     qh_memstatistics(qh ferr);
     d_1= sizeof(setT) + (qh hull_dim - 1) * SETelemsize;
     qh_fprintf(qh ferr, 8040, "\
-    size in bytes: merge %d ridge %d vertex %d facet %d\n\
-         normal %d ridge vertices %d facet vertices or neighbors %d\n",
+    size in bytes: merge %zd ridge %zd vertex %zd facet %zd\n\
+         normal %d ridge vertices %d facet vertices or neighbors %zd\n",
             (int)sizeof(mergeT), (int)sizeof(ridgeT),
             (int)sizeof(vertexT), (int)sizeof(facetT),
             qh normal_size, d_1, d_1 + SETelemsize);
diff --git src/libqhull/stat.c src/libqhull/stat.c
index 7dc60b9..5721130 100644
--- src/libqhull/stat.c
+++ src/libqhull/stat.c
@@ -472,7 +472,7 @@ void qh_initstatistics(void) {
   qh_allstatI();
   if (qhstat next > (int)sizeof(qhstat id)) {
     qh_fprintf(qhmem.ferr, 6184, "qhull error (qh_initstatistics): increase size of qhstat.id[].\n\
-      qhstat.next %d should be <= sizeof(qhstat id) %d\n", qhstat next, (int)sizeof(qhstat id));
+      qhstat.next %d should be <= sizeof(qhstat id) %zd\n", qhstat next, (int)sizeof(qhstat id));
 #if 0 /* for locating error, Znumridges should be duplicated */
     for(i=0; i < ZEND; i++) {
       int j;

