diff --git src/libqhull/io.c src/libqhull/io.c
index 6601ffd..fb1a7fd 100644
--- src/libqhull/io.c
+++ src/libqhull/io.c
@@ -2238,7 +2238,7 @@ void qh_printfacet3math(FILE *fp, facetT *facet, qh_PRINT format, int notfirst)
     qh_memfree(point, qh normal_size);
   qh_settempfree(&points);
   qh_settempfree(&vertices);
-  qh_fprintf(fp, 9110, endfmt);
+  qh_fprintf(fp, 9110, "%s", endfmt);
 } /* printfacet3math */
 
 
