---
 etc/fb.modes.ATI |   84 ++++++++++++++++++++++++++++++++++++++++++++++++++-----
 1 file changed, 77 insertions(+), 7 deletions(-)

--- a/etc/fb.modes.ATI
+++ b/etc/fb.modes.ATI
@@ -6,6 +6,8 @@
 #       Mach64 Programmer's Guide, Appendix C
 #       (C) 1998 ATI Technologies Inc.
 #
+#  Kop: this are very generic modes and not only for ATI cards.
+#
 
 #
 #	640x480, 60 Hz, Non-Interlaced (25.175 MHz dotclock)
@@ -132,6 +134,36 @@ mode "640x480-100"
     timings 22272 48 32 17 22 128 12
 endmode
  
+
+#
+#       768x576, 75 Hz, Non-Interlaced (49.188 MHz dotclock)
+#
+#                               Horizontal      Vertical
+#       Resolution              768             576
+#       Scan Frequency          46.580 kHz      75.008 Hz
+#       Sync Width              us              ms
+#                               chars           lines
+#       Front Porch             us              ms
+#                               chars           lines
+#       Back Porch              us              ms
+#                               chars           lines
+#       Active Time             us              ms
+#                               chars           lines
+#       Blank Time              us              ms
+#                               chars           lines
+#       Polarity                negative        negative
+#
+# This is a mode often used, because fbtv suggests this, since
+# this is the mode for "normal" TVs.
+#
+
+mode "768x576-75"
+    # D: 49.188 MHz, H: 46.580 kHz, V: 75.008 Hz
+    geometry 768 576 768 576 32
+    timings 20330 128 32 32 8 128 5
+endmode
+
+
 #
 #	800x600, 48 Hz, Interlaced (36.00 MHz dotclock)
 #	
@@ -430,7 +462,7 @@ endmode
 #	1024x768, 72 Hz, Non-Interlaced (75.00 MHz dotclock)
 #	
 #				Horizontal	Vertical
-#	Resolution		10224		768
+#	Resolution		1024		768
 #	Scan Frequency		58.230 kHz	72.245 Hz
 #	Sync Width		1.813 us	0.103 ms
 #				17 chars	6 lines
@@ -447,7 +479,7 @@ endmode
 
 mode "1024x768-72"
     # D: 75.00 MHz, H: 58.230 kHz, V: 72.245 Hz
-    geometry 10224 768 10224 768 8
+    geometry 1024 768 1024 768 8
     timings 13334 104 24 29 3 136 6
 endmode
  
@@ -691,7 +723,45 @@ mode "1152x864-80"
     hsync high
     vsync high
 endmode
- 
+
+#
+#       1280x960, 75 Hz, Non-Interlaced (126.00 MHz dotclock)
+#
+#                               Horizontal      Vertical
+#       Resolution              1280            960
+#       Scan Frequency          74.788 kHz      74.788 Hz
+#       Sync Width              1.018 us        0.092 ms
+#                               14 chars        7 lines
+#       Front Porch             0.127 us        0.393 ms
+#                               2 chars         30 lines
+#       Back Porch              1.473 us        0.747 ms
+#                               20 chars        57 lines
+#       Active Time             10.473 us       11.311 ms
+#                               144 chars       864 lines
+#       Blank Time              2.618 us        1.231 ms
+#                               36 chars        94 lines
+#       Polarity                positive        positive
+#
+
+mode "1280x960-75-8"
+    # D: 125.644 MHz, H: 74.788 kHz, V: 74.788 Hz
+    geometry 1280 960 1280 960 8
+    timings 7959 224 32 36 1 144 3
+endmode
+
+mode "1280x960-75"
+    # D: 125.644 MHz, H: 74.788 kHz, V: 74.788 Hz
+    geometry 1280 960 1280 960 16
+    timings 7959 224 32 36 1 144 3
+endmode
+
+mode "1280x960-75-32"
+    # D: 125.644 MHz, H: 74.788 kHz, V: 74.788 Hz
+    geometry 1280 960 1280 960 32
+    timings 7959 224 32 36 1 144 3
+endmode
+
+
 #
 #	1280x1024, 43 Hz, Interlaced (80.00 MHz dotclock)
 #	
@@ -849,13 +919,13 @@ endmode
 #
 
 mode "1280x1024-75"
-    # D: 135.00 MHz, H: 79.976 kHz, V: 75.02 Hz
-    geometry 1280 1024 1280 1024 8
-    timings 7408 248 16 38 1 144 3
+    # D: 134.880 MHz, H: 79.905 kHz, V: 74.958 Hz
+    geometry 1280 1024 1280 3264 8
+    timings 7414 232 64 38 1 112 3
     hsync high
     vsync high
 endmode
- 
+
 #
 #	1600x1200, 60 Hz, Non-Interlaced (156.00 MHz dotclock)
 #	
