Status: sent-upstream

---
 fbset.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/fbset.c
+++ b/fbset.c
@@ -710,7 +710,7 @@ static void DisplayFBInfo(struct fb_fix_
 
     puts("Frame buffer device information:");
     printf("    Name        : %s\n", fix->id);
-    printf("    Address     : %p\n", fix->smem_start);
+    printf("    Address     : %#0lx\n", fix->smem_start);
     printf("    Size        : %d\n", fix->smem_len);
     printf("    Type        : ");
     switch (fix->type) {
@@ -780,7 +780,7 @@ static void DisplayFBInfo(struct fb_fix_
     printf("    YWrapStep   : %d\n", fix->ywrapstep);
     printf("    LineLength  : %d\n", fix->line_length);
     if (fix->mmio_len) {
-	printf("    MMIO Address: %p\n", fix->mmio_start);
+	printf("    MMIO Address: %#0lx\n", fix->mmio_start);
 	printf("    MMIO Size   : %d\n", fix->mmio_len);
     }
     printf("    Accelerator : ");
