Status: sent-upstream

---
 fbset.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/fbset.c
+++ b/fbset.c
@@ -832,7 +832,8 @@ static int FillScanRates(struct VideoMod
 static void Usage(void)
 {
     puts(VERSION);
-    Die("\nUsage: %s [options] [mode]\n\n"
+    printf(
+	"\nUsage: %s [options] [mode]\n\n"
 	"Valid options:\n"
 	"  General options:\n"
 	"    -h, --help         : display this usage information\n"
@@ -887,6 +888,7 @@ static void Usage(void)
 	"    -step <value>      : step increment (in pixels or pixel lines)\n"
 	"                         (default is 8 horizontal, 2 vertical)\n",
 	ProgramName);
+	exit(0);
 }
 
 
