diff --git src/basic/parse-printf-format.c src/basic/parse-printf-format.c
index a99952b..a241760 100644
--- src/basic/parse-printf-format.c
+++ src/basic/parse-printf-format.c
@@ -25,7 +25,7 @@
 
 #include "parse-printf-format.h"
 
-#ifndef HAVE_PRINTF_H
+#if HAVE_PRINTF_H == 0
 
 static const char *consume_nonarg(const char *fmt)
 {
diff --git src/basic/parse-printf-format.h src/basic/parse-printf-format.h
index 5ca0fdf..b36b0fa 100644
--- src/basic/parse-printf-format.h
+++ src/basic/parse-printf-format.h
@@ -24,7 +24,7 @@
 
 #include "config.h"
 
-#ifdef HAVE_PRINTF_H
+#if HAVE_PRINTF_H == 1
 #include <printf.h>
 #else
 
