diff --git src/shared/musl_missing.h src/shared/musl_missing.h
index eaa7ca7..17e946f 100644
--- src/shared/musl_missing.h
+++ src/shared/musl_missing.h
@@ -44,9 +44,6 @@ void elogind_set_program_name(const char* pcall);
 #  define HAVE_SECURE_GETENV 1
 #endif // HAVE_[__]SECURE_GETENV
 
-/* Poor man's basename */
-#define basename(path) \
-        (strrchr(path, '/') ? strrchr(path, '/')+1 : path)
 
 /* strndupa may already be defined in another compatibility header */
 #if !defined(strndupa)

