diff --git src/shared/musl_missing.h src/shared/musl_missing.h
index d42f03c34..cb98836ac 100644
--- src/shared/musl_missing.h
+++ src/shared/musl_missing.h
@@ -68,9 +68,6 @@ extern char *program_invocation_short_name;
 #define canonicalize_file_name(path) \
         realpath(path, NULL)
 
-/* GLOB_BRACE is another glibc extension - ignore it for musl libc */
-#define GLOB_BRACE 0
-
 /* getnameinfo(3) glibc extensions are undefined in musl libc */
 #define NI_IDN 0
 #define NI_IDN_USE_STD3_ASCII_RULES 0

