--- src/login/logind-action.c	2015-08-29 11:16:20.584190227 +0200
+++ src/login/logind-action.c	2017-03-04 13:55:29.722627380 +0100
@@ -176,9 +176,9 @@
         return wait_for_terminate_and_warn(helper, pid, true);
 }
 
-static int write_mode(char **modes) {
+static int write_mode(const char **modes) {
         int r = 0;
-        char **mode;
+        const char **mode;
 
         STRV_FOREACH(mode, modes) {
                 int k;
@@ -199,8 +199,8 @@
         return r;
 }
 
-static int write_state(FILE **f, char **states) {
-        char **state;
+static int write_state(FILE **f, const char **states) {
+        const char **state;
         int r = 0;
 
         STRV_FOREACH(state, states) {
