--- src/libelogind/sd-bus/bus-error.c	2015-04-20 21:59:54.600440290 +0200
+++ src/libelogind/sd-bus/bus-error.c	2017-03-04 12:29:27.793186294 +0100
@@ -378,7 +378,12 @@
                         return;
 
                 errno = 0;
+#if defined(__GLIBC__)
                 x = strerror_r(error, m, k);
+#else
+		/* int strerror_r (int, char *, size_t); */
+		x = strerror_r(error, m, k) < 0 ? strdup("strerror_r() failed") : m;
+#endif
                 if (errno == ERANGE || strlen(x) >= k - 1) {
                         free(m);
                         k *= 2;
