--- src/test/test-parse-util.c
+++ src/test/test-parse-util.c
@@ -526,7 +526,10 @@ static void test_safe_atod(void) {
                 assert_se(r == -EINVAL);
 
                 errno = 0;
+/// elogind supports musl_libc, and their strtod doesn't seem to use the set locale.
+#if defined(__GLIBC__)
                 assert_se(fabs(strtod("0,5", &e) - 0.5) < 0.00001);
+#endif // __GLIBC__
         }
 
         /* And check again, reset */

