commit 2b09e13f7aec13105380d9d9fddc96ae51911b0c
Author: Daniel Kolesa <daniel@octaforge.org>
Date:   Sat Dec 18 02:53:26 2021 +0100

    reenable polkit

diff --git a/meson.build b/meson.build
index 72825a2..b14cb64 100644
--- a/meson.build
+++ b/meson.build
@@ -1155,15 +1155,6 @@ if want_polkit != 'false' and not skip_deps
                 message('Old polkit detected, will install pkla files')
                 install_polkit_pkla = true
         endif
-#if 1 /// Disable polkit completely if libpolkit is not there. See elogind issue #167
-        if not libpolkit.found()
-                if want_polkit != 'auto'
-                        error('Polkit requested but libpolkit was not found.')
-                endif
-                install_polkit = false
-                want_polkit    = false
-        endif
-#endif // 1
 endif
 conf.set10('ENABLE_POLKIT', install_polkit)
 

