diff --git ./src/basic/cgroup-util.c ./src/basic/cgroup-util.c
index 9235361b6..11102790e 100644
--- ./src/basic/cgroup-util.c
+++ ./src/basic/cgroup-util.c
@@ -824,7 +824,7 @@ int cg_attach(const char *controller, const char *path, pid_t pid) {
 
         xsprintf(c, PID_FMT "\n", pid);
 
-        r = write_string_file(fs, c, WRITE_STRING_FILE_DISABLE_BUFFER);
+        r = write_string_file(fs, c, 0);
         if (r < 0)
                 return r;
 
