diff --git ./src/basic/fileio.c ./src/basic/fileio.c
index e981cd024..ffcbf7309 100644
--- ./src/basic/fileio.c
+++ ./src/basic/fileio.c
@@ -156,7 +156,7 @@ int write_string_file_ts(
                         goto fail;
                 }
 
-                f = fdopen(fd, "w");
+                f = fdopen(fd, "we");
                 if (!f) {
                         r = -errno;
                         safe_close(fd);
