--- a/backend/cfg-obj.c
+++ b/backend/cfg-obj.c
@@ -1026,7 +1026,7 @@ _cfg_register_scsi_entry (const char *string)
       char *vendor = NULL;
       char *model  = NULL;
 
-      sscanf (string, "%*s %as %as", &vendor, &model);
+      sscanf (string, "%*s %ms %ms", &vendor, &model);
 
       if (list_append (_cfg->seen[CFG_KEY_SCSI], info))
         {
@@ -1108,7 +1108,7 @@ _cfg_register_interpreter_entry (const char *string)
       char *library  = NULL;
       char *firmware = NULL;
 
-      sscanf (string, "%*s %*s %x %x %as %as",
+      sscanf (string, "%*s %*s %x %x %ms %ms",
               &vendor, &product, &library, &firmware);
 
       if (library && _cfg_have_interpreter (library, firmware)
