Index: stremio-shell-4.4.159/mpv.cpp
===================================================================
--- stremio-shell-4.4.159.orig/mpv.cpp
+++ stremio-shell-4.4.159/mpv.cpp
@@ -65,7 +65,7 @@ class MpvRenderer : public QQuickFramebu
         // init mpv_gl:
         if (!obj->mpv_gl)
         {
-            mpv_opengl_init_params gl_init_params{get_proc_address_mpv, nullptr, nullptr};
+            mpv_opengl_init_params gl_init_params{get_proc_address_mpv, nullptr};
             mpv_render_param params[]{
                 {MPV_RENDER_PARAM_API_TYPE, const_cast<char *>(MPV_RENDER_API_TYPE_OPENGL)},
                 {MPV_RENDER_PARAM_OPENGL_INIT_PARAMS, &gl_init_params},
Index: stremio-shell-4.4.159/deps/libmpv/include/mpv/qthelper.hpp
===================================================================
--- stremio-shell-4.4.159.orig/deps/libmpv/include/mpv/qthelper.hpp
+++ stremio-shell-4.4.159/deps/libmpv/include/mpv/qthelper.hpp
@@ -18,7 +18,7 @@
 
 #include <mpv/client.h>
 
-#if !MPV_ENABLE_DEPRECATED
+#if 0
 #error "This helper is deprecated. Copy it into your project instead."
 #else
 
