--- ./screenfetch-dev.orig	2015-07-14 08:32:02.000000000 -0400
+++ ./screenfetch-dev	2015-12-30 17:39:02.668091221 -0500
@@ -500,6 +500,11 @@
 					distro="Viperr"
 					distro_codename=null
 					;;
+				"Void"|"VoidLinux")
+					distro="Void Linux"
+					distro_codename=""
+					distro_release=""
+					;;
 				*)
 					;;
 			esac
@@ -558,7 +563,6 @@
 					fi
 
 					# Hotfixes
-					[[ "${distro}" == "Void" || "${distro}" == "void" ]] && distro="Void"
 					[[ "${distro}" == "evolveos" ]] && distro="Evolve OS"
 					[[ "${distro}" == "antergos" || "${distro}" == "Antergos" ]] && distro="Antergos"
 					[[ "${distro}" == "logos" || "${distro}" == "Logos" ]] && distro="Logos"
@@ -757,7 +761,7 @@
 		trisquel) distro="Trisquel";;
 		parabolagnu|parabolagnu/linux-libre|'parabola gnu/linux-libre'|parabola) distro="Parabola GNU/Linux-libre" ;;
 		viperr) distro="Viperr" ;;
-		void) distro="Void" ;;
+		void*linux) distro="Void Linux" ;;
 		kaos) distro="KaOS";;
 		linuxdeepin) distro="LinuxDeepin" ;;
 		deepin) distro="Deepin" ;;
@@ -846,7 +850,7 @@
 		'Gentoo'|'Sabayon'|'Funtoo') pkgs=$(ls -d /var/db/pkg/*/* | wc -l) ;;
 		'NixOS') pkgs=$(ls -d -1 /nix/store/*/ | wc -l) ;;
 		'Fedora'|'Korora'|'BLAG'|'Chapeau'|'openSUSE'|'Red Hat Enterprise Linux'|'CentOS'|'Mandriva'|'Mandrake'|'Mageia'|'Viperr') pkgs=$(rpm -qa | wc -l) ;;
-		'Void') pkgs=$(xbps-query -l | wc -l) ;;
+		'Void Linux') pkgs=$(xbps-query -l | wc -l) ;;
 		'Evolve OS'|'Solus') pkgs=$(pisi list-installed | wc -l) ;;
 		'CRUX') pkgs=$(pkginfo -i | wc -l) ;;
 		'Lunar Linux') pkgs=$(lvu installed | wc -l) ;;
@@ -3655,7 +3659,7 @@
 "${c1}      \`.:+ooyyddddddddddddyyso+:.\`")
 		;;
 
-		"Void")
+		"Void Linux")
 			if [[ "$no_color" != "1" ]]; then
 				c1=$(getColor 'green')       # Dark Green
 				c2=$(getColor 'light green') # Light Green
@@ -3905,7 +3909,7 @@
 	case ${myascii} in
 		"Arch Linux - Old"|"Fedora"|"Korora"|"Chapeau"|"Mandriva"|"Mandrake"|"Chakra"|"ChromeOS"|"Sabayon"|"Slackware"|"Mac OS X"|"Trisquel"|"Kali Linux"|"Jiyuu Linux"|"Antergos"|"KaOS"|"Logos"|"gNewSense"|"NixOS") labelcolor=$(getColor 'light blue');;
 		"Arch Linux"|"Frugalware"|"Mageia"|"Deepin"|"CRUX") labelcolor=$(getColor 'light cyan');;
-		"Mint"|"LMDE"|"openSUSE"|"LinuxDeepin"|"DragonflyBSD"|"Manjaro"|"Manjaro-tree"|"Android"|"Void") labelcolor=$(getColor 'light green');;
+		"Mint"|"LMDE"|"openSUSE"|"LinuxDeepin"|"DragonflyBSD"|"Manjaro"|"Manjaro-tree"|"Android"|"Void Linux") labelcolor=$(getColor 'light green');;
 		"Ubuntu"|"FreeBSD"|"FreeBSD - Old"|"Debian"|"Raspbian"|"BSD"|"Red Hat Enterprise Linux"|"Peppermint"|"Cygwin"|"Fuduntu"|"NetBSD"|"Scientific Linux"|"DragonFlyBSD"|"BackTrack Linux") labelcolor=$(getColor 'light red');;
 		"CrunchBang"|"Solus"|"Viperr"|"elementary"*) labelcolor=$(getColor 'dark grey');;
 		"Gentoo"|"Parabola GNU/Linux-libre"|"Funtoo"|"Funtoo-text"|"BLAG") labelcolor=$(getColor 'light purple');;
