--- src/lib/util/strformat.h	2016-03-30 11:03:03.000000000 +0200
+++ src/lib/utils/strformat.h	2016-03-31 03:15:28.076259871 +0200
@@ -185,7 +185,8 @@
 #include <type_traits>
 #include <utility>
 
-#if defined(__GLIBCXX__) && (__GLIBCXX__ < 20150413)
+#if (defined(USE_INTERNAL_CBEGIN_CEND) && USE_INTERNAL_CBEGIN_CEND) || \
+    (defined(__GLIBCXX__) && (__GLIBCXX__ < 20150413))
 namespace std
 {
 template<class _Container>
