--- a/src/lib/netlist/plib/pexception.cpp	2017-01-24 23:44:30.000000000 +0100
+++ b/src/lib/netlist/plib/pexception.cpp	2017-01-25 06:48:09.658604647 +0100
@@ -10,7 +10,7 @@
 #include "pexception.h"
 #include "pfmtlog.h"
 
-#if (defined(__x86_64__) || defined(__i386__)) && defined(__linux__)
+#if (defined(__x86_64__) || defined(__i386__)) && defined(__linux__) && defined(__GLIBC__)
 #define HAS_FEENABLE_EXCEPT     (1)
 #else
 #define HAS_FEENABLE_EXCEPT     (0)
