--- config/autoconf.mk.in	2011-11-06 14:06:44.462920048 +0000
+++ config/autoconf.mk.in	2011-11-06 14:20:45.124169469 +0000
@@ -68,8 +68,8 @@
 mandir		= @mandir@
 idldir		= $(datadir)/idl/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)

-installdir	= $(libdir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
-sdkdir		= $(libdir)/$(MOZ_APP_NAME)-devel-$(MOZ_APP_VERSION)
+installdir	= $(libdir)/$(MOZ_APP_NAME)
+sdkdir		= $(libdir)/$(MOZ_APP_NAME)-devel

 DIST		= $(DEPTH)/dist
 LIBXUL_SDK      = @LIBXUL_SDK@
--- js/src/config/autoconf.mk.in	2011-11-06 14:06:45.586263727 +0000
+++ js/src/config/autoconf.mk.in	2011-11-06 14:21:01.770993669 +0000
@@ -61,8 +61,8 @@
 datadir		= @datadir@
 mandir		= @mandir@

-installdir	= $(libdir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
-sdkdir		= $(libdir)/$(MOZ_APP_NAME)-devel-$(MOZ_APP_VERSION)
+installdir	= $(libdir)/$(MOZ_APP_NAME)
+sdkdir		= $(libdir)/$(MOZ_APP_NAME)-devel

 TOP_DIST	= @TOP_DIST@
 ifneq (,$(filter /%,$(TOP_DIST)))
