diff -Nur mozilla-release.orig/browser/installer/Makefile.in mozilla-release/browser/installer/Makefile.in
--- mozilla-release.orig/browser/installer/Makefile.in	2011-09-30 08:45:28.858668017 +0000
+++ mozilla-release/browser/installer/Makefile.in	2011-09-30 08:51:01.548314685 +0000
@@ -43,6 +43,8 @@

 include $(DEPTH)/config/autoconf.mk

+MOZ_APP_VERSION=7.0
+
 include $(topsrcdir)/config/rules.mk

 MOZ_PKG_REMOVALS = $(srcdir)/removed-files.in
