--- Cargo.toml.orig	2019-01-29 11:31:40.149513234 +0100
+++ Cargo.toml	2019-01-29 11:31:54.473374380 +0100
@@ -59,3 +59,4 @@
 libudev-sys = { path = "dom/webauthn/libudev-sys" }
 serde_derive = { git = "https://github.com/servo/serde", branch = "deserialize_from_enums9" }
 winapi = { git = "https://github.com/froydnj/winapi-rs", branch = "aarch64" }
+libc = { git = "https://github.com/rust-lang/libc", rev = "914eba137b25ece7ade3986f1e34df9cf439af22" }
