this reverts https://hg.mozilla.org/mozilla-central/rev/9f8a59996994

--- toolkit/xre/nsAppRunner.cpp	2020-07-21 00:49:52.000000000 +0200
+++ toolkit/xre/nsAppRunner.cpp	2020-08-17 21:12:47.688908716 +0200
@@ -4741,8 +4741,6 @@
   CodeCoverageHandler::Init();
 #endif
 
-  NS_SetCurrentThreadName("MainThread");
-
   AUTO_BASE_PROFILER_LABEL("XREMain::XRE_main (around Gecko Profiler)", OTHER);
   AUTO_PROFILER_INIT;
   AUTO_PROFILER_LABEL("XREMain::XRE_main", OTHER);
