--- a/mozglue/misc/StackWalk.cpp	2017-04-11 04:13:21.000000000 +0200
+++ b/mozglue/misc/StackWalk.cpp	2017-11-29 15:23:07.218649970 +0100
@@ -41,7 +41,7 @@
 #define MOZ_STACKWALK_SUPPORTS_MACOSX 0
 #endif
 
-#if (defined(linux) && \
+#if defined(__GLIBC__) && (defined(linux) && \
      ((defined(__GNUC__) && (defined(__i386) || defined(PPC))) || \
       defined(HAVE__UNWIND_BACKTRACE)))
 #define MOZ_STACKWALK_SUPPORTS_LINUX 1
