--- media/webrtc/trunk/moz.build.orig	2019-01-16 14:48:37.361278044 +0100
+++ media/webrtc/trunk/moz.build	2019-01-16 14:49:21.637924835 +0100
@@ -242,6 +242,18 @@
         "/media/webrtc/trunk/webrtc/modules/video_processing/video_processing_neon_gn"
     ]
 
+
+if CONFIG["CPU_ARCH"] == "arm" and CONFIG["OS_TARGET"] == "Linux":
+
+    DIRS += [
+        "/media/webrtc/trunk/webrtc/common_audio/common_audio_neon_c_gn",
+        "/media/webrtc/trunk/webrtc/common_audio/common_audio_neon_gn",
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/isac_neon_gn",
+        "/media/webrtc/trunk/webrtc/modules/audio_processing/audio_processing_neon_c_gn",
+        "/media/webrtc/trunk/webrtc/modules/audio_processing/audio_processing_neon_gn",
+        "/media/webrtc/trunk/webrtc/modules/video_processing/video_processing_neon_gn"
+    ]
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "Linux":
 
     DIRS += [

--- media/webrtc/trunk/webrtc/modules/audio_processing/audio_processing_neon_c_gn/moz.build.orig	2019-01-16 14:58:54.713353133 +0100
+++ media/webrtc/trunk/webrtc/modules/audio_processing/audio_processing_neon_c_gn/moz.build	2019-01-08 19:56:53.000000000 +0100
@@ -71,7 +71,10 @@
     DEFINES["NO_TCMALLOC"] = True
     DEFINES["USE_NSS_CERTS"] = "1"
     DEFINES["USE_X11"] = "1"
+
+    if CONFIG["CPU_ARCH"] == "aarch64":
+        DEFINES["WEBRTC_ARCH_ARM64"] = True
+
-    DEFINES["WEBRTC_ARCH_ARM64"] = True
     DEFINES["WEBRTC_LINUX"] = True
 
     OS_LIBS += [

--- media/webrtc/trunk/webrtc/common_audio/common_audio_neon_c_gn/moz.build.orig	2019-01-16 15:16:36.947747240 +0100
+++ media/webrtc/trunk/webrtc/common_audio/common_audio_neon_c_gn/moz.build	2019-01-16 15:17:22.168379111 +0100
@@ -69,7 +69,10 @@
     DEFINES["NO_TCMALLOC"] = True
     DEFINES["USE_NSS_CERTS"] = "1"
     DEFINES["USE_X11"] = "1"
-    DEFINES["WEBRTC_ARCH_ARM64"] = True
+
+    if CONFIG["CPU_ARCH"] == "aarch64":
+        DEFINES["WEBRTC_ARCH_ARM64"] = True
+
     DEFINES["WEBRTC_LINUX"] = True
 
 if CONFIG["OS_TARGET"] == "NetBSD":

--- media/webrtc/trunk/webrtc/modules/audio_coding/isac_neon_gn/moz.build.orig	2019-01-16 15:28:38.302731713 +0100
+++ media/webrtc/trunk/webrtc/modules/audio_coding/isac_neon_gn/moz.build	2019-01-16 15:29:20.174378505 +0100
@@ -80,7 +80,10 @@
     DEFINES["NO_TCMALLOC"] = True
     DEFINES["USE_NSS_CERTS"] = "1"
     DEFINES["USE_X11"] = "1"
-    DEFINES["WEBRTC_ARCH_ARM64"] = True
+
+    if CONFIG["CPU_ARCH"] == "aarch64":
+        DEFINES["WEBRTC_ARCH_ARM64"] = True
+
     DEFINES["WEBRTC_LINUX"] = True
 
     OS_LIBS += [

--- media/webrtc/trunk/webrtc/modules/audio_processing/audio_processing_neon_gn/moz.build.orig	2019-01-16 15:31:28.281297859 +0100
+++ media/webrtc/trunk/webrtc/modules/audio_processing/audio_processing_neon_gn/moz.build	2019-01-16 15:31:06.724479707 +0100
@@ -77,7 +77,10 @@
     DEFINES["NO_TCMALLOC"] = True
     DEFINES["USE_NSS_CERTS"] = "1"
     DEFINES["USE_X11"] = "1"
-    DEFINES["WEBRTC_ARCH_ARM64"] = True
+
+    if CONFIG["CPU_ARCH"] == "aarch64":
+        DEFINES["WEBRTC_ARCH_ARM64"] = True
+
     DEFINES["WEBRTC_LINUX"] = True
 
     OS_LIBS += [

--- media/webrtc/trunk/webrtc/modules/video_processing/video_processing_neon_gn/moz.build.orig	2019-01-16 15:32:36.700720723 +0100
+++ media/webrtc/trunk/webrtc/modules/video_processing/video_processing_neon_gn/moz.build	2019-01-16 15:33:08.188455104 +0100
@@ -67,7 +67,10 @@
     DEFINES["NO_TCMALLOC"] = True
     DEFINES["USE_NSS_CERTS"] = "1"
     DEFINES["USE_X11"] = "1"
-    DEFINES["WEBRTC_ARCH_ARM64"] = True
+
+    if CONFIG["CPU_ARCH"] == "aarch64":
+        DEFINES["WEBRTC_ARCH_ARM64"] = True
+
     DEFINES["WEBRTC_LINUX"] = True
 
 if CONFIG["OS_TARGET"] == "NetBSD":

--- media/webrtc/trunk/webrtc/common_audio/common_audio_c_gn/moz.build.orig	2019-01-16 16:09:33.670752803 +0100
+++ media/webrtc/trunk/webrtc/common_audio/common_audio_c_gn/moz.build	2019-01-16 16:11:13.266888242 +0100
@@ -140,6 +140,7 @@
     DEFINES["WEBRTC_POSIX"] = True
     DEFINES["_FILE_OFFSET_BITS"] = "64"
 
+if CONFIG["OS_TARGET"] == "Linux" and CONFIG["CPU_ARCH"] != "arm":
     UNIFIED_SOURCES += [
         "/media/webrtc/trunk/webrtc/common_audio/signal_processing/complex_bit_reverse.c",
         "/media/webrtc/trunk/webrtc/common_audio/signal_processing/filter_ar_fast_q12.c",
--- media/webrtc/trunk/webrtc/modules/audio_coding/isac_fix_c_gn/moz.build.orig	2019-01-16 16:31:03.781907299 +0100
+++ media/webrtc/trunk/webrtc/modules/audio_coding/isac_fix_c_gn/moz.build	2019-01-16 16:30:44.106080812 +0100
@@ -142,6 +142,8 @@
         "rt"
     ]
 
+if CONFIG["OS_TARGET"] == "Linux" and CONFIG["CPU_ARCH"] != "arm":
+
     SOURCES += [
         "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/fix/source/pitch_filter_c.c"
     ]

--- media/webrtc/trunk/webrtc/system_wrappers/source/cpu_features.cc.orig	2019-01-16 16:52:42.676473795 +0100
+++ media/webrtc/trunk/webrtc/system_wrappers/source/cpu_features.cc	2019-01-16 16:54:22.595598235 +0100
@@ -67,7 +67,7 @@
   return 0;
 }

-#if !defined(ANDROID)
+#if !defined(ANDROID) && !defined(__linux__)
 #ifdef WEBRTC_ARCH_ARM_V7
 uint64_t WebRtc_GetCPUFeaturesARM(void) {
   return kCPUFeatureARMv7
