--- tools/profiler/core/platform.h.orig	2019-03-19 01:54:25.407952218 +0100
+++ tools/profiler/core/platform.h	2019-03-19 01:54:45.571767946 +0100
@@ -44,7 +44,7 @@
 
 // We need a definition of gettid(), but glibc doesn't provide a
 // wrapper for it.
-#if defined(__GLIBC__)
+#if defined(__linux__)
 #  include <unistd.h>
 #  include <sys/syscall.h>
 static inline pid_t gettid() { return (pid_t)syscall(SYS_gettid); }
