--- Cargo.toml.orig	2019-05-21 07:45:37.809198143 +0200
+++ Cargo.toml	2019-05-21 07:54:57.903418198 +0200
@@ -61,3 +61,4 @@
 serde_derive = { git = "https://github.com/servo/serde", branch = "deserialize_from_enums9" }
 winapi = { git = "https://github.com/froydnj/winapi-rs", branch = "aarch64" }
 cc = { git = "https://github.com/glandium/cc-rs", branch = "1.0.23-clang-cl-aarch64" }
+libc = { git = "https://github.com/rust-lang/libc", rev = "914eba137b25ece7ade3986f1e34df9cf439af22" }
