--- a/taskcluster/docker/firefox-snap/firefox.desktop
+++ b/taskcluster/docker/firefox-snap/firefox.desktop
@@ -154,11 +154,12 @@
 Terminal=false
 X-MultipleArgs=false
 Type=Application
-Icon=/browser/chrome/icons/default/default128.png
+Icon=firefox
 Categories=GNOME;GTK;Network;WebBrowser;
 MimeType=text/html;text/xml;application/xhtml+xml;application/xml;application/rss+xml;application/rdf+xml;image/gif;image/jpeg;image/png;x-scheme-handler/http;x-scheme-handler/https;x-scheme-handler/ftp;x-scheme-handler/chrome;video/webm;application/x-xpinstall;
 StartupNotify=true
 Actions=NewWindow;NewPrivateWindow;
+StartupWMClass=Firefox
 
 [Desktop Action NewWindow]
 Name=Open a New Window
