diff -ruN a/dom/media/gmp/GMPLoader.cpp b/dom/media/gmp/GMPLoader.cpp
--- a/dom/media/gmp/GMPLoader.cpp	2026-09-26 19:52:07.520679512 -0700
+++ b/dom/media/gmp/GMPLoader.cpp	2026-09-26 19:55:40.740319294 -0700
@@ -278,6 +278,27 @@
   }
   bool Start(const char* aLibPath) override {
     RegisterProfilerObserversForSandboxProfiler();
+
+#ifndef __GLIBC__
+    // For musl, load gcompat before possibly loading plugins distributed as
+    // glibc-linked binaries. Ignore failure to load.
+    PRLibSpec libSpec;
+    libSpec.type = PR_LibSpec_Pathname;
+    libSpec.value.pathname = "libgcompat.so.0";
+    PR_LoadLibraryWithFlags(libSpec, PR_LD_LAZY | PR_LD_GLOBAL);
+
+    // For libcs other than glibc, ensure that the emulation of the glibc
+    // dynamic linker is already loaded so that the plugin does not attempt to
+    // load it later. Ignore failure to load.
+#if defined(__i386__)
+    PR_LoadLibrary("ld-linux.so.2");
+#elif defined(__x86_64__)
+    PR_LoadLibrary("ld-linux-x86-64.so.2");
+#elif defined(__aarch64__)
+    PR_LoadLibrary("ld-linux-aarch64.so.2");
+#endif
+#endif
+
     mozilla::SetMediaPluginSandbox(aLibPath);
     return true;
   }
