--- a/netwerk/dns/PlatformDNSUnix.cpp
+++ b/netwerk/dns/PlatformDNSUnix.cpp
@@ -32,7 +32,7 @@
   rv = packet.FillBuffer(
       [&](unsigned char response[DNSPacket::MAX_SIZE]) -> int {
         int len = 0;
-#if defined(XP_LINUX)
+#if defined(XP_LINUX) && defined(__GLIBC__)
         len = res_nquery(&_res, host.get(), ns_c_in,
                          nsIDNSService::RESOLVE_TYPE_HTTPSSVC, response,
                          DNSPacket::MAX_SIZE);
