--- old-configure.in.orig
+++ old-configure.in
@@ -2818,6 +2818,22 @@
     _NON_GLOBAL_ACDEFINES="$_NON_GLOBAL_ACDEFINES NECKO_COOKIES"
 fi
 
+dnl ==================================
+dnl = Check sndio availability
+dnl ==================================
+
+MOZ_ARG_ENABLE_BOOL(sndio,
+[  --enable-sndio         Enable sndio support],
+   MOZ_SNDIO=1,
+   MOZ_SNDIO=)
+
+if test -n "$MOZ_SNDIO"; then
+    MOZ_SNDIO_LIBS="-lsndio"
+    AC_SUBST_LIST(MOZ_SNDIO_LIBS)
+fi
+
+AC_SUBST(MOZ_SNDIO)
+
 dnl ========================================================
 dnl =
 dnl = Maintainer debug option (no --enable equivalent)
--- media/libcubeb/src/moz.build.orig
+++ media/libcubeb/src/moz.build
@@ -44,11 +44,13 @@
     ]
     DEFINES['USE_JACK'] = True
 
-if CONFIG['OS_ARCH'] == 'OpenBSD':
+if CONFIG['MOZ_SNDIO']:
     SOURCES += [
         'cubeb_sndio.c',
     ]
     DEFINES['USE_SNDIO'] = True
+
+if CONFIG['OS_ARCH'] == 'OpenBSD':
     DEFINES['DISABLE_LIBSNDIO_DLOPEN'] = True
 
 if CONFIG['OS_TARGET'] == 'Darwin':
--- build/moz.configure/old.configure.orig
+++ build/moz.configure/old.configure
@@ -184,6 +184,7 @@
 @old_configure_options(
     '--cache-file',
     '--datadir',
+    '--enable-sndio',
     '--enable-cookies',
     '--enable-cpp-rtti',
     '--enable-crashreporter',
