From bb8c878129e872c8a4cd3534f85486ff6c000ac2 Mon Sep 17 00:00:00 2001
From: q66 <daniel@octaforge.org>
Date: Thu, 7 May 2020 19:18:18 +0200
Subject: [PATCH] fix build on non-vsx ppc64 platforms

---
 security/nss/lib/freebl/gcm.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git security/nss/lib/freebl/gcm.h security/nss/lib/freebl/gcm.h
index b2c30a7..54fb3d0 100644
--- security/nss/lib/freebl/gcm.h
+++ security/nss/lib/freebl/gcm.h
@@ -30,7 +30,7 @@
 #include <arm_neon.h>
 #endif
 
-#if defined(__powerpc64__) && !defined(NSS_DISABLE_ALTIVEC)
+#if defined(__powerpc64__) && defined(__VSX__) && !defined(NSS_DISABLE_ALTIVEC)
 #include "altivec-types.h"
 
 /* The ghash freebl test tries to use this in C++, and gcc defines conflict. */
-- 
2.25.1

