--- lib/i386-io-linux.h
+++ lib/i386-io-linux.h
@@ -6,11 +6,7 @@
  *	Can be freely distributed and used under the terms of the GNU GPL.
  */
 
-#ifdef __GLIBC__
 #include <sys/io.h>
-#else
-#include <asm/io.h>
-#endif
 
 static int
 intel_setup_io(struct pci_access *a UNUSED)
Only in pciutils-3.2.0: lib/i386-ports.o
Only in pciutils-3.2.0: lib/libpci.a
--- lib/pread.h
+++ lib/pread.h
@@ -12,7 +12,7 @@
  *  don't define it.
  */
 
-#if defined(__GLIBC__) && __GLIBC__ == 2 && __GLIBC_MINOR__ > 0
+#if 1 || defined(__GLIBC__) && __GLIBC__ == 2 && __GLIBC_MINOR__ > 0
 /* glibc 2.1 or newer -> pread/pwrite supported automatically */
 
 #elif defined(i386) && defined(__GLIBC__)
