diff --git a/board/samsung/smdk4x12/smc.c b/board/samsung/smdk4x12/smc.c
index fe3c1ed..b278193 100644
--- a/board/samsung/smdk4x12/smc.c
+++ b/board/samsung/smdk4x12/smc.c
@@ -22,6 +22,7 @@ static inline u32 exynos_smc(u32 cmd, u32 arg1, u32 arg2, u32 arg3)
 	register u32 reg3 __asm__("r3") = arg3;

 	__asm__ volatile (
+		".arch_extension sec\n"
 		"smc	0\n"
 		: "+r"(reg0), "+r"(reg1), "+r"(reg2), "+r"(reg3)

@@ -36,6 +37,7 @@ static inline u32 exynos_smc_read(u32 cmd)
 	register u32 reg1 __asm__("r1") = 0;

 	__asm__ volatile (
+		".arch_extension sec\n"
 		"smc	0\n"
 		: "+r"(reg0), "+r"(reg1)

