--- a/Makefile.am
+++ b/Makefile.am
@@ -43,14 +43,14 @@ conf_DATA = src/main.conf
 conf_DATA += profiles/input/input.conf
 conf_DATA += profiles/network/network.conf
 state_DATA =
+
+dbussystembusdir = $(DBUS_SYSTEMBUSDIR)
+dbussystembus_DATA = src/org.bluez.service
 endif
 
 if SYSTEMD
 systemdsystemunitdir = $(SYSTEMD_SYSTEMUNITDIR)
 systemdsystemunit_DATA = src/bluetooth.service
-
-dbussystembusdir = $(DBUS_SYSTEMBUSDIR)
-dbussystembus_DATA = src/org.bluez.service
 endif
 
 EXTRA_DIST += src/org.bluez.service
