--- a/Makefile.am	2025-10-08 10:50:46.724538765 +0100
+++ b/Makefile.am	2025-10-08 10:52:27.993435848 +0100
@@ -47,6 +47,9 @@
 conf_DATA += profiles/input/input.conf
 conf_DATA += profiles/network/network.conf
 state_DATA =
+
+dbussystembusdir = $(DBUS_SYSTEMBUSDIR)
+dbussystembus_DATA = src/org.bluez.service
 endif

 if SYSTEMD
@@ -55,9 +58,6 @@

 systemduserunitdir = $(SYSTEMD_USERUNITDIR)
 systemduserunit_DATA =
-
-dbussystembusdir = $(DBUS_SYSTEMBUSDIR)
-dbussystembus_DATA = src/org.bluez.service
 endif

 EXTRA_DIST += src/org.bluez.service
