From 2625ab38ae27754205d060f09e3a18f7648d5167 Mon Sep 17 00:00:00 2001
From: Doan Tran Cong Danh <congdanhqx@gmail.com>
Date: Thu, 3 Oct 2019 12:57:09 +0700
Subject: [PATCH 1/9] enable dbus without systemd

---
 Makefile.am | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 404e6a460..0884156fa 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -40,14 +40,14 @@ conf_DATA =
 
 statedir = $(localstatedir)/lib/bluetooth
 state_DATA =
+
+dbussystembusdir = $(DBUS_SYSTEMBUSDIR)
+dbussystembus_DATA = src/org.bluez.service
 endif
 
 if SYSTEMD
 systemdsystemunitdir = $(SYSTEMD_SYSTEMUNITDIR)
 systemdsystemunit_DATA = src/bluetooth.service
-
-dbussystembusdir = $(DBUS_SYSTEMBUSDIR)
-dbussystembus_DATA = src/org.bluez.service
 endif
 
 EXTRA_DIST += src/bluetooth.service.in src/org.bluez.service
-- 
2.23.0

