--- a/Makefile.obexd
+++ b/Makefile.obexd
@@ -5,9 +5,6 @@ if SYSTEMD
 systemduserunitdir = $(SYSTEMD_USERUNITDIR)
 systemduserunit_DATA = obexd/src/obex.service
 
-dbussessionbusdir = $(DBUS_SESSIONBUSDIR)
-dbussessionbus_DATA = obexd/src/org.bluez.obex.service
-
 obexd-add-service-symlink:
 	$(LN_S) -f obex.service $(DESTDIR)$(SYSTEMD_USERUNITDIR)/dbus-org.bluez.obex.service
 
@@ -18,6 +15,9 @@ obexd-add-service-symlink:
 obexd-remove-service-symlink:
 endif
 
+dbussessionbusdir = $(DBUS_SESSIONBUSDIR)
+dbussessionbus_DATA = obexd/src/org.bluez.obex.service
+
 obex_plugindir = $(libdir)/obex/plugins
 
 obexd_builtin_modules =
