From 32a425c3f9b42a55824c044a6b486c9c12e65a65 Mon Sep 17 00:00:00 2001
From: Doan Tran Cong Danh <congdanhqx@gmail.com>
Date: Thu, 3 Oct 2019 12:58:18 +0700
Subject: [PATCH 3/9] allow mesh on non-systemd system

---
 Makefile.mesh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.mesh b/Makefile.mesh
index 90979cbc0..9e6e2c2ca 100644
--- a/Makefile.mesh
+++ b/Makefile.mesh
@@ -2,11 +2,11 @@ if MESH
 
 if DATAFILES
 dbus_DATA += mesh/bluetooth-mesh.conf
+dbussystembus_DATA += mesh/org.bluez.mesh.service
 endif
 
 if SYSTEMD
 systemdsystemunit_DATA += mesh/bluetooth-mesh.service
-dbussystembus_DATA += mesh/org.bluez.mesh.service
 endif
 
 mesh_sources = mesh/mesh.h mesh/mesh.c \
-- 
2.23.0

