From fe1a76a03b9cbc28621d53680eb9ff4a2bd31778 Mon Sep 17 00:00:00 2001
From: Doan Tran Cong Danh <congdanhqx@gmail.com>
Date: Thu, 3 Oct 2019 16:31:01 +0700
Subject: [PATCH 7/9] include string.h

---
 android/client/if-hf-client.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/android/client/if-hf-client.c b/android/client/if-hf-client.c
index f701a814c..b44eb4291 100644
--- a/android/client/if-hf-client.c
+++ b/android/client/if-hf-client.c
@@ -16,6 +16,8 @@
  */
 
 #define _GNU_SOURCE
+#include <string.h>
+
 #include "if-main.h"
 #include "../hal-utils.h"
 
-- 
2.23.0

