From 05853e46e66868228468fc73d86f221b12edaef1 Mon Sep 17 00:00:00 2001
From: Doan Tran Cong Danh <congdanhqx@gmail.com>
Date: Fri, 4 Oct 2019 11:36:47 +0700
Subject: [PATCH 8/9] grant permission to bluetooth group

---
 src/bluetooth.conf | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/bluetooth.conf b/src/bluetooth.conf
index 8a1e25801..75dd7d670 100644
--- a/src/bluetooth.conf
+++ b/src/bluetooth.conf
@@ -22,6 +22,10 @@
     <allow send_interface="org.mpris.MediaPlayer2.Player"/>
   </policy>
 
+  <policy group="bluetooth">
+    <allow send_destination="org.bluez"/>
+  </policy>
+
   <policy context="default">
     <allow send_destination="org.bluez"/>
   </policy>
-- 
2.23.0

