--- tools/hid2hci.c.orig	2012-05-26 16:50:39.734627361 +0200
+++ tools/hid2hci.c	2012-05-26 16:50:52.229332774 +0200
@@ -293,7 +293,7 @@ int main(int argc, char *argv[])
 		goto exit;
 	}
 
-	snprintf(syspath, sizeof(syspath), "%s/%s", udev_get_sys_path(udev), devpath);
+	snprintf(syspath, sizeof(syspath), "%s/%s", "/sys", devpath);
 	udev_dev = udev_device_new_from_syspath(udev, syspath);
 	if (udev_dev == NULL) {
 		fprintf(stderr, "error: could not find '%s'\n", devpath);
