--- os-probes/mounted/common/90linux-distro.orig	2014-04-16 09:49:59.028554885 +0200
+++ os-probes/mounted/common/90linux-distro	2014-04-16 09:50:54.603990982 +0200
@@ -116,6 +116,13 @@
 	elif [ -e "$dir/sbin/pkgtool" ]; then
 		short="Slackware"
 		long="Slackware Linux"
+	elif [ -e "$dir/sbin/xbps-install" ]; then
+		short="Void"
+		if file "$dir/sbin/xbps-install"|grep -q 32-bit; then
+			long="Void Linux 32"
+		else
+			long="Void Linux 64"
+		fi
 	elif grep -qs OpenLinux "$dir/etc/issue"; then
 		short="Caldera"
 		long="Caldera OpenLinux"
