diff --git cargo_build.py cargo_build.py
index d78ceb9..301bf00 100644
--- cargo_build.py
+++ cargo_build.py
@@ -9,14 +9,21 @@ from pathlib import Path
 import shlex
 import subprocess
 import sys
+import os
 
 source_dir = Path(__file__).absolute().parent
 
 args = sys.argv[1:]
-binary_dir = "debug"
+if 'CROSS_BUILD' in os.environ:
+    binary_dir = os.environ['RUST_TARGET'] + "/debug"
+else:
+    binary_dir = "debug"
 
 if '--release' in args:
-    binary_dir = "release"
+    if 'CROSS_BUILD' in os.environ:
+        binary_dir = os.environ['RUST_TARGET'] + "/release"
+    else:
+        binary_dir = "release"
 
 # The file produced by Cargo will have a special name
 try:
