diff -ur setup.py setup.py
--- a/setup.py
+++ b/setup.py
@@ -69,7 +69,7 @@
                 "--message-format=json",
             ]
 
-            if platform.machine() in ("ppc64le", "ppc64", "powerpc"):
+            if platform.machine() in ("ppc64le", "ppc64", "powerpc", "ppcle", "ppc"):
                 cargo_args.extend(
                     ["--no-default-features", "--features=upload,log,human-panic"]
                 )
diff -ur src/target.rs src/target.rs
--- a/src/target.rs
+++ b/src/target.rs
@@ -40,6 +40,8 @@
     Armv7L,
     Powerpc64Le,
     Powerpc64,
+    PowerpcLe,
+    Powerpc,
     X86,
     X86_64,
     S390X,
@@ -52,6 +54,8 @@
             Arch::Armv7L => write!(f, "armv7l"),
             Arch::Powerpc64Le => write!(f, "ppc64le"),
             Arch::Powerpc64 => write!(f, "ppc64"),
+            Arch::PowerpcLe => write!(f, "ppcle"),
+            Arch::Powerpc => write!(f, "ppc"),
             Arch::X86 => write!(f, "i686"),
             Arch::X86_64 => write!(f, "x86_64"),
             Arch::S390X => write!(f, "s390x"),
@@ -67,6 +71,8 @@
             Arch::Armv7L,
             Arch::Powerpc64,
             Arch::Powerpc64Le,
+            Arch::PowerpcLe,
+            Arch::Powerpc,
             Arch::S390X,
             Arch::X86,
             Arch::X86_64,
@@ -123,6 +129,8 @@
             target_lexicon::Architecture::Aarch64(_) => Arch::Aarch64,
             target_lexicon::Architecture::Powerpc64 => Arch::Powerpc64,
             target_lexicon::Architecture::Powerpc64le => Arch::Powerpc64Le,
+            target_lexicon::Architecture::PowerpcLe => Arch::PowerpcLe,
+            target_lexicon::Architecture::Powerpc => Arch::Powerpc,
             target_lexicon::Architecture::S390x => Arch::S390X,
             unsupported => bail!("The architecture {} is not supported", unsupported),
         };
@@ -200,7 +208,7 @@
     /// Returns the default Manylinux tag for this architecture
     pub fn get_default_manylinux_tag(&self) -> PlatformTag {
         match self.arch {
-            Arch::Aarch64 | Arch::Armv7L | Arch::Powerpc64 | Arch::Powerpc64Le | Arch::S390X => {
+            Arch::Aarch64 | Arch::Armv7L | Arch::Powerpc64 | Arch::Powerpc64Le | Arch::PowerpcLe | Arch::Powerpc | Arch::S390X => {
                 PlatformTag::manylinux2014()
             }
             Arch::X86 | Arch::X86_64 => PlatformTag::manylinux2010(),
@@ -214,6 +222,8 @@
             Arch::Armv7L => 32,
             Arch::Powerpc64 => 64,
             Arch::Powerpc64Le => 64,
+            Arch::Powerpc => 32,
+            Arch::PowerpcLe => 32,
             Arch::X86 => 32,
             Arch::X86_64 => 64,
             Arch::S390X => 64,
