diff -u a/setup.py b/setup.py
--- a/setup.py
+++ b/setup.py
@@ -71,7 +71,7 @@
                 "--message-format=json",
             ]
 
-            if platform.machine() in ("ppc64le", "ppc64", "powerpc") or (
+            if platform.machine() in ("ppc64le", "ppc64", "powerpc", "ppcle", "ppc") or (
                 sys.platform == "win32" and platform.machine() == "ARM64"
             ):
                 cargo_args.extend(
diff -u a/src/auditwheel/policy.rs b/src/auditwheel/policy.rs
--- a/src/auditwheel/policy.rs
+++ b/src/auditwheel/policy.rs
@@ -100,6 +100,8 @@
                 Arch::Armv7L => "libc.musl-armv7.so.1",
                 Arch::Powerpc64Le => "libc.musl-ppc64le.so.1",
                 Arch::Powerpc64 => "", // musllinux doesn't support ppc64
+                Arch::PowerpcLe => "", // musllinux doesn't support ppcle
+                Arch::Powerpc => "", // musllinux doesn't support ppc
                 Arch::X86 => "libc.musl-x86.so.1",
                 Arch::X86_64 => "libc.musl-x86_64.so.1",
                 Arch::S390X => "libc.musl-s390x.so.1",
diff -u a/src/target.rs.orig b/src/target.rs
--- a/src/target.rs
+++ b/src/target.rs
@@ -52,6 +52,10 @@
     Powerpc64Le,
     #[serde(alias = "ppc64")]
     Powerpc64,
+    #[serde(alias = "ppcle")]
+    PowerpcLe,
+    #[serde(alias = "ppc")]
+    Powerpc,
     #[serde(alias = "i686")]
     X86,
     X86_64,
@@ -65,6 +69,8 @@
             Arch::Armv7L => write!(f, "armv7l"),
             Arch::Powerpc64Le => write!(f, "ppc64le"),
             Arch::Powerpc64 => write!(f, "ppc64"),
+            Arch::PowerpcLe => write!(f, "ppcle"),
+            Arch::Powerpc => write!(f, "ppc"),
             Arch::X86 => write!(f, "i686"),
             Arch::X86_64 => write!(f, "x86_64"),
             Arch::S390X => write!(f, "s390x"),
@@ -80,6 +86,8 @@
             Arch::Armv7L,
             Arch::Powerpc64,
             Arch::Powerpc64Le,
+            Arch::PowerpcLe,
+            Arch::Powerpc,
             Arch::S390X,
             Arch::X86,
             Arch::X86_64,
@@ -154,6 +162,8 @@
             target_lexicon::Architecture::Aarch64(_) => Arch::Aarch64,
             target_lexicon::Architecture::Powerpc64 => Arch::Powerpc64,
             target_lexicon::Architecture::Powerpc64le => Arch::Powerpc64Le,
+            target_lexicon::Architecture::PowerpcLe => Arch::PowerpcLe,
+            target_lexicon::Architecture::Powerpc => Arch::Powerpc,
             target_lexicon::Architecture::S390x => Arch::S390X,
             unsupported => bail!("The architecture {} is not supported", unsupported),
         };
@@ -303,6 +313,8 @@
             Arch::Armv7L => "armv7l",
             Arch::Powerpc64Le => "powerpc64le",
             Arch::Powerpc64 => "powerpc64",
+            Arch::PowerpcLe => "powerpcle",
+            Arch::Powerpc => "powerpc",
             Arch::X86 => "i386",
             Arch::X86_64 => "x86_64",
             Arch::S390X => "s390x",
@@ -327,7 +339,7 @@
     /// Returns the oldest possible Manylinux tag for this architecture
     pub fn get_minimum_manylinux_tag(&self) -> PlatformTag {
         match self.arch {
-            Arch::Aarch64 | Arch::Armv7L | Arch::Powerpc64 | Arch::Powerpc64Le | Arch::S390X => {
+            Arch::Aarch64 | Arch::Armv7L | Arch::Powerpc64 | Arch::Powerpc64Le | Arch::PowerpcLe | Arch::Powerpc | Arch::S390X => {
                 PlatformTag::manylinux2014()
             }
             Arch::X86 | Arch::X86_64 => PlatformTag::manylinux2010(),
@@ -341,6 +353,8 @@
             Arch::Armv7L => 32,
             Arch::Powerpc64 => 64,
             Arch::Powerpc64Le => 64,
+            Arch::Powerpc => 32,
+            Arch::PowerpcLe => 32,
             Arch::X86 => 32,
             Arch::X86_64 => 64,
             Arch::S390X => 64,
