--- setup.py	2020-04-30 22:36:01.000000000 +1000
+++ setup.py	2020-05-20 14:31:40.404994499 +1000
@@ -56,9 +56,9 @@
                     "(https://www.rust-lang.org/tools/install) and try again"
                 )
             subprocess.check_call(
-                ["cargo", "rustc", "--bin", "maturin", "--", "-C", "link-arg=-s"]
+                ["cargo", "rustc", "--bin", "maturin", "--release", "--target", os.environ["RUST_TARGET"], "--", "-C", "link-arg=-s", "-C", "linker=" + os.environ["CC"].split(" ", 1)[0]]
             )
-            source = os.path.join(source_dir, "target", "debug", executable_name)
+            source = os.path.join(source_dir, "target", os.environ["RUST_TARGET"], "release", executable_name)
         # run this after trying to build with cargo (as otherwise this leaves
         # venv in a bad state: https://github.com/benfred/py-spy/issues/69)
         install.run(self)
