--- archivers/wad.c.orig	2011-07-20 11:19:01.409198964 +0200
+++ archivers/wad.c	2011-07-20 11:19:31.465196724 +0200
@@ -245,9 +245,6 @@ static int wad_load_entries(const char *
     PHYSFS_uint32 fileCount;
     PHYSFS_uint32 directoryOffset;
     WADentry *entry;
-    char lastDirectory[9];
-
-    lastDirectory[8] = 0; /* Make sure lastDirectory stays null-terminated. */
 
     BAIL_IF_MACRO(!wad_open(name, forWriting, &fh, &fileCount,&directoryOffset), NULL, 0);
     info->entryCount = fileCount;
