--- a/sbuild/sbuild-chroot-config.cc
+++ b/sbuild/sbuild-chroot-config.cc
@@ -31,7 +31,7 @@
 #include <cstdlib>
 #include <cstring>
 
-#include <boost/filesystem/operations.hpp>
+#include <boost/filesystem.hpp>
 
 #include <sys/types.h>
 #include <sys/stat.h>
--- a/sbuild/sbuild-run-parts.cc
+++ b/sbuild/sbuild-run-parts.cc
@@ -29,6 +29,7 @@
 #include <syslog.h>
 
 #include <boost/format.hpp>
+#include <boost/filesystem.hpp>
 #include <boost/filesystem/operations.hpp>
 
 using boost::format;
--- a/sbuild/sbuild-util.cc
+++ b/sbuild/sbuild-util.cc
@@ -35,7 +35,7 @@
 #include <time.h>
 #endif
 
-#include <boost/filesystem/convenience.hpp>
+#include <boost/filesystem.hpp>
 
 using namespace sbuild;
 
--- a/test/sbuild-run-parts.cc
+++ b/test/sbuild-run-parts.cc
@@ -23,7 +23,7 @@
 #include <iostream>
 #include <sstream>
 
-#include <boost/filesystem/operations.hpp>
+#include <boost/filesystem.hpp>
 
 #include <cppunit/extensions/HelperMacros.h>
 
