From 6ffdd152d3fb07921c02e739ce5b47d1e177e7e6 Mon Sep 17 00:00:00 2001
From: Andrea Brancaleoni <miwaxe@gmail.com>
Date: Mon, 24 Aug 2015 18:39:16 +0200
Subject: [PATCH] getent

---
 etc/setup.d/20nssdatabases | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git etc/setup.d/20nssdatabases etc/setup.d/20nssdatabases
index 6f14c81..697bb1e 100755
--- a/etc/setup.d/20nssdatabases
+++ b/etc/setup.d/20nssdatabases
@@ -29,7 +29,7 @@ set -e
 dup_nss()
 {
     info "Copying $1 database to $2"
-    getent "$1" > "$2"
+    getent "$1" > "$2" 2>/dev/null || cat "/etc/$1" > "$2"
 }

 if [ $STAGE = "setup-start" ] || [ $STAGE = "setup-recover" ]; then
--
2.5.0

