From 73936a423227aa78b7682bdd3edc20643763807b Mon Sep 17 00:00:00 2001
From: Andrea Brancaleoni <abc@pompel.me>
Date: Wed, 19 Sep 2018 11:22:47 +0200
Subject: [PATCH] musl

---
 sbuild/sbuild-feature.cc | 2 ++
 1 file changed, 2 insertions(+)

diff --git sbuild/sbuild-feature.cc sbuild/sbuild-feature.cc
index 5ac4725..9902db3 100644
--- sbuild/sbuild-feature.cc
+++ sbuild/sbuild-feature.cc
@@ -20,6 +20,8 @@
 
 #include <iostream>
 
+#include <libintl.h>
+
 #include "sbuild-feature.h"
 
 using namespace sbuild;
-- 
2.19.0

