--- a/data/lxdm.conf.in	2013-08-30 17:56:16.097006000 +0200
+++ b/data/lxdm.conf.in	2013-09-04 01:54:12.942117085 +0200
@@ -23,7 +23,7 @@
 
 [server]
 ## arg used to start xserver, not fully function
-# arg=/usr/bin/X -background vt1
+arg=/usr/bin/X -background vt1
 # uncomment this if you really want xserver listen to tcp
 # tcp_listen=1
 # uncoment this if you want reset the xserver after logou
@@ -34,7 +34,7 @@
 gtk_theme=Clearlooks
 
 ## background of the greeter
-bg=/usr/share/backgrounds/default.png
+#bg=/usr/share/void-artwork/splash.png
 
 ## if show bottom pane
 bottom_pane=1
--- a/data/lxdm.in	2013-08-30 17:56:16.097006000 +0200
+++ b/data/lxdm.in	2013-09-04 01:57:56.950799946 +0200
@@ -1,21 +1,13 @@
 #!/bin/sh
 
-[ -f /etc/sysconfig/i18n ] && . /etc/sysconfig/i18n
-
-if [ -z "$LANG" -a -e /etc/sysconfig/language ]; then
-	. /etc/sysconfig/language
-	if [ -n "$RC_LANG"]; then
-		LANG=$RC_LANG
-	fi
+if [ -r /etc/profile.d/locale.sh ]; then
+	. /etc/profile.d/locale.sh
 fi
 
 if [ -n "$LANG" ]; then
 	export LANG
 fi
 
-[ -f /etc/sysconfig/desktop ] && . /etc/sysconfig/desktop
-[ -f /etc/sysconfig/windowmanager ] && . /etc/sysconfig/windowmanager
-
 if [ -n "$DEFAULT_WM" ]; then
 	PREFERRED=$DEFAULT_WM
 fi
--- a/data/Xsession	2013-08-30 17:56:16.097006000 +0200
+++ b/data/Xsession	2013-09-04 01:58:19.941348762 +0200
@@ -9,6 +9,8 @@
 	LXSESSION=/usr/bin/startlxde
 fi
 
+[ -f /etc/profile ] && . /etc/profile
+[ -f ~/.profile ] && . ~/.profile
 [ -f /etc/xprofile ] && . /etc/xprofile
 [ -f ~/.xprofile ] && . ~/.xprofile
 
