From e46c33735f81ca5607247fd0690ebbe9f1565708 Mon Sep 17 00:00:00 2001
From: Piotr Wójcik <chocimier@tlen.pl>
Date: Tue, 8 Oct 2019 19:32:15 +0200
Subject: [PATCH] tweak default PATH


diff --git a/src/lxdm.c b/src/lxdm.c
index a37f051..8a0e6c6 100644
--- a/src/lxdm.c
+++ b/src/lxdm.c
@@ -1399,7 +1399,7 @@ void lxdm_do_login(struct passwd *pw, char *session, char *lang, char *option)
 	if( G_UNLIKELY(path) && path[0] ) /* if PATH is specified in config file */
 		env=g_environ_setenv(env, "PATH", path, TRUE); /* override current $PATH with config value */
 	else /* don't use the global env, they are bad for user */
-		env=g_environ_setenv(env, "PATH", "/usr/local/bin:/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/sbin", TRUE); /* set proper default */
+		env=g_environ_setenv(env, "PATH", "/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/sbin:/sbin:/bin", TRUE); /* set proper default */
 	g_free(path);
 	/* optionally override $LANG, $LC_MESSAGES, and $LANGUAGE */
 	if( lang && lang[0] )
-- 
2.23.0

