--- data/lxdm.conf.in.orig	2012-05-02 10:53:30.903273165 +0200
+++ data/lxdm.conf.in	2012-05-02 10:54:15.357223046 +0200
@@ -16,7 +16,7 @@
 # xauth_path=/tmp
 
 ## greeter used to welcome the user
-greeter=@FULL_LIBEXECDIR@/lxdm-greeter-gtk
+greeter=/usr/libexec/lxdm-greeter-gtk
 
 [server]
 ## arg used to start xserver, not fully function
@@ -27,7 +27,7 @@ greeter=@FULL_LIBEXECDIR@/lxdm-greeter-g
 gtk_theme=Clearlooks
 
 ## background of the greeter
-bg=/usr/share/backgrounds/default.png
+bg=/usr/share/void-artwork/splash.png
 
 ## if show bottom pane
 bottom_pane=1
@@ -36,7 +36,7 @@ bottom_pane=1
 lang=1
 
 ## if show keyboard layout select control
-keyboard=0
+keyboard=0
 
 ## the theme of greeter
 theme=Industrial
