--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -267,16 +267,18 @@
     FIND_PROGRAM(LRELEASE lrelease-qt6 HINTS "${_qt_bin_dir}")
 ENDIF ()
 FILE(GLOB cppfiles *.cc)
-ADD_CUSTOM_TARGET(apvlv_lupdate
-        COMMAND ${LUPDATE}
-        -locations none -target-language zh_CN
-        ${cppfiles} -ts ${CMAKE_SOURCE_DIR}/zh_CN.ts)
-ADD_CUSTOM_TARGET(apvlv_lrelease
-        COMMAND ${LRELEASE}
-        ${CMAKE_SOURCE_DIR}/zh_CN.ts
-        -qm ${CMAKE_SOURCE_DIR}/share/doc/apvlv/translations/zh_CN.qm)
-ADD_DEPENDENCIES(apvlv_lrelease apvlv_lupdate)
-ADD_DEPENDENCIES(apvlv apvlv_lrelease)
+IF (NOT LUPDATE STREQUAL "LUPDATE-NOTFOUND" AND NOT LRELEASE STREQUAL "LRELEASE-NOTFOUND")
+    ADD_CUSTOM_TARGET(apvlv_lupdate
+            COMMAND ${LUPDATE}
+            -locations none -target-language zh_CN
+            ${cppfiles} -ts ${CMAKE_SOURCE_DIR}/zh_CN.ts)
+    ADD_CUSTOM_TARGET(apvlv_lrelease
+            COMMAND ${LRELEASE}
+            ${CMAKE_SOURCE_DIR}/zh_CN.ts
+            -qm ${CMAKE_SOURCE_DIR}/share/doc/apvlv/translations/zh_CN.qm)
+    ADD_DEPENDENCIES(apvlv_lrelease apvlv_lupdate)
+    ADD_DEPENDENCIES(apvlv apvlv_lrelease)
+ENDIF ()
 
 INSTALL(TARGETS apvlv DESTINATION bin)
