--- bin/yarn.js.orig	2018-09-18 14:33:30.318123244 -0700
+++ bin/yarn.js	2018-09-18 14:35:10.715515522 -0700
@@ -12,7 +12,7 @@
   process.exit(1); // eslint-disable-line no-process-exit
 } else {
   try {
-    require(__dirname + '/../lib/v8-compile-cache.js');
+    require('/usr/lib/yarn/v8-compile-cache.js');
   } catch (err) {
     // We don't have/need this on legacy builds and dev builds
   }
@@ -21,7 +21,7 @@
   // `require.main === module` check inside `cli/index.js` will always
   // be truthy when built with webpack :(
   // `lib/cli` may be `lib/cli/index.js` or `lib/cli.js` depending on the build.
-  var cli = require(__dirname + '/../lib/cli');
+  var cli = require('/usr/lib/yarn/cli');
   if (!cli.autoRun) {
     cli.default().catch(function(error) {
       console.error(error.stack || error.message || error);
