--- src/applet.c.orig	2011-03-01 00:01:37.106389077 +0100
+++ src/applet.c	2011-03-01 00:01:58.380344793 +0100
@@ -626,12 +626,10 @@ applet_do_notify (NMApplet *applet,
 	escaped = utils_escape_notify_message (message);
 	notify = notify_notification_new (summary,
 	                                  escaped,
-	                                  icon ? icon : GTK_STOCK_NETWORK,
-	                                  NULL);
+	                                  icon ? icon : GTK_STOCK_NETWORK);
 	g_free (escaped);
 	applet->notification = notify;
 
-	notify_notification_attach_to_status_icon (notify, applet->status_icon);
 	notify_notification_set_urgency (notify, urgency);
 	notify_notification_set_timeout (notify, NOTIFY_EXPIRES_DEFAULT);
 
