diff --git a/src/wireless-security/eap-method-leap.c b/src/wireless-security/eap-method-leap.c
index 93bb32d..3a1545e 100644
--- src/wireless-security/eap-method-leap.c
+++ src/wireless-security/eap-method-leap.c
@@ -106,7 +106,6 @@ fill_connection (EAPMethod *parent, NMConnection *connection)
 	if (method->new_connection) {
 		g_object_set (s_8021x,
 		              NM_SETTING_802_1X_PASSWORD_FLAGS, NM_SETTING_SECRET_FLAG_AGENT_OWNED,
-		              NM_SETTING_802_1X_SYSTEM_CA_CERTS, TRUE,
 		              NULL);
 	}
 }
diff --git src/wireless-security/eap-method-simple.c src/wireless-security/eap-method-simple.c
index 43063dd..6463c46 100644
--- src/wireless-security/eap-method-simple.c
+++ src/wireless-security/eap-method-simple.c
@@ -169,7 +169,6 @@ fill_connection (EAPMethod *parent, NMConnection *connection)
 	if (method->new_connection && (not_saved == FALSE)) {
 		g_object_set (s_8021x,
 		              NM_SETTING_802_1X_PASSWORD_FLAGS, NM_SETTING_SECRET_FLAG_AGENT_OWNED,
-		              NM_SETTING_802_1X_SYSTEM_CA_CERTS, TRUE,
 		              NULL);
 	}
 }
diff --git src/wireless-security/eap-method-tls.c src/wireless-security/eap-method-tls.c
index da02873..1d3222e 100644
--- src/wireless-security/eap-method-tls.c
+++ src/wireless-security/eap-method-tls.c
@@ -170,7 +170,6 @@ fill_connection (EAPMethod *parent, NMConnection *connection)
 	if (method->new_connection) {
 		g_object_set (s_8021x,
 		              secret_flag_prop, NM_SETTING_SECRET_FLAG_AGENT_OWNED,
-		              NM_SETTING_802_1X_SYSTEM_CA_CERTS, TRUE,
 		              NULL);
 	}
 
