diff --git a/.xbps-falco/userspace/libsinsp/procfs_utils.h b/.xbps-falco/userspace/libsinsp/procfs_utils.h
index f2f55e9df..bf51cd94e 100644
--- a/.xbps-falco/userspace/libsinsp/procfs_utils.h
+++ b/.xbps-falco/userspace/libsinsp/procfs_utils.h
@@ -2,6 +2,7 @@
 
 #include <istream>
 #include <string>
+#include <cstdint>
 
 namespace libsinsp {
 namespace procfs_utils {
diff --git a/.xbps-falco/userspace/libsinsp/scap_open_exception.h b/.xbps-falco/userspace/libsinsp/scap_open_exception.h
index 8c86687a6..3401c221f 100644
--- a/.xbps-falco/userspace/libsinsp/scap_open_exception.h
+++ b/.xbps-falco/userspace/libsinsp/scap_open_exception.h
@@ -17,6 +17,7 @@ limitations under the License.
 */
 #pragma once
 
+#include <cstdint>
 #include "sinsp_exception.h"
 
 /*!
