--- userspace/libsinsp/cgroup_limits.cpp	2019-12-09 23:06:46.000000000 +0100
+++ userspace/libsinsp/cgroup_limits.cpp	2019-12-20 01:14:18.816734272 +0100
@@ -9,7 +9,7 @@
 // This reports extremely large values (e.g. almost-but-not-quite 9EiB as set by k8s) as unlimited.
 // Note: we use the same maximum value for cpu shares/quotas as well; the typical values are much lower
 // and so should never exceed CGROUP_VAL_MAX either
-constexpr const int64_t CGROUP_VAL_MAX = (1UL << 42u) - 1;
+constexpr const int64_t CGROUP_VAL_MAX = (1ULL << 42u) - 1;
 
 /**
  * \brief Read a single int64_t value from cgroupfs
