diff --git a/.xbps-falco/userspace/libsinsp/CMakeLists.txt.orig b/.xbps-falco/userspace/libsinsp/CMakeLists.txt
index 6104603..7c0d8eb 100644
--- a/.xbps-falco/userspace/libsinsp/CMakeLists.txt.orig
+++ b/.xbps-falco/userspace/libsinsp/CMakeLists.txt
@@ -29,7 +29,7 @@ if(NOT MSVC)
 endif()
 
 include(ExternalProject)
-
+find_library(ZSTD_LIB NAMES zstd)
 include(jsoncpp)
 include(zlib)
 
@@ -255,6 +255,7 @@ if(NOT WIN32)
 						"${GPR_LIB}"
 						"${PROTOBUF_LIB}"
 						"${CARES_LIB}"
+						"${ZSTD_LIB}"
 			)
 
 			if(NOT MUSL_OPTIMIZED_BUILD)
@@ -280,6 +281,11 @@ if(NOT WIN32)
 			list(APPEND SINSP_PKGCONFIG_LIBRARIES stdc++fs)
 		endif()
 	endif()
+
+	list(APPEND VOID_SYSTEM_LIBRARIES
+		absl_log_internal_message absl_log_internal_nullguard
+		absl_hash absl_synchronization absl_cord absl_cordz_info absl_cordz_functions)
+	target_link_libraries(sinsp INTERFACE ${VOID_SYSTEM_LIBRARIES})
 endif() # NOT WIN32
 
 if(APPLE)
