--- a/.xbps-falco/userspace/libsinsp/CMakeLists.txt
+++ b/.xbps-falco/userspace/libsinsp/CMakeLists.txt
@@ -38,6 +38,7 @@ include(ExternalProject)
 
 if(NOT WIN32 AND NOT APPLE)
 	include(b64)
+	find_library(ZSTD_LIB NAMES zstd)
 endif()
 
 include(jsoncpp)
@@ -251,7 +252,8 @@ if(NOT WIN32)
 				"${PROTOBUF_LIB}"
 				"${CARES_LIB}"
 				"${JQ_LIB}"
-				"${B64_LIB}")
+				"${B64_LIB}"
+				"${ZSTD_LIB}")
 
 			if(NOT MUSL_OPTIMIZED_BUILD)
 				list(APPEND SINSP_LIBRARIES
@@ -301,7 +303,9 @@ else()
 	endif()
 endif() # NOT WIN32
 
-
+LIST(APPEND SINSP_LIBRARIES
+     absl_log_internal_message absl_log_internal_nullguard
+     absl_hash absl_synchronization absl_cord absl_cordz_info absl_cordz_functions)
 target_link_libraries(sinsp ${SINSP_LIBRARIES})
 
 if(APPLE)
