--- userspace/libsinsp/utils.h	2015-11-06 23:42:21.000000000 +0100
+++ userspace/libsinsp/utils.h	2015-11-07 17:17:34.171798332 +0100
@@ -79,7 +79,7 @@
 
 	static uint64_t get_current_time_ns();
 
-#ifndef _WIN32
+#if !defined(_WIN32) && defined(__GLIBC__)
 	//
 	// Print the call stack
 	//
--- userspace/libsinsp/utils.cpp.orig
+++ userspace/libsinsp/utils.cpp
@@ -21,7 +21,7 @@
 #include <limits.h>
 #include <stdlib.h>
 #include <sys/time.h>
-#ifndef CYGWING_AGENT
+#ifndef __GLIBC__
 #include <execinfo.h>
 #endif
 #include <unistd.h>
@@ -770,7 +770,7 @@
 #endif
 }
 
-#ifndef CYGWING_AGENT
+#ifndef __GLIBC__
 #ifndef _WIN32
 void sinsp_utils::bt(void)
 {
@@ -893,7 +893,7 @@
 
 	return 0;
 }
-#endif // _WIN32
+#endif // !defined(_WIN32) && defined(__GLIBC__)
 
 ///////////////////////////////////////////////////////////////////////////////
 // gethostname wrapper
