diff -r -u old/src/libhsakmt.h new/src/libhsakmt.h
--- src/libhsakmt.h	2020-03-04 15:00:47.000000000 -0500
+++ src/libhsakmt.h	2020-04-13 12:58:30.444865992 -0400
@@ -54,6 +54,10 @@
 #define CHECK_KFD_OPEN() \
 	do { if (kfd_open_count == 0 || hsakmt_forked) return HSAKMT_STATUS_KERNEL_IO_CHANNEL_NOT_OPENED; } while (0)
 
+/*On some platforms, a PAGE_SIZE macro eclipses the extern variable*/
+#ifdef PAGE_SIZE
+#undef PAGE_SIZE
+#endif
 extern int PAGE_SIZE;
 extern int PAGE_SHIFT;
 
diff -r -u old/src/topology.c new/src/topology.c
--- src/topology.c	2020-03-04 15:00:47.000000000 -0500
+++ src/topology.c	2020-04-13 13:01:31.209870701 -0400
@@ -38,6 +38,11 @@
 #include "libhsakmt.h"
 #include "fmm.h"
 
+/*Needed on MUSL*/
+#ifndef MAXNAMLEN
+#define MAXNAMLEN PATH_MAX
+#endif
+
 /* Number of memory banks added by thunk on top of topology
  * This only includes static heaps like LDS, scratch and SVM,
  * not for MMIO_REMAP heap. MMIO_REMAP memory bank is reported
