Author: Tatsuki Sugiura <sugi@nemui.org>
Origin: vendor, http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=504963
Bug-Debian: http://bugs.debian.org/504963
Forwarded: no
Last-Update: 2010-02-08

Index: libfcgi/libfcgi/fcgio.cpp
===================================================================
--- libfcgi/fcgio.cpp.orig  2010-01-23 19:13:53.000000000 +0900
+++ libfcgi/fcgio.cpp 2010-01-23 19:13:55.000000000 +0900
@@ -22,6 +22,7 @@
 #define DLLAPI  __declspec(dllexport)
 #endif
 
+#include <cstdio>
 #include <limits.h>
 #include "fcgio.h"
