diff -Naur gnome-settings-daemon-3.6.4.orig/data/org.gnome.settings-daemon.plugins.gschema.xml.in.in gnome-settings-daemon-3.6.4/data/org.gnome.settings-daemon.plugins.gschema.xml.in.in
--- gnome-settings-daemon-3.6.4.orig/data/org.gnome.settings-daemon.plugins.gschema.xml.in.in	2013-01-10 14:37:58.000000000 +0100
+++ gnome-settings-daemon-3.6.4/data/org.gnome.settings-daemon.plugins.gschema.xml.in.in	2013-04-10 17:39:36.446882275 +0200
@@ -2,7 +2,6 @@
   <schema gettext-domain="@GETTEXT_PACKAGE@" id="org.gnome.settings-daemon.plugins" path="/org/gnome/settings-daemon/plugins/">
     <child name="a11y-keyboard" schema="org.gnome.settings-daemon.plugins.a11y-keyboard"/>
     <child name="a11y-settings" schema="org.gnome.settings-daemon.plugins.a11y-settings"/>
-    <child name="background" schema="org.gnome.settings-daemon.plugins.background"/>
     <child name="clipboard" schema="org.gnome.settings-daemon.plugins.clipboard"/>
     <child name="color" schema="org.gnome.settings-daemon.plugins.color"/>
     <child name="cursor" schema="org.gnome.settings-daemon.plugins.cursor"/>
@@ -45,18 +44,6 @@
       <_summary>Priority to use for this plugin</_summary>
       <_description>Priority to use for this plugin in gnome-settings-daemon startup queue</_description>
     </key>
-  </schema>
-  <schema gettext-domain="@GETTEXT_PACKAGE@" id="org.gnome.settings-daemon.plugins.background" path="/org/gnome/settings-daemon/plugins/background/">
-    <key name="active" type="b">
-      <default>true</default>
-      <_summary>Activation of this plugin</_summary>
-      <_description>Whether this plugin would be activated by gnome-settings-daemon or not</_description>
-    </key>
-    <key name="priority" type="i">
-      <default>97</default>
-      <_summary>Priority to use for this plugin</_summary>
-      <_description>Priority to use for this plugin in gnome-settings-daemon startup queue</_description>
-    </key>
   </schema>
   <schema gettext-domain="@GETTEXT_PACKAGE@" id="org.gnome.settings-daemon.plugins.clipboard" path="/org/gnome/settings-daemon/plugins/clipboard/">
     <key name="active" type="b">
diff -Naur gnome-settings-daemon-3.6.4.orig/plugins/background/background.gnome-settings-plugin.in gnome-settings-daemon-3.6.4/plugins/background/background.gnome-settings-plugin.in
--- gnome-settings-daemon-3.6.4.orig/plugins/background/background.gnome-settings-plugin.in	2012-04-07 15:21:20.000000000 +0200
+++ gnome-settings-daemon-3.6.4/plugins/background/background.gnome-settings-plugin.in	1970-01-01 01:00:00.000000000 +0100
@@ -1,8 +0,0 @@
-[GNOME Settings Plugin]
-Module=background
-IAge=0
-_Name=Background
-_Description=Background plugin
-Authors=
-Copyright=Copyright © 2007
-Website=
diff -Naur gnome-settings-daemon-3.6.4.orig/plugins/background/gnome-fallback-background-helper.c gnome-settings-daemon-3.6.4/plugins/background/gnome-fallback-background-helper.c
--- gnome-settings-daemon-3.6.4.orig/plugins/background/gnome-fallback-background-helper.c	1970-01-01 01:00:00.000000000 +0100
+++ gnome-settings-daemon-3.6.4/plugins/background/gnome-fallback-background-helper.c	2013-04-10 17:39:36.450215685 +0200
@@ -0,0 +1,65 @@
+/* -*- Mode: C; tab-width: 8; indent-tabs-mode: nil; c-basic-offset: 8 -*-
+ *
+ * Copyright (C) 2010 Red Hat, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ *
+ * Author: Tomas Bzatek <tbzatek@redhat.com>
+ */
+
+#include "config.h"
+
+#include <glib.h>
+#include <glib/gi18n.h>
+#include <unistd.h>
+#include <gtk/gtk.h>
+
+#include "gsd-background-manager.h"
+
+int
+main (int argc,
+      char **argv)
+{
+        GMainLoop *loop;
+        GsdBackgroundManager *manager;
+        GError *error = NULL;
+
+        g_type_init ();
+        gtk_init (&argc, &argv);
+
+        bindtextdomain (GETTEXT_PACKAGE, GNOME_SETTINGS_LOCALEDIR);
+        bind_textdomain_codeset (GETTEXT_PACKAGE, "UTF-8");
+        textdomain (GETTEXT_PACKAGE);
+
+        loop = g_main_loop_new (NULL, FALSE);
+        manager = gsd_background_manager_new ();
+
+        gsd_background_manager_start (manager, &error);
+
+        if (error != NULL) {
+                g_printerr ("Unable to start the background manager: %s",
+                            error->message);
+
+                g_error_free (error);
+                _exit (1);
+        }
+
+        g_main_loop_run (loop);
+
+        gsd_background_manager_stop (manager);
+        g_main_loop_unref (loop);
+
+        return 0;
+}
diff -Naur gnome-settings-daemon-3.6.4.orig/plugins/background/gnome-fallback-background-helper.desktop.in.in gnome-settings-daemon-3.6.4/plugins/background/gnome-fallback-background-helper.desktop.in.in
--- gnome-settings-daemon-3.6.4.orig/plugins/background/gnome-fallback-background-helper.desktop.in.in	1970-01-01 01:00:00.000000000 +0100
+++ gnome-settings-daemon-3.6.4/plugins/background/gnome-fallback-background-helper.desktop.in.in	2013-04-10 17:39:36.450215685 +0200
@@ -0,0 +1,12 @@
+[Desktop Entry]
+_Name=Background
+_Comment=Background plugin
+Exec=@LIBEXECDIR@/gnome-fallback-background-helper
+Icon=desktop
+Terminal=false
+Type=Application
+Categories=
+NoDisplay=true
+OnlyShowIn=GNOME;Unity;
+X-GNOME-Autostart-Notify=true
+AutostartCondition=GNOME3 if-session gnome-fallback
diff -Naur gnome-settings-daemon-3.6.4.orig/plugins/background/gsd-background-plugin.c gnome-settings-daemon-3.6.4/plugins/background/gsd-background-plugin.c
--- gnome-settings-daemon-3.6.4.orig/plugins/background/gsd-background-plugin.c	2013-01-10 14:37:58.000000000 +0100
+++ gnome-settings-daemon-3.6.4/plugins/background/gsd-background-plugin.c	1970-01-01 01:00:00.000000000 +0100
@@ -1,104 +0,0 @@
-/* -*- Mode: C; tab-width: 8; indent-tabs-mode: nil; c-basic-offset: 8 -*-
- *
- * Copyright (C) 2007 William Jon McCann <mccann@jhu.edu>
- *
- * This program is free software; you can redistribute it and/or modify
- * it under the terms of the GNU General Public License as published by
- * the Free Software Foundation; either version 2, or (at your option)
- * any later version.
- *
- * This program is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU General Public License for more details.
- *
- * You should have received a copy of the GNU General Public License
- * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
- *
- */
-
-#include "config.h"
-
-#include <glib/gi18n-lib.h>
-#include <gmodule.h>
-
-#include "gnome-settings-plugin.h"
-#include "gsd-background-plugin.h"
-#include "gsd-background-manager.h"
-
-struct GsdBackgroundPluginPrivate {
-        GsdBackgroundManager *manager;
-};
-
-#define GSD_BACKGROUND_PLUGIN_GET_PRIVATE(object) (G_TYPE_INSTANCE_GET_PRIVATE ((object), GSD_TYPE_BACKGROUND_PLUGIN, GsdBackgroundPluginPrivate))
-
-GNOME_SETTINGS_PLUGIN_REGISTER (GsdBackgroundPlugin, gsd_background_plugin)
-
-static void
-gsd_background_plugin_init (GsdBackgroundPlugin *plugin)
-{
-        plugin->priv = GSD_BACKGROUND_PLUGIN_GET_PRIVATE (plugin);
-
-        g_debug ("GsdBackgroundPlugin initializing");
-
-        plugin->priv->manager = gsd_background_manager_new ();
-}
-
-static void
-gsd_background_plugin_finalize (GObject *object)
-{
-        GsdBackgroundPlugin *plugin;
-
-        g_return_if_fail (object != NULL);
-        g_return_if_fail (GSD_IS_BACKGROUND_PLUGIN (object));
-
-        g_debug ("GsdBackgroundPlugin finalizing");
-
-        plugin = GSD_BACKGROUND_PLUGIN (object);
-
-        g_return_if_fail (plugin->priv != NULL);
-
-        if (plugin->priv->manager != NULL) {
-                g_object_unref (plugin->priv->manager);
-        }
-
-        G_OBJECT_CLASS (gsd_background_plugin_parent_class)->finalize (object);
-}
-
-static void
-impl_activate (GnomeSettingsPlugin *plugin)
-{
-        gboolean res;
-        GError  *error;
-
-        g_debug ("Activating background plugin");
-
-        error = NULL;
-        res = gsd_background_manager_start (GSD_BACKGROUND_PLUGIN (plugin)->priv->manager, &error);
-        if (! res) {
-                g_warning ("Unable to start background manager: %s", error->message);
-                g_error_free (error);
-        }
-}
-
-static void
-impl_deactivate (GnomeSettingsPlugin *plugin)
-{
-        g_debug ("Deactivating background plugin");
-        gsd_background_manager_stop (GSD_BACKGROUND_PLUGIN (plugin)->priv->manager);
-}
-
-static void
-gsd_background_plugin_class_init (GsdBackgroundPluginClass *klass)
-{
-        GObjectClass           *object_class = G_OBJECT_CLASS (klass);
-        GnomeSettingsPluginClass *plugin_class = GNOME_SETTINGS_PLUGIN_CLASS (klass);
-
-        object_class->finalize = gsd_background_plugin_finalize;
-
-        plugin_class->activate = impl_activate;
-        plugin_class->deactivate = impl_deactivate;
-
-        g_type_class_add_private (klass, sizeof (GsdBackgroundPluginPrivate));
-}
diff -Naur gnome-settings-daemon-3.6.4.orig/plugins/background/gsd-background-plugin.h gnome-settings-daemon-3.6.4/plugins/background/gsd-background-plugin.h
--- gnome-settings-daemon-3.6.4.orig/plugins/background/gsd-background-plugin.h	2013-01-10 14:37:58.000000000 +0100
+++ gnome-settings-daemon-3.6.4/plugins/background/gsd-background-plugin.h	1970-01-01 01:00:00.000000000 +0100
@@ -1,59 +0,0 @@
-/* -*- Mode: C; tab-width: 8; indent-tabs-mode: nil; c-basic-offset: 8 -*-
- *
- * Copyright (C) 2007 William Jon McCann <mccann@jhu.edu>
- *
- * This program is free software; you can redistribute it and/or modify
- * it under the terms of the GNU General Public License as published by
- * the Free Software Foundation; either version 2, or (at your option)
- * any later version.
- *
- * This program is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU General Public License for more details.
- *
- * You should have received a copy of the GNU General Public License
- * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
- *
- */
-
-#ifndef __GSD_BACKGROUND_PLUGIN_H__
-#define __GSD_BACKGROUND_PLUGIN_H__
-
-#include <glib.h>
-#include <glib-object.h>
-#include <gmodule.h>
-
-#include "gnome-settings-plugin.h"
-
-G_BEGIN_DECLS
-
-#define GSD_TYPE_BACKGROUND_PLUGIN                (gsd_background_plugin_get_type ())
-#define GSD_BACKGROUND_PLUGIN(o)                  (G_TYPE_CHECK_INSTANCE_CAST ((o), GSD_TYPE_BACKGROUND_PLUGIN, GsdBackgroundPlugin))
-#define GSD_BACKGROUND_PLUGIN_CLASS(k)            (G_TYPE_CHECK_CLASS_CAST((k), GSD_TYPE_BACKGROUND_PLUGIN, GsdBackgroundPluginClass))
-#define GSD_IS_BACKGROUND_PLUGIN(o)               (G_TYPE_CHECK_INSTANCE_TYPE ((o), GSD_TYPE_BACKGROUND_PLUGIN))
-#define GSD_IS_BACKGROUND_PLUGIN_CLASS(k)         (G_TYPE_CHECK_CLASS_TYPE ((k), GSD_TYPE_BACKGROUND_PLUGIN))
-#define GSD_BACKGROUND_PLUGIN_GET_CLASS(o)        (G_TYPE_INSTANCE_GET_CLASS ((o), GSD_TYPE_BACKGROUND_PLUGIN, GsdBackgroundPluginClass))
-
-typedef struct GsdBackgroundPluginPrivate GsdBackgroundPluginPrivate;
-
-typedef struct
-{
-        GnomeSettingsPlugin    parent;
-        GsdBackgroundPluginPrivate *priv;
-} GsdBackgroundPlugin;
-
-typedef struct
-{
-        GnomeSettingsPluginClass parent_class;
-} GsdBackgroundPluginClass;
-
-GType   gsd_background_plugin_get_type            (void) G_GNUC_CONST;
-
-/* All the plugins must implement this function */
-G_MODULE_EXPORT GType register_gnome_settings_plugin (GTypeModule *module);
-
-G_END_DECLS
-
-#endif /* __GSD_BACKGROUND_PLUGIN_H__ */
diff -Naur gnome-settings-daemon-3.6.4.orig/plugins/background/Makefile.am gnome-settings-daemon-3.6.4/plugins/background/Makefile.am
--- gnome-settings-daemon-3.6.4.orig/plugins/background/Makefile.am	2013-01-10 14:37:58.000000000 +0100
+++ gnome-settings-daemon-3.6.4/plugins/background/Makefile.am	2013-04-10 17:40:20.117885180 +0200
@@ -1,80 +1,34 @@
-NULL =
+libexec_PROGRAMS = gnome-fallback-background-helper
 
-plugin_name = background
+gnome_fallback_background_helper_SOURCES = \
+	gnome-fallback-background-helper.c \
+	gsd-background-manager.c \
+	gsd-background-manager.h
 
-libexec_PROGRAMS = gsd-test-background
-
-gsd_test_background_SOURCES =		\
-	test-background.c		\
-	gsd-background-manager.h	\
-	gsd-background-manager.c	\
-	$(NULL)
-
-gsd_test_background_CPPFLAGS =				\
+gnome_fallback_background_helper_CPPFLAGS = \
 	-I$(top_srcdir)/gnome-settings-daemon		\
-	-I$(top_srcdir)/plugins/common			\
 	-DGNOME_SETTINGS_LOCALEDIR=\""$(datadir)/locale"\" \
 	$(AM_CPPFLAGS)
 
-gsd_test_background_CFLAGS = \
-	$(PLUGIN_CFLAGS)		\
+gnome_fallback_background_helper_CFLAGS =	\
 	$(SETTINGS_PLUGIN_CFLAGS)	\
-	$(BACKGROUND_CFLAGS)		\
-	$(AM_CFLAGS)
-
-gsd_test_background_LDADD =				\
-	$(top_builddir)/gnome-settings-daemon/libgsd.la	\
-	$(SETTINGS_PLUGIN_LIBS)				\
-	$(BACKGROUND_LIBS)				\
-	$(NULL)
-
-plugin_LTLIBRARIES = \
-	libbackground.la		\
-	$(NULL)
-
-libbackground_la_SOURCES =		\
-	gsd-background-plugin.h		\
-	gsd-background-plugin.c		\
-	gsd-background-manager.h	\
-	gsd-background-manager.c	\
-	$(NULL)
+	$(BACKGROUND_CFLAGS)
 
-libbackground_la_CPPFLAGS = \
-	-I$(top_srcdir)/gnome-settings-daemon		\
-	-I$(top_srcdir)/plugins/background/libbackground   \
-	-DGNOME_SETTINGS_LOCALEDIR=\""$(datadir)/locale"\" \
-	$(AM_CPPFLAGS)
-
-libbackground_la_CFLAGS = \
-	$(SETTINGS_PLUGIN_CFLAGS)	\
-	$(BACKGROUND_CFLAGS)		\
-	$(AM_CFLAGS)
-
-libbackground_la_LDFLAGS =		\
-	$(GSD_PLUGIN_LDFLAGS)		\
-	$(NULL)
-
-libbackground_la_LIBADD  =		\
+gnome_fallback_background_helper_LDADD =	\
 	$(SETTINGS_PLUGIN_LIBS)		\
 	$(BACKGROUND_LIBS)		\
-	$(NULL)
-
-plugin_in_files =		\
-	background.gnome-settings-plugin.in	\
-	$(NULL)
+	$(top_builddir)/gnome-settings-daemon/libgsd.la
 
-plugin_DATA = $(plugin_in_files:.gnome-settings-plugin.in=.gnome-settings-plugin)
+autostartdir = $(sysconfdir)/xdg/autostart
+autostart_in_files = gnome-fallback-background-helper.desktop.in
+autostart_in_in_files = gnome-fallback-background-helper.desktop.in.in
+autostart_DATA = $(autostart_in_files:.desktop.in=.desktop)
 
-EXTRA_DIST = 			\
-	$(plugin_in_files)	\
-	$(NULL)
+$(autostart_in_files): $(autostart_in_in_files)
+	@sed -e "s|\@LIBEXECDIR\@|$(libexecdir)|" $< > $@
 
-CLEANFILES = 			\
-	$(plugin_DATA)		\
-	$(NULL)
+@INTLTOOL_DESKTOP_RULE@
 
-DISTCLEANFILES =		\
-	$(plugin_DATA)		\
-	$(NULL)
+EXTRA_DIST = $(autostart_in_in_files)
 
-@GSD_INTLTOOL_PLUGIN_RULE@
+CLEANFILES = $(autostart_DATA) $(autostart_in_files)
diff -Naur gnome-settings-daemon-3.6.4.orig/plugins/background/test-background.c gnome-settings-daemon-3.6.4/plugins/background/test-background.c
--- gnome-settings-daemon-3.6.4.orig/plugins/background/test-background.c	2012-06-01 22:02:53.000000000 +0200
+++ gnome-settings-daemon-3.6.4/plugins/background/test-background.c	1970-01-01 01:00:00.000000000 +0100
@@ -1,7 +0,0 @@
-#define NEW gsd_background_manager_new
-#define START gsd_background_manager_start
-#define STOP gsd_background_manager_stop
-#define MANAGER GsdBackgroundManager
-#include "gsd-background-manager.h"
-
-#include "test-plugin.h"
diff -Naur gnome-settings-daemon-3.6.4.orig/po/POTFILES.in gnome-settings-daemon-3.6.4/po/POTFILES.in
--- gnome-settings-daemon-3.6.4.orig/po/POTFILES.in	2013-01-10 14:37:58.000000000 +0100
+++ gnome-settings-daemon-3.6.4/po/POTFILES.in	2013-04-10 17:39:36.450215685 +0200
@@ -21,9 +21,9 @@
 [type: gettext/glade]plugins/a11y-keyboard/gsd-a11y-preferences-dialog.ui
 [type: gettext/ini]plugins/a11y-settings/a11y-settings.gnome-settings-plugin.in
 plugins/automount/gnome-fallback-mount-helper.desktop.in.in
+plugins/background/gnome-fallback-background-helper.desktop.in.in
 plugins/automount/gsd-automount-manager.c
 plugins/automount/gsd-autorun.c
-[type: gettext/ini]plugins/background/background.gnome-settings-plugin.in
 [type: gettext/ini]plugins/clipboard/clipboard.gnome-settings-plugin.in
 [type: gettext/ini]plugins/color/color.gnome-settings-plugin.in
 plugins/color/gsd-color-manager.c
diff -Naur gnome-settings-daemon-3.6.4.orig/po/POTFILES.skip gnome-settings-daemon-3.6.4/po/POTFILES.skip
--- gnome-settings-daemon-3.6.4.orig/po/POTFILES.skip	2012-04-07 18:58:07.000000000 +0200
+++ gnome-settings-daemon-3.6.4/po/POTFILES.skip	2013-04-10 17:39:36.453549095 +0200
@@ -20,6 +20,7 @@
 data/org.gnome.settings-daemon.plugins.updates.gschema.xml.in
 data/org.gnome.settings-daemon.plugins.xrandr.gschema.xml.in
 data/org.gnome.settings-daemon.plugins.xsettings.gschema.xml.in
+plugins/background/gnome-fallback-background-helper.desktop.in
 plugins/automount/gnome-fallback-mount-helper.desktop.in
 plugins/power/org.gnome.settings-daemon.plugins.power.policy.in
 plugins/wacom/org.gnome.settings-daemon.plugins.wacom.policy.in
