--- src/MEGASync/google_breakpad/client/linux/handler/exception_handler.cc	2019-09-13 22:27:23.351723073 +0300
+++ -	2019-09-13 22:33:08.244312688 +0300
@@ -104,11 +104,13 @@
 #define PR_SET_PTRACER 0x59616d61
 #endif
 
+#if defined(__GLIBC__) && !__GLIBC_PREREQ(2, 30)
 // A wrapper for the tgkill syscall: send a signal to a specific thread.
 static int tgkill(pid_t tgid, pid_t tid, int sig) {
   return syscall(__NR_tgkill, tgid, tid, sig);
   return 0;
 }
+#endif
 
 namespace google_breakpad {
 
