--- a/src/command.c	2016-03-05 22:06:06.710989576 +0000
+++ b/src/command.c	2016-03-05 22:06:36.743614116 +0000
@@ -556,7 +556,7 @@
     ptydev = ttydev = _getpty(&fd, O_RDWR | O_NDELAY, 0622, 0);
     if (ptydev == NULL)
 	goto Failed;
-#elif defined (__svr4__) || defined(__CYGWIN32__) || defined(__lnx21__)
+#elif defined (__svr4__) || defined(__CYGWIN32__) || defined(__lnx21__) || defined(__linux__)
     {
 	extern char    *ptsname();
 
