don't use git submodule
--- a/src/CMakeLists.txt	2025-01-27 17:38:18.152414737 +1100
+++ b/src/CMakeLists.txt	2025-01-27 17:41:08.938986637 +1100
@@ -123,11 +123,12 @@
 find_package(Qt6 COMPONENTS Qml REQUIRED)
 message(STATUS "Qt6 version: ${Qt6_VERSION}")
 
+find_package(Catch2 REQUIRED)
+include_directories(${XBPS_CROSS_BASE}/usr/include/catch2)
 add_subdirectory(lib)
 add_subdirectory(gui)
 add_subdirectory(gui-qml)
 add_subdirectory(cli)
-add_subdirectory(tests)
 add_subdirectory(e2e EXCLUDE_FROM_ALL)
 add_subdirectory(crash-reporter)
 
