don't use git submodule
--- a/src/lib/CMakeLists.txt	2025-01-12 00:43:30.000000000 +1100
+++ b/src/lib/CMakeLists.txt	2025-01-27 12:33:30.219274168 +1100
@@ -19,13 +19,11 @@
 endif()
 
 # Lexbor
-add_subdirectory(vendor/lexbor)
-set(LIBS ${LIBS} lexbor_static)
-include_directories("vendor/lexbor/source/")
+set(LIBS ${LIBS} lexbor)
 
 include(ListFilterRegex)
 file(GLOB_RECURSE SOURCES "src/*.cpp" "vendor/*.cpp" "vendor/*.c")
-listFilterRegex(SOURCES "vendor/lexbor/")
+# listFilterRegex(SOURCES "vendor/lexbor/")
 include_directories("src/" ".")
 
 if(WIN_FILE_PROPS)
@@ -37,6 +35,7 @@
 
 add_library(${PROJECT_NAME} STATIC ${SOURCES})
 target_link_libraries(${PROJECT_NAME} ${QT_LIBRARIES} ${LIBS})
+target_include_directories(${PROJECT_NAME} PUBLIC lexbor)
 
 # Pre-compiled header
 if(USE_PCH)
