Update lexbor to 2.4.0
--- a/src/lib/src/utils/html-node.cpp	2025-01-12 00:43:30.000000000 +1100
+++ b/src/lib/src/utils/html-node.cpp	2025-01-27 17:57:55.493038111 +1100
@@ -136,7 +136,7 @@
 }
 
 
-lxb_status_t find_callback(lxb_dom_node_t *node, lxb_css_selector_specificity_t *spec, void *ctx)
+lxb_status_t find_callback(lxb_dom_node_t *node, lxb_css_selector_specificity_t spec, void *ctx)
 {
 	Q_UNUSED(spec)
 
@@ -154,7 +154,7 @@
 {
 	// Create CSS parser
 	auto *parser = lxb_css_parser_create();
-	auto parser_status = lxb_css_parser_init(parser, NULL, NULL);
+	auto parser_status = lxb_css_parser_init(parser, NULL);
 	if (parser_status != LXB_STATUS_OK) {
 		log(QStringLiteral("Error creating CSS parser: %1.").arg(parser_status), Logger::Error);
 		return {};
