diff --git a/build/php.m4 b/build/php.m4
index d8a5cbf..b23b456 100644
--- a/build/php.m4
+++ b/build/php.m4
@@ -1384,6 +1384,7 @@ int main(void) {
   [php_cv_type_cookie_off64_t=yes],
   [php_cv_type_cookie_off64_t=no],
   [AS_CASE([$host_alias],
+    [aarch64-linux-musl], [php_cv_type_cookie_off64_t=no],
     [*linux*], [php_cv_type_cookie_off64_t=yes],
     [php_cv_type_cookie_off64_t=no])]
   )])
