# Thanks to Chocimier for the patch!

diff --git Makefile Makefile
index 29ea60f..d586472 100644
--- Makefile
+++ Makefile
@@ -7,6 +7,7 @@ export OL_HOME=libraries
 all: release
 
 CC ?= gcc
+LD ?= ld
 UNAME ?= $(shell uname -s)
 
 # 'configure' part:
@@ -167,7 +168,7 @@ src/olvm.c: extensions/ffi.c
 	touch src/olvm.c
 
 $(repl.o): repl
-	ld -r -b binary -o $(repl.o) repl
+	$(LD) -r -b binary -o $(repl.o) repl

