diff --git a/scripts/build-binary b/scripts/build-binary
index 17230a7a..09b68121 100755
--- a/scripts/build-binary
+++ b/scripts/build-binary
@@ -18,7 +18,8 @@ if [ -n "${GOCOVER}" ]; then
     GO_BUILD_FLAGS="${GO_BUILD_FLAGS} -cover"
 fi
 
-REVISION=$(git rev-parse HEAD)$(if ! git diff --no-ext-diff --quiet --exit-code; then echo .dirty; fi)
+# REVISION=$(git rev-parse HEAD)$(if ! git diff --no-ext-diff --quiet --exit-code; then echo .dirty; fi)
+REVISION=${RKE_VERSION//+/-}
 RELEASE=${PROG}.${GOOS}-${GOARCH}
 
 BUILDTAGS="selinux netgo osusergo no_stage static_build sqlite_omit_load_extension no_embedded_executor no_cri_dockerd"
diff --git a/scripts/version.sh b/scripts/version.sh
index d05cfbae..73bc7ae2 100755
--- a/scripts/version.sh
+++ b/scripts/version.sh
@@ -29,7 +29,8 @@ fi
 GIT_TAG=$GITHUB_ACTION_TAG
 TREE_STATE=clean
 COMMIT=$DRONE_COMMIT
-REVISION=$(git rev-parse HEAD)$(if ! git diff --no-ext-diff --quiet --exit-code; then echo .dirty; fi)
+# REVISION=$(git rev-parse HEAD)$(if ! git diff --no-ext-diff --quiet --exit-code; then echo .dirty; fi)
+REVISION="release"
 PLATFORM=${GOOS}-${GOARCH}
 RELEASE=${PROG}.${PLATFORM}
 # hardcode versions unless set specifically
@@ -71,6 +72,8 @@ if [[ "${VERSION}" =~ ^v([0-9]+)\.([0-9]+)\.([0-9]+)([-+][a-zA-Z0-9.]+)?[-+]((rk
 fi
 
 DEPENDENCIES_URL="https://raw.githubusercontent.com/kubernetes/kubernetes/${KUBERNETES_VERSION}/build/dependencies.yaml"
-VERSION_GOLANG="go"$(curl -sL "${DEPENDENCIES_URL}" | yq e '.dependencies[] | select(.name == "golang: upstream version").version' -)
+#VERSION_GOLANG="go"$(curl -sL "${DEPENDENCIES_URL}" | yq e '.dependencies[] | select(.name == "golang: upstream version").version' -)
+VERSION_GOLANG=$(go version | awk '{ print $3 }')
+VERSION=${RKE_VERSION//-/+}
 
 DOCKERIZED_VERSION="${VERSION/+/-}" # this mimics what kubernetes builds do
