--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -9,16 +9,16 @@
 tilp_CPPFLAGS = -I$(top_srcdir)/intl \
 	@TICABLES_CFLAGS@ @TIFILES_CFLAGS@ @TICALCS_CFLAGS@ @TICONV_CFLAGS@ \
 	@GLIB_CFLAGS@ @GTK_CFLAGS@ \
-	@KDE_INCLUDES@ @QT_INCLUDES@ \
 	-DSHARE_DIR=\"$(pkgdatadir)\" \
 	-DLOCALEDIR=\"$(datadir)/locale\" \
 	-DSYSCONFDIR=\"$(sysconfdir)\" \
 	-DGTK_DISABLE_DEPRECATED
+#	@KDE_INCLUDES@ @QT_INCLUDES@
 tilp_LDFLAGS = -export-dynamic
 tilp_LDADD = @TICABLES_LIBS@ @TIFILES_LIBS@ @TICALCS_LIBS@ @TICONV_LIBS@ \
-	@GLIB_LIBS@ @GTK_LIBS@ \
-	@LIB_KDECORE@ @LIB_KDEUI@ @LIB_KIO@ @LIB_QT@ @KDE_LDFLAGS@ \
-	@QT_LDFLAGS@ @X_LDFLAGS@ @LIBZ@
+	@GLIB_LIBS@ @GTK_LIBS@ @LIBZ@
+#	@LIB_KDECORE@ @LIB_KDEUI@ @LIB_KIO@ @LIB_QT@ @KDE_LDFLAGS@ \
+#	@QT_LDFLAGS@ @X_LDFLAGS@
 tilp_SOURCES = *.h \
 	tilp_calcs.c tilp_cmdline.c tilp_config.c tilp_error.c \
 	tilp_files.c tilp_gif.c tilp_main.c \
--- a/configure.ac
+++ b/configure.ac
@@ -118,22 +118,23 @@
 CFLAGS="$CFLAGS $ARCH"
 
 # KDE dialogs support
-AC_ARG_WITH(kde, AC_HELP_STRING([--with-kde], [Compile with KDE support]), [kde=$withval], [kde=no])
-if test "x$kde" = "xdefault"; then
-  case $host_os in
-    *mingw*)
-      kde=no
-      ;;
-    *)
-      kde=yes
-      ;;
-  esac
-fi
-if test "x$kde" = "xyes"; then
-        AC_PROG_CXX
-        AC_PATH_KDE
-        AC_DEFINE(WITH_KDE, 1, [Use KDE support])
-fi
+#AC_ARG_WITH(kde, AC_HELP_STRING([--with-kde], [Compile with KDE support]), [kde=$withval], [kde=no])
+#if test "x$kde" = "xdefault"; then
+#  case $host_os in
+#    *mingw*)
+#      kde=no
+#      ;;
+#    *)
+#      kde=yes
+#      ;;
+#  esac
+#fi
+#if test "x$kde" = "xyes"; then
+#        AC_PROG_CXX
+#        AC_PATH_KDE
+#        AC_DEFINE(WITH_KDE, 1, [Use KDE support])
+#fi
+kde=no
 AM_CONDITIONAL(USE_KDE, test "x$kde" = "xyes")
 AC_SUBST(kde)
 
