--- a/dom/src/dae/daeDom.cpp	2017-10-21 07:05:24.000000000 +0200
+++ -	2025-01-08 16:10:01.914351183 +0100
@@ -111,7 +111,7 @@
         return ColladaDOM141::domAny::ID();
     }
 #endif
-    return NULL;
+    return 0;
 }
 
 daeInt getDomSourceID(DAE& dae)
@@ -126,7 +126,7 @@
         return ColladaDOM141::domSource::ID();
     }
 #endif
-    return NULL;
+    return 0;
 }
 
 daeInt getDomCOLLADAID(const char* specversion)
@@ -141,7 +141,7 @@
         return ColladaDOM141::domCOLLADA::ID();
     }
 #endif
-    return NULL;
+    return 0;
 }
 
 void copyElementAny(daeElementRef dstAny, daeElement* srcAny)
