diff --git libsrc/utils/Logger.cpp libsrc/utils/Logger.cpp
index 5e803dd..fbfe0f8 100644
--- libsrc/utils/Logger.cpp
+++ libsrc/utils/Logger.cpp
@@ -46,7 +46,7 @@ Logger::Logger ( std::string name, LogLevel minLevel ):
 	_syslogEnabled(true),
 	_loggerId(loggerId++)
 {
-#ifdef __GLIBC__
+#ifdef __linux__
 	_appname = std::string(program_invocation_short_name);
 #else
 	_appname = std::string(getprogname());
