diff --git a/Cargo.toml b/Cargo.toml
index 1fa32ac..815cbda 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -46,7 +46,7 @@ xdg = "2.5"
 
 [target."cfg(not(all(windows, target_arch = \"x86\", target_env = \"gnu\")))".dependencies]
 # if ssl feature is enabled compilation will fail on arm-unknown-linux-gnueabihf and i686-pc-windows-gnu
-git2 = { version = "0.18", optional = true, default-features = false }
+git2 = { version = "0.17", optional = true, default-features = false }
 
 [target.'cfg(unix)'.dependencies]
 users = { version = "0.11.3", package = "uzers" }
