--- ./setup.py.orig	2025-11-17 12:04:02.690894768 -0500
+++ ./setup.py	2025-11-17 12:04:31.274969247 -0500
@@ -1,22 +1,2 @@
 from setuptools import setup
-import ast
-import os
-currentDir = os.path.dirname(os.path.abspath(__file__))
-#currentDir = Path(__file__).parent
-
-def extractMetaInfo(src):
-    info = {}
-    a=ast.parse(src)
-    for e in a.body:
-        if isinstance(e, ast.Assign) and isinstance(e.value, ast.Str):
-            info[e.targets[0].id] = e.value.s
-    return info
-
-text = ''
-with open(currentDir  +os.path.sep+"filebytes"+ os.path.sep+"__init__.py") as f:
-    text = f.read()
-
-version = extractMetaInfo(text)["VERSION"]
-
-
-setup(version=version)
+setup(version='@@VERSION@@')
