--- a/gnome-pass-search-provider.py
+++ b/gnome-pass-search-provider.py
@@ -31,7 +31,10 @@ import re
 import subprocess
 
 from dbus.mainloop.glib import DBusGMainLoop
-from fuzzywuzzy import process, fuzz
+try:
+    from thefuzz import fuzz, process
+except ModuleNotFoundError:
+    from fuzzywuzzy import fuzz, process
 from gi.repository import GLib
 from os import getenv, walk
 from os.path import expanduser, join as path_join
