--- a/OpenEXR/OpenEXR.pc.in	2019-09-18 03:02:06.000000000 +0200
+++ b/OpenEXR/OpenEXR.pc.in	2019-12-01 17:30:08.680721453 +0100
@@ -6,8 +6,8 @@
 prefix=@prefix@
-exec_prefix=@exec_prefix@
-libdir=@libdir@
-includedir=@includedir@
-OpenEXR_includedir=@includedir@/OpenEXR
+exec_prefix=${prefix}/@exec_prefix@
+libdir=${prefix}/@libdir@
+includedir=${prefix}/@includedir@
+OpenEXR_includedir=${prefix}/@includedir@/OpenEXR
 libsuffix=@LIB_SUFFIX_DASH@
 
 Name: OpenEXR
