Index: cmake-scripts/boomerang-flags.cmake
===================================================================
--- a/cmake-scripts/boomerang-flags.cmake
+++ b/cmake-scripts/boomerang-flags.cmake
@@ -104,16 +104,6 @@ endif ()
 
 
 if (NOT MSVC)
-    if ("${CMAKE_BUILD_TYPE}" STREQUAL "Debug")
-        BOOMERANG_ADD_COMPILE_FLAGS(-O0)
-    elseif ("${CMAKE_BUILD_TYPE}" STREQUAL "RelWithDebInfo")
-        # No special flags
-    elseif ("${CMAKE_BUILD_TYPE}" STREQUAL "MinSizeRel")
-        BOOMERANG_ADD_COMPILE_FLAGS(-Os)
-    else () # Release
-        BOOMERANG_ADD_COMPILE_FLAGS(-O3)
-    endif ()
-
     if (BOOMERANG_ENABLE_COVERAGE)
         set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fprofile-arcs -ftest-coverage")
         set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fprofile-arcs -ftest-coverage")
