--- a/Configure	2020-04-16 14:12:56.000000000 -0300
+++ b/Configure	2021-11-24 16:04:43.647041822 -0300
@@ -154,10 +154,10 @@
     try_add_CFLAG $FLAG bypass && break
 done
 
 # Select the most appropriate FPU control scheme
-for FLAG in '-DISOC99_FENV' '-DFPUCONTROLH' '-Dx86'; do
+for FLAG in '' '-DISOC99_FENV' '-DFPUCONTROLH' '-Dx86'; do
     try_add_CFLAG $FLAG && break
 done
 
 # Some flags to try as last resort.  These hurt performance, so only add
 # them if needed.
