--- memory/mozjemalloc/jemalloc.c	2018-09-07 01:30:40.056777558 -0300
+++ memory/mozjemalloc/jemalloc.c	2018-09-07 01:30:45.847777478 -0300
@@ -331,9 +331,6 @@
 #endif
 #include <sys/time.h>
 #include <sys/types.h>
-#if !defined(MOZ_MEMORY_SOLARIS) && !defined(MOZ_MEMORY_ANDROID)
-#include <sys/sysctl.h>
-#endif
 #include <sys/uio.h>
 #ifndef MOZ_MEMORY
 #include <sys/ktrace.h> /* Must come after several other sys/ includes. */
