Do not generate LaTeX PDF documentation.
--- a/doc/CMakeLists.txt
+++ b/doc/CMakeLists.txt
@@ -213,7 +213,6 @@
 add_custom_target(run_doxygen
         COMMENT "Generating Latex and HTML documentation."
         COMMAND ${CMAKE_COMMAND} -E env VERSION=${VERSION} INDEX_DOC=index_html.dox GENERATE_HTML=YES GENERATE_LATEX=NO ${CMAKE_CROSSCOMPILING_EMULATOR} ${DOXYGEN_EXECUTABLE}
-        COMMAND ${CMAKE_COMMAND} -E env VERSION=${VERSION} INDEX_DOC=index_latex.dox GENERATE_HTML=NO GENERATE_LATEX=YES ${CMAKE_CROSSCOMPILING_EMULATOR} ${DOXYGEN_EXECUTABLE}
         DEPENDS ${PROJECT_BINARY_DIR}/doc/language.dox ${PROJECT_BINARY_DIR}/doc/config.dox
         DEPENDS ${OUT_DOC_FILES}
         DEPENDS examples
@@ -271,15 +270,6 @@
 )
 endif ()
 ################################################################################
-install(CODE "if(NOT EXISTS \"${PROJECT_BINARY_DIR}/latex/doxygen_manual.pdf\")
-    message(FATAL_ERROR \"\nTerminating:\n    documentation has not been generated, \n    create documentation by using the 'docs' target followed by an 'install'\n\")
-endif()"
-)
-
-install(FILES
-        "${PROJECT_BINARY_DIR}/latex/doxygen_manual.pdf"
-        DESTINATION "${CMAKE_INSTALL_PREFIX}/${DOC_INSTALL_DIR}"
-)
 
 if (build_doc_chm)
 install(CODE "if(NOT EXISTS \"${PROJECT_BINARY_DIR}/chm/doxygen_manual.chm\")
--- a/examples/CMakeLists.txt
+++ b/examples/CMakeLists.txt
@@ -1,6 +1,5 @@
 file(MAKE_DIRECTORY ${PROJECT_BINARY_DIR}/examples
-                    ${PROJECT_BINARY_DIR}/html/examples
-                    ${PROJECT_BINARY_DIR}/latex/examples)
+                    ${PROJECT_BINARY_DIR}/html/examples)
 file(GLOB EXAMPLE_FILES RELATIVE ${PROJECT_SOURCE_DIR}/examples "*")
 
 if (build_doc_chm)
@@ -81,11 +80,9 @@
         endif()
     add_custom_command(
         COMMAND ${CMAKE_COMMAND} -E make_directory ${PROJECT_BINARY_DIR}/html/examples/${f}
-        COMMAND ${CMAKE_COMMAND} -E make_directory ${PROJECT_BINARY_DIR}/latex/examples/${f}
         COMMAND ${CMAKE_COMMAND} -E env PROJECT_BINARY_DIR=${PROJECT_BINARY_DIR} EXTRA_SETTINGS=docuexample.cfg ${CMAKE_CROSSCOMPILING_EMULATOR} ${EXECUTABLE_OUTPUT_PATH}/doxygen ${f}.cfg
-        COMMAND ${Python_EXECUTABLE}  ${TOP}/examples/strip_example.py  < ${PROJECT_BINARY_DIR}/latex/examples/${f}/latex/refman.tex > ${PROJECT_BINARY_DIR}/latex/examples/${f}/latex/refman_doc.tex
         DEPENDS doxygen ${f}.${f_ext} ${f}.cfg ${TOP}/examples/strip_example.py ${f_dep} baseexample.cfg docuexample.cfg
-        OUTPUT ${PROJECT_BINARY_DIR}/html/examples/${f}/html/index.html ${PROJECT_BINARY_DIR}/latex/examples/${f}/latex/refman_doc.tex
+        OUTPUT ${PROJECT_BINARY_DIR}/html/examples/${f}/html/index.html
         WORKING_DIRECTORY ${PROJECT_BINARY_DIR}/examples
     )
     set(EXAMPLES_RES ${EXAMPLES_RES} "" ${PROJECT_BINARY_DIR}/html/examples/${f}/html/index.html)
--- a/examples/baseexample.cfg
+++ b/examples/baseexample.cfg
@@ -1,3 +1,4 @@
+GENERATE_LATEX     = NO
 PROJECT_ICON       = ../examples/doxygen.ico
 CASE_SENSE_NAMES   = NO
 QUIET              = YES
