From 5198966c8d5fec89116d025c74934ac03ea511fa Mon Sep 17 00:00:00 2001
From: Dimitri van Heesch <doxygen@gmail.com>
Date: Fri, 6 May 2022 09:55:16 +0200
Subject: [PATCH] issue #9312: Build: cache.h:53:14: error: 'exchange' is not a
 member of 'std'

---
 src/cache.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/cache.h b/src/cache.h
index 0ff3092ed5..e218eb27d2 100644
--- a/src/cache.h
+++ b/src/cache.h
@@ -19,6 +19,7 @@
 #include <list>
 #include <unordered_map>
 #include <mutex>
+#include <utility>
 #include <ctype.h>
 
 /*! Fixed size cache for value type V using keys of type K.
