--- addon/doxyapp/Makefile.doxyapp.orig	2010-06-23 01:03:15.167342549 +0200
+++ addon/doxyapp/Makefile.doxyapp	2010-06-23 01:03:28.205223709 +0200
@@ -14,7 +14,7 @@ CXXFLAGS=	-pipe -D_LARGEFILE_SOURCE -Wal
 INCPATH	=	-I../../qtools -I../../src
 LINK	=	g++
 LFLAGS	=	-g
-LIBS	=	-L../../lib -L../../lib -ldoxygen -lqtools -lmd5 -ldoxycfg -lpng 
+LIBS	=	-L../../lib -L../../lib -ldoxygen -lqtools -lmd5 -ldoxycfg -lpng -lpthread
 MOC	=	/usr/bin/moc
 
 TAR	=	tar -cf
