--- a/qtools/qdatetime.cpp	2020-01-03 23:51:18.000000000 +0100
+++ b/qtools/qdatetime.cpp	2020-12-02 10:48:03.245635836 +0100
@@ -35,7 +35,6 @@
 **
 **********************************************************************/
 
-#define gettimeofday	__hide_gettimeofday
 #include "qdatetime.h"
 #include "qdatastream.h"
 #include <stdio.h>
@@ -55,8 +54,6 @@
 #elif defined(_OS_UNIX_) || defined(_OS_MAC_)
 #include <sys/time.h>
 #include <unistd.h>
-#undef	gettimeofday
-extern "C" int gettimeofday( struct timeval *, struct timezone * );
 #endif
 
 static const uint FIRST_DAY	= 2361222;	// Julian day for 1752/09/14
