--- a/src/format_from_dict.cc
+++ b/src/format_from_dict.cc
@@ -21,6 +21,7 @@
  */
 
 #include "format_from_dict.h"
+#include <ctime>
 
 using namespace std;
 using namespace fmt;
@@ -54,7 +55,9 @@ static string scalar_to_string(scalar_t
     time_t val = get<time_t>(scalar);
     if (!spec.length())
       return to_string(val);
-    return format(runtime(format("{{:{}}}", spec)), fmt::localtime(val));
+    std::tm tm_ = {};
+    localtime_r(&val, &tm_);
+    return format(runtime(format("{{:{}}}", spec)), tm_);
   }
   throw runtime_error("Unrecognized type!"); // I shouldn't be here!
 }
