From https://sources.gentoo.org/cgi-bin/viewvc.cgi/gentoo-x86/app-forensics/sleuthkit/files/sleuthkit-4.1.0-system-sqlite.patch?revision=1.1

--- tsk/auto/db_sqlite.cpp
+++ tsk/auto/db_sqlite.cpp
@@ -14,7 +14,7 @@
  */
 
 #include "tsk_db_sqlite.h"
-#include "sqlite3.h"
+#include <sqlite3.h>
 
 #include <string.h>
 
--- tsk/auto/Makefile.am
+++ tsk/auto/Makefile.am
@@ -3,7 +3,8 @@
 
 noinst_LTLIBRARIES = libtskauto.la
 # Note that the .h files are in the top-level Makefile
-libtskauto_la_SOURCES = auto.cpp tsk_auto_i.h auto_db.cpp sqlite3.c sqlite3.h db_sqlite.cpp tsk_db_sqlite.h case_db.cpp tsk_case_db.h
+libtskauto_la_SOURCES = auto.cpp tsk_auto_i.h auto_db.cpp db_sqlite.cpp tsk_db_sqlite.h case_db.cpp tsk_case_db.h
+libtskauto_la_LIBADD = -lsqlite3
 
 indent:
 	indent *.cpp *.h
--- tsk/auto/tsk_db_sqlite.h
+++ tsk/auto/tsk_db_sqlite.h
@@ -25,7 +25,7 @@
 #include <ostream>
 
 
-#include "sqlite3.h"
+#include <sqlite3.h>
 #include "tsk_auto_i.h"
 
 using std::map;
