--- a/configure
+++ b/configure
@@ -7273,14 +7273,6 @@ fi
   if test "x$with_cfitsioinc" != x ; then
     CFITSIO_INCDIRS="$with_cfitsioinc"
   fi
-
-  CFITSIO_INCDIRS="$CFITSIO_INCDIRS   \
-           /usr/local/cfitsio/include \
-           /local/cfitsio/include"
-
-  LIBDIRS="$LIBDIRS           \
-           /usr/local/cfitsio/lib \
-           /local/cfitsio/lib"
 fi
 
 
@@ -7328,12 +7320,6 @@ fi
 
 if test "x$with_cfitsio" != xno -o \
         "x$with_pgplot"  != xno ; then
-  LIBDIRS="$LIBDIRS           \
-           /usr/local/lib     \
-           /local/lib         \
-           /opt/local/lib     \
-           /opt/SUNWspro/lib  \
-           /sw/lib"
 
   for LIBDIR in $LIBDIRS ; do
     as_ac_File=`printf "%s\n" "ac_cv_file_$LIBDIR" | $as_tr_sh`
@@ -7375,29 +7361,7 @@ fi
   # CFITSIO.
   if test "x$with_cfitsio" != xno ; then
     # Search for CFITSIO.
-    for INCDIR in $CFITSIO_INCDIRS $INCDIRS ; do
-      as_ac_File=`printf "%s\n" "ac_cv_file_$INCDIR/cfitsio/fitsio.h" | $as_tr_sh`
-{ printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking for $INCDIR/cfitsio/fitsio.h" >&5
-printf %s "checking for $INCDIR/cfitsio/fitsio.h... " >&6; }
-if eval test \${$as_ac_File+y}
-then :
-  printf %s "(cached) " >&6
-else $as_nop
-  test "$cross_compiling" = yes &&
-  as_fn_error $? "cannot check for file existence when cross compiling" "$LINENO" 5
-if test -r "$INCDIR/cfitsio/fitsio.h"; then
-  eval "$as_ac_File=yes"
-else
-  eval "$as_ac_File=no"
-fi
-fi
-eval ac_res=\$$as_ac_File
-	       { printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: $ac_res" >&5
-printf "%s\n" "$ac_res" >&6; }
-if eval test \"x\$"$as_ac_File"\" = x"yes"
-then :
-  CFITSIOINC="-I$INCDIR/cfitsio"; break
-fi
+    for INCDIR in $CFITSIO_INCDIRS ; do
 
       as_ac_File=`printf "%s\n" "ac_cv_file_$INCDIR/fitsio.h" | $as_tr_sh`
 { printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking for $INCDIR/fitsio.h" >&5
