From c3c115deab4562af72be43832aaecbd8accba51a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marc=20Andr=C3=A9=20Tanner?= <mat@brain-dump.org>
Date: Tue, 27 Jun 2017 08:50:10 +0200
Subject: [PATCH 3/4] Avoid use of BUFSIZ to guarantee system independent
 package size

---
 abduco.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git abduco.c abduco.c
index 1bd6304..335907d 100644
--- a/abduco.c
+++ b/abduco.c
@@ -75,7 +75,7 @@ typedef struct {
 	uint32_t type;
 	uint32_t len;
 	union {
-		char msg[BUFSIZ];
+		char msg[4096 - 2*sizeof(uint32_t)];
 		struct {
 			uint16_t rows;
 			uint16_t cols;
-- 
2.24.0

