--- a/configure.in	2015-08-21 23:38:47.726879936 +0200
+++ b/configure.in	2015-08-21 23:39:13.031879310 +0200
@@ -68,7 +68,7 @@ dnl Check for the fcntl header
 AC_CHECK_HEADER(fcntl.h,,AC_MSG_ERROR("fcntl.h not found"))
 
 dnl Check for the poll header
-AC_CHECK_HEADER(sys/poll.h,,AC_MSG_ERROR("sys/poll.h not found"))
+AC_CHECK_HEADER(poll.h,,AC_MSG_ERROR("poll.h not found"))
 
 dnl Other headers we're interested in
 AC_CHECK_HEADERS(unistd.h)
@@ -316,7 +316,7 @@ for testproto in 'struct pollfd *ufds, u
 do
   if test "${PROTO}" = ""; then
     AC_TRY_COMPILE([
-      #include <sys/poll.h>
+      #include <poll.h>
       int poll($testproto);
     ],,[PROTO="$testproto";],)
   fi
--- a/tsocks.c	2015-08-21 23:39:34.644878775 +0200
+++ b/tsocks.c	2015-08-21 23:39:49.338878412 +0200
@@ -41,7 +41,7 @@ char *progname = "libtsocks";
 #include <strings.h>
 #include <netinet/in.h>
 #include <arpa/inet.h>
-#include <sys/poll.h>
+#include <poll.h>
 #include <sys/time.h>
 #include <pwd.h>
 #include <errno.h>
