--- a/src/vdpau_decode.c
+++ b/src/vdpau_decode.c
@@ -1289,7 +1289,7 @@
             driver_data,
             obj_context->vdp_decoder,
             obj_surface->vdp_surface,
-            (VdpPictureInfo)&obj_context->vdp_picture_info,
+            (VdpPictureInfo *)&obj_context->vdp_picture_info,
             obj_context->vdp_bitstream_buffers_count,
             obj_context->vdp_bitstream_buffers
         );
