--- a/Makefile.in
+++ b/Makefile.in
@@ -929,13 +929,13 @@
 	cd parallel-$(YYYYMMDD) && \
 	./configure && make -j && sudo make -j install
 install-data-hook:
-	mkdir -p ${datarootdir}/bash-completion/completions
-	mkdir -p ${zshcompletiondir}
-	parallel --shell-completion bash > ${datarootdir}/bash-completion/completions/parallel
-	echo '#compdef parallel' > ${zshcompletion}
-	echo '(( $$+functions[_comp_parallel] )) ||' >> ${zshcompletion}
-	echo '  eval "$$(parallel --shell-completion auto)" &&' >> ${zshcompletion}
-	echo '  _comp_parallel' >> ${zshcompletion}
+	mkdir -p ${DESTDIR}${datarootdir}/bash-completion/completions
+	mkdir -p ${DESTDIR}${zshcompletiondir}
+	./src/parallel --shell-completion bash > ${DESTDIR}${datarootdir}/bash-completion/completions/parallel
+	echo '#compdef parallel' > ${DESTDIR}${zshcompletion}
+	echo '(( $$+functions[_comp_parallel] )) ||' >> ${DESTDIR}${zshcompletion}
+	echo '  eval "$$(parallel --shell-completion auto)" &&' >> ${DESTDIR}${zshcompletion}
+	echo '  _comp_parallel' >> ${DESTDIR}${zshcompletion}
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
