From 1aa74e96e32b4012ee0e13ada22b64811b519bab Mon Sep 17 00:00:00 2001
From: Olga Ustuzhanina <me@laserbat.pw>
Date: Thu, 24 Jan 2019 15:07:31 +0700
Subject: [PATCH] fix desktop switching on dvorak layouts

---
 src/binding.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/binding.c b/src/binding.c
index dc23b640..8c238717 100644
--- a/src/binding.c
+++ b/src/binding.c
@@ -18,6 +18,8 @@
 #include "root.h"
 #include "tray.h"
 
+#include <X11/XKBlib.h>
+
 #define MASK_NONE    0
 #define MASK_SHIFT   (1 << ShiftMapIndex)
 #define MASK_LOCK    (1 << LockMapIndex)
@@ -418,7 +420,7 @@ void InsertBinding(ActionType action, const char *modifiers,
 
             for(temp[offset] = '1'; temp[offset] <= '9'; temp[offset]++) {
 
-               sym = ParseKeyString(temp);
+               sym = XkbKeycodeToKeysym(display, temp[offset] - '1' + 10, 0, 0);
                if(sym == NoSymbol) {
                   Release(temp);
                   return;
