--- a/sources/shiboken6_generator/ApiExtractor/clangparser/triplet.cpp
+++ b/sources/shiboken6_generator/ApiExtractor/clangparser/triplet.cpp
@@ -23,9 +23,11 @@
         || a.compare("i686"_L1, Qt::CaseInsensitive) == 0) {
         return Architecture::X86;
     }
-    if (a.startsWith("armv7"_L1, Qt::CaseInsensitive))
+    if (a.startsWith("armv7"_L1, Qt::CaseInsensitive)
+        || a.startsWith("armv6"_L1, Qt::CaseInsensitive)
+        || a.startsWith("arm-"_L1, Qt::CaseInsensitive))
         return Architecture::Arm32;
-    if (a.startsWith("arm"_L1, Qt::CaseInsensitive)
+    if (a.startsWith("arm64"_L1, Qt::CaseInsensitive)
         || a.startsWith("aarch64"_L1, Qt::CaseInsensitive)) {
     }
     return Architecture::Unknown;
--- a/sources/pyside6/cmake/Macros/PySideModules.cmake
+++ b/sources/pyside6/cmake/Macros/PySideModules.cmake
@@ -126,6 +126,9 @@ macro(create_pyside_module)
     if(module_ADDITIONAL_INCLUDE_DIRS)
         list(APPEND shiboken_include_dir_list ${${module_ADDITIONAL_INCLUDE_DIRS}})
     endif()
+    if (DEFINED SYSROOT_INCLUDE_DIRS)
+        list(APPEND shiboken_include_dir_list ${SYSROOT_INCLUDE_DIRS})
+    endif()
 
     # Remove duplicates and transform the path separators into something
     # shiboken understands.
@@ -211,9 +214,15 @@ macro(create_pyside_module)
     # cross-compiling in linux, we use the clang compiler from the installed
     # libclang itself.
     if (CMAKE_CROSSCOMPILING)
+      if (DEFINED SHIBOKEN_ARCH)
+        list(APPEND shiboken_command "--platform=${CMAKE_SYSTEM_NAME}"
+                                     "--arch=${SHIBOKEN_ARCH}"
+                                     "--compiler-path=${CMAKE_CXX_COMPILER}")
+      else()
         list(APPEND shiboken_command "--platform=${CMAKE_SYSTEM_NAME}"
                                      "--arch=${CMAKE_SYSTEM_PROCESSOR}"
                                      "--compiler-path=${CMAKE_CXX_COMPILER}")
+     endif()
     endif()
 
     if(CMAKE_ANDROID_ARCH_LLVM_TRIPLE)
