--- a/sources/pyside6/libpyside/CMakeLists.txt
+++ b/sources/pyside6/libpyside/CMakeLists.txt
@@ -101,7 +101,7 @@ append_size_optimization_flags(pyside6)
 
 target_include_directories(pyside6 PUBLIC
   $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
-  $<INSTALL_INTERFACE:PySide6/include>
+  $<INSTALL_INTERFACE:include/PySide6>
 )
 
 target_compile_definitions(pyside6 PRIVATE -DQT_LEAN_HEADERS=1 -DQT_NO_KEYWORDS=1)
@@ -159,9 +159,6 @@ configure_package_config_file(
 )
 
 # Install-tree / wheel configuration
-set(PYSIDE_PYTHONPATH "")
-set(PYSIDE_TYPESYSTEMS "typesystems")
-set(PYSIDE_GLUE "glue")
 set(PYSIDE_SOVERSION "${pyside6_library_so_version}")
 
 configure_package_config_file(
@@ -177,7 +174,7 @@ configure_file("${CMAKE_CURRENT_SOURCE_D
                "${CMAKE_CURRENT_BINARY_DIR}/PySide6ConfigVersion.cmake" @ONLY)
 
 install(FILES ${libpyside_HEADERS}
-        DESTINATION ${BINDING_NAME}${pyside6_SUFFIX}/include)
+        DESTINATION include/${BINDING_NAME}${pyside6_SUFFIX})
 
 # build-time installation
 install(TARGETS pyside6 EXPORT PySide6Targets
@@ -191,9 +188,9 @@ set_target_properties(pyside6 PROPERTIES
 
 if(NOT is_pyside6_superproject_build)
     install(TARGETS pyside6 EXPORT PySide6WheelTargets
-                            LIBRARY DESTINATION "PySide6"
-                            ARCHIVE DESTINATION "PySide6"
-                            RUNTIME DESTINATION "PySide6")
+        LIBRARY DESTINATION "${LIB_INSTALL_DIR}"
+        ARCHIVE DESTINATION "${LIB_INSTALL_DIR}"
+        RUNTIME DESTINATION "${LIB_INSTALL_DIR}")
 endif()
 
 install(FILES "${CMAKE_CURRENT_BINARY_DIR}/pyside6${pyside6_SUFFIX}.pc"
--- a/sources/pyside6/PySide6/CMakeLists.txt
+++ b/sources/pyside6/PySide6/CMakeLists.txt
@@ -119,7 +119,7 @@ install(FILES ${CMAKE_CURRENT_SOURCE_DIR
 install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/templates/datavisualization_common.xml
         DESTINATION share/PySide6${pyside_SUFFIX}/typesystems)
 install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/pyside6_global.h
-        DESTINATION ${BINDING_NAME}${pyside6_SUFFIX}/include)
+        DESTINATION include/${BINDING_NAME}${pyside6_SUFFIX})
 
 install(DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/QtAsyncio"
         DESTINATION "${PYTHON_SITE_PACKAGES}/${BINDING_NAME}${pyside6_SUFFIX}")
--- a/sources/pyside6/cmake/Macros/PySideModules.cmake
+++ b/sources/pyside6/cmake/Macros/PySideModules.cmake
@@ -387,7 +387,7 @@ macro(create_pyside_module)
 
 
     install(FILES ${CMAKE_CURRENT_BINARY_DIR}/PySide6/${module_NAME}/pyside6_${lower_module_name}_python.h
-            DESTINATION PySide6${pyside6_SUFFIX}/include/${module_NAME}/)
+            DESTINATION include/PySide6${pyside6_SUFFIX}/${module_NAME}/)
     file(GLOB typesystem_files ${CMAKE_CURRENT_SOURCE_DIR}/typesystem_*.xml ${typesystem_path})
 
 #   Copy typesystem files and remove module names from the <load-typesystem> element
--- a/sources/pyside6/PySide6/QtCore/CMakeLists.txt
+++ b/sources/pyside6/PySide6/QtCore/CMakeLists.txt
@@ -306,4 +306,4 @@ if (APPLE)
 endif()
 
 install(FILES ${pyside6_SOURCE_DIR}/qtcorehelper.h ${pyside6_SOURCE_DIR}/qiopipe.h
-        DESTINATION PySide6/include/QtCore/)
+        DESTINATION include/PySide6/QtCore/)
--- a/sources/pyside6/PySide6/QtDBus/CMakeLists.txt
+++ b/sources/pyside6/PySide6/QtDBus/CMakeLists.txt
@@ -51,4 +51,4 @@ create_pyside_module(NAME QtDBus
                      TYPESYSTEM_PATH QtDBus_SOURCE_DIR
                      SOURCES QtDBus_SRC)
 
-install(FILES ${pyside6_SOURCE_DIR}/qtdbushelper.h DESTINATION PySide6/include/QtDBus)
+install(FILES ${pyside6_SOURCE_DIR}/qtdbushelper.h DESTINATION include/PySide6/QtDBus)
--- a/sources/pyside6/PySide6/QtDataVisualization/CMakeLists.txt
+++ b/sources/pyside6/PySide6/QtDataVisualization/CMakeLists.txt
@@ -68,4 +68,4 @@ create_pyside_module(NAME QtDataVisualiz
                      STATIC_SOURCES QtDataVisualization_src)
 
 install(FILES ${pyside6_SOURCE_DIR}/qtdatavisualization_helper.h
-        DESTINATION PySide6/include/QtDataVisualization)
+        DESTINATION include/PySide6/QtDataVisualization)
--- a/sources/pyside6/PySide6/QtDesigner/CMakeLists.txt
+++ b/sources/pyside6/PySide6/QtDesigner/CMakeLists.txt
@@ -67,4 +67,4 @@ create_pyside_module(NAME QtDesigner
                      STATIC_SOURCES QtDesigner_static_src
                      TYPESYSTEM_NAME ${QtDesigner_BINARY_DIR}/typesystem_designer.xml)
 
-install(FILES ${pyside6_SOURCE_DIR}/qpydesignerextensions.h DESTINATION PySide6/include/QtDesigner)
+install(FILES ${pyside6_SOURCE_DIR}/qpydesignerextensions.h DESTINATION include/PySide6/QtDesigner)
--- a/sources/pyside6/PySide6/QtGraphs/CMakeLists.txt
+++ b/sources/pyside6/PySide6/QtGraphs/CMakeLists.txt
@@ -96,4 +96,4 @@ create_pyside_module(NAME QtGraphs
                      DROPPED_ENTRIES QtGraphs_DROPPED_ENTRIES)
 
 install(FILES ${pyside6_SOURCE_DIR}/qtgraphs_helper.h
-        DESTINATION PySide6/include/QtGraphs)
+        DESTINATION include/PySide6/QtGraphs)
--- a/sources/pyside6/PySide6/QtGui/CMakeLists.txt
+++ b/sources/pyside6/PySide6/QtGui/CMakeLists.txt
@@ -366,4 +366,4 @@ create_pyside_module(NAME QtGui
 
 install(FILES ${pyside6_SOURCE_DIR}/qpytextobject.h
               ${pyside6_SOURCE_DIR}/qtguihelper.h
-        DESTINATION PySide6/include/QtGui/)
+        DESTINATION include/PySide6/QtGui/)
--- a/sources/pyside6/libpysideqml/CMakeLists.txt
+++ b/sources/pyside6/libpysideqml/CMakeLists.txt
@@ -84,7 +84,7 @@ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}
 qfp_strip_library("pyside6qml")
 
 install(FILES ${libpysideqml_HEADERS}
-        DESTINATION ${BINDING_NAME}${pyside6qml_SUFFIX}/include)
+        DESTINATION include/${BINDING_NAME}${pyside6qml_SUFFIX})
 
 #built-time installation
 install(TARGETS pyside6qml EXPORT PySide6Targets
--- a/sources/pyside6/libpysideremoteobjects/CMakeLists.txt
+++ b/sources/pyside6/libpysideremoteobjects/CMakeLists.txt
@@ -80,7 +80,7 @@ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}
 #
 
 install(FILES ${libpysideremoteobjects_HEADERS}
-        DESTINATION ${BINDING_NAME}${pyside6remoteobjects_SUFFIX}/include)
+        DESTINATION include/${BINDING_NAME}${pyside6remoteobjects_SUFFIX})
 
 install(TARGETS pyside6remoteobjects EXPORT PySide6RemoteObjectsTargets
                            LIBRARY DESTINATION "${LIB_INSTALL_DIR}"
--- a/sources/shiboken6/libshiboken/CMakeLists.txt
+++ b/sources/shiboken6/libshiboken/CMakeLists.txt
@@ -111,7 +111,7 @@ add_library(Shiboken6::libshiboken ALIAS
 target_include_directories(libshiboken PUBLIC
   $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
   $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>
-  $<INSTALL_INTERFACE:shiboken6/include>
+  $<INSTALL_INTERFACE:include/shiboken6>
 )
 
 if (NOT "${NUMPY_INCLUDE_DIR}" STREQUAL "")
@@ -197,7 +197,7 @@ install(FILES
         signature.h
         signature_p.h
 
-        DESTINATION shiboken6${shiboken6_SUFFIX}/include)
+        DESTINATION include/shiboken6${shiboken6_SUFFIX})
 install(TARGETS libshiboken EXPORT Shiboken6Targets
                             LIBRARY DESTINATION "${LIB_INSTALL_DIR}"
                             ARCHIVE DESTINATION "${LIB_INSTALL_DIR}"
