--- makedefs.orig	2015-04-19 18:16:51.880377535 +0200
+++ makedefs	2015-04-19 18:18:09.413982443 +0200
@@ -500,11 +500,11 @@
 		: ${SHLIB_ENV="LD_LIBRARY_PATH=`pwd`/lib"}
 		: ${PLUGIN_LD="${CC-gcc} -shared"}
 		;;
-    Linux.3*)	SYSTYPE=LINUX3
+    Linux.*)	SYSTYPE=LINUX3
 		case "$CCARGS" in
 		 *-DNO_DB*) ;;
 		 *-DHAS_DB*) ;;
 		 *) if [ -f /usr/include/db.h ]
 		    then
 			: we are all set
 		    elif [ -f /usr/include/db/db.h ]
@@ -731,6 +731,7 @@
 # Look for the ICU library and enable unicode email if available.
 #
 case "$CCARGS" in
+ *-DHAS_EAI*)	;;
  *-DNO_EAI*) ;;
           *) trap 'rm -f makedefs.test makedefs.test.[co]' 1 2 3 15
              cat >makedefs.test.c <<'EOF'
