--- a/postfix-install	2012-01-31 16:18:56.687116320 +0100
+++ b/postfix-install	2012-01-31 16:19:00.198032219 +0100
@@ -168,7 +168,7 @@
 # By now, shells must have functions. Ultrix users must use sh5 or lose.
 
 umask 022
-PATH=/bin:/usr/bin:/usr/sbin:/usr/etc:/sbin:/etc:/usr/contrib/bin:/usr/gnu/bin:/usr/ucb:/usr/bsd
+#PATH=/bin:/usr/bin:/usr/sbin:/usr/etc:/sbin:/etc:/usr/contrib/bin:/usr/gnu/bin:/usr/ucb:/usr/bsd
 SHELL=/bin/sh
 IFS=" 	
 "
