Index: CMakeLists.txt
===================================================================
--- CMakeLists.txt.orig
+++ CMakeLists.txt
@@ -75,11 +75,6 @@ set(CMAKE_INCLUDE_CURRENT_DIR ON)
 
 
 #RPATH setup - more below too
-if (WANT_NORPATH OR WANT_DISTROBUILD)
-	set(CMAKE_BUILD_WITH_INSTALL_RPATH FALSE)
-else()
-	set(CMAKE_BUILD_WITH_INSTALL_RPATH TRUE)
-endif()
 set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
 set(CMAKE_SKIP_RULE_DEPENDENCY TRUE)
 set(CMAKE_SKIP_BUILD_RPATH TRUE)
@@ -330,15 +325,6 @@ include (CMakeLists_Directories.cmake)
 
 #Convert our simpler command line option to the CMake style
 #None, Debug, Release, .. or custom ones
-if(WANT_DEBUG)
-	set(CMAKE_BUILD_TYPE "Debug" CACHE STRING "Set Debug Build Type" FORCE)
-endif()
-if (WANT_RELEASEWITHDEBUG)
-	set(CMAKE_BUILD_TYPE "RelWithDebInfo" CACHE STRING "Set Release with Debug Info Build Type" FORCE)
-endif()
-if(NOT WANT_DEBUG AND NOT WANT_RELEASEWITHDEBUG)
-	set(CMAKE_BUILD_TYPE "Release" CACHE STRING "Set Release Build Type" FORCE)
-endif()
 
 
 
