--- a/scribus/CMakeLists.txt	2021-04-25 22:24:52.000000000 +0200
+++ b/scribus/CMakeLists.txt	2021-06-12 16:23:22.884229251 +0200
@@ -757,7 +757,7 @@
 # Should fix bug 6897 .
 	set_target_properties(${EXE_NAME} PROPERTIES LINK_FLAGS -Wl,-E)
 endif ()
-if (GCC AND NOT WIN32)
+if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND NOT WIN32)
 # Use the -E / --export-dynamic flag to ensure that symbols in the Scribus
 # binary are in the public symbol table for plugins. The symbols must also
 # be set as SCRIBUS_API if we're building with -fvisibility=hidden .
