--- a/scribus/util.cpp	2019-07-31 00:35:04.000000000 +0200
+++ b/scribus/util.cpp	2019-07-31 00:35:04.000000000 +0200
@@ -40,7 +40,7 @@
 
 #include <csignal>
 
-#if !defined(_WIN32) && !defined(Q_OS_MAC) 
+#if defined(__GLIBC__)
 #include <execinfo.h>
 #include <cxxabi.h>
 #endif
--- a/scribus/util_debug.cpp	2019-07-31 00:35:07.000000000 +0200
+++ b/scribus/util_debug.cpp	2019-07-31 00:35:07.000000000 +0200
@@ -24,7 +24,7 @@
 #include <QDateTime>
 #include <QtGlobal>
 
-#if !defined(_WIN32) && !defined(Q_OS_MAC)
+#if defined(__GLIBC__)
 #include <execinfo.h>
 #include <cxxabi.h>
 #endif
@@ -54,7 +54,7 @@
  */
 void printBacktrace ( int nFrames )
 {
-#if !defined(_WIN32) && !defined(Q_OS_MAC) && !defined(Q_OS_OPENBSD) && !defined(Q_OS_FREEBSD)
+#if defined(__GLIBC__)
 	void ** trace = new void*[nFrames + 1];
 	char **messages = ( char ** ) nullptr;
 	int i, trace_size = 0;
