--- scribus/util.cpp.orig	2016-01-11 15:35:18.171968184 +0100
+++ scribus/util.cpp	2016-01-11 15:35:53.600967308 +0100
@@ -37,7 +37,7 @@ for which a new license (GPL+exception)
 
 #include <signal.h>
 
-#if !defined(_WIN32) && !defined(Q_OS_MAC) && !defined(Q_OS_HAIKU) 
+#ifdef __GLIBC__
 #include <execinfo.h>
 #include <cxxabi.h>
 #endif
@@ -947,7 +947,7 @@ void getDashArray(int dashtype, double l
  */
 void printBacktrace ( int nFrames )
 {
-#if !defined(_WIN32) && !defined(Q_OS_MAC) && !defined(Q_OS_OPENBSD)  && !defined(Q_OS_FREEBSD) && !defined(Q_OS_HAIKU)
+#ifdef __GLIBC__
 	void ** trace = new void*[nFrames + 1];
 	char **messages = ( char ** ) NULL;
 	int i, trace_size = 0;
