From 8152ac071bce1ebc98fac6bed962e90e9b92d8cf Mon Sep 17 00:00:00 2001
From: Ferry Huberts <ferry.huberts@pelagic.nl>
Date: Tue, 9 Mar 2021 22:04:46 +0100
Subject: [PATCH] Fix alignment compile errors on GCC 11

Signed-off-by: Ferry Huberts <ferry.huberts@pelagic.nl>
---
 src/blake2.h | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/blake2.h b/src/blake2.h
index b05208117e..b4398f9e13 100644
--- a/src/blake2.h
+++ b/src/blake2.h
@@ -57,7 +57,7 @@ extern "C" {
     uint8_t  personal[BLAKE2S_PERSONALBYTES];  // 32
   } blake2s_param;
 
-  JTR_ALIGN( 64 ) typedef struct __blake2s_state
+  typedef struct JTR_ALIGN( 64 ) __blake2s_state
   {
     uint32_t h[8];
     uint32_t t[2];
@@ -82,7 +82,7 @@ extern "C" {
     uint8_t  personal[BLAKE2B_PERSONALBYTES];  // 64
   } blake2b_param;
 
-  JTR_ALIGN( 64 ) typedef struct __blake2b_state
+  typedef struct JTR_ALIGN( 64 ) __blake2b_state
   {
     uint64_t h[8];
     uint64_t t[2];
@@ -94,7 +94,7 @@ extern "C" {
 #if defined(JOHN_NO_SIMD) || (!defined(__SSE2__) && !defined(__SSE4_1__) && !defined(__XOP__))
   typedef struct __blake2sp_state
 #else
-  JTR_ALIGN( 64 ) typedef struct __blake2sp_state
+  typedef struct JTR_ALIGN( 64 ) __blake2sp_state
 #endif
   {
     blake2s_state S[8][1];
@@ -106,7 +106,7 @@ extern "C" {
 #if defined(JOHN_NO_SIMD) || (!defined(__SSE2__) && !defined(__SSE4_1__) && !defined(__XOP__))
   typedef struct __blake2bp_state
 #else
-  JTR_ALIGN( 64 ) typedef struct __blake2bp_state
+  typedef struct JTR_ALIGN( 64 ) __blake2bp_state
 #endif
   {
     blake2b_state S[4][1];
