--- meld/build_helpers.py.orig
+++ meld/build_helpers.py
@@ -379,7 +379,7 @@
     def finalize_options(self):
         special_cases = ('debian', 'ubuntu', 'linuxmint')
         if (platform.system() == 'Linux' and
-                platform.linux_distribution()[0].lower() in special_cases):
+                False):
             # Maintain an explicit install-layout, but use deb by default
             specified_layout = getattr(self, 'install_layout', None)
             self.install_layout = specified_layout or 'deb'
