--- include/allegro5/allegro_opengl.h
+++ include/allegro5/allegro_opengl.h
@@ -103,10 +103,14 @@
 
 /* HACK: Prevent both Mesa and SGI's broken headers from screwing us */
 #define __glext_h_
+#define __gl_glext_h_
 #define __glxext_h_
+#define __glx_glxext_h_
 #include <GL/gl.h>
 #undef  __glext_h_
+#undef  __gl_glext_h_
 #undef  __glxext_h_
+#undef  __glx_glxext_h_
 
 #endif /* ALLEGRO_MACOSX */

--- include/allegro5/opengl/GLext/glx_ext_defs.h
+++ include/allegro5/opengl/GLext/glx_ext_defs.h
@@ -1,7 +1,9 @@
 /* HACK: Prevent both Mesa and SGI's broken headers from screwing us */
 #define __glxext_h_
+#define __glx_glxext_h_
 #include <GL/glx.h>
 #undef __glxext_h_
+#undef __glx_glxext_h_
 
 #ifndef GLX_VERSION_1_3
 #define _ALLEGRO_GLX_VERSION_1_3
