--- va/wayland/Makefile.am	2016-08-25 04:42:40.000000000 +0200
+++ va/wayland/Makefile.am	2016-09-12 18:05:59.410106843 +0200
@@ -24,6 +24,7 @@
 	-DLINUX			\
 	-I$(top_srcdir)		\
 	-I$(top_srcdir)/va	\
+	-I$(top_srcdir)/va/drm	\
 	$(WAYLAND_CFLAGS)	\
 	$(DRM_CFLAGS)		\
 	$(NULL)
@@ -32,7 +33,7 @@
 	va_wayland.c		\
 	va_wayland_drm.c	\
 	va_wayland_emgd.c	\
-	$(top_srcdir)/va/drm/va_drm_utils.c \
+	va_drm_utils.c		\
 	$(NULL)
 
 source_h = \
@@ -58,7 +59,7 @@
 
 # Wayland protocol
 va_wayland_drm.c: $(protocol_source_h)
-%-client-protocol.h : %.xml
+wayland-client-protocol.h : wayland.xml
 	$(AM_V_GEN)$(WAYLAND_SCANNER) client-header < $< > $@
 
 EXTRA_DIST = \
