--- a/Makefile.in	2014-11-22 12:00:00.000000000 +0100
+++ b/Makefile.in	2016-05-03 23:44:44.859376553 +0200
@@ -836,7 +836,7 @@
 	@echo "/* AUTOMATICALLY GENERATED - DO NOT EDIT MANUALLY */" > version.h
 	@echo "#ifndef W_SCAN_VERSION_H" >> version.h
 	@echo "#define W_SCAN_VERSION_H" >> version.h
-	@echo "uint version=$(__VERSION);" >> version.h
+	@echo "unsigned int version=$(__VERSION);" >> version.h
 	@echo "#endif" >> version.h
 	@echo "" >> version.h
 	@chmod a-x version.h
--- a/version.h	2014-11-22 11:59:56.000000000 +0100
+++ b/version.h	2016-05-03 23:42:59.612376864 +0200
@@ -1,6 +1,6 @@
 /* AUTOMATICALLY GENERATED - DO NOT EDIT MANUALLY */
 #ifndef W_SCAN_VERSION_H
 #define W_SCAN_VERSION_H
-uint version=20170107;
+unsigned int version=20170107;
 #endif
 
