--- a/darkplaces/sys_shared.c	2015-05-31 12:54:42.131106842 +0200
+++ b/darkplaces/sys_shared.c	2015-05-31 12:54:58.842284100 +0200
@@ -69,7 +69,7 @@ void Sys_AllowProfiling(qboolean enable)
 	else
 		moncleanup();
 #endif
-#elif defined(__linux__) || defined(__FreeBSD__)
+#elif defined(__linux__) && defined(__GLIBC__) || defined(__FreeBSD__)
 	extern int moncontrol(int);
 	moncontrol(enable);
 #endif
