--- a/utils/nfsdcld/nfsdcld.c	2019-11-13 18:09:41.000000000 +0100
+++ b/utils/nfsdcld/nfsdcld.c	2020-01-17 20:01:32.335230045 +0100
@@ -379,7 +379,7 @@
 	wsize = atomicio((void *)write, clnt->cl_fd, cmsg, bsize);
 	if (wsize != bsize)
 		xlog(L_ERROR, "%s: problem writing to cld pipe (%ld): %m",
-			 __func__, wsize);
+			 __func__, (long int)wsize);
 
 	/* reopen pipe, just to be sure */
 	ret = cld_pipe_open(clnt);
@@ -410,7 +410,7 @@
 	wsize = atomicio((void *)write, clnt->cl_fd, cmsg, bsize);
 	if (wsize != bsize) {
 		xlog(L_ERROR, "%s: problem writing to cld pipe (%ld): %m",
-			 __func__, wsize);
+			 __func__, (long int)wsize);
 		ret = cld_pipe_open(clnt);
 		if (ret) {
 			xlog(L_FATAL, "%s: unable to reopen pipe: %d",
@@ -460,7 +460,7 @@
 	wsize = atomicio((void *)write, clnt->cl_fd, cmsg, bsize);
 	if (wsize != bsize) {
 		xlog(L_ERROR, "%s: problem writing to cld pipe (%ld): %m",
-			 __func__, wsize);
+			 __func__, (long int)wsize);
 		ret = cld_pipe_open(clnt);
 		if (ret) {
 			xlog(L_FATAL, "%s: unable to reopen pipe: %d",
@@ -499,7 +499,7 @@
 	wsize = atomicio((void *)write, clnt->cl_fd, cmsg, bsize);
 	if (wsize != bsize) {
 		xlog(L_ERROR, "%s: problem writing to cld pipe (%ld): %m",
-			 __func__, wsize);
+			 __func__, (long int)wsize);
 		ret = cld_pipe_open(clnt);
 		if (ret) {
 			xlog(L_FATAL, "%s: unable to reopen pipe: %d",
@@ -549,7 +549,7 @@
 	wsize = atomicio((void *)write, clnt->cl_fd, cmsg, bsize);
 	if (wsize != bsize) {
 		xlog(L_ERROR, "%s: problem writing to cld pipe (%ld): %m",
-			 __func__, wsize);
+			 __func__, (long int)wsize);
 		ret = cld_pipe_open(clnt);
 		if (ret) {
 			xlog(L_FATAL, "%s: unable to reopen pipe: %d",
@@ -608,7 +608,7 @@
 	wsize = atomicio((void *)write, clnt->cl_fd, cmsg, bsize);
 	if (wsize != bsize) {
 		xlog(L_ERROR, "%s: problem writing to cld pipe (%ld): %m",
-			 __func__, wsize);
+			 __func__, (long int)wsize);
 		ret = cld_pipe_open(clnt);
 		if (ret) {
 			xlog(L_FATAL, "%s: unable to reopen pipe: %d",
@@ -668,7 +668,7 @@
 	wsize = atomicio((void *)write, clnt->cl_fd, cmsg, bsize);
 	if (wsize != bsize) {
 		xlog(L_ERROR, "%s: problem writing to cld pipe (%ld): %m",
-			 __func__, wsize);
+			 __func__, (long int)wsize);
 		ret = cld_pipe_open(clnt);
 		if (ret) {
 			xlog(L_FATAL, "%s: unable to reopen pipe: %d",
--- a/utils/nfsdcld/sqlite.c	2019-11-13 18:09:41.000000000 +0100
+++ b/utils/nfsdcld/sqlite.c	2020-01-17 20:08:36.425251924 +0100
@@ -512,7 +512,7 @@
 	current_epoch = tcur;
 	recovery_epoch = trec;
 	ret = 0;
-	xlog(D_GENERAL, "%s: current_epoch=%lu recovery_epoch=%lu",
-		__func__, current_epoch, recovery_epoch);
+	xlog(D_GENERAL, "%s: current_epoch=%llu recovery_epoch=%llu",
+		__func__, (long long unsigned int)current_epoch, (long long unsigned int)recovery_epoch);
 out:
 	sqlite3_finalize(stmt);
@@ -1223,7 +1223,7 @@
 
 	current_epoch = tcur;
 	recovery_epoch = trec;
-	xlog(D_GENERAL, "%s: current_epoch=%lu recovery_epoch=%lu",
-		__func__, current_epoch, recovery_epoch);
+	xlog(D_GENERAL, "%s: current_epoch=%llu recovery_epoch=%llu",
+		__func__, (long long unsigned int)current_epoch, (long long unsigned int)recovery_epoch);
 
 out:
@@ -1282,7 +1282,7 @@
 	}
 
 	recovery_epoch = 0;
-	xlog(D_GENERAL, "%s: current_epoch=%lu recovery_epoch=%lu",
-		__func__, current_epoch, recovery_epoch);
+	xlog(D_GENERAL, "%s: current_epoch=%llu recovery_epoch=%llu",
+		__func__, (long long unsigned int)current_epoch, (long long unsigned int)recovery_epoch);
 
 out:
