--- nfs-utils-2.3.1.orig/support/export/export.c
+++ nfs-utils-2.3.1/support/export/export.c
@@ -17,6 +17,7 @@
 #include <stdlib.h>
 #include <dirent.h>
 #include <errno.h>
+#include <limits.h>
 #include "xmalloc.h"
 #include "nfslib.h"
 #include "exportfs.h"
--- nfs-utils-2.3.1.orig/support/export/xtab.c
+++ nfs-utils-2.3.1/support/export/xtab.c
@@ -18,6 +18,7 @@
 #include <sys/stat.h>
 #include <errno.h>
 #include <libgen.h>
+#include <limits.h>
 
 #include "nfslib.h"
 #include "exportfs.h"
--- nfs-utils-2.3.1.orig/support/misc/file.c
+++ nfs-utils-2.3.1/support/misc/file.c
@@ -27,6 +27,7 @@
 #include <dirent.h>
 #include <stdlib.h>
 #include <stdbool.h>
+#include <limits.h>
 
 #include "xlog.h"
 #include "misc.h"
--- nfs-utils-2.3.1.orig/support/nfs/xcommon.c
+++ nfs-utils-2.3.1/support/nfs/xcommon.c
@@ -16,6 +16,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <limits.h>
 
 #include "xcommon.h"
 #include "nls.h"	/* _() */
--- nfs-utils-2.3.1.orig/support/nsm/file.c
+++ nfs-utils-2.3.1/support/nsm/file.c
@@ -85,6 +85,7 @@
 #include <fcntl.h>
 #include <dirent.h>
 #include <grp.h>
+#include <limits.h>
 
 #include "xlog.h"
 #include "nsm.h"
--- nfs-utils-2.3.1.orig/utils/blkmapd/device-discovery.c
+++ nfs-utils-2.3.1/utils/blkmapd/device-discovery.c
@@ -49,6 +49,7 @@
 #include <unistd.h>
 #include <libgen.h>
 #include <errno.h>
+#include <limits.h>
 #include <libdevmapper.h>
 
 #ifdef HAVE_CONFIG_H
--- nfs-utils-2.3.1.orig/utils/gssd/krb5_util.c
+++ nfs-utils-2.3.1/utils/gssd/krb5_util.c
@@ -120,6 +120,7 @@
 #endif
 #include <krb5.h>
 #include <rpc/auth_gss.h>
+#include <limits.h>
 
 #include "gssd.h"
 #include "err_util.h"
--- nfs-utils-2.3.1.orig/utils/mountd/cache.c
+++ nfs-utils-2.3.1/utils/mountd/cache.c
@@ -26,6 +26,7 @@
 #include <pwd.h>
 #include <grp.h>
 #include <mntent.h>
+#include <limits.h>
 #include "misc.h"
 #include "nfslib.h"
 #include "exportfs.h"
--- nfs-utils-2.3.1.orig/utils/mountd/mountd.c
+++ nfs-utils-2.3.1/utils/mountd/mountd.c
@@ -22,6 +22,7 @@
 #include <fcntl.h>
 #include <sys/resource.h>
 #include <sys/wait.h>
+#include <limits.h>
 
 #include "conffile.h"
 #include "xmalloc.h"
--- nfs-utils-2.3.1.orig/utils/mountd/rmtab.c
+++ nfs-utils-2.3.1/utils/mountd/rmtab.c
@@ -16,6 +16,7 @@
 #include <netinet/in.h>
 #include <arpa/inet.h>
 #include <netdb.h>
+#include <limits.h>
 
 #include "misc.h"
 #include "exportfs.h"
