diff -ur a/support/reexport/fsidd.c b/support/reexport/fsidd.c
--- a/support/reexport/fsidd.c	2023-11-21 09:11:42.660255810 -0500
+++ b/support/reexport/fsidd.c	2023-11-21 09:12:36.536375957 -0500
@@ -7,6 +7,7 @@
 #include <dlfcn.h>
 #endif
 #include <event2/event.h>
+#include <unistd.h>
 
 #include "conffile.h"
 #include "reexport_backend.h"
diff -ur a/support/reexport/reexport.c b/support/reexport/reexport.c
--- a/support/reexport/reexport.c	2023-11-21 09:11:42.660255810 -0500
+++ b/support/reexport/reexport.c	2023-11-21 09:12:13.654324929 -0500
@@ -8,6 +8,7 @@
 #include <sys/types.h>
 #include <sys/vfs.h>
 #include <errno.h>
+#include <unistd.h>
 
 #include "nfsd_path.h"
 #include "conffile.h"
