--- a/configure.ac
+++ b/configure.ac
@@ -547,11 +547,11 @@
  -pipe \
  -Wall \
  -Wextra \
- -Werror=strict-prototypes \
- -Werror=missing-prototypes \
- -Werror=missing-declarations \
+ -Wstrict-prototypes \
+ -Wmissing-prototypes \
+ -Wmissing-declarations \
  -Werror=format=2 \
- -Werror=undef \
+ -Wundef \
  -Werror=missing-include-dirs \
  -Werror=strict-aliasing=2 \
  -Werror=init-self \
@@ -579,10 +579,10 @@
 
 CHECK_CCSUPPORT([-Werror=format-overflow=2], [flg1])
 CHECK_CCSUPPORT([-Werror=int-conversion], [flg2])
-CHECK_CCSUPPORT([-Werror=incompatible-pointer-types], [flg3])
+#CHECK_CCSUPPORT([-Werror=incompatible-pointer-types], [flg3])
 CHECK_CCSUPPORT([-Werror=misleading-indentation], [flg4])
 
-AC_SUBST([AM_CFLAGS], ["$my_am_cflags $flg1 $flg2 $flg3 $flg4"])
+AC_SUBST([AM_CFLAGS], ["$my_am_cflags $flg1 $flg2 $flg4"])
 
 # Make sure that $ACLOCAL_FLAGS are used during a rebuild
 AC_SUBST([ACLOCAL_AMFLAGS], ["-I $ac_macro_dir \$(ACLOCAL_FLAGS)"])
