Switch to use /run directory to store pid files.

--- utils/blkmapd/device-discovery.c.orig	2012-07-09 11:20:51.221140077 +0200
+++ utils/blkmapd/device-discovery.c	2012-07-09 11:21:03.050857448 +0200
@@ -51,7 +51,7 @@
 #include "device-discovery.h"
 
 #define BL_PIPE_FILE	"/var/lib/nfs/rpc_pipefs/nfs/blocklayout"
-#define PID_FILE	"/var/run/blkmapd.pid"
+#define PID_FILE	"/run/blkmapd.pid"
 
 struct bl_disk *visible_disk_list;
 
--- utils/statd/statd.c.orig	2012-07-09 11:21:43.651887544 +0200
+++ utils/statd/statd.c	2012-07-09 11:22:02.897427788 +0200
@@ -155,7 +155,7 @@ usage(void)
 	fprintf(stderr,"      -H                   Specify a high-availability callout program.\n");
 }
 
-static const char *pidfile = "/var/run/rpc.statd.pid";
+static const char *pidfile = "/run/rpc.statd.pid";
 
 int pidfd = -1;
 static void create_pidfile(void)
--- utils/statd/sm-notify.c.orig	2012-07-09 11:22:33.106706108 +0200
+++ utils/statd/sm-notify.c	2012-07-09 11:22:49.104323933 +0200
@@ -858,7 +858,7 @@ static int record_pid(void)
 	int fd;
 
 	(void)snprintf(pid, sizeof(pid), "%d\n", (int)getpid());
-	fd = open("/var/run/sm-notify.pid", O_CREAT|O_EXCL|O_WRONLY, 0600);
+	fd = open("/run/sm-notify.pid", O_CREAT|O_EXCL|O_WRONLY, 0600);
 	if (fd < 0)
 		return 0;
 
