http://hg.python.org/cpython/rev/c816479f6aaf/
diff --git a/configure b/configure
--- a/configure
+++ b/configure
@@ -2995,6 +2995,7 @@ then
	MACHDEP="$ac_md_system$ac_md_release"

	case $MACHDEP in
+	linux*) MACHDEP="linux2";;
	cygwin*) MACHDEP="cygwin";;
	darwin*) MACHDEP="darwin";;
	atheos*) MACHDEP="atheos";;
diff --git a/configure.in b/configure.in
--- a/configure.in
+++ b/configure.in
@@ -293,6 +293,7 @@ then
	MACHDEP="$ac_md_system$ac_md_release"

	case $MACHDEP in
+	linux*) MACHDEP="linux2";;
	cygwin*) MACHDEP="cygwin";;
	darwin*) MACHDEP="darwin";;
	atheos*) MACHDEP="atheos";;
