--- setup.py	2008-09-30 10:15:45.000000000 +1000
+++ setup.py	2008-12-07 16:04:01.000000000 +1000
@@ -1013,7 +1013,7 @@
                 if self.compiler.find_library_file(lib_dirs, 'ndbm'):
                     ndbm_libs = ['ndbm']
                 else:
-                    ndbm_libs = []
+                    ndbm_libs = ['gdbm', 'gdbm_compat']
                 exts.append( Extension('dbm', ['dbmmodule.c'],
                                        define_macros=[('HAVE_NDBM_H',None)],
                                        libraries = ndbm_libs ) )
