--- configure.in.orig	2018-06-22 20:57:08.925033685 +0200
+++ configure.in	2018-06-22 20:58:19.544037328 +0200
@@ -776,42 +776,6 @@
     LAME_CFLAGS="$lame_ii"
   fi
   CPPFLAGS="$save_CPPFLAGS"
-
-  AC_MSG_CHECKING([lame version])
-  ac_save_CFLAGS="$CFLAGS"
-  ac_save_LIBS="$LIBS"
-  CFLAGS="$CFLAGS $LAME_CFLAGS"
-  LIBS="$LIBS $LAME_LIBS"
-  AC_RUN_IFELSE([AC_LANG_SOURCE([[
-#include <stdio.h>
-
-#ifdef HAVE_LAME_INC
-#include <lame/lame.h>
-#else
-#include <lame.h>
-#endif
-
-int main () {
-  lame_version_t lv;
-  get_lame_version_numerical(&lv);
-  if(lv.alpha || lv.beta) lv.minor--;
-  printf("%d%d\n", lv.major, lv.minor);
-  return 0;
-}
-]])],
-    [lame_version="`./conftest$ac_exeext`"],
-    [AC_MSG_RESULT([failed])
-     TC_PKG_ERROR(lame, lame.h, yes, lame, [http://www.mp3dev.org/],
-      [cannot compile and run a test program])],,
-    [AC_MSG_RESULT([cross compiling; assumed OK...])])
-  CFLAGS="$ac_save_CFLAGS"
-  LIBS="$ac_save_LIBS"
-
-  if test 393 -ge "$lame_version" ; then
-    have_lame="no"
-    TC_PKG_ERROR(lame, lame.h, yes, lame, [http://www.mp3dev.org/],
-      [lame version < 3.93])
-  fi
 fi
 TC_PKG_HAVE(lame, LAME)
 
