--- a/dist/Makefile.in.orig	2013-02-16 11:00:25.023739458 +0100
+++ b/dist/Makefile.in	2013-02-16 11:01:06.572741213 +0100
@@ -1689,9 +1689,9 @@ perlxsi.c:
 install-exec-local:
 	cd $(DESTDIR)$(bindir) && $(rm) -f `echo ex | sed '$(transform)'`
 	cd $(DESTDIR)$(bindir) && $(rm) -f `echo view | sed '$(transform)'`
-	cd $(DESTDIR)$(bindir) && $(ln) \
+	cd $(DESTDIR)$(bindir) && $(LN_S) \
 		`echo vi | sed '$(transform)'` `echo ex | sed '$(transform)'`
-	cd $(DESTDIR)$(bindir) && $(ln) \
+	cd $(DESTDIR)$(bindir) && $(LN_S) \
 		`echo vi | sed '$(transform)'` `echo view | sed '$(transform)'`
 
 install-data-local: install_common
@@ -1712,9 +1712,9 @@ install-data-local: install_common
 	    $(rm) -f `echo ex.0 | sed '$(transform)'`
 	@cd $(DESTDIR)$(mandir)/cat1 && \
 	    $(rm) -f `echo view.0 | sed '$(transform)'`
-	@cd $(DESTDIR)$(mandir)/cat1 && $(ln) \
+	@cd $(DESTDIR)$(mandir)/cat1 && $(LN_S) \
 	    `echo vi.0 | sed '$(transform)'` `echo ex.0 | sed '$(transform)'`
-	@cd $(DESTDIR)$(mandir)/cat1 && $(ln) \
+	@cd $(DESTDIR)$(mandir)/cat1 && $(LN_S) \
 	    `echo vi.0 | sed '$(transform)'` `echo view.0 | sed '$(transform)'`
 	@[ -d $(DESTDIR)$(mandir)/man1 ] || \
 	    ($(mkdir) $(DESTDIR)$(mandir)/man1 && \
@@ -1729,9 +1729,9 @@ install-data-local: install_common
 	    $(rm) -f `echo ex.1 | sed '$(transform)'`
 	@cd $(DESTDIR)$(mandir)/man1 && \
 	    $(rm) -f `echo view.1 | sed '$(transform)'`
-	@cd $(DESTDIR)$(mandir)/man1 && $(ln) \
+	@cd $(DESTDIR)$(mandir)/man1 && $(LN_S) \
 	    `echo vi.1 | sed '$(transform)'` `echo ex.1 | sed '$(transform)'`
-	@cd $(DESTDIR)$(mandir)/man1 && $(ln) \
+	@cd $(DESTDIR)$(mandir)/man1 && $(LN_S) \
 	    `echo vi.1 | sed '$(transform)'` `echo view.1 | sed '$(transform)'`
 install_common:
 	@[ -f $(chmod) ] || (echo 'chmod not found.'; exit 1)
