--- libstreamanalyzer/plugins/indexers/cluceneindexer/cluceneindexreader.cpp	2013-02-05 22:34:52.000000000 +0100
+++ libstreamanalyzer/plugins/indexers/cluceneindexer/cluceneindexreader.cpp	2016-09-15 09:35:02.672643338 +0200
@@ -369,8 +369,8 @@
         string size = value;
         doc.size = atoi(size.c_str());
     } else {
-        doc.properties.insert(make_pair<const string, string>(
-            wchartoutf8(name), value));
+        string uname(wchartoutf8(name));
+        doc.properties.insert(make_pair<const std::string&, std::string&>(uname, value));
     }
 }
 Variant
--- libstreamanalyzer/plugins/indexers/clucenengindexer/cluceneindexreader.cpp	2013-02-05 22:34:52.000000000 +0100
+++ libstreamanalyzer/plugins/indexers/clucenengindexer/cluceneindexreader.cpp	2016-09-15 09:39:01.049769777 +0200
@@ -321,8 +321,8 @@
         string size = value;
         doc.size = atoi(size.c_str());
     } else {
-        doc.properties.insert(make_pair<const string, string>(
-            wchartoutf8(name), value));
+        string uname(wchartoutf8(name));
+        doc.properties.insert(make_pair<const std::string&, std::string&>(uname, value));
     }
 }
 Variant
