--- a/language/update.py
+++ b/language/update.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python2
+#!/usr/bin/python3
 # usage: call here to bring all translations up to date
 # usage forms:
 # update.py [--complete] <list of language files>
@@ -203,7 +203,7 @@
         # read dictionary
         self.ReadDictionary( file )
         # store it
-        self.translations[string.split(file,".")[0]] = self.dictionary
+        self.translations[file.split(".")[0]] = self.dictionary
         del self.dictionary
 
 if __name__ == "__main__":
