From 01c1e4bbeefaedcdcebde26a6f580d1c4bd5bc9a Mon Sep 17 00:00:00 2001
From: Nathan Owens <ndowens@artixlinux.org>
Date: Mon, 23 Nov 2020 12:45:59 -0600
Subject: [PATCH] gir

---
 gdk-pixbuf/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git gdk-pixbuf/meson.build gdk-pixbuf/meson.build
index d9a254f..fd26643 100644
--- gdk-pixbuf/meson.build
+++ gdk-pixbuf/meson.build
@@ -182,7 +182,7 @@ gdkpixbuf = library('gdk_pixbuf-2.0',
                     install: true)
 
 gir = find_program('g-ir-scanner', required : get_option('introspection'))
-build_gir = gir.found() and (not meson.is_cross_build() or get_option('introspection').enabled())
+build_gir = gir.found() or get_option('introspection').enabled()
 
 if build_gir
   gir_args = [
-- 
2.29.2

