diff --git a/tests/pixbuf-jpeg.c b/tests/pixbuf-jpeg.c
index 74e3e6c54..7f10148f1 100644
--- a/tests/pixbuf-jpeg.c
+++ b/tests/pixbuf-jpeg.c
@@ -94,7 +94,11 @@ test_bug_775218 (void)
     }
 
   ref = gdk_pixbuf_new_from_file (g_test_get_filename (G_TEST_DIST, "bug775218.jpg", NULL), &error);
+#if GLIB_SIZEOF_VOID_P == 8
   g_assert_error (error, GDK_PIXBUF_ERROR, GDK_PIXBUF_ERROR_CORRUPT_IMAGE);
+#else
+  g_assert_error (error, GDK_PIXBUF_ERROR, GDK_PIXBUF_ERROR_INSUFFICIENT_MEMORY);
+#endif
   g_error_free (error);
   g_clear_object (&ref);
 }
