From 0a3e42e8292b89db602757510e367766685cc38b Mon Sep 17 00:00:00 2001
From: Dima Pasechnik <dima@pasechnik.info>
Date: Fri, 12 Dec 2025 18:18:11 -0600
Subject: [PATCH] correct cypari2.py(.in) -> cypari2.pc(.in)

---
 cypari2/meson.build | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/cypari2/meson.build b/cypari2/meson.build
index 4c37553..de1257a 100644
--- a/cypari2/meson.build
+++ b/cypari2/meson.build
@@ -49,8 +49,8 @@ endforeach
 config_data = configuration_data()
 config_data.set('version', meson.project_version())
 configure_file(
   input: 'cypari2.py.in',
-  output: 'cypari2.py',
+  output: 'cypari2.pc',
   configuration: config_data,
   install: true,
   install_dir: py.get_install_dir() / 'cypari2'
