From 796b90d19442d2f704fd2916457601cf6c65f0d9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Gonzalo=20Tornar=C3=ADa?= <tornaria@cmat.edu.uy>
Date: Wed, 30 Oct 2024 17:52:04 -0300
Subject: [PATCH] fix implicit noexcept warnings

---
 cypari2/custom_block.pyx | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/cypari2/custom_block.pyx b/cypari2/custom_block.pyx
index ddbb829..67419f8 100644
--- a/cypari2/custom_block.pyx
+++ b/cypari2/custom_block.pyx
@@ -12,14 +12,14 @@ from cysignals.signals cimport add_custom_signals
 cdef extern from "pari/pari.h":
     int     PARI_SIGINT_block, PARI_SIGINT_pending
 
-cdef int custom_signal_is_blocked():
+cdef int custom_signal_is_blocked() noexcept:
     return PARI_SIGINT_block
 
-cdef void custom_signal_unblock():
+cdef void custom_signal_unblock() noexcept:
     global PARI_SIGINT_block
     PARI_SIGINT_block = 0
 
-cdef void custom_set_pending_signal(int sig):
+cdef void custom_set_pending_signal(int sig) noexcept:
     global PARI_SIGINT_pending
     PARI_SIGINT_pending = sig
 
