commit 8ef356a4eb936c37f55a5c501f3a955e6740c0c5
Author: Gonzalo Tornaría <tornaria@cmat.edu.uy>
Date:   Wed Jul 19 19:45:23 2023 -0300

    cython3 support using legacy directives

diff --git a/cypari2/gen.pyx b/cypari2/gen.pyx
index 247b1ad..75050a0 100644
--- a/cypari2/gen.pyx
+++ b/cypari2/gen.pyx
@@ -329,7 +329,7 @@ cdef class Gen(Gen_base):
         >>> pari = Pari()
         >>> L = pari("vector(10,i,i^2)")
         >>> L.__iter__()
-        <generator object at ...>
+        <...generator object at ...>
         >>> [x for x in L]
         [1, 4, 9, 16, 25, 36, 49, 64, 81, 100]
         >>> list(L)
diff --git a/setup.py b/setup.py
index 2188711..455337f 100755
--- a/setup.py
+++ b/setup.py
@@ -36,6 +36,8 @@ class build_ext(_build_ext):
             "binding": True,
             "cdivision": True,
             "language_level": 2,
+            "legacy_implicit_noexcept": True,
+            "c_api_binop_methods": True,
         }
 
         _build_ext.finalize_options(self)
