diff --git a/ext/pdo_pgsql/config.m4 b/ext/pdo_pgsql/config.m4
index 80ffd97..d8cad47 100644
--- a/ext/pdo_pgsql/config.m4
+++ b/ext/pdo_pgsql/config.m4
@@ -20,7 +20,7 @@ if test "$PHP_PDO_PGSQL" != "no"; then
     fi
   done
 
-  if test -n "$PG_CONFIG"; then
+  if test -n "$PG_CONFIG" && test "x$cross_compiling" != "xyes"; then
     AC_MSG_RESULT([$PG_CONFIG])
     PGSQL_INCLUDE=`$PG_CONFIG --includedir`
     PGSQL_LIBDIR=`$PG_CONFIG --libdir`
diff --git a/ext/pgsql/config.m4 b/ext/pgsql/config.m4
index 669a9f7..7e5bf7f 100644
--- a/ext/pgsql/config.m4
+++ b/ext/pgsql/config.m4
@@ -18,7 +18,7 @@ if test "$PHP_PGSQL" != "no"; then
     fi
   done
 
-  if test -n "$PG_CONFIG"; then
+  if test -n "$PG_CONFIG" && test "x$cross_compiling" != "xyes"; then
     AC_MSG_RESULT([$PG_CONFIG])
     PGSQL_INCLUDE=`$PG_CONFIG --includedir`
     PGSQL_LIBDIR=`$PG_CONFIG --libdir`
