--- a/build/php.m4
+++ b/build/php.m4
@@ -1219,7 +1219,7 @@ AC_DEFUN([PHP_PWRITE_TEST],[
   AC_CACHE_CHECK(whether pwrite works,ac_cv_pwrite,[
     PHP_DOES_PWRITE_WORK
     if test "$ac_cv_pwrite" = "no"; then
-      PHP_DOES_PWRITE_WORK([ssize_t pwrite(int, void *, size_t, off64_t);])
+      PHP_DOES_PWRITE_WORK([ssize_t pwrite(int, void *, size_t, off_t);])
       if test "$ac_cv_pwrite" = "yes"; then
         ac_cv_pwrite=64
       fi
@@ -1241,7 +1241,7 @@ AC_DEFUN([PHP_PREAD_TEST],[
   AC_CACHE_CHECK(whether pread works,ac_cv_pread,[
     PHP_DOES_PREAD_WORK
     if test "$ac_cv_pread" = "no"; then
-      PHP_DOES_PREAD_WORK([ssize_t pread(int, void *, size_t, off64_t);])
+      PHP_DOES_PREAD_WORK([ssize_t pread(int, void *, size_t, off_t);])
       if test "$ac_cv_pread" = "yes"; then
         ac_cv_pread=64
       fi
@@ -1449,7 +1449,7 @@ AC_RUN_IFELSE([AC_LANG_SOURCE([[
 #include <stdlib.h>
 
 struct cookiedata {
-  off64_t pos;
+  off_t pos;
 };
 
 ssize_t reader(void *cookie, char *buffer, size_t size)
@@ -1458,7 +1458,7 @@ ssize_t writer(void *cookie, const char
 { return size; }
 int closer(void *cookie)
 { return 0; }
-int seeker(void *cookie, off64_t *position, int whence)
+int seeker(void *cookie, off_t *position, int whence)
 { ((struct cookiedata*)cookie)->pos = *position; return 0; }
 
 cookie_io_functions_t funcs = {reader, writer, seeker, closer};
--- a/main/streams/cast.c
+++ b/main/streams/cast.c
@@ -101,7 +101,7 @@ static ssize_t stream_cookie_writer(void
 }
 
 # ifdef COOKIE_SEEKER_USES_OFF64_T
-static int stream_cookie_seeker(void *cookie, off64_t *position, int whence)
+static int stream_cookie_seeker(void *cookie, off_t *position, int whence)
 {
 
 	*position = php_stream_seek((php_stream *)cookie, (zend_off_t)*position, whence);
--- a/main/php.h
+++ b/main/php.h
@@ -285,11 +285,11 @@ extern char **environ;
 #endif	/* ifndef PHP_WIN32 */
 
 #ifdef PHP_PWRITE_64
-ssize_t pwrite(int, void *, size_t, off64_t);
+ssize_t pwrite(int, void *, size_t, off_t);
 #endif
 
 #ifdef PHP_PREAD_64
-ssize_t pread(int, void *, size_t, off64_t);
+ssize_t pread(int, void *, size_t, off_t);
 #endif
 
 BEGIN_EXTERN_C()
