diff --git a/php.ini-production b/php.ini-production
index b39a960..68f14a6 100644
--- a/php.ini-production
+++ b/php.ini-production
@@ -737,7 +737,7 @@ default_charset = "UTF-8"
 ;;;;;;;;;;;;;;;;;;;;;;;;;
 
 ; UNIX: "/path1:/path2"
-;include_path = ".:/php/includes"
+include_path = ".:/usr/share/pear"
 ;
 ; Windows: "\path1;\path2"
 ;include_path = ".;c:\php\includes"
@@ -760,9 +760,7 @@ user_dir =
 
 ; Directory in which the loadable extensions (modules) reside.
 ; https://php.net/extension-dir
-;extension_dir = "./"
-; On windows:
-;extension_dir = "ext"
+extension_dir = "/usr/lib/php8.1/modules/"
 
 ; Directory where the temporary files should be placed.
 ; Defaults to the system default (see sys_get_temp_dir)
