--- src/slib.c.orig
+++ src/slib.c
@@ -14921,7 +14921,7 @@
  * Makefile).
  */
 extern	struct kwval *kw2val_lookup(register const char *str,
-    register unsigned int len);
+    register size_t len);
 #include "kw2val_lookup.c"
 
 #define	notKeyword -1
--- src/cmd.pl.orig
+++ src/cmd.pl
@@ -63,7 +63,7 @@
 	u8	remote:1;	/* always allowed as a remote command */
 } CMD;
 
-CMD	*cmd_lookup(const char *str, unsigned int len);
+CMD	*cmd_lookup(const char *str, size_t len);
 
 END
 
