Dynamically link pzstd against libzstd.so.

--- a/contrib/pzstd/Makefile
+++ b/contrib/pzstd/Makefile
@@ -167,8 +167,8 @@
 	$(TESTPROG) ./test/RoundTripTest$(EXT) $(TESTFLAGS)
 
 # Build the main binary
-pzstd$(EXT): main.o $(PROGDIR)/util.o Options.o Pzstd.o SkippableFrame.o $(ZSTDDIR)/libzstd.a
-	$(LD_COMMAND)
+pzstd$(EXT): main.o $(PROGDIR)/util.o Options.o Pzstd.o SkippableFrame.o
+	$(LD_COMMAND) -L../../lib -lzstd
 
 # Target that depends on all the tests
 .PHONY: tests
