From 6a74ec7a8747c09d77d4fed6feda2530feca5d5c Mon Sep 17 00:00:00 2001
From: maxice8 <thinkabit.ukim@gmail.com>
Date: Wed, 23 Jan 2019 01:37:29 -0200
Subject: [PATCH] add missing include <sys/types.h> for ssize_t

---
 parser.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/parser.c b/parser.c
index a4e3d78..3e3e25f 100644
--- parser.c
+++ parser.c
@@ -2,6 +2,7 @@
 #include <stddef.h>
 #include <ctype.h>
 #include <string.h>
+#include <sys/types.h>
 
 #include "internal.h"
 #include "evhtp/parser.h"
-- 
2.20.1


