--- src/xfpm-notify.c.orig	2011-02-28 23:43:55.521921469 +0100
+++ src/xfpm-notify.c	2011-02-28 23:44:16.343032607 +0100
@@ -209,14 +209,11 @@ xfpm_notify_new_notification_internal (c
 {
     NotifyNotification *n;
     
-    n = notify_notification_new (title, message, NULL, NULL);
+    n = notify_notification_new (title, message, NULL);
     
     if ( icon_name )
     	xfpm_notify_set_notification_icon (n, icon_name);
 	
-    if ( icon )
-    	notify_notification_attach_to_status_icon (n, icon);
-	
     notify_notification_set_urgency (n, (NotifyUrgency)urgency);
     
     if ( timeout != 0)
