diff --git a/utils/fstreewalk.cpp b/utils/fstreewalk.cpp
index fc43edc..5e28d6e 100644
--- a/utils/fstreewalk.cpp
+++ b/utils/fstreewalk.cpp
@@ -17,6 +17,7 @@
 
 #include "autoconfig.h"
 
+#include <sys/types.h>
 #include <stdio.h>
 
 #include <errno.h>
diff --git a/utils/pxattr.cpp b/utils/pxattr.cpp
index 3a1f8f5..dcb93fc 100644
--- a/utils/pxattr.cpp
+++ b/utils/pxattr.cpp
@@ -46,7 +46,7 @@
 
 // Not exactly true for win32, but makes my life easier by avoiding ifdefs in recoll (the calls just
 // fail, which is expected)
-#if defined(__DragonFly__) || defined(__OpenBSD__) || defined(_WIN32)
+#if defined(__DragonFly__) || defined(__OpenBSD__) || defined(_WIN32) || !defined(__GLIBC__)
 #define HAS_NO_XATTR
 #endif

diff --git a/common/conf_post.h b/common/conf_post.h
index 7093ead..0ecd56b 100644
--- a/common/conf_post.h
+++ b/common/conf_post.h
@@ -74,12 +74,4 @@ typedef int ssize_t;
 #define REAL_GCC   __GNUC__ // probably
 #endif
 
-#if defined(REAL_GCC) && defined(HAVE_FEATURES_H)
-// Older gcc versions pretended to supply std::regex, but the resulting programs mostly crashed.
-#include <features.h>
-#if ! __GNUC_PREREQ(6,0)
-#define NO_STD_REGEX 1
-#endif
-#endif
-
 #endif /* INCLUDED */
