--- got-portable-0.69/configure.ac.orig
+++ got-portable-0.69/configure.ac
@@ -204,6 +204,7 @@
 		[return (reallocarray(NULL, 1, 1) == NULL);]
 	)],
 	AC_MSG_RESULT(yes),
+	[AC_LIBOBJ(reallocarray) AC_MSG_RESULT([no])],
 	[AC_LIBOBJ(reallocarray) AC_MSG_RESULT([no])]
 )
 AC_MSG_CHECKING([for working recallocarray])
@@ -212,6 +213,7 @@
 		[return (recallocarray(NULL, 1, 1, 1) == NULL);]
 	)],
 	AC_MSG_RESULT(yes),
+	[AC_LIBOBJ(recallocarray) AC_MSG_RESULT([no])],
 	[AC_LIBOBJ(recallocarray) AC_MSG_RESULT([no])]
 )
 
