Use release-type parameters even with buildtype=plain.

--- a/src/lib/eina/meson.build
+++ b/src/lib/eina/meson.build
@@ -205,7 +205,7 @@ if get_option('buildtype') == 'debug'
      debug_threads=true
      default_mempool=true
      want_cow_magic=true
-elif get_option('buildtype') == 'release'
+else
      with_max_log_level=3
      stringshare_usage=false
      use_valgrind=false
@@ -213,14 +213,6 @@ elif get_option('buildtype') == 'release'
      debug_threads=false
      default_mempool=false
      want_cow_magic=false
- else
-     with_max_log_level=-1
-     stringshare_usage=false
-     use_valgrind=true
-     debug_malloc=false
-     debug_threads=false
-     default_mempool=false
-     want_cow_magic=false
 endif
 
 if with_max_log_level != -1
