--- a/git-annex.cabal	2020-04-01 21:09:23.653315424 +0200
+++ b/git-annex.cabal	2020-04-01 20:50:23.080838029 +0200
@@ -304,6 +304,7 @@
   Build-Depends:
    base (>= 4.11.1.0 && < 5.0),
    network-uri (>= 2.6),
+   network-bsd,
    optparse-applicative (>= 0.14.1),
    containers (>= 0.5.8),
    exceptions (>= 0.6),
@@ -404,3 +404,3 @@ if flag(NetworkBSD)
     Build-Depends: network-bsd, network (>= 3.0.0.0)
   else
-    Build-Depends: network (< 3.0.0.0), network (>= 2.6.3.0)
+    Build-Depends: network (< 3.2.0.0), network (>= 2.6.3.0)

   if flag(S3)
     Build-Depends: aws (>= 0.20)
