diff --git src/graphene-ray.c src/graphene-ray.c
index 66c3393..9151300 100644
--- src/graphene-ray.c
+++ src/graphene-ray.c
@@ -563,7 +563,7 @@ graphene_ray_intersect_box (const graphene_ray_t *r,
 #else
   if (ty_min > tx_min || fpclassify (tx_min) == FP_NAN)
     tx_min = ty_min;
-  if (ty_max > tx_max || fpclassify (tx_max) == FP_NAN)
+  if (ty_max < tx_max || fpclassify (tx_max) == FP_NAN)
     tx_max = ty_max;
 #endif
