diff -Nur a/po/POTFILES.in b/po/POTFILES.in
--- a/po/POTFILES.in	2017-06-05 18:35:51.000000000 +0200
+++ b/po/POTFILES.in	2022-06-13 02:32:35.622115937 +0200
@@ -1,10 +1,3 @@
 [encoding: UTF-8]
-src/unity_facebook_daemon.py
-src/unity_picasa_daemon.py
 src/unity_shotwell_daemon.py
-src/unity_flickr_daemon.py
-[type: gettext/ini]data/flickr.scope.in
-[type: gettext/ini]data/facebook.scope.in
 [type: gettext/ini]data/shotwell.scope.in
-[type: gettext/ini]data/picasa.scope.in
-[type: gettext/xml]unity-lens-photos.application.in
diff -Nur a/setup.cfg b/setup.cfg
--- a/setup.cfg	2017-06-05 18:35:51.000000000 +0200
+++ b/setup.cfg	2022-06-13 01:46:47.286337106 +0200
@@ -3,9 +3,4 @@
 
 [build_i18n]
 domain=unity-lens-photos
-desktop_files=[("share/unity/scopes/photos", ("data/facebook.scope.in",
-                                              "data/flickr.scope.in",
-                                              "data/picasa.scope.in",
-                                              "data/shotwell.scope.in",))]
-xml_files_no_trans=[("share/accounts/applications", ("unity-lens-photos.application.in",))]
-
+desktop_files=[("share/unity/scopes/photos", ("data/shotwell.scope.in",))]
diff -Nur a/setup.py b/setup.py
--- a/setup.py	2017-06-05 18:35:51.000000000 +0200
+++ b/setup.py	2022-06-13 01:46:47.286337106 +0200
@@ -12,13 +12,7 @@
       license="GNU General Public License (GPL)",
       data_files=[
     ('share/unity-scopes/shotwell', ['src/unity_shotwell_daemon.py']),
-    ('share/unity-scopes/flickr', ['src/unity_flickr_daemon.py']),
-    ('share/unity-scopes/facebook', ['src/unity_facebook_daemon.py']),
-    ('share/unity-scopes/picasa', ['src/unity_picasa_daemon.py']),
     ('share/dbus-1/services', ['data/unity-scope-shotwell.service']),
-    ('share/dbus-1/services', ['data/unity-scope-flickr.service']),
-    ('share/dbus-1/services', ['data/unity-scope-facebook.service']),
-    ('share/dbus-1/services', ['data/unity-scope-picasa.service']),
     ('share/applications', ['unity-lens-photos.desktop']),
     ('share/pixmaps', ['unity-lens-photos.png']),
     ], cmdclass={"build":  build_extra.build_extra,
