From d650df8af80b734f6c44a0acb2080493330b8836 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Wed, 23 Apr 2014 13:20:37 +0000
Subject: lib: Fix typo in g-i annotation

Missing colon.
---
diff --git a/libupower-glib/up-client.c b/libupower-glib/up-client.c
index 17fb02d..582fac9 100644
--- a/libupower-glib/up-client.c
+++ b/libupower-glib/up-client.c
@@ -131,7 +131,7 @@ up_client_get_devices (UpClient *client)
  * @client: a #UpClient instance.
  *
  * Get the composite display device.
- * Return value: (transfer full) a #UpClient object, or %NULL on error.
+ * Return value: (transfer full): a #UpClient object, or %NULL on error.
  *
  * Since: 1.0
  **/
--
cgit v0.9.0.2-2-gbebe
