diff --git libAfterImage/configure.in libAfterImage/configure.in
index ca49a50..780e0ff 100644
--- a/libAfterImage/configure.in
+++ b/libAfterImage/configure.in
@@ -590,21 +590,21 @@ else
 					  )
 	  if test "x$have_ft2build" = "xno" ; then
     	  	if test "x$have_freetype_freetype" = "xyes" ; then
-			AC_TRY_RUN([#include <freetype/freetype.h>
+			AC_TRY_COMPILE([#include <freetype/freetype.h>
                             	int main(){FT_Face f; return 0;}],
 	                   	[have_ttf=yes],AC_MSG_RESULT(no))
 	  	elif test "x$have_freetype" = "xyes" ; then
-			AC_TRY_RUN([#include <freetype.h>
+			AC_TRY_COMPILE([#include <freetype.h>
                             	int main(){FT_Face f; return 0;}],
 	                   	[have_ttf=yes],AC_MSG_RESULT(no))
 	  	fi;
 	  elif test "x$have_freetype_freetype" = "xyes" ; then
-		AC_TRY_RUN([#include <ft2build.h> 
+		AC_TRY_COMPILE([#include <ft2build.h>
 			    #include <freetype/freetype.h>
                             int main(){FT_Face f; return 0;}],
 	                [have_ttf=yes],AC_MSG_RESULT(no))
 	  elif test "x$have_freetype" = "xyes" ; then
-		AC_TRY_RUN([#include <ft2build.h>
+		AC_TRY_COMPILE([#include <ft2build.h>
 			#include <freetype.h>
                         int main(){FT_Face f; return 0;}],
 	                [have_ttf=yes],AC_MSG_RESULT(no))
