Index: bitcoin-22.0/src/wallet/test/init_tests.cpp
===================================================================
--- bitcoin-22.0.orig/src/wallet/test/init_tests.cpp
+++ bitcoin-22.0/src/wallet/test/init_tests.cpp
@@ -69,7 +69,7 @@ BOOST_AUTO_TEST_CASE(walletinit_verify_w
     BOOST_CHECK(result == true);
     fs::path walletdir = gArgs.GetArg("-walletdir", "");
     fs::path expected_path = fs::canonical(m_walletdir_path_cases["default"]);
-    BOOST_CHECK_EQUAL(walletdir, expected_path);
+    BOOST_TEST(fs::equivalent(walletdir, expected_path));
 }
 
 BOOST_AUTO_TEST_CASE(walletinit_verify_walletdir_no_trailing2)
@@ -79,7 +79,7 @@ BOOST_AUTO_TEST_CASE(walletinit_verify_w
     BOOST_CHECK(result == true);
     fs::path walletdir = gArgs.GetArg("-walletdir", "");
     fs::path expected_path = fs::canonical(m_walletdir_path_cases["default"]);
-    BOOST_CHECK_EQUAL(walletdir, expected_path);
+    BOOST_TEST(fs::equivalent(walletdir, expected_path));
 }
 
 BOOST_AUTO_TEST_SUITE_END()
