--- src/Makefile~	2020-03-15 17:22:29.077747681 +0100
+++ src/Makefile	2020-03-15 17:22:32.599747681 +0100
@@ -3,7 +3,9 @@
 # Complie C source codes and output object files to $(BUILD_DIR), then links
 # them together to get a executable $(TARGET).
 
-CC = gcc
+ifeq "$(origin CC)" "default"
+	CC = gcc
+endif
 DEFS = -D PACKAGE='"$(PACKAGE)"' \
 	   -D PACKAGE_NAME='"$(PACKAGE_NAME)"' \
 	   -D PACKAGE_VERSION='"$(PACKAGE_VERSION)"' \
