commit ab3daa606e215f43aa3077daad82cd2d55c38908
Author: q66 <daniel@octaforge.org>
Date:   Mon Jan 6 03:02:59 2020 +0100

    Fix build on big endian

diff --git src/audio/wav_sound_file.cpp src/audio/wav_sound_file.cpp
index c4036b1..e056f50 100644
--- src/audio/wav_sound_file.cpp
+++ src/audio/wav_sound_file.cpp
@@ -150,7 +150,7 @@ WavSoundFile::read(void* buffer, size_t buffer_size)
     throw SoundError("read error while reading samples");
 
 #ifdef WORDS_BIGENDIAN
-  if (bits_per_sample != 16)
+  if (m_bits_per_sample != 16)
     return readsize;
   char *tmp = (char*)buffer;
 
