--- a/src/ids/Makefile	2021-10-01 13:55:16.000000000 +0200
+++ b/src/ids/Makefile	2021-10-26 13:05:51.241897833 +0200
@@ -56,8 +56,8 @@
 hd_ids.c: hd_ids.h hd_ids_tiny.h
 
 hd_ids.h hd.ids: check_hd $(IDFILES)
-	./check_hd --check --sort --cfile hd_ids.h $(IDFILES)
+	check_hd --check --sort --cfile hd_ids.h $(IDFILES)
 
 hd_ids_tiny.h: check_hd hd.ids
-	./check_hd --mini --cfile hd_ids_tiny.h --log=hd_tiny.log --out=hd_tiny.ids hd.ids
+	check_hd --mini --cfile hd_ids_tiny.h --log=hd_tiny.log --out=hd_tiny.ids hd.ids
 
