--- a/lib/portage/tests/resolver/ResolverPlayground.py	2026-01-23 22:09:12.000000000 +0100
+++ b/lib/portage/tests/resolver/ResolverPlayground.py	2026-09-03 00:09:26.662475655 +0200
@@ -182,7 +182,7 @@
                 "rm",
                 "sed",
                 "sort",
-                "gtar",
+                "tar",
                 "tr",
                 "uname",
                 "uniq",
--- a/misc/emerge-delta-webrsync	2026-01-23 22:09:12.000000000 +0100
+++ b/misc/emerge-delta-webrsync	2026-09-03 00:14:44.135856633 +0200
@@ -278,7 +278,7 @@
 		*.gz)   decompressor="zcat"  ;;
 		*)      decompressor="cat"   ;;
 	esac
-	${decompressor} "${file}" | gtar "$@"
+	${decompressor} "${file}" | tar "$@"
 	_pipestatus=${PIPESTATUS[*]}
 	[[ ${_pipestatus// /} -eq 0 ]]
 }
--- a/lib/portage/dbapi/vartree.py	2026-01-23 22:09:12.000000000 +0100
+++ b/lib/portage/dbapi/vartree.py	2026-09-03 00:16:19.659369285 +0200
@@ -1110,7 +1110,7 @@
             "BINPKG_FORMAT", SUPPORTED_GENTOO_BINPKG_FORMATS[0]
         )
         if binpkg_format == "xpak":
-            tar_cmd = ("gtar", "-x", "--xattrs", "--xattrs-include=*", "-C", dest_dir)
+            tar_cmd = ("tar", "-x", "--xattrs", "--xattrs-include=*", "-C", dest_dir)
             pr, pw = multiprocessing.Pipe(duplex=False)
             proc = await asyncio.create_subprocess_exec(*tar_cmd, stdin=pr)
             pr.close()
--- a/lib/_emerge/BinpkgExtractorAsync.py	2026-01-23 22:09:12.000000000 +0100
+++ b/lib/_emerge/BinpkgExtractorAsync.py	2026-09-03 00:17:39.637959037 +0200
@@ -39,7 +39,7 @@
         tar_options = ""
         if "xattr" in self.features:
             process = subprocess.Popen(
-                ["gtar", "--help"], stdout=subprocess.PIPE, stderr=subprocess.PIPE
+                ["tar", "--help"], stdout=subprocess.PIPE, stderr=subprocess.PIPE
             )
             output = process.communicate()[0]
             if b"--xattrs" in output:
@@ -121,7 +121,7 @@
                 f"""
                     cmd0=(head -c {pkg_xpak.filestat.st_size - pkg_xpak.xpaksize} -- {shlex.quote(self.pkg_path)})
                     cmd1=({decomp_cmd})
-                    cmd2=(gtar -xp {tar_options} -C {shlex.quote(self.image_dir)} -f -);
+                    cmd2=(tar -xp {tar_options} -C {shlex.quote(self.image_dir)} -f -);
                 """
                 """
                     "${cmd0[@]}" | "${cmd1[@]}" | "${cmd2[@]}";
--- a/bin/phase-helpers.sh	2026-01-23 22:09:12.000000000 +0100
+++ b/bin/phase-helpers.sh	2026-09-03 00:19:12.204484218 +0200
@@ -488,17 +488,17 @@
 				unrar x -idq -o+ "${srcdir}${f}"
 				;;
 			tar.bz|tar.bz2|tbz|tbz2)
-				gtar -I "${bzip2_cmd-bzip2} -c" -xof "${srcdir}${f}"
+				tar -I "${bzip2_cmd-bzip2} -c" -xof "${srcdir}${f}"
 				;;
 			tar|tar.*|tgz)
 				# GNU tar recognises various file suffixes, for
 				# which it is able to execute the appropriate
 				# decompressor. They are documented by the
 				# (info) manual for the -a option.
-				gtar --warning=decompress-program -xof "${srcdir}${f}"
+				tar --warning=decompress-program -xof "${srcdir}${f}"
 				;;
 			txz)
-				gtar -xJof "${srcdir}${f}"
+				tar -xJof "${srcdir}${f}"
 				;;
 			xz)
 				xz -dc -- "${srcdir}${f}" > "${basename%.*}"
--- a/bin/misc-functions.sh	2026-01-23 22:09:12.000000000 +0100
+++ b/bin/misc-functions.sh	2026-09-03 00:20:33.681066285 +0200
@@ -582,14 +582,14 @@
 
 		[[ ${PORTAGE_VERBOSE} = 1 ]] && tar_options+=" -v"
 		if contains_word xattr "${FEATURES}" \
-			&& gtar --help 2>/dev/null | grep -q -- --xattrs
+			&& tar --help 2>/dev/null | grep -q -- --xattrs
 		then
 			tar_options+=" --xattrs"
 		fi
 
 		[[ -z "${PORTAGE_COMPRESSION_COMMAND}" ]] && die "PORTAGE_COMPRESSION_COMMAND is unset"
 
-		gtar ${tar_options} -cf - ${PORTAGE_BINPKG_TAR_OPTS} -C "${D}" . | \
+		tar ${tar_options} -cf - ${PORTAGE_BINPKG_TAR_OPTS} -C "${D}" . | \
 			${PORTAGE_COMPRESSION_COMMAND} > "${PORTAGE_BINPKG_TMPFILE}"
 		__pipestatus \
 			|| die "failed to pack binary package: '${PORTAGE_BINPKG_TMPFILE}'"
@@ -636,7 +636,7 @@
 	mkdir -p "${sources_dir}"
 	declare -a tar_args=("${EBUILD}")
 	[[ -d ${FILESDIR} ]] && tar_args=("${EBUILD}" "${FILESDIR}")
-	gtar czf "${sources_dir}/${PF}.tar.gz" \
+	tar czf "${sources_dir}/${PF}.tar.gz" \
 		"${tar_args[@]}" || \
 		die "Failed to create base rpm tarball."
 
