--- a/meson.build
+++ b/meson.build
@@ -16,7 +16,7 @@
 cc = meson.get_compiler('c')
 
 # Check if the build platform supports using malloc_trim
-malloc_trim_support_result = cc.run(
+if cc.compiles(
   '''
     #include <malloc.h>
 
@@ -27,15 +27,13 @@
   ''',
   name : 'malloc_trim is supported on the build platform',
 )
-
-if malloc_trim_support_result.compiled() and malloc_trim_support_result.returncode() == 0
   have_malloc_trim = true
 else
   have_malloc_trim = false
 endif
 
 # Check if the build platform supports using fanotify without the CAP_SYS_ADMIN capability
-fanotify_support_result = cc.run(
+if cc.compiles(
   '''
     #include <sys/fanotify.h>
     #include <fcntl.h>
@@ -47,15 +45,13 @@
   ''',
   name : 'fanotify is supported by the build platform and works without CAP_SYS_ADMIN capabilities',
 )
-
-if fanotify_support_result.compiled() and fanotify_support_result.returncode() == 0
   have_fanotify = true
 else
   have_fanotify = false
 endif
 
 # Check if the build platform supports inotify
-inotify_support_result = cc.run(
+if cc.compiles(
   '''
     #include <sys/inotify.h>
     #include <fcntl.h>
@@ -67,8 +63,6 @@
   ''',
   name : 'inotify is supported by the build platform',
 )
-
-if inotify_support_result.compiled() and inotify_support_result.returncode() == 0
   have_inotify = true
 else
   have_inotify = false
