--- lib/gvpr/Makefile.in.orig	2014-02-08 10:35:01.782106087 +0100
+++ lib/gvpr/Makefile.in	2014-02-08 10:35:20.629222313 +0100
@@ -1063,8 +1063,7 @@ $(libgvpr_C_la_OBJECTS): gdefs.h
 gvpr.3.pdf: $(srcdir)/gvpr.3
 	- @GROFF@ -Tps -man $(srcdir)/gvpr.3 | @PS2PDF@ - - >gvpr.3.pdf
 
-gdefs.h: $(top_srcdir)/lib/gvpr/gprdata mkdefs
-	./mkdefs gdefs.h < $(top_srcdir)/lib/gvpr/gprdata
+gdefs.h:
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
