index 0f1dfba..2ec58df 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -87,8 +87,6 @@ else()
     add_definitions(/DMMAPPER_NO_MINIUPNPC)
 endif()
 
-add_subdirectory(external/glm)
-
 # Extract git branch and revision
 if(EXISTS "${PROJECT_SOURCE_DIR}/.git")
     find_package(Git)
index 86738ad..3c44c89 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -508,9 +508,6 @@ set_target_properties(
   UNITY_BUILD ${USE_UNITY_BUILD}
 )
 
-target_include_directories(mmapper SYSTEM PUBLIC ${GLM_INCLUDE_DIR})
-add_dependencies(mmapper glm)
-
 if(WIN32)
     target_link_libraries(mmapper PUBLIC ws2_32)
 endif()
index 0113b5f..1cba069 100644
--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -20,7 +20,6 @@ set(clock_SRCS
     )
 set(TestClock_SRCS testclock.cpp)
 add_executable(TestClock ${TestClock_SRCS} ${clock_SRCS})
-add_dependencies(TestClock glm)
 target_link_libraries(TestClock Qt5::Test coverage_config)
 set_target_properties(
   TestClock PROPERTIES
@@ -58,7 +57,6 @@ file(GLOB_RECURSE expandoracommon_SRCS
     )
 set(TestExpandoraCommon_SRCS testexpandoracommon.cpp)
 add_executable(TestExpandoraCommon ${TestExpandoraCommon_SRCS} ${expandoracommon_SRCS})
-add_dependencies(TestExpandoraCommon glm)
 target_link_libraries(TestExpandoraCommon Qt5::Test coverage_config)
 set_target_properties(
   TestExpandoraCommon PROPERTIES
@@ -93,7 +91,6 @@ set(parser_SRCS
     )
 set(TestParser_SRCS testparser.cpp)
 add_executable(TestParser ${TestParser_SRCS} ${parser_SRCS})
-add_dependencies(TestParser glm)
 target_link_libraries(TestParser Qt5::Test coverage_config)
 set_target_properties(
   TestParser PROPERTIES
@@ -118,7 +115,6 @@ set(proxy_SRCS
     )
 set(TestProxy_SRCS TestProxy.cpp)
 add_executable(TestProxy ${TestProxy_SRCS} ${proxy_SRCS})
-add_dependencies(TestProxy glm)
 target_link_libraries(TestProxy Qt5::Test coverage_config)
 set_target_properties(
   TestProxy PROPERTIES
@@ -138,7 +134,6 @@ set(mainwindow_SRCS
     )
 set(TestMainWindow_SRCS TestMainWindow.cpp)
 add_executable(TestMainWindow ${TestMainWindow_SRCS} ${mainwindow_SRCS})
-add_dependencies(TestMainWindow glm)
 target_link_libraries(TestMainWindow Qt5::Widgets Qt5::Network Qt5::Test coverage_config)
 set_target_properties(
   TestMainWindow PROPERTIES
@@ -164,7 +159,6 @@ set(global_SRCS
     )
 set(TestGlobal_SRCS TestGlobal.cpp)
 add_executable(TestGlobal ${TestGlobal_SRCS} ${global_SRCS})
-add_dependencies(TestGlobal glm)
 target_link_libraries(TestGlobal Qt5::Widgets Qt5::Test coverage_config)
 set_target_properties(
   TestGlobal PROPERTIES
@@ -200,7 +194,6 @@ set(adventure_SRCS
         )
 set(TestAdventure_SRCS testadventure.cpp testadventure.h)
 add_executable(TestAdventure ${TestAdventure_SRCS} ${adventure_SRCS})
-add_dependencies(TestAdventure glm)
 target_link_libraries(TestAdventure Qt5::Widgets Qt5::Network Qt5::Test coverage_config)
 set_target_properties(
         TestAdventure PROPERTIES
@@ -219,7 +212,6 @@ set(timer_test
 )
 set(TestTimer TestCTimers.cpp)
 add_executable(TestTimer ${timer_test} ${TestTimer})
-add_dependencies(TestTimer glm)
 target_link_libraries(TestTimer Qt5::Test coverage_config)
 set_target_properties(
         TestTimer PROPERTIES
@@ -238,7 +230,6 @@ set(room_mob
 )
 set(RoomMob roompanel/TestRoomMob.cpp)
 add_executable(TestRoomMob ${room_mob} ${RoomMob})
-add_dependencies(TestRoomMob glm)
 target_link_libraries(TestRoomMob Qt5::Test Qt5::Gui Qt5::Widgets coverage_config)
 set_target_properties(
         TestRoomMob PROPERTIES
@@ -260,7 +251,6 @@ set(room_mobs
 )
 set(RoomMobs roompanel/TestRoomMobs.cpp)
 add_executable(TestRoomMobs ${room_mobs} ${RoomMobs})
-add_dependencies(TestRoomMobs glm)
 target_link_libraries(TestRoomMobs Qt5::Test Qt5::Gui Qt5::Widgets coverage_config)
 set_target_properties(
         TestRoomMobs PROPERTIES
@@ -285,7 +275,6 @@ set(room_manager
 )
 set(RoomManager roompanel/TestRoomManager.cpp)
 add_executable(TestRoomManager ${room_manager} ${RoomManager})
-add_dependencies(TestRoomManager glm)
 target_link_libraries(TestRoomManager Qt5::Test Qt5::Gui Qt5::Widgets coverage_config)
 set_target_properties(
         TestRoomManager PROPERTIES
