Reason: Fixes local declarations when using musl

diff -ru conf/CMakeLists.txt conf/CMakeLists.txt
--- conf/CMakeLists.txt	2020-10-28 01:45:30.000000000 +0100
+++ conf/CMakeLists.txt	2020-12-29 09:49:13.435238905 +0100
@@ -26,7 +26,7 @@
 
 # install udev rules for CM108
 if(LINUX)
-  install(FILES "${CUSTOM_CONF_DIR}/99-direwolf-cmedia.rules" DESTINATION /etc/udev/rules.d/)
+  install(FILES "${CUSTOM_CONF_DIR}/99-direwolf-cmedia.rules" DESTINATION /usr/lib/udev/rules.d/)
 endif()
 
 install(FILES "${CMAKE_BINARY_DIR}/direwolf.conf" DESTINATION ${INSTALL_CONF_DIR})
diff -ru src/decode_aprs.c src/decode_aprs.c
--- src/decode_aprs.c	2020-10-28 01:45:30.000000000 +0100
+++ src/decode_aprs.c	2020-12-29 09:53:24.021103609 +0100
@@ -3930,7 +3930,7 @@
  * models before getting to the more generic APY.
  */
 
-#if defined(__WIN32__) || defined(__OpenBSD__) || defined(__FreeBSD__) || defined(__APPLE__)
+#if ! defined(__GLIBC__) || defined(__OpenBSD__) || defined(__FreeBSD__) || defined(__APPLE__)
 	    qsort (tocalls, num_tocalls, sizeof(struct tocalls_s), tocall_cmp);
 #else
 	    qsort (tocalls, num_tocalls, sizeof(struct tocalls_s), (__compar_fn_t)tocall_cmp);
diff -ru src/direwolf.h src/direwolf.h
--- src/direwolf.h	2020-10-28 01:45:30.000000000 +0100
+++ src/direwolf.h	2020-12-29 09:50:44.182848501 +0100
@@ -282,7 +282,7 @@
 char *strcasestr(const char *S, const char *FIND);
 
 
-#if defined(__OpenBSD__) || defined(__FreeBSD__) || defined(__APPLE__)
+#if ! defined(__GLIBC__) || defined(__OpenBSD__) || defined(__FreeBSD__) || defined(__APPLE__)
 
 // strlcpy and strlcat should be in string.h and the C library.
