diff --git src/qtterminal/qt_conversion.cpp src/qtterminal/qt_conversion.cpp
index 38fd177..95ee48c 100644
--- src/qtterminal/qt_conversion.cpp
+++ src/qtterminal/qt_conversion.cpp
@@ -126,7 +126,7 @@ QImage qt_imageToQImage(int M, int N, coordval* image, t_imagecolor color_mode)
 			QRgb* line = (QRgb*)(qimage.scanLine(n));
 			for (int m = 0; m < M; m++)
 			{
-				if (isnan(*image))
+				if (std::isnan(*image))
 				{
 					image++;
 					*line++ = 0x00000000;
