--- 15.7/common/lib/modules/fglrx/build_mod/firegl_public.c	2015-07-04 16:31:23.000000000 +0200
+++ 15.7a/common/lib/modules/fglrx/build_mod/firegl_public.c	2015-07-09 17:40:58.611608136 +0200
@@ -3498,7 +3498,11 @@
 #else
         //when MSI enabled. keep irq disabled when calling the action handler,
         //exclude this IRQ from irq balancing (only on one CPU) 
-        ((useMSI) ? (IRQF_DISABLED) : (IRQF_SHARED)),    
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4,1,0)
+        ((useMSI) ? (IRQF_NOBALANCING) : (IRQF_SHARED)),
+#else
+        ((useMSI) ? (IRQF_DISABLED) : (IRQF_SHARED)),
+#endif
 #endif
         dev_name,
         context);
