Use uapi/linux/version.h if linux/version.h not found.

--- common/lib/modules/fglrx/build_mod/make.sh.orig	2013-02-07 19:17:27.405645903 +0100
+++ common/lib/modules/fglrx/build_mod/make.sh	2013-02-07 19:19:05.549150417 +0100
@@ -203,11 +203,16 @@ cd $current_wd
 # sample: #define UTS_RELEASE "2.4.0-test7"
 
 src_file=$linuxincludes/linux/version.h
+src_file_new=$linuxincludes/generated/uapi/linux/version.h
 
 if [ ! -e $src_file ]; then
+  if [ ! -e $src_file_new ]; then
   echo "kernel includes at $linuxincludes not found or incomplete" | tee -a $logfile
   echo "file: $src_file"                                           | tee -a $logfile
   exit 1
+  else
+	  src_file=$src_file_new
+  fi
 fi
 
 OsRelease=${uname_r}
