From 596667daba1ed99eda048e491ff8f36200f09429 Mon Sep 17 00:00:00 2001
From: vlazzarini <victor.lazzarini@mu.ie>
Date: Sat, 3 Jun 2023 18:06:59 +0100
Subject: [PATCH] fixing liblo calls

---
 Opcodes/OSC.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Opcodes/OSC.c b/Opcodes/OSC.c
index 4868a04c5e0..b37f5dac1e6 100644
--- a/Opcodes/OSC.c
+++ b/Opcodes/OSC.c
@@ -483,7 +483,7 @@ static int32_t OSCcounter(CSOUND *csound, OSCcount *p)
 }
 
 static int32_t OSC_handler(const char *path, const char *types,
-                       lo_arg **argv, int32_t argc, void *data, void *p)
+                       lo_arg **argv, int32_t argc, lo_message data, void *p)
 {
     IGN(argc);  IGN(data);
     OSC_PORT  *pp = (OSC_PORT*) p;
@@ -548,7 +548,7 @@ static int32_t OSC_handler(const char *path, const char *types,
             case 'b':
               {
                 int32_t len =
-                  lo_blobsize((lo_blob*)argv[i]);
+                  lo_blobsize((lo_blob)argv[i]);
                 m->args[i].blob =
                   csound->Malloc(csound,len);
                 memcpy(m->args[i].blob, argv[i], len);

From 2a071ae8ca89bc21b5c80037f8c95a01bb670ac9 Mon Sep 17 00:00:00 2001
From: John ffitch <jpff@codemist.co.uk>
Date: Tue, 31 Oct 2023 15:11:03 +0000
Subject: [PATCH] type mismatch in OSC

---
 Opcodes/OSC.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Opcodes/OSC.c b/Opcodes/OSC.c
index c67c35e6c9e..ac271c62391 100644
--- a/Opcodes/OSC.c
+++ b/Opcodes/OSC.c
@@ -965,7 +965,7 @@ static int32_t OSC_list(CSOUND *csound, OSCLISTEN *p)
 /* ******** ARRAY VERSION **** EXPERIMENTAL *** */
 
 static int32_t OSC_ahandler(const char *path, const char *types,
-                       lo_arg **argv, int32_t argc, void *data, void *p)
+                       lo_arg **argv, int32_t argc, lo_message data, void *p)
 {
     IGN(argc);  IGN(data);
     OSC_PORT  *pp = (OSC_PORT*) p;

