--- src/nm-policy.c	2011-08-23 06:41:02.099686450 +0000
+++ src/nm-policy.c	2011-08-23 06:43:38.227791737 +0000
@@ -302,6 +302,7 @@
 		return;
 	}
 
+#if 0
 	/* Try automatically determined hostname from the best device's IP config */
 	if (!best4)
 		best4 = get_best_ip4_device (policy->manager, &best_req4);
@@ -356,6 +357,7 @@
 		}
 	}
 
+#endif
 	/* If no automatically-configured hostname, try using the hostname from
 	 * when NM started up.
 	 */
