diff --git src/systemd/src/libsystemd-network/sd-dhcp6-client.c src/systemd/src/libsystemd-network/sd-dhcp6-client.c
index f67a45bd2..ec11fa605 100644
--- src/systemd/src/libsystemd-network/sd-dhcp6-client.c
+++ src/systemd/src/libsystemd-network/sd-dhcp6-client.c
@@ -8,8 +8,10 @@
 #include <errno.h>
 #include <string.h>
 #include <sys/ioctl.h>
+#ifdef __GLIBC__
 #include <linux/if_arp.h>
 #include <linux/if_infiniband.h>
+#endif
 
 #include "sd-dhcp6-client.h"
 
