--- shared/n-acd/src/n-acd.c   2018-09-09 12:10:36.626987553 +0200
+++ shared/n-acd/src/n-acd.c   2018-09-09 12:51:36.680798693 +0200
@@ -23,10 +23,10 @@
 #include <errno.h>
 #include <limits.h>
 #include <linux/filter.h>
-#include <linux/if_ether.h>
 #include <linux/if_packet.h>
 #include <net/ethernet.h>
 #include <netinet/if_ether.h>
+#include <linux/if_ether.h>
 #include <netinet/in.h>
 #include <stddef.h>
 #include <stdio.h>
--- src/platform/nm-linux-platform.c
+++ src/platform/nm-linux-platform.c
@@ -20,6 +20,7 @@
 #include "nm-default.h"
 
 #include "nm-linux-platform.h"
+#include "wpan/nm-wpan-utils.h"
 
 #include <poll.h>
 #include <endian.h>
@@ -33,7 +34,9 @@
 #include <netinet/icmp6.h>
 #include <netinet/in.h>
 #include <linux/ip.h>
-#include <linux/if_arp.h>
+#if defined(__GLIBC__)
+    #include <linux/if_arp.h>
+#endif
 #include <linux/if_link.h>
 #include <linux/if_tun.h>
 #include <linux/if_tunnel.h>
@@ -53,7 +56,6 @@
 #include "nm-platform-private.h"
 #include "wifi/nm-wifi-utils.h"
 #include "wifi/nm-wifi-utils-wext.h"
-#include "wpan/nm-wpan-utils.h"
 #include "nm-utils/unaligned.h"
 #include "nm-utils/nm-io-utils.h"
 #include "nm-utils/nm-udev-utils.h"
--- src/devices/nm-device.c
+++ src/devices/nm-device.c
@@ -34,7 +34,9 @@
 #include <arpa/inet.h>
 #include <fcntl.h>
 #include <linux/if_addr.h>
-#include <linux/if_arp.h>
+#if defined(__GLIBC__)
+    #include <linux/if_arp.h>
+#endif
 #include <linux/rtnetlink.h>
 #include <linux/pkt_sched.h>
 #include <uuid/uuid.h>

