Usual fix for musl libc
--- shared/systemd/src/basic/stdio-util.h.orig
+++ shared/systemd/src/basic/stdio-util.h
@@ -19,7 +19,9 @@
   along with systemd; If not, see <http://www.gnu.org/licenses/>.
 ***/
 
+#if defined(__GLIBC__)
 #include <printf.h>
+#endif
 #include <stdarg.h>
 #include <stdio.h>
 #include <sys/types.h>
 
--- shared/systemd/src/basic/sort-util.h	2019-06-09 17:02:48.895695896 +0200
+++ shared/systemd/src/basic/sort-util.h	2019-06-09 17:03:27.564199937 +0200
@@ -5,6 +5,12 @@
 
 #include "macro.h"
 
+#if !defined(__GLIBC__)
+typedef int (*__compar_fn_t) (const void*, const void*);
+typedef __compar_fn_t comparison_fn_t;
+typedef int (*__compar_d_fn_t) (const void *, const void *, void *);
+#endif
+
 void *xbsearch_r(const void *key, const void *base, size_t nmemb, size_t size,
                  __compar_d_fn_t compar, void *arg);
 
