--- policy/org.freedesktop.NetworkManager.policy.in.permissive	2011-03-24 14:21:35.100912222 -0500
+++ policy/org.freedesktop.NetworkManager.policy.in	2011-03-24 14:21:49.916726997 -0500
@@ -95,7 +95,7 @@
     <_message>System policy prevents modification of network settings for all users</_message>
     <defaults>
       <allow_inactive>no</allow_inactive>
-      <allow_active>auth_admin_keep</allow_active>
+      <allow_active>yes</allow_active>
     </defaults>
   </action>
