From d6b3651fa96f05aeefc318f75503fc9ca77b1f58 Mon Sep 17 00:00:00 2001
From: Vsevolod Stakhov <vsevolod@highsecure.ru>
Date: Sat, 1 Jan 2022 22:45:48 +0000
Subject: [PATCH] Fix compilation with gcc 11.x

Issue: #16
---
 libasignify/blake2.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libasignify/blake2.h b/libasignify/blake2.h
index 5d0d40c..f9967d0 100644
--- a/libasignify/blake2.h
+++ b/libasignify/blake2.h
@@ -56,7 +56,7 @@ extern "C" {
     uint8_t  personal[BLAKE2B_PERSONALBYTES];  // 64
   } blake2b_param;
 
-  BLAKE_ALIGN( 64 ) typedef struct __blake2b_state
+  typedef struct BLAKE_ALIGN( 64 ) __blake2b_state
   {
     uint64_t h[8];
     uint64_t t[2];
