--- a/src/opentimelineio/CMakeLists.txt	2024-06-25 01:08:54.000000000 +0200
+++ -	2025-04-19 23:24:03.444723937 +0200
@@ -124,7 +124,7 @@
            RUNTIME DESTINATION "${OTIO_RESOLVED_CXX_DYLIB_INSTALL_DIR}")
 
     install(EXPORT OpenTimelineIOTargets
-           DESTINATION "${OTIO_RESOLVED_CXX_INSTALL_DIR}/share/opentimelineio"
+           DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/OpenTimelineIO"
            NAMESPACE OTIO:: )
 
     include(CMakePackageConfigHelpers)
@@ -132,7 +132,7 @@
         ${CMAKE_CURRENT_SOURCE_DIR}/OpenTimelineIOConfig.cmake.in
         ${CMAKE_CURRENT_BINARY_DIR}/OpenTimelineIOConfig.cmake
         INSTALL_DESTINATION
-            ${OTIO_RESOLVED_CXX_INSTALL_DIR}/share/opentimelineio
+            ${CMAKE_INSTALL_LIBDIR}/cmake/OpenTimelineIO
         NO_SET_AND_CHECK_MACRO
         NO_CHECK_REQUIRED_COMPONENTS_MACRO
     )
@@ -141,6 +141,6 @@
         FILES
             ${CMAKE_CURRENT_BINARY_DIR}/OpenTimelineIOConfig.cmake
         DESTINATION
-            ${OTIO_RESOLVED_CXX_INSTALL_DIR}/share/opentimelineio
+            ${CMAKE_INSTALL_LIBDIR}/cmake/OpenTimelineIO
     )
 endif()
--- a/src/opentime/CMakeLists.txt	2024-06-25 01:08:54.000000000 +0200
+++ -	2025-04-19 23:24:45.851845467 +0200
@@ -49,7 +49,7 @@
             RUNTIME DESTINATION "${OTIO_RESOLVED_CXX_DYLIB_INSTALL_DIR}")
 
     install(EXPORT OpenTimeTargets
-            DESTINATION "${OTIO_RESOLVED_CXX_INSTALL_DIR}/share/opentime"
+            DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/OpenTime"
             NAMESPACE OTIO:: )
 
     include(CMakePackageConfigHelpers)
@@ -57,7 +57,7 @@
         ${CMAKE_CURRENT_SOURCE_DIR}/OpenTimeConfig.cmake.in
         ${CMAKE_CURRENT_BINARY_DIR}/OpenTimeConfig.cmake
         INSTALL_DESTINATION
-            ${OTIO_RESOLVED_CXX_INSTALL_DIR}/share/opentime
+            ${CMAKE_INSTALL_LIBDIR}/cmake/OpenTime
         NO_SET_AND_CHECK_MACRO
         NO_CHECK_REQUIRED_COMPONENTS_MACRO
     )
@@ -66,7 +66,7 @@
         FILES
             ${CMAKE_CURRENT_BINARY_DIR}/OpenTimeConfig.cmake
         DESTINATION
-            ${OTIO_RESOLVED_CXX_INSTALL_DIR}/share/opentime
+            ${CMAKE_INSTALL_LIBDIR}/cmake/OpenTime
     )
 endif()
 
--- a/CMakeLists.txt	2024-06-25 01:08:54.000000000 +0200
+++ -	2025-04-21 12:15:44.839270019 +0200
@@ -20,6 +20,8 @@

 project(OpenTimelineIO VERSION ${OTIO_VERSION} LANGUAGES C CXX)

+include(GNUInstallDirs)
+
 #------------------------------------------------------------------------------
 # Options
 # Add all options and settings here for all subprojects to aid in project
@@ -100,7 +102,7 @@
     endif()

 else()
-    set(OTIO_RESOLVED_CXX_DYLIB_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/lib")
+    set(OTIO_RESOLVED_CXX_DYLIB_INSTALL_DIR "${CMAKE_INSTALL_LIBDIR}")
     message(STATUS "OTIO C++ installing to ${CMAKE_INSTALL_PREFIX}")
 endif()

