diff --git a/src/meson.build b/src/meson.build
index 12f8e32..3366e96 100644
--- src/meson.build
+++ src/meson.build
@@ -79,7 +79,10 @@ if get_option('with_introspection')
                 install: true,
         install_dir_gir: girdir,
     install_dir_typelib: typelibdir,
-             extra_args: [ '--c-include=jsonrpc-glib.h' ],
+             extra_args: [
+                 '--c-include=jsonrpc-glib.h',
+                 '-DJSONRPC_GLIB_COMPILATION',
+             ],
   )
 
   if get_option('with_vapi')
