--- libgnomekbd/desktop_gnome_peripherals_keyboard_xkb.schemas.in.orig	2010-04-15 08:02:18.334137371 +0200
+++ libgnomekbd/desktop_gnome_peripherals_keyboard_xkb.schemas.in	2010-04-15 08:03:47.637153522 +0200
@@ -6,7 +6,7 @@
       <applyto>/desktop/gnome/peripherals/keyboard/kbd/model</applyto>
       <owner>gnome</owner>    
       <type>string</type>
-      <default></default>
+      <default>[]</default>
       <locale name="C">
         <short>Keyboard model</short>
         <long>keyboard model</long>
@@ -218,7 +218,7 @@
       <applyto>/desktop/gnome/peripherals/keyboard/indicator/fontFamily</applyto>
       <owner>gnome</owner>    
       <type>string</type>
-      <default></default>
+      <default>[]</default>
       <locale name="C">
         <short>The font family</short>
         <long>The font family for the layout indicator</long>
@@ -230,7 +230,7 @@
       <applyto>/desktop/gnome/peripherals/keyboard/indicator/fontSize</applyto>
       <owner>gnome</owner>    
       <type>int</type>
-      <default></default>
+      <default>-1</default>
       <locale name="C">
         <short>The font size</short>
         <long>The font size for the layout indicator</long>
@@ -242,7 +242,7 @@
       <applyto>/desktop/gnome/peripherals/keyboard/indicator/foregroundColor</applyto>
       <owner>gnome</owner>    
       <type>string</type>
-      <default></default>
+      <default>[]</default>
       <locale name="C">
         <short>The foreground color</short>
         <long>The foreground color for the layout indicator</long>
@@ -254,7 +254,7 @@
       <applyto>/desktop/gnome/peripherals/keyboard/indicator/backgroundColor</applyto>
       <owner>gnome</owner>    
       <type>string</type>
-      <default></default>
+      <default>[]</default>
       <locale name="C">
         <short>The background color</short>
         <long>The background color for the layout indicator</long>
