From 20dfc2002482a21b955b710af119a01aecee784b Mon Sep 17 00:00:00 2001
From: Daniel Kolesa <daniel@octaforge.org>
Date: Tue, 15 Dec 2020 20:42:17 +0100
Subject: [PATCH] add ppc32 le subarch

---
 configure | 1 +
 1 file changed, 1 insertion(+)

diff --git configure configure
index 8680128..f1ca58e 100755
--- a/configure
+++ b/configure
@@ -645,6 +645,7 @@ fi
 if test "$ARCH" = "powerpc" ; then
 trycppif "__NO_FPRS__ && !_SOFT_FLOAT" "$t" && fail \
   "$0: error: compiler's floating point configuration is unsupported"
+trycppif __LITTLE_ENDIAN__ "$t" && SUBARCH=${SUBARCH}le
 trycppif _SOFT_FLOAT "$t" && SUBARCH=${SUBARCH}-sf
 fi
 
-- 
2.29.2

