From 20dfc2002482a21b955b710af119a01aecee784b Mon Sep 17 00:00:00 2001
From: Daniel Kolesa <daniel@octaforge.org>
Date: Tue, 15 Dec 2020 20:42:17 +0100
Subject: [PATCH] add ppc32 le subarch

---
 configure | 1 +
 1 file changed, 1 insertion(+)

diff --git configure configure
index 8680128..f1ca58e 100755
--- a/configure
+++ b/configure
@@ -707,6 +707,7 @@ trycppif __mips_soft_float "$t" && SUBAR
 fi
 
 if test "$ARCH" = "powerpc" ; then
+trycppif __LITTLE_ENDIAN__ "$t" && SUBARCH=${SUBARCH}le
 trycppif "_SOFT_FLOAT || __NO_FPRS__" "$t" && SUBARCH=${SUBARCH}-sf
 printf "checking whether compiler can use 'd' constraint in asm... "
 echo 'double f(double x) { __asm__ ("fabs %0, %1" : "=d"(x) : "d"(x)); return x; }' > "$tmpc"
-- 
2.29.2

