--- a/libs/quazip/CMakeLists.txt
+++ b/libs/quazip/CMakeLists.txt
@@ -64,13 +64,8 @@ else()
 	message(FATAL_ERROR "Qt version ${QUAZIP_QT_MAJOR_VERSION} is not supported")
 endif()
 
-find_package(Qt${QUAZIP_QT_MAJOR_VERSION} OPTIONAL_COMPONENTS Zlib)
-if (Qt${QUAZIP_QT_MAJOR_VERSION}Zlib_FOUND)
-	set(QUAZIP_LIB_LIBRARIES ${QUAZIP_LIB_LIBRARIES} Qt${QUAZIP_QT_MAJOR_VERSION}::Zlib)
-else()
-	find_package(ZLIB REQUIRED)
-	set(QUAZIP_LIB_LIBRARIES ${QUAZIP_LIB_LIBRARIES} ZLIB::ZLIB)
-endif()
+find_package(ZLIB REQUIRED)
+set(QUAZIP_LIB_LIBRARIES ${QUAZIP_LIB_LIBRARIES} ZLIB::ZLIB)
 
 add_subdirectory(quazip)
 
