From 04a9e2f44d90f7b397716bb50d55e53e6ed8367c Mon Sep 17 00:00:00 2001
From: Duncaen <mail@duncano.de>
Date: Fri, 9 Mar 2018 18:53:12 +0100
Subject: [PATCH] use LDFLAGS in Makefile

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 9a124cc..0a422c3 100644
--- Makefile
+++ Makefile
@@ -62,7 +62,7 @@ installwww: www
 	install -m 0444 lowdown.tar.gz.sha512 $(WWWDIR)/snapshots
 
 lowdown: liblowdown.a main.o
-	$(CC) -o $@ main.o liblowdown.a -lm
+	$(CC) -o $@ main.o liblowdown.a $(LDFLAGS) -lm
 
 lowdown-diff: lowdown
 	ln -f lowdown lowdown-diff
--- configure.orig
+++ configure
@@ -65,7 +65,7 @@
 CFLAGS="${CFLAGS} -Wwrite-strings -Wno-unused-parameter"
 LDADD=
 CPPFLAGS=
-LDFLAGS=
+LDFLAGS=`printf "all:\\n\\t@echo \\\$(LDFLAGS)\\n" | make -sf -`
 PREFIX="/usr/local"
 BINDIR=
 SBINDIR=
