--- buffer.c.orig
+++ buffer.c
@@ -25,6 +25,7 @@
 
 #include <assert.h>
 #include <stdarg.h>
+#include <stdint.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
--- diff.c.orig
+++ diff.c
@@ -28,6 +28,7 @@
 #if HAVE_MD5
 # include <md5.h>
 #endif
+#include <stdint.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
--- nroff_escape.c.orig
+++ nroff_escape.c
@@ -20,6 +20,7 @@
 # include <sys/queue.h>
 #endif
 
+#include <stdint.h>
 #include <stdio.h>
 
 #include "lowdown.h"
--- util.c.orig
+++ util.c
@@ -20,6 +20,7 @@
 # include <sys/queue.h>
 #endif
 
+#include <stdint.h>
 #include <stdio.h>
 #include <string.h>
 
