--- a/libguile/fports.c	2017-02-12 22:06:33.000000000 +0100
+++ b/libguile/fports.c	2017-10-26 19:09:26.425213055 +0200
@@ -66,8 +66,8 @@
 #define OFF_T_MAX  LONG_MAX
 #define OFF_T_MIN  LONG_MIN
 #elif SIZEOF_OFF_T == SIZEOF_LONG_LONG
-#define OFF_T_MAX  LONG_LONG_MAX
-#define OFF_T_MIN  LONG_LONG_MIN
+#define OFF_T_MAX  LLONG_MAX
+#define OFF_T_MIN  LLONG_MIN
 #else
 #error Oops, unknown OFF_T size
 #endif
