--- libguile/finalizers.c	2014-05-02 15:15:57.000000000 +0200
+++ libguile/finalizers.c	2019-01-04 14:32:52.336586516 +0100
@@ -164,7 +164,7 @@
 
 
 #ifndef HAVE_GC_SET_FINALIZER_NOTIFIER
-static void
+void
 GC_set_finalizer_notifier (void (*notifier) (void))
 {
   GC_finalizer_notifier = notifier;
--- libguile/gc.c	2016-12-15 01:03:33.000000000 +0100
+++ libguile/gc.c	2019-01-04 14:35:56.357579179 +0100
@@ -196,7 +196,7 @@
 /* Compatibility.  */
 
 #ifndef HAVE_GC_GET_HEAP_USAGE_SAFE
-static void
+void
 GC_get_heap_usage_safe (GC_word *pheap_size, GC_word *pfree_bytes,
                         GC_word *punmapped_bytes, GC_word *pbytes_since_gc,
                         GC_word *ptotal_bytes)
@@ -214,7 +214,7 @@
 #endif
 
 #ifndef HAVE_GC_GET_FREE_SPACE_DIVISOR
-static GC_word
+GC_word
 GC_get_free_space_divisor (void)
 {
   return GC_free_space_divisor;
@@ -586,7 +586,7 @@
 }
 
 #ifndef HAVE_GC_SET_FINALIZE_ON_DEMAND
-static void
+void
 GC_set_finalize_on_demand (int foo)
 {
   GC_finalize_on_demand = foo;
