diff --git a/stdlib/MbedTLS_jll/src/MbedTLS_jll.jl b/stdlib/MbedTLS_jll/src/MbedTLS_jll.jl
index a7ca666b23..b84a42f4e7 100644
--- a/stdlib/MbedTLS_jll/src/MbedTLS_jll.jl
+++ b/stdlib/MbedTLS_jll/src/MbedTLS_jll.jl
@@ -31,9 +31,9 @@ elseif Sys.isapple()
     const libmbedtls = "@rpath/libmbedtls.13.dylib"
     const libmbedx509 = "@rpath/libmbedx509.1.dylib"
 else
-    const libmbedcrypto = "libmbedcrypto.so.5"
-    const libmbedtls = "libmbedtls.so.13"
-    const libmbedx509 = "libmbedx509.so.1"
+    const libmbedcrypto = "libmbedcrypto.so"
+    const libmbedtls = "libmbedtls.so"
+    const libmbedx509 = "libmbedx509.so"
 end
 
 function __init__()
diff --git a/stdlib/libLLVM_jll/src/libLLVM_jll.jl b/stdlib/libLLVM_jll/src/libLLVM_jll.jl
index fa45e754e5..b51812e3a8 100644
--- a/stdlib/libLLVM_jll/src/libLLVM_jll.jl
+++ b/stdlib/libLLVM_jll/src/libLLVM_jll.jl
@@ -23,7 +23,7 @@ if Sys.iswindows()
 elseif Sys.isapple()
     const libLLVM = "@rpath/libLLVM.dylib"
 else
-    const libLLVM = "libLLVM-11jl.so"
+    const libLLVM = "libLLVM.so"
 end
 
 function __init__()
diff --git a/stdlib/nghttp2_jll/src/nghttp2_jll.jl b/stdlib/nghttp2_jll/src/nghttp2_jll.jl
index 8b98c76ac5..77ad9e3800 100644
--- a/stdlib/nghttp2_jll/src/nghttp2_jll.jl
+++ b/stdlib/nghttp2_jll/src/nghttp2_jll.jl
@@ -22,7 +22,7 @@ if Sys.iswindows()
 elseif Sys.isapple()
     const libnghttp2 = "@rpath/libnghttp2.14.dylib"
 else
-    const libnghttp2 = "libnghttp2.so.14"
+    const libnghttp2 = "libnghttp2.so"
 end
 
 function __init__()
