Description: Fix FTBFS with new e2fsprogs.
Author: Hilko Bengen <bengen@debian.org>
Last-Update: 2018-04-13
Index: ext4magic/src/inode.c
===================================================================
--- ext4magic.orig/src/inode.c
+++ ext4magic/src/inode.c
@@ -406,14 +406,14 @@ void dump_inode(FILE *out, const char *p
         fprintf(out,
                         "%sFile ACL: %d    Directory ACL: %d Translator: %d\n",
                         prefix,
-                        inode->i_file_acl, LINUX_S_ISDIR(inode->i_mode) ? inode->i_dir_acl : 0,
+                        inode->i_file_acl, LINUX_S_ISDIR(inode->i_mode) ? inode->i_size_high : 0,
                         inode->osd1.hurd1.h_i_translator);
         else
                 fprintf(out, "%sFile ACL: %llu    Directory ACL: %d\n",
                         prefix,
                         inode->i_file_acl | ((long long)
                                 (inode->osd2.linux2.l_i_file_acl_high) << 32),
-                        LINUX_S_ISDIR(inode->i_mode) ? inode->i_dir_acl : 0);
+                        LINUX_S_ISDIR(inode->i_mode) ? inode->i_size_high : 0);
         if (os == EXT2_OS_LINUX)
                 fprintf(out, "%sLinks: %d   Blockcount: %llu\n",
                         prefix, inode->i_links_count,
