diff -ru qmc2.orig/lzma/CpuArch.c qmc2/lzma/CpuArch.c
--- lzma/CpuArch.c	2014-04-26 15:00:09.408414619 +0200
+++ lzma/CpuArch.c	2015-08-28 10:22:31.421881641 +0200
@@ -75,9 +75,17 @@
   #else
 
   __asm__ __volatile__ (
+  #if defined(MY_CPU_X86) && defined(__PIC__)
+    "mov %%ebx, %%edi;"
+    "cpuid;"
+    "xchgl %%ebx, %%edi;"
+    : "=a" (*a) ,
+      "=D" (*b) ,
+  #else
     "cpuid"
     : "=a" (*a) ,
       "=b" (*b) ,
+  #endif
       "=c" (*c) ,
       "=d" (*d)
     : "0" (function)) ;
