$OpenBSD: patch-doc_xio_help,v 1.1 2014/04/24 15:17:08 sthen Exp $
--- doc/xio.help.orig	Sun Jun 23 07:16:48 2013
+++ doc/xio.help	Sat Apr 19 15:47:37 2014
@@ -4566,19 +4566,6 @@ must contain certificates in PEM format and their hash
 documentation) 
 
 
-Option: openssl-egd=file
-Aliases: egd
-
-Type: FILENAME
-Option group: OPENSSL
-Phase: SPEC
-Platforms: (depends on openssl installation)
-
-On some systems, openssl requires an explicit source of random data. Specify
-the socket name where an entropy gathering daemon like egd provides random
-data, e.g. /dev/egd-pool.
-
-
 Option: openssl-pseudo
 Aliases: pseudo
 
