Index: VTK/ThirdParty/ioss/vtkioss/Ioss_FileInfo.C
===================================================================
--- VTK/ThirdParty/ioss/vtkioss/Ioss_FileInfo.C.orig
+++ VTK/ThirdParty/ioss/vtkioss/Ioss_FileInfo.C
@@ -40,7 +40,7 @@
 #include <tokenize.h>
 
 #ifndef _MSC_VER
-#include <sys/unistd.h>
+#include <unistd.h>
 #else
 #include <direct.h>
 #include <io.h>
@@ -61,9 +61,6 @@
 
 #include <cstdio>
 #include <sys/stat.h>
-#ifndef _MSC_VER
-#include <unistd.h>
-#endif
 
 namespace {
   bool internal_access(const std::string &name, int mode);
