--- src/ssl_sock.c.orig	2015-01-18 22:54:37.205257280 +0000
+++ src/ssl_sock.c	2015-01-18 22:55:12.506030051 +0000
@@ -684,7 +684,7 @@
 }
 #endif
 
-#ifdef TLSEXT_TYPE_application_layer_protocol_negotiation
+#if 0
 /* This callback is used so that the server advertises the list of
  * negociable protocols for ALPN.
  */
@@ -1633,7 +1633,7 @@
 	if (bind_conf->npn_str)
 		SSL_CTX_set_next_protos_advertised_cb(ctx, ssl_sock_advertise_npn_protos, bind_conf);
 #endif
-#ifdef TLSEXT_TYPE_application_layer_protocol_negotiation
+#if 0
 	if (bind_conf->alpn_str)
 		SSL_CTX_set_alpn_select_cb(ctx, ssl_sock_advertise_alpn_protos, bind_conf);
 #endif
@@ -3556,7 +3556,7 @@
 }
 #endif
 
-#ifdef TLSEXT_TYPE_application_layer_protocol_negotiation
+#if 0
 static int
 smp_fetch_ssl_fc_alpn(struct proxy *px, struct session *l4, void *l7, unsigned int opt,
                       const struct arg *args, struct sample *smp, const char *kw)
@@ -4117,7 +4117,7 @@
 /* parse the "alpn" bind keyword */
 static int bind_parse_alpn(char **args, int cur_arg, struct proxy *px, struct bind_conf *conf, char **err)
 {
-#ifdef TLSEXT_TYPE_application_layer_protocol_negotiation
+#if 0
 	char *p1, *p2;
 
 	if (!*args[cur_arg + 1]) {
@@ -4574,7 +4574,7 @@
 #ifdef OPENSSL_NPN_NEGOTIATED
 	{ "ssl_fc_npn",             smp_fetch_ssl_fc_npn,         0,                   NULL,    SMP_T_STR,  SMP_USE_L5CLI },
 #endif
-#ifdef TLSEXT_TYPE_application_layer_protocol_negotiation
+#if 0
 	{ "ssl_fc_alpn",            smp_fetch_ssl_fc_alpn,        0,                   NULL,    SMP_T_STR,  SMP_USE_L5CLI },
 #endif
 	{ "ssl_fc_protocol",        smp_fetch_ssl_fc_protocol,    0,                   NULL,    SMP_T_STR,  SMP_USE_L5CLI },
