--- a/data/appdata/meson.build
+++ b/data/appdata/meson.build
@@ -1,6 +1,6 @@
 appdata_conf = configuration_data()
 appdata_conf.set('application_id', application_id)
-i18n.merge_file('appdata',
+i18n.merge_file(
   input: configure_file(
     input: 'org.gnome.Dictionary.appdata.xml.in.in',
     output: 'org.gnome.Dictionary.appdata.xml.in',
--- a/data/meson.build
+++ b/data/meson.build
@@ -3,7 +3,7 @@ subdir('appdata')
 desktop_conf = configuration_data()
 desktop_conf.set('icon', application_id)
 desktop_conf.set('application_id', application_id)
-i18n.merge_file('desktop',
+i18n.merge_file(
   input: configure_file(
     input: 'org.gnome.Dictionary.desktop.in.in',
     output: 'org.gnome.Dictionary.desktop.in',
@@ -45,7 +45,7 @@ sources = [
 ]
 
 foreach s: sources
-  i18n.merge_file('sources',
+  i18n.merge_file(
     input: '@0@.in'.format(s),
     output: s,
     install: true,
