--- cmake/Dependencies.cmake	2015-08-29 23:34:01.000000000 +0200
+++ -	2018-10-05 21:42:24.139024667 +0200
@@ -58,7 +58,7 @@
 
 set(Boost_ADDITIONAL_VERSIONS "1.47.0" "1.46.1" "1.46" "1.46.0" "1.45" "1.45.0" "1.44" "1.44.0")
 
-set(LUXRAYS_BOOST_COMPONENTS thread program_options filesystem serialization iostreams regex system python)
+set(LUXRAYS_BOOST_COMPONENTS thread program_options filesystem serialization iostreams regex system python27)
 find_package(Boost ${Boost_MINIMUM_VERSION} COMPONENTS ${LUXRAYS_BOOST_COMPONENTS})
 if (NOT Boost_FOUND)
         # Try again with the other type of libs
