--- a/justfile
+++ b/justfile
@@ -12,7 +12,6 @@
 postgres_dbname_prefix := "nebula_"
 postgres_pass_prefix := "password_"
 
-COMMIT := `git rev-parse --short HEAD`
 DATE := `date "+%Y-%m-%dT%H:%M:%SZ"`
 USER := `id -un`
 VERSION := `git describe --tags --abbrev=0 || true`
@@ -23,22 +22,16 @@
 
 # build a nebula executable to the ./dist/ folder
 build:
-	go build -ldflags "-X main.version={{VERSION}} -X main.commit={{COMMIT}} -X main.date={{DATE}} -X main.builtBy={{USER}}" -o dist/nebula github.com/dennis-tra/nebula-crawler/cmd/nebula
+	go build -ldflags "-X main.version={{VERSION}} -X main.date={{DATE}} -X main.builtBy={{USER}}" -o dist/nebula github.com/dennis-tra/nebula-crawler/cmd/nebula
 
 # build a nebula docker image
 docker platform="linux/amd64":
 	docker build --platform {{platform}} \
 	  --build-arg VERSION={{VERSION}} \
-	  --build-arg COMMIT={{COMMIT}} \
 	  --build-arg BUILT_BY={{USER}} \
 	  --build-arg DATE={{DATE}} \
-	  -t dennistra/nebula:{{COMMIT}} \
 	  .
 
-# push a nebula image to docker hub
-docker-push: docker
-	docker push dennistra/nebula:{{COMMIT}}
-
 # start a clickhouse server
 start-clickhouse env="local" detached="true":
     @echo "Starting ClickHouse server..."
