--- gschem/src/Makefile.orig	2019-10-08 20:33:14.000000000 +0200
+++ gschem/src/Makefile.in	2019-10-12 15:49:39.966600662 +0200
@@ -2693,12 +2693,12 @@
 	$(AM_V_SNARF)CPP="$(CPP)" $(GUILE_SNARF) -o $@ $< $(snarf_cpp_opts)
 
 actions.decl.x: actions.decl.h actions.c Makefile
-	$(AM_V_GEN)$(CPP) -I$(srcdir) -c $< | \
+	$(AM_V_GEN)$(CPP) -I$(srcdir) $< | \
 	  sed -e '/^.*KEEP_LINE /!d' -e 's/^.*KEEP_LINE //' > $@.tmp && \
 	mv $@.tmp $@
 
 actions.init.x: actions.init.c actions.c Makefile
-	$(AM_V_GEN)$(CPP) -I$(srcdir) -c $< | \
+	$(AM_V_GEN)$(CPP) -I$(srcdir) $< | \
 	  sed -e '/^.*KEEP_LINE /!d' -e 's/^.*KEEP_LINE //' > $@.tmp && \
 	mv $@.tmp $@
 
