--- a/gschem/src/Makefile.am	2020-12-12 16:20:18.000000000 +0100
+++ b/gschem/src/Makefile.am	2020-12-20 12:55:06.921054480 +0100
@@ -152,12 +152,12 @@
 	$(AM_V_SNARF)CPP="$(CPP)" $(GUILE_SNARF) -o $@ $< $(snarf_cpp_opts)
 
 actions.decl.x: actions.decl.h actions.c Makefile
-	$(AM_V_GEN)$(CPP) -I$(srcdir) -c $< | \
+	$(AM_V_GEN)$(CPP) -I$(srcdir) $< | \
 	  sed -e '/^.*KEEP_LINE /!d' -e 's/^.*KEEP_LINE //' > $@.tmp && \
 	mv $@.tmp $@
 
 actions.init.x: actions.init.c actions.c Makefile
-	$(AM_V_GEN)$(CPP) -I$(srcdir) -c $< | \
+	$(AM_V_GEN)$(CPP) -I$(srcdir) $< | \
 	  sed -e '/^.*KEEP_LINE /!d' -e 's/^.*KEEP_LINE //' > $@.tmp && \
 	mv $@.tmp $@
 
