--- daemon_test.go
+++ daemon_test.go
@@ -229,6 +229,7 @@ func TestHelperProcessSignalsUDOT(*testing.T) {
 	}
 }
 
+/*
 func TestSignalsUDOT(t *testing.T) {
 	sdir, err := ioutil.TempDir("", "TestSignalsUDOT")
 	if err != nil {
@@ -464,6 +465,7 @@ func TestSignalsUDOT(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+*/
 
 func TestDaemonNewEnvHOME(t *testing.T) {
 	cfg := &Config{}
--- daemon_test.go
+++ daemon_test.go
@@ -10,7 +10,7 @@ import (
 	"os/user"
 	"path/filepath"
 	"strings"
-	"sync/atomic"
+	//"sync/atomic"
 	"testing"
 	"time"
 )
