From 27fb43ea79be3caefd109a082f4744efb925605c Mon Sep 17 00:00:00 2001
From: Nathan Owens <ndowens@artixlinux.org>
Date: Sun, 3 Jan 2021 08:44:42 -0600
Subject: [PATCH] fix cross

---
 configure.in | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git configure.in configure.in
index 1e4dac4..44ee54f 100644
--- configure.in
+++ configure.in
@@ -34,7 +34,7 @@ AC_CHECK_FUNCS(strerror)
 AC_SEARCH_LIBS(regcomp, [regex pcre])
 
 AC_MSG_CHECKING(for broken realloc)
-AC_TRY_RUN([#include <sys/types.h>
+AC_COMPILE_IFELSE([#include <sys/types.h>
 #include <signal.h>
 #include <stdlib.h>
 
@@ -49,7 +49,7 @@ int main(void)
   realloc((void*)0,1);
   exit(0);
 }
-],AC_MSG_RESULT(no),AC_DEFINE(BROKEN_REALLOC) AC_MSG_RESULT(yes))
+],AC_MSG_RESULT(no),AC_DEFINE([BROKEN_REALLOC, [1], [Define broken realloc]]) AC_MSG_RESULT(yes))
 
 USE_NLS=no
 AC_CHECK_PROG(MSGFMT,msgfmt,yes,no)
@@ -58,11 +58,11 @@ then
   AC_CHECK_HEADERS(libintl.h,[LIBINTL=yes])
   if test "$LIBINTL" = yes
   then
-    AC_SEARCH_LIBS(gettext,intl,[AC_DEFINE(HAVE_GETTEXT) USE_NLS=yes])
+    AC_SEARCH_LIBS(gettext,intl,[AC_DEFINE([HAVE_GETTEXT, [1], [Define HAVE_GETTEXT]]) USE_NLS=yes])
   fi
 fi
 
-AC_DEFINE_UNQUOTED(VERSION,"$VERSION")
+AC_DEFINE_UNQUOTED([VERSION], ["$VERSION"], [Define VERSION])
 AC_SUBST(VERSION)
 AC_SUBST(UPDATED)
 eval DATADIR=$datadir
-- 
2.30.0

