--- gmic-qt/src/SearchFieldWidget.cpp	2017-06-28 11:32:57.000000000 +0200
+++ gmic-qt/src/SearchFieldWidget.cpp	2019-06-07 23:46:28.301753374 +0200
@@ -30,6 +30,7 @@
 #include <QHBoxLayout>
 #include <QLineEdit>
 #include <QFrame>
+#include <QAction>
 #include "DialogSettings.h"
 
 SearchFieldWidget::SearchFieldWidget(QWidget *parent) :
--- gmic-qt/include/MainWindow.h	2017-06-28 11:32:57.000000000 +0200
+++ gmic-qt/include/MainWindow.h	2019-06-08 00:03:57.884695538 +0200
@@ -26,6 +26,7 @@
 #define _GMIC_QT_MAINWINDOW_H_
 
 #include <QPalette>
+#include <QButtonGroup>
 #include <QWidget>
 #include <QStandardItemModel>
 #include <QModelIndex>
--- gmic-qt/src/MainWindow.cpp	2017-06-28 11:32:57.000000000 +0200
+++ gmic-qt/src/MainWindow.cpp	2019-06-08 00:09:57.513675721 +0200
@@ -28,6 +28,7 @@
 #include <QSettings>
 #include <QTextDocument>
 #include <QTextStream>
+#include <QButtonGroup>
 #include <QInputDialog>
 #include <QMessageBox>
 #include <QFile>
