-Wsign-compare warning on x86.

--- src/core/monitor-kms.c.orig	2013-10-20 08:15:25.715631021 +0200
+++ src/core/monitor-kms.c	2013-10-20 08:15:38.731733070 +0200
@@ -480,7 +480,7 @@ meta_monitor_manager_kms_read_current (M
             {
               for (j = 0; j < manager->n_crtcs; j++)
                 {
-                  if (manager->crtcs[j].crtc_id == output_kms->current_encoder->crtc_id)
+                  if (manager->crtcs[j].crtc_id == (glong)output_kms->current_encoder->crtc_id)
                     {
                       meta_output->crtc = &manager->crtcs[j];
                       break;
