--- src/wayland/meta-xwayland.c.jx	2013-09-16 16:40:46.000000000 -0400
+++ src/wayland/meta-xwayland.c	2013-10-02 13:37:57.027318760 -0400
@@ -380,11 +380,9 @@ meta_xwayland_start (MetaWaylandComposit
   args[3] = "-rootless";
   args[4] = "-retro";
   args[5] = "-noreset";
-  args[6] = "-logfile";
-  args[7] = g_build_filename (g_get_user_cache_dir (), "xwayland.log", NULL);
-  args[8] = "-nolisten";
-  args[9] = "all";
-  args[10] = NULL;
+  args[6] = "-nolisten";
+  args[7] = "all";
+  args[8] = NULL;
 
   error = NULL;
   if (g_spawn_async (NULL, /* cwd */
