--- a/src/ui/x11.rs
+++ b/src/ui/x11.rs
@@ -1,5 +1,5 @@
 use std::ffi::CString;
-use std::os::raw::c_int;
+use std::os::raw::{c_int, c_long};
 
 use gdk_x11::{
     X11Display,
@@ -160,8 +160,8 @@
     s1: xlib::Atom,
     s2: xlib::Atom,
 ) {
-    const NET_WM_STATE_ADD: i64 = 1;
-    const NET_WM_STATE_APP: i64 = 1;
+    const NET_WM_STATE_ADD: c_long = 1;
+    const NET_WM_STATE_APP: c_long = 1;
 
     unsafe {
         let c = ctx.xlib();
@@ -173,8 +173,8 @@
         event.client_message.format = 32;
         let data = event.client_message.data.as_longs_mut();
         data[0] = NET_WM_STATE_ADD;
-        data[1] = s1 as i64;
-        data[2] = s2 as i64;
+        data[1] = s1 as c_long;
+        data[2] = s2 as c_long;
         data[3] = NET_WM_STATE_APP;
         data[4] = 0;
 
