--- a/ffmpegaudio.cc
+++ b/ffmpegaudio.cc
@@ -350,8 +350,9 @@
   if ( audioStream_ && codecContext_ && codec_ )
   {
     audioStream_->discard = AVDISCARD_ALL;
+#if LIBAVCODEC_VERSION_MAJOR < 62
     avcodec_close( codecContext_ );
-#if LIBAVCODEC_VERSION_MAJOR > 57 || ( LIBAVCODEC_VERSION_MAJOR == 57 && LIBAVCODEC_VERSION_MINOR >= 33 )
+#elif LIBAVCODEC_VERSION_MAJOR > 57 || ( LIBAVCODEC_VERSION_MAJOR == 57 && LIBAVCODEC_VERSION_MINOR >= 33 )
     avcodec_free_context( &codecContext_ );
 #endif
   }
