--- a/ffmpegaudio.cc.orig 2018-05-09 22:16:13.480659958 +0200
+++ b/ffmpegaudio.cc      2018-05-09 22:16:50.748322790 +0200
@@ -143,7 +143,7 @@
     return false;
   }

-  unsigned char * avioBuffer = ( unsigned char * )av_malloc( kBufferSize + FF_INPUT_BUFFER_PADDING_SIZE );
+  unsigned char * avioBuffer = ( unsigned char * )av_malloc( kBufferSize + AV_INPUT_BUFFER_PADDING_SIZE );
   if ( !avioBuffer )
   {
     errorString = QObject::tr( "av_malloc() failed." );
@@ -380,7 +380,7 @@
   }

   if ( !Qt4x5::AtomicInt::loadAcquire( isCancelled_ ) &&
-       codecContext_->codec->capabilities & CODEC_CAP_DELAY )
+       codecContext_->codec->capabilities & AV_CODEC_CAP_DELAY )
   {
     av_init_packet( &packet );
     int gotFrame = 0;

