--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -108,6 +108,10 @@
      ~/Library/Frameworks/include
      /Library/Frameworks/include
      /usr/local/include
+     /usr/local/include/openjpeg-2.5
+     /usr/include/openjpeg-2.5
+     /usr/local/include/openjpeg-2.4
+     /usr/include/openjpeg-2.4
      /usr/local/include/openjpeg-2.3
      /usr/include/openjpeg-2.3
      /usr/local/include/openjpeg-2.1
