diff -ru a/orientdb-community-2.0.4.orig/bin/dserver.sh ./bin/dserver.sh
--- a/orientdb-community-2.0.4.orig/bin/dserver.sh	2015-03-03 17:24:25.000000000 -0600
+++ ./bin/dserver.sh	2015-03-07 11:32:14.817541145 -0600
@@ -1,5 +1,5 @@
-#!/bin/sh
+#!/bin/bash
 #
 # Copyright (c) Orient Technologies LTD (http://www.orientechnologies.com)
 #
-${0/%dserver.sh/server.sh} -Ddistributed=true $*
+exec ${0/%dserver.sh/server.sh} -Ddistributed=true $*
Only in .: that.patch
