--- a/build/config/rust.gni
+++ b/build/config/rust.gni
@@ -196,7 +196,18 @@
 # a cargo project that dumps the `CARGO_CFG_TARGET_ABI` from its build.rs. See
 # https://issues.chromium.org/u/1/issues/372512092#comment5 for an example.
 rust_abi_target = ""
-if (is_linux || is_chromeos) {
+if (is_musl) {
+  if (current_cpu == "arm64") {
+    rust_abi_target = "aarch64-unknown-linux-musl"
+    cargo_target_abi = ""
+  } else if (current_cpu == "x86") {
+    rust_abi_target = "i686-unknown-linux-musl"
+    cargo_target_abi = ""
+  } else if (current_cpu == "x64") {
+    rust_abi_target = "x86_64-unknown-linux-musl"
+    cargo_target_abi = ""
+  }
+} else if (is_linux || is_chromeos) {
   if (current_cpu == "arm64") {
     rust_abi_target = "aarch64-unknown-linux-gnu"
     cargo_target_abi = ""
--- a/build/rust/known-target-triples.txt
+++ b/build/rust/known-target-triples.txt
@@ -17,6 +17,7 @@
 aarch64-pc-windows-msvc
 aarch64-unknown-fuchsia
 aarch64-unknown-linux-gnu
+aarch64-unknown-linux-musl
 arm64e-apple-ios
 arm-unknown-linux-gnueabi
 arm-unknown-linux-gnueabihf
@@ -28,6 +29,7 @@
 i686-linux-android
 i686-pc-windows-msvc
 i686-unknown-linux-gnu
+i686-unknown-linux-musl
 loongarch64-unknown-linux-gnu
 riscv64-linux-android
 riscv64gc-unknown-linux-gnu
@@ -41,5 +43,6 @@
 x86_64-pc-windows-msvc
 x86_64-unknown-fuchsia
 x86_64-unknown-linux-gnu
+x86_64-unknown-linux-musl
 powerpc64le-unknown-linux-gnu
 s390x-unknown-linux-gnu
