--- a/package.json	2022-07-06 17:31:50.000000000 +0200
+++ -	2022-07-08 23:04:43.654812957 +0200
@@ -98,7 +98,6 @@
     "precommit": "node -e 'process.exit(0)'",
     "preinstall": "npm run create-typescript-definitions",
     "prepack": "check-for-leaks",
-    "prepare": "husky",
     "repl": "node ./script/start.js --interactive",
     "start": "node ./script/start.js",
     "test": "node ./script/spec-runner.js",
