--- a/CMakeModules/Warnings.cmake
+++ b/CMakeModules/Warnings.cmake
@@ -38,13 +38,6 @@
     set( WARN_FLAGS_CXX "-Wall" )
 
 
-    # Warn about missing override specifiers
-    CHECK_CXX_COMPILER_FLAG( "-Wsuggest-override" COMPILER_SUPPORTS_WSUGGEST_OVERRIDE )
-
-    if( COMPILER_SUPPORTS_WSUGGEST_OVERRIDE )
-        set( WARN_FLAGS_CXX "${WARN_FLAGS_CXX} -Wsuggest-override" )
-        message( STATUS "Enabling warning -Wsuggest-override" )
-    endif()
