--- 3d-viewer/3d_cache/sg/CMakeLists.txt.orig	2018-08-28 02:50:47.256524770 +0200
+++ 3d-viewer/3d_cache/sg/CMakeLists.txt	2018-08-28 02:51:35.080183811 +0200
@@ -44,7 +44,7 @@
 endif()
 
 find_file( S3DSG_VERSION_FILE sg_version.h
-        PATHS ${CMAKE_SOURCE_DIR}/include/plugins/3dapi NO_DEFAULT_PATH )
+        PATHS ${CMAKE_SOURCE_DIR}/include/plugins/3dapi NO_DEFAULT_PATH NO_CMAKE_FIND_ROOT_PATH)
 
 if( NOT ${S3DSG_VERSION_FILE} STREQUAL "S3DSG_VERSION_FILE-NOTFOUND" )
