Description: Fixes the path to search for matplotlibrc file
Forwarded: not-needed
Author: Sandro Tosi <morph@debian.org>

--- lib/matplotlib/__init__.py
+++ lib/matplotlib/__init__.py
@@ -653,10 +653,12 @@ def _get_data_path():
             raise RuntimeError('Path in environment MATPLOTLIBDATA not a directory')
         return path
 
-    path = os.sep.join([os.path.dirname(__file__), 'mpl-data'])
+    path = '/usr/share/matplotlib/mpl-data'
     if os.path.isdir(path):
         return path
 
+    raise RuntimeError('Could not find the matplotlib data files')
+
     # setuptools' namespace_packages may highjack this init file
     # so need to try something known to be in matplotlib, not basemap
     import matplotlib.afm
@@ -777,7 +779,7 @@ def matplotlib_fname():
                     home, '.matplotlib', 'matplotlibrc')
             return fname
 
-    path = get_data_path()  # guaranteed to exist or raise
+    path = '/etc'  # guaranteed to exist or raise
     fname = os.path.join(path, 'matplotlibrc')
     if not os.path.exists(fname):
         warnings.warn('Could not find matplotlibrc; using defaults')
