From dba704bc7c1eb34695965bec04b33f955167a74f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Lo=C3=AFc=20Est=C3=A8ve?= <loic.esteve@ymail.com>
Date: Tue, 11 Aug 2026 13:20:57 +0200
Subject: [PATCH] Use PEP 621 metadata to fix building with flit-core>=4 (#226)

---
 pyproject.toml | 19 +++++++++----------
 1 file changed, 9 insertions(+), 10 deletions(-)

diff --git a/pyproject.toml b/pyproject.toml
index c0ea1696..f7a95ccd 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,20 +1,16 @@
 [build-system]
-requires = ["flit_core >=2,<4"]
+requires = ["flit_core >=3.2,<5"]
 build-backend = "flit_core.buildapi"
 
-# TODO: replace the following section by the standard [project] section to be able
-# to use flit v4.
-[tool.flit.metadata]
-module = "threadpoolctl"
-author = "Thomas Moreau"
-author-email = "thomas.moreau.2010@gmail.com"
-home-page = "https://github.com/joblib/threadpoolctl"
-description-file = "README.md"
+[project]
+name = "threadpoolctl"
+authors = [{name = "Thomas Moreau", email = "thomas.moreau.2010@gmail.com"}]
+readme = "README.md"
 requires-python = ">=3.9"
 license = "BSD-3-Clause"
+dynamic = ["version", "description"]
 classifiers = [
     "Intended Audience :: Developers",
-    "License :: OSI Approved :: BSD License",
     "Programming Language :: Python :: 3",
     "Programming Language :: Python :: 3.9",
     "Programming Language :: Python :: 3.10",
@@ -24,6 +20,9 @@ classifiers = [
     "Topic :: Software Development :: Libraries :: Python Modules",
 ]
 
+[project.urls]
+homepage = "https://github.com/joblib/threadpoolctl"
+
 [tool.black]
 line-length = 88
 target_version = ['py39', 'py310', 'py311', 'py312', 'py313']
