--- a/data/mate-settings-daemon.desktop.in.in
+++ b/data/mate-settings-daemon.desktop.in.in
@@ -3,7 +3,7 @@ Type=Application
 Name=MATE Settings Daemon
 Exec=@LIBEXECDIR@/mate-settings-daemon
 OnlyShowIn=MATE;
-X-MATE-Autostart-Phase=Initialization
+X-MATE-Autostart-Phase=Application
 X-MATE-Autostart-Notify=true
 X-MATE-AutoRestart=true
-NoDisplay=true
\ No newline at end of file
+NoDisplay=true
