--- include/lzo/lzodefs.h.orig	2014-06-28 09:11:26.981691990 +0200
+++ include/lzo/lzodefs.h	2014-06-28 09:11:38.989874177 +0200
@@ -970,17 +970,6 @@
 #    error "unexpected configuration - check your compiler defines"
 #  endif
 #endif
-#if (LZO_ARCH_I386)
-#  if (UINT_MAX != LZO_0xffffL) && defined(__i386_int16__)
-#    error "unexpected configuration - check your compiler defines"
-#  endif
-#  if (UINT_MAX != LZO_0xffffffffL) && !defined(__i386_int16__)
-#    error "unexpected configuration - check your compiler defines"
-#  endif
-#  if (ULONG_MAX != LZO_0xffffffffL)
-#    error "unexpected configuration - check your compiler defines"
-#  endif
-#endif
 #if (LZO_ARCH_AMD64 || LZO_ARCH_I386)
 #  if !defined(LZO_TARGET_FEATURE_SSE2)
 #    if defined(__SSE2__)
