--- fasteners/_utils.py.orig
+++ fasteners/_utils.py
@@ -19,7 +19,10 @@
 import logging
 import time
 
-from monotonic import monotonic as now  # noqa
+try:
+    from time import monotonic as now
+except ImportError:
+    from monotonic import monotonic as now
 
 # log level for low-level debugging
 BLATHER = 5
--- setup.py.orig
+++ setup.py
@@ -26,7 +26,6 @@
 
 install_requires = [
     'six',
-    'monotonic>=0.1',
 ]
 
 setup(
@@ -38,6 +37,9 @@
     url='https://github.com/harlowja/fasteners',
     license="ASL 2.0",
     install_requires=install_requires,
+    extras_require={
+        ':python_version<"3.3"': ['monotonic'],
+    },
     classifiers=[
         "Development Status :: 4 - Beta",
         "Topic :: Utilities",
