diff --git lib/Makefile.mk lib/Makefile.mk
index ed9690d..aacb2b1 100644
--- lib/Makefile.mk
+++ lib/Makefile.mk
@@ -10,13 +10,13 @@
 SubDirs :=
 
 # Add submodules.
-SubDirs += asan
+# SubDirs += asan
 SubDirs += builtins
-SubDirs += dfsan
+# SubDirs += dfsan
 SubDirs += interception
-SubDirs += lsan
-SubDirs += msan
+# SubDirs += lsan
+# SubDirs += msan
 SubDirs += profile
-SubDirs += sanitizer_common
-SubDirs += tsan
-SubDirs += ubsan
+# SubDirs += sanitizer_common
+# SubDirs += tsan
+# SubDirs += ubsan
