Index: configure.ac
===================================================================
--- a/configure.ac
+++ b/configure.ac
@@ -212,9 +212,6 @@ fi
 dnl Set version script, to limit the scope of symbols
 DC_SETVERSIONSCRIPT(libcackey.vers, libcackey.syms)
 
-dnl Upate LDFLAGS to include setting the run-time linker path to the same as our compile-time linker
-DC_SYNC_RPATH
-
 dnl If we updated LIBOBJS, update SHLIBOBJS -- must be last.
 DC_SYNC_SHLIBOBJS
 
