--- a/Makefile
+++ b/Makefile
@@ -14,11 +14,11 @@ OBJECTS := yeahconsole.o
 SOURCES := yeahconsole.c
 
 $(TARGET): $(OBJECTS) 
-	$(CC) $(DEFINES) $(INCLUDES) $(LIB_DIRS) $(LIBS) -o $@ $<
+	$(CC) $(DEFINES) $(INCLUDES) $(LIB_DIRS) $(LIBS) -o $@ $< $(LDFLAGS)
 	strip $@
 
 $(OBJECTS): $(SOURCES) 
-	$(CC) $(FLAGS)  $(DEFINES) $(INCLUDES) $(LIB_DIRS) -c -o $@ $<
+	$(CC) $(FLAGS) $(CFLAGS) $(DEFINES) $(INCLUDES) $(LIB_DIRS) -c -o $@ $<
 
 clean:
 	rm -rf $(TARGET) $(OBJECTS)
