--- a/data/meson.build
+++ b/data/meson.build
@@ -47,7 +47,6 @@ desktop_conf.set('appid', characters_application_id)
 desktop_conf.set('bindir', characters_bindir)
 
 i18n.merge_file(
-  'desktop-file',
   type: 'desktop',
   input: configure_file (
     input: characters_name + '.desktop.in',
@@ -61,7 +60,6 @@ i18n.merge_file(
 )
 
 appdata_file = i18n.merge_file(
-  'appdata-file',
   input: characters_name + '.appdata.xml.in',
   output: characters_application_id + '.appdata.xml',
   po_dir: po_dir,
