Index: sunpinyin-3.0.0.rc2/wrapper/ibus/setup/main.py
===================================================================
--- sunpinyin-3.0.0.rc2.orig/wrapper/ibus/setup/main.py
+++ sunpinyin-3.0.0.rc2/wrapper/ibus/setup/main.py
@@ -503,7 +503,6 @@ class MainWindow():
         locale.setlocale(locale.LC_ALL, "")
         localedir = os.getenv("IBUS_LOCALEDIR")
         gettext.bindtextdomain(GETTEXT_PACKAGE, localedir)
-        gettext.bind_textdomain_codeset(GETTEXT_PACKAGE, "UTF-8")
 
     def __init_options(self):
         self.__fuzzy_setup = FuzzySetupDialog()
