--- a/SConstruct
+++ b/SConstruct
@@ -7,7 +7,7 @@ import sys
 
 import SCons.Errors
 
-version = "2.0.4"
+version = "3.0.0-rc2"
 abi_major = 3
 abi_minor = 0
 
--- a/wrapper/ibus/SConstruct
+++ b/wrapper/ibus/SConstruct
@@ -14,7 +14,7 @@ sources = ['src/debug.cpp',
 locales = ['zh_CN']
 
 cflags = '-O2 -g -pipe '
-version = '2.0.3'
+version = '3.0.0-rc2'
 
 AddOption('--prefix', dest='prefix', metavar='DIR',
           help='installation prefix')
