--- lisp-kernel/platform-linuxx8632.h.orig	2015-11-06 20:10:11.000000000 +0000
+++ lisp-kernel/platform-linuxx8632.h	2017-11-14 15:25:38.758675003 +0000
@@ -20,7 +20,7 @@
 #define PLATFORM_CPU PLATFORM_CPU_X86
 #define PLATFORM_WORD_SIZE PLATFORM_WORD_SIZE_32
 
-typedef struct ucontext ExceptionInformation;
+typedef struct ucontext_t ExceptionInformation;
 
 #define MAXIMUM_MAPPABLE_MEMORY (9U<<28)
 #define IMAGE_BASE_ADDRESS 0x10000000
--- lisp-kernel/platform-linuxx8664.h.orig	2015-11-06 20:10:11.000000000 +0000
+++ lisp-kernel/platform-linuxx8664.h	2017-11-14 15:25:24.551674270 +0000
@@ -20,7 +20,7 @@
 #define PLATFORM_CPU PLATFORM_CPU_X86
 #define PLATFORM_WORD_SIZE PLATFORM_WORD_SIZE_64
 
-typedef struct ucontext ExceptionInformation;
+typedef struct ucontext_t ExceptionInformation;
 
 #define MAXIMUM_MAPPABLE_MEMORY (512L<<30L)
 #define IMAGE_BASE_ADDRESS 0x300000000000L
--- lisp-kernel/x86-exceptions.c.orig	2017-11-14 15:28:32.000000000 +0000
+++ lisp-kernel/x86-exceptions.c	2017-11-14 15:28:53.053685026 +0000
@@ -1679,7 +1679,7 @@
 	siginfo_t  *pinfo;
 	void  *puc;
 	siginfo_t info;
-	struct ucontext uc;
+	struct ucontext_t uc;
 	struct _fpstate fpstate;
 	char retcode[8];
 };
@@ -2427,7 +2427,7 @@
      It's easier to just reserve that page here than it would be to
      change copy_ucontext().
   */
-  stack.ss_size -= sizeof(struct ucontext);
+  stack.ss_size -= sizeof(struct ucontext_t);
 #endif
   if (sigaltstack(&stack, NULL) != 0) {
     perror("sigaltstack");
