--- lisp-kernel/linuxx8664/Makefile.orig
+++ lisp-kernel/linuxx8664/Makefile
@@ -69,7 +69,7 @@
 all:	../../lx86cl64
 
 
-OSLIBS = -ldl -lm -lpthread -lrt
+OSLIBS = -ldl -lm -lpthread -lrt -no-pie
 LINK_MAP = ./elf_x86_64.x
 USE_LINK_MAP = # -T ./elf_x86_64.x
 
--- lisp-kernel/linuxx8632/Makefile.orig
+++ lisp-kernel/linuxx8632/Makefile
@@ -69,7 +69,7 @@
 all:	../../lx86cl
 
 
-OSLIBS = -ldl -lm -lpthread -lrt
+OSLIBS = -ldl -lm -lpthread -lrt -no-pie
 LINK_SCRIPT = # ./elf_x86_32.x
 USE_LINK_SCRIPT = # -T $(LINK_SCRIPT)
 
