diff -up chromium-120.0.6099.56/third_party/libc++/src/include/__config.me chromium-120.0.6099.56/third_party/libc++/src/include/__config
--- chromium-120.0.6099.56/third_party/libc++/src/include/__config.me	2023-12-03 00:05:51.254483474 +0100
+++ chromium-120.0.6099.56/third_party/libc++/src/include/__config	2023-12-03 00:06:05.872863168 +0100
@@ -32,11 +32,7 @@
 
 // Warn if a compiler version is used that is not supported anymore
 // LLVM RELEASE Update the minimum compiler versions
-#  if defined(_LIBCPP_CLANG_VER)
-#    if _LIBCPP_CLANG_VER < 1500
-#      warning "Libc++ only supports Clang 15 and later"
-#    endif
-#  elif defined(_LIBCPP_APPLE_CLANG_VER)
+#  if defined(_LIBCPP_APPLE_CLANG_VER)
 #    if _LIBCPP_APPLE_CLANG_VER < 1500
 #      warning "Libc++ only supports AppleClang 15 and later"
 #    endif
