diff -up chromium-120.0.6099.35/build/config/linux/atspi2/BUILD.gn.me chromium-120.0.6099.35/build/config/linux/atspi2/BUILD.gn
--- chromium-120.0.6099.35/build/config/linux/atspi2/BUILD.gn.me	2023-11-26 16:14:15.364064126 +0100
+++ chromium-120.0.6099.35/build/config/linux/atspi2/BUILD.gn	2023-11-26 16:41:16.877321990 +0100
@@ -21,6 +21,9 @@ if (use_atk) {
     minor = atspi_version[1]
     micro = atspi_version[2]
 
+    # gn workaround for the error: Assignment had no effect
+    print("ATSPI Version: $major.$minor.$micro")
+
     # ATSPI 2.49.90 now defines these for us and it's an error for us to
     # redefine them on the compiler command line.
     # See ATSPI 927344a34cd5bf81fc64da4968241735ecb4f03b
