--- a/script/apply_all_patches.py	2023-05-03 15:27:52.000000000 +0200
+++ -	2023-05-04 23:12:01.430619050 +0200
@@ -14,6 +14,7 @@
     if os.path.exists(repo):
       git.import_patches(repo=repo, patch_data=patch_from_dir(patch_dir),
         threeway=threeway is not None,
+        exclude=['third_party/blink/tools/**', 'test/mjsunit/**', 'content/test/**', 'test/cctest/**', 'test/unittests/**', 'third_party/blink/web_tests/**', '.gitignore'],
         committer_name="Electron Scripts", committer_email="scripts@electron")
 
 
