--- metadata/grid.xml.in	2016-04-10 12:23:24.000000000 -0700
+++ metadata/grid.xml.in	2017-03-01 23:26:45.521723881 -0800
@@ -52,7 +52,505 @@
 		    <_long>Move window to the bottom right corner</_long>
 		    <default>&lt;Primary&gt;&lt;Alt&gt;KP_3</default>
 		</option>
+		<option name="put_maximize_key" type="key">
+		    <_short>Maximize</_short>
+		    <_long>Maximize window</_long>
+		    <default>&lt;Primary&gt;&lt;Alt&gt;KP_0</default>
+		</option>
+		<option name="cycle_sizes" type="key">
+		    <_short>Cycle Through Multiple Sizes</_short>
+		    <_long>Cycle through multiple different sizes by using the same keyboard shortcut multiple times in a row.</_long>
+		    <default>false</default>
+		</option>
+	    </group>
+	    <group>
+		<_short>Corners / Edges</_short>
+		<subgroup>
+		    <_short>Resize Actions</_short>
+		    <_long>Window resize action</_long>
+		    <option name="top_left_corner_action" type="int">
+			<_short>Top Left Corner</_short>
+			<_long>Action to be performed when window is dropped on the top left corner</_long>
+			<default>7</default>
+			<min>0</min>
+			<max>10</max>
+			<desc>
+			    <value>0</value>
+			    <_name>None</_name>
+			</desc>
+			<desc>
+			     <value>1</value>
+			      <_name>Bottom Left Corner</_name>
+			</desc>
+			<desc>
+			    <value>2</value>
+			    <_name>Bottom Half</_name>
+			</desc>
+			<desc>
+			    <value>3</value>
+			    <_name>Bottom Right Corner</_name>
+			</desc>
+			<desc>
+			    <value>4</value>
+			    <_name>Left Half</_name>
+			</desc>
+			<desc>
+			    <value>5</value>
+			    <_name>Fill Screen</_name>
+			</desc>
+			<desc>
+			    <value>6</value>
+			    <_name>Right Half</_name>
+			</desc>
+			<desc>
+			    <value>7</value>
+			    <_name>Top Left Corner</_name>
+			</desc>
+			<desc>
+			    <value>8</value>
+			    <_name>Top Half</_name>
+			</desc>
+			<desc>
+			    <value>9</value>
+			    <_name>Top Right Corner</_name>
+			</desc>
+			<desc>
+			    <value>10</value>
+			    <_name>Maximize</_name>
+			</desc>
+		    </option>
+		    <option name="top_edge_action" type="int">
+			<_short>Top Edge</_short>
+			<_long>Action to be performed when window is dropped on the top edge</_long>
+			<default>10</default>
+			<min>0</min>
+			<max>10</max>
+			<desc>
+			    <value>0</value>
+			    <_name>None</_name>
+			</desc>
+			<desc>
+			    <value>1</value>
+			    <_name>Bottom Left Corner</_name>
+			</desc>
+			<desc>
+			    <value>2</value>
+			    <_name>Bottom Half</_name>
+			</desc>
+			<desc>
+			    <value>3</value>
+			    <_name>Bottom Right Corner</_name>
+			</desc>
+			<desc>
+			    <value>4</value>
+			    <_name>Left Half</_name>
+			</desc>
+			<desc>
+			    <value>5</value>
+			    <_name>Fill Screen</_name>
+			</desc>
+			<desc>
+			    <value>6</value>
+			    <_name>Right Half</_name>
+			</desc>
+			<desc>
+			    <value>7</value>
+			    <_name>Top Left Corner</_name>
+			</desc>
+			<desc>
+			    <value>8</value>
+			    <_name>Top Half</_name>
+			</desc>
+			<desc>
+			    <value>9</value>
+			    <_name>Top Right Corner</_name>
+			</desc>
+			<desc>
+			    <value>10</value>
+			    <_name>Maximize</_name>
+			</desc>
+		    </option>
+		    <option name="top_right_corner_action" type="int">
+			<_short>Top Right Corner</_short>
+			<_long>Action to be performed when window is dropped on the top right corner</_long>
+			<default>9</default>
+			<min>0</min>
+			<max>10</max>
+			<desc>
+			    <value>0</value>
+			    <_name>None</_name>
+			</desc>
+			<desc>
+			    <value>1</value>
+			    <_name>Bottom Left Corner</_name>
+			</desc>
+			<desc>
+			    <value>2</value>
+			    <_name>Bottom Half</_name>
+			</desc>
+			<desc>
+			    <value>3</value>
+			    <_name>Bottom Right Corner</_name>
+			</desc>
+			<desc>
+			    <value>4</value>
+			    <_name>Left Half</_name>
+			</desc>
+			<desc>
+			    <value>5</value>
+			    <_name>Fill Screen</_name>
+			</desc>
+			<desc>
+			    <value>6</value>
+			    <_name>Right Half</_name>
+			</desc>
+			<desc>
+			    <value>7</value>
+			    <_name>Top Left Corner</_name>
+			</desc>
+			<desc>
+			    <value>8</value>
+			    <_name>Top Half</_name>
+			</desc>
+			<desc>
+			    <value>9</value>
+			    <_name>Top Right Corner</_name>
+			</desc>
+			<desc>
+			    <value>10</value>
+			    <_name>Maximize</_name>
+			</desc>
+		    </option>
+		    <option name="left_edge_action" type="int">
+			<_short>Left Edge</_short>
+			<_long>Action to be performed when window is dropped on the left edge</_long>
+			<default>4</default>
+			<min>0</min>
+			<max>10</max>
+			<desc>
+			    <value>0</value>
+			    <_name>None</_name>
+			</desc>
+			<desc>
+			    <value>1</value>
+			    <_name>Bottom Left Corner</_name>
+			</desc>
+			<desc>
+			    <value>2</value>
+			    <_name>Bottom Half</_name>
+			</desc>
+			<desc>
+			    <value>3</value>
+			    <_name>Bottom Right Corner</_name>
+			</desc>
+			<desc>
+			    <value>4</value>
+			    <_name>Left Half</_name>
+			</desc>
+			<desc>
+			    <value>5</value>
+			    <_name>Fill Screen</_name>
+			</desc>
+			<desc>
+			    <value>6</value>
+			    <_name>Right Half</_name>
+			</desc>
+			<desc>
+			    <value>7</value>
+			    <_name>Top Left Corner</_name>
+			</desc>
+			<desc>
+			    <value>8</value>
+			    <_name>Top Half</_name>
+			</desc>
+			<desc>
+			    <value>9</value>
+			    <_name>Top Right Corner</_name>
+			</desc>
+			<desc>
+			    <value>10</value>
+			    <_name>Maximize</_name>
+			</desc>
+		    </option>
+		    <option name="right_edge_action" type="int">
+			<_short>Right Edge</_short>
+			<_long>Action to be performed when window is dropped on the right edge</_long>
+			<default>6</default>
+			<min>0</min>
+			<max>10</max>
+			<desc>
+			    <value>0</value>
+			    <_name>None</_name>
+			</desc>
+			<desc>
+			    <value>1</value>
+			    <_name>Bottom Left Corner</_name>
+			</desc>
+			<desc>
+			    <value>2</value>
+			    <_name>Bottom Half</_name>
+			</desc>
+			<desc>
+			    <value>3</value>
+			    <_name>Bottom Right Corner</_name>
+			</desc>
+			<desc>
+			    <value>4</value>
+			    <_name>Left Half</_name>
+			</desc>
+			<desc>
+			    <value>5</value>
+			    <_name>Fill Screen</_name>
+			</desc>
+			<desc>
+			    <value>6</value>
+			    <_name>Right Half</_name>
+			</desc>
+			<desc>
+			    <value>7</value>
+			    <_name>Top Left Corner</_name>
+			</desc>
+			<desc>
+			    <value>8</value>
+			    <_name>Top Half</_name>
+			</desc>
+			<desc>
+			    <value>9</value>
+			    <_name>Top Right Corner</_name>
+			</desc>
+			<desc>
+			    <value>10</value>
+			    <_name>Maximize</_name>
+			</desc>
+		    </option>
+		    <option name="bottom_left_corner_action" type="int">
+			<_short>Bottom Left Corner</_short>
+			<_long>Action to be performed when window is dropped on the bottom left corner</_long>
+			<default>1</default>
+			<min>0</min>
+			<max>10</max>
+			<desc>
+			    <value>0</value>
+			    <_name>None</_name>
+			</desc>
+			<desc>
+			    <value>1</value>
+			    <_name>Bottom Left Corner</_name>
+			</desc>
+			<desc>
+			    <value>2</value>
+			    <_name>Bottom Half</_name>
+			</desc>
+			<desc>
+			    <value>3</value>
+			    <_name>Bottom Right Corner</_name>
+			</desc>
+			<desc>
+			    <value>4</value>
+			    <_name>Left Half</_name>
+			</desc>
+			<desc>
+			    <value>5</value>
+			    <_name>Fill Screen</_name>
+			</desc>
+			<desc>
+			    <value>6</value>
+			    <_name>Right Half</_name>
+			</desc>
+			<desc>
+			    <value>7</value>
+			    <_name>Top Left Corner</_name>
+			</desc>
+			<desc>
+			    <value>8</value>
+			    <_name>Top Half</_name>
+			</desc>
+			<desc>
+			    <value>9</value>
+			    <_name>Top Right Corner</_name>
+			</desc>
+			<desc>
+			    <value>10</value>
+			    <_name>Maximize</_name>
+			</desc>
+		    </option>
+		    <option name="bottom_edge_action" type="int">
+			<_short>Bottom Edge</_short>
+			<_long>Action to be performed when window is dropped on the bottom edge</_long>
+			<default>2</default>
+			<min>0</min>
+			<max>10</max>
+			<desc>
+			    <value>0</value>
+			    <_name>None</_name>
+			</desc>
+			<desc>
+			    <value>1</value>
+			    <_name>Bottom Left Corner</_name>
+			</desc>
+			<desc>
+			    <value>2</value>
+			    <_name>Bottom Half</_name>
+			</desc>
+			<desc>
+			    <value>3</value>
+			    <_name>Bottom Right Corner</_name>
+			</desc>
+			<desc>
+			    <value>4</value>
+			    <_name>Left Half</_name>
+			</desc>
+			<desc>
+			    <value>5</value>
+			    <_name>Fill Screen</_name>
+			</desc>
+			<desc>
+			    <value>6</value>
+			    <_name>Right Half</_name>
+			</desc>
+			<desc>
+			    <value>7</value>
+			    <_name>Top Left Corner</_name>
+			</desc>
+			<desc>
+			    <value>8</value>
+			    <_name>Top Half</_name>
+			</desc>
+			<desc>
+			    <value>9</value>
+			    <_name>Top Right Corner</_name>
+			</desc>
+			<desc>
+			    <value>10</value>
+			    <_name>Maximize</_name>
+			</desc>
+		    </option>
+		    <option name="bottom_right_corner_action" type="int">
+			<_short>Bottom Right Corner</_short>
+			<_long>Action to be performed when window is dropped on the bottom right corner</_long>
+			<default>3</default>
+			<min>0</min>
+			<max>10</max>
+			<desc>
+			    <value>0</value>
+			    <_name>None</_name>
+			</desc>
+			<desc>
+			    <value>1</value>
+			    <_name>Bottom Left Corner</_name>
+			</desc>
+			<desc>
+			    <value>2</value>
+			    <_name>Bottom Half</_name>
+			</desc>
+			<desc>
+			    <value>3</value>
+			    <_name>Bottom Right Corner</_name>
+			</desc>
+			<desc>
+			    <value>4</value>
+			    <_name>Left Half</_name>
+			</desc>
+			<desc>
+			    <value>5</value>
+			    <_name>Fill Screen</_name>
+			</desc>
+			<desc>
+			    <value>6</value>
+			    <_name>Right Half</_name>
+			</desc>
+			<desc>
+			    <value>7</value>
+			    <_name>Top Left Corner</_name>
+			</desc>
+			<desc>
+			    <value>8</value>
+			    <_name>Top Half</_name>
+			</desc>
+			<desc>
+			    <value>9</value>
+			    <_name>Top Right Corner</_name>
+			</desc>
+			<desc>
+			    <value>10</value>
+			    <_name>Maximize</_name>
+			</desc>
+		    </option>
+		</subgroup>
+		<subgroup>
+		    <_short>Thresholds</_short>
+		    <option name="left_edge_threshold" type="int">
+			<_short>Left Edge</_short>
+			<_long>Maximum number of pixels from the left edge a window can be dropped</_long>
+			<default>5</default>
+			<min>0</min>
+			<max>500</max>
+		    </option>
+		    <option name="right_edge_threshold" type="int">
+			<_short>Right Edge</_short>
+			<_long>Maximum number of pixels from the right edge a window can be dropped</_long>
+			<default>5</default>
+			<min>0</min>
+			<max>500</max>
+		    </option>
+		    <option name="top_edge_threshold" type="int">
+			<_short>Top Edge</_short>
+			<_long>Maximum number of pixels from the top edge a window can be dropped</_long>
+			<default>5</default>
+			<min>0</min>
+			<max>500</max>
+		    </option>
+		    <option name="bottom_edge_threshold" type="int">
+			<_short>Bottom Edge</_short>
+			<_long>Maximum number of pixels from the Bottom edge a window can be dropped</_long>
+			<default>5</default>
+			<min>0</min>
+			<max>500</max>
+		    </option>
+		</subgroup>
+	    </group>
+	    <group>
+		<_short>Appearance</_short>
+		<option name="draw_indicator" type="bool">
+		    <_short>Draw Indicator</_short>
+		    <_long>Draw Window Resize Indicator</_long>
+		    <default>true</default>
+		</option>
+		<option name="animation_duration" type="int">
+		    <_short>Preview Animation Duration</_short>
+		    <_long>Preview animation duration (in ms).</_long>
+		    <default>350</default>
+		    <min>0</min>
+		</option>
+                <option name="outline_thickness" type="float">
+                    <_short>Outline Thickness</_short>
+                    <_long>Outline Thickness</_long>
+                    <default>2.0</default>
+                    <min>1.0</min>
+                    <max>8.0</max>
+                    <precision>1.0</precision>
+                </option>
+		<option name="outline_color" type="color">
+		    <_short>Outline Color</_short>
+		    <_long>Color of the resize indicator outline</_long>
+		    <default>
+			<red>0x2f2f</red>
+			<green>0x2f2f</green>
+			<blue>0x4f4f</blue>
+			<alpha>0x9f9f</alpha>
+		    </default>
+		</option>
+		<option name="fill_color" type="color">
+		    <_short>Fill Color</_short>
+		    <_long>Fill color of the resize indicator</_long>
+		    <default>
+			<red>0x2f2f</red>
+			<green>0x2f2f</green>
+			<blue>0x4f4f</blue>
+			<alpha>0x4f4f</alpha>
+		    </default>
+		</option>
 	    </group>
 	</display>
     </plugin>
-</compiz>
+</compiz>
\ No newline at end of file
