From 8789a3365282a4f5604e090a10c960e710d240b9 Mon Sep 17 00:00:00 2001
From: "Andrew J. Hesford" <ajh@sideband.org>
Date: Tue, 6 Jun 2023 10:05:25 -0400
Subject: [PATCH] meson: allow proper selection of NumPy, Pythran in cross
 builds

---
 skimage/meson.build | 62 +++++++++++++++++++++++++++------------------
 1 file changed, 38 insertions(+), 24 deletions(-)

diff --git a/skimage/meson.build b/skimage/meson.build
index 28c831312..c168389d7 100644
--- a/skimage/meson.build
+++ b/skimage/meson.build
@@ -29,37 +29,51 @@ if is_windows
   endif
 endif
 
-# NumPy include directory - needed in all submodules
-incdir_numpy = run_command(py3,
-  [
-    '-c',
-    'import os; os.chdir(".."); import numpy; print(numpy.get_include())'
-  ],
-  check: true
-).stdout().strip()
+# Both NumPy and Pythran require header files that may differ between the build
+# system and the host system in a cross-compilation environment. To accommodate
+# these cases, we can query user-defined properties that can be specified in
+# the 'properties' section of a Meson cross file:
+#
+#   [properties]
+#   numpy-include-dir = '/path/to/numpy/includes'
+#   pythran-include-dir = '/path/to/pythran/includes'
+#
+# In the absence of explicitly configured paths, just run the build Python and
+# try to query the Python packages for their paths directly.
+
+# NumPy include directory
+incdir_numpy = meson.get_external_property('numpy-include-dir', 'not-given')
+if incdir_numpy == 'not-given'
+  # If not specified, try to query NumPy from the build python
+  incdir_numpy = run_command(py3,
+    [
+      '-c',
+      'import os; os.chdir(".."); import numpy; print(numpy.get_include())'
+    ],
+    check: true
+  ).stdout().strip()
+endif
 
 inc_np = include_directories(incdir_numpy)
 
 cc = meson.get_compiler('c')
 
-# Pythran include directory and build flags
-use_pythran = run_command(py3,
-  [
-    '-c',
-    'import os; print(os.environ.get("SCIPY_USE_PYTHRAN", 1))'
-  ],
-  check: true
-).stdout().strip() == '1'
-
-incdir_pythran = run_command(py3,
-  [
-    '-c',
-    'import os; os.chdir(".."); import pythran; print(os.path.dirname(pythran.__file__));'
-  ],
-  check: true
-).stdout().strip()
+# Pythran include directory
+incdir_pythran = meson.get_external_property('pythran-include-dir', 'not-given')
+if incdir_pythran == 'not-given'
+  # If not specified, try to query Pythran from the build python
+  incdir_pythran = run_command(py3,
+    [
+      '-c',
+      'import os; os.chdir(".."); import pythran; print(os.path.dirname(pythran.__file__));'
+    ],
+    check: true
+  ).stdout().strip()
+endif
+
 inc_pythran = include_directories(incdir_pythran)
 
+# Pythran build flags
 cpp_args_pythran = [
   '-DENABLE_PYTHON_MODULE',
   '-D__PYTHRAN__=3',
-- 
2.41.0

