--- a/include/cecloader.h	2020-07-13 12:18:33.000000000 +0200
+++ -	2023-01-15 16:19:43.758391565 +0100
@@ -172,7 +172,7 @@
     if (!g_libCEC)
     {
       std::cout << dlerror() << std::endl;
-      return NULL;
+      return false;
     }
   }
 
@@ -181,7 +181,7 @@
   if (!LibCecBootloader)
   {
     std::cout << "cannot find CECStartBootloader" << std::endl;
-    return NULL;
+    return false;
   }
 
   bool bReturn = LibCecBootloader();
