diff --git src/libcec/cmake/SetBuildInfo.cmake src/libcec/cmake/SetBuildInfo.cmake
index 3402f9f..b121361 100644
--- a/src/libcec/cmake/SetBuildInfo.cmake
+++ b/src/libcec/cmake/SetBuildInfo.cmake
@@ -40,7 +40,7 @@ else()
   # add host on which this was built to compile info
   find_program(HAVE_HOSTNAME_BIN hostname /bin /usr/bin /usr/local/bin)
   if(HAVE_HOSTNAME_BIN)
-    exec_program(hostname ARGS -f OUTPUT_VARIABLE BUILD_HOST RETURN_VALUE RETURN_HOST)
+    exec_program(hostname ARGS OUTPUT_VARIABLE BUILD_HOST RETURN_VALUE RETURN_HOST)
     if (RETURN_HOST)
       exec_program(hostname OUTPUT_VARIABLE BUILD_HOST)
     endif()
