--- ./include/cecloader.h.orig	2023-09-30 07:58:57.250505723 -0400
+++ ./include/cecloader.h	2023-09-30 08:00:08.028027063 -0400
@@ -88,7 +88,7 @@
   if (!g_libCEC)
     g_libCEC = LoadLibrary(strLib ? strLib : "cec.dll");
   if (!g_libCEC)
-    return NULL;
+    return false;
 
   typedef bool (__cdecl*_LibCecBootloader)(void);
   _LibCecBootloader LibCecBootloader;
@@ -172,7 +172,7 @@
     if (!g_libCEC)
     {
       std::cout << dlerror() << std::endl;
-      return NULL;
+      return false;
     }
   }
 
@@ -181,7 +181,7 @@
   if (!LibCecBootloader)
   {
     std::cout << "cannot find CECStartBootloader" << std::endl;
-    return NULL;
+    return false;
   }
 
   bool bReturn = LibCecBootloader();
