diff --git modules/60crypt-ssh/module-setup.sh modules/60crypt-ssh/module-setup.sh
index c3b6584..80e7dbe 100644
--- a/modules/60crypt-ssh/module-setup.sh
+++ b/modules/60crypt-ssh/module-setup.sh
@@ -40,7 +40,7 @@ install() {
     
     case ${state} in
       GENERATE )
-        ssh-keygen -t $keyType -f $osshKey -q -N "" || {
+        ssh-keygen -t $keyType -f $osshKey -q -N "" -m PEM || {
           derror "SSH ${msgKeyType} key creation failed"
           rm -rf "$tmpDir"
           return 1
