--- a/Makefile.in
+++ b/Makefile.in
@@ -42,10 +42,11 @@
 	$(CC) $(CFLAGS) -c mat.c
 
 install: all
-	$(INSTALL) -g bin -m 755 -o root adjtimex $(bindir)/adjtimex
-	$(INSTALL) -d -g root -m 755 -o root $(mandir)
+	$(INSTALL) -d -g root -m 755 -o root $(DESTDIR)$(bindir)
+	$(INSTALL) -g bin -m 755 -o root adjtimex $(DESTDIR)$(bindir)/adjtimex
+	$(INSTALL) -d -g root -m 755 -o root $(DESTDIR)$(mandir)
 	-$(INSTALL) -g root -m 644 -o root $(srcdir)/adjtimex.8 \
-		$(mandir)/adjtimex.$(manext)
+		$(DESTDIR)$(mandir)/adjtimex.$(manext)
 
 uninstall:
 	rm -f $(bindir)/adjtimex $(mandir)/adjtimex.$(manext)
