Author: Marcin Kolny <marcin.kolny@gmail.com>
Description: Gst::AudioClock: auto generate some audioclock methods
Origin: https://bugzilla.gnome.org/show_bug.cgi?id=783628
Last-update: 2017-07-01

--- a/gstreamer/gstreamermm/audioclock.cc
+++ b/gstreamer/gstreamermm/audioclock.cc
@@ -2,6 +2,7 @@
 
 
 #include <glibmm.h>
+#include <gst/gstversion.h>
 
 #include <gstreamermm/audioclock.h>
 #include <gstreamermm/private/audioclock_p.h>
