--- CMakeLists.txt	2017-03-20 02:47:25.000000000 -0600
+++ CMakeLists.txt	2017-04-05 18:02:54.871910269 -0600
@@ -524,10 +524,6 @@
 CHECK_C_SOURCE_RUNS("#include \"${CMAKE_SOURCE_DIR}/iconv-detect.c\"" _correct_iconv)
 unset(CMAKE_REQUIRED_LIBRARIES)
 
-if(NOT _correct_iconv)
-	message(FATAL_ERROR "You need to install a working iconv implementation, such as ftp://ftp.gnu.org/pub/gnu/libiconv")
-endif(NOT _correct_iconv)
-
 # ******************************
 # Backtraces for debugging
 # ******************************
