--- a/cmake/modules/FindPhonenumber.cmake
+++ b/cmake/modules/FindPhonenumber.cmake
@@ -53,20 +53,7 @@ unset(optupper)
 set(CMAKE_REQUIRED_DEFINITIONS ${PHONENUMBER_DEFINITIONS})
 set(CMAKE_REQUIRED_INCLUDES ${PHONENUMBER_INCLUDE_DIRS})
 
-foreach(lib boost_thread-mt boost_thread)
-	set(CMAKE_REQUIRED_LIBRARIES ${PHONENUMBER_LDFLAGS} -l${lib})
-	CHECK_CXX_SOURCE_COMPILES("#include <phonenumbers/phonenumberutil.h>
-
-				int main(void) {
-					i18n::phonenumbers::PhoneNumberUtil::GetInstance();
-					return 0;
-				}" phone_number_with_${lib})
-	if(phone_number_with_${lib})
-		set(ENABLE_PHONENUMBER ON)
-		set(PHONENUMBER_LDFLAGS ${CMAKE_REQUIRED_LIBRARIES})
-		break()
-	endif(phone_number_with_${lib})
-endforeach(lib)
+set(ENABLE_PHONENUMBER ON)
 
 if(NOT ENABLE_PHONENUMBER)
 	message(FATAL_ERROR "libphonenumber cannot be used. Use -DWITH_PHONENUMBER=PATH to specify the library prefix, or -DWITH_PHONENUMBER=OFF to disable it.")
