--- a/cmake/modules/GObjectIntrospection.cmake
+++ b/cmake/modules/GObjectIntrospection.cmake
@@ -122,7 +122,7 @@
 		file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/${_gir_name}_files "${_gir_files}")
 
 		add_custom_command(
-			COMMAND ${CMAKE_COMMAND} -E env "CC='${CMAKE_C_COMPILER}'" LDFLAGS=
+			COMMAND ${CMAKE_COMMAND} -E env "CC='${CMAKE_C_COMPILER}'"
 				${INTROSPECTION_SCANNER_ENV}
 				${G_IR_SCANNER}
 				${INTROSPECTION_SCANNER_ARGS}
