see http://trac.osgeo.org/geos/ticket/222

--- a/src/inlines.cpp	2020-12-09 18:17:36.000000000 +0100
+++ b/src/inlines.cpp	2021-02-02 09:57:11.270098998 +0100
@@ -62,7 +62,9 @@
 #include <geos/geomgraph/TopologyLocation.inl>
 #include <geos/geomgraph/index/SegmentIntersector.inl>
 #include <geos/noding/snapround/HotPixel.inl>
+#if ! defined(__ARM_EABI__)
 #include <geos/noding/BasicSegmentString.inl>
+#endif // !defined __ARM_EABI__
 #include <geos/noding/MCIndexNoder.inl>
 
 #endif // defined __CYGWIN__
