--- a/lib/CMakeLists.txt
+++ b/lib/CMakeLists.txt
@@ -31,14 +31,6 @@
 # Version macros
 target_compile_definitions(spotify-qt-lib PUBLIC LIB_VERSION="v${PROJECT_VERSION}")
 
-# Check if using GCC
-if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
-	# Used by crash handler
-	target_compile_definitions(spotify-qt-lib PRIVATE IS_GNU_CXX)
-	# Enable all compiler warnings
-	target_compile_options(spotify-qt-lib PRIVATE -Wall -Wextra)
-endif ()
-
 # Disable "secure" warnings on MSVC
 if (MSVC)
 	target_compile_definitions(spotify-qt-lib PRIVATE _CRT_SECURE_NO_WARNINGS)
