--- net/features.gni	2021-05-20 04:17:11.000000000 +0200
+++ -	2021-05-26 16:03:25.826113608 +0200
@@ -28,6 +28,9 @@
   # Reporting not used on iOS.
   enable_reporting = !is_ios
 
+  # Force Posix dns config
+  force_posix_dns = true
+
   # Includes the transport security state preload list. This list includes
   # mechanisms (e.g. HSTS, HPKP) to enforce trusted connections to a significant
   # set of hardcoded domains. While this list has a several hundred KB of binary
--- net/dns/BUILD.gn	2021-05-20 04:17:11.000000000 +0200
+++ -	2021-05-26 16:05:01.510622102 +0200
@@ -114,7 +114,7 @@
         "dns_config_service_android.cc",
         "dns_config_service_android.h",
       ]
-    } else if (is_linux) {
+    } else if (is_linux && !force_posix_dns) {
       sources += [
         "dns_config_service_linux.cc",
         "dns_config_service_linux.h",
@@ -433,7 +433,7 @@
 
   if (is_android) {
     sources += [ "dns_config_service_android_unittest.cc" ]
-  } else if (is_linux) {
+  } else if (is_linux && !force_posix_dns) {
     sources += [ "dns_config_service_linux_unittest.cc" ]
   } else if (is_posix) {
     sources += [ "dns_config_service_posix_unittest.cc" ]
