--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,8 @@
 CC = gcc
-CFLAGS = -Wextra -Wshadow -Wunreachable-code -Wcast-align -Wuninitialized -finline-functions -finline-small-functions
+CFLAGS = -Wextra -Wshadow -Wunreachable-code -Wcast-align -Wuninitialized
+MYCFLAGS = -finline-functions -finline-small-functions
 
-LDFLAGS = -lxcb -lxcb-util -lxcb-keysyms -lxcb-ewmh -lxcb-icccm -lxcb-randr -lxcb-xinerama -lxcb-cursor
+LIBS = -lxcb -lxcb-util -lxcb-keysyms -lxcb-ewmh -lxcb-icccm -lxcb-randr -lxcb-xinerama -lxcb-cursor
 SRC_FILES = ./src/zwm.c ./src/logger.c ./src/tree.c ./src/config_parser.c ./src/queue.c
 HEADER_FILES = ./src/logger.h ./src/tree.h ./src/type.h ./src/zwm.h ./src/config_parser.h ./src/helper.h ./src/queue.h
 OBJ_FILES = $(SRC_FILES:.c=.o)
@@ -15,15 +16,15 @@
 MANPAGE = zwm.1
 
 $(TARGET): $(OBJ_FILES)
-	$(CC) $(CFLAGS) -o $@ $^ $(LDFLAGS)
+	$(CC) $(CFLAGS) $(MYCFLAGS) -o $@ $^ $(LDFLAGS) $(LIBS)
 
 %.o: %.c $(HEADER_FILES)
-	$(CC) $(CFLAGS) -c $< -o $@
+	$(CC) $(CFLAGS) $(MYCLAGS) -c $< -o $@
 
 clean:
 	rm -f $(TARGET) $(OBJ_FILES)
 
-install: $(TARGET)
+install:
 	mkdir -p "$(DESTDIR)$(BINDIR)"
 	cp -pf $(TARGET) "$(DESTDIR)$(BINDIR)"
 	mkdir -p "$(DESTDIR)$(MANDIR)"
