From 2c95b707cafe08774c1b4d4fad7a7ab88982ea88 Mon Sep 17 00:00:00 2001
From: Piotr Wójcik <chocimier@tlen.pl>
Date: Sat, 22 Apr 2023 19:37:25 +0200
Subject: exchange-desktop-file-for-cross-compilation


diff --git a/src/kcm/CMakeLists.txt b/src/kcm/CMakeLists.txt
index beb0a56..e3b6c4c 100644
--- a/src/kcm/CMakeLists.txt
+++ b/src/kcm/CMakeLists.txt
@@ -8,4 +8,3 @@ target_link_libraries(kcm_kaccounts
     )
 
 kpackage_install_package(package kcm_kaccounts kcms)
-kcmutils_generate_desktop_file(kcm_kaccounts)
-- 
2.40.0

