--- CMakeLists.txt.orig	2015-05-23 09:25:05.721145281 +0200
+++ CMakeLists.txt	2015-05-23 09:25:49.481580405 +0200
@@ -35,6 +35,7 @@ INCLUDE(MacroLibrary)
 INCLUDE(MacroOptionalAddSubdirectory)
 INCLUDE(FindPackageHandleStandardArgs)
 
+
 # NOTE: Libraw 0.16.x is prefered version to use because it's ported to Cmake with full features supported.
 # Until libraw 0.16.0 is release (ends of 2013), we will support previous version (with limited support)
 FIND_PACKAGE(LibRaw 0.15)
@@ -73,6 +74,7 @@ INCLUDE_DIRECTORIES(${QDBUS_INCLUDE_DIRS
 
 SET(LIBKDCRAW_AREA_CODE_GENERAL 51002)
 ADD_DEFINITIONS(-DKDE_DEFAULT_DEBUG_AREA=${LIBKDCRAW_AREA_CODE_GENERAL})
+ADD_DEFINITIONS(${KDE4_ENABLE_EXCEPTIONS})
 
 # ==================================================================================================
 # Information to update before to release this library.
