--- cmd/grv/vendor/github.com/rgburke/goncurses/defs.go.orig
+++ cmd/grv/vendor/github.com/rgburke/goncurses/defs.go
@@ -4,7 +4,7 @@
 
 package goncurses
 
-// #cgo !darwin,!freebsd,!openbsd,!windows pkg-config: ncurses
+// #cgo !darwin,!freebsd,!openbsd,!windows pkg-config: ncursesw
 // #include <curses.h>
 import "C"
 
--- cmd/grv/vendor/github.com/rgburke/goncurses/form.go.orig
+++ cmd/grv/vendor/github.com/rgburke/goncurses/form.go
@@ -6,7 +6,7 @@
 
 package goncurses
 
-// #cgo !darwin,!freebsd,!openbsd pkg-config: form
+// #cgo !darwin,!freebsd,!openbsd pkg-config: formw
 // #cgo darwin freebsd openbsd LDFLAGS: -lform
 // #include <form.h>
 // #include <stdlib.h>
--- cmd/grv/vendor/github.com/rgburke/goncurses/menu.go.orig
+++ cmd/grv/vendor/github.com/rgburke/goncurses/menu.go
@@ -7,7 +7,7 @@
 package goncurses
 
 /*
-#cgo !darwin,!freebsd,!openbsd pkg-config: menu
+#cgo !darwin,!freebsd,!openbsd pkg-config: menuw
 #cgo darwin freebsd openbsd LDFLAGS: -lmenu
 #include <menu.h>
 #include <stdlib.h>
--- cmd/grv/vendor/github.com/rgburke/goncurses/mouse.go.orig
+++ cmd/grv/vendor/github.com/rgburke/goncurses/mouse.go
@@ -5,7 +5,7 @@
 
 package goncurses
 
-// #cgo !darwin,!freebsd,!openbsd,!windows pkg-config: ncurses
+// #cgo !darwin,!freebsd,!openbsd,!windows pkg-config: ncursesw
 // #cgo windows CFLAGS: -DNCURSES_MOUSE_VERSION
 // #cgo windows LDFLAGS: -lpdcurses
 // #include <curses.h>
--- cmd/grv/vendor/github.com/rgburke/goncurses/ncurses.go.orig
+++ cmd/grv/vendor/github.com/rgburke/goncurses/ncurses.go
@@ -5,7 +5,7 @@
 
 package goncurses
 
-// #cgo !darwin,!freebsd,!openbsd,!windows pkg-config: ncurses
+// #cgo !darwin,!freebsd,!openbsd,!windows pkg-config: ncursesw
 // #cgo windows CFLAGS: -DNCURSES_MOUSE_VERSION
 // #cgo windows LDFLAGS: -lpdcurses
 // #cgo darwin freebsd openbsd LDFLAGS: -lncurses
--- cmd/grv/vendor/github.com/rgburke/goncurses/panel.go.orig
+++ cmd/grv/vendor/github.com/rgburke/goncurses/panel.go
@@ -4,7 +4,7 @@
 
 package goncurses
 
-// #cgo !darwin,!freebsd,!openbsd,!windows pkg-config: panel
+// #cgo !darwin,!freebsd,!openbsd,!windows pkg-config: panelw
 // #cgo darwin freebsd openbsd LDFLAGS: -lpanel
 // #include <panel.h>
 // #include <curses.h>
