From https://github.com/ipython/ipykernel/pull/1390

From 991068ddc26cc0078ec6466897bb4f84869e7068 Mon Sep 17 00:00:00 2001
From: Ian Thomas <ianthomas23@gmail.com>
Date: Thu, 10 Apr 2025 13:36:43 +0100
Subject: [PATCH] Remove welcome_message arg from enable_pylab

---
 ipykernel/inprocess/ipkernel.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/ipykernel/inprocess/ipkernel.py b/ipykernel/inprocess/ipkernel.py
index 873b96d20..320b739b6 100644
--- a/ipykernel/inprocess/ipkernel.py
+++ b/ipykernel/inprocess/ipkernel.py
@@ -193,11 +193,11 @@ def enable_matplotlib(self, gui=None):
             gui = self.kernel.gui
         return super().enable_matplotlib(gui)
 
-    def enable_pylab(self, gui=None, import_all=True, welcome_message=False):
+    def enable_pylab(self, gui=None, import_all=True):
         """Activate pylab support at runtime."""
         if not gui:
             gui = self.kernel.gui
-        return super().enable_pylab(gui, import_all, welcome_message)
+        return super().enable_pylab(gui, import_all)
 
 
 InteractiveShellABC.register(InProcessInteractiveShell)
