From 448855acec6a6010c4bfe4b525a739b3968254f9 Mon Sep 17 00:00:00 2001
From: valoq <valoq@mailbox.org>
Date: Wed, 29 Oct 2025 21:59:49 +0100
Subject: [PATCH] add mmap2 for 32-bit arm

---
 zathura/seccomp-filters.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/zathura/seccomp-filters.c b/zathura/seccomp-filters.c
index 87649c4..f6a15b2 100644
--- a/zathura/seccomp-filters.c
+++ b/zathura/seccomp-filters.c
@@ -102,6 +102,9 @@ int seccomp_enable_strict_filter(zathura_t* zathura) {
   ALLOW_RULE(madvise);
   ALLOW_RULE(memfd_create);
   ALLOW_RULE(mmap);
+#ifdef __NR_mmap2
+  ALLOW_RULE(mmap2); /* Only required on 32-bit ARM systems */
+#endif
   ALLOW_RULE(mprotect);
   ALLOW_RULE(mremap); /* mupdf requirement */
   ALLOW_RULE(munmap);
-- 
2.51.2

