diff --git a/scripts/mk_util.py b/scripts/mk_util.py
index 014b0e40f..642a49a40 100644
--- a/scripts/mk_util.py
+++ b/scripts/mk_util.py
@@ -458,11 +458,11 @@ def check_ml():
     r = exec_cmd([OCAMLC, '-o', 'a.out', 'hello.ml'])
     if r != 0:
         raise MKException('Failed testing ocamlc compiler. Set environment variable OCAMLC with the path to the Ocaml compiler')
-    if is_verbose():
-        print ('Testing %s...' % OCAMLOPT)
-    r = exec_cmd([OCAMLOPT, '-o', 'a.out', 'hello.ml'])
-    if r != 0:
-        raise MKException('Failed testing ocamlopt compiler. Set environment variable OCAMLOPT with the path to the Ocaml native compiler. Note that ocamlopt may require flexlink to be in your path.')
+#    if is_verbose():
+#        print ('Testing %s...' % OCAMLOPT)
+#    r = exec_cmd([OCAMLOPT, '-o', 'a.out', 'hello.ml'])
+#    if r != 0:
+#        raise MKException('Failed testing ocamlopt compiler. Set environment variable OCAMLOPT with the path to the Ocaml native compiler. Note that ocamlopt may require flexlink to be in your path.')
     try:
         rmf('hello.cmi')
         rmf('hello.cmo')
@@ -2071,7 +2071,7 @@ class MLComponent(Component):
             out.write('\t%s -linkall -shared -o %s.cmxs -I . -I %s %s.cmxa\n' % (OCAMLOPTF, z3mls, self.sub_dir, z3mls))
 
             out.write('\n')
-            out.write('ml: %s.cma %s.cmxa %s.cmxs\n' % (z3mls, z3mls, z3mls))
+            out.write('ml: %s.cma\n' % (z3mls))
             if IS_OSX:
                 out.write('\tinstall_name_tool -id %s/libz3.dylib libz3.dylib\n' % (stubs_install_path))
                 out.write('\tinstall_name_tool -change libz3.dylib %s/libz3.dylib api/ml/dllz3ml.so\n' % (stubs_install_path))
@@ -2093,8 +2093,8 @@ class MLComponent(Component):
             out.write(get_component(Z3_DLL_COMPONENT).dll_name + '$(SO_EXT) ')
             out.write(os.path.join(self.sub_dir, 'META '))
             out.write(os.path.join(self.sub_dir, 'z3ml.cma '))
-            out.write(os.path.join(self.sub_dir, 'z3ml.cmxa '))
-            out.write(os.path.join(self.sub_dir, 'z3ml.cmxs '))
+#            out.write(os.path.join(self.sub_dir, 'z3ml.cmxa '))
+#            out.write(os.path.join(self.sub_dir, 'z3ml.cmxs '))
 
     def mk_install(self, out):
         if is_ml_enabled() and self._install_bindings():
@@ -2121,12 +2121,12 @@ class MLComponent(Component):
                 else:
                     out.write(' ' + os.path.join(self.sub_dir, m) + '.mli')
                 out.write(' ' + os.path.join(self.sub_dir, m) + '.cmi')
-                out.write(' ' + os.path.join(self.sub_dir, m) + '.cmx')
-            out.write(' %s' % ((os.path.join(self.sub_dir, 'libz3ml$(LIB_EXT)'))))
-            out.write(' %s' % ((os.path.join(self.sub_dir, 'z3ml$(LIB_EXT)'))))
+#                out.write(' ' + os.path.join(self.sub_dir, m) + '.cmx')
+#            out.write(' %s' % ((os.path.join(self.sub_dir, 'libz3ml$(LIB_EXT)'))))
+#            out.write(' %s' % ((os.path.join(self.sub_dir, 'z3ml$(LIB_EXT)'))))
             out.write(' %s' % ((os.path.join(self.sub_dir, 'z3ml.cma'))))
-            out.write(' %s' % ((os.path.join(self.sub_dir, 'z3ml.cmxa'))))
-            out.write(' %s' % ((os.path.join(self.sub_dir, 'z3ml.cmxs'))))
+#            out.write(' %s' % ((os.path.join(self.sub_dir, 'z3ml.cmxa'))))
+#            out.write(' %s' % ((os.path.join(self.sub_dir, 'z3ml.cmxs'))))
             out.write(' %s' % ((os.path.join(self.sub_dir, 'dllz3ml'))))
             if is_windows() or is_cygwin_mingw() or is_msys2():
                 out.write('.dll')
