--- src/libOpenImageIO/exif.cpp	2018-12-01 17:42:08.000000000 +0100
+++ src/libOpenImageIO/exif.cpp	2021-01-16 13:30:01.747255896 +0100
@@ -212,7 +212,7 @@
 };
 
 
-
+#if !defined(GPSTAG_VERSIONID)
 enum GPSTag {
     GPSTAG_VERSIONID = 0, 
     GPSTAG_LATITUDEREF = 1,  GPSTAG_LATITUDE = 2,
@@ -237,6 +237,11 @@
     GPSTAG_DIFFERENTIAL = 30,
     GPSTAG_HPOSITIONINGERROR = 31
 };
+#endif
+#if !defined(GPSTAG_HPOSITIONINGERROR)
+/* The tiff.h tag name differs from the one used here */
+#define GPSTAG_HPOSITIONINGERROR GPSTAG_GPSHPOSITIONINGERROR
+#endif
 
 static const EXIF_tag_info gps_tag_table[] = {
     { GPSTAG_VERSIONID,		"GPS:VersionID",	TIFF_BYTE, 4 }, 
