--- a/panels/power/cc-power-panel.c
+++ b/panels/power/cc-power-panel.c
@@ -934,10 +934,18 @@
                           -1);
       switch (value_tmp) {
       case CSD_POWER_ACTION_SUSPEND:
+#if UP_CHECK_VERSION(0,99,0)
+        enabled = cc_login1("CanSuspend");
+#else
         enabled = cc_login1("CanSuspend") || up_client_get_can_suspend (self->priv->up_client);
+#endif
         break;
       case CSD_POWER_ACTION_HIBERNATE:
+#if UP_CHECK_VERSION(0,99,0)
+        enabled = cc_login1("CanHibernate");
+#else
         enabled = cc_login1("CanHibernate") || up_client_get_can_hibernate (self->priv->up_client);
+#endif
         break;
       default:
         enabled = TRUE;
