--- rw-0.9/simplerw.c.newigraph 2017-02-14 00:20:35.000000000 +0900
+++ rw-0.9/simplerw.c   2022-09-11 19:39:47.033917305 +0900
@@ -134,7 +134,11 @@ int read_graph(const char *format, const
 		igraph_destroy(&igraph);
 		return(-1);
 	}
+#if (IGRAPH_VERSION_MAJOR >= 1) || ((IGRAPH_VERSION_MAJOR == 0) && (IGRAPH_VERSION_MINOR >= 10))
+	igraph_get_adjacency(&igraph, &imatrix, IGRAPH_GET_ADJACENCY_BOTH, NULL, IGRAPH_LOOPS_ONCE);
+#else
 	igraph_get_adjacency(&igraph, &imatrix, IGRAPH_GET_ADJACENCY_BOTH, 0);
+#endif
 	igraph_destroy(&igraph);
 	if(igraph_matrix_nrow(&imatrix) > MAX_VERTICES)
 	{
