*** a/CMakeLists.txt	2024-10-21 10:03:31.920419079 -0300
--- b/CMakeLists.txt	2024-10-21 10:13:21.248594699 -0300
*************** endif(EXISTS ${CMAKE_SOURCE_DIR}/.git)
*** 123,125 ****
  
! enable_testing()
  find_package(PkgConfig REQUIRED)
--- 123,125 ----
  
! 
  find_package(PkgConfig REQUIRED)
*************** add_subdirectory(po)
*** 141,143 ****
  add_subdirectory(src)
- add_subdirectory(tests)
  
--- 141,142 ----
