diff --git a/src/useradd.c b/src/useradd.c
index 677ea5a636f..49f55211a17 100644
--- a/src/useradd.c
+++ b/src/useradd.c
@@ -87,14 +87,14 @@ const char *Prog;
 /*
  * These defaults are used if there is no defaults file.
  */
-static gid_t def_group = 1000;
+static gid_t def_group = 100;
 static const char *def_groups = "";
 static const char *def_gname = "other";
 static const char *def_home = "/home";
 static const char *def_shell = "/bin/bash";
 static const char *def_template = SKEL_DIR;
 static const char *def_usrtemplate = USRSKELDIR;
-static const char *def_create_mail_spool = "yes";
+static const char *def_create_mail_spool = "no";
 static const char *def_log_init = "yes";

 static long def_inactive = -1;
