commit e9f88d39dad81b3dd25314e34fa0063aef0be59b
Author: Duncaen <mail@duncano.de>
Date:   Thu Aug 4 15:50:07 2016 +0200

    Fix mill_longjmp_ macro for other architectures than x86_64

diff --git libmill.h libmill.h
index b6d7f61..083985d 100644
--- libmill.h
+++ libmill.h
@@ -191,7 +191,7 @@ MILL_EXPORT void mill_setcls_(
 #else
 #define mill_setjmp_(ctx) \
     sigsetjmp(*ctx, 0)
-#define mill_longjmp_(ctx, 1) \
+#define mill_longjmp_(ctx) \
     siglongjmp(*ctx, 1)
 #endif
 
