This fixes build on 32-bit ppc with musl.

--- a/src/fault-linux-powerpc.h
+++ b/src/fault-linux-powerpc.h
@@ -29,9 +29,8 @@
 #if defined(__powerpc64__) || defined(_ARCH_PPC64) /* 64-bit */
 # define SIGSEGV_FAULT_STACKPOINTER  ((ucontext_t *) ucp)->uc_mcontext.gp_regs[1]
 #else /* 32-bit */
-/* both should be equivalent */
-# if 0
-#  define SIGSEGV_FAULT_STACKPOINTER  ((ucontext_t *) ucp)->uc_mcontext.regs->gpr[1]
+# if !defined(__GLIBC__)
+#  define SIGSEGV_FAULT_STACKPOINTER  ((ucontext_t *) ucp)->uc_regs->gregs[1]
 # else
 #  define SIGSEGV_FAULT_STACKPOINTER  ((ucontext_t *) ucp)->uc_mcontext.uc_regs->gregs[1]
 # endif
