diff --git parsers/guesser.go parsers/guesser.go
index 1f55738..eeac0fa 100644
--- parsers/guesser.go
+++ parsers/guesser.go
@@ -299,7 +299,7 @@ func processFile(directory string, file string, rootLicenses []LicenseMatch) Fil
 		Md5Hash:           getMd5Hash(content),
 		Sha1Hash:          getSha1Hash(content),
 		Sha256Hash:        getSha256Hash(content),
-		BytesHuman:        bytesToHuman(len(content)),
+		BytesHuman:        bytesToHuman(int64(len(content))),
 		Bytes:             len(content)}
 
 	return fileResult
diff --git parsers/helpers.go parsers/helpers.go
--- parsers/helpers.go
+++ parsers/helpers.go
@@ -80,9 +80,9 @@ func licenceListHasLicense(license LicenseMatch, licenseList []LicenseMatch) boo
 // Apache-2.0 is compatible with GPL-3.0-only
 // See https://apache.org/licenses/GPL-compatibility.html
 // SPDX-License-Identifier: Apache-2.0
-func bytesToHuman(bytes int) string {
+func bytesToHuman(bytes int64) string {
 
 	const TERABYTE = 1099511627776
 	const GIGABYTE = 1073741824
 	const MEGABYTE = 1048576
 	const KILOBYTE = 1024
