--- a/Clients/dns-sd.c	2021-01-26 20:47:47 UTC
+++ b/Clients/dns-sd.c
@@ -58,11 +58,13 @@
 //#define TEST_NEW_CLIENTSTUB 1
 
 #include <ctype.h>
+#include <stdarg.h>         // For va_list
 #include <stdio.h>          // For stdout, stderr
 #include <stdlib.h>         // For exit()
 #include <string.h>         // For strlen(), strcpy()
 #include <errno.h>          // For errno, EINTR
 #include <time.h>
+#include <sys/param.h>      // For MIN
 #include <sys/types.h>      // For u_char
 #ifdef APPLE_OSX_mDNSResponder
 #include <inttypes.h>       // For PRId64
