diff --git a/src/python_bindings/mod.rs b/src/python_bindings/mod.rs
index e69156a..2de0b3d 100644
--- a/src/python_bindings/mod.rs
+++ b/src/python_bindings/mod.rs
@@ -121,7 +121,7 @@ pub mod pyruntime {
         }
     }
 
-    #[cfg(all(target_os="linux", target_arch="x86_64"))]
+    #[cfg(all(target_os="linux", any(target_arch="x86_64", target_arch="riscv64")))]
     pub fn get_tstate_current_offset(version: &Version) -> Option<usize> {
         match version {
             Version{major: 3, minor: 7, patch: 0..=3, ..} => Some(1392),
