--- a/files/rkhunter.conf
+++ b/files/rkhunter.conf
@@ -74,7 +74,7 @@
 # value is ignored. If it is wished to keep the default value, as well as
 # any other set value, then the default must be explicitly set.
 #
-
+INSTALLDIR=/usr
 
 #
 # If this option is set to '1', it specifies that the mirrors file
@@ -157,7 +157,7 @@
 # subsequently commented out or removed, then the program will assume a
 # default directory beneath the installation directory.
 #
-#TMPDIR=/var/lib/rkhunter/tmp
+TMPDIR=/var/lib/rkhunter/tmp
 
 #
 # This option specifies the database directory to use.
@@ -166,7 +166,7 @@
 # subsequently commented out or removed, then the program will assume a
 # default directory beneath the installation directory.
 #
-#DBDIR=/var/lib/rkhunter/db
+DBDIR=/var/lib/rkhunter/db
 
 #
 # This option specifies the script directory to use.
@@ -174,7 +174,7 @@
 # The installer program will set the default directory. If this default is
 # subsequently commented out or removed, then the program will not run.
 #
-#SCRIPTDIR=/usr/local/lib/rkhunter/scripts
+SCRIPTDIR=/usr/share/rkhunter/scripts
 
 #
 # This option can be used to modify the command directory list used by rkhunter
