diff -ur python3-httpbin-0.10.1.orig/httpbin/filters.py python3-httpbin-0.10.1/httpbin/filters.py
--- python3-httpbin-0.10.1.orig/httpbin/filters.py	2023-10-17 08:12:56.814139485 -0400
+++ python3-httpbin-0.10.1/httpbin/filters.py	2023-10-17 08:14:20.979514500 -0400
@@ -10,7 +10,10 @@
 import gzip as gzip2
 import zlib
 
-import brotlicffi as _brotli
+try:
+    import brotlicffi as _brotli
+except ImportError:
+    import brotli as _brotli
 
 from six import BytesIO
 from decimal import Decimal
diff -ur python3-httpbin-0.10.1.orig/pyproject.toml python3-httpbin-0.10.1/pyproject.toml
--- python3-httpbin-0.10.1.orig/pyproject.toml	2023-10-17 08:12:56.816139494 -0400
+++ python3-httpbin-0.10.1/pyproject.toml	2023-10-17 08:13:46.165359377 -0400
@@ -32,7 +32,8 @@
 ]
 dependencies = [
     "Flask",
-    "brotlicffi",
+    "brotli; platform_python_implementation == 'CPython'",
+    "brotlicffi; platform_python_implementation != 'CPython'",
     "decorator",
     "flasgger",
     'greenlet < 3.0; python_version<"3.12"',
