--- a/retroshare-gui/src/gui/gxs/GxsCommentTreeWidget.cpp	2019-02-11 15:16:57.000000000 +0100
+++ b/retroshare-gui/src/gui/gxs/GxsCommentTreeWidget.cpp	2019-02-11 15:16:57.000000000 +0100
@@ -24,6 +24,7 @@
 #include <QDateTime>
 #include <QMenu>
 #include <QMimeData>
+#include <QPainterPath>
 #include <QPainter>
 #include <QTextDocument>
 
--- a/retroshare-gui/src/gui/gxs/GxsIdDetails.cpp	2019-02-11 15:16:57.000000000 +0100
+++ b/retroshare-gui/src/gui/gxs/GxsIdDetails.cpp	2019-02-11 15:16:57.000000000 +0100
@@ -30,6 +30,7 @@
 #include <retroshare/rspeers.h>
 
 #include <iostream>
+#include <QPainterPath>
 #include <QPainter>
 
 /* Images for tag icons */
