--- retroshare-gui/src/gui/gxs/GxsCommentTreeWidget.cpp	2019-02-11 15:16:57.000000000 +0100
+++ -	2020-07-05 00:47:16.416125978 +0200
@@ -24,6 +24,7 @@
 #include <QDateTime>
 #include <QMenu>
 #include <QMimeData>
+#include <QPainterPath>
 #include <QPainter>
 #include <QTextDocument>
 
--- retroshare-gui/src/gui/gxs/GxsIdDetails.cpp	2019-02-11 15:16:57.000000000 +0100
+++ -	2020-07-05 00:47:57.366966204 +0200
@@ -30,6 +30,7 @@
 #include <retroshare/rspeers.h>
 
 #include <iostream>
+#include <QPainterPath>
 #include <QPainter>
 
 /* Images for tag icons */
