diff -ru retroshare06-0.6.0.orig/src/libretroshare/src/libretroshare.pro retroshare06-0.6.0/src/libretroshare/src/libretroshare.pro
--- retroshare06-0.6.0.orig/src/libretroshare/src/libretroshare.pro	2015-12-29 12:44:59.441793222 +0100
+++ retroshare06-0.6.0/src/libretroshare/src/libretroshare.pro	2015-12-29 12:45:27.413886403 +0100
@@ -173,7 +173,7 @@
 		# Normal libupnp
 	} else {
 		# Patched libupnp or new unreleased version
-		DEFINES *= PATCHED_LIBUPNP
+		# DEFINES *= PATCHED_LIBUPNP
 	}
 
 	DEFINES *= UBUNTU
