--- a/scripts/jinja2/filters.py
+++ b/scripts/jinja2/filters.py
@@ -21,7 +21,7 @@ from jinja2.exceptions import FilterArgu
 from jinja2._compat import imap, string_types, text_type, iteritems
 
 
-_word_re = re.compile(r'\w+(?u)')
+_word_re = re.compile(r'\w+')
 
 
 def contextfilter(f):
--- a/scripts/jinja2/tests.py
+++ b/scripts/jinja2/tests.py
@@ -9,7 +9,7 @@
     :license: BSD, see LICENSE for more details.
 """
 import re
-from collections import Mapping
+from collections.abc import Mapping
 from jinja2.runtime import Undefined
 from jinja2._compat import text_type, string_types
 
--- a/scripts/markupsafe/__init__.py
+++ b/scripts/markupsafe/__init__.py
@@ -10,7 +10,7 @@
 """
 import re
 import string
-from collections import Mapping
+from collections.abc import Mapping
 from markupsafe._compat import text_type, string_types, int_types, \
      unichr, iteritems, PY2
 
