--- src/kpac/kpac_dhcp_helper.old	2016-04-03 21:55:42.000000000 +0100
+++ src/kpac/kpac_dhcp_helper.c	2016-07-21 08:42:04.989005565 +0100
@@ -17,6 +17,7 @@
    Boston, MA 02110-1301, USA.
 */
 
+#define _BSD_SOURCE
 #include <arpa/inet.h>
 #include <netinet/in.h>
 #include <sys/types.h>
--- ./src/ioslaves/file/kauth/fdsender.cpp	2018-02-18 09:28:09.195854002 +0100
+++ ./src/ioslaves/file/kauth/fdsender.cpp	2018-02-18 09:28:30.013527072 +0100
@@ -18,6 +18,7 @@
     License along with this library.  If not, see <http://www.gnu.org/licenses/>.
 ***/
 
+#include <string.h>
 #include "../sharefd_p.h"
 #include "fdsender.h"
