Index: nx-libs-3.5.99.26/nx-X11/config/cf/Imake.tmpl
===================================================================
--- nx-libs-3.5.99.26.orig/nx-X11/config/cf/Imake.tmpl
+++ nx-libs-3.5.99.26/nx-X11/config/cf/Imake.tmpl
@@ -1018,7 +1018,7 @@ TCLIBDIR = TclLibDir
 #if HasLargeTmp || SystemV4
 #define ArCmd ArCmdBase cq
 #else
-#define ArCmd ArCmdBase clq
+#define ArCmd ArCmdBase cq
 #endif
 #endif
 #ifndef ArAddCmd
