--- nx-X11/programs/Xserver/Xext/xf86bigfont.c	2021-02-04 14:34:56.000000000 +0100
+++ -	2021-05-10 08:55:11.108944904 +0200
@@ -45,7 +45,6 @@
 /* Linux libc4 and libc5 only (because glibc doesn't include kernel headers):
    Linux 2.0.x and 2.2.x define SHMLBA as PAGE_SIZE, but forget to define
    PAGE_SIZE. It is defined in <asm/page.h>. */
-#include <asm/page.h>
 #include <limits.h>
 #endif
 #ifdef SVR4
