--- orig/makefile.in	2010-12-22 12:56:31.610881756 +0100
+++ new/makefile.in	2010-12-22 12:57:12.344232346 +0100
@@ -89,10 +89,10 @@ EXE=
 
 #ifdef GNUMAKE
 all: check_config bcc86 cpp unproto copt as86 ar86 ld86 objdump86 \
-     library lib-bsd alt-libs elksemu
+     library lib-bsd alt-libs
 
 install: check_config install-bcc install-man \
-     install-lib install-emu
+     install-lib
 
 install-all: install install-other
 
@@ -123,10 +123,6 @@ INCLDIR= %INCLDIR%
 ASLDDIR= %ASLDDIR%
 MANDIR=  %MANDIR%
 
-#ifdef GNUMAKE
-export ELKSSRC
-#endif
-
 DISTBIN= $(DIST)$(BINDIR)
 DISTLIB= $(DIST)$(LIBDIR)
 DISTASLD=$(DIST)$(ASLDDIR)
