diff -up dev86-0.16.17/ld/x86_aout.h.long dev86-0.16.17/ld/x86_aout.h
--- dev86-0.16.17/ld/x86_aout.h.long	2003-01-28 23:17:14.000000000 +0100
+++ dev86-0.16.17/ld/x86_aout.h	2009-02-19 11:37:10.000000000 +0100
@@ -11,10 +11,11 @@
 /* If the host isn't an x86 all bets are off, use chars. */
 #if defined(i386) || defined(__BCC__) || defined(MSDOS)
 typedef long Long;
-#define __OUT_OK 1
 #else
-typedef char Long[4];
+#include <stdint.h>
+typedef int32_t Long;
 #endif
+#define __OUT_OK 1
 
 struct	exec {			/* a.out header */
   unsigned char	a_magic[2];	/* magic number */
