diff -up dev86-0.16.17/bcc/bcc.c.overflow dev86-0.16.17/bcc/bcc.c
--- dev86-0.16.17/bcc/bcc.c.overflow	2005-01-03 23:41:55.000000000 +0100
+++ dev86-0.16.17/bcc/bcc.c	2009-02-19 10:49:32.000000000 +0100
@@ -16,6 +16,7 @@
  *	-M0	A framework for the -B option.
  */
 #include <stdio.h>
+#include <limits.h>
 #ifdef __STDC__
 #include <stdlib.h>
 #ifndef MSDOS
@@ -1308,11 +1309,7 @@ void reset_prefix_path()
 
       for(d=s=ptr; d && *s; s=d)
       {
-#ifdef MAXPATHLEN
-         char buf[MAXPATHLEN];
-#else
-         char buf[1024];
-#endif
+         char buf[PATH_MAX];
 
 	 free(temp);
          d=strchr(s, ':');
