 - this ignores a warning caused by old notebook

--- a/src/sage/all.py
+++ b/src/sage/all.py
@@ -100,6 +100,9 @@ warnings.filterwarnings('ignore', category=DeprecationWarning,
                         message='The distutils(.sysconfig module| package) is deprecated',
                         module='Cython|distutils|numpy|sage.env|sage.features')
 
+warnings.filterwarnings('ignore', category=DeprecationWarning,
+                        message=r"invalid escape sequence '\\s'")
+
 ################ end setup warnings ###############################
 
 

