diff --git a/src/sage/schemes/toric/sheaf/klyachko.py b/src/sage/schemes/toric/sheaf/klyachko.py
index b1304a16913..77fae4b7545 100644
--- a/src/sage/schemes/toric/sheaf/klyachko.py
+++ b/src/sage/schemes/toric/sheaf/klyachko.py
@@ -26,7 +26,7 @@
     (0, 0, 18, 16, 1)
     sage: Gtilde = G_sum.random_deformation()
     sage: V = Gtilde.wedge(2) * K                     # long time
-    sage: V.cohomology(dim=True, weight=(0,0,0,0))    # long time
+    sage: V.cohomology(dim=True, weight=(0,0,0,0))    # long time  # random failure (see #32773)
     (0, 0, 3, 0, 0)
 
 REFERENCES:
@@ -948,7 +948,7 @@ def random_deformation(self, epsilon=None):
            sage: V.cohomology(dim=True, weight=(0,))
            (1, 0)
            sage: Vtilde = V.random_deformation()
-           sage: Vtilde.cohomology(dim=True, weight=(0,))
+           sage: Vtilde.cohomology(dim=True, weight=(0,))  # random failure (see #32773)
            (1, 0)
         """
         filt = self._filt.random_deformation(epsilon)
