diff --git a/src/sage/graphs/generators/families.py b/src/sage/graphs/generators/families.py
index ec6a5c19e60..9dd7bc1cb1e 100644
--- a/src/sage/graphs/generators/families.py
+++ b/src/sage/graphs/generators/families.py
@@ -3660,7 +3660,7 @@ def nauty_gentreeg(options="", debug=False):
 
         sage: gen = graphs.nauty_gentreeg("4", debug=True)
         sage: print(next(gen))
-        >A ...gentreeg Z=2:3 D=3 n=4
+        >A ...gentreeg ...
         sage: gen = graphs.nauty_gentreeg("4 -q", debug=True)
         sage: next(gen)
         ''
@@ -3687,7 +3687,7 @@ def nauty_gentreeg(options="", debug=False):
         sage: list(graphs.nauty_gentreeg("3 -x", debug=True))
         ['>E Usage: ...gentreeg [-D#] [-Z#:#] [-ulps] [-q] n [res/mod] ...
         sage: list(graphs.nauty_gentreeg("3", debug=True))
-        ['>A ...gentreeg Z=2:2 D=2 n=3\n', Graph on 3 vertices]
+        ['>A ...gentreeg ...\n', Graph on 3 vertices]
     """
     import shlex
     from sage.features.nauty import NautyExecutable
diff --git a/src/sage/graphs/graph_generators.py b/src/sage/graphs/graph_generators.py
index df88bbe2713..1a8016976c2 100644
--- a/src/sage/graphs/graph_generators.py
+++ b/src/sage/graphs/graph_generators.py
@@ -966,7 +966,7 @@ def nauty_geng(self, options="", debug=False):
             ...
             ValueError: wrong format of parameter option
             sage: list(graphs.nauty_geng("-c3", debug=True))
-            ['>E Usage: ...geng [-cCmtfbd#D#] [-uygsnh] [-lvq] ...
+            ['>E Usage: ...geng ...\n']
             sage: list(graphs.nauty_geng("-c 3", debug=True))
             ['>A ...geng -cd1D2 n=3 e=2-3\n', Graph on 3 vertices, Graph on 3 vertices]
         """
