diff --git a/build/pkgs/sympy/requirements.txt b/build/pkgs/sympy/requirements.txt
new file mode 100644
index 00000000000..3ef8f90b550
diff --git a/src/sage/doctest/forker.py b/src/sage/doctest/forker.py
index 7e9eab2ff2e..43365cd645e 100644
--- a/src/sage/doctest/forker.py
+++ b/src/sage/doctest/forker.py
@@ -150,14 +150,15 @@ def init_sage(controller=None):
         sage: from sympy.printing.pretty.pretty import PrettyPrinter
         sage: s = sympify('+x^'.join(str(i) for i in range(30)))
         sage: print(PrettyPrinter(settings={'wrap_line': True}).doprint(s))
-         29    28    27    26    25    24    23    22    21    20    19    18    17
-        x   + x   + x   + x   + x   + x   + x   + x   + x   + x   + x   + x   + x   +
+         29    28    27    26    25    24    23    22    21    20    19    18    17    ↪
+        x   + x   + x   + x   + x   + x   + x   + x   + x   + x   + x   + x   + x   +  ↪
         <BLANKLINE>
-         16    15    14    13    12    11    10    9    8    7    6    5    4    3
-        x   + x   + x   + x   + x   + x   + x   + x  + x  + x  + x  + x  + x  + x  + x
+        ↪  16    15    14    13    12    11    10    9    8    7    6    5    4    3   ↪
+        ↪ x   + x   + x   + x   + x   + x   + x   + x  + x  + x  + x  + x  + x  + x  + ↪
         <BLANKLINE>
-        2
-          + x
+        ↪   2
+        ↪  x  + x
+
 
     The displayhook sorts dictionary keys to simplify doctesting of
     dictionary output::
diff --git a/src/sage/functions/hypergeometric.py b/src/sage/functions/hypergeometric.py
index 010c61febe0..bbccbd20032 100644
--- a/src/sage/functions/hypergeometric.py
+++ b/src/sage/functions/hypergeometric.py
@@ -124,7 +124,7 @@
     sage: maxima(hypergeometric([1, 1, 1], [3, 3, 3], x))                               # needs sage.symbolic
     hypergeometric([1,1,1],[3,3,3],_SAGE_VAR_x)
     sage: hypergeometric((5, 4), (4, 4), 3)._sympy_()                                   # needs sympy sage.symbolic
-    hyper((5, 4), (4, 4), 3)
+    hyper((5,), (4,), 3)
     sage: hypergeometric((5, 4), (4, 4), 3)._mathematica_init_()                        # needs sage.symbolic
     'HypergeometricPFQ[{5,4},{4,4},3]'
 
diff --git a/src/sage/symbolic/expression.pyx b/src/sage/symbolic/expression.pyx
index a92fe241355..f62311423be 100644
--- a/src/sage/symbolic/expression.pyx
+++ b/src/sage/symbolic/expression.pyx
@@ -1167,7 +1167,7 @@ cdef class Expression(Expression_abc):
             sage: unicode_art(SR(13 - I))
             13 - ⅈ
             sage: unicode_art(SR(1.3 - I))
-            1.3 - ⅈ
+            1.3 - 1.0⋅ⅈ
             sage: unicode_art(cos(I))
             cosh(1)
 
diff --git a/src/sage/typeset/ascii_art.py b/src/sage/typeset/ascii_art.py
index 28024405d27..458fa8724a3 100644
--- a/src/sage/typeset/ascii_art.py
+++ b/src/sage/typeset/ascii_art.py
@@ -44,11 +44,12 @@
     sage: shell.run_cell('%display ascii_art')
     sage: shell.run_cell("i = var('i')")                                                # needs sage.symbolic
     sage: shell.run_cell('sum(factorial(i)*x^i, i, 0, 10)')                             # needs sage.symbolic
-             10           9          8         7        6        5       4      3
-    3628800*x   + 362880*x  + 40320*x  + 5040*x  + 720*x  + 120*x  + 24*x  + 6*x
+             10           9          8         7        6        5       4      3  >
+    3628800*x   + 362880*x  + 40320*x  + 5040*x  + 720*x  + 120*x  + 24*x  + 6*x   >
     <BLANKLINE>
-         2
-    + 2*x  + x + 1
+    >      2
+    > + 2*x  + x + 1
+
     sage: shell.run_cell('3/(7*x)')                                                     # needs sage.symbolic
      3
     ---
