diff --git a/src/sage/interfaces/gap3.py b/src/sage/interfaces/gap3.py
index e73308e014e..de2a7fe4289 100644
--- a/src/sage/interfaces/gap3.py
+++ b/src/sage/interfaces/gap3.py
@@ -752,7 +752,8 @@ def _latex_(self):
         r"""
         EXAMPLES::
 
-            sage: s = gap("[[1,2], [3/4, 5/6]]")
+            sage: # optional - gap3
+            sage: s = gap3("[[1,2], [3/4, 5/6]]")
             sage: s._latex_()
             '\\left(\\begin{array}{rr} 1&2\\\\ 3/4&\\frac{5}{6}\\\\ \\end{array}\\right)'
             sage: latex(s)
