diff --git a/src/sage_setup/command/sage_build_cython.py b/src/sage_setup/command/sage_build_cython.py
index f4b1357c543..3435c196d49 100644
--- a/src/sage_setup/command/sage_build_cython.py
+++ b/src/sage_setup/command/sage_build_cython.py
@@ -226,6 +226,7 @@ def run(self):
                 compiler_directives=self.cython_directives,
                 compile_time_env=self.compile_time_env,
                 create_extension=self.create_extension,
+                show_all_warnings=True,
                 # Debugging
                 gdb_debug=self.debug,
                 output_dir=os.path.join(self.build_lib, "sage"),
