diff --git a/src/pyproject.toml b/src/pyproject.toml
index 781af00a8d7..f70304534d9 100644
--- a/src/pyproject.toml
+++ b/src/pyproject.toml
@@ -11,12 +11,9 @@ requires = [
     # Exclude 3.0.3 because of https://github.com/cython/cython/issues/5748
     'cython >=3.0, != 3.0.3, <4.0',
     'gmpy2 ~=2.1.b999',
-    'jupyter_core >=4.6.3',
     'memory_allocator',
     'numpy >=1.19',
     'pkgconfig',
-    # pplpy 0.8.4 and earlier do not declare dependencies correctly (see https://github.com/sagemath/sage/issues/30922)
-    'pplpy >=0.8.6',
 ]
 build-backend = "setuptools.build_meta"
 
