diff --git a/src/sage/graphs/generators/families.py b/src/sage/graphs/generators/families.py
index e6e5312b8a8..aa10d15d4d8 100644
--- a/src/sage/graphs/generators/families.py
+++ b/src/sage/graphs/generators/families.py
@@ -3710,7 +3710,7 @@ def nauty_gentreeg(options='', debug=False):
         ...
         ValueError: wrong format of parameter options
         sage: list(graphs.nauty_gentreeg("3 -x", debug=True))
-        ['>E Usage: ...gentreeg [-D#] [-Z#:#] [-ulps] [-q] n [res/mod] ...
+        ['>E Usage: ...gentreeg [-D#] [-Z#:#] [-ulps] [-q] n... [res/mod] ...
         sage: list(graphs.nauty_gentreeg("3", debug=True))
         ['>A ...gentreeg ...\n', Graph on 3 vertices]
     """
