diff --git a/src/sage/matroids/extension.pyx b/src/sage/matroids/extension.pyx
index 778e6d6ef70..83b792e8d2b 100644
--- a/src/sage/matroids/extension.pyx
+++ b/src/sage/matroids/extension.pyx
@@ -211,6 +211,9 @@ cdef class LinearSubclassesIter:
 
         self._nodes = [first_cut]
 
+    def __iter__(self):
+        return self
+
     def __next__(self):
         """
         Return the next linear subclass.
diff --git a/src/sage/matroids/set_system.pyx b/src/sage/matroids/set_system.pyx
index d9a2f631fbf..0bf9cb2b7fa 100644
--- a/src/sage/matroids/set_system.pyx
+++ b/src/sage/matroids/set_system.pyx
@@ -772,6 +772,9 @@ cdef class SetSystemIterator:
         self._pointer = -1
         self._len = len(H)
 
+    def __iter__(self):
+        return self
+
     def __next__(self):
         """
         Return the next subset of a SetSystem.
