diff --git a/src/sage/rings/lazy_species.py b/src/sage/rings/lazy_species.py
index d8d1da3e09f..99338c29517 100644
--- a/src/sage/rings/lazy_species.py
+++ b/src/sage/rings/lazy_species.py
@@ -644,8 +644,10 @@ def __call__(self, *args):
             sage: E = L.Sets()
             sage: A = L.undefined(1)
             sage: A.define(X*E(A))
-            sage: A[5]
+            sage: A[5]  # random
             X*E_4 + X^2*E_3 + 3*X^3*E_2 + X*E_2(X^2) + 3*X^5
+            sage: A[5] == X*E[4] + X^2*E[3] + 3*X^3*E[2] + X*E[2](X[1]^2) + 3*X^5
+            True
 
             sage: C = L.Cycles()
             sage: F = E(C(A))
