diff --git a/src/sage/calculus/desolvers.py b/src/sage/calculus/desolvers.py
index 7f6ba7967e1..89a56cb54e1 100644
--- a/src/sage/calculus/desolvers.py
+++ b/src/sage/calculus/desolvers.py
@@ -1621,7 +1621,7 @@ def desolve_odeint_inner(ivar):
             assert len(des) == 1
             dvar = dvars[0]
             de = des[0]
-            func = fast_float(de, dvar, ivar)
+            func = lambda y, t: fast_float(de, dvar, ivar)(y.item(), t)
             if not compute_jac:
                 Dfun = None
             else:
