diff --git a/src/sage/topology/simplicial_set_examples.py b/src/sage/topology/simplicial_set_examples.py
index be7a4084be2..cd69adb9c5a 100644
--- a/src/sage/topology/simplicial_set_examples.py
+++ b/src/sage/topology/simplicial_set_examples.py
@@ -655,7 +655,7 @@ def simplicial_data_from_kenzo_output(filename) -> dict:
         sage: S4.homology(reduced=False)                                                # needs pyparsing
         {0: Z, 1: 0, 2: 0, 3: 0, 4: Z}
     """
-    from pyparsing import OneOrMore, nestedExpr
+    from pyparsing import OneOrMore, nested_expr
 
     with open(filename) as f:
         data = f.read()
@@ -675,7 +675,7 @@ def simplicial_data_from_kenzo_output(filename) -> dict:
             end = new_dim_idx
         if dim == 0:
             simplex_string = data[data.find('Vertices :') + len('Vertices :'):end]
-            vertices = OneOrMore(nestedExpr()).parseString(simplex_string).asList()[0]
+            vertices = OneOrMore(nested_expr()).parse_string(simplex_string).asList()[0]
             for v in vertices:
                 vertex = AbstractSimplex(0, name=v)
                 simplex_data[vertex] = None
