diff --git a/src/sage/doctest/test.py b/src/sage/doctest/test.py
index 8e7ecac8bbe..d68cc67c153 100644
--- a/src/sage/doctest/test.py
+++ b/src/sage/doctest/test.py
@@ -257,8 +257,7 @@
     sage: subprocess.call(["python3", "-m", "sage.doctest", "--warn-long", "0",    # long time
     ....:      "--random-seed=0", "--optional=sage", "interrupt.rst"], **kwds)
     Running doctests...
-    Doctesting 1 file.
-    Killing test interrupt.rst
+    Doctesting 1 file...
     ----------------------------------------------------------------------
     Doctests interrupted: 0/1 files tested
     ----------------------------------------------------------------------
