diff --git a/src/sage/matrix/matrix2.pyx b/src/sage/matrix/matrix2.pyx
index 7ff5a3a95bc..bdbb6805585 100644
--- a/src/sage/matrix/matrix2.pyx
+++ b/src/sage/matrix/matrix2.pyx
@@ -367,7 +367,7 @@ cdef class Matrix(Matrix1):
             sage: A.solve_left(b)                                                       # needs scipy
             Traceback (most recent call last):
             ...
-            LinAlgError: Matrix is singular.
+            LinAlgError: ...singular.
 
         The vector of constants needs the correct degree::
 
@@ -762,7 +762,7 @@ cdef class Matrix(Matrix1):
             sage: A.solve_right(b)
             Traceback (most recent call last):
             ...
-            LinAlgError: Matrix is singular.
+            LinAlgError: ...singular.
 
         The vector of constants needs the correct degree.  ::
 
