--- a/src/sage/geometry/polyhedral_complex.py
+++ b/src/sage/geometry/polyhedral_complex.py
@@ -932,8 +932,8 @@ class PolyhedralComplex(GenericCellComplex):
             sage: pc = PolyhedralComplex([
             ....:         Polyhedron(vertices=[(1/3, 1/3), (0, 0), (1, 2)]),
             ....:         Polyhedron(vertices=[(1, 2), (0, 0), (0, 1/2)])])
-            sage: pc._an_element_().vertices_list()
-            [[0, 0], [0, 1/2], [1, 2]]
+            sage: pc._an_element_()
+            A 2-dimensional polyhedron in QQ^2 defined as the convex hull of 3 vertices
         """
         try:
             return next(self.maximal_cell_iterator(increasing=False))
