commit 4a38d536620acd69783597f47568462e3db48c00 (HEAD -> pari-2.17)
Author: Gonzalo Tornaría <tornaria@cmat.edu.uy>
Date:   Thu Nov 7 11:01:11 2024 -0300

    make pari_bnf() deterministic

diff --git a/src/sage/rings/number_field/number_field.py b/src/sage/rings/number_field/number_field.py
index 6d43bc12d2d..f89d829c76c 100644
--- a/src/sage/rings/number_field/number_field.py
+++ b/src/sage/rings/number_field/number_field.py
@@ -4440,10 +4440,13 @@ class NumberField_generic(WithEqualityById, number_field_base.NumberField):
             bnf = self._pari_bnf
         except AttributeError:
             f = self.pari_polynomial("y")
+            # make this deterministic, it affects printing of ideals
+            _rand = pari.getrand() ; pari.setrand(1)
             if units:
                 self._pari_bnf = f.bnfinit(1)
             else:
                 self._pari_bnf = f.bnfinit()
+            pari.setrand(_rand)
             bnf = self._pari_bnf
         # Certify if needed
         if proof and not getattr(self, "_pari_bnf_certified", False):
