From c44f16c188176e1d66187884c016a61bd0fabf4d Mon Sep 17 00:00:00 2001
From: Diego Herrera <dherrera@redhat.com>
Date: Wed, 24 Jun 2026 19:30:11 -0400
Subject: [PATCH 10/11] openVINO runtime fix

---
 cmake/onnxruntime_providers_openvino.cmake | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/cmake/onnxruntime_providers_openvino.cmake b/cmake/onnxruntime_providers_openvino.cmake
index 5b5d17df8a..ce87eee040 100644
--- a/cmake/onnxruntime_providers_openvino.cmake
+++ b/cmake/onnxruntime_providers_openvino.cmake
@@ -21,14 +21,6 @@
     add_definitions(-DUSE_OVEP_NPU_MEMORY=1)
   endif()
 
-  # If building RelWithDebInfo and OV package does not have that configuration map to Release
-  get_target_property(ov_rt_implib_rwdi openvino::runtime IMPORTED_IMPLIB_RELWITHDEBINFO)
-  if ((CMAKE_BUILD_TYPE STREQUAL RelWithDebInfo) AND NOT ov_rt_implib_rwdi)
-    set_target_properties(openvino::runtime PROPERTIES
-      MAP_IMPORTED_CONFIG_RELWITHDEBINFO Release
-    )
-  endif()
-
   list(APPEND OPENVINO_LIB_LIST openvino::frontend::onnx openvino::runtime ${PYTHON_LIBRARIES})
   source_group(TREE ${ONNXRUNTIME_ROOT}/core FILES ${onnxruntime_providers_openvino_cc_srcs})
   onnxruntime_add_shared_library_module(onnxruntime_providers_openvino ${onnxruntime_providers_openvino_cc_srcs} "${ONNXRUNTIME_ROOT}/core/dll/onnxruntime.rc")
-- 
2.55.0

