--- a/cmake/external/onnxruntime_external_deps.cmake
+++ b/cmake/external/onnxruntime_external_deps.cmake
@@ -284,16 +284,22 @@
      message(STATUS "Aliasing Boost::headers to Boost::mp11")
      add_library(Boost::mp11 ALIAS Boost::headers)
   else()
-    onnxruntime_fetchcontent_declare(
-     mp11
-     URL ${DEP_URL_mp11}
-     EXCLUDE_FROM_ALL
-     FIND_PACKAGE_ARGS NAMES Boost
-    )
-    FetchContent_Populate(mp11)
-    if(NOT TARGET Boost::mp11)
-      add_library(Boost::mp11 IMPORTED INTERFACE)
-      target_include_directories(Boost::mp11 INTERFACE $<BUILD_INTERFACE:${mp11_SOURCE_DIR}/include>)
+    find_package(Boost QUIET COMPONENTS headers)
+    if(TARGET Boost::headers)
+      message(STATUS "Aliasing Boost::headers to Boost::mp11")
+      add_library(Boost::mp11 ALIAS Boost::headers)
+    else()
+      onnxruntime_fetchcontent_declare(
+       mp11
+       URL ${DEP_URL_mp11}
+       EXCLUDE_FROM_ALL
+       FIND_PACKAGE_ARGS NAMES Boost
+      )
+      FetchContent_Populate(mp11)
+      if(NOT TARGET Boost::mp11)
+        add_library(Boost::mp11 IMPORTED INTERFACE)
+        target_include_directories(Boost::mp11 INTERFACE $<BUILD_INTERFACE:${mp11_SOURCE_DIR}/include>)
+      endif()
     endif()
   endif()
 endif()
