diff --git a/setup.py b/setup.py
index 294b975a56..1612a4419c 100644
--- a/setup.py
+++ b/setup.py
@@ -523,7 +523,7 @@ classifiers = [
 if not enable_training:
     classifiers.extend(["Operating System :: Microsoft :: Windows", "Operating System :: MacOS"])
 
-if enable_training or enable_training_apis:
+if True:
     packages.append("onnxruntime.training")
     if enable_training:
         packages.extend(
@@ -565,6 +565,7 @@ if enable_training or enable_training_apis:
         ]
     )
 
+if enable_training:
     requirements_file = "requirements-training.txt"
     # with training, we want to follow this naming convention:
     # stable:
