--- a/configure	2026-08-17 22:22:13.630438436 +0100
+++ b/configure	2026-08-17 22:51:54.251923145 +0100
@@ -15753,17 +15753,18 @@
 
 GETTEXT_LIBDIR=${GETTEXT_PREFIX}/lib
 
-if test -x ${GETTEXT_PREFIX}/bin/msgfmt ; then
-    GETTEXT_BINDIR=${GETTEXT_PREFIX}/bin
-elif test -x ${GETTEXT_PREFIX}/local/bin/msgfmt ; then
-    GETTEXT_BINDIR=${GETTEXT_PREFIX}/local/bin
-else
-    { { printf '%s\n' "$as_me:${as_lineno-$LINENO}: error: in '$ac_pwd':" >&5
+if test "$GETTEXT_ENABLE" != "no"; then
+    if test -x ${GETTEXT_PREFIX}/bin/msgfmt ; then
+        GETTEXT_BINDIR=${GETTEXT_PREFIX}/bin
+    elif test -x ${GETTEXT_PREFIX}/local/bin/msgfmt ; then
+        GETTEXT_BINDIR=${GETTEXT_PREFIX}/local/bin
+    else
+        { { printf '%s\n' "$as_me:${as_lineno-$LINENO}: error: in '$ac_pwd':" >&5
 printf '%s\n' "$as_me: error: in '$ac_pwd':" >&2;}
 as_fn_error $? "\"could not find msgfmt tool\"
 See 'config.log' for more details" "$LINENO" 5; }
-    # Use a (bad) fall back value
-    GETTEXT_BINDIR=${GETTEXT_PREFIX}/bin
+        GETTEXT_BINDIR=${GETTEXT_PREFIX}/bin
+    fi
 fi
 
 
