From 1d29a21ef68d170a3fee6a58eeb4ff71399fff97 Mon Sep 17 00:00:00 2001
From: Nikias Bassen <nikias@gmx.li>
Date: Tue, 27 Oct 2020 18:02:52 +0100
Subject: [PATCH] restore: Add new baseband firmware file mapping for Mav20
 basebands (iPhone 12)

---
 src/restore.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/restore.c b/src/restore.c
index e35047a..4f167ea 100644
--- src/restore.c
+++ src/restore.c
@@ -1282,6 +1282,8 @@ static const char* restore_get_bbfw_fn_for_element(const char* elem)
 		// ICE19 firmware files
 		{ "RestorePSI2", "restorepsi2.bin" },
 		{ "PSI2", "psi_ram2.bin" },
+		// Mav20 Firmware file
+		{ "Misc", "multi_image.mbn" },
 		{ NULL, NULL }
 	};
 
