From 33285c63f585e42b32f450e4a1ab926eb9b964d2 Mon Sep 17 00:00:00 2001
From: Nikias Bassen <nikias@gmx.li>
Date: Tue, 27 Oct 2020 23:28:30 +0100
Subject: [PATCH] restore: Add SE,ChipID 0xD2 for iPhone 12 SE firmware

---
 src/restore.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/restore.c b/src/restore.c
index 841990d..adea97a 100644
--- src/restore.c
+++ src/restore.c
@@ -1985,7 +1985,7 @@ static plist_t restore_get_se_firmware_data(restored_client_t restore, struct id
 	}
 	if (chip_id == 0x20211) {
 		comp_name = "SE,Firmware";
-	} else if (chip_id == 0x73 || chip_id == 0x64 || chip_id == 0xC8) {
+	} else if (chip_id == 0x73 || chip_id == 0x64 || chip_id == 0xC8 || chip_id == 0xD2) {
 		comp_name = "SE,UpdatePayload";
 	} else {
 		info("WARNING: Unknown SE,ChipID 0x%" PRIx64 " detected. Restore might fail.\n", (uint64_t)chip_id);
