--- a/pyproject.toml
+++ b/pyproject.toml
@@ -33,21 +33,5 @@
 icecream = "^2.1.2"
 mpi4py = "^3.1.3"
 
-[tool.poetry-dynamic-versioning]
-enable = true
-vcs = "git"
-pattern  = "^(?P<base>\\d+\\.\\d+\\.\\d+)(-?((?P<stage>[a-zA-Z]+)\\.?(?P<revision>\\d+)?))?"
-format-jinja = """
-    {%- if distance == 0 -%}
-        {{- base -}}
-    {%- else -%}
-        {{- base }}.dev{{ distance }}+{{commit}}
-    {%- endif -%}
-"""
-format-jinja-imports = [
-    { module = "datetime", item = "datetime" }
-]
-
 [build-system]
-requires = ["poetry-core>=1.0.0", "poetry-dynamic-versioning"]
-build-backend = "poetry_dynamic_versioning.backend"
+requires = ["poetry-core>=1.0.0"]
