Void Linux gmime3 always support session key extraction.
--- a/configure
+++ b/configure
@@ -506,65 +506,7 @@ if pkg-config --exists "gmime-3.0 >= $GM
 
     printf "Checking for GMime session key extraction support... "
 
-    cat > _check_session_keys.c <<EOF
-#include <gmime/gmime.h>
-#include <stdio.h>
-
-int main () {
-    GError *error = NULL;
-    GMimeParser *parser = NULL;
-    GMimeMultipartEncrypted *body = NULL;
-    GMimeDecryptResult *decrypt_result = NULL;
-    GMimeObject *output = NULL;
-
-    g_mime_init ();
-    parser = g_mime_parser_new ();
-    g_mime_parser_init_with_stream (parser, g_mime_stream_file_open("$srcdir/test/corpora/crypto/basic-encrypted.eml", "r", &error));
-    if (error) return !! fprintf (stderr, "failed to instantiate parser with test/corpora/crypto/basic-encrypted.eml\n");
-
-    body = GMIME_MULTIPART_ENCRYPTED(g_mime_message_get_mime_part (g_mime_parser_construct_message (parser, NULL)));
-    if (body == NULL) return !! fprintf (stderr, "did not find a multipart encrypted message\n");
-
-    output = g_mime_multipart_encrypted_decrypt (body, GMIME_DECRYPT_EXPORT_SESSION_KEY, NULL, &decrypt_result, &error);
-    if (error || output == NULL) return !! fprintf (stderr, "decryption failed\n");
-
-    if (decrypt_result == NULL) return !! fprintf (stderr, "no GMimeDecryptResult found\n");
-    if (decrypt_result->session_key == NULL) return !! fprintf (stderr, "GMimeDecryptResult has no session key\n");
-
-    printf ("%s\n", decrypt_result->session_key);
-    return 0;
-}
-EOF
-    if ! TEMP_GPG=$(mktemp -d "${TMPDIR:-/tmp}/notmuch.XXXXXX"); then
-	printf 'No.\nCould not make tempdir for testing session-key support.\n'
-	errors=$((errors + 1))
-    elif ${CC} ${CFLAGS} ${gmime_cflags} _check_session_keys.c ${gmime_ldflags} -o _check_session_keys \
-	   && GNUPGHOME=${TEMP_GPG} gpg --batch --quiet --import < "$srcdir"/test/openpgp4-secret-key.asc \
-	   && SESSION_KEY=$(GNUPGHOME=${TEMP_GPG} ./_check_session_keys) \
-	   && [ $SESSION_KEY = 9:496A0B6D15A5E7BA762FB8E5FE6DEE421D4D9BBFCEAD1CDD0CCF636D07ADE621 ]
-    then
 	printf "OK.\n"
-    else
-	cat <<EOF
-No.
-*** Error: Could not extract session keys from encrypted message.
-
-This is likely due to your GMime having been built against a old
-version of GPGME.
-
-Please try to rebuild your version of GMime against a more recent
-version of GPGME (at least GPGME 1.8.0).
-EOF
-	if GPGME_VERS="$(pkg-config --modversion gpgme || gpgme-config --version)"; then
-	    printf 'Your current GPGME development version is: %s\n' "$GPGME_VERS"
-	else
-	    printf 'You do not have the GPGME development libraries installed.\n'
-	fi
-	errors=$((errors + 1))
-    fi
-    if [ -n "$TEMP_GPG" -a -d "$TEMP_GPG" ]; then
-	rm -rf "$TEMP_GPG"
-    fi
 
     cat > _check_gmime_cert.c <<EOF
 #include <stdio.h>
