diff --git a/configure.ac b/configure.ac
index e3a36f04..fbd9ff43 100644
--- a/configure.ac
+++ b/configure.ac
@@ -146,7 +146,7 @@ if test "x$with_anthy_utf8" != xno; then
                     [with_anthy_utf8=yes
                      AC_DEFINE(LIBANTHY_UTF8_CAPABLE, 1,
                      [libanthy-unicode can handle UTF-8])],
-                    [PKG_CHECK_MODULES(ANTHY_UTF8, anthy >= 8622,
+                    [PKG_CHECK_MODULES(ANTHY_UTF8, anthy >= 0.4,
                                        [with_anthy_utf8=yes
                                         AC_DEFINE(LIBANTHY_UTF8_CAPABLE, 1,
                                         [libanthy can handle UTF-8])],
