--- a/configure
+++ b/configure
@@ -38007,7 +38007,7 @@ if test "x$with_qt5" = xyes; then
   fi
   set -e
   # FIXME: move CMAKE_CXX_FLAGS into cmake_option
-  ${CMAKE} \
+  : ${CMAKE} \
     -B ${ac_abs_top_builddir}/qt5/applet.cmake \
     -S ${ac_abs_top_srcdir}/qt5/applet \
     -DCMAKE_CXX_FLAGS="${CXXFLAGS} ${X_CFLAGS}" \
--- a/qt5/applet/Makefile.in
+++ b/qt5/applet/Makefile.in
@@ -808,9 +808,6 @@ uninstall-am:
 .PRECIOUS: Makefile
 
 
-@QT5_TRUE@all clean install uninstall:
-@QT5_TRUE@	$(MAKE) $(AM_MAKEFLAGS) -C ../applet.cmake $@
-
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
