--- a/katarakt.pro	2020-11-21 07:57:05.333858604 +0700
+++ b/katarakt.pro	2020-11-21 07:58:23.713527253 +0700
@@ -17,9 +17,9 @@
 
     isEmpty(PKG_CONFIG):PKG_CONFIG = pkg-config    # same as in link_pkgconfig.prf
     POPPLER_VERSION = $$system($$PKG_CONFIG --modversion $$POPPLER)
-    POPPLER_VERSION_MAJOR = $$system(echo "$$POPPLER_VERSION" | cut -d . -f 1)
-    POPPLER_VERSION_MINOR = $$system(echo "$$POPPLER_VERSION" | cut -d . -f 2)
-    POPPLER_VERSION_MICRO = $$system(echo "$$POPPLER_VERSION" | cut -d . -f 3)
+    POPPLER_VERSION_MAJOR = $$system(echo "$$POPPLER_VERSION" | cut -d . -f 1 | sed 's/^0*//' | sed 's/^$/0/')
+    POPPLER_VERSION_MINOR = $$system(echo "$$POPPLER_VERSION" | cut -d . -f 2 | sed 's/^0*//' | sed 's/^$/0/')
+    POPPLER_VERSION_MICRO = $$system(echo "$$POPPLER_VERSION" | cut -d . -f 3 | sed 's/^0*//' | sed 's/^$/0/')
 
     DEFINES += POPPLER_VERSION_MAJOR=$$POPPLER_VERSION_MAJOR
     DEFINES += POPPLER_VERSION_MINOR=$$POPPLER_VERSION_MINOR
