--- a/src/tests/test-utils.c
+++ b/src/tests/test-utils.c
@@ -30,7 +30,7 @@
 void set_file_paths(void) {
 	char configpath[512], libpath[512], datapath[512];
 
-	my_strcpy(configpath, DEFAULT_CONFIG_PATH, sizeof(configpath));
+	my_strcpy(configpath, "../../lib", sizeof(configpath));
 	my_strcpy(libpath, DEFAULT_LIB_PATH, sizeof(libpath));
 	my_strcpy(datapath, DEFAULT_DATA_PATH, sizeof(datapath));
 
