--- a/meson/gpg_check_version.py
+++ b/meson/gpg_check_version.py
@@ -16,7 +16,7 @@ def check_version(gpg_version, accepted_
     gpg_major, gpg_minor, gpg_micro = [int(ver) for ver in gpg_version.split('.', 2)]
     acc_major, acc_minor, acc_micro = [int(ver) for ver in accepted_version.split('.', 2)]
     if is_gpgme:
-        return gpg_major == acc_major and gpg_minor >= acc_minor and gpg_micro >= acc_micro
+        return (gpg_major, gpg_minor, gpg_micro) >= (acc_major, acc_minor, acc_micro)
     else:
         return gpg_major == acc_major and gpg_minor == acc_minor and gpg_micro >= acc_micro
 
--- a/tool/seahorse-pgp-operation.c
+++ b/tool/seahorse-pgp-operation.c
@@ -300,7 +300,6 @@ event_cb (void *data, gpgme_event_io_t t
         break;
 
     case GPGME_EVENT_NEXT_KEY:
-    case GPGME_EVENT_NEXT_TRUSTITEM:
     default:
         /* Ignore unsupported event types */
         break;
