--- a/src/common/config-file.c
+++ b/src/common/config-file.c
@@ -23,6 +23,7 @@
 #include <netinet/in.h>
 #include <stdio.h>
 #include <stdlib.h>
+#include <string.h>
 #include <sys/socket.h>
 
 #include "config-file.h"
@@ -56,6 +57,8 @@ static const char *isolate_username_fmt
 /* Default password for the IsolatePID option. */
 static const char *isolate_password = "0";
 
+ATTR_HIDDEN
+int conf_file_set_enable_ipv6(const char *val, struct configuration *config);
 /*
  * Set the onion pool address range in the configuration object using the value
  * found in the conf file.
--- a/src/lib/torsocks.c
+++ b/src/lib/torsocks.c
@@ -59,6 +59,8 @@ static TSOCKS_INIT_ONCE(term_once);
  */
 static int is_suid;
 
+ATTR_HIDDEN
+int conf_file_set_enable_ipv6(const char *val, struct configuration *config);
 /*
  * Cleanup and exit with the given status. Note that the lib destructor will be
  * called after this call.
