diff -u -r Makefile Makefile
--- Makefile	2020-05-08 18:17:57.000000000 +1000
+++ Makefile	2020-05-19 18:40:54.345223164 +1000
@@ -92,7 +92,7 @@
 	fi
 
 .PHONY: develop
-develop: pyenv buildhash prepare
+develop: buildhash prepare
 	@set -eu -o pipefail ${SHELLFLAGS}; \
 	. "${ACTIVATE_SCRIPT}"; \
 	for dir in $(DEVEL); do \
diff -u -r pylib/Makefile pylib/Makefile
--- pylib/Makefile	2020-05-08 18:17:57.000000000 +1000
+++ pylib/Makefile	2020-05-19 18:50:01.075182994 +1000
@@ -52,7 +52,7 @@
 	python -m black anki/hooks.py
 	@touch $@
 
-BUILD_STEPS := .build/vernum .build/run-deps .build/dev-deps .build/py-proto anki/buildinfo.py .build/hooks
+BUILD_STEPS := .build/vernum .build/py-proto anki/buildinfo.py .build/hooks
 
 # Checking
 ######################
diff -u -r qt/Makefile qt/Makefile
--- qt/Makefile	2020-05-08 18:17:57.000000000 +1000
+++ qt/Makefile	2020-05-19 18:50:17.520181786 +1000
@@ -64,7 +64,7 @@
 	python -m black aqt/gui_hooks.py
 	@touch $@
 
-BUILD_STEPS := .build/vernum .build/run-deps .build/dev-deps .build/js .build/ui aqt/buildinfo.py .build/hooks .build/i18n
+BUILD_STEPS := .build/vernum .build/js .build/ui aqt/buildinfo.py .build/hooks .build/i18n
 
 # Checking
 ######################
--- Makefile	2020-05-08 18:17:57.000000000 +1000
+++ Makefile	2020-05-19 19:01:35.602131965 +1000
@@ -122,7 +122,7 @@
 	@echo "Build complete."
 
 .PHONY: build-rspy
-build-rspy: pyenv buildhash
+build-rspy: buildhash
 	@set -eu -o pipefail ${SHELLFLAGS}; \
 	. "${ACTIVATE_SCRIPT}"; \
 	$(SUBMAKE) -C rspy build BUILDFLAGS="$(BUILDFLAGS)"
--- Makefile	2020-05-19 19:41:44.191955000 +1000
+++ Makefile	2020-05-19 19:42:21.423952264 +1000
@@ -124,19 +124,16 @@
 .PHONY: build-rspy
 build-rspy: buildhash
 	@set -eu -o pipefail ${SHELLFLAGS}; \
-	. "${ACTIVATE_SCRIPT}"; \
 	$(SUBMAKE) -C rspy build BUILDFLAGS="$(BUILDFLAGS)"
 
 .PHONY: build-pylib
 build-pylib:
 	@set -eu -o pipefail ${SHELLFLAGS}; \
-	. "${ACTIVATE_SCRIPT}"; \
 	$(SUBMAKE) -C pylib build
 
 .PHONY: build-qt
 build-qt:
 	@set -eu -o pipefail ${SHELLFLAGS}; \
-	. "${ACTIVATE_SCRIPT}"; \
 	$(SUBMAKE) -C qt build
 
 .PHONY: clean
--- pylib/Makefile	2020-05-22 08:46:14.988607539 +1000
+++ pylib/Makefile	2020-05-22 08:46:38.376605821 +1000
@@ -41,7 +41,7 @@
 
 PROTODEPS := $(wildcard ../proto/*.proto)
 
-.build/py-proto: .build/dev-deps $(PROTODEPS)
+.build/py-proto: $(PROTODEPS)
 	protoc --proto_path=../proto --python_out=anki --mypy_out=anki $(PROTODEPS)
 	perl -i'' -pe 's/from fluent_pb2/from anki.fluent_pb2/' anki/backend_pb2.pyi
 	perl -i'' -pe 's/import fluent_pb2/import anki.fluent_pb2/' anki/backend_pb2.py
--- rspy/Makefile	2020-05-31 08:59:03.810152342 +1000
+++ rspy/Makefile	2020-05-08 18:17:57.000000000 +1000
@@ -51,7 +51,7 @@
 
 develop: .build/develop
 
-DEPS := .build/tools .build/vernum ../meta/buildhash \
+DEPS := .build/vernum ../meta/buildhash \
 	$(wildcard $(QT_FTL_TEMPLATES)/*.ftl) \
 	$(wildcard $(QT_FTL_LOCALES)/*/*.ftl) \
 	$(shell "${FIND}" ../rslib/src -name '*.rs') $(wildcard ../proto/*) \
