--- rspy/Makefile	2020-06-22 09:13:35.159000000 +1000
+++ rspy/Makefile	2020-06-21 18:49:39.803000000 +1000
@@ -69,7 +69,7 @@
 .build/build: $(DEPS)
 	touch ../proto/backend.proto
 	${BUILD_VARIABLES} \
-		maturin build -i "${PYTHON_FILE}" -o "$(OUTDIR)" $(BUILDFLAGS) --target $(RUST_TARGET)
+		maturin build -i "${PYTHON_FILE}" -o "$(OUTDIR)" $(BUILDFLAGS) --target $(RUST_TARGET) --manylinux off
 	touch $@
 
 check: .build/check
