--- a/build/ninja_gen/src/build.rs
+++ b/build/ninja_gen/src/build.rs
@@ -53,7 +53,7 @@
             have_n2: which::which("n2").is_ok(),
         };
 
-        build.add_action("build:configure", ConfigureBuild {})?;
+        // build.add_action("build:configure", ConfigureBuild {})?;
 
         Ok(build)
     }
--- a/build/ninja_gen/src/render.rs
+++ b/build/ninja_gen/src/render.rs
@@ -2,6 +2,7 @@
 // License: GNU AGPL, version 3 or later; http://www.gnu.org/licenses/agpl.html
 
 use std::fmt::Write;
+use std::env;
 
 use anki_io::create_dir_all;
 use anki_io::write_file_if_changed;
@@ -25,7 +26,8 @@
         writeln!(&mut buf, "builddir = {}", self.buildroot.as_str()).unwrap();
         writeln!(
             &mut buf,
-            "runner = $builddir/rust/release/{}",
+            "runner = $builddir/rust/{}/release/{}",
+            env::var("RUST_BUILD").unwrap(),
             with_exe("runner")
         )
         .unwrap();
--- a/build/ninja_gen/src/cargo.rs
+++ b/build/ninja_gen/src/cargo.rs
@@ -97,7 +97,7 @@
             .release_override
             .unwrap_or_else(|| build.build_profile());
         let release_arg = profile_arg_for_cargo(release_build).unwrap_or_default();
-        let target_arg = if let Some(target) = self.target {
+        let target_arg = if let Some(target) = &self.target {
             format!("--target {target}")
         } else {
             "".into()
@@ -115,7 +115,7 @@
         let output_root = Utf8Path::new("$builddir/rust");
         for output in self.outputs {
             let name = output.name();
-            let path = output.path(output_root, self.target, release_build);
+            let path = output.path(output_root, self.target.as_deref(), release_build);
             build.add_outputs_ext(name, vec![path], true);
         }
     }
--- a/build/configure/src/platform.rs
+++ b/build/configure/src/platform.rs
@@ -6,8 +6,8 @@
 use ninja_gen::archives::Platform;
 
 /// Please see [`overriden_python_target_platform()`] for details.
-pub fn overriden_rust_target_triple() -> Option<&'static str> {
-    overriden_python_wheel_platform().map(|p| p.as_rust_triple())
+pub fn overriden_rust_target_triple() -> Option<String> {
+    env::var("RUST_TARGET").ok()
 }
 
 /// Usually None to use the host architecture, but:
--- a/build/ninja_gen/src/cargo.rs
+++ b/build/ninja_gen/src/cargo.rs
@@ -79,7 +79,7 @@
 pub struct CargoBuild<'a> {
     pub inputs: BuildInput,
     pub outputs: &'a [RustOutput<'a>],
-    pub target: Option<&'static str>,
+    pub target: Option<String>,
     pub extra_args: &'a str,
     pub release_override: Option<BuildProfile>,
 }
