--- a/anki/mpv.py
+++ b/anki/mpv.py
@@ -104,9 +104,9 @@
         """
         self.argv = [self.executable]
         self.argv += self.default_argv
-        self.argv += ["--input-ipc-server", self._sock_filename]
+        self.argv += ["--input-ipc-server="+self._sock_filename]
         if self.window_id is not None:
-            self.argv += ["--wid", str(self.window_id)]
+            self.argv += ["--wid="+str(self.window_id)]
 
     def _start_process(self):
         """Start the mpv process.

--- a/anki/sound.py
+++ b/anki/sound.py
@@ -123,10 +123,7 @@
 
 def setMpvConfigBase(base):
     mpvConfPath = os.path.join(base, "mpv.conf")
-    MpvManager.default_argv += [
-        "--no-config",
-        "--include="+mpvConfPath,
-    ]
+    MpvManager.default_argv += ["--include="+mpvConfPath]
 
 mpvManager = None
 
