--- a/aqt/deckconf.py
+++ b/aqt/deckconf.py
@@ -172,7 +172,7 @@ class DeckConf(QDialog):
         f.lrnGradInt.setValue(c['ints'][0])
         f.lrnEasyInt.setValue(c['ints'][1])
         f.lrnEasyInt.setValue(c['ints'][1])
-        f.lrnFactor.setValue(c['initialFactor']/10.0)
+        f.lrnFactor.setValue(c['initialFactor']//10)
         f.newOrder.setCurrentIndex(c['order'])
         f.newPerDay.setValue(c['perDay'])
         f.bury.setChecked(c.get("bury", True))
@@ -180,7 +180,7 @@ class DeckConf(QDialog):
         # rev
         c = self.conf['rev']
         f.revPerDay.setValue(c['perDay'])
-        f.easyBonus.setValue(c['ease4']*100)
+        f.easyBonus.setValue(int(c['ease4'])*100)
         f.fi1.setValue(c['ivlFct']*100)
         f.maxIvl.setValue(c['maxIvl'])
         f.revplim.setText(self.parentLimText('rev'))
@@ -192,7 +192,7 @@ class DeckConf(QDialog):
         # lapse
         c = self.conf['lapse']
         f.lapSteps.setText(self.listToUser(c['delays']))
-        f.lapMult.setValue(c['mult']*100)
+        f.lapMult.setValue(int(c['mult'])*100)
         f.lapMinInt.setValue(c['minInt'])
         f.leechThreshold.setValue(c['leechFails'])
         f.leechAction.setCurrentIndex(c['leechAction'])
--- a/aqt/preferences.py
+++ b/aqt/preferences.py
@@ -77,8 +77,8 @@ class Preferences(QDialog):
             f.hwAccel.setVisible(False)
         else:
             f.hwAccel.setChecked(self.mw.pm.glMode() != "software")
-        f.lrnCutoff.setValue(qc['collapseTime']/60.0)
-        f.timeLimit.setValue(qc['timeLim']/60.0)
+        f.lrnCutoff.setValue(qc['collapseTime']//60)
+        f.timeLimit.setValue(qc['timeLim']//60)
         f.showEstimates.setChecked(qc['estTimes'])
         f.showProgress.setChecked(qc['dueCounts'])
         f.nightMode.setChecked(qc.get("nightMode", False))
