diff --git a/main.go b/main.go
index 5b1428d..2f7087a 100644
--- a/main.go
+++ b/main.go
@@ -24,7 +24,7 @@ the Matrix room (https://matrix.to/#/#ntfy:matrix.org).
 
 ntfy %s (%s), runtime %s, built at %s
 Copyright (C) 2022 Philipp C. Heckel, licensed under Apache License 2.0 & GPLv2
-`, version, commit[:7], runtime.Version(), date)
+`, version, commit, runtime.Version(), date)
 
 	app := cmd.New()
 	app.Version = version
