--- ./configure.ac.orig	2021-09-26 14:46:32.840552276 -0400
+++ ./configure.ac	2021-09-26 14:46:41.313573652 -0400
@@ -10,7 +10,7 @@
 AM_CONFIG_HEADER([config.h])
 
 # Generate python installation stuff
-m4_define_default([_AM_PYTHON_INTERPRETER_LIST],[python3.8 python3.7 python3.9])
+m4_define_default([_AM_PYTHON_INTERPRETER_LIST],[python3.8 python3.7 python3.9 python3.10 python3.11 python3.12])
 AM_PATH_PYTHON([3.7])
 AC_SUBST([idjcpythondir], [\${pkglibdir}/site-packages])
 AC_SUBST([idjcpkgpythondir], [\${pkglibdir}/site-packages/idjc])
