--- pal/shared/gpuopen/cmake/AMD.cmake	2019-06-30 09:35:25.000000000 +0200
+++ -	2019-07-02 11:07:06.366534309 +0200
@@ -46,7 +46,8 @@
         target_compile_options(${name} PRIVATE
             -Wall    # Enable warnings about questionable language constructs.
             -Wextra  # Enable extra warnings that are not enabled by -Wall.
-            -Werror) # Turn warnings into errors.
+            -Werror  # Turn warnings into errors.
+	    -Wno-error=deprecated-copy)
 
 endfunction()
 
