--- pal/shared/gpuopen/cmake/AMD.cmake	2019-10-18 11:06:42.000000000 +0200
+++ -	2019-10-29 19:06:00.059615384 +0100
@@ -119,6 +120,7 @@
                 -Wall   # Enable warnings about questionable language constructs.
                 -Wextra # Enable extra warnings that are not enabled by -Wall.
                 -Werror # Turn warnings into errors.
+		-Wno-error=deprecated-copy
         )
 
     elseif(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
--- pal/shared/gpuopen/cmake/AMD.cmake	2020-02-22 23:54:22.199302309 +0100
+++ -	2020-02-22 23:55:03.296546762 +0100
@@ -94,6 +94,7 @@
                 -fno-math-errno             # Single instruction math operations do not set ERRNO.
                 -fvisibility=hidden         # Hide export symbols by default
                 -fvisibility-inlines-hidden # Also hide inline export symbols
+		-Wno-error=deprecated-copy
         )
 
     elseif(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
