diff --git a/pal/src/core/hw/gfxip/gfx9/gfx9UniversalCmdBuffer.cpp b/pal/src/core/hw/gfxip/gfx9/gfx9UniversalCmdBuffer.cpp
index 2d62ec3ef..e14ccbb9c 100644
--- a/pal/src/core/hw/gfxip/gfx9/gfx9UniversalCmdBuffer.cpp
+++ b/pal/src/core/hw/gfxip/gfx9/gfx9UniversalCmdBuffer.cpp
@@ -10152,7 +10152,7 @@ uint32 UniversalCmdBuffer::BuildExecuteIndirectIb2Packets(
                                                                                      drawIndexReg,
                                                                                      1,
                                                                                      pParamData[cmdIndex].argBufSize,
-                                                                                     NULL,
+                                                                                     0UL,
                                                                                      PacketPredicate(),
                                                                                      usesLegacyMsFastLaunch,
                                                                                      pDeCmdIb2Space);
@@ -10164,7 +10164,7 @@ uint32 UniversalCmdBuffer::BuildExecuteIndirectIb2Packets(
                                                                                       drawIndexReg,
                                                                                       1,
                                                                                       pParamData[cmdIndex].argBufSize,
-                                                                                      NULL,
+                                                                                      0UL,
                                                                                       PacketPredicate(),
                                                                                       usesLegacyMsFastLaunch,
                                                                                       pDeCmdIb2Space);
