--- pal/shared/gpuopen/src/posix/ddPosixSocket.cpp	2019-03-01 15:31:04.000000000 +0100
+++ -	2019-03-04 15:42:00.789284605 +0100
@@ -34,12 +34,12 @@
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <sys/un.h>
-#include <sys/fcntl.h>
 #include <netinet/in.h>
 #include <netinet/tcp.h>
 #include <netdb.h>
 #include <arpa/inet.h>
 
+#include <fcntl.h>
 #include <unistd.h>
 #include <stdio.h>
 #include <stdlib.h>
 
