diff --git a/pal/src/util/lnx/lnxSysUtil.cpp b/pal/src/util/lnx/lnxSysUtil.cpp
index 3105084b..e61c438a 100644
--- a/pal/src/util/lnx/lnxSysUtil.cpp
+++ b/pal/src/util/lnx/lnxSysUtil.cpp
@@ -1207,6 +1207,7 @@ bool IsDebuggerAttached()
 Result SetRwxFilePermissions(
     const char* pFileName)
 {
+#define ACCESSPERMS (S_IRWXU|S_IRWXG|S_IRWXO|S_ISVTX)
     Result result = Result::Success;
     struct stat info;
     int ret = stat(pFileName, &info);
