From b6a0c67f6ac76fb7bf8234951678b77fbac12d76 Mon Sep 17 00:00:00 2001
From: Yarrow Angelweed <yarrow.angelweed@gmail.com>
Date: Wed, 6 Mar 2024 12:17:38 -0500
Subject: [PATCH] Take nightly clippy advice

---
 src/help.rs | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/help.rs b/src/help.rs
index 23c79aa..0e8c3c3 100644
--- a/src/help.rs
+++ b/src/help.rs
@@ -123,8 +123,7 @@ impl<'a> Section<'a> {
         Ok(0)
     }
     fn next_line_help_indent(&self) -> &'a str {
-        let max_indent =
-            self.entries.iter().map(|e| e.item.indented_by()).fold(0, std::cmp::Ord::max);
+        let max_indent = self.entries.iter().map(|e| e.item.indented_by()).fold(0, Ord::max);
         let indent_len = (max_indent + 4).min(BLANKS.len());
         &BLANKS[..indent_len]
     }
