Index: FreeFileSync-13.5/FreeFileSync/Source/Makefile
===================================================================
--- FreeFileSync-13.5.orig/FreeFileSync/Source/Makefile
+++ FreeFileSync-13.5/FreeFileSync/Source/Makefile
@@ -17,9 +17,8 @@ LDFLAGS += `pkg-config --libs   libcurl`
 CXXFLAGS  += `pkg-config --cflags libssh2`
 LDFLAGS += `pkg-config --libs   libssh2`
 
-CXXFLAGS  += `pkg-config --cflags gtk+-2.0`
-#treat as system headers so that warnings are hidden:
-CXXFLAGS  += -isystem/usr/include/gtk-2.0
+CXXFLAGS  += `pkg-config --cflags gtk+-3.0`
+LDFLAGS  += `pkg-config --libs gtk+-3.0`
 
 #support for SELinux (optional)
 SELINUX_EXISTING=$(shell pkg-config --exists libselinux && echo YES)
Index: FreeFileSync-13.5/FreeFileSync/Source/RealTimeSync/Makefile
===================================================================
--- FreeFileSync-13.5.orig/FreeFileSync/Source/RealTimeSync/Makefile
+++ FreeFileSync-13.5/FreeFileSync/Source/RealTimeSync/Makefile
@@ -8,9 +8,8 @@ CXXFLAGS += -std=c++23 -pipe -DWXINTL_NO
 LDFLAGS += -s -no-pie `wx-config --libs std, aui, richtext --debug=no` -pthread
 
 #Gtk - support "no button border"
-CXXFLAGS  += `pkg-config --cflags gtk+-2.0`
-#treat as system headers so that warnings are hidden:
-CXXFLAGS  += -isystem/usr/include/gtk-2.0
+CXXFLAGS  += `pkg-config --cflags gtk+-3.0`
+LDFLAGS  += `pkg-config --libs gtk+-3.0`
 
 cppFiles=
 cppFiles+=application.cpp
