--- GNUmakefile.orig	2015-02-16 10:07:31.188232968 +0100
+++ GNUmakefile	2015-02-16 10:09:08.028179047 +0100
@@ -16,6 +16,7 @@ IS_SUN_CC = $(shell $(CXX) -V 2>&1 | $(E
 IS_LINUX = $(shell $(CXX) -dumpmachine 2>&1 | $(EGREP) -c "linux")
 IS_MINGW = $(shell $(CXX) -dumpmachine 2>&1 | $(EGREP) -c "mingw")
 CLANG_COMPILER = $(shell $(CXX) --version 2>&1 | $(EGREP) -i -c "clang version")
+VERSION = 562
 
 # Default prefix for make install
 ifeq ($(PREFIX),)
@@ -165,7 +166,7 @@ libcryptopp.a: $(LIBOBJS)
 	$(RANLIB) $@
 
 libcryptopp.so: $(LIBOBJS)
-	$(CXX) -shared -o $@ $(LIBOBJS)
+	$(CXX) -shared -Wl,-soname=libcryptopp.so.$(VERSION) -o $@.$(VERSION) $(LIBOBJS)
 
 cryptest.exe: libcryptopp.a $(TESTOBJS)
 	$(CXX) -o $@ $(CXXFLAGS) $(TESTOBJS) ./libcryptopp.a $(LDFLAGS) $(LDLIBS)
