--- GNUmakefile.orig	2016-03-20 18:33:19.932703639 -0600
+++ GNUmakefile	2016-03-20 22:08:25.169365978 -0600
@@ -36,6 +36,8 @@
 INTEL_COMPILER := $(shell $(CXX) --version 2>&1 | $(EGREP) -c "\(ICC\)")
 MACPORTS_COMPILER := $(shell $(CXX) --version 2>&1 | $(EGREP) -i -c "macports")
 
+VERSION = 563
+
 # Default prefix for make install
 ifeq ($(PREFIX),)
 PREFIX = /usr
@@ -375,7 +377,7 @@
 	$(RANLIB) $@
 
 libcryptopp.so: public_service | $(LIBOBJS)
-	$(CXX) -shared -o $@ $(CXXFLAGS) $(GOLD_OPTION) $(LIBOBJS) $(LDLIBS)
+	$(CXX) -shared -Wl,-soname=libcryptopp.so.$(VERSION) -o $@.$(VERSION) $(CXXFLAGS) $(GOLD_OPTION) $(LIBOBJS) $(LDLIBS)
 
 libcryptopp.dylib: $(LIBOBJS)
 	$(CXX) -dynamiclib -o $@ $(CXXFLAGS) -install_name "$@" -current_version "$(LIB_MAJOR).$(LIB_MINOR).$(LIB_PATCH)" -compatibility_version "$(LIB_MAJOR).$(LIB_MINOR)" $(LIBOBJS)
