Index: VTK-9.0.1/Common/Core/vtkGenericDataArrayLookupHelper.h
===================================================================
--- VTK-9.0.1.orig/Common/Core/vtkGenericDataArrayLookupHelper.h
+++ VTK-9.0.1/Common/Core/vtkGenericDataArrayLookupHelper.h
@@ -25,6 +25,7 @@
 #include "vtkIdList.h"
 #include <algorithm>
 #include <cmath>
+#include <limits>
 #include <unordered_map>
 #include <vector>
 
Index: VTK-9.0.1/Common/DataModel/vtkPiecewiseFunction.cxx
===================================================================
--- VTK-9.0.1.orig/Common/DataModel/vtkPiecewiseFunction.cxx
+++ VTK-9.0.1/Common/DataModel/vtkPiecewiseFunction.cxx
@@ -22,6 +22,7 @@
 #include <cassert>
 #include <cmath>
 #include <iterator>
+#include <limits>
 #include <set>
 #include <vector>
 
Index: VTK-9.0.1/Rendering/Core/vtkColorTransferFunction.cxx
===================================================================
--- VTK-9.0.1.orig/Rendering/Core/vtkColorTransferFunction.cxx
+++ VTK-9.0.1/Rendering/Core/vtkColorTransferFunction.cxx
@@ -21,6 +21,7 @@
 #include <algorithm>
 #include <cmath>
 #include <iterator>
+#include <limits>
 #include <set>
 #include <vector>
 
Index: VTK-9.0.1/Common/ComputationalGeometry/Testing/Cxx/UnitTestParametricSpline.cxx
===================================================================
--- VTK-9.0.1.orig/Common/ComputationalGeometry/Testing/Cxx/UnitTestParametricSpline.cxx
+++ VTK-9.0.1/Common/ComputationalGeometry/Testing/Cxx/UnitTestParametricSpline.cxx
@@ -22,6 +22,7 @@
 #include "vtkSmartPointer.h"
 
 #include <cmath>
+#include <limits>
 
 #include "vtkTestErrorObserver.h"
 
Index: VTK-9.0.1/Common/Core/Testing/Cxx/UnitTestMath.cxx
===================================================================
--- VTK-9.0.1.orig/Common/Core/Testing/Cxx/UnitTestMath.cxx
+++ VTK-9.0.1/Common/Core/Testing/Cxx/UnitTestMath.cxx
@@ -24,6 +24,7 @@
 #include "vtkUnsignedCharArray.h"
 #include "vtkUnsignedShortArray.h"
 
+#include <limits>
 #include <vector>
 
 static int TestPi();
Index: VTK-9.0.1/Common/Core/vtkMersenneTwister.cxx
===================================================================
--- VTK-9.0.1.orig/Common/Core/vtkMersenneTwister.cxx
+++ VTK-9.0.1/Common/Core/vtkMersenneTwister.cxx
@@ -28,6 +28,7 @@
 #include "vtkMultiThreader.h"
 #include "vtkNew.h"
 
+#include <limits>
 #include <map>
 
 namespace
Index: VTK-9.0.1/Common/DataModel/Testing/Cxx/TestPentagonalPrism.cxx
===================================================================
--- VTK-9.0.1.orig/Common/DataModel/Testing/Cxx/TestPentagonalPrism.cxx
+++ VTK-9.0.1/Common/DataModel/Testing/Cxx/TestPentagonalPrism.cxx
@@ -22,6 +22,7 @@
 #include "vtkCellArray.h"
 #include "vtkMathUtilities.h"
 #include "vtkPoints.h"
+#include <limits>
 #include <map>
 #include <sstream>
 #include <string>
Index: VTK-9.0.1/Common/DataModel/Testing/Cxx/UnitTestCells.cxx
===================================================================
--- VTK-9.0.1.orig/Common/DataModel/Testing/Cxx/UnitTestCells.cxx
+++ VTK-9.0.1/Common/DataModel/Testing/Cxx/UnitTestCells.cxx
@@ -58,6 +58,7 @@
 #include "vtkMath.h"
 #include "vtkMathUtilities.h"
 #include "vtkPoints.h"
+#include <limits>
 #include <map>
 #include <sstream>
 #include <string>
Index: VTK-9.0.1/Common/DataModel/Testing/Cxx/UnitTestImplicitDataSet.cxx
===================================================================
--- VTK-9.0.1.orig/Common/DataModel/Testing/Cxx/UnitTestImplicitDataSet.cxx
+++ VTK-9.0.1/Common/DataModel/Testing/Cxx/UnitTestImplicitDataSet.cxx
@@ -20,6 +20,7 @@
 #include "vtkTestErrorObserver.h"
 
 #include <cstdio>
+#include <limits>
 #include <sstream>
 
 static vtkSmartPointer<vtkImageData> MakeVolume(int, int, int);
Index: VTK-9.0.1/Common/DataModel/Testing/Cxx/UnitTestImplicitVolume.cxx
===================================================================
--- VTK-9.0.1.orig/Common/DataModel/Testing/Cxx/UnitTestImplicitVolume.cxx
+++ VTK-9.0.1/Common/DataModel/Testing/Cxx/UnitTestImplicitVolume.cxx
@@ -20,6 +20,7 @@
 #include "vtkTestErrorObserver.h"
 
 #include <cstdio>
+#include <limits>
 #include <sstream>
 
 static vtkSmartPointer<vtkImageData> MakeVolume(int, int, int);
Index: VTK-9.0.1/Common/DataModel/Testing/Cxx/UnitTestLine.cxx
===================================================================
--- VTK-9.0.1.orig/Common/DataModel/Testing/Cxx/UnitTestLine.cxx
+++ VTK-9.0.1/Common/DataModel/Testing/Cxx/UnitTestLine.cxx
@@ -19,6 +19,7 @@
 #include "vtkMath.h"
 #include "vtkMinimalStandardRandomSequence.h"
 #include "vtkSmartPointer.h"
+#include <limits>
 
 namespace
 {
Index: VTK-9.0.1/Common/DataModel/vtkPolyhedron.cxx
===================================================================
--- VTK-9.0.1.orig/Common/DataModel/vtkPolyhedron.cxx
+++ VTK-9.0.1/Common/DataModel/vtkPolyhedron.cxx
@@ -33,6 +33,7 @@ PURPOSE.  See the above copyright notice
 #include "vtkVector.h"
 
 #include <functional>
+#include <limits>
 #include <map>
 #include <set>
 #include <unordered_map>
Index: VTK-9.0.1/Common/Misc/Testing/Cxx/UnitTestFunctionParser.cxx
===================================================================
--- VTK-9.0.1.orig/Common/Misc/Testing/Cxx/UnitTestFunctionParser.cxx
+++ VTK-9.0.1/Common/Misc/Testing/Cxx/UnitTestFunctionParser.cxx
@@ -22,6 +22,7 @@
 #include "vtkTestErrorObserver.h"
 
 #include <algorithm>
+#include <limits>
 #include <sstream>
 #include <string>
 #include <vector>
Index: VTK-9.0.1/Filters/FlowPaths/vtkEvenlySpacedStreamlines2D.cxx
===================================================================
--- VTK-9.0.1.orig/Filters/FlowPaths/vtkEvenlySpacedStreamlines2D.cxx
+++ VTK-9.0.1/Filters/FlowPaths/vtkEvenlySpacedStreamlines2D.cxx
@@ -55,6 +55,7 @@ PURPOSE.  See the above copyright notice
 #include <array>
 #include <iostream>
 #include <iterator>
+#include <limits>
 #include <vector>
 
 vtkObjectFactoryNewMacro(vtkEvenlySpacedStreamlines2D);
Index: VTK-9.0.1/Filters/FlowPaths/vtkLagrangianMatidaIntegrationModel.cxx
===================================================================
--- VTK-9.0.1.orig/Filters/FlowPaths/vtkLagrangianMatidaIntegrationModel.cxx
+++ VTK-9.0.1/Filters/FlowPaths/vtkLagrangianMatidaIntegrationModel.cxx
@@ -25,6 +25,7 @@
 #include "vtkStringArray.h"
 
 #include <cstring>
+#include <limits>
 
 vtkObjectFactoryNewMacro(vtkLagrangianMatidaIntegrationModel);
 
Index: VTK-9.0.1/Filters/HyperTree/vtkHyperTreeGridThreshold.cxx
===================================================================
--- VTK-9.0.1.orig/Filters/HyperTree/vtkHyperTreeGridThreshold.cxx
+++ VTK-9.0.1/Filters/HyperTree/vtkHyperTreeGridThreshold.cxx
@@ -28,6 +28,7 @@
 #include "vtkHyperTreeGridNonOrientedCursor.h"
 
 #include <cmath>
+#include <limits>
 
 vtkStandardNewMacro(vtkHyperTreeGridThreshold);
 
Index: VTK-9.0.1/Filters/ParallelMPI/vtkDistributedPointCloudFilter.cxx
===================================================================
--- VTK-9.0.1.orig/Filters/ParallelMPI/vtkDistributedPointCloudFilter.cxx
+++ VTK-9.0.1/Filters/ParallelMPI/vtkDistributedPointCloudFilter.cxx
@@ -31,6 +31,7 @@
 #include "vtkSmartPointer.h"
 
 #include <algorithm>
+#include <limits>
 
 // Histogram precision to divide space in two
 static const int HISTOGRAM_SIZE = 1024;
Index: VTK-9.0.1/Filters/Points/Testing/Cxx/UnitTestKernels.cxx
===================================================================
--- VTK-9.0.1.orig/Filters/Points/Testing/Cxx/UnitTestKernels.cxx
+++ VTK-9.0.1/Filters/Points/Testing/Cxx/UnitTestKernels.cxx
@@ -30,6 +30,7 @@
 #include "vtkVoronoiKernel.h"
 
 #include <cmath>
+#include <limits>
 #include <sstream>
 #include <string>
 
Index: VTK-9.0.1/Filters/Points/vtkEllipsoidalGaussianKernel.cxx
===================================================================
--- VTK-9.0.1.orig/Filters/Points/vtkEllipsoidalGaussianKernel.cxx
+++ VTK-9.0.1/Filters/Points/vtkEllipsoidalGaussianKernel.cxx
@@ -22,6 +22,7 @@
 #include "vtkObjectFactory.h"
 #include "vtkPointData.h"
 
+#include <limits>
 vtkStandardNewMacro(vtkEllipsoidalGaussianKernel);
 
 //----------------------------------------------------------------------------
Index: VTK-9.0.1/Filters/Points/vtkGaussianKernel.cxx
===================================================================
--- VTK-9.0.1.orig/Filters/Points/vtkGaussianKernel.cxx
+++ VTK-9.0.1/Filters/Points/vtkGaussianKernel.cxx
@@ -21,6 +21,7 @@
 #include "vtkMathUtilities.h"
 #include "vtkObjectFactory.h"
 #include "vtkPointData.h"
+#include <limits>
 
 vtkStandardNewMacro(vtkGaussianKernel);
 
Index: VTK-9.0.1/Filters/Points/vtkProbabilisticVoronoiKernel.cxx
===================================================================
--- VTK-9.0.1.orig/Filters/Points/vtkProbabilisticVoronoiKernel.cxx
+++ VTK-9.0.1/Filters/Points/vtkProbabilisticVoronoiKernel.cxx
@@ -21,6 +21,7 @@
 #include "vtkMathUtilities.h"
 #include "vtkObjectFactory.h"
 #include "vtkPointData.h"
+#include <limits>
 
 vtkStandardNewMacro(vtkProbabilisticVoronoiKernel);
 
Index: VTK-9.0.1/Filters/Points/vtkShepardKernel.cxx
===================================================================
--- VTK-9.0.1.orig/Filters/Points/vtkShepardKernel.cxx
+++ VTK-9.0.1/Filters/Points/vtkShepardKernel.cxx
@@ -21,6 +21,7 @@
 #include "vtkMathUtilities.h"
 #include "vtkObjectFactory.h"
 #include "vtkPointData.h"
+#include <limits>
 
 vtkStandardNewMacro(vtkShepardKernel);
 
Index: VTK-9.0.1/Filters/Statistics/Testing/Cxx/TestPCAStatistics.cxx
===================================================================
--- VTK-9.0.1.orig/Filters/Statistics/Testing/Cxx/TestPCAStatistics.cxx
+++ VTK-9.0.1/Filters/Statistics/Testing/Cxx/TestPCAStatistics.cxx
@@ -22,6 +22,7 @@
 #include "vtkTestUtilities.h"
 
 #include "vtksys/SystemTools.hxx"
+#include <limits>
 
 // Perform a fuzzy compare of floats/doubles
 template <class A>
Index: VTK-9.0.1/IO/CityGML/vtkCityGMLReader.cxx
===================================================================
--- VTK-9.0.1.orig/IO/CityGML/vtkCityGMLReader.cxx
+++ VTK-9.0.1/IO/CityGML/vtkCityGMLReader.cxx
@@ -46,6 +46,7 @@
 #include <cstdlib>
 #include <cstring>
 #include <iostream>
+#include <limits>
 #include <sstream>
 #include <stdexcept>
 #include <string>
Index: VTK-9.0.1/IO/GDAL/vtkGDALRasterReader.cxx
===================================================================
--- VTK-9.0.1.orig/IO/GDAL/vtkGDALRasterReader.cxx
+++ VTK-9.0.1/IO/GDAL/vtkGDALRasterReader.cxx
@@ -48,6 +48,7 @@
 #include <cassert>
 #include <cmath>
 #include <iostream>
+#include <limits>
 #include <sstream>
 #include <vector>
 
Index: VTK-9.0.1/IO/Geometry/vtkSTLWriter.cxx
===================================================================
--- VTK-9.0.1.orig/IO/Geometry/vtkSTLWriter.cxx
+++ VTK-9.0.1/IO/Geometry/vtkSTLWriter.cxx
@@ -34,6 +34,7 @@
 #include <io.h> /* unlink */
 #endif
 
+#include <limits>
 namespace
 {
 // For C format strings
Index: VTK-9.0.1/IO/NetCDF/vtkNetCDFCAMReader.cxx
===================================================================
--- VTK-9.0.1.orig/IO/NetCDF/vtkNetCDFCAMReader.cxx
+++ VTK-9.0.1/IO/NetCDF/vtkNetCDFCAMReader.cxx
@@ -32,6 +32,7 @@
 #include "vtkStreamingDemandDrivenPipeline.h"
 #include "vtkUnstructuredGrid.h"
 
+#include <limits>
 #include <set>
 #include <sstream>
 #include <vector>
Index: VTK-9.0.1/IO/TecplotTable/vtkTecplotTableReader.cxx
===================================================================
--- VTK-9.0.1.orig/IO/TecplotTable/vtkTecplotTableReader.cxx
+++ VTK-9.0.1/IO/TecplotTable/vtkTecplotTableReader.cxx
@@ -35,6 +35,7 @@
 #include <algorithm>
 #include <iostream>
 #include <iterator>
+#include <limits>
 #include <set>
 #include <sstream>
 #include <stdexcept>
Index: VTK-9.0.1/IO/XML/Testing/Cxx/TestXMLWriteRead.cxx
===================================================================
--- VTK-9.0.1.orig/IO/XML/Testing/Cxx/TestXMLWriteRead.cxx
+++ VTK-9.0.1/IO/XML/Testing/Cxx/TestXMLWriteRead.cxx
@@ -20,6 +20,7 @@
 #include "vtkPoints.h"
 
 #include "vtkTestUtilities.h"
+#include <limits>
 #include <string>
 
 namespace
Index: VTK-9.0.1/Rendering/OpenGL2/vtkOpenGLInstanceCulling.cxx
===================================================================
--- VTK-9.0.1.orig/Rendering/OpenGL2/vtkOpenGLInstanceCulling.cxx
+++ VTK-9.0.1/Rendering/OpenGL2/vtkOpenGLInstanceCulling.cxx
@@ -31,6 +31,7 @@
 
 #include <algorithm>
 #include <array>
+#include <limits>
 #include <sstream>
 
 vtkStandardNewMacro(vtkOpenGLInstanceCulling);
Index: VTK-9.0.1/ThirdParty/vtkm/vtkvtkm/vtk-m/vtkm/cont/StorageBasic.cxx
===================================================================
--- VTK-9.0.1.orig/ThirdParty/vtkm/vtkvtkm/vtk-m/vtkm/cont/StorageBasic.cxx
+++ VTK-9.0.1/ThirdParty/vtkm/vtkvtkm/vtk-m/vtkm/cont/StorageBasic.cxx
@@ -35,6 +35,7 @@
 
 #include <cstddef>
 #include <cstdlib>
+#include <limits>
 
 namespace vtkm
 {
Index: VTK-9.0.1/ThirdParty/vtkm/vtkvtkm/vtk-m/vtkm/worklet/testing/UnitTestKdTreeBuildNNS.cxx
===================================================================
--- VTK-9.0.1.orig/ThirdParty/vtkm/vtkvtkm/vtk-m/vtkm/worklet/testing/UnitTestKdTreeBuildNNS.cxx
+++ VTK-9.0.1/ThirdParty/vtkm/vtkvtkm/vtk-m/vtkm/worklet/testing/UnitTestKdTreeBuildNNS.cxx
@@ -11,6 +11,7 @@
 #include <random>
 #include <vtkm/cont/Algorithm.h>
 #include <vtkm/worklet/KdTree3D.h>
+#include <limits>
 
 namespace
 {
Index: VTK-9.0.1/Utilities/MetaIO/vtkmetaio/metaObject.cxx
===================================================================
--- VTK-9.0.1.orig/Utilities/MetaIO/vtkmetaio/metaObject.cxx
+++ VTK-9.0.1/Utilities/MetaIO/vtkmetaio/metaObject.cxx
@@ -19,6 +19,7 @@
 #if defined (__BORLANDC__) && (__BORLANDC__ >= 0x0580)
 #include <mem.h>
 #endif
+#include <limits>
 
 #if (METAIO_USE_NAMESPACE)
 namespace METAIO_NAMESPACE {
