--- a/LspCpp/include/LibLsp/lsp/SimpleTimer.h
+++ b/LspCpp/include/LibLsp/lsp/SimpleTimer.h
@@ -1,10 +1,18 @@
 #pragma once
 #include <thread>
 #include <atomic>
+#include <chrono>
 #include <functional>
 #include <boost/asio.hpp>
+#include <boost/version.hpp>
 
-template<typename Duration = boost::posix_time::milliseconds>
+template<typename Duration =
+#if BOOST_VERSION > 108900
+    std::chrono::milliseconds
+#else
+    boost::posix_time::milliseconds
+#endif
+    >
 class SimpleTimer
 {
 public:
@@ -44,6 +52,10 @@ private:
     std::atomic_bool is_running_;
     std::function<void()> call_back;
     boost::asio::io_context _ios;
+#if BOOST_VERSION > 108900
+    boost::asio::system_timer _deadline_timer;
+#else
     boost::asio::deadline_timer _deadline_timer;
+#endif
     std::thread _thread;
 };
--- a/LspCpp/src/lsp/ParentProcessWatcher.cpp
+++ b/LspCpp/src/lsp/ParentProcessWatcher.cpp
@@ -1,5 +1,6 @@
 #include "LibLsp/lsp/ParentProcessWatcher.h"
 #include <algorithm>
+#include <chrono>
 
 #include <boost/version.hpp>
 #if BOOST_VERSION >= 108800 // v2 is now default
@@ -29,7 +30,12 @@ using namespace std;
 
 struct ParentProcessWatcher::ParentProcessWatcherData : std::enable_shared_from_this<ParentProcessWatcherData>
 {
-    std::unique_ptr<SimpleTimer<boost::posix_time::seconds>> timer;
+#if BOOST_VERSION > 108900
+    using WatcherDuration = std::chrono::seconds;
+#else
+    using WatcherDuration = boost::posix_time::seconds;
+#endif
+    std::unique_ptr<SimpleTimer<WatcherDuration>> timer;
     lsp::Log& _log;
     std::function<void()> on_exit;
     lsp::ProcessIoService asio_io;
@@ -94,7 +100,7 @@ struct ParentProcessWatcher::ParentProce
                             );
                         }
 
-                        self->timer = std::make_unique<SimpleTimer<boost::posix_time::seconds>>(
+                        self->timer = std::make_unique<SimpleTimer<WatcherDuration>>(
                             self->_poll_delay_secs, [=]() { self->run(); }
                         );
                     }
