--- a/configure
+++ b/configure
@@ -8512,9 +8512,6 @@ then :
 
    if test "x$enable_gc" != "xno"; then
       LSP_CMAKE_OPTIONS=$LSP_CMAKE_OPTIONS"-DLSPCPP_SUPPORT_BOEHM_GC=ON "
-      if test "x$with_vcpkg" = "xno"; then
-          LSP_CMAKE_OPTIONS=$LSP_CMAKE_OPTIONS"-DLSPCPP_GC_DOWNLOADED_ROOT=../\$(GC) "
-      fi
    else
       LSP_CMAKE_OPTIONS=$LSP_CMAKE_OPTIONS"-DLSPCPP_SUPPORT_BOEHM_GC=OFF "
    fi
@@ -9164,8 +9161,8 @@ if test "x$enable_gc" != "xno" ; then
 
 printf "%s\n" "#define USEGC 1" >>confdefs.h
 
-    GCLIB="\$(GC)/.libs/libgc.a"
-    INCL=$INCL" -I\$(GC)/include"
+    GCLIB="-lgc"
+    # INCL=$INCL" -I\$(GC)/include"
     { printf "%s\n" "$as_me:${as_lineno-$LINENO}: $GCNAME is enabled" >&5
 printf "%s\n" "$as_me: $GCNAME is enabled" >&6;}
   fi
--- a/memory.h
+++ b/memory.h
@@ -27,7 +27,7 @@
 
 #ifdef GC_DEBUG
 extern "C" {
-#include <gc_backptr.h>
+#include <gc/gc_backptr.h>
 }
 #endif
 
@@ -40,8 +40,8 @@ void* asy_malloc_atomic(size_t n);
 #define GC_MALLOC(sz) asy_malloc(sz)
 #define GC_MALLOC_ATOMIC(sz) asy_malloc_atomic(sz)
 
-#include <gc_allocator.h>
-#include <gc_cpp.h>
+#include <gc/gc_allocator.h>
+#include <gc/gc_cpp.h>
 
 #define gc_allocator gc_allocator_ignore_off_page
 
--- a/Makefile.in
+++ b/Makefile.in
@@ -176,11 +176,6 @@ $(LSP_ROOT)/liblspcpp.a:
 
 all:	asy sty man faq asy-keywords.el
 
-$(GCLIB):
-	cd $(GC) && ln -sf ../libatomic_ops . && \
-		./configure CC="$(CC)" CXX="$(CXX)" $(GCOPTIONS)
-	$(MAKE) -C $(GC) check
-
 sty: doc/version.texi
 	cd doc && $(MAKE) asy-latex.pdf
 
