--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -280,14 +280,6 @@
     DIRECTORY "${PROJECT_SOURCE_DIR}/data/platform/linux/share/"
     DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}
   )
-  install(
-    CODE
-    "
-      find_program(update-mime-database_cmd NAMES update-mime-database REQUIRED)
-      mark_as_advanced(update-mime-database_cmd)
-      execute_process(COMMAND \"\${update-mime-database_cmd}\" \"\${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_DATAROOTDIR}/mime\")
-    "
-  )
 endif()
 
 if(WIN32)
