diff --git a/ell/signal.h b/ell/signal.h
index 0d4fdc8..98cf0fa 100644
--- ell/signal.h
+++ ell/signal.h
@@ -36,7 +36,7 @@ typedef void (*l_signal_notify_cb_t) (struct l_signal *signal,
 					uint32_t signo, void *user_data);
 typedef void (*l_signal_destroy_cb_t) (void *user_data);
 
-struct l_signal *l_signal_create(const __sigset_t *mask,
+struct l_signal *l_signal_create(const sigset_t *mask,
 			l_signal_notify_cb_t callback,
 			void *user_data, l_signal_destroy_cb_t destroy);
 void l_signal_remove(struct l_signal *signal);

