--- a/GNUmakefile
+++ b/GNUmakefile
@@ -35,32 +35,15 @@ OBJS=		charconv.o main.o
 
 PKG_CONFIG?=	pkg-config
 
-KDE_VER:=	3
-ifeq (${KDE_VER},3)
-KDE_INCS?=	-I/usr/include/qt3 -I/usr/include/kde
-SRCS+=		kwif3.cc
-OBJS+=		kwif3.o
-LDADD+=		-lkwalletclient -lkdecore -lqt-mt
-else
-ifeq (${KDE_VER},4)
-KDE_INCS?=	-I/usr/include/qt4 -I/usr/include/qt4/QtCore
-SRCS+=		kwif4.cc
-OBJS+=		kwif4.o
-LDADD+=		-lkdeui -lkdecore -lQtCore
-else
-ifeq (${KDE_VER},5)
-KDE_INCS?=	$(shell ${PKG_CONFIG} --cflags Qt5Gui) \
-		-I/usr/include/KF5/KI18n \
-		-I/usr/include/KF5/KCoreAddons \
-		-I/usr/include/KF5/KWallet
+KDE_VER:=	6
+KDE_INCS?=	$(shell ${PKG_CONFIG} --cflags Qt6Widgets) \
+		-I/usr/include/KF6/KI18n \
+		-I/usr/include/KF6/KCoreAddons \
+		-I/usr/include/KF6/KWallet
 SRCS+=		kwif5.cc
 OBJS+=		kwif5.o
-LDADD+=		-lKF5CoreAddons -lKF5Wallet -lKF5I18n -lQt5Core -lQt5Widgets
-else
-$(error unknown KDE_VER)
-endif
-endif
-endif
+LDADD+=		-lKF6CoreAddons -lKF6Wallet -lKF6I18n -lQt6Core -lQt6Widgets
+LDADD+=		$(shell ${PKG_CONFIG} --libs Qt6Widgets)
 
 CPPFLAGS+=	${KDE_INCS} -fPIC -D_GNU_SOURCE
 LDFLAGS+=	-fPIC
--- a/kwif5.cc
+++ b/kwif5.cc
@@ -25,7 +25,6 @@
  */
 
 #include <qstring.h>
-#include <QtWidgets/qdesktopwidget.h>
 #include <QtWidgets/qapplication.h>
 #include <klocalizedstring.h>
 #include <kaboutdata.h>
@@ -68,9 +67,7 @@ kw_io(const char *fld, const char *ent,
 	KAboutData::setApplicationData(aboutData);
 
 	localwallet = KWallet::Wallet::LocalWallet();
-	wallet = KWallet::Wallet::openWallet(localwallet,
-	    /* https://lists.qt-project.org/pipermail/interest/2020-November/035993.html */
-	    QApplication::desktop()->screen()->winId());
+	wallet = KWallet::Wallet::openWallet(localwallet, 0);
 
 	if (!wallet) {
 		rv = KWE_NOWALLET;
