--- include/h2o/socket.h.orig	2015-01-20 07:45:22.434850581 +0000
+++ include/h2o/socket.h	2015-01-20 07:45:41.732177170 +0000
@@ -39,16 +39,8 @@
 #endif
 #endif
 
-#if OPENSSL_VERSION_NUMBER >= 0x10002000L
-#define H2O_USE_ALPN 1
-#define H2O_USE_NPN 1
-#elif OPENSSL_VERSION_NUMBER >= 0x10001000L
 #define H2O_USE_ALPN 0
 #define H2O_USE_NPN 1
-#else
-#define H2O_USE_ALPN 0
-#define H2O_USE_NPN 0
-#endif
 
 #define H2O_SOCKET_INITIAL_INPUT_BUFFER_SIZE 4096
 
