diff --git a/src/window.cpp b/src/window.cpp
index 66cf765..fc8f44c 100644
--- src/window.cpp
+++ src/window.cpp
@@ -204,6 +204,9 @@ void initGL (glm::ivec4 &_viewport, bool _headless, bool _alwaysOnTop) {
             glfwWindowHint(GLFW_FLOATING, GL_TRUE);
         }
 
+        // Make it float
+        glfwWindowHint(GLFW_RESIZABLE, GLFW_FALSE);
+
         window = glfwCreateWindow(_viewport.z, _viewport.w, appTitle.c_str(), NULL, NULL);
 
         if(!window) {

