--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -125,7 +125,7 @@ else()
         set(CPACK_GENERATOR "ZIP")
 
     else()
-        target_link_libraries(glslViewer PRIVATE pthread dl lo_static)
+        target_link_libraries(glslViewer PRIVATE pthread dl lo)
         install(TARGETS glslViewer DESTINATION ${CMAKE_INSTALL_BINDIR})
         
         if (NOT APPLE)
--- a/deps/CMakeLists.txt
+++ b/deps/CMakeLists.txt
@@ -1,6 +1,2 @@
 ## Ada
 add_subdirectory(vera)
-
-if (NOT EMSCRIPTEN) 
-    add_subdirectory(liblo/cmake)
-endif()
\ No newline at end of file
