--- a/meson.build
+++ b/meson.build
@@ -20,7 +20,10 @@ if (compiler.get_id() == 'gcc' and build
 else
   thread_dep = dependency('', required:false)
 endif
-libicu_dep = dependency('icu-i18n', static:static_deps)
+libicu_dep = [
+  dependency('icu-i18n', static:static_deps),
+  dependency('icu-uc', static:static_deps)
+]
 pugixml_dep = dependency('pugixml', static:static_deps)
 libcurl_dep = dependency('libcurl', static:static_deps)
 microhttpd_dep = dependency('libmicrohttpd', static:static_deps)
