--- lib/intel_batchbuffer.c.orig	2015-03-15 09:30:52.374500012 +0100
+++ lib/intel_batchbuffer.c	2015-03-15 09:31:04.784415162 +0100
@@ -548,7 +548,7 @@ fill_object(struct drm_i915_gem_exec_obj
 	memset(obj, 0, sizeof(*obj));
 	obj->handle = gem_handle;
 	obj->relocation_count = count;
-	obj->relocs_ptr = (uint64_t)relocs;
+	obj->relocs_ptr = relocs;
 }
 
 static void exec_blit(int fd,
@@ -557,7 +557,7 @@ static void exec_blit(int fd,
 {
 	struct drm_i915_gem_execbuffer2 exec;
 
-	exec.buffers_ptr = (uint64_t)objs;
+	exec.buffers_ptr = objs;
 	exec.buffer_count = count;
 	exec.batch_start_offset = 0;
 	exec.batch_len = batch_len * 4;
