--- lib/igt_x86.h.orig	2017-10-04 10:36:48.000000000 +0000
+++ lib/igt_x86.h	2017-10-07 08:15:28.264965837 +0000
@@ -40,19 +40,7 @@
 #define AVX	0x80
 #define AVX2	0x100
 
-#if defined(__x86_64__)
 unsigned igt_x86_features(void);
 char *igt_x86_features_to_string(unsigned features, char *line);
-#else
-static inline unsigned igt_x86_features(void)
-{
-	return 0;
-}
-static inline char *igt_x86_features_to_string(unsigned features, char *line)
-{
-	line[0] = 0;
-	return line;
-}
-#endif
 
 #endif /* IGT_X86_H */
