--- b/src/pari.cc	2024-05-06 13:08:05.000000000 -0300
+++ a/src/pari.cc	2024-05-18 16:57:06.688878820 -0300
@@ -734,7 +734,7 @@
   }
 
   static gen pari_exec(const string & s,GIAC_CONTEXT){
-    static ensure_pari_is_ready now;
+    ensure_pari_is_ready now;
     long av=avma;
     void * save_pari_stack_limit = PARI_stack_limit;
     PARI_stack_limit=0; // required since the stack changed
@@ -800,7 +800,7 @@
     lidnt(v,vars,false);
     vars.erase(vars.begin());
     bool parse_all=false;
-    static ensure_pari_is_ready now;
+    ensure_pari_is_ready now;
     long av=avma;
     cb_pari_err_recover=gp_err_recover;
     if (setjmp(env)){
