--- src/History.cc.orig	2017-10-02 10:23:46.000000000 +0200
+++ src/History.cc	2017-11-14 15:42:12.119849965 +0100
@@ -3828,7 +3828,7 @@
       // mode_s += "Time: ";
       // double t=double(clock());
       // mode_s += xcas::print_DOUBLE_(t/CLOCKS_PER_SEC);
-#ifdef HAVE_MALLOC_H //
+#if defined HAVE_MALLOC_H && defined __GLIBC__ //
       struct mallinfo mem=mallinfo();
       double memd=mem.arena+mem.hblkhd;
       mode_s +=xcas::print_DOUBLE_(memd/1048576);
