--- a/src/icas.cc	2021-03-02 04:52:40.000000000 -0300
+++ b/src/icas.cc	2022-03-30 10:00:39.154056068 -0300
@@ -1810,6 +1810,7 @@
     cerr << "// Setting tex log" << '\n';
     show_tex=true;
   }
+#ifdef HAVE_LIBMICROPYTHON
   if (getenv("GIAC_MICROPY")){
     cerr << "Micropython mode\n";
     python_compat(4 | python_compat(contextptr),contextptr);
@@ -1821,6 +1822,7 @@
       return 0;
     }
   }
+#endif
 #ifdef HAVE_LIBREADLINE
   if (ARGC==1){
     int taillemax=1000;
