--- a/tools/Makefile.am	2018-03-18 17:20:02.221259868 +0100
+++ b/tools/Makefile.am	2018-03-18 17:19:53.192259402 +0100
@@ -1,6 +1,9 @@
 # hacky, but ...
 
 CXX = $(CXX_FOR_BUILD)
+CFLAGS=$(CFLAGS_FOR_BUILD)
+CXXFLAGS=$(CXXFLAGS_FOR_BUILD)
+LDFLAGS=$(LDFLAGS_FOR_BUILD)
 
 AM_CPPFLAGS = \
 	$(dbus_CFLAGS) \
@@ -13,7 +13,7 @@
 	-Wall
 
 if CROSS_COMPILING
-libdbus_cxx_la = $(BUILD_LIBDBUS_CXX_DIR)/src/libdbus-c++-1.la
+libdbus_cxx_la = /usr/lib/libdbus-c++-1.so
 else
 libdbus_cxx_la = $(top_builddir)/src/libdbus-c++-1.la
 endif
