--- a/configure.ac
+++ b/configure.ac
@@ -453,11 +453,11 @@
 echo CFLAGS="$CFLAGS"
 echo MORECFLAGS="$MORECFLAGS"
 
+AS_IF([test "$allow_clz" -eq 1],
 dnl --check if lzcnt instruction is available and requested
 AC_MSG_CHECKING(if lzcnt instruction is available and requested)
 AC_RUN_IFELSE([AC_LANG_PROGRAM([],[[if (__builtin_cpu_supports("bmi")) return 0; else return 1;]])],
   lzsup=1,lzsup=0)
-AS_IF([test "$allow_clz" -eq 1],
       [have_hwlzcnt=$lzsup
 test -f /proc/cpuinfo && $EGREP -i "abm|bmi1" /proc/cpuinfo >/dev/null && have_hwlzcnt=1 
 test $have_hwlzcnt -eq 0 && sysctl machdep.cpu.features 2>/dev/null | $EGREP -i "bmi1|abm" >/dev/null && have_hwlzcnt=1
