diff --git a/runtime/Server.cpp b/runtime/Server.cpp
index 62f4a27f7..ba2783f0e 100644
--- a/runtime/Server.cpp
+++ b/runtime/Server.cpp
@@ -127,11 +127,11 @@ Server::Server(quint16 port, QString key, QString logFileName):
     add_to_path(pythonHome, venvPath.canonicalFilePath());
 #else
     // Build (and canonicalise) the virtual environment path
-    QFileInfo venvBinPath(app_dir + "/../venv/bin");
-    QFileInfo venvLibPath(app_dir + "/../venv/lib/python");
-    QFileInfo venvDynLibPath(app_dir + "/../venv/lib/python/lib-dynload");
-    QFileInfo venvSitePackagesPath(app_dir + "/../venv/lib/python/site-packages");
-    QFileInfo venvPath(app_dir + "/../venv");
+    QFileInfo venvBinPath("/usr/bin");
+    QFileInfo venvLibPath("/usr/lib/python3.8");
+    QFileInfo venvDynLibPath("/usr/lib/python3.8/lib-dynload");
+    QFileInfo venvSitePackagesPath("/usr/lib/python3.8/site-packages");
+    QFileInfo venvPath("/usr/lib/python3.8");
 
     // Prepend the bin directory to the path
     add_to_path(path_env, venvBinPath.canonicalFilePath(), true);
@@ -265,7 +265,7 @@ bool Server::Init()
 
     // Find the webapp
     QStringList paths;
-    paths.append("../web/"); // Linux source tree
+    paths.append("/usr/lib/pgadmin4/web/"); // Linux source tree
     paths.append("../../web/"); // Windows source tree
     paths.append("../../../../web/"); // Mac source tree (in a dev env)
 #ifdef Q_OS_MAC
