Provide correct path's in texmf.cnf
--- a/texmf-dist/web2c/texmf.cnf	2024-04-03 14:06:43.529075088 +0200
+++ b/texmf-dist/web2c/texmf.cnf	2024-04-03 14:08:25.001809298 +0200
@@ -59,7 +59,7 @@
 % SELFAUTOPARENT (its grandparent = /usr/local/texlive/YYYY), and
 % SELFAUTOGRANDPARENT (its great-grandparent = /usr/local/texlive).
 % Sorry for the off-by-one-generation names.
-TEXMFROOT = $SELFAUTOPARENT
+TEXMFROOT = /usr/share
 
 % The various texmf trees used by TeX Live, follow.
 % They must must all have the TDS directory structure (https://tug.org/tds).
@@ -72,7 +72,7 @@ TEXMFDIST = $TEXMFROOT/texmf-dist
 TEXMFMAIN = $TEXMFDIST
 
 % Local additions to the distribution trees.
-TEXMFLOCAL = $SELFAUTOGRANDPARENT/texmf-local
+TEXMFLOCAL = $TEXMFROOT/texmf-local
 
 % TEXMFSYSVAR, where *-sys store cached runtime data.
 TEXMFSYSVAR = $TEXMFROOT/texmf-var
