--- source4/dsdb/samdb/ldb_modules/count_attrs.c
+++ source4/dsdb/samdb/ldb_modules/count_attrs.c
@@ -30,6 +30,10 @@
 #include "lib/tdb_wrap/tdb_wrap.h"
 #include "system/filesys.h"
 
+#ifndef HAVE_GLIBC_COMPAR_FN_T
+typedef int (*__compar_fn_t)(const void *, const void *);
+#endif
+
 #define NULL_ATTRS "__null_attrs__"
 #define EMPTY_ATTRS "__empty_attrs__"
 #define UNKNOWN_ATTR "__unknown_attribute__"

--- source4/dsdb/samdb/ldb_modules/wscript
+++ source4/dsdb/samdb/ldb_modules/wscript
@@ -30,9 +30,10 @@
         if not conf.CONFIG_SET('ENABLE_GPGME'):
             conf.fatal("GPGME support not found. "
                        "Try installing libgpgme11-dev or gpgme-devel "
                        "and python-gpgme. "
                        "Otherwise, use --without-gpgme to build without "
                        "GPGME support or --without-ad-dc to build without "
                        "the Samba AD DC. "
                        "GPGME support is required for the GPG encrypted "
                        "password sync feature")
+        conf.CHECK_TYPE("__compar_fn_t", define="HAVE_GLIBC_COMPAR_FN_T")
