--- source3/lib/system.c.orig	2019-01-23 23:55:02.171524646 +0100
+++ source3/lib/system.c	2019-01-23 23:55:32.693250813 +0100
@@ -25,6 +25,7 @@
 #include "system/capability.h"
 #include "system/passwd.h"
 #include "system/filesys.h"
+#include <sys/sysmacros.h>

 #ifdef HAVE_SYS_PRCTL_H
 #include <sys/prctl.h>

--- source3/libsmb/clifile.c.orig	2019-01-24 00:00:41.142565919 +0100
+++ source3/libsmb/clifile.c	2019-01-24 00:00:28.148674834 +0100
@@ -26,6 +26,7 @@
 #include "libsmb/clirap.h"
 #include "trans2.h"
 #include "ntioctl.h"
+#include <sys/sysmacros.h>

 /***********************************************************
  Common function for pushing stings, used by smb_bytes_push_str()

--- lib/replace/replace.h.orig	2019-01-24 00:04:11.908799366 +0100
+++ lib/replace/replace.h	2019-01-24 00:04:32.329628199 +0100
@@ -40,6 +40,7 @@
 #include <stdlib.h>
 #include <stdarg.h>
 #include <errno.h>
+#include <sys/sysmacros.h>

 #if defined(_MSC_VER) || defined(__MINGW32__)
 #include "win32_replace.h"
