--- a/lib/replace/replace.h
+++ b/lib/replace/replace.h
@@ -168,10 +168,6 @@
 #include <bsd/unistd.h>
 #endif
 
-#ifdef HAVE_UNISTD_H
-#include <unistd.h>
-#endif
-
 #ifdef HAVE_STRING_H
 #include <string.h>
 #endif
--- a/lib/replace/system/network.h
+++ b/lib/replace/system/network.h
@@ -31,10 +31,6 @@
 #error "AC_LIBREPLACE_NETWORK_CHECKS missing in configure"
 #endif
 
-#ifdef HAVE_UNISTD_H
-#include <unistd.h>
-#endif
-
 #ifdef HAVE_SYS_SOCKET_H
 #include <sys/socket.h>
 #endif
--- a/source3/rpc_server/mdssvc/mdssvc.c
+++ b/source3/rpc_server/mdssvc/mdssvc.c
@@ -18,6 +18,8 @@
    along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
 
+#include <unistd.h>
+
 #include "includes.h"
 #include "librpc/gen_ndr/auth.h"
 #include "dbwrap/dbwrap.h"
