From c7f77ad985b277f044484ecdb57b2bdd6eb7e772 Mon Sep 17 00:00:00 2001
From: Nathan Owens <ndowens@artixlinux.org>
Date: Wed, 30 Dec 2020 17:05:34 -0600
Subject: [PATCH] fix cross build

---
 configure.in | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git configure.in configure.in
index 217ac83..ae43832 100644
--- configure.in
+++ configure.in
@@ -838,7 +838,7 @@ fi
 if test "$ac_cv_type_complex_double" = yes; then
   AC_CACHE_CHECK([whether csqrt is usable],
     guile_cv_use_csqrt,
-    [AC_RUN_IFELSE([AC_LANG_SOURCE([[
+    [AC_COMPILE_IFELSE([AC_LANG_SOURCE([[
 #include <complex.h>
 /* "volatile" is meant to prevent gcc from calculating the sqrt as a
    constant, we want to test libc. */
@@ -1035,7 +1035,7 @@ AC_CHECK_MEMBERS([struct sockaddr_in6.sin6_len],,,
 AC_MSG_CHECKING(whether localtime caches TZ)
 AC_CACHE_VAL(guile_cv_localtime_cache,
 [if test x$ac_cv_func_tzset = xyes; then
-AC_RUN_IFELSE([AC_LANG_SOURCE([[#include <time.h>
+AC_COMPILE_IFELSE([AC_LANG_SOURCE([[#include <time.h>
 #if STDC_HEADERS
 # include <stdlib.h>
 #endif
@@ -1196,7 +1196,7 @@ GUILE_STRUCT_UTIMBUF
 #--------------------------------------------------------------------
 
 SCM_I_GSC_STACK_GROWS_UP=0
-AC_RUN_IFELSE([AC_LANG_SOURCE(
+AC_COMPILE_IFELSE([AC_LANG_SOURCE(
 [AC_INCLUDES_DEFAULT
 int
 find_stack_direction ()
@@ -1374,7 +1374,7 @@ if test "$with_threads" = pthreads; then
 AC_MSG_CHECKING(whether pthread_attr_getstack works for the main thread)
 old_CFLAGS="$CFLAGS"
 CFLAGS="$PTHREAD_CFLAGS $CFLAGS"
-AC_RUN_IFELSE([AC_LANG_SOURCE([[
+AC_COMPILE_IFELSE([AC_LANG_SOURCE([[
 #if HAVE_PTHREAD_ATTR_GETSTACK
 #include <pthread.h>
 
-- 
2.30.0

