--- a/libguile/fports.c	2010-12-13 18:24:40.000000000 +0100
+++ b/libguile/fports.c	2017-10-25 21:51:07.234942515 +0200
@@ -78,8 +78,8 @@
 #define OFF_T_MAX  LONG_MAX
 #define OFF_T_MIN  LONG_MIN
 #elif SIZEOF_OFF_T == SIZEOF_LONG_LONG
-#define OFF_T_MAX  LONG_LONG_MAX
-#define OFF_T_MIN  LONG_LONG_MIN
+#define OFF_T_MAX  LLONG_MAX
+#define OFF_T_MIN  LLONG_MIN
 #else
 #error Oops, unknown OFF_T size
 #endif
