From 3ef23f0db8c81fd6091a7303214901ba0d68d0c0 Mon Sep 17 00:00:00 2001
From: Daniel Santana <daniel@santana.tech>
Date: Fri, 4 Jan 2019 21:57:52 -0200
Subject: [PATCH] Use SONAMEs when loading GLX libs dinamically

---
 glx_wcb.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/glx_wcb.c b/glx_wcb.c
index 4827152..d0afc0b 100644
--- a/glx_wcb.c
+++ b/glx_wcb.c
@@ -193,8 +193,8 @@ static void init(void) {
     maximized   = false;
     transparent = false;
 
-    void* hgl  = dlopen("libGL.so", RTLD_LAZY);
-    void* hglx = dlopen("libGLX.so", RTLD_LAZY);
+    void* hgl  = dlopen("libGL.so.1", RTLD_LAZY);
+    void* hglx = dlopen("libGLX.so.0", RTLD_LAZY);
 
     if (!hgl && !hglx) {
         fprintf(stderr, "Failed to load GLX functions (libGL and libGLX do not exist!)\n");
-- 
2.20.1

