From 4705699bb680e574469b11a037acfc83f091d232 Mon Sep 17 00:00:00 2001
From: Daniel Santana <daniel@santana.tech>
Date: Fri, 4 Jan 2019 22:23:52 -0200
Subject: [PATCH] Fix musl build

---
 glava.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/glava.c b/glava.c
index dcb650f..758c8f0 100644
--- a/glava.c
+++ b/glava.c
@@ -69,6 +69,10 @@
 #error "Unsupported target system"
 #endif
 
+#ifndef ACCESSPERMS
+#define ACCESSPERMS (S_IRWXU|S_IRWXG|S_IRWXO) /* 0777 */
+#endif
+
 /* Copy installed shaders/configuration from the installed location
    (usually /etc/xdg). Modules (folders) will be linked instead of
    copied. */
-- 
2.20.1

