--- a/nmap_error.cc
+++ b/nmap_error.cc
@@ -145,6 +145,7 @@
 #  include <time.h>
 # endif
 #endif
+#include <ctime>
 
 extern NmapOps o;
 
--- a/nping/EchoServer.cc
+++ b/nping/EchoServer.cc
@@ -127,6 +127,7 @@
  *                                                                         *
  ***************************************************************************/
 
+#include <ctime>
 #include "nping.h"
 #include "EchoServer.h"
 #include "EchoHeader.h"
