diff --git twmnc/twmnc.pro twmnc/twmnc.pro
index 7f61d74..d2db8b5 100644
--- twmnc/twmnc.pro
+++ twmnc/twmnc.pro
@@ -9,7 +9,7 @@ DEPENDPATH += .
 INCLUDEPATH += .
 DESTDIR = ../bin/
 LIBS += -lboost_program_options -lboost_system -lpthread
-target.path += /usr/local/bin
+target.path += /usr/bin
 INSTALLS += target
 # Input
 SOURCES += main.cpp
diff --git twmnd/main.cpp twmnd/main.cpp
index c39525d..1637293 100644
--- twmnd/main.cpp
+++ twmnd/main.cpp
@@ -5,7 +5,6 @@
 #include <QLibraryInfo>
 #include <QTranslator>
 #include "widget.h"
-#include "xcb/xcb.h"
 #include "dbusinterface.h"
 #include <iostream>
 
diff --git twmnd/twmnd.pro twmnd/twmnd.pro
index 79624a5..1308006 100644
--- twmnd/twmnd.pro
+++ twmnd/twmnd.pro
@@ -10,10 +10,9 @@ TARGET = twmnd
 TEMPLATE = app
 CONFIG += debug
 DESTDIR = ../bin/
-LIBS += -lX11 -lXext
 QMAKE_CXXFLAGS += -Wall -Werror -pedantic -Wno-long-long
 
-target.path+=/usr/local/bin
+target.path+=/usr/bin
 INSTALLS += target
 
 SOURCES += main.cpp\
diff --git twmnd/widget.cpp twmnd/widget.cpp
index f768165..a184fac 100644
--- twmnd/widget.cpp
+++ twmnd/widget.cpp
@@ -18,7 +18,6 @@
 #include <QIcon>
 #include <QWheelEvent>
 #include <QCursor>
-#include <QtX11Extras/QX11Info>
 #include "settings.h"
 
 Widget::Widget(const char* wname) : m_settings(wname)//, m_shortcutGrabber(this, m_settings)
@@ -138,7 +137,7 @@ void Widget::processMessageQueue()
     Message& m = m_messageQueue.front();
     loadDefaults();
     if (m.data["aot"]->toBool()) {
-        setWindowFlags(windowFlags() | Qt::WindowStaysOnTopHint | Qt::X11BypassWindowManagerHint);
+        setWindowFlags(windowFlags() | Qt::WindowStaysOnTopHint | Qt::BypassWindowManagerHint);
         raise();
     }
     setupFont();
