--- a/flatpak-integrator/plugin.cpp
+++ b/flatpak-integrator/plugin.cpp
@@ -143,7 +143,14 @@ private:
     {
         constexpr auto maxBufferSize = 1024;
         thread_local std::array<char, maxBufferSize> buffer;
+#ifdef STRERROR_R_CHAR_P
         return strerror_r(error, buffer.data(), buffer.size());
+#else
+        // Won't be changed by strerror_r but not const so compiler doesn't throw an error
+        static char unknown[] = "unknown error";
+
+        return strerror_r(error, buffer.data(), buffer.size()) ? unknown : buffer.data();
+#endif
     }
 
     int openNoSymlinks(const std::filesystem::path &path, int flags, mode_t mode = 0)
