--- compiz-manager.orig	2007-10-15 21:10:06.000000000 +0200
+++ compiz-manager	2007-10-18 22:17:18.000000000 +0200
@@ -27,10 +27,10 @@
 # 
 
 
-COMPIZ_BIN_PATH="/usr/local/bin/" # For window decorators and compiz
-PLUGIN_PATH="/usr/local/lib/compiz/" 
+COMPIZ_BIN_PATH="/usr/bin/" # For window decorators and compiz
+PLUGIN_PATH="/usr/lib/compiz/" 
 GLXINFO="/usr/bin/glxinfo"
-KWIN="/usr/bin/kwin"
+KWIN="/usr/bin/kwin"
 METACITY="/usr/bin/metacity"
 COMPIZ_NAME="compiz" # Final name for compiz (compiz.real) 
 

