--- a/src/Gui/Spinner.cpp	2016-06-15 09:17:23.000000000 +0200
+++ b/src/Gui/Spinner.cpp	2020-09-02 13:51:52.962353077 +0200
@@ -26,6 +26,7 @@
 
 #include <QFontMetricsF>
 #include <QPainter>
+#include <QPainterPath>
 #include <QTimer>
 #include <QTimerEvent>
 #include <qmath.h>
--- a/src/Gui/Window.cpp	2016-06-15 09:17:23.000000000 +0200
+++ b/src/Gui/Window.cpp	2020-09-02 13:52:19.050354423 +0200
@@ -32,6 +32,7 @@
 #include <QKeyEvent>
 #include <QMenuBar>
 #include <QMessageBox>
+#include <QPainterPath>
 #include <QProgressBar>
 #include <QScrollBar>
 #include <QSplitter>
