--- src/gui/editors/notation/NotePixmapFactory.cpp	2020-06-04 04:06:44.000000000 +0200
+++ src/gui/editors/notation/NotePixmapFactory.cpp	2020-09-01 19:07:51.373873820 +0200
@@ -59,6 +59,7 @@
 #include <QFontMetrics>
 #include <QImage>
 #include <QPainter>
+#include <QPainterPath>
 #include <QPen>
 #include <QPixmap>
 #include <QPolygon>
--- src/gui/general/ThornStyle.cpp	2020-06-04 04:06:46.000000000 +0200
+++ src/gui/general/ThornStyle.cpp	2020-09-01 19:07:58.088874166 +0200
@@ -34,6 +34,7 @@
 #include <QLabel>
 #include <QLayout>
 #include <QPainter>
+#include <QPainterPath>
 #include <QRadioButton>
 #include <QStyleFactory>
 #include <QStyleOption>
--- src/gui/widgets/Panner.cpp	2020-06-04 04:06:46.000000000 +0200
+++ src/gui/widgets/Panner.cpp	2020-09-01 19:07:28.285872629 +0200
@@ -24,6 +24,7 @@
 #include "misc/Debug.h"
 #include "base/Profiler.h"
 
+#include <QPainterPath>
 #include <QPolygon>
 #include <QMouseEvent>
 
