From 17ad9c9262927a3897352162de8fe2222a2d0b24 Mon Sep 17 00:00:00 2001
From: Artem Polishchuk <ego.cordatus@gmail.com>
Date: Mon, 23 Mar 2020 12:06:29 +0200
Subject: [PATCH] Add 'include <QDBusPendingCall>'

Fix #62
---
 src/helper/helpersysctl.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/helper/helpersysctl.cpp b/src/helper/helpersysctl.cpp
index cbdc8a9..bdabd0b 100644
--- a/src/helper/helpersysctl.cpp
+++ b/src/helper/helpersysctl.cpp
@@ -24,6 +24,7 @@
 #include <QByteArray>
 #include <QDBusConnection>
 #include <QDBusInterface>
+#include <QDBusPendingCall>
 #include <QStringLiteral>
 #include <stdexcept>
 #include <utility>
-- 
2.24.1

