commit 57a0a05d9f664fca97c8531385a361fd4c68d861
Author: q66 <daniel@octaforge.org>
Date:   Sat Jan 4 23:25:33 2020 +0100

    fix build with recent vala

diff --git src/ui/dialogs/SettingsDialog/SettingsDialogTab.vala src/ui/dialogs/SettingsDialog/SettingsDialogTab.vala
index 369dd34..a952db9 100644
--- src/ui/dialogs/SettingsDialog/SettingsDialogTab.vala
+++ src/ui/dialogs/SettingsDialog/SettingsDialogTab.vala
@@ -28,11 +28,6 @@ namespace GameHub.UI.Dialogs.SettingsDialog
 	{
 		public SettingsDialog dialog { construct; protected get; }
 
-		public SettingsDialogTab(SettingsDialog dlg)
-		{
-			Object(orientation: Orientation.VERTICAL, dialog: dlg);
-		}
-
 		construct
 		{
 			margin_start = margin_end = 8;
diff --git src/ui/views/GameDetailsView/GameDetailsBlock.vala src/ui/views/GameDetailsView/GameDetailsBlock.vala
index 0a03c39..06bf4a8 100644
--- src/ui/views/GameDetailsView/GameDetailsBlock.vala
+++ src/ui/views/GameDetailsView/GameDetailsBlock.vala
@@ -30,11 +30,6 @@ namespace GameHub.UI.Views.GameDetailsView
 
 		public bool is_dialog { get; construct; }
 
-		public GameDetailsBlock(Game game, bool is_dialog)
-		{
-			Object(game: game, orientation: Orientation.VERTICAL, is_dialog: is_dialog);
-		}
-
 		public abstract bool supports_game { get; }
 
 		protected void add_info_label(string title, string? text, bool multiline=true, bool markup=false, Container? parent=null)
diff --git src/utils/downloader/Downloader.vala src/utils/downloader/Downloader.vala
index 7005965..2efcaf9 100644
--- src/utils/downloader/Downloader.vala
+++ src/utils/downloader/Downloader.vala
@@ -104,7 +104,7 @@ namespace GameHub.Utils.Downloader
 			set { _status = value; status_change(_status); }
 		}
 
-		public Download(File remote, File local, File local_tmp)
+		protected Download(File remote, File local, File local_tmp)
 		{
 			this.remote = remote;
 			this.local = local;
@@ -116,7 +116,7 @@ namespace GameHub.Utils.Downloader
 
 	public abstract class PausableDownload: Download
 	{
-		public PausableDownload(File remote, File local, File local_tmp)
+		protected PausableDownload(File remote, File local, File local_tmp)
 		{
 			base(remote, local, local_tmp);
 		}
