--- src/common/Exception.cxx
+++ src/common/Exception.cxx
@@ -39,8 +39,13 @@ SystemException::SystemException(const std::string &err) :
 
 	char msg[256];
 
+#if defined(__GLIBC__)
 	char *m = ::strerror_r(m_errno, msg, 256);
 	ss << " (\"" << m << "\", errno = " << m_errno << ")";
+#else
+	::strerror_r(m_errno, msg, 256);
+	ss << " (\"" << msg << "\", errno = " << m_errno << ")";
+#endif
 	m_error.append( ss.str() );
 }
 
--- src/fuse/Entry.hxx
+++ src/fuse/Entry.hxx
@@ -1,6 +1,8 @@
 #ifndef XWMFS_ENTRY_HXX
 #define XWMFS_ENTRY_HXX
 
+#include <sys/types.h>
+
 // C++
 #include <atomic>

