http://sourceware.org/bugzilla/show_bug.cgi?id=12403

--- nptl/sysdeps/unix/sysv/linux/x86_64/pthread_rwlock_timedrdlock.S	
+++ nptl/sysdeps/unix/sysv/linux/x86_64/pthread_rwlock_timedrdlock.S	
@@ -210,7 +210,7 @@ pthread_rwlock_timedrdlock:
 	cfi_restore(%r12)
 	retq
 
-#ifdef __ASSUME_PRIVATE_FUTEX
+#ifdef __ASSUME_FUTEX_CLOCK_REALTIME
 	cfi_adjust_cfa_offset(16)
 	cfi_rel_offset(%r12, 8)
 	cfi_rel_offset(%r13, 0)
--- nptl/sysdeps/unix/sysv/linux/x86_64/pthread_rwlock_timedwrlock.S	
+++ nptl/sysdeps/unix/sysv/linux/x86_64/pthread_rwlock_timedwrlock.S	
@@ -192,7 +192,7 @@ pthread_rwlock_timedwrlock:
 
 7:	movq	%rdx, %rax
 
-#ifndef __ASSUME_PRIVATE_FUTEX
+#ifndef __ASSUME_FUTEX_CLOCK_REALTIME
 	addq	$16, %rsp
 	cfi_adjust_cfa_offset(-16)
 	popq	%r14
@@ -207,7 +207,7 @@ pthread_rwlock_timedwrlock:
 	cfi_restore(%r12)
 	retq
 
-#ifdef __ASSUME_PRIVATE_FUTEX
+#ifdef __ASSUME_FUTEX_CLOCK_REALTIME
 	cfi_adjust_cfa_offset(16)
 	cfi_rel_offset(%r12, 8)
 	cfi_rel_offset(%r13, 0)
