From 9a168cc23962ce9fa106dc8a40407d381a3db403 Mon Sep 17 00:00:00 2001
From: Christian Persch <chpe@src.gnome.org>
Date: Fri, 14 Jan 2022 11:33:32 +0100
Subject: [PATCH] build: Fix for newer meson

---
 data/meson.build | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/data/meson.build b/data/meson.build
index 79d97e5b..e0bcbb91 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -18,7 +18,6 @@
 metainfodir = gt_datadir / 'metainfo'
 
 i18n.merge_file(
-  gt_dns_name + '.metainfo.xml',
   input: gt_dns_name + '.metainfo.xml.in',
   output: '@BASENAME@',
   po_dir: po_dir,
@@ -27,7 +26,6 @@ i18n.merge_file(
 )
 
 i18n.merge_file(
-  gt_dns_name + '.Nautilus.metainfo.xml',
   input: gt_dns_name + '.Nautilus.metainfo.xml.in',
   output: '@BASENAME@',
   po_dir: po_dir,
@@ -40,7 +38,6 @@ i18n.merge_file(
 desktopdatadir = gt_datadir / 'applications'
 
 i18n.merge_file(
-  gt_dns_name + '.desktop',
   input: gt_dns_name + '.desktop.in',
   output: '@BASENAME@',
   type: 'desktop',
@@ -57,4 +54,4 @@ meson.add_install_script(
 
 # Subdirs
 
-subdir('icons')
\ No newline at end of file
+subdir('icons')
-- 
GitLab

