--- Makefile.am
+++ Makefile.am
@@ -145,10 +145,10 @@ kcapi_hasher_links = sha1sum sha224sum sha256sum sha384sum sha512sum \
 
 install-exec-hook:
 	(cd $(DESTDIR)$(bindir) && \
-	($(foreach link, $(kcapi_hasher_links), $(LN) -f kcapi-hasher $(link);)))
+	($(foreach link, $(kcapi_hasher_links), $(LN) -f kcapi-hasher $(link)-kcapi;)))
 if HAVE_OPENSSL
 	(cd $(DESTDIR)$(bindir) && \
-	($(foreach link, $(kcapi_hasher_links), $(OPENSSL) sha256 -hmac orboDeJITITejsirpADONivirpUkvarP $(link) > $(CHECK_PREFIX)$(link).$(CHECK_SUFFIX);)))
+	($(foreach link, $(kcapi_hasher_links), $(OPENSSL) sha256 -hmac orboDeJITITejsirpADONivirpUkvarP $(link)-kcapi > $(CHECK_PREFIX)$(link).$(CHECK_SUFFIX);)))
 endif
 endif
 
