--- Makefile.in.orig	2008-10-11 06:25:26.000000000 +0200
+++ Makefile.in	2008-10-11 06:25:36.000000000 +0200
@@ -280,7 +280,7 @@ EXTRA_DIST = \
 
 CLEANFILES = freedesktop.org.xml po/stamp-it
 DISTCLEANFILES = intltool-extract intltool-merge intltool-update
-pkgconfigdir = $(datadir)/pkgconfig
+pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = shared-mime-info.pc
 all: config.h
 	$(MAKE) $(AM_MAKEFLAGS) all-recursive
