Index: libcotp-1.2.2/CMakeLists.txt
===================================================================
--- libcotp-1.2.2.orig/CMakeLists.txt
+++ libcotp-1.2.2/CMakeLists.txt
@@ -25,8 +25,7 @@ set(CMAKE_C_STANDARD 11)
 set(COTP_HEADERS src/cotp.h)
 set(SOURCE_FILES src/otp.c)
 
-set(CMAKE_C_FLAGS "-Wall -Wextra -O3 -Wno-format-truncation -fstack-protector-strong -fPIC")
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=3")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wextra -Wno-format-truncation -fstack-protector-strong -fPIC")
 
 add_library(cotp SHARED ${SOURCE_FILES})
 
