--- a/configure.ac	2021-01-18 20:39:16.546501531 +0100
+++ b/configure.ac	2021-01-18 20:57:43.667587518 +0100
@@ -142,7 +142,8 @@
   CFLAGS="$CFLAGS -Wall -Wextra -Wno-unused-parameter -std=gnu99 -pipe -g"
   CXXFLAGS="$CXXFLAGS -Wall -Wextra -Wno-unused-parameter -fno-strict-aliasing -pipe -g"
   AC_CXX_FLAG([-std=c++11])
-  AC_CXX_FLAG([-Wno-c++11-narrowing])
+  AC_CXX_FLAG([-Wno-narrowing])
+  AC_CXX_FLAG([-Wno-deprecated-copy])
   AC_C_FLAG([-Wno-unused-local-typedefs])
   AC_CXX_FLAG([-Wno-unused-local-typedefs])
 
