Subject: vendor/luajit is removed in the Debian package
Author: Sebastian Reichel <sre@debian.org>
Forwarded: not-needed
Last-Update: 2014-08-06
Upstream: Yes
Reason: Use system luajit

--- a/header.mk
+++ b/header.mk
@@ -13,7 +13,6 @@
 	tests \
 	tools \
 	vendor/luabins \
-	vendor/luajit \
 	vendor/universalchardet
 
 subdirs := $(addprefix $(TOP),$(addsuffix /Makefile,$(subdirs)))
--- a/Makefile.inc.in
+++ b/Makefile.inc.in
@@ -78,7 +78,7 @@
 CFLAGS_ICU         = @ICU_I18N_CFLAGS@
 CFLAGS_LIBASS      = @LIBASS_CFLAGS@
 CFLAGS_LIBPULSE    = @LIBPULSE_CFLAGS@
-CFLAGS_LUA         = -I$(TOP)vendor/luajit/include
+CFLAGS_LUA         = @LUAJIT_CFLAGS@
 CFLAGS_OPENAL      = @OPENAL_CFLAGS@
 CFLAGS_OSS         = @OSS_CFLAGS@
 CFLAGS_PORTAUDIO   = @PORTAUDIO_CFLAGS@
@@ -96,7 +96,7 @@
 LIBS_ICU           = @ICU_UC_LIBS@ @ICU_I18N_LIBS@
 LIBS_LIBASS        = @LIBASS_LIBS@
 LIBS_LIBPULSE      = @LIBPULSE_LIBS@
-LIBS_LUA           = $(TOP)vendor/luajit/src/libluajit.a
+LIBS_LUA           = @LUAJIT_LIBS@
 LIBS_OPENAL        = @OPENAL_LIBS@
 LIBS_PORTAUDIO     = @PORTAUDIO_LIBS@
 LIBS_PTHREAD       = @PTHREAD_LIBS@
--- a/tools/Makefile
+++ b/tools/Makefile
@@ -12,6 +12,4 @@
 
 PROGRAM += $(d)repack-thes-dict
 
-$(TOP)tools/respack.lua: $(TOP)vendor/luajit/src/host/minilua
-
 include $(TOP)Makefile.target
--- a/tools/respack.lua
+++ b/tools/respack.lua
@@ -44,7 +44,7 @@ out_cpp:write('#include "libresrc.h"\n')
 for line in manifest:lines() do
   if line:find('.') then
     local file = try_open(path..line, 'rb')
-    local id = line:gsub('^.*/', ''):gsub('\.[a-z]+$', '')
+    local id = line:gsub('^.*/', ''):gsub('%.[a-z]+$', '')
     out_cpp:write("const unsigned char " .. id .. "[] = {")
 
     local len = 0
--- a/src/libresrc/Makefile
+++ b/src/libresrc/Makefile
@@ -10,7 +10,7 @@
 
 $(resrc_OBJ): $(d)default_config.h $(d)bitmap.h $(d)default_config.cpp $(d)bitmap.cpp
 
-RESPACK := cd $(TOP)src/libresrc; $(TOP)vendor/luajit/src/host/minilua $(TOP)tools/respack.lua
+RESPACK := cd $(TOP)src/libresrc; $(TOP)tools/respack.lua
 
 $(d)bitmap.cpp: $(d)bitmap.h
 $(d)default_config.cpp: $(d)default_config.h
