--- a/DevIL/src-IL/CMakeLists.txt	2017-01-02 01:03:56.000000000 +0100
+++ b/DevIL/src-IL/CMakeLists.txt	2017-01-02 15:48:51.400299099 +0100
@@ -58,6 +58,7 @@
 
 if(BUILD_SHARED_LIBS)
     add_library(IL SHARED ${DevIL_SRCS} ${DevIL_INC} ${DevIL_RSRC} ${DevIL_TXT})
+    set_target_properties(IL PROPERTIES SOVERSION 1)
 else(BUILD_SHARED_LIBS)
     add_library(IL ${DevIL_SRCS} ${DevIL_INC} ${DevIL_RSRC} ${DevIL_TXT})
 endif(BUILD_SHARED_LIBS)
--- a/DevIL/src-ILU/CMakeLists.txt	2017-01-02 01:03:56.000000000 +0100
+++ b/DevIL/src-ILU/CMakeLists.txt	2017-01-02 15:53:22.944560222 +0100
@@ -44,6 +44,7 @@
 
 # Remove SHARED to create a static library
 add_library(ILU SHARED ${ILU_SRCS} ${ILU_INC} ${ILU_RSRC})
+set_target_properties(ILU PROPERTIES SOVERSION 1)
 
 
 ## ILU requires IL
--- a/DevIL/src-ILUT/CMakeLists.txt	2017-01-02 01:03:56.000000000 +0100
+++ b/DevIL/src-ILUT/CMakeLists.txt	2017-01-02 15:54:12.176607698 +0100
@@ -44,6 +44,7 @@
 
 # Remove SHARED to create a static library
 add_library(ILUT SHARED ${ILUT_SRCS} ${ILUT_INC} ${ILUT_RSRC})
+set_target_properties(ILUT PROPERTIES SOVERSION 1)
 
 ## add link sub library info
 target_link_libraries(ILUT
