--- DevIL/src-IL/src/il_devil.cpp	2017-01-02 01:03:56.000000000 +0100
+++ DevIL/src-IL/src/il_devil.cpp	2017-12-08 22:20:16.696024947 +0100
@@ -187,7 +187,7 @@
 // Internal version of ilTexImageSurface.
 ILboolean ILAPIENTRY ilTexImageSurface_(ILimage *Image, ILuint Width, ILuint Height, ILuint Depth, ILubyte Bpp, ILenum Format, ILenum Type, void *Data)
 {
-	ILimage* mips;
+	ILimage* mipmaps;
 	ILimage* next;
 	ILimage* faces;
 	ILimage* layers;
@@ -220,7 +220,7 @@
 	return IL_FALSE;
 	}*/
 
-	mips = Image->Mipmaps;
+	mipmaps = Image->Mipmaps;
 	next = Image->Next;
 	faces = Image->Faces;
 	layers = Image->Layers;
@@ -230,7 +230,7 @@
 	retval = ilInitImage(Image, Width, Height, Depth, Bpp, Format, Type, Data);
 
 	// reset our chains
-	Image->Mipmaps = mips;
+	Image->Mipmaps = mipmaps;
 	Image->Next = next;
 	Image->Faces = faces;
 	Image->Layers = layers;
