--- src-IL/src/il_jp2.c.orig	2016-11-13 09:08:41.000000000 +0000
+++ src-IL/src/il_jp2.c	2016-11-13 09:09:13.198423105 +0000
@@ -459,7 +459,7 @@
 			/* The buffer must be large enough to accommodate maximum
 			  putback. */
 			assert(bufsize > JAS_STREAM_MAXPUTBACK);
-			stream->bufbase_ = JAS_CAST(uchar *, buf);
+			stream->bufbase_ = JAS_CAST(jas_uchar *, buf);
 			stream->bufsize_ = bufsize - JAS_STREAM_MAXPUTBACK;
 		}
 	} else {
