From a545dc48937d29a8fc53376dd398ac92274fd1df Mon Sep 17 00:00:00 2001
From: Xiao-Long Chen <chenxiaolong@cxl.epac.to>
Date: Mon, 27 Oct 2014 21:50:02 -0400
Subject: [PATCH 4/5] Disable -Werror

---
 cmake/CompizCommon.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/CompizCommon.cmake b/cmake/CompizCommon.cmake
index e416a96..8e817f8 100644
--- a/cmake/CompizCommon.cmake
+++ b/cmake/CompizCommon.cmake
@@ -74,7 +74,7 @@ if (NOT COMPIZ_SIGN_WARNINGS)
 endif ()
 
 if (${CMAKE_PROJECT_NAME} STREQUAL "compiz")
-    set (COMPIZ_WERROR_DEFAULT ON)
+    set (COMPIZ_WERROR_DEFAULT OFF)
 else ()
     set (COMPIZ_WERROR_DEFAULT OFF)
 endif ()
-- 
2.10.0

