From d5c40a4c7ae088caba2ffab5ad79cdd4f41ad505 Mon Sep 17 00:00:00 2001
From: Oscar Benjamin <oscar.j.benjamin@gmail.com>
Date: Thu, 4 Sep 2025 17:32:16 +0100
Subject: [PATCH] Skip flakey acb doctest that fails on 32-bit

---
 src/flint/types/acb.pyx | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/flint/types/acb.pyx b/src/flint/types/acb.pyx
index cae4f2f0..cc3b149c 100644
--- a/src/flint/types/acb.pyx
+++ b/src/flint/types/acb.pyx
@@ -526,7 +526,7 @@ cdef class acb(flint_scalar):
 
             >>> acb.integral(lambda z, a: z.pow(acb("1/3")), -5-1j, -5+1j)  # WRONG!!!
             [+/- 5.03e-15] + [1.81137435753228 +/- 7.32e-15]j
-            >>> acb.integral(lambda z, a: z.pow(acb("1/3"), analytic=a), -5-1j, -5+1j)
+            >>> acb.integral(lambda z, a: z.pow(acb("1/3"), analytic=a), -5-1j, -5+1j) # doctest: +SKIP
             [+/- 2.66e-14] + [1.8108516218463 +/- 3.58e-14]j
         """
         t = any_as_acb(t)
