--- a/src/updatedb.c	2015-11-15 17:52:09.470706519 -0500
+++ b/src/updatedb.c	2015-11-15 17:52:31.754706775 -0500
@@ -35,6 +35,7 @@
 #include <sys/stat.h>
 #include <sys/time.h>
 #include <unistd.h>
+#include <paths.h>
 
 #include <mntent.h>
 #include "error.h"
@@ -388,7 +389,7 @@
 	  char *dir;
 
 #ifndef PROC_MOUNTS_PATH
-	  dir = canonicalize_file_name (me->mnt_dir);
+	  dir = realpath (me->mnt_dir, NULL);
 	  if (dir == NULL)
 	    dir = me->mnt_dir;
 #else
