From db2f422578140ba6aaf7a2a1d54caacf832fbee2 Mon Sep 17 00:00:00 2001
From: Nicolas Dufresne <nicolas.dufresne@collabora.com>
Date: Mon, 4 Jul 2016 17:54:45 -0400
Subject: build: Define MSVC _WIN32 so we can build on mingw

https://bugs.freedesktop.org/show_bug.cgi?id=96754

diff --git a/configure.ac b/configure.ac
index 92706e7..5ec8517 100644
--- a/configure.ac
+++ b/configure.ac
@@ -71,7 +71,7 @@ AS_CASE(["${host}"],
         ],
     [*-mingw32*],
         [
-         PLATFORM_CFLAGS="-DWEBRTC_WIN -U__STRICT_ANSI__"
+         PLATFORM_CFLAGS="-DWEBRTC_WIN -D_WIN32 -U__STRICT_ANSI__"
          HAVE_WIN=1
         ]
 )
-- 
cgit v0.10.2

