From 75ef0de24167daa84c11774f4bc9d1b4a9eaacfa Mon Sep 17 00:00:00 2001
From: Nicolas Dufresne <nicolas.dufresne@collabora.com>
Date: Tue, 5 Jul 2016 18:07:45 -0400
Subject: build: Protect against unsupported CPU types


diff --git a/configure.ac b/configure.ac
index 5ec8517..1fcbd53 100644
--- a/configure.ac
+++ b/configure.ac
@@ -73,7 +73,8 @@ AS_CASE(["${host}"],
         [
          PLATFORM_CFLAGS="-DWEBRTC_WIN -D_WIN32 -U__STRICT_ANSI__"
          HAVE_WIN=1
-        ]
+        ],
+    [AC_MSG_ERROR([Unsupported host $host])]
 )
 AC_SUBST(PLATFORM_CFLAGS)
 AM_CONDITIONAL(HAVE_POSIX, [test "x${HAVE_POSIX}" = "x1"])
@@ -94,8 +95,9 @@ AS_CASE(["${host_cpu}"],
         [
          HAVE_ARM=1
          ARCH_CFLAGS="-DWEBRTC_ARCH_ARM"
-        ]
+        ],
     # FIXME: Add MIPS support, see webrtc/BUILD.gn for defines
+    [AC_MSG_ERROR([Unsupported CPU type $host_cpu])]
 )
 AM_CONDITIONAL(HAVE_X86, [test "x${HAVE_X86}" = "x1"])
 AM_CONDITIONAL(HAVE_ARM, [test "x${HAVE_ARM}" = "x1"])
-- 
cgit v0.10.2

