From b8be6d1095df620bf0e2b5855818fa4340244254 Mon Sep 17 00:00:00 2001
From: Nicolas Dufresne <nicolas.dufresne@collabora.com>
Date: Wed, 6 Jul 2016 13:44:15 -0400
Subject: build: Use -no-undefined to support both clang and gcc


diff --git a/webrtc/modules/audio_processing/Makefile.am b/webrtc/modules/audio_processing/Makefile.am
index de259f0..5a98913 100644
--- a/webrtc/modules/audio_processing/Makefile.am
+++ b/webrtc/modules/audio_processing/Makefile.am
@@ -175,7 +175,7 @@ libwebrtc_audio_processing_la_LIBADD = $(top_builddir)/webrtc/libwebrtc.la \
 				       $(LIBWEBRTC_PRIVATEARCH) \
 				       $(GNUSTL_LIBS)
 libwebrtc_audio_processing_la_LDFLAGS = $(AM_LDFLAGS) $(COMMON_LDFLAGS) \
-					-Wl,--no-undefined \
+					-no-undefined \
 					-version-info $(LIBWEBRTC_AUDIO_PROCESSING_VERSION_INFO)
 
 # FIXME: The MIPS optimisations need to be hooked up once we have the
-- 
cgit v0.10.2

