From 7d15b10fb0829d456486da6a49a84492dd3eca4f Mon Sep 17 00:00:00 2001
From: Nicolas Dufresne <nicolas.dufresne@collabora.com>
Date: Wed, 6 Jul 2016 15:18:15 -0400
Subject: build: Add ARM 64bit support


diff --git a/configure.ac b/configure.ac
index a0c194a..b4b9ddf 100644
--- a/configure.ac
+++ b/configure.ac
@@ -99,6 +99,11 @@ AS_CASE(["${host_cpu}"],
          HAVE_ARM=1
          ARCH_CFLAGS="-DWEBRTC_ARCH_ARM"
         ],
+    [aarch64*],
+        [
+         HAVE_NEON=1
+         ARCH_CFLAGS="-DWEBRTC_HAS_NEON -DWEBRTC_ARCH_ARM64"
+        ],
     # FIXME: Add MIPS support, see webrtc/BUILD.gn for defines
     [AC_MSG_ERROR([Unsupported CPU type $host_cpu])]
 )
-- 
cgit v0.10.2

