--- src/modules/rlm_eap/types/rlm_eap_fast/eap_fast.c	2018-04-17 19:12:27.000000000 +0200
+++ src/modules/rlm_eap/types/rlm_eap_fast/eap_fast.c	2019-04-27 16:56:01.628410415 +0200
@@ -44,7 +44,7 @@
 {
 	const EVP_CIPHER *c;
 	const EVP_MD *h;
-#if OPENSSL_VERSION_NUMBER < 0x10100000L || defined(LIBRESSL_VERSION_NUMBER)
+#if OPENSSL_VERSION_NUMBER < 0x10100000L || (defined(LIBRESSL_VERSION_NUMBER) && LIBRESSL_VERSION_NUMBER < 0x2090000fL)
 	int md_size;
 
 	if (ssl->enc_read_ctx == NULL || ssl->enc_read_ctx->cipher == NULL ||
