diff --git a/examples/gethostszero b/examples/gethostszero
index ce02b20..ce72a58 100755
--- a/examples/gethostszero
+++ b/examples/gethostszero
@@ -6,5 +6,5 @@
 # See the downloaded file itself for the copyright notice.
 #

-curl -s https://someonewhocares.org/hosts/zero/hosts > /etc/hosts.d/02-hosts.zero
+curl -s https://raw.githubusercontent.com/StevenBlack/hosts/master/hosts > /etc/hosts.d/02-hosts.zero
