diff --git a/desktop_version/src/main.cpp b/desktop_version/src/main.cpp
index 4b87dbff..a00ff58f 100644
--- a/desktop_version/src/main.cpp
+++ b/desktop_version/src/main.cpp
@@ -383,9 +383,9 @@ static void keep_console_open(const bool open_console)
 int main(int argc, char *argv[])
 {
     char* baseDir = NULL;
-    char* assetsPath = NULL;
-    char* langDir = NULL;
-    char* fontsDir = NULL;
+    char* assetsPath = (char *)"/usr/share/vvvvvv/data.zip";
+    char* langDir = (char *)"/usr/share/vvvvvv/lang";
+    char* fontsDir = (char *)"/usr/share/vvvvvv/fonts";
     bool seed_use_sdl_getticks = false;
     bool open_console = false;
     bool print_version = false;
