--- a/tests/primer/test_primer_stdlib.py
+++ b/tests/primer/test_primer_stdlib.py
@@ -15,6 +15,7 @@
 from pytest import CaptureFixture
 
 from pylint.testutils._run import _Run as Run
+from pylint.testutils.utils import _test_cwd
 
 
 def is_module(filename: str) -> bool:
@@ -53,8 +54,8 @@
 ) -> None:
     """Test that pylint does not produce any crashes or fatal errors on stdlib modules."""
     __tracebackhide__ = True  # pylint: disable=unused-variable
-    os.chdir(test_module_location)
-    with _patch_stdout(io.StringIO()):
+    with _test_cwd(test_module_location):
+      with _patch_stdout(io.StringIO()):
         try:
             # We want to test all the code we can
             enables = ["--enable-all-extensions", "--enable=all"]
