--- src/qmmpui/uiloader.cpp	2016-10-04 11:49:39.000000000 +0200
+++ src/qmmpui/uiloader.cpp	2016-10-11 11:51:37.780613874 +0200
@@ -84,7 +84,7 @@
 {
     loadPlugins();
     QSettings settings (Qmmp::configFile(), QSettings::IniFormat);
-    QString name = settings.value("Ui/current_plugin", "skinned").toString();
+    QString name = settings.value("Ui/current_plugin", "simple").toString();
     foreach(QmmpUiPluginCache *item, *m_cache)
     {
         if(item->shortName() == name && item->uiFactory())
