From 7db32d9c8588305e17542aa696b9a1fee2fcbc0a Mon Sep 17 00:00:00 2001
From: amanusk <amanusk@pm.me>
Date: Sun, 14 Dec 2025 09:56:28 +0200
Subject: [PATCH] Fix unsucceful import of curses_display (#247)

Fixes issues #246
---
 s_tui/s_tui.py | 1 -
 1 file changed, 1 deletion(-)

diff --git a/s_tui/s_tui.py b/s_tui/s_tui.py
index 1245e97..4c1900c 100755
--- a/s_tui/s_tui.py
+++ b/s_tui/s_tui.py
@@ -36,7 +36,6 @@
 
 import psutil
 import urwid
-import urwid.curses_display
 
 try:
     import configparser
