diff --git a/src/linux/btop_collect.cpp b/src/linux/btop_collect.cpp
index 1799b0d..758fd46 100644
--- a/src/linux/btop_collect.cpp
+++ b/src/linux/btop_collect.cpp
@@ -1281,10 +1281,10 @@ namespace Gpu {
 
 			//? Try possible library paths and names for librocm_smi64.so
 			const array libRocAlts = {
+				"librocm_smi64.so.1.0", // void
 				"/opt/rocm/lib/librocm_smi64.so",
 				"librocm_smi64.so",
 				"librocm_smi64.so.5", // fedora
-				"librocm_smi64.so.1.0", // debian
 				"librocm_smi64.so.6"
 			};
 
