From 932723e0747a30a77738f654a88c052577b5cc6e Mon Sep 17 00:00:00 2001
From: Nathan Owens <ndowens@artixlinux.org>
Date: Mon, 4 Jan 2021 10:31:57 -0600
Subject: [PATCH] cross

---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git configure.ac configure.ac
index 2cfd208..17d6210 100644
--- configure.ac
+++ configure.ac
@@ -136,7 +136,7 @@ AC_FUNC_MMAP()
 dnl ***************************************
 dnl *** Check for strftime() extensions ***
 dnl ***************************************
-AC_TRY_RUN([
+AC_COMPILE_IFELSE([
   #include <string.h>
   #include <time.h>
   int
-- 
2.30.0

