--- a/Makefile	2016-10-05 08:28:54.631977409 +0200
+++ b/Makefile	2016-10-05 08:29:11.588976745 +0200
@@ -40,7 +40,7 @@ CFLAGS_OPT=-pipe $(WARN) $(DEFINE) $(OPT
 #CFLAGS=-pipe -Os -march=pentiumpro -mcpu=pentiumpro -fomit-frame-pointer -fschedule-insns2 -Wall
 
 ent: ent.c haveuint128.h
-	$(CC) -g -o ent ent.c -I.
+	cc -g -o ent ent.c -I.
 
 # CFLAGS += -fstrict-aliasing -Wstrict-aliasing=2
 
--- a/GNUmakefile	2016-10-05 09:02:43.606897920 +0200
+++ b/GNUmakefile	2016-10-05 09:02:54.656897487 +0200
@@ -39,7 +39,7 @@ CFLAGS_OPT=-pipe $(WARN) $(DEFINE) $(OPT
 #CFLAGS=-pipe -Os -march=pentiumpro -mcpu=pentiumpro -fomit-frame-pointer -fschedule-insns2 -Wall
 
 ent: ent.c haveuint128.h
-	$(CC) -g -o ent ent.c -I.
+	cc -g -o ent ent.c -I.
 
 # CFLAGS += -fstrict-aliasing -Wstrict-aliasing=2
 
