diff --git a/io/iob_send.c b/io/iob_send.c
index 10a6e8c..81c2763 100644
--- a/io/iob_send.c
+++ b/io/iob_send.c
@@ -122,6 +122,14 @@ int64 iob_send(int64 s,io_batch* b) {
 #include "io_internal.h"
 #include "iob_internal.h"
 
+/*
+ * musl defines SO_ZEROCOPY 60 in sys/socket.h
+ * glibc doesn't, for whatever reason, so do it here
+ */
+#ifndef SO_ZEROCOPY
+#define SO_ZEROCOPY 60
+#endif
+
 int64 iob_send(int64 s,io_batch* b) {
   iob_entry* e,* last;
   io_entry* E;

