--- src/settings/_default.settings.orig	2016-09-04 08:57:45.591685817 +0200
+++ src/settings/_default.settings	2016-09-04 08:57:59.528686536 +0200
@@ -232,7 +232,7 @@
     "restart": true
   },
   {
-    "value": true,
+    "value": false,
     "title": "Send Anonymous Metrics and Errors",
     "type": "bool",
     "category": "Debug",
@@ -628,4 +628,4 @@
     "value": "Ctrl+I",
     "type": "text"
   }
-]
\ No newline at end of file
+]
