diff --git a/vinagre/vinagre-connection.c b/vinagre/vinagre-connection.c
index 812ae92..3a3b9ac 100644
--- vinagre/vinagre-connection.c
+++ vinagre/vinagre-connection.c
@@ -582,7 +582,6 @@ vinagre_connection_split_string (const gchar *uri,
   gchar **server, **url;
   gint    lport;
   gchar  *lhost;
-  gboolean is_ipv6;
   gchar   ipv6_host[255] = {0,};
   VinagreProtocol *ext;
 
@@ -619,11 +618,6 @@ vinagre_connection_split_string (const gchar *uri,
       return FALSE;
     }
 
-  is_ipv6 = g_strstr_len (lhost, -1, ":") != NULL;
-
-  if (is_ipv6)
-    lhost = g_strconcat ("[", lhost, "]", NULL);
-
   if (lhost[0] == '[')
     {
       int i;
