diff -upr vinagre-3.18.1.orig/vinagre/vinagre-connection.c vinagre-3.18.1/vinagre/vinagre-connection.c
--- vinagre/vinagre-connection.c.orig	2015-09-25 13:35:42.000000000 +0200
+++ vinagre/vinagre-connection.c	2015-11-03 11:26:15.058608241 +0100
@@ -582,7 +582,7 @@ vinagre_connection_split_string (const g
   gchar **server, **url;
   gint    lport;
   gchar  *lhost;
-  gboolean is_ipv6;
+  gchar  *tmphost;
   gchar   ipv6_host[255] = {0,};
   VinagreProtocol *ext;
 
@@ -619,10 +619,12 @@ vinagre_connection_split_string (const g
       return FALSE;
     }
 
-  is_ipv6 = g_strstr_len (lhost, -1, ":") != NULL;
-
-  if (is_ipv6)
-    lhost = g_strconcat ("[", lhost, "]", NULL);
+  /* Assume an hostname contains more than 1 colons as an ipv6 address */
+  if(lhost[0] != '[') {
+    tmphost = g_strstr_len (lhost, -1, ":");
+    if(tmphost && g_strstr_len (tmphost+1, -1, ":") != NULL)
+      lhost = g_strconcat ("[", lhost, "]", NULL);
+  }
 
   if (lhost[0] == '[')
     {
