--- a/cmake/wasi-sdk-sysroot.cmake
+++ b/cmake/wasi-sdk-sysroot.cmake
@@ -357,16 +357,7 @@
   add_dependencies(build build-${target})
 endforeach()
 
-# Install a `VERSION` file in the output prefix with a dump of version
-# information.
-execute_process(
-  COMMAND ${PYTHON} ${version_script} dump
-  WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-  OUTPUT_VARIABLE version_dump)
-set(version_file_tmp ${wasi_sysroot}/VERSION)
-file(GENERATE OUTPUT ${version_file_tmp} CONTENT ${version_dump})
-add_custom_target(version-file DEPENDS ${version_file_tmp})
-add_dependencies(build version-file)
+add_dependencies(build)
 
 if(WASI_SDK_INCLUDE_TESTS)
   add_subdirectory(tests)
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -23,11 +23,8 @@
 find_program(PYTHON python3 python REQUIRED)
 
 # Set some variables based on the `version.py` script
-set(version_script ${CMAKE_CURRENT_SOURCE_DIR}/version.py)
-execute_process(
-  COMMAND ${PYTHON} ${version_script}
-  OUTPUT_VARIABLE wasi_sdk_version
-  OUTPUT_STRIP_TRAILING_WHITESPACE)
+option(WASI_SDK_VERSION "version")
+set(wasi_sdk_version ${WASI_SDK_VERSION})
 
 message(STATUS "wasi-sdk version is ${wasi_sdk_version}")
 
