From 517b27faf6e1471c2ccb25c3c22f78eb862cd552 Mon Sep 17 00:00:00 2001
From: "Kowalski Dragon (kowalski7cc)" <kowalski7cc@users.noreply.github.com>
Date: Sun, 14 Sep 2025 14:38:11 +0200
Subject: [PATCH] Build: SET CMP0004 OLD only if CMake < 4.0

Signed-off-by: Kowalski Dragon (kowalski7cc) <kowalski7cc@users.noreply.github.com>
---
 CMakeLists.txt | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index a7456337e8f5d07448e0a388446f0b7e19e46960..4021d672111d42d1b18e629a6f096f82a7355146 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -164,7 +164,9 @@ elseif(HAIKU)
 else()
     # Workaround for SDL bug #3295, which occurs in SDL2 <2.0.5
     # https://bugzilla.libsdl.org/show_bug.cgi?id=3295
-    cmake_policy(SET CMP0004 OLD)
+    if(${CMAKE_VERSION} VERSION_LESS "4.0.0")
+        cmake_policy(SET CMP0004 OLD)
+    endif()
 
     find_package(SDL2 REQUIRED)
 endif()
