--- a/src/milkyplay/drivers/jack/AudioDriver_JACK.cpp
+++ b/src/milkyplay/drivers/jack/AudioDriver_JACK.cpp	
@@ -92,7 +92,7 @@
 mp_sint32 AudioDriver_JACK::initDevice(mp_sint32 bufferSizeInWords, mp_uint32 mixFrequency, MasterMixer* mixer)
 {
 	// First load libjack
-	libJack = dlopen("libjack.so", RTLD_LAZY);
+	libJack = dlopen("libjack.so.0", RTLD_LAZY);
 	if(!libJack) {
 		fprintf(stderr, "JACK: Can't load libjack (is it installed?)\n");
 		return -1;
