diff --git a/ntpd/src/daemon/keyexchange.rs b/ntpd/src/daemon/keyexchange.rs
index b1c00ea..884208e 100644
--- a/ntpd/src/daemon/keyexchange.rs
+++ b/ntpd/src/daemon/keyexchange.rs
@@ -736,7 +736,7 @@ mod tests {
             key_exchange_client(
                 "localhost".to_string(),
                 5435,
-                &certificates_from_bufread(BufReader::new(Cursor::new(ca))).unwrap(),
+                &certificates_from_bufread(BufReader::new(Cursor::new(ca))),
             )
         )
         .await
@@ -750,7 +750,7 @@ mod tests {
             key_exchange_client(
                 "localhost".to_string(),
                 5435,
-                &certificates_from_bufread(BufReader::new(Cursor::new(ca))).unwrap(),
+                &certificates_from_bufread(BufReader::new(Cursor::new(ca))),
             ),
         )
         .await
