diff --git a/Cargo.toml b/Cargo.toml
index f3f4c80a..e14970da 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -46,10 +46,10 @@ tracing = "0.1.37"
 tracing-subscriber = { version = "0.3.0", default-features = false, features = ["std", "fmt", "ansi"] }
 
 # TLS
-rustls23 = { package = "rustls", version = "0.23.22", default-features = false, features = ["logging", "std", "tls12"] }
+rustls23 = { package = "rustls", version = "0.23.22", default-features = false, features = ["ring", "logging", "std"] }
 rustls-openssl = ">=0.3.0,<0.9"
 rustls-platform-verifier = "0.5.0"
-tokio-rustls = { version = "0.26.0", default-features = false, features = ["logging", "tls12"] }
+tokio-rustls = { version = "0.26.0", default-features = false, features = ["ring", "logging"] }
 
 # crypto
 aead = "0.5.0"
