diff --git a/Cargo.toml b/Cargo.toml
index e36919e..d5e07b8 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -45,9 +45,9 @@ clock-steering = "0.2.1"
 pps-time = "0.2.3"
 
 # TLS
-rustls23 = { package = "rustls", version = "0.23.16", features = ["logging", "std"] }
+rustls23 = { package = "rustls", version = "0.23.16", default-features = false, features = ["ring", "logging", "std"] }
 rustls-platform-verifier = "0.5.0"
-tokio-rustls = { version = "0.26.0", features = ["logging"] }
+tokio-rustls = { version = "0.26.0", default-features = false, features = ["ring", "logging"] }
 
 # crypto
 aead = "0.5.0"
