--- util.cc.orig
+++ util.cc
@@ -917,13 +917,8 @@
 
     d_errstr[0] = '\0';
 
-    STRERROR_R_T r;
-    r = strerror_r(errnum, d_errstr, 1024);
-
-    if (strlen(r) == 0)
-        rs = fmt::format("Unknown error: {}", errnum);
-    else
-        rs = std::string(d_errstr, 1024);
+    strerror_r(errnum, d_errstr, 1024);
+    rs = std::string(d_errstr, 1024);
     
     delete[] d_errstr;
     return rs;
