--- a/scripts/packages/bazel.sh
+++ b/scripts/packages/bazel.sh
@@ -104,7 +104,8 @@ function get_workspace_root() {
 
 get_workspace_root
 
-readonly wrapper_dir="$(dirname "$(get_realpath "${BASH_SOURCE[0]}")")"
+export JAVA_HOME=/usr/lib/jvm/openjdk21
+readonly wrapper_dir="/usr/libexec/bazel"
 readonly os_arch_suffix="$(uname -s | tr '[:upper:]' '[:lower:]')-$(uname -m)"
 
 function get_bazel_version() {
