--- a/tools/build_defs/pkg/path_test.py
+++ b/tools/build_defs/pkg/path_test.py
@@ -14,9 +14,18 @@
 """Testing for helper functions."""
 
 import imp
+import importlib.util
+import importlib.machinery
 import unittest
 
-pkg_bzl = imp.load_source(
+def load_source(modname, filename):
+    loader = importlib.machinery.SourceFileLoader(modname, filename)
+    spec = importlib.util.spec_from_file_location(modname, filename, loader=loader)
+    module = importlib.util.module_from_spec(spec)
+    loader.exec_module(module)
+    return module
+
+pkg_bzl = load_source(
     'pkg_bzl',
     'tools/build_defs/pkg/path.bzl')
 
