This patch eliminates the need to modify /etc/hosts

--- a/tests/LTsock.c
+++ b/tests/LTsock.c
@@ -255,10 +255,7 @@ main(argc, argv)
  * Get the host name and its IP address.  Convert the IP address to dotted
  * ASCII form.
  */
-    if (gethostname(hnm, sizeof(hnm) - 1)) {
-	cem = "ERROR!!!  can't get this host's name";
-	goto print_errno;
-    }
+    strncpy(hnm, "localhost", sizeof(hnm) - 1);
     hnm[sizeof(hnm) - 1] = '\0';
     if (!(hp = gethostbyname(hnm))) {
        (void) snprintf(buf, bufl - 1, "ERROR!!!  can't get IP address for %s",
