--- a/CMakeLists.txt	2020-12-29 14:53:13.000000000 +0100
+++ b/CMakeLists.txt	2020-12-29 14:53:13.000000000 +0100
@@ -87,6 +87,7 @@
     include (CheckIncludeFileCXX)
     include (CheckSymbolExists)
     include (CheckCSourceCompiles)
+    include (FindPkgConfig)
 
     check_include_file_cxx (istream HAVE_ISTREAM)
     check_include_file_cxx (ostream HAVE_OSTREAM)
@@ -247,6 +248,11 @@
         if (HAVE_Xi)
             list (APPEND libs Xi)
         endif()
+        if (${PKG_CONFIG_FOUND})
+             pkg_check_modules (AVAHI_COMPAT REQUIRED avahi-compat-libdns_sd)
+             include_directories (BEFORE SYSTEM ${AVAHI_COMPAT_INCLUDE_DIRS})
+             set (CMAKE_REQUIRED_INCLUDES "${CMAKE_REQUIRED_INCLUDES};${AVAHI_COMPAT_INCLUDE_DIRS}")
+        endif ()
 
         if (NOT SYNERGY_ENTERPRISE AND SYNERGY_BUILD_LEGACY_GUI)
             set (DnsSdlib "dns_sd.h")
