--- a/src/gui/src/ScreenSetupView.h	2019-06-09 19:16:38.358110544 +0200
+++ b/src/gui/src/ScreenSetupView.h	2019-06-09 19:16:53.268001862 +0200
@@ -22,6 +22,7 @@
 
 #include <QTableView>
 #include <QFlags>
+#include <QHeaderView>
 
 #include "Screen.h"
 
--- a/src/gui/src/ActionDialog.h	2019-06-09 19:17:32.518715748 +0200
+++ b/src/gui/src/ActionDialog.h	2019-06-09 19:17:50.508584615 +0200
@@ -21,6 +21,7 @@
 #define ACTIONDIALOG_H
 
 #include <QDialog>
+#include <QButtonGroup>
 
 #include "ui_ActionDialogBase.h"
 
