--- configure.orig	2014-06-09 08:20:00.584285131 +0200
+++ configure	2014-06-09 08:20:14.000360141 +0200
@@ -6019,7 +6019,7 @@ _ACEOF
 
 		{ $as_echo "$as_me:${as_lineno-$LINENO}: checking whether getaddrinfo() works" >&5
 $as_echo_n "checking whether getaddrinfo() works... " >&6; }
-		if test "$cross_compiling" = yes; then :
+		if false; then :
   { { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
 $as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
 as_fn_error $? "cannot run test program while cross compiling
