diff --git a/build.zig b/build.zig
index cdbe456..d13ad6a 100644
--- a/build.zig
+++ b/build.zig
@@ -91,6 +91,12 @@ pub fn build(b: *Build) !void {
     const wayland = b.createModule(.{ .root_source_file = scanner.result });
     const xkbcommon = b.dependency("xkbcommon", .{}).module("xkbcommon");
 
+    const translate_c = b.addTranslateC(.{
+        .root_source_file = b.path("src/c.h"),
+        .target = target,
+        .optimize = optimize,
+    });
+
     const waylock = b.addExecutable(.{
         .name = "waylock",
         .root_module = b.createModule(.{
@@ -99,6 +105,12 @@ pub fn build(b: *Build) !void {
             .optimize = optimize,
             .link_libc = true,
             .strip = strip,
+            .imports = &.{
+                .{
+                    .name = "c",
+                    .module = translate_c.createModule(),
+                },
+            },
         }),
         .use_llvm = use_llvm,
         .use_lld = use_llvm,
diff --git a/src/auth.zig b/src/auth.zig
index 83fa68f..1ad06fe 100644
--- a/src/auth.zig
+++ b/src/auth.zig
@@ -8,10 +8,7 @@ const process = std.process;
 const fatal = process.fatal;
 const system = std.posix.system;
 
-const c = @cImport({
-    @cInclude("unistd.h"); // getuid()
-    @cInclude("pwd.h"); // getpwuid()
-});
+const c = @import("c");
 
 const pam = @import("pam.zig");
 
diff --git a/src/c.h b/src/c.h
new file mode 100644
index 0000000..5d84a42
--- /dev/null
+++ b/src/c.h
@@ -0,0 +1,2 @@
+#include <unistd.h>
+#include <pwd.h>
