--- a/src/vbo.pyx
+++ b/src/vbo.pyx
@@ -188,7 +188,7 @@
         assert not self.created, """Already created the buffer"""
         buffers = self.get_implementation().glGenBuffers(1)
         try:
-            self.buffer = long( buffers )
+            self.buffer = int( buffers )
         except (TypeError,ValueError) as err:
             self.buffer = buffers[0]
         self.target = self.c_resolve( self.target_spec )
@@ -242,7 +242,7 @@
         """Add an integer to this VBO (offset)"""
         if hasattr( other, 'offset' ):
             other = other.offset
-        assert isinstance( other, (int,long) ), """Only know how to add integer/long offsets"""
+        assert isinstance( other, (int) ), """Only know how to add integer offsets"""
         return VBOOffset( self, other )
     cdef int check_live( self ):
         if self.data is _NULL:
