From 0f975b37b6b01f7e263389e62b929bf4a7965d6f Mon Sep 17 00:00:00 2001
From: Komeil Parseh <komeilparseh@disroot.org>
Date: Thu, 7 Aug 2025 20:12:39 +0330
Subject: [PATCH] Fix: Skip test_Display_banner_get_text_width_fallback on musl

---
 test/units/utils/test_display.py | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/test/units/utils/test_display.py b/test/units/utils/test_display.py
index 9d4d30f..84dde89 100644
--- a/test/units/utils/test_display.py
+++ b/test/units/utils/test_display.py
@@ -10,6 +10,7 @@ import sys
 import typing as t
 import unicodedata
 
+from pathlib import Path
 from unittest.mock import MagicMock
 
 import pytest
@@ -19,6 +20,8 @@ from ansible.module_utils._internal import _deprecator, _errors, _messages
 from ansible.utils.display import _LIBC, _MAX_INT, Display, get_text_width, _format_message
 from ansible.utils.multiprocessing import context as multiprocessing_context
 
+def is_musl() -> bool:  # FIXME
+    return Path("/tmp/__force_musl__").exists()
 
 @pytest.fixture
 def problematic_wcswidth_chars():
@@ -72,6 +75,7 @@ def test_Display_banner_get_text_width(monkeypatch, display_resource):
     assert msg.endswith(stars)
 
 
+@pytest.mark.skipif(is_musl(), reason="Fails due to musl locale handling differences")
 def test_Display_banner_get_text_width_fallback(monkeypatch, display_resource):
     locale.setlocale(locale.LC_ALL, 'C.UTF-8')
     display = Display()
-- 
2.50.1

