diff --git a/test/units/_internal/_encryption/test_crypt.py b/test/units/_internal/_encryption/test_crypt.py
index 661e2c66a6..efcb5c52af 100644
--- a/test/units/_internal/_encryption/test_crypt.py
+++ b/test/units/_internal/_encryption/test_crypt.py
@@ -80,6 +80,7 @@ class TestCryptFacade:
         assert crypt_facade._use_crypt_gensalt_rn is False
         assert crypt_facade.has_crypt_gensalt
 
+    @pytest.mark.skip(reason='no libxcrypt')
     def test_crypt_fail_errno(self, mocker: MockerFixture) -> None:
         """Test crypt() setting failure errno raises OSError."""
         mocker.patch('ctypes.get_errno', return_value=errno.EBADFD)
@@ -88,6 +89,7 @@ class TestCryptFacade:
         with pytest.raises(OSError, match=r'crypt failed:'):
             crypt_facade.crypt(b"test", b"123")
 
+    @pytest.mark.skip(reason='no libxcrypt')
     def test_crypt_result_none(self, mocker: MockerFixture) -> None:
         """Test crypt() implementation returning None raises ValueError."""
         crypt_facade = CryptFacade()
@@ -96,6 +98,7 @@ class TestCryptFacade:
         with pytest.raises(ValueError, match=r'crypt failed: invalid salt or unsupported algorithm'):
             crypt_facade.crypt(b"test", b"123")
 
+    @pytest.mark.skip(reason='no libxcrypt')
     def test_crypt_result_failure(self, mocker: MockerFixture) -> None:
         """Test crypt() implementation returning failure token raises ValueError."""
         crypt_facade = CryptFacade()
@@ -104,6 +107,7 @@ class TestCryptFacade:
         with pytest.raises(ValueError, match=r'crypt failed: invalid salt or unsupported algorithm'):
             crypt_facade.crypt(b"test", b"123")
 
+    @pytest.mark.skip(reason='no libxcrypt')
     def test_crypt_gensalt_called_with_no_impl(self, mocker: MockerFixture) -> None:
         """Calling crypt_gensalt() without impl should raise NotImplementedError."""
         crypt_facade = CryptFacade()
@@ -113,6 +117,7 @@ class TestCryptFacade:
         with pytest.raises(NotImplementedError, match=r'crypt_gensalt not available \(requires libxcrypt\)'):
             crypt_facade.crypt_gensalt(b"", 1, b"")
 
+    @pytest.mark.skip(reason='no libxcrypt')
     def test_crypt_gensalt(self, mocker: MockerFixture) -> None:
         """Test the NOT _use_crypt_gensalt_rn code path of crypt_gensalt()."""
         crypt_facade = CryptFacade()
@@ -122,6 +127,7 @@ class TestCryptFacade:
         crypt_facade.crypt_gensalt(b'', 1, b'')
         mock_impl.assert_called_once_with(b'', 1, b'', 0)
 
+    @pytest.mark.skip(reason='no libxcrypt')
     def test_crypt_gensalt_fail_errno(self, mocker: MockerFixture) -> None:
         """Test crypt_gensalt() setting failure errno raises OSError."""
         mocker.patch('ctypes.get_errno', return_value=errno.EBADFD)
@@ -130,6 +136,7 @@ class TestCryptFacade:
         with pytest.raises(OSError, match=r'crypt_gensalt failed:'):
             crypt_facade.crypt_gensalt(b'', 1, b'')
 
+    @pytest.mark.skip(reason='no libxcrypt')
     def test_crypt_gensalt_result_none(self, mocker: MockerFixture) -> None:
         """Test crypt_gensalt() implementation returning None raises ValueError."""
         crypt_facade = CryptFacade()
@@ -138,6 +145,7 @@ class TestCryptFacade:
         with pytest.raises(ValueError, match=r'crypt_gensalt failed: unable to generate salt'):
             crypt_facade.crypt_gensalt(b'', 1, b'')
 
+    @pytest.mark.skip(reason='no libxcrypt')
     def test_crypt_gensalt_result_failure(self, mocker: MockerFixture) -> None:
         """Test crypt_gensalt() implementation returning failure token raises ValueError."""
         crypt_facade = CryptFacade()
