--- a/libfs/devices.h 2005-11-23 07:43:55.000000000 +1100
+++ b/libfs/devices.h 2013-01-06 11:30:25.381538024 +1100
@@ -40,6 +40,8 @@
 #define ERROR_INVALID_PARAMETER  87
 #define ERROR_DISK_FULL    112
 
+#include <inttypes.h>
+
 struct stat;
 
 int ujfs_get_dev_size(FILE *, int64_t * size);
--- a/libfs/fssubs.c	2015-06-03 19:18:02.225624388 +0200
+++ b/libfs/fssubs.c	2015-06-03 19:18:46.216543510 +0200
@@ -30,8 +30,9 @@
 #undef HAVE_GETMNTINFO		/* we don't want both */
 #endif
 
-#if HAVE_GETMNTINFO
 #include <paths.h>
+
+#if HAVE_GETMNTINFO
 #include <sys/param.h>
 #include <sys/mount.h>
 #endif
