--- src/client.c.orig	2009-11-24 05:12:11.957336471 +0000
+++ src/client.c	2009-11-24 05:12:29.289887242 +0000
@@ -1620,7 +1620,7 @@ KeyTable	update_table[] =
 
 
 static gint
-getline(gint fd, gchar *buf, gint len)
+get_line(gint fd, gchar *buf, gint len)
 	{
 	fd_set			read_fds;
 	struct timeval	tv;
@@ -1714,7 +1714,7 @@ read_server_setup(gint fd)
 
 	while (1)
 		{
-		rs=getline(fd, buf, sizeof(buf));
+		rs=get_line(fd, buf, sizeof(buf));
         if(rs<0)
             return FALSE;
 		if (!strcmp(buf, "</gkrellmd_setup>"))
@@ -1732,7 +1732,7 @@ read_server_setup(gint fd)
 	table_size = sizeof(update_table) / sizeof(KeyTable);
 	while (1)
 		{
-		rs=getline(fd, buf, sizeof(buf));
+		rs=get_line(fd, buf, sizeof(buf));
         if(rs<0)
             return FALSE;
 		if (!strcmp(buf, "</initial_update>"))
