diff --git src/mail.c src/mail.c
index 6615f31..205b56f 100644
--- a/src/mail.c
+++ b/src/mail.c
-#if defined(OPENSSL_VERSION_NUMBER) && (OPENSSL_VERSION_NUMBER < 0x10100000L)
+#if defined(OPENSSL_VERSION_NUMBER) && (OPENSSL_VERSION_NUMBER < 0x10100000L) && !defined(LIBRESSL_VERSION_NUMBER)
 static void
 ssl_locking_cb(int mode, int n, const char *file, int line)
 	{
@@ -452,7 +452,7 @@ update_tooltip(void)
 	MailAccount	*account;
 	GString		*mboxes = NULL;
 	gchar		buf[128];
-
+
 	if (show_tooltip)
 		{
 		mboxes = g_string_sized_new(512);
@@ -4399,8 +4399,10 @@ gkrellm_init_mail_monitor(void)
 		ssl_locks[i] = g_new(GMutex, 1);
 		g_mutex_init(ssl_locks[i]);
 		}
+#ifndef LIBRESSL_VERSION_NUMBER
 	CRYPTO_set_locking_callback(ssl_locking_cb);
 #endif
+#endif
 #endif

 	mail_fetch = g_new0(Mailbox, 1);
@@ -4526,4 +4528,3 @@ gkrellm_destroy_external_mbox(gpointer mbox_ptr)
 	}

   /* =======================================================================*/
-
