diff -Nru mesa-demos-8.0.1.orig//src/egl/openvg/lion-render.h mesa-demos-8.0.1/src/egl/openvg/lion-render.h
--- src/egl/openvg/lion-render.h	1970-01-01 01:00:00.000000000 +0100
+++ src/egl/openvg/lion-render.h	2011-07-18 17:20:37.000000000 +0200
@@ -0,0 +1,16 @@
+#ifndef LION_RENDER_H
+#define LION_RENDER_H
+
+#include <VG/openvg.h>
+
+#define LION_SIZE 132
+struct lion {
+   VGPath paths[LION_SIZE];
+   VGPaint fills[LION_SIZE];
+};
+
+struct lion *lion_create(void);
+void lion_render(struct lion *l);
+void lion_destroy(struct lion *l);
+
+#endif
diff -Nru mesa-demos-8.0.1.orig//src/egl/openvg/trivial/eglcommon.h mesa-demos-8.0.1/src/egl/openvg/trivial/eglcommon.h
--- src/egl/openvg/trivial/eglcommon.h	1970-01-01 01:00:00.000000000 +0100
+++ src/egl/openvg/trivial/eglcommon.h	2011-07-18 17:20:21.921788794 +0200
@@ -0,0 +1,20 @@
+#ifndef EGLCOMMON_H
+#define EGLCOMMON_H
+
+typedef void (*init_func)();
+typedef void (*reshape_func)(int, int);
+typedef void (*draw_func)();
+typedef int  (*key_func)(unsigned key);
+
+
+void set_window_size(int width, int height);
+int window_width(void);
+int window_height(void);
+
+int run(int argc, char **argv,
+        init_func init,
+        reshape_func resh,
+        draw_func draw,
+        key_func key);
+
+#endif
