--- src/egl/opengl/Makefile.in.orig	2012-02-09 16:52:51.621534418 +0100
+++ src/egl/opengl/Makefile.in	2012-02-09 16:53:23.244536622 +0100
@@ -111,10 +111,10 @@ peglgears_OBJECTS = peglgears.$(OBJEXT)
 peglgears_LDADD = $(LDADD)
 xeglgears_SOURCES = xeglgears.c
 xeglgears_OBJECTS = xeglgears.$(OBJEXT)
-xeglgears_LDADD = $(LDADD)
+xeglgears_LDADD = $(LDADD) -lX11
 xeglthreads_SOURCES = xeglthreads.c
 xeglthreads_OBJECTS = xeglthreads.$(OBJEXT)
-xeglthreads_LDADD = $(LDADD)
+xeglthreads_LDADD = $(LDADD) -lX11 -lpthread
 DEFAULT_INCLUDES = -I.@am__isrc@
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__depfiles_maybe = depfiles
