--- a/mnamer/utils.py	2023-08-15 21:24:08.000000000 +0200
+++ b/mnamer/utils.py	2025-08-09 11:16:52.264564049 +0200
@@ -256,7 +256,7 @@
         "like Gecko) Chrome/79.0.3945.88 Safari/537.36"
     )
 
-    initial_cache_state = session._disabled  # yes, i'm a bad person
+    initial_cache_state = getattr(session, "_disabled", False)  # patched for requests-cache>=1.0
     try:
         session._disabled = not cache
         response = session.request(
