Add a library check for musl-obstack package

--- configure.ac	2013-08-26 22:55:43.000000000 +0200
+++ configure.ac	2015-09-10 09:21:53.809280964 +0200
@@ -29,6 +29,11 @@
 UUID_LIBS="$LIBS"
 AC_SUBST(UUID_LIBS)
 
+# Check for LIBOBSTACK
+AC_CHECK_LIB(obstack, _obstack_begin, , AC_MSG_WARN(libobstack could not be found))
+OBSTACK_LIBS="$LIBS"
+AC_SUBST(OBSTACK_LIBS)
+
 dnl Checks for header files.
 AC_HEADER_STDC
 AC_CHECK_HEADERS(fcntl.h limits.h malloc.h sys/ioctl.h unistd.h uuid/uuid.h)
