From 80392c12f71cb4e562099631e8ee496deb1ad719 Mon Sep 17 00:00:00 2001
From: PavanTarun <lavanya.murugan@multicorewareinc.com>
Date: Mon, 13 Jul 2026 15:54:39 +0530
Subject: [PATCH] Fix i686 build failure caused by non-PC-relative PLT
 references

Signed-off-by: PavanTarun <pavan.tarun@multicorewareinc.com>
---
 source/common/x86/cpu-a.asm | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/source/common/x86/cpu-a.asm b/source/common/x86/cpu-a.asm
index deb3b4ae3..841acd70e 100644
--- a/source/common/x86/cpu-a.asm
+++ b/source/common/x86/cpu-a.asm
@@ -177,7 +177,7 @@ cglobal safe_intel_cpu_indicator_init
 %if WIN64
     lea rax, [intel_cpu_indicator_init]
     call rax
-%elif FORMAT_ELF
+%elif UNIX64 && FORMAT_ELF
     call [rel intel_cpu_indicator_init wrt ..plt]
 %else
     call intel_cpu_indicator_init
