--- source/common/param.cpp.orig	2018-05-28 11:05:41.571934454 +0200
+++ source/common/param.cpp	2018-05-28 11:05:47.109929916 +0200
@@ -633,7 +633,7 @@ int x265_param_parse(x265_param* p, cons
         if (bValueWasNull)
             p->cpuid = atobool(value);
         else
-            p->cpuid = parseCpuName(value, bError);
+            p->cpuid = parseCpuName(value, bError, false);
 #endif
     }
     OPT("fps")
