--- a/src/calibre/headless/CMakeLists.txt
+++ b/src/calibre/headless/CMakeLists.txt
@@ -1,7 +1,7 @@
 cmake_minimum_required(VERSION 3.21)
 project(headless)
 set(CMAKE_AUTOMOC ON)
-find_package(Qt6Gui REQUIRED)
+find_package(Qt6 REQUIRED COMPONENTS Gui GuiPrivate Core CorePrivate)
 add_library(headless MODULE main.cpp headless_backingstore.cpp headless_integration.cpp)
 set_property(TARGET headless PROPERTY QT_PLUGIN_TYPE "platforms")
 set_property(TARGET headless PROPERTY QT_PLUGIN_CLASS_NAME "HeadlessIntegrationPlugin")
