--- a/meson.build.orig
+++ b/meson.build
@@ -167,6 +167,8 @@
 libjanet = library('janet', janetc,
   include_directories : incdir,
   dependencies : [m_dep, dl_dep, thread_dep],
+  version: meson.project_version(),
+  soversion: version_parts[0] + '.' + version_parts[1],
   install : true)
 
 # Extra c flags - adding -fvisibility=hidden matches the Makefile and
@@ -184,8 +186,9 @@
   extra_cross_cflags = []
 endif
 
-janet_mainclient = executable('janet', janetc, mainclient_src,
+janet_mainclient = executable('janet', mainclient_src,
   include_directories : incdir,
   dependencies : [m_dep, dl_dep, thread_dep],
   c_args : extra_native_cflags,
+  link_with : [libjanet],
   install : true)
