--- paxldso.h.orig
+++ paxldso.h
@@ -53,16 +53,15 @@
 extern array_t *ldpaths;
 extern int ldso_config_load(const char *fname);
 #else
+static array_t _ldpaths = array_init_decl;
+static array_t *ldpaths = &_ldpaths;
+
 static inline int ldso_config_load(const char *fname)
 {
 	return 0;
 }
 #endif
 
-#if PAX_LDSO_CACHE || PAX_LDSO_CONFIG
 extern void paxldso_cleanup(void);
-#else
-# define paxldso_cleanup()
-#endif
 
 #endif
