--- build/mixxx.py.orig	2014-08-27 20:20:09.652848211 +0200
+++ build/mixxx.py	2014-08-27 20:24:18.551421642 +0200
@@ -42,6 +42,7 @@
         if machine not in ['x86_64', 'x86', 'i686', 'i586',
                            'alpha', 'hppa', 'mips', 'mipsel', 's390',
                            'sparc', 'ia64', 'armel', 'armhf', 'hurd-i386',
+                           'armv5l', 'armv6l', 'armv7l',
                            'sh3', 'sh4',
                            'kfreebsd-amd64', 'kfreebsd-i386',
                            'i486', 'i386', 'powerpc', 'powerpc64',
