--- pw-inspector.c.orig	2017-07-21 11:35:56.000000000 +0000
+++ pw-inspector.c	2017-09-17 09:14:00.560000916 +0000
@@ -1,4 +1,5 @@
 #include <stdio.h>
+#include <stdint.h>
 #include <unistd.h>
 #include <stdlib.h>
 #include <string.h>
@@ -36,7 +37,7 @@
   exit(-1);
 }
 
-int32_t main(int32_t argc, char *argv[]) {
+int main(int argc, char *argv[]) {
   int32_t i, j, k;
   int32_t sets = 0, countsets = 0, minlen = 0, maxlen = MAXLENGTH, count = 0;
   int32_t set_low = 0, set_up = 0, set_no = 0, set_print = 0, set_other = 0;
