diff --git a/common/filesystem.cpp b/common/filesystem.cpp
index b2b8d65b4..5287a0c83 100644
--- a/common/filesystem.cpp
+++ b/common/filesystem.cpp
@@ -75,6 +75,9 @@ UString getAbsPath(const UString & path)
 #else
 #include <unistd.h>
 #include <stdlib.h>
+#if !defined(ACCESSPERMS)
+#define ACCESSPERMS (S_IRWXU|S_IRWXG|S_IRWXO)
+#endif
 bool isExistOnFs(const UString & path) 
 {
     struct stat buf;
@@ -103,4 +106,4 @@ UString getAbsPath(const UString & path) {
         return UString(abs);
     return path;
 }
-#endif
\ No newline at end of file
+#endif
