From 9da77ae1fbfd3f3918e1ee76b84c6519b53ab9e7 Mon Sep 17 00:00:00 2001
From: Daniel Kolesa <daniel@octaforge.org>
Date: Sat, 12 Dec 2020 05:05:04 +0100
Subject: [PATCH] void arch-generic plumbing bits

---
 Makefile | 14 +++++++++++++-
 1 file changed, 13 insertions(+), 1 deletion(-)

--- rtl8812au-dkms-20210629.orig/Makefile
+++ rtl8812au-dkms-20210629/Makefile
@@ -147,7 +147,8 @@ CONFIG_LAYER2_ROAMING = y
 #bit0: ROAM_ON_EXPIRED, #bit1: ROAM_ON_RESUME, #bit2: ROAM_ACTIVE
 CONFIG_ROAMING_FLAG = 0x3
 ###################### Platform Related #######################
-CONFIG_PLATFORM_I386_PC = y
+CONFIG_PLATFORM_VOID_NATIVE = y
+CONFIG_PLATFORM_I386_PC = n
 CONFIG_PLATFORM_ARM_RPI = n
 CONFIG_PLATFORM_ARM64_RPI = n
 CONFIG_PLATFORM_ANDROID_X86 = n
@@ -1355,6 +1356,17 @@ EXTRA_CFLAGS += -DCONFIG_RTW_MBO -DCONFI
 EXTRA_CFLAGS += -DCONFIG_RTW_80211R
 endif
 
+ifeq ($(CONFIG_PLATFORM_VOID_NATIVE), y)
+EXTRA_CFLAGS += -DCONFIG_@@VOID_ENDIAN@@_ENDIAN
+EXTRA_CFLAGS += -DCONFIG_IOCTL_CFG80211 -DRTW_USE_CFG80211_STA_EVENT
+ARCH := @@VOID_ARCH@@
+KVER ?= $(shell uname -r)
+KSRC ?= /usr/lib/modules/$(KVER)/build
+MODDESTDIR := /usr/lib/modules/$(KVER)/kernel/drivers/net/wireless/
+INSTALL_PREFIX :=
+STAGINGMODDIR := /usr/lib/modules/$(KVER)/kernel/drivers/staging
+endif
+
 ifeq ($(CONFIG_PLATFORM_I386_PC), y)
 ifeq ($(CONFIG_CONCURRENT_MODE), y)
 EXTRA_CFLAGS += -DCONFIG_CONCURRENT_MODE
