--- src/linux_sysfs.c
+++ src/linux_sysfs.c
@@ -44,6 +44,7 @@
 #include <sys/mman.h>
 #include <dirent.h>
 #include <errno.h>
+#include <limits.h>
 
 #if defined(__i386__) || defined(__x86_64__) || defined(__arm__)
 #include <sys/io.h>
--- src/pciaccess_private.h
+++ src/pciaccess_private.h
@@ -42,7 +42,7 @@
  * know of any OS we support where this isn't available in a sufficiently
  * new version, so warn unconditionally.
  */
-#include <sys/fcntl.h>
+#include <fcntl.h>
 
 #ifndef O_CLOEXEC
 #warning O_CLOEXEC not available, please upgrade.
