--- a/src/modules/localization.rs
+++ b/src/modules/localization.rs
@@ -265,19 +265,3 @@ impl Locales {
 		Ok(date)
 	}
 }
-
-#[cfg(test)]
-mod tests {
-	use super::*;
-
-	#[tokio::test]
-	async fn translate_string() -> Result<()> {
-		let (target_lang, input) = ("de_DE", "tongue-twister");
-
-		let res = Locales::translate_str(target_lang, input).await?;
-
-		assert!(res.contains("Zungenbrecher"));
-
-		Ok(())
-	}
-}
