--- a/test/hotspot/jtreg/runtime/TLS/exestack-tls.c
+++ b/test/hotspot/jtreg/runtime/TLS/exestack-tls.c
@@ -26,7 +26,6 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <gnu/libc-version.h>
 
 // Declare the thread local variable(s) in the main executable. This can be
 // used to demonstrate the issues associated with the on-stack static TLS blocks
@@ -56,13 +55,6 @@
 
 // glibc 2.15 introduced __pthread_get_minstack
 int glibc_has_pthread_get_minstack() {
-  const char* glibc_vers = gnu_get_libc_version();
-  const int glibc_vers_major = atoi(glibc_vers);
-  const int glibc_vers_minor = atoi(strchr(glibc_vers, '.') + 1);;
-  printf("GNU libc version: %s\n", glibc_vers);
-  if ((glibc_vers_major > 2) || ((glibc_vers_major == 2) && (glibc_vers_minor >= 15))) {
-    return 1;
-  }
   printf("This version does not provide __pthread_get_minstack\n");
   return 0;
 }
