--- src/player/mpvobject.cpp	2018-10-15 16:18:14.000000000 +0200
+++ -	2019-01-07 16:58:32.132459477 +0100
@@ -103,8 +103,10 @@
 };
 
 MpvObject::MpvObject(QQuickItem * parent)
-    : QQuickFramebufferObject(parent), mpv{mpv_create()}, mpv_gl(nullptr)
+    : QQuickFramebufferObject(parent), mpv_gl(nullptr)
 {
+    std::setlocale(LC_NUMERIC, "C");
+    mpv = mpv_create();
     if (!mpv)
         throw std::runtime_error("could not create mpv context");
 
