diff -up iputils-s20101006/clockdiff.c.unused iputils-s20101006/clockdiff.c
--- iputils-s20101006/clockdiff.c.unused	2011-02-09 16:17:33.988298995 +0100
+++ iputils-s20101006/clockdiff.c	2011-02-09 16:17:52.192299138 +0100
@@ -126,7 +126,7 @@ measure(struct sockaddr_in * addr)
 	int msgcount;
 	int cc, count;
 	fd_set ready;
-	long sendtime, recvtime, histime,  histime1;
+	long sendtime, recvtime, histime;
 	long min1, min2, diff;
 	long delta1, delta2;
 	struct timeval tv1, tout;
@@ -240,7 +240,6 @@ empty:
 			  rtt_sigma = (rtt_sigma *3 + abs(diff-rtt))/4;
 			  msgcount++;
 			  histime = ntohl(((__u32*)(icp+1))[1]);
-			  histime1 = ntohl(((__u32*)(icp+1))[2]);
 		/*
 		 * a hosts using a time format different from
 		 * ms. since midnight UT (as per RFC792) should
diff -up iputils-s20101006/ping6.c.unused iputils-s20101006/ping6.c
--- iputils-s20101006/ping6.c.unused	2011-02-09 16:14:11.413299738 +0100
+++ iputils-s20101006/ping6.c	2011-02-09 16:46:39.158299066 +0100
@@ -1215,7 +1215,7 @@ void pr_niquery_reply_name(struct ni_hdr
 	}
 	while (p < end) {
 		int fqdn = 1;
-		int len;
+		int buf_len;
 		int i;
 
 		memset(buf, 0xff, sizeof(buf));
@@ -1230,10 +1230,10 @@ void pr_niquery_reply_name(struct ni_hdr
 		}
 		if (p + ret < end && *(p + ret) == '\0')
 			fqdn = 0;
-		len = strlen(buf);
+		buf_len = strlen(buf);
 
 		putchar(' ');
-		for (i = 0; i < strlen(buf); i++)
+		for (i = 0; i < buf_len; i++)
 			putchar_safe(buf[i]);
 		if (fqdn)
 			putchar('.');
