diff -Naur iputils-s20150815.orig/arping.c iputils-s20150815/arping.c
--- iputils-s20150815.orig/arping.c	2015-08-15 07:07:28.000000000 -0700
+++ iputils-s20150815/arping.c	2016-03-12 14:03:20.635521646 -0800
@@ -45,6 +45,10 @@
 #ifdef USE_IDN
 #include <idna.h>
 #include <locale.h>
+
+#ifndef AI_IDN
+#define AI_IDN 0x0040
+#endif
 #endif
 
 #include "SNAPSHOT.h"
diff -Naur iputils-s20150815.orig/ping.h iputils-s20150815/ping.h
--- iputils-s20150815.orig/ping.h	2015-08-15 07:07:28.000000000 -0700
+++ iputils-s20150815/ping.h	2016-03-12 14:07:13.777715944 -0800
@@ -38,6 +38,14 @@
 #include <locale.h>
 #include <idna.h>
 #include <stringprep.h>
+
+#ifndef AI_IDN
+#define AI_IDN 0x0040
+#endif
+#ifndef NI_IDN
+#define NI_IDN 32
+#endif
+
 #define getaddrinfo_flags (AI_CANONNAME | AI_IDN | AI_CANONIDN)
 #define getnameinfo_flags NI_IDN
 #else
diff -Naur iputils-s20150815.orig/tracepath.c iputils-s20150815/tracepath.c
--- iputils-s20150815.orig/tracepath.c	2015-08-15 07:07:28.000000000 -0700
+++ iputils-s20150815/tracepath.c	2016-03-12 14:06:06.515082879 -0800
@@ -29,6 +29,13 @@
 #include <idna.h>
 #include <locale.h>
 
+#ifndef AI_IDN
+#define AI_IDN 0x0040
+#endif
+#ifndef NI_IDN
+#define NI_IDN 32
+#endif
+
 #define getnameinfo_flags	NI_IDN
 #else
 #define getnameinfo_flags	0
diff -Naur iputils-s20150815.orig/tracepath6.c iputils-s20150815/tracepath6.c
--- iputils-s20150815.orig/tracepath6.c	2015-08-15 07:07:28.000000000 -0700
+++ iputils-s20150815/tracepath6.c	2016-03-12 14:07:01.679602078 -0800
@@ -30,6 +30,13 @@
 #ifdef USE_IDN
 #include <idna.h>
 #include <locale.h>
+
+#ifndef AI_IDN
+#define AI_IDN 0x0040
+#endif
+#ifndef NI_IDN
+#define NI_IDN 32
+#endif
 #endif
 
 #ifndef SOL_IPV6
diff -Naur iputils-s20150815.orig/traceroute6.c iputils-s20150815/traceroute6.c
--- iputils-s20150815.orig/traceroute6.c	2015-08-15 07:07:28.000000000 -0700
+++ iputils-s20150815/traceroute6.c	2016-03-12 14:08:29.388427580 -0800
@@ -250,6 +250,10 @@
 #include <idna.h>
 #include <locale.h>
 
+#ifndef NI_IDN
+#define NI_IDN 32
+#endif
+
 #define getnameinfo_flags	NI_IDN
 #else
 #define getnameinfo_flags	0

