--- iputils/ping_common.c.countermeasures	Tue May 21 10:06:05 2002
+++ iputils/ping_common.c	Tue May 21 10:12:42 2002
@@ -628,7 +628,8 @@
 		tvsub(tv, &tmp_tv);
 		triptime = tv->tv_sec * 1000000 + tv->tv_usec;
 		if (triptime < 0) {
-			fprintf(stderr, "Warning: time of day goes back (%ldus), taking countermeasures.\n", triptime);
+			if (options & F_VERBOSE)
+				fprintf(stderr, "Warning: time of day goes back (%ldus), taking countermeasures.\n", triptime);
 			triptime = 0;
 			if (!(options & F_LATENCY)) {
 				gettimeofday(tv, NULL);
