diff -up iputils-s20101006/doc/ping.sgml.man iputils-s20101006/doc/ping.sgml
--- iputils-s20101006/doc/ping.sgml.man	2011-03-29 08:53:09.362500777 +0200
+++ iputils-s20101006/doc/ping.sgml	2011-03-29 08:58:03.196423782 +0200
@@ -62,8 +62,8 @@ Audible ping.
   <term><option/-A/</term>
   <listitem><para>
 Adaptive ping. Interpacket interval adapts to round-trip time, so that
-effectively not more than one (or more, if preload is set) unanswered probes
-present in the network. Minimal interval is 200msec for not super-user.
+effectively not more than one (or more, if preload is set) unanswered probe
+is present in the network. Minimal interval is 200msec for not super-user.
 On networks with low rtt this mode is essentially equivalent to flood mode.  
   </para></listitem>
  </varlistentry>
@@ -394,7 +394,7 @@ probes are answered or for some error no
   <term><option>-W <replaceable/timeout/</option></term>
   <listitem><para>
 Time to wait for a response, in seconds. The option affects only timeout
-in absense of any responses, otherwise <command/ping/ waits for two RTTs.
+in absence of any responses, otherwise <command/ping/ waits for two RTTs.
   </para></listitem>
  </varlistentry>
 </variablelist>
@@ -535,7 +535,7 @@ or
 </para>
 
 <para>
-In normal operation ping prints the ttl value from the packet it receives.
+In normal operation ping prints the TTL value from the packet it receives.
 When a remote system receives a ping packet, it can do one of three things
 with the TTL field in its response:
 </para>
diff -up iputils-s20101006/doc/tracepath.sgml.man iputils-s20101006/doc/tracepath.sgml
--- iputils-s20101006/doc/tracepath.sgml.man	2011-03-29 08:58:31.835723958 +0200
+++ iputils-s20101006/doc/tracepath.sgml	2011-03-29 09:01:41.706767424 +0200
@@ -35,7 +35,7 @@ privileges and has no fancy options.
 <command/tracepath6/ is good replacement for <command/traceroute6/
 and classic example of application of Linux error queues.
 The situation with IPv4 is worse, because commercial
-IP routers do not return enough information in icmp error messages.
+IP routers do not return enough information in ICMP error messages.
 Probably, it will change, when they will be updated.
 For now it uses Van Jacobson's trick, sweeping a range
 of UDP ports to maintain trace history.
@@ -96,7 +96,7 @@ the probe was not sent to the network.
 
 <para>
 The rest of line shows miscellaneous information about path to
-the correspinding hetwork hop. As rule it contains value of RTT.
+the correspinding network hop. As rule it contains value of RTT.
 Additionally, it can show Path MTU, when it changes.
 If the path is asymmetric
 or the probe finishes before it reach prescribed hop, difference
