diff -up iputils-s20100418/tracepath6.c.convtoint iputils-s20100418/tracepath6.c
--- iputils-s20100418/tracepath6.c.convtoint	2010-04-23 11:28:15.294593391 +0200
+++ iputils-s20100418/tracepath6.c	2010-04-23 11:28:23.544605551 +0200
@@ -89,7 +89,7 @@ void print_host(const char *a, const cha
 	}
 	if (plen >= HOST_COLUMN_SIZE)
 		plen = HOST_COLUMN_SIZE - 1;
-	printf("%*s", HOST_COLUMN_SIZE - plen, "");
+	printf("%*s", HOST_COLUMN_SIZE - (int)plen, "");
 }
 
 int recverr(int fd, int ttl)
diff -up iputils-s20100418/tracepath.c.convtoint iputils-s20100418/tracepath.c
--- iputils-s20100418/tracepath.c.convtoint	2010-04-23 11:26:20.273555629 +0200
+++ iputils-s20100418/tracepath.c	2010-04-23 11:26:20.281562493 +0200
@@ -77,7 +77,7 @@ void print_host(const char *a, const cha
 	}
 	if (plen >= HOST_COLUMN_SIZE)
 		plen = HOST_COLUMN_SIZE - 1;
-	printf("%*s", HOST_COLUMN_SIZE - plen, "");
+	printf("%*s", HOST_COLUMN_SIZE - (int)plen, "");
 }
 
 int recverr(int fd, int ttl)
