diff -Naur iputils-s20121221.orig/arping.c iputils-s20121221/arping.c
--- iputils-s20121221.orig/arping.c	2012-12-21 14:01:07.000000000 +0000
+++ iputils-s20121221/arping.c	2014-01-23 21:19:57.091268011 +0000
@@ -16,7 +16,7 @@
 #include <linux/sockios.h>
 #include <sys/file.h>
 #include <sys/time.h>
-#include <sys/signal.h>
+#include <signal.h>
 #include <sys/ioctl.h>
 #include <net/if.h>
 #include <linux/if_packet.h>
--- iputils-s20121221.orig/clockdiff.c	2012-12-21 14:01:07.000000000 +0000
+++ iputils-s20121221/clockdiff.c	2014-01-23 21:19:57.093268011 +0000
@@ -14,7 +14,6 @@
 #include <netinet/ip.h>
 #include <netinet/ip_icmp.h>
 #define TSPTYPES
-#include <protocols/timed.h>
 #include <fcntl.h>
 #include <netdb.h>
 #include <arpa/inet.h>
diff -Naur iputils-s20121221.orig/ping_common.h iputils-s20121221/ping_common.h
--- iputils-s20121221.orig/ping_common.h	2012-12-21 14:01:07.000000000 +0000
+++ iputils-s20121221/ping_common.h	2014-01-23 21:27:07.677292088 +0000
@@ -7,16 +7,17 @@
 #include <linux/sockios.h>
 #include <sys/file.h>
 #include <sys/time.h>
-#include <sys/signal.h>
+#include <signal.h>
 #include <sys/ioctl.h>
 #include <net/if.h>
 #include <sys/uio.h>
-#include <sys/poll.h>
+#include <poll.h>
 #include <ctype.h>
 #include <errno.h>
 #include <string.h>
 #include <netdb.h>
 #include <setjmp.h>
+#include <asm-generic/param.h>
 
 #ifdef CAPABILITIES
 #include <sys/prctl.h>
--- iputils-s20121221.orig/rarpd.c	2014-01-24 13:38:46.816551953 +0000
+++ iputils-s20121221/rarpd.c	2014-01-24 13:36:51.833545524 +0000
@@ -19,16 +19,17 @@
 #include <netdb.h>
 #include <arpa/inet.h>
 #include <sys/ioctl.h>
-#include <sys/poll.h>
-#include <sys/errno.h>
-#include <sys/fcntl.h>
+#include <poll.h>
+#include <errno.h>
+#include <fcntl.h>
 #include <sys/socket.h>
-#include <sys/signal.h>
+#include <signal.h>
 #include <linux/if.h>
 #include <linux/if_arp.h>
 #include <netinet/in.h>
 #include <linux/if_packet.h>
 #include <linux/filter.h>
+#include <sys/types.h>
 
 int do_reload = 1;
 
--- iputils-s20121221.orig/tracepath.c	2014-01-24 13:38:46.865551956 +0000
+++ iputils-s20121221/tracepath.c	2014-01-23 21:19:57.109268012 +0000
@@ -23,6 +23,8 @@
 #include <sys/time.h>
 #include <sys/uio.h>
 #include <arpa/inet.h>
+#include <sys/types.h>
+#include <limits.h>
 #ifdef USE_IDN
 #include <idna.h>
 #include <locale.h>
--- iputils-s20121221.orig/tracepath6.c	2014-01-24 13:38:46.865551956 +0000
+++ iputils-s20121221/tracepath6.c	2014-01-23 21:20:40.316270428 +0000
@@ -25,6 +25,7 @@
 #include <sys/time.h>
 #include <sys/uio.h>
 #include <arpa/inet.h>
+#include <limits.h>
 
 #ifdef USE_IDN
 #include <idna.h>
--- a/tracepath.c.orig    2014-11-10 16:57:16.549029140 +0100
+++ b/tracepath.c 2014-11-10 16:57:37.996240185 +0100
@@ -13,6 +13,7 @@
 #include <stdlib.h>
 #include <unistd.h>
 #include <sys/socket.h>
+#include <sys/types.h>
 #include <linux/types.h>
 #include <linux/errqueue.h>
 #include <errno.h>
@@ -23,7 +24,6 @@
 #include <sys/time.h>
 #include <sys/uio.h>
 #include <arpa/inet.h>
-#include <sys/types.h>
 #include <limits.h>
 #ifdef USE_IDN
 #include <idna.h>
