diff -Naur iputils-s20121221.orig/rdisc.c iputils-s20121221/rdisc.c
--- iputils-s20121221.orig/rdisc.c	2014-01-24 14:59:29.937822762 +0000
+++ iputils-s20121221/rdisc.c	2014-01-23 21:25:58.760288234 +0000
@@ -115,26 +114,7 @@
 
 #define MAXIFS 32
 
-#if !defined(__GLIBC__) || __GLIBC__ < 2
-/* For router advertisement */
-struct icmp_ra
-{
-	u_char	icmp_type;		/* type of message, see below */
-	u_char	icmp_code;		/* type sub code */
-	u_short	icmp_cksum;		/* ones complement cksum of struct */
-	u_char	icmp_num_addrs;
-	u_char	icmp_wpa;		/* Words per address */
-	short 	icmp_lifetime;
-};
-
-struct icmp_ra_addr
-{
-	__u32	ira_addr;
-	__u32	ira_preference;
-};
-#else
 #define icmp_ra icmp
-#endif
 
 /* Router constants */
 #define	MAX_INITIAL_ADVERT_INTERVAL	16
