--- libraries/libldap/tls_o.c.orig	2014-07-15 18:08:35.841431410 +0200
+++ libraries/libldap/tls_o.c	2014-07-15 18:09:10.681720371 +0200
@@ -1128,10 +1128,11 @@ tlso_seed_PRNG( const char *randfile )
 		 * The fact is that when $HOME is NULL, .rnd is used.
 		 */
 		randfile = RAND_file_name( buffer, sizeof( buffer ) );
-
+#ifdef HAVE_SSL_RAND_EGD
 	} else if (RAND_egd(randfile) > 0) {
 		/* EGD socket */
 		return 0;
+#endif
 	}
 
 	if (randfile == NULL) {
