diff --git a/python/tests/_test_utils.py b/python/tests/_test_utils.py
index 104e654..6cf12b0 100644
--- a/python/tests/_test_utils.py
+++ b/python/tests/_test_utils.py
@@ -36,10 +36,7 @@ TESTDATA_DIR = os.path.join(test_dir, 'testdata')
 
 TESTDATA_FILES = [
     'empty',  # Empty file
-    '10x10y',  # Small text
-    'alice29.txt',  # Large text
-    'random_org_10k.bin',  # Small data
-    'mapsdatazrh',  # Large data
+    'ukkonooa',  # Small text
 ]
 
 TESTDATA_PATHS = [os.path.join(TESTDATA_DIR, f) for f in TESTDATA_FILES]
