diff --git a/Cargo.toml b/Cargo.toml
index e84724f..9a01b29 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -20,7 +20,7 @@ codegen-units = 1
 panic = "abort"
 
 [dependencies]
-git2 = { version = "0.18.1", default-features = false, features = [] }
+git2 = { version = "0.17", default-features = false, features = [] }
 anyhow = "^1"
 colored = "^2"
 chrono = { version = "0.4.19", features = ["serde"] }
