diff --git a/module/boot@/service/local/firewall/local-iptables b/module/boot@/service/local/firewall/local-iptables
index 2ba35fb7e02b122b93f00c023ee673fbbb7b1bbe..27e224344d3c70be674a018124d1f5490a43f8f2 100644
--- a/module/boot@/service/local/firewall/local-iptables
+++ b/module/boot@/service/local/firewall/local-iptables
@@ -20,19 +20,28 @@
 )
 
 [stop]
+@build = custom
+@shebang = "/usr/bin/bash -c"
 @execute =
 (
-    if { 66-yeller -cdp local-iptables -1 /dev/console stops... }
-    if -nt {
 
-        execl-toc -x ${script_file}
-        ${script_file}
+    exec 2>&1
 
-    }
-    66-yeller -fcdp local-iptables -1 /dev/console crashed!
+    66-yeller -cdp local-iptables -1 /dev/console stops...
+
+    if ! type -p iptables &>/dev/null; then
+      66-yeller -fcdp local-iptables -1 /dev/console "unable to find iptables program"
+    fi
+
+    while read -r table; do
+        tables+=("/usr/share/iptables/empty-$table.rules")
+    done <"/proc/net/ip_tables_names"
+
+    if (( ${#tables[*]} )); then
+        cat "${tables[@]}" | iptables-restore
+    fi
 )
 
 [environment]
 script_conf=!/etc/iptables/iptables.rules
-script_file=!/usr/lib/iptables/scripts/iptables-flush
 
