diff --git a/configure b/configure
index fba297a..7a0a8da 100755
--- a/configure
+++ b/configure
@@ -23,8 +23,8 @@ Fine tuning of the installation directories:
   --with-system-module=DIR         66 module intallation directory [DATAROOTDIR/$package_macro_dir/module]
   --with-system-script=DIR         66 script installation directory of scripts [DATAROOTDIR/$package_macro_dir/script]
   --with-sysadmin-service-conf=DIR 66 sysadmin service configuration file directory [SYSDIR/$package_macro_dir/conf]
-  --with-skeleton=DIR              66 skeleton files directory installation [SYSDIR/$package_macro_dir]	
-  
+  --with-skeleton=DIR              66 skeleton files directory installation [SYSDIR/$package_macro_dir]
+
 Fine tunning of boot configuration:
   --HOSTNAME=VALUE                 hostname of the system [obarunS6]
   --HARDWARECLOCK=VALUE            set the hardware clock [UTC]
@@ -50,7 +50,7 @@ Fine tunning of boot configuration:
   --DMRAID=BOOLEAN                 mount dmraid devices [!no]
   --BTRFS=BOOLEAN                  mount btrfs devices [!no]
   --ZFS=BOOLEAN                    mount zfs devices [!no]
-  --ZFS_IMPORT=VALUE               use scan or zpoolcache method for zfs pools importation [!scan]
+  --ZFS_IMPORT=VALUE               use scan or zpoolcache method for zfs pools importation [!zpoolcache]
   --CRYPTTAB=BOOLEAN               use crypttab by default [!no]
   --FIREWALL=VALUE                 use iptables|ip6tables|nftables|ebtables|arptables []
   --CGROUPS=BOOLEAN                mount cgroups [!yes]
@@ -66,17 +66,17 @@ Fine tunning of boot configuration:
   --POPULATE_DEV=BOOLEAN           populate dev directory [!yes]
   --POPULATE_RUN=BOOLEAN           populate run directory [!yes]
   --POPULATE_TMP=BOOLEAN           populate tmp directory [!yes]
-  
- 
+
+
  Use yes or no with lowercase character to set BOOLEAN values.
- 
+
  Empty value is a commented one and not used at all.
 
  An exclamation mark '!' at the begin of the value unexport the variable
  for the environment after the use. This is avoid to poluate the environment
  with tempory variables. You need to single-quote the options as
  follow:
-	./configure --KEYMAP='!us'
+    ./configure --KEYMAP='!us'
 
 EOF
 exit 0
@@ -120,9 +120,9 @@ stripdir () {
 
 testval () {
  if test -z ${1}; then
-	return 0 ;
+    return 0 ;
  else
-	return 1 ;
+    return 1 ;
  fi
 }
 
@@ -165,7 +165,7 @@ LVM='!no'
 DMRAID='!no'
 BTRFS='!no'
 ZFS='!no'
-ZFS_IMPORT='!scan'
+ZFS_IMPORT='!zpoolcache'
 CRYPTTAB='!no'
 FIREWALL=
 CGROUPS='!yes'
@@ -256,20 +256,20 @@ fi
 
 if test -z "$sysconfdir" ; then
   if test "$sysconfdir" = '$sysconfdir' ; then
-	sysconfdir = '/etc'
+    sysconfdir = '/etc'
   fi
   if test "$adm_conf" = '$sysconfdir/$package_macro_dir/conf' ; then
-	adm_conf = '$sysconfdir/$package_macro_dir/conf'
+    adm_conf = '$sysconfdir/$package_macro_dir/conf'
   fi
   if test "$skel_directory" = '$sysconfdir/$package_macro_dir' ; then
-	skel_directory = '$sysconfdir/$package_macro_dir'
+    skel_directory = '$sysconfdir/$package_macro_dir'
   fi
 fi
 
 # Expand installation directories
 stripdir prefix
 for i in exec_prefix bindir libdir sysconfdir datarootdir mandir livedir \
-		service_directory module_directory script_directory skel_directory adm_conf ; do
+        service_directory module_directory script_directory skel_directory adm_conf ; do
   eval tmp=\${$i}
   eval $i=$tmp
   stripdir $i
@@ -290,12 +290,12 @@ for i in service_directory module_directory script_directory mandir; do
 done
 
 for i in HOSTNAME HARDWARECLOCK TZ SETUPCONSOLE TTY \
-		KEYMAP FONT FONT_MAP FONT_UNIMAP UDEV SYSCTL FORCECHCK LOCAL CONTAINER \
-		TMPFILE MODULE_KERNEL MODULE_SYSTEM RANDOMSEED \
-		FSTAB SWAP LVM DMRAID BTRFS ZFS ZFS_IMPORT \
-		CRYPTTAB FIREWALL \
-		CGROUPS MNT_PROC MNT_SYS MNT_DEV MNT_RUN MNT_TMP \
-		MNT_PTS MNT_SHM MNT_NETFS POPULATE_SYS POPULATE_DEV POPULATE_RUN POPULATE_TMP ; do
+        KEYMAP FONT FONT_MAP FONT_UNIMAP UDEV SYSCTL FORCECHCK LOCAL CONTAINER \
+        TMPFILE MODULE_KERNEL MODULE_SYSTEM RANDOMSEED \
+        FSTAB SWAP LVM DMRAID BTRFS ZFS ZFS_IMPORT \
+        CRYPTTAB FIREWALL \
+        CGROUPS MNT_PROC MNT_SYS MNT_DEV MNT_RUN MNT_TMP \
+        MNT_PTS MNT_SHM MNT_NETFS POPULATE_SYS POPULATE_DEV POPULATE_RUN POPULATE_TMP ; do
   eval tmp=\${$i}
   if testval $tmp; then
     eval $i='${i}=\"\\#${i}=\"'
diff --git a/module/boot@/service/local/firewall/local-iptables b/module/boot@/service/local/firewall/local-iptables
index 2ba35fb..4364a2f 100644
--- a/module/boot@/service/local/firewall/local-iptables
+++ b/module/boot@/service/local/firewall/local-iptables
@@ -20,19 +20,29 @@
 )
 
 [stop]
+@build = custom
+@shebang = "/usr/bin/bash -c"
 @execute =
 (
-    if { 66-yeller -cdp local-iptables -1 /dev/console stops... }
-    if -nt {
 
-        execl-toc -x ${script_file}
-        ${script_file}
+    exec 2>&1
 
-    }
-    66-yeller -fcdp local-iptables -1 /dev/console crashed!
+    66-yeller -cdp local-iptables -1 /dev/console stops...
+
+    if ! type -p iptables &>/dev/null; then
+      66-yeller -fcdp local-iptables -1 /dev/console unable to find iptables program
+      exit 111
+    fi
+
+    while read -r table; do
+        tables+=("/usr/share/iptables/empty-$table.rules")
+    done <"/proc/net/ip_tables_names"
+
+    if (( ${#tables[*]} )); then
+        cat "${tables[@]}" | iptables-restore
+    fi
 )
 
 [environment]
 script_conf=!/etc/iptables/iptables.rules
-script_file=!/usr/lib/iptables/scripts/iptables-flush
 
diff --git a/module/boot@/service/local/local-dmesg b/module/boot@/service/local/local-dmesg
index 934cd9f..fe38ef4 100644
--- a/module/boot@/service/local/local-dmesg
+++ b/module/boot@/service/local/local-dmesg
@@ -34,5 +34,5 @@ local-rc )
 )
 
 [environment]
-cmd_args=!--console-off -T
+cmd_args=!-T
 cmd_log=!/var/log/dmesg.log
diff --git a/module/boot@/service/mount/mount-swap b/module/boot@/service/mount/mount-swap
index fc7e2db..1b5b4e8 100644
--- a/module/boot@/service/mount/mount-swap
+++ b/module/boot@/service/mount/mount-swap
@@ -10,12 +10,16 @@
 @execute =
 (
     if { 66-yeller -cdp mount-swap -1 /dev/console starts... }
-    if -nt {
 
-        swapon -a
+    foreground {
+
+        if -nt {
 
+            swapon -a
+
+        }
+        66-yeller -cdp mount-swap -1 /dev/console unable to activate the swap!
     }
-    66-yeller -fcdp mount-swap -1 /dev/console crashed!
 )
 
 [stop]
diff --git a/module/boot@/service/system/devices/devices-crypttab b/module/boot@/service/system/devices/devices-crypttab
index 5b808f2..76ef6d5 100644
--- a/module/boot@/service/system/devices/devices-crypttab
+++ b/module/boot@/service/system/devices/devices-crypttab
@@ -4,7 +4,7 @@
 @description = "Active encrypted devices"
 @user = ( root )
 @options = ( !log )
-@depends = ( udevadm )
+@depends = ( udevadm devices-btrfs devices-dmraid devices-lvm )
 
 [start]
 @execute =
@@ -15,13 +15,16 @@
         execl-toc -e ${script_conf}
         if { execl-cmdline -s { 66-olexec awk -f ${script_file} ${script_conf} } }
 
-        if { 66-which -q vgchange }
-        if { 66-yeller -cdp devices-crypttab -1 /dev/console LVM detected -- activates... }
-        if -nt {
+        foreground {
 
-            vgchange --sysinit -a y
+            if { 66-which -q vgchange }
+            if { 66-yeller -cdp devices-crypttab -1 /dev/console LVM detected -- activates... }
+            if -nt {
+
+                vgchange --sysinit -a y
+            }
+            66-yeller -cdp devices-crypttab -1 /dev/console unable to activate LVM devices
         }
-        66-yeller -cdp devices-crypttab -1 /dev/console unable to activate LVM devices
     }
     66-yeller -fcdp devices-crypttab -1 /dev/console crashed!
 )
