--- ./python/enum.c.orig	2023-09-29 09:38:04.452239909 -0400
+++ ./python/enum.c	2023-09-29 09:38:26.194365475 -0400
@@ -57,7 +57,7 @@
 #else
     Py_SIZE(&self->val) = Py_SIZE(longval);
 #endif
-    self->val.ob_digit[0] = longval->ob_digit[0];
+    self->val.long_value.ob_digit[0] = longval->long_value.ob_digit[0];
     Py_DECREF(longval);
 #else
     self->val.ob_ival = val;
@@ -152,7 +152,7 @@
 #else
     Py_SIZE(&self->val) = Py_SIZE(longval);
 #endif
-    self->val.ob_digit[0] = longval->ob_digit[0];
+    self->val.long_value.ob_digit[0] = longval->long_value.ob_digit[0];
     Py_DECREF(longval);
 
     self->name = PyUnicode_FromString(name);
