--- usr/kinit/nfsmount/mount.c.orig	2009-10-16 02:59:38.000000000 +0200
+++ usr/kinit/nfsmount/mount.c	2009-10-16 03:00:08.000000000 +0200
@@ -44,6 +44,11 @@ struct mount_reply {
 
 #define MNT_REPLY_MINSIZE (sizeof(struct rpc_reply) + sizeof(uint32_t))
 
+#ifndef MNTPROC_MNT
+#  define MNTPROC_MNT	1
+#  define MNTPROC_UMNT	3
+#endif /* MNTPROC_MNT */
+
 static int get_ports(uint32_t server, const struct nfs_mount_data *data)
 {
 	uint32_t nfs_ver, mount_ver;
