--- configure.old	2020-03-20 16:11:07.048590348 +0000
+++ configure	2020-03-20 16:21:33.611151349 +0000
@@ -4865,19 +4865,14 @@
 export CROSSCOMPILE
 
 if test "x${CROSSCOMPILE}" = "x1"; then
- { $as_echo "$as_me:${as_lineno-$LINENO}: result: we are cross compiling- trying to guess from system type" >&5
-$as_echo "we are cross compiling- trying to guess from system type" >&6; }
-
- { $as_echo "$as_me:${as_lineno-$LINENO}: checking for byte order, try 2" >&5
-$as_echo_n "checking for byte order, try 2... " >&6; }
- case "${target}" in
-  *86* | *-pc-* )
-     LITTLEENDIAN=1
-     ;;
-  *sparc* | *68* | *88k* | *mac* | *Mac* | *sun* | *Sun* | *Amiga* | *amiga* )
-     BIGENDIAN=1
-     ;;
- esac
+  case $XBPS_TARGET_ENDIAN in
+    le)
+      LITTLEENDIAN=1
+      ;;
+    be)
+      BIGENDIAN=1
+      ;;
+  esac
 fi
 
 export LITTLEENDIAN
