--- Makefile.common.in.old	2020-03-19 23:16:17.660711906 +0000
+++ Makefile.common.in	2020-03-19 23:17:30.256558583 +0000
@@ -54,17 +54,17 @@
 INSTALL_DATA = @INSTALL_DATA@
 UNINSTALL_DATA = rm -f
 INSTALL_PROGRAM = @INSTALL_PROGRAM@
-prefix = @prefix@
+prefix = $(DESTDIR)@prefix@
 exec_prefix = @exec_prefix@
-bindir = @bindir@
+bindir = $(DESTDIR)@bindir@
 datadir = @datadir@
 libdir = @libdir@
 includedir = @includedir@
-sysconfdir = @sysconfdir@
+sysconfdir = $(DESTDIR)@sysconfdir@
 pkgdatadir = $(datadir)/@PACKAGE@
 
 # man pages
-mandir = @mandir@
+mandir = $(DESTDIR)@mandir@
 man1dir = $(mandir)/man1
 
 # ------------------- defines -------------------------
