--- CBLAS/CMakeLists.txt.orig	2016-12-24 00:01:32.000000000 +0100
+++ CBLAS/CMakeLists.txt	2017-01-26 15:24:38.039693320 +0100
@@ -81,5 +81,11 @@
   DESTINATION ${LIBRARY_DIR}/cmake/cblas-${LAPACK_VERSION}
   )
 
+set_target_properties(
+  cblas PROPERTIES
+  VERSION ${LAPACK_VERSION}
+  SOVERSION ${LAPACK_MAJOR_VERSION}
+  )
+
 #install(EXPORT cblas-targets
 #  DESTINATION ${LIBRARY_DIR}/cmake/cblas-${LAPACK_VERSION})
--- LAPACKE/CMakeLists.txt.orig	2016-12-24 00:01:32.000000000 +0100
+++ LAPACKE/CMakeLists.txt	2017-01-26 15:26:04.446690679 +0100
@@ -81,5 +81,11 @@
   DESTINATION ${LIBRARY_DIR}/cmake/lapacke-${LAPACK_VERSION}
   )
 
+set_target_properties(
+  lapacke PROPERTIES
+  VERSION ${LAPACK_VERSION}
+  SOVERSION ${LAPACK_MAJOR_VERSION}
+  )
+
 install(EXPORT lapacke-targets
   DESTINATION ${LIBRARY_DIR}/cmake/lapacke-${LAPACK_VERSION})
