diff --git BLAS/SRC/CMakeLists.txt BLAS/SRC/CMakeLists.txt
index 7d8066c..841e28f 100644
--- BLAS/SRC/CMakeLists.txt
+++ BLAS/SRC/CMakeLists.txt
@@ -137,6 +137,7 @@ endif()
   
   
 add_library(blas ${ALLOBJ})
+add_library(blas_static STATIC ${ALLOBJ})
 #if(UNIX)
 #  target_link_libraries(blas m)
 #endif()
@@ -145,5 +146,8 @@ set_target_properties(
   VERSION ${LAPACK_VERSION}
   SOVERSION ${LAPACK_MAJOR_VERSION}
   )
+set_target_properties(blas_static PROPERTIES OUTPUT_NAME blas)
 target_link_libraries(blas)
+target_link_libraries(blas_static)
 lapack_install_library(blas)
+lapack_install_library(blas_static)
diff --git CBLAS/src/CMakeLists.txt CBLAS/src/CMakeLists.txt
index 8093a5c..d4df765 100644
--- CBLAS/src/CMakeLists.txt
+++ CBLAS/src/CMakeLists.txt
@@ -164,5 +164,9 @@ if (CBLAS_COMPLEX16)
 endif(CBLAS_COMPLEX16)
 
 add_library(cblas ${ALLOBJ})
-target_link_libraries(cblas ${BLAS_LIBRARIES} )
+add_library(cblas_static STATIC ${ALLOBJ})
+set_target_properties(cblas_static PROPERTIES OUTPUT_NAME cblas)
+target_link_libraries(cblas ${BLAS_LIBRARIES})
+target_link_libraries(cblas_static ${BLAS_LIBRARIES})
 lapack_install_library(cblas)
+lapack_install_library(cblas_static)
diff --git LAPACKE/CMakeLists.txt LAPACKE/CMakeLists.txt
index 008f24c..807ebf0 100644
--- LAPACKE/CMakeLists.txt
+++ LAPACKE/CMakeLists.txt
@@ -46,18 +46,26 @@ append_subdir_files(UTILS_OBJ "utils")
 
 if (USE_XBLAS)
    add_library(lapacke ${SRC_OBJ} ${SRCX_OBJ} ${UTILS_OBJ})
+   add_library(lapacke_static STATIC ${SRC_OBJ} ${SRCX_OBJ} ${UTILS_OBJ})
    target_link_libraries(lapacke ${LAPACK_LIBRARIES} ${BLAS_LIBRARIES} ${XBLAS_LIBRARY})
+   target_link_libraries(lapacke_static ${LAPACK_LIBRARIES} ${BLAS_LIBRARIES} ${XBLAS_LIBRARY})
 else (USE_XBLAS)
   if (LAPACKE_WITH_TMG)
    add_library(lapacke ${SRC_OBJ} ${MATGEN_OBJ} ${UTILS_OBJ})
+   add_library(lapacke_static STATIC ${SRC_OBJ} ${MATGEN_OBJ} ${UTILS_OBJ})
    target_link_libraries(lapacke tmglib ${LAPACK_LIBRARIES} ${BLAS_LIBRARIES})
+   target_link_libraries(lapacke_static tmglib ${LAPACK_LIBRARIES} ${BLAS_LIBRARIES})
   else (LAPACKE_WITH_TMG)
    add_library(lapacke ${SRC_OBJ} ${UTILS_OBJ})
+   add_library(lapacke_static STATIC ${SRC_OBJ} ${UTILS_OBJ})
    target_link_libraries(lapacke ${LAPACK_LIBRARIES} ${BLAS_LIBRARIES})
+   target_link_libraries(lapacke_static ${LAPACK_LIBRARIES} ${BLAS_LIBRARIES})
   endif(LAPACKE_WITH_TMG)
 endif(USE_XBLAS)
 
+set_target_properties(lapacke_static PROPERTIES OUTPUT_NAME lapacke)
 lapack_install_library(lapacke)
+lapack_install_library(lapacke_static)
 INSTALL( FILES ${LAPACKE_INCLUDE} DESTINATION include )
 
 if(BUILD_TESTING)
diff --git SRC/CMakeLists.txt SRC/CMakeLists.txt
index 03441b9..e6a381b 100644
--- SRC/CMakeLists.txt
+++ SRC/CMakeLists.txt
@@ -455,6 +455,7 @@ endif()
 list(REMOVE_DUPLICATES ALLOBJ)
 
 add_library(lapack ${ALLOBJ} ${ALLXOBJ})
+add_library(lapack_static STATIC ${ALLOBJ} ${ALLXOBJ})
 target_link_libraries(lapack ${BLAS_LIBRARIES} ${XBLAS_LIBRARY})
 
 set_target_properties(
@@ -462,5 +463,7 @@ set_target_properties(
   VERSION ${LAPACK_VERSION}
   SOVERSION ${LAPACK_MAJOR_VERSION}
   )
+set_target_properties(lapack_static PROPERTIES OUTPUT_NAME lapack)
 
 lapack_install_library(lapack)
+lapack_install_library(lapack_static)
