From 18731d05b0fa59d393ca6eceb31c8161eefba5f8 Mon Sep 17 00:00:00 2001
From: John Zimmermann <me@johnnynator.dev>
Date: Sat, 3 Aug 2024 17:41:09 +0200
Subject: [PATCH 2/3] cmake cleanup

---
 linphone-app/CMakeLists.txt | 15 ++++++++-------
 1 file changed, 8 insertions(+), 7 deletions(-)

diff --git a/linphone-app/CMakeLists.txt b/linphone-app/CMakeLists.txt
index cab2cc25..d2b5dcde 100644
--- a/linphone-app/CMakeLists.txt
+++ b/linphone-app/CMakeLists.txt
@@ -105,10 +105,10 @@ foreach(PACKAGE ${LINPHONE_PACKAGES})
 endforeach()
 
 set(PLUGIN_TARGETS ${LinphoneCxx_TARGET})
-set(APP_TARGETS ${LinphoneCxx_TARGET}
+set(APP_TARGETS liblinphone++
 				${BCToolbox_TARGET}#Logger/App
 				${Mediastreamer2_TARGET}#MediastreamerUtils
-				${Belcard_TARGET}#VCard Model
+				${BelCard_TARGET}#VCard Model
 				${LibLinphone_TARGET})#MediastreamerUtils
 
 ####################################
@@ -721,6 +721,7 @@ include_directories("${LINPHONE_OUTPUT_DIR}/include/OpenGL")
 include_directories("${LINPHONE_OUTPUT_DIR}/include/")
 include_directories("${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_INCLUDEDIR}/")
 include_directories("${QTKEYCHAIN_OUTPUT_DIR}/include/")
+include_directories("${CMAKE_CURRENT_SOURCE_DIR}/include/")
 if(NOT APPLE AND NOT WIN32)
 	include_directories("${ISpell_BINARY_DIR}/include")
 endif ()
@@ -830,10 +831,10 @@ add_dependencies(${APP_LIBRARY} update_translations ${APP_PLUGIN})
 set(TOOLS_DIR "${CMAKE_BINARY_DIR}/programs")
 set(LINPHONE_BUILDER_SIGNING_IDENTITY ${LINPHONE_BUILDER_SIGNING_IDENTITY})
 
-add_custom_command(TARGET ${APP_PLUGIN} POST_BUILD COMMAND ${CMAKE_COMMAND} -E make_directory "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}/")
-add_custom_command(TARGET ${APP_PLUGIN} POST_BUILD COMMAND ${CMAKE_COMMAND} -E copy_directory "${CMAKE_CURRENT_SOURCE_DIR}/include/" "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_INCLUDEDIR}/")
-add_custom_command(TARGET ${APP_PLUGIN} POST_BUILD COMMAND ${CMAKE_COMMAND} -E copy $<TARGET_FILE:${APP_PLUGIN}> "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}/")
-add_custom_command(TARGET ${APP_PLUGIN} POST_BUILD COMMAND ${CMAKE_COMMAND} -E copy $<TARGET_LINKER_FILE:${APP_PLUGIN}> "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}/")
+#add_custom_command(TARGET ${APP_PLUGIN} POST_BUILD COMMAND ${CMAKE_COMMAND} -E make_directory "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}/")
+#add_custom_command(TARGET ${APP_PLUGIN} POST_BUILD COMMAND ${CMAKE_COMMAND} -E copy_directory "${CMAKE_CURRENT_SOURCE_DIR}/include/" "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_INCLUDEDIR}/")
+#add_custom_command(TARGET ${APP_PLUGIN} POST_BUILD COMMAND ${CMAKE_COMMAND} -E copy $<TARGET_FILE:${APP_PLUGIN}> "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}/")
+#add_custom_command(TARGET ${APP_PLUGIN} POST_BUILD COMMAND ${CMAKE_COMMAND} -E copy $<TARGET_LINKER_FILE:${APP_PLUGIN}> "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}/")
 #add_custom_command(TARGET ${TARGET_NAME} PRE_BUILD COMMAND ${CMAKE_COMMAND} -E copy "${CMAKE_CURRENT_SOURCE_DIR}/include/LinphoneApp/*" "${CMAKE_INSTALL_PREFIX}/include/LinphoneApp/")
 
 #configure_file("${CMAKE_CURRENT_SOURCE_DIR}/include/*" "${CMAKE_INSTALL_PREFIX}/include/LinphoneApp/" COPYONLY)
@@ -843,7 +844,7 @@ install(DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/include" DESTINATION "${CMAKE_INS
 add_subdirectory(build)
 add_subdirectory(cmake_builder/linphone_package)
 
-deployqt_hack(${TARGET_NAME} ${LINPHONE_QML_DIR})
+#deployqt_hack(${TARGET_NAME} ${LINPHONE_QML_DIR})
 
 
 # ------------------------------------------------------------------------------
-- 
2.46.0

