--- a/linphone-app/CMakeLists.txt	2024-07-22 11:08:44.000000000 +0200
+++ -	2024-08-04 20:12:29.115331824 +0200
@@ -24,26 +24,18 @@
 
 
 #Linphone targets
-set(LINPHONE_PACKAGES LinphoneCxx Mediastreamer2 Belcard LibLinphone)
+set(LINPHONE_PACKAGES LinphoneCxx Mediastreamer2 BelCard LibLinphone)
 
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
 
-find_package(BCToolbox)
-if(NOT BCToolbox_FOUND)
-	find_package(bctoolbox CONFIG REQUIRED)
-endif()
-if(NOT LINPHONEAPP_VERSION)
-	bc_compute_full_version(LINPHONEAPP_VERSION)
-endif()
 set(version_major)
 set(version_minor)
 set(version_patch)
 set(identifiers )
 set(metadata )
-bc_parse_full_version("${LINPHONEAPP_VERSION}" version_major version_minor version_patch identifiers metadata)
-
 
 project(linphoneqt VERSION "${version_major}.${version_minor}.${version_patch}")
+find_package(bctoolbox CONFIG)
 
 if(ENABLE_BUILD_VERBOSE)
 	#message("CMAKE_PREFIX_PATH ${CMAKE_PREFIX_PATH}")
@@ -61,13 +53,6 @@
 if(UNIX AND NOT APPLE)
 	set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
 endif()
-list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/../cmake")
-list(APPEND CMAKE_MODULE_PATH "${LINPHONE_OUTPUT_DIR}/cmake")
-list(APPEND CMAKE_MODULE_PATH "${LINPHONE_OUTPUT_DIR}/lib64/cmake")
-list(APPEND CMAKE_MODULE_PATH "${LINPHONE_OUTPUT_DIR}/lib/cmake")
-
-list(APPEND CMAKE_PREFIX_PATH "${QTKEYCHAIN_OUTPUT_DIR}/lib/cmake")
-list(APPEND CMAKE_PREFIX_PATH "${ISPELL_OUTPUT_DIR}/lib/cmake")
 
 if(APPLE)
 	list(APPEND CMAKE_FRAMEWORK_PATH "${LINPHONE_OUTPUT_DIR}/Frameworks")
@@ -847,7 +832,6 @@
 install(DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/include" DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}")
 
 
-add_subdirectory(build)
 add_subdirectory(cmake_builder/linphone_package)
 
 deployqt_hack(${TARGET_NAME} ${LINPHONE_QML_DIR})
