--- CMakeLists.txt.bak	2017-12-09 12:42:39.858783833 +0100
+++ CMakeLists.txt	2017-12-09 12:43:11.262785030 +0100
@@ -45,30 +45,6 @@
 set(CMAKE_AUTOMOC ON)
 set(CMAKE_INCLUDE_CURRENT_DIR ON)
 
-if (NOT WIN32)
-  check_cxx_compiler_flag("-Wsuggest-override" SUGGEST_OVERRIDE)
-  set(CUSTOM_FLAGS "\
--Wall \
--Wcast-align \
--Wconversion \
--Werror=return-type \
--Werror=old-style-cast \
--Wextra \
--Wfloat-equal \
--Winit-self \
--Woverloaded-virtual \
--Wpointer-arith \
--Wuninitialized \
--Wunused \
-")
-  if (NOT CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-    set(CUSTOM_FLAGS "${CUSTOM_FLAGS} -Wlogical-op")
-  endif ()
-  if (SUGGEST_OVERRIDE)
-    set(CUSTOM_FLAGS "${CUSTOM_FLAGS} -Wsuggest-override -Werror=suggest-override")
-  endif ()
-endif ()
-
 set(CUSTOM_FLAGS "${CUSTOM_FLAGS} -DQT_NO_EXCEPTIONS")
 
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${CUSTOM_FLAGS}")
