--- linphone-desktop-4.3.2/linphone-app/CMakeLists.txt.orig	2021-12-08 16:02:45.190027494 +0100
+++ linphone-desktop-4.3.2/linphone-app/CMakeLists.txt	2021-12-08 16:03:05.969026699 +0100
@@ -21,18 +21,13 @@
 ################################################################################
 cmake_minimum_required(VERSION 3.1)
 
-find_package(bctoolbox CONFIG)
-set(FULL_VERSION )
-bc_compute_full_version(FULL_VERSION)
-set(version_major )
-set(version_minor )
-set(version_patch )
-set(identifiers )
-set(metadata )
-bc_parse_full_version("${FULL_VERSION}" version_major version_minor version_patch identifiers metadata)
+set(FULL_VERSION 4.3.2)
+set(version_major 4)
+set(version_minor 3)
+set(version_patch 2)
 
 project(linphoneqt VERSION "${version_major}.${version_minor}.${version_patch}")
-
+find_package(bctoolbox CONFIG REQUIRED)
 
 if(ENABLE_BUILD_VERBOSE)
 	#message("CMAKE_PREFIX_PATH ${CMAKE_PREFIX_PATH}")
