Default to using platformdirs

--- a/jupyter_core/paths.py	2020-02-01 21:00:00.000000000 -0300
+++ b/jupyter_core/paths.py	2023-06-04 00:49:36.985844333 -0300
@@ -53,7 +53,7 @@
     We plan for this to default to False in jupyter_core version 5 and to True
     in jupyter_core version 6.
     """
-    return envset("JUPYTER_PLATFORM_DIRS", False)  # type:ignore[return-value]
+    return envset("JUPYTER_PLATFORM_DIRS", True)  # type:ignore[return-value]
 
 
 def get_home_dir() -> str:
--- a/tests/test_paths.py	2020-02-01 21:00:00.000000000 -0300
+++ b/tests/test_paths.py	2023-06-04 00:49:05.967219753 -0300
@@ -86,7 +86,7 @@
     return os.path.abspath(os.path.realpath(os.path.expanduser(path)))
 
 
-home_jupyter = realpath("~/.jupyter")
+home_jupyter = realpath("~/.config/jupyter")
 
 
 def test_envset():
