--- ./packaging/pep517_backend/_backend.py.orig	2025-11-25 11:42:24.470431431 -0500
+++ ./packaging/pep517_backend/_backend.py	2025-11-25 11:42:37.275462683 -0500
@@ -376,7 +376,7 @@
     if is_pure_python_build:
         c_ext_build_deps = []
     else:
-        c_ext_build_deps = ['Cython ~= 3.1.0']
+        c_ext_build_deps = ['Cython >= 3.1.0']
 
     return _setuptools_get_requires_for_build_wheel(
         config_settings=config_settings,
