Index: libnest2d-0.4/CMakeLists.txt
===================================================================
--- libnest2d-0.4.orig/CMakeLists.txt
+++ libnest2d-0.4/CMakeLists.txt
@@ -2,9 +2,6 @@ cmake_minimum_required(VERSION 3.0)
 
 project(Libnest2D)
 
-set(CMAKE_CXX_STANDARD 11)
-set(CMAKE_CXX_STANDARD_REQUIRED)
-
 # Add our own cmake module path.
 list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake_modules/)
 
