diff --git a/client/common/man/CMakeLists.txt b/client/common/man/CMakeLists.txt
index 40c7e3d8..68d18451 100644
--- a/client/common/man/CMakeLists.txt
+++ b/client/common/man/CMakeLists.txt
@@ -1,6 +1,3 @@
-if(CMAKE_CROSSCOMPILING)
-  find_package(GenerateArgumentManpage)
-else()
   include_directories(${CMAKE_BINARY_DIR}/include/)
 
   add_executable(generate_argument_manpage generate_argument_manpage.c ../cmdline.h)
@@ -15,13 +12,12 @@ else()
   endif()
 
   export(TARGETS generate_argument_manpage FILE "${CMAKE_BINARY_DIR}/GenerateArgumentManpageConfig.cmake")
-endif()
 
 set(MAN_OPTIONS_FILE "${CMAKE_CURRENT_BINARY_DIR}/freerdp-global-options.1")
 set_property(DIRECTORY APPEND PROPERTY CMAKE_CONFIGURE_DEPENDS ${MAN_OPTIONS_FILE})
 
 add_custom_command(
-  OUTPUT ${MAN_OPTIONS_FILE} COMMAND $<TARGET_FILE:generate_argument_manpage> "${MAN_OPTIONS_FILE}"
+  OUTPUT ${MAN_OPTIONS_FILE} COMMAND generate_argument_manpage "${MAN_OPTIONS_FILE}"
   WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR} DEPENDS generate_argument_manpage
 )
 add_custom_target(
