help cmake run executables when cross-compiling
diff --git a/cmake/InstallFreeRDPMan.cmake b/cmake/InstallFreeRDPMan.cmake
index c333f6497..2dff84966 100644
--- a/cmake/InstallFreeRDPMan.cmake
+++ b/cmake/InstallFreeRDPMan.cmake
@@ -57,7 +57,7 @@ function(generate_and_install_freerdp_man_from_xml name_base section api depende
 
 		add_custom_command(
                                         OUTPUT "${manpage}"
-					COMMAND ${CMAKE_BINARY_DIR}/client/common/man/generate_argument_docbook
+					COMMAND ${CMAKE_CROSSCOMPILING_EMULATOR} ${CMAKE_BINARY_DIR}/client/common/man/generate_argument_docbook
 					COMMAND ${XSLTPROC_EXECUTABLE} --path "${CMAKE_CURRENT_BINARY_DIR} ${CMAKE_CURRENT_SOURCE_DIR}" ${DOCBOOKXSL_DIR}/manpages/docbook.xsl ${manpage}.xml
 					WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
 					DEPENDS
diff --git a/client/SDL/dialogs/res/CMakeLists.txt b/client/SDL/dialogs/res/CMakeLists.txt
index 5591e4af6..0b0946534 100644
--- a/client/SDL/dialogs/res/CMakeLists.txt
+++ b/client/SDL/dialogs/res/CMakeLists.txt
@@ -33,7 +33,7 @@ macro(convert_to_bin FILE FILE_TYPE)
     add_custom_command(
         OUTPUT ${FILE_BYPRODUCTS}
         COMMAND ${CMAKE_COMMAND} -E make_directory ${FILE_BIN_DIR}
-        COMMAND $<TARGET_FILE:freerdp-res2bin> ${FILE} ${FILE_TYPE} ${TARGET_NAME} ${FILE_BIN_DIR}
+        COMMAND ${CMAKE_CROSSCOMPILING_EMULATOR} $<TARGET_FILE:freerdp-res2bin> ${FILE} ${FILE_TYPE} ${TARGET_NAME} ${FILE_BIN_DIR}
         COMMENT "create image resources"
         DEPENDS freerdp-res2bin
         DEPENDS ${FILE}
