diff --git a/client/common/man/CMakeLists.txt b/client/common/man/CMakeLists.txt
index 798957c78..c14d22319 100644
--- a/client/common/man/CMakeLists.txt
+++ b/client/common/man/CMakeLists.txt
@@ -1,6 +1,3 @@
-if(CMAKE_CROSSCOMPILING)
-	find_package(GenerateArgumentManpage)
-else()
     include_directories(${CMAKE_BINARY_DIR}/include/)
 
     add_executable(generate_argument_manpage
@@ -9,14 +6,13 @@ else()
     )
 	export(TARGETS generate_argument_manpage FILE
 		"${CMAKE_BINARY_DIR}/GenerateArgumentManpageConfig.cmake")
-endif()
 
 set(MAN_OPTIONS_FILE "${CMAKE_CURRENT_BINARY_DIR}/freerdp-global-options.1")
 set_property(DIRECTORY APPEND PROPERTY CMAKE_CONFIGURE_DEPENDS ${MAN_OPTIONS_FILE})
 
 add_custom_command(
 	OUTPUT ${MAN_OPTIONS_FILE}
-	COMMAND ${CMAKE_BINARY_DIR}/client/common/man/generate_argument_manpage "${MAN_OPTIONS_FILE}"
+	COMMAND generate_argument_manpage "${MAN_OPTIONS_FILE}"
 	WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
 	DEPENDS
 		generate_argument_manpage
