From 3de172e7da393e8d544b7eb38ce9db7d18c468b1 Mon Sep 17 00:00:00 2001
From: Christian Hesse <mail@eworm.de>
Date: Mon, 25 Mar 2024 20:12:08 +0100
Subject: [PATCH 1/1] Use webkit2gtk-4.1

---
 client/SDL/aad/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/client/SDL/aad/CMakeLists.txt b/client/SDL/aad/CMakeLists.txt
index 228654262..d13fb907e 100644
--- a/client/SDL/aad/CMakeLists.txt
+++ b/client/SDL/aad/CMakeLists.txt
@@ -44,7 +44,7 @@ if (WITH_WEBVIEW)
       )
     else()
       find_package(PkgConfig REQUIRED)
-      pkg_check_modules(WEBVIEW_GTK webkit2gtk-4.0 REQUIRED)
+      pkg_check_modules(WEBVIEW_GTK webkit2gtk-4.1 REQUIRED)
       include_directories(${WEBVIEW_GTK_INCLUDE_DIRS})
       list(APPEND LIBS
         ${WEBVIEW_GTK_LIBRARIES}
-- 
2.44.0

