--- a/icons/CMakeLists.txt
+++ b/icons/CMakeLists.txt
@@ -24,7 +24,7 @@ if(WITH_ICON_GENERATION)
         COMMAND ${Python_EXECUTABLE} ${CMAKE_SOURCE_DIR}/generate-24px-versions.py
                 ${CMAKE_CURRENT_SOURCE_DIR}
                 ${ICON_GEN_DIR}
-        COMMAND $<TARGET_FILE:generate-symbolic-dark> ${CMAKE_SOURCE_DIR}/icons ${ICON_GEN_DIR} ${ICON_DARK_GEN_DIR}
+        COMMAND generate-symbolic-dark ${CMAKE_SOURCE_DIR}/icons ${ICON_GEN_DIR} ${ICON_DARK_GEN_DIR}
     )
 else()
     # create the target dir, we need it for resource creation
