diff --git a/icons-dark/CMakeLists.txt b/icons-dark/CMakeLists.txt
index 5ec0138..6674add 100644
--- a/icons-dark/CMakeLists.txt
+++ b/icons-dark/CMakeLists.txt
@@ -15,7 +15,7 @@ if(WITH_ICON_GENERATION)
     # No trainling slashes on dirs. It will break scripts.
     add_custom_target(breeze-generate-symbolic-dark ALL
         COMMENT "Generating symbolic dark icons"
-        COMMAND $<TARGET_FILE:generate-symbolic-dark> ${CMAKE_SOURCE_DIR}/icons ${ICON_GEN_DIR}
+        COMMAND generate-symbolic-dark ${CMAKE_SOURCE_DIR}/icons ${ICON_GEN_DIR}
     )
     # Auto-generate 24px monochrome icons from 22px versions
     add_custom_target(breeze-generate-24px-versions-dark ALL
