--- orig/usr/local/Brother/inf/setupPrintcap	2008-09-18 08:26:25.000000000 +0200
+++ new/usr/local/Brother/inf/setupPrintcap	2019-07-08 15:03:43.515834792 +0200
@@ -46,7 +46,7 @@
         :sd=/var/spool/lpd/$BRPRINTER_NAME:\\
         :sh:\\
         :lp=$DEVICE_IF:\\
-        :if=/usr/local/Brother/lpd/filter$BRPRINTER_NAME:
+        :if=/usr/lib/brother/lpd/filter$BRPRINTER_NAME:
 EOF
 	;;
 	-e)
