--- addons/allegrogl/include/alleggl.h
+++ addons/allegrogl/include/alleggl.h
@@ -62,10 +62,14 @@
 
 /* HACK: Prevent both Mesa and SGI's broken headers from screwing us */
 #define __glext_h_
+#define __gl_glext_h_
 #define __glxext_h_
+#define __glx_glxext_h_
 #include <GL/gl.h>
 #undef  __glext_h_
+#undef  __gl_glext_h_
 #undef  __glxext_h_
+#undef  __glx_glxext_h_
 
 #endif /* ALLEGRO_MACOSX */
 
--- addons/allegrogl/include/allegrogl/GLext/glx_ext_defs.h
+++ addons/allegrogl/include/allegrogl/GLext/glx_ext_defs.h
@@ -1,6 +1,8 @@
 /* HACK: Prevent both Mesa and SGI's broken headers from screwing us */
 #define __glxext_h_
+#define __glx_glxext_h_
 #include <GL/glx.h>
+#undef __glx_glxext_h_
 #undef __glxext_h_
 
 #ifndef GLX_VERSION_1_3
