Index: makefile.in
===================================================================
--- a/makefile.in
+++ b/makefile.in
@@ -47,10 +47,10 @@ VERSION		:= @PACKAGE_VERSION@
 BUGREPORT	:= @PACKAGE_BUGREPORT@
 
 # paths
-srcdir		:= @srcdir@
-libdir		:= @libdir@
 prefix		:= @prefix@
 exec_prefix	:= @exec_prefix@
+srcdir		:= @srcdir@
+libdir		:= @libdir@
 include_dirs	:= . include
 
 # programs
@@ -1188,10 +1188,10 @@ help:
 install: all
 	@echo "installing..."
 	@echo ""
-	mkdir -p $(DESTDIR)$(exec_prefix)$(libdir)
+	mkdir -p $(DESTDIR)$(libdir)
 	mkdir -p $(DESTDIR)$(prefix)/include/liquid
-	install -m 644 -p $(SHARED_LIB) libliquid.a $(DESTDIR)$(exec_prefix)$(libdir)
-	ln -s $(SHARED_LIB) $(DESTDIR)$(exec_prefix)$(libdir)/libliquid.so
+	install -m 644 -p $(SHARED_LIB) libliquid.a $(DESTDIR)$(libdir)
+	ln -s $(SHARED_LIB) $(DESTDIR)$(libdir)/libliquid.so
 	install -m 644 -p $(addprefix include/,$(headers_install)) $(DESTDIR)$(prefix)/include/liquid
 	@echo ""
 	@echo "---------------------------------------------------------"
@@ -1201,7 +1201,7 @@ install: all
 	@echo "  libraries by running 'ldconfig' to make the shared"
 	@echo "  object available.  You might also need to modify your"
 	@echo "  LD_LIBRARY_PATH environment variable to include the"
-	@echo "  directory $(DESTDIR)$(exec_prefix)"
+	@echo "  directory $(DESTDIR)"
 	@echo ""
 	@echo "  Please report bugs to $(BUGREPORT)"
 	@echo "---------------------------------------------------------"
@@ -1214,8 +1214,8 @@ install: all
 uninstall:
 	@echo "uninstalling..."
 	$(RM) $(addprefix $(DESTDIR)$(prefix)/include/liquid/, $(headers_install))
-	$(RM) $(DESTDIR)$(exec_prefix)$(libdir)/libliquid.a
-	$(RM) $(DESTDIR)$(exec_prefix)$(libdir)/$(SHARED_LIB)
+	$(RM) $(DESTDIR)$(libdir)/libliquid.a
+	$(RM) $(DESTDIR)$(libdir)/$(SHARED_LIB)
 	@echo "done."
 
 ##
