commit 9030406bc1e0df262471708b8952dde60ab49f9d
Author: q66 <daniel@octaforge.org>
Date:   Mon Jun 13 19:12:58 2022 +0200

    fix typo

diff --git a/src/dotprod/src/dotprod_rrrf.av.c b/src/dotprod/src/dotprod_rrrf.av.c
index 4aebc45..a54763c 100644
--- a/src/dotprod/src/dotprod_rrrf.av.c
+++ b/src/dotprod/src/dotprod_rrrf.av.c
@@ -117,7 +117,7 @@ dotprod_rrrf dotprod_rrrf_create_opt(float *      _h,
     for (i=0; i<4; i++) {
         q->h[i] = calloc(1+(q->n+i-1)/4,sizeof(vector float));
         for (j=0; j<q->n; j++)
-            e->h[i][j+i] = _h[_rev ? q->n-j-1 : j];
+            q->h[i][j+i] = _h[_rev ? q->n-j-1 : j];
     }
 
     return q;
