Index: makefile.in
===================================================================
--- a/makefile.in
+++ b/makefile.in
@@ -1201,7 +1201,7 @@ install: all
 	@echo "  libraries by running 'ldconfig' to make the shared"
 	@echo "  object available.  You might also need to modify your"
 	@echo "  LD_LIBRARY_PATH environment variable to include the"
-	@echo "  directory $(DESTDIR)$(exec_prefix)"
+	@echo "  directory $(DESTDIR)"
 	@echo ""
 	@echo "  Please report bugs to $(BUGREPORT)"
 	@echo "---------------------------------------------------------"
