--- a/protoc-c/c_file.h	2018-07-01 11:52:24.135863225 +0200
+++ b/protoc-c/c_file.h	2018-07-01 11:54:02.117755492 +0200
@@ -67,6 +67,10 @@
 #include <vector>
 #include <google/protobuf/stubs/common.h>
 #include <protoc-c/c_field.h>
+#include <boost/scoped_array.hpp>
+#include <boost/scoped_ptr.hpp>
+using namespace boost;
+using namespace std;
 
 namespace google {
 namespace protobuf {
--- a/protoc-c/c_helpers.cc	2018-07-01 11:52:24.137863243 +0200
+++ b/protoc-c/c_helpers.cc	2018-07-01 11:54:05.424785612 +0200
@@ -67,6 +67,10 @@
 
 #include <protoc-c/c_helpers.h>
 #include <google/protobuf/stubs/common.h>
+#include <boost/scoped_array.hpp>
+#include <boost/scoped_ptr.hpp>
+using namespace boost;
+using namespace std;
 
 namespace google {
 namespace protobuf {
--- a/protoc-c/c_message.h	2018-07-01 11:52:24.138863253 +0200
+++ b/protoc-c/c_message.h	2018-07-01 11:54:08.375812485 +0200
@@ -66,6 +66,10 @@
 #include <string>
 #include <google/protobuf/stubs/common.h>
 #include <protoc-c/c_field.h>
+#include <boost/scoped_array.hpp>
+#include <boost/scoped_ptr.hpp>
+using namespace boost;
+using namespace std;
 
 namespace google {
 namespace protobuf {
--- a/protoc-c/c_field.h	2018-07-01 11:52:24.142863289 +0200
+++ b/protoc-c/c_field.h	2018-07-01 11:54:11.010836479 +0200
@@ -65,6 +65,10 @@
 
 #include <google/protobuf/stubs/common.h>
 #include <google/protobuf/descriptor.h>
+#include <boost/scoped_array.hpp>
+#include <boost/scoped_ptr.hpp>
+using namespace boost;
+using namespace std;
 
 namespace google {
 namespace protobuf {
