--- common/common.c	2015-04-17 23:08:32.543680010 +0200
+++ common/common.c.new	2015-04-17 23:14:02.152457972 +0200
@@ -272,7 +272,6 @@
 {
 #ifdef AICCU_GNUTLS
 	/* Allow connections to servers that have OpenPGP keys as well */
-	const int	cert_type_priority[3] = { GNUTLS_CRT_X509, GNUTLS_CRT_OPENPGP, 0 };
 	int		ret;
 #endif /* AICCU_GNUTLS*/
 
@@ -300,8 +299,7 @@
 	gnutls_set_default_priority(sock->session);
 	/* XXX: Return value is not documented in GNUTLS documentation! */
 
-	gnutls_certificate_type_set_priority(sock->session, cert_type_priority);
-	/* XXX: Return value is not documented in GNUTLS documentation! */
+	gnutls_priority_set_direct(sock->session, "NORMAL:+CTYPE-OPENPGP", NULL);
 
 	/* Configure the x509 credentials for the current session */
 	gnutls_credentials_set(sock->session, GNUTLS_CRD_CERTIFICATE, g_aiccu->tls_cred);
