--- a/assimp_qt_viewer/mainwindow.cpp
+++ b/assimp_qt_viewer/mainwindow.cpp
@@ -296,7 +296,7 @@
 {
 using namespace Assimp;
 
-	DefaultLogger::get()->detatchStream(mLoggerView, DefaultLogger::Debugging | DefaultLogger::Info | DefaultLogger::Err | DefaultLogger::Warn);
+	DefaultLogger::get()->detachStream(mLoggerView, DefaultLogger::Debugging | DefaultLogger::Info | DefaultLogger::Err | DefaultLogger::Warn);
 	DefaultLogger::kill();
 
 	if(mScene != nullptr) mImporter.FreeScene();
