diff -up notify-python-0.1.1/src/pynotify.defs.notify070 notify-python-0.1.1/src/pynotify.defs
--- notify-python-0.1.1/src/pynotify.defs.notify070	2010-11-02 17:11:14.928179237 -0400
+++ notify-python-0.1.1/src/pynotify.defs	2010-11-02 17:11:51.153180231 -0400
@@ -38,7 +38,6 @@
     '("const-gchar*" "summary")
     '("const-gchar*" "message" (null-ok) (default "NULL"))
     '("const-gchar*" "icon" (null-ok) (default "NULL"))
-    '("GtkWidget*" "attach" (null-ok) (default "NULL"))
   )
 )
 
@@ -53,24 +52,6 @@
   )
 )
 
-(define-method attach_to_widget
-  (of-object "NotifyNotification")
-  (c-name "notify_notification_attach_to_widget")
-  (return-type "none")
-  (parameters
-    '("GtkWidget*" "attach")
-  )
-)
-
-(define-method attach_to_status_icon
-  (of-object "NotifyNotification")
-  (c-name "notify_notification_attach_to_status_icon")
-  (return-type "none")
-  (parameters
-    '("GtkStatusIcon*" "attach")
-  )
-)
-
 (define-method show
   (of-object "NotifyNotification")
   (c-name "notify_notification_show")
