--- a/xkbwatch.c
+++ b/xkbwatch.c
@@ -72,7 +72,7 @@ main(int argc, char *argv[])
     static Arg vArgs[] = { {XtNorientation, (XtArgVal) XtorientVertical} };
     static Arg onArgs[] = { {XtNon, (XtArgVal) True} };
     static Arg offArgs[] = { {XtNon, (XtArgVal) False} };
-    static char *fallback_resources[] = {
+    static const char *fallback_resources[] = {
         "*Box*background: grey50",
         "*Box*borderWidth: 0",
         "*Box*vSpace: 1",
