--- a/src/core/CMakeLists.txt	2021-06-27 14:24:18.812840134 +0200
+++ b/src/core/CMakeLists.txt	2021-06-27 14:51:46.825108637 +0200
@@ -299,6 +299,8 @@
         endif()
     elseif(${config} STREQUAL "MinSizeRel")
         list(APPEND gnArgArg is_debug=false symbol_level=0 optimize_for_size=true)
+    elseif(${config} STREQUAL "None")
+        list(APPEND gnArgArg is_debug=false symbol_level=0)
     endif()
 
     list(APPEND gnArgArg
