--- a/src/core/CMakeLists.txt
+++ b/src/core/CMakeLists.txt
@@ -414,6 +414,10 @@ foreach(arch ${archs})
                 CONDITION QT_FEATURE_webengine_ozone_x11 AND XSCRNSAVER_FOUND
             )
 
+            if (cpu STREQUAL "ppc64")
+                list(APPEND gnArgArg enable_jxl_decoder=false)
+            endif()
+
             if(QT_FEATURE_webengine_kerberos)
                 list(APPEND gnArgArg
                      external_gssapi_include_dir="${GSSAPI_INCLUDE_DIRS}/gssapi"
