--- a/configure.cmake	2021-06-21 23:31:04.000000000 +0200
+++ -	2021-06-27 16:16:32.873528917 +0200
@@ -373,11 +373,6 @@
 )
 add_check_for_support(webEngineError webEngineSupport
    MODULE QtWebEngine
-   CONDITION NOT LINUX OR TEST_glibc
-   MESSAGE "A suitable version >= 2.17 of glibc is required."
-)
-add_check_for_support(webEngineError webEngineSupport
-   MODULE QtWebEngine
    CONDITION NOT LINUX OR TEST_khr
    MESSAGE "Build requires Khronos development headers for build - see mesa/libegl1-mesa-dev"
 )
