diff --git a/configure.cmake b/configure.cmake
index 28318587..8ecc66af 100644
--- a/configure.cmake
+++ b/configure.cmake
@@ -388,11 +388,6 @@ add_check_for_support(
    CONDITION NOT LINUX OR PkgConfig_FOUND
    MESSAGE "A pkg-config support is required."
 )
-add_check_for_support(
-   MODULES QtWebEngine QtPdf
-   CONDITION NOT LINUX OR TEST_glibc
-   MESSAGE "A suitable version >= 2.17 of glibc is required."
-)
 add_check_for_support(
    MODULES QtWebEngine QtPdf
    CONDITION NOT LINUX OR TEST_khr

