--- qt6-webengine-6.4.2.orig/src/3rdparty/chromium/components/autofill/core/browser/BUILD.gn
+++ qt6-webengine-6.4.2/src/3rdparty/chromium/components/autofill/core/browser/BUILD.gn
@@ -34,6 +34,11 @@ action("regex_patterns_inl_h") {
 }
 
 jumbo_static_library("browser") {
+  if (is_clang) {
+    cflags = [
+      "-fbracket-depth=1000",
+    ]
+  }
   sources = [
     "address_normalization_manager.cc",
     "address_normalization_manager.h",
