--- qt6-webengine-6.4.2.orig/src/3rdparty/chromium/third_party/node/node.py
+++ qt6-webengine-6.4.2/src/3rdparty/chromium/third_party/node/node.py
@@ -18,25 +18,7 @@ def which(cmd):
     return None
 
 def GetBinaryPath():
-  if sys.platform == 'win32':
-    nodejs = which('node.exe')
-    if nodejs:
-      return nodejs
-  else:
-    nodejs = which('nodejs')
-    if nodejs:
-      return nodejs
-    nodejs = which('node')
-    if nodejs:
-      return nodejs
-
-  darwin_name = ('node-darwin-arm64' if platform.machine() == 'arm64' else
-                 'node-darwin-x64')
-  return os_path.join(os_path.dirname(__file__), *{
-    'Darwin': ('mac', darwin_name, 'bin', 'node'),
-    'Linux': ('linux', 'node-linux-x64', 'bin', 'node'),
-    'Windows': ('win', 'node.exe'),
-  }[platform.system()])
+  return "/usr/bin/node"
 
 
 def RunNode(cmd_parts, stdout=None):
