--- qt6-webengine-6.4.2.orig/src/3rdparty/chromium/base/files/scoped_file_linux.cc
+++ qt6-webengine-6.4.2/src/3rdparty/chromium/base/files/scoped_file_linux.cc
@@ -77,15 +77,3 @@ bool IsFDOwned(int fd) {
 }
 
 }  // namespace base
-
-extern "C" {
-
-int __close(int);
-
-__attribute__((visibility("default"), noinline)) int close(int fd) {
-  if (base::IsFDOwned(fd) && g_is_ownership_enforced)
-    CrashOnFdOwnershipViolation();
-  return __close(fd);
-}
-
-}  // extern "C"
