--- a/src/3rdparty/chromium/build/config/compiler/BUILD.gn
+++ b/src/3rdparty/chromium/build/config/compiler/BUILD.gn
@@ -394,7 +394,6 @@ config("compiler") {
         "-Wno-parentheses",
         "-Wno-sign-compare", # fido
         "-Wno-stringop-overflow", # skia, libaom, protobuf, icu
-        "-Wno-stringop-overread", # libaom
         "-Wno-psabi", # skia
         "-Wno-multichar", # crashpad
         "-Wno-format-zero-length", # user_agent.cc
