--- qt6-webengine-6.4.2.orig/src/3rdparty/chromium/base/threading/platform_thread_linux.cc
+++ qt6-webengine-6.4.2/src/3rdparty/chromium/base/threading/platform_thread_linux.cc
@@ -437,8 +437,13 @@ void TerminateOnThread() {}
 
 size_t GetDefaultThreadStackSize(const pthread_attr_t& attributes) {
 #if !defined(THREAD_SANITIZER)
+#if defined(__GLIBC__)
   return 0;
 #else
+  // musl libcs default is too small, use 8mb like glibc
+  return (1 << 23);
+#endif
+#else
   // ThreadSanitizer bloats the stack heavily. Evidence has been that the
   // default stack size isn't enough for some browser tests.
   return 2 * (1 << 23);  // 2 times 8192K (the default stack size on Linux).
