--- qt6-webengine-6.4.2.orig/src/3rdparty/chromium/third_party/nasm/config/config-linux.h
+++ qt6-webengine-6.4.2/src/3rdparty/chromium/third_party/nasm/config/config-linux.h
@@ -139,7 +139,9 @@
 #define HAVE_ACCESS 1
 
 /* Define to 1 if you have the `canonicalize_file_name' function. */
+#ifdef __GLIBC__
 #define HAVE_CANONICALIZE_FILE_NAME 1
+#endif
 
 /* Define to 1 if you have the `cpu_to_le16' intrinsic function. */
 /* #undef HAVE_CPU_TO_LE16 */
