--- qt6-webengine-6.4.2.orig/src/3rdparty/chromium/BUILD.gn
+++ qt6-webengine-6.4.2/src/3rdparty/chromium/BUILD.gn
@@ -1537,14 +1537,6 @@ if (!is_ios && !use_qt) {
   }
 }
 
-# TODO(cassew): Add more OS's that don't support x86.
-is_valid_x86_target =
-    target_os != "ios" && target_os != "mac" &&
-    (target_os != "linux" || use_libfuzzer || !build_with_chromium)
-assert(
-    is_valid_x86_target || target_cpu != "x86",
-    "'target_cpu=x86' is not supported for 'target_os=$target_os'. Consider omitting 'target_cpu' (default) or using 'target_cpu=x64' instead.")
-
 group("chromium_builder_perf") {
   testonly = true
 
