--- a/src/3rdparty/chromium/third_party/crashpad/crashpad/compat/linux/sys/ptrace.h
+++ b/src/3rdparty/chromium/third_party/crashpad/crashpad/compat/linux/sys/ptrace.h
@@ -17,7 +17,9 @@
 
 #include_next <sys/ptrace.h>
 
+#ifdef __GLIBC__
 #include <sys/cdefs.h>
+#endif
 
 // https://sourceware.org/bugzilla/show_bug.cgi?id=22433
 #if !defined(PTRACE_GET_THREAD_AREA) && !defined(PT_GET_THREAD_AREA) && \
--- a/src/3rdparty/chromium/third_party/crashpad/crashpad/util/linux/ptracer.cc
+++ b/src/3rdparty/chromium/third_party/crashpad/crashpad/util/linux/ptracer.cc
@@ -26,6 +26,9 @@
 
 #if defined(ARCH_CPU_X86_FAMILY)
 #include <asm/ldt.h>
+#if !defined(__GLIBC__)
+#include <asm/ptrace-abi.h>
+#endif
 #endif
 
 namespace crashpad {
