--- qt6-webengine-6.4.2.orig/src/3rdparty/chromium/v8/src/base/cpu.cc
+++ qt6-webengine-6.4.2/src/3rdparty/chromium/v8/src/base/cpu.cc
@@ -165,6 +165,15 @@ static V8_INLINE void __cpuid(int cpu_in
 
 #if V8_HOST_ARCH_ARM || V8_HOST_ARCH_ARM64
 
+#ifndef __GLIBC__
+#include <elf.h>
+#ifdef __LP64__
+typedef Elf64_auxv_t elf_auxv_t;
+#else
+typedef Elf32_auxv_t elf_auxv_t;
+#endif
+#endif
+
 static uint32_t ReadELFHWCaps() {
   uint32_t result = 0;
 #if V8_GLIBC_PREREQ(2, 16)
