--- a/src/3rdparty/chromium/media/gpu/vaapi/BUILD.gn
+++ b/src/3rdparty/chromium/media/gpu/vaapi/BUILD.gn
@@ -14,6 +14,10 @@ import("//ui/gl/features.gni")
 assert(is_linux || is_chromeos)
 assert(use_vaapi)
 
+config("vaapi_permissive") {
+  cflags = [ "-fpermissive" ]
+}
+
 generate_stubs("libva_stubs") {
   extra_header = "va_stub_header.fragment"
   sigs = [ "va.sigs" ]
@@ -136,6 +140,10 @@ source_set("vaapi") {
     ]
   }
 
+  if (target_cpu == "x86") {
+    configs += [ ":vaapi_permissive" ]
+  }
+
   if (use_x11) {
     deps += [ "//ui/gfx/x" ]
     sources += [
