--- a/examples/CMakeLists.txt	2023-10-03 21:49:54.000000000 +0200
+++ -	2023-10-27 11:12:01.573293042 +0200
@@ -4,7 +4,7 @@
 cmake_minimum_required(VERSION 3.16)
 
 qt_examples_build_begin(EXTERNAL_BUILD)
-if(NOT CMAKE_CROSSCOMPILING) #QTBUG-86533
+#if(NOT CMAKE_CROSSCOMPILING) #QTBUG-86533
     if(TARGET Qt::WebEngineCore)
         add_subdirectory(webenginequick)
     endif()
@@ -17,5 +17,5 @@
     if(TARGET Qt::PdfWidgets)
         add_subdirectory(pdfwidgets)
     endif()
-endif()
+#endif()
 qt_examples_build_end()
