From 4ba15fa7a1607ad79b93a43be9a66957988bcec9 Mon Sep 17 00:00:00 2001
From: Brett Mandler <brettmandler@gmail.com>
Date: Fri, 11 May 2018 18:55:16 -0400
Subject: [PATCH] Fixed invalid pointer error

---
 ruler.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/ruler.c b/ruler.c
index aa1eef1..06b3256 100644
--- a/ruler.c
+++ b/ruler.c
@@ -901,7 +901,6 @@ reload_config(void)
 	if (parse_file(xdg_cfg_path) == 1 && no_of_configs == 0)
 		errx(1, "couldn't open config file '%s' (%s). No other config files supplied, exiting", xdg_cfg_path, strerror(errno));
 	free(xdg_cfg_path);
-	free(xdg_home);
 
 	for (i = 0; i < no_of_configs; i++) {
 		if (parse_file(configs[i]) != 0)

