--- configure.in	2006-11-14 21:43:56.000000000 -0600
+++ configure.in	2006-11-29 10:34:48.000000000 -0600
@@ -28,8 +28,8 @@
 dnl

 EXTRA_DEFINES=""
-EXTRA_INCLUDES=""
-EXTRA_LIBS=""
+EXTRA_INCLUDES="$EXTRA_INCLUDES"
+EXTRA_LIBS="$EXTRA_LIBS"


 dnl
@@ -130,8 +130,8 @@
   echo 'Configuring Perl-Compatible Regular Expression (PCRE) library ...'
   echo

-  REGEX_DIR='pcre-5.0'
-  REGEX_OBJS="$REGEX_DIR/pcre.o $REGEX_DIR/study.o"
+#  REGEX_DIR='pcre-5.0'
+#  REGEX_OBJS="$REGEX_DIR/pcre.o $REGEX_DIR/study.o"
   USE_PCRE="1"

 else
@@ -146,7 +146,7 @@

 fi

-( cd $REGEX_DIR && ./configure )
+#( cd $REGEX_DIR && ./configure )

 AC_SUBST(REGEX_DIR)
 AC_SUBST(REGEX_OBJS)
--- Makefile.in	2006-11-28 07:35:37.000000000 -0600
+++ Makefile.in	2006-11-29 08:20:32.000000000 -0600
@@ -38,7 +38,7 @@

 all: $(TARGET)

-$(TARGET): $(REGEX_OBJS) $(OBJS)
+$(TARGET): $(OBJS)
	$(CC) $(CFLAGS) $(LDFLAGS) $(STRIPFLAG) -o $(TARGET) $(OBJS) $(REGEX_OBJS) $(LIBS)

 debug: $(REGEX_OBJS) $(OBJS)
--- ngrep.c	2006-11-28 07:38:43.000000000 -0600
+++ ngrep.c	2006-11-29 08:10:48.000000000 -0600
@@ -92,7 +92,7 @@
 #endif

 #if USE_PCRE
-#include "pcre-5.0/pcre.h"
+#include "pcre.h"
 #else
 #include "regex-0.12/regex.h"
 #endif
