From 60c3edbc4ccb15680c0305e787bd236933c85bad Mon Sep 17 00:00:00 2001
From: lemmi <lemmi@nerd2nerd.org>
Date: Tue, 12 May 2015 08:40:56 +0200
Subject: [PATCH] musl


diff --git src/control/jobs/control_jobs.c src/control/jobs/control_jobs.c
index ea61eb1..28ec5ea 100644
--- src/control/jobs/control_jobs.c
+++ src/control/jobs/control_jobs.c
@@ -941,7 +941,7 @@ static int32_t dt_control_export_job_run(dt_job_t *job)
   const __attribute__((__unused__)) int num_threads = 1;
 #if !defined(__SUNOS__) && !defined(__NetBSD__) && !defined(__WIN32__)
 #pragma omp parallel default(none) private(imgid)                                                            \
-    shared(control, fraction, w, h, stderr, mformat, mstorage, t, sdata, job, progress, darktable, settings) \
+    shared(control, fraction, w, h, mformat, mstorage, t, sdata, job, progress, darktable, settings) \
         num_threads(num_threads) if(num_threads > 1)
 #else
 #pragma omp parallel private(imgid) shared(control, fraction, w, h, mformat, mstorage, t, sdata, job,        \
@@ -993,7 +993,6 @@ static int32_t dt_control_export_job_run(dt_job_t *job)
         if(!g_file_test(imgfilename, G_FILE_TEST_IS_REGULAR))
         {
           dt_control_log(_("image `%s' is currently unavailable"), image->filename);
-          fprintf(stderr, "image `%s' is currently unavailable", imgfilename);
           // dt_image_remove(imgid);
           dt_image_cache_read_release(darktable.image_cache, image);
         }
diff --git src/develop/blend.c src/develop/blend.c
index b39afde..e327634 100644
--- src/develop/blend.c
+++ src/develop/blend.c
@@ -2276,7 +2276,7 @@ void dt_develop_blend_process(struct dt_iop_module_t *self, struct dt_dev_pixelp
 
 #ifdef _OPENMP
 #if !defined(__SUNOS__) && !defined(__NetBSD__) && !defined(__WIN32__)
-#pragma omp parallel for default(none) shared(i, roi_out, o, mask, blend, d, stderr)
+#pragma omp parallel for default(none) shared(i, roi_out, o, mask, blend, d)
 #else
 #pragma omp parallel for shared(i, roi_out, o, mask, blend, d)
 #endif
@@ -2327,7 +2327,7 @@ void dt_develop_blend_process(struct dt_iop_module_t *self, struct dt_dev_pixelp
     {
 #ifdef _OPENMP
 #if !defined(__SUNOS__) && !defined(__WIN32__)
-#pragma omp parallel for default(none) shared(roi_out, mask, stderr)
+#pragma omp parallel for default(none) shared(roi_out, mask)
 #else
 #pragma omp parallel for shared(roi_out, mask)
 #endif
@@ -2342,7 +2342,7 @@ void dt_develop_blend_process(struct dt_iop_module_t *self, struct dt_dev_pixelp
 /* now apply blending with per-pixel opacity value as defined in mask */
 #ifdef _OPENMP
 #if !defined(__SUNOS__) && !defined(__WIN32__)
-#pragma omp parallel for default(none) shared(i, roi_out, o, mask, blend, stderr)
+#pragma omp parallel for default(none) shared(i, roi_out, o, mask, blend)
 #else
 #pragma omp parallel for shared(i, roi_out, o, mask, blend)
 #endif
-- 
2.4.0

