Index: src/external/rawspeed/cmake/build-type.cmake
===================================================================
--- a/src/external/rawspeed/cmake/build-type.cmake
+++ b/src/external/rawspeed/cmake/build-type.cmake
@@ -27,9 +27,6 @@ string(TOUPPER "${RAWSPEED_SPECIAL_BUILD
 
 # is this one of the known build types?
 list (FIND CMAKE_CONFIGURATION_TYPES_UPPERCASE ${CMAKE_BUILD_TYPE_UPPERCASE} BUILD_TYPE_IS_KNOWN)
-if (${BUILD_TYPE_IS_KNOWN} EQUAL -1)
- message(SEND_ERROR "Unknown build type: ${CMAKE_BUILD_TYPE_UPPERCASE}. Please specify one of: ${CMAKE_CONFIGURATION_TYPES}")
-endif()
 
 # is this a special build?
 list (FIND RAWSPEED_SPECIAL_BUILD_TYPES_UPPERCASE ${CMAKE_BUILD_TYPE_UPPERCASE} IS_SPECIAL_BUILD)
