--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -281,10 +281,6 @@ message(STATUS "Looking for external pro
 find_program(perl_BIN perl REQUIRED)
 message(STATUS "Found perl")
 
-# we need intltool-merge for org.darktable.darktable.desktop
-find_program(intltool_merge_BIN intltool-merge REQUIRED)
-message(STATUS "Found intltool-merge")
-
 # we need desktop-file-validate to check org.darktable.darktable.desktop
 find_program(desktop_file_validate_BIN desktop-file-validate)
 if(${desktop_file_validate_BIN} STREQUAL "desktop_file_validate_BIN-NOTFOUND")
--- a/data/CMakeLists.txt
+++ b/data/CMakeLists.txt
@@ -73,7 +73,10 @@ if(NOT WIN32)
   file(GLOB PO_FILES "${CMAKE_CURRENT_SOURCE_DIR}/../po/*.po")
   add_custom_command(
     OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/org.darktable.darktable.desktop
-    COMMAND sh -c "${intltool_merge_BIN} --desktop-style ${CMAKE_CURRENT_SOURCE_DIR}/../po ${CMAKE_CURRENT_BINARY_DIR}/org.darktable.darktable.desktop.in ${CMAKE_CURRENT_BINARY_DIR}/org.darktable.darktable.desktop"
+    COMMAND msgfmt --desktop
+        -d ${CMAKE_CURRENT_SOURCE_DIR}/../po
+        --template ${CMAKE_CURRENT_BINARY_DIR}/org.darktable.darktable.desktop.in
+        --output ${CMAKE_CURRENT_BINARY_DIR}/org.darktable.darktable.desktop
     MAIN_DEPENDENCY ${CMAKE_CURRENT_BINARY_DIR}/org.darktable.darktable.desktop.in
     DEPENDS ${PO_FILES}
     )
@@ -95,7 +98,10 @@ if(NOT WIN32)
       OUTPUT ${DARKTABLE_SHAREDIR}/metainfo/org.darktable.darktable.appdata.xml
       SOURCE ${CMAKE_CURRENT_SOURCE_DIR}/org.darktable.darktable.appdata.xml.in
       COMMAND ${CMAKE_COMMAND} -E make_directory ${DARKTABLE_SHAREDIR}/metainfo
-      COMMAND sh -c "${intltool_merge_BIN} --xml-style ${CMAKE_CURRENT_SOURCE_DIR}/../po ${CMAKE_CURRENT_SOURCE_DIR}/org.darktable.darktable.appdata.xml.in ${DARKTABLE_SHAREDIR}/metainfo/org.darktable.darktable.appdata.xml"
+      COMMAND msgfmt --xml
+            -d ${CMAKE_CURRENT_SOURCE_DIR}/../po
+            --template ${CMAKE_CURRENT_SOURCE_DIR}/org.darktable.darktable.appdata.xml.in
+            --output ${DARKTABLE_SHAREDIR}/metainfo/org.darktable.darktable.appdata.xml
       COMMAND ${appstream_util_BIN} validate --nonet ${DARKTABLE_SHAREDIR}/metainfo/org.darktable.darktable.appdata.xml
       MAIN_DEPENDENCY ${CMAKE_CURRENT_SOURCE_DIR}/org.darktable.darktable.appdata.xml.in
       DEPENDS ${PO_FILES}
@@ -104,7 +110,10 @@ if(NOT WIN32)
     add_custom_command(
       OUTPUT ${DARKTABLE_SHAREDIR}/metainfo/org.darktable.darktable.appdata.xml
       COMMAND ${CMAKE_COMMAND} -E make_directory ${DARKTABLE_SHAREDIR}/metainfo
-      COMMAND sh -c "${intltool_merge_BIN} --xml-style ${CMAKE_CURRENT_SOURCE_DIR}/../po ${CMAKE_CURRENT_SOURCE_DIR}/org.darktable.darktable.appdata.xml.in ${DARKTABLE_SHAREDIR}/metainfo/org.darktable.darktable.appdata.xml"
+      COMMAND msgfmt --xml
+            -d ${CMAKE_CURRENT_SOURCE_DIR}/../po
+            --template ${CMAKE_CURRENT_SOURCE_DIR}/org.darktable.darktable.appdata.xml.in
+            --output ${DARKTABLE_SHAREDIR}/metainfo/org.darktable.darktable.appdata.xml
       MAIN_DEPENDENCY ${CMAKE_CURRENT_SOURCE_DIR}/org.darktable.darktable.appdata.xml.in
       DEPENDS ${PO_FILES}
     )
--- a/data/org.darktable.darktable.appdata.xml.in
+++ b/data/org.darktable.darktable.appdata.xml.in
@@ -7,42 +7,42 @@
   <launchable type="desktop-id">org.darktable.darktable.desktop</launchable>
   <metadata_license>CC0-1.0</metadata_license>
   <project_license>GPL-3.0+</project_license>
-  <_summary>Organize and develop images from digital cameras</_summary>
+  <summary>Organize and develop images from digital cameras</summary>
   <translation type="gettext">darktable</translation>
   <description>
-    <_p>
+    <p>
       darktable manages your camera raw files and images in a database, lets you
       view them through lighttable mode and develop/enhance them in darkroom mode.
-    </_p>
-    <_p>
+    </p>
+    <p>
       Other modes besides lighttable and darkroom are a map for geotagging,
       tethering, print and a slideshow.
-    </_p>
-    <_p>
+    </p>
+    <p>
       darktable supports most modern cameras' raw formats, and does all
       of its processing at very high precision.
-    </_p>
+    </p>
   </description>
   <screenshots>
     <screenshot width="1920" height="1080">
       <image>https://www.darktable.org/images/darktable-appdata-lighttable.jpg</image>
-      <_caption>Virtual light table, showing a collection</_caption>
+      <caption>Virtual light table, showing a collection</caption>
     </screenshot>
     <screenshot width="1920" height="1080" type="default">
       <image>https://www.darktable.org/images/darktable-appdata-darkroom1.jpg</image>
-      <_caption>Virtual dark room with an opened image</_caption>
+      <caption>Virtual dark room with an opened image</caption>
     </screenshot>
     <screenshot width="1920" height="1080">
       <image>https://www.darktable.org/images/darktable-appdata-darkroom2.jpg</image>
-      <_caption>Virtual dark room, sharpening an image</_caption>
+      <caption>Virtual dark room, sharpening an image</caption>
     </screenshot>
     <screenshot width="1920" height="1080">
       <image>https://www.darktable.org/images/darktable-appdata-map.jpg</image>
-      <_caption>Show images on a map</_caption>
+      <caption>Show images on a map</caption>
     </screenshot>
     <screenshot width="1920" height="1080">
       <image>https://www.darktable.org/images/darktable-appdata-print.jpg</image>
-      <_caption>Print your images</_caption>
+      <caption>Print your images</caption>
     </screenshot>
   </screenshots>
   <content_rating type="oars-1.1" />
--- a/data/org.darktable.darktable.desktop.in
+++ b/data/org.darktable.darktable.desktop.in
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Name=Darktable
-_GenericName=Virtual Lighttable and Darkroom
-_Comment=Organize and develop images from digital cameras
+GenericName=Virtual Lighttable and Darkroom
+Comment=Organize and develop images from digital cameras
 
 X-GNOME-FullName=Darktable Photo Workflow Software
 
@@ -10,7 +10,7 @@ Version=1.0
 Type=Application
 Categories=Graphics;Photography;GTK;
 # TRANSLATORS: Search terms to find this application. Do NOT translate or localize the semicolons! The list MUST also end with a semicolon!
-_Keywords=graphics;photography;raw;
+Keywords=graphics;photography;raw;
 
 Exec=${CMAKE_INSTALL_PREFIX}/bin/darktable %U
 TryExec=${CMAKE_INSTALL_PREFIX}/bin/darktable
