--- lldb/source/Plugins/Process/Linux/NativeRegisterContextLinux_arm64.cpp.orig	2015-09-18 10:27:18.722489350 +0000
+++ lldb/source/Plugins/Process/Linux/NativeRegisterContextLinux_arm64.cpp	2015-09-18 09:48:27.475611875 +0000
@@ -30,6 +30,7 @@
 #include <sys/socket.h>
 // NT_PRSTATUS and NT_FPREGSET definition
 #include <elf.h>
+#include <asm/ptrace.h>
 
 #define REG_CONTEXT_SIZE (GetGPRSize() + GetFPRSize())
 

