--- source/Plugins/Process/Linux/ProcessMonitor.cpp.orig
+++ source/Plugins/Process/Linux/ProcessMonitor.cpp
@@ -27,6 +27,9 @@
 #include <sys/procfs.h>
 #endif
 #include <sys/ptrace.h>
+#ifdef __aarch64__
+#include <asm/ptrace.h>
+#endif
 #include <sys/uio.h>
 #include <sys/socket.h>
 #include <sys/syscall.h>
--- source/Plugins/Process/Linux/NativeProcessLinux.cpp.orig
+++ source/Plugins/Process/Linux/NativeProcessLinux.cpp
@@ -29,6 +29,9 @@
 #include <sys/procfs.h>
 #endif
 #include <sys/ptrace.h>
+#ifdef __aarch64__
+#include <asm/ptrace.h>
+#endif
 #include <sys/uio.h>
 #include <sys/socket.h>
 #include <sys/syscall.h>
