From f7aeba5b72f8434eb690ae3b84f43968b8041a95 Mon Sep 17 00:00:00 2001
From: Dominik Picheta <dominikpicheta@gmail.com>
Date: Sun, 21 Apr 2019 20:29:41 +0100
Subject: [PATCH] Fixes osinfo dependency conflict.

---
 choosenim.nimble | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/choosenim.nimble b/choosenim.nimble
index d826357..b775d88 100644
--- choosenim.nimble
+++ choosenim.nimble
@@ -15,7 +15,7 @@ skipExt = @["nim"]
 
 requires "nim >= 0.16.1", "nimble >= 0.8.5", "untar >= 0.1.0"
 requires "libcurl >= 1.0.0"
-requires "analytics >= 0.1.1"
+requires "analytics >= 0.2.0"
 requires "osinfo >= 0.3.0"
 
 task test, "Run the choosenim tester!":
