diff --git a/Source/LibJXR/image/decode/segdec.c b/Source/LibJXR/image/decode/segdec.c
index f2a0c8b..1e4cd7c 100644
--- a/Source/LibJXR/image/decode/segdec.c
+++ b/Source/LibJXR/image/decode/segdec.c
@@ -44,6 +44,8 @@ static Int DecodeSignificantAbsLevel (struct CAdaptiveHuffman *pAHexpt, BitIOInf
 //================================================================
 // Memory access functions
 //================================================================
+extern U32 _byteswap_ulong(U32 bits);
+
 static U32 _load4(void* pv)
 {
 #ifdef _BIG__ENDIAN_
