diff --git a/Source/LibJXR/jxrgluelib/JXRGlueJxr.c b/Source/LibJXR/jxrgluelib/JXRGlueJxr.c
index 2362806..3dd8d4f 100644
--- a/Source/LibJXR/jxrgluelib/JXRGlueJxr.c
+++ b/Source/LibJXR/jxrgluelib/JXRGlueJxr.c
@@ -28,7 +28,7 @@
 //*@@@---@@@@******************************************************************
 #include <limits.h>
 #include <JXRGlue.h>
-
+#include <wchar.h>
 
 static const char szHDPhotoFormat[] = "<dc:format>image/vnd.ms-photo</dc:format>";
 const U32 IFDEntryTypeSizes[] = { 0, 1, 1, 2, 4, 8, 1, 1, 2, 4, 8, 4, 8 };
diff --git a/Source/LibJXR/common/include/guiddef.h b/Source/LibJXR/common/include/guiddef.h
index 503b643..32b6b10 100644
--- a/Source/LibJXR/common/include/guiddef.h
+++ b/Source/LibJXR/common/include/guiddef.h
@@ -173,6 +173,7 @@ typedef FMTID *LPFMTID;
 // Faster (but makes code fatter) inline version...use sparingly
 #ifdef __cplusplus
 __inline int InlineIsEqualGUID(REFGUID rguid1, REFGUID rguid2)
+#include <wchar.h>
 {
    return (
       ((unsigned long *) &rguid1)[0] == ((unsigned long *) &rguid2)[0] &&
