--- a/dmd/compiler/src/dmd/link.d
+++ b/dmd/compiler/src/dmd/link.d
@@ -657,8 +657,12 @@ public int runLINK(bool verbose, ErrorSi
             perror("unable to create pipe to linker");
             return -1;
         }
+        version (CRuntime_Musl) {
+            childpid = fork();
+        } else {
         // vfork instead of fork to avoid https://issues.dlang.org/show_bug.cgi?id=21089
         childpid = vfork();
+        }
         if (childpid == 0)
         {
             // pipe linker stderr to fds[0]
