From 2354350bcc890c13016d67f4c060b32cb1cd693c Mon Sep 17 00:00:00 2001
From: q66 <q66@chimera-linux.org>
Date: Sat, 4 Nov 2023 08:44:01 +0100
Subject: [PATCH 01/25] llvm: always set a larger stack size explicitly

---
 llvm/lib/Support/Threading.cpp | 14 --------------
 1 file changed, 14 deletions(-)

diff --git a/llvm/lib/Support/Threading.cpp b/llvm/lib/Support/Threading.cpp
index 7cc7ba44c..b91b8f4bb 100644
--- a/llvm/lib/Support/Threading.cpp
+++ b/llvm/lib/Support/Threading.cpp
@@ -77,21 +77,7 @@ unsigned llvm::ThreadPoolStrategy::compute_thread_count() const {
 // keyword.
 #include "llvm/Support/thread.h"
 
-#if defined(__APPLE__)
-  // Darwin's default stack size for threads except the main one is only 512KB,
-  // which is not enough for some/many normal LLVM compilations. This implements
-  // the same interface as std::thread but requests the same stack size as the
-  // main thread (8MB) before creation.
 const std::optional<unsigned> llvm::thread::DefaultStackSize = 8 * 1024 * 1024;
-#elif defined(_AIX)
-  // On AIX, the default pthread stack size limit is ~192k for 64-bit programs.
-  // This limit is easily reached when doing link-time thinLTO. AIX library
-  // developers have used 4MB, so we'll do the same.
-const std::optional<unsigned> llvm::thread::DefaultStackSize = 4 * 1024 * 1024;
-#else
-const std::optional<unsigned> llvm::thread::DefaultStackSize;
-#endif
-
 
 #endif
 
-- 
2.42.0
