--- a/bin/fingerprint-helper/fingerprint-helper.pro	2016-11-21 15:24:35.315792087 +0300
+++ b/bin/fingerprint-helper/fingerprint-helper.pro	2016-11-21 15:25:24.527899001 +0300
@@ -22,8 +22,8 @@
 QMAKE_CXXFLAGS+=-fno-strict-aliasing
 
-uinput-rules.path     = $${INSTALL_ROOT}/etc/udev/rules.d
-uinput-rules.commands = install -d $(INSTALL_ROOT)/etc/udev/rules.d; \
-                        install -m 644 ./92-fingerprint-gui-uinput.rules $(INSTALL_ROOT)/etc/udev/rules.d
+uinput-rules.path     = $${INSTALL_ROOT}/usr/lib/udev/rules.d
+uinput-rules.commands = install -d $(INSTALL_ROOT)/usr/lib/udev/rules.d; \
+                        install -m 644 ./92-fingerprint-gui-uinput.rules $(INSTALL_ROOT)/usr/lib/udev/rules.d
 
 INSTALLS += target uinput-rules
  
--- a/fingerprint.pro	2016-11-21 15:33:45.215341967 +0300
+++ b/fingerprint.pro	2016-11-21 15:36:26.077954927 +0300
@@ -17,8 +17,8 @@
                             install -m 644 ./upek/lib/libbsapi.so.4.0 ./upek/lib/libbsapi.so.4.3 $(INSTALL_ROOT)$${LIBDIR}; \
                         fi; \
                         ldconfig
-upek-rules.commands   = install -d $(INSTALL_ROOT)/etc/udev/rules.d; \
-                        install -m 644 ./upek/91-fingerprint-gui-upek.rules $(INSTALL_ROOT)/etc/udev/rules.d
+upek-rules.commands   = install -d $(INSTALL_ROOT)/usr/lib/udev/rules.d; \
+                        install -m 644 ./upek/91-fingerprint-gui-upek.rules $(INSTALL_ROOT)/usr/lib/udev/rules.d
 upek-cfg.commands     = install -d $(INSTALL_ROOT)/etc; \
                         install -m 644 ./upek/upek.cfg $(INSTALL_ROOT)/etc/upek.cfg
 upek-dir.commands     = install -d $(INSTALL_ROOT)/var/upek_data; \
@@ -38,7 +38,7 @@
 uninstall-upek.commands = rm -f $(INSTALL_ROOT)$${LIBDIR}/libbsapi.so.4.0; \
                           rm -f $(INSTALL_ROOT)$${LIBDIR}/libbsapi.so.4.3; \
                           rm -rf $(INSTALL_ROOT)/var/upek_data; \
-                          rm -f $(INSTALL_ROOT)/etc/udev/rules.d/91-fingerprint-gui-upek.rules; \
+                          rm -f $(INSTALL_ROOT)/usr/lib/udev/rules.d/91-fingerprint-gui-upek.rules; \
                           ldconfig
 
 # for version 1.07 change ownership and permissions of data directories and files

