--- a/upek/91-fingerprint-gui-upek.rules        2017-01-12 19:48:38.325444912 +0300
+++ b/upek/91-fingerprint-gui-upek.rules     2017-01-12 19:50:35.799120728 +0300
@@ -7,6 +7,10 @@
 ATTRS{idVendor}=="147e", ATTRS{idProduct}=="100[0123]", SYMLINK+="input/touchchip-%k", MODE="0664", GROUP="plugdev"
 ATTRS{idVendor}=="147e", ATTRS{idProduct}=="300[01]",   SYMLINK+="input/touchchip-%k", MODE="0664", GROUP="plugdev"
 ATTRS{idVendor}=="147e", ATTRS{idProduct}=="500[23]",   SYMLINK+="input/touchchip-%k", MODE="0664", GROUP="plugdev"
+ATTRS{idVendor}=="138a", ATTRS{idProduct}=="0018",      SYMLINK+="input/touchchip-%k", MODE="0664", GROUP="plugdev"
+ATTRS{idVendor}=="138a", ATTRS{idProduct}=="003d",      SYMLINK+="input/touchchip-%k", MODE="0664", GROUP="plugdev"
+ATTRS{idVendor}=="138a", ATTRS{idProduct}=="0050",      SYMLINK+="input/touchchip-%k", MODE="0664", GROUP="plugdev"
+ATTRS{idVendor}=="138a", ATTRS{idProduct}=="0092",      SYMLINK+="input/touchchip-%k", MODE="0664", GROUP="plugdev"

 # enable power saving
 ATTRS{idVendor}=="0483", ATTRS{idProduct}=="201[56]",   ATTR{power/control}=="*", ATTR{power/control}="auto"
@@ -15,3 +16,7 @@
 ATTRS{idVendor}=="147e", ATTRS{idProduct}=="100[0123]", ATTR{power/control}=="*", ATTR{power/control}="auto"
 ATTRS{idVendor}=="147e", ATTRS{idProduct}=="300[01]",   ATTR{power/control}=="*", ATTR{power/control}="auto"
 ATTRS{idVendor}=="147e", ATTRS{idProduct}=="500[23]",   ATTR{power/control}=="*", ATTR{power/control}="auto"
+ATTRS{idVendor}=="138a", ATTRS{idProduct}=="0018",      ATTR{power/control}=="*", ATTR{power/control}="auto"
+ATTRS{idVendor}=="138a", ATTRS{idProduct}=="003d",      ATTR{power/control}=="*", ATTR{power/control}="auto"
+ATTRS{idVendor}=="138a", ATTRS{idProduct}=="0050",      ATTR{power/control}=="*", ATTR{power/control}="auto"
+ATTRS{idVendor}=="138a", ATTRS{idProduct}=="0092",      ATTR{power/control}=="*", ATTR{power/control}="auto"
