commit 7b688681f5868416d0b6b9899fbd0d2ba213477d from upstream git
diff Programs/brlapi.h.in Programs/brlapi.h.in
--- Programs/brlapi.h.in
+++ Programs/brlapi.h.in
@@ -137,8 +137,16 @@ size_t BRLAPI_STDCALL brlapi_getHandleSize(void);
 #ifdef BRLAPI_WIN32
 /* No authentication by default on Windows */
 #define BRLAPI_DEFAUTH "none"
-#else
-#define BRLAPI_DEFAUTH "polkit+keyfile:" BRLAPI_ETCDIR "/" BRLAPI_AUTHKEYFILE
+#else /* BRLAPI_WIN32 */
+#define BRLAPI_DEFAUTH_KEYFILE "keyfile:" BRLAPI_ETCDIR "/" BRLAPI_AUTHKEYFILE
+
+#ifdef USE_POLKIT
+#define BRLAPI_DEFAUTH_POLKIT "+polkit"
+#else /* USE_POLKIT */
+#define BRLAPI_DEFAUTH_POLKIT ""
+#endif /* USE_POLKIT */
+
+#define BRLAPI_DEFAUTH BRLAPI_DEFAUTH_KEYFILE BRLAPI_DEFAUTH_POLKIT
 #endif /* BRLAPI_WIN32 */
 
 #ifdef __MINGW32__
