diff -Naur brltty-5.5/Bindings/Python/Makefile.in brltty-5.5.new/Bindings/Python/Makefile.in
--- a/Bindings/Python/Makefile.in	2017-04-18 15:35:13.000000000 -0700
+++ b/Bindings/Python/Makefile.in	2017-04-23 17:36:27.797793633 -0700
@@ -1,3 +1,6 @@
+export LDFLAGS= @LDFLAGS@ @PYTHON_LIBS@
+export CFLAGS= @CFLAGS@ @PYTHON_CPPFLAGS@
+export LDSHARED= @CC@ -shared @LDFLAGS@
 ###############################################################################
 # libbrlapi - A library providing access to braille terminals for applications.
 #
