--- a/Programs/system_linux.c	2020-04-05 12:35:34.000000000 +0200
+++ b/Programs/system_linux.c	2020-12-03 14:12:42.292642038 +0100
@@ -957,9 +957,12 @@
 writeInputEvent (UinputObject *uinput, uint16_t type, uint16_t code, int32_t value) {
 #ifdef HAVE_LINUX_UINPUT_H
   struct input_event event;
+  struct timeval tv;
 
   memset(&event, 0, sizeof(event));
-  gettimeofday(&event.time, NULL);
+  gettimeofday(&tv, NULL);
+  event.input_event_sec = tv.tv_sec;
+  event.input_event_usec = tv.tv_usec;
   event.type = type;
   event.code = code;
   event.value = value;
