See: https://github.com/mesa3d/mesa/commit/434da21a7c81acbe03f0e1621e80904313c0b05c

--- src/freedreno/common/meson.build	2021-01-29 19:33:19.570870900 +0100
+++ src/freedreno/common/meson.build	2021-02-12 21:32:57.108300772 +0100
@@ -27,6 +27,7 @@
     'freedreno_uuid.c',
     'freedreno_uuid.h',
     'freedreno_guardband.h',
+    sha1_h,
   ],
   include_directories : [inc_freedreno, inc_include, inc_src, inc_gallium],
   c_args : [no_override_init_args],
--- src/gallium/frontends/lavapipe/meson.build	2021-01-29 19:33:19.784871800 +0100
+++ src/gallium/frontends/lavapipe/meson.build	2021-02-12 21:33:41.406352915 +0100
@@ -57,7 +57,7 @@
 
 liblavapipe_st = static_library(
   'lavapipe_st',
-  [liblvp_files, lvp_entrypoints, lvp_extensions_c ],
+  [liblvp_files, lvp_entrypoints, lvp_extensions_c, sha1_h],
   link_with : [ libvulkan_wsi ],
   c_args : [ lvp_flags ],
   gnu_symbol_visibility : 'hidden',
--- src/intel/common/meson.build	2021-01-29 19:33:19.836871900 +0100
+++ src/intel/common/meson.build	2021-02-12 21:35:09.226438768 +0100
@@ -39,6 +39,7 @@
   'gen_sample_positions.h',
   'gen_uuid.c',
   'gen_uuid.h',
+  sha1_h,
 )
 
 libintel_common = static_library(
