--- a/src/util/rand_xor.c	2025-09-30 15:31:17.368731386 +0300
+++ b/src/util/rand_xor.c	2025-09-30 15:24:46.938064862 +0300
@@ -28,6 +28,7 @@
 #if defined(HAVE_GETRANDOM)
 #include <sys/random.h>
 #endif
+#include <sys/types.h>		/* size_t, ssize_t */
 #include <unistd.h>
 #include <fcntl.h>
 #endif
--- a/src/amd/vulkan/winsys/amdgpu/radv_amdgpu_winsys.h	2025-09-30 15:31:17.368731386 +0300
+++ b/src/amd/vulkan/winsys/amdgpu/radv_amdgpu_winsys.h	2025-09-30 15:24:46.938064862 +0300
@@ -13,6 +13,7 @@
 
 #include <amdgpu.h>
 #include <pthread.h>
+#include <sys/types.h>
 #include "util/list.h"
 #include "util/rwlock.h"
 #include "ac_gpu_info.h"
