--- src/gallium/winsys/svga/drm/vmw_screen.h.orig	2015-05-07 14:02:28.033079796 +0200
+++ src/gallium/winsys/svga/drm/vmw_screen.h	2015-05-07 14:02:48.832054666 +0200
@@ -35,6 +35,8 @@
 #define VMW_SCREEN_H_
 
 
+#include <sys/types.h> 	/* dev_t */
+
 #include "pipe/p_compiler.h"
 #include "pipe/p_state.h"
 
--- src/gallium/state_trackers/nine/threadpool.h.orig	2015-05-07 14:10:53.443337212 +0200
+++ src/gallium/state_trackers/nine/threadpool.h	2015-05-07 14:11:04.210307653 +0200
@@ -24,6 +24,8 @@
 #ifndef _THREADPOOL_H_
 #define _THREADPOOL_H_
 
+#include <pthread.h>
+
 #define MAXTHREADS 1
 
 struct threadpool {
--- src/util/rand_xor.c.orig	2017-06-20 00:38:57.199474067 +0200
+++ src/util/rand_xor.c	2017-06-20 00:40:31.351279557 +0200
@@ -23,7 +23,9 @@
  */
 
 #if defined(__linux__)
+#include <sys/types.h>
 #include <sys/file.h>
+#include <sys/stat.h>
 #include <unistd.h>
 #include <fcntl.h>
 #else
--- src/amd/vulkan/winsys/amdgpu/radv_amdgpu_winsys.h
+++ src/amd/vulkan/winsys/amdgpu/radv_amdgpu_winsys.h
@@ -28,6 +28,8 @@
 #ifndef RADV_AMDGPU_WINSYS_H
 #define RADV_AMDGPU_WINSYS_H
 
+#include <sys/types.h>
+
 #include "radv_radeon_winsys.h"
 #include "ac_gpu_info.h"
 #include "addrlib/addrinterface.h"
--- src/gallium/state_trackers/nine/nine_debug.c.orig	2020-03-06 07:46:48.278918421 +0100
+++ src/gallium/state_trackers/nine/nine_debug.c	2020-03-06 07:51:32.919964119 +0100
@@ -65,7 +65,7 @@ _nine_debug_printf( unsigned long flag,
 {
     static boolean first = TRUE;
     static unsigned long dbg_flags = DBG_ERROR | DBG_WARN;
-    unsigned long tid = 0;
+    pthread_t tid = 0;
 
     if (first) {
         first = FALSE;
--- src/gallium/drivers/panfrost/pan_bo.h.orig	2020-03-06 08:37:26.810535178 +0100
+++ src/gallium/drivers/panfrost/pan_bo.h	2020-03-06 08:38:00.290897294 +0100
@@ -29,6 +29,7 @@
 #include <panfrost-misc.h>
 #include "pipe/p_state.h"
 #include "util/list.h"
+#include <time.h>
 
 struct panfrost_screen;
 
