--- extras/Makefile.in.orig	2009-06-19 14:51:53.084747653 +0200
+++ extras/Makefile.in	2009-06-19 14:52:05.712058999 +0200
@@ -189,15 +189,11 @@ AM_LDFLAGS = \
 
 SUBDIRS = \
 	ata_id \
-	cdrom_id \
 	edd_id \
 	path_id \
 	firmware \
-	collect \
 	floppy \
-	fstab_import \
 	rule_generator \
-	scsi_id \
 	usb_id
 
 all: all-recursive
--- udev/udev-util.c.orig	2009-06-19 14:52:40.767402973 +0200
+++ udev/udev-util.c	2009-06-19 14:54:05.981115992 +0200
@@ -126,6 +126,7 @@ int util_unlink_secure(struct udev *udev
 
 uid_t util_lookup_user(struct udev *udev, const char *user)
 {
+	/*
 	char *endptr;
 	int buflen = sysconf(_SC_GETPW_R_SIZE_MAX);
 	char buf[buflen];
@@ -147,11 +148,13 @@ uid_t util_lookup_user(struct udev *udev
 		err(udev, "specified user '%s' unknown\n", user);
 	else
 		err(udev, "error resolving user '%s': %m\n", user);
+	*/
 	return 0;
 }
 
 gid_t util_lookup_group(struct udev *udev, const char *group)
 {
+	/*
 	char *endptr;
 	int buflen = sysconf(_SC_GETGR_R_SIZE_MAX);
 	char buf[buflen];
@@ -173,6 +176,7 @@ gid_t util_lookup_group(struct udev *ude
 		err(udev, "specified group '%s' unknown\n", group);
 	else
 		err(udev, "error resolving group '%s': %m\n", group);
+	*/
 	return 0;
 }
 
