--- udev/udev-util.c	2008-11-06 02:56:15.000000000 +0100
+++ udev/udev-util.c	2008-11-09 18:28:30.000000000 +0100
@@ -126,6 +126,7 @@
 
 uid_t util_lookup_user(struct udev *udev, const char *user)
 {
+	/*
 	char *endptr;
 	int buflen = sysconf(_SC_GETPW_R_SIZE_MAX);
 	char buf[buflen];
@@ -147,11 +148,13 @@
 		err(udev, "specified user '%s' unknown\n", user);
 	else
 		err(udev, "error resolving user '%s': %m\n", user);
+	*/
 	return 0;
 }
 
 extern gid_t util_lookup_group(struct udev *udev, const char *group)
 {
+	/*
 	char *endptr;
 	int buflen = sysconf(_SC_GETGR_R_SIZE_MAX);
 	char buf[buflen];
@@ -173,6 +176,7 @@
 		err(udev, "specified group '%s' unknown\n", group);
 	else
 		err(udev, "error resolving group '%s': %m\n", group);
+	*/
 	return 0;
 }

--- extras/Makefile.in	2008-09-11 16:58:01.000000000 +0200
+++ extras/Makefile.in	2008-09-18 12:03:09.000000000 +0200
@@ -176,15 +176,11 @@
 udev_prefix = @udev_prefix@
 SUBDIRS = \
 	ata_id \
-	cdrom_id \
 	edd_id \
 	path_id \
 	firmware \
-	collect \
 	floppy \
-	fstab_import \
 	rule_generator \
-	scsi_id \
 	usb_id \
 	volume_id
 
--- extras/volume_id/lib/libvolume_id-private.h	2008-09-10 02:37:09.000000000 +0200
+++ extras/volume_id/lib/libvolume_id-private.h	2008-09-18 12:07:57.000000000 +0200
@@ -35,12 +35,8 @@
 log_null(const char *format, ...) {}
 
 #define err(format, arg...)	volume_id_log_fn(LOG_ERR, __FILE__, __LINE__, format, ##arg)
-#define info(format, arg...)	volume_id_log_fn(LOG_INFO, __FILE__, __LINE__, format, ##arg)
-#ifdef DEBUG
-#define dbg(format, arg...)	volume_id_log_fn(LOG_DEBUG, __FILE__, __LINE__, format, ##arg)
-#else
+#define info(format, arg...)    log_null(format, ##arg)
 #define dbg(format, arg...)	log_null(format, ##arg)
-#endif
 
 #if (__BYTE_ORDER == __LITTLE_ENDIAN)
 #define le16_to_cpu(x) (x)
--- extras/volume_id/lib/md5.c	2008-09-10 02:18:59.000000000 +0200
+++ extras/volume_id/lib/md5.c	2008-09-18 12:03:09.000000000 +0200
@@ -21,7 +21,7 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-#include <stdint.h>
+/*#include <stdint.h>*/
 #include <string.h>
 #include <endian.h>
 #include <byteswap.h>

