--- udev/udev-util.c	2008-11-06 02:56:15.000000000 +0100
+++ udev/udev-util.c	2008-11-09 18:28:30.000000000 +0100
@@ -126,6 +126,7 @@
 
 uid_t util_lookup_user(struct udev *udev, const char *user)
 {
+	/*
 	char *endptr;
 	int buflen = sysconf(_SC_GETPW_R_SIZE_MAX);
 	char buf[buflen];
@@ -147,11 +148,13 @@
 		err(udev, "specified user '%s' unknown\n", user);
 	else
 		err(udev, "error resolving user '%s': %m\n", user);
+	*/
 	return 0;
 }
 
 extern gid_t util_lookup_group(struct udev *udev, const char *group)
 {
+	/*
 	char *endptr;
 	int buflen = sysconf(_SC_GETGR_R_SIZE_MAX);
 	char buf[buflen];
@@ -173,6 +176,7 @@
 		err(udev, "specified group '%s' unknown\n", group);
 	else
 		err(udev, "error resolving group '%s': %m\n", group);
+	*/
 	return 0;
 }

--- Makefile.in.orig	2009-03-08 01:04:54.032608782 +0100
+++ Makefile.in	2009-03-08 01:05:10.728461250 +0100
@@ -212,10 +212,7 @@ AM_CFLAGS = 
 AM_LDFLAGS = \
 	-Wl,--as-needed
 
-SUBDIRS = \
-	udev \
-	rules \
-	extras
+SUBDIRS = udev
 
 udevconfdir = $(sysconfdir)/udev
 udevconf_DATA = \
