--- a/src/lev/Proxy.cc	2014-12-19 22:24:37.000000000 +0100
+++ b/src/lev/Proxy.cc	2017-09-01 16:55:50.009413332 +0200
@@ -933,7 +933,7 @@
                     } else if (haveLocalCopy) {
                         // plain file
                         basic_ifstream<char> ifs(absExtPath.c_str(), ios::binary | ios::in);
-                        if (ifs != NULL)
+                        if (ifs)
                             Readfile(ifs, extCode);
                         else
                             haveLocalCopy = false;
