--- include/Makefile.in.orig	2010-05-11 19:03:20.645903854 +0200
+++ include/Makefile.in	2010-05-11 19:04:02.930565181 +0200
@@ -44,7 +44,7 @@ am__aclocal_m4_deps = $(top_srcdir)/acin
 am__configure_deps = $(am__aclocal_m4_deps) $(CONFIGURE_DEPENDENCIES) \
 	$(ACLOCAL_M4)
 mkinstalldirs = $(install_sh) -d
-CONFIG_HEADER = $(top_builddir)/fficonfig.h
+CONFIG_HEADER = $(builddir)/fficonfig.h
 CONFIG_CLEAN_FILES = ffi.h ffitarget.h
 CONFIG_CLEAN_VPATH_FILES =
 SOURCES =
@@ -203,7 +203,7 @@ top_srcdir = @top_srcdir@
 AUTOMAKE_OPTIONS = foreign
 DISTCLEANFILES = ffitarget.h
 EXTRA_DIST = ffi.h.in ffi_common.h
-includesdir = $(libdir)/@PACKAGE_NAME@-@PACKAGE_VERSION@/include
+includesdir = @prefix@/include
 nodist_includes_HEADERS = ffi.h ffitarget.h
 all: all-am
 
