--- include/Makefile.in.orig	2009-11-17 03:20:27.967671016 +0100
+++ include/Makefile.in	2009-11-17 03:21:20.354047994 +0100
@@ -185,7 +185,7 @@ top_srcdir = @top_srcdir@
 AUTOMAKE_OPTIONS = foreign
 DISTCLEANFILES = ffitarget.h
 EXTRA_DIST = ffi.h.in ffi_common.h
-includesdir = $(libdir)/@PACKAGE_NAME@-@PACKAGE_VERSION@/include
+includesdir = @prefix@/include
 nodist_includes_HEADERS = ffi.h ffitarget.h 
 all: all-am
 
