From 2c1570595e365a18c85fd1b12c8f56516742b664 Mon Sep 17 00:00:00 2001
From: Jeff 'Raid' Baitis <jeff@baitis.net>
Date: Sun, 2 Dec 2018 13:30:42 -0800
Subject: [PATCH] Fix for compilation against modern LibreSSL

---
 src/shrpx_tls.cc | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/shrpx_tls.cc b/src/shrpx_tls.cc
index 9aa789d07..2a8f28744 100644
--- src/shrpx_tls.cc
+++ src/shrpx_tls.cc
@@ -364,11 +364,11 @@ int tls_session_new_cb(SSL *ssl, SSL_SESSION *session) {
 
 namespace {
 SSL_SESSION *tls_session_get_cb(SSL *ssl,
-#if OPENSSL_1_1_API
+#if OPENSSL_1_1_API || LIBRESSL_2_7_API
                                 const unsigned char *id,
-#else  // !OPENSSL_1_1_API
+#else  // !(OPENSSL_1_1_API || LIBRESSL_2_7_API)
                                 unsigned char *id,
-#endif // !OPENSSL_1_1_API
+#endif // !(OPENSSL_1_1_API || LIBRESSL_2_7_API)
                                 int idlen, int *copy) {
   auto conn = static_cast<Connection *>(SSL_get_app_data(ssl));
   auto handler = static_cast<ClientHandler *>(conn->data);
