--- a/project.py
+++ b/project.py
@@ -54,6 +54,15 @@ class PyQt(PyQtProject):
                 QtSerialPort, QtWebChannel, QtWebSockets, QtBluetooth, QtNfc,
                 QtPdf, QtPdfWidgets, QtTextToSpeech, QAxContainer]
 
+    def run_command(self, args, *, fatal=True):
+        """ Run a command and display the output if requested. """
+        qemu_machine = os.environ.get("XBPS_TARGET_QEMU_MACHINE")
+        builddir = os.environ.get("XBPS_BUILDDIR")
+        if qemu_machine and args[0].startswith(os.path.join(builddir, "python3-pyqt6")):
+            qemu = "qemu-{}-static".format(qemu_machine)
+            args.insert(0, qemu)
+        super().run_command(args, fatal=fatal)
+
     def apply_user_defaults(self, tool):
         """ Set default values where needed. """
 
