diff --git a/src/cairo-xcb-surface.c b/src/cairo-xcb-surface.c
index c350f6f..ff8d9e6 100644
--- src/cairo-xcb-surface.c
+++ src/cairo-xcb-surface.c
@@ -186,10 +186,10 @@ _cairo_xcb_surface_create_similar (void			*abstract_other,
     cairo_status_t status;
 
     if (width > XLIB_COORD_MAX || height > XLIB_COORD_MAX)
-	return NULL;
+	return _cairo_surface_create_in_error (_cairo_error (CAIRO_STATUS_INVALID_SIZE));
 
     if (width <= 0 || height <= 0)
-	return NULL;
+	return _cairo_surface_create_in_error (_cairo_error (CAIRO_STATUS_INVALID_SIZE));
 
 #if CAIRO_HAS_DRM_SURFACE && CAIRO_HAS_XCB_DRM_FUNCTIONS
     if (other->drm != NULL) {
