--- a/src/nm-l2tp-service.c
+++ b/src/nm-l2tp-service.c
@@ -1943,13 +1943,13 @@ get_localaddr (NML2tpPlugin *self,
     addr.sin_port = htons(1701);
     addr.sin_addr.s_addr = priv->naddr;
 
-    if (0 != connect (fd, &addr, sizeof(addr))) {
+    if (0 != connect (fd, (struct sockaddr *)&addr, sizeof(addr))) {
         close (fd);
         return nm_l2tp_ipsec_error(error, _("unable to connect to L2TP VPN gateway"));
     }
 
     memset( &addr, 0, sizeof(addr));
-    if (0 != getsockname( fd, &addr, &alen)) {
+    if (0 != getsockname( fd, (struct sockaddr *)&addr, &alen)) {
         close (fd);
         return nm_l2tp_ipsec_error(error, _("failed to get local IP"));
     }
