--- a/attica-kde/kdeplugin/CMakeLists.txt
+++ b/attica-kde/kdeplugin/CMakeLists.txt
@@ -22,5 +22,5 @@ target_link_libraries(attica_kde
 )
 
 install(TARGETS attica_kde                 DESTINATION ${KDE_INSTALL_PLUGINDIR})
-kaccounts_add_provider(opendesktop.provider.in)
-kaccounts_add_service(opendesktop-rating.service.in)
+kaccounts_add_provider(opendesktop.provider.in NO_INTLTOOL)
+kaccounts_add_service(opendesktop-rating.service.in NO_INTLTOOL)
--- a/attica-kde/kdeplugin/opendesktop-rating.service.in
+++ b/attica-kde/kdeplugin/opendesktop-rating.service.in
@@ -1,8 +1,8 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <service id="opendesktop-rating">
   <type>opendesktop-rating</type>
-  <_name msgctxt="OpenDesktop Rating and Commenting">Rating and Commenting</_name>
-  <_description>The service which allows you to vote and comment on anything on the OpenDesktop store</_description>
+  <name msgctxt="OpenDesktop Rating and Commenting">Rating and Commenting</name>
+  <description>The service which allows you to vote and comment on anything on the OpenDesktop store</description>
   <icon>get-hot-new-stuff</icon>
   <provider>opendesktop</provider>
   <translations>kaccounts-providers</translations>
