--- Makefile.PL.orig
+++ Makefile.PL
@@ -3,8 +3,8 @@
 # License: AGPL-3.0+ <https://www.gnu.org/licenses/agpl-3.0.txt>
 use strict;
 use ExtUtils::MakeMaker;
-my @EXE_FILES = split("\n", `git ls-files 'script/*' 2>/dev/null`);
-my $PM_FILES = `git ls-files lib '*.pm' 2>/dev/null`;
+my @EXE_FILES = split("\n", `printf '%s\n' script/* 2>/dev/null`);
+my $PM_FILES = `find lib 2>/dev/null`;
 $PM_FILES =~ tr/\n/ /;
 
 WriteMakefile(
