--- qtwebkit-2.3.4/Source/WebCore/platform/qt/MemoryUsageSupportQt.cpp	2014-09-24 13:42:05.000000000 +0200
+++ qtwebkit-2.3.4/Source/WebCore/platform/qt/MemoryUsageSupportQt.cpp	2016-09-17 23:48:23.558775636 +0200
@@ -31,7 +31,7 @@
 
 namespace WebCore {
 
-#if OS(LINUX)
+#if OS(LINUX) && defined(__GLIBC__)
 static size_t mallocMemoryUsage(bool inuse)
 {
     // Return how much memory (in bytes) has been allocated on the system heap.
