--- loudmouth/lm-sasl.c	2016-02-13 12:16:45.000000000 +0100
+++ loudmouth/lm-sasl.c	2017-11-12 14:55:20.995838985 +0100
@@ -529,7 +529,7 @@
     }
 
     nonce = g_hash_table_lookup (challenge, "nonce");
-    if (nonce == NULL || nonce == '\0') {
+    if (nonce == NULL || *nonce == '\0') {
         g_log (LM_LOG_DOMAIN, LM_LOG_LEVEL_SASL,
                "%s: server didn't provide a nonce in the challenge",
                G_STRFUNC);
