Add definitions for LONG_LONG_MIN and _MAX derived
from the values for int64_t.

--- resize_reiserfs/resize_reiserfs.c	2013-08-26 22:55:43.000000000 +0200
+++ resize_reiserfs/resize_reiserfs.c	2015-09-10 09:30:46.617279386 +0200
@@ -15,6 +15,13 @@
 #include "resize.h"
 #include <limits.h>
 
+#if !defined(__GLIBC__)
+/* These are not defined in musl libc */
+#include <stdint.h>
+#define LONG_LONG_MIN INT64_MIN
+#define LONG_LONG_MAX INT64_MAX
+#endif
+
 int opt_banner = 0;
 int opt_force = 0;
 int opt_verbose = 1;			/* now "verbose" option is default */
