diff --git a/crates/zed/Cargo.toml b/crates/zed/Cargo.toml
index b0893a7..6ec00e8 100644
--- a/crates/zed/Cargo.toml
+++ b/crates/zed/Cargo.toml
@@ -6,7 +6,7 @@ version = "1.19.2"
 publish.workspace = true
 license = "GPL-3.0-or-later"
 authors = ["Zed Team <hi@zed.dev>"]
-default-run = "zed"
+default-run = "zedit"
 
 [lints]
 workspace = true
@@ -59,7 +59,7 @@ visual-tests = [
 ]
 
 [[bin]]
-name = "zed"
+name = "zedit"
 path = "src/main.rs"
 
 [[bin]]
