--- a/libfm/iconlist.h	2019-06-04 07:47:00.000000000 +0200
+++ b/libfm/iconlist.h	2020-08-31 18:50:21.401362347 +0200
@@ -9,6 +9,7 @@
 #include <QStyledItemDelegate>
 #include <QModelIndex>
 #include <QPainter>
+#include <QPainterPath>
 
 class IconListDelegate : public QItemDelegate
 {
--- a/libfm/iconview.h	2019-06-04 07:47:00.000000000 +0200
+++ b/libfm/iconview.h	2020-08-31 18:50:30.723362828 +0200
@@ -11,6 +11,7 @@
 #include <QKeyEvent>
 #include <QModelIndex>
 #include <QPainter>
+#include <QPainterPath>
 
 class IconViewDelegate : public QStyledItemDelegate
 {
