Index: cinnamon/files/usr/share/cinnamon/applets/calendar@cinnamon.org/applet.js
===================================================================
--- cinnamon.orig/files/usr/share/cinnamon/applets/calendar@cinnamon.org/applet.js	2014-04-27 11:56:10.582581808 +0200
+++ cinnamon/files/usr/share/cinnamon/applets/calendar@cinnamon.org/applet.js	2014-07-14 00:44:37.882215797 +0200
@@ -94,7 +94,11 @@
 
             // https://bugzilla.gnome.org/show_bug.cgi?id=655129
             this._upClient = new UPowerGlib.Client();
-            this._upClient.connect('notify-resume', this._updateClockAndDate);
+            try {
+                this._upClient.connect('notify-resume', this._updateClockAndDate);
+            } catch (e) {
+                this._upClient.connect('notify::resume', this._updateClockAndDate);
+            }
 
             // Start the clock
             this.on_settings_changed();
