--- data/theme/cinnamon.css	2014-08-12 13:00:36.000000000 +0200
+++ data/theme/cinnamon.css	2014-10-01 02:51:45.859630151 +0200
@@ -1445,7 +1445,8 @@
 .applet-icon:hover,
 .applet-box:hover > .applet-icon {
 	color: #fff;
-	icon-shadow: white 0px 0px 3px;
+/* Broken with GNOME 3.14 (icon-size is not honoured)
+ *	icon-shadow: white 0px 0px 3px;*/
 }
 
 /* ===================================================================
--- files/usr/lib/cinnamon-settings/modules/cs_themes.py	2014-08-12 13:00:36.000000000 +0200
+++ files/usr/lib/cinnamon-settings/modules/cs_themes.py	2014-10-01 02:52:56.897767729 +0200
@@ -91,7 +91,7 @@
     
     def _load_icon_themes(self):
         dirs = ("/usr/share/icons", os.path.join(os.path.expanduser("~"), ".icons"))
-        valid = walk_directories(dirs, lambda d: os.path.isdir(d) and not os.path.exists(os.path.join(d, "cursors")) and os.path.exists(os.path.join(d, "index.theme")))
+        valid = walk_directories(dirs, lambda d: os.path.isdir(d) and os.path.exists(os.path.join(d, "index.theme")))
         valid.sort(lambda a,b: cmp(a.lower(), b.lower()))
         res = []
         for i in valid:
