--- a/data/cinnamon-wayland.desktop.in.in
+++ b/data/cinnamon-wayland.desktop.in.in
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Type=Application
-_Name=Cinnamon (Wayland session)
-_Comment=Window management and application launching
+Name=Cinnamon (Wayland session)
+Comment=Window management and application launching
 Exec=cinnamon --replace
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=cinnamon
--- a/data/cinnamon.desktop.in.in
+++ b/data/cinnamon.desktop.in.in
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Type=Application
-_Name=Cinnamon
-_Comment=Window management and application launching
+Name=Cinnamon
+Comment=Window management and application launching
 Exec=@bindir@/cinnamon-launcher
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=cinnamon
--- a/data/cinnamon2d.desktop.in.in
+++ b/data/cinnamon2d.desktop.in.in
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Type=Application
-_Name=Cinnamon 2D
-_Comment=Window management and application launching
+Name=Cinnamon 2D
+Comment=Window management and application launching
 Exec=@bindir@/cinnamon2d
 Categories=GNOME;GTK;System;Core;
 OnlyShowIn=X-Cinnamon;
--- a/data/meson.build
+++ b/data/meson.build
@@ -29,17 +29,15 @@ if get_option('wayland')
 endif
 
 foreach desktop_file : desktop_files
-    desktop = configure_file(
-        input: desktop_file + '.in.in',
-        output: desktop_file + '.in',
-        configuration: desktop_conf,
-    )
-
-    custom_target(
-        desktop_file,
-        input: desktop,
+    i18n.merge_file(
+        type: 'desktop',
+        input: configure_file(
+            input: desktop_file + '.in.in',
+            output: desktop_file + '.in',
+            configuration: desktop_conf,
+        ),
         output: desktop_file,
-        command: [intltool_merge, '-d', '-u', po_dir, '@INPUT@', '@OUTPUT@'],
+        po_dir: po_dir,
         install: true,
         install_dir: desktopdir,
     )
--- a/meson.build
+++ b/meson.build
@@ -65,7 +65,6 @@ endif
 cc = meson.get_compiler('c')
 math = cc.find_library('m', required: false)
 
-intltool_merge = find_program('intltool-merge')
 python = find_program('python3')
 
 # generate config.h
