From e8ba70f898179ede0170e36db49ca27570b12371 Mon Sep 17 00:00:00 2001
From: Nathan <ndowens04@gmail.com>
Date: Wed, 9 Jan 2019 21:03:56 -0600
Subject: [PATCH 1/2] Fix glibc build

---
 include/linux/netfilter.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git include/linux/netfilter.h include/linux/netfilter.h
index c3f087a..c934044 100644
--- include/linux/netfilter.h
+++ include/linux/netfilter.h
@@ -2,9 +2,10 @@
 #define __LINUX_NETFILTER_H
 
 #include <linux/types.h>
-
+#ifndef __GLIBC__
 #include <linux/in.h>
 #include <linux/in6.h>
+#endif
 #include <limits.h>
 
 /* Responses from hook functions. */
-- 
2.20.1

