From 3186b4418d5c11fd7943623722478eec9727282a Mon Sep 17 00:00:00 2001
From: Nathan <ndowens04@gmail.com>
Date: Wed, 9 Jan 2019 21:04:59 -0600
Subject: [PATCH 2/2] Fix musl build

---
 extensions/libebt_vlan.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git extensions/libebt_vlan.c extensions/libebt_vlan.c
index 4a2eb71..a5c7dc4 100644
--- extensions/libebt_vlan.c
+++ extensions/libebt_vlan.c
@@ -13,7 +13,9 @@
 #include <ctype.h>
 #include <xtables.h>
 #include <linux/netfilter_bridge/ebt_vlan.h>
+#ifdef __GLIBC__
 #include <linux/if_ether.h>
+#endif
 #include "iptables/nft.h"
 #include "iptables/nft-bridge.h"
 
-- 
2.20.1

