--- Makefile.orig
+++ Makefile
@@ -72,9 +72,9 @@
 # Make Configuration
 #
 CC = gcc
-COMPILER_OPTIONS = -Wall -O -g
+CFLAGS ?= -Wall -O -g
 
-CFLAGS= $(COMPILER_OPTIONS) -I. -DVERSION=\"$(VERSION)\" $(EXTERNAL_MD5) $(OMIT_GETOPT_LONG) $(FILEOFFSET_64BIT)
+CFLAGS += -I. -DVERSION=\"$(VERSION)\" $(EXTERNAL_MD5) $(OMIT_GETOPT_LONG) $(FILEOFFSET_64BIT)
 
 INSTALL_PROGRAM = $(INSTALL) -c -m 0755
 INSTALL_DATA    = $(INSTALL) -c -m 0644
@@ -94,7 +94,7 @@
 all: fdupes
 
 fdupes: $(OBJECT_FILES)
-	$(CC) $(CFLAGS) -o fdupes $(OBJECT_FILES)
+	$(CC) $(CFLAGS) $(LDFLAGS) -o fdupes $(OBJECT_FILES)
 
 installdirs:
 	test -d $(BIN_DIR) || $(MKDIR) $(BIN_DIR)
