--- a/re_types.h	2018-01-14 21:33:25.561341291 +0100
+++ b/include/re_types.h	2018-01-14 21:34:45.444384148 +0100
@@ -37,20 +37,25 @@
 typedef signed long long int      int64_t;
 
 #ifndef __uint32_t_defined
+#ifdef __GLIBC__
 #define __uint32_t_defined
 typedef unsigned char             uint8_t;
 typedef unsigned short int        uint16_t;
 typedef unsigned int              uint32_t;
 typedef unsigned long long int    uint64_t;
 #endif
+#endif
 
 #endif /* __BIT_TYPES_DEFINED__ */
 
 #endif /* __int8_t_defined */
 #ifndef __ssize_t_defined
+#ifndef __DEFINED_ssize_t
 typedef long     ssize_t;
+#define __DEFINED_ssize_t
 #define __ssize_t_defined
 #endif
+#endif
 
 
 #ifndef WIN32
