--- src/dns/res.c.orig	2016-12-03 15:17:53.000000000 +0100
+++ src/dns/res.c	2016-12-03 15:18:30.000000000 +0100
@@ -25,13 +25,9 @@ int get_resolv_dns(char *domain, size_t
 	uint32_t i;
 	int ret, err;
 
-#ifdef OPENBSD
 	ret = res_init();
 	state = _res;
-#else
-	memset(&state, 0, sizeof(state));
-	ret = res_ninit(&state);
-#endif
+
 	if (0 != ret)
 		return ENOENT;
 
@@ -56,10 +52,5 @@ int get_resolv_dns(char *domain, size_t
 	*n = i;
 
  out:
-#ifdef OPENBSD
-#else
-	res_nclose(&state);
-#endif
-
 	return err;
 }
