From 6f30e9034f29c3ae6ad7e617b3d5e903aa107b6a Mon Sep 17 00:00:00 2001
From: Reuben Thomas <rrt@sc3d.org>
Date: Mon, 27 Jan 2014 14:18:24 +0000
Subject: en_US.UTF-8/Compose: Fix apparent copy-paste-o, changing capital to small A.

Signed-off-by: Alan Coopersmith <alan.coopersmith@oracle.com>
---
diff --git a/nls/en_US.UTF-8/Compose.pre b/nls/en_US.UTF-8/Compose.pre
index 5434425..7cdcb49 100644
--- nls/en_US.UTF-8/Compose.pre
+++ nls/en_US.UTF-8/Compose.pre
@@ -818,8 +818,8 @@ XCOMM Part 3
 <Multi_key> <macron> <a>         	: "ā"   U0101 # LATIN SMALL LETTER A WITH MACRON
 <Multi_key> <underscore> <a>     	: "ā"   U0101 # LATIN SMALL LETTER A WITH MACRON
 <Multi_key> <a> <underscore>     	: "ā"   U0101 # LATIN SMALL LETTER A WITH MACRON
-<Multi_key> <minus> <a>          	: "ā"   U0100 # LATIN CAPITAL LETTER A WITH MACRON
-<Multi_key> <a> <minus>          	: "ā"   U0100 # LATIN CAPITAL LETTER A WITH MACRON
+<Multi_key> <minus> <a>          	: "ā"   U0101 # LATIN SMALL LETTER A WITH MACRON
+<Multi_key> <a> <minus>          	: "ā"   U0101 # LATIN SMALL LETTER A WITH MACRON
 <dead_breve> <A>                 	: "Ă"   U0102 # LATIN CAPITAL LETTER A WITH BREVE
 <Multi_key> <U> <A>              	: "Ă"   U0102 # LATIN CAPITAL LETTER A WITH BREVE
 <Multi_key> <b> <A>              	: "Ă"   U0102 # LATIN CAPITAL LETTER A WITH BREVE
--
cgit v0.9.0.2-2-gbebe
