From 61877de88d782b63b17458a61fcc078391499b29 Mon Sep 17 00:00:00 2001
From: Paul Dreik <rdfind@pauldreik.se>
Date: Sun, 15 Aug 2021 20:31:31 +0200
Subject: [PATCH] include <limits> (thanks to tastytea)

---
 rdfind.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/rdfind.cc b/rdfind.cc
index fbd6cb8..64dd8f6 100644
--- a/rdfind.cc
+++ b/rdfind.cc
@@ -9,6 +9,7 @@
 // std
 #include <algorithm>
 #include <iostream>
+#include <limits>
 #include <string>
 #include <vector>
 
