From f6c3f698dd680931b5c2f05688319290bdf0d930 Mon Sep 17 00:00:00 2001
From: Paul Dreik <rdfind@pauldreik.se>
Date: Fri, 16 Jun 2023 08:19:29 +0200
Subject: [PATCH] add missing include <cstdint>

---
 Fileinfo.hh | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Fileinfo.hh b/Fileinfo.hh
index 3ffb837..23ed54e 100644
--- a/Fileinfo.hh
+++ b/Fileinfo.hh
@@ -8,6 +8,7 @@
 #define Fileinfo_hh
 
 #include <array>
+#include <cstdint>
 #include <string>
 
 // os specific headers
