From 636e5cd2d53c8165a0dc5f857af643fd15faea9d Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Fri, 19 Oct 2012 11:09:38 +0200
Subject: [PATCH 1/4] dracut.sh: only save $kernel_cmdline, if set

---
 dracut.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dracut.sh b/dracut.sh
index e160641..aab850e 100755
--- a/dracut.sh
+++ b/dracut.sh
@@ -937,7 +937,7 @@ fi
 if [[ $kernel_only != yes ]]; then
     (( ${#install_items[@]} > 0 )) && dracut_install  ${install_items[@]}
 
-    echo "$kernel_cmdline" >> "${initdir}/etc/cmdline.d/01-default.conf"
+    [[ $kernel_cmdline ]] && echo "$kernel_cmdline" >> "${initdir}/etc/cmdline.d/01-default.conf"
 
     while pop fstab_lines line; do
         echo "$line 0 0" >> "${initdir}/etc/fstab"
-- 
1.8.0

