From b26897fb6099ed9ee67ed6e19e751be11d38fcf3 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Fri, 19 Oct 2012 11:20:07 +0200
Subject: [PATCH] dracut.sh: only 'warn' not 'error', if we don't strip

---
 dracut.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dracut.sh b/dracut.sh
index aab850e..e463404 100755
--- a/dracut.sh
+++ b/dracut.sh
@@ -1031,7 +1031,7 @@ fi
 if [[ $do_strip = yes ]] ; then
     for p in strip xargs find; do
         if ! type -P $p >/dev/null; then
-            derror "Could not find '$p'. You should run $0 with '--nostrip'."
+            dwarn "Could not find '$p'. Not stripping the initramfs."
             do_strip=no
         fi
     done
-- 
1.8.0

