--- dracut.sh.orig	2012-07-06 10:31:36.651370365 +0200
+++ dracut.sh	2012-07-06 10:31:53.234973758 +0200
@@ -877,7 +877,7 @@ if [[ $do_strip = yes ]] ; then
     find "$initdir" -type f \
         '(' -perm -0100 -or -perm -0010 -or -perm -0001 \
         -or -path '*/lib/modules/*.ko' ')' -print0 \
-        | xargs -0 strip -g 2>/dev/null
+        | xargs -r -0 strip -g 2>/dev/null
     dinfo "*** Stripping files done ***"
 fi
 
--- dracut-functions.sh.orig	2012-07-06 10:32:18.282374771 +0200
+++ dracut-functions.sh	2012-07-06 10:33:41.727379388 +0200
@@ -1182,7 +1182,7 @@ dracut_kernel_post() {
     local _moddirname=${srcmods%%/lib/modules/*}
 
     if [[ -f "$initdir/.kernelmodseen/lazylist" ]]; then
-        xargs modprobe -a ${_moddirname+-d ${_moddirname}/} --ignore-install --show-depends \
+        xargs -r modprobe -a ${_moddirname+-d ${_moddirname}/} --ignore-install --show-depends --set-version $kernel \
             < "$initdir/.kernelmodseen/lazylist" 2>/dev/null \
             | sort -u \
             | while read _cmd _modpath _options; do
@@ -1192,7 +1192,7 @@ dracut_kernel_post() {
 
         (
             if [[ -x /usr/bin/dracut-install ]] && [[ -z $_moddirname ]]; then
-                xargs dracut-install ${initdir+-D "$initdir"} -a < "$initdir/.kernelmodseen/lazylist.dep"
+                xargs -r dracut-install ${initdir+-D "$initdir"} -a < "$initdir/.kernelmodseen/lazylist.dep"
             else
                 while read _modpath; do
                     local _destpath=$_modpath
@@ -1205,14 +1205,14 @@ dracut_kernel_post() {
 
 
         if [[ -x /usr/bin/dracut-install ]]; then
-            xargs modinfo -k $kernel -F firmware < "$initdir/.kernelmodseen/lazylist.dep" \
+            xargs -r modinfo -k $kernel -F firmware < "$initdir/.kernelmodseen/lazylist.dep" \
                 | while read line; do
                 for _fwdir in $fw_dir; do
                     echo $_fwdir/$line;
                 done;
-            done |xargs dracut-install ${initdir+-D "$initdir"} -a -o
+            done |xargs -r dracut-install ${initdir+-D "$initdir"} -a -o
         else
-            for _fw in $(xargs modinfo -k $kernel -F firmware < "$initdir/.kernelmodseen/lazylist.dep"); do
+            for _fw in $(xargs -r modinfo -k $kernel -F firmware < "$initdir/.kernelmodseen/lazylist.dep"); do
                 for _fwdir in $fw_dir; do
                     if [[ -d $_fwdir && -f $_fwdir/$_fw ]]; then
                         inst_simple "$_fwdir/$_fw" "/lib/firmware/$_fw"
@@ -1258,7 +1258,7 @@ find_kernel_modules_by_path () (
         IFS=$_OLDIFS
     else
         ( cd /sys/module; echo *; ) \
-        | xargs modinfo -F filename -k $kernel 2>/dev/null
+        | xargs -r modinfo -F filename -k $kernel 2>/dev/null
     fi
     return 0
 )
