From 636cc1ef634704ffc5e037961b6fb8ede3203977 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Wed, 7 May 2014 15:30:01 +0200
Subject: resume/parse-resume.sh: correctly write timeout hook


diff --git a/modules.d/95resume/parse-resume.sh b/modules.d/95resume/parse-resume.sh
index d7b2d7f..6ab59c3 100755
--- modules.d/95resume/parse-resume.sh
+++ modules.d/95resume/parse-resume.sh
@@ -70,9 +70,10 @@ if ! getarg noresume; then
         printf '[ -e "%s" ] && { ln -s "%s" /dev/resume; rm -f -- "$job" "%s/initqueue/timeout/resume.sh"; }\n' \
             "$resume" "$resume" "$hookdir" >> $hookdir/initqueue/settled/resume.sh
 
-        printf -- "%s" 'warn "Cancelling resume operation. Device not found.";'
-        printf -- ' cancel_wait_for_dev /dev/resume; rm -f -- "$job" "%s/initqueue/settled/resume.sh";\n' \
-            "$hookdir" >> $hookdir/initqueue/timeout/resume.sh
+        {
+            printf -- "%s" 'warn "Cancelling resume operation. Device not found.";'
+            printf -- ' cancel_wait_for_dev /dev/resume; rm -f -- "$job" "%s/initqueue/settled/resume.sh";\n' "$hookdir"
+        } >> $hookdir/initqueue/timeout/resume.sh
 
         mv /lib/dracut/resume.sh /lib/dracut/hooks/pre-mount/10-resume.sh
     else
-- 
cgit v0.10.1

