typo fix that avoids '1=""' in generated vconsole.conf.

--- modules.d/10i18n/parse-i18n.sh.orig	2014-04-28 12:44:54.540715927 +0200
+++ modules.d/10i18n/parse-i18n.sh	2014-04-28 12:45:10.587813907 +0200
@@ -12,7 +12,7 @@ inst_key_val() {
     _value="$(getarg $@)"
     [ -z "${_value}" ] && _value=$_default
     if [ -n "${_value}" ]; then
-        printf '%s="%s"\n' $key ${_value} >> $_file
+        printf '%s="%s"\n' ${_key} ${_value} >> $_file
     fi
     unset _file
     unset _value
