systemd binary is in ${systemdutildir} by default.

--- modules.d/98systemd/module-setup.sh.orig	2013-03-08 11:52:43.941344749 +0100
+++ modules.d/98systemd/module-setup.sh	2013-03-08 11:52:56.437417847 +0100
@@ -17,7 +17,7 @@ depends() {
 
 install() {
 
-    SYSTEMD_VERSION=$(systemd --version | { read a b a; echo $b; })
+    SYSTEMD_VERSION=$(${systemdutildir}/systemd --version | { read a b a; echo $b; })
     if (( $SYSTEMD_VERSION < 198 )); then
         dfatal "systemd version $SYSTEMD_VERSION is too low. Need at least version 198."
         exit 1
