diff --git modules.d/99base/dracut-lib.sh modules.d/99base/dracut-lib.sh
index 99cb9dbc..3df77257 100755
--- modules.d/99base/dracut-lib.sh
+++ modules.d/99base/dracut-lib.sh
@@ -1159,11 +1159,11 @@
 
     case "$_emergency_action" in
         reboot)
-            reboot || exit 1;;
+            reboot -f || exit 1;;
         poweroff)
-            poweroff || exit 1;;
+            poweroff -f || exit 1;;
         halt)
-            halt || exit 1;;
+            halt -f || exit 1;;
     esac
 }
 
