--- modules.d/90lvm/module-setup.sh.orig	2015-08-20 16:39:33.989335973 +0200
+++ modules.d/90lvm/module-setup.sh	2015-08-20 16:39:48.163501858 +0200
@@ -117,6 +117,7 @@ install() {
     inst_script "$moddir/lvm_scan.sh" /sbin/lvm_scan
     inst_hook cmdline 30 "$moddir/parse-lvm.sh"
 
+    inst_binary wc
     inst_libdir_file "libdevmapper-event-lvm*.so"
 
     if [[ $hostonly ]] && type -P lvs &>/dev/null; then
