--- a/modules.d/95udev-rules/module-setup.sh	2018-02-19 06:00:33.000000000 -0300
+++ b/modules.d/95udev-rules/module-setup.sh	2018-05-02 00:27:53.710253893 -0300
@@ -39,6 +39,8 @@
         60-persistent-storage.rules \
         61-persistent-storage-edd.rules \
         70-uaccess.rules \
+        71-plymouth-seat.rules \
+        71-udev-seat.rules \
         71-seat.rules \
         73-seat-late.rules \
         75-net-description.rules \
