adds udev rule that allows plymouth to work

--- a/modules.d/95udev-rules/module-setup.sh	2018-02-19 06:00:33.000000000 -0300
+++ b/modules.d/95udev-rules/module-setup.sh	2018-05-02 00:27:53.710253893 -0300
@@ -49,6 +49,7 @@
         70-mouse.rules \
         70-touchpad.rules \
         70-uaccess.rules \
+        71-plymouth-seat.rules \
         71-seat.rules \
         73-seat-late.rules \
         75-net-description.rules \
