diff --git a/modules.d/50drm/module-setup.sh b/modules.d/50drm/module-setup.sh
index 382f51a9..eb3fc9d3 100755
--- modules.d/50drm/module-setup.sh
+++ modules.d/50drm/module-setup.sh
@@ -40,6 +40,6 @@ installkernel() {
             fi
         done
     else
-        dracut_instmods -s "drm_crtc_init" "=drivers/gpu/drm" "=drivers/staging"
+        dracut_instmods -o -s "drm_crtc_init" "=drivers/gpu/drm" "=drivers/staging"
     fi
 }
