From 014ec783c9bf11ae352934ee2e593f29700b2843 Mon Sep 17 00:00:00 2001
From: Hillwood Yang <hillwood@opensuse.org>
Date: Sat, 25 Jan 2020 20:39:39 +0800
Subject: [PATCH] Fix hash<QString> error above Qt 5.14

---
 src/hashqstring.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/hashqstring.h b/src/hashqstring.h
index 20522b3..71d71c4 100644
--- src/hashqstring.h
+++ src/hashqstring.h
@@ -27,6 +27,7 @@
 #include <QString>
 #include <string>
 
+#if QT_VERSION < QT_VERSION_CHECK(5, 14, 0)
 namespace std {
     template<> struct hash<QString>
     {
@@ -36,5 +37,6 @@ namespace std {
             }
     };
 }
+#endif
 
 #endif // HASHQSTRING_H
