--- a/Makefile	2015-07-25 21:11:40.397311183 +0200
+++ b/Makefile	2015-07-25 21:11:59.561311865 +0200
@@ -1,5 +1,5 @@
-CC=valac
-CFLAGS=--pkg gtk+-3.0
-LDFLAGS=-X -lm
+VALAC=valac
+VALACFLAGS=--pkg gtk+-3.0
+VALALDFLAGS=-X -lm
 SOURCE=gtk-theme-config.vala
 BINARY=gtk-theme-config
@@ -14,7 +14,7 @@ INSTALL_DATA=$(INSTALL) -Dpm 0644
 all: $(BINARY)
 
 $(BINARY): $(SOURCE)
-	$(CC) $(CFLAGS) $(LDFLAGS) $(SOURCE) -o $(BINARY)
+	$(VALAC) $(VALACFLAGS) $(VALALDFLAGS) $(SOURCE) -o $(BINARY)
 
 clean:
 	$(CLEAN) $(BINARY)
