From 81e5cf58b44be4ec0dc50722e2ed6d534df3973d Mon Sep 17 00:00:00 2001
From: Andrzej Rybczak <andrzej@rybczak.net>
Date: Thu, 30 May 2024 14:58:09 +0200
Subject: [PATCH] extras: fix compilation with taglib 2.0

---
 extras/artist_to_albumartist.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/extras/artist_to_albumartist.cpp b/extras/artist_to_albumartist.cpp
index c3ddfee3a..0349d0aed 100644
--- a/extras/artist_to_albumartist.cpp
+++ b/extras/artist_to_albumartist.cpp
@@ -34,7 +34,7 @@ enum class CopyResult { Success, NoArtist, AlbumArtistAlreadyInPlace };
 bool is_framelist_empty(const TagLib::ID3v2::FrameList &list)
 {
 	for (auto it = list.begin(); it != list.end(); ++it)
-		if ((*it)->toString() != TagLib::String::null)
+		if (!(*it)->toString().isEmpty())
 			return false;
 	return true;
 }
