--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -263,7 +263,7 @@
 	install(
 		EXPORT glm
 		NAMESPACE glm::
-		DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/glm"
+		DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/glm"
 		FILE glmConfig.cmake
 	)
 	include(CMakePackageConfigHelpers)
@@ -273,7 +273,7 @@
 	)
 	install(
 		FILES "${CMAKE_CURRENT_BINARY_DIR}/glmConfigVersion.cmake"
-		DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/glm"
+		DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/glm"
 	)
 
 	configure_file(
