--- CMakeLists.txt.orig	2016-05-30 17:56:45.738244928 +0200
+++ CMakeLists.txt	2016-05-30 17:57:44.079247938 +0200
@@ -176,7 +176,7 @@
 
 install(
     FILES 
-        "${CMAKE_CURRENT_BINARY_DIR}/${GLM_INSTALL_CONFIGDIR}/glmConfig.cmake"
+        "${CMAKE_CURRENT_BINARY_DIR}/glmConfig.cmake"
         "${CMAKE_CURRENT_BINARY_DIR}/glmVersion.cmake"
     DESTINATION ${GLM_INSTALL_CONFIGDIR}
 )
