diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
index 93abbfa..2ac0df3 100644
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -11,7 +11,6 @@
 	endif()
 
 	if(NOT GLM_DISABLE_AUTO_DETECTION)
-		add_compile_options(-Werror -Weverything)
 	endif()
 
 elseif(CMAKE_CXX_COMPILER_ID MATCHES "GNU")
@@ -20,12 +19,10 @@
 	endif()
 
 	if(NOT GLM_DISABLE_AUTO_DETECTION)
-		add_compile_options(-Werror)
 #		add_compile_options(-Wpedantic)
 #		add_compile_options(-Wall)
 #		add_compile_options(-Wextra)
 	endif()
-	add_compile_options(-O2)
 	#add_compile_options(-Wno-long-long)
 elseif(CMAKE_CXX_COMPILER_ID MATCHES "Intel")
 	if(NOT GLM_QUIET)
