--- a/qtpy/__init__.py
+++ b/qtpy/__init__.py
@@ -164,12 +164,12 @@ PYSIDE_VERSION_MIN = PYSIDE2_VERSION_MIN
 binding_specified = QT_API in os.environ
 
 API_NAMES = {
-    "pyqt5": "PyQt5",
-    "pyside2": "PySide2",
     "pyqt6": "PyQt6",
     "pyside6": "PySide6",
+    "pyqt5": "PyQt5",
+    "pyside2": "PySide2",
 }
-API = os.environ.get(QT_API, "pyqt5").lower()
+API = os.environ.get(QT_API, "pyqt6").lower()
 initial_api = API
 if API not in API_NAMES:
     raise PythonQtValueError(
@@ -178,8 +178,8 @@ if API not in API_NAMES:
     )
 
 is_old_pyqt = is_pyqt46 = False
-QT5 = PYQT5 = True
-QT4 = QT6 = PYQT4 = PYQT6 = PYSIDE = PYSIDE2 = PYSIDE6 = False
+QT6 = PYQT6 = True
+QT4 = QT5 = PYQT4 = PYQT5 = PYSIDE = PYSIDE2 = PYSIDE6 = False
 
 PYQT_VERSION = None
 PYSIDE_VERSION = None
@@ -187,14 +187,16 @@ QT_VERSION = None
 
 # Unless `FORCE_QT_API` is set, use previously imported Qt Python bindings
 if not os.environ.get("FORCE_QT_API"):
-    if "PyQt5" in sys.modules:
-        API = initial_api if initial_api in PYQT5_API else "pyqt5"
-    elif "PySide2" in sys.modules:
-        API = initial_api if initial_api in PYSIDE2_API else "pyside2"
+    if False:
+        pass
     elif "PyQt6" in sys.modules:
         API = initial_api if initial_api in PYQT6_API else "pyqt6"
     elif "PySide6" in sys.modules:
         API = initial_api if initial_api in PYSIDE6_API else "pyside6"
+    elif "PyQt5" in sys.modules:
+        API = initial_api if initial_api in PYQT5_API else "pyqt5"
+    elif "PySide2" in sys.modules:
+        API = initial_api if initial_api in PYSIDE2_API else "pyside2"
 
 if API in PYQT5_API:
     try:
@@ -206,6 +208,7 @@ if API in PYQT5_API:
         )
 
         QT5 = PYQT5 = True
+        QT6 = PYQT6 = False
 
         if sys.platform == "darwin":
             macos_version = parse(platform.mac_ver()[0])
@@ -239,6 +242,7 @@ if API in PYSIDE2_API:
 
         PYQT5 = False
         QT5 = PYSIDE2 = True
+        QT6 = PYQT6 = False
 
         if sys.platform == "darwin":
             macos_version = parse(platform.mac_ver()[0])
@@ -287,6 +291,7 @@ if API in PYSIDE6_API:
         from PySide6.QtCore import __version__ as QT_VERSION  # analysis:ignore
 
         QT5 = PYQT5 = False
+        PYQT6 = False
         QT6 = PYSIDE6 = True
 
     except ImportError:
--- a/qtpy/tests/test_cli.py
+++ b/qtpy/tests/test_cli.py
@@ -47,37 +47,37 @@ def test_cli_mypy_args():
     if qtpy.PYQT5:
         expected = " ".join(
             [
-                "--always-true=PYQT5",
-                "--always-false=PYSIDE2",
                 "--always-false=PYQT6",
                 "--always-false=PYSIDE6",
+                "--always-true=PYQT5",
+                "--always-false=PYSIDE2",
             ],
         )
     elif qtpy.PYSIDE2:
         expected = " ".join(
             [
-                "--always-false=PYQT5",
-                "--always-true=PYSIDE2",
                 "--always-false=PYQT6",
                 "--always-false=PYSIDE6",
+                "--always-false=PYQT5",
+                "--always-true=PYSIDE2",
             ],
         )
     elif qtpy.PYQT6:
         expected = " ".join(
             [
-                "--always-false=PYQT5",
-                "--always-false=PYSIDE2",
                 "--always-true=PYQT6",
                 "--always-false=PYSIDE6",
+                "--always-false=PYQT5",
+                "--always-false=PYSIDE2",
             ],
         )
     elif qtpy.PYSIDE6:
         expected = " ".join(
             [
-                "--always-false=PYQT5",
-                "--always-false=PYSIDE2",
                 "--always-false=PYQT6",
                 "--always-true=PYSIDE6",
+                "--always-false=PYQT5",
+                "--always-false=PYSIDE2",
             ],
         )
     else:
@@ -98,56 +98,56 @@ def test_cli_pyright_config():
         expected = textwrap.dedent(
             """
             pyrightconfig.json:
-            {"defineConstant": {"PYQT5": true, "PYSIDE2": false, "PYQT6": false, "PYSIDE6": false}}
+            {"defineConstant": {"PYQT6": false, "PYSIDE6": false, "PYQT5": true, "PYSIDE2": false}}
 
             pyproject.toml:
             [tool.pyright.defineConstant]
-            PYQT5 = true
-            PYSIDE2 = false
             PYQT6 = false
             PYSIDE6 = false
+            PYQT5 = true
+            PYSIDE2 = false
         """,
         )
     elif qtpy.PYSIDE2:
         expected = textwrap.dedent(
             """
             pyrightconfig.json:
-            {"defineConstant": {"PYQT5": false, "PYSIDE2": true, "PYQT6": false, "PYSIDE6": false}}
+            {"defineConstant": {"PYQT6": false, "PYSIDE6": false, "PYQT5": false, "PYSIDE2": true}}
 
             pyproject.toml:
             [tool.pyright.defineConstant]
-            PYQT5 = false
-            PYSIDE2 = true
             PYQT6 = false
             PYSIDE6 = false
+            PYQT5 = false
+            PYSIDE2 = true
         """,
         )
     elif qtpy.PYQT6:
         expected = textwrap.dedent(
             """
             pyrightconfig.json:
-            {"defineConstant": {"PYQT5": false, "PYSIDE2": false, "PYQT6": true, "PYSIDE6": false}}
+            {"defineConstant": {"PYQT6": true, "PYSIDE6": false, "PYQT5": false, "PYSIDE2": false}}
 
             pyproject.toml:
             [tool.pyright.defineConstant]
-            PYQT5 = false
-            PYSIDE2 = false
             PYQT6 = true
             PYSIDE6 = false
+            PYQT5 = false
+            PYSIDE2 = false
         """,
         )
     elif qtpy.PYSIDE6:
         expected = textwrap.dedent(
             """
             pyrightconfig.json:
-            {"defineConstant": {"PYQT5": false, "PYSIDE2": false, "PYQT6": false, "PYSIDE6": true}}
+            {"defineConstant": {"PYQT6": false, "PYSIDE6": true, "PYQT5": false, "PYSIDE2": false}}
 
             pyproject.toml:
             [tool.pyright.defineConstant]
-            PYQT5 = false
-            PYSIDE2 = false
             PYQT6 = false
             PYSIDE6 = true
+            PYQT5 = false
+            PYSIDE2 = false
         """,
         )
     else:
