diff -rupN configure.ac.in configure.ac.in
--- a/configure.ac.in	2020-03-24 05:15:11.000000000 +0100
+++ b/configure.ac.in	2020-04-09 14:54:07.189090418 +0200
@@ -26,7 +26,6 @@ AC_CHECK_HEADERS([errno.h fcntl.h stdio.
 AC_CHECK_HEADERS([sys/select.h sys/socket.h sys/ioctl.h sys/time.h])
 AC_CHECK_HEADERS([sys/un.h], [have_sys_un_h=yes], [have_sys_un_h=no])
 AM_CONDITIONAL([HAVE_SYS_UN_H], test "x$have_sys_un_h" = xyes)
-AC_CONFIG_HEADERS([pgadmin/include/libssh2/libssh2_config.h])
 
 # Checks for typedefs, structures, and compiler characteristics.
 AC_HEADER_STDBOOL
diff -rupN pgadmin/include/module.mk pgadmin/include/module.mk
--- a/pgadmin/include/module.mk	2020-03-24 05:15:11.000000000 +0100
+++ b/pgadmin/include/module.mk	2020-04-09 14:54:07.190090419 +0200
@@ -36,5 +36,4 @@ include include/gqb/module.mk
 include include/hotdraw/module.mk
 include include/utils/module.mk
 include include/ogl/module.mk
-include include/libssh2/module.mk
 
diff -rupN pgadmin/Makefile.am pgadmin/Makefile.am
--- a/pgadmin/Makefile.am	2020-03-24 05:15:11.000000000 +0100
+++ b/pgadmin/Makefile.am	2020-04-09 14:54:09.880094260 +0200
@@ -49,7 +49,6 @@ include gqb/module.mk
 include hotdraw/module.mk
 include ui/module.mk
 include utils/module.mk
-include libssh2/module.mk
 
 if SUN_CC
   __CFLAGS=""
@@ -80,10 +79,9 @@ nobase_dist_pkgdata_DATA += \
 	$(TMP_ui)
 endif
 
-if BUILD_SSH_TUNNEL
-AM_CPPFLAGS = -DDATA_DIR=\"$(pkgdatadir)/\" -I$(top_srcdir)/pgadmin/include -I$(top_srcdir)/pgadmin/include/libssh2 $(__CPPFLAGS) $(__EDB_LIBPQ)
-else
 AM_CPPFLAGS = -DDATA_DIR=\"$(pkgdatadir)/\" -I$(top_srcdir)/pgadmin/include $(__CPPFLAGS) $(__EDB_LIBPQ)
+if BUILD_SSH_TUNNEL
+AM_LDFLAGS = -lssh2
 endif
 
 # Automake trys to execute install-exec-hook if it appears anywhere in the file, so we need a dummy
@@ -100,11 +98,7 @@ nobase_dist_data_DATA += \
 	$(TMP_ui)
 endif
 
-if BUILD_SSH_TUNNEL
-AM_CPPFLAGS = -I$(top_srcdir)/pgadmin/include -I$(top_srcdir)/pgadmin/include/libssh2 $(__CPPFLAGS) $(__EDB_LIBPQ)
-else
 AM_CPPFLAGS = -I$(top_srcdir)/pgadmin/include $(__CPPFLAGS) $(__EDB_LIBPQ)
-endif
 
 install-exec-hook:
 	cd $(bindir) ;\
diff -rupN pgadmin/utils/sshTunnel.cpp pgadmin/utils/sshTunnel.cpp
--- a/pgadmin/utils/sshTunnel.cpp	2020-03-24 05:15:11.000000000 +0100
+++ b/pgadmin/utils/sshTunnel.cpp	2020-04-09 14:54:07.191090420 +0200
@@ -15,7 +15,7 @@
 
 #undef ssize_t
 #define ssize_t long
-#include "libssh2/libssh2.h"
+#include <libssh2.h>
 #include "utils/sshTunnel.h"
 #include "frm/frmMain.h"
 
