From 46b9fd687125d963b8e3b675f272c5348bdb28dd Mon Sep 17 00:00:00 2001
From: Nathan Owens <ndowens@artixlinux.org>
Date: Mon, 4 Jan 2021 09:53:39 -0600
Subject: [PATCH] cross

---
 m4/largefile.m4    |  4 ++--
 m4/sockinttypes.m4 | 16 ++++++++--------
 2 files changed, 10 insertions(+), 10 deletions(-)

diff --git m4/largefile.m4 m4/largefile.m4
index 7d98cae..f033367 100644
--- m4/largefile.m4
+++ m4/largefile.m4
@@ -32,7 +32,7 @@ dnl *
 AC_DEFUN([eb_SYS_LARGEFILE], [dnl
 AC_REQUIRE([AC_SYS_LARGEFILE])
 AC_CACHE_CHECK([for ll modifier of printf], ac_cv_func_printf_ll,
-[AC_RUN_IFELSE([
+[AC_COMPILE_IFELSE([
     #include <stdio.h>
     #include <string.h>
     #include <limits.h>
@@ -43,7 +43,7 @@ AC_CACHE_CHECK([for ll modifier of printf], ac_cv_func_printf_ll,
     }
 ], [ac_cv_func_printf_ll=yes], [ac_cv_func_printf_ll=no])])
 AC_CACHE_CHECK([for I64 modifier of printf], ac_cv_func_printf_i64,
-[AC_RUN_IFELSE([
+[AC_COMPILE_IFELSE([
     #include <stdio.h>
     #include <string.h>
     #include <limits.h>
diff --git m4/sockinttypes.m4 m4/sockinttypes.m4
index a003df4..85f6e7d 100644
--- m4/sockinttypes.m4
+++ m4/sockinttypes.m4
@@ -54,7 +54,7 @@ in_port_t in_port;
 ], [ac_cv_type_in_port_t=yes], [ac_cv_type_in_port_t=no])])
 if test "$ac_cv_type_in_port_t" != yes; then
     ac_cv_sin_port_size=unknown
-    AC_RUN_IFELSE([
+    AC_COMPILE_IFELSE([
     #include <sys/types.h>
     #include <sys/socket.h>
     #include <netinet/in.h>
@@ -63,7 +63,7 @@ if test "$ac_cv_type_in_port_t" != yes; then
 	return (sizeof(addr.sin_port) == sizeof(long)) ? 0 : 1;
     }
     ], [ac_cv_sin_port_size=long])
-    AC_RUN_IFELSE([
+    AC_COMPILE_IFELSE([
     #include <sys/types.h>
     #include <sys/socket.h>
     #include <netinet/in.h>
@@ -72,7 +72,7 @@ if test "$ac_cv_type_in_port_t" != yes; then
 	return (sizeof(addr.sin_port) == sizeof(int)) ? 0 : 1;
     }
     ], [ac_cv_sin_port_size=int])
-    AC_RUN_IFELSE([
+    AC_COMPILE_IFELSE([
     #include <sys/types.h>
     #include <sys/socket.h>
     #include <netinet/in.h>
@@ -81,7 +81,7 @@ if test "$ac_cv_type_in_port_t" != yes; then
 	return (sizeof(addr.sin_port) == sizeof(short)) ? 0 : 1;
     }
     ], [ac_cv_sin_port_size=short])
-    AC_RUN_IFELSE([
+    AC_COMPILE_IFELSE([
     #include <sys/types.h>
     #include <sys/socket.h>
     #include <netinet/in.h>
@@ -112,7 +112,7 @@ sa_family_t sa_family;
 ], [ac_cv_type_sa_family_t=yes], [ac_cv_type_sa_family_t=no])])
 if test "$ac_cv_type_sa_family_t" != yes; then
     ac_cv_sa_family_size=unknown
-    AC_RUN_IFELSE([
+    AC_COMPILE_IFELSE([
     #include <sys/types.h>
     #include <sys/socket.h>
     int main() {
@@ -120,7 +120,7 @@ if test "$ac_cv_type_sa_family_t" != yes; then
 	return (sizeof(addr.sa_family) == sizeof(long)) ? 0 : 1;
     }
     ], [ac_cv_sa_family_size=long])
-    AC_RUN_IFELSE([
+    AC_COMPILE_IFELSE([
     #include <sys/types.h>
     #include <sys/socket.h>
     int main() {
@@ -128,7 +128,7 @@ if test "$ac_cv_type_sa_family_t" != yes; then
 	return (sizeof(addr.sa_family) == sizeof(int)) ? 0 : 1;
     }
     ], [ac_cv_sa_family_size=int])
-    AC_RUN_IFELSE([
+    AC_COMPILE_IFELSE([
     #include <sys/types.h>
     #include <sys/socket.h>
     int main() {
@@ -136,7 +136,7 @@ if test "$ac_cv_type_sa_family_t" != yes; then
 	return (sizeof(addr.sa_family) == sizeof(short)) ? 0 : 1;
     }
     ], [ac_cv_sa_family_size=short])
-    AC_RUN_IFELSE([
+    AC_COMPILE_IFELSE([
     #include <sys/types.h>
     #include <sys/socket.h>
     int main() {
-- 
2.30.0

