--- a/configure.old	2017-07-01 12:43:00.384785855 +0200
+++ b/configure	2017-07-01 12:55:03.102813398 +0200
@@ -6752,7 +6752,7 @@
 
 if test $ac_cv_header_stdc = yes; then
   # /bin/cc in Irix-4.0.5 gets non-ANSI ctype macros unless using -ansi.
-  if test "$cross_compiling" = yes; then
+  if test "$cross_compiling" = yes; then :
   :
 else
   cat >conftest.$ac_ext <<_ACEOF
@@ -11607,14 +11607,7 @@
 if test "${ac_cv_func_printf_ll+set}" = set; then
   $as_echo_n "(cached) " >&6
 else
-  if test "$cross_compiling" = yes; then
-  { { $as_echo "$as_me:$LINENO: error: in \`$ac_pwd':" >&5
-$as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
-{ { $as_echo "$as_me:$LINENO: error: cannot run test program while cross compiling
-See \`config.log' for more details." >&5
-$as_echo "$as_me: error: cannot run test program while cross compiling
-See \`config.log' for more details." >&2;}
-   { (exit 1); exit 1; }; }; }
+  if test "$cross_compiling" = yes; then :
 else
   cat >conftest.$ac_ext <<_ACEOF
 
@@ -11672,14 +11665,7 @@
 if test "${ac_cv_func_printf_i64+set}" = set; then
   $as_echo_n "(cached) " >&6
 else
-  if test "$cross_compiling" = yes; then
-  { { $as_echo "$as_me:$LINENO: error: in \`$ac_pwd':" >&5
-$as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
-{ { $as_echo "$as_me:$LINENO: error: cannot run test program while cross compiling
-See \`config.log' for more details." >&5
-$as_echo "$as_me: error: cannot run test program while cross compiling
-See \`config.log' for more details." >&2;}
-   { (exit 1); exit 1; }; }; }
+  if test "$cross_compiling" = yes; then :
 else
   cat >conftest.$ac_ext <<_ACEOF
 
@@ -14066,14 +14052,7 @@
 $as_echo "$ac_cv_type_in_port_t" >&6; }
 if test "$ac_cv_type_in_port_t" != yes; then
     ac_cv_sin_port_size=unknown
-    if test "$cross_compiling" = yes; then
-  { { $as_echo "$as_me:$LINENO: error: in \`$ac_pwd':" >&5
-$as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
-{ { $as_echo "$as_me:$LINENO: error: cannot run test program while cross compiling
-See \`config.log' for more details." >&5
-$as_echo "$as_me: error: cannot run test program while cross compiling
-See \`config.log' for more details." >&2;}
-   { (exit 1); exit 1; }; }; }
+    if test "$cross_compiling" = yes; then :
 else
   cat >conftest.$ac_ext <<_ACEOF
 
@@ -14120,14 +14099,7 @@
 fi
 
 
-    if test "$cross_compiling" = yes; then
-  { { $as_echo "$as_me:$LINENO: error: in \`$ac_pwd':" >&5
-$as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
-{ { $as_echo "$as_me:$LINENO: error: cannot run test program while cross compiling
-See \`config.log' for more details." >&5
-$as_echo "$as_me: error: cannot run test program while cross compiling
-See \`config.log' for more details." >&2;}
-   { (exit 1); exit 1; }; }; }
+    if test "$cross_compiling" = yes; then :
 else
   cat >conftest.$ac_ext <<_ACEOF
 
@@ -14174,14 +14146,7 @@
 fi
 
 
-    if test "$cross_compiling" = yes; then
-  { { $as_echo "$as_me:$LINENO: error: in \`$ac_pwd':" >&5
-$as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
-{ { $as_echo "$as_me:$LINENO: error: cannot run test program while cross compiling
-See \`config.log' for more details." >&5
-$as_echo "$as_me: error: cannot run test program while cross compiling
-See \`config.log' for more details." >&2;}
-   { (exit 1); exit 1; }; }; }
+    if test "$cross_compiling" = yes; then :
 else
   cat >conftest.$ac_ext <<_ACEOF
 
@@ -14228,14 +14193,7 @@
 fi
 
 
-    if test "$cross_compiling" = yes; then
-  { { $as_echo "$as_me:$LINENO: error: in \`$ac_pwd':" >&5
-$as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
-{ { $as_echo "$as_me:$LINENO: error: cannot run test program while cross compiling
-See \`config.log' for more details." >&5
-$as_echo "$as_me: error: cannot run test program while cross compiling
-See \`config.log' for more details." >&2;}
-   { (exit 1); exit 1; }; }; }
+    if test "$cross_compiling" = yes; then :
 else
   cat >conftest.$ac_ext <<_ACEOF
 
@@ -14337,14 +14295,7 @@
 $as_echo "$ac_cv_type_sa_family_t" >&6; }
 if test "$ac_cv_type_sa_family_t" != yes; then
     ac_cv_sa_family_size=unknown
-    if test "$cross_compiling" = yes; then
-  { { $as_echo "$as_me:$LINENO: error: in \`$ac_pwd':" >&5
-$as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
-{ { $as_echo "$as_me:$LINENO: error: cannot run test program while cross compiling
-See \`config.log' for more details." >&5
-$as_echo "$as_me: error: cannot run test program while cross compiling
-See \`config.log' for more details." >&2;}
-   { (exit 1); exit 1; }; }; }
+    if test "$cross_compiling" = yes; then :
 else
   cat >conftest.$ac_ext <<_ACEOF
 
@@ -14390,14 +14341,7 @@
 fi
 
 
-    if test "$cross_compiling" = yes; then
-  { { $as_echo "$as_me:$LINENO: error: in \`$ac_pwd':" >&5
-$as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
-{ { $as_echo "$as_me:$LINENO: error: cannot run test program while cross compiling
-See \`config.log' for more details." >&5
-$as_echo "$as_me: error: cannot run test program while cross compiling
-See \`config.log' for more details." >&2;}
-   { (exit 1); exit 1; }; }; }
+    if test "$cross_compiling" = yes; then :
 else
   cat >conftest.$ac_ext <<_ACEOF
 
@@ -14443,14 +14387,7 @@
 fi
 
 
-    if test "$cross_compiling" = yes; then
-  { { $as_echo "$as_me:$LINENO: error: in \`$ac_pwd':" >&5
-$as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
-{ { $as_echo "$as_me:$LINENO: error: cannot run test program while cross compiling
-See \`config.log' for more details." >&5
-$as_echo "$as_me: error: cannot run test program while cross compiling
-See \`config.log' for more details." >&2;}
-   { (exit 1); exit 1; }; }; }
+    if test "$cross_compiling" = yes; then :
 else
   cat >conftest.$ac_ext <<_ACEOF
 
@@ -14496,14 +14433,7 @@
 fi
 
 
-    if test "$cross_compiling" = yes; then
-  { { $as_echo "$as_me:$LINENO: error: in \`$ac_pwd':" >&5
-$as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
-{ { $as_echo "$as_me:$LINENO: error: cannot run test program while cross compiling
-See \`config.log' for more details." >&5
-$as_echo "$as_me: error: cannot run test program while cross compiling
-See \`config.log' for more details." >&2;}
-   { (exit 1); exit 1; }; }; }
+    if test "$cross_compiling" = yes; then :
 else
   cat >conftest.$ac_ext <<_ACEOF
 
