From d28e105075e3ec46515c322ee5dca6967a4cc696 Mon Sep 17 00:00:00 2001
From: Daniel Diaz<dydyam at gmail dot com>
Date: Mon, 28  Mar  2011  07:06:01  +0000
Subject: compile fix: added <cstdlib> to provide declaration for 'size_t' (gcc-4.6.x likes it that way)

---
diff --git a/src/FbTk/LayerItem.hh b/src/FbTk/LayerItem.hh
index a4cdd49..ad33451 100644
--- a/src/FbTk/LayerItem.hh
+++ b/src/FbTk/LayerItem.hh
@@ -26,6 +26,7 @@
 #include "Layer.hh"
 #include "NotCopyable.hh"
 #include <vector>
+#include <cstdlib> // size_t
 
 namespace FbTk {
 
diff --git a/src/FbTk/MacroCommand.hh b/src/FbTk/MacroCommand.hh
index 932ecaf..bce2ce4 100644
--- a/src/FbTk/MacroCommand.hh
+++ b/src/FbTk/MacroCommand.hh
@@ -26,6 +26,7 @@
 #include "RefCount.hh"
 
 #include <vector>
+#include <cstdlib> // size_t
 
 namespace FbTk {
 
diff --git a/src/FbTk/MultLayers.hh b/src/FbTk/MultLayers.hh
index 62c006f..3da25f6 100644
--- a/src/FbTk/MultLayers.hh
+++ b/src/FbTk/MultLayers.hh
@@ -24,6 +24,7 @@
 #define FBTK_MULTLAYERS_HH
 
 #include <vector>
+#include <cstdlib> // size_t
 
 namespace FbTk {
 
--
cgit 
