--- Makefile.orig	2016-05-25 16:57:27.966465927 +0200
+++ Makefile	2016-05-25 16:57:32.091465618 +0200
@@ -39,8 +39,8 @@ else
   test_tmp := $(CURDIR)/t/tmp
 endif
 
-initial_setup := $(shell ./configure-version --update $(isok))
-initial_setup := $(call shout,$(initial_setup),Version configuration failed))
+#initial_setup := $(shell ./configure-version --update $(isok))
+#initial_setup := $(call shout,$(initial_setup),Version configuration failed))
 
 config/config.vars: configure config/configure config/configure.inc \
   $(wildcard config/*.in)
