--- grpc-1.39.1.orig/CMakeLists.txt
+++ grpc-1.39.1/CMakeLists.txt
@@ -15856,7 +15865,7 @@ generate_pkgconfig(
   "gpr"
   "gRPC platform support library"
   "${gRPC_CORE_VERSION}"
-  "absl_base absl_memory absl_optional absl_status absl_str_format absl_strings absl_synchronization absl_time"
+  ""
   "-lgpr"
   ""
   "gpr.pc")
@@ -15866,7 +15875,7 @@ generate_pkgconfig(
   "gRPC"
   "high performance general RPC framework"
   "${gRPC_CORE_VERSION}"
-  "gpr openssl absl_base absl_bind_front absl_flat_hash_map absl_inlined_vector absl_memory absl_optional absl_status absl_statusor absl_str_format absl_strings absl_synchronization absl_time"
+  "gpr openssl"
   "-lgrpc -laddress_sorting -lre2 -lupb -lcares -lz"
   ""
   "grpc.pc")
@@ -15876,7 +15885,7 @@ generate_pkgconfig(
   "gRPC unsecure"
   "high performance general RPC framework without SSL"
   "${gRPC_CORE_VERSION}"
-  "gpr absl_base absl_bind_front absl_flat_hash_map absl_inlined_vector absl_memory absl_optional absl_status absl_statusor absl_str_format absl_strings absl_synchronization absl_time"
+  "gpr"
   "-lgrpc_unsecure"
   ""
   "grpc_unsecure.pc")
@@ -15886,7 +15895,7 @@ generate_pkgconfig(
   "gRPC++"
   "C++ wrapper for gRPC"
   "${gRPC_CPP_VERSION}"
-  "grpc absl_base absl_bind_front absl_flat_hash_map absl_inlined_vector absl_memory absl_optional absl_status absl_statusor absl_str_format absl_strings absl_synchronization absl_time"
+  "grpc"
   "-lgrpc++"
   ""
   "grpc++.pc")
@@ -15896,7 +15905,7 @@ generate_pkgconfig(
   "gRPC++ unsecure"
   "C++ wrapper for gRPC without SSL"
   "${gRPC_CPP_VERSION}"
-  "grpc_unsecure absl_base absl_bind_front absl_flat_hash_map absl_inlined_vector absl_memory absl_optional absl_status absl_statusor absl_str_format absl_strings absl_synchronization absl_time"
+  "grpc_unsecure"
   "-lgrpc++_unsecure"
   ""
   "grpc++_unsecure.pc")
--- grpc-1.39.1.orig/cmake/pkg-config-template.pc.in
+++ grpc-1.39.1/cmake/pkg-config-template.pc.in
@@ -6,7 +6,7 @@ libdir=${exec_prefix}/lib
 Name: @PC_NAME@
 Description: @PC_DESCRIPTION@
 Version: @PC_VERSION@
-Cflags: -I${includedir}
+Cflags: -I${includedir} -I${includedir}/grpc_absl
 Requires: @PC_REQUIRES@
 Libs: -L${libdir} @PC_LIB@
 Libs.private: @PC_LIBS_PRIVATE@
