--- a/cmake/gRPCConfig.cmake.in	2023-06-14 22:21:16.000000000 +0200
+++ -	2023-06-28 11:11:32.902678412 +0200
@@ -1,5 +1,6 @@
 # Module path
 list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_LIST_DIR}/modules)
+include(CMakeFindDependencyMacro)
 
 # Depend packages
 @_gRPC_FIND_ZLIB@
--- a/CMakeLists.txt	2023-07-11 08:54:22.009027477 +0200
+++ -	2023-07-11 08:54:38.950947984 +0200
@@ -12375,6 +12375,8 @@
   ${_gRPC_ZLIB_LIBRARIES}
   ${_gRPC_ALLTARGETS_LIBRARIES}
   grpc_plugin_support
+  absl::hash
+  absl::raw_hash_set
 )
 
 
