--- xmloff/source/draw/ximp3dscene.cxx	2016-01-27 01:58:25.000000000 +0100
+++ xmloff/source/draw/ximp3dscene.cxx	2016-02-23 19:01:35.785368122 +0100
@@ -66,7 +66,7 @@
             {
                 ::basegfx::B3DVector aVal;
                 SvXMLUnitConverter::convertB3DVector(aVal, sValue);
-                if (!isnan(aVal.getX()) && !isnan(aVal.getY()) && !isnan(aVal.getZ()))
+                if (!std::isnan(aVal.getX()) && !std::isnan(aVal.getY()) && !std::isnan(aVal.getZ()))
                 {
                     maDirection = aVal;
                 }
