--- sal/osl/unx/signal.cxx	2016-01-27 01:58:25.000000000 +0100
+++ sal/osl/unx/signal.cxx	2016-02-23 19:33:44.618507179 +0100
@@ -33,7 +33,7 @@
 
 #endif /* MACOSX */
 
-#ifdef LINUX
+#if defined(LINUX) && defined(__GLIBC__)
 #include <execinfo.h>
 #include <link.h>
 #define INCLUDE_BACKTRACE
--- sal/osl/unx/backtrace.h	2017-07-25 22:26:39.000000000 +0200
+++ sal/osl/unx/backtrace.h	2017-08-04 17:53:35.182066906 +0200
@@ -20,7 +20,7 @@
 #ifndef INCLUDED_SAL_OSL_UNX_BACKTRACE_H
 #define INCLUDED_SAL_OSL_UNX_BACKTRACE_H
 
-#if defined (LINUX)
+#if defined (LINUX) && defined (__GLIBC__)
 
 #include <execinfo.h>
 
--- sal/osl/unx/backtrace.c	2017-07-25 22:26:39.000000000 +0200
+++ sal/osl/unx/backtrace.c	2017-08-04 17:54:56.831219663 +0200
@@ -282,7 +282,7 @@
     }
 }
 
-#elif !defined LINUX
+#elif !defined LINUX || !defined __GLIBC__
 
 int backtrace( void **buffer, int max_frames )
 {
