--- /dev/null	2016-02-21 14:44:37.295798706 +0100
+++ external/firebird/firebird-glibc.patch.0	2016-02-24 15:12:43.055606993 +0100
@@ -0,0 +1,11 @@
+--- src/common/classes/rwlock.h	2015-03-27 14:47:01.000000000 +0100
++++ src/common/classes/rwlock.h	2016-02-24 15:11:46.043602883 +0100
+@@ -191,7 +191,7 @@
+ 
+ 	void init()
+ 	{
+-#if defined(LINUX) && !defined(USE_VALGRIND)
++#if defined(LINUX) && defined(__GLIBC__) && !defined(USE_VALGRIND)
+ 		pthread_rwlockattr_t attr;
+ 		if (pthread_rwlockattr_init(&attr))
+ 			system_call_failed::raise("pthread_rwlockattr_init");
