--- /dev/null	2016-02-21 14:44:37.295798706 +0100
+++ external/firebird/firebird-libio.patch.0	2016-02-24 15:39:05.303721064 +0100
@@ -0,0 +1,11 @@
+--- src/jrd/perf.h	2015-03-27 14:46:58.000000000 +0100
++++ src/jrd/perf.h	2016-02-24 15:38:06.674716837 +0100
+@@ -31,7 +31,7 @@
+ #ifndef JRD_PERF_H
+ #define JRD_PERF_H
+ 
+-#ifdef LINUX
++#if defined(LINUX) && defined(__GLIBC__)
+ #include <libio.h>
+ #endif
+ 
