--- external/firebird/UnpackedTarball_firebird.mk	2016-01-27 01:58:25.000000000 +0100
+++ external/firebird/UnpackedTarball_firebird.mk	2016-02-24 15:13:34.733610719 +0100
@@ -19,6 +19,8 @@
 	external/firebird/firebird-c++11replfn.patch.0 \
         external/firebird/firebird.disable-ib-util-not-found.patch.1 \
 	external/firebird/firebird-c++14.patch \
+	external/firebird/firebird-glibc.patch.0 \
+	external/firebird/firebird-libio.patch.0 \
 ))
 
 ifeq ($(OS)-$(COM),WNT-MSC)
