--- sd/source/ui/remotecontrol/BluetoothServer.cxx	2016-01-27 01:58:25.000000000 +0100
+++ sd/source/ui/remotecontrol/BluetoothServer.cxx	2016-02-25 15:06:53.971708191 +0100
@@ -21,7 +21,7 @@
   #include <dbus/dbus.h>
   #include <errno.h>
   #include <fcntl.h>
-  #include <sys/unistd.h>
+  #include <unistd.h>
   #include <sys/socket.h>
   #include <bluetooth/bluetooth.h>
   #include <bluetooth/rfcomm.h>
