--- a/jvmfwk/plugins/sunmajor/pluginlib/sunjre.cxx
+++ b/jvmfwk/plugins/sunmajor/pluginlib/sunjre.cxx
@@ -80,7 +80,11 @@ char const* const* SunInfo::getLibraryPa
         "/lib/" JFW_PLUGIN_ARCH "/client",
         "/lib/" JFW_PLUGIN_ARCH "/server",
         "/lib/" JFW_PLUGIN_ARCH "/native_threads",
-        ("/lib/" JFW_PLUGIN_ARCH)
+        "/lib/" JFW_PLUGIN_ARCH,
+        "/lib/client",
+        "/lib/server",
+        "/lib/native_threads",
+        "/lib",
     };
     *size = std::size(ar);
     return ar;
