--- a/configure.ac
+++ b/configure.ac
@@ -8998,7 +8998,7 @@ if test -n "$ENABLE_JAVA" -a -z "$JAVAIN
         test -d "$JAVA_HOME/include/native_thread" && JAVAINC="$JAVAINC -I$JAVA_HOME/include/native_thread"
         ;;
 
-    linux-gnu*)
+    linux-gnu*|linux-musl*)
         JAVAINC="-I$JAVA_HOME/include"
         JAVAINC="$JAVAINC -I$JAVA_HOME/include/linux"
         test -d "$JAVA_HOME/include/native_thread" && JAVAINC="$JAVAINC -I$JAVA_HOME/include/native_thread"
@@ -15074,7 +15074,7 @@ else
 
     case "$host_os" in
 
-    dragonfly*|freebsd*|linux-gnu*|*netbsd*|openbsd*)
+    dragonfly*|freebsd*|linux-gnu*|linux-musl*|*netbsd*|openbsd*)
         if test "$ENABLE_JAVA" != ""; then
             pathmunge "$JAVA_HOME/bin" "after"
         fi
--- a/configure
+++ b/configure
@@ -23020,7 +23020,7 @@ if test -n "$ENABLE_JAVA" -a -z "$JAVAIN
         test -d "$JAVA_HOME/include/native_thread" && JAVAINC="$JAVAINC -I$JAVA_HOME/include/native_thread"
         ;;
 
-    linux-gnu*)
+    linux-gnu*|linux-musl*)
         JAVAINC="-I$JAVA_HOME/include"
         JAVAINC="$JAVAINC -I$JAVA_HOME/include/linux"
         test -d "$JAVA_HOME/include/native_thread" && JAVAINC="$JAVAINC -I$JAVA_HOME/include/native_thread"
@@ -48162,7 +48162,7 @@ else
 
     case "$host_os" in
 
-    dragonfly*|freebsd*|linux-gnu*|*netbsd*|openbsd*)
+    dragonfly*|freebsd*|linux-gnu*|linux-musl*|*netbsd*|openbsd*)
         if test "$ENABLE_JAVA" != ""; then
             pathmunge "$JAVA_HOME/bin" "after"
         fi
