--- a/bin/unpack-sources
+++ b/bin/unpack-sources
@@ -78,8 +78,8 @@ for tarball in $tarballs ; do
     if ! mkdir -p "$lo_src_dir/$tarname" ; then
         echo "Error: could not create directory $lo_src_dir/$tarname"
     fi
-    echo tar -xf "$tarball" -C "$lo_src_dir/$tarname" --strip-components=1
-    if ! tar -xf "$tarball" -C "$lo_src_dir/$tarname" --strip-components=1; then
+    echo bsdtar -xf "$tarball" -C "$lo_src_dir/$tarname" --strip-components=1
+    if ! bsdtar -xf "$tarball" -C "$lo_src_dir/$tarname" --strip-components=1; then
         echo "Error: could not unpack $tarname"
         exit 1
     fi
