commit 9996a4028ddc7f9a5ffda3df65d5b7d3b7df8aa5
Author: Gonzalo Tornaría <tornaria@cmat.edu.uy>
Date:   Wed Jul 19 18:34:57 2023 -0300

    cython3 support using legacy directives

diff --git a/setup.py b/setup.py
index 37acdfc..f68270b 100755
--- a/setup.py
+++ b/setup.py
@@ -157,13 +157,17 @@ class build_ext(_build_ext):
         # Run Cython with -Werror on continuous integration services
         # with Python 3.6 or later
         from Cython.Compiler import Options
-        Options.warning_errors = True
+        Options.warning_errors = False
 
         from Cython.Build.Dependencies import cythonize
         return cythonize(extensions,
                 build_dir=cythonize_dir,
                 include_path=["src", os.path.join(cythonize_dir, "src")],
-                compiler_directives=dict(binding=True, language_level=2))
+                compiler_directives=dict(
+                    binding=True,
+                    language_level=2,
+                    legacy_implicit_noexcept=True,
+                    ))
 
 
 class build_py(_build_py):
