--- Makefile
+++ Makefile
@@ -17,8 +17,8 @@ HEADERS = c_test.h charclass.h config.h edit.h expand.h ksh_limval.h lex.h \
           proto.h sh.h shf.h table.h tree.h tty.h
 
 ifeq (1,$(HAVE_LIBBSD))
-	CFLAGS += $(shell pkg-config --cflags libbsd-overlay)
-	LDFLAGS += $(shell pkg-config --libs libbsd-overlay)
+	override CFLAGS += $(shell pkg-config --cflags libbsd-overlay)
+	override LDFLAGS += $(shell pkg-config --libs libbsd-overlay)
 endif
 
 all: ksh
