diff --git a/src/config/WallpaperMatcher.cpp b/src/config/WallpaperMatcher.cpp
index e8ed02f..67f4f0c 100644
--- a/src/config/WallpaperMatcher.cpp
+++ b/src/config/WallpaperMatcher.cpp
@@ -26,7 +26,7 @@ void CWallpaperMatcher::addStates(std::vector<CConfigManager::SSetting>&& s) {
     }
 
     std::erase_if(m_settings, [&s](const auto& e) { return std::ranges::any_of(s, [&e](const auto& el) { return el.monitor == e.monitor; }); });
-    m_settings.append_range(std::move(s));
+    m_settings.insert(m_settings.end(), std::make_move_iterator(s.begin()), std::make_move_iterator(s.end()));
     recalcStates();
 }
