Description: remove compiler warnings that nowadays result in errors
Author: Thorsten Alteholz <debian@alteholz.de>
Index: uucp/uuconv.c
===================================================================
--- uucp.orig/uuconv.c	2013-04-26 16:06:16.000000000 +0200
+++ uucp/uuconv.c	2013-05-08 10:49:06.000000000 +0200
@@ -676,7 +676,7 @@
     {
       if ((*pz)[0] != '-' && pz != pzarg)
 	fprintf (e, " ");
-      fprintf (e, *pz);
+      fprintf (e, "%s", *pz);
     }
 }
 
