Author: Thorsten Alteholz <debian@alteholz.de>
Description: correct spelling error in binary
--- a/cu.c
+++ b/cu.c
@@ -1764,7 +1764,7 @@ icutake (puuconf, argc, argv, pvar, pinf
     zfrom = zbufcpy (argv[1]);
   else
     {
-      zfrom = zsysdep_terminal_line ("Remote file to retreive: ");
+      zfrom = zsysdep_terminal_line ("Remote file to retrieve: ");
       if (zfrom == NULL)
 	ucuabort ();
       zfrom[strcspn (zfrom, " \t\n")] = '\0';
--- a/uuxqt.c
+++ b/uuxqt.c
@@ -1517,7 +1517,7 @@ uqdo_xqt_file (puuconf, zfile, zbase, qs
 	  az[i++] = zQcmd;
 	  az[i++] = "\n";
 
-	  (void) fsysdep_mail (zmail, "Execution succeded", i, az);
+	  (void) fsysdep_mail (zmail, "Execution succeeded", i, az);
 	}
 
       /* Now we may have to uucp the output to some other machine.  */
