diff --git meson.build meson.build
index 56c3f6a..89c393b 100644
--- meson.build
+++ meson.build
@@ -64,8 +64,8 @@ udev           = dependency('libudev')
 pixman         = dependency('pixman-1')
 libcap         = dependency('libcap', required: get_option('libcap'))
 logind         = dependency('lib' + get_option('logind-provider'), required: get_option('logind'), version: '>=237')
-math           = cc.find_library('m')
-rt             = cc.find_library('rt')
+math           = cc.find_library('m', required: false)
+rt             = cc.find_library('rt', required: false)
 
 wlr_parts = []
 wlr_deps = []

