diff --git lualdap.c lualdap.c
index a3f42a9..026e369 100644
--- lualdap.c
+++ lualdap.c
@@ -876,7 +876,7 @@ static int lualdap_search_tostring(lua_State *L)
 */
 static int lualdap_createmeta(lua_State *L)
 {
-	const luaL_reg methods[] = {
+	const luaL_Reg methods[] = {
 		{"close", lualdap_close},
 		{"add", lualdap_add},
 		{"compare", lualdap_compare},
@@ -1043,7 +1043,7 @@ static void set_info(lua_State *L)
 */
 int luaopen_lualdap(lua_State *L)
 {
-	struct luaL_reg lualdap[] = {
+	struct luaL_Reg lualdap[] = {
 		{"open_simple", lualdap_open_simple},
 		{NULL, NULL},
 	};
