--- js/src/shell/jsoptparse.cpp	2013-10-29 21:40:20.000000000 +0100
+++ js/src/shell/jsoptparse.cpp	2017-06-14 13:00:24.076799935 +0200
@@ -253,7 +253,7 @@
     char *eq = strchr(argv[*i], '=');
     if (eq) {
         *value = eq + 1;
-        if (value[0] == '\0')
+        if (*value[0] == '\0')
             return error("A value is required for option %.*s", eq - argv[*i], argv[*i]);
         return Okay;
     }
