diff -ru enchant-1.6.0-orig/src/myspell/myspell_checker.cpp enchant-1.6.0/src/myspell/myspell_checker.cpp
--- enchant-1.6.0-orig/src/myspell/myspell_checker.cpp	2016-04-18 12:25:00.094614256 +0100
+++ enchant-1.6.0/src/myspell/myspell_checker.cpp	2016-04-18 12:26:09.108569576 +0100
@@ -148,6 +148,10 @@
 		g_iconv_close(m_translate_out);
 }
 
+#ifndef MAXWORDLEN
+#   define MAXWORDLEN 100
+#endif
+
 bool
 MySpellChecker::checkWord(const char *utf8Word, size_t len)
 {
