commit e5a77593ef6a32cdb0551e7588180bbc2b9dd392
Author: Garri Djavadyan <g.djavadyan@gmail.com>
Date:   Wed Mar 21 16:19:17 2018 +0500

    Respect CFLAGS/LDFLAGS

diff --git Makefile Makefile
index c55c29f..bd946a7 100644
--- Makefile
+++ Makefile
@@ -40,7 +40,7 @@ OBJ     =   $(SRC:.c=.o)
 
 CC      ?=   gcc
 
-CFLAGS  =   -O3 -Wall
+CFLAGS  +=   -Wall
 
 BINDIR  =   /usr/local/bin
 
@@ -76,7 +76,7 @@ endif
 ##############################
 
 $(NAME)	: $(OBJ)
-	$(CC) -o $(NAME) $(OBJ) $(LIBS)
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $(NAME) $(OBJ) $(LIBS)
 
 .PHONY: clean
 
