--- a/dnsrecon/lib/whois.py
+++ b/dnsrecon/lib/whois.py
@@ -13,11 +13,10 @@
 #    along with this program; if not, write to the Free Software
 #    Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 
+import ipaddress
 import re
 import socket
 
-from netaddr import *
-
 __name__ = 'whois.py'
 
 WHOIS_PORT_NUMBER = 43
@@ -27,19 +26,16 @@
 def get_whois(ip_addrs):
     """
     Function that returns what whois server is the one to be queried for
-    registration information, returns whois.arin.net is not in database, returns
+    registration information, returns whois.arin.net if not in database, returns
     None if private.
     """
     whois_server = None
-    ip = IPAddress(ip_addrs)
-    info_of_ip = ip.info
-    if ip.version == 4 and ip.is_private() is False:
-        for i in info_of_ip['IPv4']:
-            whois_server = i['whois']
-            if len(whois_server) == 0 and i['status'] != 'Reserved':
-                whois_server = 'whois.arin.net'
-            elif len(whois_server) == 0:
-                whois_server = None
+    try:
+        ip = ipaddress.ip_address(ip_addrs)
+        if isinstance(ip, ipaddress.IPv4Address) and not ip.is_private:
+            whois_server = 'whois.arin.net'
+    except ValueError:
+        whois_server = None
 
     return whois_server
 
@@ -61,9 +57,11 @@
         response = ''
         while True:
             d = s.recv(WHOIS_RECEIVE_BUFFER_SIZE)
-            response += str(d)
+            if not d:
+                break
+            response += d.decode('utf-8')
             counter += 1
-            if str(d) == '' or counter == 5:
+            if counter == 5:
                 break
         s.close()
     except Exception as e:
