--- a/config.h.in
+++ b/config.h.in
@@ -1,5 +1,15 @@
 /* config.h.in.  Generated from configure.ac by autoheader.  */
 
+/* taken from glibc unistd.h and fixes musl */
+#ifndef TEMP_FAILURE_RETRY
+#define TEMP_FAILURE_RETRY(expression) \
+  (__extension__                                                              \
+    ({ long int __result;                                                     \
+       do __result = (long int) (expression);                                 \
+       while (__result == -1L && errno == EINTR);                             \
+       __result; }))
+#endif
+
 /* Define if we are building with asan and ubsan */
 #undef BUILDOPT_ASAN
 
