From daef485dffd67319c114cf1bda453bf6157e11ff Mon Sep 17 00:00:00 2001
From: Paolo Bacchilega <paobac@src.gnome.org>
Date: Tue, 11 Jan 2022 12:30:44 +0100
Subject: [PATCH] LINGUAS: removed duplicated entry

---
 help/LINGUAS | 1 -
 1 file changed, 1 deletion(-)

diff --git a/help/LINGUAS b/help/LINGUAS
index b615195e..aabf71ac 100644
--- a/help/LINGUAS
+++ b/help/LINGUAS
@@ -9,7 +9,6 @@ eu
 fr
 hu
 id
-id
 nl
 oc
 sl
-- 
GitLab

