Index: kmymoney-5.1.2/kmymoney/mymoney/mymoneycontact.cpp
===================================================================
--- kmymoney-5.1.2.orig/kmymoney/mymoney/mymoneycontact.cpp
+++ kmymoney-5.1.2/kmymoney/mymoney/mymoneycontact.cpp
@@ -10,9 +10,9 @@
 #ifdef ENABLE_ADDRESSBOOK
 #include <KIdentityManagement/IdentityManager>
 #include <KIdentityManagement/Identity>
-#include <AkonadiCore/RecursiveItemFetchJob>
-#include <AkonadiCore/ItemFetchScope>
-#include <AkonadiCore/Collection>
+#include <AkonadiCore/Akonadi/RecursiveItemFetchJob>
+#include <AkonadiCore/Akonadi/ItemFetchScope>
+#include <AkonadiCore/Akonadi/Collection>
 #include <KContacts/Addressee>
 #include <QRegularExpression>
 #endif
