diff -up chromium-117.0.5938.62/net/dns/host_resolver_cache.cc.me chromium-117.0.5938.62/net/dns/host_resolver_cache.cc
diff -up chromium-117.0.5938.62/net/dns/host_resolver_cache.h.me chromium-117.0.5938.62/net/dns/host_resolver_cache.h
--- chromium-117.0.5938.62/net/dns/host_resolver_cache.h.me	2023-09-14 15:21:24.632965004 +0200
+++ chromium-117.0.5938.62/net/dns/host_resolver_cache.h	2023-09-15 09:15:48.511300845 +0200
@@ -143,12 +143,14 @@ class NET_EXPORT HostResolverCache final
     }
 
     bool operator()(const Key& lhs, const KeyRef& rhs) const {
+      const std::string rhs_domain_name{rhs.domain_name};
       return std::tie(lhs.domain_name, lhs.network_anonymization_key) <
-             std::tie(rhs.domain_name, *rhs.network_anonymization_key);
+             std::tie(rhs_domain_name, *rhs.network_anonymization_key);
     }
 
     bool operator()(const KeyRef& lhs, const Key& rhs) const {
-      return std::tie(lhs.domain_name, *lhs.network_anonymization_key) <
+      const std::string lhs_domain_name{lhs.domain_name};
+      return std::tie(lhs_domain_name, *lhs.network_anonymization_key) <
              std::tie(rhs.domain_name, rhs.network_anonymization_key);
     }
   };
