diff --git CMakeLists.txt CMakeLists.txt
index 523093d..357ad91 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -95,15 +95,12 @@ FindRequiredPackage(OpenSSL)
 FindRequiredPackage(BIND)
 FindRequiredPackage(ZLIB)
 
-if (NOT BINPAC_EXE_PATH AND
-    EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/aux/binpac/CMakeLists.txt)
+if (EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/aux/binpac/CMakeLists.txt)
     add_subdirectory(aux/binpac)
 endif ()
 FindRequiredPackage(BinPAC)
 
-if ( NOT BIFCL_EXE_PATH )
-  add_subdirectory(aux/bifcl)
-endif ()
+add_subdirectory(aux/bifcl)
 
 if (ENABLE_JEMALLOC)
     find_package(JeMalloc)
