diff -ur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2020-06-21 08:03:54.000000000 -0400
+++ b/CMakeLists.txt	2025-11-17 15:28:20.983116846 -0500
@@ -1,7 +1,7 @@
 # Keystone Assembler Engine (www.keystone-engine.org)
 # By Nguyen Anh Quynh, 2016
 
-cmake_minimum_required(VERSION 2.8.7)
+cmake_minimum_required(VERSION 3.5)
 project(keystone)
 
 set(KEYSTONE_VERSION_MAJOR 0)
@@ -18,15 +18,6 @@
   cmake_policy(SET CMP0022 NEW) # automatic when 2.8.12 is required
 endif()
 
-if (POLICY CMP0051)
-  # CMake 3.1 and higher include generator expressions of the form
-  # $<TARGETLIB:obj> in the SOURCES property.  These need to be
-  # stripped everywhere that access the SOURCES property, so we just
-  # defer to the OLD behavior of not including generator expressions
-  # in the output for now.
-  cmake_policy(SET CMP0051 OLD)
-endif()
-
 if (POLICY CMP0063)
   set(CMAKE_POLICY_DEFAULT_CMP0063 NEW) # automatic when 3.3.2 is required
 endif()
@@ -115,4 +106,4 @@
 add_custom_target(uninstall
     COMMAND ${CMAKE_COMMAND} -P ${CMAKE_CURRENT_BINARY_DIR}/CMakeUninstall.cmake)
 
-add_subdirectory(suite/fuzz)
\ No newline at end of file
+add_subdirectory(suite/fuzz)
diff -ur a/kstool/CMakeLists.txt b/kstool/CMakeLists.txt
--- a/kstool/CMakeLists.txt	2020-06-21 08:03:54.000000000 -0400
+++ b/kstool/CMakeLists.txt	2025-11-17 15:29:10.257260358 -0500
@@ -1,7 +1,7 @@
 # Kstool for Keystone assembler engine.
 # By Nguyen Anh Quynh, 2016
 
-cmake_minimum_required(VERSION 2.8)
+cmake_minimum_required(VERSION 3.5)
 
 project(kstool)
 
diff -ur a/llvm/CMakeLists.txt b/llvm/CMakeLists.txt
--- a/llvm/CMakeLists.txt	2020-06-21 08:03:54.000000000 -0400
+++ b/llvm/CMakeLists.txt	2025-11-17 15:28:26.375132550 -0500
@@ -1,6 +1,6 @@
 # See docs/CMake.html for instructions about how to build LLVM with CMake.
 
-cmake_minimum_required(VERSION 2.8.7)
+cmake_minimum_required(VERSION 3.5)
 
 set(LLVM_INSTALL_TOOLCHAIN_ONLY ON)
 
@@ -13,15 +13,6 @@
   cmake_policy(SET CMP0022 NEW) # automatic when 2.8.12 is required
 endif()
 
-if (POLICY CMP0051)
-  # CMake 3.1 and higher include generator expressions of the form
-  # $<TARGETLIB:obj> in the SOURCES property.  These need to be
-  # stripped everywhere that access the SOURCES property, so we just
-  # defer to the OLD behavior of not including generator expressions
-  # in the output for now.
-  cmake_policy(SET CMP0051 OLD)
-endif()
-
 if(CMAKE_VERSION VERSION_LESS 3.1.20141117)
   set(cmake_3_2_USES_TERMINAL)
 else()
diff -ur a/samples/CMakeLists.txt b/samples/CMakeLists.txt
--- a/samples/CMakeLists.txt	2020-06-21 08:03:54.000000000 -0400
+++ b/samples/CMakeLists.txt	2025-11-17 15:29:18.207283512 -0500
@@ -5,7 +5,7 @@
 # For Windows, be sure to copy keystone.dll or keystone.lib
 # to this directory before compiling
 
-cmake_minimum_required(VERSION 2.8)
+cmake_minimum_required(VERSION 3.5)
 
 project(sample)
 
