diff --git Makefile Makefile
index 218aa2d..77ec75e 100644
--- a/Makefile
+++ b/Makefile
@@ -47,4 +47,4 @@ CFLAGS = $(CXXFLAGS) $(CPPFLAGS) -c                \
    -I/usr/include/libchamplain-0.12/

-LIBS = `$(PKG_CONFIG) --libs gtk+-3.0`                                              \
+LIBS = `$(PKG_CONFIG) --libs gtk+-3.0` $(shell pkg-config --libs libexecinfo)    \
          -lclutter-1.0 -lclutter-gtk-1.0 -lchamplain-0.12 -lchamplain-gtk-0.12

 ALLFILES = fotoxx.o f.widgets.o f.file.o f.gallery.o f.albums.o f.area.o f.meta.o   \
@@ -146,4 +146,3 @@ uninstall:
 clean:
 	rm -f  fotoxx
 	rm -f  *.o
-
