diff --git f.file.cc f.file.cc
index 7c4f083..f2c9c07 100644
--- a/f.file.cc
+++ b/f.file.cc
@@ -3493,6 +3493,7 @@ int find_imagefiles(cchar *folder, int flags, char **&flist, int &NF, int Finit)
+#ifdef GLOB_ONLYDIR
    if (Fdirs && ! Fimages && ! Fthumbs)
       globflags += GLOB_ONLYDIR;
-
+#endif
    globdata.gl_pathc = 0;                                                        //  glob() setup
    globdata.gl_offs = 0;
    globdata.gl_pathc = 0;
