--- a/src/cfgparser/local.mk	2020-11-21 14:03:26.244348554 +0700
+++ b/src/cfgparser/local.mk	2020-11-21 14:06:09.976236365 +0700
@@ -77,6 +77,7 @@
 	$(MAKECMDPREFIX)$(BISON) -d -o "$@-tmp" "$<"
 	$(MAKECMDPREFIX)sed	-e 's/register //' \
 		-e '/^#line/s/$(call quote,$@-tmp)/$(call quote,$@)/' \
+		-e '/include.*h-tmp/s/-tmp"$$/"/' \
 		< "$@-tmp" >"$@"
 	$(MAKECMDPREFIX)mv "$(subst .c,.h,$@-tmp)" "$(subst .c,.h,$@)"
 	$(MAKECMDPREFIX)$(RM) "$@-tmp" "$(subst .c,.h,$@-tmp)"
