From 9ed4ae9cf207360000742c09007a32ad144b52db Mon Sep 17 00:00:00 2001
From: fafryd <dz1125.bug.tracker@gmail.com>
Date: Fri, 25 May 2018 20:17:35 +0200
Subject: [PATCH] fix libshadow

---
 hw/xfree86/dixmods/meson.build | 1 +
 1 file changed, 1 insertion(+)

diff --git a/hw/xfree86/dixmods/meson.build b/hw/xfree86/dixmods/meson.build
index 835d23215..6bd8029b6 100644
--- a/hw/xfree86/dixmods/meson.build
+++ b/hw/xfree86/dixmods/meson.build
@@ -38,6 +38,7 @@ shared_module(
 
     install: true,
     install_dir: module_dir,
+    install_rpath: '$ORIGIN/./',
 )
 
 if build_glx
-- 
2.17.0
