--- hw/xfree86/meson.build.orig
+++ hw/xfree86/meson.build
@@ -145,12 +145,15 @@
         install_dir: get_option('libexecdir'),
         # install_mode: ['r-sr-xr-x', 0, 0],
     )
-    configure_file(
-        input: 'Xorg.sh.in',
-        output: 'Xorg',
-        configuration: conf_data,
+    install_data(configure_file(
+			input: 'Xorg.sh.in',
+			output: 'Xorg.sh',
+			configuration: conf_data,
+    	),
+		rename: 'Xorg',
+		install_mode: 'rwxr-xr-x',
         install_dir: join_paths(get_option('prefix'), get_option('bindir')),
-    )
+	)
 endif

 executable('cvt',
