--- hw/xfree86/common/compiler.h	2015-06-05 17:19:40.000000000 +0200
+++ hw/xfree86/common/compiler.h	2015-06-20 11:07:13.548555711 +0200
@@ -758,7 +758,7 @@
     return xf86ReadMmio32Le((void *) ioBase, port);
 }
 
-#elif defined(__arm__) && defined(__linux__)
+#elif defined(__arm__) && defined(__linux__) && defined(__GLIBC__)
 
 /* for Linux on ARM, we use the LIBC inx/outx routines */
 /* note that the appropriate setup via "ioperm" needs to be done */
