diff -Naurp0 a/configure b/configure
--- a/configure	2019-12-21 16:48:39.860529868 +0300
+++ b/configure	2019-12-21 16:49:04.946979529 +0300
@@ -18049 +18049 @@ ac_compiler_gnu=$ac_cv_c_compiler_gnu
-PANGOMM_MODULES='glibmm-2.60 >= 2.59.1 cairomm-1.16 >= 1.15.1 pangocairo >= 1.41.0'
+PANGOMM_MODULES='glibmm-2.64 >= 2.59.1 cairomm-1.16 >= 1.15.1 pangocairo >= 1.41.0'
diff -Naurp0 a/tools/Makefile.am b/tools/Makefile.am
--- a/tools/Makefile.am	2019-12-21 16:48:51.875744672 +0300
+++ b/tools/Makefile.am	2019-12-21 16:49:15.519169295 +0300
@@ -33 +33 @@ extra_defs_gen_generate_extra_defs_SOURC
-extra_defs_gen_generate_extra_defs_LDADD   = $(PANGOMM_LIBS) -lglibmm_generate_extra_defs-2.60
+extra_defs_gen_generate_extra_defs_LDADD   = $(PANGOMM_LIBS) -lglibmm_generate_extra_defs-2.64
diff -Naurp0 a/tools/Makefile.in b/tools/Makefile.in
--- a/tools/Makefile.in	2019-12-21 16:49:33.502491793 +0300
+++ b/tools/Makefile.in	2019-12-21 16:49:45.390704304 +0300
@@ -378 +378 @@ extra_defs_gen_generate_extra_defs_SOURC
-extra_defs_gen_generate_extra_defs_LDADD = $(PANGOMM_LIBS) -lglibmm_generate_extra_defs-2.60
+extra_defs_gen_generate_extra_defs_LDADD = $(PANGOMM_LIBS) -lglibmm_generate_extra_defs-2.64
