--- config.mk.orig
+++ config.mk
@@ -39,5 +39,5 @@
 
 CPPFLAGS += $(NAME_DEFINES) -DSRVR_$$HOSTNAME $(DEBUG_CPPFLAGS)
 CFLAGS += -Wall -O3 $(DEBUG_CFLAGS)
-CFLAGS += -Icore -Iinclude -I/usr/include/freetype2
+CFLAGS += -Icore -Iinclude -I$(XBPS_CROSS_BASE)/usr/include/freetype2
 LDFLAGS += -lX11 -lXrandr -lXft
--- Makefile.orig
+++ Makefile
@@ -5,12 +5,12 @@
 			   -D__THIS_VERSION__=\"$(__THIS_VERSION__)\" \
 			   -D__CONFIG_NAME__=\"$(__CONFIG_NAME__)\"   \
 
-PREFIX?=/usr/X11R6
+PREFIX?=/usr
 CFLAGS?=-Os -pedantic -Wall $(NAME_DEFINES)
 
 all:
-	$(CC) $(CFLAGS) -I$(PREFIX)/include src/utils.c src/wm.c -L$(PREFIX)/lib -lX11 -lm -lXinerama -lXft -o berry
-	$(CC) $(CFLAGS) -I$(PREFIX)/include src/client.c -L$(PREFIX)/lib -lX11 -o berryc
+	$(CC) $(CFLAGS) -I$(XBPS_CROSS_BASE)/$(PREFIX)/include src/utils.c src/wm.c -L$(XBPS_CROSS_BASE)/$(PREFIX)/lib -lX11 -lm -lXinerama -lXft -o berry
+	$(CC) $(CFLAGS) -I$(XBPS_CROSS_BASE)/$(PREFIX)/include src/client.c -L$(XBPS_CROSS_BASE)/$(PREFIX)/lib -lX11 -o berryc
 
 install:
 	mkdir -p "$(DESTDIR)$(PREFIX)/bin"
