--- a/client/gui-qt/canvas.cpp	2020-02-15 16:49:02.000000000 +0100
+++ b/client/gui-qt/canvas.cpp	2020-09-01 19:25:35.075928695 +0200
@@ -18,6 +18,7 @@
 // Qt
 #include <QFontMetrics>
 #include <QPainter>
+#include <QPainterPath>
 
 // qt-client
 #include "canvas.h"
--- a/client/gui-qt/dialogs.cpp	2020-02-15 16:49:02.000000000 +0100
+++ b/client/gui-qt/dialogs.cpp	2020-09-01 19:25:42.242929065 +0200
@@ -24,6 +24,7 @@
 #include <QMessageBox>
 #include <QMouseEvent>
 #include <QPainter>
+#include <QPainterPath>
 #include <QRadioButton>
 #include <QRect>
 #include <QSignalMapper>
