diff --git a/inputremapper/configs/data.py b/inputremapper/configs/data.py
index bb868a6..438774a 100644
--- a/inputremapper/configs/data.py
+++ b/inputremapper/configs/data.py
@@ -25,8 +25,6 @@ import os
 import site
 import sys
 
-import pkg_resources
-
 from inputremapper.logging.logger import logger
 
 logged = False
@@ -56,8 +54,8 @@ def _try_python_package_location():
     """Look for the data dir at the packages installation location."""
     source = None
     try:
-        source = pkg_resources.require("input-remapper")[0].location
-        # failed in some ubuntu installations
+        import inputremapper
+        source = os.path.dirname(os.path.dirname(inputremapper.__file__))
     except Exception:
         logger.debug("failed to figure out package location")
         pass
diff --git a/inputremapper/logging/logger.py b/inputremapper/logging/logger.py
index 2547de4..ee9afb6 100644
--- a/inputremapper/logging/logger.py
+++ b/inputremapper/logging/logger.py
@@ -148,18 +148,13 @@ class Logger(logging.Logger):
 logger = Logger.bootstrap_logger()
 
 
-# using pkg_resources to figure out the version fails in many cases,
-# so we hardcode it instead
 VERSION = "2.1.1"
 EVDEV_VERSION = None
 try:
-    # pkg_resources very commonly fails/breaks
-    import pkg_resources
+    from importlib.metadata import version as pkg_version
 
-    EVDEV_VERSION = pkg_resources.require("evdev")[0].version
+    EVDEV_VERSION = pkg_version("evdev")
 except Exception as error:
-    # there have been pkg_resources.DistributionNotFound and
-    # pkg_resources.ContextualVersionConflict errors so far.
     # We can safely ignore all Exceptions here
     logger.info("Could not figure out the version")
     logger.debug(error)
