diff -p1ru a/src/service.c b/src/service.c
--- a/src/service.c	2024-08-03 10:50:29.773380597 +0800
+++ b/src/service.c	2024-08-03 10:54:10.156134073 +0800
@@ -776,4 +776,4 @@ create_menu (IndicatorPowerService * sel
   header = g_menu_item_new (NULL, "indicator._header");
-  g_menu_item_set_attribute (header, "x-canonical-type",
-                             "s", "com.canonical.indicator.root");
+  g_menu_item_set_attribute (header, "x-ayatana-type",
+                             "s", "org.ayatana.indicator.root");
   g_menu_item_set_submenu (header, G_MENU_MODEL (submenu));
