--- COLLADA2GLTF/dependencies/OpenCOLLADA/common/libBuffer/include/CommonFWriteBufferFlusher.h
+++ COLLADA2GLTF/dependencies/OpenCOLLADA/common/libBuffer/include/CommonFWriteBufferFlusher.h
@@ -35,15 +35,14 @@
 #ifdef __GNUC__
 #   include <cstdlib> /* size_t */
 #   include <cstdio>  /* FILE */
-#ifdef __FreeBSD__
-#include <stdint.h> /* int64_t */
-#endif
 #endif
 
+#include <stdint.h>
+
 #if __cplusplus > 199711L
 typedef int64_t __int64;
 #else
 #if (defined(__GNUC__) && !defined(__STRICT_ANSI__) && !defined(__MINGW32__)) || (__STDC_VERSION__ >= 199901L)
 typedef int64_t __int64;
 #endif
 #endif
