Patch from Firefox ESR 38 Void packaging.
--- configure.orig	2015-09-28 14:41:14.000000000 +0200
+++ configure	2015-10-07 13:36:23.281995377 +0200
@@ -511,8 +511,9 @@
 ac_ext=c
 # CFLAGS is not in ac_cpp because -g, -O, etc. are not valid cpp options.
 ac_cpp='$CPP $CPPFLAGS'
-ac_compile='${CC-cc} -c $CFLAGS $CPPFLAGS conftest.$ac_ext 1>&5'
-ac_link='${CC-cc} -o conftest${ac_exeext} $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS 1>&5'
+ac_compile='${CC-cc} -c conftest.$ac_ext 1>&5'
+ac_link='${CC-cc} -o conftest${ac_exeext} conftest.$ac_ext $LIBS 1>&5'
+
 cross_compiling=$ac_cv_prog_cc_cross
 
 ac_exeext=
--- mozilla/configure.orig	2015-05-29 18:30:41.474423097 +0200
+++ mozilla/configure	2015-05-29 18:30:53.451550589 +0200
@@ -10966,8 +10966,8 @@ ac_have_llvm_pr8927="no"
 ac_ext=c
 # CFLAGS is not in ac_cpp because -g, -O, etc. are not valid cpp options.
 ac_cpp='$CPP $CPPFLAGS'
-ac_compile='${CC-cc} -c $CFLAGS $CPPFLAGS conftest.$ac_ext 1>&5'
-ac_link='${CC-cc} -o conftest${ac_exeext} $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS 1>&5'
+ac_compile='${CC-cc} -c conftest.$ac_ext 1>&5'
+ac_link='${CC-cc} -o conftest${ac_exeext} conftest.$ac_ext $LIBS 1>&5'
 cross_compiling=$ac_cv_prog_cc_cross
 
 
--- mozilla/js/src/configure.orig	2015-05-29 18:31:25.049874012 +0200
+++ mozilla/js/src/configure	2015-05-29 18:31:36.898995291 +0200
@@ -9820,8 +9820,8 @@ ac_have_llvm_pr8927="no"
 ac_ext=c
 # CFLAGS is not in ac_cpp because -g, -O, etc. are not valid cpp options.
 ac_cpp='$CPP $CPPFLAGS'
-ac_compile='${CC-cc} -c $CFLAGS $CPPFLAGS conftest.$ac_ext 1>&5'
-ac_link='${CC-cc} -o conftest${ac_exeext} $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS 1>&5'
+ac_compile='${CC-cc} -c conftest.$ac_ext 1>&5'
+ac_link='${CC-cc} -o conftest${ac_exeext} conftest.$ac_ext $LIBS 1>&5'
 cross_compiling=$ac_cv_prog_cc_cross
 
 
