--- tools/profiler/core/platform-linux-android.cpp.orig	2019-07-09 22:58:30.039475686 +0200
+++ tools/profiler/core/platform-linux-android.cpp	2019-07-09 22:58:39.331437677 +0200
@@ -73,7 +73,7 @@
 
 int profiler_current_thread_id() {
   // glibc doesn't provide a wrapper for gettid().
-#if defined(__GLIBC__)
+#if defined(__linux__)
   return static_cast<int>(static_cast<pid_t>(syscall(SYS_gettid)));
 #else
   return static_cast<int>(gettid());
