--- build/moz.configure/rust.configure.orig	2018-10-26 15:02:30.276600157 +0200
+++ build/moz.configure/rust.configure	2018-10-26 15:02:14.081581242 +0200
@@ -119,7 +119,10 @@ def rust_supported_targets(rustc):
     ambiguous = set()
     per_raw_os = {}
     for t in out:
-        t = split_triplet(t, allow_unknown=True)
+        try:
+            t = split_triplet(t, allow_unknown=True)
+        except:
+            continue
         key = (t.cpu, t.endianness, t.os)
         if key in per_os:
             previous = per_os[key]
