--- a/mozglue/misc/StackWalk.cpp	2021-11-05 15:55:03.614609706 +0000
+++ b/mozglue/misc/StackWalk.cpp	2021-10-28 19:03:47.000000000 +0100
@@ -45,7 +45,7 @@
 #  define MOZ_STACKWALK_SUPPORTS_MACOSX 0
 #endif

-#if (defined(linux) &&                                            \
+#if defined(__GLIBC__) && (defined(linux) && \
      ((defined(__GNUC__) && (defined(__i386) || defined(PPC))) || \
       defined(HAVE__UNWIND_BACKTRACE)))
 #  define MOZ_STACKWALK_SUPPORTS_LINUX 1
