--- gfx/skia/skia/third_party/skcms/skcms.cc
+++ gfx/skia/skia/third_party/skcms/skcms.cc
@@ -1806,6 +1806,11 @@ typedef enum {
     Op_store_ffff,
 } Op;
 
+// work around ICE on 32-bit ppc
+#if defined(__powerpc__) && !defined(__powerpc64__)
+#  define SKCMS_PORTABLE
+#endif
+
 #if defined(__clang__)
     template <int N, typename T> using Vec = T __attribute__((ext_vector_type(N)));
 #elif defined(__GNUC__)
