--- mail/installer/Makefile.in	2009-12-01 06:47:04.000000000 +0300
+++ ../comm-1.9.1-shared/mail/installer/Makefile.in	2009-12-22 16:09:43.000000000 +0300
@@ -95,9 +95,11 @@
 # mozconfig instead.
 ifndef MAIL_PKG_SHARED
 ifndef BUILD_STATIC_LIBS
+ifeq (BUILD_STATIC_LIBS, 1)
 $(error you need an "--enable-static" build to package a build)
 endif
 endif
+endif
 
 # On mozilla-central, packager.mk uses core_abspath to redefine DIST
 # As the comm-central config.mk doesn't set it, do it ourselves instead
