--- config/baseconfig.mk	2012-11-21 03:19:15.000000000 +0200
+++ config/baseconfig.mk	2012-11-21 03:20:35.000000000 +0200
@@ -1,9 +1,9 @@
 INCLUDED_AUTOCONF_MK = 1

-includedir := $(includedir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
-idldir = $(datadir)/idl/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
-installdir = $(libdir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
-sdkdir = $(libdir)/$(MOZ_APP_NAME)-devel-$(MOZ_APP_VERSION)
+includedir := $(includedir)/$(MOZ_APP_NAME)
+idldir = $(datadir)/idl/$(MOZ_APP_NAME)
+installdir = $(libdir)/$(MOZ_APP_NAME)
+sdkdir = $(libdir)/$(MOZ_APP_NAME)-devel
 MOZILLA_SRCDIR = $(topsrcdir)/mozilla
 MOZDEPTH = $(DEPTH)/mozilla
 DIST = $(MOZDEPTH)/dist
--- mozilla/config/baseconfig.mk	2012-11-21 03:19:16.000000000 +0200
+++ mozilla/config/baseconfig.mk	2012-11-21 03:19:20.000000000 +0200
@@ -1,9 +1,9 @@
 INCLUDED_AUTOCONF_MK = 1

-includedir := $(includedir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
-idldir = $(datadir)/idl/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
-installdir = $(libdir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
-sdkdir = $(libdir)/$(MOZ_APP_NAME)-devel-$(MOZ_APP_VERSION)
+includedir := $(includedir)/$(MOZ_APP_NAME)
+idldir = $(datadir)/idl/$(MOZ_APP_NAME)
+installdir = $(libdir)/$(MOZ_APP_NAME)
+sdkdir = $(libdir)/$(MOZ_APP_NAME)-devel
 DIST = $(DEPTH)/dist

 # We do magic with OBJ_SUFFIX in config.mk, the following ensures we don't
--- mozilla/js/src/config/baseconfig.mk	2012-11-21 03:19:15.000000000 +0200
+++ mozilla/js/src/config/baseconfig.mk	2012-11-21 03:19:20.000000000 +0200
@@ -1,7 +1,7 @@
 INCLUDED_AUTOCONF_MK = 1

-installdir = $(libdir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
-sdkdir = $(libdir)/$(MOZ_APP_NAME)-devel-$(MOZ_APP_VERSION)
+installdir = $(libdir)/$(MOZ_APP_NAME)
+sdkdir = $(libdir)/$(MOZ_APP_NAME)-devel

 ifneq (,$(filter /%,$(TOP_DIST)))
 DIST = $(TOP_DIST)
