--- lib/krb5/cache.c	2009-11-22 01:41:51.000000000 +1000
+++ lib/krb5/cache.c	2009-11-23 22:27:15.000000000 +1000
@@ -1347,10 +1347,11 @@
  *
  */
 
-struct krb5_cccol_cursor {
+struct krb5_cccol_cursor_data {
     int idx;
     krb5_cc_cache_cursor cursor;
 };
+/* typedef of krb5_cccol_cursor is in krb5.h */
 
 /**
  * Get a new cache interation cursor that will interate over all
--- lib/krb5/krb5.h	2009-11-22 01:41:51.000000000 +1000
+++ lib/krb5/krb5.h	2009-11-23 22:28:10.000000000 +1000
@@ -311,7 +311,7 @@
 				 NULL)
 
 typedef void *krb5_cc_cursor;
-typedef struct krb5_cccol_cursor *krb5_cccol_cursor;
+typedef struct krb5_cccol_cursor_data *krb5_cccol_cursor;
 
 typedef struct krb5_ccache_data {
     const struct krb5_cc_ops *ops;
