diff -urb patched-easystroke-master.orig/easystroke/Makefile patched-easystroke-master/easystroke/Makefile
--- patched-easystroke-master.orig/easystroke/Makefile	2021-07-19 17:16:39.949855973 +0100
+++ patched-easystroke-master/easystroke/Makefile	2021-07-19 17:17:43.041490580 +0100
@@ -45,7 +45,7 @@
 GENFILES = gui.c desktop.c po/POTFILES.in easystroke.desktop
 GZFILES  = $(wildcard *.gz)
 
-VERSION  = $(shell test -e debian/changelog && grep '(.*)' debian/changelog | sed 's/.*(//' | sed 's/).*//' | head -n1 || (test -e version && cat version || git describe))
+VERSION  =
 GIT      = $(wildcard .git/index version)
 DIST     = easystroke-$(VERSION)
 
@@ -141,7 +141,6 @@
 tmp/$(DIST): $(GIT)
 	$(RM) -r tmp
 	mkdir tmp
-	git archive --format=tar --prefix=$(DIST)/ HEAD | (cd tmp && tar x)
 	echo $(VERSION) > $@/version
 	$(RM) $@/.gitignore $@/release
 
