--- a/desmume/src/utils/colorspacehandler/colorspacehandler_NEON.cpp	2024-05-30 16:41:15.411778633 -0400
+++ b/desmume/src/utils/colorspacehandler/colorspacehandler_NEON.cpp	2024-05-30 16:40:50.624235060 -0400
@@ -705,7 +705,7 @@
 		for (; i < pixCountVec128; i+=(sizeof(v128u32)/sizeof(u32)))
 		{
 			v128u32 dstVec = vld1q_u32(dst+i);
-			v128u32 tempDst = (SWAP_RB) ? COLOR32_SWAPRB_NEON(dstVec) : dstVec;
+			v128u32 tempDst = (SWAP_RB) ? (v128u32) COLOR32_SWAPRB_NEON(dstVec) : (v128u32) dstVec;
 			
 			v128u32 rb = vandq_u32(             tempDst,      vdupq_n_u32(0x00FF00FF) );
 			v128u32 g  = vandq_u32( vshrq_n_u32(tempDst,  8), vdupq_n_u32(0x000000FF) );

