Upstream: No
Reason: Tries to run cross compiled binary

diff --git a/m4/sasl2.m4 b/m4/sasl2.m4
index 56e0504..c1d8e35 100644
--- a/m4/sasl2.m4
+++ b/m4/sasl2.m4
@@ -315,31 +315,8 @@ if test "$gssapi" != no; then
   LIBS="$LIBS $GSSAPIBASE_LIBS"
 
   AC_MSG_CHECKING([for SPNEGO support in GSSAPI libraries])
-  AC_TRY_RUN([
-#ifdef HAVE_GSSAPI_H
-#include <gssapi.h>
-#else
-#include <gssapi/gssapi.h>
-#endif
-
-int main(void)
-{
-    gss_OID_desc spnego_oid = { 6, (void *) "\x2b\x06\x01\x05\x05\x02" };
-    gss_OID_set mech_set;
-    OM_uint32 min_stat;
-    int have_spnego = 0;
-                                                                               
-    if (gss_indicate_mechs(&min_stat, &mech_set) == GSS_S_COMPLETE) {
-	gss_test_oid_set_member(&min_stat, &spnego_oid, mech_set, &have_spnego);
-	gss_release_oid_set(&min_stat, &mech_set);
-    }
-
-    return (!have_spnego);  // 0 = success, 1 = failure
-}
-],	
-	[ AC_DEFINE(HAVE_GSS_SPNEGO,,[Define if your GSSAPI implementation supports SPNEGO])
-	AC_MSG_RESULT(yes) ],
-	AC_MSG_RESULT(no))
+  AC_DEFINE(HAVE_GSS_SPNEGO,,[Define if your GSSAPI implementation supports SPNEGO])
+  AC_MSG_RESULT(yes)
   LIBS="$cmu_save_LIBS"
 
 else
