--- plugins/media-keys/gsd-media-keys-manager.c	2011-10-20 08:19:31.818642262 +0200
+++ plugins/media-keys/gsd-media-keys-manager.c	2011-10-20 08:23:55.127492221 +0200
@@ -2066,15 +2066,15 @@
                           (GAsyncReadyCallback) power_keyboard_ready_cb,
                           manager);

-        g_dbus_proxy_new (manager->priv->connection,
-                          G_DBUS_PROXY_FLAGS_NONE,
-                          NULL,
-                          "org.freedesktop.UPower",
-                          "/org/freedesktop/UPower",
-                          "org.freedesktop.UPower",
-                          NULL,
-                          (GAsyncReadyCallback) upower_ready_cb,
-                          manager);
+        g_dbus_proxy_new_for_bus (G_BUS_TYPE_SYSTEM,
+                                  G_DBUS_PROXY_FLAGS_DO_NOT_LOAD_PROPERTIES,
+                                  NULL,
+                                  "org.freedesktop.UPower",
+                                  "/org/freedesktop/UPower",
+                                  "org.freedesktop.UPower",
+                                  NULL,
+                                  (GAsyncReadyCallback) upower_ready_cb,
+                                  manager);
 }

 static void
