From 3bb4b0d41f76483c4ae4d8d62faa4531049badeb Mon Sep 17 00:00:00 2001
From: moneromooo-monero <moneromooo-monero@users.noreply.github.com>
Date: Mon, 22 Oct 2018 22:17:15 +0000
Subject: [PATCH] miner: fix build with boost 1.69

---
 src/cryptonote_basic/miner.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git src/cryptonote_basic/miner.cpp src/cryptonote_basic/miner.cpp
index d0b03593ed..d8ca2dd357 100644
--- src/cryptonote_basic/miner.cpp
+++ src/cryptonote_basic/miner.cpp
@@ -637,7 +637,7 @@ namespace cryptonote
         boost::tribool battery_powered(on_battery_power());
         if(!indeterminate( battery_powered ))
         {
-          on_ac_power = !battery_powered;
+          on_ac_power = !(bool)battery_powered;
         }
       }
 
