Index: monero-0.18.1.2/src/common/dns_utils.cpp
===================================================================
--- monero-0.18.1.2.orig/src/common/dns_utils.cpp
+++ monero-0.18.1.2/src/common/dns_utils.cpp
@@ -38,6 +38,12 @@
 #include <boost/algorithm/string/join.hpp>
 #include <boost/optional.hpp>
 #include <boost/utility/string_ref.hpp>
+
+#include <deque>
+#include <map>
+#include <set>
+#include <string>
+
 using namespace epee;
 
 #undef MONERO_DEFAULT_LOG_CATEGORY
