--- CMakeLists.txt	2017-09-22 19:22:04.735657685 +0200
+++ CMakeLists.txt	2017-09-22 19:20:08.164169125 +0200
@@ -388,7 +388,7 @@
   include(TestCXXAcceptsFlag)
   set(ARCH native CACHE STRING "CPU to build for: -march value or 'default' to not pass -march at all")
   message(STATUS "Building on ${CMAKE_SYSTEM_PROCESSOR} for ${ARCH}")
+  if(ARCH STREQUAL "default" OR CMAKE_CROSSCOMPILING)
-  if(ARCH STREQUAL "default")
     set(ARCH_FLAG "")
   elseif(PPC64LE)
     set(ARCH_FLAG "-mcpu=${ARCH}")
