diff --git a/CMakeLists.txt b/CMakeLists.txt
index b3eff5d..defc6b2 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -76,7 +76,7 @@ endif()
 ## Compiler flags
 set(CMAKE_CXX_FLAGS
  "${CMAKE_CXX_FLAGS} -Wall -Wextra -fno-rtti")
-if (${CMAKE_HOST_SYSTEM_PROCESSOR} STREQUAL "x86_64")
+if (${CMAKE_SYSTEM_PROCESSOR} STREQUAL "x86_64")
     set(CMAKE_CXX_FLAGS
      "${CMAKE_CXX_FLAGS} -m64 -msse -msse2")
 endif()
