diff --git a/CMakeLists.txt b/CMakeLists.txt
index 82306ad..74f407a 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -105,11 +105,6 @@ endif ()
 # Use this instead of above for 32 bit
 # set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -m32")
 
-if ("${CMAKE_BUILD_TYPE}" STREQUAL Release)
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O2")
-else ()
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -ggdb -O0 -DDEBUG")
-endif ()
 
 set(COMMON_SRC_DIR "${PROJECT_SOURCE_DIR}/src")
 set(COMMON_INC_DIR "${PROJECT_SOURCE_DIR}/include/rocm_smi")
