--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -108,7 +108,6 @@
 	Netclient.$(OBJEXT) ScreenHandler.$(OBJEXT) window.$(OBJEXT) \
 	WindowlistMenu.$(OBJEXT)
 bbkeys_OBJECTS = $(am_bbkeys_OBJECTS)
-bbkeys_DEPENDENCIES = /usr/lib/libbt.a
 AM_V_P = $(am__v_P_@AM_V@)
 am__v_P_ = $(am__v_P_@AM_DEFAULT_V@)
 am__v_P_0 = false
@@ -350,7 +349,7 @@
 	WindowlistMenu.cpp \
 	WindowlistMenu.h 
 
-bbkeys_LDADD = /usr/lib/libbt.a
+bbkeys_LDADD = -lbt
 AM_INSTALLCHECK_STD_OPTIONS_EXEMPT = bbkeys
 all: all-am
 
@@ -444,7 +443,7 @@
 	  done; \
 	done; rm -f c$${pid}_.???; exit $$bad
 
-bbkeys$(EXEEXT): $(bbkeys_OBJECTS) $(bbkeys_DEPENDENCIES) $(EXTRA_bbkeys_DEPENDENCIES) 
+bbkeys$(EXEEXT): $(bbkeys_OBJECTS) $(EXTRA_bbkeys_DEPENDENCIES) 
 	@rm -f bbkeys$(EXEEXT)
 	$(AM_V_CXXLD)$(CXXLINK) $(bbkeys_OBJECTS) $(bbkeys_LDADD) $(LIBS)
 
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -31,6 +31,5 @@
 	window.hh \
 	WindowlistMenu.cpp \
 	WindowlistMenu.h 
-bbkeys_LDADD = /usr/lib/libbt.a
 
 AM_INSTALLCHECK_STD_OPTIONS_EXEMPT = bbkeys
