From 2fd12847496a9a242ca8edc448d400d3660b8009 Mon Sep 17 00:00:00 2001
From: Jehan <jehan@girinstud.io>
Date: Thu, 4 Dec 2025 23:47:00 +0100
Subject: [PATCH] Issue #15257: test failing because gegl:bevel doesn't see GPL
 ops.

---
 app/tests/tests.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/app/tests/tests.c b/app/tests/tests.c
index 2b12beed34..62431dc6a1 100644
--- a/app/tests/tests.c
+++ b/app/tests/tests.c
@@ -162,6 +162,10 @@ gimp_init_for_gui_testing_internal (gboolean  show_gui,
 
   /* from main() */
   gimp_log_init ();
+  g_object_set (gegl_config (),
+                "use-opencl",          FALSE,
+                "application-license", "GPL3",
+                NULL);
   gegl_init (NULL, NULL);
 
   /* Introduce an error margin for positions written to sessionrc */
-- 
GitLab

