--- js/src/configure.in.old	2009-11-08 19:50:54.299642792 -0500
+++ js/src/configure.in	2009-11-08 19:57:49.235621814 -0500
@@ -2465,6 +2465,10 @@
     ENABLE_JIT=1
     NANOJIT_ARCH=i386
     ;;
+x86_64*-*)
+    ENABLE_JIT=1
+    NANOJIT_ARCH=X64
+    ;;
 arm*-*)
     ENABLE_JIT=1
     NANOJIT_ARCH=ARM
@@ -2488,7 +2492,10 @@
 i?86-*)
     AC_DEFINE(AVMPLUS_IA32)
     ;;
-
+x86_64*-*)
+    AC_DEFINE(AVMPLUS_AMD64)
+    AC_DEFINE(AVMPLUS_64BIT)
+    ;;
 arm*-*)
     AC_DEFINE(AVMPLUS_ARM)
     ;;
