diff -Nur mozilla-2.0.orig/xulrunner/installer/libxul-embedding.pc.in mozilla-2.0/xulrunner/installer/libxul-embedding.pc.in
--- mozilla-2.0.orig/xulrunner/installer/libxul-embedding.pc.in	2011-03-03 14:12:04.000000000 -0800
+++ mozilla-2.0/xulrunner/installer/libxul-embedding.pc.in	2011-03-13 03:25:12.350027359 -0700
@@ -6,5 +6,6 @@
 Name: libxul-embedding
 Description: Static library for version-independent embedding of the Mozilla runtime
 Version: %MOZILLA_VERSION%
+Requires: %NSPR_NAME% >= %NSPR_VERSION%
 Libs: -L${sdkdir}/lib -lxpcomglue -ldl
 Cflags: -DXPCOM_GLUE -I${includedir} %WCHAR_CFLAGS%
diff -Nur mozilla-2.0.orig/xulrunner/installer/libxul.pc.in mozilla-2.0/xulrunner/installer/libxul.pc.in
--- mozilla-2.0.orig/xulrunner/installer/libxul.pc.in	2011-03-03 14:12:04.000000000 -0800
+++ mozilla-2.0/xulrunner/installer/libxul.pc.in	2011-03-13 03:25:28.010027359 -0700
@@ -1,5 +1,6 @@
 prefix=%prefix%
 sdkdir=%sdkdir%
+libdir=%libdir%
 includedir=%includedir%
 idldir=%idldir%
 
diff -Nur mozilla-2.0.orig/xulrunner/installer/Makefile.in mozilla-2.0/xulrunner/installer/Makefile.in
--- mozilla-2.0.orig/xulrunner/installer/Makefile.in	2011-03-03 14:12:04.000000000 -0800
+++ mozilla-2.0/xulrunner/installer/Makefile.in	2011-03-13 03:25:55.570027359 -0700
@@ -121,6 +121,7 @@
 	-e "s|%includedir%|$(includedir)|" \
 	-e "s|%idldir%|$(idldir)|" \
 	-e "s|%sdkdir%|$(sdkdir)|" \
+	-e "s|%libdir%|$(installdir)|" \
 	-e "s|%MOZ_APP_NAME%|$(MOZ_APP_NAME)|" \
 	-e "s|%MOZILLA_VERSION%|$(MOZ_APP_VERSION)|" \
 	-e "s|%WCHAR_CFLAGS%|$(WCHAR_CFLAGS)|" \
diff -Nur mozilla-2.0.orig/xulrunner/installer/mozilla-gtkmozembed-embedding.pc.in mozilla-2.0/xulrunner/installer/mozilla-gtkmozembed-embedding.pc.in
--- mozilla-2.0.orig/xulrunner/installer/mozilla-gtkmozembed-embedding.pc.in	2011-03-03 14:12:04.000000000 -0800
+++ mozilla-2.0/xulrunner/installer/mozilla-gtkmozembed-embedding.pc.in	2011-03-13 03:26:18.676694023 -0700
@@ -1,5 +1,6 @@
 prefix=%prefix%
 sdkdir=%sdkdir%
+libdir=%libdir%
 includedir=%includedir%
 
 Name: mozilla-gtkembedmoz
diff -Nur mozilla-2.0.orig/xulrunner/installer/mozilla-gtkmozembed.pc.in mozilla-2.0/xulrunner/installer/mozilla-gtkmozembed.pc.in
--- mozilla-2.0.orig/xulrunner/installer/mozilla-gtkmozembed.pc.in	2011-03-03 14:12:04.000000000 -0800
+++ mozilla-2.0/xulrunner/installer/mozilla-gtkmozembed.pc.in	2011-03-13 03:26:41.566694025 -0700
@@ -1,5 +1,6 @@
 prefix=%prefix%
 sdkdir=%sdkdir%
+libdir=%libdir%
 includedir=%includedir%
 
 Name: mozilla-gtkembedmoz
diff -Nur mozilla-2.0.orig/xulrunner/installer/mozilla-js.pc.in mozilla-2.0/xulrunner/installer/mozilla-js.pc.in
--- mozilla-2.0.orig/xulrunner/installer/mozilla-js.pc.in	2011-03-03 14:12:04.000000000 -0800
+++ mozilla-2.0/xulrunner/installer/mozilla-js.pc.in	2011-03-13 03:27:19.680027357 -0700
@@ -7,4 +7,4 @@
 Version: %MOZILLA_VERSION%
 Requires: %NSPR_NAME% >= %NSPR_VERSION%
 Libs: -L${sdkdir}/lib %MOZ_JS_LINK%
-Cflags: -I${includedir} -DXP_UNIX -DJS_THREADSAFE
+Cflags: -I${includedir} -I${includedir}/js -DXP_UNIX -DJS_THREADSAFE
