diff -Nur mozilla-2.0.orig/xulrunner/installer/Makefile.in mozilla-2.0/xulrunner/installer/Makefile.in
--- mozilla-2.0.orig/xulrunner/installer/Makefile.in	2011-03-03 14:12:04.000000000 -0800
+++ mozilla-2.0/xulrunner/installer/Makefile.in	2011-03-13 01:58:19.663360705 -0800
@@ -44,6 +44,8 @@
 
 include $(DEPTH)/config/autoconf.mk
 
+MOZ_APP_VERSION="2.0"
+
 NO_PKG_FILES = \
 	xulrunner-config \
 	regchrome* \
