--- modules/libpr0n/encoders/png/nsPNGEncoder.cpp.orig	2010-01-17 00:15:53.979744638 +0100
+++ modules/libpr0n/encoders/png/nsPNGEncoder.cpp	2010-01-17 00:16:07.855993411 +0100
@@ -135,7 +135,7 @@
 
   // initialize
   mPNG = png_create_write_struct(PNG_LIBPNG_VER_STRING,
-                                 png_voidp_NULL,
+                                 NULL,
                                  ErrorCallback,
                                  ErrorCallback);
   if (! mPNG)
