From 15e3ace08cb25e53941249fa1ee680a1e2f871b4 Mon Sep 17 00:00:00 2001
From: Xeonacid <h.dwwwwww@gmail.com>
Date: Sat, 8 Jan 2022 02:00:28 +0800
Subject: [PATCH] Meson: remove positional arguments from i18n.merge_file
 (#649)

---
 data/meson.build | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/data/meson.build b/data/meson.build
index 435b84172..a64fc120e 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -4,7 +4,6 @@ install_data(
 )
 
 i18n.merge_file (
-    'desktop',
     input: meson.project_name() + '.desktop.in',
     output: meson.project_name() + '.desktop',
     install: true,
@@ -14,7 +13,6 @@ i18n.merge_file (
 )
 
 i18n.merge_file (
-    'desktop',
     input: 'open-pantheon-terminal-here.desktop.in',
     output: 'open-pantheon-terminal-here.desktop',
     install: true,
@@ -24,7 +22,6 @@ i18n.merge_file (
 )
 
 i18n.merge_file (
-    'appdata',
     input: meson.project_name() + '.appdata.xml.in',
     output: meson.project_name() + '.appdata.xml',
     install: true,
