--- SConstruct.orig
+++ SConstruct
@@ -236,6 +237,6 @@
 env.Alias('install', env.Command(share_dst, share_src, [Copy(share_dst, share_src)]))
 env.Alias('install', env.Command(man_dst, man_src, [Copy(man_dst, man_src)]))
 env.Alias('install', env.Command(man_net_dst, man_net_src, [Copy(man_net_dst, man_net_src)]))
-env.Alias('install', env.Command(image_dst, image_src, [Copy(image_dst, image_src)]))
-env.Alias('install', env.Command(desktop_dst, desktop_src, [Copy(desktop_dst, desktop_src)]))
+env.Alias('install', env.Command(image_dst, image_src, [Mkdir(image_dst),Copy(image_dst, image_src)]))
+env.Alias('install', env.Command(desktop_dst, desktop_src, [Mkdir(desktop_dst),Copy(desktop_dst, desktop_src)]))
 env.Alias('install', (prefix + "/bin/"))
