---
 Makefile |    4 ++++
 1 file changed, 4 insertions(+)

Index: linux-3.16-ck1/Makefile
===================================================================
--- linux-3.16-ck1.orig/Makefile	2014-08-16 14:46:12.574916784 +1000
+++ linux-3.16-ck1/Makefile	2014-08-16 14:46:12.573916784 +1000
@@ -10,6 +10,10 @@ NAME = Shuffling Zombie Juror
 # Comments in this file are targeted only to the developer, do not
 # expect to learn how to build the kernel reading this file.
 
+CKVERSION = -ck1
+CKNAME = BFS Powered
+EXTRAVERSION := $(EXTRAVERSION)$(CKVERSION)
+
 # Do not:
 # o  use make's built-in rules and variables
 #    (this increases performance and avoids hard-to-debug behaviour);
