diff --git a/chat.stoat.StoatDesktop.desktop b/chat.stoat.StoatDesktop.desktop
index b0e0b55..0aac665 100644
--- a/chat.stoat.StoatDesktop.desktop
+++ b/chat.stoat.StoatDesktop.desktop
@@ -1,10 +1,10 @@
 [Desktop Entry]
 Name=Stoat
 Comment=Open source, user-first chat platform
-Exec=stoat-desktop
+Exec="/usr/bin/stoat-desktop"
 Terminal=false
 Type=Application
-Icon=chat.stoat.StoatDesktop
+Icon=stoat
 Categories=Network;InstantMessaging
 StartupWMClass=stoat-desktop
