diff -u -r vlc-3.0.10/modules/gui/qt/components/playlist/views.cpp vlc-3.0.10-qt-5.15/modules/gui/qt/components/playlist/views.cpp
--- modules/gui/qt/components/playlist/views.cpp	2017-11-24 15:29:17.000000000 +0000
+++ modules/gui/qt/components/playlist/views.cpp	2020-05-29 20:50:09.267959371 +0000
@@ -27,6 +27,7 @@
 #include "input_manager.hpp"                      /* THEMIM */
 
 #include <QPainter>
+#include <QPainterPath>
 #include <QRect>
 #include <QStyleOptionViewItem>
 #include <QFontMetrics>
diff -u -r vlc-3.0.10/modules/gui/qt/dialogs/plugins.cpp vlc-3.0.10-qt-5.15/modules/gui/qt/dialogs/plugins.cpp
--- modules/gui/qt/dialogs/plugins.cpp	2017-11-24 15:29:17.000000000 +0000
+++ modules/gui/qt/dialogs/plugins.cpp	2020-05-29 20:49:44.114275108 +0000
@@ -53,6 +53,7 @@
 #include <QListView>
 #include <QListWidget>
 #include <QPainter>
+#include <QPainterPath>
 #include <QStyleOptionViewItem>
 #include <QKeyEvent>
 #include <QPushButton>
diff -u -r vlc-3.0.10/modules/gui/qt/util/timetooltip.hpp vlc-3.0.10-qt-5.15/modules/gui/qt/util/timetooltip.hpp
--- modules/gui/qt/util/timetooltip.hpp	2018-05-03 12:03:21.000000000 +0000
+++ modules/gui/qt/util/timetooltip.hpp	2020-05-29 20:40:30.306628713 +0000
@@ -24,6 +24,7 @@
 
 #include "qt.hpp"
 
+#include <QPainterPath>
 #include <QWidget>
 
 class TimeTooltip : public QWidget
