Fix build libmatroska-1.1.0, from Gentoo.

--- modules/demux/mkv/matroska_segment.cpp	2011-02-02 13:45:56.115863316 +0100
+++ modules/demux/mkv/matroska_segment.cpp	2011-02-02 13:55:23.957416146 +0100
@@ -220,74 +220,6 @@
                     }
                     ep->Up();
                 }
-                else if( MKV_IS_ID( el, KaxTagGeneral ) )
-                {
-                    msg_Dbg( &sys.demuxer, "|   + General" );
-                    ep->Down();
-                    while( ( el = ep->Get() ) != NULL )
-                    {
-                        msg_Dbg( &sys.demuxer, "|   |   + Unknown (%s)", typeid( *el ).name() );
-                    }
-                    ep->Up();
-                }
-                else if( MKV_IS_ID( el, KaxTagGenres ) )
-                {
-                    msg_Dbg( &sys.demuxer, "|   + Genres" );
-                    ep->Down();
-                    while( ( el = ep->Get() ) != NULL )
-                    {
-                        msg_Dbg( &sys.demuxer, "|   |   + Unknown (%s)", typeid( *el ).name() );
-                    }
-                    ep->Up();
-                }
-                else if( MKV_IS_ID( el, KaxTagAudioSpecific ) )
-                {
-                    msg_Dbg( &sys.demuxer, "|   + Audio Specific" );
-                    ep->Down();
-                    while( ( el = ep->Get() ) != NULL )
-                    {
-                        msg_Dbg( &sys.demuxer, "|   |   + Unknown (%s)", typeid( *el ).name() );
-                    }
-                    ep->Up();
-                }
-                else if( MKV_IS_ID( el, KaxTagImageSpecific ) )
-                {
-                    msg_Dbg( &sys.demuxer, "|   + Images Specific" );
-                    ep->Down();
-                    while( ( el = ep->Get() ) != NULL )
-                    {
-                        msg_Dbg( &sys.demuxer, "|   |   + Unknown (%s)", typeid( *el ).name() );
-                    }
-                    ep->Up();
-                }
-                else if( MKV_IS_ID( el, KaxTagMultiComment ) )
-                {
-                    msg_Dbg( &sys.demuxer, "|   + Multi Comment" );
-                }
-                else if( MKV_IS_ID( el, KaxTagMultiCommercial ) )
-                {
-                    msg_Dbg( &sys.demuxer, "|   + Multi Commercial" );
-                }
-                else if( MKV_IS_ID( el, KaxTagMultiDate ) )
-                {
-                    msg_Dbg( &sys.demuxer, "|   + Multi Date" );
-                }
-                else if( MKV_IS_ID( el, KaxTagMultiEntity ) )
-                {
-                    msg_Dbg( &sys.demuxer, "|   + Multi Entity" );
-                }
-                else if( MKV_IS_ID( el, KaxTagMultiIdentifier ) )
-                {
-                    msg_Dbg( &sys.demuxer, "|   + Multi Identifier" );
-                }
-                else if( MKV_IS_ID( el, KaxTagMultiLegal ) )
-                {
-                    msg_Dbg( &sys.demuxer, "|   + Multi Legal" );
-                }
-                else if( MKV_IS_ID( el, KaxTagMultiTitle ) )
-                {
-                    msg_Dbg( &sys.demuxer, "|   + Multi Title" );
-                }
                 else
                 {
                     msg_Dbg( &sys.demuxer, "|   + LoadTag Unknown (%s)", typeid( *el ).name() );
--- modules/demux/mkv/mkv.hpp	2011-02-02 13:45:56.114863318 +0100
+++ modules/demux/mkv/mkv.hpp	2011-02-02 13:46:08.647832555 +0100
@@ -92,7 +92,6 @@
 #include "matroska/KaxSegment.h"
 #include "matroska/KaxTag.h"
 #include "matroska/KaxTags.h"
-#include "matroska/KaxTagMulti.h"
 #include "matroska/KaxTracks.h"
 #include "matroska/KaxTrackAudio.h"
 #include "matroska/KaxTrackVideo.h"
