--- modules/misc/notify/notify.c.orig	2011-02-28 23:54:18.794722978 +0100
+++ modules/misc/notify/notify.c	2011-02-28 23:54:49.425738093 +0100
@@ -312,7 +312,7 @@ static int Notify( vlc_object_t *p_this,
     }
 
     notification = notify_notification_new( _("Now Playing"),
-                                            psz_temp, NULL, NULL );
+                                            psz_temp, NULL );
     notify_notification_set_timeout( notification,
                                 var_InheritInteger(p_this, "notify-timeout") );
     notify_notification_set_urgency( notification, NOTIFY_URGENCY_LOW );
