*** a/modules/codec/x265.c	2025-06-01 22:43:12.511233665 -0400
--- b/modules/codec/x265.c	2025-06-01 23:35:53.645226763 -0400
*************** static block_t *Encode(encoder_t *p_enc,
*** 90,93 ****
      uint32_t i_nal = 0;
      x265_encoder_encode(p_sys->h, &nal, &i_nal,
!             likely(p_pict) ? &pic : NULL, &pic);
  
--- 90,94 ----
      uint32_t i_nal = 0;
+     x265_picture *pic_out = NULL;
      x265_encoder_encode(p_sys->h, &nal, &i_nal,
!             likely(p_pict) ? &pic : NULL, &pic_out);
  
