diff -up cups-1.5b1/cups/ppd.c.strict-ppd-line-length cups-1.5b1/cups/ppd.c
--- cups-1.5b1/cups/ppd.c.strict-ppd-line-length	2011-05-20 05:49:49.000000000 +0200
+++ cups-1.5b1/cups/ppd.c	2011-05-24 15:46:13.000000000 +0200
@@ -2786,7 +2786,7 @@ ppd_read(cups_file_t    *fp,		/* I - Fil
 	*lineptr++ = ch;
 	col ++;
 
-	if (col > (PPD_MAX_LINE - 1))
+	if (col > (PPD_MAX_LINE - 1) && cg->ppd_conform == PPD_CONFORM_STRICT)
 	{
 	 /*
           * Line is too long...
@@ -2847,7 +2847,7 @@ ppd_read(cups_file_t    *fp,		/* I - Fil
 	{
 	  col ++;
 
-	  if (col > (PPD_MAX_LINE - 1))
+	  if (col > (PPD_MAX_LINE - 1) && cg->ppd_conform == PPD_CONFORM_STRICT)
 	  {
 	   /*
             * Line is too long...
@@ -2906,7 +2906,7 @@ ppd_read(cups_file_t    *fp,		/* I - Fil
 	{
 	  col ++;
 
-	  if (col > (PPD_MAX_LINE - 1))
+	  if (col > (PPD_MAX_LINE - 1) && cg->ppd_conform == PPD_CONFORM_STRICT)
 	  {
 	   /*
             * Line is too long...
