diff -up cups-1.5b1/Makedefs.in.build cups-1.5b1/Makedefs.in
--- cups-1.5b1/Makedefs.in.build	2011-05-04 06:28:00.000000000 +0200
+++ cups-1.5b1/Makedefs.in	2011-05-24 15:54:03.000000000 +0200
@@ -138,7 +138,7 @@ BACKLIBS	=	@BACKLIBS@
 BANNERTOPS	=	@BANNERTOPS@
 BUILDDIRS	=	@BUILDDIRS@
 CFLAGS		=	@CPPFLAGS@ @CFLAGS@
-COMMONLIBS	=	@LIBS@
+COMMONLIBS	=	@LIBS@ $(DNSSDLIBS)
 CXXFLAGS	=	@CPPFLAGS@ @CXXFLAGS@
 CXXLIBS		=	@CXXLIBS@
 DBUS_NOTIFIER	=	@DBUS_NOTIFIER@
diff -up cups-1.5b1/scheduler/dirsvc.c.build cups-1.5b1/scheduler/dirsvc.c
--- cups-1.5b1/scheduler/dirsvc.c.build	2011-05-20 05:49:49.000000000 +0200
+++ cups-1.5b1/scheduler/dirsvc.c	2011-05-24 15:55:26.000000000 +0200
@@ -2047,7 +2047,7 @@ cupsdUpdateDNSSDName(void)
 
     WebIFRef = DNSSDRef;
     if ((error = DNSServiceRegister(&WebIFRef,
-				    kDNSServiceFlagsShareConnection,
+				    0,
 				    0, webif, "_http._tcp", NULL,
 				    NULL, htons(DNSSDPort), 7,
 				    "\006path=/", dnssdRegisterCallback,
@@ -2769,7 +2769,7 @@ dnssdRegisterPrinter(cupsd_printer_t *p)
     do
     {
       p->ipp_ref = DNSSDRef;
-      if ((se = DNSServiceRegister(&p->ipp_ref, kDNSServiceFlagsShareConnection,
+      if ((se = DNSServiceRegister(&p->ipp_ref, 0,
                                    0, name, regtype, NULL, NULL,
 				   htons(DNSSDPort), ipp_len, ipp_txt,
 				   dnssdRegisterCallback,
@@ -2866,7 +2866,7 @@ dnssdRegisterPrinter(cupsd_printer_t *p)
 
     p->printer_ref = DNSSDRef;
     if ((se = DNSServiceRegister(&p->printer_ref,
-				 kDNSServiceFlagsShareConnection,
+				 0,
 				 0, name, "_printer._tcp", NULL, NULL,
 				 htons(printer_port), printer_len, printer_txt,
 				 dnssdRegisterCallback,
