diff -uraN cups/conf/cupsd.conf.in cups-new/conf/cupsd.conf.in
--- cups/conf/cupsd.conf.in	2012-05-25 15:28:25.000000000 +0200
+++ cups-new/conf/cupsd.conf.in	2012-08-04 20:56:26.699062177 +0200
@@ -19,8 +19,6 @@

 # Show shared printers on the local network.
 Browsing On
-BrowseOrder allow,deny
-BrowseAllow all
 BrowseLocalProtocols @CUPS_BROWSE_LOCAL_PROTOCOLS@

 # Default authentication type, when authentication is required...
