diff -up cups-2.0rc1/config-scripts/cups-ssl.m4.no-gcry cups-2.0rc1/config-scripts/cups-ssl.m4
--- cups-2.0rc1/config-scripts/cups-ssl.m4.no-gcry	2014-09-12 15:41:23.324760213 +0200
+++ cups-2.0rc1/config-scripts/cups-ssl.m4	2014-09-12 15:43:13.124203363 +0200
@@ -60,7 +60,6 @@ if test x$enable_ssl != xno; then
     dnl Then look for GNU TLS...
     if test $have_ssl = 0 -a "x$enable_gnutls" != "xno" -a "x$PKGCONFIG" != x; then
     	AC_PATH_TOOL(LIBGNUTLSCONFIG,libgnutls-config)
-    	AC_PATH_TOOL(LIBGCRYPTCONFIG,libgcrypt-config)
 	if $PKGCONFIG --exists gnutls; then
 	    have_ssl=1
 	    SSLLIBS=`$PKGCONFIG --libs gnutls`
