diff -ruN cups-1.6.2.orig/config-scripts/cups-directories.m4 cups-1.6.2/config-scripts/cups-directories.m4
--- cups-1.6.2.orig/config-scripts/cups-directories.m4	2012-10-01 03:55:23.000000000 +0200
+++ cups-1.6.2/config-scripts/cups-directories.m4	2013-04-02 00:11:41.000000000 +0200
@@ -429,7 +429,7 @@
 		;;
 	*)
 		# All others
-		CUPS_STATEDIR="$localstatedir/run/cups"
+		CUPS_STATEDIR="/run/cups"
 		;;
 esac
 AC_DEFINE_UNQUOTED(CUPS_STATEDIR, "$CUPS_STATEDIR")
