diff -ruN cups-1.4.7.orig//config-scripts/cups-ssl.m4 cups-1.4.7/config-scripts/cups-ssl.m4
--- cups-1.5.0.orig//config-scripts/cups-ssl.m4	2011-01-11 08:05:58.000000000 +0100
+++ cups-1.5.0/config-scripts/cups-ssl.m4	2011-08-02 10:44:26.011047900 +0200
@@ -96,7 +96,6 @@
     dnl Then look for GNU TLS...
     if test $have_ssl = 0 -a "x$enable_gnutls" != "xno" -a "x$PKGCONFIG" != x; then
     	AC_PATH_PROG(LIBGNUTLSCONFIG,libgnutls-config)
-    	AC_PATH_PROG(LIBGCRYPTCONFIG,libgcrypt-config)
 	if $PKGCONFIG --exists gnutls; then
 	    have_ssl=1
 	    SSLLIBS=`$PKGCONFIG --libs gnutls`
@@ -108,18 +107,6 @@
 	    AC_DEFINE(HAVE_GNUTLS)
 	fi
 
-	if test $have_ssl = 1; then
-	    CUPS_SERVERCERT="ssl/server.crt"
-	    CUPS_SERVERKEY="ssl/server.key"
-
-            if $PKGCONFIG --exists gcrypt; then
-	        SSLLIBS="$SSLLIBS `$PKGCONFIG --libs gcrypt`"
-	        SSLFLAGS="$SSLFLAGS `$PKGCONFIG --cflags gcrypt`"
-	    elif test "x$LIBGCRYPTCONFIG" != x; then
-	        SSLLIBS="$SSLLIBS `$LIBGCRYPTCONFIG --libs`"
-	        SSLFLAGS="$SSLFLAGS `$LIBGCRYPTCONFIG --cflags`"
-	    fi
-	fi
     fi
 
     dnl Check for the OpenSSL library last...
--- cups-1.5.0.orig//cups/http-private.h	2011-01-22 01:07:22.000000000 +0100
+++ cups-1.5.0/cups/http-private.h	2011-08-02 10:42:43.341604107 +0200
@@ -93,7 +93,6 @@
 #  elif defined HAVE_GNUTLS
 #    include <gnutls/gnutls.h>
 #    include <gnutls/x509.h>
-#    include <gcrypt.h>
 #  elif defined(HAVE_CDSASSL)
 #    include <CoreFoundation/CoreFoundation.h>
 #    include <Security/Security.h>
