diff -up cups-1.5b1/backend/serial.c.serial cups-1.5b1/backend/serial.c
--- cups-1.5b1/backend/serial.c.serial	2011-05-20 05:49:49.000000000 +0200
+++ cups-1.5b1/backend/serial.c	2011-05-23 17:34:33.000000000 +0200
@@ -75,6 +75,7 @@
 #endif /* __APPLE__ */
 
 #if defined(__linux) && defined(TIOCGSERIAL)
+#  include <linux/types.h>
 #  include <linux/serial.h>
 #  include <linux/ioctl.h>
 #endif /* __linux && TIOCGSERIAL */
