--- src/libpng/pngpriv.h.orig	2017-01-24 01:56:17.000000000 +0100
+++ src/libpng/pngpriv.h	2018-02-23 14:33:41.864333473 +0100
@@ -105,6 +105,7 @@
  * this in $(CC), e.g. "CC=gcc -mfpu=neon", but people who build libpng rarely
  * do this.
  */
+#define PNG_ARM_NEON_OPT 0
 #ifndef PNG_ARM_NEON_OPT
    /* ARM NEON optimizations are being controlled by the compiler settings,
     * typically the target FPU.  If the FPU has been set to NEON (-mfpu=neon
