From 831e2ee8e9cf08ea1ee9736cde8370f9d0312abc Mon Sep 17 00:00:00 2001
From: Nicolas Werner <nicolas.werner@hotmail.de>
Date: Wed, 28 Jun 2023 13:01:58 +0200
Subject: [PATCH] Fix build with fmt10

---
 lib/client.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/lib/client.cpp b/lib/client.cpp
index 624f046..3666623 100644
--- a/lib/client.cpp
+++ b/lib/client.cpp
@@ -6,6 +6,7 @@
 #include <thread>
 
 #include "coeurl/request.hpp"
+#include "coeurl/errors.hpp"
 
 namespace coeurl {
 std::shared_ptr<spdlog::logger> Client::log = spdlog::null_logger_mt("coeurl_null");
@@ -347,7 +348,7 @@ void Client::remove_request(Request *r) {
         long http_code;
         curl_easy_getinfo(req->easy, CURLINFO_RESPONSE_CODE, &http_code);
 
-        Client::log->trace("DONE: {} => {} ({}) http: {}", req->url_, req->curl_error, req->error, http_code);
+        Client::log->trace("DONE: {} => {} ({}) http: {}", req->url_, coeurl::to_string(req->curl_error), req->error, http_code);
 
         if (req->on_complete_)
             req->on_complete_(*req.get());
-- 
GitLab

