--- a/Makefile
+++ b/Makefile
@@ -18,18 +18,6 @@ ifdef DPKG
 	PDEBLDFLAGS := $(shell dpkg-buildflags --get LDFLAGS)
 endif
 
-MACHINE := $(shell sh -c 'uname -m 2> /dev/null || echo not')
-ifneq (,$(findstring aarch,$(MACHINE)))
-	SIMDFLAGS := -march=armv8-a -mtune=cortex-a53
-else
-ifneq (,$(findstring arm,$(MACHINE)))
-	SIMDFLAGS := -march=armv8-a -mtune=cortex-a53 -mfpu=neon-fp-armv8 -mfloat-abi=hard
-else
-	SIMDFLAGS := -msse2
-endif
-endif
-
-
 PROGRAM = helm
 BIN     = $(DESTDIR)/usr/bin
 BINFILE = $(BIN)/$(PROGRAM)
