--- a/Makefile.in
+++ b/Makefile.in
@@ -2702,9 +2702,6 @@
 	$(am__append_40) $(am__append_43) $(am__append_45)
 EXTRA_libostree_1_la_DEPENDENCIES = $(symbol_files)
 
-# XXX: work around clang being passed -fstack-clash-protection which it doesn't understand
-# See: https://bugzilla.redhat.com/show_bug.cgi?id=1672012
-INTROSPECTION_SCANNER_ENV = CC=gcc
 @BUILDOPT_INTROSPECTION_TRUE@OSTree_1_0_gir_EXPORT_PACKAGES = ostree-1
 @BUILDOPT_INTROSPECTION_TRUE@OSTree_1_0_gir_INCLUDES = Gio-2.0
 @BUILDOPT_INTROSPECTION_TRUE@OSTree_1_0_gir_CFLAGS =  \
