From 13a2e3ecb4b6cf6aad26c6b1d2388ef6922a5c92 Mon Sep 17 00:00:00 2001
From: Akbarkhon Variskhanov <akbarkhon.variskhanov@gmail.com>
Date: Tue, 20 Dec 2022 13:50:40 +0500
Subject: [PATCH] build: Fix failure due to wrong var type (Fixes #25)

---
 panel-plugin/network.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/panel-plugin/network.cc b/panel-plugin/network.cc
index 6a85dab..3de831d 100644
--- a/panel-plugin/network.cc
+++ b/panel-plugin/network.cc
@@ -114,7 +114,7 @@ read_netload_proc (gulong *bytes)
 gint
 read_netload (gulong *net, gulong *NTotal)
 {
-    static guint64 bytes[2];
+    static gulong bytes[2];
     static gint64 time[2];
 
     *net = 0;
-- 
GitLab

