--- configure.ac.orig	2015-09-16 11:08:07.715932841 -0400
+++ configure.ac	2015-09-16 11:08:11.162932736 -0400
@@ -49,6 +49,7 @@
 AC_PROG_YACC
 PKG_PROG_PKG_CONFIG
 
+AC_GNU_SOURCE
 AC_CHECK_PROG([have_protoc_c], [protoc-c], [yes], [no])
 AC_CHECK_HEADERS([google/protobuf-c/protobuf-c.h],
                  [have_protobuf_c_h="yes"],

--- ./src/daemon/Makefile.am.orig	2015-09-16 11:08:07.715932841 -0400
+++ ./src/daemon/Makefile.am	2015-09-16 11:08:07.715932841 -0400
@@ -49,7 +49,7 @@
 collectd_CFLAGS = $(AM_CFLAGS)
 collectd_LDFLAGS = -export-dynamic
 collectd_LDADD = libavltree.la libcommon.la libheap.la -lm
-collectd_DEPENDENCIES =
+collectd_DEPENDENCIES = libavltree.la libcommon.la libheap.la
 
 # Link to these libraries..
 if BUILD_WITH_LIBRT
