--- a/src/crypto/mac/macs.c.in	2015-01-29 06:41:45.000000000 +0100
+++ b/src/crypto/mac/macs.c.in	2015-12-19 14:41:42.579958439 +0100
@@ -67,7 +67,7 @@ static inline bool mac_available(const f
 void fastd_mac_init(void) {
 	size_t i, j;
 	for (i = 0; i < array_size(macs); i++) {
-		for (j = 0; macs[i].impls[j].impl; j++) {
+		for (j = 0; macs[i].impls[j].impl == NULL; j++) {
 			if (mac_available(macs[i].impls[j].impl))
 				break;
 		}
