--- a/test/ioctl/mock.c
+++ b/test/ioctl/mock.c
@@ -114,7 +114,11 @@
 	} \
 })
 
+#if __USE_GNU
 int ioctl(int fd, unsigned long request, ...)
+#else
+int ioctl(int fd, int request, ...)
+#endif
 {
 	struct mock_cmds *mock_cmds;
 	bool result64;
@@ -141,7 +141,11 @@
 		result64 = true;
 		break;
 	default:
+#if __USE_GNU
 		fail("unexpected %s %lu", __func__, request);
+#else
+		fail("unexpected %s %u", __func__, request);
+#endif
 	}
 	check(mock_cmds->remaining_cmds,
 	      "unexpected %s command", mock_cmds->name);
