--- ./go/channelling/server/users.go.orig	2018-01-15 14:20:19.997694620 +0100
+++ ./go/channelling/server/users.go	2018-01-15 14:20:41.151902026 +0100
@@ -456,7 +456,7 @@ func (users *Users) Post(request *http.R
 	}
 
 	// Do this before session validation to avoid timing information.
-	userid := fmt.Sprintf("%s@%s", uuid.NewV4().String(), users.realm)
+	userid := fmt.Sprintf("%s@%s", uuid.Must(uuid.NewV4()).String(), users.realm)
 
 	// Make sure Sid matches session and is valid.
 	if !users.ValidateSession(snr.Id, snr.Sid) {
