From 1484c9ee0a60468dfd88954011fae0e28c0f73de Mon Sep 17 00:00:00 2001
From: Peter Jung <admin@ptr1337.dev>
Date: Fri, 18 Oct 2024 22:40:58 +0200
Subject: [PATCH 3/6] Add IBT support

Signed-off-by: Peter Jung <admin@ptr1337.dev>
---
 src/nvidia-modeset/Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/nvidia-modeset/Makefile b/src/nvidia-modeset/Makefile
index 2b96f3fa..ed934014 100644
--- a/src/nvidia-modeset/Makefile
+++ b/src/nvidia-modeset/Makefile
@@ -151,6 +151,7 @@ ifeq ($(TARGET_ARCH),x86_64)
   CONDITIONAL_CFLAGS += $(call TEST_CC_ARG, -fno-jump-tables)
   CONDITIONAL_CFLAGS += $(call TEST_CC_ARG, -mindirect-branch=thunk-extern)
   CONDITIONAL_CFLAGS += $(call TEST_CC_ARG, -mindirect-branch-register)
+  CONDITIONAL_CFLAGS += $(call TEST_CC_ARG, -mharden-sls=all)
 endif
 
 CFLAGS += $(CONDITIONAL_CFLAGS)
-- 
2.47.0

