--- configure-ng.py.orig	2014-09-11 19:36:11.000000000 +0200
+++ configure-ng.py	2014-10-21 00:47:10.948729709 +0200
@@ -1355,9 +1355,6 @@
             generate_application_makefile(target_config, verbose, tool)
             subdirs.append(tool)

-        # Generate the pyuic4 wrapper.
-        pyuic_wrapper = generate_pyuic4_wrapper(target_config)
-
     # Generate the Qt Designer plugin.
     if not target_config.no_designer_plugin and 'QtDesigner' in target_config.pyqt_modules:
         if generate_plugin_makefile(target_config, verbose, 'designer', target_config.designer_plugin_dir, "Qt Designer"):
@@ -1412,20 +1409,13 @@
 INSTALLS += init_py
 ''' % (' '.join(subdirs), source_path('__init__.py'), target_config.pyqt_module_dir))

-    # Install the uic module and the pyuic4 wrapper.
+    # Install the uic module.
     out_f.write('''
 uic_package.files = %s
 uic_package.path = %s/PyQt4
 INSTALLS += uic_package
 ''' % (source_path('pyuic', 'uic'), target_config.pyqt_module_dir))

-    if not target_config.no_tools:
-        out_f.write('''
-pyuic4.files = %s
-pyuic4.path = %s
-INSTALLS += pyuic4
-''' % (pyuic_wrapper, target_config.pyqt_bin_dir))
-
     # Install the QScintilla .api file.
     if target_config.qsci_api:
         out_f.write('''
@@ -1444,18 +1434,6 @@

     out_f.close()

-    # Make the pyuic4 wrapper executable on platforms that support it.  This
-    # means that qmake will try and strip it after installing it resulting in
-    # an (ignored) error.  However if we did it after running qmake then (on
-    # Linux) the execute bits would be stripped on installation.
-    if not target_config.no_tools and target_config.py_platform != 'win32':
-        inform("Making the %s wrapper executable..." % pyuic_wrapper)
-
-        sbuf = os.stat(pyuic_wrapper)
-        mode = sbuf.st_mode
-        mode |= (stat.S_IXUSR | stat.S_IXGRP | stat.S_IXOTH)
-        os.chmod(pyuic_wrapper, mode)
-
     # Generate the top-level makefile.
     inform("Generating the top-level Makefile...")
     run_qmake(target_config, verbose, toplevel_pro)
