https://github.com/kislyuk/yq/issues/114

diff --git a/test/test.py b/test/test.py
index f25dced..ffa7147 100755
--- a/test/test.py
+++ b/test/test.py
@@ -77,7 +77,7 @@ class TestYq(unittest.TestCase):
         unusable_tty_input = mock.Mock()
         unusable_tty_input.isatty = mock.Mock(return_value=True)
 
-        self.run_yq("{}", [], expect_exit_codes={0} if sys.stdin.isatty() else {2})
+        self.run_yq("{}", [], expect_exit_codes={0})
         self.run_yq("{}", ["."])
         self.run_yq(unusable_non_tty_input, [".", test_doc])
         self.run_yq(unusable_non_tty_input, [".", test_doc, test_doc])
