--- apps/CA.pl.in	2006-04-28 02:30:49.000000000 +0200
+++ apps/CA.pl.in	2010-04-01 00:35:02.600553509 +0200
@@ -53,7 +53,7 @@
 $X509="$openssl x509";
 $PKCS12="$openssl pkcs12";
 
-$CATOP="./demoCA";
+$CATOP="/etc/ssl";
 $CAKEY="cakey.pem";
 $CAREQ="careq.pem";
 $CACERT="cacert.pem";
--- apps/CA.sh	2009-10-15 19:27:47.000000000 +0200
+++ apps/CA.sh	2010-04-01 00:35:02.600553509 +0200
@@ -68,7 +68,7 @@
 X509="$OPENSSL x509"
 PKCS12="openssl pkcs12"
 
-if [ -z "$CATOP" ] ; then CATOP=./demoCA ; fi
+if [ -z "$CATOP" ] ; then CATOP=/etc/ssl ; fi
 CAKEY=./cakey.pem
 CAREQ=./careq.pem
 CACERT=./cacert.pem
--- apps/openssl.cnf	2009-04-04 20:09:43.000000000 +0200
+++ apps/openssl.cnf	2010-04-01 00:35:02.607220681 +0200
@@ -39,7 +39,7 @@
 ####################################################################
 [ CA_default ]
 
-dir		= ./demoCA		# Where everything is kept
+dir		= /etc/ssl		# Where everything is kept
 certs		= $dir/certs		# Where the issued certs are kept
 crl_dir		= $dir/crl		# Where the issued crl are kept
 database	= $dir/index.txt	# database index file.
