diff --git a/crypto/ppccap.c b/crypto/ppccap.c
index eeaa47c..e6eeb14 100644
--- a/crypto/ppccap.c
+++ b/crypto/ppccap.c
@@ -207,17 +207,9 @@ size_t OPENSSL_instrument_bus2(unsigned int *out, size_t cnt, size_t max)
         return 0;
 }
 
-#if defined(__GLIBC__) && defined(__GLIBC_PREREQ)
-# if __GLIBC_PREREQ(2, 16)
-#  include <sys/auxv.h>
-#  define OSSL_IMPLEMENT_GETAUXVAL
-# elif defined(__ANDROID_API__)
-/* see https://developer.android.google.cn/ndk/guides/cpu-features */
-#  if __ANDROID_API__ >= 18
-#   include <sys/auxv.h>
-#   define OSSL_IMPLEMENT_GETAUXVAL
-#  endif
-# endif
+#if defined(__linux__)
+# include <sys/auxv.h>
+# define OSSL_IMPLEMENT_GETAUXVAL
 #endif
 
 #if defined(__FreeBSD__)
