--- a/crypto/ppccap.c.old	2020-03-17 14:31:17.000000000 +0000
+++ b/crypto/ppccap.c	2020-03-30 06:32:25.943988524 +0000
@@ -207,11 +207,9 @@
         return 0;
 }
 
-#if defined(__GLIBC__) && defined(__GLIBC_PREREQ)
-# if __GLIBC_PREREQ(2, 16)
-#  include <sys/auxv.h>
-#  define OSSL_IMPLEMENT_GETAUXVAL
-# endif
+#if defined(__linux__)
+# include <sys/auxv.h>
+# define OSSL_IMPLEMENT_GETAUXVAL
 #endif
 
 /* I wish <sys/auxv.h> was universally available */
