diff --git configure configure
index c4957bb..b94faba 100755
--- configure
+++ configure
@@ -5443,41 +5443,6 @@ if test "${with_tcl+set}" = set; then :
 fi
 
 
-	# This will need to be extended in the future, but I don't think
-	# it's a good idea to stuff a shitload of random stuff in here right now
-	for path in $TCL_DIR /usr/lib /usr/lib/tcl8.4 /usr/lib/tcl8.5
-	do
-		file="${path}/tclConfig.sh"
-		{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for ${file}" >&5
-$as_echo_n "checking for ${file}... " >&6; }
-		if test -r ${file}
-		then
-			TCL_CONF=${file}
-			{ $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
-$as_echo "yes" >&6; }
-			break
-		fi
-		{ $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
-$as_echo "no" >&6; }
-	done
-
-	if test x"${TCL_CONF}" = x
-	then
-		# They --enable-tcl'd, so give them some sane default
-		TCL_FLAGS="-I/usr/include/tcl -ltcl"
-		{ $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: Could not find tclConfig.sh, using some sane defaults." >&5
-$as_echo "$as_me: WARNING: Could not find tclConfig.sh, using some sane defaults." >&2;}
-	else
-		{ $as_echo "$as_me:${as_lineno-$LINENO}: checking modtcl flags" >&5
-$as_echo_n "checking modtcl flags... " >&6; }
-		. ${TCL_CONF}
-		# eval because those vars depend on other vars in there
-		eval "TCL_LIB_SPEC=\"${TCL_LIB_SPEC}\""
-		eval "TCL_INCLUDE_SPEC=\"${TCL_INCLUDE_SPEC}\""
-		TCL_FLAGS="$TCL_INCLUDE_SPEC $TCL_LIB_SPEC"
-		{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $TCL_FLAGS" >&5
-$as_echo "$TCL_FLAGS" >&6; }
-	fi
 	my_saved_LIBS="$LIBS"
 	appendLib "$TCL_FLAGS"
 	ac_fn_cxx_check_func "$LINENO" "Tcl_CreateInterp" "ac_cv_func_Tcl_CreateInterp"
