--- a/capi/include/ResvgQt.h	2019-03-21 21:50:09.719296805 +0100
+++ b/capi/include/ResvgQt.h	2020-09-01 01:20:43.903570695 +0200
@@ -23,6 +23,7 @@
 #include <QFile>
 #include <QGuiApplication>
 #include <QPainter>
+#include <QPainterPath>
 #include <QRectF>
 #include <QScopedPointer>
 #include <QScreen>
--- a/resvg-qt/cpp/qt_capi.cpp	2019-03-21 21:50:09.719296805 +0100
+++ b/resvg-qt/cpp/qt_capi.cpp	2020-09-01 01:23:19.594578727 +0200
@@ -1,6 +1,7 @@
 #include <QGuiApplication>
 #include <QImage>
 #include <QPainter>
+#include <QPainterPath>
 #include <QDebug>
 
 #include "qt_capi.hpp"
