--- a/lldb/bindings/python/python.swig
+++ b/lldb/bindings/python/python.swig
@@ -54,7 +54,7 @@ except ImportError:
 %enddef
 // These versions will not generate working python modules, so error out early.
 #if SWIG_VERSION >= 0x030009 && SWIG_VERSION < 0x030011
-#error Swig versions 3.0.9 and 3.0.10 are incompatible with lldb.
+#warning Swig versions 3.0.9 and 3.0.10 are incompatible with lldb.
 #endif
 
 // The name of the module to be created.
