From f8b87697e8b893b6293ebbfc34670c32bfb49397 Mon Sep 17 00:00:00 2001
From: Tristan Gingold <tgingold@free.fr>
Date: Thu, 12 May 2022 18:16:38 +0200
Subject: [PATCH] ghdllocal.adb: move pragma suppress.  Fix #2056

---
 src/ghdldrv/ghdllocal.adb | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ghdldrv/ghdllocal.adb b/src/ghdldrv/ghdllocal.adb
index 4ed71a590a..756589602b 100644
--- a/src/ghdldrv/ghdllocal.adb
+++ b/src/ghdldrv/ghdllocal.adb
@@ -1259,11 +1259,11 @@ package body Ghdllocal is
             Src : String_Access;
             Dst : String_Access;
             Success : Boolean;
-            pragma Unreferenced (Success);
          begin
             Src := new String'(Image (Dir) & Basename & Get_Object_Suffix.all);
             Dst := new String'(Basename & Get_Object_Suffix.all);
             Copy_File (Src.all, Dst.all, Success, Overwrite, Full);
+            pragma Unreferenced (Success);
             --  Be silent in case of error.
             Free (Src);
             Free (Dst);
