diff --git a/configure b/configure
index c1c1a6d..dad7c38 100755
--- a/configure
+++ b/configure
@@ -280,6 +280,7 @@ if test $backend = llvm; then
        check_version  9.0 $llvm_version ||
        check_version 10.0 $llvm_version ||
        check_version 11.0 $llvm_version ||
+       check_version 12.0 $llvm_version ||
        false; then
     echo "Debugging is enabled with llvm $llvm_version"
   else
