--- a/cmake/ETLPlatform.cmake.ori	2021-03-17 09:51:48.000000000 +0100
+++ b/cmake/ETLPlatform.cmake	2021-09-13 21:57:30.274791781 +0200
@@ -320,6 +320,10 @@
 		else()
 			set(ARCH "arm64-v8a")
 		endif()
+	elseif("${CMAKE_SYSTEM_PROCESSOR}" MATCHES "ppc64*")
+		set(ARCH "ppc64")
+	elseif("${CMAKE_SYSTEM_PROCESSOR}" MATCHES "ppc*")
+		set(ARCH "ppc")
 	else()
 		set(ARCH "${CMAKE_SYSTEM_PROCESSOR}")
 		message(STATUS "Warning: processor architecture not recognised (${CMAKE_SYSTEM_PROCESSOR})")
