Instead of defining the integer types locally include <stdint.h>

--- AP_commons.h	2006-08-27 04:35:15.000000000 +0200
+++ AP_commons.h	2017-01-27 18:01:09.520651784 +0100
@@ -21,36 +21,7 @@
 
 #include <sys/types.h>
 #include <sys/stat.h>
-
-#ifndef _UINT8_T
-#define _UINT8_T
-typedef unsigned char         uint8_t;
-#endif /*_UINT8_T */
-
-#ifndef _UINT16_T
-#define _UINT16_T
-typedef unsigned short       uint16_t;
-#endif /* _UINT16_T */
-
-#ifndef _UINT32_T
-#ifndef __uint32_t_defined
-typedef unsigned int         uint32_t;
-#endif
-#endif /*_UINT32_T */
-
-#ifndef _UINT64_T
-#define _UINT64_T
-#if defined (_MSC_VER)
-typedef unsigned __int64	 uint64_t;
-#else
-typedef unsigned long long   uint64_t;
-#endif /* _MSC_VER */
-#endif /* _UINT64_T */
-
-#ifndef _INT16_T
-#define _INT16_T
-typedef short       int16_t;
-#endif /* _INT16_T */
+#include <stdint.h>
 
 #if defined (__ppc__) || defined (__ppc64__)
 #define SWAP16(x) (x)
