From 5d5245e14c6d38af7f5db3642426fe319e92926f Mon Sep 17 00:00:00 2001
From: Nick Cao <nickcao@nichi.co>
Date: Sat, 1 Jul 2023 17:49:05 +0800
Subject: [PATCH 72/72] drm/verisilicon: import DMA_BUF namespace

---
 drivers/gpu/drm/verisilicon/vs_drv.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/gpu/drm/verisilicon/vs_drv.c b/drivers/gpu/drm/verisilicon/vs_drv.c
index 1255a373f7a2..8627027047aa 100644
--- a/drivers/gpu/drm/verisilicon/vs_drv.c
+++ b/drivers/gpu/drm/verisilicon/vs_drv.c
@@ -299,4 +299,5 @@ module_init(vs_drm_init);
 module_exit(vs_drm_fini);
 
 MODULE_DESCRIPTION("VeriSilicon DRM Driver");
+MODULE_IMPORT_NS(DMA_BUF);
 MODULE_LICENSE("GPL");
-- 
2.41.0

