Explicit cast to float to avoid ambiguity.

--- es-core/src/components/HelpComponent.cpp	2016-03-13 20:31:00.000000000 +0100
+++ es-core/src/components/HelpComponent.cpp	2016-09-15 10:28:21.700078076 +0200
@@ -71,7 +71,7 @@
 	std::vector< std::shared_ptr<TextComponent> > labels;
 
 	float width = 0;
-	const float height = round(font->getLetterHeight() * 1.25f);
+	const float height = round(static_cast<float>(font->getLetterHeight() * 1.25f));
 	for(auto it = mPrompts.begin(); it != mPrompts.end(); it++)
 	{
 		auto icon = std::make_shared<ImageComponent>(mWindow);
