--- Makefile.in.orig	2009-03-08 01:47:08.023659652 +0100
+++ Makefile.in	2009-03-08 01:48:53.801346700 +0100
@@ -251,7 +251,7 @@ EXTRA_modinfo_SOURCES = backwards_compat
 MAN5 = modprobe.conf.5 modules.dep.5 depmod.conf.5
 MAN8 = depmod.8 insmod.8 lsmod.8 rmmod.8 modprobe.8 modinfo.8
 SGML = $(addprefix doc/,  $(MAN5:%.5=%.sgml) $(MAN8:%.8=%.sgml))
-man_MANS = $(MAN5) $(MAN8)
+man_MANS =
 TESTSUITE := $(shell find tests -type f ! -name '*~') tests/vg-suppressions
 EXTRA_DIST = generate-modprobe.conf modprobe.devfs FAQ CODING stress_modules.sh install-with-care $(SGML) $(man_MANS) $(TESTSUITE)
 sbin_SCRIPTS = generate-modprobe.conf
