diff --git a/tests/describe.scm b/tests/describe.scm
index 3eb5ed6..68ca621 100644
--- a/tests/describe.scm
+++ b/tests/describe.scm
@@ -74,14 +74,14 @@
                                           #:pattern "0.1")))
     (describe-format (describe-workdir repository options))))
 
-  (test-equal "describe workdir, invalid strategy"
-    #f
-    (let* ((repository (repository-open directory))
-           (oid (reference-target (repository-head repository)))
-           (commit (commit-lookup repository oid))
-           (options (make-describe-options
-                     #:strategy 'foo)))
-      (describe-format (describe-commit commit options))))
+;;  (test-equal "describe workdir, invalid strategy"
+;;    #f
+;;    (let* ((repository (repository-open directory))
+;;           (oid (reference-target (repository-head repository)))
+;;           (commit (commit-lookup repository oid))
+;;           (options (make-describe-options
+;;                     #:strategy 'foo)))
+;;      (describe-format (describe-commit commit options))))
 
   (test-equal "describe HEAD^, default options"
     "0.1"
