From 8d709c3b55dc92621d5871fb429a02c78cc4b2d8 Mon Sep 17 00:00:00 2001
From: Daniel Kolesa <daniel@octaforge.org>
Date: Wed, 16 Sep 2020 23:57:32 +0200
Subject: [PATCH] fix build

---
 src/libtracker-miner/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git src/libtracker-miner/meson.build src/libtracker-miner/meson.build
index a226623..9eacee3 100644
--- a/src/libtracker-miner/meson.build
+++ b/src/libtracker-miner/meson.build
@@ -40,7 +40,7 @@ endif
 
 libtracker_miner = library(
     'tracker-miner-' + tracker_api_version,
-    miner_enums[0], miner_enums[1], miner_sources,
+    miner_enums[0], miner_enums[1], tracker_miners_common_enum_header, miner_sources,
     c_args: tracker_c_args,
     install: true,
     install_dir: tracker_internal_libs_dir,
-- 
2.28.0

