--- a/CMakeLists.txt	2023-12-09 13:53:39.723972612 -0500
+++ b/CMakeLists.txt	2023-12-09 13:54:10.126651729 -0500
@@ -50,14 +50,7 @@
     endif()
 endif()
 
-#execute_process(COMMAND ${QT_QTPATHS_EXECUTABLE} -query QT_INSTALL_PLUGINS OUTPUT_VARIABLE PLUGINDIR OUTPUT_STRIP_TRAILING_WHITESPACE)
-execute_process(COMMAND ${QT_QTPATHS_EXECUTABLE} --plugin-dir OUTPUT_VARIABLE PLUGINDIR OUTPUT_STRIP_TRAILING_WHITESPACE)
-
-if(PLUGINDIR)
-   message(STATUS "Plugin path: ${PLUGINDIR}")
-else()
-   message(FATAL_ERROR "Could not get plugin path")
-endif()
+set(PLUGINDIR /usr/${QT6_INSTALL_PLUGINS})
 
 if(Qt6LinguistTools_FOUND)
     message(STATUS "Generating translations ...")
