--- a/daemon/sigsegv.c	2017-10-21 17:29:14.166064983 +0200
+++ b/daemon/sigsegv.c	2017-10-21 17:28:45.685063897 +0200
@@ -61,7 +61,7 @@
 # define REGFORMAT "%x"
 #endif
 
-#if defined(__arm__) || defined(__powerpc__) || defined (__ia64__) || defined (__alpha__) || defined (__FreeBSD_kernel__) || defined (__sh__)
+#if defined(__arm__) || defined(__aarch64__) || defined(__powerpc__) || defined (__ia64__) || defined (__alpha__) || defined (__FreeBSD_kernel__) || defined (__sh__)
 # define DISABLE_STACKTRACE
 #endif
 
