--- a/proxies/graph_proxy.h	2016-06-06 16:05:42.186102302 +0200
+++ b/proxies/graph_proxy.h	2016-06-06 16:06:09.741103724 +0200
@@ -28,7 +28,7 @@
 #define GRAPH_PROXY_H__61D1ED56_E33B_4F50_B45B_F520979E8AA7__INCLUDED
 
 #include "common.h"
-
+#include <unistd.h> /* pid_t */
 typedef struct graph_proxy_tag { int unused; } * graph_proxy_handle;
 
 #ifdef __cplusplus
--- a/alsapid/alsapid.h	2016-06-06 16:26:09.314165609 +0200
+++ b/alsapid/alsapid.h	2016-06-06 16:26:23.903166361 +0200
@@ -28,6 +28,7 @@
 #define ALSAPID_H__0A27F284_7538_4791_8023_0FBED929EAF3__INCLUDED
 
 #include "../common.h"
+#include <unistd.h> /* pid_t */
 
 void alsapid_compose_src_link(int alsa_client_id, char * buffer);
 void alsapid_compose_dst_link(char * buffer);
