--- common/cmd_cfgload.c
+++ common/cmd_cfgload.c
@@ -52,7 +52,7 @@ static char* read_cfgload(void)
 
 	setenv("filesize", "0");
 
-	sprintf(cmd, "fatload mmc 0:1 0x%p boot.ini", (void *)p);
+	sprintf(cmd, "load mmc 0:1 0x%p /boot/boot.ini", (void *)p);
 	run_command(cmd, 0);
 
 	filesize = getenv_ulong("filesize", 16, 0);
--- include/configs/odroidc2.h
+++ include/configs/odroidc2.h
@@ -75,6 +75,13 @@
 #define CONFIG_BOOTDELAY		1	// Seconds
 #define CONFIG_ABORTBOOT_WITH_ENTERKEY
 
+/* Void */
+#define CONFIG_CMD_FS_GENERIC
+#define CONFIG_SUPPORT_RAW_INITRD
+#define CONFIG_IDENT_STRING		" Void Linux"
+#define CONFIG_CMD_PART
+#define CONFIG_PARTITION_UUIDS
+
 /* args/envs */
 #define CONFIG_SYS_MAXARGS		64
 
-- 
2.7.1

