diff --git a/src/Makefile.am b/src/Makefile.am
index 135f2f0..ede1fc3 100644
--- src/Makefile.am
+++ src/Makefile.am
@@ -13,10 +13,12 @@ telepathy_PYTHON = \
 spec_dir = $(top_srcdir)/spec
 spec_files := $(patsubst $(spec_dir)%.xml,_generated%.py,$(wildcard $(spec_dir)/*.xml))
 
+# We leave _generated/errors.py out of here because there exists a
+# spec/errors.xml file, so that means there'll be a
+# _generated/errors.py in $(spec_files). See fd.o#32526
 BUILT_SOURCES = \
 	_generated/interfaces.py \
 	_generated/constants.py \
-	_generated/errors.py \
 	_generated/__init__.py \
 	$(spec_files)
