--- a/src/node_file-inl.h
+++ b/src/node_file-inl.h
@@ -151,9 +151,15 @@
   fields->SetValue(static_cast<size_t>(FsStatFsOffset::field),                 \
                    static_cast<NativeT>(stat))
 
-  SET_FIELD(kType, s->f_type);
+  #ifdef UV_STATFS_HAS_F_FRSIZE
+	SET_FIELD(kFrSize, s->f_bsize);
+  #else
+	SET_FIELD(kFrSize, s->f_bsize);
+  #endif
+	SET_FIELD(kBFree, s->f_bfree);
+
   SET_FIELD(kBSize, s->f_bsize);
-  SET_FIELD(kFrSize, s->f_frsize);
+  SET_FIELD(kFrSize, s->f_bsize);
   SET_FIELD(kBlocks, s->f_blocks);
   SET_FIELD(kBFree, s->f_bfree);
   SET_FIELD(kBAvail, s->f_bavail);
