--- a/ffmpeg/libavutil/ppc/util_altivec.h
+++ b/ffmpeg/libavutil/ppc/util_altivec.h
@@ -54,6 +54,15 @@
 #if HAVE_ALTIVEC
 #include <altivec.h>
 
+/* starting with gcc 8.3, the vector keyword no longer allows usage with
+ * typedefs (e.g. vector uint32_t), only with their respective basic types
+ * (e.g. vector unsigned int) - with __vector this still works and there
+ * is code relying on it
+ */
+#ifndef vector
+#define vector __vector
+#endif
+
 // used to build registers permutation vectors (vcprm)
 // the 's' are for words in the _s_econd vector
 #define WORD_0 0x00,0x01,0x02,0x03
