--- a/src/FileModule.cc
+++ b/src/FileModule.cc
@@ -65,7 +65,7 @@ void FileModule::registerUse(const std::
 	auto ext = fs::path(path).extension().generic_string();
 
 	if (boost::iequals(ext, ".otf") || boost::iequals(ext, ".ttf")) {
-		if (fs::is_regular(path)) {
+		if (fs::is_regular_file(path)) {
 			FontCache::instance()->register_font_file(path);
 		} else {
 			LOG(message_group::Error,Location::NONE,"","Can't read font with path '%1$s'",path);
--- a/features/boost.prf
+++ b/features/boost.prf
@@ -36,7 +36,7 @@ isEmpty(BOOST_LINK_FLAGS) {
       BOOST_LINK_FLAGS = -lboost_thread-mt -lboost_program_options-mt -lboost_filesystem-mt -lboost_system-mt -lboost_regex-mt
     } else {
       exists($$OPENSCAD_LIBDIR/lib/libboost*thread*) {
-        BOOST_LINK_FLAGS = -lboost_thread -lboost_program_options -lboost_filesystem -lboost_system -lboost_regex
+        BOOST_LINK_FLAGS = -lboost_thread -lboost_program_options -lboost_filesystem
       }
     }
   }
@@ -50,7 +50,7 @@ isEmpty(BOOST_LINK_FLAGS) {
       BOOST_LINK_FLAGS = -lboost_thread-mt -lboost_program_options-mt -lboost_filesystem-mt -lboost_system-mt -lboost_regex-mt
     } else {
       exists($$BOOST_DIR/lib/libboost*thread*) {
-        BOOST_LINK_FLAGS = -lboost_thread -lboost_program_options -lboost_filesystem -lboost_system -lboost_regex
+        BOOST_LINK_FLAGS = -lboost_thread -lboost_program_options -lboost_filesystem
       }
     }
   }
@@ -71,7 +71,7 @@ isEmpty(BOOST_LINK_FLAGS) {
 
 isEmpty(BOOST_LINK_FLAGS) {
   unix|macx {
-    BOOST_LINK_FLAGS = -lboost_thread -lboost_program_options -lboost_filesystem -lboost_system -lboost_regex
+    BOOST_LINK_FLAGS = -lboost_thread -lboost_program_options -lboost_filesystem
   }
 }
 
--- a/c++std.pri
+++ b/c++std.pri
@@ -26,8 +26,8 @@ macx {
 }
 
 c++std {
-  CONFIG += c++14 strict_c++
-  message("Using C++14")
+  CONFIG += c++17 strict_c++
+  message("Using C++17")
 
   *clang*: {
       # 3rd party libraries will probably violate this for a long time
