--- setup.py.orig
+++ setup.py
@@ -85,7 +85,11 @@ def is_platform_mac():
 try:
     if not _CYTHON_INSTALLED:
         raise ImportError('No supported version of Cython installed.')
-    from Cython.Distutils import build_ext as _build_ext
+    try:
+        from Cython.Distutils.old_build_ext import old_build_ext as _build_ext
+    except ImportError:
+        # Pre 0.25
+        from Cython.Distutils import build_ext as _build_ext
     cython = True
 except ImportError:
     cython = False
