--- a/checker/checker.pri	2017-05-25 10:26:10.000000000 +0200
+++ b/checker/checker.pri	2017-08-15 10:32:05.033468727 +0200
@@ -1,3 +1,4 @@
+include(../config.pri)
 
 CONFIG += qt stl c++11 exceptions console warn_on
 QT -= xml network gui widgets
--- a/checker/helper.pro	2017-05-25 10:26:10.000000000 +0200
+++ b/checker/helper.pro	2017-08-15 10:32:44.558652327 +0200
@@ -1,3 +1,4 @@
+include(../config.pri)
 
 TEMPLATE = app
 
