From: Daniel Kahn Gillmor <dkg@fifthhorseman.net>
Date: Fri, 1 Apr 2016 06:04:20 -0400
Subject: avoid looking for git info during autoreconf

---
 configure.ac | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/configure.ac b/configure.ac
index b92440c..6222be6 100644
--- a/configure.ac
+++ b/configure.ac
@@ -32,13 +32,10 @@ m4_define([mym4_version], [1.4.23])
 # the decimalized short revision number, a beta version string and a
 # flag indicating a development version (mym4_isgit).  Note that the
 # m4 processing is done by autoconf and not during the configure run.
-m4_define([mym4_revision],
-          m4_esyscmd([git rev-parse --short HEAD | tr -d '\n\r']))
+m4_define([mym4_revision],[])
 m4_define([mym4_revision_dec],
           m4_esyscmd_s([echo $((0x$(echo ]mym4_revision[|head -c 4)))]))
-m4_define([mym4_betastring],
-          m4_esyscmd_s([git describe --match 'gnupg-1.[0-9].*[0-9]' --long|\
-                        awk -F- '$3!=0{print"-beta"$3}']))
+m4_define([mym4_betastring],[])
 m4_define([mym4_isgit],m4_if(mym4_betastring,[],[no],[yes]))
 m4_define([mym4_full_version],[mym4_version[]mym4_betastring])
 
