--- src/app/main.cpp	2015-05-10 03:31:33.000000000 +0200
+++ src/app/main.cpp	2015-06-23 02:03:55.384011534 +0200
@@ -53,7 +53,9 @@
 
 #ifdef Q_OS_UNIX
 #include <signal.h>
+#if defined(__GLIBC__)
 #include <execinfo.h>
 #include "stacktrace.h"
+#endif
 #endif // Q_OS_UNIX
 
@@ -300,7 +302,7 @@
 {
     signal(SIGABRT, 0);
     signal(SIGSEGV, 0);
-#if !defined Q_OS_WIN && !defined Q_OS_HAIKU
+#if !defined Q_OS_WIN && !defined Q_OS_HAIKU && defined __GLIBC__
     std::cerr << "\n\n*************************************************************\n";
     std::cerr << "Catching SIGSEGV, please report a bug at http://bug.qbittorrent.org\nand provide the following backtrace:\n";
     std::cerr << "qBittorrent version: " << VERSION << std::endl;
@@ -319,7 +321,7 @@
 {
     signal(SIGABRT, 0);
     signal(SIGSEGV, 0);
-#if !defined Q_OS_WIN && !defined Q_OS_HAIKU
+#if !defined Q_OS_WIN && !defined Q_OS_HAIKU && defined __GLIBC__
     std::cerr << "\n\n*************************************************************\n";
     std::cerr << "Catching SIGABRT, please report a bug at http://bug.qbittorrent.org\nand provide the following backtrace:\n";
     std::cerr << "qBittorrent version: " << VERSION << std::endl;
