--- support/apachectl.in.orig	2010-03-06 18:09:11.000000000 +0100
+++ support/apachectl.in	2010-03-06 18:12:31.000000000 +0100
@@ -56,6 +56,12 @@
 # the URL to your server's mod_status status page.  If you do not
 # have one, then status and fullstatus will not work.
 STATUSURL="http://localhost:@PORT@/server-status"
+
+# Source /etc/conf.d/apache for $HTTPD setting, etc.
+if [ -r /etc/conf.d/apache ]; then
+    . /etc/conf.d/apache
+fi
+
 #
 # Set this variable to a command that increases the maximum
 # number of file descriptors allowed per child process. This is
