--- a/modules/ssl/ssl_engine_init.c.orig  2018-10-30 16:25:09.084840429 +0100
+++ b/modules/ssl/ssl_engine_init.c       2018-10-30 16:25:55.019411636 +0100
@@ -1492,7 +1492,7 @@
     X509_STORE_CTX *sctx;
     X509_STORE *store = SSL_CTX_get_cert_store(mctx->ssl_ctx);

-#if OPENSSL_VERSION_NUMBER >= 0x1010100fL
+#if OPENSSL_VERSION_NUMBER >= 0x1010100fL && !defined(LIBRESSL_VERSION_NUMBER)
     /* For OpenSSL >=1.1.1, turn on client cert support which is
      * otherwise turned off by default (by design).
      * https://github.com/openssl/openssl/issues/6933 */
