From 9785f58c2c09f62aa8968d30982d4c15ec793752 Mon Sep 17 00:00:00 2001
From: q66 <daniel@octaforge.org>
Date: Tue, 4 Feb 2020 20:55:34 +0100
Subject: [PATCH] fix build on big endian

---
 backend/genesys/low.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git backend/genesys/low.cpp backend/genesys/low.cpp
index 7937fcc..4020456 100644
--- backend/genesys/low.cpp
+++ backend/genesys/low.cpp
@@ -539,7 +539,7 @@ Image read_unshuffled_image_from_scanner(Genesys_Device* dev, const ScanSession&
     }
 
 #ifdef WORDS_BIGENDIAN
-    if (depth == 16) {
+    if (session.params.depth == 16) {
         dev->pipeline.push_node<ImagePipelineNodeSwap16BitEndian>();
     }
 #endif
-- 
2.24.0

