--- a/data/meson.build
+++ b/data/meson.build
@@ -1,4 +1,4 @@
-desktop_file = i18n.merge_file ('desktop-file',
+desktop_file = i18n.merge_file (
   type: 'desktop',
   input: configure_file(
     input: files('org.gnome.Tau.desktop.in.in'),
@@ -22,7 +22,7 @@ if desktop_file_validate.found()
     )
 endif
 
-appdata_file = i18n.merge_file ('appdata-file',
+appdata_file = i18n.merge_file (
   input: configure_file(
     input: files('org.gnome.Tau.appdata.xml.in.in'),
     output: 'org.gnome.Tau.appdata.xml.in',
