diff --git configure.in configure.in
index e5b678c..f4937d0 100644
--- a/configure.in
+++ b/configure.in
@@ -59,10 +59,6 @@ m4_define([gmodule_required_version], [2.0.0])
 m4_define([pango_pkg], [pango])
 m4_define([pango_required_version], [1.0.0])
 
-# Pangox
-m4_define([pangox_pkg], [pangox])
-m4_define([pangox_required_version], [1.0.0])
-
 # PangoFT2
 m4_define([pangoft2_pkg], [pangoft2])
 m4_define([pangoft2_required_version], [1.0.0])
@@ -349,7 +345,6 @@ PKG_CHECK_MODULES([BASE_DEPENDENCIES], [\
 gtk_pkg >= gtk_required_version \
 gdk_pkg >= gdk_required_version \
 pango_pkg >= pango_required_version \
-pangox_pkg >= pangox_required_version \
 gmodule_pkg >= gmodule_required_version \
 ])
 
@@ -794,7 +789,7 @@ LIBS="$save_LIBS"
 # CFLAGS and LIBS
 ##################################################
 
-GDKGLEXT_PACKAGES="gdk_pkg pango_pkg pangox_pkg gmodule_pkg"
+GDKGLEXT_PACKAGES="gdk_pkg pango_pkg gmodule_pkg"
 GDKGLEXT_EXTRA_CFLAGS="$GL_CFLAGS $GDKGLEXT_WIN_CFLAGS"
 GDKGLEXT_EXTRA_LIBS="$GL_LIBS $GDKGLEXT_WIN_LIBS"
 GDKGLEXT_DEP_CFLAGS="$GDKGLEXT_EXTRA_CFLAGS `$PKG_CONFIG --cflags $GDKGLEXT_PACKAGES`"
diff --git gdk/Makefile.am gdk/Makefile.am
index 700213e..09f6ba7 100644
--- a/gdk/Makefile.am
+++ b/gdk/Makefile.am
@@ -73,7 +73,6 @@ gdkglext_public_h_sources = \
 	gdkgldrawable.h		\
 	gdkglpixmap.h		\
 	gdkglwindow.h		\
-	gdkglfont.h		\
 	gdkglshapes.h		\
 	gdkglglext.h
 
diff --git gdk/gdkgl.h gdk/gdkgl.h
index f22a31c..fc80db3 100644
--- a/gdk/gdkgl.h
+++ b/gdk/gdkgl.h
@@ -33,7 +33,6 @@
 #include <gdk/gdkgldrawable.h>
 #include <gdk/gdkglpixmap.h>
 #include <gdk/gdkglwindow.h>
-#include <gdk/gdkglfont.h>
 #include <gdk/gdkglshapes.h>
 
 #endif /* __GDK_GL_H__ */
diff --git gdk/x11/Makefile.am gdk/x11/Makefile.am
index a7fb6f6..526b7cc 100644
--- a/gdk/x11/Makefile.am
+++ b/gdk/x11/Makefile.am
@@ -35,7 +35,6 @@ gdkglext_x11_c_sources = \
 	gdkgldrawable-x11.c	\
 	gdkglpixmap-x11.c	\
 	gdkglwindow-x11.c	\
-	gdkglfont-x11.c		\
 	gdkglglxext.c
 
 gdkglext_x11_headers = \
