From da26cd80362e939510fedd51c9486946a61dc159 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jan=20Christian=20Gr=C3=BCnhage?=
 <jan.christian@gruenhage.xyz>
Date: Wed, 24 Aug 2022 23:48:08 +0200
Subject: [PATCH] Use PREFIX and DESTDIR in Makefile

---
 Makefile | 25 +++++++++++++------------
 1 file changed, 13 insertions(+), 12 deletions(-)

diff --git a/Makefile b/Makefile
index dc78cfe..845e23b 100755
--- a/Makefile
+++ b/Makefile
@@ -13,20 +13,21 @@ bin2c: bin2c.c
 	$(CC) -Wall -Wextra -g -o $@ $<
 
 install: rpiboot
-	install -m 755 rpiboot /usr/bin/
-	install -d /usr/share/rpiboot
-	install -m 644 msd/bootcode.bin  /usr/share/rpiboot/
-	install -m 644 msd/bootcode4.bin /usr/share/rpiboot/
-	install -m 644 msd/start.elf  /usr/share/rpiboot/
-	install -m 644 msd/start4.elf /usr/share/rpiboot/
+	install -d $(DESTDIR)$(PREFIX)/bin
+	install -m 755 rpiboot $(DESTDIR)$(PREFIX)/bin/
+	install -d $(DESTDIR)$(PREFIX)/share/rpiboot
+	install -m 644 msd/bootcode.bin $(DESTDIR)$(PREFIX)/share/rpiboot/
+	install -m 644 msd/bootcode4.bin $(DESTDIR)$(PREFIX)/share/rpiboot/
+	install -m 644 msd/start.elf $(DESTDIR)$(PREFIX)/share/rpiboot/
+	install -m 644 msd/start4.elf $(DESTDIR)$(PREFIX)/share/rpiboot/
 
 uninstall:
-	rm -f /usr/bin/rpiboot
-	rm -f /usr/share/rpiboot/bootcode.bin
-	rm -f /usr/share/rpiboot/bootcode4.bin
-	rm -f /usr/share/rpiboot/start.elf
-	rm -f /usr/share/rpiboot/start4.elf
-	rmdir --ignore-fail-on-non-empty /usr/share/rpiboot/
+	rm -f $(DESTDIR)$(PREFIX)/bin/rpiboot
+	rm -f $(DESTDIR)$(PREFIX)/share/rpiboot/bootcode.bin
+	rm -f $(DESTDIR)$(PREFIX)/share/rpiboot/bootcode4.bin
+	rm -f $(DESTDIR)$(PREFIX)/share/rpiboot/start.elf
+	rm -f $(DESTDIR)$(PREFIX)/share/rpiboot/start4.elf
+	rmdir --ignore-fail-on-non-empty $(DESTDIR)$(PREFIX)/share/rpiboot/
 
 clean:
 	rm -f rpiboot msd/*.h bin2c
-- 
2.37.2

