--- avidemux_core/ADM_core/src/ADM_crashdump_unix.cpp	2015-06-10 07:42:47.000000000 +0200
+++ avidemux_core/ADM_core/src/ADM_crashdump_unix.cpp	2015-09-28 08:15:19.568394743 +0200
@@ -31,8 +31,10 @@
 #endif
 #include <sys/machelf.h>
 #else
+#if defined(__GLIBC__)
 #include <execinfo.h>
 #endif
+#endif
 
 #include "ADM_default.h"
 
@@ -166,6 +168,7 @@
 	if(mysaveFunction)
 		mysaveFunction();
 
+#if defined(__GLIBC__)
 #if !defined(__HAIKU__)
 	char wholeStuff[2048];
 #if !defined(__sun__)
@@ -209,7 +212,7 @@
 	backtrace(fileno(stdout));
 #endif
 	printf("*********** BACKTRACK **************\n");
-
+#endif
 	if(myFatalFunction)
 		myFatalFunction("Crash", wholeStuff); // FIXME
 
