--- a/avidemux_core/ADM_core/src/ADM_crashdump_unix.cpp	2016-03-05 09:22:54.000000000 +0100
+++ b/avidemux_core/ADM_core/src/ADM_crashdump_unix.cpp	2016-03-08 03:06:07.328943375 +0100
@@ -30,7 +30,7 @@
 #define	_ELF64
 #endif
 #include <sys/machelf.h>
-#else
+#elif defined(__GLIBC__)
 #include <execinfo.h>
 #endif
 
@@ -165,7 +165,7 @@
 	if(mysaveFunction)
 		mysaveFunction();
 #define MAX_BACKTRACK 30
-#if !defined(__HAIKU__) && !defined(__sun__)
+#if defined(__GLIBC__) && !defined(__HAIKU__) && !defined(__sun__)
     char wholeStuff[2048];
     char buffer[4096];
     char in[2048];
