--- a/avidemux_core/ADM_core/src/ADM_folder_linux.cpp	2016-11-19 08:58:39.000000000 +0100
+++ b/avidemux_core/ADM_core/src/ADM_folder_linux.cpp	2016-11-21 07:05:07.700699647 +0100
@@ -39,7 +39,7 @@
 static std::string canonize(const std::string &in)
 {
     std::string out;
-    char *simple2=canonicalize_file_name(in.c_str());
+    char *simple2=realpath(in.c_str(), NULL);
     if(simple2)
     {
         out=std::string(simple2)+std::string("/");
