From cd7de12f2ebe5a2ad03c2086da4c3b5317a4506e Mon Sep 17 00:00:00 2001
From: Chih-Hsuan Yen <yan12125@gmail.com>
Date: Sun, 15 Dec 2019 09:32:55 +0800
Subject: [PATCH] Fix building with Qt 5.14

---
 lxqt-config-input/touchpaddevice.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lxqt-config-input/touchpaddevice.cpp b/lxqt-config-input/touchpaddevice.cpp
index 31adbe5f..41690c2c 100644
--- lxqt-config-input/touchpaddevice.cpp
+++ lxqt-config-input/touchpaddevice.cpp
@@ -21,6 +21,7 @@
 #include <cmath>
 #include <QDebug>
 #include <QX11Info>
+#include <QUrl>
 #include <libudev.h>
 #include <LXQt/Settings>
 #include <X11/Xatom.h>
