diff --git spamc/configure spamc/configure
index d8e5dcf..3db5772 100755
--- a/spamc/configure
+++ b/spamc/configure
@@ -1737,10 +1737,10 @@ if test "$cross_compiling" != yes; then
 	{ { echo "$as_me:$LINENO: error: cannot run C compiled programs.
 If you meant to cross compile, use \`--host'.
 See \`config.log' for more details." >&5
-echo "$as_me: error: cannot run C compiled programs.
+echo "$as_echo: cannot run C compiled programs.
 If you meant to cross compile, use \`--host'.
 See \`config.log' for more details." >&2;}
-   { (exit 1); exit 1; }; }
+    }
     fi
   fi
 fi
