From 20d5fc3b99675b64d40e3815b1fe635a5e32cdd3 Mon Sep 17 00:00:00 2001
From: fl0pp5 <fl0pp5@altlinux.org>
Date: Fri, 19 Apr 2024 18:47:59 +0300
Subject: [PATCH 1/2] fix invalid link to Data directory & remove link to
 Credits.txt

---
 Resources/cccp.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Resources/cccp.sh b/Resources/cccp.sh
index f1fd0aabe..8ded4ae25 100644
--- a/Resources/cccp.sh
+++ b/Resources/cccp.sh
@@ -5,7 +5,7 @@ link_base_files() {
         exit 1
     fi
 
-    ln -s "${base_data_path}"/*.rte "${base_data_path}/Credits.txt" $tmp_dir
+	ln -s "${base_data_path}"/Data "$tmp_dir"
 }
 
 link_user_files() {

From 07a9105ce67b771de4f45effbecc331b5db221bf Mon Sep 17 00:00:00 2001
From: HeliumAnt <lesgoets@gmail.com>
Date: Sat, 20 Apr 2024 16:16:23 +0200
Subject: [PATCH 2/2] update remaining userfiles

---
 Resources/cccp.sh | 18 +++---------------
 1 file changed, 3 insertions(+), 15 deletions(-)

diff --git a/Resources/cccp.sh b/Resources/cccp.sh
index 8ded4ae25..abef3c1db 100644
--- a/Resources/cccp.sh
+++ b/Resources/cccp.sh
@@ -5,12 +5,12 @@ link_base_files() {
         exit 1
     fi
 
-	ln -s "${base_data_path}"/Data "$tmp_dir"
+    ln -s "${base_data_path}"/Data "$tmp_dir"
 }
 
 link_user_files() {
-    local user_files=("LogConsole.txt" "LogLoading.txt" "LogLoadingWarning.txt" "AbortScreen.bmp" "AbortScreen.png" "Settings.ini")
-    local user_directories=("Metagames.rte" "Scenes.rte" "_ScreenShots")
+    local user_files=("LogConsole.txt" "LogLoading.txt" "LogLoadingWarning.txt" "AbortLog.txt" "AbortScreen.bmp" "AbortScreen.png")
+    local user_directories=("Mods" "Userdata" "ScreenShots")
 
     if ! [[ -d "${user_data}" ]]; then
         mkdir -p "${user_data}"
@@ -20,16 +20,6 @@ link_user_files() {
         ln -s "${user_data}/$file" $tmp_dir
     done
 
-    if ! [[ -d "${user_data}/Metagames.rte" ]]; then
-        mkdir -p "${user_data}/Metagames.rte"
-        echo -e "DataModule\n\tModuleName = Metagame Saves" > "${user_data}/Metagames.rte/Index.ini"
-    fi
-
-    if ! [[ -d "${user_data}/Scenes.rte" ]]; then
-        mkdir -p "${user_data}/Scenes.rte"
-        echo -e "DataModule\n\tModuleName = Saves" > "${user_data}/Scenes.rte/Index.ini"
-    fi
-
     for directory in ${user_directories[@]}; do
         if ! [[ -d "${user_data}/${directory}" ]]; then
             mkdir -p "${user_data}/${directory}"
@@ -51,8 +41,6 @@ link_user_files
 
 cd "${tmp_dir}"
 
-export CCCP_SETTINGSPATH="Settings.ini"
-
 "@EXEPATH@/@EXENAME@" $@
 
 exit_code=$?

