From 5580df46df9fe48a960474934c82d53dcf070364 Mon Sep 17 00:00:00 2001
From: Naz <ndpm13@ch-naseem.com>
Date: Fri, 17 Apr 2026 16:32:35 +0100
Subject: [PATCH] build: don't output "none required" to LIBM

---
 configure.ac | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/configure.ac b/configure.ac
index b03cf24..1ab4897 100644
--- a/configure.ac
+++ b/configure.ac
@@ -253,10 +253,10 @@ AC_DEFINE_UNQUOTED([HAVE_ASAN], [$HAVE_ASAN], [Define to 1 if ASAN is enabled])
 
 # Link libm on platforms where trunc() lives outside libc.
 AC_SEARCH_LIBS([trunc], [m])
-LIBM="$ac_cv_search_trunc"
-if test "x$LIBM" = "xno" || test "x$LIBM" = "xyes" ; then
-	LIBM=""
-fi
+case "$ac_cv_search_trunc" in
+	no|yes|"none required") LIBM="" ;;
+	*) LIBM="$ac_cv_search_trunc"
+esac
 AC_SUBST(LIBM)
 
 # check if there's a qsort_r available (musl does not have it)
-- 
2.53.0

