Adapted from Alpine Linux.

--- a/tools/f2fs_io/f2fs_io.c
+++ b/tools/f2fs_io/f2fs_io.c
@@ -40,6 +40,7 @@
 #include <sys/xattr.h>
 #define _GNU_SOURCE
 #include <dirent.h>
+#include <linux/falloc.h>
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
@@ -946,6 +947,13 @@
 	do_write_with_advice(argc, argv, cmd, true);
 }
 
+#ifndef MADV_POPULATE_READ
+#  if defined(__arm__) && !defined(__aarch64__)
+#    define MADV_POPULATE_READ  22
+#    define MADV_POPULATE_WRITE 23
+#  endif
+#endif
+
 #define read_desc "read data from file"
 #define read_help					\
 "f2fs_io read [chunk_size in 4kb] [offset in chunk_size] [count] [IO] [advice] [print_nbytes] [file_path]\n\n"	\
