--- cpp/include/IceUtil/Mutex.h.orig	2016-03-27 17:49:17.579001730 +0200
+++ cpp/include/IceUtil/Mutex.h	2016-03-27 17:51:56.591989829 +0200
@@ -231,7 +231,7 @@ Mutex::init(MutexProtocol
     // Enable mutex error checking in debug builds
     //
 #ifndef NDEBUG
-#if defined(__linux) && !defined(__USE_UNIX98)
+#if defined(__linux) && !defined(__USE_UNIX98) && defined(PTHREAD_MUTEX_ERRORCHECK_NP)
     rc = pthread_mutexattr_settype(&attr, PTHREAD_MUTEX_ERRORCHECK_NP);
 #else
     rc = pthread_mutexattr_settype(&attr, PTHREAD_MUTEX_ERRORCHECK);
