--- cpp/src/IceUtil/FileUtil.cpp.orig	2012-05-24 11:42:01.584952006 +0200
+++ cpp/src/IceUtil/FileUtil.cpp	2012-05-24 11:42:13.839655835 +0200
@@ -13,6 +13,7 @@
 #include <IceUtil/Exception.h>
 #include <climits>
 #include <string.h>
+#include <unistd.h>
 
 #ifdef _WIN32
 #  include <process.h>
--- cpp/src/Slice/FileTracker.cpp.orig	2012-05-24 11:43:53.797240067 +0200
+++ cpp/src/Slice/FileTracker.cpp	2012-05-24 11:44:09.908850687 +0200
@@ -9,9 +9,7 @@
 
 #include <Slice/FileTracker.h>
 
-#ifdef __sun
-#   include <unistd.h>
-#endif
+#include <unistd.h>
 
 #ifdef _WIN32
 #   include <direct.h>
