--- sysdeps/unix/sysv/linux/getpagesize.c.orig	2010-05-08 17:30:11.713092813 +0200
+++ sysdeps/unix/sysv/linux/getpagesize.c	2010-05-08 17:30:40.031108986 +0200
@@ -28,10 +28,6 @@
 int
 __getpagesize ()
 {
-#ifdef __ASSUME_AT_PAGESIZE
-  assert (GLRO(dl_pagesize) != 0);
-  return GLRO(dl_pagesize);
-#else
   if (GLRO(dl_pagesize) != 0)
     return GLRO(dl_pagesize);
 
@@ -47,7 +43,6 @@ __getpagesize ()
   return NBPC;
 #  endif	/* NBPG.  */
 # endif	/* EXEC_PAGESIZE.  */
-#endif
 }
 libc_hidden_def (__getpagesize)
 weak_alias (__getpagesize, getpagesize)
