diff -Naur glibc-old//sysdeps/i386/Makefile glibc//sysdeps/i386/Makefile
--- sysdeps/i386/Makefile.orig	2010-03-18 11:52:30.000000000 +1000
+++ sysdeps/i386/Makefile	2010-04-16 15:05:50.000000000 +1000
@@ -1,6 +1,7 @@
 # The mpn functions need a #define for asm syntax flavor.
-# Every i386 port in use uses gas syntax (I think).
-asm-CPPFLAGS += -DGAS_SYNTAX
+# Every i386 port in use uses gas syntax (I think).  Don't replace
+# __i686 in __i686.get_pc_thunk.bx.
+asm-CPPFLAGS += -DGAS_SYNTAX  -U __i686
 
 # The i386 `long double' is a distinct type we support.
 long-double-fcts = yes
