--- misc/sys/xattr.h	2013-08-11 08:52:55.000000000 +1000
+++ misc/sys/xattr.h	2014-01-07 15:45:50.533969040 +1000
@@ -26,13 +26,8 @@
 
 /* The following constants should be used for the fifth parameter of
    `*setxattr'.  */
-enum
-{
-  XATTR_CREATE = 1,	/* set value, fail if attr already exists.  */
-#define XATTR_CREATE	XATTR_CREATE
-  XATTR_REPLACE = 2	/* set value, fail if attr does not exist.  */
-#define XATTR_REPLACE	XATTR_REPLACE
-};
+#define XATTR_CREATE	1
+#define XATTR_REPLACE	2
 
 /* Set the attribute NAME of the file pointed to by PATH to VALUE (which
    is SIZE bytes long).  Return 0 on success, -1 for errors.  */
