From 32cf40699346d37fabfa887bbd95e95004799ae1 Mon Sep 17 00:00:00 2001
From: Andreas Schwab <schwab@redhat.com>
Date: Mon, 6 Sep 2010 14:55:59 +0200
Subject: [PATCH 1/1] Don't mix pattern rules with normal rules

---
 ChangeLog       |    4 ++++
 manual/Makefile |    5 ++++-
 2 files changed, 8 insertions(+), 1 deletions(-)

diff --git a/manual/Makefile b/manual/Makefile
index c5866eb..b1f5fa7 100644
--- manual/Makefile
+++ manual/Makefile
@@ -232,7 +232,10 @@ ifdef objpfx
 .PHONY: stubs
 stubs: $(objpfx)stubs
 endif
-$(objpfx)stubs ../po/manual.pot $(objpfx)stamp%:
+$(objpfx)stubs ../po/manual.pot:
+	$(make-target-directory)
+	touch $@
+$(objpfx)stamp%:
 	$(make-target-directory)
 	touch $@
 
-- 
1.7.2

