--- a/dutil.h
+++ b/dutil.h
@@ -28,6 +28,10 @@
 #define __pure __attribute__ ((pure))
 #endif
 
+#ifndef __always_inline
+#define __always_inline inline
+#endif
+
 #define roundup(x,y) ((((x) + ((y) - 1)) / (y)) * (y))
 
 static inline __attribute__((const)) bool is_power_of_2(unsigned long n)
--- a/lib/bpf/src/libbpf.c
+++ b/lib/bpf/src/libbpf.c
@@ -10,6 +10,10 @@
  * Copyright (C) 2019 Isovalent, Inc.
  */

+#ifndef __always_inline
+#define __always_inline inline
+#endif
+
 #ifndef _GNU_SOURCE
 #define _GNU_SOURCE
 #endif
