diff --git dwarves.c dwarves.c
index 8cb359f..98b8b6b 100644
--- a/dwarves.c
+++ b/dwarves.c
@@ -22,6 +22,7 @@
 #include <unistd.h>
 #include <sys/stat.h>
 #include <sys/utsname.h>
+#include <limits.h>
 
 #include "config.h"
 #include "list.h"
diff --git pahole.c pahole.c
index dac7a49..15c0c29 100644
--- a/pahole.c
+++ b/pahole.c
@@ -10,6 +10,7 @@
 #include <assert.h>
 #include <stdio.h>
 #include <dwarf.h>
+#include <limits.h>
 #include <inttypes.h>
 #include <search.h>
 #include <stdarg.h>
