diff --git a/conf/glances.conf b/conf/glances.conf
index 153ff3d..ef0a7ff 100644
--- a/conf/glances.conf
+++ b/conf/glances.conf
@@ -8,7 +8,7 @@
 # It is also possible to overwrite it in each plugin sections
 refresh=2
 # Does Glances should check if a newer version is available on PyPI ?
-check_update=true
+check_update=false
 # History size (maximum number of values)
 # Default is 1200 values (~1h with the default refresh rate)
 history_size=1200
