--- cmake/ConkyPlatformChecks.cmake.orig	2015-09-09 16:43:45.145815169 -0400
+++ cmake/ConkyPlatformChecks.cmake	2015-09-09 16:44:03.281814616 -0400
@@ -105,7 +105,7 @@
 if(BUILD_IPV6)
 	find_file(IF_INET6 if_inet6 PATHS /proc/net)
 	if(NOT IF_INET6)
-		message(FATAL_ERROR "/proc/net/if_inet6 unavailable")
+		message(WARNING "/proc/net/if_inet6 unavailable")
 	endif(NOT IF_INET6)
 endif(BUILD_IPV6)
 
