From cb7e05c703331dd38613dad8304ed49581853abe Mon Sep 17 00:00:00 2001
From: q66 <daniel@octaforge.org>
Date: Wed, 10 Jun 2020 21:12:43 +0200
Subject: [PATCH] e_util_suid: need signal.h

---
 src/bin/e_util_suid.h | 1 +
 1 file changed, 1 insertion(+)

diff --git src/bin/e_util_suid.h src/bin/e_util_suid.h
index e09091146..be961a520 100644
--- src/bin/e_util_suid.h
+++ src/bin/e_util_suid.h
@@ -28,6 +28,7 @@
 #endif
 #include <sys/types.h>
 #include <pwd.h>
+#include <signal.h>
 #ifdef HAVE_PRCTL
 # include <sys/prctl.h>
 #elif defined(HAVE_PROCCTL)
-- 
2.27.0

