--- a/tests/linux/tunnel_test
+++ b/tests/linux/tunnel_test
@@ -17,9 +17,9 @@
 	atf_check -s exit:0 \
 		-o match:"ip -4 tunnel add tun0" \
 		-o match:"mode gre" \
-		-o match:"ttl '255'" \
-		-o match:"local '1.2.3.4'" \
-		-o match:"remote '5.6.7.8'" \
+		-o match:"ttl 255" \
+		-o match:"local 1.2.3.4" \
+		-o match:"remote 5.6.7.8" \
 		${EXECUTOR}
 }
 
@@ -39,8 +39,8 @@
 	atf_check -s exit:0 \
 		-o match:"ip -4 link add foo" \
 		-o match:"type gretap" \
-		-o match:"local '1.2.3.4'" \
-		-o match:"remote '5.6.7.8'" \
+		-o match:"local 1.2.3.4" \
+		-o match:"remote 5.6.7.8" \
 		${EXECUTOR}
 }
 
@@ -58,8 +58,8 @@
 	atf_check -s exit:0 \
 		-o match:"ip -6 link add foo" \
 		-o match:"type gretap" \
-		-o match:"local '2001:db8::aaaa'" \
-		-o match:"remote '2001:db8::eeee'" \
+		-o match:"local 2001:db8::aaaa" \
+		-o match:"remote 2001:db8::eeee" \
 		${EXECUTOR}
 }
 
