--- a/src/cmarkgfm/build_cmark.py
+++ b/src/cmarkgfm/build_cmark.py
@@ -72,11 +72,11 @@ else:
 
 ffibuilder = cffi.FFI()
 ffibuilder.cdef(CMARK_DEF_H)
-ffibuilder.set_source(
+ffibuilder.set_source_pkgconfig(
     'cmarkgfm._cmark',
+    ['libcmark-gfm'],
     CMARK_MODULE_H,
-    sources=SOURCES,
-    include_dirs=[SRC_DIR, EXTENSIONS_SRC_DIR, GENERATED_SRC_DIR],
+    include_dirs=[GENERATED_SRC_DIR],
     extra_compile_args=EXTRA_COMPILE_ARGS
 )
 
