$NetBSD: patch-ac,v 1.1.1.1 2002/10/03 19:25:04 wiz Exp $

--- a/autoreconf.sh	Tue Jan  5 14:28:40 1999
+++ b/autoreconf.sh
@@ -143,7 +143,7 @@ while read dir; do
 	:
      else
 	test $verbose = yes && echo running aclocal in $dir, creating $aclocal
-	aclocal --output=$aclocal -I `echo $aclocal | sed 's,/*[^/]*$,,;s,^$,.,'`
+	aclocal-1.4 --output=$aclocal -I `echo $aclocal | sed 's,/*[^/]*$,,;s,^$,.,'`
      fi
   fi
 
@@ -154,7 +154,7 @@ while read dir; do
      amforce=
      test $force = no && amforce=--no-force
      test $verbose = yes && echo running automake`test x"$amforce" = x || echo " ($amforce)"` in $dir
-     automake $amforce $automake_mode $automake_deps
+     automake-1.4 $amforce $automake_mode $automake_deps
   fi
 
   test ! -f $aclocal && aclocal=
