--- Makefile	2010-07-21 21:33:13.000000000 +0200
+++ Makefile	2010-07-22 11:14:03.325522937 +0200
@@ -54,7 +54,7 @@
 # files that depend only on the B phase, but may have to be regenerated
 # for "make installer".
 BSUBDIRS = codepage com32 lzo core memdisk mbr memdump gpxe sample \
-	   diag libinstaller dos win32 win64 dosutil txt
+	   libinstaller txt
 ITARGET  =
 IOBJECTS = $(ITARGET) \
 	utils/gethostip utils/isohybrid utils/mkdiskimage \
