diff -ur a/configure b/configure
--- a/configure
+++ b/configure
@@ -5952,6 +5952,14 @@
 printf "%s\n" "$ac_cv_path_EGREP" >&6; }
  EGREP="$ac_cv_path_EGREP"
 
+if test x$MULTIARCH = xpowerpc-linux-musl
+then
+	MULTIARCH="powerpc-linux-gnu"
+fi
+if test x$MULTIARCH = xpowerpcle-linux-musl
+then
+	MULTIARCH="powerpcle-linux-gnu"
+fi
 
 
 { printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking for CC compiler name" >&5
@@ -6859,7 +6867,11 @@
         powerpc64-linux-gnu
 #  endif
 # elif defined(__powerpc__)
+#  if defined(__LITTLE_ENDIAN__)
+        powerpcle-linux-gnu
+#  else
         powerpc-linux-gnu
+#  endif
 # elif defined(__s390x__)
         s390x-linux-gnu
 # elif defined(__s390__)
diff -ur a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -1039,7 +1039,11 @@
         powerpc64-linux-gnu
 #  endif
 # elif defined(__powerpc__)
+#  if defined(__LITTLE_ENDIAN__)
+        powerpcle-linux-gnu
+#  else
         powerpc-linux-gnu
+#  endif
 # elif defined(__s390x__)
         s390x-linux-gnu
 # elif defined(__s390__)
