--- a/src/Makefile.in	2021-01-18 20:34:19.338872822 +0700
+++ b/src/Makefile.in	2021-01-18 20:35:11.314028147 +0700
@@ -93,13 +93,13 @@
 	$(LN) -s @sharedname@ $(SHAREDLIB)
 
 install-headers:
-	install -d $(DESTDIR)$(PREFIX)$(INCLUDEDIR)/stk
-	cp -R ../include/*.h $(DESTDIR)$(PREFIX)$(INCLUDEDIR)/stk
+	install -d $(DESTDIR)$(INCLUDEDIR)/stk
+	cp -R ../include/*.h $(DESTDIR)$(INCLUDEDIR)/stk
 
 install: $(SHAREDLIB) install-headers
-	install -d $(DESTDIR)$(PREFIX)$(LIBDIR)
-	install -m 644 @sharedname@ $(DESTDIR)$(PREFIX)$(LIBDIR)
-	ln -sf @sharedname@ $(DESTDIR)$(PREFIX)$(LIBDIR)/$(SHAREDLIB)
+	install -d $(DESTDIR)$(LIBDIR)
+	install -m 644 @sharedname@ $(DESTDIR)$(LIBDIR)
+	ln -sf @sharedname@ $(DESTDIR)$(LIBDIR)/$(SHAREDLIB)
 
 
 $(OBJECTS) : Stk.h
