--- a/cmake/BuildTypes.cmake
+++ b/cmake/BuildTypes.cmake
@@ -49,7 +49,7 @@
     )
     list(APPEND KNOWN_BUILD_TYPES ${build})
 endforeach()
-list(APPEND KNOWN_BUILD_TYPES Debug Release RelWithDebInfo MinSizeRel)
+list(APPEND KNOWN_BUILD_TYPES Debug Release None RelWithDebInfo MinSizeRel)
 
 if (NOT CMAKE_BUILD_TYPE IN_LIST KNOWN_BUILD_TYPES)
     message(FATAL_ERROR "Unknown build type: ${CMAKE_BUILD_TYPE}. Choices are ${KNOWN_BUILD_TYPES}")
