Index: vigra-1.11.1/include/vigra/stdconvolution.hxx
===================================================================
--- vigra-1.11.1.orig/include/vigra/stdconvolution.hxx
+++ vigra-1.11.1/include/vigra/stdconvolution.hxx
@@ -792,7 +792,9 @@ public:
         {}
 
         ~InitProxy()
-#ifndef _MSC_VER
+#if __cplusplus >= 201700L
+            noexcept(false)
+#elif !defined(_MSC_VER)
             throw(PreconditionViolation)
 #elif _MSC_VER >= 1900
             noexcept(false)
Index: vigra-1.11.1/include/vigra/separableconvolution.hxx
===================================================================
--- vigra-1.11.1.orig/include/vigra/separableconvolution.hxx
+++ vigra-1.11.1/include/vigra/separableconvolution.hxx
@@ -1409,7 +1409,9 @@ class Kernel1D
         {}
 
         ~InitProxy()
-#ifndef _MSC_VER
+#if __cplusplus >= 201700L
+            noexcept(false)
+#elif !defined(_MSC_VER)
             throw(PreconditionViolation)
 #elif _MSC_VER >= 1900
             noexcept(false)
