--- icedtea-2.6.17.orig/acinclude.m4	2019-05-30 09:49:34.952140696 +0300
+++ icedtea-2.6.17/acinclude.m4	2019-05-30 09:50:28.231483929 +0300
@@ -2271,8 +2271,8 @@
     dnl Check for syscalls
     AC_CHECK_FUNCS([openat64 fstatat64 fgetxattr fsetxattr fremovexattr flistxattr unlinkat renameat futimesat fdopendir epoll_create epoll_ctl epoll_wait],,
       [AC_MSG_ERROR([Could not find required syscalls; check config.log and use --disable-compile-against-syscalls if necessary.])])
-    AC_CHECK_HEADERS([sys/epoll.h attr/xattr.h],
-      , [AC_MSG_ERROR([Could not find required system headers; install the appropriate files from glibc-headers, libc6-dev and/or libattr-devel, libattr1-dev or use --disable-compile-against-syscalls if necessary.])])
+    AC_CHECK_HEADERS([sys/epoll.h sys/xattr.h],
+      , [AC_MSG_ERROR([Could not find required system headers; install the appropriate files from glibc-headers or libc6-dev or use --disable-compile-against-syscalls if necessary.])])
     ENABLE_SYSCALL_COMPILATION=true
   fi
   AM_CONDITIONAL(USE_SYSCALL_COMPILATION, test x"${ENABLE_SYSCALL_COMPILATION}" = "xtrue")
