--- a/configure.ac	2015-06-21 03:57:51.000000000 +0200
+++ b/configure.ac	2017-09-11 16:41:33.565358208 +0200
@@ -120,9 +120,11 @@
   AC_HELP_STRING(--with-unicode-version=VERSION,
   [version of Unicode Standard @<:@default=current version@:>@]),
   UNICODE_VERSION=$withval,UNICODE_VERSION=$DEFAULT_UNICODE_VERSION)
-AC_CHECK_FILE([lib/$UNICODE_VERSION.c],
-  AC_MSG_RESULT($UNICODE_VERSION),
-  AC_MSG_ERROR(Unknown Unicode version $UNICODE_VERSION.))
+if test -f "lib/$UNICODE_VERSION.c"; then
+  AC_MSG_RESULT($UNICODE_VERSION)
+else
+  AC_MSG_ERROR(Unknown Unicode version $UNICODE_VERSION.)
+fi
 AC_SUBST(UNICODE_VERSION)
 
 # check if code to debug memory allocation is enabled.
