diff --git a/src/log.h b/src/log.h
index 6d953d0..63e489d 100644
--- a/src/log.h
+++ b/src/log.h
@@ -4,20 +4,8 @@
 
 #include "config.h"
 
-// If glog is not installed on the compiling machine,
-// then these macros will do nothing.
-#if GLOG_FOUND
-#include <glog/logging.h>
-#define WM_INIT_LOGGING(executable_name) google::InitGoogleLogging(executable_name)
-#define WM_LOG(severity, msg)                \
-  do {                                       \
-    LOG(severity) << msg;                    \
-    google::FlushLogFiles(google::severity); \
-  } while (0)
-#else
 #define WM_INIT_LOGGING(executable_name)
 #define WM_LOG(severity, msg)
-#endif
 
 
 #define WM_LOG_WITH_ERRNO(reason, errno)               \
