--- a/src/log.h
+++ b/src/log.h
@@ -6,7 +6,6 @@
 
 // If glog is not installed on the compiling machine,
 // then these macros will do nothing.
-#if GLOG_FOUND
 #include <glog/logging.h>
 #define WM_INIT_LOGGING(executable_name) google::InitGoogleLogging(executable_name)
 #define WM_LOG(severity, msg)                \
@@ -14,10 +13,6 @@
     LOG(severity) << msg;                    \
     google::FlushLogFiles(google::severity); \
   } while (0)
-#else
-#define WM_INIT_LOGGING(executable_name)
-#define WM_LOG(severity, msg)
-#endif
 
 
 #define WM_LOG_WITH_ERRNO(reason, errno)               \
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -18,7 +18,7 @@
 endif()
 
 find_package(X11 REQUIRED)
-find_package(glog)
+find_package (glog CONFIG REQUIRED)
 
 # CMake will generate config.h from config.h.in
 include_directories("src")
@@ -43,10 +43,6 @@
   src/workspace.cc
 )
 
-set(LINK_LIBRARIES X11)
-if (GLOG_FOUND)
-  set(LINK_LIBRARIES ${LINK_LIBRARIES} glog)
-endif()
-target_link_libraries(${PROJECT_NAME} ${LINK_LIBRARIES})
+target_link_libraries(${PROJECT_NAME} X11 glog::glog)
 
 install(TARGETS ${PROJECT_NAME} DESTINATION bin)
