diff --git Tests/ConfigSources/CMakeLists.txt Tests/ConfigSources/CMakeLists.txt
index a3d98f685f..84ef26b0fd 100644
--- Tests/ConfigSources/CMakeLists.txt
+++ Tests/ConfigSources/CMakeLists.txt
@@ -75,7 +75,7 @@ add_custom_command(APPEND
   )
 foreach(n RANGE 1 5)
   set_property(SOURCE custom${n}_Debug.cpp PROPERTY COMPILE_DEFINITIONS CUSTOM_CFG_DEBUG)
-  foreach(other Release RelWithDebInfo MinSizeRel)
+  foreach(other None Release RelWithDebInfo MinSizeRel)
     set_property(SOURCE custom${n}_${other}.cpp PROPERTY COMPILE_DEFINITIONS CUSTOM_CFG_OTHER)
   endforeach()
 endforeach()
