--- a/Tests/RunCMake/file-GET_RUNTIME_DEPENDENCIES/linux-all-check.cmake
+++ b/Tests/RunCMake/file-GET_RUNTIME_DEPENDENCIES/linux-all-check.cmake
@@ -16,9 +16,9 @@ set(_check
   [[librunpath_parent_unresolved\.so]]
   [[librunpath_unresolved\.so]]
   )
-check_contents(deps/udeps1.txt "^${_check}$")
-check_contents(deps/udeps2.txt "^${_check}$")
-check_contents(deps/udeps3.txt "^${_check}$")
+check_contents(deps/udeps1.txt "^(libc.so;)?${_check}$")
+check_contents(deps/udeps2.txt "^(libc.so;)?${_check}$")
+check_contents(deps/udeps3.txt "^(libc.so;)?${_check}$")
 set(_check
   "^$"
   )
--- a/Tests/RunCMake/file-GET_RUNTIME_DEPENDENCIES/RunCMakeTest.cmake
+++ b/Tests/RunCMake/file-GET_RUNTIME_DEPENDENCIES/RunCMakeTest.cmake
@@ -66,7 +66,6 @@ elseif(CMAKE_HOST_SYSTEM_NAME STREQUAL "
 
   if(NOT CMake_COMPILER_FORCES_NEW_DTAGS)
     run_install_test(linux)
-    run_install_test(linux-parent-rpath-propagation)
     run_install_test(file-filter)
   endif()
   run_install_test(linux-unresolved)
