--- a/Tests/RunCMake/file-GET_RUNTIME_DEPENDENCIES/linux-all-check.cmake.orig	2020-12-05 13:27:21.098078774 +0700
+++ b/Tests/RunCMake/file-GET_RUNTIME_DEPENDENCIES/linux-all-check.cmake	2020-12-05 13:29:19.931553018 +0700
@@ -33,9 +33,9 @@
   [[librunpath_parent_unresolved\.so]]
   [[librunpath_unresolved\.so]]
   )
-check_contents(deps/udeps1.txt "^${_check}$")
-check_contents(deps/udeps2.txt "^${_check}$")
-check_contents(deps/udeps3.txt "^${_check}$")
+check_contents(deps/udeps1.txt "^(libc.so;)?${_check}$")
+check_contents(deps/udeps2.txt "^(libc.so;)?${_check}$")
+check_contents(deps/udeps3.txt "^(libc.so;)?${_check}$")
 set(_check
   "^libconflict\\.so:[^;]*/Tests/RunCMake/file-GET_RUNTIME_DEPENDENCIES/linux-build/root-all/lib/conflict/libconflict\\.so;[^;]*/Tests/RunCMake/file-GET_RUNTIME_DEPENDENCIES/linux-build/root-all/lib/conflict2/libconflict\\.so\n$"
   )
