--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -533,12 +533,6 @@ if(BUILD_TESTING)
 endif()
 
 if(NOT CMake_TEST_EXTERNAL_CMAKE)
-  # Install license file as it requires.
-  install(FILES
-    "${CMake_LICENSE_FILE}"
-    "${CMake_SOURCE_DIR}/CONTRIBUTORS.rst"
-    DESTINATION ${CMAKE_DOC_DIR})
-
   # Install script directories.
   install(
     DIRECTORY Help Modules Templates
--- a/Source/kwsys/CMakeLists.txt
+++ b/Source/kwsys/CMakeLists.txt
@@ -602,11 +602,6 @@ if(KWSYS_INSTALL_DOC_DIR)
       COMPONENT ${KWSYS_INSTALL_COMPONENT_NAME_RUNTIME}
       )
   endif()
-
-  # Install the license under the documentation directory.
-  install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/Copyright.txt
-    DESTINATION ${KWSYS_INSTALL_DOC_DIR}/${KWSYS_NAMESPACE}
-    ${KWSYS_INSTALL_LICENSE_OPTIONS})
 endif()
 
 #-----------------------------------------------------------------------------
