--- tdb.mk.orig	2011-11-16 14:59:11.561978836 +0100
+++ tdb.mk	2011-11-16 15:01:20.037847681 +0100
@@ -15,16 +15,16 @@ bin/tdbtest$(EXEEXT): tools/tdbtest.o $(
 	$(CC) $(CFLAGS) $(LDFLAGS) -o bin/tdbtest tools/tdbtest.o -L. -ltdb -lgdbm $(TDB_DEPS)
 
 bin/tdbtool$(EXEEXT): tools/tdbtool.o $(TDB_LIB)
-	$(CC) $(CFLAGS) $(LDFLAGS) -o bin/tdbtool tools/tdbtool.o -L. -ltdb $(TDB_DEPS)
+	$(CC) $(CFLAGS) $(LDFLAGS) tools/tdbtool.o $(TDB_DEPS) $(srcdir)/$(TDB_SOLIB) -o bin/tdbtool
 
 bin/tdbtorture$(EXEEXT): tools/tdbtorture.o $(TDB_LIB)
 	$(CC) $(CFLAGS) $(LDFLAGS) -o bin/tdbtorture tools/tdbtorture.o -L. -ltdb $(TDB_DEPS)
 
 bin/tdbdump$(EXEEXT): tools/tdbdump.o $(TDB_LIB)
-	$(CC) $(CFLAGS) $(LDFLAGS) -o bin/tdbdump tools/tdbdump.o -L. -ltdb $(TDB_DEPS)
+	$(CC) $(CFLAGS) $(LDFLAGS) tools/tdbdump.o $(TDB_DEPS) $(srcdir)/$(TDB_SOLIB) -o bin/tdbdump
 
 bin/tdbbackup$(EXEEXT): tools/tdbbackup.o $(TDB_LIB)
-	$(CC) $(CFLAGS) $(LDFLAGS) -o bin/tdbbackup tools/tdbbackup.o -L. -ltdb $(TDB_DEPS)
+	$(CC) $(CFLAGS) $(LDFLAGS) tools/tdbbackup.o $(TDB_DEPS) $(srcdir)/$(TDB_SOLIB) -o bin/tdbbackup
 
 test:: abi_checks
 
--- Makefile.in.orig	2011-11-18 09:26:39.467966708 +0100
+++ Makefile.in	2011-11-18 09:27:01.555448949 +0100
@@ -48,7 +48,7 @@ default: all
 include $(tdbdir)/tdb.mk
 include $(tdbdir)/rules.mk
 
-all:: showflags dirs $(PROGS) $(TDB_SOLIB) libtdb.a $(PYTHON_BUILD_TARGET) $(EXTRA_TARGETS)
+all:: showflags dirs $(TDB_SOLIB) libtdb.a $(PROGS) $(PYTHON_BUILD_TARGET) $(EXTRA_TARGETS)
 
 install:: all
 $(TDB_SOLIB): $(TDB_OBJ)
