--- src/bin/pdf/main.cpp    2013-12-11 14:56:58.000000000 +0100
+++ src/bin/pdf/main.cpp   2015-03-02 19:09:49.201401971 +0100
@@ -62,11 +62,6 @@
    if (!eina_init())
      goto del_global_param;
 
-   if (globalParams->getAntialias())
-     globalParams->setAntialias((char *)"yes");
-   if (globalParams->getVectorAntialias())
-     globalParams->setVectorAntialias((char *)"yes");
-
    pdfdoc = new PDFDoc(new GooString(file), NULL);
    if (!pdfdoc)
      goto del_global_param;

