Index: internal/utils/linux.go
===================================================================
--- internal/utils/linux.go.orig
+++ internal/utils/linux.go
@@ -20,7 +20,7 @@ func QT_DOC_DIR() string {
 	switch QT_DISTRO() {
 	case "arch":
 		return "/usr/share/doc/qt"
-	case "fedora":
+	case "fedora", "void":
 		return "/usr/share/doc/qt5"
 	case "suse":
 		return "/usr/share/doc/packages/qt5"
@@ -56,6 +56,9 @@ func QT_DISTRO() string {
 	if runtime.GOOS == "darwin" || runtime.GOOS == "freebsd" {
 		return runtime.GOOS
 	}
+	if _, err := exec.LookPath("xbps-query"); err == nil {
+		return "void"
+	}
 	if _, err := exec.LookPath("pacman"); err == nil {
 		return "arch"
 	}
