=== modified file 'Makefile.in'
--- a/Makefile.in	2010-05-27 13:08:26 +0000
+++ b/Makefile.in	2010-05-27 13:08:41 +0000
@@ -3,7 +3,7 @@
 WVPACKAGE_TARNAME=@PACKAGE_TARNAME@
 WVPACKAGE_VERSION=@PACKAGE_VERSION@
 
-prefix=/usr/local
+prefix=${DESTDIR}/usr
 WVSTREAMS_INC=
 WVSTREAMS_LIB=
 WVSTREAMS_BIN=$(prefix)/bin
@@ -23,7 +23,7 @@
 
 BINDIR=${prefix}/bin
 MANDIR=${prefix}/share/man
-PPPDIR=/etc/ppp/peers
+PPPDIR=${DESTDIR}/etc/ppp/peers
 
 include wvrules.mk
 

=== modified file 'config.defaults.mk'
--- a/config.defaults.mk	2010-05-27 13:08:26 +0000
+++ b/config.defaults.mk	2010-05-27 13:43:03 +0000
@@ -13,7 +13,7 @@
 LIBS_PAM=-lpam
 LIBS_TCL=
 
-prefix=/usr/local
+prefix=${DESTDIR}/usr
 datadir=${prefix}/share
 includedir=${prefix}/include
 infodir=${prefix}/share/info

