--- a/wvrules-posix.mk
+++ b/wvrules-posix.mk
@@ -85,12 +85,15 @@
 	$(AR) s $1
 endef
 
-CC: FORCE
+CC:
 	@CC="$(CC)" CFLAGS="$(CFLAGS)" CPPFLAGS="$(CPPFLAGS)" \
 	  $(WVSTREAMS_SRC)/gen-cc CC c
 
-CXX: FORCE
+CXX:
 	@CC="$(CXX)" CFLAGS="$(CXXFLAGS)" CPPFLAGS="$(CPPFLAGS)" \
 	  $(WVSTREAMS_SRC)/gen-cc CXX cc
 
+#All files must depend on the above two rules. This is a godawful hack.
+$(shell find -type f '(' -name '*.c' -o -name '*.cc' ')' ): CC CXX
+
 wvlink=$(LINK_MSG)$(WVLINK_CC) $(LDFLAGS) $($1-LDFLAGS) -o $1 $(filter %.o %.a %.so, $2) $($1-LIBS) $(XX_LIBS) $(LDLIBS) $(PRELIBS) $(LIBS)
