--- libnemo-extension/meson.build
+++ libnemo-extension/meson.build
@@ -61,7 +61,7 @@
   dependencies: nemo_extension_deps,
 )
 
-typelibdir = go_intr.get_pkgconfig_variable('typelibdir', define_variable: ['libdir', get_option('libdir')])
+typelibdir = go_intr.get_pkgconfig_variable('typelibdir')
 
 gnome.generate_gir(nemo_extension_lib,
   sources: nemo_extension_sources + nemo_extension_headers,
