diff --git libnemo-extension/meson.build libnemo-extension/meson.build
index 31a1618..84a6cf7 100644
--- libnemo-extension/meson.build
+++ libnemo-extension/meson.build
@@ -61,7 +61,9 @@ nemo_extension = declare_dependency(
   dependencies: nemo_extension_deps,
 )
 
-typelibdir = go_intr.get_pkgconfig_variable('typelibdir', define_variable: ['libdir', get_option('libdir')])
+typelibdir = go_intr.get_pkgconfig_variable('typelibdir',
+  define_variable: ['libdir', join_paths(get_option('prefix'), get_option('libdir'))]
+)
 
 gnome.generate_gir(nemo_extension_lib,
   sources: nemo_extension_sources + nemo_extension_headers,
