--- a/src/headers/gx_parameter.h	2018-07-22 06:24:50.000000000 +0200
+++ b/src/headers/gx_parameter.h	2018-07-22 06:24:50.000000000 +0200
@@ -28,6 +28,8 @@
 #ifndef SRC_HEADERS_GX_PARAMETER_H_
 #define SRC_HEADERS_GX_PARAMETER_H_
 
+#include <boost/noncopyable.hpp>
+
 namespace gx_system { class JsonWriter; class JsonParser; }
 
 namespace gx_engine {
--- a/src/headers/gx_system.h	2018-07-22 06:24:50.000000000 +0200
+++ b/src/headers/gx_system.h	2018-07-22 06:24:50.000000000 +0200
@@ -25,6 +25,8 @@
 #ifndef SRC_HEADERS_GX_SYSTEM_H_
 #define SRC_HEADERS_GX_SYSTEM_H_
 
+#include <boost/noncopyable.hpp>
+
 #ifndef NDEBUG
 #include <fenv.h>
 
