--- a/wscript
+++ b/wscript
@@ -203,7 +203,7 @@ def check_boost(conf):
     int main() { boost::system::error_code c; }
     """
     msg = "Checking for boost-system "
-    conf.check_cxx(msg = msg, fragment=code, lib="boost_system", uselib_store="BOOST_SYSTEM", mandatory=1,includes='/opt/local/include', libpath='/opt/local/lib')
+    conf.check_cxx(msg = msg, fragment=code, mandatory=1,includes='/opt/local/include', libpath='/opt/local/lib')
     # some boost (>1.49) versions depend on boost-system so we will check for it first
     # and later link against boost_system were boost headers are included.
     boost_atleast_version = 104200
@@ -218,11 +218,11 @@ def check_boost(conf):
     int main(){ return 0; }
     """ % boost_atleast_version
     msg = "Checking for boost >= %s" % boost_atleast_vermsg
-    conf.check_cxx(msg = msg, fragment=code, lib="boost_system", mandatory=1)
+    conf.check_cxx(msg = msg, fragment=code, mandatory=1)
     #conf.check_cxx(msg = msg, fragment=code, lib="boost_system-mt", mandatory=1,includes='/opt/local/include', libpath='/opt/local/lib')
 
     msg = "Checking for boost_iostreams >= %s" % boost_atleast_vermsg
-    conf.check_cxx(msg = msg, fragment=code, lib=["boost_iostreams","boost_system"], uselib_store="BOOST_IOSTREAMS", mandatory=1)
+    conf.check_cxx(msg = msg, fragment=code, lib=["boost_iostreams"], uselib_store="BOOST_IOSTREAMS", mandatory=1)
     #conf.check_cxx(msg = msg, fragment=code, lib=["boost_iostreams","boost_system-mt"], uselib_store="BOOST_IOSTREAMS", mandatory=1, includes='/opt/local/include', libpath='/opt/local/lib')
  
     #try:
