--- dbus/dnsmasq.conf.orig	2019-11-15 12:36:44.303440024 +0100
+++ dbus/dnsmasq.conf	2019-11-15 12:36:56.357452970 +0100
@@ -2,7 +2,7 @@
  "-//freedesktop//DTD D-BUS Bus Configuration 1.0//EN"
  "http://www.freedesktop.org/standards/dbus/1.0/busconfig.dtd">
 <busconfig>
-	<policy user="root">
+	<policy user="dnsmasq">
 		<allow own="uk.org.thekelleys.dnsmasq"/>
 		<allow send_destination="uk.org.thekelleys.dnsmasq"/>
 	</policy>
