--- a/font_freetype/Makefile.am	2005-10-18 11:45:40.000000000 +0100
+++ b/font_freetype/Makefile.am	2006-07-10 15:11:55.000000000 +0100
@@ -4,8 +4,9 @@
 agginclude_HEADERS = agg_font_freetype.h
 lib_LTLIBRARIES = libaggfontfreetype.la
 
-libaggfontfreetype_la_LDFLAGS = -version-info @AGG_LIB_VERSION@ @FREETYPE_LIBS@ 
+libaggfontfreetype_la_LDFLAGS = -version-info @AGG_LIB_VERSION@
 libaggfontfreetype_la_SOURCES = agg_font_freetype.cpp
 libaggfontfreetype_la_CXXFLAGS = -I$(top_srcdir)/include @FREETYPE_CFLAGS@ 
+libaggfontfreetype_la_LIBADD = ../src/libagg.la @FREETYPE_LIBS@
 endif
 
--- a/src/platform/sdl/Makefile.am	2005-10-17 23:49:35.000000000 +0100
+++ b/src/platform/sdl/Makefile.am	2006-07-10 15:11:55.000000000 +0100
@@ -5,6 +5,6 @@
 libaggplatformsdl_la_LDFLAGS = -version-info @AGG_LIB_VERSION@
 libaggplatformsdl_la_SOURCES = agg_platform_support.cpp
 libaggplatformsdl_la_CXXFLAGS = -I$(top_srcdir)/include @SDL_CFLAGS@
-libaggplatformsdl_la_LIBADD = @SDL_LIBS@
+libaggplatformsdl_la_LIBADD = ../../libagg.la @SDL_LIBS@
 endif
 
--- a/configure.ac	2023-04-01 07:28:38.000000000 -0300
+++ b/configure.ac	2023-07-05 02:14:00.092376054 -0300
@@ -123,7 +123,8 @@
 fi
 AM_CONDITIONAL(ENABLE_X11,[test x$no_x = x -a xno != x$enable_platform -a x$win32_host != xyes])
 AC_SUBST(x_includes)
-AC_SUBST(x_libraries)
+test -n "$x_libraries" && X_LDFLAGS="-L$x_libraries"
+AC_SUBST(X_LDFLAGS)
 dnl ###############################################
 
 dnl Settung up library version
--- a/src/platform/X11/Makefile.am	2023-04-01 07:28:38.000000000 -0300
+++ b/src/platform/X11/Makefile.am	2023-07-05 01:49:38.771562173 -0300
@@ -1,8 +1,8 @@
 if ENABLE_X11
 lib_LTLIBRARIES = libaggplatformX11.la
 
-libaggplatformX11_la_LDFLAGS = -version-info @AGG_LIB_VERSION@
+libaggplatformX11_la_LDFLAGS = -version-info @AGG_LIB_VERSION@ @X_LDFLAGS@
 libaggplatformX11_la_SOURCES = agg_platform_support.cpp
 libaggplatformX11_la_CXXFLAGS =  -I$(top_srcdir)/include @x_includes@
-libaggplatformX11_la_LIBADD = -lX11
+libaggplatformX11_la_LIBADD = ../../libagg.la -lX11
 endif
