diff --git a/bootstrap b/bootstrap
index f8de2d3..bd2ab07 100755
--- a/bootstrap
+++ b/bootstrap
@@ -5,7 +5,7 @@ set -e
 
 # Generate a VERSION file that is included in the dist tarball to avoid needed git
 # when calling autoreconf in a release tarball.
-git describe --tags --always --dirty > VERSION
+# git describe --tags --always --dirty > VERSION
 
 # generate list of source files for use in Makefile.am
 # if you add new source files, you must run ./bootstrap again
