--- configure.ac.orig	2014-07-30 00:57:35.525532956 +0200
+++ configure.ac	2014-07-30 01:24:04.531476426 +0200
@@ -717,39 +717,8 @@
 
 IF_PERL=\#
 if test x$support_perl = xyes; then
-  AC_PATH_PROG(PERL, perl5, perl)
-
-  AC_MSG_CHECKING(for $PERL suitability)
-  if $PERL -MExtUtils::Embed -e "use v5.8" >/dev/null 2>/dev/null; then
-
-     save_CXXFLAGS="$CXXFLAGS"
-     save_LIBS="$LIBS"
-     CXXFLAGS="$CXXFLAGS `$PERL -MExtUtils::Embed -e ccopts`"
-     LIBS="$LIBS `$PERL -MExtUtils::Embed -e ldopts`"
-     AC_LINK_IFELSE([AC_LANG_PROGRAM([[
-#include <EXTERN.h>
-#include <perl.h>
-#include <XSUB.h>
-]], [[
-     PerlInterpreter *perl = perl_alloc ();
-]])],[rxvt_perl_link=yes],[rxvt_perl_link=no])
-     CXXFLAGS="$save_CXXFLAGS"
-     LIBS="$save_LIBS"
-
-     if test x$rxvt_perl_link = xyes; then
-        AC_MSG_RESULT(ok)
-        AC_DEFINE(ENABLE_PERL, 1, Define if you can embed a perl interpreter)
-        IF_PERL=
-        PERL_O=rxvtperl.o
-        PERLFLAGS="`$PERL -MExtUtils::Embed -e ccopts`"
-        PERLLIB="`$PERL -MExtUtils::Embed -e ldopts`"
-        PERLPRIVLIBEXP="`$PERL -MConfig -e 'print $Config{privlibexp}'`"
-     else
-        AC_MSG_ERROR([no, unable to link])
-     fi
-  else
-     AC_MSG_ERROR([no working perl found, or perl not version >= 5.8])
-  fi
+   AC_DEFINE(ENABLE_PERL, 1, Define if you can embed a perl interpreter)
+   IF_PERL=
 fi
 AC_SUBST(PERLLIB)
 AC_SUBST(PERLFLAGS)
