Index: qt-everywhere-src-5.15.2/qtbase/src/corelib/global/qfloat16.h
===================================================================
--- qt-everywhere-src-5.15.2.orig/qtbase/src/corelib/global/qfloat16.h
+++ qt-everywhere-src-5.15.2/qtbase/src/corelib/global/qfloat16.h
@@ -43,6 +43,7 @@
 
 #include <QtCore/qglobal.h>
 #include <QtCore/qmetatype.h>
+#include <limits>
 #include <string.h>
 
 #if defined(QT_COMPILER_SUPPORTS_F16C) && defined(__AVX2__) && !defined(__F16C__)
Index: qt-everywhere-src-5.15.2/qtbase/src/corelib/text/qbytearraymatcher.h
===================================================================
--- qt-everywhere-src-5.15.2.orig/qtbase/src/corelib/text/qbytearraymatcher.h
+++ qt-everywhere-src-5.15.2/qtbase/src/corelib/text/qbytearraymatcher.h
@@ -42,6 +42,8 @@
 
 #include <QtCore/qbytearray.h>
 
+#include <limits>
+
 QT_BEGIN_NAMESPACE
 
 
