--- src/kvirc/kernel/KviIrcConnection.cpp.orig
+++ src/kvirc/kernel/KviIrcConnection.cpp
@@ -669,7 +669,7 @@
 	while(m_pChannelList->first())
 	{
 		m_pChannelList->first()->close();
-		QApplication::processEvents(QEventLoop::ExcludeSocketNotifiers & QEventLoop::ExcludeUserInputEvents);
+		QApplication::processEvents(QEventLoop::ExcludeSocketNotifiers | QEventLoop::ExcludeUserInputEvents);
 	}
 }
 
@@ -678,7 +678,7 @@
 	while(m_pQueryList->first())
 	{
 		m_pQueryList->first()->close();
-		QApplication::processEvents(QEventLoop::ExcludeSocketNotifiers & QEventLoop::ExcludeUserInputEvents);
+		QApplication::processEvents(QEventLoop::ExcludeSocketNotifiers | QEventLoop::ExcludeUserInputEvents);
 	}
 }
 
