OpenSSH 6.8 changed fingerprint format to SHA256:base64.

--- keychain.sh.orig	2015-03-18 10:43:15.441403693 +0100
+++ keychain.sh	2015-03-18 10:44:51.825161724 +0100
@@ -666,6 +666,11 @@ extract_fingerprints() {
                 #   2048 00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00 /home/barney/.ssh/id_rsa.pub
                 echo "$ef_line" | cut -f3 -d' '
                 ;;
+	    *\ SHA256:*)
+                # The new OpenSSH format from 6.8, we hope
+                #   1024 SHA256:[base64] /home/barney/.ssh/id_dsa (RSA)
+                echo "$ef_line"|awk '{print $2}'|cut -f2 -d':'
+                ;;
             *\ [0-9a-fA-F][0-9a-fA-F]:[0-9a-fA-F][0-9a-fA-F]:*)
                 # The more consistent OpenSSH format, we hope
                 #   1024 00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00 /home/barney/.ssh/id_dsa (DSA)
