--- a/cmake/coin.cmake
+++ b/cmake/coin.cmake
@@ -23,7 +23,7 @@ add_library(COIN ${COIN_LIBRARY_TYPE} ${
 group_files(COIN_SOURCES "coin")
 target_include_directories(COIN SYSTEM PUBLIC
   $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include/coin>
-  $<INSTALL_INTERFACE:include/coin>)
+  $<INSTALL_INTERFACE:include/ogdf/coin>)
 target_compile_definitions(COIN PRIVATE
     -DCLP_BUILD -DCOINUTILS_BUILD -DOSI_BUILD -D__OSI_CLP__
     -DCOMPILE_IN_CG -DCOMPILE_IN_CP -DCOMPILE_IN_LP -DCOMPILE_IN_TM
