--- a/cmake/compiler-specifics.cmake
+++ b/cmake/compiler-specifics.cmake
@@ -39,17 +39,6 @@ if(MSVC)
   endif()
 endif()
 
-# use specified or native arch (ie, activate things like SSE)
-if(CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang" AND NOT ${CMAKE_SYSTEM_PROCESSOR} MATCHES "^arm")
-  set(OGDF_ARCH "native" CACHE STRING "Target CPU (micro)architecture passed to the compiler via `-march`.")
-  mark_as_advanced(OGDF_ARCH)
-
-  # cannot use add_definitions() here because it does not work with check-sse3.cmake
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -march=${OGDF_ARCH}")
-else()
-  unset(OGDF_ARCH CACHE)
-endif()
-
 # set default warning flags for OGDF and tests
 set(available_default_warning_flags "")
 set(available_default_warning_flags_debug "")
@@ -96,6 +85,5 @@ elseif(CMAKE_CXX_COMPILER_ID MATCHES "GN
     set(available_default_warning_flags_release "-Wno-error=unused-but-set-variable -Wno-error=strict-overflow")
   endif()
   set(available_default_warning_flags_release "${available_default_warning_flags_release} -Wno-error=unused-variable")
-  set(warnings_as_errors_flag "-Werror")
   set(warnings_not_as_errors_flag "-Wno-error")
 endif()
