Index: b/cmake/compiler-specifics.cmake
===================================================================
--- a/cmake/compiler-specifics.cmake
+++ b/cmake/compiler-specifics.cmake
@@ -12,12 +12,6 @@ if(MSVC)
   set(COIN_LIBRARY_TYPE STATIC)
 endif()
 
-# use native arch (ie, activate things like SSE)
-if(CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
-  # cannot use add_definitions() here because it does not work with check-sse3.cmake
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -march=native")
-endif()
-
 # set default warning flags for OGDF and tests
 set(available_default_warning_flags "")
 set(available_default_warning_flags_debug "")
@@ -38,7 +32,6 @@ if(CMAKE_CXX_COMPILER_ID MATCHES "GNU|Cl
     set(available_default_warning_flags_release "-Wno-error=unused-but-set-variable -Wno-error=strict-overflow")
   endif()
   set(available_default_warning_flags_release "${available_default_warning_flags_release} -Wno-error=unused-variable")
-  set(warnings_as_errors_flag "-Werror")
 elseif(MSVC)
   set(available_default_warning_flags "/W3 /wd4018 /wd4068 /wd4101 /wd4244 /wd4250 /wd4267 /wd4373 /wd4800 /wd4996")
   # this has to be explained because MSVC is so cryptic:
