--- examples/dvbinfo/buffer.h
+++ examples/dvbinfo/buffer.h
@@ -23,6 +23,9 @@
 #ifndef DVBINFO_BUFFER_H_
 #define DVBINFO_BUFFER_H_
 
+#include <sys/types.h>
+#include <stdint.h>
+
 typedef struct buffer_s buffer_t;
 
 struct buffer_s
--- src/descriptor.h
+++ src/descriptor.h
@@ -35,6 +35,9 @@
 #ifndef _DVBPSI_DESCRIPTOR_H_
 #define _DVBPSI_DESCRIPTOR_H_
 
+#include <sys/types.h>
+#include <stdint.h>
+
 #ifdef __cplusplus
 extern "C" {
 #endif
