--- a/configure
+++ b/configure
@@ -4092,9 +4092,9 @@ ac_compile='$CXX -c $CXXFLAGS $CPPFLAGS
 ac_link='$CXX -o conftest$ac_exeext $CXXFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS >&5'
 ac_compiler_gnu=$ac_cv_cxx_compiler_gnu
 
-  { $as_echo "$as_me:${as_lineno-$LINENO}: checking if $CXX accepts -std=c++11" >&5
-$as_echo_n "checking if $CXX accepts -std=c++11... " >&6; }
-  CXXFLAGS="-std=c++11 $CXXFLAGS"
+  { $as_echo "$as_me:${as_lineno-$LINENO}: checking if $CXX accepts -std=c++17" >&5
+$as_echo_n "checking if $CXX accepts -std=c++17... " >&6; }
+  CXXFLAGS="-std=c++17 $CXXFLAGS"
         cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 int main(){}
