commit 34d8f3fb025e930450ce44f7aac501da4c2adc9c
Author: Daniel Kolesa <daniel@octaforge.org>
Date:   Sun Jan 23 21:09:42 2022 +0100

    disable spidermonkey jslint

diff --git a/js/src/build/moz.build b/js/src/build/moz.build
index 55c66b97..243f7db2 100644
--- a/js/src/build/moz.build
+++ b/js/src/build/moz.build
@@ -97,15 +97,3 @@ NO_EXPAND_LIBS = True
 
 DIST_INSTALL = True
 
-# Run SpiderMonkey style checker after linking the static library. This avoids
-# running the script for no-op builds.
-GeneratedFile(
-    "spidermonkey_checks",
-    script="/config/run_spidermonkey_checks.py",
-    inputs=[
-        "!%sjs_static.%s" % (CONFIG["LIB_PREFIX"], CONFIG["LIB_SUFFIX"]),
-        "/config/check_spidermonkey_style.py",
-        "/config/check_macroassembler_style.py",
-        "/config/check_js_opcode.py",
-    ],
-)
