commit 32e52128ce8fe7704a29a5a4fb921a8394540fbc
Author: q66 <daniel@octaforge.org>
Date:   Mon Jun 13 14:01:59 2022 +0200

    fix ftbfs on powerpc32

diff --git a/js/src/wasm/WasmFrame.h b/js/src/wasm/WasmFrame.h
index 85f2612d..df5c9629 100644
--- a/js/src/wasm/WasmFrame.h
+++ b/js/src/wasm/WasmFrame.h
@@ -230,7 +230,7 @@ class DebugFrame {
   // Avoid -Wunused-private-field warnings.
  protected:
 #if defined(JS_CODEGEN_MIPS32) || defined(JS_CODEGEN_ARM) || \
-    defined(JS_CODEGEN_X86) || defined(__wasi__)
+    defined(JS_CODEGEN_X86) || defined(__wasi__) || (defined(__powerpc__) && !defined(__powerpc64__))
   // See alignmentStaticAsserts().  For MIPS32, ARM32 and X86 DebugFrame is only
   // 4-byte aligned, so we add another word to get up to 8-byte
   // alignment.
