--- a/Makefile	2018-08-21 18:07:51.000000000 +0200
+++ b/Makefile.new	2018-10-02 11:16:49.335221638 +0200
@@ -61,16 +61,7 @@
 
 $(CONFIGURE): aclocal.m4
 	autoconf
-	./configure \
-		--prefix=/ \
-		--exec-prefix=/ \
-		--sbindir=/bin \
-		--bindir=/usr/bin \
-		--libdir=/lib \
-		--libexecdir=/usr/lib \
-		--includedir=/usr/include \
-		--mandir=/usr/share/man \
-		--datadir=/usr/share \
+	./configure @configure_args@ \
 		$$LOCAL_CONFIGURE_OPTIONS
 	touch .census
 
