From 146e628e68487110607a82e338e399c238a2b371 Mon Sep 17 00:00:00 2001
From: Jasper Chan <jasperchan515@gmail.com>
Date: Thu, 16 Aug 2018 17:16:27 -0700
Subject: [PATCH] Remove osx features

---
 Makefile | 14 ++++----------
 1 file changed, 4 insertions(+), 10 deletions(-)

diff --git Makefile Makefile
index 6287cbc..eb4b38d 100644
--- Makefile
+++ Makefile
@@ -395,7 +395,7 @@ JBGOPTS=-m 16 -d 0 -p 92	# Equivalent options for pbmtojbg
 # The usual build rules
 #
 all:	all-test $(PROGS) $(BINPROGS) $(SHELLS) getweb \
-	all-icc2ps all-osx-hotplug man doc \
+	all-icc2ps man doc \
 	all-done
 
 all-test:
@@ -628,7 +628,7 @@ command2foo2lava-pjl.o: command2foo2lava-pjl.c
 #
 # Installation rules
 #
-install: all install-test install-prog install-icc2ps install-osx-hotplug \
+install: all install-test install-prog install-icc2ps \
 	    install-extra install-crd install-foo install-ppd \
 	    install-gui install-desktop install-filter \
 	    install-man install-doc
@@ -969,7 +969,7 @@ RULES=hplj10xx.rules
 DEVDDIR=/etc/devd
 
 ifeq ($(UNAME),Darwin)
-install-hotplug: install-hotplug-test install-hotplug-osx
+install-hotplug: install-hotplug-test
 else
 install-hotplug: install-hotplug-test install-hotplug-prog
 endif
@@ -1155,7 +1155,6 @@ cups:	FRC
 # Uninstall
 #
 uninstall:
-	cd osx-hotplug; $(MAKE) PREFIX=$(PREFIX) uninstall
 	-rm -f /etc/hotplug/usb/hplj1000
 	-rm -f /etc/hotplug/usb/hplj1005
 	-rm -f /etc/hotplug/usb/hplj1018
@@ -1254,7 +1253,6 @@ clean:
 	-rm -f pksm2bitcmyk
 	-rm -f *.icm.*.ps
 	cd icc2ps; $(MAKE) $@
-	cd osx-hotplug; $(MAKE) $@
 
 #
 # Header dependencies
@@ -1511,7 +1509,7 @@ oldppd:
 # Manpage generation.  No, I am not interested in "info" files or
 # HTML documentation.
 #
-man: $(MANPAGES) man-icc2ps man-osx-hotplug
+man: $(MANPAGES) man-icc2ps
 
 $(MANPAGES): macros.man includer-man
 
@@ -1583,9 +1581,6 @@ install-man: man
 	$(INSTALL) -c -m 644 usb_printerid.1 $(MANDIR)/man1/
 	$(INSTALL) -c -m 644 printer-profile.1 $(MANDIR)/man1/
 	cd icc2ps; $(MAKE) install-man
-ifeq ($(UNAME),Darwin)
-	cd osx-hotplug; $(MAKE) install-man
-endif
 
 doc: README INSTALL manual.pdf
 
@@ -1597,7 +1592,6 @@ install-doc: doc
 	$(INSTALL) -c -m 644 manual.pdf $(DOCDIR)
 	$(INSTALL) -c -m 644 COPYING $(DOCDIR)
 	$(INSTALL) -c -m 644 INSTALL $(DOCDIR)
-	$(INSTALL) -c -m 644 INSTALL.osx $(DOCDIR)
 	$(INSTALL) -c -m 644 README $(DOCDIR)
 	$(INSTALL) -c -m 644 ChangeLog $(DOCDIR)
 
-- 
2.18.0

