From c026b43d6ac929950e430a39f4c89d73d52dd659 Mon Sep 17 00:00:00 2001
From: Jasper Chan <jasperchan515@gmail.com>
Date: Sun, 19 Aug 2018 17:22:45 -0700
Subject: [PATCH] Skip foomatic-rip check

---
 Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git Makefile Makefile
index 6287cbc..2edbbae 100644
--- Makefile
+++ Makefile
@@ -659,11 +659,11 @@ install-test:
 	    : ; \
 	elif ! type foomatic-rip >/dev/null 2>&1; then \
 	    echo "      ***"; \
-	    echo "      *** Error: foomatic-rip is not installed!"; \
+	    echo "      *** Warning: foomatic-rip is not installed!"; \
 	    echo "      ***"; \
 	    echo "      *** Install foomatic package(s) for your OS"; \
 	    echo "      ***"; \
-	    exit 1; \
+		: ; \
 	fi
 	# ... OK!
 	#
-- 
2.18.0

