OSError: [Errno 97] Address family not supported by protocol

--- a/src/gevent/tests/test__socket.py
+++ b/src/gevent/tests/test__socket.py
@@ -483,7 +483,7 @@
                 # use', which makes even less sense. The manylinux
                 # 2010 environment produces 'errno 99 Cannot assign
                 # requested address', which, I guess?
-                'refused|not known|already in use|assign'
+                'refused|not known|already in use|assign|not supported'
         ):
             socket.create_connection(
                 (greentest.DEFAULT_BIND_ADDR, connect_port),
