--- setup.py.orig	2018-06-08 12:56:05.000000000 +0200
+++ setup.py	2018-06-09 18:13:19.783191927 +0200
@@ -306,15 +306,6 @@
 
 
 def run_setup(ext_modules, run_make):
-    if run_make:
-        if (not LIBEV_EMBED and not WIN and cffi_modules) or PYPY:
-            # We're not embedding libev but we do want
-            # to build the CFFI module. We need to configure libev
-            # because the CORE Extension won't.
-            # TODO: Generalize this.
-            if LIBEV_CFFI_MODULE in cffi_modules and not WIN:
-                system(libev_configure_command)
-
     setup(
         name='gevent',
         version=__version__,
