--- a/waflib/Context.py	2024-06-04 07:49:13.399793819 -0400
+++ b/waflib/Context.py	2024-06-04 08:00:35.473824306 -0400
@@ -2,7 +2,7 @@
 # encoding: utf-8
 # WARNING! Do not edit! https://waf.io/book/index.html#_obtaining_the_waf_file

-import os,re,imp,sys
+import os,re,types,sys
 from waflib import Utils,Errors,Logs
 import waflib.Node
 HEXVERSION=0x2000e00
@@ -344,7 +344,7 @@
 		return cache_modules[path]
 	except KeyError:
 		pass
-	module=imp.new_module(WSCRIPT_FILE)
+	module=types.ModuleType(WSCRIPT_FILE)
 	try:
 		code=Utils.readf(path,m='r',encoding=encoding)
 	except EnvironmentError:

