Index: pam_zfscrypt-0.2.0/include/zfscrypt_dataset.h
===================================================================
--- pam_zfscrypt-0.2.0.orig/include/zfscrypt_dataset.h
+++ pam_zfscrypt-0.2.0/include/zfscrypt_dataset.h
@@ -71,18 +71,3 @@ zfscrypt_err_t zfscrypt_dataset_iter(zfs
 // private constants
 
 extern const char ZFSCRYPT_USER_PROPERTY[];
-
-// FIXME Copied from /usr/include/libzfs/sys/zio.h because including <sys/zio.h> results in
-// compiler error about unknown type rlim64_t.
-enum zio_encrypt {
-    ZIO_CRYPT_INHERIT = 0,
-    ZIO_CRYPT_ON,
-    ZIO_CRYPT_OFF,
-    ZIO_CRYPT_AES_128_CCM,
-    ZIO_CRYPT_AES_192_CCM,
-    ZIO_CRYPT_AES_256_CCM,
-    ZIO_CRYPT_AES_128_GCM,
-    ZIO_CRYPT_AES_192_GCM,
-    ZIO_CRYPT_AES_256_GCM,
-    ZIO_CRYPT_FUNCTIONS
-};
