--- a/engine/core/vfs/zip/zipnode.cpp
+++ b/engine/core/vfs/zip/zipnode.cpp
@@ -29,6 +29,8 @@
 
 #include "zipnode.h"
 
+#include <algorithm>
+
 namespace {
     /** helper function to find a value in a ZipNodeContainer
      *  @param container the ZipNodeContainer to search through
@@ -266,4 +268,4 @@ std::ostream& operator<<(std::ostream& o
     }
 
     return os;
-}
\ No newline at end of file
+}
