--- src/libutil/PosixMessageQueue.h	2009-12-19 17:26:20.000000000 +0100
+++ src/libtuil/PosixMessageQueue.h	2016-10-03 18:40:48.996217777 +0200
@@ -36,7 +36,13 @@
 #include <mqueue.h>
+#include <signal.h>
 #include <string>
 
+#if !defined(__GLIBC__)
+/* In musl libc this type is not defined. */
+typedef union sigval sigval_t;
+#endif
+
 namespace Util
 {
 
 
--- src/libutil/IpcRingBuffer.h	2008-06-08 19:17:32.000000000 +0200
+++ src/libutil/IpcRingBuffer.h	2016-10-03 18:53:00.310710281 +0200
@@ -37,6 +37,7 @@
 #include "PosixMessageQueue.h"
 
 #include <semaphore.h>
+#include <signal.h>
 
 namespace Util {
 
