--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -93,7 +93,7 @@ if(APPLE)
 	find_library(SECURITY_LIBRARY Security REQUIRED)
 endif()
 
-find_package(Boost ${BOOST_VERSION} REQUIRED COMPONENTS iostreams program_options regex system thread random coroutine locale filesystem graph)
+find_package(Boost ${BOOST_VERSION} REQUIRED COMPONENTS iostreams program_options thread random coroutine locale filesystem graph)
 find_package(ICU REQUIRED COMPONENTS data i18n uc)
 
 # no, gettext executables are not required when NLS is deactivated
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -240,8 +240,6 @@ if(ENABLE_GAME)
 		OpenSSL::SSL
 		Boost::iostreams
 		Boost::program_options
-		Boost::regex
-		Boost::system
 		Boost::random
 		Boost::coroutine
 		Boost::locale
@@ -287,8 +285,6 @@ if(ENABLE_TESTS)
 		OpenSSL::SSL
 		Boost::iostreams
 		Boost::program_options
-		Boost::regex
-		Boost::system
 		Boost::random
 		Boost::coroutine
 		Boost::locale
@@ -336,8 +332,6 @@ if(ENABLE_SERVER)
 		OpenSSL::SSL
 		Boost::iostreams
 		Boost::program_options
-		Boost::regex
-		Boost::system
 		Boost::random
 		Boost::coroutine
 		Boost::locale
@@ -377,8 +371,6 @@ if(ENABLE_CAMPAIGN_SERVER)
 		OpenSSL::SSL
 		Boost::iostreams
 		Boost::program_options
-		Boost::regex
-		Boost::system
 		Boost::random
 		Boost::coroutine
 		Boost::locale
