From 182d0ff548b0149a17a6fefcf061074db05c82de Mon Sep 17 00:00:00 2001
From: Pentarctagon <pentarctagon@tutamail.com>
Date: Thu, 17 Aug 2023 23:16:09 -0500
Subject: [PATCH] Compile fix for boost 1.83.

Fixes #7849
---
 src/gettext.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/gettext.cpp b/src/gettext.cpp
index 2d27ef6abc00..5e0d2f6c3ea5 100644
--- a/src/gettext.cpp
+++ b/src/gettext.cpp
@@ -166,7 +166,11 @@ namespace
 			return msg;
 		}
 
+#if BOOST_VERSION < 108300
 		const char* get(int domain_id, const char* ctx, const char* sid, int n) const override
+#else
+		const char* get(int domain_id, const char* ctx, const char* sid, bl::count_type n) const override
+#endif
 		{
 			auto& base = get_base();
 			const char* msg = base.get(domain_id, ctx, sid, n);
