--- src/replay.cpp.orig	2016-05-18 00:57:46.000000000 +0000
+++ src/replay.cpp	2017-09-22 20:22:56.660358034 +0000
@@ -403,7 +403,7 @@
 		last_location = *loc_it;
 		
 		const config &speak = command(last_location).child("speak");
-		assert(speak);
+		assert(static_cast<bool>(speak));
 		add_chat_log_entry(speak, chat_log_appender);
 
 	}
@@ -581,7 +581,7 @@
 config &replay::command(int n)
 {
 	config & retv = cfg_.child("command", n);
-	assert(retv);
+	assert(static_cast<bool>(retv));
 	return retv;
 }
 
--- src/playcampaign.cpp.orig	2016-05-18 00:57:46.000000000 +0000
+++ src/playcampaign.cpp	2017-09-22 20:25:32.094366053 +0000
@@ -192,7 +192,7 @@
 	if (gamestate.replay_start().empty()){
 		// Backwards compatibility code for 1.2 and 1.2.1
 		const config &scenario = game_config.find_child(type,"id",gamestate.carryover_sides_start["next_scenario"]);
-		assert(scenario);
+		assert(static_cast<bool>(scenario));
 		gamestate.replay_start() = scenario;
 	}
 	starting_pos = gamestate.replay_start();
