From 6c16e1ee44f0ce02fdac3a83df744a79305af13e Mon Sep 17 00:00:00 2001
From: "Andrew J. Hesford" <ajh@sideband.org>
Date: Fri, 26 Nov 2021 07:57:31 -0500
Subject: [PATCH] Fix format overflow warning with 32-bit gcc

---
 Quake/gl_model.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Quake/gl_model.c b/Quake/gl_model.c
index 75d2ae6..062ce2a 100644
--- a/Quake/gl_model.c
+++ b/Quake/gl_model.c
@@ -2465,7 +2465,7 @@ visdone:
 
 		if (i < mod->numsubmodels-1)
 		{	// duplicate the basic information
-			char	name[10];
+			char	name[12];
 
 			sprintf (name, "*%i", i+1);
 			loadmodel = Mod_FindName (name);
-- 
2.34.1

