Index: p11-kit-0.23.22/doc/manual/meson.build
===================================================================
--- p11-kit-0.23.22.orig/doc/manual/meson.build
+++ p11-kit-0.23.22/doc/manual/meson.build
@@ -60,7 +60,7 @@ if get_option('gtk_doc')
     'annotation-glossary.xml'
   ]
 
-  glib_prefix = dependency('glib-2.0').get_pkgconfig_variable('prefix')
+  glib_prefix = '/usr'
   fixxref_args = [
     '--html-dir=' + (prefix / gnome.gtkdoc_html_dir(meson.project_name())),
     '--extra-dir=' + (glib_prefix / gnome.gtkdoc_html_dir('glib')),
