--- libunity-7.1.4+19.04.20190319.orig/extras/Makefile.am
+++ libunity-7.1.4+19.04.20190319/extras/Makefile.am
@@ -24,6 +24,7 @@
   --library unity-extras \
   --internal-vapi=unity-extras-internal.vapi \
   --internal-header=unity-extras-internal.h \
+  --shared-library=libunity-extras.so.9 \
   --thread \
   --use-header \
   --vapidir $(top_srcdir)/vapi \
@@ -82,7 +83,6 @@
 
 libunity_extras_la_vala.stamp: $(libunity_extras_la_VALASOURCES)
 	$(AM_V_GEN) $(VALAC) $(libunity_extras_la_VALAFLAGS) $^
-	@sed -i -e 's/<namespace name="UnityExtras" version="@GIR_VERSION@" c:prefix="Unity">/<namespace name="UnityExtras" version="@GIR_VERSION@" c:prefix="UnityExtras" shared-library="libunity-extras.so.@LIBUNITY_LT_CURRENT@">/g' UnityExtras-@GIR_VERSION@.gir
 	@sed -i -e 's/"Extras/"/g;s/"extras_/"/' UnityExtras-@GIR_VERSION@.gir
 	@sed -i -e 's/<parameter name="scope_creation_cb" transfer-ownership="none"/<parameter name="scope_creation_cb" transfer-ownership="none" scope="call"/' UnityExtras-@GIR_VERSION@.gir
 	@touch $@
@@ -97,12 +97,19 @@
 # Compile .typelib from .gir
 ##
 if HAVE_INTROSPECTION
--include $(INTROSPECTION_MAKEFILE)
-INTROSPECTION_GIRS =
-INTROSPECTION_COMPILER_ARGS = --includedir=$(srcdir)
-
 typelibdir = $(libdir)/girepository-1.0
 typelib_DATA = $(extrasgir_DATA:.gir=.typelib)
 
+# Extract dlname from libunity.la
+# This is what g-ir-scanner does.
+libunity_extra_dlname = \
+	`$(SED) -nE "s/^dlname='([A-Za-z0-9.+-]+)'/\1/p" libunity-extras.la`
+INTROSPECTION_COMPILER_ARGS= \
+	--includedir=$(srcdir) \
+	-l $(libunity_extra_dlname)
+$(extrasgir_DATA): libunity_extras_la_vala.stamp
+$(typelib_DATA): $(extrasgir_DATA) libunity-extras.la
+	@INTROSPECTION_COMPILER@ $(INTROSPECTION_COMPILER_ARGS) -o $@ $^
+
 CLEANFILES += $(typelib_DATA)
 endif
--- libunity-7.1.4+19.04.20190319.orig/protocol/protocol-icon.vala
+++ libunity-7.1.4+19.04.20190319/protocol/protocol-icon.vala
@@ -185,7 +185,7 @@
   }
 
   /* Added to GIcon interface in 2.37 */
-  private Variant serialize ()
+  private Variant? serialize ()
   {
     Variant? ret = null;
     return ret;
--- libunity-7.1.4+19.04.20190319.orig/src/Makefile.am
+++ libunity-7.1.4+19.04.20190319/src/Makefile.am
@@ -75,6 +75,7 @@
   --library unity \
   --internal-vapi=unity-internal.vapi \
   --internal-header=unity-internal.h \
+  --shared-library=libunity.so.9 \
   --target-glib=2.32 \
   --thread \
   --use-header \
@@ -135,21 +136,8 @@
 
 $(libunity_la_GENERATED): libunity_la_vala.stamp
 
-#
-# The reason for the first 'sed' below is:
-# https://bugzilla.gnome.org/show_bug.cgi?id=642576
-#
-# The second is there because forced cname in Vala is not properly propagated
-# into the gir file. Need https://bugzilla.gnome.org/show_bug.cgi?id=681356
-#
-# And the third because we need to specify scope manually.
-#
 libunity_la_vala.stamp: $(libunity_la_VALASOURCES)
 	$(AM_V_GEN) $(VALAC) $(libunity_la_VALAFLAGS) $^
-	@sed -i -e 's/<namespace name="Unity" version="@GIR_VERSION@" c:prefix="Unity">/<namespace name="Unity" version="@GIR_VERSION@" c:prefix="Unity" shared-library="libunity.so.@LIBUNITY_LT_CURRENT@">/g' Unity-@GIR_VERSION@.gir
-	@sed -i -e 's/emit_preview_ready/preview_ready/g' Unity-@GIR_VERSION@.gir
-	@sed -i -e 's/<parameter name="async_callback" transfer-ownership="none" /<parameter name="async_callback" transfer-ownership="none" scope="async" /g' Unity-@GIR_VERSION@.gir
-	@sed -i -e 's/<record name="ScopeResult">/<record name="ScopeResult" c:type="UnityScopeResult" glib:type-name="UnityScopeResult" glib:get-type="unity_scope_result_get_type">/;s/<record name="SearchContext">/<record name="SearchContext" c:type="UnitySearchContext" glib:type-name="UnitySearchContext" glib:get-type="unity_search_context_get_type">/' Unity-@GIR_VERSION@.gir
 	@touch $@
 
 BUILT_SOURCES += libunity_la_vala.stamp
@@ -169,12 +157,19 @@
 # Compile .typelib from .gir
 ##
 if HAVE_INTROSPECTION
--include $(INTROSPECTION_MAKEFILE)
-INTROSPECTION_GIRS =
-INTROSPECTION_COMPILER_ARGS = --includedir=$(srcdir)
-
 typelibdir = $(libdir)/girepository-1.0
 typelib_DATA = $(unitygir_DATA:.gir=.typelib)
 
+# Extract dlname from libunity.la
+# This is what g-ir-scanner does.
+libunity_dlname = \
+	`$(SED) -nE "s/^dlname='([A-Za-z0-9.+-]+)'/\1/p" libunity.la`
+INTROSPECTION_COMPILER_ARGS= \
+	--includedir=$(srcdir) \
+	-l $(libunity_dlname)
+$(unitygir_DATA): libunity_la_vala.stamp
+$(typelib_DATA): $(unitygir_DATA) libunity.la
+	@INTROSPECTION_COMPILER@ $(INTROSPECTION_COMPILER_ARGS) -o $@ $^
+
 CLEANFILES += $(typelib_DATA)
 endif
--- libunity-7.1.4+19.04.20190319.orig/src/unity-aggregator-scope.vala
+++ libunity-7.1.4+19.04.20190319/src/unity-aggregator-scope.vala
@@ -51,7 +51,7 @@
    */
   public abstract int category_index_for_scope_id (string scope_id);
 
-  public AggregatorScope (string dbus_path_, string id_, MergeMode merge_mode = AggregatorScope.MergeMode.OWNER_SCOPE, bool proxy_filter_hints = false)
+  protected AggregatorScope (string dbus_path_, string id_, MergeMode merge_mode = AggregatorScope.MergeMode.OWNER_SCOPE, bool proxy_filter_hints = false)
   {
     Object (dbus_path: dbus_path_, id: id_, is_master: true,
             merge_mode: merge_mode, proxy_filter_hints: proxy_filter_hints);
--- libunity-7.1.4+19.04.20190319.orig/src/unity-deprecated-scope.vala
+++ libunity-7.1.4+19.04.20190319/src/unity-deprecated-scope.vala
@@ -61,7 +61,7 @@
   internal CategorySet _categories;
   internal FilterSet _filters;
    
-  public DeprecatedScopeBase (string dbus_path_, string id_)
+  protected DeprecatedScopeBase (string dbus_path_, string id_)
   {
     Object (dbus_path: dbus_path_, id: id_);
   }
--- libunity-7.1.4+19.04.20190319.orig/src/unity-scope-channel.vala
+++ libunity-7.1.4+19.04.20190319/src/unity-scope-channel.vala
@@ -312,7 +312,7 @@
           DBusSignalFlags.NONE, this.owner_changed);
     }
 
-    private void owner_changed (DBusConnection con, string sender_name,
+    private void owner_changed (DBusConnection con, string? sender_name,
                                 string obj_path, string ifc_name,
                                 string sig_name, Variant parameters)
     {
--- libunity-7.1.4+19.04.20190319.orig/tools/preview-renderer.vala
+++ libunity-7.1.4+19.04.20190319/tools/preview-renderer.vala
@@ -63,7 +63,7 @@
      */
     public abstract class GridRenderer: PreviewRenderer
     {
-        public GridRenderer()
+        protected GridRenderer()
         {
             Object();
         }
