Source: https://github.com/void-linux/void-packages/issues/23557

diff --git a/linux-user/syscall.c b/linux-user/syscall.c
index 14fdebd..de280af 100644
--- a/linux-user/syscall.c
+++ b/linux-user/syscall.c
@@ -3229,7 +3229,7 @@ static abi_long do_sendrecvmsg_locked(int fd, struct target_msghdr *msgp,
                                       int flags, int send)
 {
     abi_long ret, len;
-    struct msghdr msg;
+    struct msghdr msg = {0};
     abi_ulong count;
     struct iovec *vec;
     abi_ulong target_vec;
