--- kded/CMakeLists.txt	2009-10-02 14:55:17.000000000 +0000
+++ kded/CMakeLists.txt	2010-01-31 22:16:13.946933892 +0000
@@ -69,7 +69,7 @@
 if (WIN32)
 install( FILES applications.menu  DESTINATION  ${SHARE_INSTALL_PREFIX}/xdg/menus )
 else (WIN32)
-install( FILES applications.menu  DESTINATION  ${SYSCONF_INSTALL_DIR}/xdg/menus )
+install( FILES applications.menu  DESTINATION  ${SYSCONF_INSTALL_DIR}/xdg/menus RENAME kde-applications.menu )
 endif (WIN32)
 install( FILES kdedmodule.desktop DESTINATION  ${SERVICETYPES_INSTALL_DIR} )
 install( FILES kded.upd           DESTINATION  ${DATA_INSTALL_DIR}/kconf_update )
--- kded/kbuildsycoca.cpp	2009-12-04 23:10:18.000000000 +0000
+++ kded/kbuildsycoca.cpp	2010-01-31 22:16:13.962766572 +0000
@@ -302,7 +302,7 @@
      if (!m_trackId.isEmpty())
         g_vfolder->setTrackId(m_trackId);
 
-     VFolderMenu::SubMenu *kdeMenu = g_vfolder->parseMenu("applications.menu", true);
+     VFolderMenu::SubMenu *kdeMenu = g_vfolder->parseMenu("kde-applications.menu", true);
 
      KServiceGroup::Ptr entry = g_bsgf->addNew("/", kdeMenu->directoryFile, KServiceGroup::Ptr(), false);
      entry->setLayoutInfo(kdeMenu->layoutList);
