--- kdecore/localization/kcatalog.cpp
+++ kdecore/localization/kcatalog.cpp
@@ -193,12 +193,14 @@ void KCatalogPrivate::setupGettextEnv ()
     //kDebug() << "bindtextdomain" << name << localeDir;
     bindtextdomain(name, localeDir);
 
+#if defined(__GLIBC__)
     // Magic to make sure Gettext doesn't use stale cached translation
     // from previous language.
 #ifndef _MSC_VER
     extern int _nl_msg_cat_cntr;
 #endif
     ++_nl_msg_cat_cntr;
+#endif
   }
 }
 
