--- alsaMixer/src/applet-struct.h.orig	2017-04-28 15:38:01.448075750 +0200
+++ alsaMixer/src/applet-struct.h	2017-04-28 15:38:10.281059992 +0200
@@ -23,6 +23,8 @@
 #include <sys/types.h>
 #include <alsa/asoundlib.h>
 #define _STRUCT_TIMEVAL
+#define __timeval_defined 1
+#define __timespec_defined 1
 #include <cairo-dock.h>
 
 #ifdef INDICATOR_SOUNDMENU_WITH_IND3
--- alsaMixer/src/applet-notifications.c.orig	2017-04-28 15:48:09.445991115 +0200
+++ alsaMixer/src/applet-notifications.c	2017-04-28 15:48:21.549969521 +0200
@@ -17,6 +17,7 @@
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
 
+#define __timespec_defined 1
 #include <glib/gi18n.h>
 
 #include "applet-struct.h"

--- alsaMixer/src/applet-draw.c.orig	2017-04-28 15:51:33.625626869 +0200
+++ alsaMixer/src/applet-draw.c	2017-04-28 15:51:15.744658767 +0200
@@ -19,6 +19,7 @@
 
 #include <stdlib.h>
 #include <string.h>
+#define __timespec_defined 1
 #include <glib/gi18n.h>
 
 #include "applet-struct.h"

--- alsaMixer/src/applet-backend-alsamixer.c.orig	2017-04-28 15:56:14.587125649 +0200
+++ alsaMixer/src/applet-backend-alsamixer.c	2017-04-28 15:55:56.683157588 +0200
@@ -20,6 +20,7 @@
 #include <stdlib.h>
 #include <math.h>
 #include <string.h>
+#define __timespec_defined 1
 #include <glib/gi18n.h>
 
 #include "applet-struct.h"

--- Sound-Effects/src/applet-struct.h.orig	2017-04-28 16:00:52.732629452 +0200
+++ Sound-Effects/src/applet-struct.h	2017-04-28 16:00:36.919657661 +0200
@@ -24,6 +24,8 @@
 #include <sys/types.h>
 #include <alsa/asoundlib.h>
 #define _STRUCT_TIMEVAL
+#define __timeval_defined 1
+#define __timespec_defined 1
 #include <cairo-dock.h>
 
 //\___________ structure containing the applet's configuration parameters.

