From 274b58cef2501ff45e18e8ba98ce7a2e0f1478a3 Mon Sep 17 00:00:00 2001
From: Alan Lacerda <alacerda@alpinelinux.org>
Date: Fri, 17 Apr 2015 20:02:16 +0000
Subject: [PATCH] get_nprocs patch

---
 src/gpm-load.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/gpm-load.c b/src/gpm-load.c
index 3b0f572..5e78cb7 100644
--- a/src/gpm-load.c
+++ b/src/gpm-load.c
@@ -29,9 +29,9 @@
 #include <string.h>
 #include <sys/time.h>
 #include <sys/types.h>
+#include <sys/sysinfo.h>
 #if defined(sun) && defined(__SVR4)
 #include <kstat.h>
-#include <sys/sysinfo.h>
 #endif
 #ifdef HAVE_UNISTD_H
 #include <unistd.h>
-- 
2.3.3

