--- a/src/egg-debug.c	2015-11-07 07:50:36.623167156 +0100
+++ b/src/egg-debug.c	2015-11-07 07:51:46.719072134 +0100
@@ -39,7 +39,9 @@
 #include <sys/stat.h>
 #include <fcntl.h>
 #include <time.h>
+#ifdef __GLIBC__
 #include <execinfo.h>
+#endif
 
 #include "egg-debug.h"
 
@@ -78,6 +80,7 @@ pk_set_console_mode (guint console_code)
 void
 egg_debug_backtrace (void)
 {
+#ifdef __GLIBC__
 	void *call_stack[512];
 	int  call_stack_size;
 	char **symbols;
@@ -95,6 +98,7 @@ egg_debug_backtrace (void)
 		pk_set_console_mode (CONSOLE_RESET);
 		free (symbols);
 	}
+#endif
 }
 
 /**
--- a/applets/brightness/egg-debug.c	2015-11-07 07:52:33.320008962 +0100
+++ b/applets/brightness/egg-debug.c	2015-11-07 07:53:09.302960183 +0100
@@ -39,7 +39,9 @@
 #include <sys/stat.h>
 #include <fcntl.h>
 #include <time.h>
+#ifdef __GLIBC__
 #include <execinfo.h>
+#endif
 
 #include "egg-debug.h"
 
@@ -78,6 +80,7 @@ pk_set_console_mode (guint console_code)
 void
 egg_debug_backtrace (void)
 {
+#ifdef __GLIBC__
 	void *call_stack[512];
 	int  call_stack_size;
 	char **symbols;
@@ -95,6 +98,7 @@ egg_debug_backtrace (void)
 		pk_set_console_mode (CONSOLE_RESET);
 		free (symbols);
 	}
+#endif
 }
 
 /**
--- a/applets/inhibit/egg-debug.c	2015-11-07 07:53:32.815928309 +0100
+++ b/applets/inhibit/egg-debug.c	2015-11-07 07:53:55.503897553 +0100
@@ -39,7 +39,9 @@
 #include <sys/stat.h>
 #include <fcntl.h>
 #include <time.h>
+#ifdef __GLIBC__
 #include <execinfo.h>
+#endif
 
 #include "egg-debug.h"
 
@@ -78,6 +80,7 @@ pk_set_console_mode (guint console_code)
 void
 egg_debug_backtrace (void)
 {
+#ifdef __GLIBC__
 	void *call_stack[512];
 	int  call_stack_size;
 	char **symbols;
@@ -95,6 +98,7 @@ egg_debug_backtrace (void)
 		pk_set_console_mode (CONSOLE_RESET);
 		free (symbols);
 	}
+#endif
 }
 
 /**
