From e49a4bfc495e347834f7b3e8fc3a1501a0b313d7 Mon Sep 17 00:00:00 2001
From: q66 <daniel@octaforge.org>
Date: Fri, 9 Jul 2021 02:28:55 +0200
Subject: [PATCH] [PATCH] fix on platforms without 64-bit atomics

---
 src/aggregate.rs | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/aggregate.rs b/src/aggregate.rs
index 1255515..9d0f266 100644
--- a/src/aggregate.rs
+++ b/src/aggregate.rs
@@ -6,7 +6,7 @@ use std::{borrow::Cow, io, path::Path};
 #[cfg(feature = "aggregate-scan-progress")]
 use std::{
     sync::{
-        atomic::{AtomicU64, Ordering},
+        atomic::{AtomicUsize, Ordering},
         Arc,
     },
     thread,
@@ -37,7 +37,7 @@ pub fn aggregate(
     let mut inodes = InodeFilter::default();
 
     #[cfg(feature = "aggregate-scan-progress")]
-    let shared_count = Arc::new(AtomicU64::new(0));
+    let shared_count = Arc::new(AtomicUsize::new(0));
 
     #[cfg(feature = "aggregate-scan-progress")]
     if let Some(mut out) = err {
-- 
2.33.1

