--- Makefile.orig	2008-10-13 00:36:54.000000000 +0200
+++ Makefile	2008-10-13 00:37:03.000000000 +0200
@@ -691,9 +691,9 @@ ifeq ($(uname_S),NetBSD)
 	ifeq ($(shell expr "$(uname_R)" : '[01]\.'),2)
 		NEEDS_LIBICONV = YesPlease
 	endif
-	BASIC_CFLAGS += -I/usr/pkg/include
-	BASIC_LDFLAGS += -L/usr/pkg/lib
-	ALL_LDFLAGS += -Wl,-rpath,/usr/pkg/lib
+	#BASIC_CFLAGS += -I/usr/pkg/include
+	#BASIC_LDFLAGS += -L/usr/pkg/lib
+	#ALL_LDFLAGS += -Wl,-rpath,/usr/pkg/lib
 endif
 ifeq ($(uname_S),AIX)
 	NO_STRCASESTR=YesPlease
