--- makefile.cfg.orig	1998-03-21 20:08:57.000000000 +0100
+++ makefile.cfg	2008-10-02 18:35:21.000000000 +0200
@@ -191,6 +191,8 @@ wrjpgcom: wrjpgcom.$(O)
 # Installation rules:
 
 install: cjpeg djpeg jpegtran rdjpgcom wrjpgcom @FORCE_INSTALL_LIB@
+	$(INSTALL_DATA) -m 755 -d $(bindir)
+	$(INSTALL_DATA) -m 755 -d $(mandir)
 	$(INSTALL_PROGRAM) cjpeg $(bindir)/$(binprefix)cjpeg
 	$(INSTALL_PROGRAM) djpeg $(bindir)/$(binprefix)djpeg
 	$(INSTALL_PROGRAM) jpegtran $(bindir)/$(binprefix)jpegtran
@@ -203,9 +205,11 @@ install: cjpeg djpeg jpegtran rdjpgcom w
 	$(INSTALL_DATA) $(srcdir)/wrjpgcom.1 $(mandir)/$(manprefix)wrjpgcom.$(manext)
 
 install-lib: libjpeg.$(A) install-headers
+	$(INSTALL_DATA) -m 755 -d $(libdir)
 	$(INSTALL_LIB) libjpeg.$(A) $(libdir)/$(binprefix)libjpeg.$(A)
 
 install-headers: jconfig.h
+	$(INSTALL_DATA) -m 755 -d $(includedir)
 	$(INSTALL_DATA) jconfig.h $(includedir)/jconfig.h
 	$(INSTALL_DATA) $(srcdir)/jpeglib.h $(includedir)/jpeglib.h
 	$(INSTALL_DATA) $(srcdir)/jmorecfg.h $(includedir)/jmorecfg.h
