diff --git lib/Driver/ToolChains.cpp lib/Driver/ToolChains.cpp
index f789fd5..07fc182 100644
--- lib/Driver/ToolChains.cpp
+++ lib/Driver/ToolChains.cpp
@@ -3385,6 +3385,10 @@ void Linux::AddClangCXXStdlibIncludeArgs(const ArgList &DriverArgs,
   }
 }
 
+ToolChain::RuntimeLibType Linux::GetDefaultRuntimeLibType() const {
+  return ToolChain::RLT_CompilerRT;
+}
+
 bool Linux::isPIEDefault() const {
   return getSanitizerArgs().requiresPIE();
 }
diff --git lib/Driver/ToolChains.h lib/Driver/ToolChains.h
index 47fb10d..88aee51 100644
--- lib/Driver/ToolChains.h
+++ lib/Driver/ToolChains.h
@@ -640,6 +640,7 @@ public:
   void
   AddClangCXXStdlibIncludeArgs(const llvm::opt::ArgList &DriverArgs,
                                llvm::opt::ArgStringList &CC1Args) const override;
+  RuntimeLibType GetDefaultRuntimeLibType() const override;
   bool isPIEDefault() const override;
 
   std::string Linker;
