--- lib/Driver/Tools.cpp	2015-08-11 13:59:22.324639527 +0200
+++ lib/Driver/Tools.cpp	2015-08-11 13:59:34.058576064 +0200
@@ -2158,8 +2158,6 @@
   CmdArgs.push_back(Args.MakeArgString(getCompilerRT(TC, "builtins")));
 
   if (!TC.getTriple().isOSWindows()) {
-    // FIXME: why do we link against gcc when we are using compiler-rt?
-    CmdArgs.push_back("-lgcc_s");
     if (TC.getDriver().CCCIsCXX())
       CmdArgs.push_back("-lgcc_eh");
   }
