--- a/scripts/install-sh	2011-01-13 10:34:38.853561570 +0100
+++ b/scripts/install-sh	2011-01-13 10:35:04.468525074 +0100
@@ -91,7 +91,6 @@ while ${MORETODO} ; do
     X-G)
 	GROUP="$2"
 	shift
-	${ROOT} && CHGROUPIT=true
 	;;
     X-G*)
 	if ${ROOT} ; then
@@ -110,12 +109,10 @@ while ${MORETODO} ; do
 	;;
     X-M)
 	MODE="$2"
-	${ROOT} && CHMODIT=true
 	shift
 	;;
     X-M*)
 	MODE="`echo \"$1\" | sed 's/^..//'`"
-	${ROOT} && CHMODIT=true
 	;;
     X-n)
 	TOUCHIT=false
@@ -132,7 +129,6 @@ while ${MORETODO} ; do
     X-O)
 	OWNER="$2"
 	shift
-	${ROOT} && CHOWNIT=true
 	;;
     X-O*)
 	if ${ROOT} ; then
