diff --git a/read.c b/read.c
index 05f7b4fc7b76508912e537c7b1026ab1cdef7eab..9f5af3639c47b304df74968e7bd3516dd0b44fdc 100644
--- a/read.c
+++ b/read.c
@@ -255,20 +255,20 @@ readFile(char *filename, FontPtr font)
         if(verbose_flag)
             fprintf(stderr, "size %d: %dx%d\n",
                     i, 
-                    (int)(face->available_sizes[i].x_ppem >> 6), 
-                    (int)(face->available_sizes[i].y_ppem >> 6));
+                    (int)((face->available_sizes[i].x_ppem + 32) >> 6),
+                    (int)((face->available_sizes[i].y_ppem + 32) >> 6));
 
         rc = FT_Set_Pixel_Sizes(face,
-                                face->available_sizes[i].x_ppem >> 6,
-                                face->available_sizes[i].y_ppem >> 6);
+                                (face->available_sizes[i].x_ppem + 32) >> 6,
+                                (face->available_sizes[i].y_ppem + 32) >> 6);
         if(rc != 0) {
             fprintf(stderr, "Couldn't set size.\n");
             return -1;
         }
 
         strike = makeStrike(font, 
-                            face->available_sizes[i].x_ppem >> 6,
-                            face->available_sizes[i].y_ppem >> 6);
+                            (face->available_sizes[i].x_ppem + 32) >> 6,
+                            (face->available_sizes[i].y_ppem + 32) >> 6);
         if(strike == NULL) {
             fprintf(stderr, "Couldn't allocate strike.\n");
             return -1;

