--- Makefile.orig	2019-11-14 09:48:03.854735606 +0100
+++ Makefile	2019-11-14 09:58:50.671671457 +0100
@@ -118,9 +118,9 @@
 	D_BUILD_ENV	= -DBUILD_ENV_UNIX
 	PCAL		= pcal
 	CC		= /usr/bin/gcc
-	PACK		= compress
+	# PACK		= compress
 	# PACK		= pack
-	# PACK		= gzip
+	PACK		= gzip -f
 endif
 
 # 
@@ -147,9 +147,9 @@
 	MANDIR = $(DJDIR)/man/man1
 	CATDIR = $(DJDIR)/man/cat1
 else   # Unix
-	BINDIR = /usr/local/bin
-	MANDIR = /usr/man/man1
-	CATDIR = /usr/man/cat1
+	BINDIR = /usr/bin
+	MANDIR = /usr/share/man/man1
+	CATDIR = /usr/share/man/cat1
 endif
 
 OBJECTS = $(OBJDIR)/pcal.o \
