--- a/src/ucon64.c
+++ b/src/ucon64.c
@@ -2123,7 +2123,7 @@
 #elif   defined __APPLE__                       // Mac OS X actually
     "libdiscmage.dylib";
 #elif   defined __unix__ || defined __BEOS__
-    "libdiscmage.so";
+    "/usr/lib/libdiscmage.so";
 #else
     "library";
 #endif
--- a/src/ucon64_misc.c
+++ b/src/ucon64_misc.c
@@ -580,7 +580,7 @@
 {
   uint32_t version;
 #ifdef  DLOPEN
-  const char *p = get_property (ucon64.configfile, "discmage_path", PROPERTY_MODE_FILENAME);
+  const char *p = "/usr/lib/libdiscmage.so";
   if (p)
     strcpy (ucon64.discmage_path, p);
   else
@@ -1229,7 +1229,7 @@
 #elif   defined __APPLE__                       // Mac OS X actually
                        PROPERTY_MODE_DIR ("ucon64") "discmage.dylib",
 #elif   defined __unix__ || defined __BEOS__
-                       PROPERTY_MODE_DIR ("ucon64") "discmage.so",
+                       "/usr/lib/libdiscmage.so",
 #else
                        "",
 #endif
