--- a/src/Makefile
+++ b/src/Makefile
@@ -11,7 +11,7 @@
 CFLAGS0:=$(CFLAGS)
 include config.mak
 
-CFLAGS+=-I. -Wall -W -Wno-implicit-fallthrough -O3 -fsigned-char
+CFLAGS+=-I. -Wall -W -Wno-implicit-fallthrough -fsigned-char
 ifneq ($(findstring clang,$(CC)),clang)
 CFLAGS+=-Wno-format-truncation -Wno-stringop-overflow
 endif
--- a/src/backup/libcd64/Makefile
+++ b/src/backup/libcd64/Makefile
@@ -10,7 +10,7 @@
 ifdef DEBUG
 CFLAGS+=-pg -g -pedantic -ansi -DDEBUG
 else
-CFLAGS+=-O3 -funroll-loops
+CFLAGS+=-funroll-loops
 ifneq ($(findstring clang,$(CC)),clang)
 CFLAGS+=-fexpensive-optimizations
 endif
--- a/src/libdiscmage/Makefile.in
+++ b/src/libdiscmage/Makefile.in
@@ -23,7 +23,7 @@
 ifeq ($(CC),cc)
 CC=@CC@
 endif
-CFLAGS0:=$(CFLAGS) -I. -Wall -W -Wno-implicit-fallthrough -O3 @DEFS@
+CFLAGS0:=$(CFLAGS) -I. -Wall -W -Wno-implicit-fallthrough @DEFS@
 ifneq ($(findstring clang,$(CC)),clang)
 CFLAGS0+=-Wno-format-truncation -Wno-stringop-overflow
 endif
