From 51bd18034c510a6d4711823e08b69af2f7aa063e Mon Sep 17 00:00:00 2001
From: Vincent Sanders <vincent.sanders@collabora.co.uk>
Date: Fri, 03 May 2013 10:39:49 +0000
Subject: fix building teh lexers with bison versions at and after 2.6

---
diff --git a/src/nsgenbind-lexer.l b/src/nsgenbind-lexer.l
index b257c83..8189a72 100644
--- src/nsgenbind-lexer.l
+++ src/nsgenbind-lexer.l
@@ -19,6 +19,16 @@
     yylloc->first_column = yylloc->last_column + 1; \
     yylloc->last_column += yyleng;
 
+
+/* Ensure compatability with bison 2.6 and later */
+#if ! defined YYSTYPE && ! defined YYSTYPE_IS_DECLARED && defined NSGENBIND_STYPE_IS_DECLARED
+#define YYSTYPE NSGENBIND_STYPE
+#endif
+
+#if ! defined YYLTYPE && ! defined YYLTYPE_IS_DECLARED && defined NSGENBIND_LTYPE_IS_DECLARED
+#define YYLTYPE NSGENBIND_LTYPE
+#endif
+
 %}
 
 /* lexer options */
diff --git a/src/webidl-lexer.l b/src/webidl-lexer.l
index 5369d99..74b9bb8 100644
--- src/webidl-lexer.l
+++ src/webidl-lexer.l
@@ -27,6 +27,15 @@
     yylloc->first_column = yylloc->last_column + 1; \
     yylloc->last_column += yyleng;
 
+/* Ensure compatability with bison 2.6 and later */
+#if ! defined YYSTYPE && ! defined YYSTYPE_IS_DECLARED && defined WEBIDL_STYPE_IS_DECLARED
+#define YYSTYPE WEBIDL_STYPE
+#endif
+
+#if ! defined YYLTYPE && ! defined YYLTYPE_IS_DECLARED && defined WEBIDL_LTYPE_IS_DECLARED
+#define YYLTYPE WEBIDL_LTYPE
+#endif
+
 %}
 
 
--
cgit v0.9.0.3-65-g4555
