--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -9,11 +9,12 @@
     endif()
 endforeach()

-foreach(hwpolicy CMP0026 CMP0068)
-    if(POLICY ${hwpolicy})
-        cmake_policy(SET ${hwpolicy} OLD)
-    endif()
-endforeach()
+# doesnt work anymore :(
+#foreach(hwpolicy CMP0026 CMP0068)
+#    if(POLICY ${hwpolicy})
+#        cmake_policy(SET ${hwpolicy} OLD)
+#    endif()
+#endforeach()

 set(CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake_modules")
 include(${CMAKE_MODULE_PATH}/utils.cmake)

--- a/misc/libphyslayer/CMakeLists.txt
+++ b/misc/libphyslayer/CMakeLists.txt
@@ -21,7 +21,7 @@
 install(TARGETS physlayer RUNTIME DESTINATION ${target_binary_install_dir}
                           LIBRARY DESTINATION ${target_library_install_dir}
                           ARCHIVE DESTINATION ${target_library_install_dir})
-get_target_property(physlayer_fullpath physlayer LOCATION)
+get_target_property(physlayer_fullpath physlayer CMAKE_CURRENT_SOURCE_DIR)


 ## added standard variables (FORCE or cmake won't pick 'em)
