--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -65,29 +65,9 @@ if (WIN32)
 
 else(WIN32)
 
-try_run(RUN_RESULT_2 COMPILE_SUCCESS_2 ${CMAKE_CURRENT_BINARY_DIR} ${config_SRCS} COMPILE_DEFINITIONS -DTEST_DECLARE_QSORT_R=ON)
-if(COMPILE_SUCCESS_2 AND (RUN_RESULT_2 EQUAL 0))
-    SET(VAR_2 0)
-else()
-    SET(VAR_2 1)
-endif()
-file(APPEND "${config_FN}" "#define NEED_DECLARE_QSORT_R ${VAR_2}\n")
-
-try_run(RUN_RESULT_3 COMPILE_SUCCESS_3 ${CMAKE_CURRENT_BINARY_DIR} ${config_SRCS} COMPILE_DEFINITIONS -DTEST_QSORT_R=ON)
-if(COMPILE_SUCCESS_3 AND (RUN_RESULT_3 EQUAL 0))
-    SET(VAR_3 1)
-else()
-    SET(VAR_3 0)
-endif()
-file(APPEND "${config_FN}" "#define NEED_QSORT_R ${VAR_3}\n")
-
-try_run(RUN_RESULT_4 COMPILE_SUCCESS_4 ${CMAKE_CURRENT_BINARY_DIR} ${config_SRCS} COMPILE_DEFINITIONS -DTEST_SWAP_QSORT_R=ON)
-if(COMPILE_SUCCESS_4 AND (RUN_RESULT_4 EQUAL 0))
-    SET(VAR_4 1)
-else()
-    SET(VAR_4 0)
-endif()
-file(APPEND "${config_FN}" "#define NEED_SWAP_QSORT_R ${VAR_4}\n")
+file(APPEND "${config_FN}" "#define NEED_DECLARE_QSORT_R 0\n")
+file(APPEND "${config_FN}" "#define NEED_QSORT_R 0\n")
+file(APPEND "${config_FN}" "#define NEED_SWAP_QSORT_R 1\n")
 
 endif(WIN32)
 
--- a/stellarsolver/astrometry/include/astrometry/ioutils.h
+++ b/stellarsolver/astrometry/include/astrometry/ioutils.h
@@ -28,17 +28,21 @@ char* dirname(const char* path);
 #include "astrometry/an-bool.h"
 #include "astrometry/bl.h"
 #include "astrometry/keywords.h"
+#include "astrometry/os-features.h"
 
 extern uint32_t ENDIAN_DETECTOR;
 
+#if NEED_DECLARE_QSORT_R
 void QSORT_R(void* base, size_t nmembers, size_t member_size,
-             void* token, int (*compar)(void *, const void *, const void *));
+             void* token,
+             int QSORT_COMPARISON_FUNCTION((*compar),void *, const void*, const void*));
 
 /**
    You should define the "comparison" function like this:
    static int QSORT_COMPARISON_FUNCTION(my_comparison, void* token, const void* v1, const void* v2) {
  */
 #define QSORT_COMPARISON_FUNCTION(func, thunk, v1, v2) func(thunk, v1, v2)
+#endif
 
 int copy_file(const char* infn, const char* outfn);
 
--- a/stellarsolver/astrometry/util/ioutils.c
+++ b/stellarsolver/astrometry/util/ioutils.c
@@ -43,7 +43,9 @@
 
 #include "os-features.h"
 #include "ioutils.h"
+#if NEED_QSORT_R
 #include "qsort_reentrant.c"
+#endif
 #include "errors.h"
 #include "log.h"
 
