--- unix/gtkwin.c.orig	2017-04-24 13:52:45.000000000 +0000
+++ unix/gtkwin.c	2017-04-30 22:43:56.901428607 +0000
@@ -2188,11 +2188,11 @@
     free(data);
     free(col_css);
 #else
-    if (gtk_widget_get_window(win)) {
+    if (gtk_widget_get_window(widget)) {
         /* For GTK1, which doesn't have a 'const' on
          * gdk_window_set_background's second parameter type. */
         GdkColor col_mutable = *col;
-        gdk_window_set_background(gtk_widget_get_window(win), &col_mutable);
+        gdk_window_set_background(gtk_widget_get_window(widget), &col_mutable);
     }
 #endif
 }
