--- src/spaceorb.c.orig	2009-03-31 10:32:33.238047887 +0200
+++ src/spaceorb.c	2009-03-31 10:32:43.299024563 +0200
@@ -337,7 +337,7 @@ DeviceInit (DeviceIntPtr dev)
     return !Success;
   }
 
-  if (InitValuatorClassDeviceStruct (dev, 6, xf86GetMotionEvents,
+  if (InitValuatorClassDeviceStruct (dev, 6,
 				     pInfo->history_size, Absolute) == FALSE)
     {
       ErrorF ("Unable to allocate SPACEORB ValuatorClassDeviceStruct\n");
