Disable phoning home:
https://github.com/saulpw/visidata/issues/913

--- a/visidata/motd.py
+++ b/visidata/motd.py
@@ -12,7 +12,7 @@
 
 from visidata import options, asyncsingle, vd, VisiData
 
-vd.option('motd_url', 'https://visidata.org/motd-'+vd.version, 'source of randomized startup messages', sheettype=None)
+vd.option('motd_url', '', 'source of randomized startup messages', sheettype=None)
 
 
 vd.motd = 'Support VisiData: https://github.com/sponsors/saulpw'
