Index: easytag-2.4.3/src/tags/id3lib/c_wrapper.cpp
===================================================================
--- easytag-2.4.3.orig/src/tags/id3lib/c_wrapper.cpp
+++ easytag-2.4.3/src/tags/id3lib/c_wrapper.cpp
@@ -39,10 +39,10 @@ extern "C"
   // Tag wrappers
   //
 
-  ID3_C_EXPORT bool CCONV
+  ID3_C_EXPORT ID3_Bool CCONV
   ID3Field_SetEncoding(ID3Field *field, ID3_TextEnc enc)
   {
-    bool changed = false;
+    ID3_Bool changed = ID3_False;
     if (field)
     {
       ID3_CATCH(changed = reinterpret_cast<ID3_Field *>(field)->SetEncoding(enc));
@@ -61,10 +61,10 @@ extern "C"
     return enc;
   }
 
-  ID3_C_EXPORT bool CCONV
+  ID3_C_EXPORT ID3_Bool CCONV
   ID3Field_IsEncodable(const ID3Field *field)
   {
-    bool isEncodable = false;
+    ID3_Bool isEncodable = ID3_False;
     if (field)
     {
       ID3_CATCH(isEncodable = reinterpret_cast<const ID3_Field *>(field)->IsEncodable());
Index: easytag-2.4.3/src/tags/id3lib/id3_bugfix.h
===================================================================
--- easytag-2.4.3.orig/src/tags/id3lib/id3_bugfix.h
+++ easytag-2.4.3/src/tags/id3lib/id3_bugfix.h
@@ -29,13 +29,13 @@
 G_BEGIN_DECLS
 
 #if !HAVE_DECL_ID3FIELD_SETENCODING
-ID3_C_EXPORT bool                  CCONV ID3Field_SetEncoding    (ID3Field *field, ID3_TextEnc enc);
+ID3_C_EXPORT ID3_Bool                  CCONV ID3Field_SetEncoding    (ID3Field *field, ID3_TextEnc enc);
 #endif /* !HAVE_DECL_ID3FIELD_SETENCODING */
 #if !HAVE_DECL_ID3FIELD_GETENCODING
 ID3_C_EXPORT ID3_TextEnc           CCONV ID3Field_GetEncoding    (const ID3Field *field);
 #endif /* !HAVE_DECL_ID3FIELD_GETENCODING */
 #if !HAVE_DECL_ID3FIELD_ISENCODABLE
-ID3_C_EXPORT bool                  CCONV ID3Field_IsEncodable    (const ID3Field *field);
+ID3_C_EXPORT ID3_Bool                  CCONV ID3Field_IsEncodable    (const ID3Field *field);
 #endif /* !HAVE_DECL_ID3FIELD_ISENCODABLE */
 ID3_C_EXPORT ID3_FieldType         CCONV ID3Field_GetType        (const ID3Field *field);
 //ID3_C_EXPORT ID3_FieldID           CCONV ID3Field_GetID          (const ID3Field *field);
