diff --git Makefile.am Makefile.am
index 435bc87..a599603 100644
--- Makefile.am
+++ Makefile.am
@@ -134,14 +134,14 @@ noinst_PROGRAMS =
 if DBUS_POLICY
 dbus_datadir = @DBUS_DATADIR@/dbus-1/system.d
 dist_dbus_data_DATA =
+
+dbus_busdir = @DBUS_BUSDIR@
+dbus_bus_DATA =
 endif
 
 if SYSTEMD_SERVICE
 systemd_unitdir = @SYSTEMD_UNITDIR@
 systemd_unit_DATA =
-
-dbus_busdir = @DBUS_BUSDIR@
-dbus_bus_DATA =
 endif
 
 eap_sources = src/eap.c src/eap.h src/eap-private.h \
@@ -212,13 +212,13 @@ endif
 
 if DBUS_POLICY
 dist_dbus_data_DATA += src/iwd-dbus.conf
+dbus_bus_DATA += src/net.connman.iwd.service
 endif
 
 if SYSTEMD_SERVICE
 src_iwd_DEPENDENCIES += src/iwd.service
 
 systemd_unit_DATA += src/iwd.service
-dbus_bus_DATA += src/net.connman.iwd.service
 endif
 endif
 
@@ -271,13 +271,13 @@ wired_ead_DEPENDENCIES = $(ell_dependencies)
 
 if DBUS_POLICY
 dist_dbus_data_DATA += wired/ead-dbus.conf
+dbus_bus_DATA += wired/net.connman.ead.service
 endif
 
 if SYSTEMD_SERVICE
 wired_ead_DEPENDENCIES += wired/ead.service
 
 systemd_unit_DATA += wired/ead.service
-dbus_bus_DATA += wired/net.connman.ead.service
 endif
 endif
 

