--- src/resolve.c	2020-09-05 09:43:41.000000000 +0200
+++ -	2020-10-27 20:27:39.193991880 +0100
@@ -554,7 +554,7 @@
 			l_warn("[General].dns_resolve_method is deprecated, "
 				"use [Network].NameResolvingService");
 		else /* Default to systemd-resolved service. */
-			method_name = "systemd";
+			method_name = "resolvconf";
 	}
 
 	for (i = 0; resolve_method_ops_list[i].name; i++) {
--- src/iwd.config.5	2020-09-05 09:47:11.000000000 +0200
+++ -	2020-11-01 22:37:16.517573954 +0100
@@ -198,7 +198,7 @@
 \fBEnableNetworkConfiguration\fP and provides the choice of system
 resolver integration.
 .sp
-If not specified, \fBsystemd\fP is used as default.
+If not specified, \fBresolvconf\fP is used as default.
 T}
 _
 T{
