commit b70919e0d9dadc93893e9d18bc3ef13b88756ecf
Author: dana <dana@dana.is>
Date:   Sun Jan 27 00:26:31 2019 -0600

    44020: VCS_INFO_detect_p4: Fix infinite recursion

diff --git ChangeLog ChangeLog
index 81668ccc9..be243c1b8 100644
--- ChangeLog
+++ ChangeLog
@@ -1,3 +1,8 @@
+2018-01-27  dana  <dana@dana.is>
+
+	* 44020: Functions/VCS_Info/Backends/VCS_INFO_detect_p4: Fix
+	infinite recursion
+
 2018-01-24  dana  <dana@dana.is>
 
 	* unposted: Config/version.mk: Post-release version bump
diff --git Functions/VCS_Info/Backends/VCS_INFO_detect_p4 Functions/VCS_Info/Backends/VCS_INFO_detect_p4
index d171c68ee..5ec21da9f 100644
--- Functions/VCS_Info/Backends/VCS_INFO_detect_p4
+++ Functions/VCS_Info/Backends/VCS_INFO_detect_p4
@@ -44,7 +44,6 @@ VCS_INFO_p4_get_server() {
 }
 
 
-(( ${+functions[VCS_INFO_detect_p4]} )) ||
 VCS_INFO_detect_p4() {
   local serverport p4where
 
