--- toonz/sources/common/tcore/tstopwatch.cpp.orig
+++ toonz/sources/common/tcore/tstopwatch.cpp
@@ -11,7 +11,7 @@
 #if defined(__APPLE_CC__)
 #include <unistd.h>
 #else
-#include <sys/unistd.h>
+#include <unistd.h>
 #endif
 #include <limits.h>
 #include <sys/times.h>
--- toonz/sources/toonzfarm/tfarmserver/tfarmserver.cpp.orig
+++ toonz/sources/toonzfarm/tfarmserver/tfarmserver.cpp
@@ -51,7 +51,7 @@
 #endif
 
 #ifdef LINUX
-#include <sys/sysctl.h>
+#include <linux/sysctl.h>
 #endif
 
 // forward declaration
