--- src/include/xm_lua.h	2011-10-11 22:18:17.000000000 +0200
+++ src/include/xm_lua.h	2015-10-02 11:44:15.981530419 +0200
@@ -4,7 +4,15 @@
     #include "lauxlib.h"
     #include "lualib.h"
   #else
-    #if defined(HAVE_LUA5_1_LUA_H)
+    #if defined(HAVE_LUA5_2_LUA_H)
+      #include <lua5.2/lua.h>
+      #include <lua5.2/lauxlib.h>
+      #include <lua5.2/lualib.h>
+    #elif defined(HAVE_LUA52_LUA_H)
+      #include <lua52/lua.h>
+      #include <lua52/lauxlib.h>
+      #include <lua52/lualib.h>      
+    #elif defined(HAVE_LUA5_1_LUA_H)
       #include <lua5.1/lua.h>
       #include <lua5.1/lauxlib.h>
       #include <lua5.1/lualib.h>
