--- src/net/NetActions.cpp.orig	2018-01-16 11:03:55.152026167 +0100
+++ src/net/NetActions.cpp	2018-01-16 11:03:59.910007186 +0100
@@ -30,7 +30,7 @@ Foundation, Inc., 59 Temple Place, Suite
 #include <sstream>
 #include "../helpers/SwapEndian.h"
 #include "helpers/Net.h"
-#include "helpers/utf8.h"
+#include "../helpers/utf8.h"
 
 char NetAction::m_buffer[NETACTION_MAX_PACKET_SIZE];
 unsigned int NetAction::m_biggestTCPPacketSent = 0;
--- src/helpers/utf8.h.orig	2018-01-16 11:08:10.042009362 +0100
+++ src/helpers/utf8.h	2018-01-16 11:08:17.306980381 +0100
@@ -18,8 +18,8 @@ along with XMOTO; if not, write to the F
 Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 =============================================================================*/
 
-#ifndef __UTF8_H__
-#define __UTF8_H__
+#ifndef __MYUTF8_H__
+#define __MYUTF8_H__
 
 #include <vector>
 #include <string>

