diff --git a/meson.build b/meson.build
index d7062af..93aae1a 100644
--- meson.build
+++ meson.build
@@ -2,7 +2,7 @@
 # project definition
 #
 project('spice-gtk', 'c',
-         version : run_command('build-aux/git-version-gen', '${MESON_SOURCE_ROOT}/.tarball-version', check : true).stdout().strip(),
+		 version : '0.36',
          license : 'LGPLv2.1',
          meson_version : '>= 0.49')
 
diff --git a/src/meson.build b/src/meson.build
index d9614cb..2378c83 100644
--- src/meson.build
+++ src/meson.build
@@ -8,10 +8,7 @@ spice_gtk_include += [include_directories('.')]
 version_info = meson.project_version().split('.')
 major = '@0@'.format(version_info[0])
 minor = '@0@'.format(version_info[1])
-micro = version_info[2].split('-')[0]
-if micro == ''
-  micro = '0'
-endif
+micro = '0'
 version_data = configuration_data()
 version_data.set('SPICE_GTK_MAJOR_VERSION', major)
 version_data.set('SPICE_GTK_MINOR_VERSION', minor)

