diff --git a/src/meson.build b/src/meson.build
index c55db44..667582b 100644
--- src/meson.build
+++ src/meson.build
@@ -203,8 +203,8 @@ spice_client_glib_dep = declare_dependency(sources : [spice_marshals[1], spice_c
 #
 # SpiceClientGLib-2.0.gir
 #
+if spice_gtk_has_introspection
 spice_client_glib_gir = gnome.generate_gir(spice_client_glib_lib,
-                                           build_by_default : spice_gtk_has_introspection,
                                            export_packages : 'spice-client-glib-2.0',
                                            extra_args : ['--accept-unprefixed'],
                                            header : 'spice-client.h',
@@ -217,7 +217,7 @@ spice_client_glib_gir = gnome.generate_gir(spice_client_glib_lib,
                                            namespace : 'SpiceClientGLib',
                                            nsversion : '2.0',
                                            sources : spice_client_glib_introspection_sources)
-
+endif
 #
 # spice-client-glib-usb-acl-helper
 #
@@ -340,11 +340,11 @@ if spice_gtk_has_gtk
                                             link_with : spice_client_gtk_lib,
                                             dependencies : [spice_client_glib_dep, spice_gtk_deps])
 
+if spice_gtk_has_introspection
   #
   # SpiceClientGtk-3.0.gir
   #
   spice_client_gtk_gir = gnome.generate_gir(spice_client_gtk_lib,
-                                            build_by_default : spice_gtk_has_introspection,
                                             export_packages : 'spice-client-gtk-3.0',
                                             extra_args : ['--accept-unprefixed'],
                                             header : 'spice-widget.h',
@@ -358,3 +358,4 @@ if spice_gtk_has_gtk
                                             nsversion : '3.0',
                                             sources : spice_client_gtk_introspection_sources)
 endif
+endif
