From 70e8b9f2bc718eeb5642a6680fd0525baaadc6c1 Mon Sep 17 00:00:00 2001
From: Nathan Owens <ndowens@artixlinux.org>
Date: Tue, 1 Dec 2020 14:40:31 -0600
Subject: vala-0.50 fix

---
 valabindwriter.vala | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git valabindwriter.vala valabindwriter.vala
index 5b71abd..d9e36fd 100644
--- a/valabindwriter.vala
+++ b/valabindwriter.vala
@@ -36,7 +36,7 @@ public class ValabindWriter : CodeVisitor {
 #if VALA_0_50
 		// required to avoid ugly runtime errors
 #else
-		context.profile = Profile.GOBJECT;
+	context.set_target_profile(Profile.GOBJECT);
 #endif
 	}
 
-- 
2.29.2

