--- a/meson.build
+++ b/meson.build
@@ -26,7 +26,8 @@ libvala_dep = dependency('libvala-@0@'.format(libvala_version))
 glib_dep = dependency('glib-2.0')
 gobject_dep = dependency('gobject-2.0')
 
-valaccodegen_dep = cc.find_library('valaccodegen', dirs : [libvala_dep.get_pkgconfig_variable('pkglibdir')])
+cc_sysroot = run_command(cc.cmd_array()[0], '-print-sysroot').stdout().strip()
+valaccodegen_dep = cc.find_library('valaccodegen', dirs : [cc_sysroot + libvala_dep.get_pkgconfig_variable('pkglibdir')])
 
 executable(
   'valabind',
