From f942315402e3fc38def14a974dd5b210e5d2c38d Mon Sep 17 00:00:00 2001
From: Daniel Kolesa <daniel@octaforge.org>
Date: Wed, 16 Sep 2020 20:39:29 +0200
Subject: [PATCH] fix with vala-0.50

---
 valabindwriter.vala | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git valabindwriter.vala valabindwriter.vala
index 165cdf5..c793c9f 100644
--- valabindwriter.vala
+++ valabindwriter.vala
@@ -33,7 +33,7 @@ public class ValabindWriter : CodeVisitor {
 		if (glibmode)
 			context.add_define ("GOBJECT");
 		// required to avoid ugly runtime errors
-		context.profile = Profile.GOBJECT;
+		context.set_target_profile(Profile.GOBJECT);
 	}
 
 	public void parse () {
-- 
2.28.0

