--- a/Configure
+++ b/Configure
@@ -107,28 +107,6 @@
 	fi
 fi
 
-: Proper PATH setting
-paths='/bin /usr/bin /usr/local/bin /usr/ucb /usr/local /usr/lbin'
-paths="$paths /opt/bin /opt/local/bin /opt/local /opt/lbin"
-paths="$paths /usr/5bin /etc /usr/gnu/bin /usr/new /usr/new/bin /usr/nbin"
-paths="$paths /opt/gnu/bin /opt/new /opt/new/bin /opt/nbin"
-paths="$paths /sys5.3/bin /sys5.3/usr/bin /bsd4.3/bin /bsd4.3/usr/ucb"
-paths="$paths /bsd4.3/usr/bin /usr/bsd /bsd43/bin /opt/ansic/bin /usr/ccs/bin"
-paths="$paths /etc /usr/lib /usr/ucblib /lib /usr/ccs/lib"
-paths="$paths /sbin /usr/sbin /usr/libexec"
-paths="$paths /system/gnu_library/bin"
-
-for p in $paths
-do
-	case "$p_$PATH$p_" in
-	*$p_$p$p_*) ;;
-	*) test -d $p && PATH=$PATH$p_$p ;;
-	esac
-done
-
-PATH=.$p_$PATH
-export PATH
-
 : shall we be using ksh?
 inksh=''
 needksh=''
@@ -8795,57 +8773,6 @@
 	;;
 esac
 
-# How will the perl executable find the installed shared $libperl?
-# Add $xxx to ccdlflags.
-# If we can't figure out a command-line option, use $shrpenv to
-# set env LD_RUN_PATH.  The main perl makefile uses this.
-shrpdir=$archlibexp/CORE
-xxx=''
-tmp_shrpenv=''
-if "$useshrplib"; then
-    case "$osname" in
-	aix)
-		# We'll set it in Makefile.SH...
-		;;
-	solaris)
-		xxx="-R $shrpdir"
-		;;
-	freebsd|minix|mirbsd|netbsd|openbsd|interix|dragonfly|bitrig)
-		xxx="-Wl,-R$shrpdir"
-		;;
-	bsdos|linux|irix*|dec_osf|gnu*|haiku)
-		xxx="-Wl,-rpath,$shrpdir"
-		;;
-	hpux*)
-		# hpux doesn't like the default, either.
-		tmp_shrpenv="env LDOPTS=\"+s +b${shrpdir}\""
-		;;
-	cygwin)
-		# cygwin needs only ldlibpth
-		;;
-	*)
-		tmp_shrpenv="env LD_RUN_PATH=$shrpdir"
-		;;
-	esac
-	case "$xxx" in
-	'') ;;
-	*)
-		# Only add $xxx if it isn't already in ccdlflags.
-		case " $ccdlflags " in
-		*" $xxx "*)	;;
-		*)	ccdlflags="$ccdlflags $xxx"
-			cat <<EOM >&4
-
-Adding $xxx to the flags
-passed to $ld so that the perl executable will find the
-installed shared $libperl.
-
-EOM
-			;;
-		esac
-		;;
-	esac
-fi
 # Fix ccdlflags in AIX for building external extensions.
 # (For building Perl itself bare -bE:perl.exp is needed,
 #  Makefile.SH takes care of this.)
