https://github.com/arsv/perl-cross/pull/143

diff --git a/cnf/configure_pfmt.sh b/cnf/configure_pfmt.sh
index 8f93da1..3aa7731 100644
--- a/cnf/configure_pfmt.sh
+++ b/cnf/configure_pfmt.sh
@@ -52,3 +52,9 @@ else
 	define uvxformat '"lx"'
 	define uvXUformat '"lX"'
 fi
+
+define i32dformat 'PRId32'
+define u32uformat 'PRIu32'
+define u32oformat 'PRIo32'
+define u32xformat 'PRIx32'
+define u32XUformat 'PRIX32'
