Don't run scrollbar test, it fails on x86.

diff --git a/meson.build b/meson.build
index 6e1105a..f707cd9 100644
--- a/meson.build
+++ b/meson.build
@@ -410,24 +410,6 @@
     dependencies: deps,
 ))

-test('scrollbar test', executable('scrollbar.test', [
-        'test/scrollbar-test.c',
-        theme_parser,
-        theme_lexer,
-        default_theme,
-    ],
-    objects: rofi.extract_objects([
-        'source/widgets/widget.c',
-        'source/widgets/scrollbar.c',
-        'source/theme.c',
-        'source/css-colors.c',
-        'source/rofi-types.c',
-        'source/css-colors.c',
-        'config/config.c',
-    ]),
-    dependencies: deps,
-))
-
 test('textbox test', executable('textbox.test', [
         'test/textbox-test.c',
         theme_parser,

