diff --git a/subprojects/libnkutils/tests/format-string.c b/subprojects/libnkutils/tests/format-string.c
index d00ad2d..cf69b20 100644
--- a/subprojects/libnkutils/tests/format-string.c
+++ b/subprojects/libnkutils/tests/format-string.c
@@ -899,7 +899,7 @@ static const struct {
         .testpath = "/nkutils/name/replace/escaping/forwardslash/3",
         .data = {
             .identifier = '$',
-            .source = "${data/a/\\//b/x}",
+            .source = "${data/ab/\\/x}",
             .data = {
                 { .name = "data", .content = "'ab'" },
                 { .name = NULL }
