From 371da358d1580dc06218d18a12a99611cac39e4e Mon Sep 17 00:00:00 2001
From: Jan Grulich <jgrulich@redhat.com>
Date: Wed, 2 Jan 2019 10:05:40 +0100
Subject: [PATCH] Avoid invalid conversion error with C++ compilators

---
 src/pipewire/utils.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git src/pipewire/utils.h src/pipewire/utils.h
index 2ce9fdde..7262ff42 100644
--- src/pipewire/utils.h
+++ src/pipewire/utils.h
@@ -58,7 +58,7 @@ pw_spa_pod_copy(const struct spa_pod *pod)
 		return NULL;
 
 	size = SPA_POD_SIZE(pod);
-	if ((c = malloc(size)) == NULL)
+	if ((c = (struct spa_pod *) malloc(size)) == NULL)
 		return NULL;
 
 	return (struct spa_pod *) memcpy(c, pod, size);
