From aaa015d0c0c66b64c7198a34ad3c4b43445667d2 Mon Sep 17 00:00:00 2001
From: Wim Taymans <wtaymans@redhat.com>
Date: Tue, 19 Jul 2022 13:06:52 +0200
Subject: [PATCH] avb: fix compilation on big endian

---
 spa/plugins/avb/avbtp/packets.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/spa/plugins/avb/avbtp/packets.h b/spa/plugins/avb/avbtp/packets.h
index 7047456bf..3d4a652ee 100644
--- a/spa/plugins/avb/avbtp/packets.h
+++ b/spa/plugins/avb/avbtp/packets.h
@@ -116,7 +116,7 @@ struct spa_avbtp_packet_aaf {
 	unsigned gv:1;
 	unsigned tv:1;
 
-	uint8_t seq_number;
+	uint8_t seq_num;
 
 	unsigned _r2:7;
 	unsigned tu:1;
