--- src/pipewire/data-loop.c
+++ src/pipewire/data-loop.c
@@ -47,6 +47,7 @@ static void make_realtime(struct pw_data_loop *this)
 	system_bus = pw_rtkit_bus_get_system();
 
 	rl.rlim_cur = rl.rlim_max = rttime;
+#ifdef RLIMIT_RTTIME
 	if ((r = setrlimit(RLIMIT_RTTIME, &rl)) < 0)
 		pw_log_debug("setrlimit() failed: %s", strerror(errno));
 
@@ -60,6 +61,7 @@ static void make_realtime(struct pw_data_loop *this)
 				pw_log_debug("setrlimit() failed: %s", strerror(errno));
 		}
 	}
+#endif
 
 	if ((r = pw_rtkit_make_realtime(system_bus, 0, rtprio)) < 0) {
 		pw_log_debug("could not make thread realtime: %s", strerror(r));
