--- meson.build.orig	2021-01-05 18:40:24.345620177 +0100
+++ meson.build	2021-01-05 18:40:43.209619527 +0100
@@ -325,7 +325,7 @@
 pthread_lib = dependency('threads')
 dbus_dep = dependency('dbus-1')
 sdl_dep = dependency('sdl2', required : false)
-ncurses_dep = dependency('ncurses', required : false)
+ncurses_dep = dependency('ncursesw', required : false)
 sndfile_dep = dependency('sndfile', version : '>= 1.0.20', required : false)
 
 if get_option('gstreamer')
