--- src/util/common.c.orig	2016-01-29 08:21:02.000701749 +0100
+++ src/util/common.c	2016-01-29 08:21:49.545577920 +0100
@@ -38,9 +38,12 @@
 #include <sys/time.h>
 #include <sys/resource.h>
 #include <errno.h>
-#include <sys/sysctl.h>
 #include <dirent.h>
 
+#if !defined(__linux__) || defined(__linux__) && defined(__GLIBC__)
+#include <sys/sysctl.h>
+#endif
+
 #ifdef HAVE_RSVG
 #include <librsvg/rsvg.h>
 #endif
