From 5c634b7e59eb183054df875d33f6f380a9ca2bc5 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Piotr=20W=C3=B3jcik?= <chocimier@tlen.pl>
Date: Thu, 2 Sep 2021 23:15:19 +0200
Subject: [PATCH] online-test


diff --git a/tests/musicbrainztest-severaldiscs.cpp b/tests/musicbrainztest-severaldiscs.cpp
index c572a7c..943f1fb 100644
--- a/tests/musicbrainztest-severaldiscs.cpp
+++ b/tests/musicbrainztest-severaldiscs.cpp
@@ -101,7 +101,7 @@ void MusicBrainzTestSeveralDiscs::testLookup()
   for (int j=12; j < 15; j++)
     QCOMPARE(i.track(j).get(Artist).toString(),QString::fromUtf8("Billy Bragg & Wilco"));
   for (int j=15; j < 20; j++)
-    QCOMPARE(i.track(j).get(Artist).toString(),QString::fromUtf8("Billy Bragg and The Blokes"));
+    QCOMPARE(i.track(j).get(Artist).toString(),QString::fromUtf8("Billy Bragg and the Blokes"));
   // comments not supported in a simple way
   for (int j=0; j < 19; j++)
     QCOMPARE(i.track(j).get(Comment).toString(),QString());
-- 
2.33.0

