--- src/Standard/Standard_CLocaleSentry.hxx.orig	2016-01-02 06:11:52.000000000 +0100
+++ src/Standard/Standard_CLocaleSentry.hxx	2016-06-03 14:12:17.399501218 +0200
@@ -30,7 +30,7 @@
   #endif
 
   //! We check _GNU_SOURCE for glibc extensions here and it is always defined by g++ compiler.
-  #if defined(_GNU_SOURCE) && !defined(__ANDROID__)
+  #if defined(_GNU_SOURCE) && defined(__GLIBC__) && !defined(__ANDROID__)
     #define HAVE_XLOCALE_H
   #endif
 #endif // ifndef HAVE_LOCALE_H
