Description: Fix import path for wpath in setup.py
Author: Axel Beckert <abe@debian.org>
Forwarded: no

Index: wicd/setup.py
===================================================================
--- wicd.orig/setup.py	2019-09-29 04:13:32.874964424 +0200
+++ wicd/setup.py	2019-09-29 04:13:51.127519471 +0200
@@ -60,10 +60,10 @@
 
     def run(self):
         try:
-            import wpath
-        except ImportError:
+            from wicd import wpath
+        except (ImportError, ModuleNotFoundError):
             self.run_command('configure')
-            import wpath
+            from wicd import wpath
             #raise Exception, 'Please run "./setup.py configure" first.'
         _build.run(self)
 
@@ -423,10 +423,10 @@
 class install(_install):
     def run(self):
         try:
-            import wpath
-        except ImportError:
+            from wicd import wpath
+        except (ImportError, ModuleNotFoundError):
             self.run_command('build')
-            import wpath
+            from wicd import wpath
 
         print("Using init file",(wpath.init, wpath.initfile))
         data.extend([
@@ -610,11 +610,11 @@
  
     def run(self):
         try:
-            import wpath
-        except ImportError:
+            from wicd import wpath
+        except (ImportError, ModuleNotFoundError):
             # if there's no wpath.py, then run configure+build
             self.run_command('build')
-            import wpath
+            from wicd import wpath
 
         if not wpath.no_install_i18n:
             if os.path.exists('translations'):
