--- a/curses/netentry_curses.py	2015-02-15 18:29:05.000000000 +0100
+++ b/curses/netentry_curses.py	2015-02-15 18:31:24.000000000 +0100
@@ -538,11 +538,12 @@
         self.bitrates = wireless.GetAvailableBitrates()
         self.bitrates.append('auto')
         self.bitrate_combo.set_list(self.bitrates)
-        self.bitrate_combo.set_focus(
-            self.bitrates.index(
-                wireless.GetWirelessProperty(networkID, 'bitrate')
+        if wireless.GetWirelessProperty(networkID, 'bitrate'):
+            self.bitrate_combo.set_focus(
+                self.bitrates.index(
+                    wireless.GetWirelessProperty(networkID, 'bitrate')
+                )
             )
-        )
         self.allow_lower_bitrates_chkbox.set_state(
             to_bool(self.format_entry(networkID, 'allow_lower_bitrates'))
         )
