=== modified file 'curses/wicd-curses.py'
--- a/curses/wicd-curses.py	2014-12-21 16:57:33 +0000
+++ b/curses/wicd-curses.py	2015-02-24 23:41:01 +0000
@@ -532,7 +532,10 @@
     def get_selected_profile(self):
         """Get the selected wired profile"""
         loc = self.get_focus()[1]
-        return self.theList[loc]
+        if len(self.theList) > loc:
+            return self.theList[loc]
+        else:
+            return self.theList[-1]
 
 
 class AdHocDialog(Dialog2):

