commit 3315d1f6cae5a3d7f24e16798587999a5804fad6
Author: q66 <daniel@octaforge.org>
Date:   Wed Jan 8 13:19:14 2020 +0100

    Add missing include to fix build on musl/BE

diff --git src/endian.c src/endian.c
index b7724f3..09b2315 100644
--- src/endian.c
+++ src/endian.c
@@ -15,6 +15,7 @@ void xle64enc(uint8_t *d, uint64_t n) {
 #include <stdint.h>
 #ifdef __linux__
 	#include <endian.h>
+	#include <byteswap.h>
 #else
 	#include <sys/endian.h>
 #endif
