Description: Fix for deprecated symbols in gtk3-3.13
Author: Yaakov Selkowitz <yselkowi@redhat.com>
Bug-Debian: http://bugs.debian.org/763220

--- config.mk.orig
+++ config.mk
@@ -172,10 +172,6 @@ endif
 CFLAGS_OPTIONS := $(CFLAGS)
 
 ifeq (${USEGTK3}, 1) 
-CPPFLAGS+=-DGTK_DISABLE_SINGLE_INCLUDES
-CPPFLAGS+=-DGTK_DISABLE_DEPRECATED
-CPPFLAGS+=-DGDK_DISABLE_DEPRECATED
-CPPFLAGS+=-DGSEAL_ENABLE
 M4FLAGS += -DWITH_GTK3=1
 endif
 
