--- a/krusader/DiskUsage/filelightParts/fileTree.h
+++ b/krusader/DiskUsage/filelightParts/fileTree.h
@@ -26,6 +26,7 @@
 #include <time.h>
 #include <stdlib.h>
 #include <unistd.h>
+#include <sys/stat.h>
 
 // TODO these are pointlessly general purpose now, make them incredibly specific
 
--- a/krusader/FileSystem/krpermhandler.h
+++ b/krusader/FileSystem/krpermhandler.h
@@ -42,6 +42,7 @@
 #include <time.h>
 #include <stdlib.h>
 #include <unistd.h>
+#include <sys/stat.h>
 
 #define NO_PERM      0
 #define UNKNOWN_PERM 1
