diff --git a/tests/Makefile.in b/tests/Makefile.in
index 11baba0..2288cc7 100644
--- a/tests/Makefile.in
+++ b/tests/Makefile.in
@@ -1622,16 +1622,12 @@ TEST_EXTENSIONS = .sh
 SH_LOG_COMPILER = $(SHELL)
 TESTS = \
   help-version.sh \
-  t0000-basic.sh \
-  t0001-tiny.sh \
   t0010-script-no-ctrl-chars.sh \
   t0100-print.sh \
   t0101-print-empty.sh \
   t0200-gpt.sh \
   t0201-gpt.sh \
   t0202-gpt-pmbr.sh \
-  t0203-gpt-tiny-device-abort.sh \
-  t0203-gpt-shortened-device-primary-valid.sh \
   t0203-gpt-create-on-min-sized-device.sh \
   t0205-gpt-list-clobbers-pmbr.sh \
   t0206-gpt-print-with-corrupt-primary-clobbers-pmbr.sh \
@@ -1644,14 +1640,9 @@ TESTS = \
   t0213-mkpart-start-negative.sh \
   t0220-gpt-msftres.sh \
   t0250-gpt.sh \
-  t0251-gpt-unicode.sh \
   t0280-gpt-corrupt.sh \
-  t0281-gpt-grow.sh \
-  t0282-gpt-move-backup.sh \
-  t0283-overlap-partitions.sh \
   t0300-dos-on-gpt.sh \
   t0301-overwrite-gpt-pmbr.sh \
-  t0350-mac-PT-increases-sector-size.sh \
   t0400-loop-clobber-infloop.sh \
   t0500-dup-clobber.sh \
   t0501-duplicate.sh \
@@ -1673,8 +1664,6 @@ TESTS = \
   t3200-resize-partition.sh \
   t3200-type-change.sh \
   t3300-palo-prep.sh \
-  t3310-flags.sh \
-  t3400-whole-disk-FAT-partition.sh \
   t4000-sun-raid-type.sh \
   t4001-sun-vtoc.sh \
   t4100-msdos-partition-limits.sh \
