From 0a96bb015d9dc30ff98cfc21256851cb2005cea1 Mon Sep 17 00:00:00 2001
From: Piotr Wójcik <chocimier@tlen.pl>
Date: Sat, 1 Jul 2023 17:28:17 +0200
Subject: adjust-PATH-for-usrmerge


diff --git a/cfg.cpp b/cfg.cpp
index 02379f2..56a584b 100644
--- a/cfg.cpp
+++ b/cfg.cpp
@@ -29,7 +29,7 @@ Cfg::Cfg()
 	: currentSession(-1)
 {
 	/* Configuration options */
-	options.insert(option("default_path","/bin:/usr/bin:/usr/local/bin"));
+	options.insert(option("default_path","/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/sbin:/sbin:/bin"));
 	options.insert(option("default_xserver","/usr/bin/X"));
 	options.insert(option("xserver_arguments",""));
 	options.insert(option("numlock",""));
diff --git a/slim.conf b/slim.conf
index a8e2e1c..1c14ade 100644
--- a/slim.conf
+++ b/slim.conf
@@ -1,6 +1,6 @@
 # Path, X server and arguments (if needed)
 # Note: -xauth $authfile is automatically appended
-default_path        /bin:/usr/bin:/usr/local/bin
+default_path        /usr/local/sbin:/usr/local/bin:/usr/bin:/usr/sbin:/sbin:/bin
 default_xserver     /usr/bin/X
 #xserver_arguments   -dpi 75
 
-- 
2.41.0

