--- panel.cpp	2010-10-24 06:01:15.117000019 +0000
+++ panel.cpp	2010-10-24 17:26:30.987999967 +0000
@@ -23,7 +23,7 @@
     Root = root;
     cfg = config;

-    session = "";
+    session = cfg->nextSession(session);

     // Init GC
     XGCValues gcv;
@@ -188,7 +188,7 @@
 }

 void Panel::ClearPanel() {
-    session = "";
+    session = cfg->nextSession(session);
     Reset();
     XClearWindow(Dpy, Root);
     XClearWindow(Dpy, Win);
