From 741e43960a4ea3a8bc230b1f599311d800c2cc83 Mon Sep 17 00:00:00 2001
From: Nobuhiro Iwamatsu <iwamatsu@nigauri.org>
Date: Wed, 02 Oct 2013 23:23:24 +0000
Subject: Update slim.conf. Add sessiondir.

Signed-off-by: Nobuhiro Iwamatsu <iwamatsu@nigauri.org>
---
diff --git a/slim.conf b/slim.conf
index a8e2e1c..c5f1d38 100644
--- a/slim.conf
+++ b/slim.conf
@@ -47,11 +47,9 @@ login_cmd           exec /bin/bash -login ~/.xinitrc %session
 # options "-d" and "-nodaemon"
 # daemon	yes
 
-# Available sessions (first one is the default).
-# The current chosen session name is replaced in the login_cmd
-# above, so your login command can handle different sessions.
-# see the xinitrc.sample file shipped with slim sources
-sessions            xfce4,icewm-session,wmaker,blackbox
+# Set directory that contains the xsessions.
+# slim reads xsesion from this directory, and be able to select.
+sessiondir            /usr/share/xsessions/
 
 # Executed when pressing F11 (requires imagemagick)
 screenshot_cmd      import -window root /slim.png
--
cgit v0.9.2
