diff --git a/imagick.c b/imagick.c
index 7733e2b..e1cbd55 100644
--- a/imagick.c
+++ b/imagick.c
@@ -603,7 +603,7 @@ static zval *php_imagick_read_property(zend_object *object, zend_string *member,
 					if (format) {
 						retval = rv;
 						ZVAL_STRING(retval, format);
-						php_strtolower(Z_STRVAL_P(retval), Z_STRLEN_P(retval));
+						zend_str_tolower(Z_STRVAL_P(retval), Z_STRLEN_P(retval));
 						IMAGICK_FREE_MAGICK_MEMORY(format);
 					} else {
 						retval = rv;
@@ -676,7 +676,7 @@ static zval *php_imagick_read_property(zval *object, zval *member, int type, voi
 					if (format) {
 						retval = rv;
 						ZVAL_STRING(retval, format);
-						php_strtolower(Z_STRVAL_P(retval), Z_STRLEN_P(retval));
+						zend_str_tolower(Z_STRVAL_P(retval), Z_STRLEN_P(retval));
 						IMAGICK_FREE_MAGICK_MEMORY(format);
 					} else {
 						retval = rv;
@@ -759,7 +759,7 @@ static zval *php_imagick_read_property(zval *object, zval *member, int type, con
 
 					if (format) {
 						ZVAL_STRING(retval, format, 1);
-						php_strtolower(Z_STRVAL_P(retval), Z_STRLEN_P(retval));
+						zend_str_tolower(Z_STRVAL_P(retval), Z_STRLEN_P(retval));
 						IMAGICK_FREE_MAGICK_MEMORY(format);
 					} else {
 						ZVAL_STRING(retval, "", 1);
