--- src/os/unix/dir.cpp.orig	2016-07-03 20:48:33.000000000 +0200
+++ src/os/unix/dir.cpp	2016-07-20 21:03:49.299919267 +0200
@@ -174,10 +174,10 @@
 
 stringT pws_os::getxmldir(void)
 {
-  return _S("/usr/share/pwsafe/xml/");
+  return _S("/usr/share/passwordsafe/xml/");
 }
 
 stringT pws_os::gethelpdir(void)
 {
-  return _S("/usr/share/doc/passwordsafe/help/");
+  return _S("/usr/share/passwordsafe/help/");
 }
