--- src/os/unix/dir.cpp.orig
+++ src/os/unix/dir.cpp
@@ -176,7 +176,7 @@
 #ifdef __FreeBSD__
   return _S("/usr/local/share/pwsafe/xml/");
 #else
-  return _S("/usr/share/pwsafe/xml/");
+  return _S("/usr/share/passwordsafe/xml/");
 #endif
 }
 
@@ -185,6 +185,6 @@
 #ifdef __FreeBSD__
   return _S("/usr/local/share/doc/passwordsafe/help/");
 #else
-  return _S("/usr/share/doc/passwordsafe/help/");
+  return _S("/usr/share/passwordsafe/help/");
 #endif
 }
