--- src/os/linux/dir.cpp.orig	2015-10-09 10:35:00.000000000 +0200
+++ src/os/linux/dir.cpp	2015-11-18 12:40:54.216572503 +0100
@@ -151,10 +151,10 @@
 
 stringT pws_os::getxmldir(void)
 {
-  return _S("/usr/share/pwsafe/xml/");
+  return _S("/usr/share/passwordsafe/xml/");
 }
 
 stringT pws_os::gethelpdir(void)
 {
-  return _S("/usr/share/doc/passwordsafe/help/");
+  return _S("/usr/share/passwordsafe/help/");
 }
