--- a/CMakeLists.txt	2021-02-25 23:45:37.917034534 +0100
+++ b/CMakeLists.txt	2021-02-26 00:05:34.138397928 +0100
@@ -255,34 +255,6 @@
 endif (USE_ASAN)
 endif (MSVC)
 
-if (NOT NO_GTEST)
-   # Download and unpack googletest at configure time
-   # See https://crascit.com/2015/07/25/cmake-gtest/
-   configure_file(Misc/CMakeLists.gtest.txt.in googletest-download/CMakeLists.txt)
-   execute_process(COMMAND "${CMAKE_COMMAND}" -G "${CMAKE_GENERATOR}" .
-       WORKING_DIRECTORY "${CMAKE_BINARY_DIR}/googletest-download" )
-   execute_process(COMMAND "${CMAKE_COMMAND}" --build .
-       WORKING_DIRECTORY "${CMAKE_BINARY_DIR}/googletest-download" )
-
-   # Prevent GoogleTest from overriding our compiler/linker options
-   # when building with Visual Studio
-   set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
-   # Set some other gtest configurations:
-   set(BUILD_GMOCK OFF CACHE BOOL "" FORCE)
-   set(BUILD_GTEST ON CACHE BOOL "" FORCE)
-   set(INSTALL_GTEST OFF CACHE BOOL "" FORCE)
-
-   # Add googletest directly to our build. This adds
-   # the following targets: gtest, gtest_main, gmock
-   # and gmock_main
-   add_subdirectory("${CMAKE_BINARY_DIR}/googletest-src"
-                    "${CMAKE_BINARY_DIR}/googletest-build")
-
-   include_directories("${gtest_SOURCE_DIR}/include"
-#                       "${gmock_SOURCE_DIR}/include"
-                       )
-endif(NOT NO_GTEST)
-
 if (WIN32 AND NOT WX_WINDOWS)
   set(VERSION_IN "${PROJECT_SOURCE_DIR}/src/ui/Windows/version.in")
 else (WIN32 AND NOT WX_WINDOWS)
