diff --git a/conduit-example.toml b/conduit-example.toml
index 5eed070..bee1ded 100644
--- a/conduit-example.toml
+++ b/conduit-example.toml
@@ -23,7 +23,7 @@
 #server_name = "your.server.name"
 
 # This is the only directory where Conduit will save its data
-database_path = "/var/lib/matrix-conduit/"
+database_path = "/var/lib/conduit/"
 database_backend = "rocksdb"
 
 # The port Conduit will be running on. You need to set up a reverse proxy in
