--- src/benchmark.cpp	2012-07-08 21:57:55.246042862 +0800
+++ src/benchmark.cpp	2012-07-08 21:58:20.015908672 +0800
@@ -26,6 +26,7 @@
 #include <stdlib.h>
 #include <string.h>
 #include <config.h>
+#include <unistd.h>
 #include <calf/audio_fx.h>
 
 #ifdef BENCHMARK_PLUGINS
--- src/osctlnet.cpp	2012-07-08 21:57:55.246042862 +0800
+++ src/osctlnet.cpp	2012-07-08 21:58:19.002580827 +0800
@@ -26,6 +26,7 @@
 #include <stdlib.h>
 #include <sstream>
 #include <stdio.h>
+#include <unistd.h>
 using namespace osctl;
 using namespace std;
 
--- src/preset.cpp	2012-07-08 21:57:55.246042862 +0800
+++ src/preset.cpp	2012-07-08 21:58:18.989247567 +0800
@@ -20,6 +20,7 @@
  */
 
 #include <config.h>
+#include <unistd.h>
 #include <fcntl.h>
 #include <stdlib.h>
 #include <errno.h>
