upstream: yes

From 02eb963f11de11a1115e815625ae38f6dad6b276 Mon Sep 17 00:00:00 2001
From: Miek Gieben <miek@miek.nl>
Date: Wed, 2 Mar 2016 15:51:45 +0000
Subject: [PATCH] Release 1.1.15

---
 ChangeLog    | 3 ++-
 configure.ac | 2 +-
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/ChangeLog b/ChangeLog
index 1c79ba6..5a6812a 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,4 +1,4 @@
-xx Yyy 2013: 1.1.15 Miek Gieben <miek@miek.nl>
+02 Mar 2016: 1.1.15 Miek Gieben <miek@miek.nl>
   * [rdup] Change -P so any arguments are executed with 'sh -c', 
     this removes the 7 option limit (Alexander Neumann)
   * [rdup-tr] Fix crashes with filename encryption and symlinks (reported by
@@ -6,6 +6,7 @@ xx Yyy 2013: 1.1.15 Miek Gieben <miek@miek.nl>
   * [rdup-tr] Fix memory leak
   * [rdup] Fix the linked list that tracks the child's PIDs.
   * [rdup] Drop -a option, it's racy and must proper or not at all.
+  * [rdup] Compilation fixed for Mac OS X
 
 
 19 Jan 2013: 1.1.14 Miek Gieben <miek@miek.nl>
diff --git a/configure.ac b/configure.ac
index 5b77cfe..bc3ada5 100644
--- a/configure.ac
+++ b/configure.ac
@@ -2,7 +2,7 @@
 # Process this file with autoconf to produce a configure script.
 
 AC_PREREQ(2.57)
-AC_INIT(rdup, 1.1.14, miek@miek.nl, rdup)
+AC_INIT(rdup, 1.1.15, miek@miek.nl, rdup)
 AC_CONFIG_SRCDIR([rdup.c])
 AC_PREFIX_DEFAULT(/usr/local)
 
