--- obnamlib/plugins/backup_plugin.py
+++ obnamlib/plugins/backup_plugin.py
@@ -593,7 +593,7 @@ class BackupPlugin(obnamlib.ObnamPlugin):
         self.open_or_reopen_fs(self.fs.reinit, root_url)
 
     def open_or_reopen_fs(self, func, root_url):
-        scheme, neloc, path, params, query, fragment = \
+        scheme, netloc, path, params, query, fragment = \
             urlparse.urlparse(root_url)
 
         try:
@@ -613,7 +613,7 @@ class BackupPlugin(obnamlib.ObnamPlugin):
         for root_url in root_urls:
             self.progress.what('determining absolute path for %s' % root_url)
 
-            scheme, neloc, path, params, query, fragment = \
+            scheme, netloc, path, params, query, fragment = \
                 urlparse.urlparse(root_url)
 
             if self.fs.isdir(path):
