--- ext/test/fail_fs/fail_fs.c	2017-05-26 20:37:54.000000000 +0200
+++ ext/test/fail_fs/fail_fs.c	2017-05-27 23:53:50.203903712 +0200
@@ -35,7 +35,9 @@
 #include <unistd.h>
 #include <fcntl.h>
 #include <sys/stat.h>
+#ifdef __GLIBC__
 #include <execinfo.h>
+#endif
 
 #include <wiredtiger_ext.h>
 #include "queue.h"
@@ -697,6 +699,7 @@
 		(void)wtext->msg_printf(wtext, session,
 		    "fail_fs: %s: simulated failure after %" PRId64
 		    " %s operations", fail_fh->iface.name, nops, opkind);
+#ifdef __GLIBC__
 #ifdef __FreeBSD__
 		btret = backtrace(bt, sizeof(bt) / sizeof(bt[0]));
 #else
@@ -708,6 +711,7 @@
 				    btstr[i]);
 			free(btstr);
 		}
+#endif
 	}
 	return (EIO);
 }
