diff --git a/GNUmakefile b/GNUmakefile
index 93bd983..50d7005 100644
--- a/GNUmakefile
+++ b/GNUmakefile
@@ -113,8 +113,8 @@ bin/retro-unu: tools/retro-unu.c
 # basic image target
 
 ngaImage: image/retro.muri image/retro.forth bin/retro-muri bin/retro-extend
-	./bin/retro-muri image/retro.muri
-	./bin/retro-extend ngaImage image/retro.forth image/build.retro
+	@$(ASSEMBLE) image/retro.muri
+	@$(EXTEND) ngaImage image/retro.forth image/build.retro
 
 # minimal system
 
@@ -130,7 +130,7 @@ update-extensions: bin/retro
 
 vm/nga-c/image.c: ngaImage bin/retro-embedimage bin/retro-extend $(DEVICES) interface/retro-unix.retro
 	cp ngaImage rre.image
-	./bin/retro-extend rre.image $(DEVICES) interface/retro-unix.retro
+	@$(EXTEND) rre.image $(DEVICES) interface/retro-unix.retro
 	./bin/retro-embedimage rre.image >vm/nga-c/image.c
 
 bin/retro: vm/nga-c/image.c vm/nga-c/retro.c package/list.forth package/load-extensions.retro
