diff --git a/numbat-exchange-rates/Cargo.toml b/numbat-exchange-rates/Cargo.toml
index a34b2ae..b3f8a08 100644
--- a/numbat-exchange-rates/Cargo.toml
+++ b/numbat-exchange-rates/Cargo.toml
@@ -11,7 +11,7 @@ rust-version = "1.88"
 
 [dependencies]
 attohttpc = { version = "0.27.0", default-features = false, features = [
-    "tls-rustls-webpki-roots",
+    "tls-native",
 ], optional = true }
 quick-xml = "0.37.5"
 
