Relax this attribute, otherwise GL context creation fails
--- mymcplus/gui/icon_window.py.orig	2020-06-08 18:05:05.646326599 +0200
+++ mymcplus/gui/icon_window.py	2020-06-08 18:05:14.093326318 +0200
@@ -121,7 +121,6 @@
             glcanvas.WX_GL_RGBA,
             glcanvas.WX_GL_DOUBLEBUFFER,
             glcanvas.WX_GL_DEPTH_SIZE, 24,
-            glcanvas.WX_GL_SAMPLES, 16
         ]
 
         self.sizer = wx.BoxSizer(wx.VERTICAL)
