--- cfe-3.8.0.src/lib/Driver/ToolChains.cpp.orig
+++ cfe-3.8.0.src/lib/Driver/ToolChains.cpp
@@ -2428,6 +2428,7 @@
                                         ArgStringList &CC1Args) const {
   const Generic_GCC::GCCVersion &V = GCCInstallation.getVersion();
   bool UseInitArrayDefault =
+      getTriple().getEnvironment() == llvm::Triple::Musl ||
       getTriple().getArch() == llvm::Triple::aarch64 ||
       getTriple().getArch() == llvm::Triple::aarch64_be ||
       (getTriple().getOS() == llvm::Triple::Linux &&
