--- libfuse/Makefile	2020-02-23 04:40:32.000000000 +0100
+++ libfuse/Makefile	2020-03-09 16:30:59.282571220 +0100
@@ -93,11 +93,8 @@
 	strip --strip-all build/mount.mergerfs
 	strip --strip-all build/mergerfs-fusermount
 
-install: mergerfs-fusermount mount.mergerfs strip
-	install -D build/mergerfs-fusermount "$(INSTALLBINDIR)/mergerfs-fusermount"
-	install -D build/mount.mergerfs "$(INSTALLSBINDIR)/mount.mergerfs"
-	chown root:root "$(INSTALLBINDIR)/mergerfs-fusermount"
-	chmod u+s "$(INSTALLBINDIR)/mergerfs-fusermount"
+install:
+	echo "---------"
 
 .PHONY: objects strip
 
