--- Makefile.orig	2017-06-04 20:12:49.459463768 +0200
+++ Makefile	2017-06-04 20:14:04.535236799 +0200
@@ -77,9 +77,8 @@ help:
 	@echo "usage: make"
 	@echo "make XATTR_AVAILABLE=0 - to build program without xattrs functionality (auto discovered otherwise)"
 
-$(TARGET): src/version.hpp obj/obj-stamp libfuse/lib/.libs/libfuse.a $(OBJ)
-	cd libfuse && make
-	$(CXX) $(CFLAGS) $(OBJ) -o $@ libfuse/lib/.libs/libfuse.a -ldl $(LDFLAGS)
+$(TARGET): src/version.hpp obj/obj-stamp $(OBJ)
+	$(CXX) $(CFLAGS) $(OBJ) -o $@ -lfuse -ldl $(LDFLAGS)
 
 mount.mergerfs: $(TARGET)
 	$(LN) -fs "$<" "$@"
