diff --git a/codelitegcc/main.cpp b/codelitegcc/main.cpp
index 1b78e0825..a4ffdc3e7 100644
--- a/codelitegcc/main.cpp
+++ b/codelitegcc/main.cpp
@@ -33,6 +33,7 @@ extern int ExecuteProcessWIN(const std::string& commandline);
 #include <sys/file.h>
 #include <sys/stat.h>
 #include <sys/stat.h>
+#include <fcntl.h>
 
 void WriteContent( const std::string& logfile, const std::string& filename, const std::string& flags )
 {
