--- a/glaxnimate-0.5.1/external/QtAppSetup/CMakeLists.txt
+++ b/glaxnimate-0.5.1/external/QtAppSetup/CMakeLists.txt
@@ -31,11 +31,11 @@
 if(Python3_FOUND AND SCRIPTING_PY)
     set(PYTHON_ENABLED ON PARENT_SCOPE)
     message(STATUS "Python3 enabled")
-    add_subdirectory(external/pybind11)
+    find_package(pybind11 REQUIRED)
     add_subdirectory(src/app/scripting/python)
     target_include_directories (${PROJECT_SLUG} PUBLIC
         ${Python3_INCLUDE_DIRS}
-        ${CMAKE_CURRENT_SOURCE_DIR}/external/pybind11/include
+        ${pybind11_INCLUDE_DIR}
     )
 else()
     set(PYTHON_ENABLED OFF PARENT_SCOPE)
--- a/glaxnimate-0.5.1/src/python/CMakeLists.txt
+++ b/glaxnimate-0.5.1/src/python/CMakeLists.txt
@@ -1,3 +1,4 @@
+find_package(pybind11 REQUIRED)
 set(SOURCES
     python.cpp
     miscdefs.cpp
