--- sile.in	2016-06-09 10:57:37.000000000 +0000
+++ -	2018-06-21 20:22:13.514804438 +0000
@@ -1,4 +1,4 @@
-#!@LUA@
+#!@LUA@@LUA_VERSION@
 package.path = (os.getenv("SILE_PATH") and
 os.getenv("SILE_PATH").."/?.lua" or "") .. ';?.lua;@SILE_PATH@/?.lua;@SILE_PATH@/lua-libraries/?.lua;@SILE_PATH@/lua-libraries/?/init.lua;lua-libraries/?.lua;lua-libraries/?/init.lua;' .. package.path
 package.cpath = package.cpath .. ";core/?.@SHARED_LIB_EXT@;@SILE_LIB_PATH@/?.@SHARED_LIB_EXT@;"
