--- dumb/cmake/CMakeLists.txt.orig
+++ dumb/cmake/CMakeLists.txt
@@ -1,7 +1,7 @@
 cmake_minimum_required(VERSION 2.6)
 project(libdumb C)
 
-set(CMAKE_C_FLAGS "-Wall -DDUMB_DECLARE_DEPRECATED -D_USE_SSE -msse -Wno-unused-variable -Wno-unused-but-set-variable")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -DDUMB_DECLARE_DEPRECATED -Wno-unused-variable -Wno-unused-but-set-variable")
 set(CMAKE_C_FLAGS_DEBUG "-ggdb -DDEBUGMODE=1 -D_DEBUG")
 set(CMAKE_C_FLAGS_RELEASE "-ffast-math -O2 -DNDEBUG")
 set(CMAKE_C_FLAGS_RELWITHDEBINFO "-ffast-math -g -O2 -DNDEBUG")
@@ -103,6 +103,7 @@
 )
 
 add_library(dumb ${SOURCES})
+set_target_properties(dumb PROPERTIES VERSION 1.0)
 set_target_properties(dumb PROPERTIES DEBUG_POSTFIX d)
 
 # Make sure the dylib install name path is set on OSX so you can include dumb in app bundles
