--- rkt-1.25.0/stage1/prepare-app/prepare-app.c	2017-02-20 11:16:05.000000000 +0100
+++ rkt-1.25.0/stage1/prepare-app/prepare-app.c	2017-03-06 00:55:19.536542504 +0100
@@ -27,6 +27,10 @@
 #include <inttypes.h>
 #include <stdbool.h>
 
+#if !defined(ACCESSPERMS)
+#define ACCESSPERMS (S_IRWXU|S_IRWXG|S_IRWXO) /* 0777 */
+#endif
+
 #define err_out(_fmt, _args...)						\
 		fprintf(stderr, "Error: " _fmt "\n", ##_args);
 static int exit_err;
