--- sh/init.sh.Linux.in.orig	2010-02-26 09:13:26.682795699 +0100
+++ sh/init.sh.Linux.in	2010-02-26 09:20:18.411807297 +0100
@@ -70,4 +70,14 @@ if $mountproc; then
 	eend $?
 fi
 
+# Try to mount xenfs as early as possible, otherwise rc_sys() will always
+# return RC_SYS_XENU and will think that we are in a domU while it's not.
+if grep -Eq "[[:space:]]+xenfs$" /proc/filesystems; then
+	ebegin "Mounting xenfs"
+	if ! fstabinfo --mount /proc/xen; then
+		mount -n -t xenfs xenfs /proc/xen -o nosuid,nodev,noexec
+	fi
+	eend $?
+fi
+
 . "$RC_LIBEXECDIR"/sh/init-common-post.sh
