--- pkgconfig/Makefile.orig	2010-10-28 03:47:39.000000000 +0200
+++ pkgconfig/Makefile	2010-10-28 03:48:00.000000000 +0200
@@ -2,7 +2,8 @@ DIR=	${LIBDIR}/pkgconfig
 SRCS=	einfo.pc.in openrc.pc.in
 INC=	einfo.pc openrc.pc
 
-sed -n -e 's/^VERSION=[[:space:]]*\([^[:space:]]*\).*/#define VERSION "\1${GITVER}\"/p' ../../Makefile > version.h
+all:
+	sed -n -e 's/^VERSION=[[:space:]]*\([^[:space:]]*\).*/#define VERSION "\1${GITVER}\"/p' ../Makefile.inc > version.h
 
 SED_EXTRA=	-e 's:@VERSION@:${VERSION}:g'
 
