diff -rauN src/tests/allocs-by-size.c src/tests/allocs-by-size.c
--- src/tests/allocs-by-size.c	2020-03-22 20:00:12.677048822 +0100
+++ src/tests/allocs-by-size.c	2020-03-22 20:01:07.000418361 +0100
@@ -109,7 +109,7 @@
     {
       const Item *item = &g_array_index (ar, Item, i);
 
-      g_print ("%"G_GUINT64_FORMAT",%"G_GUINT64_FORMAT",%"G_GUINT64_FORMAT"\n",
+      g_print ("%"G_GSIZE_FORMAT",%"G_GSIZE_FORMAT",%"G_GSIZE_FORMAT"\n",
                item->size, item->cmp, item->count);
     }
 }
