From 0cfe2ed0d265fe610915b5447ca61f4bfaf01715 Mon Sep 17 00:00:00 2001
From: Jonathan Mercier-Ganady <jmgr@jmgr.info>
Date: Tue, 17 Mar 2020 22:40:09 +0000
Subject: [PATCH] Add a missing include

---
 actiontools/keyboardkeyedit.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/actiontools/keyboardkeyedit.cpp b/actiontools/keyboardkeyedit.cpp
index 80eaa8d9..377385af 100644
--- actiontools/keyboardkeyedit.cpp
+++ actiontools/keyboardkeyedit.cpp
@@ -20,6 +20,8 @@
 
 #include "keyboardkeyedit.h"
 
+#include <QKeyEvent>
+
 namespace ActionTools
 {
     KeyboardKeyEdit::KeyboardKeyEdit(QWidget *parent):
