--- a/actiontools/CMakeLists.txt
+++ b/actiontools/CMakeLists.txt
@@ -294,11 +294,17 @@ setup_target(${PROJECT})
 find_package(OpenCV REQUIRED core imgproc)
 find_package(Qt6 ${ACT_MINIMUM_QT_VERSION} COMPONENTS Widgets Qml QmlPrivate REQUIRED)
 
+set(_toolchain_file "")
+if (DEFINED CMAKE_TOOLCHAIN_FILE)
+    set(_toolchain_file "-DCMAKE_TOOLCHAIN_FILE=${CMAKE_TOOLCHAIN_FILE}")
+endif()
+
 ExternalProject_Add(external_qtjsapi
     SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/3rdparty/qtjsapi
     BINARY_DIR ${CMAKE_BINARY_DIR}/3rdparty/qtjsapi
     INSTALL_COMMAND cmake -E echo "Skipping install step."
     CMAKE_ARGS
+            ${_toolchain_file}
             -DCMAKE_PREFIX_PATH=${CMAKE_PREFIX_PATH}
             -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
 )
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -56,7 +56,7 @@ function(setup_target TARGET)
 endfunction()
 
 function(compile_translations TARGET)
-    find_package(Qt6 ${ACT_MINIMUM_QT_VERSION} COMPONENTS LinguistTools REQUIRED)
+    find_package(Qt6 ${ACT_MINIMUM_QT_VERSION} COMPONENTS Linguist REQUIRED)
 
     set(TS_FILES
         ${CMAKE_CURRENT_SOURCE_DIR}/translations/${TARGET}_fr_FR.ts
