--- actiontools/version.cpp	2015-06-22 17:22:46.000000000 +0200
+++ actiontools/version.cpp	2015-10-08 11:03:52.104274402 +0200
@@ -20,6 +20,7 @@
 
 #include "version.h"
 
+#include <QDataStream>
 #include <QRegExp>
 #include <QStringList>
 
--- tools/version.cpp	2015-06-22 17:22:46.000000000 +0200
+++ tools/version.cpp	2015-10-08 11:10:09.000244997 +0200
@@ -20,6 +20,7 @@
 
 #include "version.h"
 
+#include <QDataStream>
 #include <QRegExp>
 #include <QStringList>
 
--- actiontools/qtsingleapplication/qtlocalpeer.h	2015-06-22 17:22:46.000000000 +0200
+++ actiontools/qtsingleapplication/qtlocalpeer.h	2015-10-08 11:27:48.169162361 +0200
@@ -48,6 +48,7 @@
 #include <QtNetwork/QLocalServer>
 #include <QtNetwork/QLocalSocket>
 #include <QtCore/QDir>
+#include <QDataStream>
 
 namespace QtLP_Private {
 #include "qtlockedfile.h"
--- actiontools/datacopyactioninstance.h	2015-06-22 17:22:46.000000000 +0200
+++ actiontools/datacopyactioninstance.h	2015-10-08 11:30:36.177149253 +0200
@@ -24,6 +24,7 @@
 #include "actiontools_global.h"
 #include "actioninstance.h"
 
+#include <QDataStream>
 #include <QTimer>
 
 namespace ActionTools
--- actiontools/parameter.h	2015-06-22 17:22:46.000000000 +0200
+++ actiontools/parameter.h	2015-10-08 11:37:16.880117991 +0200
@@ -24,6 +24,7 @@
 #include "subparameter.h"
 #include "actiontools_global.h"
 
+#include <QDataStream>
 #include <QSharedData>
 
 namespace ActionTools
--- actiontools/actionexception.cpp	2015-06-22 17:22:46.000000000 +0200
+++ actiontools/actionexception.cpp	2015-10-08 11:40:59.383100631 +0200
@@ -21,6 +21,7 @@
 #include "actionexception.h"
 
 #include <QDebug>
+#include <QDataStream>
 
 namespace ActionTools
 {
--- actiontools/subparameter.h	2015-06-22 17:22:46.000000000 +0200
+++ actiontools/subparameter.h	2015-10-08 11:43:23.632089377 +0200
@@ -23,6 +23,7 @@
 
 #include "actiontools_global.h"
 
+#include <QDataStream>
 #include <QSharedData>
 #include <QVariant>
 
--- actiontools/qxtsmtp/qxtmailattachment.h	2015-06-22 17:22:46.000000000 +0200
+++ actiontools/qxtsmtp/qxtmailattachment.h	2015-10-08 11:21:06.874193670 +0200
@@ -28,6 +28,7 @@
 
 #include "qxtglobal.h"
 
+#include <QIODevice>
 #include <QStringList>
 #include <QHash>
 #include <QByteArray>
--- actiontools/numberformat.h	2015-06-22 17:22:46.000000000 +0200
+++ actiontools/numberformat.h	2015-10-08 11:33:13.130137008 +0200
@@ -24,6 +24,7 @@
 #include "actiontools_global.h"
 
 #include <QString>
+#include <QObject>
 
 namespace ActionTools
 {
