--- gui/mainwindow.cpp
+++ gui/mainwindow.cpp
@@ -128,11 +128,16 @@ MainWindow::MainWindow(QxtCommandOptions *commandOptions, ProgressSplashScreen *
 
 	ui->setupUi(this);
 
+    mScriptProgressDialog->close();
     mScriptProgressDialog->setWindowModality(Qt::ApplicationModal);
     mScriptProgressDialog->setCancelButton(0);
     mScriptProgressDialog->setAutoClose(false);
     mScriptProgressDialog->setMinimumDuration(0);
 
+#ifndef ACT_NO_UPDATER
+    mUpdaterProgressDialog->close();
+#endif
+
     if(Global::ACTIONA_VERSION >= Tools::Version(1, 0, 0))
 		ui->reportBugPushButton->setVisible(false);
 	

