--- a/actions/actionpacksystem/systeminfo/linux/qnetworkinfo_linux.cpp	2018-07-17 20:29:12.000000000 +0200
+++ b/actions/actionpacksystem/systeminfo/linux/qnetworkinfo_linux.cpp	2018-07-27 20:12:49.889568736 +0200
@@ -62,11 +62,11 @@
 #endif // QT_NO_UDEV
 
 #include <cmath>
-#include <linux/wireless.h>
 #include <sys/ioctl.h>
 #include <sys/socket.h>
 #include <sys/types.h>
 #include <unistd.h>
+#include <linux/wireless.h>
 
 QT_BEGIN_NAMESPACE
 
