--- a/compat/blake2.h
+++ b/compat/blake2.h
@@ -61,7 +61,7 @@
     uint8_t  personal[BLAKE2S_PERSONALBYTES];  // 32
   } blake2s_param;
 
-  BLAKE2_ALIGN( 64 ) typedef struct __blake2s_state
+  typedef struct BLAKE2_ALIGN( 64 ) __blake2s_state
   {
     uint32_t h[8];
     uint32_t t[2];
@@ -86,7 +86,7 @@
     uint8_t  personal[BLAKE2B_PERSONALBYTES];  // 64
   } blake2b_param;
 
-  BLAKE2_ALIGN( 64 ) typedef struct __blake2b_state
+  typedef struct BLAKE2_ALIGN( 64 ) __blake2b_state
   {
     uint64_t h[8];
     uint64_t t[2];
