--- src/pidgin-libnotify.c.orig	2007-10-01 20:52:38.000000000 +1000
+++ src/pidgin-libnotify.c	2007-10-01 20:53:20.000000000 +1000
@@ -307,6 +307,7 @@
 	g_hash_table_insert (buddy_hash, contact, notification);
 
 	g_object_set_data (G_OBJECT(notification), "contact", contact);
+        g_object_set_data (G_OBJECT(notification), "buddy", buddy);
 
 	g_signal_connect (notification, "closed", G_CALLBACK(closed_cb), NULL);
 
