--- line/src/unix.rs.orig       2018-11-19 19:17:02.606568203 +0100
+++ line/src/unix.rs    2018-11-19 19:17:13.762473164 +0100
@@ -6,7 +6,7 @@
 fn size() -> (usize, usize) {
     unsafe {
         let mut size: libc::winsize = std::mem::zeroed();
-        if libc::ioctl(1, libc::TIOCGWINSZ as libc::c_ulong, &mut size) == 0 {
+        if libc::ioctl(1, libc::TIOCGWINSZ, &mut size) == 0 {
             (size.ws_col as usize, size.ws_row as usize)
         } else {
             (0, 0)
