--- a/ResultTable.cc	2018-01-10 19:18:31.278796587 +0100
+++ b/ResultTable.cc	2018-01-10 19:18:18.193831083 +0100
@@ -909,8 +909,8 @@ void ResultTable::lmsint(double &a, doub
 	l = 0;
 	for (i = 0; i < columns; i++) {
 	    if (partialmins[i] != timeoutresult) {
-		residuals[l] = abs(partialmins[i] - 
-				   ((currentslope * 
+		residuals[l] = abs((long int)partialmins[i] - 
+				   (long int)((currentslope * 
 				     column2size(i) /
 				     slopescale) +
 				    currentintercept));
@@ -930,7 +930,7 @@ void ResultTable::lmsint(double &a, doub
 	l = 0;
 	for (i = 0; i < columns; i++) {
 	    if (partialmins[i] != timeoutresult) {
-		ys[l] = abs(partialmins[i] - mediany);
+		ys[l] = abs((long int)partialmins[i] - (long int)mediany);
 		l++;
 	    }
 	}
