diff --git a/configure b/configure
index a58dc5629..e01a8f4ec 100755
--- a/configure
+++ b/configure
@@ -14305,7 +14305,7 @@ if test -n "$DEP_CFLAGS"; then
 	gtk+-3.0 >= 3.22
 	gspell-1 >= 1.8
 	libgedit-amtk-5
-	tepl-6 >= 6.8
+	libgedit-tepl-6 >= 6.8
 	gee-0.8 >= 0.10
 	gsettings-desktop-schemas
 \""; } >&5
@@ -14314,7 +14314,7 @@ if test -n "$DEP_CFLAGS"; then
 	gtk+-3.0 >= 3.22
 	gspell-1 >= 1.8
 	libgedit-amtk-5
-	tepl-6 >= 6.8
+	libgedit-tepl-6 >= 6.8
 	gee-0.8 >= 0.10
 	gsettings-desktop-schemas
 ") 2>&5
@@ -14326,7 +14326,7 @@ if test -n "$DEP_CFLAGS"; then
 	gtk+-3.0 >= 3.22
 	gspell-1 >= 1.8
 	libgedit-amtk-5
-	tepl-6 >= 6.8
+	libgedit-tepl-6 >= 6.8
 	gee-0.8 >= 0.10
 	gsettings-desktop-schemas
 " 2>/dev/null`
@@ -14346,7 +14346,7 @@ if test -n "$DEP_LIBS"; then
 	gtk+-3.0 >= 3.22
 	gspell-1 >= 1.8
 	libgedit-amtk-5
-	tepl-6 >= 6.8
+	libgedit-tepl-6 >= 6.8
 	gee-0.8 >= 0.10
 	gsettings-desktop-schemas
 \""; } >&5
@@ -14355,7 +14355,7 @@ if test -n "$DEP_LIBS"; then
 	gtk+-3.0 >= 3.22
 	gspell-1 >= 1.8
 	libgedit-amtk-5
-	tepl-6 >= 6.8
+	libgedit-tepl-6 >= 6.8
 	gee-0.8 >= 0.10
 	gsettings-desktop-schemas
 ") 2>&5
@@ -14367,7 +14367,7 @@ if test -n "$DEP_LIBS"; then
 	gtk+-3.0 >= 3.22
 	gspell-1 >= 1.8
 	libgedit-amtk-5
-	tepl-6 >= 6.8
+	libgedit-tepl-6 >= 6.8
 	gee-0.8 >= 0.10
 	gsettings-desktop-schemas
 " 2>/dev/null`
@@ -14396,7 +14396,7 @@ fi
 	gtk+-3.0 >= 3.22
 	gspell-1 >= 1.8
 	libgedit-amtk-5
-	tepl-6 >= 6.8
+	libgedit-tepl-6 >= 6.8
 	gee-0.8 >= 0.10
 	gsettings-desktop-schemas
 " 2>&1`
@@ -14406,7 +14406,7 @@ fi
 	gtk+-3.0 >= 3.22
 	gspell-1 >= 1.8
 	libgedit-amtk-5
-	tepl-6 >= 6.8
+	libgedit-tepl-6 >= 6.8
 	gee-0.8 >= 0.10
 	gsettings-desktop-schemas
 " 2>&1`
@@ -14419,7 +14419,7 @@ fi
 	gtk+-3.0 >= 3.22
 	gspell-1 >= 1.8
 	libgedit-amtk-5
-	tepl-6 >= 6.8
+	libgedit-tepl-6 >= 6.8
 	gee-0.8 >= 0.10
 	gsettings-desktop-schemas
 ) were not met:
