--- a/zam-plugins/dpf/utils/lv2-ttl-generator/GNUmakefile
+++ b/zam-plugins/dpf/utils/lv2-ttl-generator/GNUmakefile
@@ -20,8 +20,22 @@ endif
 
 build: ../lv2_ttl_generator
 
+ifdef CROSS_BUILD
+
+HOST_CC := gcc
+HOST_CFLAGS := $(XBPS_CFLAGS)
+HOST_LDFLAGS := $(patsubst -L%, -L/usr/lib, $(LDFLAGS))
+
+else
+
+HOST_CC := $(CC)
+HOST_CFLAGS := $(CFLAGS)
+HOST_LDFLAGS := $(LDFLAGS)
+
+endif
+
 ../lv2_ttl_generator: lv2_ttl_generator.c
-	$(CC) $< $(CFLAGS) -o $@ $(LDFLAGS)
+	$(HOST_CC) $< $(HOST_CFLAGS) -o $@ $(HOST_LDFLAGS)
 
 endif # WINDOWS
 
