From https://gitweb.gentoo.org/repo/gentoo.git/tree/dev-libs/thrift/files/thrift-0.16.0-network-tests.patch

diff --git a/lib/cpp/test/ToStringTest.cpp b/lib/cpp/test/ToStringTest.cpp
index 736b33c0a..f66909873 100644
--- a/lib/cpp/test/ToStringTest.cpp
+++ b/lib/cpp/test/ToStringTest.cpp
@@ -41,31 +41,6 @@ BOOST_AUTO_TEST_CASE(base_types_to_string) {
   BOOST_CHECK_EQUAL(to_string("abc"), "abc");
 }
 
-// NOTE: Currently (as of 2021.08.12) the locale-based tests do not work on
-// Windows in the AppVeyor Thrift CI build correctly. Therefore disabled on
-// Windows:
-#ifndef _WIN32
-BOOST_AUTO_TEST_CASE(locale_en_US_int_to_string) {
-#ifdef _WIN32
-  std::locale::global(std::locale("en-US.UTF-8"));
-#else
-  std::locale::global(std::locale("en_US.UTF-8"));
-#endif
-  BOOST_CHECK_EQUAL(to_string(1000000), "1000000");
-}
-
-BOOST_AUTO_TEST_CASE(locale_de_DE_floating_point_to_string) {
-#ifdef _WIN32
-  std::locale::global(std::locale("de-DE.UTF-8"));
-#else
-  std::locale::global(std::locale("de_DE.UTF-8"));
-#endif
-  BOOST_CHECK_EQUAL(to_string(1.5), "1.5");
-  BOOST_CHECK_EQUAL(to_string(1.5f), "1.5");
-  BOOST_CHECK_EQUAL(to_string(1.5L), "1.5");
-}
-#endif
-
 BOOST_AUTO_TEST_CASE(empty_vector_to_string) {
   std::vector<int> l;
   BOOST_CHECK_EQUAL(to_string(l), "[]");

