diff --git a/lib/cpp/src/thrift/transport/TServerSocket.cpp b/lib/cpp/src/thrift/transport/TServerSocket.cpp
index 0000000..0000000 100644
--- a/lib/cpp/src/thrift/transport/TServerSocket.cpp
+++ b/lib/cpp/src/thrift/transport/TServerSocket.cpp
@@ -506,7 +506,7 @@ void TServerSocket::listen() {
       errno_copy = THRIFT_GET_SOCKET_ERROR;
 
       // use short circuit evaluation here to only sleep if we need to
-    } while ((retries++ < retryLimit_) && (THRIFT_SLEEP_SEC(retryDelay_) == 0));
+    } while (addr_iter || ((retries++ < retryLimit_) && (THRIFT_SLEEP_SEC(retryDelay_) == 0)));
 
   } // TCP socket //
