diff --git a/src/efibootmgr/efibootmgr.c b/src/efibootmgr/efibootmgr.c
index 199af39..3826e03 100644
--- src/efibootmgr/efibootmgr.c
+++ src/efibootmgr/efibootmgr.c
@@ -18,7 +18,7 @@
     Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 
 
-  This must tie the EFI_DEVICE_PATH to /boot/efi/elilo.efi
+  This must tie the EFI_DEVICE_PATH to /boot/efi/EFI/void_grub/grubx64.efi
   The  EFI_DEVICE_PATH will look something like:
     ACPI device path, length 12 bytes
     Hardware Device Path, PCI, length 6 bytes
@@ -26,7 +26,7 @@
     Media Device Path, Hard Drive, partition XX, length 30 bytes
     Media Device Path, File Path, length ??
     End of Hardware Device Path, length 4
-    Arguments passed to elilo, as UCS-2 characters, length ??
+    Arguments passed to grub2, as UCS-2 characters, length ??
 
 */
 
@@ -780,8 +780,8 @@ usage()
 	printf("\t-g | --gpt            force disk with invalid PMBR to be treated as GPT\n");
 	printf("\t-H | --acpi_hid XXXX  set the ACPI HID (used with -i)\n");
 	printf("\t-i | --iface name     create a netboot entry for the named interface\n");
-	printf("\t-l | --loader name     (defaults to \\elilo.efi)\n");
-	printf("\t-L | --label label     Boot manager display label (defaults to \"Linux\")\n");
+	printf("\t-l | --loader name     (defaults to \\EFI\\void_grub\\grubx64.efi)\n");
+	printf("\t-L | --label label     Boot manager display label (defaults to \"Void Linux (GRUB2)\")\n");
 	printf("\t-n | --bootnext XXXX   set BootNext to XXXX (hex)\n");
 	printf("\t-N | --delete-bootnext delete BootNext\n");
 	printf("\t-o | --bootorder XXXX,YYYY,ZZZZ,...     explicitly set BootOrder (hex)\n");
@@ -808,8 +808,8 @@ set_default_opts()
 	opts.active          = -1;   /* Don't set it */
 	opts.timeout         = -1;   /* Don't set it */
 	opts.edd10_devicenum = 0x80;
-	opts.loader          = "\\elilo.efi";
-	opts.label           = "Linux";
+	opts.loader          = "\\EFI\\void_grub\\grubx64.efi";
+	opts.label           = "Void Linux (GRUB2)";
 	opts.disk            = "/dev/sda";
 	opts.iface           = NULL;
 	opts.part            = 1;
