From 2af29c3dfba668f341b007261954a829ec148f82 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andr=C3=A9=20Cerqueira?= <acerqueira021@gmail.com>
Date: Thu, 28 Jul 2022 21:42:22 +0100
Subject: [PATCH 1/1] efibootmgr_default_to_grub2

---
 src/efibootmgr.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/efibootmgr.c b/src/efibootmgr.c
index b980bcd..c9ff07b 100644
--- a/src/efibootmgr.c
+++ b/src/efibootmgr.c
@@ -1412,7 +1412,7 @@ usage()
 	printf("\t-i | --iface name     Create a netboot entry for the named interface.\n");
 	printf("\t-I | --index number   When creating an entry, insert it in bootorder at specified position (default: 0).\n");
 	printf("\t-l | --loader name     (Defaults to \""DEFAULT_LOADER"\").\n");
-	printf("\t-L | --label label     Boot manager display label (defaults to \"Linux\").\n");
+	printf("\t-L | --label label     Boot manager display label (defaults to \"Void Linux (GRUB2)\")\n");
 	printf("\t-m | --mirror-below-4G t|f Mirror memory below 4GB.\n");
 	printf("\t-M | --mirror-above-4G X Percentage memory to mirror above 4GB.\n");
 	printf("\t-n | --bootnext XXXX   Set BootNext to XXXX (hex).\n");
@@ -1444,7 +1444,7 @@ set_default_opts()
 	opts.timeout         = -1;   /* Don't set it */
 	opts.edd10_devicenum = 0x80;
 	opts.loader          = DEFAULT_LOADER;
-	opts.label           = (unsigned char *)"Linux";
+	opts.label           = (unsigned char *)"Void Linux (GRUB2)";
 	opts.disk            = "/dev/sda";
 	opts.part            = -1;
 }
-- 
2.37.1

