--- a/src/efibootmgr.c
+++ b/src/efibootmgr.c
@@ -1239,7 +1239,7 @@ usage()
 	printf("\t     --ip-origin { {dhcp|static} | { static|stateless|stateful} }\n");
 #endif
 	printf("\t-l | --loader name     (defaults to \""DEFAULT_LOADER"\")\n");
-	printf("\t-L | --label label     Boot manager display label (defaults to \"Linux\")\n");
+	printf("\t-L | --label label     Boot manager display label (defaults to \"Void Linux (GRUB2)\")\n");
 	printf("\t-m | --mirror-below-4G t|f mirror memory below 4GB\n");
 	printf("\t-M | --mirror-above-4G X percentage memory to mirror above 4GB\n");
 	printf("\t-n | --bootnext XXXX   set BootNext to XXXX (hex)\n");
@@ -1269,7 +1269,7 @@ set_default_opts()
 	opts.timeout         = -1;   /* Don't set it */
 	opts.edd10_devicenum = 0x80;
 	opts.loader          = DEFAULT_LOADER;
-	opts.label           = (unsigned char *)"Linux";
+	opts.label           = (unsigned char *)"Void Linux (GRUB2)";
 	opts.disk            = "/dev/sda";
 	opts.part            = 1;
 }
