--- a/src/efibootmgr.c	2019-01-16 16:38:24.694719888 +0100
+++ b/src/efibootmgr.c	2019-01-16 16:39:35.773715651 +0100
@@ -1536,9 +1536,6 @@
 					       "invalid numeric value %s\n",
 					       optarg);
 			}
-                        /* XXX efivar-36 accidentally doesn't have a public
-                         * header for this */
-			extern int efi_set_verbose(int verbosity, FILE *errlog);
 			efi_set_verbose(opts.verbose - 2, stderr);
 			break;
 		case 'V':
