Use pci.h from linux... sys/pci.h is glibc specific.

--- src/lib/scsi_ioctls.c.orig	2014-12-28 10:43:35.513305649 +0100
+++ src/lib/scsi_ioctls.c	2014-12-28 10:43:45.024306074 +0100
@@ -19,11 +19,11 @@
  */
 
 #include <stdio.h>
-#include <sys/pci.h>
 #include <stdint.h>
 #include <sys/stat.h>
 #include <fcntl.h>
 #include <sys/ioctl.h>
+#include <linux/pci.h>
 #include <linux/nvme.h>
 #include "scsi_ioctls.h"
 
