Index: config/acinclude.m4
===================================================================
--- a/config/acinclude.m4
+++ b/config/acinclude.m4
@@ -302,19 +302,8 @@ Please define variable QMAKE and possibl
 Defining QTDIR can help although it is deprecated.])
   fi
   mkdir conftest.d
-  cat > conftest.d/conftest.pro <<\EOF
-changequote(<<, >>)dnl
-message(QMAKE_UIC="$$QMAKE_UIC")dnl qt4 only
-message(QMAKE_MOC="$$QMAKE_MOC")dnl qt4 only
-message(QT_VERSION="$$[QT_VERSION]")
-message(QT_INSTALL_PREFIX="$$[QT_INSTALL_PREFIX]")
-message(QT_INSTALL_DATA="$$[QT_INSTALL_DATA]")
-message(QT_INSTALL_HEADERS="$$[QT_INSTALL_HEADERS]")
-message(QT_INSTALL_BINS="$$[QT_INSTALL_BINS]")
-changequote([, ])dnl
-EOF
-  if ( cd conftest.d && $QMAKE > conftest.out 2>&1 ) ; then
-    sed -e 's/^.*: *//' < conftest.d/conftest.out > conftest.d/conftest.sh
+  if ( cd conftest.d && $QMAKE -query > conftest.out 2>/dev/null ) ; then
+    sed -e 's/:/=/' < conftest.d/conftest.out > conftest.d/conftest.sh
     . conftest.d/conftest.sh
     rm -rf conftest.d
   else
