https://github.com/uncrustify/uncrustify/issues/3196

diff --git a/tests/cli/test_cli_options.py b/tests/cli/test_cli_options.py
index d835822..ea289ca 100755
--- a/tests/cli/test_cli_options.py
+++ b/tests/cli/test_cli_options.py
@@ -705,8 +705,8 @@ def main(args):
     #
     # Test --tracking_space FILE
     #
-    if parsed_args.config == 'Release':
-        eprint("Release")
+    if parsed_args.config == 'Release' or parsed_args.config == 'None':
+        eprint(parsed_args.config)
         if not check_uncrustify_output(
                 uncr_bin,
                 parsed_args,
