--- a/libiberty/Makefile.in	2020-12-03 20:45:14.000000000 +0100
+++ b/libiberty/Makefile.in	2021-01-23 18:39:45.891324445 +0100
@@ -111,7 +111,7 @@
 
 INCDIR=$(srcdir)/$(MULTISRCTOP)../include
 
-COMPILE.c = $(CC) -c @DEFS@ $(CFLAGS) $(CPPFLAGS) -I. -I$(INCDIR) \
-               $(HDEFINES) @ac_libiberty_warn_cflags@ -D_GNU_SOURCE \
+COMPILE.c = $(CC) -c @DEFS@ $(CFLAGS) $(CPPFLAGS) -fPIC -I. -I$(INCDIR) \
+               $(HDEFINES) @ac_libiberty_warn_cflags@ -D_DEFAULT_SOURCE \
                @CET_HOST_FLAGS@
 
