From dc9701e18775c01d0b69fabaa350147f70096da8 Mon Sep 17 00:00:00 2001
From: Mae Dartmann <hello@maedartmann.name>
Date: Wed, 16 Feb 2022 06:08:41 +0000
Subject: [PATCH] meson: fix invalid positional argument

---
 data/meson.build | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/data/meson.build b/data/meson.build
index fd6d5da..74f7e82 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -1,5 +1,5 @@
 appdata_file = 'org.gnome.tweaks.appdata.xml'
-i18n.merge_file(appdata_file,
+i18n.merge_file(
   input: appdata_file + '.in',
   output: appdata_file,
   po_dir: '../po',
@@ -8,7 +8,7 @@ i18n.merge_file(appdata_file,
 )
 
 desktop_file = 'org.gnome.tweaks.desktop'
-i18n.merge_file(desktop_file,
+i18n.merge_file(
   input: desktop_file + '.in',
   output: desktop_file,
   po_dir: '../po',
-- 
GitLab

