--- setup.py.orig
+++ setup.py
@@ -43,7 +43,7 @@
     ),
     python_requires='>=2.7,!=3.0.*,!=3.1.*,!=3.2.*,!=3.3.*',
     install_requires=[
-        'backports.functools_lru_cache',
+        'backports.functools_lru_cache;python_version<"3.4"',
         'six>=1.11.0',
         'more_itertools>=2.6',
     ],
@@ -78,7 +78,7 @@
             'codecov',
 
             'pytest-cov',
-            'backports.unittest_mock',
+            'backports.unittest_mock;python_version<"3.4"',

            # TLS
            'trustme>=0.4.0',

