--- a/makefile
+++ b/makefile
@@ -57,10 +57,10 @@ zxcvbn-inline.o: zxcvbn.c dict-src.h zxc
 	$(CC) $(CPPFLAGS) $(CFLAGS) -c -o zxcvbn-inline.o zxcvbn.c
 
 dict-src.h: dictgen $(WORDS)
-	./dictgen -o dict-src.h $(WORDS)
+	$(VOID_QEMU) ./dictgen -o dict-src.h $(WORDS)
 
 dict-crc.h: dictgen $(WORDS)
-	./dictgen -b -o zxcvbn.dict -h dict-crc.h $(WORDS)
+	$(VOID_QEMU) ./dictgen -b -o zxcvbn.dict -h dict-crc.h $(WORDS)
 
 dictgen: dict-generate.cpp makefile
 	$(CXX) $(CPPFLAGS) -std=c++11 $(CXXFLAGS) \
