--- a/src/names/main.cpp
+++ b/src/names/main.cpp
@@ -365,12 +365,14 @@ ConflictTrackerNotifyEntryRemoved (CName
 CNameConflictTracker::CNameConflictTracker (CTxMemPool &p)
   : txNameConflicts(std::make_shared<std::vector<CTransactionRef>>()), pool(p)
 {
+  using namespace boost::placeholders;
   pool.NotifyEntryRemoved.connect (
     boost::bind (&ConflictTrackerNotifyEntryRemoved, this, _1, _2));
 }
 
 CNameConflictTracker::~CNameConflictTracker ()
 {
+  using namespace boost::placeholders;
   pool.NotifyEntryRemoved.disconnect (
     boost::bind (&ConflictTrackerNotifyEntryRemoved, this, _1, _2));
 }
--- a/src/wallet/test/init_test_fixture.cpp
+++ b/src/wallet/test/init_test_fixture.cpp
@@ -29,7 +29,7 @@ InitWalletDirTestingSetup::InitWalletDir
     fs::create_directories(m_walletdir_path_cases["default"]);
     fs::create_directories(m_walletdir_path_cases["custom"]);
     fs::create_directories(m_walletdir_path_cases["relative"]);
-    std::ofstream f(m_walletdir_path_cases["file"].BOOST_FILESYSTEM_C_STR);
+    std::ofstream f(BOOST_FILESYSTEM_C_STR(m_walletdir_path_cases["file"]));
     f.close();
 }
 
@@ -41,4 +41,4 @@ InitWalletDirTestingSetup::~InitWalletDi
 void InitWalletDirTestingSetup::SetWalletDir(const fs::path& walletdir_path)
 {
     gArgs.ForceSetArg("-walletdir", walletdir_path.string());
-}
\ No newline at end of file
+}
--- a/src/wallet/test/db_tests.cpp
+++ b/src/wallet/test/db_tests.cpp
@@ -18,7 +18,7 @@ BOOST_AUTO_TEST_CASE(getwalletenv_file)
     std::string test_name = "test_name.dat";
     fs::path datadir = SetDataDir("tempdir");
     fs::path file_path = datadir / test_name;
-    std::ofstream f(file_path.BOOST_FILESYSTEM_C_STR);
+    std::ofstream f(BOOST_FILESYSTEM_C_STR(file_path));
     f.close();
 
     std::string filename;
--- a/src/fs.cpp
+++ b/src/fs.cpp
@@ -206,7 +206,7 @@ void ofstream::close()
 }
 #else // __GLIBCXX__
 
-static_assert(sizeof(*fs::path().BOOST_FILESYSTEM_C_STR) == sizeof(wchar_t),
+static_assert(sizeof(BOOST_FILESYSTEM_C_STR(*fs::path())) == sizeof(wchar_t),
     "Warning: This build is using boost::filesystem ofstream and ifstream "
     "implementations which will fail to open paths containing multibyte "
     "characters. You should delete this static_assert to ignore this warning, "
