voids triplets are armv7l-* and arm-* for for armv6.
--- a/compiler-rt/cmake/base-config-ix.cmake
+++ b/compiler-rt/cmake/base-config-ix.cmake
@@ -278,8 +278,9 @@
       else()
         test_target_arch(armv4t "" "-march=armv4t" "-mfloat-abi=soft")
         test_target_arch(armv6m "" "-march=armv6m" "-mfloat-abi=soft")
-        test_target_arch(arm "" "-march=armv7-a" "-mfloat-abi=soft")
-        test_target_arch(armhf "" "-march=armv7-a" "-mfloat-abi=hard")
+        test_target_arch(arm "" "-march=armv6" "-mfloat-abi=hard")
+        test_target_arch(armv7 "" "-march=armv7-a" "-mfloat-abi=soft")
+        test_target_arch(armv7l "" "-march=armv7-a" "-mfloat-abi=hard")
       endif()
     elseif("${COMPILER_RT_DEFAULT_TARGET_ARCH}" MATCHES "avr")
       test_target_arch(avr "__AVR__" "--target=avr")
