From e1ba6dba2d88f004587ea9f987bd3218b7079ace Mon Sep 17 00:00:00 2001
From: Jardel Weyrich <jweyrich@gmail.com>
Date: Tue, 18 Jul 2017 11:45:41 -0300
Subject: [PATCH 07/12] Fix memory leak related to scope stack.

---
 src/output.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git src/output.c src/output.c
index ed2e48e..8c23e92 100644
--- src/output.c
+++ src/output.c
@@ -144,7 +144,7 @@ void output_term(void) {
 
 	// TODO(jweyrich): Should we loop to pop + close + output every scope?
 	if (g_scope_stack != NULL)
-		free(g_scope_stack);
+		STACK_DEALLOC(g_scope_stack);
 
 	_unregister_all_formats();
 }
-- 
2.26.2.672.g232c24e857

