--- a/cmake/FindPolarSSL.cmake
+++ b/cmake/FindPolarSSL.cmake
@@ -67,7 +67,7 @@
 else()
   execute_process(
-    COMMAND grep -w "MBEDTLS_VERSION_STRING" ${POLARSSL_INCLUDE_DIRS}/${POLARSSL_INC_FOLDER}/version.h
+    COMMAND grep -w "MBEDTLS_VERSION_STRING" ${POLARSSL_INCLUDE_DIRS}/${POLARSSL_INC_FOLDER}/build_info.h
-    COMMAND sed -e "s@\s\+@ @g"
+    COMMAND sed -e "s@\\s\\+@ @g"
     COMMAND cut -d\  -f3
     COMMAND sed -e "s@\"@@g"
     OUTPUT_VARIABLE POLARSSL_VERSION_STRING
