diff --git a/gdk/gdkevents.c b/gdk/gdkevents.c
index 58d9020c33..85e4d13216 100644
--- a/gdk/gdkevents.c
+++ b/gdk/gdkevents.c
@@ -2614,6 +2614,16 @@ gdk_scroll_event_get_unit (GdkEvent *event)
   return self->unit;
 }
 
+/**
+ * gdk_scroll_event_get_relative_direction:
+ * @event: (type GdkScrollRelativeDirection): a relative scroll direction.
+ *
+ * Extracts the scroll direction relative to the physical motion.
+ *
+ * Returns: the relative scroll direction.
+ *
+ * Since: 4.20
+ */
 GdkScrollRelativeDirection
 gdk_scroll_event_get_relative_direction (GdkEvent *event)
 {
diff --git a/gdk/gdktoplevel.c b/gdk/gdktoplevel.c
index 75a6b43767..0f4a8aac30 100644
--- a/gdk/gdktoplevel.c
+++ b/gdk/gdktoplevel.c
@@ -872,6 +872,8 @@ gdk_toplevel_unexport_handle (GdkToplevel *toplevel,
  *
  * The capabilities that are available for this toplevel.
  *
+ * Returns: the capabilities of the `GdkToplevel`.
+ *
  * Since: 4.20
  */
 GdkToplevelCapabilities
diff --git a/gsk/gskrendernode.h b/gsk/gskrendernode.h
index 97568d411a..5b2c70ff27 100644
--- a/gsk/gskrendernode.h
+++ b/gsk/gskrendernode.h
@@ -203,11 +203,6 @@ typedef struct _GskBlurNode                     GskBlurNode;
 typedef struct _GskMaskNode                     GskMaskNode;
 typedef struct _GskGLShaderNode                 GskGLShaderNode GDK_DEPRECATED_TYPE_IN_4_16_FOR(GtkGLArea);
 typedef struct _GskSubsurfaceNode               GskSubsurfaceNode;
-/**
- * GskComponentTransferNode:
- *
- * Since: 4.20
- */
 typedef struct _GskComponentTransferNode        GskComponentTransferNode;
 
 GDK_AVAILABLE_IN_ALL
diff --git a/gsk/gskrendernodeimpl.c b/gsk/gskrendernodeimpl.c
index 2b80fbb8ab..d270240801 100644
--- a/gsk/gskrendernodeimpl.c
+++ b/gsk/gskrendernodeimpl.c
@@ -8605,6 +8605,14 @@ gsk_subsurface_node_get_subsurface (const GskRenderNode *node)
 /* }}} */
 /* {{{ GSK_COMPONENT_TRANSFER_NODE */
 
+/**
+ * GskComponentTransferNode:
+ *
+ * A render node for applying a `GskComponentTransfer` for each color
+ * component of the child node.
+ *
+ * Since: 4.20
+ */
 struct _GskComponentTransferNode
 {
   GskRenderNode render_node;
