--- a/gtk/meson.build
+++ b/gtk/meson.build
@@ -1001,6 +1001,7 @@ gtk_deps = [
   epoxy_dep,
   libm,
   graphene_dep,
+  libsysprof_dep,
 ]
 
 if harfbuzz_dep.found() and pangoft_dep.found()
diff --git a/tools/meson.build b/tools/meson.build
index 1811b69..804e37f 100644
--- a/tools/meson.build
+++ b/tools/meson.build
@@ -30,7 +30,7 @@ gtk_tools = [
                          'gtk-builder-tool-enumerate.c',
                          'gtk-builder-tool-preview.c'], [libgtk_dep] ],
   ['gtk4-update-icon-cache', ['updateiconcache.c'] + extra_update_icon_cache_objs, [ libgtk_static_dep ] ],
-  ['gtk4-encode-symbolic-svg', ['encodesymbolic.c'], [ libgtk_static_dep ] ],
+  ['gtk4-encode-symbolic-svg', ['encodesymbolic.c'], [ libgtk_static_dep, libsysprof_dep ] ],
 ]
 
 if os_unix
