diff --git a/src/platform/cmake/modules/FindEspeak.cmake b/src/platform/cmake/modules/FindEspeak.cmake
index 6815e874..a548a232 100644
--- a/src/platform/cmake/modules/FindEspeak.cmake
+++ b/src/platform/cmake/modules/FindEspeak.cmake
@@ -1,5 +1,5 @@
-FIND_PATH( ESPEAK_INCLUDE_DIR espeak/speak_lib.h  )
-FIND_LIBRARY(ESPEAK_LIBRARIES NAMES espeak )
+FIND_PATH( ESPEAK_INCLUDE_DIR espeak/speak_lib.h)
+FIND_LIBRARY(ESPEAK_LIBRARIES NAMES espeak espeak-ng)
 
 IF( ESPEAK_LIBRARIES )
 	SET( ESPEAK_FOUND TRUE )
