--- a/arcan/src/frameserver/encode/default/encode_presets.c
+++ b/arcan/src/frameserver/encode/default/encode_presets.c
@@ -63,7 +63,6 @@
 		dst->storage.video.codec, NULL) != 0){
 		dst->storage.video.codec   = NULL;
 		dst->storage.video.context = NULL;
-		avcodec_close(dst->storage.video.context);
 		return false;
 	}

@@ -91,7 +90,6 @@

 	if (avcodec_open2(
 		dst->storage.audio.context, dst->storage.audio.codec, NULL) != 0){
-		avcodec_close(dst->storage.audio.context);
 		dst->storage.audio.context = NULL;
 		dst->storage.audio.codec   = NULL;
 		return false;
@@ -181,7 +179,6 @@

 	if (avcodec_open2(dst->storage.video.context,
 		dst->storage.video.codec, &opts) != 0){
-		avcodec_close(dst->storage.video.context);
 		dst->storage.video.context = NULL;
 		dst->storage.video.codec   = NULL;
 		return false;
@@ -247,7 +244,6 @@
 		width, height, fps, vbr / 1024);
 	if (avcodec_open2(dst->storage.video.context,
 		dst->storage.video.codec, &opts) != 0){
-		avcodec_close(dst->storage.video.context);
 		dst->storage.video.context = NULL;
 		dst->storage.video.codec   = NULL;
 		return false;
