Index: a/arcan/src/frameserver/decode/default/decode_av.c
===================================================================
--- a/arcan/src/frameserver/decode/default/decode_av.c
+++ b/arcan/src/frameserver/decode/default/decode_av.c
@@ -310,7 +310,7 @@
 /* this just flushes the buffer regardless without respect for pts, this is far
  * from ideal depending on how far into the future pts is versus audio pipeline
  * buffer bloat */
-static void audio_flush(void* vtx, long pts)
+static void audio_flush(void* vtx, int64_t pts)
 {
 	arcan_shmif_lock(&decctx.shmcont);
 	arcan_event ev = {
