Index: a/arcan/src/CMakeLists.txt
===================================================================
--- a/arcan/src/CMakeLists.txt
+++ b/arcan/src/CMakeLists.txt
@@ -1,16 +1,8 @@
 # tried playing nice by using REQUIRE_QUIET but some built-in modules
 # (CheckLibraryExists, ...) wouldn't listen so how about a nice cup
 # of shut up.
-function(message)
-	list(GET ARGV 0 TYPE)
-	if (TYPE STREQUAL "FATAL_ERROR")
-		list(REMOVE_AT ARGV 0)
-		_message(${TYPE} ${CL_RED} "${ARGV}" ${CL_RST})
-	endif()
-endfunction()
-
 function(amsg msg)
-	_message("" ${msg})
+	message("" ${msg})
 endfunction()
 
 if (CLIENT_LIBRARY_BUILD OR BUILD_PRESET STREQUAL "client")
