Fixes failing tests when compiling with musl

diff --git a/src/main.c b/src/main.c
index b9163b6..2e925ab 100644
--- a/src/main.c
+++ b/src/main.c
@@ -19,6 +19,7 @@
 */
 
 #include <errno.h>
+#include <locale.h>
 #include <stdlib.h>
 #include <string.h>
 #include <syslog.h>
@@ -215,6 +216,8 @@ localed_started ()
 gint
 main (gint argc, gchar *argv[])
 {
+	setlocale(LC_CTYPE, "");
+	
     GError *error = NULL;
     GOptionContext *option_context;
     pid_t pid;
diff --git a/tests/bad-read-userconf b/tests/bad-read-userconf
index ffa2308..f20bbb9 100755
--- a/tests/bad-read-userconf
+++ b/tests/bad-read-userconf
@@ -20,7 +20,7 @@ EOF
 sleep 0.1
 sed -i 's/[^]]*]//' scratch/error
 cmp scratch/error << EOF
-: ERROR: Failed to parse configuration: Key file does not have group ?settings?
+: ERROR: Failed to parse configuration: Key file does not have group “settings”
 EOF
 RES=$?
 
diff --git a/tests/try-options b/tests/try-options
index ea8acff..bcac5f8 100755
--- a/tests/try-options
+++ b/tests/try-options
@@ -41,7 +41,7 @@ if [ $RES = 0 ]; then
     sleep 0.1
     cmp scratch/help << EOF
 Usage:
-  mylocaled [OPTION?] - locale settings D-Bus service
+  mylocaled [OPTION…] - locale settings D-Bus service
 
 Help Options:
   -h, --help        Show help options
