# workaround for https://sourceforge.net/p/graphicsmagick/bugs/580/

--- a/Makefile.am
+++ b/Makefile.am
@@ -79,7 +79,6 @@ TESTS_ENVIRONMENT = \
 
 # Tests to run
 TESTS = \
-	$(PERLMAGICK_TESTS) \
 	$(TESTS_TESTS) \
 	$(MAGICKPP_TESTS) \
 	$(WAND_TESTS) \
--- a/Makefile.in
+++ b/Makefile.in
@@ -4767,7 +4767,7 @@ PERLMAGICK_MAINTAINER_TARGETS = $(top_sr
 @WITH_PERL_TRUE@PERLMAGICK_DISTCLEAN_LOCAL_TARGETS = clean-perl
 @WITH_PERL_TRUE@PERLMAGICK_MAINTAINER_CLEAN_LOCAL_TARGETS = distclean-local
 #PERLMAGICK_CHECK_LOCAL_TARGETS = check-perl
-@WITH_PERL_TRUE@PERLMAGICK_TESTS = PerlMagick/PerlMagickCheck.sh
+#@WITH_PERL_TRUE@PERLMAGICK_TESTS = PerlMagick/PerlMagickCheck.sh
 @WITH_PERL_DYNAMIC_FALSE@@WITH_PERL_STATIC_TRUE@@WITH_PERL_TRUE@PERLMAGICK_CHECKSCRIPTS = perl-build
 
 #PERLMAGICK_CHECKSCRIPTS = perl-build
