# workaround for https://sourceforge.net/p/graphicsmagick/bugs/580/

--- a/Makefile.am.orig    2021-02-14 00:46:02.239418403 -0800
+++ b/Makefile.am 2021-02-14 00:46:48.431547444 -0800
@@ -79,7 +79,6 @@
 
 # Tests to run
 TESTS = \
-       $(PERLMAGICK_TESTS) \
        $(TESTS_TESTS) \
        $(MAGICKPP_TESTS) \
        $(WAND_TESTS) \

--- a/Makefile.in.orig    2021-02-14 00:49:51.414058625 -0800
+++ b/Makefile.in 2021-02-14 00:50:19.996138472 -0800
@@ -4767,7 +4767,7 @@
 @WITH_PERL_TRUE@PERLMAGICK_DISTCLEAN_LOCAL_TARGETS = clean-perl
 @WITH_PERL_TRUE@PERLMAGICK_MAINTAINER_CLEAN_LOCAL_TARGETS = distclean-local
 #PERLMAGICK_CHECK_LOCAL_TARGETS = check-perl
-@WITH_PERL_TRUE@PERLMAGICK_TESTS = PerlMagick/PerlMagickCheck.sh
+#@WITH_PERL_TRUE@PERLMAGICK_TESTS = PerlMagick/PerlMagickCheck.sh
 @WITH_PERL_DYNAMIC_FALSE@@WITH_PERL_STATIC_TRUE@@WITH_PERL_TRUE@PERLMAGICK_CHECKSCRIPTS = perl-build
 
 #PERLMAGICK_CHECKSCRIPTS = perl-build
