--- a/contrib/zip/src/zip.h	2019-09-24 19:28:59.000000000 +0200
+++ b/contrib/zip/src/zip.h	2019-09-24 19:28:59.000000000 +0200
@@ -20,7 +20,7 @@
 #endif
 
 #if !defined(_SSIZE_T_DEFINED) && !defined(_SSIZE_T_DEFINED_) &&               \
-    !defined(_SSIZE_T) && !defined(_SSIZE_T_) && !defined(__ssize_t_defined)
+    !defined(_SSIZE_T) && !defined(_SSIZE_T_) && !defined(__ssize_t_defined) && !defined(__DEFINED_ssize_t)
 #define _SSIZE_T
 // 64-bit Windows is the only mainstream platform
 // where sizeof(long) != sizeof(void*)
