--- Makefile.in.orig	2014-01-08 09:01:48.970245932 +0100
+++ Makefile.in	2014-01-08 09:02:23.283386875 +0100
@@ -7390,7 +7390,7 @@ check-am: all-am
 	$(MAKE) $(AM_MAKEFLAGS) check-TESTS check-local
 check: $(BUILT_SOURCES)
 	$(MAKE) $(AM_MAKEFLAGS) check-recursive
-all-am: Makefile $(INFO_DEPS) $(LIBRARIES) $(PROGRAMS) $(MANS) \
+all-am: Makefile $(LIBRARIES) $(PROGRAMS) \
 		$(HEADERS) all-local
 installdirs: installdirs-recursive
 installdirs-am:
@@ -7558,7 +7558,7 @@ install-info-am: $(INFO_DEPS)
 	    install-info --info-dir="$(DESTDIR)$(infodir)" "$(DESTDIR)$(infodir)/$$relfile" || :;\
 	  done; \
 	else : ; fi
-install-man: install-man1
+install-man:
 
 install-pdf: install-pdf-recursive
 
