--- python/ccnet/packet.py.ORIG
+++ python/ccnet/packet.py
@@ -72,7 +72,7 @@
 def parse_header(buf):
     try:
         ver, ptype, length, id = struct.unpack(CCNET_HEADER_FORMAT, buf)
-    except struct.error, e:
+    except struct.error as e:
         raise NetworkError('error when unpack packet header: %s' % e)
     
     return PacketHeader(ver, ptype, length, id)
--- python/ccnet/async/async_client.py.ORIG
+++ python/ccnet/async/async_client.py
@@ -26,7 +26,7 @@
 ]
 
 def debug_print(msg):
-    print msg
+    print(msg)
 
 class AsyncClient(Client):
     '''Async mode client'''
