--- kopete/cmake/modules/FindLibOTR.cmake	2012-08-12 18:08:59.000000000 -0600
+++ kopete/cmake/modules/FindLibOTR.cmake	2012-11-09 12:14:24.251450904 -0700
@@ -13,13 +13,13 @@
     SET(LIBOTR_FIND_QUIETLY TRUE)
 ENDIF (LIBOTR_INCLUDE_DIR AND LIBOTR_LIBRARY)
 
-FIND_PATH(LIBOTR_INCLUDE_DIR libotr/version.h)
+FIND_PATH(LIBOTR_INCLUDE_DIR libotr3/version.h)
 
-FIND_LIBRARY(LIBOTR_LIBRARY NAMES otr libotr)
+FIND_LIBRARY(LIBOTR_LIBRARY NAMES otr3 libotr3)
 
-# Determine version information from libotr/version.h
+# Determine version information from libotr3/version.h
 IF( LIBOTR_INCLUDE_DIR AND LIBOTR_LIBRARY )
-  EXECUTE_PROCESS(COMMAND grep "OTRL_VERSION" "${LIBOTR_INCLUDE_DIR}/libotr/version.h" OUTPUT_VARIABLE output)
+  EXECUTE_PROCESS(COMMAND grep "OTRL_VERSION" "${LIBOTR_INCLUDE_DIR}/libotr3/version.h" OUTPUT_VARIABLE output)
   STRING(REGEX MATCH "OTRL_VERSION \"[0-9]+\\.[0-9]+\\.[0-9]+" LIBOTR_VERSION "${output}")
   STRING(REGEX REPLACE "^OTRL_VERSION \"" "" LIBOTR_VERSION "${LIBOTR_VERSION}")
   # Check if version is at least 3.2.0
--- kopete/plugins/otr/authenticationwizard.h	2012-08-12 18:09:00.000000000 -0600
+++ kopete/plugins/otr/authenticationwizard.h	2012-11-09 12:15:11.181265336 -0700
@@ -26,7 +26,7 @@
   */
 
 extern "C"{
-#include "libotr/proto.h"
+#include "libotr3/proto.h"
 }
 
 #include "kopetechatsession.h"
--- kopete/plugins/otr/otrlchatinterface.h	2012-08-12 18:09:00.000000000 -0600
+++ kopete/plugins/otr/otrlchatinterface.h	2012-11-09 12:16:10.141032744 -0700
@@ -36,10 +36,10 @@
 #include "authenticationwizard.h"
 
 extern "C" {
-#include <libotr/privkey.h>
-#include <libotr/proto.h>
-#include <libotr/message.h>
-#include <libotr/userstate.h>
+#include <libotr3/privkey.h>
+#include <libotr3/proto.h>
+#include <libotr3/message.h>
+#include <libotr3/userstate.h>
 }
 
 
--- kopete/plugins/otr/otrlconfinterface.h	2012-08-12 18:09:00.000000000 -0600
+++ kopete/plugins/otr/otrlconfinterface.h	2012-11-09 12:15:56.961084991 -0700
@@ -32,11 +32,11 @@
 #include <kopetechatsession.h>
 
 extern "C" {
-#include <libotr/privkey.h>
-#include <libotr/proto.h>
-#include <libotr/message.h>
-#include <libotr/userstate.h>
-#include <libotr/context.h>
+#include <libotr3/privkey.h>
+#include <libotr3/proto.h>
+#include <libotr3/message.h>
+#include <libotr3/userstate.h>
+#include <libotr3/context.h>
 }
 
 class KOPETE_OTR_SHARED_EXPORT OtrlConfInterface : public QObject
