--- module/libblueman.c.orig	2016-03-27 09:13:07.203123088 +0200
+++ module/libblueman.c	2016-03-27 09:13:19.497969382 +0200
@@ -28,6 +28,7 @@
 #include <string.h>
 #include <sys/ioctl.h>
 #include <sys/socket.h>
+#include <sys/time.h>
 #include <unistd.h>
 #include <linux/sockios.h>
 #include <linux/if.h>
--- module/modem-prober.c.orig	2016-03-27 09:19:20.635454542 +0200
+++ module/modem-prober.c	2016-03-27 09:19:28.550355592 +0200
@@ -38,6 +38,10 @@
 #define PyString_FromString PyUnicode_FromString
 #endif
 
+#ifndef XCASE
+# define XCASE 4
+#endif
+
 static gboolean verbose = FALSE;
 
 void set_probe_debug(gboolean debug) {
