--- src/tspi/tsp_tcsi_param.c.orig	2015-11-18 10:44:20.263987906 +0100
+++ src/tspi/tsp_tcsi_param.c	2015-11-18 10:45:20.524547393 +0100
@@ -11,7 +11,11 @@
 #include <stdlib.h>
 #include <string.h>
 #include <stdio.h>
+#if defined(__GLIBC__)
 #include <bits/local_lim.h>
+#else
+#define HOST_NAME_MAX 64
+#endif
 #include "trousers/tss.h"
 #include "trousers/trousers.h"
 #include "trousers_types.h" 
