From 9d699d00aada7eaabbc787a8584b0279abe5fa83 Mon Sep 17 00:00:00 2001
From: illiliti <illiliti@protonmail.com>
Date: Sat, 15 Apr 2023 03:36:22 +0300
Subject: [PATCH] tinyramfs: copy modules.builtin.modinfo file

---
 tinyramfs | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/tinyramfs b/tinyramfs
index f2f1d96..7b7a6b0 100755
--- a/tinyramfs
+++ b/tinyramfs
@@ -143,6 +143,9 @@ copy_modules()
     copy_file "${moddir}/${kernel}/modules.builtin" \
               "/lib/modules/${kernel}/modules.builtin" 0644
 
+    copy_file "${moddir}/${kernel}/modules.builtin.modinfo" \
+              "/lib/modules/${kernel}/modules.builtin.modinfo" 0644
+
     depmod -b "$tmpdir" "$kernel"
 }
