From e14a88e5a254c6cd55fe3b83dd1fe6aba6abdfb7 Mon Sep 17 00:00:00 2001
From: Rasmus Thomsen <rasmus.thomsen@protonmail.com>
Date: Thu, 31 Jan 2019 14:36:01 +0100
Subject: [PATCH] feat(gettext): build against system gettext

---
 Cargo.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Cargo.toml b/Cargo.toml
index 96e9d18..df2e068 100644
--- Cargo.toml
+++ Cargo.toml
@@ -35,7 +35,7 @@ xi-rpc = { version="0.2.0", path="vendor/xi-editor/rust/rpc" }
 dirs = "1"
 toml = "0.4"
 tempfile = "3"
-gettext-rs = "0.4"
+gettext-rs = { version="0.4", features=["gettext-system"] }
 crossbeam-deque = "0.7"
 
 [profile.release]
-- 
2.20.1

