--- src/mongo/util/log.cpp.orig	2016-12-09 15:51:39.493596899 +0100
+++ src/mongo/util/log.cpp	2016-12-09 15:51:52.286765108 +0100
@@ -101,7 +101,7 @@ string errnoWithDescription(int errNumbe
     char buf[kBuflen];
     char* msg{nullptr};
 
-#if defined(__GNUC__) && defined(_GNU_SOURCE)
+#if defined(__GNUC__) && defined(_GNU_SOURCE) && defined(__GLIBC__)
     msg = strerror_r(errNumber, buf, kBuflen);
 #elif defined(_WIN32)
 
