--- src/mongo/crypto/crypto_openssl.cpp.orig	2017-06-08 16:11:03.000000000 +0000
+++ src/mongo/crypto/crypto_openssl.cpp	2017-07-05 17:52:36.991580944 +0000
@@ -41,7 +41,7 @@
 #include <openssl/evp.h>
 #include <openssl/hmac.h>
 #include <openssl/sha.h>
-#if OPENSSL_VERSION_NUMBER < 0x10100000L
+#if OPENSSL_VERSION_NUMBER < 0x10100000L || defined(LIBRESSL_VERSION_NUMBER)
 namespace {
 // Copies of OpenSSL after 1.1.0 define new EVP digest routines. We must
 // polyfill used definitions to interact with older OpenSSL versions.
--- src/mongo/util/net/ssl_manager.cpp.orig	2017-06-08 16:11:03.000000000 +0000
+++ src/mongo/util/net/ssl_manager.cpp	2017-07-05 17:59:28.566556412 +0000
@@ -125,7 +125,7 @@
 #endif // MONGO_CONFIG_NEEDS_ASN1_ANY_DEFINITIONS
 // clang-format on
 
-#if OPENSSL_VERSION_NUMBER < 0x10100000L
+#if OPENSSL_VERSION_NUMBER < 0x10100000L || defined(LIBRESSL_VERSION_NUMBER)
 // Copies of OpenSSL after 1.1.0 define new functions for interaction with
 // X509 structure. We must polyfill used definitions to interact with older
 // OpenSSL versions.
