--- SConstruct.orig	2017-07-05 17:34:21.000000000 +0000
+++ SConstruct	2017-07-05 17:35:17.126642925 +0000
@@ -2799,8 +2799,6 @@
     if use_system_version_of_library("boost"):
         if not conf.CheckCXXHeader( "boost/filesystem/operations.hpp" ):
             myenv.ConfError("can't find boost headers")
-        if not conf.CheckBoostMinVersion():
-            myenv.ConfError("system's version of boost is too old. version 1.49 or better required")
 
         # Note that on Windows with using-system-boost builds, the following
         # FindSysLibDep calls do nothing useful (but nothing problematic either)
