From 4ed0f5103bd25f5bdaca5b11bb69e87cd9746f9f Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Sun, 27 Dec 2015 09:13:09 -0700
Subject: [PATCH 4/5] use gnu++11 instead of c++11

---
 src/third_party/gperftools-2.2/SConscript | 3 +++
 1 file changed, 3 insertions(+)

diff --git src/third_party/gperftools-2.2/SConscript src/third_party/gperftools-2.2/SConscript
index 2948eaa..f6d5c9e 100644
--- src/third_party/gperftools-2.2/SConscript
+++ src/third_party/gperftools-2.2/SConscript
@@ -102,6 +102,9 @@ def removeIfPresent(lst, item):
 for to_remove in ['-Werror', "-Wsign-compare","-Wall"]:
     removeIfPresent(env['CCFLAGS'], to_remove)
 
+removeIfPresent(env['CXXFLAGS'], "-std=c++11")
+env.Append( CXXFLAGS=["-std=gnu++11"] )
+
 env.Library(
     target='tcmalloc_minimal',
     source=files,
-- 
2.6.4

