--- a/plugins/netstat/devproc.c
+++ b/plugins/netstat/devproc.c
@@ -25,15 +25,15 @@
 #include <net/if.h>
 #include <net/if_arp.h>
 #include <arpa/inet.h>
-#include <linux/sockios.h>
-#include <linux/types.h>
-#include <linux/ethtool.h>
 #include <iwlib.h>
 #include "nsconfig.h"
 #include "netstat.h"
 #include "statusicon.h"
 #include "devproc.h"
 #include "dbg.h"
+#include <linux/sockios.h>
+#include <linux/types.h>
+#include <linux/ethtool.h>
 
 /* network device list */
 static void netproc_netdevlist_add(NETDEVLIST_PTR *netdev_list,
