--- kodi-addon-inputstream-adaptive-2.6.7/wvdecrypter/cdm/build/build_config.h	2021-02-10 21:23:12.000000000 +0100
+++ -	2023-04-08 14:05:51.583643126 +0200
@@ -138,6 +138,17 @@
 #define ARCH_CPU_32_BITS 1
 #define ARCH_CPU_LITTLE_ENDIAN 1
 #endif
+#elif defined(__riscv) || defined(__riscv__)
+#define ARCH_CPU_RISCV_FAMILY 1
+#define ARCH_CPU_RISCV 1
+#define ARCH_CPU_LITTLE_ENDIAN 1
+#if defined(__riscv_xlen)
+# if (__riscv_xlen == 64)
+#  define ARCH_CPU_64_BITS 1
+# else
+#  define ARCH_CPU_32_BITS 1
+# endif
+#endif
 #else
 #error Please add support for your architecture in build/build_config.h
 #endif
