diff --git a/libjxr.pc.in b/libjxr.pc.in
index b58145105b3..72e4c1de2a5 100644
--- a/libjxr.pc.in
+++ b/libjxr.pc.in
@@ -1,4 +1,4 @@
-prefix=%(DIR_INSTALL)s
+prefix=@CMAKE_INSTALL_PREFIX@
 exec_prefix=${prefix}
 libdir=${exec_prefix}/lib
 includedir=${prefix}/include
@@ -6,7 +6,7 @@ includedir=${prefix}/include
 Name: libjxr
 Description: A library for reading JPEG XR images.
 
-Version: %(JXR_VERSION)s
+Version: @JXRLIB_LIB_VERSION@
 Libs: -L${libdir} -ljpegxr -ljxrglue
 Libs.private: -lm 
-Cflags: -I${includedir}/libjxr/common -I${includedir}/libjxr/image/x86 -I${includedir}/libjxr/image -I${includedir}/libjxr/glue -I${includedir}/libjxr/test -D__ANSI__ -DDISABLE_PERF_MEASUREMENT %(JXR_ENDIAN)s
+Cflags: -I${includedir}/jxrlib -D__ANSI__ -DDISABLE_PERF_MEASUREMENT
