commit 4763141dcb61464fdc3ac11273d0a21874670bb5
Author: Duncaen <mail@duncano.de>
Date:   Wed Aug 3 16:56:29 2016 +0200

    Fix segfault in crypto by increasing algo_cipher array

diff --git lib/crypto/c_src/crypto.c lib/crypto/c_src/crypto.c
index 7183c39..d0044fe 100644
--- lib/crypto/c_src/crypto.c
+++ lib/crypto/c_src/crypto.c
@@ -749,7 +749,7 @@ static ERL_NIF_TERM algo_hash[8];   /* increase when extending the list */
 static int algo_pubkey_cnt;
 static ERL_NIF_TERM algo_pubkey[7]; /* increase when extending the list */
 static int algo_cipher_cnt;
-static ERL_NIF_TERM algo_cipher[20]; /* increase when extending the list */
+static ERL_NIF_TERM algo_cipher[21]; /* increase when extending the list */
 
 static void init_algorithms_types(ErlNifEnv* env)
 {
