--- ./otp_src_20.3/lib/crypto/c_src/crypto.c.orig	2018-04-23 18:04:06.652997515 +0200
+++ ./otp_src_20.3/lib/crypto/c_src/crypto.c	2018-04-23 18:03:47.480968547 +0200
@@ -102,8 +102,10 @@
 #  undef FIPS_SUPPORT
 # endif
 
+# if LIBRESSL_VERSION_NUMBER < 0x2070000fL
 /* LibreSSL wants the 1.0.1 API */
-# define NEED_EVP_COMPATIBILITY_FUNCTIONS
+#  define NEED_EVP_COMPATIBILITY_FUNCTIONS
+# endif
 #endif
 
 
