diff -p1ru a/configure.ac b/configure.ac
--- a/configure.ac	2024-07-24 14:19:43.674000000 +0800
+++ b/configure.ac	2024-07-24 15:48:03.604000000 +0800
@@ -42,6 +42,6 @@ AC_CONFIG_FILES([
 PKG_CHECK_MODULES(APPLET, gtk+-3.0 >= 3.0
-                          indicator3-0.4 >= 0.2
+                          ayatana-indicator3-0.4 >= 0.2
                           dbusmenu-gtk3-0.4 >= 0.2)
 PKG_CHECK_MODULES(SERVICE, gtk+-3.0 >= 3.0
-                           indicator3-0.4 >= 0.2
+                           ayatana-indicator3-0.4 >= 0.2
                            dbusmenu-glib-0.4 >= 0.2)
@@ -72,4 +72,4 @@ if test "x$with_localinstall" = "xyes";
 else
-	INDICATORDIR=`$PKG_CONFIG --variable=indicatordir indicator3-0.4`
-	INDICATORICONSDIR=`$PKG_CONFIG --variable=iconsdir indicator3-0.4`
+	INDICATORDIR=`$PKG_CONFIG --variable=indicatordir ayatana-indicator3-0.4`
+	INDICATORICONSDIR=`$PKG_CONFIG --variable=iconsdir ayatana-indicator3-0.4`
 fi
diff -p1ru a/src/indicator-printers-service.c b/src/indicator-printers-service.c
--- a/src/indicator-printers-service.c	2024-07-24 14:19:43.622000000 +0800
+++ b/src/indicator-printers-service.c	2024-07-24 15:53:32.546000000 +0800
@@ -18,3 +18,3 @@
 
-#include <libindicator/indicator-service.h>
+#include <libayatana-indicator/indicator-service.h>
 #include <libdbusmenu-glib/dbusmenu-glib.h>
diff -p1ru a/src/indicator-printers.c b/src/indicator-printers.c
--- a/src/indicator-printers.c	2017-11-17 23:09:26.000000000 +0800
+++ b/src/indicator-printers.c	2024-07-24 16:01:35.326000000 +0800
@@ -28,4 +28,4 @@
 
-#include <libindicator/indicator.h>
-#include <libindicator/indicator-image-helper.h>
+#include <libayatana-indicator/indicator.h>
+#include <libayatana-indicator/indicator-image-helper.h>
 
diff -p1ru a/src/indicator-printers.h b/src/indicator-printers.h
--- a/src/indicator-printers.h	2017-11-17 23:09:26.000000000 +0800
+++ b/src/indicator-printers.h	2024-07-24 15:56:49.431000000 +0800
@@ -21,3 +21,3 @@
 
-#include <libindicator/indicator-object.h>
+#include <libayatana-indicator/indicator-object.h>
 
