--- SConstruct.orig	2010-03-13 13:59:50.000000000 +0300
+++ SConstruct	2013-04-23 19:31:39.042439981 +0400
@@ -119,6 +119,13 @@
     env.Append( CPPDEFINES=[ "WIN32", "NDEBUG", "_MT" ] )
 elif platform.startswith('linux-gcc'):
     env.Tool( 'default' )
+    env.Replace(AS = os.environ['AS'])
+    env.Replace(AR = os.environ['AR'])
+    env.Replace(NM = os.environ['NM'])
+    env.Replace(CPP = os.environ['CPP'])
+    env.Replace(LD = os.environ['LD'])
+    env.Replace(CXX = os.environ['CXX'])
+    env.Replace(GCC = os.environ['GCC'])
     env.Append( LIBS = ['pthread'], CCFLAGS = "-Wall" )
     env['SHARED_LIB_ENABLED'] = True
 else:
