--- scripts/CMakeLists.txt.orig	2015-03-25 03:19:09 UTC
+++ scripts/CMakeLists.txt
@@ -16,19 +16,16 @@ if(DEFINED BUILD_DEFINES)
 	add_definitions(${BUILD_DEFINES})
 endif()
 
-if(BUILD_SHARED_LIBS)
-    add_library(pugixml SHARED ${SOURCES})
-else()
-    add_library(pugixml STATIC ${SOURCES})
-endif()
+add_library(pugixml SHARED ${SOURCES})
+add_library(pugixml_static STATIC ${SOURCES})
 
 set_target_properties(pugixml PROPERTIES VERSION 1.6 SOVERSION 1)
+set_target_properties(pugixml_static PROPERTIES OUTPUT_NAME pugixml)
 
-install(TARGETS pugixml EXPORT pugixml-config 
+install(TARGETS pugixml pugixml_static
 	ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
 	LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
 	RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
 )
 
 install(FILES ${HEADERS} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
-install(EXPORT pugixml-config DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/pugixml)
