--- a/src/pugiconfig.hpp	2025-01-10 10:15:31.000000000 -0800
+++ b/src/pugiconfig.hpp	2026-09-15 17:56:02.888563275 -0700
@@ -47,7 +47,7 @@
 // #define PUGIXML_HEADER_ONLY
 
 // Uncomment this to enable long long support (usually enabled automatically)
-// #define PUGIXML_HAS_LONG_LONG
+#define PUGIXML_HAS_LONG_LONG
 
 // Uncomment this to enable support for std::string_view (usually enabled automatically)
 // #define PUGIXML_HAS_STRING_VIEW
