--- CMakeLists.txt.bak	2017-09-30 16:33:46.956774231 +0200
+++ CMakeLists.txt	2017-09-30 16:34:08.857775066 +0200
@@ -52,7 +52,7 @@
 
 if(BUILD_PKGCONFIG)
 	# Install library into its own directory under LIBDIR
-	set(INSTALL_SUFFIX /pugixml-${PUGIXML_VERSION_STRING})
+	# set(INSTALL_SUFFIX /pugixml-${PUGIXML_VERSION_STRING})
 endif()
 
 install(TARGETS pugixml EXPORT pugixml-config
@@ -75,4 +75,4 @@
 	add_executable(check ${TEST_SOURCES})
 	target_link_libraries(check pugixml)
 	add_custom_command(TARGET check POST_BUILD COMMAND check WORKING_DIRECTORY ${CMAKE_SOURCE_DIR})
-endif()
\ No newline at end of file
+endif()
