--- openjdk/hotspot/src/os_cpu/linux_ppc/vm/os_linux_ppc.cpp
+++ openjdk/hotspot/src/os_cpu/linux_ppc/vm/os_linux_ppc.cpp
@@ -80,7 +80,7 @@
 # include <pwd.h>
 # include <poll.h>
 # include <ucontext.h>
-
+# include <asm/ptrace.h>
 
 address os::current_stack_pointer() {
   intptr_t* csp;
--- openjdk/hotspot/src/cpu/ppc/vm/macroAssembler_ppc.cpp
+++ openjdk/hotspot/src/cpu/ppc/vm/macroAssembler_ppc.cpp
@@ -46,6 +46,8 @@
 #include "gc_implementation/g1/heapRegion.hpp"
 #endif
 
+#include <asm/ptrace.h>
+
 #ifdef PRODUCT
 #define BLOCK_COMMENT(str) // nothing
 #else
--- openjdk/hotspot/src/os_cpu/linux_ppc/vm/thread_linux_ppc.cpp
+++ openjdk/hotspot/src/os_cpu/linux_ppc/vm/thread_linux_ppc.cpp
@@ -27,6 +27,8 @@
 #include "runtime/frame.inline.hpp"
 #include "runtime/thread.hpp"
 
+#include <asm/ptrace.h>
+
 bool JavaThread::pd_get_top_frame_for_profiling(frame* fr_addr, void* ucontext, bool isInJava) {
   assert(this->is_Java_thread(), "must be JavaThread");
 
