diff --git src/CmdLine.hs src/CmdLine.hs
index 8c3d530..ccf7aac 100644
--- a/src/CmdLine.hs
+++ b/src/CmdLine.hs
@@ -52,7 +52,7 @@ automatic cmd = case cmd of
                 b <- doesDirectoryExist x
                 if b then return cmd{cmdDataDir=x} else do
                     exe <- getExecutablePath
-                    return cmd{cmdDataDir = takeDirectory exe </> "data"}
+                    return cmd{cmdDataDir = takeDirectory exe </> "../share/hlint/data"}
         git cmd
             | cmdGit cmd = do
                 mgit <- findExecutable "git"
