--- a/src/CmdLine.hs
+++ b/src/CmdLine.hs
@@ -58,7 +58,7 @@
                 b <- doesDirectoryExist x
                 if b then pure cmd{cmdDataDir=x} else do
                     exe <- getExecutablePath
-                    pure cmd{cmdDataDir = takeDirectory exe </> "data"}
+                    pure cmd{cmdDataDir = takeDirectory exe </> "../share/hlint/data"}
         git cmd
             | cmdGit cmd = do
                 mgit <- findExecutable "git"
