--- a/bindings/qt/sdk.pri
+++ bindings/qt/sdk.pri
@@ -2,7 +2,7 @@
 MEGASDK_BASE_PATH = $$PWD/../../
 
 VPATH += $$MEGASDK_BASE_PATH
-SOURCES += src/attrmap.cpp \
+SOURCES -= src/attrmap.cpp \
     src/backofftimer.cpp \
     src/base64.cpp \
     src/command.cpp \
@@ -41,7 +41,7 @@ SOURCES += src/attrmap.cpp \
     src/mega_utf8proc.cpp
 
 CONFIG(USE_MEGAAPI) {
-    SOURCES += src/megaapi.cpp src/megaapi_impl.cpp \
+    SOURCES += \
         bindings/qt/QTMegaRequestListener.cpp \
         bindings/qt/QTMegaTransferListener.cpp \
         bindings/qt/QTMegaGlobalListener.cpp \
@@ -52,7 +52,6 @@ CONFIG(USE_MEGAAPI) {
 
 # CONFIG += USE_LIBUV
 CONFIG(USE_LIBUV) {
-    SOURCES += src/mega_http_parser.cpp
     DEFINES += HAVE_LIBUV
     INCLUDEPATH += $$MEGASDK_BASE_PATH/bindings/qt/3rdparty/include/libuv
     win32 {
@@ -98,12 +97,6 @@ win32 {
 }
 
 
-unix {
-SOURCES += src/posix/net.cpp  \
-    src/posix/fs.cpp  \
-    src/posix/waiter.cpp
-}
-
 HEADERS  += include/mega.h \
             include/mega/account.h \
             include/mega/attrmap.h \
@@ -169,9 +162,6 @@ win32 {
 }
 
 unix {
-    !exists($$MEGASDK_BASE_PATH/include/mega/config.h) {
-        error("Configuration file not found! Please re-run configure script located in the project's root directory!")
-    }
     HEADERS  += include/mega/posix/meganet.h  \
             include/mega/posix/megasys.h  \
             include/mega/posix/megafs.h  \
@@ -231,6 +221,7 @@ win32 {
 }
 
 unix:!macx {
+   LIBS += -lmega
    INCLUDEPATH += $$MEGASDK_BASE_PATH/include/mega/posix
    LIBS += -lsqlite3 -lrt
 
