diff --git bindings/qt/sdk.pri bindings/qt/sdk.pri
index c64482fc..0c9c4520 100644
--- bindings/qt/sdk.pri
+++ bindings/qt/sdk.pri
@@ -2,7 +2,7 @@
 MEGASDK_BASE_PATH = $$PWD/../../
 
 VPATH += $$MEGASDK_BASE_PATH
-SOURCES += src/attrmap.cpp \
+SOURCES -= src/attrmap.cpp \
     src/backofftimer.cpp \
     src/base64.cpp \
     src/command.cpp \
@@ -42,7 +42,7 @@ SOURCES += src/attrmap.cpp \
     src/mediafileattribute.cpp
 
 CONFIG(USE_MEGAAPI) {
-  SOURCES += src/megaapi.cpp src/megaapi_impl.cpp
+  SOURCES -= src/megaapi.cpp src/megaapi_impl.cpp
 
   CONFIG(qt) {
     SOURCES += bindings/qt/QTMegaRequestListener.cpp \
@@ -67,7 +67,7 @@ CONFIG(USE_LIBWEBSOCKETS) {
 }
 
 CONFIG(USE_LIBUV) {
-    SOURCES += src/mega_http_parser.cpp
+    SOURCES -= src/mega_http_parser.cpp
     DEFINES += HAVE_LIBUV
     INCLUDEPATH += $$MEGASDK_BASE_PATH/bindings/qt/3rdparty/include/libuv
     win32 {
@@ -197,7 +197,7 @@ win32 {
     CONFIG += USE_CURL
 
     CONFIG(USE_CURL) {
-        SOURCES += src/wincurl/net.cpp  \
+        SOURCES -= src/wincurl/net.cpp  \
             src/wincurl/fs.cpp  \
             src/wincurl/waiter.cpp
         HEADERS += include/mega/wincurl/meganet.h
@@ -205,7 +205,7 @@ win32 {
         LIBS +=  -llibcurl -lcares -llibeay32 -lssleay32
     }
     else {
-        SOURCES += src/win32/net.cpp \
+        SOURCES -= src/win32/net.cpp \
             src/win32/fs.cpp \
             src/win32/waiter.cpp
         HEADERS += include/mega/win32/meganet.h
@@ -218,7 +218,7 @@ win32 {
 
 
 unix {
-SOURCES += src/posix/net.cpp  \
+SOURCES -= src/posix/net.cpp  \
     src/posix/fs.cpp  \
     src/posix/waiter.cpp
 }
@@ -293,7 +293,7 @@ win32 {
 }
 
 unix {
-    !exists($$MEGASDK_BASE_PATH/include/mega/config.h) {
+    exists($$MEGASDK_BASE_PATH/include/mega/config.h) {
         error("Configuration file not found! Please re-run configure script located in the project's root directory!")
     }
     HEADERS  += include/mega/posix/meganet.h  \
@@ -309,7 +309,7 @@ CONFIG(USE_PCRE) {
 
 CONFIG(qt) {
   DEFINES += USE_QT MEGA_QT_LOGGING
-  SOURCES += src/gfx/qt.cpp src/thread/qtthread.cpp
+  SOURCES -= src/gfx/qt.cpp src/thread/qtthread.cpp
 }
 else {
     DEFINES += USE_FREEIMAGE
@@ -385,7 +385,7 @@ win32 {
 
 unix:!macx {
    INCLUDEPATH += $$MEGASDK_BASE_PATH/include/mega/posix
-   LIBS += -lsqlite3 -lrt
+   LIBS += -lmega -lsqlite3 -lrt
 
    exists($$MEGASDK_BASE_PATH/bindings/qt/3rdparty/libs/libcurl.a) {
     LIBS += $$MEGASDK_BASE_PATH/bindings/qt/3rdparty/libs/libcurl.a
