--- pysearpc/server.py.ORIG	2018-06-21 04:49:39.000000000 +0200
+++ pysearpc/server.py	2018-07-10 08:44:54.404532828 +0200
@@ -25,7 +25,7 @@
         """input str -> output str"""
         try:
             argv = json.loads(fcallstr)
-        except Exception, e:
+        except Exception as e:
             raise SearpcError('bad call str: ' + str(e))
 
         service = self.services[svcname]
@@ -41,7 +41,7 @@
     def call_function(self, svcname, fcallstr):
         try:
             retVal = self._call_function(svcname, fcallstr)
-        except Exception, e:
+        except Exception as e:
             ret = {'err_code': 555, 'err_msg': str(e)}
         else:
             ret = {'ret': retVal}
