commit d9d1fffa224d222a4adee088dd94055c9c6a6cec
Author: Wim Taymans <wtaymans@redhat.com>
Date:   Tue Jan 10 21:34:15 2023 +0100

    enable deprecated properties

diff --git a/src/policy-ep.c b/src/policy-ep.c
index 01e442b5e..2a4f9bfc1 100644
--- a/src/policy-ep.c
+++ b/src/policy-ep.c
@@ -28,6 +28,8 @@
 #include <math.h>
 #include <time.h>
 
+#define PW_ENABLE_DEPRECATED
+
 #include "config.h"
 
 #include <spa/node/node.h>
diff --git a/src/policy-node.c b/src/policy-node.c
index eeb74a15c..25002e8dc 100644
--- a/src/policy-node.c
+++ b/src/policy-node.c
@@ -28,6 +28,8 @@
 #include <math.h>
 #include <time.h>
 
+#define PW_ENABLE_DEPRECATED
+
 #include "config.h"
 
 #include <spa/node/node.h>
diff --git a/src/stream-endpoint.c b/src/stream-endpoint.c
index 7d480aa6d..c60d06ee9 100644
--- a/src/stream-endpoint.c
+++ b/src/stream-endpoint.c
@@ -28,6 +28,8 @@
 #include <math.h>
 #include <time.h>
 
+#define PW_ENABLE_DEPRECATED
+
 #include "config.h"
 
 #include <spa/node/node.h>
