--- a/record/drivers/record_ffmpeg.c
+++ b/record/drivers/record_ffmpeg.c
@@ -372,7 +372,7 @@
    if (!audio->buffer)
       return false;

-   audio->outbuf_size = AV_INPUT_BUFFER_MIN_SIZE;
+   audio->outbuf_size = AV_INPUT_BUFFER_PADDING_SIZE;
    audio->outbuf      = (uint8_t*)av_malloc(audio->outbuf_size);

    if (!audio->outbuf)
@@ -949,7 +949,6 @@

    if (handle->audio.codec)
    {
-      avcodec_close(handle->audio.codec);
       av_free(handle->audio.codec);
    }

@@ -957,7 +956,6 @@

    if (handle->video.codec)
    {
-      avcodec_close(handle->video.codec);
       av_free(handle->video.codec);
    }

--- a/cores/libretro-ffmpeg/ffmpeg_core.c
+++ b/cores/libretro-ffmpeg/ffmpeg_core.c
@@ -2088,17 +2088,12 @@

    for (i = 0; i < MAX_STREAMS; i++)
    {
-      if (sctx[i])
-         avcodec_close(sctx[i]);
-      if (actx[i])
-         avcodec_close(actx[i]);
       sctx[i] = NULL;
       actx[i] = NULL;
    }

    if (vctx)
    {
-      avcodec_close(vctx);
       vctx = NULL;
    }
