From fb8ddbcc06afa2a8f25637032fbb8211f6c6c122 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Albin=20Ahlb=C3=A4ck?= <albin.ahlback@gmail.com>
Date: Fri, 20 May 2022 14:16:38 +0200
Subject: [PATCH] Fix bug in test code for GCC 12

---
 test/t-sdiv_qrnnd.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/test/t-sdiv_qrnnd.c b/test/t-sdiv_qrnnd.c
index 2e9822e848..2676903826 100644
--- a/test/t-sdiv_qrnnd.c
+++ b/test/t-sdiv_qrnnd.c
@@ -33,7 +33,7 @@ int main(void)
         {
             d = n_randtest_not_zero(state);
             nh = n_randtest(state);
-        } while ((FLINT_ABS(nh) >= FLINT_ABS(d)/2) || (nh == WORD_MIN));
+        } while ((nh == WORD_MIN) || (FLINT_ABS(nh) >= FLINT_ABS(d)/2));
 
         nl = n_randtest(state);
 

