diff --git a/meson.build b/meson.build
index 535b3db..eb00f50 100644
--- a/meson.build
+++ b/meson.build
@@ -432,11 +432,11 @@ ladspa_symbol_args = []
 lv2_symbol_args = []
 vamp_symbol_args = []
 
-if get_option('buildtype').startswith('release')
+if get_option('debug')
+  config_summary += { 'Build type': 'Debug' }
+else
   config_summary += { 'Build type': 'Release' }
   feature_defines += ['-DNO_THREAD_CHECKS', '-DNO_TIMING', '-DNDEBUG']
-else
-  config_summary += { 'Build type': 'Debug' }
 endif
 
 if system == 'darwin'
