--- src/system/VectorOps.h    Mon Mar 21 21:51:33 2011 +0000
+++ src/system/VectorOps.h    Sun Oct 02 21:12:42 2011 +0100
@@ -84,8 +84,8 @@
 }

 template<typename T>
-inline void v_move(T *const R__ dst,
-                   const T *const R__ src,
+inline void v_move(T *const dst,
+                   const T *const src,
                    const int count)
 {
     memmove(dst, src, count * sizeof(T));
