--- a/Makefile
+++ b/Makefile
@@ -19,10 +19,10 @@
 # http://www.gnu.org/prep/standards/html_node/Directory-Variables.html
 PREFIX ?= /usr/local
 EXEC_PREFIX = $(PREFIX)
-BINDIR = $(EXEC_PREFIX)/bin
-DATAROOTDIR = $(PREFIX)/share
+BINDIR = $(DESTDIR)$(EXEC_PREFIX)/bin
+DATAROOTDIR = $(DESTDIR)$(PREFIX)/share
 DOCDIR = $(DATAROOTDIR)/doc/$(CPROG)
-SYSCONFDIR ?= $(PREFIX)/etc
+SYSCONFDIR ?= $(DESTDIR)/etc
 HTMLDIR = $(DOCDIR)
 INCLUDEDIR = $(DESTDIR)$(PREFIX)/include
 LIBDIR = $(DESTDIR)$(EXEC_PREFIX)/lib
