diff --git a/extras/tables/table-ldap/ber.h b/extras/tables/table-ldap/ber.h
index d656508..b92656c 100644
--- a/extras/tables/table-ldap/ber.h
+++ b/extras/tables/table-ldap/ber.h
@@ -75,7 +75,9 @@ struct ber_oid {
 	size_t		bo_n;
 };
 
-__BEGIN_DECLS
+#ifdef __cpluplus
+extern "C" {
+#endif
 struct ber_element	*ber_get_element(unsigned long);
 void			 ber_set_header(struct ber_element *, int,
 			    unsigned long);
@@ -123,4 +125,6 @@ size_t			 ber_calc_len(struct ber_element *);
 void			 ber_set_application(struct ber *,
 			    unsigned long (*)(struct ber_element *));
 void			 ber_free(struct ber *);
-__END_DECLS
+#ifdef __cpluplus
+}
+#endif

