--- a/dbus-proxy.c	2019-01-15 18:02:01.000000000 +0000
+++ b/dbus-proxy.c	2019-01-15 18:02:01.000000000 +0000
@@ -31,6 +31,16 @@
 
 #include "flatpak-proxy.h"
 
+// Taken from glibc unistd.h
+#ifndef TEMP_FAILURE_RETRY
+# define TEMP_FAILURE_RETRY(expression) \
+  (__extension__                                                              \
+    ({ long int __result;                                                     \
+       do __result = (long int) (expression);                                 \
+       while (__result == -1L && errno == EINTR);                             \
+       __result; }))
+#endif
+
 static const char *argv0;
 static GList *proxies;
 static int sync_fd = -1;
