From c85018e0de56a78ba8809d82ce4570d5c2d0aeeb Mon Sep 17 00:00:00 2001
From: elbachir-one <bachiralfa@gmail.com>
Date: Tue, 3 Dec 2024 16:53:35 +0100
Subject: [PATCH] Fix atomic_flag test() usage for compatibility with older C++
 standards

---
 BasicUsageEnvironment/BasicTaskScheduler.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/BasicUsageEnvironment/BasicTaskScheduler.cpp b/BasicUsageEnvironment/BasicTaskScheduler.cpp
index 0809fc2..d2e2fdf 100644
--- a/BasicUsageEnvironment/BasicTaskScheduler.cpp
+++ b/BasicUsageEnvironment/BasicTaskScheduler.cpp
@@ -188,8 +188,8 @@ void BasicTaskScheduler::SingleStep(unsigned maxDelayTime) {
       if (mask == 0) mask = EVENT_TRIGGER_ID_HIGH_BIT;
 
 #ifndef NO_STD_LIB
-      if (fTriggersAwaitingHandling[i].test()) {
-	fTriggersAwaitingHandling[i].clear();
+	if (fTriggersAwaitingHandling[i].test_and_set()) {
+		fTriggersAwaitingHandling[i].clear(); // Reset after testing, if necessary
 #else
       if (fTriggersAwaitingHandling[i]) {
 	fTriggersAwaitingHandling[i] = False;
-- 
2.46.2

