--- a/cmake/data.cmake
+++ b/cmake/data.cmake
@@ -60,16 +60,18 @@ file(COPY "${DATA_SRC_DIR}/lang" DESTINA
 file(COPY "${DATA_SRC_DIR}/voices/!v" DESTINATION "${DATA_DIST_DIR}/voices")
 file(COPY "${PHONEME_SRC_DIR}" DESTINATION "${DATA_DIST_ROOT}")
 
+if(NOT ESPEAK_RUN_CMD)
 set(ESPEAK_RUN_ENV ${CMAKE_COMMAND} -E env "ESPEAK_DATA_PATH=${DATA_DIST_ROOT}")
 set(ESPEAK_RUN_CMD ${ESPEAK_RUN_ENV} $ENV{VALGRIND} "$<TARGET_FILE:espeak-ng-bin>")
+endif()
 
 add_custom_command(
   OUTPUT "${DATA_DIST_DIR}/intonations"
-  COMMAND ${ESPEAK_RUN_CMD} --compile-intonations
+  COMMAND espeak-ng-bin --compile-intonations
   WORKING_DIRECTORY "${PHONEME_SRC_DIR}"
   COMMENT "Compile intonations"
   DEPENDS
-    "$<TARGET_FILE:espeak-ng-bin>"
+    espeak-ng-bin
     "${PHONEME_SRC_DIR}/intonation"
 )
 
@@ -95,12 +97,12 @@ add_custom_command(
     "${DATA_DIST_DIR}/phondata-manifest"
     "${DATA_DIST_DIR}/phonindex"
     "${DATA_DIST_DIR}/phontab"
-  COMMAND ${ESPEAK_RUN_CMD} --compile-phonemes
+  COMMAND espeak-ng-bin --compile-phonemes
   WORKING_DIRECTORY "${PHONEME_SRC_DIR}"
   COMMENT "Compile phonemes"
   DEPENDS
     "${DATA_DIST_DIR}/intonations"
-    "$<TARGET_FILE:espeak-ng-bin>"
+    espeak-ng-bin
     ${_phon_deps}
 )
 
@@ -133,10 +135,10 @@ foreach(_dict_name ${_dict_compile_list}
   add_custom_command(
     OUTPUT "${_dict_target}"
     COMMAND ${CMAKE_COMMAND} -E copy ${_dict_deps} "${DICT_TMP_DIR}/"
-    COMMAND ${ESPEAK_RUN_CMD} --compile=${_dict_name}
+    COMMAND espeak-ng-bin --compile=${_dict_name}
     WORKING_DIRECTORY "${DICT_TMP_DIR}"
     DEPENDS
-      "$<TARGET_FILE:espeak-ng-bin>"
+      espeak-ng-bin
       "${DATA_DIST_DIR}/phondata"
       "${DATA_DIST_DIR}/intonations"
       ${_dict_deps}
@@ -152,8 +154,8 @@ if (HAVE_MBROLA AND USE_MBROLA)
     list(APPEND _mbr_targets ${_mbl_out})
     add_custom_command(
       OUTPUT "${_mbl_out}"
-      COMMAND ${ESPEAK_RUN_CMD} --compile-mbrola="${_mbl_src}"
-      DEPENDS "$<TARGET_FILE:espeak-ng-bin>" "${_mbl_src}"
+      COMMAND espeak-ng-bin --compile-mbrola="${_mbl_src}"
+      DEPENDS espeak-ng-bin "${_mbl_src}"
     )
   endforeach(_mbl)
 endif()
