--- a/src/libmtp.c
+++ b/src/libmtp.c
@@ -7498,7 +7498,8 @@ static LIBMTP_folder_t *get_subfolders_for_folder(LIBMTP_folder_t *list, uint32_
      * we basically don't care. Hopefully parent_id is maintained for all
      * children, because we rely on that instead.
      */
-    if (ob->oi.AssociationDesc != 0x00000000U) {
+    if (ob->oi.AssociationDesc != 0x00000000U &&
+        ob->oi.AssociationDesc != 0x00000001U) {
       LIBMTP_INFO("MTP extended association type 0x%08x encountered\n", ob->oi.AssociationDesc);
     }
 
