Author: Milan P. Stanić <mps@arvanta.net>
Date:   Sun May 30 07:40:11 2021 +0000

Fix missing PATH_MAX on ppc64le

--- a/libbridge/libbridge_devif.c	2021-05-30 07:59:46.533574878 +0000
+++ b/libbridge/libbridge_devif.c	2021-05-30 08:01:09.964036452 +0000
@@ -24,6 +24,7 @@
 #include <string.h>
 #include <dirent.h>
 #include <fcntl.h>
+#include <limits.h>
 
 #include "libbridge.h"
 #include "libbridge_private.h"

--- a/libbridge/libbridge_init.c	2021-05-30 08:00:15.983737797 +0000
+++ b/libbridge/libbridge_init.c	2021-05-30 08:00:49.573923635 +0000
@@ -24,6 +24,7 @@
 #include <dirent.h>
 #include <sys/types.h>
 #include <sys/stat.h>
+#include <limits.h>
 
 #include "libbridge.h"
 #include "libbridge_private.h"
