From de80deea6575c2148dbd032e2cc530ade1508331 Mon Sep 17 00:00:00 2001
From: Cimbali <me@cimba.li>
Date: Sun, 20 Apr 2025 08:34:04 +0100
Subject: [PATCH] Modernize cfg for changing setuptools requirements

---
 setup.cfg | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/setup.cfg b/setup.cfg
index c8b2f921..a7fa8caa 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -22,7 +22,6 @@ classifiers =
 	Intended Audience :: End Users/Desktop
 	Intended Audience :: Information Technology
 	Intended Audience :: Science/Research
-	License :: OSI Approved :: GNU General Public License v2 or later (GPLv2+)
 	Natural Language :: English
 	Natural Language :: French
 	Natural Language :: German
@@ -111,13 +110,13 @@ no_location = true
 no_wrap = true
 sort_output = true
 omit_header = true
-output-file = pympress/share/locale/pympress.pot
+output_file = pympress/share/locale/pympress.pot
 mapping_file = pympress/share/locale/babel_mapping.cfg
 
 [compile_catalog]
 domain = pympress
 directory = pympress/share/locale/
-use-fuzzy = false
+use_fuzzy = false
 statistics = true
 
 [pysrpm]

