diff --git a/alexandria-tests.asd b/alexandria-tests.asd
index 6b785d3..fa0bb88 100644
--- a/alexandria-tests.asd
+++ b/alexandria-tests.asd
@@ -8,5 +8,6 @@
   :perform (test-op (o c)
              (flet ((run-tests (&rest args)
                       (apply (intern (string '#:run-tests) '#:alexandria-tests) args)))
-               (run-tests :compiled nil)
-               (run-tests :compiled t))))
+               (unless (and (run-tests :compiled nil)
+                            (run-tests :compiled t))
+                 (error "Some tests failed.")))))
