From: Wilfred Hughes <me@wilfred.me.uk>
Date: Fri, 31 Mar 2023 08:19:23 -0700
Subject: [PATCH] Remove tests that no longer apply after 8b842387a

---
 src/lines.rs | 10 ----------
 1 file changed, 10 deletions(-)

diff --git a/src/lines.rs b/src/lines.rs
index 6dcb67d7e..7b670ec05 100644
--- a/src/lines.rs
+++ b/src/lines.rs
@@ -298,11 +298,6 @@ mod tests {
         assert_eq!(codepoint_len("ƒoo"), 3);
     }
 
-    #[test]
-    fn test_split_line_empty() {
-        assert_eq!(split_on_newlines(""), vec![""]);
-    }
-
     #[test]
     fn test_split_line_single() {
         assert_eq!(split_on_newlines("foo"), vec!["foo"]);
@@ -318,11 +313,6 @@ mod tests {
         assert_eq!(split_on_newlines("foo\r\nbar"), vec!["foo", "bar"]);
     }
 
-    #[test]
-    fn test_split_line_with_trailing_newline() {
-        assert_eq!(split_on_newlines("foo\nbar\n"), vec!["foo", "bar", ""]);
-    }
-
     #[test]
     fn test_is_all_whiteapce() {
         assert!(is_all_whitespace(" \n\t"));
