From b2882c699b1444efadd9faee36855a965ac6a215 Mon Sep 17 00:00:00 2001
From: bartoszek <bartoszek.github@bartus.33mail.com>
Date: Thu, 4 Jun 2020 14:10:08 +0200
Subject: [PATCH] Fix qt5:5.15.0 build.

---
 src/Mod/Image/Gui/OpenGLImageBox.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git src/Mod/Image/Gui/OpenGLImageBox.cpp src/Mod/Image/Gui/OpenGLImageBox.cpp
index 404a1a19f..bb22ae0f8 100644
--- src/Mod/Image/Gui/OpenGLImageBox.cpp
+++ src/Mod/Image/Gui/OpenGLImageBox.cpp
@@ -26,6 +26,7 @@
 # include <QSurfaceFormat>
 # include <QMessageBox>
 # include <QPainter>
+# include <QPainterPath>
 #endif
 
 #if defined(__MINGW32__)
-- 
2.27.0

