--- a/src/Base/UnlimitedUnsigned.h
+++ b/src/Base/UnlimitedUnsigned.h
@@ -31,6 +31,7 @@
 #include <vector>
 #include <string>
 #include <stdexcept>
+#include <cstdint>
 
 // ----------------------------------------------------------------------------
 
--- a/src/3rdParty/libE57Format/src/ReaderImpl.cpp
+++ b/src/3rdParty/libE57Format/src/ReaderImpl.cpp
@@ -30,6 +30,8 @@
 #include "Common.h"
 #include "StringFunctions.h"
 
+#include <limits>
+
 namespace e57
 {
    /*!
