--- a/src/Base/Builder3D.h
+++ b/src/Base/Builder3D.h
@@ -26,6 +26,7 @@
 
 // Std. configurations
 
+#include <cstdint>
 #include <sstream>
 #include <vector>
 #include <Base/Tools3D.h>
--- a/src/Base/FileInfo.cpp
+++ b/src/Base/FileInfo.cpp
@@ -22,6 +22,7 @@
  ***************************************************************************/
 
 
+#include <limits.h>
 #include "PreCompiled.h"
 
 #ifndef _PreComp_
