--- a/desktop/ida.desktop	2008-08-28 00:33:58.000000000 +0200
+++ b/desktop/ida.desktop	2008-08-28 00:33:58.000000000 +0200
@@ -1,9 +1,10 @@
 [Desktop Entry]
 Type=Application
-Encoding=UTF-8
 Name=ida
 GenericName=Image Viewer
+Comment=Image Viewer
 Exec=ida %F
-Terminal=no
-Categories=Motif;Graphics;Viewer
-MimeType=image/jpeg;image/tiff;image/png
+Terminal=false
+Categories=Motif;Graphics;Viewer;
+MimeType=image/jpeg;image/tiff;image/png;
+Icon=ida

