From 80b2aecc8e763e00b39c0425be90947a7687745b Mon Sep 17 00:00:00 2001
From: Gerd Hoffmann <kraxel@redhat.com>
Date: Thu, 20 Dec 2018 11:28:18 +0100
Subject: [PATCH] drop cairo-gl dep from meson.build

---
 meson.build | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/meson.build b/meson.build
index 35045ad..3639b97 100644
--- a/meson.build
+++ b/meson.build
@@ -17,7 +17,6 @@ fontconf_dep = dependency('fontconfig')
 pixman_dep   = dependency('pixman-1')
 poppler_dep  = dependency('poppler-glib')
 cairo_dep    = dependency('cairo')
-cairo_gl_dep = dependency('cairo-gl')
 drm_dep      = dependency('libdrm')
 gbm_dep      = dependency('gbm')
 epoxy_dep    = dependency('epoxy')
@@ -68,9 +67,6 @@ trans_src    = ''.join(['jpeg/', jpeg_ver,  '/transupp.c'])
 trans_inc    = include_directories(''.join(['jpeg/', jpeg_ver]))
 
 # finish & write configuration
-if cairo_gl_dep.found()
-    config.set('HAVE_CAIRO_GL', true)
-endif
 if motif_dep.found()
     config.set('HAVE_MOTIF', true)
 endif
@@ -119,7 +115,7 @@ executable('thumbnail.cgi',
 fbpdf_srcs   = [ 'fbpdf.c', 'parseconfig.c', 'fbiconfig.c',
                  'vt.c', 'kbd.c', 'fbtools.c', 'drmtools.c', 'drmtools-egl.c' ]
 fbpdf_deps   = [ drm_dep, gbm_dep, epoxy_dep,
-                 pixman_dep, poppler_dep, cairo_dep, cairo_gl_dep ]
+                 pixman_dep, poppler_dep, cairo_dep ]
 
 executable('fbpdf',
            sources             : fbpdf_srcs,
