--- a/botocore/httpsession.py
+++ b/botocore/httpsession.py
@@ -53,13 +53,10 @@ filter_ssl_warnings()
 logger = logging.getLogger(__name__)
 DEFAULT_TIMEOUT = 60
 MAX_POOL_CONNECTIONS = 10
-DEFAULT_CA_BUNDLE = os.path.join(os.path.dirname(__file__), 'cacert.pem')
+DEFAULT_CA_BUNDLE = '/etc/ssl/certs.pem'
 
-try:
-    from certifi import where
-except ImportError:
-    def where():
-        return DEFAULT_CA_BUNDLE
+def where():
+    return DEFAULT_CA_BUNDLE
 
 
 def get_cert_path(verify):
