--- tests/__init__.py
+++ tests/__init__.py
@@ -13,7 +13,7 @@
 
 import os
 import sys
-import mock
+from unittest import mock
 import time
 import random
 import shutil
@@ -37,6 +37,9 @@
 
 from nose.tools import assert_equal
 
+testroot = os.path.abspath(os.path.dirname(__file__) or os.curdir)
+sys.path.insert(0, os.path.normpath(os.path.join(testroot, '..')))
+
 import botocore.loaders
 import botocore.session
 from botocore.awsrequest import AWSResponse
