--- plug-ins/postscript/diapsft2renderer.c.orig	2014-05-07 13:13:51.997988792 +0200
+++ plug-ins/postscript/diapsft2renderer.c	2014-05-07 13:14:35.810986950 +0200
@@ -29,8 +29,8 @@
 #include <pango/pangoft2.h>
 #include <pango/pango-engine.h>
 /* I'd really rather avoid this */
-#include <freetype/ftglyph.h>
-#include <freetype/ftoutln.h>
+#include <ftglyph.h>
+#include <ftoutln.h>
 
 #define DPI 300
 
