From b68013ed349a625887b5c022bb18245ee20f1292 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Miro=20Hron=C4=8Dok?= <miro@hroncok.cz>
Date: Thu, 20 Aug 2026 11:42:09 +0200
Subject: [PATCH] Update project metadata to support flit-core version 4+

---
 pyproject.toml | 22 +++++++++++-----------
 1 file changed, 11 insertions(+), 11 deletions(-)

diff --git a/pyproject.toml b/pyproject.toml
index bfef87d..7509e74 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,29 +1,29 @@
 [build-system]
-requires = ["flit_core"]
+requires = ["flit_core>=3.11,<5"]
 build-backend = "flit_core.buildapi"
 
-[tool.flit.metadata]
-module = "nkdfu"
-dist-name = "nkdfu"
-author = "Nitrokey"
-author-email = "pypi@nitrokey.com"
-home-page = "https://github.com/Nitrokey/nkdfu"
+[project]
+name = "nkdfu"
+dynamic = ["version", "description"]
+authors = [{ name = "Nitrokey", email = "pypi@nitrokey.com" }]
+urls.Homepage = "https://github.com/Nitrokey/nkdfu"
 requires-python = ">=3.6"
-description-file = "README.md"
-requires = [
+readme = "README.md"
+dependencies = [
   "fire",
   "tqdm",
   "intelhex >= 2.3.0",
   "libusb1 >= 1.9.3",
 ]
 classifiers=[
-    "License :: OSI Approved :: GNU General Public License v2 (GPLv2)",
     "Intended Audience :: Developers",
     "Intended Audience :: End Users/Desktop",
     "Programming Language :: Python :: 3 :: Only",
     "Programming Language :: Python :: 3.6",
     "Programming Language :: Python :: 3.7",
 ]
+license = "GPL-2.0-only"
+license-files = ["LICENSE"]
 
-[tool.flit.scripts]
+[project.scripts]
 nkdfu = "nkdfu.dfu_flash:cli"
