--- utils/utils.go.bck	2018-02-13 03:06:34.295215926 -0500
+++ utils/utils.go	2018-02-13 03:06:52.708205990 -0500
@@ -24,7 +24,7 @@
 
 // GenerateUID returns a uid
 func GenerateUID() string {
-	return uuid.NewV4().String()
+	return uuid.Must(uuid.NewV4()).String()
 }
 
 func GetInput() (string, error) {
--- migrate/migrations.go.bck	2018-02-13 03:10:28.827089361 -0500
+++ migrate/migrations.go	2018-02-13 03:10:51.090077347 -0500
@@ -48,7 +48,7 @@
 		notes := []migrateToV2PostNote{}
 		for _, note := range book {
 			newNote := migrateToV2PostNote{
-				UUID:     uuid.NewV4().String(),
+				UUID:     uuid.Must(uuid.NewV4()).String(),
 				Content:  note.Content,
 				AddedOn:  note.AddedOn,
 				EditedOn: 0,
