From 368134cf950f242a1690bd57d435c887760fb19c Mon Sep 17 00:00:00 2001
From: Jasper Chan <jasperchan515@gmail.com>
Date: Fri, 13 Jul 2018 21:10:12 -0700
Subject: [PATCH] Add NULL repo argument (is optional)

---
 libgit2-glib/ggit-config.c | 1 +
 1 file changed, 1 insertion(+)

diff --git libgit2-glib/ggit-config.c libgit2-glib/ggit-config.c
index b6536ce..2785574 100644
--- libgit2-glib/ggit-config.c
+++ libgit2-glib/ggit-config.c
@@ -295,6 +295,7 @@ ggit_config_add_file (GgitConfig      *config,
 	ret = git_config_add_file_ondisk (_ggit_native_get (config),
 	                                  path,
 	                                  (git_config_level_t)level,
+	                                  NULL,
 	                                  force);
 	g_free (path);
 
-- 
2.18.0

