diff --git a/libgit2-glib/ggit-clone-options.c b/libgit2-glib/ggit-clone-options.c
index 1288a82..9234efc 100644
--- libgit2-glib/ggit-clone-options.c
+++ libgit2-glib/ggit-clone-options.c
@@ -149,7 +149,7 @@ create_repository_wrapper (git_repository **out,
 
 	if (error != NULL)
 	{
-		giterr_set_str (GIT_ERROR, error->message);
+		git_error_set_str (GIT_ERROR, error->message);
 		g_error_free (error);
 
 		if (repository != NULL)
@@ -187,7 +187,7 @@ create_remote_wrapper (git_remote     **out,
 
 	if (error)
 	{
-		giterr_set_str (GIT_ERROR, error->message);
+		git_error_set_str (GIT_ERROR, error->message);
 		g_error_free (error);
 
 		if (remote != NULL)
diff --git a/libgit2-glib/ggit-config.c b/libgit2-glib/ggit-config.c
index acdad95..84601de 100644
--- libgit2-glib/ggit-config.c
+++ libgit2-glib/ggit-config.c
@@ -191,7 +191,7 @@ ggit_config_find_global (void)
 	if (git_config_find_global (&buf) == GIT_OK)
 	{
 		path = g_file_new_for_path (buf.ptr);
-		git_buf_free (&buf);
+		git_buf_dispose (&buf);
 	}
 
 	return path;
@@ -220,7 +220,7 @@ ggit_config_find_system (void)
 	if (git_config_find_system (&buf) == GIT_OK)
 	{
 		path = g_file_new_for_path (buf.ptr);
-		git_buf_free (&buf);
+		git_buf_dispose (&buf);
 	}
 
 	return path;
diff --git a/libgit2-glib/ggit-diff.c b/libgit2-glib/ggit-diff.c
index db9361d..1d62328 100644
--- libgit2-glib/ggit-diff.c
+++ libgit2-glib/ggit-diff.c
@@ -786,13 +786,13 @@ ggit_diff_format_email (GgitDiff                    *diff,
 
 	if (ret != GIT_OK)
 	{
-		git_buf_free (&buf);
+		git_buf_dispose (&buf);
 		_ggit_error_set (error, ret);
 		return NULL;
 	}
 
 	retval = g_strndup (buf.ptr, buf.size);
-	git_buf_free (&buf);
+	git_buf_dispose (&buf);
 
 	return retval;
 }
diff --git a/libgit2-glib/ggit-error.c b/libgit2-glib/ggit-error.c
index 06a21a7..acc1eb2 100644
--- libgit2-glib/ggit-error.c
+++ libgit2-glib/ggit-error.c
@@ -51,7 +51,7 @@ _ggit_error_set (GError **error,
 	g_return_if_fail (err < 0);
 
 	/* TODO: add more kind of errors, see git_error_t */
-	git2_err = giterr_last ();
+	git2_err = git_error_last ();
 	g_set_error_literal (error, GGIT_ERROR,
 	                     err,
 	                     git2_err == NULL ? "" : git2_err->message);
diff --git a/libgit2-glib/ggit-message.c b/libgit2-glib/ggit-message.c
index 0fa9e8c..5100e52 100644
--- libgit2-glib/ggit-message.c
+++ libgit2-glib/ggit-message.c
@@ -46,7 +46,7 @@ ggit_message_prettify (const gchar *message,
 	git_message_prettify (&buf, message, strip_comments, comment_char);
 
 	d = g_strdup (buf.ptr);
-	git_buf_free (&buf);
+	git_buf_dispose (&buf);
 
 	return d;
 }
diff --git a/libgit2-glib/ggit-patch.c b/libgit2-glib/ggit-patch.c
index ee04e4d..eff9e84 100644
--- libgit2-glib/ggit-patch.c
+++ libgit2-glib/ggit-patch.c
@@ -194,7 +194,7 @@ ggit_patch_to_string (GgitPatch  *patch,
 	if (ret == GIT_OK)
 	{
 		result = g_strdup (buf.ptr);
-		git_buf_free (&buf);
+		git_buf_dispose (&buf);
 	}
 
 	return result;
diff --git a/libgit2-glib/ggit-remote-callbacks.c b/libgit2-glib/ggit-remote-callbacks.c
index a4bc130..03064eb 100644
--- libgit2-glib/ggit-remote-callbacks.c
+++ libgit2-glib/ggit-remote-callbacks.c
@@ -160,7 +160,7 @@ credentials_wrap (git_cred     **cred,
 		{
 			if (error)
 			{
-				giterr_set_str (GIT_ERROR, error->message);
+				git_error_set_str (GIT_ERROR, error->message);
 				g_error_free (error);
 
 				return GIT_ERROR;
diff --git a/libgit2-glib/ggit-repository.c b/libgit2-glib/ggit-repository.c
index b9bd2b7..40e06df 100644
--- libgit2-glib/ggit-repository.c
+++ libgit2-glib/ggit-repository.c
@@ -1013,7 +1013,7 @@ ggit_repository_discover_full (GFile        *location,
 	if (ret == GIT_OK)
 	{
 		rep = g_file_new_for_path (buf.ptr);
-		git_buf_free (&buf);
+		git_buf_dispose (&buf);
 	}
 	else
 	{
@@ -3680,7 +3680,7 @@ ggit_repository_get_default_notes_ref (GgitRepository  *repository,
 	}
 
 	ref = g_strdup (buf.ptr);
-	git_buf_free (&buf);
+	git_buf_dispose (&buf);
 
 	return ref;
 }

