--- a/meson_vapi_link.py	2020-08-22 14:36:38.509123984 +0700
+++ b/meson_vapi_link.py	2020-08-22 14:37:05.199021653 +0700
@@ -4,7 +4,7 @@
 import subprocess
 import sys
 
-vapidir = os.path.join(os.environ['MESON_INSTALL_DESTDIR_PREFIX'], sys.argv[1], 'vala', 'vapi')
+vapidir = os.path.join(os.environ['DESTDIR'] + sys.argv[1], 'vala', 'vapi')
 
 # FIXME: meson will not track the creation of these files
 #        https://github.com/mesonbuild/meson/blob/master/mesonbuild/scripts/uninstall.py#L39
