--- a/procdbase.c
+++ b/procdbase.c
@@ -67,7 +67,7 @@
 #include <fcntl.h>
 #include <unistd.h>
 #include <string.h>
-#include <malloc.h>
+#include <stdlib.h>
 
 #include "atop.h"
 #include "photoproc.h"
--- a/photosyst.c	2021-10-11 10:19:03.436752176 -0400
+++ b/photosyst.c	2021-10-11 10:19:17.451816812 -0400
@@ -165,6 +165,7 @@
 #include <dirent.h>
 #include <sys/ioctl.h>
 #include <sys/sysmacros.h>
+#include <limits.h>
 
 #define SCALINGMAXCPU	8	// threshold for scaling info per CPU
 
