From ad4a431513470f8386dd7a8c28aeef35187193fe Mon Sep 17 00:00:00 2001
From: Aldo Gunsing <dev@aldogunsing.nl>
Date: Fri, 17 Dec 2021 00:11:58 +0100
Subject: [PATCH 3/6] More fixes for GHC 9.0

---
 src/Layout/Modifier.hs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Layout/Modifier.hs b/src/Layout/Modifier.hs
index 1fe456e..0d5e44c 100644
--- a/src/Layout/Modifier.hs
+++ b/src/Layout/Modifier.hs
@@ -22,7 +22,7 @@ module Layout.Modifier
     , activatedBy
     ) where
 
-import BasePrelude hiding (Alt, Control, empty, fromList)
+import BasePrelude hiding (Alt, Control, empty, fromList, singleton)
 import Prelude.Unicode hiding ((∈))
 import Data.Foldable.Unicode ((∈))
 import Data.Monoid.Unicode ((∅), (⊕))
-- 
2.36.0

