From c35906df452081d10473468cf4b06aa8623d100b Mon Sep 17 00:00:00 2001
From: Piotr Wójcik <chocimier@tlen.pl>
Date: Mon, 31 Jan 2022 22:20:24 +0100
Subject: version


diff --git a/setup.cfg b/setup.cfg
index 3e496ce..70cda6f 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -1,5 +1,6 @@
 [metadata]
 name = filelock
+version = attr: src.filelock.version.version
 description = A platform independent file lock.
 long_description = file: README.md
 long_description_content_type = text/markdown
-- 
2.35.0

