--- ./src/3ds/SoStream.cpp.orig	2018-01-16 11:18:58.965420684 +0100
+++ ./src/3ds/SoStream.cpp	2018-01-16 11:18:23.140563596 +0100
@@ -176,7 +176,7 @@ do { \
   if (!gotNum) { setBadBit(); return FALSE; } \
  \
   char *ce; \
-  s = '\0'; \
+  *s = '\0'; \
   _convertType_ tempVal = _convertFunc_(buf, &ce, 0); \
  \
   if (ce != s) \
@@ -273,7 +273,7 @@ do { \
 gotAll: \
   \
   char *ce; \
-  s = '\0'; \
+  *s = '\0'; \
   double tempVal = _convertFunc_(buf, &ce); \
  \
   if (ce != s) \
