--- src/main/main.cpp	2015-01-26 15:26:15.000000000 +0100
+++ src/main/main.cpp	2015-07-06 09:35:37.741814165 +0200
@@ -22,7 +22,7 @@
 #include <QMessageBox> // For QT_REQUIRE_VERSION
 #include <iostream>
 
-#if defined(Q_OS_LINUX) || defined(__GLIBC__) || defined(__FreeBSD__) || defined(__HAIKU__)
+#if (defined(Q_OS_LINUX) && defined(__GLIBC__)) || defined(__FreeBSD__) || defined(__HAIKU__)
 #include <signal.h>
 #include <execinfo.h>
 
@@ -166,7 +166,7 @@
     QApplication::setGraphicsSystem(QSL("raster"));
 #endif
 
-#if defined(Q_OS_LINUX) || defined(__GLIBC__) || defined(__FreeBSD__)
+#if (defined(Q_OS_LINUX) && defined(__GLIBC__)) || defined(__FreeBSD__)
     signal(SIGSEGV, qupzilla_signal_handler);
 #endif
 
