https://github.com/sweord/qupzilla/commit/5bdd369c9514d481cf3d0761a383e087cead97c9

--- src/lib/webtab/searchtoolbar.ui
+++ src/lib/webtab/searchtoolbar.ui
@@ -64,9 +64,6 @@
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
-     <property name="shortcut">
-      <string notr="true">Ctrl+Shift+G</string>
-     </property>
      <property name="autoRaise">
       <bool>true</bool>
      </property>
@@ -80,9 +77,6 @@
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
-     <property name="shortcut">
-      <string notr="true">Ctrl+G</string>
-     </property>
      <property name="autoRaise">
       <bool>true</bool>
      </property>
--- src/lib/webtab/searchtoolbar.cpp.orig
+++ src/lib/webtab/searchtoolbar.cpp
@@ -35,7 +35,9 @@
 
     ui->closeButton->setIcon(IconProvider::standardIcon(QStyle::SP_DialogCloseButton));
     ui->next->setIcon(IconProvider::standardIcon(QStyle::SP_ArrowDown));
+    ui->next->setShortcut(QKeySequence("Ctrl+G"));
     ui->previous->setIcon(IconProvider::standardIcon(QStyle::SP_ArrowUp));
+    ui->previous->setShortcut(QKeySequence("Ctrl+Shift+G"));
 
     connect(ui->closeButton, SIGNAL(clicked()), this, SLOT(hide()));
     connect(ui->lineEdit, SIGNAL(textChanged(QString)), this, SLOT(findNext()));
