--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -29,6 +29,12 @@
 set(CMAKE_MODULE_PATH "${IMHEX_BASE_FOLDER}/cmake/modules")
 include("${IMHEX_BASE_FOLDER}/cmake/build_helpers.cmake")
 
+# Void Linux flags for cross compiling libromfs
+set(NATIVE_CMAKE_C_COMPILER "$ENV{CC_host}")
+set(NATIVE_CMAKE_CXX_COMPILER "$ENV{CXX_host}")
+set(NATIVE_CMAKE_C_FLAGS "$ENV{CFLAGS_host}")
+set(NATIVE_CMAKE_CXX_FLAGS "$ENV{CXXFLAGS_host}")
+
 # Setup project
 loadVersion(IMHEX_VERSION)
 setVariableInParent(IMHEX_VERSION ${IMHEX_VERSION})
