--- a/lib/third_party/miniaudio/include/miniaudio.h
+++ b/lib/third_party/miniaudio/include/miniaudio.h
@@ -13124,15 +13124,7 @@
         return ma_result_from_errno(err);
     }
 #else
-#if defined(_WIN32) || defined(__APPLE__)
     *ppFile = fopen(pFilePath, pOpenMode);
-#else
-    #if defined(_FILE_OFFSET_BITS) && _FILE_OFFSET_BITS == 64 && defined(_LARGEFILE64_SOURCE)
-        *ppFile = fopen64(pFilePath, pOpenMode);
-    #else
-        *ppFile = fopen(pFilePath, pOpenMode);
-    #endif
-#endif
     if (*ppFile == NULL) {
         ma_result result = ma_result_from_errno(errno);
         if (result == MA_SUCCESS) {
--- a/plugins/builtin/source/content/providers/disk_provider.cpp
+++ b/plugins/builtin/source/content/providers/disk_provider.cpp
@@ -43,13 +43,6 @@
     #include <sys/disk.h>
 #endif
 
-#if defined(OS_LINUX) && !defined(OS_FREEBSD)
-    #define lseek lseek64
-#elif defined(OS_FREEBSD)
-    #include <sys/disk.h>
-    #define DEFAULT_SECTOR_SIZE 512
-#endif
-
 namespace hex::plugin::builtin {
 
     bool DiskProvider::isAvailable() const {

