commit 135160be16da4f8dd8631bbd8780b8d8ac1e682d
Author: Alexis Ballier <aballier@gentoo.org>
Date:   Thu Mar 1 11:51:31 2012 -0300

    Include libavfilter/avcodec.h for av_vsrc_buffer_add_frame.

diff --git a/lib/DllAvFilter.h b/lib/DllAvFilter.h
index cf91ee9..a6ce6d0 100644
--- a/lib/DllAvFilter.h
+++ b/lib/DllAvFilter.h
@@ -44,13 +44,16 @@ extern "C" {
   #if (defined HAVE_LIBAVFILTER_AVFILTER_H)
     #include <libavfilter/avfiltergraph.h>
     #include <libavfilter/buffersink.h>
+    #include <libavfilter/avcodec.h>
   #elif (defined HAVE_FFMPEG_AVFILTER_H)
     #include <ffmpeg/avfiltergraph.h>
     #include <ffmpeg/buffersink.h>
+    #include <ffmpeg/avcodec.h>
   #endif
 #else
   #include "libavfilter/avfiltergraph.h"
   #include "libavfilter/buffersink.h"
+  #include "libavfilter/avcodec.h"
 #endif
 }
 
