commit 47afab56db3f6f5e8b76f889a6873539b752e9c4
Author: Alexis Ballier <aballier@gentoo.org>
Date:   Sat Apr 14 14:32:38 2012 -0300

    DllAvCodec: Remove now unused private avcodec_guess_channel_layout.

diff --git a/lib/DllAvCodec.h b/lib/DllAvCodec.h
index 1ab00ea..9f3b16c 100644
--- a/lib/DllAvCodec.h
+++ b/lib/DllAvCodec.h
@@ -60,7 +60,6 @@ extern "C" {
   #endif
 
   /* From non-public audioconvert.h */
-  int64_t avcodec_guess_channel_layout(int nb_channels, enum CodecID codec_id, const char *fmt_name);
   struct AVAudioConvert;
   typedef struct AVAudioConvert AVAudioConvert;
   AVAudioConvert *av_audio_convert_alloc(enum AVSampleFormat out_fmt, int out_channels,
@@ -125,7 +124,6 @@ public:
                                const void * const  in[6], const int  in_stride[6], int len)=0;
   virtual int av_dup_packet(AVPacket *pkt)=0;
   virtual void av_init_packet(AVPacket *pkt)=0;
-  virtual int64_t avcodec_guess_channel_layout(int nb_channels, enum CodecID codec_id, const char *fmt_name)=0;
 };
 
 #if (defined USE_EXTERNAL_FFMPEG) || (defined TARGET_DARWIN)
@@ -205,7 +203,6 @@ public:
 
   virtual int av_dup_packet(AVPacket *pkt) { return ::av_dup_packet(pkt); }
   virtual void av_init_packet(AVPacket *pkt) { return ::av_init_packet(pkt); }
-  virtual int64_t avcodec_guess_channel_layout(int nb_channels, enum CodecID codec_id, const char *fmt_name) { return ::avcodec_guess_channel_layout(nb_channels, codec_id, fmt_name); }
 
   // DLL faking.
   virtual bool ResolveExports() { return true; }
@@ -230,7 +227,6 @@ class DllAvCodec : public DllDynamic, DllAvCodecInterface
   DEFINE_FUNC_ALIGNED9(int, __cdecl, av_parser_parse2, AVCodecParserContext*,AVCodecContext*, uint8_t**, int*, const uint8_t*, int, int64_t, int64_t, int64_t)
   DEFINE_METHOD1(int, av_dup_packet, (AVPacket *p1))
   DEFINE_METHOD1(void, av_init_packet, (AVPacket *p1))
-  DEFINE_METHOD3(int64_t, avcodec_guess_channel_layout, (int p1, enum CodecID p2, const char *p3))
 
   LOAD_SYMBOLS();
 
@@ -297,7 +293,6 @@ class DllAvCodec : public DllDynamic, DllAvCodecInterface
     RESOLVE_METHOD(av_audio_convert)
     RESOLVE_METHOD(av_dup_packet)
     RESOLVE_METHOD(av_init_packet)
-    RESOLVE_METHOD(avcodec_guess_channel_layout)
   END_METHOD_RESOLVE()
 
   /* dependencies of libavcodec */
