--- tools/TexturePacker/Makefile.in.orig	2014-05-05 14:06:10.574102302 +0200
+++ tools/TexturePacker/Makefile.in	2014-05-05 14:06:30.389236221 +0200
@@ -20,7 +20,7 @@ CXXFLAGS_FOR_BUILD += \
   -I@abs_top_srcdir@/xbmc/linux
 
 LDFLAGS_FOR_BUILD += -lSDL_image -lSDL -llzo2
-LDFLAGS_FOR_BUILD += -L@abs_top_srcdir@/lib/libsquish -lsquish-native
+LDFLAGS_FOR_BUILD += @abs_top_srcdir@/lib/libsquish/libsquish.a
 
 ifeq ($(findstring Darwin,$(shell uname -s)),Darwin)
 DEFINES += -DTARGET_DARWIN
