--- a/meson.build
+++ b/meson.build
@@ -25,7 +25,10 @@ with_xapian_support = compiler.has_heade
 find_library_in_compiler = meson.version().version_compare('>=0.31.0')
 rt_dep = dependency('rt', required:false)
 docopt_dep = dependency('docopt', static:static_linkage)
-icu_dep = dependency('icu-i18n', static:static_linkage)
+icu_dep = [
+  dependency('icu-i18n', static:static_linkage),
+  dependency('icu-uc', static:static_linkage),
+]
 
 with_writer = host_machine.system() != 'windows'
 
