--- a/gdb/ser-unix.c
+++ b/gdb/ser-unix.c
@@ -529,12 +529,20 @@
   /* Clear the current output baud rate and fill a new value.  */
   tio.c_cflag &= ~CBAUD;
   tio.c_cflag |= BOTHER;
+#ifdef __GLIBC__
   tio.c_ospeed = rate;
+#else
+  tio.__c_ospeed = rate;
+#endif
 
   /* Clear the current input baud rate and fill a new value.  */
   tio.c_cflag &= ~(CBAUD << IBSHIFT);
   tio.c_cflag |= BOTHER << IBSHIFT;
+#ifdef __GLIBC__
   tio.c_ispeed = rate;
+#else
+  tio.__c_ispeed = rate;
+#endif
 
   if (ioctl (fd, req_set, &tio) < 0)
     perror_with_name (_("Can not set custom baud rate"));
