--- a/gdb/python/lib/gdb/command/prompt.py
+++ b/gdb/python/lib/gdb/command/prompt.py
@@ -45,7 +45,7 @@
         self.hook_set = False
 
     def get_show_string (self, pvalue):
-        if self.value is not '':
+        if self.value is not None and self.value != '':
            return "The extended prompt is: " + self.value
         else:
            return "The extended prompt is not set."
@@ -57,7 +57,7 @@
         return ""
 
     def before_prompt_hook(self, current):
-        if self.value is not '':
+        if self.value is not None and self.value != '':
             return gdb.prompt.substitute_prompt(self.value)
         else:
             return None
