Add HOSTCXXFLAGS and HOSTLDFLAGS to the substituted vars.

--- configure.in	2013-11-17 17:54:39.000000000 +0100
+++ configure.in	2016-12-08 19:18:41.299412824 +0100
@@ -89,6 +89,19 @@
 
 AC_SUBST(HOSTCXX_LINK)
 
+if test x"$HOSTCXXFLAGS" = x; then
+  HOSTCXXFLAGS="$CXXFLAGS"
+fi
+
+AC_SUBST(HOSTCXXFLAGS)
+
+if test x"$HOSTLDFLAGS" = x; then
+  HOSTCXXFLAGS="$LDFLAGS"
+fi
+
+AC_SUBST(HOSTLDFLAGS)
+
+
 #this test is broken, because AC_TRY_LINK calls g++
 #AC_MSG_CHECKING([if we need our own C++ allocation operators])
 #AC_TRY_LINK([ void icewm_alloc() {
--- src/Makefile.in	2016-12-08 19:21:46.490133362 +0100
+++ src/Makefile.in	2016-12-08 19:20:49.259218801 +0100
@@ -11,8 +11,10 @@
 
 CXX =           @CXX@
 HOSTCXX =       @HOSTCXX@
+HOSTCXXFLAGS =  @HOSTCXXFLAGS@
 LD =            @CXX_LINK@
 HOSTLD =        @HOSTCXX_LINK@
+HOSTLDFLAGS =   @HOSTLDFLAGS@
 EXEEXT =        @EXEEXT@
 
 DEBUG =         @DEBUG@
@@ -211,11 +213,11 @@
 
 genpref.o: genpref.cc
 	@echo "  HOSTCXX " $@
-	@$(HOSTCXX) $(CXXFLAGS) $(GCCDEP) -c $<
+	@$(HOSTCXX) $(HOSTCXXFLAGS) -DVERSION='"@VERSION@"' $(GCCDEP) -c $<
 	
 genpref$(EXEEXT):
 	@echo "  HOSTLD  " $@
-	@$(HOSTLD) -o $@ $(genpref_OBJS)
+	@$(HOSTLD) $(HOSTLDFLAGS) -o $@ $(genpref_OBJS)
 
 ################################################################################
 
