diff --git a/openmp/libomptarget/include/Shared/SourceInfo.h b/openmp/libomptarget/include/Shared/SourceInfo.h
index 7ce5fd43efc0..c9ff20c59b43 100644
--- a/openmp/libomptarget/include/Shared/SourceInfo.h
+++ b/openmp/libomptarget/include/Shared/SourceInfo.h
@@ -14,6 +14,7 @@
 #define OMPTARGET_SHARED_SOURCE_INFO_H
 
 #include <string>
+#include <stdint.h>
 
 #ifdef _WIN32
 constexpr bool OSWindows = true;
