--- base/debug/stack_trace.cc.orig	2018-12-08 14:11:25.303475116 +0100
+++ base/debug/stack_trace.cc	2018-12-08 18:00:43.874946999 +0100
@@ -229,7 +229,7 @@
 }
 std::string StackTrace::ToStringWithPrefix(const char* prefix_string) const {
   std::stringstream stream;
-#if !defined(__UCLIBC__) && !defined(_AIX)
+#if defined(__GLIBC__) && !defined(_AIX)
   OutputToStreamWithPrefix(&stream, prefix_string);
 #endif
   return stream.str();
--- net/socket/udp_socket_posix.cc.orig	2019-07-03 13:13:46.034342649 -0400
+++ net/socket/udp_socket_posix.cc	2019-07-03 13:23:53.117081909 -0400
@@ -1194,7 +1194,7 @@
     msg_iov->push_back({const_cast<char*>(buffer->data()), buffer->length()});
   msgvec->reserve(buffers.size());
   for (size_t j = 0; j < buffers.size(); j++)
-    msgvec->push_back({{nullptr, 0, &msg_iov[j], 1, nullptr, 0, 0}, 0});
+    msgvec->push_back({{nullptr, 0, &msg_iov[j], 1, 0, 0, 0}, 0});
   int result = HANDLE_EINTR(Sendmmsg(fd, &msgvec[0], buffers.size(), 0));
   SendResult send_result(0, 0, std::move(buffers));
   if (result < 0) {
