--- chrome/browser/metrics/chrome_browser_main_extra_parts_metrics.cc.orig	2021-03-02 18:45:03.000000000 +0100
+++ chrome/browser/metrics/chrome_browser_main_extra_parts_metrics.cc	2021-03-03 17:39:14.944100746 +0100
@@ -55,7 +55,9 @@
 // TODO(crbug.com/1052397): Revisit the macro expression once build flag switch
 // of lacros-chrome is complete.
 #if defined(OS_LINUX) || BUILDFLAG(IS_CHROMEOS_LACROS)
+#if defined(__GLIBC__)
 #include <gnu/libc-version.h>
+#endif
 
 #include "base/linux_util.h"
 #include "base/strings/string_split.h"
@@ -317,6 +319,7 @@
 // TODO(crbug.com/1052397): Revisit the macro expression once build flag switch
 // of lacros-chrome is complete.
 #if defined(OS_LINUX) || BUILDFLAG(IS_CHROMEOS_LACROS)
+#if defined(__GLIBC__)
   base::Version version(gnu_get_libc_version());
 
   UMALinuxGlibcVersion glibc_version_result = UMA_LINUX_GLIBC_NOT_PARSEABLE;
@@ -338,6 +341,7 @@
   }
   base::UmaHistogramSparse("Linux.GlibcVersion", glibc_version_result);
 #endif
+#endif
 }
 
 void RecordTouchEventState() {
