--- ./third_party/usrsctp/usrsctplib/usrsctplib/netinet/sctp_userspace.c.orig
+++ ./third_party/usrsctp/usrsctplib/usrsctplib/netinet/sctp_userspace.c
@@ -72,7 +72,7 @@
 	pthread_setname_np(name);
 #endif
 #if defined(__Userspace_os_Linux)
-	pthread_setname_np(pthread_self(), name);
+	//pthread_setname_np(pthread_self(), name);
 #endif
 #if defined(__Userspace_os_FreeBSD)
 	pthread_set_name_np(pthread_self(), name);
--- ./base/debug/stack_trace.cc.orig
+++ ./base/debug/stack_trace.cc
@@ -33,7 +33,7 @@
 
 std::string StackTrace::ToString() const {
   std::stringstream stream;
-#if !defined(__UCLIBC__)
+#if defined(__GLIBC__)
   OutputToStream(&stream);
 #endif
   return stream.str();
