--- base/debug/stack_trace.cc.orig	2018-12-08 14:11:25.303475116 +0100
+++ base/debug/stack_trace.cc	2018-12-08 18:00:43.874946999 +0100
@@ -229,7 +229,7 @@
 }
 std::string StackTrace::ToStringWithPrefix(const char* prefix_string) const {
   std::stringstream stream;
-#if !defined(__UCLIBC__) && !defined(_AIX)
+#if defined(__GLIBC__) && !defined(_AIX)
   OutputToStreamWithPrefix(&stream, prefix_string);
 #endif
   return stream.str();
