--- ./net/third_party/quiche/src/quiche/quic/core/quic_one_block_arena.h
+++ ./net/third_party/quiche/src/quiche/quic/core/quic_one_block_arena.h
@@ -70,7 +70,7 @@
 
 // QuicConnections currently use around 1KB of polymorphic types which would
 // ordinarily be on the heap. Instead, store them inline in an arena.
-using QuicConnectionArena = QuicOneBlockArena<1280>;
+using QuicConnectionArena = QuicOneBlockArena<1504>;
 
 }  // namespace quic
 
