--- src/animation.cpp	2019-07-08 08:41:51.000000000 +0200
+++ -	2020-07-08 16:48:26.348650427 +0200
@@ -25,6 +25,7 @@
 #include "utils.h"
 #include <QApplication>
 #include <QColor>
+#include <QPainterPath>
 #include <QPainter>
 #include <QDebug>
 
--- src/picker.cpp	2019-07-08 08:41:51.000000000 +0200
+++ -	2020-07-08 16:48:52.281052014 +0200
@@ -26,6 +26,7 @@
 #include "settings.h"
 #include "utils.h"
 #include "colormenu.h"
+#include <QPainterPath>
 #include <QPainter>
 #include <QBitmap>
 #include <QPixmap>
