From 4a8e09f9ebdc5c8efcbc75933138c9337c98e1bb Mon Sep 17 00:00:00 2001
From: Nathan Owens <ndowens@artixlinux.org>
Date: Wed, 30 Dec 2020 15:31:53 -0600
Subject: [PATCH] fix cross build

Replace AC_RUN_* with AC_COMPILE_*
---
diff --git configure.ac configure.ac
index cfe6d41..0fca5f3 100644
--- configure.ac
+++ configure.ac
@@ -30,7 +30,7 @@ AC_DEFINE_UNQUOTED(DAR_VERSION, "AC_PACKAGE_VERSION", [dar and dar_suite version
 
 AM_INIT_AUTOMAKE([subdir-objects])
 AM_GNU_GETTEXT([external])
-AM_GNU_GETTEXT_VERSION
+AM_GNU_GETTEXT_VERSION([0.19])
 XGETTEXT_EXTRA_OPTIONS='--keyword=dar_gettext'
 AM_ICONV
 
@@ -634,7 +634,7 @@ AS_IF(      [test "x$enable_libgcrypt_linking" != "xyes"],
                  min_version_gcrypt_hash_bug="1.6.0"
                  AC_DEFINE_UNQUOTED(MIN_VERSION_GCRYPT_HASH_BUG, "$min_version_gcrypt_hash_bug", [ligcrypt minimum version without hash bug])
                  AC_MSG_CHECKING([for libgcrypt usability])
-                 AC_RUN_IFELSE([AC_LANG_PROGRAM([[ extern "C"
+                 AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[ extern "C"
                                                   {
                                                        #if HAVE_GCRYPT_H
                                                        #include <gcrypt.h>
@@ -655,7 +655,7 @@ AS_IF(      [test "x$enable_libgcrypt_linking" != "xyes"],
                                 ],
                                 [ AC_DEFINE(CRYPTO_AVAILABLE, 1, [header and linking is available to have strong encryption works])
                                   AC_MSG_RESULT([ok])
-                                  AC_RUN_IFELSE([AC_LANG_PROGRAM([[ extern "C"
+                                  AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[ extern "C"
                                                                     {
                                                                      #if HAVE_GCRYPT_H
                                                                      #include <gcrypt.h>
@@ -1464,7 +1464,7 @@ AS_IF(       [ test "x$enable_threadar" != "xyes" ],
                                [ AC_MSG_WARN([Cannot find libthreadar.h header file]) ]
                               )
                AC_MSG_CHECKING([for libthreadar() operationability])
-               AC_RUN_IFELSE([AC_LANG_PROGRAM( [[
+               AC_COMPILE_IFELSE([AC_LANG_PROGRAM( [[
                                     #if HAVE_LIBTHREADAR_LIBTHREADAR_HPP
                                     #include <libthreadar/libthreadar.hpp>
                                     #endif
-- 
2.30.0

