Fix Zsh completion "bad pattern: [" error.

--- a/src/bin/uudoc.rs
+++ b/src/bin/uudoc.rs
@@ -203,7 +203,14 @@
     };
     let bin_name = std::env::var("PROG_PREFIX").unwrap_or_default() + utility;
 
-    clap_complete::generate(shell, &mut command, bin_name, &mut io::stdout());
+    let mut buffer = Vec::new();
+    clap_complete::generate(shell, &mut command, bin_name, &mut buffer);
+    let mut completion = String::from_utf8(buffer).unwrap();
+    if shell == Shell::Zsh {
+        completion = completion.replace("([)", "(\\[)")
+                               .replace("[_commands", "_bracket_commands");
+    }
+    io::stdout().write_all(completion.as_bytes()).unwrap();
     io::stdout().flush().unwrap();
     process::exit(0);
 }
