--- super-intel.c-	2014-07-06 16:31:53.865139451 +0200
+++ super-intel.c	2014-07-06 16:31:45.646139350 +0200
@@ -5075,7 +5075,7 @@
 	spare->num_disks = 1,
 	spare->num_raid_devs = 0,
 	spare->cache_size = mpb->cache_size,
-	spare->pwr_cycle_count = __cpu_to_le32(1),
+	spare->pwr_cycle_count = __cpu_to_le32(1);
 
 	snprintf((char *) spare->sig, MAX_SIGNATURE_LENGTH,
 		 MPB_SIGNATURE MPB_VERSION_RAID0);
