--- mdadm-4.3/util.c
+++ mdadm-4.3/util.c
@@ -32,6 +32,7 @@
 #include	<sys/vfs.h>
 #include	<sys/mman.h>
 #include	<linux/magic.h>
+#include	<limits.h>
 #include	<poll.h>
 #include	<ctype.h>
 #include	<dirent.h>
--- mdadm-4.3/Create.c
+++ mdadm-4.3/Create.c
@@ -28,6 +28,7 @@
 #include	"md_p.h"
 #include	<ctype.h>
 #include	<fcntl.h>
+#include	<linux/falloc.h>
 #include	<signal.h>
 #include	<sys/signalfd.h>
 #include	<sys/wait.h>
