--- a/setup.py	2022-07-20 16:25:14.945476061 -0500
+++ b/setup.py	2022-07-20 16:25:32.661119877 -0500
@@ -11,7 +11,7 @@
 
 LICENSE = 'The MIT License (MIT)'
 DESCRIPTION = 'Finviz Finance. Information downloader.'
-LONG_DESCRIPTION = (HERE / "README_pypi.md").read_text()
+LONG_DESCRIPTION = (HERE / "README.md").read_text()
 LONG_DESC_TYPE = "text/markdown"
 
 INSTALL_REQUIRES = [
@@ -43,4 +43,4 @@
       install_requires=INSTALL_REQUIRES,
       packages=find_packages(),
       python_requires=PYTHON_REQUIRES
-      )
\ No newline at end of file
+      )
