--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -152,7 +152,7 @@ else( NOT CRYPTSETUP_FOUND )
 
 	get_filename_component(libcrypsetupFullPath ${cryptsetup_lib} REALPATH)
 	MESSAGE( STATUS "commiting to cryptsetup library path of: ${libcrypsetupFullPath}" )
-	file( WRITE ${PROJECT_BINARY_DIR}/cryptsetup_library_path.h "\n#define CRYPTSETUP_LIBRARY_PATH \"${libcrypsetupFullPath}\"" )
+	file( WRITE ${PROJECT_BINARY_DIR}/cryptsetup_library_path.h "\n" )
 
 endif( NOT CRYPTSETUP_FOUND )
 
--- a/zuluCrypt-gui/utility.cpp
+++ b/zuluCrypt-gui/utility.cpp
@@ -3174,5 +3174,9 @@ bool utility::canShowKeySlotProperties()
 
 bool utility::libCryptSetupLibraryNotFound()
 {
-	return !utility::pathExists( CRYPTSETUP_LIBRARY_PATH ) ;
+	/*
+	 * always present in Void Linux because
+	 * zulucrypt-gui depends on zulucrypt which depends on libcryptsetup
+	 */
+	return false;
 }
