diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4dd4ac0..efda8cb 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -574,8 +574,8 @@ if(FUSE3_FOUND OR WINFSP)
                        COMMAND ${CMAKE_COMMAND} -E create_symlink dwarfs mount.dwarfs
                        DEPENDS dwarfs-bin)
     list(APPEND SYMLINKS mount.dwarfs)
-    install(FILES ${CMAKE_CURRENT_BINARY_DIR}/mount.dwarfs DESTINATION sbin)
-    install(TARGETS dwarfs-bin RUNTIME DESTINATION sbin)
+    install(FILES ${CMAKE_CURRENT_BINARY_DIR}/mount.dwarfs DESTINATION bin)
+    install(TARGETS dwarfs-bin RUNTIME DESTINATION bin)
   endif()
   list(APPEND BINARY_TARGETS dwarfs-bin)
   list(APPEND MAIN_TARGETS dwarfs_main)
@@ -596,8 +596,8 @@ if(FUSE_FOUND AND (WITH_LEGACY_FUSE OR NOT FUSE3_FOUND))
                      COMMAND ${CMAKE_COMMAND} -E create_symlink dwarfs2 mount.dwarfs2
                      DEPENDS dwarfs2-bin)
   list(APPEND SYMLINKS mount.dwarfs2)
-  install(TARGETS dwarfs2-bin RUNTIME DESTINATION sbin)
-  install(FILES ${CMAKE_CURRENT_BINARY_DIR}/mount.dwarfs2 DESTINATION sbin)
+  install(TARGETS dwarfs2-bin RUNTIME DESTINATION bin)
+  install(FILES ${CMAKE_CURRENT_BINARY_DIR}/mount.dwarfs2 DESTINATION bin)
   list(APPEND BINARY_TARGETS dwarfs2-bin)
   list(APPEND MAIN_TARGETS dwarfs2_main)
 endif()
