diff --git a/src/backing/eds/backing-eds-calendar-source.vala b/src/backing/eds/backing-eds-calendar-source.vala
index 7e30e08..eba3ac6 100644
--- a/src/backing/eds/backing-eds-calendar-source.vala
+++ b/src/backing/eds/backing-eds-calendar-source.vala
@@ -257,7 +257,7 @@ internal class EdsCalendarSource : CalendarSource {
     // Invoked by EdsStore prior to making it available outside of unit
     internal async void open_async(Cancellable? cancellable) throws Error {
         client = (E.CalClient) yield E.CalClient.connect(eds_source, E.CalClientSourceType.EVENTS,
-            cancellable);
+            30, cancellable);
         
         client.bind_property("readonly", this, PROP_READONLY, BindingFlags.SYNC_CREATE);
         client.notify["readonly"].connect(() => {
diff --git a/vapi/libecal-1.2.vapi b/vapi/libecal-1.2.vapi
index 6ead3ec..1f39d14 100644
--- a/vapi/libecal-1.2.vapi
+++ b/vapi/libecal-1.2.vapi
@@ -23,8 +23,8 @@ namespace E {
 		public bool check_save_schedules ();
 		public static bool check_timezones (iCal.icalcomponent comp, GLib.List comps, GLib.Callback tzlookup, void* ecalclient, GLib.Cancellable cancellable) throws GLib.Error;
 		[CCode (finish_name = "e_cal_client_connect_finish")]
-		public static async unowned E.Client connect (E.Source source, E.CalClientSourceType source_type, GLib.Cancellable cancellable) throws GLib.Error;
-		public static unowned E.Client connect_sync (E.Source source, E.CalClientSourceType source_type, GLib.Cancellable cancellable) throws GLib.Error;
+		public static async unowned E.Client connect (E.Source source, E.CalClientSourceType source_type, uint32 wait_for_connected_seconds, GLib.Cancellable cancellable) throws GLib.Error;
+		public static unowned E.Client connect_sync (E.Source source, E.CalClientSourceType source_type, uint32 wait_for_connected_seconds, GLib.Cancellable cancellable) throws GLib.Error;
 		[CCode (finish_name = "e_cal_client_create_object_finish")]
 		public async void create_object (iCal.icalcomponent icalcomp, GLib.Cancellable? cancellable, out string out_uid) throws GLib.Error;
 		public bool create_object_sync (iCal.icalcomponent icalcomp, out string out_uid, GLib.Cancellable? cancellable) throws GLib.Error;
@@ -599,6 +599,8 @@ namespace E {
 	[CCode (cheader_filename = "libecal/libecal.h")]
 	public const string CAL_STATIC_CAPABILITY_NO_GEN_OPTIONS;
 	[CCode (cheader_filename = "libecal/libecal.h")]
+	public const string CAL_STATIC_CAPABILITY_NO_MEMO_START_DATE;
+	[CCode (cheader_filename = "libecal/libecal.h")]
 	public const string CAL_STATIC_CAPABILITY_NO_ORGANIZER;
 	[CCode (cheader_filename = "libecal/libecal.h")]
 	public const string CAL_STATIC_CAPABILITY_NO_PROCEDURE_ALARMS;
