--- california-0.4.0/src/application/california-application.vala	2015-03-24 00:36:24.000000000 +0100
+++ california-0.4.0/src/application/california-application.vala	2017-02-12 16:21:50.338328785 +0100
@@ -55,7 +55,7 @@
         }
     }
     
-    private static const ActionEntry[] action_entries = {
+    private const ActionEntry[] action_entries = {
         // public actions
         { ACTION_CALENDAR_MANAGER, on_calendar_manager },
         { ACTION_HELP, on_help },
--- california-0.4.0/src/host/host-main-window.vala	2015-03-24 00:36:24.000000000 +0100
+++ california-0.4.0/src/host/host-main-window.vala	2017-02-12 16:22:17.723370162 +0100
@@ -71,7 +71,7 @@
     private const string ACTION_RESET_FONT = "reset-font";
     private const string ACCEL_RESET_FONT = "KP_Multiply";
     
-    private static const ActionEntry[] action_entries = {
+    private const ActionEntry[] action_entries = {
         { ACTION_QUICK_CREATE_EVENT, on_quick_create_event },
         { ACTION_JUMP_TO_TODAY, on_jump_to_today },
         { ACTION_NEXT, on_next },
