From d6d25017f28d39a3b6d096ab707d96e1a5d4dfe9 Mon Sep 17 00:00:00 2001
From: Neel Chotai <neel@chot.ai>
Date: Fri, 14 Feb 2020 17:56:23 +0000
Subject: [PATCH] fix musl compilation

---
 plugins/micron/micron-nvme.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/plugins/micron/micron-nvme.c b/plugins/micron/micron-nvme.c
index 8240887..165fcf0 100644
--- plugins/micron/micron-nvme.c
+++ plugins/micron/micron-nvme.c
@@ -13,6 +13,7 @@
 #include "nvme-print.h"
 #include "nvme-ioctl.h"
 #include <sys/ioctl.h>
+#include <limits.h>
 
 #define CREATE_CMD
 #include "micron-nvme.h"
-- 
2.25.0

