commit ddf3610e8326d014360c9550d54a73f97b40e862
Author: q66 <daniel@octaforge.org>
Date:   Wed Apr 27 04:20:41 2022 +0200

    fix build on musl

diff --git a/util/base64.c b/util/base64.c
index e386c11..dad4492 100644
--- a/util/base64.c
+++ b/util/base64.c
@@ -22,6 +22,7 @@
 #include <stdlib.h>
 #include <string.h>
 #include <errno.h>
+#include <sys/types.h>
 
 static const char base64_table[65] =
 	"ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";
