--- a/libs/random/test/multiprecision_float_test.cpp
+++ b/libs/random/test/multiprecision_float_test.cpp
@@ -15,6 +15,7 @@
 #include <boost/test/unit_test.hpp>
 
 #include <boost/core/ignore_unused.hpp>
+#include <boost/mpl/list.hpp>
 #include <boost/multiprecision/cpp_bin_float.hpp>
 #include <boost/multiprecision/cpp_int.hpp>
 #include <boost/multiprecision/debug_adaptor.hpp>
--- a/libs/random/test/multiprecision_int_test.cpp
+++ b/libs/random/test/multiprecision_int_test.cpp
@@ -21,6 +21,7 @@
 #include <boost/test/unit_test.hpp>
 
 #include <boost/core/ignore_unused.hpp>
+#include <boost/mpl/list.hpp>
 #include <boost/multiprecision/debug_adaptor.hpp>
 #include <boost/multiprecision/cpp_bin_float.hpp>
 #include <boost/multiprecision/cpp_int.hpp>
