--- a/libs/phoenix/test/Jamfile
+++ b/libs/phoenix/test/Jamfile
@@ -146,7 +146,7 @@ test-suite phoenix_scope :
     [ run scope/lambda_tests19.cpp ]
     [ run scope/lambda_tests20.cpp ]
     [ run scope/lambda_tests21.cpp ]
-    [ run scope/lambda_tests22.cpp ]
+#    [ run scope/lambda_tests22.cpp ]
     [ run scope/lambda_tests23.cpp ]
     [ run scope/let_tests.cpp ]
     [ run scope/let_tests_113.cpp ]
