--- a/libs/charconv/test/test_float128.cpp
+++ b/libs/charconv/test/test_float128.cpp
@@ -575,12 +575,12 @@ void spot_check_inf(const std::string& b
 
 void test_nanq()
 {
-    BOOST_TEST(isnanq(boost::charconv::detail::nanq())) && BOOST_TEST(!issignaling(boost::charconv::detail::nanq()));
+    BOOST_TEST(isnanq(boost::charconv::detail::nanq())) && BOOST_TEST(!boost::charconv::detail::issignaling(boost::charconv::detail::nanq()));
 }
 
 void test_nans()
 {
-    BOOST_TEST(isnanq(boost::charconv::detail::nans())) && BOOST_TEST(issignaling(boost::charconv::detail::nans()));
+    BOOST_TEST(isnanq(boost::charconv::detail::nans())) && BOOST_TEST(boost::charconv::detail::issignaling(boost::charconv::detail::nans()));
 }
 
 #if defined(__GNUC__) && __GNUC__ < 9 && __GNUC__ >= 5
