--- a/libs/nowide/test/test_stdio.cpp
+++ b/libs/nowide/test/test_stdio.cpp
@@ -101,6 +101,7 @@ void test_main(int, char** argv, char**)
         if(!f2)
             f2 = boost::nowide::freopen(filename.c_str(), "r", f);
         std::cout << " -- no write possible" << std::endl;
+#ifdef __GLIBC__
         {
             TEST(f2 == f);
             TEST(std::fputs("not-written\n", f) < 0);
@@ -111,6 +112,7 @@ void test_main(int, char** argv, char**)
             TEST(std::fgets(buf, 32, f) != 0);
             TEST(strcmp(buf, "foobar\n") == 0);
         }
+#endif
         std::cout << " -- Reopen different file" << std::endl;
         const std::string filename2 = filename + ".1.txt";
         TEST(boost::nowide::freopen(filename2.c_str(), "w", f) == f);
