--- a/libs/system/test/error_code_test.cpp
+++ b/libs/system/test/error_code_test.cpp
@@ -199,7 +199,11 @@ int main( int, char ** )
   // BOOST_TEST( ec_0_system.message().substr(0,36) == "The operation completed successfully" );
 #elif  defined(linux) || defined(__linux) || defined(__linux__)
   // Linux appends value to message as unsigned, so it varies with # of bits
+#ifdef __GLIBC__
   BOOST_TEST( ec.message().substr(0,13) == "Unknown error" );
+#else
+  BOOST_TEST( ec.message() == "No error information" );
+#endif
 #elif defined(__hpux)
   BOOST_TEST( ec.message() == "" );
 #elif defined(__osf__)
@@ -222,7 +226,11 @@ int main( int, char ** )
   BOOST_TEST( econd_ok.message().substr(0,8) == "No error" );
 #elif  defined(linux) || defined(__linux) || defined(__linux__)
   // Linux appends value to message as unsigned, so it varies with # of bits
+#ifdef __GLIBC__
   BOOST_TEST( econd.message().substr(0,13) == "Unknown error" );
+#else
+  BOOST_TEST( econd.message() == "No error information" );
+#endif
 #elif defined(__hpux)
   BOOST_TEST( econd.message() == "" );
 #elif defined(__osf__)
