--- a/libfprint/nbis/include/bozorth.h	2019-08-11 15:17:26.878344708 +0200
+++ b/libfprint/nbis/include/bozorth.h	2019-08-11 15:17:10.303132150 +0200
@@ -218,7 +218,9 @@ struct xytq_struct {
 /* Globals supporting command line options */
 extern int verbose_threshold;
 /* Global supporting error reporting */
+#ifdef __GLIBC__
 extern FILE *stderr;
+#endif
 
 /**************************************************************************/
 /* In: BZ_GBLS.C */
