--- src/invoker/invoker.c	2018-08-31 07:19:31.000000000 +0200
+++ -	2018-08-31 12:26:25.514301761 +0200
@@ -26,7 +26,6 @@
 #include <string.h>
 #include <signal.h>
 #include <sys/socket.h>
-#include <bits/socket.h>
 #include <sys/un.h>
 #include <sys/uio.h>
 #include <sys/time.h>
--- src/launcherlib/booster.cpp	2018-08-31 12:08:21.962843593 +0200
+++ -	2018-08-31 12:30:57.972278758 +0200
@@ -586,7 +586,7 @@
     else
         dlopenFlags |= RTLD_LOCAL;
 
-#if (PLATFORM_ID == Linux)
+#if defined(__GLIBC__)
     if (m_appData->dlopenDeep())
         dlopenFlags |= RTLD_DEEPBIND;
 #endif
--- src/launcherlib/booster.cpp	2018-08-31 12:49:54.813076172 +0200
+++ -	2018-08-31 12:50:06.767009507 +0200
@@ -49,6 +49,10 @@
 
 #include "coverage.h"
 
+#if !defined(__GLIBC__)
+#define basename(src) (strrchr(src,'/') ? strrchr(src,'/')+1 : src)
+#endif
+
 Booster::Booster() :
     m_appData(new AppData),
     m_connection(NULL),
