--- a/accel-pppd/CMakeLists.txt	2023-05-01 16:13:43.302315843 +1100
+++ b/accel-pppd/CMakeLists.txt	2023-05-01 16:14:13.826317298 +1100
@@ -130,7 +130,7 @@
 set_property(TARGET accel-pppd PROPERTY INSTALL_RPATH ${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}/accel-ppp)

 INSTALL(TARGETS accel-pppd
-	RUNTIME DESTINATION sbin
+	RUNTIME DESTINATION bin
 )

 INSTALL(FILES accel-ppp.conf.5 DESTINATION share/man/man5)
