--- a/libgrive/CMakeLists.txt
+++ b/libgrive/CMakeLists.txt
@@ -5,7 +5,7 @@ set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PAT
 find_package(LibGcrypt REQUIRED)
 find_package(CURL REQUIRED)
 find_package(Backtrace)
-find_package(Boost 1.40.0 COMPONENTS program_options filesystem unit_test_framework regex system REQUIRED)
+find_package(Boost 1.40.0 COMPONENTS program_options filesystem unit_test_framework REQUIRED)
 find_package(BFD)
 find_package(CppUnit)
 find_package(Iberty)
