--- include/rapidjson/document.h	2016-08-25 09:59:05.000000000 +0200
+++ include/rapidjson/document.h	2018-05-30 09:55:35.028135241 +0200
@@ -1936,7 +1936,7 @@
         if (count) {
             GenericValue* e = static_cast<GenericValue*>(allocator.Malloc(count * sizeof(GenericValue)));
             SetElementsPointer(e);
-            std::memcpy(e, values, count * sizeof(GenericValue));
+            *e = *values;
         }
         else
             SetElementsPointer(0);
@@ -1949,7 +1949,7 @@
         if (count) {
             Member* m = static_cast<Member*>(allocator.Malloc(count * sizeof(Member)));
             SetMembersPointer(m);
-            std::memcpy(m, members, count * sizeof(Member));
+            *m = *members;
         }
         else
             SetMembersPointer(0);
