--- a/include/rapidjson/internal/regex.h	2016-08-25 09:59:05.000000000 +0200
+++ b/include/rapidjson/internal/regex.h		2018-01-13 21:51:33.959472047 +0100
@@ -300,7 +300,7 @@
                 case '\\': // Escape character
                     if (!CharacterEscape(ds, &codepoint))
                         return; // Unsupported escape character
-                    // fall through to default
+                    // fall-through
 
                 default: // Pattern character
                     PushOperand(operandStack, codepoint);
@@ -534,7 +534,7 @@
                 }
                 else if (!CharacterEscape(ds, &codepoint))
                     return false;
-                // fall through to default
+                // fall-through
 
             default:
                 switch (step) {
@@ -543,7 +543,7 @@
                         step++;
                         break;
                     }
-                    // fall through to step 0 for other characters
+                    // fall-through
 
                 case 0:
                     {
