https://github.com/jangernert/FeedReader/issues/410

--- Constants.vala.in.orig
+++ Constants.vala.in
@@ -43,11 +43,11 @@ namespace FeedReader {
 	}
 
 	public class MediaButton : GLib.Object {
-		public const string PLAY = N_("Play");
-		public const string PAUSE = N_("Pause");
-		public const string MUTE = N_("Mute");
-		public const string UNMUTE = N_("Unmute");
-		public const string CLOSE = N_("Close");
+		public const string PLAY = "Play";
+		public const string PAUSE = "Pause";
+		public const string MUTE = "Mute";
+		public const string UNMUTE = "Unmute";
+		public const string CLOSE = "Close";
 	}
 
 	public class Constants : GLib.Object {
