--- a/pm_python/setup.py.old	2010-11-05 14:43:04.000000000 +0000
+++ b/pm_python/setup.py	2010-11-05 17:40:26.000000000 +0000
@@ -173,7 +173,8 @@
     EXTENSION = dict(
         ext_modules=[ 
             Extension("pyportmidi._pyportmidi", [os.path.join("pyportmidi", "_pyportmidi.pyx")],
-                      library_dirs=["./linux"],
+                      library_dirs=["../build/Release", "../build/Debug"],
+                      include_dirs=["../pm_common", "../porttime"],
                       libraries = ["portmidi", "asound", "pthread"]
                       )
         ]
