--- a/xinetd/confparse.c	2015-05-31 12:28:33.148468478 +0200
+++ b/xinetd/confparse.c	2015-05-31 12:28:41.588557956 +0200
@@ -746,7 +746,7 @@ static status_e check_entry( struct serv
    }
 
 /* #ifndef NO_RPC */
-#if defined(HAVE_RPC_RPCENT_H) || defined(HAVE_NETDB_H)
+#if defined(HAVE_RPC_RPCENT_H) && defined(HAVE_NETDB_H)
    if ( SC_IS_RPC( scp ) && !SC_IS_UNLISTED( scp ) )
    {
       struct rpcent *rep = (struct rpcent *)getrpcbyname( SC_NAME(scp) ) ;
