diff --git a/main-beta.lua b/main-beta.lua
index 1d1eecae..6a2a3e30 100644
--- a/main-beta.lua
+++ b/main-beta.lua
@@ -1,7 +1,11 @@
 local currentPath = debug.getinfo(1, 'S').source:sub(2)
 local rootPath = currentPath:gsub('[/\\]*[^/\\]-$', '')
-loadfile(rootPath .. '/platform.lua')('script-beta')
+
+loadfile(rootPath .. '/platform.lua')(rootPath, 'script-beta')
+
+local math = require 'math'
 local fs = require 'bee.filesystem'
+
 ROOT = fs.path(rootPath)
 LANG = LANG or 'en-US'
 
@@ -9,7 +13,7 @@ collectgarbage('generational', 10, 50)
 --collectgarbage('incremental', 120, 120, 0)
 
 log = require 'log'
-log.init(ROOT, ROOT / 'log' / 'service.log')
+log.init(ROOT, fs.path('/tmp/lua-language-server' .. math.random(1000, 9000)) / 'log' / 'service.log')
 log.info('Lua Lsp startup, root: ', ROOT)
 log.debug('ROOT:', ROOT:string())
 
diff --git a/main.lua b/main.lua
index b7c10ed5..6d3e4035 100644
--- a/main.lua
+++ b/main.lua
@@ -1,21 +1,27 @@
 local currentPath = debug.getinfo(1, 'S').source:sub(2)
 local rootPath = currentPath:gsub('[/\\]*[^/\\]-$', '')
-loadfile(rootPath .. '/platform.lua')('script')
+
+loadfile(rootPath .. '/platform.lua')(rootPath, 'script')
+
+local math = require 'math'
 local fs = require 'bee.filesystem'
+
+rootPath = fs.path(rootPath)
 ROOT = fs.current_path() / rootPath
 LANG = LANG or 'en-US'
+LOG_PATH = fs.path('/tmp/lua-lua-language-server' .. math.random(1000, 9000)) / 'log'
 
 --collectgarbage('generational')
 collectgarbage("setpause", 100)
 collectgarbage("setstepmul", 1000)
 
 log = require 'log'
-log.init(ROOT, ROOT / 'log' / 'service.log')
+log.init(ROOT, LOG_PATH / 'service.log')
 log.info('Lua Lsp startup, root: ', ROOT)
 log.debug('ROOT:', ROOT:string())
 ac = {}
 
-xpcall(dofile, log.debug, rootPath .. '/debugger.lua')
+xpcall(dofile, log.debug, tostring(rootPath / 'debugger.lua'))
 require 'utility'
 local service = require 'service'
 local session = service()
diff --git a/platform.lua b/platform.lua
index 46656fda..a1c1989c 100644
--- a/platform.lua
+++ b/platform.lua
@@ -1,4 +1,4 @@
-local script = ...
+local rootPath, script = ...
 
 local function findExePath()
     local n = 0
@@ -30,13 +30,7 @@ cpath:    %s
 end
 
 local fs = require 'bee.filesystem'
-local rootPath = fs.path(exePath):parent_path():parent_path():remove_filename():string()
-if dll == '.dll' then
-    rootPath = rootPath:gsub('/', '\\')
-    package.path  = rootPath .. script .. '\\?.lua'
-          .. ';' .. rootPath .. script .. '\\?\\init.lua'
-else
-    rootPath = rootPath:gsub('\\', '/')
-    package.path  = rootPath .. script .. '/?.lua'
-          .. ';' .. rootPath .. script .. '/?/init.lua'
-end
+rootPath = fs.absolute(fs.path(rootPath))
+
+package.path  = tostring(rootPath / script / '?.lua')
+      .. ';' .. tostring(rootPath / script / '?' / 'init.lua')
diff --git a/script/workspace.lua b/script/workspace.lua
index 75bc27a0..ba855700 100644
--- a/script/workspace.lua
+++ b/script/workspace.lua
@@ -187,7 +187,7 @@ function mt:init(rootUri)
     end
     log.info('Workspace inited, root: ', self.root)
     log.info('Workspace inited, uri: ', rootUri)
-    local logPath = ROOT / 'log' / (rootUri:gsub('[/:]+', '_') .. '.log')
+    local logPath = LOG_PATH / (rootUri:gsub('[/:]+', '_') .. '.log')
     log.info('Log path: ', logPath)
     log.init(ROOT, logPath)
 end
diff --git a/test.lua b/test.lua
index 72178860..da8f46dc 100644
--- a/test.lua
+++ b/test.lua
@@ -3,16 +3,17 @@ local rootPath = currentPath:gsub('[^/\\]-$', '')
 if rootPath == '' then
     rootPath = './'
 end
-loadfile(rootPath .. 'platform.lua')('script')
+loadfile(rootPath .. 'platform.lua')(rootPath, 'script')
 package.path  = package.path
       .. ';' .. rootPath .. 'test/?.lua'
       .. ';' .. rootPath .. 'test/?/init.lua'
 local fs = require 'bee.filesystem'
 ROOT = fs.absolute(fs.path(rootPath):parent_path())
 LANG = 'en-US'
+LOG_PATH = ROOT / 'log'
 
 log = require 'log'
-log.init(ROOT, ROOT / 'log' / 'test.log')
+log.init(ROOT, LOG_PATH / 'test.log')
 log.debug('测试开始')
 ac = {}
 
