diff --git a/make/detect_platform.lua b/make/detect_platform.lua
index 8dba298..fe8ae3e 100644
--- a/make/detect_platform.lua
+++ b/make/detect_platform.lua
@@ -46,9 +46,8 @@
     else
         -- Use default GCC
         if     lm.platform == nil then
-        elseif lm.platform == "linux-x64" then
-        elseif lm.platform == "linux-arm64" then
-            lm.cc = 'aarch64-linux-gnu-gcc'
+        elseif os.getenv("CC") then
+            lm.cc = os.getenv("CC")
         else
             error "unknown platform"
         end
    end
