--- a/src/Toolbar.cc	2020-12-13 23:57:20.285876509 +0100
+++ b/src/Toolbar.cc	2020-12-13 23:58:32.570856812 +0100
@@ -44,7 +44,7 @@
 {
   timeval now;
   gettimeofday(&now, 0);
-  return (std::max(1000l, ((((resolution - (now.tv_sec % resolution)) * 1000l))
+  return (std::max(static_cast<uint64_t>(1000l), static_cast<uint64_t>((((resolution - (now.tv_sec % resolution)) * 1000l))
                            - (now.tv_usec / 1000l))));
 }
 
