--- a/panel/backends/wayland/wlroots/CMakeLists.txt
+++ b/panel/backends/wayland/wlroots/CMakeLists.txt
@@ -6,7 +6,7 @@ set(BACKEND "backend")
 set(NAME ${PREFIX_NAME}_${PLATFORM_NAME})
 project(${PROGRAM}_${BACKEND}_${NAME})
 
-find_package(Qt6 ${REQUIRED_QT_VERSION} REQUIRED COMPONENTS WaylandClient Concurrent)
+find_package(Qt6 ${REQUIRED_QT_VERSION} REQUIRED COMPONENTS WaylandClient WaylandClientPrivate Concurrent GuiPrivate)
 find_package(Qt6Xdg)
 
 set(PROG_SHARE_DIR ${CMAKE_INSTALL_FULL_DATAROOTDIR}/lxqt/${PROGRAM}/${BACKEND})
