Index: gview_v4l2core/jpeg_decoder.c
===================================================================
--- gview_v4l2core/jpeg_decoder.c.orig
+++ gview_v4l2core/jpeg_decoder.c
@@ -1436,7 +1436,7 @@ int jpeg_init_decoder(int width, int hei
 		exit(-1);
 	}
 
-	codec_data->context->pix_fmt = PIX_FMT_YUV422P;
+	codec_data->context->pix_fmt = AV_PIX_FMT_YUV422P;
 	codec_data->context->width = width;
 	codec_data->context->height = height;
 	//jpeg_ctx->context->dsp_mask = (FF_MM_MMX | FF_MM_MMXEXT | FF_MM_SSE);
Index: gview_v4l2core/uvc_h264.c
===================================================================
--- gview_v4l2core/uvc_h264.c.orig
+++ gview_v4l2core/uvc_h264.c
@@ -970,7 +970,7 @@ int h264_init_decoder(int width, int hei
 	}
 	
 	h264_ctx->context->flags2 |= CODEC_FLAG2_FAST;
-	h264_ctx->context->pix_fmt = PIX_FMT_YUV420P;
+	h264_ctx->context->pix_fmt = AV_PIX_FMT_YUV420P;
 	h264_ctx->context->width = width;
 	h264_ctx->context->height = height;
 	//h264_ctx->context->dsp_mask = (FF_MM_MMX | FF_MM_MMXEXT | FF_MM_SSE);
