--- a/pyproject.toml
+++ b/pyproject.toml
@@ -35,7 +35,7 @@
 '''
 
 [tool.cibuildwheel]
-test-command = "pytest -nauto {project}/tests"
+test-command = "pytest {project}/tests"
 test-extras = "test"
 skip = "cp310-*"
 test-skip = "*_arm64 *_universal2:arm64"
@@ -82,7 +82,6 @@
 minversion = "6.0"
 norecursedirs = ["lib", ".pc", ".git", "venv"]
 testpaths = ["tests"]
-addopts = "-n auto"
 
 [tool.coverage.run]
 concurrency = ["multiprocessing"]
