--- src/xf86DMC.c.orig	2009-03-31 10:09:55.207071509 +0200
+++ src/xf86DMC.c	2009-03-31 10:10:07.844204952 +0200
@@ -301,7 +301,7 @@ DeviceControl (DeviceIntPtr dev, int mod
 		 * Device reports motions on 2 axes in absolute coordinates.
 		 * Axes min and max values are reported in raw coordinates.
 		 */
-		if (InitValuatorClassDeviceStruct (dev, 2, xf86GetMotionEvents,
+		if (InitValuatorClassDeviceStruct (dev, 2,
 						   pInfo->history_size, Absolute) == FALSE)
 			{
 				ErrorF ("Unable to allocate DMC ValuatorClassDeviceStruct\n");
