--- /InstallRules/CMakeLists.txt
+++ /InstallRules/CMakeLists.txt
@@ -16,6 +16,9 @@
 
 if(NOT BUILD_HEADLESS)
   #plugins loaded by rvu at runtime
+ 
+
+  if(WIN32)
   get_target_property(GIF_PLUGIN Qt5::QGifPlugin LOCATION)
   get_target_property(ICO_PLUGIN Qt5::QICOPlugin LOCATION)
   get_target_property(JPEG_PLUGIN Qt5::QJpegPlugin LOCATION)
@@ -25,12 +28,11 @@
   get_filename_component(JPEG_PLUGIN_NAME "${JPEG_PLUGIN}" NAME)
   get_filename_component(TIFF_PLUGIN_NAME "${TIFF_PLUGIN}" NAME)
 
-  if(WIN32)
     get_target_property(QPA_PLUGIN Qt5::QWindowsIntegrationPlugin LOCATION)
   elseif(APPLE)
     get_target_property(QPA_PLUGIN Qt5::QCocoaIntegrationPlugin LOCATION)
   elseif(UNIX)
-    get_target_property(QPA_PLUGIN Qt5::QXcbIntegrationPlugin LOCATION)
+    #  get_target_property(QPA_PLUGIN Qt5::QXcbIntegrationPlugin LOCATION)
   endif()
   get_filename_component(QPA_PLUGIN_NAME "${QPA_PLUGIN}" NAME)
 
