--- a/src/system.hh.in
+++ b/src/system.hh.in
@@ -134,7 +134,7 @@
 
 #include <boost/algorithm/string.hpp>
 #include <boost/any.hpp>
-#include <boost/bind.hpp>
+#include <boost/bind/bind.hpp>
 #include <boost/cast.hpp>
 #include <boost/current_function.hpp>
 
@@ -184,6 +184,8 @@
 #include <boost/regex.hpp>
 #endif // HAVE_BOOST_REGEX_UNICODE
 
+#include <boost/scoped_ptr.hpp>
+
 #include <boost/tokenizer.hpp>
 
 #include <boost/tuple/tuple.hpp>
--- a/src/utils.h
+++ b/src/utils.h
@@ -81,6 +81,7 @@
 
 namespace ledger {
   using namespace boost;
+  using namespace boost::placeholders;
 
   typedef std::string string;
   typedef std::list<string> strings_list;
