--- src/item.h.orig
+++ src/item.h
@@ -92,7 +92,7 @@ class item_t : public supports_flags<uint_least16_t>, public scope_t
 
   typedef std::pair<optional<value_t>, bool> tag_data_t;
   typedef std::map<string, tag_data_t,
-                   function<bool(string, string)> > string_map;
+                   std::function<bool(string, string)> > string_map;
 
   state_t              _state;
   optional<date_t>     _date;
