--- a/Makefile.in
+++ b/Makefile.in
@@ -573,7 +573,7 @@
 				   src/misc.c src/compat.c \
                                    tests/test-shunting-yard.c
 
-tests_test_shunting_yard_LDADD = -lcunit
+tests_test_shunting_yard_LDADD = -lbcunit
 @COND_FUZZER_TRUE@tests_fuzz_shunting_yard_SOURCES = src/shunting-yard.c inc/shunting-yard.h \
 @COND_FUZZER_TRUE@                                   src/stack.c inc/stack.h inc/bitwise.h \
 @COND_FUZZER_TRUE@				   src/misc.c src/compat.c \
--- a/tests/test-shunting-yard.c
+++ b/tests/test-shunting-yard.c
@@ -7,7 +7,7 @@
 
 #include "shunting-yard.h"
 
-#include <CUnit/Basic.h>
+#include <BCUnit/Basic.h>
 #include <stdint.h>
 #include <stdlib.h>
 #include <string.h>
