This function is now defined in glib-2.60

--- libmatekbd/matekbd-keyboard-config.c	2018-06-13 10:50:49.000000000 +0200
+++ libmatekbd/matekbd-keyboard-config.c	2019-04-11 15:47:04.968640566 +0200
@@ -48,35 +48,6 @@
 	MATEKBD_KEYBOARD_CONFIG_KEY_OPTIONS
 };
 
-/*
- * static common functions
- */
-
-static gboolean
-g_strv_equal (gchar ** l1, gchar ** l2)
-{
-	if (l1 == l2)
-		return TRUE;
-	if (l1 == NULL)
-		return g_strv_length (l2) == 0;
-	if (l2 == NULL)
-		return g_strv_length (l1) == 0;
-
-	while ((*l1 != NULL) && (*l2 != NULL)) {
-		if (*l1 != *l2) {
-			if (*l1 && *l2) {
-				if (g_ascii_strcasecmp (*l1, *l2))
-					return FALSE;
-			} else
-				return FALSE;
-		}
-
-		l1++;
-		l2++;
-	}
-	return (*l1 == NULL) && (*l2 == NULL);
-}
-
 gboolean
 matekbd_keyboard_config_get_lv_descriptions (XklConfigRegistry *
 					  config_registry,
