--- a/Makefile.in
+++ b/Makefile.in
@@ -56,8 +56,6 @@
 SVN_ZLIB_LIBS = @SVN_ZLIB_LIBS@
 SVN_LZ4_LIBS = @SVN_LZ4_LIBS@
 SVN_UTF8PROC_LIBS = @SVN_UTF8PROC_LIBS@
-SVN_MACOS_PLIST_LIBS = @SVN_MACOS_PLIST_LIBS@
-SVN_MACOS_KEYCHAIN_LIBS = @SVN_MACOS_KEYCHAIN_LIBS@
 
 LIBS = @LIBS@
 
--- a/configure.ac
+++ b/configure.ac
@@ -479,12 +479,6 @@
             [Defined if Cyrus SASL v2 is present on the system])
 fi
 
-dnl Mac OS specific features -------------------
-
-SVN_LIB_MACHO_ITERATE
-SVN_LIB_MACOS_PLIST
-SVN_LIB_MACOS_KEYCHAIN
-
 dnl APR_HAS_DSO -------------------
 
 AC_MSG_CHECKING([whether APR has support for DSOs])
