--- bCNC	2017-10-02 14:07:33.000000000 +0200
+++ bCNC	2018-03-14 01:20:39.731648471 +0100
@@ -1,10 +1,10 @@
 #!/usr/bin/env sh
 
-DIR=`dirname $0`
+DIR="$(dirname "$(readlink -f "$0")")"
 PYTHONPATH=${DIR}:${DIR}/lib:${DIR}/plugins
 export DIR PYTHONPATH
 if [ .$PYTHON = . ]; then
 	PYTHON=python
 fi
-${PYTHON} ${DIR}/bCNC.py $*
+python2 ${DIR}/bCNC.py $*
 #python -m cProfile -o bCNC.out ${DIR}/bCNC.py $*
