diff -ur a/unittests/linuxliketests.py b/unittests/linuxliketests.py
--- a/unittests/linuxliketests.py	2025-09-22 13:49:50.000000000 -0400
+++ b/unittests/linuxliketests.py	2025-11-08 22:13:29.868759969 -0500
@@ -308,6 +308,7 @@
                              r'Run-time dependency qt5 \(modules: Core\) found: YES 5.* \(pkg-config\)')
 
     @skip_if_not_base_option('b_sanitize')
+    @unittest.skip('asan not available on musl')
     def test_generate_gir_with_address_sanitizer(self):
         if is_cygwin():
             raise SkipTest('asan not available on Cygwin')
@@ -447,6 +448,7 @@
         self._test_soname_impl(libdir, True)
 
     @skip_if_not_base_option('b_sanitize')
+    @unittest.skip('asan not available on musl')
     def test_c_link_args_and_env(self):
         '''
         Test that the CFLAGS / CXXFLAGS environment variables are
@@ -1004,6 +1006,7 @@
             self.assertEqual(got_rpath, yonder_libdir, rpath_format)
 
     @skip_if_not_base_option('b_sanitize')
+    @unittest.skip('asan not available on musl')
     def test_env_cflags_ldflags(self):
         if is_cygwin():
             raise SkipTest('asan not available on Cygwin')
@@ -1020,6 +1023,7 @@
         self.wipe()
 
     @skip_if_not_base_option('b_sanitize')
+    @unittest.skip('asan not available on musl')
     def test_pch_with_address_sanitizer(self):
         if is_cygwin():
             raise SkipTest('asan not available on Cygwin')
