From 4cbfb48faf37bb52de76596c8a8007ca4c1f902e Mon Sep 17 00:00:00 2001
From: q66 <q66@chimera-linux.org>
Date: Fri, 30 May 2025 12:33:02 +0200
Subject: [PATCH 02/15] remove a bunch of unused junk

diff --git a/compiler/rustc_target/src/spec/base/linux_musl.rs b/compiler/rustc_target/src/spec/base/linux_musl.rs
index 6d3124b55..056eeaa81 100644
--- a/compiler/rustc_target/src/spec/base/linux_musl.rs
+++ b/compiler/rustc_target/src/spec/base/linux_musl.rs
@@ -1,11 +1,8 @@
-use crate::spec::{Env, LinkSelfContainedDefault, TargetOptions, base, crt_objects};
+use crate::spec::{Env, TargetOptions, base};
 
 pub(crate) fn opts() -> TargetOptions {
     TargetOptions {
         env: Env::Musl,
-        pre_link_objects_self_contained: crt_objects::pre_musl_self_contained(),
-        post_link_objects_self_contained: crt_objects::post_musl_self_contained(),
-        link_self_contained: LinkSelfContainedDefault::InferredForMusl,
         ..base::linux::opts()
     }
 }
diff --git a/compiler/rustc_target/src/spec/crt_objects.rs b/compiler/rustc_target/src/spec/crt_objects.rs
index 2d84e78f2..607a777eb 100644
--- a/compiler/rustc_target/src/spec/crt_objects.rs
+++ b/compiler/rustc_target/src/spec/crt_objects.rs
@@ -48,7 +48,10 @@ use crate::spec::LinkOutputKind;
 pub type CrtObjects = BTreeMap<LinkOutputKind, Vec<Cow<'static, str>>>;
 
 pub(super) fn new(obj_table: &[(LinkOutputKind, &[&'static str])]) -> CrtObjects {
-    obj_table.iter().map(|(z, k)| (*z, k.iter().map(|b| (*b).into()).collect())).collect()
+    obj_table
+        .iter()
+        .map(|(z, k)| (*z, k.iter().map(|b| (*b).into()).collect()))
+        .collect()
 }
 
 pub(super) fn all(obj: &'static str) -> CrtObjects {
@@ -62,28 +65,6 @@ pub(super) fn all(obj: &'static str) -> CrtObjects {
     ])
 }
 
-pub(super) fn pre_musl_self_contained() -> CrtObjects {
-    new(&[
-        (LinkOutputKind::DynamicNoPicExe, &["crt1.o", "crti.o", "crtbegin.o"]),
-        (LinkOutputKind::DynamicPicExe, &["Scrt1.o", "crti.o", "crtbeginS.o"]),
-        (LinkOutputKind::StaticNoPicExe, &["crt1.o", "crti.o", "crtbegin.o"]),
-        (LinkOutputKind::StaticPicExe, &["rcrt1.o", "crti.o", "crtbeginS.o"]),
-        (LinkOutputKind::DynamicDylib, &["crti.o", "crtbeginS.o"]),
-        (LinkOutputKind::StaticDylib, &["crti.o", "crtbeginS.o"]),
-    ])
-}
-
-pub(super) fn post_musl_self_contained() -> CrtObjects {
-    new(&[
-        (LinkOutputKind::DynamicNoPicExe, &["crtend.o", "crtn.o"]),
-        (LinkOutputKind::DynamicPicExe, &["crtendS.o", "crtn.o"]),
-        (LinkOutputKind::StaticNoPicExe, &["crtend.o", "crtn.o"]),
-        (LinkOutputKind::StaticPicExe, &["crtendS.o", "crtn.o"]),
-        (LinkOutputKind::DynamicDylib, &["crtendS.o", "crtn.o"]),
-        (LinkOutputKind::StaticDylib, &["crtendS.o", "crtn.o"]),
-    ])
-}
-
 pub(super) fn pre_mingw_self_contained() -> CrtObjects {
     new(&[
         (LinkOutputKind::DynamicNoPicExe, &["crt2.o"]),
diff --git a/src/bootstrap/configure.py b/src/bootstrap/configure.py
index 1915986be..7f5721594 100755
--- a/src/bootstrap/configure.py
+++ b/src/bootstrap/configure.py
@@ -180,106 +180,6 @@ v("llvm-config", None, "set path to llvm-config")
 v("llvm-filecheck", None, "set path to LLVM's FileCheck utility")
 v("python", "build.python", "set path to python")
 v("android-ndk", "build.android-ndk", "set path to Android NDK")
-v(
-    "musl-root",
-    "target.x86_64-unknown-linux-musl.musl-root",
-    "MUSL root installation directory (deprecated)",
-)
-v(
-    "musl-root-x86_64",
-    "target.x86_64-unknown-linux-musl.musl-root",
-    "x86_64-unknown-linux-musl install directory",
-)
-v(
-    "musl-root-i586",
-    "target.i586-unknown-linux-musl.musl-root",
-    "i586-unknown-linux-musl install directory",
-)
-v(
-    "musl-root-i686",
-    "target.i686-unknown-linux-musl.musl-root",
-    "i686-unknown-linux-musl install directory",
-)
-v(
-    "musl-root-arm",
-    "target.arm-unknown-linux-musleabi.musl-root",
-    "arm-unknown-linux-musleabi install directory",
-)
-v(
-    "musl-root-armhf",
-    "target.arm-unknown-linux-musleabihf.musl-root",
-    "arm-unknown-linux-musleabihf install directory",
-)
-v(
-    "musl-root-armv5te",
-    "target.armv5te-unknown-linux-musleabi.musl-root",
-    "armv5te-unknown-linux-musleabi install directory",
-)
-v(
-    "musl-root-armv7",
-    "target.armv7-unknown-linux-musleabi.musl-root",
-    "armv7-unknown-linux-musleabi install directory",
-)
-v(
-    "musl-root-armv7hf",
-    "target.armv7-unknown-linux-musleabihf.musl-root",
-    "armv7-unknown-linux-musleabihf install directory",
-)
-v(
-    "musl-root-aarch64",
-    "target.aarch64-unknown-linux-musl.musl-root",
-    "aarch64-unknown-linux-musl install directory",
-)
-v(
-    "musl-root-mips",
-    "target.mips-unknown-linux-musl.musl-root",
-    "mips-unknown-linux-musl install directory",
-)
-v(
-    "musl-root-mipsel",
-    "target.mipsel-unknown-linux-musl.musl-root",
-    "mipsel-unknown-linux-musl install directory",
-)
-v(
-    "musl-root-mips64",
-    "target.mips64-unknown-linux-muslabi64.musl-root",
-    "mips64-unknown-linux-muslabi64 install directory",
-)
-v(
-    "musl-root-mips64el",
-    "target.mips64el-unknown-linux-muslabi64.musl-root",
-    "mips64el-unknown-linux-muslabi64 install directory",
-)
-v(
-    "musl-root-powerpc64",
-    "target.powerpc64-unknown-linux-musl.musl-root",
-    "powerpc64-unknown-linux-musl install directory",
-)
-v(
-    "musl-root-powerpc64le",
-    "target.powerpc64le-unknown-linux-musl.musl-root",
-    "powerpc64le-unknown-linux-musl install directory",
-)
-v(
-    "musl-root-riscv32gc",
-    "target.riscv32gc-unknown-linux-musl.musl-root",
-    "riscv32gc-unknown-linux-musl install directory",
-)
-v(
-    "musl-root-riscv64gc",
-    "target.riscv64gc-unknown-linux-musl.musl-root",
-    "riscv64gc-unknown-linux-musl install directory",
-)
-v(
-    "musl-root-loongarch64",
-    "target.loongarch64-unknown-linux-musl.musl-root",
-    "loongarch64-unknown-linux-musl install directory",
-)
-v(
-    "musl-root-wali-wasm32",
-    "target.wasm32-wali-linux-musl.musl-root",
-    "wasm32-wali-linux-musl install directory",
-)
 v(
     "qemu-armhf-rootfs",
     "target.arm-unknown-linux-gnueabihf.qemu-rootfs",
diff --git a/src/bootstrap/src/core/build_steps/compile.rs b/src/bootstrap/src/core/build_steps/compile.rs
index 02940a802..d105f128b 100644
--- a/src/bootstrap/src/core/build_steps/compile.rs
+++ b/src/bootstrap/src/core/build_steps/compile.rs
@@ -375,54 +375,7 @@ fn copy_self_contained_objects(
     t!(fs::create_dir_all(&libdir_self_contained));
     let mut target_deps = vec![];
 
-    // Copies the libc and CRT objects.
-    //
-    // rustc historically provides a more self-contained installation for musl targets
-    // not requiring the presence of a native musl toolchain. For example, it can fall back
-    // to using gcc from a glibc-targeting toolchain for linking.
-    // To do that we have to distribute musl startup objects as a part of Rust toolchain
-    // and link with them manually in the self-contained mode.
-    if target.needs_crt_begin_end() {
-        let srcdir = builder.musl_libdir(target).unwrap_or_else(|| {
-            panic!("Target {:?} does not have a \"musl-libdir\" key", target.triple)
-        });
-        if !target.starts_with("wasm32") {
-            for &obj in &["libc.a", "crt1.o", "Scrt1.o", "rcrt1.o", "crti.o", "crtn.o"] {
-                copy_and_stamp(
-                    builder,
-                    &libdir_self_contained,
-                    &srcdir,
-                    obj,
-                    &mut target_deps,
-                    DependencyType::TargetSelfContained,
-                );
-            }
-            let crt_path = builder.ensure(llvm::CrtBeginEnd { target });
-            for &obj in &["crtbegin.o", "crtbeginS.o", "crtend.o", "crtendS.o"] {
-                let src = crt_path.join(obj);
-                let target = libdir_self_contained.join(obj);
-                builder.copy_link(&src, &target, FileType::NativeLibrary);
-                target_deps.push((target, DependencyType::TargetSelfContained));
-            }
-        } else {
-            // For wasm32 targets, we need to copy the libc.a and crt1-command.o files from the
-            // musl-libdir, but we don't need the other files.
-            for &obj in &["libc.a", "crt1-command.o"] {
-                copy_and_stamp(
-                    builder,
-                    &libdir_self_contained,
-                    &srcdir,
-                    obj,
-                    &mut target_deps,
-                    DependencyType::TargetSelfContained,
-                );
-            }
-        }
-        if !target.starts_with("s390x") {
-            let libunwind_path = copy_llvm_libunwind(builder, target, &libdir_self_contained);
-            target_deps.push((libunwind_path, DependencyType::TargetSelfContained));
-        }
-    } else if target.contains("-wasi") {
+    if target.contains("-wasi") {
         let srcdir = builder.wasi_libdir(target).unwrap_or_else(|| {
             panic!(
                 "Target {:?} does not have a \"wasi-root\" key in bootstrap.toml \
@@ -656,15 +609,6 @@ pub fn std_cargo(
             .arg("--manifest-path")
             .arg(builder.src.join("library/sysroot/Cargo.toml"));
 
-        // Help the libc crate compile by assisting it in finding various
-        // sysroot native libraries.
-        if target.contains("musl")
-            && let Some(p) = builder.musl_libdir(target)
-        {
-            let root = format!("native={}", p.to_str().unwrap());
-            cargo.rustflag("-L").rustflag(&root);
-        }
-
         if target.contains("-wasi")
             && let Some(dir) = builder.wasi_libdir(target)
         {
diff --git a/src/bootstrap/src/core/build_steps/llvm.rs b/src/bootstrap/src/core/build_steps/llvm.rs
index db2a76c4a..d08d8d8d0 100644
--- a/src/bootstrap/src/core/build_steps/llvm.rs
+++ b/src/bootstrap/src/core/build_steps/llvm.rs
@@ -1320,10 +1320,7 @@ impl Step for CrtBeginEnd {
         run.path("src/llvm-project/compiler-rt/lib/crt")
     }
 
-    fn make_run(run: RunConfig<'_>) {
-        if run.target.needs_crt_begin_end() {
-            run.builder.ensure(CrtBeginEnd { target: run.target });
-        }
+    fn make_run(_run: RunConfig<'_>) {
     }
 
     /// Build crtbegin.o/crtend.o for musl target.
diff --git a/src/bootstrap/src/core/config/config.rs b/src/bootstrap/src/core/config/config.rs
index 2f493658e..0236d8170 100644
--- a/src/bootstrap/src/core/config/config.rs
+++ b/src/bootstrap/src/core/config/config.rs
@@ -262,8 +262,6 @@ pub struct Config {
     pub print_step_timings: bool,
     pub print_step_rusage: bool,
 
-    // Fallback musl-root for all targets
-    pub musl_root: Option<PathBuf>,
     pub prefix: Option<PathBuf>,
     pub sysconfdir: Option<PathBuf>,
     pub datadir: Option<PathBuf>,
@@ -539,7 +537,6 @@ impl Config {
             randomize_layout: rust_randomize_layout,
             default_linker: rust_default_linker,
             channel: rust_channel,
-            musl_root: rust_musl_root,
             rpath: rust_rpath,
             verbose_tests: rust_verbose_tests,
             optimize_tests: rust_optimize_tests,
@@ -868,8 +865,6 @@ impl Config {
                     rpath: target_rpath,
                     rustflags: target_rustflags,
                     crt_static: target_crt_static,
-                    musl_root: target_musl_root,
-                    musl_libdir: target_musl_libdir,
                     wasi_root: target_wasi_root,
                     qemu_rootfs: target_qemu_rootfs,
                     no_std: target_no_std,
@@ -942,8 +937,6 @@ impl Config {
                 target.crt_static = target_crt_static;
                 target.default_linker = target_default_linker;
                 target.default_linker_linux_override = default_linker_linux_override;
-                target.musl_root = target_musl_root.map(PathBuf::from);
-                target.musl_libdir = target_musl_libdir.map(PathBuf::from);
                 target.wasi_root = target_wasi_root.map(PathBuf::from);
                 target.qemu_rootfs = target_qemu_rootfs.map(PathBuf::from);
                 target.runner = target_runner;
@@ -1393,7 +1386,6 @@ impl Config {
             low_priority: build_low_priority.unwrap_or(false),
             mandir: install_mandir.map(PathBuf::from),
             miri_info,
-            musl_root: rust_musl_root.map(PathBuf::from),
             ninja_in_file: llvm_ninja.unwrap_or(true),
             nodejs: build_nodejs.map(PathBuf::from),
             omit_git_hash,
diff --git a/src/bootstrap/src/core/config/target_selection.rs b/src/bootstrap/src/core/config/target_selection.rs
index 47f6d6f38..a519b6655 100644
--- a/src/bootstrap/src/core/config/target_selection.rs
+++ b/src/bootstrap/src/core/config/target_selection.rs
@@ -96,10 +96,6 @@ impl TargetSelection {
         env::var("OSTYPE").is_ok_and(|v| v.to_lowercase().contains("cygwin"))
     }
 
-    pub fn needs_crt_begin_end(&self) -> bool {
-        self.contains("musl") && !self.contains("unikraft")
-    }
-
     /// Path to the file defining the custom target, if any.
     pub fn filepath(&self) -> Option<&Path> {
         self.file.as_ref().map(Path::new)
diff --git a/src/bootstrap/src/core/config/toml/rust.rs b/src/bootstrap/src/core/config/toml/rust.rs
index 48042ba56..083970f1b 100644
--- a/src/bootstrap/src/core/config/toml/rust.rs
+++ b/src/bootstrap/src/core/config/toml/rust.rs
@@ -31,7 +31,6 @@ define_config! {
         incremental: Option<bool> = "incremental",
         default_linker: Option<String> = "default-linker",
         channel: Option<String> = "channel",
-        musl_root: Option<String> = "musl-root",
         rpath: Option<bool> = "rpath",
         rustflags: Option<Vec<String>> = "rustflags",
         strip: Option<bool> = "strip",
@@ -346,7 +345,6 @@ pub fn check_incompatible_options_for_ci_rustc(
         debuginfo_level_tools: _,
         debuginfo_level_tests: _,
         backtrace: _,
-        musl_root: _,
         verbose_tests: _,
         optimize_tests: _,
         codegen_tests: _,
diff --git a/src/bootstrap/src/core/config/toml/target.rs b/src/bootstrap/src/core/config/toml/target.rs
index 847b75e69..5ad42917a 100644
--- a/src/bootstrap/src/core/config/toml/target.rs
+++ b/src/bootstrap/src/core/config/toml/target.rs
@@ -39,8 +39,6 @@ define_config! {
         rpath: Option<bool> = "rpath",
         rustflags: Option<Vec<String>> = "rustflags",
         crt_static: Option<bool> = "crt-static",
-        musl_root: Option<String> = "musl-root",
-        musl_libdir: Option<String> = "musl-libdir",
         wasi_root: Option<String> = "wasi-root",
         qemu_rootfs: Option<String> = "qemu-rootfs",
         no_std: Option<bool> = "no-std",
@@ -73,8 +71,6 @@ pub struct Target {
     pub rpath: Option<bool>,
     pub rustflags: Vec<String>,
     pub crt_static: Option<bool>,
-    pub musl_root: Option<PathBuf>,
-    pub musl_libdir: Option<PathBuf>,
     pub wasi_root: Option<PathBuf>,
     pub qemu_rootfs: Option<PathBuf>,
     pub runner: Option<String>,
diff --git a/src/bootstrap/src/core/sanity.rs b/src/bootstrap/src/core/sanity.rs
index 50d815401..3818e081b 100644
--- a/src/bootstrap/src/core/sanity.rs
+++ b/src/bootstrap/src/core/sanity.rs
@@ -12,7 +12,7 @@
 use std::collections::{HashMap, HashSet};
 use std::ffi::{OsStr, OsString};
 use std::path::PathBuf;
-use std::{env, fs};
+use std::env;
 
 #[cfg(not(test))]
 use crate::builder::Builder;
@@ -367,22 +367,6 @@ than building it.
             continue;
         }
 
-        // Make sure musl-root is valid.
-        if target.contains("musl") && !target.contains("unikraft") {
-            match build.musl_libdir(*target) {
-                Some(libdir) => {
-                    if fs::metadata(libdir.join("libc.a")).is_err() {
-                        panic!("couldn't find libc.a in musl libdir: {}", libdir.display());
-                    }
-                }
-                None => panic!(
-                    "when targeting MUSL either the rust.musl-root \
-                            option or the target.$TARGET.musl-root option must \
-                            be specified in bootstrap.toml"
-                ),
-            }
-        }
-
         if need_cmake && target.is_msvc() {
             // There are three builds of cmake on windows: MSVC, MinGW, and
             // Cygwin. The Cygwin build does not have generators for Visual
diff --git a/src/bootstrap/src/lib.rs b/src/bootstrap/src/lib.rs
index a31eb0c1c..a78506c39 100644
--- a/src/bootstrap/src/lib.rs
+++ b/src/bootstrap/src/lib.rs
@@ -1394,35 +1394,6 @@ impl Build {
         }
     }
 
-    /// Returns the "musl root" for this `target`, if defined.
-    ///
-    /// If this is a native target (host is also musl) and no musl-root is given,
-    /// it falls back to the system toolchain in /usr.
-    fn musl_root(&self, target: TargetSelection) -> Option<&Path> {
-        let configured_root = self
-            .config
-            .target_config
-            .get(&target)
-            .and_then(|t| t.musl_root.as_ref())
-            .or(self.config.musl_root.as_ref())
-            .map(|p| &**p);
-
-        if self.config.is_host_target(target) && configured_root.is_none() {
-            Some(Path::new("/usr"))
-        } else {
-            configured_root
-        }
-    }
-
-    /// Returns the "musl libdir" for this `target`.
-    fn musl_libdir(&self, target: TargetSelection) -> Option<PathBuf> {
-        self.config
-            .target_config
-            .get(&target)
-            .and_then(|t| t.musl_libdir.clone())
-            .or_else(|| self.musl_root(target).map(|root| root.join("lib")))
-    }
-
     /// Returns the `lib` directory for the WASI target specified, if
     /// configured.
     ///
diff --git a/src/bootstrap/src/utils/cc_detect.rs b/src/bootstrap/src/utils/cc_detect.rs
index 0662ae304..f6373f44e 100644
--- a/src/bootstrap/src/utils/cc_detect.rs
+++ b/src/bootstrap/src/utils/cc_detect.rs
@@ -195,30 +195,6 @@ fn default_compiler(
             }
         }
 
-        "mips-unknown-linux-musl" if compiler == Language::C => {
-            if cfg.get_compiler().path().to_str() == Some("gcc") {
-                Some(PathBuf::from("mips-linux-musl-gcc"))
-            } else {
-                None
-            }
-        }
-        "mipsel-unknown-linux-musl" if compiler == Language::C => {
-            if cfg.get_compiler().path().to_str() == Some("gcc") {
-                Some(PathBuf::from("mipsel-linux-musl-gcc"))
-            } else {
-                None
-            }
-        }
-
-        t if t.contains("musl") && compiler == Language::C => {
-            if let Some(root) = build.musl_root(target) {
-                let guess = root.join("bin/musl-gcc");
-                if guess.exists() { Some(guess) } else { None }
-            } else {
-                None
-            }
-        }
-
         t if t.contains("-wasi") => {
             let root = if let Some(path) = build.wasi_sdk_path.as_ref() {
                 path
