From d7d7fc24d1744a3db70a9fb010c31daa14a46df7 Mon Sep 17 00:00:00 2001
From: Daniel Kolesa <daniel@octaforge.org>
Date: Mon, 20 Dec 2021 00:55:18 +0100
Subject: [PATCH 04/15] fix shebang on install-template.sh

diff --git a/src/tools/rust-installer/install-template.sh b/src/tools/rust-installer/install-template.sh
index b477c3eac..7bf7ed7fb 100644
--- a/src/tools/rust-installer/install-template.sh
+++ b/src/tools/rust-installer/install-template.sh
@@ -1,4 +1,4 @@
-#!/usr/bin/env bash
+#!/bin/sh
 
 # No undefined variables
 set -u
