From 361bc09082dbb15c3586df12f664a9ffdef869a6 Mon Sep 17 00:00:00 2001
From: Duncan Overbruck <mail@duncano.de>
Date: Mon, 3 Nov 2025 18:26:14 +0100
Subject: [PATCH 07/15] void: fix linkage against host/target LLVM in cross

---
 compiler/rustc_llvm/build.rs | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/compiler/rustc_llvm/build.rs b/compiler/rustc_llvm/build.rs
index d5c43c4fa..3093fcb48 100644
--- a/compiler/rustc_llvm/build.rs
+++ b/compiler/rustc_llvm/build.rs
@@ -350,9 +350,17 @@ fn main() {
     for lib in output(&mut cmd).split_whitespace() {
         if is_crossed {
             if let Some(stripped) = lib.strip_prefix("-LIBPATH:") {
-                println!("cargo:rustc-link-search=native={}", stripped.replace(&host, &target));
+                println!(
+                    "cargo:rustc-link-search=native={}{}",
+                    env::var("XBPS_CROSS_BASE").unwrap(),
+                    stripped.replace(&host, &target)
+                );
             } else if let Some(stripped) = lib.strip_prefix("-L") {
-                println!("cargo:rustc-link-search=native={}", stripped.replace(&host, &target));
+                println!(
+                    "cargo:rustc-link-search=native={}{}",
+                    env::var("XBPS_CROSS_BASE").unwrap(),
+                    stripped.replace(&host, &target)
+                );
             }
         } else if let Some(stripped) = lib.strip_prefix("-LIBPATH:") {
             println!("cargo:rustc-link-search=native={stripped}");
-- 
2.51.2

