From d60d7331d50b592252f03b34601bfc14a512df35 Mon Sep 17 00:00:00 2001
From: q66 <q66@chimera-linux.org>
Date: Thu, 20 Feb 2025 20:44:45 +0100
Subject: [PATCH 08/15] do not export LD_LIBRARY_PATH

---
 src/bootstrap/bootstrap.py | 11 -----------
 1 file changed, 11 deletions(-)

diff --git a/src/bootstrap/bootstrap.py b/src/bootstrap/bootstrap.py
index 4dd465edb..30c107209 100644
--- a/src/bootstrap/bootstrap.py
+++ b/src/bootstrap/bootstrap.py
@@ -1045,17 +1045,6 @@ class RustBuild(object):
             env["CARGO_INCREMENTAL"] = "0"
         env["CARGO_TARGET_DIR"] = build_dir
         env["RUSTC"] = self.rustc()
-        env["LD_LIBRARY_PATH"] = (
-            os.path.join(self.bin_root(), "lib") + (os.pathsep + env["LD_LIBRARY_PATH"])
-            if "LD_LIBRARY_PATH" in env
-            else ""
-        )
-        env["DYLD_LIBRARY_PATH"] = (
-            os.path.join(self.bin_root(), "lib")
-            + (os.pathsep + env["DYLD_LIBRARY_PATH"])
-            if "DYLD_LIBRARY_PATH" in env
-            else ""
-        )
         env["LIBRARY_PATH"] = (
             os.path.join(self.bin_root(), "lib") + (os.pathsep + env["LIBRARY_PATH"])
             if "LIBRARY_PATH" in env
-- 
2.51.2

