From fb66003fb2c75b7d6c782705549b9fe2247883aa Mon Sep 17 00:00:00 2001
From: q66 <q66@chimera-linux.org>
Date: Thu, 20 Feb 2025 20:44:45 +0100
Subject: [PATCH 08/15] do not export LD_LIBRARY_PATH

---
 src/bootstrap/bootstrap.py | 11 -----------
 1 file changed, 11 deletions(-)

diff --git a/src/bootstrap/bootstrap.py b/src/bootstrap/bootstrap.py
index d8c6be782..9b625bb31 100644
--- a/src/bootstrap/bootstrap.py
+++ b/src/bootstrap/bootstrap.py
@@ -1024,17 +1024,6 @@ class RustBuild(object):
             del env["CARGO_BUILD_TARGET"]
         env["CARGO_TARGET_DIR"] = build_dir
         env["RUSTC"] = self.rustc()
-        env["LD_LIBRARY_PATH"] = (
-            os.path.join(self.bin_root(), "lib") + (os.pathsep + env["LD_LIBRARY_PATH"])
-            if "LD_LIBRARY_PATH" in env
-            else ""
-        )
-        env["DYLD_LIBRARY_PATH"] = (
-            os.path.join(self.bin_root(), "lib")
-            + (os.pathsep + env["DYLD_LIBRARY_PATH"])
-            if "DYLD_LIBRARY_PATH" in env
-            else ""
-        )
         env["LIBRARY_PATH"] = (
             os.path.join(self.bin_root(), "lib") + (os.pathsep + env["LIBRARY_PATH"])
             if "LIBRARY_PATH" in env
-- 
2.51.2

