diff --git a/src/bootstrap/builder.rs b/src/bootstrap/builder.rs
index 8b144f146..acfa7cf04 100644
--- a/src/bootstrap/builder.rs
+++ b/src/bootstrap/builder.rs
@@ -1856,7 +1856,7 @@ impl<'a> Builder<'a> {
         // efficient initial-exec TLS model. This doesn't work with `dlopen`,
         // so we can't use it by default in general, but we can use it for tools
         // and our own internal libraries.
-        if !mode.must_support_dlopen() && !target.triple.starts_with("powerpc-") {
+        if !mode.must_support_dlopen() && !target.triple.starts_with("powerpc-") && !target.triple.starts_with("powerpcle-") {
             cargo.env("RUSTC_TLS_MODEL_INITIAL_EXEC", "1");
         }
 
