From 0b057b456a5b24ce5cf898e47eddb7cbc7448fe2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=C3=89rico=20Nogueira?= <erico.erc@gmail.com>
Date: Fri, 16 Apr 2021 03:34:16 -0300
Subject: [PATCH 14/15] copy new version of need-ssp_nonshared.patch from
 alpine

diff --git a/library/std/src/sys/pal/unix/mod.rs b/library/std/src/sys/pal/unix/mod.rs
index c0b56d8d2..b92ecb5e0 100644
--- a/library/std/src/sys/pal/unix/mod.rs
+++ b/library/std/src/sys/pal/unix/mod.rs
@@ -374,6 +374,10 @@ cfg_if::cfg_if! {
         #[link(name = "dl", cfg(not(target_feature = "crt-static")))]
         #[link(name = "log", cfg(not(target_feature = "crt-static")))]
         unsafe extern "C" {}
+    } else if #[cfg(all(target_os = "linux", target_env = "musl"))] {
+        #[link(name = "ssp_nonshared")]
+        #[link(name = "execinfo")]
+        unsafe extern "C" {}
     } else if #[cfg(target_os = "freebsd")] {
         #[link(name = "execinfo")]
         #[link(name = "pthread")]
