diff --git a/src/bootstrap/src/lib.rs b/src/bootstrap/src/lib.rs
index 82b640f54..83fe7660c 100644
--- a/src/bootstrap/src/lib.rs
+++ b/src/bootstrap/src/lib.rs
@@ -484,7 +484,7 @@ impl Build {
     pub fn require_submodule(&self, submodule: &str, err_hint: Option<&str>) {
         // When testing bootstrap itself, it is much faster to ignore
         // submodules. Almost all Steps work fine without their submodules.
-        if cfg!(test) && !self.config.submodules() {
+        if cfg!(test) || !self.config.submodules() {
             return;
         }
         self.config.update_submodule(submodule);
