--- rustc-1.16.0-src/src/tools/rustbook/Cargo.toml.orig
+++ rustc-1.16.0-src/src/tools/rustbook/Cargo.toml
@@ -2,7 +2,6 @@
 authors = ["The Rust Project Developers"]
 name = "rustbook"
 version = "0.0.0"
-build = false
 
 [[bin]]
 name = "rustbook"
--- rustc-1.16.0-src/src/librustc_plugin/Cargo.toml.orig
+++ rustc-1.16.0-src/src/librustc_plugin/Cargo.toml
@@ -2,7 +2,6 @@
 authors = ["The Rust Project Developers"]
 name = "rustc_plugin"
 version = "0.0.0"
-build = false
 
 [lib]
 name = "rustc_plugin"
