commit 486217d7a2779d35c1177c7b1b7e96c1e5d30441 (HEAD, refs/remotes/origin/python, refs/heads/python)
Author: Daniel Micay <danielmicay@gmail.com>
Date:   Thu Jan 3 02:53:05 2013 -0500

    add missing $(CFG_PYTHON) prefixes

diff --git a/mk/docs.mk b/mk/docs.mk
index 2a7b3ec..9f64712 100644
--- mk/docs.mk
+++ mk/docs.mk
@@ -152,7 +152,7 @@ else
 
 doc/rust.g: rust.md $(S)src/etc/extract_grammar.py
 	@$(call E, extract_grammar: $@)
-	$(Q)$(S)src/etc/extract_grammar.py $< >$@
+	$(Q)$(CFG_PYTHON) $(S)src/etc/extract_grammar.py $< >$@
 
 verify-grammar: doc/rust.g
 	@$(call E, LLnextgen: $<)
diff --git a/mk/snap.mk b/mk/snap.mk
index 382f3cb..21bdc3b 100644
--- mk/snap.mk
+++ mk/snap.mk
@@ -14,10 +14,10 @@ define DEF_SNAP_FOR_STAGE_H
 
 ifdef CFG_INSTALL_SNAP
 snap-stage$(1)-H-$(2): $$(HSREQ$(1)_H_$(2))
-	$(S)src/etc/make-snapshot.py stage$(1) $(2) install
+	$(CFG_PYTHON) $(S)src/etc/make-snapshot.py stage$(1) $(2) install
 else
 snap-stage$(1)-H-$(2): $$(HSREQ$(1)_H_$(2))
-	$(S)src/etc/make-snapshot.py stage$(1) $(2)
+	$(CFG_PYTHON) $(S)src/etc/make-snapshot.py stage$(1) $(2)
 endif
 
 endef
@@ -30,4 +30,4 @@ snap-stage1: snap-stage1-H-$(CFG_HOST_TRIPLE)
 
 snap-stage2: snap-stage2-H-$(CFG_HOST_TRIPLE)
 
-snap-stage3: snap-stage3-H-$(CFG_HOST_TRIPLE)
\ No newline at end of file
+snap-stage3: snap-stage3-H-$(CFG_HOST_TRIPLE)
diff --git a/mk/stage0.mk b/mk/stage0.mk
index b64f5df..e1a83c1 100644
--- mk/stage0.mk
+++ mk/stage0.mk
@@ -11,7 +11,7 @@ $(HBIN0_H_$(CFG_HOST_TRIPLE))/rustc$(X):		\
 ifdef CFG_ENABLE_LOCAL_RUST
 	$(Q)$(S)src/etc/local_stage0.sh $(CFG_HOST_TRIPLE) $(CFG_LOCAL_RUST_ROOT)
 else 
-	$(Q)$(S)src/etc/get-snapshot.py $(CFG_HOST_TRIPLE) $(SNAPSHOT_FILE)
+	$(Q)$(CFG_PYTHON) $(S)src/etc/get-snapshot.py $(CFG_HOST_TRIPLE) $(SNAPSHOT_FILE)
 ifdef CFG_ENABLE_PAX_FLAGS
 	@$(call E, apply PaX flags: $@)
 	@"$(CFG_PAXCTL)" -cm "$@"
diff --git a/mk/tests.mk b/mk/tests.mk
index 3f08549..454d540 100644
--- mk/tests.mk
+++ mk/tests.mk
@@ -701,7 +701,7 @@ tmp/$(FT).rc tmp/$(FT_DRIVER).rs: \
 		$(RPASS_TESTS) \
 		$(S)src/etc/combine-tests.py
 	@$(call E, check: building combined stage2 test runner)
-	$(Q)$(S)src/etc/combine-tests.py
+	$(Q)$(CFG_PYTHON) $(S)src/etc/combine-tests.py
 
 define DEF_CHECK_FAST_FOR_T_H
 # $(1) unused
