From: Jakub Jirutka <jakub@jirutka.cz>
Date: Thu, 04 Aug 2016 17:53:00 +0200
Subject: [PATCH] Do not require FileCheck

--- a/src/bootstrap/build/sanity.rs
+++ b/src/bootstrap/build/sanity.rs
@@ -76,12 +76,6 @@
         need_cmd(build.cxx(host).as_ref());
     }
 
-    // Externally configured LLVM requires FileCheck to exist
-    let filecheck = build.llvm_filecheck(&build.config.build);
-    if !filecheck.starts_with(&build.out) && !filecheck.exists() {
-        panic!("filecheck executable {:?} does not exist", filecheck);
-    }
-
     for target in build.config.target.iter() {
         // Either can't build or don't want to run jemalloc on these targets
         if target.contains("rumprun") ||
