--- rustc-1.47.0-src/library/std/build.rs.orig
+++ rustc-1.47.0-src/library/std/build.rs
@@ -8,6 +8,8 @@
             println!("cargo:rustc-link-lib=dl");
             println!("cargo:rustc-link-lib=log");
             println!("cargo:rustc-link-lib=gcc");
+        } else if target.contains("musl") {
+            println!("cargo:rustc-link-lib=ssp_nonshared");
         }
     } else if target.contains("freebsd") {
         println!("cargo:rustc-link-lib=execinfo");
--- rustc-1.44.0-src/library/unwind/build.rs.orig
+++ rustc-1.44.0-src/library/unwind/build.rs
@@ -12,6 +12,9 @@ fn main() {
     } else if target.contains("linux") {
         if !target.contains("android") {
             println!("cargo:rustc-link-lib=gcc_s");
+            if target.contains("musl") {
+                println!("cargo:rustc-link-lib=ssp_nonshared");
+            }
         }
     } else if target.contains("freebsd") {
         println!("cargo:rustc-link-lib=gcc_s");
