--- a/library/std/src/sys/unix/mod.rs.orig	2021-02-11 18:34:14.479832268 +0100
+++ b/library/std/src/sys/unix/mod.rs	2021-02-11 18:38:28.078987749 +0100
@@ -243,6 +243,9 @@
         #[link(name = "log")]
         #[link(name = "gcc")]
         extern "C" {}
+    } else if #[cfg(all(target_os = "linux", target_env = "musl"))] {
+	#[link(name = "ssp_nonshared")]
+	extern "C" {}
     } else if #[cfg(target_os = "freebsd")] {
         #[link(name = "execinfo")]
         #[link(name = "pthread")]

