--- a/library/std/src/sys/unix/mod.rs
+++ b/library/std/src/sys/unix/mod.rs
@@ -212,6 +212,10 @@ cfg_if::cfg_if! {
         #[link(name = "log")]
         #[link(name = "gcc")]
         extern "C" {}
+    } else if #[cfg(all(target_os = "linux", target_env = "musl"))] {
+	#[link(name = "ssp_nonshared")]
+        #[link(name = "execinfo")]
+	extern "C" {}
     } else if #[cfg(target_os = "freebsd")] {
         #[link(name = "execinfo")]
         #[link(name = "pthread")]
