From dc1de5be2b7908adb493b36f5079064083ed9507 Mon Sep 17 00:00:00 2001
From: q66 <daniel@octaforge.org>
Date: Sat, 3 Jul 2021 19:00:46 +0200
Subject: [PATCH 1/2] cc-1.0.60: fix powerpcle being matched as as msvc

---
 vendor/cc-1.0.60/src/lib.rs | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/vendor/cc-1.0.60/src/lib.rs b/vendor/cc-1.0.60/src/lib.rs
index aff6be86..f428ecbb 100644
--- a/vendor/cc-1.0.60/src/lib.rs
+++ b/vendor/cc-1.0.60/src/lib.rs
@@ -2556,11 +2556,7 @@ impl Tool {
         let family = if let Some(fname) = path.file_name().and_then(|p| p.to_str()) {
             if fname.contains("clang-cl") {
                 ToolFamily::Msvc { clang_cl: true }
-            } else if fname.contains("cl")
-                && !fname.contains("cloudabi")
-                && !fname.contains("uclibc")
-                && !fname.contains("clang")
-            {
+            } else if fname.ends_with("cl") || fname == "cl.exe" {
                 ToolFamily::Msvc { clang_cl: false }
             } else if fname.contains("clang") {
                 match clang_driver {
-- 
2.31.1

