commit e6f3e66154138ce3e31e803e74b8c71787c70acc
Author: Gonzalo Tornaría <tornaria@cmat.edu.uy>
Date:   Wed Jul 19 20:47:52 2023 -0300

    cython3 support using legacy directives

diff --git a/setup.py b/setup.py
index 13d543b..55dcd34 100755
--- a/setup.py
+++ b/setup.py
@@ -38,7 +38,11 @@ class build_ext(_build_ext):
         self.extensions[:] = cythonize(
             self.extensions,
             include_path=sys.path,
-            compiler_directives={'embedsignature': True})
+            compiler_directives={
+                'embedsignature': True,
+                'legacy_implicit_noexcept': True,
+                'c_api_binop_methods': True,
+                })
 
         _build_ext.run(self)
 
