Index: qt5-fsarchiver-1.8.6pl1/src/oper_restore.c
===================================================================
--- qt5-fsarchiver-1.8.6pl1.orig/src/oper_restore.c
+++ qt5-fsarchiver-1.8.6pl1/src/oper_restore.c
@@ -24,7 +24,7 @@
 #include <assert.h>
 #include <string.h>
 #include <stdlib.h>
-#include <attr/xattr.h>
+#include <sys/xattr.h>
 #include <sys/time.h>
 #include <sys/stat.h>
 #include <gcrypt.h>
