--- a/qt5-fsarchiver.pro	2018-06-22 00:01:06.101404975 +0200
+++ b/qt5-fsarchiver.pro	2018-06-22 00:01:45.933869386 +0200
@@ -9,7 +9,7 @@
 QT += widgets gui core
 
 DEFINES +=  HAVE_CONFIG_H _REENTRANT _FILE_OFFSET_BITS=64 _LARGEFILE64_SOURCE _GNU_SOURCE HAVE_QT5
-INCLUDEPATH += . src /usr/include/ext2fs /usr/include/et /usr/include/e2p /usr/include/blkid /usr/include/uuid 
+INCLUDEPATH += . src $$(XBPS_CROSS_BASE)/usr/include/ext2fs $$(XBPS_CROSS_BASE)/usr/include/et $$(XBPS_CROSS_BASE)/usr/include/e2p $$(XBPS_CROSS_BASE)/usr/include/blkid $$(XBPS_CROSS_BASE)/usr/include/uuid 
 LIBS += -L/usr/lib -lQt5Gui -lQt5Core -lpthread -lrt -llzma -lext2fs -lcom_err -le2p -lblkid -luuid -lz -llz4 -lbz2 -llzo2 -lgcrypt -lgpg-error -lzstd
 OBJECTS_DIR = build
 MOC_DIR = build
