--- qt5-fsarchiver.pro.orig	2018-06-21 23:15:21.665117797 +0200
+++ qt5-fsarchiver.pro	2018-06-21 23:15:36.159276907 +0200
@@ -147,7 +147,7 @@
            src/writebuf.c
 RESOURCES += src/icon.qrc
 # install
- target.path = /usr/sbin
+ target.path = /usr/bin
  icon.files = src/images/harddrive1.png
  icon.path = /usr/share/app-install/icons
  autostart.files = starter/gnome-qt5-fsarchiver.desktop
@@ -158,7 +158,7 @@
  autostart2.path = /usr/share/applications
  doc.files = doc
  doc.path = /usr/share/doc/qt5-fsarchiver/doc
- smbfind.files = src/sbin
+ smbfind.files = src/bin
  smbfind.path = /usr
 TRANSLATIONS += translations/qt5-fsarchiver_ar.ts \
                 translations/qt5-fsarchiver_ca.ts \
