Index: qt5-fsarchiver-1.8.6pl1/qt-fsarchiver.pro
===================================================================
--- qt5-fsarchiver-1.8.6pl1.orig/qt-fsarchiver.pro
+++ qt5-fsarchiver-1.8.6pl1/qt-fsarchiver.pro
@@ -153,9 +153,9 @@ isEmpty(DOC_DIR) {
   DOC_DIR = /usr/share/doc/qt-fsarchiver
 }
 # install
- target.path = /usr/sbin
+ target.path = /usr/bin
  target1.files = src/sbin/qt-fsarchiver.sh
- target1.path = /usr/sbin
+ target1.path = /usr/bin
  icon.files = src/images/qt-fsarchiver.png
  icon.path = /usr/share/pixmaps
  autostart.files = starter/qt-fsarchiver.desktop
Index: qt5-fsarchiver-1.8.6pl1/src/mainWindow.cpp
===================================================================
--- qt5-fsarchiver-1.8.6pl1.orig/src/mainWindow.cpp
+++ qt5-fsarchiver-1.8.6pl1/src/mainWindow.cpp
@@ -315,10 +315,10 @@ MWindow::MWindow()
       }
    j = 0; 
    // qt-fsarchiver-terminal entfernen
-   QFile file1("/usr/sbin/qt-fsarchiver-terminal");
+   QFile file1("/usr/bin/qt-fsarchiver-terminal");
    if (file1.exists()) 
       {
-      befehl = "rm /usr/sbin/qt-fsarchiver-terminal";
+      befehl = "rm /usr/bin/qt-fsarchiver-terminal";
       if(system (befehl.toLatin1().data()))
          befehl = "";
       }
Index: qt5-fsarchiver-1.8.6pl1/src/sbin/qt-fsarchiver.sh
===================================================================
--- qt5-fsarchiver-1.8.6pl1.orig/src/sbin/qt-fsarchiver.sh
+++ qt5-fsarchiver-1.8.6pl1/src/sbin/qt-fsarchiver.sh
@@ -1,2 +1,2 @@
 #!/bin/bash
-pkexec env DISPLAY=$DISPLAY XAUTHORITY=$XAUTHORITY '/usr/sbin/qt-fsarchiver' 
+pkexec env DISPLAY=$DISPLAY XAUTHORITY=$XAUTHORITY /usr/bin/qt-fsarchiver
Index: qt5-fsarchiver-1.8.6pl1/starter/qt-fsarchiver.desktop
===================================================================
--- qt5-fsarchiver-1.8.6pl1.orig/starter/qt-fsarchiver.desktop
+++ qt5-fsarchiver-1.8.6pl1/starter/qt-fsarchiver.desktop
@@ -2,7 +2,7 @@
 Name=qt-fsarchiver
 Name[de_DE]=qt-fsarchiver
 Comment=Qt GUI for fsarchiver
-Exec=/usr/sbin/qt-fsarchiver.sh
+Exec=/usr/bin/qt-fsarchiver.sh
 Icon=qt-fsarchiver
 Terminal=false
 Type=Application
