--- src/common.c.orig	2018-06-21 23:29:12.238999543 +0200
+++ src/common.c	2018-06-21 23:30:28.347795214 +0200
@@ -34,7 +34,7 @@
 #include <time.h>
 #include <limits.h>
 
-#ifdef HAVE_EXECINFO_H
+#if defined(HAVE_EXECINFO_H) && defined(__GLIBC__)
 #include <execinfo.h>
 #endif
 
@@ -567,7 +567,7 @@
 
 int format_stacktrace(char *buffer, int bufsize)
 {
-#ifdef HAVE_EXECINFO_H
+#if defined(HAVE_EXECINFO_H) && defined(__GLIBC__)
     const int stack_depth=20;
     void *temp[stack_depth];
     char **strings;
