Index: qt5-fsarchiver-1.8.6pl1/src/common.c
===================================================================
--- qt5-fsarchiver-1.8.6pl1.orig/src/common.c
+++ qt5-fsarchiver-1.8.6pl1/src/common.c
@@ -34,7 +34,7 @@
 #include <time.h>
 #include <limits.h>
 
-#ifdef HAVE_EXECINFO_H
+#if defined(HAVE_EXECINFO_H) && defined(__GLIBC__)
 #include <execinfo.h>
 #endif
 
@@ -567,7 +567,7 @@ u64 stats_errcount(cstats stats)
 
 int format_stacktrace(char *buffer, int bufsize)
 {
-#ifdef HAVE_EXECINFO_H
+#if defined(HAVE_EXECINFO_H) && defined(__GLIBC__)
     const int stack_depth=20;
     void *temp[stack_depth];
     char **strings;
