--- CMakeLists.txt
+++ CMakeLists.txt
@@ -1,4 +1,11 @@
 PROJECT(miniz C)
+
+set(MINIZ_API_VERSION 2)
+set(MINIZ_MINOR_VERSION 1)
+set(MINIZ_PATCH_VERSION 0)
+set(MINIZ_VERSION
+    ${MINIZ_API_VERSION}.${MINIZ_MINOR_VERSION}.${MINIZ_PATCH_VERSION})
+
 cmake_minimum_required(VERSION 2.8)
 if(CMAKE_BUILD_TYPE STREQUAL "")
   # CMake defaults to leaving CMAKE_BUILD_TYPE empty. This screws up
@@ -12,5 +19,12 @@ set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_SOURCE_DIR}/bin)

 set(miniz_SOURCE miniz.c miniz_zip.c miniz_tinfl.c miniz_tdef.c)

-add_library(miniz ${miniz_SOURCE})
+add_library(objlibs OBJECT ${miniz_SOURCE})
+add_library(miniz_static STATIC $<TARGET_OBJECTS:objlibs>)
+add_library(miniz SHARED $<TARGET_OBJECTS:objlibs>)
+add_compile_options(-fPIC)
+set_property(TARGET ${PROJECT_NAME} PROPERTY SOVERSION ${MINIZ_API_VERSION})
+
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC")
+
 target_include_directories(miniz PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}")

 set(EXAMPLE1_SRC_LIST "${CMAKE_CURRENT_SOURCE_DIR}/examples/example1.c")
