commit a6bf0fe8d243bb7637410985dac6dfee42f56ff7
Author: T.J. Townsend <blakkheim@archlinux.org>
Date:   Sun Oct 9 22:51:53 2022 -0400

    Fetch modules over HTTPS instead of FTP by default.

diff --git a/src/backend/module_manager.cc b/src/backend/module_manager.cc
index 1b53a601..b0b3b0e0 100644
--- a/src/backend/module_manager.cc
+++ b/src/backend/module_manager.cc
@@ -554,13 +554,13 @@ void backend_init_module_mgr_config(void)
 
 	SWConfig config(confPath.c_str());
 
-	InstallSource is("FTP");
+	InstallSource is("HTTPS");
 	is.caption = "CrossWire";
-	is.source = "ftp.crosswire.org";
-	is.directory = "/pub/sword/raw";
+	is.source = "www.crosswire.org";
+	is.directory = "/ftpmirror/pub/sword/raw";
 
 	config["General"]["PassiveFTP"] = "true";
-	config["Sources"]["FTPSource"] = is.getConfEnt();
+	config["Sources"]["HTTPSSource"] = is.getConfEnt();
 	config.save();
 
 	InstallSource is_local("DIR");
