diff --git a/src/ui/graph/GraphNode.cpp b/src/ui/graph/GraphNode.cpp
index dfefc5d..65071b3 100644
--- a/src/ui/graph/GraphNode.cpp
+++ b/src/ui/graph/GraphNode.cpp
@@ -80,9 +80,9 @@ void CGraphNode::update() {
         const auto& p = m_node->m_ports.at(i);
 
         if (p->m_output)
-            outputVec.append_range(p->m_channels);
+            outputVec.insert(outputVec.end(), p->m_channels.begin(), p->m_channels.end());
         else
-            inputVec.append_range(p->m_channels);
+            inputVec.insert(inputVec.end(), p->m_channels.begin(), p->m_channels.end());
     }
 
     size_t      maxRows = std::max(inputVec.size(), outputVec.size());
