Fix for https://bugs.launchpad.net/onboard/+bug/1948723


--
--- a/Onboard/LayoutLoaderSVG.py
+++ b/Onboard/LayoutLoaderSVG.py
@@ -445,7 +445,7 @@
             except KeyError as ex:
                 (strerror) = ex
                 raise Exceptions.LayoutFileError("Unrecognized modifier %s in" \
-                    "definition of %s" (strerror, full_id))
+                    "definition of %s" % (strerror, full_id))
 
         value = attributes.get("action")
         if value:
