--- a/debug-me.cabal	2025-07-05 13:11:54.095974453 +0100
+++ b/debug-me.cabal	2025-07-05 13:11:59.253974699 +0100
@@ -55,7 +55,7 @@
   Build-Depends:
       base (>= 4.9 && < 5.0)
     , network (>= 2.6)
-    , bytestring < 0.12
+    , bytestring < 0.13
     , cryptonite (>= 0.20)
     , unix (>= 2.7)
     , process (>= 1.4)
@@ -64,7 +64,7 @@
     , stm-chans (>= 3.0)
     , posix-pty (>= 0.2.1)
     , terminal-size (>= 0.3)
-    , aeson (>= 0.11 && < 2.1)
+    , aeson (>= 0.11 && < 2.3)
     , sandi (>= 0.4)
     , text (>= 1.2.2)
     , optparse-applicative (>= 0.12)
