--- Role/User.hs
+++ Role/User.hs
@@ -247,7 +247,7 @@ getDeveloperMessage ichan ochan us now = maybe
 getDeveloperMessage' :: MissingHashes (Message Entered) -> TMChan (Message Seen) -> TVar UserState -> POSIXTime -> STM Input
 getDeveloperMessage' (MissingHashes wiremsg) ochan us now = do
 	st <- readTVar us
-	Developer msg <- restoreHashes (userStateRecentActivity us) (MissingHashes (Developer wiremsg))
+	~(Developer msg) <- restoreHashes (userStateRecentActivity us) (MissingHashes (Developer wiremsg))
 	case msg of
 		ControlMessage (Control (SessionKey spk _) _) -> do
 			let sigverifier = mkSigVerifier $ case spk of
