--- doc/Makefile.in.orig	2017-02-08 15:54:38.164525908 +0100
+++ doc/Makefile.in	2017-02-08 15:57:04.101172972 +0100
@@ -235,7 +235,6 @@ GETTEXT_MACRO_VERSION = @GETTEXT_MACRO_V
 GMSGFMT = @GMSGFMT@
 GMSGFMT_015 = @GMSGFMT_015@
 GREP = @GREP@
-HELP2MAN = @HELP2MAN@
 INDENT = @INDENT@
 INSTALL = @INSTALL@
 INSTALL_DATA = @INSTALL_DATA@
@@ -351,7 +350,6 @@ target_alias = @target_alias@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-help2man = @HELP2MAN@
 FLEX = $(top_builddir)/src/flex$(EXEEXT)
 info_TEXINFOS = flex.texi
 dist_man_MANS = flex.1
@@ -880,11 +878,8 @@ uninstall-man: uninstall-man1
 .PRECIOUS: Makefile
 
 
-flex.1: $(top_srcdir)/configure.ac $(top_srcdir)/src/flex.skl $(top_srcdir)/src/options.c $(top_srcdir)/src/options.h | $(FLEX)
-	$(help2man) --name='$(PACKAGE_NAME)' --section=1 \
-	--source='The Flex Project' --manual='Programming' \
-	--output=$@ $(FLEX) \
-	|| rm -f $@
+flex.1:
+	$(INSTALL) -Dm644 doc/flex.1 $(DESTDIR)/usr/share/man/man1/flex.1
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
