--- a/doc/Makefile.in	2017-05-17 09:26:34.592412829 +0200
+++ b/doc/Makefile.in	2017-05-17 09:27:07.176392146 +0200
@@ -241,7 +241,6 @@ GETTEXT_MACRO_VERSION = @GETTEXT_MACRO_V
 GMSGFMT = @GMSGFMT@
 GMSGFMT_015 = @GMSGFMT_015@
 GREP = @GREP@
-HELP2MAN = @HELP2MAN@
 INDENT = @INDENT@
 INSTALL = @INSTALL@
 INSTALL_DATA = @INSTALL_DATA@
@@ -872,11 +871,8 @@ uninstall-man: uninstall-man1
 .PRECIOUS: Makefile
 
 
-flex.1: $(top_srcdir)/configure.ac $(top_srcdir)/src/flex.skl $(top_srcdir)/src/options.c $(top_srcdir)/src/options.h | $(FLEX)
-	$(HELP2MAN) --name='$(PACKAGE_NAME)' --section=1 \
-	--source='The Flex Project' --manual='Programming' \
-	--output=$@ $(FLEX) \
-	|| rm -f $@
+flex.1:
+	$(INSTALL) -Dm644 doc/flex.1 $(DESTDIR)/usr/share/man/man1/flex.1
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
