diff -rupN Makefile.cfg Makefile.cfg
--- a/Makefile.cfg	2009-05-31 13:20:39.000000000 -0500
+++ b/Makefile.cfg	2014-11-13 22:42:41.460682486 -0600
@@ -19,11 +19,11 @@
 #
 
 export libdir ?= lib
-export prefix ?= /usr/local
+export prefix ?= /usr
 export DESTDIR ?= /
 
 ifneq ($(shell spl-config --cflags 2>/dev/null),)
-FOUND_SPL = 1
+FOUND_SPL = 0
 else
 FOUND_SPL = 0
 endif
@@ -35,19 +35,19 @@ FOUND_SWIG = 0
 endif
 
 ifneq ($(shell perl -le 'print 1' 2>/dev/null),)
-FOUND_PERL5 = 1
+FOUND_PERL5 = 0
 else
 FOUND_PERL5 = 0
 endif
 
 ifneq ($(shell python -c 'print 1' 2>/dev/null),)
-FOUND_PYTHON = 1
+FOUND_PYTHON = 0
 else
 FOUND_PYTHON = 0
 endif
 
 ifneq ($(shell ruby -e 'puts 1' 2>/dev/null),)
-FOUND_RUBY = 1
+FOUND_RUBY = 0
 else
 FOUND_RUBY = 0
 endif
diff -rupN stfl_internals.h stfl_internals.h
--- a/stfl_internals.h	2014-04-25 06:36:07.000000000 -0500
+++ b/stfl_internals.h	2014-11-13 22:41:38.413685526 -0600
@@ -28,7 +28,7 @@ extern "C" {
 #endif
 
 #include "stfl.h"
-#include <ncursesw/ncurses.h>
+#include <ncurses.h>
 #include <pthread.h>
 
 struct stfl_widget_type;
