--- a/Makefile.in
+++ b/Makefile.in
@@ -1955,7 +1955,7 @@
 
 # Use distutils to install the module.
 install-exec-local: .stamp-distutils-in-builddir
-	$(PYTHON) setup.py install --prefix=$(DESTDIR)$(prefix)
+	$(PYTHON) setup.py install --prefix=$(prefix) --root=$(DESTDIR) --single-version-externally-managed
 
 # Uninstall the module, crossing our fingers that we know enough
 # about how distutils works to do this.  Unfortunately, distutils
