--- configure.ac.orig	2018-01-31 21:22:40.957541567 -0800
+++ configure.ac	2018-01-31 21:23:27.838559722 -0800
@@ -49,15 +49,6 @@
 
 LT_INIT([win32-dll])
 
-AM_MISSING_PROG(HELP2ADOC, help2adoc, $missing_dir)
-AM_MISSING_PROG(HELP2MAN, help2man, $missing_dir)
-if ! which help2man >/dev/null ; then
-   AC_MSG_ERROR([help2man missing])
-fi
-if ! which gengetopt >/dev/null ; then
-   AC_MSG_ERROR([gengetopt missing])
-fi
-
 gl_LD_VERSION_SCRIPT
 
 PKG_CHECK_MODULES([LIBJSON], [json-c], [], [
