--- a/src/Makefile.am	2018-01-31 21:34:44.725821832 -0800
+++ b/src/Makefile.am	2018-01-31 21:35:32.885840481 -0800
@@ -40,24 +40,6 @@
 BUILT_SOURCES = cmdline.c cmdline.h
 MAINTAINERCLEANFILES = $(BUILT_SOURCES)
 
-dist_man_MANS = $(top_srcdir)/man/u2f-server.1
-DISTCLEANFILES = $(dist_man_MANS)
-
-$(top_srcdir)/man/u2f-server.1: $(srcdir)/u2f-server.c $(srcdir)/cmdline.ggo $(top_srcdir)/configure.ac | $(builddir)/u2f-server$(EXEEXT)
-	$(HELP2MAN) \
-		--output=$(top_builddir)/man/$@ $(builddir)/u2f-server$(EXEEXT) \
-		--name="Yubico Universal 2nd Factor (U2F) Server Tool" \
-		--no-info
-if ENABLE_H2A
-	$(HELP2ADOC) \
-		-e $(builddir)/u2f-server$(EXEEXT) \
-		--name="Yubico Universal 2nd Factor (U2F) Server Tool" \
-		> $(top_builddir)/man/$@.txt
-endif
-
-clean-local:
-	rm -f $(top_builddir)/man/*.[1-9]
-
 if ENABLE_COV
 AM_CFLAGS += --coverage
 AM_LDFLAGS = --coverage
