--- a/lib/ansible/modules/system/service.py
+++ b/lib/ansible/modules/system/service.py
@@ -414,7 +414,7 @@ class Service(object):
 
             # Write out the contents of the list into our temporary file.
             for rcline in new_rc_conf:
-                os.write(TMP_RCCONF, rcline)
+                os.write(TMP_RCCONF, to_bytes(rcline))
 
             # Close temporary file.
             os.close(TMP_RCCONF)
@@ -1111,7 +1111,7 @@ class DragonFlyBsdService(FreeBsdService
             if os.path.isfile(rcfile):
                 self.rcconf_file = rcfile
 
-        self.rcconf_key = "%s" % string.replace(self.name, "-", "_")
+        self.rcconf_key = "%s" % self.name.replace("-", "_")
 
         return self.service_enable_rcconf()
 
@@ -1299,7 +1299,7 @@ class NetBsdService(Service):
             if os.path.isfile(rcfile):
                 self.rcconf_file = rcfile
 
-        self.rcconf_key = "%s" % string.replace(self.name, "-", "_")
+        self.rcconf_key = "%s" % self.name.replace("-", "_")
 
         return self.service_enable_rcconf()
 
