From 501b46c364c5608d1b3b84dd48fe57abcabd4aea Mon Sep 17 00:00:00 2001
From: Ionut Biru <ibiru@archlinux.org>
Date: Mon, 6 Dec 2010 04:22:45 -0800
Subject: [PATCH] bugfix: remove bad import from im module

Signed-off-by: Ionut Biru <ibiru@archlinux.org>
---
 butterfly/channel/im.py |    2 --
 1 files changed, 0 insertions(+), 2 deletions(-)

diff --git a/butterfly/channel/im.py b/butterfly/channel/im.py
index 1781234..ac8dd44 100644
--- butterfly/channel/im.py
+++ butterfly/channel/im.py
@@ -30,8 +30,6 @@ import papyon.event
 
 from butterfly.channel.text import ButterflyTextChannel
 
-from butterfly.Channel_Interface_Conference import CHANNEL_INTERFACE_CONFERENCE
-
 __all__ = ['ButterflyImChannel']
 
 logger = logging.getLogger('Butterfly.ImChannel')
-- 
1.7.3.2

