diff --git a/Makefile b/Makefile
index 065f1dd..15d5322 100644
--- a/Makefile
+++ b/Makefile
@@ -13,21 +13,21 @@ man1_targets = \
 	s6-taiclockd.1 \
 	s6-tcpclient.1 \
 	s6-tcpserver.1 \
-	s6-tcpserver-access.1.in \
-	s6-tcpserver4.1.in \
-	s6-tcpserver4-socketbinder.1.in \
-	s6-tcpserver4d.1.in \
-	s6-tcpserver6.1.in \
-	s6-tcpserver6-socketbinder.1.in \
-	s6-tcpserver6d.1.in \
-	s6-tlsc.1.in \
-	s6-tlsc-io.1.in \
-	s6-tlsclient.1.in \
-	s6-tlsd.1.in \
-	s6-tlsd-io.1.in \
-	s6-tlsserver.1.in \
-	s6-ucspitlsc.1.in \
-	s6-ucspitlsd.1.in
+	s6-tcpserver-access.1 \
+	s6-tcpserver4.1 \
+	s6-tcpserver4-socketbinder.1 \
+	s6-tcpserver4d.1 \
+	s6-tcpserver6.1 \
+	s6-tcpserver6-socketbinder.1 \
+	s6-tcpserver6d.1 \
+	s6-tlsc.1 \
+	s6-tlsc-io.1 \
+	s6-tlsclient.1 \
+	s6-tlsd.1 \
+	s6-tlsd-io.1 \
+	s6-tlsserver.1 \
+	s6-ucspitlsc.1 \
+	s6-ucspitlsd.1
 
 man7_targets = \
 	s6-tls.7
