--- a/conf/icecast.xml.in	2026-03-29 16:57:32.453024858 +0200
+++ b/conf/icecast.xml.in	2026-03-29 17:00:30.581264225 +0200
@@ -298,12 +298,10 @@
 
     <security>
         <chroot>false</chroot>
-        <!--
         <changeowner>
             <user>nobody</user>
             <group>nogroup</group>
         </changeowner>
-        -->
 
         <tls-context>
             <!-- The certificate file containng public and optionally private key.
