--- a/src/event_stream.c	2026-03-29 18:18:28.331745052 +0200
+++ b/src/event_stream.c	2026-03-29 18:19:44.219106053 +0200
@@ -161,7 +161,8 @@
     if (event_stream_thread)
         thread_join(event_stream_thread);
 
-    avl_tree_free(client_tree, _free_client);
+    if (client_tree)
+        avl_tree_free(client_tree, _free_client);
 
     thread_mutex_lock(&event_stream_event_mutex);
     igloo_ro_unref(&event_queue);

