diff -Naur ./plugins/adplug/adplug/s3m.cpp ../deadbeef-0.7.2/plugins/adplug/adplug/s3m.cpp
--- ./plugins/adplug/adplug/s3m.cpp	2016-06-19 13:26:18.000000000 +0200
+++ ../deadbeef-0.7.2/plugins/adplug/adplug/s3m.cpp	2017-07-22 10:38:22.694537750 +0200
@@ -26,7 +26,7 @@
 #include <string.h>
 #include "s3m.h"
 
-const char Cs3mPlayer::chnresolv[] =	// S3M -> adlib channel conversion
+const signed char Cs3mPlayer::chnresolv[] =	// S3M -> adlib channel conversion
   {-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,0,1,2,3,4,5,6,7,8,-1,-1,-1,-1,-1,-1,-1};
 
 const unsigned short Cs3mPlayer::notetable[12] =		// S3M adlib note table
diff -Naur ./plugins/adplug/adplug/s3m.h ../deadbeef-0.7.2/plugins/adplug/adplug/s3m.h
--- ./plugins/adplug/adplug/s3m.h	2016-06-19 13:26:18.000000000 +0200
+++ ../deadbeef-0.7.2/plugins/adplug/adplug/s3m.h	2017-07-22 10:43:59.836740736 +0200
@@ -92,7 +92,7 @@
   char filetype[30];
 
  private:
-  static const char chnresolv[];
+  static const signed char chnresolv[];
   static const unsigned short notetable[12];
   static const unsigned char vibratotab[32];
 
diff -Naur ./plugins/sid/sidplay-libs/libsidplay/src/xsid/xsid.cpp ../deadbeef.old/plugins/sid/sidplay-libs/libsidplay/src/xsid/xsid.cpp
--- ./plugins/sid/sidplay-libs/libsidplay/src/xsid/xsid.cpp	2017-07-22 11:09:15.117275762 +0200
+++ ../deadbeef.old/plugins/sid/sidplay-libs/libsidplay/src/xsid/xsid.cpp	2016-06-19 13:26:18.000000000 +0200
@@ -81,7 +81,7 @@
     '\x08', '\x19', '\x2a', '\x3b', '\x4c', '\x5d', '\x6e', '\x7f'
 };
 */
-const int8_t XSID::sampleConvertTable[16] =
+const char XSID::sampleConvertTable[16] =
 {
     '\x80', '\x94', '\xa9', '\xbc', '\xce', '\xe1', '\xf2', '\x03',
     '\x1b', '\x2a', '\x3b', '\x49', '\x58', '\x66', '\x73', '\x7f'
diff -Naur ./plugins/sid/sidplay-libs/libsidplay/src/xsid/xsid.h ../deadbeef.old/plugins/sid/sidplay-libs/libsidplay/src/xsid/xsid.h
--- ./plugins/sid/sidplay-libs/libsidplay/src/xsid/xsid.h	2017-07-22 11:08:43.054046487 +0200
+++ ../deadbeef.old/plugins/sid/sidplay-libs/libsidplay/src/xsid/xsid.h	2016-06-19 13:26:18.000000000 +0200
@@ -234,7 +234,7 @@
     uint8_t             sidData0x18;
     bool                _sidSamples;
     int8_t              sampleOffset;
-    static const int8_t sampleConvertTable[16];
+    static const char sampleConvertTable[16];
     bool                wasRunning;
 
 private:
