diff -ru ./plugins/artwork/artwork.c ../deadbeef-0.7.0_/plugins/artwork/artwork.c
--- ./plugins/artwork/artwork.c	2016-01-17 23:10:44.000000000 +0600
+++ ../deadbeef-0.7.0_/plugins/artwork/artwork.c	2016-02-03 17:41:23.000000000 +0600
@@ -27,6 +27,7 @@
 #endif
 #include <stdio.h>
 #include <stdlib.h>
+#include <limits.h>
 #include <string.h>
 #include <ctype.h>
 #include <libgen.h>
Only in ../deadbeef-0.7.0_/plugins/artwork: artwork.c.orig
diff -ru ./plugins/artwork/artwork_internal.c ../deadbeef-0.7.0_/plugins/artwork/artwork_internal.c
--- ./plugins/artwork/artwork_internal.c	2016-01-03 02:28:08.000000000 +0600
+++ ../deadbeef-0.7.0_/plugins/artwork/artwork_internal.c	2016-02-03 17:46:29.000000000 +0600
@@ -26,6 +26,7 @@
     #include "../../config.h"
 #endif
 #include <stdlib.h>
+#include <limits.h>
 #include <string.h>
 #include <libgen.h>
 #include <errno.h>
diff -ru ./plugins/artwork/cache.c ../deadbeef-0.7.0_/plugins/artwork/cache.c
--- ./plugins/artwork/cache.c	2016-01-03 02:28:08.000000000 +0600
+++ ../deadbeef-0.7.0_/plugins/artwork/cache.c	2016-02-03 17:46:05.000000000 +0600
@@ -25,6 +25,7 @@
     #include "../../config.h"
 #endif
 #include <stdlib.h>
+#include <limits.h>
 #include <string.h>
 #include <time.h>
 #include <libgen.h>
diff -ru ./plugins/cdda/cdda.c ../deadbeef-0.7.0_/plugins/cdda/cdda.c
--- ./plugins/cdda/cdda.c	2015-07-15 13:06:25.000000000 +0600
+++ ../deadbeef-0.7.0_/plugins/cdda/cdda.c	2016-02-03 17:41:23.000000000 +0600
@@ -20,6 +20,7 @@
 
 #include <stdio.h>
 #include <stdlib.h>
+#include <limits.h>
 #include <string.h>
 #include <sys/types.h>
 
Only in ../deadbeef-0.7.0_/plugins/cdda: cdda.c.orig
diff -ru ./plugins/flac/flac.c ../deadbeef-0.7.0_/plugins/flac/flac.c
--- ./plugins/flac/flac.c	2016-01-03 02:27:39.000000000 +0600
+++ ../deadbeef-0.7.0_/plugins/flac/flac.c	2016-02-03 17:41:23.000000000 +0600
@@ -36,6 +36,7 @@
 #include <string.h>
 #include <stdio.h>
 #include <stdlib.h>
+#include <limits.h>
 #include <stdbool.h>
 #include <math.h>
 #include <FLAC/stream_decoder.h>
diff -ru ./plugins/liboggedit/oggedit_flac.c ../deadbeef-0.7.0_/plugins/liboggedit/oggedit_flac.c
--- ./plugins/liboggedit/oggedit_flac.c	2016-01-03 02:28:08.000000000 +0600
+++ ../deadbeef-0.7.0_/plugins/liboggedit/oggedit_flac.c	2016-02-03 17:41:23.000000000 +0600
@@ -28,6 +28,7 @@
 #include <config.h>
 #endif
 #include <stdlib.h>
+#include <limits.h>
 #include <string.h>
 #include <unistd.h>
 #include <limits.h>
Only in ../deadbeef-0.7.0_/plugins/liboggedit: oggedit_flac.c.orig
diff -ru ./plugins/liboggedit/oggedit_internal.c ../deadbeef-0.7.0_/plugins/liboggedit/oggedit_internal.c
--- ./plugins/liboggedit/oggedit_internal.c	2016-01-03 02:28:08.000000000 +0600
+++ ../deadbeef-0.7.0_/plugins/liboggedit/oggedit_internal.c	2016-02-03 17:41:23.000000000 +0600
@@ -29,6 +29,7 @@
 #endif
 #include <stdio.h>
 #include <stdlib.h>
+#include <limits.h>
 #include <string.h>
 #include <unistd.h>
 #include <libgen.h>
diff -ru ./plugins/liboggedit/oggedit_vorbis.c ../deadbeef-0.7.0_/plugins/liboggedit/oggedit_vorbis.c
--- ./plugins/liboggedit/oggedit_vorbis.c	2016-01-03 02:28:08.000000000 +0600
+++ ../deadbeef-0.7.0_/plugins/liboggedit/oggedit_vorbis.c	2016-02-03 17:41:23.000000000 +0600
@@ -28,6 +28,7 @@
 #include <config.h>
 #endif
 #include <stdlib.h>
+#include <limits.h>
 #include <string.h>
 #include <unistd.h>
 #include <limits.h>
Only in ../deadbeef-0.7.0_/plugins/liboggedit: oggedit_vorbis.c.orig
