diff -ru ./conf.c ../_patched_deadbeef-0.6.2/conf.c
--- ./conf.c	2014-08-05 17:54:03.000000000 +0000
+++ ../_patched_deadbeef-0.6.2/conf.c	2015-08-05 13:56:05.209000000 +0000
@@ -29,6 +29,7 @@
 #include <stdint.h>
 #include <string.h>
 #include <stdlib.h>
+#include <limits.h>
 #include <inttypes.h>
 #include <errno.h>
 #include <unistd.h>
Only in ../_patched_deadbeef-0.6.2/: main.c.orig
Only in ./: musl_limits_h.patch
diff -ru ./plugins/artwork/artwork.c ../_patched_deadbeef-0.6.2/plugins/artwork/artwork.c
--- ./plugins/artwork/artwork.c	2014-07-31 19:03:51.000000000 +0000
+++ ../_patched_deadbeef-0.6.2/plugins/artwork/artwork.c	2015-08-05 14:20:35.774000000 +0000
@@ -27,6 +27,7 @@
 #endif
 #include <stdio.h>
 #include <stdlib.h>
+#include <limits.h>
 #include <string.h>
 #include <sys/stat.h>
 #ifdef __linux__
diff -ru ./plugins/cdda/cdda.c ../_patched_deadbeef-0.6.2/plugins/cdda/cdda.c
--- ./plugins/cdda/cdda.c	2014-07-12 20:30:07.000000000 +0000
+++ ../_patched_deadbeef-0.6.2/plugins/cdda/cdda.c	2015-08-05 14:17:37.715000000 +0000
@@ -23,6 +23,7 @@
 #endif
 #include <stdio.h>
 #include <stdlib.h>
+#include <limits.h>
 #include <string.h>
 #include <assert.h>
 #include <sys/types.h>
diff -ru ./plugins/converter/converter.c ../_patched_deadbeef-0.6.2/plugins/converter/converter.c
--- ./plugins/converter/converter.c	2014-07-31 19:16:23.000000000 +0000
+++ ../_patched_deadbeef-0.6.2/plugins/converter/converter.c	2015-08-05 13:56:05.209000000 +0000
@@ -26,6 +26,7 @@
 #include <sys/syslimits.h>
 #endif
 #include <stdlib.h>
+#include <limits.h>
 #include <string.h>
 #include <sys/types.h>
 #include <sys/stat.h>
diff -ru ./plugins/flac/flac.c ../_patched_deadbeef-0.6.2/plugins/flac/flac.c
--- ./plugins/flac/flac.c	2014-08-05 17:54:03.000000000 +0000
+++ ../_patched_deadbeef-0.6.2/plugins/flac/flac.c	2015-08-05 14:15:56.192000000 +0000
@@ -36,6 +36,7 @@
 #include <string.h>
 #include <stdio.h>
 #include <stdlib.h>
+#include <limits.h>
 #include <stdbool.h>
 #include <math.h>
 #include <FLAC/stream_decoder.h>
diff -ru ./plugins/liboggedit/oggedit_flac.c ../_patched_deadbeef-0.6.2/plugins/liboggedit/oggedit_flac.c
--- ./plugins/liboggedit/oggedit_flac.c	2014-07-31 19:02:29.000000000 +0000
+++ ../_patched_deadbeef-0.6.2/plugins/liboggedit/oggedit_flac.c	2015-08-05 14:13:00.237000000 +0000
@@ -28,6 +28,7 @@
 #include <config.h>
 #endif
 #include <stdlib.h>
+#include <limits.h>
 #include <string.h>
 #include <unistd.h>
 #include "oggedit_internal.h"
diff -ru ./plugins/liboggedit/oggedit_internal.c ../_patched_deadbeef-0.6.2/plugins/liboggedit/oggedit_internal.c
--- ./plugins/liboggedit/oggedit_internal.c	2014-07-31 19:02:29.000000000 +0000
+++ ../_patched_deadbeef-0.6.2/plugins/liboggedit/oggedit_internal.c	2015-08-05 14:10:49.465000000 +0000
@@ -29,6 +29,7 @@
 #endif
 #include <stdio.h>
 #include <stdlib.h>
+#include <limits.h>
 #include <string.h>
 #include <unistd.h>
 #include <libgen.h>
diff -ru ./plugins/liboggedit/oggedit_vorbis.c ../_patched_deadbeef-0.6.2/plugins/liboggedit/oggedit_vorbis.c
--- ./plugins/liboggedit/oggedit_vorbis.c	2014-07-31 19:02:29.000000000 +0000
+++ ../_patched_deadbeef-0.6.2/plugins/liboggedit/oggedit_vorbis.c	2015-08-05 14:12:05.786000000 +0000
@@ -28,6 +28,7 @@
 #include <config.h>
 #endif
 #include <stdlib.h>
+#include <limits.h>
 #include <string.h>
 #include <unistd.h>
 #include "oggedit_internal.h"
diff -ru ./plugins/m3u/m3u.c ../_patched_deadbeef-0.6.2/plugins/m3u/m3u.c
--- ./plugins/m3u/m3u.c	2014-07-31 19:02:29.000000000 +0000
+++ ../_patched_deadbeef-0.6.2/plugins/m3u/m3u.c	2015-08-05 13:56:05.209000000 +0000
@@ -26,6 +26,7 @@
 #endif
 #include <string.h>
 #include <stdlib.h>
+#include <limits.h>
 #include <math.h> // for ceil
 #if HAVE_SYS_SYSLIMITS_H
 #include <sys/syslimits.h>
