From 6d39333448dd7c48fb3f4e07ff5d7e05d0527fff Mon Sep 17 00:00:00 2001
From: Carlo Landmeter <clandmeter@gmail.com>
Date: Wed, 29 Oct 2014 10:36:57 +0100
Subject: [PATCH] musl libc does not have execinfo.h

---
 main.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/main.c b/main.c
index d7dab49..797889c 100644
--- a/main.c
+++ b/main.c
@@ -51,7 +51,7 @@
 #include <sys/fcntl.h>
 #include <sys/errno.h>
 #include <signal.h>
-#ifdef __linux__
+#ifdef __GLIBC__
 #include <execinfo.h>
 #endif
 #include <unistd.h>
@@ -642,7 +642,7 @@ player_mainloop (void) {
     }
 }
 
-#ifdef __linux__
+#ifdef __GLIBC__
 void
 sigsegv_handler (int sig) {
     fprintf (stderr, "Segmentation Fault\n");
@@ -761,7 +761,7 @@ main (int argc, char *argv[]) {
     }
 #endif
 
-#ifdef __linux__
+#ifdef __GLIBC__
     signal (SIGSEGV, sigsegv_handler);
 #endif
     setlocale (LC_ALL, "");
