--- KF5AuthConfig.cmake.in.ORIG
+++ KF5AuthConfig.cmake.in
@@ -21,4 +21,8 @@
     else()
         include("${CMAKE_CURRENT_LIST_DIR}/KF5AuthToolsTargets.cmake")
     endif()
+    if(CMAKE_CROSSCOMPILING)
+        set_target_properties(KF5::kauth-policy-gen PROPERTIES IMPORTED_LOCATION_NONE /usr/lib/libexec/kauth/kauth-policy-gen)
+        set_target_properties(KF5::kauth-policy-gen PROPERTIES IMPORTED_LOCATION /usr/lib/libexec/kauth/kauth-policy-gen)
+    endif()
 endif()
--- cmake/KF5AuthMacros.cmake.ORIG
+++ cmake/KF5AuthMacros.cmake
@@ -65,0 +66,4 @@
+    set(KAUTH_POLICY_GEN_LOCATION KF5::kauth-policy-gen)
+    if(CMAKE_CROSSCOMPILING)
+        get_target_property(KAUTH_POLICY_GEN_LOCATION KF5::kauth-policy-gen LOCATION)
+    endif()
@@ -67 +71 @@
-                       COMMAND KF5::kauth-policy-gen ${_input} ${_output}
+                       COMMAND ${KAUTH_POLICY_GEN_LOCATION} ${_input} ${_output}
