*** a/libavcodec/libx265.c	2024-08-01 19:22:48.000000000 -0400
--- b/libavcodec/libx265.c	2025-05-31 01:04:16.111611835 -0400
*************** FF_ENABLE_DEPRECATION_WARNINGS
*** 705,709 ****
      }
! 
      ret = ctx->api->encoder_encode(ctx->encoder, &nal, &nnal,
!                                    pic ? &x265pic : NULL, &x265pic_out);
  
--- 705,709 ----
      }
!     x265_picture *x265pic_out_ptr = &x265pic_out;
      ret = ctx->api->encoder_encode(ctx->encoder, &nal, &nnal,
!                                    pic ? &x265pic : NULL, &x265pic_out_ptr);
  
