--- a/setup.py
+++ b/setup.py
@@ -209,7 +209,11 @@
             # generate src/SWIG/x509_v_flag.h to overcome weaknesses of swig
             # https://todo.sr.ht/~mcepl/m2crypto/298
             with open("src/SWIG/x509_v_flag.h", "w", encoding="utf-8") as x509_v_h:
-                cmd = [shutil.which(os.environ.get("CC", "gcc"))]
+                cmd = shlex.split(os.environ.get("CC", "gcc"))
+                if cmd:
+                    cmd[0] = shutil.which(cmd[0])
+                if not cmd and cmd[0]:
+                    raise RuntimeError("compiler is not in path")
                 cflags = os.environ.get("CFLAGS")
                 if cflags is not None:
                     cmd += cflags.split()
