--- a/src/debug.h	2018-03-27 11:19:24.000000000 +0200
+++ b/src/debug.h	2018-03-27 11:19:24.000000000 +0200
@@ -24,7 +24,7 @@
 # define ASSERT(left, operator, right) (void(0))
 #else
 
-#if !defined(_WIN32) && !defined(__APPLE__) && !defined(__ANDROID__)
+#if !defined(_WIN32) && !defined(__APPLE__) && !defined(__ANDROID__) && defined(__GLIBC__)
 #include <execinfo.h>
 #endif
 
@@ -35,7 +35,7 @@
       vara << "[" << a << "] " << op << " " << varb << "[" << b << "]" <<
       std::endl;
 
-#if !defined(_WIN32) && !defined(__APPLE__) && !defined(__ANDROID__)
+#if !defined(_WIN32) && !defined(__APPLE__) && !defined(__ANDROID__) && defined(__GLIBC__)
   void *callstack[64];
   size_t size;
   size = backtrace(callstack, 64);
