diff -up SDL-1.2.14/include/SDL_endian.h.byteorder SDL-1.2.14/include/SDL_endian.h
--- SDL-1.2.14/include/SDL_endian.h.byteorder	2009-10-13 01:07:19.000000000 +0200
+++ SDL-1.2.14/include/SDL_endian.h	2009-10-26 14:27:39.000000000 +0100
@@ -39,6 +39,10 @@
 /*@}*/
 
 #ifndef SDL_BYTEORDER	/* Not defined in SDL_config.h? */
+#ifdef __linux__
+#include <endian.h>
+#define SDL_BYTEORDER  __BYTE_ORDER
+#else /* __linux __ */
 #if defined(__hppa__) || \
     defined(__m68k__) || defined(mc68000) || defined(_M_M68K) || \
     (defined(__MIPS__) && defined(__MISPEB__)) || \
@@ -48,6 +52,7 @@
 #else
 #define SDL_BYTEORDER	SDL_LIL_ENDIAN
 #endif
+#endif /* __linux __ */
 #endif /* !SDL_BYTEORDER */
 
 
