Description: Fix crash with joystick detection.
Index: libsdl1.2-1.2.14/src/joystick/linux/SDL_sysjoystick.c
===================================================================
--- libsdl1.2-1.2.14.orig/src/joystick/linux/SDL_sysjoystick.c	2010-01-12 12:37:36.000000000 -0500
+++ libsdl1.2-1.2.14/src/joystick/linux/SDL_sysjoystick.c	2010-01-12 12:38:27.000000000 -0500
@@ -700,7 +700,7 @@
 				continue;
 			}
 			if ( test_bit(i, absbit) ) {
-				int values[5];
+				int values[6];
 
 				if ( ioctl(fd, EVIOCGABS(i), values) < 0 )
 					continue;
