Index: loudgain-0.6.8/CMakeLists.txt
===================================================================
--- loudgain-0.6.8.orig/CMakeLists.txt
+++ loudgain-0.6.8/CMakeLists.txt
@@ -69,12 +69,12 @@ TARGET_LINK_LIBRARIES(loudgain
 )
 
 SET_TARGET_PROPERTIES(loudgain PROPERTIES
-  COMPILE_FLAGS "-Wall -pedantic -g"
+  COMPILE_FLAGS "-Wall -pedantic"
 )
 
-SET(CMAKE_C_FLAGS "-std=gnu99 -D_GNU_SOURCE")
+SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=gnu99 -D_GNU_SOURCE")
 
-SET(CMAKE_CXX_FLAGS "-std=gnu++11 -D_GNU_SOURCE")
+SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=gnu++11 -D_GNU_SOURCE")
 
 INSTALL(TARGETS loudgain DESTINATION ${CMAKE_INSTALL_PREFIX}/bin)
 
