--- a/libs/image/tests/kis_paint_device_test.cpp	2019-09-30 15:18:28.000000000 +0200
+++ b/libs/image/tests/kis_paint_device_test.cpp	2019-09-30 15:18:28.000000000 +0200
@@ -2354,7 +2354,7 @@
 
         qDebug() << "Iteration:" << i;
 
-#ifdef Q_OS_LINUX
+#ifdef __GLIBC__ 
         struct mallinfo info = mallinfo();
         qDebug() << "Allocated on heap:" << (info.arena >> 20) << "MiB";
         qDebug() << "Mmaped regions:" << info.hblks << (info.hblkhd >> 20) << "MiB";
