*** razorqt-session/src/main.cpp	2012-02-13 11:24:51.000000000 +0100
--- razorqt-session/src/main.cpp	2012-04-10 13:30:59.331210956 +0200
***************
*** 28,33 ****
--- 28,34 ----
  #include <QApplication>
  #include <QIcon>
  
+ #include <unistd.h>
  #include "razormodman.h"
  #include "sessiondbusadaptor.h"
  #include "razortranslate.h"
*** razorqt-session/src/razormodman.cpp	2012-02-13 11:24:51.000000000 +0100
--- razorqt-session/src/razormodman.cpp	2012-04-10 13:31:13.251209998 +0200
***************
*** 35,40 ****
--- 35,41 ----
  #include <QMessageBox>
  #include "wmselectdialog.h"
  #include <razorqt/xfitman.h>
+ #include <unistd.h>
  
  #define MAX_CRASHES_PER_APP 5
  
