index 2b34536..9f55dba 100644
--- a/servatrice/CMakeLists.txt
+++ b/servatrice/CMakeLists.txt
@@ -30,7 +30,7 @@ if(WIN32)
 endif(WIN32)
 
 # Under FreeBSD we need libexecinfo to use backtrace_symbols_fd()
-if(CMAKE_HOST_SYSTEM MATCHES "FreeBSD")
+if(CMAKE_HOST_SYSTEM MATCHES "FreeBSD" OR (ENABLE_LIBEXECINFO))
   find_package(Libexecinfo REQUIRED)
   set(SYSTEM_LIBRARIES ${EXECINFO_LIBRARY} ${SYSTEM_LIBRARIES})
 endif()
@@ -105,7 +105,7 @@ include_directories(${CMAKE_CURRENT_BINARY_DIR})
 # Build servatrice binary and link it
 add_executable(servatrice MACOSX_BUNDLE ${servatrice_MOC_SRCS} ${servatrice_RESOURCES_RCC} ${servatrice_SOURCES})
 
-if(CMAKE_HOST_SYSTEM MATCHES "FreeBSD")
+if(CMAKE_HOST_SYSTEM MATCHES "FreeBSD" OR (ENABLE_LIBEXECINFO))
   target_link_libraries(servatrice cockatrice_common Threads::Threads ${SERVATRICE_QT_MODULES} ${LIBEXECINFO_LIBRARY})
 else()
   target_link_libraries(servatrice cockatrice_common Threads::Threads ${SERVATRICE_QT_MODULES})
