diff --git a/Cargo.toml b/Cargo.toml
index d7ff624f..9f2e6360 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -53,7 +53,7 @@ hashbrown           = { version = "0.17.1", features = [ "serde" ] }
 indexmap            = { version = "2.14.0", features = [ "serde" ] }
 libc                = "0.2.186"
 lru                 = "0.18.0"
-mlua                = { version = "0.11.6", features = [ "anyhow", "async", "error-send", "lua55", "macros", "serde" ] }
+mlua                = { version = "0.11.6", features = [ "anyhow", "async", "error-send", "lua54", "macros", "serde" ] }
 objc2               = "0.6.4"
 ordered-float       = { version = "5.3.0", features = [ "serde" ] }
 parking_lot         = "0.12.5"
diff --git a/yazi-actor/src/lives/tab.rs b/yazi-actor/src/lives/tab.rs
index 69d5abb0..f376ab54 100644
--- a/yazi-actor/src/lives/tab.rs
+++ b/yazi-actor/src/lives/tab.rs
@@ -47,7 +47,7 @@ impl UserData for Tab {
 		cached_field!(fields, name, |lua, me| {
 			match me.name() {
 				Cow::Borrowed(s) => lua.create_string(s),
-				Cow::Owned(s) => lua.create_external_string(s),
+				Cow::Owned(s) => lua.create_string(s),
 			}
 		});
 
diff --git a/yazi-binding/src/error.rs b/yazi-binding/src/error.rs
index ff075c4b..4adada26 100644
--- a/yazi-binding/src/error.rs
+++ b/yazi-binding/src/error.rs
@@ -74,7 +74,7 @@ impl UserData for Error {
 	fn add_methods<M: UserDataMethods<Self>>(methods: &mut M) {
 		methods.add_meta_method(MetaMethod::ToString, |lua, me, ()| {
 			Ok(match me {
-				Self::Io(_) | Self::Fs(_) | Self::Serde(_) => lua.create_external_string(me.to_string()),
+				Self::Io(_) | Self::Fs(_) | Self::Serde(_) => lua.create_string(me.to_string()),
 				Self::Custom(s) => lua.create_string(&**s),
 			})
 		});
@@ -82,11 +82,11 @@ impl UserData for Error {
 			match (lhs, rhs) {
 				(Value::String(l), Value::UserData(r)) => {
 					let r = r.borrow::<Self>()?;
-					lua.create_external_string([&l.as_bytes(), r.to_string().as_bytes()].concat())
+					lua.create_string([&l.as_bytes(), r.to_string().as_bytes()].concat())
 				}
 				(Value::UserData(l), Value::String(r)) => {
 					let l = l.borrow::<Self>()?;
-					lua.create_external_string([l.to_string().as_bytes(), &r.as_bytes()].concat())
+					lua.create_string([l.to_string().as_bytes(), &r.as_bytes()].concat())
 				}
 				_ => Err("only string can be concatenated with Error".into_lua_err()),
 			}
diff --git a/yazi-binding/src/fd.rs b/yazi-binding/src/fd.rs
index 931329a7..01c6e64d 100644
--- a/yazi-binding/src/fd.rs
+++ b/yazi-binding/src/fd.rs
@@ -18,7 +18,7 @@ impl UserData for Fd {
 			match me.0.read(&mut buf).await {
 				Ok(n) => {
 					buf.truncate(n);
-					lua.create_external_string(buf)?.into_lua_multi(&lua)
+					lua.create_string(buf)?.into_lua_multi(&lua)
 				}
 				Err(e) => (Value::Nil, Error::Io(e)).into_lua_multi(&lua),
 			}
diff --git a/yazi-binding/src/path.rs b/yazi-binding/src/path.rs
index 09cc38d2..47ab66ae 100644
--- a/yazi-binding/src/path.rs
+++ b/yazi-binding/src/path.rs
@@ -132,7 +132,7 @@ impl UserData for Path {
 		methods.add_method("strip_prefix", |_, me, base: Value| me.strip_prefix(base));
 
 		methods.add_meta_method(MetaMethod::Concat, |lua, lhs, rhs: mlua::String| {
-			lua.create_external_string([lhs.encoded_bytes(), &rhs.as_bytes()].concat())
+			lua.create_string([lhs.encoded_bytes(), &rhs.as_bytes()].concat())
 		});
 		methods.add_meta_method(MetaMethod::Eq, |_, me, other: PathRef| Ok(me.inner == other.inner));
 		methods
diff --git a/yazi-binding/src/process/child.rs b/yazi-binding/src/process/child.rs
index b0550167..ada5d215 100644
--- a/yazi-binding/src/process/child.rs
+++ b/yazi-binding/src/process/child.rs
@@ -117,7 +117,7 @@ impl UserData for Child {
 		});
 		methods.add_async_method_mut("read_line", |lua, mut me, ()| async move {
 			match me.read_line().await {
-				(Some(b), event) => (lua.create_external_string(b)?, event).into_lua_multi(&lua),
+				(Some(b), event) => (lua.create_string(b)?, event).into_lua_multi(&lua),
 				(None, event) => (Value::Nil, event).into_lua_multi(&lua),
 			}
 		});
@@ -128,7 +128,7 @@ impl UserData for Child {
 				return (Value::Nil, 3u8).into_lua_multi(&lua);
 			};
 			match result {
-				(Some(b), event) => (lua.create_external_string(b)?, event).into_lua_multi(&lua),
+				(Some(b), event) => (lua.create_string(b)?, event).into_lua_multi(&lua),
 				(None, event) => (Value::Nil, event).into_lua_multi(&lua),
 			}
 		});
diff --git a/yazi-binding/src/process/output.rs b/yazi-binding/src/process/output.rs
index 97508849..7e2d2145 100644
--- a/yazi-binding/src/process/output.rs
+++ b/yazi-binding/src/process/output.rs
@@ -23,10 +23,10 @@ impl UserData for Output {
 	fn add_fields<F: UserDataFields<Self>>(fields: &mut F) {
 		cached_field!(fields, status, |_, me| Ok(Status::new(me.inner.status)));
 		cached_field_mut!(fields, stdout, |lua, me| {
-			lua.create_external_string(mem::take(&mut me.inner.stdout))
+			lua.create_string(mem::take(&mut me.inner.stdout))
 		});
 		cached_field_mut!(fields, stderr, |lua, me| {
-			lua.create_external_string(mem::take(&mut me.inner.stderr))
+			lua.create_string(mem::take(&mut me.inner.stderr))
 		});
 	}
 }
diff --git a/yazi-binding/src/url.rs b/yazi-binding/src/url.rs
index dba0b1b3..e01ee316 100644
--- a/yazi-binding/src/url.rs
+++ b/yazi-binding/src/url.rs
@@ -213,7 +213,7 @@ impl UserData for Url {
 			lua.create_string(me.to_strand().encoded_bytes())
 		});
 		methods.add_meta_method(MetaMethod::Concat, |lua, lhs, rhs: mlua::String| {
-			lua.create_external_string([lhs.to_strand().encoded_bytes(), &rhs.as_bytes()].concat())
+			lua.create_string([lhs.to_strand().encoded_bytes(), &rhs.as_bytes()].concat())
 		});
 	}
 }
diff --git a/yazi-dds/src/sendable.rs b/yazi-dds/src/sendable.rs
index 860213e7..c960fd1a 100644
--- a/yazi-dds/src/sendable.rs
+++ b/yazi-dds/src/sendable.rs
@@ -72,7 +72,7 @@ impl Sendable {
 
 	pub fn data_to_value(lua: &Lua, data: Data) -> mlua::Result<Value> {
 		Ok(match data {
-			Data::String(Cow::Owned(s)) => Value::String(lua.create_external_string(s)?),
+			Data::String(Cow::Owned(s)) => Value::String(lua.create_string(s)?),
 			Data::List(l) => {
 				let mut vec = Vec::with_capacity(l.len());
 				for v in l.into_iter() {
@@ -90,7 +90,7 @@ impl Sendable {
 			}
 			Data::Url(u) => yazi_binding::Url::new(u).into_lua(lua)?,
 			Data::Path(u) => yazi_binding::Path::new(u).into_lua(lua)?,
-			Data::Bytes(b) => Value::String(lua.create_external_string(b)?),
+			Data::Bytes(b) => Value::String(lua.create_string(b)?),
 			Data::Any(b) => {
 				let mut b = b.into_any();
 				macro_rules! try_cast {
@@ -249,10 +249,10 @@ impl Sendable {
 
 	fn key_to_value(lua: &Lua, key: DataKey) -> mlua::Result<Value> {
 		match key {
-			DataKey::String(Cow::Owned(s)) => lua.create_external_string(s).map(Value::String),
+			DataKey::String(Cow::Owned(s)) => lua.create_string(s).map(Value::String),
 			DataKey::Url(u) => yazi_binding::Url::new(u).into_lua(lua),
 			DataKey::Path(u) => yazi_binding::Path::new(u).into_lua(lua),
-			DataKey::Bytes(b) => lua.create_external_string(b).map(Value::String),
+			DataKey::Bytes(b) => lua.create_string(b).map(Value::String),
 			_ => Self::key_to_value_ref(lua, &key),
 		}
 	}
diff --git a/yazi-plugin/src/elements/elements.rs b/yazi-plugin/src/elements/elements.rs
index 84e06003..0cbe1e12 100644
--- a/yazi-plugin/src/elements/elements.rs
+++ b/yazi-plugin/src/elements/elements.rs
@@ -98,7 +98,7 @@ pub(super) fn printable(lua: &Lua) -> mlua::Result<Value> {
 	let f = lua.create_function(|lua, s: mlua::String| {
 		Ok(match replace_to_printable(&s.as_bytes(), false, 1, true) {
 			Cow::Borrowed(_) => s,
-			Cow::Owned(new) => lua.create_external_string(new)?,
+			Cow::Owned(new) => lua.create_string(new)?,
 		})
 	})?;
 
@@ -194,7 +194,7 @@ pub(super) fn truncate(lua: &Lua) -> mlua::Result<Value> {
 				"…".bytes().chain(lossy[idx + len..].bytes()).collect()
 			}
 		};
-		lua.create_external_string(result)
+		lua.create_string(result)
 	})?;
 
 	f.into_lua(lua)
diff --git a/yazi-plugin/src/utils/text.rs b/yazi-plugin/src/utils/text.rs
index 8e3e8e95..fcf40fef 100644
--- a/yazi-plugin/src/utils/text.rs
+++ b/yazi-plugin/src/utils/text.rs
@@ -19,7 +19,7 @@ impl Utils {
 				Some(false) => yazi_shared::shell::windows::escape_os_bytes(&b),
 				None => yazi_shared::shell::escape_os_bytes(&b),
 			};
-			lua.create_external_string(s)
+			lua.create_string(s)
 		})
 	}
 
@@ -29,7 +29,7 @@ impl Utils {
 				CLIPBOARD.set(text).await;
 				Ok(None)
 			} else {
-				Some(lua.create_external_string(CLIPBOARD.get().await)).transpose()
+				Some(lua.create_string(CLIPBOARD.get().await)).transpose()
 			}
 		})
 	}
