--- a/src/bzfs/bzfs.cxx	2018-09-27 08:34:45.781489776 +0200
+++ b/src/bzfs/bzfs.cxx	2018-09-27 08:35:47.688492106 +0200
@@ -1312,7 +1312,7 @@
     // accept incoming connection on our well known port
     struct sockaddr_in clientAddr;
     AddrLen addr_len = sizeof(clientAddr);
-    int fd = accept(wksSocket, (struct sockaddr*)&clientAddr, &addr_len);
+    int fd = accept(wksSocket, (struct sockaddr*)&clientAddr, (socklen_t*)&addr_len);
     if (fd == -1)
     {
         nerror("accepting on wks");
