--- src/bzfs/bzfs.cxx-orig	2018-01-22 14:58:45.381583835 -0600
+++ src/bzfs/bzfs.cxx	2018-01-22 14:58:55.584456281 -0600
@@ -1260,7 +1260,7 @@
   // accept incoming connection on our well known port
   struct sockaddr_in clientAddr;
   AddrLen addr_len = sizeof(clientAddr);
-  int fd = accept(wksSocket, (struct sockaddr*)&clientAddr, &addr_len);
+  int fd = accept(wksSocket, (struct sockaddr*)&clientAddr, (socklen_t*)&addr_len);
   if (fd == -1) {
     nerror("accepting on wks");
     return;
