--- zlibrary/ui/src/qt4/tree/QtWaitingSpinner.cpp.orig	2018-03-17 07:35:31.000000000 +0000
+++ zlibrary/ui/src/qt4/tree/QtWaitingSpinner.cpp	2018-03-17 07:37:07.801751090 +0000
@@ -86,7 +86,7 @@
 }
 
 void QtWaitingSpinner::setRoundness(qreal roundness) {
-	myRoundness = std::max(0.0, std::min(100.0, roundness));
+	myRoundness = std::max((qreal)0.0, std::min((qreal)100.0, roundness));
 }
 
 void QtWaitingSpinner::setColor(QColor color) {
@@ -148,7 +148,7 @@
 	qreal alphaDiff = color.alphaF() - (qreal)minAlphaF;
 	qreal gradation = alphaDiff / (qreal)(distanceThreshold + 1);
 	qreal resultAlpha = color.alphaF() - gradation * distance;
-	resultAlpha = std::min(1.0, std::max(0.0, resultAlpha)); //if alpha is out of bound, force it to bounds
+	resultAlpha = std::min((qreal)1.0, std::max((qreal)0.0, (qreal)resultAlpha)); //if alpha is out of bound, force it to bounds
 	color.setAlphaF(resultAlpha);
 	return color;
 }
