diff --git a/makefiles/arch/desktop.mk b/makefiles/arch/desktop.mk
index f7f696f..e96ee33 100644
--- a/makefiles/arch/desktop.mk
+++ b/makefiles/arch/desktop.mk
@@ -7,10 +7,8 @@ IMAGEDIR = $(INSTALLDIR)/share/pixmaps
 APPIMAGEDIR = $(INSTALLDIR)/share/pixmaps/%APPLICATION_NAME%
 
 
-CCACHE = $(shell if which ccache > /dev/null; then echo "ccache"; fi) #if ccache is not installed, do not use it
-CC = $(CCACHE) gcc
-AR = ar rsu
-LD = g++
+AR := ${AR} rsu
+LD := ${CXX}
 
 CFLAGS += -pipe -fno-exceptions -Wall -Wno-ctor-dtor-privacy -W -DLIBICONV_PLUG -fPIC
 LDFLAGS =

