--- kernel/conftest.sh.orig
+++ kernel/conftest.sh
@@ -160,6 +160,7 @@ build_cflags() {

     if [ "$ARCH" = "i386" -o "$ARCH" = "x86_64" ]; then
         CFLAGS="$CFLAGS -I$SOURCES/arch/x86/include"
+        CFLAGS="$CFLAGS -I$SOURCES/arch/x86/include/uapi"
         CFLAGS="$CFLAGS -I$OUTPUT/arch/x86/include/generated"
         CFLAGS="$CFLAGS -I$OUTPUT/arch/x86/include/generated/uapi"
     elif [ "$ARCH" = "arm" ]; then
