--- a/pyproject.toml	2025-02-27 00:16:18.000000000 +0100
+++ b/pyproject.toml	2025-03-17 15:20:09.011808694 +0100
@@ -16,7 +16,7 @@
   "img2pdf>=0.5",
   "packaging>=20",
   "pdfminer.six>=20220319",
-  "pi-heif",                # Heif image format - maintainers: if this is removed, it will NOT break
+  "pillow-heif",                # Heif image format - maintainers: if this is removed, it will NOT break
   "pikepdf>=8.10.1",
   "Pillow>=10.0.1",
   "pluggy>=1",
--- a/src/ocrmypdf/_pipeline.py	2025-02-27 00:16:18.000000000 +0100
+++ b/src/ocrmypdf/_pipeline.py	2025-03-17 15:21:10.155848376 +0100
@@ -42,7 +42,7 @@
 from ocrmypdf.pluginspec import OrientationConfidence
 
 try:
-    from pi_heif import register_heif_opener
+    from pillow_heif import register_heif_opener
 except ImportError:
 
     def register_heif_opener():
