--- a/pyproject.toml    2020-02-02 01:00:00.000000000 +0100
+++ b/pyproject.toml    2025-11-22 17:00:55.440822910 +0100
@@ -16,7 +16,7 @@
   "img2pdf>=0.5",
   "packaging>=20",
   "pdfminer.six>=20220319",
-  "pi-heif",                # Heif image format - maintainers: if this is removed, it will NOT break
+  "pillow-heif",                # Heif image format - maintainers: if this is removed, it will NOT break
   "pikepdf>=10",
   "Pillow>=10.0.1",
   "pluggy>=1",
--- a/src/ocrmypdf/_pipeline.py 2020-02-02 01:00:00.000000000 +0100
+++ b/src/ocrmypdf/_pipeline.py 2025-11-22 17:27:57.466003395 +0100
@@ -42,7 +42,7 @@
 from ocrmypdf.pluginspec import OrientationConfidence
 
 try:
-    from pi_heif import register_heif_opener
+    from pillow_heif import register_heif_opener
 except ImportError:
 
     def register_heif_opener():
