From a65f29c4cddccd22545746b077e6bd19d56e6bfb Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Piotr=20W=C3=B3jcik?= <chocimier@tlen.pl>
Date: Fri, 24 Jul 2020 23:07:44 +0200
Subject: [PATCH] Absolute paths can't match both cross and native use


diff --git include/armadillo_bits/config.hpp.cmake include/armadillo_bits/config.hpp.cmake
index be42243..a285e35 100644
--- a/include/armadillo_bits/config.hpp.cmake
+++ b/include/armadillo_bits/config.hpp.cmake
@@ -50,7 +50,7 @@
 #endif
 
 #if !defined(ARMA_SUPERLU_INCLUDE_DIR)
-#define ARMA_SUPERLU_INCLUDE_DIR ${ARMA_SUPERLU_INCLUDE_DIR}/
+#define ARMA_SUPERLU_INCLUDE_DIR ./
 //// If you're using SuperLU and want to explicitly include the SuperLU headers,
 //// uncomment the above define and specify the appropriate include directory.
 //// Make sure the directory has a trailing /
@@ -94,7 +94,7 @@
 //// ARMA_BLAS_LONG, ARMA_BLAS_LONG_LONG, ARMA_USE_FORTRAN_HIDDEN_ARGS
 
 #cmakedefine ARMA_USE_ATLAS
-#define ARMA_ATLAS_INCLUDE_DIR ${ARMA_ATLAS_INCLUDE_DIR}/
+#define ARMA_ATLAS_INCLUDE_DIR ./
 //// If you're using ATLAS and the compiler can't find cblas.h and/or clapack.h
 //// uncomment the above define and specify the appropriate include directory.
 //// Make sure the directory has a trailing /
@@ -140,7 +140,7 @@
   #undef  ARMA_USE_HDF5
   #define ARMA_USE_HDF5
   
-  #define ARMA_HDF5_INCLUDE_DIR ${ARMA_HDF5_INCLUDE_DIR}/
+  #define ARMA_HDF5_INCLUDE_DIR ./
 #endif
 
 #if !defined(ARMA_MAT_PREALLOC)
-- 
2.27.0

