diff --git a/hyprctl/src/main.cpp b/hyprctl/src/main.cpp
index d44a954..2d76056 100644
--- a/hyprctl/src/main.cpp
+++ b/hyprctl/src/main.cpp
@@ -271,7 +271,7 @@ int requestIPC(std::string_view filename, std::string_view arg) {
         return 1;
     }
 
-    std::string socketPath = getRuntimeDir() + "/" + instanceSignature + "/" + filename;
+    std::string socketPath = getRuntimeDir() + "/" + instanceSignature + "/" + std::string{filename};
 
     strncpy(serverAddress.sun_path, socketPath.c_str(), sizeof(serverAddress.sun_path) - 1);
