--- a/src/helpers/Monitor.cpp
+++ b/src/helpers/Monitor.cpp
@@ -693,7 +693,7 @@ void CMonitor::onConnect(bool noRule) {
         std::ranges::sort(sortedModes, sortFunc);
         if (sortedModes.size() > 3)
             sortedModes.erase(sortedModes.begin() + 3, sortedModes.end());
-        requestedModes.insert_range(requestedModes.end(), sortedModes | std::views::reverse);
+        requestedModes.insert(requestedModes.end(), sortedModes.rbegin(), sortedModes.rend());
     };
 
     // last fallback is always preferred mode
