diff --git src/Spago/Prelude.hs src/Spago/Prelude.hs
index ba18578..debc651 100644
--- src/Spago/Prelude.hs
+++ src/Spago/Prelude.hs
@@ -1,5 +1,5 @@
 module Spago.Prelude
-  ( 
+  (
   -- * Basic exports
     module X
   , Proxy(..)
@@ -12,7 +12,7 @@ module Spago.Prelude
   , headMay
   , lastMay
   , empty
-  
+
   -- * Logging, errors, printing, etc
   , Pretty
   , pretty
@@ -24,7 +24,6 @@ module Spago.Prelude
 
   -- * Lens
   , (</>)
-  , (^..)
   , transformMOf
 
   -- * Files and directories
@@ -95,7 +94,6 @@ import           Data.Sequence                         (Seq (..))
 import           Data.Text.Prettyprint.Doc             (Pretty)
 import           Data.Text.IO.Utf8                     (readFile, writeFile)
 import           Dhall.Optics                          (transformMOf)
-import           Lens.Family                           ((^..))
 import           RIO                                   as X hiding (FilePath, first, force, second)
 import           RIO.Orphans                           as X
 import           Safe                                  (headMay, lastMay)
@@ -166,7 +164,7 @@ cptree from to' = Turtle.cptree (Turtle.decodeString from) (Turtle.decodeString
 
 
 -- | Code from: https://github.com/dhall-lang/dhall-haskell/blob/d8f2787745bb9567a4542973f15e807323de4a1a/dhall/src/Dhall/Import.hs#L578
-assertDirectory :: (MonadIO m, HasLogFunc env, MonadReader env m) => FilePath.FilePath -> m ()
+assertDirectory :: (MonadIO m, MonadThrow m, HasLogFunc env, MonadReader env m) => FilePath.FilePath -> m ()
 assertDirectory directory = do
   let private = transform Directory.emptyPermissions
         where
@@ -188,7 +186,6 @@ assertDirectory directory = do
       unless (accessible permissions) $ do
         die [ "Directory " <> displayShow directory <> " is not accessible. " <> displayShow permissions ]
     else do
-      logDebug $ "Directory " <> displayShow directory <> " does not exist, creating..."
       assertDirectory (FilePath.takeDirectory directory)
 
       Directory.createDirectory directory
@@ -198,7 +195,7 @@ assertDirectory directory = do
 
 -- | Release tag for the `purescript-docs-search` app.
 docsSearchVersion :: Text
-docsSearchVersion = "v0.0.10"
+docsSearchVersion = "v0.0.8"
 
 
 githubTokenEnvVar :: IsString t => t
