--- qtwebengine/src/3rdparty/chromium/build/compiler_version.py	2016-05-26 14:58:54.000000000 +0200
+++ qtwebengine/src/3rdparty/chromium/build/compiler_version.py	2016-08-08 20:46:31.311068824 +0200
@@ -41,7 +41,9 @@
   """Look up an environment variable from a possible list of variable names."""
   for var in var_list:
     if var in os.environ:
-      return os.environ[var]
+      val = os.environ[var]
+      if val != "":
+        return val
   return default
 
 
--- qtwebengine/src/3rdparty/ninja/configure.py	2016-05-26 14:58:54.000000000 +0200
+++ qtwebengine/src/3rdparty/ninja/configure.py	2016-08-08 20:58:43.594054074 +0200
@@ -245,6 +245,8 @@
 n.newline()
 
 CXX = configure_env.get('CXX', 'g++')
+if CXX == '':
+    CXX = 'g++'
 objext = '.o'
 if platform.is_msvc():
     CXX = 'cl'
@@ -270,9 +272,12 @@
 n.variable('builddir', 'build')
 n.variable('cxx', CXX)
 if platform.is_msvc():
-    n.variable('ar', 'link')
+    AR = 'link'
 else:
-    n.variable('ar', configure_env.get('AR', 'ar'))
+    AR = configure_env.get('AR', 'ar')
+    if AR == '':
+        AR = 'ar'
+n.variable('ar', AR)
 
 if platform.is_msvc():
     cflags = ['/showIncludes',
