--- qtwebengine/tools/qmake/mkspecs/features/functions.prf	2016-05-26 20:01:25.000000000 +0200
+++ qtwebengine/tools/qmake/mkspecs/features/functions.prf	2016-08-07 23:17:46.870626280 +0200
@@ -63,11 +63,7 @@
 }
 
 defineTest(isGCCVersionSupported) {
-  # The below will work for gcc 4.7 and up and also match gcc 5
-  greaterThan(QT_GCC_MINOR_VERSION, 6):return(true)
-  greaterThan(QT_GCC_MAJOR_VERSION, 4):return(true)
-  skipBuild("Using gcc version "$$QT_GCC_MAJOR_VERSION"."$$QT_GCC_MINOR_VERSION", but at least gcc version 4.7 is required to build Qt WebEngine.")
-  return(false)
+  return(true)
 }
 
 defineTest(isQMLTestSupportApiEnabled) {
