diff -Nru indicator-application-12.10.1daily13.01.11.orig/configure.ac indicator-application-12.10.1daily13.01.11/configure.ac
--- indicator-application-12.10.1daily13.01.11.orig/configure.ac	2013-01-12 10:22:02.405959692 -0500
+++ indicator-application-12.10.1daily13.01.11/configure.ac	2013-01-12 10:22:12.069242219 -0500
@@ -4,7 +4,7 @@
 
 AC_PREREQ(2.53)
 
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 AM_INIT_AUTOMAKE(indicator-application, 12.10.0)
 
 AM_MAINTAINER_MODE
