diff -p1ru a/configure.ac b/configure.ac
--- a/configure.ac	2024-07-22 17:34:53.413000000 +0800
+++ b/configure.ac	2024-07-22 17:43:07.897000000 +0800
@@ -34,3 +34,2 @@ GTK_REQUIRED_VERSION=3.0.0
 GIO_REQUIRED_VERSION=2.26
-INDICATOR_REQUIRED_VERSION=0.3.5
 DBUSMENUGTK_REQUIRED_VERSION=0.5.90
@@ -39,4 +38,2 @@ DBUS_GLIB_REQUIRED_VERSION=0.82
 
-LIBAPPINDICATOR_REQUIRED_VERSION=0.2.9
-
 PKG_CHECK_MODULES(INDICATOR,  glib-2.0 >= $GLIB_REQUIRED_VERSION
@@ -44,3 +41,3 @@ PKG_CHECK_MODULES(INDICATOR,  glib-2.0 >
                               gtk+-3.0 >= $GTK_REQUIRED_VERSION
-                              indicator3-0.4 >= $INDICATOR_REQUIRED_VERSION
+                              ayatana-indicator3-0.4
                               json-glib-1.0 >= $JSON_GLIB_REQUIRED_VERSION
@@ -48,3 +45,3 @@ PKG_CHECK_MODULES(INDICATOR,  glib-2.0 >
                               dbusmenu-gtk3-0.4 >= $DBUSMENUGTK_REQUIRED_VERSION)
-PKG_CHECK_MODULES(APPINDICATOR, appindicator3-0.1 >= $LIBAPPINDICATOR_REQUIRED_VERSION)
+PKG_CHECK_MODULES(APPINDICATOR, ayatana-appindicator3-0.1)
 
@@ -87,4 +84,4 @@ elif test "x$with_gtk" = x2; then
 else
-	INDICATORDIR=`$PKG_CONFIG --variable=indicatordir indicator3-0.4`
-	INDICATORICONSDIR=`$PKG_CONFIG --variable=iconsdir indicator3-0.4`
+	INDICATORDIR=`$PKG_CONFIG --variable=indicatordir ayatana-indicator3-0.4`
+	INDICATORICONSDIR=`$PKG_CONFIG --variable=iconsdir ayatana-indicator3-0.4`
 fi
diff -p1ru a/src/application-service-appstore.c b/src/application-service-appstore.c
--- a/src/application-service-appstore.c	2019-03-08 23:35:42.000000000 +0800
+++ b/src/application-service-appstore.c	2024-07-22 17:48:27.569000000 +0800
@@ -26,5 +26,5 @@ with this program.  If not, see <http://
 
-#include <libindicator/indicator-object.h>
-#include "libappindicator/app-indicator.h"
-#include "libappindicator/app-indicator-enum-types.h"
+#include <libayatana-indicator/indicator-object.h>
+#include "libayatana-appindicator/app-indicator.h"
+#include "libayatana-appindicator/app-indicator-enum-types.h"
 #include "application-service-appstore.h"
diff -p1ru a/src/generate-id.h b/src/generate-id.h
--- a/src/generate-id.h	2019-03-08 23:35:42.000000000 +0800
+++ b/src/generate-id.h	2024-07-22 17:51:52.750000000 +0800
@@ -25,3 +25,3 @@ with this program.  If not, see <http://
 #include <glib.h>
-#include "libappindicator/app-indicator.h"
+#include "libayatana-appindicator/app-indicator.h"
 
diff -p1ru a/src/indicator-application.c b/src/indicator-application.c
--- a/src/indicator-application.c	2019-03-08 23:35:42.000000000 +0800
+++ b/src/indicator-application.c	2024-07-22 17:55:31.850000000 +0800
@@ -37,5 +37,5 @@ with this program.  If not, see <http://
 /* Indicator Stuff */
-#include <libindicator/indicator.h>
-#include <libindicator/indicator-object.h>
-#include <libindicator/indicator-image-helper.h>
+#include <libayatana-indicator/indicator.h>
+#include <libayatana-indicator/indicator-object.h>
+#include <libayatana-indicator/indicator-image-helper.h>
 
