diff --git a/Makefile b/Makefile
index 2d9cb79..138d221 100644
--- a/Makefile
+++ b/Makefile
@@ -108,8 +108,6 @@ trace: export EXE_FLAGS     += -g3
 trace: compile
 
 test: OBJDIR                 = $(TESTDIR)
-test: export CFLAGS         += -O2 -DLSP_TESTING -DLSP_TRACE -g3 -fstack-protector
-test: export CXXFLAGS       += -O2 -DLSP_TESTING -DLSP_TRACE -g3 -fstack-protector
 test: export EXE_TEST_FLAGS += -g3
 test: export MAKE_OPTS      += LSP_TESTING=1
 test: export BUILD_MODULES   = jack
