--- a/modules/lsp-runtime-lib/src/main/ipc/Library.cpp	2026-08-19 14:02:04.349078479 +0100
+++ b/modules/lsp-runtime-lib/src/main/ipc/Library.cpp	2026-08-19 14:03:13.887642314 +0100
@@ -72,14 +72,13 @@
                 const char *str  = path->get_utf8();
                 if (str == NULL)
                     return STATUS_NO_MEM;
-
-            #ifdef _GNU_SOURCE
+            #ifdef __GLIBC__
                 void * const handle    = (new_namespace) ?
                     ::dlmopen(LM_ID_NEWLM, str, RTLD_NOW) :
                     ::dlopen(str, RTLD_NOW);
             #else
                 void * const handle    = ::dlopen(str, RTLD_NOW);
-            #endif /* _GNU_SOURCE */
+            #endif /* __GLIBC__ */
                 if (handle == NULL)
                 {
                     lsp_warn("Error loading module %s: %s", path->get_native(), ::dlerror());
