diff --git a/Makefile b/Makefile
index c861fef..6668520 100644
--- a/Makefile
+++ b/Makefile
@@ -15,16 +15,16 @@ THIRDPARTYDIR=$(HARESRCDIR)/third-party
 all: himitsud himitsu-store hiq hiprompt-tty docs
 
 himitsud:
-	hare build -o $@ cmd/$@/
+	hare build $(HAREFLAGS) -o $@ cmd/$@/
 
 himitsu-store:
-	hare build -o $@ cmd/$@/
+	hare build $(HAREFLAGS) -o $@ cmd/$@/
 
 hiprompt-tty:
-	hare build -o $@ cmd/$@/
+	hare build $(HAREFLAGS) -o $@ cmd/$@/
 
 hiq:
-	hare build -o $@ cmd/$@/
+	hare build $(HAREFLAGS) -o $@ cmd/$@/
 
 check:
 	hare test
