From 90afb407bfc3c5550e7192935993c35a7447f6ae Mon Sep 17 00:00:00 2001
From: Nathan Owens <ndowens04@gmail.com>
Date: Wed, 8 Jan 2020 18:55:59 -0600
Subject: [PATCH] Add pkgconfig

---
 libexecinfo.pc | 5 +++++
 1 file changed, 5 insertions(+)
 create mode 100644 libexecinfo.pc

diff --git a/libexecinfo.pc b/libexecinfo.pc
new file mode 100644
index 0000000..dc61a97
--- /dev/null
+++ b/libexecinfo.pc
@@ -0,0 +1,10 @@
+prefix=/usr
+exec_prefix=${prefix}
+includedir=${prefix}/include
+libdir=${exec_prefix}/lib
+
+Name: libexecinfo
+Description: libexecinfo library
+Version: version
+CFlags: -I${includedir}
+Libs: -L${libdir} -lexecinfo
