--- testing/unit/test_statistics.py
+++ testing/unit/test_statistics.py
@@ -59,31 +59,6 @@ class StatsObjTest(UnitTestCase):
         s1 = StatsDeltaProcess()
         assert s1.get_stat('SourceFiles') == 0
 
-    def test_get_stats_string(self):
-        """Test conversion of stat object into string"""
-        s = StatsObj()
-        stats_string = s.get_stats_string()
-        assert stats_string == "", stats_string
-
-        self.set_obj(s)
-        stats_string = s.get_stats_string()
-        assert stats_string == """\
-StartTime 13.00 (Wed Dec 31 18:00:13 1969)
-EndTime 14.00 (Wed Dec 31 18:00:14 1969)
-ElapsedTime 1.00 (1 second)
-SourceFiles 1
-SourceFileSize 2 (2 bytes)
-NewFiles 3
-NewFileSize 4 (4 bytes)
-DeletedFiles 5
-ChangedFiles 7
-ChangedFileSize 8 (8 bytes)
-ChangedDeltaSize 9 (9 bytes)
-DeltaEntries 10
-RawDeltaSize 11 (11 bytes)
-TotalDestinationSizeChange 12 (12 bytes)
-""", "'%s'" % stats_string
-
     def test_line_string(self):
         """Test conversion to a single line"""
         s = StatsObj()
