--- a/CMakeLists.txt	2020-11-22 21:44:04.000000000 +0100
+++ -	2024-08-04 12:14:48.905942371 +0200
@@ -87,7 +87,7 @@
 
 
 include(CMakePackageConfigHelpers)
-set(CONFIG_PACKAGE_LOCATION "${CMAKE_INSTALL_DATADIR}/Bcg729/cmake")
+set(CONFIG_PACKAGE_LOCATION "${CMAKE_INSTALL_LIBDIR}/cmake/Bcg729/")
 write_basic_package_version_file(
 	"${CMAKE_CURRENT_BINARY_DIR}/Bcg729ConfigVersion.cmake"
 	VERSION ${PACKAGE_VERSION}
