--- steam.orig	2015-03-22 11:22:16.000197500 +0100
+++ steam	2015-03-22 11:23:10.780279019 +0100
@@ -183,6 +183,13 @@ if ! check_bootstrap "$LAUNCHSTEAMDIR";
 	exit 1
 fi
 
+# -- void --
+# Preload lib{crypto,ssl} from openssl, our libressl seems to be incompatible.
+# -- void --
+STEAM32_BSTRAP=${DEFAULTSTEAMDIR}/ubuntu12_32/steam-runtime/i386/lib/i386-linux-gnu
+STEAM64_BSTRAP=${DEFAULTSTEAMDIR}/ubuntu12_32/steam-runtime/amd64/lib/x86_64-linux-gnu
+export LD_PRELOAD="$STEAM32_BSTRAP/libcrypto.so.1.0.0:$STEAM32_BSTRAP/libssl.so.1.0.0"
+
 # go to the install directory and run the client
 cp "$LAUNCHSTEAMBOOTSTRAPFILE" "$LAUNCHSTEAMDIR/bootstrap.tar.xz"
 cd "$LAUNCHSTEAMDIR"
