--- steam.orig	2016-02-16 08:28:31.191216344 +0100
+++ steam	2016-02-16 08:30:59.849711436 +0100
@@ -183,6 +183,13 @@ if ! check_bootstrap "$LAUNCHSTEAMDIR";
 	exit 1
 fi
 
+# --*-- void --*--
+if ! xbps-uhelper version nvidia &>/dev/null; then
+	STEAM_RUNTIME=${DEFAULTSTEAMDIR}/ubuntu12_32/steam-runtime
+	export LD_PRELOAD="$STEAM_RUNTIME/i386/lib/i386-linux-gnu/libcrypto.so.1.0.0:$STEAM_RUNTIME/i386/lib/i386-linux-gnu/libssl.so.1.0.0"
+fi
+# --*-- void --*--
+
 # go to the install directory and run the client
 cp "$LAUNCHSTEAMBOOTSTRAPFILE" "$LAUNCHSTEAMDIR/bootstrap.tar.xz"
 cd "$LAUNCHSTEAMDIR"
