--- steam.orig 2016-05-07 13:51:39.465996146 +0200
+++ steam      2016-05-07 13:52:51.155112972 +0200
@@ -183,6 +183,10 @@
       exit 1
 fi
 
+# --*-- void --*--
+export  LD_PRELOAD='/usr/$LIB/libstdc++.so.6 /usr/$LIB/libgcc_s.so.1 /usr/$LIB/libxcb.so.1'
+# --*-- void --*--
+
 # go to the install directory and run the client
 cp "$LAUNCHSTEAMBOOTSTRAPFILE" "$LAUNCHSTEAMDIR/bootstrap.tar.xz"
 cd "$LAUNCHSTEAMDIR"
@@ -190,7 +198,7 @@
       mkdir -p --mode=777 /tmp/dumps
     if [[ -f "/tmp/dumps/${USER}_stdout.txt" ]]; then
         rm -f "/tmp/dumps/${USER}_stdout.txt"
-    fi 
+    fi
       exec "$LAUNCHSTEAMDIR/$STEAMBOOTSTRAP" "$@" 2>&1 | tee "/tmp/dumps/${USER}_stdout.txt"
 else
       exec "$LAUNCHSTEAMDIR/$STEAMBOOTSTRAP" "$@"

