diff --git a/src/configdialog.ui b/src/configdialog.ui
index e42a323f7..8730d4ca7 100644
--- a/src/configdialog.ui
+++ b/src/configdialog.ui
@@ -495,6 +495,9 @@
                </item>
                <item>
                 <widget class="QGroupBox" name="groupBox_Update">
+                 <property name="enabled">
+                  <bool>false</bool>
+                 </property>
                  <property name="title">
                   <string>Update</string>
                  </property>
diff --git a/src/configmanager.cpp b/src/configmanager.cpp
index 4f1e41670..4627897cd 100644
--- a/src/configmanager.cpp
+++ b/src/configmanager.cpp
@@ -512,10 +512,12 @@ ConfigManager::ConfigManager(QObject *parent): QObject (parent),
     registerOption("Macros/RepositoryURL", &URLmacroRepository, "https://api.github.com/repos/texstudio-org/texstudio-macro/contents/", nullptr);
 
 	//updates
+	/*
 	registerOption("Update/AutoCheck", &autoUpdateCheck, true, &pseudoDialog->checkBoxAutoUpdateCheck);
 	registerOption("Update/UpdateLevel", &updateLevel, 0, &pseudoDialog->comboBoxUpdateLevel);
 	registerOption("Update/AutoCheckInvervalDays", &autoUpdateCheckIntervalDays, 7, &pseudoDialog->spinBoxAutoUpdateCheckIntervalDays);
 	registerOption("Update/LastCheck", &lastUpdateCheck, QDateTime());
+	*/
 
 	//editor
 	registerOption("Editor/WordWrapMode", &editorConfig->wordwrap, 1, &pseudoDialog->comboBoxLineWrap);
