--- a/setup.py
+++ b/setup.py
@@ -1,16 +1,9 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 from distutils.core import setup
 from distutils.command import install as distutils_install
 from iotop.version import VERSION
 
-# Dirty hack to make setup.py install the iotop script to sbin/ instead of bin/
-# while still honoring the choice of installing into local/ or not.
-if hasattr(distutils_install, 'INSTALL_SCHEMES'):
-    for d in distutils_install.INSTALL_SCHEMES.itervalues():
-        if d.get('scripts', '').endswith('/bin'):
-            d['scripts'] = d['scripts'][:-len('/bin')] + '/sbin'
-
 setup(name='iotop',
       version=VERSION,
       description='Per process I/O bandwidth monitor',
