--- a/hardinfo.h	2023-04-14 16:50:28.324202873 -0400
+++ b/hardinfo.h	2023-04-14 16:51:11.470129493 -0400
@@ -64,9 +64,9 @@
 };
 
 /* String utility functions */
-inline void  remove_quotes(gchar *str);
-inline char *strend(gchar *str, gchar chr);
-inline void  remove_linefeed(gchar *str);
+void  remove_quotes(gchar *str);
+char *strend(gchar *str, gchar chr);
+void  remove_linefeed(gchar *str);
 gchar       *strreplace(gchar *string, gchar *replace, gchar new_char);
 
 /* Widget utility functions */
@@ -91,7 +91,7 @@
 
 
 gchar	     *find_program(gchar *program_name);
-inline gchar *size_human_readable(gfloat size);
+gchar        *size_human_readable(gfloat size);
 void          nonblock_sleep(guint msec);
 void          open_url(gchar *url);
 GSList	     *modules_load_selected(void);
--- a/util.c 16:50:36.611380820 -0400
+++ b/util.c	2023-04-14 16:51:25.031420828 -0400
@@ -111,7 +111,7 @@
 			   plural(hours), minutes, plural(minutes));
 }
 
-inline gchar *size_human_readable(gfloat size)
+gchar *size_human_readable(gfloat size)
 {
     if (size < KiB)
 	return g_strdup_printf("%.1f B", size);
@@ -123,7 +123,7 @@
     return g_strdup_printf("%.1f GiB", size / GiB);
 }
 
-inline char *strend(gchar * str, gchar chr)
+char *strend(gchar * str, gchar chr)
 {
     if (!str)
 	return NULL;
@@ -135,7 +135,7 @@
     return str;
 }
 
-inline void remove_quotes(gchar * str)
+void remove_quotes(gchar * str)
 {
     if (!str)
 	return;
@@ -146,7 +146,7 @@
     strend(str, '"');
 }
 
-inline void remove_linefeed(gchar * str)
+void remove_linefeed(gchar * str)
 {
     strend(str, '\n');
 }
