diff --git hardinfo.h hardinfo.h
index 9f52f63..aeb84d9 100644
--- a/hardinfo.h
+++ b/hardinfo.h
@@ -64,8 +64,8 @@ struct _ModuleAbout {
 };
 
 /* String utility functions */
-inline void  remove_quotes(gchar *str);
-inline char *strend(gchar *str, gchar chr);
+void  remove_quotes(gchar *str);
+char *strend(gchar *str, gchar chr);
 inline void  remove_linefeed(gchar *str);
 gchar       *strreplace(gchar *string, gchar *replace, gchar new_char);
 
@@ -91,7 +91,7 @@ gpointer __idle_free(gpointer ptr, gchar *f, gint l);
 
 
 gchar	     *find_program(gchar *program_name);
-inline gchar *size_human_readable(gfloat size);
+gchar        *size_human_readable(gfloat size);
 void          nonblock_sleep(guint msec);
 void          open_url(gchar *url);
 GSList	     *modules_load_selected(void);
diff --git util.c util.c
index 266ce4e..e6c81fd 100644
--- a/util.c
+++ b/util.c
@@ -111,7 +111,7 @@ gchar *seconds_to_string(unsigned int seconds)
 			   plural(hours), minutes, plural(minutes));
 }
 
-inline gchar *size_human_readable(gfloat size)
+gchar *size_human_readable(gfloat size)
 {
     if (size < KiB)
 	return g_strdup_printf("%.1f B", size);
@@ -123,7 +123,7 @@ inline gchar *size_human_readable(gfloat size)
     return g_strdup_printf("%.1f GiB", size / GiB);
 }
 
-inline char *strend(gchar * str, gchar chr)
+char *strend(gchar * str, gchar chr)
 {
     if (!str)
 	return NULL;
@@ -135,7 +135,7 @@ inline char *strend(gchar * str, gchar chr)
     return str;
 }
 
-inline void remove_quotes(gchar * str)
+void remove_quotes(gchar * str)
 {
     if (!str)
 	return;

