From f11d5624be68e95e9c2a8c7779b182ef97b7973c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Piotr=20W=C3=B3jcik?= <chocimier@tlen.pl>
Date: Fri, 30 Nov 2018 11:29:13 +0100
Subject: [PATCH] Fix inline strend


diff --git hardinfo.h hardinfo.h
index 9f52f63..6c7044d 100644
--- hardinfo.h
+++ hardinfo.h
@@ -65,7 +65,7 @@ struct _ModuleAbout {
 
 /* String utility functions */
 inline void  remove_quotes(gchar *str);
-inline char *strend(gchar *str, gchar chr);
+char *strend(gchar *str, gchar chr);
 inline void  remove_linefeed(gchar *str);
 gchar       *strreplace(gchar *string, gchar *replace, gchar new_char);
 
diff --git util.c util.c
index 266ce4e..068889f 100644
--- util.c
+++ util.c
@@ -123,7 +123,7 @@ inline gchar *size_human_readable(gfloat size)
     return g_strdup_printf("%.1f GiB", size / GiB);
 }
 
-inline char *strend(gchar * str, gchar chr)
+char *strend(gchar * str, gchar chr)
 {
     if (!str)
 	return NULL;
-- 
2.19.1

