--- build.sh.org	2013-01-11 16:52:50.064204113 +0000
+++ build.sh	2013-01-11 18:14:24.595441089 +0000
@@ -5,18 +5,18 @@
 
 # directory containing the ARM shared libraries (rootfs, lib/ of SD card)
 # specifically libEGL.so and libGLESv2.so
-ARM_LIBS=/opt/bcm-rootfs/opt/vc/lib
+ARM_LIBS=/opt/vc/lib
 SDL_LIB=lib
 
 # directory containing baseq3/ containing .pk3 files - baseq3 on CD
-BASEQ3_DIR="/home/${USER}/"
+BASEQ3_DIR="/opt/ioquake3-rpi/"
 
 # directory to find khronos linux make files (with include/ containing
 # headers! Make needs them.)
-INCLUDES="-I/opt/bcm-rootfs/opt/vc/include -I/opt/bcm-rootfs/opt/vc/include/interface/vcos/pthreads"
+INCLUDES="-I/opt/vc/include -I/opt/vc/include/interface/vcos/pthreads"
 
 # prefix of arm cross compiler installed
-CROSS_COMPILE=bcm2708-
+CROSS_COMPILE=@XBPS_CROSS_TRIPLET@
 
 # clean
 if [ $# -ge 1 ] && [ $1 = clean ]; then
