--- configure.ac.orig
+++ configure.ac
@@ -184,6 +184,7 @@
 	OSSH_CHECK_CFLAG_COMPILE([-Wall])
 	OSSH_CHECK_CFLAG_COMPILE([-Wpointer-arith])
 	OSSH_CHECK_CFLAG_COMPILE([-Wuninitialized])
+	OSSH_CHECK_CFLAG_COMPILE([-Wimplicit-function-declaration])
 	OSSH_CHECK_CFLAG_COMPILE([-Wsign-compare])
 	OSSH_CHECK_CFLAG_COMPILE([-Wformat-security])
 	OSSH_CHECK_CFLAG_COMPILE([-Wsizeof-pointer-memaccess])
@@ -1692,7 +1693,6 @@
 	endgrent \
 	err \
 	errx \
-	explicit_bzero \
 	fchmod \
 	fchown \
 	freeaddrinfo \
@@ -1732,7 +1732,6 @@
 	prctl \
 	pstat \
 	readpassphrase \
-	reallocarray \
 	recvmsg \
 	rresvport_af \
 	sendmsg \
@@ -1761,8 +1760,6 @@
 	strcasestr \
 	strdup \
 	strerror \
-	strlcat \
-	strlcpy \
 	strmode \
 	strnlen \
 	strnvis \
@@ -1774,7 +1771,6 @@
 	swap32 \
 	sysconf \
 	tcgetpgrp \
-	timingsafe_bcmp \
 	truncate \
 	unsetenv \
 	updwtmpx \
@@ -1785,6 +1781,13 @@
 	waitpid \
 	warn \
 ])
+AC_CHECK_DECLS([ \
+	explicit_bzero, \
+	strlcat, \
+	strlcpy, 
+	timingsafe_bcmp \
+], [], [], [[#include <string.h>]])
+AC_CHECK_DECLS([reallocarray], [], [], [[#include <stdlib.h>]])
 
 dnl Wide character support.
 AC_CHECK_FUNCS([mblen mbtowc nl_langinfo wcwidth])
