--- openssh-9.2p1.orig/channels.c
+++ openssh-9.2p1/channels.c
@@ -2558,7 +2558,7 @@ channel_handler(struct ssh *ssh, int tab
 			if (table == CHAN_PRE &&
 			    c->type == SSH_CHANNEL_OPEN &&
 			    c->inactive_deadline != 0 && c->lastused != 0 &&
-			    now >= c->lastused + c->inactive_deadline) {
+			    now >= c->lastused + (time_t)c->inactive_deadline) {
 				/* channel closed for inactivity */
 				verbose("channel %d: closing after %u seconds "
 				    "of inactivity", c->self,
