--- servconf.c.orig
+++ servconf.c
@@ -935,13 +935,6 @@
 	{ "no",				0 },
 	{ NULL, -1 }
 };
-static const struct multistate multistate_privsep[] = {
-	{ "yes",			PRIVSEP_NOSANDBOX },
-	{ "sandbox",			PRIVSEP_ON },
-	{ "nosandbox",			PRIVSEP_NOSANDBOX },
-	{ "no",				PRIVSEP_OFF },
-	{ NULL, -1 }
-};
 static const struct multistate multistate_tcpfwd[] = {
 	{ "yes",			FORWARD_ALLOW },
 	{ "all",			FORWARD_ALLOW },
--- configure.ac.orig
+++ configure.ac
@@ -399,7 +399,6 @@
 	sys/bitypes.h \
 	sys/bsdtty.h \
 	sys/capability.h \
-	sys/cdefs.h \
 	sys/dir.h \
 	sys/mman.h \
 	sys/ndir.h \
