--- configure.ac.orig	2015-10-23 14:18:17.868326070 +0200
+++ configure.ac	2015-10-23 14:20:17.736065420 +0200
@@ -294,20 +294,29 @@ dnl ====================================
 
 dnl strftime checks
 
-AC_TRY_RUN([#include <time.h>
-                int main ()
+AC_CACHE_CHECK([Define if strftime supports %E and %O modifiers], ac_cv_strftime_extensions,
+  [AC_TRY_RUN([
+   #include <string.h>
+   #include <time.h>
+    int
+    main (int argc, char **argv)
                 {
-                  char buf[100];
-                  struct tm tm = {0};
-                  tm.tm_year = 99;
-                  if (strftime(buf, 100, "%EY", &tm) == 4 &&
-                      strcmp (buf, "1999")==0)
+       struct tm tm;
+       char      buffer[16];
+       tm.tm_year = 81;
+       if (strftime (buffer, 16, "%EY", &tm) == 4 && strcmp (buffer, "1981") == 0)
                     return 0;
                   return 1;
                 }
             ],
-    AC_DEFINE(HAVE_STRFTIME_EXTENSION, 1, [Define if strftime supports %E and %O modifiers.])
+  ac_cv_strftime_extensions=yes,
+  ac_cv_strftime_extensions=no,
+  [AC_MSG_ERROR([cross-compiling, preset ac_cv_strftime_extensions=yes/no])]
+  )]
 )
+if test x"$ac_cv_strftime_extensions" = x"yes"; then
+    AC_DEFINE(HAVE_STRFTIME_EXTENSION, 1, [Define if strftime supports %E and %O modifiers.])
+fi
 
 dnl ===========================================================================
 
