--- ext/standard/config.m4.orig
+++ ext/standard/config.m4
@@ -314,7 +314,7 @@
   fi
   AC_DEFINE_UNQUOTED(PHP_MD5_CRYPT, $ac_result, [Whether the system supports MD5 salt])  
   
-  if test "$ac_cv_crypt_sha512" = "yes"; then
+  if test "$ac_cv_crypt_SHA512" = "yes"; then
     ac_result=1
     ac_crypt_sha512=1
   else
@@ -323,7 +323,7 @@
   fi
   AC_DEFINE_UNQUOTED(PHP_SHA512_CRYPT, $ac_result, [Whether the system supports SHA512 salt])
 
-  if test "$ac_cv_crypt_sha256" = "yes"; then
+  if test "$ac_cv_crypt_SHA256" = "yes"; then
     ac_result=1
     ac_crypt_sha256=1
   else
