diff --git ext/openssl/openssl.c ext/openssl/openssl.c.new
index 18c822ed67..4c0f5b916f 100644
--- ext/openssl/openssl.c
+++ ext/openssl/openssl.c.new
@@ -6901,7 +6901,7 @@ static void php_openssl_load_cipher_mode(struct php_openssl_cipher_mode *mode, c
 	int cipher_mode = EVP_CIPHER_mode(cipher_type);
 	memset(mode, 0, sizeof(struct php_openssl_cipher_mode));
 	switch (cipher_mode) {
-#if PHP_OPENSSL_API_VERSION >= 0x10100
+#if PHP_OPENSSL_API_VERSION >= 0x10100 && !defined(LIBRESSL_VERSION_NUMBER)
 		case EVP_CIPH_GCM_MODE:
 		case EVP_CIPH_OCB_MODE:
 		case EVP_CIPH_CCM_MODE:
