--- a/configure	2020-03-17 11:40:20.000000000 +0100
+++ b/configure	2020-04-12 19:51:57.216350534 +0200
@@ -6374,10 +6374,12 @@ IFS="- /.
     as_fn_error $? "Please note that Apache version >= 2.0.44 is required" "$LINENO" 5
   fi
 
+  APXS_PREFIX='$(INSTALL_ROOT)'/usr
   APXS_LIBEXECDIR='$(INSTALL_ROOT)'`$APXS -q LIBEXECDIR`
   if test -z `$APXS -q SYSCONFDIR`; then
     INSTALL_IT="\$(mkinstalldirs) '$APXS_LIBEXECDIR' && \
                  $APXS -S LIBEXECDIR='$APXS_LIBEXECDIR' \
+                       -S PREFIX='$APXS_PREFIX' \
                        -i -n php7"
   else
     APXS_SYSCONFDIR='$(INSTALL_ROOT)'`$APXS -q SYSCONFDIR`
@@ -6385,6 +6387,7 @@ IFS="- /.
                 \$(mkinstalldirs) '$APXS_SYSCONFDIR' && \
                  $APXS -S LIBEXECDIR='$APXS_LIBEXECDIR' \
                        -S SYSCONFDIR='$APXS_SYSCONFDIR' \
+                       -S PREFIX='$APXS_PREFIX' \
                        -i -a -n php7"
   fi
 
@@ -57170,7 +57173,7 @@ $as_echo_n "checking for pg_config... "
     fi
   done
 
-  if test -n "$PG_CONFIG"; then
+  if test -n "$PG_CONFIG" && test "x$cross_compiling" != "xyes"; then
     { $as_echo "$as_me:${as_lineno-$LINENO}: result: $PG_CONFIG" >&5
 $as_echo "$PG_CONFIG" >&6; }
     PGSQL_INCLUDE=`$PG_CONFIG --includedir`
@@ -58821,7 +58824,7 @@ $as_echo_n "checking for pg_config... "
     fi
   done
 
-  if test -n "$PG_CONFIG"; then
+  if test -n "$PG_CONFIG" && test "x$cross_compiling" != "xyes"; then
     { $as_echo "$as_me:${as_lineno-$LINENO}: result: $PG_CONFIG" >&5
 $as_echo "$PG_CONFIG" >&6; }
     PGSQL_INCLUDE=`$PG_CONFIG --includedir`
