--- ./php.ini-production.orig	2016-07-20 09:11:36.000000000 -0400
+++ ./php.ini-production	2016-07-21 18:15:02.471993953 -0400
@@ -291,7 +291,7 @@
 ; and below.  This directive makes most sense if used in a per-directory
 ; or per-virtualhost web server configuration file.
 ; http://php.net/open-basedir
-;open_basedir =
+open_basedir = /srv/www/:/home/:/tmp/:/usr/share/pear/:/usr/share/webapps/
 
 ; This directive allows you to disable certain functions for security reasons.
 ; It receives a comma-delimited list of function names.
@@ -696,7 +696,7 @@
 ;;;;;;;;;;;;;;;;;;;;;;;;;
 
 ; UNIX: "/path1:/path2"
-;include_path = ".:/php/includes"
+include_path = ".:/usr/share/pear"
 ;
 ; Windows: "\path1;\path2"
 ;include_path = ".;c:\php\includes"
@@ -719,7 +719,7 @@
 
 ; Directory in which the loadable extensions (modules) reside.
 ; http://php.net/extension-dir
-; extension_dir = "./"
+extension_dir = "/usr/lib/php/modules/"
 ; On windows:
 ; extension_dir = "ext"
 
