diff --git php.ini-production php.ini-production
index 794d3e8ddb..c7acf9c883 100644
--- php.ini-production
+++ php.ini-production
@@ -306,7 +306,7 @@ serialize_precision = -1
 ; and below.  This directive makes most sense if used in a per-directory
 ; or per-virtualhost web server configuration file.
 ; http://php.net/open-basedir
-;open_basedir =
+open_basedir = /srv/www/:/home/:/tmp/:/usr/share/pear/:/usr/share/webapps/
 
 ; This directive allows you to disable certain functions for security reasons.
 ; It receives a comma-delimited list of function names.
@@ -729,7 +729,7 @@ default_charset = "UTF-8"
 ;;;;;;;;;;;;;;;;;;;;;;;;;
 
 ; UNIX: "/path1:/path2"
-;include_path = ".:/php/includes"
+include_path = ".:/usr/share/pear"
 ;
 ; Windows: "\path1;\path2"
 ;include_path = ".;c:\php\includes"
@@ -752,9 +752,7 @@ user_dir =
 
 ; Directory in which the loadable extensions (modules) reside.
 ; http://php.net/extension-dir
-;extension_dir = "./"
-; On windows:
-;extension_dir = "ext"
+extension_dir = "/usr/lib/php/modules/"
 
 ; Directory where the temporary files should be placed.
 ; Defaults to the system default (see sys_get_temp_dir)
