--- Makefile.tgl	2020-01-06 11:43:44.000000000 +0100
+++ Makefile.tgl.new	2020-02-04 16:11:57.434863220 +0100
@@ -5,8 +5,11 @@
 
 TLD_OBJECTS=${OBJ}/dump-tl-file.o
 GENERATE_OBJECTS=${OBJ}/generate.o
+GENERATE_OBJECTS_HOST=${OBJ_HOST}/generate.o
 TGL_COMMON_OBJECTS=${OBJ}/tools.o ${OBJ}/crypto/rand_openssl.o ${OBJ}/crypto/rand_altern.o ${OBJ}/crypto/err_openssl.o ${OBJ}/crypto/err_altern.o
-TGL_OBJ_C=${GENERATE_OBJECTS} ${TGL_COMMON_OBJECTS} ${TGL_OBJECTS} ${TLD_OBJECTS}
+TGL_COMMON_OBJECTS_HOST=${OBJ_HOST}/tools.o ${OBJ_HOST}/crypto/rand_openssl.o ${OBJ_HOST}/crypto/rand_altern.o ${OBJ_HOST}/crypto/err_openssl.o ${OBJ_HOST}/crypto/err_altern.o
+TGL_OBJ_C=${GENERATE_OBJECTS} ${TGL_COMMON_OBJECTS} ${TGL_OBJECTS} ${TLD_OBJECTS }
+TGL_OBJ_C_HOST=${GENERATE_OBJECTS_HOST} ${TGL_COMMON_OBJECTS_HOST}
 
 AR?=ar
 
@@ -18,6 +21,9 @@
 
 ${TGL_OBJECTS}: ${AUTO}/constants.h ${AUTO}/auto-skip.h ${AUTO}/auto-fetch.h ${AUTO}/auto-store.h ${AUTO}/auto-autocomplete.h ${AUTO}/auto-types.h ${AUTO}/auto-fetch-ds.h ${AUTO}/auto-free-ds.h ${AUTO}/auto-store-ds.h ${AUTO}/auto-print-ds.h
 
+${TGL_OBJ_C_HOST}: ${OBJ_HOST}/%.o: ${srcdir}/tgl/%.c | create_dirs
+	${CC_FOR_BUILD} ${INCLUDE} ${COMPILE_FLAGS_HOST} -iquote ${srcdir}/tgl -c -MP -MD -MF ${DEP}/$*.d -MQ ${OBJ_HOST}/$*.o -o $@ $<
+
 ${TGL_OBJ_C}: ${OBJ}/%.o: ${srcdir}/tgl/%.c | create_dirs
 	${CC} ${INCLUDE} ${COMPILE_FLAGS} -iquote ${srcdir}/tgl -c -MP -MD -MF ${DEP}/$*.d -MQ ${OBJ}/$*.o -o $@ $<
 
@@ -30,23 +36,26 @@
 ${LIB}/libtgl.a: ${TGL_OBJECTS} ${TGL_COMMON_OBJECTS} ${TGL_OBJECTS_AUTO}
 	rm -f $@ && ${AR} ruv $@ $^
 
-${EXE}/generate: ${GENERATE_OBJECTS} ${TGL_COMMON_OBJECTS}
-	${CC} ${GENERATE_OBJECTS} ${TGL_COMMON_OBJECTS} ${LINK_FLAGS} -o $@
+${EXE}/generate: ${GENERATE_OBJECTS_HOST} ${TGL_COMMON_OBJECTS_HOST}
+	${CC_FOR_BUILD} ${GENERATE_OBJECTS_HOST} ${TGL_COMMON_OBJECTS_HOST} ${LINK_FLAGS} -o $@
+
+${EXE_HOST}/generate: ${GENERATE_OBJECTS_HOST} ${TGL_COMMON_OBJECTS_HOST}
+	${CC_FOR_BUILD} ${GENERATE_OBJECTS_HOST} ${TGL_COMMON_OBJECTS_HOST} ${LINK_FLAGS} -o $@
 
-${AUTO}/scheme.tlo: ${AUTO}/scheme.tl ${EXE}/tl-parser
-	${EXE}/tl-parser -e $@ ${AUTO}/scheme.tl
+${AUTO}/scheme.tlo: ${AUTO}/scheme.tl ${EXE_HOST}/tl-parser
+	${EXE_HOST}/tl-parser -e $@ ${AUTO}/scheme.tl
 
 ${AUTO}/scheme.tl: ${srcdir}/tgl/scheme.tl ${srcdir}/tgl/encrypted_scheme.tl ${srcdir}/tgl/binlog.tl ${srcdir}/tgl/append.tl ${srcdir}/tgl/mtproto.tl | ${AUTO}
 	cat $^ > $@
 
-${AUTO}/scheme2.tl: ${AUTO}/scheme.tl ${EXE}/tl-parser
-	${EXE}/tl-parser -E ${AUTO}/scheme.tl 2> $@  || ( cat $@ && rm $@ && false )
+${AUTO}/scheme2.tl: ${AUTO}/scheme.tl ${EXE_HOST}/tl-parser
+	${EXE_HOST}/tl-parser -E ${AUTO}/scheme.tl 2> $@  || ( cat $@ && rm $@ && false )
 
-${AUTO}/auto-%.c: ${AUTO}/scheme.tlo ${EXE}/generate auto/constants.h ${AUTO}/auto-%.h | create_dirs_and_headers
-	${EXE}/generate -g $(patsubst ${AUTO}/auto-%.c,%,$@) ${AUTO}/scheme.tlo > $@ || ( rm $@ && false )
+${AUTO}/auto-%.c: ${AUTO}/scheme.tlo ${EXE_HOST}/generate auto/constants.h ${AUTO}/auto-%.h | create_dirs_and_headers
+	${EXE_HOST}/generate -g $(patsubst ${AUTO}/auto-%.c,%,$@) ${AUTO}/scheme.tlo > $@ || ( rm $@ && false )
 
-${AUTO}/auto-%.h: ${AUTO}/scheme.tlo ${EXE}/generate
-	${EXE}/generate -g $(patsubst ${AUTO}/auto-%.h,%-header,$@) ${AUTO}/scheme.tlo > $@  || ( rm $@ && false )
+${AUTO}/auto-%.h: ${AUTO}/scheme.tlo ${EXE_HOST}/generate
+	${EXE_HOST}/generate -g $(patsubst ${AUTO}/auto-%.h,%-header,$@) ${AUTO}/scheme.tlo > $@  || ( rm $@ && false )
 
 ${AUTO}/constants.h: ${AUTO}/scheme2.tl ${srcdir}/tgl/gen_constants_h.awk
 	awk -f ${srcdir}/tgl/gen_constants_h.awk < $< > $@
