--- Makefile.tl-parser	2020-01-06 11:43:44.000000000 +0100
+++ Makefile.tl-parser.new	2020-02-04 15:42:13.876880229 +0100
@@ -1,7 +1,15 @@
 TL_PARSER_OBJECTS=${OBJ}/tl-parser.o ${OBJ}/tlc.o
 
+TL_PARSER_OBJECTS_HOST=${OBJ_HOST}/tl-parser.o ${OBJ_HOST}/tlc.o
+
 ${TL_PARSER_OBJECTS}: ${OBJ}/%.o: ${srcdir}/tgl/tl-parser/%.c | create_dirs
 	${CC} ${INCLUDE} ${COMPILE_FLAGS} -iquote ${srcdir}/tgl/tl-parser -c -MP -MD -MF ${DEP}/$*.d -MQ ${OBJ}/$*.o -o $@ $<
 
+${TL_PARSER_OBJECTS_HOST}: ${OBJ_HOST}/%.o: ${srcdir}/tgl/tl-parser/%.c | create_dirs
+	${CC_FOR_BUILD} ${INCLUDE} ${COMPILE_FLAGS_HOST} -iquote ${srcdir}/tgl/tl-parser -c -MP -MD -MF ${DEP}/$*.d -MQ ${OBJ}/$*.o -o $@ $<
+
 ${EXE}/tl-parser: ${TL_PARSER_OBJECTS}
 	${CC} $^ ${LINK_FLAGS} -o $@
+
+${EXE_HOST}/tl-parser: ${TL_PARSER_OBJECTS_HOST}
+	${CC_FOR_BUILD} $^ ${LINK_FLAGS} -o $@
