diff --git a/src/jobs/meltjob.cpp b/src/jobs/meltjob.cpp
index f5390e4..693a089 100644
--- src/jobs/meltjob.cpp
+++ src/jobs/meltjob.cpp
@@ -50,7 +50,7 @@ void MeltJob::start()
 #ifdef Q_OS_WIN
     QFileInfo meltPath(shotcutPath, "qmelt.exe");
 #else
-    QFileInfo meltPath(shotcutPath, "qmelt");
+    QFileInfo meltPath(shotcutPath, "melt");
 #endif
     setReadChannel(QProcess::StandardError);
     QStringList args;
