--- a/eduvpn/data/share/applications/org.eduvpn.client.desktop	2024-07-30 10:00:49.000000000 +0200
+++ -	2024-08-11 17:45:12.089603897 +0200
@@ -6,5 +6,5 @@
 Exec=eduvpn-gui
 Icon=org.eduvpn.client
 Terminal=false
-StartupWMClass=org.eduvpn.client
+StartupWMClass=eduvpn-gui
 Categories=Network;Dialup
--- a/eduvpn/data/share/applications/org.letsconnect-vpn.client.desktop	2024-07-30 10:00:49.000000000 +0200
+++ -	2024-08-11 17:45:33.379549600 +0200
@@ -6,5 +6,5 @@
 Exec=letsconnect-gui
 Icon=org.letsconnect-vpn.client
 Terminal=false
-StartupWMClass=org.letsconnect-vpn.client
+StartupWMClass=letsconnect-gui
 Categories=Network;Dialup
