diff -Nru nux-4.0.0daily12.12.05.orig/configure.ac nux-4.0.0daily12.12.05/configure.ac
--- nux-4.0.0daily12.12.05.orig/configure.ac	2013-01-07 13:35:53.028194074 -0500
+++ nux-4.0.0daily12.12.05/configure.ac	2013-01-07 13:36:07.038120083 -0500
@@ -60,7 +60,7 @@
 m4_ifdef([AM_SILENT_RULES],[AM_SILENT_RULES([yes])])
 
 AC_CONFIG_SRCDIR([Makefile.am])
-AM_CONFIG_HEADER([config.h])
+AC_CONFIG_HEADER([config.h])
 AM_INIT_AUTOMAKE([1.10])
 
 NUX_MAJOR_VERSION=nux_major_version
