--- a/NuxGraphics/IOpenGLFrameBufferObject.cpp	2022-11-09 20:34:57.826292791 +0530
+++ b/NuxGraphics/IOpenGLFrameBufferObject.cpp	2022-11-09 20:44:30.187949413 +0530
@@ -72,6 +72,10 @@
     texture_attachment_array_.clear();
     surface_attachment_array_.clear();
 
+    auto const& gpu_info = GetGraphicsDisplay()->GetGpuDevice()->GetGpuInfo();
+    texture_attachment_array_.resize(gpu_info.GetMaxFboAttachment());
+    surface_attachment_array_.resize(gpu_info.GetMaxFboAttachment());
+
     depth_surface_attachment_ = ObjectPtr<IOpenGLSurface> (0);
     stencil_surface_attachment_ = ObjectPtr<IOpenGLSurface> (0);
 
