diff -p1ru a/Nux/TimerProc.h b/Nux/TimerProc.h
--- a/Nux/TimerProc.h	2013-08-15 06:08:42.000000000 +0800
+++ b/Nux/TimerProc.h	2024-07-16 18:21:08.638000000 +0800
@@ -21,3 +21,3 @@
 
-
+#include <memory>
 #ifndef TIMERPROC_H
diff -p1ru a/Nux/WindowThread.h b/Nux/WindowThread.h
--- a/Nux/WindowThread.h	2014-02-25 02:14:54.000000000 +0800
+++ b/Nux/WindowThread.h	2024-07-16 18:21:10.262000000 +0800
@@ -21,3 +21,3 @@
 
-
+#include <memory>
 #ifndef WINDOWTHREAD_H
