--- a/NuxGraphics/FreetypeFont.cpp
+++ b/NuxGraphics/FreetypeFont.cpp
@@ -27,7 +27,7 @@
 //#include "OpenGL/GLsdk/GL/gl.h"
 //#include "OpenGL/GLsdk/GL/glprocs.h"
 //#include <GL/glu.h>
-#include <GL/glew.h>
+#include <glew-1.13.0/GL/glew.h>
 #include "GfxServer.h"
 #include "OpenGLEngine/GLCheck.h"
 #include "OpenGLEngine/OpenGLEngine.h"
--- a/NuxGraphics/GLResource.h
+++ b/NuxGraphics/GLResource.h
@@ -72,8 +72,8 @@ namespace nux
 #define NUX_ENABLE_CG_SHADERS 0
 
 #if defined(NUX_OS_WINDOWS)
-  #include "GL/glew.h"
-  #include "GL/wglew.h"
+  #include "GL/glewmx.h"
+  #include "GL/wglewmx.h"
 
   GLEWContext *glewGetContext();
   WGLEWContext *wglewGetContext();
@@ -96,8 +96,8 @@ namespace nux
     #ifndef GLEW_MX
       #define GLEW_MX
     #endif
-    #include "GL/glew.h"
-    #include "GL/glxew.h"
+    #include "GL/glewmx.h"
+    #include "GL/glxewmx.h"
 
     GLEWContext *glewGetContext();
     GLXEWContext *glxewGetContext();
