diff --git a/CMakeLists.txt b/CMakeLists.txt
index 67b61fa..d928dae 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -36,6 +36,7 @@ set(SOURCES
     src/data/graphelementdata.cpp
     src/data/graphelementproperty.cpp
     src/data/node.cpp
+    src/data/path.cpp
     src/data/pdfdocument.cpp
     src/data/style.cpp
     src/data/stylelist.cpp
@@ -50,6 +51,7 @@ set(SOURCES
     src/gui/mainmenu.cpp
     src/gui/mainwindow.cpp
     src/gui/nodeitem.cpp
+    src/gui/pathitem.cpp
     src/gui/preferencedialog.cpp
     src/gui/previewwindow.cpp
     src/gui/propertypalette.cpp
