From ad6a14a2f28c3d54755f3a1a17069089ad896666 Mon Sep 17 00:00:00 2001
From: Andreas Kloeckner <inform@tiker.net>
Date: Mon, 28 Aug 2023 12:51:04 -0500
Subject: [PATCH] Fix numpy finding in aksetup

---
 aksetup_helper.py | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/aksetup_helper.py b/aksetup_helper.py
index d36a2d70..8c9bcec5 100644
--- a/aksetup_helper.py
+++ b/aksetup_helper.py
@@ -43,13 +43,13 @@


 def get_numpy_incpath():
-    from os.path import join, basename
+    from os.path import join, dirname
     from importlib.util import find_spec
     origin = find_spec("numpy").origin
     if origin is None:
         raise RuntimeError("origin of numpy package not found")

-    pathname = basename(origin)
+    pathname = dirname(origin)
     return join(pathname, "core", "include")


