From 3f0a36985a046dfb9432e57fb26e5ee4913fc7fe Mon Sep 17 00:00:00 2001
From: Andrea Brancaleoni <miwaxe@gmail.com>
Date: Tue, 25 Aug 2015 19:06:12 +0200
Subject: [PATCH] musl

---
 code/unix/unix_main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git code/unix/unix_main.c code/unix/unix_main.c
index ab3608b..f823800 100644
--- a/code/unix/unix_main.c
+++ b/code/unix/unix_main.c
@@ -53,7 +53,7 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 #include <dlfcn.h>
 #endif

-#ifdef __linux__
+#ifdef __GLIBC__
   #include <fpu_control.h> // bk001213 - force dumps on divide by zero
 #endif

--
2.5.0

