--- a/Makefile
+++ b/Makefile
@@ -827,7 +827,7 @@ check: check_back check_mild
 check_mild:	check_character_set check_white_space check_links \
 		  check_name_lengths check_now \
 		  check_slashed_abbrs check_sorted \
-		  check_tables check_web check_ziguard check_zishrink check_tzs
+		  check_tables check_ziguard check_zishrink check_tzs
 
 # True if UTF8_LOCALE does not work;
 # otherwise, false but with LC_ALL set to $(UTF8_LOCALE).
