Index: tzutils-2021b/Makefile
===================================================================
--- tzutils-2021b.orig/Makefile
+++ tzutils-2021b/Makefile
@@ -751,9 +751,8 @@ tzselect:	tzselect.ksh version
 		chmod +x $@.out
 		mv $@.out $@
 
-check:		check_character_set check_white_space check_links \
-		  check_name_lengths check_sorted \
-		  check_tables check_web check_zishrink check_tzs
+check:		check_links check_name_lengths check_sorted \
+		  check_tables check_zishrink check_tzs
 
 check_character_set: $(ENCHILADA)
 	test ! '$(UTF8_LOCALE)' || \
