--- a/OgreMain/include/OgreString.h	2018-04-17 09:04:05.425658306 +0200
+++ b/OgreMain/include/OgreString.h	2018-04-17 09:14:39.457143961 +0200
@@ -76,6 +76,11 @@ namespace __gnu_cxx
 #   define strtol_l(ptr, end, base, l) strtol(ptr, end, base)
 #endif
 
+#if OGRE_PLATFORM == OGRE_PLATFORM_LINUX && !defined(__GLIBC__)
+#   define strtoul_l(ptr, end, base, l) strtoul(ptr, end, base)
+#   define strtol_l(ptr, end, base, l) strtol(ptr, end, base)
+#endif
+
 // If compiling with make on macOS, these headers need to be included to get
 // definitions of locale_t, strtod_l, etc...
 // See: http://www.unix.com/man-page/osx/3/strtod_l/
