--- a/src/meson.build	2023-04-17 23:31:55.000000000 +0100
+++ b/src/meson.build	2023-04-27 06:21:59.187169515 +0100
@@ -3,10 +3,10 @@
 # Needs prefix for configure_file()
 if get_option('append_libdir_mangohud')
   libdir_mangohud = join_paths(get_option('prefix'), get_option('libdir'), 'mangohud')
-  ld_libdir_mangohud = get_option('prefix') + '/\$LIB/mangohud/'
+  ld_libdir_mangohud = join_paths(get_option('prefix') ,get_option('libdir'), 'mangohud')
 else
   libdir_mangohud = join_paths(get_option('prefix'), get_option('libdir'))
-  ld_libdir_mangohud = get_option('prefix') + '/\$LIB/'
+  ld_libdir_mangohud = join_paths(get_option('prefix') ,get_option('libdir'))
 endif
 
 conf_data = configuration_data()
--- a/bin/mangohud.in	2023-04-17 23:31:55.000000000 +0100
+++ b/bin/mangohud.in	2023-04-27 06:30:23.169862796 +0100
@@ -8,11 +8,11 @@
 	exit 1
 fi
 
-MANGOHUD_LIB_NAME="@ld_libdir_mangohud@libMangoHud.so"
+MANGOHUD_LIB_NAME="@ld_libdir_mangohud@/libMangoHud.so"
 
 if [ "$1" = "--dlsym" ]; then
 	MANGOHUD_DLSYM=1
-	MANGOHUD_LIB_NAME="@ld_libdir_mangohud@libMangoHud_dlsym.so:${MANGOHUD_LIB_NAME}"
+	MANGOHUD_LIB_NAME="@ld_libdir_mangohud@/libMangoHud_dlsym.so:${MANGOHUD_LIB_NAME}"
 	shift
 fi
