diff --git commands.c commands.c
index a3d83c9..a6a98ed 100644
--- commands.c
+++ commands.c
@@ -1144,7 +1144,7 @@ void command_retr(char *filename)
                     wait_time.tv_usec = xfer_delay;
                     select( 0, NULL, NULL, NULL, &wait_time);
                 }
-                i = read(phile, buffer, my_buffer_size)
+                i = read(phile, buffer, my_buffer_size);
             }
             free(buffer);
             close(phile);
