--- a/Makefile.in
+++ b/Makefile.in
@@ -15,7 +15,7 @@ OBJS=bftpdutmp.o commands.o commands_adm
 SRCS=bftpdutmp.c commands.c commands_admin.c cwd.c dirlist.c list.c login.c logging.c main.c mystring.c options.c md5.c
 
 OBJS2LINK=$(OBJS) $(PAX)
-LDFLAGS=
+LDFLAGS=@LDFLAGS@
 
 all: bftpd
 
@@ -37,18 +37,12 @@ endif
 %.o: %.d
 
 install: all
-	mkdir -p $(DESTDIR)/$(prefix)/sbin
-	$(INSTALL) -g 0 -m 700 -o 0 bftpd $(DESTDIR)/$(prefix)/sbin
+	mkdir -p $(DESTDIR)/$(prefix)/bin
+	$(INSTALL) -g 0 -m 700 -o 0 bftpd $(DESTDIR)/$(prefix)/bin
 	mkdir -p $(DESTDIR)/$(mandir)/man8
 	$(INSTALL) -g 0 -m 644 -o 0 bftpd.8 $(DESTDIR)/$(mandir)/man8
-	mkdir -p $(DESTDIR)/$(prefix)/etc
-	[ -f $(DESTDIR)/$(prefix)/etc/bftpd.conf ] || \
-		$(INSTALL) -g 0 -m 600 -o 0 bftpd.conf $(DESTDIR)/$(prefix)/etc
-	mkdir -p $(DESTDIR)/$(prefix)/var/log
-	touch $(DESTDIR)/$(prefix)/var/log/bftpd.log
-	chmod 644 $(DESTDIR)/$(prefix)/var/log/bftpd.log
-	mkdir -p $(DESTDIR)/$(prefix)/var/run/bftpd
-	chmod 755 $(DESTDIR)/$(prefix)/var/run/bftpd
+	mkdir -p $(DESTDIR)/etc
+	$(INSTALL) -g 0 -m 600 -o 0 bftpd.conf $(DESTDIR)/etc
 
 clean distclean:
 	rm -f *~ $(OBJS) bftpd mksources.finished config.cache
--- a/mypaths.h
+++ b/mypaths.h
@@ -1,5 +1,5 @@
 #define PATH_BFTPD_CONF "/etc/bftpd.conf"
-#ifdef PREFIX
+#if 0
 #define PATH_BFTPD_CONF_WITH_PREFIX PREFIX"/etc/bftpd.conf"
 #endif
 #define PATH_STATUSLOG "/dev/null"
