--- Makefile.orig	2010-12-14 00:44:40.207776131 +0100
+++ Makefile	2010-12-14 00:44:44.599106588 +0100
@@ -39,7 +39,7 @@ PROGRAMS += $(shell test `uname -m 2>/de
 # BSD make doesn't understand the $(shell) syntax above, it wants the !=
 # syntax below. GNU make ignores the line below so in the end both BSD
 # make and GNU make are happy.
-PROGRAMS != echo dmidecode ; test `uname -m 2>/dev/null` != ia64 && echo biosdecode ownership vpddecode
+#PROGRAMS != echo dmidecode ; test `uname -m 2>/dev/null` != ia64 && echo biosdecode ownership vpddecode
 
 all : $(PROGRAMS)
 
