--- src/widgets/private/settings/shortcutedit.cpp	2019-07-30 13:07:11.000000000 +0200
+++ -	2020-06-22 16:42:22.506228977 +0200
@@ -19,6 +19,7 @@
 
 #include <QDebug>
 #include <QPainter>
+#include <QPainterPath>
 #include <QKeySequence>
 #include <QKeyEvent>
 
--- src/widgets/dsegmentedcontrol.cpp	2019-07-30 13:07:11.000000000 +0200
+++ -	2020-06-22 16:43:42.149506303 +0200
@@ -17,6 +17,7 @@
 
 #include <QDebug>
 #include <QPainter>
+#include <QPainterPath>
 #include <QPaintEvent>
 #include <QApplication>
 
--- src/widgets/dvideowidget.cpp	2019-07-30 13:07:11.000000000 +0200
+++ -	2020-06-22 16:44:21.292219385 +0200
@@ -24,6 +24,7 @@
 #include <QMediaPlayer>
 #include <QMediaPlaylist>
 #include <QPainter>
+#include <QPainterPath>
 #include <QPointer>
 
 DWIDGET_BEGIN_NAMESPACE
--- src/widgets/dsimplelistview.cpp	2019-07-30 13:07:11.000000000 +0200
+++ -	2020-06-22 16:45:02.445556466 +0200
@@ -30,6 +30,7 @@
 #include <QStyleFactory>
 #include <QWheelEvent>
 #include <QtMath>
+#include <QPainterPath>
 #include <QPointer>
 
 #include "dhidpihelper.h"
--- src/widgets/dclipeffectwidget.cpp	2019-07-30 13:07:11.000000000 +0200
+++ -	2020-06-22 16:45:36.908358398 +0200
@@ -21,6 +21,7 @@
 #include <QEvent>
 #include <QBackingStore>
 #include <QPainter>
+#include <QPainterPath>
 #include <QPaintEvent>
 #include <QDebug>
 
--- src/widgets/dgraphicsclipeffect.cpp	2019-07-30 13:07:11.000000000 +0200
+++ -	2020-06-22 16:46:09.212494581 +0200
@@ -19,6 +19,7 @@
 #include <DObjectPrivate>
 
 #include <QPainter>
+#include <QPainterPath>
 #include <QDebug>
 
 DWIDGET_BEGIN_NAMESPACE
--- src/widgets/dwaterprogress.cpp	2019-07-30 13:07:11.000000000 +0200
+++ -	2020-06-22 16:46:35.208048436 +0200
@@ -20,6 +20,7 @@
 #include <QtMath>
 #include <QTimer>
 #include <QPainter>
+#include <QPainterPath>
 #include <QGraphicsDropShadowEffect>
 
 #include <DObjectPrivate>
--- src/widgets/private/dblureffectwidget_p.h	2019-07-30 13:07:11.000000000 +0200
+++ -	2020-06-22 16:47:08.709658671 +0200
@@ -19,6 +19,7 @@
 #define DBLUREFFECTWIDGET_P_H
 
 #include <DObjectPrivate>
+#include <QPainterPath>
 #include "dblureffectwidget.h"
 
 DWIDGET_BEGIN_NAMESPACE
--- src/widgets/dspinner.cpp	2019-07-30 13:07:11.000000000 +0200
+++ -	2020-06-22 16:47:35.308523365 +0200
@@ -2,6 +2,7 @@
 
 #include <QtMath>
 #include <QPainter>
+#include <QPainterPath>
 #include <QTimer>
 
 #include <DObjectPrivate>
--- src/widgets/dborderlesswidget.cpp	2019-07-30 13:07:11.000000000 +0200
+++ -	2020-06-22 16:48:29.369148052 +0200
@@ -25,6 +25,7 @@
 #include <QTemporaryFile>
 #include <QImage>
 #include <QPainter>
+#include <QPainterPath>
 #include <QGuiApplication>
 #include <QWindow>
 
--- src/widgets/dcrumbedit.cpp	2019-07-30 13:07:11.000000000 +0200
+++ -	2020-06-22 16:48:50.759824370 +0200
@@ -23,6 +23,7 @@
 
 #include <QAbstractTextDocumentLayout>
 #include <QPainter>
+#include <QPainterPath>
 #include <QTextBlock>
 #include <QStyleOptionFrame>
 #include <QMouseEvent>
--- src/dtkwidget_global.h	2019-07-30 13:07:11.000000000 +0200
+++ -	2020-06-22 16:54:00.773382584 +0200
@@ -21,6 +21,7 @@
 #include <dtkwidget_config.h>
 
 #include <QtCore/QMetaMethod>
+#include <QtGui/QPainterPath>
 
 #define DWIDGET_NAMESPACE Widget
 #define DTK_WIDGET_NAMESPACE DTK_NAMESPACE::Widget
