From bf0601df89f611b33d8dbc27722351bf99e2cc18 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Piotr=20W=C3=B3jcik?= <chocimier@tlen.pl>
Date: Sat, 18 Jan 2020 12:19:45 +0100
Subject: [PATCH] relocate


diff --git src/port/path.c src/port/path.c
index 710988b..20dc57d 100644
--- a/src/port/path.c
+++ b/src/port/path.c
@@ -544,6 +544,8 @@ make_relative_path(char *ret_path, const char *target_path,
 	int			tail_len;
 	int			i;
 
+	goto no_match; // we relocate only binaries
+
 	/*
 	 * Determine the common prefix --- note we require it to end on a
 	 * directory separator, consider eg '/usr/lib' and '/usr/libexec'.
-- 
2.23.0

