--- a/src/meson.build
+++ b/src/mseon.build
@@ -1,5 +1,4 @@
-add_project_arguments('-rdynamic', '-DWLR_USE_UNSTABLE', '-Werror',
-'-Wno-unused-function', '-Wno-uninitialized', '-Wno-sizeof-pointer-div', language: 'c')
+add_project_arguments('-rdynamic', '-DWLR_USE_UNSTABLE', '-Wno-sizeof-pointer-div', language: 'c')
 
 foreach name : ['lua', 'lua5.3', 'lua-5.3', 'lua53']
     luaDep = dependency(name, version: '>=5.3', required: false)
