--- dialogs/keyboard-settings/xfce-keyboard-settings.c
+++ dialogs/keyboard-settings/xfce-keyboard-settings.c
@@ -457,7 +457,7 @@ xfce_keyboard_settings_finalize (GObject *object)
 
 #ifdef HAVE_LIBXKLAVIER
   /* Stop xklavier engine */
-  xkl_engine_stop_listen (settings->priv->xkl_engine);
+  xkl_engine_stop_listen (settings->priv->xkl_engine, XKLL_TRACK_KEYBOARD_STATE);
 #endif /* HAVE_LIBXKLAVIER */
 
   g_object_unref (settings->priv->provider);
