--- Makefile.orig	2010-05-02 14:30:00.962750572 +0200
+++ Makefile	2010-05-02 14:31:50.935750160 +0200
@@ -3,10 +3,10 @@ CXX=g++
 BIN=${DESTDIR}/usr/bin
 
 # For 'make man': sudo apt-get install xsltproc docbook-xsl docbook-xml
-DB2MAN=/usr/share/sgml/docbook/stylesheet/xsl/nwalsh/manpages/docbook.xsl
+DB2MAN=/usr/share/xsl/docbook/manpages/docbook.xsl
 XP=xsltproc -''-nonet -''-param man.charmap.use.subset "0"
 MAN_SOURCE=man/cppcheck.1.xml
-
+MANDIR=$(DESTDIR)/usr/share/man
 
 ###### Object Files
 
@@ -77,13 +77,14 @@ test:	all
 clean:
 	rm -f lib/*.o cli/*.o test/*.o testrunner cppcheck
 
-man:	$(MAN_SOURCE)
-	$(XP) $(DB2MAN) $(MAN_SOURCE)
+cppcheck.1:	$(MAN_SOURCE)
+	$(XP) $(DB2MAN) $<
 
 install:	cppcheck
 	install -d ${BIN}
-	install cppcheck ${BIN}
-
+	install -m755 cppcheck ${BIN}
+	install -d $(MANDIR)/man1
+	install -m644 cppcheck.1 $(MANDIR)/man1
 
 ###### Build
 
