--- a/test/signal/test-signalhandler.cpp
+++ b/test/signal/test-signalhandler.cpp
@@ -52,7 +52,7 @@
     ++*(int*)nullptr;
 }
 
-#if !defined(__APPLE__)
+#if !defined(__APPLE__) && defined(__GLIBC__)
 /*static*/ int my_fpe() // NOLINT(misc-use-internal-linkage)
 {
     if (feenableexcept(FE_ALL_EXCEPT) == -1)
@@ -78,7 +78,7 @@
         my_abort();
     else if (strcmp(argv[1], "segv") == 0)
         my_segv();
-#if !defined(__APPLE__)
+#if !defined(__APPLE__) && defined(__GLIBC__)
     else if (strcmp(argv[1], "fpe") == 0)
         return my_fpe();
 #endif
