diff --git a/life_line_chart/SimpleSVGItems.py b/life_line_chart/SimpleSVGItems.py
index 09d5ede..7e2cb44 100644
--- a/life_line_chart/SimpleSVGItems.py
+++ b/life_line_chart/SimpleSVGItems.py
@@ -1,4 +1,8 @@
-from collections import MutableSequence
+try:
+    from collections.abc import MutableSequence
+except ImportError:
+    # Required for python versions < 3.9
+    from collections import MutableSequence
 
 
 class Line(object):
