From 46c37c04967c06d1f8ded3b40161b9d9fcb8ff8d Mon Sep 17 00:00:00 2001
From: Dominique Leuenberger <dimstar@opensuse.org>
Date: Fri, 19 Jun 2015 09:14:14 +0200
Subject: libgnac-debug.h: fix typo (critial vs critical)

---
 libgnac/libgnac-debug.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libgnac/libgnac-debug.h b/libgnac/libgnac-debug.h
index 3f2705d..087e251 100644
--- a/libgnac/libgnac-debug.h
+++ b/libgnac/libgnac-debug.h
@@ -52,7 +52,7 @@
 
 #if defined(__STDC_VERSION__) && __STDC_VERSION__ >= 199901L
 #define libgnac_critical(...) \
-        libgnac_critial_real (__func__, __FILE__, __LINE__, TRUE, __VA_ARGS__)
+        libgnac_critical_real (__func__, __FILE__, __LINE__, TRUE, __VA_ARGS__)
 #elif defined(__GNUC__) && __GNUC__ >= 3
 #define libgnac_critical(...) \
         libgnac_critical_real (__FUNCTION__, __FILE__, __LINE__, __VA_ARGS__)
-- 
cgit v0.12

