--- a/src/photos-base-manager.h	2021-04-20 13:16:07.042348375 +0200
+++ b/src/photos-base-manager.h	2021-04-20 13:16:16.974347953 +0200
@@ -78,8 +78,6 @@ guint               photos_base_manager_
 
 GObject            *photos_base_manager_get_previous_object      (PhotosBaseManager *self, GObject *object);
 
-PhotosSparqlTemplate *photos_base_manager_get_sparql_template    (PhotosBaseManager *self, gint flags);
-
 const gchar        *photos_base_manager_get_title                (PhotosBaseManager *self);
 
 gchar              *photos_base_manager_get_where                (PhotosBaseManager *self, gint flags);
