From a0ac868a19d55dc52100e54fe4b2d29041bc6752 Mon Sep 17 00:00:00 2001
From: rvalue <i@rvalue.moe>
Date: Fri, 21 Jan 2022 12:36:02 +0000
Subject: [PATCH] meson: remove incorrect args for i18n.merge_file

---
 data/meson.build | 2 --
 1 file changed, 2 deletions(-)

diff --git a/data/meson.build b/data/meson.build
index 6e7192e0..94271de2 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -3,7 +3,6 @@ subdir('icons')
 appdata = photos_namespace + '.appdata.xml'
 
 i18n.merge_file(
-  appdata,
   input: appdata + '.in',
   output: appdata,
   po_dir: po_dir,
@@ -23,7 +22,6 @@ desktop_in = configure_file(
 )
 
 i18n.merge_file(
-  desktop,
   type: 'desktop',
   input: desktop_in,
   output: desktop,
-- 
GitLab

