--- a/configure.in
+++ b/configure.in
@@ -1,4 +1,4 @@
-AC_INIT(src/swmain.c)
+AC_INIT([sopwith], [1.8.4])
 AC_CONFIG_AUX_DIR(autotools)
 
 VERSION=1.8.4
@@ -46,7 +46,8 @@
 	;;
 esac
 
-AM_INIT_AUTOMAKE(sopwith,$VERSION)
+AM_INIT_AUTOMAKE([-Wall foreign])
+AM_PROG_AR
 
 AM_CONFIG_HEADER(config.h:config.hin)
 
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -6,7 +6,7 @@
 bin_PROGRAMS = @SOPWITH_BINS@
 EXTRA_PROGRAMS = sopwith gtksopwith psopwith
 
-CFLAGS = @CFLAGS@ -I..
+AM_CFLAGS = @CFLAGS@ @SDL_CFLAGS@ -I..
 
 COMMON_SRC =       \
 video.h	    swcollsn.c  swgames.h   swmain.h    swutil.h 	\
--- a/src/sdl/Makefile.am
+++ b/src/sdl/Makefile.am
@@ -1,4 +1,4 @@
-CFLAGS=@CFLAGS@ @SDL_CFLAGS@ -I..
+AM_CFLAGS=@CFLAGS@ @SDL_CFLAGS@ -I..
 
 noinst_LIBRARIES = libsdlsopwith.a
 
