--- istream.c	2011-01-04 20:22:22.000000000 +1100
+++ istream.c	2012-07-04 21:50:51.529661517 +1000
@@ -22,8 +22,8 @@
 static void basic_close(int *handle);
 static int basic_read(int *handle, char *buf, int len);
 
-static void file_close(struct file_handle *handle);
-static int file_read(struct file_handle *handle, char *buf, int len);
+static void file_close(struct file_handle_rofl *handle);
+static int file_read(struct file_handle_rofl *handle, char *buf, int len);
 
 static int str_read(Str handle, char *buf, int len);
 
@@ -114,7 +114,7 @@
     stream = New(union input_stream);
     init_base_stream(&stream->base, STREAM_BUF_SIZE);
     stream->file.type = IST_FILE;
-    stream->file.handle = New(struct file_handle);
+    stream->file.handle = New(struct file_handle_rofl);
     stream->file.handle->f = f;
     if (closep)
 	stream->file.handle->close = closep;
@@ -658,13 +658,13 @@
 }
 
 static void
-file_close(struct file_handle *handle)
+file_close(struct file_handle_rofl *handle)
 {
     handle->close(handle->f);
 }
 
 static int
-file_read(struct file_handle *handle, char *buf, int len)
+file_read(struct file_handle_rofl *handle, char *buf, int len)
 {
     return fread(buf, 1, len, handle->f);
 }
--- istream.h	2003-10-21 02:41:56.000000000 +1000
+++ istream.h	2012-07-04 21:50:51.529661517 +1000
@@ -20,7 +20,7 @@
 
 typedef struct stream_buffer *StreamBuffer;
 
-struct file_handle {
+struct file_handle_rofl {
     FILE *f;
     void (*close) ();
 };
@@ -53,7 +53,7 @@
 
 struct file_stream {
     struct stream_buffer stream;
-    struct file_handle *handle;
+    struct file_handle_rofl *handle;
     char type;
     char iseos;
     int (*read) ();
