--- ./scripts/w3mhelp.cgi.in.orig	2011-01-04 04:22:28.000000000 -0500
+++ ./scripts/w3mhelp.cgi.in	2015-11-03 16:42:22.593003004 -0500
@@ -48,7 +48,7 @@
 	}
 	# print "tlang=$tlang\n";
 	eval {require "w3mhelp-funcdesc.$tlang.pl";};
-	if (defined(%funcdesc)) {
+	if (%funcdesc) {
 	    $lang = $tlang;
 	}
     }
--- ./scripts/w3mman/w3mman2html.cgi.in.orig	2015-11-03 17:08:16.109020865 -0500
+++ ./scripts/w3mman/w3mman2html.cgi.in	2015-11-03 17:09:02.749021401 -0500
@@ -220,7 +220,7 @@
   local($p);
 
   (! -d && -x) || return 0;
-  if (! defined(%PATH)) {
+  if (!%PATH) {
     for $p (split(":", $ENV{'PATH'})) {
       $p =~ s@/+$@@;
       $PATH{$p} = 1;
