Disable ALPN with libressl; won't work until 2.1.2.

--- src/cryptography/hazmat/bindings/openssl/ssl.py.orig	2015-01-21 16:43:51.457822382 +0100
+++ src/cryptography/hazmat/bindings/openssl/ssl.py	2015-01-21 16:44:24.320971331 +0100
@@ -530,7 +530,7 @@ static const long Cryptography_HAS_NEXTP
 #endif
 
 /* ALPN was added in OpenSSL 1.0.2. */
-#if OPENSSL_VERSION_NUMBER < 0x10002001L
+#if 1
 int (*SSL_CTX_set_alpn_protos)(SSL_CTX *,
                                const unsigned char *,
                                unsigned) = NULL;
