--- src/cryptography/hazmat/bindings/openssl/ec.py
+++ src/cryptography/hazmat/bindings/openssl/ec.py
@@ -390,7 +390,7 @@ static const long Cryptography_HAS_EC2M = 1;
 #endif

 #if defined(OPENSSL_NO_EC) || OPENSSL_VERSION_NUMBER < 0x1000200f || \
-    defined(LIBRESSL_VERSION_NUMBER)
+    !defined(LIBRESSL_VERSION_NUMBER)
 static const long Cryptography_HAS_EC_1_0_2 = 0;
 const char *(*EC_curve_nid2nist)(int) = NULL;
 #else
