diff --git a/_packaging/appimage/org.gaphor.Gaphor.desktop b/_packaging/appimage/org.gaphor.Gaphor.desktop
index 93d3bdb..94651cb 100644
--- a/_packaging/appimage/org.gaphor.Gaphor.desktop
+++ b/_packaging/appimage/org.gaphor.Gaphor.desktop
@@ -6,7 +6,7 @@ Comment[fr]=L'outil de modélisation facile
 Comment[hu]=Egyszerű modellező eszköz
 Comment[nl]=De eenvoudige modelleertool
 Comment[pl]=Proste narzędzie do modelowania
-Exec=gaphor-exe %f
+Exec=gaphor %f
 Icon=org.gaphor.Gaphor
 Terminal=false
 Type=Application
