From e0a029f305f199ad46f35d001931e9b531e1abcd Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=B6ren=20Tempel?= <soeren+git@soeren-tempel.net>
Date: Mon, 1 Jun 2020 00:27:18 +0200
Subject: [PATCH] CMakeLists.txt: Install pkgconfig files again

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 86282775..60f9087d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -49,7 +49,7 @@ find_package(libusb REQUIRED)
 
 ## Package configuration (pkg-config) on unix-based systems
 if (NOT WIN32 AND NOT CMAKE_CROSSCOMPILING)
-    #add_subdirectory(cmake/pkgconfig)
+    add_subdirectory(cmake/pkgconfig)
     find_package(PkgConfig)
     pkg_check_modules(GTK3 gtk+-3.0)
 endif ()
