--- a/client/Makefile.am	2020-05-31 01:43:58.000000000 -0400
+++ b/client/Makefile.am	2020-06-06 11:31:53.051972845 -0400
@@ -113,6 +113,7 @@
 
 boinc_clientdir = $(bindir)
 
+if ANDROID
 if OS_ARM_LINUX
 boinc_client_LDADD += libwhetneon.a libwhetvfp.a
 noinst_LIBRARIES = libwhetneon.a libwhetvfp.a
@@ -122,6 +123,7 @@
 libwhetvfp_a_SOURCES = whetstone.cpp
 libwhetvfp_a_CXXFLAGS = $(boinc_client_CXXFLAGS) -DANDROID_VFP -mfloat-abi=softfp -mfpu=vfp
 endif
+endif
 
 switcher_SOURCES = switcher.cpp
 switcher_LDFLAGS = $(AM_LDFLAGS) -L../lib
--- a/client/whetstone.cpp	2018-01-14 11:07:52.000000000 +0100
+++ b/client/whetstone.cpp	2018-02-26 20:49:52.185904752 +0100
@@ -290,6 +290,8 @@
     return 0;
 }
 
+#ifdef ANDROID
 #if defined(ANDROID_NEON) || defined(ANDROID_VFP)
   }
 #endif // namespace closure
+#endif
--- a/configure.ac	2020-05-31 01:43:58.000000000 -0400
+++ b/configure.ac	2020-06-06 11:41:14.259984481 -0400
@@ -1051,6 +1051,7 @@
 dnl or OS2
 AM_CONDITIONAL(OS_OS2, [echo $host_os | grep '^os2' > /dev/null])
 AM_CONDITIONAL(OS_ARM_LINUX, [echo $host_alias | grep '^arm-linux' > /dev/null])
+AM_CONDITIONAL([ANDROID],  [AS_CASE([$host_os],[linux-android*],[true], [false])])
 
 dnl Whether to build fcgi components
 AM_CONDITIONAL(ENABLE_FCGI,[test "${enable_fcgi}" = yes])
