 client/whetstone.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/client/whetstone.cpp b/client/whetstone.cpp
index 230efb1210..6e2472ca9e 100644
--- a/client/whetstone.cpp 
+++ b/client/whetstone.cpp 
@@ -290,6 +290,8 @@ int whetstone(double& flops, double& cpu_time, double min_cpu_time) { 
     return 0;
 }

+#ifdef ANDROID 
 #if defined(ANDROID_NEON) || defined(ANDROID_VFP)
   }
 #endif // namespace closure
+#endif 
--  
2.39.1 
