Add objective c++ program and check for gtk2

--- a/configure.ac
+++ b/configure.ac
@@ -39,6 +39,7 @@ AC_PROG_CC
 AC_PROG_CXX
 AC_PROG_F77
 AC_PROG_CPP
+AC_PROG_OBJCXX
 AC_PROG_MAKE_SET
 SAH_LINKS
 AC_LANG_PUSH(C)
@@ -744,6 +745,8 @@ AM_CONDITIONAL(ENABLE_LIBRARIES, [test "
 AM_CONDITIONAL(INSTALL_HEADERS, [test "${enable_install_headers}" = yes])
 AM_CONDITIONAL(HAVE_CUDA_LIB, [test "${enable_client}" = yes -a -f ./coprocs/CUDA/posix/${boinc_platform}/libcudart.so])
 
+PKG_CHECK_MODULES([GTK2], [gtk+-2.0])
+
 dnl ======================================================================
 dnl some more vodoo required for building portable client-binary (client, clientgui)
 dnl ======================================================================

