--- clientgui/AsyncRPC.cpp	2015-09-29 11:16:41.000000000 +0200
+++ clientgui/AsyncRPC.cpp	2015-10-13 16:17:54.313343388 +0200
@@ -19,7 +19,7 @@
 #pragma implementation "AsyncRPC.h"
 #endif
 
-#if !(defined(_WIN32) || (defined(__WXMAC__) && (MAC_OS_X_VERSION_MIN_REQUIRED < MAC_OS_X_VERSION_10_4)))
+#if !(defined(_WIN32) || (defined(__WXMAC__) && (MAC_OS_X_VERSION_MIN_REQUIRED < MAC_OS_X_VERSION_10_4))) && defined(__GLIBC__)
 #include <xlocale.h>
 #endif
 
