diff --git absl/debugging/internal/stacktrace_config.h absl/debugging/internal/stacktrace_config.h
index d4e8480..ee1ba54 100644
--- absl/debugging/internal/stacktrace_config.h
+++ absl/debugging/internal/stacktrace_config.h
@@ -40,7 +40,7 @@
 # elif defined(__aarch64__)
 #define ABSL_STACKTRACE_INL_HEADER \
     "absl/debugging/internal/stacktrace_aarch64-inl.inc"
-# elif defined(__arm__)
+# elif defined(__arm__) && defined(__GLIBC__)
 // Note: When using glibc this may require -funwind-tables to function properly.
 #define ABSL_STACKTRACE_INL_HEADER \
   "absl/debugging/internal/stacktrace_generic-inl.inc"
