From 3f41e39c020d5b135e98fe025f259a5ac999129e Mon Sep 17 00:00:00 2001
From: Duncan Overbruck <mail@duncano.de>
Date: Mon, 20 Nov 2023 21:59:50 +0100
Subject: [PATCH] allow some missing syscalls for i386 with glibc

---
 sandbox-seccomp-filter.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/sandbox-seccomp-filter.c b/sandbox-seccomp-filter.c
index a2e009d..46db374 100644
--- a/sandbox-seccomp-filter.c
+++ b/sandbox-seccomp-filter.c
@@ -102,6 +102,9 @@ static const struct sock_filter preauth_ctrl[] = {
 #endif
 #ifdef __NR_accept /* not defined for __i386__ (linux) */
 	SC_ALLOW(accept),
+#endif
+#ifdef __NR_socketcall /* used for accept on __i386__ (linux) */
+	SC_ALLOW(socketcall),
 #endif
 	SC_ALLOW(fcntl),
 #ifdef __NR_fcntl64 /* only noted on arm */
@@ -190,6 +193,9 @@ static const struct sock_filter preauth_work[] = {
 	SC_ALLOW(fstat), /* for kutil_openlog logging */
 #ifdef __NR_newfstatat
 	SC_ALLOW(newfstatat), /* for kutil_openlog logging */
+#endif
+#ifdef __NR_statx
+	SC_ALLOW(statx), /* for kutil_openlog logging */
 #endif
 	SC_ALLOW(write),
 	SC_ALLOW(writev),
-- 
2.42.0

