--- configure
+++ configure
@@ -438,6 +438,12 @@ if [ ${HAVE_SECCOMP_FILTER} -eq 1 ]; then
 		arm*)
 			echo "#define SECCOMP_AUDIT_ARCH AUDIT_ARCH_ARM"
 			;;
+                ppc64le)
+                        echo "#define SECCOMP_AUDIT_ARCH AUDIT_ARCH_PPC64LE"
+                        ;;
+                ppc64)
+                        echo "#define SECCOMP_AUDIT_ARCH AUDIT_ARCH_PPC64"
+                        ;;
 	esac
 fi
 
