From d34e7aabd04018f70b15e54ccfffe93781f97bf6 Mon Sep 17 00:00:00 2001
From: Eric Koegel <eric.koegel@gmail.com>
Date: Wed, 12 Nov 2014 19:21:37 +0300
Subject: [PATCH 04/12] trivial: ignore non-empty log dir on uninstall

---
 data/Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/data/Makefile.am b/data/Makefile.am
index 6b47655..abd9403 100644
--- a/data/Makefile.am
+++ b/data/Makefile.am
@@ -83,7 +83,7 @@ install-data-local: install-logrotate install-90-consolekit
 	$(MKDIR_P) $(DESTDIR)$(localstatedir)/log/ConsoleKit/
 
 uninstall-local:: uninstall-logrotate uninstall-90-consolekit
-	rmdir $(DESTDIR)$(localstatedir)/log/ConsoleKit/
+	rmdir --ignore-fail-on-non-empty $(DESTDIR)$(localstatedir)/log/ConsoleKit/
 
 LOGROTATE_CONFS = consolekit.logrotate
 
-- 
2.2.1

