Found on Arch User Repository, adapted for -p0


--- a/fbxkb.c	2006-12-18 23:47:52.000000000 +0200
+++ b/fbxkb.c	2012-01-15 22:27:30.571427088 +0200
@@ -378,10 +378,11 @@ read_kbd_description()
             g_assert((no >= 0) && (no < ngroups));
             if (group2info[no].sym != NULL) {
                 ERR("xkb group #%d is already defined\n", no);
+            } else {
+                group2info[no].sym = g_strdup(tok);
+                group2info[no].flag = sym2flag(tok);
+                group2info[no].name = XGetAtomName(dpy, kbd_desc_ptr->names->groups[no]);           
             }
-            group2info[no].sym = g_strdup(tok);
-            group2info[no].flag = sym2flag(tok);
-            group2info[no].name = XGetAtomName(dpy, kbd_desc_ptr->names->groups[no]);           
         }
         XFree(sym_name);
     }
--- a/Makefile.common	2004-10-16 01:18:59.000000000 +0300
+++ b/Makefile.common	2012-01-15 23:20:02.081020564 +0200

@@ -16,7 +16,7 @@ endif
 endif
 
 CC = gcc
-LIBS = $(shell pkg-config --libs gtk+-2.0 gdk-pixbuf-2.0 gdk-pixbuf-xlib-2.0) -L/usr/X11R6/lib  -lXmu
+LIBS = $(shell pkg-config --libs gtk+-2.0 gdk-pixbuf-2.0 gdk-pixbuf-xlib-2.0) -lX11 -L/usr/X11R6/lib  -lXmu
 INCS = $(shell pkg-config --cflags gtk+-2.0 gdk-pixbuf-2.0 gdk-pixbuf-xlib-2.0)
 CFLAGS = -O2    # overwriten by command line or env. variable
 CFLAGS += -Wall # always nice to have 
@@ -24,9 +24,6 @@ ifneq (,$(DEVEL))
 CFLAGS := -g -Wall 
 endif
 
-# -DGTK_DISABLE_DEPRECATED does not work yet
-CFLAGS += -g -DG_DISABLE_DEPRECATED -DGDK_DISABLE_DEPRECATED -DGTK_DISABLE_DEPRECATED
-
 %.o: %.c
 	$(CC) $(CFLAGS) $(INCS) -c $<
