From e139a679d7cc3f04ad001054a6b1815b7afc92e3 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jan=20Christian=20Gr=C3=BCnhage?=
 <jan.christian@gruenhage.xyz>
Date: Thu, 7 Oct 2021 18:04:15 +0200
Subject: [PATCH 1/3] don't run tests unless asked

---
 Makefile | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/Makefile b/Makefile
index 9c840c4..d4c241e 100644
--- a/Makefile
+++ b/Makefile
@@ -14,16 +14,12 @@ BEES_VERSION ?= $(shell git describe --always --dirty || echo UNKNOWN)
 
 DEFAULT_MAKE_TARGET ?= reallyall
 
-ifeq ($(DEFAULT_MAKE_TARGET),reallyall)
-	RUN_INSTALL_TESTS = test
-endif
-
 include Defines.mk
 
 default: $(DEFAULT_MAKE_TARGET)
 
 all: lib src scripts
-reallyall: all doc test
+reallyall: all doc
 
 clean: ## Cleanup
 	git clean -dfx -e localconf
@@ -55,7 +51,7 @@ install_tools: src
 	install -Dm755 bin/fiewalk $(DESTDIR)$(PREFIX)/sbin/fiewalk
 
 install_bees: ## Install bees + libs
-install_bees: src $(RUN_INSTALL_TESTS)
+install_bees: src
 	install -Dm755 bin/bees	$(DESTDIR)$(LIBEXEC_PREFIX)/bees
 
 install_scripts: ## Install scipts
-- 
2.35.1

