From 0a9982bc98fb3a664247c554d7b051eec2f8c359 Mon Sep 17 00:00:00 2001
From: Rohit Vighne <rohit.vighne@gmail.com>
Date: Fri, 7 Aug 2026 15:40:31 -0400
Subject: [PATCH 2/2] move sbin/* into bin/*

---
 Makefile                  | 2 +-
 scripts/beesd@.service.in | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 1918372..5389aa5 100644
--- a/Makefile
+++ b/Makefile
@@ -53,11 +53,11 @@ install_bees: ## Install bees + libs
 install_bees: src $(RUN_INSTALL_TESTS)
 	install -Dm755 bin/bees	$(DESTDIR)$(LIBEXEC_PREFIX)/bees
 
 install_scripts: ## Install scipts
 install_scripts: scripts
-	install -Dm755 scripts/beesd $(DESTDIR)$(PREFIX)/sbin/beesd
+	install -Dm755 scripts/beesd $(DESTDIR)$(PREFIX)/bin/beesd
 	install -Dm644 scripts/beesd.conf.sample $(DESTDIR)$(ETC_PREFIX)/bees/beesd.conf.sample
 ifneq ($(SYSTEMD_SYSTEM_UNIT_DIR),)
 	install -Dm644 scripts/beesd@.service $(DESTDIR)$(SYSTEMD_SYSTEM_UNIT_DIR)/beesd@.service
 endif
 
diff --git a/scripts/beesd@.service.in b/scripts/beesd@.service.in
index 32a6f85..78727a4 100644
--- a/scripts/beesd@.service.in
+++ b/scripts/beesd@.service.in
@@ -3,11 +3,11 @@ Description=Bees (%i)
 Documentation=https://github.com/Zygo/bees
 After=sysinit.target
 
 [Service]
 Type=simple
-ExecStart=@PREFIX@/sbin/beesd --no-timestamps %i
+ExecStart=@PREFIX@/bin/beesd --no-timestamps %i
 CPUAccounting=true
 CPUSchedulingPolicy=batch
 CPUWeight=12
 IOSchedulingClass=idle
 IOSchedulingPriority=7
-- 
2.55.0

