--- a/lib/task.cc
+++ b/lib/task.cc
@@ -289,7 +289,6 @@
 		lock.unlock();
 
 		char buf[24] = { 0 };
-		DIE_IF_MINUS_ERRNO(pthread_getname_np(pthread_self(), buf, sizeof(buf)));
 		DIE_IF_MINUS_ERRNO(pthread_setname_np(pthread_self(), m_title.c_str()));
 
 		TaskStatePtr this_task = shared_from_this();
--- a/src/bees-trace.cc
+++ b/src/bees-trace.cc
@@ -136,10 +136,6 @@
 	// OK try the pthread name next.
 	char buf[24];
 	memset(buf, '\0', sizeof(buf));
-	int err = pthread_getname_np(pthread_self(), buf, sizeof(buf));
-	if (err) {
-		return string("pthread_getname_np: ") + strerror(err);
-	}
 	buf[sizeof(buf) - 1] = '\0';
 
 	// thread_getname_np returns process name
