--- Makefile_bak	2021-08-22 22:58:00.224045573 -0700
+++ Makefile	2021-08-22 23:08:13.971375382 -0700
@@ -49,18 +49,15 @@
 
 scripts: scripts/beesd scripts/beesd@.service
 
-install_tools: ## Install support tools + libs
-install_tools: src
-	install -Dm755 bin/fiemap $(DESTDIR)$(PREFIX)/bin/fiemap
-	install -Dm755 bin/fiewalk $(DESTDIR)$(PREFIX)/sbin/fiewalk
-
 install_bees: ## Install bees + libs
 install_bees: src $(RUN_INSTALL_TESTS)
-	install -Dm755 bin/bees	$(DESTDIR)$(LIBEXEC_PREFIX)/bees
+	install -Dm755 bin/bees	$(DESTDIR)$(PREFIX)/bin/bees
+	install -Dm755 bin/fiemap $(DESTDIR)$(PREFIX)/bin/fiemap
+	install -Dm755 bin/fiewalk $(DESTDIR)$(PREFIX)/bin/fiewalk
 
 install_scripts: ## Install scipts
 install_scripts: scripts
-	install -Dm755 scripts/beesd $(DESTDIR)$(PREFIX)/sbin/beesd
+	install -Dm755 scripts/beesd $(DESTDIR)$(PREFIX)/bin/beesd
 	install -Dm644 scripts/beesd.conf.sample $(DESTDIR)/$(ETC_PREFIX)/bees/beesd.conf.sample
 ifneq ($(SYSTEMD_SYSTEM_UNIT_DIR),)
 	install -Dm644 scripts/beesd@.service $(DESTDIR)$(SYSTEMD_SYSTEM_UNIT_DIR)/beesd@.service
