--- src/config_pepperflash.c	2015-07-05 12:34:13.000000000 -0300
+++ src/config_pepperflash.c	2015-07-25 18:59:05.692993023 -0300
@@ -39,6 +39,7 @@
     "/usr/lib64/chromium/PepperFlash",          // chromium-pepper-flash (Old Build) (CentOS/Read Hat/Fedora)
     "/opt/chromium/PepperFlash",                // chromium-pepper-flash (New Build) (CentOS/Read Hat/Fedora)
     "/usr/lib/pepflashplugin-installer",        // pepflashplugin-installer (Ubuntu)
+	"/usr/lib/chromium-pepper-flash", 			// chromium-pepper-flash (Void Linux)
     NULL,
 };
 
