--- a/src/ppb_video_decoder.c.orig        2018-05-03 22:08:33.026677145 +0200
+++ b/src/ppb_video_decoder.c     2018-05-03 22:09:26.176206079 +0200
@@ -542,9 +542,9 @@
         goto err;
     }

-    if (vd->avcodec->capabilities & CODEC_CAP_TRUNCATED) {
-        trace_info("%s, codec have CODEC_CAP_TRUNCATED\n", __func__);
-        vd->avctx->flags |= CODEC_FLAG_TRUNCATED;
+    if (vd->avcodec->capabilities & AV_CODEC_CAP_TRUNCATED) {
+        trace_info("%s, codec have AV_CODEC_CAP_TRUNCATED\n", __func__);
+        vd->avctx->flags |= AV_CODEC_FLAG_TRUNCATED;
     }

     vd->avctx->opaque = vd;

