From ecebb21e9d64081ca436aebcdb32ba0960ac8a60 Mon Sep 17 00:00:00 2001
From: Jens Georg <mail@jensge.org>
Date: Thu, 4 Apr 2019 20:28:07 +0200
Subject: [PATCH] build: Properly export common resources

Fixes #3
---
 src/common/meson.build | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/common/meson.build b/src/common/meson.build
index 68b1ca5..75b82db 100644
--- a/src/common/meson.build
+++ b/src/common/meson.build
@@ -1,7 +1,9 @@
 util_resource = gnome.compile_resources('common-resource',
     join_paths(resource_data, 'org.gupnp.Tools.Common.gresource.xml'),
+    c_name : 'gupnp_tools_common',
+    export : true,
     source_dir : resource_data,
-    extra_args : ['--manual-register', '--c-name', 'gupnp_tools_common']
+    extra_args : ['--manual-register']
 )
 
 libutil = static_library('util', ['icons.c', 'icons.h', 'pretty-print.c', 'pretty-print.h'] + util_resource,
-- 
2.18.1


