From e5781feb4403e7f2de19f7c027cda5e5ef092a5c Mon Sep 17 00:00:00 2001
From: Nathan Owens <ndowens@artixlinux.org>
Date: Fri, 1 Jan 2021 15:46:05 -0600
Subject: [PATCH] fix test for musl

---
 test/memfake.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git test/memfake.c test/memfake.c
index 1c4fa94..bc2da8e 100644
--- test/memfake.c
+++ test/memfake.c
@@ -22,19 +22,21 @@
 #include <stdlib.h>
 #include <time.h>
 #include <unistd.h>
+#include <assert.h>
 
 int main(int argc, char **argv)
 {
-    uint MB_to_allocate = 1;
+    size_t MB_to_allocate = 1;
     bool add_randomness = false;
 
-    if (argc >= 2) MB_to_allocate = atoi(argv[1]);
-    if (argc >= 3) add_randomness = atoi(argv[2]);
+    if (argc >= 2) MB_to_allocate = strtoul(argv[1], 0, 10);
+    if (argc >= 3) add_randomness = strtoul(argv[2], 0, 10);
     if (argc >= 4) return 1;
 
     size_t array_size = MB_to_allocate * 1024 * 1024 / sizeof(int);
 
     int* array = calloc(array_size, sizeof(int));
+    assert(array != NULL);
 
     // Fill half with random values and leave an half of zeroes, if asked to
     if (add_randomness) {
-- 
2.30.0

