diff --git a/scanroutines.c b/scanroutines.c
index 2c5f33a..edc7c29 100644
--- a/scanroutines.c
+++ b/scanroutines.c
@@ -20,6 +20,8 @@
     along with this library.  If not, see <http://www.gnu.org/licenses/>.
 */
 
+#define _GNU_SOURCE 1
+#include <sys/types.h>
 #include <assert.h>
 #include <stdbool.h>
 
diff --git a/targetmem.h b/targetmem.h
index b62a4f2..25b5c9c 100644
--- a/targetmem.h
+++ b/targetmem.h
@@ -24,6 +24,9 @@
 #ifndef TARGETMEM_H
 #define TARGETMEM_H
 
+#define _GNU_SOURCE 1
+#include <sys/types.h>
+
 #include <assert.h>
 #include <string.h>
 #include <stdlib.h>

