--- Makefile.in     2016-04-15 19:56:08.257716796 -0400
+++ Makefile.in     2016-04-15 19:56:03.663451813 -0400
@@ -3088,12 +3088,6 @@
    uninstall-local uninstall-man uninstall-man1


-compute.1: compute
-   PATH='$(abs_top_builddir)$(PATH_SEPARATOR)'$$PATH \
-      $(abs_top_srcdir)/man/help2man --include=$(top_srcdir)/man/compute.x compute -o $@-t
-   chmod u=rw,go=r $@-t
-   mv -f $@-t $@
-
 # Verify that all source files using _() are listed in po/POTFILES.in.
 # The idea is to run this before making pretests, as well as official
 # releases, so that translators will be sure to have all the messages.
