--- ocaml-4.08.0/asmcomp/emitaux.ml
+++ ocaml-4.08.0/asmcomp/emitaux.ml
@@ -213,6 +213,7 @@ let emit_frames a =
     | None -> a.efa_word 0
     end
   in
+  a.efa_align Arch.size_addr;
   a.efa_word (List.length !frame_descriptors);
   List.iter emit_frame !frame_descriptors;
   Label_table.iter emit_debuginfo debuginfos;
--- ocaml-4.08.0/runtime/power.S
+++ ocaml-4.08.0/runtime/power.S
@@ -642,6 +642,7 @@ caml_system__code_end:
 /* Frame table */
 
         .section ".data"
+        .align  wordalign
         .globl  caml_system__frametable
         .type   caml_system__frametable, @object
 caml_system__frametable:
