From: Nathan Owens <ndowens04@gmail.com>
On musl-based systems, <string.h> causes redefintions
---
diff --git posixovl.c posixovl.c
index ae64f21..04e41fb 100644
--- a/posixovl.c
+++ b/posixovl.c
@@ -30,7 +30,9 @@
 #include <stdbool.h>
 #include <stdio.h>
 #include <stdlib.h>
+#if defined(__GLIBC__)
 #include <string.h>
+#endif
 #include <unistd.h>
 #ifdef HAVE_XATTR_XATTR_H
 #	include <attr/xattr.h>
