diff -ruN a/SPIRV/CMakeLists.txt b/SPIRV/CMakeLists.txt
--- a/SPIRV/CMakeLists.txt	2022-12-06 23:34:33.000000000 +0100
+++ b/SPIRV/CMakeLists.txt	2022-12-16 12:15:03.231070779 +0100
@@ -100,7 +100,7 @@
     target_link_libraries(SPIRV PRIVATE MachineIndependent SPIRV-Tools-opt)
     target_include_directories(SPIRV PUBLIC
         $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../External>
-        $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/External>)
+    )
 else()
     target_link_libraries(SPIRV PRIVATE MachineIndependent)
 endif()
diff -ruN a/StandAlone/CMakeLists.txt b/StandAlone/CMakeLists.txt
--- a/StandAlone/CMakeLists.txt	2022-12-06 23:34:33.000000000 +0100
+++ b/StandAlone/CMakeLists.txt	2022-12-16 12:15:20.527812776 +0100
@@ -83,7 +83,7 @@
 target_link_libraries(glslangValidator ${LIBRARIES})
 target_include_directories(glslangValidator PUBLIC
     $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../External>
-    $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/External>)
+)
 
 if(ENABLE_OPT)
     target_include_directories(glslangValidator
