poppler-0.72 renamed GooString::getCString() to c_str()

diff --git a/src/poppler-core/destname.cc b/src/poppler-core/destname.cc
index 7177990..200c267 100644
--- src/poppler-core/destname.cc
+++ src/poppler-core/destname.cc
@@ -32,7 +32,7 @@
 
 inline std::string goo_to_string (GooString *goo)
 {
-  return std::string {goo->getCString (),
+  return std::string {goo->c_str (),
       static_cast<std::string::size_type>(goo->getLength ())};
 }
 

