diff --git src/poppler-core/destname.cc src/poppler-core/destname.cc
index 33e05e2..5f8d4f7 100644
--- src/poppler-core/destname.cc
+++ src/poppler-core/destname.cc
@@ -82,7 +82,7 @@ std::string poppler_core::build_destname (const std::string &name,
       if (link_dest->isPageRef ())
         {
           Ref page_ref = link_dest->getPageRef ();
-          pagenum = doc->findPage (page_ref.num, page_ref.gen);
+          pagenum = doc->findPage (page_ref);
         }
       else
         {

