--- lightcrafts/coprocesses/dcraw/GNUmakefile	2015-11-22 15:22:44.000000000 +0100
+++ lightcrafts/coprocesses/dcraw/GNUmakefile	2015-12-02 17:39:48.316543552 +0100
@@ -6,7 +6,7 @@
 # Uncomment to compile in debug mode.
 #DEBUG:=		true
 
-EXEC_EXTRA_CFLAGS:=	-fopenmp
+EXEC_EXTRA_CFLAGS:=
 EXEC_WINDOWS_CFLAGS:=	-static
 EXEC_EXTRA_DEFINES:=	-DNODEPS -DLIGHTZONE
 EXEC_LINUX_DEFINES:=	-Dfgetc=getc_unlocked
--- lightcrafts/coprocesses/dcraw/dcraw_lz.c	2015-11-22 15:22:44.000000000 +0100
+++ lightcrafts/coprocesses/dcraw/dcraw_lz.c	2015-12-02 17:48:03.900579281 +0100
@@ -101,7 +101,6 @@
 #endif
 
 #ifdef _OPENMP
-#include <omp.h>
 #define uf_omp_get_thread_num() omp_get_thread_num()
 #define uf_omp_get_num_threads() omp_get_num_threads()
 #else
@@ -4795,7 +4794,7 @@
 #if defined(_STATIC_BUFFER)
    static char buffer [TS*TS*(8*11+6)] __attribute__((aligned(64)));
 #else
-   char *buffer;
+   char *buffer = NULL;
 #endif
 
   if (verbose)
@@ -4828,9 +4827,13 @@
   {
   cielab3 (0,0);
   border_interpolate(6);
+#if ! defined(_FIXED_NDIR)
   ndir = 4 << (passes > 1);
+#endif
+#if ! defined(_STATIC_BUFFER)
   buffer = (char *) malloc (TS*TS*(ndir*11+6));
   merror (buffer, "xtrans_interpolate()");
+#endif
   rgb  = (ushort(*)[TS][TS][3]) buffer;
   lab  = (short (*)    [TS][3])(buffer + TS*TS*(ndir*6));
   drv  = (float (*)[TS][TS])   (buffer + TS*TS*(ndir*6+6));
@@ -4885,8 +4888,10 @@
 #endif
 
 #if ! defined(_STATIC_BUFFER)
-  buffer = (char *) malloc (TS*TS*(ndir*11+6));
-  merror (buffer, "xtrans_interpolate()");
+  if (buffer == NULL) {
+    buffer = (char *) malloc (TS*TS*(ndir*11+6));
+    merror (buffer, "xtrans_interpolate()");
+  }
 #endif
 
   rgb  = (ushort(*)[TS][TS][3]) buffer;
--- lightcrafts/jnisrc/dcraw/dcrawUtils.cpp	2015-11-22 15:22:44.000000000 +0100
+++ lightcrafts/jnisrc/dcraw/dcrawUtils.cpp	2015-12-02 17:54:53.663608822 +0100
@@ -13,7 +13,6 @@
         name4(Java_,com_lightcrafts_utils_DCRaw,_,method)
 
-#include<limits.h>
+#include <limits.h>
-#include<omp.h>
 
 JNIEXPORT void JNICALL DCRaw_METHOD(interpolateGreen)
 ( JNIEnv *env, jclass cls,
--- lightcrafts/jnisrc/dcraw/GNUmakefile	2015-11-22 15:22:44.000000000 +0100
+++ lightcrafts/jnisrc/dcrawGNUmakefile	2015-12-02 17:58:22.962623911 +0100
@@ -13,7 +13,7 @@
   JNI_EXTRA_LINK:=      -lstdc++
 endif
 
-JNI_EXTRA_CFLAGS+=	-fopenmp
+#JNI_EXTRA_CFLAGS+=	-fopenmp
 
 JNI_EXTRA_DEFINES:=	-DNO_JPEG -DNO_LCMS
 JNI_LINUX_DEFINES:=	-Dfgetc=getc_unlocked
