Fixes bad macro checks, breaking altivec build.

--- a/lightcrafts/jnisrc/jai/bilateralFilter_sse.c
+++ b/lightcrafts/jnisrc/jai/bilateralFilter_sse.c
@@ -188,7 +188,7 @@ static inline void reset_denormals(csr_context oldMXCSR) {
 
 #endif // defined(__i386__) || defined(__x86_64__)
 
-#if defined(__ppc__) || defined(__ppc64__)
+#if defined(__powerpc__) || defined(__powerpc64__)
 #include <altivec.h>
 
 static inline vFloat loadUnalignedFloat( float *target )
@@ -253,7 +253,7 @@ static void reset_denormals( csr_context oldJavaMode ) {
     vec_mtvscr ( oldJavaMode );
 }
 
-#endif // defined(__ppc__) || defined(__ppc64__)
+#endif // defined(__powerpc__) || defined(__powerpc64__)
 #endif // defined(__SSE2__) || defined (__VEC__)
 
 #if USE_VECTOR
