--- a/src/DoConfig
+++ b/src/DoConfig
@@ -571,7 +571,7 @@
       # icc just gives warnings for bad command line args
       RemoveProg("$name") and return 0;
    }
-   system("./$name") and RemoveProg("$name") and return 0;
+   system("\$RUNTEST ./$name") and RemoveProg("$name") and return 0;
    RemoveProg("$name") and return 1;
 }
 
@@ -763,7 +763,7 @@
    die "Goodbye!\n";
 }
 
-my $config_info = `./GenConfigInfo`;
+my $config_info = `\$RUNTEST ./GenConfigInfo`;
 
 my ($compiler_name, $language_standard, $cpu_type, $os_name);
 ($compiler_name, $language_standard, $cpu_type, $os_name) =
--- a/src/MakeCheckFeatures
+++ b/src/MakeCheckFeatures
@@ -14,7 +14,7 @@
 
    if test -f CheckFeatures
    then
-      if ./CheckFeatures
+      if $RUNTEST ./CheckFeatures
       then
 	 echo "[yes]"
 	 echo "#ifndef NTL_HAVE_$f" >  "../include/NTL/HAVE_$f.h"
--- a/src/mfile
+++ b/src/mfile
@@ -354,7 +354,7 @@
 setup1: 
 	$(COMPILE) MakeDescAux.cpp
 	$(LINK) -o MakeDesc MakeDesc.cpp MakeDescAux.o $(LDLIBS)
-	./MakeDesc
+	$(RUNTEST) ./MakeDesc
 	mv mach_desc.h ../include/NTL/mach_desc.h
 
 
@@ -378,7 +378,7 @@
 
 setup3:
 	$(LINK) $(GMP_OPT_INCDIR) -o gen_gmp_aux gen_gmp_aux.cpp $(GMP_OPT_LIBPATH) $(GMP_OPT_LIB) $(LDLIBS)
-	./gen_gmp_aux > ../include/NTL/gmp_aux.h 
+	$(RUNTEST) ./gen_gmp_aux > ../include/NTL/gmp_aux.h 
 	$(LINK) $(GF2X_OPT_INCDIR) -o gf2x_version_1_2_or_later_required gf2x_version_1_2_or_later_required.cpp $(GF2X_OPT_LIBPATH) $(GF2X_OPT_LIB) $(LDLIBS) 
 
 # setup4 runs the wizard
