--- packages/insomnia-app/scripts/build.js	2021-03-09 22:33:38.000000000 -0600
+++ packages/insomnia-app/scripts/build.js	2021-03-10 11:13:42.894148116 -0600
@@ -43,11 +43,6 @@
   console.log(`[build] npm: ${childProcess.spawnSync('npm', ['--version']).stdout}`.trim());
   console.log(`[build] node: ${childProcess.spawnSync('node', ['--version']).stdout}`.trim());
 
-  if (process.version.indexOf('v12.') !== 0) {
-    console.log('[build] Node v12.x.x is required to build');
-    process.exit(1);
-  }
-
   // Remove folders first
   console.log('[build] Removing existing directories');
   await emptyDir(buildFolder);
