--- a/libpurple/network.c	2018-03-06 07:00:49.000000000 +0100
+++ b/libpurple/network.c	2020-02-11 04:30:20.163644344 +0100
@@ -939,8 +939,16 @@
 #if NM_CHECK_VERSION(0,8,992)
 		case NM_STATE_DISCONNECTING:
 #endif
+#if NM_CHECK_VERSION(0,8,992)
+			if (prev != NM_STATE_CONNECTED_LOCAL &&
+				prev != NM_STATE_CONNECTED_SITE &&
+				prev != NM_STATE_CONNECTED_GLOBAL &&
+				prev != NM_STATE_UNKNOWN)
+				break;
+#else
 			if (prev != NM_STATE_CONNECTED && prev != NM_STATE_UNKNOWN)
 				break;
+#endif
 			if (ui_ops != NULL && ui_ops->network_disconnected != NULL)
 				ui_ops->network_disconnected();
 			break;
