--- a/meson.build	2026-04-27 02:02:51.000000000 +0800
+++ b/meson.build	2026-05-03 11:00:04.049346792 +0800
@@ -265,7 +265,7 @@
 )

 neatvnc = library(
-	'neatvnc',
+	'neatvnc1',
 	[],
 	version: meson.project_version(),
 	dependencies: dependencies,
@@ -297,13 +297,14 @@
 	subdir('test')
 endif

-install_headers('include/neatvnc.h')
+install_headers('include/neatvnc.h', subdir: meson.project_name() + '1')

 pkgconfig = import('pkgconfig')
 pkgconfig.generate(
 	libraries: neatvnc,
 	version: meson.project_version(),
-	filebase: meson.project_name(),
+	filebase: meson.project_name() + '1',
+	subdirs: meson.project_name() + '1',
 	name: meson.project_name(),
 	description: 'A Neat VNC server library'
 )
