--- a/openvdb/openvdb/CMakeLists.txt
+++ b/openvdb/openvdb/CMakeLists.txt
@@ -117,7 +117,7 @@ if(OPENVDB_CORE_SHARED AND NOT Boost_USE
   set(Boost_USE_STATIC_LIBS OFF)
 endif()
 
-find_package(Boost ${MINIMUM_BOOST_VERSION} REQUIRED COMPONENTS iostreams system)
+find_package(Boost ${MINIMUM_BOOST_VERSION} REQUIRED COMPONENTS iostreams)
 if(OPENVDB_FUTURE_DEPRECATION AND FUTURE_MINIMUM_BOOST_VERSION)
   # The X.Y.Z boost version value isn't available until CMake 3.14
   set(FULL_BOOST_VERSION "${Boost_MAJOR_VERSION}.${Boost_MINOR_VERSION}.${Boost_SUBMINOR_VERSION}")
@@ -251,8 +251,7 @@ endif()
 # other headers.
 
 list(APPEND OPENVDB_CORE_DEPENDENT_LIBS
-  Boost::iostreams
-  Boost::system)
+  Boost::iostreams)
 
 if(WIN32)
   # Boost headers contain #pragma commands on Windows which causes Boost
--- a/cmake/FindOpenVDB.cmake
+++ b/cmake/FindOpenVDB.cmake
@@ -496,7 +496,7 @@ if(NOT OPENVDB_USE_STATIC_LIBS AND NOT B
   set(Boost_USE_STATIC_LIBS OFF)
 endif()
 
-find_package(Boost REQUIRED COMPONENTS iostreams system)
+find_package(Boost REQUIRED COMPONENTS iostreams)
 
 # Add deps for pyopenvdb
 # @todo track for numpy
@@ -673,7 +673,6 @@ endif()
 
 set(_OPENVDB_VISIBLE_DEPENDENCIES
   Boost::iostreams
-  Boost::system
 )
 
 if(OpenVDB_USES_IMATH_HALF)
