--- src/Makefile.am.orig	2015-10-22 09:13:22.528361894 +0200
+++ src/Makefile.am	2015-10-22 09:13:45.520587919 +0200
@@ -31,12 +31,3 @@ cmdline.c cmdline.h: cmdline.ggo Makefil
 
 BUILT_SOURCES = cmdline.c cmdline.h
 MAINTAINERCLEANFILES = $(BUILT_SOURCES)
-
-dist_man_MANS = u2f-host.1
-DISTCLEANFILES = $(dist_man_MANS)
-
-u2f-host.1: $(srcdir)/u2f-host.c $(srcdir)/cmdline.ggo $(top_srcdir)/configure.ac
-	$(AM_V_GEN)$(HELP2MAN) \
-		--output=$@ $(builddir)/u2f-host$(EXEEXT) \
-		--name="Yubico Universal 2nd Factor (U2F) Host Tool" \
-		--no-info
