diff --git a/build.zig b/build.zig
index 231c2e6..f92c70d 100644
--- a/build.zig
+++ b/build.zig
@@ -6,7 +6,7 @@ const std = @import("std");
 pub fn build(b: *std.Build) void {
     const target = b.standardTargetOptions(.{});
     const optimize = b.standardOptimizeOption(.{
-        .preferred_optimize_mode = .ReleaseFast,
+        //.preferred_optimize_mode = .ReleaseFast,
     });
 
     const pie = b.option(bool, "pie", "Build with PIE support (by default false)") orelse false;
