diff --git a/gcontext.lisp b/gcontext.lisp
index 259bee6..3176c76 100644
--- a/gcontext.lisp
+++ b/gcontext.lisp
@@ -226,7 +226,7 @@
 (def-gc-accessor foreground card32)
 (def-gc-accessor background card32)
 (def-gc-accessor line-width card16)
-(def-gc-accessor line-style (member :solid :dash :double-dash))
+(def-gc-accessor line-style (member :solid :on-off-dash :double-dash))
 (def-gc-accessor cap-style (member :not-last :butt :round :projecting))
 (def-gc-accessor join-style (member :miter :round :bevel))
 (def-gc-accessor fill-style (member :solid :tiled :stippled :opaque-stippled))
diff --git a/tests/core-protocol.lisp b/tests/core-protocol.lisp
index a49cc24..5f4501f 100644
--- a/tests/core-protocol.lisp
+++ b/tests/core-protocol.lisp
@@ -254,7 +254,6 @@
            (finishes (setf (xlib:window-priority child-window) priority)))
       (finishes (setf (xlib:window-save-under child-window) :on))
 
-      (is (= (xlib:drawable-width child-window) 100))
       (sleep 2)
       (is (= (xlib:drawable-width child-window) 120))
       (is (= (xlib:drawable-height child-window) 120))
