--- a/test/build-id.sh
+++ b/test/build-id.sh
@@ -12,7 +12,7 @@ readelf -n $t/exe | grep -qv 'GNU.*0x00000010.*NT_GNU_BUILD_ID'
 
 clang -o $t/exe $t/a.c -fuse-ld=`pwd`/../mold -Wl,-build-id=uuid
 readelf -nW $t/exe |
-  grep -Pq 'GNU.*0x00000010.*NT_GNU_BUILD_ID.*Build ID: ............4...[89abcdef]'
+  rg -Pq 'GNU.*0x00000010.*NT_GNU_BUILD_ID.*Build ID: ............4...[89abcdef]'
 
 clang -o $t/exe $t/a.c -fuse-ld=`pwd`/../mold -Wl,-build-id=md5
 readelf -n $t/exe | grep -q 'GNU.*0x00000010.*NT_GNU_BUILD_ID'
--- a/test/copyrel-relro.sh
+++ b/test/copyrel-relro.sh
@@ -22,7 +22,7 @@ EOF
 clang -fuse-ld=`pwd`/../mold $t/a.o $t/b.so -o $t/exe
 readelf -a $t/exe > $t/log
 
-grep -Pqz '(?s)\[(\d+)\] .dynbss.rel.ro .* \1 readonly' $t/log
-grep -Pqz '(?s)\[(\d+)\] .dynbss .* \1 readwrite' $t/log
+rg -PUq '(?s)\[(\d+)\] .dynbss.rel.ro .* \1 readonly' $t/log
+rg -PUq '(?s)\[(\d+)\] .dynbss .* \1 readwrite' $t/log
 
 echo OK
--- a/test/demangle.sh
+++ b/test/demangle.sh
@@ -16,10 +16,10 @@ EOF
 grep -q 'undefined symbol: .*: _Z3fooii' $t/log
 
 ! clang -fuse-ld=`pwd`/../mold -o $t/exe $t/a.o -Wl,-demangle 2> $t/log || false
-grep -Pq 'undefined symbol: .*: foo\(int, int\)' $t/log
+rg -Pq 'undefined symbol: .*: foo\(int, int\)' $t/log
 
 ! clang -fuse-ld=`pwd`/../mold -o $t/exe $t/a.o 2> $t/log || false
-grep -Pq 'undefined symbol: .*: foo\(int, int\)' $t/log
+rg -Pq 'undefined symbol: .*: foo\(int, int\)' $t/log
 
 cat <<EOF | clang -c -o $t/b.o -xc -
 extern int Pi;
--- a/test/dt_init.sh
+++ b/test/dt_init.sh
@@ -18,15 +18,15 @@ EOF
 clang -fuse-ld=`pwd`/../mold -o $t/exe $t/a.o
 readelf -a $t/exe > $t/log
 
-grep -Pqz '(?s)\(INIT\)\s+0x([0-9a-f]+)\b.*\1\s+0 FUNC    GLOBAL HIDDEN\s+\d+ _init\b' $t/log
+rg -PUq '(?s)\(INIT\)\s+0x([0-9a-f]+)\b.*\1\s+0 FUNC    GLOBAL HIDDEN\s+\d+ _init\b' $t/log
 
-grep -Pqz '(?s)\(FINI\)\s+0x([0-9a-f]+)\b.*\1\s+0 FUNC    GLOBAL HIDDEN\s+\d+ _fini\b' $t/log
+rg -PUq '(?s)\(FINI\)\s+0x([0-9a-f]+)\b.*\1\s+0 FUNC    GLOBAL HIDDEN\s+\d+ _fini\b' $t/log
 
 clang -fuse-ld=`pwd`/../mold -o $t/exe $t/a.o -Wl,-init,init -Wl,-fini,fini
 readelf -a $t/exe > $t/log
 
-grep -Pqz '(?s)\(INIT\)\s+0x([0-9a-f]+)\b.*\1\s+0 NOTYPE  GLOBAL DEFAULT\s+\d+ init\b' $t/log
+rg -PUq '(?s)\(INIT\)\s+0x([0-9a-f]+)\b.*\1\s+0 NOTYPE  GLOBAL DEFAULT\s+\d+ init\b' $t/log
 
-grep -Pqz '(?s)\(FINI\)\s+0x([0-9a-f]+)\b.*\1\s+0 NOTYPE  GLOBAL DEFAULT\s+\d+ fini\b' $t/log
+rg -PUq '(?s)\(FINI\)\s+0x([0-9a-f]+)\b.*\1\s+0 NOTYPE  GLOBAL DEFAULT\s+\d+ fini\b' $t/log
 
 echo OK
--- a/test/dt_needed.sh
+++ b/test/dt_needed.sh
@@ -23,7 +23,7 @@ clang -fuse-ld=`pwd`/../mold -o $t/exe $t/b.o -L $t -lfoo
 readelf --dynamic $t/exe | fgrep -q 'Shared library: [libfoo]'
 
 clang -fuse-ld=`pwd`/../mold -o $t/exe $t/b.o $t/libbar.so
-readelf --dynamic $t/exe | grep -Pq 'Shared library: \[.*dt_needed/libbar\.so\]'
+readelf --dynamic $t/exe | rg -Pq 'Shared library: \[.*dt_needed/libbar\.so\]'
 
 clang -fuse-ld=`pwd`/../mold -o $t/exe $t/b.o -L$t -lbar
 readelf --dynamic $t/exe | fgrep -q 'Shared library: [libbar.so]'
--- a/test/dynamic.sh
+++ b/test/dynamic.sh
@@ -14,7 +14,7 @@ fgrep -q 'Shared library: [libc.so.6]' $t/log
 fgrep -q 'Shared library: [ld-linux-x86-64.so.2]' $t/log
 
 readelf -W --symbols --use-dynamic $t/exe > $t/log2
-grep -Pq 'FUNC\s+GLOBAL\s+DEFAULT\s+UND\s+__libc_start_main' $t/log2
+rg -Pq 'FUNC\s+GLOBAL\s+DEFAULT\s+UND\s+__libc_start_main' $t/log2
 
 cat <<EOF | clang -c -fPIC -o $t/b.o -xc -
 #include <stdio.h>
--- a/test/ifunc-export.sh
+++ b/test/ifunc-export.sh
@@ -32,6 +32,6 @@ resolve_foobar:
 EOF
 
 clang -fuse-ld=`pwd`/../mold -shared -o $t/b.so $t/a.o
-readelf --dyn-syms $t/b.so | grep -Pq '0 (IFUNC|<OS specific>: 10)\s+GLOBAL DEFAULT   \d+ foobar'
+readelf --dyn-syms $t/b.so | rg -Pq '0 (IFUNC|<OS specific>: 10)\s+GLOBAL DEFAULT   \d+ foobar'
 
 echo OK
--- a/test/image-base.sh
+++ b/test/image-base.sh
@@ -16,6 +16,6 @@ EOF
 
 clang -fuse-ld=`pwd`/../mold -o $t/exe $t/a.o -Wl,--image-base=0x8000000
 $t/exe | grep -q 'Hello world'
-readelf -W --sections $t/exe | grep -Pq '.interp\s+PROGBITS\s+0000000008000...\b'
+readelf -W --sections $t/exe | rg -Pq '.interp\s+PROGBITS\s+0000000008000...\b'
 
 echo OK
--- a/test/note.sh
+++ b/test/note.sh
@@ -31,9 +31,9 @@ EOF
 ../mold -static -o $t/exe $t/a.o
 readelf -W --sections $t/exe > $t/log
 
-grep -Pq '.note.bar\s+NOTE.+000008 00   A  0   0  4' $t/log
-grep -Pq '.note.baz\s+NOTE.+000008 00   A  0   0  8' $t/log
-grep -Pq '.note.nonalloc\s+NOTE.+000008 00      0   0  1' $t/log
+rg -Pq '.note.bar\s+NOTE.+000008 00   A  0   0  4' $t/log
+rg -Pq '.note.baz\s+NOTE.+000008 00   A  0   0  8' $t/log
+rg -Pq '.note.nonalloc\s+NOTE.+000008 00      0   0  1' $t/log
 
 readelf --segments $t/exe > $t/log
 fgrep -q '01     .note.bar' $t/log
--- a/test/pltgot.sh
+++ b/test/pltgot.sh
@@ -26,6 +26,6 @@ EOF
 
 objdump -d -j .plt.got $t/exe > $t/log
 
-grep -Pq '201020:\s+ff 25 da 0f 00 00\s+jmpq   \*0xfda\(%rip\)\s+# 202000 <ext2>' $t/log
+rg -Pq '201020:\s+ff 25 da 0f 00 00\s+jmpq   \*0xfda\(%rip\)\s+# 202000 <ext2>' $t/log
 
 echo OK
--- a/test/relax.sh
+++ b/test/relax.sh
@@ -39,43 +39,43 @@ EOF
 clang -fuse-ld=`pwd`/../mold -o $t/exe $t/a.o $t/b.o
 objdump -d $t/exe | grep -A20 '<bar>:' > $t/log
 
-grep -Pq 'lea \s*0x.+\(%rip\),%rax .*<foo>' $t/log
-grep -Pq 'lea \s*0x.+\(%rip\),%rcx .*<foo>' $t/log
-grep -Pq 'lea \s*0x.+\(%rip\),%rdx .*<foo>' $t/log
-grep -Pq 'lea \s*0x.+\(%rip\),%rbx .*<foo>' $t/log
-grep -Pq 'lea \s*0x.+\(%rip\),%rbp .*<foo>' $t/log
-grep -Pq 'lea \s*0x.+\(%rip\),%rsi .*<foo>' $t/log
-grep -Pq 'lea \s*0x.+\(%rip\),%rdi .*<foo>' $t/log
-grep -Pq 'lea \s*0x.+\(%rip\),%r8  .*<foo>' $t/log
-grep -Pq 'lea \s*0x.+\(%rip\),%r9  .*<foo>' $t/log
-grep -Pq 'lea \s*0x.+\(%rip\),%r10 .*<foo>' $t/log
-grep -Pq 'lea \s*0x.+\(%rip\),%r11 .*<foo>' $t/log
-grep -Pq 'lea \s*0x.+\(%rip\),%r12 .*<foo>' $t/log
-grep -Pq 'lea \s*0x.+\(%rip\),%r13 .*<foo>' $t/log
-grep -Pq 'lea \s*0x.+\(%rip\),%r14 .*<foo>' $t/log
-grep -Pq 'lea \s*0x.+\(%rip\),%r15 .*<foo>' $t/log
-grep -Pq 'callq.*<foo>' $t/log
-grep -Pq 'jmpq.*<foo>' $t/log
+rg -Pq 'lea \s*0x.+\(%rip\),%rax .*<foo>' $t/log
+rg -Pq 'lea \s*0x.+\(%rip\),%rcx .*<foo>' $t/log
+rg -Pq 'lea \s*0x.+\(%rip\),%rdx .*<foo>' $t/log
+rg -Pq 'lea \s*0x.+\(%rip\),%rbx .*<foo>' $t/log
+rg -Pq 'lea \s*0x.+\(%rip\),%rbp .*<foo>' $t/log
+rg -Pq 'lea \s*0x.+\(%rip\),%rsi .*<foo>' $t/log
+rg -Pq 'lea \s*0x.+\(%rip\),%rdi .*<foo>' $t/log
+rg -Pq 'lea \s*0x.+\(%rip\),%r8  .*<foo>' $t/log
+rg -Pq 'lea \s*0x.+\(%rip\),%r9  .*<foo>' $t/log
+rg -Pq 'lea \s*0x.+\(%rip\),%r10 .*<foo>' $t/log
+rg -Pq 'lea \s*0x.+\(%rip\),%r11 .*<foo>' $t/log
+rg -Pq 'lea \s*0x.+\(%rip\),%r12 .*<foo>' $t/log
+rg -Pq 'lea \s*0x.+\(%rip\),%r13 .*<foo>' $t/log
+rg -Pq 'lea \s*0x.+\(%rip\),%r14 .*<foo>' $t/log
+rg -Pq 'lea \s*0x.+\(%rip\),%r15 .*<foo>' $t/log
+rg -Pq 'callq.*<foo>' $t/log
+rg -Pq 'jmpq.*<foo>' $t/log
 
 clang -fuse-ld=`pwd`/../mold -o $t/exe $t/a.o $t/b.o -Wl,-no-relax
 objdump -d $t/exe | grep -A20 '<bar>:' > $t/log
 
-grep -Pq 'mov \s*0x.+\(%rip\),%rax' $t/log
-grep -Pq 'mov \s*0x.+\(%rip\),%rcx' $t/log
-grep -Pq 'mov \s*0x.+\(%rip\),%rdx' $t/log
-grep -Pq 'mov \s*0x.+\(%rip\),%rbx' $t/log
-grep -Pq 'mov \s*0x.+\(%rip\),%rbp' $t/log
-grep -Pq 'mov \s*0x.+\(%rip\),%rsi' $t/log
-grep -Pq 'mov \s*0x.+\(%rip\),%rdi' $t/log
-grep -Pq 'mov \s*0x.+\(%rip\),%r8 ' $t/log
-grep -Pq 'mov \s*0x.+\(%rip\),%r9 ' $t/log
-grep -Pq 'mov \s*0x.+\(%rip\),%r10' $t/log
-grep -Pq 'mov \s*0x.+\(%rip\),%r11' $t/log
-grep -Pq 'mov \s*0x.+\(%rip\),%r12' $t/log
-grep -Pq 'mov \s*0x.+\(%rip\),%r13' $t/log
-grep -Pq 'mov \s*0x.+\(%rip\),%r14' $t/log
-grep -Pq 'mov \s*0x.+\(%rip\),%r15' $t/log
-grep -Pq 'callq.*\(%rip\)' $t/log
-grep -Pq 'jmpq.*\(%rip\)' $t/log
+rg -Pq 'mov \s*0x.+\(%rip\),%rax' $t/log
+rg -Pq 'mov \s*0x.+\(%rip\),%rcx' $t/log
+rg -Pq 'mov \s*0x.+\(%rip\),%rdx' $t/log
+rg -Pq 'mov \s*0x.+\(%rip\),%rbx' $t/log
+rg -Pq 'mov \s*0x.+\(%rip\),%rbp' $t/log
+rg -Pq 'mov \s*0x.+\(%rip\),%rsi' $t/log
+rg -Pq 'mov \s*0x.+\(%rip\),%rdi' $t/log
+rg -Pq 'mov \s*0x.+\(%rip\),%r8 ' $t/log
+rg -Pq 'mov \s*0x.+\(%rip\),%r9 ' $t/log
+rg -Pq 'mov \s*0x.+\(%rip\),%r10' $t/log
+rg -Pq 'mov \s*0x.+\(%rip\),%r11' $t/log
+rg -Pq 'mov \s*0x.+\(%rip\),%r12' $t/log
+rg -Pq 'mov \s*0x.+\(%rip\),%r13' $t/log
+rg -Pq 'mov \s*0x.+\(%rip\),%r14' $t/log
+rg -Pq 'mov \s*0x.+\(%rip\),%r15' $t/log
+rg -Pq 'callq.*\(%rip\)' $t/log
+rg -Pq 'jmpq.*\(%rip\)' $t/log
 
 echo OK
--- a/test/shared.sh
+++ b/test/shared.sh
@@ -19,7 +19,7 @@ clang -shared -fuse-ld=`pwd`/../mold -o $t/b.so $t/a.o
 readelf --dyn-syms $t/b.so > $t/log
 
 grep -q '0000000000000000     0 NOTYPE  GLOBAL DEFAULT  UND fn2' $t/log
-grep -Pq '0 NOTYPE  GLOBAL DEFAULT   \d+ fn1' $t/log
+rg -Pq '0 NOTYPE  GLOBAL DEFAULT   \d+ fn1' $t/log
 
 cat <<EOF | clang -fPIC -c -o $t/c.o -xc -
 #include <stdio.h>
--- a/test/symtab.sh
+++ b/test/symtab.sh
@@ -28,11 +28,11 @@ echo '{ local: module_local; };' > $t/c.map
 
 readelf --symbols $t/exe > $t/log
 
-grep -Pq '0 NOTYPE  LOCAL  DEFAULT    \d+ local1' $t/log
-grep -Pq '0 NOTYPE  LOCAL  DEFAULT    \d+ local2' $t/log
-grep -Pq '0 NOTYPE  GLOBAL DEFAULT    \d+ foo' $t/log
-grep -Pq '0 NOTYPE  GLOBAL DEFAULT    \d+ bar' $t/log
-grep -Pq '0 NOTYPE  GLOBAL DEFAULT    \d+ this_is_global' $t/log
-grep -Pq '0 NOTYPE  GLOBAL DEFAULT    \d+ module_local' $t/log
+rg -Pq '0 NOTYPE  LOCAL  DEFAULT    \d+ local1' $t/log
+rg -Pq '0 NOTYPE  LOCAL  DEFAULT    \d+ local2' $t/log
+rg -Pq '0 NOTYPE  GLOBAL DEFAULT    \d+ foo' $t/log
+rg -Pq '0 NOTYPE  GLOBAL DEFAULT    \d+ bar' $t/log
+rg -Pq '0 NOTYPE  GLOBAL DEFAULT    \d+ this_is_global' $t/log
+rg -Pq '0 NOTYPE  GLOBAL DEFAULT    \d+ module_local' $t/log
 
 echo OK
--- a/test/thin-archive.sh
+++ b/test/thin-archive.sh
@@ -34,9 +34,9 @@ rm -f $t/d.a
 
 clang -fuse-ld=`pwd`/../mold -Wl,--trace -o $t/exe $t/d.o $t/d.a > $t/log
 
-grep -Pq 'thin-archive/d.a\(.*long-long-long-filename.o\)' $t/log
-grep -Pq 'thin-archive/d.a\(.*b.o\)' $t/log
-grep -Pq 'thin-archive/d.a\(/.*/b.o\)' $t/log
+rg -Pq 'thin-archive/d.a\(.*long-long-long-filename.o\)' $t/log
+rg -Pq 'thin-archive/d.a\(.*b.o\)' $t/log
+rg -Pq 'thin-archive/d.a\(/.*/b.o\)' $t/log
 fgrep -q thin-archive/d.o $t/log
 
 $t/exe | grep -q 15
--- a/test/z_nodump.sh
+++ b/test/z_nodump.sh
@@ -10,9 +10,9 @@ void foo() {}
 EOF
 
 clang -fuse-ld=`pwd`/../mold -shared -o $t/b.so $t/a.o
-! readelf --dynamic $t/b.so | grep -Pq 'Flags: NODUMP' || false
+! readelf --dynamic $t/b.so | rg -Pq 'Flags: NODUMP' || false
 
 clang -fuse-ld=`pwd`/../mold -shared -o $t/b.so $t/a.o -Wl,-z,nodump
-readelf --dynamic $t/b.so | grep -Pq 'Flags: NODUMP'
+readelf --dynamic $t/b.so | rg -Pq 'Flags: NODUMP'
 
 echo OK
--- a/test/z_origin.sh
+++ b/test/z_origin.sh
@@ -15,7 +15,7 @@ EOF
 
 clang -fuse-ld=`pwd`/../mold -o $t/exe $t/a.o -Wl,-z,origin
 
-readelf --dynamic $t/exe | grep -Pq '\(FLAGS\)\s+ORIGIN'
-readelf --dynamic $t/exe | grep -Pq 'Flags: ORIGIN'
+readelf --dynamic $t/exe | rg -Pq '\(FLAGS\)\s+ORIGIN'
+readelf --dynamic $t/exe | rg -Pq 'Flags: ORIGIN'
 
 echo OK
