Index: MellowPlayer-3.6.6/src/3rdparty/boost/di.hpp
===================================================================
--- MellowPlayer-3.6.6.orig/src/3rdparty/boost/di.hpp
+++ MellowPlayer-3.6.6/src/3rdparty/boost/di.hpp
@@ -67,7 +67,7 @@ BOOST_DI_CFG_FWD
 #pragma clang diagnostic error "-Wundefined-internal"
 #pragma clang diagnostic ignored "-Wmissing-field-initializers"
 #elif defined(__GCC__)
-#pragma GCC diagnostic error "-Wdeprecated-declarations"
+#pragma GCC diagnostic warning "-Wdeprecated-declarations"
 #pragma GCC diagnostic push
 #if (__GNUC__ < 6)
 #pragma GCC diagnostic error "-Werror"
