--- a/core/src/spdlog/details/os-inl.h
+++ b/core/src/spdlog/details/os-inl.h
@@ -231,7 +231,7 @@
     int fd = ::fileno(f);
 #endif
 // 64 bits(but not in osx or cygwin, where fstat64 is deprecated)
-#if (defined(__linux__) || defined(__sun) || defined(_AIX)) && (defined(__LP64__) || defined(_LP64))
+#if ((defined(__linux__) && defined(__GNU_SOURCE_)) || defined(__sun) || defined(_AIX)) && (defined(__LP64__) || defined(_LP64))
     struct stat64 st;
     if (::fstat64(fd, &st) == 0)
     {
