--- a/Cargo.toml
+++ b/Cargo.toml
@@ -45,7 +45,7 @@

 [dependencies.ffmpeg]
 package = "ffmpeg-next"
-version = "6"
+version = "8"
 optional = true
 default-features = false
 features = ["codec", "format", "filter", "software-resampling", "software-scaling"]
@@ -100,4 +100,4 @@

 [patch.crates-io]
 # ffmpeg-sys-next does not support cross-compilation, which I use to produce binaries https://github.com/zmwangx/rust-ffmpeg-sys/pull/30
-ffmpeg-sys-next = { rev = "fd5784d645df2ebe022a204ac36582074da1edf7", git = "https://github.com/kornelski/rust-ffmpeg-sys-1"}
+ffmpeg-sys-next = { rev = "36eca19434217b585748e0355a6d10c16deca8a2", git = "https://github.com/zmwangx/rust-ffmpeg-sys"}
