--- ./liblabjackusb/Makefile.orig	2021-06-29 19:26:03.060647459 -0400
+++ ./liblabjackusb/Makefile	2021-06-29 19:25:31.946314812 -0400
@@ -7,9 +7,9 @@
 
 VERSION = 2.6.0
 PREFIX ?= /usr/local
-DESTINATION = $(PREFIX)/lib
+DESTINATION = $(DESTDIR)$(PREFIX)/lib
 HEADER = labjackusb.h
-HEADER_DESTINATION = $(PREFIX)/include
+HEADER_DESTINATION = $(DESTDIR)$(PREFIX)/include
 LIBFLAGS = -lusb-1.0 -lc
 ADD_LDCONFIG_PATH = ./add_ldconfig_path.sh
