--- scripts/sign-file.c.orig	2015-11-02 08:09:38.249671776 +0100
+++ scripts/sign-file.c	2015-11-02 08:10:04.203937498 +0100
@@ -39,7 +39,7 @@
  * signing with anything other than SHA1 - so we're stuck with that if such is
  * the case.
  */
-#if OPENSSL_VERSION_NUMBER < 0x10000000L
+#if defined(LIBRESSL_VERSION_NUMBER) || OPENSSL_VERSION_NUMBER < 0x10000000L
 #define USE_PKCS7
 #endif
 #ifndef USE_PKCS7
