--- src/include.am.orig	2017-12-01 23:56:36.199044786 +0100
+++ src/include.am	2017-12-01 23:56:54.073045467 +0100
@@ -7,7 +7,7 @@
 
 MEGACMD = mega-cmd mega-exec mega-cmd-server
 bin_PROGRAMS += $(MEGACMD)
-$(MEGACMD): $(MEGA_LIBS)
+$(MEGACMD):
 noinst_HEADERS += src/comunicationsmanager.h src/configurationmanager.h src/megacmd.h src/megacmdlogger.h src/megacmdsandbox.h src/megacmdutils.h src/listeners.h src/megacmdexecuter.h src/megacmdversion.h src/megacmdplatform.h src/comunicationsmanagerportsockets.h
 megacmdcompletiondir = $(sysconfdir)/bash_completion.d/
 megacmdcompletion_DATA = src/client/megacmd_completion.sh
