diff --git examples/megacmd/megacmdshell/megacmdshell.cpp examples/megacmd/megacmdshell/megacmdshell.cpp
index be281cb7..979ab793 100644
--- examples/megacmd/megacmdshell/megacmdshell.cpp
+++ examples/megacmd/megacmdshell/megacmdshell.cpp
@@ -33,6 +33,8 @@
 #include <algorithm>
 #include <stdio.h>
 
+#include <string.h>
+
 enum
 {
     MCMD_OK = 0,              ///< Everything OK
diff --git examples/megacmd/megacmdshell/megacmdshellcommunications.cpp examples/megacmd/megacmdshell/megacmdshellcommunications.cpp
index 8046a8b5..a01d07ab 100644
--- examples/megacmd/megacmdshell/megacmdshellcommunications.cpp
+++ examples/megacmd/megacmdshell/megacmdshellcommunications.cpp
@@ -27,6 +27,8 @@
 #include <iostream>
 #include <sstream>
 
+#include <string.h>
+
 #ifdef _WIN32
 #include <shlobj.h> //SHGetFolderPath
 #include <Shlwapi.h> //PathAppend
