--- net.cpp	2017-11-18 11:58:03.524955739 +0100
+++ src/posix/net.cpp	2017-11-18 11:59:41.082959457 +0100
@@ -53,7 +53,7 @@
     }
 }
 
-#if OPENSSL_VERSION_NUMBER >= 0x10000000
+#if OPENSSL_VERSION_NUMBER >= 0x10000000 || defined (LIBRESSL_VERSION_NUMBER)
 void CurlHttpIO::id_function(CRYPTO_THREADID* id)
 {
     CRYPTO_THREADID_set_pointer(id, (void *)THREAD_CLASS::currentThreadId());
@@ -137,7 +137,7 @@
 #if defined(USE_OPENSSL) && !defined(OPENSSL_IS_BORINGSSL)
 
     if (!CRYPTO_get_locking_callback()
-#if OPENSSL_VERSION_NUMBER >= 0x10000000
+#if OPENSSL_VERSION_NUMBER >= 0x10000000  || defined (LIBRESSL_VERSION_NUMBER)
         && !CRYPTO_THREADID_get_callback())
 #else
         && !CRYPTO_get_id_callback())
@@ -147,7 +147,7 @@
         int numLocks = CRYPTO_num_locks();
         sslMutexes = new MUTEX_CLASS*[numLocks];
         memset(sslMutexes, 0, numLocks * sizeof(MUTEX_CLASS*));
-#if OPENSSL_VERSION_NUMBER >= 0x10000000
+#if OPENSSL_VERSION_NUMBER >= 0x10000000  || defined (LIBRESSL_VERSION_NUMBER)
         CRYPTO_THREADID_set_callback(CurlHttpIO::id_function);
 #else
         CRYPTO_set_id_callback(CurlHttpIO::id_function);
