diff --git a/cmake/modules/configuration.cmake b/cmake/modules/configuration.cmake
index ad102d9cc..8707f5c6c 100644
--- a/cmake/modules/configuration.cmake
+++ b/cmake/modules/configuration.cmake
@@ -38,7 +38,7 @@ else()
     include(CheckSymbolExists)
     check_symbol_exists(glob glob.h HAVE_GLOB_H)
 
-    check_function_exists(aio_write, HAVE_AIO_RT)
+    check_function_exists(aio_write HAVE_AIO_RT)
 
     # Check if our toolchain supports TI emulation mode.
     try_compile(SUPPORTS_TI_EMULATION_MODE
diff --git a/cmake/modules/sdklib_libraries.cmake b/cmake/modules/sdklib_libraries.cmake
index e51fb8ede..ecc170198 100644
--- a/cmake/modules/sdklib_libraries.cmake
+++ b/cmake/modules/sdklib_libraries.cmake
@@ -109,7 +109,7 @@ macro(load_sdklib_libraries)
 
         find_package(PkgConfig REQUIRED) # For libraries loaded using pkg-config
 
-        pkg_check_modules(cryptopp REQUIRED IMPORTED_TARGET libcrypto++)
+        pkg_check_modules(cryptopp REQUIRED IMPORTED_TARGET libcryptopp)
         target_link_libraries(SDKlib PUBLIC PkgConfig::cryptopp) # TODO: Private for SDK core
 
         pkg_check_modules(sodium REQUIRED IMPORTED_TARGET libsodium)
