*** configure.old	2020-05-27 06:34:42.422898841 -0300
--- configure.ac	2020-05-27 06:34:25.983868381 -0300
***************
*** 14,20 ****
  m4_pattern_forbid([PKG_CHECK_MODULES])
  m4_pattern_forbid([GUILE_PKG])
  
! GUILE_PKG([2.2 2.0])
  GUILE_PROGS
  
  PKG_CHECK_MODULES([SQLITE], [sqlite3])
--- 14,20 ----
  m4_pattern_forbid([PKG_CHECK_MODULES])
  m4_pattern_forbid([GUILE_PKG])
  
! GUILE_PKG([3.0 2.2 2.0])
  GUILE_PROGS
  
  PKG_CHECK_MODULES([SQLITE], [sqlite3])
