--- scripts/fakeroot.in.orig	2010-01-15 20:06:08.160073189 +0100
+++ scripts/fakeroot.in	2010-01-15 20:06:19.679197650 +0100
@@ -43,7 +43,7 @@ export FAKED_MODE
 
 libfound=no
 
-GETOPTEST=`getopt --version`
+GETOPTEST=`getopt --version 2>/dev/null`
 case $GETOPTEST in
 getopt*) # GNU getopt
     FAKE_TEMP=`getopt -l lib: -l faked: -l unknown-is-real -l fd-base: -l version -l help -- +l:f:i:s:ub:vh "$@"`
