--- a/src/manager.rs
+++ b/src/manager.rs
@@ -143,6 +143,6 @@
             if let Some(threshold) = Config::get().idle_threshold {
                 match maths_utility::query_screensaver_info(&self.base_window) {
                     Ok(info) => {
-                        if info.idle / 1000 >= threshold {
+                        if info.idle as u64 / 1000 >= threshold {
                             self.layout_windows
                                 .values_mut()
                                 .flatten()
