diff --git a/CMakeLists.txt b/CMakeLists.txt
index c694166..405dc90 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -199,7 +199,7 @@ set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/bin )
 
 if(CMAKE_BUILD_TYPE)
     # Validate build type
-    set(CONFIGURATION_TYPES "Debug;Release;RelWithDebInfo")
+    set(CONFIGURATION_TYPES "Debug;Release;RelWithDebInfo;None")
 
     string(FIND "${CONFIGURATION_TYPES}" "${CMAKE_BUILD_TYPE}" MATCHED_CONFIG)
     if (${MATCHED_CONFIG} EQUAL -1)
