diff -rup ispc-1.9.2/ispc.cpp ispc-1.9.2.new/ispc.cpp
--- ispc-1.9.2/ispc.cpp	2017-11-11 00:07:30.000000000 +0100
+++ ispc-1.9.2.new/ispc.cpp	2018-03-17 20:23:01.104377910 +0100
@@ -59,7 +59,7 @@
   #include <llvm/IR/Module.h>
   #include <llvm/IR/Instructions.h>
 #endif
-#if ISPC_LLVM_VERSION >= ISPC_LLVM_3_6 // LLVM 3.6+
+#if ISPC_LLVM_VERSION >= ISPC_LLVM_3_6 && ISPC_LLVM_VERSION < ISPC_LLVM_6_0 // LLVM 3.6-6.0
   #include <llvm/Target/TargetSubtargetInfo.h>
   #if ISPC_LLVM_VERSION >= ISPC_LLVM_3_7 // LLVM 3.7+
     #include <llvm/Target/TargetLowering.h>
 
