From 08ef19e848176e58511f1a66992b5a26cb3c51a2 Mon Sep 17 00:00:00 2001
From: Alessandro Bono <alessandro.bono369@gmail.com>
Date: Wed, 1 Nov 2023 12:35:44 +0100
Subject: [PATCH] gdm-session: Pass the correct pointer type

---
 daemon/gdm-session.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/daemon/gdm-session.c b/daemon/gdm-session.c
index 06f648ee0..be19a049c 100644
--- a/daemon/gdm-session.c
+++ b/daemon/gdm-session.c
@@ -3045,7 +3045,7 @@ free_pending_worker_connection (GdmSession      *self,
         worker_manager_interface = g_object_get_data (G_OBJECT (connection),
                                                       "gdm-dbus-worker-manager-interface");
         if (worker_manager_interface != NULL) {
-                g_dbus_interface_skeleton_unexport (worker_manager_interface);
+                g_dbus_interface_skeleton_unexport (G_DBUS_INTERFACE_SKELETON (worker_manager_interface));
                 g_object_set_data (G_OBJECT (connection),
                                    "gdm-dbus-worker-manager-interface",
                                    NULL);
-- 
GitLab

