--- a/meson.build
+++ b/meson.build
@@ -94,19 +94,8 @@ if xdmcp_dep.found() and get_option('tcp-wrappers')
   libwrap_dep = cc.find_library('libwrap')
 endif
 # systemd
-systemd_dep = dependency('systemd')
 libsystemd_dep = dependency('libsystemd')
-if meson.version().version_compare('>= 0.53')
-  systemd_multiseat_x = find_program('systemd-multi-seat-x',
-    required: false,
-    dirs: [
-      systemd_dep.get_pkgconfig_variable('systemdutildir'),
-      '/lib/systemd',
-      '/usr/lib/systemd',
-    ])
-else
-  systemd_multiseat_x = find_program('systemd-multi-seat-x', required: false)
-endif
+systemd_multiseat_x = find_program('systemd-multi-seat-x', required: false)
 systemd_x_server = systemd_multiseat_x.found()? systemd_multiseat_x.path() : '/lib/systemd/systemd-multi-seat-x'
 # Plymouth
 plymouth_dep = dependency('ply-boot-client', required: get_option('plymouth'))
