--- a/src/3rdparty/qt5-x11embed/CMakeLists.txt	2017-11-06 19:24:12.000000000 +0100
+++ b/src/3rdparty/qt5-x11embed/CMakeLists.txt	2017-11-06 19:24:12.000000000 +0100
@@ -1,7 +1,7 @@
 cmake_minimum_required(VERSION 2.8.12 FATAL_ERROR)
 project(qt5-x11embed)
 
-list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/3rdparty/ECM/find-modules")
+list(APPEND CMAKE_MODULE_PATH "/usr/share/ECM/find-modules")
 
 find_package(Qt5Core COMPONENTS Private REQUIRED)
 find_package(Qt5Widgets COMPONENTS Private REQUIRED)
