--- CMakeLists.txt.orig
+++ CMakeLists.txt
@@ -57,6 +57,7 @@
 OPTION(WANT_OGGVORBIS	"Include OGG/Vorbis support" ON)
 OPTION(WANT_PULSEAUDIO	"Include PulseAudio support" ON)
 OPTION(WANT_PORTAUDIO	"Include PortAudio support" ON)
+OPTION(WANT_SNDIO	"Include sndio support" ON)
 OPTION(WANT_SOUNDIO	"Include libsoundio support" ON)
 OPTION(WANT_SDL		"Include SDL (Simple DirectMedia Layer) support" ON)
 OPTION(WANT_SF2		"Include SoundFont2 player plugin" ON)
@@ -429,7 +430,7 @@
 	FIND_PACKAGE(Threads)
 ENDIF(LMMS_BUILD_LINUX OR LMMS_BUILD_APPLE OR LMMS_BUILD_OPENBSD)
 
-IF(LMMS_BUILD_OPENBSD)
+IF(WANT_SNDIO)
 	FIND_PACKAGE(Sndio)
 	IF(SNDIO_FOUND)
 		SET(LMMS_HAVE_SNDIO TRUE)
@@ -437,7 +438,7 @@
     	ELSE()
         	SET(STATUS_SNDIO "<not found or not supported on this platform>")
 	ENDIF(SNDIO_FOUND)
-ENDIF(LMMS_BUILD_OPENBSD)
+ENDIF(WANT_SNDIO)
 
 # check for WINE
 IF(WANT_VST)
