--- mupdf.c.orig
+++ mupdf.c
@@ -33,10 +33,10 @@
 	w = rect.x1 - rect.x0;
 	h = rect.y1 - rect.y0;
 
-	pix = fz_new_pixmap_with_bbox(doc->ctx, fz_device_rgb(doc->ctx), &bbox);
+	pix = fz_new_pixmap_with_bbox(doc->ctx, fz_device_rgb(doc->ctx), &bbox, 1);
 	fz_clear_pixmap_with_value(doc->ctx, pix, 0xff);
 
-	dev = fz_new_draw_device(doc->ctx, pix);
+	dev = fz_new_draw_device(doc->ctx, NULL, pix);
 	fz_run_page(doc->ctx, page, dev, &ctm, NULL);
 	fz_drop_device(doc->ctx, dev);
 
