diff --git a/src/Cache.cpp b/src/Cache.cpp
index c575ebf7..0c5e879c 100644
--- a/src/Cache.cpp
+++ b/src/Cache.cpp
@@ -438,7 +438,7 @@ Cache::loadSecretsFromStore(
                 if (job->error() && job->error() != QKeychain::Error::EntryNotFound) {
                     nhlog::db()->error("Restoring secret '{}' failed ({}): {}",
                                        name.toStdString(),
-                                       job->error(),
+                                       static_cast<int>(job->error()),
                                        job->errorString().toStdString());
 
                     fatalSecretError();
diff --git a/src/ChatPage.cpp b/src/ChatPage.cpp
index 12d83f49..61daacb5 100644
--- a/src/ChatPage.cpp
+++ b/src/ChatPage.cpp
@@ -1248,7 +1248,7 @@ ChatPage::getBackupVersion()
                       nhlog::crypto()->info("Our backup key {} does not match the one "
                                             "used in the online backup {}",
                                             pubkey,
-                                            auth_data["public_key"]);
+                                            auth_data["public_key"].get<std::string>());
                       cache::client()->deleteBackupVersion();
                       return;
                   }
diff --git a/src/encryption/DeviceVerificationFlow.cpp b/src/encryption/DeviceVerificationFlow.cpp
index 0e9043dd..7dee9e6b 100644
--- a/src/encryption/DeviceVerificationFlow.cpp
+++ b/src/encryption/DeviceVerificationFlow.cpp
@@ -39,7 +39,7 @@ DeviceVerificationFlow::DeviceVerificationFlow(QObject *,
   , deviceIds(std::move(deviceIds_))
   , model_(model)
 {
-    nhlog::crypto()->debug("CREATING NEW FLOW, {}, {}", flow_type, (void *)this);
+    nhlog::crypto()->debug("CREATING NEW FLOW, {}, {}", static_cast<int>(flow_type), (void *)this);
     if (deviceIds.size() == 1)
         deviceId = deviceIds.front();
 
diff --git a/src/timeline/InputBar.cpp b/src/timeline/InputBar.cpp
index 94944337..1fb210d8 100644
--- a/src/timeline/InputBar.cpp
+++ b/src/timeline/InputBar.cpp
@@ -948,14 +948,14 @@ MediaUpload::MediaUpload(std::unique_ptr<QIODevice> source_,
                 this,
                 [mediaPlayer](QMediaPlayer::Error error) {
                     nhlog::ui()->debug("Media player error {} and errorStr {}",
-                                       error,
+                                       static_cast<int>(error),
                                        mediaPlayer->errorString().toStdString());
                 });
         connect(mediaPlayer,
                 &QMediaPlayer::mediaStatusChanged,
                 [mediaPlayer](QMediaPlayer::MediaStatus status) {
                     nhlog::ui()->debug(
-                      "Media player status {} and error {}", status, mediaPlayer->error());
+                      "Media player status {} and error {}", static_cast<int>(status), static_cast<int>(mediaPlayer->error()));
                 });
         connect(mediaPlayer,
                 qOverload<const QString &, const QVariant &>(&QMediaPlayer::metaDataChanged),
diff --git a/src/ui/MxcMediaProxy.cpp b/src/ui/MxcMediaProxy.cpp
index da5a2231..dbe63469 100644
--- a/src/ui/MxcMediaProxy.cpp
+++ b/src/ui/MxcMediaProxy.cpp
@@ -37,11 +37,11 @@ MxcMediaProxy::MxcMediaProxy(QObject *parent)
             qOverload<QMediaPlayer::Error>(&MxcMediaProxy::error),
             [this](QMediaPlayer::Error error) {
                 nhlog::ui()->info("Media player error {} and errorStr {}",
-                                  error,
+                                  static_cast<int>(error),
                                   this->errorString().toStdString());
             });
     connect(this, &MxcMediaProxy::mediaStatusChanged, [this](QMediaPlayer::MediaStatus status) {
-        nhlog::ui()->info("Media player status {} and error {}", status, this->error());
+        nhlog::ui()->info("Media player status {} and error {}", static_cast<int>(status), static_cast<int>(this->error()));
     });
     connect(this,
             qOverload<const QString &, const QVariant &>(&MxcMediaProxy::metaDataChanged),
