From 62888d310396476b54b46689da48477501490033 Mon Sep 17 00:00:00 2001
From: Juan RP <xtraeme@gmail.com>
Date: Thu, 13 Mar 2014 20:35:26 +0100
Subject: [PATCH 1/4] xbps_transaction_commit: add some debugging in error
 paths.

---
 lib/transaction_commit.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/lib/transaction_commit.c b/lib/transaction_commit.c
index dee8876..ef6b0ce 100644
--- lib/transaction_commit.c
+++ lib/transaction_commit.c
@@ -256,7 +256,7 @@ xbps_transaction_commit(struct xbps_handle *xhp)
 			rv = xbps_remove_pkg(xhp, pkgver, update);
 			if (rv != 0) {
 				xbps_dbg_printf(xhp, "[trans] failed to "
-				    "remove %s\n", pkgver);
+				    "remove %s: %s\n", pkgver, strerror(rv));
 				goto out;
 			}
 			continue;
@@ -266,9 +266,11 @@ xbps_transaction_commit(struct xbps_handle *xhp)
 			 * Reconfigure pending package.
 			 */
 			rv = xbps_configure_pkg(xhp, pkgver, false, false);
-			if (rv != 0)
+			if (rv != 0) {
+				xbps_dbg_printf(xhp, "[trans] failed to "
+				    "configure %s: %s\n", pkgver, strerror(rv));
 				goto out;
-
+			}
 			continue;
 
 		} else if (strcmp(tract, "update") == 0) {
-- 
1.9.0

