From 4d40179a8ae86f16aeea1418b6a0b58c8ed8f3d7 Mon Sep 17 00:00:00 2001
From: Duncan Overbruck <mail@duncano.de>
Date: Wed, 4 Mar 2020 12:46:44 +0100
Subject: [PATCH] bin/xbps-install: fix regression in small transaction stats

fixes #242
---
 bin/xbps-install/transaction.c | 1 +
 1 file changed, 1 insertion(+)

diff --git bin/xbps-install/transaction.c bin/xbps-install/transaction.c
index ae1ddcf2..89d1b8a4 100644
--- bin/xbps-install/transaction.c
+++ bin/xbps-install/transaction.c
@@ -125,6 +125,7 @@ show_package_list(struct transaction *trans, xbps_trans_type_t ttype, unsigned i
 		xbps_dictionary_get_bool(obj, "download", &dload);
 		tt = xbps_transaction_pkg_type(obj);
 
+		buf = NULL;
 		if (tt == XBPS_TRANS_UPDATE) {
 			/* get installed pkgver */
 			ipkgd = xbps_pkgdb_get_pkg(trans->xhp, pkgname);
-- 
2.25.1

