From 3a5c1f3a71200f45f06dace1f55faf782ece9352 Mon Sep 17 00:00:00 2001
From: Enno Boland <g@s01.de>
Date: Tue, 26 Apr 2016 13:56:10 +0200
Subject: [PATCH] lib/fetch/http.c: fix EOL at HTTP CONNECT

fixes #172.
---
 lib/fetch/http.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/fetch/http.c b/lib/fetch/http.c
index a027c39..433bd11 100644
--- lib/fetch/http.c
+++ lib/fetch/http.c
@@ -733,7 +733,7 @@ http_connect(struct url *URL, struct url *purl, const char *flags, int *cached)
 		/* fetch_connect() has already set an error code */
 		return (NULL);
 	if (strcasecmp(URL->scheme, SCHEME_HTTPS) == 0 && purl) {
-		http_cmd(conn, "CONNECT %s:%d HTTP/1.1",
+		http_cmd(conn, "CONNECT %s:%d HTTP/1.1\r\n",
 		    URL->host, URL->port);
 		if (http_get_reply(conn) != HTTP_OK) {
 			fetch_close(conn);
-- 
2.8.1

