From 33a1bea633048a9d52f1ba52ec19852d321ffb37 Mon Sep 17 00:00:00 2001
From: Juan RP <xtraeme@voidlinux.eu>
Date: Sat, 31 Oct 2015 14:07:12 +0100
Subject: [PATCH 1/2] alternatives: fix a warning.

---
 lib/package_alternatives.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/package_alternatives.c b/lib/package_alternatives.c
index b71081e..58caa41 100644
--- lib/package_alternatives.c
+++ lib/package_alternatives.c
@@ -102,7 +102,7 @@ create_symlinks(struct xbps_handle *xhp, xbps_array_t a, const char *grname)
 	cnt = xbps_array_count(a);
 	for (i = 0; i < cnt; i++) {
 		xbps_string_t str;
-		char *l, *lnk, *tgt;
+		char *l, *lnk, *tgt = NULL;
 		const char *tgt0;
 		int rv;
 
-- 
2.6.2

