From daa1b446821646d8923623f8a0d6e1b65db0929b Mon Sep 17 00:00:00 2001
From: Duncaen <mail@duncano.de>
Date: Wed, 2 May 2018 00:45:24 +0200
Subject: [PATCH] xbps-install: fix "Space available on disk:" padding

---
 bin/xbps-install/transaction.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/bin/xbps-install/transaction.c b/bin/xbps-install/transaction.c
index 00850f86..2601b845 100644
--- ./bin/xbps-install/transaction.c
+++ ./bin/xbps-install/transaction.c
@@ -211,7 +211,7 @@ show_transaction_sizes(struct transaction *trans, int cols)
 			    "%s\n", strerror(errno));
 			return -1;
 		}
-		printf("Space available on disk:           %6s\n", size);
+		printf("Space available on disk:      %6s\n", size);
 	}
 	printf("\n");
 
