From fa1d543dfae2383745d4789e51874a33ef6b7179 Mon Sep 17 00:00:00 2001
From: Juan RP <xtraeme@gmail.com>
Date: Sat, 2 Feb 2013 14:34:55 +0100
Subject: [PATCH] Ignore package conflicts against themselves, due to virtual
 packages.

---
 NEWS                    | 3 +++
 lib/package_conflicts.c | 6 ++++++
 2 files changed, 9 insertions(+)

diff --git a/lib/package_conflicts.c b/lib/package_conflicts.c
index dd07677..658fdd6 100644
--- lib/package_conflicts.c
+++ lib/package_conflicts.c
@@ -63,6 +63,9 @@ xbps_pkg_find_conflicts(struct xbps_handle *xhp,
 		    (pkgd = xbps_pkgdb_get_virtualpkg(xhp, cfpkg))) {
 			prop_dictionary_get_cstring_nocopy(pkgd,
 			    "pkgver", &pkgver);
+			if (strcmp(pkgver, repopkgver) == 0)
+				continue;
+
 			xbps_dbg_printf(xhp, "found conflicting installed "
 			    "pkg %s with pkg in transaction %s\n", pkgver,
 			    repopkgver);
@@ -79,6 +82,9 @@ xbps_pkg_find_conflicts(struct xbps_handle *xhp,
 		    (pkgd = xbps_find_virtualpkg_in_array(xhp, unsorted, cfpkg))) {
 			prop_dictionary_get_cstring_nocopy(pkgd,
 			    "pkgver", &pkgver);
+			if (strcmp(pkgver, repopkgver) == 0)
+				continue;
+
 			xbps_dbg_printf(xhp, "found conflicting pkgs in "
 			    "transaction %s <-> %s\n", pkgver, repopkgver);
 			buf = xbps_xasprintf("%s conflicts with "
-- 
1.8.1.1

