From f896b17ba73a51451b221ebfc1bae59022cabcec Mon Sep 17 00:00:00 2001
From: Juan RP <xtraeme@voidlinux.eu>
Date: Sat, 31 Oct 2015 14:19:50 +0100
Subject: [PATCH 2/2] alternatives: fix an assertion!

---
 lib/package_alternatives.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/package_alternatives.c b/lib/package_alternatives.c
index 58caa41..19c1422 100644
--- lib/package_alternatives.c
+++ lib/package_alternatives.c
@@ -110,7 +110,7 @@ create_symlinks(struct xbps_handle *xhp, xbps_array_t a, const char *grname)
 		l = left(xbps_string_cstring_nocopy(str));
 		assert(l);
 		tgt0 = right(xbps_string_cstring_nocopy(str));
-		assert(tgt);
+		assert(tgt0);
 		if (l[0] != '/') {
 			char *tgt_dup, *tgt_dir;
 			tgt_dup = strdup(tgt0);
-- 
2.6.2

