From 12e18ae5796cdf0b1ec05b31a41380c6dbbe241f Mon Sep 17 00:00:00 2001
From: Juan RP <xtraeme@gmail.com>
Date: Thu, 30 Oct 2014 10:59:42 +0100
Subject: [PATCH 1/4] bin/xbps-install/util.c: use STDOUT_FILENO.

---
 bin/xbps-install/util.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/bin/xbps-install/util.c b/bin/xbps-install/util.c
index 3be2baf..9594877 100644
--- bin/xbps-install/util.c
+++ bin/xbps-install/util.c
@@ -40,7 +40,7 @@ get_maxcols(void)
 {
 	struct winsize ws;
 
-	if (ioctl(fileno(stdin), TIOCGWINSZ, &ws) == 0)
+	if (ioctl(STDOUT_FILENO, TIOCGWINSZ, &ws) == 0)
 		return ws.ws_col ? ws.ws_col : 80;
 
 	return 80;
-- 
2.1.3

