From d0a9bbb912630fee4fe7d29b6ec0655142f42fad Mon Sep 17 00:00:00 2001
From: Juan RP <xtraeme@gmail.com>
Date: Tue, 18 Mar 2014 14:08:40 +0100
Subject: [PATCH] Fix #34 (make libfetch always add "Accept: */*" in the HTTP
 header).

---
 NEWS             | 6 ++++++
 lib/fetch/http.c | 8 ++++++++
 2 files changed, 14 insertions(+)

--- lib/fetch/http.c
+++ lib/fetch/http.c
@@ -939,8 +939,16 @@ http_request(struct url *URL, const char *op, struct url_stat *us,
 			http_cmd(conn, "User-Agent: %s\r\n", p);
 		else
 			http_cmd(conn, "User-Agent: %s\r\n", _LIBFETCH_VER);
+
+		/*
+		 * Some servers returns 406 (Not Acceptable) if the Accept field is not
+		 * provided by the user agent, such example is http://alioth.debian.org.
+		 */
+		http_cmd(conn, "Accept: */*\r\n");
+
 		if (url->offset > 0)
 			http_cmd(conn, "Range: bytes=%lld-\r\n", (long long)url->offset);
+
 		http_cmd(conn, "\r\n");
 
 		/*
-- 
1.9.0

