From 6c96fe32ccff435d42ff19ef8c7755beb25bfd97 Mon Sep 17 00:00:00 2001
From: Juan RP <xtraeme@gmail.com>
Date: Tue, 5 Feb 2013 09:32:43 +0100
Subject: [PATCH 1/2] lib/initend.c: print dbg msg about successful vpkg conf
 files read.

---
 lib/initend.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/lib/initend.c b/lib/initend.c
index 6f66ded..e38edc1 100644
--- lib/initend.c
+++ lib/initend.c
@@ -108,12 +108,13 @@ config_inject_vpkgs(struct xbps_handle *xh)
 			path = xbps_xasprintf("%s/%s", vpkgdir, dp->d_name);
 			fp = fopen(path, "r");
 			assert(fp);
-			free(path);
 			if (cfg_parse_fp(xh->cfg, fp) != 0) {
 				xbps_error_printf("Failed to parse "
 				    "vpkg conf file %s:\n", dp->d_name);
 			}
 			fclose(fp);
+			xbps_dbg_printf(xh, "Injected vpkgs from %s\n", path);
+			free(path);
 		}
 	}
 	closedir(dirp);
-- 
1.8.1.1

