From 3f7cd86d9d4e291a148016be9da379db52ac2d37 Mon Sep 17 00:00:00 2001
From: Juan RP <xtraeme@gmail.com>
Date: Thu, 30 Oct 2014 11:00:01 +0100
Subject: [PATCH 2/4] bin/xbps-query/list.c: remove unused header.

---
 bin/xbps-query/list.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/bin/xbps-query/list.c b/bin/xbps-query/list.c
index 73ba667..8532dc9 100644
--- bin/xbps-query/list.c
+++ bin/xbps-query/list.c
@@ -29,7 +29,6 @@
 #include <errno.h>
 #include <string.h>
 #include <assert.h>
-#include <sys/ioctl.h>
 
 #include "defs.h"
 
-- 
2.1.3

