From 59781f4e91b798d63ad38e0d43ef0c0c2fe5f34c Mon Sep 17 00:00:00 2001
From: Juan RP <xtraeme@gmail.com>
Date: Thu, 30 Oct 2014 11:25:37 +0100
Subject: [PATCH 4/4] bin/xbps-install/fetch_cb.c: use STDOUT_FILENO.

---
 bin/xbps-install/fetch_cb.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/bin/xbps-install/fetch_cb.c b/bin/xbps-install/fetch_cb.c
index fb1a069..8b715a9 100644
--- bin/xbps-install/fetch_cb.c
+++ bin/xbps-install/fetch_cb.c
@@ -162,7 +162,7 @@ fetch_file_progress_cb(const struct xbps_fetch_cb_data *xfpd, void *cbdata)
 
 	if (xfpd->cb_start) {
 		/* start transfer stats */
-		v_tty = isatty(STDERR_FILENO);
+		v_tty = isatty(STDOUT_FILENO);
 		get_time(&xfer->start);
 		xfer->last.tv_sec = xfer->last.tv_usec = 0;
 	} else if (xfpd->cb_update) {
-- 
2.1.3

