From 743eabb88ee734d094174c509b4fd82ff9ed00b9 Mon Sep 17 00:00:00 2001
From: Juan RP <xtraeme@gmail.com>
Date: Sat, 24 Nov 2012 08:50:52 +0100
Subject: [PATCH] xbps-rindex: dont panic if old binpkg doesn't exist.

---
 bin/xbps-rindex/common.c | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/bin/xbps-rindex/common.c b/bin/xbps-rindex/common.c
index cca7871..43e55fe 100644
--- bin/xbps-rindex/common.c
+++ bin/xbps-rindex/common.c
@@ -42,18 +42,18 @@ remove_pkg(const char *repodir, const char *arch, const char *file)
 	char *filepath;
 	int rv;
 
-	/* Remove real binpkg */
 	filepath = xbps_xasprintf("%s/%s/%s", repodir, arch, file);
 	if (remove(filepath) == -1) {
-		rv = errno;
-		xbps_error_printf("failed to remove old binpkg `%s': %s\n",
-		    file, strerror(rv));
-		free(filepath);
-		return rv;
+		if (errno != ENOENT) {
+			rv = errno;
+			xbps_error_printf("failed to remove old binpkg "
+			    "`%s': %s\n", file, strerror(rv));
+			free(filepath);
+			return rv;
+		}
 	}
 	free(filepath);
 
-	/* Remove symlink to binpkg */
 	filepath = xbps_xasprintf("%s/%s", repodir, file);
 	if (remove(filepath) == -1) {
 		rv = errno;
-- 
1.8.0

