From 7b728f5ac0adc28f3d7270c26e14ecff73a7f16f Mon Sep 17 00:00:00 2001
From: Duncaen <mail@duncano.de>
Date: Wed, 26 Jun 2019 22:24:47 +0200
Subject: [PATCH] bin/xbps-remove: fix skipping executing orphans transaction

---
 bin/xbps-remove/main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/bin/xbps-remove/main.c b/bin/xbps-remove/main.c
index 1ef328a8..4ff943c0 100644
--- bin/xbps-remove/main.c
+++ bin/xbps-remove/main.c
@@ -295,7 +295,7 @@ main(int argc, char **argv)
 			exit(rv);
 		}
 	}
-	if (missing == argc) {
+	if (!orphans && missing == argc) {
 		goto out;
 	}
 	if (orphans || (argc > optind)) {
