commit 7b728f5ac0adc28f3d7270c26e14ecff73a7f16f
Author: Duncaen <mail@duncano.de>
Date:   Wed Jun 26 22:24:47 2019 +0200

    bin/xbps-remove: fix skipping executing orphans transaction

diff --git bin/xbps-remove/main.c bin/xbps-remove/main.c
index 1ef328a8..4ff943c0 100644
--- bin/xbps-remove/main.c
+++ bin/xbps-remove/main.c
@@ -295,7 +295,7 @@ main(int argc, char **argv)
 			exit(rv);
 		}
 	}
-	if (missing == argc) {
+	if (!orphans && missing == argc) {
 		goto out;
 	}
 	if (orphans || (argc > optind)) {
