commit 7e762cc3573df496752c3698ce9949300e61d9c6
Author: Duncaen <mail@duncano.de>
Date:   Fri Jun 21 20:23:55 2019 +0200

    lib/fetch.c: ignore EADDRNOTAVAIL and try next ip

diff --git lib/fetch/common.c lib/fetch/common.c
index 6bd8f510..df7f33dd 100644
--- lib/fetch/common.c
+++ lib/fetch/common.c
@@ -580,6 +580,10 @@ happy_eyeballs_connect(struct addrinfo *res0, int verbose)
 					unreach |= UNREACH_IPV6;
 				}
 				continue;
+			} else if (errno == EADDRNOTAVAIL) {
+				err = errno;
+				close(sd);
+				continue;
 			} else {
 				err = errno;
 				rv = -1;
