diff --git lib/package_alternatives.c lib/package_alternatives.c
index ecdc40ae..2dbb6546 100644
--- lib/package_alternatives.c
+++ lib/package_alternatives.c
@@ -113,6 +113,9 @@ remove_symlinks(struct xbps_handle *xhp, xbps_array_t a, const char *grname)
 	unsigned int i, cnt;
 	struct stat st;
 
+	if (strcmp(grname, "cc") == 0)
+		return 0;
+
 	cnt = xbps_array_count(a);
 	for (i = 0; i < cnt; i++) {
 		xbps_string_t str;
