commit 9c300a93d13a70c4a9e7af50f5f1d53e48cc52db
Author: Duncaen <mail@duncano.de>
Date:   Fri Jun 21 00:32:16 2019 +0200

    bin/xbps-checkvers: fix --format long form

diff --git a/bin/xbps-checkvers/main.c b/bin/xbps-checkvers/main.c
index 43ac0e89..d792e008 100644
--- bin/xbps-checkvers/main.c
+++ bin/xbps-checkvers/main.c
@@ -645,7 +645,7 @@ main(int argc, char **argv)
 		{ "config", required_argument, NULL, 'C' },
 		{ "distdir", required_argument, NULL, 'D' },
 		{ "debug", no_argument, NULL, 'd' },
-		{ "format", no_argument, NULL, 'f' },
+		{ "format", required_argument, NULL, 'f' },
 		{ "installed", no_argument, NULL, 'I' },
 		{ "ignore-conf-repos", no_argument, NULL, 'i' },
 		{ "manual", no_argument, NULL, 'm' },
