From b7b6f7f1b4bc4f30d6641504e81bb038a0b14a86 Mon Sep 17 00:00:00 2001
From: Juan RP <xtraeme@gmail.com>
Date: Sun, 25 Nov 2012 11:03:03 +0100
Subject: [PATCH] xbps-create: use the right var in chdir(2) doh.

---
 bin/xbps-create/main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/bin/xbps-create/main.c b/bin/xbps-create/main.c
index df302eb..53b9ad5 100644
--- bin/xbps-create/main.c
+++ bin/xbps-create/main.c
@@ -710,7 +710,7 @@ main(int argc, char **argv)
 	process_destdir(mutable_files);
 
 	/* Back to original cwd after file tree walk processing */
-	if (chdir(cwd) == -1)
+	if (chdir(p) == -1)
 		die("cannot chdir() to cwd %s:", cwd);
 
 	/*
-- 
1.8.0

