commit d72091a2060817ac10f55fbd8de4e0611d6d6940
Author: Duncaen <mail@duncano.de>
Date:   Wed Jun 26 13:07:16 2019 +0200

    lib/transaction_files.c: fix returning error on hash mismatch

diff --git lib/transaction_files.c lib/transaction_files.c
index 4dbf4584..9a6ca630 100644
--- lib/transaction_files.c
+++ lib/transaction_files.c
@@ -331,6 +331,7 @@ collect_obsoletes(struct xbps_handle *xhp)
 				continue;
 			case ERANGE:
 				/* hash mismatch don't delete it */
+				rv = 0;
 				continue;
 			default:
 				break;
