--- bin/xbps-checkvers/main.c	2018-07-30 13:47:08.000000000 +0200
+++ bin/xbps-checkvers/main.c	2018-08-07 08:47:03.070720109 +0200
@@ -599,9 +599,9 @@
 	else
 		rcv->pkgd = xbps_rpool_get_pkg(&rcv->xhp, srcver);
 
-	srcver = strncat(srcver, "-", 1);
+	srcver = strncat(srcver, "-", 2);
 	srcver = strncat(srcver, version.v.s, version.v.len);
-	srcver = strncat(srcver, "_", 1);
+	srcver = strncat(srcver, "_", 2);
 	srcver = strncat(srcver, revision.v.s, revision.v.len);
 
 	xbps_dictionary_get_cstring_nocopy(rcv->pkgd, "pkgver", &repover);
