Fix will be included in 0.6.3.

--- lib/Makefile.orig	2010-11-02 07:40:41.000000000 +0100
+++ lib/Makefile	2010-11-02 07:40:54.000000000 +0100
@@ -66,7 +66,7 @@ $(OBJS): %.o: %.c
 
 libxbps.so: $(OBJS) $(LIBFETCH_OBJS) $(LIBPROP_OBJS)
 	@printf " [CCLD]\t\t$@\n"
-	@$(CC) $(LDFLAGS) $^ -o $(LIBXBPS_SHLIB)
+	@$(CC) $^ $(LDFLAGS) -o $(LIBXBPS_SHLIB)
 	@-ln -sf $(LIBXBPS_SHLIB) libxbps.so.$(LIBXBPS_MAJOR)
 	@-ln -sf $(LIBXBPS_SHLIB) libxbps.so
 
