--- a/data/_xbps_src
+++ b/data/_xbps_src
@@ -26,6 +26,7 @@
 }
 
 _arguments -s : \
+	'-A[Host architecture]:architecture:($archs)' \
 	'-a[Cross compile packages]:architecture:($archs)' \
 	'-C[Do not remove build directory/autodeps/destdir]' \
 	'-E[Exit immediately when binary package already exists]' \
@@ -64,9 +65,7 @@
 		case $words[1] in
 			build|check|configure|extract|fetch|install|patch|pkg|show|show-avail|show-build-deps|show-hostmakedepends|show-makedepends|show-options|update-check)
 				_arguments ':package:_xbps_src_all_packages' && ret=0;;
-			binary-bootstrap)
-				_arguments '::architecture:($archs)' && ret=0;;
-			bootstrap|bootstrap-update|chroot|clean-repocache|consistency-check|list|purge-distfiles|remove-autodeps|show-repo-updates|show-sys-updates|update-bulk|update-sys|update-hash-cache)
+			binary-bootstrap|bootstrap|bootstrap-update|chroot|clean-repocache|consistency-check|list|purge-distfiles|remove-autodeps|show-repo-updates|show-sys-updates|update-bulk|update-sys|update-hash-cache)
 				# no further arguments
 				ret=0;;
 			clean)
