--- a/data/_xbps_src
+++ b/data/_xbps_src
@@ -25,25 +25,35 @@
 	compadd "$@" -- ${${dirs:t}%-*}
 }
 
 _arguments -s : \
+	'-1[Fail if dependencies are missing]' \
+	'-A[Host architecture]:architecture:($archs)' \
 	'-a[Cross compile packages]:architecture:($archs)' \
+	'-b[Build broken, nocross, and excluded packages]' \
+	'-c[Configuration file]:config: ' \
 	'-C[Do not remove build directory/autodeps/destdir]' \
 	'-E[Exit immediately when binary package already exists]' \
 	'-f[Force building and registering binary packages]' \
 	'-G[Enable XBPS_USE_GIT_REVS]' \
-	'-Q[Enable running the check stage]' \
 	'-g[Enable building -dbg packages]' \
 	'-H[Absolute path to hostdir]:hostdir:_files -/' \
 	'-h[Help]' \
 	'-I[Ignore required dependencies]' \
+	'-i[Make internal errors non-fatal]' \
 	'-j[Number of parallel build jobs]:number: ' \
+	'-K[Enable extended checks]' \
 	'-L[Disable ASCII colors]' \
 	'-m[Absolute path to masterdir]:masterdir:_files -/' \
 	'-N[Disable use of remote repositories]' \
+	'-p[Show additional variables]:variables: ' \
 	'-o[Set package build options]:options: ' \
+	'-Q[Enable running the check stage]' \
 	'-q[Suppress output of xbps-src]' \
 	'-r[Use alternative local repository]:repo:_files -/' \
+	'-s[Make some warnings errors]' \
 	'-t[Create a temporary masterdir]' \
+	'-v[Show verbose messages]' \
+	'-V[Print version]' \
 	'1:target:->target' \
 	'*::args:->args' && ret=0
 
@@ -64,9 +80,7 @@
 		case $words[1] in
 			build|check|configure|extract|fetch|install|patch|pkg|show|show-avail|show-build-deps|show-hostmakedepends|show-makedepends|show-options|update-check)
 				_arguments ':package:_xbps_src_all_packages' && ret=0;;
-			binary-bootstrap)
-				_arguments '::architecture:($archs)' && ret=0;;
-			bootstrap|bootstrap-update|chroot|clean-repocache|consistency-check|list|purge-distfiles|remove-autodeps|show-repo-updates|show-sys-updates|update-bulk|update-sys|update-hash-cache)
+			binary-bootstrap|bootstrap|bootstrap-update|chroot|clean-repocache|consistency-check|list|purge-distfiles|remove-autodeps|show-repo-updates|show-sys-updates|update-bulk|update-sys|update-hash-cache)
 				# no further arguments
 				ret=0;;
 			clean)
