diff --git a/cmake/UraniumTests.cmake b/cmake/UraniumTests.cmake
index 332b24c0..68a34acb 100644
--- a/cmake/UraniumTests.cmake
+++ b/cmake/UraniumTests.cmake
@@ -49,8 +49,3 @@ foreach(_plugin ${_plugins})
     endif()
 endforeach()
 
-#Add code style test.
-add_test(
-    NAME "code-style"
-    COMMAND ${PYTHON_EXECUTABLE} run_mypy.py WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
-)
\ No newline at end of file
