--- src/wfuzz/__init__.py.orig
+++ src/wfuzz/__init__.py
@@ -20,8 +20,8 @@
 try:
     import pycurl

-    if "openssl".lower() not in pycurl.version.lower():
-        print("\nWarning: Pycurl is not compiled against Openssl. Wfuzz might not work correctly when fuzzing SSL sites. Check Wfuzz's documentation for more information.\n")
+    if "ssl".lower() not in pycurl.version.lower():
+        print("\nWarning: Pycurl is not compiled against Open- or LibreSSL. Wfuzz might not work correctly when fuzzing SSL sites. Check Wfuzz's documentation for more information.\n")

     if not hasattr(pycurl, "PATH_AS_IS"):
         print("\nWarning: Pycurl and/or libcurl version is old. PATH_AS_IS option is missing. Wfuzz might not correctly fuzz URLS with '..'.\n")
