--- xpcom/base/nsMemoryReporterManager.cpp.orig    2018-10-25 01:03:36.267772091 +0200
+++ xpcom/base/nsMemoryReporterManager.cpp 2018-10-25 01:42:08.863831518 +0200
@@ -133,6 +133,7 @@
   return GetProcSelfSmapsPrivate(aN);
 }

+#ifdef __GLIBC__
 #ifdef HAVE_MALLINFO
 #define HAVE_SYSTEM_HEAP_REPORTER 1
 static MOZ_MUST_USE nsresult
@@ -154,6 +155,7 @@
     return NS_OK;
 }
 #endif
+#endif

 #elif defined(__DragonFly__) || defined(__FreeBSD__) \
     || defined(__NetBSD__) || defined(__OpenBSD__) \
