Include <netdb.h> to define IPPORT_RESERVED for musl libc

--- netwerk/sctp/src/netinet/sctp_pcb.c	2015-07-14 00:06:57.000000000 +0200
+++ netwerk/sctp/src/netinet/sctp_pcb.c	2015-08-21 10:13:21.509211726 +0200
@@ -35,6 +35,7 @@
 __FBSDID("$FreeBSD: head/sys/netinet/sctp_pcb.c 258765 2013-11-30 12:51:19Z tuexen $");
 #endif
 
+#include <netdb.h>
 #include <netinet/sctp_os.h>
 #ifdef __FreeBSD__
 #include <sys/proc.h>
