Disable fpehandler() for musl libc as it requires access
to the internals of ucontext_t which are not available.

--- toolkit/xre/nsSigHandlers.cpp	2015-07-14 00:08:05.000000000 +0200
+++ toolkit/sre/nsSigHandlers.cpp	2015-08-23 13:04:47.979746292 +0200
@@ -132,7 +132,7 @@
 
 #endif
 
-#ifdef SA_SIGINFO
+#if defined(SA_SIGINFO) && defined(__GLIBC__)
 static void fpehandler(int signum, siginfo_t *si, void *context)
 {
   /* Integer divide by zero or integer overflow. */
@@ -237,7 +237,7 @@
   }
 #endif // CRAWL_STACK_ON_SIGSEGV
 
-#ifdef SA_SIGINFO
+#if defined(SA_SIGINFO) && defined(__GLIBC__)
   /* Install a handler for floating point exceptions and disable them if they occur. */
   struct sigaction sa, osa;
   sa.sa_flags = SA_ONSTACK | SA_RESTART | SA_SIGINFO;
