--- ./media/audioipc/audioipc/src/cmsg.rs.orig	2018-05-09 22:19:14.748631939 +0200
+++ ./media/audioipc/audioipc/src/cmsg.rs	2018-05-09 22:19:22.961620862 +0200
@@ -106,11 +106,11 @@ impl ControlMsgBuilder {
                 return Err(Error::NoSpace);
             }
 
-            let cmsghdr = cmsghdr {
-                cmsg_len: cmsg_len as _,
-                cmsg_level: level,
-                cmsg_type: kind,
-            };
+            use std::mem;
+            let mut cmsghdr: cmsghdr =  unsafe { mem::zeroed() };
+            cmsghdr.cmsg_len = cmsg_len as _;
+            cmsghdr.cmsg_level = level;
+            cmsghdr.cmsg_type = kind;
 
             let cmsghdr = unsafe {
                 slice::from_raw_parts(&cmsghdr as *const _ as *const _, mem::size_of::<cmsghdr>())
