--- dom/system/OSFileConstants.cpp.orig	2013-04-17 06:17:29.798371189 +0000
+++ dom/system/OSFileConstants.cpp	2013-04-17 06:30:30.032285977 +0000
@@ -509,6 +509,11 @@
   INT_CONSTANT(_STAT_VER),
 #endif // defined(_STAT_VER)
 
+  // glibc's stat/lstat/fstat are macros while uclibc's are not
+#if defined(__GLIBC__) && !defined(__UCLIBC__)
+   { "OSFILE_STAT_MACROS", JS::Int32Value(1) },
+#endif // defined(stat)
+
   PROP_END
 };
 
--- toolkit/components/osfile/modules/osfile_unix_back.jsm.orig	2014-04-03 13:08:26.686409787 +0000
+++ toolkit/components/osfile/modules/osfile_unix_back.jsm	2014-04-03 13:34:14.101716259 +0000
@@ -512,7 +512,7 @@
                       /*path*/   Type.fd,
                       /*buf*/    Type.stat.out_ptr
                      );
-       } else if (Const._STAT_VER != undefined) {
+       } else if (Const.OSFILE_STAT_MACROS != undefined) {
          const ver = Const._STAT_VER;
          let xstat_name, lxstat_name, fxstat_name;
          if (OS.Constants.Sys.Name == "SunOS") {
