--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -49,6 +49,7 @@ if (WITH_QT6)
     find_package(Qt6Widgets 6.1.0 REQUIRED)
     set(copyq_qt Qt6)
     add_definitions( -DQT_DEPRECATED_WARNINGS_SINCE=0x051500 )
+    find_package(Qt6 6.10.0 REQUIRED COMPONENTS Gui GuiPrivate Linguist)
 else()
     set(QT_DEFAULT_MAJOR_VERSION 5)
     find_package(Qt5Widgets 5.15.0 REQUIRED)
--- a/src/platform/x11/systemclipboard/CMakeLists.txt
+++ b/src/platform/x11/systemclipboard/CMakeLists.txt
@@ -33,6 +33,7 @@ endif()
 
 target_link_libraries(systemclipboard
                       ${copyq_qt}::Gui
+                      ${copyq_qt}::GuiPrivate
                       ${copyq_qt}::WaylandClient
                       Wayland::Client
 )
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -45,7 +45,6 @@ endif()
 # translations
 file(GLOB copyq_TRANSLATIONS ../translations/*.ts)
 
-find_package(${copyq_qt}LinguistTools)
 if (${copyq_qt}LinguistTools_FOUND)
     qt_add_translation(copyq_QM ${copyq_TRANSLATIONS})
 endif()
