--- a/pkg/atlasrep/gap/userpref.g	2022-09-06 17:41:17.000000000 -0300
+++ b/pkg/atlasrep/gap/userpref.g	2022-10-29 07:46:48.580833277 -0300
@@ -46,7 +46,7 @@
 this information depends on the value of the preference at the time \
 when the AtlasRep package and its data extensions get loaded."
     ],
-  default:= true,
+  default:= false,
   values:= [ true, false ],
   multi:= false,
   package:= "AtlasRep",
