--- Modules/posixmodule.c.orig
+++ Modules/posixmodule.c
@@ -103,8 +103,9 @@ corresponding Unix manual entries for more information on calls.");
 #undef HAVE_SCHED_SETAFFINITY
 #endif
 
-#if defined(HAVE_SYS_XATTR_H) && defined(__GLIBC__) && !defined(__FreeBSD_kernel__) && !defined(__GNU__)
+#if defined(HAVE_SYS_XATTR_H) && defined(__linux__) && !defined(__FreeBSD_kernel__) && !defined(__GNU__)
 #define USE_XATTRS
+#include <linux/limits.h>
 #endif
 
 #ifdef USE_XATTRS
