commit 82a14c87aaa44ae004b9d76af42144155e426290
Author: Göran Weinholt <goran@weinholt.se>
Date:   Fri May 4 18:18:22 2018 +0200

    - Double FMTBUFSIZE to fix compilation with gcc-8
        c/prim5.c

diff --git c/prim5.c c/prim5.c
index 5bee635..69c6bfd 100644
--- c/prim5.c
+++ c/prim5.c
@@ -219,6 +219,6 @@ static ptr s_decode_float(x) ptr x; {
     return S_decode_float(FLODAT(x));
 }

-#define FMTBUFSIZE 60
+#define FMTBUFSIZE 120
 #define CHUNKADDRLT(x, y) (((chunkinfo *)(Scar(x)))->addr < ((chunkinfo *)(Scar(y)))->addr)
 mkmergesort(sort_chunks, merge_chunks, ptr, Snil, CHUNKADDRLT, INITCDR)
