--- a/Makefile
+++ b/Makefile
@@ -308,7 +308,7 @@ smlnj-mlton-x16:
 bootstrap-polyml:
 	$(MAKE) polyml-mlton
 	$(RM) "$(BIN)/$(MLTON)"
-	$(MAKE) BOOTSTRAP_MLTON=$(MLTON).polyml all
+	$(MAKE) OLD_MLTON="$(BIN)/$(MLTON).polyml" all
 	$(RM) "$(LIB)/$(MLTON)-polyml$(EXE)"
 	$(RM) "$(BIN)/$(MLTON).polyml"
 
@@ -317,7 +317,7 @@ polyml-mlton:
 	$(MAKE) dirs
 	$(MAKE) runtime
 	$(MAKE) -C "$(SRC)/mlton" polyml-mlton
-	$(CP) "$(SRC)/mlton/mlton-polyml$(EXE)" "$(LIB)/"
+	$(CP) "$(SRC)/mlton/$(MLTON_OUTPUT)-polyml$(EXE)" "$(LIB)/"
 	$(MAKE) script
 	$(MAKE) basis-no-check
 	$(MAKE) libraries-no-check
