--- a/mlton/Makefile
+++ b/mlton/Makefile
@@ -136,13 +136,13 @@ mlton-stubs.mlb: $(shell "$(RUN_MLTON_DEPS)" -stop f ../lib/stubs/mlton-stubs/so
 		echo '$$(SML_LIB)/basis/sml-nj.mlb';				\
 		echo '$$(SML_LIB)/basis/mlton.mlb';				\
 		echo '$$(SML_LIB)/basis/basis.mlb';				\
-		$(RUN_MLTON) -stop f mlton.mlb |				\
+		$(SRC)/bin/mlbdeps mlton.mlb |					\
 			$(GREP) -v 'mlb$$' |					\
 			$(GREP) 'mlyacc';					\
-		$(RUN_MLTON) -stop f ../lib/stubs/mlton-stubs/sources.mlb |	\
+		$(SRC)/bin/mlbdeps ../lib/stubs/mlton-stubs/sources.mlb |	\
 			$(GREP) -v 'mlb$$' |					\
 			$(GREP) 'mlton-stubs';					\
-		$(RUN_MLTON) -stop f mlton.mlb |				\
+		$(SRC)/bin/mlbdeps mlton.mlb |					\
 			$(GREP) -v 'mlb$$' |					\
 			$(GREP) -v 'sml/basis' |				\
 			$(GREP) -v 'targets' |					\
