--- src/plugins/telegram/telegramplugin.json	2018-08-04 13:02:51.000000000 +0200
+++ -	2018-08-14 17:08:58.242306671 +0200
@@ -58,7 +58,7 @@
     "X-Purpose-Constraints": [
         [
             "application:org.telegram.desktop.desktop",
-            "application:telegramdesktop.desktop"
+            "application:telegram-desktop.desktop"
         ]
     ],
     "X-Purpose-PluginTypes": [
--- src/plugins/telegram/telegramplugin.cpp	2018-08-14 17:16:20.962221257 +0200
+++ -	2018-08-14 17:16:28.212418466 +0200
@@ -56,7 +56,7 @@
 
         void start() override
         {
-            for (const QString &desktopFile: {QStringLiteral("org.telegram.desktop.desktop"), QStringLiteral("telegramdesktop.desktop") }) {
+            for (const QString &desktopFile: {QStringLiteral("org.telegram.desktop.desktop"), QStringLiteral("telegram-desktop.desktop") }) {
                 const auto path = QStandardPaths::locate(QStandardPaths::ApplicationsLocation, desktopFile);
                 if (!path.isEmpty()) {
                     useDesktopFile(path);
