--- pki.c.orig
+++ pki.c
@@ -30,7 +30,7 @@
 #endif
 /* OpenSSL 1.1 made a lot of structures opaque, so we need to
  * define the 1.1 wrappers in previous versions. */
-#if OPENSSL_VERSION_NUMBER < 0x10100000L
+#if OPENSSL_VERSION_NUMBER < 0x10100000L || defined(LIBRESSL_VERSION_NUMBER)
 #define EVP_PKEY_base_id(o)  ((o)->type)
 #define EVP_PKEY_get0_DSA(o) ((o)->pkey.dsa)
 static EVP_MD_CTX *EVP_MD_CTX_new(void)
