diff -rup src/CMakeLists.txt.orig src/CMakeLists.txt
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -542,7 +542,7 @@ SET (AEFFECT_H_FOUND FALSE)
 SET (VST_NATIVE_SUPPORT FALSE)
 
 if (ENABLE_VST_NATIVE)
-      find_file(VST_HEADER_CHECK aeffectx.h PATHS ${VST_HEADER_PATH} NO_DEFAULT_PATH)
+      find_file(VST_HEADER_CHECK aeffectx.h PATHS ${VST_HEADER_PATH} NO_DEFAULT_PATH CMAKE_FIND_ROOT_PATH_BOTH)
       if (VST_HEADER_CHECK STREQUAL "VST_HEADER_CHECK-NOTFOUND")
         message("Native VST support disabled")
       else (VST_HEADER_CHECK STREQUAL "VST_HEADER_CHECK-NOTFOUND")

