From 40afa0ad60e221c7b02fb8febbf9baff42c13cdc Mon Sep 17 00:00:00 2001
From: Michal Vasilek <michal@vasilek.cz>
Date: Mon, 25 Apr 2022 10:19:30 +0200
Subject: [PATCH] Fix intptr_t was not declared in this scope on i386

---
 src/al/dsp.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/al/dsp.cpp b/src/al/dsp.cpp
index 8a4803e4c..3f601dc63 100644
--- a/src/al/dsp.cpp
+++ b/src/al/dsp.cpp
@@ -21,7 +21,7 @@
 //=============================================================================
 
 #include <stdio.h>
-//#include <stdint.h>
+#include <stdint.h>
 //#include "config.h"
 #include "al.h"
 #include "dsp.h"
