--- a/src/java.base/share/classes/sun/security/util/DomainName.java
+++ b/src/java.base/share/classes/sun/security/util/DomainName.java
@@ -206,8 +206,11 @@ class DomainName {
                 new PrivilegedAction<>() {
                     @Override
                     public InputStream run() {
-                        File f = new File(System.getProperty("java.home"),
-                            "lib/security/public_suffix_list.dat");
+                        File f = new File("/usr/share/publicsuffix/public_suffix_list.dat");
+                        if (!f.exists()) {
+                            f = new File(System.getProperty("java.home"),
+                                "lib/security/public_suffix_list.dat");
+                        }
                         try {
                             return new FileInputStream(f);
                         } catch (FileNotFoundException e) {
