--- src/LV2_Plugin/YoshimiLV2Plugin.h	2020-04-20 11:31:08.000000000 +0200
+++ src/LV2_Plugin/YoshimiLV2Plugin.h	2020-12-27 09:24:22.402582559 +0100
@@ -106,7 +106,7 @@
    virtual void registerAudioPort(int) {}
 
    //static methods
-   static LV2_Handle	instantiate (const struct _LV2_Descriptor *, double sample_rate, const char *bundle_path, const LV2_Feature *const *features);
+   static LV2_Handle	instantiate (const struct LV2_Descriptor *, double sample_rate, const char *bundle_path, const LV2_Feature *const *features);
    static void connect_port(LV2_Handle instance, uint32_t port, void *data_location);
    static void activate(LV2_Handle instance);
    static void deactivate(LV2_Handle instance);
@@ -159,7 +159,7 @@
     YoshimiLV2PluginUI(const char *, LV2UI_Write_Function, LV2UI_Controller, LV2UI_Widget *widget, const LV2_Feature *const *features);
     ~YoshimiLV2PluginUI();
     bool init();
-    static LV2UI_Handle	instantiate(const struct _LV2UI_Descriptor *descriptor, const char *plugin_uri, const char *bundle_path, LV2UI_Write_Function write_function, LV2UI_Controller controller, LV2UI_Widget *widget, const LV2_Feature *const *features);
+    static LV2UI_Handle	instantiate(const struct LV2UI_Descriptor *descriptor, const char *plugin_uri, const char *bundle_path, LV2UI_Write_Function write_function, LV2UI_Controller controller, LV2UI_Widget *widget, const LV2_Feature *const *features);
     static void cleanup(LV2UI_Handle ui);
     static void static_guiClosed(void *arg);
     void run();
--- src/LV2_Plugin/YoshimiLV2Plugin.cpp	2020-04-20 11:31:08.000000000 +0200
+++ src/LV2_Plugin/YoshimiLV2Plugin.cpp	2020-12-27 09:27:36.598189783 +0100
@@ -378,7 +378,7 @@
 }
 
 
-LV2_Handle	YoshimiLV2Plugin::instantiate (const struct _LV2_Descriptor *desc, double sample_rate, const char *bundle_path, const LV2_Feature *const *features)
+LV2_Handle	YoshimiLV2Plugin::instantiate (const struct LV2_Descriptor *desc, double sample_rate, const char *bundle_path, const LV2_Feature *const *features)
 {
     SynthEngine *synth = new SynthEngine(0, NULL, true);
     if (synth == NULL || !synth->getRuntime().isRuntimeSetupCompleted()){
@@ -711,9 +711,9 @@
 }
 
 
-LV2UI_Handle YoshimiLV2PluginUI::instantiate(const _LV2UI_Descriptor *descriptor, const char *plugin_uri, const char *bundle_path, LV2UI_Write_Function write_function, LV2UI_Controller controller, LV2UI_Widget *widget, const LV2_Feature * const *features)
+LV2UI_Handle YoshimiLV2PluginUI::instantiate(const LV2UI_Descriptor *descriptor, const char *plugin_uri, const char *bundle_path, LV2UI_Write_Function write_function, LV2UI_Controller controller, LV2UI_Widget *widget, const LV2_Feature * const *features)
 {
-    const _LV2UI_Descriptor *desc = descriptor;
+    const LV2UI_Descriptor *desc = descriptor;
     descriptor = desc;
     const char *plug = plugin_uri;
     plugin_uri = plug;
