--- src/main.cpp.orig	2017-10-15 22:35:17.000000000 +0200
+++ src/main.cpp	2017-10-22 15:26:56.153363369 +0200
@@ -97,6 +97,8 @@
 
 static void *mainGuiThread(void *arg)
 {
+    Fl::lock();
+
     sem_post((sem_t *)arg);
 
     map<SynthEngine *, MusicClient *>::iterator it;
@@ -334,9 +336,6 @@
     int minVmajor = 1; // need to improve this idea
     int minVminor = 5;
 
-    // moved from mainGuiThread() to prevent leaking from early GuiThreadMessage
-    Fl::lock();
-
     if (!mainCreateNewInstance(0))
     {
         goto bail_out;
