--- CMakeLists.txt.orig	2016-11-06 06:11:37.041598678 +0100
+++ CMakeLists.txt	2016-11-06 06:19:21.978622663 +0100
@@ -106,10 +106,10 @@
 ENDIF()
 
 #SET(CMAKE_CXX_FLAGS "-fno-implement-inlines -finline-small-functions -findirect-inlining -fpartial-inlining")
-SET(CMAKE_CXX_FLAGS "-O2 -std=gnu++11 -Wno-deprecated-register")
-IF (NOT CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-    SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-implement-inlines")
-ENDIF()
+#SET(CMAKE_CXX_FLAGS "-O2 -std=gnu++11 -Wno-deprecated-register")
+#IF (NOT CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+#    SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-implement-inlines")
+#ENDIF()
 
 SET(CMAKE_CXX_FLAGS_RELEASE "")
 
