diff --git libelf/elf_getarsym.c libelf/elf_getarsym.c
index 40633aa..132d1a2 100644
--- libelf/elf_getarsym.c
+++ libelf/elf_getarsym.c
@@ -166,7 +166,7 @@ elf_getarsym (elf, ptr)
 
       /* We have an archive.  The first word in there is the number of
 	 entries in the table.  */
-      uint64_t n;
+      uint64_t n = 0;
       size_t off = elf->start_offset + SARMAG + sizeof (struct ar_hdr);
       if (read_number_entries (&n, elf, &off, index64_p) < 0)
 	{
