From cb5f396d2360991c3c333f52f005cb32c48b73a1 Mon Sep 17 00:00:00 2001
From: Nathan Owens <ndowens@artixlinux.org>
Date: Fri, 6 Nov 2020 13:55:34 -0600
Subject: [PATCH] fix-aarch64_fregs

---
 backends/aarch64_initreg.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git backends/aarch64_initreg.c backends/aarch64_initreg.c
index daf6f37..8a09904 100644
--- backends/aarch64_initreg.c
+++ backends/aarch64_initreg.c
@@ -82,8 +82,7 @@ aarch64_set_initial_registers_tid (pid_t tid __attribute__ ((unused)),
 
   Dwarf_Word dwarf_fregs[32];
   for (int r = 0; r < 32; r++)
-    dwarf_fregs[r] = fregs.vregs[r] & 0xFFFFFFFF;
-
+		dwarf_fregs[r] = (Dwarf_Word)fregs.vregs[r];
   if (! setfunc (64, 32, dwarf_fregs, arg))
     return false;
 
-- 
2.29.2

