From 4a6602ae33489bc74da76bb374cb2c22a2a35d88 Mon Sep 17 00:00:00 2001
From: Nathan Owens <ndowens@artixlinux.org>
Date: Fri, 6 Nov 2020 09:58:19 -0600
Subject: [PATCH] musl-cdefs

---
 lib/fixedsizehash.h |  1 -
 libelf/elf.h        | 10 ++++++++++
 2 files changed, 10 insertions(+), 1 deletion(-)

diff --git lib/fixedsizehash.h lib/fixedsizehash.h
index dac2a5f..43016fc 100644
--- lib/fixedsizehash.h
+++ lib/fixedsizehash.h
@@ -30,7 +30,6 @@
 #include <errno.h>
 #include <stdlib.h>
 #include <string.h>
-#include <sys/cdefs.h>
 
 #include <system.h>
 
diff --git libelf/elf.h libelf/elf.h
index 6439c1a..b9c80ba 100644
--- libelf/elf.h
+++ libelf/elf.h
@@ -22,6 +22,16 @@
 /* Standard ELF types.  */
 
 #include <stdint.h>
+#if !defined(__BEGIN_DECLS)
+	/* C++ needs to know that types and declarations are C, not C++.  */
+	#ifdef  __cplusplus
+	# define __BEGIN_DECLS  extern "C" {
+	# define __END_DECLS    }
+	#else
+	# define __BEGIN_DECLS
+	# define __END_DECLS
+	#endif
+	#endif
 
 /* Type for a 16-bit quantity.  */
 typedef uint16_t Elf32_Half;
-- 
2.29.2

