--- a/common/autoconf/generated-configure.sh	2017-12-20 00:25:20.000000000 +0100
+++ b/common/autoconf/generated-configure.sh	2020-08-18 23:29:24.482332056 +0200
@@ -34082,7 +34082,7 @@
     COMPILER_VERSION_STRING=`$ECHO $COMPILER_VERSION_OUTPUT | \
         $SED -e 's/ *[Uu]sage:.*//'`
     COMPILER_VERSION_NUMBER=`$ECHO $COMPILER_VERSION_OUTPUT | \
-        $SED -e "s/^.*[ ,\t]$COMPILER_NAME[ ,\t]\([1-9]\.[0-9][0-9]*\).*/\1/"`
+        $SED -e "s/^.*[ ,\t]$COMPILER_NAME[ ,\t]\([1-9][0-9]*\.[0-9][0-9]*\).*/\1/"`
   elif test  "x$TOOLCHAIN_TYPE" = xxlc; then
     # xlc -qversion output typically looks like
     #     IBM XL C/C++ for AIX, V11.1 (5724-X13)
@@ -34143,7 +34143,7 @@
     COMPILER_VERSION_STRING=`$ECHO $COMPILER_VERSION_OUTPUT | \
         $SED -e 's/ *Copyright .*//'`
     COMPILER_VERSION_NUMBER=`$ECHO $COMPILER_VERSION_OUTPUT | \
-        $SED -e 's/^.* \([1-9]\.[0-9.]*\)[^0-9.].*$/\1/'`
+        $SED -e 's/^.* \([1-9][0-9]*\.[0-9.]*\)[^0-9.].*$/\1/'`
   elif test  "x$TOOLCHAIN_TYPE" = xclang; then
     # clang --version output typically looks like
     #    Apple LLVM version 5.0 (clang-500.2.79) (based on LLVM 3.3svn)
--- a/common/autoconf/toolchain.m4	2017-12-20 00:25:20.000000000 +0100
+++ b/common/autoconf/toolchain.m4	2020-08-18 23:32:12.832325343 +0200
@@ -424,7 +424,7 @@
     # Collapse compiler output into a single line
     COMPILER_VERSION_STRING=`$ECHO $COMPILER_VERSION_OUTPUT`
     COMPILER_VERSION_NUMBER=`$ECHO $COMPILER_VERSION_OUTPUT | \
-        $SED -e 's/^.* version \(@<:@1-9@:>@@<:@0-9.@:>@*\).*$/\1/'`
+        $SED -e 's/^.* version \(@<:@1-9@:>@@<:@0-9@:>@*\.@<:@0-9.@:>@*\).*$/\1/'`
   else
       AC_MSG_ERROR([Unknown toolchain type $TOOLCHAIN_TYPE.])
   fi
