--- a/iouyap.c
+++ b/iouyap.c
@@ -35,7 +35,9 @@
 #include <linux/if_tun.h>
 #include <arpa/inet.h>
 #include <netpacket/packet.h>
+#if HAVE_NETINET_IF_ETHER_H && !HAVE_LINUX_IF_TUN_H
 #include <net/ethernet.h>
+#endif
 #include <netdb.h>
 #include <net/if_arp.h>
 #include <linux/rtnetlink.h>
--- a/iouyap.h
+++ b/iouyap.h
@@ -24,6 +24,7 @@
 #include <netinet/in.h>
 #include <errno.h>
 #include <sys/un.h>
+#include <sys/types.h>
 
 #define NAME                  "iouyap"
 #define VERSION               "0.97.0"
--- a/netmap.c
+++ b/netmap.c
@@ -28,6 +28,7 @@
 #include <string.h>
 #include <netdb.h>
 #include <sys/socket.h>
+#include <sys/types.h>
 #include <netinet/in.h>
 #include <arpa/inet.h>
 
