From 94b3bedd1ed56d70168dfd09594d2f0d8d6dea96 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=C4=90o=C3=A0n=20Tr=E1=BA=A7n=20C=C3=B4ng=20Danh?=
 <congdanhqx@gmail.com>
Date: Sun, 29 Sep 2019 12:32:46 +0700
Subject: [PATCH 1/2] use versioned pcsclite library on musl-libc as well

---
 scd/scdaemon.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scd/scdaemon.c b/scd/scdaemon.c
index 6d068d3d3..2f2586452 100644
--- a/scd/scdaemon.c
+++ b/scd/scdaemon.c
@@ -207,7 +207,7 @@ static struct debug_flags_s debug_flags [] =
 #define DEFAULT_PCSC_DRIVER "winscard.dll"
 #elif defined(__APPLE__)
 #define DEFAULT_PCSC_DRIVER "/System/Library/Frameworks/PCSC.framework/PCSC"
-#elif defined(__GLIBC__)
+#elif defined(__linux__)
 #define DEFAULT_PCSC_DRIVER "libpcsclite.so.1"
 #else
 #define DEFAULT_PCSC_DRIVER "libpcsclite.so"
-- 
2.37.1

