--- a/scd/scdaemon.c	2019-09-29 12:18:47.810111976 +0700
+++ b/scd/scdaemon.c	2019-09-29 12:20:38.079117234 +0700
@@ -189,7 +189,7 @@
 #define DEFAULT_PCSC_DRIVER "winscard.dll"
 #elif defined(__APPLE__)
 #define DEFAULT_PCSC_DRIVER "/System/Library/Frameworks/PCSC.framework/PCSC"
-#elif defined(__GLIBC__)
+#elif defined(__linux__)
 #define DEFAULT_PCSC_DRIVER "libpcsclite.so.1"
 #else
 #define DEFAULT_PCSC_DRIVER "libpcsclite.so"
