From 9f9fd4ec05587aae2b65a20fadabccb02c593ce2 Mon Sep 17 00:00:00 2001
From: Leah Neukirchen <leah@vuxu.org>
Date: Mon, 7 Feb 2022 18:52:43 +0100
Subject: [PATCH 1/2] unbreak i128 support

---
 Cargo.lock | 5 +++--
 Cargo.toml | 3 ++-
 2 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/Cargo.lock b/Cargo.lock
index 73e90fb..1b0c1e9 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -418,9 +418,9 @@ dependencies = [
 
 [[package]]
 name = "parse_int"
-version = "0.5.0"
+version = "0.6.0"
 source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "15f96500577cfa0a3bad8a88a3c4daa66684828af2e7d349012fa7fc3c725f0c"
+checksum = "2d695b79916a2c08bcff7be7647ab60d1402885265005a6658ffe6d763553c5a"
 dependencies = [
  "num-traits 0.2.14",
 ]
@@ -792,6 +792,7 @@ dependencies = [
  "log",
  "nix",
  "nom",
+ "num-traits 0.2.14",
  "parse_int",
  "rand 0.3.23",
  "structopt",
diff --git a/Cargo.toml b/Cargo.toml
index 235a30b..1bd3319 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -28,7 +28,8 @@ json = "0.11"
 rand = "0.3"
 unix_socket = "0.5"
 unicode-segmentation = "1.2"
-parse_int = "0.5"
+parse_int = "0.6"
+num-traits = { version = "0.2", features = ["i128"] }
 
 unsegen = "0.3"
 unsegen_signals = "0.3"
-- 
2.40.0

