commit db6e990cc46d69526beae930e0048f65d5510c34
Author: q66 <daniel@octaforge.org>
Date:   Tue Aug 10 18:52:51 2021 +0200

    fix build on platforms without 64-bit atomics

diff --git a/src/main.rs b/src/main.rs
index ccfccab..fe2b52c 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -284,7 +284,7 @@ struct MpscTimer {
     next_sender: Option<Sender<Event>>,
     sender: Sender<Event>,
     evt_fn: Box<dyn Fn() -> Event>,
-    counter: std::sync::Arc<std::sync::atomic::AtomicU64>,
+    counter: std::sync::Arc<std::sync::atomic::AtomicUsize>,
 }
 
 impl MpscTimer {
@@ -293,7 +293,7 @@ impl MpscTimer {
             next_sender: Some(sender.clone()),
             sender,
             evt_fn,
-            counter: std::sync::Arc::new(std::sync::atomic::AtomicU64::new(0)),
+            counter: std::sync::Arc::new(std::sync::atomic::AtomicUsize::new(0)),
         }
     }
 
