--- a/cmake/modules/FindSndFile.cmake
+++ b/cmake/modules/FindSndFile.cmake
@@ -5,22 +5,21 @@
 # SndFile::SndFile - the sndfile library
 
 find_package(PkgConfig QUIET)
-if(PkgConfig_FOUND)
-    pkg_check_modules(SndFile sndfile QUIET)
-else()
-    find_path(
-        SndFile_INCLUDEDIR
-        NAMES sndfile.h
-        PATH_SUFFIXES sndfile
-        DOC "SndFile include directory"
-    )
+pkg_check_modules(SndFile sndfile QUIET)
+find_path(
+    SndFile_INCLUDEDIR
+    NAMES sndfile.h
+    PATH_SUFFIXES sndfile
+    HINTS ${PC_SndFile_INCLUDE_DIRS}
+    DOC "SndFile include directory"
+)
 
-    find_library(
-        SndFile_LINK_LIBRARIES
-        NAMES sndfile sndfile-1
-        DOC "SndFile library"
-    )
-endif()
+find_library(
+    SndFile_LINK_LIBRARIES
+    NAMES sndfile sndfile-1
+    HINTS ${PC_SndFile_LIBRARY_DIRS}
+    DOC "SndFile library"
+)
 
 include(FindPackageHandleStandardArgs)
 find_package_handle_standard_args(
@@ -30,17 +29,6 @@
     SndFile_INCLUDEDIR
 )
 
-file(STRINGS "${SndFile_INCLUDEDIR}/sndfile.h"
-     SndFile_SUPPORTS_SET_COMPRESSION_LEVEL
-     REGEX ".*SFC_SET_COMPRESSION_LEVEL.*"
-)
-if(SndFile_SUPPORTS_SET_COMPRESSION_LEVEL)
-    set(SndFile_SUPPORTS_SET_COMPRESSION_LEVEL ON)
-else()
-    set(SndFile_SUPPORTS_SET_COMPRESSION_LEVEL OFF)
-endif()
-mark_as_advanced(SndFile_SUPPORTS_SET_COMPRESSION_LEVEL)
-
 if(SndFile_FOUND)
     if(NOT TARGET SndFile::sndfile)
         add_library(SndFile::sndfile INTERFACE IMPORTED)
