--- a/src/plugins/vumeter/vumeterwidget.cpp
+++ b/src/plugins/vumeter/vumeterwidget.cpp
@@ -33,6 +33,7 @@
 #include <QActionGroup>
 #include <QBasicTimer>
 #include <QContextMenuEvent>
+#include <QElapsedTimer>
 #include <QJsonObject>
 #include <QMenu>
 #include <QPainter>
--- a/src/utils/starrating.cpp
+++ b/src/utils/starrating.cpp
@@ -104,7 +104,7 @@ void StarRating::paint(QPainter* painter
                                  .arg(m_maxCount)
                                  .arg(mode == EditMode::Editable ? 1 : 0)
                                  .arg(rect.width())
-                                 .arg(alignment);
+                                 .arg(static_cast<int>(alignment));
 
     QPixmap pixmap;
     if(!QPixmapCache::find(cacheKey, &pixmap)) {
