From f57700420a3c72300e5135426d5a953547f90872 Mon Sep 17 00:00:00 2001
From: Ivan V <ivv@i1v.org>
Date: Sun, 30 Nov 2025 16:03:43 +0000
Subject: [PATCH] [utils] Fix compatibility with Qt 6.10.1

---
 src/utils/starrating.cpp | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/src/utils/starrating.cpp b/src/utils/starrating.cpp
index 1e71e1816..0be778632 100644
--- a/src/utils/starrating.cpp
+++ b/src/utils/starrating.cpp
@@ -99,12 +99,12 @@ void StarRating::setStarScale(int scale)
 void StarRating::paint(QPainter* painter, const QRect& rect, const QPalette& palette, EditMode mode,
                        Qt::Alignment alignment) const
 {
-    const QString cacheKey = u"StarRating:%1|%2|%3|%4|%5|%6"_s.arg(m_rating)
-                                 .arg(m_scale)
-                                 .arg(m_maxCount)
-                                 .arg(mode == EditMode::Editable ? 1 : 0)
-                                 .arg(rect.width())
-                                 .arg(alignment);
+    const QString cacheKey = u"StarRating:%1|%2|%3|%4|%5|%6"_s.arg(static_cast<float>(m_rating))
+                                 .arg(static_cast<int>(m_scale))
+                                 .arg(static_cast<int>(m_maxCount))
+                                 .arg(static_cast<int>(mode == EditMode::Editable ? 1 : 0))
+                                 .arg(static_cast<int>(rect.width()))
+                                 .arg(static_cast<int>(alignment));

     QPixmap pixmap;
     if(!QPixmapCache::find(cacheKey, &pixmap)) {

--- a/src/plugins/vumeter/vumeterwidget.cpp
+++ b/src/plugins/vumeter/vumeterwidget.cpp
@@ -37,6 +37,7 @@
 #include <QMenu>
 #include <QPainter>
 #include <QTimerEvent>
+#include <QElapsedTimer>

 using namespace Qt::StringLiterals;
