--- a/modules/readline/readline.lisp	2016-09-20 22:23:51.876046496 +0200
+++ b/modules/readline/readline.lisp	2016-09-20 22:24:11.070047486 +0200
@@ -424,7 +424,7 @@
    "The version of this incarnation of the readline library, e.g., 0x0402."))
 (def-c-var gnu-readline-p (:name "rl_gnu_readline_p") (:type int)
   (:documentation "True if this is real GNU readline."))
-(def-c-var readline-state (:name "rl_readline_state") (:type int)
+(def-c-var readline-state (:name "rl_readline_state") (:type ulong)
   (:documentation "Flags word encapsulating the current readline state."))
 (def-c-var editing-mode (:name "rl_editing_mode") (:type int)
   (:documentation "Says which editing mode readline is currently using.
