--- a/crates/engine_xetex/build.rs
+++ b/crates/engine_xetex/build.rs
@@ -184,7 +184,7 @@ const C_FILES: &[&str] = &[
 ];
 
 const CXX_FLAGS: &[&str] = &[
-    "-std=c++14",
+    "-std=c++17",
     "-Wall",
     "-Wdate-time",
     "-Wendif-labels",
--- a/crates/xetex_layout/build.rs
+++ b/crates/xetex_layout/build.rs
@@ -86,7 +86,7 @@ fn main() {
     let mut cppcfg = cc::Build::new();
 
     let cppflags = [
-        "-std=c++14",
+        "-std=c++17",
         "-Wall",
         "-Wdate-time",
         "-Wendif-labels",
