--- main.c
+++ main.c
@@ -19,6 +19,10 @@
 #include "archcommon.h"
 #include "header.h"
 
+#if !defined(FNM_EXTMATCH)
+#define FNM_EXTMATCH 0
+#endif
+
 static const char *argv0;
 bool no_act = false;
 bool run_scripts = true;
--- archhurd.c
+++ archhurd.c
@@ -6,6 +6,10 @@
 
 #include "archcommon.h"
 
+#if !defined(FNM_EXTMATCH)
+#define FNM_EXTMATCH 0
+#endif
+
 bool variable_match(const char *iface, const char *variable, const char *pattern) {
 	if (!strcasecmp(variable, "name"))
 		return fnmatch(pattern, iface, FNM_EXTMATCH) == 0;
--- archkfreebsd.c
+++ archkfreebsd.c
@@ -9,6 +9,10 @@
 
 #include "archcommon.h"
 
+#if !defined(FNM_EXTMATCH)
+#define FNM_EXTMATCH 0
+#endif
+
 static bool match_mac(const char *iface, const char *pattern) {
 	for (struct ifaddrs *ifa = ifap; ifa; ifa = ifa->ifa_next) {
 		if (ifa->ifa_addr->sa_family != AF_LINK)
--- archlinux.c
+++ archlinux.c
@@ -10,6 +10,10 @@
 
 #include "archcommon.h"
 
+#if !defined(FNM_EXTMATCH)
+#define FNM_EXTMATCH 0
+#endif
+
 bool variable_match(const char *iface, const char *variable, const char *pattern) {
 	// Map platform-independent variables to sysfs names
 	if(!strcasecmp(variable, "mac"))
