--- a/Makefile	2017-08-31 13:27:17.009081874 +0200
+++ b/Makefile	2017-08-31 13:28:55.868955322 +0200
@@ -22,13 +22,13 @@ all : ifup ifdown ifquery ifup.8 ifdown.
 .SECONDARY: link.c ipx.c can.c meta.c inet6.c inet.c
 
 install :
-	install -m 0755 -d     ${BASEDIR}/sbin
-	install -m 0755 ifup   ${BASEDIR}/sbin
-	ln -s /sbin/ifup ${BASEDIR}/sbin/ifdown
-	ln -s /sbin/ifup ${BASEDIR}/sbin/ifquery
-	install -D -m 0755 settle-dad.sh $(BASEDIR)/lib/ifupdown/settle-dad.sh
-	install -D -m 0755 wait-for-ll6.sh $(BASEDIR)/lib/ifupdown/wait-for-ll6.sh
-	install -D -m 0755 wait-online.sh $(BASEDIR)/lib/ifupdown/wait-online.sh
+	install -m 0755 -d     ${BASEDIR}/usr/bin
+	install -m 0755 ifup   ${BASEDIR}/usr/bin
+	ln -sr /usr/bin/ifup ${BASEDIR}/usr/bin/ifdown
+	ln -sr /usr/bin/ifup ${BASEDIR}/usr/bin/ifquery
+	install -D -m 0755 settle-dad.sh $(BASEDIR)/usr/lib/ifupdown/settle-dad.sh
+	install -D -m 0755 wait-for-ll6.sh $(BASEDIR)/usr/lib/ifupdown/wait-for-ll6.sh
+	install -D -m 0755 wait-online.sh $(BASEDIR)/usr/lib/ifupdown/wait-online.sh
 
 clean :
 	rm -f *.o $(patsubst %.defn,%.c,$(DEFNFILES)) *~

