--- src/lib/FHCollector.cpp	2017-09-16 12:28:50.000000000 +0200
+++ src/lib/FHCollector.cpp	2018-08-09 11:05:32.164697936 +0200
@@ -1907,7 +1907,7 @@
   textObjectProps.insert("svg:width", width);
   for (int i=0; i<4; ++i) // osnola: let assume that there is no padding
   {
-    char const *(padding[])= {"fo:padding-left","fo:padding-right","fo:padding-top","fo:padding-bottom"};
+    char const *padding[] = {"fo:padding-left","fo:padding-right","fo:padding-top","fo:padding-bottom"};
     textObjectProps.insert(padding[i],0,librevenge::RVNG_POINT);
   }
   if (!FH_ALMOST_ZERO(rotation))
