From b10b26c41dd0f32e975fb319fba68eda3a3a8d3f Mon Sep 17 00:00:00 2001
From: t0b3 <t0b3@users.noreply.github.com>
Date: Tue, 25 Nov 2025 15:21:15 +0100
Subject: [PATCH] fix: support Qt 6.10.1

fixes: #64055
---
 src/core/qgssnappingutils.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/core/qgssnappingutils.cpp b/src/core/qgssnappingutils.cpp
index 8f2f33a05dd4..73b418ec6ec5 100644
--- a/src/core/qgssnappingutils.cpp
+++ b/src/core/qgssnappingutils.cpp
@@ -646,7 +646,7 @@ QString QgsSnappingUtils::dump()
     msg += QString( "layer : %1\n"
                     "config: %2   tolerance %3 %4\n" )
            .arg( layer.layer->name() )
-           .arg( layer.type ).arg( layer.tolerance ).arg( static_cast<int>( layer.unit ) );
+           .arg( static_cast<int>( layer.type ) ).arg( layer.tolerance ).arg( static_cast<int>( layer.unit ) );
 
     if ( mStrategy == IndexAlwaysFull || mStrategy == IndexHybrid || mStrategy == IndexExtent )
     {
