diff -up uberwriter_bugfixes/uberwriter/UberwriterWindow.py.orig uberwriter_bugfixes/uberwriter/UberwriterWindow.py
--- uberwriter_bugfixes/uberwriter/UberwriterWindow.py.orig	2012-12-10 13:24:22.947869013 -0600
+++ uberwriter_bugfixes/uberwriter/UberwriterWindow.py	2012-12-10 13:24:38.839927266 -0600
@@ -19,7 +19,6 @@ import subprocess
 import os
 import codecs
 import webbrowser
-import apt
 import urllib
 
 from locale import gettext as _
@@ -528,25 +527,6 @@ class UberwriterWindow(Window):
         self.export("html")
 
     def export_as_pdf(self, widget, data=None):
-        if self.texlive_installed == False:
-            try:
-                cache = apt.Cache()
-                inst = cache["texlive"].is_installed
-            except:
-                inst = True
-
-            if inst == False:
-                dialog = Gtk.MessageDialog(self,
-                    Gtk.DialogFlags.MODAL | Gtk.DialogFlags.DESTROY_WITH_PARENT,
-                    Gtk.MessageType.INFO,
-                    None, 
-                    _("You can not export to PDF.")
-                )
-                dialog.format_secondary_markup(_("Please install <a href=\"apt:texlive\">texlive</a> from the software center."))
-                response = dialog.run()
-                return
-            else:
-                self.texlive_installed = True
         self.export("pdf")
 
     def copy_html_to_clipboard(self, widget, date=None):
@@ -851,10 +831,6 @@ class UberwriterWindow(Window):
         self.UberwriterAdvancedExportDialog = UberwriterAdvancedExportDialog
 
         # Code for other initialization actions should be added here.
-        
-        # Texlive checker
-
-        self.texlive_installed = False
 
         # Draw background
         self.background_image = helpers.get_media_path('bg_light.png')
