--- a/libetpan-config.in	2015-03-29 14:26:51.317434432 +0200
+++ b/libetpan-config.in	2015-03-29 14:34:29.172407320 +0200
@@ -47,8 +47,7 @@
       echo $includedir
       ;;
     --libs)
-      libdir=-L@libdir@
-      echo $libdir -letpan@LIBSUFFIX@ @LDFLAGS@ @SSLLIBS@ @GNUTLSLIB@ @LIBICONV@ @DBLIB@ @LIBS@ @SASLLIBS@
+      echo -letpan@LIBSUFFIX@ @LDFLAGS@ @SSLLIBS@ @GNUTLSLIB@ @LIBICONV@ @DBLIB@ @LIBS@ @SASLLIBS@
       ;;
     *)
       echo "${usage}" 1>&2
