diff --git a/DaemonLinux.cpp b/DaemonLinux.cpp
index edd15f07..126b4689 100644
--- DaemonLinux.cpp
+++ DaemonLinux.cpp
@@ -78,9 +78,9 @@ namespace i2p
 
 #if !defined(__OpenBSD__)
 				// point std{in,out,err} descriptors to /dev/null
-                stdin  = freopen("/dev/null", "r", stdin);
-				stdout = freopen("/dev/null", "w", stdout);
-				stderr = freopen("/dev/null", "w", stderr);
+                freopen("/dev/null", "r", stdin);
+				freopen("/dev/null", "w", stdout);
+				freopen("/dev/null", "w", stderr);
 #endif
 			}
 
