diff --git Makefile Makefile
index 34efed6..6ad418e 100644
--- Makefile
+++ Makefile
@@ -10,21 +10,21 @@ RM	:= rm
 # flags
 CFLAGS	+= -std=c11 -O2 -fPIC -Wall -Werror
 ifneq ($(wildcard /usr/include/iniparser),)
-CFLAGS	+= -I/usr/include/iniparser
+override CFLAGS	+= -I/usr/include/iniparser
 endif
-CFLAGS	+= -liniparser
+override CFLAGS	+= -liniparser
 CFLAGS_SYSTEMD := $(shell pkg-config --cflags --libs libsystemd 2>/dev/null)
 ifneq ($(CFLAGS_SYSTEMD),)
-CFLAGS	+= -DHAVE_SYSTEMD $(CFLAGS_SYSTEMD)
+override CFLAGS	+= -DHAVE_SYSTEMD $(CFLAGS_SYSTEMD)
 endif
-CFLAGS	+= $(shell pkg-config --cflags --libs libmpdclient)
-CFLAGS	+= $(shell pkg-config --cflags --libs libnotify)
-CFLAGS_LIBAV := $(shell pkg-config --cflags --libs libavformat libavutil 2>/dev/null)
+override CFLAGS	+= $(shell pkg-config --cflags --libs libmpdclient)
+override CFLAGS	+= $(shell pkg-config --cflags --libs libnotify)
+override CFLAGS_LIBAV := $(shell pkg-config --cflags --libs libavformat libavutil 2>/dev/null)
 ifneq ($(CFLAGS_LIBAV),)
-CFLAGS	+= -DHAVE_LIBAV $(CFLAGS_LIBAV)
-CFLAGS	+= -lmagic
+override CFLAGS	+= -DHAVE_LIBAV $(CFLAGS_LIBAV)
+override CFLAGS	+= -lmagic
 endif
-LDFLAGS	+= -Wl,-z,now -Wl,-z,relro -pie
+override LDFLAGS	+= -Wl,-z,now -Wl,-z,relro -pie
 
 # this is just a fallback in case you do not use git but downloaded
 # a release tarball...

