--- druntime/src/core/sys/linux/execinfo.d
+++ druntime/src/core/sys/linux/execinfo.d
@@ -7,7 +7,7 @@
  */
 module core.sys.linux.execinfo;
 
-version (linux):
+version (none):
 extern (C):
 nothrow:
 
--- druntime/src/rt/backtrace/dwarf.d
+++ druntime/src/rt/backtrace/dwarf.d
@@ -15,7 +15,7 @@ module rt.backtrace.dwarf;
 version(CRuntime_Glibc) version = linux_or_freebsd;
 else version(FreeBSD) version = linux_or_freebsd;
 
-version(linux_or_freebsd):
+version(none):
 
 import rt.util.container.array;
 import rt.backtrace.elf;
