From 6bbdd200e89bdbf180071e5281c0feb9617de81d Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Fri, 19 Feb 2010 03:32:00 +0100
Subject: [PATCH 47/62] vala: fix definition of the GLib mainloop adapter

---
 vala/libpulse.vapi |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/vala/libpulse.vapi b/vala/libpulse.vapi
index b02bb56..d47ef73 100644
--- a/vala/libpulse.vapi
+++ b/vala/libpulse.vapi
@@ -879,11 +879,11 @@ namespace PulseAudio {
         }
 
         [Compact]
-        [CCode (cname="pa_glib_mainloop", cprefix="pa_glib_mainloop_", free_function="pa_glib_mainloop_free")]
+        [CCode (cheader_filename="pulse/glib-mainloop.h", cname="pa_glib_mainloop", cprefix="pa_glib_mainloop_", free_function="pa_glib_mainloop_free")]
         public class GLibMainLoop {
 
                 [CCode (cname="pa_glib_mainloop_new")]
-                public GLibMainLoop();
+                public GLibMainLoop(MainContext? c = null);
 
                 public unowned MainLoopApi get_api();
         }
-- 
1.7.0

