--- pulseaudio-0.9.21/src/pulse/context.c-orig	2010-09-29 08:56:41.000000000 +0200
+++ pulseaudio-0.9.21/src/pulse/context.c	2010-09-29 13:04:15.000000000 +0200
@@ -624,7 +624,7 @@ static pa_strlist *prepend_per_user(pa_s
     char *ufn;
 
 #ifdef ENABLE_LEGACY_RUNTIME_DIR
-    static char *legacy_dir;
+    char *legacy_dir;
 
     /* The very old per-user instance path (< 0.9.11). This is supported only to ease upgrades */
     if ((legacy_dir = get_very_old_legacy_runtime_dir())) {
