--- setup.py.orig	2016-03-27 00:51:03.561989163 -0400
+++ setup.py	2016-03-27 00:56:28.082046356 -0400
@@ -132,7 +132,7 @@
         # Certifi is also optional on 2.7.9+, although making our dependencies
         # conditional on micro version numbers seems like a bad idea
         # until we have more declarative metadata.
-        install_requires.append('certifi')
+        pass
     if sys.version_info < (3, 5):
         install_requires.append('backports_abc>=0.4')
     kwargs['install_requires'] = install_requires
