diff --git setup.py setup.py
index f09169f..d42c486 100644
--- setup.py
+++ setup.py
@@ -120,7 +120,7 @@ if (platform.python_implementation() == 'CPython' and
         # Certifi is also optional on 2.7.9+, although making our dependencies
         # conditional on micro version numbers seems like a bad idea
         # until we have more declarative metadata.
-        install_requires.append('certifi')
+        pass
     kwargs['install_requires'] = install_requires

 setup(
diff --git tornado/simple_httpclient.py tornado/simple_httpclient.py
index f0f73fa..ffe3e40 100644
--- tornado/simple_httpclient.py
+++ tornado/simple_httpclient.py
@@ -33,17 +33,9 @@ except ImportError:
     # ssl is not available on Google App Engine.
     ssl = None

-try:
-    import certifi
-except ImportError:
-    certifi = None
-

 def _default_ca_certs():
-    if certifi is None:
-        raise Exception("The 'certifi' package is required to use https "
-                        "in simple_httpclient")
-    return certifi.where()
+    return "/etc/ssl/certs/ca-certificates.crt"


 class SimpleAsyncHTTPClient(AsyncHTTPClient):
