diff --git setup.py setup.py
index f09169f..f795807 100644
--- setup.py
+++ setup.py
@@ -121,7 +121,7 @@ def build_extension(self, ext):
 if setuptools is not None:
     # If setuptools is not available, you're on your own for dependencies.
     install_requires = []
-    if sys.version_info < (3, 2):
+    if sys.version_info < (2, 7, 9) or (3, 0) <= sys.version_info < (3, 2):
         install_requires.append('backports.ssl_match_hostname')
     kwargs['install_requires'] = install_requires
 

