--- a/Makefile
+++ b/Makefile
@@ -57,7 +57,7 @@
 CARGO_PROFILE=release
 # CARGO_PROFILE=debug
 
-CARGO_BUILD_ARGS=--$(CARGO_PROFILE)
+CARGO_BUILD_ARGS=--$(CARGO_PROFILE) --target $(RUST_TARGET)
 CARGO_BUILD=$(CARGO) build $(CARGO_BUILD_ARGS)
 
 CARGO_CLEAN=$(CARGO) clean $(CARGO_CLEAN_ARGS)
--- a/Makefile
+++ b/Makefile
@@ -200,7 +200,7 @@
 install: INITRAMFS_HOOK=$(INITRAMFS_DIR)/hooks/bcachefs
 install: INITRAMFS_SCRIPT=$(INITRAMFS_DIR)/scripts/local-premount/bcachefs
 install: bcachefs $(optional_install)
-	$(INSTALL) -m0755 -D target/release/bcachefs -t $(DESTDIR)$(ROOT_SBINDIR)
+	$(INSTALL) -m0755 -D target/*/release/bcachefs -t $(DESTDIR)$(ROOT_SBINDIR)
 	$(INSTALL) -m0644 -D bcachefs.8    -t $(DESTDIR)$(PREFIX)/share/man/man8/
 	$(INSTALL) -m0755 -D initramfs/script $(DESTDIR)$(INITRAMFS_SCRIPT)
 	$(INSTALL) -m0755 -D initramfs/hook   $(DESTDIR)$(INITRAMFS_HOOK)
--- a/Makefile
+++ b/Makefile
@@ -199,7 +199,7 @@
 .PHONY: install
 install: INITRAMFS_HOOK=$(INITRAMFS_DIR)/hooks/bcachefs
 install: INITRAMFS_SCRIPT=$(INITRAMFS_DIR)/scripts/local-premount/bcachefs
-install: bcachefs $(optional_install)
+install: $(optional_install)
 	$(INSTALL) -m0755 -D target/*/release/bcachefs -t $(DESTDIR)$(ROOT_SBINDIR)
 	$(INSTALL) -m0644 -D bcachefs.8    -t $(DESTDIR)$(PREFIX)/share/man/man8/
 	$(INSTALL) -m0755 -D initramfs/script $(DESTDIR)$(INITRAMFS_SCRIPT)
