--- a/Makefile
+++ b/Makefile
@@ -57,7 +57,7 @@
 # CARGO_PROFILE=debug
 CARGO_MANIFEST=--manifest-path rust-src/Cargo.toml
 
-CARGO_BUILD_ARGS=--$(CARGO_PROFILE)
+CARGO_BUILD_ARGS=--$(CARGO_PROFILE) --target $(RUST_TARGET)
 CARGO_BUILD=$(CARGO) build $(CARGO_BUILD_ARGS) $(CARGO_MANIFEST)
 
 CARGO_CLEAN=$(CARGO) clean $(CARGO_CLEAN_ARGS) $(CARGO_MANIFEST)
@@ -190,6 +190,9 @@
 RUST_SRCS:=$(shell find rust-src/src rust-src/bch_bindgen/src -type f -iname '*.rs')
 rust-src/target/release/libbcachefs_rust.a: $(RUST_SRCS)
 	$(CARGO_BUILD)
+	if [ -f rust-src/target/*/release/libbcachefs_rust.a ]; then \
+		cp rust-src/target/*/release/libbcachefs_rust.a $@; \
+	fi
 
 tests/test_helper: $(filter ./tests/%.o, $(OBJS))
 	@echo "    [LD]     $@"
