diff --git a/utils/logprof.conf b/utils/logprof.conf
index a3d7a31..296dd8b 100644
--- a/utils/logprof.conf
+++ b/utils/logprof.conf
@@ -14,7 +14,7 @@
   inactive_profiledir = /usr/share/apparmor/extra-profiles 
   logfiles = /var/log/audit/audit.log /var/log/syslog /var/log/messages
 
-  parser = /sbin/apparmor_parser /sbin/subdomain_parser
+  parser = /usr/bin/apparmor_parser /usr/bin/subdomain_parser
   ldd = /usr/bin/ldd
   logger = /bin/logger /usr/bin/logger
 
@@ -36,10 +36,10 @@
   custom_includes =
 
 
-[repository]
-  distro         = ubuntu-intrepid
-  url            = http://apparmor.test.opensuse.org/backend/api
-  preferred_user = ubuntu
+#[repository]
+#  distro         = ubuntu-intrepid
+#  url            = http://apparmor.test.opensuse.org/backend/api
+#  preferred_user = ubuntu
 
 [qualifiers]
   # things will be painfully broken if bash has a profile

