modify logprof.conf for Void Linux
--- a/utils/logprof.conf
+++ b/utils/logprof.conf
@@ -12,10 +12,10 @@
 [settings]
   profiledir = /etc/apparmor.d /etc/subdomain.d
   inactive_profiledir = /usr/share/apparmor/extra-profiles
-  logfiles = /var/log/audit/audit.log /var/log/syslog /var/log/messages
+  logfiles = /var/log/audit/audit.log /var/log/socklog/kernel/current /var/log/syslog /var/log/messages
 
-  parser = /sbin/apparmor_parser /sbin/subdomain_parser
-  logger = /bin/logger /usr/bin/logger
+  parser = /usr/bin/apparmor_parser /usr/bin/subdomain_parser
+  logger = /usr/bin/logger
 
   # customize how file ownership permissions are presented
