From d295a72f7bf03839f13067e3d9689bb6d33e8f65 Mon Sep 17 00:00:00 2001
From: Komeil Parseh <komeilparseh@disroot.org>
Date: Wed, 16 Jul 2025 15:51:30 +0330
Subject: [PATCH] Handle missing XDG_CURRENT_DESKTOP env var

---
 persepolis/scripts/useful_tools.py | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/persepolis/scripts/useful_tools.py b/persepolis/scripts/useful_tools.py
index 898ec0c..04a152c 100644
--- a/persepolis/scripts/useful_tools.py
+++ b/persepolis/scripts/useful_tools.py
@@ -86,8 +86,15 @@ def osAndDesktopEnvironment():
     desktop_env = None
     if os_type in OS.UNIX_LIKE:
         # find desktop environment('KDE', 'GNOME', ...)
-        desktop_env = os.environ.get('XDG_CURRENT_DESKTOP').lower()
-
+        xdg_desktop = os.environ.get("XDG_CURRENT_DESKTOP")
+        if xdg_desktop is not None:
+            desktop_env = xdg_desktop.lower()
+        else:
+            desktop_env = "unknown"
+            if logger_availability:
+                logger.sendToLog(
+                    "XDG_CURRENT_DESKTOP environment variable not found.", "WARNING"
+                )
     return os_type, desktop_env
 
 
-- 
2.50.1

