--- a/test/test_libwacom.py	2026-07-20 05:12:17.000000000 +0200
+++ b/test/test_libwacom.py	2026-09-02 12:46:54.390895889 +0200
@@ -583,6 +583,7 @@
 @pytest.mark.parametrize(
     "fallback", (WacomDatabase.Fallback.NONE, WacomDatabase.Fallback.GENERIC)
 )
+@pytest.mark.skip(reason="can't create devices in build env")
 def test_new_from_path_known_device(db, fallback):
     name = "Wacom Intuos4 WL"
     vid = 0x056A
@@ -602,6 +603,7 @@
 @pytest.mark.parametrize(
     "fallback", (WacomDatabase.Fallback.NONE, WacomDatabase.Fallback.GENERIC)
 )
+@pytest.mark.skip(reason="can't create devices in build env")
 def test_new_from_path_unknown_device(db, fallback, bustype):
     bus = {
         WacomBustype.UNKNOWN: 0,
