--- CMakeLists.txt
+++ CMakeLists.txt
@@ -320,7 +320,7 @@ if(FLATBUFFERS_INSTALL)
     install(
       TARGETS flatc EXPORT FlatcTargets
       RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
-      CONFIGURATIONS Release
+      CONFIGURATIONS Release RelWithDebInfo
     )
 
     install(
@@ -328,7 +328,7 @@ if(FLATBUFFERS_INSTALL)
       FILE FlatcTargets.cmake
       NAMESPACE flatbuffers::
       DESTINATION ${FB_CMAKE_DIR}
-      CONFIGURATIONS Release
+      CONFIGURATIONS Release RelWithDebInfo
     )
   endif()
 
