From 1f0faa7f7dc6d13b49ed9b91110b98e1a55a9ba3 Mon Sep 17 00:00:00 2001
From: "Andrew J. Hesford" <ajh@sideband.org>
Date: Fri, 16 Oct 2020 12:27:06 -0400
Subject: [PATCH] tox.ini: support Python 3.9

---
 tox.ini | 14 +++++++++++++-
 1 file changed, 13 insertions(+), 1 deletion(-)

diff --git a/tox.ini b/tox.ini
index 45893c0..5df9174 100644
--- tox.ini
+++ tox.ini
@@ -5,7 +5,7 @@
 
 [tox]
 minversion = 2.0
-envlist = py27,py3{4,5,6,7,8},py{27,38}-flake8,pypy
+envlist = py27,py3{4,5,6,7,8,9},py{27,38,39}-flake8,pypy
 
 [testenv]
 basepython =
@@ -15,6 +15,7 @@ basepython =
     py36: {env:TOXPYTHON:python3.6}
     py37: {env:TOXPYTHON:python3.7}
     py38: {env:TOXPYTHON:python3.8}
+    py39: {env:TOXPYTHON:python3.9}
     pypy: {env:TOXPYTHON:pypy}
 commands =
     {envpython} --version
@@ -63,6 +64,10 @@ commands =
 commands =
     {[tests]commands}
 
+[testenv:py39]
+commands =
+    {[tests]commands}
+
 [testenv:py27-flake8]
 deps =
     flake8
@@ -77,6 +82,13 @@ commands =
     {[testenv]commands}
     flake8 .
 
+[testenv:py39-flake8]
+deps =
+    flake8
+commands =
+    {[testenv]commands}
+    flake8 .
+
 [testenv:pypy]
 commands =
     {[tests]commands}
