--- Config.mk.orig	2014-04-29 20:54:43.117010822 +0100
+++ Config.mk	2014-04-29 20:58:23.100055325 +0100
@@ -188,6 +188,8 @@
 CFLAGS += -std=gnu99
 
 CFLAGS += -Wall -Wstrict-prototypes
+
+CFLAGS += -Wno-error=maybe-uninitialized
 
 # Clang complains about macros that expand to 'if ( ( foo == bar ) ) ...'
 # and is over-zealous with the printf format lint
--- tools/blktap2/drivers/block-qcow.c.orig	2014-03-10 10:43:57.000000000 +0000
+++ tools/blktap2/drivers/block-qcow.c	2014-04-30 21:10:26.270699187 +0100
@@ -427,6 +427,7 @@
 
 		if (posix_memalign((void **)&tmp_ptr, 4096, 4096) != 0) {
 			DPRINTF("ERROR allocating memory for L1 table\n");
+			return -1;
 		}
 		memcpy(tmp_ptr, l1_ptr, 4096);
 
@@ -600,6 +601,7 @@
 		
 		if (posix_memalign((void **)&tmp_ptr2, 4096, 4096) != 0) {
 			DPRINTF("ERROR allocating memory for L1 table\n");
+			return -1;
 		}
 		memcpy(tmp_ptr2, l2_ptr, 4096);
 		lseek(s->fd, l2_offset + (l2_sector << 12), SEEK_SET);
