From 18c163276f9b1f7f94e17a4f51f37a58b1ee4052 Mon Sep 17 00:00:00 2001
From: John Thomson <git@johnthomson.fastmail.com.au>
Date: Thu, 25 Jan 2018 17:06:56 +1000
Subject: [PATCH] Fix ocaml 4.06.0 build, use unsafe strings

---
 tools/ocaml/common.make | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tools/ocaml/common.make b/tools/ocaml/common.make
index d5478f6..21ac33c 100644
--- tools/ocaml/common.make
+++ tools/ocaml/common.make
@@ -12,8 +12,8 @@ OCAMLFIND ?= ocamlfind
 CFLAGS += -fPIC -Werror -I$(shell ocamlc -where)
 
 OCAMLOPTFLAG_G := $(shell $(OCAMLOPT) -h 2>&1 | sed -n 's/^  *\(-g\) .*/\1/p')
-OCAMLOPTFLAGS = $(OCAMLOPTFLAG_G) -ccopt "$(LDFLAGS)" -dtypes $(OCAMLINCLUDE) -cc $(CC) -w F -warn-error F
-OCAMLCFLAGS += -g $(OCAMLINCLUDE) -w F -warn-error F
+OCAMLOPTFLAGS = $(OCAMLOPTFLAG_G) -ccopt "$(LDFLAGS)" -dtypes $(OCAMLINCLUDE) -cc $(CC) -w F -warn-error F -unsafe-string
+OCAMLCFLAGS += -g $(OCAMLINCLUDE) -w F -warn-error F -unsafe-string
 
 VERSION := 4.1
 
