--- tools/qemu-xen/util/memfd.c.orig	2018-09-22 17:42:37.087624225 +0200
+++ tools/qemu-xen/util/memfd.c	2018-09-22 17:42:51.368623656 +0200
@@ -37,7 +37,7 @@
 #include <sys/syscall.h>
 #include <asm/unistd.h>
 
-static int memfd_create(const char *name, unsigned int flags)
+int memfd_create(const char *name, unsigned int flags)
 {
 #ifdef __NR_memfd_create
     return syscall(__NR_memfd_create, name, flags);
