Index: CMakeLists.txt
===================================================================
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -5,16 +5,8 @@ CMAKE_MINIMUM_REQUIRED(VERSION 2.8)
 
 project(xstarter)
 
-if(NOT CMAKE_BUILD_TYPE)
-  set(CMAKE_BUILD_TYPE Debug CACHE STRING "Choose the type of build (Debug or Release)" FORCE)
-endif()
-
 set(PROJECT_VERSION "0.8.0")
-if (CMAKE_BUILD_TYPE STREQUAL Debug)
-  set(CMAKE_C_FLAGS "-g -Wall -pedantic")
-else()
-  set(CMAKE_C_FLAGS "-Wall -pedantic -O3")
-endif()
+set(CMAKE_C_FLAGS "-Wall -pedantic ${CMAKE_C_FLAGS}")
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY "${CMAKE_SOURCE_DIR}/bin")
 
 set (CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake")
