--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2,18 +2,11 @@
 
 project(xstarter)
 
-if (NOT CMAKE_BUILD_TYPE)
-    set(CMAKE_BUILD_TYPE Debug CACHE STRING "Choose the type of build (Debug or Release)" FORCE)
-endif()
-
 set (CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/../cmake")
 
 set(PROJECT_VERSION "0.8.2")
-if (CMAKE_BUILD_TYPE STREQUAL Debug)
-    set(CMAKE_C_FLAGS "-g -Wall -pedantic")
-else()
-    set(CMAKE_C_FLAGS "-Wall -pedantic -O3")
-endif()
+
+set(CMAKE_C_FLAGS "-fcommon -Wall -pedantic ${CMAKE_C_FLAGS}")
 
 # cpack
 set(CPACK_PACKAGE_CONTACT "Maciej Lechowski")
@@ -42,7 +35,7 @@
 message(STATUS "Cmake build type: " ${CMAKE_BUILD_TYPE})
 message(STATUS "Curses include directory: " ${CURSES_INCLUDE_DIR})
 
-pkg_check_modules(CURSES REQUIRED ncurses)
+pkg_check_modules(CURSES REQUIRED ncursesw)
 
 add_library(lib-inih STATIC
     external/inih/ini.c
