diff --git a/sftp/CMakeLists.txt b/sftp/CMakeLists.txt
index 5e8d2cb1..aa88f054 100644
--- sftp/CMakeLists.txt
+++ sftp/CMakeLists.txt
@@ -9,6 +9,9 @@ target_link_libraries(kio_sftp
    KF5::I18n
    Qt5::Network
    ${LIBSSH_LIBRARIES})
+if(TARGET ssh) # libssh 0.9
+    target_link_libraries(kio_sftp ssh)
+endif()
 set_target_properties(kio_sftp PROPERTIES OUTPUT_NAME "sftp")
 
 install(TARGETS kio_sftp DESTINATION ${PLUGIN_INSTALL_DIR}/kf5/kio)
