--- a/Makefile
+++ b/Makefile
@@ -54,8 +54,8 @@
 uninstall: uninstall-headers uninstall-docs
 
 install-headers:
-	mkdir -p $(PREFIX)/include/png++
-	cp $(headers) $(PREFIX)/include/png++
+	mkdir -p $(DESTDIR)/$(PREFIX)/include/png++
+	cp $(headers) $(DESTDIR)/$(PREFIX)/include/png++
 
 uninstall-headers:
 	rm -rf $(PREFIX)/include/png++
@@ -103,7 +103,7 @@
 
 install-docs:
 	if [ -d doc ]; then \
-		dir=$(PREFIX)/share/doc/$(dist_dir); \
+		dir=$(DESTDIR)/$(PREFIX)/share/doc/$(dist_dir); \
 		rm -rf $$dir; \
 		mkdir -p $$dir \
 		&& cp -r $(doc_files) doc/html $$dir; \
