--- a/src/net_uuid.c
+++ b/src/net_uuid.c
@@ -28,6 +28,7 @@
 #include <stdlib.h>
 #include <string.h>
 #include <fcntl.h>
+#include <unistd.h>
 
 #if defined(HAVE_INTTYPES_H)
 #include <inttypes.h>
--- a/src/netlib.c
+++ b/src/netlib.c
@@ -779,6 +779,7 @@ random_ip_address(struct addrinfo *res,
 }
 
 #if defined(HAVE_SENDFILE)
+#include <sys/sendfile.h>
 int netperf_sendfile(SOCKET send_socket, struct ring_elt *send_ring) {
 
   int  len;
@@ -2309,6 +2310,7 @@ bind_to_specific_processor(int processor
      value will not tell you if you are bound vs unbound. */
   bindprocessor(BINDPROCESS,getpid(),(cpu_t)mapped_affinity);
 #elif HAVE_SCHED_SETAFFINITY
+#define _GNU_SOURCE
 #include <sched.h>
   /* in theory this should cover systems with more CPUs than bits in a
      long, without having to specify __USE_GNU.  we "cheat" by taking
