From 015f975586317cf851319e9616bdccfafdd81741 Mon Sep 17 00:00:00 2001
From: Jouke Witteveen <j.witteveen@gmail.com>
Date: Wed, 05 Mar 2014 22:00:56 +0000
Subject: Fix adding of interfaces

`shift 3` doesn't do anything if there are only 2 parameters.
---
diff --git a/src/lib/network b/src/lib/network
index a875101..0250c3d 100755
--- src/lib/network
+++ src/lib/network
@@ -17,8 +17,7 @@ is_interface() {
 # $4...: additional arguments
 interface_add() {
     local type="$1" name="$2" link="$3"
-    shift 3
-    ip link add ${link:+link "$link"} name "$name" type "$type" "$@" || return
+    do_debug ip link add ${link:+link "$link"} name "$name" type "$type" "${@:4}" || return
     if [[ -x "$PROFILE_DIR/interfaces/$name" ]]; then
         source "$PROFILE_DIR/interfaces/$name"
     fi
--
cgit v0.9.2-18-g5137
