upstreamed in cjcliffe/CubicSDR#950

--- a/src/visual/WaterfallCanvas.cpp
+++ b/src/visual/WaterfallCanvas.cpp
@@ -482,7 +482,7 @@
     }
 
 }
-void WaterfallCanvas::OnIdle(wxIdleEvent & /* event */) {
+void WaterfallCanvas::OnIdle(wxIdleEvent & event) {
     processInputQueue();
     Refresh();
     event.RequestMore();
--- a/src/visual/TuningCanvas.cpp
+++ b/src/visual/TuningCanvas.cpp
@@ -269,7 +269,7 @@
     }
 }
 
-void TuningCanvas::OnIdle(wxIdleEvent & /* event */) {
+void TuningCanvas::OnIdle(wxIdleEvent & event) {
     if (mouseTracker.mouseDown()) {
         if (downState != TUNING_HOVER_NONE) {
             dragAccum += 5.0*mouseTracker.getOriginDeltaMouseX();
