--- src/openvpn/ssl_openssl.c	2019-03-05 18:39:14.924080534 +0100
+++ src/openvpn/ssl_openssl.c	2019-03-05 18:49:33.737131153 +0100
@@ -459,7 +459,7 @@
         return;
     }
 
-#if (OPENSSL_VERSION_NUMBER < 0x1010100fL)
+#if (OPENSSL_VERSION_NUMBER < 0x1010100fL || defined(LIBRESSL_VERSION_NUMBER))
         crypto_msg(M_WARN, "Not compiled with OpenSSL 1.1.1 or higher. "
                        "Ignoring TLS 1.3 only tls-ciphersuites '%s' setting.",
                         ciphers);
@@ -1846,7 +1846,7 @@
         crypto_msg(M_FATAL, "Cannot create SSL_CTX object");
     }
 
-#if (OPENSSL_VERSION_NUMBER >= 0x1010100fL)
+#if (OPENSSL_VERSION_NUMBER >= 0x1010100fL && !defined(LIBRESSL_VERSION_NUMBER))
     if (tls13)
     {
         SSL_CTX_set_min_proto_version(tls_ctx.ctx, TLS1_3_VERSION);
