--- matlib/mat.hh.orig	2018-01-17 09:37:50.871239713 +0100
+++ matlib/mat.hh	2018-01-17 09:37:59.927203590 +0100
@@ -91,7 +91,7 @@ template<typename T> void mmul(T *c, con
                 memcpy(c, r, d1 * d3 * sizeof(c[0]));
 }
 
-template<typename T> void mdet(const T *c, unsigned int d)
+template<typename T> T mdet(const T *c, unsigned int d)
 {
         T *c2;
         unsigned int i, j, k, l;
