--- a/lib/printf_test.c
+++ b/lib/printf_test.c
@@ -22,6 +22,7 @@ static int
 t_simple(void)
 {
   char buf[256];
+  const char *errormsg;
   memset(buf, 0xa5, 256);
 
   BSPRINTF(0, "", buf, "", NULL);
@@ -32,11 +33,12 @@ t_simple(void)
   BSPRINTF(1, "@", buf, "@", 64);
   BSPRINTF(1, "\xff", buf, "%c", 0xff);
 
-  errno = 5;
-  BSPRINTF(18, "Input/output error", buf, "%m");
+  errormsg = strerror(EIO);
+  errno = EIO;
+  BSPRINTF(strlen(errormsg), errormsg, buf, "%m");
   errno = 0;
 
-  BSPRINTF(18, "Input/output error", buf, "%M", 5);
+  BSPRINTF(strlen(errormsg), errormsg, buf, "%M", EIO);
 
   BSPRINTF(11, "TeSt%StRiNg", buf, "%s", "TeSt%StRiNg");
