--- progs/capsh.c.orig	2009-05-10 19:57:26.897493561 +0200
+++ progs/capsh.c	2009-05-10 19:57:40.092519179 +0200
@@ -9,6 +9,7 @@
  * capability manipulations work as expected (or not).
  */
 
+#include <sys/wait.h>
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
@@ -16,7 +17,6 @@
 #include <sys/capability.h>
 #include <unistd.h>
 #include <errno.h>
-#include <sys/wait.h>
 
 /* prctl based API for altering character of current process */
 #define PR_GET_KEEPCAPS    7
