--- lib/ext2fs/swapfs.c	2018-01-02 05:52:58.000000000 +0100
+++ lib/ext2fs/swapfs.c	2018-01-08 21:54:26.223011078 +0100
@@ -19,6 +19,7 @@
 
+#include "ext2fsP.h"
 #include "ext2_fs.h"
 #include "ext2fs.h"
 #include <ext2fs/ext2_ext_attr.h>
 
 #ifdef WORDS_BIGENDIAN
@@ -359,7 +359,7 @@
 	if (inode_includes(inode_size, i_projid))
                 t->i_projid = ext2fs_swab16(f->i_projid);
 	/* catch new static fields added after i_projid */
-	EXT2FS_BUILD_BUG_ON(sizeof(ext2_inode_large) != 160);
+	EXT2FS_BUILD_BUG_ON(sizeof(struct ext2_inode_large) != 160);
 
 	i = sizeof(struct ext2_inode) + extra_isize + sizeof(__u32);
 	if (bufsize < (int) i)
