From c50802b213c325cfa957d66955f6de96908710d9 Mon Sep 17 00:00:00 2001
From: Rico Tzschichholz <ricotz@ubuntu.com>
Date: Sat, 6 Nov 2021 19:46:07 +0100
Subject: [PATCH] Fix accessibility conflict of constant and its value

---
 src/simple-scan.vala | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/simple-scan.vala b/src/simple-scan.vala
index c15a5418..71f32de9 100644
--- a/src/simple-scan.vala
+++ b/src/simple-scan.vala
@@ -14,7 +14,7 @@ public class SimpleScan : Gtk.Application
     static bool show_version;
     static bool debug_enabled;
     static string? fix_pdf_filename = null;
-    public const OptionEntry[] options =
+    const OptionEntry[] options =
     {
         { "version", 'v', 0, OptionArg.NONE, ref show_version,
           /* Help string for command line --version flag */
-- 
GitLab

