--- OpenSSL/rand/rand.c.orig	2014-07-21 08:43:25.068650199 +0200
+++ OpenSSL/rand/rand.c	2014-07-21 08:45:10.724488166 +0200
@@ -245,7 +245,9 @@ static PyMethodDef rand_methods[] = {
 #ifdef MS_WINDOWS
     { "screen",    (PyCFunction)rand_screen,       METH_VARARGS, rand_screen_doc },
 #endif
+#ifdef HAVE_OPENSSL_RAND_EGD
     { "egd",       (PyCFunction)rand_egd,          METH_VARARGS, rand_egd_doc },
+#endif
     { "cleanup",   (PyCFunction)rand_cleanup,      METH_VARARGS, rand_cleanup_doc },
     { "load_file", (PyCFunction)rand_load_file,    METH_VARARGS, rand_load_file_doc },
     { "write_file",(PyCFunction)rand_write_file,   METH_VARARGS, rand_write_file_doc },
--- OpenSSL/ssl/context.c.orig	2014-07-21 08:48:55.940289949 +0200
+++ OpenSSL/ssl/context.c	2014-07-21 08:52:02.068848946 +0200
@@ -1109,9 +1109,11 @@ ssl_Context_init(ssl_ContextObj *self, i
     SSL_METHOD *method;
 
     switch (i_method) {
+#if OPENSSL_VERSION_NUMBER < 0x20000000L
         case ssl_SSLv2_METHOD:
             method = SSLv2_method();
             break;
+#endif
         case ssl_SSLv23_METHOD:
             method = SSLv23_method();
             break;
