diff --git a/src/session-child.c b/src/session-child.c
--- a/src/session-child.c
+++ b/src/session-child.c
@@ -397,7 +397,7 @@ session_child_run (int argc, char **argv)
         else
         {
             /* Set POSIX variables */
-            pam_putenv (pam_handle, "PATH=/usr/local/bin:/usr/bin:/bin");
+            pam_putenv (pam_handle, "PATH=/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/sbin:/sbin:/bin");
             g_autofree gchar* user_env = g_strdup_printf ("USER=%s", username);
             pam_putenv (pam_handle, user_env);
             g_autofree gchar* logname_env = g_strdup_printf ("LOGNAME=%s", username);
