--- data/lightdm.conf.orig	2012-10-05 09:43:48.307845764 +0200
+++ data/lightdm.conf	2012-10-05 09:44:57.956957476 +0200
@@ -19,9 +19,9 @@
 #start-default-seat=true
 #greeter-user=lightdm
 #minimum-display-number=0
-#minimum-vt=7
+minimum-vt=1
 #lock-memory=true
-#user-authority-in-system-dir=false
+user-authority-in-system-dir=true
 #guest-account-script=guest-account
 #log-directory=/var/log/lightdm
 #run-directory=/var/run/lightdm
@@ -69,7 +69,7 @@
 #xdmcp-manager=
 #xdmcp-port=177
 #xdmcp-key=
-#greeter-session=example-gtk-gnome
+greeter-session=lightdm-gtk-greeter
 #greeter-hide-users=false
 #greeter-allow-guest=true
 #greeter-show-manual-login=false
@@ -77,7 +77,7 @@
 #user-session=default
 #allow-guest=true
 #guest-session=UNIMPLEMENTED
-#session-wrapper=lightdm-session
+#session-wrapper=lightdm-session
 #display-setup-script=
 #greeter-setup-script=
 #session-setup-script=
