--- data/lightdm.conf 2014-03-12 20:56:34.000000000 +0100
+++ data/lightdm.conf 2014-03-14 15:48:32.089948842 +0100
@@ -19,12 +19,12 @@
 #start-default-seat=true
 #greeter-user=lightdm
 #minimum-display-number=0
-#minimum-vt=7
+#minimum-vt=1
 #lock-memory=true
 #user-authority-in-system-dir=false
 #guest-account-script=guest-account
 #log-directory=/var/log/lightdm
-#run-directory=/var/run/lightdm
+run-directory=/run/lightdm
 #cache-directory=/var/cache/lightdm
 #sessions-directory=/usr/share/lightdm/sessions:/usr/share/xsessions
 #remote-sessions-directory=/usr/share/lightdm/remote-sessions
@@ -93,7 +93,7 @@
 #user-session=default
 #allow-guest=true
 #guest-session=
-#session-wrapper=lightdm-session
+session-wrapper=/etc/lightdm/Xsession
 #greeter-wrapper=
 #guest-wrapper=
 #display-setup-script=
@@ -107,6 +107,7 @@
 #autologin-in-background=false
 #autologin-session=UNIMPLEMENTED
 #exit-on-failure=false
+pam-service=lightdm-autologin

 #
 # Seat configuration
--- data/users.conf 2013-09-02 17:17:41.167211771 +0200
+++ data/users.conf 2013-09-02 17:21:27.955545717 +0200
@@ -9,6 +9,6 @@
 # hidden-shells = Shells that indicate a user cannot login
 #
 [UserAccounts]
-minimum-uid=500
+minimum-uid=1000
 hidden-users=nobody nobody4 noaccess
-hidden-shells=/bin/false /usr/sbin/nologin
+hidden-shells=/bin/false /sbin/nologin
