--- data/lightdm.conf	2013-09-02 17:17:41.167211771 +0200
+++ data/lightdm.conf	2013-09-02 17:20:41.278613565 +0200
@@ -18,14 +18,14 @@
 #
 [LightDM]
 #start-default-seat=true
-#greeter-user=lightdm
+greeter-user=lightdm
 #minimum-display-number=0
-#minimum-vt=7
+minimum-vt=1
 #lock-memory=true
 #user-authority-in-system-dir=false
 #guest-account-script=guest-account
-#log-directory=/var/log/lightdm
-#run-directory=/var/run/lightdm
+log-directory=/var/log/lightdm
+run-directory=/run/lightdm
 #cache-directory=/var/cache/lightdm
 #sessions-directory=/usr/share/lightdm/sessions:/usr/share/xsessions
 #remote-sessions-directory=/usr/share/lightdm/remote-sessions
@@ -81,7 +81,7 @@
 #xdmcp-key=
 #unity-compositor-command=unity-system-compositor
 #unity-compositor-timeout=60
-#greeter-session=example-gtk-gnome
+greeter-session=lightdm-gtk-greeter
 #greeter-hide-users=false
 #greeter-allow-guest=true
 #greeter-show-manual-login=false
@@ -89,7 +89,7 @@
 #user-session=default
 #allow-guest=true
 #guest-session=UNIMPLEMENTED
-#session-wrapper=lightdm-session
+session-wrapper=/etc/lightdm/Xsession
 #greeter-wrapper=
 #display-setup-script=
 #greeter-setup-script=
@@ -100,6 +100,7 @@
 #autologin-user-timeout=0
 #autologin-in-background=false
 #autologin-session=UNIMPLEMENTED
+pam-service=lightdm-autologin
 #exit-on-failure=false
 
 #
--- data/users.conf	2013-09-02 17:17:41.167211771 +0200
+++ data/users.conf	2013-09-02 17:21:27.955545717 +0200
@@ -9,6 +9,6 @@
 # hidden-shells = Shells that indicate a user cannot login
 #
 [UserAccounts]
-minimum-uid=500
+minimum-uid=1000
 hidden-users=nobody nobody4 noaccess
-hidden-shells=/bin/false /usr/sbin/nologin
+hidden-shells=/bin/false /sbin/nologin
