--- themes/gnome/gtkrc.orig	2011-02-16 15:30:48.425744100 +0100
+++ themes/gnome/gtkrc	2011-02-16 15:32:41.960886724 +0100
@@ -1,7 +1,7 @@
-gtk_theme_name = "HumanLogin"
-gtk_icon_theme_name = "LoginIcons"
+gtk_theme_name = "Clearlooks"
+gtk_icon_theme_name = "gnome"
 gtk_cursor_theme_name = "DMZ-White"
-gtk_font_name = "UbuntuBeta 10"
+gtk_font_name = "Arial 10"
 # Not working, but set to 96*1024 (=96dpi)
 gtk_xft_dpi = 98304
 gtk_xft_hinting = 1
