--- eiskaltdcpp-qt/src/main.cpp	2015-04-20 02:57:16.000000000 +0500
+++ eiskaltdcpp-qt/src/main.cpp.new	2018-01-07 08:47:19.501488838 +0500
@@ -72,7 +72,7 @@
 #if !defined(Q_WS_WIN)
 #include <unistd.h>
 #include <signal.h>
-#if !defined (__HAIKU__)
+#if !defined (__HAIKU__) && defined (__GLIBC__)
 #include <execinfo.h>
 
 #ifdef ENABLE_STACKTRACE
@@ -127,7 +127,7 @@
         return 0;
     }
 
-#if !defined (Q_WS_WIN) && !defined (__HAIKU__)
+#if !defined (Q_WS_WIN) && !defined (__HAIKU__) && defined (__GLIBC__)
     installHandlers();
 #endif
 
