https://github.com/eiskaltdcpp/eiskaltdcpp/commit/fc9bc0c2

--- dcpp/Util.h
+++ dcpp/Util.h.new
@@ -243,7 +243,7 @@ public:
     static int64_t toInt64(const string& aString) {
 #ifdef _WIN32
         return _atoi64(aString.c_str());
-#elif defined(__HAIKU__)
+#elif defined(__HAIKU__) || !defined(__GLIBC__)
         return strtoll(aString.c_str(), (char **)NULL, 10);
 #else
         return strtoq(aString.c_str(), (char **)NULL, 10);
--- eiskaltdcpp-qt/src/main.cpp
+++ eiskaltdcpp-qt/src/main.cpp.new
@@ -86,7 +86,7 @@ void parseCmdLine(const QStringList &);
 #if !defined(Q_OS_WIN)
 #include <unistd.h>
 #include <signal.h>
-#if !defined (Q_OS_HAIKU)
+#if !defined (Q_OS_HAIKU) && defined (__GLIBC__)
 #include <execinfo.h>
 
 #ifdef ENABLE_STACKTRACE
@@ -141,7 +141,7 @@ int main(int argc, char *argv[])
         return 0;
     }
 
-#if !defined (Q_OS_WIN) && !defined (Q_OS_HAIKU)
+#if !defined (Q_OS_WIN) && !defined (Q_OS_HAIKU) && defined (__GLIBC__)
     installHandlers();
 #endif
 
