From aa43f762540d25acbdfbd218e5efde4c20851ce1 Mon Sep 17 00:00:00 2001
From: "lauren n. liberda" <lauren@selfisekai.rocks>
Date: Sat, 8 Jun 2024 02:39:04 +0200
Subject: [PATCH] zig 0.13: require zig 0.13 now

---
 src/main.zig | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/main.zig b/src/main.zig
index b477ad7..6ba77e2 100644
--- a/src/main.zig
+++ b/src/main.zig
@@ -12,8 +12,8 @@ const utildbg = @import("./util/utildbg.zig");
 pub fn main() !void {
     comptime {
         const v = builtin.zig_version;
-        if (v.major != 0 or v.minor != 12)
-            @panic("Must be built against Zig 0.12.x");
+        if (v.major != 0 or v.minor != 13)
+            @panic("Must be built against Zig 0.13.x");
     }
 
     const allocator = std.heap.c_allocator;
