From e370ef83141ac5caa41b65c665e86233020ff125 Mon Sep 17 00:00:00 2001
From: "lauren n. liberda" <lauren@selfisekai.rocks>
Date: Sat, 8 Jun 2024 02:51:12 +0200
Subject: [PATCH] zig 0.13: adapt to new Dir.writeFile API (aka Dir.writeFile2)

---
 src/TileCache.zig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/TileCache.zig b/src/TileCache.zig
index 0b1fa63..17e662c 100644
--- a/src/TileCache.zig
+++ b/src/TileCache.zig
@@ -425,7 +425,7 @@ fn download_loop_transfer_complete(tile_cache: *@This(), msg: *curl.CURLMsg) !vo
                     // Save to FS
                     if (is_valid_png_data) {
                         const path = try png_path(tile_cache.allocator, p.get(p.pref.tile_cache_url).t.?, coords);
-                        try cache_dir.writeFile(path, datum_array);
+                        try cache_dir.writeFile(.{ .sub_path = path, .data = datum_array });
                     }
                 }
                 if (tile_cache.transfer_map.get(coords).?.load_to_texture) {
