From 1c066f5ea010994c790113bf6cf4a520227a9e06 Mon Sep 17 00:00:00 2001
From: Thomas Deutschmann <whissi@gentoo.org>
Date: Thu, 30 Apr 2020 19:59:26 +0200
Subject: [PATCH 2/5] cmake: Remove -Werror

Signed-off-by: Thomas Deutschmann <whissi@gentoo.org>
---
 cmake/maintainer.cmake | 6 ------
 1 file changed, 6 deletions(-)

diff --git cmake/maintainer.cmake cmake/maintainer.cmake
index 6aa3ac866..b2f63bdaa 100644
--- cmake/maintainer.cmake
+++ cmake/maintainer.cmake
@@ -157,12 +157,6 @@ IF(MY_COMPILER_IS_CLANG)
   # -Wzero-as-null-pointer-constant
 ENDIF()
 
-# Turn on Werror (warning => error) when using maintainer mode.
-IF(MYSQL_MAINTAINER_MODE)
-  STRING_APPEND(MY_C_WARNING_FLAGS   " -Werror")
-  STRING_APPEND(MY_CXX_WARNING_FLAGS " -Werror")
-ENDIF()
-
 # Set warning flags for gcc/g++/clang/clang++
 IF(MY_COMPILER_IS_GNU_OR_CLANG)
   STRING_APPEND(CMAKE_C_FLAGS   " ${MY_C_WARNING_FLAGS}")
-- 
2.27.0

