--- a/probe-rs-tools/src/bin/probe-rs/util/cargo.rs
+++ b/probe-rs-tools/src/bin/probe-rs/util/cargo.rs
@@ -164,45 +164,6 @@
     }
 
     #[test]
-    fn get_binary_artifact_with_cargo_config() {
-        let work_dir = test_project_dir("binary_cargo_config");
-
-        let mut expected_path = work_dir.join("target");
-        expected_path.push("thumbv7m-none-eabi");
-        expected_path.push("debug");
-        expected_path.push("binary_cargo_config");
-
-        let args = [];
-
-        let binary_artifact =
-            build_artifact(&work_dir, &args).expect("Failed to read artifact path.");
-
-        assert_eq!(
-            binary_artifact.path(),
-            dunce::canonicalize(expected_path).expect("Failed to canonicalize path")
-        );
-    }
-
-    #[test]
-    fn get_binary_artifact_with_cargo_config_toml() {
-        let work_dir = test_project_dir("binary_cargo_config_toml");
-        let mut expected_path = work_dir.join("target");
-        expected_path.push("thumbv7m-none-eabi");
-        expected_path.push("debug");
-        expected_path.push("binary_cargo_config_toml");
-
-        let args = [];
-
-        let binary_artifact =
-            build_artifact(&work_dir, &args).expect("Failed to read artifact path.");
-
-        assert_eq!(
-            binary_artifact.path(),
-            dunce::canonicalize(expected_path).expect("Failed to canonicalize path")
-        );
-    }
-
-    #[test]
     fn get_library_artifact_fails() {
         let work_dir = test_project_dir("library_project");
 
