--- a/CMakeLists.txt	2020-12-20 19:39:26.000000000 +0100
+++ b/CMakeLists.txt	2021-08-13 09:50:54.728623869 +0200
@@ -100,12 +100,12 @@
 ### On Windows cmake fails to find the opencl.lib, as we only use Khronos Headers which doesn't come with the library
 ### So we bypass that by not making it required and double check for the header after.
 ### On linux typically the library always comes with the package.
-if (WIN32)
-	FIND_PACKAGE(OpenCL)
-else()
-	FIND_PACKAGE(OpenCL REQUIRED)
-endif()
-if (WIN32)
+#if (WIN32)
+#	FIND_PACKAGE(OpenCL)
+#else()
+#	FIND_PACKAGE(OpenCL REQUIRED)
+#endif()
+#if (WIN32)
 	set(OpenCL_FOUND FALSE)
 	foreach(_DIR ${OpenCL_INCLUDE_DIRS})
 		if(EXISTS "${_DIR}/CL/cl.h")
@@ -119,9 +119,17 @@
 		message(STATUS "On Windows: Errors about 'missing: OpenCL_LIBRARY' can be ignored, that is not needed.")
 		include_directories(${OpenCL_INCLUDE_DIRS})
 	endif()
-endif()
+#endif()
 
-add_executable(conv conv.c)
+#add_executable(conv conv.c)
+add_custom_target("conv")
+add_custom_command(
+	OUTPUT "build/conv"
+	COMMAND cc
+	ARGS -o conv conv.c
+	DEPENDS "build/conv"
+	COMMENT "Generator"
+)
 
 set(HAVE_OPENCV FALSE)
 set(HAVE_OPENCV_3_X FALSE)
@@ -355,7 +363,8 @@
 	target_link_libraries(w2xc ${OPENCV_LIBRARIES} ${CMAKE_DL_LIBS} pthread ${FILE_SYSTEM_LIB})
 endif()
 
-set(CONV_EXE "$<TARGET_FILE_DIR:conv>/conv")
+#set(CONV_EXE "$<TARGET_FILE_DIR:conv>/conv")
+set(CONV_EXE ${CMAKE_BINARY_DIR}/conv)
 
 add_custom_command(
 	OUTPUT modelHandler_OpenCL.cl.h
