diff --git a/resources/com.chatterino.chatterino.desktop b/resources/com.chatterino.chatterino.desktop
index 6ee45354..ece81f5c 100644
--- a/resources/com.chatterino.chatterino.desktop
+++ b/resources/com.chatterino.chatterino.desktop
@@ -4,7 +4,7 @@ Version=1.0
 Name=Chatterino
 Comment=Chat client for Twitch
 Exec=chatterino
-Icon=chatterino
+Icon=com.chatterino.chatterino
 Terminal=false
 Categories=Network;InstantMessaging;
 StartupWMClass=chatterino
