diff --git chatterino.pro chatterino.pro
index d8b14bbe..f7657f06 100644
--- a/chatterino.pro
+++ b/chatterino.pro
@@ -12,10 +12,17 @@
 #  - BOOST_DIRECTORY (C:\local\boost\ by default) (Windows only)

 QT                += widgets core gui network multimedia svg concurrent
-CONFIG            += communi
-COMMUNI           += core model util

 INCLUDEPATH       += src/
+# libcommuni
+INCLUDEPATH       += /usr/include/qt5/Communi/IrcCore
+INCLUDEPATH       += /usr/include/qt5/Communi/IrcModel
+INCLUDEPATH       += /usr/include/qt5/Communi/IrcUtil
+DEFINES           += IRC_SHARED
+LIBS              += -lIrcCore -lIrcModel -lIrcUtil
+# qtkeychain
+INCLUDEPATH       += /usr/include/qt5keychain
+LIBS              += -lqt5keychain
 TARGET             = chatterino
 TEMPLATE           = app
 PRECOMPILED_HEADER = src/PrecompiledHeader.hpp
@@ -78,7 +85,6 @@ CONFIG(debug, debug|release) {
 include(lib/warnings.pri)
 include(lib/fmt.pri)
 include(lib/humanize.pri)
-include(lib/libcommuni.pri)
 include(lib/websocketpp.pri)
 include(lib/wintoast.pri)
 include(lib/signals.pri)
@@ -86,7 +92,6 @@ include(lib/settings.pri)
 include(lib/serialize.pri)
 include(lib/winsdk.pri)
 include(lib/rapidjson.pri)
-include(lib/qtkeychain.pri)

 exists( $$OUT_PWD/conanbuildinfo.pri ) {
     message("Using conan packages")
diff --git src/common/Credentials.cpp src/common/Credentials.cpp
index 87abecb9..5de78130 100644
--- a/src/common/Credentials.cpp
+++ b/src/common/Credentials.cpp
@@ -1,12 +1,12 @@
 #include "Credentials.hpp"
 
 #include "debug/AssertInGuiThread.hpp"
-#include "keychain.h"
 #include "singletons/Paths.hpp"
 #include "singletons/Settings.hpp"
 #include "util/CombinePath.hpp"
 #include "util/Overloaded.hpp"
 
+#include <qt5keychain/keychain.h>
 #include <QSaveFile>
 #include <boost/variant.hpp>
 
