diff --git a/cmake/FindBoostCertify.cmake b/cmake/FindBoostCertify.cmake
index 7dc04c3a..5f538248 100644
--- a/cmake/FindBoostCertify.cmake
+++ b/cmake/FindBoostCertify.cmake
@@ -1,6 +1,6 @@
 include(FindPackageHandleStandardArgs)
 
-find_path(BoostCertify_INCLUDE_DIR boost/certify/https_verification.hpp HINTS ${CMAKE_SOURCE_DIR}/lib/certify/include)
+find_path(BoostCertify_INCLUDE_DIR boost/certify/https_verification.hpp HINTS ${CMAKE_SOURCE_DIR}/lib/certify/include NO_CMAKE_FIND_ROOT_PATH)
 
 find_package_handle_standard_args(BoostCertify DEFAULT_MSG BoostCertify_INCLUDE_DIR)
 
diff --git a/cmake/FindLRUCache.cmake b/cmake/FindLRUCache.cmake
index 82905436..b0b89a4e 100644
--- a/cmake/FindLRUCache.cmake
+++ b/cmake/FindLRUCache.cmake
@@ -1,6 +1,6 @@
 include(FindPackageHandleStandardArgs)
 
-find_path(LRUCache_INCLUDE_DIR lrucache/lrucache.hpp HINTS ${CMAKE_SOURCE_DIR}/lib/lrucache)
+find_path(LRUCache_INCLUDE_DIR lrucache/lrucache.hpp HINTS ${CMAKE_SOURCE_DIR}/lib/lrucache NO_CMAKE_FIND_ROOT_PATH)
 
 find_package_handle_standard_args(LRUCache DEFAULT_MSG LRUCache_INCLUDE_DIR)
 
diff --git a/cmake/FindMagicEnum.cmake b/cmake/FindMagicEnum.cmake
index b595075c..9c4e0f03 100644
--- a/cmake/FindMagicEnum.cmake
+++ b/cmake/FindMagicEnum.cmake
@@ -1,6 +1,6 @@
 include(FindPackageHandleStandardArgs)
 
-find_path(MagicEnum_INCLUDE_DIR magic_enum/magic_enum.hpp HINTS ${CMAKE_SOURCE_DIR}/lib/magic_enum/include)
+find_path(MagicEnum_INCLUDE_DIR magic_enum/magic_enum.hpp HINTS ${CMAKE_SOURCE_DIR}/lib/magic_enum/include NO_CMAKE_FIND_ROOT_PATH)
 
 find_package_handle_standard_args(MagicEnum DEFAULT_MSG MagicEnum_INCLUDE_DIR)
 
diff --git a/cmake/FindPajladaSerialize.cmake b/cmake/FindPajladaSerialize.cmake
index 4671874c..119284c2 100644
--- a/cmake/FindPajladaSerialize.cmake
+++ b/cmake/FindPajladaSerialize.cmake
@@ -1,6 +1,6 @@
 include(FindPackageHandleStandardArgs)
 
-find_path(PajladaSerialize_INCLUDE_DIR pajlada/serialize.hpp HINTS ${CMAKE_SOURCE_DIR}/lib/serialize/include)
+find_path(PajladaSerialize_INCLUDE_DIR pajlada/serialize.hpp HINTS ${CMAKE_SOURCE_DIR}/lib/serialize/include NO_CMAKE_FIND_ROOT_PATH)
 
 find_package_handle_standard_args(PajladaSerialize DEFAULT_MSG PajladaSerialize_INCLUDE_DIR)
 
diff --git a/cmake/FindPajladaSignals.cmake b/cmake/FindPajladaSignals.cmake
index f4c964ec..75efba20 100644
--- a/cmake/FindPajladaSignals.cmake
+++ b/cmake/FindPajladaSignals.cmake
@@ -1,6 +1,6 @@
 include(FindPackageHandleStandardArgs)
 
-find_path(PajladaSignals_INCLUDE_DIR pajlada/signals/signal.hpp HINTS ${CMAKE_SOURCE_DIR}/lib/signals/include)
+find_path(PajladaSignals_INCLUDE_DIR pajlada/signals/signal.hpp HINTS ${CMAKE_SOURCE_DIR}/lib/signals/include NO_CMAKE_FIND_ROOT_PATH)
 
 find_package_handle_standard_args(PajladaSignals DEFAULT_MSG PajladaSignals_INCLUDE_DIR)
 
diff --git a/cmake/FindSol2.cmake b/cmake/FindSol2.cmake
index be64d000..2b18a5ce 100644
--- a/cmake/FindSol2.cmake
+++ b/cmake/FindSol2.cmake
@@ -1,6 +1,6 @@
 include(FindPackageHandleStandardArgs)
 
-find_path(Sol2_INCLUDE_DIR sol/sol.hpp HINTS ${CMAKE_SOURCE_DIR}/lib/sol2/include)
+find_path(Sol2_INCLUDE_DIR sol/sol.hpp HINTS ${CMAKE_SOURCE_DIR}/lib/sol2/include NO_CMAKE_FIND_ROOT_PATH)
 
 find_package_handle_standard_args(Sol2 DEFAULT_MSG Sol2_INCLUDE_DIR)
 
diff --git a/cmake/FindWebsocketpp.cmake b/cmake/FindWebsocketpp.cmake
index f8d0bb65..cd9b8b0e 100644
--- a/cmake/FindWebsocketpp.cmake
+++ b/cmake/FindWebsocketpp.cmake
@@ -1,6 +1,6 @@
 include(FindPackageHandleStandardArgs)
 
-find_path(Websocketpp_INCLUDE_DIR websocketpp/version.hpp HINTS ${CMAKE_SOURCE_DIR}/lib/websocketpp)
+find_path(Websocketpp_INCLUDE_DIR websocketpp/version.hpp HINTS ${CMAKE_SOURCE_DIR}/lib/websocketpp NO_CMAKE_FIND_ROOT_PATH)
 
 find_package_handle_standard_args(Websocketpp DEFAULT_MSG Websocketpp_INCLUDE_DIR)
 
