--- podcasts-gtk/src/widgets/episode.rs.orig	2019-06-09 15:54:59.793451047 +0200
+++ podcasts-gtk/src/widgets/episode.rs	2019-06-09 15:55:19.191311512 +0200
@@ -559,7 +559,7 @@
             .total_size
             .get_text()
             .as_ref()
-            .map(|s| s.trim_right_matches(" MB"))
+            .map(|s| s.trim_end_matches(" MB"))
             .and_then(|s| s.parse::<i32>().ok())
             .is_none()
         {
--- podcasts-gtk/src/widgets/shows_view.rs.orig	2019-06-09 15:54:00.320878867 +0200
+++ podcasts-gtk/src/widgets/shows_view.rs	2019-06-09 15:54:13.151786563 +0200
@@ -80,7 +80,6 @@
 }
 
 fn on_child_activate(child: &gtk::FlowBoxChild, sender: &Sender<Action>) -> Result<(), Error> {
-    use gtk::WidgetExt;
 
     // This is such an ugly hack...
     let id = WidgetExt::get_name(child)
