--- podcasts-gtk/Cargo.toml
+++ podcasts-gtk/Cargo.toml
@@ -28,7 +28,7 @@ reqwest = "0.8.8"
 serde_json = "1.0.26"
 # html2text = "0.1.8"
 html2text = { git = "https://github.com/alatiera/rust-html2text" }
-gettext-rs = { git = "https://github.com/danigm/gettext-rs", branch = "no-gettext", features = ["gettext-system"] }
+gettext-rs = { git = "https://github.com/danigm/gettext-rs", features = ["gettext-system"] }
 
 [dependencies.gtk]
 features = ["v3_22"]
