--- a/Accounts/AccountsQt5Config.cmake.in	2019-09-18 21:00:34.000000000 +0200
+++ b/Accounts/AccountsQt5Config.cmake.in	2019-09-18 21:00:34.000000000 +0200
@@ -1,5 +1,20 @@
 #  ACCOUNTSQT_INCLUDE_DIRS - The libaccounts-qt include directories
 #  ACCOUNTSQT_LIBRARIES - The libraries needed to use libaccounts-qt
+ 
+get_filename_component(_IMPORT_PREFIX "${CMAKE_CURRENT_LIST_FILE}" PATH)
+if(_realCurr STREQUAL _realOrig)
+  set(_IMPORT_PREFIX "/usr/lib/cmake/AccountsQt5")
+endif()
+unset(_realOrig)
+unset(_realCurr)
+get_filename_component(_IMPORT_PREFIX "${_IMPORT_PREFIX}" PATH)
+get_filename_component(_IMPORT_PREFIX "${_IMPORT_PREFIX}" PATH)
+get_filename_component(_IMPORT_PREFIX "${_IMPORT_PREFIX}" PATH)
+if(_IMPORT_PREFIX STREQUAL "/")
+  set(_IMPORT_PREFIX "")
+endif()
+ 
+set(ACCOUNTSQT_LIBRARIES ${_IMPORT_PREFIX}/lib/lib$${TARGET}.so)
+set(ACCOUNTSQT_INCLUDE_DIRS ${_IMPORT_PREFIX}/include/$${TARGET}/)
 
-set(ACCOUNTSQT_LIBRARIES $${INSTALL_LIBDIR}/lib$${TARGET}.so)
-set(ACCOUNTSQT_INCLUDE_DIRS $${INSTALL_PREFIX}/include/$${TARGET}/)
\ No newline at end of file
+set(_IMPORT_PREFIX)
