Returns 403 if user-agent is not set

--- a/autotest/pymod/gdaltest.py
+++ b/autotest/pymod/gdaltest.py
@@ -1895,8 +1895,9 @@ def gdalurlopen(url, timeout=10):
 
     import http.client
 
+    rq = urllib.request.Request(url, headers={'User-Agent': 'Mozilla/5.0'})
     try:
-        handle = urllib.request.urlopen(url)
+        handle = urllib.request.urlopen(rq)
         socket.setdefaulttimeout(old_timeout)
         return handle
     except urllib.error.HTTPError as e:
