--- frmts/jpeg2000/jpeg2000_vsil_io.cpp	2017-04-28 18:07:35.000000000 +0200
+++ frmts/jpeg2000/jpeg2000_vsil_io.cpp	2017-05-13 13:47:12.058749833 +0200
@@ -207,7 +207,7 @@
             /* The buffer must be large enough to accommodate maximum
                putback. */
             assert(bufsize > JAS_STREAM_MAXPUTBACK);
-            stream->bufbase_ = JAS_CAST(uchar *, buf);
+            stream->bufbase_ = JAS_CAST(jas_uchar *, buf);
             stream->bufsize_ = bufsize - JAS_STREAM_MAXPUTBACK;
         }
     } else {
