diff --git frmts/pdf/pdfobject.cpp frmts/pdf/pdfobject.cpp
index aeedcfa..d01dbda 100644
--- frmts/pdf/pdfobject.cpp
+++ frmts/pdf/pdfobject.cpp
@@ -1195,7 +1195,7 @@ GDALPDFObject* GDALPDFDictionaryPoppler::Get(const char* pszKey)
         return oIter->second;
 
 #if POPPLER_MAJOR_VERSION >= 1 || POPPLER_MINOR_VERSION >= 58
-    Object o = m_poDict->lookupNF(((char*)pszKey));
+    const Object &o = m_poDict->lookupNF(((char*)pszKey));
     if (!o.isNull())
     {
         int nRefNum = 0;
@@ -1329,7 +1329,7 @@ GDALPDFObject* GDALPDFArrayPoppler::Get(int nIndex)
         return m_v[nIndex];
 
 #if POPPLER_MAJOR_VERSION >= 1 || POPPLER_MINOR_VERSION >= 58
-    Object o = m_poArray->getNF(nIndex);
+    const Object &o = m_poArray->getNF(nIndex);
     if( !o.isNull() )
     {
         int nRefNum = 0;

