diff --git a/frmts/pdf/pdfio.cpp b/frmts/pdf/pdfio.cpp
index b00560d..f9b4955 100644
--- frmts/pdf/pdfio.cpp
+++ frmts/pdf/pdfio.cpp
@@ -389,7 +389,7 @@ GBool VSIPDFFileStream::hasGetChars()
 /*                            getChars()                                */
 /************************************************************************/
 
-int VSIPDFFileStream::getChars(int nChars, Guchar *buffer)
+int VSIPDFFileStream::getChars(int nChars, unsigned char *buffer)
 {
     int nRead = 0;
     while (nRead < nChars)
diff --git a/frmts/pdf/pdfio.h b/frmts/pdf/pdfio.h
index 795fc2b..f73bb99 100644
--- frmts/pdf/pdfio.h
+++ frmts/pdf/pdfio.h
@@ -30,6 +30,7 @@
 #ifndef PDFIO_H_INCLUDED
 #define PDFIO_H_INCLUDED
 
+#include <goo/gfile.h>
 #include "cpl_vsi_virtual.h"
 
 /************************************************************************/
@@ -46,9 +47,9 @@
 #define moveStart_delta_type Goffset
 #else
 #define getPos_ret_type int
-#define getStart_ret_type Guint
-#define makeSubStream_offset_type Guint
-#define setPos_offset_type Guint
+#define getStart_ret_type unsigned int
+#define makeSubStream_offset_type unsigned int
+#define setPos_offset_type unsigned int
 #define moveStart_delta_type int
 #endif
 
@@ -109,10 +110,10 @@ class VSIPDFFileStream final: public BaseStream
          * but will still compile correctly.
          */
         virtual GBool hasGetChars() override;
-        virtual int getChars(int nChars, Guchar *buffer) override;
+        virtual int getChars(int nChars, unsigned char *buffer) override;
 #else
         virtual GBool hasGetChars() ;
-        virtual int getChars(int nChars, Guchar *buffer) ;
+        virtual int getChars(int nChars, unsigned char *buffer) ;
 #endif
 
         VSIPDFFileStream  *poParent;
diff --git a/frmts/pdf/pdfsdk_headers.h b/frmts/pdf/pdfsdk_headers.h
index 9150b0f..5e0f669 100644
--- frmts/pdf/pdfsdk_headers.h
+++ frmts/pdf/pdfsdk_headers.h
@@ -50,7 +50,6 @@
 #pragma warning( disable : 4244 ) /* conversion from 'const int' to 'Guchar', possible loss of data */
 #endif
 
-#include <goo/gtypes.h>
 #include <goo/GooList.h>
 
 /* begin of poppler xpdf includes */

