--- a/stdlib/CMakeLists.txt
+++ b/stdlib/CMakeLists.txt
@@ -1,15 +1,26 @@
-add_executable(to_c_array EXCLUDE_FROM_ALL to_c_array.cpp)
-configure_target_cc_props(to_c_array)
+set(TO_C_ARRAY_DEP)
+if(CMAKE_CROSSCOMPILING)
+	find_program(TO_C_ARRAY to_c_array)
+	if(NOT TO_C_ARRAY)
+		message(FATAL_ERROR "Unable to find to_c_array")
+	endif()
+	set(TO_C_ARRAY_COMMAND ${TO_C_ARRAY})
+else()
+	add_executable(to_c_array EXCLUDE_FROM_ALL to_c_array.cpp)
+	configure_target_cc_props(to_c_array)
+	list(APPEND TO_C_ARRAY_DEP to_c_array)
+	set(TO_C_ARRAY_COMMAND $<TARGET_FILE:to_c_array>)
+endif()
 
 add_custom_command(
 	COMMENT "Packing std.jsonnet into std.jsonnet.h"
 	COMMAND
-		to_c_array
+		${TO_C_ARRAY_COMMAND}
 			"${CMAKE_CURRENT_SOURCE_DIR}/std.jsonnet"
 			"${CMAKE_CURRENT_BINARY_DIR}/std.jsonnet.h"
 	OUTPUT "${CMAKE_CURRENT_BINARY_DIR}/std.jsonnet.h"
 	MAIN_DEPENDENCY std.jsonnet
-	DEPENDS to_c_array)
+	DEPENDS ${TO_C_ARRAY_DEP})
 
 add_library(stdlib_h INTERFACE "${CMAKE_CURRENT_BINARY_DIR}/std.jsonnet.h")
 target_include_directories(stdlib_h INTERFACE "${CMAKE_CURRENT_BINARY_DIR}")
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -372,6 +372,7 @@
 #
 
 if(BUILD_MAN_PAGES AND (BUILD_JSONNET OR BUILD_JSONNETFMT))
+    set(EXEC_WRAPPER "" CACHE STRING "Executable wrapper for cross builds")
     find_program(HELP2MAN_BINARY NAMES help2man)
     if (HELP2MAN_BINARY)
         message(STATUS "help2man found, man pages will be generated.")
@@ -380,7 +381,7 @@
             add_custom_command(
                 OUTPUT "${PROJECT_BINARY_DIR}/man/man1/jsonnet.1"
                 COMMAND "${HELP2MAN_BINARY}"
-                ARGS --section=1 --no-info --name="Jsonnet data templating language interpreter" --output="${PROJECT_BINARY_DIR}/man/man1/jsonnet.1" "$<TARGET_FILE:jsonnet>"
+                ARGS --section=1 --no-info --name="Jsonnet data templating language interpreter" --output="${PROJECT_BINARY_DIR}/man/man1/jsonnet.1" "${EXEC_WRAPPER} $<TARGET_FILE:jsonnet>"
                 DEPENDS jsonnet
             )
         endif()
@@ -388,7 +389,7 @@
             add_custom_command(
                 OUTPUT "${PROJECT_BINARY_DIR}/man/man1/jsonnetfmt.1"
                 COMMAND "${HELP2MAN_BINARY}"
-                ARGS --section=1 --no-info --name="Jsonnet data templating language auto-formatter" --output="${PROJECT_BINARY_DIR}/man/man1/jsonnetfmt.1" "$<TARGET_FILE:jsonnetfmt>"
+                ARGS --section=1 --no-info --name="Jsonnet data templating language auto-formatter" --output="${PROJECT_BINARY_DIR}/man/man1/jsonnetfmt.1" "${EXEC_WRAPPER} $<TARGET_FILE:jsonnetfmt>"
                 DEPENDS jsonnetfmt
             )
         endif()
