--- a/third_party/rapidyaml/rapidyaml/ext/c4core/src/c4/ext/fast_float/include/fast_float/float_common.h
+++ b/third_party/rapidyaml/rapidyaml/ext/c4core/src/c4/ext/fast_float/include/fast_float/float_common.h
@@ -9,10 +9,12 @@
        || defined(__amd64) || defined(__aarch64__) || defined(_M_ARM64) \
        || defined(__MINGW64__)                                          \
        || defined(__s390x__)                                            \
+       || defined(__powerpc64__)                                        \
        || (defined(__ppc64__) || defined(__PPC64__) || defined(__ppc64le__) || defined(__PPC64LE__)))
 #define FASTFLOAT_64BIT
 #elif (defined(__i386) || defined(__i386__) || defined(_M_IX86)   \
      || defined(__arm__)                                        \
+     || defined(__powerpc__)                                    \
      || defined(__MINGW32__))
 #define FASTFLOAT_32BIT
 #else
@@ -337,4 +339,4 @@ inline OStream& operator<<(OStream &out, const fast_float::decimal &d) {
   return out;
 }
 
-#endif
\ No newline at end of file
+#endif
