--- viewer/frame/extensionpanel.cpp	2019-07-05 11:40:24.000000000 +0200
+++ viewer/frame/extensionpanel.cpp	2020-09-01 18:57:17.236841105 +0200
@@ -19,6 +19,7 @@
 #include "controller/signalmanager.h"
 #include "darrowbutton.h"
 #include <QPainter>
+#include <QPainterPath>
 
 using namespace Dtk::Widget;
 
--- viewer/frame/toptoolbar.cpp	2019-07-05 11:40:24.000000000 +0200
+++ viewer/frame/toptoolbar.cpp	2020-09-01 18:57:11.238840796 +0200
@@ -35,6 +35,7 @@
 #include <QDebug>
 #include <QHBoxLayout>
 #include <QPainter>
+#include <QPainterPath>
 #include <QProcess>
 #include <QResizeEvent>
 #include <QShortcut>
--- viewer/module/album/albumdelegate.cpp	2019-07-05 11:40:24.000000000 +0200
+++ viewer/module/album/albumdelegate.cpp	2020-09-01 18:56:54.229839918 +0200
@@ -22,6 +22,7 @@
 #include <QDateTime>
 #include <QLineEdit>
 #include <QPainter>
+#include <QPainterPath>
 #include <QDebug>
 #include <QHBoxLayout>
 #include <QImageReader>
--- viewer/module/timeline/mvc/timelinedelegate.cpp	2019-07-05 11:40:24.000000000 +0200
+++ viewer/module/timeline/mvc/timelinedelegate.cpp	2020-09-01 18:56:59.205840175 +0200
@@ -24,6 +24,7 @@
 #include <QHBoxLayout>
 #include <QLineEdit>
 #include <QPainter>
+#include <QPainterPath>
 #include <QPixmapCache>
 #include <QStandardItemModel>
 #include <QThread>
--- viewer/module/timeline/mvc/timelineview.cpp	2019-07-05 11:40:24.000000000 +0200
+++ viewer/module/timeline/mvc/timelineview.cpp	2020-09-01 18:57:04.935840471 +0200
@@ -29,6 +29,7 @@
 #include <QHeaderView>
 #include <QPaintEvent>
 #include <QPainter>
+#include <QPainterPath>
 #include <QtConcurrent>
 #include <QtMath>
 
--- viewer/widgets/blureframe.cpp	2019-07-05 11:40:24.000000000 +0200
+++ viewer/widgets/blureframe.cpp	2020-09-01 18:56:35.239838939 +0200
@@ -19,6 +19,7 @@
 #include <QApplication>
 #include <QDebug>
 #include <QPainter>
+#include <QPainterPath>
 #include <QGraphicsScene>
 #include <QGraphicsPixmapItem>
 #include <QGraphicsBlurEffect>
--- viewer/widgets/dialogs/albumdeletedialog.cpp	2019-07-05 11:40:24.000000000 +0200
+++ viewer/widgets/dialogs/albumdeletedialog.cpp	2020-09-01 18:56:24.002838359 +0200
@@ -20,6 +20,7 @@
 #include "utils/baseutils.h"
 #include "utils/imageutils.h"
 #include <QPainter>
+#include <QPainterPath>
 #include <QDebug>
 
 AlbumDeleteDialog::AlbumDeleteDialog(const QStringList &paths, QWidget *parent)
--- viewer/widgets/dialogs/filedeletedialog.cpp	2019-07-05 11:40:24.000000000 +0200
+++ viewer/widgets/dialogs/filedeletedialog.cpp	2020-09-01 18:56:15.056837898 +0200
@@ -20,6 +20,7 @@
 #include "utils/baseutils.h"
 #include "utils/imageutils.h"
 #include <QPainter>
+#include <QPainterPath>
 #include <QDebug>
 
 FileDeleteDialog::FileDeleteDialog(const QStringList &paths, QWidget *parent)
--- viewer/widgets/dialogs/imginfodialog.cpp	2019-07-05 11:40:24.000000000 +0200
+++ viewer/widgets/dialogs/imginfodialog.cpp	2020-09-01 18:56:30.424838690 +0200
@@ -26,6 +26,7 @@
 #include <QKeyEvent>
 #include <QLabel>
 #include <QPainter>
+#include <QPainterPath>
 #include <QVBoxLayout>
 
 namespace {
--- viewer/widgets/dspinner.cpp	2019-07-05 11:40:24.000000000 +0200
+++ viewer/widgets/dspinner.cpp	2020-09-01 18:56:44.085839395 +0200
@@ -2,6 +2,7 @@
 
 #include <QtMath>
 #include <QPainter>
+#include <QPainterPath>
 #include <QTimer>
 
 #include <DObjectPrivate>
--- viewer/widgets/popupmenustyle.cpp	2019-07-05 11:40:24.000000000 +0200
+++ viewer/widgets/popupmenustyle.cpp	2020-09-01 18:56:40.639839217 +0200
@@ -19,6 +19,7 @@
 #include <QPixmap>
 #include <QPixmapCache>
 #include <QPainter>
+#include <QPainterPath>
 #include <QStyleOption>
 #include <QStyleOptionMenuItem>
 
--- viewer/widgets/thumbnaildelegate.cpp	2019-07-05 11:40:24.000000000 +0200
+++ viewer/widgets/thumbnaildelegate.cpp	2020-09-01 18:56:02.445837247 +0200
@@ -23,6 +23,7 @@
 #include <QHBoxLayout>
 #include <QLineEdit>
 #include <QPainter>
+#include <QPainterPath>
 #include <QPixmapCache>
 #include <QStandardItemModel>
 #include <QThread>
--- viewer/widgets/thumbnaillistview.cpp	2019-07-05 11:40:24.000000000 +0200
+++ viewer/widgets/thumbnaillistview.cpp	2020-09-01 18:56:09.028837587 +0200
@@ -30,6 +30,7 @@
 #include <QMutex>
 #include <QPaintEvent>
 #include <QPainter>
+#include <QPainterPath>
 #include <QPen>
 #include <QScrollBar>
 #include <QStandardItemModel>
