Do not install unrelated programs alongside mesonlsp.
--- a/subprojects/muon/meson.build
+++ b/subprojects/muon/meson.build
@@ -173,16 +173,5 @@
     include_directories: include_dir,
 )
 
-muon = executable(
-    'muon',
-    src,
-    dependencies: muon_dep,
-    include_directories: include_dir,
-    link_args: link_args,
-    c_args: c_args,
-    cpp_args: c_args,
-    install: true,
-)
-
 subdir('tests')
 subdir('doc')
