https://github.com/JCWasmx86/mesonlsp/pull/177
--- a/src/libcxathrow/meson.build
+++ b/src/libcxathrow/meson.build
@@ -17,7 +17,11 @@ if cxx.get_id() == 'gcc'
         src_files += 'backtrace_stacktrace.cpp'
         cxathrow_deps += backtrace_dep
     else
-        if is_static or host_machine.system() == 'windows'
+        if not cxx.has_header('execinfo.h')
+            # Musl libc doesn't have execinfo.h
+            src_files += null_backtrace
+            used_libunwind = libunwind_dep.found()
+        elif is_static or host_machine.system() == 'windows'
             # Should only happen in crappy distributions....
             src_files += null_backtrace
             used_libunwind = true
