--- a/meson.build
+++ b/meson.build
@@ -94,8 +94,7 @@
 sha256_proj = subproject('sha256', default_options: ['default_library=static'])
 sha256_dep = sha256_proj.get_variable('sha256_dep')
 
-ada_proj = subproject('ada', default_options: ['default_library=static'])
-ada_dep = ada_proj.get_variable('ada_dep')
+ada_dep = dependency('ada')
 
 muon_proj = subproject('muon')
 muon_dep = muon_proj.get_variable('muon_dep')
