--- a/src/ssl.c	2014-11-21 09:48:57.000000000 -0500
+++ b/src/ssl.c.fixed	2017-09-27 01:46:07.908463177 -0400
@@ -1,7 +1,7 @@
 #include "vpn-ws.h"
 
 #ifndef __WIN32__
-static int _vpn_ws_ssl_wait_read(fd) {
+static int _vpn_ws_ssl_wait_read(int fd) {
         fd_set rset;
         FD_ZERO(&rset);
         FD_SET(fd, &rset);
@@ -12,7 +12,7 @@
         return 0;
 }
 
-static int _vpn_ws_ssl_wait_write(fd) {
+static int _vpn_ws_ssl_wait_write(int fd) {
         fd_set wset;
         FD_ZERO(&wset);
         FD_SET(fd, &wset);
