--- omxplayer.cpp.orig	2013-02-02 09:31:33.347600923 +0000
+++ omxplayer.cpp	2013-02-02 09:32:03.477096232 +0000
@@ -71,7 +71,7 @@
 int               m_use_hw_audio        = false;
 std::string       m_external_subtitles_path;
 bool              m_has_external_subtitles = false;
-std::string       m_font_path           = "/usr/share/fonts/truetype/freefont/FreeSans.ttf";
+std::string       m_font_path           = "/usr/share/fonts/TTF/FreeSans.ttf";
 bool              m_has_font            = false;
 float             m_font_size           = 0.055f;
 bool              m_centered            = false;
