From: Ruben Undheim <ruben.undheim@gmail.com>
Date: Fri, 27 Jul 2018 18:46:13 +0000
Subject: Fix adding of sys.path in yosys-smtbmc

---
 backends/smt2/Makefile.inc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git backends/smt2/Makefile.inc backends/smt2/Makefile.inc
index 92941d4..2c0b7c6 100644
--- a/backends/smt2/Makefile.inc
+++ b/backends/smt2/Makefile.inc
@@ -22,7 +22,7 @@ else
 TARGETS += yosys-smtbmc
 
 yosys-smtbmc: backends/smt2/smtbmc.py
-	$(P) sed 's|##yosys-sys-path##|sys.path += [os.path.dirname(os.path.realpath(__file__)) + p for p in ["/share/python3", "/../share/yosys/python3"]]|;' < $< > $@.new
+	$(P) sed 's|##yosys-sys-path##|sys.path += ["/usr/share/yosys"]|;' < $< > $@.new
 	$(Q) chmod +x $@.new
 	$(Q) mv $@.new $@
 endif
