diff --git src/CMakeLists.txt src/CMakeLists.txt
index 2e453fa5..f2ac7461 100644
--- src/CMakeLists.txt
+++ src/CMakeLists.txt
@@ -45,7 +45,7 @@ else()
 	set(GIT_BRANCH "Unknown")
 endif()
 add_definitions(-DGIT_BRANCH=${GIT_BRANCH})
-add_definitions(-DDEFAULT_PLUGIN_PATH=${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}/edb)
+add_definitions(-DDEFAULT_PLUGIN_PATH=${DEFAULT_PLUGIN_PATH})
 
 include_directories(
 	"capstone-edb"
