diff --git src/DialogInputValue.cpp src/DialogInputValue.cpp
index aa26b4b7..0ca19575 100644
--- src/DialogInputValue.cpp
+++ src/DialogInputValue.cpp
@@ -26,6 +26,14 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 #include "ui_DialogInputValue.h"
 
+#if !defined(__GLIBC__)
+/* These are not defined in musl libc */
+#include <stdint.h>
+#define LONG_LONG_MIN INT64_MIN
+#define LONG_LONG_MAX INT64_MAX
+#define ULONG_LONG_MAX UINT64_MAX
+#endif
+
 //------------------------------------------------------------------------------
 // Name: DialogInputValue
 // Desc:
