diff --git plugins/DebuggerCore/unix/linux/DebuggerCore.cpp plugins/DebuggerCore/unix/linux/DebuggerCore.cpp
index f9351aa5..b298152f 100644
--- plugins/DebuggerCore/unix/linux/DebuggerCore.cpp
+++ plugins/DebuggerCore/unix/linux/DebuggerCore.cpp
@@ -45,6 +45,7 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #include <sys/ptrace.h>
 #include <sys/mman.h>
 #include <sys/personality.h>
+#include <linux/wait.h> /* __WALL */
 
 // doesn't always seem to be defined in the headers
 
diff --git plugins/DebuggerCore/unix/linux/detect/proc-pid-mem-write.cpp plugins/DebuggerCore/unix/linux/detect/proc-pid-mem-write.cpp
index 6c64ff81..67198a59 100644
--- plugins/DebuggerCore/unix/linux/detect/proc-pid-mem-write.cpp
+++ plugins/DebuggerCore/unix/linux/detect/proc-pid-mem-write.cpp
@@ -15,6 +15,7 @@
 #include <sys/wait.h>
 #include <unistd.h>
 #include <fcntl.h>
+#include <linux/wait.h> /* __WALL */
 
 enum Method {
 	Detected,
