diff --git plugins/DebuggerCore/unix/linux/PlatformThread.cpp plugins/DebuggerCore/unix/linux/PlatformThread.cpp
index 2ed6d704..2b963039 100644
--- plugins/DebuggerCore/unix/linux/PlatformThread.cpp
+++ plugins/DebuggerCore/unix/linux/PlatformThread.cpp
@@ -34,6 +34,10 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #include <sys/ptrace.h>
 #include <sys/user.h>
 
+#if !defined(__GLIBC__)
+#define __ptrace_request int
+#endif
+
 // doesn't always seem to be defined in the headers
 #ifndef PTRACE_GET_THREAD_AREA
 #define PTRACE_GET_THREAD_AREA static_cast<__ptrace_request>(25)
