diff --color -ruN 1/include/IState.h 2/include/IState.h
--- 1/include/IState.h	2020-12-14 02:09:07.000000000 +0100
+++ 2/include/IState.h	2022-06-22 05:48:20.394258961 +0200
@@ -19,6 +19,7 @@
 #ifndef ISTATE_H_20110315_
 #define ISTATE_H_20110315_
 
+#include <memory>
 #include "Register.h"
 #include "Types.h"
 
diff --color -ruN 1/include/RegisterViewModelBase.h 2/include/RegisterViewModelBase.h
--- 1/include/RegisterViewModelBase.h	2020-12-14 02:09:07.000000000 +0100
+++ 2/include/RegisterViewModelBase.h	2022-06-22 05:49:05.600877962 +0200
@@ -6,6 +6,7 @@
 #include "util/Integer.h"
 #include <QAbstractItemModel>
 #include <deque>
+#include <memory>
 #include <vector>
 
 Q_DECLARE_METATYPE(std::vector<NumberDisplayMode>)
diff --color -ruN 1/include/State.h 2/include/State.h
--- 1/include/State.h	2020-12-14 02:09:07.000000000 +0100
+++ 2/include/State.h	2022-06-22 05:49:27.911029610 +0200
@@ -19,6 +19,7 @@
 #ifndef STATE_H_20060715_
 #define STATE_H_20060715_
 
+#include <memory>
 #include "API.h"
 #include "Types.h"
 
