--- a/configure.in	2004-11-07 11:01:34.000000000 +0100
+++ b/configure.in	2015-08-30 13:46:25.944394407 +0200
@@ -1,7 +1,8 @@
 dnl process this file with autoconf to generate the configure script
 
-AC_INIT(ogmmerge.cpp)
-AM_INIT_AUTOMAKE(ogmtools,1.5)
+AC_INIT([ogmtools],[1.5])
+AC_CONFIG_SRCDIR([ogmmerge.cpp])
+AM_INIT_AUTOMAKE
 
 AC_PROG_CC
 AC_PROG_CXX
@@ -19,7 +20,7 @@
 dnl libdvdread support
 dnl
 
-have_dvdread=no
+have_dvdread=yes
 AM_PATH_LIBDVDREAD(AC_DEFINE(HAVE_LIBDVDREAD))
 AM_CONDITIONAL(HAVE_LIBDVDREAD, test x"$have_dvdread" = "xyes")
 
@@ -31,7 +32,7 @@
 dnl Check for headers
 AC_HEADER_STDC()
 
-CFLAGS="-Wall -Wno-sign-compare @OGG_CFLAGS@ @VORBIS_CFLAGS@ @AVILIB_CFLAGS@ @DEBUG_CFLAGS@ @PROFILING_CFLAGS@"
-CXXFLAGS="-Wall -Wno-sign-compare @OGG_CFLAGS@ @VORBIS_CFLAGS@ @AVILIB_CFLAGS@ @DEBUG_CFLAGS@ @PROFILING_CFLAGS@"
+AM_CFLAGS="-Wall -Wno-sign-compare @OGG_CFLAGS@ @VORBIS_CFLAGS@ @AVILIB_CFLAGS@ @DEBUG_CFLAGS@ @PROFILING_CFLAGS@"
+AM_CXXFLAGS="-Wall -Wno-sign-compare @OGG_CFLAGS@ @VORBIS_CFLAGS@ @AVILIB_CFLAGS@ @DEBUG_CFLAGS@ @PROFILING_CFLAGS@"
 
 AC_OUTPUT(Makefile avilib/Makefile)
