--- setup.py	2018-05-04 21:16:28.000000000 -0700
+++ setup.py	2018-05-05 06:30:39.111289470 -0700
@@ -559,7 +559,7 @@
     shutil.copytree('kitty', os.path.join(libdir, 'kitty'), ignore=src_ignore)
     shutil.copytree('kittens', os.path.join(libdir, 'kittens'), ignore=src_ignore)
     import compileall
-    compileall.compile_dir(libdir, quiet=1, workers=4)
+    compileall.compile_dir(libdir, quiet=1, workers=1)
     for root, dirs, files in os.walk(libdir):
         for f in files:
             path = os.path.join(root, f)
