Reason: Apparently doesn't work with more build jobs
--- setup.py.orig	2019-01-16 07:04:11.515988522 +0100
+++ setup.py	2019-01-16 07:04:24.905987494 +0100
@@ -594,7 +594,7 @@
         for f in files:
             if f.rpartition('.')[-1] in ('pyc', 'pyo'):
                 os.remove(os.path.join(root, f))
-    kwargs = dict(ddir='', force=True, optimize=1, quiet=1, workers=num_workers)
+    kwargs = dict(ddir='', force=True, optimize=1, quiet=1, workers=1)
     if hasattr(py_compile, 'PycInvalidationMode'):
         kwargs['invalidation_mode'] = py_compile.PycInvalidationMode.UNCHECKED_HASH
     compileall.compile_dir(base_path, **kwargs)