--- setup.py
+++ setup.py
@@ -584,7 +584,7 @@ def compile_python(base_path):
         for f in files:
             if f.rpartition('.')[-1] in ('pyc', 'pyo'):
                 os.remove(os.path.join(root, f))
-    compileall.compile_dir(base_path, ddir='', force=True, optimize=1, quiet=1, workers=num_workers)
+    compileall.compile_dir(base_path, ddir='', force=True, optimize=1, quiet=1, workers=1)
 
 
 def package(args, for_bundle=False, sh_launcher=False):
