From f60beb400eeb1e4778bbfd738bc4a4ccef3de539 Mon Sep 17 00:00:00 2001
From: John Lindgren <john@jlindgren.net>
Date: Sat, 5 Mar 2022 12:05:34 -0500
Subject: [PATCH] Fix build with FFmpeg 5.0

---
 src/ffaudio/ffaudio-core.cc | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/ffaudio/ffaudio-core.cc b/src/ffaudio/ffaudio-core.cc
index 16dc63e3a..a2b95fd6c 100644
--- a/src/ffaudio/ffaudio-core.cc
+++ b/src/ffaudio/ffaudio-core.cc
@@ -274,7 +274,7 @@ static AVInputFormat * get_format_by_content (const char * name, VFSFile & file)
         AVProbeData d = {name, buf, filled};
         score = target;
 
-        f = av_probe_input_format2 (& d, true, & score);
+        f = (AVInputFormat *) av_probe_input_format2 (& d, true, & score);
         if (f)
             break;
 
@@ -347,7 +347,7 @@ static bool find_codec (AVFormatContext * c, CodecInfo * cinfo)
 #endif
         if (stream && stream->codecpar && stream->codecpar->codec_type == AVMEDIA_TYPE_AUDIO)
         {
-            AVCodec * codec = avcodec_find_decoder (stream->codecpar->codec_id);
+            AVCodec * codec = (AVCodec *) avcodec_find_decoder (stream->codecpar->codec_id);
 
             if (codec)
             {
