From 2008b179c5b5cd0442c9954531d1c4dd952ad1ab Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?I=C3=B1aki=20=C3=9Acar?= <iucar@fedoraproject.org>
Date: Fri, 8 Jul 2022 15:27:43 +0200
Subject: [PATCH] unbundle dependencies common

---
 src/cpp/session/CMakeLists.txt | 58 ----------------------------------
 1 file changed, 58 deletions(-)

diff --git a/src/cpp/session/CMakeLists.txt b/src/cpp/session/CMakeLists.txt
index 2e58351..aa9625b 100644
--- a/src/cpp/session/CMakeLists.txt
+++ b/src/cpp/session/CMakeLists.txt
@@ -64,28 +64,6 @@ else()
 endif()
 
 
-# validate our dependencies exist
-foreach(VAR RSTUDIO_DEPENDENCIES_DICTIONARIES_DIR
-            RSTUDIO_DEPENDENCIES_MATHJAX_DIR
-            RSTUDIO_DEPENDENCIES_PANDOC_DIR
-            RSTUDIO_DEPENDENCIES_QUARTO_DIR)
-
- 
-   # skip quarto if not enabled
-   if("${VAR}" STREQUAL "RSTUDIO_DEPENDENCIES_QUARTO_DIR" AND NOT QUARTO_ENABLED)
-      continue()
-   endif()
-
-   # validate existence
-   if(NOT EXISTS "${${VAR}}")
-      message(FATAL_ERROR "${${VAR}} not found (re-run install-dependencies script to install")
-   endif()
-
-   # cache variables so they can be seen by configure_file
-   set("${VAR}" "${${VAR}}" CACHE INTERNAL "")
-   
-endforeach()
-
 # verify embedded packages are available
 foreach(PKG ${RSTUDIO_EMBEDDED_PACKAGES})
    file(GLOB PKG_FILES "${RSTUDIO_DEPENDENCIES_DIR}/common/${PKG}*.tar.gz")
@@ -667,42 +645,6 @@ if(NOT RSTUDIO_SESSION_WIN32 AND NOT RSESSION_ALTERNATE_BUILD)
    install(DIRECTORY "resources/dependencies"
            DESTINATION ${RSTUDIO_INSTALL_SUPPORTING}/resources)
 
-   # install hunspell dictionaries
-   install(DIRECTORY "${RSTUDIO_DEPENDENCIES_DICTIONARIES_DIR}"
-           DESTINATION "${RSTUDIO_INSTALL_SUPPORTING}/resources")
-
-   # install mathjax for local html preview
-   install(DIRECTORY "${RSTUDIO_DEPENDENCIES_MATHJAX_DIR}"
-           DESTINATION "${RSTUDIO_INSTALL_SUPPORTING}/resources")
-
-   # install quarto (or pandoc if quarto disabled)
-   if(QUARTO_ENABLED)
-      # install some quarto folders into Resources, as needed
-      if(APPLE)
-         if (RSTUDIO_ELECTRON)
-            install(DIRECTORY "${RSTUDIO_DEPENDENCIES_QUARTO_DIR}" 
-                  DESTINATION "${RSTUDIO_INSTALL_RESOURCES}/app"
-                  USE_SOURCE_PERMISSIONS)
-         else()
-            install(DIRECTORY "${RSTUDIO_DEPENDENCIES_QUARTO_DIR}"
-                  DESTINATION "${RSTUDIO_INSTALL_BIN}"
-                  USE_SOURCE_PERMISSIONS
-                  PATTERN "*/share" EXCLUDE)
-            install(DIRECTORY "${RSTUDIO_DEPENDENCIES_QUARTO_DIR}/share" 
-                  DESTINATION "${RSTUDIO_INSTALL_SUPPORTING}/quarto"
-                  USE_SOURCE_PERMISSIONS)
-         endif()
-      else()
-         install(DIRECTORY "${RSTUDIO_DEPENDENCIES_QUARTO_DIR}"
-               DESTINATION "${RSTUDIO_INSTALL_BIN}"
-               USE_SOURCE_PERMISSIONS)
-      endif()
-   else()
-      install(DIRECTORY "${RSTUDIO_DEPENDENCIES_PANDOC_DIR}/"
-              DESTINATION "${RSTUDIO_INSTALL_BIN}/pandoc"
-              USE_SOURCE_PERMISSIONS)
-   endif()
-
    # install embedded packages
    foreach(PKG ${RSTUDIO_EMBEDDED_PACKAGES})
       file(GLOB PKG_FILES "${RSTUDIO_DEPENDENCIES_DIR}/common/${PKG}*.tar.gz")
-- 
2.34.1

