From 521eff4cfbbfed2ab51dcbf4cfb3cb1fc2ceb417 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?I=C3=B1aki=20=C3=9Acar?= <iucar@fedoraproject.org>
Date: Wed, 11 Nov 2020 12:43:46 +0100
Subject: [PATCH] fix resources path

---
 src/cpp/desktop/DesktopOptions.cpp | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/src/cpp/desktop/DesktopOptions.cpp b/src/cpp/desktop/DesktopOptions.cpp
index 673870d..d49c141 100644
--- a/src/cpp/desktop/DesktopOptions.cpp
+++ b/src/cpp/desktop/DesktopOptions.cpp
@@ -464,15 +464,9 @@ FilePath Options::resourcesPath() const
 {
    if (resourcesPath_.isEmpty())
    {
-#ifdef RSTUDIO_PACKAGE_BUILD
       // release configuration: the 'resources' folder is
       // part of the supporting files folder
       resourcesPath_ = supportingFilePath().completePath("resources");
-#else
-      // developer configuration: the 'resources' folder is
-      // a sibling of the RStudio executable
-      resourcesPath_ = scriptsPath().completePath("resources");
-#endif
    }
 
    return resourcesPath_;
-- 
2.28.0

