From 4b9b0aa10768df68e80eb8eb03b41b937bc45e9e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?I=C3=B1aki=20=C3=9Acar?= <iucar@fedoraproject.org>
Date: Sat, 5 Jun 2021 18:21:24 +0200
Subject: [PATCH] unbundle dependencies common
 
---
 src/cpp/session/CMakeLists.txt | 22 ----------------------
 1 file changed, 22 deletions(-)
 
diff --git a/src/cpp/session/CMakeLists.txt b/src/cpp/session/CMakeLists.txt
index 54d15d9..a6b41cb 100644
--- a/src/cpp/session/CMakeLists.txt
+++ b/src/cpp/session/CMakeLists.txt
@@ -48,14 +48,11 @@ else()
 
 endif()
 
-foreach(DEP "${RSTUDIO_DEPENDENCIES_DICTIONARIES_DIR}"
+foreach(DEP "${RSTUDIO_DEPENDENCIES_DICTIONARIES_DIR}")
-            "${RSTUDIO_DEPENDENCIES_MATHJAX_DIR}"
-            "${RSTUDIO_DEPENDENCIES_PANDOC_DIR}")
    if(NOT EXISTS "${DEP}")
       message(FATAL_ERROR "${DEP} not found (re-run install-dependencies script to install")
    endif()
 endforeach()
-
 # verify embedded packages are available
 foreach(PKG ${RSTUDIO_EMBEDDED_PACKAGES})
    file(GLOB PKG_FILES "${RSTUDIO_DEPENDENCIES_DIR}/common/${PKG}*.tar.gz")
@@ -574,16 +574,6 @@
    install(DIRECTORY "${RSTUDIO_DEPENDENCIES_DICTIONARIES_DIR}"
            DESTINATION "${RSTUDIO_INSTALL_SUPPORTING}/resources")
 
-   # install mathjax for local html preview
-   install(DIRECTORY "${RSTUDIO_DEPENDENCIES_MATHJAX_DIR}"
-           DESTINATION "${RSTUDIO_INSTALL_SUPPORTING}/resources")
-
-   # install pandoc
-   file(GLOB PANDOC_FILES "${RSTUDIO_DEPENDENCIES_PANDOC_DIR}/pandoc*")
-   install(FILES ${PANDOC_FILES}
-           PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE
-           DESTINATION "${RSTUDIO_INSTALL_BIN}/pandoc")
-
    # install embedded packages
    foreach(PKG ${RSTUDIO_EMBEDDED_PACKAGES})
       file(GLOB PKG_FILES "${RSTUDIO_DEPENDENCIES_DIR}/common/${PKG}*.tar.gz")
-- 
2.31.1
