From c109ef6f7b7f879b24db8574ada3a731a3c3b4a3 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?I=C3=B1aki=20=C3=9Acar?= <iucar@fedoraproject.org>
Date: Fri, 3 Jun 2022 16:10:44 +0200
Subject: [PATCH] do not disable seccomp filter sandbox

---
 src/cpp/desktop/DesktopMain.cpp | 14 --------------
 1 file changed, 14 deletions(-)

diff --git a/src/cpp/desktop/DesktopMain.cpp b/src/cpp/desktop/DesktopMain.cpp
index 7ab82e1..5cf6127 100644
--- a/src/cpp/desktop/DesktopMain.cpp
+++ b/src/cpp/desktop/DesktopMain.cpp
@@ -643,20 +643,6 @@ int main(int argc, char* argv[])
          arguments.push_back(noSandbox);
       }
 
-      static char disableSeccompFilterSandbox[] = "--disable-seccomp-filter-sandbox";
-
-      // newer versions of glibc require us to disable the seccomp filter
-      // sandbox, as the sandbox included with the version of chromium bundled
-      // with Qt 5.12.x does not play well with newer versions of glibc.
-      //
-      // the seccomp filter sandbox is used to prevent user-mode applications
-      // from executing potentially malicious system calls; however, it doesn't
-      // understand some of the newer syscalls introduced in newer versions of
-      // Linux (and used by newer versions of glibc)
-      const char* libcVersion = gnu_get_libc_version();
-      if (core::Version(libcVersion) >= core::Version("2.34"))
-         arguments.push_back(disableSeccompFilterSandbox);
-
 #endif
 
 #ifdef Q_OS_WIN32
-- 
2.34.1

