From b9fe819ef89989c06e69c49437f1c1d59cb942b5 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?I=C3=B1aki=20=C3=9Acar?= <iucar@fedoraproject.org>
Date: Fri, 8 Jul 2022 18:19:55 +0200
Subject: [PATCH] depend on Java source files only

---
 src/gwt/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/gwt/CMakeLists.txt b/src/gwt/CMakeLists.txt
index d9ac4cd..f84619f 100644
--- a/src/gwt/CMakeLists.txt
+++ b/src/gwt/CMakeLists.txt
@@ -71,7 +71,7 @@ if(GWT_BUILD)
    endif()
 
    # depend on Java source files
-   file(GLOB_RECURSE GWT_SOURCE_FILES CONFIGURE_DEPENDS "${CMAKE_CURRENT_SOURCE_DIR}/src/*")
+   file(GLOB_RECURSE GWT_SOURCE_FILES CONFIGURE_DEPENDS "${CMAKE_CURRENT_SOURCE_DIR}/src/*.java")
 
    # generated during GWT build command
    set(GWT_BUILD_TIMESTAMP "${CMAKE_CURRENT_BINARY_DIR}/timestamp")
-- 
2.34.1

