--- a/src/cpp/session/SessionClientInit.cpp
+++ b/src/cpp/session/SessionClientInit.cpp
@@ -443,8 +443,7 @@
 
    sessionInfo["disable_packages"] = module_context::disablePackages();
 
-   sessionInfo["disable_check_for_updates"] =
-          !core::system::getenv("RSTUDIO_DISABLE_CHECK_FOR_UPDATES").empty();
+   sessionInfo["disable_check_for_updates"] = true;
 
    sessionInfo["allow_vcs_exe_edit"] = options.allowVcsExecutableEdit();
    sessionInfo["allow_cran_repos_edit"] = options.allowCRANReposEdit();
