--- src/cpp/core/system/PosixChildProcessTracker.cpp	2020-04-01 18:16:24.000000000 +0200
+++ -	2020-05-01 18:29:14.359052359 +0200
@@ -60,7 +60,7 @@
    std::for_each(processes.begin(),
                  processes.end(),
                  boost::bind(&ChildProcessTracker::attemptToReapProcess,
-                             this, _1));
+                             this, boost::placeholders::_1));
 }
 
 void ChildProcessTracker::attemptToReapProcess(
--- src/cpp/core/system/PosixOutputCapture.cpp	2020-04-01 18:16:24.000000000 +0200
+++ -	2020-05-01 18:30:02.132088380 +0200
@@ -80,10 +80,10 @@
     };
 
    if (dupStdoutFd != -1)
-      outHandler = boost::bind<void>(wrapHandler, stdoutHandler, dupStdoutFd, _1);
+      outHandler = boost::bind<void>(wrapHandler, stdoutHandler, dupStdoutFd, boost::placeholders::_1);
 
    if (dupStderrFd != -1)
-      errHandler = boost::bind<void>(wrapHandler, stderrHandler, dupStderrFd, _1);
+      errHandler = boost::bind<void>(wrapHandler, stderrHandler, dupStderrFd, boost::placeholders::_1);
 
    try
    {
