--- a/src/cpp/core/system/PosixChildProcessTracker.cpp	2020-04-01 18:16:24.000000000 +0200
+++ b/src/cpp/core/system/PosixChildProcessTracker.cpp	2020-04-01 18:16:24.000000000 +0200
@@ -60,7 +60,7 @@
    std::for_each(processes.begin(),
                  processes.end(),
                  boost::bind(&ChildProcessTracker::attemptToReapProcess,
-                             this, _1));
+                             this, boost::placeholders::_1));
 }
 
 void ChildProcessTracker::attemptToReapProcess(
--- a/src/cpp/core/system/PosixOutputCapture.cpp	2021-02-11 00:26:02.000000000 +0100
+++ b/src/cpp/core/system/PosixOutputCapture.cpp	2021-02-11 00:26:02.000000000 +0100
@@ -104,13 +104,13 @@
    if (dupStdoutFd != -1)
    {
       boost::shared_ptr<bool> pSkipStdoutWrite = boost::make_shared<bool>(false);
-      outHandler = boost::bind<void>(wrapHandler, stdoutHandler, dupStdoutFd, "Stdout", pSkipStdoutWrite, _1);
+      outHandler = boost::bind<void>(wrapHandler, stdoutHandler, dupStdoutFd, "Stdout", pSkipStdoutWrite, boost::placeholders::_1);
    }
 
    if (dupStderrFd != -1)
    {
       boost::shared_ptr<bool> pSkipStderrWrite = boost::make_shared<bool>(false);
-      errHandler = boost::bind<void>(wrapHandler, stderrHandler, dupStderrFd, "Stderr", pSkipStderrWrite, _1);
+      errHandler = boost::bind<void>(wrapHandler, stderrHandler, dupStderrFd, "Stderr", pSkipStderrWrite, boost::placeholders::_1);
    }
 
    try
--- a/src/cpp/core/HtmlUtils.cpp	2021-05-30 15:23:37.948407527 +0200
+++ b/src/cpp/core/HtmlUtils.cpp	2021-05-30 15:23:37.948407527 +0200
@@ -56,7 +56,7 @@
    : boost::iostreams::regex_filter(
        boost::regex(
              R"((<\s*[Ii][Mm][Gg] [^\>]*[Ss][Rr][Cc]\s*=\s*)(["'])(.*?)(\2))"),
-        boost::bind(&Base64ImageFilter::toBase64Image, this, _1)),
+        boost::bind(&Base64ImageFilter::toBase64Image, this, boost::placeholders::_1)),
      basePath_(basePath)
 {
 }
@@ -98,7 +98,7 @@
 CssUrlFilter::CssUrlFilter(const FilePath& basePath)
    : boost::iostreams::regex_filter(
         boost::regex("url\\('([^'']+)'\\)"),
-        boost::bind(&CssUrlFilter::toBase64Url, this, _1)),
+        boost::bind(&CssUrlFilter::toBase64Url, this, boost::placeholders::_1)),
      basePath_(basePath)
 {
 }
--- a/src/cpp/r/session/RStdCallbacks.cpp	2021-02-11 00:26:02.000000000 +0100
+++ b/src/cpp/r/session/RStdCallbacks.cpp	2021-02-11 00:26:02.000000000 +0100
@@ -577,7 +577,7 @@
    try
    {
       doHistoryFileOperation(args, boost::bind(&ConsoleHistory::loadFromFile,
-                                               &consoleHistory(), _1, true));
+                                               &consoleHistory(), boost::placeholders::_1, true));
       
       s_callbacks.consoleHistoryReset();
    }
@@ -592,7 +592,7 @@
    try
    {
       doHistoryFileOperation(args, boost::bind(&ConsoleHistory::saveToFile,
-                                               &consoleHistory(), _1));
+                                               &consoleHistory(), boost::placeholders::_1));
    }
    catch(r::exec::RErrorException& e)
    {
@@ -614,7 +614,7 @@
       ConsoleHistory& history = consoleHistory();
       std::for_each(commands.begin(), 
                     commands.end(),
-                    boost::bind(&ConsoleHistory::add, &history, _1));
+                    boost::bind(&ConsoleHistory::add, &history, boost::placeholders::_1));
    }
    catch(r::exec::RErrorException& e)
    {
