--- a/src/cpp/core/HtmlUtils.cpp	2021-02-11 00:26:02.000000000 +0100
+++ b/src/cpp/core/HtmlUtils.cpp	2021-02-11 00:26:02.000000000 +0100
@@ -17,6 +17,7 @@
 
 #include <core/system/System.hpp>
 
+#include <boost/bind/bind.hpp>
 #include <boost/format.hpp>
 #include <boost/algorithm/string/predicate.hpp>
 
--- a/src/cpp/core/file_lock/FileLock.cpp	2021-02-11 00:26:02.000000000 +0100
+++ b/src/cpp/core/file_lock/FileLock.cpp	2021-02-11 00:26:02.000000000 +0100
@@ -26,6 +26,7 @@
 #include <core/system/Environment.hpp>
 #include <core/system/Xdg.hpp>
 
+#include <boost/bind/bind.hpp>
 #include <boost/algorithm/string.hpp>
 
 // borrowed from SessionConstants.hpp
