--- rstudio-1.1.463-orig/src/cpp/server/ServerSecureKeyFile.cpp	2018-10-25 09:42:55.000000000 +1100
+++ src/cpp/server/ServerSecureKeyFile.cpp	2018-11-17 22:17:18.110012955 +1100
@@ -79,8 +79,8 @@
          return error;
 
       // change mode it so it is only readable and writeable by this user
-      if (changeFileMode(secureKeyPath,
-                         core::system::UserReadWriteMode) < 0)
+      if (!!changeFileMode(secureKeyPath,
+                           core::system::UserReadWriteMode))
       {
          return systemError(errno, ERROR_LOCATION);
       }
