--- a/libogmrip/ogmrip-srt.c	2015-01-10 14:39:43.000000000 +0100
+++ b/libogmrip/ogmrip-srt.c	2017-10-16 21:00:19.921665865 +0200
@@ -228,7 +228,7 @@
   g_return_val_if_fail (input != NULL, NULL);
 
   argv = g_ptr_array_new ();
-  g_ptr_array_add (argv, g_strdup ("tesseract"));
+  g_ptr_array_add (argv, g_strdup ("tesseract-ocr"));
   g_ptr_array_add (argv, g_strdup (input));
   g_ptr_array_add (argv, g_strdup (input));
 
@@ -555,7 +555,7 @@
   }
 
 #ifdef HAVE_TESSERACT_SUPPORT
-  fullname = g_find_program_in_path ("tesseract");
+  fullname = g_find_program_in_path ("tesseract-ocr");
   use_tesseract = fullname != NULL;
   g_free (fullname);
 
