diff --git setup.py setup.py
index 1a84791..1a6d7c2 100644
--- a/setup.py
+++ b/setup.py
@@ -14,9 +14,6 @@ with open(os.path.join(here, "README.md")) as f:
 with open(os.path.join(here, "requirements.txt")) as f:
     install_requires = f.read()
 
-with open(os.path.join(here, "requirements-dev.txt")) as f:
-    dev_requires = f.read()
-
 setup(
     name="pylibgen",
     version=version,
@@ -43,12 +40,6 @@ setup(
     packages=find_packages(exclude=("tests",)),
     python_requires=">=3.6",
     install_requires=install_requires,
-    extras_require={
-        # linting and formatting tools (flake8, black) we let pre-commit pin versions
-        # build and publishing tools (setuptools, wheel, twine) we always want latest (see makefile)
-        # only dev dependencies that should be pinned in extras (imo) is testing (pytest) and pre-commit
-        "dev": dev_requires,
-    },
     classifiers=[
         "Development Status :: 7 - Inactive",
         "Intended Audience :: Developers",
