From e5701bb817321555b2da56eec45ecb953f47fd50 Mon Sep 17 00:00:00 2001
From: Xiao-Long Chen <chenxiaolong@cxl.epac.to>
Date: Sun, 15 Mar 2015 17:52:14 -0400
Subject: [PATCH 1/2] autoconf fix

---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index b9c8800..bf751ba 100644
--- a/configure.ac
+++ b/configure.ac
@@ -3,7 +3,7 @@ AC_INIT(indicator-messages, 12.10.5)
 
 AC_PREREQ(2.62)
 
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 AM_INIT_AUTOMAKE([check-news])
 
 AM_MAINTAINER_MODE
-- 
2.3.3

