--- pastebinit.orig	2020-12-18 07:59:28.960857870 +0700
+++ pastebinit	2020-12-18 08:00:47.131001268 +0700
@@ -39,13 +39,7 @@
 # Now try to override it with a distributor pastebin
 try:
     import platform
-    release = platform.linux_distribution()[0].lower()
-    if release == 'debian':
-        defaultPB = "paste.debian.net"
-    elif release == 'fedora':
-        defaultPB = "fpaste.org"
-    elif release == 'ubuntu':
-        defaultPB = "paste.ubuntu.com"
+    release = 'void'
 except ImportError:
     pass
 
