From 86f552b48c835b5f20208830d287656c9d71874c Mon Sep 17 00:00:00 2001
From: classabbyamp <5366828+classabbyamp@users.noreply.github.com>
Date: Thu, 2 Jan 2025 15:04:38 -0500
Subject: [PATCH] Makefile: install udev rules under PREFIX

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 7119820..6f7bb25 100644
--- a/Makefile
+++ b/Makefile
@@ -22,7 +22,7 @@ VERS=$$(python3 ./sendto_silhouette.py --version)
 
 DEST=$(DESTDIR)$(PREFIX)/share/inkscape/extensions
 LOCALE=$(DESTDIR)$(PREFIX)/share/locale
-UDEV=$(DESTDIR)/lib/udev
+UDEV=$(DESTDIR)$(PREFIX)/lib/udev
 INKSCAPE_TEMPLATES=$(DESTDIR)$(PREFIX)/share/inkscape/templates
 
 # User-specifc inkscape extensions folder for local install
