--- src/appsys/AppSystem.vala
+++ src/appsys/AppSystem.vala
@@ -70,7 +70,7 @@ public class AppSystem : GLib.Object
     }
 
     private void signal_received(GLib.DBusConnection connection,
-                                 string sender,
+                                 string? sender,
                                  string object_path,
                                  string interface_name,
                                  string signal_name,
