--- subprojects/gvc/meson.build.orig	2017-08-14 22:00:04.000000000 +0000
+++ subprojects/gvc/meson.build	2017-10-10 18:10:05.177178298 +0000
@@ -15,17 +15,17 @@
 ]
 
 libgvc_sources = [
-    '../../src/gvc/gvc-channel-map.c',
-    '../../src/gvc/gvc-mixer-card.c',
-    '../../src/gvc/gvc-mixer-control.c',
-    '../../src/gvc/gvc-mixer-event-role.c',
-    '../../src/gvc/gvc-mixer-sink.c',
-    '../../src/gvc/gvc-mixer-sink-input.c',
-    '../../src/gvc/gvc-mixer-source.c',
-    '../../src/gvc/gvc-mixer-source-output.c',
-    '../../src/gvc/gvc-mixer-stream.c',
-    '../../src/gvc/gvc-mixer-ui-device.c',
-    '../../src/gvc/test-audio-device-selection.c',
+    'gvc-channel-map.c',
+    'gvc-mixer-card.c',
+    'gvc-mixer-control.c',
+    'gvc-mixer-event-role.c',
+    'gvc-mixer-sink.c',
+    'gvc-mixer-sink-input.c',
+    'gvc-mixer-source.c',
+    'gvc-mixer-source-output.c',
+    'gvc-mixer-stream.c',
+    'gvc-mixer-ui-device.c',
+    'test-audio-device-selection.c',
 ]
 
 libgvc = static_library(
@@ -34,8 +34,7 @@
     dependencies: libgvc_deps,
     include_directories: [
         include_directories('.'),
         include_directories(join_paths('..', '..')),
-        include_directories(join_paths('..', '..', 'src', 'gvc')),
     ],
     c_args: [
         '-DG_LOG_DOMAIN="Gvc"',
@@ -50,7 +49,6 @@
     link_with: libgvc,
     include_directories: [
         include_directories('.'),
         include_directories(join_paths('..', '..')),
-        include_directories(join_paths('..', '..', 'src', 'gvc')),
     ],
 )
