--- Makefile	2017-09-15 14:12:23.848961346 +1000
+++ Makefile	2017-09-15 15:01:32.364872161 +1000
@@ -2,6 +2,7 @@
 
 #  Copyright (c) 1995,1996 Sascha Demetrio
 #  Copyright (c) 2009 - 2011, 2014 - 2016 Peter Pentchev
+#  Copyright (c) 2017 Alastair Knowles
 
 #  It might be helpful to read the `README'-file first.
 
@@ -70,11 +71,10 @@ $(HEXER): $(OBJECTS)
 $(MYC): calc.c config.h
 	$(CC) $(LDFLAGS) $(CPPFLAGS) $(CFLAGS) -DMYCALC=1 -o $@ calc.c -lm
 
-bin2c: bin2c.c
-	$(CC) $(CPPFLAGS) $(CFLAGS) $(LDFLAGS) -o $@ bin2c.c
-
-helptext.c: help.txt bin2c
-	./bin2c -n helptext -o $@ help.txt
+helptext.c: help.txt
+	$(file >$@,char helptext[] = {)
+	$(file >>$@,$(shell cat help.txt | xxd -i))
+	$(file >>$@,, 0};)
 
 .c.o:	config.h
 	$(CC) $(CPPFLAGS) $(CFLAGS) -c -o $@ $<
