diff --git a/git-cliff-core/Cargo.toml b/git-cliff-core/Cargo.toml
index 0b8db3e..08ba673 100644
--- a/git-cliff-core/Cargo.toml
+++ b/git-cliff-core/Cargo.toml
@@ -9,7 +9,7 @@ homepage = "https://github.com/orhun/git-cliff"
 repository = "https://github.com/orhun/git-cliff"
 keywords = ["changelog", "generator", "conventional", "commit"]
 edition = "2021"
-rust-version = "1.83.0"
+rust-version = "1.82.0"
 
 [features]
 default = ["repo"]
diff --git a/git-cliff/Cargo.toml b/git-cliff/Cargo.toml
index 75696a7..b586a4a 100644
--- a/git-cliff/Cargo.toml
+++ b/git-cliff/Cargo.toml
@@ -11,7 +11,7 @@ keywords = ["changelog", "generator", "conventional", "commit"]
 categories = ["command-line-utilities"]
 default-run = "git-cliff"
 edition = "2021"
-rust-version = "1.83.0"
+rust-version = "1.82.0"
 
 [[bin]]
 name = "git-cliff-completions"
