--- a/src/hotspot/os_cpu/linux_arm/os_linux_arm.cpp
+++ b/src/hotspot/os_cpu/linux_arm/os_linux_arm.cpp
@@ -73,7 +73,18 @@
 # include <ucontext.h>
-# include <fpu_control.h>
 # include <asm/ptrace.h>
 
+// Stupid hack as the origin if below doesnt compile with gcc 8.2.0:
+//
+// os_linux_arm.cpp:114:5: error: missing binary operator before token "("
+//  #if NGREG == 16
+//       ^~~~~
+//
+// The NGREG is 18 (bits/signal.h:10), so force it to that value.
+#ifdef NGREG
+#  undef NGREG
+#endif
+#define NGREG 18
+
 #define SPELL_REG_SP  "sp"
 
 // Don't #define SPELL_REG_FP for thumb because it is not safe to use, so this makes sure we never fetch it.
