--- a/brotli/c/dec/decode.c
+++ b/brotli/c/dec/decode.c
@@ -434,10 +434,10 @@ static BROTLI_INLINE void PreloadSymbol(
                                         BrotliBitReader* br,
                                         brotli_reg_t* bits,
                                         brotli_reg_t* value) {
+  BROTLI_HC_MARK_TABLE_FOR_FAST_LOAD(table);
   if (safe) {
     return;
   }
-  BROTLI_HC_MARK_TABLE_FOR_FAST_LOAD(table);
   BROTLI_HC_ADJUST_TABLE_INDEX(table, BrotliGetBits(br, HUFFMAN_TABLE_BITS));
   *bits = BROTLI_HC_FAST_LOAD_BITS(table);
   *value = BROTLI_HC_FAST_LOAD_VALUE(table);
