--- ./base/gsicc_lcms2.c.jlg	2021-09-27 03:44:02.000000000 -0400
+++ ./base/gsicc_lcms2.c	2021-10-05 10:50:03.428000000 -0400
@@ -462,7 +462,7 @@
 gscms_transform_color(gx_device *dev, gsicc_link_t *icclink, void *inputcolor,
                              void *outputcolor, int num_bytes)
 {
-    return gscms_transformm_color_const(dev, icclink, inputcolor, outputcolor, num_bytes);
+    return gscms_transform_color_const(dev, icclink, inputcolor, outputcolor, num_bytes);
 }
 
 int
