--- a/linssid-app/linssid-app.pro
+++ b/linssid-app/linssid-app.pro
@@ -3,9 +3,9 @@
 TARGET = linssid
 VERSION = 3.6
 CONFIG -= debug_and_release app_bundle lib_bundle
-CONFIG += release 
+CONFIG += release qwt
 PKGCONFIG +=
-QT = core gui widgets svg
+QT = core gui widgets
 SOURCES += AboutBox.cpp Getter.cpp MainForm.cpp main.cpp prefsDialog.cpp
 HEADERS += AboutBox.h Custom.h Getter.h MainForm.h prefsDialog.h
 FORMS += AboutBox.ui MainForm.ui prefsDialog.ui
@@ -15,18 +15,16 @@
 MOC_DIR = 
 RCC_DIR = 
 UI_DIR = 
-QMAKE_CC = gcc
-QMAKE_CXX = g++
 DEFINES += 
 INCLUDEPATH += /usr/include/boost
 INCLUDEPATH += /usr/include/qt5
 INCLUDEPATH += /usr/include/qwt
 LIBS += -lboost_regex
-LIBS += /usr/lib/libqwt-qt5.so.6
+LIBS += -lqwt
 QMAKE_CXXFLAGS += -std=c++11
 #
 TARGET = linssid
-target.path = /usr/sbin
+target.path = /usr/bin
 INSTALLS += target
 targetexec.path = /usr/bin
 targetexec.files=linssid-pkexec
