--- ./linssid-app/MainForm.cpp.orig	2023-10-25 14:47:34.411697129 -0400
+++ ./linssid-app/MainForm.cpp	2023-10-25 14:50:20.485975281 -0400
@@ -924,10 +924,10 @@
                 setData(Qt::DisplayRole, MainForm::cellDataRay[row].cenChan);
         MainForm::cellDataRay[row].pTableItem[CENCHAN]->setTextAlignment(Qt::AlignCenter);
         MainForm::cellDataRay[row].pTableItem[FIRST_SEEN]->
-                setText(QDateTime::fromTime_t(MainForm::cellDataRay[row].firstSeen).toString("MM/dd-hh:mm:ss"));
+                setText(QDateTime::fromSecsSinceEpoch(MainForm::cellDataRay[row].firstSeen).toString("MM/dd-hh:mm:ss"));
         MainForm::cellDataRay[row].pTableItem[FIRST_SEEN]->setTextAlignment(Qt::AlignCenter);
         MainForm::cellDataRay[row].pTableItem[LAST_SEEN]->
-                setText(QDateTime::fromTime_t(MainForm::cellDataRay[row].lastSeen).toString("MM/dd-hh:mm:ss"));
+                setText(QDateTime::fromSecsSinceEpoch(MainForm::cellDataRay[row].lastSeen).toString("MM/dd-hh:mm:ss"));
         MainForm::cellDataRay[row].pTableItem[LAST_SEEN]->setTextAlignment(Qt::AlignCenter);
         MainForm::cellDataRay[row].pTableItem[VENDOR]->
                 setText(MainForm::cellDataRay[row].vendor.c_str());
@@ -1127,7 +1127,7 @@
     MainForm::cellDataRay[tbi].pTableItem[PLOT]->setFlags(
             Qt::ItemIsEnabled | Qt::ItemIsUserCheckable | Qt::ItemIsEditable);
     MainForm::cellDataRay[tbi].pTableItem[PLOT]->setCheckState(Qt::Checked);
-    MainForm::cellDataRay[tbi].pTableItem[SSID]->setTextColor(MainForm::cellDataRay[tbi].color);
+    MainForm::cellDataRay[tbi].pTableItem[SSID]->setForeground(MainForm::cellDataRay[tbi].color);
 }
 
 void MainForm::extractData(string tl, int &tbi, int &newBSS) {
