--- a/linssid-app/MainForm.cpp
+++ b/linssid-app/MainForm.cpp
@@ -31,6 +31,7 @@
 #include <qwt_plot_marker.h>
 #include <QPointF>
 #include <qwt_scale_draw.h>
+#include <qwt_text.h>
 #include <sys/stat.h>
 #include <unistd.h>
 #include <sys/types.h>
@@ -1036,7 +1037,7 @@
                     MainForm::cellDataRay[tbi].signal ) );
             }
         } else {
-            MainForm::cellDataRay[tbi].pBandCurve->setSamples(0, 0, 0);
+            MainForm::cellDataRay[tbi].pBandCurve->setSamples(static_cast<double *>(NULL), static_cast<double *>(NULL), 0);
         }
         // now the signal history plot
         int ixStart;
@@ -1055,7 +1056,7 @@
                     &(MainForm::cellDataRay[tbi].pHistory->signal[ixStart]), ixLength);
         } else {
 
-            MainForm::cellDataRay[tbi].pTimeCurve->setSamples(0, 0, 0);
+            MainForm::cellDataRay[tbi].pTimeCurve->setSamples(static_cast<double *>(NULL), static_cast<double *>(NULL), 0);
         }
     }
     MainForm::mainFormWidget.chan24Plot->replot();
