--- a/linssid-app/MainForm.cpp	2018-06-23 18:36:57.000000000 +0200
+++ b/linssid-app/MainForm.cpp	2021-02-13 00:46:29.112580143 +0100
@@ -81,8 +81,8 @@
     int cenChan; // center channel in 40/80/160 MHz bandwidths
     string vendor;
     bool firstPlot;
-    long firstSeen;
-    long lastSeen;
+    time_t firstSeen;
+    time_t lastSeen;
     string netType;
     QColor color;
     QwtPlotCurve* pBandCurve;
@@ -162,11 +162,11 @@
 int MainForm::maxTableIndex; // holds the highest index pointer into cellData
 int MainForm::numVendors;
 int MainForm::maxVendorRecL;
-long MainForm::runStartTime;
-long MainForm::now; // absolute time of the block
+time_t MainForm::runStartTime;
+time_t MainForm::now; // absolute time of the block
 pageBlockType pageBlock; // which section of page is data coming from
 int MainForm::logDataState;
-long MainForm::blockSampleTime; // time of the block relative to runStartTime
+time_t MainForm::blockSampleTime; // time of the block relative to runStartTime
 bool MainForm::firstScan; // do we need to get sudo privileges?
 
 string MainForm::chan24Freq[15] {
--- a/linssid-app/MainForm.h	2018-06-23 18:34:52.000000000 +0200
+++ b/linssid-app/MainForm.h	2021-02-13 00:48:01.969756452 +0100
@@ -86,9 +86,9 @@
     static int maxTableIndex;
     static int numVendors;
     static int maxVendorRecL;
-    static long runStartTime;
-    static long blockSampleTime;
-    static long now;
+    static time_t runStartTime;
+    static time_t blockSampleTime;
+    static time_t now;
     static int logDataState;
     static bool firstScan;
     static string wlPrivacies[3];
