From d6c96d629183e8bab2167ef56457f994017e7c85 Mon Sep 17 00:00:00 2001
From: neeshy <neeshy@tfwno.gf>
Date: Sat, 18 Apr 2020 23:23:51 -0400
Subject: Fix warning related to gettext

---
 deluge/i18n/util.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/deluge/i18n/util.py b/deluge/i18n/util.py
index bd002f7a2..90568fc5b 100644
--- a/deluge/i18n/util.py
+++ b/deluge/i18n/util.py
@@ -114,7 +114,7 @@ def setup_translation():
         # Workaround for Python 2 unicode gettext (keyword removed in Py3).
         kwargs = {} if not deluge.common.PY2 else {'unicode': True}
 
-        gettext.install(I18N_DOMAIN, translations_path, names='ngettext', **kwargs)
+        gettext.install(I18N_DOMAIN, translations_path, names=['ngettext'], **kwargs)
         builtins.__dict__['_n'] = builtins.__dict__['ngettext']
 
         libintl = None
-- 
2.27.0

