--- Makefile.orig	2009-01-08 22:08:17.000000000 +0100
+++ Makefile	2009-09-18 11:14:04.183561770 +0200
@@ -15,7 +15,8 @@ CFLAGS += -D_LARGEFILE64_SOURCE -D_FILE_
 #CFLAGS += -DPBZIP_NO_LOADAVG
 
 # Where you want pbzip2 installed when you do 'make install'
-PREFIX = /usr
+PREFIX = $(DESTDIR)/usr
+MANDIR = $(PREFIX)/share/man
 
 all: pbzip2
 
@@ -29,15 +30,10 @@ pbzip2-static: pbzip2.cpp libbz2.a
 
 # Install the binary pbzip2 program and man page
 install: pbzip2
-	if ( test ! -d $(PREFIX)/bin ) ; then mkdir -p $(PREFIX)/bin ; fi
-	if ( test ! -d $(PREFIX)/man ) ; then mkdir -p $(PREFIX)/man ; fi
-	if ( test ! -d $(PREFIX)/man/man1 ) ; then mkdir -p $(PREFIX)/man/man1 ; fi
-	cp -f pbzip2 $(PREFIX)/bin/pbzip2
-	chmod a+x $(PREFIX)/bin/pbzip2
-	ln -s -f $(PREFIX)/bin/pbzip2 $(PREFIX)/bin/pbunzip2
-	ln -s -f $(PREFIX)/bin/pbzip2 $(PREFIX)/bin/pbzcat
-	cp -f pbzip2.1 $(PREFIX)/man/man1
-	chmod a+r $(PREFIX)/man/man1/pbzip2.1
+	install -d $(PREFIX)/bin
+	install -d $(MANDIR)/man1
+	install -m 755 pbzip2 $(PREFIX)/bin/pbzip2
+	install -m 644 pbzip2.1 $(MANDIR)/man1/pbzip2.1
 
 clean:
 	rm -f *.o pbzip2
