--- lib/mali/Makefile.split.orig	2014-01-29 13:32:24.446817426 +0100
+++ lib/mali/Makefile.split	2014-01-29 13:33:03.822133629 +0100
@@ -1,16 +1,16 @@
 # Makefile for when everything is split up nicely.
 
 install: libMali.so libEGL.so.1.4 libGLESv1_CM.so.1.1 libGLESv2.so.2.0
-        $(INSTALL_DATA) $^ $(libdir)
+	$(INSTALL_DATA) $^ $(libdir)
 
-        $(RM) $(libdir)libEGL.so.1 $(libdir)libEGL.so
-        $(LN) libEGL.so.1.4 $(libdir)libEGL.so.1
-        $(LN) libEGL.so.1 $(libdir)libEGL.so
+	$(RM) $(libdir)libEGL.so.1 $(libdir)libEGL.so
+	$(LN) libEGL.so.1.4 $(libdir)libEGL.so.1
+	$(LN) libEGL.so.1 $(libdir)libEGL.so
 
-        $(RM) $(libdir)libGLESv1_CM.so.1 $(libdir)libGLESv1_CM.so
-        $(LN) libGLESv1_CM.so.1.1 $(libdir)libGLESv1_CM.so.1
-        $(LN) libGLESv1_CM.so.1 $(libdir)libGLESv1_CM.so
+	$(RM) $(libdir)libGLESv1_CM.so.1 $(libdir)libGLESv1_CM.so
+	$(LN) libGLESv1_CM.so.1.1 $(libdir)libGLESv1_CM.so.1
+	$(LN) libGLESv1_CM.so.1 $(libdir)libGLESv1_CM.so
 
-        $(RM) $(libdir)libGLESv2.so.2 $(libdir)libGLESv2.so
-        $(LN) libGLESv2.so.2.0 $(libdir)libGLESv2.so.2
-        $(LN) libGLESv2.so.2 $(libdir)libGLESv2.so
+	$(RM) $(libdir)libGLESv2.so.2 $(libdir)libGLESv2.so
+	$(LN) libGLESv2.so.2.0 $(libdir)libGLESv2.so.2
+	$(LN) libGLESv2.so.2 $(libdir)libGLESv2.so
