--- configure.orig	2016-05-08 15:18:37.105172865 +0900
+++ configure	2016-05-08 15:18:25.159172097 +0900
@@ -138,7 +138,7 @@
   exit 0
 fi
 
-PKGCONFIG=`whereis "pkg-config" | tr -s ' ' '\n' | grep "bin/""pkg-config""$" | head -n 1`
+PKGCONFIG=`which pkg-config`
 if [ -z "$PKGCONFIG" ]; then
   ERR "\n  ERROR: \`"
   WARN "pkg-config"
--- src/Makefile.orig	2016-05-08 15:19:20.502175654 +0900
+++ src/Makefile	2016-05-08 15:20:22.909179665 +0900
@@ -22,7 +22,7 @@
 include ../.default
 -include ../.config
 
-PKGCONFIG := $(shell whereis "pkg-config" | tr -s ' ' '\n' | grep "bin/""pkg-config""$$" | head -n 1)
+PKGCONFIG := $(shell which pkg-config)
 
 VTE ?= ""
 
