diff --git src/profile.c src/profile.c
index 3de9fb5..b7dc71f 100644
--- a/src/profile.c
+++ b/src/profile.c
@@ -2974,12 +2974,8 @@ GString *save_user_settings(GtkWidget *widget, struct Window *win_data)
 	// g_debug("save_user_settings(): profile = %s", profile);
 
 	// g_debug("\n%s", contents->str);
-#  ifdef BSD
 	gchar resolved_patch[PATH_MAX+1];
 	gchar *real_file_name = g_strdup(realpath((const gchar *)profile, resolved_patch));
-#  else
-	gchar *real_file_name = canonicalize_file_name((const gchar *)profile);
-#  endif
 	if (real_file_name==NULL)
 	{
 		//if (access(profile, F_OK))
