--- src/profile.c.orig	2016-05-07 00:22:09.702161948 +0900
+++ src/profile.c	2016-05-07 00:24:21.566170424 +0900
@@ -2660,11 +2660,7 @@
 	// g_debug("save_user_settings(): profile = %s", profile);
 
 	// g_debug("\n%s", contents->str);
-#ifdef BSD
 	gchar *real_file_name = realpath((const gchar *)profile, NULL);
-#else
-	gchar *real_file_name = canonicalize_file_name((const gchar *)profile);
-#endif
 	if (real_file_name==NULL)
 	{
 		//if (access(profile, F_OK))
