diff --git qomui/update.py qomui/update.py
index 9230b58..f247741 100755
--- qomui/update.py
+++ qomui/update.py
@@ -991,9 +991,7 @@ class AddServers(QtCore.QThread):
 
                 if ip != 0:
 
-                    country_check = check_output(["geoiplookup", "{}".format(ip)]).decode("utf-8")
-                    cc = country_check.split(" ")[3].split(",")[0]
-                    country = country_translate(cc)
+                    country = "undefined"
                     self.log.emit(("debug", "importing {}".format(name)))
                     custom_servers[name] = {
                                                 "name": name,
