--- a/qomui/update.py
+++ b/qomui/update.py
@@ -991,9 +991,7 @@
 
                 if ip != 0:
 
-                    country_check = check_output(["geoiplookup", "{}".format(ip)]).decode("utf-8")
-                    cc = country_check.split(" ")[3].split(",")[0]
-                    country = country_translate(cc)
+                    country = "Unknown"
                     self.log.emit(("debug", "importing {}".format(name)))
                     custom_servers[name] = {
                                                 "name": name,
