--- a/src/base/cl_macros.h	2014-07-30 15:19:17.474034947 +0200
+++ b/src/base/cl_macros.h	2014-07-30 15:19:28.210034058 +0200
@@ -66,11 +66,7 @@
 
 // Ignore a value (instead of assigning it to a variable).
 // unused ...
-  #if defined(__GNUC__) || defined(__KCC) // avoid a gcc warning "statement with no effect"
-    #define unused  (void)
-  #else
-    #define unused
-  #endif
+  #define unused
 
 // Denotes a point where control flow can never arrive.
 // NOTREACHED
