This is needed to build on armv6/7 musl.
--- a/mupen64plus-core/src/r4300/new_dynarec/new_dynarec.c
+++ b/mupen64plus-core/src/r4300/new_dynarec/new_dynarec.c
@@ -25,12 +25,8 @@
 #include <stdlib.h>
 #include <string.h>
 
-#if defined(__APPLE__)
 #include <sys/types.h> // needed for u_int, u_char, etc
 
-#define MAP_ANONYMOUS MAP_ANON
-#endif
-
 #ifdef __cplusplus
 extern "C" {
 #endif
--- a/mupen64plus-core/src/r4300/new_dynarec/new_dynarec_64.c
+++ b/mupen64plus-core/src/r4300/new_dynarec/new_dynarec_64.c
@@ -24,10 +24,7 @@
 #include <string.h>
 #include <assert.h>
 
-#if defined(__APPLE__)
 #include <sys/types.h> // needed for u_int, u_char, etc
-#define MAP_ANONYMOUS MAP_ANON
-#endif
 
 #include "new_dynarec.h"
 #include "main/main.h"
