--- a/feature.cpp
+++ b/feature.cpp
@@ -42,7 +42,8 @@ float minimum_distance(const Node *first, const Node *last,
 
   const float a = last->x - first->x;
   const float b = last->y - first->y;
-  const float c = last->y * first->x - last->x * first->y;
+  const float c = static_cast<double>(last->y * first->x)
+    - static_cast<double>(last->x * first->y);
 
   float max = -1.0;
   for (const Node *n = first; n != last; ++n) {
