--- a/ftpfs.c	2019-03-21 19:21:55.591218591 +0100
+++ b/ftpfs.c	2019-03-21 19:22:10.270696908 +0100
@@ -687,7 +687,7 @@ static int test_exists(const char* path)
 	return ftpfs_getattr(path, &sbuf);
 }
 
-static __off_t test_size(const char* path)
+static off_t test_size(const char* path)
 {
 	struct stat sbuf;
 	int err = ftpfs_getattr(path, &sbuf);
@@ -952,7 +952,7 @@ static int ftpfs_truncate(const char* pa
 
   /* fix openoffice problem, truncating exactly to file length */
   
-  __off_t size = (long long int)test_size(path); 
+  off_t size = (long long int)test_size(path); 
   DEBUG(1, "ftpfs_truncate: %s check filesize=%lld\n", path, (long long int)size);
   
   if (offset == size)  
@@ -980,7 +980,7 @@ static int ftpfs_ftruncate(const char *
   }
   /* fix openoffice problem, truncating exactly to file length */
   
-  __off_t size = test_size(path); 
+  off_t size = test_size(path); 
   DEBUG(1, "ftpfs_ftruncate: %s check filesize=%lld\n", path, (long long int)size);
   
   if (offset == size)  
