diff --git Makefile.in Makefile.in
index 7bb5cba..e557452 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -205,8 +205,8 @@ top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 @GUILE_FALSE@AM_CFLAGS = @CFLAGS@ @X_CFLAGS@ @FORK_FLAG@ @GUILE_FLAG@ -I. -Wall
 @GUILE_TRUE@AM_CFLAGS = @CFLAGS@ @X_CFLAGS@ @FORK_FLAG@ @GUILE_FLAG@ @GUILE_CFLAGS@ -I. -Wall
-@GUILE_FALSE@xbindkeys_LDADD = @X_LIBS@
-@GUILE_TRUE@xbindkeys_LDADD = @X_LIBS@ @GUILE_LDFLAGS@
+@GUILE_FALSE@xbindkeys_LDADD = @X_LIBS@ -lguile-2.2 -lgc
+@GUILE_TRUE@xbindkeys_LDADD = @X_LIBS@ -lguile-2.2 -lgc
 xbindkeys_SOURCES = xbindkeys.c xbindkeys.h \
 		keys.h keys.c \
 		options.c options.h \
