commit 42bceea265f53fe7fd4a41037b936deed975fc6c
Author: Alberto Fanjul <albertofanjul@gmail.com>
Date:   Sun Nov 4 03:33:21 2018 +0100

    meson: Bump libgit2-glib dependency

diff --git meson.build meson.build
index 44a6e40f..33b82fa8 100644
--- meson.build
+++ meson.build
@@ -130,7 +130,7 @@ gsettings_desktop_schemas_dep = dependency('gsettings-desktop-schemas')
 gtk_dep = dependency('gtk+-3.0', version: '>= 3.20.0')
 gtkspell_dep = dependency('gtkspell3-3.0')
 gtksourceview_dep = dependency('gtksourceview-3.0', version: '>= 3.10')
-libgit2_glib_dep = dependency('libgit2-glib-1.0', version: ['>= 0.25.0', '< 0.27.0'])
+libgit2_glib_dep = dependency('libgit2-glib-1.0', version: ['>= 0.27.7'])
 libpeas_dep = dependency('libpeas-1.0')
 libsecret_dep = dependency('libsecret-1')
 libsoup_dep = dependency('libsoup-2.4')
