--- a/libgitg/gitg-diff-view-file-selectable.vala	2019-07-25 13:13:01.000000000 +0200
+++ b/libgitg/gitg-diff-view-file-selectable.vala	2020-09-06 14:36:26.374965220 +0200
@@ -54,7 +54,7 @@
 
 		buffer.get_start_iter(out iter);
 
-		while (buffer.forward_iter_to_source_mark(ref iter, d_selection_category))
+		while (buffer.forward_iter_to_source_mark(iter, d_selection_category))
 		{
 			ret += iter.get_line();
 		}
@@ -321,7 +321,7 @@
 
 		var buffer = source_view.buffer as Gtk.SourceBuffer;
 
-		if (!buffer.forward_iter_to_source_mark(ref iter, "header"))
+		if (!buffer.forward_iter_to_source_mark(iter, "header"))
 		{
 			iter.forward_to_end();
 		}
@@ -493,7 +493,7 @@
 		}
 		else
 		{
-			something_selected = (buffer as Gtk.SourceBuffer).forward_iter_to_source_mark(ref iter, d_selection_category);
+			something_selected = (buffer as Gtk.SourceBuffer).forward_iter_to_source_mark(iter, d_selection_category);
 		}
 
 		if (something_selected != has_selection)
