--- a/tests/test-encoding.t
+++ b/tests/test-encoding.t
@@ -33,10 +33,10 @@
   $ . $TESTDIR/latin-1-encoding
   Warning: commit message (did|does) not conform to UTF-8. (re)
   You may want to amend it after fixing the message, or set the config
-  variable i18n.commitencoding to the encoding your project uses.
+  variable i18n.commitEncoding to the encoding your project uses.
   Warning: commit message (did|does) not conform to UTF-8. (re)
   You may want to amend it after fixing the message, or set the config
-  variable i18n.commitencoding to the encoding your project uses.
+  variable i18n.commitEncoding to the encoding your project uses.
 
   $ cd ..
   $ git init -q --bare repo.git
