--- unix/Makefile.orig	2009-03-21 01:54:48.460031655 +0100
+++ unix/Makefile	2009-03-21 01:55:23.230935902 +0100
@@ -102,9 +102,9 @@ INSTALL_PROGRAM = $(INSTALL)
 INSTALL_D = mkdir -p
 # on some systems, manext=l and MANDIR=/usr/man/man$(manext) may be appropriate
 manext = 1
-prefix = /usr/local
-BINDIR = $(prefix)/bin#			where to install executables
-MANDIR = $(prefix)/man/man$(manext)#	where to install man pages
+DESTDIR = /usr/local
+BINDIR = $(DESTDIR)/usr/bin#			where to install executables
+MANDIR = $(DESTDIR)/usr/share/man/man$(manext)#	where to install man pages
 INSTALLEDBIN = $(BINDIR)/funzip$E $(BINDIR)/unzip$E $(BINDIR)/unzipsfx$E \
 	$(BINDIR)/zipgrep$E $(BINDIR)/zipinfo$E
 INSTALLEDMAN = $(MANDIR)/funzip.$(manext) $(MANDIR)/unzip.$(manext) \
