--- Makefile.in.bck	2016-03-27 02:52:28.209509627 -0400
+++ Makefile.in	2016-03-27 02:54:28.288878752 -0400
@@ -17,10 +17,10 @@
 
 srcdir = @srcdir@
 VPATH = @srcdir@
-bindir = @bindir@
-libdir = @libdir@
-datadir = @datarootdir@
-docdir = @docdir@
+bindir = $(DESTDIR)@bindir@
+libdir = $(DESTDIR)@libdir@
+datadir = $(DESTDIR)@datarootdir@
+docdir = $(DESTDIR)@docdir@
 
 # unix
 OBJECTS=abcm2ps.o \
