diff --git a/subprojects/debugbreak.wrap b/subprojects/debugbreak.wrap
index 3ec9e45..a46a315 100644
--- a/subprojects/debugbreak.wrap
+++ b/subprojects/debugbreak.wrap
@@ -1,8 +1,5 @@
-[wrap-git]
+[wrap-file]
 directory = debugbreak
-url = https://github.com/MrAnno/debugbreak.git
-revision = 83bf7e933311b88613cbaadeced9c2e2c811054a
-patch_directory = debugbreak
 
 [provide]
 debugbreak = debugbreak_dep
diff --git a/subprojects/klib.wrap b/subprojects/klib.wrap
index dbaa5de..446f4da 100644
--- a/subprojects/klib.wrap
+++ b/subprojects/klib.wrap
@@ -1,8 +1,5 @@
-[wrap-git]
+[wrap-file]
 directory = klib
-url = https://github.com/attractivechaos/klib.git
-revision = cdb7e9236dc47abf8da7ebd702cc6f7f21f0c502
-patch_directory = klib
 
 [provide]
 klib = klib_dep
diff --git a/subprojects/nanopb-cmake.wrap b/subprojects/nanopb-cmake.wrap
index cdfa9e1..960ab61 100644
--- a/subprojects/nanopb-cmake.wrap
+++ b/subprojects/nanopb-cmake.wrap
@@ -1,4 +1,2 @@
-[wrap-git]
+[wrap-file]
 directory = nanopb
-url = https://github.com/nanopb/nanopb.git
-revision = nanopb-0.4.9.1
diff --git a/subprojects/nanopb-cmake.wrap.orig b/subprojects/nanopb-cmake.wrap.orig
new file mode 100644
index 0000000..d3b5d77
--- /dev/null
+++ b/subprojects/nanopb-cmake.wrap.orig
@@ -0,0 +1,4 @@
+[wrap-git]
+directory = nanopb
+url = https://github.com/nanopb/nanopb
+revision = nanopb-0.4.9.1
diff --git a/subprojects/nanopb-cmake.wrap.rej b/subprojects/nanopb-cmake.wrap.rej
new file mode 100644
index 0000000..8bdf4ee
--- /dev/null
+++ b/subprojects/nanopb-cmake.wrap.rej
@@ -0,0 +1,8 @@
+--- subprojects/nanopb-cmake.wrap
++++ subprojects/nanopb-cmake.wrap
+@@ -1,4 +1,2 @@
+-[wrap-git]
++[wrap-file]
+ directory = nanopb
+-url = https://github.com/nanopb/nanopb
+-revision = nanopb-0.4.7
