# remove mfx
--- a/cmake/dependencies/common.cmake
+++ b/cmake/dependencies/common.cmake
@@ -39,7 +39,6 @@
 elseif(UNIX)
     set(FFMPEG_PLATFORM_LIBRARIES va va-drm va-x11 vdpau X11)
     if(CMAKE_SYSTEM_PROCESSOR STREQUAL "x86_64")
-        list(APPEND FFMPEG_PLATFORM_LIBRARIES mfx)
         set(FFMPEG_PREPARED_BINARIES "${CMAKE_SOURCE_DIR}/third-party/build-deps/ffmpeg/linux-x86_64")
         set(CPACK_DEB_PLATFORM_PACKAGE_DEPENDS "libmfx1,")
         set(CPACK_RPM_PLATFORM_PACKAGE_REQUIRES "intel-mediasdk >= 22.3.0,")
