--- a/cmake/packaging/linux.cmake
+++ b/cmake/packaging/linux.cmake
@@ -11,13 +11,10 @@ if(${SUNSHINE_BUILD_APPIMAGE} OR ${SUNSHINE_BUILD_FLATPAK})
     install(FILES "${CMAKE_CURRENT_BINARY_DIR}/sunshine.service"
             DESTINATION "${SUNSHINE_ASSETS_DIR}/systemd/user")
 else()
-    find_package(Systemd)
     find_package(Udev)
 
     install(FILES "${SUNSHINE_SOURCE_ASSETS_DIR}/linux/misc/60-sunshine.rules"
             DESTINATION "${UDEV_RULES_INSTALL_DIR}")
-    install(FILES "${CMAKE_CURRENT_BINARY_DIR}/sunshine.service"
-            DESTINATION "${SYSTEMD_USER_UNIT_INSTALL_DIR}")
 endif()
 
 # Post install

--- a/cmake/dependencies/common.cmake
+++ b/cmake/dependencies/common.cmake
@@ -39,10 +39,7 @@ elseif(APPLE)
 elseif(UNIX)
     set(FFMPEG_PLATFORM_LIBRARIES va va-drm va-x11 vdpau X11)
     if(CMAKE_SYSTEM_PROCESSOR STREQUAL "x86_64")
-        list(APPEND FFMPEG_PLATFORM_LIBRARIES mfx)
         set(FFMPEG_PREPARED_BINARIES "${CMAKE_SOURCE_DIR}/third-party/build-deps/ffmpeg/linux-x86_64")
-        set(CPACK_DEB_PLATFORM_PACKAGE_DEPENDS "libmfx1,")
-        set(CPACK_RPM_PLATFORM_PACKAGE_REQUIRES "intel-mediasdk >= 22.3.0,")
     elseif(CMAKE_SYSTEM_PROCESSOR STREQUAL "aarch64")
         set(FFMPEG_PREPARED_BINARIES "${CMAKE_SOURCE_DIR}/third-party/build-deps/ffmpeg/linux-aarch64")
     elseif(CMAKE_SYSTEM_PROCESSOR STREQUAL "ppc64le" OR CMAKE_SYSTEM_PROCESSOR STREQUAL "ppc64")

--- a/packaging/linux/sunshine.desktop
+++ b/packaging/linux/sunshine.desktop
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Type=Application
 Name=@PROJECT_NAME@
-Exec=/usr/bin/env systemctl start --u sunshine
+Exec=sunshine
 Version=1.0
 Comment=@PROJECT_DESCRIPTION@
 Icon=sunshine
