--- a/cmake/dependencies/common.cmake	2023-11-26 17:17:31.826060231 +0100
+++ -	2023-11-26 17:17:36.655892216 +0100
@@ -17,6 +17,14 @@
 add_subdirectory(third-party/miniupnp/miniupnpc)
 include_directories(SYSTEM third-party/miniupnp/miniupnpc/include)
 
+find_package(PkgConfig REQUIRED)
+if(SYSTEM_FFMPEG)
+    pkg_check_modules(FFMPEG REQUIRED
+            libavcodec
+            libswscale
+            libavutil
+	)
+else()
 # ffmpeg pre-compiled binaries
 if(WIN32)
     if(NOT CMAKE_SYSTEM_PROCESSOR STREQUAL "AMD64")
@@ -65,10 +73,9 @@
         ${FFMPEG_PREPARED_BINARIES}/lib/libx265.a
         ${HDR10_PLUS_LIBRARY}
         ${FFMPEG_PLATFORM_LIBRARIES})
-
+endif()
 # common dependencies
 find_package(OpenSSL REQUIRED)
-find_package(PkgConfig REQUIRED)
 find_package(Threads REQUIRED)
 pkg_check_modules(CURL REQUIRED libcurl)
 
