--- Makefile
+++ Makefile
@@ -1,8 +1,8 @@
-CFLAGS=-std=c99 -Wall -Wextra -O2 `pkg-config --cflags gio-2.0 gio-unix-2.0 mpv`
-LDFLAGS=`pkg-config --libs gio-2.0 gio-unix-2.0`
+CFLAGS += -std=c99 -Wall -Wextra -O2 $(shell pkg-config --cflags gio-2.0 gio-unix-2.0 mpv)
+LDFLAGS += $(shell pkg-config --libs gio-2.0 gio-unix-2.0)

 mpris.so: mpris.c
-	gcc mpris.c -o mpris.so $(CFLAGS) $(LDFLAGS) -shared -fPIC
+	$(CC) mpris.c -o mpris.so $(CFLAGS) $(LDFLAGS) -shared -fPIC
