From aec15d243b7c19017583227de0c72c79c6e771e4 Mon Sep 17 00:00:00 2001
From: Duncan Overbruck <mail@duncano.de>
Date: Sun, 1 Jun 2025 22:56:36 +0200
Subject: [PATCH] fix memory corruption due to wrong sizeof in
 label_selection_new

---
 src/label.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/label.c b/src/label.c
index 0df5db6..b742d3c 100644
--- a/src/label.c
+++ b/src/label.c
@@ -88,7 +88,7 @@ int label_symbols_find_idx(label_symbols_t *label_symbols, char *s) {
 label_selection_t *
 label_selection_new(label_symbols_t *label_symbols, int num_labels) {
     label_selection_t *l =
-        malloc(sizeof(label_symbols_t) + label_symbols->num_symbols);
+        malloc(sizeof(label_selection_t) + label_symbols->num_symbols);
 
     l->num_labels = num_labels;
 
-- 
2.49.0

