--- a/doc/Makefile.am	2020-10-24 15:35:54.045900486 +1100
+++ b/doc/Makefile.am	2020-10-24 15:36:15.274015032 +1100
@@ -214,26 +214,15 @@
 mup_uguide__DATA = $(uguide_files)
 
 muschar.ps: ../src/include/muschar.h ../tools/doc/mkmuscharps.sh
-	../tools/doc/mkmuscharps.sh < ../src/include/muschar.h
 
 quickref.ps: quickref.t ../src/mup/prolog.ps muschar.ps
-	$(GROFF) -t -Tps quickref.t | $(SED) -e "/BeginProlog/r ../src/mup/prolog.ps" > quickref.ps
 
 uguide.mm: uguide.t ../tools/doc/ugexi muschar.ps ext_1.ps ext_2.ps ext_3.ps ext_4.ps ../src/mup/mup
-	MUPPATH=$$(pwd)/../mup-input/includes MUPQUIET=1 PATH=$$(pwd)/../src/mup:$$PATH ../tools/doc/ugexi < uguide.t > uguide.mm
 
 uguide.ps: uguide.mm ../tools/doc/mvtoc ../src/mup/prolog.ps
-	$(GROFF) -t -Tps -mm -rN2 uguide.mm | $(SED) -e "/BeginProlog/r ../src/mup/prolog.ps" > uguide.ps.tmp
-	../tools/doc/mvtoc uguide.ps.tmp > uguide.ps
-	rm uguide.ps.tmp
 
 ext_1.mup ext_2.mup ext_3.mup ext_4.mup: charlist ../src/mup/fontdata.c ../tools/doc/gen_ext_mup
-	../tools/doc/gen_ext_mup
 
 ext_1.ps ext_2.ps ext_3.ps ext_4.ps: ext_1.mup ext_2.mup ext_3.mup ext_4.mup ../src/mup/mup
-	for p in 1 2 3 4 ; do ../src/mup/mup ext_$$p.mup | $(SED) -e "/Copyright/,/Set up extended character set fonts/d" -e "/BeginFeature/,/EndFeature/d" -e "/%%Trailer/,/%%DocumentFonts/d"  -e "/%%Page/d" > ext_$$p.ps; done
 
 $(uguide_files): uguide.mm ../tools/doc/tr2html ../tools/doc/mkuggifs ../src/mup/prolog.ps muschar.ps
-	mkdir -p uguide
-	(cd uguide ; cp ../param_index . ; ../../tools/doc/tr2html < ../uguide.mm > /dev/null ; rm param_index)
-	(cd uguide; ../../tools/doc/mkuggifs  CAT=$(CAT) GS=$(GS) EXPR=$(EXPR) SORT=$(SORT) PPMTOGIF=$(PPMTOGIF) GREP=$(GREP) EGREP=$(EGREP) )
