From 4ffcd59113fa26d7c2e9d5897b1eef919fd4b457 Mon Sep 17 00:00:00 2001
From: Adam Hupp <adam@hupp.org>
Date: Mon, 9 Jan 2023 12:55:15 -0800
Subject: [PATCH] update test for upstream added gzip extensions

---
 test/python_magic_test.py | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/test/python_magic_test.py b/test/python_magic_test.py
index 624a443..d51587c 100755
--- a/test/python_magic_test.py
+++ b/test/python_magic_test.py
@@ -134,7 +134,7 @@ def test_extension(self):
             self.assert_values(m, {
                 # some versions return '' for the extensions of a gz file,
                 # including w/ the command line.  Who knows...
-                'test.gz': ('gz/tgz/tpz/zabw/svgz', '', '???'),
+                'test.gz': ('gz/tgz/tpz/zabw/svgz/adz/kmy/xcfgz', 'gz/tgz/tpz/zabw/svgz', '', '???'),
                 'name_use.jpg': 'jpeg/jpg/jpe/jfif',
             })
         except NotImplementedError:
@@ -227,6 +227,5 @@ def test_pathlike(self):
         m = magic.Magic(mime=True)
         self.assertEqual('application/pdf', m.from_file(path))
 
-
 if __name__ == '__main__':
     unittest.main()
