From 55355e1e7c82e9bccda9cfb72b65c64c6d8b208c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Trung=20L=C3=AA?= <trung.le@ruby-journal.com>
Date: Thu, 24 Feb 2022 10:42:35 +1100
Subject: [PATCH] Add utility lib to fix missing std::exchange error

---
 src/interpreter.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/interpreter.h b/src/interpreter.h
index 72724ee..aac4916 100644
--- a/src/teakra/src/interpreter.h
+++ b/src/teakra/src/interpreter.h
@@ -1,4 +1,5 @@
 #pragma once
+#include <utility>
 #include <atomic>
 #include <stdexcept>
 #include <tuple>
