--- Makefile	2018-05-09 12:22:50.994990401 +0200
+++ -	2018-05-09 12:45:12.744048126 +0200
@@ -476,30 +476,30 @@
 BIN_PREFIX=$(PREFIX)/bin
 LOCALE_DIR=$(PREFIX)/share/locale
 install: version $(TARGET)
-	mkdir -p $(DATA_PREFIX)
-	mkdir -p $(BIN_PREFIX)
-	install --mode=755 $(TARGET) $(BIN_PREFIX)
-	cp -R --no-preserve=ownership data/font $(DATA_PREFIX)
-	cp -R --no-preserve=ownership data/json $(DATA_PREFIX)
-	cp -R --no-preserve=ownership data/mods $(DATA_PREFIX)
-	cp -R --no-preserve=ownership data/names $(DATA_PREFIX)
-	cp -R --no-preserve=ownership data/raw $(DATA_PREFIX)
-	cp -R --no-preserve=ownership data/recycling $(DATA_PREFIX)
-	cp -R --no-preserve=ownership data/motd $(DATA_PREFIX)
-	cp -R --no-preserve=ownership data/credits $(DATA_PREFIX)
-	cp -R --no-preserve=ownership data/title $(DATA_PREFIX)
+	mkdir -p $(DESTDIR)$(DATA_PREFIX)
+	mkdir -p $(DESTDIR)$(BIN_PREFIX)
+	install --mode=755 $(TARGET) $(DESTDIR)$(BIN_PREFIX)
+	cp -R --no-preserve=ownership data/font $(DESTDIR)$(DATA_PREFIX)
+	cp -R --no-preserve=ownership data/json $(DESTDIR)$(DATA_PREFIX)
+	cp -R --no-preserve=ownership data/mods $(DESTDIR)$(DATA_PREFIX)
+	cp -R --no-preserve=ownership data/names $(DESTDIR)$(DATA_PREFIX)
+	cp -R --no-preserve=ownership data/raw $(DESTDIR)$(DATA_PREFIX)
+	cp -R --no-preserve=ownership data/recycling $(DESTDIR)$(DATA_PREFIX)
+	cp -R --no-preserve=ownership data/motd $(DESTDIR)$(DATA_PREFIX)
+	cp -R --no-preserve=ownership data/credits $(DESTDIR)$(DATA_PREFIX)
+	cp -R --no-preserve=ownership data/title $(DESTDIR)$(DATA_PREFIX)
 ifdef TILES
-	cp -R --no-preserve=ownership gfx $(DATA_PREFIX)
+	cp -R --no-preserve=ownership gfx $(DESTDIR)$(DATA_PREFIX)
 endif
 ifdef LUA
-	mkdir -p $(DATA_PREFIX)/lua
-	install --mode=644 lua/autoexec.lua $(DATA_PREFIX)/lua
-	install --mode=644 lua/class_definitions.lua $(DATA_PREFIX)/lua
+	mkdir -p $(DESTDIR)$(DATA_PREFIX)/lua
+	install --mode=644 lua/autoexec.lua $(DESTDIR)$(DATA_PREFIX)/lua
+	install --mode=644 lua/class_definitions.lua $(DESTDIR)$(DATA_PREFIX)/lua
 endif
 	install --mode=644 data/changelog.txt data/cataicon.ico data/fontdata.json \
-                   README.txt LICENSE.txt -t $(DATA_PREFIX)
-	mkdir -p $(LOCALE_DIR)
-	LOCALE_DIR=$(LOCALE_DIR) lang/compile_mo.sh
+                   README.txt LICENSE.txt -t $(DESTDIR)$(DATA_PREFIX)
+	mkdir -p $(DESTDIR)$(LOCALE_DIR)
+	LOCALE_DIR=$(DESTDIR)$(LOCALE_DIR) lang/compile_mo.sh
 endif
 
 ifeq ($(TARGETSYSTEM), CYGWIN)
