Index: Cataclysm-DDA-0.E/CMakeLists.txt
===================================================================
--- Cataclysm-DDA-0.E.orig/CMakeLists.txt
+++ Cataclysm-DDA-0.E/CMakeLists.txt
@@ -222,7 +222,7 @@ IF(MSVC)
     endif()
 ELSE()
     SET(CATA_WARNINGS
-      "-Werror -Wall -Wextra \
+      "-Wall -Wextra \
        -Wformat-signedness \
        -Wlogical-op \
        -Wmissing-declarations \
