--- a/mesonbuild/modules/gnome.py	2020-06-29 19:00:15.000000000 +0200
+++ b/mesonbuild/modules/gnome.py	2020-06-29 19:00:15.000000000 +0200
@@ -403,7 +403,7 @@
 
     def _get_gir_dep(self, state):
         if not self.gir_dep:
-            kwargs = {'native': True, 'required': True}
+            kwargs = {'native': False, 'required': True}
             holder = self.interpreter.func_dependency(state.current_node, ['gobject-introspection-1.0'], kwargs)
             self.gir_dep = holder.held_object
             giscanner = state.environment.lookup_binary_entry(MachineChoice.HOST, 'g-ir-scanner')
