--- mesonbuild/modules/gnome.py	2021-06-07 17:35:31.000000000 +0000
+++ -	2021-06-17 07:12:07.442352693 +0000
@@ -87,7 +87,7 @@
                      once=True)
 
     def _get_native_dep(self, state, depname, required=True):
-        kwargs = {'native': True, 'required': required}
+        kwargs = {'native': False, 'required': required}
         holder = self.interpreter.func_dependency(state.current_node, [depname], kwargs)
         return holder.held_object
 
