--- lmon16f.c.orig	2016-05-14 18:38:40.459977271 +0200
+++ lmon16f.c	2016-05-14 18:38:58.012263072 +0200
@@ -706,7 +706,6 @@
 int isroot = 0;
 
 #include <mntent.h>
-#include <fstab.h>
 #include <sys/stat.h>
 #include <sys/statfs.h>
 #include <net/if.h>
@@ -3456,7 +3455,7 @@
 		mp->mnt_dir[JFSNAMELEN - 1] = 0;
 		mp->mnt_type[JFSTYPELEN - 1] = 0;
 	    }
-	    endfsent();
+	    endmntent(mfp);
 	    jfs_loaded = 1;
 	    jfses = i;
 	}
