--- lmon16i.c	2019-01-03 14:05:52.400371456 +0200
+++ lmon16i.c	2019-01-03 14:07:04.872711631 +0200
@@ -63,7 +63,7 @@ static char *SccsId = "nmon " VERSION;
 #include <fcntl.h>
 #include <math.h>
 #include <time.h>
-#include <sys/errno.h>
+#include <errno.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <sys/ioctl.h>
@@ -72,6 +72,10 @@ static char *SccsId = "nmon " VERSION;
 #include <sys/socket.h>
 #include <sys/wait.h>
 
+#ifndef __STRING
+#define __STRING(x) #x
+#endif
+
 /* Windows moved here so they can be cleared when the screen mode changes */
 WINDOW *padwelcome = NULL;
 WINDOW *padtop = NULL;
@@ -584,7 +588,6 @@ struct procsinfo {
 int isroot = 0;
 
 #include <mntent.h>
-#include <fstab.h>
 #include <sys/stat.h>
 #include <sys/statfs.h>
 #include <net/if.h>
