--- lmon16g.c.orig
+++ lmon16g.c
@@ -63,7 +63,7 @@
 #include <fcntl.h>
 #include <math.h>
 #include <time.h>
-#include <sys/errno.h>
+#include <errno.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <sys/ioctl.h>
@@ -72,6 +72,10 @@
 #include <sys/socket.h>
 #include <sys/wait.h>
 
+#ifndef __STRING
+#define __STRING(x) #x
+#endif
+
 /* Windows moved here so they can be cleared when the screen mode changes */
 WINDOW *padwelcome = NULL;
 WINDOW *padtop = NULL;
@@ -576,7 +580,6 @@
 int isroot = 0;
 
 #include <mntent.h>
-#include <fstab.h>
 #include <sys/stat.h>
 #include <sys/statfs.h>
 #include <net/if.h>
